# Comparing `tmp/wgd-2.0.37.tar.gz` & `tmp/wgd-2.0.38.tar.gz`

## filetype from file(1)

```diff
@@ -1 +1 @@
-gzip compressed data, was "dist/wgd-2.0.37.tar", last modified: Wed May  8 16:14:26 2024, max compression
+gzip compressed data, was "dist/wgd-2.0.38.tar", last modified: Sat May 11 16:47:15 2024, max compression
```

## Comparing `wgd-2.0.37.tar` & `wgd-2.0.38.tar`

### file list

```diff
@@ -1,31 +1,31 @@
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-08 16:14:26.436413 wgd-2.0.37/
--rwxrwxrwx   0 heche     (1000) heche     (1000)    35071 2022-09-25 05:52:37.000000 wgd-2.0.37/LICENSE
--rwxrwxrwx   0 heche     (1000) heche     (1000)    97683 2024-05-08 16:14:26.437504 wgd-2.0.37/PKG-INFO
--rwxrwxrwx   0 heche     (1000) heche     (1000)    97445 2024-05-01 07:38:13.000000 wgd-2.0.37/README.md
--rwxrwxrwx   0 heche     (1000) heche     (1000)    62375 2024-05-01 10:49:12.000000 wgd-2.0.37/cli.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)      103 2024-05-08 16:14:26.438508 wgd-2.0.37/setup.cfg
--rwxrwxrwx   0 heche     (1000) heche     (1000)     1964 2024-05-08 16:13:39.000000 wgd-2.0.37/setup.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-08 16:14:26.414384 wgd-2.0.37/test/
--rwxrwxrwx   0 heche     (1000) heche     (1000)     4521 2023-06-06 22:05:06.000000 wgd-2.0.37/test/test_core.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-08 16:14:26.428193 wgd-2.0.37/wgd/
--rwxrwxrwx   0 heche     (1000) heche     (1000)      711 2022-09-25 05:52:37.000000 wgd-2.0.37/wgd/__init__.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    31126 2022-12-22 23:06:45.000000 wgd-2.0.37/wgd/beast.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)     1685 2022-09-30 07:30:21.000000 wgd-2.0.37/wgd/cluster.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    12579 2023-11-20 15:14:14.000000 wgd-2.0.37/wgd/codeml.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   157655 2024-04-22 13:37:59.000000 wgd-2.0.37/wgd/core.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    17245 2024-04-22 09:47:26.000000 wgd-2.0.37/wgd/mcmctree.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    14140 2023-03-01 13:14:50.000000 wgd-2.0.37/wgd/mix.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   121315 2024-05-03 06:53:28.000000 wgd-2.0.37/wgd/peak.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)     4722 2023-06-06 22:05:06.000000 wgd-2.0.37/wgd/postplot.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    54236 2024-04-16 08:22:14.000000 wgd-2.0.37/wgd/ratecorrect.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    18305 2024-03-04 04:24:50.000000 wgd-2.0.37/wgd/syn.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    21638 2023-10-09 15:32:16.000000 wgd-2.0.37/wgd/tree.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)    31523 2023-06-23 20:25:14.000000 wgd-2.0.37/wgd/utils.py
--rwxrwxrwx   0 heche     (1000) heche     (1000)   176151 2024-05-01 10:43:46.000000 wgd-2.0.37/wgd/viz.py
-drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-08 16:14:26.435414 wgd-2.0.37/wgd.egg-info/
--rwxrwxrwx   0 heche     (1000) heche     (1000)    97683 2024-05-08 16:14:26.000000 wgd-2.0.37/wgd.egg-info/PKG-INFO
--rwxrwxrwx   0 heche     (1000) heche     (1000)      416 2024-05-08 16:14:26.000000 wgd-2.0.37/wgd.egg-info/SOURCES.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)        1 2024-05-08 16:14:26.000000 wgd-2.0.37/wgd.egg-info/dependency_links.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)       32 2024-05-08 16:14:26.000000 wgd-2.0.37/wgd.egg-info/entry_points.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)      779 2024-05-08 16:14:26.000000 wgd-2.0.37/wgd.egg-info/requires.txt
--rwxrwxrwx   0 heche     (1000) heche     (1000)        8 2024-05-08 16:14:26.000000 wgd-2.0.37/wgd.egg-info/top_level.txt
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-11 16:47:15.432341 wgd-2.0.38/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    35071 2022-09-25 05:52:37.000000 wgd-2.0.38/LICENSE
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   111207 2024-05-11 16:47:15.433340 wgd-2.0.38/PKG-INFO
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   110969 2024-05-10 08:28:30.000000 wgd-2.0.38/README.md
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    62375 2024-05-01 10:49:12.000000 wgd-2.0.38/cli.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      103 2024-05-11 16:47:15.435338 wgd-2.0.38/setup.cfg
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     1964 2024-05-11 16:39:39.000000 wgd-2.0.38/setup.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-11 16:47:15.256131 wgd-2.0.38/test/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     4521 2023-06-06 22:05:06.000000 wgd-2.0.38/test/test_core.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-11 16:47:15.419340 wgd-2.0.38/wgd/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      711 2022-09-25 05:52:37.000000 wgd-2.0.38/wgd/__init__.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    31126 2022-12-22 23:06:45.000000 wgd-2.0.38/wgd/beast.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     1685 2022-09-30 07:30:21.000000 wgd-2.0.38/wgd/cluster.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    12579 2023-11-20 15:14:14.000000 wgd-2.0.38/wgd/codeml.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   157655 2024-04-22 13:37:59.000000 wgd-2.0.38/wgd/core.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    17245 2024-04-22 09:47:26.000000 wgd-2.0.38/wgd/mcmctree.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    14140 2023-03-01 13:14:50.000000 wgd-2.0.38/wgd/mix.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   121397 2024-05-11 16:39:19.000000 wgd-2.0.38/wgd/peak.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)     4722 2023-06-06 22:05:06.000000 wgd-2.0.38/wgd/postplot.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    54236 2024-04-16 08:22:14.000000 wgd-2.0.38/wgd/ratecorrect.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    18305 2024-03-04 04:24:50.000000 wgd-2.0.38/wgd/syn.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    21638 2023-10-09 15:32:16.000000 wgd-2.0.38/wgd/tree.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)    31523 2023-06-23 20:25:14.000000 wgd-2.0.38/wgd/utils.py
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   176151 2024-05-01 10:43:46.000000 wgd-2.0.38/wgd/viz.py
+drwxrwxrwx   0 heche     (1000) heche     (1000)        0 2024-05-11 16:47:15.431346 wgd-2.0.38/wgd.egg-info/
+-rwxrwxrwx   0 heche     (1000) heche     (1000)   111207 2024-05-11 16:47:14.000000 wgd-2.0.38/wgd.egg-info/PKG-INFO
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      416 2024-05-11 16:47:15.000000 wgd-2.0.38/wgd.egg-info/SOURCES.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)        1 2024-05-11 16:47:14.000000 wgd-2.0.38/wgd.egg-info/dependency_links.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)       32 2024-05-11 16:47:14.000000 wgd-2.0.38/wgd.egg-info/entry_points.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)      779 2024-05-11 16:47:14.000000 wgd-2.0.38/wgd.egg-info/requires.txt
+-rwxrwxrwx   0 heche     (1000) heche     (1000)        8 2024-05-11 16:47:14.000000 wgd-2.0.38/wgd.egg-info/top_level.txt
```

### Comparing `wgd-2.0.37/LICENSE` & `wgd-2.0.38/LICENSE`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/PKG-INFO` & `wgd-2.0.38/PKG-INFO`

 * *Files 10% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
 00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
-00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 370a  Version: 2.0.37.
+00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 380a  Version: 2.0.38.
 00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
 00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
 00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
 00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
 00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
 00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
 00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
@@ -72,6035 +72,6880 @@
 00000470: 2a2c 2056 4942 2d55 4765 6e74 2043 656e  *, VIB-UGent Cen
 00000480: 7465 7220 666f 7220 506c 616e 7420 5379  ter for Plant Sy
 00000490: 7374 656d 7320 4269 6f6c 6f67 792a 2a0a  stems Biology**.
 000004a0: 0a5b 2a2a 496e 7472 6f64 7563 7469 6f6e  .[**Introduction
 000004b0: 2a2a 5d28 2369 6e74 726f 6475 6374 696f  **](#introductio
 000004c0: 6e29 207c 200a 5b2a 2a49 6e73 7461 6c6c  n) | .[**Install
 000004d0: 6174 696f 6e2a 2a5d 2823 696e 7374 616c  ation**](#instal
-000004e0: 6c61 7469 6f6e 2920 7c20 0a5b 2a2a 5061  lation) | .[**Pa
-000004f0: 7261 6d65 7465 7273 2a2a 5d28 2370 6172  rameters**](#par
-00000500: 616d 6574 6572 7329 207c 200a 5b2a 2a55  ameters) | .[**U
-00000510: 7361 6765 2a2a 5d28 2375 7361 6765 2920  sage**](#usage) 
-00000520: 7c20 0a5b 2a2a 496c 6c75 7374 7261 7469  | .[**Illustrati
-00000530: 6f6e 2a2a 5d28 2369 6c6c 7573 7472 6174  on**](#illustrat
-00000540: 696f 6e29 207c 0a5b 2a2a 446f 6375 6d65  ion) |.[**Docume
-00000550: 6e74 6174 696f 6e2a 2a5d 2868 7474 7073  ntation**](https
-00000560: 3a2f 2f77 6764 7632 2e72 6561 6474 6865  ://wgdv2.readthe
-00000570: 646f 6373 2e69 6f2f 656e 2f6c 6174 6573  docs.io/en/lates
-00000580: 742f 3f62 6164 6765 3d6c 6174 6573 7429  t/?badge=latest)
-00000590: 207c 0a5b 2a2a 4369 7461 7469 6f6e 2a2a   |.[**Citation**
-000005a0: 5d28 2363 6974 6174 696f 6e29 0a0a 3c2f  ](#citation)..</
-000005b0: 6469 763e 0a0a 6077 6764 2076 3260 2069  div>..`wgd v2` i
-000005c0: 7320 6120 7079 7468 6f6e 2070 6163 6b61  s a python packa
-000005d0: 6765 2075 7067 7261 6465 6420 6672 6f6d  ge upgraded from
-000005e0: 2074 6865 206f 7269 6769 6e61 6c20 6077   the original `w
-000005f0: 6764 6020 7061 636b 6167 6520 6169 6d69  gd` package aimi
-00000600: 6e67 2066 6f72 2074 6865 2069 6e66 6572  ng for the infer
-00000610: 656e 6365 2061 6e64 2074 696d 696e 6720  ence and timing 
-00000620: 6f66 2061 6e63 6965 6e74 2077 686f 6c65  of ancient whole
-00000630: 2d67 656e 6f6d 6520 6475 706c 6963 6174  -genome duplicat
-00000640: 696f 6e20 2857 4744 2920 6576 656e 7473  ion (WGD) events
-00000650: 2e20 466f 7220 7468 6520 7072 6f70 6f73  . For the propos
-00000660: 6520 6f66 2069 6c6c 7573 7472 6174 696e  e of illustratin
-00000670: 6720 7468 6520 7072 696e 6369 706c 6520  g the principle 
-00000680: 616e 6420 7573 6167 6520 6f66 2060 7767  and usage of `wg
-00000690: 6420 7632 602c 2077 6520 636f 6d70 696c  d v2`, we compil
-000006a0: 6564 2074 6869 7320 646f 6375 6d65 6e74  ed this document
-000006b0: 6174 696f 6e2e 2042 656c 6f77 2077 6520  ation. Below we 
-000006c0: 6669 7273 7420 6761 7665 2061 6e20 696e  first gave an in
-000006d0: 7472 6f64 7563 7469 6f6e 206f 7665 7220  troduction over 
-000006e0: 7468 6520 7363 6f70 6520 616e 6420 6d65  the scope and me
-000006f0: 6368 616e 6973 6d20 6f66 2060 7767 6420  chanism of `wgd 
-00000700: 7632 6020 616e 6420 7468 656e 2074 6865  v2` and then the
-00000710: 2070 7261 6374 6963 616c 2069 6e66 6f72   practical infor
-00000720: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-00000730: 6720 746f 2074 6865 2069 6e73 7461 6c6c  g to the install
-00000740: 6174 696f 6e20 616e 6420 7573 6167 652e  ation and usage.
-00000750: 2041 6e20 6578 616d 706c 6172 2077 6f72   An examplar wor
-00000760: 6b66 6c6f 7720 6973 2070 726f 7669 6465  kflow is provide
-00000770: 6420 696e 2074 6865 2074 7574 6f72 6961  d in the tutoria
-00000780: 6c20 7365 6374 696f 6e20 6f6e 2068 6f77  l section on how
-00000790: 2074 6f20 7365 656b 2065 7669 6465 6e63   to seek evidenc
-000007a0: 6520 666f 7220 6120 7075 7461 7469 7665  e for a putative
-000007b0: 2057 4744 2065 7665 6e74 2061 6e64 2070   WGD event and p
-000007c0: 6572 666f 726d 2070 726f 7065 7220 7469  erform proper ti
-000007d0: 6d69 6e67 2077 6974 6820 6120 6672 6573  ming with a fres
-000007e0: 686c 7920 6f62 7461 696e 6564 2067 656e  hly obtained gen
-000007f0: 6f6d 6520 6173 7365 6d62 6c79 2069 6e20  ome assembly in 
-00000800: 6861 6e64 2e20 466f 7220 7468 6f73 6520  hand. For those 
-00000810: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
-00000820: 6564 2069 6e20 6d6f 7265 2074 6865 6f72  ed in more theor
-00000830: 6574 6963 616c 2064 6574 6169 6c73 2c20  etical details, 
-00000840: 7765 2072 6563 6f6d 6d65 6e64 2074 7572  we recommend tur
-00000850: 6e69 6e67 2074 6f20 6f75 7220 7061 7065  ning to our pape
-00000860: 7220 616e 6420 626f 6f6b 2063 6861 7074  r and book chapt
-00000870: 6572 2066 6f72 206d 6f72 6520 6465 7461  er for more deta
-00000880: 696c 6564 2064 6573 6372 6970 7469 6f6e  iled description
-00000890: 2061 6e64 2069 6e73 6967 6874 6675 6c20   and insightful 
-000008a0: 6469 7363 7573 7369 6f6e 732e 2054 6865  discussions. The
-000008b0: 206b 6579 2069 6d70 726f 7665 6420 6665   key improved fe
-000008c0: 6174 7572 6573 206f 6620 6077 6764 2076  atures of `wgd v
-000008d0: 3260 2061 7265 2064 656d 6f6e 7374 7261  2` are demonstra
-000008e0: 7465 6420 696e 206f 7572 206c 6174 6573  ted in our lates
-000008f0: 7420 6d61 6e75 7363 7269 7074 206f 6e20  t manuscript on 
-00000900: 5b42 696f 696e 666f 726d 6174 6963 735d  [Bioinformatics]
-00000910: 2868 7474 7073 3a2f 2f64 6f69 2e6f 7267  (https://doi.org
-00000920: 2f31 302e 3130 3933 2f62 696f 696e 666f  /10.1093/bioinfo
-00000930: 726d 6174 6963 732f 6274 6165 3237 3229  rmatics/btae272)
-00000940: 2e20 4966 2079 6f75 2075 7365 2060 7767  . If you use `wg
-00000950: 6420 7632 6020 696e 2079 6f75 7220 7265  d v2` in your re
-00000960: 7365 6172 6368 2c20 706c 6561 7365 2063  search, please c
-00000970: 6974 6520 7573 2061 7320 7375 6767 6573  ite us as sugges
-00000980: 7465 6420 696e 205b 4369 7461 7469 6f6e  ted in [Citation
-00000990: 5d28 2363 6974 6174 696f 6e29 2073 6563  ](#citation) sec
-000009a0: 7469 6f6e 2e0a 0a23 2320 496e 7472 6f64  tion...## Introd
-000009b0: 7563 7469 6f6e 0a0a 506f 6c79 706c 6f69  uction..Polyploi
-000009c0: 6469 7a61 7469 6f6e 732c 2074 6865 2065  dizations, the e
-000009d0: 766f 6c75 7469 6f6e 6172 7920 7072 6f63  volutionary proc
-000009e0: 6573 7320 7468 6174 2074 6865 2065 6e74  ess that the ent
-000009f0: 6972 6520 6765 6e6f 6d65 206f 6620 616e  ire genome of an
-00000a00: 206f 7267 616e 6973 6d20 6973 2064 7570   organism is dup
-00000a10: 6c69 6361 7465 642c 2061 6c73 6f20 6e61  licated, also na
-00000a20: 6d65 6420 6173 2077 686f 6c65 2d67 656e  med as whole-gen
-00000a30: 6f6d 6520 6475 706c 6963 6174 696f 6e73  ome duplications
-00000a40: 2028 5747 4473 292c 206f 6363 7572 2072   (WGDs), occur r
-00000a50: 6563 7572 7265 6e74 6c79 2061 6372 6f73  ecurrently acros
-00000a60: 7320 7468 6520 7472 6565 206f 6620 6c69  s the tree of li
-00000a70: 6665 2e20 5468 6572 6520 6172 6520 7477  fe. There are tw
-00000a80: 6f20 6d6f 6465 7320 6f66 2070 6f6c 7970  o modes of polyp
-00000a90: 6c6f 6964 697a 6174 696f 6e73 2c20 6175  loidizations, au
-00000aa0: 746f 706f 6c79 706c 6f69 6469 7a61 7469  topolyploidizati
-00000ab0: 6f6e 7320 616e 6420 616c 6c6f 706f 6c79  ons and allopoly
-00000ac0: 706c 6f69 6469 7a61 7469 6f6e 732e 2041  ploidizations. A
-00000ad0: 7574 6f70 6f6c 7970 6c6f 6964 697a 6174  utopolyploidizat
-00000ae0: 696f 6e73 2061 7265 2074 6865 2064 7570  ions are the dup
-00000af0: 6c69 6361 7469 6f6e 206f 6620 7468 6520  lication of the 
-00000b00: 7361 6d65 2067 656e 6f6d 652c 2072 6573  same genome, res
-00000b10: 756c 7469 6e67 2069 6e20 7477 6f20 6964  ulting in two id
-00000b20: 656e 7469 6361 6c20 7375 6267 656e 6f6d  entical subgenom
-00000b30: 6573 2061 7420 7468 6520 7469 6d65 2069  es at the time i
-00000b40: 7420 656d 6572 6765 642e 2057 6869 6c65  t emerged. While
-00000b50: 2074 6865 2061 6c6c 6f70 6f6c 7970 6c6f   the allopolyplo
-00000b60: 6964 697a 6174 696f 6e73 2061 7265 206e  idizations are n
-00000b70: 6f72 6d61 6c6c 7920 6163 6869 6576 6564  ormally achieved
-00000b80: 2069 6e20 7477 6f20 7374 6570 732c 2066   in two steps, f
-00000b90: 6972 7374 2074 6865 2068 7962 7269 6469  irst the hybridi
-00000ba0: 7a61 7469 6f6e 2062 6574 7765 656e 2074  zation between t
-00000bb0: 776f 2064 6966 6665 7265 6e74 2073 7065  wo different spe
-00000bc0: 6369 6573 2c20 7265 7375 6c74 696e 6720  cies, resulting 
-00000bd0: 696e 2074 6865 2061 7269 7369 6e67 206f  in the arising o
-00000be0: 6620 7472 616e 7369 656e 7420 686f 6d6f  f transient homo
-00000bf0: 706c 6f69 6479 2c73 6563 6f6e 6420 7468  ploidy,second th
-00000c00: 6520 6475 706c 6963 6174 696f 6e20 6f66  e duplication of
-00000c10: 2074 6865 2068 6f6d 6f70 6c6f 6964 792c   the homoploidy,
-00000c20: 2072 6573 756c 7469 6e67 2069 6e20 7468   resulting in th
-00000c30: 6520 656d 6572 6765 6e63 6520 6f66 2061  e emergence of a
-00000c40: 6c6c 6f70 6f6c 7970 6c6f 6964 792e 2044  llopolyploidy. D
-00000c50: 7565 2074 6f20 7468 6520 756e 7374 6162  ue to the unstab
-00000c60: 696c 6974 7920 616e 6420 756e 6261 6c61  ility and unbala
-00000c70: 6e63 6564 2074 6574 7261 736f 6d69 6320  nced tetrasomic 
-00000c80: 696e 6865 7269 7461 6e63 652c 2066 6f72  inheritance, for
-00000c90: 2069 6e73 7461 6e63 6520 7468 6520 6e75   instance the nu
-00000ca0: 636c 6561 722d 6379 746f 706c 6173 6d69  clear-cytoplasmi
-00000cb0: 6320 696e 636f 6d70 6174 6962 696c 6974  c incompatibilit
-00000cc0: 792c 2074 6865 2070 6f6c 7970 6c6f 6964  y, the polyploid
-00000cd0: 7920 6765 6e6f 6d65 2077 696c 6c20 7468  y genome will th
-00000ce0: 656e 2065 7870 6572 6965 6e63 6520 6120  en experience a 
-00000cf0: 7072 6f63 6573 7320 6361 6c6c 6564 2064  process called d
-00000d00: 6970 6c6f 6964 697a 6174 696f 6e2c 2061  iploidization, a
-00000d10: 6c73 6f20 6e61 6d65 6420 6173 2066 7261  lso named as fra
-00000d20: 6374 696f 6e61 7469 6f6e 2c20 6475 7269  ctionation, duri
-00000d30: 6e67 2077 6869 6368 2061 206c 6172 6765  ng which a large
-00000d40: 2070 6f72 7469 6f6e 206f 6620 6765 6e65   portion of gene
-00000d50: 2064 7570 6c69 6361 7465 7320 7769 6c6c   duplicates will
-00000d60: 2067 6574 206c 6f73 7420 616e 6420 6f6e   get lost and on
-00000d70: 6c79 2061 2066 7261 6374 696f 6e20 6361  ly a fraction ca
-00000d80: 6e20 6265 2072 6574 6169 6e65 642e 2054  n be retained. T
-00000d90: 6865 2074 7261 6365 7320 6f66 2070 6f6c  he traces of pol
-00000da0: 7970 6c6f 6964 697a 6174 696f 6e73 2063  yploidizations c
-00000db0: 616e 2062 6520 7468 7573 2075 6e65 6172  an be thus unear
-00000dc0: 7468 6564 2066 726f 6d20 7468 6573 6520  thed from these 
-00000dd0: 7265 7461 696e 6564 2067 656e 6520 6475  retained gene du
-00000de0: 706c 6963 6174 6573 2e20 5468 7265 6520  plicates. Three 
-00000df0: 6170 7072 6f61 6368 6573 2062 6173 6564  approaches based
-00000e00: 206f 6e20 6765 6e65 2064 7570 6c69 6361   on gene duplica
-00000e10: 7465 732c 206e 616d 656c 792c 202a 4b2a  tes, namely, *K*
-00000e20: 3c73 7562 3e53 3c2f 7375 623e 206d 6574  <sub>S</sub> met
-00000e30: 686f 642c 2067 656e 6520 7472 6565 202d  hod, gene tree -
-00000e40: 2073 7065 6369 6573 2074 7265 6520 7265   species tree re
-00000e50: 636f 6e63 696c 6961 7469 6f6e 206d 6574  conciliation met
-00000e60: 686f 6420 616e 6420 7379 6e74 656e 7920  hod and synteny 
-00000e70: 6d65 7468 6f64 2c20 6172 6520 636f 6d6d  method, are comm
-00000e80: 6f6e 6c79 2075 7365 6420 696e 2064 6574  only used in det
-00000e90: 6563 7469 6e67 2065 7669 6465 6e63 6520  ecting evidence 
-00000ea0: 666f 7220 5747 4473 2e20 5468 6520 6765  for WGDs. The ge
-00000eb0: 6e65 2074 7265 6520 2d20 7370 6563 6965  ne tree - specie
-00000ec0: 7320 7472 6565 2072 6563 6f6e 6369 6c69  s tree reconcili
-00000ed0: 6174 696f 6e20 6d65 7468 6f64 2069 7320  ation method is 
-00000ee0: 6e6f 7420 7769 7468 696e 2074 6865 2073  not within the s
-00000ef0: 636f 7065 206f 6620 6077 6764 2076 3260  cope of `wgd v2`
-00000f00: 2c20 6275 7420 7765 206b 696e 646c 7920  , but we kindly 
-00000f10: 7265 6665 7220 7265 6164 6572 7320 7768  refer readers wh
-00000f20: 6f20 6172 6520 696e 7465 7265 7374 6564  o are interested
-00000f30: 2074 6f20 7468 6520 7068 796c 6f67 656e   to the phylogen
-00000f40: 6f6d 6963 2070 726f 6772 616d 2064 6576  omic program dev
-00000f50: 656c 6f70 6564 2062 7920 4172 7468 7572  eloped by Arthur
-00000f60: 205a 7761 656e 6570 6f65 6c20 6e61 6d65   Zwaenepoel name
-00000f70: 6420 5b57 4841 4c45 5d28 6874 7470 733a  d [WHALE](https:
-00000f80: 2f2f 6769 7468 7562 2e63 6f6d 2f61 727a  //github.com/arz
-00000f90: 7761 2f57 6861 6c65 2e6a 6c29 2061 6e64  wa/Whale.jl) and
-00000fa0: 2074 6865 2061 7373 6f63 6961 7465 6420   the associated 
-00000fb0: 5b70 6170 6572 5d28 6874 7470 733a 2f2f  [paper](https://
-00000fc0: 646f 692e 6f72 672f 3130 2e31 3039 332f  doi.org/10.1093/
-00000fd0: 6d6f 6c62 6576 2f6d 737a 3038 3829 2066  molbev/msz088) f
-00000fe0: 6f72 206d 6f72 6520 7465 6368 6e69 6361  or more technica
-00000ff0: 6c20 616e 6420 7468 656f 7265 7469 6361  l and theoretica
-00001000: 6c20 6465 7461 696c 732e 0a0a 5468 6520  l details...The 
-00001010: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00001020: 6d65 7468 6f64 2069 7320 6573 7461 626c  method is establ
-00001030: 6973 6865 6420 6f6e 2061 206d 6f64 656c  ished on a model
-00001040: 206f 6620 6765 6e65 2066 616d 696c 7920   of gene family 
-00001050: 6576 6f6c 7574 696f 6e20 7468 6174 2065  evolution that e
-00001060: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
-00001070: 6973 2061 6c6c 6f77 6564 2074 6f20 6576  is allowed to ev
-00001080: 6f6c 7665 2076 6961 2067 656e 6520 6475  olve via gene du
-00001090: 706c 6963 6174 696f 6e20 616e 6420 6c6f  plication and lo
-000010a0: 7373 2e20 4e6f 7465 2074 6861 7420 7468  ss. Note that th
-000010b0: 6520 6765 6e65 2066 616d 696c 7920 6865  e gene family he
-000010c0: 7265 2069 7320 6173 7375 6d65 6420 746f  re is assumed to
-000010d0: 2062 6520 7468 6520 636c 7573 7465 7220   be the cluster 
-000010e0: 6f66 2061 6c6c 2067 656e 6573 2064 6573  of all genes des
-000010f0: 6365 6e64 6564 2066 726f 6d20 616e 2061  cended from an a
-00001100: 6e63 6573 7472 616c 2067 656e 6520 696e  ncestral gene in
-00001110: 2061 2073 696e 676c 6520 6765 6e6f 6d65   a single genome
-00001120: 2e20 5265 636f 7665 7269 6e67 2074 6865  . Recovering the
-00001130: 2067 656e 6520 7472 6565 206f 6620 7375   gene tree of su
-00001140: 6368 2067 656e 6520 6661 6d69 6c79 2069  ch gene family i
-00001150: 6e66 6f72 6d73 2074 6865 2074 696d 696e  nforms the timin
-00001160: 672c 2073 6369 6c69 6365 7420 7468 6520  g, scilicet the 
-00001170: 6167 652c 206f 6620 6765 6e65 2064 7570  age, of gene dup
-00001180: 6c69 6361 7469 6f6e 2065 7665 6e74 732e  lication events.
-00001190: 2054 6865 2061 6765 2072 6566 6572 6564   The age refered
-000011a0: 2068 6572 652c 2069 7320 6e6f 7420 696e   here, is not in
-000011b0: 2072 6561 6c20 6765 6f6c 6f67 6963 616c   real geological
-000011c0: 2074 696d 652c 2062 7574 2069 6e20 7468   time, but in th
-000011d0: 6520 756e 6974 206f 6620 6576 6f6c 7574  e unit of evolut
-000011e0: 696f 6e61 7279 2064 6973 7461 6e63 652c  ionary distance,
-000011f0: 2069 2e65 2e2c 2074 6865 206e 756d 6265   i.e., the numbe
-00001200: 7220 6f66 2073 7562 7374 6974 7574 696f  r of substitutio
-00001210: 6e73 2070 6572 2073 6974 652e 2057 6865  ns per site. Whe
-00001220: 6e20 7468 6520 6576 6f6c 7574 696f 6e61  n the evolutiona
-00001230: 7279 2072 6174 6520 7265 6d61 696e 7320  ry rate remains 
-00001240: 6170 7072 6f78 696d 6174 656c 7920 636f  approximately co
-00001250: 6e73 7461 6e74 2c20 7468 6520 6576 6f6c  nstant, the evol
-00001260: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
-00001270: 6520 6973 2074 6865 6e20 7375 7070 6f73  e is then suppos
-00001280: 6564 2074 6f20 6265 2070 726f 706f 7274  ed to be proport
-00001290: 696f 6e61 6c20 746f 2074 6865 2072 6561  ional to the rea
-000012a0: 6c20 6576 6f6c 7574 696f 6e61 7279 2074  l evolutionary t
-000012b0: 696d 652e 2054 6865 2073 796e 6f6e 796d  ime. The synonym
-000012c0: 6f75 7320 6469 7374 616e 6365 202a 4b2a  ous distance *K*
-000012d0: 3c73 7562 3e53 3c2f 7375 623e 2c20 7468  <sub>S</sub>, th
-000012e0: 6520 6e75 6d62 6572 206f 6620 7379 6e6f  e number of syno
-000012f0: 6e79 6d6f 7573 2073 7562 7374 6974 7574  nymous substitut
-00001300: 696f 6e73 2070 6572 2073 796e 6f6e 796d  ions per synonym
-00001310: 6f75 7320 7369 7465 2c20 6973 2073 7563  ous site, is suc
-00001320: 6820 6361 6e64 6964 6174 6520 7468 6174  h candidate that
-00001330: 2073 796e 6f6e 796d 6f75 7320 7375 6273   synonymous subs
-00001340: 7469 7475 7469 6f6e 7320 776f 756c 6420  titutions would 
-00001350: 6e6f 7420 696e 6375 7220 7468 6520 6368  not incur the ch
-00001360: 616e 6765 206f 6620 616d 696e 6f20 6163  ange of amino ac
-00001370: 6964 2061 6e64 2061 7265 2074 6875 7320  id and are thus 
-00001380: 7265 6761 7264 6564 2061 7320 6e65 7574  regarded as neut
-00001390: 7261 6c2c 2077 6869 6368 2061 6363 6f72  ral, which accor
-000013a0: 6469 6e67 2074 6f20 7468 6520 6e65 7574  ding to the neut
-000013b0: 7261 6c20 7468 656f 7279 2073 686f 756c  ral theory shoul
-000013c0: 6420 6f63 6375 7220 696e 2063 6f6e 7374  d occur in const
-000013d0: 616e 7420 7261 7465 2e20 4769 7665 6e20  ant rate. Given 
-000013e0: 6120 6d6f 6465 6c20 6f66 2067 656e 6520  a model of gene 
-000013f0: 6661 6d69 6c79 2074 6861 7420 616c 6c6f  family that allo
-00001400: 7773 2074 6865 2067 656e 6520 746f 2064  ws the gene to d
-00001410: 7570 6c69 6361 7465 2061 6e64 2067 6574  uplicate and get
-00001420: 206c 6f73 7420 696e 2061 2066 6978 6564   lost in a fixed
-00001430: 2072 6174 652c 206f 6e65 2063 616e 2064   rate, one can d
-00001440: 6572 6976 6520 7468 6174 2074 6865 2070  erive that the p
-00001450: 726f 6261 6269 6c69 7479 2064 656e 7369  robability densi
-00001460: 7479 2066 756e 6374 696f 6e20 6f66 2074  ty function of t
-00001470: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-00001480: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00001490: 696f 6e20 6f66 2072 6574 6169 6e65 6420  ion of retained 
-000014a0: 6765 6e65 2064 7570 6c69 6361 7465 7320  gene duplicates 
-000014b0: 6973 2061 2071 7561 7369 2d65 7870 6f6e  is a quasi-expon
-000014c0: 656e 7469 616c 2066 756e 6374 696f 6e20  ential function 
-000014d0: 7468 6174 206d 6f73 7420 7265 7461 696e  that most retain
-000014e0: 6564 2067 656e 6520 6475 706c 6963 6174  ed gene duplicat
-000014f0: 6573 2061 7265 2072 6563 656e 746c 7920  es are recently 
-00001500: 626f 726e 6564 2077 6974 6820 7e30 2061  borned with ~0 a
-00001510: 6765 2077 6869 6c65 2061 7320 7468 6520  ge while as the 
-00001520: 6167 6520 676f 696e 6720 6f6c 6465 7220  age going older 
-00001530: 7468 6520 6173 736f 6369 6174 6564 206e  the associated n
-00001540: 756d 6265 7220 6f66 2072 6574 6169 6e65  umber of retaine
-00001550: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
-00001560: 7320 6465 6361 7920 7175 6173 692d 6578  s decay quasi-ex
-00001570: 706f 6e65 6e74 6961 6c6c 792e 2054 6865  ponentially. The
-00001580: 7265 666f 7265 2c20 7468 6520 6f63 6375  refore, the occu
-00001590: 7261 6e63 6520 6f66 206c 6172 6765 2d73  rance of large-s
-000015a0: 6361 6c65 2067 656e 6520 6475 706c 6963  cale gene duplic
-000015b0: 6174 696f 6e20 6576 656e 7473 2c20 666f  ation events, fo
-000015c0: 7220 696e 7374 616e 6520 5747 4473 2c20  r instane WGDs, 
-000015d0: 7769 7468 2076 6172 6965 6420 7265 7465  with varied rete
-000015e0: 6e74 696f 6e20 7261 7465 2c20 7769 6c6c  ntion rate, will
-000015f0: 206c 6561 7665 2061 6e20 6167 6520 7065   leave an age pe
-00001600: 616b 2066 726f 6d20 7468 6520 6275 7273  ak from the burs
-00001610: 7420 6f66 2067 656e 6520 6475 706c 6963  t of gene duplic
-00001620: 6174 6573 2069 6e20 6120 7368 6f72 7420  ates in a short 
-00001630: 7469 6d65 2d66 7261 6d65 2075 706f 6e20  time-frame upon 
-00001640: 7468 6520 696e 6974 6961 6c20 6167 6520  the initial age 
-00001650: 6469 7374 7269 6275 7469 6f6e 2c20 616e  distribution, an
-00001660: 6420 6361 6e20 6265 2075 6e76 6569 6c65  d can be unveile
-00001670: 6420 6672 6f6d 206d 6978 7475 7265 206d  d from mixture m
-00001680: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
-00001690: 2e20 486f 7765 7665 722c 2057 4744 7320  . However, WGDs 
-000016a0: 6964 656e 7469 6669 6564 2066 726f 6d20  identified from 
-000016b0: 7468 6520 7061 7261 6c6f 676f 7573 202a  the paralogous *
-000016c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-000016d0: 6765 2064 6973 7472 6962 7574 696f 6e73  ge distributions
-000016e0: 2063 616e 206f 6e6c 7920 696e 666f 726d   can only inform
-000016f0: 2074 6865 2057 4744 2074 696d 696e 6720   the WGD timing 
-00001700: 696e 2074 6865 2074 696d 652d 7363 616c  in the time-scal
-00001710: 6520 6f66 2074 6861 7420 7370 6563 6966  e of that specif
-00001720: 6963 2073 7065 6369 6573 2c20 7768 6963  ic species, whic
-00001730: 6820 6973 206e 6f74 2063 6f6d 7061 7261  h is not compara
-00001740: 626c 6520 696e 2074 6865 2070 6879 6c6f  ble in the phylo
-00001750: 6765 6e65 7469 6320 636f 6e74 6578 742e  genetic context.
-00001760: 204f 6e6c 7920 7769 7468 2074 6865 206f   Only with the o
-00001770: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00001780: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-00001790: 6973 7472 6962 7574 696f 6e73 2c20 7768  istributions, wh
-000017a0: 6963 6820 636f 6e76 6572 7420 7468 6520  ich convert the 
-000017b0: 6573 7469 6d61 7465 6420 626f 6479 2066  estimated body f
-000017c0: 726f 6d20 7061 7261 6c6f 6775 6573 2074  rom paralogues t
-000017d0: 6f20 6f72 7468 6f6c 6f67 7565 7320 616e  o orthologues an
-000017e0: 6420 696e 666f 726d 2074 6865 2072 656c  d inform the rel
-000017f0: 6174 6976 6520 7469 6d69 6e67 206f 6620  ative timing of 
-00001800: 7370 6563 6961 7469 6f6e 2065 7665 6e74  speciation event
-00001810: 732c 2063 616e 2077 6520 6465 6369 7068  s, can we deciph
-00001820: 6572 2074 6865 2070 6879 6c6f 6765 6e65  er the phylogene
-00001830: 7469 6320 706c 6163 656d 656e 7420 6f66  tic placement of
-00001840: 2057 4744 7320 6166 7465 7220 7072 6f70   WGDs after prop
-00001850: 6572 2072 6174 6520 636f 7272 6563 7469  er rate correcti
-00001860: 6f6e 2e20 6077 6764 2076 3260 2069 7320  on. `wgd v2` is 
-00001870: 7375 6368 2070 726f 6772 616d 2074 6861  such program tha
-00001880: 7420 6865 6c70 7320 7573 6572 7320 636f  t helps users co
-00001890: 6e73 7472 7563 7420 7061 7261 6c6f 676f  nstruct paralogo
-000018a0: 7573 2061 6e64 206f 7274 686f 6c6f 676f  us and orthologo
-000018b0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-000018c0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-000018d0: 696f 6e73 2061 6e64 2072 6561 6c69 7a65  ions and realize
-000018e0: 2062 6f74 6820 7468 6520 6964 656e 7469   both the identi
-000018f0: 6669 6361 7469 6f6e 2061 6e64 2070 6c61  fication and pla
-00001900: 6365 6d65 6e74 206f 6620 5747 4473 2e0a  cement of WGDs..
-00001910: 0a49 6e20 7468 6520 7072 656d 6973 6520  .In the premise 
-00001920: 6f66 2070 6879 6c6f 6765 6e65 7469 6361  of phylogenetica
-00001930: 6c6c 7920 6c6f 6361 7465 6420 5747 4473  lly located WGDs
-00001940: 2c20 7468 6520 6162 736f 6c75 7465 2061  , the absolute a
-00001950: 6765 2028 696e 2067 656f 6c6f 6769 6361  ge (in geologica
-00001960: 6c20 7469 6d65 2920 6f66 2057 4744 7320  l time) of WGDs 
-00001970: 6361 6e20 616c 736f 2062 6520 696e 6665  can also be infe
-00001980: 7272 6564 2066 726f 6d20 7468 6f73 6520  rred from those 
-00001990: 5747 442d 7265 7461 696e 6564 2067 656e  WGD-retained gen
-000019a0: 6520 6475 706c 6963 6174 6573 2c20 616c  e duplicates, al
-000019b0: 7468 6f75 6768 2074 6865 7265 2068 6173  though there has
-000019c0: 2062 6565 6e20 6e6f 2065 6173 7920 6f72   been no easy or
-000019d0: 2073 7472 6169 6768 7466 6f72 7761 7264   straightforward
-000019e0: 2070 6970 656c 696e 6520 666f 7220 7468   pipeline for th
-000019f0: 6973 206a 6f62 2073 6f20 6661 722e 2049  is job so far. I
-00001a00: 6e20 6077 6764 2076 3260 2c20 7765 2064  n `wgd v2`, we d
-00001a10: 6576 656c 6f70 6564 2061 2066 6561 7369  eveloped a feasi
-00001a20: 626c 6520 696e 7465 6772 6174 6564 2070  ble integrated p
-00001a30: 6970 6c65 6c69 6e65 2066 6f72 2061 6273  ipleline for abs
-00001a40: 6f6c 7574 6520 6461 7469 6e67 206f 6620  olute dating of 
-00001a50: 5747 4473 2e20 5468 6520 7069 7065 6c69  WGDs. The pipeli
-00001a60: 6e65 2063 616e 2062 6520 726f 7567 686c  ne can be roughl
-00001a70: 7920 6469 7669 6465 6420 696e 746f 2074  y divided into t
-00001a80: 6872 6565 206d 6169 6e20 7374 6570 732e  hree main steps.
-00001a90: 2031 2920 5468 6520 636f 6e73 7472 7563   1) The construc
-00001aa0: 7469 6f6e 206f 6620 616e 6368 6f72 202a  tion of anchor *
-00001ab0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
-00001ac0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
-00001ad0: 7468 6520 6465 6c69 6e65 6174 696f 6e20  the delineation 
-00001ae0: 6f66 2063 7265 6469 6162 6c65 202a 4b2a  of crediable *K*
-00001af0: 3c73 7562 3e53 3c2f 7375 623e 2072 616e  <sub>S</sub> ran
-00001b00: 6765 2061 646f 7074 6564 2066 6f72 2070  ge adopted for p
-00001b10: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
-00001b20: 6e67 2c20 7573 696e 6720 6077 6764 2064  ng, using `wgd d
-00001b30: 6d64 602c 2060 7767 6420 6b73 6460 2c20  md`, `wgd ksd`, 
-00001b40: 6077 6764 2073 796e 6020 616e 6420 6077  `wgd syn` and `w
-00001b50: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
-00001b60: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
-00001b70: 2063 6f6e 7369 6465 7220 6765 6e6f 6d65   consider genome
-00001b80: 2061 7373 656d 626c 7920 6265 6361 7573   assembly becaus
-00001b90: 6520 666f 7220 7472 616e 7363 7269 7074  e for transcript
-00001ba0: 6f6d 6520 6173 7365 6d62 6c79 2069 7427  ome assembly it'
-00001bb0: 7320 696d 706f 7373 6962 6c65 2074 6f20  s impossible to 
-00001bc0: 6469 7374 696e 6775 6973 6820 5747 442d  distinguish WGD-
-00001bd0: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
-00001be0: 6573 2066 726f 6d20 736d 616c 6c2d 7363  es from small-sc
-00001bf0: 616c 6520 6475 706c 6963 6174 696f 6e2d  ale duplication-
-00001c00: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
-00001c10: 6573 2c20 7768 6963 6820 6861 7070 656e  es, which happen
-00001c20: 6564 2069 6e20 6120 636f 6e74 696e 756f  ed in a continuo
-00001c30: 7573 2074 696d 652d 6672 616d 6520 696e  us time-frame in
-00001c40: 7374 6561 6420 6f66 206f 6e6c 7920 6120  stead of only a 
-00001c50: 7365 7061 7261 7465 2073 686f 7274 2074  separate short t
-00001c60: 696d 652d 6672 616d 6520 616e 6420 7468  ime-frame and th
-00001c70: 7573 2072 6566 6c65 6374 7320 7468 6520  us reflects the 
-00001c80: 6475 7261 7469 6f6e 206f 6620 7468 6174  duration of that
-00001c90: 2062 7261 6e63 6820 7261 7468 6572 2074   branch rather t
-00001ca0: 6861 6e20 7468 6520 7469 6d65 2061 7420  han the time at 
-00001cb0: 7768 6963 6820 5747 4420 6f63 6375 7272  which WGD occurr
-00001cc0: 6564 2e20 3229 2054 6865 2066 6f72 6d75  ed. 2) The formu
-00001cd0: 6c61 7469 6f6e 206f 6620 6120 7374 6172  lation of a star
-00001ce0: 7469 6e67 2074 7265 6520 7573 6564 2069  ting tree used i
-00001cf0: 6e20 7468 6520 7068 796c 6f67 656e 6574  n the phylogenet
-00001d00: 6963 2064 6174 696e 672c 2063 6f6d 706f  ic dating, compo
-00001d10: 7365 6420 6f66 2061 2066 6577 2073 7065  sed of a few spe
-00001d20: 6369 6573 2061 6e64 2061 6e6e 6f74 6174  cies and annotat
-00001d30: 6564 2077 6974 6820 666f 7373 696c 2063  ed with fossil c
-00001d40: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-00001d50: 6d61 7469 6f6e 2e20 5468 6973 2073 7465  mation. This ste
-00001d60: 7020 6973 2065 7373 656e 7469 616c 2066  p is essential f
-00001d70: 6f72 2074 6865 2072 6573 756c 7420 6f66  or the result of
-00001d80: 2061 6273 6f6c 7574 6520 5747 4420 6461   absolute WGD da
-00001d90: 7469 6e67 2074 6861 7420 7765 2073 7567  ting that we sug
-00001da0: 6765 7374 2075 7365 7273 206f 6620 7461  gest users of ta
-00001db0: 6b69 6e67 2067 7265 6174 2063 6175 7469  king great cauti
-00001dc0: 6f6e 2074 6f20 6173 7375 7265 2074 6865  on to assure the
-00001dd0: 2074 7265 6520 746f 706f 6c6f 6779 2061   tree topology a
-00001de0: 6e64 2070 726f 7065 7220 626f 756e 6473  nd proper bounds
-00001df0: 2066 6f72 2066 6f73 7369 6c20 6361 6c69   for fossil cali
-00001e00: 6272 6174 696f 6e73 2e20 3329 2054 6865  brations. 3) The
-00001e10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
-00001e20: 206f 7274 686f 6772 6f75 7073 2063 6f6e   orthogroups con
-00001e30: 7369 7374 696e 6720 6f66 2063 6f6c 6c69  sisting of colli
-00001e40: 6e65 6172 2064 7570 6c69 6361 7465 7320  near duplicates 
-00001e50: 6f66 2074 6865 2066 6f63 616c 2073 7065  of the focal spe
-00001e60: 6369 6573 2061 6e64 2074 6865 6972 2072  cies and their r
-00001e70: 6563 6970 726f 6361 6c20 6265 7374 2068  eciprocal best h
-00001e80: 6974 7320 2852 4248 7329 2061 6761 696e  its (RBHs) again
-00001e90: 7374 206f 7468 6572 2073 7065 6369 6573  st other species
-00001ea0: 2069 6e20 7468 6520 7374 6172 7469 6e67   in the starting
-00001eb0: 2074 7265 6520 616e 6420 7468 6520 7068   tree and the ph
-00001ec0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-00001ed0: 6720 7573 696e 6720 6120 6d6f 6c65 6375  g using a molecu
-00001ee0: 6c61 7220 6461 7469 6e67 2070 726f 6772  lar dating progr
-00001ef0: 616d 2066 6f72 2069 6e73 7461 6e63 6520  am for instance 
-00001f00: 606d 636d 6374 7265 6560 2c20 7669 6120  `mcmctree`, via 
-00001f10: 7072 6f67 7261 6d20 6077 6764 2064 6d64  program `wgd dmd
-00001f20: 6020 616e 6420 6077 6764 2066 6f63 7573  ` and `wgd focus
-00001f30: 602e 2057 6520 7265 636f 6d6d 656e 6420  `. We recommend 
-00001f40: 7468 6520 7573 6167 6520 6f66 2042 6179  the usage of Bay
-00001f50: 6573 6961 6e20 6d6f 6c65 6375 6c61 7220  esian molecular 
-00001f60: 6461 7469 6e67 2070 726f 6772 616d 2060  dating program `
-00001f70: 6d63 6d63 7472 6565 602c 2077 6869 6368  mcmctree`, which
-00001f80: 2070 726f 7669 6465 7320 6120 7661 7269   provides a vari
-00001f90: 6574 7920 6f66 2064 6966 6665 7265 6e74  ety of different
-00001fa0: 2073 7562 7374 6974 7574 696f 6e20 616e   substitution an
-00001fb0: 6420 7261 7465 206d 6f64 656c 732e 204e  d rate models. N
-00001fc0: 6f6e 6574 6865 6c65 7373 2c20 7765 2075  onetheless, we u
-00001fd0: 7267 6520 7573 6572 7320 746f 2073 6574  rge users to set
-00001fe0: 2070 7269 6f72 2064 6973 7472 6962 7574   prior distribut
-00001ff0: 696f 6e20 6f66 2064 6966 6665 7265 6e74  ion of different
-00002000: 2070 6172 616d 6574 6572 7320 7769 7468   parameters with
-00002010: 2063 6175 7469 6f6e 2061 6e64 2061 7373   caution and ass
-00002020: 7572 6520 6164 6571 7561 7465 2073 616d  ure adequate sam
-00002030: 706c 696e 6720 6f66 2064 6966 6665 7265  pling of differe
-00002040: 6e74 2070 6172 616d 6574 6572 732e 0a0a  nt parameters...
-00002050: 2323 2049 6e73 7461 6c6c 6174 696f 6e0a  ## Installation.
-00002060: 0a54 6865 2065 6173 6965 7374 2077 6179  .The easiest way
-00002070: 2074 6f20 696e 7374 616c 6c20 6077 6764   to install `wgd
-00002080: 2076 3260 2069 7320 7573 696e 6720 6050   v2` is using `P
-00002090: 5950 4960 2e20 4e6f 7465 2074 6861 7420  YPI`. Note that 
-000020a0: 6966 2079 6f75 2077 616e 7420 746f 2067  if you want to g
-000020b0: 6574 2074 6865 206c 6174 6573 7420 7570  et the latest up
-000020c0: 6461 7465 2c20 7765 2073 7567 6765 7374  date, we suggest
-000020d0: 2069 6e73 7461 6c6c 696e 6720 6672 6f6d   installing from
-000020e0: 2074 6865 2073 6f75 7263 652c 2073 696e   the source, sin
-000020f0: 6365 2074 6865 2075 7064 6174 6520 6f6e  ce the update on
-00002100: 2060 5059 5049 6020 7769 6c6c 2062 6520   `PYPI` will be 
-00002110: 6465 6c61 7965 6420 636f 6d70 6172 6564  delayed compared
-00002120: 2074 6f20 6865 7265 206f 6620 736f 7572   to here of sour
-00002130: 6365 2e0a 0a60 6060 0a70 6970 2069 6e73  ce...```.pip ins
-00002140: 7461 6c6c 2077 6764 0a60 6060 0a0a 546f  tall wgd.```..To
-00002150: 2069 6e73 7461 6c6c 2060 7767 6420 7632   install `wgd v2
-00002160: 6020 696e 2061 2076 6972 7475 616c 2065  ` in a virtual e
-00002170: 6e76 6972 6f6e 6d65 6e74 2028 7768 6963  nvironment (whic
-00002180: 6820 7765 2073 7472 6f6e 676c 7920 7265  h we strongly re
-00002190: 636f 6d6d 656e 642c 206e 6f20 6d61 7474  commend, no matt
-000021a0: 6572 2079 6f75 2069 6e73 7461 6c6c 2066  er you install f
-000021b0: 726f 6d20 736f 7572 6365 2c20 6050 5950  rom source, `PYP
-000021c0: 4960 206f 7220 6062 696f 636f 6e64 6160  I` or `bioconda`
-000021d0: 292c 2074 6865 2066 6f6c 6c6f 7769 6e67  ), the following
-000021e0: 2063 6f6d 6d61 6e64 206c 696e 6573 2063   command lines c
-000021f0: 6f75 6c64 2062 6520 7573 6564 2e0a 0a60  ould be used...`
-00002200: 6060 0a67 6974 2063 6c6f 6e65 2068 7474  ``.git clone htt
-00002210: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00002220: 6865 6368 652d 7073 622f 7767 640a 6364  heche-psb/wgd.cd
-00002230: 2077 6764 0a76 6972 7475 616c 656e 7620   wgd.virtualenv 
-00002240: 2d70 3d70 7974 686f 6e33 2045 4e56 2028  -p=python3 ENV (
-00002250: 6f72 2070 7974 686f 6e33 202d 6d20 7665  or python3 -m ve
-00002260: 6e76 2045 4e56 290a 736f 7572 6365 2045  nv ENV).source E
-00002270: 4e56 2f62 696e 2f61 6374 6976 6174 650a  NV/bin/activate.
-00002280: 7069 7020 696e 7374 616c 6c20 2d72 2072  pip install -r r
-00002290: 6571 7569 7265 6d65 6e74 732e 7478 740a  equirements.txt.
-000022a0: 7069 7020 696e 7374 616c 6c20 2e0a 6060  pip install ..``
-000022b0: 600a 0a57 6865 6e20 6d65 7420 7769 7468  `..When met with
-000022c0: 2070 6572 6d69 7373 696f 6e20 7072 6f62   permission prob
-000022d0: 6c65 6d20 696e 2069 6e73 7461 6c6c 6174  lem in installat
-000022e0: 696f 6e2c 2070 6c65 6173 6520 7472 7920  ion, please try 
-000022f0: 7468 6520 666f 6c6c 6f77 696e 6720 636f  the following co
-00002300: 6d6d 616e 6420 6c69 6e65 2e0a 0a60 6060  mmand line...```
-00002310: 0a70 6970 2069 6e73 7461 6c6c 202d 6520  .pip install -e 
-00002320: 2e0a 6060 600a 0a49 6620 6d75 6c74 6970  ..```..If multip
-00002330: 6c79 2076 6572 7369 6f6e 7320 6f66 2060  ly versions of `
-00002340: 7767 6460 2077 6572 6520 696e 7374 616c  wgd` were instal
-00002350: 6c65 6420 696e 2074 6865 2073 7973 7465  led in the syste
-00002360: 6d2c 2070 6c65 6173 6520 6164 6420 7468  m, please add th
-00002370: 6520 7269 6768 7420 7061 7468 206f 6620  e right path of 
-00002380: 696e 7465 7265 7374 6564 2076 6572 7369  interested versi
-00002390: 6f6e 2069 6e74 6f20 7468 6520 656e 7669  on into the envi
-000023a0: 726f 6e6d 656e 7420 7661 7269 6162 6c65  ronment variable
-000023b0: 732c 2066 6f72 2065 7861 6d70 6c65 0a0a  s, for example..
-000023c0: 6060 600a 6578 706f 7274 2050 4154 483d  ```.export PATH=
-000023d0: 2224 5041 5448 3a7e 2f2e 6c6f 6361 6c2f  "$PATH:~/.local/
-000023e0: 6269 6e2f 7767 6422 0a60 6060 0a0a 4e6f  bin/wgd".```..No
-000023f0: 7465 2074 6861 7420 7468 6520 7665 7273  te that the vers
-00002400: 696f 6e20 6f66 2060 6e75 6d70 7960 2069  ion of `numpy` i
-00002410: 7320 696d 706f 7274 616e 7420 2866 6f72  s important (for
-00002420: 206d 616e 7920 6f74 6865 7220 7061 636b   many other pack
-00002430: 6167 6573 2061 7265 2074 6865 2073 616d  ages are the sam
-00002440: 6520 6f66 2063 6f75 7273 6529 2c20 6573  e of course), es
-00002450: 7065 6369 616c 6c79 2066 6f72 2060 6661  pecially for `fa
-00002460: 7374 636c 7573 7465 7260 2070 6163 6b61  stcluster` packa
-00002470: 6765 2e20 496e 206f 7572 2074 6573 742c  ge. In our test,
-00002480: 2074 6865 2060 6e75 6d70 7960 2031 2e31   the `numpy` 1.1
-00002490: 392e 3020 776f 726b 7320 6669 6e65 206f  9.0 works fine o
-000024a0: 6e20 6070 7974 686f 6e33 2e36 2f38 602e  n `python3.6/8`.
-000024b0: 2049 6620 796f 7520 6d65 7420 736f 6d65   If you met some
-000024c0: 2065 7272 6f72 7320 6f72 2077 6172 6e69   errors or warni
-000024d0: 6e67 7320 6162 6f75 7420 606e 756d 7079  ngs about `numpy
-000024e0: 602c 206d 6179 6265 2063 6f6e 7369 6465  `, maybe conside
-000024f0: 7269 6e67 2070 7265 2d69 6e73 7461 6c6c  ring pre-install
-00002500: 2060 6e75 6d70 7960 2061 7320 312e 3139   `numpy` as 1.19
-00002510: 2e30 206f 7220 6f74 6865 7220 636c 6f73  .0 or other clos
-00002520: 652d 6279 2076 6572 7369 6f6e 7320 6265  e-by versions be
-00002530: 666f 7265 2079 6f75 2069 6e73 7461 6c6c  fore you install
-00002540: 2060 7767 6460 2e20 6077 6764 6020 7265   `wgd`. `wgd` re
-00002550: 6c69 6573 206f 6e20 6578 7465 726e 616c  lies on external
-00002560: 2073 6f66 7477 6172 6573 2069 6e63 6c75   softwares inclu
-00002570: 6469 6e67 2060 6469 616d 6f6e 6460 2061  ding `diamond` a
-00002580: 6e64 2060 6d63 6c60 2066 6f72 2060 7767  nd `mcl` for `wg
-00002590: 6420 646d 6460 2c20 6070 616d 6c20 7634  d dmd`, `paml v4
-000025a0: 2e39 6a60 2c20 606d 6166 6674 6020 2860  .9j`, `mafft` (`
-000025b0: 6d75 7363 6c65 6020 616e 6420 6070 7261  muscle` and `pra
-000025c0: 6e6b 6020 6966 2073 6574 292c 2060 6661  nk` if set), `fa
-000025d0: 7374 7472 6565 6020 286f 7220 6069 7174  sttree` (or `iqt
-000025e0: 7265 6560 2069 6620 7365 7429 2066 6f72  ree` if set) for
-000025f0: 2060 7767 6420 6b73 6460 2061 6e64 206f   `wgd ksd` and o
-00002600: 7074 696f 6e61 6c6c 7920 606d 7262 6179  ptionally `mrbay
-00002610: 6573 6020 666f 7220 7468 6520 7068 796c  es` for the phyl
-00002620: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
-00002630: 6365 2066 756e 6374 696f 6e20 696e 2060  ce function in `
-00002640: 7767 6420 646d 6460 2061 6e64 2060 7767  wgd dmd` and `wg
-00002650: 6420 666f 6375 7360 2028 606d 6166 6674  d focus` (`mafft
-00002660: 602c 2060 6d75 7363 6c65 6020 616e 6420  `, `muscle` and 
-00002670: 6070 7261 6e6b 6020 6173 2077 656c 6c20  `prank` as well 
-00002680: 7768 656e 2074 6865 2061 6e61 6c79 7369  when the analysi
-00002690: 7320 7265 7175 6972 6573 2073 6571 7565  s requires seque
-000026a0: 6e63 6520 616c 6967 6e6d 656e 7429 2e20  nce alignment). 
-000026b0: 536f 6d65 206f 7468 6572 206f 7074 696f  Some other optio
-000026c0: 6e61 6c20 736f 6674 7761 7265 7320 696e  nal softwares in
-000026d0: 636c 7564 696e 6720 6070 616d 6c20 7634  cluding `paml v4
-000026e0: 2e39 6a60 2c20 6072 3873 602c 2060 6265  .9j`, `r8s`, `be
-000026f0: 6173 7460 2c20 6065 6767 6e6f 6760 2c20  ast`, `eggnog`, 
-00002700: 6064 6961 6d6f 6e64 602c 2060 696e 7465  `diamond`, `inte
-00002710: 7270 726f 7363 616e 602c 2060 686d 6d65  rproscan`, `hmme
-00002720: 7220 7633 2e31 6232 6020 616e 6420 6061  r v3.1b2` and `a
-00002730: 7374 7261 6c2d 7072 6f60 2028 6068 6d6d  stral-pro` (`hmm
-00002740: 6572 2076 332e 3162 3260 2069 7320 616c  er v3.1b2` is al
-00002750: 736f 2066 6f72 2074 6865 205b 6f72 7468  so for the [orth
-00002760: 6f67 726f 7570 2061 7373 6967 6e6d 656e  ogroup assignmen
-00002770: 745d 2868 7474 7073 3a2f 2f77 6764 7632  t](https://wgdv2
-00002780: 2e72 6561 6474 6865 646f 6373 2e69 6f2f  .readthedocs.io/
-00002790: 656e 2f6c 6174 6573 742f 7573 6167 652e  en/latest/usage.
-000027a0: 6874 6d6c 236f 7274 686f 6772 6f75 702d  html#orthogroup-
-000027b0: 6173 7369 676e 6d65 6e74 2920 6675 6e63  assignment) func
-000027c0: 7469 6f6e 2069 6e20 6077 6764 2064 6d64  tion in `wgd dmd
-000027d0: 6020 616e 6420 6061 7374 7261 6c2d 7072  ` and `astral-pr
-000027e0: 6f60 2069 7320 616c 736f 2066 6f72 2074  o` is also for t
-000027f0: 6865 2066 756e 6374 696f 6e20 602d 2d63  he function `--c
-00002800: 6f6c 6c69 6e65 6172 636f 616c 6573 6365  ollinearcoalesce
-00002810: 6e63 6560 2069 6e20 6077 6764 2064 6d64  nce` in `wgd dmd
-00002820: 6029 2061 7265 2066 6f72 2074 6865 206d  `) are for the m
-00002830: 6f6c 6563 756c 6172 2064 6174 696e 672c  olecular dating,
-00002840: 2067 656e 6520 6661 6d69 6c79 2066 756e   gene family fun
-00002850: 6374 696f 6e20 616e 6e6f 7461 7469 6f6e  ction annotation
-00002860: 206f 7220 7068 796c 6f67 656e 6574 6963   or phylogenetic
-00002870: 2069 6e66 6572 656e 6365 2069 6e20 6077   inference in `w
-00002880: 6764 2066 6f63 7573 602e 0a0a 2323 2050  gd focus`...## P
-00002890: 6172 616d 6574 6572 730a 0a54 6865 7265  arameters..There
-000028a0: 2061 7265 2037 206d 6169 6e20 7072 6f67   are 7 main prog
-000028b0: 7261 6d73 2069 6e20 6077 6764 2076 3260  rams in `wgd v2`
-000028c0: 3a20 6064 6d64 602c 6066 6f63 7573 602c  : `dmd`,`focus`,
-000028d0: 606b 7364 602c 606d 6978 602c 6070 6561  `ksd`,`mix`,`pea
-000028e0: 6b60 2c60 7379 6e60 2c60 7669 7a60 2e20  k`,`syn`,`viz`. 
-000028f0: 4865 7265 6166 7465 7220 7765 2077 696c  Hereafter we wil
-00002900: 6c20 7072 6f76 6964 6520 6120 6465 7461  l provide a deta
-00002910: 696c 6564 2065 6c75 6369 6461 7469 6f6e  iled elucidation
-00002920: 206f 6e20 6561 6368 206f 6620 7468 6520   on each of the 
-00002930: 7072 6f67 7261 6d20 616e 6420 6974 7320  program and its 
-00002940: 6173 736f 6369 6174 6564 2070 6172 616d  associated param
-00002950: 6574 6572 732e 2050 6c65 6173 6520 7265  eters. Please re
-00002960: 6665 7220 746f 2074 6865 205b 5573 6167  fer to the [Usag
-00002970: 655d 2823 7573 6167 6529 2066 6f72 2074  e](#usage) for t
-00002980: 6865 2073 6365 6e61 7269 6f73 2074 6f20  he scenarios to 
-00002990: 7768 6963 6820 6561 6368 2070 6172 616d  which each param
-000029a0: 6574 6572 2061 7070 6c69 6573 2e0a 0a54  eter applies...T
-000029b0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-000029c0: 646d 6460 2063 616e 2072 6561 6c69 7a65  dmd` can realize
-000029d0: 2074 6865 2064 656c 696e 6561 7469 6f6e   the delineation
-000029e0: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
-000029f0: 6d65 2c20 5242 4873 2028 5265 6369 7072  me, RBHs (Recipr
-00002a00: 6f63 616c 2042 6573 7420 4869 7473 292c  ocal Best Hits),
-00002a10: 204d 5242 4873 2028 4d75 6c74 6970 6c65   MRBHs (Multiple
-00002a20: 2052 6563 6970 726f 6361 6c20 4265 7374   Reciprocal Best
-00002a30: 2048 6974 7329 2c20 6f72 7468 6f67 726f   Hits), orthogro
-00002a40: 7570 7320 616e 6420 736f 6d65 206f 7468  ups and some oth
-00002a50: 6572 206f 7274 686f 6772 6f75 702d 7265  er orthogroup-re
-00002a60: 6c61 7465 6420 6675 6e63 7469 6f6e 732c  lated functions,
-00002a70: 2069 6e63 6c75 6469 6e67 2063 6972 6375   including circu
-00002a80: 6d73 6372 6970 7469 6f6e 206f 6620 6e65  mscription of ne
-00002a90: 7374 6564 2073 696e 676c 652d 636f 7079  sted single-copy
-00002aa0: 206f 7274 686f 6772 6f75 7073 2028 4e53   orthogroups (NS
-00002ab0: 4f47 7329 2c20 756e 6269 6173 6564 2074  OGs), unbiased t
-00002ac0: 6573 7420 6f66 2073 696e 676c 652d 636f  est of single-co
-00002ad0: 7079 206f 7274 686f 6772 6f75 7073 2028  py orthogroups (
-00002ae0: 534f 4773 2920 6f76 6572 206d 6973 7369  SOGs) over missi
-00002af0: 6e67 2069 6e70 6172 616c 6f67 732c 2063  ng inparalogs, c
-00002b00: 6f6e 7374 7275 6374 696f 6e20 6f66 2042  onstruction of B
-00002b10: 5553 434f 2d67 7569 6465 6420 7369 6e67  USCO-guided sing
-00002b20: 6c65 2d63 6f70 7920 6f72 7468 6f67 726f  le-copy orthogro
-00002b30: 7570 7320 2853 4f47 7329 2c61 6e64 2074  ups (SOGs),and t
-00002b40: 6865 2063 6f6c 6c69 6e65 6172 2063 6f61  he collinear coa
-00002b50: 6c65 7363 656e 6365 2069 6e66 6572 656e  lescence inferen
-00002b60: 6365 206f 6620 7068 796c 6f67 656e 792e  ce of phylogeny.
-00002b70: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
-00002b80: 7565 6e63 6573 2028 6f70 7469 6f6e 290a  uences (option).
-00002b90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ba0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bb0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bc0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002be0: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00002bf0: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00002c00: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00002c10: 5f64 6d64 0a2d 742c 202d 2d74 6d70 6469  _dmd.-t, --tmpdi
-00002c20: 722c 2074 6865 2074 656d 706f 7261 7279  r, the temporary
-00002c30: 2077 6f72 6b69 6e67 2064 6972 6563 746f   working directo
-00002c40: 7279 2c20 6465 6661 756c 7420 4e6f 6e65  ry, default None
-00002c50: 2c20 6966 204e 6f6e 6520 7761 7320 6769  , if None was gi
-00002c60: 7665 6e2c 2074 6865 2074 6d70 6469 7220  ven, the tmpdir 
-00002c70: 776f 756c 6420 6265 2061 7373 6967 6e65  would be assigne
-00002c80: 6420 7261 6e64 6f6d 206e 616d 6573 2069  d random names i
-00002c90: 6e20 6375 7272 656e 7420 6469 7265 6374  n current direct
-00002ca0: 6f72 7920 616e 6420 6175 746f 6d61 7465  ory and automate
-00002cb0: 6c79 2072 656d 6f76 6564 2061 7420 7468  ly removed at th
-00002cc0: 6520 636f 6d70 6c65 7469 6f6e 206f 6620  e completion of 
-00002cd0: 7072 6f67 7261 6d2c 2065 6c73 6520 7468  program, else th
-00002ce0: 6520 746d 7064 6972 2077 6f75 6c64 2062  e tmpdir would b
-00002cf0: 6520 6b65 7074 0a2d 702c 202d 2d70 726f  e kept.-p, --pro
-00002d00: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
-00002d10: 7768 6574 6865 7220 7573 696e 6720 7072  whether using pr
-00002d20: 6f74 6569 6e20 6f72 206e 7563 6c65 6f74  otein or nucleot
-00002d30: 6964 6520 7365 7175 656e 6365 730a 2d63  ide sequences.-c
-00002d40: 2c20 2d2d 6373 636f 7265 2c20 7468 6520  , --cscore, the 
-00002d50: 632d 7363 6f72 6520 746f 2072 6573 7472  c-score to restr
-00002d60: 6963 7420 7468 6520 686f 6d6f 6c6f 6720  ict the homolog 
-00002d70: 7369 6d69 6c61 7269 7479 206f 6620 4d52  similarity of MR
-00002d80: 4248 732c 2064 6566 6175 6c74 204e 6f6e  BHs, default Non
-00002d90: 652c 2069 6620 4e6f 6e65 2077 6173 2067  e, if None was g
-00002da0: 6976 656e 2c20 7468 6520 632d 7363 6f72  iven, the c-scor
-00002db0: 6520 6675 6e63 696f 6e20 776f 756c 646e  e funcion wouldn
-00002dc0: 2774 2062 6520 6163 7469 7661 7465 642c  't be activated,
-00002dd0: 2065 6c73 6520 6578 7065 6374 696e 6720   else expecting 
-00002de0: 6120 6465 6369 6d61 6c20 7769 7468 696e  a decimal within
-00002df0: 2074 6865 2072 616e 6765 206f 6620 3020   the range of 0 
-00002e00: 616e 6420 310a 2d49 2c20 2d2d 696e 666c  and 1.-I, --infl
-00002e10: 6174 696f 6e2c 2074 6865 2069 6e66 6c61  ation, the infla
-00002e20: 7469 6f6e 2066 6163 746f 7220 666f 7220  tion factor for 
-00002e30: 4d43 4c20 7072 6f67 7261 6d2c 2064 6566  MCL program, def
-00002e40: 6175 6c74 2032 2e30 2c20 7769 7468 2068  ault 2.0, with h
-00002e50: 6967 6865 7220 7661 6c75 6520 6c65 6164  igher value lead
-00002e60: 696e 6720 746f 206d 6f72 6520 6275 7420  ing to more but 
-00002e70: 736d 616c 6c65 7220 636c 7573 7465 7273  smaller clusters
-00002e80: 0a2d 652c 202d 2d65 7661 6c2c 2074 6865  .-e, --eval, the
-00002e90: 2065 2d76 616c 7565 2063 7574 2d6f 6666   e-value cut-off
-00002ea0: 2066 6f72 2073 696d 696c 6172 6974 7920   for similarity 
-00002eb0: 696e 2064 6961 6d6f 6e64 2061 6e64 2f6f  in diamond and/o
-00002ec0: 7220 686d 6d65 722c 2064 6566 6175 6c74  r hmmer, default
-00002ed0: 2031 652d 3130 0a2d 2d74 6f5f 7374 6f70   1e-10.--to_stop
-00002ee0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00002ef0: 6865 7468 6572 2074 6f20 7472 616e 736c  hether to transl
-00002f00: 6174 6520 7468 726f 7567 6820 5354 4f50  ate through STOP
-00002f10: 2063 6f64 6f6e 732c 2069 6620 7468 6520   codons, if the 
-00002f20: 666c 6167 2077 6173 2073 6574 2c20 7472  flag was set, tr
-00002f30: 616e 736c 6174 696f 6e20 7769 6c6c 2062  anslation will b
-00002f40: 6520 7465 726d 696e 6174 6564 2061 7420  e terminated at 
-00002f50: 7468 6520 6669 7273 7420 696e 2d66 7261  the first in-fra
-00002f60: 6d65 2073 746f 7020 636f 646f 6e2c 2065  me stop codon, e
-00002f70: 6c73 6520 6120 6675 6c6c 2074 7261 6e73  lse a full trans
-00002f80: 6c61 7469 6f6e 2063 6f6e 7469 6e75 696e  lation continuin
-00002f90: 6720 6f6e 2070 6173 7369 6e67 2061 6e79  g on passing any
-00002fa0: 2073 746f 7020 636f 646f 6e73 2077 6f75   stop codons wou
-00002fb0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
-00002fc0: 2d2d 6364 732c 2066 6c61 6720 6f70 7469  --cds, flag opti
-00002fd0: 6f6e 2c20 7768 6574 6865 7220 746f 206f  on, whether to o
-00002fe0: 6e6c 7920 7472 616e 736c 6174 6520 7468  nly translate th
-00002ff0: 6520 636f 6d70 6c65 7465 2043 4453 2074  e complete CDS t
-00003000: 6861 7420 7374 6172 7473 2077 6974 6820  hat starts with 
-00003010: 6120 7661 6c69 6420 7374 6172 7420 636f  a valid start co
-00003020: 646f 6e20 616e 6420 6f6e 6c79 2063 6f6e  don and only con
-00003030: 7461 696e 7320 6120 7369 6e67 6c65 2069  tains a single i
-00003040: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
-00003050: 6f6e 2061 7420 7468 6520 656e 6420 616e  on at the end an
-00003060: 6420 6d75 7374 2062 6520 6469 7669 6461  d must be divida
-00003070: 626c 6520 6279 2074 6872 6565 2c20 6966  ble by three, if
-00003080: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00003090: 742c 206f 6e6c 7920 7468 6520 636f 6d70  t, only the comp
-000030a0: 6c65 7465 2043 4453 2077 6f75 6c64 2062  lete CDS would b
-000030b0: 6520 7472 616e 736c 6174 6564 0a2d 662c  e translated.-f,
-000030c0: 202d 2d66 6f63 7573 2c20 7468 6520 7370   --focus, the sp
-000030d0: 6563 6965 7320 746f 2062 6520 6d65 7267  ecies to be merg
-000030e0: 6564 206f 6e20 6c6f 6361 6c20 4d52 4248  ed on local MRBH
-000030f0: 732c 2064 6566 6175 6c74 204e 6f6e 652c  s, default None,
-00003100: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
-00003110: 656e 2c20 7468 6520 6c6f 6361 6c20 4d52  en, the local MR
-00003120: 4248 7320 776f 756c 646e 2774 2062 6520  BHs wouldn't be 
-00003130: 696e 6665 7272 6564 0a2d 6170 2c20 2d2d  inferred.-ap, --
-00003140: 616e 6368 6f72 706f 696e 7473 2c20 7468  anchorpoints, th
-00003150: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-00003160: 6461 7461 2066 696c 6520 6672 6f6d 2069  data file from i
-00003170: 2d61 6468 6f72 6520 666f 7220 636f 6e73  -adhore for cons
-00003180: 7472 7563 7469 6e67 2074 6865 206f 7274  tructing the ort
-00003190: 686f 6772 6f75 7073 2077 6974 6820 616e  hogroups with an
-000031a0: 6368 6f72 2070 6169 7273 2c20 6465 6661  chor pairs, defa
-000031b0: 756c 7420 4e6f 6e65 0a2d 736d 2c20 2d2d  ult None.-sm, --
-000031c0: 7365 676d 656e 7473 2c20 7468 6520 7365  segments, the se
-000031d0: 676d 656e 7473 2064 6174 6166 696c 6520  gments datafile 
-000031e0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
-000031f0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
-00003200: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
-00003210: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
-00003220: 650a 2d6c 652c 202d 2d6c 6973 7465 6c65  e.-le, --listele
-00003230: 6d65 6e74 732c 2074 6865 206c 6973 7473  ments, the lists
-00003240: 6567 6d65 6e74 7320 6461 7461 2066 696c  egments data fil
-00003250: 6520 7573 6564 2069 6e20 636f 6c6c 696e  e used in collin
-00003260: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-00003270: 616e 616c 7973 6973 2069 6620 696e 6974  analysis if init
-00003280: 6961 7465 642c 2064 6566 6175 6c74 204e  iated, default N
-00003290: 6f6e 650a 2d67 742c 202d 2d67 656e 6574  one.-gt, --genet
-000032a0: 6162 6c65 2c20 7468 6520 6765 6e65 2074  able, the gene t
-000032b0: 6162 6c65 2064 6174 6166 696c 6520 7573  able datafile us
-000032c0: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
-000032d0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
-000032e0: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
-000032f0: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
-00003300: 2d63 6f63 2c20 2d2d 636f 6c6c 696e 6561  -coc, --collinea
-00003310: 7263 6f61 6c65 7363 656e 6365 2c20 666c  rcoalescence, fl
-00003320: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00003330: 6572 2074 6f20 696e 6974 6961 7465 2074  er to initiate t
-00003340: 6865 2063 6f6c 6c69 6e65 6172 2063 6f61  he collinear coa
-00003350: 6c65 7363 656e 6365 2061 6e61 6c79 7369  lescence analysi
-00003360: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00003370: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
-00003380: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
-00003390: 6974 6961 7465 640a 2d6b 662c 202d 2d6b  itiated.-kf, --k
-000033a0: 6565 7066 6173 7461 2c20 666c 6167 206f  eepfasta, flag o
-000033b0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000033c0: 6f20 6f75 7470 7574 2074 6865 2073 6571  o output the seq
-000033d0: 7565 6e63 6520 696e 666f 726d 6174 696f  uence informatio
-000033e0: 6e20 6f66 204d 5242 4873 2c20 6966 2074  n of MRBHs, if t
-000033f0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00003400: 2074 6865 2073 6571 7565 6e63 6573 206f   the sequences o
-00003410: 6620 4d52 4248 7320 776f 756c 6420 6265  f MRBHs would be
-00003420: 2069 6e20 6f75 7470 7574 0a2d 6b64 2c20   in output.-kd, 
-00003430: 2d2d 6b65 6570 6475 706c 6963 6174 6573  --keepduplicates
-00003440: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00003450: 6865 7468 6572 2074 6f20 616c 6c6f 7720  hether to allow 
-00003460: 7468 6520 7361 6d65 2067 656e 6520 746f  the same gene to
-00003470: 206f 6363 7572 2069 6e20 6469 6666 6572   occur in differ
-00003480: 656e 7420 4d52 4248 7320 286f 6e6c 7920  ent MRBHs (only 
-00003490: 6d65 616e 696e 6766 756c 2077 6865 6e20  meaningful when 
-000034a0: 7468 6520 6373 636f 7265 2077 6173 2075  the cscore was u
-000034b0: 7365 6429 2c20 6966 2074 6865 2066 6c61  sed), if the fla
-000034c0: 6720 7761 7320 7365 742c 2074 6865 2073  g was set, the s
-000034d0: 616d 6520 6765 6e65 2063 6f75 6c64 2062  ame gene could b
-000034e0: 6520 6173 7369 676e 6564 2074 6f20 6469  e assigned to di
-000034f0: 6666 6572 656e 7420 4d52 4248 730a 2d67  fferent MRBHs.-g
-00003500: 6d2c 202d 2d67 6c6f 6261 6c6d 7262 682c  m, --globalmrbh,
-00003510: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00003520: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00003530: 6520 676c 6f62 616c 204d 5242 4873 2063  e global MRBHs c
-00003540: 6f6e 7374 7275 6374 696f 6e2c 2069 6620  onstruction, if 
-00003550: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003560: 2c20 7468 6520 2d2d 666f 6375 7320 6f70  , the --focus op
-00003570: 7469 6f6e 2077 6f75 6c64 2062 6520 6967  tion would be ig
-00003580: 6e6f 7265 6420 616e 6420 6f6e 6c79 2067  nored and only g
-00003590: 6c6f 6261 6c20 4d52 4248 7320 776f 756c  lobal MRBHs woul
-000035a0: 6420 6265 2062 7569 6c74 0a2d 6e2c 202d  d be built.-n, -
-000035b0: 2d6e 7468 7265 6164 732c 2074 6865 206e  -nthreads, the n
-000035c0: 756d 6265 7220 6f66 2074 6872 6561 6473  umber of threads
-000035d0: 2074 6f20 7573 652c 2064 6566 6175 6c74   to use, default
-000035e0: 2034 0a2d 6f69 2c20 2d2d 6f72 7468 6f69   4.-oi, --orthoi
-000035f0: 6e66 6572 2c20 666c 6167 206f 7074 696f  nfer, flag optio
-00003600: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00003610: 6974 6961 7465 206f 7274 686f 6772 6f75  itiate orthogrou
-00003620: 7020 696e 6665 726e 6563 652c 2069 6620  p infernece, if 
-00003630: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003640: 2c20 7468 6520 6f72 7468 6f67 726f 7570  , the orthogroup
-00003650: 2069 6e66 6572 6e65 6365 2070 726f 6772   infernece progr
-00003660: 616d 2077 6f75 6c64 2062 6520 696e 6974  am would be init
-00003670: 6961 7465 640a 2d6f 6f2c 202d 2d6f 6e6c  iated.-oo, --onl
-00003680: 796f 7274 686f 2c20 666c 6167 206f 7074  yortho, flag opt
-00003690: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000036a0: 6f6e 6c79 2063 6f6e 6475 6374 206f 7274  only conduct ort
-000036b0: 686f 6772 6f75 7020 696e 6665 726e 6563  hogroup infernec
-000036c0: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
-000036d0: 6173 2073 6574 2c20 6f6e 6c79 2074 6865  as set, only the
-000036e0: 206f 7274 686f 6772 6f75 7020 696e 6665   orthogroup infe
-000036f0: 726e 6563 6520 7069 7065 6c69 6e65 2077  rnece pipeline w
-00003700: 6f75 6c64 2062 6520 7065 7266 6f72 6d65  ould be performe
-00003710: 6420 7768 696c 6520 7468 6520 6f74 6865  d while the othe
-00003720: 7220 616e 616c 7973 6973 2077 6f75 6c64  r analysis would
-00003730: 6e27 7420 6265 2069 6e69 7469 6174 6564  n't be initiated
-00003740: 0a2d 676e 2c20 2d2d 6765 746e 736f 672c  .-gn, --getnsog,
-00003750: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00003760: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00003770: 6520 7468 6520 7365 6172 6368 696e 6720  e the searching 
-00003780: 666f 7220 6e65 7374 6564 2073 696e 676c  for nested singl
-00003790: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
-000037a0: 6c69 6573 2028 4e53 4f47 7329 2028 6f6e  lies (NSOGs) (on
-000037b0: 6c79 206d 6561 6e69 6e67 6675 6c20 7768  ly meaningful wh
-000037c0: 656e 2074 6865 206f 7274 686f 6772 6f75  en the orthogrou
-000037d0: 7020 696e 6665 726e 6563 6520 7069 7065  p infernece pipe
-000037e0: 6c69 6e65 2077 6173 2061 6374 6976 6174  line was activat
-000037f0: 6564 292c 2069 6620 7468 6520 666c 6167  ed), if the flag
-00003800: 2077 6173 2073 6574 2c20 6164 6469 7469   was set, additi
-00003810: 6f6e 616c 204e 534f 4773 2061 6e61 6c79  onal NSOGs analy
-00003820: 7369 7320 776f 756c 6420 6265 2070 6572  sis would be per
-00003830: 666f 726d 6564 2062 6573 6964 6573 2074  formed besides t
-00003840: 6865 2062 6173 6963 206f 7274 686f 6772  he basic orthogr
-00003850: 6f75 7020 696e 6665 726e 6563 650a 2d74  oup infernece.-t
-00003860: 7265 652c 202d 2d74 7265 655f 6d65 7468  ree, --tree_meth
-00003870: 6f64 2c20 7768 6963 6820 6765 6e65 2074  od, which gene t
-00003880: 7265 6520 696e 6665 7265 6e63 6520 7072  ree inference pr
-00003890: 6f67 7261 6d20 746f 2069 6e76 6f6b 6520  ogram to invoke 
-000038a0: 286f 6e6c 7920 6d65 616e 696e 6766 756c  (only meaningful
-000038b0: 2077 6865 6e20 7468 6520 636f 6c6c 696e   when the collin
-000038c0: 6561 7220 636f 616c 6573 6365 6e63 652c  ear coalescence,
-000038d0: 2067 656e 652d 746f 2d66 616d 696c 7920   gene-to-family 
-000038e0: 6173 7369 676e 6d65 6e74 206f 7220 4e53  assignment or NS
-000038f0: 4f47 7320 616e 616c 7973 6973 2077 6572  OGs analysis wer
-00003900: 6520 6163 7469 7661 7465 6429 2c20 6465  e activated), de
-00003910: 6661 756c 7420 6661 7374 7472 6565 0a2d  fault fasttree.-
-00003920: 7473 2c20 2d2d 7472 6565 7365 742c 2074  ts, --treeset, t
-00003930: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
-00003940: 7474 696e 6720 666f 7220 6765 6e65 2074  tting for gene t
-00003950: 7265 6520 696e 6665 7265 6e63 652c 2064  ree inference, d
-00003960: 6566 6175 6c74 204e 6f6e 652c 2074 6869  efault None, thi
-00003970: 7320 6f70 7469 6f6e 2063 616e 2062 6520  s option can be 
-00003980: 7072 6f76 6964 6564 206d 756c 7469 706c  provided multipl
-00003990: 6520 7469 6d65 730a 2d6d 632c 202d 2d6d  e times.-mc, --m
-000039a0: 736f 6763 7574 2c20 7468 6520 7261 7469  sogcut, the rati
-000039b0: 6f20 6375 746f 6666 2066 6f72 206d 6f73  o cutoff for mos
-000039c0: 746c 7920 7369 6e67 6c65 2d63 6f70 7920  tly single-copy 
-000039d0: 6661 6d69 6c79 2028 6d65 616e 696e 6766  family (meaningf
-000039e0: 756c 2077 6865 6e20 6163 7469 7661 7469  ul when activati
-000039f0: 6e67 2074 6865 206f 7274 686f 6772 6f75  ng the orthogrou
-00003a00: 7020 696e 6665 726e 6563 6520 7069 7065  p infernece pipe
-00003a10: 6c69 6e65 2920 616e 6420 7370 6563 6965  line) and specie
-00003a20: 7320 7265 7072 6573 656e 7461 7469 6f6e  s representation
-00003a30: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
-00003a40: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
-00003a50: 6973 2c20 6465 6661 756c 7420 302e 382e  is, default 0.8.
-00003a60: 0a2d 6761 2c20 2d2d 6765 6e65 6173 7369  .-ga, --geneassi
-00003a70: 676e 2c20 666c 6167 206f 7074 696f 6e2c  gn, flag option,
-00003a80: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00003a90: 6961 7465 2074 6865 2067 656e 652d 746f  iate the gene-to
-00003aa0: 2d66 616d 696c 7920 6173 7369 676e 6d65  -family assignme
-00003ab0: 6e74 2061 6e61 6c79 7369 732c 2069 6620  nt analysis, if 
-00003ac0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003ad0: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
-00003ae0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-00003af0: 640a 2d73 612c 202d 2d73 6571 3261 7373  d.-sa, --seq2ass
-00003b00: 6967 6e2c 2074 6865 2071 7565 7269 6564  ign, the queried
-00003b10: 2073 6571 7565 6e63 6573 2064 6174 6120   sequences data 
-00003b20: 6669 6c65 2069 6e20 6765 6e65 2d74 6f2d  file in gene-to-
-00003b30: 6661 6d69 6c79 2061 7373 6967 6e6d 656e  family assignmen
-00003b40: 7420 616e 616c 7973 6973 2c20 6465 6661  t analysis, defa
-00003b50: 756c 7420 4e6f 6e65 2c20 7468 6973 206f  ult None, this o
-00003b60: 7074 696f 6e20 6361 6e20 6265 2070 726f  ption can be pro
-00003b70: 7669 6465 6420 6d75 6c74 6970 6c65 2074  vided multiple t
-00003b80: 696d 6573 0a2d 6661 2c20 2d2d 6661 6d32  imes.-fa, --fam2
-00003b90: 6173 7369 676e 2c20 7468 6520 7175 6572  assign, the quer
-00003ba0: 6965 6420 6661 6d69 6c69 7920 6461 7461  ied familiy data
-00003bb0: 2066 696c 6520 696e 2067 656e 652d 746f   file in gene-to
-00003bc0: 2d66 616d 696c 7920 6173 7369 676e 6d65  -family assignme
-00003bd0: 6e74 2061 6e61 6c79 7369 732c 2064 6566  nt analysis, def
-00003be0: 6175 6c74 204e 6f6e 650a 2d63 632c 202d  ault None.-cc, -
-00003bf0: 2d63 6f6e 6361 742c 2066 6c61 6720 6f70  -concat, flag op
-00003c00: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00003c10: 2069 6e69 7469 6174 6520 7468 6520 636f   initiate the co
-00003c20: 6e63 6174 656e 6174 696f 6e20 7069 7065  ncatenation pipe
-00003c30: 6c69 6e65 2066 6f72 206f 7274 686f 6772  line for orthogr
-00003c40: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
-00003c50: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003c60: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-00003c70: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00003c80: 7465 640a 2d74 652c 202d 2d74 6573 7473  ted.-te, --tests
-00003c90: 6f67 2c20 666c 6167 206f 7074 696f 6e2c  og, flag option,
-00003ca0: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00003cb0: 6961 7465 2074 6865 2075 6e62 6961 7365  iate the unbiase
-00003cc0: 6420 7465 7374 206f 6620 7369 6e67 6c65  d test of single
-00003cd0: 2d63 6f70 7920 6765 6e65 2066 616d 696c  -copy gene famil
-00003ce0: 6965 732c 2069 6620 7468 6520 666c 6167  ies, if the flag
-00003cf0: 2077 6173 2073 6574 2c20 7468 6520 616e   was set, the an
-00003d00: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
-00003d10: 696e 6974 6961 7465 640a 2d62 732c 202d  initiated.-bs, -
-00003d20: 2d62 696e 732c 2074 6865 206e 756d 6265  -bins, the numbe
-00003d30: 7220 6f66 2062 696e 7320 6469 7669 6465  r of bins divide
-00003d40: 6420 696e 2074 6865 2067 656e 6520 6c65  d in the gene le
-00003d50: 6e67 7468 206e 6f72 6d61 6c69 7a61 7469  ngth normalizati
-00003d60: 6f6e 2c20 6465 6661 756c 7420 3130 300a  on, default 100.
-00003d70: 2d6e 702c 202d 2d6e 6f72 6d61 6c69 7a65  -np, --normalize
-00003d80: 6470 6572 6365 6e74 2c20 7468 6520 7065  dpercent, the pe
-00003d90: 7263 656e 7461 6765 206f 6620 7570 7065  rcentage of uppe
-00003da0: 7220 6869 7473 2075 7365 6420 666f 7220  r hits used for 
-00003db0: 6765 6e65 206c 656e 6774 6820 6e6f 726d  gene length norm
-00003dc0: 616c 697a 6174 696f 6e2c 2064 6566 6175  alization, defau
-00003dd0: 6c74 2035 0a2d 6e6e 2c20 2d2d 6e6f 6e6f  lt 5.-nn, --nono
-00003de0: 726d 616c 697a 6174 696f 6e2c 2066 6c61  rmalization, fla
-00003df0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00003e00: 7220 746f 2063 616c 6c20 6f66 6620 7468  r to call off th
-00003e10: 6520 6e6f 726d 616c 697a 6174 696f 6e2c  e normalization,
-00003e20: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003e30: 2073 6574 2c20 6e6f 206e 6f72 6d61 6c69   set, no normali
-00003e40: 7a61 7469 6f6e 2077 6f75 6c64 2062 6520  zation would be 
-00003e50: 636f 6e64 7563 7465 640a 2d62 736f 672c  conducted.-bsog,
-00003e60: 202d 2d62 7573 636f 736f 672c 2066 6c61   --buscosog, fla
-00003e70: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00003e80: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
-00003e90: 6520 6275 7363 6f2d 6775 6964 6564 2073  e busco-guided s
-00003ea0: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
-00003eb0: 6661 6d69 6c79 2061 6e61 6c79 7369 732c  family analysis,
-00003ec0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003ed0: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
-00003ee0: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
-00003ef0: 6961 7465 640a 2d62 686d 6d2c 202d 2d62  iated.-bhmm, --b
-00003f00: 7573 636f 686d 6d2c 2074 6865 2048 4d4d  uscohmm, the HMM
-00003f10: 2070 726f 6669 6c65 2064 6174 6166 696c   profile datafil
-00003f20: 6520 696e 2074 6865 2062 7573 636f 2d67  e in the busco-g
-00003f30: 7569 6465 6420 7369 6e67 6c65 2d63 6f70  uided single-cop
-00003f40: 7920 6765 6e65 2066 616d 696c 7920 616e  y gene family an
-00003f50: 616c 7973 6973 2c20 6465 6661 756c 7420  alysis, default 
-00003f60: 4e6f 6e65 0a2d 6263 7466 2c20 2d2d 6275  None.-bctf, --bu
-00003f70: 7363 6f63 7574 6f66 662c 2074 6865 2048  scocutoff, the H
-00003f80: 4d4d 2073 636f 7265 2063 7574 6f66 6620  MM score cutoff 
-00003f90: 6461 7461 6669 6c65 2069 6e20 7468 6520  datafile in the 
-00003fa0: 6275 7363 6f2d 6775 6964 6564 2073 696e  busco-guided sin
-00003fb0: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
-00003fc0: 6d69 6c79 2061 6e61 6c79 7369 732c 2064  mily analysis, d
-00003fd0: 6566 6175 6c74 204e 6f6e 650a 2d6f 6620  efault None.-of 
-00003fe0: 2c2d 2d6f 6766 6f72 6d61 742c 2066 6c61  ,--ogformat, fla
-00003ff0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004000: 7220 746f 2061 6464 2069 6e64 6578 2074  r to add index t
-00004010: 6f20 7468 6520 5242 4820 6661 6d69 6c69  o the RBH famili
-00004020: 6573 0a60 6060 0a0a 5468 6520 7072 6f67  es.```..The prog
-00004030: 7261 6d20 6077 6764 2066 6f63 7573 6020  ram `wgd focus` 
-00004040: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
-00004050: 636f 6e63 6174 656e 6174 696f 6e2d 6261  concatenation-ba
-00004060: 7365 6420 616e 6420 636f 616c 6573 6365  sed and coalesce
-00004070: 6e63 652d 6261 7365 6420 7068 796c 6f67  nce-based phylog
-00004080: 656e 6574 6963 2069 6e66 6572 656e 6365  enetic inference
-00004090: 2061 6e64 2070 6879 6c6f 6765 6e65 7469   and phylogeneti
-000040a0: 6320 6461 7469 6e67 206f 6620 5747 4473  c dating of WGDs
-000040b0: 2065 7463 2e0a 6060 600a 7767 6420 666f   etc..```.wgd fo
-000040c0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
-000040d0: 7565 6e63 6573 2028 6f70 7469 6f6e 290a  uences (option).
-000040e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000040f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004100: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004110: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004120: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004130: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00004140: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00004150: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00004160: 5f66 6f63 7573 0a2d 742c 202d 2d74 6d70  _focus.-t, --tmp
-00004170: 6469 722c 2074 6865 2074 656d 706f 7261  dir, the tempora
-00004180: 7279 2077 6f72 6b69 6e67 2064 6972 6563  ry working direc
-00004190: 746f 7279 2c20 6465 6661 756c 7420 4e6f  tory, default No
-000041a0: 6e65 2c20 6966 204e 6f6e 6520 7761 7320  ne, if None was 
-000041b0: 6769 7665 6e2c 2074 6865 2074 6d70 6469  given, the tmpdi
-000041c0: 7220 7769 6c6c 2062 6520 6173 7369 676e  r will be assign
-000041d0: 6564 2072 616e 646f 6d20 6e61 6d65 7320  ed random names 
-000041e0: 696e 2063 7572 7265 6e74 2064 6972 6563  in current direc
-000041f0: 746f 7279 2061 6e64 2061 7574 6f6d 6174  tory and automat
-00004200: 656c 7920 7265 6d6f 7665 6420 6174 2074  ely removed at t
-00004210: 6865 2063 6f6d 706c 6574 696f 6e20 6f66  he completion of
-00004220: 2070 726f 6772 616d 2c20 656c 7365 2074   program, else t
-00004230: 6865 2074 6d70 6469 7220 776f 756c 6420  he tmpdir would 
-00004240: 6265 206b 6570 740a 2d6e 2c20 2d2d 6e74  be kept.-n, --nt
-00004250: 6872 6561 6473 2c20 7468 6520 6e75 6d62  hreads, the numb
-00004260: 6572 206f 6620 7468 7265 6164 7320 746f  er of threads to
-00004270: 2075 7365 2c20 6465 6661 756c 7420 340a   use, default 4.
-00004280: 2d2d 746f 5f73 746f 702c 2066 6c61 6720  --to_stop, flag 
-00004290: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000042a0: 746f 2074 7261 6e73 6c61 7465 2074 6872  to translate thr
-000042b0: 6f75 6768 2053 544f 5020 636f 646f 6e73  ough STOP codons
-000042c0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-000042d0: 7320 7365 742c 2074 7261 6e73 6c61 7469  s set, translati
-000042e0: 6f6e 2077 696c 6c20 6265 2074 6572 6d69  on will be termi
-000042f0: 6e61 7465 6420 6174 2074 6865 2066 6972  nated at the fir
-00004300: 7374 2069 6e2d 6672 616d 6520 7374 6f70  st in-frame stop
-00004310: 2063 6f64 6f6e 2c20 656c 7365 2061 2066   codon, else a f
-00004320: 756c 6c20 7472 616e 736c 6174 696f 6e20  ull translation 
-00004330: 636f 6e74 696e 7569 6e67 206f 6e20 7061  continuing on pa
-00004340: 7374 2061 6e79 2073 746f 7020 636f 646f  st any stop codo
-00004350: 6e73 2077 6f75 6c64 2062 6520 696e 6974  ns would be init
-00004360: 6961 7465 640a 2d2d 6364 732c 2066 6c61  iated.--cds, fla
-00004370: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004380: 7220 746f 206f 6e6c 7920 7472 616e 736c  r to only transl
-00004390: 6174 6520 7468 6520 636f 6d70 6c65 7465  ate the complete
-000043a0: 2043 4453 2074 6861 7420 7374 6172 7473   CDS that starts
-000043b0: 2077 6974 6820 6120 7661 6c69 6420 7374   with a valid st
-000043c0: 6172 7420 636f 646f 6e20 616e 6420 6f6e  art codon and on
-000043d0: 6c79 2063 6f6e 7461 696e 7320 6120 7369  ly contains a si
-000043e0: 6e67 6c65 2069 6e2d 6672 616d 6520 7374  ngle in-frame st
-000043f0: 6f70 2063 6f64 6f6e 2061 7420 7468 6520  op codon at the 
-00004400: 656e 6420 616e 6420 6d75 7374 2062 6520  end and must be 
-00004410: 6469 7669 6461 626c 6520 6279 2074 6872  dividable by thr
-00004420: 6565 2c20 6966 2074 6865 2066 6c61 6720  ee, if the flag 
-00004430: 7761 7320 7365 742c 206f 6e6c 7920 7468  was set, only th
-00004440: 6520 636f 6d70 6c65 7465 2043 4453 2077  e complete CDS w
-00004450: 6f75 6c64 2062 6520 7472 616e 736c 6174  ould be translat
-00004460: 6564 0a2d 2d73 7472 6970 5f67 6170 732c  ed.--strip_gaps,
-00004470: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004480: 6574 6865 7220 746f 2064 726f 7020 616c  ether to drop al
-00004490: 6c20 6761 7073 2069 6e20 6d75 6c74 6970  l gaps in multip
-000044a0: 6c65 2073 6571 7565 6e63 6520 616c 6967  le sequence alig
-000044b0: 6e6d 656e 742c 2069 6620 7468 6520 666c  nment, if the fl
-000044c0: 6167 2077 6173 2073 6574 2c20 616c 6c20  ag was set, all 
-000044d0: 6761 7073 2077 6f75 6c64 2062 6520 6472  gaps would be dr
-000044e0: 6f70 7065 640a 2d61 2c20 2d2d 616c 6967  opped.-a, --alig
-000044f0: 6e65 722c 2077 6869 6368 2061 6c69 676e  ner, which align
-00004500: 6d65 6e74 2070 726f 6772 616d 2074 6f20  ment program to 
-00004510: 7573 652c 2064 6566 6175 6c74 206d 6166  use, default maf
-00004520: 6674 0a2d 7472 6565 2c20 2d2d 7472 6565  ft.-tree, --tree
-00004530: 5f6d 6574 686f 642c 2077 6869 6368 2067  _method, which g
-00004540: 656e 6520 7472 6565 2069 6e66 6572 656e  ene tree inferen
-00004550: 6365 2070 726f 6772 616d 2074 6f20 696e  ce program to in
-00004560: 766f 6b65 2c20 6465 6661 756c 7420 6661  voke, default fa
-00004570: 7374 7472 6565 0a2d 7473 2c20 2d2d 7472  sttree.-ts, --tr
-00004580: 6565 7365 742c 2074 6865 2070 6172 616d  eeset, the param
-00004590: 6574 6572 7320 7365 7474 696e 6720 666f  eters setting fo
-000045a0: 7220 6765 6e65 2074 7265 6520 696e 6665  r gene tree infe
-000045b0: 7265 6e63 652c 2064 6566 6175 6c74 204e  rence, default N
-000045c0: 6f6e 652c 2074 6869 7320 6f70 7469 6f6e  one, this option
-000045d0: 2063 616e 2062 6520 7072 6f76 6964 6564   can be provided
-000045e0: 206d 756c 7469 706c 6520 7469 6d65 730a   multiple times.
-000045f0: 2d2d 636f 6e63 6174 656e 6174 696f 6e2c  --concatenation,
-00004600: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004610: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00004620: 6520 7468 6520 636f 6e63 6174 656e 6174  e the concatenat
-00004630: 696f 6e2d 6261 7365 6420 7370 6563 6965  ion-based specie
-00004640: 7320 7472 6565 2069 6e66 6572 656e 6365  s tree inference
-00004650: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00004660: 7320 7365 742c 2063 6f6e 6361 7465 6e61  s set, concatena
-00004670: 7469 6f6e 2d62 6173 6564 2073 7065 6369  tion-based speci
-00004680: 6573 2074 7265 6520 776f 756c 6420 6265  es tree would be
-00004690: 2069 6e66 6572 6564 0a2d 2d63 6f61 6c65   infered.--coale
-000046a0: 7363 656e 6365 2c20 666c 6167 206f 7074  scence, flag opt
-000046b0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000046c0: 696e 6974 6961 7465 2074 6865 2063 6f61  initiate the coa
-000046d0: 6c65 7363 656e 6365 2d62 6173 6564 2073  lescence-based s
-000046e0: 7065 6369 6573 2074 7265 6520 696e 6665  pecies tree infe
-000046f0: 7265 6e63 652c 2069 6620 7468 6520 666c  rence, if the fl
-00004700: 6167 2077 6173 2073 6574 2c20 636f 616c  ag was set, coal
-00004710: 6573 6365 6e63 652d 6261 7365 6420 7370  escence-based sp
-00004720: 6563 6965 7320 7472 6565 2077 6f75 6c64  ecies tree would
-00004730: 2062 6520 696e 6665 7265 640a 2d73 702c   be infered.-sp,
-00004740: 202d 2d73 7065 6369 6573 7472 6565 2c20   --speciestree, 
-00004750: 7370 6563 6965 7320 7472 6565 2064 6174  species tree dat
-00004760: 6166 696c 6520 666f 7220 6461 7469 6e67  afile for dating
-00004770: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00004780: 642c 202d 2d64 6174 696e 672c 2077 6869  d, --dating, whi
-00004790: 6368 206d 6f6c 6563 756c 6172 2064 6174  ch molecular dat
-000047a0: 696e 6720 7072 6f67 7261 6d20 746f 2075  ing program to u
-000047b0: 7365 2c20 6465 6661 756c 7420 6e6f 6e65  se, default none
-000047c0: 0a2d 6473 2c20 2d2d 6461 7469 6e67 7365  .-ds, --datingse
-000047d0: 742c 2074 6865 2070 6172 616d 6574 6572  t, the parameter
-000047e0: 7320 7365 7474 696e 6720 666f 7220 6461  s setting for da
-000047f0: 7469 6e67 2070 726f 6772 616d 2c20 6465  ting program, de
-00004800: 6661 756c 7420 4e6f 6e65 2c20 7468 6973  fault None, this
-00004810: 206f 7074 696f 6e20 6361 6e20 6265 2070   option can be p
-00004820: 726f 7669 6465 6420 6d75 6c74 6970 6c65  rovided multiple
-00004830: 2074 696d 6573 0a2d 6e73 2c20 2d2d 6e73   times.-ns, --ns
-00004840: 6974 6573 2c20 7468 6520 6e73 6974 6573  ites, the nsites
-00004850: 2069 6e66 6f72 6d61 7469 6f6e 2066 6f72   information for
-00004860: 2072 3873 2064 6174 696e 672c 2064 6566   r8s dating, def
-00004870: 6175 6c74 204e 6f6e 650a 2d6f 742c 202d  ault None.-ot, -
-00004880: 2d6f 7574 6772 6f75 702c 2074 6865 206f  -outgroup, the o
-00004890: 7574 6772 6f75 7020 696e 666f 726d 6174  utgroup informat
-000048a0: 696f 6e20 666f 7220 7238 7320 6461 7469  ion for r8s dati
-000048b0: 6e67 2c20 6465 6661 756c 7420 4e6f 6e65  ng, default None
-000048c0: 0a2d 7074 2c20 2d2d 7061 7274 6974 696f  .-pt, --partitio
-000048d0: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
-000048e0: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-000048f0: 6174 6520 7061 7274 6974 696f 6e20 6461  ate partition da
-00004900: 7469 6e67 2061 6e61 6c79 7369 7320 666f  ting analysis fo
-00004910: 7220 636f 646f 6e2c 2069 6620 7468 6520  r codon, if the 
-00004920: 666c 6167 2077 6173 2073 6574 2c20 616e  flag was set, an
-00004930: 2061 6464 6974 696f 6e61 6c20 7061 7274   additional part
-00004940: 6974 696f 6e20 6461 7469 6e67 2061 6e61  ition dating ana
-00004950: 6c79 7369 7320 776f 756c 6420 6265 2069  lysis would be i
-00004960: 6e69 7469 6174 6564 0a2d 616d 2c20 2d2d  nitiated.-am, --
-00004970: 6161 6d6f 6465 6c2c 2077 6869 6368 2070  aamodel, which p
-00004980: 726f 7465 696e 206d 6f64 656c 2074 6f20  rotein model to 
-00004990: 6265 2075 7365 6420 696e 206d 636d 6374  be used in mcmct
-000049a0: 7265 652c 2064 6566 6175 6c74 2070 6f69  ree, default poi
-000049b0: 7373 6f6e 0a2d 6b73 2c20 666c 6167 206f  sson.-ks, flag o
-000049c0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000049d0: 6f20 696e 6974 6961 7465 204b 7320 6361  o initiate Ks ca
-000049e0: 6c63 756c 6174 696f 6e20 666f 7220 686f  lculation for ho
-000049f0: 6d6f 6c6f 6775 6573 2069 6e20 7468 6520  mologues in the 
-00004a00: 7072 6f76 6964 6564 206f 7274 686f 6c6f  provided ortholo
-00004a10: 676f 7573 2067 656e 6520 6661 6d69 6c79  gous gene family
-00004a20: 0a2d 2d61 6e6e 6f74 6174 696f 6e2c 2077  .--annotation, w
-00004a30: 6869 6368 2061 6e6e 6f74 6174 696f 6e20  hich annotation 
-00004a40: 7072 6f67 7261 6d20 746f 2075 7365 2c20  program to use, 
-00004a50: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d70  default None.--p
-00004a60: 6169 7277 6973 652c 2066 6c61 6720 6f70  airwise, flag op
-00004a70: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00004a80: 2069 6e69 7469 6174 6520 7061 6972 7769   initiate pairwi
-00004a90: 7365 204b 7320 6573 7469 6d61 7469 6f6e  se Ks estimation
-00004aa0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00004ab0: 7320 7365 742c 2070 6169 7277 6973 6520  s set, pairwise 
-00004ac0: 4b73 2076 616c 7565 7320 776f 756c 6420  Ks values would 
-00004ad0: 6265 2065 7374 696d 6174 6564 0a2d 6564  be estimated.-ed
-00004ae0: 2c20 2d2d 6567 676e 6f67 6461 7461 2c20  , --eggnogdata, 
-00004af0: 7468 6520 6567 676e 6f67 2061 6e6e 6f74  the eggnog annot
-00004b00: 6174 696f 6e20 6461 7461 6669 6c65 2c20  ation datafile, 
-00004b10: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d70  default None.--p
-00004b20: 6661 6d2c 2077 6869 6368 206f 7074 696f  fam, which optio
-00004b30: 6e20 746f 2075 7365 2066 6f72 2070 6661  n to use for pfa
-00004b40: 6d20 616e 6e6f 7461 7469 6f6e 2c20 6465  m annotation, de
-00004b50: 6661 756c 7420 4e6f 6e65 0a2d 2d64 6d6e  fault None.--dmn
-00004b60: 622c 2074 6865 2064 6961 6d6f 6e64 2064  b, the diamond d
-00004b70: 6174 6162 6173 6520 666f 7220 616e 6e6f  atabase for anno
-00004b80: 7461 7469 6f6e 2c20 6465 6661 756c 7420  tation, default 
-00004b90: 4e6f 6e65 0a2d 2d68 6d6d 2c20 7468 6520  None.--hmm, the 
-00004ba0: 484d 4d20 7072 6f66 696c 6520 666f 7220  HMM profile for 
-00004bb0: 616e 6e6f 7461 7469 6f6e 2c20 6465 6661  annotation, defa
-00004bc0: 756c 7420 4e6f 6e65 0a2d 2d65 7661 6c75  ult None.--evalu
-00004bd0: 652c 2074 6865 2065 2d76 616c 7565 2063  e, the e-value c
-00004be0: 7574 2d6f 6666 2066 6f72 2061 6e6e 6f74  ut-off for annot
-00004bf0: 6174 696f 6e2c 2064 6566 6175 6c74 2031  ation, default 1
-00004c00: 652d 3130 0a2d 2d65 7865 7061 7468 2c20  e-10.--exepath, 
-00004c10: 7468 6520 7061 7468 2074 6f20 7468 6520  the path to the 
-00004c20: 696e 7465 7270 726f 7363 616e 2065 7865  interproscan exe
-00004c30: 6375 7461 626c 652c 2064 6566 6175 6c74  cutable, default
-00004c40: 204e 6f6e 650a 2d66 2c20 2d2d 666f 7373   None.-f, --foss
-00004c50: 696c 2c20 7468 6520 666f 7373 696c 2063  il, the fossil c
-00004c60: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-00004c70: 6d61 7469 6f6e 2069 6e20 4265 6173 742c  mation in Beast,
-00004c80: 2064 6566 6175 6c74 2028 2763 6c61 6465   default ('clade
-00004c90: 313b 636c 6164 6532 272c 2027 7461 7861  1;clade2', 'taxa
-00004ca0: 312c 7461 7861 323b 7461 7861 332c 7461  1,taxa2;taxa3,ta
-00004cb0: 7861 3427 2c20 2734 3b35 272c 2027 302e  xa4', '4;5', '0.
-00004cc0: 353b 302e 3627 2c20 2734 3030 3b35 3030  5;0.6', '400;500
-00004cd0: 2729 0a2d 7268 2c20 2d2d 726f 6f74 6865  ').-rh, --roothe
-00004ce0: 6967 6874 2c20 7468 6520 726f 6f74 2068  ight, the root h
-00004cf0: 6569 6768 7420 6361 6c69 6272 6174 696f  eight calibratio
-00004d00: 6e20 696e 666f 2069 6e20 4265 6173 742c  n info in Beast,
-00004d10: 2064 6566 6175 6c74 2028 342c 302e 352c   default (4,0.5,
-00004d20: 3430 3029 0a2d 6373 2c20 2d2d 6368 6169  400).-cs, --chai
-00004d30: 6e73 6574 2c20 7468 6520 7061 7261 6d65  nset, the parame
-00004d40: 7465 7273 206f 6620 4d43 4d43 2063 6861  ters of MCMC cha
-00004d50: 696e 2069 6e20 4265 6173 742c 2064 6566  in in Beast, def
-00004d60: 6175 6c74 2028 3130 3030 302c 3130 3029  ault (10000,100)
-00004d70: 0a2d 2d62 6561 7374 6c67 6a61 722c 2074  .--beastlgjar, t
-00004d80: 6865 2070 6174 6820 746f 2062 6561 7374  he path to beast
-00004d90: 4c47 2e6a 6172 2c20 6465 6661 756c 7420  LG.jar, default 
-00004da0: 4e6f 6e65 0a2d 2d62 6561 676c 652c 2066  None.--beagle, f
-00004db0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004dc0: 6865 7220 746f 2075 7365 2062 6561 676c  her to use beagl
-00004dd0: 6520 696e 2042 6561 7374 2c20 6966 2074  e in Beast, if t
-00004de0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00004df0: 2062 6561 676c 6520 776f 756c 6420 6265   beagle would be
-00004e00: 2075 7365 640a 2d2d 7072 6f74 636f 6364   used.--protcocd
-00004e10: 6174 696e 672c 2066 6c61 6720 6f70 7469  ating, flag opti
-00004e20: 6f6e 2c20 7768 6574 6865 7220 746f 206f  on, whether to o
-00004e30: 6e6c 7920 696e 6974 6961 7465 2074 6865  nly initiate the
-00004e40: 2070 726f 7465 696e 2d63 6f6e 6361 7465   protein-concate
-00004e50: 6e61 7469 6f6e 2d62 6173 6564 2064 6174  nation-based dat
-00004e60: 696e 6720 616e 616c 7973 6973 2c20 6966  ing analysis, if
-00004e70: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00004e80: 742c 2074 6865 2061 6e61 6c79 7369 7320  t, the analysis 
-00004e90: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
-00004ea0: 6564 0a2d 2d70 726f 7464 6174 696e 672c  ed.--protdating,
-00004eb0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004ec0: 6574 6865 7220 746f 206f 6e6c 7920 696e  ether to only in
-00004ed0: 6974 6961 7465 2074 6865 2070 726f 7465  itiate the prote
-00004ee0: 696e 2d62 6173 6564 2064 6174 696e 6720  in-based dating 
-00004ef0: 616e 616c 7973 6973 2c20 6966 2074 6865  analysis, if the
-00004f00: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00004f10: 6865 2061 6e61 6c79 7369 7320 776f 756c  he analysis woul
-00004f20: 6420 6265 2069 6e69 7469 6174 6564 0a60  d be initiated.`
-00004f30: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
-00004f40: 6077 6764 206b 7364 6020 6361 6e20 7265  `wgd ksd` can re
-00004f50: 616c 697a 6520 7468 6520 636f 6e73 7472  alize the constr
-00004f60: 7563 7469 6f6e 206f 6620 2a4b 2a3c 7375  uction of *K*<su
-00004f70: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00004f80: 7374 7269 6275 7469 6f6e 2061 6e64 2072  stribution and r
-00004f90: 6174 6520 636f 7272 6563 7469 6f6e 2e0a  ate correction..
-00004fa0: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
-00004fb0: 6c69 6573 2073 6571 7565 6e63 6573 2028  lies sequences (
-00004fc0: 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d  option).--------
-00004fd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004fe0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004ff0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005000: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005010: 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c 202d 2d6f  --------.-o, --o
-00005020: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
-00005030: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
-00005040: 6175 6c74 2077 6764 5f6b 7364 0a2d 742c  ault wgd_ksd.-t,
-00005050: 202d 2d74 6d70 6469 722c 2074 6865 2074   --tmpdir, the t
-00005060: 656d 706f 7261 7279 2077 6f72 6b69 6e67  emporary working
-00005070: 2064 6972 6563 746f 7279 2c20 6465 6661   directory, defa
-00005080: 756c 7420 4e6f 6e65 2c20 6966 204e 6f6e  ult None, if Non
-00005090: 6520 7761 7320 6769 7665 6e2c 2074 6865  e was given, the
-000050a0: 2074 6d70 6469 7220 7769 6c6c 2062 6520   tmpdir will be 
-000050b0: 6173 7369 676e 6564 2072 616e 646f 6d20  assigned random 
-000050c0: 6e61 6d65 7320 696e 2063 7572 7265 6e74  names in current
-000050d0: 2064 6972 6563 746f 7279 2061 6e64 2061   directory and a
-000050e0: 7574 6f6d 6174 656c 7920 7265 6d6f 7665  utomately remove
-000050f0: 6420 6174 2074 6865 2063 6f6d 706c 6574  d at the complet
-00005100: 696f 6e20 6f66 2070 726f 6772 616d 2c20  ion of program, 
-00005110: 656c 7365 2074 6865 2074 6d70 6469 7220  else the tmpdir 
-00005120: 776f 756c 6420 6265 206b 6570 740a 2d6e  would be kept.-n
-00005130: 2c20 2d2d 6e74 6872 6561 6473 2c20 7468  , --nthreads, th
-00005140: 6520 6e75 6d62 6572 206f 6620 7468 7265  e number of thre
-00005150: 6164 7320 746f 2075 7365 2c20 6465 6661  ads to use, defa
-00005160: 756c 7420 340a 2d2d 746f 5f73 746f 702c  ult 4.--to_stop,
-00005170: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005180: 6574 6865 7220 746f 2074 7261 6e73 6c61  ether to transla
-00005190: 7465 2074 6872 6f75 6768 2053 544f 5020  te through STOP 
-000051a0: 636f 646f 6e73 2c20 6966 2074 6865 2066  codons, if the f
-000051b0: 6c61 6720 7761 7320 7365 742c 2074 7261  lag was set, tra
-000051c0: 6e73 6c61 7469 6f6e 2077 696c 6c20 6265  nslation will be
-000051d0: 2074 6572 6d69 6e61 7465 6420 6174 2074   terminated at t
-000051e0: 6865 2066 6972 7374 2069 6e2d 6672 616d  he first in-fram
-000051f0: 6520 7374 6f70 2063 6f64 6f6e 2c20 656c  e stop codon, el
-00005200: 7365 2061 2066 756c 6c20 7472 616e 736c  se a full transl
-00005210: 6174 696f 6e20 636f 6e74 696e 7569 6e67  ation continuing
-00005220: 206f 6e20 7061 7374 2061 6e79 2073 746f   on past any sto
-00005230: 7020 636f 646f 6e73 2077 6f75 6c64 2062  p codons would b
-00005240: 6520 696e 6974 6961 7465 640a 2d2d 6364  e initiated.--cd
-00005250: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
-00005260: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
-00005270: 7472 616e 736c 6174 6520 7468 6520 636f  translate the co
-00005280: 6d70 6c65 7465 2043 4453 2074 6861 7420  mplete CDS that 
-00005290: 7374 6172 7473 2077 6974 6820 6120 7661  starts with a va
-000052a0: 6c69 6420 7374 6172 7420 636f 646f 6e20  lid start codon 
-000052b0: 616e 6420 6f6e 6c79 2063 6f6e 7461 696e  and only contain
-000052c0: 7320 6120 7369 6e67 6c65 2069 6e2d 6672  s a single in-fr
-000052d0: 616d 6520 7374 6f70 2063 6f64 6f6e 2061  ame stop codon a
-000052e0: 7420 7468 6520 656e 6420 616e 6420 6d75  t the end and mu
-000052f0: 7374 2062 6520 6469 7669 6461 626c 6520  st be dividable 
-00005300: 6279 2074 6872 6565 2c20 6966 2074 6865  by three, if the
-00005310: 2066 6c61 6720 7761 7320 7365 742c 206f   flag was set, o
-00005320: 6e6c 7920 7468 6520 636f 6d70 6c65 7465  nly the complete
-00005330: 2043 4453 2077 6f75 6c64 2062 6520 7472   CDS would be tr
-00005340: 616e 736c 6174 6564 0a2d 2d70 6169 7277  anslated.--pairw
-00005350: 6973 652c 2066 6c61 6720 6f70 7469 6f6e  ise, flag option
-00005360: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
-00005370: 7469 6174 6520 7061 6972 7769 7365 204b  tiate pairwise K
-00005380: 7320 6573 7469 6d61 7469 6f6e 2c20 6966  s estimation, if
-00005390: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-000053a0: 742c 2070 6169 7277 6973 6520 4b73 2076  t, pairwise Ks v
-000053b0: 616c 7565 7320 776f 756c 6420 6265 2065  alues would be e
-000053c0: 7374 696d 6174 6564 0a2d 2d73 7472 6970  stimated.--strip
-000053d0: 5f67 6170 732c 2066 6c61 6720 6f70 7469  _gaps, flag opti
-000053e0: 6f6e 2c20 7768 6574 6865 7220 746f 2064  on, whether to d
-000053f0: 726f 7020 616c 6c20 6761 7073 2069 6e20  rop all gaps in 
-00005400: 6d75 6c74 6970 6c65 2073 6571 7565 6e63  multiple sequenc
-00005410: 6520 616c 6967 6e6d 656e 742c 2069 6620  e alignment, if 
-00005420: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00005430: 2c20 616c 6c20 6761 7073 2077 6f75 6c64  , all gaps would
-00005440: 2062 6520 6472 6f70 7065 640a 2d61 2c20   be dropped.-a, 
-00005450: 2d2d 616c 6967 6e65 722c 2077 6869 6368  --aligner, which
-00005460: 2061 6c69 676e 6d65 6e74 2070 726f 6772   alignment progr
-00005470: 616d 2074 6f20 7573 652c 2064 6566 6175  am to use, defau
-00005480: 6c74 206d 6166 6674 200a 2d74 7265 652c  lt mafft .-tree,
-00005490: 202d 2d74 7265 655f 6d65 7468 6f64 2c20   --tree_method, 
-000054a0: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
-000054b0: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-000054c0: 6d20 746f 2069 6e76 6f6b 652c 2064 6566  m to invoke, def
-000054d0: 6175 6c74 2066 6173 7474 7265 650a 2d2d  ault fasttree.--
-000054e0: 7472 6565 5f6f 7074 696f 6e73 2c20 6f70  tree_options, op
-000054f0: 7469 6f6e 7320 696e 2074 7265 6520 696e  tions in tree in
-00005500: 6665 7265 6e63 6520 6173 2061 2063 6f6d  ference as a com
-00005510: 6d61 2073 6570 6172 6174 6564 2073 7472  ma separated str
-00005520: 696e 672c 2064 6566 6175 6c74 204e 6f6e  ing, default Non
-00005530: 650a 2d2d 6e6f 6465 5f61 7665 7261 6765  e.--node_average
-00005540: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00005550: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
-00005560: 7465 206e 6f64 652d 6176 6572 6167 6520  te node-average 
-00005570: 7761 7920 6f66 2064 652d 7265 6475 6e64  way of de-redund
-00005580: 616e 6379 2069 6e73 7465 6164 206f 6620  ancy instead of 
-00005590: 6e6f 6465 2d77 6569 6768 7465 642c 2069  node-weighted, i
-000055a0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-000055b0: 6574 2c20 7468 6520 6e6f 6465 2d61 7665  et, the node-ave
-000055c0: 7261 6769 6e67 2064 652d 7265 6475 6e64  raging de-redund
-000055d0: 616e 6379 2077 6f75 6c64 2062 6520 696e  ancy would be in
-000055e0: 6974 6961 7465 640a 2d73 722c 202d 2d73  itiated.-sr, --s
-000055f0: 7061 6972 2c20 7468 6520 7370 6563 6965  pair, the specie
-00005600: 7320 7061 6972 2074 6f20 6265 2070 6c6f  s pair to be plo
-00005610: 7474 6564 2c20 6465 6661 756c 7420 4e6f  tted, default No
-00005620: 6e65 2c20 7468 6973 206f 7074 696f 6e20  ne, this option 
-00005630: 6361 6e20 6265 2070 726f 7669 6465 6420  can be provided 
-00005640: 6d75 6c74 6970 6c65 2074 696d 6573 0a2d  multiple times.-
-00005650: 7370 2c20 2d2d 7370 6563 6965 7374 7265  sp, --speciestre
-00005660: 652c 2074 6865 2073 7065 6369 6573 2074  e, the species t
-00005670: 7265 6520 746f 2070 6572 666f 726d 2072  ree to perform r
-00005680: 6174 6520 636f 7272 6563 7469 6f6e 2c20  ate correction, 
-00005690: 6465 6661 756c 7420 4e6f 6e65 2c20 6966  default None, if
-000056a0: 204e 6f6e 6520 7761 7320 6769 7665 6e2c   None was given,
-000056b0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
-000056c0: 7469 6f6e 2061 6e61 6c79 7369 7320 776f  tion analysis wo
-000056d0: 756c 6420 6265 2063 616c 6c65 6420 6f66  uld be called of
-000056e0: 660a 2d72 772c 202d 2d72 6577 6569 6768  f.-rw, --reweigh
-000056f0: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
-00005700: 7768 6574 6865 7220 746f 2072 6563 616c  whether to recal
-00005710: 6375 6c61 7465 2074 6865 2077 6569 6768  culate the weigh
-00005720: 7420 7065 7220 7370 6563 6965 7320 7061  t per species pa
-00005730: 6972 2c20 6966 2074 6865 2066 6c61 6720  ir, if the flag 
-00005740: 7761 7320 7365 742c 2074 6865 2077 6569  was set, the wei
-00005750: 6768 7420 776f 756c 6420 6265 2072 6563  ght would be rec
-00005760: 616c 6375 6c61 7465 640a 2d6f 722c 202d  alculated.-or, -
-00005770: 2d6f 6e6c 7972 6f6f 746f 7574 2c20 666c  -onlyrootout, fl
-00005780: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00005790: 6572 2074 6f20 6f6e 6c79 2063 6f6e 6475  er to only condu
-000057a0: 6374 2072 6174 6520 636f 7272 6563 7469  ct rate correcti
-000057b0: 6f6e 2075 7369 6e67 2074 6865 206f 7574  on using the out
-000057c0: 6772 6f75 7020 6174 2072 6f6f 7420 6173  group at root as
-000057d0: 206f 7574 6772 6f75 702c 2069 6620 7468   outgroup, if th
-000057e0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000057f0: 6f6e 6c79 2074 6865 206f 7574 6772 6f75  only the outgrou
-00005800: 7020 6174 2072 6f6f 7420 776f 756c 6420  p at root would 
-00005810: 6265 2075 7365 6420 6173 206f 7574 6772  be used as outgr
-00005820: 6f75 700a 2d65 706b 2c20 2d2d 6578 7472  oup.-epk, --extr
-00005830: 6170 6172 616e 6f6d 656b 732c 2065 7874  aparanomeks, ext
-00005840: 7261 2070 6172 616e 6f6d 6520 4b73 2064  ra paranome Ks d
-00005850: 6174 6120 746f 2070 6c6f 7420 696e 2074  ata to plot in t
-00005860: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
-00005870: 7269 6275 7469 6f6e 2c20 6465 6661 756c  ribution, defaul
-00005880: 7420 4e6f 6e65 0a2d 6170 2c20 2d2d 616e  t None.-ap, --an
-00005890: 6368 6f72 706f 696e 7473 2c20 616e 6368  chorpoints, anch
-000058a0: 6f72 706f 696e 7473 2e74 7874 2066 696c  orpoints.txt fil
-000058b0: 6520 746f 2070 6c6f 7420 616e 6368 6f72  e to plot anchor
-000058c0: 204b 7320 696e 2074 6865 206d 6978 6564   Ks in the mixed
-000058d0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-000058e0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-000058f0: 706b 2c20 2d2d 706c 6f74 6b64 652c 2066  pk, --plotkde, f
-00005900: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005910: 6865 7220 746f 2070 6c6f 7420 6b64 6520  her to plot kde 
-00005920: 6375 7276 6520 6f66 206f 7274 686f 6c6f  curve of ortholo
-00005930: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
-00005940: 7469 6f6e 206f 7665 7220 6869 7374 6f67  tion over histog
-00005950: 7261 6d20 696e 2074 6865 206d 6978 6564  ram in the mixed
-00005960: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005970: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005980: 7320 7365 742c 2074 6865 206b 6465 2063  s set, the kde c
-00005990: 7572 7665 2077 6f75 6c64 2062 6520 706c  urve would be pl
-000059a0: 6f74 7465 640a 2d70 6167 2c20 2d2d 706c  otted.-pag, --pl
-000059b0: 6f74 6170 676d 6d2c 2066 6c61 6720 6f70  otapgmm, flag op
-000059c0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-000059d0: 2070 6572 666f 726d 2061 6e64 2070 6c6f   perform and plo
-000059e0: 7420 6d69 7874 7572 6520 6d6f 6465 6c69  t mixture modeli
-000059f0: 6e67 206f 6620 616e 6368 6f72 204b 7320  ng of anchor Ks 
-00005a00: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
-00005a10: 6469 7374 7269 6275 7469 6f6e 2c20 6966  distribution, if
-00005a20: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00005a30: 742c 2074 6865 206d 6978 7475 7265 206d  t, the mixture m
-00005a40: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
-00005a50: 7220 4b73 2077 6f75 6c64 2062 6520 706c  r Ks would be pl
-00005a60: 6f74 7465 640a 2d70 656d 2c20 2d2d 706c  otted.-pem, --pl
-00005a70: 6f74 656c 6d6d 2c20 666c 6167 206f 7074  otelmm, flag opt
-00005a80: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00005a90: 7065 7266 6f72 6d20 616e 6420 706c 6f74  perform and plot
-00005aa0: 2065 6c6d 6d20 6d69 7874 7572 6520 6d6f   elmm mixture mo
-00005ab0: 6465 6c69 6e67 206f 6620 7061 7261 6e6f  deling of parano
-00005ac0: 6d65 204b 7320 696e 2074 6865 206d 6978  me Ks in the mix
-00005ad0: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
-00005ae0: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-00005af0: 7761 7320 7365 742c 2074 6865 2065 6c6d  was set, the elm
-00005b00: 6d20 6d69 7874 7572 6520 6d6f 6465 6c69  m mixture modeli
-00005b10: 6e67 206f 6620 7061 7261 6e6f 6d65 204b  ng of paranome K
-00005b20: 7320 776f 756c 6420 6265 2070 6c6f 7474  s would be plott
-00005b30: 6564 0a2d 632c 202d 2d63 6f6d 706f 6e65  ed.-c, --compone
-00005b40: 6e74 732c 2074 6865 2072 616e 6765 206f  nts, the range o
-00005b50: 6620 7468 6520 6e75 6d62 6572 206f 6620  f the number of 
-00005b60: 636f 6d70 6f6e 656e 7473 2074 6f20 6669  components to fi
-00005b70: 7420 696e 2061 6e63 686f 7220 4b73 206d  t in anchor Ks m
-00005b80: 6978 7475 7265 206d 6f64 656c 696e 672c  ixture modeling,
-00005b90: 2064 6566 6175 6c74 2028 312c 3429 0a2d   default (1,4).-
-00005ba0: 786c 2c20 2d2d 786c 696d 2c20 7468 6520  xl, --xlim, the 
-00005bb0: 7820 6178 6973 206c 696d 6974 206f 6620  x axis limit of 
-00005bc0: 4b73 2064 6973 7472 6962 7574 696f 6e0a  Ks distribution.
-00005bd0: 2d79 6c2c 202d 2d79 6c69 6d2c 2074 6865  -yl, --ylim, the
-00005be0: 2079 2061 7869 7320 6c69 6d69 7420 6f66   y axis limit of
-00005bf0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005c00: 0a2d 6164 6f2c 202d 2d61 646a 7573 746f  .-ado, --adjusto
-00005c10: 7274 686f 2c20 666c 6167 206f 7074 696f  rtho, flag optio
-00005c20: 6e2c 2077 6865 7468 6572 2074 6f20 6164  n, whether to ad
-00005c30: 6a75 7374 2074 6865 2068 6973 746f 6772  just the histogr
-00005c40: 616d 2068 6569 6768 7420 6f66 206f 7274  am height of ort
-00005c50: 686f 6c6f 676f 7573 204b 7320 6173 2074  hologous Ks as t
-00005c60: 6f20 6d61 7463 6820 7468 6520 6865 6967  o match the heig
-00005c70: 6874 206f 6620 7061 7261 6c6f 676f 7573  ht of paralogous
-00005c80: 204b 732c 2069 6620 7468 6520 666c 6167   Ks, if the flag
-00005c90: 2077 6173 2073 6574 2c20 7468 6520 6164   was set, the ad
-00005ca0: 6a75 7374 6d65 6e74 2077 6f75 6c64 2062  justment would b
-00005cb0: 6520 636f 6e64 7563 7465 640a 2d61 6466  e conducted.-adf
-00005cc0: 2c20 2d2d 6164 6a75 7374 6661 6374 6f72  , --adjustfactor
-00005cd0: 2c20 7468 6520 6164 6a75 7374 6d65 6e74  , the adjustment
-00005ce0: 2066 6163 746f 7220 6f66 206f 7274 686f   factor of ortho
-00005cf0: 6c6f 676f 7573 204b 732c 2064 6566 6175  logous Ks, defau
-00005d00: 6c74 2030 2e35 0a2d 6f61 2c20 2d2d 6f6b  lt 0.5.-oa, --ok
-00005d10: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
-00005d20: 7479 206f 6620 6f72 7468 6f6c 6f67 6f75  ty of orthologou
-00005d30: 7320 4b73 2064 6973 7472 6962 7574 696f  s Ks distributio
-00005d40: 6e20 696e 206d 6978 6564 2070 6c6f 742c  n in mixed plot,
-00005d50: 2064 6566 6175 6c74 2030 2e35 0a2d 6661   default 0.5.-fa
-00005d60: 2c20 2d2d 666f 6375 7332 616c 6c2c 2073  , --focus2all, s
-00005d70: 6574 2066 6f63 616c 2073 7065 6369 6573  et focal species
-00005d80: 2061 6e64 206c 6574 2073 7065 6369 6573   and let species
-00005d90: 2070 6169 7220 746f 2062 6520 6265 7477   pair to be betw
-00005da0: 6565 6e20 666f 6361 6c20 616e 6420 616c  een focal and al
-00005db0: 6c20 7468 6520 7265 6d61 696e 696e 6720  l the remaining 
-00005dc0: 7370 6563 6965 732c 2064 6566 6175 6c74  species, default
-00005dd0: 204e 6f6e 650a 2d6b 732c 202d 2d6b 7374   None.-ks, --kst
-00005de0: 7265 652c 2066 6c61 6720 6f70 7469 6f6e  ree, flag option
-00005df0: 2c20 7768 6574 6865 7220 746f 2069 6e66  , whether to inf
-00005e00: 6572 204b 7320 7472 6565 2c20 6966 2074  er Ks tree, if t
-00005e10: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00005e20: 2074 6865 204b 7320 7472 6565 2069 6e66   the Ks tree inf
-00005e30: 6572 656e 6365 2061 6e61 6c79 7369 7320  erence analysis 
-00005e40: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
-00005e50: 6564 0a2d 6f63 6b2c 202d 2d6f 6e6c 7963  ed.-ock, --onlyc
-00005e60: 6f6e 6361 746b 7374 7265 652c 2066 6c61  oncatkstree, fla
-00005e70: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00005e80: 7220 746f 206f 6e6c 7920 696e 6665 7220  r to only infer 
-00005e90: 4b73 2074 7265 6520 756e 6465 7220 636f  Ks tree under co
-00005ea0: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
-00005eb0: 6d65 6e74 2c20 6966 2074 6865 2066 6c61  ment, if the fla
-00005ec0: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
-00005ed0: 7468 6520 4b73 2074 7265 6520 756e 6465  the Ks tree unde
-00005ee0: 7220 636f 6e63 6174 656e 6174 6564 2061  r concatenated a
-00005ef0: 6c69 676e 6d65 6e74 2077 6f75 6c64 2062  lignment would b
-00005f00: 6520 6361 6c63 756c 6174 6564 0a2d 6373  e calculated.-cs
-00005f10: 2c20 2d2d 636c 6173 7369 632c 2066 6c61  , --classic, fla
-00005f20: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00005f30: 7220 746f 2064 7261 7720 6d69 7865 6420  r to draw mixed 
-00005f40: 4b73 2070 6c6f 7420 696e 2061 2063 6c61  Ks plot in a cla
-00005f50: 7373 6963 206d 616e 6e65 7220 7768 6572  ssic manner wher
-00005f60: 6520 7468 6520 6675 6c6c 206f 7274 686f  e the full ortho
-00005f70: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
-00005f80: 6275 7469 6f6e 2069 7320 6472 6177 6564  bution is drawed
-00005f90: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005fa0: 7320 7365 742c 2074 6865 2063 6c61 7373  s set, the class
-00005fb0: 6963 206d 6978 6564 204b 7320 706c 6f74  ic mixed Ks plot
-00005fc0: 2077 6f75 6c64 2062 6520 6472 6177 6e0a   would be drawn.
-00005fd0: 2d74 612c 202d 2d74 6f70 6172 726f 772c  -ta, --toparrow,
-00005fe0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005ff0: 6574 6865 7220 746f 2061 646a 7573 7420  ether to adjust 
-00006000: 7468 6520 6172 726f 7720 746f 2062 6520  the arrow to be 
-00006010: 6174 2074 6865 2074 6f70 206f 6620 7468  at the top of th
-00006020: 6520 706c 6f74 2c20 696e 7374 6561 6420  e plot, instead 
-00006030: 6f66 2062 6569 6e67 2063 6f6f 7264 696e  of being coordin
-00006040: 6174 6564 2061 7320 7468 6520 4b44 4520  ated as the KDE 
-00006050: 6f66 2074 6865 206f 7274 686f 6c6f 676f  of the orthologo
-00006060: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
-00006070: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-00006080: 7761 7320 7365 742c 2074 6865 2061 7272  was set, the arr
-00006090: 6f77 2077 6f75 6c64 2062 6520 7365 7420  ow would be set 
-000060a0: 6174 2074 6865 2074 6f70 0a2d 6273 2c20  at the top.-bs, 
-000060b0: 2d2d 626f 6f74 7374 7261 702c 2074 6865  --bootstrap, the
-000060c0: 206e 756d 6265 7220 6f66 2062 6f6f 7473   number of boots
-000060d0: 7472 6170 2072 6570 6c69 6361 7465 7320  trap replicates 
-000060e0: 6f66 206f 7274 686f 6c6f 6720 4b73 2064  of ortholog Ks d
-000060f0: 6973 7472 6962 7574 696f 6e20 696e 206d  istribution in m
-00006100: 6978 6564 2070 6c6f 740a 6060 600a 0a54  ixed plot.```..T
-00006110: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-00006120: 6d69 7860 2063 616e 2072 6561 6c69 7a65  mix` can realize
-00006130: 2074 6865 206d 6978 7475 7265 206d 6f64   the mixture mod
-00006140: 656c 2063 6c75 7374 6572 696e 6720 616e  el clustering an
-00006150: 616c 7973 6973 206f 6620 2a4b 2a3c 7375  alysis of *K*<su
-00006160: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00006170: 7374 7269 6275 7469 6f6e 2e0a 6060 600a  stribution..```.
-00006180: 7767 6420 6d69 7820 6b73 5f64 6174 6166  wgd mix ks_dataf
-00006190: 696c 6520 286f 7074 696f 6e29 0a2d 2d2d  ile (option).---
-000061a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d66  -------------.-f
-000061f0: 2c20 2d2d 6669 6c74 6572 732c 2074 6865  , --filters, the
-00006200: 2063 7574 6f66 6620 616c 6967 6e6d 656e   cutoff alignmen
-00006210: 7420 6c65 6e67 7468 2c20 6465 6661 756c  t length, defaul
-00006220: 7420 3330 300a 2d72 2c20 2d2d 6b73 5f72  t 300.-r, --ks_r
-00006230: 616e 6765 2c20 7468 6520 4b73 2072 616e  ange, the Ks ran
-00006240: 6765 2074 6f20 6265 2063 6f6e 7369 6465  ge to be conside
-00006250: 7265 642c 2064 6566 6175 6c74 2028 302c  red, default (0,
-00006260: 2035 290a 2d62 2c20 2d2d 6269 6e73 2c20   5).-b, --bins, 
-00006270: 7468 6520 6e75 6d62 6572 206f 6620 6269  the number of bi
-00006280: 6e73 2069 6e20 4b73 2064 6973 7472 6962  ns in Ks distrib
-00006290: 7574 696f 6e2c 2064 6566 6175 6c74 2035  ution, default 5
-000062a0: 300a 2d6f 2c20 2d2d 6f75 7464 6972 2c20  0.-o, --outdir, 
-000062b0: 7468 6520 6f75 7470 7574 2064 6972 6563  the output direc
-000062c0: 746f 7279 2c20 6465 6661 756c 7420 7767  tory, default wg
-000062d0: 645f 6d69 780a 2d2d 6d65 7468 6f64 2c20  d_mix.--method, 
-000062e0: 7768 6963 6820 6d69 7874 7572 6520 6d6f  which mixture mo
-000062f0: 6465 6c20 746f 2075 7365 2c20 6465 6661  del to use, defa
-00006300: 756c 7420 676d 6d0a 2d6e 2c20 2d2d 636f  ult gmm.-n, --co
-00006310: 6d70 6f6e 656e 7473 2c20 7468 6520 7261  mponents, the ra
-00006320: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-00006330: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-00006340: 746f 2066 6974 2c20 6465 6661 756c 7420  to fit, default 
-00006350: 2831 2c20 3429 0a2d 672c 202d 2d67 616d  (1, 4).-g, --gam
-00006360: 6d61 2c20 7468 6520 6761 6d6d 6120 7061  ma, the gamma pa
-00006370: 7261 6d65 7465 7220 666f 7220 6267 6d6d  rameter for bgmm
-00006380: 206d 6f64 656c 732c 2064 6566 6175 6c74   models, default
-00006390: 2030 2e30 3031 0a2d 6e69 2c20 2d2d 6e5f   0.001.-ni, --n_
-000063a0: 696e 6974 2c20 7468 6520 6e75 6d62 6572  init, the number
-000063b0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-000063c0: 6961 6c69 7a61 7469 6f6e 732c 2064 6566  ializations, def
-000063d0: 6175 6c74 2032 3030 0a2d 6d69 2c20 2d2d  ault 200.-mi, --
-000063e0: 6d61 785f 6974 6572 2c20 7468 6520 6d61  max_iter, the ma
-000063f0: 7869 6d75 6d20 6e75 6d62 6572 206f 6620  ximum number of 
-00006400: 6974 6572 6174 696f 6e73 2c20 6465 6661  iterations, defa
-00006410: 756c 7420 3230 300a 6060 600a 0a54 6865  ult 200.```..The
-00006420: 2070 726f 6772 616d 2060 7767 6420 7065   program `wgd pe
-00006430: 616b 6020 6361 6e20 7265 616c 697a 6520  ak` can realize 
-00006440: 7468 6520 7365 6172 6368 206f 6620 6372  the search of cr
-00006450: 6564 6961 626c 6520 2a4b 2a3c 7375 623e  ediable *K*<sub>
-00006460: 533c 2f73 7562 3e20 7261 6e67 6520 7573  S</sub> range us
-00006470: 6564 2069 6e20 5747 4420 6461 7469 6e67  ed in WGD dating
-00006480: 2e0a 6060 600a 7767 6420 7065 616b 206b  ..```.wgd peak k
-00006490: 735f 6461 7461 6669 6c65 2028 6f70 7469  s_datafile (opti
-000064a0: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
-000064b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064f0: 2d2d 2d2d 0a2d 6170 2c20 2d2d 616e 6368  ----.-ap, --anch
-00006500: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
-00006510: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
-00006520: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00006530: 6e65 0a2d 736d 2c20 2d2d 7365 676d 656e  ne.-sm, --segmen
-00006540: 7473 2c20 7468 6520 7365 676d 656e 7473  ts, the segments
-00006550: 2064 6174 6166 696c 652c 2064 6566 6175   datafile, defau
-00006560: 6c74 204e 6f6e 650a 2d6c 652c 202d 2d6c  lt None.-le, --l
-00006570: 6973 7465 6c65 6d65 6e74 732c 2074 6865  istelements, the
-00006580: 206c 6973 7473 6567 6d65 6e74 7320 6461   listsegments da
-00006590: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-000065a0: 4e6f 6e65 200a 2d6d 702c 202d 2d6d 756c  None .-mp, --mul
-000065b0: 7469 706c 6963 6f6e 7061 6972 732c 2074  tipliconpairs, t
-000065c0: 6865 206d 756c 7469 706c 6963 6f6e 7061  he multipliconpa
-000065d0: 6972 7320 6461 7461 6669 6c65 2c20 6465  irs datafile, de
-000065e0: 6661 756c 7420 4e6f 6e65 0a2d 6f2c 202d  fault None.-o, -
-000065f0: 2d6f 7574 6469 722c 2074 6865 206f 7574  -outdir, the out
-00006600: 7075 7420 6469 7265 6374 6f72 792c 2064  put directory, d
-00006610: 6566 6175 6c74 2077 6764 5f70 6561 6b0a  efault wgd_peak.
-00006620: 2d61 662c 202d 2d61 6c69 676e 6669 6c74  -af, --alignfilt
-00006630: 6572 2c20 6375 746f 6666 2066 6f72 2061  er, cutoff for a
-00006640: 6c69 676e 6d65 6e74 2069 6465 6e74 6974  lignment identit
-00006650: 792c 206c 656e 6774 6820 616e 6420 636f  y, length and co
-00006660: 7665 7261 6765 2c20 6465 6661 756c 7420  verage, default 
-00006670: 302e 302c 2030 2c20 302e 300a 2d72 2c20  0.0, 0, 0.0.-r, 
-00006680: 2d2d 6b73 7261 6e67 652c 2072 616e 6765  --ksrange, range
-00006690: 206f 6620 4b73 2074 6f20 6265 2061 6e61   of Ks to be ana
-000066a0: 6c79 7a65 642c 2064 6566 6175 6c74 2028  lyzed, default (
-000066b0: 302c 2035 290a 2d62 772c 202d 2d62 696e  0, 5).-bw, --bin
-000066c0: 5f77 6964 7468 2c20 6261 6e64 7769 6474  _width, bandwidt
-000066d0: 6820 6f66 204b 7320 6469 7374 7269 6275  h of Ks distribu
-000066e0: 7469 6f6e 2c20 6465 6661 756c 7420 302e  tion, default 0.
-000066f0: 310a 2d69 632c 202d 2d77 6569 6768 7473  1.-ic, --weights
-00006700: 5f6f 7574 6c69 6572 735f 696e 636c 7564  _outliers_includ
-00006710: 6564 2c20 666c 6167 206f 7074 696f 6e2c  ed, flag option,
-00006720: 2077 6865 7468 6572 2074 6f20 696e 636c   whether to incl
-00006730: 7564 6520 4b73 206f 7574 6c69 6572 732c  ude Ks outliers,
-00006740: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00006750: 2073 6574 2c20 4b73 206f 7574 6c69 6572   set, Ks outlier
-00006760: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
-00006770: 6465 6420 696e 2074 6865 2061 6e61 6c79  ded in the analy
-00006780: 7369 730a 2d6d 2c20 2d2d 6d65 7468 6f64  sis.-m, --method
-00006790: 2c20 7768 6963 6820 6d69 7874 7572 6520  , which mixture 
-000067a0: 6d6f 6465 6c20 746f 2075 7365 2c20 6465  model to use, de
-000067b0: 6661 756c 7420 676d 6d0a 2d2d 7365 6564  fault gmm.--seed
-000067c0: 2c20 7261 6e64 6f6d 2073 6565 6420 6769  , random seed gi
-000067d0: 7665 6e20 746f 2069 6e69 7469 616c 697a  ven to initializ
-000067e0: 6174 696f 6e2c 2064 6566 6175 6c74 2032  ation, default 2
-000067f0: 3335 3238 3930 0a2d 6569 2c20 2d2d 656d  352890.-ei, --em
-00006800: 5f69 7465 722c 2074 6865 206e 756d 6265  _iter, the numbe
-00006810: 7220 6f66 2045 4d20 6974 6572 6174 696f  r of EM iteratio
-00006820: 6e73 2074 6f20 7065 7266 6f72 6d2c 2064  ns to perform, d
-00006830: 6566 6175 6c74 2032 3030 0a2d 6e69 2c20  efault 200.-ni, 
-00006840: 2d2d 6e5f 696e 6974 2c20 7468 6520 6e75  --n_init, the nu
-00006850: 6d62 6572 206f 6620 6b2d 6d65 616e 7320  mber of k-means 
-00006860: 696e 6974 6961 6c69 7a61 7469 6f6e 732c  initializations,
-00006870: 2064 6566 6175 6c74 2032 3030 0a2d 6e2c   default 200.-n,
-00006880: 202d 2d63 6f6d 706f 6e65 6e74 732c 2074   --components, t
-00006890: 6865 2072 616e 6765 206f 6620 7468 6520  he range of the 
-000068a0: 6e75 6d62 6572 206f 6620 636f 6d70 6f6e  number of compon
-000068b0: 656e 7473 2074 6f20 6669 742c 2064 6566  ents to fit, def
-000068c0: 6175 6c74 2028 312c 2034 290a 2d67 2c20  ault (1, 4).-g, 
-000068d0: 2d2d 6761 6d6d 612c 2074 6865 2067 616d  --gamma, the gam
-000068e0: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
-000068f0: 2062 676d 6d20 6d6f 6465 6c73 2c20 6465   bgmm models, de
-00006900: 6661 756c 7420 3165 2d33 0a2d 2d62 6f6f  fault 1e-3.--boo
-00006910: 7473 2c20 7468 6520 6e75 6d62 6572 206f  ts, the number o
-00006920: 6620 626f 6f74 7374 7261 7020 7265 706c  f bootstrap repl
-00006930: 6963 6174 6573 206f 6620 6b64 652c 2064  icates of kde, d
-00006940: 6566 6175 6c74 2032 3030 0a2d 2d77 6569  efault 200.--wei
-00006950: 6768 7465 642c 2066 6c61 6720 6f70 7469  ghted, flag opti
-00006960: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
-00006970: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
-00006980: 206d 6574 686f 6420 6f66 2064 652d 7265   method of de-re
-00006990: 6475 6e64 616e 6379 2c20 6966 2074 6865  dundancy, if the
-000069a0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-000069b0: 6865 206e 6f64 652d 7765 6967 6874 6564  he node-weighted
-000069c0: 206d 6574 686f 6420 776f 756c 6420 6265   method would be
-000069d0: 2075 7365 640a 2d70 2c20 2d2d 706c 6f74   used.-p, --plot
-000069e0: 2c20 7468 6520 706c 6f74 7469 6e67 206d  , the plotting m
-000069f0: 6574 686f 6420 746f 2062 6520 7573 6564  ethod to be used
-00006a00: 2c20 6465 6661 756c 7420 6964 656e 7469  , default identi
-00006a10: 6361 6c0a 2d62 6d2c 202d 2d62 775f 6d65  cal.-bm, --bw_me
-00006a20: 7468 6f64 2c20 7468 6520 6261 6e64 7769  thod, the bandwi
-00006a30: 6474 6820 6d65 7468 6f64 2074 6f20 6265  dth method to be
-00006a40: 2075 7365 6420 696e 2061 6e61 6c79 7a69   used in analyzi
-00006a50: 6e67 2074 6865 2070 6561 6b20 6f66 2057  ng the peak of W
-00006a60: 4744 2064 6174 6573 2c20 6465 6661 756c  GD dates, defaul
-00006a70: 7420 7369 6c76 6572 6d61 6e0a 2d2d 6e5f  t silverman.--n_
-00006a80: 6d65 646f 6964 732c 2074 6865 206e 756d  medoids, the num
-00006a90: 6265 7220 6f66 206d 6564 6f69 6473 2074  ber of medoids t
-00006aa0: 6f20 6669 742c 2064 6566 6175 6c74 2032  o fit, default 2
-00006ab0: 0a2d 6b6d 2c20 2d2d 6b64 656d 6574 686f  .-km, --kdemetho
-00006ac0: 642c 2074 6865 206b 6465 206d 6574 686f  d, the kde metho
-00006ad0: 6420 746f 2062 6520 7573 6564 2069 6e20  d to be used in 
-00006ae0: 616e 616c 797a 696e 6720 7468 6520 7065  analyzing the pe
-00006af0: 616b 206f 6620 5747 4420 6461 7465 732c  ak of WGD dates,
-00006b00: 206b 6d65 646f 6964 7320 616e 616c 7973   kmedoids analys
-00006b10: 6973 206f 7220 7468 6520 6261 7369 6320  is or the basic 
-00006b20: 4b73 2070 6c6f 7474 696e 672c 2064 6566  Ks plotting, def
-00006b30: 6175 6c74 2073 6369 7079 0a2d 2d6e 5f63  ault scipy.--n_c
-00006b40: 6c75 7374 6572 732c 2074 6865 206e 756d  lusters, the num
-00006b50: 6265 7220 6f66 2063 6c75 7374 6572 7320  ber of clusters 
-00006b60: 746f 2070 6c6f 7420 456c 626f 7720 6c6f  to plot Elbow lo
-00006b70: 7373 2066 756e 6374 696f 6e2c 2064 6566  ss function, def
-00006b80: 6175 6c74 2035 0a2d 6764 2c20 2d2d 6775  ault 5.-gd, --gu
-00006b90: 6964 652c 2074 6865 2072 6567 696d 6520  ide, the regime 
-00006ba0: 7265 7369 6469 6e67 2061 6e63 686f 7273  residing anchors
-00006bb0: 2c20 6465 6661 756c 7420 5365 676d 656e  , default Segmen
-00006bc0: 740a 2d70 7263 742c 202d 2d70 726f 6d69  t.-prct, --promi
-00006bd0: 6e65 6e63 655f 6375 746f 6666 2c20 7468  nence_cutoff, th
-00006be0: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
-00006bf0: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
-00006c00: 6520 7065 616b 7320 696e 2070 6561 6b20  e peaks in peak 
-00006c10: 6669 6e64 696e 6720 7374 6570 732c 2064  finding steps, d
-00006c20: 6566 6175 6c74 2030 2e31 0a2d 7268 2c20  efault 0.1.-rh, 
-00006c30: 2d2d 7265 6c5f 6865 6967 6874 2c20 7468  --rel_height, th
-00006c40: 6520 7265 6c61 7469 7665 2068 6569 6768  e relative heigh
-00006c50: 7420 6174 2077 6869 6368 2074 6865 2070  t at which the p
-00006c60: 6561 6b20 7769 6474 6820 6973 206d 6561  eak width is mea
-00006c70: 7375 7265 642c 2064 6566 6175 6c74 2030  sured, default 0
-00006c80: 2e34 0a2d 6b64 2c20 2d2d 6b73 746f 6461  .4.-kd, --kstoda
-00006c90: 7465 2c20 7468 6520 7261 6e67 6520 6f66  te, the range of
-00006ca0: 204b 7320 746f 2062 6520 6461 7465 6420   Ks to be dated 
-00006cb0: 696e 2068 6575 7269 7374 6963 2073 6561  in heuristic sea
-00006cc0: 7263 682c 2064 6566 6175 6c74 2028 302e  rch, default (0.
-00006cd0: 352c 2031 2e35 290a 2d78 6c2c 202d 2d78  5, 1.5).-xl, --x
-00006ce0: 6c69 6d2c 2074 6865 2078 2061 7869 7320  lim, the x axis 
-00006cf0: 6c69 6d69 7420 6f66 2047 4d4d 204b 7320  limit of GMM Ks 
-00006d00: 6469 7374 7269 6275 7469 6f6e 0a2d 796c  distribution.-yl
-00006d10: 2c20 2d2d 796c 696d 2c20 7468 6520 7920  , --ylim, the y 
-00006d20: 6178 6973 206c 696d 6974 206f 6620 474d  axis limit of GM
-00006d30: 4d20 4b73 2064 6973 7472 6962 7574 696f  M Ks distributio
-00006d40: 6e0a 2d2d 6d61 6e75 616c 7365 742c 2066  n.--manualset, f
-00006d50: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00006d60: 6865 7220 746f 206f 7574 7075 7420 616e  her to output an
-00006d70: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
-00006d80: 6d61 6e75 616c 6c79 2073 6574 204b 7320  manually set Ks 
-00006d90: 7261 6e67 652c 2069 6620 7468 6520 666c  range, if the fl
-00006da0: 6167 2077 6173 2073 6574 2c20 6d61 6e75  ag was set, manu
-00006db0: 616c 6c79 2073 6574 204b 7320 7261 6e67  ally set Ks rang
-00006dc0: 6520 776f 756c 6420 6265 206f 7574 7075  e would be outpu
-00006dd0: 7474 6564 0a2d 2d63 692c 2074 6865 2063  tted.--ci, the c
-00006de0: 6f6e 6669 6465 6e63 6520 6c65 7665 6c20  onfidence level 
-00006df0: 6f66 206c 6f67 2d6e 6f72 6d61 6c20 6469  of log-normal di
-00006e00: 7374 7269 6275 7469 6f6e 2074 6f20 6461  stribution to da
-00006e10: 7465 2c20 6465 6661 756c 7420 3935 0a2d  te, default 95.-
-00006e20: 2d68 6472 2c20 7468 6520 6869 6768 6573  -hdr, the highes
-00006e30: 7420 6465 6e73 6974 7920 7265 6769 6f6e  t density region
-00006e40: 2028 4844 5229 2061 7070 6c69 6564 2069   (HDR) applied i
-00006e50: 6e20 7468 6520 7365 676d 656e 742d 6775  n the segment-gu
-00006e60: 6964 6564 2061 6e63 686f 7220 7061 6972  ided anchor pair
-00006e70: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00006e80: 2c20 6465 6661 756c 7420 3935 0a2d 2d68  , default 95.--h
-00006e90: 6575 7269 7374 6963 2c20 666c 6167 206f  euristic, flag o
-00006ea0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-00006eb0: 6f20 696e 6974 6961 7465 2068 6575 7269  o initiate heuri
-00006ec0: 7374 6963 206d 6574 686f 6420 6f66 2064  stic method of d
-00006ed0: 6566 696e 696e 6720 4349 2066 6f72 2064  efining CI for d
-00006ee0: 6174 696e 672c 2069 6620 7468 6520 666c  ating, if the fl
-00006ef0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-00006f00: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
-00006f10: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00006f20: 7465 640a 2d6b 632c 202d 2d6b 7363 7574  ted.-kc, --kscut
-00006f30: 6f66 662c 2074 6865 204b 7320 7361 7475  off, the Ks satu
-00006f40: 7261 7469 6f6e 2063 7574 6f66 6620 696e  ration cutoff in
-00006f50: 2064 6174 696e 672c 2064 6566 6175 6c74   dating, default
-00006f60: 2035 0a2d 2d6b 6565 7074 6d70 6669 672c   5.--keeptmpfig,
-00006f70: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00006f80: 6574 6865 7220 746f 206b 6565 7020 7465  ether to keep te
-00006f90: 6d70 6f72 6172 7920 6669 6775 7265 7320  mporary figures 
-00006fa0: 696e 2070 6561 6b20 6669 6e64 696e 6720  in peak finding 
-00006fb0: 7072 6f63 6573 732c 2069 6620 7468 6520  process, if the 
-00006fc0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00006fd0: 6f73 6520 6669 6775 7265 7320 776f 756c  ose figures woul
-00006fe0: 6420 6265 206b 6570 740a 6060 600a 0a54  d be kept.```..T
-00006ff0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-00007000: 7379 6e60 2063 616e 2072 6561 6c69 7a65  syn` can realize
-00007010: 2074 6865 2069 6e74 7261 2d20 616e 6420   the intra- and 
-00007020: 696e 7465 722d 7370 6563 6966 6963 2073  inter-specific s
-00007030: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
-00007040: 2e0a 6060 600a 7767 6420 7379 6e20 6661  ..```.wgd syn fa
-00007050: 6d69 6c69 6573 2067 6666 7320 286f 7074  milies gffs (opt
-00007060: 696f 6e29 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d  ion).-----------
-00007070: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007080: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007090: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000070a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000070b0: 2d2d 2d2d 2d0a 2d6b 732c 202d 2d6b 735f  -----.-ks, --ks_
-000070c0: 6469 7374 7269 6275 7469 6f6e 2c20 6b73  distribution, ks
-000070d0: 2064 6973 7472 6962 7574 696f 6e20 6461   distribution da
-000070e0: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-000070f0: 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574 6469  None.-o, --outdi
-00007100: 722c 2074 6865 206f 7574 7075 7420 6469  r, the output di
-00007110: 7265 6374 6f72 792c 2064 6566 6175 6c74  rectory, default
-00007120: 2077 6764 5f73 796e 0a2d 662c 202d 2d66   wgd_syn.-f, --f
-00007130: 6561 7475 7265 2c20 7468 6520 6665 6174  eature, the feat
-00007140: 7572 6520 666f 7220 7061 7273 696e 6720  ure for parsing 
-00007150: 6765 6e65 2049 4473 2066 726f 6d20 4746  gene IDs from GF
-00007160: 4620 6669 6c65 732c 2064 6566 6175 6c74  F files, default
-00007170: 2067 656e 650a 2d61 2c20 2d2d 6174 7472   gene.-a, --attr
-00007180: 6962 7574 652c 2074 6865 2061 7474 7269  ibute, the attri
-00007190: 6275 7465 2066 6f72 2070 6172 7369 6e67  bute for parsing
-000071a0: 2074 6865 2067 656e 6520 4944 7320 6672   the gene IDs fr
-000071b0: 6f6d 2074 6865 2047 4646 2066 696c 6573  om the GFF files
-000071c0: 2c20 6465 6661 756c 7420 4944 0a2d 6174  , default ID.-at
-000071d0: 672c 202d 2d61 6464 6974 696f 6e61 6c67  g, --additionalg
-000071e0: 6666 696e 666f 2c20 7468 6520 6665 6174  ffinfo, the feat
-000071f0: 7572 6520 616e 6420 6174 7472 6962 7574  ure and attribut
-00007200: 6520 696e 666f 726d 6174 696f 6e20 6f66  e information of
-00007210: 2061 6464 6974 696f 6e61 6c20 6766 6633   additional gff3
-00007220: 2066 696c 6573 2069 6620 6469 6666 6572   files if differ
-00007230: 656e 7420 696e 2074 6865 2066 6f72 6d61  ent in the forma
-00007240: 7420 6f66 2028 6665 6174 7572 653b 6174  t of (feature;at
-00007250: 7472 6962 7574 6529 272c 2064 6566 6175  tribute)', defau
-00007260: 6c74 204e 6f6e 650a 2d6d 6c2c 202d 2d6d  lt None.-ml, --m
-00007270: 696e 6c65 6e2c 2074 6865 206d 696e 696d  inlen, the minim
-00007280: 756d 206c 656e 6774 6820 6f66 2061 2073  um length of a s
-00007290: 6361 6666 6f6c 6420 746f 2062 6520 696e  caffold to be in
-000072a0: 636c 7564 6564 2069 6e20 646f 7470 6c6f  cluded in dotplo
-000072b0: 742c 2064 6566 6175 6c74 202d 312c 2069  t, default -1, i
-000072c0: 6620 2d31 2077 6173 2073 6574 2c20 7468  f -1 was set, th
-000072d0: 6520 3130 2520 6f66 2074 6865 206c 6f6e  e 10% of the lon
-000072e0: 6765 7374 2073 6361 6666 6f6c 6420 776f  gest scaffold wo
-000072f0: 756c 6420 6265 2073 6574 0a2d 6d73 2c20  uld be set.-ms, 
-00007300: 2d2d 6d61 7873 697a 652c 2074 6865 206d  --maxsize, the m
-00007310: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
-00007320: 7a65 2074 6f20 6265 2069 6e63 6c75 6465  ze to be include
-00007330: 642c 2064 6566 6175 6c74 2032 3030 0a2d  d, default 200.-
-00007340: 722c 202d 2d6b 735f 7261 6e67 652c 2074  r, --ks_range, t
-00007350: 6865 204b 7320 7261 6e67 6520 696e 2063  he Ks range in c
-00007360: 6f6c 6f72 6564 2064 6f74 706c 6f74 2c20  olored dotplot, 
-00007370: 6465 6661 756c 7420 2830 2c20 3529 0a2d  default (0, 5).-
-00007380: 2d69 6164 686f 7265 5f6f 7074 696f 6e73  -iadhore_options
-00007390: 2c20 7468 6520 7061 7261 6d65 7465 7220  , the parameter 
-000073a0: 7365 7474 696e 6720 696e 2069 6164 686f  setting in iadho
-000073b0: 7265 2c20 6465 6661 756c 7420 6173 2061  re, default as a
-000073c0: 2073 7472 696e 6720 6f66 206c 656e 6774   string of lengt
-000073d0: 6820 7a65 726f 0a2d 6d67 2c20 2d2d 6d69  h zero.-mg, --mi
-000073e0: 6e73 6567 6c65 6e2c 2074 6865 206d 696e  nseglen, the min
-000073f0: 696d 756d 206c 656e 6774 6820 6f66 2073  imum length of s
-00007400: 6567 6d65 6e74 7320 746f 2069 6e63 6c75  egments to inclu
-00007410: 6465 2069 6e20 7261 7469 6f20 6966 203c  de in ratio if <
-00007420: 3d20 312c 2064 6566 6175 6c74 2031 3030  = 1, default 100
-00007430: 3030 0a2d 6b72 2c20 2d2d 6b65 6570 7265  00.-kr, --keepre
-00007440: 6475 6e2c 2066 6c61 6720 6f70 7469 6f6e  dun, flag option
-00007450: 2c20 7768 6574 6865 7220 746f 206b 6565  , whether to kee
-00007460: 7020 7265 6475 6e64 616e 7420 6d75 6c74  p redundant mult
-00007470: 6970 6c69 636f 6e73 2c20 6966 2074 6865  iplicons, if the
-00007480: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007490: 6865 2072 6564 756e 6461 6e74 206d 756c  he redundant mul
-000074a0: 7469 706c 6963 6f6e 7320 776f 756c 6420  tiplicons would 
-000074b0: 6265 206b 6570 740a 2d6d 676e 2c20 2d2d  be kept.-mgn, --
-000074c0: 6d69 6e67 656e 656e 756d 2c20 7468 6520  mingenenum, the 
-000074d0: 6d69 6e69 6d75 6d20 6e75 6d62 6572 206f  minimum number o
-000074e0: 6620 6765 6e65 7320 666f 7220 6120 7365  f genes for a se
-000074f0: 676d 656e 7420 746f 2062 6520 636f 6e73  gment to be cons
-00007500: 6964 6572 6564 2c20 6465 6661 756c 7420  idered, default 
-00007510: 3330 0a2d 6473 2c20 2d2d 646f 7473 697a  30.-ds, --dotsiz
-00007520: 652c 2074 6865 2064 6f74 2073 697a 6520  e, the dot size 
-00007530: 696e 2064 6f74 2070 6c6f 742c 2064 6566  in dot plot, def
-00007540: 6175 6c74 2030 2e33 0a2d 6161 2c20 2d2d  ault 0.3.-aa, --
-00007550: 6170 616c 7068 612c 2074 6865 206f 7061  apalpha, the opa
-00007560: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
-00007570: 6f74 732c 2064 6566 6175 6c74 2031 0a2d  ots, default 1.-
-00007580: 6861 2c20 2d2d 686f 616c 7068 612c 2074  ha, --hoalpha, t
-00007590: 6865 206f 7061 6369 7479 206f 6620 686f  he opacity of ho
-000075a0: 6d6f 6c6f 6720 646f 7473 2c20 6465 6661  molog dots, defa
-000075b0: 756c 7420 300a 2d73 7274 2c20 2d2d 7368  ult 0.-srt, --sh
-000075c0: 6f77 7265 616c 7469 636b 2c20 666c 6167  owrealtick, flag
-000075d0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000075e0: 2074 6f20 7368 6f77 2074 6865 2072 6561   to show the rea
-000075f0: 6c20 7469 636b 2069 6e20 6765 6e65 7320  l tick in genes 
-00007600: 6f72 2062 6173 6573 2c20 6966 2074 6865  or bases, if the
-00007610: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007620: 6865 2072 6561 6c20 7469 636b 2077 6f75  he real tick wou
-00007630: 6c64 2062 6520 7368 6f77 6564 0a2d 746c  ld be showed.-tl
-00007640: 732c 202d 2d74 6963 6b6c 6162 656c 7369  s, --ticklabelsi
-00007650: 7a65 2c20 7468 6520 6c61 6265 6c20 7369  ze, the label si
-00007660: 7a65 206f 6620 7469 636b 2c20 6465 6661  ze of tick, defa
-00007670: 756c 7420 350a 2d67 722c 202d 2d67 6973  ult 5.-gr, --gis
-00007680: 7472 622c 2066 6c61 6720 6f70 7469 6f6e  trb, flag option
-00007690: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
-000076a0: 2067 6973 745f 7261 696e 626f 7720 6173   gist_rainbow as
-000076b0: 2063 6f6c 6f72 206d 6170 206f 6620 646f   color map of do
-000076c0: 7470 6c6f 740a 2d6e 2c20 2d2d 6e74 6872  tplot.-n, --nthr
-000076d0: 6561 6473 2c20 7468 6520 6e75 6d62 6572  eads, the number
-000076e0: 206f 6620 7468 7265 6164 7320 746f 2075   of threads to u
-000076f0: 7365 2069 6e20 7379 6e74 656e 7920 696e  se in synteny in
-00007700: 6665 7265 6e63 652c 2064 6566 6175 6c74  ference, default
-00007710: 2034 0a60 6060 0a0a 5468 6520 7072 6f67   4.```..The prog
-00007720: 7261 6d20 6077 6764 2076 697a 6020 6361  ram `wgd viz` ca
-00007730: 6e20 7265 616c 697a 6520 7468 6520 7669  n realize the vi
-00007740: 7375 616c 697a 6174 696f 6e20 6f66 202a  sualization of *
-00007750: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-00007760: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-00007770: 616e 6420 7379 6e74 656e 792e 0a60 6060  and synteny..```
-00007780: 0a77 6764 2076 697a 2028 6f70 7469 6f6e  .wgd viz (option
-00007790: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
-000077a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077e0: 2d2d 0a2d 642c 202d 2d64 6174 6166 696c  --.-d, --datafil
-000077f0: 652c 2074 6865 204b 7320 6461 7461 6669  e, the Ks datafi
-00007800: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
-00007810: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00007820: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00007830: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00007840: 5f76 697a 0a2d 7372 2c20 2d2d 7370 6169  _viz.-sr, --spai
-00007850: 722c 2074 6865 2073 7065 6369 6573 2070  r, the species p
-00007860: 6169 7220 746f 2062 6520 706c 6f74 7465  air to be plotte
-00007870: 642c 2064 6566 6175 6c74 204e 6f6e 652c  d, default None,
-00007880: 2074 6869 7320 6f70 7469 6f6e 2063 616e   this option can
-00007890: 2062 6520 7072 6f76 6964 6564 206d 756c   be provided mul
-000078a0: 7469 706c 6520 7469 6d65 730a 2d66 612c  tiple times.-fa,
-000078b0: 202d 2d66 6f63 7573 3261 6c6c 2c20 7365   --focus2all, se
-000078c0: 7420 666f 6361 6c20 7370 6563 6965 7320  t focal species 
-000078d0: 616e 6420 6c65 7420 7370 6563 6965 7320  and let species 
-000078e0: 7061 6972 2074 6f20 6265 2062 6574 7765  pair to be betwe
-000078f0: 656e 2066 6f63 616c 2061 6e64 2061 6c6c  en focal and all
-00007900: 2074 6865 2072 656d 6169 6e69 6e67 2073   the remaining s
-00007910: 7065 6369 6573 2c20 6465 6661 756c 7420  pecies, default 
-00007920: 4e6f 6e65 0a2d 6773 2c20 2d2d 6773 6d61  None.-gs, --gsma
-00007930: 702c 2074 6865 2067 656e 6520 6e61 6d65  p, the gene name
-00007940: 2d73 7065 6369 6573 206e 616d 6520 6d61  -species name ma
-00007950: 702c 2064 6566 6175 6c74 204e 6f6e 650a  p, default None.
-00007960: 2d73 702c 202d 2d73 7065 6369 6573 7472  -sp, --speciestr
-00007970: 6565 2c20 7468 6520 7370 6563 6965 7320  ee, the species 
-00007980: 7472 6565 2074 6f20 7065 7266 6f72 6d20  tree to perform 
-00007990: 7261 7465 2063 6f72 7265 6374 696f 6e2c  rate correction,
-000079a0: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
-000079b0: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
-000079c0: 2c20 7468 6520 7261 7465 2063 6f72 7265  , the rate corre
-000079d0: 6374 696f 6e20 616e 616c 7973 6973 2077  ction analysis w
-000079e0: 6f75 6c64 2062 6520 6361 6c6c 6564 206f  ould be called o
-000079f0: 6666 0a2d 706b 2c20 2d2d 706c 6f74 6b64  ff.-pk, --plotkd
-00007a00: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
-00007a10: 7768 6574 6865 7220 746f 2070 6c6f 7420  whether to plot 
-00007a20: 6b64 6520 6375 7276 6520 7570 6f6e 2068  kde curve upon h
-00007a30: 6973 746f 6772 616d 2c20 6966 2074 6865  istogram, if the
-00007a40: 2066 6c61 6720 7761 7320 7365 742c 206b   flag was set, k
-00007a50: 6465 2063 7572 7665 2077 6f75 6c64 2062  de curve would b
-00007a60: 6520 6164 6465 640a 2d72 772c 202d 2d72  e added.-rw, --r
-00007a70: 6577 6569 6768 742c 2066 6c61 6720 6f70  eweight, flag op
-00007a80: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00007a90: 2072 6563 616c 6375 6c61 7465 2074 6865   recalculate the
-00007aa0: 2077 6569 6768 7420 7065 7220 7370 6563   weight per spec
-00007ab0: 6965 7320 7061 6972 2c20 6966 2074 6865  ies pair, if the
-00007ac0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007ad0: 6865 2077 6569 6768 7420 776f 756c 6420  he weight would 
-00007ae0: 6265 2072 6563 616c 6375 6c61 7465 640a  be recalculated.
-00007af0: 2d6f 722c 202d 2d6f 6e6c 7972 6f6f 746f  -or, --onlyrooto
-00007b00: 7574 2c20 666c 6167 206f 7074 696f 6e2c  ut, flag option,
-00007b10: 2077 6865 7468 6572 2074 6f20 6f6e 6c79   whether to only
-00007b20: 2063 6f6e 6475 6374 2072 6174 6520 636f   conduct rate co
-00007b30: 7272 6563 7469 6f6e 2075 7369 6e67 2074  rrection using t
-00007b40: 6865 206f 7574 6772 6f75 7020 6174 2072  he outgroup at r
-00007b50: 6f6f 7420 6173 206f 7574 6772 6f75 702c  oot as outgroup,
-00007b60: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00007b70: 2073 6574 2c20 6f6e 6c79 2074 6865 206f   set, only the o
-00007b80: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
-00007b90: 776f 756c 6420 6265 2075 7365 6420 6173  would be used as
-00007ba0: 206f 7574 6772 6f75 700a 2d69 7465 722c   outgroup.-iter,
-00007bb0: 202d 2d65 6d5f 6974 6572 6174 696f 6e73   --em_iterations
-00007bc0: 2c20 7468 6520 6d61 7869 6d75 6d20 454d  , the maximum EM
-00007bd0: 2069 7465 7261 7469 6f6e 732c 2064 6566   iterations, def
-00007be0: 6175 6c74 2032 3030 0a2d 696e 6974 2c20  ault 200.-init, 
-00007bf0: 2d2d 656d 5f69 6e69 7469 616c 697a 6174  --em_initializat
-00007c00: 696f 6e73 2c20 7468 6520 6d61 7869 6d75  ions, the maximu
-00007c10: 6d20 454d 2069 6e69 7469 616c 697a 6174  m EM initializat
-00007c20: 696f 6e73 2c20 6465 6661 756c 7420 3230  ions, default 20
-00007c30: 300a 2d70 7263 742c 202d 2d70 726f 6d69  0.-prct, --promi
-00007c40: 6e65 6e63 655f 6375 746f 6666 2c20 7468  nence_cutoff, th
-00007c50: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
-00007c60: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
-00007c70: 6520 7065 616b 732c 2064 6566 6175 6c74  e peaks, default
-00007c80: 2030 2e31 0a2d 7268 2c20 2d2d 7265 6c5f   0.1.-rh, --rel_
-00007c90: 6865 6967 6874 2c20 7468 6520 7265 6c61  height, the rela
-00007ca0: 7469 7665 2068 6569 6768 7420 6174 2077  tive height at w
-00007cb0: 6869 6368 2074 6865 2070 6561 6b20 7769  hich the peak wi
-00007cc0: 6474 6820 6973 206d 6561 7375 7265 642c  dth is measured,
-00007cd0: 2064 6566 6175 6c74 2030 2e34 0a2d 736d   default 0.4.-sm
-00007ce0: 2c20 2d2d 7365 676d 656e 7473 2c20 7468  , --segments, th
-00007cf0: 6520 7365 676d 656e 7473 2064 6174 6166  e segments dataf
-00007d00: 696c 652c 2064 6566 6175 6c74 204e 6f6e  ile, default Non
-00007d10: 650a 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c  e.-ml, --minlen,
-00007d20: 2074 6865 206d 696e 696d 756d 206c 656e   the minimum len
-00007d30: 6774 6820 6f66 2061 2073 6361 6666 6f6c  gth of a scaffol
-00007d40: 6420 746f 2062 6520 696e 636c 7564 6564  d to be included
-00007d50: 2069 6e20 646f 7470 6c6f 742c 2064 6566   in dotplot, def
-00007d60: 6175 6c74 202d 312c 2069 6620 2d31 2077  ault -1, if -1 w
-00007d70: 6173 2073 6574 2c20 7468 6520 3130 2520  as set, the 10% 
-00007d80: 6f66 2074 6865 206c 6f6e 6765 7374 2073  of the longest s
-00007d90: 6361 6666 6f6c 6473 2077 696c 6c20 6265  caffolds will be
-00007da0: 2073 6574 0a2d 6d73 2c20 2d2d 6d61 7873   set.-ms, --maxs
-00007db0: 697a 652c 2074 6865 206d 6178 696d 756d  ize, the maximum
-00007dc0: 2066 616d 696c 7920 7369 7a65 2074 6f20   family size to 
-00007dd0: 6265 2069 6e63 6c75 6465 642c 2064 6566  be included, def
-00007de0: 6175 6c74 2032 3030 0a2d 6170 2c20 2d2d  ault 200.-ap, --
-00007df0: 616e 6368 6f72 706f 696e 7473 2c20 7468  anchorpoints, th
-00007e00: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-00007e10: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
-00007e20: 7420 4e6f 6e65 0a2d 6d74 2c20 2d2d 6d75  t None.-mt, --mu
-00007e30: 6c74 6970 6c69 636f 6e2c 2074 6865 206d  ltiplicon, the m
-00007e40: 756c 7469 706c 6963 6f6e 7320 6461 7461  ultiplicons data
-00007e50: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00007e60: 6e65 0a2d 6774 2c20 2d2d 6765 6e65 7461  ne.-gt, --geneta
-00007e70: 626c 652c 2074 6865 2067 656e 6520 7461  ble, the gene ta
-00007e80: 626c 6520 6461 7461 6669 6c65 2c20 6465  ble datafile, de
-00007e90: 6661 756c 7420 4e6f 6e65 0a2d 6d67 2c20  fault None.-mg, 
-00007ea0: 2d2d 6d69 6e73 6567 6c65 6e2c 2074 6865  --minseglen, the
-00007eb0: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
-00007ec0: 6f66 2073 6567 6d65 6e74 7320 746f 2069  of segments to i
-00007ed0: 6e63 6c75 6465 2c20 696e 2072 6174 696f  nclude, in ratio
-00007ee0: 2069 6620 3c3d 2031 2c20 6465 6661 756c   if <= 1, defaul
-00007ef0: 7420 3130 3030 300a 2d6d 676e 2c20 2d2d  t 10000.-mgn, --
-00007f00: 6d69 6e67 656e 656e 756d 2c20 7468 6520  mingenenum, the 
-00007f10: 6d69 6e69 6d75 6d20 6e75 6d62 6572 206f  minimum number o
-00007f20: 6620 6765 6e65 7320 666f 7220 6120 7365  f genes for a se
-00007f30: 676d 656e 7420 746f 2062 6520 636f 6e73  gment to be cons
-00007f40: 6964 6572 6564 2c20 6465 6661 756c 7420  idered, default 
-00007f50: 3330 0a2d 6b72 2c20 2d2d 6b65 6570 7265  30.-kr, --keepre
-00007f60: 6475 6e2c 2066 6c61 6720 6f70 7469 6f6e  dun, flag option
-00007f70: 2c20 7768 6574 6865 7220 746f 206b 6565  , whether to kee
-00007f80: 7020 7265 6475 6e64 616e 7420 6d75 6c74  p redundant mult
-00007f90: 6970 6c69 636f 6e73 2c20 6966 2074 6865  iplicons, if the
-00007fa0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007fb0: 6865 2072 6564 756e 6461 6e74 206d 756c  he redundant mul
-00007fc0: 7469 706c 6963 6f6e 7320 776f 756c 6420  tiplicons would 
-00007fd0: 6265 206b 6570 740a 2d65 706b 2c20 2d2d  be kept.-epk, --
-00007fe0: 6578 7472 6170 6172 616e 6f6d 656b 732c  extraparanomeks,
-00007ff0: 2065 7874 7261 2070 6172 616e 6f6d 6520   extra paranome 
-00008000: 4b73 2064 6174 6120 746f 2070 6c6f 7420  Ks data to plot 
-00008010: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
-00008020: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
-00008030: 6661 756c 7420 4e6f 6e65 0a2d 7061 672c  fault None.-pag,
-00008040: 202d 2d70 6c6f 7461 7067 6d6d 2c20 666c   --plotapgmm, fl
-00008050: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00008060: 6572 2074 6f20 636f 6e64 7563 7420 616e  er to conduct an
-00008070: 6420 706c 6f74 206d 6978 7475 7265 206d  d plot mixture m
-00008080: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
-00008090: 7220 4b73 2069 6e20 7468 6520 6d69 7865  r Ks in the mixe
-000080a0: 6420 4b73 2064 6973 7472 6962 7574 696f  d Ks distributio
-000080b0: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
-000080c0: 6173 2073 6574 2c20 7468 6520 6d69 7874  as set, the mixt
-000080d0: 7572 6520 6d6f 6465 6c69 6e67 206f 6620  ure modeling of 
-000080e0: 616e 6368 6f72 204b 7320 776f 756c 6420  anchor Ks would 
-000080f0: 6265 2063 6f6e 6475 6374 6564 2061 6e64  be conducted and
-00008100: 2070 6c6f 7474 6564 0a2d 7065 6d2c 202d   plotted.-pem, -
-00008110: 2d70 6c6f 7465 6c6d 6d2c 2066 6c61 6720  -plotelmm, flag 
-00008120: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00008130: 746f 2063 6f6e 6475 6374 2061 6e64 2070  to conduct and p
-00008140: 6c6f 7420 656c 6d6d 206d 6978 7475 7265  lot elmm mixture
-00008150: 206d 6f64 656c 696e 6720 6f66 2070 6172   modeling of par
-00008160: 616e 6f6d 6520 4b73 2069 6e20 7468 6520  anome Ks in the 
-00008170: 6d69 7865 6420 4b73 2064 6973 7472 6962  mixed Ks distrib
-00008180: 7574 696f 6e2c 2069 6620 7468 6520 666c  ution, if the fl
-00008190: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-000081a0: 656c 6d6d 206d 6978 7475 7265 206d 6f64  elmm mixture mod
-000081b0: 656c 696e 6720 6f66 2070 6172 616e 6f6d  eling of paranom
-000081c0: 6520 4b73 2077 6f75 6c64 2062 6520 636f  e Ks would be co
-000081d0: 6e64 7563 7465 6420 616e 6420 706c 6f74  nducted and plot
-000081e0: 7465 640a 2d63 2c20 2d2d 636f 6d70 6f6e  ted.-c, --compon
-000081f0: 656e 7473 2c20 7468 6520 7261 6e67 6520  ents, the range 
-00008200: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
-00008210: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
-00008220: 6974 2069 6e20 616e 6368 6f72 204b 7320  it in anchor Ks 
-00008230: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00008240: 2c20 6465 6661 756c 7420 2831 2c34 290a  , default (1,4).
-00008250: 2d70 7379 2c20 2d2d 706c 6f74 7379 6e2c  -psy, --plotsyn,
-00008260: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00008270: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00008280: 6520 7468 6520 7379 6e74 656e 7920 706c  e the synteny pl
-00008290: 6f74 2c20 6f6e 6c79 2077 6865 6e20 7468  ot, only when th
-000082a0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000082b0: 7468 6520 7379 6e74 656e 7920 706c 6f74  the synteny plot
-000082c0: 2077 6f75 6c64 2062 6520 7072 6f64 7563   would be produc
-000082d0: 6564 0a2d 6473 2c20 2d2d 646f 7473 697a  ed.-ds, --dotsiz
-000082e0: 652c 2074 6865 2064 6f74 2073 697a 6520  e, the dot size 
-000082f0: 696e 2064 6f74 2070 6c6f 742c 2064 6566  in dot plot, def
-00008300: 6175 6c74 2030 2e33 0a2d 6161 2c20 2d2d  ault 0.3.-aa, --
-00008310: 6170 616c 7068 612c 2074 6865 206f 7061  apalpha, the opa
-00008320: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
-00008330: 6f74 732c 2064 6566 6175 6c74 2031 0a2d  ots, default 1.-
-00008340: 6861 2c20 2d2d 686f 616c 7068 612c 2074  ha, --hoalpha, t
-00008350: 6865 206f 7061 6369 7479 206f 6620 686f  he opacity of ho
-00008360: 6d6f 6c6f 6720 646f 7473 2c20 6465 6661  molog dots, defa
-00008370: 756c 7420 300a 2d73 7274 2c20 2d2d 7368  ult 0.-srt, --sh
-00008380: 6f77 7265 616c 7469 636b 2c20 666c 6167  owrealtick, flag
-00008390: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000083a0: 2074 6f20 7368 6f77 2074 6865 2072 6561   to show the rea
-000083b0: 6c20 7469 636b 2069 6e20 6765 6e65 7320  l tick in genes 
-000083c0: 6f72 2062 6173 6573 2c20 6966 2074 6865  or bases, if the
-000083d0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-000083e0: 6865 2072 6561 6c20 7469 636b 2077 6f75  he real tick wou
-000083f0: 6c64 2062 6520 7368 6f77 6564 0a2d 746c  ld be showed.-tl
-00008400: 732c 202d 2d74 6963 6b6c 6162 656c 7369  s, --ticklabelsi
-00008410: 7a65 2c20 7468 6520 6c61 6265 6c20 7369  ze, the label si
-00008420: 7a65 206f 6620 7469 636b 2c20 6465 6661  ze of tick, defa
-00008430: 756c 7420 350a 2d78 6c2c 202d 2d78 6c69  ult 5.-xl, --xli
-00008440: 6d2c 2074 6865 2078 2061 7869 7320 6c69  m, the x axis li
-00008450: 6d69 7420 6f66 204b 7320 6469 7374 7269  mit of Ks distri
-00008460: 6275 7469 6f6e 0a2d 796c 2c20 2d2d 796c  bution.-yl, --yl
-00008470: 696d 2c20 7468 6520 7920 6178 6973 206c  im, the y axis l
-00008480: 696d 6974 206f 6620 4b73 2064 6973 7472  imit of Ks distr
-00008490: 6962 7574 696f 6e0a 2d61 646f 2c20 2d2d  ibution.-ado, --
-000084a0: 6164 6a75 7374 6f72 7468 6f2c 2066 6c61  adjustortho, fla
-000084b0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000084c0: 7220 746f 2061 646a 7573 7420 7468 6520  r to adjust the 
-000084d0: 6869 7374 6f67 7261 6d20 6865 6967 6874  histogram height
-000084e0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-000084f0: 4b73 2061 7320 746f 206d 6174 6368 2074  Ks as to match t
-00008500: 6865 2068 6569 6768 7420 6f66 2070 6172  he height of par
-00008510: 616c 6f67 6f75 7320 4b73 2c20 6966 2074  alogous Ks, if t
-00008520: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00008530: 2074 6865 2061 646a 7573 746d 656e 7420   the adjustment 
-00008540: 776f 756c 6420 6265 2063 6f6e 6475 6374  would be conduct
-00008550: 6564 0a2d 6164 662c 202d 2d61 646a 7573  ed.-adf, --adjus
-00008560: 7466 6163 746f 722c 2074 6865 2061 646a  tfactor, the adj
-00008570: 7573 746d 656e 7420 6661 6374 6f72 206f  ustment factor o
-00008580: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
-00008590: 2c20 6465 6661 756c 7420 302e 350a 2d6f  , default 0.5.-o
-000085a0: 612c 202d 2d6f 6b61 6c70 6861 2c20 7468  a, --okalpha, th
-000085b0: 6520 6f70 6163 6974 7920 6f66 206f 7274  e opacity of ort
-000085c0: 686f 6c6f 676f 7573 204b 7320 6469 7374  hologous Ks dist
-000085d0: 7269 6275 7469 6f6e 2069 6e20 6d69 7865  ribution in mixe
-000085e0: 6420 706c 6f74 2c20 6465 6661 756c 7420  d plot, default 
-000085f0: 302e 350a 2d63 732c 202d 2d63 6c61 7373  0.5.-cs, --class
-00008600: 6963 2c20 666c 6167 206f 7074 696f 6e2c  ic, flag option,
-00008610: 2077 6865 7468 6572 2074 6f20 6472 6177   whether to draw
-00008620: 206d 6978 6564 204b 7320 706c 6f74 2069   mixed Ks plot i
-00008630: 6e20 6120 636c 6173 7369 6320 6d61 6e6e  n a classic mann
-00008640: 6572 2077 6865 7265 2074 6865 2066 756c  er where the ful
-00008650: 6c20 6f72 7468 6f6c 6f67 6f75 7320 4b73  l orthologous Ks
-00008660: 2064 6973 7472 6962 7574 696f 6e20 6973   distribution is
-00008670: 2064 7261 7765 642c 2069 6620 7468 6520   drawed, if the 
-00008680: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00008690: 6520 636c 6173 7369 6320 6d69 7865 6420  e classic mixed 
-000086a0: 4b73 2070 6c6f 7420 776f 756c 6420 6265  Ks plot would be
-000086b0: 2064 7261 776e 0a2d 7461 2c20 2d2d 746f   drawn.-ta, --to
-000086c0: 7061 7272 6f77 2c20 666c 6167 206f 7074  parrow, flag opt
-000086d0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000086e0: 6164 6a75 7374 2074 6865 2061 7272 6f77  adjust the arrow
-000086f0: 2074 6f20 6265 2061 7420 7468 6520 746f   to be at the to
-00008700: 7020 6f66 2074 6865 2070 6c6f 742c 2069  p of the plot, i
-00008710: 6e73 7465 6164 206f 6620 6265 696e 6720  nstead of being 
-00008720: 636f 6f72 6469 6e61 7465 6420 6173 2074  coordinated as t
-00008730: 6865 204b 4445 206f 6620 7468 6520 6f72  he KDE of the or
-00008740: 7468 6f6c 6f67 6f75 7320 4b73 2064 6973  thologous Ks dis
-00008750: 7472 6962 7574 696f 6e2c 2069 6620 7468  tribution, if th
-00008760: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00008770: 7468 6520 6172 726f 7720 776f 756c 6420  the arrow would 
-00008780: 6265 2073 6574 2061 7420 7468 6520 746f  be set at the to
-00008790: 700a 2d6e 612c 202d 2d6e 6f64 6561 7665  p.-na, --nodeave
-000087a0: 7261 6765 642c 2066 6c61 6720 6f70 7469  raged, flag opti
-000087b0: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
-000087c0: 7365 206e 6f64 652d 6176 6572 6167 6564  se node-averaged
-000087d0: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
-000087e0: 6564 756e 6461 6e63 792c 2069 6620 7468  edundancy, if th
-000087f0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00008800: 7468 6520 6e6f 6465 2d61 7665 7261 6765  the node-average
-00008810: 6420 6d65 7468 6f64 2077 6f75 6c64 2062  d method would b
-00008820: 6520 696e 6974 6961 7465 640a 2d62 732c  e initiated.-bs,
-00008830: 202d 2d62 6f6f 7473 7472 6170 2c20 7468   --bootstrap, th
-00008840: 6520 6e75 6d62 6572 206f 6620 626f 6f74  e number of boot
-00008850: 7374 7261 7020 7265 706c 6963 6174 6573  strap replicates
-00008860: 206f 6620 6f72 7468 6f6c 6f67 204b 7320   of ortholog Ks 
-00008870: 6469 7374 7269 6275 7469 6f6e 2069 6e20  distribution in 
-00008880: 6d69 7865 6420 706c 6f74 0a2d 6772 2c20  mixed plot.-gr, 
-00008890: 2d2d 6769 7374 7262 2c20 666c 6167 206f  --gistrb, flag o
-000088a0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000088b0: 6f20 7573 6520 6769 7374 5f72 6169 6e62  o use gist_rainb
-000088c0: 6f77 2061 7320 636f 6c6f 7220 6d61 7020  ow as color map 
-000088d0: 6f66 2064 6f74 706c 6f74 0a2d 6e2c 202d  of dotplot.-n, -
-000088e0: 2d6e 7468 7265 6164 732c 2074 6865 206e  -nthreads, the n
-000088f0: 756d 6265 7220 6f66 2074 6872 6561 6473  umber of threads
-00008900: 2074 6f20 7573 6520 696e 2062 6f6f 7473   to use in boots
-00008910: 7472 6170 2073 616d 706c 696e 672c 2064  trap sampling, d
-00008920: 6566 6175 6c74 2031 0a60 6060 0a0a 2323  efault 1.```..##
-00008930: 2055 7361 6765 0a0a 4865 7265 2077 6520   Usage..Here we 
-00008940: 7072 6f76 6964 6564 2074 6865 2062 6173  provided the bas
-00008950: 6963 2075 7361 6765 2066 6f72 2065 6163  ic usage for eac
-00008960: 6820 7072 6f67 7261 6d20 616e 6420 7468  h program and th
-00008970: 6520 7265 6c65 7661 6e74 2070 6172 616d  e relevant param
-00008980: 6574 6572 7320 616e 6420 7375 6767 6573  eters and sugges
-00008990: 7469 6f6e 7320 6f6e 2070 6172 616d 6574  tions on paramet
-000089a0: 6572 697a 6174 696f 6e2e 2041 2072 656d  erization. A rem
-000089b0: 696e 6465 7220 7468 6174 2074 6865 2067  inder that the g
-000089c0: 6976 656e 2063 6f72 6573 2061 6e64 2074  iven cores and t
-000089d0: 6872 6561 6473 2063 616e 2073 6967 6e69  hreads can signi
-000089e0: 6669 6361 6e74 6c79 2069 6d70 6163 7420  ficantly impact 
-000089f0: 7468 6520 7275 6e20 7469 6d65 2061 6e64  the run time and
-00008a00: 2074 6875 7320 7765 2061 6464 6564 2073   thus we added s
-00008a10: 6f6d 6520 7265 706f 7274 2069 6e66 6f72  ome report infor
-00008a20: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-00008a30: 6720 746f 2074 6865 2073 7973 7465 6d20  g to the system 
-00008a40: 6f66 2075 7365 7273 2074 6f20 6661 6369  of users to faci
-00008a50: 6c69 7461 7465 2074 6865 2065 6666 6963  litate the effic
-00008a60: 6965 6e74 2073 6574 7469 6e67 206f 6620  ient setting of 
-00008a70: 7468 7265 6164 7320 616e 6420 6d65 6d6f  threads and memo
-00008a80: 7279 2e20 5468 6520 6c6f 6769 6361 6c20  ry. The logical 
-00008a90: 4350 5573 2072 6570 6f72 7465 6420 7265  CPUs reported re
-00008aa0: 7072 6573 656e 7473 2074 6865 206e 756d  presents the num
-00008ab0: 6265 7220 6f66 2070 6879 7369 6361 6c20  ber of physical 
-00008ac0: 636f 7265 7320 6d75 6c74 6970 6c69 6564  cores multiplied
-00008ad0: 2062 7920 7468 6520 6e75 6d62 6572 206f   by the number o
-00008ae0: 6620 7468 7265 6164 7320 7468 6174 2063  f threads that c
-00008af0: 616e 2072 756e 206f 6e20 6561 6368 2063  an run on each c
-00008b00: 6f72 652c 2061 6c73 6f20 6b6e 6f77 6e20  ore, also known 
-00008b10: 6173 2048 7970 6572 2054 6872 6561 6469  as Hyper Threadi
-00008b20: 6e67 2e20 5468 6520 6e75 6d62 6572 206f  ng. The number o
-00008b30: 6620 6c6f 6769 6361 6c20 4350 5573 206d  f logical CPUs m
-00008b40: 6179 206e 6f74 206e 6563 6573 7361 7269  ay not necessari
-00008b50: 6c79 2062 6520 6571 7569 7661 6c65 6e74  ly be equivalent
-00008b60: 2074 6f20 7468 6520 6163 7475 616c 206e   to the actual n
-00008b70: 756d 6265 7220 6f66 2043 5055 7320 7468  umber of CPUs th
-00008b80: 6520 6375 7272 656e 7420 7072 6f63 6573  e current proces
-00008b90: 7320 6361 6e20 7573 652e 2054 6865 2061  s can use. The a
-00008ba0: 7661 696c 6162 6c65 206d 656d 6f72 7920  vailable memory 
-00008bb0: 7265 6665 7273 2074 6f20 7468 6520 6d65  refers to the me
-00008bc0: 6d6f 7279 2074 6861 7420 6361 6e20 6265  mory that can be
-00008bd0: 2067 6976 656e 2069 6e73 7461 6e74 6c79   given instantly
-00008be0: 2074 6f20 7072 6f63 6573 7365 7320 7769   to processes wi
-00008bf0: 7468 6f75 7420 7468 6520 7379 7374 656d  thout the system
-00008c00: 2067 6f69 6e67 2069 6e74 6f20 7377 6170   going into swap
-00008c10: 2061 6e64 2072 6566 6c65 6374 7320 7468   and reflects th
-00008c20: 6520 6163 7475 616c 206d 656d 6f72 7920  e actual memory 
-00008c30: 6176 6169 6c61 626c 652e 2054 6865 2066  available. The f
-00008c40: 7265 6520 6d65 6d6f 7279 2072 6566 6572  ree memory refer
-00008c50: 7320 746f 2074 6865 206d 656d 6f72 7920  s to the memory 
-00008c60: 6e6f 7420 6265 696e 6720 7573 6564 2061  not being used a
-00008c70: 7420 616c 6c20 287a 6572 6f65 6429 2074  t all (zeroed) t
-00008c80: 6861 7420 6973 2072 6561 6469 6c79 2061  hat is readily a
-00008c90: 7661 696c 6162 6c65 2e20 5468 6520 6465  vailable. The de
-00008ca0: 7363 7269 7074 696f 6e20 6162 6f76 6520  scription above 
-00008cb0: 7265 6665 7273 2074 6f20 7468 6520 646f  refers to the do
-00008cc0: 6375 6d65 6e74 6174 696f 6e20 6f66 2060  cumentation of `
-00008cd0: 7073 7574 696c 602e 0a0a 2323 2320 7767  psutil`...### wg
-00008ce0: 6420 646d 640a 0a2a 2a54 6865 2064 656c  d dmd..**The del
-00008cf0: 696e 6561 7469 6f6e 206f 6620 7768 6f6c  ineation of whol
-00008d00: 6520 7061 7261 6e6f 6d65 2a2a 0a60 6060  e paranome**.```
-00008d10: 0a77 6764 2064 6d64 2041 7175 696c 6567  .wgd dmd Aquileg
-00008d20: 6961 5f63 6f65 7275 6c65 6120 2d49 2032  ia_coerulea -I 2
-00008d30: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-00008d40: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-00008d50: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
-00008d60: 2920 282d 6e20 3429 2028 2d6f 2077 6764  ) (-n 4) (-o wgd
-00008d70: 5f64 6d64 2920 282d 7420 776f 726b 696e  _dmd) (-t workin
-00008d80: 675f 746d 7029 0a60 6060 200a 0a4e 6f74  g_tmp).``` ..Not
-00008d90: 6520 7468 6174 2077 6520 646f 6e27 7420  e that we don't 
-00008da0: 7072 6f76 6964 6520 7468 6520 6461 7461  provide the data
-00008db0: 206f 6620 7468 6973 2063 6f64 696e 6720   of this coding 
-00008dc0: 7365 7175 656e 6365 2028 6364 7329 2066  sequence (cds) f
-00008dd0: 696c 6520 6041 7175 696c 6567 6961 5f63  ile `Aquilegia_c
-00008de0: 6f65 7275 6c65 6160 2062 7574 2069 7420  oerulea` but it 
-00008df0: 6361 6e20 6265 2065 6173 696c 7920 646f  can be easily do
-00008e00: 776e 6c6f 6164 6564 2061 7420 5b50 6879  wnloaded at [Phy
-00008e10: 746f 7a6f 6d65 5d28 6874 7470 733a 2f2f  tozome](https://
-00008e20: 7068 7974 6f7a 6f6d 652d 6e65 7874 2e6a  phytozome-next.j
-00008e30: 6769 2e64 6f65 2e67 6f76 2f69 6e66 6f2f  gi.doe.gov/info/
-00008e40: 4163 6f65 7275 6c65 615f 7633 5f31 2920  Acoerulea_v3_1) 
-00008e50: 2873 616d 6520 666f 7220 6f74 6865 7220  (same for other 
-00008e60: 5573 6167 6520 646f 6329 2e20 4120 7265  Usage doc). A re
-00008e70: 6d69 6e64 6572 2074 6861 7420 696e 2074  minder that in t
-00008e80: 6865 2069 7373 7565 7320 736f 6d65 2075  he issues some u
-00008e90: 7365 7273 2064 6964 6e27 7420 646f 776e  sers didn't down
-00008ea0: 6c6f 6164 2074 6865 2060 4163 6f65 7275  load the `Acoeru
-00008eb0: 6c65 615f 3332 325f 7633 2e31 2e63 6473  lea_322_v3.1.cds
-00008ec0: 5f70 7269 6d61 7279 5472 616e 7363 7269  _primaryTranscri
-00008ed0: 7074 4f6e 6c79 2e66 612e 677a 6020 6275  ptOnly.fa.gz` bu
-00008ee0: 7420 696e 7374 6561 6420 7468 6520 6041  t instead the `A
-00008ef0: 636f 6572 756c 6561 5f33 3232 5f76 332e  coerulea_322_v3.
-00008f00: 312e 6364 732e 6661 2e67 7a60 2066 696c  1.cds.fa.gz` fil
-00008f10: 652e 2046 6f72 2074 6865 2063 6f6e 7374  e. For the const
-00008f20: 7275 6374 696f 6e20 6f66 2077 686f 6c65  ruction of whole
-00008f30: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
-00008f40: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
-00008f50: 6275 7469 6f6e 2c20 2a2a 6f6e 6c79 206f  bution, **only o
-00008f60: 6e65 2074 7261 6e73 6372 6970 7420 2874  ne transcript (t
-00008f70: 6865 2070 7269 6d61 7279 206f 6e65 2920  he primary one) 
-00008f80: 7065 7220 6765 6e65 2073 686f 756c 6420  per gene should 
-00008f90: 6265 2069 6e63 6c75 6465 642a 2a20 7375  be included** su
-00008fa0: 6368 2074 6861 7420 7468 6520 2a4b 2a3c  ch that the *K*<
-00008fb0: 7375 623e 533c 2f73 7562 3e20 6973 2072  sub>S</sub> is r
-00008fc0: 6561 6c6c 7920 696e 6469 6361 7469 6e67  eally indicating
-00008fd0: 2074 6865 2061 6765 206f 6620 6765 6e65   the age of gene
-00008fe0: 2064 7570 6c69 6361 7469 6f6e 2065 7665   duplication eve
-00008ff0: 6e74 2c20 696e 7374 6561 6420 6f66 2061  nt, instead of a
-00009000: 6c74 6572 6e61 7469 7665 2073 706c 6963  lternative splic
-00009010: 696e 672e 2054 7261 6e73 6372 6970 746f  ing. Transcripto
-00009020: 6d65 2064 6174 6120 7368 6f75 6c64 2062  me data should b
-00009030: 6520 6361 7265 6675 6c6c 7920 7472 6561  e carefully trea
-00009040: 7465 6420 7769 7468 2064 652d 7265 6475  ted with de-redu
-00009050: 6e64 616e 6379 2073 6f20 6173 2074 6f20  ndancy so as to 
-00009060: 7265 6475 6365 2074 6865 2066 616c 7365  reduce the false
-00009070: 2070 6f73 6974 6976 6520 6475 706c 6963   positive duplic
-00009080: 6174 696f 6e20 6275 6d70 2063 6175 7365  ation bump cause
-00009090: 6420 6279 2070 6572 7661 7369 7665 2061  d by pervasive a
-000090a0: 6c74 6572 6e61 7469 7665 2073 706c 6963  lternative splic
-000090b0: 696e 672e 2048 6572 6520 7468 6520 696e  ing. Here the in
-000090c0: 666c 6174 696f 6e20 6661 6374 6f72 2070  flation factor p
-000090d0: 6172 616d 6574 6572 2c20 6769 7665 6e20  arameter, given 
-000090e0: 6279 2060 2d49 6020 6f72 2060 2d2d 696e  by `-I` or `--in
-000090f0: 666c 6174 696f 6e60 2c20 6166 6665 6374  flation`, affect
-00009100: 7320 7468 6520 6772 616e 756c 6172 6974  s the granularit
-00009110: 7920 6f72 2072 6573 6f6c 7574 696f 6e20  y or resolution 
-00009120: 6f66 2074 6865 2063 6c75 7374 6572 696e  of the clusterin
-00009130: 6720 6f75 7463 6f6d 6520 616e 6420 696d  g outcome and im
-00009140: 706c 6963 6974 6c79 2063 6f6e 7472 6f6c  plicitly control
-00009150: 6c73 2074 6865 206e 756d 6265 7220 6f66  ls the number of
-00009160: 2063 6c75 7374 6572 732c 2077 6974 6820   clusters, with 
-00009170: 6c6f 7720 7661 6c75 6573 2073 7563 6820  low values such 
-00009180: 6173 2031 2e33 206f 7220 312e 3420 6c65  as 1.3 or 1.4 le
-00009190: 6164 696e 6720 746f 2066 6577 6572 2062  ading to fewer b
-000091a0: 7574 206c 6172 6765 7220 636c 7573 7465  ut larger cluste
-000091b0: 7273 2061 6e64 2068 6967 6820 7661 6c75  rs and high valu
-000091c0: 6573 2073 7563 6820 6173 2035 206f 7220  es such as 5 or 
-000091d0: 3620 6c65 6164 696e 6720 746f 206d 6f72  6 leading to mor
-000091e0: 6520 6275 7420 736d 616c 6c65 7220 636c  e but smaller cl
-000091f0: 7573 7465 7273 2e20 5765 2073 6574 2074  usters. We set t
-00009200: 6865 2064 6566 6175 6c74 2076 616c 7565  he default value
-00009210: 2061 7320 3220 6173 2073 7567 6765 7374   as 2 as suggest
-00009220: 6564 2062 7920 5b4d 434c 5d28 6874 7470  ed by [MCL](http
-00009230: 733a 2f2f 6d69 6361 6e73 2e6f 7267 2f6d  s://micans.org/m
-00009240: 636c 2f29 2e20 5468 6520 652d 7661 6c75  cl/). The e-valu
-00009250: 6520 6375 742d 6f66 6620 666f 7220 7365  e cut-off for se
-00009260: 7175 656e 6365 2073 696d 696c 6172 6974  quence similarit
-00009270: 792c 2067 6976 656e 2062 7920 602d 6560  y, given by `-e`
-00009280: 206f 7220 602d 2d65 7661 6c60 2c20 7768   or `--eval`, wh
-00009290: 6963 6820 6465 6e6f 7465 7320 7468 6520  ich denotes the 
-000092a0: 6578 7065 6374 6564 2076 616c 7565 206f  expected value o
-000092b0: 6620 7468 6520 6869 7420 7175 616e 7469  f the hit quanti
-000092c0: 6669 6573 2074 6865 206e 756d 6265 7220  fies the number 
-000092d0: 6f66 2061 6c69 676e 6d65 6e74 7320 6f66  of alignments of
-000092e0: 2073 696d 696c 6172 206f 7220 6265 7474   similar or bett
-000092f0: 6572 2071 7561 6c69 7479 2074 6861 7420  er quality that 
-00009300: 796f 7520 6578 7065 6374 2074 6f20 6669  you expect to fi
-00009310: 6e64 2073 6561 7263 6869 6e67 2074 6869  nd searching thi
-00009320: 7320 7175 6572 7920 6167 6169 6e73 7420  s query against 
-00009330: 6120 6461 7461 6261 7365 206f 6620 7261  a database of ra
-00009340: 6e64 6f6d 2073 6571 7565 6e63 6573 2074  ndom sequences t
-00009350: 6865 2073 616d 6520 7369 7a65 2061 7320  he same size as 
-00009360: 7468 6520 6163 7475 616c 2074 6172 6765  the actual targe
-00009370: 7420 6461 7461 6261 7365 2c20 6973 2074  t database, is t
-00009380: 6865 206b 6579 2070 6172 616d 6574 6572  he key parameter
-00009390: 206d 6561 7375 7269 6e67 2074 6865 2073   measuring the s
-000093a0: 6967 6e69 6669 6361 6e63 6520 6f66 2061  ignificance of a
-000093b0: 2068 6974 2c20 7768 6963 6820 6973 2073   hit, which is s
-000093c0: 6574 2068 6572 6520 6173 2064 6566 6175  et here as defau
-000093d0: 6c74 2031 652d 3130 2e20 4e6f 7465 2074  lt 1e-10. Note t
-000093e0: 6861 7420 5b44 4941 4d4f 4e44 5d28 6874  hat [DIAMOND](ht
-000093f0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-00009400: 2f62 6275 6368 6669 6e6b 2f64 6961 6d6f  /bbuchfink/diamo
-00009410: 6e64 2f77 696b 692f 312e 2d54 7574 6f72  nd/wiki/1.-Tutor
-00009420: 6961 6c29 2069 7473 656c 6620 6279 2064  ial) itself by d
-00009430: 6566 6175 6c74 206f 6e6c 7920 7265 706f  efault only repo
-00009440: 7274 7320 616c 6c20 616c 6967 6e6d 656e  rts all alignmen
-00009450: 7473 2077 6974 6820 652d 7661 6c75 6520  ts with e-value 
-00009460: 3c20 302e 3030 312e 2054 6865 2070 6572  < 0.001. The per
-00009470: 6365 6e74 6167 6520 6f66 2075 7070 6572  centage of upper
-00009480: 2068 6974 7320 7573 6564 2066 6f72 2067   hits used for g
-00009490: 656e 6520 6c65 6e67 7468 206e 6f72 6d61  ene length norma
-000094a0: 6c69 7a61 7469 6f6e 2c20 6769 7665 6e20  lization, given 
-000094b0: 6279 2060 2d6e 7060 206f 7220 602d 2d6e  by `-np` or `--n
-000094c0: 6f72 6d61 6c69 7a65 6470 6572 6365 6e74  ormalizedpercent
-000094d0: 602c 2077 6869 6368 2064 6574 6572 6d69  `, which determi
-000094e0: 6e65 7320 7468 6520 7570 7065 7220 7065  nes the upper pe
-000094f0: 7263 656e 7469 6c65 206f 6620 6869 7473  rcentile of hits
-00009500: 2070 6572 2062 696e 2028 6361 7465 676f   per bin (catego
-00009510: 7269 7a65 6420 6279 2067 656e 6520 6c65  rized by gene le
-00009520: 6e67 7468 2920 7573 6564 2069 6e20 7468  ngth) used in th
-00009530: 6520 6669 7420 6f66 206c 696e 6561 7220  e fit of linear 
-00009540: 7265 6772 6573 7369 6f6e 2c20 636f 6e73  regression, cons
-00009550: 6964 6572 696e 6720 7468 6174 206e 6f74  idering that not
-00009560: 2061 6c6c 2068 6974 7320 7065 7220 6269   all hits per bi
-00009570: 6e20 7368 6f77 2061 7070 6172 656e 7420  n show apparent 
-00009580: 6c69 6e65 6172 2072 656c 6174 696f 6e73  linear relations
-00009590: 6869 702c 2069 7320 7365 7420 6173 2064  hip, is set as d
-000095a0: 6566 6175 6c74 2035 2c20 696e 6469 6361  efault 5, indica
-000095b0: 7469 6e67 2074 6865 2075 7361 6765 206f  ting the usage o
-000095c0: 6620 746f 7020 3525 2068 6974 7320 7065  f top 5% hits pe
-000095d0: 7220 6269 6e2e 2054 6865 206e 756d 6265  r bin. The numbe
-000095e0: 7220 6f66 2062 696e 7320 6469 7669 6465  r of bins divide
-000095f0: 6420 696e 2067 656e 6520 6c65 6e67 7468  d in gene length
-00009600: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
-00009610: 6769 7665 6e20 6279 2060 2d62 7360 206f  given by `-bs` o
-00009620: 7220 602d 2d62 696e 7360 2c20 6465 7465  r `--bins`, dete
-00009630: 726d 696e 6573 2074 6865 206e 756d 6265  rmines the numbe
-00009640: 7220 6f66 2062 696e 7320 746f 2063 6174  r of bins to cat
-00009650: 6567 6f72 697a 6520 7468 6520 6765 6e65  egorize the gene
-00009660: 206c 656e 6774 682c 2069 7320 7365 7420   length, is set 
-00009670: 6173 2064 6566 6175 6c74 2031 3030 2e20  as default 100. 
-00009680: 5468 6520 7061 7261 6d65 7465 7220 602d  The parameter `-
-00009690: 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f 726d  nn` or `--nonorm
-000096a0: 616c 697a 6174 696f 6e60 2063 616e 2062  alization` can b
-000096b0: 6520 7365 7420 746f 2063 616c 6c20 6f66  e set to call of
-000096c0: 6620 7468 6520 6e6f 726d 616c 697a 6174  f the normalizat
-000096d0: 696f 6e20 7072 6f63 6573 732c 2061 6c74  ion process, alt
-000096e0: 686f 7567 6820 6974 2773 2073 7567 6765  hough it's sugge
-000096f0: 7374 6564 2074 6f20 636f 6e64 7563 7420  sted to conduct 
-00009700: 7468 6520 6e6f 726d 616c 697a 6174 696f  the normalizatio
-00009710: 6e20 746f 2061 6371 7569 7265 206d 6f72  n to acquire mor
-00009720: 6520 6163 6375 7261 7465 2067 656e 6520  e accurate gene 
-00009730: 6661 6d69 6c79 2063 6c75 7374 6572 696e  family clusterin
-00009740: 6720 7265 7375 6c74 2e20 5468 6520 7061  g result. The pa
-00009750: 7261 6d65 7465 7273 2060 2d2d 746f 5f73  rameters `--to_s
-00009760: 746f 7060 2061 6e64 2060 2d2d 6364 7360  top` and `--cds`
-00009770: 2063 6f6e 7472 6f6c 2074 6865 2062 6568   control the beh
-00009780: 6176 696f 7572 206f 6620 7472 616e 736c  aviour of transl
-00009790: 6174 696e 6720 636f 6469 6e67 2073 6571  ating coding seq
-000097a0: 7565 6e63 6520 696e 746f 2061 6d69 6e6f  uence into amino
-000097b0: 2061 6369 6420 7365 7175 656e 6365 2e20   acid sequence. 
-000097c0: 4966 2074 6865 2060 2d2d 746f 5f73 746f  If the `--to_sto
-000097d0: 7060 2077 6173 2073 6574 2c20 7468 6520  p` was set, the 
-000097e0: 7472 616e 736c 6174 696f 6e20 776f 756c  translation woul
-000097f0: 6420 6265 2074 6572 6d69 6e61 7465 6420  d be terminated 
-00009800: 6174 2074 6865 2066 6972 7374 2069 6e2d  at the first in-
-00009810: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
-00009820: 2c20 6f74 6865 7277 6973 6520 7468 6520  , otherwise the 
-00009830: 7472 616e 736c 6174 696f 6e20 776f 756c  translation woul
-00009840: 6420 7369 6d70 6c79 2073 6b69 7020 616e  d simply skip an
-00009850: 7920 7374 6f70 2063 6f64 6f6e 732e 2049  y stop codons. I
-00009860: 6620 7468 6520 602d 2d63 6473 6020 7761  f the `--cds` wa
-00009870: 7320 7365 742c 2073 6571 7565 6e63 6573  s set, sequences
-00009880: 2074 6861 7420 646f 6573 6e27 7420 7374   that doesn't st
-00009890: 6172 7420 7769 7468 2061 2076 616c 6964  art with a valid
-000098a0: 2073 7461 7274 2063 6f64 6f6e 2c20 6f72   start codon, or
-000098b0: 2063 6f6e 7461 696e 7320 6d6f 7265 2074   contains more t
-000098c0: 6861 6e20 6f6e 6520 696e 2d66 7261 6d65  han one in-frame
-000098d0: 2073 746f 7020 636f 646f 6e2c 206f 7220   stop codon, or 
-000098e0: 6973 206e 6f74 2064 6976 6964 6162 6c65  is not dividable
-000098f0: 2062 7920 332c 2077 6f75 6c64 2062 6520   by 3, would be 
-00009900: 7369 6d70 6c79 2064 726f 7070 6564 2c20  simply dropped, 
-00009910: 7375 6368 2074 6861 7420 6f6e 6c79 2073  such that only s
-00009920: 7472 6963 7420 636f 6d70 6c65 7465 2063  trict complete c
-00009930: 6f64 696e 6720 7365 7175 656e 6365 7320  oding sequences 
-00009940: 776f 756c 6420 6265 2069 6e63 6c75 6465  would be include
-00009950: 6420 696e 2074 6865 2073 7562 7365 7175  d in the subsequ
-00009960: 656e 7420 616e 616c 7973 6973 2e20 5468  ent analysis. Th
-00009970: 6520 6578 6163 7420 6265 6861 7669 6f75  e exact behaviou
-00009980: 7220 6f66 2060 2d2d 746f 5f73 746f 7060  r of `--to_stop`
-00009990: 2061 6e64 2060 2d2d 6364 7360 2069 7320   and `--cds` is 
-000099a0: 6465 6669 6e65 6420 616e 6420 6465 7363  defined and desc
-000099b0: 7269 6265 6420 696e 2074 6865 205b 6269  ribed in the [bi
-000099c0: 6f70 7974 686f 6e5d 2868 7474 7073 3a2f  opython](https:/
-000099d0: 2f62 696f 7079 7468 6f6e 2e6f 7267 2f64  /biopython.org/d
-000099e0: 6f63 732f 312e 3735 2f61 7069 2f42 696f  ocs/1.75/api/Bio
-000099f0: 2e53 6571 2e68 746d 6c23 4269 6f2e 5365  .Seq.html#Bio.Se
-00009a00: 712e 5365 712e 7472 616e 736c 6174 6529  q.Seq.translate)
-00009a10: 206c 6962 7261 7279 2e20 5468 6520 6e75   library. The nu
-00009a20: 6d62 6572 206f 6620 7061 7261 6c6c 656c  mber of parallel
-00009a30: 2074 6872 6561 6473 2062 7920 7468 6520   threads by the 
-00009a40: 6f70 7469 6f6e 2060 2d6e 6020 6f72 2060  option `-n` or `
-00009a50: 2d2d 6e74 6872 6561 6473 6020 6361 6e20  --nthreads` can 
-00009a60: 6265 2073 6574 2074 6f20 6163 6365 6c65  be set to accele
-00009a70: 7261 7465 2074 6865 2063 616c 6375 6c61  rate the calcula
-00009a80: 7469 6f6e 2077 6974 6869 6e20 6064 6961  tion within `dia
-00009a90: 6d6f 6e64 602e 2054 6865 2064 6972 6563  mond`. The direc
-00009aa0: 746f 7279 206f 6620 6f75 7470 7574 206f  tory of output o
-00009ab0: 7220 696e 7465 726d 6564 6961 7465 2066  r intermediate f
-00009ac0: 696c 6573 2069 7320 6465 7465 726d 696e  iles is determin
-00009ad0: 6564 2062 7920 7468 6520 7061 7261 6d65  ed by the parame
-00009ae0: 7465 7220 602d 6f60 206f 7220 602d 2d6f  ter `-o` or `--o
-00009af0: 7574 6469 7260 2c20 616e 6420 602d 7460  utdir`, and `-t`
-00009b00: 206f 7220 602d 2d74 6d70 6469 7260 2c20   or `--tmpdir`, 
-00009b10: 7768 6963 6820 7769 6c6c 2062 6520 6372  which will be cr
-00009b20: 6561 7465 6420 6279 2074 6865 2070 726f  eated by the pro
-00009b30: 6772 616d 2069 7473 656c 6620 616e 6420  gram itself and 
-00009b40: 6265 206f 7665 7277 7269 7474 656e 2069  be overwritten i
-00009b50: 6620 7468 6520 666f 6c64 6572 2068 6173  f the folder has
-00009b60: 2061 6c72 6561 6479 2062 6565 6e20 6372   already been cr
-00009b70: 6561 7465 642e 204e 6f74 6520 7468 6174  eated. Note that
-00009b80: 2074 6865 2073 6f66 7477 6172 6520 6064   the software `d
-00009b90: 6961 6d6f 6e64 6020 7368 6f75 6c64 2062  iamond` should b
-00009ba0: 6520 7072 652d 696e 7374 616c 6c65 6420  e pre-installed 
-00009bb0: 616e 6420 7365 7420 696e 2074 6865 2065  and set in the e
-00009bc0: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
-00009bd0: 696e 2061 6c6c 2074 6865 2061 6e61 6c79  in all the analy
-00009be0: 7369 7320 7065 7266 6f72 6d65 6420 6279  sis performed by
-00009bf0: 2060 7767 6420 646d 6460 2065 7863 6570   `wgd dmd` excep
-00009c00: 7420 666f 7220 7468 6520 636f 6c6c 696e  t for the collin
-00009c10: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-00009c20: 616e 616c 7973 6973 2e0a 0a2a 2a57 6520  analysis...**We 
-00009c30: 7375 6767 6573 7420 7468 6174 2074 6865  suggest that the
-00009c40: 2064 6566 6175 6c74 2073 6574 7469 6e67   default setting
-00009c50: 2069 6e20 7768 6963 6820 7468 6520 696e   in which the in
-00009c60: 666c 6174 696f 6e20 6661 6374 6f72 2069  flation factor i
-00009c70: 7320 7365 7420 6173 2032 2c20 652d 7661  s set as 2, e-va
-00009c80: 6c75 6520 6375 742d 6f66 6620 6173 2031  lue cut-off as 1
-00009c90: 652d 3130 2061 6e64 206f 7468 6572 2070  e-10 and other p
-00009ca0: 6172 616d 6574 6572 7320 696e 2064 6566  arameters in def
-00009cb0: 6175 6c74 2069 7320 6120 676f 6f64 2073  ault is a good s
-00009cc0: 7461 7274 696e 6720 706f 696e 742c 2075  tarting point, u
-00009cd0: 6e6c 6573 7320 796f 7520 7370 6563 6966  nless you specif
-00009ce0: 6963 616c 6c79 2077 616e 7420 746f 2065  ically want to e
-00009cf0: 7870 6c6f 7265 2074 6865 2065 6666 6563  xplore the effec
-00009d00: 7473 206f 6620 6469 6666 6572 656e 7420  ts of different 
-00009d10: 7061 7261 6d65 7465 7273 2e2a 2a20 5375  parameters.** Su
-00009d20: 6368 2074 6861 7420 7468 6520 636f 6d6d  ch that the comm
-00009d30: 616e 6420 666f 7220 7468 6520 6465 6c69  and for the deli
-00009d40: 6e65 6174 696f 6e20 6f66 2077 686f 6c65  neation of whole
-00009d50: 2070 6172 616e 6f6d 6520 6973 2073 696d   paranome is sim
-00009d60: 706c 7920 6173 2062 656c 6f77 2e0a 0a60  ply as below...`
-00009d70: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
-00009d80: 6567 6961 5f63 6f65 7275 6c65 610a 6060  egia_coerulea.``
-00009d90: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
-00009da0: 7469 6f6e 206f 6620 5242 4873 2a2a 0a60  tion of RBHs**.`
-00009db0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-00009dc0: 6e63 6531 2073 6571 7565 6e63 6532 202d  nce1 sequence2 -
-00009dd0: 6520 3165 2d31 3020 2d62 7320 3130 3020  e 1e-10 -bs 100 
-00009de0: 2d6e 7020 3520 282d 6e6e 2920 282d 6e20  -np 5 (-nn) (-n 
-00009df0: 3429 2028 2d63 2030 2e39 2920 282d 2d6f  4) (-c 0.9) (--o
-00009e00: 6766 6f72 6d61 7429 2028 2d2d 746f 5f73  gformat) (--to_s
-00009e10: 746f 7029 2028 2d2d 6364 7329 2028 2d6f  top) (--cds) (-o
-00009e20: 2077 6764 5f64 6d64 2920 282d 7420 776f   wgd_dmd) (-t wo
-00009e30: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
-00009e40: 546f 2064 656c 696e 6561 7465 2052 4248  To delineate RBH
-00009e50: 7320 6265 7477 6565 6e20 7477 6f20 6364  s between two cd
-00009e60: 7320 7365 7175 656e 6365 2066 696c 6573  s sequence files
-00009e70: 2c20 7468 6520 7265 6c65 7661 6e74 2070  , the relevant p
-00009e80: 6172 616d 6574 6572 2069 7320 6d6f 7374  arameter is most
-00009e90: 6c79 2074 6865 2073 616d 6520 6173 2077  ly the same as w
-00009ea0: 686f 6c65 2070 6172 616e 6f6d 6520 696e  hole paranome in
-00009eb0: 6665 7265 6e63 652c 2065 7863 6570 7420  ference, except 
-00009ec0: 666f 7220 7468 6520 7061 7261 6d65 7465  for the paramete
-00009ed0: 7220 602d 6360 206f 7220 602d 2d63 7363  r `-c` or `--csc
-00009ee0: 6f72 6560 2c20 7768 6963 6820 7261 6e67  ore`, which rang
-00009ef0: 6573 2062 6574 7765 656e 2030 2061 6e64  es between 0 and
-00009f00: 2031 2061 6e64 2069 7320 7573 6564 2074   1 and is used t
-00009f10: 6f20 7265 6c61 7820 7468 6520 7369 6d69  o relax the simi
-00009f20: 6c61 7269 7479 2063 7574 6f66 6620 6672  larity cutoff fr
-00009f30: 6f6d 2074 6865 2065 7863 6c75 7369 7665  om the exclusive
-00009f40: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
-00009f50: 2068 6974 7320 746f 2061 2063 6572 7461   hits to a certa
-00009f60: 696e 2072 6174 696f 2061 7320 746f 2074  in ratio as to t
-00009f70: 6865 2062 6573 7420 6869 7473 2e20 466f  he best hits. Fo
-00009f80: 7220 696e 7374 616e 6365 2c20 6966 2074  r instance, if t
-00009f90: 6865 2067 656e 6520 6231 2066 726f 6d20  he gene b1 from 
-00009fa0: 6765 6e6f 6d65 2042 2068 6173 2074 6865  genome B has the
-00009fb0: 2062 6573 7420 6869 7420 6765 6e65 2061   best hit gene a
-00009fc0: 3120 6672 6f6d 2067 656e 6f6d 6520 4120  1 from genome A 
-00009fd0: 7769 7468 2074 6865 2062 6974 2073 636f  with the bit sco
-00009fe0: 7265 2061 7320 3130 302c 2077 6869 6368  re as 100, which
-00009ff0: 2069 7320 6120 7363 6f72 696e 6720 6d61   is a scoring ma
-0000a000: 7472 6978 2069 6e64 6570 656e 6465 6e74  trix independent
-0000a010: 206d 6561 7375 7265 206f 6620 7468 6520   measure of the 
-0000a020: 286c 6f63 616c 2920 7369 6d69 6c61 7269  (local) similari
-0000a030: 7479 206f 6620 7468 6520 7477 6f20 616c  ty of the two al
-0000a040: 6967 6e65 6420 7365 7175 656e 6365 732c  igned sequences,
-0000a050: 2077 6974 6820 6c61 7267 6572 2076 616c   with larger val
-0000a060: 7565 7320 696e 6469 6361 7469 6e67 2068  ues indicating h
-0000a070: 6967 6865 7220 7369 6d69 6c61 7269 7469  igher similariti
-0000a080: 6573 2c20 6769 7665 6e20 7468 6520 602d  es, given the `-
-0000a090: 6320 302e 3960 2c20 6765 6e65 7320 6672  c 0.9`, genes fr
-0000a0a0: 6f6d 2067 656e 6f6d 6520 4120 7768 6963  om genome A whic
-0000a0b0: 6820 6861 7320 7468 6520 6269 7420 7363  h has the bit sc
-0000a0c0: 6f72 6520 7769 7468 2067 656e 6520 6231  ore with gene b1
-0000a0d0: 2068 6967 6865 7220 7468 616e 2030 2e39   higher than 0.9
-0000a0e0: 7831 3030 2077 696c 6c20 616c 736f 2062  x100 will also b
-0000a0f0: 6520 7772 6974 7465 6e20 696e 2074 6865  e written in the
-0000a100: 2072 6573 756c 7420 6669 6c65 2c20 7768   result file, wh
-0000a110: 6963 6820 696e 2061 2073 656e 7365 2061  ich in a sense a
-0000a120: 7265 206e 6f74 2052 4248 7320 616e 796d  re not RBHs anym
-0000a130: 6f72 6520 6f66 2063 6f75 7273 652c 2062  ore of course, b
-0000a140: 7574 2074 6865 2068 6967 686c 7920 7369  ut the highly si
-0000a150: 6d69 6c61 7220 686f 6d6f 6c6f 6775 6520  milar homologue 
-0000a160: 7061 6972 732e 2049 6620 6d6f 7265 2074  pairs. If more t
-0000a170: 6861 6e20 3220 7365 7175 656e 6365 2066  han 2 sequence f
-0000a180: 696c 6573 2077 6572 6520 7072 6f76 6964  iles were provid
-0000a190: 6564 2c20 6576 6572 7920 7061 6972 2d77  ed, every pair-w
-0000a1a0: 6973 6520 5242 4873 2077 6f75 6c64 2062  ise RBHs would b
-0000a1b0: 6520 6361 6c63 756c 6174 6564 2065 7863  e calculated exc
-0000a1c0: 6570 7420 666f 7220 7175 6572 7969 6e67  ept for querying
-0000a1d0: 2074 6865 2073 616d 6520 7365 7175 656e   the same sequen
-0000a1e0: 6365 2069 7473 656c 662e 2054 6865 206e  ce itself. The n
-0000a1f0: 756d 6265 7220 6f66 2070 6172 616c 6c65  umber of paralle
-0000a200: 6c20 7468 7265 6164 7320 746f 2062 6f6f  l threads to boo
-0000a210: 7374 6572 2074 6865 2072 756e 6e69 6e67  ster the running
-0000a220: 2073 7065 6564 2063 616e 2062 6520 7365   speed can be se
-0000a230: 7420 6279 2074 6865 206f 7074 696f 6e20  t by the option 
-0000a240: 602d 6e60 206f 7220 602d 2d6e 7468 7265  `-n` or `--nthre
-0000a250: 6164 7360 2077 6869 6368 2069 7320 7375  ads` which is su
-0000a260: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000a270: 7420 6173 2028 4e2d 3129 4e2f 3220 7768  t as (N-1)N/2 wh
-0000a280: 6572 6520 4e20 6973 2074 6865 206e 756d  ere N is the num
-0000a290: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
-0000a2a0: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-0000a2b0: 6869 6768 6573 7420 6566 6669 6369 656e  highest efficien
-0000a2c0: 6379 2e20 5468 6520 6f70 7469 6f6e 2060  cy. The option `
-0000a2d0: 2d2d 6f67 666f 726d 6174 6020 6361 6e20  --ogformat` can 
-0000a2e0: 6265 2073 6574 2074 6f20 6164 6420 696e  be set to add in
-0000a2f0: 6465 7820 2866 6f72 2069 6e73 7461 6e63  dex (for instanc
-0000a300: 6520 4746 3030 3030 3030 3031 2920 746f  e GF00000001) to
-0000a310: 2074 6865 206f 7574 7075 7420 5242 4820   the output RBH 
-0000a320: 6765 6e65 2066 616d 696c 6965 7320 7768  gene families wh
-0000a330: 6963 6820 6361 6e20 6265 2066 7572 7468  ich can be furth
-0000a340: 6572 2075 7365 6420 696e 2074 6865 202a  er used in the *
-0000a350: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2063  K*<sub>S</sub> c
-0000a360: 616c 6375 6c61 7469 6f6e 2062 7920 6077  alculation by `w
-0000a370: 6764 206b 7364 602e 0a0a 2a2a 5468 6520  gd ksd`...**The 
-0000a380: 7375 6767 6573 7465 6420 636f 6d6d 616e  suggested comman
-0000a390: 6420 746f 2073 7461 7274 2077 6974 6820  d to start with 
-0000a3a0: 6973 2061 6c73 6f20 756e 6465 7220 7468  is also under th
-0000a3b0: 6520 6465 6661 756c 7420 7365 7474 696e  e default settin
-0000a3c0: 6720 7769 7468 2074 6865 2063 6f6d 6d61  g with the comma
-0000a3d0: 6e64 2073 686f 776e 2062 656c 6f77 2a2a  nd shown below**
-0000a3e0: 0a0a 6060 600a 7767 6420 646d 6420 7365  ..```.wgd dmd se
-0000a3f0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000a400: 320a 6060 600a 0a2a 2a54 6865 2064 656c  2.```..**The del
-0000a410: 696e 6561 7469 6f6e 206f 6620 6c6f 6361  ineation of loca
-0000a420: 6c20 4d52 4248 732a 2a0a 6060 600a 7767  l MRBHs**.```.wg
-0000a430: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
-0000a440: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000a450: 6365 3320 2d66 2073 6571 7565 6e63 6531  ce3 -f sequence1
-0000a460: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-0000a470: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-0000a480: 6e20 3429 2028 2d6b 6629 2028 2d6b 6429  n 4) (-kf) (-kd)
-0000a490: 2028 2d63 2030 2e39 2920 282d 2d74 6f5f   (-c 0.9) (--to_
-0000a4a0: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
-0000a4b0: 6f20 7767 645f 646d 6429 2028 2d74 2077  o wgd_dmd) (-t w
-0000a4c0: 6f72 6b69 6e67 5f74 6d70 290a 6060 600a  orking_tmp).```.
-0000a4d0: 0a2a 2a54 6865 2064 6973 7469 6e63 7469  .**The distincti
-0000a4e0: 6f6e 2062 6574 7765 656e 206c 6f63 616c  on between local
-0000a4f0: 2061 6e64 2067 6c6f 6261 6c20 2868 6572   and global (her
-0000a500: 6575 6e64 6572 2920 4d52 4248 7320 6973  eunder) MRBHs is
-0000a510: 2074 6861 7420 6c6f 6361 6c20 4d52 4248   that local MRBH
-0000a520: 7320 6172 6520 7468 6520 7265 7375 6c74  s are the result
-0000a530: 7320 6f66 206d 6572 6765 6420 5242 4873  s of merged RBHs
-0000a540: 206f 6e20 6120 6a6f 696e 7420 666f 6361   on a joint foca
-0000a550: 6c20 7370 6563 6965 732c 2066 6f72 2069  l species, for i
-0000a560: 6e73 7461 6e63 6520 696e 2061 2074 6872  nstance in a thr
-0000a570: 6565 2073 7065 6369 6573 2073 7973 7465  ee species syste
-0000a580: 6d20 2841 2c28 422c 4329 292c 2074 6865  m (A,(B,C)), the
-0000a590: 206c 6f63 616c 204d 5242 4873 206f 6620   local MRBHs of 
-0000a5a0: 4120 6f6e 6c79 2072 6571 7569 7265 2074  A only require t
-0000a5b0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
-0000a5c0: 6620 5242 4873 2062 6574 7765 656e 2041  f RBHs between A
-0000a5d0: 2061 6e64 2043 2028 6465 6e6f 7465 6420   and C (denoted 
-0000a5e0: 6173 2041 4329 2061 6e64 2041 4220 616e  as AC) and AB an
-0000a5f0: 6420 7468 656e 2074 6865 206d 6572 6769  d then the mergi
-0000a600: 6e67 206f 6620 4142 2061 6e64 2041 4320  ng of AB and AC 
-0000a610: 6174 2074 6865 2061 7869 7320 6f66 2041  at the axis of A
-0000a620: 2c20 7768 696c 6520 7468 6520 676c 6f61  , while the gloa
-0000a630: 626c 204d 5242 4873 2061 7265 2069 6e64  bl MRBHs are ind
-0000a640: 6570 656e 6465 6e74 206f 6620 666f 6361  ependent of foca
-0000a650: 6c20 7370 6563 6965 7320 696e 2074 6861  l species in tha
-0000a660: 7420 6974 206a 7573 7420 6361 6c6c 7320  t it just calls 
-0000a670: 7468 6520 6361 6c63 756c 6174 696f 6e73  the calculations
-0000a680: 206f 6620 616c 6c20 706f 7373 6962 6c65   of all possible
-0000a690: 2073 7065 6369 6573 2070 6169 7220 286e   species pair (n
-0000a6a0: 6f74 2073 656c 6620 746f 2073 656c 6620  ot self to self 
-0000a6b0: 7370 6563 6965 7320 7061 6972 292c 2073  species pair), s
-0000a6c0: 7563 6820 7468 6174 2041 422c 2041 432c  uch that AB, AC,
-0000a6d0: 2061 6e64 2042 4320 6172 6520 616c 6c20   and BC are all 
-0000a6e0: 746f 2062 6520 6361 6c63 756c 6174 6564  to be calculated
-0000a6f0: 2061 6e64 206d 6572 6765 642e 2a2a 0a0a   and merged.**..
-0000a700: 5477 6f20 7479 7065 7320 6f66 204d 5242  Two types of MRB
-0000a710: 4873 2061 7320 696e 7465 7072 6574 6174  Hs as intepretat
-0000a720: 6564 2061 626f 7665 2063 616e 2062 6520  ed above can be 
-0000a730: 6465 6c69 6e65 6174 6564 2062 7920 6077  delineated by `w
-0000a740: 6764 2064 6d64 602c 2074 6865 206c 6f63  gd dmd`, the loc
-0000a750: 616c 204d 5242 4873 2061 6e64 2074 6865  al MRBHs and the
-0000a760: 2067 6c6f 6261 6c20 4d52 4248 732e 2054   global MRBHs. T
-0000a770: 6865 206c 6f63 616c 204d 5242 4873 2061  he local MRBHs a
-0000a780: 7265 2063 6f6e 7374 7275 6374 6564 2062  re constructed b
-0000a790: 7920 6d65 7267 696e 6720 616c 6c20 7468  y merging all th
-0000a7a0: 6520 7265 6c65 7661 6e74 2052 4248 7320  e relevant RBHs 
-0000a7b0: 6f6e 6c79 2077 6974 6820 7468 6520 666f  only with the fo
-0000a7c0: 6361 6c20 7370 6563 6965 732c 2077 6869  cal species, whi
-0000a7d0: 6368 2069 7320 7365 7420 6279 2060 2d66  ch is set by `-f
-0000a7e0: 6020 6f72 2060 2d2d 666f 6375 7360 2e20  ` or `--focus`. 
-0000a7f0: 5468 6520 7061 7261 6d65 7465 7220 602d  The parameter `-
-0000a800: 6b66 6020 6f72 2060 2d2d 6b65 6570 6661  kf` or `--keepfa
-0000a810: 7374 6160 2063 616e 2062 6520 7365 7420  sta` can be set 
-0000a820: 746f 2072 6574 6169 6e20 7468 6520 7365  to retain the se
-0000a830: 7175 656e 6365 2069 6e66 6f72 6d61 7469  quence informati
-0000a840: 6f6e 206f 6620 6561 6368 204d 5242 482e  on of each MRBH.
-0000a850: 2054 6865 2070 6172 616d 6574 6572 2060   The parameter `
-0000a860: 2d6b 6460 206f 7220 602d 2d6b 6565 7064  -kd` or `--keepd
-0000a870: 7570 6c69 6361 7465 7360 2064 6574 6572  uplicates` deter
-0000a880: 6d69 6e65 7320 7768 6574 6865 7220 7468  mines whether th
-0000a890: 6520 7361 6d65 2067 656e 6573 2063 616e  e same genes can
-0000a8a0: 2061 7070 6561 7220 696e 2064 6966 6665   appear in diffe
-0000a8b0: 7265 6e74 206c 6f63 616c 204d 5242 4873  rent local MRBHs
-0000a8c0: 2e20 4e6f 726d 616c 6c79 2074 6865 7265  . Normally there
-0000a8d0: 2077 696c 6c20 6265 206e 6f20 6475 706c   will be no dupl
-0000a8e0: 6963 6174 6573 2069 6e20 7468 6520 6c6f  icates in the lo
-0000a8f0: 6361 6c20 4d52 4248 7320 6275 7420 6966  cal MRBHs but if
-0000a900: 2075 7365 7273 2073 6574 2074 6865 2060   users set the `
-0000a910: 2d63 6020 6173 2030 2e39 2028 6f72 2061  -c` as 0.9 (or a
-0000a920: 6e79 2076 616c 7565 2073 6d61 6c6c 6572  ny value smaller
-0000a930: 2074 6861 6e20 3129 2c20 6974 2773 206c   than 1), it's l
-0000a940: 696b 656c 7920 7468 6174 2074 6865 2073  ikely that the s
-0000a950: 616d 6520 6765 6e65 2077 696c 6c20 6861  ame gene will ha
-0000a960: 7665 2063 6861 6e63 6520 746f 2061 7070  ve chance to app
-0000a970: 6561 7220 6d75 6c74 6970 6c65 2074 696d  ear multiple tim
-0000a980: 6573 2069 6e20 6469 6666 6572 656e 7420  es in different 
-0000a990: 6c6f 6361 6c20 4d52 4248 732e 2054 6861  local MRBHs. Tha
-0000a9a0: 7420 6973 2074 6f20 7361 792c 2074 6865  t is to say, the
-0000a9b0: 2070 6172 616d 6574 6572 2060 2d6b 6460   parameter `-kd`
-0000a9c0: 2069 7320 6d65 616e 696e 6766 756c 206f   is meaningful o
-0000a9d0: 6e6c 7920 7768 656e 2069 7427 7320 7365  nly when it's se
-0000a9e0: 7420 746f 6765 7468 6572 2077 6974 6820  t together with 
-0000a9f0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
-0000aa00: 6360 2e20 5468 6520 6e75 6d62 6572 206f  c`. The number o
-0000aa10: 6620 7061 7261 6c6c 656c 2074 6872 6561  f parallel threa
-0000aa20: 6473 2069 7320 7375 6767 6573 7465 6420  ds is suggested 
-0000aa30: 746f 2062 6520 7365 7420 6173 2074 6865  to be set as the
-0000aa40: 206e 756d 6265 7220 6f66 2063 6473 2066   number of cds f
-0000aa50: 696c 6573 206d 696e 7573 2031 2e0a 0a2a  iles minus 1...*
-0000aa60: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
-0000aa70: 7274 696e 6720 7275 6e20 6973 2075 6e64  rting run is und
-0000aa80: 6572 2074 6865 2064 6566 6175 6c74 2070  er the default p
-0000aa90: 6172 616d 6574 6572 2077 6974 6820 7468  arameter with th
-0000aaa0: 6520 636f 6d6d 616e 6420 7368 6f77 6e20  e command shown 
-0000aab0: 6173 2062 656c 6f77 2e2a 2a0a 0a60 6060  as below.**..```
-0000aac0: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000aad0: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000aae0: 7565 6e63 6533 202d 6620 7365 7175 656e  uence3 -f sequen
-0000aaf0: 6365 310a 6060 600a 0a2a 2a54 6865 2064  ce1.```..**The d
-0000ab00: 656c 696e 6561 7469 6f6e 206f 6620 676c  elineation of gl
-0000ab10: 6f62 616c 204d 5242 4873 2a2a 0a60 6060  obal MRBHs**.```
-0000ab20: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000ab30: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000ab40: 7565 6e63 6533 202d 676d 202d 6520 3165  uence3 -gm -e 1e
-0000ab50: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
-0000ab60: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
-0000ab70: 2d6b 6629 2028 2d6b 6429 2028 2d63 2030  -kf) (-kd) (-c 0
-0000ab80: 2e39 2920 282d 2d74 6f5f 7374 6f70 2920  .9) (--to_stop) 
-0000ab90: 282d 2d63 6473 2920 282d 6f20 7767 645f  (--cds) (-o wgd_
-0000aba0: 646d 6429 2028 2d74 2077 6f72 6b69 6e67  dmd) (-t working
-0000abb0: 5f74 6d70 290a 6060 600a 0a54 6865 2067  _tmp).```..The g
-0000abc0: 6c6f 6261 6c20 4d52 4248 7320 6973 2063  lobal MRBHs is c
-0000abd0: 6f6e 7374 7275 6374 6564 2062 7920 6578  onstructed by ex
-0000abe0: 6861 7573 7469 7665 6c79 206d 6572 6769  haustively mergi
-0000abf0: 6e67 2061 6c6c 2074 6865 2070 6f73 7369  ng all the possi
-0000ac00: 626c 6520 7061 6972 2d77 6973 6520 5242  ble pair-wise RB
-0000ac10: 4873 2065 7863 6570 7420 666f 7220 7175  Hs except for qu
-0000ac20: 6572 7969 6e67 2074 6865 2073 6571 7565  erying the seque
-0000ac30: 6e63 6520 6974 7365 6c66 2c20 7768 6963  nce itself, whic
-0000ac40: 6820 6361 6e20 6265 2069 6e69 7469 6174  h can be initiat
-0000ac50: 6564 2062 7920 6164 6420 7468 6520 666c  ed by add the fl
-0000ac60: 6167 2060 2d67 6d60 206f 7220 602d 2d67  ag `-gm` or `--g
-0000ac70: 6c6f 6261 6c6d 7262 6860 2e20 5468 6520  lobalmrbh`. The 
-0000ac80: 7265 7374 206f 6620 7265 6c65 7661 6e74  rest of relevant
-0000ac90: 2070 6172 616d 6574 6572 7320 7374 6179   parameters stay
-0000aca0: 7320 7468 6520 7361 6d65 2061 7320 7468  s the same as th
-0000acb0: 6520 6c6f 6361 6c20 4d52 4248 732e 2054  e local MRBHs. T
-0000acc0: 6865 206e 756d 6265 7220 6f66 2070 6172  he number of par
-0000acd0: 616c 6c65 6c20 7468 7265 6164 7320 6973  allel threads is
-0000ace0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000acf0: 2073 6574 2061 7320 284e 2d31 294e 2f32   set as (N-1)N/2
-0000ad00: 2074 6f6f 2077 6865 7265 204e 2069 7320   too where N is 
-0000ad10: 7468 6520 6e75 6d62 6572 206f 6620 6364  the number of cd
-0000ad20: 7320 6669 6c65 7320 746f 2061 6368 6965  s files to achie
-0000ad30: 7665 2074 6865 2068 6967 6865 7374 2065  ve the highest e
-0000ad40: 6666 6963 6965 6e63 792e 0a0a 2a2a 4120  fficiency...**A 
-0000ad50: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
-0000ad60: 6e67 2072 756e 2069 7320 756e 6465 7220  ng run is under 
-0000ad70: 7468 6520 6465 6661 756c 7420 7061 7261  the default para
-0000ad80: 6d65 7465 7220 7769 7468 2074 6865 2063  meter with the c
-0000ad90: 6f6d 6d61 6e64 2073 686f 776e 2061 7320  ommand shown as 
-0000ada0: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
-0000adb0: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
-0000adc0: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000add0: 6365 3320 2d67 6d0a 6060 600a 0a2a 2a54  ce3 -gm.```..**T
-0000ade0: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
-0000adf0: 6620 6f72 7468 6f67 726f 7570 732a 2a0a  f orthogroups**.
-0000ae00: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
-0000ae10: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000ae20: 7365 7175 656e 6365 3320 2d6f 6920 2d6f  sequence3 -oi -o
-0000ae30: 6f20 2d65 2031 652d 3130 202d 6273 2031  o -e 1e-10 -bs 1
-0000ae40: 3030 202d 6e70 2035 2028 2d6e 6e29 2028  00 -np 5 (-nn) (
-0000ae50: 2d63 6329 2028 2d74 6529 2028 2d6d 6320  -cc) (-te) (-mc 
-0000ae60: 302e 3829 2028 2d67 6e29 2028 2d74 7265  0.8) (-gn) (-tre
-0000ae70: 6520 2766 6173 7474 7265 6527 2920 282d  e 'fasttree') (-
-0000ae80: 7473 2027 2d66 6173 7465 7374 2729 2028  ts '-fastest') (
-0000ae90: 2d6e 2034 2920 282d 2d74 6f5f 7374 6f70  -n 4) (--to_stop
-0000aea0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
-0000aeb0: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
-0000aec0: 6e67 5f74 6d70 290a 6060 600a 0a49 6e20  ng_tmp).```..In 
-0000aed0: 6077 6764 2076 3260 2c20 7765 2061 6c73  `wgd v2`, we als
-0000aee0: 6f20 696d 706c 656d 656e 7465 6420 616e  o implemented an
-0000aef0: 2061 6c67 6f72 6974 686d 206f 6620 6465   algorithm of de
-0000af00: 6c69 6e65 6174 696e 6720 6f72 7468 6f67  lineating orthog
-0000af10: 726f 7570 732c 2077 6869 6368 2063 616e  roups, which can
-0000af20: 2062 6520 696e 6974 6961 7465 6420 7769   be initiated wi
-0000af30: 7468 2074 6865 2070 6172 616d 6574 6572  th the parameter
-0000af40: 2060 2d6f 6960 206f 7220 602d 2d6f 7274   `-oi` or `--ort
-0000af50: 686f 696e 6665 7260 2e20 5477 6f20 7761  hoinfer`. Two wa
-0000af60: 7973 206f 6620 6465 6c69 6e65 6174 696f  ys of delineatio
-0000af70: 6e20 6361 6e20 6265 2063 686f 7365 6e2c  n can be chosen,
-0000af80: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
-0000af90: 6f6e 2077 6179 2028 7365 7420 6279 2074  on way (set by t
-0000afa0: 6865 2070 6172 616d 6574 6572 2060 2d63  he parameter `-c
-0000afb0: 6360 206f 7220 602d 2d63 6f6e 6361 7460  c` or `--concat`
-0000afc0: 2920 6f72 2074 6865 206e 6f6e 2d63 6f6e  ) or the non-con
-0000afd0: 6361 7465 6e61 7469 6f6e 2028 6465 6661  catenation (defa
-0000afe0: 756c 7429 2077 6179 2e20 496e 2062 7269  ult) way. In bri
-0000aff0: 6566 2c20 7468 6520 636f 6e63 6174 656e  ef, the concaten
-0000b000: 6174 696f 6e20 7761 7920 6f66 2064 656c  ation way of del
-0000b010: 696e 6561 7469 6e67 206f 7274 686f 6772  ineating orthogr
-0000b020: 6f75 7073 2073 7461 7274 7320 7769 7468  oups starts with
-0000b030: 2063 6f6e 6361 7465 6e61 7469 6e67 2061   concatenating a
-0000b040: 6c6c 2074 6865 2073 6571 7565 6e63 6573  ll the sequences
-0000b050: 2069 6e74 6f20 6120 7369 6e67 6c65 2073   into a single s
-0000b060: 6571 7565 6e63 6520 6669 6c65 2061 6e64  equence file and
-0000b070: 2074 6865 6e20 696e 6665 7272 696e 6720   then inferring 
-0000b080: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
-0000b090: 6d65 206f 6620 7468 6973 2073 696e 676c  me of this singl
-0000b0a0: 6520 7365 7175 656e 6365 2066 696c 6520  e sequence file 
-0000b0b0: 7769 7468 2074 6865 2063 6c75 7374 6572  with the cluster
-0000b0c0: 696e 6720 7265 7375 6c74 7320 6d61 7070  ing results mapp
-0000b0d0: 6564 2062 6163 6b20 746f 2074 6865 2062  ed back to the b
-0000b0e0: 656c 6f6e 6769 6e67 2073 7065 6369 6573  elonging species
-0000b0f0: 2e20 5768 696c 6520 7468 6520 6e6f 6e2d  . While the non-
-0000b100: 636f 6e63 6174 656e 6174 696f 6e20 7761  concatenation wa
-0000b110: 7920 7374 6172 7473 2077 6974 6820 7265  y starts with re
-0000b120: 7370 6563 7469 7665 2070 6169 722d 7769  spective pair-wi
-0000b130: 7365 2064 6961 6d6f 6e64 2073 6561 7263  se diamond searc
-0000b140: 6820 2869 6e63 6c75 6469 6e67 2071 7565  h (including que
-0000b150: 7279 696e 6720 7468 6520 7361 6d65 2073  rying the same s
-0000b160: 6571 7565 6e63 6520 6974 7365 6c66 2920  equence itself) 
-0000b170: 616e 6420 7468 656e 2061 6c6c 2074 6865  and then all the
-0000b180: 2073 6571 7565 6e63 6520 7369 6d69 6c61   sequence simila
-0000b190: 7269 7479 2074 6162 6c65 7320 7769 6c6c  rity tables will
-0000b1a0: 2062 6520 636f 6e63 6174 656e 6174 6564   be concatenated
-0000b1b0: 2061 6e64 2063 6c75 7374 6572 6564 2069   and clustered i
-0000b1c0: 6e74 6f20 6f72 7468 6f67 726f 7570 732e  nto orthogroups.
-0000b1d0: 2053 6f6d 6520 6f74 6865 7220 706f 7373   Some other poss
-0000b1e0: 6962 6c79 2075 7365 6675 6c20 706f 7374  ibly useful post
-0000b1f0: 2d63 6c75 7374 6572 696e 6720 6675 6e63  -clustering func
-0000b200: 7469 6f6e 7320 6361 6e20 6265 2069 6e69  tions can be ini
-0000b210: 7469 6174 6564 2c20 696e 636c 7564 696e  tiated, includin
-0000b220: 6720 7468 6520 7061 7261 6d65 7465 7220  g the parameter 
-0000b230: 602d 7465 6020 6f72 2060 2d2d 7465 7374  `-te` or `--test
-0000b240: 736f 6760 2c20 7768 6963 6820 6361 6e20  sog`, which can 
-0000b250: 6265 2073 6574 2074 6f20 7374 6172 7420  be set to start 
-0000b260: 7468 6520 756e 6269 6173 6564 2074 6573  the unbiased tes
-0000b270: 7420 6f66 2073 696e 676c 652d 636f 7079  t of single-copy
-0000b280: 2067 656e 6520 6661 6d69 6c69 6573 2028   gene families (
-0000b290: 6e6f 7465 2074 6861 7420 7468 6973 2066  note that this f
-0000b2a0: 756e 6374 696f 6e20 6e65 6564 7320 6068  unction needs `h
-0000b2b0: 6d6d 6572 6020 2876 332e 3162 3229 2074  mmer` (v3.1b2) t
-0000b2c0: 6f20 6265 2069 6e73 7461 6c6c 6564 2069  o be installed i
-0000b2d0: 6e20 7468 6520 656e 7669 726f 6e6d 656e  n the environmen
-0000b2e0: 7420 7061 7468 292c 2074 6865 2070 6172  t path), the par
-0000b2f0: 616d 6574 6572 2060 2d6d 6360 206f 7220  ameter `-mc` or 
-0000b300: 602d 2d6d 736f 6763 7574 602c 2072 616e  `--msogcut`, ran
-0000b310: 6769 6e67 2062 6574 7765 656e 2030 2074  ging between 0 t
-0000b320: 6f20 312c 2077 6869 6368 2063 616e 2062  o 1, which can b
-0000b330: 6520 7365 7420 746f 2073 6561 7263 6820  e set to search 
-0000b340: 7468 6520 736f 2d63 616c 6c65 6420 6d6f  the so-called mo
-0000b350: 7374 6c79 2073 696e 676c 652d 636f 7079  stly single-copy
-0000b360: 2066 616d 696c 7920 7768 6963 6820 6861   family which ha
-0000b370: 7320 6869 6768 6572 2074 6861 6e20 6365  s higher than ce
-0000b380: 7274 6169 6e20 6375 742d 6f66 6620 7065  rtain cut-off pe
-0000b390: 7263 656e 7461 6765 206f 6620 7370 6563  rcentage of spec
-0000b3a0: 6965 7320 636f 7665 7261 6765 2c20 7468  ies coverage, th
-0000b3b0: 6520 7061 7261 6d65 7465 7220 602d 676e  e parameter `-gn
-0000b3c0: 6020 6f72 2060 2d2d 6765 746e 736f 6760  ` or `--getnsog`
-0000b3d0: 2c20 7768 6963 6820 6361 6e20 6265 2073  , which can be s
-0000b3e0: 6574 2074 6f20 7365 6172 6368 2066 6f72  et to search for
-0000b3f0: 206e 6573 7465 6420 7369 6e67 6c65 2d63   nested single-c
-0000b400: 6f70 7920 6765 6e65 2066 616d 696c 6965  opy gene familie
-0000b410: 7320 284e 534f 4773 2920 7768 6963 6820  s (NSOGs) which 
-0000b420: 6973 206f 7269 6769 6e61 6c6c 7920 6d75  is originally mu
-0000b430: 6c74 6979 2d63 6f70 7920 6275 7420 6861  ltiy-copy but ha
-0000b440: 7320 6120 286d 6f73 746c 7929 2073 696e  s a (mostly) sin
-0000b450: 676c 652d 636f 7079 2062 7261 6e63 6820  gle-copy branch 
-0000b460: 2877 6869 6368 2072 6571 7569 7265 7320  (which requires 
-0000b470: 7468 6520 6368 6f73 656e 2074 7265 652d  the chosen tree-
-0000b480: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-0000b490: 6d20 7365 7420 6279 2060 2d74 7265 6560  m set by `-tree`
-0000b4a0: 206f 7220 602d 2d74 7265 655f 6d65 7468   or `--tree_meth
-0000b4b0: 6f64 6020 746f 2062 6520 7072 652d 696e  od` to be pre-in
-0000b4c0: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
-0000b4d0: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
-0000b4e0: 7769 7468 2074 6865 2070 6172 616d 6574  with the paramet
-0000b4f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
-0000b500: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-0000b510: 6e63 6520 636f 6e74 726f 6c6c 6564 2062  nce controlled b
-0000b520: 7920 602d 7473 6020 6f72 2060 2d2d 7472  y `-ts` or `--tr
-0000b530: 6565 7365 7460 292e 2054 6865 2070 726f  eeset`). The pro
-0000b540: 6772 616d 2060 7767 6420 646d 6460 2077  gram `wgd dmd` w
-0000b550: 6f75 6c64 2073 7469 6c6c 2063 6f6e 6475  ould still condu
-0000b560: 6374 2074 6865 2052 4248 7320 6361 6c63  ct the RBHs calc
-0000b570: 756c 6174 696f 6e20 756e 6c65 7373 2074  ulation unless t
-0000b580: 6865 2070 6172 616d 6574 6572 2060 2d6f  he parameter `-o
-0000b590: 6f60 206f 7220 602d 2d6f 6e6c 796f 7274  o` or `--onlyort
-0000b5a0: 686f 6020 7761 7320 7365 742e 2049 6620  ho` was set. If 
-0000b5b0: 6f6e 6520 6f6e 6c79 2077 616e 7473 2074  one only wants t
-0000b5c0: 6f20 696e 6665 7220 7468 6520 6f72 7468  o infer the orth
-0000b5d0: 6f67 726f 7570 732c 2069 7427 7320 7375  ogroups, it's su
-0000b5e0: 6767 6573 7465 6420 746f 2061 6464 2074  ggested to add t
-0000b5f0: 6865 2066 6c61 6720 602d 6f6f 6020 746f  he flag `-oo` to
-0000b600: 206a 7573 7420 696d 706c 656d 656e 7420   just implement 
-0000b610: 7468 6520 6f72 7468 6f67 726f 7570 7320  the orthogroups 
-0000b620: 6465 6c69 6e65 6174 696f 6e20 616e 616c  delineation anal
-0000b630: 7973 6973 2e20 5468 6520 6e75 6d62 6572  ysis. The number
-0000b640: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
-0000b650: 6561 6473 2069 7320 7375 6767 6573 7465  eads is suggeste
-0000b660: 6420 746f 2062 6520 7365 7420 6173 2028  d to be set as (
-0000b670: 4e2b 3129 4e2f 3220 7768 6572 6520 4e20  N+1)N/2 where N 
-0000b680: 6973 2074 6865 206e 756d 6265 7220 6f66  is the number of
-0000b690: 2063 6473 2066 696c 6573 2074 6f20 6163   cds files to ac
-0000b6a0: 6869 6576 6520 7468 6520 6869 6768 6573  hieve the highes
-0000b6b0: 7420 6566 6669 6369 656e 6379 2073 696e  t efficiency sin
-0000b6c0: 6365 2074 6865 2073 656c 662d 636f 6d70  ce the self-comp
-0000b6d0: 6172 6973 6f6e 2069 7320 616c 736f 2069  arison is also i
-0000b6e0: 6e63 6c75 6465 642e 0a0a 2a2a 5468 6520  ncluded...**The 
-0000b6f0: 6465 6661 756c 7420 7365 7474 696e 6720  default setting 
-0000b700: 6f66 2070 6172 616d 6574 6572 7320 6973  of parameters is
-0000b710: 2061 2072 6561 736f 6e61 626c 6520 7374   a reasonable st
-0000b720: 6172 7469 6e67 2070 6f69 6e74 2077 6974  arting point wit
-0000b730: 6820 7468 6520 636f 6d6d 616e 6420 6173  h the command as
-0000b740: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
-0000b750: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
-0000b760: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
-0000b770: 6e63 6533 202d 6f69 202d 6f6f 0a60 6060  nce3 -oi -oo.```
-0000b780: 0a0a 2a2a 5468 6520 636f 6c6c 696e 6561  ..**The collinea
-0000b790: 7220 636f 616c 6573 6365 6e63 6520 696e  r coalescence in
-0000b7a0: 6665 7265 6e63 6520 6f66 2070 6879 6c6f  ference of phylo
-0000b7b0: 6765 6e79 2a2a 0a60 6060 0a77 6764 2064  geny**.```.wgd d
-0000b7c0: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
-0000b7d0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000b7e0: 202d 6170 2061 7064 6174 6120 2d73 6d20   -ap apdata -sm 
-0000b7f0: 736d 6461 7461 202d 6c65 206c 6564 6174  smdata -le ledat
-0000b800: 6120 2d67 7420 6774 6461 7461 202d 636f  a -gt gtdata -co
-0000b810: 6320 282d 7472 6565 2027 6661 7374 7472  c (-tree 'fasttr
-0000b820: 6565 2729 2028 2d74 7320 272d 6661 7374  ee') (-ts '-fast
-0000b830: 6573 7427 2920 282d 6e20 3429 2028 2d2d  est') (-n 4) (--
-0000b840: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
-0000b850: 2028 2d6f 2077 6764 5f64 6d64 2920 282d   (-o wgd_dmd) (-
-0000b860: 7420 776f 726b 696e 675f 746d 7029 0a60  t working_tmp).`
-0000b870: 6060 0a0a 4120 6e6f 7665 6c20 7068 796c  ``..A novel phyl
-0000b880: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
-0000b890: 6365 206d 6574 686f 6420 6e61 6d65 6420  ce method named 
-0000b8a0: 2263 6f6c 6c69 6e65 6172 2063 6f61 6c65  "collinear coale
-0000b8b0: 7363 656e 6365 2069 6e66 6572 656e 6365  scence inference
-0000b8c0: 2220 6973 2061 6c73 6f20 696d 706c 656d  " is also implem
-0000b8d0: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
-0000b8e0: 602e 2046 6f72 2074 6869 7320 616e 616c  `. For this anal
-0000b8f0: 7973 6973 2c20 7573 6572 7320 6e65 6564  ysis, users need
-0000b900: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-0000b910: 616e 6368 6f72 2070 6f69 6e74 7320 6669  anchor points fi
-0000b920: 6c65 2062 7920 602d 6170 6020 6f72 2060  le by `-ap` or `
-0000b930: 2d2d 616e 6368 6f72 706f 696e 7473 602c  --anchorpoints`,
-0000b940: 2074 6865 2063 6f6c 6c69 6e65 6172 2073   the collinear s
-0000b950: 6567 6d65 6e74 7320 6669 6c65 2062 7920  egments file by 
-0000b960: 602d 736d 6020 6f72 2060 2d2d 7365 676d  `-sm` or `--segm
-0000b970: 656e 7473 602c 2074 6865 206c 6973 7473  ents`, the lists
-0000b980: 6567 6d65 6e74 7320 6669 6c65 2062 7920  egments file by 
-0000b990: 602d 6c65 6020 6f72 2060 2d2d 6c69 7374  `-le` or `--list
-0000b9a0: 656c 656d 656e 7473 602c 2061 6e64 2074  elements`, and t
-0000b9b0: 6865 2067 656e 6520 7461 626c 6520 6669  he gene table fi
-0000b9c0: 6c65 2062 7920 602d 6774 6020 6f72 2060  le by `-gt` or `
-0000b9d0: 2d2d 6765 6e65 7461 626c 6560 2c20 616c  --genetable`, al
-0000b9e0: 6c20 6f66 2077 6869 6368 2063 616e 2062  l of which can b
-0000b9f0: 6520 7072 6f64 7563 6564 2069 6e20 7468  e produced in th
-0000ba00: 6520 7072 6f67 7261 6d20 6077 6764 2073  e program `wgd s
-0000ba10: 796e 602e 2054 6865 2070 6172 616d 6574  yn`. The paramet
-0000ba20: 6572 2060 2d63 6f63 6020 6f72 2060 2d2d  er `-coc` or `--
-0000ba30: 636f 6c6c 696e 6561 7263 6f61 6c65 7363  collinearcoalesc
-0000ba40: 656e 6365 6020 6e65 6564 7320 746f 2062  ence` needs to b
-0000ba50: 6520 7365 7420 746f 2073 7461 7274 2074  e set to start t
-0000ba60: 6869 7320 616e 616c 7973 6973 2e20 5468  his analysis. Th
-0000ba70: 6520 7472 6565 2d69 6e66 6572 656e 6365  e tree-inference
-0000ba80: 2070 726f 6772 616d 2061 6e64 2074 6865   program and the
-0000ba90: 2061 7373 6f63 6961 7465 6420 7061 7261   associated para
-0000baa0: 6d65 7465 7273 2063 616e 2062 6520 7365  meters can be se
-0000bab0: 7420 6a75 7374 2061 7320 6162 6f76 6520  t just as above 
-0000bac0: 6279 2060 2d74 7265 6560 206f 7220 602d  by `-tree` or `-
-0000bad0: 2d74 7265 655f 6d65 7468 6f64 6020 616e  -tree_method` an
-0000bae0: 6420 602d 7473 6020 6f72 2060 2d2d 7472  d `-ts` or `--tr
-0000baf0: 6565 7365 7460 2e20 506c 6561 7365 2061  eeset`. Please a
-0000bb00: 6c73 6f20 6d61 6b65 2073 7572 6520 7468  lso make sure th
-0000bb10: 6520 6368 6f73 656e 2074 7265 652d 696e  e chosen tree-in
-0000bb20: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
-0000bb30: 6973 2069 6e73 7461 6c6c 6564 2069 6e20  is installed in 
-0000bb40: 7468 6520 656e 7669 726f 6e6d 656e 7420  the environment 
-0000bb50: 7061 7468 2e20 5468 6520 7072 6f67 7261  path. The progra
-0000bb60: 6d20 6061 7374 7261 6c2d 7072 6f60 2069  m `astral-pro` i
-0000bb70: 7320 7265 7175 6972 6564 2074 6f20 6265  s required to be
-0000bb80: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
-0000bb90: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
-0000bba0: 7468 2074 6f6f 2e20 4e6f 7465 2074 6861  th too. Note tha
-0000bbb0: 7420 7468 6572 6520 7368 6f75 6c64 2062  t there should b
-0000bbc0: 6520 6e6f 2064 7570 6c69 6361 7465 6420  e no duplicated 
-0000bbd0: 6765 6e65 2049 4473 2069 6e20 7468 6520  gene IDs in the 
-0000bbe0: 7365 7175 656e 6365 2066 696c 652e 2054  sequence file. T
-0000bbf0: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
-0000bc00: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
-0000bc10: 6163 6365 6c65 7261 7465 2074 6865 2073  accelerate the s
-0000bc20: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
-0000bc30: 7420 616e 6420 6765 6e65 2074 7265 6520  t and gene tree 
-0000bc40: 696e 6665 7265 6e63 6520 666f 7220 6561  inference for ea
-0000bc50: 6368 2067 656e 6520 6661 6d69 6c79 2061  ch gene family a
-0000bc60: 6e64 2074 6875 7320 7375 6767 6573 7465  nd thus suggeste
-0000bc70: 6420 746f 2062 6520 7365 7420 6173 206d  d to be set as m
-0000bc80: 7563 6820 6173 2074 6865 206e 756d 6265  uch as the numbe
-0000bc90: 7220 6f66 2067 656e 6520 6661 6d69 6c69  r of gene famili
-0000bca0: 6573 2e0a 0a2a 2a41 2073 7567 6765 7374  es...**A suggest
-0000bcb0: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-0000bcc0: 6f66 2074 6869 7320 616e 616c 7973 6973  of this analysis
-0000bcd0: 2069 7320 7769 7468 2074 6865 2073 696d   is with the sim
-0000bce0: 706c 6520 636f 6d6d 616e 6420 6265 6c6f  ple command belo
-0000bcf0: 772e 2a2a 0a0a 6060 600a 7767 6420 646d  w.**..```.wgd dm
-0000bd00: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
-0000bd10: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000bd20: 2d61 7020 6170 6461 7461 202d 736d 2073  -ap apdata -sm s
-0000bd30: 6d64 6174 6120 2d6c 6520 6c65 6461 7461  mdata -le ledata
-0000bd40: 202d 6774 2067 7464 6174 6120 2d63 6f63   -gt gtdata -coc
-0000bd50: 0a60 6060 0a0a 2323 2320 7767 6420 666f  .```..### wgd fo
-0000bd60: 6375 730a 0a2a 2a54 6865 2063 6f6e 6361  cus..**The conca
-0000bd70: 7465 6e61 7469 6f6e 2d62 6173 6564 2f63  tenation-based/c
-0000bd80: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
-0000bd90: 2070 6879 6c6f 6765 6e65 7469 6320 696e   phylogenetic in
-0000bda0: 6665 7265 6e63 652a 2a0a 6060 600a 7767  ference**.```.wg
-0000bdb0: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000bdc0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000bdd0: 6e63 6532 2073 6571 7565 6e63 6533 2028  nce2 sequence3 (
-0000bde0: 2d2d 636f 6e63 6174 656e 6174 696f 6e29  --concatenation)
-0000bdf0: 2028 2d2d 636f 616c 6573 6365 6e63 6529   (--coalescence)
-0000be00: 2028 2d74 7265 6520 2766 6173 7474 7265   (-tree 'fasttre
-0000be10: 6527 2920 282d 7473 2027 2d66 6173 7465  e') (-ts '-faste
-0000be20: 7374 2729 2028 2d6e 2034 2920 282d 2d74  st') (-n 4) (--t
-0000be30: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
-0000be40: 282d 6f20 7767 645f 666f 6375 7329 2028  (-o wgd_focus) (
-0000be50: 2d74 2077 6f72 6b69 6e67 5f74 6d70 290a  -t working_tmp).
-0000be60: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
-0000be70: 2060 7767 6420 666f 6375 7360 2069 6d70   `wgd focus` imp
-0000be80: 6c65 6d65 6e74 6564 2074 776f 2062 6173  lemented two bas
-0000be90: 6963 2070 6879 6c6f 6765 6e65 7469 6320  ic phylogenetic 
-0000bea0: 696e 6665 7265 6e63 6520 6d65 7468 6f64  inference method
-0000beb0: 732c 2069 2e65 2e2c 2063 6f6e 6361 7465  s, i.e., concate
-0000bec0: 6e61 7469 6f6e 2d62 6173 6564 2061 6e64  nation-based and
-0000bed0: 2063 6f61 6c65 7363 656e 6365 2d62 6173   coalescence-bas
-0000bee0: 6564 206d 6574 686f 6473 2e20 546f 2069  ed methods. To i
-0000bef0: 6e69 7469 6174 6520 7468 6573 6520 616e  nitiate these an
-0000bf00: 616c 7973 6973 2c20 7573 6572 7320 6e65  alysis, users ne
-0000bf10: 6564 2074 6f20 7365 7420 7468 6520 666c  ed to set the fl
-0000bf20: 6167 206f 7074 696f 6e20 602d 2d63 6f6e  ag option `--con
-0000bf30: 6361 7465 6e61 7469 6f6e 6020 6f72 2060  catenation` or `
-0000bf40: 2d2d 636f 616c 6573 6365 6e63 6560 2e20  --coalescence`. 
-0000bf50: 5468 6520 636f 6e63 6174 656e 6174 696f  The concatenatio
-0000bf60: 6e2d 6261 7365 6420 6d65 7468 6f64 2069  n-based method i
-0000bf70: 6e63 6c75 6465 7320 6120 6665 7720 6d61  ncludes a few ma
-0000bf80: 6a6f 7220 7374 6570 732c 2069 2e65 2e2c  jor steps, i.e.,
-0000bf90: 2074 6865 206d 756c 7469 706c 6520 7365   the multiple se
-0000bfa0: 7175 656e 6365 2061 6c69 676e 6d65 6e74  quence alignment
-0000bfb0: 2028 4d53 4129 206f 6620 6561 6368 2067   (MSA) of each g
-0000bfc0: 656e 6520 6661 6d69 6c79 2c20 7468 6520  ene family, the 
-0000bfd0: 636f 6e63 6174 656e 6174 696f 6e20 6f66  concatenation of
-0000bfe0: 2061 6c6c 2067 656e 6520 6661 6d69 6c69   all gene famili
-0000bff0: 6573 2061 6e64 2074 6865 6e20 7468 6520  es and then the 
-0000c000: 6765 6e65 2074 7265 6520 2861 6c73 6f20  gene tree (also 
-0000c010: 7370 6563 6965 7320 7472 6565 2069 6e20  species tree in 
-0000c020: 7468 6973 2063 6173 6529 2069 6e66 6572  this case) infer
-0000c030: 656e 6365 2e20 5468 6520 636f 616c 6573  ence. The coales
-0000c040: 6365 6e63 652d 6261 7365 6420 6d65 7468  cence-based meth
-0000c050: 6f64 2077 696c 6c20 696e 7374 6561 6420  od will instead 
-0000c060: 7065 7266 6f72 6d20 7468 6520 4d53 4120  perform the MSA 
-0000c070: 6f66 2065 6163 6820 6765 6e65 2066 616d  of each gene fam
-0000c080: 696c 7920 616e 6420 7468 6520 6765 6e65  ily and the gene
-0000c090: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-0000c0a0: 6261 7365 6420 6f6e 2065 6163 6820 4d53  based on each MS
-0000c0b0: 412c 2061 6e64 2074 6865 6e20 696e 6665  A, and then infe
-0000c0c0: 7220 7468 6520 7370 6563 6965 7320 7472  r the species tr
-0000c0d0: 6565 2062 6173 6564 206f 6e20 7468 6573  ee based on thes
-0000c0e0: 6520 696e 6469 7669 6475 616c 2067 656e  e individual gen
-0000c0f0: 6520 7472 6565 732e 2054 6865 2074 7265  e trees. The tre
-0000c100: 652d 696e 6665 7265 6e63 6520 7072 6f67  e-inference prog
-0000c110: 7261 6d20 616e 6420 7468 6520 6173 736f  ram and the asso
-0000c120: 6369 6174 6564 2070 6172 616d 6574 6572  ciated parameter
-0000c130: 7320 6361 6e20 6265 2073 6574 206a 7573  s can be set jus
-0000c140: 7420 6173 2061 626f 7665 2062 7920 602d  t as above by `-
-0000c150: 7472 6565 6020 6f72 2060 2d2d 7472 6565  tree` or `--tree
-0000c160: 5f6d 6574 686f 6460 2061 6e64 2060 2d74  _method` and `-t
-0000c170: 7360 206f 7220 602d 2d74 7265 6573 6574  s` or `--treeset
-0000c180: 602e 2050 6c65 6173 6520 616c 736f 206d  `. Please also m
-0000c190: 616b 6520 7375 7265 2074 6865 2063 686f  ake sure the cho
-0000c1a0: 7365 6e20 7472 6565 2d69 6e66 6572 656e  sen tree-inferen
-0000c1b0: 6365 2070 726f 6772 616d 2069 7320 696e  ce program is in
-0000c1c0: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
-0000c1d0: 6e76 6972 6f6e 6d65 6e74 2070 6174 682e  nvironment path.
-0000c1e0: 2054 6865 2070 726f 6772 616d 2060 6173   The program `as
-0000c1f0: 7472 616c 2d70 726f 6020 6973 2072 6571  tral-pro` is req
-0000c200: 7569 7265 6420 746f 2062 6520 696e 7374  uired to be inst
-0000c210: 616c 6c65 6420 696e 2074 6865 2065 6e76  alled in the env
-0000c220: 6972 6f6e 6d65 6e74 2070 6174 6820 6966  ironment path if
-0000c230: 2074 6865 2063 6f61 6c65 7363 656e 6365   the coalescence
-0000c240: 2d62 6173 6564 206d 6574 686f 6420 6973  -based method is
-0000c250: 2063 686f 7365 6e2e 2054 6865 2070 6172   chosen. The par
-0000c260: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000c270: 7265 2061 7265 2074 6f20 6163 6365 6c65  re are to accele
-0000c280: 7261 7465 2074 6865 2073 6571 7565 6e63  rate the sequenc
-0000c290: 6520 616c 6967 6e6d 656e 7420 616e 6420  e alignment and 
-0000c2a0: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-0000c2b0: 6e63 6520 666f 7220 6561 6368 2067 656e  nce for each gen
-0000c2c0: 6520 6661 6d69 6c79 2074 6f6f 2061 6e64  e family too and
-0000c2d0: 2074 6875 7320 7375 6767 6573 7465 6420   thus suggested 
-0000c2e0: 746f 2062 6520 7365 7420 6173 206d 7563  to be set as muc
-0000c2f0: 6820 6173 2074 6865 206e 756d 6265 7220  h as the number 
-0000c300: 6f66 2067 656e 6520 6661 6d69 6c69 6573  of gene families
-0000c310: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
-0000c320: 2073 7461 7274 696e 6720 7275 6e20 6f66   starting run of
-0000c330: 2074 6869 7320 616e 616c 7973 6973 2069   this analysis i
-0000c340: 7320 7769 7468 2074 6865 2073 696d 706c  s with the simpl
-0000c350: 6520 636f 6d6d 616e 6420 6265 6c6f 772e  e command below.
-0000c360: 2a2a 0a0a 6060 600a 7767 6420 666f 6375  **..```.wgd focu
-0000c370: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-0000c380: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000c390: 6571 7565 6e63 6533 202d 2d63 6f6e 6361  equence3 --conca
-0000c3a0: 7465 6e61 7469 6f6e 0a77 6764 2066 6f63  tenation.wgd foc
-0000c3b0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000c3c0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000c3d0: 7365 7175 656e 6365 3320 2d2d 636f 616c  sequence3 --coal
-0000c3e0: 6573 6365 6e63 650a 6060 600a 0a2a 2a54  escence.```..**T
-0000c3f0: 6865 2066 756e 6374 696f 6e61 6c20 616e  he functional an
-0000c400: 6e6f 7461 7469 6f6e 206f 6620 6765 6e65  notation of gene
-0000c410: 2066 616d 696c 6965 732a 2a0a 6060 600a   families**.```.
-0000c420: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000c430: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000c440: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000c450: 202d 2d61 6e6e 6f74 6174 696f 6e20 6567   --annotation eg
-0000c460: 676e 6f67 202d 6564 2065 6464 6174 6120  gnog -ed eddata 
-0000c470: 2d2d 646d 6e62 2064 6264 6174 610a 6060  --dmnb dbdata.``
-0000c480: 600a 0a54 6865 2070 726f 6772 616d 2060  `..The program `
-0000c490: 7767 6420 666f 6375 7360 2061 6c73 6f20  wgd focus` also 
-0000c4a0: 6164 6465 6420 736f 6d65 2077 7261 7070  added some wrapp
-0000c4b0: 696e 6720 6675 6e63 7469 6f6e 7320 666f  ing functions fo
-0000c4c0: 7220 6675 6e63 7469 6f6e 616c 2061 6e6e  r functional ann
-0000c4d0: 6f74 6174 696f 6e20 6f66 2067 656e 6520  otation of gene 
-0000c4e0: 6661 6d69 6c69 6573 206f 6e20 7468 6520  families on the 
-0000c4f0: 686f 6f64 206f 6620 6461 7461 6261 7365  hood of database
-0000c500: 7320 616e 6420 736f 6674 7761 7265 7320  s and softwares 
-0000c510: 666f 7220 696e 7374 616e 6365 2060 4567  for instance `Eg
-0000c520: 674e 4f47 6020 616e 6420 6045 6767 4e4f  gNOG` and `EggNO
-0000c530: 472d 6d61 7070 6572 602e 2046 6f72 2074  G-mapper`. For t
-0000c540: 6865 2061 6e6e 6f74 6174 696f 6e20 7573  he annotation us
-0000c550: 696e 6720 6045 6767 4e4f 472d 6d61 7070  ing `EggNOG-mapp
-0000c560: 6572 602c 2075 7365 7273 206e 6565 6420  er`, users need 
-0000c570: 746f 2070 726f 7669 6465 2074 6865 2070  to provide the p
-0000c580: 6174 6820 746f 2074 6865 2065 6767 4e4f  ath to the eggNO
-0000c590: 4720 616e 6e6f 7461 7469 6f6e 2064 6174  G annotation dat
-0000c5a0: 6162 6173 6520 7669 6120 7468 6520 6f70  abase via the op
-0000c5b0: 7469 6f6e 2060 2d65 6460 2c20 7468 6520  tion `-ed`, the 
-0000c5c0: 7061 7468 2074 6f20 7468 6520 6469 616d  path to the diam
-0000c5d0: 6f6e 642d 636f 6d70 6174 6962 6c65 2064  ond-compatible d
-0000c5e0: 6174 6162 6173 6520 7669 6120 7468 6520  atabase via the 
-0000c5f0: 6f70 7469 6f6e 2060 2d2d 646d 6e64 5f64  option `--dmnd_d
-0000c600: 6260 2061 6e64 2074 6865 206f 7074 696f  b` and the optio
-0000c610: 6e20 602d 2d61 6e6e 6f74 6174 696f 6e60  n `--annotation`
-0000c620: 2073 6574 2061 7320 2265 6767 6e6f 6722   set as "eggnog"
-0000c630: 2e20 5468 6520 6d61 6e6e 6572 2068 6f77  . The manner how
-0000c640: 2050 4641 4d20 616e 6e6f 7461 7469 6f6e   PFAM annotation
-0000c650: 2077 696c 6c20 6265 2070 6572 666f 726d   will be perform
-0000c660: 6564 2063 616e 2062 6520 636f 6e74 726f  ed can be contro
-0000c670: 6c6c 6564 2076 6961 2074 6865 206f 7074  lled via the opt
-0000c680: 696f 6e20 6070 6661 6d60 2c20 6569 7468  ion `pfam`, eith
-0000c690: 6572 2022 6e6f 6e65 222c 2022 7265 616c  er "none", "real
-0000c6a0: 6967 6e22 206f 7220 2264 656e 6f76 6f22  ign" or "denovo"
-0000c6b0: 2c20 7468 6520 6465 7461 696c 6564 2065  , the detailed e
-0000c6c0: 7870 6c61 6e61 7469 6f6e 2063 616e 2062  xplanation can b
-0000c6d0: 6520 666f 756e 6420 6174 2074 6865 2077  e found at the w
-0000c6e0: 696b 6920 6f66 205b 4567 674e 4f47 2d6d  iki of [EggNOG-m
-0000c6f0: 6170 7065 725d 2868 7474 7073 3a2f 2f67  apper](https://g
-0000c700: 6974 6875 622e 636f 6d2f 6567 676e 6f67  ithub.com/eggnog
-0000c710: 6462 2f65 6767 6e6f 672d 6d61 7070 6572  db/eggnog-mapper
-0000c720: 2f77 696b 6929 2e20 506c 6561 7365 2070  /wiki). Please p
-0000c730: 7265 2d69 6e73 7461 6c6c 2074 6865 2060  re-install the `
-0000c740: 4567 674e 4f47 2d6d 6170 7065 7260 2070  EggNOG-mapper` p
-0000c750: 7974 686f 6e20 7061 636b 6167 6520 6966  ython package if
-0000c760: 2075 7369 6e67 2074 6869 7320 6675 6e63   using this func
-0000c770: 7469 6f6e 2e20 466f 7220 7468 6520 616e  tion. For the an
-0000c780: 6e6f 7461 7469 6f6e 2075 7369 6e67 2060  notation using `
-0000c790: 686d 6d73 6361 6e60 2c20 7768 6174 2069  hmmscan`, what i
-0000c7a0: 7320 696d 706c 656d 656e 7465 6420 696e  s implemented in
-0000c7b0: 2060 7767 6420 7632 6020 6973 2061 2073   `wgd v2` is a s
-0000c7c0: 696d 706c 6520 6275 6e64 6c65 2066 756e  imple bundle fun
-0000c7d0: 6374 696f 6e20 746f 2070 6572 666f 726d  ction to perform
-0000c7e0: 2060 686d 6d73 6361 6e60 2061 6e61 6c79   `hmmscan` analy
-0000c7f0: 7369 7320 666f 7220 6120 6769 7665 6e20  sis for a given 
-0000c800: 686d 6d20 7072 6f66 696c 6520 616e 6420  hmm profile and 
-0000c810: 7365 7420 6f66 2067 656e 6520 6661 6d69  set of gene fami
-0000c820: 6c69 6573 2073 7563 6820 7468 6174 2075  lies such that u
-0000c830: 7365 7273 206e 6565 6420 746f 2073 6574  sers need to set
-0000c840: 2074 6865 206f 7074 696f 6e20 602d 2d61   the option `--a
-0000c850: 6e6e 6f74 6174 696f 6e60 2061 7320 2268  nnotation` as "h
-0000c860: 6d6d 7066 616d 2220 616e 6420 7072 6f76  mmpfam" and prov
-0000c870: 6964 6520 7468 6520 686d 6d70 726f 6669  ide the hmmprofi
-0000c880: 6c65 2076 6961 2074 6865 206f 7074 696f  le via the optio
-0000c890: 6e20 602d 2d68 6d6d 602e 2046 6f72 2074  n `--hmm`. For t
-0000c8a0: 6865 2061 6e6e 6f74 6174 696f 6e20 7573  he annotation us
-0000c8b0: 696e 6720 6069 6e74 6572 7072 6f73 6361  ing `interprosca
-0000c8c0: 6e60 2c20 7573 6572 7320 6e65 6564 2074  n`, users need t
-0000c8d0: 6f20 7072 6f76 6964 6520 7468 6520 7061  o provide the pa
-0000c8e0: 7468 2074 6f20 7468 6520 696e 7465 7270  th to the interp
-0000c8f0: 726f 7363 616e 2069 6e73 7461 6c6c 6174  roscan installat
-0000c900: 696f 6e20 666f 6c64 6572 2077 6865 7265  ion folder where
-0000c910: 2074 6865 7265 2069 7320 6120 6069 6e74   there is a `int
-0000c920: 6572 7072 6f73 6361 6e2e 7368 6020 6669  erproscan.sh` fi
-0000c930: 6c65 2076 6961 2074 6865 206f 7074 696f  le via the optio
-0000c940: 6e20 602d 2d65 7865 7061 7468 6020 616e  n `--exepath` an
-0000c950: 6420 7365 7420 7468 6520 7468 6520 6f70  d set the the op
-0000c960: 7469 6f6e 2060 2d2d 616e 6e6f 7461 7469  tion `--annotati
-0000c970: 6f6e 6020 6173 2022 696e 7465 7270 726f  on` as "interpro
-0000c980: 7363 616e 222e 2054 6865 2070 6172 616c  scan". The paral
-0000c990: 6c65 6c20 7468 7265 6164 7320 6865 7265  lel threads here
-0000c9a0: 2061 7265 2074 6f20 7061 7261 6c6c 656c   are to parallel
-0000c9b0: 697a 6520 7468 6520 616e 616c 7973 6973  ize the analysis
-0000c9c0: 2066 6f72 2065 6163 6820 6765 6e65 2066   for each gene f
-0000c9d0: 616d 696c 7920 616e 6420 7468 7573 2073  amily and thus s
-0000c9e0: 7567 6765 7374 6564 2074 6f20 6265 2073  uggested to be s
-0000c9f0: 6574 2061 7320 6d75 6368 2061 7320 7468  et as much as th
-0000ca00: 6520 6e75 6d62 6572 206f 6620 6765 6e65  e number of gene
-0000ca10: 2066 616d 696c 6965 732e 0a0a 2a2a 4120   families...**A 
-0000ca20: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
-0000ca30: 6e67 2072 756e 206f 6620 7468 6973 2061  ng run of this a
-0000ca40: 6e61 6c79 7369 7320 6361 6e20 6265 2077  nalysis can be w
-0000ca50: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
-0000ca60: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
-0000ca70: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000ca80: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000ca90: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000caa0: 2d61 6e6e 6f74 6174 696f 6e20 6567 676e  -annotation eggn
-0000cab0: 6f67 202d 6564 2065 6464 6174 6120 2d2d  og -ed eddata --
-0000cac0: 646d 6e62 2064 6264 6174 610a 7767 6420  dmnb dbdata.wgd 
-0000cad0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
-0000cae0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
-0000caf0: 6532 2073 6571 7565 6e63 6533 202d 2d61  e2 sequence3 --a
-0000cb00: 6e6e 6f74 6174 696f 6e20 686d 6d70 6661  nnotation hmmpfa
-0000cb10: 6d20 2d2d 686d 6d20 686d 6d64 6174 610a  m --hmm hmmdata.
-0000cb20: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000cb30: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000cb40: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000cb50: 202d 2d61 6e6e 6f74 6174 696f 6e20 696e   --annotation in
-0000cb60: 7465 7270 726f 7363 616e 202d 2d65 7865  terproscan --exe
-0000cb70: 7061 7468 2024 5041 5448 0a60 6060 0a0a  path $PATH.```..
-0000cb80: 2a2a 5468 6520 7068 796c 6f67 656e 6574  **The phylogenet
-0000cb90: 6963 2064 6174 696e 6720 6f66 2057 4744  ic dating of WGD
-0000cba0: 732a 2a0a 6060 600a 7767 6420 666f 6375  s**.```.wgd focu
-0000cbb0: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-0000cbc0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000cbd0: 6571 7565 6e63 6533 202d 6420 6d63 6d63  equence3 -d mcmc
-0000cbe0: 7472 6565 202d 7370 2073 7064 6174 6120  tree -sp spdata 
-0000cbf0: 282d 2d70 726f 7463 6f63 6461 7469 6e67  (--protcocdating
-0000cc00: 2920 282d 2d70 6172 7469 7469 6f6e 2920  ) (--partition) 
-0000cc10: 282d 2d61 616d 6f64 656c 206c 6729 2028  (--aamodel lg) (
-0000cc20: 2d64 7320 2762 7572 6e69 6e20 3d20 3230  -ds 'burnin = 20
-0000cc30: 3030 2729 2028 2d64 7320 2773 616d 7066  00') (-ds 'sampf
-0000cc40: 7265 7120 3d20 3130 3030 2729 2028 2d64  req = 1000') (-d
-0000cc50: 7320 276e 7361 6d70 6c65 203d 2032 3030  s 'nsample = 200
-0000cc60: 3030 2729 2020 282d 6e20 3429 2028 2d2d  00')  (-n 4) (--
-0000cc70: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
-0000cc80: 2028 2d6f 2077 6764 5f66 6f63 7573 2920   (-o wgd_focus) 
-0000cc90: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
-0000cca0: 0a60 6060 0a0a 5468 6520 6162 736f 6c75  .```..The absolu
-0000ccb0: 7465 2064 6174 696e 6720 6f66 2057 4744  te dating of WGD
-0000ccc0: 7320 6973 2061 2073 7065 6369 6669 6320  s is a specific 
-0000ccd0: 7069 7065 6c69 6e65 2069 6d70 6c65 6d65  pipeline impleme
-0000cce0: 6e74 6564 2069 6e20 6077 6764 2076 3260  nted in `wgd v2`
-0000ccf0: 2075 7369 6e67 2074 6865 206d 6574 686f   using the metho
-0000cd00: 6420 6f66 2070 6879 6c6f 6765 6e65 7469  d of phylogeneti
-0000cd10: 6320 6461 7469 6e67 2e20 5468 6520 6661  c dating. The fa
-0000cd20: 6d69 6c69 6573 2075 7365 6420 696e 2074  milies used in t
-0000cd30: 6869 7320 7374 6570 2063 616e 2062 6520  his step can be 
-0000cd40: 7072 6f64 7563 6564 2066 726f 6d20 6077  produced from `w
-0000cd50: 6764 2064 6d64 6020 616e 6420 6077 6764  gd dmd` and `wgd
-0000cd60: 2070 6561 6b60 2e20 4e6f 7465 2074 6861   peak`. Note tha
-0000cd70: 7420 6865 7265 2077 6520 6f6e 6c79 2064  t here we only d
-0000cd80: 6973 6375 7373 2068 6f77 2074 6f20 6461  iscuss how to da
-0000cd90: 7465 2057 4744 7320 7769 7468 2067 656e  te WGDs with gen
-0000cda0: 6f6d 6520 6173 7365 6d62 6c79 2c20 696e  ome assembly, in
-0000cdb0: 7374 6561 6420 6f66 2074 7261 6e73 6372  stead of transcr
-0000cdc0: 6970 746f 6d65 2061 7373 656d 626c 7920  iptome assembly 
-0000cdd0: 2877 6869 6368 2077 696c 6c20 6265 2064  (which will be d
-0000cde0: 6973 6375 7373 6564 2069 6e20 6120 7365  iscussed in a se
-0000cdf0: 7061 7261 7465 2073 6563 7469 6f6e 2068  parate section h
-0000ce00: 6572 6575 6e64 6572 292e 2054 6865 2061  ereunder). The a
-0000ce10: 7373 756d 7074 696f 6e20 7765 206d 6164  ssumption we mad
-0000ce20: 6520 6865 7265 2069 7320 7468 6174 206e  e here is that n
-0000ce30: 6f74 2061 6c6c 2061 6e63 686f 7220 7061  ot all anchor pa
-0000ce40: 6972 7320 2863 6f6c 6c69 6e65 6172 2064  irs (collinear d
-0000ce50: 7570 6c69 6361 7465 7329 2061 7265 2073  uplicates) are s
-0000ce60: 7569 7461 626c 6520 666f 7220 7068 796c  uitable for phyl
-0000ce70: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
-0000ce80: 2066 6f72 2069 6e73 7461 6e63 6520 7468   for instance th
-0000ce90: 6f73 6520 6661 7374 6c79 206f 7220 736c  ose fastly or sl
-0000cea0: 6f77 6c79 2065 766f 6c76 696e 6720 6765  owly evolving ge
-0000ceb0: 6e65 2064 7570 6c69 6361 7465 732c 2062  ne duplicates, b
-0000cec0: 6563 6175 7365 2074 6865 7927 7265 2070  ecause they're p
-0000ced0: 726f 6e65 2074 6f20 6769 7665 2062 6961  rone to give bia
-0000cee0: 7365 6420 6461 7469 6e67 2065 7374 696d  sed dating estim
-0000cef0: 6174 696f 6e2e 2053 6f20 6173 2074 6f20  ation. So as to 
-0000cf00: 7265 7472 6965 7665 2074 6865 2022 7265  retrieve the "re
-0000cf10: 6c69 6162 6c65 2220 616e 6368 6f72 2070  liable" anchor p
-0000cf20: 6169 7273 2c20 7765 2069 6d70 6c65 6d65  airs, we impleme
-0000cf30: 6e74 6564 2073 6f6d 6520 6d65 7468 6f64  nted some method
-0000cf40: 7320 6f66 2069 6465 6e74 6966 7969 6e67  s of identifying
-0000cf50: 2063 7265 6469 6162 6c65 2061 6e63 686f   crediable ancho
-0000cf60: 7220 7061 6972 7320 6261 7365 6420 6f6e  r pairs based on
-0000cf70: 2074 6865 6972 204b 7320 7661 6c75 6573   their Ks values
-0000cf80: 2061 6e64 2f6f 7220 7265 7369 6469 6e67   and/or residing
-0000cf90: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
-0000cfa0: 6e74 732e 2046 6f72 2061 2067 656e 6f6d  nts. For a genom
-0000cfb0: 6520 7769 7468 2063 6c65 6172 2073 6967  e with clear sig
-0000cfc0: 6e61 6c73 206f 6620 7075 7461 7469 7665  nals of putative
-0000cfd0: 2057 4744 732c 2073 7563 6820 6173 2074   WGDs, such as t
-0000cfe0: 6865 202a 4171 7569 6c65 6769 6120 636f  he *Aquilegia co
-0000cff0: 6572 756c 6561 2a20 696e 2074 6865 2065  erulea* in the e
-0000d000: 7861 6d70 6c65 2c20 6120 6865 7572 6973  xample, a heuris
-0000d010: 7469 6320 6d65 7468 6f64 2074 6861 7420  tic method that 
-0000d020: 6170 706c 6965 7320 7468 6520 7072 696e  applies the prin
-0000d030: 6369 706c 6520 6f66 2068 6f77 205b 6b73  ciple of how [ks
-0000d040: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
-0000d050: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
-0000d060: 422f 6b73 7261 7465 7329 2066 696e 6420  B/ksrates) find 
-0000d070: 7468 6520 696e 6974 6961 6c20 7065 616b  the initial peak
-0000d080: 7320 616e 6420 7468 6569 7220 7061 7261  s and their para
-0000d090: 6d65 7465 7273 2077 6173 2069 6d70 6c65  meters was imple
-0000d0a0: 6d65 6e74 6564 2074 6f20 6669 6e64 2074  mented to find t
-0000d0b0: 6865 2039 3525 2063 6f6e 6669 6465 6e63  he 95% confidenc
-0000d0c0: 6520 6c65 7665 6c20 6f66 2074 6865 2061  e level of the a
-0000d0d0: 7373 756d 6564 206c 6f67 6e6f 726d 616c  ssumed lognormal
-0000d0e0: 2064 6973 7472 6962 7574 696f 6e20 6f66   distribution of
-0000d0f0: 2074 6865 2061 6e63 686f 7220 7061 6972   the anchor pair
-0000d100: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d110: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000d120: 6e20 746f 2066 696c 7465 7220 616e 6368  n to filter anch
-0000d130: 6f72 2070 6169 7273 2077 6974 6820 746f  or pairs with to
-0000d140: 6f20 6869 6768 206f 7220 6c6f 7720 2a4b  o high or low *K
-0000d150: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-0000d160: 6573 2c20 7468 6520 6578 616d 706c 6172  es, the examplar
-0000d170: 2063 6f6d 6d61 6e64 206f 6620 7768 6963   command of whic
-0000d180: 6820 6973 2073 686f 7765 6420 696e 2074  h is showed in t
-0000d190: 6865 2060 7767 6420 7065 616b 6020 7365  he `wgd peak` se
-0000d1a0: 6374 696f 6e20 6f66 205b 496c 6c75 7374  ction of [Illust
-0000d1b0: 7261 7469 6f6e 5d28 6874 7470 733a 2f2f  ration](https://
-0000d1c0: 6769 7468 7562 2e63 6f6d 2f68 6563 6865  github.com/heche
-0000d1d0: 2d70 7362 2f77 6764 3f74 6162 3d72 6561  -psb/wgd?tab=rea
-0000d1e0: 646d 652d 6f76 2d66 696c 6523 696c 6c75  dme-ov-file#illu
-0000d1f0: 7374 7261 7469 6f6e 292e 2049 6620 7468  stration). If th
-0000d200: 6973 2068 6575 7269 7374 6963 206d 6574  is heuristic met
-0000d210: 686f 6420 6661 696c 6564 2074 6f20 6769  hod failed to gi
-0000d220: 7665 2072 6561 736f 6e61 626c 6520 7265  ve reasonable re
-0000d230: 7375 6c74 732c 2077 6869 6368 2069 7320  sults, which is 
-0000d240: 7573 7561 6c6c 7920 6475 6520 746f 2074  usually due to t
-0000d250: 6865 2065 6666 6563 7420 6f66 206d 756c  he effect of mul
-0000d260: 7469 706c 6520 6164 6a61 6365 6e74 2057  tiple adjacent W
-0000d270: 4744 7320 7468 6174 2062 6c75 7273 2074  GDs that blurs t
-0000d280: 6865 2070 6561 6b2d 6669 6e64 696e 6720  he peak-finding 
-0000d290: 7072 6f63 6573 732c 2075 7365 7273 2063  process, users c
-0000d2a0: 616e 2074 7572 6e20 746f 2074 6865 2063  an turn to the c
-0000d2b0: 6f6c 6c69 6e65 6172 2073 6567 6d65 6e74  ollinear segment
-0000d2c0: 732d 6775 6964 6564 2061 6e63 686f 7220  s-guided anchor 
-0000d2d0: 7061 6972 2063 6c75 7374 6572 696e 6720  pair clustering 
-0000d2e0: 696d 706c 656d 656e 7465 6420 696e 2060  implemented in `
-0000d2f0: 7767 6420 7632 602c 2069 6e20 7768 6963  wgd v2`, in whic
-0000d300: 6820 6120 636f 6c6c 696e 6561 7220 7365  h a collinear se
-0000d310: 676d 656e 742d 7769 7365 2047 4d4d 2063  gment-wise GMM c
-0000d320: 6c75 7374 6572 696e 6720 7769 6c6c 2062  lustering will b
-0000d330: 6520 6669 7273 7420 636f 6e64 7563 7465  e first conducte
-0000d340: 6420 6261 7365 6420 6f6e 2074 6865 2022  d based on the "
-0000d350: 736f 2d63 616c 6c65 6422 2073 6567 6d65  so-called" segme
-0000d360: 6e74 202a 4b2a 3c73 7562 3e53 3c2f 7375  nt *K*<sub>S</su
-0000d370: 623e 2061 6765 2072 6570 7265 7365 6e74  b> age represent
-0000d380: 6564 2062 7920 7468 6520 6d65 6469 616e  ed by the median
-0000d390: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d3a0: 2061 6765 206f 6620 616c 6c20 7468 6520   age of all the 
-0000d3b0: 7265 7369 6469 6e67 2073 796e 7465 6c6f  residing syntelo
-0000d3c0: 6773 2028 6e6f 7465 2074 6861 7420 7468  gs (note that th
-0000d3d0: 6520 6765 6e65 2064 7570 6c69 6361 7465  e gene duplicate
-0000d3e0: 2070 6169 7273 2061 646f 7074 6564 2069   pairs adopted i
-0000d3f0: 6e20 7468 6973 2073 7465 7020 6973 2066  n this step is f
-0000d400: 726f 6d20 7468 6520 6669 6c65 2060 6d75  rom the file `mu
-0000d410: 6c74 6970 6c69 636f 6e5f 7061 6972 732e  ltiplicon_pairs.
-0000d420: 7478 7460 2077 6869 6368 2063 6f6e 7461  txt` which conta
-0000d430: 696e 7320 7468 6520 6675 6c6c 2073 6574  ins the full set
-0000d440: 206f 6620 7379 6e74 656c 6f67 7329 2c20   of syntelogs), 
-0000d450: 696e 7374 6561 6420 6f66 2074 6865 2073  instead of the s
-0000d460: 6d61 6c6c 6572 2067 656e 6520 7365 7420  maller gene set 
-0000d470: 6f66 2061 6e63 686f 7220 7061 6972 732e  of anchor pairs.
-0000d480: 2054 6865 2064 6973 7469 6e63 7469 6f6e   The distinction
-0000d490: 2062 6574 7765 656e 2061 6e63 686f 7220   between anchor 
-0000d4a0: 7061 6972 7320 616e 6420 7379 6e74 656c  pairs and syntel
-0000d4b0: 6f67 7320 6973 2074 6861 7420 7468 6520  ogs is that the 
-0000d4c0: 6c61 7474 6572 2072 6566 6572 7320 746f  latter refers to
-0000d4d0: 206d 756c 7469 706c 6520 7365 7473 206f   multiple sets o
-0000d4e0: 6620 6765 6e65 7320 6465 7269 7665 6420  f genes derived 
-0000d4f0: 6672 6f6d 2074 6865 2073 616d 6520 616e  from the same an
-0000d500: 6365 7374 7261 6c20 6765 6e6f 6d69 6320  cestral genomic 
-0000d510: 7265 6769 6f6e 2077 6869 6c65 2074 6865  region while the
-0000d520: 2066 6f72 6d65 7220 696d 706c 6965 7320   former implies 
-0000d530: 7468 6520 6c61 7474 6572 2062 7574 2072  the latter but r
-0000d540: 6571 7569 7265 7320 6164 6469 7469 6f6e  equires addition
-0000d550: 616c 6c79 2074 6865 2063 6f6e 7365 7276  ally the conserv
-0000d560: 6564 2067 656e 6520 6f72 6465 722c 2062  ed gene order, b
-0000d570: 6f74 6820 6f66 2077 6869 6368 2061 7265  oth of which are
-0000d580: 2c20 7769 7468 696e 2061 2067 656e 6f6d  , within a genom
-0000d590: 652c 2061 7373 756d 6564 2074 6f20 6265  e, assumed to be
-0000d5a0: 206f 7269 6769 6e61 7465 6420 6672 6f6d   originated from
-0000d5b0: 2074 6865 2064 7570 6c69 6361 7469 6f6e   the duplication
-0000d5c0: 206f 6620 6120 636f 6d6d 6f6e 2061 6e63   of a common anc
-0000d5d0: 6573 7472 616c 2067 656e 6f6d 6963 2072  estral genomic r
-0000d5e0: 6567 696f 6e20 616e 6420 6173 2073 7563  egion and as suc
-0000d5f0: 6820 6465 656d 6564 2065 7669 6465 6e63  h deemed evidenc
-0000d600: 6520 666f 7220 5747 442e 2054 6865 6e20  e for WGD. Then 
-0000d610: 7468 6520 7379 6e74 656c 6f67 7320 7769  the syntelogs wi
-0000d620: 6c6c 2062 6520 6d61 7070 6564 2062 6163  ll be mapped bac
-0000d630: 6b20 6163 636f 7264 696e 6720 746f 2074  k according to t
-0000d640: 6865 2063 6c75 7374 6572 696e 6720 7265  he clustering re
-0000d650: 7375 6c74 7320 6f66 2074 6865 6972 2061  sults of their a
-0000d660: 6666 696c 6961 7465 6420 7365 676d 656e  ffiliated segmen
-0000d670: 7473 2e20 5369 6e63 6520 7468 6520 4761  ts. Since the Ga
-0000d680: 7573 7369 616e 2073 6861 7065 206f 6620  ussian shape of 
-0000d690: 7468 6520 7365 676d 656e 7420 636c 7573  the segment clus
-0000d6a0: 7465 7220 646f 6573 6e27 7420 6e65 6365  ter doesn't nece
-0000d6b0: 7373 6172 696c 7920 696d 6974 6174 6520  ssarily imitate 
-0000d6c0: 7468 6520 7368 6170 6520 6f66 2074 6865  the shape of the
-0000d6d0: 2072 6573 6964 696e 6720 7379 6e74 656c   residing syntel
-0000d6e0: 6f67 732c 2073 6f20 6173 2074 6f20 7265  ogs, so as to re
-0000d6f0: 7472 6965 7665 2074 6865 2022 7265 6c69  trieve the "reli
-0000d700: 6162 6c65 2220 6765 6e65 2073 6574 7320  able" gene sets 
-0000d710: 666f 7220 7068 796c 6f67 656e 6574 6963  for phylogenetic
-0000d720: 2064 6174 696e 672c 2077 6520 6164 706f   dating, we adpo
-0000d730: 7465 6420 7468 6520 2839 3525 2920 6869  ted the (95%) hi
-0000d740: 6768 6573 7420 6465 6e73 6974 7920 7265  ghest density re
-0000d750: 6769 6f6e 2028 4844 5229 206f 6620 7468  gion (HDR) of th
-0000d760: 6520 7379 6e74 656c 6f67 202a 4b2a 3c73  e syntelog *K*<s
-0000d770: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000d780: 6973 7472 6962 7574 696f 6e20 666f 7220  istribution for 
-0000d790: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
-0000d7a0: 2064 6174 696e 672c 2074 6865 2063 616c   dating, the cal
-0000d7b0: 6375 6c61 7469 6f6e 206f 6620 7468 6520  culation of the 
-0000d7c0: 2839 3525 2920 4844 5220 696e 2074 6865  (95%) HDR in the
-0000d7d0: 2066 756e 6374 696f 6e20 5b63 616c 6375   function [calcu
-0000d7e0: 6c61 7465 4850 445d 2868 7474 7073 3a2f  lateHPD](https:/
-0000d7f0: 2f67 6974 6875 622e 636f 6d2f 6865 6368  /github.com/hech
-0000d800: 652d 7073 622f 7767 642f 626c 6f62 2f70  e-psb/wgd/blob/p
-0000d810: 6879 6c6f 6461 7469 6e67 2f77 6764 2f70  hylodating/wgd/p
-0000d820: 6561 6b2e 7079 234c 3137 3537 2920 7365  eak.py#L1757) se
-0000d830: 656b 7320 7468 6520 7368 6f72 7465 7374  eks the shortest
-0000d840: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d850: 2072 616e 6765 2028 612c 6229 2077 6869   range (a,b) whi
-0000d860: 6368 2073 6174 6973 6669 6573 2074 6865  ch satisfies the
-0000d870: 2072 6571 7569 7265 6d65 6e74 206f 6620   requirement of 
-0000d880: 7370 616e 6e69 6e67 206d 6f72 6520 7468  spanning more th
-0000d890: 616e 2028 3935 2529 206f 6620 616c 6c20  an (95%) of all 
-0000d8a0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-0000d8b0: 7562 3e20 7661 6c75 6573 2e20 4f6e 2074  ub> values. On t
-0000d8c0: 6865 2070 7265 6d69 7365 206f 6620 6964  he premise of id
-0000d8d0: 656e 7469 6669 6564 2061 6e63 686f 7220  entified anchor 
-0000d8e0: 7061 6972 7320 286e 6f74 6520 7468 6174  pairs (note that
-0000d8f0: 2074 6865 2073 796e 7465 6c6f 6773 2061   the syntelogs a
-0000d900: 7265 2061 6c73 6f20 7265 6665 7272 6564  re also referred
-0000d910: 2061 7320 616e 6368 6f72 2070 6169 7273   as anchor pairs
-0000d920: 2068 6572 6561 6674 6572 292c 2077 6520   hereafter), we 
-0000d930: 696d 706c 656d 656e 7465 6420 696e 2074  implemented in t
-0000d940: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-0000d950: 646d 6460 2074 6865 2073 6f2d 6361 6c6c  dmd` the so-call
-0000d960: 6564 2061 6e63 686f 722d 6177 6172 6520  ed anchor-aware 
-0000d970: 6c6f 6361 6c20 4d52 4248 7320 6f72 206f  local MRBHs or o
-0000d980: 7274 686f 6772 6f75 7073 2c20 696e 2077  rthogroups, in w
-0000d990: 6869 6368 2074 6865 206f 7269 6769 6e61  hich the origina
-0000d9a0: 6c20 6c6f 6361 6c20 4d52 4248 7320 6172  l local MRBHs ar
-0000d9b0: 6520 6675 7274 6865 7220 6d65 7267 6564  e further merged
-0000d9c0: 2077 6974 6820 616e 6368 6f72 2070 6169   with anchor pai
-0000d9d0: 7273 2073 7563 6820 7468 6174 2065 6163  rs such that eac
-0000d9e0: 6820 6f72 7468 6f67 726f 7570 2063 6f6e  h orthogroup con
-0000d9f0: 7461 696e 7320 7468 6520 616e 6368 6f72  tains the anchor
-0000da00: 2070 6169 7220 616e 6420 7468 6520 6f72   pair and the or
-0000da10: 7468 6f6c 6f67 7565 732e 2057 6974 6820  thologues. With 
-0000da20: 7468 6973 206f 7274 686f 6772 6f75 702c  this orthogroup,
-0000da30: 2075 7365 7273 2074 6865 6e20 6e65 6564   users then need
-0000da40: 206f 6e65 2073 7461 7274 696e 6720 7472   one starting tr
-0000da50: 6565 2066 696c 6520 2861 7320 7368 6f77  ee file (as show
-0000da60: 6e20 696e 2074 6865 205b 496c 6c75 7374  n in the [Illust
-0000da70: 7261 7469 6f6e 5d28 6874 7470 733a 2f2f  ration](https://
-0000da80: 6769 7468 7562 2e63 6f6d 2f68 6563 6865  github.com/heche
-0000da90: 2d70 7362 2f77 6764 2f74 7265 652f 7068  -psb/wgd/tree/ph
-0000daa0: 796c 6f64 6174 696e 673f 7461 623d 7265  ylodating?tab=re
-0000dab0: 6164 6d65 2d6f 762d 6669 6c65 2369 6c6c  adme-ov-file#ill
-0000dac0: 7573 7472 6174 696f 6e29 2070 6172 7429  ustration) part)
-0000dad0: 2069 6e64 6963 6174 696e 6720 7468 6520   indicating the 
-0000dae0: 7472 6565 2074 6f70 6f6c 6f67 7920 616e  tree topology an
-0000daf0: 6420 666f 7373 696c 2063 616c 6962 7261  d fossil calibra
-0000db00: 7469 6f6e 2069 6e66 6f72 6d61 7469 6f6e  tion information
-0000db10: 2066 6f72 2074 6865 2066 696e 616c 2070   for the final p
-0000db20: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
-0000db30: 6e67 2075 7369 6e67 2074 6865 2070 726f  ng using the pro
-0000db40: 6772 616d 2060 7767 6420 666f 6375 7360  gram `wgd focus`
-0000db50: 2e20 5573 6572 7320 6361 6e20 7365 7420  . Users can set 
-0000db60: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-0000db70: 602d 2d70 726f 7463 6f63 6461 7469 6e67  `--protcocdating
-0000db80: 6020 746f 206f 6e6c 7920 636f 6e64 7563  ` to only conduc
-0000db90: 7420 7468 6520 6461 7469 6e67 206f 6620  t the dating of 
-0000dba0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
-0000dbb0: 6e20 7072 6f74 6569 6e20 4d53 4120 6f72  n protein MSA or
-0000dbc0: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
-0000dbd0: 2060 2d2d 7072 6f74 6461 7469 6e67 6020   `--protdating` 
-0000dbe0: 746f 206f 6e6c 7920 636f 6e64 7563 7420  to only conduct 
-0000dbf0: 7468 6520 6461 7469 6e67 206f 6620 7468  the dating of th
-0000dc00: 6520 7072 6f74 6569 6e20 4d53 4173 2c20  e protein MSAs, 
-0000dc10: 6e6f 7465 6420 7468 6174 2074 6865 7365  noted that these
-0000dc20: 2074 776f 206f 7074 696f 6e73 206f 6e6c   two options onl
-0000dc30: 7920 776f 726b 2066 6f72 2074 6865 2060  y work for the `
-0000dc40: 6d63 6d63 7472 6565 6020 6f70 7469 6f6e  mcmctree` option
-0000dc50: 2073 6f20 6661 722e 2054 6865 2066 6c61   so far. The fla
-0000dc60: 6720 6f70 7469 6f6e 2060 2d2d 7061 7274  g option `--part
-0000dc70: 6974 696f 6e60 2063 616e 2062 6520 7365  ition` can be se
-0000dc80: 7420 746f 2070 6572 666f 726d 2060 6d63  t to perform `mc
-0000dc90: 6d63 7472 6565 6020 616e 616c 7973 6973  mctree` analysis
-0000dca0: 2075 7369 6e67 2074 6865 2070 6172 7469   using the parti
-0000dcb0: 7469 6f6e 6564 2064 6174 6120 2869 2e65  tioned data (i.e
-0000dcc0: 2e2c 2031 7374 2c20 326e 6420 616e 6420  ., 1st, 2nd and 
-0000dcd0: 3372 6420 706f 7369 7469 6f6e 206f 6620  3rd position of 
-0000dce0: 636f 646f 6e29 2069 6e73 7465 6164 206f  codon) instead o
-0000dcf0: 6620 7573 696e 6720 7468 6520 636f 646f  f using the codo
-0000dd00: 6e20 6173 2061 2077 686f 6c65 2e20 5468  n as a whole. Th
-0000dd10: 6520 6f70 7469 6f6e 2060 2d2d 6161 6d6f  e option `--aamo
-0000dd20: 6465 6c60 2063 616e 2062 6520 7365 7420  del` can be set 
-0000dd30: 746f 2064 6574 6572 6d69 6e65 2074 6865  to determine the
-0000dd40: 2061 6d69 6e6f 2061 6369 6420 6d6f 6465   amino acid mode
-0000dd50: 6c20 6170 706c 6965 6420 696e 2060 6d63  l applied in `mc
-0000dd60: 6d63 7472 6565 6020 616e 616c 7973 6973  mctree` analysis
-0000dd70: 2e20 5468 6520 6f70 7469 6f6e 2060 2d64  . The option `-d
-0000dd80: 7360 2063 616e 2062 6520 7573 6564 2074  s` can be used t
-0000dd90: 6f20 7365 7420 7061 7261 6d65 7465 7273  o set parameters
-0000dda0: 2066 6f72 2074 6865 206d 6f6c 6563 756c   for the molecul
-0000ddb0: 6172 2064 6174 696e 6720 7072 6f67 7261  ar dating progra
-0000ddc0: 6d2e 2054 6865 2070 6172 616c 6c65 6c20  m. The parallel 
-0000ddd0: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
-0000dde0: 2074 6f20 7061 7261 6c6c 656c 697a 6520   to parallelize 
-0000ddf0: 7468 6520 616e 616c 7973 6973 2066 6f72  the analysis for
-0000de00: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
-0000de10: 7920 616e 6420 7468 7573 2073 7567 6765  y and thus sugge
-0000de20: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
-0000de30: 7320 6d75 6368 2061 7320 7468 6520 6e75  s much as the nu
-0000de40: 6d62 6572 206f 6620 6765 6e65 2066 616d  mber of gene fam
-0000de50: 696c 6965 732e 0a0a 2a2a 4120 7375 6767  ilies...**A sugg
-0000de60: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-0000de70: 756e 2063 616e 2062 6520 7769 7468 2074  un can be with t
-0000de80: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-0000de90: 2e2a 2a0a 0a60 6060 0a77 6764 2066 6f63  .**..```.wgd foc
-0000dea0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000deb0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000dec0: 7365 7175 656e 6365 3320 2d64 206d 636d  sequence3 -d mcm
-0000ded0: 6374 7265 6520 2d73 7020 7370 6461 7461  ctree -sp spdata
-0000dee0: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
-0000def0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000df00: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000df10: 3320 2d64 2072 3873 202d 7370 2073 7064  3 -d r8s -sp spd
-0000df20: 6174 6120 2d2d 6e73 6974 6573 206e 7369  ata --nsites nsi
-0000df30: 7465 696e 666f 0a77 6764 2066 6f63 7573  teinfo.wgd focus
-0000df40: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000df50: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
-0000df60: 7175 656e 6365 3320 2d64 2062 6561 7374  quence3 -d beast
-0000df70: 202d 2d66 6f73 7369 6c20 666f 7373 696c   --fossil fossil
-0000df80: 696e 666f 202d 2d72 6f6f 7468 6569 6768  info --rootheigh
-0000df90: 7420 726f 6f74 6865 6967 6874 696e 666f  t rootheightinfo
-0000dfa0: 202d 2d63 6861 696e 7365 7420 6368 6169   --chainset chai
-0000dfb0: 6e73 6574 696e 666f 202d 2d62 6561 7374  nsetinfo --beast
-0000dfc0: 6c67 6a61 7220 2450 4154 480a 6060 600a  lgjar $PATH.```.
-0000dfd0: 0a23 2323 2077 6764 206b 7364 0a0a 2a2a  .### wgd ksd..**
-0000dfe0: 5468 6520 636f 6e73 7472 7563 7469 6f6e  The construction
-0000dff0: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
-0000e000: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
-0000e010: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000e020: 696f 6e2a 2a0a 6060 600a 7767 6420 6b73  ion**.```.wgd ks
-0000e030: 6420 6661 6d69 6c69 6573 2073 6571 7565  d families seque
-0000e040: 6e63 6520 282d 6f20 7767 645f 6b73 6420  nce (-o wgd_ksd 
-0000e050: 2d74 2077 6764 5f6b 7364 5f74 6d70 202d  -t wgd_ksd_tmp -
-0000e060: 2d6e 7468 7265 6164 7320 3420 2d2d 746f  -nthreads 4 --to
-0000e070: 5f73 746f 7020 2d2d 6364 7320 2d2d 7061  _stop --cds --pa
-0000e080: 6972 7769 7365 202d 2d73 7472 6970 5f67  irwise --strip_g
-0000e090: 6170 7320 2d2d 616c 6967 6e65 7220 6d61  aps --aligner ma
-0000e0a0: 6666 7420 2d2d 7472 6565 5f6d 6574 686f  fft --tree_metho
-0000e0b0: 6420 6661 7374 7472 6565 202d 2d6e 6f64  d fasttree --nod
-0000e0c0: 655f 6176 6572 6167 6529 0a60 6060 0a0a  e_average).```..
-0000e0d0: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
-0000e0e0: 206b 7364 602c 2061 7320 696d 7069 6564   ksd`, as impied
-0000e0f0: 2062 7920 6974 7320 6e61 6d65 2c20 6973   by its name, is
-0000e100: 2066 6f72 2074 6865 2063 6f6e 7374 7275   for the constru
-0000e110: 6374 696f 6e20 6f66 202a 4b2a 3c73 7562  ction of *K*<sub
-0000e120: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000e130: 7472 6962 7574 696f 6e2e 2045 7863 6570  tribution. Excep
-0000e140: 7420 666f 7220 7468 6520 6166 6f72 656d  t for the aforem
-0000e150: 656e 7469 6f6e 6564 2070 6172 616d 6574  entioned paramet
-0000e160: 6572 7320 7375 6368 2061 7320 602d 2d74  ers such as `--t
-0000e170: 6f5f 7374 6f70 6020 616e 6420 602d 2d63  o_stop` and `--c
-0000e180: 6473 602c 2074 6865 7265 2061 7265 2073  ds`, there are s
-0000e190: 6f6d 6520 696d 706f 7274 616e 7420 7061  ome important pa
-0000e1a0: 7261 6d65 7465 7273 2074 6861 7420 6861  rameters that ha
-0000e1b0: 7665 2063 7275 6369 616c 2069 6d70 6163  ve crucial impac
-0000e1c0: 7420 6f6e 2074 6865 202a 4b2a 3c73 7562  t on the *K*<sub
-0000e1d0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-0000e1e0: 696f 6e2e 2054 6865 206f 7074 696f 6e20  ion. The option 
-0000e1f0: 602d 2d70 6169 7277 6973 6560 2069 7320  `--pairwise` is 
-0000e200: 6120 7665 7279 2069 6d70 6f72 7461 6e74  a very important
-0000e210: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-0000e220: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-0000e230: 623e 2065 7374 696d 6174 696f 6e2c 2077  b> estimation, w
-0000e240: 6974 6820 7768 6963 6820 7468 6520 6043  ith which the `C
-0000e250: 4f44 454d 4c60 2077 696c 6c20 6361 6c63  ODEML` will calc
-0000e260: 756c 6174 6520 7468 6520 2a4b 2a3c 7375  ulate the *K*<su
-0000e270: 623e 533c 2f73 7562 3e20 666f 7220 6561  b>S</sub> for ea
-0000e280: 6368 2067 656e 6520 7061 6972 2073 6570  ch gene pair sep
-0000e290: 6172 6174 656c 7920 6261 7365 6420 6f6e  arately based on
-0000e2a0: 2074 6865 2061 6c69 676e 6d65 6e74 206f   the alignment o
-0000e2b0: 6620 6f6e 6c79 2074 6865 7365 2074 776f  f only these two
-0000e2c0: 2067 656e 6573 2069 6e73 7465 6164 206f   genes instead o
-0000e2d0: 6620 7468 6520 7768 6f6c 6520 616c 6967  f the whole alig
-0000e2e0: 6e6d 656e 7420 6f66 2074 6865 2066 616d  nment of the fam
-0000e2f0: 696c 792c 2073 7563 6820 7468 6174 206c  ily, such that l
-0000e300: 6573 7320 6761 7073 2061 7265 2065 7870  ess gaps are exp
-0000e310: 6563 7465 6420 616e 6420 7468 7573 2074  ected and thus t
-0000e320: 6865 2061 6c69 676e 6d65 6e74 2069 6e20  he alignment in 
-0000e330: 7468 6520 636f 6e73 6964 6572 6174 696f  the consideratio
-0000e340: 6e20 6f66 2060 434f 4445 4d4c 6020 7769  n of `CODEML` wi
-0000e350: 6c6c 2062 6520 6c6f 6e67 6572 2028 6265  ll be longer (be
-0000e360: 6361 7573 6520 6043 4f44 454d 4c60 2077  cause `CODEML` w
-0000e370: 696c 6c20 6175 746f 6d61 7465 6c79 2073  ill automately s
-0000e380: 6b69 7020 6576 6572 7920 636f 6c75 6d6e  kip every column
-0000e390: 2077 6974 6820 6761 702c 2072 6567 6172   with gap, regar
-0000e3a0: 646c 6573 7320 6f66 2077 6865 7468 6572  dless of whether
-0000e3b0: 2069 7427 7320 616e 206f 7665 7261 6c6c   it's an overall
-0000e3c0: 206f 7220 7061 7274 6961 6c20 6761 7029   or partial gap)
-0000e3d0: 2c20 7769 7468 6f75 7420 7768 6963 6820  , without which 
-0000e3e0: 7468 6520 6043 4f44 454d 4c60 2077 696c  the `CODEML` wil
-0000e3f0: 6c20 6361 6c63 756c 6174 6520 7468 6520  l calculate the 
-0000e400: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e410: 6261 7365 6420 6f6e 2074 6865 2077 686f  based on the who
-0000e420: 6c65 2061 6c69 676e 6d65 6e74 206f 6620  le alignment of 
-0000e430: 7468 6520 6661 6d69 6c79 2c20 7768 6963  the family, whic
-0000e440: 6820 6d69 6768 7420 6861 7665 206e 6f20  h might have no 
-0000e450: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e460: 7265 7375 6c74 2061 7420 616c 6c20 6966  result at all if
-0000e470: 2074 6865 2073 7472 6970 7065 6420 616c   the stripped al
-0000e480: 6967 6e6d 656e 7420 6c65 6e67 7468 2028  ignment length (
-0000e490: 7265 6d6f 7669 6e67 2061 6c6c 2067 6170  removing all gap
-0000e4a0: 2d63 6f6e 7461 696e 696e 6720 636f 6c75  -containing colu
-0000e4b0: 6d6e 7329 2077 6173 207a 6572 6f2c 2061  mns) was zero, a
-0000e4c0: 2063 6175 7365 206f 6620 6469 6666 6572   cause of differ
-0000e4d0: 656e 7420 6e75 6d62 6572 206f 6620 2a4b  ent number of *K
-0000e4e0: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
-0000e4f0: 7469 6d61 7465 7320 6265 7477 6565 6e20  timates between 
-0000e500: 2270 6169 7277 6973 6520 6d6f 6465 2220  "pairwise mode" 
-0000e510: 616e 6420 226e 6f6e 2d70 6169 7277 6973  and "non-pairwis
-0000e520: 6520 6d6f 6465 222e 2049 7427 7320 6469  e mode". It's di
-0000e530: 6666 6963 756c 7420 746f 2073 6179 2077  fficult to say w
-0000e540: 6869 6368 206d 6f64 6520 6973 206d 6f72  hich mode is mor
-0000e550: 6520 6964 6561 6c2c 2061 6c74 686f 7567  e ideal, althoug
-0000e560: 6820 7468 6520 226e 6f6e 2d70 6169 7277  h the "non-pairw
-0000e570: 6973 6520 6d6f 6465 2220 2864 6566 6175  ise mode" (defau
-0000e580: 6c74 2073 6574 7469 6e67 2920 7768 6963  lt setting) whic
-0000e590: 6820 7275 6e73 206f 6e20 7468 6520 7768  h runs on the wh
-0000e5a0: 6f6c 6520 616c 6967 6e6d 656e 7420 696e  ole alignment in
-0000e5b0: 7374 6561 6420 6f66 2061 206c 6f63 616c  stead of a local
-0000e5c0: 2061 6c69 676e 6d65 6e74 2c20 6d69 6768   alignment, migh
-0000e5d0: 7420 6265 206d 6f72 6520 6269 6f6c 6f67  t be more biolog
-0000e5e0: 6963 616c 6c79 2063 6f6e 7365 7276 6564  ically conserved
-0000e5f0: 2069 6e20 7468 6174 2069 7420 6173 7375   in that it assu
-0000e600: 7265 7320 7468 6520 6576 6f6c 7574 696f  res the evolutio
-0000e610: 6e20 6f66 2065 6163 6820 636f 6c75 6d6e  n of each column
-0000e620: 2074 6f20 6265 2073 7461 7274 6564 2066   to be started f
-0000e630: 726f 6d20 7468 6520 726f 6f74 206f 6620  rom the root of 
-0000e640: 7468 6520 6661 6d69 6c79 2061 6e64 2061  the family and a
-0000e650: 6c6c 2074 6865 2067 656e 6520 6475 706c  ll the gene dupl
-0000e660: 6963 6174 6573 2061 7265 2074 616b 656e  icates are taken
-0000e670: 2069 6e74 6f20 6163 636f 756e 7420 696e   into account in
-0000e680: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-0000e690: 7375 623e 2065 7374 696d 6174 696f 6e20  sub> estimation 
-0000e6a0: 7072 6f63 6573 732e 2054 6865 206f 7074  process. The opt
-0000e6b0: 696f 6e20 602d 2d73 7472 6970 5f67 6170  ion `--strip_gap
-0000e6c0: 7360 2063 616e 2072 656d 6f76 6520 616c  s` can remove al
-0000e6d0: 6c20 7468 6520 6761 702d 636f 6e74 6169  l the gap-contai
-0000e6e0: 6e69 6e67 2063 6f6c 756d 6e73 2c20 7769  ning columns, wi
-0000e6f0: 7468 206f 7220 7769 7468 6f75 7420 7768  th or without wh
-0000e700: 6963 6820 7468 6520 7265 7375 6c74 206f  ich the result o
-0000e710: 6620 226e 6f6e 2d70 6169 7277 6973 6520  f "non-pairwise 
-0000e720: 6d6f 6465 2220 776f 6e27 7420 6265 2061  mode" won't be a
-0000e730: 6666 6563 7465 642c 2077 6869 6c65 2077  ffected, while w
-0000e740: 6974 6820 7768 6963 6820 7468 6520 7265  ith which the re
-0000e750: 7375 6c74 206f 6620 2270 6169 7277 6973  sult of "pairwis
-0000e760: 6520 6d6f 6465 2220 7769 6c6c 2062 6520  e mode" will be 
-0000e770: 616c 7465 7265 642e 2054 6865 206f 7074  altered. The opt
-0000e780: 696f 6e20 602d 2d61 6c69 676e 6572 6020  ion `--aligner` 
-0000e790: 616e 6420 602d 2d61 6c6e 5f6f 7074 696f  and `--aln_optio
-0000e7a0: 6e73 6020 7768 6963 6820 6465 6369 6465  ns` which decide
-0000e7b0: 2077 6869 6368 2061 6c69 676e 6d65 6e74   which alignment
-0000e7c0: 2070 726f 6772 616d 2074 6f20 6265 2075   program to be u
-0000e7d0: 7365 6420 616e 6420 7768 6963 6820 7061  sed and which pa
-0000e7e0: 7261 6d65 7465 7220 746f 2062 6520 7365  rameter to be se
-0000e7f0: 742c 2077 696c 6c20 6861 7665 2069 6d70  t, will have imp
-0000e800: 6163 7420 6f6e 2074 6865 202a 4b2a 3c73  act on the *K*<s
-0000e810: 7562 3e53 3c2f 7375 623e 2072 6573 756c  ub>S</sub> resul
-0000e820: 7473 2c20 6e6f 7465 6420 7468 6174 2074  ts, noted that t
-0000e830: 6865 2064 6566 6175 6c74 2070 726f 6772  he default progr
-0000e840: 616d 2069 7320 606d 6166 6674 6020 616e  am is `mafft` an
-0000e850: 6420 7468 6520 7061 7261 6d65 7465 7220  d the parameter 
-0000e860: 6973 2060 2d2d 6175 746f 602e 2054 6865  is `--auto`. The
-0000e870: 206f 7074 696f 6e20 602d 2d74 7265 655f   option `--tree_
-0000e880: 6d65 7468 6f64 6020 616e 6420 602d 2d74  method` and `--t
-0000e890: 7265 655f 6f70 7469 6f6e 7360 2064 6563  ree_options` dec
-0000e8a0: 6964 6520 7768 6963 6820 6765 6e65 2074  ide which gene t
-0000e8b0: 7265 6520 696e 6665 7265 6e63 6520 7072  ree inference pr
-0000e8c0: 6f67 7261 6d20 746f 2062 6520 7573 6564  ogram to be used
-0000e8d0: 2061 6e64 2077 6869 6368 2070 6172 616d   and which param
-0000e8e0: 6574 6572 2074 6f20 6265 2073 6574 2c20  eter to be set, 
-0000e8f0: 776f 6e27 7420 6166 6665 6374 2074 6865  won't affect the
-0000e900: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000e910: 2065 7374 696d 6174 696f 6e20 6974 7365   estimation itse
-0000e920: 6c66 2062 7574 2074 6865 2072 6573 756c  lf but the resul
-0000e930: 7420 6f66 2064 652d 7265 6475 6e64 616e  t of de-redundan
-0000e940: 6379 2c20 6e6f 7465 6420 7468 6174 2077  cy, noted that w
-0000e950: 6520 696d 706c 656d 656e 7465 6420 6120  e implemented a 
-0000e960: 6275 696c 742d 696e 2067 656e 6520 7472  built-in gene tr
-0000e970: 6565 2069 6e66 6572 656e 6365 206d 6574  ee inference met
-0000e980: 686f 6420 6261 7365 6420 6f6e 2074 6865  hod based on the
-0000e990: 2041 7665 7261 6765 204c 696e 6b61 6765   Average Linkage
-0000e9a0: 2043 6c75 7374 6572 696e 6720 2841 4c43   Clustering (ALC
-0000e9b0: 2920 2874 6875 7320 6120 6469 7374 616e  ) (thus a distan
-0000e9c0: 6365 2d62 6173 6564 2074 7265 6529 2077  ce-based tree) w
-0000e9d0: 6974 6820 7468 6520 602d 2d74 7265 655f  ith the `--tree_
-0000e9e0: 6d65 7468 6f64 6020 7365 7420 6173 2022  method` set as "
-0000e9f0: 636c 7573 7465 7222 2e20 5477 6f20 6d65  cluster". Two me
-0000ea00: 7468 6f64 7320 6f66 2064 652d 7265 6475  thods of de-redu
-0000ea10: 6e64 616e 6379 2061 7265 2069 6d70 6c65  ndancy are imple
-0000ea20: 6d65 6e74 6564 2069 6e20 6077 6764 2076  mented in `wgd v
-0000ea30: 3260 2c20 6e61 6d65 6c79 206e 6f64 652d  2`, namely node-
-0000ea40: 7765 6967 6874 6564 2061 6e64 206e 6f64  weighted and nod
-0000ea50: 652d 6176 6572 6167 6564 206d 6574 686f  e-averaged metho
-0000ea60: 6473 2e20 5468 6520 6e6f 6465 2d77 6569  ds. The node-wei
-0000ea70: 6768 7465 6420 6d65 7468 6f64 2061 6368  ghted method ach
-0000ea80: 6965 7665 7320 7468 6520 6465 2d72 6564  ieves the de-red
-0000ea90: 756e 6461 6e63 7920 7669 6120 7765 6967  undancy via weig
-0000eaa0: 6869 6e67 2074 6865 202a 4b2a 3c73 7562  hing the *K*<sub
-0000eab0: 3e53 3c2f 7375 623e 2076 616c 7565 2061  >S</sub> value a
-0000eac0: 7373 6f63 6961 7465 6420 7769 7468 2065  ssociated with e
-0000ead0: 6163 6820 6765 6e65 2070 6169 7220 7375  ach gene pair su
-0000eae0: 6368 2074 6861 7420 7468 6520 7765 6967  ch that the weig
-0000eaf0: 6874 206f 6620 6120 7369 6e67 6c65 2064  ht of a single d
-0000eb00: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
-0000eb10: 2073 756d 7320 7570 2074 6f20 3120 286e   sums up to 1 (n
-0000eb20: 6f74 6564 2074 6861 7420 7468 6520 6e75  oted that the nu
-0000eb30: 6d62 6572 206f 6620 2a4b 2a3c 7375 623e  mber of *K*<sub>
-0000eb40: 533c 2f73 7562 3e20 6573 7469 6d61 7469  S</sub> estimati
-0000eb50: 6f6e 2072 656d 6169 6e73 2074 6865 2073  on remains the s
-0000eb60: 616d 6529 2077 6869 6c65 2074 6865 206e  ame) while the n
-0000eb70: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
-0000eb80: 686f 6420 7265 616c 697a 6573 2074 6865  hod realizes the
-0000eb90: 2064 652d 7265 6475 6e64 616e 6379 2076   de-redundancy v
-0000eba0: 6961 2063 616c 6375 6c61 7469 6e67 2070  ia calculating p
-0000ebb0: 6572 2067 656e 6520 7472 6565 206e 6f64  er gene tree nod
-0000ebc0: 6520 6f6e 6520 6176 6572 6167 6564 202a  e one averaged *
-0000ebd0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
-0000ebe0: 616c 7565 2074 6f20 7265 7072 6573 656e  alue to represen
-0000ebf0: 7420 7468 6520 6167 6520 6f66 2065 6163  t the age of eac
-0000ec00: 6820 6475 706c 6963 6174 696f 6e20 6576  h duplication ev
-0000ec10: 656e 742e 2054 6865 206f 7074 696f 6e20  ent. The option 
-0000ec20: 602d 2d6e 6f64 655f 6176 6572 6167 6560  `--node_average`
-0000ec30: 2063 616e 2062 6520 7365 7420 746f 2063   can be set to c
-0000ec40: 686f 6f73 6520 7468 6520 6e6f 6465 2d61  hoose the node-a
-0000ec50: 7665 7261 6765 6420 7761 7920 6f66 2064  veraged way of d
-0000ec60: 652d 7265 6475 6e64 616e 6379 2e20 4469  e-redundancy. Di
-0000ec70: 6666 6572 656e 7420 6d65 7468 6f64 7320  fferent methods 
-0000ec80: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
-0000ec90: 2077 696c 6c20 6861 7665 2069 6d70 6163   will have impac
-0000eca0: 7420 6f6e 2074 6865 2064 6574 6563 7469  t on the detecti
-0000ecb0: 6f6e 206f 6620 5747 4420 7369 676e 616c  on of WGD signal
-0000ecc0: 732c 2077 6869 6368 2068 6173 2062 6565  s, which has bee
-0000ecd0: 6e20 696e 7665 7374 6967 6174 6564 2069  n investigated i
-0000ece0: 6e20 7468 6973 205b 6c69 7465 7261 7475  n this [literatu
-0000ecf0: 7265 5d28 6874 7470 733a 2f2f 646f 692e  re](https://doi.
-0000ed00: 6f72 672f 3130 2e31 3039 332f 6762 652f  org/10.1093/gbe/
-0000ed10: 6576 7932 3030 292e 2054 6865 2070 6172  evy200). The par
-0000ed20: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000ed30: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
-0000ed40: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
-0000ed50: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
-0000ed60: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
-0000ed70: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000ed80: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
-0000ed90: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
-0000eda0: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
-0000edb0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000edc0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-0000edd0: 2063 6f6d 6d61 6e64 2061 7320 6265 6c6f   command as belo
-0000ede0: 772a 2a0a 0a60 6060 0a77 6764 206b 7364  w**..```.wgd ksd
-0000edf0: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000ee00: 6365 0a60 6060 0a0a 2a2a 5468 6520 636f  ce.```..**The co
-0000ee10: 6e73 7472 7563 7469 6f6e 206f 6620 6f72  nstruction of or
-0000ee20: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-0000ee30: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-0000ee40: 7374 7269 6275 7469 6f6e 2a2a 0a60 6060  stribution**.```
-0000ee50: 0a77 6764 206b 7364 2066 616d 696c 6965  .wgd ksd familie
-0000ee60: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000ee70: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000ee80: 282d 2d72 6577 6569 6768 7429 0a60 6060  (--reweight).```
-0000ee90: 0a0a 4672 6f6d 2070 6172 616c 6f67 6f75  ..From paralogou
-0000eea0: 7320 746f 206f 7274 686f 6c6f 676f 7573  s to orthologous
-0000eeb0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000eec0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000eed0: 6e2c 2075 7365 7273 206f 6e6c 7920 6e65  n, users only ne
-0000eee0: 6564 2074 6f20 7072 6f76 6964 6520 6d6f  ed to provide mo
-0000eef0: 7265 2063 6473 2066 696c 6573 2e20 4e6f  re cds files. No
-0000ef00: 7465 2074 6861 7420 7769 7468 206f 7274  te that with ort
-0000ef10: 686f 6c6f 676f 7573 2067 656e 6520 6661  hologous gene fa
-0000ef20: 6d69 6c69 6573 2074 6865 2077 6569 6768  milies the weigh
-0000ef30: 7469 6e67 206d 6574 686f 6420 6361 6e20  ting method can 
-0000ef40: 6265 2073 6574 2074 6f20 6265 2063 616c  be set to be cal
-0000ef50: 6375 6c61 7465 6420 7065 7220 7370 6563  culated per spec
-0000ef60: 6965 7320 7061 6972 2069 6e73 7465 6164  ies pair instead
-0000ef70: 206f 6620 636f 6e73 6964 6572 696e 6720   of considering 
-0000ef80: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
-0000ef90: 2062 6563 6175 7365 2077 6865 6e20 706c   because when pl
-0000efa0: 6f74 7469 6e67 206f 7274 686f 6c6f 676f  otting orthologo
-0000efb0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-0000efc0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000efd0: 696f 6e20 6265 7477 6565 6e20 7477 6f20  ion between two 
-0000efe0: 7370 6563 6965 7320 7468 6520 7765 6967  species the weig
-0000eff0: 6874 2063 616c 6375 6c61 7465 6420 6672  ht calculated fr
-0000f000: 6f6d 2074 6869 7320 7370 6563 6966 6963  om this specific
-0000f010: 2073 7065 6369 6573 2070 6169 7220 7368   species pair sh
-0000f020: 6f75 6c64 2062 6520 636f 6e73 6572 7665  ould be conserve
-0000f030: 6420 7768 696c 6520 7468 6520 6f6e 6520  d while the one 
-0000f040: 6361 6c63 756c 6174 6564 2066 726f 6d20  calculated from 
-0000f050: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
-0000f060: 2077 696c 6c20 7661 7279 2077 6974 6820   will vary with 
-0000f070: 7468 6520 6e75 6d62 6572 206f 6620 7370  the number of sp
-0000f080: 6563 6965 732e 2054 6f20 696e 6974 6961  ecies. To initia
-0000f090: 7465 2074 6865 2077 6569 6768 7469 6e67  te the weighting
-0000f0a0: 2070 6572 2073 7065 6369 6573 2070 6169   per species pai
-0000f0b0: 722c 2074 6865 206f 7074 696f 6e20 602d  r, the option `-
-0000f0c0: 2d72 6577 6569 6768 7460 2063 616e 2062  -reweight` can b
-0000f0d0: 6520 7365 742e 0a0a 2a2a 4120 7375 6767  e set...**A sugg
-0000f0e0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-0000f0f0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-0000f100: 6e64 2061 7320 6265 6c6f 772a 2a0a 0a60  nd as below**..`
-0000f110: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
-0000f120: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000f130: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000f140: 330a 6060 600a 0a2a 2a54 6865 2063 6f6e  3.```..**The con
-0000f150: 7374 7275 6374 696f 6e20 6f66 202a 4b2a  struction of *K*
-0000f160: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-0000f170: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
-0000f180: 7468 2072 6174 6520 636f 7272 6563 7469  th rate correcti
-0000f190: 6f6e 2a2a 0a60 6060 0a77 6764 206b 7364  on**.```.wgd ksd
-0000f1a0: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000f1b0: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
-0000f1c0: 7175 656e 6365 3320 2d2d 666f 6375 7332  quence3 --focus2
-0000f1d0: 616c 6c20 7365 7175 656e 6365 3120 2d73  all sequence1 -s
-0000f1e0: 7020 7370 6461 7461 202d 2d65 7874 7261  p spdata --extra
-0000f1f0: 7061 7261 6e6f 6d65 6b73 2070 6172 616e  paranomeks paran
-0000f200: 6f6d 654b 7364 6174 6120 282d 2d70 6c6f  omeKsdata (--plo
-0000f210: 7465 6c6d 6d20 2d2d 706c 6f74 6170 676d  telmm --plotapgm
-0000f220: 6d20 2d2d 616e 6368 6f72 706f 696e 7473  m --anchorpoints
-0000f230: 2061 7064 6174 6120 2d2d 7265 7765 6967   apdata --reweig
-0000f240: 6874 202d 2d6f 6e6c 7972 6f6f 746f 7574  ht --onlyrootout
-0000f250: 290a 6060 600a 0a49 6e73 7069 7265 6420  ).```..Inspired 
-0000f260: 6279 2074 6865 2072 6174 6520 636f 7272  by the rate corr
-0000f270: 6563 7469 6f6e 2061 6c67 6f72 6974 686d  ection algorithm
-0000f280: 2069 6e20 5b6b 7372 6174 6573 5d28 6874   in [ksrates](ht
-0000f290: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-0000f2a0: 2f56 4942 2d50 5342 2f6b 7372 6174 6573  /VIB-PSB/ksrates
-0000f2b0: 292c 2077 6520 696d 706c 656d 656e 7465  ), we implemente
-0000f2c0: 6420 7468 6520 7261 7465 2063 6f72 7265  d the rate corre
-0000f2d0: 6374 696f 6e20 616e 616c 7973 6973 2061  ction analysis a
-0000f2e0: 6c73 6f20 696e 2060 7767 6420 7632 602c  lso in `wgd v2`,
-0000f2f0: 2077 6869 6368 2069 7320 6d6f 7374 6c79   which is mostly
-0000f300: 2074 6865 2073 616d 6520 6173 205b 6b73   the same as [ks
-0000f310: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
-0000f320: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
-0000f330: 422f 6b73 7261 7465 7329 2062 7574 2064  B/ksrates) but d
-0000f340: 6966 6665 7273 2069 6e20 7468 6520 6361  iffers in the ca
-0000f350: 6c63 756c 6174 696f 6e20 6f66 2074 6865  lculation of the
-0000f360: 2073 7461 6e64 6172 6420 6465 7669 6174   standard deviat
-0000f370: 696f 6e20 6f66 2072 6573 6361 6c65 6420  ion of rescaled 
-0000f380: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000f390: 6167 6573 2e20 546f 2070 6572 666f 726d  ages. To perform
-0000f3a0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
-0000f3b0: 7469 6f6e 2061 6e61 6c79 7369 732c 2075  tion analysis, u
-0000f3c0: 7365 7273 2063 616e 2075 7365 2062 6f74  sers can use bot
-0000f3d0: 6820 7468 6520 6077 6764 206b 7364 6020  h the `wgd ksd` 
-0000f3e0: 616e 6420 6077 6764 2076 697a 6020 7072  and `wgd viz` pr
-0000f3f0: 6f67 7261 6d2e 2046 6f72 2074 6865 2060  ogram. For the `
-0000f400: 7767 6420 6b73 6460 2070 726f 6772 616d  wgd ksd` program
-0000f410: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
-0000f420: 7072 6f76 6964 6520 6120 7370 6563 6965  provide a specie
-0000f430: 7320 7472 6565 2076 6961 2074 6865 206f  s tree via the o
-0000f440: 7074 696f 6e20 602d 2d73 7065 6369 6573  ption `--species
-0000f450: 7472 6565 6020 6f6e 2077 6869 6368 2072  tree` on which r
-0000f460: 6174 6520 636f 7272 6563 7469 6f6e 2063  ate correction c
-0000f470: 616e 2062 6520 636f 6e64 7563 7465 642e  an be conducted.
-0000f480: 204e 6f74 6520 7468 6174 2075 6e6e 6563   Note that unnec
-0000f490: 6573 7361 7279 2062 7261 636b 6574 7320  essary brackets 
-0000f4a0: 6d69 6768 7420 6c65 6164 2074 6f20 756e  might lead to un
-0000f4b0: 6578 7065 6374 6564 2065 7272 6f72 732c  expected errors,
-0000f4c0: 2066 6f72 2069 6e73 7461 6e63 6520 6120   for instance a 
-0000f4d0: 7472 6565 2060 2841 2c28 422c 4329 293b  tree `(A,(B,C));
-0000f4e0: 6020 7368 6f75 6c64 206e 6f74 2062 6520  ` should not be 
-0000f4f0: 7265 7072 6573 656e 7465 6420 6173 2060  represented as `
-0000f500: 2841 2c28 2842 2c43 2929 293b 602e 2054  (A,((B,C)));`. T
-0000f510: 6865 2073 6574 206f 6620 7370 6563 6965  he set of specie
-0000f520: 7320 7061 6972 7320 746f 2062 6520 7368  s pairs to be sh
-0000f530: 6f77 6e20 6973 2066 6c65 7869 626c 6520  own is flexible 
-0000f540: 7468 6174 2074 6865 206d 6f73 7420 636f  that the most co
-0000f550: 6e76 656e 6965 6e74 206f 7074 696f 6e20  nvenient option 
-0000f560: 6973 2060 2d2d 666f 6375 7332 616c 6c60  is `--focus2all`
-0000f570: 2077 6869 6368 2073 696d 706c 7920 7368   which simply sh
-0000f580: 6f77 7320 616c 6c20 7468 6520 706f 7373  ows all the poss
-0000f590: 6962 6c65 2066 6f63 616c 2d73 6973 7465  ible focal-siste
-0000f5a0: 7220 7370 6563 6965 7320 7061 6972 732c  r species pairs,
-0000f5b0: 206f 7220 7573 6572 7320 6361 6e20 6d61   or users can ma
-0000f5c0: 6e75 616c 6c79 2073 6574 2074 6865 2073  nually set the s
-0000f5d0: 7065 6369 6573 2070 6169 7273 2076 6961  pecies pairs via
-0000f5e0: 2074 6865 206f 7074 696f 6e20 602d 2d73   the option `--s
-0000f5f0: 7061 6972 602e 204e 6f74 6520 7468 6174  pair`. Note that
-0000f600: 2069 6620 7468 6520 7370 6563 6965 7320   if the species 
-0000f610: 7061 6972 7320 7765 7265 206d 616e 7561  pairs were manua
-0000f620: 6c6c 7920 7365 742c 2069 7420 776f 756c  lly set, it woul
-0000f630: 6420 6265 206e 6565 6465 6420 746f 2063  d be needed to c
-0000f640: 6f2d 7365 7420 7468 6520 6f70 7469 6f6e  o-set the option
-0000f650: 2060 2d2d 636c 6173 7369 6360 2e20 5468   `--classic`. Th
-0000f660: 6520 6f70 7469 6f6e 2060 2d2d 6f6e 6c79  e option `--only
-0000f670: 726f 6f74 6f75 7460 2063 616e 2062 6520  rootout` can be 
-0000f680: 7365 7420 746f 206f 6e6c 7920 636f 6e73  set to only cons
-0000f690: 6964 6572 206f 7574 6772 6f75 7020 6174  ider outgroup at
-0000f6a0: 2074 6865 2072 6f6f 742c 2069 6e73 7465   the root, inste
-0000f6b0: 6164 206f 6620 616c 6c20 7468 6520 706f  ad of all the po
-0000f6c0: 7373 6962 6c65 206f 7574 6772 6f75 7073  ssible outgroups
-0000f6d0: 2070 6572 2066 6f63 616c 2d73 6973 7465   per focal-siste
-0000f6e0: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
-0000f6f0: 7768 6963 6820 6861 7320 696d 7061 6374  which has impact
-0000f700: 206f 6e20 7468 6520 6669 6e61 6c20 636f   on the final co
-0000f710: 7272 6563 7465 6420 2a4b 2a3c 7375 623e  rrected *K*<sub>
-0000f720: 533c 2f73 7562 3e20 6167 6573 2e20 5765  S</sub> ages. We
-0000f730: 2073 7567 6765 7374 206f 6620 7573 696e   suggest of usin
-0000f740: 6720 616c 6c20 7468 6520 706f 7373 6962  g all the possib
-0000f750: 6c65 206f 7574 6772 6f75 7073 2070 6572  le outgroups per
-0000f760: 2066 6f63 616c 2d73 6973 7465 7220 7370   focal-sister sp
-0000f770: 6563 6965 7320 7061 6972 2061 7320 666f  ecies pair as fo
-0000f780: 7220 6120 6c65 7373 2062 6961 7365 6420  r a less biased 
-0000f790: 7265 7375 6c74 2e20 5468 6520 7061 7261  result. The para
-0000f7a0: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-0000f7b0: 7375 623e 2064 6174 6120 7368 6f75 6c64  sub> data should
-0000f7c0: 2062 6520 7072 6f76 6964 6564 2076 6961   be provided via
-0000f7d0: 2074 6865 206f 7074 696f 6e20 602d 2d65   the option `--e
-0000f7e0: 7874 7261 7061 7261 6e6f 6d65 6b73 602e  xtraparanomeks`.
-0000f7f0: 200a 0a53 6f6d 6520 6f74 6865 7220 6f70   ..Some other op
-0000f800: 7469 6f6e 7320 7768 6963 6820 6861 7665  tions which have
-0000f810: 206e 6f20 696d 7061 6374 206f 6e20 7468   no impact on th
-0000f820: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
-0000f830: 6e20 6275 7420 6164 6420 6d6f 7265 206c  n but add more l
-0000f840: 6179 6572 7320 6f72 2063 6861 6e67 6520  ayers or change 
-0000f850: 7468 6520 6170 7065 6172 616e 6365 206f  the appearance o
-0000f860: 6e20 7468 6520 6d69 7865 6420 2a4b 2a3c  n the mixed *K*<
-0000f870: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-0000f880: 6469 7374 7269 6275 7469 6f6e 2c20 696e  distribution, in
-0000f890: 636c 7564 6520 602d 2d70 6c6f 7461 7067  clude `--plotapg
-0000f8a0: 6d6d 6020 616e 6420 602d 2d70 6c6f 7465  mm` and `--plote
-0000f8b0: 6c6d 6d60 2065 7463 2e20 5468 6520 6f70  lmm` etc. The op
-0000f8c0: 7469 6f6e 2060 2d2d 706c 6f74 6170 676d  tion `--plotapgm
-0000f8d0: 6d60 2063 616e 2062 6520 7365 7420 746f  m` can be set to
-0000f8e0: 2063 616c 6c20 7468 6520 474d 4d20 616e   call the GMM an
-0000f8f0: 616c 7973 6973 2075 706f 6e20 7468 6520  alysis upon the 
-0000f900: 616e 6368 6f72 2070 6169 7220 2a4b 2a3c  anchor pair *K*<
-0000f910: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-0000f920: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
-0000f930: 2070 6c6f 7420 7468 6520 636c 7573 7465   plot the cluste
-0000f940: 7269 6e67 2072 6573 756c 7420 7570 6f6e  ring result upon
-0000f950: 2074 6865 206d 6978 6564 202a 4b2a 3c73   the mixed *K*<s
-0000f960: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000f970: 6973 7472 6962 7574 696f 6e2c 2077 6869  istribution, whi
-0000f980: 6368 2068 6173 2074 6f20 6265 2063 6f2d  ch has to be co-
-0000f990: 7365 7420 7769 7468 2074 6865 206f 7074  set with the opt
-0000f9a0: 696f 6e20 602d 2d61 6e63 686f 7270 6f69  ion `--anchorpoi
-0000f9b0: 6e74 7360 2070 726f 7669 6469 6e67 2074  nts` providing t
-0000f9c0: 6865 2061 6e63 686f 7220 7061 6972 7320  he anchor pairs 
-0000f9d0: 696e 666f 726d 6174 696f 6e2e 2054 6865  information. The
-0000f9e0: 206f 7074 696f 6e20 602d 2d70 6c6f 7465   option `--plote
-0000f9f0: 6c6d 6d60 2063 616e 2062 6520 7365 7420  lmm` can be set 
-0000fa00: 746f 2063 616c 6c20 7468 6520 454c 4d4d  to call the ELMM
-0000fa10: 2061 6e61 6c79 7369 7320 7570 6f6e 2074   analysis upon t
-0000fa20: 6865 2077 686f 6c65 2070 6172 616e 6f6d  he whole paranom
-0000fa30: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-0000fa40: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
-0000fa50: 6f6e 2e20 2a2a 4e6f 7465 2074 6861 7420  on. **Note that 
-0000fa60: 7468 6520 7370 6563 6965 7320 6e61 6d65  the species name
-0000fa70: 7320 7072 6573 656e 7420 696e 2074 6865  s present in the
-0000fa80: 2073 7065 6369 6573 2074 7265 6520 6669   species tree fi
-0000fa90: 6c65 2073 686f 756c 6420 6d61 7463 6820  le should match 
-0000faa0: 7468 6520 6e61 6d65 7320 6f66 2074 6865  the names of the
-0000fab0: 2063 6f72 7265 7370 6f6e 6469 6e67 2073   corresponding s
-0000fac0: 6571 7565 6e63 6520 6669 6c65 732a 2a20  equence files** 
-0000fad0: 466f 7220 696e 7374 616e 6365 2c20 6769  For instance, gi
-0000fae0: 7665 6e20 7468 6520 6364 7320 6669 6c65  ven the cds file
-0000faf0: 206e 616d 6573 2027 412e 6364 7327 2c27   names 'A.cds','
-0000fb00: 422e 6364 7327 2c27 432e 6364 7327 2c20  B.cds','C.cds', 
-0000fb10: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
-0000fb20: 2063 6f75 6c64 2062 6520 2728 412e 6364   could be '(A.cd
-0000fb30: 732c 2842 2e63 6473 2c43 2e63 6473 2929  s,(B.cds,C.cds))
-0000fb40: 3b27 2072 6174 6865 7220 7468 616e 2027  ;' rather than '
-0000fb50: 2841 2c28 422c 4329 293b 272e 2054 6865  (A,(B,C));'. The
-0000fb60: 7265 2069 7320 6e6f 2072 6571 7569 7265  re is no require
-0000fb70: 6d65 6e74 2066 6f72 2074 6865 206e 616d  ment for the nam
-0000fb80: 6520 6f66 2074 6865 2070 6172 616e 6f6d  e of the paranom
-0000fb90: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-0000fba0: 3e20 6461 7461 6669 6c65 2077 6869 6368  > datafile which
-0000fbb0: 2063 616e 2062 6520 6e61 6d65 6420 696e   can be named in
-0000fbc0: 2077 6861 7465 7665 7220 6d61 6e6e 6572   whatever manner
-0000fbd0: 2075 7365 7273 2070 7265 6665 722e 2054   users prefer. T
-0000fbe0: 6865 2027 474d 4d27 2069 7320 7468 6520  he 'GMM' is the 
-0000fbf0: 6162 6272 6576 6961 7469 6f6e 206f 6620  abbreviation of 
-0000fc00: 4761 7573 7369 616e 204d 6978 7475 7265  Gaussian Mixture
-0000fc10: 204d 6f64 656c 696e 6720 7768 696c 6520   Modeling while 
-0000fc20: 7468 6520 2745 4c4d 4d27 2072 6566 6572  the 'ELMM' refer
-0000fc30: 7320 746f 2045 7870 6f6e 656e 7469 616c  s to Exponential
-0000fc40: 2d4c 6f67 6e6f 726d 616c 204d 6978 7475  -Lognormal Mixtu
-0000fc50: 7265 204d 6f64 656c 696e 6720 6173 205b  re Modeling as [
-0000fc60: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
-0000fc70: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
-0000fc80: 5053 422f 6b73 7261 7465 7329 2069 6e74  PSB/ksrates) int
-0000fc90: 6572 7072 6574 732e 0a0a 5468 6572 6520  erprets...There 
-0000fca0: 6172 6520 3231 2063 6f6c 756d 6e73 2069  are 21 columns i
-0000fcb0: 6e20 7468 6520 7265 7375 6c74 2060 2e6b  n the result `.k
-0000fcc0: 732e 7473 7660 2066 696c 6520 6265 7369  s.tsv` file besi
-0000fcd0: 6465 7320 7468 6520 696e 6465 7820 636f  des the index co
-0000fce0: 6c75 6d6e 7320 6070 6169 7260 2061 7320  lumns `pair` as 
-0000fcf0: 7468 6520 756e 6971 7565 2069 6465 6e74  the unique ident
-0000fd00: 6966 6965 7220 666f 7220 6561 6368 2067  ifier for each g
-0000fd10: 656e 6520 7061 6972 2e20 5468 6520 604e  ene pair. The `N
-0000fd20: 602c 2060 5360 2c20 6064 4e60 2c20 6064  `, `S`, `dN`, `d
-0000fd30: 4e2f 6453 602c 2060 6453 602c 2060 6c60  N/dS`, `dS`, `l`
-0000fd40: 2061 6e64 2060 7460 2061 7265 2066 726f   and `t` are fro
-0000fd50: 6d20 7468 6520 636f 6465 6d6c 2072 6573  m the codeml res
-0000fd60: 756c 7473 2c20 7265 7072 6573 656e 7469  ults, representi
-0000fd70: 6e67 2074 6865 204e 2065 7374 696d 6174  ng the N estimat
-0000fd80: 652c 2074 6865 2053 2065 7374 696d 6174  e, the S estimat
-0000fd90: 652c 2074 6865 2064 4e20 6573 7469 6d61  e, the dN estima
-0000fda0: 7465 2c20 7468 6520 644e 2f64 5320 286f  te, the dN/dS (o
-0000fdb0: 6d65 6761 2920 6573 7469 6d61 7465 2c20  mega) estimate, 
-0000fdc0: 7468 6520 6453 2065 7374 696d 6174 652c  the dS estimate,
-0000fdd0: 2074 6865 206c 6f67 2d6c 696b 656c 6968   the log-likelih
-0000fde0: 6f6f 6420 616e 6420 7468 6520 7420 6573  ood and the t es
-0000fdf0: 7469 6d61 7465 2c20 7265 7370 6563 7469  timate, respecti
-0000fe00: 7665 6c79 2e20 5468 6520 6061 6c69 676e  vely. The `align
-0000fe10: 6d65 6e74 636f 7665 7261 6765 602c 2060  mentcoverage`, `
-0000fe20: 616c 6967 6e6d 656e 7469 6465 6e74 6974  alignmentidentit
-0000fe30: 7960 2061 6e64 2060 616c 6967 6e6d 656e  y` and `alignmen
-0000fe40: 746c 656e 6774 6860 2061 7265 2074 6865  tlength` are the
-0000fe50: 2069 6e66 6f72 6d61 7469 6f6e 2070 6572   information per
-0000fe60: 7461 696e 696e 6720 746f 2074 6865 2061  taining to the a
-0000fe70: 6c69 676e 6d65 6e74 2066 6f72 2065 6163  lignment for eac
-0000fe80: 6820 6661 6d69 6c79 2c20 7265 7072 6573  h family, repres
-0000fe90: 656e 7469 6e67 2074 6865 2072 6174 696f  enting the ratio
-0000fea0: 206f 6620 7468 6520 7374 7269 7070 6564   of the stripped
-0000feb0: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
-0000fec0: 6820 636f 6d70 6172 6564 2074 6f20 7468  h compared to th
-0000fed0: 6520 6675 6c6c 2061 6c69 676e 6d65 6e74  e full alignment
-0000fee0: 206c 656e 6774 682c 2074 6865 2072 6174   length, the rat
-0000fef0: 696f 206f 6620 636f 6c75 6d6e 7320 7769  io of columns wi
-0000ff00: 7468 2069 6465 6e74 6963 616c 206e 7563  th identical nuc
-0000ff10: 6c65 6f74 6964 6573 2063 6f6d 7061 7265  leotides compare
-0000ff20: 6420 746f 2074 6865 206f 7665 7261 6c6c  d to the overall
-0000ff30: 2063 6f6c 756d 6e73 206f 6620 7468 6520   columns of the 
-0000ff40: 7374 7269 7070 6564 2061 6c69 676e 6d65  stripped alignme
-0000ff50: 6e74 2c20 616e 6420 7468 6520 6c65 6e67  nt, and the leng
-0000ff60: 7468 206f 6620 7468 6520 6675 6c6c 2061  th of the full a
-0000ff70: 6c69 676e 6d65 6e74 2c20 7265 7370 6563  lignment, respec
-0000ff80: 7469 7665 6c79 2e0a 2323 2320 7767 6420  tively..### wgd 
-0000ff90: 6d69 780a 0a2a 2a54 6865 206d 6978 7475  mix..**The mixtu
-0000ffa0: 7265 206d 6f64 656c 2063 6c75 7374 6572  re model cluster
-0000ffb0: 696e 6720 616e 616c 7973 6973 206f 6620  ing analysis of 
-0000ffc0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000ffd0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-0000ffe0: 2a2a 0a60 6060 0a77 6764 206d 6978 206b  **.```.wgd mix k
-0000fff0: 7364 6174 6120 282d 2d6e 5f69 6e69 7420  sdata (--n_init 
-00010000: 3230 3020 2d2d 6d61 785f 6974 6572 2032  200 --max_iter 2
-00010010: 3030 202d 2d6b 735f 7261 6e67 6520 3020  00 --ks_range 0 
-00010020: 3520 2d2d 6669 6c74 6572 7320 3330 3020  5 --filters 300 
-00010030: 2d2d 6269 6e73 2035 3020 2d2d 636f 6d70  --bins 50 --comp
-00010040: 6f6e 656e 7473 2031 2034 202d 2d67 616d  onents 1 4 --gam
-00010050: 6d61 2030 2e30 3031 290a 6060 600a 0a54  ma 0.001).```..T
-00010060: 6869 7320 7061 7274 206f 6620 4761 7573  his part of Gaus
-00010070: 7369 616e 206d 6978 7475 7265 206d 6f64  sian mixture mod
-00010080: 656c 696e 6720 2847 4d4d 2920 616e 616c  eling (GMM) anal
-00010090: 7973 6973 2069 7320 696e 6865 7269 7465  ysis is inherite
-000100a0: 6420 6672 6f6d 2074 6865 206f 7269 6769  d from the origi
-000100b0: 6e61 6c20 6077 6764 6020 7072 6f67 7261  nal `wgd` progra
-000100c0: 6d2c 2062 7574 2077 7269 7465 7320 6164  m, but writes ad
-000100d0: 6469 7469 6f6e 616c 6c79 2074 6865 2070  ditionally the p
-000100e0: 726f 6261 6269 6c69 7479 206f 6620 6561  robability of ea
-000100f0: 6368 202a 4b2a 3c73 7562 3e53 3c2f 7375  ch *K*<sub>S</su
-00010100: 623e 2076 616c 7565 2069 6e74 6f20 7468  b> value into th
-00010110: 6520 6669 6e61 6c20 6461 7461 6672 616d  e final datafram
-00010120: 652e 2042 6173 6963 616c 6c79 2c20 7573  e. Basically, us
-00010130: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
-00010140: 6964 6520 7769 7468 2061 2028 6e6f 726d  ide with a (norm
-00010150: 616c 6c79 2066 726f 6d20 7768 6f6c 652d  ally from whole-
-00010160: 7061 7261 6e6f 6d65 206f 7220 616e 6368  paranome or anch
-00010170: 6f72 2d70 6169 7273 2920 2a4b 2a3c 7375  or-pairs) *K*<su
-00010180: 623e 533c 2f73 7562 3e20 6461 7461 6669  b>S</sub> datafi
-00010190: 6c65 2061 6e64 2074 6865 2047 4d4d 2061  le and the GMM a
-000101a0: 6e61 6c79 7369 7320 7769 6c6c 2062 6520  nalysis will be 
-000101b0: 636f 6e64 7563 7465 6420 7570 6f6e 2074  conducted upon t
-000101c0: 6865 2064 6174 6166 696c 652e 2053 6f6d  he datafile. Som
-000101d0: 6520 7061 7261 6d65 7465 7273 2063 616e  e parameters can
-000101e0: 2061 6666 6563 7420 7468 6520 7265 7375   affect the resu
-000101f0: 6c74 732c 2069 6e63 6c75 6469 6e67 2060  lts, including `
-00010200: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
-00010210: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-00010220: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-00010230: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
-00010240: 6175 6c74 2032 3030 292c 2060 2d2d 6d61  ault 200), `--ma
-00010250: 785f 6974 6572 602c 2077 6869 6368 2073  x_iter`, which s
-00010260: 6574 7320 7468 6520 6d61 7869 6d75 6d20  ets the maximum 
-00010270: 6e75 6d62 6572 206f 6620 6974 6572 6174  number of iterat
-00010280: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
-00010290: 3029 2c20 602d 2d6d 6574 686f 6460 2c20  0), `--method`, 
-000102a0: 7768 6963 6820 6465 7465 726d 696e 6573  which determines
-000102b0: 2077 6869 6368 2063 6c75 7374 6572 696e   which clusterin
-000102c0: 6720 6d65 7468 6f64 2074 6f20 7573 6520  g method to use 
-000102d0: 2864 6566 6175 6c74 2067 6d6d 292c 2060  (default gmm), `
-000102e0: 2d2d 6761 6d6d 6160 2c20 7768 6963 6820  --gamma`, which 
-000102f0: 7365 7473 2074 6865 2067 616d 6d61 2070  sets the gamma p
-00010300: 6172 616d 6574 6572 2066 6f72 2074 6865  arameter for the
-00010310: 2062 676d 6d20 6d6f 6465 6c20 2864 6566   bgmm model (def
-00010320: 6175 6c74 2030 2e30 3031 292c 2060 2d2d  ault 0.001), `--
-00010330: 636f 6d70 6f6e 656e 7473 602c 2077 6869  components`, whi
-00010340: 6368 2073 6574 7320 7468 6520 7261 6e67  ch sets the rang
-00010350: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
-00010360: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
-00010370: 2066 6974 2028 6465 6661 756c 7420 3120   fit (default 1 
-00010380: 3429 2c20 616e 6420 7468 6520 6461 7461  4), and the data
-00010390: 2066 696c 7465 7269 6e67 2070 6172 616d   filtering param
-000103a0: 6574 6572 7320 602d 2d66 696c 7465 7273  eters `--filters
-000103b0: 6020 7768 6963 6820 6669 6c74 6572 7320  ` which filters 
-000103c0: 6461 7461 2062 6173 6564 206f 6e20 616c  data based on al
-000103d0: 6967 6e6d 656e 7420 6c65 6e67 7468 2c20  ignment length, 
-000103e0: 602d 2d6b 735f 7261 6e67 6560 2077 6869  `--ks_range` whi
-000103f0: 6368 2066 696c 7465 7273 2064 6174 6120  ch filters data 
-00010400: 6261 7365 6420 6f6e 202a 4b2a 3c73 7562  based on *K*<sub
-00010410: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
-00010420: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
-00010430: 7220 602d 2d62 696e 7360 2077 6869 6368  r `--bins` which
-00010440: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-00010450: 206f 6620 6269 6e73 2069 6e20 2a4b 2a3c   of bins in *K*<
-00010460: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00010470: 7269 6275 7469 6f6e 2028 6465 6661 756c  ribution (defaul
-00010480: 7420 3530 292e 0a0a 2a2a 4120 7375 6767  t 50)...**A sugg
-00010490: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-000104a0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-000104b0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-000104c0: 6f77 2a2a 0a0a 6060 600a 7767 6420 6d69  ow**..```.wgd mi
-000104d0: 7820 6b73 6461 7461 0a60 6060 0a0a 2323  x ksdata.```..##
-000104e0: 2320 7767 6420 7065 616b 0a0a 2a2a 5468  # wgd peak..**Th
-000104f0: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
-00010500: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
-00010510: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
-00010520: 2069 6e20 5747 4420 6461 7469 6e67 2a2a   in WGD dating**
-00010530: 0a60 6060 0a77 6764 2070 6561 6b20 6b73  .```.wgd peak ks
-00010540: 6461 7461 202d 6170 2061 7064 6174 6120  data -ap apdata 
-00010550: 2d73 6d20 736d 6461 7461 202d 6c65 206c  -sm smdata -le l
-00010560: 6564 6174 6120 2d6d 7020 6d70 6461 7461  edata -mp mpdata
-00010570: 202d 2d68 6575 7269 7374 6963 2028 2d2d   --heuristic (--
-00010580: 616c 6967 6e66 696c 7465 7220 302e 3020  alignfilter 0.0 
-00010590: 3020 302e 3020 2d2d 6b73 7261 6e67 6520  0 0.0 --ksrange 
-000105a0: 3020 3520 2d2d 6269 6e5f 7769 6474 6820  0 5 --bin_width 
-000105b0: 302e 3120 2d2d 6775 6964 6520 7365 676d  0.1 --guide segm
-000105c0: 656e 7420 2d2d 7072 6f6d 696e 656e 6365  ent --prominence
-000105d0: 5f63 7574 6f66 6620 302e 3120 2d2d 7265  _cutoff 0.1 --re
-000105e0: 6c5f 6865 6967 6874 2030 2e34 202d 2d63  l_height 0.4 --c
-000105f0: 6920 3935 202d 2d68 6472 2039 3520 2d2d  i 95 --hdr 95 --
-00010600: 6b73 6375 746f 6666 2035 290a 6060 600a  kscutoff 5).```.
-00010610: 0a41 7320 6d65 6e74 696f 6e65 6420 7072  .As mentioned pr
-00010620: 6576 696f 7573 6c79 2c20 6120 6865 7572  eviously, a heur
-00010630: 6973 7469 6320 6d65 7468 6f64 2061 6e64  istic method and
-00010640: 2061 2063 6f6c 6c69 6e65 6172 2073 6567   a collinear seg
-00010650: 6d65 6e74 732d 6775 6964 6564 2061 6e63  ments-guided anc
-00010660: 686f 7220 7061 6972 2063 6c75 7374 6572  hor pair cluster
-00010670: 696e 6720 666f 7220 7468 6520 7365 6172  ing for the sear
-00010680: 6368 206f 6620 6372 6564 6961 626c 6520  ch of crediable 
-00010690: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000106a0: 7261 6e67 6520 7573 6564 2069 6e20 5747  range used in WG
-000106b0: 4420 6461 7469 6e67 2061 7265 2069 6d70  D dating are imp
-000106c0: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
-000106d0: 2076 3260 2e20 5573 6572 7320 6e65 6564   v2`. Users need
-000106e0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-000106f0: 616e 6368 6f72 2070 6f69 6e74 732c 2073  anchor points, s
-00010700: 6567 6d65 6e74 732c 206c 6973 7473 6567  egments, listseg
-00010710: 6d65 6e74 732c 206d 756c 7469 706c 6963  ments, multiplic
-00010720: 6f6e 7061 6972 7320 6461 7461 6669 6c65  onpairs datafile
-00010730: 2066 726f 6d20 6069 2d61 6468 6f72 6560   from `i-adhore`
-00010740: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-00010750: 636c 7573 7465 7269 6e67 2066 756e 6374  clustering funct
-00010760: 696f 6e2e 2053 6f6d 6520 7061 7261 6d65  ion. Some parame
-00010770: 7465 7273 2074 6861 7420 6361 6e20 696d  ters that can im
-00010780: 7061 6374 2074 6865 2072 6573 756c 7473  pact the results
-00010790: 2069 6e63 6c75 6465 2060 2d2d 616c 6967   include `--alig
-000107a0: 6e66 696c 7465 7260 2c20 7768 6963 6820  nfilter`, which 
-000107b0: 6669 6c74 6572 7320 7468 6520 6461 7461  filters the data
-000107c0: 2062 6173 6564 206f 6e20 616c 6967 6e6d   based on alignm
-000107d0: 656e 7420 6964 656e 7469 7479 2c20 6c65  ent identity, le
-000107e0: 6e67 7468 2061 6e64 2063 6f76 6572 6167  ngth and coverag
-000107f0: 652c 2060 2d2d 6b73 7261 6e67 6560 2c20  e, `--ksrange`, 
-00010800: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
-00010810: 616e 6765 206f 6620 4b73 2074 6f20 6265  ange of Ks to be
-00010820: 2061 6e61 6c79 7a65 642c 2060 2d2d 6269   analyzed, `--bi
-00010830: 6e5f 7769 6474 6860 2c20 7768 6963 6820  n_width`, which 
-00010840: 7365 7473 2074 6865 2062 616e 6477 6964  sets the bandwid
-00010850: 7468 206f 6620 2a4b 2a3c 7375 623e 533c  th of *K*<sub>S<
-00010860: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00010870: 6f6e 2c20 602d 2d77 6569 6768 7473 5f6f  on, `--weights_o
-00010880: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
-00010890: 6020 7768 6963 6820 6465 7465 726d 696e  ` which determin
-000108a0: 6573 2077 6865 7468 6572 2074 6f20 696e  es whether to in
-000108b0: 636c 7564 6520 2a4b 2a3c 7375 623e 533c  clude *K*<sub>S<
-000108c0: 2f73 7562 3e20 6f75 746c 6965 7273 2028  /sub> outliers (
-000108d0: 7768 6f73 6520 7661 6c75 6520 6973 206f  whose value is o
-000108e0: 7665 7220 3529 2069 6e20 616e 616c 7973  ver 5) in analys
-000108f0: 6973 2c20 602d 2d6d 6574 686f 6460 2077  is, `--method` w
-00010900: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
-00010910: 7768 6963 6820 636c 7573 7465 7269 6e67  which clustering
-00010920: 206d 6574 686f 6420 746f 2075 7365 2028   method to use (
-00010930: 6465 6661 756c 7420 676d 6d29 2c20 602d  default gmm), `-
-00010940: 2d73 6565 6460 2077 6869 6368 2073 6574  -seed` which set
-00010950: 7320 7468 6520 7261 6e64 6f6d 2073 6565  s the random see
-00010960: 6420 6769 7665 6e20 746f 2069 6e69 7469  d given to initi
-00010970: 616c 697a 6174 696f 6e20 2864 6566 6175  alization (defau
-00010980: 6c74 2032 3335 3238 3930 292c 2060 2d2d  lt 2352890), `--
-00010990: 6e5f 696e 6974 602c 2077 6869 6368 2073  n_init`, which s
-000109a0: 6574 7320 7468 6520 6e75 6d62 6572 206f  ets the number o
-000109b0: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
-000109c0: 6c69 7a61 7469 6f6e 7320 2864 6566 6175  lizations (defau
-000109d0: 6c74 2032 3030 292c 2060 2d2d 656d 5f69  lt 200), `--em_i
-000109e0: 7465 7260 2c20 7768 6963 6820 7365 7473  ter`, which sets
-000109f0: 2074 6865 206d 6178 696d 756d 206e 756d   the maximum num
-00010a00: 6265 7220 6f66 2069 7465 7261 7469 6f6e  ber of iteration
-00010a10: 7320 2864 6566 6175 6c74 2032 3030 292c  s (default 200),
-00010a20: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
-00010a30: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
-00010a40: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-00010a50: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
-00010a60: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
-00010a70: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
-00010a80: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
-00010a90: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-00010aa0: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-00010ab0: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
-00010ac0: 3120 3429 2c20 602d 2d77 6569 6768 7465  1 4), `--weighte
-00010ad0: 6460 2077 6869 6368 2064 6574 6572 6d69  d` which determi
-00010ae0: 6e65 7320 7768 6574 6865 7220 746f 2075  nes whether to u
-00010af0: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
-00010b00: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
-00010b10: 6564 756e 6461 6e63 792c 2060 2d2d 6775  edundancy, `--gu
-00010b20: 6964 6560 2077 6869 6368 2064 6574 6572  ide` which deter
-00010b30: 6d69 6e65 7320 7768 6963 6820 7265 6769  mines which regi
-00010b40: 6d65 2072 6573 6964 696e 6720 616e 6368  me residing anch
-00010b50: 6f72 7320 746f 2062 6520 7573 6564 2028  ors to be used (
-00010b60: 6465 6661 756c 7420 5365 676d 656e 7429  default Segment)
-00010b70: 2c20 602d 2d70 726f 6d69 6e65 6e63 655f  , `--prominence_
-00010b80: 6375 746f 6666 6020 7768 6963 6820 7365  cutoff` which se
-00010b90: 7473 2074 6865 2070 726f 6d69 6e65 6e63  ts the prominenc
-00010ba0: 6520 6375 746f 6666 206f 6620 6163 6365  e cutoff of acce
-00010bb0: 7074 6162 6c65 2070 6561 6b73 2069 6e20  ptable peaks in 
-00010bc0: 7065 616b 2066 696e 6469 6e67 2070 726f  peak finding pro
-00010bd0: 6365 7373 2c20 602d 2d72 656c 5f68 6569  cess, `--rel_hei
-00010be0: 6768 7460 2077 6869 6368 2073 6574 7320  ght` which sets 
-00010bf0: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
-00010c00: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
-00010c10: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
-00010c20: 6561 7375 7265 642c 2060 2d2d 6b73 746f  easured, `--ksto
-00010c30: 6461 7465 6020 7768 6963 6820 6d61 6e75  date` which manu
-00010c40: 616c 6c79 2073 6574 7320 7468 6520 7261  ally sets the ra
-00010c50: 6e67 6520 6f66 202a 4b2a 3c73 7562 3e53  nge of *K*<sub>S
-00010c60: 3c2f 7375 623e 2074 6f20 6265 2064 6174  </sub> to be dat
-00010c70: 6564 2069 6e20 6865 7572 6973 7469 6320  ed in heuristic 
-00010c80: 7365 6172 6368 2061 6e64 206e 6565 6473  search and needs
-00010c90: 2074 6f20 6265 2063 6f2d 7365 7420 7769   to be co-set wi
-00010ca0: 7468 206f 7074 696f 6e20 602d 2d6d 616e  th option `--man
-00010cb0: 7561 6c73 6574 602c 2060 2d2d 786c 696d  ualset`, `--xlim
-00010cc0: 6020 616e 6420 602d 2d79 6c69 6d60 2064  ` and `--ylim` d
-00010cd0: 6574 6572 6d69 6e69 6e67 2074 6865 2078  etermining the x
-00010ce0: 2061 6e64 2079 2061 7869 7320 6c69 6d69   and y axis limi
-00010cf0: 7420 6f66 2047 4d4d 202a 4b2a 3c73 7562  t of GMM *K*<sub
-00010d00: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-00010d10: 7574 696f 6e2c 2060 2d2d 6369 6020 7365  ution, `--ci` se
-00010d20: 7474 696e 6720 7468 6520 636f 6e66 6964  tting the confid
-00010d30: 656e 6365 206c 6576 656c 206f 6620 6c6f  ence level of lo
-00010d40: 672d 6e6f 726d 616c 2064 6973 7472 6962  g-normal distrib
-00010d50: 7574 696f 6e20 746f 2064 6174 6520 2864  ution to date (d
-00010d60: 6566 6175 6c74 2039 3529 2c20 602d 2d68  efault 95), `--h
-00010d70: 6472 6020 7365 7474 696e 6720 7468 6520  dr` setting the 
-00010d80: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
-00010d90: 7265 6769 6f6e 2028 4844 5229 2061 7070  region (HDR) app
-00010da0: 6c69 6564 2069 6e20 7468 6520 7365 676d  lied in the segm
-00010db0: 656e 742d 6775 6964 6564 2061 6e63 686f  ent-guided ancho
-00010dc0: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
-00010dd0: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
-00010de0: 696f 6e20 2864 6566 6175 6c74 2039 3529  ion (default 95)
-00010df0: 2c20 602d 2d68 6575 7269 7374 6963 6020  , `--heuristic` 
-00010e00: 6465 7465 726d 696e 696e 6720 7768 6574  determining whet
-00010e10: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00010e20: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
-00010e30: 206f 6620 6465 6669 6e69 6e67 2043 4920   of defining CI 
-00010e40: 666f 7220 6461 7469 6e67 2c20 602d 2d6b  for dating, `--k
-00010e50: 7363 7574 6f66 6660 2073 6574 7469 6e67  scutoff` setting
-00010e60: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00010e70: 7375 623e 2073 6174 7572 6174 696f 6e20  sub> saturation 
-00010e80: 6375 746f 6666 2069 6e20 6461 7469 6e67  cutoff in dating
-00010e90: 2028 6465 6661 756c 7420 3529 2e0a 0a46   (default 5)...F
-00010ea0: 6f75 7220 7265 7375 6c74 2073 7562 666c  our result subfl
-00010eb0: 6f64 6572 7320 7769 6c6c 2062 6520 7072  oders will be pr
-00010ec0: 6f64 7563 6564 2c20 6e61 6d65 6c79 2060  oduced, namely `
-00010ed0: 416e 6368 6f72 4b73 5f46 696e 6450 6561  AnchorKs_FindPea
-00010ee0: 6b60 2c20 6041 6e63 686f 724b 735f 474d  k`, `AnchorKs_GM
-00010ef0: 4d60 2c20 6053 6567 6d65 6e74 4775 6964  M`, `SegmentGuid
-00010f00: 654b 735f 474d 4d60 2061 6e64 2060 5365  eKs_GMM` and `Se
-00010f10: 676d 656e 744b 735f 4669 6e64 5065 616b  gmentKs_FindPeak
-00010f20: 602e 2054 6865 2060 416e 6368 6f72 4b73  `. The `AnchorKs
-00010f30: 5f46 696e 6450 6561 6b60 2073 7562 666c  _FindPeak` subfl
-00010f40: 6f64 6572 2063 6f6e 7461 696e 7320 7265  oder contains re
-00010f50: 7375 6c74 7320 6f66 2074 6865 2064 6574  sults of the det
-00010f60: 6563 7465 6420 7065 616b 7320 6279 2074  ected peaks by t
-00010f70: 6865 2060 7369 676e 616c 6020 6d6f 6475  he `signal` modu
-00010f80: 6c65 206f 6620 6053 6369 5079 6020 6c69  le of `SciPy` li
-00010f90: 6272 6172 7920 616e 6420 7468 6520 6173  brary and the as
-00010fa0: 7375 6d65 6420 6869 6768 6573 7420 6d61  sumed highest ma
-00010fb0: 7373 2070 6172 7420 2872 6566 6572 7265  ss part (referre
-00010fc0: 6420 746f 2061 7320 4869 6768 4d61 7373  d to as HighMass
-00010fd0: 2068 6572 6561 6674 6572 2920 6f66 2065   hereafter) of e
-00010fe0: 6163 6820 7065 616b 2c20 7768 6963 6820  ach peak, which 
-00010ff0: 6361 6e20 6265 2075 7365 6420 666f 7220  can be used for 
-00011000: 6675 7274 6865 7220 5747 4420 6461 7469  further WGD dati
-00011010: 6e67 2e20 5468 6520 6041 6e63 686f 724b  ng. The `AnchorK
-00011020: 735f 474d 4d60 2073 686f 7773 2074 6865  s_GMM` shows the
-00011030: 2047 4d4d 2072 6573 756c 7473 2075 706f   GMM results upo
-00011040: 6e20 7468 6520 6f72 6967 696e 616c 2061  n the original a
-00011050: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
-00011060: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00011070: 6f6e 2062 7920 7468 6520 606d 6978 7475  on by the `mixtu
-00011080: 7265 6020 6d6f 6475 6c65 206f 6620 6073  re` module of `s
-00011090: 6369 6b69 742d 6c65 6172 6e60 206c 6962  cikit-learn` lib
-000110a0: 7261 7279 2061 6e64 2074 776f 2073 7562  rary and two sub
-000110b0: 666c 6f64 6572 732c 2060 4c6f 6747 4d4d  floders, `LogGMM
-000110c0: 5f43 4960 2063 6f6e 7461 696e 696e 6720  _CI` containing 
-000110d0: 7468 6520 7265 7375 6c74 7320 6f66 2039  the results of 9
-000110e0: 3525 2043 4920 6f66 2065 6163 6820 636f  5% CI of each co
-000110f0: 6d70 6f6e 656e 742c 2060 4869 6768 4d61  mponent, `HighMa
-00011100: 7373 5f43 4960 2063 6f6e 7461 696e 696e  ss_CI` containin
-00011110: 6720 7468 6520 4869 6768 4d61 7373 206f  g the HighMass o
-00011120: 6620 6561 6368 2063 6f6d 706f 6e65 6e74  f each component
-00011130: 2c20 7768 6963 6820 6361 6e20 6265 2075  , which can be u
-00011140: 7365 6420 666f 7220 6675 7274 6865 7220  sed for further 
-00011150: 5747 4420 6461 7469 6e67 2e20 5468 6520  WGD dating. The 
-00011160: 6053 6567 6d65 6e74 4775 6964 654b 735f  `SegmentGuideKs_
-00011170: 474d 4d60 2073 7562 666f 6c64 6572 2070  GMM` subfolder p
-00011180: 7265 7365 6e74 7320 7265 7375 6c74 7320  resents results 
-00011190: 6f66 2073 6567 6d65 6e74 202a 4b2a 3c73  of segment *K*<s
-000111a0: 7562 3e53 3c2f 7375 623e 2047 4d4d 2077  ub>S</sub> GMM w
-000111b0: 6869 6368 2061 7265 206d 6170 7065 6420  hich are mapped 
-000111c0: 6261 636b 2074 6f20 7468 6520 7265 7369  back to the resi
-000111d0: 6469 6e67 2061 6e63 686f 7220 7061 6972  ding anchor pair
-000111e0: 7320 616e 6420 7468 6520 6173 736f 6369  s and the associ
-000111f0: 6174 6564 2039 3525 2048 4452 2061 6e64  ated 95% HDR and
-00011200: 2048 6967 684d 6173 7320 6f66 2065 6163   HighMass of eac
-00011210: 6820 7365 676d 656e 7420 636c 7573 7465  h segment cluste
-00011220: 7220 696e 2073 7562 666c 6f64 6572 7320  r in subfloders 
-00011230: 6f66 2060 4844 525f 4349 6020 616e 6420  of `HDR_CI` and 
-00011240: 6048 6967 684d 6173 735f 4349 602e 2054  `HighMass_CI`. T
-00011250: 6865 2060 5365 676d 656e 744b 735f 4669  he `SegmentKs_Fi
-00011260: 6e64 5065 616b 6020 7375 6266 6f6c 6465  ndPeak` subfolde
-00011270: 7220 6973 2073 696d 696c 6172 2074 6f20  r is similar to 
-00011280: 6041 6e63 686f 724b 735f 4669 6e64 5065  `AnchorKs_FindPe
-00011290: 616b 6020 6275 7420 7769 7468 2073 6567  ak` but with seg
-000112a0: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
-000112b0: 7375 623e 2069 6e73 7465 6164 2e20 5468  sub> instead. Th
-000112c0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000112d0: 3e20 696e 2060 4d75 6c74 6970 6c69 636f  > in `Multiplico
-000112e0: 6e60 2063 616e 2061 6c73 6f20 6265 2063  n` can also be c
-000112f0: 616c 6375 6c61 7465 6420 696e 2070 6c61  alculated in pla
-00011300: 6365 206f 6620 6053 6567 6d65 6e74 6020  ce of `Segment` 
-00011310: 7573 696e 6720 7468 6520 6f70 7469 6f6e  using the option
-00011320: 2060 2d2d 6775 6964 6560 2061 7320 7375   `--guide` as su
-00011330: 6368 2074 6865 2072 6573 756c 7420 7469  ch the result ti
-00011340: 746c 652c 206c 6162 656c 2c20 6669 6c65  tle, label, file
-00011350: 2061 6e64 2066 6f6c 6465 7220 6e61 6d65   and folder name
-00011360: 7320 7769 6c6c 2062 6520 6368 616e 6765  s will be change
-00011370: 6420 6163 636f 7264 696e 676c 792e 0a0a  d accordingly...
-00011380: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-00011390: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-000113a0: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
-000113b0: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
-000113c0: 600a 7767 6420 7065 616b 206b 7364 6174  `.wgd peak ksdat
-000113d0: 6120 2d61 7020 6170 6461 7461 202d 736d  a -ap apdata -sm
-000113e0: 2073 6d64 6174 6120 2d6c 6520 6c65 6461   smdata -le leda
-000113f0: 7461 202d 6d70 206d 7064 6174 6120 2d2d  ta -mp mpdata --
-00011400: 6865 7572 6973 7469 630a 6060 600a 0a23  heuristic.```..#
-00011410: 2323 2077 6764 2073 796e 0a0a 2a2a 5468  ## wgd syn..**Th
-00011420: 6520 696e 7472 612d 7370 6563 6966 6963  e intra-specific
-00011430: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
-00011440: 6365 2a2a 0a60 6060 0a77 6764 2073 796e  ce**.```.wgd syn
-00011450: 2066 616d 696c 6965 7320 6766 6620 282d   families gff (-
-00011460: 2d6b 735f 6469 7374 7269 6275 7469 6f6e  -ks_distribution
-00011470: 206b 7364 6174 6120 2d66 2067 656e 6520   ksdata -f gene 
-00011480: 2d61 2049 4420 2d2d 6d69 6e6c 656e 202d  -a ID --minlen -
-00011490: 3120 2d2d 6d69 6e73 6567 6c65 6e20 3130  1 --minseglen 10
-000114a0: 3030 3020 2d2d 6d69 6e67 656e 656e 756d  000 --mingenenum
-000114b0: 2033 3029 0a60 6060 0a0a 5468 6520 7072   30).```..The pr
-000114c0: 6f67 7261 6d20 6077 6764 2073 796e 6020  ogram `wgd syn` 
-000114d0: 6973 206d 6169 6e6c 7920 6465 616c 696e  is mainly dealin
-000114e0: 6720 7769 7468 2063 6f6c 6c69 6e65 6172  g with collinear
-000114f0: 6974 7920 6f72 2073 796e 7465 6e79 2028  ity or synteny (
-00011500: 626f 7468 2072 6566 6572 7265 6420 746f  both referred to
-00011510: 2061 7320 7379 6e74 656e 7920 6865 7265   as synteny here
-00011520: 6166 7465 7229 2061 6e61 6c79 7369 732e  after) analysis.
-00011530: 2054 776f 2069 6e70 7574 2066 696c 6573   Two input files
-00011540: 2061 7265 2065 7373 656e 7469 616c 2c20   are essential, 
-00011550: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
-00011560: 6669 6c65 2061 6e64 2074 6865 2067 6666  file and the gff
-00011570: 3320 6669 6c65 2e20 5468 6520 6765 6e65  3 file. The gene
-00011580: 2066 616d 696c 7920 6669 6c65 2069 7320   family file is 
-00011590: 696e 2074 6865 2066 6f72 6d61 7420 6173  in the format as
-000115a0: 2060 4f72 7468 6f46 696e 6465 7260 2e20   `OrthoFinder`. 
-000115b0: 5468 6520 736f 6674 7761 7265 2060 692d  The software `i-
-000115c0: 6164 686f 7265 6020 6973 2061 2070 7265  adhore` is a pre
-000115d0: 7265 7175 6973 6974 652e 2057 6974 6820  requisite. With 
-000115e0: 6465 6661 756c 7420 7061 7261 6d65 7465  default paramete
-000115f0: 7273 2c20 7468 6520 7072 6f67 7261 6d20  rs, the program 
-00011600: 6261 7369 6361 6c6c 7920 636f 6e64 7563  basically conduc
-00011610: 7473 2031 2920 6669 6c74 6572 696e 6720  ts 1) filtering 
-00011620: 6765 6e65 2066 616d 696c 6965 7320 6261  gene families ba
-00011630: 7365 6420 6f6e 206d 6178 696d 756d 2066  sed on maximum f
-00011640: 616d 696c 7920 7369 7a65 2032 2920 7265  amily size 2) re
-00011650: 7472 6965 7669 6e67 2067 656e 6520 706f  trieving gene po
-00011660: 7369 7469 6f6e 2061 6e64 2073 6361 6666  sition and scaff
-00011670: 6f6c 6420 696e 666f 726d 6174 696f 6e20  old information 
-00011680: 6672 6f6d 2067 6666 3320 6669 6c65 2033  from gff3 file 3
-00011690: 2920 7072 6f64 7563 696e 6720 7468 6520  ) producing the 
-000116a0: 636f 6e66 6967 7572 6174 696f 6e20 6669  configuration fi
-000116b0: 6c65 2061 6e64 2061 7373 6f63 6961 7465  le and associate
-000116c0: 6420 6461 7461 6669 6c65 7320 666f 7220  d datafiles for 
-000116d0: 6069 2d61 6468 6f72 6560 2034 2920 6361  `i-adhore` 4) ca
-000116e0: 6c6c 696e 6720 6069 2d61 6468 6f72 6560  lling `i-adhore`
-000116f0: 2067 6976 656e 2074 6865 2070 6172 616d   given the param
-00011700: 6574 6572 7320 7365 7420 746f 2069 6e66  eters set to inf
-00011710: 6572 2073 796e 7465 6e79 2035 2920 7669  er synteny 5) vi
-00011720: 7375 616c 697a 696e 6720 7468 6520 7379  sualizing the sy
-00011730: 6e74 656e 7920 696e 2022 646f 7470 6c6f  nteny in "dotplo
-00011740: 7422 2069 6e20 7468 6520 756e 6974 206f  t" in the unit o
-00011750: 6620 6765 6e65 7320 616e 6420 6261 7365  f genes and base
-00011760: 732c 2069 6e20 2253 796e 6465 7074 6822  s, in "Syndepth"
-00011770: 2070 6c6f 7420 7368 6f77 696e 6720 7468   plot showing th
-00011780: 6520 6469 7374 7269 6275 7469 6f6e 206f  e distribution o
-00011790: 6620 6469 6666 6572 656e 7420 6361 7465  f different cate
-000117a0: 676f 7269 6573 206f 6620 636f 6c6c 696e  gories of collin
-000117b0: 6561 7269 7479 2072 6174 696f 7320 7769  earity ratios wi
-000117c0: 7468 696e 2061 6e64 2062 6574 7765 656e  thin and between
-000117d0: 2073 7065 6369 6573 2c20 696e 2022 6475   species, in "du
-000117e0: 7053 7461 636b 2220 706c 6f74 2073 686f  pStack" plot sho
-000117f0: 7769 6e67 206d 756c 7469 706c 6963 6f6e  wing multiplicon
-00011800: 7320 7769 7468 2064 6966 6665 7265 6e74  s with different
-00011810: 206d 756c 7469 706c 6963 6174 696f 6e20   multiplication 
-00011820: 6c65 7665 6c73 2e20 3629 2069 6620 7769  levels. 6) if wi
-00011830: 7468 202a 4b2a 3c73 7562 3e53 3c2f 7375  th *K*<sub>S</su
-00011840: 623e 2064 6174 612c 2061 2022 2a4b 2a3c  b> data, a "*K*<
-00011850: 7375 623e 533c 2f73 7562 3e20 646f 7470  sub>S</sub> dotp
-00011860: 6c6f 7422 2077 6974 6820 646f 7473 2061  lot" with dots a
-00011870: 6e6e 6f74 6174 6564 2069 6e20 2a4b 2a3c  nnotated in *K*<
-00011880: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
-00011890: 6573 2061 6e64 2061 202a 4b2a 3c73 7562  es and a *K*<sub
-000118a0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-000118b0: 7574 696f 6e20 7769 7468 2061 6e63 686f  ution with ancho
-000118c0: 7220 7061 6972 7320 6465 6e6f 7465 6420  r pairs denoted 
-000118d0: 7769 6c6c 2062 6520 7072 6f64 7563 6564  will be produced
-000118e0: 2e20 5468 6520 6765 6e65 2069 6e66 6f72  . The gene infor
-000118f0: 6d61 7469 6f6e 2069 6e20 7468 6520 6765  mation in the ge
-00011900: 6e65 2066 616d 696c 7920 6669 6c65 2061  ne family file a
-00011910: 6e64 2067 6666 3320 6669 6c65 2073 686f  nd gff3 file sho
-00011920: 756c 6420 6265 206d 6174 6368 6564 2077  uld be matched w
-00011930: 6869 6368 2072 6571 7569 7265 7320 7573  hich requires us
-00011940: 6572 7320 746f 2073 6574 2070 726f 7065  ers to set prope
-00011950: 7220 602d 2d66 6561 7475 7265 6020 616e  r `--feature` an
-00011960: 6420 602d 2d61 7474 7269 6275 7465 602e  d `--attribute`.
-00011970: 2054 6865 206d 6178 696d 756d 2066 616d   The maximum fam
-00011980: 696c 7920 7369 7a65 2074 6f20 6265 2069  ily size to be i
-00011990: 6e63 6c75 6465 6420 6361 6e20 6265 2073  ncluded can be s
-000119a0: 6574 2076 6961 2074 6865 206f 7074 696f  et via the optio
-000119b0: 6e20 602d 2d6d 6178 7369 7a65 602c 206e  n `--maxsize`, n
-000119c0: 6f74 6564 2074 6861 7420 7468 6973 2066  oted that this f
-000119d0: 696c 7465 7269 6e67 2069 7320 6d61 696e  iltering is main
-000119e0: 6c79 2074 6f20 6472 6f70 2074 686f 7365  ly to drop those
-000119f0: 2068 7567 6520 7461 6e64 656d 2064 7570   huge tandem dup
-00011a00: 6c69 6361 7465 7320 6661 6d69 6c79 2061  licates family a
-00011a10: 6e64 2074 7261 6e73 706f 7361 626c 6520  nd transposable 
-00011a20: 656c 656d 656e 7473 2028 5445 7329 2066  elements (TEs) f
-00011a30: 616d 696c 792c 2061 6e64 206e 6f74 206d  amily, and not m
-00011a40: 616e 6461 746f 7279 2e20 5573 6572 7320  andatory. Users 
-00011a50: 6361 6e20 6669 6c74 6572 2074 686f 7365  can filter those
-00011a60: 2066 7261 676d 656e 7461 7279 2073 6361   fragmentary sca
-00011a70: 6666 6f6c 6473 2076 6961 2074 6865 206f  ffolds via the o
-00011a80: 7074 696f 6e20 602d 2d6d 696e 6c65 6e60  ption `--minlen`
-00011a90: 2e20 5468 6520 6d69 6e69 6d75 6d20 6c65  . The minimum le
-00011aa0: 6e67 7468 2061 6e64 206e 756d 6265 7220  ngth and number 
-00011ab0: 6f66 2067 656e 6573 2066 6f72 2061 2073  of genes for a s
-00011ac0: 6567 6d65 6e74 2074 6f20 6265 2063 6f6e  egment to be con
-00011ad0: 7369 6465 7265 6420 6361 6e20 6265 2073  sidered can be s
-00011ae0: 6574 2076 6961 2074 6865 206f 7074 696f  et via the optio
-00011af0: 6e20 602d 2d6d 696e 7365 676c 656e 6020  n `--minseglen` 
-00011b00: 616e 6420 602d 2d6d 696e 6765 6e65 6e75  and `--mingenenu
-00011b10: 6d60 2e20 5265 6475 6e64 616e 7420 6d75  m`. Redundant mu
-00011b20: 6c74 6970 6c69 636f 6e73 2063 616e 2062  ltiplicons can b
-00011b30: 6520 6b65 7074 2062 7920 7365 7420 7468  e kept by set th
-00011b40: 6520 666c 6167 206f 7074 696f 6e20 602d  e flag option `-
-00011b50: 2d6b 6565 7072 6564 756e 602e 0a0a 2a2a  -keepredun`...**
-00011b60: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-00011b70: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-00011b80: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
-00011b90: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
-00011ba0: 7767 6420 7379 6e20 6661 6d69 6c69 6573  wgd syn families
-00011bb0: 2067 6666 0a60 6060 0a0a 2a2a 5468 6520   gff.```..**The 
-00011bc0: 696e 7465 722d 7370 6563 6966 6963 2073  inter-specific s
-00011bd0: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
-00011be0: 2a2a 0a60 6060 0a77 6764 2073 796e 2066  **.```.wgd syn f
-00011bf0: 616d 696c 6965 7320 6766 6631 2067 6666  amilies gff1 gff
-00011c00: 3220 282d 2d61 6464 6974 696f 6e61 6c67  2 (--additionalg
-00011c10: 6666 696e 666f 2022 6d52 4e41 3b4e 616d  ffinfo "mRNA;Nam
-00011c20: 6522 202d 2d61 6464 6974 696f 6e61 6c67  e" --additionalg
-00011c30: 6666 696e 666f 2022 6765 6e65 3b49 4422  ffinfo "gene;ID"
-00011c40: 290a 6060 600a 0a46 6f72 206d 756c 7469  ).```..For multi
-00011c50: 2d73 7065 6369 6573 2073 796e 7465 6e79  -species synteny
-00011c60: 2069 6e66 6572 656e 6365 2c20 6966 2075   inference, if u
-00011c70: 7365 7273 2068 6176 6520 6766 6633 2066  sers have gff3 f
-00011c80: 696c 6573 2077 6869 6368 2068 6176 6520  iles which have 
-00011c90: 6469 6666 6572 656e 7420 6665 6174 7572  different featur
-00011ca0: 6573 206f 7220 6174 7472 6962 7574 6573  es or attributes
-00011cb0: 2066 6f72 2067 656e 6520 706f 7369 7469   for gene positi
-00011cc0: 6f6e 2069 6e66 6f72 6d61 7469 6f6e 2072  on information r
-00011cd0: 6574 7269 6576 616c 2c20 7468 6520 6f70  etrieval, the op
-00011ce0: 7469 6f6e 2060 2d2d 6164 6469 7469 6f6e  tion `--addition
-00011cf0: 616c 6766 6669 6e66 6f60 2063 616e 2062  algffinfo` can b
-00011d00: 6520 7365 7420 746f 2070 726f 7669 6465  e set to provide
-00011d10: 2074 6865 2061 6464 6974 696f 6e61 6c20   the additional 
-00011d20: 696e 666f 726d 6174 696f 6e2e 2054 6865  information. The
-00011d30: 2072 656d 6169 6e69 6e67 2070 6172 616d   remaining param
-00011d40: 6574 6572 2073 6574 7469 6e67 2069 7320  eter setting is 
-00011d50: 7468 6520 7361 6d65 2061 7320 7468 6520  the same as the 
-00011d60: 696e 7472 612d 7370 6563 6966 6963 2073  intra-specific s
-00011d70: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
-00011d80: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
-00011d90: 2073 7461 7274 696e 6720 7275 6e20 6361   starting run ca
-00011da0: 6e20 7573 6520 636f 6d6d 616e 6420 7369  n use command si
-00011db0: 6d70 6c79 2061 7320 6265 6c6f 772a 2a0a  mply as below**.
-00011dc0: 0a60 6060 0a77 6764 2073 796e 2066 616d  .```.wgd syn fam
-00011dd0: 696c 6965 7320 6766 6631 2067 6666 320a  ilies gff1 gff2.
-00011de0: 6060 600a 0a23 2323 2077 6764 2076 697a  ```..### wgd viz
-00011df0: 0a0a 2a2a 5468 6520 7669 7375 616c 697a  ..**The visualiz
-00011e00: 6174 696f 6e20 6f66 202a 4b2a 3c73 7562  ation of *K*<sub
-00011e10: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-00011e20: 7472 6962 7574 696f 6e20 616e 6420 454c  tribution and EL
-00011e30: 4d4d 2061 6e61 6c79 7369 732a 2a0a 6060  MM analysis**.``
-00011e40: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
-00011e50: 6174 610a 6060 600a 0a54 6865 2070 726f  ata.```..The pro
-00011e60: 6772 616d 2060 7767 6420 7669 7a60 2069  gram `wgd viz` i
-00011e70: 7320 6d61 696e 6c79 2066 6f72 2074 6865  s mainly for the
-00011e80: 2070 7572 706f 7365 206f 6620 2a4b 2a3c   purpose of *K*<
-00011e90: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00011ea0: 7269 6275 7469 6f6e 2061 6e64 2073 796e  ribution and syn
-00011eb0: 7465 6e79 2076 6973 7561 6c69 7a61 7469  teny visualizati
-00011ec0: 6f6e 2c20 7769 7468 2073 6f6d 6520 6f70  on, with some op
-00011ed0: 7469 6f6e 616c 206d 6978 7475 7265 206d  tional mixture m
-00011ee0: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
-00011ef0: 2e20 5468 6520 6261 7369 6320 6675 6e63  . The basic func
-00011f00: 7469 6f6e 2069 7320 6a75 7374 2074 6f20  tion is just to 
-00011f10: 706c 6f74 2074 6865 202a 4b2a 3c73 7562  plot the *K*<sub
-00011f20: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-00011f30: 7574 696f 6e20 616e 6420 636f 6e64 7563  ution and conduc
-00011f40: 7420 616e 2045 4c4d 4d20 616e 616c 7973  t an ELMM analys
-00011f50: 6973 2069 6e20 7365 6172 6368 206f 6620  is in search of 
-00011f60: 706f 7465 6e74 6961 6c20 5747 4420 636f  potential WGD co
-00011f70: 6d70 6f6e 656e 7473 2e20 536f 6d65 206b  mponents. Some k
-00011f80: 6579 2070 6172 616d 6574 6572 7320 6166  ey parameters af
-00011f90: 6665 6374 696e 6720 7468 6520 454c 4d4d  fecting the ELMM
-00011fa0: 2072 6573 756c 7420 696e 636c 7564 6520   result include 
-00011fb0: 602d 2d70 726f 6d69 6e65 6e63 655f 6375  `--prominence_cu
-00011fc0: 746f 6666 6020 616e 6420 602d 2d72 656c  toff` and `--rel
-00011fd0: 5f68 6569 6768 7460 2c20 7768 6963 6820  _height`, which 
-00011fe0: 6861 7665 2062 6565 6e20 6578 706c 6169  have been explai
-00011ff0: 6e65 6420 6168 6561 642c 2060 2d2d 656d  ned ahead, `--em
-00012000: 5f69 7465 7261 7469 6f6e 7360 2061 6e64  _iterations` and
-00012010: 2060 2d2d 656d 5f69 6e69 7469 616c 697a   `--em_initializ
-00012020: 6174 696f 6e73 6020 6465 7465 726d 696e  ations` determin
-00012030: 696e 6720 7468 6520 6d61 7869 6d75 6d20  ing the maximum 
-00012040: 6974 6572 6174 696f 6e73 2061 6e64 2069  iterations and i
-00012050: 6e69 7469 616c 697a 6174 696f 6e73 2069  nitializations i
-00012060: 6e20 7468 6520 454d 2061 6c67 6f72 6974  n the EM algorit
-00012070: 686d 2e0a 0a2a 2a41 2073 7567 6765 7374  hm...**A suggest
-00012080: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-00012090: 6361 6e20 7573 6520 636f 6d6d 616e 6420  can use command 
-000120a0: 7369 6d70 6c79 2061 7320 6265 6c6f 772a  simply as below*
-000120b0: 2a0a 0a60 6060 0a77 6764 2076 697a 202d  *..```.wgd viz -
-000120c0: 6420 6b73 6461 7461 0a60 6060 0a0a 2a2a  d ksdata.```..**
-000120d0: 5468 6520 7669 7375 616c 697a 6174 696f  The visualizatio
-000120e0: 6e20 6f66 202a 4b2a 3c73 7562 3e53 3c2f  n of *K*<sub>S</
-000120f0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-00012100: 7574 696f 6e20 7769 7468 2072 6174 6520  ution with rate 
-00012110: 636f 7272 6563 7469 6f6e 2a2a 0a60 6060  correction**.```
-00012120: 0a77 6764 2076 697a 202d 6420 6b73 6461  .wgd viz -d ksda
-00012130: 7461 202d 7370 2073 7064 6174 6120 2d2d  ta -sp spdata --
-00012140: 666f 6375 7332 616c 6c20 666f 6361 6c5f  focus2all focal_
-00012150: 7370 6563 6965 7320 2d2d 6578 7472 6170  species --extrap
-00012160: 6172 616e 6f6d 656b 7320 6b73 6461 7461  aranomeks ksdata
-00012170: 2028 2d2d 616e 6368 6f72 706f 696e 7473   (--anchorpoints
-00012180: 2061 7064 6174 6120 2d2d 706c 6f74 6170   apdata --plotap
-00012190: 676d 6d20 2d2d 706c 6f74 656c 6d6d 290a  gmm --plotelmm).
-000121a0: 6060 600a 0a42 6573 6964 6573 2074 6865  ```..Besides the
-000121b0: 2062 6173 6963 202a 4b2a 3c73 7562 3e53   basic *K*<sub>S
-000121c0: 3c2f 7375 623e 2070 6c6f 742c 2073 7562  </sub> plot, sub
-000121d0: 7374 6974 7574 696f 6e20 7261 7465 2063  stitution rate c
-000121e0: 6f72 7265 6374 696f 6e20 6361 6e20 616c  orrection can al
-000121f0: 736f 2062 6520 6163 6869 6576 6564 2067  so be achieved g
-00012200: 6976 656e 2061 7420 6c65 6173 7420 6120  iven at least a 
-00012210: 7370 6563 6965 7320 7472 6565 2028 7669  species tree (vi
-00012220: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
-00012230: 7370 6563 6965 7374 7265 6560 2920 616e  speciestree`) an
-00012240: 6420 6120 666f 6361 6c20 7370 6563 6965  d a focal specie
-00012250: 7320 2865 6974 6865 7220 7669 6120 7468  s (either via th
-00012260: 6520 6f70 7469 6f6e 2060 2d2d 666f 6375  e option `--focu
-00012270: 7332 616c 6c60 206f 7220 7669 6120 7468  s2all` or via th
-00012280: 6520 6f70 7469 6f6e 2060 2d2d 7370 6169  e option `--spai
-00012290: 7260 2069 6e20 7468 6520 666f 726d 206f  r` in the form o
-000122a0: 6620 2224 666f 6361 6c5f 7370 6563 6965  f "$focal_specie
-000122b0: 733b 2466 6f63 616c 5f73 7065 6369 6573  s;$focal_species
-000122c0: 2229 2e20 4974 2773 2073 7567 6765 7374  "). It's suggest
-000122d0: 6564 2074 6861 7420 7468 6520 6f72 7468  ed that the orth
-000122e0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-000122f0: 533c 2f73 7562 3e20 6461 7461 2069 7320  S</sub> data is 
-00012300: 7072 6f76 6964 6564 2062 7920 7468 6520  provided by the 
-00012310: 602d 2d64 6174 6166 696c 6560 206f 7074  `--datafile` opt
-00012320: 696f 6e20 616e 6420 7468 6520 7061 7261  ion and the para
-00012330: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00012340: 3c2f 7375 623e 2064 6174 6120 6973 2070  </sub> data is p
-00012350: 726f 7669 6465 6420 6279 2074 6865 2060  rovided by the `
-00012360: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
-00012370: 7360 206f 7074 696f 6e2c 2061 6c74 686f  s` option, altho
-00012380: 7567 6820 6974 2773 2061 6c6c 6f77 6564  ugh it's allowed
-00012390: 2074 6f20 6f6e 6c79 2070 726f 7669 6465   to only provide
-000123a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000123b0: 2064 6174 6120 7669 6120 7468 6520 602d   data via the `-
-000123c0: 2d64 6174 6166 696c 6560 206f 7074 696f  -datafile` optio
-000123d0: 6e20 616e 6420 6465 706f 7369 7420 626f  n and deposit bo
-000123e0: 7468 206f 7274 686f 6c6f 676f 7573 2061  th orthologous a
-000123f0: 6e64 2070 6172 616c 6f67 6f75 7320 2a4b  nd paralogous *K
-00012400: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
-00012410: 7461 2074 6865 7265 6f6e 2e20 5468 6572  ta thereon. Ther
-00012420: 6520 6172 6520 7477 6f20 7479 7065 7320  e are two types 
-00012430: 6f66 206d 6978 6564 2070 6c6f 7473 2028  of mixed plots (
-00012440: 7468 6520 226d 6978 6564 2220 6865 7265  the "mixed" here
-00012450: 2072 6566 6572 7320 746f 206d 6978 6564   refers to mixed
-00012460: 206f 7274 686f 6c6f 676f 7573 2061 6e64   orthologous and
-00012470: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
-00012480: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00012490: 7269 6275 7469 6f6e 7329 2c20 6f6e 6520  ributions), one 
-000124a0: 6f66 2077 6869 6368 2069 7320 7369 6d69  of which is simi
-000124b0: 6c61 7220 746f 2077 6861 7420 606b 7372  lar to what `ksr
-000124c0: 6174 6573 6020 706c 6f74 732c 2077 6869  ates` plots, whi
-000124d0: 6c65 2074 6865 206f 7468 6572 206f 6620  le the other of 
-000124e0: 7768 6963 6820 6973 206c 696b 6520 7468  which is like th
-000124f0: 6520 636f 6e76 656e 7469 6f6e 616c 202a  e conventional *
-00012500: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
-00012510: 6c6f 7420 7468 6174 2062 6f74 6820 7468  lot that both th
-00012520: 6520 6f72 6967 696e 616c 206f 7274 686f  e original ortho
-00012530: 6c6f 676f 7573 2061 6e64 2070 6172 616c  logous and paral
-00012540: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-00012550: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00012560: 6f6e 7320 6172 6520 7472 7574 6866 756c  ons are truthful
-00012570: 6c79 2070 6c6f 7474 6564 2069 6e73 7465  ly plotted inste
-00012580: 6164 206f 6620 6a75 7374 2062 6569 6e67  ad of just being
-00012590: 2072 6570 7265 7365 6e74 6564 2062 7920   represented by 
-000125a0: 736f 6d65 2076 6572 7469 6361 6c20 6c69  some vertical li
-000125b0: 6e65 732e 2057 6520 7375 6767 6573 7420  nes. We suggest 
-000125c0: 7573 6572 7320 746f 2061 646f 7074 2074  users to adopt t
-000125d0: 6865 2060 6b73 7261 7465 7360 2d6c 696b  he `ksrates`-lik
-000125e0: 6520 706c 6f74 732c 2077 6869 6368 2069  e plots, which i
-000125f0: 7320 7468 6520 6465 6661 756c 742e 204f  s the default. O
-00012600: 7468 6572 7769 7365 2074 6865 206f 7074  therwise the opt
-00012610: 696f 6e20 602d 2d63 6c61 7373 6963 6020  ion `--classic` 
-00012620: 7769 6c6c 2062 6520 6e65 6564 6564 2074  will be needed t
-00012630: 6f20 7365 742e 2045 7874 7261 206d 6978  o set. Extra mix
-00012640: 7475 7265 206d 6f64 656c 696e 6720 616e  ture modeling an
-00012650: 616c 7973 6973 2063 616e 2062 6520 696e  alysis can be in
-00012660: 6974 6961 7465 6420 7669 6120 7468 6520  itiated via the 
-00012670: 6f70 7469 6f6e 2060 2d2d 706c 6f74 656c  option `--plotel
-00012680: 6d6d 6020 616e 6420 602d 2d70 6c6f 7461  mm` and `--plota
-00012690: 7067 6d6d 6020 7769 7468 2074 6865 2061  pgmm` with the a
-000126a0: 6e63 686f 7220 706f 696e 7473 2064 6174  nchor points dat
-000126b0: 6166 696c 6520 7072 6f76 6964 6564 2062  afile provided b
-000126c0: 7920 7468 6520 602d 2d61 6e63 686f 7270  y the `--anchorp
-000126d0: 6f69 6e74 7360 206f 7074 696f 6e2e 0a0a  oints` option...
-000126e0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-000126f0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-00012700: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
-00012710: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
-00012720: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
-00012730: 6174 6120 2d73 7020 7370 6461 7461 202d  ata -sp spdata -
-00012740: 2d66 6f63 7573 3261 6c6c 2066 6f63 616c  -focus2all focal
-00012750: 5f73 7065 6369 6573 202d 2d65 7874 7261  _species --extra
-00012760: 7061 7261 6e6f 6d65 6b73 206b 7364 6174  paranomeks ksdat
-00012770: 610a 6060 600a 0a2a 2a54 6865 2076 6973  a.```..**The vis
-00012780: 7561 6c69 7a61 7469 6f6e 206f 6620 7379  ualization of sy
-00012790: 6e74 656e 792a 2a0a 6060 600a 7767 6420  nteny**.```.wgd 
-000127a0: 7669 7a20 2d61 7020 6170 6461 7461 202d  viz -ap apdata -
-000127b0: 736d 2073 6d64 6174 6120 2d6d 7420 6d74  sm smdata -mt mt
-000127c0: 6461 7461 202d 6774 2067 7464 6174 6120  data -gt gtdata 
-000127d0: 2d2d 706c 6f74 7379 6e20 282d 2d6d 696e  --plotsyn (--min
-000127e0: 6c65 6e20 2d31 202d 2d6d 696e 7365 676c  len -1 --minsegl
-000127f0: 656e 2031 3030 3030 202d 2d6d 696e 6765  en 10000 --minge
-00012800: 6e65 6e75 6d20 3330 290a 6060 600a 0a43  nenum 30).```..C
-00012810: 6f6d 7061 7265 6420 746f 2074 6865 206f  ompared to the o
-00012820: 7269 6769 6e61 6c20 6077 6764 602c 2074  riginal `wgd`, t
-00012830: 6865 2060 7767 6420 7669 7a60 2070 726f  he `wgd viz` pro
-00012840: 6772 616d 2061 6464 6564 2073 796e 7465  gram added synte
-00012850: 6e79 2076 6973 7561 6c69 7a61 7469 6f6e  ny visualization
-00012860: 2070 6970 656c 696e 652e 2055 7365 7273   pipeline. Users
-00012870: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
-00012880: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
-00012890: 2060 2d2d 706c 6f74 7379 6e60 2074 6f20   `--plotsyn` to 
-000128a0: 696e 6974 6961 7465 2074 6869 7320 7061  initiate this pa
-000128b0: 7274 206f 6620 7069 7065 6c69 6e65 2e20  rt of pipeline. 
-000128c0: 5468 6973 2073 7465 7020 6173 7375 6d65  This step assume
-000128d0: 7320 7468 6174 2075 7365 7273 2068 6176  s that users hav
-000128e0: 6520 6f62 7461 696e 6564 2061 6c72 6561  e obtained alrea
-000128f0: 6479 2074 6865 2073 796e 7465 6e69 6320  dy the syntenic 
-00012900: 7265 7375 6c74 7320 6672 6f6d 2060 692d  results from `i-
-00012910: 6164 686f 7265 6020 616e 6420 7573 6573  adhore` and uses
-00012920: 2074 686f 7365 2072 6573 756c 7420 6669   those result fi
-00012930: 6c65 7320 746f 2072 6561 6c69 7a65 2074  les to realize t
-00012940: 6865 2073 796e 7465 6e79 2076 6973 7561  he synteny visua
-00012950: 6c69 7a61 7469 6f6e 2e20 5369 6d6c 6961  lization. Simlia
-00012960: 7220 746f 2060 7767 6420 7379 6e60 2c20  r to `wgd syn`, 
-00012970: 7468 6520 6578 7472 6120 2a4b 2a3c 7375  the extra *K*<su
-00012980: 623e 533c 2f73 7562 3e20 6461 7461 2066  b>S</sub> data f
-00012990: 696c 6520 6361 6e20 6265 2074 7261 6e73  ile can be trans
-000129a0: 6d69 7474 6564 2076 6961 2074 6865 206f  mitted via the o
-000129b0: 7074 696f 6e20 602d 2d64 6174 6166 696c  ption `--datafil
-000129c0: 6560 2028 696e 7374 6561 6420 6f66 2060  e` (instead of `
-000129d0: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
-000129e0: 7360 206f 7074 696f 6e29 2e20 4261 7369  s` option). Basi
-000129f0: 6361 6c6c 792c 2074 6865 2073 796e 7465  cally, the synte
-00012a00: 6e69 6320 7265 7375 6c74 2066 696c 6573  nic result files
-00012a10: 2072 6571 7569 7265 6420 6172 6520 7468   required are th
-00012a20: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-00012a30: 6461 7461 6669 6c65 2c20 6d75 6c74 6970  datafile, multip
-00012a40: 6c69 636f 6e73 2064 6174 6166 696c 652c  licons datafile,
-00012a50: 2067 656e 6520 7461 626c 6520 6461 7461   gene table data
-00012a60: 6669 6c65 2028 6175 746f 6d61 7465 6c79  file (automately
-00012a70: 2070 726f 6475 6365 6420 6279 2060 7767   produced by `wg
-00012a80: 6420 7379 6e60 292c 2061 6e64 2073 6567  d syn`), and seg
-00012a90: 6d65 6e74 7320 6461 7461 6669 6c65 2e0a  ments datafile..
-00012aa0: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
-00012ab0: 7461 7274 696e 6720 7275 6e20 6361 6e20  tarting run can 
-00012ac0: 7573 6520 636f 6d6d 616e 6420 7369 6d70  use command simp
-00012ad0: 6c79 2061 7320 6265 6c6f 772a 2a0a 0a60  ly as below**..`
-00012ae0: 6060 0a77 6764 2076 697a 202d 6170 2061  ``.wgd viz -ap a
-00012af0: 7064 6174 6120 2d73 6d20 736d 6461 7461  pdata -sm smdata
-00012b00: 202d 6d74 206d 7464 6174 6120 2d67 7420   -mt mtdata -gt 
-00012b10: 6774 6461 7461 202d 2d70 6c6f 7473 796e  gtdata --plotsyn
-00012b20: 0a60 6060 0a0a 2323 2049 6c6c 7573 7472  .```..## Illustr
-00012b30: 6174 696f 6e0a 0a57 6520 696c 6c75 7374  ation..We illust
-00012b40: 7261 7465 206f 7572 2070 726f 6772 616d  rate our program
-00012b50: 206f 6e20 616e 2065 7865 6d70 6c61 7279   on an exemplary
-00012b60: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
-00012b70: 6e64 2064 6174 696e 6720 7570 6f6e 2073  nd dating upon s
-00012b80: 7065 6369 6573 202a 4171 7569 6c65 6769  pecies *Aquilegi
-00012b90: 6120 636f 6572 756c 6561 2a2e 0a0a 5468  a coerulea*...Th
-00012ba0: 6520 2a41 7175 696c 6567 6961 2063 6f65  e *Aquilegia coe
-00012bb0: 7275 6c65 612a 2077 6173 2072 6570 6f72  rulea* was repor
-00012bc0: 7465 6420 746f 2065 7870 6572 6965 6e63  ted to experienc
-00012bd0: 6520 616e 2070 616c 656f 2d70 6f6c 7970  e an paleo-polyp
-00012be0: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
-00012bf0: 7420 6166 7465 7220 7468 6520 6469 7665  t after the dive
-00012c00: 7267 656e 6365 206f 6620 636f 7265 2065  rgence of core e
-00012c10: 7564 6963 6f74 732c 2077 6869 6368 2069  udicots, which i
-00012c20: 7320 6c69 6b65 6c79 2073 6861 7265 6420  s likely shared 
-00012c30: 6279 2061 6c6c 2052 616e 756e 6375 6c61  by all Ranuncula
-00012c40: 6c65 732e 0a0a 4669 7273 7420 6162 6f76  les...First abov
-00012c50: 6520 616c 6c2c 206c 6574 2773 2064 656c  e all, let's del
-00012c60: 696e 6561 7465 2074 6865 2077 686f 6c65  ineate the whole
-00012c70: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
-00012c80: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00012c90: 7374 7269 6275 7469 6f6e 2061 6e64 2068  stribution and h
-00012ca0: 6176 6520 6120 6261 7369 6320 6f62 7365  ave a basic obse
-00012cb0: 7276 6174 696f 6e20 666f 7220 706f 7465  rvation for pote
-00012cc0: 6e74 6961 6c6c 7920 636f 6e63 6569 7661  ntially conceiva
-00012cd0: 626c 6520 5747 4473 2c20 7573 696e 6720  ble WGDs, using 
-00012ce0: 7468 6520 636f 6d6d 616e 6420 6c69 6e65  the command line
-00012cf0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00012d00: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
-00012d10: 6f65 7275 6c65 610a 7767 6420 6b73 6420  oerulea.wgd ksd 
-00012d20: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
-00012d30: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
-00012d40: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00012d50: 610a 6060 600a 0a54 6865 2063 6f6e 7374  a.```..The const
-00012d60: 7275 6374 6564 2077 686f 6c65 2070 6172  ructed whole par
-00012d70: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
-00012d80: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
-00012d90: 6275 7469 6f6e 206f 6620 2a41 7175 696c  bution of *Aquil
-00012da0: 6567 6961 2063 6f65 7275 6c65 612a 2069  egia coerulea* i
-00012db0: 7320 6173 2062 656c 6f77 2c20 7765 2063  s as below, we c
-00012dc0: 616e 2073 6565 2074 6861 7420 7468 6572  an see that ther
-00012dd0: 6520 7365 656d 7320 746f 2062 6520 6120  e seems to be a 
-00012de0: 6875 6d70 2061 7420 2a4b 2a3c 7375 623e  hump at *K*<sub>
-00012df0: 533c 2f73 7562 3e20 3120 6275 7420 6e6f  S</sub> 1 but no
-00012e00: 7420 636c 6561 722e 0a0a 215b 5d28 6461  t clear...![](da
-00012e10: 7461 2f41 7175 696c 6567 6961 5f63 6f65  ta/Aquilegia_coe
-00012e20: 7275 6c65 612e 7473 762e 6b73 645f 7770  rulea.tsv.ksd_wp
-00012e30: 2e73 7667 290a 0a57 6520 7468 656e 2063  .svg)..We then c
-00012e40: 6f6e 7374 7275 6374 2074 6865 2061 6e63  onstruct the anc
-00012e50: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
-00012e60: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
-00012e70: 7469 6f6e 2075 7369 6e67 2074 6865 2063  tion using the c
-00012e80: 6f6d 6d61 6e64 206c 696e 6520 6265 6c6f  ommand line belo
-00012e90: 772e 0a0a 6060 600a 7767 6420 7379 6e20  w...```.wgd syn 
-00012ea0: 2d66 206d 524e 4120 2d61 204e 616d 6520  -f mRNA -a Name 
-00012eb0: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
-00012ec0: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
-00012ed0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00012ee0: 612e 6766 6633 202d 6b73 2077 6764 5f6b  a.gff3 -ks wgd_k
-00012ef0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
-00012f00: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
-00012f10: 0a60 6060 0a0a 4173 2073 686f 776e 2062  .```..As shown b
-00012f20: 656c 6f77 2c20 7468 6572 6520 6172 6520  elow, there are 
-00012f30: 736f 6d65 2072 6574 6169 6e65 6420 616e  some retained an
-00012f40: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
-00012f50: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00012f60: 6265 7477 6565 6e20 3120 616e 6420 322c  between 1 and 2,
-00012f70: 2077 6869 6368 2073 6565 6d73 2074 6f20   which seems to 
-00012f80: 7375 6767 6573 7420 6120 5747 4420 6576  suggest a WGD ev
-00012f90: 656e 742e 0a0a 215b 5d28 6461 7461 2f41  ent...![](data/A
-00012fa0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00012fb0: 612e 7473 762e 6b73 645f 7770 5f61 702e  a.tsv.ksd_wp_ap.
-00012fc0: 7376 6729 0a0a 5468 6520 6173 736f 6369  svg)..The associ
-00012fd0: 6174 6564 2060 6475 7053 7461 636b 6020  ated `dupStack` 
-00012fe0: 706c 6f74 2073 686f 7773 2074 6861 7420  plot shows that 
-00012ff0: 7468 6572 6520 6172 6520 6e75 6d65 726f  there are numero
-00013000: 7573 2064 7570 6c69 6361 7465 6420 7365  us duplicated se
-00013010: 676d 656e 7473 2061 6372 6f73 7320 6d6f  gments across mo
-00013020: 7374 206f 6620 7468 6520 6368 726f 6d6f  st of the chromo
-00013030: 736f 6d65 732e 0a0a 215b 5d28 6461 7461  somes...![](data
-00013040: 2f73 796e 5f72 6573 756c 7473 2f41 7175  /syn_results/Aqu
-00013050: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
-00013060: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013070: 6561 5f6d 756c 7469 706c 6963 6f6e 735f  ea_multiplicons_
-00013080: 6c65 7665 6c2e 706e 6729 0a0a 5765 2069  level.png)..We i
-00013090: 6d70 6c65 6d65 6e74 6564 2074 776f 2074  mplemented two t
-000130a0: 7970 6573 206f 6620 646f 7420 706c 6f74  ypes of dot plot
-000130b0: 7320 696e 206f 7866 6f72 6420 6772 6964  s in oxford grid
-000130c0: 3a20 6f6e 6520 696e 2074 6865 2075 6e69  : one in the uni
-000130d0: 7420 6f66 2062 6173 6573 2061 6e64 2074  t of bases and t
-000130e0: 6865 206f 7468 6572 2069 6e20 7468 6520  he other in the 
-000130f0: 756e 6974 206f 6620 6765 6e65 732c 2077  unit of genes, w
-00013100: 6869 6368 2063 616e 2062 6520 636f 6c6f  hich can be colo
-00013110: 7265 6420 6279 202a 4b2a 3c73 7562 3e53  red by *K*<sub>S
-00013120: 3c2f 7375 623e 2076 616c 7565 7320 6769  </sub> values gi
-00013130: 7665 6e20 2a4b 2a3c 7375 623e 533c 2f73  ven *K*<sub>S</s
-00013140: 7562 3e20 6461 7461 2e0a 0a21 5b5d 2864  ub> data...![](d
-00013150: 6174 612f 7379 6e5f 7265 7375 6c74 732f  ata/syn_results/
-00013160: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013170: 6561 2d76 732d 4171 7569 6c65 6769 615f  ea-vs-Aquilegia_
-00013180: 636f 6572 756c 6561 5f4b 732e 646f 745f  coerulea_Ks.dot_
-00013190: 756e 6974 5f67 656e 652e 706e 6729 0a0a  unit_gene.png)..
-000131a0: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
-000131b0: 7468 6520 646f 7420 706c 6f74 2069 6e20  the dot plot in 
-000131c0: 7468 6520 756e 6974 206f 6620 6765 6e65  the unit of gene
-000131d0: 7320 7072 6573 656e 7473 206e 756d 6572  s presents numer
-000131e0: 6f75 7320 6465 6e73 656c 7920 6167 6772  ous densely aggr
-000131f0: 6567 6174 6564 2028 6c69 6e65 2d6c 696b  egated (line-lik
-00013200: 6529 2061 6e63 686f 7220 706f 696e 7473  e) anchor points
-00013210: 2061 7420 6d6f 7374 206f 6620 7468 6520   at most of the 
-00013220: 6368 726f 6d6f 736f 6d65 7320 7769 7468  chromosomes with
-00013230: 2063 6f6e 7369 7374 656e 7420 2a4b 2a3c   consistent *K*<
-00013240: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-00013250: 6265 7477 6565 6e20 3120 616e 6420 322e  between 1 and 2.
-00013260: 2054 6865 2064 6f74 2070 6c6f 7420 696e   The dot plot in
-00013270: 2074 6865 2075 6e69 7420 6f66 2062 6173   the unit of bas
-00013280: 6573 2073 686f 7773 2074 6865 2073 616d  es shows the sam
-00013290: 6520 7061 7474 6572 6e2c 2061 7320 6d61  e pattern, as ma
-000132a0: 6e69 6665 7374 6564 2062 656c 6f77 2e0a  nifested below..
-000132b0: 0a21 5b5d 2864 6174 612f 7379 6e5f 7265  .![](data/syn_re
-000132c0: 7375 6c74 732f 4171 7569 6c65 6769 615f  sults/Aquilegia_
-000132d0: 636f 6572 756c 6561 2d76 732d 4171 7569  coerulea-vs-Aqui
-000132e0: 6c65 6769 615f 636f 6572 756c 6561 5f4b  legia_coerulea_K
-000132f0: 732e 646f 742e 706e 6729 0a0a 5468 6520  s.dot.png)..The 
-00013300: 646f 7420 706c 6f74 7320 7769 7468 6f75  dot plots withou
-00013310: 7420 2a4b 2a3c 7375 623e 533c 2f73 7562  t *K*<sub>S</sub
-00013320: 3e20 616e 6e6f 7461 7469 6f6e 2077 696c  > annotation wil
-00013330: 6c20 616c 736f 2062 6520 6175 746f 6d61  l also be automa
-00013340: 7465 6c79 2070 726f 6475 6365 642c 2061  tely produced, a
-00013350: 7320 7368 6f77 6e20 6265 6c6f 772e 0a0a  s shown below...
-00013360: 215b 5d28 6461 7461 2f73 796e 5f72 6573  ![](data/syn_res
-00013370: 756c 7473 2f41 7175 696c 6567 6961 5f63  ults/Aquilegia_c
-00013380: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
-00013390: 6567 6961 5f63 6f65 7275 6c65 612e 646f  egia_coerulea.do
-000133a0: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
-000133b0: 0a0a 215b 5d28 6461 7461 2f73 796e 5f72  ..![](data/syn_r
-000133c0: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
-000133d0: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
-000133e0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-000133f0: 646f 742e 706e 6729 0a0a 4e6f 7465 2074  dot.png)..Note t
-00013400: 6861 7420 7468 6520 6f70 6163 6974 7920  hat the opacity 
-00013410: 6f66 2061 6e63 686f 7220 646f 7473 2061  of anchor dots a
-00013420: 6e64 2061 6c6c 2068 6f6d 6f6c 6f67 2064  nd all homolog d
-00013430: 6f74 7320 6361 6e20 6265 2073 6574 2062  ots can be set b
-00013440: 7920 7468 6520 6f70 7469 6f6e 2060 2d2d  y the option `--
-00013450: 6170 616c 7068 6160 2061 6e64 2060 2d2d  apalpha` and `--
-00013460: 686f 616c 7068 6160 2073 6570 6172 6174  hoalpha` separat
-00013470: 656c 792e 2049 6620 6f6e 6520 6a75 7374  ely. If one just
-00013480: 2077 616e 7473 2074 6f20 7365 6520 7468   wants to see th
-00013490: 6520 616e 6368 6f72 2064 6f74 732c 2073  e anchor dots, s
-000134a0: 6574 7469 6e67 2074 6865 2060 686f 616c  etting the `hoal
-000134b0: 7068 6160 2061 7320 3020 286f 7220 6f74  pha` as 0 (or ot
-000134c0: 6865 7220 6d69 6e75 7363 756c 6520 7661  her minuscule va
-000134d0: 6c75 6573 2920 7769 6c6c 2064 6f2e 2049  lues) will do. I
-000134e0: 6620 6f6e 6520 7761 6e74 7320 746f 2073  f one wants to s
-000134f0: 6565 2074 6865 2064 6973 7472 6962 7574  ee the distribut
-00013500: 696f 6e20 6f66 2077 686f 6c65 2064 6f74  ion of whole dot
-00013510: 7320 6265 7474 6572 2c20 7365 7474 696e  s better, settin
-00013520: 6720 7468 6520 6068 6f61 6c70 6861 6020  g the `hoalpha` 
-00013530: 6869 6768 6572 2028 616e 6420 6061 7061  higher (and `apa
-00013540: 6c70 6861 6020 6c6f 7765 7229 2077 696c  lpha` lower) wil
-00013550: 6c20 646f 2e20 5468 6520 6064 6f74 7369  l do. The `dotsi
-00013560: 7a65 6020 6f70 7469 6f6e 2063 616e 2062  ze` option can b
-00013570: 6520 6361 6c6c 6564 2074 6f20 6164 6a75  e called to adju
-00013580: 7374 2074 6865 2073 697a 6520 6f66 2064  st the size of d
-00013590: 6f74 732e 0a0a 4120 6675 7274 6865 7220  ots...A further 
-000135a0: 6173 736f 6369 6174 6564 2053 796e 6465  associated Synde
-000135b0: 7074 6820 706c 6f74 2073 686f 7773 2074  pth plot shows t
-000135c0: 6861 7420 7468 6572 6520 6172 6520 6d6f  hat there are mo
-000135d0: 7265 2074 6861 6e20 3530 2064 7570 6c69  re than 50 dupli
-000135e0: 6361 7465 6420 7365 676d 656e 7473 206c  cated segments l
-000135f0: 6f6e 6765 7220 7468 616e 2031 3030 3030  onger than 10000
-00013600: 2062 7020 616e 6420 3330 2067 656e 6573   bp and 30 genes
-00013610: 2028 736f 2061 7320 746f 2064 726f 7020   (so as to drop 
-00013620: 6672 6167 6d65 6e74 6172 7920 7365 676d  fragmentary segm
-00013630: 656e 7473 292c 2077 6869 6368 2064 6f6d  ents), which dom
-00013640: 696e 6174 6573 2074 6865 2077 686f 6c65  inates the whole
-00013650: 2063 6f6c 6c69 6e65 6172 2072 6174 696f   collinear ratio
-00013660: 2063 6174 6567 6f72 792e 0a0a 215b 5d28   category...![](
-00013670: 6461 7461 2f73 796e 5f72 6573 756c 7473  data/syn_results
-00013680: 2f53 796e 6465 7074 682e 7376 6729 0a0a  /Syndepth.svg)..
-00013690: 4d6f 7265 2065 7871 7569 7369 7465 2070  More exquisite p
-000136a0: 6c6f 7473 2069 6e63 6c75 6469 6e67 2062  lots including b
-000136b0: 6f74 6820 696e 7472 612d 7370 6563 6966  oth intra-specif
-000136c0: 6963 2061 6e64 2069 6e74 6572 2d73 7065  ic and inter-spe
-000136d0: 6369 6669 6320 636f 6d70 6172 6973 6f6e  cific comparison
-000136e0: 7320 7573 696e 6720 7468 6520 6f72 7468  s using the orth
-000136f0: 6f67 726f 7570 7320 2863 6f6d 706f 7365  ogroups (compose
-00013700: 6420 6f66 202a 4171 7569 6c65 6769 6120  d of *Aquilegia 
-00013710: 636f 6572 756c 6561 2a2c 202a 5072 6f74  coerulea*, *Prot
-00013720: 6561 2063 796e 6172 6f69 6465 732a 2c20  ea cynaroides*, 
-00013730: 2a41 636f 7275 7320 616d 6572 6963 616e  *Acorus american
-00013740: 7573 2a20 616e 6420 2a56 6974 6973 2076  us* and *Vitis v
-00013750: 696e 6966 6572 612a 2c20 7365 6520 6675  inifera*, see fu
-00013760: 7274 6865 7220 666f 7220 7468 6520 636f  rther for the co
-00013770: 6e74 6578 7429 2069 6e66 6572 7265 6420  ntext) inferred 
-00013780: 6865 7265 696e 6166 7465 7220 6361 6e20  hereinafter can 
-00013790: 6265 2061 6c73 6f20 7072 6f64 7563 6564  be also produced
-000137a0: 2075 7369 6e67 2060 7767 6420 7379 6e60   using `wgd syn`
-000137b0: 2e20 4e6f 7465 2074 6861 7420 6469 6666  . Note that diff
-000137c0: 6572 656e 7420 6765 6e6f 6d65 2061 7373  erent genome ass
-000137d0: 656d 626c 6965 7320 6d69 6768 7420 6861  emblies might ha
-000137e0: 7665 2064 6966 6665 7265 6e74 2066 6561  ve different fea
-000137f0: 7475 7265 7320 616e 6420 6174 7472 6962  tures and attrib
-00013800: 7574 6573 2077 6869 6368 2063 616e 2062  utes which can b
-00013810: 6520 6163 636f 6d6d 6f64 6174 6564 2076  e accommodated v
-00013820: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00013830: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
-00013840: 666f 6020 666f 7220 6561 6368 2067 656e  fo` for each gen
-00013850: 6f6d 6520 6173 7365 6d62 6c79 2077 686f  ome assembly who
-00013860: 7365 206f 7264 6572 206e 6565 6473 2074  se order needs t
-00013870: 6f20 666f 6c6c 6f77 2074 6865 206f 7264  o follow the ord
-00013880: 6572 206f 6620 6766 6633 2066 696c 6573  er of gff3 files
-00013890: 2c20 666f 7220 696e 7374 616e 6365 2027  , for instance '
-000138a0: 6d4e 5241 3b4e 616d 6527 2066 6f72 2060  mNRA;Name' for `
-000138b0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000138c0: 6561 2e67 6666 3360 2c20 276d 4e52 413b  ea.gff3`, 'mNRA;
-000138d0: 4944 2720 666f 7220 6050 726f 7465 615f  ID' for `Protea_
-000138e0: 6379 6e61 726f 6964 6573 2e67 6666 3360  cynaroides.gff3`
-000138f0: 2c20 276d 4e52 413b 4e61 6d65 2720 666f  , 'mNRA;Name' fo
-00013900: 7220 6041 636f 7275 735f 616d 6572 6963  r `Acorus_americ
-00013910: 616e 7573 2e67 6666 3360 2061 6e64 2027  anus.gff3` and '
-00013920: 6d4e 5241 3b4e 616d 6527 2066 6f72 2060  mNRA;Name' for `
-00013930: 5669 7469 735f 7669 6e69 6665 7261 2e67  Vitis_vinifera.g
-00013940: 6666 3360 2e0a 0a60 6060 0a77 6764 2073  ff3`...```.wgd s
-00013950: 796e 2077 6764 5f6f 7274 686f 2f4f 7274  yn wgd_ortho/Ort
-00013960: 686f 6772 6f75 7073 2e73 702e 7473 7620  hogroups.sp.tsv 
-00013970: 2d6b 7320 7767 645f 6f72 7468 6f5f 6b73  -ks wgd_ortho_ks
-00013980: 2f4f 7274 686f 6772 6f75 7073 2e73 702e  /Orthogroups.sp.
-00013990: 7473 762e 6b73 2e74 7376 2041 7175 696c  tsv.ks.tsv Aquil
-000139a0: 6567 6961 5f63 6f65 7275 6c65 612e 6766  egia_coerulea.gf
-000139b0: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
-000139c0: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
-000139d0: 6527 2050 726f 7465 615f 6379 6e61 726f  e' Protea_cynaro
-000139e0: 6964 6573 2e67 6666 3320 2d2d 6164 6469  ides.gff3 --addi
-000139f0: 7469 6f6e 616c 6766 6669 6e66 6f20 276d  tionalgffinfo 'm
-00013a00: 4e52 413b 4944 2720 4163 6f72 7573 5f61  NRA;ID' Acorus_a
-00013a10: 6d65 7269 6361 6e75 732e 6766 6633 202d  mericanus.gff3 -
-00013a20: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
-00013a30: 666f 2027 6d4e 5241 3b4e 616d 6527 2056  fo 'mNRA;Name' V
-00013a40: 6974 6973 5f76 696e 6966 6572 612e 6766  itis_vinifera.gf
-00013a50: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
-00013a60: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
-00013a70: 6527 202d 6f20 7767 645f 6f72 7468 6f5f  e' -o wgd_ortho_
-00013a80: 7379 6e0a 6060 600a 0a55 706f 6e20 7468  syn.```..Upon th
-00013a90: 6520 6163 7175 6973 6974 696f 6e20 6f66  e acquisition of
-00013aa0: 2074 6865 2063 6f6c 6c69 6e65 6172 2072   the collinear r
-00013ab0: 6573 756c 7473 2075 7369 6e67 2060 7767  esults using `wg
-00013ac0: 6420 7379 6e60 2c20 7468 6520 7361 6d65  d syn`, the same
-00013ad0: 2063 6f6c 6c69 6e65 6172 2070 6c6f 7473   collinear plots
-00013ae0: 2063 616e 2062 6520 616c 736f 2070 726f   can be also pro
-00013af0: 6475 6365 6420 6279 2060 7767 6420 7669  duced by `wgd vi
-00013b00: 7a60 2075 7369 6e67 2074 6865 2063 6f6d  z` using the com
-00013b10: 6d61 6e64 2062 656c 6f77 2e0a 6060 600a  mand below..```.
-00013b20: 7767 6420 7669 7a20 2d2d 706c 6f74 7379  wgd viz --plotsy
-00013b30: 6e20 2d73 6d20 7767 645f 6f72 7468 6f5f  n -sm wgd_ortho_
-00013b40: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
-00013b50: 7365 676d 656e 7473 2e74 7874 202d 6170  segments.txt -ap
-00013b60: 2077 6764 5f6f 7274 686f 5f73 796e 2f69   wgd_ortho_syn/i
-00013b70: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00013b80: 7270 6f69 6e74 732e 7478 7420 2d6d 7420  rpoints.txt -mt 
-00013b90: 7767 645f 6f72 7468 6f5f 7379 6e2f 6961  wgd_ortho_syn/ia
-00013ba0: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
-00013bb0: 6c69 636f 6e73 2e74 7874 202d 6774 2077  licons.txt -gt w
-00013bc0: 6764 5f6f 7274 686f 5f73 796e 2f67 656e  gd_ortho_syn/gen
-00013bd0: 652d 7461 626c 652e 6373 7620 2d64 2077  e-table.csv -d w
-00013be0: 6764 5f6f 7274 686f 5f6b 732f 4f72 7468  gd_ortho_ks/Orth
-00013bf0: 6f67 726f 7570 732e 7370 2e74 7376 2e6b  ogroups.sp.tsv.k
-00013c00: 732e 7473 7620 2d6f 2077 6764 5f6f 7274  s.tsv -o wgd_ort
-00013c10: 686f 5f76 697a 0a60 6060 0a0a 215b 5d28  ho_viz.```..![](
-00013c20: 6461 7461 2f6f 7274 686f 5f73 796e 5f72  data/ortho_syn_r
-00013c30: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
-00013c40: 5f63 6f65 7275 6c65 615f 5669 7469 735f  _coerulea_Vitis_
-00013c50: 7669 6e69 6665 7261 5f6d 756c 7469 706c  vinifera_multipl
-00013c60: 6963 6f6e 735f 6c65 7665 6c2e 706e 6729  icons_level.png)
-00013c70: 0a54 6865 2061 626f 7665 2060 6475 7053  .The above `dupS
-00013c80: 7461 636b 6020 706c 6f74 2073 686f 7773  tack` plot shows
-00013c90: 2074 6865 2064 6973 7472 6962 7574 696f   the distributio
-00013ca0: 6e20 6f66 2064 7570 6c69 6361 7465 6420  n of duplicated 
-00013cb0: 7365 676d 656e 7473 206f 6620 2a41 7175  segments of *Aqu
-00013cc0: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
-00013cd0: 2063 6f6d 7061 7265 6420 746f 2069 7473   compared to its
-00013ce0: 656c 6620 2869 6e20 6772 6565 6e29 2061  elf (in green) a
-00013cf0: 6e64 2063 6f6d 7061 7265 6420 746f 202a  nd compared to *
-00013d00: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
-00013d10: 2869 6e20 626c 7565 2920 6f76 6572 2074  (in blue) over t
-00013d20: 6865 2063 6872 6f6d 6f73 6f6d 6573 206f  he chromosomes o
-00013d30: 6620 2a41 2e20 636f 6572 756c 6561 2a2e  f *A. coerulea*.
-00013d40: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
-00013d50: 7379 6e5f 7265 7375 6c74 732f 4f76 6572  syn_results/Over
-00013d60: 616c 6c73 7065 6369 6573 5f4b 732e 646f  allspecies_Ks.do
-00013d70: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
-00013d80: 0a54 6865 2061 626f 7665 202a 4b2a 3c73  .The above *K*<s
-00013d90: 7562 3e53 3c2f 7375 623e 2064 6f74 706c  ub>S</sub> dotpl
-00013da0: 6f74 2069 6e20 756e 6974 206f 6620 6765  ot in unit of ge
-00013db0: 6e65 2073 686f 7773 2074 6865 206f 7665  ne shows the ove
-00013dc0: 7261 6c6c 2064 6973 7472 6962 7574 696f  rall distributio
-00013dd0: 6e20 6f66 2063 6f6c 6c69 6e65 6172 6974  n of collinearit
-00013de0: 7920 6163 6f72 7373 2074 6865 2066 6f75  y acorss the fou
-00013df0: 7220 7370 6563 6965 7320 696e 766f 6c76  r species involv
-00013e00: 6564 2e0a 215b 5d28 6461 7461 2f6f 7274  ed..![](data/ort
-00013e10: 686f 5f73 796e 5f72 6573 756c 7473 2f4f  ho_syn_results/O
-00013e20: 7665 7261 6c6c 7370 6563 6965 732e 646f  verallspecies.do
-00013e30: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
-00013e40: 0a54 6865 2061 626f 7665 2064 6f74 706c  .The above dotpl
-00013e50: 6f74 2069 7320 7769 7468 6f75 7420 7468  ot is without th
-00013e60: 6520 616e 6e6f 7461 7469 6f6e 206f 6620  e annotation of 
-00013e70: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013e80: 6167 6573 2063 6f6d 7061 7265 6420 746f  ages compared to
-00013e90: 2074 6865 206c 6173 7420 6f6e 652e 0a21   the last one..!
-00013ea0: 5b5d 2864 6174 612f 6f72 7468 6f5f 7379  [](data/ortho_sy
-00013eb0: 6e5f 7265 7375 6c74 732f 5379 6e64 6570  n_results/Syndep
-00013ec0: 7468 2e73 7667 290a 5468 6520 6162 6f76  th.svg).The abov
-00013ed0: 6520 5379 6e64 6570 7468 2070 6c6f 7420  e Syndepth plot 
-00013ee0: 7368 6f77 7320 7468 6520 636f 6c6c 696e  shows the collin
-00013ef0: 6561 7220 7261 7469 6f20 6163 6f72 7373  ear ratio acorss
-00013f00: 2061 6c6c 2073 7065 6369 6573 2070 6169   all species pai
-00013f10: 7273 2028 696e 7472 612d 7370 6563 6966  rs (intra-specif
-00013f20: 6963 2063 6f6d 7061 7269 736f 6e20 696e  ic comparison in
-00013f30: 2067 7265 656e 2077 6869 6c65 2069 6e74   green while int
-00013f40: 6572 2d73 7065 6369 6669 6320 636f 6d70  er-specific comp
-00013f50: 6172 6973 6f6e 2069 6e20 626c 7565 292e  arison in blue).
-00013f60: 0a0a 5765 2063 616e 2066 6974 2061 6e20  ..We can fit an 
-00013f70: 454c 4d4d 206d 6978 7475 7265 206d 6f64  ELMM mixture mod
-00013f80: 656c 2075 706f 6e20 7468 6520 7768 6f6c  el upon the whol
-00013f90: 6520 7061 7261 6e6f 6d65 202a 4b2a 3c73  e paranome *K*<s
-00013fa0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-00013fb0: 6973 7472 6962 7574 696f 6e20 746f 2073  istribution to s
-00013fc0: 6565 206d 6f72 6520 6163 6375 7261 7465  ee more accurate
-00013fd0: 6c79 2074 6865 2073 6967 6e69 6669 6361  ly the significa
-00013fe0: 6e63 6520 616e 6420 6c6f 6361 7469 6f6e  nce and location
-00013ff0: 206f 6620 706f 7465 6e74 6961 6c20 5747   of potential WG
-00014000: 4473 2c20 7573 696e 6720 7468 6520 636f  Ds, using the co
-00014010: 6d6d 616e 6420 6c69 6e65 2062 656c 6f77  mmand line below
-00014020: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
-00014030: 6420 7767 645f 6b73 642f 4171 7569 6c65  d wgd_ksd/Aquile
-00014040: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
-00014050: 2e6b 732e 7473 760a 6060 600a 0a54 6865  .ks.tsv.```..The
-00014060: 2072 6573 756c 7420 6f66 2045 4c4d 4d20   result of ELMM 
-00014070: 6d69 7874 7572 6520 6d6f 6465 6c20 636c  mixture model cl
-00014080: 7573 7465 7269 6e67 2073 686f 7773 2074  ustering shows t
-00014090: 6861 7420 7468 6572 6520 6973 2061 206c  hat there is a l
-000140a0: 696b 656c 7920 5747 4420 636f 6d70 6f6e  ikely WGD compon
-000140b0: 656e 7420 6174 202a 4b2a 3c73 7562 3e53  ent at *K*<sub>S
-000140c0: 3c2f 7375 623e 2031 2e31 392e 0a0a 215b  </sub> 1.19...![
-000140d0: 5d28 6461 7461 2f65 6c6d 6d5f 4171 7569  ](data/elmm_Aqui
-000140e0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
-000140f0: 7376 2e6b 732e 7473 765f 6265 7374 5f6d  sv.ks.tsv_best_m
-00014100: 6f64 656c 735f 7765 6967 6874 6564 2e73  odels_weighted.s
-00014110: 7667 290a 0a4c 6574 2773 2064 6f20 6120  vg)..Let's do a 
-00014120: 6d69 7874 7572 6520 6d6f 6465 6c20 636c  mixture model cl
-00014130: 7573 7465 7269 6e67 2066 6f72 2061 6e63  ustering for anc
-00014140: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
-00014150: 7562 3e20 746f 6f2c 2075 7369 6e67 2074  ub> too, using t
-00014160: 6865 2063 6f6d 6d61 6e64 206c 696e 6520  he command line 
-00014170: 6265 6c6f 772e 204e 6f74 6520 7468 6174  below. Note that
-00014180: 2074 6869 7320 7374 6570 2077 696c 6c20   this step will 
-00014190: 6175 746f 6d61 7465 6c79 2063 616c 6c20  automately call 
-000141a0: 7468 6520 7365 676d 656e 7420 2a4b 2a3c  the segment *K*<
-000141b0: 7375 623e 533c 2f73 7562 3e20 636c 7573  sub>S</sub> clus
-000141c0: 7465 7269 6e67 2061 6e61 6c79 7369 7320  tering analysis 
-000141d0: 746f 6f2e 0a0a 6060 600a 7767 6420 7065  too...```.wgd pe
-000141e0: 616b 2077 6764 5f6b 7364 2f41 7175 696c  ak wgd_ksd/Aquil
-000141f0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00014200: 762e 6b73 2e74 7376 202d 2d61 6e63 686f  v.ks.tsv --ancho
-00014210: 7270 6f69 6e74 7320 7767 645f 7379 6e2f  rpoints wgd_syn/
-00014220: 6961 6468 6f72 652d 6f75 742f 616e 6368  iadhore-out/anch
-00014230: 6f72 706f 696e 7473 2e74 7874 202d 2d73  orpoints.txt --s
-00014240: 6567 6d65 6e74 7320 7767 645f 7379 6e2f  egments wgd_syn/
-00014250: 6961 6468 6f72 652d 6f75 742f 7365 676d  iadhore-out/segm
-00014260: 656e 7473 2e74 7874 202d 2d6c 6973 7465  ents.txt --liste
-00014270: 6c65 6d65 6e74 7320 7767 645f 7379 6e2f  lements wgd_syn/
-00014280: 6961 6468 6f72 652d 6f75 742f 6c69 7374  iadhore-out/list
-00014290: 5f65 6c65 6d65 6e74 732e 7478 7420 2d2d  _elements.txt --
-000142a0: 6d75 6c74 6970 6c69 636f 6e70 6169 7273  multipliconpairs
-000142b0: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
-000142c0: 2d6f 7574 2f6d 756c 7469 706c 6963 6f6e  -out/multiplicon
-000142d0: 5f70 6169 7273 2e74 7874 2028 2d2d 7765  _pairs.txt (--we
-000142e0: 6967 6874 6564 290a 6060 600a 0a54 6865  ighted).```..The
-000142f0: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
-00014300: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
-00014310: 7269 6275 7469 6f6e 2061 6c73 6f20 6861  ribution also ha
-00014320: 7320 6120 6c69 6b65 6c79 2057 4744 2063  s a likely WGD c
-00014330: 6f6d 706f 6e65 6e74 2077 6974 6820 6d6f  omponent with mo
-00014340: 6465 2031 2e32 382e 0a0a 215b 5d28 6461  de 1.28...![](da
-00014350: 7461 2f4f 7269 6769 6e61 6c5f 416e 6368  ta/Original_Anch
-00014360: 6f72 4b73 5f47 4d4d 5f43 6f6d 706f 6e65  orKs_GMM_Compone
-00014370: 6e74 335f 6e6f 6465 5f61 7665 7261 6765  nt3_node_average
-00014380: 645f 4c6f 676e 6f72 6d61 6c2e 7376 6729  d_Lognormal.svg)
-00014390: 0a0a 4e6f 7720 7468 6174 2077 6520 6861  ..Now that we ha
-000143a0: 7665 2073 6565 6e20 7468 6520 6576 6964  ve seen the evid
-000143b0: 656e 6365 206f 6620 6e75 6d65 726f 7573  ence of numerous
-000143c0: 2064 7570 6c69 6361 7465 6420 7365 676d   duplicated segm
-000143d0: 656e 7473 2061 6e64 2074 6865 2061 6767  ents and the agg
-000143e0: 7265 6761 7469 6f6e 206f 6620 6475 706c  regation of dupl
-000143f0: 6963 6174 6573 2061 6765 2061 7420 2a4b  icates age at *K
-00014400: 2a3c 7375 623e 533c 2f73 7562 3e20 312e  *<sub>S</sub> 1.
-00014410: 3238 206f 7220 312e 3139 2066 6f72 2061  28 or 1.19 for a
-00014420: 6e63 686f 7220 7061 6972 7320 616e 6420  nchor pairs and 
-00014430: 6e6f 6e2d 616e 6368 6f72 2070 6169 7273  non-anchor pairs
-00014440: 2074 6872 6f75 6768 6f75 7420 7468 6520   throughout the 
-00014450: 7768 6f6c 6520 6765 6e6f 6d65 2e20 5765  whole genome. We
-00014460: 2063 616e 2063 6c61 696d 2077 6974 6820   can claim with 
-00014470: 736f 6d65 2063 6f6e 6669 6465 6e63 6520  some confidence 
-00014480: 7468 6174 202a 4171 7569 6c65 6769 6120  that *Aquilegia 
-00014490: 636f 6572 756c 6561 2a20 6d69 6768 7420  coerulea* might 
-000144a0: 6861 7665 2065 7870 6572 6965 6e63 6564  have experienced
-000144b0: 2061 2070 616c 656f 2d70 6f6c 7970 6c6f   a paleo-polyplo
-000144c0: 6964 697a 6174 696f 6e20 6576 656e 742e  idization event.
-000144d0: 204e 6578 742c 204c 6574 2773 2068 6176   Next, Let's hav
-000144e0: 6520 6120 6675 7274 6865 7220 6c6f 6f6b  e a further look
-000144f0: 2061 626f 7574 2069 7473 2070 6879 6c6f   about its phylo
-00014500: 6765 6e65 7469 6320 6c6f 6361 7469 6f6e  genetic location
-00014510: 2e20 5765 206b 6e6f 7720 7468 6174 2074  . We know that t
-00014520: 6865 7265 2061 7265 2075 6e63 6572 7461  here are uncerta
-00014530: 696e 7469 6573 2061 626f 7574 2077 6865  inties about whe
-00014540: 7468 6572 2074 6869 7320 7075 7461 7469  ther this putati
-00014550: 7665 2070 616c 656f 2d70 6f6c 7970 6c6f  ve paleo-polyplo
-00014560: 6964 697a 6174 696f 6e20 6576 656e 7420  idization event 
-00014570: 6973 2073 6861 7265 6420 7769 7468 2061  is shared with a
-00014580: 6c6c 2065 7564 6963 6f74 7320 6f72 206e  ll eudicots or n
-00014590: 6f74 2e20 5765 2063 616e 2063 686f 6f73  ot. We can choos
-000145a0: 6520 736f 6d65 206f 7468 6572 2065 7564  e some other eud
-000145b0: 6963 6f74 2067 656e 6f6d 6573 2074 6f20  icot genomes to 
-000145c0: 7365 6520 7468 6520 6f72 6465 7269 6e67  see the ordering
-000145d0: 206f 6620 7370 6563 6961 7469 6f6e 2061   of speciation a
-000145e0: 6e64 2070 6f6c 7970 6c6f 6964 697a 6174  nd polyploidizat
-000145f0: 696f 6e20 6576 656e 7473 2e20 4865 7265  ion events. Here
-00014600: 2077 6520 6368 6f6f 7365 202a 5669 7469   we choose *Viti
-00014610: 7320 7669 6e69 6665 7261 2a2c 202a 5072  s vinifera*, *Pr
-00014620: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
-00014630: 2061 6e64 202a 4163 6f72 7573 2061 6d65   and *Acorus ame
-00014640: 7269 6361 6e75 732a 2069 6e20 7468 6520  ricanus* in the 
-00014650: 666f 6c6c 6f77 696e 6720 2a4b 2a3c 7375  following *K*<su
-00014660: 623e 533c 2f73 7562 3e20 616e 616c 7973  b>S</sub> analys
-00014670: 6973 2e20 4669 7273 742c 2077 6520 6275  is. First, we bu
-00014680: 696c 7420 6120 676c 6f62 616c 204d 5242  ilt a global MRB
-00014690: 4820 6661 6d69 6c79 2075 7369 6e67 2074  H family using t
-000146a0: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-000146b0: 2e0a 0a60 6060 0a77 6764 2064 6d64 202d  ...```.wgd dmd -
-000146c0: 2d67 6c6f 6261 6c6d 7262 6820 4171 7569  -globalmrbh Aqui
-000146d0: 6c65 6769 615f 636f 6572 756c 6561 2050  legia_coerulea P
-000146e0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-000146f0: 2041 636f 7275 735f 616d 6572 6963 616e   Acorus_american
-00014700: 7573 2056 6974 6973 5f76 696e 6966 6572  us Vitis_vinifer
-00014710: 6120 2d6f 2077 6764 5f67 6c6f 6261 6c6d  a -o wgd_globalm
-00014720: 7262 680a 6060 600a 0a49 6e20 7468 6520  rbh.```..In the 
-00014730: 676c 6f62 616c 204d 5242 4820 6661 6d69  global MRBH fami
-00014740: 6c79 2c20 6576 6572 7920 7061 6972 206f  ly, every pair o
-00014750: 6620 6f72 7468 6f6c 6f67 6f75 7320 6765  f orthologous ge
-00014760: 6e65 7320 6973 2074 6865 2072 6563 6970  nes is the recip
-00014770: 726f 6361 6c20 6265 7374 2068 6974 2c20  rocal best hit, 
-00014780: 7375 6767 6573 7469 6e67 2074 7275 6520  suggesting true 
-00014790: 6f72 7468 6f6c 6f67 6f75 7320 7265 6c61  orthologous rela
-000147a0: 7469 6f6e 7368 6970 732e 2057 6520 776f  tionships. We wo
-000147b0: 756c 6420 7573 6520 7468 6520 2a4b 2a3c  uld use the *K*<
-000147c0: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
-000147d0: 6573 2061 7373 6f63 6961 7465 6420 7769  es associated wi
-000147e0: 7468 2074 6865 7365 206f 7274 686f 6c6f  th these ortholo
-000147f0: 676f 7573 2070 6169 7273 2074 6f20 6465  gous pairs to de
-00014800: 6c69 6d69 7420 7468 6520 6469 7665 7267  limit the diverg
-00014810: 656e 6365 202a 4b2a 3c73 7562 3e53 3c2f  ence *K*<sub>S</
-00014820: 7375 623e 2070 6561 6b2e 2054 6f67 6574  sub> peak. Toget
-00014830: 6865 7220 7769 7468 2074 6865 2077 686f  her with the who
-00014840: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
-00014850: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00014860: 7269 6275 7469 6f6e 2c20 7765 2063 6f6e  ribution, we con
-00014870: 6475 6374 2074 6865 2072 6174 6520 636f  duct the rate co
-00014880: 7272 6563 7469 6f6e 2075 7369 6e67 2074  rrection using t
-00014890: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-000148a0: 2e0a 0a21 2153 696e 6365 2060 7767 6460  ...!!Since `wgd`
-000148b0: 2076 6572 7369 6f6e 2032 2e30 2e32 342c   version 2.0.24,
-000148c0: 2077 6520 7265 7772 6f74 6520 6120 636c   we rewrote a cl
-000148d0: 6561 6e65 7220 616e 6420 7175 6963 6b65  eaner and quicke
-000148e0: 7220 7761 7920 6f66 2064 6f69 6e67 2073  r way of doing s
-000148f0: 7562 7374 6974 7574 696f 6e20 7261 7465  ubstitution rate
-00014900: 2063 6f72 7265 6374 696f 6e2e 2049 7427   correction. It'
-00014910: 7320 6e6f 7420 7265 7175 6972 6564 2074  s not required t
-00014920: 6f20 7479 7065 2069 6e20 616e 7920 7370  o type in any sp
-00014930: 6569 6365 7320 7061 6972 2061 6e64 2061  eices pair and a
-00014940: 2073 6572 6965 7320 6f66 202a 4b2a 3c73   series of *K*<s
-00014950: 7562 3e53 3c2f 7375 623e 2070 6c6f 7473  ub>S</sub> plots
-00014960: 2077 696c 6c20 6265 2070 726f 6475 6365   will be produce
-00014970: 642e 2054 6865 2072 6571 7569 7265 6420  d. The required 
-00014980: 6669 6c65 7320 6172 6520 6f72 7468 6f6c  files are orthol
-00014990: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-000149a0: 2f73 7562 3e20 6461 7461 6669 6c65 2c20  /sub> datafile, 
-000149b0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-000149c0: 7562 3e53 3c2f 7375 623e 2064 6174 6166  ub>S</sub> dataf
-000149d0: 696c 652c 2061 2073 7065 6369 6573 2074  ile, a species t
-000149e0: 7265 6520 616e 6420 6120 666f 6361 6c20  ree and a focal 
-000149f0: 7370 6563 6965 7320 2874 6865 206f 6e65  species (the one
-00014a00: 2069 6e70 7574 7465 6420 7769 7468 2070   inputted with p
-00014a10: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
-00014a20: 623e 533c 2f73 7562 3e20 6461 7461 292e  b>S</sub> data).
-00014a30: 2055 7365 7273 2063 616e 2063 686f 6f73   Users can choos
-00014a40: 6520 746f 2061 6464 206f 6e65 206d 6f72  e to add one mor
-00014a50: 6520 6c61 7965 7220 6f66 2045 4c4d 4d20  e layer of ELMM 
-00014a60: 616e 616c 7973 6973 206f 6e20 7061 7261  analysis on para
-00014a70: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00014a80: 3c2f 7375 623e 2076 616c 7565 7320 616e  </sub> values an
-00014a90: 642f 6f72 2047 4d4d 2061 6e61 6c79 7369  d/or GMM analysi
-00014aa0: 7320 6f6e 2061 6e63 686f 7220 2a4b 2a3c  s on anchor *K*<
-00014ab0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00014ac0: 7269 6275 7469 6f6e 2e20 5468 6520 6f72  ribution. The or
-00014ad0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-00014ae0: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
-00014af0: 6275 7469 6f6e 2063 616e 2062 6520 6361  bution can be ca
-00014b00: 6c63 756c 6174 6564 2075 7369 6e67 2074  lculated using t
-00014b10: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-00014b20: 2e0a 0a60 6060 0a77 6764 206b 7364 2077  ...```.wgd ksd w
-00014b30: 6764 5f67 6c6f 6261 6c6d 7262 682f 676c  gd_globalmrbh/gl
-00014b40: 6f62 616c 5f4d 5242 482e 7473 7620 4171  obal_MRBH.tsv Aq
-00014b50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00014b60: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
-00014b70: 6573 2041 636f 7275 735f 616d 6572 6963  es Acorus_americ
-00014b80: 616e 7573 2056 6974 6973 5f76 696e 6966  anus Vitis_vinif
-00014b90: 6572 6120 2d6f 2077 6764 5f67 6c6f 6261  era -o wgd_globa
-00014ba0: 6c6d 7262 685f 6b73 0a60 6060 0a0a 5769  lmrbh_ks.```..Wi
-00014bb0: 7468 2074 6865 2063 616c 6375 6c61 7465  th the calculate
-00014bc0: 6420 6f72 7468 6f6c 6f67 6f75 7320 2a4b  d orthologous *K
-00014bd0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00014be0: 7374 7269 6275 7469 6f6e 2c20 7765 2063  stribution, we c
-00014bf0: 616e 2075 7365 2074 6865 2063 6f6d 6d61  an use the comma
-00014c00: 6e64 2062 656c 6f77 2074 6f20 636f 6e64  nd below to cond
-00014c10: 7563 7420 7468 6520 7261 7465 2063 6f72  uct the rate cor
-00014c20: 7265 6374 696f 6e20 616e 642f 6f72 206d  rection and/or m
-00014c30: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-00014c40: 616e 616c 7973 6973 2e0a 0a60 6060 0a77  analysis...```.w
-00014c50: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
-00014c60: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
-00014c70: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
-00014c80: 7376 202d 6661 2041 7175 696c 6567 6961  sv -fa Aquilegia
-00014c90: 5f63 6f65 7275 6c65 6120 2d65 706b 2077  _coerulea -epk w
-00014ca0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00014cb0: 5f63 6f65 7275 6c65 612e 6b73 2e74 7376  _coerulea.ks.tsv
-00014cc0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
-00014cd0: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00014ce0: 6f69 6e74 732e 7478 7420 2d73 7020 7370  oints.txt -sp sp
-00014cf0: 6563 6965 7374 7265 652e 6e77 202d 6f20  eciestree.nw -o 
-00014d00: 7767 645f 7669 7a5f 6d69 7865 645f 4b73  wgd_viz_mixed_Ks
-00014d10: 202d 2d70 6c6f 7465 6c6d 6d20 2d2d 706c   --plotelmm --pl
-00014d20: 6f74 6170 676d 6d20 2d2d 7265 7765 6967  otapgmm --reweig
-00014d30: 6874 0a60 6060 0a0a 6f72 2075 7369 6e67  ht.```..or using
-00014d40: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-00014d50: 6f77 2c20 7768 6963 6820 636f 6d62 696e  ow, which combin
-00014d60: 6573 2074 6865 2074 776f 2073 7465 7073  es the two steps
-00014d70: 2061 626f 7665 2069 6e20 6f6e 652e 204e   above in one. N
-00014d80: 6f74 6520 7468 6174 2077 6520 7375 6767  ote that we sugg
-00014d90: 6573 7420 6f66 2074 616b 696e 6720 7477  est of taking tw
-00014da0: 6f20 7365 7061 7261 7465 2073 7465 7073  o separate steps
-00014db0: 2069 6e20 7768 6963 6820 6077 6764 206b   in which `wgd k
-00014dc0: 7364 6020 756e 6465 7274 616b 6573 2074  sd` undertakes t
-00014dd0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
-00014de0: 6620 6f72 7468 6f6c 6f67 6f75 7320 2a4b  f orthologous *K
-00014df0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00014e00: 7374 7269 6275 7469 6f6e 2077 6869 6c65  stribution while
-00014e10: 2060 7767 6420 7669 7a60 2063 6172 7269   `wgd viz` carri
-00014e20: 6573 206f 7574 2074 6865 2072 6174 6520  es out the rate 
-00014e30: 636f 7272 6563 7469 6f6e 2061 6e64 2047  correction and G
-00014e40: 4d4d 2061 6e61 6c79 7369 7320 7375 6368  MM analysis such
-00014e50: 2074 6861 7420 6974 2773 2065 6173 6965   that it's easie
-00014e60: 7220 746f 2064 6562 7567 2e0a 6060 600a  r to debug..```.
-00014e70: 7767 6420 6b73 6420 7767 645f 676c 6f62  wgd ksd wgd_glob
-00014e80: 616c 6d72 6268 2f67 6c6f 6261 6c5f 4d52  almrbh/global_MR
-00014e90: 4248 2e74 7376 2041 7175 696c 6567 6961  BH.tsv Aquilegia
-00014ea0: 5f63 6f65 7275 6c65 6120 5072 6f74 6561  _coerulea Protea
-00014eb0: 5f63 796e 6172 6f69 6465 7320 4163 6f72  _cynaroides Acor
-00014ec0: 7573 5f61 6d65 7269 6361 6e75 7320 5669  us_americanus Vi
-00014ed0: 7469 735f 7669 6e69 6665 7261 202d 2d65  tis_vinifera --e
-00014ee0: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
-00014ef0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00014f00: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00014f10: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-00014f20: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-00014f30: 6874 202d 6f20 7767 645f 676c 6f62 616c  ht -o wgd_global
-00014f40: 6d72 6268 5f6b 735f 7261 7465 5f63 6f72  mrbh_ks_rate_cor
-00014f50: 7265 6374 696f 6e20 2d66 6120 4171 7569  rection -fa Aqui
-00014f60: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
-00014f70: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
-00014f80: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
-00014f90: 6e74 732e 7478 7420 2d2d 706c 6f74 656c  nts.txt --plotel
-00014fa0: 6d6d 202d 2d70 6c6f 7461 7067 6d6d 0a60  mm --plotapgmm.`
-00014fb0: 6060 0a0a 5468 6520 6669 6c65 2060 7370  ``..The file `sp
-00014fc0: 6563 6965 7374 7265 652e 6e77 6020 6973  eciestree.nw` is
-00014fd0: 2074 6865 2074 6578 7420 6669 6c65 206f   the text file o
-00014fe0: 6620 7370 6563 6965 7320 7472 6565 2069  f species tree i
-00014ff0: 6e20 6e65 7769 636b 2074 6861 7420 7261  n newick that ra
-00015000: 7465 2063 6f72 7265 6374 696f 6e20 776f  te correction wo
-00015010: 756c 6420 6265 2063 6f6e 6475 6374 6564  uld be conducted
-00015020: 206f 6e2e 2049 7473 2063 6f6e 7465 6e74   on. Its content
-00015030: 2069 7320 6173 2062 656c 6f77 2e20 5573   is as below. Us
-00015040: 6572 7320 6361 6e20 6f70 7469 6f6e 616c  ers can optional
-00015050: 6c79 2070 726f 7669 6465 2074 6865 2073  ly provide the s
-00015060: 7065 6369 6573 2070 6169 7273 2074 6f20  pecies pairs to 
-00015070: 6265 2070 6c6f 7474 6564 2062 7574 2077  be plotted but w
-00015080: 6520 7375 6767 6573 7420 6f66 206a 7573  e suggest of jus
-00015090: 7420 7573 696e 6720 602d 6661 2041 7175  t using `-fa Aqu
-000150a0: 696c 6567 6961 5f63 6f65 7275 6c65 6160  ilegia_coerulea`
-000150b0: 2074 6f20 706c 6f74 2061 6c6c 2070 6f73   to plot all pos
-000150c0: 7369 626c 6520 666f 6361 6c2d 7369 7374  sible focal-sist
-000150d0: 6572 2073 7065 6369 6573 2070 6169 7273  er species pairs
-000150e0: 2e20 5765 2073 7567 6765 7374 2061 6464  . We suggest add
-000150f0: 696e 6720 7468 6520 6f70 7469 6f6e 2060  ing the option `
-00015100: 2d2d 7265 7765 6967 6874 6020 746f 2072  --reweight` to r
-00015110: 6563 616c 6375 6c61 7465 2074 6865 2077  ecalculate the w
-00015120: 6569 6768 7420 7065 7220 7370 6563 6965  eight per specie
-00015130: 7320 7061 6972 2073 7563 6820 7468 6174  s pair such that
-00015140: 2074 6865 2077 6569 6768 7420 6f66 206f   the weight of o
-00015150: 7274 686f 6c6f 676f 7573 2067 656e 6520  rthologous gene 
-00015160: 7061 6972 7320 7769 6c6c 2062 6563 6f6d  pairs will becom
-00015170: 6520 312e 2045 7874 7261 2063 6f6c 6c69  e 1. Extra colli
-00015180: 6e65 6172 2064 6174 6120 6361 6e20 6265  near data can be
-00015190: 2061 6464 6564 2062 7920 7468 6520 6f70   added by the op
-000151a0: 7469 6f6e 2060 2d61 7060 2061 6e64 2061  tion `-ap` and a
-000151b0: 6464 6974 696f 6e61 6c20 636c 7573 7465  dditional cluste
-000151c0: 7269 6e67 2061 6e61 6c79 7369 7320 6361  ring analysis ca
-000151d0: 6e20 6265 2069 6e69 7469 6174 6564 2062  n be initiated b
-000151e0: 7920 7365 7474 696e 6720 7468 6520 6f70  y setting the op
-000151f0: 7469 6f6e 2060 2d2d 706c 6f74 6170 676d  tion `--plotapgm
-00015200: 6d60 2061 6e64 2060 2d2d 706c 6f74 656c  m` and `--plotel
-00015210: 6d6d 602e 0a0a 6060 600a 2828 2856 6974  mm`...```.(((Vit
-00015220: 6973 5f76 696e 6966 6572 612c 5072 6f74  is_vinifera,Prot
-00015230: 6561 5f63 796e 6172 6f69 6465 7329 2c41  ea_cynaroides),A
-00015240: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015250: 6129 2c41 636f 7275 735f 616d 6572 6963  a),Acorus_americ
-00015260: 616e 7573 293b 0a60 6060 0a0a 215b 5d28  anus);.```..![](
-00015270: 6461 7461 2f4d 6978 6564 2e6b 732e 4171  data/Mixed.ks.Aq
-00015280: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015290: 2e6e 6f64 652e 7765 6967 6874 6564 2e70  .node.weighted.p
-000152a0: 6e67 290a 0a54 6865 206d 6978 6564 202a  ng)..The mixed *
-000152b0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
-000152c0: 6973 7472 6962 7574 696f 6e20 7368 6f77  istribution show
-000152d0: 6e20 6162 6f76 6520 6973 2061 2070 7562  n above is a pub
-000152e0: 6c69 6361 7469 6f6e 2d72 6561 6479 2066  lication-ready f
-000152f0: 6967 7572 6520 7468 6174 2061 7373 656d  igure that assem
-00015300: 626c 6573 2074 6865 2072 6573 756c 7473  bles the results
-00015310: 206f 6620 454c 4d4d 2c20 474d 4d20 616e   of ELMM, GMM an
-00015320: 6420 7261 7465 2063 6f72 7265 6374 696f  d rate correctio
-00015330: 6e2e 2054 6865 206f 6e65 2d76 732d 6f6e  n. The one-vs-on
-00015340: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00015350: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00015360: 7374 7269 6275 7469 6f6e 7320 6973 2061  stributions is a
-00015370: 6c73 6f20 6175 746f 6d61 7469 6361 6c6c  lso automaticall
-00015380: 7920 7072 6f64 7563 6564 2077 6974 6820  y produced with 
-00015390: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
-000153a0: 7265 7375 6c74 7320 7375 7065 7269 6d70  results superimp
-000153b0: 6f73 6564 2077 6865 7265 2061 7661 696c  osed where avail
-000153c0: 6162 6c65 2061 7320 7368 6f77 6e20 6265  able as shown be
-000153d0: 6c6f 772e 0a0a 215b 5d28 6461 7461 2f46  low...![](data/F
-000153e0: 6f63 7573 5f73 6973 7465 725f 7061 6972  ocus_sister_pair
-000153f0: 732e 6b73 2e6e 6f64 652e 7765 6967 6874  s.ks.node.weight
-00015400: 6564 2e70 6e67 290a 215b 5d28 6461 7461  ed.png).![](data
-00015410: 2f41 6c6c 5f70 6169 7273 2e6b 732e 6e6f  /All_pairs.ks.no
-00015420: 6465 2e77 6569 6768 7465 642e 706e 6729  de.weighted.png)
-00015430: 0a0a 4265 7369 6465 7320 7468 6520 6162  ..Besides the ab
-00015440: 6f76 6520 6b73 7261 7465 732d 6c69 6b65  ove ksrates-like
-00015450: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00015460: 2070 6c6f 742c 2061 206d 6f72 6520 636c   plot, a more cl
-00015470: 6173 7369 6320 2a4b 2a3c 7375 623e 533c  assic *K*<sub>S<
-00015480: 2f73 7562 3e20 706c 6f74 2063 616e 2062  /sub> plot can b
-00015490: 6520 6d61 6465 2062 7920 6164 6469 6e67  e made by adding
-000154a0: 2074 6865 206f 7074 696f 6e20 602d 2d63   the option `--c
-000154b0: 6c61 7373 6963 6020 746f 2074 6170 206d  lassic` to tap m
-000154c0: 6f72 6520 6465 7461 696c 6564 6c79 2069  ore detailedly i
-000154d0: 6e74 6f20 7468 6520 7661 7269 6174 696f  nto the variatio
-000154e0: 6e20 6f66 2073 796e 6f6e 796d 6f75 7320  n of synonymous 
-000154f0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00015500: 652e 0a0a 5573 696e 6720 7468 6520 636f  e...Using the co
-00015510: 6d6d 616e 6420 6265 6c6f 772c 2074 6865  mmand below, the
-00015520: 2064 6972 6563 7469 6f6e 206f 6620 7261   direction of ra
-00015530: 7465 2063 6f72 7265 6374 696f 6e20 616e  te correction an
-00015540: 6420 6465 6772 6565 206f 6620 7261 7465  d degree of rate
-00015550: 2076 6172 6961 7469 6f6e 2063 616e 2062   variation can b
-00015560: 6520 6f62 7365 7276 6564 206d 6f72 6520  e observed more 
-00015570: 6469 7265 6374 6c79 2e0a 0a60 6060 0a77  directly...```.w
-00015580: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
-00015590: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
-000155a0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
-000155b0: 7376 202d 6661 2041 7175 696c 6567 6961  sv -fa Aquilegia
-000155c0: 5f63 6f65 7275 6c65 6120 2d65 706b 2077  _coerulea -epk w
-000155d0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-000155e0: 5f63 6f65 7275 6c65 612e 6b73 2e74 7376  _coerulea.ks.tsv
-000155f0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
-00015600: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00015610: 6f69 6e74 732e 7478 7420 2d73 7020 7370  oints.txt -sp sp
-00015620: 6563 6965 7374 7265 652e 6e77 202d 6f20  eciestree.nw -o 
-00015630: 7767 645f 7669 7a5f 6d69 7865 645f 4b73  wgd_viz_mixed_Ks
-00015640: 202d 2d70 6c6f 7465 6c6d 6d20 2d2d 706c   --plotelmm --pl
-00015650: 6f74 6170 676d 6d20 2d2d 7265 7765 6967  otapgmm --reweig
-00015660: 6874 202d 2d70 6c6f 746b 6465 202d 2d63  ht --plotkde --c
-00015670: 6c61 7373 6963 0a60 6060 0a0a 215b 5d28  lassic.```..![](
-00015680: 6461 7461 2f41 7175 696c 6567 6961 5f63  data/Aquilegia_c
-00015690: 6f65 7275 6c65 615f 476c 6f62 616c 6d72  oerulea_Globalmr
-000156a0: 6268 4b73 5f45 6c6d 6d5f 4170 676d 6d5f  bhKs_Elmm_Apgmm_
-000156b0: 436f 7272 6563 7465 642e 6b73 642e 7376  Corrected.ksd.sv
-000156c0: 6729 0a0a 4173 2073 686f 776e 2061 626f  g)..As shown abo
-000156d0: 7665 2c20 6265 6361 7573 6520 6f66 2074  ve, because of t
-000156e0: 6865 2068 6967 6865 7220 7375 6273 7469  he higher substi
-000156f0: 7475 7469 6f6e 2072 6174 6520 6f66 202a  tution rate of *
-00015700: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00015710: 6561 2a2c 2074 6865 206f 7269 6769 6e61  ea*, the origina
-00015720: 6c20 6f72 7468 6f6c 6f67 6f75 7320 2a4b  l orthologous *K
-00015730: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-00015740: 6c75 6573 2077 6572 6520 6163 7475 616c  lues were actual
-00015750: 6c79 2075 6e64 6572 6573 7469 6d61 7465  ly underestimate
-00015760: 6420 696e 2074 6865 2074 696d 652d 6672  d in the time-fr
-00015770: 616d 6520 6f66 202a 4171 7569 6c65 6769  ame of *Aquilegi
-00015780: 6120 636f 6572 756c 6561 2a2e 2057 6865  a coerulea*. Whe
-00015790: 6e20 7765 2072 6563 6f76 6572 6564 2074  n we recovered t
-000157a0: 6865 2064 6976 6572 6765 6e63 6520 7375  he divergence su
-000157b0: 6273 7469 7475 7469 6f6e 2064 6973 7461  bstitution dista
-000157c0: 6e63 6520 696e 2074 6572 6d73 206f 6620  nce in terms of 
-000157d0: 7477 6f20 7469 6d65 7320 6f66 2074 6865  two times of the
-000157e0: 2062 7261 6e63 682d 7370 6563 6966 6963   branch-specific
-000157f0: 2063 6f6e 7472 6962 7574 696f 6e20 6f66   contribution of
-00015800: 202a 412e 2063 6f65 7275 6c65 612a 2073   *A. coerulea* s
-00015810: 696e 6365 2069 7473 2064 6976 6572 6765  ince its diverge
-00015820: 6e63 6520 7769 7468 2074 6865 2073 6973  nce with the sis
-00015830: 7465 7220 7370 6563 6965 7320 706c 7573  ter species plus
-00015840: 2074 6865 2073 6861 7265 6420 7375 6273   the shared subs
-00015850: 7469 7475 7469 6f6e 2064 6973 7461 6e63  titution distanc
-00015860: 6520 6265 666f 7265 2064 6976 6572 6765  e before diverge
-00015870: 6e63 6520 2869 6e20 7265 6c61 7469 7665  nce (in relative
-00015880: 2074 6f20 7468 6520 6f75 7467 726f 7570   to the outgroup
-00015890: 292c 2074 6865 2063 6f72 7265 6374 6564  ), the corrected
-000158a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000158b0: 206d 6f64 6520 6265 6361 6d65 206c 6172   mode became lar
-000158c0: 6765 722e 0a0a 4e6f 7465 2074 6861 7420  ger...Note that 
-000158d0: 7765 2063 616e 2065 6173 696c 7920 7368  we can easily sh
-000158e0: 6f77 2074 6861 7420 2a41 7175 696c 6567  ow that *Aquileg
-000158f0: 6961 2063 6f65 7275 6c65 612a 2068 6173  ia coerulea* has
-00015900: 2068 6967 6865 7220 7375 6273 7469 7475   higher substitu
-00015910: 7469 6f6e 2072 6174 6520 7468 616e 202a  tion rate than *
-00015920: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
-00015930: 732a 2061 6e64 202a 5669 7469 7320 7669  s* and *Vitis vi
-00015940: 6e69 6665 7261 2a20 6279 2063 6f6d 7061  nifera* by compa
-00015950: 7269 6e67 2074 6865 6972 2073 7562 7374  ring their subst
-00015960: 6974 7574 696f 6e20 6469 7374 616e 6365  itution distance
-00015970: 2069 6e20 7265 6761 7264 2074 6f20 7468   in regard to th
-00015980: 6520 7361 6d65 2064 6976 6572 6765 6e63  e same divergenc
-00015990: 6520 6576 656e 7420 7769 7468 206f 7574  e event with out
-000159a0: 6772 6f75 7020 7370 6563 6965 7320 2a41  group species *A
-000159b0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-000159c0: 2a2c 2075 7369 6e67 2063 6f6d 6d61 6e64  *, using command
-000159d0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-000159e0: 2076 697a 202d 6420 7767 645f 676c 6f62   viz -d wgd_glob
-000159f0: 616c 6d72 6268 5f6b 732f 676c 6f62 616c  almrbh_ks/global
-00015a00: 5f4d 5242 482e 7473 762e 6b73 2e74 7376  _MRBH.tsv.ks.tsv
-00015a10: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
-00015a20: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
-00015a30: 6f20 7767 645f 7669 7a5f 436f 6d70 6172  o wgd_viz_Compar
-00015a40: 655f 7261 7465 202d 2d73 7061 6972 2022  e_rate --spair "
-00015a50: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-00015a60: 733b 5072 6f74 6561 5f63 796e 6172 6f69  s;Protea_cynaroi
-00015a70: 6465 7322 202d 2d73 7061 6972 2022 4171  des" --spair "Aq
-00015a80: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015a90: 3b41 636f 7275 735f 616d 6572 6963 616e  ;Acorus_american
-00015aa0: 7573 2220 2d2d 7370 6169 7220 2256 6974  us" --spair "Vit
-00015ab0: 6973 5f76 696e 6966 6572 613b 4163 6f72  is_vinifera;Acor
-00015ac0: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
-00015ad0: 2d70 6c6f 746b 6465 202d 2d63 6c61 7373  -plotkde --class
-00015ae0: 6963 0a60 6060 0a0a 215b 5d28 6461 7461  ic.```..![](data
-00015af0: 2f52 6177 5f4f 7274 686f 6c6f 6775 6573  /Raw_Orthologues
-00015b00: 5f43 6f6d 7061 7265 5f72 6174 652e 6b73  _Compare_rate.ks
-00015b10: 642e 7376 6729 0a0a 4173 2064 6973 706c  d.svg)..As displ
-00015b20: 6179 6564 2061 626f 7665 2c20 7468 6520  ayed above, the 
-00015b30: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-00015b40: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
-00015b50: 6573 2062 6577 7465 656e 202a 4171 7569  es bewteen *Aqui
-00015b60: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-00015b70: 616e 6420 2a41 636f 7275 7320 616d 6572  and *Acorus amer
-00015b80: 6963 616e 7573 2a20 6861 7320 7468 6520  icanus* has the 
-00015b90: 6869 6768 6573 7420 6d6f 6465 2c20 696e  highest mode, in
-00015ba0: 6469 6361 7469 6e67 2074 6865 2066 6173  dicating the fas
-00015bb0: 7465 7220 7375 6273 7469 7475 7469 6f6e  ter substitution
-00015bc0: 2072 6174 6520 6f66 202a 412e 2063 6f65   rate of *A. coe
-00015bd0: 7275 6c65 612a 2063 6f6d 7061 7265 6420  rulea* compared 
-00015be0: 746f 202a 5072 6f74 6561 2063 796e 6172  to *Protea cynar
-00015bf0: 6f69 6465 732a 2061 6e64 202a 5669 7469  oides* and *Viti
-00015c00: 7320 7669 6e69 6665 7261 2a2e 0a0a 4265  s vinifera*...Be
-00015c10: 666f 7265 2076 322e 302e 3231 2c20 7468  fore v2.0.21, th
-00015c20: 6520 6765 6e65 2d73 7065 6369 6573 206d  e gene-species m
-00015c30: 6170 2066 696c 6520 6973 206e 6563 6365  ap file is necce
-00015c40: 7373 6172 696c 7920 6e65 6564 6564 2066  ssarily needed f
-00015c50: 6f72 2069 7473 2069 6d70 6c65 6d65 6e74  or its implement
-00015c60: 6174 696f 6e20 696e 2060 7767 6420 7669  ation in `wgd vi
-00015c70: 7a60 2c20 7768 6963 6820 7368 6f75 6c64  z`, which should
-00015c80: 2062 6520 6175 746f 6d61 7465 6c79 2070   be automately p
-00015c90: 726f 6475 6365 6420 6279 2074 6865 206c  roduced by the l
-00015ca0: 6173 7420 6077 6764 206b 7364 6020 7374  ast `wgd ksd` st
-00015cb0: 6570 2067 6976 656e 2074 6865 2060 7370  ep given the `sp
-00015cc0: 6169 7260 2061 6e64 2060 7370 6563 6965  air` and `specie
-00015cd0: 7374 7265 6560 2070 6172 616d 6574 6572  stree` parameter
-00015ce0: 732e 2054 6865 2060 6765 6e65 5f73 7065  s. The `gene_spe
-00015cf0: 6369 6573 2e6d 6170 6020 6861 7320 636f  cies.map` has co
-00015d00: 6e74 656e 7473 2061 7320 6265 6c6f 7720  ntents as below 
-00015d10: 696e 2077 6869 6368 2065 6163 6820 6c69  in which each li
-00015d20: 6e65 2069 7320 7468 6520 6a6f 696e 6564  ne is the joined
-00015d30: 2073 7472 696e 6720 6f66 2067 656e 6520   string of gene 
-00015d40: 6e61 6d65 2061 6e64 2073 7065 6369 6573  name and species
-00015d50: 206e 616d 6520 6279 2073 7061 6365 2e20   name by space. 
-00015d60: 4166 7465 7220 7632 2e30 2e32 3120 2869  After v2.0.21 (i
-00015d70: 6e63 6c75 6465 6429 2c20 7468 6520 6765  ncluded), the ge
-00015d80: 6e65 2d73 7065 6369 6573 206d 6170 2066  ne-species map f
-00015d90: 696c 6520 6973 206e 6f74 206e 6563 6365  ile is not necce
-00015da0: 7373 6172 696c 7920 6e65 6564 6564 2061  ssarily needed a
-00015db0: 6e79 6d6f 7265 2e0a 0a60 6060 0a41 7163  nymore...```.Aqc
-00015dc0: 6f65 3647 3035 3738 3030 2e31 2041 7175  oe6G057800.1 Aqu
-00015dd0: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
-00015de0: 5676 695f 5649 545f 3230 3173 3030 3131  Vvi_VIT_201s0011
-00015df0: 6730 3135 3330 2e31 2056 6974 6973 5f76  g01530.1 Vitis_v
-00015e00: 696e 6966 6572 610a 5063 795f 5072 6f63  inifera.Pcy_Proc
-00015e10: 7930 3167 3038 3531 3020 5072 6f74 6561  y01g08510 Protea
-00015e20: 5f63 796e 6172 6f69 6465 730a 4161 6d5f  _cynaroides.Aam_
-00015e30: 4163 6f72 612e 3034 4731 3432 3930 302e  Acora.04G142900.
-00015e40: 3120 4163 6f72 7573 5f61 6d65 7269 6361  1 Acorus_america
-00015e50: 6e75 730a 6060 600a 0a41 6e20 616c 7465  nus.```..An alte
-00015e60: 726e 6174 6976 6520 7761 7920 746f 2063  rnative way to c
-00015e70: 616c 6375 6c61 7465 2074 6865 206f 7274  alculate the ort
-00015e80: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-00015e90: 3e53 3c2f 7375 623e 2069 7320 746f 2064  >S</sub> is to d
-00015ea0: 6972 6563 746c 7920 7573 6520 7468 6520  irectly use the 
-00015eb0: 6f72 7468 6f67 726f 7570 7320 696e 7374  orthogroups inst
-00015ec0: 6561 6420 6f66 2067 6c6f 6261 6c20 4d52  ead of global MR
-00015ed0: 4248 2066 616d 696c 792e 2054 6861 7420  BH family. That 
-00015ee0: 7761 7920 7765 2064 6f6e 2774 2075 7365  way we don't use
-00015ef0: 2074 6865 2070 7265 2d69 6e66 6572 7265   the pre-inferre
-00015f00: 6420 7061 7261 6e6f 6d65 202a 4b2a 3c73  d paranome *K*<s
-00015f10: 7562 3e53 3c2f 7375 623e 2062 7574 2074  ub>S</sub> but t
-00015f20: 6865 2070 6172 616c 6f67 6f75 7320 6765  he paralogous ge
-00015f30: 6e65 2070 6169 7273 2069 6e73 6964 6520  ne pairs inside 
-00015f40: 6561 6368 206f 7274 686f 6772 6f75 7020  each orthogroup 
-00015f50: 696e 7374 6561 642e 2054 6f20 6163 6869  instead. To achi
-00015f60: 6576 6520 7468 6174 2c20 7765 2066 6972  eve that, we fir
-00015f70: 7374 206e 6565 6420 746f 2069 6e66 6572  st need to infer
-00015f80: 206f 7274 686f 6772 6f75 7073 2075 7369   orthogroups usi
-00015f90: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015fa0: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
-00015fb0: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
-00015fc0: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
-00015fd0: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
-00015fe0: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
-00015ff0: 7669 6e69 6665 7261 202d 2d6f 7274 686f  vinifera --ortho
-00016000: 696e 6665 7220 2d6f 2077 6764 5f6f 7274  infer -o wgd_ort
-00016010: 686f 2028 2d2d 6f6e 6c79 6f72 7468 6f29  ho (--onlyortho)
-00016020: 200a 6060 600a 0a55 7365 7273 2063 616e   .```..Users can
-00016030: 2064 6563 6964 6520 746f 206f 6e6c 7920   decide to only 
-00016040: 636f 6e64 7563 7420 7468 6520 6f72 7468  conduct the orth
-00016050: 6f67 726f 7570 2061 6e61 6c79 7369 7320  ogroup analysis 
-00016060: 7768 696c 6520 736b 6970 7069 6e67 206f  while skipping o
-00016070: 7468 6572 2061 6e61 6c79 7369 7320 6279  ther analysis by
-00016080: 2061 6464 696e 6720 7468 6520 666c 6167   adding the flag
-00016090: 2060 2d2d 6f6e 6c79 6f72 7468 6f60 2e20   `--onlyortho`. 
-000160a0: 4e65 7874 2073 7465 7020 6973 2074 6865  Next step is the
-000160b0: 2073 616d 6520 7769 7468 2067 6c6f 6261   same with globa
-000160c0: 6c20 4d52 4248 2066 616d 696c 792e 0a0a  l MRBH family...
-000160d0: 6060 600a 7767 6420 6b73 6420 7767 645f  ```.wgd ksd wgd_
-000160e0: 6f72 7468 6f2f 4f72 7468 6f67 726f 7570  ortho/Orthogroup
-000160f0: 732e 7370 2e74 7376 2041 7175 696c 6567  s.sp.tsv Aquileg
-00016100: 6961 5f63 6f65 7275 6c65 6120 5072 6f74  ia_coerulea Prot
-00016110: 6561 5f63 796e 6172 6f69 6465 7320 4163  ea_cynaroides Ac
-00016120: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
-00016130: 5669 7469 735f 7669 6e69 6665 7261 202d  Vitis_vinifera -
-00016140: 6f20 7767 645f 6f72 7468 6f5f 6b73 0a77  o wgd_ortho_ks.w
-00016150: 6764 2076 697a 202d 6420 7767 645f 6f72  gd viz -d wgd_or
-00016160: 7468 6f5f 6b73 2f4f 7274 686f 6772 6f75  tho_ks/Orthogrou
-00016170: 7073 2e73 702e 7473 762e 6b73 2e74 7376  ps.sp.tsv.ks.tsv
-00016180: 202d 6570 6b20 7767 645f 6b73 642f 4171   -epk wgd_ksd/Aq
-00016190: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000161a0: 2e6b 732e 7473 7620 2d73 7020 7370 6563  .ks.tsv -sp spec
-000161b0: 6965 7374 7265 652e 6e77 202d 2d72 6577  iestree.nw --rew
-000161c0: 6569 6768 7420 2d61 7020 7767 645f 7379  eight -ap wgd_sy
-000161d0: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
-000161e0: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
-000161f0: 706c 6f74 656c 6d6d 202d 2d70 6c6f 7461  plotelmm --plota
-00016200: 7067 6d6d 202d 6f20 7767 645f 6f72 7468  pgmm -o wgd_orth
-00016210: 6f5f 6b73 5f72 6174 655f 636f 7272 6563  o_ks_rate_correc
-00016220: 7469 6f6e 0a60 6060 0a0a 215b 5d28 6461  tion.```..![](da
-00016230: 7461 2f4d 6978 6564 2e6b 732e 4171 7569  ta/Mixed.ks.Aqui
-00016240: 6c65 6769 615f 636f 6572 756c 6561 2e6e  legia_coerulea.n
-00016250: 6f64 652e 7765 6967 6874 6564 5f4f 7274  ode.weighted_Ort
-00016260: 686f 5f4b 732e 706e 6729 0a21 5b5d 2864  ho_Ks.png).![](d
-00016270: 6174 612f 466f 6375 735f 7369 7374 6572  ata/Focus_sister
-00016280: 5f70 6169 7273 2e6b 732e 6e6f 6465 2e77  _pairs.ks.node.w
-00016290: 6569 6768 7465 645f 4f72 7468 6f5f 4b73  eighted_Ortho_Ks
-000162a0: 2e70 6e67 290a 215b 5d28 6461 7461 2f41  .png).![](data/A
-000162b0: 6c6c 5f70 6169 7273 2e6b 732e 6e6f 6465  ll_pairs.ks.node
-000162c0: 2e77 6569 6768 7465 645f 4f72 7468 6f5f  .weighted_Ortho_
-000162d0: 4b73 2e70 6e67 290a 0a41 7320 7368 6f77  Ks.png)..As show
-000162e0: 6e20 6162 6f76 652c 2074 6865 206e 756d  n above, the num
-000162f0: 6265 7220 6f66 206f 7274 686f 6c6f 676f  ber of orthologo
-00016300: 7573 2067 656e 6520 7061 6972 7320 6973  us gene pairs is
-00016310: 2064 6966 6665 7265 6e74 2074 6861 6e20   different than 
-00016320: 7468 6520 6f6e 6520 6672 6f6d 2067 6c6f  the one from glo
-00016330: 6261 6c20 4d52 4248 2066 616d 696c 6965  bal MRBH familie
-00016340: 7320 696e 2074 6861 7420 6865 7265 2077  s in that here w
-00016350: 6520 706c 6f74 7465 6420 616c 6c20 6f72  e plotted all or
-00016360: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
-00016370: 6169 7273 2069 6e73 7465 6164 206f 6620  airs instead of 
-00016380: 6f6e 6c79 2067 6c6f 6261 6c20 4d52 4248  only global MRBH
-00016390: 2066 616d 696c 6965 732c 2074 6f67 6574   families, toget
-000163a0: 6865 7220 7769 7468 2064 6966 6665 7265  her with differe
-000163b0: 6e74 2072 6563 616c 6375 6c61 7465 6420  nt recalculated 
-000163c0: 7765 6967 6874 732e 0a0a 4166 7465 7220  weights...After 
-000163d0: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
-000163e0: 2074 696d 696e 6720 6f66 2074 6865 2052   timing of the R
-000163f0: 616e 756e 6375 6c61 6c65 7320 5747 442c  anunculales WGD,
-00016400: 2077 6520 6361 6e20 6675 7274 6865 7220   we can further 
-00016410: 696e 6665 7220 6974 7320 6162 736f 6c75  infer its absolu
-00016420: 7465 2061 6765 2e20 4669 7273 7420 7765  te age. First we
-00016430: 2069 6e66 6572 2074 6865 2063 7265 6469   infer the credi
-00016440: 626c 6520 7261 6e67 6520 6f66 2061 6e63  ble range of anc
-00016450: 686f 7220 7061 6972 7320 6279 202a 4b2a  hor pairs by *K*
-00016460: 3c73 7562 3e53 3c2f 7375 623e 2068 6575  <sub>S</sub> heu
-00016470: 7269 7374 6963 616c 6c79 2075 7369 6e67  ristically using
-00016480: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
-00016490: 6420 7065 616b 602e 0a0a 6060 600a 7767  d peak`...```.wg
-000164a0: 6420 7065 616b 202d 2d68 6575 7269 7374  d peak --heurist
-000164b0: 6963 2077 6764 5f6b 7364 2f41 7175 696c  ic wgd_ksd/Aquil
-000164c0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-000164d0: 762e 6b73 2e74 7376 202d 6170 2077 6764  v.ks.tsv -ap wgd
-000164e0: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-000164f0: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
-00016500: 7420 2d73 6d20 7767 645f 7379 6e2f 6961  t -sm wgd_syn/ia
-00016510: 6468 6f72 652d 6f75 742f 7365 676d 656e  dhore-out/segmen
-00016520: 7473 2e74 7874 202d 6c65 2077 6764 5f73  ts.txt -le wgd_s
-00016530: 796e 2f69 6164 686f 7265 2d6f 7574 2f6c  yn/iadhore-out/l
-00016540: 6973 745f 656c 656d 656e 7473 2e74 7874  ist_elements.txt
-00016550: 202d 6d70 2077 6764 5f73 796e 2f69 6164   -mp wgd_syn/iad
-00016560: 686f 7265 2d6f 7574 2f6d 756c 7469 706c  hore-out/multipl
-00016570: 6963 6f6e 5f70 6169 7273 2e74 7874 202d  icon_pairs.txt -
-00016580: 6f20 7767 645f 7065 616b 0a60 6060 0a0a  o wgd_peak.```..
-00016590: 215b 5d28 6461 7461 2f41 6e63 686f 724b  ![](data/AnchorK
-000165a0: 735f 5065 616b 4349 5f41 7175 696c 6567  s_PeakCI_Aquileg
-000165b0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-000165c0: 6b73 2e74 7376 5f6e 6f64 655f 7765 6967  ks.tsv_node_weig
-000165d0: 6874 6564 2e73 7667 290a 0a41 7320 7368  hted.svg)..As sh
-000165e0: 6f77 6e20 6162 6f76 652c 2077 6520 6173  own above, we as
-000165f0: 7375 6d65 6420 6120 6c6f 676e 6f72 6d61  sumed a lognorma
-00016600: 6c20 6469 7374 7269 6275 7469 6f6e 2061  l distribution a
-00016610: 7420 7468 6520 7065 616b 206c 6f63 6174  t the peak locat
-00016620: 696f 6e20 6465 7465 6374 6564 2062 7920  ion detected by 
-00016630: 7468 6520 6073 6967 6e61 6c60 206d 6f64  the `signal` mod
-00016640: 756c 6520 6f66 2060 7363 6970 7960 206c  ule of `scipy` l
-00016650: 6962 7261 7279 2e20 5468 6520 3935 2520  ibrary. The 95% 
-00016660: 636f 6e66 6964 656e 6365 206c 6576 656c  confidence level
-00016670: 206f 6620 7468 6520 6c6f 676e 6f72 6d61   of the lognorma
-00016680: 6c20 6469 7374 7269 6275 7469 6f6e 2077  l distribution w
-00016690: 6173 2061 7070 6c69 6564 2c20 692e 652e  as applied, i.e.
-000166a0: 2c20 302e 3638 2d32 2e37 342c 2069 6e20  , 0.68-2.74, in 
-000166b0: 6675 7274 6865 7220 6d6f 6c65 6375 6c61  further molecula
-000166c0: 7220 6461 7469 6e67 2e20 5468 6520 6669  r dating. The fi
-000166d0: 6c65 2060 4171 7569 6c65 6769 615f 636f  le `Aquilegia_co
-000166e0: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-000166f0: 765f 3935 2543 495f 4150 5f66 6f72 5f64  v_95%CI_AP_for_d
-00016700: 6174 696e 675f 7765 6967 6874 6564 5f66  ating_weighted_f
-00016710: 6f72 6d61 742e 7473 7660 2069 7320 7768  ormat.tsv` is wh
-00016720: 6174 2077 6520 6e65 6564 2066 6f72 206e  at we need for n
-00016730: 6578 7420 7374 6570 2e20 546f 2062 7569  ext step. To bui
-00016740: 6c64 2074 6865 206f 7274 686f 6772 6f75  ld the orthogrou
-00016750: 7073 2075 7365 6420 696e 2070 6879 6c6f  ps used in phylo
-00016760: 6765 6e65 7469 6320 6461 7469 6e67 2c20  genetic dating, 
-00016770: 7765 206e 6565 6420 746f 2073 656c 6563  we need to selec
-00016780: 7420 736f 6d65 2073 7065 6369 6573 2061  t some species a
-00016790: 6e64 2066 6f72 6d20 6120 7374 6172 7469  nd form a starti
-000167a0: 6e67 2074 7265 6520 7769 7468 2070 726f  ng tree with pro
-000167b0: 7065 7220 666f 7373 696c 2063 616c 6962  per fossil calib
-000167c0: 7261 7469 6f6e 732e 2057 6520 7072 6f76  rations. We prov
-000167d0: 6964 6520 6f6e 6520 696e 206d 636d 6374  ide one in mcmct
-000167e0: 7265 6520 666f 726d 6174 2061 7320 6265  ree format as be
-000167f0: 6c6f 772e 0a0a 6060 600a 3137 2031 0a28  low...```.17 1.(
-00016800: 2828 2850 6f74 616d 6f67 6574 6f6e 5f61  (((Potamogeton_a
-00016810: 6375 7469 666f 6c69 7573 2c28 5370 6972  cutifolius,(Spir
-00016820: 6f64 656c 615f 696e 7465 726d 6564 6961  odela_intermedia
-00016830: 2c41 6d6f 7270 686f 7068 616c 6c75 735f  ,Amorphophallus_
-00016840: 6b6f 6e6a 6163 2929 2c28 4163 616e 7468  konjac)),(Acanth
-00016850: 6f63 686c 616d 7973 5f62 7261 6374 6561  ochlamys_bractea
-00016860: 7461 2c28 4469 6f73 636f 7265 615f 616c  ta,(Dioscorea_al
-00016870: 6174 612c 4469 6f73 636f 7265 615f 726f  ata,Dioscorea_ro
-00016880: 7475 6e64 6174 6129 2927 3e30 2e35 3630  tundata))'>0.560
-00016890: 303c 312e 3238 3633 2729 273e 302e 3833  0<1.2863')'>0.83
-000168a0: 3630 3c31 2e32 3836 3327 2c28 4163 6f72  60<1.2863',(Acor
-000168b0: 7573 5f61 6d65 7269 6361 6e75 732c 4163  us_americanus,Ac
-000168c0: 6f72 7573 5f74 6174 6172 696e 6f77 6969  orus_tatarinowii
-000168d0: 2929 273e 302e 3833 3630 3c31 2e32 3836  ))'>0.8360<1.286
-000168e0: 3327 2c28 2828 2854 6574 7261 6365 6e74  3',((((Tetracent
-000168f0: 726f 6e5f 7369 6e65 6e73 652c 5472 6f63  ron_sinense,Troc
-00016900: 686f 6465 6e64 726f 6e5f 6172 616c 696f  hodendron_aralio
-00016910: 6964 6573 292c 2842 7578 7573 5f61 7573  ides),(Buxus_aus
-00016920: 7472 6f79 756e 6e61 6e65 6e73 6973 2c42  troyunnanensis,B
-00016930: 7578 7573 5f73 696e 6963 6129 2927 3e31  uxus_sinica))'>1
-00016940: 2e31 3038 303c 312e 3238 3633 272c 284e  .1080<1.2863',(N
-00016950: 656c 756d 626f 5f6e 7563 6966 6572 612c  elumbo_nucifera,
-00016960: 2854 656c 6f70 6561 5f73 7065 6369 6f73  (Telopea_specios
-00016970: 6973 7369 6d61 2c50 726f 7465 615f 6379  issima,Protea_cy
-00016980: 6e61 726f 6964 6573 2929 2927 3e31 2e31  naroides)))'>1.1
-00016990: 3038 303c 312e 3238 3633 272c 2841 7175  080<1.2863',(Aqu
-000169a0: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
-000169b0: 6170 312c 4171 7569 6c65 6769 615f 636f  ap1,Aquilegia_co
-000169c0: 6572 756c 6561 5f61 7032 2929 273e 312e  erulea_ap2))'>1.
-000169d0: 3130 3830 3c31 2e32 3836 3327 2927 3e31  1080<1.2863')'>1
-000169e0: 2e32 3732 303c 322e 3437 3230 273b 0a60  .2720<2.4720';.`
-000169f0: 6060 0a0a 4173 2070 7265 7365 6e74 6564  ``..As presented
-00016a00: 2061 626f 7665 2c20 7468 6520 666f 6375   above, the focu
-00016a10: 7320 7370 6563 6965 7320 7468 6174 2069  s species that i
-00016a20: 7320 6162 6f75 7420 746f 2062 6520 6461  s about to be da
-00016a30: 7465 6420 6e65 6564 7320 746f 2062 6520  ted needs to be 
-00016a40: 7265 706c 6163 6564 2077 6974 6820 6028  replaced with `(
-00016a50: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016a60: 6561 5f61 7031 2c41 7175 696c 6567 6961  ea_ap1,Aquilegia
-00016a70: 5f63 6f65 7275 6c65 615f 6170 3229 602e  _coerulea_ap2)`.
-00016a80: 2057 6974 6820 7468 6973 2073 7461 7274   With this start
-00016a90: 696e 6720 7472 6565 2061 6e64 2070 7265  ing tree and pre
-00016aa0: 646f 776e 6c6f 6164 6564 2063 6473 2066  downloaded cds f
-00016ab0: 696c 6573 206f 6620 616c 6c20 7468 6520  iles of all the 
-00016ac0: 7370 6563 6965 732c 2077 6520 6361 6e20  species, we can 
-00016ad0: 6275 696c 6420 7468 6520 6f72 7468 6f67  build the orthog
-00016ae0: 726f 7570 2075 7365 6420 696e 2074 6865  roup used in the
-00016af0: 2066 696e 616c 206d 6f6c 6563 756c 6172   final molecular
-00016b00: 2064 6174 696e 6720 7573 696e 6720 7468   dating using th
-00016b10: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
-00016b20: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
-00016b30: 202d 6620 4171 7569 6c65 6769 615f 636f   -f Aquilegia_co
-00016b40: 6572 756c 6561 202d 6170 2077 6764 5f70  erulea -ap wgd_p
-00016b50: 6561 6b2f 4171 7569 6c65 6769 615f 636f  eak/Aquilegia_co
-00016b60: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-00016b70: 765f 3935 2543 495f 4150 5f66 6f72 5f64  v_95%CI_AP_for_d
-00016b80: 6174 696e 675f 7765 6967 6874 6564 5f66  ating_weighted_f
-00016b90: 6f72 6d61 742e 7473 7620 2d6f 2077 6764  ormat.tsv -o wgd
-00016ba0: 5f64 6d64 5f6f 7274 686f 2050 6f74 616d  _dmd_ortho Potam
-00016bb0: 6f67 6574 6f6e 5f61 6375 7469 666f 6c69  ogeton_acutifoli
-00016bc0: 7573 2053 7069 726f 6465 6c61 5f69 6e74  us Spirodela_int
-00016bd0: 6572 6d65 6469 6120 416d 6f72 7068 6f70  ermedia Amorphop
-00016be0: 6861 6c6c 7573 5f6b 6f6e 6a61 6320 4163  hallus_konjac Ac
-00016bf0: 616e 7468 6f63 686c 616d 7973 5f62 7261  anthochlamys_bra
-00016c00: 6374 6561 7461 2044 696f 7363 6f72 6561  cteata Dioscorea
-00016c10: 5f61 6c61 7461 2044 696f 7363 6f72 6561  _alata Dioscorea
-00016c20: 5f72 6f74 756e 6461 7461 2041 636f 7275  _rotundata Acoru
-00016c30: 735f 616d 6572 6963 616e 7573 2041 636f  s_americanus Aco
-00016c40: 7275 735f 7461 7461 7269 6e6f 7769 6920  rus_tatarinowii 
-00016c50: 5465 7472 6163 656e 7472 6f6e 5f73 696e  Tetracentron_sin
-00016c60: 656e 7365 2054 726f 6368 6f64 656e 6472  ense Trochodendr
-00016c70: 6f6e 5f61 7261 6c69 6f69 6465 7320 4275  on_aralioides Bu
-00016c80: 7875 735f 6175 7374 726f 7975 6e6e 616e  xus_austroyunnan
-00016c90: 656e 7369 7320 4275 7875 735f 7369 6e69  ensis Buxus_sini
-00016ca0: 6361 204e 656c 756d 626f 5f6e 7563 6966  ca Nelumbo_nucif
-00016cb0: 6572 6120 5465 6c6f 7065 615f 7370 6563  era Telopea_spec
-00016cc0: 696f 7369 7373 696d 6120 5072 6f74 6561  iosissima Protea
-00016cd0: 5f63 796e 6172 6f69 6465 7320 4171 7569  _cynaroides Aqui
-00016ce0: 6c65 6769 615f 636f 6572 756c 6561 0a60  legia_coerulea.`
-00016cf0: 6060 0a0a 5468 6520 7265 7375 6c74 2066  ``..The result f
-00016d00: 696c 6520 606d 6572 6765 5f66 6f63 7573  ile `merge_focus
-00016d10: 5f61 702e 7473 7660 2069 7320 7768 6174  _ap.tsv` is what
-00016d20: 2077 6520 6e65 6564 2066 6f72 2074 6865   we need for the
-00016d30: 2066 696e 616c 2073 7465 7020 6f66 206d   final step of m
-00016d40: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
-00016d50: 696e 2070 726f 6772 616d 2060 7767 6420  in program `wgd 
-00016d60: 666f 6375 7360 2e0a 0a60 6060 0a77 6764  focus`...```.wgd
-00016d70: 2066 6f63 7573 202d 2d70 726f 7464 6174   focus --protdat
-00016d80: 696e 6720 2d2d 6161 6d6f 6465 6c20 6c67  ing --aamodel lg
-00016d90: 2077 6764 5f64 6d64 5f6f 7274 686f 2f6d   wgd_dmd_ortho/m
-00016da0: 6572 6765 5f66 6f63 7573 5f61 702e 7473  erge_focus_ap.ts
-00016db0: 7620 2d73 7020 6461 7469 6e67 5f74 7265  v -sp dating_tre
-00016dc0: 652e 6e77 202d 6f20 7767 645f 6461 7469  e.nw -o wgd_dati
-00016dd0: 6e67 202d 6420 6d63 6d63 7472 6565 202d  ng -d mcmctree -
-00016de0: 6473 2027 6275 726e 696e 203d 2032 3030  ds 'burnin = 200
-00016df0: 3027 202d 6473 2027 7361 6d70 6672 6571  0' -ds 'sampfreq
-00016e00: 203d 2031 3030 3027 202d 6473 2027 6e73   = 1000' -ds 'ns
-00016e10: 616d 706c 6520 3d20 3230 3030 3027 2050  ample = 20000' P
-00016e20: 6f74 616d 6f67 6574 6f6e 5f61 6375 7469  otamogeton_acuti
-00016e30: 666f 6c69 7573 2053 7069 726f 6465 6c61  folius Spirodela
-00016e40: 5f69 6e74 6572 6d65 6469 6120 416d 6f72  _intermedia Amor
-00016e50: 7068 6f70 6861 6c6c 7573 5f6b 6f6e 6a61  phophallus_konja
-00016e60: 6320 4163 616e 7468 6f63 686c 616d 7973  c Acanthochlamys
-00016e70: 5f62 7261 6374 6561 7461 2044 696f 7363  _bracteata Diosc
-00016e80: 6f72 6561 5f61 6c61 7461 2044 696f 7363  orea_alata Diosc
-00016e90: 6f72 6561 5f72 6f74 756e 6461 7461 2041  orea_rotundata A
-00016ea0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-00016eb0: 2041 636f 7275 735f 7461 7461 7269 6e6f   Acorus_tatarino
-00016ec0: 7769 6920 5465 7472 6163 656e 7472 6f6e  wii Tetracentron
-00016ed0: 5f73 696e 656e 7365 2054 726f 6368 6f64  _sinense Trochod
-00016ee0: 656e 6472 6f6e 5f61 7261 6c69 6f69 6465  endron_aralioide
-00016ef0: 7320 4275 7875 735f 6175 7374 726f 7975  s Buxus_austroyu
-00016f00: 6e6e 616e 656e 7369 7320 4275 7875 735f  nnanensis Buxus_
-00016f10: 7369 6e69 6361 204e 656c 756d 626f 5f6e  sinica Nelumbo_n
-00016f20: 7563 6966 6572 6120 5465 6c6f 7065 615f  ucifera Telopea_
-00016f30: 7370 6563 696f 7369 7373 696d 6120 5072  speciosissima Pr
-00016f40: 6f74 6561 5f63 796e 6172 6f69 6465 7320  otea_cynaroides 
-00016f50: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016f60: 6561 0a60 6060 0a0a 4865 7265 2077 6520  ea.```..Here we 
-00016f70: 6f6e 6c79 2069 6d70 6c65 6d65 6e74 6564  only implemented
-00016f80: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
-00016f90: 6f6e 2061 6e61 6c79 7369 7320 7573 696e  on analysis usin
-00016fa0: 6720 7072 6f74 6569 6e20 7365 7175 656e  g protein sequen
-00016fb0: 6365 2062 7920 6164 6469 6e67 2074 6865  ce by adding the
-00016fc0: 2066 6c61 6720 602d 2d70 726f 7464 6174   flag `--protdat
-00016fd0: 696e 6760 2061 6e64 2077 6520 7365 7420  ing` and we set 
-00016fe0: 7468 6520 7061 7261 6d65 7465 7220 666f  the parameter fo
-00016ff0: 7220 606d 636d 6374 7265 6560 2076 6961  r `mcmctree` via
-00017000: 2074 6865 206f 7074 696f 6e20 602d 6473   the option `-ds
-00017010: 602e 204e 6f74 6520 7468 6174 206f 7468  `. Note that oth
-00017020: 6572 2064 6174 696e 6720 7072 6f67 7261  er dating progra
-00017030: 6d20 7375 6368 2061 7320 6072 3873 6020  m such as `r8s` 
-00017040: 616e 6420 6062 6561 7374 6020 6172 6520  and `beast` are 
-00017050: 616c 736f 2061 7661 696c 6162 6c65 2067  also available g
-00017060: 6976 656e 2073 6f6d 6520 6d61 6e64 6174  iven some mandat
-00017070: 6f72 7920 7061 7261 6d65 7465 7273 2e20  ory parameters. 
-00017080: 5468 6520 6669 6e61 6c20 6c6f 6720 6f66  The final log of
-00017090: 2074 6865 2073 7563 6365 7373 6675 6c20   the successful 
-000170a0: 7275 6e20 6973 2061 7320 6265 6c6f 772e  run is as below.
-000170b0: 0a0a 6060 600a 3136 3a30 343a 3235 2049  ..```.16:04:25 I
-000170c0: 4e46 4f20 2020 2020 5275 6e6e 696e 6720  NFO     Running 
-000170d0: 6d63 6d63 7472 6565 2075 7369 6e67 2048  mcmctree using H
-000170e0: 6573 7369 616e 206d 6174 7269 7820 6f66  essian matrix of
-000170f0: 204c 472b 4761 6d6d 6120 2063 6f72 652e   LG+Gamma  core.
-00017100: 7079 3a39 3637 0a20 2020 2020 2020 2020  py:967.         
-00017110: 2020 2020 2020 2020 2066 6f72 2070 726f           for pro
-00017120: 7465 696e 206d 6f64 656c 0a32 333a 3439  tein model.23:49
-00017130: 3a33 3720 494e 464f 2020 2020 2050 6f73  :37 INFO     Pos
-00017140: 7465 7269 6f72 206d 6561 6e20 666f 7220  terior mean for 
-00017150: 7468 6520 6167 6573 206f 6620 7767 6420  the ages of wgd 
-00017160: 6973 2031 3132 2e38 3934 3520 6d63 6d63  is 112.8945 mcmc
-00017170: 7472 6565 2e70 793a 3239 360a 2020 2020  tree.py:296.    
-00017180: 2020 2020 2020 2020 2020 2020 2020 6d69                mi
-00017190: 6c6c 696f 6e20 7965 6172 7320 6672 6f6d  llion years from
-000171a0: 2043 6f6e 6361 7465 6e61 7465 6420 7065   Concatenated pe
-000171b0: 7074 6964 650a 2020 2020 2020 2020 2020  ptide.          
-000171c0: 2020 2020 2020 2020 616c 6967 6e6d 656e          alignmen
-000171d0: 7420 616e 6420 3935 2520 6372 6564 6962  t and 95% credib
-000171e0: 696c 6974 7920 696e 7465 7276 616c 7320  ility intervals 
-000171f0: 2843 4929 0a20 2020 2020 2020 2020 2020  (CI).           
-00017200: 2020 2020 2020 2069 7320 3130 312e 3232         is 101.22
-00017210: 342d 3132 332e 3132 3120 6d69 6c6c 696f  4-123.121 millio
-00017220: 6e20 7965 6172 730a 2020 2020 2020 2020  n years.        
-00017230: 2049 4e46 4f20 2020 2020 546f 7461 6c20   INFO     Total 
-00017240: 7275 6e20 7469 6d65 3a20 3239 3137 3573  run time: 29175s
-00017250: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017260: 2020 2020 2020 2020 2020 2020 2020 636c                cl
-00017270: 692e 7079 3a32 3431 0a20 2020 2020 2020  i.py:241.       
-00017280: 2020 494e 464f 2020 2020 2044 6f6e 6520    INFO     Done 
-00017290: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000172a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000172b0: 2020 2020 2020 2020 2020 2020 2020 2063                 c
-000172c0: 6c69 2e70 793a 3234 320a 6060 600a 0a54  li.py:242.```..T
-000172d0: 6f20 7669 7375 616c 697a 6520 7468 6520  o visualize the 
-000172e0: 6461 7465 2c20 7765 2061 6c73 6f20 7072  date, we also pr
-000172f0: 6f76 6964 6564 2061 2070 7974 686f 6e20  ovided a python 
-00017300: 7363 7269 7074 2074 6f20 706c 6f74 2074  script to plot t
-00017310: 6865 2057 4744 2064 6174 6573 2069 6e20  he WGD dates in 
-00017320: 7468 6520 6077 6764 6020 666f 6c64 6572  the `wgd` folder
-00017330: 2e20 5573 6572 7320 6e65 6564 2074 6f20  . Users need to 
-00017340: 6578 7472 6163 7420 7468 6520 7261 7720  extract the raw 
-00017350: 6461 7465 7320 6672 6f6d 2074 6865 2060  dates from the `
-00017360: 6d63 6d63 2e74 7874 6020 666f 7220 7468  mcmc.txt` for th
-00017370: 6520 5747 4420 6e6f 6465 2066 6972 7374  e WGD node first
-00017380: 2061 6e64 2073 6176 6520 6974 2061 7320   and save it as 
-00017390: 6669 6c65 2060 6461 7465 732e 7478 7460  file `dates.txt`
-000173a0: 2028 6f72 2077 6861 7465 7665 7220 7072   (or whatever pr
-000173b0: 6566 6572 7265 6420 6e61 6d65 292e 2041  eferred name). A
-000173c0: 6e20 6578 616d 706c 6520 636f 6d6d 616e  n example comman
-000173d0: 6420 6973 2061 7320 6265 6c6f 772e 0a0a  d is as below...
-000173e0: 6060 600a 7079 7468 6f6e 2024 5041 5448  ```.python $PATH
-000173f0: 2f70 6f73 7470 6c6f 742e 7079 2070 6f73  /postplot.py pos
-00017400: 7464 6973 2064 6174 6573 2e74 7874 202d  tdis dates.txt -
-00017410: 2d70 6572 6365 6e74 696c 6520 3930 202d  -percentile 90 -
-00017420: 2d74 6974 6c65 2022 5747 4420 6461 7465  -title "WGD date
-00017430: 2220 2d2d 6870 6420 2d6f 2022 5261 6e75  " --hpd -o "Ranu
-00017440: 6e63 756c 616c 6573 5f57 4744 5f64 6174  nculales_WGD_dat
-00017450: 652e 7376 6722 0a60 6060 0a0a 215b 5d28  e.svg".```..![](
-00017460: 6461 7461 2f52 616e 756e 6375 6c61 6c65  data/Ranunculale
-00017470: 735f 5747 445f 6461 7465 2e73 7667 290a  s_WGD_date.svg).
-00017480: 0a54 6865 2070 6f73 7465 7269 6f72 206d  .The posterior m
-00017490: 6561 6e2c 206d 6564 6961 6e20 616e 6420  ean, median and 
-000174a0: 6d6f 6465 206f 6620 7468 6520 5261 6e75  mode of the Ranu
-000174b0: 6e63 756c 616c 6573 2057 4744 2061 6765  nculales WGD age
-000174c0: 2069 7320 3131 322e 3932 2c20 3131 332e   is 112.92, 113.
-000174d0: 3434 2061 6e64 2031 3132 2e35 3420 6d79  44 and 112.54 my
-000174e0: 612c 2077 6974 6820 3930 2520 4850 4420  a, with 90% HPD 
-000174f0: 3130 352e 3037 202d 2031 3232 2e33 3220  105.07 - 122.32 
-00017500: 6d79 6120 6173 206d 616e 6966 6573 7465  mya as manifeste
-00017510: 6420 6162 6f76 652e 0a0a 2323 2320 4b73  d above...### Ks
-00017520: 7472 6565 0a0a 496e 2061 6464 6974 696f  tree..In additio
-00017530: 6e20 746f 2070 6169 7277 6973 6520 2a4b  n to pairwise *K
-00017540: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
-00017550: 7469 6d61 7469 6f6e 2c20 6120 2a4b 2a3c  timation, a *K*<
-00017560: 7375 623e 533c 2f73 7562 3e20 7472 6565  sub>S</sub> tree
-00017570: 2077 6974 6820 6272 616e 6368 206c 656e   with branch len
-00017580: 6774 6820 696e 202a 4b2a 3c73 7562 3e53  gth in *K*<sub>S
-00017590: 3c2f 7375 623e 2075 6e69 7420 6361 6e20  </sub> unit can 
-000175a0: 616c 736f 2062 6520 6465 7269 7665 6420  also be derived 
-000175b0: 6672 6f6d 2074 6865 2070 726f 6772 616d  from the program
-000175c0: 2060 7767 6420 6b73 6460 2067 6976 656e   `wgd ksd` given
-000175d0: 2074 6865 206f 7074 696f 6e20 602d 2d6b   the option `--k
-000175e0: 7374 7265 6560 2061 6e64 2060 2d2d 7370  stree` and `--sp
-000175f0: 6563 6965 7374 7265 6560 2e20 4e6f 7465  eciestree`. Note
-00017600: 2074 6861 7420 7468 6520 6164 6469 7469   that the additi
-00017610: 6f6e 616c 206f 7074 696f 6e20 602d 2d6f  onal option `--o
-00017620: 6e6c 7963 6f6e 6361 746b 7374 7265 6560  nlyconcatkstree`
-00017630: 2077 696c 6c20 6f6e 6c79 2063 616c 6c20   will only call 
-00017640: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-00017650: 7562 3e20 6573 7469 6d61 7469 6f6e 2066  ub> estimation f
-00017660: 6f72 2074 6865 2063 6f6e 6361 7465 6e61  or the concatena
-00017670: 7465 6420 616c 6967 6e6d 656e 7420 7261  ted alignment ra
-00017680: 7468 6572 2074 6861 6e20 616c 6c20 7468  ther than all th
-00017690: 6520 616c 6967 6e6d 656e 7473 2e20 5573  e alignments. Us
-000176a0: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
-000176b0: 6964 6520 6120 7072 6573 6574 2073 7065  ide a preset spe
-000176c0: 6369 6573 2074 7265 6520 666f 7220 7468  cies tree for th
-000176d0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000176e0: 3e20 7472 6565 2069 6e66 6572 656e 6365  > tree inference
-000176f0: 206f 6620 7468 6520 636f 6e63 6174 656e   of the concaten
-00017700: 6174 6564 2061 6c69 676e 6d65 6e74 2077  ated alignment w
-00017710: 6869 6c65 2074 6865 2072 656d 6169 6e69  hile the remaini
-00017720: 6e67 2061 6c69 676e 6d65 6e74 7320 7769  ng alignments wi
-00017730: 6c6c 2062 6520 6167 6169 6e73 7420 616e  ll be against an
-00017740: 2061 7574 6f6d 6174 656c 7920 696e 6665   automately infe
-00017750: 7272 6564 2074 7265 6520 6672 6f6d 2060  rred tree from `
-00017760: 6661 7374 7472 6565 6020 6f72 2060 6971  fasttree` or `iq
-00017770: 7472 6565 602e 2049 6e20 7468 6520 656e  tree`. In the en
-00017780: 642c 2075 7365 7273 2077 696c 6c20 6765  d, users will ge
-00017790: 7420 6120 2a4b 2a3c 7375 623e 533c 2f73  t a *K*<sub>S</s
-000177a0: 7562 3e20 7472 6565 2c20 6120 2a4b 2a3c  ub> tree, a *K*<
-000177b0: 7375 623e 413c 2f73 7562 3e20 7472 6565  sub>A</sub> tree
-000177c0: 2061 6e64 2061 20cf 8920 7472 6565 2070   and a .. tree p
-000177d0: 6572 2066 616d 2061 6e64 2066 6f72 2074  er fam and for t
-000177e0: 6865 2063 6f6e 6361 7465 6e61 7465 6420  he concatenated 
-000177f0: 616c 6967 6e6d 656e 742e 0a0a 6060 600a  alignment...```.
-00017800: 7767 6420 6b73 6420 6461 7461 2f6b 7374  wgd ksd data/kst
-00017810: 7265 655f 6461 7461 2f66 616d 2e74 7376  ree_data/fam.tsv
-00017820: 2064 6174 612f 6b73 7472 6565 5f64 6174   data/kstree_dat
-00017830: 612f 4163 6f72 7573 5f74 6174 6172 696e  a/Acorus_tatarin
-00017840: 6f77 6969 2064 6174 612f 6b73 7472 6565  owii data/kstree
-00017850: 5f64 6174 612f 416d 626f 7265 6c6c 615f  _data/Amborella_
-00017860: 7472 6963 686f 706f 6461 2064 6174 612f  trichopoda data/
-00017870: 6b73 7472 6565 5f64 6174 612f 4171 7569  kstree_data/Aqui
-00017880: 6c65 6769 615f 636f 6572 756c 6561 2064  legia_coerulea d
-00017890: 6174 612f 6b73 7472 6565 5f64 6174 612f  ata/kstree_data/
-000178a0: 4172 6973 746f 6c6f 6368 6961 5f66 696d  Aristolochia_fim
-000178b0: 6272 6961 7461 2064 6174 612f 6b73 7472  briata data/kstr
-000178c0: 6565 5f64 6174 612f 4379 6361 735f 7061  ee_data/Cycas_pa
-000178d0: 6e7a 6869 6875 6165 6e73 6920 2d2d 6b73  nzhihuaensi --ks
-000178e0: 7472 6565 202d 2d73 7065 6369 6573 7472  tree --speciestr
-000178f0: 6565 2064 6174 612f 6b73 7472 6565 5f64  ee data/kstree_d
-00017900: 6174 612f 7370 6563 6965 735f 7472 6565  ata/species_tree
-00017910: 312e 6e77 202d 2d6f 6e6c 7963 6f6e 6361  1.nw --onlyconca
-00017920: 746b 7374 7265 6520 2d6f 2077 6764 5f6b  tkstree -o wgd_k
-00017930: 7374 7265 655f 746f 706f 6c6f 6779 310a  stree_topology1.
-00017940: 6060 600a 0a21 5b5d 2864 6174 612f 6b73  ```..![](data/ks
-00017950: 7472 6565 5f72 6573 756c 7473 2f6b 7374  tree_results/kst
-00017960: 7265 652e 7376 6729 0a0a 4162 6f76 6520  ree.svg)..Above 
-00017970: 7765 2075 7365 6420 7468 7265 6520 616c  we used three al
-00017980: 7465 726e 6174 6976 6520 746f 706f 6c6f  ternative topolo
-00017990: 6769 6573 2074 6f20 696e 6665 7220 7468  gies to infer th
-000179a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000179b0: 3e20 7472 6565 2077 6869 6368 206c 6564  > tree which led
-000179c0: 2074 6f20 6469 6666 6572 656e 7420 6272   to different br
-000179d0: 616e 6368 206c 656e 6774 6820 6573 7469  anch length esti
-000179e0: 6d61 7469 6f6e 2e20 4e6f 7465 2074 6861  mation. Note tha
-000179f0: 7420 7468 6520 6661 6d69 6c69 6573 2077  t the families w
-00017a00: 6520 7573 6564 2077 6572 6520 6f6e 6c79  e used were only
-00017a10: 2074 776f 2067 6c6f 6261 6c20 4d52 4248   two global MRBH
-00017a20: 2066 616d 696c 6965 7320 666f 7220 7468   families for th
-00017a30: 6520 7075 7270 6f73 6520 6f66 2069 6c6c  e purpose of ill
-00017a40: 7573 7472 6174 696f 6e2e 2054 6f20 6163  ustration. To ac
-00017a50: 7175 6972 6520 616e 2061 6363 7572 6174  quire an accurat
-00017a60: 6520 7072 6f66 696c 6520 6f66 2074 6865  e profile of the
-00017a70: 2073 7562 7374 6974 7574 696f 6e20 7261   substitution ra
-00017a80: 7465 2076 6172 6961 7469 6f6e 2c20 6f72  te variation, or
-00017a90: 7468 6f6c 6f67 7565 7320 6174 2074 6865  thologues at the
-00017aa0: 2077 686f 6c65 2067 656e 6f6d 6520 7363   whole genome sc
-00017ab0: 616c 6520 7368 6f75 6c64 2062 6520 7573  ale should be us
-00017ac0: 6564 2e0a 0a23 2320 4369 7461 7469 6f6e  ed...## Citation
-00017ad0: 0a20 0a50 6c65 6173 6520 6369 7465 2075  . .Please cite u
-00017ae0: 7320 6174 2068 7474 7073 3a2f 2f64 6f69  s at https://doi
-00017af0: 2e6f 7267 2f31 302e 3130 3037 2f39 3738  .org/10.1007/978
-00017b00: 2d31 2d30 3731 362d 3235 3631 2d33 5f31  -1-0716-2561-3_1
-00017b10: 2061 6e64 2068 7474 7073 3a2f 2f64 6f69   and https://doi
-00017b20: 2e6f 7267 2f31 302e 3130 3933 2f62 696f  .org/10.1093/bio
-00017b30: 696e 666f 726d 6174 6963 732f 6274 6165  informatics/btae
-00017b40: 3237 322e 0a0a 6060 600a 4865 6e67 6368  272...```.Hengch
-00017b50: 6920 4368 656e 2c20 4172 7468 7572 205a  i Chen, Arthur Z
-00017b60: 7761 656e 6570 6f65 6c20 2832 3032 3329  waenepoel (2023)
-00017b70: 2e20 496e 6665 7265 6e63 6520 6f66 2041  . Inference of A
-00017b80: 6e63 6965 6e74 2050 6f6c 7970 6c6f 6964  ncient Polyploid
-00017b90: 7920 6672 6f6d 2047 656e 6f6d 6963 2044  y from Genomic D
-00017ba0: 6174 612e 2049 6e3a 2056 616e 2064 6520  ata. In: Van de 
-00017bb0: 5065 6572 2c20 592e 2028 6564 7329 2050  Peer, Y. (eds) P
-00017bc0: 6f6c 7970 6c6f 6964 792e 204d 6574 686f  olyploidy. Metho
-00017bd0: 6473 2069 6e20 4d6f 6c65 6375 6c61 7220  ds in Molecular 
-00017be0: 4269 6f6c 6f67 792c 2076 6f6c 2032 3534  Biology, vol 254
-00017bf0: 352e 2048 756d 616e 612c 204e 6577 2059  5. Humana, New Y
-00017c00: 6f72 6b2c 204e 592e 2068 7474 7073 3a2f  ork, NY. https:/
-00017c10: 2f64 6f69 2e6f 7267 2f31 302e 3130 3037  /doi.org/10.1007
-00017c20: 2f39 3738 2d31 2d30 3731 362d 3235 3631  /978-1-0716-2561
-00017c30: 2d33 5f31 0a48 656e 6763 6869 2043 6865  -3_1.Hengchi Che
-00017c40: 6e2c 2041 7274 6875 7220 5a77 6165 6e65  n, Arthur Zwaene
-00017c50: 706f 656c 2c20 5976 6573 2056 616e 2064  poel, Yves Van d
-00017c60: 6520 5065 6572 2028 3230 3234 292e 2077  e Peer (2024). w
-00017c70: 6764 2076 323a 2061 2073 7569 7465 206f  gd v2: a suite o
-00017c80: 6620 746f 6f6c 7320 746f 2075 6e63 6f76  f tools to uncov
-00017c90: 6572 2061 6e64 2064 6174 6520 616e 6369  er and date anci
-00017ca0: 656e 7420 706f 6c79 706c 6f69 6479 2061  ent polyploidy a
-00017cb0: 6e64 2077 686f 6c65 2d67 656e 6f6d 6520  nd whole-genome 
-00017cc0: 6475 706c 6963 6174 696f 6e2e 2042 696f  duplication. Bio
-00017cd0: 696e 666f 726d 6174 6963 732c 2068 7474  informatics, htt
-00017ce0: 7073 3a2f 2f64 6f69 2e6f 7267 2f31 302e  ps://doi.org/10.
-00017cf0: 3130 3933 2f62 696f 696e 666f 726d 6174  1093/bioinformat
-00017d00: 6963 732f 6274 6165 3237 320a 6060 600a  ics/btae272.```.
-00017d10: 0a46 6f72 2063 6974 6174 696f 6e20 6f66  .For citation of
-00017d20: 2074 6865 2074 6f6f 6c73 2075 7365 6420   the tools used 
-00017d30: 696e 2077 6764 2c20 706c 6561 7365 2063  in wgd, please c
-00017d40: 6f6e 7375 6c74 2074 6865 2064 6f63 756d  onsult the docum
-00017d50: 656e 7461 7469 6f6e 2061 740a 6874 7470  entation at.http
-00017d60: 733a 2f2f 7767 6476 322e 7265 6164 7468  s://wgdv2.readth
-00017d70: 6564 6f63 732e 696f 2f65 6e2f 6c61 7465  edocs.io/en/late
-00017d80: 7374 2f63 6974 6174 696f 6e2e 6874 6d6c  st/citation.html
-00017d90: 2e0a 0a                                  ...
+000004e0: 6c61 7469 6f6e 2920 7c0a 5b2a 2a50 6970  lation) |.[**Pip
+000004f0: 656c 696e 6573 2a2a 5d28 2370 6970 656c  elines**](#pipel
+00000500: 696e 6573 2920 7c0a 5b2a 2a50 6172 616d  ines) |.[**Param
+00000510: 6574 6572 732a 2a5d 2823 7061 7261 6d65  eters**](#parame
+00000520: 7465 7273 2920 7c20 0a5b 2a2a 5573 6167  ters) | .[**Usag
+00000530: 652a 2a5d 2823 7573 6167 6529 207c 200a  e**](#usage) | .
+00000540: 5b2a 2a49 6c6c 7573 7472 6174 696f 6e2a  [**Illustration*
+00000550: 2a5d 2823 696c 6c75 7374 7261 7469 6f6e  *](#illustration
+00000560: 2920 7c0a 5b2a 2a44 6f63 756d 656e 7461  ) |.[**Documenta
+00000570: 7469 6f6e 2a2a 5d28 6874 7470 733a 2f2f  tion**](https://
+00000580: 7767 6476 322e 7265 6164 7468 6564 6f63  wgdv2.readthedoc
+00000590: 732e 696f 2f65 6e2f 6c61 7465 7374 2f3f  s.io/en/latest/?
+000005a0: 6261 6467 653d 6c61 7465 7374 2920 7c0a  badge=latest) |.
+000005b0: 5b2a 2a43 6974 6174 696f 6e2a 2a5d 2823  [**Citation**](#
+000005c0: 6369 7461 7469 6f6e 290a 0a3c 2f64 6976  citation)..</div
+000005d0: 3e0a 0a60 7767 6420 7632 6020 6973 2061  >..`wgd v2` is a
+000005e0: 2070 7974 686f 6e20 7061 636b 6167 6520   python package 
+000005f0: 7570 6772 6164 6564 2066 726f 6d20 7468  upgraded from th
+00000600: 6520 6f72 6967 696e 616c 2060 7767 6460  e original `wgd`
+00000610: 2070 6163 6b61 6765 2061 696d 696e 6720   package aiming 
+00000620: 666f 7220 7468 6520 696e 6665 7265 6e63  for the inferenc
+00000630: 6520 616e 6420 7469 6d69 6e67 206f 6620  e and timing of 
+00000640: 616e 6369 656e 7420 7768 6f6c 652d 6765  ancient whole-ge
+00000650: 6e6f 6d65 2064 7570 6c69 6361 7469 6f6e  nome duplication
+00000660: 2028 5747 4429 2065 7665 6e74 732e 2046   (WGD) events. F
+00000670: 6f72 2074 6865 2070 726f 706f 7365 206f  or the propose o
+00000680: 6620 696c 6c75 7374 7261 7469 6e67 2074  f illustrating t
+00000690: 6865 2070 7269 6e63 6970 6c65 2061 6e64  he principle and
+000006a0: 2075 7361 6765 206f 6620 6077 6764 2076   usage of `wgd v
+000006b0: 3260 2c20 7765 2063 6f6d 7069 6c65 6420  2`, we compiled 
+000006c0: 7468 6973 2064 6f63 756d 656e 7461 7469  this documentati
+000006d0: 6f6e 2e20 4265 6c6f 7720 7765 2066 6972  on. Below we fir
+000006e0: 7374 2067 6176 6520 616e 2069 6e74 726f  st gave an intro
+000006f0: 6475 6374 696f 6e20 6f76 6572 2074 6865  duction over the
+00000700: 2073 636f 7065 2061 6e64 206d 6563 6861   scope and mecha
+00000710: 6e69 736d 206f 6620 6077 6764 2076 3260  nism of `wgd v2`
+00000720: 2061 6e64 2074 6865 6e20 7468 6520 7072   and then the pr
+00000730: 6163 7469 6361 6c20 696e 666f 726d 6174  actical informat
+00000740: 696f 6e20 7065 7274 6169 6e69 6e67 2074  ion pertaining t
+00000750: 6f20 7468 6520 696e 7374 616c 6c61 7469  o the installati
+00000760: 6f6e 2061 6e64 2075 7361 6765 2e20 416e  on and usage. An
+00000770: 2065 7861 6d70 6c61 7220 776f 726b 666c   examplar workfl
+00000780: 6f77 2069 7320 7072 6f76 6964 6564 2069  ow is provided i
+00000790: 6e20 7468 6520 7475 746f 7269 616c 2073  n the tutorial s
+000007a0: 6563 7469 6f6e 206f 6e20 686f 7720 746f  ection on how to
+000007b0: 2073 6565 6b20 6576 6964 656e 6365 2066   seek evidence f
+000007c0: 6f72 2061 2070 7574 6174 6976 6520 5747  or a putative WG
+000007d0: 4420 6576 656e 7420 616e 6420 7065 7266  D event and perf
+000007e0: 6f72 6d20 7072 6f70 6572 2074 696d 696e  orm proper timin
+000007f0: 6720 7769 7468 2061 2066 7265 7368 6c79  g with a freshly
+00000800: 206f 6274 6169 6e65 6420 6765 6e6f 6d65   obtained genome
+00000810: 2061 7373 656d 626c 7920 696e 2068 616e   assembly in han
+00000820: 642e 2046 6f72 2074 686f 7365 2077 686f  d. For those who
+00000830: 2061 7265 2069 6e74 6572 6573 7465 6420   are interested 
+00000840: 696e 206d 6f72 6520 7468 656f 7265 7469  in more theoreti
+00000850: 6361 6c20 6465 7461 696c 732c 2077 6520  cal details, we 
+00000860: 7265 636f 6d6d 656e 6420 7475 726e 696e  recommend turnin
+00000870: 6720 746f 206f 7572 2070 6170 6572 2061  g to our paper a
+00000880: 6e64 2062 6f6f 6b20 6368 6170 7465 7220  nd book chapter 
+00000890: 666f 7220 6d6f 7265 2064 6574 6169 6c65  for more detaile
+000008a0: 6420 6465 7363 7269 7074 696f 6e20 616e  d description an
+000008b0: 6420 696e 7369 6768 7466 756c 2064 6973  d insightful dis
+000008c0: 6375 7373 696f 6e73 2e20 5468 6520 6b65  cussions. The ke
+000008d0: 7920 696d 7072 6f76 6564 2066 6561 7475  y improved featu
+000008e0: 7265 7320 6f66 2060 7767 6420 7632 6020  res of `wgd v2` 
+000008f0: 6172 6520 6465 6d6f 6e73 7472 6174 6564  are demonstrated
+00000900: 2069 6e20 6f75 7220 6c61 7465 7374 206d   in our latest m
+00000910: 616e 7573 6372 6970 7420 6f6e 205b 4269  anuscript on [Bi
+00000920: 6f69 6e66 6f72 6d61 7469 6373 5d28 6874  oinformatics](ht
+00000930: 7470 733a 2f2f 646f 692e 6f72 672f 3130  tps://doi.org/10
+00000940: 2e31 3039 332f 6269 6f69 6e66 6f72 6d61  .1093/bioinforma
+00000950: 7469 6373 2f62 7461 6532 3732 292e 2049  tics/btae272). I
+00000960: 6620 796f 7520 7573 6520 6077 6764 2076  f you use `wgd v
+00000970: 3260 2069 6e20 796f 7572 2072 6573 6561  2` in your resea
+00000980: 7263 682c 2070 6c65 6173 6520 6369 7465  rch, please cite
+00000990: 2075 7320 6173 2073 7567 6765 7374 6564   us as suggested
+000009a0: 2069 6e20 5b43 6974 6174 696f 6e5d 2823   in [Citation](#
+000009b0: 6369 7461 7469 6f6e 2920 7365 6374 696f  citation) sectio
+000009c0: 6e2e 0a0a 2323 2049 6e74 726f 6475 6374  n...## Introduct
+000009d0: 696f 6e0a 0a50 6f6c 7970 6c6f 6964 697a  ion..Polyploidiz
+000009e0: 6174 696f 6e73 2c20 7468 6520 6576 6f6c  ations, the evol
+000009f0: 7574 696f 6e61 7279 2070 726f 6365 7373  utionary process
+00000a00: 2074 6861 7420 7468 6520 656e 7469 7265   that the entire
+00000a10: 2067 656e 6f6d 6520 6f66 2061 6e20 6f72   genome of an or
+00000a20: 6761 6e69 736d 2069 7320 6475 706c 6963  ganism is duplic
+00000a30: 6174 6564 2c20 616c 736f 206e 616d 6564  ated, also named
+00000a40: 2061 7320 7768 6f6c 652d 6765 6e6f 6d65   as whole-genome
+00000a50: 2064 7570 6c69 6361 7469 6f6e 7320 2857   duplications (W
+00000a60: 4744 7329 2c20 6f63 6375 7220 7265 6375  GDs), occur recu
+00000a70: 7272 656e 746c 7920 6163 726f 7373 2074  rrently across t
+00000a80: 6865 2074 7265 6520 6f66 206c 6966 652e  he tree of life.
+00000a90: 2054 6865 7265 2061 7265 2074 776f 206d   There are two m
+00000aa0: 6f64 6573 206f 6620 706f 6c79 706c 6f69  odes of polyploi
+00000ab0: 6469 7a61 7469 6f6e 732c 2061 7574 6f70  dizations, autop
+00000ac0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e73  olyploidizations
+00000ad0: 2061 6e64 2061 6c6c 6f70 6f6c 7970 6c6f   and allopolyplo
+00000ae0: 6964 697a 6174 696f 6e73 2e20 4175 746f  idizations. Auto
+00000af0: 706f 6c79 706c 6f69 6469 7a61 7469 6f6e  polyploidization
+00000b00: 7320 6172 6520 7468 6520 6475 706c 6963  s are the duplic
+00000b10: 6174 696f 6e20 6f66 2074 6865 2073 616d  ation of the sam
+00000b20: 6520 6765 6e6f 6d65 2c20 7265 7375 6c74  e genome, result
+00000b30: 696e 6720 696e 2074 776f 2069 6465 6e74  ing in two ident
+00000b40: 6963 616c 2073 7562 6765 6e6f 6d65 7320  ical subgenomes 
+00000b50: 6174 2074 6865 2074 696d 6520 6974 2065  at the time it e
+00000b60: 6d65 7267 6564 2e20 5768 696c 6520 7468  merged. While th
+00000b70: 6520 616c 6c6f 706f 6c79 706c 6f69 6469  e allopolyploidi
+00000b80: 7a61 7469 6f6e 7320 6172 6520 6e6f 726d  zations are norm
+00000b90: 616c 6c79 2061 6368 6965 7665 6420 696e  ally achieved in
+00000ba0: 2074 776f 2073 7465 7073 2c20 6669 7273   two steps, firs
+00000bb0: 7420 7468 6520 6879 6272 6964 697a 6174  t the hybridizat
+00000bc0: 696f 6e20 6265 7477 6565 6e20 7477 6f20  ion between two 
+00000bd0: 6469 6666 6572 656e 7420 7370 6563 6965  different specie
+00000be0: 732c 2072 6573 756c 7469 6e67 2069 6e20  s, resulting in 
+00000bf0: 7468 6520 6172 6973 696e 6720 6f66 2074  the arising of t
+00000c00: 7261 6e73 6965 6e74 2068 6f6d 6f70 6c6f  ransient homoplo
+00000c10: 6964 792c 7365 636f 6e64 2074 6865 2064  idy,second the d
+00000c20: 7570 6c69 6361 7469 6f6e 206f 6620 7468  uplication of th
+00000c30: 6520 686f 6d6f 706c 6f69 6479 2c20 7265  e homoploidy, re
+00000c40: 7375 6c74 696e 6720 696e 2074 6865 2065  sulting in the e
+00000c50: 6d65 7267 656e 6365 206f 6620 616c 6c6f  mergence of allo
+00000c60: 706f 6c79 706c 6f69 6479 2e20 4475 6520  polyploidy. Due 
+00000c70: 746f 2074 6865 2075 6e73 7461 6269 6c69  to the unstabili
+00000c80: 7479 2061 6e64 2075 6e62 616c 616e 6365  ty and unbalance
+00000c90: 6420 7465 7472 6173 6f6d 6963 2069 6e68  d tetrasomic inh
+00000ca0: 6572 6974 616e 6365 2c20 666f 7220 696e  eritance, for in
+00000cb0: 7374 616e 6365 2074 6865 206e 7563 6c65  stance the nucle
+00000cc0: 6172 2d63 7974 6f70 6c61 736d 6963 2069  ar-cytoplasmic i
+00000cd0: 6e63 6f6d 7061 7469 6269 6c69 7479 2c20  ncompatibility, 
+00000ce0: 7468 6520 706f 6c79 706c 6f69 6479 2067  the polyploidy g
+00000cf0: 656e 6f6d 6520 7769 6c6c 2074 6865 6e20  enome will then 
+00000d00: 6578 7065 7269 656e 6365 2061 2070 726f  experience a pro
+00000d10: 6365 7373 2063 616c 6c65 6420 6469 706c  cess called dipl
+00000d20: 6f69 6469 7a61 7469 6f6e 2c20 616c 736f  oidization, also
+00000d30: 206e 616d 6564 2061 7320 6672 6163 7469   named as fracti
+00000d40: 6f6e 6174 696f 6e2c 2064 7572 696e 6720  onation, during 
+00000d50: 7768 6963 6820 6120 6c61 7267 6520 706f  which a large po
+00000d60: 7274 696f 6e20 6f66 2067 656e 6520 6475  rtion of gene du
+00000d70: 706c 6963 6174 6573 2077 696c 6c20 6765  plicates will ge
+00000d80: 7420 6c6f 7374 2061 6e64 206f 6e6c 7920  t lost and only 
+00000d90: 6120 6672 6163 7469 6f6e 2063 616e 2062  a fraction can b
+00000da0: 6520 7265 7461 696e 6564 2e20 5468 6520  e retained. The 
+00000db0: 7472 6163 6573 206f 6620 706f 6c79 706c  traces of polypl
+00000dc0: 6f69 6469 7a61 7469 6f6e 7320 6361 6e20  oidizations can 
+00000dd0: 6265 2074 6875 7320 756e 6561 7274 6865  be thus unearthe
+00000de0: 6420 6672 6f6d 2074 6865 7365 2072 6574  d from these ret
+00000df0: 6169 6e65 6420 6765 6e65 2064 7570 6c69  ained gene dupli
+00000e00: 6361 7465 732e 2054 6872 6565 2061 7070  cates. Three app
+00000e10: 726f 6163 6865 7320 6261 7365 6420 6f6e  roaches based on
+00000e20: 2067 656e 6520 6475 706c 6963 6174 6573   gene duplicates
+00000e30: 2c20 6e61 6d65 6c79 2c20 2a4b 2a3c 7375  , namely, *K*<su
+00000e40: 623e 533c 2f73 7562 3e20 6d65 7468 6f64  b>S</sub> method
+00000e50: 2c20 6765 6e65 2074 7265 6520 2d20 7370  , gene tree - sp
+00000e60: 6563 6965 7320 7472 6565 2072 6563 6f6e  ecies tree recon
+00000e70: 6369 6c69 6174 696f 6e20 6d65 7468 6f64  ciliation method
+00000e80: 2061 6e64 2073 796e 7465 6e79 206d 6574   and synteny met
+00000e90: 686f 642c 2061 7265 2063 6f6d 6d6f 6e6c  hod, are commonl
+00000ea0: 7920 7573 6564 2069 6e20 6465 7465 6374  y used in detect
+00000eb0: 696e 6720 6576 6964 656e 6365 2066 6f72  ing evidence for
+00000ec0: 2057 4744 732e 2054 6865 2067 656e 6520   WGDs. The gene 
+00000ed0: 7472 6565 202d 2073 7065 6369 6573 2074  tree - species t
+00000ee0: 7265 6520 7265 636f 6e63 696c 6961 7469  ree reconciliati
+00000ef0: 6f6e 206d 6574 686f 6420 6973 206e 6f74  on method is not
+00000f00: 2077 6974 6869 6e20 7468 6520 7363 6f70   within the scop
+00000f10: 6520 6f66 2060 7767 6420 7632 602c 2062  e of `wgd v2`, b
+00000f20: 7574 2077 6520 6b69 6e64 6c79 2072 6566  ut we kindly ref
+00000f30: 6572 2072 6561 6465 7273 2077 686f 2061  er readers who a
+00000f40: 7265 2069 6e74 6572 6573 7465 6420 746f  re interested to
+00000f50: 2074 6865 2070 6879 6c6f 6765 6e6f 6d69   the phylogenomi
+00000f60: 6320 7072 6f67 7261 6d20 6465 7665 6c6f  c program develo
+00000f70: 7065 6420 6279 2041 7274 6875 7220 5a77  ped by Arthur Zw
+00000f80: 6165 6e65 706f 656c 206e 616d 6564 205b  aenepoel named [
+00000f90: 5748 414c 455d 2868 7474 7073 3a2f 2f67  WHALE](https://g
+00000fa0: 6974 6875 622e 636f 6d2f 6172 7a77 612f  ithub.com/arzwa/
+00000fb0: 5768 616c 652e 6a6c 2920 616e 6420 7468  Whale.jl) and th
+00000fc0: 6520 6173 736f 6369 6174 6564 205b 7061  e associated [pa
+00000fd0: 7065 725d 2868 7474 7073 3a2f 2f64 6f69  per](https://doi
+00000fe0: 2e6f 7267 2f31 302e 3130 3933 2f6d 6f6c  .org/10.1093/mol
+00000ff0: 6265 762f 6d73 7a30 3838 2920 666f 7220  bev/msz088) for 
+00001000: 6d6f 7265 2074 6563 686e 6963 616c 2061  more technical a
+00001010: 6e64 2074 6865 6f72 6574 6963 616c 2064  nd theoretical d
+00001020: 6574 6169 6c73 2e0a 0a54 6865 202a 4b2a  etails...The *K*
+00001030: 3c73 7562 3e53 3c2f 7375 623e 206d 6574  <sub>S</sub> met
+00001040: 686f 6420 6973 2065 7374 6162 6c69 7368  hod is establish
+00001050: 6564 206f 6e20 6120 6d6f 6465 6c20 6f66  ed on a model of
+00001060: 2067 656e 6520 6661 6d69 6c79 2065 766f   gene family evo
+00001070: 6c75 7469 6f6e 2074 6861 7420 6561 6368  lution that each
+00001080: 2067 656e 6520 6661 6d69 6c79 2069 7320   gene family is 
+00001090: 616c 6c6f 7765 6420 746f 2065 766f 6c76  allowed to evolv
+000010a0: 6520 7669 6120 6765 6e65 2064 7570 6c69  e via gene dupli
+000010b0: 6361 7469 6f6e 2061 6e64 206c 6f73 732e  cation and loss.
+000010c0: 204e 6f74 6520 7468 6174 2074 6865 2067   Note that the g
+000010d0: 656e 6520 6661 6d69 6c79 2068 6572 6520  ene family here 
+000010e0: 6973 2061 7373 756d 6564 2074 6f20 6265  is assumed to be
+000010f0: 2074 6865 2063 6c75 7374 6572 206f 6620   the cluster of 
+00001100: 616c 6c20 6765 6e65 7320 6465 7363 656e  all genes descen
+00001110: 6465 6420 6672 6f6d 2061 6e20 616e 6365  ded from an ance
+00001120: 7374 7261 6c20 6765 6e65 2069 6e20 6120  stral gene in a 
+00001130: 7369 6e67 6c65 2067 656e 6f6d 652e 2052  single genome. R
+00001140: 6563 6f76 6572 696e 6720 7468 6520 6765  ecovering the ge
+00001150: 6e65 2074 7265 6520 6f66 2073 7563 6820  ne tree of such 
+00001160: 6765 6e65 2066 616d 696c 7920 696e 666f  gene family info
+00001170: 726d 7320 7468 6520 7469 6d69 6e67 2c20  rms the timing, 
+00001180: 7363 696c 6963 6574 2074 6865 2061 6765  scilicet the age
+00001190: 2c20 6f66 2067 656e 6520 6475 706c 6963  , of gene duplic
+000011a0: 6174 696f 6e20 6576 656e 7473 2e20 5468  ation events. Th
+000011b0: 6520 6167 6520 7265 6665 7265 6420 6865  e age refered he
+000011c0: 7265 2c20 6973 206e 6f74 2069 6e20 7265  re, is not in re
+000011d0: 616c 2067 656f 6c6f 6769 6361 6c20 7469  al geological ti
+000011e0: 6d65 2c20 6275 7420 696e 2074 6865 2075  me, but in the u
+000011f0: 6e69 7420 6f66 2065 766f 6c75 7469 6f6e  nit of evolution
+00001200: 6172 7920 6469 7374 616e 6365 2c20 692e  ary distance, i.
+00001210: 652e 2c20 7468 6520 6e75 6d62 6572 206f  e., the number o
+00001220: 6620 7375 6273 7469 7475 7469 6f6e 7320  f substitutions 
+00001230: 7065 7220 7369 7465 2e20 5768 656e 2074  per site. When t
+00001240: 6865 2065 766f 6c75 7469 6f6e 6172 7920  he evolutionary 
+00001250: 7261 7465 2072 656d 6169 6e73 2061 7070  rate remains app
+00001260: 726f 7869 6d61 7465 6c79 2063 6f6e 7374  roximately const
+00001270: 616e 742c 2074 6865 2065 766f 6c75 7469  ant, the evoluti
+00001280: 6f6e 6172 7920 6469 7374 616e 6365 2069  onary distance i
+00001290: 7320 7468 656e 2073 7570 706f 7365 6420  s then supposed 
+000012a0: 746f 2062 6520 7072 6f70 6f72 7469 6f6e  to be proportion
+000012b0: 616c 2074 6f20 7468 6520 7265 616c 2065  al to the real e
+000012c0: 766f 6c75 7469 6f6e 6172 7920 7469 6d65  volutionary time
+000012d0: 2e20 5468 6520 7379 6e6f 6e79 6d6f 7573  . The synonymous
+000012e0: 2064 6973 7461 6e63 6520 2a4b 2a3c 7375   distance *K*<su
+000012f0: 623e 533c 2f73 7562 3e2c 2074 6865 206e  b>S</sub>, the n
+00001300: 756d 6265 7220 6f66 2073 796e 6f6e 796d  umber of synonym
+00001310: 6f75 7320 7375 6273 7469 7475 7469 6f6e  ous substitution
+00001320: 7320 7065 7220 7379 6e6f 6e79 6d6f 7573  s per synonymous
+00001330: 2073 6974 652c 2069 7320 7375 6368 2063   site, is such c
+00001340: 616e 6469 6461 7465 2074 6861 7420 7379  andidate that sy
+00001350: 6e6f 6e79 6d6f 7573 2073 7562 7374 6974  nonymous substit
+00001360: 7574 696f 6e73 2077 6f75 6c64 206e 6f74  utions would not
+00001370: 2069 6e63 7572 2074 6865 2063 6861 6e67   incur the chang
+00001380: 6520 6f66 2061 6d69 6e6f 2061 6369 6420  e of amino acid 
+00001390: 616e 6420 6172 6520 7468 7573 2072 6567  and are thus reg
+000013a0: 6172 6465 6420 6173 206e 6575 7472 616c  arded as neutral
+000013b0: 2c20 7768 6963 6820 6163 636f 7264 696e  , which accordin
+000013c0: 6720 746f 2074 6865 206e 6575 7472 616c  g to the neutral
+000013d0: 2074 6865 6f72 7920 7368 6f75 6c64 206f   theory should o
+000013e0: 6363 7572 2069 6e20 636f 6e73 7461 6e74  ccur in constant
+000013f0: 2072 6174 652e 2047 6976 656e 2061 206d   rate. Given a m
+00001400: 6f64 656c 206f 6620 6765 6e65 2066 616d  odel of gene fam
+00001410: 696c 7920 7468 6174 2061 6c6c 6f77 7320  ily that allows 
+00001420: 7468 6520 6765 6e65 2074 6f20 6475 706c  the gene to dupl
+00001430: 6963 6174 6520 616e 6420 6765 7420 6c6f  icate and get lo
+00001440: 7374 2069 6e20 6120 6669 7865 6420 7261  st in a fixed ra
+00001450: 7465 2c20 6f6e 6520 6361 6e20 6465 7269  te, one can deri
+00001460: 7665 2074 6861 7420 7468 6520 7072 6f62  ve that the prob
+00001470: 6162 696c 6974 7920 6465 6e73 6974 7920  ability density 
+00001480: 6675 6e63 7469 6f6e 206f 6620 7468 6520  function of the 
+00001490: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000014a0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+000014b0: 206f 6620 7265 7461 696e 6564 2067 656e   of retained gen
+000014c0: 6520 6475 706c 6963 6174 6573 2069 7320  e duplicates is 
+000014d0: 6120 7175 6173 692d 6578 706f 6e65 6e74  a quasi-exponent
+000014e0: 6961 6c20 6675 6e63 7469 6f6e 2074 6861  ial function tha
+000014f0: 7420 6d6f 7374 2072 6574 6169 6e65 6420  t most retained 
+00001500: 6765 6e65 2064 7570 6c69 6361 7465 7320  gene duplicates 
+00001510: 6172 6520 7265 6365 6e74 6c79 2062 6f72  are recently bor
+00001520: 6e65 6420 7769 7468 207e 3020 6167 6520  ned with ~0 age 
+00001530: 7768 696c 6520 6173 2074 6865 2061 6765  while as the age
+00001540: 2067 6f69 6e67 206f 6c64 6572 2074 6865   going older the
+00001550: 2061 7373 6f63 6961 7465 6420 6e75 6d62   associated numb
+00001560: 6572 206f 6620 7265 7461 696e 6564 2067  er of retained g
+00001570: 656e 6520 6475 706c 6963 6174 6573 2064  ene duplicates d
+00001580: 6563 6179 2071 7561 7369 2d65 7870 6f6e  ecay quasi-expon
+00001590: 656e 7469 616c 6c79 2e20 5468 6572 6566  entially. Theref
+000015a0: 6f72 652c 2074 6865 206f 6363 7572 616e  ore, the occuran
+000015b0: 6365 206f 6620 6c61 7267 652d 7363 616c  ce of large-scal
+000015c0: 6520 6765 6e65 2064 7570 6c69 6361 7469  e gene duplicati
+000015d0: 6f6e 2065 7665 6e74 732c 2066 6f72 2069  on events, for i
+000015e0: 6e73 7461 6e65 2057 4744 732c 2077 6974  nstane WGDs, wit
+000015f0: 6820 7661 7269 6564 2072 6574 656e 7469  h varied retenti
+00001600: 6f6e 2072 6174 652c 2077 696c 6c20 6c65  on rate, will le
+00001610: 6176 6520 616e 2061 6765 2070 6561 6b20  ave an age peak 
+00001620: 6672 6f6d 2074 6865 2062 7572 7374 206f  from the burst o
+00001630: 6620 6765 6e65 2064 7570 6c69 6361 7465  f gene duplicate
+00001640: 7320 696e 2061 2073 686f 7274 2074 696d  s in a short tim
+00001650: 652d 6672 616d 6520 7570 6f6e 2074 6865  e-frame upon the
+00001660: 2069 6e69 7469 616c 2061 6765 2064 6973   initial age dis
+00001670: 7472 6962 7574 696f 6e2c 2061 6e64 2063  tribution, and c
+00001680: 616e 2062 6520 756e 7665 696c 6564 2066  an be unveiled f
+00001690: 726f 6d20 6d69 7874 7572 6520 6d6f 6465  rom mixture mode
+000016a0: 6c69 6e67 2061 6e61 6c79 7369 732e 2048  ling analysis. H
+000016b0: 6f77 6576 6572 2c20 5747 4473 2069 6465  owever, WGDs ide
+000016c0: 6e74 6966 6965 6420 6672 6f6d 2074 6865  ntified from the
+000016d0: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
+000016e0: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+000016f0: 6469 7374 7269 6275 7469 6f6e 7320 6361  distributions ca
+00001700: 6e20 6f6e 6c79 2069 6e66 6f72 6d20 7468  n only inform th
+00001710: 6520 5747 4420 7469 6d69 6e67 2069 6e20  e WGD timing in 
+00001720: 7468 6520 7469 6d65 2d73 6361 6c65 206f  the time-scale o
+00001730: 6620 7468 6174 2073 7065 6369 6669 6320  f that specific 
+00001740: 7370 6563 6965 732c 2077 6869 6368 2069  species, which i
+00001750: 7320 6e6f 7420 636f 6d70 6172 6162 6c65  s not comparable
+00001760: 2069 6e20 7468 6520 7068 796c 6f67 656e   in the phylogen
+00001770: 6574 6963 2063 6f6e 7465 7874 2e20 4f6e  etic context. On
+00001780: 6c79 2077 6974 6820 7468 6520 6f72 7468  ly with the orth
+00001790: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000017a0: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+000017b0: 7269 6275 7469 6f6e 732c 2077 6869 6368  ributions, which
+000017c0: 2063 6f6e 7665 7274 2074 6865 2065 7374   convert the est
+000017d0: 696d 6174 6564 2062 6f64 7920 6672 6f6d  imated body from
+000017e0: 2070 6172 616c 6f67 7565 7320 746f 206f   paralogues to o
+000017f0: 7274 686f 6c6f 6775 6573 2061 6e64 2069  rthologues and i
+00001800: 6e66 6f72 6d20 7468 6520 7265 6c61 7469  nform the relati
+00001810: 7665 2074 696d 696e 6720 6f66 2073 7065  ve timing of spe
+00001820: 6369 6174 696f 6e20 6576 656e 7473 2c20  ciation events, 
+00001830: 6361 6e20 7765 2064 6563 6970 6865 7220  can we decipher 
+00001840: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
+00001850: 2070 6c61 6365 6d65 6e74 206f 6620 5747   placement of WG
+00001860: 4473 2061 6674 6572 2070 726f 7065 7220  Ds after proper 
+00001870: 7261 7465 2063 6f72 7265 6374 696f 6e2e  rate correction.
+00001880: 2060 7767 6420 7632 6020 6973 2073 7563   `wgd v2` is suc
+00001890: 6820 7072 6f67 7261 6d20 7468 6174 2068  h program that h
+000018a0: 656c 7073 2075 7365 7273 2063 6f6e 7374  elps users const
+000018b0: 7275 6374 2070 6172 616c 6f67 6f75 7320  ruct paralogous 
+000018c0: 616e 6420 6f72 7468 6f6c 6f67 6f75 7320  and orthologous 
+000018d0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000018e0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+000018f0: 7320 616e 6420 7265 616c 697a 6520 626f  s and realize bo
+00001900: 7468 2074 6865 2069 6465 6e74 6966 6963  th the identific
+00001910: 6174 696f 6e20 616e 6420 706c 6163 656d  ation and placem
+00001920: 656e 7420 6f66 2057 4744 732e 0a0a 496e  ent of WGDs...In
+00001930: 2074 6865 2070 7265 6d69 7365 206f 6620   the premise of 
+00001940: 7068 796c 6f67 656e 6574 6963 616c 6c79  phylogenetically
+00001950: 206c 6f63 6174 6564 2057 4744 732c 2074   located WGDs, t
+00001960: 6865 2061 6273 6f6c 7574 6520 6167 6520  he absolute age 
+00001970: 2869 6e20 6765 6f6c 6f67 6963 616c 2074  (in geological t
+00001980: 696d 6529 206f 6620 5747 4473 2063 616e  ime) of WGDs can
+00001990: 2061 6c73 6f20 6265 2069 6e66 6572 7265   also be inferre
+000019a0: 6420 6672 6f6d 2074 686f 7365 2057 4744  d from those WGD
+000019b0: 2d72 6574 6169 6e65 6420 6765 6e65 2064  -retained gene d
+000019c0: 7570 6c69 6361 7465 732c 2061 6c74 686f  uplicates, altho
+000019d0: 7567 6820 7468 6572 6520 6861 7320 6265  ugh there has be
+000019e0: 656e 206e 6f20 6561 7379 206f 7220 7374  en no easy or st
+000019f0: 7261 6967 6874 666f 7277 6172 6420 7069  raightforward pi
+00001a00: 7065 6c69 6e65 2066 6f72 2074 6869 7320  peline for this 
+00001a10: 6a6f 6220 736f 2066 6172 2e20 496e 2060  job so far. In `
+00001a20: 7767 6420 7632 602c 2077 6520 6465 7665  wgd v2`, we deve
+00001a30: 6c6f 7065 6420 6120 6665 6173 6962 6c65  loped a feasible
+00001a40: 2069 6e74 6567 7261 7465 6420 7069 706c   integrated pipl
+00001a50: 656c 696e 6520 666f 7220 6162 736f 6c75  eline for absolu
+00001a60: 7465 2064 6174 696e 6720 6f66 2057 4744  te dating of WGD
+00001a70: 732e 2054 6865 2070 6970 656c 696e 6520  s. The pipeline 
+00001a80: 6361 6e20 6265 2072 6f75 6768 6c79 2064  can be roughly d
+00001a90: 6976 6964 6564 2069 6e74 6f20 7468 7265  ivided into thre
+00001aa0: 6520 6d61 696e 2073 7465 7073 2e20 3129  e main steps. 1)
+00001ab0: 2054 6865 2063 6f6e 7374 7275 6374 696f   The constructio
+00001ac0: 6e20 6f66 2061 6e63 686f 7220 2a4b 2a3c  n of anchor *K*<
+00001ad0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00001ae0: 7269 6275 7469 6f6e 2061 6e64 2074 6865  ribution and the
+00001af0: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
+00001b00: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
+00001b10: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
+00001b20: 6164 6f70 7465 6420 666f 7220 7068 796c  adopted for phyl
+00001b30: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
+00001b40: 2075 7369 6e67 2060 7767 6420 646d 6460   using `wgd dmd`
+00001b50: 2c20 6077 6764 206b 7364 602c 2060 7767  , `wgd ksd`, `wg
+00001b60: 6420 7379 6e60 2061 6e64 2060 7767 6420  d syn` and `wgd 
+00001b70: 7065 616b 602e 204e 6f74 6520 7468 6174  peak`. Note that
+00001b80: 2068 6572 6520 7765 206f 6e6c 7920 636f   here we only co
+00001b90: 6e73 6964 6572 2067 656e 6f6d 6520 6173  nsider genome as
+00001ba0: 7365 6d62 6c79 2062 6563 6175 7365 2066  sembly because f
+00001bb0: 6f72 2074 7261 6e73 6372 6970 746f 6d65  or transcriptome
+00001bc0: 2061 7373 656d 626c 7920 6974 2773 2069   assembly it's i
+00001bd0: 6d70 6f73 7369 626c 6520 746f 2064 6973  mpossible to dis
+00001be0: 7469 6e67 7569 7368 2057 4744 2d64 6572  tinguish WGD-der
+00001bf0: 6976 6564 2064 7570 6c69 6361 7465 7320  ived duplicates 
+00001c00: 6672 6f6d 2073 6d61 6c6c 2d73 6361 6c65  from small-scale
+00001c10: 2064 7570 6c69 6361 7469 6f6e 2d64 6572   duplication-der
+00001c20: 6976 6564 2064 7570 6c69 6361 7465 732c  ived duplicates,
+00001c30: 2077 6869 6368 2068 6170 7065 6e65 6420   which happened 
+00001c40: 696e 2061 2063 6f6e 7469 6e75 6f75 7320  in a continuous 
+00001c50: 7469 6d65 2d66 7261 6d65 2069 6e73 7465  time-frame inste
+00001c60: 6164 206f 6620 6f6e 6c79 2061 2073 6570  ad of only a sep
+00001c70: 6172 6174 6520 7368 6f72 7420 7469 6d65  arate short time
+00001c80: 2d66 7261 6d65 2061 6e64 2074 6875 7320  -frame and thus 
+00001c90: 7265 666c 6563 7473 2074 6865 2064 7572  reflects the dur
+00001ca0: 6174 696f 6e20 6f66 2074 6861 7420 6272  ation of that br
+00001cb0: 616e 6368 2072 6174 6865 7220 7468 616e  anch rather than
+00001cc0: 2074 6865 2074 696d 6520 6174 2077 6869   the time at whi
+00001cd0: 6368 2057 4744 206f 6363 7572 7265 642e  ch WGD occurred.
+00001ce0: 2032 2920 5468 6520 666f 726d 756c 6174   2) The formulat
+00001cf0: 696f 6e20 6f66 2061 2073 7461 7274 696e  ion of a startin
+00001d00: 6720 7472 6565 2075 7365 6420 696e 2074  g tree used in t
+00001d10: 6865 2070 6879 6c6f 6765 6e65 7469 6320  he phylogenetic 
+00001d20: 6461 7469 6e67 2c20 636f 6d70 6f73 6564  dating, composed
+00001d30: 206f 6620 6120 6665 7720 7370 6563 6965   of a few specie
+00001d40: 7320 616e 6420 616e 6e6f 7461 7465 6420  s and annotated 
+00001d50: 7769 7468 2066 6f73 7369 6c20 6361 6c69  with fossil cali
+00001d60: 6272 6174 696f 6e20 696e 666f 726d 6174  bration informat
+00001d70: 696f 6e2e 2054 6869 7320 7374 6570 2069  ion. This step i
+00001d80: 7320 6573 7365 6e74 6961 6c20 666f 7220  s essential for 
+00001d90: 7468 6520 7265 7375 6c74 206f 6620 6162  the result of ab
+00001da0: 736f 6c75 7465 2057 4744 2064 6174 696e  solute WGD datin
+00001db0: 6720 7468 6174 2077 6520 7375 6767 6573  g that we sugges
+00001dc0: 7420 7573 6572 7320 6f66 2074 616b 696e  t users of takin
+00001dd0: 6720 6772 6561 7420 6361 7574 696f 6e20  g great caution 
+00001de0: 746f 2061 7373 7572 6520 7468 6520 7472  to assure the tr
+00001df0: 6565 2074 6f70 6f6c 6f67 7920 616e 6420  ee topology and 
+00001e00: 7072 6f70 6572 2062 6f75 6e64 7320 666f  proper bounds fo
+00001e10: 7220 666f 7373 696c 2063 616c 6962 7261  r fossil calibra
+00001e20: 7469 6f6e 732e 2033 2920 5468 6520 636f  tions. 3) The co
+00001e30: 6e73 7472 7563 7469 6f6e 206f 6620 6f72  nstruction of or
+00001e40: 7468 6f67 726f 7570 7320 636f 6e73 6973  thogroups consis
+00001e50: 7469 6e67 206f 6620 636f 6c6c 696e 6561  ting of collinea
+00001e60: 7220 6475 706c 6963 6174 6573 206f 6620  r duplicates of 
+00001e70: 7468 6520 666f 6361 6c20 7370 6563 6965  the focal specie
+00001e80: 7320 616e 6420 7468 6569 7220 7265 6369  s and their reci
+00001e90: 7072 6f63 616c 2062 6573 7420 6869 7473  procal best hits
+00001ea0: 2028 5242 4873 2920 6167 6169 6e73 7420   (RBHs) against 
+00001eb0: 6f74 6865 7220 7370 6563 6965 7320 696e  other species in
+00001ec0: 2074 6865 2073 7461 7274 696e 6720 7472   the starting tr
+00001ed0: 6565 2061 6e64 2074 6865 2070 6879 6c6f  ee and the phylo
+00001ee0: 6765 6e65 7469 6320 6461 7469 6e67 2075  genetic dating u
+00001ef0: 7369 6e67 2061 206d 6f6c 6563 756c 6172  sing a molecular
+00001f00: 2064 6174 696e 6720 7072 6f67 7261 6d20   dating program 
+00001f10: 666f 7220 696e 7374 616e 6365 2060 6d63  for instance `mc
+00001f20: 6d63 7472 6565 602c 2076 6961 2070 726f  mctree`, via pro
+00001f30: 6772 616d 2060 7767 6420 646d 6460 2061  gram `wgd dmd` a
+00001f40: 6e64 2060 7767 6420 666f 6375 7360 2e20  nd `wgd focus`. 
+00001f50: 5765 2072 6563 6f6d 6d65 6e64 2074 6865  We recommend the
+00001f60: 2075 7361 6765 206f 6620 4261 7965 7369   usage of Bayesi
+00001f70: 616e 206d 6f6c 6563 756c 6172 2064 6174  an molecular dat
+00001f80: 696e 6720 7072 6f67 7261 6d20 606d 636d  ing program `mcm
+00001f90: 6374 7265 6560 2c20 7768 6963 6820 7072  ctree`, which pr
+00001fa0: 6f76 6964 6573 2061 2076 6172 6965 7479  ovides a variety
+00001fb0: 206f 6620 6469 6666 6572 656e 7420 7375   of different su
+00001fc0: 6273 7469 7475 7469 6f6e 2061 6e64 2072  bstitution and r
+00001fd0: 6174 6520 6d6f 6465 6c73 2e20 4e6f 6e65  ate models. None
+00001fe0: 7468 656c 6573 732c 2077 6520 7572 6765  theless, we urge
+00001ff0: 2075 7365 7273 2074 6f20 7365 7420 7072   users to set pr
+00002000: 696f 7220 6469 7374 7269 6275 7469 6f6e  ior distribution
+00002010: 206f 6620 6469 6666 6572 656e 7420 7061   of different pa
+00002020: 7261 6d65 7465 7273 2077 6974 6820 6361  rameters with ca
+00002030: 7574 696f 6e20 616e 6420 6173 7375 7265  ution and assure
+00002040: 2061 6465 7175 6174 6520 7361 6d70 6c69   adequate sampli
+00002050: 6e67 206f 6620 6469 6666 6572 656e 7420  ng of different 
+00002060: 7061 7261 6d65 7465 7273 2e0a 0a23 2320  parameters...## 
+00002070: 496e 7374 616c 6c61 7469 6f6e 0a0a 5468  Installation..Th
+00002080: 6520 6561 7369 6573 7420 7761 7920 746f  e easiest way to
+00002090: 2069 6e73 7461 6c6c 2060 7767 6420 7632   install `wgd v2
+000020a0: 6020 6973 2075 7369 6e67 2060 5059 5049  ` is using `PYPI
+000020b0: 602e 204e 6f74 6520 7468 6174 2069 6620  `. Note that if 
+000020c0: 796f 7520 7761 6e74 2074 6f20 6765 7420  you want to get 
+000020d0: 7468 6520 6c61 7465 7374 2075 7064 6174  the latest updat
+000020e0: 652c 2077 6520 7375 6767 6573 7420 696e  e, we suggest in
+000020f0: 7374 616c 6c69 6e67 2066 726f 6d20 7468  stalling from th
+00002100: 6520 736f 7572 6365 2c20 7369 6e63 6520  e source, since 
+00002110: 7468 6520 7570 6461 7465 206f 6e20 6050  the update on `P
+00002120: 5950 4960 2077 696c 6c20 6265 2064 656c  YPI` will be del
+00002130: 6179 6564 2063 6f6d 7061 7265 6420 746f  ayed compared to
+00002140: 2068 6572 6520 6f66 2073 6f75 7263 652e   here of source.
+00002150: 0a0a 6060 600a 7069 7020 696e 7374 616c  ..```.pip instal
+00002160: 6c20 7767 640a 6060 600a 0a54 6f20 696e  l wgd.```..To in
+00002170: 7374 616c 6c20 6077 6764 2076 3260 2069  stall `wgd v2` i
+00002180: 6e20 6120 7669 7274 7561 6c20 656e 7669  n a virtual envi
+00002190: 726f 6e6d 656e 7420 2877 6869 6368 2077  ronment (which w
+000021a0: 6520 7374 726f 6e67 6c79 2072 6563 6f6d  e strongly recom
+000021b0: 6d65 6e64 2c20 6e6f 206d 6174 7465 7220  mend, no matter 
+000021c0: 796f 7520 696e 7374 616c 6c20 6672 6f6d  you install from
+000021d0: 2073 6f75 7263 652c 2060 5059 5049 6020   source, `PYPI` 
+000021e0: 6f72 2060 6269 6f63 6f6e 6461 6029 2c20  or `bioconda`), 
+000021f0: 7468 6520 666f 6c6c 6f77 696e 6720 636f  the following co
+00002200: 6d6d 616e 6420 6c69 6e65 7320 636f 756c  mmand lines coul
+00002210: 6420 6265 2075 7365 642e 0a0a 6060 600a  d be used...```.
+00002220: 6769 7420 636c 6f6e 6520 6874 7470 733a  git clone https:
+00002230: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
+00002240: 6865 2d70 7362 2f77 6764 0a63 6420 7767  he-psb/wgd.cd wg
+00002250: 640a 7669 7274 7561 6c65 6e76 202d 703d  d.virtualenv -p=
+00002260: 7079 7468 6f6e 3320 454e 5620 286f 7220  python3 ENV (or 
+00002270: 7079 7468 6f6e 3320 2d6d 2076 656e 7620  python3 -m venv 
+00002280: 454e 5629 0a73 6f75 7263 6520 454e 562f  ENV).source ENV/
+00002290: 6269 6e2f 6163 7469 7661 7465 0a70 6970  bin/activate.pip
+000022a0: 2069 6e73 7461 6c6c 202d 7220 7265 7175   install -r requ
+000022b0: 6972 656d 656e 7473 2e74 7874 0a70 6970  irements.txt.pip
+000022c0: 2069 6e73 7461 6c6c 202e 0a60 6060 0a0a   install ..```..
+000022d0: 5768 656e 206d 6574 2077 6974 6820 7065  When met with pe
+000022e0: 726d 6973 7369 6f6e 2070 726f 626c 656d  rmission problem
+000022f0: 2069 6e20 696e 7374 616c 6c61 7469 6f6e   in installation
+00002300: 2c20 706c 6561 7365 2074 7279 2074 6865  , please try the
+00002310: 2066 6f6c 6c6f 7769 6e67 2063 6f6d 6d61   following comma
+00002320: 6e64 206c 696e 652e 0a0a 6060 600a 7069  nd line...```.pi
+00002330: 7020 696e 7374 616c 6c20 2d65 202e 0a60  p install -e ..`
+00002340: 6060 0a0a 4966 206d 756c 7469 706c 7920  ``..If multiply 
+00002350: 7665 7273 696f 6e73 206f 6620 6077 6764  versions of `wgd
+00002360: 6020 7765 7265 2069 6e73 7461 6c6c 6564  ` were installed
+00002370: 2069 6e20 7468 6520 7379 7374 656d 2c20   in the system, 
+00002380: 706c 6561 7365 2061 6464 2074 6865 2072  please add the r
+00002390: 6967 6874 2070 6174 6820 6f66 2069 6e74  ight path of int
+000023a0: 6572 6573 7465 6420 7665 7273 696f 6e20  erested version 
+000023b0: 696e 746f 2074 6865 2065 6e76 6972 6f6e  into the environ
+000023c0: 6d65 6e74 2076 6172 6961 626c 6573 2c20  ment variables, 
+000023d0: 666f 7220 6578 616d 706c 650a 0a60 6060  for example..```
+000023e0: 0a65 7870 6f72 7420 5041 5448 3d22 2450  .export PATH="$P
+000023f0: 4154 483a 7e2f 2e6c 6f63 616c 2f62 696e  ATH:~/.local/bin
+00002400: 2f77 6764 220a 6060 600a 0a4e 6f74 6520  /wgd".```..Note 
+00002410: 7468 6174 2074 6865 2076 6572 7369 6f6e  that the version
+00002420: 206f 6620 606e 756d 7079 6020 6973 2069   of `numpy` is i
+00002430: 6d70 6f72 7461 6e74 2028 666f 7220 6d61  mportant (for ma
+00002440: 6e79 206f 7468 6572 2070 6163 6b61 6765  ny other package
+00002450: 7320 6172 6520 7468 6520 7361 6d65 206f  s are the same o
+00002460: 6620 636f 7572 7365 292c 2065 7370 6563  f course), espec
+00002470: 6961 6c6c 7920 666f 7220 6066 6173 7463  ially for `fastc
+00002480: 6c75 7374 6572 6020 7061 636b 6167 652e  luster` package.
+00002490: 2049 6e20 6f75 7220 7465 7374 2c20 7468   In our test, th
+000024a0: 6520 606e 756d 7079 6020 312e 3139 2e30  e `numpy` 1.19.0
+000024b0: 2077 6f72 6b73 2066 696e 6520 6f6e 2060   works fine on `
+000024c0: 7079 7468 6f6e 332e 362f 3860 2e20 4966  python3.6/8`. If
+000024d0: 2079 6f75 206d 6574 2073 6f6d 6520 6572   you met some er
+000024e0: 726f 7273 206f 7220 7761 726e 696e 6773  rors or warnings
+000024f0: 2061 626f 7574 2060 6e75 6d70 7960 2c20   about `numpy`, 
+00002500: 6d61 7962 6520 636f 6e73 6964 6572 696e  maybe considerin
+00002510: 6720 7072 652d 696e 7374 616c 6c20 606e  g pre-install `n
+00002520: 756d 7079 6020 6173 2031 2e31 392e 3020  umpy` as 1.19.0 
+00002530: 6f72 206f 7468 6572 2063 6c6f 7365 2d62  or other close-b
+00002540: 7920 7665 7273 696f 6e73 2062 6566 6f72  y versions befor
+00002550: 6520 796f 7520 696e 7374 616c 6c20 6077  e you install `w
+00002560: 6764 602e 2060 7767 6460 2072 656c 6965  gd`. `wgd` relie
+00002570: 7320 6f6e 2065 7874 6572 6e61 6c20 736f  s on external so
+00002580: 6674 7761 7265 7320 696e 636c 7564 696e  ftwares includin
+00002590: 6720 6064 6961 6d6f 6e64 6020 616e 6420  g `diamond` and 
+000025a0: 606d 636c 6020 666f 7220 6077 6764 2064  `mcl` for `wgd d
+000025b0: 6d64 602c 2060 7061 6d6c 2076 342e 396a  md`, `paml v4.9j
+000025c0: 602c 2060 6d61 6666 7460 2028 606d 7573  `, `mafft` (`mus
+000025d0: 636c 6560 2061 6e64 2060 7072 616e 6b60  cle` and `prank`
+000025e0: 2069 6620 7365 7429 2c20 6066 6173 7474   if set), `fastt
+000025f0: 7265 6560 2028 6f72 2060 6971 7472 6565  ree` (or `iqtree
+00002600: 6020 6966 2073 6574 2920 666f 7220 6077  ` if set) for `w
+00002610: 6764 206b 7364 6020 616e 6420 6f70 7469  gd ksd` and opti
+00002620: 6f6e 616c 6c79 2060 6d72 6261 7965 7360  onally `mrbayes`
+00002630: 2066 6f72 2074 6865 2070 6879 6c6f 6765   for the phyloge
+00002640: 6e65 7469 6320 696e 6665 7265 6e63 6520  netic inference 
+00002650: 6675 6e63 7469 6f6e 2069 6e20 6077 6764  function in `wgd
+00002660: 2064 6d64 6020 616e 6420 6077 6764 2066   dmd` and `wgd f
+00002670: 6f63 7573 6020 2860 6d61 6666 7460 2c20  ocus` (`mafft`, 
+00002680: 606d 7573 636c 6560 2061 6e64 2060 7072  `muscle` and `pr
+00002690: 616e 6b60 2061 7320 7765 6c6c 2077 6865  ank` as well whe
+000026a0: 6e20 7468 6520 616e 616c 7973 6973 2072  n the analysis r
+000026b0: 6571 7569 7265 7320 7365 7175 656e 6365  equires sequence
+000026c0: 2061 6c69 676e 6d65 6e74 292e 2053 6f6d   alignment). Som
+000026d0: 6520 6f74 6865 7220 6f70 7469 6f6e 616c  e other optional
+000026e0: 2073 6f66 7477 6172 6573 2069 6e63 6c75   softwares inclu
+000026f0: 6469 6e67 2060 7061 6d6c 2076 342e 396a  ding `paml v4.9j
+00002700: 602c 2060 7238 7360 2c20 6062 6561 7374  `, `r8s`, `beast
+00002710: 602c 2060 6567 676e 6f67 602c 2060 6469  `, `eggnog`, `di
+00002720: 616d 6f6e 6460 2c20 6069 6e74 6572 7072  amond`, `interpr
+00002730: 6f73 6361 6e60 2c20 6068 6d6d 6572 2076  oscan`, `hmmer v
+00002740: 332e 3162 3260 2061 6e64 2060 6173 7472  3.1b2` and `astr
+00002750: 616c 2d70 726f 6020 2860 686d 6d65 7220  al-pro` (`hmmer 
+00002760: 7633 2e31 6232 6020 6973 2061 6c73 6f20  v3.1b2` is also 
+00002770: 666f 7220 7468 6520 5b6f 7274 686f 6772  for the [orthogr
+00002780: 6f75 7020 6173 7369 676e 6d65 6e74 5d28  oup assignment](
+00002790: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
+000027a0: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
+000027b0: 6c61 7465 7374 2f75 7361 6765 2e68 746d  latest/usage.htm
+000027c0: 6c23 6f72 7468 6f67 726f 7570 2d61 7373  l#orthogroup-ass
+000027d0: 6967 6e6d 656e 7429 2066 756e 6374 696f  ignment) functio
+000027e0: 6e20 696e 2060 7767 6420 646d 6460 2061  n in `wgd dmd` a
+000027f0: 6e64 2060 6173 7472 616c 2d70 726f 6020  nd `astral-pro` 
+00002800: 6973 2061 6c73 6f20 666f 7220 7468 6520  is also for the 
+00002810: 6675 6e63 7469 6f6e 2060 2d2d 636f 6c6c  function `--coll
+00002820: 696e 6561 7263 6f61 6c65 7363 656e 6365  inearcoalescence
+00002830: 6020 696e 2060 7767 6420 646d 6460 2920  ` in `wgd dmd`) 
+00002840: 6172 6520 666f 7220 7468 6520 6d6f 6c65  are for the mole
+00002850: 6375 6c61 7220 6461 7469 6e67 2c20 6765  cular dating, ge
+00002860: 6e65 2066 616d 696c 7920 6675 6e63 7469  ne family functi
+00002870: 6f6e 2061 6e6e 6f74 6174 696f 6e20 6f72  on annotation or
+00002880: 2070 6879 6c6f 6765 6e65 7469 6320 696e   phylogenetic in
+00002890: 6665 7265 6e63 6520 696e 2060 7767 6420  ference in `wgd 
+000028a0: 666f 6375 7360 2e0a 0a23 2320 5069 7065  focus`...## Pipe
+000028b0: 6c69 6e65 730a 0a54 6f20 7175 6963 6b6c  lines..To quickl
+000028c0: 7920 6765 7420 6661 6d69 6c69 6172 2077  y get familiar w
+000028d0: 6974 6820 6077 6764 2076 3260 2c20 7765  ith `wgd v2`, we
+000028e0: 2070 726f 7669 6465 6420 736f 6d65 2063   provided some c
+000028f0: 6f6d 6d6f 6e20 7069 7065 6c69 6e65 7320  ommon pipelines 
+00002900: 666f 7220 7573 6572 7320 7769 7468 2061  for users with a
+00002910: 2066 7265 7368 2067 656e 6f6d 6520 6173   fresh genome as
+00002920: 7365 6d62 6c79 2069 6e20 6861 6e64 2e0a  sembly in hand..
+00002930: 0a23 2323 2050 6970 656c 696e 6520 310a  .### Pipeline 1.
+00002940: 2a2a 5369 6d70 6c65 2063 6f6e 7374 7275  **Simple constru
+00002950: 6374 696f 6e20 6f66 2061 6765 2064 6973  ction of age dis
+00002960: 7472 6962 7574 696f 6e2a 2a0a 6060 600a  tribution**.```.
+00002970: 7767 6420 646d 6420 4171 7569 6c65 6769  wgd dmd Aquilegi
+00002980: 615f 636f 6572 756c 6561 202d 6f20 7767  a_coerulea -o wg
+00002990: 645f 646d 640a 7767 6420 6b73 6420 7767  d_dmd.wgd ksd wg
+000029a0: 645f 646d 642f 4171 7569 6c65 6769 615f  d_dmd/Aquilegia_
+000029b0: 636f 6572 756c 6561 2e74 7376 2041 7175  coerulea.tsv Aqu
+000029c0: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+000029d0: 2d6f 2077 6764 5f6b 7364 0a60 6060 0a60  -o wgd_ksd.```.`
+000029e0: 6060 0a23 7768 6174 2077 696c 6c20 6265  ``.#what will be
+000029f0: 2069 6e20 7468 6520 7265 7375 6c74 2064   in the result d
+00002a00: 6972 6563 746f 7279 0a2d 7767 645f 646d  irectory.-wgd_dm
+00002a10: 640a 2d2d 4171 7569 6c65 6769 615f 636f  d.--Aquilegia_co
+00002a20: 6572 756c 6561 2e74 7376 0a2d 7767 645f  erulea.tsv.-wgd_
+00002a30: 6b73 640a 2d2d 4171 7569 6c65 6769 615f  ksd.--Aquilegia_
+00002a40: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00002a50: 7473 7620 4171 7569 6c65 6769 615f 636f  tsv Aquilegia_co
+00002a60: 6572 756c 6561 2e74 7376 2e6b 732e 7376  erulea.tsv.ks.sv
+00002a70: 672f 7064 660a 6060 600a 5468 6520 7265  g/pdf.```.The re
+00002a80: 7375 6c74 696e 6720 6041 7175 696c 6567  sulting `Aquileg
+00002a90: 6961 5f63 6f65 7275 6c65 612e 7473 7660  ia_coerulea.tsv`
+00002aa0: 2069 7320 7468 6520 7768 6f6c 6520 7061   is the whole pa
+00002ab0: 7261 6e6f 6d65 2066 616d 696c 7920 6669  ranome family fi
+00002ac0: 6c65 2c20 6041 7175 696c 6567 6961 5f63  le, `Aquilegia_c
+00002ad0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00002ae0: 7376 6020 6669 6c65 2069 7320 7468 6520  sv` file is the 
+00002af0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00002b00: 6469 7374 7269 6275 6974 6f6e 2066 696c  distribuiton fil
+00002b10: 652c 2060 4171 7569 6c65 6769 615f 636f  e, `Aquilegia_co
+00002b20: 6572 756c 6561 2e74 7376 2e6b 732e 7376  erulea.tsv.ks.sv
+00002b30: 6760 2061 6e64 2060 4171 7569 6c65 6769  g` and `Aquilegi
+00002b40: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00002b50: 732e 7064 6660 2061 7265 2074 6865 202a  s.pdf` are the *
+00002b60: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
+00002b70: 6c6f 7473 2e0a 0a23 2323 2050 6970 656c  lots...### Pipel
+00002b80: 696e 6520 320a 2a2a 5369 6d70 6c65 2063  ine 2.**Simple c
+00002b90: 6f6e 7374 7275 6374 696f 6e20 6f66 2061  onstruction of a
+00002ba0: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00002bb0: 7769 7468 2063 6f6c 6c69 6e65 6172 6974  with collinearit
+00002bc0: 792a 2a0a 6060 600a 7767 6420 646d 6420  y**.```.wgd dmd 
+00002bd0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00002be0: 6561 202d 6f20 7767 645f 646d 640a 7767  ea -o wgd_dmd.wg
+00002bf0: 6420 6b73 6420 7767 645f 646d 642f 4171  d ksd wgd_dmd/Aq
+00002c00: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00002c10: 2e74 7376 2041 7175 696c 6567 6961 5f63  .tsv Aquilegia_c
+00002c20: 6f65 7275 6c65 6120 2d6f 2077 6764 5f6b  oerulea -o wgd_k
+00002c30: 7364 0a77 6764 2073 796e 202d 6620 6d52  sd.wgd syn -f mR
+00002c40: 4e41 202d 6120 4e61 6d65 2077 6764 5f64  NA -a Name wgd_d
+00002c50: 6d64 2f41 7175 696c 6567 6961 5f63 6f65  md/Aquilegia_coe
+00002c60: 7275 6c65 612e 7473 7620 4171 7569 6c65  rulea.tsv Aquile
+00002c70: 6769 615f 636f 6572 756c 6561 2e67 6666  gia_coerulea.gff
+00002c80: 3320 2d6b 7320 7767 645f 6b73 642f 4171  3 -ks wgd_ksd/Aq
+00002c90: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00002ca0: 2e74 7376 2e6b 732e 7473 7620 2d6f 2077  .tsv.ks.tsv -o w
+00002cb0: 6764 5f73 796e 0a60 6060 0a60 6060 0a23  gd_syn.```.```.#
+00002cc0: 2077 6861 7420 7769 6c6c 2062 6520 696e   what will be in
+00002cd0: 2074 6865 2077 6764 5f73 796e 2072 6573   the wgd_syn res
+00002ce0: 756c 7420 6469 7265 6374 6f72 790a 2d77  ult directory.-w
+00002cf0: 6764 5f73 796e 0a2d 2d61 6e63 686f 7273  gd_syn.--anchors
+00002d00: 2e63 7376 0a2d 2d66 616d 696c 6965 732e  .csv.--families.
+00002d10: 7473 760a 2d2d 6765 6e65 2d74 6162 6c65  tsv.--gene-table
+00002d20: 2e63 7376 0a2d 2d69 6164 686f 7265 2e63  .csv.--iadhore.c
+00002d30: 6f6e 660a 2d2d 7363 6166 666f 6c64 5f6c  onf.--scaffold_l
+00002d40: 656e 6774 682e 7473 760a 2d2d 7365 676d  ength.tsv.--segm
+00002d50: 656e 7473 5f63 6f6f 7264 696e 6174 6573  ents_coordinates
+00002d60: 2e74 7376 0a2d 2d53 6567 7072 6f66 696c  .tsv.--Segprofil
+00002d70: 652e 6373 760a 2d2d 5379 6e64 6570 7468  e.csv.--Syndepth
+00002d80: 2e70 6466 2f73 7667 0a2d 2d41 7175 696c  .pdf/svg.--Aquil
+00002d90: 6567 6961 5f63 6f65 7275 6c65 615f 4171  egia_coerulea_Aq
+00002da0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00002db0: 5f6d 756c 7469 706c 6963 6f6e 735f 6c65  _multiplicons_le
+00002dc0: 7665 6c2e 7064 662f 706e 672f 7376 670a  vel.pdf/png/svg.
+00002dd0: 2d2d 4171 7569 6c65 6769 615f 636f 6572  --Aquilegia_coer
+00002de0: 756c 6561 5f67 656e 655f 6f72 6465 725f  ulea_gene_order_
+00002df0: 7065 7263 6872 6f6d 2e74 7376 0a2d 2d41  perchrom.tsv.--A
+00002e00: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00002e10: 612e 7473 762e 616e 6368 6f72 732e 6b73  a.tsv.anchors.ks
+00002e20: 2e74 7376 0a2d 2d41 7175 696c 6567 6961  .tsv.--Aquilegia
+00002e30: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00002e40: 642e 7064 662f 7376 670a 2d2d 4171 7569  d.pdf/svg.--Aqui
+00002e50: 6c65 6769 615f 636f 6572 756c 6561 2d76  legia_coerulea-v
+00002e60: 732d 4171 7569 6c65 6769 615f 636f 6572  s-Aquilegia_coer
+00002e70: 756c 6561 2e64 6f74 2e70 6466 2f70 6e67  ulea.dot.pdf/png
+00002e80: 2f73 7667 0a2d 2d41 7175 696c 6567 6961  /svg.--Aquilegia
+00002e90: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
+00002ea0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00002eb0: 646f 745f 756e 6974 5f67 656e 652e 7064  dot_unit_gene.pd
+00002ec0: 662f 706e 672f 7376 670a 2d2d 4171 7569  f/png/svg.--Aqui
+00002ed0: 6c65 6769 615f 636f 6572 756c 6561 2d76  legia_coerulea-v
+00002ee0: 732d 4171 7569 6c65 6769 615f 636f 6572  s-Aquilegia_coer
+00002ef0: 756c 6561 5f4b 732e 646f 742e 7064 662f  ulea_Ks.dot.pdf/
+00002f00: 706e 672f 7376 670a 2d2d 4171 7569 6c65  png/svg.--Aquile
+00002f10: 6769 615f 636f 6572 756c 6561 2d76 732d  gia_coerulea-vs-
+00002f20: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00002f30: 6561 5f4b 732e 646f 745f 756e 6974 5f67  ea_Ks.dot_unit_g
+00002f40: 656e 652e 7064 662f 706e 672f 7376 670a  ene.pdf/png/svg.
+00002f50: 2d2d 6961 6468 6f72 652d 6f75 740a 2d2d  --iadhore-out.--
+00002f60: 2d61 6c69 676e 6d65 6e74 2e74 7874 0a2d  -alignment.txt.-
+00002f70: 2d2d 616e 6368 6f72 706f 696e 7473 2e74  --anchorpoints.t
+00002f80: 7874 0a2d 2d2d 6261 7365 636c 7573 7465  xt.---basecluste
+00002f90: 7273 2e74 7874 0a2d 2d2d 6765 6e65 732e  rs.txt.---genes.
+00002fa0: 7478 740a 2d2d 2d6c 6973 745f 656c 656d  txt.---list_elem
+00002fb0: 656e 7473 2e74 7874 0a2d 2d2d 6d75 6c74  ents.txt.---mult
+00002fc0: 6970 6c69 636f 6e5f 7061 6972 732e 7478  iplicon_pairs.tx
+00002fd0: 740a 2d2d 2d6d 756c 7469 706c 6963 6f6e  t.---multiplicon
+00002fe0: 732e 7478 740a 2d2d 2d73 6567 6d65 6e74  s.txt.---segment
+00002ff0: 732e 7478 740a 6060 600a 5468 6520 6061  s.txt.```.The `a
+00003000: 6e63 686f 7273 2e63 7376 602c 2060 6661  nchors.csv`, `fa
+00003010: 6d69 6c69 6573 2e74 7376 602c 2060 6765  milies.tsv`, `ge
+00003020: 6e65 2d74 6162 6c65 2e63 7376 602c 2060  ne-table.csv`, `
+00003030: 7363 6166 666f 6c64 5f6c 656e 6774 682e  scaffold_length.
+00003040: 7473 7660 2c20 6073 6567 6d65 6e74 735f  tsv`, `segments_
+00003050: 636f 6f72 6469 6e61 7465 732e 7473 7660  coordinates.tsv`
+00003060: 2c20 6053 6567 7072 6f66 696c 652e 6373  , `Segprofile.cs
+00003070: 7660 2c20 6041 7175 696c 6567 6961 5f63  v`, `Aquilegia_c
+00003080: 6f65 7275 6c65 615f 6765 6e65 5f6f 7264  oerulea_gene_ord
+00003090: 6572 5f70 6572 6368 726f 6d2e 7473 7660  er_perchrom.tsv`
+000030a0: 2061 7265 2074 6865 2062 6173 6963 2066   are the basic f
+000030b0: 696c 6573 2073 756d 6d61 7269 7a69 6e67  iles summarizing
+000030c0: 2074 6865 2067 656e 6520 6f72 6465 722c   the gene order,
+000030d0: 2066 616d 696c 792c 2061 6e63 686f 7273   family, anchors
+000030e0: 2061 6e64 2063 6f6c 6c69 6e65 6172 2073   and collinear s
+000030f0: 6567 6d65 6e74 732e 0a54 6865 2060 4171  egments..The `Aq
+00003100: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00003110: 2e74 7376 2e61 6e63 686f 7273 2e6b 732e  .tsv.anchors.ks.
+00003120: 7473 7660 2069 7320 7468 6520 2a4b 2a3c  tsv` is the *K*<
+00003130: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00003140: 7269 6275 7469 6f6e 2066 696c 6520 6f66  ribution file of
+00003150: 2061 6e63 686f 7220 7061 6972 732e 0a54   anchor pairs..T
+00003160: 6865 2060 5379 6e64 6570 7468 2e70 6466  he `Syndepth.pdf
+00003170: 2f73 7667 6020 7368 6f77 7320 7468 6520  /svg` shows the 
+00003180: 636f 6c6c 696e 6561 7220 7261 7469 6f20  collinear ratio 
+00003190: 6f66 206d 756c 7469 706c 6963 6f6e 732e  of multiplicons.
+000031a0: 0a54 6865 2060 4171 7569 6c65 6769 615f  .The `Aquilegia_
+000031b0: 636f 6572 756c 6561 5f41 7175 696c 6567  coerulea_Aquileg
+000031c0: 6961 5f63 6f65 7275 6c65 615f 6d75 6c74  ia_coerulea_mult
+000031d0: 6970 6c69 636f 6e73 5f6c 6576 656c 2e70  iplicons_level.p
+000031e0: 6466 2f70 6e67 2f73 7667 6020 6973 2074  df/png/svg` is t
+000031f0: 6865 2022 6475 7053 7461 636b 2220 706c  he "dupStack" pl
+00003200: 6f74 2073 686f 7769 6e67 206d 756c 7469  ot showing multi
+00003210: 706c 6963 6f6e 7320 7769 7468 2064 6966  plicons with dif
+00003220: 6665 7265 6e74 206c 6576 656c 7320 2864  ferent levels (d
+00003230: 6566 696e 6564 2061 7320 7468 6520 6e75  efined as the nu
+00003240: 6d62 6572 206f 6620 7365 676d 656e 7473  mber of segments
+00003250: 2077 6974 6869 6e29 2e0a 5468 6520 6041   within)..The `A
+00003260: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003270: 612e 7473 762e 6b73 642e 7064 662f 7376  a.tsv.ksd.pdf/sv
+00003280: 6760 2069 7320 7468 6520 2a4b 2a3c 7375  g` is the *K*<su
+00003290: 623e 533c 2f73 7562 3e20 706c 6f74 2077  b>S</sub> plot w
+000032a0: 6974 6820 616e 6368 6f72 2070 6169 7273  ith anchor pairs
+000032b0: 2061 6e6e 6f74 6174 6564 2e0a 5468 6520   annotated..The 
+000032c0: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
+000032d0: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
+000032e0: 5f63 6f65 7275 6c65 612e 646f 742e 7064  _coerulea.dot.pd
+000032f0: 662f 706e 672f 7376 6760 2069 7320 7468  f/png/svg` is th
+00003300: 6520 646f 7420 706c 6f74 2077 6974 686f  e dot plot witho
+00003310: 7574 202a 4b2a 3c73 7562 3e53 3c2f 7375  ut *K*<sub>S</su
+00003320: 623e 2061 6e6e 6f74 6174 696f 6e20 616e  b> annotation an
+00003330: 6420 7769 7468 2063 6f6f 7264 696e 6174  d with coordinat
+00003340: 6573 2069 6e20 7468 6520 6e75 6d62 6572  es in the number
+00003350: 206f 6620 6261 7365 2e0a 5468 6520 6041   of base..The `A
+00003360: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003370: 612d 7673 2d41 7175 696c 6567 6961 5f63  a-vs-Aquilegia_c
+00003380: 6f65 7275 6c65 612e 646f 745f 756e 6974  oerulea.dot_unit
+00003390: 5f67 656e 652e 7064 662f 706e 672f 7376  _gene.pdf/png/sv
+000033a0: 6760 2069 7320 7468 6520 646f 7420 706c  g` is the dot pl
+000033b0: 6f74 2077 6974 686f 7574 202a 4b2a 3c73  ot without *K*<s
+000033c0: 7562 3e53 3c2f 7375 623e 2061 6e6e 6f74  ub>S</sub> annot
+000033d0: 6174 696f 6e20 616e 6420 7769 7468 2063  ation and with c
+000033e0: 6f6f 7264 696e 6174 6573 2069 6e20 7468  oordinates in th
+000033f0: 6520 6e75 6d62 6572 206f 6620 6765 6e65  e number of gene
+00003400: 2e0a 5468 6520 6041 7175 696c 6567 6961  ..The `Aquilegia
+00003410: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
+00003420: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+00003430: 4b73 2e64 6f74 2e70 6466 2f70 6e67 2f73  Ks.dot.pdf/png/s
+00003440: 7667 6020 6973 2074 6865 2064 6f74 2070  vg` is the dot p
+00003450: 6c6f 7420 7769 7468 202a 4b2a 3c73 7562  lot with *K*<sub
+00003460: 3e53 3c2f 7375 623e 2061 6e6e 6f74 6174  >S</sub> annotat
+00003470: 696f 6e20 616e 6420 7769 7468 2063 6f6f  ion and with coo
+00003480: 7264 696e 6174 6573 2069 6e20 7468 6520  rdinates in the 
+00003490: 6e75 6d62 6572 206f 6620 6261 7365 2e0a  number of base..
+000034a0: 5468 6520 6041 7175 696c 6567 6961 5f63  The `Aquilegia_c
+000034b0: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
+000034c0: 6567 6961 5f63 6f65 7275 6c65 615f 4b73  egia_coerulea_Ks
+000034d0: 2e64 6f74 5f75 6e69 745f 6765 6e65 2e70  .dot_unit_gene.p
+000034e0: 6466 2f70 6e67 2f73 7667 6020 6973 2074  df/png/svg` is t
+000034f0: 6865 2064 6f74 2070 6c6f 7420 7769 7468  he dot plot with
+00003500: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00003510: 2061 6e6e 6f74 6174 696f 6e20 616e 6420   annotation and 
+00003520: 7769 7468 2063 6f6f 7264 696e 6174 6573  with coordinates
+00003530: 2069 6e20 7468 6520 6e75 6d62 6572 206f   in the number o
+00003540: 6620 6765 6e65 2e0a 5468 6520 6069 6164  f gene..The `iad
+00003550: 686f 7265 2e63 6f6e 6660 2069 7320 7468  hore.conf` is th
+00003560: 6520 636f 6e66 6967 7572 6174 696f 6e20  e configuration 
+00003570: 6669 6c65 2066 6f72 2060 692d 6164 686f  file for `i-adho
+00003580: 7265 602e 0a54 6865 2060 6961 6468 6f72  re`..The `iadhor
+00003590: 652d 6f75 7460 2073 7562 666f 6c64 6572  e-out` subfolder
+000035a0: 2063 6f6e 7461 696e 7320 7468 6520 6f72   contains the or
+000035b0: 6967 696e 616c 2063 6f6c 6c69 6e65 6172  iginal collinear
+000035c0: 2072 6573 756c 7473 2066 726f 6d20 6069   results from `i
+000035d0: 2d61 6468 6f72 6560 2028 706c 6561 7365  -adhore` (please
+000035e0: 2072 6566 6572 2074 6f20 7468 6520 6d61   refer to the ma
+000035f0: 6e75 616c 2074 6865 7265 696e 2066 6f72  nual therein for
+00003600: 2064 6574 6169 6c65 6420 6465 7363 7269   detailed descri
+00003610: 7074 696f 6e29 2e0a 0a23 2323 2050 6970  ption)...### Pip
+00003620: 656c 696e 6520 330a 2a2a 436f 6e73 7472  eline 3.**Constr
+00003630: 7563 7469 6f6e 206f 6620 6167 6520 6469  uction of age di
+00003640: 7374 7269 6275 7469 6f6e 2061 6e64 2045  stribution and E
+00003650: 4c4d 4d20 616e 616c 7973 6973 2a2a 0a60  LMM analysis**.`
+00003660: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
+00003670: 6567 6961 5f63 6f65 7275 6c65 6120 2d6f  egia_coerulea -o
+00003680: 2077 6764 5f64 6d64 0a77 6764 206b 7364   wgd_dmd.wgd ksd
+00003690: 2077 6764 5f64 6d64 2f41 7175 696c 6567   wgd_dmd/Aquileg
+000036a0: 6961 5f63 6f65 7275 6c65 612e 7473 7620  ia_coerulea.tsv 
+000036b0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000036c0: 6561 202d 6f20 7767 645f 6b73 640a 7767  ea -o wgd_ksd.wg
+000036d0: 6420 7669 7a20 2d64 2077 6764 5f6b 7364  d viz -d wgd_ksd
+000036e0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000036f0: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00003700: 6f20 7767 645f 454c 4d4d 0a60 6060 0a60  o wgd_ELMM.```.`
+00003710: 6060 0a23 2077 6861 7420 7769 6c6c 2062  ``.# what will b
+00003720: 6520 696e 2074 6865 2077 6764 5f45 4c4d  e in the wgd_ELM
+00003730: 4d20 7265 7375 6c74 2064 6972 6563 746f  M result directo
+00003740: 7279 0a2d 7767 645f 454c 4d4d 0a2d 2d41  ry.-wgd_ELMM.--A
+00003750: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003760: 612e 7473 762e 6b73 2e74 7376 2e6b 7364  a.tsv.ks.tsv.ksd
+00003770: 2e70 6466 2f73 7667 0a2d 2d41 7175 696c  .pdf/svg.--Aquil
+00003780: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00003790: 762e 6b73 2e74 7376 2e73 706c 696e 655f  v.ks.tsv.spline_
+000037a0: 6e6f 6465 5f61 7665 7261 6765 642e 7064  node_averaged.pd
+000037b0: 662f 7376 670a 2d2d 4171 7569 6c65 6769  f/svg.--Aquilegi
+000037c0: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+000037d0: 732e 7473 762e 7370 6c69 6e65 5f77 6569  s.tsv.spline_wei
+000037e0: 6768 7465 642e 7064 662f 7376 670a 2d2d  ghted.pdf/svg.--
+000037f0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00003800: 6561 2e74 7376 2e6b 732e 7473 765f 7065  ea.tsv.ks.tsv_pe
+00003810: 616b 5f64 6574 6563 7469 6f6e 5f6e 6f64  ak_detection_nod
+00003820: 655f 6176 6572 6167 6564 2e70 6466 2f73  e_averaged.pdf/s
+00003830: 7667 0a2d 2d41 7175 696c 6567 6961 5f63  vg.--Aquilegia_c
+00003840: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00003850: 7376 5f70 6561 6b5f 6465 7465 6374 696f  sv_peak_detectio
+00003860: 6e5f 7765 6967 6874 6564 2e70 6466 2f73  n_weighted.pdf/s
+00003870: 7667 0a2d 2d65 6c6d 6d5f 4171 7569 6c65  vg.--elmm_Aquile
+00003880: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00003890: 2e6b 732e 7473 765f 6265 7374 5f6d 6f64  .ks.tsv_best_mod
+000038a0: 656c 735f 6e6f 6465 5f61 7665 7261 6765  els_node_average
+000038b0: 642e 7064 662f 7376 670a 2d2d 656c 6d6d  d.pdf/svg.--elmm
+000038c0: 5f41 7175 696c 6567 6961 5f63 6f65 7275  _Aquilegia_coeru
+000038d0: 6c65 612e 7473 762e 6b73 2e74 7376 5f62  lea.tsv.ks.tsv_b
+000038e0: 6573 745f 6d6f 6465 6c73 5f77 6569 6768  est_models_weigh
+000038f0: 7465 642e 7064 662f 7376 670a 2d2d 656c  ted.pdf/svg.--el
+00003900: 6d6d 5f41 7175 696c 6567 6961 5f63 6f65  mm_Aquilegia_coe
+00003910: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00003920: 5f6d 6f64 656c 735f 6461 7461 5f64 7269  _models_data_dri
+00003930: 7665 6e5f 6e6f 6465 5f61 7665 7261 6765  ven_node_average
+00003940: 642e 7064 662f 7376 670a 2d2d 656c 6d6d  d.pdf/svg.--elmm
+00003950: 5f41 7175 696c 6567 6961 5f63 6f65 7275  _Aquilegia_coeru
+00003960: 6c65 612e 7473 762e 6b73 2e74 7376 5f6d  lea.tsv.ks.tsv_m
+00003970: 6f64 656c 735f 6461 7461 5f64 7269 7665  odels_data_drive
+00003980: 6e5f 7765 6967 6874 6564 2e70 6466 2f73  n_weighted.pdf/s
+00003990: 7667 0a2d 2d65 6c6d 6d5f 4171 7569 6c65  vg.--elmm_Aquile
+000039a0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+000039b0: 2e6b 732e 7473 765f 6d6f 6465 6c73 5f72  .ks.tsv_models_r
+000039c0: 616e 646f 6d5f 6e6f 6465 5f61 7665 7261  andom_node_avera
+000039d0: 6765 642e 7064 662f 7376 670a 2d2d 656c  ged.pdf/svg.--el
+000039e0: 6d6d 5f41 7175 696c 6567 6961 5f63 6f65  mm_Aquilegia_coe
+000039f0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00003a00: 5f6d 6f64 656c 735f 7261 6e64 6f6d 5f77  _models_random_w
+00003a10: 6569 6768 7465 642e 7064 662f 7376 670a  eighted.pdf/svg.
+00003a20: 2d2d 656c 6d6d 5f42 4943 5f41 7175 696c  --elmm_BIC_Aquil
+00003a30: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00003a40: 762e 6b73 2e74 7376 5f6e 6f64 655f 6176  v.ks.tsv_node_av
+00003a50: 6572 6167 6564 2e70 6466 2f73 7667 0a2d  eraged.pdf/svg.-
+00003a60: 2d65 6c6d 6d5f 4249 435f 4171 7569 6c65  -elmm_BIC_Aquile
+00003a70: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00003a80: 2e6b 732e 7473 765f 7765 6967 6874 6564  .ks.tsv_weighted
+00003a90: 2e70 6466 2f73 7667 0a60 6060 0a54 6865  .pdf/svg.```.The
+00003aa0: 2060 4171 7569 6c65 6769 615f 636f 6572   `Aquilegia_coer
+00003ab0: 756c 6561 2e74 7376 2e6b 732e 7473 762e  ulea.tsv.ks.tsv.
+00003ac0: 6b73 642e 7064 662f 7376 6760 2069 7320  ksd.pdf/svg` is 
+00003ad0: 7468 6520 6261 7369 6320 2a4b 2a3c 7375  the basic *K*<su
+00003ae0: 623e 533c 2f73 7562 3e20 706c 6f74 2e20  b>S</sub> plot. 
+00003af0: 5468 6520 6041 7175 696c 6567 6961 5f63  The `Aquilegia_c
+00003b00: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00003b10: 7376 2e73 706c 696e 655f 6e6f 6465 5f61  sv.spline_node_a
+00003b20: 7665 7261 6765 642e 7064 662f 7376 6760  veraged.pdf/svg`
+00003b30: 2061 6e64 2060 4171 7569 6c65 6769 615f   and `Aquilegia_
+00003b40: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00003b50: 7473 762e 7370 6c69 6e65 5f77 6569 6768  tsv.spline_weigh
+00003b60: 7465 642e 7064 662f 7376 6760 2061 7265  ted.pdf/svg` are
+00003b70: 2074 6865 2073 706c 696e 6520 706c 6f74   the spline plot
+00003b80: 7320 7573 6564 2066 6f72 2070 6561 6b20  s used for peak 
+00003b90: 6465 7465 6374 696f 6e2e 2054 6865 2060  detection. The `
+00003ba0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00003bb0: 6561 2e74 7376 2e6b 732e 7473 765f 7065  ea.tsv.ks.tsv_pe
+00003bc0: 616b 5f64 6574 6563 7469 6f6e 5f6e 6f64  ak_detection_nod
+00003bd0: 655f 6176 6572 6167 6564 2e70 6466 2f73  e_averaged.pdf/s
+00003be0: 7667 6020 616e 6420 6041 7175 696c 6567  vg` and `Aquileg
+00003bf0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00003c00: 6b73 2e74 7376 5f70 6561 6b5f 6465 7465  ks.tsv_peak_dete
+00003c10: 6374 696f 6e5f 7765 6967 6874 6564 2e70  ction_weighted.p
+00003c20: 6466 2f73 7667 6020 6172 6520 7468 6520  df/svg` are the 
+00003c30: 7265 7375 6c74 7320 6f66 2070 6561 6b20  results of peak 
+00003c40: 6465 7465 6374 696f 6e2e 2054 6865 2060  detection. The `
+00003c50: 656c 6d6d 5f41 7175 696c 6567 6961 5f63  elmm_Aquilegia_c
+00003c60: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00003c70: 7376 5f62 6573 745f 6d6f 6465 6c73 5f6e  sv_best_models_n
+00003c80: 6f64 655f 6176 6572 6167 6564 2e70 6466  ode_averaged.pdf
+00003c90: 2f73 7667 6020 616e 6420 6065 6c6d 6d5f  /svg` and `elmm_
+00003ca0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00003cb0: 6561 2e74 7376 2e6b 732e 7473 765f 6265  ea.tsv.ks.tsv_be
+00003cc0: 7374 5f6d 6f64 656c 735f 7765 6967 6874  st_models_weight
+00003cd0: 6564 2e70 6466 2f73 7667 6020 6172 6520  ed.pdf/svg` are 
+00003ce0: 7468 6520 7265 7375 6c74 7320 6672 6f6d  the results from
+00003cf0: 2074 6865 2062 6573 7420 6d6f 6465 6c20   the best model 
+00003d00: 696e 666f 726d 6564 2062 7920 6042 4943  informed by `BIC
+00003d10: 602e 2054 6865 2060 656c 6d6d 5f41 7175  `. The `elmm_Aqu
+00003d20: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00003d30: 7473 762e 6b73 2e74 7376 5f6d 6f64 656c  tsv.ks.tsv_model
+00003d40: 735f 6461 7461 5f64 7269 7665 6e5f 6e6f  s_data_driven_no
+00003d50: 6465 5f61 7665 7261 6765 642e 7064 662f  de_averaged.pdf/
+00003d60: 7376 6760 2061 6e64 2060 656c 6d6d 5f41  svg` and `elmm_A
+00003d70: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003d80: 612e 7473 762e 6b73 2e74 7376 5f6d 6f64  a.tsv.ks.tsv_mod
+00003d90: 656c 735f 6461 7461 5f64 7269 7665 6e5f  els_data_driven_
+00003da0: 7765 6967 6874 6564 2e70 6466 2f73 7667  weighted.pdf/svg
+00003db0: 6020 6172 6520 7468 6520 6d6f 6465 6c20  ` are the model 
+00003dc0: 7265 7375 6c74 7320 6672 6f6d 2064 6174  results from dat
+00003dd0: 612d 6472 6976 656e 2069 6e69 7469 6174  a-driven initiat
+00003de0: 696f 6e2e 2054 6865 2060 656c 6d6d 5f41  ion. The `elmm_A
+00003df0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003e00: 612e 7473 762e 6b73 2e74 7376 5f6d 6f64  a.tsv.ks.tsv_mod
+00003e10: 656c 735f 7261 6e64 6f6d 5f6e 6f64 655f  els_random_node_
+00003e20: 6176 6572 6167 6564 2e70 6466 2f73 7667  averaged.pdf/svg
+00003e30: 6020 616e 6420 6065 6c6d 6d5f 4171 7569  ` and `elmm_Aqui
+00003e40: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00003e50: 7376 2e6b 732e 7473 765f 6d6f 6465 6c73  sv.ks.tsv_models
+00003e60: 5f72 616e 646f 6d5f 7765 6967 6874 6564  _random_weighted
+00003e70: 2e70 6466 2f73 7667 6020 6172 6520 7468  .pdf/svg` are th
+00003e80: 6520 6d6f 6465 6c20 7265 7375 6c74 7320  e model results 
+00003e90: 6672 6f6d 2072 616e 646f 6d20 696e 6974  from random init
+00003ea0: 6961 7469 6f6e 2e20 5468 6520 6065 6c6d  iation. The `elm
+00003eb0: 6d5f 4249 435f 4171 7569 6c65 6769 615f  m_BIC_Aquilegia_
+00003ec0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00003ed0: 7473 765f 6e6f 6465 5f61 7665 7261 6765  tsv_node_average
+00003ee0: 642e 7064 662f 7376 6760 2061 6e64 2060  d.pdf/svg` and `
+00003ef0: 656c 6d6d 5f42 4943 5f41 7175 696c 6567  elmm_BIC_Aquileg
+00003f00: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00003f10: 6b73 2e74 7376 5f77 6569 6768 7465 642e  ks.tsv_weighted.
+00003f20: 7064 662f 7376 6760 2061 7265 2074 6865  pdf/svg` are the
+00003f30: 2042 4943 2070 6c6f 7473 2066 6f72 2065   BIC plots for e
+00003f40: 6163 6820 6d6f 6465 6c2e 0a0a 2323 2320  ach model...### 
+00003f50: 5069 7065 6c69 6e65 2034 0a2a 2a43 6f6e  Pipeline 4.**Con
+00003f60: 7374 7275 6374 696f 6e20 6f66 2061 6765  struction of age
+00003f70: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
+00003f80: 7468 2063 6f6c 6c69 6e65 6172 6974 7920  th collinearity 
+00003f90: 616e 6420 7065 616b 2066 696e 6469 6e67  and peak finding
+00003fa0: 2061 6e61 6c79 7369 732a 2a0a 6060 600a   analysis**.```.
+00003fb0: 7767 6420 646d 6420 4171 7569 6c65 6769  wgd dmd Aquilegi
+00003fc0: 615f 636f 6572 756c 6561 202d 6f20 7767  a_coerulea -o wg
+00003fd0: 645f 646d 640a 7767 6420 6b73 6420 7767  d_dmd.wgd ksd wg
+00003fe0: 645f 646d 642f 4171 7569 6c65 6769 615f  d_dmd/Aquilegia_
+00003ff0: 636f 6572 756c 6561 2e74 7376 2041 7175  coerulea.tsv Aqu
+00004000: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+00004010: 2d6f 2077 6764 5f6b 7364 0a77 6764 2073  -o wgd_ksd.wgd s
+00004020: 796e 202d 6620 6d52 4e41 202d 6120 4e61  yn -f mRNA -a Na
+00004030: 6d65 2077 6764 5f64 6d64 2f41 7175 696c  me wgd_dmd/Aquil
+00004040: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00004050: 7620 4171 7569 6c65 6769 615f 636f 6572  v Aquilegia_coer
+00004060: 756c 6561 2e67 6666 3320 2d6b 7320 7767  ulea.gff3 -ks wg
+00004070: 645f 6b73 642f 4171 7569 6c65 6769 615f  d_ksd/Aquilegia_
+00004080: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00004090: 7473 7620 2d6f 2077 6764 5f73 796e 0a77  tsv -o wgd_syn.w
+000040a0: 6764 2070 6561 6b20 2d2d 6865 7572 6973  gd peak --heuris
+000040b0: 7469 6320 7767 645f 6b73 642f 4171 7569  tic wgd_ksd/Aqui
+000040c0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000040d0: 7376 2e6b 732e 7473 7620 2d61 7020 7767  sv.ks.tsv -ap wg
+000040e0: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+000040f0: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
+00004100: 7874 202d 736d 2077 6764 5f73 796e 2f69  xt -sm wgd_syn/i
+00004110: 6164 686f 7265 2d6f 7574 2f73 6567 6d65  adhore-out/segme
+00004120: 6e74 732e 7478 7420 2d6c 6520 7767 645f  nts.txt -le wgd_
+00004130: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00004140: 6c69 7374 5f65 6c65 6d65 6e74 732e 7478  list_elements.tx
+00004150: 7420 2d6d 7020 7767 645f 7379 6e2f 6961  t -mp wgd_syn/ia
+00004160: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
+00004170: 6c69 636f 6e5f 7061 6972 732e 7478 7420  licon_pairs.txt 
+00004180: 2d6e 2031 2034 202d 6b63 2033 202d 6f20  -n 1 4 -kc 3 -o 
+00004190: 7767 645f 7065 616b 0a60 6060 0a60 6060  wgd_peak.```.```
+000041a0: 0a23 2077 6861 7420 7769 6c6c 2062 6520  .# what will be 
+000041b0: 696e 2074 6865 2077 6764 5f70 6561 6b20  in the wgd_peak 
+000041c0: 7265 7375 6c74 2064 6972 6563 746f 7279  result directory
+000041d0: 0a2d 7767 645f 7065 616b 0a2d 2d41 6e63  .-wgd_peak.--Anc
+000041e0: 686f 724b 735f 4669 6e64 5065 616b 0a2d  horKs_FindPeak.-
+000041f0: 2d2d 416e 6368 6f72 4b73 5f50 6561 6b43  --AnchorKs_PeakC
+00004200: 495f 4171 7569 6c65 6769 615f 636f 6572  I_Aquilegia_coer
+00004210: 756c 6561 2e74 7376 2e6b 732e 7473 765f  ulea.tsv.ks.tsv_
+00004220: 6e6f 6465 5f61 7665 7261 6765 642f 7765  node_averaged/we
+00004230: 6967 6874 6564 2e70 6466 0a2d 2d2d 4171  ighted.pdf.---Aq
+00004240: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00004250: 2e74 7376 2e6b 732e 7473 765f 3935 2543  .tsv.ks.tsv_95%C
+00004260: 495f 4150 5f66 6f72 5f64 6174 696e 675f  I_AP_for_dating_
+00004270: 7765 6967 6874 6564 5f66 6f72 6d61 742e  weighted_format.
+00004280: 7473 760a 2d2d 2d41 7175 696c 6567 6961  tsv.---Aquilegia
+00004290: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+000042a0: 2e74 7376 5f39 3525 4349 5f41 505f 666f  .tsv_95%CI_AP_fo
+000042b0: 725f 6461 7469 6e67 5f77 6569 6768 7465  r_dating_weighte
+000042c0: 642e 7473 760a 2d2d 2d41 7175 696c 6567  d.tsv.---Aquileg
+000042d0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+000042e0: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
+000042f0: 666f 725f 6461 7469 6e67 5f6e 6f64 655f  for_dating_node_
+00004300: 6176 6572 6167 6564 5f66 6f72 6d61 742e  averaged_format.
+00004310: 7473 760a 2d2d 2d41 7175 696c 6567 6961  tsv.---Aquilegia
+00004320: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00004330: 2e74 7376 5f39 3525 4349 5f41 505f 666f  .tsv_95%CI_AP_fo
+00004340: 725f 6461 7469 6e67 5f6e 6f64 655f 6176  r_dating_node_av
+00004350: 6572 6167 6564 2e74 7376 0a2d 2d41 6e63  eraged.tsv.--Anc
+00004360: 686f 724b 735f 474d 4d0a 2d2d 2d47 4d4d  horKs_GMM.---GMM
+00004370: 5f45 6c62 6f77 2d4c 6f73 735f 6f72 6967  _Elbow-Loss_orig
+00004380: 696e 616c 5f4b 732e 7064 660a 2d2d 2d47  inal_Ks.pdf.---G
+00004390: 4d4d 5f4f 7269 6769 6e61 6c5f 416e 6368  MM_Original_Anch
+000043a0: 6f72 4b73 5f43 6c75 7374 6572 696e 675f  orKs_Clustering_
+000043b0: 5369 6c68 6f75 6574 7465 5f43 6f65 6666  Silhouette_Coeff
+000043c0: 6963 6965 6e74 2e70 6466 0a2d 2d2d 4f72  icient.pdf.---Or
+000043d0: 6967 696e 616c 5f41 6e63 686f 724b 735f  iginal_AnchorKs_
+000043e0: 474d 4d5f 312f 322f 332f 3463 6f6d 706f  GMM_1/2/3/4compo
+000043f0: 6e65 6e74 735f 7072 6564 6963 7469 6f6e  nents_prediction
+00004400: 2e74 7376 0a2d 2d2d 4f72 6967 696e 616c  .tsv.---Original
+00004410: 5f41 6e63 686f 724b 735f 474d 4d5f 4149  _AnchorKs_GMM_AI
+00004420: 435f 4249 432e 7064 660a 2d2d 2d4f 7269  C_BIC.pdf.---Ori
+00004430: 6769 6e61 6c5f 416e 6368 6f72 4b73 5f47  ginal_AnchorKs_G
+00004440: 4d4d 5f43 6f6d 706f 6e65 6e74 312f 322f  MM_Component1/2/
+00004450: 332f 345f 6e6f 6465 5f61 7665 7261 6765  3/4_node_average
+00004460: 645f 4c6f 676e 6f72 6d61 6c2e 7064 660a  d_Lognormal.pdf.
+00004470: 2d2d 2d4f 7269 6769 6e61 6c5f 416e 6368  ---Original_Anch
+00004480: 6f72 4b73 5f47 4d4d 5f43 6f6d 706f 6e65  orKs_GMM_Compone
+00004490: 6e74 312f 322f 332f 345f 6e6f 6465 5f61  nt1/2/3/4_node_a
+000044a0: 7665 7261 6765 642e 7064 660a 2d2d 2d4c  veraged.pdf.---L
+000044b0: 6f67 474d 4d5f 4349 0a2d 2d2d 2d47 4d4d  ogGMM_CI.----GMM
+000044c0: 5f31 2f32 2f33 2f34 636f 6d70 6f6e 656e  _1/2/3/4componen
+000044d0: 7473 5f43 302f 312f 322f 335f 3935 2543  ts_C0/1/2/3_95%C
+000044e0: 492e 7473 760a 2d2d 2d2d 474d 4d5f 436f  I.tsv.----GMM_Co
+000044f0: 6d70 6f6e 656e 7431 2f32 2f33 2f34 5f6e  mponent1/2/3/4_n
+00004500: 6f64 655f 6176 6572 6167 6564 5f4c 6f67  ode_averaged_Log
+00004510: 6e6f 726d 616c 2e70 6466 0a2d 2d2d 4869  normal.pdf.---Hi
+00004520: 6768 4d61 7373 5f43 490a 2d2d 2d2d 474d  ghMass_CI.----GM
+00004530: 4d5f 312f 322f 332f 3463 6f6d 706f 6e65  M_1/2/3/4compone
+00004540: 6e74 735f 4330 2f31 2f32 2f33 5f48 6967  nts_C0/1/2/3_Hig
+00004550: 684d 6173 735f 3935 2543 492e 7473 760a  hMass_95%CI.tsv.
+00004560: 2d2d 2d2d 474d 4d5f 312f 322f 332f 3463  ----GMM_1/2/3/4c
+00004570: 6f6d 706f 6e65 6e74 735f 4869 6768 4d61  omponents_HighMa
+00004580: 7373 5f39 3525 4349 2e70 6466 0a2d 2d53  ss_95%CI.pdf.--S
+00004590: 6567 6d65 6e74 4775 6964 654b 735f 474d  egmentGuideKs_GM
+000045a0: 4d0a 2d2d 2d47 4d4d 5f45 6c62 6f77 2d4c  M.---GMM_Elbow-L
+000045b0: 6f73 735f 5365 676d 656e 745f 4b73 2e70  oss_Segment_Ks.p
+000045c0: 6466 0a2d 2d2d 474d 4d5f 5365 676d 656e  df.---GMM_Segmen
+000045d0: 745f 4b73 5f43 6c75 7374 6572 696e 675f  t_Ks_Clustering_
+000045e0: 5369 6c68 6f75 6574 7465 5f43 6f65 6666  Silhouette_Coeff
+000045f0: 6963 6965 6e74 2e70 6466 0a2d 2d2d 5365  icient.pdf.---Se
+00004600: 676d 656e 742d 6775 6964 6564 5f41 6e63  gment-guided_Anc
+00004610: 686f 724b 735f 474d 4d5f 312f 322f 332f  horKs_GMM_1/2/3/
+00004620: 3463 6f6d 706f 6e65 6e74 735f 7072 6564  4components_pred
+00004630: 6963 7469 6f6e 2e74 7376 0a2d 2d2d 5365  iction.tsv.---Se
+00004640: 676d 656e 745f 4b73 5f43 6c75 7374 6572  gment_Ks_Cluster
+00004650: 735f 474d 4d5f 436f 6d70 6f6e 656e 7431  s_GMM_Component1
+00004660: 2f32 2f33 2f34 2e70 6466 0a2d 2d2d 5365  /2/3/4.pdf.---Se
+00004670: 676d 656e 745f 4b73 5f43 6c75 7374 6572  gment_Ks_Cluster
+00004680: 735f 4c6f 676e 6f72 6d61 6c5f 474d 4d5f  s_Lognormal_GMM_
+00004690: 436f 6d70 6f6e 656e 7431 2f32 2f33 2f34  Component1/2/3/4
+000046a0: 2e70 6466 0a2d 2d2d 5365 676d 656e 745f  .pdf.---Segment_
+000046b0: 4b73 5f47 4d4d 5f41 4943 5f42 4943 2e70  Ks_GMM_AIC_BIC.p
+000046c0: 6466 0a2d 2d2d 5365 676d 656e 745f 4b73  df.---Segment_Ks
+000046d0: 2e74 7376 0a2d 2d2d 4869 6768 4d61 7373  .tsv.---HighMass
+000046e0: 5f43 490a 2d2d 2d2d 5365 676d 656e 745f  _CI.----Segment_
+000046f0: 6775 6964 6564 5f31 2f32 2f33 2f34 636f  guided_1/2/3/4co
+00004700: 6d70 6f6e 656e 7473 5f43 302f 312f 322f  mponents_C0/1/2/
+00004710: 335f 4869 6768 4d61 7373 5f39 3525 4349  3_HighMass_95%CI
+00004720: 2e74 7376 0a2d 2d2d 2d53 6567 6d65 6e74  .tsv.----Segment
+00004730: 5f67 7569 6465 645f 312f 322f 332f 3463  _guided_1/2/3/4c
+00004740: 6f6d 706f 6e65 6e74 735f 4869 6768 4d61  omponents_HighMa
+00004750: 7373 5f39 3525 4349 2e70 6466 0a2d 2d2d  ss_95%CI.pdf.---
+00004760: 4844 525f 4349 0a2d 2d2d 2d53 6567 6d65  HDR_CI.----Segme
+00004770: 6e74 5f67 7569 6465 645f 3935 2548 4452  nt_guided_95%HDR
+00004780: 5f41 505f 312f 322f 332f 3463 6f6d 706f  _AP_1/2/3/4compo
+00004790: 6e65 6e74 735f 4330 2f31 2f32 2f33 2e74  nents_C0/1/2/3.t
+000047a0: 7376 0a2d 2d2d 2d53 6567 6d65 6e74 5f67  sv.----Segment_g
+000047b0: 7569 6465 645f 416e 6368 6f72 4b73 5f47  uided_AnchorKs_G
+000047c0: 4d4d 5f43 6f6d 706f 6e65 6e74 312f 322f  MM_Component1/2/
+000047d0: 332f 345f 6e6f 6465 5f61 7665 7261 6765  3/4_node_average
+000047e0: 645f 6b64 652e 7064 660a 2d2d 2d2d 5365  d_kde.pdf.----Se
+000047f0: 676d 656e 745f 6775 6964 6564 5f41 6e63  gment_guided_Anc
+00004800: 686f 724b 735f 474d 4d5f 436f 6d70 6f6e  horKs_GMM_Compon
+00004810: 656e 7431 2f32 2f33 2f34 5f6e 6f64 655f  ent1/2/3/4_node_
+00004820: 6176 6572 6167 6564 2e70 6466 0a2d 2d53  averaged.pdf.--S
+00004830: 6567 6d65 6e74 4b73 5f46 696e 6450 6561  egmentKs_FindPea
+00004840: 6b0a 2d2d 2d53 6567 6d65 6e74 4b73 5f50  k.---SegmentKs_P
+00004850: 6561 6b43 495f 4171 7569 6c65 6769 615f  eakCI_Aquilegia_
+00004860: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00004870: 7473 762e 7064 660a 2d2d 2d50 6561 6b5f  tsv.pdf.---Peak_
+00004880: 312f 325f 5365 676d 656e 745f 6775 6964  1/2_Segment_guid
+00004890: 6564 5f41 7175 696c 6567 6961 5f63 6f65  ed_Aquilegia_coe
+000048a0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+000048b0: 5f39 3525 4349 5f4d 505f 666f 725f 6461  _95%CI_MP_for_da
+000048c0: 7469 6e67 5f66 6f72 6d61 742e 7473 760a  ting_format.tsv.
+000048d0: 2d2d 2d50 6561 6b5f 312f 325f 5365 676d  ---Peak_1/2_Segm
+000048e0: 656e 745f 6775 6964 6564 5f41 7175 696c  ent_guided_Aquil
+000048f0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00004900: 762e 6b73 2e74 7376 5f39 3525 4349 5f4d  v.ks.tsv_95%CI_M
+00004910: 505f 666f 725f 6461 7469 6e67 2e74 7376  P_for_dating.tsv
+00004920: 0a60 6060 0a46 6f75 7220 7265 7375 6c74  .```.Four result
+00004930: 2073 7562 666c 6f64 6572 7320 7769 6c6c   subfloders will
+00004940: 2062 6520 7072 6f64 7563 6564 2c20 6e61   be produced, na
+00004950: 6d65 6c79 2060 416e 6368 6f72 4b73 5f46  mely `AnchorKs_F
+00004960: 696e 6450 6561 6b60 2c20 6041 6e63 686f  indPeak`, `Ancho
+00004970: 724b 735f 474d 4d60 2c20 6053 6567 6d65  rKs_GMM`, `Segme
+00004980: 6e74 4775 6964 654b 735f 474d 4d60 2061  ntGuideKs_GMM` a
+00004990: 6e64 2060 5365 676d 656e 744b 735f 4669  nd `SegmentKs_Fi
+000049a0: 6e64 5065 616b 602e 2054 6865 2060 416e  ndPeak`. The `An
+000049b0: 6368 6f72 4b73 5f46 696e 6450 6561 6b60  chorKs_FindPeak`
+000049c0: 2073 7562 666c 6f64 6572 2063 6f6e 7461   subfloder conta
+000049d0: 696e 7320 7265 7375 6c74 7320 6f66 2074  ins results of t
+000049e0: 6865 2064 6574 6563 7465 6420 7065 616b  he detected peak
+000049f0: 7320 6279 2074 6865 2060 7369 676e 616c  s by the `signal
+00004a00: 6020 6d6f 6475 6c65 206f 6620 6053 6369  ` module of `Sci
+00004a10: 5079 6020 6c69 6272 6172 7920 616e 6420  Py` library and 
+00004a20: 7468 6520 6173 7375 6d65 6420 6869 6768  the assumed high
+00004a30: 6573 7420 6d61 7373 2070 6172 7420 2872  est mass part (r
+00004a40: 6566 6572 7265 6420 746f 2061 7320 4869  eferred to as Hi
+00004a50: 6768 4d61 7373 2068 6572 6561 6674 6572  ghMass hereafter
+00004a60: 2920 6f66 2065 6163 6820 7065 616b 2c20  ) of each peak, 
+00004a70: 7768 6963 6820 6361 6e20 6265 2075 7365  which can be use
+00004a80: 6420 666f 7220 6675 7274 6865 7220 5747  d for further WG
+00004a90: 4420 6461 7469 6e67 2e20 5468 6520 6041  D dating. The `A
+00004aa0: 6e63 686f 724b 735f 474d 4d60 2073 686f  nchorKs_GMM` sho
+00004ab0: 7773 2074 6865 2047 4d4d 2072 6573 756c  ws the GMM resul
+00004ac0: 7473 2075 706f 6e20 7468 6520 6f72 6967  ts upon the orig
+00004ad0: 696e 616c 2061 6e63 686f 7220 2a4b 2a3c  inal anchor *K*<
+00004ae0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00004af0: 7269 6275 7469 6f6e 2062 7920 7468 6520  ribution by the 
+00004b00: 606d 6978 7475 7265 6020 6d6f 6475 6c65  `mixture` module
+00004b10: 206f 6620 6073 6369 6b69 742d 6c65 6172   of `scikit-lear
+00004b20: 6e60 206c 6962 7261 7279 2061 6e64 2074  n` library and t
+00004b30: 776f 2073 7562 666c 6f64 6572 732c 2060  wo subfloders, `
+00004b40: 4c6f 6747 4d4d 5f43 4960 2063 6f6e 7461  LogGMM_CI` conta
+00004b50: 696e 696e 6720 7468 6520 7265 7375 6c74  ining the result
+00004b60: 7320 6f66 2039 3525 2043 4920 6f66 2065  s of 95% CI of e
+00004b70: 6163 6820 636f 6d70 6f6e 656e 742c 2060  ach component, `
+00004b80: 4869 6768 4d61 7373 5f43 4960 2063 6f6e  HighMass_CI` con
+00004b90: 7461 696e 696e 6720 7468 6520 4869 6768  taining the High
+00004ba0: 4d61 7373 206f 6620 6561 6368 2063 6f6d  Mass of each com
+00004bb0: 706f 6e65 6e74 2c20 7768 6963 6820 6361  ponent, which ca
+00004bc0: 6e20 6265 2075 7365 6420 666f 7220 6675  n be used for fu
+00004bd0: 7274 6865 7220 5747 4420 6461 7469 6e67  rther WGD dating
+00004be0: 2e20 5468 6520 6053 6567 6d65 6e74 4775  . The `SegmentGu
+00004bf0: 6964 654b 735f 474d 4d60 2073 7562 666f  ideKs_GMM` subfo
+00004c00: 6c64 6572 2070 7265 7365 6e74 7320 7265  lder presents re
+00004c10: 7375 6c74 7320 6f66 2073 6567 6d65 6e74  sults of segment
+00004c20: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00004c30: 2047 4d4d 2077 6869 6368 2061 7265 206d   GMM which are m
+00004c40: 6170 7065 6420 6261 636b 2074 6f20 7468  apped back to th
+00004c50: 6520 7265 7369 6469 6e67 2061 6e63 686f  e residing ancho
+00004c60: 7220 7061 6972 7320 616e 6420 7468 6520  r pairs and the 
+00004c70: 6173 736f 6369 6174 6564 2039 3525 2048  associated 95% H
+00004c80: 4452 2061 6e64 2048 6967 684d 6173 7320  DR and HighMass 
+00004c90: 6f66 2065 6163 6820 7365 676d 656e 7420  of each segment 
+00004ca0: 636c 7573 7465 7220 696e 2073 7562 666c  cluster in subfl
+00004cb0: 6f64 6572 7320 6f66 2060 4844 525f 4349  oders of `HDR_CI
+00004cc0: 6020 616e 6420 6048 6967 684d 6173 735f  ` and `HighMass_
+00004cd0: 4349 602e 2054 6865 2060 5365 676d 656e  CI`. The `Segmen
+00004ce0: 744b 735f 4669 6e64 5065 616b 6020 7375  tKs_FindPeak` su
+00004cf0: 6266 6f6c 6465 7220 6973 2073 696d 696c  bfolder is simil
+00004d00: 6172 2074 6f20 6041 6e63 686f 724b 735f  ar to `AnchorKs_
+00004d10: 4669 6e64 5065 616b 6020 6275 7420 7769  FindPeak` but wi
+00004d20: 7468 2073 6567 6d65 6e74 202a 4b2a 3c73  th segment *K*<s
+00004d30: 7562 3e53 3c2f 7375 623e 2069 6e73 7465  ub>S</sub> inste
+00004d40: 6164 2e20 5468 6520 2a4b 2a3c 7375 623e  ad. The *K*<sub>
+00004d50: 533c 2f73 7562 3e20 696e 2060 4d75 6c74  S</sub> in `Mult
+00004d60: 6970 6c69 636f 6e60 2063 616e 2061 6c73  iplicon` can als
+00004d70: 6f20 6265 2063 616c 6375 6c61 7465 6420  o be calculated 
+00004d80: 696e 2070 6c61 6365 206f 6620 6053 6567  in place of `Seg
+00004d90: 6d65 6e74 6020 7573 696e 6720 7468 6520  ment` using the 
+00004da0: 6f70 7469 6f6e 2060 2d2d 6775 6964 6560  option `--guide`
+00004db0: 2061 7320 7375 6368 2074 6865 2072 6573   as such the res
+00004dc0: 756c 7420 7469 746c 652c 206c 6162 656c  ult title, label
+00004dd0: 2c20 6669 6c65 2061 6e64 2066 6f6c 6465  , file and folde
+00004de0: 7220 6e61 6d65 7320 7769 6c6c 2062 6520  r names will be 
+00004df0: 6368 616e 6765 6420 6163 636f 7264 696e  changed accordin
+00004e00: 676c 792e 0a0a 2323 2320 5069 7065 6c69  gly...### Pipeli
+00004e10: 6e65 2035 0a2a 2a43 6f6e 7374 7275 6374  ne 5.**Construct
+00004e20: 696f 6e20 6f66 2061 6765 2064 6973 7472  ion of age distr
+00004e30: 6962 7574 696f 6e20 7769 7468 2063 6f6c  ibution with col
+00004e40: 6c69 6e65 6172 6974 7920 616e 6420 7261  linearity and ra
+00004e50: 7465 2063 6f72 7265 6374 696f 6e2a 2a0a  te correction**.
+00004e60: 6060 600a 7767 6420 646d 6420 4171 7569  ```.wgd dmd Aqui
+00004e70: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
+00004e80: 6f20 7767 645f 646d 640a 7767 6420 6b73  o wgd_dmd.wgd ks
+00004e90: 6420 7767 645f 646d 642f 4171 7569 6c65  d wgd_dmd/Aquile
+00004ea0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00004eb0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+00004ec0: 6c65 6120 2d6f 2077 6764 5f6b 7364 0a77  lea -o wgd_ksd.w
+00004ed0: 6764 2073 796e 202d 6620 6d52 4e41 202d  gd syn -f mRNA -
+00004ee0: 6120 4e61 6d65 2077 6764 5f64 6d64 2f41  a Name wgd_dmd/A
+00004ef0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00004f00: 612e 7473 7620 4171 7569 6c65 6769 615f  a.tsv Aquilegia_
+00004f10: 636f 6572 756c 6561 2e67 6666 3320 2d6b  coerulea.gff3 -k
+00004f20: 7320 7767 645f 6b73 642f 4171 7569 6c65  s wgd_ksd/Aquile
+00004f30: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00004f40: 2e6b 732e 7473 7620 2d6f 2077 6764 5f73  .ks.tsv -o wgd_s
+00004f50: 796e 0a77 6764 2064 6d64 202d 2d67 6c6f  yn.wgd dmd --glo
+00004f60: 6261 6c6d 7262 6820 4171 7569 6c65 6769  balmrbh Aquilegi
+00004f70: 615f 636f 6572 756c 6561 2050 726f 7465  a_coerulea Prote
+00004f80: 615f 6379 6e61 726f 6964 6573 2041 636f  a_cynaroides Aco
+00004f90: 7275 735f 616d 6572 6963 616e 7573 2056  rus_americanus V
+00004fa0: 6974 6973 5f76 696e 6966 6572 6120 2d6f  itis_vinifera -o
+00004fb0: 2077 6764 5f67 6c6f 6261 6c6d 7262 680a   wgd_globalmrbh.
+00004fc0: 7767 6420 6b73 6420 7767 645f 676c 6f62  wgd ksd wgd_glob
+00004fd0: 616c 6d72 6268 2f67 6c6f 6261 6c5f 4d52  almrbh/global_MR
+00004fe0: 4248 2e74 7376 2041 7175 696c 6567 6961  BH.tsv Aquilegia
+00004ff0: 5f63 6f65 7275 6c65 6120 5072 6f74 6561  _coerulea Protea
+00005000: 5f63 796e 6172 6f69 6465 7320 4163 6f72  _cynaroides Acor
+00005010: 7573 5f61 6d65 7269 6361 6e75 7320 5669  us_americanus Vi
+00005020: 7469 735f 7669 6e69 6665 7261 202d 6f20  tis_vinifera -o 
+00005030: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
+00005040: 730a 7767 6420 7669 7a20 2d64 2077 6764  s.wgd viz -d wgd
+00005050: 5f67 6c6f 6261 6c6d 7262 685f 6b73 2f67  _globalmrbh_ks/g
+00005060: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
+00005070: 732e 7473 7620 2d66 6120 4171 7569 6c65  s.tsv -fa Aquile
+00005080: 6769 615f 636f 6572 756c 6561 202d 6570  gia_coerulea -ep
+00005090: 6b20 7767 645f 6b73 642f 4171 7569 6c65  k wgd_ksd/Aquile
+000050a0: 6769 615f 636f 6572 756c 6561 2e6b 732e  gia_coerulea.ks.
+000050b0: 7473 7620 2d61 7020 7767 645f 7379 6e2f  tsv -ap wgd_syn/
+000050c0: 6961 6468 6f72 652d 6f75 742f 616e 6368  iadhore-out/anch
+000050d0: 6f72 706f 696e 7473 2e74 7874 202d 7370  orpoints.txt -sp
+000050e0: 2073 7065 6369 6573 7472 6565 2e6e 7720   speciestree.nw 
+000050f0: 2d6f 2077 6764 5f76 697a 5f6d 6978 6564  -o wgd_viz_mixed
+00005100: 5f4b 7320 2d2d 706c 6f74 656c 6d6d 202d  _Ks --plotelmm -
+00005110: 2d70 6c6f 7461 7067 6d6d 202d 2d72 6577  -plotapgmm --rew
+00005120: 6569 6768 740a 6060 600a 6060 600a 2320  eight.```.```.# 
+00005130: 7768 6174 2077 696c 6c20 6265 2069 6e20  what will be in 
+00005140: 7468 6520 7767 645f 7669 7a5f 6d69 7865  the wgd_viz_mixe
+00005150: 645f 4b73 2072 6573 756c 7420 6469 7265  d_Ks result dire
+00005160: 6374 6f72 790a 2d77 6764 5f76 697a 5f6d  ctory.-wgd_viz_m
+00005170: 6978 6564 5f4b 730a 2d2d 416c 6c5f 7061  ixed_Ks.--All_pa
+00005180: 6972 732e 6b73 2e6e 6f64 652e 7765 6967  irs.ks.node.weig
+00005190: 6874 6564 2e70 6466 0a2d 2d46 6f63 7573  hted.pdf.--Focus
+000051a0: 5f73 6973 7465 725f 7061 6972 732e 6b73  _sister_pairs.ks
+000051b0: 2e6e 6f64 652e 7765 6967 6874 6564 2e70  .node.weighted.p
+000051c0: 6466 0a2d 2d67 6c6f 6261 6c5f 4d52 4248  df.--global_MRBH
+000051d0: 2e74 7376 2e6b 732e 7473 762e 6b73 642e  .tsv.ks.tsv.ksd.
+000051e0: 7064 662f 7376 670a 2d2d 4d69 7865 642e  pdf/svg.--Mixed.
+000051f0: 6b73 2e41 7175 696c 6567 6961 5f63 6f65  ks.Aquilegia_coe
+00005200: 7275 6c65 612e 6e6f 6465 2e77 6569 6768  rulea.node.weigh
+00005210: 7465 642e 7064 660a 2d2d 7370 6169 722e  ted.pdf.--spair.
+00005220: 636f 7272 6563 7465 642e 6b73 2e69 6e66  corrected.ks.inf
+00005230: 6f2e 7473 760a 2d2d 7370 6169 722e 6f72  o.tsv.--spair.or
+00005240: 6967 696e 616c 2e6b 732e 696e 666f 2e74  iginal.ks.info.t
+00005250: 7376 0a2d 2d53 696d 706c 655f 4b73 5f44  sv.--Simple_Ks_D
+00005260: 6973 7472 6962 7574 696f 6e73 0a2d 2d2d  istributions.---
+00005270: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+00005280: 732f 4171 7569 6c65 6769 615f 636f 6572  s/Aquilegia_coer
+00005290: 756c 6561 2f50 726f 7465 615f 6379 6e61  ulea/Protea_cyna
+000052a0: 726f 6964 6573 2f56 6974 6973 5f76 696e  roides/Vitis_vin
+000052b0: 6966 6572 615f 5f41 7175 696c 6567 6961  ifera__Aquilegia
+000052c0: 5f63 6f65 7275 6c65 612f 5072 6f74 6561  _coerulea/Protea
+000052d0: 5f63 796e 6172 6f69 6465 732f 5669 7469  _cynaroides/Viti
+000052e0: 735f 7669 6e69 6665 7261 2e6b 732e 6e6f  s_vinifera.ks.no
+000052f0: 6465 2e77 6569 6768 7465 642e 7064 660a  de.weighted.pdf.
+00005300: 6060 600a 5468 6520 6041 6c6c 5f70 6169  ```.The `All_pai
+00005310: 7273 2e6b 732e 6e6f 6465 2e77 6569 6768  rs.ks.node.weigh
+00005320: 7465 642e 7064 6660 2069 7320 7468 6520  ted.pdf` is the 
+00005330: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00005340: 706c 6f74 206f 6620 616c 6c20 7370 6563  plot of all spec
+00005350: 6965 7320 7061 6972 732e 2054 6865 2060  ies pairs. The `
+00005360: 466f 6375 735f 7369 7374 6572 5f70 6169  Focus_sister_pai
+00005370: 7273 2e6b 732e 6e6f 6465 2e77 6569 6768  rs.ks.node.weigh
+00005380: 7465 642e 7064 6660 2069 7320 7468 6520  ted.pdf` is the 
+00005390: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000053a0: 706c 6f74 206f 6620 616c 6c20 666f 6361  plot of all foca
+000053b0: 6c2d 7369 7374 6572 2073 7065 6369 6573  l-sister species
+000053c0: 2070 6169 7273 2e20 5468 6520 6067 6c6f   pairs. The `glo
+000053d0: 6261 6c5f 4d52 4248 2e74 7376 2e6b 732e  bal_MRBH.tsv.ks.
+000053e0: 7473 762e 6b73 642e 7064 662f 7376 6760  tsv.ksd.pdf/svg`
+000053f0: 2069 7320 7468 6520 2a4b 2a3c 7375 623e   is the *K*<sub>
+00005400: 533c 2f73 7562 3e20 706c 6f74 206f 6620  S</sub> plot of 
+00005410: 7468 6520 6461 7461 6669 6c65 2060 676c  the datafile `gl
+00005420: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
+00005430: 2e74 7376 602e 2054 6865 2060 4d69 7865  .tsv`. The `Mixe
+00005440: 642e 6b73 2e41 7175 696c 6567 6961 5f63  d.ks.Aquilegia_c
+00005450: 6f65 7275 6c65 612e 6e6f 6465 2e77 6569  oerulea.node.wei
+00005460: 6768 7465 642e 7064 6660 2069 7320 7468  ghted.pdf` is th
+00005470: 6520 2a2a 6669 6e61 6c20 7265 7375 6c74  e **final result
+00005480: 2a2a 206f 6620 7261 7465 2063 6f72 7265  ** of rate corre
+00005490: 6374 696f 6e20 2877 6974 6820 6d69 7874  ction (with mixt
+000054a0: 7572 6520 6d6f 6465 6c69 6e67 2072 6573  ure modeling res
+000054b0: 756c 7473 2069 6620 7365 7429 2e20 5468  ults if set). Th
+000054c0: 6520 6073 7061 6972 2e63 6f72 7265 6374  e `spair.correct
+000054d0: 6564 2e6b 732e 696e 666f 2e74 7376 6020  ed.ks.info.tsv` 
+000054e0: 616e 6420 6073 7061 6972 2e6f 7269 6769  and `spair.origi
+000054f0: 6e61 6c2e 6b73 2e69 6e66 6f2e 7473 7660  nal.ks.info.tsv`
+00005500: 2064 6f63 756d 656e 7420 7468 6520 2a4b   document the *K
+00005510: 2a3c 7375 623e 533c 2f73 7562 3e20 696e  *<sub>S</sub> in
+00005520: 666f 726d 6174 696f 6e20 6f66 2061 6c6c  formation of all
+00005530: 2073 7065 6369 6573 2070 6169 7273 2062   species pairs b
+00005540: 6566 6f72 6520 616e 6420 6166 7465 7220  efore and after 
+00005550: 7261 7465 2063 6f72 7265 6374 696f 6e2e  rate correction.
+00005560: 2054 6865 2073 7562 666f 6c64 6572 2060   The subfolder `
+00005570: 5369 6d70 6c65 5f4b 735f 4469 7374 7269  Simple_Ks_Distri
+00005580: 6275 7469 6f6e 7360 2063 6f6e 7461 696e  butions` contain
+00005590: 7320 7468 6520 7369 6e67 6c65 202a 4b2a  s the single *K*
+000055a0: 3c73 7562 3e53 3c2f 7375 623e 2070 6c6f  <sub>S</sub> plo
+000055b0: 7473 206f 6620 616c 6c20 7370 6563 6965  ts of all specie
+000055c0: 7320 7061 6972 732e 0a0a 2323 2320 5069  s pairs...### Pi
+000055d0: 7065 6c69 6e65 2036 0a2a 2a43 6f6e 7374  peline 6.**Const
+000055e0: 7275 6374 696f 6e20 6f66 2061 6765 2064  ruction of age d
+000055f0: 6973 7472 6962 7574 696f 6e20 7769 7468  istribution with
+00005600: 2063 6f6c 6c69 6e65 6172 6974 7920 616e   collinearity an
+00005610: 6420 5747 4420 6461 7469 6e67 2a2a 0a60  d WGD dating**.`
+00005620: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
+00005630: 6567 6961 5f63 6f65 7275 6c65 6120 2d6f  egia_coerulea -o
+00005640: 2077 6764 5f64 6d64 0a77 6764 206b 7364   wgd_dmd.wgd ksd
+00005650: 2077 6764 5f64 6d64 2f41 7175 696c 6567   wgd_dmd/Aquileg
+00005660: 6961 5f63 6f65 7275 6c65 612e 7473 7620  ia_coerulea.tsv 
+00005670: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00005680: 6561 202d 6f20 7767 645f 6b73 640a 7767  ea -o wgd_ksd.wg
+00005690: 6420 7379 6e20 2d66 206d 524e 4120 2d61  d syn -f mRNA -a
+000056a0: 204e 616d 6520 7767 645f 646d 642f 4171   Name wgd_dmd/Aq
+000056b0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000056c0: 2e74 7376 2041 7175 696c 6567 6961 5f63  .tsv Aquilegia_c
+000056d0: 6f65 7275 6c65 612e 6766 6633 202d 6b73  oerulea.gff3 -ks
+000056e0: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
+000056f0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00005700: 6b73 2e74 7376 202d 6f20 7767 645f 7379  ks.tsv -o wgd_sy
+00005710: 6e0a 7767 6420 7065 616b 202d 2d68 6575  n.wgd peak --heu
+00005720: 7269 7374 6963 2077 6764 5f6b 7364 2f41  ristic wgd_ksd/A
+00005730: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00005740: 612e 7473 762e 6b73 2e74 7376 202d 6170  a.tsv.ks.tsv -ap
+00005750: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+00005760: 2d6f 7574 2f61 6e63 686f 7270 6f69 6e74  -out/anchorpoint
+00005770: 732e 7478 7420 2d73 6d20 7767 645f 7379  s.txt -sm wgd_sy
+00005780: 6e2f 6961 6468 6f72 652d 6f75 742f 7365  n/iadhore-out/se
+00005790: 676d 656e 7473 2e74 7874 202d 6c65 2077  gments.txt -le w
+000057a0: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+000057b0: 7574 2f6c 6973 745f 656c 656d 656e 7473  ut/list_elements
+000057c0: 2e74 7874 202d 6d70 2077 6764 5f73 796e  .txt -mp wgd_syn
+000057d0: 2f69 6164 686f 7265 2d6f 7574 2f6d 756c  /iadhore-out/mul
+000057e0: 7469 706c 6963 6f6e 5f70 6169 7273 2e74  tiplicon_pairs.t
+000057f0: 7874 202d 6e20 3120 3420 2d6b 6320 3320  xt -n 1 4 -kc 3 
+00005800: 2d6f 2077 6764 5f70 6561 6b0a 7767 6420  -o wgd_peak.wgd 
+00005810: 646d 6420 2d66 2041 7175 696c 6567 6961  dmd -f Aquilegia
+00005820: 5f63 6f65 7275 6c65 6120 2d61 7020 7767  _coerulea -ap wg
+00005830: 645f 7065 616b 2f41 6e63 686f 724b 735f  d_peak/AnchorKs_
+00005840: 4669 6e64 5065 616b 2f41 7175 696c 6567  FindPeak/Aquileg
+00005850: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00005860: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
+00005870: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
+00005880: 7465 645f 666f 726d 6174 2e74 7376 202d  ted_format.tsv -
+00005890: 6f20 7767 645f 646d 645f 6f72 7468 6f20  o wgd_dmd_ortho 
+000058a0: 506f 7461 6d6f 6765 746f 6e5f 6163 7574  Potamogeton_acut
+000058b0: 6966 6f6c 6975 7320 5370 6972 6f64 656c  ifolius Spirodel
+000058c0: 615f 696e 7465 726d 6564 6961 2041 6d6f  a_intermedia Amo
+000058d0: 7270 686f 7068 616c 6c75 735f 6b6f 6e6a  rphophallus_konj
+000058e0: 6163 2041 6361 6e74 686f 6368 6c61 6d79  ac Acanthochlamy
+000058f0: 735f 6272 6163 7465 6174 6120 4469 6f73  s_bracteata Dios
+00005900: 636f 7265 615f 616c 6174 6120 4469 6f73  corea_alata Dios
+00005910: 636f 7265 615f 726f 7475 6e64 6174 6120  corea_rotundata 
+00005920: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+00005930: 7320 4163 6f72 7573 5f74 6174 6172 696e  s Acorus_tatarin
+00005940: 6f77 6969 2054 6574 7261 6365 6e74 726f  owii Tetracentro
+00005950: 6e5f 7369 6e65 6e73 6520 5472 6f63 686f  n_sinense Trocho
+00005960: 6465 6e64 726f 6e5f 6172 616c 696f 6964  dendron_aralioid
+00005970: 6573 2042 7578 7573 5f61 7573 7472 6f79  es Buxus_austroy
+00005980: 756e 6e61 6e65 6e73 6973 2042 7578 7573  unnanensis Buxus
+00005990: 5f73 696e 6963 6120 4e65 6c75 6d62 6f5f  _sinica Nelumbo_
+000059a0: 6e75 6369 6665 7261 2054 656c 6f70 6561  nucifera Telopea
+000059b0: 5f73 7065 6369 6f73 6973 7369 6d61 2050  _speciosissima P
+000059c0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
+000059d0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+000059e0: 6c65 610a 7767 6420 666f 6375 7320 2d2d  lea.wgd focus --
+000059f0: 7072 6f74 636f 6364 6174 696e 6720 2d2d  protcocdating --
+00005a00: 6161 6d6f 6465 6c20 6c67 2077 6764 5f64  aamodel lg wgd_d
+00005a10: 6d64 5f6f 7274 686f 2f6d 6572 6765 5f66  md_ortho/merge_f
+00005a20: 6f63 7573 5f61 702e 7473 7620 2d73 7020  ocus_ap.tsv -sp 
+00005a30: 6461 7469 6e67 5f74 7265 652e 6e77 202d  dating_tree.nw -
+00005a40: 6f20 7767 645f 6461 7469 6e67 202d 6420  o wgd_dating -d 
+00005a50: 6d63 6d63 7472 6565 202d 6473 2027 6275  mcmctree -ds 'bu
+00005a60: 726e 696e 203d 2032 3030 3027 202d 6473  rnin = 2000' -ds
+00005a70: 2027 7361 6d70 6672 6571 203d 2031 3030   'sampfreq = 100
+00005a80: 3027 202d 6473 2027 6e73 616d 706c 6520  0' -ds 'nsample 
+00005a90: 3d20 3230 3030 3027 2050 6f74 616d 6f67  = 20000' Potamog
+00005aa0: 6574 6f6e 5f61 6375 7469 666f 6c69 7573  eton_acutifolius
+00005ab0: 2053 7069 726f 6465 6c61 5f69 6e74 6572   Spirodela_inter
+00005ac0: 6d65 6469 6120 416d 6f72 7068 6f70 6861  media Amorphopha
+00005ad0: 6c6c 7573 5f6b 6f6e 6a61 6320 4163 616e  llus_konjac Acan
+00005ae0: 7468 6f63 686c 616d 7973 5f62 7261 6374  thochlamys_bract
+00005af0: 6561 7461 2044 696f 7363 6f72 6561 5f61  eata Dioscorea_a
+00005b00: 6c61 7461 2044 696f 7363 6f72 6561 5f72  lata Dioscorea_r
+00005b10: 6f74 756e 6461 7461 2041 636f 7275 735f  otundata Acorus_
+00005b20: 616d 6572 6963 616e 7573 2041 636f 7275  americanus Acoru
+00005b30: 735f 7461 7461 7269 6e6f 7769 6920 5465  s_tatarinowii Te
+00005b40: 7472 6163 656e 7472 6f6e 5f73 696e 656e  tracentron_sinen
+00005b50: 7365 2054 726f 6368 6f64 656e 6472 6f6e  se Trochodendron
+00005b60: 5f61 7261 6c69 6f69 6465 7320 4275 7875  _aralioides Buxu
+00005b70: 735f 6175 7374 726f 7975 6e6e 616e 656e  s_austroyunnanen
+00005b80: 7369 7320 4275 7875 735f 7369 6e69 6361  sis Buxus_sinica
+00005b90: 204e 656c 756d 626f 5f6e 7563 6966 6572   Nelumbo_nucifer
+00005ba0: 6120 5465 6c6f 7065 615f 7370 6563 696f  a Telopea_specio
+00005bb0: 7369 7373 696d 6120 5072 6f74 6561 5f63  sissima Protea_c
+00005bc0: 796e 6172 6f69 6465 7320 4171 7569 6c65  ynaroides Aquile
+00005bd0: 6769 615f 636f 6572 756c 6561 0a60 6060  gia_coerulea.```
+00005be0: 0a60 6060 0a23 2077 6861 7420 7769 6c6c  .```.# what will
+00005bf0: 2062 6520 696e 2074 6865 2077 6764 5f64   be in the wgd_d
+00005c00: 6174 696e 6720 7265 7375 6c74 2064 6972  ating result dir
+00005c10: 6563 746f 7279 0a2d 7767 645f 6461 7469  ectory.-wgd_dati
+00005c20: 6e67 0a2d 2d43 6f6e 6361 7465 6e61 7465  ng.--Concatenate
+00005c30: 642e 7061 6c6e 0a2d 2d43 6f6e 6361 7465  d.paln.--Concate
+00005c40: 6e61 7465 642e 7061 6c6e 2e70 616d 6c0a  nated.paln.paml.
+00005c50: 2d2d 4732 532e 4d61 700a 2d2d 4746 3030  --G2S.Map.--GF00
+00005c60: 3030 3030 3031 2e70 616c 6e0a 2d2d 2e2e  000001.paln.--..
+00005c70: 0a2d 2d47 4630 3030 3030 3138 372e 7061  .--GF00000187.pa
+00005c80: 6c6e 0a2d 2d6d 636d 6374 7265 650a 2d2d  ln.--mcmctree.--
+00005c90: 2d43 6f6e 6361 7465 6e61 7465 640a 2d2d  -Concatenated.--
+00005ca0: 2d2d 7065 700a 2d2d 2d2d 2d43 6f6e 6361  --pep.-----Conca
+00005cb0: 7465 6e61 7465 642e 7061 6c6e 2e70 616d  tenated.paln.pam
+00005cc0: 6c0a 2d2d 2d2d 2d64 6174 696e 675f 7472  l.-----dating_tr
+00005cd0: 6565 2e6e 770a 2d2d 2d2d 2d46 6967 5472  ee.nw.-----FigTr
+00005ce0: 6565 2e74 7265 0a2d 2d2d 2d2d 696e 2e42  ee.tre.-----in.B
+00005cf0: 560a 2d2d 2d2d 2d6c 672e 6461 740a 2d2d  V.-----lg.dat.--
+00005d00: 2d2d 2d6c 6e66 0a2d 2d2d 2d2d 6d63 6d63  ---lnf.-----mcmc
+00005d10: 7472 6565 2e63 7472 6c0a 2d2d 2d2d 2d6d  tree.ctrl.-----m
+00005d20: 636d 6374 7265 652e 6f75 740a 2d2d 2d2d  cmctree.out.----
+00005d30: 2d6d 636d 632e 7478 740a 2d2d 2d2d 2d72  -mcmc.txt.-----r
+00005d40: 6174 6573 0a2d 2d2d 2d2d 7273 740a 2d2d  ates.-----rst.--
+00005d50: 2d2d 2d72 7374 310a 2d2d 2d2d 2d72 7562  ---rst1.-----rub
+00005d60: 0a2d 2d2d 2d2d 746d 7030 3030 312e 6374  .-----tmp0001.ct
+00005d70: 6c0a 2d2d 2d2d 2d74 6d70 3030 3031 2e6f  l.-----tmp0001.o
+00005d80: 7574 0a2d 2d2d 2d2d 746d 7030 3030 312e  ut.-----tmp0001.
+00005d90: 7472 6565 730a 2d2d 2d2d 2d74 6d70 3030  trees.-----tmp00
+00005da0: 3031 2e74 7874 0a60 6060 0a54 6865 2060  01.txt.```.The `
+00005db0: 436f 6e63 6174 656e 6174 6564 2e70 616c  Concatenated.pal
+00005dc0: 6e60 2061 6e64 2060 436f 6e63 6174 656e  n` and `Concaten
+00005dd0: 6174 6564 2e70 616c 6e2e 7061 6d6c 6020  ated.paln.paml` 
+00005de0: 6172 6520 7468 6520 636f 6e63 6174 656e  are the concaten
+00005df0: 6174 6564 2070 726f 7465 696e 2061 6c69  ated protein ali
+00005e00: 676e 6d65 6e74 7320 696e 2066 6173 7461  gnments in fasta
+00005e10: 2061 6e64 2070 616d 6c20 666f 726d 6174   and paml format
+00005e20: 2e20 5468 6520 6047 3253 2e4d 6170 6020  . The `G2S.Map` 
+00005e30: 6973 2074 6865 206d 6170 2062 6574 7765  is the map betwe
+00005e40: 656e 2067 656e 6520 616e 6420 7370 6563  en gene and spec
+00005e50: 6965 7320 6e61 6d65 732e 2054 6865 2060  ies names. The `
+00005e60: 4746 3030 3030 3030 3031 2e70 616c 6e60  GF00000001.paln`
+00005e70: 2c2e 2e20 616e 6420 6047 4630 3030 3030  ,.. and `GF00000
+00005e80: 3138 372e 7061 6c6e 6020 6172 6520 7468  187.paln` are th
+00005e90: 6520 7072 6f74 6569 6e20 616c 6967 6e6d  e protein alignm
+00005ea0: 656e 7473 2066 6f72 2065 6163 6820 6765  ents for each ge
+00005eb0: 6e65 2066 616d 696c 792e 2054 6865 2060  ne family. The `
+00005ec0: 6d63 6d63 7472 6565 6020 7375 6266 6f6c  mcmctree` subfol
+00005ed0: 6465 7220 636f 6e74 6169 6e73 2074 6865  der contains the
+00005ee0: 2064 6174 696e 6720 7265 7375 6c74 7320   dating results 
+00005ef0: 666f 7220 7468 6520 636f 6e63 6174 656e  for the concaten
+00005f00: 6174 6564 2066 616d 696c 7920 2861 6e64  ated family (and
+00005f10: 2070 6572 2067 656e 6520 6661 6d69 6c79   per gene family
+00005f20: 2069 6620 7365 7429 2e20 5468 6520 6465   if set). The de
+00005f30: 6570 6572 2060 436f 6e63 6174 656e 6174  eper `Concatenat
+00005f40: 6564 6020 7375 6266 6f6c 6465 7220 636f  ed` subfolder co
+00005f50: 6e74 6169 6e73 2074 6865 2064 6174 696e  ntains the datin
+00005f60: 6720 7265 7375 6c74 7320 6f66 2063 6f6e  g results of con
+00005f70: 6361 7465 6e61 7465 6420 7072 6f74 6569  catenated protei
+00005f80: 6e20 616c 6967 6e6d 656e 7420 286f 7220  n alignment (or 
+00005f90: 6e75 636c 656f 7469 6465 2061 6c69 676e  nucleotide align
+00005fa0: 6d65 6e74 2069 6620 7365 7429 2e20 5468  ment if set). Th
+00005fb0: 6520 6465 6570 6573 7420 6070 6570 6020  e deepest `pep` 
+00005fc0: 7375 6266 6f6c 6465 7220 636f 6e74 6169  subfolder contai
+00005fd0: 6e73 2074 6865 2066 696e 616c 2064 6174  ns the final dat
+00005fe0: 696e 6720 7265 7375 6c74 7320 286f 6620  ing results (of 
+00005ff0: 636f 6e63 6174 656e 6174 6564 2070 726f  concatenated pro
+00006000: 7465 696e 2061 6c69 676e 6d65 6e74 2069  tein alignment i
+00006010: 6e20 7468 6973 2063 6173 6529 2e20 506c  n this case). Pl
+00006020: 6561 7365 2072 6566 6572 2074 6f20 606d  ease refer to `m
+00006030: 636d 6374 7265 6560 206d 616e 7561 6c20  cmctree` manual 
+00006040: 666f 7220 6465 7461 696c 6564 2064 6573  for detailed des
+00006050: 6372 6970 7469 6f6e 206f 6620 6561 6368  cription of each
+00006060: 2066 696c 6520 7072 6f64 7563 6564 2062   file produced b
+00006070: 7920 606d 636d 6374 7265 6560 2e20 5468  y `mcmctree`. Th
+00006080: 6520 696d 706f 7274 616e 7420 7265 7375  e important resu
+00006090: 6c74 2066 696c 6573 2061 7265 2060 4669  lt files are `Fi
+000060a0: 6754 7265 652e 7472 6560 2c20 606d 636d  gTree.tre`, `mcm
+000060b0: 6374 7265 652e 6f75 7460 2061 6e64 2060  ctree.out` and `
+000060c0: 6d63 6d63 2e74 7874 6020 7768 6963 6820  mcmc.txt` which 
+000060d0: 646f 6375 6d65 6e74 2074 6865 2066 696e  document the fin
+000060e0: 616c 2064 6174 6520 6573 7469 6d61 7469  al date estimati
+000060f0: 6f6e 2c20 6c6f 6720 696e 666f 726d 6174  on, log informat
+00006100: 696f 6e20 616e 6420 706f 7374 6572 696f  ion and posterio
+00006110: 7220 7361 6d70 6c65 7320 666f 7220 6561  r samples for ea
+00006120: 6368 206e 6f64 6520 7265 7370 6563 7469  ch node respecti
+00006130: 7665 6c79 2e0a 200a 2323 2050 6172 616d  vely.. .## Param
+00006140: 6574 6572 730a 0a54 6865 7265 2061 7265  eters..There are
+00006150: 2037 206d 6169 6e20 7072 6f67 7261 6d73   7 main programs
+00006160: 2069 6e20 6077 6764 2076 3260 3a20 6064   in `wgd v2`: `d
+00006170: 6d64 602c 6066 6f63 7573 602c 606b 7364  md`,`focus`,`ksd
+00006180: 602c 606d 6978 602c 6070 6561 6b60 2c60  `,`mix`,`peak`,`
+00006190: 7379 6e60 2c60 7669 7a60 2e20 4865 7265  syn`,`viz`. Here
+000061a0: 6166 7465 7220 7765 2077 696c 6c20 7072  after we will pr
+000061b0: 6f76 6964 6520 6120 6465 7461 696c 6564  ovide a detailed
+000061c0: 2065 6c75 6369 6461 7469 6f6e 206f 6e20   elucidation on 
+000061d0: 6561 6368 206f 6620 7468 6520 7072 6f67  each of the prog
+000061e0: 7261 6d20 616e 6420 6974 7320 6173 736f  ram and its asso
+000061f0: 6369 6174 6564 2070 6172 616d 6574 6572  ciated parameter
+00006200: 732e 2050 6c65 6173 6520 7265 6665 7220  s. Please refer 
+00006210: 746f 2074 6865 205b 5573 6167 655d 2823  to the [Usage](#
+00006220: 7573 6167 6529 2066 6f72 2074 6865 2073  usage) for the s
+00006230: 6365 6e61 7269 6f73 2074 6f20 7768 6963  cenarios to whic
+00006240: 6820 6561 6368 2070 6172 616d 6574 6572  h each parameter
+00006250: 2061 7070 6c69 6573 2e0a 0a54 6865 2070   applies...The p
+00006260: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
+00006270: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+00006280: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
+00006290: 7768 6f6c 6520 7061 7261 6e6f 6d65 2c20  whole paranome, 
+000062a0: 5242 4873 2028 5265 6369 7072 6f63 616c  RBHs (Reciprocal
+000062b0: 2042 6573 7420 4869 7473 292c 204d 5242   Best Hits), MRB
+000062c0: 4873 2028 4d75 6c74 6970 6c65 2052 6563  Hs (Multiple Rec
+000062d0: 6970 726f 6361 6c20 4265 7374 2048 6974  iprocal Best Hit
+000062e0: 7329 2c20 6f72 7468 6f67 726f 7570 7320  s), orthogroups 
+000062f0: 616e 6420 736f 6d65 206f 7468 6572 206f  and some other o
+00006300: 7274 686f 6772 6f75 702d 7265 6c61 7465  rthogroup-relate
+00006310: 6420 6675 6e63 7469 6f6e 732c 2069 6e63  d functions, inc
+00006320: 6c75 6469 6e67 2063 6972 6375 6d73 6372  luding circumscr
+00006330: 6970 7469 6f6e 206f 6620 6e65 7374 6564  iption of nested
+00006340: 2073 696e 676c 652d 636f 7079 206f 7274   single-copy ort
+00006350: 686f 6772 6f75 7073 2028 4e53 4f47 7329  hogroups (NSOGs)
+00006360: 2c20 756e 6269 6173 6564 2074 6573 7420  , unbiased test 
+00006370: 6f66 2073 696e 676c 652d 636f 7079 206f  of single-copy o
+00006380: 7274 686f 6772 6f75 7073 2028 534f 4773  rthogroups (SOGs
+00006390: 2920 6f76 6572 206d 6973 7369 6e67 2069  ) over missing i
+000063a0: 6e70 6172 616c 6f67 732c 2063 6f6e 7374  nparalogs, const
+000063b0: 7275 6374 696f 6e20 6f66 2042 5553 434f  ruction of BUSCO
+000063c0: 2d67 7569 6465 6420 7369 6e67 6c65 2d63  -guided single-c
+000063d0: 6f70 7920 6f72 7468 6f67 726f 7570 7320  opy orthogroups 
+000063e0: 2853 4f47 7329 2c61 6e64 2074 6865 2063  (SOGs),and the c
+000063f0: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
+00006400: 656e 6365 2069 6e66 6572 656e 6365 206f  ence inference o
+00006410: 6620 7068 796c 6f67 656e 792e 0a60 6060  f phylogeny..```
+00006420: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
+00006430: 6573 2028 6f70 7469 6f6e 290a 2d2d 2d2d  es (option).----
+00006440: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006450: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006460: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006470: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006480: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c  ------------.-o,
+00006490: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+000064a0: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+000064b0: 2064 6566 6175 6c74 2077 6764 5f64 6d64   default wgd_dmd
+000064c0: 0a2d 742c 202d 2d74 6d70 6469 722c 2074  .-t, --tmpdir, t
+000064d0: 6865 2074 656d 706f 7261 7279 2077 6f72  he temporary wor
+000064e0: 6b69 6e67 2064 6972 6563 746f 7279 2c20  king directory, 
+000064f0: 6465 6661 756c 7420 4e6f 6e65 2c20 6966  default None, if
+00006500: 204e 6f6e 6520 7761 7320 6769 7665 6e2c   None was given,
+00006510: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
+00006520: 6420 6265 2061 7373 6967 6e65 6420 7261  d be assigned ra
+00006530: 6e64 6f6d 206e 616d 6573 2069 6e20 6375  ndom names in cu
+00006540: 7272 656e 7420 6469 7265 6374 6f72 7920  rrent directory 
+00006550: 616e 6420 6175 746f 6d61 7465 6c79 2072  and automately r
+00006560: 656d 6f76 6564 2061 7420 7468 6520 636f  emoved at the co
+00006570: 6d70 6c65 7469 6f6e 206f 6620 7072 6f67  mpletion of prog
+00006580: 7261 6d2c 2065 6c73 6520 7468 6520 746d  ram, else the tm
+00006590: 7064 6972 2077 6f75 6c64 2062 6520 6b65  pdir would be ke
+000065a0: 7074 0a2d 702c 202d 2d70 726f 742c 2066  pt.-p, --prot, f
+000065b0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+000065c0: 6865 7220 7573 696e 6720 7072 6f74 6569  her using protei
+000065d0: 6e20 6f72 206e 7563 6c65 6f74 6964 6520  n or nucleotide 
+000065e0: 7365 7175 656e 6365 730a 2d63 2c20 2d2d  sequences.-c, --
+000065f0: 6373 636f 7265 2c20 7468 6520 632d 7363  cscore, the c-sc
+00006600: 6f72 6520 746f 2072 6573 7472 6963 7420  ore to restrict 
+00006610: 7468 6520 686f 6d6f 6c6f 6720 7369 6d69  the homolog simi
+00006620: 6c61 7269 7479 206f 6620 4d52 4248 732c  larity of MRBHs,
+00006630: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
+00006640: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
+00006650: 2c20 7468 6520 632d 7363 6f72 6520 6675  , the c-score fu
+00006660: 6e63 696f 6e20 776f 756c 646e 2774 2062  ncion wouldn't b
+00006670: 6520 6163 7469 7661 7465 642c 2065 6c73  e activated, els
+00006680: 6520 6578 7065 6374 696e 6720 6120 6465  e expecting a de
+00006690: 6369 6d61 6c20 7769 7468 696e 2074 6865  cimal within the
+000066a0: 2072 616e 6765 206f 6620 3020 616e 6420   range of 0 and 
+000066b0: 310a 2d49 2c20 2d2d 696e 666c 6174 696f  1.-I, --inflatio
+000066c0: 6e2c 2074 6865 2069 6e66 6c61 7469 6f6e  n, the inflation
+000066d0: 2066 6163 746f 7220 666f 7220 4d43 4c20   factor for MCL 
+000066e0: 7072 6f67 7261 6d2c 2064 6566 6175 6c74  program, default
+000066f0: 2032 2e30 2c20 7769 7468 2068 6967 6865   2.0, with highe
+00006700: 7220 7661 6c75 6520 6c65 6164 696e 6720  r value leading 
+00006710: 746f 206d 6f72 6520 6275 7420 736d 616c  to more but smal
+00006720: 6c65 7220 636c 7573 7465 7273 0a2d 652c  ler clusters.-e,
+00006730: 202d 2d65 7661 6c2c 2074 6865 2065 2d76   --eval, the e-v
+00006740: 616c 7565 2063 7574 2d6f 6666 2066 6f72  alue cut-off for
+00006750: 2073 696d 696c 6172 6974 7920 696e 2064   similarity in d
+00006760: 6961 6d6f 6e64 2061 6e64 2f6f 7220 686d  iamond and/or hm
+00006770: 6d65 722c 2064 6566 6175 6c74 2031 652d  mer, default 1e-
+00006780: 3130 0a2d 2d74 6f5f 7374 6f70 2c20 666c  10.--to_stop, fl
+00006790: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+000067a0: 6572 2074 6f20 7472 616e 736c 6174 6520  er to translate 
+000067b0: 7468 726f 7567 6820 5354 4f50 2063 6f64  through STOP cod
+000067c0: 6f6e 732c 2069 6620 7468 6520 666c 6167  ons, if the flag
+000067d0: 2077 6173 2073 6574 2c20 7472 616e 736c   was set, transl
+000067e0: 6174 696f 6e20 7769 6c6c 2062 6520 7465  ation will be te
+000067f0: 726d 696e 6174 6564 2061 7420 7468 6520  rminated at the 
+00006800: 6669 7273 7420 696e 2d66 7261 6d65 2073  first in-frame s
+00006810: 746f 7020 636f 646f 6e2c 2065 6c73 6520  top codon, else 
+00006820: 6120 6675 6c6c 2074 7261 6e73 6c61 7469  a full translati
+00006830: 6f6e 2063 6f6e 7469 6e75 696e 6720 6f6e  on continuing on
+00006840: 2070 6173 7369 6e67 2061 6e79 2073 746f   passing any sto
+00006850: 7020 636f 646f 6e73 2077 6f75 6c64 2062  p codons would b
+00006860: 6520 696e 6974 6961 7465 640a 2d2d 6364  e initiated.--cd
+00006870: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
+00006880: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
+00006890: 7472 616e 736c 6174 6520 7468 6520 636f  translate the co
+000068a0: 6d70 6c65 7465 2043 4453 2074 6861 7420  mplete CDS that 
+000068b0: 7374 6172 7473 2077 6974 6820 6120 7661  starts with a va
+000068c0: 6c69 6420 7374 6172 7420 636f 646f 6e20  lid start codon 
+000068d0: 616e 6420 6f6e 6c79 2063 6f6e 7461 696e  and only contain
+000068e0: 7320 6120 7369 6e67 6c65 2069 6e2d 6672  s a single in-fr
+000068f0: 616d 6520 7374 6f70 2063 6f64 6f6e 2061  ame stop codon a
+00006900: 7420 7468 6520 656e 6420 616e 6420 6d75  t the end and mu
+00006910: 7374 2062 6520 6469 7669 6461 626c 6520  st be dividable 
+00006920: 6279 2074 6872 6565 2c20 6966 2074 6865  by three, if the
+00006930: 2066 6c61 6720 7761 7320 7365 742c 206f   flag was set, o
+00006940: 6e6c 7920 7468 6520 636f 6d70 6c65 7465  nly the complete
+00006950: 2043 4453 2077 6f75 6c64 2062 6520 7472   CDS would be tr
+00006960: 616e 736c 6174 6564 0a2d 662c 202d 2d66  anslated.-f, --f
+00006970: 6f63 7573 2c20 7468 6520 7370 6563 6965  ocus, the specie
+00006980: 7320 746f 2062 6520 6d65 7267 6564 206f  s to be merged o
+00006990: 6e20 6c6f 6361 6c20 4d52 4248 732c 2064  n local MRBHs, d
+000069a0: 6566 6175 6c74 204e 6f6e 652c 2069 6620  efault None, if 
+000069b0: 4e6f 6e65 2077 6173 2067 6976 656e 2c20  None was given, 
+000069c0: 7468 6520 6c6f 6361 6c20 4d52 4248 7320  the local MRBHs 
+000069d0: 776f 756c 646e 2774 2062 6520 696e 6665  wouldn't be infe
+000069e0: 7272 6564 0a2d 6170 2c20 2d2d 616e 6368  rred.-ap, --anch
+000069f0: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
+00006a00: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
+00006a10: 2066 696c 6520 6672 6f6d 2069 2d61 6468   file from i-adh
+00006a20: 6f72 6520 666f 7220 636f 6e73 7472 7563  ore for construc
+00006a30: 7469 6e67 2074 6865 206f 7274 686f 6772  ting the orthogr
+00006a40: 6f75 7073 2077 6974 6820 616e 6368 6f72  oups with anchor
+00006a50: 2070 6169 7273 2c20 6465 6661 756c 7420   pairs, default 
+00006a60: 4e6f 6e65 0a2d 736d 2c20 2d2d 7365 676d  None.-sm, --segm
+00006a70: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
+00006a80: 7473 2064 6174 6166 696c 6520 7573 6564  ts datafile used
+00006a90: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
+00006aa0: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
+00006ab0: 6973 2069 6620 696e 6974 6961 7465 642c  is if initiated,
+00006ac0: 2064 6566 6175 6c74 204e 6f6e 650a 2d6c   default None.-l
+00006ad0: 652c 202d 2d6c 6973 7465 6c65 6d65 6e74  e, --listelement
+00006ae0: 732c 2074 6865 206c 6973 7473 6567 6d65  s, the listsegme
+00006af0: 6e74 7320 6461 7461 2066 696c 6520 7573  nts data file us
+00006b00: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
+00006b10: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+00006b20: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
+00006b30: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
+00006b40: 2d67 742c 202d 2d67 656e 6574 6162 6c65  -gt, --genetable
+00006b50: 2c20 7468 6520 6765 6e65 2074 6162 6c65  , the gene table
+00006b60: 2064 6174 6166 696c 6520 7573 6564 2069   datafile used i
+00006b70: 6e20 636f 6c6c 696e 6561 7220 636f 616c  n collinear coal
+00006b80: 6573 6365 6e63 6520 616e 616c 7973 6973  escence analysis
+00006b90: 2069 6620 696e 6974 6961 7465 642c 2064   if initiated, d
+00006ba0: 6566 6175 6c74 204e 6f6e 650a 2d63 6f63  efault None.-coc
+00006bb0: 2c20 2d2d 636f 6c6c 696e 6561 7263 6f61  , --collinearcoa
+00006bc0: 6c65 7363 656e 6365 2c20 666c 6167 206f  lescence, flag o
+00006bd0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00006be0: 6f20 696e 6974 6961 7465 2074 6865 2063  o initiate the c
+00006bf0: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
+00006c00: 656e 6365 2061 6e61 6c79 7369 732c 2069  ence analysis, i
+00006c10: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00006c20: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
+00006c30: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00006c40: 7465 640a 2d6b 662c 202d 2d6b 6565 7066  ted.-kf, --keepf
+00006c50: 6173 7461 2c20 666c 6167 206f 7074 696f  asta, flag optio
+00006c60: 6e2c 2077 6865 7468 6572 2074 6f20 6f75  n, whether to ou
+00006c70: 7470 7574 2074 6865 2073 6571 7565 6e63  tput the sequenc
+00006c80: 6520 696e 666f 726d 6174 696f 6e20 6f66  e information of
+00006c90: 204d 5242 4873 2c20 6966 2074 6865 2066   MRBHs, if the f
+00006ca0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00006cb0: 2073 6571 7565 6e63 6573 206f 6620 4d52   sequences of MR
+00006cc0: 4248 7320 776f 756c 6420 6265 2069 6e20  BHs would be in 
+00006cd0: 6f75 7470 7574 0a2d 6b64 2c20 2d2d 6b65  output.-kd, --ke
+00006ce0: 6570 6475 706c 6963 6174 6573 2c20 666c  epduplicates, fl
+00006cf0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+00006d00: 6572 2074 6f20 616c 6c6f 7720 7468 6520  er to allow the 
+00006d10: 7361 6d65 2067 656e 6520 746f 206f 6363  same gene to occ
+00006d20: 7572 2069 6e20 6469 6666 6572 656e 7420  ur in different 
+00006d30: 4d52 4248 7320 286f 6e6c 7920 6d65 616e  MRBHs (only mean
+00006d40: 696e 6766 756c 2077 6865 6e20 7468 6520  ingful when the 
+00006d50: 6373 636f 7265 2077 6173 2075 7365 6429  cscore was used)
+00006d60: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00006d70: 7320 7365 742c 2074 6865 2073 616d 6520  s set, the same 
+00006d80: 6765 6e65 2063 6f75 6c64 2062 6520 6173  gene could be as
+00006d90: 7369 676e 6564 2074 6f20 6469 6666 6572  signed to differ
+00006da0: 656e 7420 4d52 4248 730a 2d67 6d2c 202d  ent MRBHs.-gm, -
+00006db0: 2d67 6c6f 6261 6c6d 7262 682c 2066 6c61  -globalmrbh, fla
+00006dc0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00006dd0: 7220 746f 2069 6e69 7469 6174 6520 676c  r to initiate gl
+00006de0: 6f62 616c 204d 5242 4873 2063 6f6e 7374  obal MRBHs const
+00006df0: 7275 6374 696f 6e2c 2069 6620 7468 6520  ruction, if the 
+00006e00: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00006e10: 6520 2d2d 666f 6375 7320 6f70 7469 6f6e  e --focus option
+00006e20: 2077 6f75 6c64 2062 6520 6967 6e6f 7265   would be ignore
+00006e30: 6420 616e 6420 6f6e 6c79 2067 6c6f 6261  d and only globa
+00006e40: 6c20 4d52 4248 7320 776f 756c 6420 6265  l MRBHs would be
+00006e50: 2062 7569 6c74 0a2d 6e2c 202d 2d6e 7468   built.-n, --nth
+00006e60: 7265 6164 732c 2074 6865 206e 756d 6265  reads, the numbe
+00006e70: 7220 6f66 2074 6872 6561 6473 2074 6f20  r of threads to 
+00006e80: 7573 652c 2064 6566 6175 6c74 2034 0a2d  use, default 4.-
+00006e90: 6f69 2c20 2d2d 6f72 7468 6f69 6e66 6572  oi, --orthoinfer
+00006ea0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00006eb0: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
+00006ec0: 7465 206f 7274 686f 6772 6f75 7020 696e  te orthogroup in
+00006ed0: 6665 726e 6563 652c 2069 6620 7468 6520  fernece, if the 
+00006ee0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00006ef0: 6520 6f72 7468 6f67 726f 7570 2069 6e66  e orthogroup inf
+00006f00: 6572 6e65 6365 2070 726f 6772 616d 2077  ernece program w
+00006f10: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00006f20: 640a 2d6f 6f2c 202d 2d6f 6e6c 796f 7274  d.-oo, --onlyort
+00006f30: 686f 2c20 666c 6167 206f 7074 696f 6e2c  ho, flag option,
+00006f40: 2077 6865 7468 6572 2074 6f20 6f6e 6c79   whether to only
+00006f50: 2063 6f6e 6475 6374 206f 7274 686f 6772   conduct orthogr
+00006f60: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
+00006f70: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00006f80: 6574 2c20 6f6e 6c79 2074 6865 206f 7274  et, only the ort
+00006f90: 686f 6772 6f75 7020 696e 6665 726e 6563  hogroup infernec
+00006fa0: 6520 7069 7065 6c69 6e65 2077 6f75 6c64  e pipeline would
+00006fb0: 2062 6520 7065 7266 6f72 6d65 6420 7768   be performed wh
+00006fc0: 696c 6520 7468 6520 6f74 6865 7220 616e  ile the other an
+00006fd0: 616c 7973 6973 2077 6f75 6c64 6e27 7420  alysis wouldn't 
+00006fe0: 6265 2069 6e69 7469 6174 6564 0a2d 676e  be initiated.-gn
+00006ff0: 2c20 2d2d 6765 746e 736f 672c 2066 6c61  , --getnsog, fla
+00007000: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00007010: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+00007020: 6520 7365 6172 6368 696e 6720 666f 7220  e searching for 
+00007030: 6e65 7374 6564 2073 696e 676c 652d 636f  nested single-co
+00007040: 7079 2067 656e 6520 6661 6d69 6c69 6573  py gene families
+00007050: 2028 4e53 4f47 7329 2028 6f6e 6c79 206d   (NSOGs) (only m
+00007060: 6561 6e69 6e67 6675 6c20 7768 656e 2074  eaningful when t
+00007070: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+00007080: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
+00007090: 2077 6173 2061 6374 6976 6174 6564 292c   was activated),
+000070a0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+000070b0: 2073 6574 2c20 6164 6469 7469 6f6e 616c   set, additional
+000070c0: 204e 534f 4773 2061 6e61 6c79 7369 7320   NSOGs analysis 
+000070d0: 776f 756c 6420 6265 2070 6572 666f 726d  would be perform
+000070e0: 6564 2062 6573 6964 6573 2074 6865 2062  ed besides the b
+000070f0: 6173 6963 206f 7274 686f 6772 6f75 7020  asic orthogroup 
+00007100: 696e 6665 726e 6563 650a 2d74 7265 652c  infernece.-tree,
+00007110: 202d 2d74 7265 655f 6d65 7468 6f64 2c20   --tree_method, 
+00007120: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
+00007130: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
+00007140: 6d20 746f 2069 6e76 6f6b 6520 286f 6e6c  m to invoke (onl
+00007150: 7920 6d65 616e 696e 6766 756c 2077 6865  y meaningful whe
+00007160: 6e20 7468 6520 636f 6c6c 696e 6561 7220  n the collinear 
+00007170: 636f 616c 6573 6365 6e63 652c 2067 656e  coalescence, gen
+00007180: 652d 746f 2d66 616d 696c 7920 6173 7369  e-to-family assi
+00007190: 676e 6d65 6e74 206f 7220 4e53 4f47 7320  gnment or NSOGs 
+000071a0: 616e 616c 7973 6973 2077 6572 6520 6163  analysis were ac
+000071b0: 7469 7661 7465 6429 2c20 6465 6661 756c  tivated), defaul
+000071c0: 7420 6661 7374 7472 6565 0a2d 7473 2c20  t fasttree.-ts, 
+000071d0: 2d2d 7472 6565 7365 742c 2074 6865 2070  --treeset, the p
+000071e0: 6172 616d 6574 6572 7320 7365 7474 696e  arameters settin
+000071f0: 6720 666f 7220 6765 6e65 2074 7265 6520  g for gene tree 
+00007200: 696e 6665 7265 6e63 652c 2064 6566 6175  inference, defau
+00007210: 6c74 204e 6f6e 652c 2074 6869 7320 6f70  lt None, this op
+00007220: 7469 6f6e 2063 616e 2062 6520 7072 6f76  tion can be prov
+00007230: 6964 6564 206d 756c 7469 706c 6520 7469  ided multiple ti
+00007240: 6d65 730a 2d6d 632c 202d 2d6d 736f 6763  mes.-mc, --msogc
+00007250: 7574 2c20 7468 6520 7261 7469 6f20 6375  ut, the ratio cu
+00007260: 746f 6666 2066 6f72 206d 6f73 746c 7920  toff for mostly 
+00007270: 7369 6e67 6c65 2d63 6f70 7920 6661 6d69  single-copy fami
+00007280: 6c79 2028 6d65 616e 696e 6766 756c 2077  ly (meaningful w
+00007290: 6865 6e20 6163 7469 7661 7469 6e67 2074  hen activating t
+000072a0: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+000072b0: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
+000072c0: 2920 616e 6420 7370 6563 6965 7320 7265  ) and species re
+000072d0: 7072 6573 656e 7461 7469 6f6e 2069 6e20  presentation in 
+000072e0: 636f 6c6c 696e 6561 7220 636f 616c 6573  collinear coales
+000072f0: 6365 6e63 6520 616e 616c 7973 6973 2c20  cence analysis, 
+00007300: 6465 6661 756c 7420 302e 382e 0a2d 6761  default 0.8..-ga
+00007310: 2c20 2d2d 6765 6e65 6173 7369 676e 2c20  , --geneassign, 
+00007320: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00007330: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+00007340: 2074 6865 2067 656e 652d 746f 2d66 616d   the gene-to-fam
+00007350: 696c 7920 6173 7369 676e 6d65 6e74 2061  ily assignment a
+00007360: 6e61 6c79 7369 732c 2069 6620 7468 6520  nalysis, if the 
+00007370: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00007380: 6520 616e 616c 7973 6973 2077 6f75 6c64  e analysis would
+00007390: 2062 6520 696e 6974 6961 7465 640a 2d73   be initiated.-s
+000073a0: 612c 202d 2d73 6571 3261 7373 6967 6e2c  a, --seq2assign,
+000073b0: 2074 6865 2071 7565 7269 6564 2073 6571   the queried seq
+000073c0: 7565 6e63 6573 2064 6174 6120 6669 6c65  uences data file
+000073d0: 2069 6e20 6765 6e65 2d74 6f2d 6661 6d69   in gene-to-fami
+000073e0: 6c79 2061 7373 6967 6e6d 656e 7420 616e  ly assignment an
+000073f0: 616c 7973 6973 2c20 6465 6661 756c 7420  alysis, default 
+00007400: 4e6f 6e65 2c20 7468 6973 206f 7074 696f  None, this optio
+00007410: 6e20 6361 6e20 6265 2070 726f 7669 6465  n can be provide
+00007420: 6420 6d75 6c74 6970 6c65 2074 696d 6573  d multiple times
+00007430: 0a2d 6661 2c20 2d2d 6661 6d32 6173 7369  .-fa, --fam2assi
+00007440: 676e 2c20 7468 6520 7175 6572 6965 6420  gn, the queried 
+00007450: 6661 6d69 6c69 7920 6461 7461 2066 696c  familiy data fil
+00007460: 6520 696e 2067 656e 652d 746f 2d66 616d  e in gene-to-fam
+00007470: 696c 7920 6173 7369 676e 6d65 6e74 2061  ily assignment a
+00007480: 6e61 6c79 7369 732c 2064 6566 6175 6c74  nalysis, default
+00007490: 204e 6f6e 650a 2d63 632c 202d 2d63 6f6e   None.-cc, --con
+000074a0: 6361 742c 2066 6c61 6720 6f70 7469 6f6e  cat, flag option
+000074b0: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
+000074c0: 7469 6174 6520 7468 6520 636f 6e63 6174  tiate the concat
+000074d0: 656e 6174 696f 6e20 7069 7065 6c69 6e65  enation pipeline
+000074e0: 2066 6f72 206f 7274 686f 6772 6f75 7020   for orthogroup 
+000074f0: 696e 6665 726e 6563 652c 2069 6620 7468  infernece, if th
+00007500: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00007510: 7468 6520 616e 616c 7973 6973 2077 6f75  the analysis wou
+00007520: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+00007530: 2d74 652c 202d 2d74 6573 7473 6f67 2c20  -te, --testsog, 
+00007540: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00007550: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+00007560: 2074 6865 2075 6e62 6961 7365 6420 7465   the unbiased te
+00007570: 7374 206f 6620 7369 6e67 6c65 2d63 6f70  st of single-cop
+00007580: 7920 6765 6e65 2066 616d 696c 6965 732c  y gene families,
+00007590: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+000075a0: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
+000075b0: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
+000075c0: 6961 7465 640a 2d62 732c 202d 2d62 696e  iated.-bs, --bin
+000075d0: 732c 2074 6865 206e 756d 6265 7220 6f66  s, the number of
+000075e0: 2062 696e 7320 6469 7669 6465 6420 696e   bins divided in
+000075f0: 2074 6865 2067 656e 6520 6c65 6e67 7468   the gene length
+00007600: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
+00007610: 6465 6661 756c 7420 3130 300a 2d6e 702c  default 100.-np,
+00007620: 202d 2d6e 6f72 6d61 6c69 7a65 6470 6572   --normalizedper
+00007630: 6365 6e74 2c20 7468 6520 7065 7263 656e  cent, the percen
+00007640: 7461 6765 206f 6620 7570 7065 7220 6869  tage of upper hi
+00007650: 7473 2075 7365 6420 666f 7220 6765 6e65  ts used for gene
+00007660: 206c 656e 6774 6820 6e6f 726d 616c 697a   length normaliz
+00007670: 6174 696f 6e2c 2064 6566 6175 6c74 2035  ation, default 5
+00007680: 0a2d 6e6e 2c20 2d2d 6e6f 6e6f 726d 616c  .-nn, --nonormal
+00007690: 697a 6174 696f 6e2c 2066 6c61 6720 6f70  ization, flag op
+000076a0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+000076b0: 2063 616c 6c20 6f66 6620 7468 6520 6e6f   call off the no
+000076c0: 726d 616c 697a 6174 696f 6e2c 2069 6620  rmalization, if 
+000076d0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+000076e0: 2c20 6e6f 206e 6f72 6d61 6c69 7a61 7469  , no normalizati
+000076f0: 6f6e 2077 6f75 6c64 2062 6520 636f 6e64  on would be cond
+00007700: 7563 7465 640a 2d62 736f 672c 202d 2d62  ucted.-bsog, --b
+00007710: 7573 636f 736f 672c 2066 6c61 6720 6f70  uscosog, flag op
+00007720: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00007730: 2069 6e69 7469 6174 6520 7468 6520 6275   initiate the bu
+00007740: 7363 6f2d 6775 6964 6564 2073 696e 676c  sco-guided singl
+00007750: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
+00007760: 6c79 2061 6e61 6c79 7369 732c 2069 6620  ly analysis, if 
+00007770: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00007780: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
+00007790: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+000077a0: 640a 2d62 686d 6d2c 202d 2d62 7573 636f  d.-bhmm, --busco
+000077b0: 686d 6d2c 2074 6865 2048 4d4d 2070 726f  hmm, the HMM pro
+000077c0: 6669 6c65 2064 6174 6166 696c 6520 696e  file datafile in
+000077d0: 2074 6865 2062 7573 636f 2d67 7569 6465   the busco-guide
+000077e0: 6420 7369 6e67 6c65 2d63 6f70 7920 6765  d single-copy ge
+000077f0: 6e65 2066 616d 696c 7920 616e 616c 7973  ne family analys
+00007800: 6973 2c20 6465 6661 756c 7420 4e6f 6e65  is, default None
+00007810: 0a2d 6263 7466 2c20 2d2d 6275 7363 6f63  .-bctf, --buscoc
+00007820: 7574 6f66 662c 2074 6865 2048 4d4d 2073  utoff, the HMM s
+00007830: 636f 7265 2063 7574 6f66 6620 6461 7461  core cutoff data
+00007840: 6669 6c65 2069 6e20 7468 6520 6275 7363  file in the busc
+00007850: 6f2d 6775 6964 6564 2073 696e 676c 652d  o-guided single-
+00007860: 636f 7079 2067 656e 6520 6661 6d69 6c79  copy gene family
+00007870: 2061 6e61 6c79 7369 732c 2064 6566 6175   analysis, defau
+00007880: 6c74 204e 6f6e 650a 2d6f 6620 2c2d 2d6f  lt None.-of ,--o
+00007890: 6766 6f72 6d61 742c 2066 6c61 6720 6f70  gformat, flag op
+000078a0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+000078b0: 2061 6464 2069 6e64 6578 2074 6f20 7468   add index to th
+000078c0: 6520 5242 4820 6661 6d69 6c69 6573 0a60  e RBH families.`
+000078d0: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+000078e0: 6077 6764 2066 6f63 7573 6020 6361 6e20  `wgd focus` can 
+000078f0: 7265 616c 697a 6520 7468 6520 636f 6e63  realize the conc
+00007900: 6174 656e 6174 696f 6e2d 6261 7365 6420  atenation-based 
+00007910: 616e 6420 636f 616c 6573 6365 6e63 652d  and coalescence-
+00007920: 6261 7365 6420 7068 796c 6f67 656e 6574  based phylogenet
+00007930: 6963 2069 6e66 6572 656e 6365 2061 6e64  ic inference and
+00007940: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
+00007950: 7469 6e67 206f 6620 5747 4473 2065 7463  ting of WGDs etc
+00007960: 2e0a 6060 600a 7767 6420 666f 6375 7320  ..```.wgd focus 
+00007970: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
+00007980: 6573 2028 6f70 7469 6f6e 290a 2d2d 2d2d  es (option).----
+00007990: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000079a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000079b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000079c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000079d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c  ------------.-o,
+000079e0: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+000079f0: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+00007a00: 2064 6566 6175 6c74 2077 6764 5f66 6f63   default wgd_foc
+00007a10: 7573 0a2d 742c 202d 2d74 6d70 6469 722c  us.-t, --tmpdir,
+00007a20: 2074 6865 2074 656d 706f 7261 7279 2077   the temporary w
+00007a30: 6f72 6b69 6e67 2064 6972 6563 746f 7279  orking directory
+00007a40: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+00007a50: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
+00007a60: 6e2c 2074 6865 2074 6d70 6469 7220 7769  n, the tmpdir wi
+00007a70: 6c6c 2062 6520 6173 7369 676e 6564 2072  ll be assigned r
+00007a80: 616e 646f 6d20 6e61 6d65 7320 696e 2063  andom names in c
+00007a90: 7572 7265 6e74 2064 6972 6563 746f 7279  urrent directory
+00007aa0: 2061 6e64 2061 7574 6f6d 6174 656c 7920   and automately 
+00007ab0: 7265 6d6f 7665 6420 6174 2074 6865 2063  removed at the c
+00007ac0: 6f6d 706c 6574 696f 6e20 6f66 2070 726f  ompletion of pro
+00007ad0: 6772 616d 2c20 656c 7365 2074 6865 2074  gram, else the t
+00007ae0: 6d70 6469 7220 776f 756c 6420 6265 206b  mpdir would be k
+00007af0: 6570 740a 2d6e 2c20 2d2d 6e74 6872 6561  ept.-n, --nthrea
+00007b00: 6473 2c20 7468 6520 6e75 6d62 6572 206f  ds, the number o
+00007b10: 6620 7468 7265 6164 7320 746f 2075 7365  f threads to use
+00007b20: 2c20 6465 6661 756c 7420 340a 2d2d 746f  , default 4.--to
+00007b30: 5f73 746f 702c 2066 6c61 6720 6f70 7469  _stop, flag opti
+00007b40: 6f6e 2c20 7768 6574 6865 7220 746f 2074  on, whether to t
+00007b50: 7261 6e73 6c61 7465 2074 6872 6f75 6768  ranslate through
+00007b60: 2053 544f 5020 636f 646f 6e73 2c20 6966   STOP codons, if
+00007b70: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00007b80: 742c 2074 7261 6e73 6c61 7469 6f6e 2077  t, translation w
+00007b90: 696c 6c20 6265 2074 6572 6d69 6e61 7465  ill be terminate
+00007ba0: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
+00007bb0: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
+00007bc0: 6f6e 2c20 656c 7365 2061 2066 756c 6c20  on, else a full 
+00007bd0: 7472 616e 736c 6174 696f 6e20 636f 6e74  translation cont
+00007be0: 696e 7569 6e67 206f 6e20 7061 7374 2061  inuing on past a
+00007bf0: 6e79 2073 746f 7020 636f 646f 6e73 2077  ny stop codons w
+00007c00: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00007c10: 640a 2d2d 6364 732c 2066 6c61 6720 6f70  d.--cds, flag op
+00007c20: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00007c30: 206f 6e6c 7920 7472 616e 736c 6174 6520   only translate 
+00007c40: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
+00007c50: 2074 6861 7420 7374 6172 7473 2077 6974   that starts wit
+00007c60: 6820 6120 7661 6c69 6420 7374 6172 7420  h a valid start 
+00007c70: 636f 646f 6e20 616e 6420 6f6e 6c79 2063  codon and only c
+00007c80: 6f6e 7461 696e 7320 6120 7369 6e67 6c65  ontains a single
+00007c90: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
+00007ca0: 6f64 6f6e 2061 7420 7468 6520 656e 6420  odon at the end 
+00007cb0: 616e 6420 6d75 7374 2062 6520 6469 7669  and must be divi
+00007cc0: 6461 626c 6520 6279 2074 6872 6565 2c20  dable by three, 
+00007cd0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00007ce0: 7365 742c 206f 6e6c 7920 7468 6520 636f  set, only the co
+00007cf0: 6d70 6c65 7465 2043 4453 2077 6f75 6c64  mplete CDS would
+00007d00: 2062 6520 7472 616e 736c 6174 6564 0a2d   be translated.-
+00007d10: 2d73 7472 6970 5f67 6170 732c 2066 6c61  -strip_gaps, fla
+00007d20: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00007d30: 7220 746f 2064 726f 7020 616c 6c20 6761  r to drop all ga
+00007d40: 7073 2069 6e20 6d75 6c74 6970 6c65 2073  ps in multiple s
+00007d50: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
+00007d60: 742c 2069 6620 7468 6520 666c 6167 2077  t, if the flag w
+00007d70: 6173 2073 6574 2c20 616c 6c20 6761 7073  as set, all gaps
+00007d80: 2077 6f75 6c64 2062 6520 6472 6f70 7065   would be droppe
+00007d90: 640a 2d61 2c20 2d2d 616c 6967 6e65 722c  d.-a, --aligner,
+00007da0: 2077 6869 6368 2061 6c69 676e 6d65 6e74   which alignment
+00007db0: 2070 726f 6772 616d 2074 6f20 7573 652c   program to use,
+00007dc0: 2064 6566 6175 6c74 206d 6166 6674 0a2d   default mafft.-
+00007dd0: 7472 6565 2c20 2d2d 7472 6565 5f6d 6574  tree, --tree_met
+00007de0: 686f 642c 2077 6869 6368 2067 656e 6520  hod, which gene 
+00007df0: 7472 6565 2069 6e66 6572 656e 6365 2070  tree inference p
+00007e00: 726f 6772 616d 2074 6f20 696e 766f 6b65  rogram to invoke
+00007e10: 2c20 6465 6661 756c 7420 6661 7374 7472  , default fasttr
+00007e20: 6565 0a2d 7473 2c20 2d2d 7472 6565 7365  ee.-ts, --treese
+00007e30: 742c 2074 6865 2070 6172 616d 6574 6572  t, the parameter
+00007e40: 7320 7365 7474 696e 6720 666f 7220 6765  s setting for ge
+00007e50: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
+00007e60: 652c 2064 6566 6175 6c74 204e 6f6e 652c  e, default None,
+00007e70: 2074 6869 7320 6f70 7469 6f6e 2063 616e   this option can
+00007e80: 2062 6520 7072 6f76 6964 6564 206d 756c   be provided mul
+00007e90: 7469 706c 6520 7469 6d65 730a 2d2d 636f  tiple times.--co
+00007ea0: 6e63 6174 656e 6174 696f 6e2c 2066 6c61  ncatenation, fla
+00007eb0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00007ec0: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+00007ed0: 6520 636f 6e63 6174 656e 6174 696f 6e2d  e concatenation-
+00007ee0: 6261 7365 6420 7370 6563 6965 7320 7472  based species tr
+00007ef0: 6565 2069 6e66 6572 656e 6365 2c20 6966  ee inference, if
+00007f00: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00007f10: 742c 2063 6f6e 6361 7465 6e61 7469 6f6e  t, concatenation
+00007f20: 2d62 6173 6564 2073 7065 6369 6573 2074  -based species t
+00007f30: 7265 6520 776f 756c 6420 6265 2069 6e66  ree would be inf
+00007f40: 6572 6564 0a2d 2d63 6f61 6c65 7363 656e  ered.--coalescen
+00007f50: 6365 2c20 666c 6167 206f 7074 696f 6e2c  ce, flag option,
+00007f60: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
+00007f70: 6961 7465 2074 6865 2063 6f61 6c65 7363  iate the coalesc
+00007f80: 656e 6365 2d62 6173 6564 2073 7065 6369  ence-based speci
+00007f90: 6573 2074 7265 6520 696e 6665 7265 6e63  es tree inferenc
+00007fa0: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
+00007fb0: 6173 2073 6574 2c20 636f 616c 6573 6365  as set, coalesce
+00007fc0: 6e63 652d 6261 7365 6420 7370 6563 6965  nce-based specie
+00007fd0: 7320 7472 6565 2077 6f75 6c64 2062 6520  s tree would be 
+00007fe0: 696e 6665 7265 640a 2d73 702c 202d 2d73  infered.-sp, --s
+00007ff0: 7065 6369 6573 7472 6565 2c20 7370 6563  peciestree, spec
+00008000: 6965 7320 7472 6565 2064 6174 6166 696c  ies tree datafil
+00008010: 6520 666f 7220 6461 7469 6e67 2c20 6465  e for dating, de
+00008020: 6661 756c 7420 4e6f 6e65 0a2d 642c 202d  fault None.-d, -
+00008030: 2d64 6174 696e 672c 2077 6869 6368 206d  -dating, which m
+00008040: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
+00008050: 7072 6f67 7261 6d20 746f 2075 7365 2c20  program to use, 
+00008060: 6465 6661 756c 7420 6e6f 6e65 0a2d 6473  default none.-ds
+00008070: 2c20 2d2d 6461 7469 6e67 7365 742c 2074  , --datingset, t
+00008080: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
+00008090: 7474 696e 6720 666f 7220 6461 7469 6e67  tting for dating
+000080a0: 2070 726f 6772 616d 2c20 6465 6661 756c   program, defaul
+000080b0: 7420 4e6f 6e65 2c20 7468 6973 206f 7074  t None, this opt
+000080c0: 696f 6e20 6361 6e20 6265 2070 726f 7669  ion can be provi
+000080d0: 6465 6420 6d75 6c74 6970 6c65 2074 696d  ded multiple tim
+000080e0: 6573 0a2d 6e73 2c20 2d2d 6e73 6974 6573  es.-ns, --nsites
+000080f0: 2c20 7468 6520 6e73 6974 6573 2069 6e66  , the nsites inf
+00008100: 6f72 6d61 7469 6f6e 2066 6f72 2072 3873  ormation for r8s
+00008110: 2064 6174 696e 672c 2064 6566 6175 6c74   dating, default
+00008120: 204e 6f6e 650a 2d6f 742c 202d 2d6f 7574   None.-ot, --out
+00008130: 6772 6f75 702c 2074 6865 206f 7574 6772  group, the outgr
+00008140: 6f75 7020 696e 666f 726d 6174 696f 6e20  oup information 
+00008150: 666f 7220 7238 7320 6461 7469 6e67 2c20  for r8s dating, 
+00008160: 6465 6661 756c 7420 4e6f 6e65 0a2d 7074  default None.-pt
+00008170: 2c20 2d2d 7061 7274 6974 696f 6e2c 2066  , --partition, f
+00008180: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00008190: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+000081a0: 7061 7274 6974 696f 6e20 6461 7469 6e67  partition dating
+000081b0: 2061 6e61 6c79 7369 7320 666f 7220 636f   analysis for co
+000081c0: 646f 6e2c 2069 6620 7468 6520 666c 6167  don, if the flag
+000081d0: 2077 6173 2073 6574 2c20 616e 2061 6464   was set, an add
+000081e0: 6974 696f 6e61 6c20 7061 7274 6974 696f  itional partitio
+000081f0: 6e20 6461 7469 6e67 2061 6e61 6c79 7369  n dating analysi
+00008200: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
+00008210: 6174 6564 0a2d 616d 2c20 2d2d 6161 6d6f  ated.-am, --aamo
+00008220: 6465 6c2c 2077 6869 6368 2070 726f 7465  del, which prote
+00008230: 696e 206d 6f64 656c 2074 6f20 6265 2075  in model to be u
+00008240: 7365 6420 696e 206d 636d 6374 7265 652c  sed in mcmctree,
+00008250: 2064 6566 6175 6c74 2070 6f69 7373 6f6e   default poisson
+00008260: 0a2d 6b73 2c20 666c 6167 206f 7074 696f  .-ks, flag optio
+00008270: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00008280: 6974 6961 7465 204b 7320 6361 6c63 756c  itiate Ks calcul
+00008290: 6174 696f 6e20 666f 7220 686f 6d6f 6c6f  ation for homolo
+000082a0: 6775 6573 2069 6e20 7468 6520 7072 6f76  gues in the prov
+000082b0: 6964 6564 206f 7274 686f 6c6f 676f 7573  ided orthologous
+000082c0: 2067 656e 6520 6661 6d69 6c79 0a2d 2d61   gene family.--a
+000082d0: 6e6e 6f74 6174 696f 6e2c 2077 6869 6368  nnotation, which
+000082e0: 2061 6e6e 6f74 6174 696f 6e20 7072 6f67   annotation prog
+000082f0: 7261 6d20 746f 2075 7365 2c20 6465 6661  ram to use, defa
+00008300: 756c 7420 4e6f 6e65 0a2d 2d70 6169 7277  ult None.--pairw
+00008310: 6973 652c 2066 6c61 6720 6f70 7469 6f6e  ise, flag option
+00008320: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
+00008330: 7469 6174 6520 7061 6972 7769 7365 204b  tiate pairwise K
+00008340: 7320 6573 7469 6d61 7469 6f6e 2c20 6966  s estimation, if
+00008350: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00008360: 742c 2070 6169 7277 6973 6520 4b73 2076  t, pairwise Ks v
+00008370: 616c 7565 7320 776f 756c 6420 6265 2065  alues would be e
+00008380: 7374 696d 6174 6564 0a2d 6564 2c20 2d2d  stimated.-ed, --
+00008390: 6567 676e 6f67 6461 7461 2c20 7468 6520  eggnogdata, the 
+000083a0: 6567 676e 6f67 2061 6e6e 6f74 6174 696f  eggnog annotatio
+000083b0: 6e20 6461 7461 6669 6c65 2c20 6465 6661  n datafile, defa
+000083c0: 756c 7420 4e6f 6e65 0a2d 2d70 6661 6d2c  ult None.--pfam,
+000083d0: 2077 6869 6368 206f 7074 696f 6e20 746f   which option to
+000083e0: 2075 7365 2066 6f72 2070 6661 6d20 616e   use for pfam an
+000083f0: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
+00008400: 7420 4e6f 6e65 0a2d 2d64 6d6e 622c 2074  t None.--dmnb, t
+00008410: 6865 2064 6961 6d6f 6e64 2064 6174 6162  he diamond datab
+00008420: 6173 6520 666f 7220 616e 6e6f 7461 7469  ase for annotati
+00008430: 6f6e 2c20 6465 6661 756c 7420 4e6f 6e65  on, default None
+00008440: 0a2d 2d68 6d6d 2c20 7468 6520 484d 4d20  .--hmm, the HMM 
+00008450: 7072 6f66 696c 6520 666f 7220 616e 6e6f  profile for anno
+00008460: 7461 7469 6f6e 2c20 6465 6661 756c 7420  tation, default 
+00008470: 4e6f 6e65 0a2d 2d65 7661 6c75 652c 2074  None.--evalue, t
+00008480: 6865 2065 2d76 616c 7565 2063 7574 2d6f  he e-value cut-o
+00008490: 6666 2066 6f72 2061 6e6e 6f74 6174 696f  ff for annotatio
+000084a0: 6e2c 2064 6566 6175 6c74 2031 652d 3130  n, default 1e-10
+000084b0: 0a2d 2d65 7865 7061 7468 2c20 7468 6520  .--exepath, the 
+000084c0: 7061 7468 2074 6f20 7468 6520 696e 7465  path to the inte
+000084d0: 7270 726f 7363 616e 2065 7865 6375 7461  rproscan executa
+000084e0: 626c 652c 2064 6566 6175 6c74 204e 6f6e  ble, default Non
+000084f0: 650a 2d66 2c20 2d2d 666f 7373 696c 2c20  e.-f, --fossil, 
+00008500: 7468 6520 666f 7373 696c 2063 616c 6962  the fossil calib
+00008510: 7261 7469 6f6e 2069 6e66 6f72 6d61 7469  ration informati
+00008520: 6f6e 2069 6e20 4265 6173 742c 2064 6566  on in Beast, def
+00008530: 6175 6c74 2028 2763 6c61 6465 313b 636c  ault ('clade1;cl
+00008540: 6164 6532 272c 2027 7461 7861 312c 7461  ade2', 'taxa1,ta
+00008550: 7861 323b 7461 7861 332c 7461 7861 3427  xa2;taxa3,taxa4'
+00008560: 2c20 2734 3b35 272c 2027 302e 353b 302e  , '4;5', '0.5;0.
+00008570: 3627 2c20 2734 3030 3b35 3030 2729 0a2d  6', '400;500').-
+00008580: 7268 2c20 2d2d 726f 6f74 6865 6967 6874  rh, --rootheight
+00008590: 2c20 7468 6520 726f 6f74 2068 6569 6768  , the root heigh
+000085a0: 7420 6361 6c69 6272 6174 696f 6e20 696e  t calibration in
+000085b0: 666f 2069 6e20 4265 6173 742c 2064 6566  fo in Beast, def
+000085c0: 6175 6c74 2028 342c 302e 352c 3430 3029  ault (4,0.5,400)
+000085d0: 0a2d 6373 2c20 2d2d 6368 6169 6e73 6574  .-cs, --chainset
+000085e0: 2c20 7468 6520 7061 7261 6d65 7465 7273  , the parameters
+000085f0: 206f 6620 4d43 4d43 2063 6861 696e 2069   of MCMC chain i
+00008600: 6e20 4265 6173 742c 2064 6566 6175 6c74  n Beast, default
+00008610: 2028 3130 3030 302c 3130 3029 0a2d 2d62   (10000,100).--b
+00008620: 6561 7374 6c67 6a61 722c 2074 6865 2070  eastlgjar, the p
+00008630: 6174 6820 746f 2062 6561 7374 4c47 2e6a  ath to beastLG.j
+00008640: 6172 2c20 6465 6661 756c 7420 4e6f 6e65  ar, default None
+00008650: 0a2d 2d62 6561 676c 652c 2066 6c61 6720  .--beagle, flag 
+00008660: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00008670: 746f 2075 7365 2062 6561 676c 6520 696e  to use beagle in
+00008680: 2042 6561 7374 2c20 6966 2074 6865 2066   Beast, if the f
+00008690: 6c61 6720 7761 7320 7365 742c 2062 6561  lag was set, bea
+000086a0: 676c 6520 776f 756c 6420 6265 2075 7365  gle would be use
+000086b0: 640a 2d2d 7072 6f74 636f 6364 6174 696e  d.--protcocdatin
+000086c0: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
+000086d0: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
+000086e0: 696e 6974 6961 7465 2074 6865 2070 726f  initiate the pro
+000086f0: 7465 696e 2d63 6f6e 6361 7465 6e61 7469  tein-concatenati
+00008700: 6f6e 2d62 6173 6564 2064 6174 696e 6720  on-based dating 
+00008710: 616e 616c 7973 6973 2c20 6966 2074 6865  analysis, if the
+00008720: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00008730: 6865 2061 6e61 6c79 7369 7320 776f 756c  he analysis woul
+00008740: 6420 6265 2069 6e69 7469 6174 6564 0a2d  d be initiated.-
+00008750: 2d70 726f 7464 6174 696e 672c 2066 6c61  -protdating, fla
+00008760: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00008770: 7220 746f 206f 6e6c 7920 696e 6974 6961  r to only initia
+00008780: 7465 2074 6865 2070 726f 7465 696e 2d62  te the protein-b
+00008790: 6173 6564 2064 6174 696e 6720 616e 616c  ased dating anal
+000087a0: 7973 6973 2c20 6966 2074 6865 2066 6c61  ysis, if the fla
+000087b0: 6720 7761 7320 7365 742c 2074 6865 2061  g was set, the a
+000087c0: 6e61 6c79 7369 7320 776f 756c 6420 6265  nalysis would be
+000087d0: 2069 6e69 7469 6174 6564 0a60 6060 0a0a   initiated.```..
+000087e0: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
+000087f0: 206b 7364 6020 6361 6e20 7265 616c 697a   ksd` can realiz
+00008800: 6520 7468 6520 636f 6e73 7472 7563 7469  e the constructi
+00008810: 6f6e 206f 6620 2a4b 2a3c 7375 623e 533c  on of *K*<sub>S<
+00008820: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00008830: 6275 7469 6f6e 2061 6e64 2072 6174 6520  bution and rate 
+00008840: 636f 7272 6563 7469 6f6e 2e0a 6060 600a  correction..```.
+00008850: 7767 6420 6b73 6420 6661 6d69 6c69 6573  wgd ksd families
+00008860: 2073 6571 7565 6e63 6573 2028 6f70 7469   sequences (opti
+00008870: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
+00008880: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00008890: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000088a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000088b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000088c0: 2d2d 2d2d 0a2d 6f2c 202d 2d6f 7574 6469  ----.-o, --outdi
+000088d0: 722c 2074 6865 206f 7574 7075 7420 6469  r, the output di
+000088e0: 7265 6374 6f72 792c 2064 6566 6175 6c74  rectory, default
+000088f0: 2077 6764 5f6b 7364 0a2d 742c 202d 2d74   wgd_ksd.-t, --t
+00008900: 6d70 6469 722c 2074 6865 2074 656d 706f  mpdir, the tempo
+00008910: 7261 7279 2077 6f72 6b69 6e67 2064 6972  rary working dir
+00008920: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
+00008930: 4e6f 6e65 2c20 6966 204e 6f6e 6520 7761  None, if None wa
+00008940: 7320 6769 7665 6e2c 2074 6865 2074 6d70  s given, the tmp
+00008950: 6469 7220 7769 6c6c 2062 6520 6173 7369  dir will be assi
+00008960: 676e 6564 2072 616e 646f 6d20 6e61 6d65  gned random name
+00008970: 7320 696e 2063 7572 7265 6e74 2064 6972  s in current dir
+00008980: 6563 746f 7279 2061 6e64 2061 7574 6f6d  ectory and autom
+00008990: 6174 656c 7920 7265 6d6f 7665 6420 6174  ately removed at
+000089a0: 2074 6865 2063 6f6d 706c 6574 696f 6e20   the completion 
+000089b0: 6f66 2070 726f 6772 616d 2c20 656c 7365  of program, else
+000089c0: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
+000089d0: 6420 6265 206b 6570 740a 2d6e 2c20 2d2d  d be kept.-n, --
+000089e0: 6e74 6872 6561 6473 2c20 7468 6520 6e75  nthreads, the nu
+000089f0: 6d62 6572 206f 6620 7468 7265 6164 7320  mber of threads 
+00008a00: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
+00008a10: 340a 2d2d 746f 5f73 746f 702c 2066 6c61  4.--to_stop, fla
+00008a20: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00008a30: 7220 746f 2074 7261 6e73 6c61 7465 2074  r to translate t
+00008a40: 6872 6f75 6768 2053 544f 5020 636f 646f  hrough STOP codo
+00008a50: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
+00008a60: 7761 7320 7365 742c 2074 7261 6e73 6c61  was set, transla
+00008a70: 7469 6f6e 2077 696c 6c20 6265 2074 6572  tion will be ter
+00008a80: 6d69 6e61 7465 6420 6174 2074 6865 2066  minated at the f
+00008a90: 6972 7374 2069 6e2d 6672 616d 6520 7374  irst in-frame st
+00008aa0: 6f70 2063 6f64 6f6e 2c20 656c 7365 2061  op codon, else a
+00008ab0: 2066 756c 6c20 7472 616e 736c 6174 696f   full translatio
+00008ac0: 6e20 636f 6e74 696e 7569 6e67 206f 6e20  n continuing on 
+00008ad0: 7061 7374 2061 6e79 2073 746f 7020 636f  past any stop co
+00008ae0: 646f 6e73 2077 6f75 6c64 2062 6520 696e  dons would be in
+00008af0: 6974 6961 7465 640a 2d2d 6364 732c 2066  itiated.--cds, f
+00008b00: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00008b10: 6865 7220 746f 206f 6e6c 7920 7472 616e  her to only tran
+00008b20: 736c 6174 6520 7468 6520 636f 6d70 6c65  slate the comple
+00008b30: 7465 2043 4453 2074 6861 7420 7374 6172  te CDS that star
+00008b40: 7473 2077 6974 6820 6120 7661 6c69 6420  ts with a valid 
+00008b50: 7374 6172 7420 636f 646f 6e20 616e 6420  start codon and 
+00008b60: 6f6e 6c79 2063 6f6e 7461 696e 7320 6120  only contains a 
+00008b70: 7369 6e67 6c65 2069 6e2d 6672 616d 6520  single in-frame 
+00008b80: 7374 6f70 2063 6f64 6f6e 2061 7420 7468  stop codon at th
+00008b90: 6520 656e 6420 616e 6420 6d75 7374 2062  e end and must b
+00008ba0: 6520 6469 7669 6461 626c 6520 6279 2074  e dividable by t
+00008bb0: 6872 6565 2c20 6966 2074 6865 2066 6c61  hree, if the fla
+00008bc0: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
+00008bd0: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
+00008be0: 2077 6f75 6c64 2062 6520 7472 616e 736c   would be transl
+00008bf0: 6174 6564 0a2d 2d70 6169 7277 6973 652c  ated.--pairwise,
+00008c00: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00008c10: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00008c20: 6520 7061 6972 7769 7365 204b 7320 6573  e pairwise Ks es
+00008c30: 7469 6d61 7469 6f6e 2c20 6966 2074 6865  timation, if the
+00008c40: 2066 6c61 6720 7761 7320 7365 742c 2070   flag was set, p
+00008c50: 6169 7277 6973 6520 4b73 2076 616c 7565  airwise Ks value
+00008c60: 7320 776f 756c 6420 6265 2065 7374 696d  s would be estim
+00008c70: 6174 6564 0a2d 2d73 7472 6970 5f67 6170  ated.--strip_gap
+00008c80: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
+00008c90: 7768 6574 6865 7220 746f 2064 726f 7020  whether to drop 
+00008ca0: 616c 6c20 6761 7073 2069 6e20 6d75 6c74  all gaps in mult
+00008cb0: 6970 6c65 2073 6571 7565 6e63 6520 616c  iple sequence al
+00008cc0: 6967 6e6d 656e 742c 2069 6620 7468 6520  ignment, if the 
+00008cd0: 666c 6167 2077 6173 2073 6574 2c20 616c  flag was set, al
+00008ce0: 6c20 6761 7073 2077 6f75 6c64 2062 6520  l gaps would be 
+00008cf0: 6472 6f70 7065 640a 2d61 2c20 2d2d 616c  dropped.-a, --al
+00008d00: 6967 6e65 722c 2077 6869 6368 2061 6c69  igner, which ali
+00008d10: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
+00008d20: 6f20 7573 652c 2064 6566 6175 6c74 206d  o use, default m
+00008d30: 6166 6674 200a 2d74 7265 652c 202d 2d74  afft .-tree, --t
+00008d40: 7265 655f 6d65 7468 6f64 2c20 7768 6963  ree_method, whic
+00008d50: 6820 6765 6e65 2074 7265 6520 696e 6665  h gene tree infe
+00008d60: 7265 6e63 6520 7072 6f67 7261 6d20 746f  rence program to
+00008d70: 2069 6e76 6f6b 652c 2064 6566 6175 6c74   invoke, default
+00008d80: 2066 6173 7474 7265 650a 2d2d 7472 6565   fasttree.--tree
+00008d90: 5f6f 7074 696f 6e73 2c20 6f70 7469 6f6e  _options, option
+00008da0: 7320 696e 2074 7265 6520 696e 6665 7265  s in tree infere
+00008db0: 6e63 6520 6173 2061 2063 6f6d 6d61 2073  nce as a comma s
+00008dc0: 6570 6172 6174 6564 2073 7472 696e 672c  eparated string,
+00008dd0: 2064 6566 6175 6c74 204e 6f6e 650a 2d2d   default None.--
+00008de0: 6e6f 6465 5f61 7665 7261 6765 2c20 666c  node_average, fl
+00008df0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+00008e00: 6572 2074 6f20 696e 6974 6961 7465 206e  er to initiate n
+00008e10: 6f64 652d 6176 6572 6167 6520 7761 7920  ode-average way 
+00008e20: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
+00008e30: 2069 6e73 7465 6164 206f 6620 6e6f 6465   instead of node
+00008e40: 2d77 6569 6768 7465 642c 2069 6620 7468  -weighted, if th
+00008e50: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00008e60: 7468 6520 6e6f 6465 2d61 7665 7261 6769  the node-averagi
+00008e70: 6e67 2064 652d 7265 6475 6e64 616e 6379  ng de-redundancy
+00008e80: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00008e90: 7465 640a 2d73 722c 202d 2d73 7061 6972  ted.-sr, --spair
+00008ea0: 2c20 7468 6520 7370 6563 6965 7320 7061  , the species pa
+00008eb0: 6972 2074 6f20 6265 2070 6c6f 7474 6564  ir to be plotted
+00008ec0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+00008ed0: 7468 6973 206f 7074 696f 6e20 6361 6e20  this option can 
+00008ee0: 6265 2070 726f 7669 6465 6420 6d75 6c74  be provided mult
+00008ef0: 6970 6c65 2074 696d 6573 0a2d 7370 2c20  iple times.-sp, 
+00008f00: 2d2d 7370 6563 6965 7374 7265 652c 2074  --speciestree, t
+00008f10: 6865 2073 7065 6369 6573 2074 7265 6520  he species tree 
+00008f20: 746f 2070 6572 666f 726d 2072 6174 6520  to perform rate 
+00008f30: 636f 7272 6563 7469 6f6e 2c20 6465 6661  correction, defa
+00008f40: 756c 7420 4e6f 6e65 2c20 6966 204e 6f6e  ult None, if Non
+00008f50: 6520 7761 7320 6769 7665 6e2c 2074 6865  e was given, the
+00008f60: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
+00008f70: 2061 6e61 6c79 7369 7320 776f 756c 6420   analysis would 
+00008f80: 6265 2063 616c 6c65 6420 6f66 660a 2d72  be called off.-r
+00008f90: 772c 202d 2d72 6577 6569 6768 742c 2066  w, --reweight, f
+00008fa0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00008fb0: 6865 7220 746f 2072 6563 616c 6375 6c61  her to recalcula
+00008fc0: 7465 2074 6865 2077 6569 6768 7420 7065  te the weight pe
+00008fd0: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
+00008fe0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00008ff0: 7365 742c 2074 6865 2077 6569 6768 7420  set, the weight 
+00009000: 776f 756c 6420 6265 2072 6563 616c 6375  would be recalcu
+00009010: 6c61 7465 640a 2d6f 722c 202d 2d6f 6e6c  lated.-or, --onl
+00009020: 7972 6f6f 746f 7574 2c20 666c 6167 206f  yrootout, flag o
+00009030: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00009040: 6f20 6f6e 6c79 2063 6f6e 6475 6374 2072  o only conduct r
+00009050: 6174 6520 636f 7272 6563 7469 6f6e 2075  ate correction u
+00009060: 7369 6e67 2074 6865 206f 7574 6772 6f75  sing the outgrou
+00009070: 7020 6174 2072 6f6f 7420 6173 206f 7574  p at root as out
+00009080: 6772 6f75 702c 2069 6620 7468 6520 666c  group, if the fl
+00009090: 6167 2077 6173 2073 6574 2c20 6f6e 6c79  ag was set, only
+000090a0: 2074 6865 206f 7574 6772 6f75 7020 6174   the outgroup at
+000090b0: 2072 6f6f 7420 776f 756c 6420 6265 2075   root would be u
+000090c0: 7365 6420 6173 206f 7574 6772 6f75 700a  sed as outgroup.
+000090d0: 2d65 706b 2c20 2d2d 6578 7472 6170 6172  -epk, --extrapar
+000090e0: 616e 6f6d 656b 732c 2065 7874 7261 2070  anomeks, extra p
+000090f0: 6172 616e 6f6d 6520 4b73 2064 6174 6120  aranome Ks data 
+00009100: 746f 2070 6c6f 7420 696e 2074 6865 206d  to plot in the m
+00009110: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
+00009120: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
+00009130: 6e65 0a2d 6170 2c20 2d2d 616e 6368 6f72  ne.-ap, --anchor
+00009140: 706f 696e 7473 2c20 616e 6368 6f72 706f  points, anchorpo
+00009150: 696e 7473 2e74 7874 2066 696c 6520 746f  ints.txt file to
+00009160: 2070 6c6f 7420 616e 6368 6f72 204b 7320   plot anchor Ks 
+00009170: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
+00009180: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
+00009190: 6661 756c 7420 4e6f 6e65 0a2d 706b 2c20  fault None.-pk, 
+000091a0: 2d2d 706c 6f74 6b64 652c 2066 6c61 6720  --plotkde, flag 
+000091b0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000091c0: 746f 2070 6c6f 7420 6b64 6520 6375 7276  to plot kde curv
+000091d0: 6520 6f66 206f 7274 686f 6c6f 676f 7573  e of orthologous
+000091e0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+000091f0: 206f 7665 7220 6869 7374 6f67 7261 6d20   over histogram 
+00009200: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
+00009210: 6469 7374 7269 6275 7469 6f6e 2c20 6966  distribution, if
+00009220: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00009230: 742c 2074 6865 206b 6465 2063 7572 7665  t, the kde curve
+00009240: 2077 6f75 6c64 2062 6520 706c 6f74 7465   would be plotte
+00009250: 640a 2d70 6167 2c20 2d2d 706c 6f74 6170  d.-pag, --plotap
+00009260: 676d 6d2c 2066 6c61 6720 6f70 7469 6f6e  gmm, flag option
+00009270: 2c20 7768 6574 6865 7220 746f 2070 6572  , whether to per
+00009280: 666f 726d 2061 6e64 2070 6c6f 7420 6d69  form and plot mi
+00009290: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
+000092a0: 6620 616e 6368 6f72 204b 7320 696e 2074  f anchor Ks in t
+000092b0: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
+000092c0: 7269 6275 7469 6f6e 2c20 6966 2074 6865  ribution, if the
+000092d0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+000092e0: 6865 206d 6978 7475 7265 206d 6f64 656c  he mixture model
+000092f0: 696e 6720 6f66 2061 6e63 686f 7220 4b73  ing of anchor Ks
+00009300: 2077 6f75 6c64 2062 6520 706c 6f74 7465   would be plotte
+00009310: 640a 2d70 656d 2c20 2d2d 706c 6f74 656c  d.-pem, --plotel
+00009320: 6d6d 2c20 666c 6167 206f 7074 696f 6e2c  mm, flag option,
+00009330: 2077 6865 7468 6572 2074 6f20 7065 7266   whether to perf
+00009340: 6f72 6d20 616e 6420 706c 6f74 2065 6c6d  orm and plot elm
+00009350: 6d20 6d69 7874 7572 6520 6d6f 6465 6c69  m mixture modeli
+00009360: 6e67 206f 6620 7061 7261 6e6f 6d65 204b  ng of paranome K
+00009370: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
+00009380: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00009390: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+000093a0: 7365 742c 2074 6865 2065 6c6d 6d20 6d69  set, the elmm mi
+000093b0: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
+000093c0: 6620 7061 7261 6e6f 6d65 204b 7320 776f  f paranome Ks wo
+000093d0: 756c 6420 6265 2070 6c6f 7474 6564 0a2d  uld be plotted.-
+000093e0: 632c 202d 2d63 6f6d 706f 6e65 6e74 732c  c, --components,
+000093f0: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
+00009400: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
+00009410: 6f6e 656e 7473 2074 6f20 6669 7420 696e  onents to fit in
+00009420: 2061 6e63 686f 7220 4b73 206d 6978 7475   anchor Ks mixtu
+00009430: 7265 206d 6f64 656c 696e 672c 2064 6566  re modeling, def
+00009440: 6175 6c74 2028 312c 3429 0a2d 786c 2c20  ault (1,4).-xl, 
+00009450: 2d2d 786c 696d 2c20 7468 6520 7820 6178  --xlim, the x ax
+00009460: 6973 206c 696d 6974 206f 6620 4b73 2064  is limit of Ks d
+00009470: 6973 7472 6962 7574 696f 6e0a 2d79 6c2c  istribution.-yl,
+00009480: 202d 2d79 6c69 6d2c 2074 6865 2079 2061   --ylim, the y a
+00009490: 7869 7320 6c69 6d69 7420 6f66 204b 7320  xis limit of Ks 
+000094a0: 6469 7374 7269 6275 7469 6f6e 0a2d 6164  distribution.-ad
+000094b0: 6f2c 202d 2d61 646a 7573 746f 7274 686f  o, --adjustortho
+000094c0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+000094d0: 6865 7468 6572 2074 6f20 6164 6a75 7374  hether to adjust
+000094e0: 2074 6865 2068 6973 746f 6772 616d 2068   the histogram h
+000094f0: 6569 6768 7420 6f66 206f 7274 686f 6c6f  eight of ortholo
+00009500: 676f 7573 204b 7320 6173 2074 6f20 6d61  gous Ks as to ma
+00009510: 7463 6820 7468 6520 6865 6967 6874 206f  tch the height o
+00009520: 6620 7061 7261 6c6f 676f 7573 204b 732c  f paralogous Ks,
+00009530: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00009540: 2073 6574 2c20 7468 6520 6164 6a75 7374   set, the adjust
+00009550: 6d65 6e74 2077 6f75 6c64 2062 6520 636f  ment would be co
+00009560: 6e64 7563 7465 640a 2d61 6466 2c20 2d2d  nducted.-adf, --
+00009570: 6164 6a75 7374 6661 6374 6f72 2c20 7468  adjustfactor, th
+00009580: 6520 6164 6a75 7374 6d65 6e74 2066 6163  e adjustment fac
+00009590: 746f 7220 6f66 206f 7274 686f 6c6f 676f  tor of orthologo
+000095a0: 7573 204b 732c 2064 6566 6175 6c74 2030  us Ks, default 0
+000095b0: 2e35 0a2d 6f61 2c20 2d2d 6f6b 616c 7068  .5.-oa, --okalph
+000095c0: 612c 2074 6865 206f 7061 6369 7479 206f  a, the opacity o
+000095d0: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
+000095e0: 2064 6973 7472 6962 7574 696f 6e20 696e   distribution in
+000095f0: 206d 6978 6564 2070 6c6f 742c 2064 6566   mixed plot, def
+00009600: 6175 6c74 2030 2e35 0a2d 6661 2c20 2d2d  ault 0.5.-fa, --
+00009610: 666f 6375 7332 616c 6c2c 2073 6574 2066  focus2all, set f
+00009620: 6f63 616c 2073 7065 6369 6573 2061 6e64  ocal species and
+00009630: 206c 6574 2073 7065 6369 6573 2070 6169   let species pai
+00009640: 7220 746f 2062 6520 6265 7477 6565 6e20  r to be between 
+00009650: 666f 6361 6c20 616e 6420 616c 6c20 7468  focal and all th
+00009660: 6520 7265 6d61 696e 696e 6720 7370 6563  e remaining spec
+00009670: 6965 732c 2064 6566 6175 6c74 204e 6f6e  ies, default Non
+00009680: 650a 2d6b 732c 202d 2d6b 7374 7265 652c  e.-ks, --kstree,
+00009690: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+000096a0: 6574 6865 7220 746f 2069 6e66 6572 204b  ether to infer K
+000096b0: 7320 7472 6565 2c20 6966 2074 6865 2066  s tree, if the f
+000096c0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+000096d0: 204b 7320 7472 6565 2069 6e66 6572 656e   Ks tree inferen
+000096e0: 6365 2061 6e61 6c79 7369 7320 776f 756c  ce analysis woul
+000096f0: 6420 6265 2069 6e69 7469 6174 6564 0a2d  d be initiated.-
+00009700: 6f63 6b2c 202d 2d6f 6e6c 7963 6f6e 6361  ock, --onlyconca
+00009710: 746b 7374 7265 652c 2066 6c61 6720 6f70  tkstree, flag op
+00009720: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00009730: 206f 6e6c 7920 696e 6665 7220 4b73 2074   only infer Ks t
+00009740: 7265 6520 756e 6465 7220 636f 6e63 6174  ree under concat
+00009750: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
+00009760: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00009770: 7320 7365 742c 206f 6e6c 7920 7468 6520  s set, only the 
+00009780: 4b73 2074 7265 6520 756e 6465 7220 636f  Ks tree under co
+00009790: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
+000097a0: 6d65 6e74 2077 6f75 6c64 2062 6520 6361  ment would be ca
+000097b0: 6c63 756c 6174 6564 0a2d 6373 2c20 2d2d  lculated.-cs, --
+000097c0: 636c 6173 7369 632c 2066 6c61 6720 6f70  classic, flag op
+000097d0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+000097e0: 2064 7261 7720 6d69 7865 6420 4b73 2070   draw mixed Ks p
+000097f0: 6c6f 7420 696e 2061 2063 6c61 7373 6963  lot in a classic
+00009800: 206d 616e 6e65 7220 7768 6572 6520 7468   manner where th
+00009810: 6520 6675 6c6c 206f 7274 686f 6c6f 676f  e full orthologo
+00009820: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
+00009830: 6f6e 2069 7320 6472 6177 6564 2c20 6966  on is drawed, if
+00009840: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00009850: 742c 2074 6865 2063 6c61 7373 6963 206d  t, the classic m
+00009860: 6978 6564 204b 7320 706c 6f74 2077 6f75  ixed Ks plot wou
+00009870: 6c64 2062 6520 6472 6177 6e0a 2d74 612c  ld be drawn.-ta,
+00009880: 202d 2d74 6f70 6172 726f 772c 2066 6c61   --toparrow, fla
+00009890: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+000098a0: 7220 746f 2061 646a 7573 7420 7468 6520  r to adjust the 
+000098b0: 6172 726f 7720 746f 2062 6520 6174 2074  arrow to be at t
+000098c0: 6865 2074 6f70 206f 6620 7468 6520 706c  he top of the pl
+000098d0: 6f74 2c20 696e 7374 6561 6420 6f66 2062  ot, instead of b
+000098e0: 6569 6e67 2063 6f6f 7264 696e 6174 6564  eing coordinated
+000098f0: 2061 7320 7468 6520 4b44 4520 6f66 2074   as the KDE of t
+00009900: 6865 206f 7274 686f 6c6f 676f 7573 204b  he orthologous K
+00009910: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00009920: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00009930: 7365 742c 2074 6865 2061 7272 6f77 2077  set, the arrow w
+00009940: 6f75 6c64 2062 6520 7365 7420 6174 2074  ould be set at t
+00009950: 6865 2074 6f70 0a2d 6273 2c20 2d2d 626f  he top.-bs, --bo
+00009960: 6f74 7374 7261 702c 2074 6865 206e 756d  otstrap, the num
+00009970: 6265 7220 6f66 2062 6f6f 7473 7472 6170  ber of bootstrap
+00009980: 2072 6570 6c69 6361 7465 7320 6f66 206f   replicates of o
+00009990: 7274 686f 6c6f 6720 4b73 2064 6973 7472  rtholog Ks distr
+000099a0: 6962 7574 696f 6e20 696e 206d 6978 6564  ibution in mixed
+000099b0: 2070 6c6f 740a 6060 600a 0a54 6865 2070   plot.```..The p
+000099c0: 726f 6772 616d 2060 7767 6420 6d69 7860  rogram `wgd mix`
+000099d0: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+000099e0: 206d 6978 7475 7265 206d 6f64 656c 2063   mixture model c
+000099f0: 6c75 7374 6572 696e 6720 616e 616c 7973  lustering analys
+00009a00: 6973 206f 6620 2a4b 2a3c 7375 623e 533c  is of *K*<sub>S<
+00009a10: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00009a20: 6275 7469 6f6e 2e0a 6060 600a 7767 6420  bution..```.wgd 
+00009a30: 6d69 7820 6b73 5f64 6174 6166 696c 6520  mix ks_datafile 
+00009a40: 286f 7074 696f 6e29 0a2d 2d2d 2d2d 2d2d  (option).-------
+00009a50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009a60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009a70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009a80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009a90: 2d2d 2d2d 2d2d 2d2d 2d0a 2d66 2c20 2d2d  ---------.-f, --
+00009aa0: 6669 6c74 6572 732c 2074 6865 2063 7574  filters, the cut
+00009ab0: 6f66 6620 616c 6967 6e6d 656e 7420 6c65  off alignment le
+00009ac0: 6e67 7468 2c20 6465 6661 756c 7420 3330  ngth, default 30
+00009ad0: 300a 2d72 2c20 2d2d 6b73 5f72 616e 6765  0.-r, --ks_range
+00009ae0: 2c20 7468 6520 4b73 2072 616e 6765 2074  , the Ks range t
+00009af0: 6f20 6265 2063 6f6e 7369 6465 7265 642c  o be considered,
+00009b00: 2064 6566 6175 6c74 2028 302c 2035 290a   default (0, 5).
+00009b10: 2d62 2c20 2d2d 6269 6e73 2c20 7468 6520  -b, --bins, the 
+00009b20: 6e75 6d62 6572 206f 6620 6269 6e73 2069  number of bins i
+00009b30: 6e20 4b73 2064 6973 7472 6962 7574 696f  n Ks distributio
+00009b40: 6e2c 2064 6566 6175 6c74 2035 300a 2d6f  n, default 50.-o
+00009b50: 2c20 2d2d 6f75 7464 6972 2c20 7468 6520  , --outdir, the 
+00009b60: 6f75 7470 7574 2064 6972 6563 746f 7279  output directory
+00009b70: 2c20 6465 6661 756c 7420 7767 645f 6d69  , default wgd_mi
+00009b80: 780a 2d2d 6d65 7468 6f64 2c20 7768 6963  x.--method, whic
+00009b90: 6820 6d69 7874 7572 6520 6d6f 6465 6c20  h mixture model 
+00009ba0: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
+00009bb0: 676d 6d0a 2d6e 2c20 2d2d 636f 6d70 6f6e  gmm.-n, --compon
+00009bc0: 656e 7473 2c20 7468 6520 7261 6e67 6520  ents, the range 
+00009bd0: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
+00009be0: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
+00009bf0: 6974 2c20 6465 6661 756c 7420 2831 2c20  it, default (1, 
+00009c00: 3429 0a2d 672c 202d 2d67 616d 6d61 2c20  4).-g, --gamma, 
+00009c10: 7468 6520 6761 6d6d 6120 7061 7261 6d65  the gamma parame
+00009c20: 7465 7220 666f 7220 6267 6d6d 206d 6f64  ter for bgmm mod
+00009c30: 656c 732c 2064 6566 6175 6c74 2030 2e30  els, default 0.0
+00009c40: 3031 0a2d 6e69 2c20 2d2d 6e5f 696e 6974  01.-ni, --n_init
+00009c50: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+00009c60: 6b2d 6d65 616e 7320 696e 6974 6961 6c69  k-means initiali
+00009c70: 7a61 7469 6f6e 732c 2064 6566 6175 6c74  zations, default
+00009c80: 2032 3030 0a2d 6d69 2c20 2d2d 6d61 785f   200.-mi, --max_
+00009c90: 6974 6572 2c20 7468 6520 6d61 7869 6d75  iter, the maximu
+00009ca0: 6d20 6e75 6d62 6572 206f 6620 6974 6572  m number of iter
+00009cb0: 6174 696f 6e73 2c20 6465 6661 756c 7420  ations, default 
+00009cc0: 3230 300a 6060 600a 0a54 6865 2070 726f  200.```..The pro
+00009cd0: 6772 616d 2060 7767 6420 7065 616b 6020  gram `wgd peak` 
+00009ce0: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
+00009cf0: 7365 6172 6368 206f 6620 6372 6564 6961  search of credia
+00009d00: 626c 6520 2a4b 2a3c 7375 623e 533c 2f73  ble *K*<sub>S</s
+00009d10: 7562 3e20 7261 6e67 6520 7573 6564 2069  ub> range used i
+00009d20: 6e20 5747 4420 6461 7469 6e67 2e0a 6060  n WGD dating..``
+00009d30: 600a 7767 6420 7065 616b 206b 735f 6461  `.wgd peak ks_da
+00009d40: 7461 6669 6c65 2028 6f70 7469 6f6e 290a  tafile (option).
+00009d50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009d60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009d70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009d80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009d90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009da0: 0a2d 6170 2c20 2d2d 616e 6368 6f72 706f  .-ap, --anchorpo
+00009db0: 696e 7473 2c20 7468 6520 616e 6368 6f72  ints, the anchor
+00009dc0: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
+00009dd0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00009de0: 736d 2c20 2d2d 7365 676d 656e 7473 2c20  sm, --segments, 
+00009df0: 7468 6520 7365 676d 656e 7473 2064 6174  the segments dat
+00009e00: 6166 696c 652c 2064 6566 6175 6c74 204e  afile, default N
+00009e10: 6f6e 650a 2d6c 652c 202d 2d6c 6973 7465  one.-le, --liste
+00009e20: 6c65 6d65 6e74 732c 2074 6865 206c 6973  lements, the lis
+00009e30: 7473 6567 6d65 6e74 7320 6461 7461 6669  tsegments datafi
+00009e40: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+00009e50: 200a 2d6d 702c 202d 2d6d 756c 7469 706c   .-mp, --multipl
+00009e60: 6963 6f6e 7061 6972 732c 2074 6865 206d  iconpairs, the m
+00009e70: 756c 7469 706c 6963 6f6e 7061 6972 7320  ultipliconpairs 
+00009e80: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+00009e90: 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574  t None.-o, --out
+00009ea0: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
+00009eb0: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
+00009ec0: 6c74 2077 6764 5f70 6561 6b0a 2d61 662c  lt wgd_peak.-af,
+00009ed0: 202d 2d61 6c69 676e 6669 6c74 6572 2c20   --alignfilter, 
+00009ee0: 6375 746f 6666 2066 6f72 2061 6c69 676e  cutoff for align
+00009ef0: 6d65 6e74 2069 6465 6e74 6974 792c 206c  ment identity, l
+00009f00: 656e 6774 6820 616e 6420 636f 7665 7261  ength and covera
+00009f10: 6765 2c20 6465 6661 756c 7420 302e 302c  ge, default 0.0,
+00009f20: 2030 2c20 302e 300a 2d72 2c20 2d2d 6b73   0, 0.0.-r, --ks
+00009f30: 7261 6e67 652c 2072 616e 6765 206f 6620  range, range of 
+00009f40: 4b73 2074 6f20 6265 2061 6e61 6c79 7a65  Ks to be analyze
+00009f50: 642c 2064 6566 6175 6c74 2028 302c 2035  d, default (0, 5
+00009f60: 290a 2d62 772c 202d 2d62 696e 5f77 6964  ).-bw, --bin_wid
+00009f70: 7468 2c20 6261 6e64 7769 6474 6820 6f66  th, bandwidth of
+00009f80: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00009f90: 2c20 6465 6661 756c 7420 302e 310a 2d69  , default 0.1.-i
+00009fa0: 632c 202d 2d77 6569 6768 7473 5f6f 7574  c, --weights_out
+00009fb0: 6c69 6572 735f 696e 636c 7564 6564 2c20  liers_included, 
+00009fc0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00009fd0: 7468 6572 2074 6f20 696e 636c 7564 6520  ther to include 
+00009fe0: 4b73 206f 7574 6c69 6572 732c 2069 6620  Ks outliers, if 
+00009ff0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+0000a000: 2c20 4b73 206f 7574 6c69 6572 7320 776f  , Ks outliers wo
+0000a010: 756c 6420 6265 2069 6e63 6c75 6465 6420  uld be included 
+0000a020: 696e 2074 6865 2061 6e61 6c79 7369 730a  in the analysis.
+0000a030: 2d6d 2c20 2d2d 6d65 7468 6f64 2c20 7768  -m, --method, wh
+0000a040: 6963 6820 6d69 7874 7572 6520 6d6f 6465  ich mixture mode
+0000a050: 6c20 746f 2075 7365 2c20 6465 6661 756c  l to use, defaul
+0000a060: 7420 676d 6d0a 2d2d 7365 6564 2c20 7261  t gmm.--seed, ra
+0000a070: 6e64 6f6d 2073 6565 6420 6769 7665 6e20  ndom seed given 
+0000a080: 746f 2069 6e69 7469 616c 697a 6174 696f  to initializatio
+0000a090: 6e2c 2064 6566 6175 6c74 2032 3335 3238  n, default 23528
+0000a0a0: 3930 0a2d 6569 2c20 2d2d 656d 5f69 7465  90.-ei, --em_ite
+0000a0b0: 722c 2074 6865 206e 756d 6265 7220 6f66  r, the number of
+0000a0c0: 2045 4d20 6974 6572 6174 696f 6e73 2074   EM iterations t
+0000a0d0: 6f20 7065 7266 6f72 6d2c 2064 6566 6175  o perform, defau
+0000a0e0: 6c74 2032 3030 0a2d 6e69 2c20 2d2d 6e5f  lt 200.-ni, --n_
+0000a0f0: 696e 6974 2c20 7468 6520 6e75 6d62 6572  init, the number
+0000a100: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
+0000a110: 6961 6c69 7a61 7469 6f6e 732c 2064 6566  ializations, def
+0000a120: 6175 6c74 2032 3030 0a2d 6e2c 202d 2d63  ault 200.-n, --c
+0000a130: 6f6d 706f 6e65 6e74 732c 2074 6865 2072  omponents, the r
+0000a140: 616e 6765 206f 6620 7468 6520 6e75 6d62  ange of the numb
+0000a150: 6572 206f 6620 636f 6d70 6f6e 656e 7473  er of components
+0000a160: 2074 6f20 6669 742c 2064 6566 6175 6c74   to fit, default
+0000a170: 2028 312c 2034 290a 2d67 2c20 2d2d 6761   (1, 4).-g, --ga
+0000a180: 6d6d 612c 2074 6865 2067 616d 6d61 2070  mma, the gamma p
+0000a190: 6172 616d 6574 6572 2066 6f72 2062 676d  arameter for bgm
+0000a1a0: 6d20 6d6f 6465 6c73 2c20 6465 6661 756c  m models, defaul
+0000a1b0: 7420 3165 2d33 0a2d 2d62 6f6f 7473 2c20  t 1e-3.--boots, 
+0000a1c0: 7468 6520 6e75 6d62 6572 206f 6620 626f  the number of bo
+0000a1d0: 6f74 7374 7261 7020 7265 706c 6963 6174  otstrap replicat
+0000a1e0: 6573 206f 6620 6b64 652c 2064 6566 6175  es of kde, defau
+0000a1f0: 6c74 2032 3030 0a2d 2d77 6569 6768 7465  lt 200.--weighte
+0000a200: 642c 2066 6c61 6720 6f70 7469 6f6e 2c20  d, flag option, 
+0000a210: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
+0000a220: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
+0000a230: 686f 6420 6f66 2064 652d 7265 6475 6e64  hod of de-redund
+0000a240: 616e 6379 2c20 6966 2074 6865 2066 6c61  ancy, if the fla
+0000a250: 6720 7761 7320 7365 742c 2074 6865 206e  g was set, the n
+0000a260: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
+0000a270: 686f 6420 776f 756c 6420 6265 2075 7365  hod would be use
+0000a280: 640a 2d70 2c20 2d2d 706c 6f74 2c20 7468  d.-p, --plot, th
+0000a290: 6520 706c 6f74 7469 6e67 206d 6574 686f  e plotting metho
+0000a2a0: 6420 746f 2062 6520 7573 6564 2c20 6465  d to be used, de
+0000a2b0: 6661 756c 7420 6964 656e 7469 6361 6c0a  fault identical.
+0000a2c0: 2d62 6d2c 202d 2d62 775f 6d65 7468 6f64  -bm, --bw_method
+0000a2d0: 2c20 7468 6520 6261 6e64 7769 6474 6820  , the bandwidth 
+0000a2e0: 6d65 7468 6f64 2074 6f20 6265 2075 7365  method to be use
+0000a2f0: 6420 696e 2061 6e61 6c79 7a69 6e67 2074  d in analyzing t
+0000a300: 6865 2070 6561 6b20 6f66 2057 4744 2064  he peak of WGD d
+0000a310: 6174 6573 2c20 6465 6661 756c 7420 7369  ates, default si
+0000a320: 6c76 6572 6d61 6e0a 2d2d 6e5f 6d65 646f  lverman.--n_medo
+0000a330: 6964 732c 2074 6865 206e 756d 6265 7220  ids, the number 
+0000a340: 6f66 206d 6564 6f69 6473 2074 6f20 6669  of medoids to fi
+0000a350: 742c 2064 6566 6175 6c74 2032 0a2d 6b6d  t, default 2.-km
+0000a360: 2c20 2d2d 6b64 656d 6574 686f 642c 2074  , --kdemethod, t
+0000a370: 6865 206b 6465 206d 6574 686f 6420 746f  he kde method to
+0000a380: 2062 6520 7573 6564 2069 6e20 616e 616c   be used in anal
+0000a390: 797a 696e 6720 7468 6520 7065 616b 206f  yzing the peak o
+0000a3a0: 6620 5747 4420 6461 7465 732c 206b 6d65  f WGD dates, kme
+0000a3b0: 646f 6964 7320 616e 616c 7973 6973 206f  doids analysis o
+0000a3c0: 7220 7468 6520 6261 7369 6320 4b73 2070  r the basic Ks p
+0000a3d0: 6c6f 7474 696e 672c 2064 6566 6175 6c74  lotting, default
+0000a3e0: 2073 6369 7079 0a2d 2d6e 5f63 6c75 7374   scipy.--n_clust
+0000a3f0: 6572 732c 2074 6865 206e 756d 6265 7220  ers, the number 
+0000a400: 6f66 2063 6c75 7374 6572 7320 746f 2070  of clusters to p
+0000a410: 6c6f 7420 456c 626f 7720 6c6f 7373 2066  lot Elbow loss f
+0000a420: 756e 6374 696f 6e2c 2064 6566 6175 6c74  unction, default
+0000a430: 2035 0a2d 6764 2c20 2d2d 6775 6964 652c   5.-gd, --guide,
+0000a440: 2074 6865 2072 6567 696d 6520 7265 7369   the regime resi
+0000a450: 6469 6e67 2061 6e63 686f 7273 2c20 6465  ding anchors, de
+0000a460: 6661 756c 7420 5365 676d 656e 740a 2d70  fault Segment.-p
+0000a470: 7263 742c 202d 2d70 726f 6d69 6e65 6e63  rct, --prominenc
+0000a480: 655f 6375 746f 6666 2c20 7468 6520 7072  e_cutoff, the pr
+0000a490: 6f6d 696e 656e 6365 2063 7574 6f66 6620  ominence cutoff 
+0000a4a0: 6f66 2061 6363 6570 7461 626c 6520 7065  of acceptable pe
+0000a4b0: 616b 7320 696e 2070 6561 6b20 6669 6e64  aks in peak find
+0000a4c0: 696e 6720 7374 6570 732c 2064 6566 6175  ing steps, defau
+0000a4d0: 6c74 2030 2e31 0a2d 7268 2c20 2d2d 7265  lt 0.1.-rh, --re
+0000a4e0: 6c5f 6865 6967 6874 2c20 7468 6520 7265  l_height, the re
+0000a4f0: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
+0000a500: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
+0000a510: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
+0000a520: 642c 2064 6566 6175 6c74 2030 2e34 0a2d  d, default 0.4.-
+0000a530: 6b64 2c20 2d2d 6b73 746f 6461 7465 2c20  kd, --kstodate, 
+0000a540: 7468 6520 7261 6e67 6520 6f66 204b 7320  the range of Ks 
+0000a550: 746f 2062 6520 6461 7465 6420 696e 2068  to be dated in h
+0000a560: 6575 7269 7374 6963 2073 6561 7263 682c  euristic search,
+0000a570: 2064 6566 6175 6c74 2028 302e 352c 2031   default (0.5, 1
+0000a580: 2e35 290a 2d78 6c2c 202d 2d78 6c69 6d2c  .5).-xl, --xlim,
+0000a590: 2074 6865 2078 2061 7869 7320 6c69 6d69   the x axis limi
+0000a5a0: 7420 6f66 2047 4d4d 204b 7320 6469 7374  t of GMM Ks dist
+0000a5b0: 7269 6275 7469 6f6e 0a2d 796c 2c20 2d2d  ribution.-yl, --
+0000a5c0: 796c 696d 2c20 7468 6520 7920 6178 6973  ylim, the y axis
+0000a5d0: 206c 696d 6974 206f 6620 474d 4d20 4b73   limit of GMM Ks
+0000a5e0: 2064 6973 7472 6962 7574 696f 6e0a 2d2d   distribution.--
+0000a5f0: 6d61 6e75 616c 7365 742c 2066 6c61 6720  manualset, flag 
+0000a600: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+0000a610: 746f 206f 7574 7075 7420 616e 6368 6f72  to output anchor
+0000a620: 2070 6169 7273 2077 6974 6820 6d61 6e75   pairs with manu
+0000a630: 616c 6c79 2073 6574 204b 7320 7261 6e67  ally set Ks rang
+0000a640: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
+0000a650: 6173 2073 6574 2c20 6d61 6e75 616c 6c79  as set, manually
+0000a660: 2073 6574 204b 7320 7261 6e67 6520 776f   set Ks range wo
+0000a670: 756c 6420 6265 206f 7574 7075 7474 6564  uld be outputted
+0000a680: 0a2d 2d63 692c 2074 6865 2063 6f6e 6669  .--ci, the confi
+0000a690: 6465 6e63 6520 6c65 7665 6c20 6f66 206c  dence level of l
+0000a6a0: 6f67 2d6e 6f72 6d61 6c20 6469 7374 7269  og-normal distri
+0000a6b0: 6275 7469 6f6e 2074 6f20 6461 7465 2c20  bution to date, 
+0000a6c0: 6465 6661 756c 7420 3935 0a2d 2d68 6472  default 95.--hdr
+0000a6d0: 2c20 7468 6520 6869 6768 6573 7420 6465  , the highest de
+0000a6e0: 6e73 6974 7920 7265 6769 6f6e 2028 4844  nsity region (HD
+0000a6f0: 5229 2061 7070 6c69 6564 2069 6e20 7468  R) applied in th
+0000a700: 6520 7365 676d 656e 742d 6775 6964 6564  e segment-guided
+0000a710: 2061 6e63 686f 7220 7061 6972 204b 7320   anchor pair Ks 
+0000a720: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
+0000a730: 6661 756c 7420 3935 0a2d 2d68 6575 7269  fault 95.--heuri
+0000a740: 7374 6963 2c20 666c 6167 206f 7074 696f  stic, flag optio
+0000a750: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+0000a760: 6974 6961 7465 2068 6575 7269 7374 6963  itiate heuristic
+0000a770: 206d 6574 686f 6420 6f66 2064 6566 696e   method of defin
+0000a780: 696e 6720 4349 2066 6f72 2064 6174 696e  ing CI for datin
+0000a790: 672c 2069 6620 7468 6520 666c 6167 2077  g, if the flag w
+0000a7a0: 6173 2073 6574 2c20 7468 6520 6865 7572  as set, the heur
+0000a7b0: 6973 7469 6320 6d65 7468 6f64 2077 6f75  istic method wou
+0000a7c0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+0000a7d0: 2d6b 632c 202d 2d6b 7363 7574 6f66 662c  -kc, --kscutoff,
+0000a7e0: 2074 6865 204b 7320 7361 7475 7261 7469   the Ks saturati
+0000a7f0: 6f6e 2063 7574 6f66 6620 696e 2064 6174  on cutoff in dat
+0000a800: 696e 672c 2064 6566 6175 6c74 2035 0a2d  ing, default 5.-
+0000a810: 2d6b 6565 7074 6d70 6669 672c 2066 6c61  -keeptmpfig, fla
+0000a820: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+0000a830: 7220 746f 206b 6565 7020 7465 6d70 6f72  r to keep tempor
+0000a840: 6172 7920 6669 6775 7265 7320 696e 2070  ary figures in p
+0000a850: 6561 6b20 6669 6e64 696e 6720 7072 6f63  eak finding proc
+0000a860: 6573 732c 2069 6620 7468 6520 666c 6167  ess, if the flag
+0000a870: 2077 6173 2073 6574 2c20 7468 6f73 6520   was set, those 
+0000a880: 6669 6775 7265 7320 776f 756c 6420 6265  figures would be
+0000a890: 206b 6570 740a 6060 600a 0a54 6865 2070   kept.```..The p
+0000a8a0: 726f 6772 616d 2060 7767 6420 7379 6e60  rogram `wgd syn`
+0000a8b0: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+0000a8c0: 2069 6e74 7261 2d20 616e 6420 696e 7465   intra- and inte
+0000a8d0: 722d 7370 6563 6966 6963 2073 796e 7465  r-specific synte
+0000a8e0: 6e79 2069 6e66 6572 656e 6365 2e0a 6060  ny inference..``
+0000a8f0: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
+0000a900: 6573 2067 6666 7320 286f 7074 696f 6e29  es gffs (option)
+0000a910: 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  .---------------
+0000a920: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a930: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a940: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a950: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a960: 2d0a 2d6b 732c 202d 2d6b 735f 6469 7374  -.-ks, --ks_dist
+0000a970: 7269 6275 7469 6f6e 2c20 6b73 2064 6973  ribution, ks dis
+0000a980: 7472 6962 7574 696f 6e20 6461 7461 6669  tribution datafi
+0000a990: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+0000a9a0: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
+0000a9b0: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
+0000a9c0: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
+0000a9d0: 5f73 796e 0a2d 662c 202d 2d66 6561 7475  _syn.-f, --featu
+0000a9e0: 7265 2c20 7468 6520 6665 6174 7572 6520  re, the feature 
+0000a9f0: 666f 7220 7061 7273 696e 6720 6765 6e65  for parsing gene
+0000aa00: 2049 4473 2066 726f 6d20 4746 4620 6669   IDs from GFF fi
+0000aa10: 6c65 732c 2064 6566 6175 6c74 2067 656e  les, default gen
+0000aa20: 650a 2d61 2c20 2d2d 6174 7472 6962 7574  e.-a, --attribut
+0000aa30: 652c 2074 6865 2061 7474 7269 6275 7465  e, the attribute
+0000aa40: 2066 6f72 2070 6172 7369 6e67 2074 6865   for parsing the
+0000aa50: 2067 656e 6520 4944 7320 6672 6f6d 2074   gene IDs from t
+0000aa60: 6865 2047 4646 2066 696c 6573 2c20 6465  he GFF files, de
+0000aa70: 6661 756c 7420 4944 0a2d 6174 672c 202d  fault ID.-atg, -
+0000aa80: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+0000aa90: 666f 2c20 7468 6520 6665 6174 7572 6520  fo, the feature 
+0000aaa0: 616e 6420 6174 7472 6962 7574 6520 696e  and attribute in
+0000aab0: 666f 726d 6174 696f 6e20 6f66 2061 6464  formation of add
+0000aac0: 6974 696f 6e61 6c20 6766 6633 2066 696c  itional gff3 fil
+0000aad0: 6573 2069 6620 6469 6666 6572 656e 7420  es if different 
+0000aae0: 696e 2074 6865 2066 6f72 6d61 7420 6f66  in the format of
+0000aaf0: 2028 6665 6174 7572 653b 6174 7472 6962   (feature;attrib
+0000ab00: 7574 6529 272c 2064 6566 6175 6c74 204e  ute)', default N
+0000ab10: 6f6e 650a 2d6d 6c2c 202d 2d6d 696e 6c65  one.-ml, --minle
+0000ab20: 6e2c 2074 6865 206d 696e 696d 756d 206c  n, the minimum l
+0000ab30: 656e 6774 6820 6f66 2061 2073 6361 6666  ength of a scaff
+0000ab40: 6f6c 6420 746f 2062 6520 696e 636c 7564  old to be includ
+0000ab50: 6564 2069 6e20 646f 7470 6c6f 742c 2064  ed in dotplot, d
+0000ab60: 6566 6175 6c74 202d 312c 2069 6620 2d31  efault -1, if -1
+0000ab70: 2077 6173 2073 6574 2c20 7468 6520 3130   was set, the 10
+0000ab80: 2520 6f66 2074 6865 206c 6f6e 6765 7374  % of the longest
+0000ab90: 2073 6361 6666 6f6c 6420 776f 756c 6420   scaffold would 
+0000aba0: 6265 2073 6574 0a2d 6d73 2c20 2d2d 6d61  be set.-ms, --ma
+0000abb0: 7873 697a 652c 2074 6865 206d 6178 696d  xsize, the maxim
+0000abc0: 756d 2066 616d 696c 7920 7369 7a65 2074  um family size t
+0000abd0: 6f20 6265 2069 6e63 6c75 6465 642c 2064  o be included, d
+0000abe0: 6566 6175 6c74 2032 3030 0a2d 722c 202d  efault 200.-r, -
+0000abf0: 2d6b 735f 7261 6e67 652c 2074 6865 204b  -ks_range, the K
+0000ac00: 7320 7261 6e67 6520 696e 2063 6f6c 6f72  s range in color
+0000ac10: 6564 2064 6f74 706c 6f74 2c20 6465 6661  ed dotplot, defa
+0000ac20: 756c 7420 2830 2c20 3529 0a2d 2d69 6164  ult (0, 5).--iad
+0000ac30: 686f 7265 5f6f 7074 696f 6e73 2c20 7468  hore_options, th
+0000ac40: 6520 7061 7261 6d65 7465 7220 7365 7474  e parameter sett
+0000ac50: 696e 6720 696e 2069 6164 686f 7265 2c20  ing in iadhore, 
+0000ac60: 6465 6661 756c 7420 6173 2061 2073 7472  default as a str
+0000ac70: 696e 6720 6f66 206c 656e 6774 6820 7a65  ing of length ze
+0000ac80: 726f 0a2d 6d67 2c20 2d2d 6d69 6e73 6567  ro.-mg, --minseg
+0000ac90: 6c65 6e2c 2074 6865 206d 696e 696d 756d  len, the minimum
+0000aca0: 206c 656e 6774 6820 6f66 2073 6567 6d65   length of segme
+0000acb0: 6e74 7320 746f 2069 6e63 6c75 6465 2069  nts to include i
+0000acc0: 6e20 7261 7469 6f20 6966 203c 3d20 312c  n ratio if <= 1,
+0000acd0: 2064 6566 6175 6c74 2031 3030 3030 0a2d   default 10000.-
+0000ace0: 6b72 2c20 2d2d 6b65 6570 7265 6475 6e2c  kr, --keepredun,
+0000acf0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+0000ad00: 6574 6865 7220 746f 206b 6565 7020 7265  ether to keep re
+0000ad10: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
+0000ad20: 636f 6e73 2c20 6966 2074 6865 2066 6c61  cons, if the fla
+0000ad30: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+0000ad40: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+0000ad50: 6963 6f6e 7320 776f 756c 6420 6265 206b  icons would be k
+0000ad60: 6570 740a 2d6d 676e 2c20 2d2d 6d69 6e67  ept.-mgn, --ming
+0000ad70: 656e 656e 756d 2c20 7468 6520 6d69 6e69  enenum, the mini
+0000ad80: 6d75 6d20 6e75 6d62 6572 206f 6620 6765  mum number of ge
+0000ad90: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+0000ada0: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+0000adb0: 6564 2c20 6465 6661 756c 7420 3330 0a2d  ed, default 30.-
+0000adc0: 6473 2c20 2d2d 646f 7473 697a 652c 2074  ds, --dotsize, t
+0000add0: 6865 2064 6f74 2073 697a 6520 696e 2064  he dot size in d
+0000ade0: 6f74 2070 6c6f 742c 2064 6566 6175 6c74  ot plot, default
+0000adf0: 2030 2e33 0a2d 6161 2c20 2d2d 6170 616c   0.3.-aa, --apal
+0000ae00: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+0000ae10: 206f 6620 616e 6368 6f72 2064 6f74 732c   of anchor dots,
+0000ae20: 2064 6566 6175 6c74 2031 0a2d 6861 2c20   default 1.-ha, 
+0000ae30: 2d2d 686f 616c 7068 612c 2074 6865 206f  --hoalpha, the o
+0000ae40: 7061 6369 7479 206f 6620 686f 6d6f 6c6f  pacity of homolo
+0000ae50: 6720 646f 7473 2c20 6465 6661 756c 7420  g dots, default 
+0000ae60: 300a 2d73 7274 2c20 2d2d 7368 6f77 7265  0.-srt, --showre
+0000ae70: 616c 7469 636b 2c20 666c 6167 206f 7074  altick, flag opt
+0000ae80: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+0000ae90: 7368 6f77 2074 6865 2072 6561 6c20 7469  show the real ti
+0000aea0: 636b 2069 6e20 6765 6e65 7320 6f72 2062  ck in genes or b
+0000aeb0: 6173 6573 2c20 6966 2074 6865 2066 6c61  ases, if the fla
+0000aec0: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+0000aed0: 6561 6c20 7469 636b 2077 6f75 6c64 2062  eal tick would b
+0000aee0: 6520 7368 6f77 6564 0a2d 746c 732c 202d  e showed.-tls, -
+0000aef0: 2d74 6963 6b6c 6162 656c 7369 7a65 2c20  -ticklabelsize, 
+0000af00: 7468 6520 6c61 6265 6c20 7369 7a65 206f  the label size o
+0000af10: 6620 7469 636b 2c20 6465 6661 756c 7420  f tick, default 
+0000af20: 350a 2d67 722c 202d 2d67 6973 7472 622c  5.-gr, --gistrb,
+0000af30: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+0000af40: 6574 6865 7220 746f 2075 7365 2067 6973  ether to use gis
+0000af50: 745f 7261 696e 626f 7720 6173 2063 6f6c  t_rainbow as col
+0000af60: 6f72 206d 6170 206f 6620 646f 7470 6c6f  or map of dotplo
+0000af70: 740a 2d6e 2c20 2d2d 6e74 6872 6561 6473  t.-n, --nthreads
+0000af80: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+0000af90: 7468 7265 6164 7320 746f 2075 7365 2069  threads to use i
+0000afa0: 6e20 7379 6e74 656e 7920 696e 6665 7265  n synteny infere
+0000afb0: 6e63 652c 2064 6566 6175 6c74 2034 0a60  nce, default 4.`
+0000afc0: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+0000afd0: 6077 6764 2076 697a 6020 6361 6e20 7265  `wgd viz` can re
+0000afe0: 616c 697a 6520 7468 6520 7669 7375 616c  alize the visual
+0000aff0: 697a 6174 696f 6e20 6f66 202a 4b2a 3c73  ization of *K*<s
+0000b000: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000b010: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+0000b020: 7379 6e74 656e 792e 0a60 6060 0a77 6764  synteny..```.wgd
+0000b030: 2076 697a 2028 6f70 7469 6f6e 290a 2d2d   viz (option).--
+0000b040: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000b050: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000b060: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000b070: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000b080: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
+0000b090: 642c 202d 2d64 6174 6166 696c 652c 2074  d, --datafile, t
+0000b0a0: 6865 204b 7320 6461 7461 6669 6c65 2c20  he Ks datafile, 
+0000b0b0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
+0000b0c0: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+0000b0d0: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+0000b0e0: 2064 6566 6175 6c74 2077 6764 5f76 697a   default wgd_viz
+0000b0f0: 0a2d 7372 2c20 2d2d 7370 6169 722c 2074  .-sr, --spair, t
+0000b100: 6865 2073 7065 6369 6573 2070 6169 7220  he species pair 
+0000b110: 746f 2062 6520 706c 6f74 7465 642c 2064  to be plotted, d
+0000b120: 6566 6175 6c74 204e 6f6e 652c 2074 6869  efault None, thi
+0000b130: 7320 6f70 7469 6f6e 2063 616e 2062 6520  s option can be 
+0000b140: 7072 6f76 6964 6564 206d 756c 7469 706c  provided multipl
+0000b150: 6520 7469 6d65 730a 2d66 612c 202d 2d66  e times.-fa, --f
+0000b160: 6f63 7573 3261 6c6c 2c20 7365 7420 666f  ocus2all, set fo
+0000b170: 6361 6c20 7370 6563 6965 7320 616e 6420  cal species and 
+0000b180: 6c65 7420 7370 6563 6965 7320 7061 6972  let species pair
+0000b190: 2074 6f20 6265 2062 6574 7765 656e 2066   to be between f
+0000b1a0: 6f63 616c 2061 6e64 2061 6c6c 2074 6865  ocal and all the
+0000b1b0: 2072 656d 6169 6e69 6e67 2073 7065 6369   remaining speci
+0000b1c0: 6573 2c20 6465 6661 756c 7420 4e6f 6e65  es, default None
+0000b1d0: 0a2d 6773 2c20 2d2d 6773 6d61 702c 2074  .-gs, --gsmap, t
+0000b1e0: 6865 2067 656e 6520 6e61 6d65 2d73 7065  he gene name-spe
+0000b1f0: 6369 6573 206e 616d 6520 6d61 702c 2064  cies name map, d
+0000b200: 6566 6175 6c74 204e 6f6e 650a 2d73 702c  efault None.-sp,
+0000b210: 202d 2d73 7065 6369 6573 7472 6565 2c20   --speciestree, 
+0000b220: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
+0000b230: 2074 6f20 7065 7266 6f72 6d20 7261 7465   to perform rate
+0000b240: 2063 6f72 7265 6374 696f 6e2c 2064 6566   correction, def
+0000b250: 6175 6c74 204e 6f6e 652c 2069 6620 4e6f  ault None, if No
+0000b260: 6e65 2077 6173 2067 6976 656e 2c20 7468  ne was given, th
+0000b270: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+0000b280: 6e20 616e 616c 7973 6973 2077 6f75 6c64  n analysis would
+0000b290: 2062 6520 6361 6c6c 6564 206f 6666 0a2d   be called off.-
+0000b2a0: 706b 2c20 2d2d 706c 6f74 6b64 652c 2066  pk, --plotkde, f
+0000b2b0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+0000b2c0: 6865 7220 746f 2070 6c6f 7420 6b64 6520  her to plot kde 
+0000b2d0: 6375 7276 6520 7570 6f6e 2068 6973 746f  curve upon histo
+0000b2e0: 6772 616d 2c20 6966 2074 6865 2066 6c61  gram, if the fla
+0000b2f0: 6720 7761 7320 7365 742c 206b 6465 2063  g was set, kde c
+0000b300: 7572 7665 2077 6f75 6c64 2062 6520 6164  urve would be ad
+0000b310: 6465 640a 2d72 772c 202d 2d72 6577 6569  ded.-rw, --rewei
+0000b320: 6768 742c 2066 6c61 6720 6f70 7469 6f6e  ght, flag option
+0000b330: 2c20 7768 6574 6865 7220 746f 2072 6563  , whether to rec
+0000b340: 616c 6375 6c61 7465 2074 6865 2077 6569  alculate the wei
+0000b350: 6768 7420 7065 7220 7370 6563 6965 7320  ght per species 
+0000b360: 7061 6972 2c20 6966 2074 6865 2066 6c61  pair, if the fla
+0000b370: 6720 7761 7320 7365 742c 2074 6865 2077  g was set, the w
+0000b380: 6569 6768 7420 776f 756c 6420 6265 2072  eight would be r
+0000b390: 6563 616c 6375 6c61 7465 640a 2d6f 722c  ecalculated.-or,
+0000b3a0: 202d 2d6f 6e6c 7972 6f6f 746f 7574 2c20   --onlyrootout, 
+0000b3b0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+0000b3c0: 7468 6572 2074 6f20 6f6e 6c79 2063 6f6e  ther to only con
+0000b3d0: 6475 6374 2072 6174 6520 636f 7272 6563  duct rate correc
+0000b3e0: 7469 6f6e 2075 7369 6e67 2074 6865 206f  tion using the o
+0000b3f0: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
+0000b400: 6173 206f 7574 6772 6f75 702c 2069 6620  as outgroup, if 
+0000b410: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+0000b420: 2c20 6f6e 6c79 2074 6865 206f 7574 6772  , only the outgr
+0000b430: 6f75 7020 6174 2072 6f6f 7420 776f 756c  oup at root woul
+0000b440: 6420 6265 2075 7365 6420 6173 206f 7574  d be used as out
+0000b450: 6772 6f75 700a 2d69 7465 722c 202d 2d65  group.-iter, --e
+0000b460: 6d5f 6974 6572 6174 696f 6e73 2c20 7468  m_iterations, th
+0000b470: 6520 6d61 7869 6d75 6d20 454d 2069 7465  e maximum EM ite
+0000b480: 7261 7469 6f6e 732c 2064 6566 6175 6c74  rations, default
+0000b490: 2032 3030 0a2d 696e 6974 2c20 2d2d 656d   200.-init, --em
+0000b4a0: 5f69 6e69 7469 616c 697a 6174 696f 6e73  _initializations
+0000b4b0: 2c20 7468 6520 6d61 7869 6d75 6d20 454d  , the maximum EM
+0000b4c0: 2069 6e69 7469 616c 697a 6174 696f 6e73   initializations
+0000b4d0: 2c20 6465 6661 756c 7420 3230 300a 2d70  , default 200.-p
+0000b4e0: 7263 742c 202d 2d70 726f 6d69 6e65 6e63  rct, --prominenc
+0000b4f0: 655f 6375 746f 6666 2c20 7468 6520 7072  e_cutoff, the pr
+0000b500: 6f6d 696e 656e 6365 2063 7574 6f66 6620  ominence cutoff 
+0000b510: 6f66 2061 6363 6570 7461 626c 6520 7065  of acceptable pe
+0000b520: 616b 732c 2064 6566 6175 6c74 2030 2e31  aks, default 0.1
+0000b530: 0a2d 7268 2c20 2d2d 7265 6c5f 6865 6967  .-rh, --rel_heig
+0000b540: 6874 2c20 7468 6520 7265 6c61 7469 7665  ht, the relative
+0000b550: 2068 6569 6768 7420 6174 2077 6869 6368   height at which
+0000b560: 2074 6865 2070 6561 6b20 7769 6474 6820   the peak width 
+0000b570: 6973 206d 6561 7375 7265 642c 2064 6566  is measured, def
+0000b580: 6175 6c74 2030 2e34 0a2d 736d 2c20 2d2d  ault 0.4.-sm, --
+0000b590: 7365 676d 656e 7473 2c20 7468 6520 7365  segments, the se
+0000b5a0: 676d 656e 7473 2064 6174 6166 696c 652c  gments datafile,
+0000b5b0: 2064 6566 6175 6c74 204e 6f6e 650a 2d6d   default None.-m
+0000b5c0: 6c2c 202d 2d6d 696e 6c65 6e2c 2074 6865  l, --minlen, the
+0000b5d0: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
+0000b5e0: 6f66 2061 2073 6361 6666 6f6c 6420 746f  of a scaffold to
+0000b5f0: 2062 6520 696e 636c 7564 6564 2069 6e20   be included in 
+0000b600: 646f 7470 6c6f 742c 2064 6566 6175 6c74  dotplot, default
+0000b610: 202d 312c 2069 6620 2d31 2077 6173 2073   -1, if -1 was s
+0000b620: 6574 2c20 7468 6520 3130 2520 6f66 2074  et, the 10% of t
+0000b630: 6865 206c 6f6e 6765 7374 2073 6361 6666  he longest scaff
+0000b640: 6f6c 6473 2077 696c 6c20 6265 2073 6574  olds will be set
+0000b650: 0a2d 6d73 2c20 2d2d 6d61 7873 697a 652c  .-ms, --maxsize,
+0000b660: 2074 6865 206d 6178 696d 756d 2066 616d   the maximum fam
+0000b670: 696c 7920 7369 7a65 2074 6f20 6265 2069  ily size to be i
+0000b680: 6e63 6c75 6465 642c 2064 6566 6175 6c74  ncluded, default
+0000b690: 2032 3030 0a2d 6170 2c20 2d2d 616e 6368   200.-ap, --anch
+0000b6a0: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
+0000b6b0: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
+0000b6c0: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+0000b6d0: 6e65 0a2d 6d74 2c20 2d2d 6d75 6c74 6970  ne.-mt, --multip
+0000b6e0: 6c69 636f 6e2c 2074 6865 206d 756c 7469  licon, the multi
+0000b6f0: 706c 6963 6f6e 7320 6461 7461 6669 6c65  plicons datafile
+0000b700: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+0000b710: 6774 2c20 2d2d 6765 6e65 7461 626c 652c  gt, --genetable,
+0000b720: 2074 6865 2067 656e 6520 7461 626c 6520   the gene table 
+0000b730: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+0000b740: 7420 4e6f 6e65 0a2d 6d67 2c20 2d2d 6d69  t None.-mg, --mi
+0000b750: 6e73 6567 6c65 6e2c 2074 6865 206d 696e  nseglen, the min
+0000b760: 696d 756d 206c 656e 6774 6820 6f66 2073  imum length of s
+0000b770: 6567 6d65 6e74 7320 746f 2069 6e63 6c75  egments to inclu
+0000b780: 6465 2c20 696e 2072 6174 696f 2069 6620  de, in ratio if 
+0000b790: 3c3d 2031 2c20 6465 6661 756c 7420 3130  <= 1, default 10
+0000b7a0: 3030 300a 2d6d 676e 2c20 2d2d 6d69 6e67  000.-mgn, --ming
+0000b7b0: 656e 656e 756d 2c20 7468 6520 6d69 6e69  enenum, the mini
+0000b7c0: 6d75 6d20 6e75 6d62 6572 206f 6620 6765  mum number of ge
+0000b7d0: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+0000b7e0: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+0000b7f0: 6564 2c20 6465 6661 756c 7420 3330 0a2d  ed, default 30.-
+0000b800: 6b72 2c20 2d2d 6b65 6570 7265 6475 6e2c  kr, --keepredun,
+0000b810: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+0000b820: 6574 6865 7220 746f 206b 6565 7020 7265  ether to keep re
+0000b830: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
+0000b840: 636f 6e73 2c20 6966 2074 6865 2066 6c61  cons, if the fla
+0000b850: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+0000b860: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+0000b870: 6963 6f6e 7320 776f 756c 6420 6265 206b  icons would be k
+0000b880: 6570 740a 2d65 706b 2c20 2d2d 6578 7472  ept.-epk, --extr
+0000b890: 6170 6172 616e 6f6d 656b 732c 2065 7874  aparanomeks, ext
+0000b8a0: 7261 2070 6172 616e 6f6d 6520 4b73 2064  ra paranome Ks d
+0000b8b0: 6174 6120 746f 2070 6c6f 7420 696e 2074  ata to plot in t
+0000b8c0: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
+0000b8d0: 7269 6275 7469 6f6e 2c20 6465 6661 756c  ribution, defaul
+0000b8e0: 7420 4e6f 6e65 0a2d 7061 672c 202d 2d70  t None.-pag, --p
+0000b8f0: 6c6f 7461 7067 6d6d 2c20 666c 6167 206f  lotapgmm, flag o
+0000b900: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+0000b910: 6f20 636f 6e64 7563 7420 616e 6420 706c  o conduct and pl
+0000b920: 6f74 206d 6978 7475 7265 206d 6f64 656c  ot mixture model
+0000b930: 696e 6720 6f66 2061 6e63 686f 7220 4b73  ing of anchor Ks
+0000b940: 2069 6e20 7468 6520 6d69 7865 6420 4b73   in the mixed Ks
+0000b950: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
+0000b960: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+0000b970: 6574 2c20 7468 6520 6d69 7874 7572 6520  et, the mixture 
+0000b980: 6d6f 6465 6c69 6e67 206f 6620 616e 6368  modeling of anch
+0000b990: 6f72 204b 7320 776f 756c 6420 6265 2063  or Ks would be c
+0000b9a0: 6f6e 6475 6374 6564 2061 6e64 2070 6c6f  onducted and plo
+0000b9b0: 7474 6564 0a2d 7065 6d2c 202d 2d70 6c6f  tted.-pem, --plo
+0000b9c0: 7465 6c6d 6d2c 2066 6c61 6720 6f70 7469  telmm, flag opti
+0000b9d0: 6f6e 2c20 7768 6574 6865 7220 746f 2063  on, whether to c
+0000b9e0: 6f6e 6475 6374 2061 6e64 2070 6c6f 7420  onduct and plot 
+0000b9f0: 656c 6d6d 206d 6978 7475 7265 206d 6f64  elmm mixture mod
+0000ba00: 656c 696e 6720 6f66 2070 6172 616e 6f6d  eling of paranom
+0000ba10: 6520 4b73 2069 6e20 7468 6520 6d69 7865  e Ks in the mixe
+0000ba20: 6420 4b73 2064 6973 7472 6962 7574 696f  d Ks distributio
+0000ba30: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
+0000ba40: 6173 2073 6574 2c20 7468 6520 656c 6d6d  as set, the elmm
+0000ba50: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
+0000ba60: 6720 6f66 2070 6172 616e 6f6d 6520 4b73  g of paranome Ks
+0000ba70: 2077 6f75 6c64 2062 6520 636f 6e64 7563   would be conduc
+0000ba80: 7465 6420 616e 6420 706c 6f74 7465 640a  ted and plotted.
+0000ba90: 2d63 2c20 2d2d 636f 6d70 6f6e 656e 7473  -c, --components
+0000baa0: 2c20 7468 6520 7261 6e67 6520 6f66 2074  , the range of t
+0000bab0: 6865 206e 756d 6265 7220 6f66 2063 6f6d  he number of com
+0000bac0: 706f 6e65 6e74 7320 746f 2066 6974 2069  ponents to fit i
+0000bad0: 6e20 616e 6368 6f72 204b 7320 6d69 7874  n anchor Ks mixt
+0000bae0: 7572 6520 6d6f 6465 6c69 6e67 2c20 6465  ure modeling, de
+0000baf0: 6661 756c 7420 2831 2c34 290a 2d70 7379  fault (1,4).-psy
+0000bb00: 2c20 2d2d 706c 6f74 7379 6e2c 2066 6c61  , --plotsyn, fla
+0000bb10: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+0000bb20: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+0000bb30: 6520 7379 6e74 656e 7920 706c 6f74 2c20  e synteny plot, 
+0000bb40: 6f6e 6c79 2077 6865 6e20 7468 6520 666c  only when the fl
+0000bb50: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+0000bb60: 7379 6e74 656e 7920 706c 6f74 2077 6f75  synteny plot wou
+0000bb70: 6c64 2062 6520 7072 6f64 7563 6564 0a2d  ld be produced.-
+0000bb80: 6473 2c20 2d2d 646f 7473 697a 652c 2074  ds, --dotsize, t
+0000bb90: 6865 2064 6f74 2073 697a 6520 696e 2064  he dot size in d
+0000bba0: 6f74 2070 6c6f 742c 2064 6566 6175 6c74  ot plot, default
+0000bbb0: 2030 2e33 0a2d 6161 2c20 2d2d 6170 616c   0.3.-aa, --apal
+0000bbc0: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+0000bbd0: 206f 6620 616e 6368 6f72 2064 6f74 732c   of anchor dots,
+0000bbe0: 2064 6566 6175 6c74 2031 0a2d 6861 2c20   default 1.-ha, 
+0000bbf0: 2d2d 686f 616c 7068 612c 2074 6865 206f  --hoalpha, the o
+0000bc00: 7061 6369 7479 206f 6620 686f 6d6f 6c6f  pacity of homolo
+0000bc10: 6720 646f 7473 2c20 6465 6661 756c 7420  g dots, default 
+0000bc20: 300a 2d73 7274 2c20 2d2d 7368 6f77 7265  0.-srt, --showre
+0000bc30: 616c 7469 636b 2c20 666c 6167 206f 7074  altick, flag opt
+0000bc40: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+0000bc50: 7368 6f77 2074 6865 2072 6561 6c20 7469  show the real ti
+0000bc60: 636b 2069 6e20 6765 6e65 7320 6f72 2062  ck in genes or b
+0000bc70: 6173 6573 2c20 6966 2074 6865 2066 6c61  ases, if the fla
+0000bc80: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+0000bc90: 6561 6c20 7469 636b 2077 6f75 6c64 2062  eal tick would b
+0000bca0: 6520 7368 6f77 6564 0a2d 746c 732c 202d  e showed.-tls, -
+0000bcb0: 2d74 6963 6b6c 6162 656c 7369 7a65 2c20  -ticklabelsize, 
+0000bcc0: 7468 6520 6c61 6265 6c20 7369 7a65 206f  the label size o
+0000bcd0: 6620 7469 636b 2c20 6465 6661 756c 7420  f tick, default 
+0000bce0: 350a 2d78 6c2c 202d 2d78 6c69 6d2c 2074  5.-xl, --xlim, t
+0000bcf0: 6865 2078 2061 7869 7320 6c69 6d69 7420  he x axis limit 
+0000bd00: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
+0000bd10: 6f6e 0a2d 796c 2c20 2d2d 796c 696d 2c20  on.-yl, --ylim, 
+0000bd20: 7468 6520 7920 6178 6973 206c 696d 6974  the y axis limit
+0000bd30: 206f 6620 4b73 2064 6973 7472 6962 7574   of Ks distribut
+0000bd40: 696f 6e0a 2d61 646f 2c20 2d2d 6164 6a75  ion.-ado, --adju
+0000bd50: 7374 6f72 7468 6f2c 2066 6c61 6720 6f70  stortho, flag op
+0000bd60: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+0000bd70: 2061 646a 7573 7420 7468 6520 6869 7374   adjust the hist
+0000bd80: 6f67 7261 6d20 6865 6967 6874 206f 6620  ogram height of 
+0000bd90: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2061  orthologous Ks a
+0000bda0: 7320 746f 206d 6174 6368 2074 6865 2068  s to match the h
+0000bdb0: 6569 6768 7420 6f66 2070 6172 616c 6f67  eight of paralog
+0000bdc0: 6f75 7320 4b73 2c20 6966 2074 6865 2066  ous Ks, if the f
+0000bdd0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+0000bde0: 2061 646a 7573 746d 656e 7420 776f 756c   adjustment woul
+0000bdf0: 6420 6265 2063 6f6e 6475 6374 6564 0a2d  d be conducted.-
+0000be00: 6164 662c 202d 2d61 646a 7573 7466 6163  adf, --adjustfac
+0000be10: 746f 722c 2074 6865 2061 646a 7573 746d  tor, the adjustm
+0000be20: 656e 7420 6661 6374 6f72 206f 6620 6f72  ent factor of or
+0000be30: 7468 6f6c 6f67 6f75 7320 4b73 2c20 6465  thologous Ks, de
+0000be40: 6661 756c 7420 302e 350a 2d6f 612c 202d  fault 0.5.-oa, -
+0000be50: 2d6f 6b61 6c70 6861 2c20 7468 6520 6f70  -okalpha, the op
+0000be60: 6163 6974 7920 6f66 206f 7274 686f 6c6f  acity of ortholo
+0000be70: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
+0000be80: 7469 6f6e 2069 6e20 6d69 7865 6420 706c  tion in mixed pl
+0000be90: 6f74 2c20 6465 6661 756c 7420 302e 350a  ot, default 0.5.
+0000bea0: 2d63 732c 202d 2d63 6c61 7373 6963 2c20  -cs, --classic, 
+0000beb0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+0000bec0: 7468 6572 2074 6f20 6472 6177 206d 6978  ther to draw mix
+0000bed0: 6564 204b 7320 706c 6f74 2069 6e20 6120  ed Ks plot in a 
+0000bee0: 636c 6173 7369 6320 6d61 6e6e 6572 2077  classic manner w
+0000bef0: 6865 7265 2074 6865 2066 756c 6c20 6f72  here the full or
+0000bf00: 7468 6f6c 6f67 6f75 7320 4b73 2064 6973  thologous Ks dis
+0000bf10: 7472 6962 7574 696f 6e20 6973 2064 7261  tribution is dra
+0000bf20: 7765 642c 2069 6620 7468 6520 666c 6167  wed, if the flag
+0000bf30: 2077 6173 2073 6574 2c20 7468 6520 636c   was set, the cl
+0000bf40: 6173 7369 6320 6d69 7865 6420 4b73 2070  assic mixed Ks p
+0000bf50: 6c6f 7420 776f 756c 6420 6265 2064 7261  lot would be dra
+0000bf60: 776e 0a2d 7461 2c20 2d2d 746f 7061 7272  wn.-ta, --toparr
+0000bf70: 6f77 2c20 666c 6167 206f 7074 696f 6e2c  ow, flag option,
+0000bf80: 2077 6865 7468 6572 2074 6f20 6164 6a75   whether to adju
+0000bf90: 7374 2074 6865 2061 7272 6f77 2074 6f20  st the arrow to 
+0000bfa0: 6265 2061 7420 7468 6520 746f 7020 6f66  be at the top of
+0000bfb0: 2074 6865 2070 6c6f 742c 2069 6e73 7465   the plot, inste
+0000bfc0: 6164 206f 6620 6265 696e 6720 636f 6f72  ad of being coor
+0000bfd0: 6469 6e61 7465 6420 6173 2074 6865 204b  dinated as the K
+0000bfe0: 4445 206f 6620 7468 6520 6f72 7468 6f6c  DE of the orthol
+0000bff0: 6f67 6f75 7320 4b73 2064 6973 7472 6962  ogous Ks distrib
+0000c000: 7574 696f 6e2c 2069 6620 7468 6520 666c  ution, if the fl
+0000c010: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+0000c020: 6172 726f 7720 776f 756c 6420 6265 2073  arrow would be s
+0000c030: 6574 2061 7420 7468 6520 746f 700a 2d6e  et at the top.-n
+0000c040: 612c 202d 2d6e 6f64 6561 7665 7261 6765  a, --nodeaverage
+0000c050: 642c 2066 6c61 6720 6f70 7469 6f6e 2c20  d, flag option, 
+0000c060: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
+0000c070: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
+0000c080: 686f 6420 666f 7220 6465 2d72 6564 756e  hod for de-redun
+0000c090: 6461 6e63 792c 2069 6620 7468 6520 666c  dancy, if the fl
+0000c0a0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+0000c0b0: 6e6f 6465 2d61 7665 7261 6765 6420 6d65  node-averaged me
+0000c0c0: 7468 6f64 2077 6f75 6c64 2062 6520 696e  thod would be in
+0000c0d0: 6974 6961 7465 640a 2d62 732c 202d 2d62  itiated.-bs, --b
+0000c0e0: 6f6f 7473 7472 6170 2c20 7468 6520 6e75  ootstrap, the nu
+0000c0f0: 6d62 6572 206f 6620 626f 6f74 7374 7261  mber of bootstra
+0000c100: 7020 7265 706c 6963 6174 6573 206f 6620  p replicates of 
+0000c110: 6f72 7468 6f6c 6f67 204b 7320 6469 7374  ortholog Ks dist
+0000c120: 7269 6275 7469 6f6e 2069 6e20 6d69 7865  ribution in mixe
+0000c130: 6420 706c 6f74 0a2d 6772 2c20 2d2d 6769  d plot.-gr, --gi
+0000c140: 7374 7262 2c20 666c 6167 206f 7074 696f  strb, flag optio
+0000c150: 6e2c 2077 6865 7468 6572 2074 6f20 7573  n, whether to us
+0000c160: 6520 6769 7374 5f72 6169 6e62 6f77 2061  e gist_rainbow a
+0000c170: 7320 636f 6c6f 7220 6d61 7020 6f66 2064  s color map of d
+0000c180: 6f74 706c 6f74 0a2d 6e2c 202d 2d6e 7468  otplot.-n, --nth
+0000c190: 7265 6164 732c 2074 6865 206e 756d 6265  reads, the numbe
+0000c1a0: 7220 6f66 2074 6872 6561 6473 2074 6f20  r of threads to 
+0000c1b0: 7573 6520 696e 2062 6f6f 7473 7472 6170  use in bootstrap
+0000c1c0: 2073 616d 706c 696e 672c 2064 6566 6175   sampling, defau
+0000c1d0: 6c74 2031 0a60 6060 0a0a 2323 2055 7361  lt 1.```..## Usa
+0000c1e0: 6765 0a0a 4865 7265 2077 6520 7072 6f76  ge..Here we prov
+0000c1f0: 6964 6564 2074 6865 2062 6173 6963 2075  ided the basic u
+0000c200: 7361 6765 2066 6f72 2065 6163 6820 7072  sage for each pr
+0000c210: 6f67 7261 6d20 616e 6420 7468 6520 7265  ogram and the re
+0000c220: 6c65 7661 6e74 2070 6172 616d 6574 6572  levant parameter
+0000c230: 7320 616e 6420 7375 6767 6573 7469 6f6e  s and suggestion
+0000c240: 7320 6f6e 2070 6172 616d 6574 6572 697a  s on parameteriz
+0000c250: 6174 696f 6e2e 2041 2072 656d 696e 6465  ation. A reminde
+0000c260: 7220 7468 6174 2074 6865 2067 6976 656e  r that the given
+0000c270: 2063 6f72 6573 2061 6e64 2074 6872 6561   cores and threa
+0000c280: 6473 2063 616e 2073 6967 6e69 6669 6361  ds can significa
+0000c290: 6e74 6c79 2069 6d70 6163 7420 7468 6520  ntly impact the 
+0000c2a0: 7275 6e20 7469 6d65 2061 6e64 2074 6875  run time and thu
+0000c2b0: 7320 7765 2061 6464 6564 2073 6f6d 6520  s we added some 
+0000c2c0: 7265 706f 7274 2069 6e66 6f72 6d61 7469  report informati
+0000c2d0: 6f6e 2070 6572 7461 696e 696e 6720 746f  on pertaining to
+0000c2e0: 2074 6865 2073 7973 7465 6d20 6f66 2075   the system of u
+0000c2f0: 7365 7273 2074 6f20 6661 6369 6c69 7461  sers to facilita
+0000c300: 7465 2074 6865 2065 6666 6963 6965 6e74  te the efficient
+0000c310: 2073 6574 7469 6e67 206f 6620 7468 7265   setting of thre
+0000c320: 6164 7320 616e 6420 6d65 6d6f 7279 2e20  ads and memory. 
+0000c330: 5468 6520 6c6f 6769 6361 6c20 4350 5573  The logical CPUs
+0000c340: 2072 6570 6f72 7465 6420 7265 7072 6573   reported repres
+0000c350: 656e 7473 2074 6865 206e 756d 6265 7220  ents the number 
+0000c360: 6f66 2070 6879 7369 6361 6c20 636f 7265  of physical core
+0000c370: 7320 6d75 6c74 6970 6c69 6564 2062 7920  s multiplied by 
+0000c380: 7468 6520 6e75 6d62 6572 206f 6620 7468  the number of th
+0000c390: 7265 6164 7320 7468 6174 2063 616e 2072  reads that can r
+0000c3a0: 756e 206f 6e20 6561 6368 2063 6f72 652c  un on each core,
+0000c3b0: 2061 6c73 6f20 6b6e 6f77 6e20 6173 2048   also known as H
+0000c3c0: 7970 6572 2054 6872 6561 6469 6e67 2e20  yper Threading. 
+0000c3d0: 5468 6520 6e75 6d62 6572 206f 6620 6c6f  The number of lo
+0000c3e0: 6769 6361 6c20 4350 5573 206d 6179 206e  gical CPUs may n
+0000c3f0: 6f74 206e 6563 6573 7361 7269 6c79 2062  ot necessarily b
+0000c400: 6520 6571 7569 7661 6c65 6e74 2074 6f20  e equivalent to 
+0000c410: 7468 6520 6163 7475 616c 206e 756d 6265  the actual numbe
+0000c420: 7220 6f66 2043 5055 7320 7468 6520 6375  r of CPUs the cu
+0000c430: 7272 656e 7420 7072 6f63 6573 7320 6361  rrent process ca
+0000c440: 6e20 7573 652e 2054 6865 2061 7661 696c  n use. The avail
+0000c450: 6162 6c65 206d 656d 6f72 7920 7265 6665  able memory refe
+0000c460: 7273 2074 6f20 7468 6520 6d65 6d6f 7279  rs to the memory
+0000c470: 2074 6861 7420 6361 6e20 6265 2067 6976   that can be giv
+0000c480: 656e 2069 6e73 7461 6e74 6c79 2074 6f20  en instantly to 
+0000c490: 7072 6f63 6573 7365 7320 7769 7468 6f75  processes withou
+0000c4a0: 7420 7468 6520 7379 7374 656d 2067 6f69  t the system goi
+0000c4b0: 6e67 2069 6e74 6f20 7377 6170 2061 6e64  ng into swap and
+0000c4c0: 2072 6566 6c65 6374 7320 7468 6520 6163   reflects the ac
+0000c4d0: 7475 616c 206d 656d 6f72 7920 6176 6169  tual memory avai
+0000c4e0: 6c61 626c 652e 2054 6865 2066 7265 6520  lable. The free 
+0000c4f0: 6d65 6d6f 7279 2072 6566 6572 7320 746f  memory refers to
+0000c500: 2074 6865 206d 656d 6f72 7920 6e6f 7420   the memory not 
+0000c510: 6265 696e 6720 7573 6564 2061 7420 616c  being used at al
+0000c520: 6c20 287a 6572 6f65 6429 2074 6861 7420  l (zeroed) that 
+0000c530: 6973 2072 6561 6469 6c79 2061 7661 696c  is readily avail
+0000c540: 6162 6c65 2e20 5468 6520 6465 7363 7269  able. The descri
+0000c550: 7074 696f 6e20 6162 6f76 6520 7265 6665  ption above refe
+0000c560: 7273 2074 6f20 7468 6520 646f 6375 6d65  rs to the docume
+0000c570: 6e74 6174 696f 6e20 6f66 2060 7073 7574  ntation of `psut
+0000c580: 696c 602e 0a0a 2323 2320 7767 6420 646d  il`...### wgd dm
+0000c590: 640a 0a2a 2a54 6865 2064 656c 696e 6561  d..**The delinea
+0000c5a0: 7469 6f6e 206f 6620 7768 6f6c 6520 7061  tion of whole pa
+0000c5b0: 7261 6e6f 6d65 2a2a 0a60 6060 0a77 6764  ranome**.```.wgd
+0000c5c0: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+0000c5d0: 6f65 7275 6c65 6120 2d49 2032 202d 6520  oerulea -I 2 -e 
+0000c5e0: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+0000c5f0: 7020 3520 282d 6e6e 2920 282d 2d74 6f5f  p 5 (-nn) (--to_
+0000c600: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
+0000c610: 6e20 3429 2028 2d6f 2077 6764 5f64 6d64  n 4) (-o wgd_dmd
+0000c620: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
+0000c630: 7029 0a60 6060 200a 0a4e 6f74 6520 7468  p).``` ..Note th
+0000c640: 6174 2077 6520 646f 6e27 7420 7072 6f76  at we don't prov
+0000c650: 6964 6520 7468 6520 6461 7461 206f 6620  ide the data of 
+0000c660: 7468 6973 2063 6f64 696e 6720 7365 7175  this coding sequ
+0000c670: 656e 6365 2028 6364 7329 2066 696c 6520  ence (cds) file 
+0000c680: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
+0000c690: 6c65 6160 2062 7574 2069 7420 6361 6e20  lea` but it can 
+0000c6a0: 6265 2065 6173 696c 7920 646f 776e 6c6f  be easily downlo
+0000c6b0: 6164 6564 2061 7420 5b50 6879 746f 7a6f  aded at [Phytozo
+0000c6c0: 6d65 5d28 6874 7470 733a 2f2f 7068 7974  me](https://phyt
+0000c6d0: 6f7a 6f6d 652d 6e65 7874 2e6a 6769 2e64  ozome-next.jgi.d
+0000c6e0: 6f65 2e67 6f76 2f69 6e66 6f2f 4163 6f65  oe.gov/info/Acoe
+0000c6f0: 7275 6c65 615f 7633 5f31 2920 2873 616d  rulea_v3_1) (sam
+0000c700: 6520 666f 7220 6f74 6865 7220 5573 6167  e for other Usag
+0000c710: 6520 646f 6329 2e20 4120 7265 6d69 6e64  e doc). A remind
+0000c720: 6572 2074 6861 7420 696e 2074 6865 2069  er that in the i
+0000c730: 7373 7565 7320 736f 6d65 2075 7365 7273  ssues some users
+0000c740: 2064 6964 6e27 7420 646f 776e 6c6f 6164   didn't download
+0000c750: 2074 6865 2060 4163 6f65 7275 6c65 615f   the `Acoerulea_
+0000c760: 3332 325f 7633 2e31 2e63 6473 5f70 7269  322_v3.1.cds_pri
+0000c770: 6d61 7279 5472 616e 7363 7269 7074 4f6e  maryTranscriptOn
+0000c780: 6c79 2e66 612e 677a 6020 6275 7420 696e  ly.fa.gz` but in
+0000c790: 7374 6561 6420 7468 6520 6041 636f 6572  stead the `Acoer
+0000c7a0: 756c 6561 5f33 3232 5f76 332e 312e 6364  ulea_322_v3.1.cd
+0000c7b0: 732e 6661 2e67 7a60 2066 696c 652e 2046  s.fa.gz` file. F
+0000c7c0: 6f72 2074 6865 2063 6f6e 7374 7275 6374  or the construct
+0000c7d0: 696f 6e20 6f66 2077 686f 6c65 2070 6172  ion of whole par
+0000c7e0: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+0000c7f0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+0000c800: 6f6e 2c20 2a2a 6f6e 6c79 206f 6e65 2074  on, **only one t
+0000c810: 7261 6e73 6372 6970 7420 2874 6865 2070  ranscript (the p
+0000c820: 7269 6d61 7279 206f 6e65 2920 7065 7220  rimary one) per 
+0000c830: 6765 6e65 2073 686f 756c 6420 6265 2069  gene should be i
+0000c840: 6e63 6c75 6465 642a 2a20 7375 6368 2074  ncluded** such t
+0000c850: 6861 7420 7468 6520 2a4b 2a3c 7375 623e  hat the *K*<sub>
+0000c860: 533c 2f73 7562 3e20 6973 2072 6561 6c6c  S</sub> is reall
+0000c870: 7920 696e 6469 6361 7469 6e67 2074 6865  y indicating the
+0000c880: 2061 6765 206f 6620 6765 6e65 2064 7570   age of gene dup
+0000c890: 6c69 6361 7469 6f6e 2065 7665 6e74 2c20  lication event, 
+0000c8a0: 696e 7374 6561 6420 6f66 2061 6c74 6572  instead of alter
+0000c8b0: 6e61 7469 7665 2073 706c 6963 696e 672e  native splicing.
+0000c8c0: 2054 7261 6e73 6372 6970 746f 6d65 2064   Transcriptome d
+0000c8d0: 6174 6120 7368 6f75 6c64 2062 6520 6361  ata should be ca
+0000c8e0: 7265 6675 6c6c 7920 7472 6561 7465 6420  refully treated 
+0000c8f0: 7769 7468 2064 652d 7265 6475 6e64 616e  with de-redundan
+0000c900: 6379 2073 6f20 6173 2074 6f20 7265 6475  cy so as to redu
+0000c910: 6365 2074 6865 2066 616c 7365 2070 6f73  ce the false pos
+0000c920: 6974 6976 6520 6475 706c 6963 6174 696f  itive duplicatio
+0000c930: 6e20 6275 6d70 2063 6175 7365 6420 6279  n bump caused by
+0000c940: 2070 6572 7661 7369 7665 2061 6c74 6572   pervasive alter
+0000c950: 6e61 7469 7665 2073 706c 6963 696e 672e  native splicing.
+0000c960: 2048 6572 6520 7468 6520 696e 666c 6174   Here the inflat
+0000c970: 696f 6e20 6661 6374 6f72 2070 6172 616d  ion factor param
+0000c980: 6574 6572 2c20 6769 7665 6e20 6279 2060  eter, given by `
+0000c990: 2d49 6020 6f72 2060 2d2d 696e 666c 6174  -I` or `--inflat
+0000c9a0: 696f 6e60 2c20 6166 6665 6374 7320 7468  ion`, affects th
+0000c9b0: 6520 6772 616e 756c 6172 6974 7920 6f72  e granularity or
+0000c9c0: 2072 6573 6f6c 7574 696f 6e20 6f66 2074   resolution of t
+0000c9d0: 6865 2063 6c75 7374 6572 696e 6720 6f75  he clustering ou
+0000c9e0: 7463 6f6d 6520 616e 6420 696d 706c 6963  tcome and implic
+0000c9f0: 6974 6c79 2063 6f6e 7472 6f6c 6c73 2074  itly controlls t
+0000ca00: 6865 206e 756d 6265 7220 6f66 2063 6c75  he number of clu
+0000ca10: 7374 6572 732c 2077 6974 6820 6c6f 7720  sters, with low 
+0000ca20: 7661 6c75 6573 2073 7563 6820 6173 2031  values such as 1
+0000ca30: 2e33 206f 7220 312e 3420 6c65 6164 696e  .3 or 1.4 leadin
+0000ca40: 6720 746f 2066 6577 6572 2062 7574 206c  g to fewer but l
+0000ca50: 6172 6765 7220 636c 7573 7465 7273 2061  arger clusters a
+0000ca60: 6e64 2068 6967 6820 7661 6c75 6573 2073  nd high values s
+0000ca70: 7563 6820 6173 2035 206f 7220 3620 6c65  uch as 5 or 6 le
+0000ca80: 6164 696e 6720 746f 206d 6f72 6520 6275  ading to more bu
+0000ca90: 7420 736d 616c 6c65 7220 636c 7573 7465  t smaller cluste
+0000caa0: 7273 2e20 5765 2073 6574 2074 6865 2064  rs. We set the d
+0000cab0: 6566 6175 6c74 2076 616c 7565 2061 7320  efault value as 
+0000cac0: 3220 6173 2073 7567 6765 7374 6564 2062  2 as suggested b
+0000cad0: 7920 5b4d 434c 5d28 6874 7470 733a 2f2f  y [MCL](https://
+0000cae0: 6d69 6361 6e73 2e6f 7267 2f6d 636c 2f29  micans.org/mcl/)
+0000caf0: 2e20 5468 6520 652d 7661 6c75 6520 6375  . The e-value cu
+0000cb00: 742d 6f66 6620 666f 7220 7365 7175 656e  t-off for sequen
+0000cb10: 6365 2073 696d 696c 6172 6974 792c 2067  ce similarity, g
+0000cb20: 6976 656e 2062 7920 602d 6560 206f 7220  iven by `-e` or 
+0000cb30: 602d 2d65 7661 6c60 2c20 7768 6963 6820  `--eval`, which 
+0000cb40: 6465 6e6f 7465 7320 7468 6520 6578 7065  denotes the expe
+0000cb50: 6374 6564 2076 616c 7565 206f 6620 7468  cted value of th
+0000cb60: 6520 6869 7420 7175 616e 7469 6669 6573  e hit quantifies
+0000cb70: 2074 6865 206e 756d 6265 7220 6f66 2061   the number of a
+0000cb80: 6c69 676e 6d65 6e74 7320 6f66 2073 696d  lignments of sim
+0000cb90: 696c 6172 206f 7220 6265 7474 6572 2071  ilar or better q
+0000cba0: 7561 6c69 7479 2074 6861 7420 796f 7520  uality that you 
+0000cbb0: 6578 7065 6374 2074 6f20 6669 6e64 2073  expect to find s
+0000cbc0: 6561 7263 6869 6e67 2074 6869 7320 7175  earching this qu
+0000cbd0: 6572 7920 6167 6169 6e73 7420 6120 6461  ery against a da
+0000cbe0: 7461 6261 7365 206f 6620 7261 6e64 6f6d  tabase of random
+0000cbf0: 2073 6571 7565 6e63 6573 2074 6865 2073   sequences the s
+0000cc00: 616d 6520 7369 7a65 2061 7320 7468 6520  ame size as the 
+0000cc10: 6163 7475 616c 2074 6172 6765 7420 6461  actual target da
+0000cc20: 7461 6261 7365 2c20 6973 2074 6865 206b  tabase, is the k
+0000cc30: 6579 2070 6172 616d 6574 6572 206d 6561  ey parameter mea
+0000cc40: 7375 7269 6e67 2074 6865 2073 6967 6e69  suring the signi
+0000cc50: 6669 6361 6e63 6520 6f66 2061 2068 6974  ficance of a hit
+0000cc60: 2c20 7768 6963 6820 6973 2073 6574 2068  , which is set h
+0000cc70: 6572 6520 6173 2064 6566 6175 6c74 2031  ere as default 1
+0000cc80: 652d 3130 2e20 4e6f 7465 2074 6861 7420  e-10. Note that 
+0000cc90: 5b44 4941 4d4f 4e44 5d28 6874 7470 733a  [DIAMOND](https:
+0000cca0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 6275  //github.com/bbu
+0000ccb0: 6368 6669 6e6b 2f64 6961 6d6f 6e64 2f77  chfink/diamond/w
+0000ccc0: 696b 692f 312e 2d54 7574 6f72 6961 6c29  iki/1.-Tutorial)
+0000ccd0: 2069 7473 656c 6620 6279 2064 6566 6175   itself by defau
+0000cce0: 6c74 206f 6e6c 7920 7265 706f 7274 7320  lt only reports 
+0000ccf0: 616c 6c20 616c 6967 6e6d 656e 7473 2077  all alignments w
+0000cd00: 6974 6820 652d 7661 6c75 6520 3c20 302e  ith e-value < 0.
+0000cd10: 3030 312e 2054 6865 2070 6572 6365 6e74  001. The percent
+0000cd20: 6167 6520 6f66 2075 7070 6572 2068 6974  age of upper hit
+0000cd30: 7320 7573 6564 2066 6f72 2067 656e 6520  s used for gene 
+0000cd40: 6c65 6e67 7468 206e 6f72 6d61 6c69 7a61  length normaliza
+0000cd50: 7469 6f6e 2c20 6769 7665 6e20 6279 2060  tion, given by `
+0000cd60: 2d6e 7060 206f 7220 602d 2d6e 6f72 6d61  -np` or `--norma
+0000cd70: 6c69 7a65 6470 6572 6365 6e74 602c 2077  lizedpercent`, w
+0000cd80: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
+0000cd90: 7468 6520 7570 7065 7220 7065 7263 656e  the upper percen
+0000cda0: 7469 6c65 206f 6620 6869 7473 2070 6572  tile of hits per
+0000cdb0: 2062 696e 2028 6361 7465 676f 7269 7a65   bin (categorize
+0000cdc0: 6420 6279 2067 656e 6520 6c65 6e67 7468  d by gene length
+0000cdd0: 2920 7573 6564 2069 6e20 7468 6520 6669  ) used in the fi
+0000cde0: 7420 6f66 206c 696e 6561 7220 7265 6772  t of linear regr
+0000cdf0: 6573 7369 6f6e 2c20 636f 6e73 6964 6572  ession, consider
+0000ce00: 696e 6720 7468 6174 206e 6f74 2061 6c6c  ing that not all
+0000ce10: 2068 6974 7320 7065 7220 6269 6e20 7368   hits per bin sh
+0000ce20: 6f77 2061 7070 6172 656e 7420 6c69 6e65  ow apparent line
+0000ce30: 6172 2072 656c 6174 696f 6e73 6869 702c  ar relationship,
+0000ce40: 2069 7320 7365 7420 6173 2064 6566 6175   is set as defau
+0000ce50: 6c74 2035 2c20 696e 6469 6361 7469 6e67  lt 5, indicating
+0000ce60: 2074 6865 2075 7361 6765 206f 6620 746f   the usage of to
+0000ce70: 7020 3525 2068 6974 7320 7065 7220 6269  p 5% hits per bi
+0000ce80: 6e2e 2054 6865 206e 756d 6265 7220 6f66  n. The number of
+0000ce90: 2062 696e 7320 6469 7669 6465 6420 696e   bins divided in
+0000cea0: 2067 656e 6520 6c65 6e67 7468 206e 6f72   gene length nor
+0000ceb0: 6d61 6c69 7a61 7469 6f6e 2c20 6769 7665  malization, give
+0000cec0: 6e20 6279 2060 2d62 7360 206f 7220 602d  n by `-bs` or `-
+0000ced0: 2d62 696e 7360 2c20 6465 7465 726d 696e  -bins`, determin
+0000cee0: 6573 2074 6865 206e 756d 6265 7220 6f66  es the number of
+0000cef0: 2062 696e 7320 746f 2063 6174 6567 6f72   bins to categor
+0000cf00: 697a 6520 7468 6520 6765 6e65 206c 656e  ize the gene len
+0000cf10: 6774 682c 2069 7320 7365 7420 6173 2064  gth, is set as d
+0000cf20: 6566 6175 6c74 2031 3030 2e20 5468 6520  efault 100. The 
+0000cf30: 7061 7261 6d65 7465 7220 602d 6e6e 6020  parameter `-nn` 
+0000cf40: 6f72 2060 2d2d 6e6f 6e6f 726d 616c 697a  or `--nonormaliz
+0000cf50: 6174 696f 6e60 2063 616e 2062 6520 7365  ation` can be se
+0000cf60: 7420 746f 2063 616c 6c20 6f66 6620 7468  t to call off th
+0000cf70: 6520 6e6f 726d 616c 697a 6174 696f 6e20  e normalization 
+0000cf80: 7072 6f63 6573 732c 2061 6c74 686f 7567  process, althoug
+0000cf90: 6820 6974 2773 2073 7567 6765 7374 6564  h it's suggested
+0000cfa0: 2074 6f20 636f 6e64 7563 7420 7468 6520   to conduct the 
+0000cfb0: 6e6f 726d 616c 697a 6174 696f 6e20 746f  normalization to
+0000cfc0: 2061 6371 7569 7265 206d 6f72 6520 6163   acquire more ac
+0000cfd0: 6375 7261 7465 2067 656e 6520 6661 6d69  curate gene fami
+0000cfe0: 6c79 2063 6c75 7374 6572 696e 6720 7265  ly clustering re
+0000cff0: 7375 6c74 2e20 5468 6520 7061 7261 6d65  sult. The parame
+0000d000: 7465 7273 2060 2d2d 746f 5f73 746f 7060  ters `--to_stop`
+0000d010: 2061 6e64 2060 2d2d 6364 7360 2063 6f6e   and `--cds` con
+0000d020: 7472 6f6c 2074 6865 2062 6568 6176 696f  trol the behavio
+0000d030: 7572 206f 6620 7472 616e 736c 6174 696e  ur of translatin
+0000d040: 6720 636f 6469 6e67 2073 6571 7565 6e63  g coding sequenc
+0000d050: 6520 696e 746f 2061 6d69 6e6f 2061 6369  e into amino aci
+0000d060: 6420 7365 7175 656e 6365 2e20 4966 2074  d sequence. If t
+0000d070: 6865 2060 2d2d 746f 5f73 746f 7060 2077  he `--to_stop` w
+0000d080: 6173 2073 6574 2c20 7468 6520 7472 616e  as set, the tran
+0000d090: 736c 6174 696f 6e20 776f 756c 6420 6265  slation would be
+0000d0a0: 2074 6572 6d69 6e61 7465 6420 6174 2074   terminated at t
+0000d0b0: 6865 2066 6972 7374 2069 6e2d 6672 616d  he first in-fram
+0000d0c0: 6520 7374 6f70 2063 6f64 6f6e 2c20 6f74  e stop codon, ot
+0000d0d0: 6865 7277 6973 6520 7468 6520 7472 616e  herwise the tran
+0000d0e0: 736c 6174 696f 6e20 776f 756c 6420 7369  slation would si
+0000d0f0: 6d70 6c79 2073 6b69 7020 616e 7920 7374  mply skip any st
+0000d100: 6f70 2063 6f64 6f6e 732e 2049 6620 7468  op codons. If th
+0000d110: 6520 602d 2d63 6473 6020 7761 7320 7365  e `--cds` was se
+0000d120: 742c 2073 6571 7565 6e63 6573 2074 6861  t, sequences tha
+0000d130: 7420 646f 6573 6e27 7420 7374 6172 7420  t doesn't start 
+0000d140: 7769 7468 2061 2076 616c 6964 2073 7461  with a valid sta
+0000d150: 7274 2063 6f64 6f6e 2c20 6f72 2063 6f6e  rt codon, or con
+0000d160: 7461 696e 7320 6d6f 7265 2074 6861 6e20  tains more than 
+0000d170: 6f6e 6520 696e 2d66 7261 6d65 2073 746f  one in-frame sto
+0000d180: 7020 636f 646f 6e2c 206f 7220 6973 206e  p codon, or is n
+0000d190: 6f74 2064 6976 6964 6162 6c65 2062 7920  ot dividable by 
+0000d1a0: 332c 2077 6f75 6c64 2062 6520 7369 6d70  3, would be simp
+0000d1b0: 6c79 2064 726f 7070 6564 2c20 7375 6368  ly dropped, such
+0000d1c0: 2074 6861 7420 6f6e 6c79 2073 7472 6963   that only stric
+0000d1d0: 7420 636f 6d70 6c65 7465 2063 6f64 696e  t complete codin
+0000d1e0: 6720 7365 7175 656e 6365 7320 776f 756c  g sequences woul
+0000d1f0: 6420 6265 2069 6e63 6c75 6465 6420 696e  d be included in
+0000d200: 2074 6865 2073 7562 7365 7175 656e 7420   the subsequent 
+0000d210: 616e 616c 7973 6973 2e20 5468 6520 6578  analysis. The ex
+0000d220: 6163 7420 6265 6861 7669 6f75 7220 6f66  act behaviour of
+0000d230: 2060 2d2d 746f 5f73 746f 7060 2061 6e64   `--to_stop` and
+0000d240: 2060 2d2d 6364 7360 2069 7320 6465 6669   `--cds` is defi
+0000d250: 6e65 6420 616e 6420 6465 7363 7269 6265  ned and describe
+0000d260: 6420 696e 2074 6865 205b 6269 6f70 7974  d in the [biopyt
+0000d270: 686f 6e5d 2868 7474 7073 3a2f 2f62 696f  hon](https://bio
+0000d280: 7079 7468 6f6e 2e6f 7267 2f64 6f63 732f  python.org/docs/
+0000d290: 312e 3735 2f61 7069 2f42 696f 2e53 6571  1.75/api/Bio.Seq
+0000d2a0: 2e68 746d 6c23 4269 6f2e 5365 712e 5365  .html#Bio.Seq.Se
+0000d2b0: 712e 7472 616e 736c 6174 6529 206c 6962  q.translate) lib
+0000d2c0: 7261 7279 2e20 5468 6520 6e75 6d62 6572  rary. The number
+0000d2d0: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
+0000d2e0: 6561 6473 2062 7920 7468 6520 6f70 7469  eads by the opti
+0000d2f0: 6f6e 2060 2d6e 6020 6f72 2060 2d2d 6e74  on `-n` or `--nt
+0000d300: 6872 6561 6473 6020 6361 6e20 6265 2073  hreads` can be s
+0000d310: 6574 2074 6f20 6163 6365 6c65 7261 7465  et to accelerate
+0000d320: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
+0000d330: 2077 6974 6869 6e20 6064 6961 6d6f 6e64   within `diamond
+0000d340: 602e 2054 6865 2064 6972 6563 746f 7279  `. The directory
+0000d350: 206f 6620 6f75 7470 7574 206f 7220 696e   of output or in
+0000d360: 7465 726d 6564 6961 7465 2066 696c 6573  termediate files
+0000d370: 2069 7320 6465 7465 726d 696e 6564 2062   is determined b
+0000d380: 7920 7468 6520 7061 7261 6d65 7465 7220  y the parameter 
+0000d390: 602d 6f60 206f 7220 602d 2d6f 7574 6469  `-o` or `--outdi
+0000d3a0: 7260 2c20 616e 6420 602d 7460 206f 7220  r`, and `-t` or 
+0000d3b0: 602d 2d74 6d70 6469 7260 2c20 7768 6963  `--tmpdir`, whic
+0000d3c0: 6820 7769 6c6c 2062 6520 6372 6561 7465  h will be create
+0000d3d0: 6420 6279 2074 6865 2070 726f 6772 616d  d by the program
+0000d3e0: 2069 7473 656c 6620 616e 6420 6265 206f   itself and be o
+0000d3f0: 7665 7277 7269 7474 656e 2069 6620 7468  verwritten if th
+0000d400: 6520 666f 6c64 6572 2068 6173 2061 6c72  e folder has alr
+0000d410: 6561 6479 2062 6565 6e20 6372 6561 7465  eady been create
+0000d420: 642e 204e 6f74 6520 7468 6174 2074 6865  d. Note that the
+0000d430: 2073 6f66 7477 6172 6520 6064 6961 6d6f   software `diamo
+0000d440: 6e64 6020 7368 6f75 6c64 2062 6520 7072  nd` should be pr
+0000d450: 652d 696e 7374 616c 6c65 6420 616e 6420  e-installed and 
+0000d460: 7365 7420 696e 2074 6865 2065 6e76 6972  set in the envir
+0000d470: 6f6e 6d65 6e74 2070 6174 6820 696e 2061  onment path in a
+0000d480: 6c6c 2074 6865 2061 6e61 6c79 7369 7320  ll the analysis 
+0000d490: 7065 7266 6f72 6d65 6420 6279 2060 7767  performed by `wg
+0000d4a0: 6420 646d 6460 2065 7863 6570 7420 666f  d dmd` except fo
+0000d4b0: 7220 7468 6520 636f 6c6c 696e 6561 7220  r the collinear 
+0000d4c0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+0000d4d0: 7973 6973 2e0a 0a2a 2a57 6520 7375 6767  ysis...**We sugg
+0000d4e0: 6573 7420 7468 6174 2074 6865 2064 6566  est that the def
+0000d4f0: 6175 6c74 2073 6574 7469 6e67 2069 6e20  ault setting in 
+0000d500: 7768 6963 6820 7468 6520 696e 666c 6174  which the inflat
+0000d510: 696f 6e20 6661 6374 6f72 2069 7320 7365  ion factor is se
+0000d520: 7420 6173 2032 2c20 652d 7661 6c75 6520  t as 2, e-value 
+0000d530: 6375 742d 6f66 6620 6173 2031 652d 3130  cut-off as 1e-10
+0000d540: 2061 6e64 206f 7468 6572 2070 6172 616d   and other param
+0000d550: 6574 6572 7320 696e 2064 6566 6175 6c74  eters in default
+0000d560: 2069 7320 6120 676f 6f64 2073 7461 7274   is a good start
+0000d570: 696e 6720 706f 696e 742c 2075 6e6c 6573  ing point, unles
+0000d580: 7320 796f 7520 7370 6563 6966 6963 616c  s you specifical
+0000d590: 6c79 2077 616e 7420 746f 2065 7870 6c6f  ly want to explo
+0000d5a0: 7265 2074 6865 2065 6666 6563 7473 206f  re the effects o
+0000d5b0: 6620 6469 6666 6572 656e 7420 7061 7261  f different para
+0000d5c0: 6d65 7465 7273 2e2a 2a20 5375 6368 2074  meters.** Such t
+0000d5d0: 6861 7420 7468 6520 636f 6d6d 616e 6420  hat the command 
+0000d5e0: 666f 7220 7468 6520 6465 6c69 6e65 6174  for the delineat
+0000d5f0: 696f 6e20 6f66 2077 686f 6c65 2070 6172  ion of whole par
+0000d600: 616e 6f6d 6520 6973 2073 696d 706c 7920  anome is simply 
+0000d610: 6173 2062 656c 6f77 2e0a 0a60 6060 0a77  as below...```.w
+0000d620: 6764 2064 6d64 2041 7175 696c 6567 6961  gd dmd Aquilegia
+0000d630: 5f63 6f65 7275 6c65 610a 6060 600a 0a2a  _coerulea.```..*
+0000d640: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
+0000d650: 206f 6620 5242 4873 2a2a 0a60 6060 0a77   of RBHs**.```.w
+0000d660: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
+0000d670: 2073 6571 7565 6e63 6532 202d 6520 3165   sequence2 -e 1e
+0000d680: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
+0000d690: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
+0000d6a0: 2d63 2030 2e39 2920 282d 2d6f 6766 6f72  -c 0.9) (--ogfor
+0000d6b0: 6d61 7429 2028 2d2d 746f 5f73 746f 7029  mat) (--to_stop)
+0000d6c0: 2028 2d2d 6364 7329 2028 2d6f 2077 6764   (--cds) (-o wgd
+0000d6d0: 5f64 6d64 2920 282d 7420 776f 726b 696e  _dmd) (-t workin
+0000d6e0: 675f 746d 7029 0a60 6060 0a0a 546f 2064  g_tmp).```..To d
+0000d6f0: 656c 696e 6561 7465 2052 4248 7320 6265  elineate RBHs be
+0000d700: 7477 6565 6e20 7477 6f20 6364 7320 7365  tween two cds se
+0000d710: 7175 656e 6365 2066 696c 6573 2c20 7468  quence files, th
+0000d720: 6520 7265 6c65 7661 6e74 2070 6172 616d  e relevant param
+0000d730: 6574 6572 2069 7320 6d6f 7374 6c79 2074  eter is mostly t
+0000d740: 6865 2073 616d 6520 6173 2077 686f 6c65  he same as whole
+0000d750: 2070 6172 616e 6f6d 6520 696e 6665 7265   paranome infere
+0000d760: 6e63 652c 2065 7863 6570 7420 666f 7220  nce, except for 
+0000d770: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
+0000d780: 6360 206f 7220 602d 2d63 7363 6f72 6560  c` or `--cscore`
+0000d790: 2c20 7768 6963 6820 7261 6e67 6573 2062  , which ranges b
+0000d7a0: 6574 7765 656e 2030 2061 6e64 2031 2061  etween 0 and 1 a
+0000d7b0: 6e64 2069 7320 7573 6564 2074 6f20 7265  nd is used to re
+0000d7c0: 6c61 7820 7468 6520 7369 6d69 6c61 7269  lax the similari
+0000d7d0: 7479 2063 7574 6f66 6620 6672 6f6d 2074  ty cutoff from t
+0000d7e0: 6865 2065 7863 6c75 7369 7665 2072 6563  he exclusive rec
+0000d7f0: 6970 726f 6361 6c20 6265 7374 2068 6974  iprocal best hit
+0000d800: 7320 746f 2061 2063 6572 7461 696e 2072  s to a certain r
+0000d810: 6174 696f 2061 7320 746f 2074 6865 2062  atio as to the b
+0000d820: 6573 7420 6869 7473 2e20 466f 7220 696e  est hits. For in
+0000d830: 7374 616e 6365 2c20 6966 2074 6865 2067  stance, if the g
+0000d840: 656e 6520 6231 2066 726f 6d20 6765 6e6f  ene b1 from geno
+0000d850: 6d65 2042 2068 6173 2074 6865 2062 6573  me B has the bes
+0000d860: 7420 6869 7420 6765 6e65 2061 3120 6672  t hit gene a1 fr
+0000d870: 6f6d 2067 656e 6f6d 6520 4120 7769 7468  om genome A with
+0000d880: 2074 6865 2062 6974 2073 636f 7265 2061   the bit score a
+0000d890: 7320 3130 302c 2077 6869 6368 2069 7320  s 100, which is 
+0000d8a0: 6120 7363 6f72 696e 6720 6d61 7472 6978  a scoring matrix
+0000d8b0: 2069 6e64 6570 656e 6465 6e74 206d 6561   independent mea
+0000d8c0: 7375 7265 206f 6620 7468 6520 286c 6f63  sure of the (loc
+0000d8d0: 616c 2920 7369 6d69 6c61 7269 7479 206f  al) similarity o
+0000d8e0: 6620 7468 6520 7477 6f20 616c 6967 6e65  f the two aligne
+0000d8f0: 6420 7365 7175 656e 6365 732c 2077 6974  d sequences, wit
+0000d900: 6820 6c61 7267 6572 2076 616c 7565 7320  h larger values 
+0000d910: 696e 6469 6361 7469 6e67 2068 6967 6865  indicating highe
+0000d920: 7220 7369 6d69 6c61 7269 7469 6573 2c20  r similarities, 
+0000d930: 6769 7665 6e20 7468 6520 602d 6320 302e  given the `-c 0.
+0000d940: 3960 2c20 6765 6e65 7320 6672 6f6d 2067  9`, genes from g
+0000d950: 656e 6f6d 6520 4120 7768 6963 6820 6861  enome A which ha
+0000d960: 7320 7468 6520 6269 7420 7363 6f72 6520  s the bit score 
+0000d970: 7769 7468 2067 656e 6520 6231 2068 6967  with gene b1 hig
+0000d980: 6865 7220 7468 616e 2030 2e39 7831 3030  her than 0.9x100
+0000d990: 2077 696c 6c20 616c 736f 2062 6520 7772   will also be wr
+0000d9a0: 6974 7465 6e20 696e 2074 6865 2072 6573  itten in the res
+0000d9b0: 756c 7420 6669 6c65 2c20 7768 6963 6820  ult file, which 
+0000d9c0: 696e 2061 2073 656e 7365 2061 7265 206e  in a sense are n
+0000d9d0: 6f74 2052 4248 7320 616e 796d 6f72 6520  ot RBHs anymore 
+0000d9e0: 6f66 2063 6f75 7273 652c 2062 7574 2074  of course, but t
+0000d9f0: 6865 2068 6967 686c 7920 7369 6d69 6c61  he highly simila
+0000da00: 7220 686f 6d6f 6c6f 6775 6520 7061 6972  r homologue pair
+0000da10: 732e 2049 6620 6d6f 7265 2074 6861 6e20  s. If more than 
+0000da20: 3220 7365 7175 656e 6365 2066 696c 6573  2 sequence files
+0000da30: 2077 6572 6520 7072 6f76 6964 6564 2c20   were provided, 
+0000da40: 6576 6572 7920 7061 6972 2d77 6973 6520  every pair-wise 
+0000da50: 5242 4873 2077 6f75 6c64 2062 6520 6361  RBHs would be ca
+0000da60: 6c63 756c 6174 6564 2065 7863 6570 7420  lculated except 
+0000da70: 666f 7220 7175 6572 7969 6e67 2074 6865  for querying the
+0000da80: 2073 616d 6520 7365 7175 656e 6365 2069   same sequence i
+0000da90: 7473 656c 662e 2054 6865 206e 756d 6265  tself. The numbe
+0000daa0: 7220 6f66 2070 6172 616c 6c65 6c20 7468  r of parallel th
+0000dab0: 7265 6164 7320 746f 2062 6f6f 7374 6572  reads to booster
+0000dac0: 2074 6865 2072 756e 6e69 6e67 2073 7065   the running spe
+0000dad0: 6564 2063 616e 2062 6520 7365 7420 6279  ed can be set by
+0000dae0: 2074 6865 206f 7074 696f 6e20 602d 6e60   the option `-n`
+0000daf0: 206f 7220 602d 2d6e 7468 7265 6164 7360   or `--nthreads`
+0000db00: 2077 6869 6368 2069 7320 7375 6767 6573   which is sugges
+0000db10: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
+0000db20: 2028 4e2d 3129 4e2f 3220 7768 6572 6520   (N-1)N/2 where 
+0000db30: 4e20 6973 2074 6865 206e 756d 6265 7220  N is the number 
+0000db40: 6f66 2063 6473 2066 696c 6573 2074 6f20  of cds files to 
+0000db50: 6163 6869 6576 6520 7468 6520 6869 6768  achieve the high
+0000db60: 6573 7420 6566 6669 6369 656e 6379 2e20  est efficiency. 
+0000db70: 5468 6520 6f70 7469 6f6e 2060 2d2d 6f67  The option `--og
+0000db80: 666f 726d 6174 6020 6361 6e20 6265 2073  format` can be s
+0000db90: 6574 2074 6f20 6164 6420 696e 6465 7820  et to add index 
+0000dba0: 2866 6f72 2069 6e73 7461 6e63 6520 4746  (for instance GF
+0000dbb0: 3030 3030 3030 3031 2920 746f 2074 6865  00000001) to the
+0000dbc0: 206f 7574 7075 7420 5242 4820 6765 6e65   output RBH gene
+0000dbd0: 2066 616d 696c 6965 7320 7768 6963 6820   families which 
+0000dbe0: 6361 6e20 6265 2066 7572 7468 6572 2075  can be further u
+0000dbf0: 7365 6420 696e 2074 6865 202a 4b2a 3c73  sed in the *K*<s
+0000dc00: 7562 3e53 3c2f 7375 623e 2063 616c 6375  ub>S</sub> calcu
+0000dc10: 6c61 7469 6f6e 2062 7920 6077 6764 206b  lation by `wgd k
+0000dc20: 7364 602e 0a0a 2a2a 5468 6520 7375 6767  sd`...**The sugg
+0000dc30: 6573 7465 6420 636f 6d6d 616e 6420 746f  ested command to
+0000dc40: 2073 7461 7274 2077 6974 6820 6973 2061   start with is a
+0000dc50: 6c73 6f20 756e 6465 7220 7468 6520 6465  lso under the de
+0000dc60: 6661 756c 7420 7365 7474 696e 6720 7769  fault setting wi
+0000dc70: 7468 2074 6865 2063 6f6d 6d61 6e64 2073  th the command s
+0000dc80: 686f 776e 2062 656c 6f77 2a2a 0a0a 6060  hown below**..``
+0000dc90: 600a 7767 6420 646d 6420 7365 7175 656e  `.wgd dmd sequen
+0000dca0: 6365 3120 7365 7175 656e 6365 320a 6060  ce1 sequence2.``
+0000dcb0: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
+0000dcc0: 7469 6f6e 206f 6620 6c6f 6361 6c20 4d52  tion of local MR
+0000dcd0: 4248 732a 2a0a 6060 600a 7767 6420 646d  BHs**.```.wgd dm
+0000dce0: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
+0000dcf0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000dd00: 2d66 2073 6571 7565 6e63 6531 202d 6520  -f sequence1 -e 
+0000dd10: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+0000dd20: 7020 3520 282d 6e6e 2920 282d 6e20 3429  p 5 (-nn) (-n 4)
+0000dd30: 2028 2d6b 6629 2028 2d6b 6429 2028 2d63   (-kf) (-kd) (-c
+0000dd40: 2030 2e39 2920 282d 2d74 6f5f 7374 6f70   0.9) (--to_stop
+0000dd50: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
+0000dd60: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
+0000dd70: 6e67 5f74 6d70 290a 6060 600a 0a2a 2a54  ng_tmp).```..**T
+0000dd80: 6865 2064 6973 7469 6e63 7469 6f6e 2062  he distinction b
+0000dd90: 6574 7765 656e 206c 6f63 616c 2061 6e64  etween local and
+0000dda0: 2067 6c6f 6261 6c20 2868 6572 6575 6e64   global (hereund
+0000ddb0: 6572 2920 4d52 4248 7320 6973 2074 6861  er) MRBHs is tha
+0000ddc0: 7420 6c6f 6361 6c20 4d52 4248 7320 6172  t local MRBHs ar
+0000ddd0: 6520 7468 6520 7265 7375 6c74 7320 6f66  e the results of
+0000dde0: 206d 6572 6765 6420 5242 4873 206f 6e20   merged RBHs on 
+0000ddf0: 6120 6a6f 696e 7420 666f 6361 6c20 7370  a joint focal sp
+0000de00: 6563 6965 732c 2066 6f72 2069 6e73 7461  ecies, for insta
+0000de10: 6e63 6520 696e 2061 2074 6872 6565 2073  nce in a three s
+0000de20: 7065 6369 6573 2073 7973 7465 6d20 2841  pecies system (A
+0000de30: 2c28 422c 4329 292c 2074 6865 206c 6f63  ,(B,C)), the loc
+0000de40: 616c 204d 5242 4873 206f 6620 4120 6f6e  al MRBHs of A on
+0000de50: 6c79 2072 6571 7569 7265 2074 6865 2063  ly require the c
+0000de60: 616c 6375 6c61 7469 6f6e 206f 6620 5242  alculation of RB
+0000de70: 4873 2062 6574 7765 656e 2041 2061 6e64  Hs between A and
+0000de80: 2043 2028 6465 6e6f 7465 6420 6173 2041   C (denoted as A
+0000de90: 4329 2061 6e64 2041 4220 616e 6420 7468  C) and AB and th
+0000dea0: 656e 2074 6865 206d 6572 6769 6e67 206f  en the merging o
+0000deb0: 6620 4142 2061 6e64 2041 4320 6174 2074  f AB and AC at t
+0000dec0: 6865 2061 7869 7320 6f66 2041 2c20 7768  he axis of A, wh
+0000ded0: 696c 6520 7468 6520 676c 6f61 626c 204d  ile the gloabl M
+0000dee0: 5242 4873 2061 7265 2069 6e64 6570 656e  RBHs are indepen
+0000def0: 6465 6e74 206f 6620 666f 6361 6c20 7370  dent of focal sp
+0000df00: 6563 6965 7320 696e 2074 6861 7420 6974  ecies in that it
+0000df10: 206a 7573 7420 6361 6c6c 7320 7468 6520   just calls the 
+0000df20: 6361 6c63 756c 6174 696f 6e73 206f 6620  calculations of 
+0000df30: 616c 6c20 706f 7373 6962 6c65 2073 7065  all possible spe
+0000df40: 6369 6573 2070 6169 7220 286e 6f74 2073  cies pair (not s
+0000df50: 656c 6620 746f 2073 656c 6620 7370 6563  elf to self spec
+0000df60: 6965 7320 7061 6972 292c 2073 7563 6820  ies pair), such 
+0000df70: 7468 6174 2041 422c 2041 432c 2061 6e64  that AB, AC, and
+0000df80: 2042 4320 6172 6520 616c 6c20 746f 2062   BC are all to b
+0000df90: 6520 6361 6c63 756c 6174 6564 2061 6e64  e calculated and
+0000dfa0: 206d 6572 6765 642e 2a2a 0a0a 5477 6f20   merged.**..Two 
+0000dfb0: 7479 7065 7320 6f66 204d 5242 4873 2061  types of MRBHs a
+0000dfc0: 7320 696e 7465 7072 6574 6174 6564 2061  s intepretated a
+0000dfd0: 626f 7665 2063 616e 2062 6520 6465 6c69  bove can be deli
+0000dfe0: 6e65 6174 6564 2062 7920 6077 6764 2064  neated by `wgd d
+0000dff0: 6d64 602c 2074 6865 206c 6f63 616c 204d  md`, the local M
+0000e000: 5242 4873 2061 6e64 2074 6865 2067 6c6f  RBHs and the glo
+0000e010: 6261 6c20 4d52 4248 732e 2054 6865 206c  bal MRBHs. The l
+0000e020: 6f63 616c 204d 5242 4873 2061 7265 2063  ocal MRBHs are c
+0000e030: 6f6e 7374 7275 6374 6564 2062 7920 6d65  onstructed by me
+0000e040: 7267 696e 6720 616c 6c20 7468 6520 7265  rging all the re
+0000e050: 6c65 7661 6e74 2052 4248 7320 6f6e 6c79  levant RBHs only
+0000e060: 2077 6974 6820 7468 6520 666f 6361 6c20   with the focal 
+0000e070: 7370 6563 6965 732c 2077 6869 6368 2069  species, which i
+0000e080: 7320 7365 7420 6279 2060 2d66 6020 6f72  s set by `-f` or
+0000e090: 2060 2d2d 666f 6375 7360 2e20 5468 6520   `--focus`. The 
+0000e0a0: 7061 7261 6d65 7465 7220 602d 6b66 6020  parameter `-kf` 
+0000e0b0: 6f72 2060 2d2d 6b65 6570 6661 7374 6160  or `--keepfasta`
+0000e0c0: 2063 616e 2062 6520 7365 7420 746f 2072   can be set to r
+0000e0d0: 6574 6169 6e20 7468 6520 7365 7175 656e  etain the sequen
+0000e0e0: 6365 2069 6e66 6f72 6d61 7469 6f6e 206f  ce information o
+0000e0f0: 6620 6561 6368 204d 5242 482e 2054 6865  f each MRBH. The
+0000e100: 2070 6172 616d 6574 6572 2060 2d6b 6460   parameter `-kd`
+0000e110: 206f 7220 602d 2d6b 6565 7064 7570 6c69   or `--keepdupli
+0000e120: 6361 7465 7360 2064 6574 6572 6d69 6e65  cates` determine
+0000e130: 7320 7768 6574 6865 7220 7468 6520 7361  s whether the sa
+0000e140: 6d65 2067 656e 6573 2063 616e 2061 7070  me genes can app
+0000e150: 6561 7220 696e 2064 6966 6665 7265 6e74  ear in different
+0000e160: 206c 6f63 616c 204d 5242 4873 2e20 4e6f   local MRBHs. No
+0000e170: 726d 616c 6c79 2074 6865 7265 2077 696c  rmally there wil
+0000e180: 6c20 6265 206e 6f20 6475 706c 6963 6174  l be no duplicat
+0000e190: 6573 2069 6e20 7468 6520 6c6f 6361 6c20  es in the local 
+0000e1a0: 4d52 4248 7320 6275 7420 6966 2075 7365  MRBHs but if use
+0000e1b0: 7273 2073 6574 2074 6865 2060 2d63 6020  rs set the `-c` 
+0000e1c0: 6173 2030 2e39 2028 6f72 2061 6e79 2076  as 0.9 (or any v
+0000e1d0: 616c 7565 2073 6d61 6c6c 6572 2074 6861  alue smaller tha
+0000e1e0: 6e20 3129 2c20 6974 2773 206c 696b 656c  n 1), it's likel
+0000e1f0: 7920 7468 6174 2074 6865 2073 616d 6520  y that the same 
+0000e200: 6765 6e65 2077 696c 6c20 6861 7665 2063  gene will have c
+0000e210: 6861 6e63 6520 746f 2061 7070 6561 7220  hance to appear 
+0000e220: 6d75 6c74 6970 6c65 2074 696d 6573 2069  multiple times i
+0000e230: 6e20 6469 6666 6572 656e 7420 6c6f 6361  n different loca
+0000e240: 6c20 4d52 4248 732e 2054 6861 7420 6973  l MRBHs. That is
+0000e250: 2074 6f20 7361 792c 2074 6865 2070 6172   to say, the par
+0000e260: 616d 6574 6572 2060 2d6b 6460 2069 7320  ameter `-kd` is 
+0000e270: 6d65 616e 696e 6766 756c 206f 6e6c 7920  meaningful only 
+0000e280: 7768 656e 2069 7427 7320 7365 7420 746f  when it's set to
+0000e290: 6765 7468 6572 2077 6974 6820 7468 6520  gether with the 
+0000e2a0: 7061 7261 6d65 7465 7220 602d 6360 2e20  parameter `-c`. 
+0000e2b0: 5468 6520 6e75 6d62 6572 206f 6620 7061  The number of pa
+0000e2c0: 7261 6c6c 656c 2074 6872 6561 6473 2069  rallel threads i
+0000e2d0: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+0000e2e0: 6520 7365 7420 6173 2074 6865 206e 756d  e set as the num
+0000e2f0: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
+0000e300: 206d 696e 7573 2031 2e0a 0a2a 2a41 2073   minus 1...**A s
+0000e310: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+0000e320: 6720 7275 6e20 6973 2075 6e64 6572 2074  g run is under t
+0000e330: 6865 2064 6566 6175 6c74 2070 6172 616d  he default param
+0000e340: 6574 6572 2077 6974 6820 7468 6520 636f  eter with the co
+0000e350: 6d6d 616e 6420 7368 6f77 6e20 6173 2062  mmand shown as b
+0000e360: 656c 6f77 2e2a 2a0a 0a60 6060 0a77 6764  elow.**..```.wgd
+0000e370: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+0000e380: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000e390: 6533 202d 6620 7365 7175 656e 6365 310a  e3 -f sequence1.
+0000e3a0: 6060 600a 0a2a 2a54 6865 2064 656c 696e  ```..**The delin
+0000e3b0: 6561 7469 6f6e 206f 6620 676c 6f62 616c  eation of global
+0000e3c0: 204d 5242 4873 2a2a 0a60 6060 0a77 6764   MRBHs**.```.wgd
+0000e3d0: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+0000e3e0: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000e3f0: 6533 202d 676d 202d 6520 3165 2d31 3020  e3 -gm -e 1e-10 
+0000e400: 2d62 7320 3130 3020 2d6e 7020 3520 282d  -bs 100 -np 5 (-
+0000e410: 6e6e 2920 282d 6e20 3429 2028 2d6b 6629  nn) (-n 4) (-kf)
+0000e420: 2028 2d6b 6429 2028 2d63 2030 2e39 2920   (-kd) (-c 0.9) 
+0000e430: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
+0000e440: 6473 2920 282d 6f20 7767 645f 646d 6429  ds) (-o wgd_dmd)
+0000e450: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
+0000e460: 290a 6060 600a 0a54 6865 2067 6c6f 6261  ).```..The globa
+0000e470: 6c20 4d52 4248 7320 6973 2063 6f6e 7374  l MRBHs is const
+0000e480: 7275 6374 6564 2062 7920 6578 6861 7573  ructed by exhaus
+0000e490: 7469 7665 6c79 206d 6572 6769 6e67 2061  tively merging a
+0000e4a0: 6c6c 2074 6865 2070 6f73 7369 626c 6520  ll the possible 
+0000e4b0: 7061 6972 2d77 6973 6520 5242 4873 2065  pair-wise RBHs e
+0000e4c0: 7863 6570 7420 666f 7220 7175 6572 7969  xcept for queryi
+0000e4d0: 6e67 2074 6865 2073 6571 7565 6e63 6520  ng the sequence 
+0000e4e0: 6974 7365 6c66 2c20 7768 6963 6820 6361  itself, which ca
+0000e4f0: 6e20 6265 2069 6e69 7469 6174 6564 2062  n be initiated b
+0000e500: 7920 6164 6420 7468 6520 666c 6167 2060  y add the flag `
+0000e510: 2d67 6d60 206f 7220 602d 2d67 6c6f 6261  -gm` or `--globa
+0000e520: 6c6d 7262 6860 2e20 5468 6520 7265 7374  lmrbh`. The rest
+0000e530: 206f 6620 7265 6c65 7661 6e74 2070 6172   of relevant par
+0000e540: 616d 6574 6572 7320 7374 6179 7320 7468  ameters stays th
+0000e550: 6520 7361 6d65 2061 7320 7468 6520 6c6f  e same as the lo
+0000e560: 6361 6c20 4d52 4248 732e 2054 6865 206e  cal MRBHs. The n
+0000e570: 756d 6265 7220 6f66 2070 6172 616c 6c65  umber of paralle
+0000e580: 6c20 7468 7265 6164 7320 6973 2073 7567  l threads is sug
+0000e590: 6765 7374 6564 2074 6f20 6265 2073 6574  gested to be set
+0000e5a0: 2061 7320 284e 2d31 294e 2f32 2074 6f6f   as (N-1)N/2 too
+0000e5b0: 2077 6865 7265 204e 2069 7320 7468 6520   where N is the 
+0000e5c0: 6e75 6d62 6572 206f 6620 6364 7320 6669  number of cds fi
+0000e5d0: 6c65 7320 746f 2061 6368 6965 7665 2074  les to achieve t
+0000e5e0: 6865 2068 6967 6865 7374 2065 6666 6963  he highest effic
+0000e5f0: 6965 6e63 792e 0a0a 2a2a 4120 7375 6767  iency...**A sugg
+0000e600: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000e610: 756e 2069 7320 756e 6465 7220 7468 6520  un is under the 
+0000e620: 6465 6661 756c 7420 7061 7261 6d65 7465  default paramete
+0000e630: 7220 7769 7468 2074 6865 2063 6f6d 6d61  r with the comma
+0000e640: 6e64 2073 686f 776e 2061 7320 6265 6c6f  nd shown as belo
+0000e650: 772e 2a2a 0a0a 6060 600a 7767 6420 646d  w.**..```.wgd dm
+0000e660: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
+0000e670: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000e680: 2d67 6d0a 6060 600a 0a2a 2a54 6865 2064  -gm.```..**The d
+0000e690: 656c 696e 6561 7469 6f6e 206f 6620 6f72  elineation of or
+0000e6a0: 7468 6f67 726f 7570 732a 2a0a 6060 600a  thogroups**.```.
+0000e6b0: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
+0000e6c0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
+0000e6d0: 656e 6365 3320 2d6f 6920 2d6f 6f20 2d65  ence3 -oi -oo -e
+0000e6e0: 2031 652d 3130 202d 6273 2031 3030 202d   1e-10 -bs 100 -
+0000e6f0: 6e70 2035 2028 2d6e 6e29 2028 2d63 6329  np 5 (-nn) (-cc)
+0000e700: 2028 2d74 6529 2028 2d6d 6320 302e 3829   (-te) (-mc 0.8)
+0000e710: 2028 2d67 6e29 2028 2d74 7265 6520 2766   (-gn) (-tree 'f
+0000e720: 6173 7474 7265 6527 2920 282d 7473 2027  asttree') (-ts '
+0000e730: 2d66 6173 7465 7374 2729 2028 2d6e 2034  -fastest') (-n 4
+0000e740: 2920 282d 2d74 6f5f 7374 6f70 2920 282d  ) (--to_stop) (-
+0000e750: 2d63 6473 2920 282d 6f20 7767 645f 646d  -cds) (-o wgd_dm
+0000e760: 6429 2028 2d74 2077 6f72 6b69 6e67 5f74  d) (-t working_t
+0000e770: 6d70 290a 6060 600a 0a49 6e20 6077 6764  mp).```..In `wgd
+0000e780: 2076 3260 2c20 7765 2061 6c73 6f20 696d   v2`, we also im
+0000e790: 706c 656d 656e 7465 6420 616e 2061 6c67  plemented an alg
+0000e7a0: 6f72 6974 686d 206f 6620 6465 6c69 6e65  orithm of deline
+0000e7b0: 6174 696e 6720 6f72 7468 6f67 726f 7570  ating orthogroup
+0000e7c0: 732c 2077 6869 6368 2063 616e 2062 6520  s, which can be 
+0000e7d0: 696e 6974 6961 7465 6420 7769 7468 2074  initiated with t
+0000e7e0: 6865 2070 6172 616d 6574 6572 2060 2d6f  he parameter `-o
+0000e7f0: 6960 206f 7220 602d 2d6f 7274 686f 696e  i` or `--orthoin
+0000e800: 6665 7260 2e20 5477 6f20 7761 7973 206f  fer`. Two ways o
+0000e810: 6620 6465 6c69 6e65 6174 696f 6e20 6361  f delineation ca
+0000e820: 6e20 6265 2063 686f 7365 6e2c 2074 6865  n be chosen, the
+0000e830: 2063 6f6e 6361 7465 6e61 7469 6f6e 2077   concatenation w
+0000e840: 6179 2028 7365 7420 6279 2074 6865 2070  ay (set by the p
+0000e850: 6172 616d 6574 6572 2060 2d63 6360 206f  arameter `-cc` o
+0000e860: 7220 602d 2d63 6f6e 6361 7460 2920 6f72  r `--concat`) or
+0000e870: 2074 6865 206e 6f6e 2d63 6f6e 6361 7465   the non-concate
+0000e880: 6e61 7469 6f6e 2028 6465 6661 756c 7429  nation (default)
+0000e890: 2077 6179 2e20 496e 2062 7269 6566 2c20   way. In brief, 
+0000e8a0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+0000e8b0: 6e20 7761 7920 6f66 2064 656c 696e 6561  n way of delinea
+0000e8c0: 7469 6e67 206f 7274 686f 6772 6f75 7073  ting orthogroups
+0000e8d0: 2073 7461 7274 7320 7769 7468 2063 6f6e   starts with con
+0000e8e0: 6361 7465 6e61 7469 6e67 2061 6c6c 2074  catenating all t
+0000e8f0: 6865 2073 6571 7565 6e63 6573 2069 6e74  he sequences int
+0000e900: 6f20 6120 7369 6e67 6c65 2073 6571 7565  o a single seque
+0000e910: 6e63 6520 6669 6c65 2061 6e64 2074 6865  nce file and the
+0000e920: 6e20 696e 6665 7272 696e 6720 7468 6520  n inferring the 
+0000e930: 7768 6f6c 6520 7061 7261 6e6f 6d65 206f  whole paranome o
+0000e940: 6620 7468 6973 2073 696e 676c 6520 7365  f this single se
+0000e950: 7175 656e 6365 2066 696c 6520 7769 7468  quence file with
+0000e960: 2074 6865 2063 6c75 7374 6572 696e 6720   the clustering 
+0000e970: 7265 7375 6c74 7320 6d61 7070 6564 2062  results mapped b
+0000e980: 6163 6b20 746f 2074 6865 2062 656c 6f6e  ack to the belon
+0000e990: 6769 6e67 2073 7065 6369 6573 2e20 5768  ging species. Wh
+0000e9a0: 696c 6520 7468 6520 6e6f 6e2d 636f 6e63  ile the non-conc
+0000e9b0: 6174 656e 6174 696f 6e20 7761 7920 7374  atenation way st
+0000e9c0: 6172 7473 2077 6974 6820 7265 7370 6563  arts with respec
+0000e9d0: 7469 7665 2070 6169 722d 7769 7365 2064  tive pair-wise d
+0000e9e0: 6961 6d6f 6e64 2073 6561 7263 6820 2869  iamond search (i
+0000e9f0: 6e63 6c75 6469 6e67 2071 7565 7279 696e  ncluding queryin
+0000ea00: 6720 7468 6520 7361 6d65 2073 6571 7565  g the same seque
+0000ea10: 6e63 6520 6974 7365 6c66 2920 616e 6420  nce itself) and 
+0000ea20: 7468 656e 2061 6c6c 2074 6865 2073 6571  then all the seq
+0000ea30: 7565 6e63 6520 7369 6d69 6c61 7269 7479  uence similarity
+0000ea40: 2074 6162 6c65 7320 7769 6c6c 2062 6520   tables will be 
+0000ea50: 636f 6e63 6174 656e 6174 6564 2061 6e64  concatenated and
+0000ea60: 2063 6c75 7374 6572 6564 2069 6e74 6f20   clustered into 
+0000ea70: 6f72 7468 6f67 726f 7570 732e 2053 6f6d  orthogroups. Som
+0000ea80: 6520 6f74 6865 7220 706f 7373 6962 6c79  e other possibly
+0000ea90: 2075 7365 6675 6c20 706f 7374 2d63 6c75   useful post-clu
+0000eaa0: 7374 6572 696e 6720 6675 6e63 7469 6f6e  stering function
+0000eab0: 7320 6361 6e20 6265 2069 6e69 7469 6174  s can be initiat
+0000eac0: 6564 2c20 696e 636c 7564 696e 6720 7468  ed, including th
+0000ead0: 6520 7061 7261 6d65 7465 7220 602d 7465  e parameter `-te
+0000eae0: 6020 6f72 2060 2d2d 7465 7374 736f 6760  ` or `--testsog`
+0000eaf0: 2c20 7768 6963 6820 6361 6e20 6265 2073  , which can be s
+0000eb00: 6574 2074 6f20 7374 6172 7420 7468 6520  et to start the 
+0000eb10: 756e 6269 6173 6564 2074 6573 7420 6f66  unbiased test of
+0000eb20: 2073 696e 676c 652d 636f 7079 2067 656e   single-copy gen
+0000eb30: 6520 6661 6d69 6c69 6573 2028 6e6f 7465  e families (note
+0000eb40: 2074 6861 7420 7468 6973 2066 756e 6374   that this funct
+0000eb50: 696f 6e20 6e65 6564 7320 6068 6d6d 6572  ion needs `hmmer
+0000eb60: 6020 2876 332e 3162 3229 2074 6f20 6265  ` (v3.1b2) to be
+0000eb70: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
+0000eb80: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
+0000eb90: 7468 292c 2074 6865 2070 6172 616d 6574  th), the paramet
+0000eba0: 6572 2060 2d6d 6360 206f 7220 602d 2d6d  er `-mc` or `--m
+0000ebb0: 736f 6763 7574 602c 2072 616e 6769 6e67  sogcut`, ranging
+0000ebc0: 2062 6574 7765 656e 2030 2074 6f20 312c   between 0 to 1,
+0000ebd0: 2077 6869 6368 2063 616e 2062 6520 7365   which can be se
+0000ebe0: 7420 746f 2073 6561 7263 6820 7468 6520  t to search the 
+0000ebf0: 736f 2d63 616c 6c65 6420 6d6f 7374 6c79  so-called mostly
+0000ec00: 2073 696e 676c 652d 636f 7079 2066 616d   single-copy fam
+0000ec10: 696c 7920 7768 6963 6820 6861 7320 6869  ily which has hi
+0000ec20: 6768 6572 2074 6861 6e20 6365 7274 6169  gher than certai
+0000ec30: 6e20 6375 742d 6f66 6620 7065 7263 656e  n cut-off percen
+0000ec40: 7461 6765 206f 6620 7370 6563 6965 7320  tage of species 
+0000ec50: 636f 7665 7261 6765 2c20 7468 6520 7061  coverage, the pa
+0000ec60: 7261 6d65 7465 7220 602d 676e 6020 6f72  rameter `-gn` or
+0000ec70: 2060 2d2d 6765 746e 736f 6760 2c20 7768   `--getnsog`, wh
+0000ec80: 6963 6820 6361 6e20 6265 2073 6574 2074  ich can be set t
+0000ec90: 6f20 7365 6172 6368 2066 6f72 206e 6573  o search for nes
+0000eca0: 7465 6420 7369 6e67 6c65 2d63 6f70 7920  ted single-copy 
+0000ecb0: 6765 6e65 2066 616d 696c 6965 7320 284e  gene families (N
+0000ecc0: 534f 4773 2920 7768 6963 6820 6973 206f  SOGs) which is o
+0000ecd0: 7269 6769 6e61 6c6c 7920 6d75 6c74 6979  riginally multiy
+0000ece0: 2d63 6f70 7920 6275 7420 6861 7320 6120  -copy but has a 
+0000ecf0: 286d 6f73 746c 7929 2073 696e 676c 652d  (mostly) single-
+0000ed00: 636f 7079 2062 7261 6e63 6820 2877 6869  copy branch (whi
+0000ed10: 6368 2072 6571 7569 7265 7320 7468 6520  ch requires the 
+0000ed20: 6368 6f73 656e 2074 7265 652d 696e 6665  chosen tree-infe
+0000ed30: 7265 6e63 6520 7072 6f67 7261 6d20 7365  rence program se
+0000ed40: 7420 6279 2060 2d74 7265 6560 206f 7220  t by `-tree` or 
+0000ed50: 602d 2d74 7265 655f 6d65 7468 6f64 6020  `--tree_method` 
+0000ed60: 746f 2062 6520 7072 652d 696e 7374 616c  to be pre-instal
+0000ed70: 6c65 6420 696e 2074 6865 2065 6e76 6972  led in the envir
+0000ed80: 6f6e 6d65 6e74 2070 6174 6820 7769 7468  onment path with
+0000ed90: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
+0000eda0: 7365 7474 696e 6720 666f 7220 6765 6e65  setting for gene
+0000edb0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0000edc0: 636f 6e74 726f 6c6c 6564 2062 7920 602d  controlled by `-
+0000edd0: 7473 6020 6f72 2060 2d2d 7472 6565 7365  ts` or `--treese
+0000ede0: 7460 292e 2054 6865 2070 726f 6772 616d  t`). The program
+0000edf0: 2060 7767 6420 646d 6460 2077 6f75 6c64   `wgd dmd` would
+0000ee00: 2073 7469 6c6c 2063 6f6e 6475 6374 2074   still conduct t
+0000ee10: 6865 2052 4248 7320 6361 6c63 756c 6174  he RBHs calculat
+0000ee20: 696f 6e20 756e 6c65 7373 2074 6865 2070  ion unless the p
+0000ee30: 6172 616d 6574 6572 2060 2d6f 6f60 206f  arameter `-oo` o
+0000ee40: 7220 602d 2d6f 6e6c 796f 7274 686f 6020  r `--onlyortho` 
+0000ee50: 7761 7320 7365 742e 2049 6620 6f6e 6520  was set. If one 
+0000ee60: 6f6e 6c79 2077 616e 7473 2074 6f20 696e  only wants to in
+0000ee70: 6665 7220 7468 6520 6f72 7468 6f67 726f  fer the orthogro
+0000ee80: 7570 732c 2069 7427 7320 7375 6767 6573  ups, it's sugges
+0000ee90: 7465 6420 746f 2061 6464 2074 6865 2066  ted to add the f
+0000eea0: 6c61 6720 602d 6f6f 6020 746f 206a 7573  lag `-oo` to jus
+0000eeb0: 7420 696d 706c 656d 656e 7420 7468 6520  t implement the 
+0000eec0: 6f72 7468 6f67 726f 7570 7320 6465 6c69  orthogroups deli
+0000eed0: 6e65 6174 696f 6e20 616e 616c 7973 6973  neation analysis
+0000eee0: 2e20 5468 6520 6e75 6d62 6572 206f 6620  . The number of 
+0000eef0: 7061 7261 6c6c 656c 2074 6872 6561 6473  parallel threads
+0000ef00: 2069 7320 7375 6767 6573 7465 6420 746f   is suggested to
+0000ef10: 2062 6520 7365 7420 6173 2028 4e2b 3129   be set as (N+1)
+0000ef20: 4e2f 3220 7768 6572 6520 4e20 6973 2074  N/2 where N is t
+0000ef30: 6865 206e 756d 6265 7220 6f66 2063 6473  he number of cds
+0000ef40: 2066 696c 6573 2074 6f20 6163 6869 6576   files to achiev
+0000ef50: 6520 7468 6520 6869 6768 6573 7420 6566  e the highest ef
+0000ef60: 6669 6369 656e 6379 2073 696e 6365 2074  ficiency since t
+0000ef70: 6865 2073 656c 662d 636f 6d70 6172 6973  he self-comparis
+0000ef80: 6f6e 2069 7320 616c 736f 2069 6e63 6c75  on is also inclu
+0000ef90: 6465 642e 0a0a 2a2a 5468 6520 6465 6661  ded...**The defa
+0000efa0: 756c 7420 7365 7474 696e 6720 6f66 2070  ult setting of p
+0000efb0: 6172 616d 6574 6572 7320 6973 2061 2072  arameters is a r
+0000efc0: 6561 736f 6e61 626c 6520 7374 6172 7469  easonable starti
+0000efd0: 6e67 2070 6f69 6e74 2077 6974 6820 7468  ng point with th
+0000efe0: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
+0000eff0: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2064  ow.**..```.wgd d
+0000f000: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
+0000f010: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000f020: 202d 6f69 202d 6f6f 0a60 6060 0a0a 2a2a   -oi -oo.```..**
+0000f030: 5468 6520 636f 6c6c 696e 6561 7220 636f  The collinear co
+0000f040: 616c 6573 6365 6e63 6520 696e 6665 7265  alescence infere
+0000f050: 6e63 6520 6f66 2070 6879 6c6f 6765 6e79  nce of phylogeny
+0000f060: 2a2a 0a60 6060 0a77 6764 2064 6d64 2073  **.```.wgd dmd s
+0000f070: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
+0000f080: 6532 2073 6571 7565 6e63 6533 202d 6170  e2 sequence3 -ap
+0000f090: 2061 7064 6174 6120 2d73 6d20 736d 6461   apdata -sm smda
+0000f0a0: 7461 202d 6c65 206c 6564 6174 6120 2d67  ta -le ledata -g
+0000f0b0: 7420 6774 6461 7461 202d 636f 6320 282d  t gtdata -coc (-
+0000f0c0: 7472 6565 2027 6661 7374 7472 6565 2729  tree 'fasttree')
+0000f0d0: 2028 2d74 7320 272d 6661 7374 6573 7427   (-ts '-fastest'
+0000f0e0: 2920 282d 6e20 3429 2028 2d2d 746f 5f73  ) (-n 4) (--to_s
+0000f0f0: 746f 7029 2028 2d2d 6364 7329 2028 2d6f  top) (--cds) (-o
+0000f100: 2077 6764 5f64 6d64 2920 282d 7420 776f   wgd_dmd) (-t wo
+0000f110: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
+0000f120: 4120 6e6f 7665 6c20 7068 796c 6f67 656e  A novel phylogen
+0000f130: 6574 6963 2069 6e66 6572 656e 6365 206d  etic inference m
+0000f140: 6574 686f 6420 6e61 6d65 6420 2263 6f6c  ethod named "col
+0000f150: 6c69 6e65 6172 2063 6f61 6c65 7363 656e  linear coalescen
+0000f160: 6365 2069 6e66 6572 656e 6365 2220 6973  ce inference" is
+0000f170: 2061 6c73 6f20 696d 706c 656d 656e 7465   also implemente
+0000f180: 6420 696e 2060 7767 6420 7632 602e 2046  d in `wgd v2`. F
+0000f190: 6f72 2074 6869 7320 616e 616c 7973 6973  or this analysis
+0000f1a0: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000f1b0: 7072 6f76 6964 6520 7468 6520 616e 6368  provide the anch
+0000f1c0: 6f72 2070 6f69 6e74 7320 6669 6c65 2062  or points file b
+0000f1d0: 7920 602d 6170 6020 6f72 2060 2d2d 616e  y `-ap` or `--an
+0000f1e0: 6368 6f72 706f 696e 7473 602c 2074 6865  chorpoints`, the
+0000f1f0: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
+0000f200: 6e74 7320 6669 6c65 2062 7920 602d 736d  nts file by `-sm
+0000f210: 6020 6f72 2060 2d2d 7365 676d 656e 7473  ` or `--segments
+0000f220: 602c 2074 6865 206c 6973 7473 6567 6d65  `, the listsegme
+0000f230: 6e74 7320 6669 6c65 2062 7920 602d 6c65  nts file by `-le
+0000f240: 6020 6f72 2060 2d2d 6c69 7374 656c 656d  ` or `--listelem
+0000f250: 656e 7473 602c 2061 6e64 2074 6865 2067  ents`, and the g
+0000f260: 656e 6520 7461 626c 6520 6669 6c65 2062  ene table file b
+0000f270: 7920 602d 6774 6020 6f72 2060 2d2d 6765  y `-gt` or `--ge
+0000f280: 6e65 7461 626c 6560 2c20 616c 6c20 6f66  netable`, all of
+0000f290: 2077 6869 6368 2063 616e 2062 6520 7072   which can be pr
+0000f2a0: 6f64 7563 6564 2069 6e20 7468 6520 7072  oduced in the pr
+0000f2b0: 6f67 7261 6d20 6077 6764 2073 796e 602e  ogram `wgd syn`.
+0000f2c0: 2054 6865 2070 6172 616d 6574 6572 2060   The parameter `
+0000f2d0: 2d63 6f63 6020 6f72 2060 2d2d 636f 6c6c  -coc` or `--coll
+0000f2e0: 696e 6561 7263 6f61 6c65 7363 656e 6365  inearcoalescence
+0000f2f0: 6020 6e65 6564 7320 746f 2062 6520 7365  ` needs to be se
+0000f300: 7420 746f 2073 7461 7274 2074 6869 7320  t to start this 
+0000f310: 616e 616c 7973 6973 2e20 5468 6520 7472  analysis. The tr
+0000f320: 6565 2d69 6e66 6572 656e 6365 2070 726f  ee-inference pro
+0000f330: 6772 616d 2061 6e64 2074 6865 2061 7373  gram and the ass
+0000f340: 6f63 6961 7465 6420 7061 7261 6d65 7465  ociated paramete
+0000f350: 7273 2063 616e 2062 6520 7365 7420 6a75  rs can be set ju
+0000f360: 7374 2061 7320 6162 6f76 6520 6279 2060  st as above by `
+0000f370: 2d74 7265 6560 206f 7220 602d 2d74 7265  -tree` or `--tre
+0000f380: 655f 6d65 7468 6f64 6020 616e 6420 602d  e_method` and `-
+0000f390: 7473 6020 6f72 2060 2d2d 7472 6565 7365  ts` or `--treese
+0000f3a0: 7460 2e20 506c 6561 7365 2061 6c73 6f20  t`. Please also 
+0000f3b0: 6d61 6b65 2073 7572 6520 7468 6520 6368  make sure the ch
+0000f3c0: 6f73 656e 2074 7265 652d 696e 6665 7265  osen tree-infere
+0000f3d0: 6e63 6520 7072 6f67 7261 6d20 6973 2069  nce program is i
+0000f3e0: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+0000f3f0: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+0000f400: 2e20 5468 6520 7072 6f67 7261 6d20 6061  . The program `a
+0000f410: 7374 7261 6c2d 7072 6f60 2069 7320 7265  stral-pro` is re
+0000f420: 7175 6972 6564 2074 6f20 6265 2069 6e73  quired to be ins
+0000f430: 7461 6c6c 6564 2069 6e20 7468 6520 656e  talled in the en
+0000f440: 7669 726f 6e6d 656e 7420 7061 7468 2074  vironment path t
+0000f450: 6f6f 2e20 4e6f 7465 2074 6861 7420 7468  oo. Note that th
+0000f460: 6572 6520 7368 6f75 6c64 2062 6520 6e6f  ere should be no
+0000f470: 2064 7570 6c69 6361 7465 6420 6765 6e65   duplicated gene
+0000f480: 2049 4473 2069 6e20 7468 6520 7365 7175   IDs in the sequ
+0000f490: 656e 6365 2066 696c 652e 2054 6865 2070  ence file. The p
+0000f4a0: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
+0000f4b0: 6865 7265 2061 7265 2074 6f20 6163 6365  here are to acce
+0000f4c0: 6c65 7261 7465 2074 6865 2073 6571 7565  lerate the seque
+0000f4d0: 6e63 6520 616c 6967 6e6d 656e 7420 616e  nce alignment an
+0000f4e0: 6420 6765 6e65 2074 7265 6520 696e 6665  d gene tree infe
+0000f4f0: 7265 6e63 6520 666f 7220 6561 6368 2067  rence for each g
+0000f500: 656e 6520 6661 6d69 6c79 2061 6e64 2074  ene family and t
+0000f510: 6875 7320 7375 6767 6573 7465 6420 746f  hus suggested to
+0000f520: 2062 6520 7365 7420 6173 206d 7563 6820   be set as much 
+0000f530: 6173 2074 6865 206e 756d 6265 7220 6f66  as the number of
+0000f540: 2067 656e 6520 6661 6d69 6c69 6573 2e0a   gene families..
+0000f550: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+0000f560: 7461 7274 696e 6720 7275 6e20 6f66 2074  tarting run of t
+0000f570: 6869 7320 616e 616c 7973 6973 2069 7320  his analysis is 
+0000f580: 7769 7468 2074 6865 2073 696d 706c 6520  with the simple 
+0000f590: 636f 6d6d 616e 6420 6265 6c6f 772e 2a2a  command below.**
+0000f5a0: 0a0a 6060 600a 7767 6420 646d 6420 7365  ..```.wgd dmd se
+0000f5b0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000f5c0: 3220 7365 7175 656e 6365 3320 2d61 7020  2 sequence3 -ap 
+0000f5d0: 6170 6461 7461 202d 736d 2073 6d64 6174  apdata -sm smdat
+0000f5e0: 6120 2d6c 6520 6c65 6461 7461 202d 6774  a -le ledata -gt
+0000f5f0: 2067 7464 6174 6120 2d63 6f63 0a60 6060   gtdata -coc.```
+0000f600: 0a0a 2323 2320 7767 6420 6b73 640a 0a2a  ..### wgd ksd..*
+0000f610: 2a54 6865 2063 6f6e 7374 7275 6374 696f  *The constructio
+0000f620: 6e20 6f66 2077 686f 6c65 2070 6172 616e  n of whole paran
+0000f630: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
+0000f640: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+0000f650: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206b  tion**.```.wgd k
+0000f660: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
+0000f670: 656e 6365 2028 2d6f 2077 6764 5f6b 7364  ence (-o wgd_ksd
+0000f680: 202d 7420 7767 645f 6b73 645f 746d 7020   -t wgd_ksd_tmp 
+0000f690: 2d2d 6e74 6872 6561 6473 2034 202d 2d74  --nthreads 4 --t
+0000f6a0: 6f5f 7374 6f70 202d 2d63 6473 202d 2d70  o_stop --cds --p
+0000f6b0: 6169 7277 6973 6520 2d2d 7374 7269 705f  airwise --strip_
+0000f6c0: 6761 7073 202d 2d61 6c69 676e 6572 206d  gaps --aligner m
+0000f6d0: 6166 6674 202d 2d74 7265 655f 6d65 7468  afft --tree_meth
+0000f6e0: 6f64 2066 6173 7474 7265 6520 2d2d 6e6f  od fasttree --no
+0000f6f0: 6465 5f61 7665 7261 6765 290a 6060 600a  de_average).```.
+0000f700: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
+0000f710: 6420 6b73 6460 2c20 6173 2069 6d70 6965  d ksd`, as impie
+0000f720: 6420 6279 2069 7473 206e 616d 652c 2069  d by its name, i
+0000f730: 7320 666f 7220 7468 6520 636f 6e73 7472  s for the constr
+0000f740: 7563 7469 6f6e 206f 6620 2a4b 2a3c 7375  uction of *K*<su
+0000f750: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+0000f760: 7374 7269 6275 7469 6f6e 2e20 4578 6365  stribution. Exce
+0000f770: 7074 2066 6f72 2074 6865 2061 666f 7265  pt for the afore
+0000f780: 6d65 6e74 696f 6e65 6420 7061 7261 6d65  mentioned parame
+0000f790: 7465 7273 2073 7563 6820 6173 2060 2d2d  ters such as `--
+0000f7a0: 746f 5f73 746f 7060 2061 6e64 2060 2d2d  to_stop` and `--
+0000f7b0: 6364 7360 2c20 7468 6572 6520 6172 6520  cds`, there are 
+0000f7c0: 736f 6d65 2069 6d70 6f72 7461 6e74 2070  some important p
+0000f7d0: 6172 616d 6574 6572 7320 7468 6174 2068  arameters that h
+0000f7e0: 6176 6520 6372 7563 6961 6c20 696d 7061  ave crucial impa
+0000f7f0: 6374 206f 6e20 7468 6520 2a4b 2a3c 7375  ct on the *K*<su
+0000f800: 623e 533c 2f73 7562 3e20 6573 7469 6d61  b>S</sub> estima
+0000f810: 7469 6f6e 2e20 5468 6520 6f70 7469 6f6e  tion. The option
+0000f820: 2060 2d2d 7061 6972 7769 7365 6020 6973   `--pairwise` is
+0000f830: 2061 2076 6572 7920 696d 706f 7274 616e   a very importan
+0000f840: 7420 7061 7261 6d65 7465 7220 666f 7220  t parameter for 
+0000f850: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+0000f860: 7562 3e20 6573 7469 6d61 7469 6f6e 2c20  ub> estimation, 
+0000f870: 7769 7468 2077 6869 6368 2074 6865 2060  with which the `
+0000f880: 434f 4445 4d4c 6020 7769 6c6c 2063 616c  CODEML` will cal
+0000f890: 6375 6c61 7465 2074 6865 202a 4b2a 3c73  culate the *K*<s
+0000f8a0: 7562 3e53 3c2f 7375 623e 2066 6f72 2065  ub>S</sub> for e
+0000f8b0: 6163 6820 6765 6e65 2070 6169 7220 7365  ach gene pair se
+0000f8c0: 7061 7261 7465 6c79 2062 6173 6564 206f  parately based o
+0000f8d0: 6e20 7468 6520 616c 6967 6e6d 656e 7420  n the alignment 
+0000f8e0: 6f66 206f 6e6c 7920 7468 6573 6520 7477  of only these tw
+0000f8f0: 6f20 6765 6e65 7320 696e 7374 6561 6420  o genes instead 
+0000f900: 6f66 2074 6865 2077 686f 6c65 2061 6c69  of the whole ali
+0000f910: 676e 6d65 6e74 206f 6620 7468 6520 6661  gnment of the fa
+0000f920: 6d69 6c79 2c20 7375 6368 2074 6861 7420  mily, such that 
+0000f930: 6c65 7373 2067 6170 7320 6172 6520 6578  less gaps are ex
+0000f940: 7065 6374 6564 2061 6e64 2074 6875 7320  pected and thus 
+0000f950: 7468 6520 616c 6967 6e6d 656e 7420 696e  the alignment in
+0000f960: 2074 6865 2063 6f6e 7369 6465 7261 7469   the considerati
+0000f970: 6f6e 206f 6620 6043 4f44 454d 4c60 2077  on of `CODEML` w
+0000f980: 696c 6c20 6265 206c 6f6e 6765 7220 2862  ill be longer (b
+0000f990: 6563 6175 7365 2060 434f 4445 4d4c 6020  ecause `CODEML` 
+0000f9a0: 7769 6c6c 2061 7574 6f6d 6174 656c 7920  will automately 
+0000f9b0: 736b 6970 2065 7665 7279 2063 6f6c 756d  skip every colum
+0000f9c0: 6e20 7769 7468 2067 6170 2c20 7265 6761  n with gap, rega
+0000f9d0: 7264 6c65 7373 206f 6620 7768 6574 6865  rdless of whethe
+0000f9e0: 7220 6974 2773 2061 6e20 6f76 6572 616c  r it's an overal
+0000f9f0: 6c20 6f72 2070 6172 7469 616c 2067 6170  l or partial gap
+0000fa00: 292c 2077 6974 686f 7574 2077 6869 6368  ), without which
+0000fa10: 2074 6865 2060 434f 4445 4d4c 6020 7769   the `CODEML` wi
+0000fa20: 6c6c 2063 616c 6375 6c61 7465 2074 6865  ll calculate the
+0000fa30: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000fa40: 2062 6173 6564 206f 6e20 7468 6520 7768   based on the wh
+0000fa50: 6f6c 6520 616c 6967 6e6d 656e 7420 6f66  ole alignment of
+0000fa60: 2074 6865 2066 616d 696c 792c 2077 6869   the family, whi
+0000fa70: 6368 206d 6967 6874 2068 6176 6520 6e6f  ch might have no
+0000fa80: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000fa90: 2072 6573 756c 7420 6174 2061 6c6c 2069   result at all i
+0000faa0: 6620 7468 6520 7374 7269 7070 6564 2061  f the stripped a
+0000fab0: 6c69 676e 6d65 6e74 206c 656e 6774 6820  lignment length 
+0000fac0: 2872 656d 6f76 696e 6720 616c 6c20 6761  (removing all ga
+0000fad0: 702d 636f 6e74 6169 6e69 6e67 2063 6f6c  p-containing col
+0000fae0: 756d 6e73 2920 7761 7320 7a65 726f 2c20  umns) was zero, 
+0000faf0: 6120 6361 7573 6520 6f66 2064 6966 6665  a cause of diffe
+0000fb00: 7265 6e74 206e 756d 6265 7220 6f66 202a  rent number of *
+0000fb10: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
+0000fb20: 7374 696d 6174 6573 2062 6574 7765 656e  stimates between
+0000fb30: 2022 7061 6972 7769 7365 206d 6f64 6522   "pairwise mode"
+0000fb40: 2061 6e64 2022 6e6f 6e2d 7061 6972 7769   and "non-pairwi
+0000fb50: 7365 206d 6f64 6522 2e20 4974 2773 2064  se mode". It's d
+0000fb60: 6966 6669 6375 6c74 2074 6f20 7361 7920  ifficult to say 
+0000fb70: 7768 6963 6820 6d6f 6465 2069 7320 6d6f  which mode is mo
+0000fb80: 7265 2069 6465 616c 2c20 616c 7468 6f75  re ideal, althou
+0000fb90: 6768 2074 6865 2022 6e6f 6e2d 7061 6972  gh the "non-pair
+0000fba0: 7769 7365 206d 6f64 6522 2028 6465 6661  wise mode" (defa
+0000fbb0: 756c 7420 7365 7474 696e 6729 2077 6869  ult setting) whi
+0000fbc0: 6368 2072 756e 7320 6f6e 2074 6865 2077  ch runs on the w
+0000fbd0: 686f 6c65 2061 6c69 676e 6d65 6e74 2069  hole alignment i
+0000fbe0: 6e73 7465 6164 206f 6620 6120 6c6f 6361  nstead of a loca
+0000fbf0: 6c20 616c 6967 6e6d 656e 742c 206d 6967  l alignment, mig
+0000fc00: 6874 2062 6520 6d6f 7265 2062 696f 6c6f  ht be more biolo
+0000fc10: 6769 6361 6c6c 7920 636f 6e73 6572 7665  gically conserve
+0000fc20: 6420 696e 2074 6861 7420 6974 2061 7373  d in that it ass
+0000fc30: 7572 6573 2074 6865 2065 766f 6c75 7469  ures the evoluti
+0000fc40: 6f6e 206f 6620 6561 6368 2063 6f6c 756d  on of each colum
+0000fc50: 6e20 746f 2062 6520 7374 6172 7465 6420  n to be started 
+0000fc60: 6672 6f6d 2074 6865 2072 6f6f 7420 6f66  from the root of
+0000fc70: 2074 6865 2066 616d 696c 7920 616e 6420   the family and 
+0000fc80: 616c 6c20 7468 6520 6765 6e65 2064 7570  all the gene dup
+0000fc90: 6c69 6361 7465 7320 6172 6520 7461 6b65  licates are take
+0000fca0: 6e20 696e 746f 2061 6363 6f75 6e74 2069  n into account i
+0000fcb0: 6e20 7468 6520 2a4b 2a3c 7375 623e 533c  n the *K*<sub>S<
+0000fcc0: 2f73 7562 3e20 6573 7469 6d61 7469 6f6e  /sub> estimation
+0000fcd0: 2070 726f 6365 7373 2e20 5468 6520 6f70   process. The op
+0000fce0: 7469 6f6e 2060 2d2d 7374 7269 705f 6761  tion `--strip_ga
+0000fcf0: 7073 6020 6361 6e20 7265 6d6f 7665 2061  ps` can remove a
+0000fd00: 6c6c 2074 6865 2067 6170 2d63 6f6e 7461  ll the gap-conta
+0000fd10: 696e 696e 6720 636f 6c75 6d6e 732c 2077  ining columns, w
+0000fd20: 6974 6820 6f72 2077 6974 686f 7574 2077  ith or without w
+0000fd30: 6869 6368 2074 6865 2072 6573 756c 7420  hich the result 
+0000fd40: 6f66 2022 6e6f 6e2d 7061 6972 7769 7365  of "non-pairwise
+0000fd50: 206d 6f64 6522 2077 6f6e 2774 2062 6520   mode" won't be 
+0000fd60: 6166 6665 6374 6564 2c20 7768 696c 6520  affected, while 
+0000fd70: 7769 7468 2077 6869 6368 2074 6865 2072  with which the r
+0000fd80: 6573 756c 7420 6f66 2022 7061 6972 7769  esult of "pairwi
+0000fd90: 7365 206d 6f64 6522 2077 696c 6c20 6265  se mode" will be
+0000fda0: 2061 6c74 6572 6564 2e20 5468 6520 6f70   altered. The op
+0000fdb0: 7469 6f6e 2060 2d2d 616c 6967 6e65 7260  tion `--aligner`
+0000fdc0: 2061 6e64 2060 2d2d 616c 6e5f 6f70 7469   and `--aln_opti
+0000fdd0: 6f6e 7360 2077 6869 6368 2064 6563 6964  ons` which decid
+0000fde0: 6520 7768 6963 6820 616c 6967 6e6d 656e  e which alignmen
+0000fdf0: 7420 7072 6f67 7261 6d20 746f 2062 6520  t program to be 
+0000fe00: 7573 6564 2061 6e64 2077 6869 6368 2070  used and which p
+0000fe10: 6172 616d 6574 6572 2074 6f20 6265 2073  arameter to be s
+0000fe20: 6574 2c20 7769 6c6c 2068 6176 6520 696d  et, will have im
+0000fe30: 7061 6374 206f 6e20 7468 6520 2a4b 2a3c  pact on the *K*<
+0000fe40: 7375 623e 533c 2f73 7562 3e20 7265 7375  sub>S</sub> resu
+0000fe50: 6c74 732c 206e 6f74 6564 2074 6861 7420  lts, noted that 
+0000fe60: 7468 6520 6465 6661 756c 7420 7072 6f67  the default prog
+0000fe70: 7261 6d20 6973 2060 6d61 6666 7460 2061  ram is `mafft` a
+0000fe80: 6e64 2074 6865 2070 6172 616d 6574 6572  nd the parameter
+0000fe90: 2069 7320 602d 2d61 7574 6f60 2e20 5468   is `--auto`. Th
+0000fea0: 6520 6f70 7469 6f6e 2060 2d2d 7472 6565  e option `--tree
+0000feb0: 5f6d 6574 686f 6460 2061 6e64 2060 2d2d  _method` and `--
+0000fec0: 7472 6565 5f6f 7074 696f 6e73 6020 6465  tree_options` de
+0000fed0: 6369 6465 2077 6869 6368 2067 656e 6520  cide which gene 
+0000fee0: 7472 6565 2069 6e66 6572 656e 6365 2070  tree inference p
+0000fef0: 726f 6772 616d 2074 6f20 6265 2075 7365  rogram to be use
+0000ff00: 6420 616e 6420 7768 6963 6820 7061 7261  d and which para
+0000ff10: 6d65 7465 7220 746f 2062 6520 7365 742c  meter to be set,
+0000ff20: 2077 6f6e 2774 2061 6666 6563 7420 7468   won't affect th
+0000ff30: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+0000ff40: 3e20 6573 7469 6d61 7469 6f6e 2069 7473  > estimation its
+0000ff50: 656c 6620 6275 7420 7468 6520 7265 7375  elf but the resu
+0000ff60: 6c74 206f 6620 6465 2d72 6564 756e 6461  lt of de-redunda
+0000ff70: 6e63 792c 206e 6f74 6564 2074 6861 7420  ncy, noted that 
+0000ff80: 7765 2069 6d70 6c65 6d65 6e74 6564 2061  we implemented a
+0000ff90: 2062 7569 6c74 2d69 6e20 6765 6e65 2074   built-in gene t
+0000ffa0: 7265 6520 696e 6665 7265 6e63 6520 6d65  ree inference me
+0000ffb0: 7468 6f64 2062 6173 6564 206f 6e20 7468  thod based on th
+0000ffc0: 6520 4176 6572 6167 6520 4c69 6e6b 6167  e Average Linkag
+0000ffd0: 6520 436c 7573 7465 7269 6e67 2028 414c  e Clustering (AL
+0000ffe0: 4329 2028 7468 7573 2061 2064 6973 7461  C) (thus a dista
+0000fff0: 6e63 652d 6261 7365 6420 7472 6565 2920  nce-based tree) 
+00010000: 7769 7468 2074 6865 2060 2d2d 7472 6565  with the `--tree
+00010010: 5f6d 6574 686f 6460 2073 6574 2061 7320  _method` set as 
+00010020: 2263 6c75 7374 6572 222e 2054 776f 206d  "cluster". Two m
+00010030: 6574 686f 6473 206f 6620 6465 2d72 6564  ethods of de-red
+00010040: 756e 6461 6e63 7920 6172 6520 696d 706c  undancy are impl
+00010050: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
+00010060: 7632 602c 206e 616d 656c 7920 6e6f 6465  v2`, namely node
+00010070: 2d77 6569 6768 7465 6420 616e 6420 6e6f  -weighted and no
+00010080: 6465 2d61 7665 7261 6765 6420 6d65 7468  de-averaged meth
+00010090: 6f64 732e 2054 6865 206e 6f64 652d 7765  ods. The node-we
+000100a0: 6967 6874 6564 206d 6574 686f 6420 6163  ighted method ac
+000100b0: 6869 6576 6573 2074 6865 2064 652d 7265  hieves the de-re
+000100c0: 6475 6e64 616e 6379 2076 6961 2077 6569  dundancy via wei
+000100d0: 6768 696e 6720 7468 6520 2a4b 2a3c 7375  ghing the *K*<su
+000100e0: 623e 533c 2f73 7562 3e20 7661 6c75 6520  b>S</sub> value 
+000100f0: 6173 736f 6369 6174 6564 2077 6974 6820  associated with 
+00010100: 6561 6368 2067 656e 6520 7061 6972 2073  each gene pair s
+00010110: 7563 6820 7468 6174 2074 6865 2077 6569  uch that the wei
+00010120: 6768 7420 6f66 2061 2073 696e 676c 6520  ght of a single 
+00010130: 6475 706c 6963 6174 696f 6e20 6576 656e  duplication even
+00010140: 7420 7375 6d73 2075 7020 746f 2031 2028  t sums up to 1 (
+00010150: 6e6f 7465 6420 7468 6174 2074 6865 206e  noted that the n
+00010160: 756d 6265 7220 6f66 202a 4b2a 3c73 7562  umber of *K*<sub
+00010170: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
+00010180: 696f 6e20 7265 6d61 696e 7320 7468 6520  ion remains the 
+00010190: 7361 6d65 2920 7768 696c 6520 7468 6520  same) while the 
+000101a0: 6e6f 6465 2d61 7665 7261 6765 6420 6d65  node-averaged me
+000101b0: 7468 6f64 2072 6561 6c69 7a65 7320 7468  thod realizes th
+000101c0: 6520 6465 2d72 6564 756e 6461 6e63 7920  e de-redundancy 
+000101d0: 7669 6120 6361 6c63 756c 6174 696e 6720  via calculating 
+000101e0: 7065 7220 6765 6e65 2074 7265 6520 6e6f  per gene tree no
+000101f0: 6465 206f 6e65 2061 7665 7261 6765 6420  de one averaged 
+00010200: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00010210: 7661 6c75 6520 746f 2072 6570 7265 7365  value to represe
+00010220: 6e74 2074 6865 2061 6765 206f 6620 6561  nt the age of ea
+00010230: 6368 2064 7570 6c69 6361 7469 6f6e 2065  ch duplication e
+00010240: 7665 6e74 2e20 5468 6520 6f70 7469 6f6e  vent. The option
+00010250: 2060 2d2d 6e6f 6465 5f61 7665 7261 6765   `--node_average
+00010260: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
+00010270: 6368 6f6f 7365 2074 6865 206e 6f64 652d  choose the node-
+00010280: 6176 6572 6167 6564 2077 6179 206f 6620  averaged way of 
+00010290: 6465 2d72 6564 756e 6461 6e63 792e 2044  de-redundancy. D
+000102a0: 6966 6665 7265 6e74 206d 6574 686f 6473  ifferent methods
+000102b0: 206f 6620 6465 2d72 6564 756e 6461 6e63   of de-redundanc
+000102c0: 7920 7769 6c6c 2068 6176 6520 696d 7061  y will have impa
+000102d0: 6374 206f 6e20 7468 6520 6465 7465 6374  ct on the detect
+000102e0: 696f 6e20 6f66 2057 4744 2073 6967 6e61  ion of WGD signa
+000102f0: 6c73 2c20 7768 6963 6820 6861 7320 6265  ls, which has be
+00010300: 656e 2069 6e76 6573 7469 6761 7465 6420  en investigated 
+00010310: 696e 2074 6869 7320 5b6c 6974 6572 6174  in this [literat
+00010320: 7572 655d 2868 7474 7073 3a2f 2f64 6f69  ure](https://doi
+00010330: 2e6f 7267 2f31 302e 3130 3933 2f67 6265  .org/10.1093/gbe
+00010340: 2f65 7679 3230 3029 2e20 5468 6520 7061  /evy200). The pa
+00010350: 7261 6c6c 656c 2074 6872 6561 6473 2068  rallel threads h
+00010360: 6572 6520 6172 6520 746f 2070 6172 616c  ere are to paral
+00010370: 6c65 6c69 7a65 2074 6865 2061 6e61 6c79  lelize the analy
+00010380: 7369 7320 666f 7220 6561 6368 2067 656e  sis for each gen
+00010390: 6520 6661 6d69 6c79 2061 6e64 2074 6875  e family and thu
+000103a0: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+000103b0: 6520 7365 7420 6173 206d 7563 6820 6173  e set as much as
+000103c0: 2074 6865 206e 756d 6265 7220 6f66 2067   the number of g
+000103d0: 656e 6520 6661 6d69 6c69 6573 2e0a 0a2a  ene families...*
+000103e0: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
+000103f0: 7274 696e 6720 7275 6e20 6361 6e20 7573  rting run can us
+00010400: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
+00010410: 6f77 2a2a 0a0a 6060 600a 7767 6420 6b73  ow**..```.wgd ks
+00010420: 6420 6661 6d69 6c69 6573 2073 6571 7565  d families seque
+00010430: 6e63 650a 6060 600a 0a2a 2a54 6865 2063  nce.```..**The c
+00010440: 6f6e 7374 7275 6374 696f 6e20 6f66 206f  onstruction of o
+00010450: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
+00010460: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+00010470: 6973 7472 6962 7574 696f 6e2a 2a0a 6060  istribution**.``
+00010480: 600a 7767 6420 6b73 6420 6661 6d69 6c69  `.wgd ksd famili
+00010490: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
+000104a0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+000104b0: 2028 2d2d 7265 7765 6967 6874 290a 6060   (--reweight).``
+000104c0: 600a 0a46 726f 6d20 7061 7261 6c6f 676f  `..From paralogo
+000104d0: 7573 2074 6f20 6f72 7468 6f6c 6f67 6f75  us to orthologou
+000104e0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+000104f0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00010500: 6f6e 2c20 7573 6572 7320 6f6e 6c79 206e  on, users only n
+00010510: 6565 6420 746f 2070 726f 7669 6465 206d  eed to provide m
+00010520: 6f72 6520 6364 7320 6669 6c65 732e 204e  ore cds files. N
+00010530: 6f74 6520 7468 6174 2077 6974 6820 6f72  ote that with or
+00010540: 7468 6f6c 6f67 6f75 7320 6765 6e65 2066  thologous gene f
+00010550: 616d 696c 6965 7320 7468 6520 7765 6967  amilies the weig
+00010560: 6874 696e 6720 6d65 7468 6f64 2063 616e  hting method can
+00010570: 2062 6520 7365 7420 746f 2062 6520 6361   be set to be ca
+00010580: 6c63 756c 6174 6564 2070 6572 2073 7065  lculated per spe
+00010590: 6369 6573 2070 6169 7220 696e 7374 6561  cies pair instea
+000105a0: 6420 6f66 2063 6f6e 7369 6465 7269 6e67  d of considering
+000105b0: 2074 6865 2077 686f 6c65 2066 616d 696c   the whole famil
+000105c0: 7920 6265 6361 7573 6520 7768 656e 2070  y because when p
+000105d0: 6c6f 7474 696e 6720 6f72 7468 6f6c 6f67  lotting ortholog
+000105e0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+000105f0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00010600: 7469 6f6e 2062 6574 7765 656e 2074 776f  tion between two
+00010610: 2073 7065 6369 6573 2074 6865 2077 6569   species the wei
+00010620: 6768 7420 6361 6c63 756c 6174 6564 2066  ght calculated f
+00010630: 726f 6d20 7468 6973 2073 7065 6369 6669  rom this specifi
+00010640: 6320 7370 6563 6965 7320 7061 6972 2073  c species pair s
+00010650: 686f 756c 6420 6265 2063 6f6e 7365 7276  hould be conserv
+00010660: 6564 2077 6869 6c65 2074 6865 206f 6e65  ed while the one
+00010670: 2063 616c 6375 6c61 7465 6420 6672 6f6d   calculated from
+00010680: 2074 6865 2077 686f 6c65 2066 616d 696c   the whole famil
+00010690: 7920 7769 6c6c 2076 6172 7920 7769 7468  y will vary with
+000106a0: 2074 6865 206e 756d 6265 7220 6f66 2073   the number of s
+000106b0: 7065 6369 6573 2e20 546f 2069 6e69 7469  pecies. To initi
+000106c0: 6174 6520 7468 6520 7765 6967 6874 696e  ate the weightin
+000106d0: 6720 7065 7220 7370 6563 6965 7320 7061  g per species pa
+000106e0: 6972 2c20 7468 6520 6f70 7469 6f6e 2060  ir, the option `
+000106f0: 2d2d 7265 7765 6967 6874 6020 6361 6e20  --reweight` can 
+00010700: 6265 2073 6574 2e0a 0a2a 2a41 2073 7567  be set...**A sug
+00010710: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+00010720: 7275 6e20 6361 6e20 7573 6520 636f 6d6d  run can use comm
+00010730: 616e 6420 6173 2062 656c 6f77 2a2a 0a0a  and as below**..
+00010740: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
+00010750: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+00010760: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+00010770: 6533 0a60 6060 0a0a 2a2a 5468 6520 636f  e3.```..**The co
+00010780: 6e73 7472 7563 7469 6f6e 206f 6620 2a4b  nstruction of *K
+00010790: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+000107a0: 6520 6469 7374 7269 6275 7469 6f6e 2077  e distribution w
+000107b0: 6974 6820 7261 7465 2063 6f72 7265 6374  ith rate correct
+000107c0: 696f 6e2a 2a0a 6060 600a 7767 6420 6b73  ion**.```.wgd ks
+000107d0: 6420 6661 6d69 6c69 6573 2073 6571 7565  d families seque
+000107e0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+000107f0: 6571 7565 6e63 6533 202d 2d66 6f63 7573  equence3 --focus
+00010800: 3261 6c6c 2073 6571 7565 6e63 6531 202d  2all sequence1 -
+00010810: 7370 2073 7064 6174 6120 2d2d 6578 7472  sp spdata --extr
+00010820: 6170 6172 616e 6f6d 656b 7320 7061 7261  aparanomeks para
+00010830: 6e6f 6d65 4b73 6461 7461 2028 2d2d 706c  nomeKsdata (--pl
+00010840: 6f74 656c 6d6d 202d 2d70 6c6f 7461 7067  otelmm --plotapg
+00010850: 6d6d 202d 2d61 6e63 686f 7270 6f69 6e74  mm --anchorpoint
+00010860: 7320 6170 6461 7461 202d 2d72 6577 6569  s apdata --rewei
+00010870: 6768 7420 2d2d 6f6e 6c79 726f 6f74 6f75  ght --onlyrootou
+00010880: 7429 0a60 6060 0a0a 496e 7370 6972 6564  t).```..Inspired
+00010890: 2062 7920 7468 6520 7261 7465 2063 6f72   by the rate cor
+000108a0: 7265 6374 696f 6e20 616c 676f 7269 7468  rection algorith
+000108b0: 6d20 696e 205b 6b73 7261 7465 735d 2868  m in [ksrates](h
+000108c0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+000108d0: 6d2f 5649 422d 5053 422f 6b73 7261 7465  m/VIB-PSB/ksrate
+000108e0: 7329 2c20 7765 2069 6d70 6c65 6d65 6e74  s), we implement
+000108f0: 6564 2074 6865 2072 6174 6520 636f 7272  ed the rate corr
+00010900: 6563 7469 6f6e 2061 6e61 6c79 7369 7320  ection analysis 
+00010910: 616c 736f 2069 6e20 6077 6764 2076 3260  also in `wgd v2`
+00010920: 2c20 7768 6963 6820 6973 206d 6f73 746c  , which is mostl
+00010930: 7920 7468 6520 7361 6d65 2061 7320 5b6b  y the same as [k
+00010940: 7372 6174 6573 5d28 6874 7470 733a 2f2f  srates](https://
+00010950: 6769 7468 7562 2e63 6f6d 2f56 4942 2d50  github.com/VIB-P
+00010960: 5342 2f6b 7372 6174 6573 2920 6275 7420  SB/ksrates) but 
+00010970: 6469 6666 6572 7320 696e 2074 6865 2063  differs in the c
+00010980: 616c 6375 6c61 7469 6f6e 206f 6620 7468  alculation of th
+00010990: 6520 7374 616e 6461 7264 2064 6576 6961  e standard devia
+000109a0: 7469 6f6e 206f 6620 7265 7363 616c 6564  tion of rescaled
+000109b0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000109c0: 2061 6765 732e 2054 6f20 7065 7266 6f72   ages. To perfor
+000109d0: 6d20 7468 6520 7261 7465 2063 6f72 7265  m the rate corre
+000109e0: 6374 696f 6e20 616e 616c 7973 6973 2c20  ction analysis, 
+000109f0: 7573 6572 7320 6361 6e20 7573 6520 626f  users can use bo
+00010a00: 7468 2074 6865 2060 7767 6420 6b73 6460  th the `wgd ksd`
+00010a10: 2061 6e64 2060 7767 6420 7669 7a60 2070   and `wgd viz` p
+00010a20: 726f 6772 616d 2e20 466f 7220 7468 6520  rogram. For the 
+00010a30: 6077 6764 206b 7364 6020 7072 6f67 7261  `wgd ksd` progra
+00010a40: 6d2c 2075 7365 7273 206e 6565 6420 746f  m, users need to
+00010a50: 2070 726f 7669 6465 2061 2073 7065 6369   provide a speci
+00010a60: 6573 2074 7265 6520 7669 6120 7468 6520  es tree via the 
+00010a70: 6f70 7469 6f6e 2060 2d2d 7370 6563 6965  option `--specie
+00010a80: 7374 7265 6560 206f 6e20 7768 6963 6820  stree` on which 
+00010a90: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
+00010aa0: 6361 6e20 6265 2063 6f6e 6475 6374 6564  can be conducted
+00010ab0: 2e20 4e6f 7465 2074 6861 7420 756e 6e65  . Note that unne
+00010ac0: 6365 7373 6172 7920 6272 6163 6b65 7473  cessary brackets
+00010ad0: 206d 6967 6874 206c 6561 6420 746f 2075   might lead to u
+00010ae0: 6e65 7870 6563 7465 6420 6572 726f 7273  nexpected errors
+00010af0: 2c20 666f 7220 696e 7374 616e 6365 2061  , for instance a
+00010b00: 2074 7265 6520 6028 412c 2842 2c43 2929   tree `(A,(B,C))
+00010b10: 3b60 2073 686f 756c 6420 6e6f 7420 6265  ;` should not be
+00010b20: 2072 6570 7265 7365 6e74 6564 2061 7320   represented as 
+00010b30: 6028 412c 2828 422c 4329 2929 3b60 2e20  `(A,((B,C)));`. 
+00010b40: 5468 6520 7365 7420 6f66 2073 7065 6369  The set of speci
+00010b50: 6573 2070 6169 7273 2074 6f20 6265 2073  es pairs to be s
+00010b60: 686f 776e 2069 7320 666c 6578 6962 6c65  hown is flexible
+00010b70: 2074 6861 7420 7468 6520 6d6f 7374 2063   that the most c
+00010b80: 6f6e 7665 6e69 656e 7420 6f70 7469 6f6e  onvenient option
+00010b90: 2069 7320 602d 2d66 6f63 7573 3261 6c6c   is `--focus2all
+00010ba0: 6020 7768 6963 6820 7369 6d70 6c79 2073  ` which simply s
+00010bb0: 686f 7773 2061 6c6c 2074 6865 2070 6f73  hows all the pos
+00010bc0: 7369 626c 6520 666f 6361 6c2d 7369 7374  sible focal-sist
+00010bd0: 6572 2073 7065 6369 6573 2070 6169 7273  er species pairs
+00010be0: 2c20 6f72 2075 7365 7273 2063 616e 206d  , or users can m
+00010bf0: 616e 7561 6c6c 7920 7365 7420 7468 6520  anually set the 
+00010c00: 7370 6563 6965 7320 7061 6972 7320 7669  species pairs vi
+00010c10: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00010c20: 7370 6169 7260 2e20 4e6f 7465 2074 6861  spair`. Note tha
+00010c30: 7420 6966 2074 6865 2073 7065 6369 6573  t if the species
+00010c40: 2070 6169 7273 2077 6572 6520 6d61 6e75   pairs were manu
+00010c50: 616c 6c79 2073 6574 2c20 6974 2077 6f75  ally set, it wou
+00010c60: 6c64 2062 6520 6e65 6564 6564 2074 6f20  ld be needed to 
+00010c70: 636f 2d73 6574 2074 6865 206f 7074 696f  co-set the optio
+00010c80: 6e20 602d 2d63 6c61 7373 6963 602e 2054  n `--classic`. T
+00010c90: 6865 206f 7074 696f 6e20 602d 2d6f 6e6c  he option `--onl
+00010ca0: 7972 6f6f 746f 7574 6020 6361 6e20 6265  yrootout` can be
+00010cb0: 2073 6574 2074 6f20 6f6e 6c79 2063 6f6e   set to only con
+00010cc0: 7369 6465 7220 6f75 7467 726f 7570 2061  sider outgroup a
+00010cd0: 7420 7468 6520 726f 6f74 2c20 696e 7374  t the root, inst
+00010ce0: 6561 6420 6f66 2061 6c6c 2074 6865 2070  ead of all the p
+00010cf0: 6f73 7369 626c 6520 6f75 7467 726f 7570  ossible outgroup
+00010d00: 7320 7065 7220 666f 6361 6c2d 7369 7374  s per focal-sist
+00010d10: 6572 2073 7065 6369 6573 2070 6169 722c  er species pair,
+00010d20: 2077 6869 6368 2068 6173 2069 6d70 6163   which has impac
+00010d30: 7420 6f6e 2074 6865 2066 696e 616c 2063  t on the final c
+00010d40: 6f72 7265 6374 6564 202a 4b2a 3c73 7562  orrected *K*<sub
+00010d50: 3e53 3c2f 7375 623e 2061 6765 732e 2057  >S</sub> ages. W
+00010d60: 6520 7375 6767 6573 7420 6f66 2075 7369  e suggest of usi
+00010d70: 6e67 2061 6c6c 2074 6865 2070 6f73 7369  ng all the possi
+00010d80: 626c 6520 6f75 7467 726f 7570 7320 7065  ble outgroups pe
+00010d90: 7220 666f 6361 6c2d 7369 7374 6572 2073  r focal-sister s
+00010da0: 7065 6369 6573 2070 6169 7220 6173 2066  pecies pair as f
+00010db0: 6f72 2061 206c 6573 7320 6269 6173 6564  or a less biased
+00010dc0: 2072 6573 756c 742e 2054 6865 2070 6172   result. The par
+00010dd0: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00010de0: 2f73 7562 3e20 6461 7461 2073 686f 756c  /sub> data shoul
+00010df0: 6420 6265 2070 726f 7669 6465 6420 7669  d be provided vi
+00010e00: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00010e10: 6578 7472 6170 6172 616e 6f6d 656b 7360  extraparanomeks`
+00010e20: 2e20 0a0a 536f 6d65 206f 7468 6572 206f  . ..Some other o
+00010e30: 7074 696f 6e73 2077 6869 6368 2068 6176  ptions which hav
+00010e40: 6520 6e6f 2069 6d70 6163 7420 6f6e 2074  e no impact on t
+00010e50: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+00010e60: 6f6e 2062 7574 2061 6464 206d 6f72 6520  on but add more 
+00010e70: 6c61 7965 7273 206f 7220 6368 616e 6765  layers or change
+00010e80: 2074 6865 2061 7070 6561 7261 6e63 6520   the appearance 
+00010e90: 6f6e 2074 6865 206d 6978 6564 202a 4b2a  on the mixed *K*
+00010ea0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00010eb0: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
+00010ec0: 6e63 6c75 6465 2060 2d2d 706c 6f74 6170  nclude `--plotap
+00010ed0: 676d 6d60 2061 6e64 2060 2d2d 706c 6f74  gmm` and `--plot
+00010ee0: 656c 6d6d 6020 6574 632e 2054 6865 206f  elmm` etc. The o
+00010ef0: 7074 696f 6e20 602d 2d70 6c6f 7461 7067  ption `--plotapg
+00010f00: 6d6d 6020 6361 6e20 6265 2073 6574 2074  mm` can be set t
+00010f10: 6f20 6361 6c6c 2074 6865 2047 4d4d 2061  o call the GMM a
+00010f20: 6e61 6c79 7369 7320 7570 6f6e 2074 6865  nalysis upon the
+00010f30: 2061 6e63 686f 7220 7061 6972 202a 4b2a   anchor pair *K*
+00010f40: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00010f50: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
+00010f60: 6420 706c 6f74 2074 6865 2063 6c75 7374  d plot the clust
+00010f70: 6572 696e 6720 7265 7375 6c74 2075 706f  ering result upo
+00010f80: 6e20 7468 6520 6d69 7865 6420 2a4b 2a3c  n the mixed *K*<
+00010f90: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00010fa0: 6469 7374 7269 6275 7469 6f6e 2c20 7768  distribution, wh
+00010fb0: 6963 6820 6861 7320 746f 2062 6520 636f  ich has to be co
+00010fc0: 2d73 6574 2077 6974 6820 7468 6520 6f70  -set with the op
+00010fd0: 7469 6f6e 2060 2d2d 616e 6368 6f72 706f  tion `--anchorpo
+00010fe0: 696e 7473 6020 7072 6f76 6964 696e 6720  ints` providing 
+00010ff0: 7468 6520 616e 6368 6f72 2070 6169 7273  the anchor pairs
+00011000: 2069 6e66 6f72 6d61 7469 6f6e 2e20 5468   information. Th
+00011010: 6520 6f70 7469 6f6e 2060 2d2d 706c 6f74  e option `--plot
+00011020: 656c 6d6d 6020 6361 6e20 6265 2073 6574  elmm` can be set
+00011030: 2074 6f20 6361 6c6c 2074 6865 2045 4c4d   to call the ELM
+00011040: 4d20 616e 616c 7973 6973 2075 706f 6e20  M analysis upon 
+00011050: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
+00011060: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
+00011070: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+00011080: 696f 6e2e 202a 2a4e 6f74 6520 7468 6174  ion. **Note that
+00011090: 2074 6865 2073 7065 6369 6573 206e 616d   the species nam
+000110a0: 6573 2070 7265 7365 6e74 2069 6e20 7468  es present in th
+000110b0: 6520 7370 6563 6965 7320 7472 6565 2066  e species tree f
+000110c0: 696c 6520 7368 6f75 6c64 206d 6174 6368  ile should match
+000110d0: 2074 6865 206e 616d 6573 206f 6620 7468   the names of th
+000110e0: 6520 636f 7272 6573 706f 6e64 696e 6720  e corresponding 
+000110f0: 7365 7175 656e 6365 2066 696c 6573 2a2a  sequence files**
+00011100: 2046 6f72 2069 6e73 7461 6e63 652c 2067   For instance, g
+00011110: 6976 656e 2074 6865 2063 6473 2066 696c  iven the cds fil
+00011120: 6520 6e61 6d65 7320 2741 2e63 6473 272c  e names 'A.cds',
+00011130: 2742 2e63 6473 272c 2743 2e63 6473 272c  'B.cds','C.cds',
+00011140: 2074 6865 2073 7065 6369 6573 2074 7265   the species tre
+00011150: 6520 636f 756c 6420 6265 2027 2841 2e63  e could be '(A.c
+00011160: 6473 2c28 422e 6364 732c 432e 6364 7329  ds,(B.cds,C.cds)
+00011170: 293b 2720 7261 7468 6572 2074 6861 6e20  );' rather than 
+00011180: 2728 412c 2842 2c43 2929 3b27 2e20 5468  '(A,(B,C));'. Th
+00011190: 6572 6520 6973 206e 6f20 7265 7175 6972  ere is no requir
+000111a0: 656d 656e 7420 666f 7220 7468 6520 6e61  ement for the na
+000111b0: 6d65 206f 6620 7468 6520 7061 7261 6e6f  me of the parano
+000111c0: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
+000111d0: 623e 2064 6174 6166 696c 6520 7768 6963  b> datafile whic
+000111e0: 6820 6361 6e20 6265 206e 616d 6564 2069  h can be named i
+000111f0: 6e20 7768 6174 6576 6572 206d 616e 6e65  n whatever manne
+00011200: 7220 7573 6572 7320 7072 6566 6572 2e20  r users prefer. 
+00011210: 5468 6520 2747 4d4d 2720 6973 2074 6865  The 'GMM' is the
+00011220: 2061 6262 7265 7669 6174 696f 6e20 6f66   abbreviation of
+00011230: 2047 6175 7373 6961 6e20 4d69 7874 7572   Gaussian Mixtur
+00011240: 6520 4d6f 6465 6c69 6e67 2077 6869 6c65  e Modeling while
+00011250: 2074 6865 2027 454c 4d4d 2720 7265 6665   the 'ELMM' refe
+00011260: 7273 2074 6f20 4578 706f 6e65 6e74 6961  rs to Exponentia
+00011270: 6c2d 4c6f 676e 6f72 6d61 6c20 4d69 7874  l-Lognormal Mixt
+00011280: 7572 6520 4d6f 6465 6c69 6e67 2061 7320  ure Modeling as 
+00011290: 5b6b 7372 6174 6573 5d28 6874 7470 733a  [ksrates](https:
+000112a0: 2f2f 6769 7468 7562 2e63 6f6d 2f56 4942  //github.com/VIB
+000112b0: 2d50 5342 2f6b 7372 6174 6573 2920 696e  -PSB/ksrates) in
+000112c0: 7465 7270 7265 7473 2e0a 0a54 6865 7265  terprets...There
+000112d0: 2061 7265 2032 3120 636f 6c75 6d6e 7320   are 21 columns 
+000112e0: 696e 2074 6865 2072 6573 756c 7420 602e  in the result `.
+000112f0: 6b73 2e74 7376 6020 6669 6c65 2062 6573  ks.tsv` file bes
+00011300: 6964 6573 2074 6865 2069 6e64 6578 2063  ides the index c
+00011310: 6f6c 756d 6e73 2060 7061 6972 6020 6173  olumns `pair` as
+00011320: 2074 6865 2075 6e69 7175 6520 6964 656e   the unique iden
+00011330: 7469 6669 6572 2066 6f72 2065 6163 6820  tifier for each 
+00011340: 6765 6e65 2070 6169 722e 2054 6865 2060  gene pair. The `
+00011350: 4e60 2c20 6053 602c 2060 644e 602c 2060  N`, `S`, `dN`, `
+00011360: 644e 2f64 5360 2c20 6064 5360 2c20 606c  dN/dS`, `dS`, `l
+00011370: 6020 616e 6420 6074 6020 6172 6520 6672  ` and `t` are fr
+00011380: 6f6d 2074 6865 2063 6f64 656d 6c20 7265  om the codeml re
+00011390: 7375 6c74 732c 2072 6570 7265 7365 6e74  sults, represent
+000113a0: 696e 6720 7468 6520 4e20 6573 7469 6d61  ing the N estima
+000113b0: 7465 2c20 7468 6520 5320 6573 7469 6d61  te, the S estima
+000113c0: 7465 2c20 7468 6520 644e 2065 7374 696d  te, the dN estim
+000113d0: 6174 652c 2074 6865 2064 4e2f 6453 2028  ate, the dN/dS (
+000113e0: 6f6d 6567 6129 2065 7374 696d 6174 652c  omega) estimate,
+000113f0: 2074 6865 2064 5320 6573 7469 6d61 7465   the dS estimate
+00011400: 2c20 7468 6520 6c6f 672d 6c69 6b65 6c69  , the log-likeli
+00011410: 686f 6f64 2061 6e64 2074 6865 2074 2065  hood and the t e
+00011420: 7374 696d 6174 652c 2072 6573 7065 6374  stimate, respect
+00011430: 6976 656c 792e 2054 6865 2060 616c 6967  ively. The `alig
+00011440: 6e6d 656e 7463 6f76 6572 6167 6560 2c20  nmentcoverage`, 
+00011450: 6061 6c69 676e 6d65 6e74 6964 656e 7469  `alignmentidenti
+00011460: 7479 6020 616e 6420 6061 6c69 676e 6d65  ty` and `alignme
+00011470: 6e74 6c65 6e67 7468 6020 6172 6520 7468  ntlength` are th
+00011480: 6520 696e 666f 726d 6174 696f 6e20 7065  e information pe
+00011490: 7274 6169 6e69 6e67 2074 6f20 7468 6520  rtaining to the 
+000114a0: 616c 6967 6e6d 656e 7420 666f 7220 6561  alignment for ea
+000114b0: 6368 2066 616d 696c 792c 2072 6570 7265  ch family, repre
+000114c0: 7365 6e74 696e 6720 7468 6520 7261 7469  senting the rati
+000114d0: 6f20 6f66 2074 6865 2073 7472 6970 7065  o of the strippe
+000114e0: 6420 616c 6967 6e6d 656e 7420 6c65 6e67  d alignment leng
+000114f0: 7468 2063 6f6d 7061 7265 6420 746f 2074  th compared to t
+00011500: 6865 2066 756c 6c20 616c 6967 6e6d 656e  he full alignmen
+00011510: 7420 6c65 6e67 7468 2c20 7468 6520 7261  t length, the ra
+00011520: 7469 6f20 6f66 2063 6f6c 756d 6e73 2077  tio of columns w
+00011530: 6974 6820 6964 656e 7469 6361 6c20 6e75  ith identical nu
+00011540: 636c 656f 7469 6465 7320 636f 6d70 6172  cleotides compar
+00011550: 6564 2074 6f20 7468 6520 6f76 6572 616c  ed to the overal
+00011560: 6c20 636f 6c75 6d6e 7320 6f66 2074 6865  l columns of the
+00011570: 2073 7472 6970 7065 6420 616c 6967 6e6d   stripped alignm
+00011580: 656e 742c 2061 6e64 2074 6865 206c 656e  ent, and the len
+00011590: 6774 6820 6f66 2074 6865 2066 756c 6c20  gth of the full 
+000115a0: 616c 6967 6e6d 656e 742c 2072 6573 7065  alignment, respe
+000115b0: 6374 6976 656c 792e 0a0a 2323 2320 7767  ctively...### wg
+000115c0: 6420 7379 6e0a 0a2a 2a54 6865 2069 6e74  d syn..**The int
+000115d0: 7261 2d73 7065 6369 6669 6320 7379 6e74  ra-specific synt
+000115e0: 656e 7920 696e 6665 7265 6e63 652a 2a0a  eny inference**.
+000115f0: 6060 600a 7767 6420 7379 6e20 6661 6d69  ```.wgd syn fami
+00011600: 6c69 6573 2067 6666 2028 2d2d 6b73 5f64  lies gff (--ks_d
+00011610: 6973 7472 6962 7574 696f 6e20 6b73 6461  istribution ksda
+00011620: 7461 202d 6620 6765 6e65 202d 6120 4944  ta -f gene -a ID
+00011630: 202d 2d6d 696e 6c65 6e20 2d31 202d 2d6d   --minlen -1 --m
+00011640: 696e 7365 676c 656e 2031 3030 3030 202d  inseglen 10000 -
+00011650: 2d6d 696e 6765 6e65 6e75 6d20 3330 290a  -mingenenum 30).
+00011660: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
+00011670: 2060 7767 6420 7379 6e60 2069 7320 6d61   `wgd syn` is ma
+00011680: 696e 6c79 2064 6561 6c69 6e67 2077 6974  inly dealing wit
+00011690: 6820 636f 6c6c 696e 6561 7269 7479 206f  h collinearity o
+000116a0: 7220 7379 6e74 656e 7920 2862 6f74 6820  r synteny (both 
+000116b0: 7265 6665 7272 6564 2074 6f20 6173 2073  referred to as s
+000116c0: 796e 7465 6e79 2068 6572 6561 6674 6572  ynteny hereafter
+000116d0: 2920 616e 616c 7973 6973 2e20 5477 6f20  ) analysis. Two 
+000116e0: 696e 7075 7420 6669 6c65 7320 6172 6520  input files are 
+000116f0: 6573 7365 6e74 6961 6c2c 2074 6865 2067  essential, the g
+00011700: 656e 6520 6661 6d69 6c79 2066 696c 6520  ene family file 
+00011710: 616e 6420 7468 6520 6766 6633 2066 696c  and the gff3 fil
+00011720: 652e 2054 6865 2067 656e 6520 6661 6d69  e. The gene fami
+00011730: 6c79 2066 696c 6520 6973 2069 6e20 7468  ly file is in th
+00011740: 6520 666f 726d 6174 2061 7320 604f 7274  e format as `Ort
+00011750: 686f 4669 6e64 6572 602e 2054 6865 2073  hoFinder`. The s
+00011760: 6f66 7477 6172 6520 6069 2d61 6468 6f72  oftware `i-adhor
+00011770: 6560 2069 7320 6120 7072 6572 6571 7569  e` is a prerequi
+00011780: 7369 7465 2e20 5769 7468 2064 6566 6175  site. With defau
+00011790: 6c74 2070 6172 616d 6574 6572 732c 2074  lt parameters, t
+000117a0: 6865 2070 726f 6772 616d 2062 6173 6963  he program basic
+000117b0: 616c 6c79 2063 6f6e 6475 6374 7320 3129  ally conducts 1)
+000117c0: 2066 696c 7465 7269 6e67 2067 656e 6520   filtering gene 
+000117d0: 6661 6d69 6c69 6573 2062 6173 6564 206f  families based o
+000117e0: 6e20 6d61 7869 6d75 6d20 6661 6d69 6c79  n maximum family
+000117f0: 2073 697a 6520 3229 2072 6574 7269 6576   size 2) retriev
+00011800: 696e 6720 6765 6e65 2070 6f73 6974 696f  ing gene positio
+00011810: 6e20 616e 6420 7363 6166 666f 6c64 2069  n and scaffold i
+00011820: 6e66 6f72 6d61 7469 6f6e 2066 726f 6d20  nformation from 
+00011830: 6766 6633 2066 696c 6520 3329 2070 726f  gff3 file 3) pro
+00011840: 6475 6369 6e67 2074 6865 2063 6f6e 6669  ducing the confi
+00011850: 6775 7261 7469 6f6e 2066 696c 6520 616e  guration file an
+00011860: 6420 6173 736f 6369 6174 6564 2064 6174  d associated dat
+00011870: 6166 696c 6573 2066 6f72 2060 692d 6164  afiles for `i-ad
+00011880: 686f 7265 6020 3429 2063 616c 6c69 6e67  hore` 4) calling
+00011890: 2060 692d 6164 686f 7265 6020 6769 7665   `i-adhore` give
+000118a0: 6e20 7468 6520 7061 7261 6d65 7465 7273  n the parameters
+000118b0: 2073 6574 2074 6f20 696e 6665 7220 7379   set to infer sy
+000118c0: 6e74 656e 7920 3529 2076 6973 7561 6c69  nteny 5) visuali
+000118d0: 7a69 6e67 2074 6865 2073 796e 7465 6e79  zing the synteny
+000118e0: 2069 6e20 2264 6f74 706c 6f74 2220 696e   in "dotplot" in
+000118f0: 2074 6865 2075 6e69 7420 6f66 2067 656e   the unit of gen
+00011900: 6573 2061 6e64 2062 6173 6573 2c20 696e  es and bases, in
+00011910: 2022 5379 6e64 6570 7468 2220 706c 6f74   "Syndepth" plot
+00011920: 2073 686f 7769 6e67 2074 6865 2064 6973   showing the dis
+00011930: 7472 6962 7574 696f 6e20 6f66 2064 6966  tribution of dif
+00011940: 6665 7265 6e74 2063 6174 6567 6f72 6965  ferent categorie
+00011950: 7320 6f66 2063 6f6c 6c69 6e65 6172 6974  s of collinearit
+00011960: 7920 7261 7469 6f73 2077 6974 6869 6e20  y ratios within 
+00011970: 616e 6420 6265 7477 6565 6e20 7370 6563  and between spec
+00011980: 6965 732c 2069 6e20 2264 7570 5374 6163  ies, in "dupStac
+00011990: 6b22 2070 6c6f 7420 7368 6f77 696e 6720  k" plot showing 
+000119a0: 6d75 6c74 6970 6c69 636f 6e73 2077 6974  multiplicons wit
+000119b0: 6820 6469 6666 6572 656e 7420 6d75 6c74  h different mult
+000119c0: 6970 6c69 6361 7469 6f6e 206c 6576 656c  iplication level
+000119d0: 732e 2036 2920 6966 2077 6974 6820 2a4b  s. 6) if with *K
+000119e0: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+000119f0: 7461 2c20 6120 222a 4b2a 3c73 7562 3e53  ta, a "*K*<sub>S
+00011a00: 3c2f 7375 623e 2064 6f74 706c 6f74 2220  </sub> dotplot" 
+00011a10: 7769 7468 2064 6f74 7320 616e 6e6f 7461  with dots annota
+00011a20: 7465 6420 696e 202a 4b2a 3c73 7562 3e53  ted in *K*<sub>S
+00011a30: 3c2f 7375 623e 2076 616c 7565 7320 616e  </sub> values an
+00011a40: 6420 6120 2a4b 2a3c 7375 623e 533c 2f73  d a *K*<sub>S</s
+00011a50: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
+00011a60: 2077 6974 6820 616e 6368 6f72 2070 6169   with anchor pai
+00011a70: 7273 2064 656e 6f74 6564 2077 696c 6c20  rs denoted will 
+00011a80: 6265 2070 726f 6475 6365 642e 2054 6865  be produced. The
+00011a90: 2067 656e 6520 696e 666f 726d 6174 696f   gene informatio
+00011aa0: 6e20 696e 2074 6865 2067 656e 6520 6661  n in the gene fa
+00011ab0: 6d69 6c79 2066 696c 6520 616e 6420 6766  mily file and gf
+00011ac0: 6633 2066 696c 6520 7368 6f75 6c64 2062  f3 file should b
+00011ad0: 6520 6d61 7463 6865 6420 7768 6963 6820  e matched which 
+00011ae0: 7265 7175 6972 6573 2075 7365 7273 2074  requires users t
+00011af0: 6f20 7365 7420 7072 6f70 6572 2060 2d2d  o set proper `--
+00011b00: 6665 6174 7572 6560 2061 6e64 2060 2d2d  feature` and `--
+00011b10: 6174 7472 6962 7574 6560 2e20 5468 6520  attribute`. The 
+00011b20: 6d61 7869 6d75 6d20 6661 6d69 6c79 2073  maximum family s
+00011b30: 697a 6520 746f 2062 6520 696e 636c 7564  ize to be includ
+00011b40: 6564 2063 616e 2062 6520 7365 7420 7669  ed can be set vi
+00011b50: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00011b60: 6d61 7873 697a 6560 2c20 6e6f 7465 6420  maxsize`, noted 
+00011b70: 7468 6174 2074 6869 7320 6669 6c74 6572  that this filter
+00011b80: 696e 6720 6973 206d 6169 6e6c 7920 746f  ing is mainly to
+00011b90: 2064 726f 7020 7468 6f73 6520 6875 6765   drop those huge
+00011ba0: 2074 616e 6465 6d20 6475 706c 6963 6174   tandem duplicat
+00011bb0: 6573 2066 616d 696c 7920 616e 6420 7472  es family and tr
+00011bc0: 616e 7370 6f73 6162 6c65 2065 6c65 6d65  ansposable eleme
+00011bd0: 6e74 7320 2854 4573 2920 6661 6d69 6c79  nts (TEs) family
+00011be0: 2c20 616e 6420 6e6f 7420 6d61 6e64 6174  , and not mandat
+00011bf0: 6f72 792e 2055 7365 7273 2063 616e 2066  ory. Users can f
+00011c00: 696c 7465 7220 7468 6f73 6520 6672 6167  ilter those frag
+00011c10: 6d65 6e74 6172 7920 7363 6166 666f 6c64  mentary scaffold
+00011c20: 7320 7669 6120 7468 6520 6f70 7469 6f6e  s via the option
+00011c30: 2060 2d2d 6d69 6e6c 656e 602e 2054 6865   `--minlen`. The
+00011c40: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
+00011c50: 616e 6420 6e75 6d62 6572 206f 6620 6765  and number of ge
+00011c60: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+00011c70: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+00011c80: 6564 2063 616e 2062 6520 7365 7420 7669  ed can be set vi
+00011c90: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00011ca0: 6d69 6e73 6567 6c65 6e60 2061 6e64 2060  minseglen` and `
+00011cb0: 2d2d 6d69 6e67 656e 656e 756d 602e 2052  --mingenenum`. R
+00011cc0: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+00011cd0: 6963 6f6e 7320 6361 6e20 6265 206b 6570  icons can be kep
+00011ce0: 7420 6279 2073 6574 2074 6865 2066 6c61  t by set the fla
+00011cf0: 6720 6f70 7469 6f6e 2060 2d2d 6b65 6570  g option `--keep
+00011d00: 7265 6475 6e60 2e0a 0a2a 2a41 2073 7567  redun`...**A sug
+00011d10: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+00011d20: 7275 6e20 6361 6e20 7573 6520 636f 6d6d  run can use comm
+00011d30: 616e 6420 7369 6d70 6c79 2061 7320 6265  and simply as be
+00011d40: 6c6f 772a 2a0a 0a60 6060 0a77 6764 2073  low**..```.wgd s
+00011d50: 796e 2066 616d 696c 6965 7320 6766 660a  yn families gff.
+00011d60: 6060 600a 0a2a 2a54 6865 2069 6e74 6572  ```..**The inter
+00011d70: 2d73 7065 6369 6669 6320 7379 6e74 656e  -specific synten
+00011d80: 7920 696e 6665 7265 6e63 652a 2a0a 6060  y inference**.``
+00011d90: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
+00011da0: 6573 2067 6666 3120 6766 6632 2028 2d2d  es gff1 gff2 (--
+00011db0: 6164 6469 7469 6f6e 616c 6766 6669 6e66  additionalgffinf
+00011dc0: 6f20 226d 524e 413b 4e61 6d65 2220 2d2d  o "mRNA;Name" --
+00011dd0: 6164 6469 7469 6f6e 616c 6766 6669 6e66  additionalgffinf
+00011de0: 6f20 2267 656e 653b 4944 2229 0a60 6060  o "gene;ID").```
+00011df0: 0a0a 466f 7220 6d75 6c74 692d 7370 6563  ..For multi-spec
+00011e00: 6965 7320 7379 6e74 656e 7920 696e 6665  ies synteny infe
+00011e10: 7265 6e63 652c 2069 6620 7573 6572 7320  rence, if users 
+00011e20: 6861 7665 2067 6666 3320 6669 6c65 7320  have gff3 files 
+00011e30: 7768 6963 6820 6861 7665 2064 6966 6665  which have diffe
+00011e40: 7265 6e74 2066 6561 7475 7265 7320 6f72  rent features or
+00011e50: 2061 7474 7269 6275 7465 7320 666f 7220   attributes for 
+00011e60: 6765 6e65 2070 6f73 6974 696f 6e20 696e  gene position in
+00011e70: 666f 726d 6174 696f 6e20 7265 7472 6965  formation retrie
+00011e80: 7661 6c2c 2074 6865 206f 7074 696f 6e20  val, the option 
+00011e90: 602d 2d61 6464 6974 696f 6e61 6c67 6666  `--additionalgff
+00011ea0: 696e 666f 6020 6361 6e20 6265 2073 6574  info` can be set
+00011eb0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+00011ec0: 6164 6469 7469 6f6e 616c 2069 6e66 6f72  additional infor
+00011ed0: 6d61 7469 6f6e 2e20 5468 6520 7265 6d61  mation. The rema
+00011ee0: 696e 696e 6720 7061 7261 6d65 7465 7220  ining parameter 
+00011ef0: 7365 7474 696e 6720 6973 2074 6865 2073  setting is the s
+00011f00: 616d 6520 6173 2074 6865 2069 6e74 7261  ame as the intra
+00011f10: 2d73 7065 6369 6669 6320 7379 6e74 656e  -specific synten
+00011f20: 7920 696e 6665 7265 6e63 652e 0a0a 2a2a  y inference...**
+00011f30: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+00011f40: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
+00011f50: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
+00011f60: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
+00011f70: 7767 6420 7379 6e20 6661 6d69 6c69 6573  wgd syn families
+00011f80: 2067 6666 3120 6766 6632 0a60 6060 0a0a   gff1 gff2.```..
+00011f90: 2323 2320 7767 6420 7669 7a0a 0a2a 2a54  ### wgd viz..**T
+00011fa0: 6865 2076 6973 7561 6c69 7a61 7469 6f6e  he visualization
+00011fb0: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
+00011fc0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00011fd0: 7469 6f6e 2061 6e64 2045 4c4d 4d20 616e  tion and ELMM an
+00011fe0: 616c 7973 6973 2a2a 0a60 6060 0a77 6764  alysis**.```.wgd
+00011ff0: 2076 697a 202d 6420 6b73 6461 7461 0a60   viz -d ksdata.`
+00012000: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+00012010: 6077 6764 2076 697a 6020 6973 206d 6169  `wgd viz` is mai
+00012020: 6e6c 7920 666f 7220 7468 6520 7075 7270  nly for the purp
+00012030: 6f73 6520 6f66 202a 4b2a 3c73 7562 3e53  ose of *K*<sub>S
+00012040: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
+00012050: 696f 6e20 616e 6420 7379 6e74 656e 7920  ion and synteny 
+00012060: 7669 7375 616c 697a 6174 696f 6e2c 2077  visualization, w
+00012070: 6974 6820 736f 6d65 206f 7074 696f 6e61  ith some optiona
+00012080: 6c20 6d69 7874 7572 6520 6d6f 6465 6c69  l mixture modeli
+00012090: 6e67 2061 6e61 6c79 7369 732e 2054 6865  ng analysis. The
+000120a0: 2062 6173 6963 2066 756e 6374 696f 6e20   basic function 
+000120b0: 6973 206a 7573 7420 746f 2070 6c6f 7420  is just to plot 
+000120c0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+000120d0: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
+000120e0: 2061 6e64 2063 6f6e 6475 6374 2061 6e20   and conduct an 
+000120f0: 454c 4d4d 2061 6e61 6c79 7369 7320 696e  ELMM analysis in
+00012100: 2073 6561 7263 6820 6f66 2070 6f74 656e   search of poten
+00012110: 7469 616c 2057 4744 2063 6f6d 706f 6e65  tial WGD compone
+00012120: 6e74 732e 2053 6f6d 6520 6b65 7920 7061  nts. Some key pa
+00012130: 7261 6d65 7465 7273 2061 6666 6563 7469  rameters affecti
+00012140: 6e67 2074 6865 2045 4c4d 4d20 7265 7375  ng the ELMM resu
+00012150: 6c74 2069 6e63 6c75 6465 2060 2d2d 7072  lt include `--pr
+00012160: 6f6d 696e 656e 6365 5f63 7574 6f66 6660  ominence_cutoff`
+00012170: 2061 6e64 2060 2d2d 7265 6c5f 6865 6967   and `--rel_heig
+00012180: 6874 602c 2077 6869 6368 2068 6176 6520  ht`, which have 
+00012190: 6265 656e 2065 7870 6c61 696e 6564 2061  been explained a
+000121a0: 6865 6164 2c20 602d 2d65 6d5f 6974 6572  head, `--em_iter
+000121b0: 6174 696f 6e73 6020 616e 6420 602d 2d65  ations` and `--e
+000121c0: 6d5f 696e 6974 6961 6c69 7a61 7469 6f6e  m_initialization
+000121d0: 7360 2064 6574 6572 6d69 6e69 6e67 2074  s` determining t
+000121e0: 6865 206d 6178 696d 756d 2069 7465 7261  he maximum itera
+000121f0: 7469 6f6e 7320 616e 6420 696e 6974 6961  tions and initia
+00012200: 6c69 7a61 7469 6f6e 7320 696e 2074 6865  lizations in the
+00012210: 2045 4d20 616c 676f 7269 7468 6d2e 0a0a   EM algorithm...
+00012220: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+00012230: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+00012240: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
+00012250: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
+00012260: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
+00012270: 6174 610a 6060 600a 0a2a 2a54 6865 2076  ata.```..**The v
+00012280: 6973 7561 6c69 7a61 7469 6f6e 206f 6620  isualization of 
+00012290: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000122a0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+000122b0: 2077 6974 6820 7261 7465 2063 6f72 7265   with rate corre
+000122c0: 6374 696f 6e2a 2a0a 6060 600a 7767 6420  ction**.```.wgd 
+000122d0: 7669 7a20 2d64 206b 7364 6174 6120 2d73  viz -d ksdata -s
+000122e0: 7020 7370 6461 7461 202d 2d66 6f63 7573  p spdata --focus
+000122f0: 3261 6c6c 2066 6f63 616c 5f73 7065 6369  2all focal_speci
+00012300: 6573 202d 2d65 7874 7261 7061 7261 6e6f  es --extraparano
+00012310: 6d65 6b73 206b 7364 6174 6120 282d 2d61  meks ksdata (--a
+00012320: 6e63 686f 7270 6f69 6e74 7320 6170 6461  nchorpoints apda
+00012330: 7461 202d 2d70 6c6f 7461 7067 6d6d 202d  ta --plotapgmm -
+00012340: 2d70 6c6f 7465 6c6d 6d29 0a60 6060 0a0a  -plotelmm).```..
+00012350: 4265 7369 6465 7320 7468 6520 6261 7369  Besides the basi
+00012360: 6320 2a4b 2a3c 7375 623e 533c 2f73 7562  c *K*<sub>S</sub
+00012370: 3e20 706c 6f74 2c20 7375 6273 7469 7475  > plot, substitu
+00012380: 7469 6f6e 2072 6174 6520 636f 7272 6563  tion rate correc
+00012390: 7469 6f6e 2063 616e 2061 6c73 6f20 6265  tion can also be
+000123a0: 2061 6368 6965 7665 6420 6769 7665 6e20   achieved given 
+000123b0: 6174 206c 6561 7374 2061 2073 7065 6369  at least a speci
+000123c0: 6573 2074 7265 6520 2876 6961 2074 6865  es tree (via the
+000123d0: 206f 7074 696f 6e20 602d 2d73 7065 6369   option `--speci
+000123e0: 6573 7472 6565 6029 2061 6e64 2061 2066  estree`) and a f
+000123f0: 6f63 616c 2073 7065 6369 6573 2028 6569  ocal species (ei
+00012400: 7468 6572 2076 6961 2074 6865 206f 7074  ther via the opt
+00012410: 696f 6e20 602d 2d66 6f63 7573 3261 6c6c  ion `--focus2all
+00012420: 6020 6f72 2076 6961 2074 6865 206f 7074  ` or via the opt
+00012430: 696f 6e20 602d 2d73 7061 6972 6020 696e  ion `--spair` in
+00012440: 2074 6865 2066 6f72 6d20 6f66 2022 2466   the form of "$f
+00012450: 6f63 616c 5f73 7065 6369 6573 3b24 666f  ocal_species;$fo
+00012460: 6361 6c5f 7370 6563 6965 7322 292e 2049  cal_species"). I
+00012470: 7427 7320 7375 6767 6573 7465 6420 7468  t's suggested th
+00012480: 6174 2074 6865 206f 7274 686f 6c6f 676f  at the orthologo
+00012490: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+000124a0: 623e 2064 6174 6120 6973 2070 726f 7669  b> data is provi
+000124b0: 6465 6420 6279 2074 6865 2060 2d2d 6461  ded by the `--da
+000124c0: 7461 6669 6c65 6020 6f70 7469 6f6e 2061  tafile` option a
+000124d0: 6e64 2074 6865 2070 6172 616c 6f67 6f75  nd the paralogou
+000124e0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+000124f0: 3e20 6461 7461 2069 7320 7072 6f76 6964  > data is provid
+00012500: 6564 2062 7920 7468 6520 602d 2d65 7874  ed by the `--ext
+00012510: 7261 7061 7261 6e6f 6d65 6b73 6020 6f70  raparanomeks` op
+00012520: 7469 6f6e 2c20 616c 7468 6f75 6768 2069  tion, although i
+00012530: 7427 7320 616c 6c6f 7765 6420 746f 206f  t's allowed to o
+00012540: 6e6c 7920 7072 6f76 6964 6520 2a4b 2a3c  nly provide *K*<
+00012550: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
+00012560: 2076 6961 2074 6865 2060 2d2d 6461 7461   via the `--data
+00012570: 6669 6c65 6020 6f70 7469 6f6e 2061 6e64  file` option and
+00012580: 2064 6570 6f73 6974 2062 6f74 6820 6f72   deposit both or
+00012590: 7468 6f6c 6f67 6f75 7320 616e 6420 7061  thologous and pa
+000125a0: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
+000125b0: 3e53 3c2f 7375 623e 2064 6174 6120 7468  >S</sub> data th
+000125c0: 6572 656f 6e2e 2054 6865 7265 2061 7265  ereon. There are
+000125d0: 2074 776f 2074 7970 6573 206f 6620 6d69   two types of mi
+000125e0: 7865 6420 706c 6f74 7320 2874 6865 2022  xed plots (the "
+000125f0: 6d69 7865 6422 2068 6572 6520 7265 6665  mixed" here refe
+00012600: 7273 2074 6f20 6d69 7865 6420 6f72 7468  rs to mixed orth
+00012610: 6f6c 6f67 6f75 7320 616e 6420 7061 7261  ologous and para
+00012620: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00012630: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
+00012640: 696f 6e73 292c 206f 6e65 206f 6620 7768  ions), one of wh
+00012650: 6963 6820 6973 2073 696d 696c 6172 2074  ich is similar t
+00012660: 6f20 7768 6174 2060 6b73 7261 7465 7360  o what `ksrates`
+00012670: 2070 6c6f 7473 2c20 7768 696c 6520 7468   plots, while th
+00012680: 6520 6f74 6865 7220 6f66 2077 6869 6368  e other of which
+00012690: 2069 7320 6c69 6b65 2074 6865 2063 6f6e   is like the con
+000126a0: 7665 6e74 696f 6e61 6c20 2a4b 2a3c 7375  ventional *K*<su
+000126b0: 623e 533c 2f73 7562 3e20 706c 6f74 2074  b>S</sub> plot t
+000126c0: 6861 7420 626f 7468 2074 6865 206f 7269  hat both the ori
+000126d0: 6769 6e61 6c20 6f72 7468 6f6c 6f67 6f75  ginal orthologou
+000126e0: 7320 616e 6420 7061 7261 6c6f 676f 7573  s and paralogous
+000126f0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00012700: 2064 6973 7472 6962 7574 696f 6e73 2061   distributions a
+00012710: 7265 2074 7275 7468 6675 6c6c 7920 706c  re truthfully pl
+00012720: 6f74 7465 6420 696e 7374 6561 6420 6f66  otted instead of
+00012730: 206a 7573 7420 6265 696e 6720 7265 7072   just being repr
+00012740: 6573 656e 7465 6420 6279 2073 6f6d 6520  esented by some 
+00012750: 7665 7274 6963 616c 206c 696e 6573 2e20  vertical lines. 
+00012760: 5765 2073 7567 6765 7374 2075 7365 7273  We suggest users
+00012770: 2074 6f20 6164 6f70 7420 7468 6520 606b   to adopt the `k
+00012780: 7372 6174 6573 602d 6c69 6b65 2070 6c6f  srates`-like plo
+00012790: 7473 2c20 7768 6963 6820 6973 2074 6865  ts, which is the
+000127a0: 2064 6566 6175 6c74 2e20 4f74 6865 7277   default. Otherw
+000127b0: 6973 6520 7468 6520 6f70 7469 6f6e 2060  ise the option `
+000127c0: 2d2d 636c 6173 7369 6360 2077 696c 6c20  --classic` will 
+000127d0: 6265 206e 6565 6465 6420 746f 2073 6574  be needed to set
+000127e0: 2e20 4578 7472 6120 6d69 7874 7572 6520  . Extra mixture 
+000127f0: 6d6f 6465 6c69 6e67 2061 6e61 6c79 7369  modeling analysi
+00012800: 7320 6361 6e20 6265 2069 6e69 7469 6174  s can be initiat
+00012810: 6564 2076 6961 2074 6865 206f 7074 696f  ed via the optio
+00012820: 6e20 602d 2d70 6c6f 7465 6c6d 6d60 2061  n `--plotelmm` a
+00012830: 6e64 2060 2d2d 706c 6f74 6170 676d 6d60  nd `--plotapgmm`
+00012840: 2077 6974 6820 7468 6520 616e 6368 6f72   with the anchor
+00012850: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
+00012860: 2070 726f 7669 6465 6420 6279 2074 6865   provided by the
+00012870: 2060 2d2d 616e 6368 6f72 706f 696e 7473   `--anchorpoints
+00012880: 6020 6f70 7469 6f6e 2e0a 0a2a 2a41 2073  ` option...**A s
+00012890: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+000128a0: 6720 7275 6e20 6361 6e20 7573 6520 636f  g run can use co
+000128b0: 6d6d 616e 6420 7369 6d70 6c79 2061 7320  mmand simply as 
+000128c0: 6265 6c6f 772a 2a0a 0a60 6060 0a77 6764  below**..```.wgd
+000128d0: 2076 697a 202d 6420 6b73 6461 7461 202d   viz -d ksdata -
+000128e0: 7370 2073 7064 6174 6120 2d2d 666f 6375  sp spdata --focu
+000128f0: 7332 616c 6c20 666f 6361 6c5f 7370 6563  s2all focal_spec
+00012900: 6965 7320 2d2d 6578 7472 6170 6172 616e  ies --extraparan
+00012910: 6f6d 656b 7320 6b73 6461 7461 0a60 6060  omeks ksdata.```
+00012920: 0a0a 2a2a 5468 6520 7669 7375 616c 697a  ..**The visualiz
+00012930: 6174 696f 6e20 6f66 2073 796e 7465 6e79  ation of synteny
+00012940: 2a2a 0a60 6060 0a77 6764 2076 697a 202d  **.```.wgd viz -
+00012950: 6170 2061 7064 6174 6120 2d73 6d20 736d  ap apdata -sm sm
+00012960: 6461 7461 202d 6d74 206d 7464 6174 6120  data -mt mtdata 
+00012970: 2d67 7420 6774 6461 7461 202d 2d70 6c6f  -gt gtdata --plo
+00012980: 7473 796e 2028 2d2d 6d69 6e6c 656e 202d  tsyn (--minlen -
+00012990: 3120 2d2d 6d69 6e73 6567 6c65 6e20 3130  1 --minseglen 10
+000129a0: 3030 3020 2d2d 6d69 6e67 656e 656e 756d  000 --mingenenum
+000129b0: 2033 3029 0a60 6060 0a0a 436f 6d70 6172   30).```..Compar
+000129c0: 6564 2074 6f20 7468 6520 6f72 6967 696e  ed to the origin
+000129d0: 616c 2060 7767 6460 2c20 7468 6520 6077  al `wgd`, the `w
+000129e0: 6764 2076 697a 6020 7072 6f67 7261 6d20  gd viz` program 
+000129f0: 6164 6465 6420 7379 6e74 656e 7920 7669  added synteny vi
+00012a00: 7375 616c 697a 6174 696f 6e20 7069 7065  sualization pipe
+00012a10: 6c69 6e65 2e20 5573 6572 7320 6e65 6564  line. Users need
+00012a20: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+00012a30: 666c 6167 206f 7074 696f 6e20 602d 2d70  flag option `--p
+00012a40: 6c6f 7473 796e 6020 746f 2069 6e69 7469  lotsyn` to initi
+00012a50: 6174 6520 7468 6973 2070 6172 7420 6f66  ate this part of
+00012a60: 2070 6970 656c 696e 652e 2054 6869 7320   pipeline. This 
+00012a70: 7374 6570 2061 7373 756d 6573 2074 6861  step assumes tha
+00012a80: 7420 7573 6572 7320 6861 7665 206f 6274  t users have obt
+00012a90: 6169 6e65 6420 616c 7265 6164 7920 7468  ained already th
+00012aa0: 6520 7379 6e74 656e 6963 2072 6573 756c  e syntenic resul
+00012ab0: 7473 2066 726f 6d20 6069 2d61 6468 6f72  ts from `i-adhor
+00012ac0: 6560 2061 6e64 2075 7365 7320 7468 6f73  e` and uses thos
+00012ad0: 6520 7265 7375 6c74 2066 696c 6573 2074  e result files t
+00012ae0: 6f20 7265 616c 697a 6520 7468 6520 7379  o realize the sy
+00012af0: 6e74 656e 7920 7669 7375 616c 697a 6174  nteny visualizat
+00012b00: 696f 6e2e 2053 696d 6c69 6172 2074 6f20  ion. Simliar to 
+00012b10: 6077 6764 2073 796e 602c 2074 6865 2065  `wgd syn`, the e
+00012b20: 7874 7261 202a 4b2a 3c73 7562 3e53 3c2f  xtra *K*<sub>S</
+00012b30: 7375 623e 2064 6174 6120 6669 6c65 2063  sub> data file c
+00012b40: 616e 2062 6520 7472 616e 736d 6974 7465  an be transmitte
+00012b50: 6420 7669 6120 7468 6520 6f70 7469 6f6e  d via the option
+00012b60: 2060 2d2d 6461 7461 6669 6c65 6020 2869   `--datafile` (i
+00012b70: 6e73 7465 6164 206f 6620 602d 2d65 7874  nstead of `--ext
+00012b80: 7261 7061 7261 6e6f 6d65 6b73 6020 6f70  raparanomeks` op
+00012b90: 7469 6f6e 292e 2042 6173 6963 616c 6c79  tion). Basically
+00012ba0: 2c20 7468 6520 7379 6e74 656e 6963 2072  , the syntenic r
+00012bb0: 6573 756c 7420 6669 6c65 7320 7265 7175  esult files requ
+00012bc0: 6972 6564 2061 7265 2074 6865 2061 6e63  ired are the anc
+00012bd0: 686f 7220 706f 696e 7473 2064 6174 6166  hor points dataf
+00012be0: 696c 652c 206d 756c 7469 706c 6963 6f6e  ile, multiplicon
+00012bf0: 7320 6461 7461 6669 6c65 2c20 6765 6e65  s datafile, gene
+00012c00: 2074 6162 6c65 2064 6174 6166 696c 6520   table datafile 
+00012c10: 2861 7574 6f6d 6174 656c 7920 7072 6f64  (automately prod
+00012c20: 7563 6564 2062 7920 6077 6764 2073 796e  uced by `wgd syn
+00012c30: 6029 2c20 616e 6420 7365 676d 656e 7473  `), and segments
+00012c40: 2064 6174 6166 696c 652e 0a0a 2a2a 4120   datafile...**A 
+00012c50: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
+00012c60: 6e67 2072 756e 2063 616e 2075 7365 2063  ng run can use c
+00012c70: 6f6d 6d61 6e64 2073 696d 706c 7920 6173  ommand simply as
+00012c80: 2062 656c 6f77 2a2a 0a0a 6060 600a 7767   below**..```.wg
+00012c90: 6420 7669 7a20 2d61 7020 6170 6461 7461  d viz -ap apdata
+00012ca0: 202d 736d 2073 6d64 6174 6120 2d6d 7420   -sm smdata -mt 
+00012cb0: 6d74 6461 7461 202d 6774 2067 7464 6174  mtdata -gt gtdat
+00012cc0: 6120 2d2d 706c 6f74 7379 6e0a 6060 600a  a --plotsyn.```.
+00012cd0: 0a23 2323 2077 6764 206d 6978 0a0a 2a2a  .### wgd mix..**
+00012ce0: 5468 6520 6d69 7874 7572 6520 6d6f 6465  The mixture mode
+00012cf0: 6c20 636c 7573 7465 7269 6e67 2061 6e61  l clustering ana
+00012d00: 6c79 7369 7320 6f66 202a 4b2a 3c73 7562  lysis of *K*<sub
+00012d10: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
+00012d20: 7472 6962 7574 696f 6e2a 2a0a 6060 600a  tribution**.```.
+00012d30: 7767 6420 6d69 7820 6b73 6461 7461 2028  wgd mix ksdata (
+00012d40: 2d2d 6e5f 696e 6974 2032 3030 202d 2d6d  --n_init 200 --m
+00012d50: 6178 5f69 7465 7220 3230 3020 2d2d 6b73  ax_iter 200 --ks
+00012d60: 5f72 616e 6765 2030 2035 202d 2d66 696c  _range 0 5 --fil
+00012d70: 7465 7273 2033 3030 202d 2d62 696e 7320  ters 300 --bins 
+00012d80: 3530 202d 2d63 6f6d 706f 6e65 6e74 7320  50 --components 
+00012d90: 3120 3420 2d2d 6761 6d6d 6120 302e 3030  1 4 --gamma 0.00
+00012da0: 3129 0a60 6060 0a0a 5468 6973 2070 6172  1).```..This par
+00012db0: 7420 6f66 2047 6175 7373 6961 6e20 6d69  t of Gaussian mi
+00012dc0: 7874 7572 6520 6d6f 6465 6c69 6e67 2028  xture modeling (
+00012dd0: 474d 4d29 2061 6e61 6c79 7369 7320 6973  GMM) analysis is
+00012de0: 2069 6e68 6572 6974 6564 2066 726f 6d20   inherited from 
+00012df0: 7468 6520 6f72 6967 696e 616c 2060 7767  the original `wg
+00012e00: 6460 2070 726f 6772 616d 2c20 6275 7420  d` program, but 
+00012e10: 7772 6974 6573 2061 6464 6974 696f 6e61  writes additiona
+00012e20: 6c6c 7920 7468 6520 7072 6f62 6162 696c  lly the probabil
+00012e30: 6974 7920 6f66 2065 6163 6820 2a4b 2a3c  ity of each *K*<
+00012e40: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+00012e50: 6520 696e 746f 2074 6865 2066 696e 616c  e into the final
+00012e60: 2064 6174 6166 7261 6d65 2e20 4261 7369   dataframe. Basi
+00012e70: 6361 6c6c 792c 2075 7365 7273 206e 6565  cally, users nee
+00012e80: 6420 746f 2070 726f 7669 6465 2077 6974  d to provide wit
+00012e90: 6820 6120 286e 6f72 6d61 6c6c 7920 6672  h a (normally fr
+00012ea0: 6f6d 2077 686f 6c65 2d70 6172 616e 6f6d  om whole-paranom
+00012eb0: 6520 6f72 2061 6e63 686f 722d 7061 6972  e or anchor-pair
+00012ec0: 7329 202a 4b2a 3c73 7562 3e53 3c2f 7375  s) *K*<sub>S</su
+00012ed0: 623e 2064 6174 6166 696c 6520 616e 6420  b> datafile and 
+00012ee0: 7468 6520 474d 4d20 616e 616c 7973 6973  the GMM analysis
+00012ef0: 2077 696c 6c20 6265 2063 6f6e 6475 6374   will be conduct
+00012f00: 6564 2075 706f 6e20 7468 6520 6461 7461  ed upon the data
+00012f10: 6669 6c65 2e20 536f 6d65 2070 6172 616d  file. Some param
+00012f20: 6574 6572 7320 6361 6e20 6166 6665 6374  eters can affect
+00012f30: 2074 6865 2072 6573 756c 7473 2c20 696e   the results, in
+00012f40: 636c 7564 696e 6720 602d 2d6e 5f69 6e69  cluding `--n_ini
+00012f50: 7460 2c20 7768 6963 6820 7365 7473 2074  t`, which sets t
+00012f60: 6865 206e 756d 6265 7220 6f66 206b 2d6d  he number of k-m
+00012f70: 6561 6e73 2069 6e69 7469 616c 697a 6174  eans initializat
+00012f80: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
+00012f90: 3029 2c20 602d 2d6d 6178 5f69 7465 7260  0), `--max_iter`
+00012fa0: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
+00012fb0: 206d 6178 696d 756d 206e 756d 6265 7220   maximum number 
+00012fc0: 6f66 2069 7465 7261 7469 6f6e 7320 2864  of iterations (d
+00012fd0: 6566 6175 6c74 2032 3030 292c 2060 2d2d  efault 200), `--
+00012fe0: 6d65 7468 6f64 602c 2077 6869 6368 2064  method`, which d
+00012ff0: 6574 6572 6d69 6e65 7320 7768 6963 6820  etermines which 
+00013000: 636c 7573 7465 7269 6e67 206d 6574 686f  clustering metho
+00013010: 6420 746f 2075 7365 2028 6465 6661 756c  d to use (defaul
+00013020: 7420 676d 6d29 2c20 602d 2d67 616d 6d61  t gmm), `--gamma
+00013030: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
+00013040: 6520 6761 6d6d 6120 7061 7261 6d65 7465  e gamma paramete
+00013050: 7220 666f 7220 7468 6520 6267 6d6d 206d  r for the bgmm m
+00013060: 6f64 656c 2028 6465 6661 756c 7420 302e  odel (default 0.
+00013070: 3030 3129 2c20 602d 2d63 6f6d 706f 6e65  001), `--compone
+00013080: 6e74 7360 2c20 7768 6963 6820 7365 7473  nts`, which sets
+00013090: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
+000130a0: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
+000130b0: 6f6e 656e 7473 2074 6f20 6669 7420 2864  onents to fit (d
+000130c0: 6566 6175 6c74 2031 2034 292c 2061 6e64  efault 1 4), and
+000130d0: 2074 6865 2064 6174 6120 6669 6c74 6572   the data filter
+000130e0: 696e 6720 7061 7261 6d65 7465 7273 2060  ing parameters `
+000130f0: 2d2d 6669 6c74 6572 7360 2077 6869 6368  --filters` which
+00013100: 2066 696c 7465 7273 2064 6174 6120 6261   filters data ba
+00013110: 7365 6420 6f6e 2061 6c69 676e 6d65 6e74  sed on alignment
+00013120: 206c 656e 6774 682c 2060 2d2d 6b73 5f72   length, `--ks_r
+00013130: 616e 6765 6020 7768 6963 6820 6669 6c74  ange` which filt
+00013140: 6572 7320 6461 7461 2062 6173 6564 206f  ers data based o
+00013150: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+00013160: 3e20 7661 6c75 6573 2061 6e64 2074 6865  > values and the
+00013170: 2070 6172 616d 6574 6572 2060 2d2d 6269   parameter `--bi
+00013180: 6e73 6020 7768 6963 6820 7365 7473 2074  ns` which sets t
+00013190: 6865 206e 756d 6265 7220 6f66 2062 696e  he number of bin
+000131a0: 7320 696e 202a 4b2a 3c73 7562 3e53 3c2f  s in *K*<sub>S</
+000131b0: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
+000131c0: 6e20 2864 6566 6175 6c74 2035 3029 2e0a  n (default 50)..
+000131d0: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+000131e0: 7461 7274 696e 6720 7275 6e20 6361 6e20  tarting run can 
+000131f0: 7573 6520 636f 6d6d 616e 6420 7369 6d70  use command simp
+00013200: 6c79 2061 7320 6265 6c6f 772a 2a0a 0a60  ly as below**..`
+00013210: 6060 0a77 6764 206d 6978 206b 7364 6174  ``.wgd mix ksdat
+00013220: 610a 6060 600a 0a23 2323 2077 6764 2070  a.```..### wgd p
+00013230: 6561 6b0a 0a2a 2a54 6865 2073 6561 7263  eak..**The searc
+00013240: 6820 6f66 2063 7265 6469 6162 6c65 202a  h of crediable *
+00013250: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2072  K*<sub>S</sub> r
+00013260: 616e 6765 2075 7365 6420 696e 2057 4744  ange used in WGD
+00013270: 2064 6174 696e 672a 2a0a 6060 600a 7767   dating**.```.wg
+00013280: 6420 7065 616b 206b 7364 6174 6120 2d61  d peak ksdata -a
+00013290: 7020 6170 6461 7461 202d 736d 2073 6d64  p apdata -sm smd
+000132a0: 6174 6120 2d6c 6520 6c65 6461 7461 202d  ata -le ledata -
+000132b0: 6d70 206d 7064 6174 6120 2d2d 6865 7572  mp mpdata --heur
+000132c0: 6973 7469 6320 282d 2d61 6c69 676e 6669  istic (--alignfi
+000132d0: 6c74 6572 2030 2e30 2030 2030 2e30 202d  lter 0.0 0 0.0 -
+000132e0: 2d6b 7372 616e 6765 2030 2035 202d 2d62  -ksrange 0 5 --b
+000132f0: 696e 5f77 6964 7468 2030 2e31 202d 2d67  in_width 0.1 --g
+00013300: 7569 6465 2073 6567 6d65 6e74 202d 2d70  uide segment --p
+00013310: 726f 6d69 6e65 6e63 655f 6375 746f 6666  rominence_cutoff
+00013320: 2030 2e31 202d 2d72 656c 5f68 6569 6768   0.1 --rel_heigh
+00013330: 7420 302e 3420 2d2d 6369 2039 3520 2d2d  t 0.4 --ci 95 --
+00013340: 6864 7220 3935 202d 2d6b 7363 7574 6f66  hdr 95 --kscutof
+00013350: 6620 3529 0a60 6060 0a0a 4173 206d 656e  f 5).```..As men
+00013360: 7469 6f6e 6564 2070 7265 7669 6f75 736c  tioned previousl
+00013370: 792c 2061 2068 6575 7269 7374 6963 206d  y, a heuristic m
+00013380: 6574 686f 6420 616e 6420 6120 636f 6c6c  ethod and a coll
+00013390: 696e 6561 7220 7365 676d 656e 7473 2d67  inear segments-g
+000133a0: 7569 6465 6420 616e 6368 6f72 2070 6169  uided anchor pai
+000133b0: 7220 636c 7573 7465 7269 6e67 2066 6f72  r clustering for
+000133c0: 2074 6865 2073 6561 7263 6820 6f66 2063   the search of c
+000133d0: 7265 6469 6162 6c65 202a 4b2a 3c73 7562  rediable *K*<sub
+000133e0: 3e53 3c2f 7375 623e 2072 616e 6765 2075  >S</sub> range u
+000133f0: 7365 6420 696e 2057 4744 2064 6174 696e  sed in WGD datin
+00013400: 6720 6172 6520 696d 706c 656d 656e 7465  g are implemente
+00013410: 6420 696e 2060 7767 6420 7632 602e 2055  d in `wgd v2`. U
+00013420: 7365 7273 206e 6565 6420 746f 2070 726f  sers need to pro
+00013430: 7669 6465 2074 6865 2061 6e63 686f 7220  vide the anchor 
+00013440: 706f 696e 7473 2c20 7365 676d 656e 7473  points, segments
+00013450: 2c20 6c69 7374 7365 676d 656e 7473 2c20  , listsegments, 
+00013460: 6d75 6c74 6970 6c69 636f 6e70 6169 7273  multipliconpairs
+00013470: 2064 6174 6166 696c 6520 6672 6f6d 2060   datafile from `
+00013480: 692d 6164 686f 7265 6020 746f 2061 6368  i-adhore` to ach
+00013490: 6965 7665 2074 6865 2063 6c75 7374 6572  ieve the cluster
+000134a0: 696e 6720 6675 6e63 7469 6f6e 2e20 536f  ing function. So
+000134b0: 6d65 2070 6172 616d 6574 6572 7320 7468  me parameters th
+000134c0: 6174 2063 616e 2069 6d70 6163 7420 7468  at can impact th
+000134d0: 6520 7265 7375 6c74 7320 696e 636c 7564  e results includ
+000134e0: 6520 602d 2d61 6c69 676e 6669 6c74 6572  e `--alignfilter
+000134f0: 602c 2077 6869 6368 2066 696c 7465 7273  `, which filters
+00013500: 2074 6865 2064 6174 6120 6261 7365 6420   the data based 
+00013510: 6f6e 2061 6c69 676e 6d65 6e74 2069 6465  on alignment ide
+00013520: 6e74 6974 792c 206c 656e 6774 6820 616e  ntity, length an
+00013530: 6420 636f 7665 7261 6765 2c20 602d 2d6b  d coverage, `--k
+00013540: 7372 616e 6765 602c 2077 6869 6368 2073  srange`, which s
+00013550: 6574 7320 7468 6520 7261 6e67 6520 6f66  ets the range of
+00013560: 204b 7320 746f 2062 6520 616e 616c 797a   Ks to be analyz
+00013570: 6564 2c20 602d 2d62 696e 5f77 6964 7468  ed, `--bin_width
+00013580: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
+00013590: 6520 6261 6e64 7769 6474 6820 6f66 202a  e bandwidth of *
+000135a0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+000135b0: 6973 7472 6962 7574 696f 6e2c 2060 2d2d  istribution, `--
+000135c0: 7765 6967 6874 735f 6f75 746c 6965 7273  weights_outliers
+000135d0: 5f69 6e63 6c75 6465 6460 2077 6869 6368  _included` which
+000135e0: 2064 6574 6572 6d69 6e65 7320 7768 6574   determines whet
+000135f0: 6865 7220 746f 2069 6e63 6c75 6465 202a  her to include *
+00013600: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206f  K*<sub>S</sub> o
+00013610: 7574 6c69 6572 7320 2877 686f 7365 2076  utliers (whose v
+00013620: 616c 7565 2069 7320 6f76 6572 2035 2920  alue is over 5) 
+00013630: 696e 2061 6e61 6c79 7369 732c 2060 2d2d  in analysis, `--
+00013640: 6d65 7468 6f64 6020 7768 6963 6820 6465  method` which de
+00013650: 7465 726d 696e 6573 2077 6869 6368 2063  termines which c
+00013660: 6c75 7374 6572 696e 6720 6d65 7468 6f64  lustering method
+00013670: 2074 6f20 7573 6520 2864 6566 6175 6c74   to use (default
+00013680: 2067 6d6d 292c 2060 2d2d 7365 6564 6020   gmm), `--seed` 
+00013690: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
+000136a0: 616e 646f 6d20 7365 6564 2067 6976 656e  andom seed given
+000136b0: 2074 6f20 696e 6974 6961 6c69 7a61 7469   to initializati
+000136c0: 6f6e 2028 6465 6661 756c 7420 3233 3532  on (default 2352
+000136d0: 3839 3029 2c20 602d 2d6e 5f69 6e69 7460  890), `--n_init`
+000136e0: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
+000136f0: 206e 756d 6265 7220 6f66 206b 2d6d 6561   number of k-mea
+00013700: 6e73 2069 6e69 7469 616c 697a 6174 696f  ns initializatio
+00013710: 6e73 2028 6465 6661 756c 7420 3230 3029  ns (default 200)
+00013720: 2c20 602d 2d65 6d5f 6974 6572 602c 2077  , `--em_iter`, w
+00013730: 6869 6368 2073 6574 7320 7468 6520 6d61  hich sets the ma
+00013740: 7869 6d75 6d20 6e75 6d62 6572 206f 6620  ximum number of 
+00013750: 6974 6572 6174 696f 6e73 2028 6465 6661  iterations (defa
+00013760: 756c 7420 3230 3029 2c20 602d 2d67 616d  ult 200), `--gam
+00013770: 6d61 602c 2077 6869 6368 2073 6574 7320  ma`, which sets 
+00013780: 7468 6520 6761 6d6d 6120 7061 7261 6d65  the gamma parame
+00013790: 7465 7220 666f 7220 7468 6520 6267 6d6d  ter for the bgmm
+000137a0: 206d 6f64 656c 2028 6465 6661 756c 7420   model (default 
+000137b0: 302e 3030 3129 2c20 602d 2d63 6f6d 706f  0.001), `--compo
+000137c0: 6e65 6e74 7360 2c20 7768 6963 6820 7365  nents`, which se
+000137d0: 7473 2074 6865 2072 616e 6765 206f 6620  ts the range of 
+000137e0: 7468 6520 6e75 6d62 6572 206f 6620 636f  the number of co
+000137f0: 6d70 6f6e 656e 7473 2074 6f20 6669 7420  mponents to fit 
+00013800: 2864 6566 6175 6c74 2031 2034 292c 2060  (default 1 4), `
+00013810: 2d2d 7765 6967 6874 6564 6020 7768 6963  --weighted` whic
+00013820: 6820 6465 7465 726d 696e 6573 2077 6865  h determines whe
+00013830: 7468 6572 2074 6f20 7573 6520 6e6f 6465  ther to use node
+00013840: 2d77 6569 6768 7465 6420 6d65 7468 6f64  -weighted method
+00013850: 2066 6f72 2064 652d 7265 6475 6e64 616e   for de-redundan
+00013860: 6379 2c20 602d 2d67 7569 6465 6020 7768  cy, `--guide` wh
+00013870: 6963 6820 6465 7465 726d 696e 6573 2077  ich determines w
+00013880: 6869 6368 2072 6567 696d 6520 7265 7369  hich regime resi
+00013890: 6469 6e67 2061 6e63 686f 7273 2074 6f20  ding anchors to 
+000138a0: 6265 2075 7365 6420 2864 6566 6175 6c74  be used (default
+000138b0: 2053 6567 6d65 6e74 292c 2060 2d2d 7072   Segment), `--pr
+000138c0: 6f6d 696e 656e 6365 5f63 7574 6f66 6660  ominence_cutoff`
+000138d0: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
+000138e0: 7072 6f6d 696e 656e 6365 2063 7574 6f66  prominence cutof
+000138f0: 6620 6f66 2061 6363 6570 7461 626c 6520  f of acceptable 
+00013900: 7065 616b 7320 696e 2070 6561 6b20 6669  peaks in peak fi
+00013910: 6e64 696e 6720 7072 6f63 6573 732c 2060  nding process, `
+00013920: 2d2d 7265 6c5f 6865 6967 6874 6020 7768  --rel_height` wh
+00013930: 6963 6820 7365 7473 2074 6865 2072 656c  ich sets the rel
+00013940: 6174 6976 6520 6865 6967 6874 2061 7420  ative height at 
+00013950: 7768 6963 6820 7468 6520 7065 616b 2077  which the peak w
+00013960: 6964 7468 2069 7320 6d65 6173 7572 6564  idth is measured
+00013970: 2c20 602d 2d6b 7374 6f64 6174 6560 2077  , `--kstodate` w
+00013980: 6869 6368 206d 616e 7561 6c6c 7920 7365  hich manually se
+00013990: 7473 2074 6865 2072 616e 6765 206f 6620  ts the range of 
+000139a0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000139b0: 746f 2062 6520 6461 7465 6420 696e 2068  to be dated in h
+000139c0: 6575 7269 7374 6963 2073 6561 7263 6820  euristic search 
+000139d0: 616e 6420 6e65 6564 7320 746f 2062 6520  and needs to be 
+000139e0: 636f 2d73 6574 2077 6974 6820 6f70 7469  co-set with opti
+000139f0: 6f6e 2060 2d2d 6d61 6e75 616c 7365 7460  on `--manualset`
+00013a00: 2c20 602d 2d78 6c69 6d60 2061 6e64 2060  , `--xlim` and `
+00013a10: 2d2d 796c 696d 6020 6465 7465 726d 696e  --ylim` determin
+00013a20: 696e 6720 7468 6520 7820 616e 6420 7920  ing the x and y 
+00013a30: 6178 6973 206c 696d 6974 206f 6620 474d  axis limit of GM
+00013a40: 4d20 2a4b 2a3c 7375 623e 533c 2f73 7562  M *K*<sub>S</sub
+00013a50: 3e20 6469 7374 7269 6275 7469 6f6e 2c20  > distribution, 
+00013a60: 602d 2d63 6960 2073 6574 7469 6e67 2074  `--ci` setting t
+00013a70: 6865 2063 6f6e 6669 6465 6e63 6520 6c65  he confidence le
+00013a80: 7665 6c20 6f66 206c 6f67 2d6e 6f72 6d61  vel of log-norma
+00013a90: 6c20 6469 7374 7269 6275 7469 6f6e 2074  l distribution t
+00013aa0: 6f20 6461 7465 2028 6465 6661 756c 7420  o date (default 
+00013ab0: 3935 292c 2060 2d2d 6864 7260 2073 6574  95), `--hdr` set
+00013ac0: 7469 6e67 2074 6865 2068 6967 6865 7374  ting the highest
+00013ad0: 2064 656e 7369 7479 2072 6567 696f 6e20   density region 
+00013ae0: 2848 4452 2920 6170 706c 6965 6420 696e  (HDR) applied in
+00013af0: 2074 6865 2073 6567 6d65 6e74 2d67 7569   the segment-gui
+00013b00: 6465 6420 616e 6368 6f72 2070 6169 7220  ded anchor pair 
+00013b10: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00013b20: 6469 7374 7269 6275 7469 6f6e 2028 6465  distribution (de
+00013b30: 6661 756c 7420 3935 292c 2060 2d2d 6865  fault 95), `--he
+00013b40: 7572 6973 7469 6360 2064 6574 6572 6d69  uristic` determi
+00013b50: 6e69 6e67 2077 6865 7468 6572 2074 6f20  ning whether to 
+00013b60: 696e 6974 6961 7465 2068 6575 7269 7374  initiate heurist
+00013b70: 6963 206d 6574 686f 6420 6f66 2064 6566  ic method of def
+00013b80: 696e 696e 6720 4349 2066 6f72 2064 6174  ining CI for dat
+00013b90: 696e 672c 2060 2d2d 6b73 6375 746f 6666  ing, `--kscutoff
+00013ba0: 6020 7365 7474 696e 6720 7468 6520 2a4b  ` setting the *K
+00013bb0: 2a3c 7375 623e 533c 2f73 7562 3e20 7361  *<sub>S</sub> sa
+00013bc0: 7475 7261 7469 6f6e 2063 7574 6f66 6620  turation cutoff 
+00013bd0: 696e 2064 6174 696e 6720 2864 6566 6175  in dating (defau
+00013be0: 6c74 2035 292e 0a0a 2a2a 4120 7375 6767  lt 5)...**A sugg
+00013bf0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+00013c00: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+00013c10: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
+00013c20: 6f77 2a2a 0a0a 6060 600a 7767 6420 7065  ow**..```.wgd pe
+00013c30: 616b 206b 7364 6174 6120 2d61 7020 6170  ak ksdata -ap ap
+00013c40: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
+00013c50: 2d6c 6520 6c65 6461 7461 202d 6d70 206d  -le ledata -mp m
+00013c60: 7064 6174 6120 2d2d 6865 7572 6973 7469  pdata --heuristi
+00013c70: 630a 6060 600a 0a23 2323 2077 6764 2066  c.```..### wgd f
+00013c80: 6f63 7573 0a0a 2a2a 5468 6520 636f 6e63  ocus..**The conc
+00013c90: 6174 656e 6174 696f 6e2d 6261 7365 642f  atenation-based/
+00013ca0: 636f 616c 6573 6365 6e63 652d 6261 7365  coalescence-base
+00013cb0: 6420 7068 796c 6f67 656e 6574 6963 2069  d phylogenetic i
+00013cc0: 6e66 6572 656e 6365 2a2a 0a60 6060 0a77  nference**.```.w
+00013cd0: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
+00013ce0: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
+00013cf0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+00013d00: 282d 2d63 6f6e 6361 7465 6e61 7469 6f6e  (--concatenation
+00013d10: 2920 282d 2d63 6f61 6c65 7363 656e 6365  ) (--coalescence
+00013d20: 2920 282d 7472 6565 2027 6661 7374 7472  ) (-tree 'fasttr
+00013d30: 6565 2729 2028 2d74 7320 272d 6661 7374  ee') (-ts '-fast
+00013d40: 6573 7427 2920 282d 6e20 3429 2028 2d2d  est') (-n 4) (--
+00013d50: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
+00013d60: 2028 2d6f 2077 6764 5f66 6f63 7573 2920   (-o wgd_focus) 
+00013d70: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
+00013d80: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+00013d90: 6d20 6077 6764 2066 6f63 7573 6020 696d  m `wgd focus` im
+00013da0: 706c 656d 656e 7465 6420 7477 6f20 6261  plemented two ba
+00013db0: 7369 6320 7068 796c 6f67 656e 6574 6963  sic phylogenetic
+00013dc0: 2069 6e66 6572 656e 6365 206d 6574 686f   inference metho
+00013dd0: 6473 2c20 692e 652e 2c20 636f 6e63 6174  ds, i.e., concat
+00013de0: 656e 6174 696f 6e2d 6261 7365 6420 616e  enation-based an
+00013df0: 6420 636f 616c 6573 6365 6e63 652d 6261  d coalescence-ba
+00013e00: 7365 6420 6d65 7468 6f64 732e 2054 6f20  sed methods. To 
+00013e10: 696e 6974 6961 7465 2074 6865 7365 2061  initiate these a
+00013e20: 6e61 6c79 7369 732c 2075 7365 7273 206e  nalysis, users n
+00013e30: 6565 6420 746f 2073 6574 2074 6865 2066  eed to set the f
+00013e40: 6c61 6720 6f70 7469 6f6e 2060 2d2d 636f  lag option `--co
+00013e50: 6e63 6174 656e 6174 696f 6e60 206f 7220  ncatenation` or 
+00013e60: 602d 2d63 6f61 6c65 7363 656e 6365 602e  `--coalescence`.
+00013e70: 2054 6865 2063 6f6e 6361 7465 6e61 7469   The concatenati
+00013e80: 6f6e 2d62 6173 6564 206d 6574 686f 6420  on-based method 
+00013e90: 696e 636c 7564 6573 2061 2066 6577 206d  includes a few m
+00013ea0: 616a 6f72 2073 7465 7073 2c20 692e 652e  ajor steps, i.e.
+00013eb0: 2c20 7468 6520 6d75 6c74 6970 6c65 2073  , the multiple s
+00013ec0: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
+00013ed0: 7420 284d 5341 2920 6f66 2065 6163 6820  t (MSA) of each 
+00013ee0: 6765 6e65 2066 616d 696c 792c 2074 6865  gene family, the
+00013ef0: 2063 6f6e 6361 7465 6e61 7469 6f6e 206f   concatenation o
+00013f00: 6620 616c 6c20 6765 6e65 2066 616d 696c  f all gene famil
+00013f10: 6965 7320 616e 6420 7468 656e 2074 6865  ies and then the
+00013f20: 2067 656e 6520 7472 6565 2028 616c 736f   gene tree (also
+00013f30: 2073 7065 6369 6573 2074 7265 6520 696e   species tree in
+00013f40: 2074 6869 7320 6361 7365 2920 696e 6665   this case) infe
+00013f50: 7265 6e63 652e 2054 6865 2063 6f61 6c65  rence. The coale
+00013f60: 7363 656e 6365 2d62 6173 6564 206d 6574  scence-based met
+00013f70: 686f 6420 7769 6c6c 2069 6e73 7465 6164  hod will instead
+00013f80: 2070 6572 666f 726d 2074 6865 204d 5341   perform the MSA
+00013f90: 206f 6620 6561 6368 2067 656e 6520 6661   of each gene fa
+00013fa0: 6d69 6c79 2061 6e64 2074 6865 2067 656e  mily and the gen
+00013fb0: 6520 7472 6565 2069 6e66 6572 656e 6365  e tree inference
+00013fc0: 2062 6173 6564 206f 6e20 6561 6368 204d   based on each M
+00013fd0: 5341 2c20 616e 6420 7468 656e 2069 6e66  SA, and then inf
+00013fe0: 6572 2074 6865 2073 7065 6369 6573 2074  er the species t
+00013ff0: 7265 6520 6261 7365 6420 6f6e 2074 6865  ree based on the
+00014000: 7365 2069 6e64 6976 6964 7561 6c20 6765  se individual ge
+00014010: 6e65 2074 7265 6573 2e20 5468 6520 7472  ne trees. The tr
+00014020: 6565 2d69 6e66 6572 656e 6365 2070 726f  ee-inference pro
+00014030: 6772 616d 2061 6e64 2074 6865 2061 7373  gram and the ass
+00014040: 6f63 6961 7465 6420 7061 7261 6d65 7465  ociated paramete
+00014050: 7273 2063 616e 2062 6520 7365 7420 6a75  rs can be set ju
+00014060: 7374 2061 7320 6162 6f76 6520 6279 2060  st as above by `
+00014070: 2d74 7265 6560 206f 7220 602d 2d74 7265  -tree` or `--tre
+00014080: 655f 6d65 7468 6f64 6020 616e 6420 602d  e_method` and `-
+00014090: 7473 6020 6f72 2060 2d2d 7472 6565 7365  ts` or `--treese
+000140a0: 7460 2e20 506c 6561 7365 2061 6c73 6f20  t`. Please also 
+000140b0: 6d61 6b65 2073 7572 6520 7468 6520 6368  make sure the ch
+000140c0: 6f73 656e 2074 7265 652d 696e 6665 7265  osen tree-infere
+000140d0: 6e63 6520 7072 6f67 7261 6d20 6973 2069  nce program is i
+000140e0: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+000140f0: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+00014100: 2e20 5468 6520 7072 6f67 7261 6d20 6061  . The program `a
+00014110: 7374 7261 6c2d 7072 6f60 2069 7320 7265  stral-pro` is re
+00014120: 7175 6972 6564 2074 6f20 6265 2069 6e73  quired to be ins
+00014130: 7461 6c6c 6564 2069 6e20 7468 6520 656e  talled in the en
+00014140: 7669 726f 6e6d 656e 7420 7061 7468 2069  vironment path i
+00014150: 6620 7468 6520 636f 616c 6573 6365 6e63  f the coalescenc
+00014160: 652d 6261 7365 6420 6d65 7468 6f64 2069  e-based method i
+00014170: 7320 6368 6f73 656e 2e20 5468 6520 7061  s chosen. The pa
+00014180: 7261 6c6c 656c 2074 6872 6561 6473 2068  rallel threads h
+00014190: 6572 6520 6172 6520 746f 2061 6363 656c  ere are to accel
+000141a0: 6572 6174 6520 7468 6520 7365 7175 656e  erate the sequen
+000141b0: 6365 2061 6c69 676e 6d65 6e74 2061 6e64  ce alignment and
+000141c0: 2067 656e 6520 7472 6565 2069 6e66 6572   gene tree infer
+000141d0: 656e 6365 2066 6f72 2065 6163 6820 6765  ence for each ge
+000141e0: 6e65 2066 616d 696c 7920 746f 6f20 616e  ne family too an
+000141f0: 6420 7468 7573 2073 7567 6765 7374 6564  d thus suggested
+00014200: 2074 6f20 6265 2073 6574 2061 7320 6d75   to be set as mu
+00014210: 6368 2061 7320 7468 6520 6e75 6d62 6572  ch as the number
+00014220: 206f 6620 6765 6e65 2066 616d 696c 6965   of gene familie
+00014230: 732e 0a0a 2a2a 4120 7375 6767 6573 7465  s...**A suggeste
+00014240: 6420 7374 6172 7469 6e67 2072 756e 206f  d starting run o
+00014250: 6620 7468 6973 2061 6e61 6c79 7369 7320  f this analysis 
+00014260: 6973 2077 6974 6820 7468 6520 7369 6d70  is with the simp
+00014270: 6c65 2063 6f6d 6d61 6e64 2062 656c 6f77  le command below
+00014280: 2e2a 2a0a 0a60 6060 0a77 6764 2066 6f63  .**..```.wgd foc
+00014290: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
+000142a0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+000142b0: 7365 7175 656e 6365 3320 2d2d 636f 6e63  sequence3 --conc
+000142c0: 6174 656e 6174 696f 6e0a 7767 6420 666f  atenation.wgd fo
+000142d0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+000142e0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+000142f0: 2073 6571 7565 6e63 6533 202d 2d63 6f61   sequence3 --coa
+00014300: 6c65 7363 656e 6365 0a60 6060 0a0a 2a2a  lescence.```..**
+00014310: 5468 6520 6675 6e63 7469 6f6e 616c 2061  The functional a
+00014320: 6e6e 6f74 6174 696f 6e20 6f66 2067 656e  nnotation of gen
+00014330: 6520 6661 6d69 6c69 6573 2a2a 0a60 6060  e families**.```
+00014340: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+00014350: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+00014360: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+00014370: 3320 2d2d 616e 6e6f 7461 7469 6f6e 2065  3 --annotation e
+00014380: 6767 6e6f 6720 2d65 6420 6564 6461 7461  ggnog -ed eddata
+00014390: 202d 2d64 6d6e 6220 6462 6461 7461 0a60   --dmnb dbdata.`
+000143a0: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+000143b0: 6077 6764 2066 6f63 7573 6020 616c 736f  `wgd focus` also
+000143c0: 2061 6464 6564 2073 6f6d 6520 7772 6170   added some wrap
+000143d0: 7069 6e67 2066 756e 6374 696f 6e73 2066  ping functions f
+000143e0: 6f72 2066 756e 6374 696f 6e61 6c20 616e  or functional an
+000143f0: 6e6f 7461 7469 6f6e 206f 6620 6765 6e65  notation of gene
+00014400: 2066 616d 696c 6965 7320 6f6e 2074 6865   families on the
+00014410: 2068 6f6f 6420 6f66 2064 6174 6162 6173   hood of databas
+00014420: 6573 2061 6e64 2073 6f66 7477 6172 6573  es and softwares
+00014430: 2066 6f72 2069 6e73 7461 6e63 6520 6045   for instance `E
+00014440: 6767 4e4f 4760 2061 6e64 2060 4567 674e  ggNOG` and `EggN
+00014450: 4f47 2d6d 6170 7065 7260 2e20 466f 7220  OG-mapper`. For 
+00014460: 7468 6520 616e 6e6f 7461 7469 6f6e 2075  the annotation u
+00014470: 7369 6e67 2060 4567 674e 4f47 2d6d 6170  sing `EggNOG-map
+00014480: 7065 7260 2c20 7573 6572 7320 6e65 6564  per`, users need
+00014490: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+000144a0: 7061 7468 2074 6f20 7468 6520 6567 674e  path to the eggN
+000144b0: 4f47 2061 6e6e 6f74 6174 696f 6e20 6461  OG annotation da
+000144c0: 7461 6261 7365 2076 6961 2074 6865 206f  tabase via the o
+000144d0: 7074 696f 6e20 602d 6564 602c 2074 6865  ption `-ed`, the
+000144e0: 2070 6174 6820 746f 2074 6865 2064 6961   path to the dia
+000144f0: 6d6f 6e64 2d63 6f6d 7061 7469 626c 6520  mond-compatible 
+00014500: 6461 7461 6261 7365 2076 6961 2074 6865  database via the
+00014510: 206f 7074 696f 6e20 602d 2d64 6d6e 645f   option `--dmnd_
+00014520: 6462 6020 616e 6420 7468 6520 6f70 7469  db` and the opti
+00014530: 6f6e 2060 2d2d 616e 6e6f 7461 7469 6f6e  on `--annotation
+00014540: 6020 7365 7420 6173 2022 6567 676e 6f67  ` set as "eggnog
+00014550: 222e 2054 6865 206d 616e 6e65 7220 686f  ". The manner ho
+00014560: 7720 5046 414d 2061 6e6e 6f74 6174 696f  w PFAM annotatio
+00014570: 6e20 7769 6c6c 2062 6520 7065 7266 6f72  n will be perfor
+00014580: 6d65 6420 6361 6e20 6265 2063 6f6e 7472  med can be contr
+00014590: 6f6c 6c65 6420 7669 6120 7468 6520 6f70  olled via the op
+000145a0: 7469 6f6e 2060 7066 616d 602c 2065 6974  tion `pfam`, eit
+000145b0: 6865 7220 226e 6f6e 6522 2c20 2272 6561  her "none", "rea
+000145c0: 6c69 676e 2220 6f72 2022 6465 6e6f 766f  lign" or "denovo
+000145d0: 222c 2074 6865 2064 6574 6169 6c65 6420  ", the detailed 
+000145e0: 6578 706c 616e 6174 696f 6e20 6361 6e20  explanation can 
+000145f0: 6265 2066 6f75 6e64 2061 7420 7468 6520  be found at the 
+00014600: 7769 6b69 206f 6620 5b45 6767 4e4f 472d  wiki of [EggNOG-
+00014610: 6d61 7070 6572 5d28 6874 7470 733a 2f2f  mapper](https://
+00014620: 6769 7468 7562 2e63 6f6d 2f65 6767 6e6f  github.com/eggno
+00014630: 6764 622f 6567 676e 6f67 2d6d 6170 7065  gdb/eggnog-mappe
+00014640: 722f 7769 6b69 292e 2050 6c65 6173 6520  r/wiki). Please 
+00014650: 7072 652d 696e 7374 616c 6c20 7468 6520  pre-install the 
+00014660: 6045 6767 4e4f 472d 6d61 7070 6572 6020  `EggNOG-mapper` 
+00014670: 7079 7468 6f6e 2070 6163 6b61 6765 2069  python package i
+00014680: 6620 7573 696e 6720 7468 6973 2066 756e  f using this fun
+00014690: 6374 696f 6e2e 2046 6f72 2074 6865 2061  ction. For the a
+000146a0: 6e6e 6f74 6174 696f 6e20 7573 696e 6720  nnotation using 
+000146b0: 6068 6d6d 7363 616e 602c 2077 6861 7420  `hmmscan`, what 
+000146c0: 6973 2069 6d70 6c65 6d65 6e74 6564 2069  is implemented i
+000146d0: 6e20 6077 6764 2076 3260 2069 7320 6120  n `wgd v2` is a 
+000146e0: 7369 6d70 6c65 2062 756e 646c 6520 6675  simple bundle fu
+000146f0: 6e63 7469 6f6e 2074 6f20 7065 7266 6f72  nction to perfor
+00014700: 6d20 6068 6d6d 7363 616e 6020 616e 616c  m `hmmscan` anal
+00014710: 7973 6973 2066 6f72 2061 2067 6976 656e  ysis for a given
+00014720: 2068 6d6d 2070 726f 6669 6c65 2061 6e64   hmm profile and
+00014730: 2073 6574 206f 6620 6765 6e65 2066 616d   set of gene fam
+00014740: 696c 6965 7320 7375 6368 2074 6861 7420  ilies such that 
+00014750: 7573 6572 7320 6e65 6564 2074 6f20 7365  users need to se
+00014760: 7420 7468 6520 6f70 7469 6f6e 2060 2d2d  t the option `--
+00014770: 616e 6e6f 7461 7469 6f6e 6020 6173 2022  annotation` as "
+00014780: 686d 6d70 6661 6d22 2061 6e64 2070 726f  hmmpfam" and pro
+00014790: 7669 6465 2074 6865 2068 6d6d 7072 6f66  vide the hmmprof
+000147a0: 696c 6520 7669 6120 7468 6520 6f70 7469  ile via the opti
+000147b0: 6f6e 2060 2d2d 686d 6d60 2e20 466f 7220  on `--hmm`. For 
+000147c0: 7468 6520 616e 6e6f 7461 7469 6f6e 2075  the annotation u
+000147d0: 7369 6e67 2060 696e 7465 7270 726f 7363  sing `interprosc
+000147e0: 616e 602c 2075 7365 7273 206e 6565 6420  an`, users need 
+000147f0: 746f 2070 726f 7669 6465 2074 6865 2070  to provide the p
+00014800: 6174 6820 746f 2074 6865 2069 6e74 6572  ath to the inter
+00014810: 7072 6f73 6361 6e20 696e 7374 616c 6c61  proscan installa
+00014820: 7469 6f6e 2066 6f6c 6465 7220 7768 6572  tion folder wher
+00014830: 6520 7468 6572 6520 6973 2061 2060 696e  e there is a `in
+00014840: 7465 7270 726f 7363 616e 2e73 6860 2066  terproscan.sh` f
+00014850: 696c 6520 7669 6120 7468 6520 6f70 7469  ile via the opti
+00014860: 6f6e 2060 2d2d 6578 6570 6174 6860 2061  on `--exepath` a
+00014870: 6e64 2073 6574 2074 6865 2074 6865 206f  nd set the the o
+00014880: 7074 696f 6e20 602d 2d61 6e6e 6f74 6174  ption `--annotat
+00014890: 696f 6e60 2061 7320 2269 6e74 6572 7072  ion` as "interpr
+000148a0: 6f73 6361 6e22 2e20 5468 6520 7061 7261  oscan". The para
+000148b0: 6c6c 656c 2074 6872 6561 6473 2068 6572  llel threads her
+000148c0: 6520 6172 6520 746f 2070 6172 616c 6c65  e are to paralle
+000148d0: 6c69 7a65 2074 6865 2061 6e61 6c79 7369  lize the analysi
+000148e0: 7320 666f 7220 6561 6368 2067 656e 6520  s for each gene 
+000148f0: 6661 6d69 6c79 2061 6e64 2074 6875 7320  family and thus 
+00014900: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
+00014910: 7365 7420 6173 206d 7563 6820 6173 2074  set as much as t
+00014920: 6865 206e 756d 6265 7220 6f66 2067 656e  he number of gen
+00014930: 6520 6661 6d69 6c69 6573 2e0a 0a2a 2a41  e families...**A
+00014940: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
+00014950: 696e 6720 7275 6e20 6f66 2074 6869 7320  ing run of this 
+00014960: 616e 616c 7973 6973 2063 616e 2062 6520  analysis can be 
+00014970: 7769 7468 2074 6865 2063 6f6d 6d61 6e64  with the command
+00014980: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
+00014990: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
+000149a0: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
+000149b0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+000149c0: 2d2d 616e 6e6f 7461 7469 6f6e 2065 6767  --annotation egg
+000149d0: 6e6f 6720 2d65 6420 6564 6461 7461 202d  nog -ed eddata -
+000149e0: 2d64 6d6e 6220 6462 6461 7461 0a77 6764  -dmnb dbdata.wgd
+000149f0: 2066 6f63 7573 2066 616d 696c 6965 7320   focus families 
+00014a00: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+00014a10: 6365 3220 7365 7175 656e 6365 3320 2d2d  ce2 sequence3 --
+00014a20: 616e 6e6f 7461 7469 6f6e 2068 6d6d 7066  annotation hmmpf
+00014a30: 616d 202d 2d68 6d6d 2068 6d6d 6461 7461  am --hmm hmmdata
+00014a40: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+00014a50: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+00014a60: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+00014a70: 3320 2d2d 616e 6e6f 7461 7469 6f6e 2069  3 --annotation i
+00014a80: 6e74 6572 7072 6f73 6361 6e20 2d2d 6578  nterproscan --ex
+00014a90: 6570 6174 6820 2450 4154 480a 6060 600a  epath $PATH.```.
+00014aa0: 0a2a 2a54 6865 2070 6879 6c6f 6765 6e65  .**The phylogene
+00014ab0: 7469 6320 6461 7469 6e67 206f 6620 5747  tic dating of WG
+00014ac0: 4473 2a2a 0a60 6060 0a77 6764 2066 6f63  Ds**.```.wgd foc
+00014ad0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
+00014ae0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+00014af0: 7365 7175 656e 6365 3320 2d64 206d 636d  sequence3 -d mcm
+00014b00: 6374 7265 6520 2d73 7020 7370 6461 7461  ctree -sp spdata
+00014b10: 2028 2d2d 7072 6f74 636f 6364 6174 696e   (--protcocdatin
+00014b20: 6729 2028 2d2d 7061 7274 6974 696f 6e29  g) (--partition)
+00014b30: 2028 2d2d 6161 6d6f 6465 6c20 6c67 2920   (--aamodel lg) 
+00014b40: 282d 6473 2027 6275 726e 696e 203d 2032  (-ds 'burnin = 2
+00014b50: 3030 3027 2920 282d 6473 2027 7361 6d70  000') (-ds 'samp
+00014b60: 6672 6571 203d 2031 3030 3027 2920 282d  freq = 1000') (-
+00014b70: 6473 2027 6e73 616d 706c 6520 3d20 3230  ds 'nsample = 20
+00014b80: 3030 3027 2920 2028 2d6e 2034 2920 282d  000')  (-n 4) (-
+00014b90: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
+00014ba0: 2920 282d 6f20 7767 645f 666f 6375 7329  ) (-o wgd_focus)
+00014bb0: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
+00014bc0: 290a 6060 600a 0a54 6865 2061 6273 6f6c  ).```..The absol
+00014bd0: 7574 6520 6461 7469 6e67 206f 6620 5747  ute dating of WG
+00014be0: 4473 2069 7320 6120 7370 6563 6966 6963  Ds is a specific
+00014bf0: 2070 6970 656c 696e 6520 696d 706c 656d   pipeline implem
+00014c00: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
+00014c10: 6020 7573 696e 6720 7468 6520 6d65 7468  ` using the meth
+00014c20: 6f64 206f 6620 7068 796c 6f67 656e 6574  od of phylogenet
+00014c30: 6963 2064 6174 696e 672e 2054 6865 2066  ic dating. The f
+00014c40: 616d 696c 6965 7320 7573 6564 2069 6e20  amilies used in 
+00014c50: 7468 6973 2073 7465 7020 6361 6e20 6265  this step can be
+00014c60: 2070 726f 6475 6365 6420 6672 6f6d 2060   produced from `
+00014c70: 7767 6420 646d 6460 2061 6e64 2060 7767  wgd dmd` and `wg
+00014c80: 6420 7065 616b 602e 204e 6f74 6520 7468  d peak`. Note th
+00014c90: 6174 2068 6572 6520 7765 206f 6e6c 7920  at here we only 
+00014ca0: 6469 7363 7573 7320 686f 7720 746f 2064  discuss how to d
+00014cb0: 6174 6520 5747 4473 2077 6974 6820 6765  ate WGDs with ge
+00014cc0: 6e6f 6d65 2061 7373 656d 626c 792c 2069  nome assembly, i
+00014cd0: 6e73 7465 6164 206f 6620 7472 616e 7363  nstead of transc
+00014ce0: 7269 7074 6f6d 6520 6173 7365 6d62 6c79  riptome assembly
+00014cf0: 2028 7768 6963 6820 7769 6c6c 2062 6520   (which will be 
+00014d00: 6469 7363 7573 7365 6420 696e 2061 2073  discussed in a s
+00014d10: 6570 6172 6174 6520 7365 6374 696f 6e20  eparate section 
+00014d20: 6865 7265 756e 6465 7229 2e20 5468 6520  hereunder). The 
+00014d30: 6173 7375 6d70 7469 6f6e 2077 6520 6d61  assumption we ma
+00014d40: 6465 2068 6572 6520 6973 2074 6861 7420  de here is that 
+00014d50: 6e6f 7420 616c 6c20 616e 6368 6f72 2070  not all anchor p
+00014d60: 6169 7273 2028 636f 6c6c 696e 6561 7220  airs (collinear 
+00014d70: 6475 706c 6963 6174 6573 2920 6172 6520  duplicates) are 
+00014d80: 7375 6974 6162 6c65 2066 6f72 2070 6879  suitable for phy
+00014d90: 6c6f 6765 6e65 7469 6320 6461 7469 6e67  logenetic dating
+00014da0: 2c20 666f 7220 696e 7374 616e 6365 2074  , for instance t
+00014db0: 686f 7365 2066 6173 746c 7920 6f72 2073  hose fastly or s
+00014dc0: 6c6f 776c 7920 6576 6f6c 7669 6e67 2067  lowly evolving g
+00014dd0: 656e 6520 6475 706c 6963 6174 6573 2c20  ene duplicates, 
+00014de0: 6265 6361 7573 6520 7468 6579 2772 6520  because they're 
+00014df0: 7072 6f6e 6520 746f 2067 6976 6520 6269  prone to give bi
+00014e00: 6173 6564 2064 6174 696e 6720 6573 7469  ased dating esti
+00014e10: 6d61 7469 6f6e 2e20 536f 2061 7320 746f  mation. So as to
+00014e20: 2072 6574 7269 6576 6520 7468 6520 2272   retrieve the "r
+00014e30: 656c 6961 626c 6522 2061 6e63 686f 7220  eliable" anchor 
+00014e40: 7061 6972 732c 2077 6520 696d 706c 656d  pairs, we implem
+00014e50: 656e 7465 6420 736f 6d65 206d 6574 686f  ented some metho
+00014e60: 6473 206f 6620 6964 656e 7469 6679 696e  ds of identifyin
+00014e70: 6720 6372 6564 6961 626c 6520 616e 6368  g crediable anch
+00014e80: 6f72 2070 6169 7273 2062 6173 6564 206f  or pairs based o
+00014e90: 6e20 7468 6569 7220 4b73 2076 616c 7565  n their Ks value
+00014ea0: 7320 616e 642f 6f72 2072 6573 6964 696e  s and/or residin
+00014eb0: 6720 636f 6c6c 696e 6561 7220 7365 676d  g collinear segm
+00014ec0: 656e 7473 2e20 466f 7220 6120 6765 6e6f  ents. For a geno
+00014ed0: 6d65 2077 6974 6820 636c 6561 7220 7369  me with clear si
+00014ee0: 676e 616c 7320 6f66 2070 7574 6174 6976  gnals of putativ
+00014ef0: 6520 5747 4473 2c20 7375 6368 2061 7320  e WGDs, such as 
+00014f00: 7468 6520 2a41 7175 696c 6567 6961 2063  the *Aquilegia c
+00014f10: 6f65 7275 6c65 612a 2069 6e20 7468 6520  oerulea* in the 
+00014f20: 6578 616d 706c 652c 2061 2068 6575 7269  example, a heuri
+00014f30: 7374 6963 206d 6574 686f 6420 7468 6174  stic method that
+00014f40: 2061 7070 6c69 6573 2074 6865 2070 7269   applies the pri
+00014f50: 6e63 6970 6c65 206f 6620 686f 7720 5b6b  nciple of how [k
+00014f60: 7372 6174 6573 5d28 6874 7470 733a 2f2f  srates](https://
+00014f70: 6769 7468 7562 2e63 6f6d 2f56 4942 2d50  github.com/VIB-P
+00014f80: 5342 2f6b 7372 6174 6573 2920 6669 6e64  SB/ksrates) find
+00014f90: 2074 6865 2069 6e69 7469 616c 2070 6561   the initial pea
+00014fa0: 6b73 2061 6e64 2074 6865 6972 2070 6172  ks and their par
+00014fb0: 616d 6574 6572 7320 7761 7320 696d 706c  ameters was impl
+00014fc0: 656d 656e 7465 6420 746f 2066 696e 6420  emented to find 
+00014fd0: 7468 6520 3935 2520 636f 6e66 6964 656e  the 95% confiden
+00014fe0: 6365 206c 6576 656c 206f 6620 7468 6520  ce level of the 
+00014ff0: 6173 7375 6d65 6420 6c6f 676e 6f72 6d61  assumed lognorma
+00015000: 6c20 6469 7374 7269 6275 7469 6f6e 206f  l distribution o
+00015010: 6620 7468 6520 616e 6368 6f72 2070 6169  f the anchor pai
+00015020: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
+00015030: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00015040: 6f6e 2074 6f20 6669 6c74 6572 2061 6e63  on to filter anc
+00015050: 686f 7220 7061 6972 7320 7769 7468 2074  hor pairs with t
+00015060: 6f6f 2068 6967 6820 6f72 206c 6f77 202a  oo high or low *
+00015070: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00015080: 6765 732c 2074 6865 2065 7861 6d70 6c61  ges, the exampla
+00015090: 7220 636f 6d6d 616e 6420 6f66 2077 6869  r command of whi
+000150a0: 6368 2069 7320 7368 6f77 6564 2069 6e20  ch is showed in 
+000150b0: 7468 6520 6077 6764 2070 6561 6b60 2073  the `wgd peak` s
+000150c0: 6563 7469 6f6e 206f 6620 5b49 6c6c 7573  ection of [Illus
+000150d0: 7472 6174 696f 6e5d 2868 7474 7073 3a2f  tration](https:/
+000150e0: 2f67 6974 6875 622e 636f 6d2f 6865 6368  /github.com/hech
+000150f0: 652d 7073 622f 7767 643f 7461 623d 7265  e-psb/wgd?tab=re
+00015100: 6164 6d65 2d6f 762d 6669 6c65 2369 6c6c  adme-ov-file#ill
+00015110: 7573 7472 6174 696f 6e29 2e20 4966 2074  ustration). If t
+00015120: 6869 7320 6865 7572 6973 7469 6320 6d65  his heuristic me
+00015130: 7468 6f64 2066 6169 6c65 6420 746f 2067  thod failed to g
+00015140: 6976 6520 7265 6173 6f6e 6162 6c65 2072  ive reasonable r
+00015150: 6573 756c 7473 2c20 7768 6963 6820 6973  esults, which is
+00015160: 2075 7375 616c 6c79 2064 7565 2074 6f20   usually due to 
+00015170: 7468 6520 6566 6665 6374 206f 6620 6d75  the effect of mu
+00015180: 6c74 6970 6c65 2061 646a 6163 656e 7420  ltiple adjacent 
+00015190: 5747 4473 2074 6861 7420 626c 7572 7320  WGDs that blurs 
+000151a0: 7468 6520 7065 616b 2d66 696e 6469 6e67  the peak-finding
+000151b0: 2070 726f 6365 7373 2c20 7573 6572 7320   process, users 
+000151c0: 6361 6e20 7475 726e 2074 6f20 7468 6520  can turn to the 
+000151d0: 636f 6c6c 696e 6561 7220 7365 676d 656e  collinear segmen
+000151e0: 7473 2d67 7569 6465 6420 616e 6368 6f72  ts-guided anchor
+000151f0: 2070 6169 7220 636c 7573 7465 7269 6e67   pair clustering
+00015200: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
+00015210: 6077 6764 2076 3260 2c20 696e 2077 6869  `wgd v2`, in whi
+00015220: 6368 2061 2063 6f6c 6c69 6e65 6172 2073  ch a collinear s
+00015230: 6567 6d65 6e74 2d77 6973 6520 474d 4d20  egment-wise GMM 
+00015240: 636c 7573 7465 7269 6e67 2077 696c 6c20  clustering will 
+00015250: 6265 2066 6972 7374 2063 6f6e 6475 6374  be first conduct
+00015260: 6564 2062 6173 6564 206f 6e20 7468 6520  ed based on the 
+00015270: 2273 6f2d 6361 6c6c 6564 2220 7365 676d  "so-called" segm
+00015280: 656e 7420 2a4b 2a3c 7375 623e 533c 2f73  ent *K*<sub>S</s
+00015290: 7562 3e20 6167 6520 7265 7072 6573 656e  ub> age represen
+000152a0: 7465 6420 6279 2074 6865 206d 6564 6961  ted by the media
+000152b0: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+000152c0: 3e20 6167 6520 6f66 2061 6c6c 2074 6865  > age of all the
+000152d0: 2072 6573 6964 696e 6720 7379 6e74 656c   residing syntel
+000152e0: 6f67 7320 286e 6f74 6520 7468 6174 2074  ogs (note that t
+000152f0: 6865 2067 656e 6520 6475 706c 6963 6174  he gene duplicat
+00015300: 6520 7061 6972 7320 6164 6f70 7465 6420  e pairs adopted 
+00015310: 696e 2074 6869 7320 7374 6570 2069 7320  in this step is 
+00015320: 6672 6f6d 2074 6865 2066 696c 6520 606d  from the file `m
+00015330: 756c 7469 706c 6963 6f6e 5f70 6169 7273  ultiplicon_pairs
+00015340: 2e74 7874 6020 7768 6963 6820 636f 6e74  .txt` which cont
+00015350: 6169 6e73 2074 6865 2066 756c 6c20 7365  ains the full se
+00015360: 7420 6f66 2073 796e 7465 6c6f 6773 292c  t of syntelogs),
+00015370: 2069 6e73 7465 6164 206f 6620 7468 6520   instead of the 
+00015380: 736d 616c 6c65 7220 6765 6e65 2073 6574  smaller gene set
+00015390: 206f 6620 616e 6368 6f72 2070 6169 7273   of anchor pairs
+000153a0: 2e20 5468 6520 6469 7374 696e 6374 696f  . The distinctio
+000153b0: 6e20 6265 7477 6565 6e20 616e 6368 6f72  n between anchor
+000153c0: 2070 6169 7273 2061 6e64 2073 796e 7465   pairs and synte
+000153d0: 6c6f 6773 2069 7320 7468 6174 2074 6865  logs is that the
+000153e0: 206c 6174 7465 7220 7265 6665 7273 2074   latter refers t
+000153f0: 6f20 6d75 6c74 6970 6c65 2073 6574 7320  o multiple sets 
+00015400: 6f66 2067 656e 6573 2064 6572 6976 6564  of genes derived
+00015410: 2066 726f 6d20 7468 6520 7361 6d65 2061   from the same a
+00015420: 6e63 6573 7472 616c 2067 656e 6f6d 6963  ncestral genomic
+00015430: 2072 6567 696f 6e20 7768 696c 6520 7468   region while th
+00015440: 6520 666f 726d 6572 2069 6d70 6c69 6573  e former implies
+00015450: 2074 6865 206c 6174 7465 7220 6275 7420   the latter but 
+00015460: 7265 7175 6972 6573 2061 6464 6974 696f  requires additio
+00015470: 6e61 6c6c 7920 7468 6520 636f 6e73 6572  nally the conser
+00015480: 7665 6420 6765 6e65 206f 7264 6572 2c20  ved gene order, 
+00015490: 626f 7468 206f 6620 7768 6963 6820 6172  both of which ar
+000154a0: 652c 2077 6974 6869 6e20 6120 6765 6e6f  e, within a geno
+000154b0: 6d65 2c20 6173 7375 6d65 6420 746f 2062  me, assumed to b
+000154c0: 6520 6f72 6967 696e 6174 6564 2066 726f  e originated fro
+000154d0: 6d20 7468 6520 6475 706c 6963 6174 696f  m the duplicatio
+000154e0: 6e20 6f66 2061 2063 6f6d 6d6f 6e20 616e  n of a common an
+000154f0: 6365 7374 7261 6c20 6765 6e6f 6d69 6320  cestral genomic 
+00015500: 7265 6769 6f6e 2061 6e64 2061 7320 7375  region and as su
+00015510: 6368 2064 6565 6d65 6420 6576 6964 656e  ch deemed eviden
+00015520: 6365 2066 6f72 2057 4744 2e20 5468 656e  ce for WGD. Then
+00015530: 2074 6865 2073 796e 7465 6c6f 6773 2077   the syntelogs w
+00015540: 696c 6c20 6265 206d 6170 7065 6420 6261  ill be mapped ba
+00015550: 636b 2061 6363 6f72 6469 6e67 2074 6f20  ck according to 
+00015560: 7468 6520 636c 7573 7465 7269 6e67 2072  the clustering r
+00015570: 6573 756c 7473 206f 6620 7468 6569 7220  esults of their 
+00015580: 6166 6669 6c69 6174 6564 2073 6567 6d65  affiliated segme
+00015590: 6e74 732e 2053 696e 6365 2074 6865 2047  nts. Since the G
+000155a0: 6175 7373 6961 6e20 7368 6170 6520 6f66  aussian shape of
+000155b0: 2074 6865 2073 6567 6d65 6e74 2063 6c75   the segment clu
+000155c0: 7374 6572 2064 6f65 736e 2774 206e 6563  ster doesn't nec
+000155d0: 6573 7361 7269 6c79 2069 6d69 7461 7465  essarily imitate
+000155e0: 2074 6865 2073 6861 7065 206f 6620 7468   the shape of th
+000155f0: 6520 7265 7369 6469 6e67 2073 796e 7465  e residing synte
+00015600: 6c6f 6773 2c20 736f 2061 7320 746f 2072  logs, so as to r
+00015610: 6574 7269 6576 6520 7468 6520 2272 656c  etrieve the "rel
+00015620: 6961 626c 6522 2067 656e 6520 7365 7473  iable" gene sets
+00015630: 2066 6f72 2070 6879 6c6f 6765 6e65 7469   for phylogeneti
+00015640: 6320 6461 7469 6e67 2c20 7765 2061 6470  c dating, we adp
+00015650: 6f74 6564 2074 6865 2028 3935 2529 2068  oted the (95%) h
+00015660: 6967 6865 7374 2064 656e 7369 7479 2072  ighest density r
+00015670: 6567 696f 6e20 2848 4452 2920 6f66 2074  egion (HDR) of t
+00015680: 6865 2073 796e 7465 6c6f 6720 2a4b 2a3c  he syntelog *K*<
+00015690: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+000156a0: 6469 7374 7269 6275 7469 6f6e 2066 6f72  distribution for
+000156b0: 2074 6865 2070 6879 6c6f 6765 6e65 7469   the phylogeneti
+000156c0: 6320 6461 7469 6e67 2c20 7468 6520 6361  c dating, the ca
+000156d0: 6c63 756c 6174 696f 6e20 6f66 2074 6865  lculation of the
+000156e0: 2028 3935 2529 2048 4452 2069 6e20 7468   (95%) HDR in th
+000156f0: 6520 6675 6e63 7469 6f6e 205b 6361 6c63  e function [calc
+00015700: 756c 6174 6548 5044 5d28 6874 7470 733a  ulateHPD](https:
+00015710: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
+00015720: 6865 2d70 7362 2f77 6764 2f62 6c6f 622f  he-psb/wgd/blob/
+00015730: 7068 796c 6f64 6174 696e 672f 7767 642f  phylodating/wgd/
+00015740: 7065 616b 2e70 7923 4c31 3735 3729 2073  peak.py#L1757) s
+00015750: 6565 6b73 2074 6865 2073 686f 7274 6573  eeks the shortes
+00015760: 7420 2a4b 2a3c 7375 623e 533c 2f73 7562  t *K*<sub>S</sub
+00015770: 3e20 7261 6e67 6520 2861 2c62 2920 7768  > range (a,b) wh
+00015780: 6963 6820 7361 7469 7366 6965 7320 7468  ich satisfies th
+00015790: 6520 7265 7175 6972 656d 656e 7420 6f66  e requirement of
+000157a0: 2073 7061 6e6e 696e 6720 6d6f 7265 2074   spanning more t
+000157b0: 6861 6e20 2839 3525 2920 6f66 2061 6c6c  han (95%) of all
+000157c0: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+000157d0: 7375 623e 2076 616c 7565 732e 204f 6e20  sub> values. On 
+000157e0: 7468 6520 7072 656d 6973 6520 6f66 2069  the premise of i
+000157f0: 6465 6e74 6966 6965 6420 616e 6368 6f72  dentified anchor
+00015800: 2070 6169 7273 2028 6e6f 7465 2074 6861   pairs (note tha
+00015810: 7420 7468 6520 7379 6e74 656c 6f67 7320  t the syntelogs 
+00015820: 6172 6520 616c 736f 2072 6566 6572 7265  are also referre
+00015830: 6420 6173 2061 6e63 686f 7220 7061 6972  d as anchor pair
+00015840: 7320 6865 7265 6166 7465 7229 2c20 7765  s hereafter), we
+00015850: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
+00015860: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
+00015870: 2064 6d64 6020 7468 6520 736f 2d63 616c   dmd` the so-cal
+00015880: 6c65 6420 616e 6368 6f72 2d61 7761 7265  led anchor-aware
+00015890: 206c 6f63 616c 204d 5242 4873 206f 7220   local MRBHs or 
+000158a0: 6f72 7468 6f67 726f 7570 732c 2069 6e20  orthogroups, in 
+000158b0: 7768 6963 6820 7468 6520 6f72 6967 696e  which the origin
+000158c0: 616c 206c 6f63 616c 204d 5242 4873 2061  al local MRBHs a
+000158d0: 7265 2066 7572 7468 6572 206d 6572 6765  re further merge
+000158e0: 6420 7769 7468 2061 6e63 686f 7220 7061  d with anchor pa
+000158f0: 6972 7320 7375 6368 2074 6861 7420 6561  irs such that ea
+00015900: 6368 206f 7274 686f 6772 6f75 7020 636f  ch orthogroup co
+00015910: 6e74 6169 6e73 2074 6865 2061 6e63 686f  ntains the ancho
+00015920: 7220 7061 6972 2061 6e64 2074 6865 206f  r pair and the o
+00015930: 7274 686f 6c6f 6775 6573 2e20 5769 7468  rthologues. With
+00015940: 2074 6869 7320 6f72 7468 6f67 726f 7570   this orthogroup
+00015950: 2c20 7573 6572 7320 7468 656e 206e 6565  , users then nee
+00015960: 6420 6f6e 6520 7374 6172 7469 6e67 2074  d one starting t
+00015970: 7265 6520 6669 6c65 2028 6173 2073 686f  ree file (as sho
+00015980: 776e 2069 6e20 7468 6520 5b49 6c6c 7573  wn in the [Illus
+00015990: 7472 6174 696f 6e5d 2868 7474 7073 3a2f  tration](https:/
+000159a0: 2f67 6974 6875 622e 636f 6d2f 6865 6368  /github.com/hech
+000159b0: 652d 7073 622f 7767 642f 7472 6565 2f70  e-psb/wgd/tree/p
+000159c0: 6879 6c6f 6461 7469 6e67 3f74 6162 3d72  hylodating?tab=r
+000159d0: 6561 646d 652d 6f76 2d66 696c 6523 696c  eadme-ov-file#il
+000159e0: 6c75 7374 7261 7469 6f6e 2920 7061 7274  lustration) part
+000159f0: 2920 696e 6469 6361 7469 6e67 2074 6865  ) indicating the
+00015a00: 2074 7265 6520 746f 706f 6c6f 6779 2061   tree topology a
+00015a10: 6e64 2066 6f73 7369 6c20 6361 6c69 6272  nd fossil calibr
+00015a20: 6174 696f 6e20 696e 666f 726d 6174 696f  ation informatio
+00015a30: 6e20 666f 7220 7468 6520 6669 6e61 6c20  n for the final 
+00015a40: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
+00015a50: 696e 6720 7573 696e 6720 7468 6520 7072  ing using the pr
+00015a60: 6f67 7261 6d20 6077 6764 2066 6f63 7573  ogram `wgd focus
+00015a70: 602e 2055 7365 7273 2063 616e 2073 6574  `. Users can set
+00015a80: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
+00015a90: 2060 2d2d 7072 6f74 636f 6364 6174 696e   `--protcocdatin
+00015aa0: 6760 2074 6f20 6f6e 6c79 2063 6f6e 6475  g` to only condu
+00015ab0: 6374 2074 6865 2064 6174 696e 6720 6f66  ct the dating of
+00015ac0: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
+00015ad0: 6f6e 2070 726f 7465 696e 204d 5341 206f  on protein MSA o
+00015ae0: 7220 7468 6520 666c 6167 206f 7074 696f  r the flag optio
+00015af0: 6e20 602d 2d70 726f 7464 6174 696e 6760  n `--protdating`
+00015b00: 2074 6f20 6f6e 6c79 2063 6f6e 6475 6374   to only conduct
+00015b10: 2074 6865 2064 6174 696e 6720 6f66 2074   the dating of t
+00015b20: 6865 2070 726f 7465 696e 204d 5341 732c  he protein MSAs,
+00015b30: 206e 6f74 6564 2074 6861 7420 7468 6573   noted that thes
+00015b40: 6520 7477 6f20 6f70 7469 6f6e 7320 6f6e  e two options on
+00015b50: 6c79 2077 6f72 6b20 666f 7220 7468 6520  ly work for the 
+00015b60: 606d 636d 6374 7265 6560 206f 7074 696f  `mcmctree` optio
+00015b70: 6e20 736f 2066 6172 2e20 5468 6520 666c  n so far. The fl
+00015b80: 6167 206f 7074 696f 6e20 602d 2d70 6172  ag option `--par
+00015b90: 7469 7469 6f6e 6020 6361 6e20 6265 2073  tition` can be s
+00015ba0: 6574 2074 6f20 7065 7266 6f72 6d20 606d  et to perform `m
+00015bb0: 636d 6374 7265 6560 2061 6e61 6c79 7369  cmctree` analysi
+00015bc0: 7320 7573 696e 6720 7468 6520 7061 7274  s using the part
+00015bd0: 6974 696f 6e65 6420 6461 7461 2028 692e  itioned data (i.
+00015be0: 652e 2c20 3173 742c 2032 6e64 2061 6e64  e., 1st, 2nd and
+00015bf0: 2033 7264 2070 6f73 6974 696f 6e20 6f66   3rd position of
+00015c00: 2063 6f64 6f6e 2920 696e 7374 6561 6420   codon) instead 
+00015c10: 6f66 2075 7369 6e67 2074 6865 2063 6f64  of using the cod
+00015c20: 6f6e 2061 7320 6120 7768 6f6c 652e 2054  on as a whole. T
+00015c30: 6865 206f 7074 696f 6e20 602d 2d61 616d  he option `--aam
+00015c40: 6f64 656c 6020 6361 6e20 6265 2073 6574  odel` can be set
+00015c50: 2074 6f20 6465 7465 726d 696e 6520 7468   to determine th
+00015c60: 6520 616d 696e 6f20 6163 6964 206d 6f64  e amino acid mod
+00015c70: 656c 2061 7070 6c69 6564 2069 6e20 606d  el applied in `m
+00015c80: 636d 6374 7265 6560 2061 6e61 6c79 7369  cmctree` analysi
+00015c90: 732e 2054 6865 206f 7074 696f 6e20 602d  s. The option `-
+00015ca0: 6473 6020 6361 6e20 6265 2075 7365 6420  ds` can be used 
+00015cb0: 746f 2073 6574 2070 6172 616d 6574 6572  to set parameter
+00015cc0: 7320 666f 7220 7468 6520 6d6f 6c65 6375  s for the molecu
+00015cd0: 6c61 7220 6461 7469 6e67 2070 726f 6772  lar dating progr
+00015ce0: 616d 2e20 5468 6520 7061 7261 6c6c 656c  am. The parallel
+00015cf0: 2074 6872 6561 6473 2068 6572 6520 6172   threads here ar
+00015d00: 6520 746f 2070 6172 616c 6c65 6c69 7a65  e to parallelize
+00015d10: 2074 6865 2061 6e61 6c79 7369 7320 666f   the analysis fo
+00015d20: 7220 6561 6368 2067 656e 6520 6661 6d69  r each gene fami
+00015d30: 6c79 2061 6e64 2074 6875 7320 7375 6767  ly and thus sugg
+00015d40: 6573 7465 6420 746f 2062 6520 7365 7420  ested to be set 
+00015d50: 6173 206d 7563 6820 6173 2074 6865 206e  as much as the n
+00015d60: 756d 6265 7220 6f66 2067 656e 6520 6661  umber of gene fa
+00015d70: 6d69 6c69 6573 2e0a 0a2a 2a41 2073 7567  milies...**A sug
+00015d80: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+00015d90: 7275 6e20 6361 6e20 6265 2077 6974 6820  run can be with 
+00015da0: 7468 6520 636f 6d6d 616e 6420 6265 6c6f  the command belo
+00015db0: 772e 2a2a 0a0a 6060 600a 7767 6420 666f  w.**..```.wgd fo
+00015dc0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+00015dd0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+00015de0: 2073 6571 7565 6e63 6533 202d 6420 6d63   sequence3 -d mc
+00015df0: 6d63 7472 6565 202d 7370 2073 7064 6174  mctree -sp spdat
+00015e00: 610a 7767 6420 666f 6375 7320 6661 6d69  a.wgd focus fami
+00015e10: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+00015e20: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+00015e30: 6533 202d 6420 7238 7320 2d73 7020 7370  e3 -d r8s -sp sp
+00015e40: 6461 7461 202d 2d6e 7369 7465 7320 6e73  data --nsites ns
+00015e50: 6974 6569 6e66 6f0a 7767 6420 666f 6375  iteinfo.wgd focu
+00015e60: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+00015e70: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+00015e80: 6571 7565 6e63 6533 202d 6420 6265 6173  equence3 -d beas
+00015e90: 7420 2d2d 666f 7373 696c 2066 6f73 7369  t --fossil fossi
+00015ea0: 6c69 6e66 6f20 2d2d 726f 6f74 6865 6967  linfo --rootheig
+00015eb0: 6874 2072 6f6f 7468 6569 6768 7469 6e66  ht rootheightinf
+00015ec0: 6f20 2d2d 6368 6169 6e73 6574 2063 6861  o --chainset cha
+00015ed0: 696e 7365 7469 6e66 6f20 2d2d 6265 6173  insetinfo --beas
+00015ee0: 746c 676a 6172 2024 5041 5448 0a60 6060  tlgjar $PATH.```
+00015ef0: 0a0a 2323 2049 6c6c 7573 7472 6174 696f  ..## Illustratio
+00015f00: 6e0a 0a57 6520 696c 6c75 7374 7261 7465  n..We illustrate
+00015f10: 206f 7572 2070 726f 6772 616d 206f 6e20   our program on 
+00015f20: 616e 2065 7865 6d70 6c61 7279 2057 4744  an exemplary WGD
+00015f30: 2069 6e66 6572 656e 6365 2061 6e64 2064   inference and d
+00015f40: 6174 696e 6720 7570 6f6e 2073 7065 6369  ating upon speci
+00015f50: 6573 202a 4171 7569 6c65 6769 6120 636f  es *Aquilegia co
+00015f60: 6572 756c 6561 2a2e 0a0a 5468 6520 2a41  erulea*...The *A
+00015f70: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
+00015f80: 612a 2077 6173 2072 6570 6f72 7465 6420  a* was reported 
+00015f90: 746f 2065 7870 6572 6965 6e63 6520 616e  to experience an
+00015fa0: 2070 616c 656f 2d70 6f6c 7970 6c6f 6964   paleo-polyploid
+00015fb0: 697a 6174 696f 6e20 6576 656e 7420 6166  ization event af
+00015fc0: 7465 7220 7468 6520 6469 7665 7267 656e  ter the divergen
+00015fd0: 6365 206f 6620 636f 7265 2065 7564 6963  ce of core eudic
+00015fe0: 6f74 732c 2077 6869 6368 2069 7320 6c69  ots, which is li
+00015ff0: 6b65 6c79 2073 6861 7265 6420 6279 2061  kely shared by a
+00016000: 6c6c 2052 616e 756e 6375 6c61 6c65 732e  ll Ranunculales.
+00016010: 0a0a 4669 7273 7420 6162 6f76 6520 616c  ..First above al
+00016020: 6c2c 206c 6574 2773 2064 656c 696e 6561  l, let's delinea
+00016030: 7465 2074 6865 2077 686f 6c65 2070 6172  te the whole par
+00016040: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00016050: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00016060: 6275 7469 6f6e 2061 6e64 2068 6176 6520  bution and have 
+00016070: 6120 6261 7369 6320 6f62 7365 7276 6174  a basic observat
+00016080: 696f 6e20 666f 7220 706f 7465 6e74 6961  ion for potentia
+00016090: 6c6c 7920 636f 6e63 6569 7661 626c 6520  lly conceivable 
+000160a0: 5747 4473 2c20 7573 696e 6720 7468 6520  WGDs, using the 
+000160b0: 636f 6d6d 616e 6420 6c69 6e65 2062 656c  command line bel
+000160c0: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
+000160d0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+000160e0: 6c65 610a 7767 6420 6b73 6420 7767 645f  lea.wgd ksd wgd_
+000160f0: 646d 642f 4171 7569 6c65 6769 615f 636f  dmd/Aquilegia_co
+00016100: 6572 756c 6561 2e74 7376 2041 7175 696c  erulea.tsv Aquil
+00016110: 6567 6961 5f63 6f65 7275 6c65 610a 6060  egia_coerulea.``
+00016120: 600a 0a54 6865 2063 6f6e 7374 7275 6374  `..The construct
+00016130: 6564 2077 686f 6c65 2070 6172 616e 6f6d  ed whole paranom
+00016140: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00016150: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00016160: 6f6e 206f 6620 2a41 7175 696c 6567 6961  on of *Aquilegia
+00016170: 2063 6f65 7275 6c65 612a 2069 7320 6173   coerulea* is as
+00016180: 2062 656c 6f77 2c20 7765 2063 616e 2073   below, we can s
+00016190: 6565 2074 6861 7420 7468 6572 6520 7365  ee that there se
+000161a0: 656d 7320 746f 2062 6520 6120 6875 6d70  ems to be a hump
+000161b0: 2061 7420 2a4b 2a3c 7375 623e 533c 2f73   at *K*<sub>S</s
+000161c0: 7562 3e20 3120 6275 7420 6e6f 7420 636c  ub> 1 but not cl
+000161d0: 6561 722e 0a0a 215b 5d28 6461 7461 2f41  ear...![](data/A
+000161e0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+000161f0: 612e 7473 762e 6b73 645f 7770 2e73 7667  a.tsv.ksd_wp.svg
+00016200: 290a 0a57 6520 7468 656e 2063 6f6e 7374  )..We then const
+00016210: 7275 6374 2074 6865 2061 6e63 686f 7220  ruct the anchor 
+00016220: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00016230: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+00016240: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
+00016250: 6e64 206c 696e 6520 6265 6c6f 772e 0a0a  nd line below...
+00016260: 6060 600a 7767 6420 7379 6e20 2d66 206d  ```.wgd syn -f m
+00016270: 524e 4120 2d61 204e 616d 6520 7767 645f  RNA -a Name wgd_
+00016280: 646d 642f 4171 7569 6c65 6769 615f 636f  dmd/Aquilegia_co
+00016290: 6572 756c 6561 2e74 7376 2041 7175 696c  erulea.tsv Aquil
+000162a0: 6567 6961 5f63 6f65 7275 6c65 612e 6766  egia_coerulea.gf
+000162b0: 6633 202d 6b73 2077 6764 5f6b 7364 2f41  f3 -ks wgd_ksd/A
+000162c0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+000162d0: 612e 7473 762e 6b73 2e74 7376 0a60 6060  a.tsv.ks.tsv.```
+000162e0: 0a0a 4173 2073 686f 776e 2062 656c 6f77  ..As shown below
+000162f0: 2c20 7468 6572 6520 6172 6520 736f 6d65  , there are some
+00016300: 2072 6574 6169 6e65 6420 616e 6368 6f72   retained anchor
+00016310: 2070 6169 7273 2077 6974 6820 2a4b 2a3c   pairs with *K*<
+00016320: 7375 623e 533c 2f73 7562 3e20 6265 7477  sub>S</sub> betw
+00016330: 6565 6e20 3120 616e 6420 322c 2077 6869  een 1 and 2, whi
+00016340: 6368 2073 6565 6d73 2074 6f20 7375 6767  ch seems to sugg
+00016350: 6573 7420 6120 5747 4420 6576 656e 742e  est a WGD event.
+00016360: 0a0a 215b 5d28 6461 7461 2f41 7175 696c  ..![](data/Aquil
+00016370: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00016380: 762e 6b73 645f 7770 5f61 702e 7376 6729  v.ksd_wp_ap.svg)
+00016390: 0a0a 5468 6520 6173 736f 6369 6174 6564  ..The associated
+000163a0: 2060 6475 7053 7461 636b 6020 706c 6f74   `dupStack` plot
+000163b0: 2073 686f 7773 2074 6861 7420 7468 6572   shows that ther
+000163c0: 6520 6172 6520 6e75 6d65 726f 7573 2064  e are numerous d
+000163d0: 7570 6c69 6361 7465 6420 7365 676d 656e  uplicated segmen
+000163e0: 7473 2061 6372 6f73 7320 6d6f 7374 206f  ts across most o
+000163f0: 6620 7468 6520 6368 726f 6d6f 736f 6d65  f the chromosome
+00016400: 732e 0a0a 215b 5d28 6461 7461 2f73 796e  s...![](data/syn
+00016410: 5f72 6573 756c 7473 2f41 7175 696c 6567  _results/Aquileg
+00016420: 6961 5f63 6f65 7275 6c65 615f 4171 7569  ia_coerulea_Aqui
+00016430: 6c65 6769 615f 636f 6572 756c 6561 5f6d  legia_coerulea_m
+00016440: 756c 7469 706c 6963 6f6e 735f 6c65 7665  ultiplicons_leve
+00016450: 6c2e 706e 6729 0a0a 5765 2069 6d70 6c65  l.png)..We imple
+00016460: 6d65 6e74 6564 2074 776f 2074 7970 6573  mented two types
+00016470: 206f 6620 646f 7420 706c 6f74 7320 696e   of dot plots in
+00016480: 206f 7866 6f72 6420 6772 6964 3a20 6f6e   oxford grid: on
+00016490: 6520 696e 2074 6865 2075 6e69 7420 6f66  e in the unit of
+000164a0: 2062 6173 6573 2061 6e64 2074 6865 206f   bases and the o
+000164b0: 7468 6572 2069 6e20 7468 6520 756e 6974  ther in the unit
+000164c0: 206f 6620 6765 6e65 732c 2077 6869 6368   of genes, which
+000164d0: 2063 616e 2062 6520 636f 6c6f 7265 6420   can be colored 
+000164e0: 6279 202a 4b2a 3c73 7562 3e53 3c2f 7375  by *K*<sub>S</su
+000164f0: 623e 2076 616c 7565 7320 6769 7665 6e20  b> values given 
+00016500: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00016510: 6461 7461 2e0a 0a21 5b5d 2864 6174 612f  data...![](data/
+00016520: 7379 6e5f 7265 7375 6c74 732f 4171 7569  syn_results/Aqui
+00016530: 6c65 6769 615f 636f 6572 756c 6561 2d76  legia_coerulea-v
+00016540: 732d 4171 7569 6c65 6769 615f 636f 6572  s-Aquilegia_coer
+00016550: 756c 6561 5f4b 732e 646f 745f 756e 6974  ulea_Ks.dot_unit
+00016560: 5f67 656e 652e 706e 6729 0a0a 4173 2073  _gene.png)..As s
+00016570: 686f 776e 2061 626f 7665 2c20 7468 6520  hown above, the 
+00016580: 646f 7420 706c 6f74 2069 6e20 7468 6520  dot plot in the 
+00016590: 756e 6974 206f 6620 6765 6e65 7320 7072  unit of genes pr
+000165a0: 6573 656e 7473 206e 756d 6572 6f75 7320  esents numerous 
+000165b0: 6465 6e73 656c 7920 6167 6772 6567 6174  densely aggregat
+000165c0: 6564 2028 6c69 6e65 2d6c 696b 6529 2061  ed (line-like) a
+000165d0: 6e63 686f 7220 706f 696e 7473 2061 7420  nchor points at 
+000165e0: 6d6f 7374 206f 6620 7468 6520 6368 726f  most of the chro
+000165f0: 6d6f 736f 6d65 7320 7769 7468 2063 6f6e  mosomes with con
+00016600: 7369 7374 656e 7420 2a4b 2a3c 7375 623e  sistent *K*<sub>
+00016610: 533c 2f73 7562 3e20 6167 6520 6265 7477  S</sub> age betw
+00016620: 6565 6e20 3120 616e 6420 322e 2054 6865  een 1 and 2. The
+00016630: 2064 6f74 2070 6c6f 7420 696e 2074 6865   dot plot in the
+00016640: 2075 6e69 7420 6f66 2062 6173 6573 2073   unit of bases s
+00016650: 686f 7773 2074 6865 2073 616d 6520 7061  hows the same pa
+00016660: 7474 6572 6e2c 2061 7320 6d61 6e69 6665  ttern, as manife
+00016670: 7374 6564 2062 656c 6f77 2e0a 0a21 5b5d  sted below...![]
+00016680: 2864 6174 612f 7379 6e5f 7265 7375 6c74  (data/syn_result
+00016690: 732f 4171 7569 6c65 6769 615f 636f 6572  s/Aquilegia_coer
+000166a0: 756c 6561 2d76 732d 4171 7569 6c65 6769  ulea-vs-Aquilegi
+000166b0: 615f 636f 6572 756c 6561 5f4b 732e 646f  a_coerulea_Ks.do
+000166c0: 742e 706e 6729 0a0a 5468 6520 646f 7420  t.png)..The dot 
+000166d0: 706c 6f74 7320 7769 7468 6f75 7420 2a4b  plots without *K
+000166e0: 2a3c 7375 623e 533c 2f73 7562 3e20 616e  *<sub>S</sub> an
+000166f0: 6e6f 7461 7469 6f6e 2077 696c 6c20 616c  notation will al
+00016700: 736f 2062 6520 6175 746f 6d61 7465 6c79  so be automately
+00016710: 2070 726f 6475 6365 642c 2061 7320 7368   produced, as sh
+00016720: 6f77 6e20 6265 6c6f 772e 0a0a 215b 5d28  own below...![](
+00016730: 6461 7461 2f73 796e 5f72 6573 756c 7473  data/syn_results
+00016740: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00016750: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
+00016760: 5f63 6f65 7275 6c65 612e 646f 745f 756e  _coerulea.dot_un
+00016770: 6974 5f67 656e 652e 706e 6729 0a0a 215b  it_gene.png)..![
+00016780: 5d28 6461 7461 2f73 796e 5f72 6573 756c  ](data/syn_resul
+00016790: 7473 2f41 7175 696c 6567 6961 5f63 6f65  ts/Aquilegia_coe
+000167a0: 7275 6c65 612d 7673 2d41 7175 696c 6567  rulea-vs-Aquileg
+000167b0: 6961 5f63 6f65 7275 6c65 612e 646f 742e  ia_coerulea.dot.
+000167c0: 706e 6729 0a0a 4e6f 7465 2074 6861 7420  png)..Note that 
+000167d0: 7468 6520 6f70 6163 6974 7920 6f66 2061  the opacity of a
+000167e0: 6e63 686f 7220 646f 7473 2061 6e64 2061  nchor dots and a
+000167f0: 6c6c 2068 6f6d 6f6c 6f67 2064 6f74 7320  ll homolog dots 
+00016800: 6361 6e20 6265 2073 6574 2062 7920 7468  can be set by th
+00016810: 6520 6f70 7469 6f6e 2060 2d2d 6170 616c  e option `--apal
+00016820: 7068 6160 2061 6e64 2060 2d2d 686f 616c  pha` and `--hoal
+00016830: 7068 6160 2073 6570 6172 6174 656c 792e  pha` separately.
+00016840: 2049 6620 6f6e 6520 6a75 7374 2077 616e   If one just wan
+00016850: 7473 2074 6f20 7365 6520 7468 6520 616e  ts to see the an
+00016860: 6368 6f72 2064 6f74 732c 2073 6574 7469  chor dots, setti
+00016870: 6e67 2074 6865 2060 686f 616c 7068 6160  ng the `hoalpha`
+00016880: 2061 7320 3020 286f 7220 6f74 6865 7220   as 0 (or other 
+00016890: 6d69 6e75 7363 756c 6520 7661 6c75 6573  minuscule values
+000168a0: 2920 7769 6c6c 2064 6f2e 2049 6620 6f6e  ) will do. If on
+000168b0: 6520 7761 6e74 7320 746f 2073 6565 2074  e wants to see t
+000168c0: 6865 2064 6973 7472 6962 7574 696f 6e20  he distribution 
+000168d0: 6f66 2077 686f 6c65 2064 6f74 7320 6265  of whole dots be
+000168e0: 7474 6572 2c20 7365 7474 696e 6720 7468  tter, setting th
+000168f0: 6520 6068 6f61 6c70 6861 6020 6869 6768  e `hoalpha` high
+00016900: 6572 2028 616e 6420 6061 7061 6c70 6861  er (and `apalpha
+00016910: 6020 6c6f 7765 7229 2077 696c 6c20 646f  ` lower) will do
+00016920: 2e20 5468 6520 6064 6f74 7369 7a65 6020  . The `dotsize` 
+00016930: 6f70 7469 6f6e 2063 616e 2062 6520 6361  option can be ca
+00016940: 6c6c 6564 2074 6f20 6164 6a75 7374 2074  lled to adjust t
+00016950: 6865 2073 697a 6520 6f66 2064 6f74 732e  he size of dots.
+00016960: 0a0a 4120 6675 7274 6865 7220 6173 736f  ..A further asso
+00016970: 6369 6174 6564 2053 796e 6465 7074 6820  ciated Syndepth 
+00016980: 706c 6f74 2073 686f 7773 2074 6861 7420  plot shows that 
+00016990: 7468 6572 6520 6172 6520 6d6f 7265 2074  there are more t
+000169a0: 6861 6e20 3530 2064 7570 6c69 6361 7465  han 50 duplicate
+000169b0: 6420 7365 676d 656e 7473 206c 6f6e 6765  d segments longe
+000169c0: 7220 7468 616e 2031 3030 3030 2062 7020  r than 10000 bp 
+000169d0: 616e 6420 3330 2067 656e 6573 2028 736f  and 30 genes (so
+000169e0: 2061 7320 746f 2064 726f 7020 6672 6167   as to drop frag
+000169f0: 6d65 6e74 6172 7920 7365 676d 656e 7473  mentary segments
+00016a00: 292c 2077 6869 6368 2064 6f6d 696e 6174  ), which dominat
+00016a10: 6573 2074 6865 2077 686f 6c65 2063 6f6c  es the whole col
+00016a20: 6c69 6e65 6172 2072 6174 696f 2063 6174  linear ratio cat
+00016a30: 6567 6f72 792e 0a0a 215b 5d28 6461 7461  egory...![](data
+00016a40: 2f73 796e 5f72 6573 756c 7473 2f53 796e  /syn_results/Syn
+00016a50: 6465 7074 682e 7376 6729 0a0a 5765 2063  depth.svg)..We c
+00016a60: 616e 2066 6974 2061 6e20 454c 4d4d 206d  an fit an ELMM m
+00016a70: 6978 7475 7265 206d 6f64 656c 2075 706f  ixture model upo
+00016a80: 6e20 7468 6520 7768 6f6c 6520 7061 7261  n the whole para
+00016a90: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+00016aa0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+00016ab0: 7574 696f 6e20 746f 2073 6565 206d 6f72  ution to see mor
+00016ac0: 6520 6163 6375 7261 7465 6c79 2074 6865  e accurately the
+00016ad0: 2073 6967 6e69 6669 6361 6e63 6520 616e   significance an
+00016ae0: 6420 6c6f 6361 7469 6f6e 206f 6620 706f  d location of po
+00016af0: 7465 6e74 6961 6c20 5747 4473 2c20 7573  tential WGDs, us
+00016b00: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
+00016b10: 6c69 6e65 2062 656c 6f77 2e0a 0a60 6060  line below...```
+00016b20: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
+00016b30: 6b73 642f 4171 7569 6c65 6769 615f 636f  ksd/Aquilegia_co
+00016b40: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
+00016b50: 760a 6060 600a 0a54 6865 2072 6573 756c  v.```..The resul
+00016b60: 7420 6f66 2045 4c4d 4d20 6d69 7874 7572  t of ELMM mixtur
+00016b70: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
+00016b80: 6e67 2073 686f 7773 2074 6861 7420 7468  ng shows that th
+00016b90: 6572 6520 6973 2061 206c 696b 656c 7920  ere is a likely 
+00016ba0: 5747 4420 636f 6d70 6f6e 656e 7420 6174  WGD component at
+00016bb0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00016bc0: 2031 2e31 392e 0a0a 215b 5d28 6461 7461   1.19...![](data
+00016bd0: 2f65 6c6d 6d5f 4171 7569 6c65 6769 615f  /elmm_Aquilegia_
+00016be0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00016bf0: 7473 765f 6265 7374 5f6d 6f64 656c 735f  tsv_best_models_
+00016c00: 7765 6967 6874 6564 2e73 7667 290a 0a4c  weighted.svg)..L
+00016c10: 6574 2773 2064 6f20 6120 6d69 7874 7572  et's do a mixtur
+00016c20: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
+00016c30: 6e67 2066 6f72 2061 6e63 686f 7220 2a4b  ng for anchor *K
+00016c40: 2a3c 7375 623e 533c 2f73 7562 3e20 746f  *<sub>S</sub> to
+00016c50: 6f2c 2075 7369 6e67 2074 6865 2063 6f6d  o, using the com
+00016c60: 6d61 6e64 206c 696e 6520 6265 6c6f 772e  mand line below.
+00016c70: 204e 6f74 6520 7468 6174 2074 6869 7320   Note that this 
+00016c80: 7374 6570 2077 696c 6c20 6175 746f 6d61  step will automa
+00016c90: 7465 6c79 2063 616c 6c20 7468 6520 7365  tely call the se
+00016ca0: 676d 656e 7420 2a4b 2a3c 7375 623e 533c  gment *K*<sub>S<
+00016cb0: 2f73 7562 3e20 636c 7573 7465 7269 6e67  /sub> clustering
+00016cc0: 2061 6e61 6c79 7369 7320 746f 6f2e 0a0a   analysis too...
+00016cd0: 6060 600a 7767 6420 7065 616b 2077 6764  ```.wgd peak wgd
+00016ce0: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
+00016cf0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00016d00: 7376 202d 2d61 6e63 686f 7270 6f69 6e74  sv --anchorpoint
+00016d10: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
+00016d20: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
+00016d30: 7473 2e74 7874 202d 2d73 6567 6d65 6e74  ts.txt --segment
+00016d40: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
+00016d50: 652d 6f75 742f 7365 676d 656e 7473 2e74  e-out/segments.t
+00016d60: 7874 202d 2d6c 6973 7465 6c65 6d65 6e74  xt --listelement
+00016d70: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
+00016d80: 652d 6f75 742f 6c69 7374 5f65 6c65 6d65  e-out/list_eleme
+00016d90: 6e74 732e 7478 7420 2d2d 6d75 6c74 6970  nts.txt --multip
+00016da0: 6c69 636f 6e70 6169 7273 2077 6764 5f73  liconpairs wgd_s
+00016db0: 796e 2f69 6164 686f 7265 2d6f 7574 2f6d  yn/iadhore-out/m
+00016dc0: 756c 7469 706c 6963 6f6e 5f70 6169 7273  ultiplicon_pairs
+00016dd0: 2e74 7874 2028 2d2d 7765 6967 6874 6564  .txt (--weighted
+00016de0: 290a 6060 600a 0a54 6865 2061 6e63 686f  ).```..The ancho
+00016df0: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
+00016e00: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00016e10: 6f6e 2061 6c73 6f20 6861 7320 6120 6c69  on also has a li
+00016e20: 6b65 6c79 2057 4744 2063 6f6d 706f 6e65  kely WGD compone
+00016e30: 6e74 2077 6974 6820 6d6f 6465 2031 2e32  nt with mode 1.2
+00016e40: 382e 0a0a 215b 5d28 6461 7461 2f4f 7269  8...![](data/Ori
+00016e50: 6769 6e61 6c5f 416e 6368 6f72 4b73 5f47  ginal_AnchorKs_G
+00016e60: 4d4d 5f43 6f6d 706f 6e65 6e74 335f 6e6f  MM_Component3_no
+00016e70: 6465 5f61 7665 7261 6765 645f 4c6f 676e  de_averaged_Logn
+00016e80: 6f72 6d61 6c2e 7376 6729 0a0a 4e6f 7720  ormal.svg)..Now 
+00016e90: 7468 6174 2077 6520 6861 7665 2073 6565  that we have see
+00016ea0: 6e20 7468 6520 6576 6964 656e 6365 206f  n the evidence o
+00016eb0: 6620 6e75 6d65 726f 7573 2064 7570 6c69  f numerous dupli
+00016ec0: 6361 7465 6420 7365 676d 656e 7473 2061  cated segments a
+00016ed0: 6e64 2074 6865 2061 6767 7265 6761 7469  nd the aggregati
+00016ee0: 6f6e 206f 6620 6475 706c 6963 6174 6573  on of duplicates
+00016ef0: 2061 6765 2061 7420 2a4b 2a3c 7375 623e   age at *K*<sub>
+00016f00: 533c 2f73 7562 3e20 312e 3238 206f 7220  S</sub> 1.28 or 
+00016f10: 312e 3139 2066 6f72 2061 6e63 686f 7220  1.19 for anchor 
+00016f20: 7061 6972 7320 616e 6420 6e6f 6e2d 616e  pairs and non-an
+00016f30: 6368 6f72 2070 6169 7273 2074 6872 6f75  chor pairs throu
+00016f40: 6768 6f75 7420 7468 6520 7768 6f6c 6520  ghout the whole 
+00016f50: 6765 6e6f 6d65 2e20 5765 2063 616e 2063  genome. We can c
+00016f60: 6c61 696d 2077 6974 6820 736f 6d65 2063  laim with some c
+00016f70: 6f6e 6669 6465 6e63 6520 7468 6174 202a  onfidence that *
+00016f80: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
+00016f90: 6561 2a20 6d69 6768 7420 6861 7665 2065  ea* might have e
+00016fa0: 7870 6572 6965 6e63 6564 2061 2070 616c  xperienced a pal
+00016fb0: 656f 2d70 6f6c 7970 6c6f 6964 697a 6174  eo-polyploidizat
+00016fc0: 696f 6e20 6576 656e 742e 204e 6578 742c  ion event. Next,
+00016fd0: 204c 6574 2773 2068 6176 6520 6120 6675   Let's have a fu
+00016fe0: 7274 6865 7220 6c6f 6f6b 2061 626f 7574  rther look about
+00016ff0: 2069 7473 2070 6879 6c6f 6765 6e65 7469   its phylogeneti
+00017000: 6320 6c6f 6361 7469 6f6e 2e20 5765 206b  c location. We k
+00017010: 6e6f 7720 7468 6174 2074 6865 7265 2061  now that there a
+00017020: 7265 2075 6e63 6572 7461 696e 7469 6573  re uncertainties
+00017030: 2061 626f 7574 2077 6865 7468 6572 2074   about whether t
+00017040: 6869 7320 7075 7461 7469 7665 2070 616c  his putative pal
+00017050: 656f 2d70 6f6c 7970 6c6f 6964 697a 6174  eo-polyploidizat
+00017060: 696f 6e20 6576 656e 7420 6973 2073 6861  ion event is sha
+00017070: 7265 6420 7769 7468 2061 6c6c 2065 7564  red with all eud
+00017080: 6963 6f74 7320 6f72 206e 6f74 2e20 5765  icots or not. We
+00017090: 2063 616e 2063 686f 6f73 6520 736f 6d65   can choose some
+000170a0: 206f 7468 6572 2065 7564 6963 6f74 2067   other eudicot g
+000170b0: 656e 6f6d 6573 2074 6f20 7365 6520 7468  enomes to see th
+000170c0: 6520 6f72 6465 7269 6e67 206f 6620 7370  e ordering of sp
+000170d0: 6563 6961 7469 6f6e 2061 6e64 2070 6f6c  eciation and pol
+000170e0: 7970 6c6f 6964 697a 6174 696f 6e20 6576  yploidization ev
+000170f0: 656e 7473 2e20 4865 7265 2077 6520 6368  ents. Here we ch
+00017100: 6f6f 7365 202a 5669 7469 7320 7669 6e69  oose *Vitis vini
+00017110: 6665 7261 2a2c 202a 5072 6f74 6561 2063  fera*, *Protea c
+00017120: 796e 6172 6f69 6465 732a 2061 6e64 202a  ynaroides* and *
+00017130: 4163 6f72 7573 2061 6d65 7269 6361 6e75  Acorus americanu
+00017140: 732a 2069 6e20 7468 6520 666f 6c6c 6f77  s* in the follow
+00017150: 696e 6720 2a4b 2a3c 7375 623e 533c 2f73  ing *K*<sub>S</s
+00017160: 7562 3e20 616e 616c 7973 6973 2e20 4669  ub> analysis. Fi
+00017170: 7273 742c 2077 6520 6275 696c 7420 6120  rst, we built a 
+00017180: 676c 6f62 616c 204d 5242 4820 6661 6d69  global MRBH fami
+00017190: 6c79 2075 7369 6e67 2074 6865 2063 6f6d  ly using the com
+000171a0: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
+000171b0: 0a77 6764 2064 6d64 202d 2d67 6c6f 6261  .wgd dmd --globa
+000171c0: 6c6d 7262 6820 4171 7569 6c65 6769 615f  lmrbh Aquilegia_
+000171d0: 636f 6572 756c 6561 2050 726f 7465 615f  coerulea Protea_
+000171e0: 6379 6e61 726f 6964 6573 2041 636f 7275  cynaroides Acoru
+000171f0: 735f 616d 6572 6963 616e 7573 2056 6974  s_americanus Vit
+00017200: 6973 5f76 696e 6966 6572 6120 2d6f 2077  is_vinifera -o w
+00017210: 6764 5f67 6c6f 6261 6c6d 7262 680a 6060  gd_globalmrbh.``
+00017220: 600a 0a49 6e20 7468 6520 676c 6f62 616c  `..In the global
+00017230: 204d 5242 4820 6661 6d69 6c79 2c20 6576   MRBH family, ev
+00017240: 6572 7920 7061 6972 206f 6620 6f72 7468  ery pair of orth
+00017250: 6f6c 6f67 6f75 7320 6765 6e65 7320 6973  ologous genes is
+00017260: 2074 6865 2072 6563 6970 726f 6361 6c20   the reciprocal 
+00017270: 6265 7374 2068 6974 2c20 7375 6767 6573  best hit, sugges
+00017280: 7469 6e67 2074 7275 6520 6f72 7468 6f6c  ting true orthol
+00017290: 6f67 6f75 7320 7265 6c61 7469 6f6e 7368  ogous relationsh
+000172a0: 6970 732e 2057 6520 776f 756c 6420 7573  ips. We would us
+000172b0: 6520 7468 6520 2a4b 2a3c 7375 623e 533c  e the *K*<sub>S<
+000172c0: 2f73 7562 3e20 7661 6c75 6573 2061 7373  /sub> values ass
+000172d0: 6f63 6961 7465 6420 7769 7468 2074 6865  ociated with the
+000172e0: 7365 206f 7274 686f 6c6f 676f 7573 2070  se orthologous p
+000172f0: 6169 7273 2074 6f20 6465 6c69 6d69 7420  airs to delimit 
+00017300: 7468 6520 6469 7665 7267 656e 6365 202a  the divergence *
+00017310: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
+00017320: 6561 6b2e 2054 6f67 6574 6865 7220 7769  eak. Together wi
+00017330: 7468 2074 6865 2077 686f 6c65 2070 6172  th the whole par
+00017340: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00017350: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00017360: 6f6e 2c20 7765 2063 6f6e 6475 6374 2074  on, we conduct t
+00017370: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+00017380: 6f6e 2075 7369 6e67 2074 6865 2063 6f6d  on using the com
+00017390: 6d61 6e64 2062 656c 6f77 2e0a 0a21 2153  mand below...!!S
+000173a0: 696e 6365 2060 7767 6460 2076 6572 7369  ince `wgd` versi
+000173b0: 6f6e 2032 2e30 2e32 342c 2077 6520 7265  on 2.0.24, we re
+000173c0: 7772 6f74 6520 6120 636c 6561 6e65 7220  wrote a cleaner 
+000173d0: 616e 6420 7175 6963 6b65 7220 7761 7920  and quicker way 
+000173e0: 6f66 2064 6f69 6e67 2073 7562 7374 6974  of doing substit
+000173f0: 7574 696f 6e20 7261 7465 2063 6f72 7265  ution rate corre
+00017400: 6374 696f 6e2e 2049 7427 7320 6e6f 7420  ction. It's not 
+00017410: 7265 7175 6972 6564 2074 6f20 7479 7065  required to type
+00017420: 2069 6e20 616e 7920 7370 6569 6365 7320   in any speices 
+00017430: 7061 6972 2061 6e64 2061 2073 6572 6965  pair and a serie
+00017440: 7320 6f66 202a 4b2a 3c73 7562 3e53 3c2f  s of *K*<sub>S</
+00017450: 7375 623e 2070 6c6f 7473 2077 696c 6c20  sub> plots will 
+00017460: 6265 2070 726f 6475 6365 642e 2054 6865  be produced. The
+00017470: 2072 6571 7569 7265 6420 6669 6c65 7320   required files 
+00017480: 6172 6520 6f72 7468 6f6c 6f67 6f75 7320  are orthologous 
+00017490: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000174a0: 6461 7461 6669 6c65 2c20 7061 7261 6c6f  datafile, paralo
+000174b0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+000174c0: 7375 623e 2064 6174 6166 696c 652c 2061  sub> datafile, a
+000174d0: 2073 7065 6369 6573 2074 7265 6520 616e   species tree an
+000174e0: 6420 6120 666f 6361 6c20 7370 6563 6965  d a focal specie
+000174f0: 7320 2874 6865 206f 6e65 2069 6e70 7574  s (the one input
+00017500: 7465 6420 7769 7468 2070 6172 616c 6f67  ted with paralog
+00017510: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00017520: 7562 3e20 6461 7461 292e 2055 7365 7273  ub> data). Users
+00017530: 2063 616e 2063 686f 6f73 6520 746f 2061   can choose to a
+00017540: 6464 206f 6e65 206d 6f72 6520 6c61 7965  dd one more laye
+00017550: 7220 6f66 2045 4c4d 4d20 616e 616c 7973  r of ELMM analys
+00017560: 6973 206f 6e20 7061 7261 6c6f 676f 7573  is on paralogous
+00017570: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00017580: 2076 616c 7565 7320 616e 642f 6f72 2047   values and/or G
+00017590: 4d4d 2061 6e61 6c79 7369 7320 6f6e 2061  MM analysis on a
+000175a0: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
+000175b0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+000175c0: 6f6e 2e20 5468 6520 6f72 7468 6f6c 6f67  on. The ortholog
+000175d0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+000175e0: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
+000175f0: 2063 616e 2062 6520 6361 6c63 756c 6174   can be calculat
+00017600: 6564 2075 7369 6e67 2074 6865 2063 6f6d  ed using the com
+00017610: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
+00017620: 0a77 6764 206b 7364 2077 6764 5f67 6c6f  .wgd ksd wgd_glo
+00017630: 6261 6c6d 7262 682f 676c 6f62 616c 5f4d  balmrbh/global_M
+00017640: 5242 482e 7473 7620 4171 7569 6c65 6769  RBH.tsv Aquilegi
+00017650: 615f 636f 6572 756c 6561 2050 726f 7465  a_coerulea Prote
+00017660: 615f 6379 6e61 726f 6964 6573 2041 636f  a_cynaroides Aco
+00017670: 7275 735f 616d 6572 6963 616e 7573 2056  rus_americanus V
+00017680: 6974 6973 5f76 696e 6966 6572 6120 2d6f  itis_vinifera -o
+00017690: 2077 6764 5f67 6c6f 6261 6c6d 7262 685f   wgd_globalmrbh_
+000176a0: 6b73 0a60 6060 0a0a 5769 7468 2074 6865  ks.```..With the
+000176b0: 2063 616c 6375 6c61 7465 6420 6f72 7468   calculated orth
+000176c0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000176d0: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+000176e0: 7469 6f6e 2c20 7765 2063 616e 2075 7365  tion, we can use
+000176f0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00017700: 6f77 2074 6f20 636f 6e64 7563 7420 7468  ow to conduct th
+00017710: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+00017720: 6e20 616e 642f 6f72 206d 6978 7475 7265  n and/or mixture
+00017730: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
+00017740: 6973 2e0a 0a60 6060 0a77 6764 2076 697a  is...```.wgd viz
+00017750: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
+00017760: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
+00017770: 482e 7473 762e 6b73 2e74 7376 202d 6661  H.tsv.ks.tsv -fa
+00017780: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+00017790: 6c65 6120 2d65 706b 2077 6764 5f6b 7364  lea -epk wgd_ksd
+000177a0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000177b0: 6c65 612e 6b73 2e74 7376 202d 6170 2077  lea.ks.tsv -ap w
+000177c0: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+000177d0: 7574 2f61 6e63 686f 7270 6f69 6e74 732e  ut/anchorpoints.
+000177e0: 7478 7420 2d73 7020 7370 6563 6965 7374  txt -sp speciest
+000177f0: 7265 652e 6e77 202d 6f20 7767 645f 7669  ree.nw -o wgd_vi
+00017800: 7a5f 6d69 7865 645f 4b73 202d 2d70 6c6f  z_mixed_Ks --plo
+00017810: 7465 6c6d 6d20 2d2d 706c 6f74 6170 676d  telmm --plotapgm
+00017820: 6d20 2d2d 7265 7765 6967 6874 0a60 6060  m --reweight.```
+00017830: 0a0a 6f72 2075 7369 6e67 2074 6865 2063  ..or using the c
+00017840: 6f6d 6d61 6e64 2062 656c 6f77 2c20 7768  ommand below, wh
+00017850: 6963 6820 636f 6d62 696e 6573 2074 6865  ich combines the
+00017860: 2074 776f 2073 7465 7073 2061 626f 7665   two steps above
+00017870: 2069 6e20 6f6e 652e 204e 6f74 6520 7468   in one. Note th
+00017880: 6174 2077 6520 7375 6767 6573 7420 6f66  at we suggest of
+00017890: 2074 616b 696e 6720 7477 6f20 7365 7061   taking two sepa
+000178a0: 7261 7465 2073 7465 7073 2069 6e20 7768  rate steps in wh
+000178b0: 6963 6820 6077 6764 206b 7364 6020 756e  ich `wgd ksd` un
+000178c0: 6465 7274 616b 6573 2074 6865 2063 616c  dertakes the cal
+000178d0: 6375 6c61 7469 6f6e 206f 6620 6f72 7468  culation of orth
+000178e0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000178f0: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00017900: 7469 6f6e 2077 6869 6c65 2060 7767 6420  tion while `wgd 
+00017910: 7669 7a60 2063 6172 7269 6573 206f 7574  viz` carries out
+00017920: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+00017930: 7469 6f6e 2061 6e64 2047 4d4d 2061 6e61  tion and GMM ana
+00017940: 6c79 7369 7320 7375 6368 2074 6861 7420  lysis such that 
+00017950: 6974 2773 2065 6173 6965 7220 746f 2064  it's easier to d
+00017960: 6562 7567 2e0a 6060 600a 7767 6420 6b73  ebug..```.wgd ks
+00017970: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
+00017980: 2f67 6c6f 6261 6c5f 4d52 4248 2e74 7376  /global_MRBH.tsv
+00017990: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+000179a0: 6c65 6120 5072 6f74 6561 5f63 796e 6172  lea Protea_cynar
+000179b0: 6f69 6465 7320 4163 6f72 7573 5f61 6d65  oides Acorus_ame
+000179c0: 7269 6361 6e75 7320 5669 7469 735f 7669  ricanus Vitis_vi
+000179d0: 6e69 6665 7261 202d 2d65 7874 7261 7061  nifera --extrapa
+000179e0: 7261 6e6f 6d65 6b73 2077 6764 5f6b 7364  ranomeks wgd_ksd
+000179f0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00017a00: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00017a10: 7370 2073 7065 6369 6573 7472 6565 2e6e  sp speciestree.n
+00017a20: 7720 2d2d 7265 7765 6967 6874 202d 6f20  w --reweight -o 
+00017a30: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
+00017a40: 735f 7261 7465 5f63 6f72 7265 6374 696f  s_rate_correctio
+00017a50: 6e20 2d66 6120 4171 7569 6c65 6769 615f  n -fa Aquilegia_
+00017a60: 636f 6572 756c 6561 202d 6170 2077 6764  coerulea -ap wgd
+00017a70: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
+00017a80: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
+00017a90: 7420 2d2d 706c 6f74 656c 6d6d 202d 2d70  t --plotelmm --p
+00017aa0: 6c6f 7461 7067 6d6d 0a60 6060 0a0a 5468  lotapgmm.```..Th
+00017ab0: 6520 6669 6c65 2060 7370 6563 6965 7374  e file `speciest
+00017ac0: 7265 652e 6e77 6020 6973 2074 6865 2074  ree.nw` is the t
+00017ad0: 6578 7420 6669 6c65 206f 6620 7370 6563  ext file of spec
+00017ae0: 6965 7320 7472 6565 2069 6e20 6e65 7769  ies tree in newi
+00017af0: 636b 2074 6861 7420 7261 7465 2063 6f72  ck that rate cor
+00017b00: 7265 6374 696f 6e20 776f 756c 6420 6265  rection would be
+00017b10: 2063 6f6e 6475 6374 6564 206f 6e2e 2049   conducted on. I
+00017b20: 7473 2063 6f6e 7465 6e74 2069 7320 6173  ts content is as
+00017b30: 2062 656c 6f77 2e20 5573 6572 7320 6361   below. Users ca
+00017b40: 6e20 6f70 7469 6f6e 616c 6c79 2070 726f  n optionally pro
+00017b50: 7669 6465 2074 6865 2073 7065 6369 6573  vide the species
+00017b60: 2070 6169 7273 2074 6f20 6265 2070 6c6f   pairs to be plo
+00017b70: 7474 6564 2062 7574 2077 6520 7375 6767  tted but we sugg
+00017b80: 6573 7420 6f66 206a 7573 7420 7573 696e  est of just usin
+00017b90: 6720 602d 6661 2041 7175 696c 6567 6961  g `-fa Aquilegia
+00017ba0: 5f63 6f65 7275 6c65 6160 2074 6f20 706c  _coerulea` to pl
+00017bb0: 6f74 2061 6c6c 2070 6f73 7369 626c 6520  ot all possible 
+00017bc0: 666f 6361 6c2d 7369 7374 6572 2073 7065  focal-sister spe
+00017bd0: 6369 6573 2070 6169 7273 2e20 5765 2073  cies pairs. We s
+00017be0: 7567 6765 7374 2061 6464 696e 6720 7468  uggest adding th
+00017bf0: 6520 6f70 7469 6f6e 2060 2d2d 7265 7765  e option `--rewe
+00017c00: 6967 6874 6020 746f 2072 6563 616c 6375  ight` to recalcu
+00017c10: 6c61 7465 2074 6865 2077 6569 6768 7420  late the weight 
+00017c20: 7065 7220 7370 6563 6965 7320 7061 6972  per species pair
+00017c30: 2073 7563 6820 7468 6174 2074 6865 2077   such that the w
+00017c40: 6569 6768 7420 6f66 206f 7274 686f 6c6f  eight of ortholo
+00017c50: 676f 7573 2067 656e 6520 7061 6972 7320  gous gene pairs 
+00017c60: 7769 6c6c 2062 6563 6f6d 6520 312e 2045  will become 1. E
+00017c70: 7874 7261 2063 6f6c 6c69 6e65 6172 2064  xtra collinear d
+00017c80: 6174 6120 6361 6e20 6265 2061 6464 6564  ata can be added
+00017c90: 2062 7920 7468 6520 6f70 7469 6f6e 2060   by the option `
+00017ca0: 2d61 7060 2061 6e64 2061 6464 6974 696f  -ap` and additio
+00017cb0: 6e61 6c20 636c 7573 7465 7269 6e67 2061  nal clustering a
+00017cc0: 6e61 6c79 7369 7320 6361 6e20 6265 2069  nalysis can be i
+00017cd0: 6e69 7469 6174 6564 2062 7920 7365 7474  nitiated by sett
+00017ce0: 696e 6720 7468 6520 6f70 7469 6f6e 2060  ing the option `
+00017cf0: 2d2d 706c 6f74 6170 676d 6d60 2061 6e64  --plotapgmm` and
+00017d00: 2060 2d2d 706c 6f74 656c 6d6d 602e 0a0a   `--plotelmm`...
+00017d10: 6060 600a 2828 2856 6974 6973 5f76 696e  ```.(((Vitis_vin
+00017d20: 6966 6572 612c 5072 6f74 6561 5f63 796e  ifera,Protea_cyn
+00017d30: 6172 6f69 6465 7329 2c41 7175 696c 6567  aroides),Aquileg
+00017d40: 6961 5f63 6f65 7275 6c65 6129 2c41 636f  ia_coerulea),Aco
+00017d50: 7275 735f 616d 6572 6963 616e 7573 293b  rus_americanus);
+00017d60: 0a60 6060 0a0a 215b 5d28 6461 7461 2f4d  .```..![](data/M
+00017d70: 6978 6564 2e6b 732e 4171 7569 6c65 6769  ixed.ks.Aquilegi
+00017d80: 615f 636f 6572 756c 6561 2e6e 6f64 652e  a_coerulea.node.
+00017d90: 7765 6967 6874 6564 2e70 6e67 290a 0a54  weighted.png)..T
+00017da0: 6865 206d 6978 6564 202a 4b2a 3c73 7562  he mixed *K*<sub
+00017db0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00017dc0: 7574 696f 6e20 7368 6f77 6e20 6162 6f76  ution shown abov
+00017dd0: 6520 6973 2061 2070 7562 6c69 6361 7469  e is a publicati
+00017de0: 6f6e 2d72 6561 6479 2066 6967 7572 6520  on-ready figure 
+00017df0: 7468 6174 2061 7373 656d 626c 6573 2074  that assembles t
+00017e00: 6865 2072 6573 756c 7473 206f 6620 454c  he results of EL
+00017e10: 4d4d 2c20 474d 4d20 616e 6420 7261 7465  MM, GMM and rate
+00017e20: 2063 6f72 7265 6374 696f 6e2e 2054 6865   correction. The
+00017e30: 206f 6e65 2d76 732d 6f6e 6520 6f72 7468   one-vs-one orth
+00017e40: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00017e50: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00017e60: 7469 6f6e 7320 6973 2061 6c73 6f20 6175  tions is also au
+00017e70: 746f 6d61 7469 6361 6c6c 7920 7072 6f64  tomatically prod
+00017e80: 7563 6564 2077 6974 6820 7261 7465 2063  uced with rate c
+00017e90: 6f72 7265 6374 696f 6e20 7265 7375 6c74  orrection result
+00017ea0: 7320 7375 7065 7269 6d70 6f73 6564 2077  s superimposed w
+00017eb0: 6865 7265 2061 7661 696c 6162 6c65 2061  here available a
+00017ec0: 7320 7368 6f77 6e20 6265 6c6f 772e 0a0a  s shown below...
+00017ed0: 215b 5d28 6461 7461 2f46 6f63 7573 5f73  ![](data/Focus_s
+00017ee0: 6973 7465 725f 7061 6972 732e 6b73 2e6e  ister_pairs.ks.n
+00017ef0: 6f64 652e 7765 6967 6874 6564 2e70 6e67  ode.weighted.png
+00017f00: 290a 215b 5d28 6461 7461 2f41 6c6c 5f70  ).![](data/All_p
+00017f10: 6169 7273 2e6b 732e 6e6f 6465 2e77 6569  airs.ks.node.wei
+00017f20: 6768 7465 642e 706e 6729 0a0a 4265 7369  ghted.png)..Besi
+00017f30: 6465 7320 7468 6520 6162 6f76 6520 6b73  des the above ks
+00017f40: 7261 7465 732d 6c69 6b65 202a 4b2a 3c73  rates-like *K*<s
+00017f50: 7562 3e53 3c2f 7375 623e 2070 6c6f 742c  ub>S</sub> plot,
+00017f60: 2061 206d 6f72 6520 636c 6173 7369 6320   a more classic 
+00017f70: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00017f80: 706c 6f74 2063 616e 2062 6520 6d61 6465  plot can be made
+00017f90: 2062 7920 6164 6469 6e67 2074 6865 206f   by adding the o
+00017fa0: 7074 696f 6e20 602d 2d63 6c61 7373 6963  ption `--classic
+00017fb0: 6020 746f 2074 6170 206d 6f72 6520 6465  ` to tap more de
+00017fc0: 7461 696c 6564 6c79 2069 6e74 6f20 7468  tailedly into th
+00017fd0: 6520 7661 7269 6174 696f 6e20 6f66 2073  e variation of s
+00017fe0: 796e 6f6e 796d 6f75 7320 7375 6273 7469  ynonymous substi
+00017ff0: 7475 7469 6f6e 2072 6174 652e 0a0a 5573  tution rate...Us
+00018000: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
+00018010: 6265 6c6f 772c 2074 6865 2064 6972 6563  below, the direc
+00018020: 7469 6f6e 206f 6620 7261 7465 2063 6f72  tion of rate cor
+00018030: 7265 6374 696f 6e20 616e 6420 6465 6772  rection and degr
+00018040: 6565 206f 6620 7261 7465 2076 6172 6961  ee of rate varia
+00018050: 7469 6f6e 2063 616e 2062 6520 6f62 7365  tion can be obse
+00018060: 7276 6564 206d 6f72 6520 6469 7265 6374  rved more direct
+00018070: 6c79 2e0a 0a60 6060 0a77 6764 2076 697a  ly...```.wgd viz
+00018080: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
+00018090: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
+000180a0: 482e 7473 762e 6b73 2e74 7376 202d 6661  H.tsv.ks.tsv -fa
+000180b0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+000180c0: 6c65 6120 2d65 706b 2077 6764 5f6b 7364  lea -epk wgd_ksd
+000180d0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000180e0: 6c65 612e 6b73 2e74 7376 202d 6170 2077  lea.ks.tsv -ap w
+000180f0: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+00018100: 7574 2f61 6e63 686f 7270 6f69 6e74 732e  ut/anchorpoints.
+00018110: 7478 7420 2d73 7020 7370 6563 6965 7374  txt -sp speciest
+00018120: 7265 652e 6e77 202d 6f20 7767 645f 7669  ree.nw -o wgd_vi
+00018130: 7a5f 6d69 7865 645f 4b73 202d 2d70 6c6f  z_mixed_Ks --plo
+00018140: 7465 6c6d 6d20 2d2d 706c 6f74 6170 676d  telmm --plotapgm
+00018150: 6d20 2d2d 7265 7765 6967 6874 202d 2d70  m --reweight --p
+00018160: 6c6f 746b 6465 202d 2d63 6c61 7373 6963  lotkde --classic
+00018170: 0a60 6060 0a0a 215b 5d28 6461 7461 2f41  .```..![](data/A
+00018180: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00018190: 615f 476c 6f62 616c 6d72 6268 4b73 5f45  a_GlobalmrbhKs_E
+000181a0: 6c6d 6d5f 4170 676d 6d5f 436f 7272 6563  lmm_Apgmm_Correc
+000181b0: 7465 642e 6b73 642e 7376 6729 0a0a 4173  ted.ksd.svg)..As
+000181c0: 2073 686f 776e 2061 626f 7665 2c20 6265   shown above, be
+000181d0: 6361 7573 6520 6f66 2074 6865 2068 6967  cause of the hig
+000181e0: 6865 7220 7375 6273 7469 7475 7469 6f6e  her substitution
+000181f0: 2072 6174 6520 6f66 202a 4171 7569 6c65   rate of *Aquile
+00018200: 6769 6120 636f 6572 756c 6561 2a2c 2074  gia coerulea*, t
+00018210: 6865 206f 7269 6769 6e61 6c20 6f72 7468  he original orth
+00018220: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00018230: 533c 2f73 7562 3e20 7661 6c75 6573 2077  S</sub> values w
+00018240: 6572 6520 6163 7475 616c 6c79 2075 6e64  ere actually und
+00018250: 6572 6573 7469 6d61 7465 6420 696e 2074  erestimated in t
+00018260: 6865 2074 696d 652d 6672 616d 6520 6f66  he time-frame of
+00018270: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
+00018280: 756c 6561 2a2e 2057 6865 6e20 7765 2072  ulea*. When we r
+00018290: 6563 6f76 6572 6564 2074 6865 2064 6976  ecovered the div
+000182a0: 6572 6765 6e63 6520 7375 6273 7469 7475  ergence substitu
+000182b0: 7469 6f6e 2064 6973 7461 6e63 6520 696e  tion distance in
+000182c0: 2074 6572 6d73 206f 6620 7477 6f20 7469   terms of two ti
+000182d0: 6d65 7320 6f66 2074 6865 2062 7261 6e63  mes of the branc
+000182e0: 682d 7370 6563 6966 6963 2063 6f6e 7472  h-specific contr
+000182f0: 6962 7574 696f 6e20 6f66 202a 412e 2063  ibution of *A. c
+00018300: 6f65 7275 6c65 612a 2073 696e 6365 2069  oerulea* since i
+00018310: 7473 2064 6976 6572 6765 6e63 6520 7769  ts divergence wi
+00018320: 7468 2074 6865 2073 6973 7465 7220 7370  th the sister sp
+00018330: 6563 6965 7320 706c 7573 2074 6865 2073  ecies plus the s
+00018340: 6861 7265 6420 7375 6273 7469 7475 7469  hared substituti
+00018350: 6f6e 2064 6973 7461 6e63 6520 6265 666f  on distance befo
+00018360: 7265 2064 6976 6572 6765 6e63 6520 2869  re divergence (i
+00018370: 6e20 7265 6c61 7469 7665 2074 6f20 7468  n relative to th
+00018380: 6520 6f75 7467 726f 7570 292c 2074 6865  e outgroup), the
+00018390: 2063 6f72 7265 6374 6564 202a 4b2a 3c73   corrected *K*<s
+000183a0: 7562 3e53 3c2f 7375 623e 206d 6f64 6520  ub>S</sub> mode 
+000183b0: 6265 6361 6d65 206c 6172 6765 722e 0a0a  became larger...
+000183c0: 4e6f 7465 2074 6861 7420 7765 2063 616e  Note that we can
+000183d0: 2065 6173 696c 7920 7368 6f77 2074 6861   easily show tha
+000183e0: 7420 2a41 7175 696c 6567 6961 2063 6f65  t *Aquilegia coe
+000183f0: 7275 6c65 612a 2068 6173 2068 6967 6865  rulea* has highe
+00018400: 7220 7375 6273 7469 7475 7469 6f6e 2072  r substitution r
+00018410: 6174 6520 7468 616e 202a 5072 6f74 6561  ate than *Protea
+00018420: 2063 796e 6172 6f69 6465 732a 2061 6e64   cynaroides* and
+00018430: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
+00018440: 2a20 6279 2063 6f6d 7061 7269 6e67 2074  * by comparing t
+00018450: 6865 6972 2073 7562 7374 6974 7574 696f  heir substitutio
+00018460: 6e20 6469 7374 616e 6365 2069 6e20 7265  n distance in re
+00018470: 6761 7264 2074 6f20 7468 6520 7361 6d65  gard to the same
+00018480: 2064 6976 6572 6765 6e63 6520 6576 656e   divergence even
+00018490: 7420 7769 7468 206f 7574 6772 6f75 7020  t with outgroup 
+000184a0: 7370 6563 6965 7320 2a41 636f 7275 735f  species *Acorus_
+000184b0: 616d 6572 6963 616e 7573 2a2c 2075 7369  americanus*, usi
+000184c0: 6e67 2063 6f6d 6d61 6e64 2062 656c 6f77  ng command below
+000184d0: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
+000184e0: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
+000184f0: 5f6b 732f 676c 6f62 616c 5f4d 5242 482e  _ks/global_MRBH.
+00018500: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
+00018510: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
+00018520: 7265 7765 6967 6874 202d 6f20 7767 645f  reweight -o wgd_
+00018530: 7669 7a5f 436f 6d70 6172 655f 7261 7465  viz_Compare_rate
+00018540: 202d 2d73 7061 6972 2022 4163 6f72 7573   --spair "Acorus
+00018550: 5f61 6d65 7269 6361 6e75 733b 5072 6f74  _americanus;Prot
+00018560: 6561 5f63 796e 6172 6f69 6465 7322 202d  ea_cynaroides" -
+00018570: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00018580: 615f 636f 6572 756c 6561 3b41 636f 7275  a_coerulea;Acoru
+00018590: 735f 616d 6572 6963 616e 7573 2220 2d2d  s_americanus" --
+000185a0: 7370 6169 7220 2256 6974 6973 5f76 696e  spair "Vitis_vin
+000185b0: 6966 6572 613b 4163 6f72 7573 5f61 6d65  ifera;Acorus_ame
+000185c0: 7269 6361 6e75 7322 202d 2d70 6c6f 746b  ricanus" --plotk
+000185d0: 6465 202d 2d63 6c61 7373 6963 0a60 6060  de --classic.```
+000185e0: 0a0a 215b 5d28 6461 7461 2f52 6177 5f4f  ..![](data/Raw_O
+000185f0: 7274 686f 6c6f 6775 6573 5f43 6f6d 7061  rthologues_Compa
+00018600: 7265 5f72 6174 652e 6b73 642e 7376 6729  re_rate.ksd.svg)
+00018610: 0a0a 4173 2064 6973 706c 6179 6564 2061  ..As displayed a
+00018620: 626f 7665 2c20 7468 6520 6f72 7468 6f6c  bove, the orthol
+00018630: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+00018640: 2f73 7562 3e20 7661 6c75 6573 2062 6577  /sub> values bew
+00018650: 7465 656e 202a 4171 7569 6c65 6769 6120  teen *Aquilegia 
+00018660: 636f 6572 756c 6561 2a20 616e 6420 2a41  coerulea* and *A
+00018670: 636f 7275 7320 616d 6572 6963 616e 7573  corus americanus
+00018680: 2a20 6861 7320 7468 6520 6869 6768 6573  * has the highes
+00018690: 7420 6d6f 6465 2c20 696e 6469 6361 7469  t mode, indicati
+000186a0: 6e67 2074 6865 2066 6173 7465 7220 7375  ng the faster su
+000186b0: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
+000186c0: 6f66 202a 412e 2063 6f65 7275 6c65 612a  of *A. coerulea*
+000186d0: 2063 6f6d 7061 7265 6420 746f 202a 5072   compared to *Pr
+000186e0: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
+000186f0: 2061 6e64 202a 5669 7469 7320 7669 6e69   and *Vitis vini
+00018700: 6665 7261 2a2e 0a0a 4265 666f 7265 2076  fera*...Before v
+00018710: 322e 302e 3231 2c20 7468 6520 6765 6e65  2.0.21, the gene
+00018720: 2d73 7065 6369 6573 206d 6170 2066 696c  -species map fil
+00018730: 6520 6973 206e 6563 6365 7373 6172 696c  e is neccessaril
+00018740: 7920 6e65 6564 6564 2066 6f72 2069 7473  y needed for its
+00018750: 2069 6d70 6c65 6d65 6e74 6174 696f 6e20   implementation 
+00018760: 696e 2060 7767 6420 7669 7a60 2c20 7768  in `wgd viz`, wh
+00018770: 6963 6820 7368 6f75 6c64 2062 6520 6175  ich should be au
+00018780: 746f 6d61 7465 6c79 2070 726f 6475 6365  tomately produce
+00018790: 6420 6279 2074 6865 206c 6173 7420 6077  d by the last `w
+000187a0: 6764 206b 7364 6020 7374 6570 2067 6976  gd ksd` step giv
+000187b0: 656e 2074 6865 2060 7370 6169 7260 2061  en the `spair` a
+000187c0: 6e64 2060 7370 6563 6965 7374 7265 6560  nd `speciestree`
+000187d0: 2070 6172 616d 6574 6572 732e 2054 6865   parameters. The
+000187e0: 2060 6765 6e65 5f73 7065 6369 6573 2e6d   `gene_species.m
+000187f0: 6170 6020 6861 7320 636f 6e74 656e 7473  ap` has contents
+00018800: 2061 7320 6265 6c6f 7720 696e 2077 6869   as below in whi
+00018810: 6368 2065 6163 6820 6c69 6e65 2069 7320  ch each line is 
+00018820: 7468 6520 6a6f 696e 6564 2073 7472 696e  the joined strin
+00018830: 6720 6f66 2067 656e 6520 6e61 6d65 2061  g of gene name a
+00018840: 6e64 2073 7065 6369 6573 206e 616d 6520  nd species name 
+00018850: 6279 2073 7061 6365 2e20 4166 7465 7220  by space. After 
+00018860: 7632 2e30 2e32 3120 2869 6e63 6c75 6465  v2.0.21 (include
+00018870: 6429 2c20 7468 6520 6765 6e65 2d73 7065  d), the gene-spe
+00018880: 6369 6573 206d 6170 2066 696c 6520 6973  cies map file is
+00018890: 206e 6f74 206e 6563 6365 7373 6172 696c   not neccessaril
+000188a0: 7920 6e65 6564 6564 2061 6e79 6d6f 7265  y needed anymore
+000188b0: 2e0a 0a60 6060 0a41 7163 6f65 3647 3035  ...```.Aqcoe6G05
+000188c0: 3738 3030 2e31 2041 7175 696c 6567 6961  7800.1 Aquilegia
+000188d0: 5f63 6f65 7275 6c65 610a 5676 695f 5649  _coerulea.Vvi_VI
+000188e0: 545f 3230 3173 3030 3131 6730 3135 3330  T_201s0011g01530
+000188f0: 2e31 2056 6974 6973 5f76 696e 6966 6572  .1 Vitis_vinifer
+00018900: 610a 5063 795f 5072 6f63 7930 3167 3038  a.Pcy_Procy01g08
+00018910: 3531 3020 5072 6f74 6561 5f63 796e 6172  510 Protea_cynar
+00018920: 6f69 6465 730a 4161 6d5f 4163 6f72 612e  oides.Aam_Acora.
+00018930: 3034 4731 3432 3930 302e 3120 4163 6f72  04G142900.1 Acor
+00018940: 7573 5f61 6d65 7269 6361 6e75 730a 6060  us_americanus.``
+00018950: 600a 0a41 6e20 616c 7465 726e 6174 6976  `..An alternativ
+00018960: 6520 7761 7920 746f 2063 616c 6375 6c61  e way to calcula
+00018970: 7465 2074 6865 206f 7274 686f 6c6f 676f  te the orthologo
+00018980: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+00018990: 623e 2069 7320 746f 2064 6972 6563 746c  b> is to directl
+000189a0: 7920 7573 6520 7468 6520 6f72 7468 6f67  y use the orthog
+000189b0: 726f 7570 7320 696e 7374 6561 6420 6f66  roups instead of
+000189c0: 2067 6c6f 6261 6c20 4d52 4248 2066 616d   global MRBH fam
+000189d0: 696c 792e 2054 6861 7420 7761 7920 7765  ily. That way we
+000189e0: 2064 6f6e 2774 2075 7365 2074 6865 2070   don't use the p
+000189f0: 7265 2d69 6e66 6572 7265 6420 7061 7261  re-inferred para
+00018a00: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+00018a10: 7375 623e 2062 7574 2074 6865 2070 6172  sub> but the par
+00018a20: 616c 6f67 6f75 7320 6765 6e65 2070 6169  alogous gene pai
+00018a30: 7273 2069 6e73 6964 6520 6561 6368 206f  rs inside each o
+00018a40: 7274 686f 6772 6f75 7020 696e 7374 6561  rthogroup instea
+00018a50: 642e 2054 6f20 6163 6869 6576 6520 7468  d. To achieve th
+00018a60: 6174 2c20 7765 2066 6972 7374 206e 6565  at, we first nee
+00018a70: 6420 746f 2069 6e66 6572 206f 7274 686f  d to infer ortho
+00018a80: 6772 6f75 7073 2075 7369 6e67 2074 6865  groups using the
+00018a90: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
+00018aa0: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
+00018ab0: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+00018ac0: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
+00018ad0: 7320 4163 6f72 7573 5f61 6d65 7269 6361  s Acorus_america
+00018ae0: 6e75 7320 5669 7469 735f 7669 6e69 6665  nus Vitis_vinife
+00018af0: 7261 202d 2d6f 7274 686f 696e 6665 7220  ra --orthoinfer 
+00018b00: 2d6f 2077 6764 5f6f 7274 686f 2028 2d2d  -o wgd_ortho (--
+00018b10: 6f6e 6c79 6f72 7468 6f29 200a 6060 600a  onlyortho) .```.
+00018b20: 0a55 7365 7273 2063 616e 2064 6563 6964  .Users can decid
+00018b30: 6520 746f 206f 6e6c 7920 636f 6e64 7563  e to only conduc
+00018b40: 7420 7468 6520 6f72 7468 6f67 726f 7570  t the orthogroup
+00018b50: 2061 6e61 6c79 7369 7320 7768 696c 6520   analysis while 
+00018b60: 736b 6970 7069 6e67 206f 7468 6572 2061  skipping other a
+00018b70: 6e61 6c79 7369 7320 6279 2061 6464 696e  nalysis by addin
+00018b80: 6720 7468 6520 666c 6167 2060 2d2d 6f6e  g the flag `--on
+00018b90: 6c79 6f72 7468 6f60 2e20 4e65 7874 2073  lyortho`. Next s
+00018ba0: 7465 7020 6973 2074 6865 2073 616d 6520  tep is the same 
+00018bb0: 7769 7468 2067 6c6f 6261 6c20 4d52 4248  with global MRBH
+00018bc0: 2066 616d 696c 792e 0a0a 6060 600a 7767   family...```.wg
+00018bd0: 6420 6b73 6420 7767 645f 6f72 7468 6f2f  d ksd wgd_ortho/
+00018be0: 4f72 7468 6f67 726f 7570 732e 7370 2e74  Orthogroups.sp.t
+00018bf0: 7376 2041 7175 696c 6567 6961 5f63 6f65  sv Aquilegia_coe
+00018c00: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
+00018c10: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
+00018c20: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
+00018c30: 7669 6e69 6665 7261 202d 6f20 7767 645f  vinifera -o wgd_
+00018c40: 6f72 7468 6f5f 6b73 0a77 6764 2076 697a  ortho_ks.wgd viz
+00018c50: 202d 6420 7767 645f 6f72 7468 6f5f 6b73   -d wgd_ortho_ks
+00018c60: 2f4f 7274 686f 6772 6f75 7073 2e73 702e  /Orthogroups.sp.
+00018c70: 7473 762e 6b73 2e74 7376 202d 6570 6b20  tsv.ks.tsv -epk 
+00018c80: 7767 645f 6b73 642f 4171 7569 6c65 6769  wgd_ksd/Aquilegi
+00018c90: 615f 636f 6572 756c 6561 2e6b 732e 7473  a_coerulea.ks.ts
+00018ca0: 7620 2d73 7020 7370 6563 6965 7374 7265  v -sp speciestre
+00018cb0: 652e 6e77 202d 2d72 6577 6569 6768 7420  e.nw --reweight 
+00018cc0: 2d61 7020 7767 645f 7379 6e2f 6961 6468  -ap wgd_syn/iadh
+00018cd0: 6f72 652d 6f75 742f 616e 6368 6f72 706f  ore-out/anchorpo
+00018ce0: 696e 7473 2e74 7874 202d 706c 6f74 656c  ints.txt -plotel
+00018cf0: 6d6d 202d 2d70 6c6f 7461 7067 6d6d 202d  mm --plotapgmm -
+00018d00: 6f20 7767 645f 6f72 7468 6f5f 6b73 5f72  o wgd_ortho_ks_r
+00018d10: 6174 655f 636f 7272 6563 7469 6f6e 0a60  ate_correction.`
+00018d20: 6060 0a0a 215b 5d28 6461 7461 2f4d 6978  ``..![](data/Mix
+00018d30: 6564 2e6b 732e 4171 7569 6c65 6769 615f  ed.ks.Aquilegia_
+00018d40: 636f 6572 756c 6561 2e6e 6f64 652e 7765  coerulea.node.we
+00018d50: 6967 6874 6564 5f4f 7274 686f 5f4b 732e  ighted_Ortho_Ks.
+00018d60: 706e 6729 0a21 5b5d 2864 6174 612f 466f  png).![](data/Fo
+00018d70: 6375 735f 7369 7374 6572 5f70 6169 7273  cus_sister_pairs
+00018d80: 2e6b 732e 6e6f 6465 2e77 6569 6768 7465  .ks.node.weighte
+00018d90: 645f 4f72 7468 6f5f 4b73 2e70 6e67 290a  d_Ortho_Ks.png).
+00018da0: 215b 5d28 6461 7461 2f41 6c6c 5f70 6169  ![](data/All_pai
+00018db0: 7273 2e6b 732e 6e6f 6465 2e77 6569 6768  rs.ks.node.weigh
+00018dc0: 7465 645f 4f72 7468 6f5f 4b73 2e70 6e67  ted_Ortho_Ks.png
+00018dd0: 290a 0a41 7320 7368 6f77 6e20 6162 6f76  )..As shown abov
+00018de0: 652c 2074 6865 206e 756d 6265 7220 6f66  e, the number of
+00018df0: 206f 7274 686f 6c6f 676f 7573 2067 656e   orthologous gen
+00018e00: 6520 7061 6972 7320 6973 2064 6966 6665  e pairs is diffe
+00018e10: 7265 6e74 2074 6861 6e20 7468 6520 6f6e  rent than the on
+00018e20: 6520 6672 6f6d 2067 6c6f 6261 6c20 4d52  e from global MR
+00018e30: 4248 2066 616d 696c 6965 7320 696e 2074  BH families in t
+00018e40: 6861 7420 6865 7265 2077 6520 706c 6f74  hat here we plot
+00018e50: 7465 6420 616c 6c20 6f72 7468 6f6c 6f67  ted all ortholog
+00018e60: 6f75 7320 6765 6e65 2070 6169 7273 2069  ous gene pairs i
+00018e70: 6e73 7465 6164 206f 6620 6f6e 6c79 2067  nstead of only g
+00018e80: 6c6f 6261 6c20 4d52 4248 2066 616d 696c  lobal MRBH famil
+00018e90: 6965 732c 2074 6f67 6574 6865 7220 7769  ies, together wi
+00018ea0: 7468 2064 6966 6665 7265 6e74 2072 6563  th different rec
+00018eb0: 616c 6375 6c61 7465 6420 7765 6967 6874  alculated weight
+00018ec0: 732e 0a0a 4166 7465 7220 7468 6520 7068  s...After the ph
+00018ed0: 796c 6f67 656e 6574 6963 2074 696d 696e  ylogenetic timin
+00018ee0: 6720 6f66 2074 6865 2052 616e 756e 6375  g of the Ranuncu
+00018ef0: 6c61 6c65 7320 5747 442c 2077 6520 6361  lales WGD, we ca
+00018f00: 6e20 6675 7274 6865 7220 696e 6665 7220  n further infer 
+00018f10: 6974 7320 6162 736f 6c75 7465 2061 6765  its absolute age
+00018f20: 2e20 4669 7273 7420 7765 2069 6e66 6572  . First we infer
+00018f30: 2074 6865 2063 7265 6469 626c 6520 7261   the credible ra
+00018f40: 6e67 6520 6f66 2061 6e63 686f 7220 7061  nge of anchor pa
+00018f50: 6972 7320 6279 202a 4b2a 3c73 7562 3e53  irs by *K*<sub>S
+00018f60: 3c2f 7375 623e 2068 6575 7269 7374 6963  </sub> heuristic
+00018f70: 616c 6c79 2075 7369 6e67 2074 6865 2070  ally using the p
+00018f80: 726f 6772 616d 2060 7767 6420 7065 616b  rogram `wgd peak
+00018f90: 602e 0a0a 6060 600a 7767 6420 7065 616b  `...```.wgd peak
+00018fa0: 202d 2d68 6575 7269 7374 6963 2077 6764   --heuristic wgd
+00018fb0: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
+00018fc0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00018fd0: 7376 202d 6170 2077 6764 5f73 796e 2f69  sv -ap wgd_syn/i
+00018fe0: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+00018ff0: 7270 6f69 6e74 732e 7478 7420 2d73 6d20  rpoints.txt -sm 
+00019000: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
+00019010: 6f75 742f 7365 676d 656e 7473 2e74 7874  out/segments.txt
+00019020: 202d 6c65 2077 6764 5f73 796e 2f69 6164   -le wgd_syn/iad
+00019030: 686f 7265 2d6f 7574 2f6c 6973 745f 656c  hore-out/list_el
+00019040: 656d 656e 7473 2e74 7874 202d 6d70 2077  ements.txt -mp w
+00019050: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+00019060: 7574 2f6d 756c 7469 706c 6963 6f6e 5f70  ut/multiplicon_p
+00019070: 6169 7273 2e74 7874 202d 6f20 7767 645f  airs.txt -o wgd_
+00019080: 7065 616b 0a60 6060 0a0a 215b 5d28 6461  peak.```..![](da
+00019090: 7461 2f41 6e63 686f 724b 735f 5065 616b  ta/AnchorKs_Peak
+000190a0: 4349 5f41 7175 696c 6567 6961 5f63 6f65  CI_Aquilegia_coe
+000190b0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+000190c0: 5f6e 6f64 655f 7765 6967 6874 6564 2e73  _node_weighted.s
+000190d0: 7667 290a 0a41 7320 7368 6f77 6e20 6162  vg)..As shown ab
+000190e0: 6f76 652c 2077 6520 6173 7375 6d65 6420  ove, we assumed 
+000190f0: 6120 6c6f 676e 6f72 6d61 6c20 6469 7374  a lognormal dist
+00019100: 7269 6275 7469 6f6e 2061 7420 7468 6520  ribution at the 
+00019110: 7065 616b 206c 6f63 6174 696f 6e20 6465  peak location de
+00019120: 7465 6374 6564 2062 7920 7468 6520 6073  tected by the `s
+00019130: 6967 6e61 6c60 206d 6f64 756c 6520 6f66  ignal` module of
+00019140: 2060 7363 6970 7960 206c 6962 7261 7279   `scipy` library
+00019150: 2e20 5468 6520 3935 2520 636f 6e66 6964  . The 95% confid
+00019160: 656e 6365 206c 6576 656c 206f 6620 7468  ence level of th
+00019170: 6520 6c6f 676e 6f72 6d61 6c20 6469 7374  e lognormal dist
+00019180: 7269 6275 7469 6f6e 2077 6173 2061 7070  ribution was app
+00019190: 6c69 6564 2c20 692e 652e 2c20 302e 3638  lied, i.e., 0.68
+000191a0: 2d32 2e37 342c 2069 6e20 6675 7274 6865  -2.74, in furthe
+000191b0: 7220 6d6f 6c65 6375 6c61 7220 6461 7469  r molecular dati
+000191c0: 6e67 2e20 5468 6520 6669 6c65 2060 4171  ng. The file `Aq
+000191d0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000191e0: 2e74 7376 2e6b 732e 7473 765f 3935 2543  .tsv.ks.tsv_95%C
+000191f0: 495f 4150 5f66 6f72 5f64 6174 696e 675f  I_AP_for_dating_
+00019200: 7765 6967 6874 6564 5f66 6f72 6d61 742e  weighted_format.
+00019210: 7473 7660 2069 7320 7768 6174 2077 6520  tsv` is what we 
+00019220: 6e65 6564 2066 6f72 206e 6578 7420 7374  need for next st
+00019230: 6570 2e20 546f 2062 7569 6c64 2074 6865  ep. To build the
+00019240: 206f 7274 686f 6772 6f75 7073 2075 7365   orthogroups use
+00019250: 6420 696e 2070 6879 6c6f 6765 6e65 7469  d in phylogeneti
+00019260: 6320 6461 7469 6e67 2c20 7765 206e 6565  c dating, we nee
+00019270: 6420 746f 2073 656c 6563 7420 736f 6d65  d to select some
+00019280: 2073 7065 6369 6573 2061 6e64 2066 6f72   species and for
+00019290: 6d20 6120 7374 6172 7469 6e67 2074 7265  m a starting tre
+000192a0: 6520 7769 7468 2070 726f 7065 7220 666f  e with proper fo
+000192b0: 7373 696c 2063 616c 6962 7261 7469 6f6e  ssil calibration
+000192c0: 732e 2057 6520 7072 6f76 6964 6520 6f6e  s. We provide on
+000192d0: 6520 696e 206d 636d 6374 7265 6520 666f  e in mcmctree fo
+000192e0: 726d 6174 2061 7320 6265 6c6f 772e 0a0a  rmat as below...
+000192f0: 6060 600a 3137 2031 0a28 2828 2850 6f74  ```.17 1.((((Pot
+00019300: 616d 6f67 6574 6f6e 5f61 6375 7469 666f  amogeton_acutifo
+00019310: 6c69 7573 2c28 5370 6972 6f64 656c 615f  lius,(Spirodela_
+00019320: 696e 7465 726d 6564 6961 2c41 6d6f 7270  intermedia,Amorp
+00019330: 686f 7068 616c 6c75 735f 6b6f 6e6a 6163  hophallus_konjac
+00019340: 2929 2c28 4163 616e 7468 6f63 686c 616d  )),(Acanthochlam
+00019350: 7973 5f62 7261 6374 6561 7461 2c28 4469  ys_bracteata,(Di
+00019360: 6f73 636f 7265 615f 616c 6174 612c 4469  oscorea_alata,Di
+00019370: 6f73 636f 7265 615f 726f 7475 6e64 6174  oscorea_rotundat
+00019380: 6129 2927 3e30 2e35 3630 303c 312e 3238  a))'>0.5600<1.28
+00019390: 3633 2729 273e 302e 3833 3630 3c31 2e32  63')'>0.8360<1.2
+000193a0: 3836 3327 2c28 4163 6f72 7573 5f61 6d65  863',(Acorus_ame
+000193b0: 7269 6361 6e75 732c 4163 6f72 7573 5f74  ricanus,Acorus_t
+000193c0: 6174 6172 696e 6f77 6969 2929 273e 302e  atarinowii))'>0.
+000193d0: 3833 3630 3c31 2e32 3836 3327 2c28 2828  8360<1.2863',(((
+000193e0: 2854 6574 7261 6365 6e74 726f 6e5f 7369  (Tetracentron_si
+000193f0: 6e65 6e73 652c 5472 6f63 686f 6465 6e64  nense,Trochodend
+00019400: 726f 6e5f 6172 616c 696f 6964 6573 292c  ron_aralioides),
+00019410: 2842 7578 7573 5f61 7573 7472 6f79 756e  (Buxus_austroyun
+00019420: 6e61 6e65 6e73 6973 2c42 7578 7573 5f73  nanensis,Buxus_s
+00019430: 696e 6963 6129 2927 3e31 2e31 3038 303c  inica))'>1.1080<
+00019440: 312e 3238 3633 272c 284e 656c 756d 626f  1.2863',(Nelumbo
+00019450: 5f6e 7563 6966 6572 612c 2854 656c 6f70  _nucifera,(Telop
+00019460: 6561 5f73 7065 6369 6f73 6973 7369 6d61  ea_speciosissima
+00019470: 2c50 726f 7465 615f 6379 6e61 726f 6964  ,Protea_cynaroid
+00019480: 6573 2929 2927 3e31 2e31 3038 303c 312e  es)))'>1.1080<1.
+00019490: 3238 3633 272c 2841 7175 696c 6567 6961  2863',(Aquilegia
+000194a0: 5f63 6f65 7275 6c65 615f 6170 312c 4171  _coerulea_ap1,Aq
+000194b0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000194c0: 5f61 7032 2929 273e 312e 3130 3830 3c31  _ap2))'>1.1080<1
+000194d0: 2e32 3836 3327 2927 3e31 2e32 3732 303c  .2863')'>1.2720<
+000194e0: 322e 3437 3230 273b 0a60 6060 0a0a 4173  2.4720';.```..As
+000194f0: 2070 7265 7365 6e74 6564 2061 626f 7665   presented above
+00019500: 2c20 7468 6520 666f 6375 7320 7370 6563  , the focus spec
+00019510: 6965 7320 7468 6174 2069 7320 6162 6f75  ies that is abou
+00019520: 7420 746f 2062 6520 6461 7465 6420 6e65  t to be dated ne
+00019530: 6564 7320 746f 2062 6520 7265 706c 6163  eds to be replac
+00019540: 6564 2077 6974 6820 6028 4171 7569 6c65  ed with `(Aquile
+00019550: 6769 615f 636f 6572 756c 6561 5f61 7031  gia_coerulea_ap1
+00019560: 2c41 7175 696c 6567 6961 5f63 6f65 7275  ,Aquilegia_coeru
+00019570: 6c65 615f 6170 3229 602e 2057 6974 6820  lea_ap2)`. With 
+00019580: 7468 6973 2073 7461 7274 696e 6720 7472  this starting tr
+00019590: 6565 2061 6e64 2070 7265 646f 776e 6c6f  ee and predownlo
+000195a0: 6164 6564 2063 6473 2066 696c 6573 206f  aded cds files o
+000195b0: 6620 616c 6c20 7468 6520 7370 6563 6965  f all the specie
+000195c0: 732c 2077 6520 6361 6e20 6275 696c 6420  s, we can build 
+000195d0: 7468 6520 6f72 7468 6f67 726f 7570 2075  the orthogroup u
+000195e0: 7365 6420 696e 2074 6865 2066 696e 616c  sed in the final
+000195f0: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
+00019600: 6720 7573 696e 6720 7468 6520 636f 6d6d  g using the comm
+00019610: 616e 6420 6173 2062 656c 6f77 2e0a 0a60  and as below...`
+00019620: 6060 0a77 6764 2064 6d64 202d 6620 4171  ``.wgd dmd -f Aq
+00019630: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00019640: 202d 6170 2077 6764 5f70 6561 6b2f 416e   -ap wgd_peak/An
+00019650: 6368 6f72 4b73 5f46 696e 6450 6561 6b2f  chorKs_FindPeak/
+00019660: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00019670: 6561 2e74 7376 2e6b 732e 7473 765f 3935  ea.tsv.ks.tsv_95
+00019680: 2543 495f 4150 5f66 6f72 5f64 6174 696e  %CI_AP_for_datin
+00019690: 675f 7765 6967 6874 6564 5f66 6f72 6d61  g_weighted_forma
+000196a0: 742e 7473 7620 2d6f 2077 6764 5f64 6d64  t.tsv -o wgd_dmd
+000196b0: 5f6f 7274 686f 2050 6f74 616d 6f67 6574  _ortho Potamoget
+000196c0: 6f6e 5f61 6375 7469 666f 6c69 7573 2053  on_acutifolius S
+000196d0: 7069 726f 6465 6c61 5f69 6e74 6572 6d65  pirodela_interme
+000196e0: 6469 6120 416d 6f72 7068 6f70 6861 6c6c  dia Amorphophall
+000196f0: 7573 5f6b 6f6e 6a61 6320 4163 616e 7468  us_konjac Acanth
+00019700: 6f63 686c 616d 7973 5f62 7261 6374 6561  ochlamys_bractea
+00019710: 7461 2044 696f 7363 6f72 6561 5f61 6c61  ta Dioscorea_ala
+00019720: 7461 2044 696f 7363 6f72 6561 5f72 6f74  ta Dioscorea_rot
+00019730: 756e 6461 7461 2041 636f 7275 735f 616d  undata Acorus_am
+00019740: 6572 6963 616e 7573 2041 636f 7275 735f  ericanus Acorus_
+00019750: 7461 7461 7269 6e6f 7769 6920 5465 7472  tatarinowii Tetr
+00019760: 6163 656e 7472 6f6e 5f73 696e 656e 7365  acentron_sinense
+00019770: 2054 726f 6368 6f64 656e 6472 6f6e 5f61   Trochodendron_a
+00019780: 7261 6c69 6f69 6465 7320 4275 7875 735f  ralioides Buxus_
+00019790: 6175 7374 726f 7975 6e6e 616e 656e 7369  austroyunnanensi
+000197a0: 7320 4275 7875 735f 7369 6e69 6361 204e  s Buxus_sinica N
+000197b0: 656c 756d 626f 5f6e 7563 6966 6572 6120  elumbo_nucifera 
+000197c0: 5465 6c6f 7065 615f 7370 6563 696f 7369  Telopea_speciosi
+000197d0: 7373 696d 6120 5072 6f74 6561 5f63 796e  ssima Protea_cyn
+000197e0: 6172 6f69 6465 7320 4171 7569 6c65 6769  aroides Aquilegi
+000197f0: 615f 636f 6572 756c 6561 0a60 6060 0a0a  a_coerulea.```..
+00019800: 5468 6520 7265 7375 6c74 2066 696c 6520  The result file 
+00019810: 606d 6572 6765 5f66 6f63 7573 5f61 702e  `merge_focus_ap.
+00019820: 7473 7660 2069 7320 7768 6174 2077 6520  tsv` is what we 
+00019830: 6e65 6564 2066 6f72 2074 6865 2066 696e  need for the fin
+00019840: 616c 2073 7465 7020 6f66 206d 6f6c 6563  al step of molec
+00019850: 756c 6172 2064 6174 696e 6720 696e 2070  ular dating in p
+00019860: 726f 6772 616d 2060 7767 6420 666f 6375  rogram `wgd focu
+00019870: 7360 2e0a 0a60 6060 0a77 6764 2066 6f63  s`...```.wgd foc
+00019880: 7573 202d 2d70 726f 7463 6f63 6461 7469  us --protcocdati
+00019890: 6e67 202d 2d61 616d 6f64 656c 206c 6720  ng --aamodel lg 
+000198a0: 7767 645f 646d 645f 6f72 7468 6f2f 6d65  wgd_dmd_ortho/me
+000198b0: 7267 655f 666f 6375 735f 6170 2e74 7376  rge_focus_ap.tsv
+000198c0: 202d 7370 2064 6174 696e 675f 7472 6565   -sp dating_tree
+000198d0: 2e6e 7720 2d6f 2077 6764 5f64 6174 696e  .nw -o wgd_datin
+000198e0: 6720 2d64 206d 636d 6374 7265 6520 2d64  g -d mcmctree -d
+000198f0: 7320 2762 7572 6e69 6e20 3d20 3230 3030  s 'burnin = 2000
+00019900: 2720 2d64 7320 2773 616d 7066 7265 7120  ' -ds 'sampfreq 
+00019910: 3d20 3130 3030 2720 2d64 7320 276e 7361  = 1000' -ds 'nsa
+00019920: 6d70 6c65 203d 2032 3030 3030 2720 506f  mple = 20000' Po
+00019930: 7461 6d6f 6765 746f 6e5f 6163 7574 6966  tamogeton_acutif
+00019940: 6f6c 6975 7320 5370 6972 6f64 656c 615f  olius Spirodela_
+00019950: 696e 7465 726d 6564 6961 2041 6d6f 7270  intermedia Amorp
+00019960: 686f 7068 616c 6c75 735f 6b6f 6e6a 6163  hophallus_konjac
+00019970: 2041 6361 6e74 686f 6368 6c61 6d79 735f   Acanthochlamys_
+00019980: 6272 6163 7465 6174 6120 4469 6f73 636f  bracteata Diosco
+00019990: 7265 615f 616c 6174 6120 4469 6f73 636f  rea_alata Diosco
+000199a0: 7265 615f 726f 7475 6e64 6174 6120 4163  rea_rotundata Ac
+000199b0: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
+000199c0: 4163 6f72 7573 5f74 6174 6172 696e 6f77  Acorus_tatarinow
+000199d0: 6969 2054 6574 7261 6365 6e74 726f 6e5f  ii Tetracentron_
+000199e0: 7369 6e65 6e73 6520 5472 6f63 686f 6465  sinense Trochode
+000199f0: 6e64 726f 6e5f 6172 616c 696f 6964 6573  ndron_aralioides
+00019a00: 2042 7578 7573 5f61 7573 7472 6f79 756e   Buxus_austroyun
+00019a10: 6e61 6e65 6e73 6973 2042 7578 7573 5f73  nanensis Buxus_s
+00019a20: 696e 6963 6120 4e65 6c75 6d62 6f5f 6e75  inica Nelumbo_nu
+00019a30: 6369 6665 7261 2054 656c 6f70 6561 5f73  cifera Telopea_s
+00019a40: 7065 6369 6f73 6973 7369 6d61 2050 726f  peciosissima Pro
+00019a50: 7465 615f 6379 6e61 726f 6964 6573 2041  tea_cynaroides A
+00019a60: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00019a70: 610a 6060 600a 0a48 6572 6520 7765 206f  a.```..Here we o
+00019a80: 6e6c 7920 696d 706c 656d 656e 7465 6420  nly implemented 
+00019a90: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+00019aa0: 6e20 616e 616c 7973 6973 2075 7369 6e67  n analysis using
+00019ab0: 2070 726f 7465 696e 2073 6571 7565 6e63   protein sequenc
+00019ac0: 6520 6279 2061 6464 696e 6720 7468 6520  e by adding the 
+00019ad0: 666c 6167 2060 2d2d 7072 6f74 6461 7469  flag `--protdati
+00019ae0: 6e67 6020 616e 6420 7765 2073 6574 2074  ng` and we set t
+00019af0: 6865 2070 6172 616d 6574 6572 2066 6f72  he parameter for
+00019b00: 2060 6d63 6d63 7472 6565 6020 7669 6120   `mcmctree` via 
+00019b10: 7468 6520 6f70 7469 6f6e 2060 2d64 7360  the option `-ds`
+00019b20: 2e20 4e6f 7465 2074 6861 7420 6f74 6865  . Note that othe
+00019b30: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
+00019b40: 2073 7563 6820 6173 2060 7238 7360 2061   such as `r8s` a
+00019b50: 6e64 2060 6265 6173 7460 2061 7265 2061  nd `beast` are a
+00019b60: 6c73 6f20 6176 6169 6c61 626c 6520 6769  lso available gi
+00019b70: 7665 6e20 736f 6d65 206d 616e 6461 746f  ven some mandato
+00019b80: 7279 2070 6172 616d 6574 6572 732e 2054  ry parameters. T
+00019b90: 6865 2066 696e 616c 206c 6f67 206f 6620  he final log of 
+00019ba0: 7468 6520 7375 6363 6573 7366 756c 2072  the successful r
+00019bb0: 756e 2069 7320 6173 2062 656c 6f77 2e0a  un is as below..
+00019bc0: 0a60 6060 0a31 363a 3034 3a32 3520 494e  .```.16:04:25 IN
+00019bd0: 464f 2020 2020 2052 756e 6e69 6e67 206d  FO     Running m
+00019be0: 636d 6374 7265 6520 7573 696e 6720 4865  cmctree using He
+00019bf0: 7373 6961 6e20 6d61 7472 6978 206f 6620  ssian matrix of 
+00019c00: 4c47 2b47 616d 6d61 2020 636f 7265 2e70  LG+Gamma  core.p
+00019c10: 793a 3936 370a 2020 2020 2020 2020 2020  y:967.          
+00019c20: 2020 2020 2020 2020 666f 7220 7072 6f74          for prot
+00019c30: 6569 6e20 6d6f 6465 6c0a 3233 3a34 393a  ein model.23:49:
+00019c40: 3337 2049 4e46 4f20 2020 2020 506f 7374  37 INFO     Post
+00019c50: 6572 696f 7220 6d65 616e 2066 6f72 2074  erior mean for t
+00019c60: 6865 2061 6765 7320 6f66 2077 6764 2069  he ages of wgd i
+00019c70: 7320 3131 322e 3839 3435 206d 636d 6374  s 112.8945 mcmct
+00019c80: 7265 652e 7079 3a32 3936 0a20 2020 2020  ree.py:296.     
+00019c90: 2020 2020 2020 2020 2020 2020 206d 696c               mil
+00019ca0: 6c69 6f6e 2079 6561 7273 2066 726f 6d20  lion years from 
+00019cb0: 436f 6e63 6174 656e 6174 6564 2070 6570  Concatenated pep
+00019cc0: 7469 6465 0a20 2020 2020 2020 2020 2020  tide.           
+00019cd0: 2020 2020 2020 2061 6c69 676e 6d65 6e74         alignment
+00019ce0: 2061 6e64 2039 3525 2063 7265 6469 6269   and 95% credibi
+00019cf0: 6c69 7479 2069 6e74 6572 7661 6c73 2028  lity intervals (
+00019d00: 4349 290a 2020 2020 2020 2020 2020 2020  CI).            
+00019d10: 2020 2020 2020 6973 2031 3031 2e32 3234        is 101.224
+00019d20: 2d31 3233 2e31 3231 206d 696c 6c69 6f6e  -123.121 million
+00019d30: 2079 6561 7273 0a20 2020 2020 2020 2020   years.         
+00019d40: 494e 464f 2020 2020 2054 6f74 616c 2072  INFO     Total r
+00019d50: 756e 2074 696d 653a 2032 3931 3735 7320  un time: 29175s 
+00019d60: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019d70: 2020 2020 2020 2020 2020 2020 2063 6c69               cli
+00019d80: 2e70 793a 3234 310a 2020 2020 2020 2020  .py:241.        
+00019d90: 2049 4e46 4f20 2020 2020 446f 6e65 2020   INFO     Done  
+00019da0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019db0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019dc0: 2020 2020 2020 2020 2020 2020 2020 636c                cl
+00019dd0: 692e 7079 3a32 3432 0a60 6060 0a0a 546f  i.py:242.```..To
+00019de0: 2076 6973 7561 6c69 7a65 2074 6865 2064   visualize the d
+00019df0: 6174 652c 2077 6520 616c 736f 2070 726f  ate, we also pro
+00019e00: 7669 6465 6420 6120 7079 7468 6f6e 2073  vided a python s
+00019e10: 6372 6970 7420 746f 2070 6c6f 7420 7468  cript to plot th
+00019e20: 6520 5747 4420 6461 7465 7320 696e 2074  e WGD dates in t
+00019e30: 6865 2060 7767 6460 2066 6f6c 6465 722e  he `wgd` folder.
+00019e40: 2055 7365 7273 206e 6565 6420 746f 2065   Users need to e
+00019e50: 7874 7261 6374 2074 6865 2072 6177 2064  xtract the raw d
+00019e60: 6174 6573 2066 726f 6d20 7468 6520 606d  ates from the `m
+00019e70: 636d 632e 7478 7460 2066 6f72 2074 6865  cmc.txt` for the
+00019e80: 2057 4744 206e 6f64 6520 6669 7273 7420   WGD node first 
+00019e90: 616e 6420 7361 7665 2069 7420 6173 2066  and save it as f
+00019ea0: 696c 6520 6064 6174 6573 2e74 7874 6020  ile `dates.txt` 
+00019eb0: 286f 7220 7768 6174 6576 6572 2070 7265  (or whatever pre
+00019ec0: 6665 7272 6564 206e 616d 6529 2e20 416e  ferred name). An
+00019ed0: 2065 7861 6d70 6c65 2063 6f6d 6d61 6e64   example command
+00019ee0: 2069 7320 6173 2062 656c 6f77 2e0a 0a60   is as below...`
+00019ef0: 6060 0a70 7974 686f 6e20 2450 4154 482f  ``.python $PATH/
+00019f00: 706f 7374 706c 6f74 2e70 7920 706f 7374  postplot.py post
+00019f10: 6469 7320 6461 7465 732e 7478 7420 2d2d  dis dates.txt --
+00019f20: 7065 7263 656e 7469 6c65 2039 3020 2d2d  percentile 90 --
+00019f30: 7469 746c 6520 2257 4744 2064 6174 6522  title "WGD date"
+00019f40: 202d 2d68 7064 202d 6f20 2252 616e 756e   --hpd -o "Ranun
+00019f50: 6375 6c61 6c65 735f 5747 445f 6461 7465  culales_WGD_date
+00019f60: 2e73 7667 220a 6060 600a 0a21 5b5d 2864  .svg".```..![](d
+00019f70: 6174 612f 5261 6e75 6e63 756c 616c 6573  ata/Ranunculales
+00019f80: 5f57 4744 5f64 6174 652e 7376 6729 0a0a  _WGD_date.svg)..
+00019f90: 5468 6520 706f 7374 6572 696f 7220 6d65  The posterior me
+00019fa0: 616e 2c20 6d65 6469 616e 2061 6e64 206d  an, median and m
+00019fb0: 6f64 6520 6f66 2074 6865 2052 616e 756e  ode of the Ranun
+00019fc0: 6375 6c61 6c65 7320 5747 4420 6167 6520  culales WGD age 
+00019fd0: 6973 2031 3132 2e39 322c 2031 3133 2e34  is 112.92, 113.4
+00019fe0: 3420 616e 6420 3131 322e 3534 206d 7961  4 and 112.54 mya
+00019ff0: 2c20 7769 7468 2039 3025 2048 5044 2031  , with 90% HPD 1
+0001a000: 3035 2e30 3720 2d20 3132 322e 3332 206d  05.07 - 122.32 m
+0001a010: 7961 2061 7320 6d61 6e69 6665 7374 6564  ya as manifested
+0001a020: 2061 626f 7665 2e0a 0a23 2323 204b 7374   above...### Kst
+0001a030: 7265 650a 0a49 6e20 6164 6469 7469 6f6e  ree..In addition
+0001a040: 2074 6f20 7061 6972 7769 7365 202a 4b2a   to pairwise *K*
+0001a050: 3c73 7562 3e53 3c2f 7375 623e 2065 7374  <sub>S</sub> est
+0001a060: 696d 6174 696f 6e2c 2061 202a 4b2a 3c73  imation, a *K*<s
+0001a070: 7562 3e53 3c2f 7375 623e 2074 7265 6520  ub>S</sub> tree 
+0001a080: 7769 7468 2062 7261 6e63 6820 6c65 6e67  with branch leng
+0001a090: 7468 2069 6e20 2a4b 2a3c 7375 623e 533c  th in *K*<sub>S<
+0001a0a0: 2f73 7562 3e20 756e 6974 2063 616e 2061  /sub> unit can a
+0001a0b0: 6c73 6f20 6265 2064 6572 6976 6564 2066  lso be derived f
+0001a0c0: 726f 6d20 7468 6520 7072 6f67 7261 6d20  rom the program 
+0001a0d0: 6077 6764 206b 7364 6020 6769 7665 6e20  `wgd ksd` given 
+0001a0e0: 7468 6520 6f70 7469 6f6e 2060 2d2d 6b73  the option `--ks
+0001a0f0: 7472 6565 6020 616e 6420 602d 2d73 7065  tree` and `--spe
+0001a100: 6369 6573 7472 6565 602e 204e 6f74 6520  ciestree`. Note 
+0001a110: 7468 6174 2074 6865 2061 6464 6974 696f  that the additio
+0001a120: 6e61 6c20 6f70 7469 6f6e 2060 2d2d 6f6e  nal option `--on
+0001a130: 6c79 636f 6e63 6174 6b73 7472 6565 6020  lyconcatkstree` 
+0001a140: 7769 6c6c 206f 6e6c 7920 6361 6c6c 2074  will only call t
+0001a150: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0001a160: 623e 2065 7374 696d 6174 696f 6e20 666f  b> estimation fo
+0001a170: 7220 7468 6520 636f 6e63 6174 656e 6174  r the concatenat
+0001a180: 6564 2061 6c69 676e 6d65 6e74 2072 6174  ed alignment rat
+0001a190: 6865 7220 7468 616e 2061 6c6c 2074 6865  her than all the
+0001a1a0: 2061 6c69 676e 6d65 6e74 732e 2055 7365   alignments. Use
+0001a1b0: 7273 206e 6565 6420 746f 2070 726f 7669  rs need to provi
+0001a1c0: 6465 2061 2070 7265 7365 7420 7370 6563  de a preset spec
+0001a1d0: 6965 7320 7472 6565 2066 6f72 2074 6865  ies tree for the
+0001a1e0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0001a1f0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0001a200: 6f66 2074 6865 2063 6f6e 6361 7465 6e61  of the concatena
+0001a210: 7465 6420 616c 6967 6e6d 656e 7420 7768  ted alignment wh
+0001a220: 696c 6520 7468 6520 7265 6d61 696e 696e  ile the remainin
+0001a230: 6720 616c 6967 6e6d 656e 7473 2077 696c  g alignments wil
+0001a240: 6c20 6265 2061 6761 696e 7374 2061 6e20  l be against an 
+0001a250: 6175 746f 6d61 7465 6c79 2069 6e66 6572  automately infer
+0001a260: 7265 6420 7472 6565 2066 726f 6d20 6066  red tree from `f
+0001a270: 6173 7474 7265 6560 206f 7220 6069 7174  asttree` or `iqt
+0001a280: 7265 6560 2e20 496e 2074 6865 2065 6e64  ree`. In the end
+0001a290: 2c20 7573 6572 7320 7769 6c6c 2067 6574  , users will get
+0001a2a0: 2061 202a 4b2a 3c73 7562 3e53 3c2f 7375   a *K*<sub>S</su
+0001a2b0: 623e 2074 7265 652c 2061 202a 4b2a 3c73  b> tree, a *K*<s
+0001a2c0: 7562 3e41 3c2f 7375 623e 2074 7265 6520  ub>A</sub> tree 
+0001a2d0: 616e 6420 6120 cf89 2074 7265 6520 7065  and a .. tree pe
+0001a2e0: 7220 6661 6d20 616e 6420 666f 7220 7468  r fam and for th
+0001a2f0: 6520 636f 6e63 6174 656e 6174 6564 2061  e concatenated a
+0001a300: 6c69 676e 6d65 6e74 2e0a 0a60 6060 0a77  lignment...```.w
+0001a310: 6764 206b 7364 2064 6174 612f 6b73 7472  gd ksd data/kstr
+0001a320: 6565 5f64 6174 612f 6661 6d2e 7473 7620  ee_data/fam.tsv 
+0001a330: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
+0001a340: 2f41 636f 7275 735f 7461 7461 7269 6e6f  /Acorus_tatarino
+0001a350: 7769 6920 6461 7461 2f6b 7374 7265 655f  wii data/kstree_
+0001a360: 6461 7461 2f41 6d62 6f72 656c 6c61 5f74  data/Amborella_t
+0001a370: 7269 6368 6f70 6f64 6120 6461 7461 2f6b  richopoda data/k
+0001a380: 7374 7265 655f 6461 7461 2f41 7175 696c  stree_data/Aquil
+0001a390: 6567 6961 5f63 6f65 7275 6c65 6120 6461  egia_coerulea da
+0001a3a0: 7461 2f6b 7374 7265 655f 6461 7461 2f41  ta/kstree_data/A
+0001a3b0: 7269 7374 6f6c 6f63 6869 615f 6669 6d62  ristolochia_fimb
+0001a3c0: 7269 6174 6120 6461 7461 2f6b 7374 7265  riata data/kstre
+0001a3d0: 655f 6461 7461 2f43 7963 6173 5f70 616e  e_data/Cycas_pan
+0001a3e0: 7a68 6968 7561 656e 7369 202d 2d6b 7374  zhihuaensi --kst
+0001a3f0: 7265 6520 2d2d 7370 6563 6965 7374 7265  ree --speciestre
+0001a400: 6520 6461 7461 2f6b 7374 7265 655f 6461  e data/kstree_da
+0001a410: 7461 2f73 7065 6369 6573 5f74 7265 6531  ta/species_tree1
+0001a420: 2e6e 7720 2d2d 6f6e 6c79 636f 6e63 6174  .nw --onlyconcat
+0001a430: 6b73 7472 6565 202d 6f20 7767 645f 6b73  kstree -o wgd_ks
+0001a440: 7472 6565 5f74 6f70 6f6c 6f67 7931 0a60  tree_topology1.`
+0001a450: 6060 0a0a 215b 5d28 6461 7461 2f6b 7374  ``..![](data/kst
+0001a460: 7265 655f 7265 7375 6c74 732f 6b73 7472  ree_results/kstr
+0001a470: 6565 2e73 7667 290a 0a41 626f 7665 2077  ee.svg)..Above w
+0001a480: 6520 7573 6564 2074 6872 6565 2061 6c74  e used three alt
+0001a490: 6572 6e61 7469 7665 2074 6f70 6f6c 6f67  ernative topolog
+0001a4a0: 6965 7320 746f 2069 6e66 6572 2074 6865  ies to infer the
+0001a4b0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0001a4c0: 2074 7265 6520 7768 6963 6820 6c65 6420   tree which led 
+0001a4d0: 746f 2064 6966 6665 7265 6e74 2062 7261  to different bra
+0001a4e0: 6e63 6820 6c65 6e67 7468 2065 7374 696d  nch length estim
+0001a4f0: 6174 696f 6e2e 204e 6f74 6520 7468 6174  ation. Note that
+0001a500: 2074 6865 2066 616d 696c 6965 7320 7765   the families we
+0001a510: 2075 7365 6420 7765 7265 206f 6e6c 7920   used were only 
+0001a520: 7477 6f20 676c 6f62 616c 204d 5242 4820  two global MRBH 
+0001a530: 6661 6d69 6c69 6573 2066 6f72 2074 6865  families for the
+0001a540: 2070 7572 706f 7365 206f 6620 696c 6c75   purpose of illu
+0001a550: 7374 7261 7469 6f6e 2e20 546f 2061 6371  stration. To acq
+0001a560: 7569 7265 2061 6e20 6163 6375 7261 7465  uire an accurate
+0001a570: 2070 726f 6669 6c65 206f 6620 7468 6520   profile of the 
+0001a580: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
+0001a590: 6520 7661 7269 6174 696f 6e2c 206f 7274  e variation, ort
+0001a5a0: 686f 6c6f 6775 6573 2061 7420 7468 6520  hologues at the 
+0001a5b0: 7768 6f6c 6520 6765 6e6f 6d65 2073 6361  whole genome sca
+0001a5c0: 6c65 2073 686f 756c 6420 6265 2075 7365  le should be use
+0001a5d0: 642e 0a0a 496e 2061 6464 6974 696f 6e2c  d...In addition,
+0001a5e0: 206d 6f72 6520 6578 7175 6973 6974 6520   more exquisite 
+0001a5f0: 636f 6c6c 696e 6561 7220 706c 6f74 7320  collinear plots 
+0001a600: 696e 636c 7564 696e 6720 626f 7468 2069  including both i
+0001a610: 6e74 7261 2d73 7065 6369 6669 6320 616e  ntra-specific an
+0001a620: 6420 696e 7465 722d 7370 6563 6966 6963  d inter-specific
+0001a630: 2063 6f6d 7061 7269 736f 6e73 2075 7369   comparisons usi
+0001a640: 6e67 2074 6865 206f 7274 686f 6772 6f75  ng the orthogrou
+0001a650: 7073 2028 636f 6d70 6f73 6564 206f 6620  ps (composed of 
+0001a660: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
+0001a670: 6c65 612a 2c20 2a50 726f 7465 6120 6379  lea*, *Protea cy
+0001a680: 6e61 726f 6964 6573 2a2c 202a 4163 6f72  naroides*, *Acor
+0001a690: 7573 2061 6d65 7269 6361 6e75 732a 2061  us americanus* a
+0001a6a0: 6e64 202a 5669 7469 7320 7669 6e69 6665  nd *Vitis vinife
+0001a6b0: 7261 2a29 2069 6e66 6572 7265 6420 6361  ra*) inferred ca
+0001a6c0: 6e20 6265 2061 6c73 6f20 7072 6f64 7563  n be also produc
+0001a6d0: 6564 2075 7369 6e67 2060 7767 6420 7379  ed using `wgd sy
+0001a6e0: 6e60 2e20 4e6f 7465 2074 6861 7420 6469  n`. Note that di
+0001a6f0: 6666 6572 656e 7420 6765 6e6f 6d65 2061  fferent genome a
+0001a700: 7373 656d 626c 6965 7320 6d69 6768 7420  ssemblies might 
+0001a710: 6861 7665 2064 6966 6665 7265 6e74 2066  have different f
+0001a720: 6561 7475 7265 7320 616e 6420 6174 7472  eatures and attr
+0001a730: 6962 7574 6573 2077 6869 6368 2063 616e  ibutes which can
+0001a740: 2062 6520 6163 636f 6d6d 6f64 6174 6564   be accommodated
+0001a750: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
+0001a760: 602d 2d61 6464 6974 696f 6e61 6c67 6666  `--additionalgff
+0001a770: 696e 666f 6020 666f 7220 6561 6368 2067  info` for each g
+0001a780: 656e 6f6d 6520 6173 7365 6d62 6c79 2077  enome assembly w
+0001a790: 686f 7365 206f 7264 6572 206e 6565 6473  hose order needs
+0001a7a0: 2074 6f20 666f 6c6c 6f77 2074 6865 206f   to follow the o
+0001a7b0: 7264 6572 206f 6620 6766 6633 2066 696c  rder of gff3 fil
+0001a7c0: 6573 2c20 666f 7220 696e 7374 616e 6365  es, for instance
+0001a7d0: 2027 6d4e 5241 3b4e 616d 6527 2066 6f72   'mNRA;Name' for
+0001a7e0: 2060 4171 7569 6c65 6769 615f 636f 6572   `Aquilegia_coer
+0001a7f0: 756c 6561 2e67 6666 3360 2c20 276d 4e52  ulea.gff3`, 'mNR
+0001a800: 413b 4944 2720 666f 7220 6050 726f 7465  A;ID' for `Prote
+0001a810: 615f 6379 6e61 726f 6964 6573 2e67 6666  a_cynaroides.gff
+0001a820: 3360 2c20 276d 4e52 413b 4e61 6d65 2720  3`, 'mNRA;Name' 
+0001a830: 666f 7220 6041 636f 7275 735f 616d 6572  for `Acorus_amer
+0001a840: 6963 616e 7573 2e67 6666 3360 2061 6e64  icanus.gff3` and
+0001a850: 2027 6d4e 5241 3b4e 616d 6527 2066 6f72   'mNRA;Name' for
+0001a860: 2060 5669 7469 735f 7669 6e69 6665 7261   `Vitis_vinifera
+0001a870: 2e67 6666 3360 2e0a 0a60 6060 0a77 6764  .gff3`...```.wgd
+0001a880: 2064 6d64 202d 6f6f 202d 6f69 2041 7175   dmd -oo -oi Aqu
+0001a890: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+0001a8a0: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
+0001a8b0: 7320 4163 6f72 7573 5f61 6d65 7269 6361  s Acorus_america
+0001a8c0: 6e75 7320 5669 7469 735f 7669 6e69 6665  nus Vitis_vinife
+0001a8d0: 7261 202d 6f20 7767 645f 6f72 7468 6f0a  ra -o wgd_ortho.
+0001a8e0: 7767 6420 6b73 6420 7767 645f 6f72 7468  wgd ksd wgd_orth
+0001a8f0: 6f2f 4f72 7468 6f67 726f 7570 732e 7370  o/Orthogroups.sp
+0001a900: 2e74 7376 2041 7175 696c 6567 6961 5f63  .tsv Aquilegia_c
+0001a910: 6f65 7275 6c65 6120 5072 6f74 6561 5f63  oerulea Protea_c
+0001a920: 796e 6172 6f69 6465 7320 4163 6f72 7573  ynaroides Acorus
+0001a930: 5f61 6d65 7269 6361 6e75 7320 5669 7469  _americanus Viti
+0001a940: 735f 7669 6e69 6665 7261 202d 6f20 7767  s_vinifera -o wg
+0001a950: 645f 6f72 7468 6f5f 6b73 0a77 6764 2073  d_ortho_ks.wgd s
+0001a960: 796e 2077 6764 5f6f 7274 686f 2f4f 7274  yn wgd_ortho/Ort
+0001a970: 686f 6772 6f75 7073 2e73 702e 7473 7620  hogroups.sp.tsv 
+0001a980: 2d6b 7320 7767 645f 6f72 7468 6f5f 6b73  -ks wgd_ortho_ks
+0001a990: 2f4f 7274 686f 6772 6f75 7073 2e73 702e  /Orthogroups.sp.
+0001a9a0: 7473 762e 6b73 2e74 7376 2041 7175 696c  tsv.ks.tsv Aquil
+0001a9b0: 6567 6961 5f63 6f65 7275 6c65 612e 6766  egia_coerulea.gf
+0001a9c0: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
+0001a9d0: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
+0001a9e0: 6527 2050 726f 7465 615f 6379 6e61 726f  e' Protea_cynaro
+0001a9f0: 6964 6573 2e67 6666 3320 2d2d 6164 6469  ides.gff3 --addi
+0001aa00: 7469 6f6e 616c 6766 6669 6e66 6f20 276d  tionalgffinfo 'm
+0001aa10: 4e52 413b 4944 2720 4163 6f72 7573 5f61  NRA;ID' Acorus_a
+0001aa20: 6d65 7269 6361 6e75 732e 6766 6633 202d  mericanus.gff3 -
+0001aa30: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+0001aa40: 666f 2027 6d4e 5241 3b4e 616d 6527 2056  fo 'mNRA;Name' V
+0001aa50: 6974 6973 5f76 696e 6966 6572 612e 6766  itis_vinifera.gf
+0001aa60: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
+0001aa70: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
+0001aa80: 6527 202d 6f20 7767 645f 6f72 7468 6f5f  e' -o wgd_ortho_
+0001aa90: 7379 6e0a 6060 600a 0a55 706f 6e20 7468  syn.```..Upon th
+0001aaa0: 6520 6163 7175 6973 6974 696f 6e20 6f66  e acquisition of
+0001aab0: 2074 6865 2063 6f6c 6c69 6e65 6172 2072   the collinear r
+0001aac0: 6573 756c 7473 2075 7369 6e67 2060 7767  esults using `wg
+0001aad0: 6420 7379 6e60 2c20 7468 6520 7361 6d65  d syn`, the same
+0001aae0: 2063 6f6c 6c69 6e65 6172 2070 6c6f 7473   collinear plots
+0001aaf0: 2063 616e 2062 6520 616c 736f 2070 726f   can be also pro
+0001ab00: 6475 6365 6420 6279 2060 7767 6420 7669  duced by `wgd vi
+0001ab10: 7a60 2075 7369 6e67 2074 6865 2063 6f6d  z` using the com
+0001ab20: 6d61 6e64 2062 656c 6f77 2e0a 6060 600a  mand below..```.
+0001ab30: 7767 6420 7669 7a20 2d2d 706c 6f74 7379  wgd viz --plotsy
+0001ab40: 6e20 2d73 6d20 7767 645f 6f72 7468 6f5f  n -sm wgd_ortho_
+0001ab50: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+0001ab60: 7365 676d 656e 7473 2e74 7874 202d 6170  segments.txt -ap
+0001ab70: 2077 6764 5f6f 7274 686f 5f73 796e 2f69   wgd_ortho_syn/i
+0001ab80: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+0001ab90: 7270 6f69 6e74 732e 7478 7420 2d6d 7420  rpoints.txt -mt 
+0001aba0: 7767 645f 6f72 7468 6f5f 7379 6e2f 6961  wgd_ortho_syn/ia
+0001abb0: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
+0001abc0: 6c69 636f 6e73 2e74 7874 202d 6774 2077  licons.txt -gt w
+0001abd0: 6764 5f6f 7274 686f 5f73 796e 2f67 656e  gd_ortho_syn/gen
+0001abe0: 652d 7461 626c 652e 6373 7620 2d64 2077  e-table.csv -d w
+0001abf0: 6764 5f6f 7274 686f 5f6b 732f 4f72 7468  gd_ortho_ks/Orth
+0001ac00: 6f67 726f 7570 732e 7370 2e74 7376 2e6b  ogroups.sp.tsv.k
+0001ac10: 732e 7473 7620 2d6f 2077 6764 5f6f 7274  s.tsv -o wgd_ort
+0001ac20: 686f 5f76 697a 0a60 6060 0a0a 215b 5d28  ho_viz.```..![](
+0001ac30: 6461 7461 2f6f 7274 686f 5f73 796e 5f72  data/ortho_syn_r
+0001ac40: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
+0001ac50: 5f63 6f65 7275 6c65 615f 5669 7469 735f  _coerulea_Vitis_
+0001ac60: 7669 6e69 6665 7261 5f6d 756c 7469 706c  vinifera_multipl
+0001ac70: 6963 6f6e 735f 6c65 7665 6c2e 706e 6729  icons_level.png)
+0001ac80: 0a54 6865 2061 626f 7665 2060 6475 7053  .The above `dupS
+0001ac90: 7461 636b 6020 706c 6f74 2073 686f 7773  tack` plot shows
+0001aca0: 2074 6865 2064 6973 7472 6962 7574 696f   the distributio
+0001acb0: 6e20 6f66 2064 7570 6c69 6361 7465 6420  n of duplicated 
+0001acc0: 7365 676d 656e 7473 206f 6620 2a41 7175  segments of *Aqu
+0001acd0: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+0001ace0: 2063 6f6d 7061 7265 6420 746f 2069 7473   compared to its
+0001acf0: 656c 6620 2869 6e20 6772 6565 6e29 2061  elf (in green) a
+0001ad00: 6e64 2063 6f6d 7061 7265 6420 746f 202a  nd compared to *
+0001ad10: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
+0001ad20: 2869 6e20 626c 7565 2920 6f76 6572 2074  (in blue) over t
+0001ad30: 6865 2063 6872 6f6d 6f73 6f6d 6573 206f  he chromosomes o
+0001ad40: 6620 2a41 2e20 636f 6572 756c 6561 2a2e  f *A. coerulea*.
+0001ad50: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
+0001ad60: 7379 6e5f 7265 7375 6c74 732f 4f76 6572  syn_results/Over
+0001ad70: 616c 6c73 7065 6369 6573 5f4b 732e 646f  allspecies_Ks.do
+0001ad80: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+0001ad90: 0a54 6865 2061 626f 7665 202a 4b2a 3c73  .The above *K*<s
+0001ada0: 7562 3e53 3c2f 7375 623e 2064 6f74 706c  ub>S</sub> dotpl
+0001adb0: 6f74 2069 6e20 756e 6974 206f 6620 6765  ot in unit of ge
+0001adc0: 6e65 2073 686f 7773 2074 6865 206f 7665  ne shows the ove
+0001add0: 7261 6c6c 2064 6973 7472 6962 7574 696f  rall distributio
+0001ade0: 6e20 6f66 2063 6f6c 6c69 6e65 6172 6974  n of collinearit
+0001adf0: 7920 6163 6f72 7373 2074 6865 2066 6f75  y acorss the fou
+0001ae00: 7220 7370 6563 6965 7320 696e 766f 6c76  r species involv
+0001ae10: 6564 2e0a 215b 5d28 6461 7461 2f6f 7274  ed..![](data/ort
+0001ae20: 686f 5f73 796e 5f72 6573 756c 7473 2f4f  ho_syn_results/O
+0001ae30: 7665 7261 6c6c 7370 6563 6965 732e 646f  verallspecies.do
+0001ae40: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+0001ae50: 0a54 6865 2061 626f 7665 2064 6f74 706c  .The above dotpl
+0001ae60: 6f74 2069 7320 7769 7468 6f75 7420 7468  ot is without th
+0001ae70: 6520 616e 6e6f 7461 7469 6f6e 206f 6620  e annotation of 
+0001ae80: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0001ae90: 6167 6573 2063 6f6d 7061 7265 6420 746f  ages compared to
+0001aea0: 2074 6865 206c 6173 7420 6f6e 652e 0a21   the last one..!
+0001aeb0: 5b5d 2864 6174 612f 6f72 7468 6f5f 7379  [](data/ortho_sy
+0001aec0: 6e5f 7265 7375 6c74 732f 5379 6e64 6570  n_results/Syndep
+0001aed0: 7468 2e73 7667 290a 5468 6520 6162 6f76  th.svg).The abov
+0001aee0: 6520 5379 6e64 6570 7468 2070 6c6f 7420  e Syndepth plot 
+0001aef0: 7368 6f77 7320 7468 6520 636f 6c6c 696e  shows the collin
+0001af00: 6561 7220 7261 7469 6f20 6163 6f72 7373  ear ratio acorss
+0001af10: 2061 6c6c 2073 7065 6369 6573 2070 6169   all species pai
+0001af20: 7273 2028 696e 7472 612d 7370 6563 6966  rs (intra-specif
+0001af30: 6963 2063 6f6d 7061 7269 736f 6e20 696e  ic comparison in
+0001af40: 2067 7265 656e 2077 6869 6c65 2069 6e74   green while int
+0001af50: 6572 2d73 7065 6369 6669 6320 636f 6d70  er-specific comp
+0001af60: 6172 6973 6f6e 2069 6e20 626c 7565 292e  arison in blue).
+0001af70: 0a0a 2323 2043 6974 6174 696f 6e0a 200a  ..## Citation. .
+0001af80: 506c 6561 7365 2063 6974 6520 7573 2061  Please cite us a
+0001af90: 7420 6874 7470 733a 2f2f 646f 692e 6f72  t https://doi.or
+0001afa0: 672f 3130 2e31 3030 372f 3937 382d 312d  g/10.1007/978-1-
+0001afb0: 3037 3136 2d32 3536 312d 335f 3120 616e  0716-2561-3_1 an
+0001afc0: 6420 6874 7470 733a 2f2f 646f 692e 6f72  d https://doi.or
+0001afd0: 672f 3130 2e31 3039 332f 6269 6f69 6e66  g/10.1093/bioinf
+0001afe0: 6f72 6d61 7469 6373 2f62 7461 6532 3732  ormatics/btae272
+0001aff0: 2e0a 0a60 6060 0a48 656e 6763 6869 2043  ...```.Hengchi C
+0001b000: 6865 6e2c 2041 7274 6875 7220 5a77 6165  hen, Arthur Zwae
+0001b010: 6e65 706f 656c 2028 3230 3233 292e 2049  nepoel (2023). I
+0001b020: 6e66 6572 656e 6365 206f 6620 416e 6369  nference of Anci
+0001b030: 656e 7420 506f 6c79 706c 6f69 6479 2066  ent Polyploidy f
+0001b040: 726f 6d20 4765 6e6f 6d69 6320 4461 7461  rom Genomic Data
+0001b050: 2e20 496e 3a20 5661 6e20 6465 2050 6565  . In: Van de Pee
+0001b060: 722c 2059 2e20 2865 6473 2920 506f 6c79  r, Y. (eds) Poly
+0001b070: 706c 6f69 6479 2e20 4d65 7468 6f64 7320  ploidy. Methods 
+0001b080: 696e 204d 6f6c 6563 756c 6172 2042 696f  in Molecular Bio
+0001b090: 6c6f 6779 2c20 766f 6c20 3235 3435 2e20  logy, vol 2545. 
+0001b0a0: 4875 6d61 6e61 2c20 4e65 7720 596f 726b  Humana, New York
+0001b0b0: 2c20 4e59 2e20 6874 7470 733a 2f2f 646f  , NY. https://do
+0001b0c0: 692e 6f72 672f 3130 2e31 3030 372f 3937  i.org/10.1007/97
+0001b0d0: 382d 312d 3037 3136 2d32 3536 312d 335f  8-1-0716-2561-3_
+0001b0e0: 310a 4865 6e67 6368 6920 4368 656e 2c20  1.Hengchi Chen, 
+0001b0f0: 4172 7468 7572 205a 7761 656e 6570 6f65  Arthur Zwaenepoe
+0001b100: 6c2c 2059 7665 7320 5661 6e20 6465 2050  l, Yves Van de P
+0001b110: 6565 7220 2832 3032 3429 2e20 7767 6420  eer (2024). wgd 
+0001b120: 7632 3a20 6120 7375 6974 6520 6f66 2074  v2: a suite of t
+0001b130: 6f6f 6c73 2074 6f20 756e 636f 7665 7220  ools to uncover 
+0001b140: 616e 6420 6461 7465 2061 6e63 6965 6e74  and date ancient
+0001b150: 2070 6f6c 7970 6c6f 6964 7920 616e 6420   polyploidy and 
+0001b160: 7768 6f6c 652d 6765 6e6f 6d65 2064 7570  whole-genome dup
+0001b170: 6c69 6361 7469 6f6e 2e20 4269 6f69 6e66  lication. Bioinf
+0001b180: 6f72 6d61 7469 6373 2c20 566f 6c75 6d65  ormatics, Volume
+0001b190: 2034 302c 2049 7373 7565 2035 2c20 4d61   40, Issue 5, Ma
+0001b1a0: 7920 3230 3234 2c20 6274 6165 3237 322c  y 2024, btae272,
+0001b1b0: 2068 7474 7073 3a2f 2f64 6f69 2e6f 7267   https://doi.org
+0001b1c0: 2f31 302e 3130 3933 2f62 696f 696e 666f  /10.1093/bioinfo
+0001b1d0: 726d 6174 6963 732f 6274 6165 3237 320a  rmatics/btae272.
+0001b1e0: 6060 600a 0a46 6f72 2063 6974 6174 696f  ```..For citatio
+0001b1f0: 6e20 6f66 2074 6865 2074 6f6f 6c73 2075  n of the tools u
+0001b200: 7365 6420 696e 2077 6764 2c20 706c 6561  sed in wgd, plea
+0001b210: 7365 2063 6f6e 7375 6c74 2074 6865 2064  se consult the d
+0001b220: 6f63 756d 656e 7461 7469 6f6e 2061 740a  ocumentation at.
+0001b230: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
+0001b240: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
+0001b250: 6c61 7465 7374 2f63 6974 6174 696f 6e2e  latest/citation.
+0001b260: 6874 6d6c 2e0a 0a                        html...
```

### Comparing `wgd-2.0.37/README.md` & `wgd-2.0.38/wgd.egg-info/PKG-INFO`

 * *Files 16% similar despite different names*

```diff
@@ -1,6091 +1,6951 @@
-00000000: 3c64 6976 2061 6c69 676e 3d22 6365 6e74  <div align="cent
-00000010: 6572 223e 0a0a 2320 6077 6764 2076 3260  er">..# `wgd v2`
-00000020: 203a 2061 2073 7569 7465 2074 6f6f 6c20   : a suite tool 
-00000030: 6f66 2057 4744 2069 6e66 6572 656e 6365  of WGD inference
-00000040: 2061 6e64 2074 696d 696e 670a 0a5b 215b   and timing..[![
-00000050: 4275 696c 6420 5374 6174 7573 5d28 6874  Build Status](ht
-00000060: 7470 733a 2f2f 6170 702e 7472 6176 6973  tps://app.travis
-00000070: 2d63 692e 636f 6d2f 6865 6368 652d 7073  -ci.com/heche-ps
-00000080: 622f 7767 642e 7376 673f 6272 616e 6368  b/wgd.svg?branch
-00000090: 3d70 6879 6c6f 6461 7469 6e67 295d 2868  =phylodating)](h
-000000a0: 7474 7073 3a2f 2f61 7070 2e74 7261 7669  ttps://app.travi
-000000b0: 732d 6369 2e63 6f6d 2f68 6563 6865 2d70  s-ci.com/heche-p
-000000c0: 7362 2f77 6764 290a 5b21 5b44 6f63 756d  sb/wgd).[![Docum
-000000d0: 656e 7461 7469 6f6e 2053 7461 7475 735d  entation Status]
-000000e0: 2868 7474 7073 3a2f 2f72 6561 6474 6865  (https://readthe
-000000f0: 646f 6373 2e6f 7267 2f70 726f 6a65 6374  docs.org/project
-00000100: 732f 7767 6476 322f 6261 6467 652f 3f76  s/wgdv2/badge/?v
-00000110: 6572 7369 6f6e 3d6c 6174 6573 7429 5d28  ersion=latest)](
-00000120: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
-00000130: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
-00000140: 6c61 7465 7374 2f3f 6261 6467 653d 6c61  latest/?badge=la
-00000150: 7465 7374 290a 5b21 5b6c 6963 656e 7365  test).[![license
-00000160: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
-00000170: 6965 6c64 732e 696f 2f70 7970 692f 6c2f  ields.io/pypi/l/
-00000180: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
-00000190: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
-000001a0: 672f 7079 7069 2f77 6764 290a 5b21 5b4c  g/pypi/wgd).[![L
-000001b0: 6174 6573 7420 5079 5049 2076 6572 7369  atest PyPI versi
-000001c0: 6f6e 5d28 6874 7470 733a 2f2f 696d 672e  on](https://img.
-000001d0: 7368 6965 6c64 732e 696f 2f70 7970 692f  shields.io/pypi/
-000001e0: 762f 7767 642e 7376 6729 5d28 6874 7470  v/wgd.svg)](http
-000001f0: 733a 2f2f 7079 7069 2e70 7974 686f 6e2e  s://pypi.python.
-00000200: 6f72 672f 7079 7069 2f77 6764 290a 5b21  org/pypi/wgd).[!
-00000210: 5b69 6e73 7461 6c6c 2077 6974 6820 6269  [install with bi
-00000220: 6f63 6f6e 6461 5d28 6874 7470 733a 2f2f  oconda](https://
-00000230: 696d 672e 7368 6965 6c64 732e 696f 2f62  img.shields.io/b
-00000240: 6164 6765 2f69 6e73 7461 6c6c 2532 3077  adge/install%20w
-00000250: 6974 682d 6269 6f63 6f6e 6461 2d62 7269  ith-bioconda-bri
-00000260: 6768 7467 7265 656e 2e73 7667 3f73 7479  ghtgreen.svg?sty
-00000270: 6c65 3d66 6c61 7429 5d28 6874 7470 3a2f  le=flat)](http:/
-00000280: 2f62 696f 636f 6e64 612e 6769 7468 7562  /bioconda.github
-00000290: 2e69 6f2f 7265 6369 7065 732f 7767 642f  .io/recipes/wgd/
-000002a0: 5245 4144 4d45 2e68 746d 6c29 0a5b 215b  README.html).[![
-000002b0: 446f 776e 6c6f 6164 735d 2868 7474 7073  Downloads](https
-000002c0: 3a2f 2f70 6570 792e 7465 6368 2f62 6164  ://pepy.tech/bad
-000002d0: 6765 2f77 6764 295d 2868 7474 7073 3a2f  ge/wgd)](https:/
-000002e0: 2f70 6570 792e 7465 6368 2f70 726f 6a65  /pepy.tech/proje
-000002f0: 6374 2f77 6764 290a 0a2a 2a48 656e 6763  ct/wgd)..**Hengc
-00000300: 6869 2043 6865 6e2c 2041 7274 6875 7220  hi Chen, Arthur 
-00000310: 5a77 6165 6e65 706f 656c 2c20 5976 6573  Zwaenepoel, Yves
-00000320: 2056 616e 2064 6520 5065 6572 2a2a 0a0a   Van de Peer**..
-00000330: 5b2a 2a42 696f 696e 666f 726d 6174 6963  [**Bioinformatic
-00000340: 7320 2620 4576 6f6c 7574 696f 6e61 7279  s & Evolutionary
-00000350: 2047 656e 6f6d 6963 7320 4772 6f75 702a   Genomics Group*
-00000360: 2a5d 2868 7474 7073 3a2f 2f77 7777 2e76  *](https://www.v
-00000370: 616e 6465 7065 6572 6c61 622e 6f72 672f  andepeerlab.org/
-00000380: 292a 2a2c 2056 4942 2d55 4765 6e74 2043  )**, VIB-UGent C
-00000390: 656e 7465 7220 666f 7220 506c 616e 7420  enter for Plant 
-000003a0: 5379 7374 656d 7320 4269 6f6c 6f67 792a  Systems Biology*
-000003b0: 2a0a 0a5b 2a2a 496e 7472 6f64 7563 7469  *..[**Introducti
-000003c0: 6f6e 2a2a 5d28 2369 6e74 726f 6475 6374  on**](#introduct
-000003d0: 696f 6e29 207c 200a 5b2a 2a49 6e73 7461  ion) | .[**Insta
-000003e0: 6c6c 6174 696f 6e2a 2a5d 2823 696e 7374  llation**](#inst
-000003f0: 616c 6c61 7469 6f6e 2920 7c20 0a5b 2a2a  allation) | .[**
-00000400: 5061 7261 6d65 7465 7273 2a2a 5d28 2370  Parameters**](#p
-00000410: 6172 616d 6574 6572 7329 207c 200a 5b2a  arameters) | .[*
-00000420: 2a55 7361 6765 2a2a 5d28 2375 7361 6765  *Usage**](#usage
-00000430: 2920 7c20 0a5b 2a2a 496c 6c75 7374 7261  ) | .[**Illustra
-00000440: 7469 6f6e 2a2a 5d28 2369 6c6c 7573 7472  tion**](#illustr
-00000450: 6174 696f 6e29 207c 0a5b 2a2a 446f 6375  ation) |.[**Docu
-00000460: 6d65 6e74 6174 696f 6e2a 2a5d 2868 7474  mentation**](htt
-00000470: 7073 3a2f 2f77 6764 7632 2e72 6561 6474  ps://wgdv2.readt
-00000480: 6865 646f 6373 2e69 6f2f 656e 2f6c 6174  hedocs.io/en/lat
-00000490: 6573 742f 3f62 6164 6765 3d6c 6174 6573  est/?badge=lates
-000004a0: 7429 207c 0a5b 2a2a 4369 7461 7469 6f6e  t) |.[**Citation
-000004b0: 2a2a 5d28 2363 6974 6174 696f 6e29 0a0a  **](#citation)..
-000004c0: 3c2f 6469 763e 0a0a 6077 6764 2076 3260  </div>..`wgd v2`
-000004d0: 2069 7320 6120 7079 7468 6f6e 2070 6163   is a python pac
-000004e0: 6b61 6765 2075 7067 7261 6465 6420 6672  kage upgraded fr
-000004f0: 6f6d 2074 6865 206f 7269 6769 6e61 6c20  om the original 
-00000500: 6077 6764 6020 7061 636b 6167 6520 6169  `wgd` package ai
-00000510: 6d69 6e67 2066 6f72 2074 6865 2069 6e66  ming for the inf
-00000520: 6572 656e 6365 2061 6e64 2074 696d 696e  erence and timin
-00000530: 6720 6f66 2061 6e63 6965 6e74 2077 686f  g of ancient who
-00000540: 6c65 2d67 656e 6f6d 6520 6475 706c 6963  le-genome duplic
-00000550: 6174 696f 6e20 2857 4744 2920 6576 656e  ation (WGD) even
-00000560: 7473 2e20 466f 7220 7468 6520 7072 6f70  ts. For the prop
-00000570: 6f73 6520 6f66 2069 6c6c 7573 7472 6174  ose of illustrat
-00000580: 696e 6720 7468 6520 7072 696e 6369 706c  ing the principl
-00000590: 6520 616e 6420 7573 6167 6520 6f66 2060  e and usage of `
-000005a0: 7767 6420 7632 602c 2077 6520 636f 6d70  wgd v2`, we comp
-000005b0: 696c 6564 2074 6869 7320 646f 6375 6d65  iled this docume
-000005c0: 6e74 6174 696f 6e2e 2042 656c 6f77 2077  ntation. Below w
-000005d0: 6520 6669 7273 7420 6761 7665 2061 6e20  e first gave an 
-000005e0: 696e 7472 6f64 7563 7469 6f6e 206f 7665  introduction ove
-000005f0: 7220 7468 6520 7363 6f70 6520 616e 6420  r the scope and 
-00000600: 6d65 6368 616e 6973 6d20 6f66 2060 7767  mechanism of `wg
-00000610: 6420 7632 6020 616e 6420 7468 656e 2074  d v2` and then t
-00000620: 6865 2070 7261 6374 6963 616c 2069 6e66  he practical inf
-00000630: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
-00000640: 696e 6720 746f 2074 6865 2069 6e73 7461  ing to the insta
-00000650: 6c6c 6174 696f 6e20 616e 6420 7573 6167  llation and usag
-00000660: 652e 2041 6e20 6578 616d 706c 6172 2077  e. An examplar w
-00000670: 6f72 6b66 6c6f 7720 6973 2070 726f 7669  orkflow is provi
-00000680: 6465 6420 696e 2074 6865 2074 7574 6f72  ded in the tutor
-00000690: 6961 6c20 7365 6374 696f 6e20 6f6e 2068  ial section on h
-000006a0: 6f77 2074 6f20 7365 656b 2065 7669 6465  ow to seek evide
-000006b0: 6e63 6520 666f 7220 6120 7075 7461 7469  nce for a putati
-000006c0: 7665 2057 4744 2065 7665 6e74 2061 6e64  ve WGD event and
-000006d0: 2070 6572 666f 726d 2070 726f 7065 7220   perform proper 
-000006e0: 7469 6d69 6e67 2077 6974 6820 6120 6672  timing with a fr
-000006f0: 6573 686c 7920 6f62 7461 696e 6564 2067  eshly obtained g
-00000700: 656e 6f6d 6520 6173 7365 6d62 6c79 2069  enome assembly i
-00000710: 6e20 6861 6e64 2e20 466f 7220 7468 6f73  n hand. For thos
-00000720: 6520 7768 6f20 6172 6520 696e 7465 7265  e who are intere
-00000730: 7374 6564 2069 6e20 6d6f 7265 2074 6865  sted in more the
-00000740: 6f72 6574 6963 616c 2064 6574 6169 6c73  oretical details
-00000750: 2c20 7765 2072 6563 6f6d 6d65 6e64 2074  , we recommend t
-00000760: 7572 6e69 6e67 2074 6f20 6f75 7220 7061  urning to our pa
-00000770: 7065 7220 616e 6420 626f 6f6b 2063 6861  per and book cha
-00000780: 7074 6572 2066 6f72 206d 6f72 6520 6465  pter for more de
-00000790: 7461 696c 6564 2064 6573 6372 6970 7469  tailed descripti
-000007a0: 6f6e 2061 6e64 2069 6e73 6967 6874 6675  on and insightfu
-000007b0: 6c20 6469 7363 7573 7369 6f6e 732e 2054  l discussions. T
-000007c0: 6865 206b 6579 2069 6d70 726f 7665 6420  he key improved 
-000007d0: 6665 6174 7572 6573 206f 6620 6077 6764  features of `wgd
-000007e0: 2076 3260 2061 7265 2064 656d 6f6e 7374   v2` are demonst
-000007f0: 7261 7465 6420 696e 206f 7572 206c 6174  rated in our lat
-00000800: 6573 7420 6d61 6e75 7363 7269 7074 206f  est manuscript o
-00000810: 6e20 5b42 696f 696e 666f 726d 6174 6963  n [Bioinformatic
-00000820: 735d 2868 7474 7073 3a2f 2f64 6f69 2e6f  s](https://doi.o
-00000830: 7267 2f31 302e 3130 3933 2f62 696f 696e  rg/10.1093/bioin
-00000840: 666f 726d 6174 6963 732f 6274 6165 3237  formatics/btae27
-00000850: 3229 2e20 4966 2079 6f75 2075 7365 2060  2). If you use `
-00000860: 7767 6420 7632 6020 696e 2079 6f75 7220  wgd v2` in your 
-00000870: 7265 7365 6172 6368 2c20 706c 6561 7365  research, please
-00000880: 2063 6974 6520 7573 2061 7320 7375 6767   cite us as sugg
-00000890: 6573 7465 6420 696e 205b 4369 7461 7469  ested in [Citati
-000008a0: 6f6e 5d28 2363 6974 6174 696f 6e29 2073  on](#citation) s
-000008b0: 6563 7469 6f6e 2e0a 0a23 2320 496e 7472  ection...## Intr
-000008c0: 6f64 7563 7469 6f6e 0a0a 506f 6c79 706c  oduction..Polypl
-000008d0: 6f69 6469 7a61 7469 6f6e 732c 2074 6865  oidizations, the
-000008e0: 2065 766f 6c75 7469 6f6e 6172 7920 7072   evolutionary pr
-000008f0: 6f63 6573 7320 7468 6174 2074 6865 2065  ocess that the e
-00000900: 6e74 6972 6520 6765 6e6f 6d65 206f 6620  ntire genome of 
-00000910: 616e 206f 7267 616e 6973 6d20 6973 2064  an organism is d
-00000920: 7570 6c69 6361 7465 642c 2061 6c73 6f20  uplicated, also 
-00000930: 6e61 6d65 6420 6173 2077 686f 6c65 2d67  named as whole-g
-00000940: 656e 6f6d 6520 6475 706c 6963 6174 696f  enome duplicatio
-00000950: 6e73 2028 5747 4473 292c 206f 6363 7572  ns (WGDs), occur
-00000960: 2072 6563 7572 7265 6e74 6c79 2061 6372   recurrently acr
-00000970: 6f73 7320 7468 6520 7472 6565 206f 6620  oss the tree of 
-00000980: 6c69 6665 2e20 5468 6572 6520 6172 6520  life. There are 
-00000990: 7477 6f20 6d6f 6465 7320 6f66 2070 6f6c  two modes of pol
-000009a0: 7970 6c6f 6964 697a 6174 696f 6e73 2c20  yploidizations, 
-000009b0: 6175 746f 706f 6c79 706c 6f69 6469 7a61  autopolyploidiza
-000009c0: 7469 6f6e 7320 616e 6420 616c 6c6f 706f  tions and allopo
-000009d0: 6c79 706c 6f69 6469 7a61 7469 6f6e 732e  lyploidizations.
-000009e0: 2041 7574 6f70 6f6c 7970 6c6f 6964 697a   Autopolyploidiz
-000009f0: 6174 696f 6e73 2061 7265 2074 6865 2064  ations are the d
-00000a00: 7570 6c69 6361 7469 6f6e 206f 6620 7468  uplication of th
-00000a10: 6520 7361 6d65 2067 656e 6f6d 652c 2072  e same genome, r
-00000a20: 6573 756c 7469 6e67 2069 6e20 7477 6f20  esulting in two 
-00000a30: 6964 656e 7469 6361 6c20 7375 6267 656e  identical subgen
-00000a40: 6f6d 6573 2061 7420 7468 6520 7469 6d65  omes at the time
-00000a50: 2069 7420 656d 6572 6765 642e 2057 6869   it emerged. Whi
-00000a60: 6c65 2074 6865 2061 6c6c 6f70 6f6c 7970  le the allopolyp
-00000a70: 6c6f 6964 697a 6174 696f 6e73 2061 7265  loidizations are
-00000a80: 206e 6f72 6d61 6c6c 7920 6163 6869 6576   normally achiev
-00000a90: 6564 2069 6e20 7477 6f20 7374 6570 732c  ed in two steps,
-00000aa0: 2066 6972 7374 2074 6865 2068 7962 7269   first the hybri
-00000ab0: 6469 7a61 7469 6f6e 2062 6574 7765 656e  dization between
-00000ac0: 2074 776f 2064 6966 6665 7265 6e74 2073   two different s
-00000ad0: 7065 6369 6573 2c20 7265 7375 6c74 696e  pecies, resultin
-00000ae0: 6720 696e 2074 6865 2061 7269 7369 6e67  g in the arising
-00000af0: 206f 6620 7472 616e 7369 656e 7420 686f   of transient ho
-00000b00: 6d6f 706c 6f69 6479 2c73 6563 6f6e 6420  moploidy,second 
-00000b10: 7468 6520 6475 706c 6963 6174 696f 6e20  the duplication 
-00000b20: 6f66 2074 6865 2068 6f6d 6f70 6c6f 6964  of the homoploid
-00000b30: 792c 2072 6573 756c 7469 6e67 2069 6e20  y, resulting in 
-00000b40: 7468 6520 656d 6572 6765 6e63 6520 6f66  the emergence of
-00000b50: 2061 6c6c 6f70 6f6c 7970 6c6f 6964 792e   allopolyploidy.
-00000b60: 2044 7565 2074 6f20 7468 6520 756e 7374   Due to the unst
-00000b70: 6162 696c 6974 7920 616e 6420 756e 6261  ability and unba
-00000b80: 6c61 6e63 6564 2074 6574 7261 736f 6d69  lanced tetrasomi
-00000b90: 6320 696e 6865 7269 7461 6e63 652c 2066  c inheritance, f
-00000ba0: 6f72 2069 6e73 7461 6e63 6520 7468 6520  or instance the 
-00000bb0: 6e75 636c 6561 722d 6379 746f 706c 6173  nuclear-cytoplas
-00000bc0: 6d69 6320 696e 636f 6d70 6174 6962 696c  mic incompatibil
-00000bd0: 6974 792c 2074 6865 2070 6f6c 7970 6c6f  ity, the polyplo
-00000be0: 6964 7920 6765 6e6f 6d65 2077 696c 6c20  idy genome will 
-00000bf0: 7468 656e 2065 7870 6572 6965 6e63 6520  then experience 
-00000c00: 6120 7072 6f63 6573 7320 6361 6c6c 6564  a process called
-00000c10: 2064 6970 6c6f 6964 697a 6174 696f 6e2c   diploidization,
-00000c20: 2061 6c73 6f20 6e61 6d65 6420 6173 2066   also named as f
-00000c30: 7261 6374 696f 6e61 7469 6f6e 2c20 6475  ractionation, du
-00000c40: 7269 6e67 2077 6869 6368 2061 206c 6172  ring which a lar
-00000c50: 6765 2070 6f72 7469 6f6e 206f 6620 6765  ge portion of ge
-00000c60: 6e65 2064 7570 6c69 6361 7465 7320 7769  ne duplicates wi
-00000c70: 6c6c 2067 6574 206c 6f73 7420 616e 6420  ll get lost and 
-00000c80: 6f6e 6c79 2061 2066 7261 6374 696f 6e20  only a fraction 
-00000c90: 6361 6e20 6265 2072 6574 6169 6e65 642e  can be retained.
-00000ca0: 2054 6865 2074 7261 6365 7320 6f66 2070   The traces of p
-00000cb0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e73  olyploidizations
-00000cc0: 2063 616e 2062 6520 7468 7573 2075 6e65   can be thus une
-00000cd0: 6172 7468 6564 2066 726f 6d20 7468 6573  arthed from thes
-00000ce0: 6520 7265 7461 696e 6564 2067 656e 6520  e retained gene 
-00000cf0: 6475 706c 6963 6174 6573 2e20 5468 7265  duplicates. Thre
-00000d00: 6520 6170 7072 6f61 6368 6573 2062 6173  e approaches bas
-00000d10: 6564 206f 6e20 6765 6e65 2064 7570 6c69  ed on gene dupli
-00000d20: 6361 7465 732c 206e 616d 656c 792c 202a  cates, namely, *
-00000d30: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206d  K*<sub>S</sub> m
-00000d40: 6574 686f 642c 2067 656e 6520 7472 6565  ethod, gene tree
-00000d50: 202d 2073 7065 6369 6573 2074 7265 6520   - species tree 
-00000d60: 7265 636f 6e63 696c 6961 7469 6f6e 206d  reconciliation m
-00000d70: 6574 686f 6420 616e 6420 7379 6e74 656e  ethod and synten
-00000d80: 7920 6d65 7468 6f64 2c20 6172 6520 636f  y method, are co
-00000d90: 6d6d 6f6e 6c79 2075 7365 6420 696e 2064  mmonly used in d
-00000da0: 6574 6563 7469 6e67 2065 7669 6465 6e63  etecting evidenc
-00000db0: 6520 666f 7220 5747 4473 2e20 5468 6520  e for WGDs. The 
-00000dc0: 6765 6e65 2074 7265 6520 2d20 7370 6563  gene tree - spec
-00000dd0: 6965 7320 7472 6565 2072 6563 6f6e 6369  ies tree reconci
-00000de0: 6c69 6174 696f 6e20 6d65 7468 6f64 2069  liation method i
-00000df0: 7320 6e6f 7420 7769 7468 696e 2074 6865  s not within the
-00000e00: 2073 636f 7065 206f 6620 6077 6764 2076   scope of `wgd v
-00000e10: 3260 2c20 6275 7420 7765 206b 696e 646c  2`, but we kindl
-00000e20: 7920 7265 6665 7220 7265 6164 6572 7320  y refer readers 
-00000e30: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
-00000e40: 6564 2074 6f20 7468 6520 7068 796c 6f67  ed to the phylog
-00000e50: 656e 6f6d 6963 2070 726f 6772 616d 2064  enomic program d
-00000e60: 6576 656c 6f70 6564 2062 7920 4172 7468  eveloped by Arth
-00000e70: 7572 205a 7761 656e 6570 6f65 6c20 6e61  ur Zwaenepoel na
-00000e80: 6d65 6420 5b57 4841 4c45 5d28 6874 7470  med [WHALE](http
-00000e90: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f61  s://github.com/a
-00000ea0: 727a 7761 2f57 6861 6c65 2e6a 6c29 2061  rzwa/Whale.jl) a
-00000eb0: 6e64 2074 6865 2061 7373 6f63 6961 7465  nd the associate
-00000ec0: 6420 5b70 6170 6572 5d28 6874 7470 733a  d [paper](https:
-00000ed0: 2f2f 646f 692e 6f72 672f 3130 2e31 3039  //doi.org/10.109
-00000ee0: 332f 6d6f 6c62 6576 2f6d 737a 3038 3829  3/molbev/msz088)
-00000ef0: 2066 6f72 206d 6f72 6520 7465 6368 6e69   for more techni
-00000f00: 6361 6c20 616e 6420 7468 656f 7265 7469  cal and theoreti
-00000f10: 6361 6c20 6465 7461 696c 732e 0a0a 5468  cal details...Th
-00000f20: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-00000f30: 3e20 6d65 7468 6f64 2069 7320 6573 7461  > method is esta
-00000f40: 626c 6973 6865 6420 6f6e 2061 206d 6f64  blished on a mod
-00000f50: 656c 206f 6620 6765 6e65 2066 616d 696c  el of gene famil
-00000f60: 7920 6576 6f6c 7574 696f 6e20 7468 6174  y evolution that
-00000f70: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
-00000f80: 7920 6973 2061 6c6c 6f77 6564 2074 6f20  y is allowed to 
-00000f90: 6576 6f6c 7665 2076 6961 2067 656e 6520  evolve via gene 
-00000fa0: 6475 706c 6963 6174 696f 6e20 616e 6420  duplication and 
-00000fb0: 6c6f 7373 2e20 4e6f 7465 2074 6861 7420  loss. Note that 
-00000fc0: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
-00000fd0: 6865 7265 2069 7320 6173 7375 6d65 6420  here is assumed 
-00000fe0: 746f 2062 6520 7468 6520 636c 7573 7465  to be the cluste
-00000ff0: 7220 6f66 2061 6c6c 2067 656e 6573 2064  r of all genes d
-00001000: 6573 6365 6e64 6564 2066 726f 6d20 616e  escended from an
-00001010: 2061 6e63 6573 7472 616c 2067 656e 6520   ancestral gene 
-00001020: 696e 2061 2073 696e 676c 6520 6765 6e6f  in a single geno
-00001030: 6d65 2e20 5265 636f 7665 7269 6e67 2074  me. Recovering t
-00001040: 6865 2067 656e 6520 7472 6565 206f 6620  he gene tree of 
-00001050: 7375 6368 2067 656e 6520 6661 6d69 6c79  such gene family
-00001060: 2069 6e66 6f72 6d73 2074 6865 2074 696d   informs the tim
-00001070: 696e 672c 2073 6369 6c69 6365 7420 7468  ing, scilicet th
-00001080: 6520 6167 652c 206f 6620 6765 6e65 2064  e age, of gene d
-00001090: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
-000010a0: 732e 2054 6865 2061 6765 2072 6566 6572  s. The age refer
-000010b0: 6564 2068 6572 652c 2069 7320 6e6f 7420  ed here, is not 
-000010c0: 696e 2072 6561 6c20 6765 6f6c 6f67 6963  in real geologic
-000010d0: 616c 2074 696d 652c 2062 7574 2069 6e20  al time, but in 
-000010e0: 7468 6520 756e 6974 206f 6620 6576 6f6c  the unit of evol
-000010f0: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
-00001100: 652c 2069 2e65 2e2c 2074 6865 206e 756d  e, i.e., the num
-00001110: 6265 7220 6f66 2073 7562 7374 6974 7574  ber of substitut
-00001120: 696f 6e73 2070 6572 2073 6974 652e 2057  ions per site. W
-00001130: 6865 6e20 7468 6520 6576 6f6c 7574 696f  hen the evolutio
-00001140: 6e61 7279 2072 6174 6520 7265 6d61 696e  nary rate remain
-00001150: 7320 6170 7072 6f78 696d 6174 656c 7920  s approximately 
-00001160: 636f 6e73 7461 6e74 2c20 7468 6520 6576  constant, the ev
-00001170: 6f6c 7574 696f 6e61 7279 2064 6973 7461  olutionary dista
-00001180: 6e63 6520 6973 2074 6865 6e20 7375 7070  nce is then supp
-00001190: 6f73 6564 2074 6f20 6265 2070 726f 706f  osed to be propo
-000011a0: 7274 696f 6e61 6c20 746f 2074 6865 2072  rtional to the r
-000011b0: 6561 6c20 6576 6f6c 7574 696f 6e61 7279  eal evolutionary
-000011c0: 2074 696d 652e 2054 6865 2073 796e 6f6e   time. The synon
-000011d0: 796d 6f75 7320 6469 7374 616e 6365 202a  ymous distance *
-000011e0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2c20  K*<sub>S</sub>, 
-000011f0: 7468 6520 6e75 6d62 6572 206f 6620 7379  the number of sy
-00001200: 6e6f 6e79 6d6f 7573 2073 7562 7374 6974  nonymous substit
-00001210: 7574 696f 6e73 2070 6572 2073 796e 6f6e  utions per synon
-00001220: 796d 6f75 7320 7369 7465 2c20 6973 2073  ymous site, is s
-00001230: 7563 6820 6361 6e64 6964 6174 6520 7468  uch candidate th
-00001240: 6174 2073 796e 6f6e 796d 6f75 7320 7375  at synonymous su
-00001250: 6273 7469 7475 7469 6f6e 7320 776f 756c  bstitutions woul
-00001260: 6420 6e6f 7420 696e 6375 7220 7468 6520  d not incur the 
-00001270: 6368 616e 6765 206f 6620 616d 696e 6f20  change of amino 
-00001280: 6163 6964 2061 6e64 2061 7265 2074 6875  acid and are thu
-00001290: 7320 7265 6761 7264 6564 2061 7320 6e65  s regarded as ne
-000012a0: 7574 7261 6c2c 2077 6869 6368 2061 6363  utral, which acc
-000012b0: 6f72 6469 6e67 2074 6f20 7468 6520 6e65  ording to the ne
-000012c0: 7574 7261 6c20 7468 656f 7279 2073 686f  utral theory sho
-000012d0: 756c 6420 6f63 6375 7220 696e 2063 6f6e  uld occur in con
-000012e0: 7374 616e 7420 7261 7465 2e20 4769 7665  stant rate. Give
-000012f0: 6e20 6120 6d6f 6465 6c20 6f66 2067 656e  n a model of gen
-00001300: 6520 6661 6d69 6c79 2074 6861 7420 616c  e family that al
-00001310: 6c6f 7773 2074 6865 2067 656e 6520 746f  lows the gene to
-00001320: 2064 7570 6c69 6361 7465 2061 6e64 2067   duplicate and g
-00001330: 6574 206c 6f73 7420 696e 2061 2066 6978  et lost in a fix
-00001340: 6564 2072 6174 652c 206f 6e65 2063 616e  ed rate, one can
-00001350: 2064 6572 6976 6520 7468 6174 2074 6865   derive that the
-00001360: 2070 726f 6261 6269 6c69 7479 2064 656e   probability den
-00001370: 7369 7479 2066 756e 6374 696f 6e20 6f66  sity function of
-00001380: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00001390: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000013a0: 7574 696f 6e20 6f66 2072 6574 6169 6e65  ution of retaine
-000013b0: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
-000013c0: 7320 6973 2061 2071 7561 7369 2d65 7870  s is a quasi-exp
-000013d0: 6f6e 656e 7469 616c 2066 756e 6374 696f  onential functio
-000013e0: 6e20 7468 6174 206d 6f73 7420 7265 7461  n that most reta
-000013f0: 696e 6564 2067 656e 6520 6475 706c 6963  ined gene duplic
-00001400: 6174 6573 2061 7265 2072 6563 656e 746c  ates are recentl
-00001410: 7920 626f 726e 6564 2077 6974 6820 7e30  y borned with ~0
-00001420: 2061 6765 2077 6869 6c65 2061 7320 7468   age while as th
-00001430: 6520 6167 6520 676f 696e 6720 6f6c 6465  e age going olde
-00001440: 7220 7468 6520 6173 736f 6369 6174 6564  r the associated
-00001450: 206e 756d 6265 7220 6f66 2072 6574 6169   number of retai
-00001460: 6e65 6420 6765 6e65 2064 7570 6c69 6361  ned gene duplica
-00001470: 7465 7320 6465 6361 7920 7175 6173 692d  tes decay quasi-
-00001480: 6578 706f 6e65 6e74 6961 6c6c 792e 2054  exponentially. T
-00001490: 6865 7265 666f 7265 2c20 7468 6520 6f63  herefore, the oc
-000014a0: 6375 7261 6e63 6520 6f66 206c 6172 6765  curance of large
-000014b0: 2d73 6361 6c65 2067 656e 6520 6475 706c  -scale gene dupl
-000014c0: 6963 6174 696f 6e20 6576 656e 7473 2c20  ication events, 
-000014d0: 666f 7220 696e 7374 616e 6520 5747 4473  for instane WGDs
-000014e0: 2c20 7769 7468 2076 6172 6965 6420 7265  , with varied re
-000014f0: 7465 6e74 696f 6e20 7261 7465 2c20 7769  tention rate, wi
-00001500: 6c6c 206c 6561 7665 2061 6e20 6167 6520  ll leave an age 
-00001510: 7065 616b 2066 726f 6d20 7468 6520 6275  peak from the bu
-00001520: 7273 7420 6f66 2067 656e 6520 6475 706c  rst of gene dupl
-00001530: 6963 6174 6573 2069 6e20 6120 7368 6f72  icates in a shor
-00001540: 7420 7469 6d65 2d66 7261 6d65 2075 706f  t time-frame upo
-00001550: 6e20 7468 6520 696e 6974 6961 6c20 6167  n the initial ag
-00001560: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
-00001570: 616e 6420 6361 6e20 6265 2075 6e76 6569  and can be unvei
-00001580: 6c65 6420 6672 6f6d 206d 6978 7475 7265  led from mixture
-00001590: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
-000015a0: 6973 2e20 486f 7765 7665 722c 2057 4744  is. However, WGD
-000015b0: 7320 6964 656e 7469 6669 6564 2066 726f  s identified fro
-000015c0: 6d20 7468 6520 7061 7261 6c6f 676f 7573  m the paralogous
-000015d0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000015e0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-000015f0: 6e73 2063 616e 206f 6e6c 7920 696e 666f  ns can only info
-00001600: 726d 2074 6865 2057 4744 2074 696d 696e  rm the WGD timin
-00001610: 6720 696e 2074 6865 2074 696d 652d 7363  g in the time-sc
-00001620: 616c 6520 6f66 2074 6861 7420 7370 6563  ale of that spec
-00001630: 6966 6963 2073 7065 6369 6573 2c20 7768  ific species, wh
-00001640: 6963 6820 6973 206e 6f74 2063 6f6d 7061  ich is not compa
-00001650: 7261 626c 6520 696e 2074 6865 2070 6879  rable in the phy
-00001660: 6c6f 6765 6e65 7469 6320 636f 6e74 6578  logenetic contex
-00001670: 742e 204f 6e6c 7920 7769 7468 2074 6865  t. Only with the
-00001680: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
-00001690: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-000016a0: 2064 6973 7472 6962 7574 696f 6e73 2c20   distributions, 
-000016b0: 7768 6963 6820 636f 6e76 6572 7420 7468  which convert th
-000016c0: 6520 6573 7469 6d61 7465 6420 626f 6479  e estimated body
-000016d0: 2066 726f 6d20 7061 7261 6c6f 6775 6573   from paralogues
-000016e0: 2074 6f20 6f72 7468 6f6c 6f67 7565 7320   to orthologues 
-000016f0: 616e 6420 696e 666f 726d 2074 6865 2072  and inform the r
-00001700: 656c 6174 6976 6520 7469 6d69 6e67 206f  elative timing o
-00001710: 6620 7370 6563 6961 7469 6f6e 2065 7665  f speciation eve
-00001720: 6e74 732c 2063 616e 2077 6520 6465 6369  nts, can we deci
-00001730: 7068 6572 2074 6865 2070 6879 6c6f 6765  pher the phyloge
-00001740: 6e65 7469 6320 706c 6163 656d 656e 7420  netic placement 
-00001750: 6f66 2057 4744 7320 6166 7465 7220 7072  of WGDs after pr
-00001760: 6f70 6572 2072 6174 6520 636f 7272 6563  oper rate correc
-00001770: 7469 6f6e 2e20 6077 6764 2076 3260 2069  tion. `wgd v2` i
-00001780: 7320 7375 6368 2070 726f 6772 616d 2074  s such program t
-00001790: 6861 7420 6865 6c70 7320 7573 6572 7320  hat helps users 
-000017a0: 636f 6e73 7472 7563 7420 7061 7261 6c6f  construct paralo
-000017b0: 676f 7573 2061 6e64 206f 7274 686f 6c6f  gous and ortholo
-000017c0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-000017d0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-000017e0: 7574 696f 6e73 2061 6e64 2072 6561 6c69  utions and reali
-000017f0: 7a65 2062 6f74 6820 7468 6520 6964 656e  ze both the iden
-00001800: 7469 6669 6361 7469 6f6e 2061 6e64 2070  tification and p
-00001810: 6c61 6365 6d65 6e74 206f 6620 5747 4473  lacement of WGDs
-00001820: 2e0a 0a49 6e20 7468 6520 7072 656d 6973  ...In the premis
-00001830: 6520 6f66 2070 6879 6c6f 6765 6e65 7469  e of phylogeneti
-00001840: 6361 6c6c 7920 6c6f 6361 7465 6420 5747  cally located WG
-00001850: 4473 2c20 7468 6520 6162 736f 6c75 7465  Ds, the absolute
-00001860: 2061 6765 2028 696e 2067 656f 6c6f 6769   age (in geologi
-00001870: 6361 6c20 7469 6d65 2920 6f66 2057 4744  cal time) of WGD
-00001880: 7320 6361 6e20 616c 736f 2062 6520 696e  s can also be in
-00001890: 6665 7272 6564 2066 726f 6d20 7468 6f73  ferred from thos
-000018a0: 6520 5747 442d 7265 7461 696e 6564 2067  e WGD-retained g
-000018b0: 656e 6520 6475 706c 6963 6174 6573 2c20  ene duplicates, 
-000018c0: 616c 7468 6f75 6768 2074 6865 7265 2068  although there h
-000018d0: 6173 2062 6565 6e20 6e6f 2065 6173 7920  as been no easy 
-000018e0: 6f72 2073 7472 6169 6768 7466 6f72 7761  or straightforwa
-000018f0: 7264 2070 6970 656c 696e 6520 666f 7220  rd pipeline for 
-00001900: 7468 6973 206a 6f62 2073 6f20 6661 722e  this job so far.
-00001910: 2049 6e20 6077 6764 2076 3260 2c20 7765   In `wgd v2`, we
-00001920: 2064 6576 656c 6f70 6564 2061 2066 6561   developed a fea
-00001930: 7369 626c 6520 696e 7465 6772 6174 6564  sible integrated
-00001940: 2070 6970 6c65 6c69 6e65 2066 6f72 2061   pipleline for a
-00001950: 6273 6f6c 7574 6520 6461 7469 6e67 206f  bsolute dating o
-00001960: 6620 5747 4473 2e20 5468 6520 7069 7065  f WGDs. The pipe
-00001970: 6c69 6e65 2063 616e 2062 6520 726f 7567  line can be roug
-00001980: 686c 7920 6469 7669 6465 6420 696e 746f  hly divided into
-00001990: 2074 6872 6565 206d 6169 6e20 7374 6570   three main step
-000019a0: 732e 2031 2920 5468 6520 636f 6e73 7472  s. 1) The constr
-000019b0: 7563 7469 6f6e 206f 6620 616e 6368 6f72  uction of anchor
-000019c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000019d0: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
-000019e0: 6420 7468 6520 6465 6c69 6e65 6174 696f  d the delineatio
-000019f0: 6e20 6f66 2063 7265 6469 6162 6c65 202a  n of crediable *
-00001a00: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2072  K*<sub>S</sub> r
-00001a10: 616e 6765 2061 646f 7074 6564 2066 6f72  ange adopted for
-00001a20: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
-00001a30: 7469 6e67 2c20 7573 696e 6720 6077 6764  ting, using `wgd
-00001a40: 2064 6d64 602c 2060 7767 6420 6b73 6460   dmd`, `wgd ksd`
-00001a50: 2c20 6077 6764 2073 796e 6020 616e 6420  , `wgd syn` and 
-00001a60: 6077 6764 2070 6561 6b60 2e20 4e6f 7465  `wgd peak`. Note
-00001a70: 2074 6861 7420 6865 7265 2077 6520 6f6e   that here we on
-00001a80: 6c79 2063 6f6e 7369 6465 7220 6765 6e6f  ly consider geno
-00001a90: 6d65 2061 7373 656d 626c 7920 6265 6361  me assembly beca
-00001aa0: 7573 6520 666f 7220 7472 616e 7363 7269  use for transcri
-00001ab0: 7074 6f6d 6520 6173 7365 6d62 6c79 2069  ptome assembly i
-00001ac0: 7427 7320 696d 706f 7373 6962 6c65 2074  t's impossible t
-00001ad0: 6f20 6469 7374 696e 6775 6973 6820 5747  o distinguish WG
-00001ae0: 442d 6465 7269 7665 6420 6475 706c 6963  D-derived duplic
-00001af0: 6174 6573 2066 726f 6d20 736d 616c 6c2d  ates from small-
-00001b00: 7363 616c 6520 6475 706c 6963 6174 696f  scale duplicatio
-00001b10: 6e2d 6465 7269 7665 6420 6475 706c 6963  n-derived duplic
-00001b20: 6174 6573 2c20 7768 6963 6820 6861 7070  ates, which happ
-00001b30: 656e 6564 2069 6e20 6120 636f 6e74 696e  ened in a contin
-00001b40: 756f 7573 2074 696d 652d 6672 616d 6520  uous time-frame 
-00001b50: 696e 7374 6561 6420 6f66 206f 6e6c 7920  instead of only 
-00001b60: 6120 7365 7061 7261 7465 2073 686f 7274  a separate short
-00001b70: 2074 696d 652d 6672 616d 6520 616e 6420   time-frame and 
-00001b80: 7468 7573 2072 6566 6c65 6374 7320 7468  thus reflects th
-00001b90: 6520 6475 7261 7469 6f6e 206f 6620 7468  e duration of th
-00001ba0: 6174 2062 7261 6e63 6820 7261 7468 6572  at branch rather
-00001bb0: 2074 6861 6e20 7468 6520 7469 6d65 2061   than the time a
-00001bc0: 7420 7768 6963 6820 5747 4420 6f63 6375  t which WGD occu
-00001bd0: 7272 6564 2e20 3229 2054 6865 2066 6f72  rred. 2) The for
-00001be0: 6d75 6c61 7469 6f6e 206f 6620 6120 7374  mulation of a st
-00001bf0: 6172 7469 6e67 2074 7265 6520 7573 6564  arting tree used
-00001c00: 2069 6e20 7468 6520 7068 796c 6f67 656e   in the phylogen
-00001c10: 6574 6963 2064 6174 696e 672c 2063 6f6d  etic dating, com
-00001c20: 706f 7365 6420 6f66 2061 2066 6577 2073  posed of a few s
-00001c30: 7065 6369 6573 2061 6e64 2061 6e6e 6f74  pecies and annot
-00001c40: 6174 6564 2077 6974 6820 666f 7373 696c  ated with fossil
-00001c50: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
-00001c60: 6f72 6d61 7469 6f6e 2e20 5468 6973 2073  ormation. This s
-00001c70: 7465 7020 6973 2065 7373 656e 7469 616c  tep is essential
-00001c80: 2066 6f72 2074 6865 2072 6573 756c 7420   for the result 
-00001c90: 6f66 2061 6273 6f6c 7574 6520 5747 4420  of absolute WGD 
-00001ca0: 6461 7469 6e67 2074 6861 7420 7765 2073  dating that we s
-00001cb0: 7567 6765 7374 2075 7365 7273 206f 6620  uggest users of 
-00001cc0: 7461 6b69 6e67 2067 7265 6174 2063 6175  taking great cau
-00001cd0: 7469 6f6e 2074 6f20 6173 7375 7265 2074  tion to assure t
-00001ce0: 6865 2074 7265 6520 746f 706f 6c6f 6779  he tree topology
-00001cf0: 2061 6e64 2070 726f 7065 7220 626f 756e   and proper boun
-00001d00: 6473 2066 6f72 2066 6f73 7369 6c20 6361  ds for fossil ca
-00001d10: 6c69 6272 6174 696f 6e73 2e20 3329 2054  librations. 3) T
-00001d20: 6865 2063 6f6e 7374 7275 6374 696f 6e20  he construction 
-00001d30: 6f66 206f 7274 686f 6772 6f75 7073 2063  of orthogroups c
-00001d40: 6f6e 7369 7374 696e 6720 6f66 2063 6f6c  onsisting of col
-00001d50: 6c69 6e65 6172 2064 7570 6c69 6361 7465  linear duplicate
-00001d60: 7320 6f66 2074 6865 2066 6f63 616c 2073  s of the focal s
-00001d70: 7065 6369 6573 2061 6e64 2074 6865 6972  pecies and their
-00001d80: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
-00001d90: 2068 6974 7320 2852 4248 7329 2061 6761   hits (RBHs) aga
-00001da0: 696e 7374 206f 7468 6572 2073 7065 6369  inst other speci
-00001db0: 6573 2069 6e20 7468 6520 7374 6172 7469  es in the starti
-00001dc0: 6e67 2074 7265 6520 616e 6420 7468 6520  ng tree and the 
-00001dd0: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
-00001de0: 696e 6720 7573 696e 6720 6120 6d6f 6c65  ing using a mole
-00001df0: 6375 6c61 7220 6461 7469 6e67 2070 726f  cular dating pro
-00001e00: 6772 616d 2066 6f72 2069 6e73 7461 6e63  gram for instanc
-00001e10: 6520 606d 636d 6374 7265 6560 2c20 7669  e `mcmctree`, vi
-00001e20: 6120 7072 6f67 7261 6d20 6077 6764 2064  a program `wgd d
-00001e30: 6d64 6020 616e 6420 6077 6764 2066 6f63  md` and `wgd foc
-00001e40: 7573 602e 2057 6520 7265 636f 6d6d 656e  us`. We recommen
-00001e50: 6420 7468 6520 7573 6167 6520 6f66 2042  d the usage of B
-00001e60: 6179 6573 6961 6e20 6d6f 6c65 6375 6c61  ayesian molecula
-00001e70: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
-00001e80: 2060 6d63 6d63 7472 6565 602c 2077 6869   `mcmctree`, whi
-00001e90: 6368 2070 726f 7669 6465 7320 6120 7661  ch provides a va
-00001ea0: 7269 6574 7920 6f66 2064 6966 6665 7265  riety of differe
-00001eb0: 6e74 2073 7562 7374 6974 7574 696f 6e20  nt substitution 
-00001ec0: 616e 6420 7261 7465 206d 6f64 656c 732e  and rate models.
-00001ed0: 204e 6f6e 6574 6865 6c65 7373 2c20 7765   Nonetheless, we
-00001ee0: 2075 7267 6520 7573 6572 7320 746f 2073   urge users to s
-00001ef0: 6574 2070 7269 6f72 2064 6973 7472 6962  et prior distrib
-00001f00: 7574 696f 6e20 6f66 2064 6966 6665 7265  ution of differe
-00001f10: 6e74 2070 6172 616d 6574 6572 7320 7769  nt parameters wi
-00001f20: 7468 2063 6175 7469 6f6e 2061 6e64 2061  th caution and a
-00001f30: 7373 7572 6520 6164 6571 7561 7465 2073  ssure adequate s
-00001f40: 616d 706c 696e 6720 6f66 2064 6966 6665  ampling of diffe
-00001f50: 7265 6e74 2070 6172 616d 6574 6572 732e  rent parameters.
-00001f60: 0a0a 2323 2049 6e73 7461 6c6c 6174 696f  ..## Installatio
-00001f70: 6e0a 0a54 6865 2065 6173 6965 7374 2077  n..The easiest w
-00001f80: 6179 2074 6f20 696e 7374 616c 6c20 6077  ay to install `w
-00001f90: 6764 2076 3260 2069 7320 7573 696e 6720  gd v2` is using 
-00001fa0: 6050 5950 4960 2e20 4e6f 7465 2074 6861  `PYPI`. Note tha
-00001fb0: 7420 6966 2079 6f75 2077 616e 7420 746f  t if you want to
-00001fc0: 2067 6574 2074 6865 206c 6174 6573 7420   get the latest 
-00001fd0: 7570 6461 7465 2c20 7765 2073 7567 6765  update, we sugge
-00001fe0: 7374 2069 6e73 7461 6c6c 696e 6720 6672  st installing fr
-00001ff0: 6f6d 2074 6865 2073 6f75 7263 652c 2073  om the source, s
-00002000: 696e 6365 2074 6865 2075 7064 6174 6520  ince the update 
-00002010: 6f6e 2060 5059 5049 6020 7769 6c6c 2062  on `PYPI` will b
-00002020: 6520 6465 6c61 7965 6420 636f 6d70 6172  e delayed compar
-00002030: 6564 2074 6f20 6865 7265 206f 6620 736f  ed to here of so
-00002040: 7572 6365 2e0a 0a60 6060 0a70 6970 2069  urce...```.pip i
-00002050: 6e73 7461 6c6c 2077 6764 0a60 6060 0a0a  nstall wgd.```..
-00002060: 546f 2069 6e73 7461 6c6c 2060 7767 6420  To install `wgd 
-00002070: 7632 6020 696e 2061 2076 6972 7475 616c  v2` in a virtual
-00002080: 2065 6e76 6972 6f6e 6d65 6e74 2028 7768   environment (wh
-00002090: 6963 6820 7765 2073 7472 6f6e 676c 7920  ich we strongly 
-000020a0: 7265 636f 6d6d 656e 642c 206e 6f20 6d61  recommend, no ma
-000020b0: 7474 6572 2079 6f75 2069 6e73 7461 6c6c  tter you install
-000020c0: 2066 726f 6d20 736f 7572 6365 2c20 6050   from source, `P
-000020d0: 5950 4960 206f 7220 6062 696f 636f 6e64  YPI` or `biocond
-000020e0: 6160 292c 2074 6865 2066 6f6c 6c6f 7769  a`), the followi
-000020f0: 6e67 2063 6f6d 6d61 6e64 206c 696e 6573  ng command lines
-00002100: 2063 6f75 6c64 2062 6520 7573 6564 2e0a   could be used..
-00002110: 0a60 6060 0a67 6974 2063 6c6f 6e65 2068  .```.git clone h
-00002120: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-00002130: 6d2f 6865 6368 652d 7073 622f 7767 640a  m/heche-psb/wgd.
-00002140: 6364 2077 6764 0a76 6972 7475 616c 656e  cd wgd.virtualen
-00002150: 7620 2d70 3d70 7974 686f 6e33 2045 4e56  v -p=python3 ENV
-00002160: 2028 6f72 2070 7974 686f 6e33 202d 6d20   (or python3 -m 
-00002170: 7665 6e76 2045 4e56 290a 736f 7572 6365  venv ENV).source
-00002180: 2045 4e56 2f62 696e 2f61 6374 6976 6174   ENV/bin/activat
-00002190: 650a 7069 7020 696e 7374 616c 6c20 2d72  e.pip install -r
-000021a0: 2072 6571 7569 7265 6d65 6e74 732e 7478   requirements.tx
-000021b0: 740a 7069 7020 696e 7374 616c 6c20 2e0a  t.pip install ..
-000021c0: 6060 600a 0a57 6865 6e20 6d65 7420 7769  ```..When met wi
-000021d0: 7468 2070 6572 6d69 7373 696f 6e20 7072  th permission pr
-000021e0: 6f62 6c65 6d20 696e 2069 6e73 7461 6c6c  oblem in install
-000021f0: 6174 696f 6e2c 2070 6c65 6173 6520 7472  ation, please tr
-00002200: 7920 7468 6520 666f 6c6c 6f77 696e 6720  y the following 
-00002210: 636f 6d6d 616e 6420 6c69 6e65 2e0a 0a60  command line...`
-00002220: 6060 0a70 6970 2069 6e73 7461 6c6c 202d  ``.pip install -
-00002230: 6520 2e0a 6060 600a 0a49 6620 6d75 6c74  e ..```..If mult
-00002240: 6970 6c79 2076 6572 7369 6f6e 7320 6f66  iply versions of
-00002250: 2060 7767 6460 2077 6572 6520 696e 7374   `wgd` were inst
-00002260: 616c 6c65 6420 696e 2074 6865 2073 7973  alled in the sys
-00002270: 7465 6d2c 2070 6c65 6173 6520 6164 6420  tem, please add 
-00002280: 7468 6520 7269 6768 7420 7061 7468 206f  the right path o
-00002290: 6620 696e 7465 7265 7374 6564 2076 6572  f interested ver
-000022a0: 7369 6f6e 2069 6e74 6f20 7468 6520 656e  sion into the en
-000022b0: 7669 726f 6e6d 656e 7420 7661 7269 6162  vironment variab
-000022c0: 6c65 732c 2066 6f72 2065 7861 6d70 6c65  les, for example
-000022d0: 0a0a 6060 600a 6578 706f 7274 2050 4154  ..```.export PAT
-000022e0: 483d 2224 5041 5448 3a7e 2f2e 6c6f 6361  H="$PATH:~/.loca
-000022f0: 6c2f 6269 6e2f 7767 6422 0a60 6060 0a0a  l/bin/wgd".```..
-00002300: 4e6f 7465 2074 6861 7420 7468 6520 7665  Note that the ve
-00002310: 7273 696f 6e20 6f66 2060 6e75 6d70 7960  rsion of `numpy`
-00002320: 2069 7320 696d 706f 7274 616e 7420 2866   is important (f
-00002330: 6f72 206d 616e 7920 6f74 6865 7220 7061  or many other pa
-00002340: 636b 6167 6573 2061 7265 2074 6865 2073  ckages are the s
-00002350: 616d 6520 6f66 2063 6f75 7273 6529 2c20  ame of course), 
-00002360: 6573 7065 6369 616c 6c79 2066 6f72 2060  especially for `
-00002370: 6661 7374 636c 7573 7465 7260 2070 6163  fastcluster` pac
-00002380: 6b61 6765 2e20 496e 206f 7572 2074 6573  kage. In our tes
-00002390: 742c 2074 6865 2060 6e75 6d70 7960 2031  t, the `numpy` 1
-000023a0: 2e31 392e 3020 776f 726b 7320 6669 6e65  .19.0 works fine
-000023b0: 206f 6e20 6070 7974 686f 6e33 2e36 2f38   on `python3.6/8
-000023c0: 602e 2049 6620 796f 7520 6d65 7420 736f  `. If you met so
-000023d0: 6d65 2065 7272 6f72 7320 6f72 2077 6172  me errors or war
-000023e0: 6e69 6e67 7320 6162 6f75 7420 606e 756d  nings about `num
-000023f0: 7079 602c 206d 6179 6265 2063 6f6e 7369  py`, maybe consi
-00002400: 6465 7269 6e67 2070 7265 2d69 6e73 7461  dering pre-insta
-00002410: 6c6c 2060 6e75 6d70 7960 2061 7320 312e  ll `numpy` as 1.
-00002420: 3139 2e30 206f 7220 6f74 6865 7220 636c  19.0 or other cl
-00002430: 6f73 652d 6279 2076 6572 7369 6f6e 7320  ose-by versions 
-00002440: 6265 666f 7265 2079 6f75 2069 6e73 7461  before you insta
-00002450: 6c6c 2060 7767 6460 2e20 6077 6764 6020  ll `wgd`. `wgd` 
-00002460: 7265 6c69 6573 206f 6e20 6578 7465 726e  relies on extern
-00002470: 616c 2073 6f66 7477 6172 6573 2069 6e63  al softwares inc
-00002480: 6c75 6469 6e67 2060 6469 616d 6f6e 6460  luding `diamond`
-00002490: 2061 6e64 2060 6d63 6c60 2066 6f72 2060   and `mcl` for `
-000024a0: 7767 6420 646d 6460 2c20 6070 616d 6c20  wgd dmd`, `paml 
-000024b0: 7634 2e39 6a60 2c20 606d 6166 6674 6020  v4.9j`, `mafft` 
-000024c0: 2860 6d75 7363 6c65 6020 616e 6420 6070  (`muscle` and `p
-000024d0: 7261 6e6b 6020 6966 2073 6574 292c 2060  rank` if set), `
-000024e0: 6661 7374 7472 6565 6020 286f 7220 6069  fasttree` (or `i
-000024f0: 7174 7265 6560 2069 6620 7365 7429 2066  qtree` if set) f
-00002500: 6f72 2060 7767 6420 6b73 6460 2061 6e64  or `wgd ksd` and
-00002510: 206f 7074 696f 6e61 6c6c 7920 606d 7262   optionally `mrb
-00002520: 6179 6573 6020 666f 7220 7468 6520 7068  ayes` for the ph
-00002530: 796c 6f67 656e 6574 6963 2069 6e66 6572  ylogenetic infer
-00002540: 656e 6365 2066 756e 6374 696f 6e20 696e  ence function in
-00002550: 2060 7767 6420 646d 6460 2061 6e64 2060   `wgd dmd` and `
-00002560: 7767 6420 666f 6375 7360 2028 606d 6166  wgd focus` (`maf
-00002570: 6674 602c 2060 6d75 7363 6c65 6020 616e  ft`, `muscle` an
-00002580: 6420 6070 7261 6e6b 6020 6173 2077 656c  d `prank` as wel
-00002590: 6c20 7768 656e 2074 6865 2061 6e61 6c79  l when the analy
-000025a0: 7369 7320 7265 7175 6972 6573 2073 6571  sis requires seq
-000025b0: 7565 6e63 6520 616c 6967 6e6d 656e 7429  uence alignment)
-000025c0: 2e20 536f 6d65 206f 7468 6572 206f 7074  . Some other opt
-000025d0: 696f 6e61 6c20 736f 6674 7761 7265 7320  ional softwares 
-000025e0: 696e 636c 7564 696e 6720 6070 616d 6c20  including `paml 
-000025f0: 7634 2e39 6a60 2c20 6072 3873 602c 2060  v4.9j`, `r8s`, `
-00002600: 6265 6173 7460 2c20 6065 6767 6e6f 6760  beast`, `eggnog`
-00002610: 2c20 6064 6961 6d6f 6e64 602c 2060 696e  , `diamond`, `in
-00002620: 7465 7270 726f 7363 616e 602c 2060 686d  terproscan`, `hm
-00002630: 6d65 7220 7633 2e31 6232 6020 616e 6420  mer v3.1b2` and 
-00002640: 6061 7374 7261 6c2d 7072 6f60 2028 6068  `astral-pro` (`h
-00002650: 6d6d 6572 2076 332e 3162 3260 2069 7320  mmer v3.1b2` is 
-00002660: 616c 736f 2066 6f72 2074 6865 205b 6f72  also for the [or
-00002670: 7468 6f67 726f 7570 2061 7373 6967 6e6d  thogroup assignm
-00002680: 656e 745d 2868 7474 7073 3a2f 2f77 6764  ent](https://wgd
-00002690: 7632 2e72 6561 6474 6865 646f 6373 2e69  v2.readthedocs.i
-000026a0: 6f2f 656e 2f6c 6174 6573 742f 7573 6167  o/en/latest/usag
-000026b0: 652e 6874 6d6c 236f 7274 686f 6772 6f75  e.html#orthogrou
-000026c0: 702d 6173 7369 676e 6d65 6e74 2920 6675  p-assignment) fu
-000026d0: 6e63 7469 6f6e 2069 6e20 6077 6764 2064  nction in `wgd d
-000026e0: 6d64 6020 616e 6420 6061 7374 7261 6c2d  md` and `astral-
-000026f0: 7072 6f60 2069 7320 616c 736f 2066 6f72  pro` is also for
-00002700: 2074 6865 2066 756e 6374 696f 6e20 602d   the function `-
-00002710: 2d63 6f6c 6c69 6e65 6172 636f 616c 6573  -collinearcoales
-00002720: 6365 6e63 6560 2069 6e20 6077 6764 2064  cence` in `wgd d
-00002730: 6d64 6029 2061 7265 2066 6f72 2074 6865  md`) are for the
-00002740: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
-00002750: 672c 2067 656e 6520 6661 6d69 6c79 2066  g, gene family f
-00002760: 756e 6374 696f 6e20 616e 6e6f 7461 7469  unction annotati
-00002770: 6f6e 206f 7220 7068 796c 6f67 656e 6574  on or phylogenet
-00002780: 6963 2069 6e66 6572 656e 6365 2069 6e20  ic inference in 
-00002790: 6077 6764 2066 6f63 7573 602e 0a0a 2323  `wgd focus`...##
-000027a0: 2050 6172 616d 6574 6572 730a 0a54 6865   Parameters..The
-000027b0: 7265 2061 7265 2037 206d 6169 6e20 7072  re are 7 main pr
-000027c0: 6f67 7261 6d73 2069 6e20 6077 6764 2076  ograms in `wgd v
-000027d0: 3260 3a20 6064 6d64 602c 6066 6f63 7573  2`: `dmd`,`focus
-000027e0: 602c 606b 7364 602c 606d 6978 602c 6070  `,`ksd`,`mix`,`p
-000027f0: 6561 6b60 2c60 7379 6e60 2c60 7669 7a60  eak`,`syn`,`viz`
-00002800: 2e20 4865 7265 6166 7465 7220 7765 2077  . Hereafter we w
-00002810: 696c 6c20 7072 6f76 6964 6520 6120 6465  ill provide a de
-00002820: 7461 696c 6564 2065 6c75 6369 6461 7469  tailed elucidati
-00002830: 6f6e 206f 6e20 6561 6368 206f 6620 7468  on on each of th
-00002840: 6520 7072 6f67 7261 6d20 616e 6420 6974  e program and it
-00002850: 7320 6173 736f 6369 6174 6564 2070 6172  s associated par
-00002860: 616d 6574 6572 732e 2050 6c65 6173 6520  ameters. Please 
-00002870: 7265 6665 7220 746f 2074 6865 205b 5573  refer to the [Us
-00002880: 6167 655d 2823 7573 6167 6529 2066 6f72  age](#usage) for
-00002890: 2074 6865 2073 6365 6e61 7269 6f73 2074   the scenarios t
-000028a0: 6f20 7768 6963 6820 6561 6368 2070 6172  o which each par
-000028b0: 616d 6574 6572 2061 7070 6c69 6573 2e0a  ameter applies..
-000028c0: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
-000028d0: 6420 646d 6460 2063 616e 2072 6561 6c69  d dmd` can reali
-000028e0: 7a65 2074 6865 2064 656c 696e 6561 7469  ze the delineati
-000028f0: 6f6e 206f 6620 7768 6f6c 6520 7061 7261  on of whole para
-00002900: 6e6f 6d65 2c20 5242 4873 2028 5265 6369  nome, RBHs (Reci
-00002910: 7072 6f63 616c 2042 6573 7420 4869 7473  procal Best Hits
-00002920: 292c 204d 5242 4873 2028 4d75 6c74 6970  ), MRBHs (Multip
-00002930: 6c65 2052 6563 6970 726f 6361 6c20 4265  le Reciprocal Be
-00002940: 7374 2048 6974 7329 2c20 6f72 7468 6f67  st Hits), orthog
-00002950: 726f 7570 7320 616e 6420 736f 6d65 206f  roups and some o
-00002960: 7468 6572 206f 7274 686f 6772 6f75 702d  ther orthogroup-
-00002970: 7265 6c61 7465 6420 6675 6e63 7469 6f6e  related function
-00002980: 732c 2069 6e63 6c75 6469 6e67 2063 6972  s, including cir
-00002990: 6375 6d73 6372 6970 7469 6f6e 206f 6620  cumscription of 
-000029a0: 6e65 7374 6564 2073 696e 676c 652d 636f  nested single-co
-000029b0: 7079 206f 7274 686f 6772 6f75 7073 2028  py orthogroups (
-000029c0: 4e53 4f47 7329 2c20 756e 6269 6173 6564  NSOGs), unbiased
-000029d0: 2074 6573 7420 6f66 2073 696e 676c 652d   test of single-
-000029e0: 636f 7079 206f 7274 686f 6772 6f75 7073  copy orthogroups
-000029f0: 2028 534f 4773 2920 6f76 6572 206d 6973   (SOGs) over mis
-00002a00: 7369 6e67 2069 6e70 6172 616c 6f67 732c  sing inparalogs,
-00002a10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
-00002a20: 2042 5553 434f 2d67 7569 6465 6420 7369   BUSCO-guided si
-00002a30: 6e67 6c65 2d63 6f70 7920 6f72 7468 6f67  ngle-copy orthog
-00002a40: 726f 7570 7320 2853 4f47 7329 2c61 6e64  roups (SOGs),and
-00002a50: 2074 6865 2063 6f6c 6c69 6e65 6172 2063   the collinear c
-00002a60: 6f61 6c65 7363 656e 6365 2069 6e66 6572  oalescence infer
-00002a70: 656e 6365 206f 6620 7068 796c 6f67 656e  ence of phylogen
-00002a80: 792e 0a60 6060 0a77 6764 2064 6d64 2073  y..```.wgd dmd s
-00002a90: 6571 7565 6e63 6573 2028 6f70 7469 6f6e  equences (option
-00002aa0: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
-00002ab0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ac0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ad0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ae0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002af0: 2d2d 0a2d 6f2c 202d 2d6f 7574 6469 722c  --.-o, --outdir,
-00002b00: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
-00002b10: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
-00002b20: 6764 5f64 6d64 0a2d 742c 202d 2d74 6d70  gd_dmd.-t, --tmp
-00002b30: 6469 722c 2074 6865 2074 656d 706f 7261  dir, the tempora
-00002b40: 7279 2077 6f72 6b69 6e67 2064 6972 6563  ry working direc
-00002b50: 746f 7279 2c20 6465 6661 756c 7420 4e6f  tory, default No
-00002b60: 6e65 2c20 6966 204e 6f6e 6520 7761 7320  ne, if None was 
-00002b70: 6769 7665 6e2c 2074 6865 2074 6d70 6469  given, the tmpdi
-00002b80: 7220 776f 756c 6420 6265 2061 7373 6967  r would be assig
-00002b90: 6e65 6420 7261 6e64 6f6d 206e 616d 6573  ned random names
-00002ba0: 2069 6e20 6375 7272 656e 7420 6469 7265   in current dire
-00002bb0: 6374 6f72 7920 616e 6420 6175 746f 6d61  ctory and automa
-00002bc0: 7465 6c79 2072 656d 6f76 6564 2061 7420  tely removed at 
-00002bd0: 7468 6520 636f 6d70 6c65 7469 6f6e 206f  the completion o
-00002be0: 6620 7072 6f67 7261 6d2c 2065 6c73 6520  f program, else 
-00002bf0: 7468 6520 746d 7064 6972 2077 6f75 6c64  the tmpdir would
-00002c00: 2062 6520 6b65 7074 0a2d 702c 202d 2d70   be kept.-p, --p
-00002c10: 726f 742c 2066 6c61 6720 6f70 7469 6f6e  rot, flag option
-00002c20: 2c20 7768 6574 6865 7220 7573 696e 6720  , whether using 
-00002c30: 7072 6f74 6569 6e20 6f72 206e 7563 6c65  protein or nucle
-00002c40: 6f74 6964 6520 7365 7175 656e 6365 730a  otide sequences.
-00002c50: 2d63 2c20 2d2d 6373 636f 7265 2c20 7468  -c, --cscore, th
-00002c60: 6520 632d 7363 6f72 6520 746f 2072 6573  e c-score to res
-00002c70: 7472 6963 7420 7468 6520 686f 6d6f 6c6f  trict the homolo
-00002c80: 6720 7369 6d69 6c61 7269 7479 206f 6620  g similarity of 
-00002c90: 4d52 4248 732c 2064 6566 6175 6c74 204e  MRBHs, default N
-00002ca0: 6f6e 652c 2069 6620 4e6f 6e65 2077 6173  one, if None was
-00002cb0: 2067 6976 656e 2c20 7468 6520 632d 7363   given, the c-sc
-00002cc0: 6f72 6520 6675 6e63 696f 6e20 776f 756c  ore funcion woul
-00002cd0: 646e 2774 2062 6520 6163 7469 7661 7465  dn't be activate
-00002ce0: 642c 2065 6c73 6520 6578 7065 6374 696e  d, else expectin
-00002cf0: 6720 6120 6465 6369 6d61 6c20 7769 7468  g a decimal with
-00002d00: 696e 2074 6865 2072 616e 6765 206f 6620  in the range of 
-00002d10: 3020 616e 6420 310a 2d49 2c20 2d2d 696e  0 and 1.-I, --in
-00002d20: 666c 6174 696f 6e2c 2074 6865 2069 6e66  flation, the inf
-00002d30: 6c61 7469 6f6e 2066 6163 746f 7220 666f  lation factor fo
-00002d40: 7220 4d43 4c20 7072 6f67 7261 6d2c 2064  r MCL program, d
-00002d50: 6566 6175 6c74 2032 2e30 2c20 7769 7468  efault 2.0, with
-00002d60: 2068 6967 6865 7220 7661 6c75 6520 6c65   higher value le
-00002d70: 6164 696e 6720 746f 206d 6f72 6520 6275  ading to more bu
-00002d80: 7420 736d 616c 6c65 7220 636c 7573 7465  t smaller cluste
-00002d90: 7273 0a2d 652c 202d 2d65 7661 6c2c 2074  rs.-e, --eval, t
-00002da0: 6865 2065 2d76 616c 7565 2063 7574 2d6f  he e-value cut-o
-00002db0: 6666 2066 6f72 2073 696d 696c 6172 6974  ff for similarit
-00002dc0: 7920 696e 2064 6961 6d6f 6e64 2061 6e64  y in diamond and
-00002dd0: 2f6f 7220 686d 6d65 722c 2064 6566 6175  /or hmmer, defau
-00002de0: 6c74 2031 652d 3130 0a2d 2d74 6f5f 7374  lt 1e-10.--to_st
-00002df0: 6f70 2c20 666c 6167 206f 7074 696f 6e2c  op, flag option,
-00002e00: 2077 6865 7468 6572 2074 6f20 7472 616e   whether to tran
-00002e10: 736c 6174 6520 7468 726f 7567 6820 5354  slate through ST
-00002e20: 4f50 2063 6f64 6f6e 732c 2069 6620 7468  OP codons, if th
-00002e30: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00002e40: 7472 616e 736c 6174 696f 6e20 7769 6c6c  translation will
-00002e50: 2062 6520 7465 726d 696e 6174 6564 2061   be terminated a
-00002e60: 7420 7468 6520 6669 7273 7420 696e 2d66  t the first in-f
-00002e70: 7261 6d65 2073 746f 7020 636f 646f 6e2c  rame stop codon,
-00002e80: 2065 6c73 6520 6120 6675 6c6c 2074 7261   else a full tra
-00002e90: 6e73 6c61 7469 6f6e 2063 6f6e 7469 6e75  nslation continu
-00002ea0: 696e 6720 6f6e 2070 6173 7369 6e67 2061  ing on passing a
-00002eb0: 6e79 2073 746f 7020 636f 646f 6e73 2077  ny stop codons w
-00002ec0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-00002ed0: 640a 2d2d 6364 732c 2066 6c61 6720 6f70  d.--cds, flag op
-00002ee0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00002ef0: 206f 6e6c 7920 7472 616e 736c 6174 6520   only translate 
-00002f00: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
-00002f10: 2074 6861 7420 7374 6172 7473 2077 6974   that starts wit
-00002f20: 6820 6120 7661 6c69 6420 7374 6172 7420  h a valid start 
-00002f30: 636f 646f 6e20 616e 6420 6f6e 6c79 2063  codon and only c
-00002f40: 6f6e 7461 696e 7320 6120 7369 6e67 6c65  ontains a single
-00002f50: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
-00002f60: 6f64 6f6e 2061 7420 7468 6520 656e 6420  odon at the end 
-00002f70: 616e 6420 6d75 7374 2062 6520 6469 7669  and must be divi
-00002f80: 6461 626c 6520 6279 2074 6872 6565 2c20  dable by three, 
-00002f90: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00002fa0: 7365 742c 206f 6e6c 7920 7468 6520 636f  set, only the co
-00002fb0: 6d70 6c65 7465 2043 4453 2077 6f75 6c64  mplete CDS would
-00002fc0: 2062 6520 7472 616e 736c 6174 6564 0a2d   be translated.-
-00002fd0: 662c 202d 2d66 6f63 7573 2c20 7468 6520  f, --focus, the 
-00002fe0: 7370 6563 6965 7320 746f 2062 6520 6d65  species to be me
-00002ff0: 7267 6564 206f 6e20 6c6f 6361 6c20 4d52  rged on local MR
-00003000: 4248 732c 2064 6566 6175 6c74 204e 6f6e  BHs, default Non
-00003010: 652c 2069 6620 4e6f 6e65 2077 6173 2067  e, if None was g
-00003020: 6976 656e 2c20 7468 6520 6c6f 6361 6c20  iven, the local 
-00003030: 4d52 4248 7320 776f 756c 646e 2774 2062  MRBHs wouldn't b
-00003040: 6520 696e 6665 7272 6564 0a2d 6170 2c20  e inferred.-ap, 
-00003050: 2d2d 616e 6368 6f72 706f 696e 7473 2c20  --anchorpoints, 
-00003060: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
-00003070: 7320 6461 7461 2066 696c 6520 6672 6f6d  s data file from
-00003080: 2069 2d61 6468 6f72 6520 666f 7220 636f   i-adhore for co
-00003090: 6e73 7472 7563 7469 6e67 2074 6865 206f  nstructing the o
-000030a0: 7274 686f 6772 6f75 7073 2077 6974 6820  rthogroups with 
-000030b0: 616e 6368 6f72 2070 6169 7273 2c20 6465  anchor pairs, de
-000030c0: 6661 756c 7420 4e6f 6e65 0a2d 736d 2c20  fault None.-sm, 
-000030d0: 2d2d 7365 676d 656e 7473 2c20 7468 6520  --segments, the 
-000030e0: 7365 676d 656e 7473 2064 6174 6166 696c  segments datafil
-000030f0: 6520 7573 6564 2069 6e20 636f 6c6c 696e  e used in collin
-00003100: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-00003110: 616e 616c 7973 6973 2069 6620 696e 6974  analysis if init
-00003120: 6961 7465 642c 2064 6566 6175 6c74 204e  iated, default N
-00003130: 6f6e 650a 2d6c 652c 202d 2d6c 6973 7465  one.-le, --liste
-00003140: 6c65 6d65 6e74 732c 2074 6865 206c 6973  lements, the lis
-00003150: 7473 6567 6d65 6e74 7320 6461 7461 2066  tsegments data f
-00003160: 696c 6520 7573 6564 2069 6e20 636f 6c6c  ile used in coll
-00003170: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
-00003180: 6520 616e 616c 7973 6973 2069 6620 696e  e analysis if in
-00003190: 6974 6961 7465 642c 2064 6566 6175 6c74  itiated, default
-000031a0: 204e 6f6e 650a 2d67 742c 202d 2d67 656e   None.-gt, --gen
-000031b0: 6574 6162 6c65 2c20 7468 6520 6765 6e65  etable, the gene
-000031c0: 2074 6162 6c65 2064 6174 6166 696c 6520   table datafile 
-000031d0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
-000031e0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
-000031f0: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
-00003200: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
-00003210: 650a 2d63 6f63 2c20 2d2d 636f 6c6c 696e  e.-coc, --collin
-00003220: 6561 7263 6f61 6c65 7363 656e 6365 2c20  earcoalescence, 
-00003230: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00003240: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
-00003250: 2074 6865 2063 6f6c 6c69 6e65 6172 2063   the collinear c
-00003260: 6f61 6c65 7363 656e 6365 2061 6e61 6c79  oalescence analy
-00003270: 7369 732c 2069 6620 7468 6520 666c 6167  sis, if the flag
-00003280: 2077 6173 2073 6574 2c20 7468 6520 616e   was set, the an
-00003290: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
-000032a0: 696e 6974 6961 7465 640a 2d6b 662c 202d  initiated.-kf, -
-000032b0: 2d6b 6565 7066 6173 7461 2c20 666c 6167  -keepfasta, flag
-000032c0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000032d0: 2074 6f20 6f75 7470 7574 2074 6865 2073   to output the s
-000032e0: 6571 7565 6e63 6520 696e 666f 726d 6174  equence informat
-000032f0: 696f 6e20 6f66 204d 5242 4873 2c20 6966  ion of MRBHs, if
-00003300: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00003310: 742c 2074 6865 2073 6571 7565 6e63 6573  t, the sequences
-00003320: 206f 6620 4d52 4248 7320 776f 756c 6420   of MRBHs would 
-00003330: 6265 2069 6e20 6f75 7470 7574 0a2d 6b64  be in output.-kd
-00003340: 2c20 2d2d 6b65 6570 6475 706c 6963 6174  , --keepduplicat
-00003350: 6573 2c20 666c 6167 206f 7074 696f 6e2c  es, flag option,
-00003360: 2077 6865 7468 6572 2074 6f20 616c 6c6f   whether to allo
-00003370: 7720 7468 6520 7361 6d65 2067 656e 6520  w the same gene 
-00003380: 746f 206f 6363 7572 2069 6e20 6469 6666  to occur in diff
-00003390: 6572 656e 7420 4d52 4248 7320 286f 6e6c  erent MRBHs (onl
-000033a0: 7920 6d65 616e 696e 6766 756c 2077 6865  y meaningful whe
-000033b0: 6e20 7468 6520 6373 636f 7265 2077 6173  n the cscore was
-000033c0: 2075 7365 6429 2c20 6966 2074 6865 2066   used), if the f
-000033d0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
-000033e0: 2073 616d 6520 6765 6e65 2063 6f75 6c64   same gene could
-000033f0: 2062 6520 6173 7369 676e 6564 2074 6f20   be assigned to 
-00003400: 6469 6666 6572 656e 7420 4d52 4248 730a  different MRBHs.
-00003410: 2d67 6d2c 202d 2d67 6c6f 6261 6c6d 7262  -gm, --globalmrb
-00003420: 682c 2066 6c61 6720 6f70 7469 6f6e 2c20  h, flag option, 
-00003430: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00003440: 6174 6520 676c 6f62 616c 204d 5242 4873  ate global MRBHs
-00003450: 2063 6f6e 7374 7275 6374 696f 6e2c 2069   construction, i
-00003460: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003470: 6574 2c20 7468 6520 2d2d 666f 6375 7320  et, the --focus 
-00003480: 6f70 7469 6f6e 2077 6f75 6c64 2062 6520  option would be 
-00003490: 6967 6e6f 7265 6420 616e 6420 6f6e 6c79  ignored and only
-000034a0: 2067 6c6f 6261 6c20 4d52 4248 7320 776f   global MRBHs wo
-000034b0: 756c 6420 6265 2062 7569 6c74 0a2d 6e2c  uld be built.-n,
-000034c0: 202d 2d6e 7468 7265 6164 732c 2074 6865   --nthreads, the
-000034d0: 206e 756d 6265 7220 6f66 2074 6872 6561   number of threa
-000034e0: 6473 2074 6f20 7573 652c 2064 6566 6175  ds to use, defau
-000034f0: 6c74 2034 0a2d 6f69 2c20 2d2d 6f72 7468  lt 4.-oi, --orth
-00003500: 6f69 6e66 6572 2c20 666c 6167 206f 7074  oinfer, flag opt
-00003510: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00003520: 696e 6974 6961 7465 206f 7274 686f 6772  initiate orthogr
-00003530: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
-00003540: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003550: 6574 2c20 7468 6520 6f72 7468 6f67 726f  et, the orthogro
-00003560: 7570 2069 6e66 6572 6e65 6365 2070 726f  up infernece pro
-00003570: 6772 616d 2077 6f75 6c64 2062 6520 696e  gram would be in
-00003580: 6974 6961 7465 640a 2d6f 6f2c 202d 2d6f  itiated.-oo, --o
-00003590: 6e6c 796f 7274 686f 2c20 666c 6167 206f  nlyortho, flag o
-000035a0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000035b0: 6f20 6f6e 6c79 2063 6f6e 6475 6374 206f  o only conduct o
-000035c0: 7274 686f 6772 6f75 7020 696e 6665 726e  rthogroup infern
-000035d0: 6563 652c 2069 6620 7468 6520 666c 6167  ece, if the flag
-000035e0: 2077 6173 2073 6574 2c20 6f6e 6c79 2074   was set, only t
-000035f0: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
-00003600: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
-00003610: 2077 6f75 6c64 2062 6520 7065 7266 6f72   would be perfor
-00003620: 6d65 6420 7768 696c 6520 7468 6520 6f74  med while the ot
-00003630: 6865 7220 616e 616c 7973 6973 2077 6f75  her analysis wou
-00003640: 6c64 6e27 7420 6265 2069 6e69 7469 6174  ldn't be initiat
-00003650: 6564 0a2d 676e 2c20 2d2d 6765 746e 736f  ed.-gn, --getnso
-00003660: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
-00003670: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00003680: 6174 6520 7468 6520 7365 6172 6368 696e  ate the searchin
-00003690: 6720 666f 7220 6e65 7374 6564 2073 696e  g for nested sin
-000036a0: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
-000036b0: 6d69 6c69 6573 2028 4e53 4f47 7329 2028  milies (NSOGs) (
-000036c0: 6f6e 6c79 206d 6561 6e69 6e67 6675 6c20  only meaningful 
-000036d0: 7768 656e 2074 6865 206f 7274 686f 6772  when the orthogr
-000036e0: 6f75 7020 696e 6665 726e 6563 6520 7069  oup infernece pi
-000036f0: 7065 6c69 6e65 2077 6173 2061 6374 6976  peline was activ
-00003700: 6174 6564 292c 2069 6620 7468 6520 666c  ated), if the fl
-00003710: 6167 2077 6173 2073 6574 2c20 6164 6469  ag was set, addi
-00003720: 7469 6f6e 616c 204e 534f 4773 2061 6e61  tional NSOGs ana
-00003730: 6c79 7369 7320 776f 756c 6420 6265 2070  lysis would be p
-00003740: 6572 666f 726d 6564 2062 6573 6964 6573  erformed besides
-00003750: 2074 6865 2062 6173 6963 206f 7274 686f   the basic ortho
-00003760: 6772 6f75 7020 696e 6665 726e 6563 650a  group infernece.
-00003770: 2d74 7265 652c 202d 2d74 7265 655f 6d65  -tree, --tree_me
-00003780: 7468 6f64 2c20 7768 6963 6820 6765 6e65  thod, which gene
-00003790: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-000037a0: 7072 6f67 7261 6d20 746f 2069 6e76 6f6b  program to invok
-000037b0: 6520 286f 6e6c 7920 6d65 616e 696e 6766  e (only meaningf
-000037c0: 756c 2077 6865 6e20 7468 6520 636f 6c6c  ul when the coll
-000037d0: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
-000037e0: 652c 2067 656e 652d 746f 2d66 616d 696c  e, gene-to-famil
-000037f0: 7920 6173 7369 676e 6d65 6e74 206f 7220  y assignment or 
-00003800: 4e53 4f47 7320 616e 616c 7973 6973 2077  NSOGs analysis w
-00003810: 6572 6520 6163 7469 7661 7465 6429 2c20  ere activated), 
-00003820: 6465 6661 756c 7420 6661 7374 7472 6565  default fasttree
-00003830: 0a2d 7473 2c20 2d2d 7472 6565 7365 742c  .-ts, --treeset,
-00003840: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
-00003850: 7365 7474 696e 6720 666f 7220 6765 6e65  setting for gene
-00003860: 2074 7265 6520 696e 6665 7265 6e63 652c   tree inference,
-00003870: 2064 6566 6175 6c74 204e 6f6e 652c 2074   default None, t
-00003880: 6869 7320 6f70 7469 6f6e 2063 616e 2062  his option can b
-00003890: 6520 7072 6f76 6964 6564 206d 756c 7469  e provided multi
-000038a0: 706c 6520 7469 6d65 730a 2d6d 632c 202d  ple times.-mc, -
-000038b0: 2d6d 736f 6763 7574 2c20 7468 6520 7261  -msogcut, the ra
-000038c0: 7469 6f20 6375 746f 6666 2066 6f72 206d  tio cutoff for m
-000038d0: 6f73 746c 7920 7369 6e67 6c65 2d63 6f70  ostly single-cop
-000038e0: 7920 6661 6d69 6c79 2028 6d65 616e 696e  y family (meanin
-000038f0: 6766 756c 2077 6865 6e20 6163 7469 7661  gful when activa
-00003900: 7469 6e67 2074 6865 206f 7274 686f 6772  ting the orthogr
-00003910: 6f75 7020 696e 6665 726e 6563 6520 7069  oup infernece pi
-00003920: 7065 6c69 6e65 2920 616e 6420 7370 6563  peline) and spec
-00003930: 6965 7320 7265 7072 6573 656e 7461 7469  ies representati
-00003940: 6f6e 2069 6e20 636f 6c6c 696e 6561 7220  on in collinear 
-00003950: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
-00003960: 7973 6973 2c20 6465 6661 756c 7420 302e  ysis, default 0.
-00003970: 382e 0a2d 6761 2c20 2d2d 6765 6e65 6173  8..-ga, --geneas
-00003980: 7369 676e 2c20 666c 6167 206f 7074 696f  sign, flag optio
-00003990: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-000039a0: 6974 6961 7465 2074 6865 2067 656e 652d  itiate the gene-
-000039b0: 746f 2d66 616d 696c 7920 6173 7369 676e  to-family assign
-000039c0: 6d65 6e74 2061 6e61 6c79 7369 732c 2069  ment analysis, i
-000039d0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-000039e0: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-000039f0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00003a00: 7465 640a 2d73 612c 202d 2d73 6571 3261  ted.-sa, --seq2a
-00003a10: 7373 6967 6e2c 2074 6865 2071 7565 7269  ssign, the queri
-00003a20: 6564 2073 6571 7565 6e63 6573 2064 6174  ed sequences dat
-00003a30: 6120 6669 6c65 2069 6e20 6765 6e65 2d74  a file in gene-t
-00003a40: 6f2d 6661 6d69 6c79 2061 7373 6967 6e6d  o-family assignm
-00003a50: 656e 7420 616e 616c 7973 6973 2c20 6465  ent analysis, de
-00003a60: 6661 756c 7420 4e6f 6e65 2c20 7468 6973  fault None, this
-00003a70: 206f 7074 696f 6e20 6361 6e20 6265 2070   option can be p
-00003a80: 726f 7669 6465 6420 6d75 6c74 6970 6c65  rovided multiple
-00003a90: 2074 696d 6573 0a2d 6661 2c20 2d2d 6661   times.-fa, --fa
-00003aa0: 6d32 6173 7369 676e 2c20 7468 6520 7175  m2assign, the qu
-00003ab0: 6572 6965 6420 6661 6d69 6c69 7920 6461  eried familiy da
-00003ac0: 7461 2066 696c 6520 696e 2067 656e 652d  ta file in gene-
-00003ad0: 746f 2d66 616d 696c 7920 6173 7369 676e  to-family assign
-00003ae0: 6d65 6e74 2061 6e61 6c79 7369 732c 2064  ment analysis, d
-00003af0: 6566 6175 6c74 204e 6f6e 650a 2d63 632c  efault None.-cc,
-00003b00: 202d 2d63 6f6e 6361 742c 2066 6c61 6720   --concat, flag 
-00003b10: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00003b20: 746f 2069 6e69 7469 6174 6520 7468 6520  to initiate the 
-00003b30: 636f 6e63 6174 656e 6174 696f 6e20 7069  concatenation pi
-00003b40: 7065 6c69 6e65 2066 6f72 206f 7274 686f  peline for ortho
-00003b50: 6772 6f75 7020 696e 6665 726e 6563 652c  group infernece,
-00003b60: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003b70: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
-00003b80: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
-00003b90: 6961 7465 640a 2d74 652c 202d 2d74 6573  iated.-te, --tes
-00003ba0: 7473 6f67 2c20 666c 6167 206f 7074 696f  tsog, flag optio
-00003bb0: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00003bc0: 6974 6961 7465 2074 6865 2075 6e62 6961  itiate the unbia
-00003bd0: 7365 6420 7465 7374 206f 6620 7369 6e67  sed test of sing
-00003be0: 6c65 2d63 6f70 7920 6765 6e65 2066 616d  le-copy gene fam
-00003bf0: 696c 6965 732c 2069 6620 7468 6520 666c  ilies, if the fl
-00003c00: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-00003c10: 616e 616c 7973 6973 2077 6f75 6c64 2062  analysis would b
-00003c20: 6520 696e 6974 6961 7465 640a 2d62 732c  e initiated.-bs,
-00003c30: 202d 2d62 696e 732c 2074 6865 206e 756d   --bins, the num
-00003c40: 6265 7220 6f66 2062 696e 7320 6469 7669  ber of bins divi
-00003c50: 6465 6420 696e 2074 6865 2067 656e 6520  ded in the gene 
-00003c60: 6c65 6e67 7468 206e 6f72 6d61 6c69 7a61  length normaliza
-00003c70: 7469 6f6e 2c20 6465 6661 756c 7420 3130  tion, default 10
-00003c80: 300a 2d6e 702c 202d 2d6e 6f72 6d61 6c69  0.-np, --normali
-00003c90: 7a65 6470 6572 6365 6e74 2c20 7468 6520  zedpercent, the 
-00003ca0: 7065 7263 656e 7461 6765 206f 6620 7570  percentage of up
-00003cb0: 7065 7220 6869 7473 2075 7365 6420 666f  per hits used fo
-00003cc0: 7220 6765 6e65 206c 656e 6774 6820 6e6f  r gene length no
-00003cd0: 726d 616c 697a 6174 696f 6e2c 2064 6566  rmalization, def
-00003ce0: 6175 6c74 2035 0a2d 6e6e 2c20 2d2d 6e6f  ault 5.-nn, --no
-00003cf0: 6e6f 726d 616c 697a 6174 696f 6e2c 2066  normalization, f
-00003d00: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00003d10: 6865 7220 746f 2063 616c 6c20 6f66 6620  her to call off 
-00003d20: 7468 6520 6e6f 726d 616c 697a 6174 696f  the normalizatio
-00003d30: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
-00003d40: 6173 2073 6574 2c20 6e6f 206e 6f72 6d61  as set, no norma
-00003d50: 6c69 7a61 7469 6f6e 2077 6f75 6c64 2062  lization would b
-00003d60: 6520 636f 6e64 7563 7465 640a 2d62 736f  e conducted.-bso
-00003d70: 672c 202d 2d62 7573 636f 736f 672c 2066  g, --buscosog, f
-00003d80: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00003d90: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00003da0: 7468 6520 6275 7363 6f2d 6775 6964 6564  the busco-guided
-00003db0: 2073 696e 676c 652d 636f 7079 2067 656e   single-copy gen
-00003dc0: 6520 6661 6d69 6c79 2061 6e61 6c79 7369  e family analysi
-00003dd0: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00003de0: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
-00003df0: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
-00003e00: 6974 6961 7465 640a 2d62 686d 6d2c 202d  itiated.-bhmm, -
-00003e10: 2d62 7573 636f 686d 6d2c 2074 6865 2048  -buscohmm, the H
-00003e20: 4d4d 2070 726f 6669 6c65 2064 6174 6166  MM profile dataf
-00003e30: 696c 6520 696e 2074 6865 2062 7573 636f  ile in the busco
-00003e40: 2d67 7569 6465 6420 7369 6e67 6c65 2d63  -guided single-c
-00003e50: 6f70 7920 6765 6e65 2066 616d 696c 7920  opy gene family 
-00003e60: 616e 616c 7973 6973 2c20 6465 6661 756c  analysis, defaul
-00003e70: 7420 4e6f 6e65 0a2d 6263 7466 2c20 2d2d  t None.-bctf, --
-00003e80: 6275 7363 6f63 7574 6f66 662c 2074 6865  buscocutoff, the
-00003e90: 2048 4d4d 2073 636f 7265 2063 7574 6f66   HMM score cutof
-00003ea0: 6620 6461 7461 6669 6c65 2069 6e20 7468  f datafile in th
-00003eb0: 6520 6275 7363 6f2d 6775 6964 6564 2073  e busco-guided s
-00003ec0: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
-00003ed0: 6661 6d69 6c79 2061 6e61 6c79 7369 732c  family analysis,
-00003ee0: 2064 6566 6175 6c74 204e 6f6e 650a 2d6f   default None.-o
-00003ef0: 6620 2c2d 2d6f 6766 6f72 6d61 742c 2066  f ,--ogformat, f
-00003f00: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00003f10: 6865 7220 746f 2061 6464 2069 6e64 6578  her to add index
-00003f20: 2074 6f20 7468 6520 5242 4820 6661 6d69   to the RBH fami
-00003f30: 6c69 6573 0a60 6060 0a0a 5468 6520 7072  lies.```..The pr
-00003f40: 6f67 7261 6d20 6077 6764 2066 6f63 7573  ogram `wgd focus
-00003f50: 6020 6361 6e20 7265 616c 697a 6520 7468  ` can realize th
-00003f60: 6520 636f 6e63 6174 656e 6174 696f 6e2d  e concatenation-
-00003f70: 6261 7365 6420 616e 6420 636f 616c 6573  based and coales
-00003f80: 6365 6e63 652d 6261 7365 6420 7068 796c  cence-based phyl
-00003f90: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
-00003fa0: 6365 2061 6e64 2070 6879 6c6f 6765 6e65  ce and phylogene
-00003fb0: 7469 6320 6461 7469 6e67 206f 6620 5747  tic dating of WG
-00003fc0: 4473 2065 7463 2e0a 6060 600a 7767 6420  Ds etc..```.wgd 
-00003fd0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
-00003fe0: 6571 7565 6e63 6573 2028 6f70 7469 6f6e  equences (option
-00003ff0: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
-00004000: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004010: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004020: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004030: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004040: 2d2d 0a2d 6f2c 202d 2d6f 7574 6469 722c  --.-o, --outdir,
-00004050: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
-00004060: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
-00004070: 6764 5f66 6f63 7573 0a2d 742c 202d 2d74  gd_focus.-t, --t
-00004080: 6d70 6469 722c 2074 6865 2074 656d 706f  mpdir, the tempo
-00004090: 7261 7279 2077 6f72 6b69 6e67 2064 6972  rary working dir
-000040a0: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
-000040b0: 4e6f 6e65 2c20 6966 204e 6f6e 6520 7761  None, if None wa
-000040c0: 7320 6769 7665 6e2c 2074 6865 2074 6d70  s given, the tmp
-000040d0: 6469 7220 7769 6c6c 2062 6520 6173 7369  dir will be assi
-000040e0: 676e 6564 2072 616e 646f 6d20 6e61 6d65  gned random name
-000040f0: 7320 696e 2063 7572 7265 6e74 2064 6972  s in current dir
-00004100: 6563 746f 7279 2061 6e64 2061 7574 6f6d  ectory and autom
-00004110: 6174 656c 7920 7265 6d6f 7665 6420 6174  ately removed at
-00004120: 2074 6865 2063 6f6d 706c 6574 696f 6e20   the completion 
-00004130: 6f66 2070 726f 6772 616d 2c20 656c 7365  of program, else
-00004140: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
-00004150: 6420 6265 206b 6570 740a 2d6e 2c20 2d2d  d be kept.-n, --
-00004160: 6e74 6872 6561 6473 2c20 7468 6520 6e75  nthreads, the nu
-00004170: 6d62 6572 206f 6620 7468 7265 6164 7320  mber of threads 
-00004180: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
-00004190: 340a 2d2d 746f 5f73 746f 702c 2066 6c61  4.--to_stop, fla
-000041a0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000041b0: 7220 746f 2074 7261 6e73 6c61 7465 2074  r to translate t
-000041c0: 6872 6f75 6768 2053 544f 5020 636f 646f  hrough STOP codo
-000041d0: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
-000041e0: 7761 7320 7365 742c 2074 7261 6e73 6c61  was set, transla
-000041f0: 7469 6f6e 2077 696c 6c20 6265 2074 6572  tion will be ter
-00004200: 6d69 6e61 7465 6420 6174 2074 6865 2066  minated at the f
-00004210: 6972 7374 2069 6e2d 6672 616d 6520 7374  irst in-frame st
-00004220: 6f70 2063 6f64 6f6e 2c20 656c 7365 2061  op codon, else a
-00004230: 2066 756c 6c20 7472 616e 736c 6174 696f   full translatio
-00004240: 6e20 636f 6e74 696e 7569 6e67 206f 6e20  n continuing on 
-00004250: 7061 7374 2061 6e79 2073 746f 7020 636f  past any stop co
-00004260: 646f 6e73 2077 6f75 6c64 2062 6520 696e  dons would be in
-00004270: 6974 6961 7465 640a 2d2d 6364 732c 2066  itiated.--cds, f
-00004280: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004290: 6865 7220 746f 206f 6e6c 7920 7472 616e  her to only tran
-000042a0: 736c 6174 6520 7468 6520 636f 6d70 6c65  slate the comple
-000042b0: 7465 2043 4453 2074 6861 7420 7374 6172  te CDS that star
-000042c0: 7473 2077 6974 6820 6120 7661 6c69 6420  ts with a valid 
-000042d0: 7374 6172 7420 636f 646f 6e20 616e 6420  start codon and 
-000042e0: 6f6e 6c79 2063 6f6e 7461 696e 7320 6120  only contains a 
-000042f0: 7369 6e67 6c65 2069 6e2d 6672 616d 6520  single in-frame 
-00004300: 7374 6f70 2063 6f64 6f6e 2061 7420 7468  stop codon at th
-00004310: 6520 656e 6420 616e 6420 6d75 7374 2062  e end and must b
-00004320: 6520 6469 7669 6461 626c 6520 6279 2074  e dividable by t
-00004330: 6872 6565 2c20 6966 2074 6865 2066 6c61  hree, if the fla
-00004340: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
-00004350: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
-00004360: 2077 6f75 6c64 2062 6520 7472 616e 736c   would be transl
-00004370: 6174 6564 0a2d 2d73 7472 6970 5f67 6170  ated.--strip_gap
-00004380: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
-00004390: 7768 6574 6865 7220 746f 2064 726f 7020  whether to drop 
-000043a0: 616c 6c20 6761 7073 2069 6e20 6d75 6c74  all gaps in mult
-000043b0: 6970 6c65 2073 6571 7565 6e63 6520 616c  iple sequence al
-000043c0: 6967 6e6d 656e 742c 2069 6620 7468 6520  ignment, if the 
-000043d0: 666c 6167 2077 6173 2073 6574 2c20 616c  flag was set, al
-000043e0: 6c20 6761 7073 2077 6f75 6c64 2062 6520  l gaps would be 
-000043f0: 6472 6f70 7065 640a 2d61 2c20 2d2d 616c  dropped.-a, --al
-00004400: 6967 6e65 722c 2077 6869 6368 2061 6c69  igner, which ali
-00004410: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
-00004420: 6f20 7573 652c 2064 6566 6175 6c74 206d  o use, default m
-00004430: 6166 6674 0a2d 7472 6565 2c20 2d2d 7472  afft.-tree, --tr
-00004440: 6565 5f6d 6574 686f 642c 2077 6869 6368  ee_method, which
-00004450: 2067 656e 6520 7472 6565 2069 6e66 6572   gene tree infer
-00004460: 656e 6365 2070 726f 6772 616d 2074 6f20  ence program to 
-00004470: 696e 766f 6b65 2c20 6465 6661 756c 7420  invoke, default 
-00004480: 6661 7374 7472 6565 0a2d 7473 2c20 2d2d  fasttree.-ts, --
-00004490: 7472 6565 7365 742c 2074 6865 2070 6172  treeset, the par
-000044a0: 616d 6574 6572 7320 7365 7474 696e 6720  ameters setting 
-000044b0: 666f 7220 6765 6e65 2074 7265 6520 696e  for gene tree in
-000044c0: 6665 7265 6e63 652c 2064 6566 6175 6c74  ference, default
-000044d0: 204e 6f6e 652c 2074 6869 7320 6f70 7469   None, this opti
-000044e0: 6f6e 2063 616e 2062 6520 7072 6f76 6964  on can be provid
-000044f0: 6564 206d 756c 7469 706c 6520 7469 6d65  ed multiple time
-00004500: 730a 2d2d 636f 6e63 6174 656e 6174 696f  s.--concatenatio
-00004510: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
-00004520: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00004530: 6174 6520 7468 6520 636f 6e63 6174 656e  ate the concaten
-00004540: 6174 696f 6e2d 6261 7365 6420 7370 6563  ation-based spec
-00004550: 6965 7320 7472 6565 2069 6e66 6572 656e  ies tree inferen
-00004560: 6365 2c20 6966 2074 6865 2066 6c61 6720  ce, if the flag 
-00004570: 7761 7320 7365 742c 2063 6f6e 6361 7465  was set, concate
-00004580: 6e61 7469 6f6e 2d62 6173 6564 2073 7065  nation-based spe
-00004590: 6369 6573 2074 7265 6520 776f 756c 6420  cies tree would 
-000045a0: 6265 2069 6e66 6572 6564 0a2d 2d63 6f61  be infered.--coa
-000045b0: 6c65 7363 656e 6365 2c20 666c 6167 206f  lescence, flag o
-000045c0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000045d0: 6f20 696e 6974 6961 7465 2074 6865 2063  o initiate the c
-000045e0: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
-000045f0: 2073 7065 6369 6573 2074 7265 6520 696e   species tree in
-00004600: 6665 7265 6e63 652c 2069 6620 7468 6520  ference, if the 
-00004610: 666c 6167 2077 6173 2073 6574 2c20 636f  flag was set, co
-00004620: 616c 6573 6365 6e63 652d 6261 7365 6420  alescence-based 
-00004630: 7370 6563 6965 7320 7472 6565 2077 6f75  species tree wou
-00004640: 6c64 2062 6520 696e 6665 7265 640a 2d73  ld be infered.-s
-00004650: 702c 202d 2d73 7065 6369 6573 7472 6565  p, --speciestree
-00004660: 2c20 7370 6563 6965 7320 7472 6565 2064  , species tree d
-00004670: 6174 6166 696c 6520 666f 7220 6461 7469  atafile for dati
-00004680: 6e67 2c20 6465 6661 756c 7420 4e6f 6e65  ng, default None
-00004690: 0a2d 642c 202d 2d64 6174 696e 672c 2077  .-d, --dating, w
-000046a0: 6869 6368 206d 6f6c 6563 756c 6172 2064  hich molecular d
-000046b0: 6174 696e 6720 7072 6f67 7261 6d20 746f  ating program to
-000046c0: 2075 7365 2c20 6465 6661 756c 7420 6e6f   use, default no
-000046d0: 6e65 0a2d 6473 2c20 2d2d 6461 7469 6e67  ne.-ds, --dating
-000046e0: 7365 742c 2074 6865 2070 6172 616d 6574  set, the paramet
-000046f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
-00004700: 6461 7469 6e67 2070 726f 6772 616d 2c20  dating program, 
-00004710: 6465 6661 756c 7420 4e6f 6e65 2c20 7468  default None, th
-00004720: 6973 206f 7074 696f 6e20 6361 6e20 6265  is option can be
-00004730: 2070 726f 7669 6465 6420 6d75 6c74 6970   provided multip
-00004740: 6c65 2074 696d 6573 0a2d 6e73 2c20 2d2d  le times.-ns, --
-00004750: 6e73 6974 6573 2c20 7468 6520 6e73 6974  nsites, the nsit
-00004760: 6573 2069 6e66 6f72 6d61 7469 6f6e 2066  es information f
-00004770: 6f72 2072 3873 2064 6174 696e 672c 2064  or r8s dating, d
-00004780: 6566 6175 6c74 204e 6f6e 650a 2d6f 742c  efault None.-ot,
-00004790: 202d 2d6f 7574 6772 6f75 702c 2074 6865   --outgroup, the
-000047a0: 206f 7574 6772 6f75 7020 696e 666f 726d   outgroup inform
-000047b0: 6174 696f 6e20 666f 7220 7238 7320 6461  ation for r8s da
-000047c0: 7469 6e67 2c20 6465 6661 756c 7420 4e6f  ting, default No
-000047d0: 6e65 0a2d 7074 2c20 2d2d 7061 7274 6974  ne.-pt, --partit
-000047e0: 696f 6e2c 2066 6c61 6720 6f70 7469 6f6e  ion, flag option
-000047f0: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
-00004800: 7469 6174 6520 7061 7274 6974 696f 6e20  tiate partition 
-00004810: 6461 7469 6e67 2061 6e61 6c79 7369 7320  dating analysis 
-00004820: 666f 7220 636f 646f 6e2c 2069 6620 7468  for codon, if th
-00004830: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00004840: 616e 2061 6464 6974 696f 6e61 6c20 7061  an additional pa
-00004850: 7274 6974 696f 6e20 6461 7469 6e67 2061  rtition dating a
-00004860: 6e61 6c79 7369 7320 776f 756c 6420 6265  nalysis would be
-00004870: 2069 6e69 7469 6174 6564 0a2d 616d 2c20   initiated.-am, 
-00004880: 2d2d 6161 6d6f 6465 6c2c 2077 6869 6368  --aamodel, which
-00004890: 2070 726f 7465 696e 206d 6f64 656c 2074   protein model t
-000048a0: 6f20 6265 2075 7365 6420 696e 206d 636d  o be used in mcm
-000048b0: 6374 7265 652c 2064 6566 6175 6c74 2070  ctree, default p
-000048c0: 6f69 7373 6f6e 0a2d 6b73 2c20 666c 6167  oisson.-ks, flag
-000048d0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000048e0: 2074 6f20 696e 6974 6961 7465 204b 7320   to initiate Ks 
-000048f0: 6361 6c63 756c 6174 696f 6e20 666f 7220  calculation for 
-00004900: 686f 6d6f 6c6f 6775 6573 2069 6e20 7468  homologues in th
-00004910: 6520 7072 6f76 6964 6564 206f 7274 686f  e provided ortho
-00004920: 6c6f 676f 7573 2067 656e 6520 6661 6d69  logous gene fami
-00004930: 6c79 0a2d 2d61 6e6e 6f74 6174 696f 6e2c  ly.--annotation,
-00004940: 2077 6869 6368 2061 6e6e 6f74 6174 696f   which annotatio
-00004950: 6e20 7072 6f67 7261 6d20 746f 2075 7365  n program to use
-00004960: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00004970: 2d70 6169 7277 6973 652c 2066 6c61 6720  -pairwise, flag 
-00004980: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00004990: 746f 2069 6e69 7469 6174 6520 7061 6972  to initiate pair
-000049a0: 7769 7365 204b 7320 6573 7469 6d61 7469  wise Ks estimati
-000049b0: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-000049c0: 7761 7320 7365 742c 2070 6169 7277 6973  was set, pairwis
-000049d0: 6520 4b73 2076 616c 7565 7320 776f 756c  e Ks values woul
-000049e0: 6420 6265 2065 7374 696d 6174 6564 0a2d  d be estimated.-
-000049f0: 6564 2c20 2d2d 6567 676e 6f67 6461 7461  ed, --eggnogdata
-00004a00: 2c20 7468 6520 6567 676e 6f67 2061 6e6e  , the eggnog ann
-00004a10: 6f74 6174 696f 6e20 6461 7461 6669 6c65  otation datafile
-00004a20: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00004a30: 2d70 6661 6d2c 2077 6869 6368 206f 7074  -pfam, which opt
-00004a40: 696f 6e20 746f 2075 7365 2066 6f72 2070  ion to use for p
-00004a50: 6661 6d20 616e 6e6f 7461 7469 6f6e 2c20  fam annotation, 
-00004a60: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d64  default None.--d
-00004a70: 6d6e 622c 2074 6865 2064 6961 6d6f 6e64  mnb, the diamond
-00004a80: 2064 6174 6162 6173 6520 666f 7220 616e   database for an
-00004a90: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
-00004aa0: 7420 4e6f 6e65 0a2d 2d68 6d6d 2c20 7468  t None.--hmm, th
-00004ab0: 6520 484d 4d20 7072 6f66 696c 6520 666f  e HMM profile fo
-00004ac0: 7220 616e 6e6f 7461 7469 6f6e 2c20 6465  r annotation, de
-00004ad0: 6661 756c 7420 4e6f 6e65 0a2d 2d65 7661  fault None.--eva
-00004ae0: 6c75 652c 2074 6865 2065 2d76 616c 7565  lue, the e-value
-00004af0: 2063 7574 2d6f 6666 2066 6f72 2061 6e6e   cut-off for ann
-00004b00: 6f74 6174 696f 6e2c 2064 6566 6175 6c74  otation, default
-00004b10: 2031 652d 3130 0a2d 2d65 7865 7061 7468   1e-10.--exepath
-00004b20: 2c20 7468 6520 7061 7468 2074 6f20 7468  , the path to th
-00004b30: 6520 696e 7465 7270 726f 7363 616e 2065  e interproscan e
-00004b40: 7865 6375 7461 626c 652c 2064 6566 6175  xecutable, defau
-00004b50: 6c74 204e 6f6e 650a 2d66 2c20 2d2d 666f  lt None.-f, --fo
-00004b60: 7373 696c 2c20 7468 6520 666f 7373 696c  ssil, the fossil
-00004b70: 2063 616c 6962 7261 7469 6f6e 2069 6e66   calibration inf
-00004b80: 6f72 6d61 7469 6f6e 2069 6e20 4265 6173  ormation in Beas
-00004b90: 742c 2064 6566 6175 6c74 2028 2763 6c61  t, default ('cla
-00004ba0: 6465 313b 636c 6164 6532 272c 2027 7461  de1;clade2', 'ta
-00004bb0: 7861 312c 7461 7861 323b 7461 7861 332c  xa1,taxa2;taxa3,
-00004bc0: 7461 7861 3427 2c20 2734 3b35 272c 2027  taxa4', '4;5', '
-00004bd0: 302e 353b 302e 3627 2c20 2734 3030 3b35  0.5;0.6', '400;5
-00004be0: 3030 2729 0a2d 7268 2c20 2d2d 726f 6f74  00').-rh, --root
-00004bf0: 6865 6967 6874 2c20 7468 6520 726f 6f74  height, the root
-00004c00: 2068 6569 6768 7420 6361 6c69 6272 6174   height calibrat
-00004c10: 696f 6e20 696e 666f 2069 6e20 4265 6173  ion info in Beas
-00004c20: 742c 2064 6566 6175 6c74 2028 342c 302e  t, default (4,0.
-00004c30: 352c 3430 3029 0a2d 6373 2c20 2d2d 6368  5,400).-cs, --ch
-00004c40: 6169 6e73 6574 2c20 7468 6520 7061 7261  ainset, the para
-00004c50: 6d65 7465 7273 206f 6620 4d43 4d43 2063  meters of MCMC c
-00004c60: 6861 696e 2069 6e20 4265 6173 742c 2064  hain in Beast, d
-00004c70: 6566 6175 6c74 2028 3130 3030 302c 3130  efault (10000,10
-00004c80: 3029 0a2d 2d62 6561 7374 6c67 6a61 722c  0).--beastlgjar,
-00004c90: 2074 6865 2070 6174 6820 746f 2062 6561   the path to bea
-00004ca0: 7374 4c47 2e6a 6172 2c20 6465 6661 756c  stLG.jar, defaul
-00004cb0: 7420 4e6f 6e65 0a2d 2d62 6561 676c 652c  t None.--beagle,
-00004cc0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004cd0: 6574 6865 7220 746f 2075 7365 2062 6561  ether to use bea
-00004ce0: 676c 6520 696e 2042 6561 7374 2c20 6966  gle in Beast, if
-00004cf0: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00004d00: 742c 2062 6561 676c 6520 776f 756c 6420  t, beagle would 
-00004d10: 6265 2075 7365 640a 2d2d 7072 6f74 636f  be used.--protco
-00004d20: 6364 6174 696e 672c 2066 6c61 6720 6f70  cdating, flag op
-00004d30: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00004d40: 206f 6e6c 7920 696e 6974 6961 7465 2074   only initiate t
-00004d50: 6865 2070 726f 7465 696e 2d63 6f6e 6361  he protein-conca
-00004d60: 7465 6e61 7469 6f6e 2d62 6173 6564 2064  tenation-based d
-00004d70: 6174 696e 6720 616e 616c 7973 6973 2c20  ating analysis, 
-00004d80: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00004d90: 7365 742c 2074 6865 2061 6e61 6c79 7369  set, the analysi
-00004da0: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
-00004db0: 6174 6564 0a2d 2d70 726f 7464 6174 696e  ated.--protdatin
-00004dc0: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
-00004dd0: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
-00004de0: 696e 6974 6961 7465 2074 6865 2070 726f  initiate the pro
-00004df0: 7465 696e 2d62 6173 6564 2064 6174 696e  tein-based datin
-00004e00: 6720 616e 616c 7973 6973 2c20 6966 2074  g analysis, if t
-00004e10: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00004e20: 2074 6865 2061 6e61 6c79 7369 7320 776f   the analysis wo
-00004e30: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
-00004e40: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
-00004e50: 6d20 6077 6764 206b 7364 6020 6361 6e20  m `wgd ksd` can 
-00004e60: 7265 616c 697a 6520 7468 6520 636f 6e73  realize the cons
-00004e70: 7472 7563 7469 6f6e 206f 6620 2a4b 2a3c  truction of *K*<
-00004e80: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-00004e90: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
-00004ea0: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-00004eb0: 2e0a 6060 600a 7767 6420 6b73 6420 6661  ..```.wgd ksd fa
-00004ec0: 6d69 6c69 6573 2073 6571 7565 6e63 6573  milies sequences
-00004ed0: 2028 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d   (option).------
-00004ee0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004ef0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004f00: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004f10: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004f20: 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c 202d  ----------.-o, -
-00004f30: 2d6f 7574 6469 722c 2074 6865 206f 7574  -outdir, the out
-00004f40: 7075 7420 6469 7265 6374 6f72 792c 2064  put directory, d
-00004f50: 6566 6175 6c74 2077 6764 5f6b 7364 0a2d  efault wgd_ksd.-
-00004f60: 742c 202d 2d74 6d70 6469 722c 2074 6865  t, --tmpdir, the
-00004f70: 2074 656d 706f 7261 7279 2077 6f72 6b69   temporary worki
-00004f80: 6e67 2064 6972 6563 746f 7279 2c20 6465  ng directory, de
-00004f90: 6661 756c 7420 4e6f 6e65 2c20 6966 204e  fault None, if N
-00004fa0: 6f6e 6520 7761 7320 6769 7665 6e2c 2074  one was given, t
-00004fb0: 6865 2074 6d70 6469 7220 7769 6c6c 2062  he tmpdir will b
-00004fc0: 6520 6173 7369 676e 6564 2072 616e 646f  e assigned rando
-00004fd0: 6d20 6e61 6d65 7320 696e 2063 7572 7265  m names in curre
-00004fe0: 6e74 2064 6972 6563 746f 7279 2061 6e64  nt directory and
-00004ff0: 2061 7574 6f6d 6174 656c 7920 7265 6d6f   automately remo
-00005000: 7665 6420 6174 2074 6865 2063 6f6d 706c  ved at the compl
-00005010: 6574 696f 6e20 6f66 2070 726f 6772 616d  etion of program
-00005020: 2c20 656c 7365 2074 6865 2074 6d70 6469  , else the tmpdi
-00005030: 7220 776f 756c 6420 6265 206b 6570 740a  r would be kept.
-00005040: 2d6e 2c20 2d2d 6e74 6872 6561 6473 2c20  -n, --nthreads, 
-00005050: 7468 6520 6e75 6d62 6572 206f 6620 7468  the number of th
-00005060: 7265 6164 7320 746f 2075 7365 2c20 6465  reads to use, de
-00005070: 6661 756c 7420 340a 2d2d 746f 5f73 746f  fault 4.--to_sto
-00005080: 702c 2066 6c61 6720 6f70 7469 6f6e 2c20  p, flag option, 
-00005090: 7768 6574 6865 7220 746f 2074 7261 6e73  whether to trans
-000050a0: 6c61 7465 2074 6872 6f75 6768 2053 544f  late through STO
-000050b0: 5020 636f 646f 6e73 2c20 6966 2074 6865  P codons, if the
-000050c0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-000050d0: 7261 6e73 6c61 7469 6f6e 2077 696c 6c20  ranslation will 
-000050e0: 6265 2074 6572 6d69 6e61 7465 6420 6174  be terminated at
-000050f0: 2074 6865 2066 6972 7374 2069 6e2d 6672   the first in-fr
-00005100: 616d 6520 7374 6f70 2063 6f64 6f6e 2c20  ame stop codon, 
-00005110: 656c 7365 2061 2066 756c 6c20 7472 616e  else a full tran
-00005120: 736c 6174 696f 6e20 636f 6e74 696e 7569  slation continui
-00005130: 6e67 206f 6e20 7061 7374 2061 6e79 2073  ng on past any s
-00005140: 746f 7020 636f 646f 6e73 2077 6f75 6c64  top codons would
-00005150: 2062 6520 696e 6974 6961 7465 640a 2d2d   be initiated.--
-00005160: 6364 732c 2066 6c61 6720 6f70 7469 6f6e  cds, flag option
-00005170: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
-00005180: 7920 7472 616e 736c 6174 6520 7468 6520  y translate the 
-00005190: 636f 6d70 6c65 7465 2043 4453 2074 6861  complete CDS tha
-000051a0: 7420 7374 6172 7473 2077 6974 6820 6120  t starts with a 
-000051b0: 7661 6c69 6420 7374 6172 7420 636f 646f  valid start codo
-000051c0: 6e20 616e 6420 6f6e 6c79 2063 6f6e 7461  n and only conta
-000051d0: 696e 7320 6120 7369 6e67 6c65 2069 6e2d  ins a single in-
-000051e0: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
-000051f0: 2061 7420 7468 6520 656e 6420 616e 6420   at the end and 
-00005200: 6d75 7374 2062 6520 6469 7669 6461 626c  must be dividabl
-00005210: 6520 6279 2074 6872 6565 2c20 6966 2074  e by three, if t
-00005220: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00005230: 206f 6e6c 7920 7468 6520 636f 6d70 6c65   only the comple
-00005240: 7465 2043 4453 2077 6f75 6c64 2062 6520  te CDS would be 
-00005250: 7472 616e 736c 6174 6564 0a2d 2d70 6169  translated.--pai
-00005260: 7277 6973 652c 2066 6c61 6720 6f70 7469  rwise, flag opti
-00005270: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
-00005280: 6e69 7469 6174 6520 7061 6972 7769 7365  nitiate pairwise
-00005290: 204b 7320 6573 7469 6d61 7469 6f6e 2c20   Ks estimation, 
-000052a0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-000052b0: 7365 742c 2070 6169 7277 6973 6520 4b73  set, pairwise Ks
-000052c0: 2076 616c 7565 7320 776f 756c 6420 6265   values would be
-000052d0: 2065 7374 696d 6174 6564 0a2d 2d73 7472   estimated.--str
-000052e0: 6970 5f67 6170 732c 2066 6c61 6720 6f70  ip_gaps, flag op
-000052f0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00005300: 2064 726f 7020 616c 6c20 6761 7073 2069   drop all gaps i
-00005310: 6e20 6d75 6c74 6970 6c65 2073 6571 7565  n multiple seque
-00005320: 6e63 6520 616c 6967 6e6d 656e 742c 2069  nce alignment, i
-00005330: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00005340: 6574 2c20 616c 6c20 6761 7073 2077 6f75  et, all gaps wou
-00005350: 6c64 2062 6520 6472 6f70 7065 640a 2d61  ld be dropped.-a
-00005360: 2c20 2d2d 616c 6967 6e65 722c 2077 6869  , --aligner, whi
-00005370: 6368 2061 6c69 676e 6d65 6e74 2070 726f  ch alignment pro
-00005380: 6772 616d 2074 6f20 7573 652c 2064 6566  gram to use, def
-00005390: 6175 6c74 206d 6166 6674 200a 2d74 7265  ault mafft .-tre
-000053a0: 652c 202d 2d74 7265 655f 6d65 7468 6f64  e, --tree_method
-000053b0: 2c20 7768 6963 6820 6765 6e65 2074 7265  , which gene tre
-000053c0: 6520 696e 6665 7265 6e63 6520 7072 6f67  e inference prog
-000053d0: 7261 6d20 746f 2069 6e76 6f6b 652c 2064  ram to invoke, d
-000053e0: 6566 6175 6c74 2066 6173 7474 7265 650a  efault fasttree.
-000053f0: 2d2d 7472 6565 5f6f 7074 696f 6e73 2c20  --tree_options, 
-00005400: 6f70 7469 6f6e 7320 696e 2074 7265 6520  options in tree 
-00005410: 696e 6665 7265 6e63 6520 6173 2061 2063  inference as a c
-00005420: 6f6d 6d61 2073 6570 6172 6174 6564 2073  omma separated s
-00005430: 7472 696e 672c 2064 6566 6175 6c74 204e  tring, default N
-00005440: 6f6e 650a 2d2d 6e6f 6465 5f61 7665 7261  one.--node_avera
-00005450: 6765 2c20 666c 6167 206f 7074 696f 6e2c  ge, flag option,
-00005460: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00005470: 6961 7465 206e 6f64 652d 6176 6572 6167  iate node-averag
-00005480: 6520 7761 7920 6f66 2064 652d 7265 6475  e way of de-redu
-00005490: 6e64 616e 6379 2069 6e73 7465 6164 206f  ndancy instead o
-000054a0: 6620 6e6f 6465 2d77 6569 6768 7465 642c  f node-weighted,
-000054b0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-000054c0: 2073 6574 2c20 7468 6520 6e6f 6465 2d61   set, the node-a
-000054d0: 7665 7261 6769 6e67 2064 652d 7265 6475  veraging de-redu
-000054e0: 6e64 616e 6379 2077 6f75 6c64 2062 6520  ndancy would be 
-000054f0: 696e 6974 6961 7465 640a 2d73 722c 202d  initiated.-sr, -
-00005500: 2d73 7061 6972 2c20 7468 6520 7370 6563  -spair, the spec
-00005510: 6965 7320 7061 6972 2074 6f20 6265 2070  ies pair to be p
-00005520: 6c6f 7474 6564 2c20 6465 6661 756c 7420  lotted, default 
-00005530: 4e6f 6e65 2c20 7468 6973 206f 7074 696f  None, this optio
-00005540: 6e20 6361 6e20 6265 2070 726f 7669 6465  n can be provide
-00005550: 6420 6d75 6c74 6970 6c65 2074 696d 6573  d multiple times
-00005560: 0a2d 7370 2c20 2d2d 7370 6563 6965 7374  .-sp, --speciest
-00005570: 7265 652c 2074 6865 2073 7065 6369 6573  ree, the species
-00005580: 2074 7265 6520 746f 2070 6572 666f 726d   tree to perform
-00005590: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-000055a0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
-000055b0: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
-000055c0: 6e2c 2074 6865 2072 6174 6520 636f 7272  n, the rate corr
-000055d0: 6563 7469 6f6e 2061 6e61 6c79 7369 7320  ection analysis 
-000055e0: 776f 756c 6420 6265 2063 616c 6c65 6420  would be called 
-000055f0: 6f66 660a 2d72 772c 202d 2d72 6577 6569  off.-rw, --rewei
-00005600: 6768 742c 2066 6c61 6720 6f70 7469 6f6e  ght, flag option
-00005610: 2c20 7768 6574 6865 7220 746f 2072 6563  , whether to rec
-00005620: 616c 6375 6c61 7465 2074 6865 2077 6569  alculate the wei
-00005630: 6768 7420 7065 7220 7370 6563 6965 7320  ght per species 
-00005640: 7061 6972 2c20 6966 2074 6865 2066 6c61  pair, if the fla
-00005650: 6720 7761 7320 7365 742c 2074 6865 2077  g was set, the w
-00005660: 6569 6768 7420 776f 756c 6420 6265 2072  eight would be r
-00005670: 6563 616c 6375 6c61 7465 640a 2d6f 722c  ecalculated.-or,
-00005680: 202d 2d6f 6e6c 7972 6f6f 746f 7574 2c20   --onlyrootout, 
-00005690: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-000056a0: 7468 6572 2074 6f20 6f6e 6c79 2063 6f6e  ther to only con
-000056b0: 6475 6374 2072 6174 6520 636f 7272 6563  duct rate correc
-000056c0: 7469 6f6e 2075 7369 6e67 2074 6865 206f  tion using the o
-000056d0: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
-000056e0: 6173 206f 7574 6772 6f75 702c 2069 6620  as outgroup, if 
-000056f0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00005700: 2c20 6f6e 6c79 2074 6865 206f 7574 6772  , only the outgr
-00005710: 6f75 7020 6174 2072 6f6f 7420 776f 756c  oup at root woul
-00005720: 6420 6265 2075 7365 6420 6173 206f 7574  d be used as out
-00005730: 6772 6f75 700a 2d65 706b 2c20 2d2d 6578  group.-epk, --ex
-00005740: 7472 6170 6172 616e 6f6d 656b 732c 2065  traparanomeks, e
-00005750: 7874 7261 2070 6172 616e 6f6d 6520 4b73  xtra paranome Ks
-00005760: 2064 6174 6120 746f 2070 6c6f 7420 696e   data to plot in
-00005770: 2074 6865 206d 6978 6564 204b 7320 6469   the mixed Ks di
-00005780: 7374 7269 6275 7469 6f6e 2c20 6465 6661  stribution, defa
-00005790: 756c 7420 4e6f 6e65 0a2d 6170 2c20 2d2d  ult None.-ap, --
-000057a0: 616e 6368 6f72 706f 696e 7473 2c20 616e  anchorpoints, an
-000057b0: 6368 6f72 706f 696e 7473 2e74 7874 2066  chorpoints.txt f
-000057c0: 696c 6520 746f 2070 6c6f 7420 616e 6368  ile to plot anch
-000057d0: 6f72 204b 7320 696e 2074 6865 206d 6978  or Ks in the mix
-000057e0: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
-000057f0: 6f6e 2c20 6465 6661 756c 7420 4e6f 6e65  on, default None
-00005800: 0a2d 706b 2c20 2d2d 706c 6f74 6b64 652c  .-pk, --plotkde,
-00005810: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005820: 6574 6865 7220 746f 2070 6c6f 7420 6b64  ether to plot kd
-00005830: 6520 6375 7276 6520 6f66 206f 7274 686f  e curve of ortho
-00005840: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
-00005850: 6275 7469 6f6e 206f 7665 7220 6869 7374  bution over hist
-00005860: 6f67 7261 6d20 696e 2074 6865 206d 6978  ogram in the mix
-00005870: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
-00005880: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-00005890: 7761 7320 7365 742c 2074 6865 206b 6465  was set, the kde
-000058a0: 2063 7572 7665 2077 6f75 6c64 2062 6520   curve would be 
-000058b0: 706c 6f74 7465 640a 2d70 6167 2c20 2d2d  plotted.-pag, --
-000058c0: 706c 6f74 6170 676d 6d2c 2066 6c61 6720  plotapgmm, flag 
-000058d0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000058e0: 746f 2070 6572 666f 726d 2061 6e64 2070  to perform and p
-000058f0: 6c6f 7420 6d69 7874 7572 6520 6d6f 6465  lot mixture mode
-00005900: 6c69 6e67 206f 6620 616e 6368 6f72 204b  ling of anchor K
-00005910: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
-00005920: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
-00005930: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
-00005940: 7365 742c 2074 6865 206d 6978 7475 7265  set, the mixture
-00005950: 206d 6f64 656c 696e 6720 6f66 2061 6e63   modeling of anc
-00005960: 686f 7220 4b73 2077 6f75 6c64 2062 6520  hor Ks would be 
-00005970: 706c 6f74 7465 640a 2d70 656d 2c20 2d2d  plotted.-pem, --
-00005980: 706c 6f74 656c 6d6d 2c20 666c 6167 206f  plotelmm, flag o
-00005990: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000059a0: 6f20 7065 7266 6f72 6d20 616e 6420 706c  o perform and pl
-000059b0: 6f74 2065 6c6d 6d20 6d69 7874 7572 6520  ot elmm mixture 
-000059c0: 6d6f 6465 6c69 6e67 206f 6620 7061 7261  modeling of para
-000059d0: 6e6f 6d65 204b 7320 696e 2074 6865 206d  nome Ks in the m
-000059e0: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
-000059f0: 7469 6f6e 2c20 6966 2074 6865 2066 6c61  tion, if the fla
-00005a00: 6720 7761 7320 7365 742c 2074 6865 2065  g was set, the e
-00005a10: 6c6d 6d20 6d69 7874 7572 6520 6d6f 6465  lmm mixture mode
-00005a20: 6c69 6e67 206f 6620 7061 7261 6e6f 6d65  ling of paranome
-00005a30: 204b 7320 776f 756c 6420 6265 2070 6c6f   Ks would be plo
-00005a40: 7474 6564 0a2d 632c 202d 2d63 6f6d 706f  tted.-c, --compo
-00005a50: 6e65 6e74 732c 2074 6865 2072 616e 6765  nents, the range
-00005a60: 206f 6620 7468 6520 6e75 6d62 6572 206f   of the number o
-00005a70: 6620 636f 6d70 6f6e 656e 7473 2074 6f20  f components to 
-00005a80: 6669 7420 696e 2061 6e63 686f 7220 4b73  fit in anchor Ks
-00005a90: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00005aa0: 672c 2064 6566 6175 6c74 2028 312c 3429  g, default (1,4)
-00005ab0: 0a2d 786c 2c20 2d2d 786c 696d 2c20 7468  .-xl, --xlim, th
-00005ac0: 6520 7820 6178 6973 206c 696d 6974 206f  e x axis limit o
-00005ad0: 6620 4b73 2064 6973 7472 6962 7574 696f  f Ks distributio
-00005ae0: 6e0a 2d79 6c2c 202d 2d79 6c69 6d2c 2074  n.-yl, --ylim, t
-00005af0: 6865 2079 2061 7869 7320 6c69 6d69 7420  he y axis limit 
-00005b00: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
-00005b10: 6f6e 0a2d 6164 6f2c 202d 2d61 646a 7573  on.-ado, --adjus
-00005b20: 746f 7274 686f 2c20 666c 6167 206f 7074  tortho, flag opt
-00005b30: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00005b40: 6164 6a75 7374 2074 6865 2068 6973 746f  adjust the histo
-00005b50: 6772 616d 2068 6569 6768 7420 6f66 206f  gram height of o
-00005b60: 7274 686f 6c6f 676f 7573 204b 7320 6173  rthologous Ks as
-00005b70: 2074 6f20 6d61 7463 6820 7468 6520 6865   to match the he
-00005b80: 6967 6874 206f 6620 7061 7261 6c6f 676f  ight of paralogo
-00005b90: 7573 204b 732c 2069 6620 7468 6520 666c  us Ks, if the fl
-00005ba0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-00005bb0: 6164 6a75 7374 6d65 6e74 2077 6f75 6c64  adjustment would
-00005bc0: 2062 6520 636f 6e64 7563 7465 640a 2d61   be conducted.-a
-00005bd0: 6466 2c20 2d2d 6164 6a75 7374 6661 6374  df, --adjustfact
-00005be0: 6f72 2c20 7468 6520 6164 6a75 7374 6d65  or, the adjustme
-00005bf0: 6e74 2066 6163 746f 7220 6f66 206f 7274  nt factor of ort
-00005c00: 686f 6c6f 676f 7573 204b 732c 2064 6566  hologous Ks, def
-00005c10: 6175 6c74 2030 2e35 0a2d 6f61 2c20 2d2d  ault 0.5.-oa, --
-00005c20: 6f6b 616c 7068 612c 2074 6865 206f 7061  okalpha, the opa
-00005c30: 6369 7479 206f 6620 6f72 7468 6f6c 6f67  city of ortholog
-00005c40: 6f75 7320 4b73 2064 6973 7472 6962 7574  ous Ks distribut
-00005c50: 696f 6e20 696e 206d 6978 6564 2070 6c6f  ion in mixed plo
-00005c60: 742c 2064 6566 6175 6c74 2030 2e35 0a2d  t, default 0.5.-
-00005c70: 6661 2c20 2d2d 666f 6375 7332 616c 6c2c  fa, --focus2all,
-00005c80: 2073 6574 2066 6f63 616c 2073 7065 6369   set focal speci
-00005c90: 6573 2061 6e64 206c 6574 2073 7065 6369  es and let speci
-00005ca0: 6573 2070 6169 7220 746f 2062 6520 6265  es pair to be be
-00005cb0: 7477 6565 6e20 666f 6361 6c20 616e 6420  tween focal and 
-00005cc0: 616c 6c20 7468 6520 7265 6d61 696e 696e  all the remainin
-00005cd0: 6720 7370 6563 6965 732c 2064 6566 6175  g species, defau
-00005ce0: 6c74 204e 6f6e 650a 2d6b 732c 202d 2d6b  lt None.-ks, --k
-00005cf0: 7374 7265 652c 2066 6c61 6720 6f70 7469  stree, flag opti
-00005d00: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
-00005d10: 6e66 6572 204b 7320 7472 6565 2c20 6966  nfer Ks tree, if
-00005d20: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00005d30: 742c 2074 6865 204b 7320 7472 6565 2069  t, the Ks tree i
-00005d40: 6e66 6572 656e 6365 2061 6e61 6c79 7369  nference analysi
-00005d50: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
-00005d60: 6174 6564 0a2d 6f63 6b2c 202d 2d6f 6e6c  ated.-ock, --onl
-00005d70: 7963 6f6e 6361 746b 7374 7265 652c 2066  yconcatkstree, f
-00005d80: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005d90: 6865 7220 746f 206f 6e6c 7920 696e 6665  her to only infe
-00005da0: 7220 4b73 2074 7265 6520 756e 6465 7220  r Ks tree under 
-00005db0: 636f 6e63 6174 656e 6174 6564 2061 6c69  concatenated ali
-00005dc0: 676e 6d65 6e74 2c20 6966 2074 6865 2066  gnment, if the f
-00005dd0: 6c61 6720 7761 7320 7365 742c 206f 6e6c  lag was set, onl
-00005de0: 7920 7468 6520 4b73 2074 7265 6520 756e  y the Ks tree un
-00005df0: 6465 7220 636f 6e63 6174 656e 6174 6564  der concatenated
-00005e00: 2061 6c69 676e 6d65 6e74 2077 6f75 6c64   alignment would
-00005e10: 2062 6520 6361 6c63 756c 6174 6564 0a2d   be calculated.-
-00005e20: 6373 2c20 2d2d 636c 6173 7369 632c 2066  cs, --classic, f
-00005e30: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005e40: 6865 7220 746f 2064 7261 7720 6d69 7865  her to draw mixe
-00005e50: 6420 4b73 2070 6c6f 7420 696e 2061 2063  d Ks plot in a c
-00005e60: 6c61 7373 6963 206d 616e 6e65 7220 7768  lassic manner wh
-00005e70: 6572 6520 7468 6520 6675 6c6c 206f 7274  ere the full ort
-00005e80: 686f 6c6f 676f 7573 204b 7320 6469 7374  hologous Ks dist
-00005e90: 7269 6275 7469 6f6e 2069 7320 6472 6177  ribution is draw
-00005ea0: 6564 2c20 6966 2074 6865 2066 6c61 6720  ed, if the flag 
-00005eb0: 7761 7320 7365 742c 2074 6865 2063 6c61  was set, the cla
-00005ec0: 7373 6963 206d 6978 6564 204b 7320 706c  ssic mixed Ks pl
-00005ed0: 6f74 2077 6f75 6c64 2062 6520 6472 6177  ot would be draw
-00005ee0: 6e0a 2d74 612c 202d 2d74 6f70 6172 726f  n.-ta, --toparro
-00005ef0: 772c 2066 6c61 6720 6f70 7469 6f6e 2c20  w, flag option, 
-00005f00: 7768 6574 6865 7220 746f 2061 646a 7573  whether to adjus
-00005f10: 7420 7468 6520 6172 726f 7720 746f 2062  t the arrow to b
-00005f20: 6520 6174 2074 6865 2074 6f70 206f 6620  e at the top of 
-00005f30: 7468 6520 706c 6f74 2c20 696e 7374 6561  the plot, instea
-00005f40: 6420 6f66 2062 6569 6e67 2063 6f6f 7264  d of being coord
-00005f50: 696e 6174 6564 2061 7320 7468 6520 4b44  inated as the KD
-00005f60: 4520 6f66 2074 6865 206f 7274 686f 6c6f  E of the ortholo
-00005f70: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
-00005f80: 7469 6f6e 2c20 6966 2074 6865 2066 6c61  tion, if the fla
-00005f90: 6720 7761 7320 7365 742c 2074 6865 2061  g was set, the a
-00005fa0: 7272 6f77 2077 6f75 6c64 2062 6520 7365  rrow would be se
-00005fb0: 7420 6174 2074 6865 2074 6f70 0a2d 6273  t at the top.-bs
-00005fc0: 2c20 2d2d 626f 6f74 7374 7261 702c 2074  , --bootstrap, t
-00005fd0: 6865 206e 756d 6265 7220 6f66 2062 6f6f  he number of boo
-00005fe0: 7473 7472 6170 2072 6570 6c69 6361 7465  tstrap replicate
-00005ff0: 7320 6f66 206f 7274 686f 6c6f 6720 4b73  s of ortholog Ks
-00006000: 2064 6973 7472 6962 7574 696f 6e20 696e   distribution in
-00006010: 206d 6978 6564 2070 6c6f 740a 6060 600a   mixed plot.```.
-00006020: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
-00006030: 6420 6d69 7860 2063 616e 2072 6561 6c69  d mix` can reali
-00006040: 7a65 2074 6865 206d 6978 7475 7265 206d  ze the mixture m
-00006050: 6f64 656c 2063 6c75 7374 6572 696e 6720  odel clustering 
-00006060: 616e 616c 7973 6973 206f 6620 2a4b 2a3c  analysis of *K*<
-00006070: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-00006080: 6469 7374 7269 6275 7469 6f6e 2e0a 6060  distribution..``
-00006090: 600a 7767 6420 6d69 7820 6b73 5f64 6174  `.wgd mix ks_dat
-000060a0: 6166 696c 6520 286f 7074 696f 6e29 0a2d  afile (option).-
-000060b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000060c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000060d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000060e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000060f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a  ---------------.
-00006100: 2d66 2c20 2d2d 6669 6c74 6572 732c 2074  -f, --filters, t
-00006110: 6865 2063 7574 6f66 6620 616c 6967 6e6d  he cutoff alignm
-00006120: 656e 7420 6c65 6e67 7468 2c20 6465 6661  ent length, defa
-00006130: 756c 7420 3330 300a 2d72 2c20 2d2d 6b73  ult 300.-r, --ks
-00006140: 5f72 616e 6765 2c20 7468 6520 4b73 2072  _range, the Ks r
-00006150: 616e 6765 2074 6f20 6265 2063 6f6e 7369  ange to be consi
-00006160: 6465 7265 642c 2064 6566 6175 6c74 2028  dered, default (
-00006170: 302c 2035 290a 2d62 2c20 2d2d 6269 6e73  0, 5).-b, --bins
-00006180: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
-00006190: 6269 6e73 2069 6e20 4b73 2064 6973 7472  bins in Ks distr
-000061a0: 6962 7574 696f 6e2c 2064 6566 6175 6c74  ibution, default
-000061b0: 2035 300a 2d6f 2c20 2d2d 6f75 7464 6972   50.-o, --outdir
-000061c0: 2c20 7468 6520 6f75 7470 7574 2064 6972  , the output dir
-000061d0: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
-000061e0: 7767 645f 6d69 780a 2d2d 6d65 7468 6f64  wgd_mix.--method
-000061f0: 2c20 7768 6963 6820 6d69 7874 7572 6520  , which mixture 
-00006200: 6d6f 6465 6c20 746f 2075 7365 2c20 6465  model to use, de
-00006210: 6661 756c 7420 676d 6d0a 2d6e 2c20 2d2d  fault gmm.-n, --
-00006220: 636f 6d70 6f6e 656e 7473 2c20 7468 6520  components, the 
-00006230: 7261 6e67 6520 6f66 2074 6865 206e 756d  range of the num
-00006240: 6265 7220 6f66 2063 6f6d 706f 6e65 6e74  ber of component
-00006250: 7320 746f 2066 6974 2c20 6465 6661 756c  s to fit, defaul
-00006260: 7420 2831 2c20 3429 0a2d 672c 202d 2d67  t (1, 4).-g, --g
-00006270: 616d 6d61 2c20 7468 6520 6761 6d6d 6120  amma, the gamma 
-00006280: 7061 7261 6d65 7465 7220 666f 7220 6267  parameter for bg
-00006290: 6d6d 206d 6f64 656c 732c 2064 6566 6175  mm models, defau
-000062a0: 6c74 2030 2e30 3031 0a2d 6e69 2c20 2d2d  lt 0.001.-ni, --
-000062b0: 6e5f 696e 6974 2c20 7468 6520 6e75 6d62  n_init, the numb
-000062c0: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
-000062d0: 6974 6961 6c69 7a61 7469 6f6e 732c 2064  itializations, d
-000062e0: 6566 6175 6c74 2032 3030 0a2d 6d69 2c20  efault 200.-mi, 
-000062f0: 2d2d 6d61 785f 6974 6572 2c20 7468 6520  --max_iter, the 
-00006300: 6d61 7869 6d75 6d20 6e75 6d62 6572 206f  maximum number o
-00006310: 6620 6974 6572 6174 696f 6e73 2c20 6465  f iterations, de
-00006320: 6661 756c 7420 3230 300a 6060 600a 0a54  fault 200.```..T
-00006330: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-00006340: 7065 616b 6020 6361 6e20 7265 616c 697a  peak` can realiz
-00006350: 6520 7468 6520 7365 6172 6368 206f 6620  e the search of 
-00006360: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
-00006370: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
-00006380: 7573 6564 2069 6e20 5747 4420 6461 7469  used in WGD dati
-00006390: 6e67 2e0a 6060 600a 7767 6420 7065 616b  ng..```.wgd peak
-000063a0: 206b 735f 6461 7461 6669 6c65 2028 6f70   ks_datafile (op
-000063b0: 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d  tion).----------
-000063c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000063d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000063e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000063f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006400: 2d2d 2d2d 2d2d 0a2d 6170 2c20 2d2d 616e  ------.-ap, --an
-00006410: 6368 6f72 706f 696e 7473 2c20 7468 6520  chorpoints, the 
-00006420: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
-00006430: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-00006440: 4e6f 6e65 0a2d 736d 2c20 2d2d 7365 676d  None.-sm, --segm
-00006450: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
-00006460: 7473 2064 6174 6166 696c 652c 2064 6566  ts datafile, def
-00006470: 6175 6c74 204e 6f6e 650a 2d6c 652c 202d  ault None.-le, -
-00006480: 2d6c 6973 7465 6c65 6d65 6e74 732c 2074  -listelements, t
-00006490: 6865 206c 6973 7473 6567 6d65 6e74 7320  he listsegments 
-000064a0: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
-000064b0: 7420 4e6f 6e65 200a 2d6d 702c 202d 2d6d  t None .-mp, --m
-000064c0: 756c 7469 706c 6963 6f6e 7061 6972 732c  ultipliconpairs,
-000064d0: 2074 6865 206d 756c 7469 706c 6963 6f6e   the multiplicon
-000064e0: 7061 6972 7320 6461 7461 6669 6c65 2c20  pairs datafile, 
-000064f0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
-00006500: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
-00006510: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
-00006520: 2064 6566 6175 6c74 2077 6764 5f70 6561   default wgd_pea
-00006530: 6b0a 2d61 662c 202d 2d61 6c69 676e 6669  k.-af, --alignfi
-00006540: 6c74 6572 2c20 6375 746f 6666 2066 6f72  lter, cutoff for
-00006550: 2061 6c69 676e 6d65 6e74 2069 6465 6e74   alignment ident
-00006560: 6974 792c 206c 656e 6774 6820 616e 6420  ity, length and 
-00006570: 636f 7665 7261 6765 2c20 6465 6661 756c  coverage, defaul
-00006580: 7420 302e 302c 2030 2c20 302e 300a 2d72  t 0.0, 0, 0.0.-r
-00006590: 2c20 2d2d 6b73 7261 6e67 652c 2072 616e  , --ksrange, ran
-000065a0: 6765 206f 6620 4b73 2074 6f20 6265 2061  ge of Ks to be a
-000065b0: 6e61 6c79 7a65 642c 2064 6566 6175 6c74  nalyzed, default
-000065c0: 2028 302c 2035 290a 2d62 772c 202d 2d62   (0, 5).-bw, --b
-000065d0: 696e 5f77 6964 7468 2c20 6261 6e64 7769  in_width, bandwi
-000065e0: 6474 6820 6f66 204b 7320 6469 7374 7269  dth of Ks distri
-000065f0: 6275 7469 6f6e 2c20 6465 6661 756c 7420  bution, default 
-00006600: 302e 310a 2d69 632c 202d 2d77 6569 6768  0.1.-ic, --weigh
-00006610: 7473 5f6f 7574 6c69 6572 735f 696e 636c  ts_outliers_incl
-00006620: 7564 6564 2c20 666c 6167 206f 7074 696f  uded, flag optio
-00006630: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00006640: 636c 7564 6520 4b73 206f 7574 6c69 6572  clude Ks outlier
-00006650: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00006660: 6173 2073 6574 2c20 4b73 206f 7574 6c69  as set, Ks outli
-00006670: 6572 7320 776f 756c 6420 6265 2069 6e63  ers would be inc
-00006680: 6c75 6465 6420 696e 2074 6865 2061 6e61  luded in the ana
-00006690: 6c79 7369 730a 2d6d 2c20 2d2d 6d65 7468  lysis.-m, --meth
-000066a0: 6f64 2c20 7768 6963 6820 6d69 7874 7572  od, which mixtur
-000066b0: 6520 6d6f 6465 6c20 746f 2075 7365 2c20  e model to use, 
-000066c0: 6465 6661 756c 7420 676d 6d0a 2d2d 7365  default gmm.--se
-000066d0: 6564 2c20 7261 6e64 6f6d 2073 6565 6420  ed, random seed 
-000066e0: 6769 7665 6e20 746f 2069 6e69 7469 616c  given to initial
-000066f0: 697a 6174 696f 6e2c 2064 6566 6175 6c74  ization, default
-00006700: 2032 3335 3238 3930 0a2d 6569 2c20 2d2d   2352890.-ei, --
-00006710: 656d 5f69 7465 722c 2074 6865 206e 756d  em_iter, the num
-00006720: 6265 7220 6f66 2045 4d20 6974 6572 6174  ber of EM iterat
-00006730: 696f 6e73 2074 6f20 7065 7266 6f72 6d2c  ions to perform,
-00006740: 2064 6566 6175 6c74 2032 3030 0a2d 6e69   default 200.-ni
-00006750: 2c20 2d2d 6e5f 696e 6974 2c20 7468 6520  , --n_init, the 
-00006760: 6e75 6d62 6572 206f 6620 6b2d 6d65 616e  number of k-mean
-00006770: 7320 696e 6974 6961 6c69 7a61 7469 6f6e  s initialization
-00006780: 732c 2064 6566 6175 6c74 2032 3030 0a2d  s, default 200.-
-00006790: 6e2c 202d 2d63 6f6d 706f 6e65 6e74 732c  n, --components,
-000067a0: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
-000067b0: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
-000067c0: 6f6e 656e 7473 2074 6f20 6669 742c 2064  onents to fit, d
-000067d0: 6566 6175 6c74 2028 312c 2034 290a 2d67  efault (1, 4).-g
-000067e0: 2c20 2d2d 6761 6d6d 612c 2074 6865 2067  , --gamma, the g
-000067f0: 616d 6d61 2070 6172 616d 6574 6572 2066  amma parameter f
-00006800: 6f72 2062 676d 6d20 6d6f 6465 6c73 2c20  or bgmm models, 
-00006810: 6465 6661 756c 7420 3165 2d33 0a2d 2d62  default 1e-3.--b
-00006820: 6f6f 7473 2c20 7468 6520 6e75 6d62 6572  oots, the number
-00006830: 206f 6620 626f 6f74 7374 7261 7020 7265   of bootstrap re
-00006840: 706c 6963 6174 6573 206f 6620 6b64 652c  plicates of kde,
-00006850: 2064 6566 6175 6c74 2032 3030 0a2d 2d77   default 200.--w
-00006860: 6569 6768 7465 642c 2066 6c61 6720 6f70  eighted, flag op
-00006870: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00006880: 2075 7365 206e 6f64 652d 7765 6967 6874   use node-weight
-00006890: 6564 206d 6574 686f 6420 6f66 2064 652d  ed method of de-
-000068a0: 7265 6475 6e64 616e 6379 2c20 6966 2074  redundancy, if t
-000068b0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000068c0: 2074 6865 206e 6f64 652d 7765 6967 6874   the node-weight
-000068d0: 6564 206d 6574 686f 6420 776f 756c 6420  ed method would 
-000068e0: 6265 2075 7365 640a 2d70 2c20 2d2d 706c  be used.-p, --pl
-000068f0: 6f74 2c20 7468 6520 706c 6f74 7469 6e67  ot, the plotting
-00006900: 206d 6574 686f 6420 746f 2062 6520 7573   method to be us
-00006910: 6564 2c20 6465 6661 756c 7420 6964 656e  ed, default iden
-00006920: 7469 6361 6c0a 2d62 6d2c 202d 2d62 775f  tical.-bm, --bw_
-00006930: 6d65 7468 6f64 2c20 7468 6520 6261 6e64  method, the band
-00006940: 7769 6474 6820 6d65 7468 6f64 2074 6f20  width method to 
-00006950: 6265 2075 7365 6420 696e 2061 6e61 6c79  be used in analy
-00006960: 7a69 6e67 2074 6865 2070 6561 6b20 6f66  zing the peak of
-00006970: 2057 4744 2064 6174 6573 2c20 6465 6661   WGD dates, defa
-00006980: 756c 7420 7369 6c76 6572 6d61 6e0a 2d2d  ult silverman.--
-00006990: 6e5f 6d65 646f 6964 732c 2074 6865 206e  n_medoids, the n
-000069a0: 756d 6265 7220 6f66 206d 6564 6f69 6473  umber of medoids
-000069b0: 2074 6f20 6669 742c 2064 6566 6175 6c74   to fit, default
-000069c0: 2032 0a2d 6b6d 2c20 2d2d 6b64 656d 6574   2.-km, --kdemet
-000069d0: 686f 642c 2074 6865 206b 6465 206d 6574  hod, the kde met
-000069e0: 686f 6420 746f 2062 6520 7573 6564 2069  hod to be used i
-000069f0: 6e20 616e 616c 797a 696e 6720 7468 6520  n analyzing the 
-00006a00: 7065 616b 206f 6620 5747 4420 6461 7465  peak of WGD date
-00006a10: 732c 206b 6d65 646f 6964 7320 616e 616c  s, kmedoids anal
-00006a20: 7973 6973 206f 7220 7468 6520 6261 7369  ysis or the basi
-00006a30: 6320 4b73 2070 6c6f 7474 696e 672c 2064  c Ks plotting, d
-00006a40: 6566 6175 6c74 2073 6369 7079 0a2d 2d6e  efault scipy.--n
-00006a50: 5f63 6c75 7374 6572 732c 2074 6865 206e  _clusters, the n
-00006a60: 756d 6265 7220 6f66 2063 6c75 7374 6572  umber of cluster
-00006a70: 7320 746f 2070 6c6f 7420 456c 626f 7720  s to plot Elbow 
-00006a80: 6c6f 7373 2066 756e 6374 696f 6e2c 2064  loss function, d
-00006a90: 6566 6175 6c74 2035 0a2d 6764 2c20 2d2d  efault 5.-gd, --
-00006aa0: 6775 6964 652c 2074 6865 2072 6567 696d  guide, the regim
-00006ab0: 6520 7265 7369 6469 6e67 2061 6e63 686f  e residing ancho
-00006ac0: 7273 2c20 6465 6661 756c 7420 5365 676d  rs, default Segm
-00006ad0: 656e 740a 2d70 7263 742c 202d 2d70 726f  ent.-prct, --pro
-00006ae0: 6d69 6e65 6e63 655f 6375 746f 6666 2c20  minence_cutoff, 
-00006af0: 7468 6520 7072 6f6d 696e 656e 6365 2063  the prominence c
-00006b00: 7574 6f66 6620 6f66 2061 6363 6570 7461  utoff of accepta
-00006b10: 626c 6520 7065 616b 7320 696e 2070 6561  ble peaks in pea
-00006b20: 6b20 6669 6e64 696e 6720 7374 6570 732c  k finding steps,
-00006b30: 2064 6566 6175 6c74 2030 2e31 0a2d 7268   default 0.1.-rh
-00006b40: 2c20 2d2d 7265 6c5f 6865 6967 6874 2c20  , --rel_height, 
-00006b50: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
-00006b60: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
-00006b70: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
-00006b80: 6561 7375 7265 642c 2064 6566 6175 6c74  easured, default
-00006b90: 2030 2e34 0a2d 6b64 2c20 2d2d 6b73 746f   0.4.-kd, --ksto
-00006ba0: 6461 7465 2c20 7468 6520 7261 6e67 6520  date, the range 
-00006bb0: 6f66 204b 7320 746f 2062 6520 6461 7465  of Ks to be date
-00006bc0: 6420 696e 2068 6575 7269 7374 6963 2073  d in heuristic s
-00006bd0: 6561 7263 682c 2064 6566 6175 6c74 2028  earch, default (
-00006be0: 302e 352c 2031 2e35 290a 2d78 6c2c 202d  0.5, 1.5).-xl, -
-00006bf0: 2d78 6c69 6d2c 2074 6865 2078 2061 7869  -xlim, the x axi
-00006c00: 7320 6c69 6d69 7420 6f66 2047 4d4d 204b  s limit of GMM K
-00006c10: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
-00006c20: 796c 2c20 2d2d 796c 696d 2c20 7468 6520  yl, --ylim, the 
-00006c30: 7920 6178 6973 206c 696d 6974 206f 6620  y axis limit of 
-00006c40: 474d 4d20 4b73 2064 6973 7472 6962 7574  GMM Ks distribut
-00006c50: 696f 6e0a 2d2d 6d61 6e75 616c 7365 742c  ion.--manualset,
-00006c60: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00006c70: 6574 6865 7220 746f 206f 7574 7075 7420  ether to output 
-00006c80: 616e 6368 6f72 2070 6169 7273 2077 6974  anchor pairs wit
-00006c90: 6820 6d61 6e75 616c 6c79 2073 6574 204b  h manually set K
-00006ca0: 7320 7261 6e67 652c 2069 6620 7468 6520  s range, if the 
-00006cb0: 666c 6167 2077 6173 2073 6574 2c20 6d61  flag was set, ma
-00006cc0: 6e75 616c 6c79 2073 6574 204b 7320 7261  nually set Ks ra
-00006cd0: 6e67 6520 776f 756c 6420 6265 206f 7574  nge would be out
-00006ce0: 7075 7474 6564 0a2d 2d63 692c 2074 6865  putted.--ci, the
-00006cf0: 2063 6f6e 6669 6465 6e63 6520 6c65 7665   confidence leve
-00006d00: 6c20 6f66 206c 6f67 2d6e 6f72 6d61 6c20  l of log-normal 
-00006d10: 6469 7374 7269 6275 7469 6f6e 2074 6f20  distribution to 
-00006d20: 6461 7465 2c20 6465 6661 756c 7420 3935  date, default 95
-00006d30: 0a2d 2d68 6472 2c20 7468 6520 6869 6768  .--hdr, the high
-00006d40: 6573 7420 6465 6e73 6974 7920 7265 6769  est density regi
-00006d50: 6f6e 2028 4844 5229 2061 7070 6c69 6564  on (HDR) applied
-00006d60: 2069 6e20 7468 6520 7365 676d 656e 742d   in the segment-
-00006d70: 6775 6964 6564 2061 6e63 686f 7220 7061  guided anchor pa
-00006d80: 6972 204b 7320 6469 7374 7269 6275 7469  ir Ks distributi
-00006d90: 6f6e 2c20 6465 6661 756c 7420 3935 0a2d  on, default 95.-
-00006da0: 2d68 6575 7269 7374 6963 2c20 666c 6167  -heuristic, flag
-00006db0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-00006dc0: 2074 6f20 696e 6974 6961 7465 2068 6575   to initiate heu
-00006dd0: 7269 7374 6963 206d 6574 686f 6420 6f66  ristic method of
-00006de0: 2064 6566 696e 696e 6720 4349 2066 6f72   defining CI for
-00006df0: 2064 6174 696e 672c 2069 6620 7468 6520   dating, if the 
+00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
+00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
+00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 380a  Version: 2.0.38.
+00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
+00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
+00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
+00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
+00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
+00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
+00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
+000000a0: 650a 4c69 6365 6e73 653a 2047 504c 0a44  e.License: GPL.D
+000000b0: 6573 6372 6970 7469 6f6e 2d43 6f6e 7465  escription-Conte
+000000c0: 6e74 2d54 7970 653a 2074 6578 742f 6d61  nt-Type: text/ma
+000000d0: 726b 646f 776e 0a4c 6963 656e 7365 2d46  rkdown.License-F
+000000e0: 696c 653a 204c 4943 454e 5345 0a0a 3c64  ile: LICENSE..<d
+000000f0: 6976 2061 6c69 676e 3d22 6365 6e74 6572  iv align="center
+00000100: 223e 0a0a 2320 6077 6764 2076 3260 203a  ">..# `wgd v2` :
+00000110: 2061 2073 7569 7465 2074 6f6f 6c20 6f66   a suite tool of
+00000120: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
+00000130: 6e64 2074 696d 696e 670a 0a5b 215b 4275  nd timing..[![Bu
+00000140: 696c 6420 5374 6174 7573 5d28 6874 7470  ild Status](http
+00000150: 733a 2f2f 6170 702e 7472 6176 6973 2d63  s://app.travis-c
+00000160: 692e 636f 6d2f 6865 6368 652d 7073 622f  i.com/heche-psb/
+00000170: 7767 642e 7376 673f 6272 616e 6368 3d70  wgd.svg?branch=p
+00000180: 6879 6c6f 6461 7469 6e67 295d 2868 7474  hylodating)](htt
+00000190: 7073 3a2f 2f61 7070 2e74 7261 7669 732d  ps://app.travis-
+000001a0: 6369 2e63 6f6d 2f68 6563 6865 2d70 7362  ci.com/heche-psb
+000001b0: 2f77 6764 290a 5b21 5b44 6f63 756d 656e  /wgd).[![Documen
+000001c0: 7461 7469 6f6e 2053 7461 7475 735d 2868  tation Status](h
+000001d0: 7474 7073 3a2f 2f72 6561 6474 6865 646f  ttps://readthedo
+000001e0: 6373 2e6f 7267 2f70 726f 6a65 6374 732f  cs.org/projects/
+000001f0: 7767 6476 322f 6261 6467 652f 3f76 6572  wgdv2/badge/?ver
+00000200: 7369 6f6e 3d6c 6174 6573 7429 5d28 6874  sion=latest)](ht
+00000210: 7470 733a 2f2f 7767 6476 322e 7265 6164  tps://wgdv2.read
+00000220: 7468 6564 6f63 732e 696f 2f65 6e2f 6c61  thedocs.io/en/la
+00000230: 7465 7374 2f3f 6261 6467 653d 6c61 7465  test/?badge=late
+00000240: 7374 290a 5b21 5b6c 6963 656e 7365 5d28  st).[![license](
+00000250: 6874 7470 733a 2f2f 696d 672e 7368 6965  https://img.shie
+00000260: 6c64 732e 696f 2f70 7970 692f 6c2f 7767  lds.io/pypi/l/wg
+00000270: 642e 7376 6729 5d28 6874 7470 733a 2f2f  d.svg)](https://
+00000280: 7079 7069 2e70 7974 686f 6e2e 6f72 672f  pypi.python.org/
+00000290: 7079 7069 2f77 6764 290a 5b21 5b4c 6174  pypi/wgd).[![Lat
+000002a0: 6573 7420 5079 5049 2076 6572 7369 6f6e  est PyPI version
+000002b0: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
+000002c0: 6965 6c64 732e 696f 2f70 7970 692f 762f  ields.io/pypi/v/
+000002d0: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
+000002e0: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
+000002f0: 672f 7079 7069 2f77 6764 290a 5b21 5b69  g/pypi/wgd).[![i
+00000300: 6e73 7461 6c6c 2077 6974 6820 6269 6f63  nstall with bioc
+00000310: 6f6e 6461 5d28 6874 7470 733a 2f2f 696d  onda](https://im
+00000320: 672e 7368 6965 6c64 732e 696f 2f62 6164  g.shields.io/bad
+00000330: 6765 2f69 6e73 7461 6c6c 2532 3077 6974  ge/install%20wit
+00000340: 682d 6269 6f63 6f6e 6461 2d62 7269 6768  h-bioconda-brigh
+00000350: 7467 7265 656e 2e73 7667 3f73 7479 6c65  tgreen.svg?style
+00000360: 3d66 6c61 7429 5d28 6874 7470 3a2f 2f62  =flat)](http://b
+00000370: 696f 636f 6e64 612e 6769 7468 7562 2e69  ioconda.github.i
+00000380: 6f2f 7265 6369 7065 732f 7767 642f 5245  o/recipes/wgd/RE
+00000390: 4144 4d45 2e68 746d 6c29 0a5b 215b 446f  ADME.html).[![Do
+000003a0: 776e 6c6f 6164 735d 2868 7474 7073 3a2f  wnloads](https:/
+000003b0: 2f70 6570 792e 7465 6368 2f62 6164 6765  /pepy.tech/badge
+000003c0: 2f77 6764 295d 2868 7474 7073 3a2f 2f70  /wgd)](https://p
+000003d0: 6570 792e 7465 6368 2f70 726f 6a65 6374  epy.tech/project
+000003e0: 2f77 6764 290a 0a2a 2a48 656e 6763 6869  /wgd)..**Hengchi
+000003f0: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
+00000400: 6165 6e65 706f 656c 2c20 5976 6573 2056  aenepoel, Yves V
+00000410: 616e 2064 6520 5065 6572 2a2a 0a0a 5b2a  an de Peer**..[*
+00000420: 2a42 696f 696e 666f 726d 6174 6963 7320  *Bioinformatics 
+00000430: 2620 4576 6f6c 7574 696f 6e61 7279 2047  & Evolutionary G
+00000440: 656e 6f6d 6963 7320 4772 6f75 702a 2a5d  enomics Group**]
+00000450: 2868 7474 7073 3a2f 2f77 7777 2e76 616e  (https://www.van
+00000460: 6465 7065 6572 6c61 622e 6f72 672f 292a  depeerlab.org/)*
+00000470: 2a2c 2056 4942 2d55 4765 6e74 2043 656e  *, VIB-UGent Cen
+00000480: 7465 7220 666f 7220 506c 616e 7420 5379  ter for Plant Sy
+00000490: 7374 656d 7320 4269 6f6c 6f67 792a 2a0a  stems Biology**.
+000004a0: 0a5b 2a2a 496e 7472 6f64 7563 7469 6f6e  .[**Introduction
+000004b0: 2a2a 5d28 2369 6e74 726f 6475 6374 696f  **](#introductio
+000004c0: 6e29 207c 200a 5b2a 2a49 6e73 7461 6c6c  n) | .[**Install
+000004d0: 6174 696f 6e2a 2a5d 2823 696e 7374 616c  ation**](#instal
+000004e0: 6c61 7469 6f6e 2920 7c0a 5b2a 2a50 6970  lation) |.[**Pip
+000004f0: 656c 696e 6573 2a2a 5d28 2370 6970 656c  elines**](#pipel
+00000500: 696e 6573 2920 7c0a 5b2a 2a50 6172 616d  ines) |.[**Param
+00000510: 6574 6572 732a 2a5d 2823 7061 7261 6d65  eters**](#parame
+00000520: 7465 7273 2920 7c20 0a5b 2a2a 5573 6167  ters) | .[**Usag
+00000530: 652a 2a5d 2823 7573 6167 6529 207c 200a  e**](#usage) | .
+00000540: 5b2a 2a49 6c6c 7573 7472 6174 696f 6e2a  [**Illustration*
+00000550: 2a5d 2823 696c 6c75 7374 7261 7469 6f6e  *](#illustration
+00000560: 2920 7c0a 5b2a 2a44 6f63 756d 656e 7461  ) |.[**Documenta
+00000570: 7469 6f6e 2a2a 5d28 6874 7470 733a 2f2f  tion**](https://
+00000580: 7767 6476 322e 7265 6164 7468 6564 6f63  wgdv2.readthedoc
+00000590: 732e 696f 2f65 6e2f 6c61 7465 7374 2f3f  s.io/en/latest/?
+000005a0: 6261 6467 653d 6c61 7465 7374 2920 7c0a  badge=latest) |.
+000005b0: 5b2a 2a43 6974 6174 696f 6e2a 2a5d 2823  [**Citation**](#
+000005c0: 6369 7461 7469 6f6e 290a 0a3c 2f64 6976  citation)..</div
+000005d0: 3e0a 0a60 7767 6420 7632 6020 6973 2061  >..`wgd v2` is a
+000005e0: 2070 7974 686f 6e20 7061 636b 6167 6520   python package 
+000005f0: 7570 6772 6164 6564 2066 726f 6d20 7468  upgraded from th
+00000600: 6520 6f72 6967 696e 616c 2060 7767 6460  e original `wgd`
+00000610: 2070 6163 6b61 6765 2061 696d 696e 6720   package aiming 
+00000620: 666f 7220 7468 6520 696e 6665 7265 6e63  for the inferenc
+00000630: 6520 616e 6420 7469 6d69 6e67 206f 6620  e and timing of 
+00000640: 616e 6369 656e 7420 7768 6f6c 652d 6765  ancient whole-ge
+00000650: 6e6f 6d65 2064 7570 6c69 6361 7469 6f6e  nome duplication
+00000660: 2028 5747 4429 2065 7665 6e74 732e 2046   (WGD) events. F
+00000670: 6f72 2074 6865 2070 726f 706f 7365 206f  or the propose o
+00000680: 6620 696c 6c75 7374 7261 7469 6e67 2074  f illustrating t
+00000690: 6865 2070 7269 6e63 6970 6c65 2061 6e64  he principle and
+000006a0: 2075 7361 6765 206f 6620 6077 6764 2076   usage of `wgd v
+000006b0: 3260 2c20 7765 2063 6f6d 7069 6c65 6420  2`, we compiled 
+000006c0: 7468 6973 2064 6f63 756d 656e 7461 7469  this documentati
+000006d0: 6f6e 2e20 4265 6c6f 7720 7765 2066 6972  on. Below we fir
+000006e0: 7374 2067 6176 6520 616e 2069 6e74 726f  st gave an intro
+000006f0: 6475 6374 696f 6e20 6f76 6572 2074 6865  duction over the
+00000700: 2073 636f 7065 2061 6e64 206d 6563 6861   scope and mecha
+00000710: 6e69 736d 206f 6620 6077 6764 2076 3260  nism of `wgd v2`
+00000720: 2061 6e64 2074 6865 6e20 7468 6520 7072   and then the pr
+00000730: 6163 7469 6361 6c20 696e 666f 726d 6174  actical informat
+00000740: 696f 6e20 7065 7274 6169 6e69 6e67 2074  ion pertaining t
+00000750: 6f20 7468 6520 696e 7374 616c 6c61 7469  o the installati
+00000760: 6f6e 2061 6e64 2075 7361 6765 2e20 416e  on and usage. An
+00000770: 2065 7861 6d70 6c61 7220 776f 726b 666c   examplar workfl
+00000780: 6f77 2069 7320 7072 6f76 6964 6564 2069  ow is provided i
+00000790: 6e20 7468 6520 7475 746f 7269 616c 2073  n the tutorial s
+000007a0: 6563 7469 6f6e 206f 6e20 686f 7720 746f  ection on how to
+000007b0: 2073 6565 6b20 6576 6964 656e 6365 2066   seek evidence f
+000007c0: 6f72 2061 2070 7574 6174 6976 6520 5747  or a putative WG
+000007d0: 4420 6576 656e 7420 616e 6420 7065 7266  D event and perf
+000007e0: 6f72 6d20 7072 6f70 6572 2074 696d 696e  orm proper timin
+000007f0: 6720 7769 7468 2061 2066 7265 7368 6c79  g with a freshly
+00000800: 206f 6274 6169 6e65 6420 6765 6e6f 6d65   obtained genome
+00000810: 2061 7373 656d 626c 7920 696e 2068 616e   assembly in han
+00000820: 642e 2046 6f72 2074 686f 7365 2077 686f  d. For those who
+00000830: 2061 7265 2069 6e74 6572 6573 7465 6420   are interested 
+00000840: 696e 206d 6f72 6520 7468 656f 7265 7469  in more theoreti
+00000850: 6361 6c20 6465 7461 696c 732c 2077 6520  cal details, we 
+00000860: 7265 636f 6d6d 656e 6420 7475 726e 696e  recommend turnin
+00000870: 6720 746f 206f 7572 2070 6170 6572 2061  g to our paper a
+00000880: 6e64 2062 6f6f 6b20 6368 6170 7465 7220  nd book chapter 
+00000890: 666f 7220 6d6f 7265 2064 6574 6169 6c65  for more detaile
+000008a0: 6420 6465 7363 7269 7074 696f 6e20 616e  d description an
+000008b0: 6420 696e 7369 6768 7466 756c 2064 6973  d insightful dis
+000008c0: 6375 7373 696f 6e73 2e20 5468 6520 6b65  cussions. The ke
+000008d0: 7920 696d 7072 6f76 6564 2066 6561 7475  y improved featu
+000008e0: 7265 7320 6f66 2060 7767 6420 7632 6020  res of `wgd v2` 
+000008f0: 6172 6520 6465 6d6f 6e73 7472 6174 6564  are demonstrated
+00000900: 2069 6e20 6f75 7220 6c61 7465 7374 206d   in our latest m
+00000910: 616e 7573 6372 6970 7420 6f6e 205b 4269  anuscript on [Bi
+00000920: 6f69 6e66 6f72 6d61 7469 6373 5d28 6874  oinformatics](ht
+00000930: 7470 733a 2f2f 646f 692e 6f72 672f 3130  tps://doi.org/10
+00000940: 2e31 3039 332f 6269 6f69 6e66 6f72 6d61  .1093/bioinforma
+00000950: 7469 6373 2f62 7461 6532 3732 292e 2049  tics/btae272). I
+00000960: 6620 796f 7520 7573 6520 6077 6764 2076  f you use `wgd v
+00000970: 3260 2069 6e20 796f 7572 2072 6573 6561  2` in your resea
+00000980: 7263 682c 2070 6c65 6173 6520 6369 7465  rch, please cite
+00000990: 2075 7320 6173 2073 7567 6765 7374 6564   us as suggested
+000009a0: 2069 6e20 5b43 6974 6174 696f 6e5d 2823   in [Citation](#
+000009b0: 6369 7461 7469 6f6e 2920 7365 6374 696f  citation) sectio
+000009c0: 6e2e 0a0a 2323 2049 6e74 726f 6475 6374  n...## Introduct
+000009d0: 696f 6e0a 0a50 6f6c 7970 6c6f 6964 697a  ion..Polyploidiz
+000009e0: 6174 696f 6e73 2c20 7468 6520 6576 6f6c  ations, the evol
+000009f0: 7574 696f 6e61 7279 2070 726f 6365 7373  utionary process
+00000a00: 2074 6861 7420 7468 6520 656e 7469 7265   that the entire
+00000a10: 2067 656e 6f6d 6520 6f66 2061 6e20 6f72   genome of an or
+00000a20: 6761 6e69 736d 2069 7320 6475 706c 6963  ganism is duplic
+00000a30: 6174 6564 2c20 616c 736f 206e 616d 6564  ated, also named
+00000a40: 2061 7320 7768 6f6c 652d 6765 6e6f 6d65   as whole-genome
+00000a50: 2064 7570 6c69 6361 7469 6f6e 7320 2857   duplications (W
+00000a60: 4744 7329 2c20 6f63 6375 7220 7265 6375  GDs), occur recu
+00000a70: 7272 656e 746c 7920 6163 726f 7373 2074  rrently across t
+00000a80: 6865 2074 7265 6520 6f66 206c 6966 652e  he tree of life.
+00000a90: 2054 6865 7265 2061 7265 2074 776f 206d   There are two m
+00000aa0: 6f64 6573 206f 6620 706f 6c79 706c 6f69  odes of polyploi
+00000ab0: 6469 7a61 7469 6f6e 732c 2061 7574 6f70  dizations, autop
+00000ac0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e73  olyploidizations
+00000ad0: 2061 6e64 2061 6c6c 6f70 6f6c 7970 6c6f   and allopolyplo
+00000ae0: 6964 697a 6174 696f 6e73 2e20 4175 746f  idizations. Auto
+00000af0: 706f 6c79 706c 6f69 6469 7a61 7469 6f6e  polyploidization
+00000b00: 7320 6172 6520 7468 6520 6475 706c 6963  s are the duplic
+00000b10: 6174 696f 6e20 6f66 2074 6865 2073 616d  ation of the sam
+00000b20: 6520 6765 6e6f 6d65 2c20 7265 7375 6c74  e genome, result
+00000b30: 696e 6720 696e 2074 776f 2069 6465 6e74  ing in two ident
+00000b40: 6963 616c 2073 7562 6765 6e6f 6d65 7320  ical subgenomes 
+00000b50: 6174 2074 6865 2074 696d 6520 6974 2065  at the time it e
+00000b60: 6d65 7267 6564 2e20 5768 696c 6520 7468  merged. While th
+00000b70: 6520 616c 6c6f 706f 6c79 706c 6f69 6469  e allopolyploidi
+00000b80: 7a61 7469 6f6e 7320 6172 6520 6e6f 726d  zations are norm
+00000b90: 616c 6c79 2061 6368 6965 7665 6420 696e  ally achieved in
+00000ba0: 2074 776f 2073 7465 7073 2c20 6669 7273   two steps, firs
+00000bb0: 7420 7468 6520 6879 6272 6964 697a 6174  t the hybridizat
+00000bc0: 696f 6e20 6265 7477 6565 6e20 7477 6f20  ion between two 
+00000bd0: 6469 6666 6572 656e 7420 7370 6563 6965  different specie
+00000be0: 732c 2072 6573 756c 7469 6e67 2069 6e20  s, resulting in 
+00000bf0: 7468 6520 6172 6973 696e 6720 6f66 2074  the arising of t
+00000c00: 7261 6e73 6965 6e74 2068 6f6d 6f70 6c6f  ransient homoplo
+00000c10: 6964 792c 7365 636f 6e64 2074 6865 2064  idy,second the d
+00000c20: 7570 6c69 6361 7469 6f6e 206f 6620 7468  uplication of th
+00000c30: 6520 686f 6d6f 706c 6f69 6479 2c20 7265  e homoploidy, re
+00000c40: 7375 6c74 696e 6720 696e 2074 6865 2065  sulting in the e
+00000c50: 6d65 7267 656e 6365 206f 6620 616c 6c6f  mergence of allo
+00000c60: 706f 6c79 706c 6f69 6479 2e20 4475 6520  polyploidy. Due 
+00000c70: 746f 2074 6865 2075 6e73 7461 6269 6c69  to the unstabili
+00000c80: 7479 2061 6e64 2075 6e62 616c 616e 6365  ty and unbalance
+00000c90: 6420 7465 7472 6173 6f6d 6963 2069 6e68  d tetrasomic inh
+00000ca0: 6572 6974 616e 6365 2c20 666f 7220 696e  eritance, for in
+00000cb0: 7374 616e 6365 2074 6865 206e 7563 6c65  stance the nucle
+00000cc0: 6172 2d63 7974 6f70 6c61 736d 6963 2069  ar-cytoplasmic i
+00000cd0: 6e63 6f6d 7061 7469 6269 6c69 7479 2c20  ncompatibility, 
+00000ce0: 7468 6520 706f 6c79 706c 6f69 6479 2067  the polyploidy g
+00000cf0: 656e 6f6d 6520 7769 6c6c 2074 6865 6e20  enome will then 
+00000d00: 6578 7065 7269 656e 6365 2061 2070 726f  experience a pro
+00000d10: 6365 7373 2063 616c 6c65 6420 6469 706c  cess called dipl
+00000d20: 6f69 6469 7a61 7469 6f6e 2c20 616c 736f  oidization, also
+00000d30: 206e 616d 6564 2061 7320 6672 6163 7469   named as fracti
+00000d40: 6f6e 6174 696f 6e2c 2064 7572 696e 6720  onation, during 
+00000d50: 7768 6963 6820 6120 6c61 7267 6520 706f  which a large po
+00000d60: 7274 696f 6e20 6f66 2067 656e 6520 6475  rtion of gene du
+00000d70: 706c 6963 6174 6573 2077 696c 6c20 6765  plicates will ge
+00000d80: 7420 6c6f 7374 2061 6e64 206f 6e6c 7920  t lost and only 
+00000d90: 6120 6672 6163 7469 6f6e 2063 616e 2062  a fraction can b
+00000da0: 6520 7265 7461 696e 6564 2e20 5468 6520  e retained. The 
+00000db0: 7472 6163 6573 206f 6620 706f 6c79 706c  traces of polypl
+00000dc0: 6f69 6469 7a61 7469 6f6e 7320 6361 6e20  oidizations can 
+00000dd0: 6265 2074 6875 7320 756e 6561 7274 6865  be thus unearthe
+00000de0: 6420 6672 6f6d 2074 6865 7365 2072 6574  d from these ret
+00000df0: 6169 6e65 6420 6765 6e65 2064 7570 6c69  ained gene dupli
+00000e00: 6361 7465 732e 2054 6872 6565 2061 7070  cates. Three app
+00000e10: 726f 6163 6865 7320 6261 7365 6420 6f6e  roaches based on
+00000e20: 2067 656e 6520 6475 706c 6963 6174 6573   gene duplicates
+00000e30: 2c20 6e61 6d65 6c79 2c20 2a4b 2a3c 7375  , namely, *K*<su
+00000e40: 623e 533c 2f73 7562 3e20 6d65 7468 6f64  b>S</sub> method
+00000e50: 2c20 6765 6e65 2074 7265 6520 2d20 7370  , gene tree - sp
+00000e60: 6563 6965 7320 7472 6565 2072 6563 6f6e  ecies tree recon
+00000e70: 6369 6c69 6174 696f 6e20 6d65 7468 6f64  ciliation method
+00000e80: 2061 6e64 2073 796e 7465 6e79 206d 6574   and synteny met
+00000e90: 686f 642c 2061 7265 2063 6f6d 6d6f 6e6c  hod, are commonl
+00000ea0: 7920 7573 6564 2069 6e20 6465 7465 6374  y used in detect
+00000eb0: 696e 6720 6576 6964 656e 6365 2066 6f72  ing evidence for
+00000ec0: 2057 4744 732e 2054 6865 2067 656e 6520   WGDs. The gene 
+00000ed0: 7472 6565 202d 2073 7065 6369 6573 2074  tree - species t
+00000ee0: 7265 6520 7265 636f 6e63 696c 6961 7469  ree reconciliati
+00000ef0: 6f6e 206d 6574 686f 6420 6973 206e 6f74  on method is not
+00000f00: 2077 6974 6869 6e20 7468 6520 7363 6f70   within the scop
+00000f10: 6520 6f66 2060 7767 6420 7632 602c 2062  e of `wgd v2`, b
+00000f20: 7574 2077 6520 6b69 6e64 6c79 2072 6566  ut we kindly ref
+00000f30: 6572 2072 6561 6465 7273 2077 686f 2061  er readers who a
+00000f40: 7265 2069 6e74 6572 6573 7465 6420 746f  re interested to
+00000f50: 2074 6865 2070 6879 6c6f 6765 6e6f 6d69   the phylogenomi
+00000f60: 6320 7072 6f67 7261 6d20 6465 7665 6c6f  c program develo
+00000f70: 7065 6420 6279 2041 7274 6875 7220 5a77  ped by Arthur Zw
+00000f80: 6165 6e65 706f 656c 206e 616d 6564 205b  aenepoel named [
+00000f90: 5748 414c 455d 2868 7474 7073 3a2f 2f67  WHALE](https://g
+00000fa0: 6974 6875 622e 636f 6d2f 6172 7a77 612f  ithub.com/arzwa/
+00000fb0: 5768 616c 652e 6a6c 2920 616e 6420 7468  Whale.jl) and th
+00000fc0: 6520 6173 736f 6369 6174 6564 205b 7061  e associated [pa
+00000fd0: 7065 725d 2868 7474 7073 3a2f 2f64 6f69  per](https://doi
+00000fe0: 2e6f 7267 2f31 302e 3130 3933 2f6d 6f6c  .org/10.1093/mol
+00000ff0: 6265 762f 6d73 7a30 3838 2920 666f 7220  bev/msz088) for 
+00001000: 6d6f 7265 2074 6563 686e 6963 616c 2061  more technical a
+00001010: 6e64 2074 6865 6f72 6574 6963 616c 2064  nd theoretical d
+00001020: 6574 6169 6c73 2e0a 0a54 6865 202a 4b2a  etails...The *K*
+00001030: 3c73 7562 3e53 3c2f 7375 623e 206d 6574  <sub>S</sub> met
+00001040: 686f 6420 6973 2065 7374 6162 6c69 7368  hod is establish
+00001050: 6564 206f 6e20 6120 6d6f 6465 6c20 6f66  ed on a model of
+00001060: 2067 656e 6520 6661 6d69 6c79 2065 766f   gene family evo
+00001070: 6c75 7469 6f6e 2074 6861 7420 6561 6368  lution that each
+00001080: 2067 656e 6520 6661 6d69 6c79 2069 7320   gene family is 
+00001090: 616c 6c6f 7765 6420 746f 2065 766f 6c76  allowed to evolv
+000010a0: 6520 7669 6120 6765 6e65 2064 7570 6c69  e via gene dupli
+000010b0: 6361 7469 6f6e 2061 6e64 206c 6f73 732e  cation and loss.
+000010c0: 204e 6f74 6520 7468 6174 2074 6865 2067   Note that the g
+000010d0: 656e 6520 6661 6d69 6c79 2068 6572 6520  ene family here 
+000010e0: 6973 2061 7373 756d 6564 2074 6f20 6265  is assumed to be
+000010f0: 2074 6865 2063 6c75 7374 6572 206f 6620   the cluster of 
+00001100: 616c 6c20 6765 6e65 7320 6465 7363 656e  all genes descen
+00001110: 6465 6420 6672 6f6d 2061 6e20 616e 6365  ded from an ance
+00001120: 7374 7261 6c20 6765 6e65 2069 6e20 6120  stral gene in a 
+00001130: 7369 6e67 6c65 2067 656e 6f6d 652e 2052  single genome. R
+00001140: 6563 6f76 6572 696e 6720 7468 6520 6765  ecovering the ge
+00001150: 6e65 2074 7265 6520 6f66 2073 7563 6820  ne tree of such 
+00001160: 6765 6e65 2066 616d 696c 7920 696e 666f  gene family info
+00001170: 726d 7320 7468 6520 7469 6d69 6e67 2c20  rms the timing, 
+00001180: 7363 696c 6963 6574 2074 6865 2061 6765  scilicet the age
+00001190: 2c20 6f66 2067 656e 6520 6475 706c 6963  , of gene duplic
+000011a0: 6174 696f 6e20 6576 656e 7473 2e20 5468  ation events. Th
+000011b0: 6520 6167 6520 7265 6665 7265 6420 6865  e age refered he
+000011c0: 7265 2c20 6973 206e 6f74 2069 6e20 7265  re, is not in re
+000011d0: 616c 2067 656f 6c6f 6769 6361 6c20 7469  al geological ti
+000011e0: 6d65 2c20 6275 7420 696e 2074 6865 2075  me, but in the u
+000011f0: 6e69 7420 6f66 2065 766f 6c75 7469 6f6e  nit of evolution
+00001200: 6172 7920 6469 7374 616e 6365 2c20 692e  ary distance, i.
+00001210: 652e 2c20 7468 6520 6e75 6d62 6572 206f  e., the number o
+00001220: 6620 7375 6273 7469 7475 7469 6f6e 7320  f substitutions 
+00001230: 7065 7220 7369 7465 2e20 5768 656e 2074  per site. When t
+00001240: 6865 2065 766f 6c75 7469 6f6e 6172 7920  he evolutionary 
+00001250: 7261 7465 2072 656d 6169 6e73 2061 7070  rate remains app
+00001260: 726f 7869 6d61 7465 6c79 2063 6f6e 7374  roximately const
+00001270: 616e 742c 2074 6865 2065 766f 6c75 7469  ant, the evoluti
+00001280: 6f6e 6172 7920 6469 7374 616e 6365 2069  onary distance i
+00001290: 7320 7468 656e 2073 7570 706f 7365 6420  s then supposed 
+000012a0: 746f 2062 6520 7072 6f70 6f72 7469 6f6e  to be proportion
+000012b0: 616c 2074 6f20 7468 6520 7265 616c 2065  al to the real e
+000012c0: 766f 6c75 7469 6f6e 6172 7920 7469 6d65  volutionary time
+000012d0: 2e20 5468 6520 7379 6e6f 6e79 6d6f 7573  . The synonymous
+000012e0: 2064 6973 7461 6e63 6520 2a4b 2a3c 7375   distance *K*<su
+000012f0: 623e 533c 2f73 7562 3e2c 2074 6865 206e  b>S</sub>, the n
+00001300: 756d 6265 7220 6f66 2073 796e 6f6e 796d  umber of synonym
+00001310: 6f75 7320 7375 6273 7469 7475 7469 6f6e  ous substitution
+00001320: 7320 7065 7220 7379 6e6f 6e79 6d6f 7573  s per synonymous
+00001330: 2073 6974 652c 2069 7320 7375 6368 2063   site, is such c
+00001340: 616e 6469 6461 7465 2074 6861 7420 7379  andidate that sy
+00001350: 6e6f 6e79 6d6f 7573 2073 7562 7374 6974  nonymous substit
+00001360: 7574 696f 6e73 2077 6f75 6c64 206e 6f74  utions would not
+00001370: 2069 6e63 7572 2074 6865 2063 6861 6e67   incur the chang
+00001380: 6520 6f66 2061 6d69 6e6f 2061 6369 6420  e of amino acid 
+00001390: 616e 6420 6172 6520 7468 7573 2072 6567  and are thus reg
+000013a0: 6172 6465 6420 6173 206e 6575 7472 616c  arded as neutral
+000013b0: 2c20 7768 6963 6820 6163 636f 7264 696e  , which accordin
+000013c0: 6720 746f 2074 6865 206e 6575 7472 616c  g to the neutral
+000013d0: 2074 6865 6f72 7920 7368 6f75 6c64 206f   theory should o
+000013e0: 6363 7572 2069 6e20 636f 6e73 7461 6e74  ccur in constant
+000013f0: 2072 6174 652e 2047 6976 656e 2061 206d   rate. Given a m
+00001400: 6f64 656c 206f 6620 6765 6e65 2066 616d  odel of gene fam
+00001410: 696c 7920 7468 6174 2061 6c6c 6f77 7320  ily that allows 
+00001420: 7468 6520 6765 6e65 2074 6f20 6475 706c  the gene to dupl
+00001430: 6963 6174 6520 616e 6420 6765 7420 6c6f  icate and get lo
+00001440: 7374 2069 6e20 6120 6669 7865 6420 7261  st in a fixed ra
+00001450: 7465 2c20 6f6e 6520 6361 6e20 6465 7269  te, one can deri
+00001460: 7665 2074 6861 7420 7468 6520 7072 6f62  ve that the prob
+00001470: 6162 696c 6974 7920 6465 6e73 6974 7920  ability density 
+00001480: 6675 6e63 7469 6f6e 206f 6620 7468 6520  function of the 
+00001490: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000014a0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+000014b0: 206f 6620 7265 7461 696e 6564 2067 656e   of retained gen
+000014c0: 6520 6475 706c 6963 6174 6573 2069 7320  e duplicates is 
+000014d0: 6120 7175 6173 692d 6578 706f 6e65 6e74  a quasi-exponent
+000014e0: 6961 6c20 6675 6e63 7469 6f6e 2074 6861  ial function tha
+000014f0: 7420 6d6f 7374 2072 6574 6169 6e65 6420  t most retained 
+00001500: 6765 6e65 2064 7570 6c69 6361 7465 7320  gene duplicates 
+00001510: 6172 6520 7265 6365 6e74 6c79 2062 6f72  are recently bor
+00001520: 6e65 6420 7769 7468 207e 3020 6167 6520  ned with ~0 age 
+00001530: 7768 696c 6520 6173 2074 6865 2061 6765  while as the age
+00001540: 2067 6f69 6e67 206f 6c64 6572 2074 6865   going older the
+00001550: 2061 7373 6f63 6961 7465 6420 6e75 6d62   associated numb
+00001560: 6572 206f 6620 7265 7461 696e 6564 2067  er of retained g
+00001570: 656e 6520 6475 706c 6963 6174 6573 2064  ene duplicates d
+00001580: 6563 6179 2071 7561 7369 2d65 7870 6f6e  ecay quasi-expon
+00001590: 656e 7469 616c 6c79 2e20 5468 6572 6566  entially. Theref
+000015a0: 6f72 652c 2074 6865 206f 6363 7572 616e  ore, the occuran
+000015b0: 6365 206f 6620 6c61 7267 652d 7363 616c  ce of large-scal
+000015c0: 6520 6765 6e65 2064 7570 6c69 6361 7469  e gene duplicati
+000015d0: 6f6e 2065 7665 6e74 732c 2066 6f72 2069  on events, for i
+000015e0: 6e73 7461 6e65 2057 4744 732c 2077 6974  nstane WGDs, wit
+000015f0: 6820 7661 7269 6564 2072 6574 656e 7469  h varied retenti
+00001600: 6f6e 2072 6174 652c 2077 696c 6c20 6c65  on rate, will le
+00001610: 6176 6520 616e 2061 6765 2070 6561 6b20  ave an age peak 
+00001620: 6672 6f6d 2074 6865 2062 7572 7374 206f  from the burst o
+00001630: 6620 6765 6e65 2064 7570 6c69 6361 7465  f gene duplicate
+00001640: 7320 696e 2061 2073 686f 7274 2074 696d  s in a short tim
+00001650: 652d 6672 616d 6520 7570 6f6e 2074 6865  e-frame upon the
+00001660: 2069 6e69 7469 616c 2061 6765 2064 6973   initial age dis
+00001670: 7472 6962 7574 696f 6e2c 2061 6e64 2063  tribution, and c
+00001680: 616e 2062 6520 756e 7665 696c 6564 2066  an be unveiled f
+00001690: 726f 6d20 6d69 7874 7572 6520 6d6f 6465  rom mixture mode
+000016a0: 6c69 6e67 2061 6e61 6c79 7369 732e 2048  ling analysis. H
+000016b0: 6f77 6576 6572 2c20 5747 4473 2069 6465  owever, WGDs ide
+000016c0: 6e74 6966 6965 6420 6672 6f6d 2074 6865  ntified from the
+000016d0: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
+000016e0: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+000016f0: 6469 7374 7269 6275 7469 6f6e 7320 6361  distributions ca
+00001700: 6e20 6f6e 6c79 2069 6e66 6f72 6d20 7468  n only inform th
+00001710: 6520 5747 4420 7469 6d69 6e67 2069 6e20  e WGD timing in 
+00001720: 7468 6520 7469 6d65 2d73 6361 6c65 206f  the time-scale o
+00001730: 6620 7468 6174 2073 7065 6369 6669 6320  f that specific 
+00001740: 7370 6563 6965 732c 2077 6869 6368 2069  species, which i
+00001750: 7320 6e6f 7420 636f 6d70 6172 6162 6c65  s not comparable
+00001760: 2069 6e20 7468 6520 7068 796c 6f67 656e   in the phylogen
+00001770: 6574 6963 2063 6f6e 7465 7874 2e20 4f6e  etic context. On
+00001780: 6c79 2077 6974 6820 7468 6520 6f72 7468  ly with the orth
+00001790: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000017a0: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+000017b0: 7269 6275 7469 6f6e 732c 2077 6869 6368  ributions, which
+000017c0: 2063 6f6e 7665 7274 2074 6865 2065 7374   convert the est
+000017d0: 696d 6174 6564 2062 6f64 7920 6672 6f6d  imated body from
+000017e0: 2070 6172 616c 6f67 7565 7320 746f 206f   paralogues to o
+000017f0: 7274 686f 6c6f 6775 6573 2061 6e64 2069  rthologues and i
+00001800: 6e66 6f72 6d20 7468 6520 7265 6c61 7469  nform the relati
+00001810: 7665 2074 696d 696e 6720 6f66 2073 7065  ve timing of spe
+00001820: 6369 6174 696f 6e20 6576 656e 7473 2c20  ciation events, 
+00001830: 6361 6e20 7765 2064 6563 6970 6865 7220  can we decipher 
+00001840: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
+00001850: 2070 6c61 6365 6d65 6e74 206f 6620 5747   placement of WG
+00001860: 4473 2061 6674 6572 2070 726f 7065 7220  Ds after proper 
+00001870: 7261 7465 2063 6f72 7265 6374 696f 6e2e  rate correction.
+00001880: 2060 7767 6420 7632 6020 6973 2073 7563   `wgd v2` is suc
+00001890: 6820 7072 6f67 7261 6d20 7468 6174 2068  h program that h
+000018a0: 656c 7073 2075 7365 7273 2063 6f6e 7374  elps users const
+000018b0: 7275 6374 2070 6172 616c 6f67 6f75 7320  ruct paralogous 
+000018c0: 616e 6420 6f72 7468 6f6c 6f67 6f75 7320  and orthologous 
+000018d0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000018e0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+000018f0: 7320 616e 6420 7265 616c 697a 6520 626f  s and realize bo
+00001900: 7468 2074 6865 2069 6465 6e74 6966 6963  th the identific
+00001910: 6174 696f 6e20 616e 6420 706c 6163 656d  ation and placem
+00001920: 656e 7420 6f66 2057 4744 732e 0a0a 496e  ent of WGDs...In
+00001930: 2074 6865 2070 7265 6d69 7365 206f 6620   the premise of 
+00001940: 7068 796c 6f67 656e 6574 6963 616c 6c79  phylogenetically
+00001950: 206c 6f63 6174 6564 2057 4744 732c 2074   located WGDs, t
+00001960: 6865 2061 6273 6f6c 7574 6520 6167 6520  he absolute age 
+00001970: 2869 6e20 6765 6f6c 6f67 6963 616c 2074  (in geological t
+00001980: 696d 6529 206f 6620 5747 4473 2063 616e  ime) of WGDs can
+00001990: 2061 6c73 6f20 6265 2069 6e66 6572 7265   also be inferre
+000019a0: 6420 6672 6f6d 2074 686f 7365 2057 4744  d from those WGD
+000019b0: 2d72 6574 6169 6e65 6420 6765 6e65 2064  -retained gene d
+000019c0: 7570 6c69 6361 7465 732c 2061 6c74 686f  uplicates, altho
+000019d0: 7567 6820 7468 6572 6520 6861 7320 6265  ugh there has be
+000019e0: 656e 206e 6f20 6561 7379 206f 7220 7374  en no easy or st
+000019f0: 7261 6967 6874 666f 7277 6172 6420 7069  raightforward pi
+00001a00: 7065 6c69 6e65 2066 6f72 2074 6869 7320  peline for this 
+00001a10: 6a6f 6220 736f 2066 6172 2e20 496e 2060  job so far. In `
+00001a20: 7767 6420 7632 602c 2077 6520 6465 7665  wgd v2`, we deve
+00001a30: 6c6f 7065 6420 6120 6665 6173 6962 6c65  loped a feasible
+00001a40: 2069 6e74 6567 7261 7465 6420 7069 706c   integrated pipl
+00001a50: 656c 696e 6520 666f 7220 6162 736f 6c75  eline for absolu
+00001a60: 7465 2064 6174 696e 6720 6f66 2057 4744  te dating of WGD
+00001a70: 732e 2054 6865 2070 6970 656c 696e 6520  s. The pipeline 
+00001a80: 6361 6e20 6265 2072 6f75 6768 6c79 2064  can be roughly d
+00001a90: 6976 6964 6564 2069 6e74 6f20 7468 7265  ivided into thre
+00001aa0: 6520 6d61 696e 2073 7465 7073 2e20 3129  e main steps. 1)
+00001ab0: 2054 6865 2063 6f6e 7374 7275 6374 696f   The constructio
+00001ac0: 6e20 6f66 2061 6e63 686f 7220 2a4b 2a3c  n of anchor *K*<
+00001ad0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00001ae0: 7269 6275 7469 6f6e 2061 6e64 2074 6865  ribution and the
+00001af0: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
+00001b00: 6372 6564 6961 626c 6520 2a4b 2a3c 7375  crediable *K*<su
+00001b10: 623e 533c 2f73 7562 3e20 7261 6e67 6520  b>S</sub> range 
+00001b20: 6164 6f70 7465 6420 666f 7220 7068 796c  adopted for phyl
+00001b30: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
+00001b40: 2075 7369 6e67 2060 7767 6420 646d 6460   using `wgd dmd`
+00001b50: 2c20 6077 6764 206b 7364 602c 2060 7767  , `wgd ksd`, `wg
+00001b60: 6420 7379 6e60 2061 6e64 2060 7767 6420  d syn` and `wgd 
+00001b70: 7065 616b 602e 204e 6f74 6520 7468 6174  peak`. Note that
+00001b80: 2068 6572 6520 7765 206f 6e6c 7920 636f   here we only co
+00001b90: 6e73 6964 6572 2067 656e 6f6d 6520 6173  nsider genome as
+00001ba0: 7365 6d62 6c79 2062 6563 6175 7365 2066  sembly because f
+00001bb0: 6f72 2074 7261 6e73 6372 6970 746f 6d65  or transcriptome
+00001bc0: 2061 7373 656d 626c 7920 6974 2773 2069   assembly it's i
+00001bd0: 6d70 6f73 7369 626c 6520 746f 2064 6973  mpossible to dis
+00001be0: 7469 6e67 7569 7368 2057 4744 2d64 6572  tinguish WGD-der
+00001bf0: 6976 6564 2064 7570 6c69 6361 7465 7320  ived duplicates 
+00001c00: 6672 6f6d 2073 6d61 6c6c 2d73 6361 6c65  from small-scale
+00001c10: 2064 7570 6c69 6361 7469 6f6e 2d64 6572   duplication-der
+00001c20: 6976 6564 2064 7570 6c69 6361 7465 732c  ived duplicates,
+00001c30: 2077 6869 6368 2068 6170 7065 6e65 6420   which happened 
+00001c40: 696e 2061 2063 6f6e 7469 6e75 6f75 7320  in a continuous 
+00001c50: 7469 6d65 2d66 7261 6d65 2069 6e73 7465  time-frame inste
+00001c60: 6164 206f 6620 6f6e 6c79 2061 2073 6570  ad of only a sep
+00001c70: 6172 6174 6520 7368 6f72 7420 7469 6d65  arate short time
+00001c80: 2d66 7261 6d65 2061 6e64 2074 6875 7320  -frame and thus 
+00001c90: 7265 666c 6563 7473 2074 6865 2064 7572  reflects the dur
+00001ca0: 6174 696f 6e20 6f66 2074 6861 7420 6272  ation of that br
+00001cb0: 616e 6368 2072 6174 6865 7220 7468 616e  anch rather than
+00001cc0: 2074 6865 2074 696d 6520 6174 2077 6869   the time at whi
+00001cd0: 6368 2057 4744 206f 6363 7572 7265 642e  ch WGD occurred.
+00001ce0: 2032 2920 5468 6520 666f 726d 756c 6174   2) The formulat
+00001cf0: 696f 6e20 6f66 2061 2073 7461 7274 696e  ion of a startin
+00001d00: 6720 7472 6565 2075 7365 6420 696e 2074  g tree used in t
+00001d10: 6865 2070 6879 6c6f 6765 6e65 7469 6320  he phylogenetic 
+00001d20: 6461 7469 6e67 2c20 636f 6d70 6f73 6564  dating, composed
+00001d30: 206f 6620 6120 6665 7720 7370 6563 6965   of a few specie
+00001d40: 7320 616e 6420 616e 6e6f 7461 7465 6420  s and annotated 
+00001d50: 7769 7468 2066 6f73 7369 6c20 6361 6c69  with fossil cali
+00001d60: 6272 6174 696f 6e20 696e 666f 726d 6174  bration informat
+00001d70: 696f 6e2e 2054 6869 7320 7374 6570 2069  ion. This step i
+00001d80: 7320 6573 7365 6e74 6961 6c20 666f 7220  s essential for 
+00001d90: 7468 6520 7265 7375 6c74 206f 6620 6162  the result of ab
+00001da0: 736f 6c75 7465 2057 4744 2064 6174 696e  solute WGD datin
+00001db0: 6720 7468 6174 2077 6520 7375 6767 6573  g that we sugges
+00001dc0: 7420 7573 6572 7320 6f66 2074 616b 696e  t users of takin
+00001dd0: 6720 6772 6561 7420 6361 7574 696f 6e20  g great caution 
+00001de0: 746f 2061 7373 7572 6520 7468 6520 7472  to assure the tr
+00001df0: 6565 2074 6f70 6f6c 6f67 7920 616e 6420  ee topology and 
+00001e00: 7072 6f70 6572 2062 6f75 6e64 7320 666f  proper bounds fo
+00001e10: 7220 666f 7373 696c 2063 616c 6962 7261  r fossil calibra
+00001e20: 7469 6f6e 732e 2033 2920 5468 6520 636f  tions. 3) The co
+00001e30: 6e73 7472 7563 7469 6f6e 206f 6620 6f72  nstruction of or
+00001e40: 7468 6f67 726f 7570 7320 636f 6e73 6973  thogroups consis
+00001e50: 7469 6e67 206f 6620 636f 6c6c 696e 6561  ting of collinea
+00001e60: 7220 6475 706c 6963 6174 6573 206f 6620  r duplicates of 
+00001e70: 7468 6520 666f 6361 6c20 7370 6563 6965  the focal specie
+00001e80: 7320 616e 6420 7468 6569 7220 7265 6369  s and their reci
+00001e90: 7072 6f63 616c 2062 6573 7420 6869 7473  procal best hits
+00001ea0: 2028 5242 4873 2920 6167 6169 6e73 7420   (RBHs) against 
+00001eb0: 6f74 6865 7220 7370 6563 6965 7320 696e  other species in
+00001ec0: 2074 6865 2073 7461 7274 696e 6720 7472   the starting tr
+00001ed0: 6565 2061 6e64 2074 6865 2070 6879 6c6f  ee and the phylo
+00001ee0: 6765 6e65 7469 6320 6461 7469 6e67 2075  genetic dating u
+00001ef0: 7369 6e67 2061 206d 6f6c 6563 756c 6172  sing a molecular
+00001f00: 2064 6174 696e 6720 7072 6f67 7261 6d20   dating program 
+00001f10: 666f 7220 696e 7374 616e 6365 2060 6d63  for instance `mc
+00001f20: 6d63 7472 6565 602c 2076 6961 2070 726f  mctree`, via pro
+00001f30: 6772 616d 2060 7767 6420 646d 6460 2061  gram `wgd dmd` a
+00001f40: 6e64 2060 7767 6420 666f 6375 7360 2e20  nd `wgd focus`. 
+00001f50: 5765 2072 6563 6f6d 6d65 6e64 2074 6865  We recommend the
+00001f60: 2075 7361 6765 206f 6620 4261 7965 7369   usage of Bayesi
+00001f70: 616e 206d 6f6c 6563 756c 6172 2064 6174  an molecular dat
+00001f80: 696e 6720 7072 6f67 7261 6d20 606d 636d  ing program `mcm
+00001f90: 6374 7265 6560 2c20 7768 6963 6820 7072  ctree`, which pr
+00001fa0: 6f76 6964 6573 2061 2076 6172 6965 7479  ovides a variety
+00001fb0: 206f 6620 6469 6666 6572 656e 7420 7375   of different su
+00001fc0: 6273 7469 7475 7469 6f6e 2061 6e64 2072  bstitution and r
+00001fd0: 6174 6520 6d6f 6465 6c73 2e20 4e6f 6e65  ate models. None
+00001fe0: 7468 656c 6573 732c 2077 6520 7572 6765  theless, we urge
+00001ff0: 2075 7365 7273 2074 6f20 7365 7420 7072   users to set pr
+00002000: 696f 7220 6469 7374 7269 6275 7469 6f6e  ior distribution
+00002010: 206f 6620 6469 6666 6572 656e 7420 7061   of different pa
+00002020: 7261 6d65 7465 7273 2077 6974 6820 6361  rameters with ca
+00002030: 7574 696f 6e20 616e 6420 6173 7375 7265  ution and assure
+00002040: 2061 6465 7175 6174 6520 7361 6d70 6c69   adequate sampli
+00002050: 6e67 206f 6620 6469 6666 6572 656e 7420  ng of different 
+00002060: 7061 7261 6d65 7465 7273 2e0a 0a23 2320  parameters...## 
+00002070: 496e 7374 616c 6c61 7469 6f6e 0a0a 5468  Installation..Th
+00002080: 6520 6561 7369 6573 7420 7761 7920 746f  e easiest way to
+00002090: 2069 6e73 7461 6c6c 2060 7767 6420 7632   install `wgd v2
+000020a0: 6020 6973 2075 7369 6e67 2060 5059 5049  ` is using `PYPI
+000020b0: 602e 204e 6f74 6520 7468 6174 2069 6620  `. Note that if 
+000020c0: 796f 7520 7761 6e74 2074 6f20 6765 7420  you want to get 
+000020d0: 7468 6520 6c61 7465 7374 2075 7064 6174  the latest updat
+000020e0: 652c 2077 6520 7375 6767 6573 7420 696e  e, we suggest in
+000020f0: 7374 616c 6c69 6e67 2066 726f 6d20 7468  stalling from th
+00002100: 6520 736f 7572 6365 2c20 7369 6e63 6520  e source, since 
+00002110: 7468 6520 7570 6461 7465 206f 6e20 6050  the update on `P
+00002120: 5950 4960 2077 696c 6c20 6265 2064 656c  YPI` will be del
+00002130: 6179 6564 2063 6f6d 7061 7265 6420 746f  ayed compared to
+00002140: 2068 6572 6520 6f66 2073 6f75 7263 652e   here of source.
+00002150: 0a0a 6060 600a 7069 7020 696e 7374 616c  ..```.pip instal
+00002160: 6c20 7767 640a 6060 600a 0a54 6f20 696e  l wgd.```..To in
+00002170: 7374 616c 6c20 6077 6764 2076 3260 2069  stall `wgd v2` i
+00002180: 6e20 6120 7669 7274 7561 6c20 656e 7669  n a virtual envi
+00002190: 726f 6e6d 656e 7420 2877 6869 6368 2077  ronment (which w
+000021a0: 6520 7374 726f 6e67 6c79 2072 6563 6f6d  e strongly recom
+000021b0: 6d65 6e64 2c20 6e6f 206d 6174 7465 7220  mend, no matter 
+000021c0: 796f 7520 696e 7374 616c 6c20 6672 6f6d  you install from
+000021d0: 2073 6f75 7263 652c 2060 5059 5049 6020   source, `PYPI` 
+000021e0: 6f72 2060 6269 6f63 6f6e 6461 6029 2c20  or `bioconda`), 
+000021f0: 7468 6520 666f 6c6c 6f77 696e 6720 636f  the following co
+00002200: 6d6d 616e 6420 6c69 6e65 7320 636f 756c  mmand lines coul
+00002210: 6420 6265 2075 7365 642e 0a0a 6060 600a  d be used...```.
+00002220: 6769 7420 636c 6f6e 6520 6874 7470 733a  git clone https:
+00002230: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
+00002240: 6865 2d70 7362 2f77 6764 0a63 6420 7767  he-psb/wgd.cd wg
+00002250: 640a 7669 7274 7561 6c65 6e76 202d 703d  d.virtualenv -p=
+00002260: 7079 7468 6f6e 3320 454e 5620 286f 7220  python3 ENV (or 
+00002270: 7079 7468 6f6e 3320 2d6d 2076 656e 7620  python3 -m venv 
+00002280: 454e 5629 0a73 6f75 7263 6520 454e 562f  ENV).source ENV/
+00002290: 6269 6e2f 6163 7469 7661 7465 0a70 6970  bin/activate.pip
+000022a0: 2069 6e73 7461 6c6c 202d 7220 7265 7175   install -r requ
+000022b0: 6972 656d 656e 7473 2e74 7874 0a70 6970  irements.txt.pip
+000022c0: 2069 6e73 7461 6c6c 202e 0a60 6060 0a0a   install ..```..
+000022d0: 5768 656e 206d 6574 2077 6974 6820 7065  When met with pe
+000022e0: 726d 6973 7369 6f6e 2070 726f 626c 656d  rmission problem
+000022f0: 2069 6e20 696e 7374 616c 6c61 7469 6f6e   in installation
+00002300: 2c20 706c 6561 7365 2074 7279 2074 6865  , please try the
+00002310: 2066 6f6c 6c6f 7769 6e67 2063 6f6d 6d61   following comma
+00002320: 6e64 206c 696e 652e 0a0a 6060 600a 7069  nd line...```.pi
+00002330: 7020 696e 7374 616c 6c20 2d65 202e 0a60  p install -e ..`
+00002340: 6060 0a0a 4966 206d 756c 7469 706c 7920  ``..If multiply 
+00002350: 7665 7273 696f 6e73 206f 6620 6077 6764  versions of `wgd
+00002360: 6020 7765 7265 2069 6e73 7461 6c6c 6564  ` were installed
+00002370: 2069 6e20 7468 6520 7379 7374 656d 2c20   in the system, 
+00002380: 706c 6561 7365 2061 6464 2074 6865 2072  please add the r
+00002390: 6967 6874 2070 6174 6820 6f66 2069 6e74  ight path of int
+000023a0: 6572 6573 7465 6420 7665 7273 696f 6e20  erested version 
+000023b0: 696e 746f 2074 6865 2065 6e76 6972 6f6e  into the environ
+000023c0: 6d65 6e74 2076 6172 6961 626c 6573 2c20  ment variables, 
+000023d0: 666f 7220 6578 616d 706c 650a 0a60 6060  for example..```
+000023e0: 0a65 7870 6f72 7420 5041 5448 3d22 2450  .export PATH="$P
+000023f0: 4154 483a 7e2f 2e6c 6f63 616c 2f62 696e  ATH:~/.local/bin
+00002400: 2f77 6764 220a 6060 600a 0a4e 6f74 6520  /wgd".```..Note 
+00002410: 7468 6174 2074 6865 2076 6572 7369 6f6e  that the version
+00002420: 206f 6620 606e 756d 7079 6020 6973 2069   of `numpy` is i
+00002430: 6d70 6f72 7461 6e74 2028 666f 7220 6d61  mportant (for ma
+00002440: 6e79 206f 7468 6572 2070 6163 6b61 6765  ny other package
+00002450: 7320 6172 6520 7468 6520 7361 6d65 206f  s are the same o
+00002460: 6620 636f 7572 7365 292c 2065 7370 6563  f course), espec
+00002470: 6961 6c6c 7920 666f 7220 6066 6173 7463  ially for `fastc
+00002480: 6c75 7374 6572 6020 7061 636b 6167 652e  luster` package.
+00002490: 2049 6e20 6f75 7220 7465 7374 2c20 7468   In our test, th
+000024a0: 6520 606e 756d 7079 6020 312e 3139 2e30  e `numpy` 1.19.0
+000024b0: 2077 6f72 6b73 2066 696e 6520 6f6e 2060   works fine on `
+000024c0: 7079 7468 6f6e 332e 362f 3860 2e20 4966  python3.6/8`. If
+000024d0: 2079 6f75 206d 6574 2073 6f6d 6520 6572   you met some er
+000024e0: 726f 7273 206f 7220 7761 726e 696e 6773  rors or warnings
+000024f0: 2061 626f 7574 2060 6e75 6d70 7960 2c20   about `numpy`, 
+00002500: 6d61 7962 6520 636f 6e73 6964 6572 696e  maybe considerin
+00002510: 6720 7072 652d 696e 7374 616c 6c20 606e  g pre-install `n
+00002520: 756d 7079 6020 6173 2031 2e31 392e 3020  umpy` as 1.19.0 
+00002530: 6f72 206f 7468 6572 2063 6c6f 7365 2d62  or other close-b
+00002540: 7920 7665 7273 696f 6e73 2062 6566 6f72  y versions befor
+00002550: 6520 796f 7520 696e 7374 616c 6c20 6077  e you install `w
+00002560: 6764 602e 2060 7767 6460 2072 656c 6965  gd`. `wgd` relie
+00002570: 7320 6f6e 2065 7874 6572 6e61 6c20 736f  s on external so
+00002580: 6674 7761 7265 7320 696e 636c 7564 696e  ftwares includin
+00002590: 6720 6064 6961 6d6f 6e64 6020 616e 6420  g `diamond` and 
+000025a0: 606d 636c 6020 666f 7220 6077 6764 2064  `mcl` for `wgd d
+000025b0: 6d64 602c 2060 7061 6d6c 2076 342e 396a  md`, `paml v4.9j
+000025c0: 602c 2060 6d61 6666 7460 2028 606d 7573  `, `mafft` (`mus
+000025d0: 636c 6560 2061 6e64 2060 7072 616e 6b60  cle` and `prank`
+000025e0: 2069 6620 7365 7429 2c20 6066 6173 7474   if set), `fastt
+000025f0: 7265 6560 2028 6f72 2060 6971 7472 6565  ree` (or `iqtree
+00002600: 6020 6966 2073 6574 2920 666f 7220 6077  ` if set) for `w
+00002610: 6764 206b 7364 6020 616e 6420 6f70 7469  gd ksd` and opti
+00002620: 6f6e 616c 6c79 2060 6d72 6261 7965 7360  onally `mrbayes`
+00002630: 2066 6f72 2074 6865 2070 6879 6c6f 6765   for the phyloge
+00002640: 6e65 7469 6320 696e 6665 7265 6e63 6520  netic inference 
+00002650: 6675 6e63 7469 6f6e 2069 6e20 6077 6764  function in `wgd
+00002660: 2064 6d64 6020 616e 6420 6077 6764 2066   dmd` and `wgd f
+00002670: 6f63 7573 6020 2860 6d61 6666 7460 2c20  ocus` (`mafft`, 
+00002680: 606d 7573 636c 6560 2061 6e64 2060 7072  `muscle` and `pr
+00002690: 616e 6b60 2061 7320 7765 6c6c 2077 6865  ank` as well whe
+000026a0: 6e20 7468 6520 616e 616c 7973 6973 2072  n the analysis r
+000026b0: 6571 7569 7265 7320 7365 7175 656e 6365  equires sequence
+000026c0: 2061 6c69 676e 6d65 6e74 292e 2053 6f6d   alignment). Som
+000026d0: 6520 6f74 6865 7220 6f70 7469 6f6e 616c  e other optional
+000026e0: 2073 6f66 7477 6172 6573 2069 6e63 6c75   softwares inclu
+000026f0: 6469 6e67 2060 7061 6d6c 2076 342e 396a  ding `paml v4.9j
+00002700: 602c 2060 7238 7360 2c20 6062 6561 7374  `, `r8s`, `beast
+00002710: 602c 2060 6567 676e 6f67 602c 2060 6469  `, `eggnog`, `di
+00002720: 616d 6f6e 6460 2c20 6069 6e74 6572 7072  amond`, `interpr
+00002730: 6f73 6361 6e60 2c20 6068 6d6d 6572 2076  oscan`, `hmmer v
+00002740: 332e 3162 3260 2061 6e64 2060 6173 7472  3.1b2` and `astr
+00002750: 616c 2d70 726f 6020 2860 686d 6d65 7220  al-pro` (`hmmer 
+00002760: 7633 2e31 6232 6020 6973 2061 6c73 6f20  v3.1b2` is also 
+00002770: 666f 7220 7468 6520 5b6f 7274 686f 6772  for the [orthogr
+00002780: 6f75 7020 6173 7369 676e 6d65 6e74 5d28  oup assignment](
+00002790: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
+000027a0: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
+000027b0: 6c61 7465 7374 2f75 7361 6765 2e68 746d  latest/usage.htm
+000027c0: 6c23 6f72 7468 6f67 726f 7570 2d61 7373  l#orthogroup-ass
+000027d0: 6967 6e6d 656e 7429 2066 756e 6374 696f  ignment) functio
+000027e0: 6e20 696e 2060 7767 6420 646d 6460 2061  n in `wgd dmd` a
+000027f0: 6e64 2060 6173 7472 616c 2d70 726f 6020  nd `astral-pro` 
+00002800: 6973 2061 6c73 6f20 666f 7220 7468 6520  is also for the 
+00002810: 6675 6e63 7469 6f6e 2060 2d2d 636f 6c6c  function `--coll
+00002820: 696e 6561 7263 6f61 6c65 7363 656e 6365  inearcoalescence
+00002830: 6020 696e 2060 7767 6420 646d 6460 2920  ` in `wgd dmd`) 
+00002840: 6172 6520 666f 7220 7468 6520 6d6f 6c65  are for the mole
+00002850: 6375 6c61 7220 6461 7469 6e67 2c20 6765  cular dating, ge
+00002860: 6e65 2066 616d 696c 7920 6675 6e63 7469  ne family functi
+00002870: 6f6e 2061 6e6e 6f74 6174 696f 6e20 6f72  on annotation or
+00002880: 2070 6879 6c6f 6765 6e65 7469 6320 696e   phylogenetic in
+00002890: 6665 7265 6e63 6520 696e 2060 7767 6420  ference in `wgd 
+000028a0: 666f 6375 7360 2e0a 0a23 2320 5069 7065  focus`...## Pipe
+000028b0: 6c69 6e65 730a 0a54 6f20 7175 6963 6b6c  lines..To quickl
+000028c0: 7920 6765 7420 6661 6d69 6c69 6172 2077  y get familiar w
+000028d0: 6974 6820 6077 6764 2076 3260 2c20 7765  ith `wgd v2`, we
+000028e0: 2070 726f 7669 6465 6420 736f 6d65 2063   provided some c
+000028f0: 6f6d 6d6f 6e20 7069 7065 6c69 6e65 7320  ommon pipelines 
+00002900: 666f 7220 7573 6572 7320 7769 7468 2061  for users with a
+00002910: 2066 7265 7368 2067 656e 6f6d 6520 6173   fresh genome as
+00002920: 7365 6d62 6c79 2069 6e20 6861 6e64 2e0a  sembly in hand..
+00002930: 0a23 2323 2050 6970 656c 696e 6520 310a  .### Pipeline 1.
+00002940: 2a2a 5369 6d70 6c65 2063 6f6e 7374 7275  **Simple constru
+00002950: 6374 696f 6e20 6f66 2061 6765 2064 6973  ction of age dis
+00002960: 7472 6962 7574 696f 6e2a 2a0a 6060 600a  tribution**.```.
+00002970: 7767 6420 646d 6420 4171 7569 6c65 6769  wgd dmd Aquilegi
+00002980: 615f 636f 6572 756c 6561 202d 6f20 7767  a_coerulea -o wg
+00002990: 645f 646d 640a 7767 6420 6b73 6420 7767  d_dmd.wgd ksd wg
+000029a0: 645f 646d 642f 4171 7569 6c65 6769 615f  d_dmd/Aquilegia_
+000029b0: 636f 6572 756c 6561 2e74 7376 2041 7175  coerulea.tsv Aqu
+000029c0: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+000029d0: 2d6f 2077 6764 5f6b 7364 0a60 6060 0a60  -o wgd_ksd.```.`
+000029e0: 6060 0a23 7768 6174 2077 696c 6c20 6265  ``.#what will be
+000029f0: 2069 6e20 7468 6520 7265 7375 6c74 2064   in the result d
+00002a00: 6972 6563 746f 7279 0a2d 7767 645f 646d  irectory.-wgd_dm
+00002a10: 640a 2d2d 4171 7569 6c65 6769 615f 636f  d.--Aquilegia_co
+00002a20: 6572 756c 6561 2e74 7376 0a2d 7767 645f  erulea.tsv.-wgd_
+00002a30: 6b73 640a 2d2d 4171 7569 6c65 6769 615f  ksd.--Aquilegia_
+00002a40: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00002a50: 7473 7620 4171 7569 6c65 6769 615f 636f  tsv Aquilegia_co
+00002a60: 6572 756c 6561 2e74 7376 2e6b 732e 7376  erulea.tsv.ks.sv
+00002a70: 672f 7064 660a 6060 600a 5468 6520 7265  g/pdf.```.The re
+00002a80: 7375 6c74 696e 6720 6041 7175 696c 6567  sulting `Aquileg
+00002a90: 6961 5f63 6f65 7275 6c65 612e 7473 7660  ia_coerulea.tsv`
+00002aa0: 2069 7320 7468 6520 7768 6f6c 6520 7061   is the whole pa
+00002ab0: 7261 6e6f 6d65 2066 616d 696c 7920 6669  ranome family fi
+00002ac0: 6c65 2c20 6041 7175 696c 6567 6961 5f63  le, `Aquilegia_c
+00002ad0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00002ae0: 7376 6020 6669 6c65 2069 7320 7468 6520  sv` file is the 
+00002af0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00002b00: 6469 7374 7269 6275 6974 6f6e 2066 696c  distribuiton fil
+00002b10: 652c 2060 4171 7569 6c65 6769 615f 636f  e, `Aquilegia_co
+00002b20: 6572 756c 6561 2e74 7376 2e6b 732e 7376  erulea.tsv.ks.sv
+00002b30: 6760 2061 6e64 2060 4171 7569 6c65 6769  g` and `Aquilegi
+00002b40: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00002b50: 732e 7064 6660 2061 7265 2074 6865 202a  s.pdf` are the *
+00002b60: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
+00002b70: 6c6f 7473 2e0a 0a23 2323 2050 6970 656c  lots...### Pipel
+00002b80: 696e 6520 320a 2a2a 5369 6d70 6c65 2063  ine 2.**Simple c
+00002b90: 6f6e 7374 7275 6374 696f 6e20 6f66 2061  onstruction of a
+00002ba0: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00002bb0: 7769 7468 2063 6f6c 6c69 6e65 6172 6974  with collinearit
+00002bc0: 792a 2a0a 6060 600a 7767 6420 646d 6420  y**.```.wgd dmd 
+00002bd0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00002be0: 6561 202d 6f20 7767 645f 646d 640a 7767  ea -o wgd_dmd.wg
+00002bf0: 6420 6b73 6420 7767 645f 646d 642f 4171  d ksd wgd_dmd/Aq
+00002c00: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00002c10: 2e74 7376 2041 7175 696c 6567 6961 5f63  .tsv Aquilegia_c
+00002c20: 6f65 7275 6c65 6120 2d6f 2077 6764 5f6b  oerulea -o wgd_k
+00002c30: 7364 0a77 6764 2073 796e 202d 6620 6d52  sd.wgd syn -f mR
+00002c40: 4e41 202d 6120 4e61 6d65 2077 6764 5f64  NA -a Name wgd_d
+00002c50: 6d64 2f41 7175 696c 6567 6961 5f63 6f65  md/Aquilegia_coe
+00002c60: 7275 6c65 612e 7473 7620 4171 7569 6c65  rulea.tsv Aquile
+00002c70: 6769 615f 636f 6572 756c 6561 2e67 6666  gia_coerulea.gff
+00002c80: 3320 2d6b 7320 7767 645f 6b73 642f 4171  3 -ks wgd_ksd/Aq
+00002c90: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00002ca0: 2e74 7376 2e6b 732e 7473 7620 2d6f 2077  .tsv.ks.tsv -o w
+00002cb0: 6764 5f73 796e 0a60 6060 0a60 6060 0a23  gd_syn.```.```.#
+00002cc0: 2077 6861 7420 7769 6c6c 2062 6520 696e   what will be in
+00002cd0: 2074 6865 2077 6764 5f73 796e 2072 6573   the wgd_syn res
+00002ce0: 756c 7420 6469 7265 6374 6f72 790a 2d77  ult directory.-w
+00002cf0: 6764 5f73 796e 0a2d 2d61 6e63 686f 7273  gd_syn.--anchors
+00002d00: 2e63 7376 0a2d 2d66 616d 696c 6965 732e  .csv.--families.
+00002d10: 7473 760a 2d2d 6765 6e65 2d74 6162 6c65  tsv.--gene-table
+00002d20: 2e63 7376 0a2d 2d69 6164 686f 7265 2e63  .csv.--iadhore.c
+00002d30: 6f6e 660a 2d2d 7363 6166 666f 6c64 5f6c  onf.--scaffold_l
+00002d40: 656e 6774 682e 7473 760a 2d2d 7365 676d  ength.tsv.--segm
+00002d50: 656e 7473 5f63 6f6f 7264 696e 6174 6573  ents_coordinates
+00002d60: 2e74 7376 0a2d 2d53 6567 7072 6f66 696c  .tsv.--Segprofil
+00002d70: 652e 6373 760a 2d2d 5379 6e64 6570 7468  e.csv.--Syndepth
+00002d80: 2e70 6466 2f73 7667 0a2d 2d41 7175 696c  .pdf/svg.--Aquil
+00002d90: 6567 6961 5f63 6f65 7275 6c65 615f 4171  egia_coerulea_Aq
+00002da0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00002db0: 5f6d 756c 7469 706c 6963 6f6e 735f 6c65  _multiplicons_le
+00002dc0: 7665 6c2e 7064 662f 706e 672f 7376 670a  vel.pdf/png/svg.
+00002dd0: 2d2d 4171 7569 6c65 6769 615f 636f 6572  --Aquilegia_coer
+00002de0: 756c 6561 5f67 656e 655f 6f72 6465 725f  ulea_gene_order_
+00002df0: 7065 7263 6872 6f6d 2e74 7376 0a2d 2d41  perchrom.tsv.--A
+00002e00: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00002e10: 612e 7473 762e 616e 6368 6f72 732e 6b73  a.tsv.anchors.ks
+00002e20: 2e74 7376 0a2d 2d41 7175 696c 6567 6961  .tsv.--Aquilegia
+00002e30: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00002e40: 642e 7064 662f 7376 670a 2d2d 4171 7569  d.pdf/svg.--Aqui
+00002e50: 6c65 6769 615f 636f 6572 756c 6561 2d76  legia_coerulea-v
+00002e60: 732d 4171 7569 6c65 6769 615f 636f 6572  s-Aquilegia_coer
+00002e70: 756c 6561 2e64 6f74 2e70 6466 2f70 6e67  ulea.dot.pdf/png
+00002e80: 2f73 7667 0a2d 2d41 7175 696c 6567 6961  /svg.--Aquilegia
+00002e90: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
+00002ea0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00002eb0: 646f 745f 756e 6974 5f67 656e 652e 7064  dot_unit_gene.pd
+00002ec0: 662f 706e 672f 7376 670a 2d2d 4171 7569  f/png/svg.--Aqui
+00002ed0: 6c65 6769 615f 636f 6572 756c 6561 2d76  legia_coerulea-v
+00002ee0: 732d 4171 7569 6c65 6769 615f 636f 6572  s-Aquilegia_coer
+00002ef0: 756c 6561 5f4b 732e 646f 742e 7064 662f  ulea_Ks.dot.pdf/
+00002f00: 706e 672f 7376 670a 2d2d 4171 7569 6c65  png/svg.--Aquile
+00002f10: 6769 615f 636f 6572 756c 6561 2d76 732d  gia_coerulea-vs-
+00002f20: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00002f30: 6561 5f4b 732e 646f 745f 756e 6974 5f67  ea_Ks.dot_unit_g
+00002f40: 656e 652e 7064 662f 706e 672f 7376 670a  ene.pdf/png/svg.
+00002f50: 2d2d 6961 6468 6f72 652d 6f75 740a 2d2d  --iadhore-out.--
+00002f60: 2d61 6c69 676e 6d65 6e74 2e74 7874 0a2d  -alignment.txt.-
+00002f70: 2d2d 616e 6368 6f72 706f 696e 7473 2e74  --anchorpoints.t
+00002f80: 7874 0a2d 2d2d 6261 7365 636c 7573 7465  xt.---basecluste
+00002f90: 7273 2e74 7874 0a2d 2d2d 6765 6e65 732e  rs.txt.---genes.
+00002fa0: 7478 740a 2d2d 2d6c 6973 745f 656c 656d  txt.---list_elem
+00002fb0: 656e 7473 2e74 7874 0a2d 2d2d 6d75 6c74  ents.txt.---mult
+00002fc0: 6970 6c69 636f 6e5f 7061 6972 732e 7478  iplicon_pairs.tx
+00002fd0: 740a 2d2d 2d6d 756c 7469 706c 6963 6f6e  t.---multiplicon
+00002fe0: 732e 7478 740a 2d2d 2d73 6567 6d65 6e74  s.txt.---segment
+00002ff0: 732e 7478 740a 6060 600a 5468 6520 6061  s.txt.```.The `a
+00003000: 6e63 686f 7273 2e63 7376 602c 2060 6661  nchors.csv`, `fa
+00003010: 6d69 6c69 6573 2e74 7376 602c 2060 6765  milies.tsv`, `ge
+00003020: 6e65 2d74 6162 6c65 2e63 7376 602c 2060  ne-table.csv`, `
+00003030: 7363 6166 666f 6c64 5f6c 656e 6774 682e  scaffold_length.
+00003040: 7473 7660 2c20 6073 6567 6d65 6e74 735f  tsv`, `segments_
+00003050: 636f 6f72 6469 6e61 7465 732e 7473 7660  coordinates.tsv`
+00003060: 2c20 6053 6567 7072 6f66 696c 652e 6373  , `Segprofile.cs
+00003070: 7660 2c20 6041 7175 696c 6567 6961 5f63  v`, `Aquilegia_c
+00003080: 6f65 7275 6c65 615f 6765 6e65 5f6f 7264  oerulea_gene_ord
+00003090: 6572 5f70 6572 6368 726f 6d2e 7473 7660  er_perchrom.tsv`
+000030a0: 2061 7265 2074 6865 2062 6173 6963 2066   are the basic f
+000030b0: 696c 6573 2073 756d 6d61 7269 7a69 6e67  iles summarizing
+000030c0: 2074 6865 2067 656e 6520 6f72 6465 722c   the gene order,
+000030d0: 2066 616d 696c 792c 2061 6e63 686f 7273   family, anchors
+000030e0: 2061 6e64 2063 6f6c 6c69 6e65 6172 2073   and collinear s
+000030f0: 6567 6d65 6e74 732e 0a54 6865 2060 4171  egments..The `Aq
+00003100: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00003110: 2e74 7376 2e61 6e63 686f 7273 2e6b 732e  .tsv.anchors.ks.
+00003120: 7473 7660 2069 7320 7468 6520 2a4b 2a3c  tsv` is the *K*<
+00003130: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00003140: 7269 6275 7469 6f6e 2066 696c 6520 6f66  ribution file of
+00003150: 2061 6e63 686f 7220 7061 6972 732e 0a54   anchor pairs..T
+00003160: 6865 2060 5379 6e64 6570 7468 2e70 6466  he `Syndepth.pdf
+00003170: 2f73 7667 6020 7368 6f77 7320 7468 6520  /svg` shows the 
+00003180: 636f 6c6c 696e 6561 7220 7261 7469 6f20  collinear ratio 
+00003190: 6f66 206d 756c 7469 706c 6963 6f6e 732e  of multiplicons.
+000031a0: 0a54 6865 2060 4171 7569 6c65 6769 615f  .The `Aquilegia_
+000031b0: 636f 6572 756c 6561 5f41 7175 696c 6567  coerulea_Aquileg
+000031c0: 6961 5f63 6f65 7275 6c65 615f 6d75 6c74  ia_coerulea_mult
+000031d0: 6970 6c69 636f 6e73 5f6c 6576 656c 2e70  iplicons_level.p
+000031e0: 6466 2f70 6e67 2f73 7667 6020 6973 2074  df/png/svg` is t
+000031f0: 6865 2022 6475 7053 7461 636b 2220 706c  he "dupStack" pl
+00003200: 6f74 2073 686f 7769 6e67 206d 756c 7469  ot showing multi
+00003210: 706c 6963 6f6e 7320 7769 7468 2064 6966  plicons with dif
+00003220: 6665 7265 6e74 206c 6576 656c 7320 2864  ferent levels (d
+00003230: 6566 696e 6564 2061 7320 7468 6520 6e75  efined as the nu
+00003240: 6d62 6572 206f 6620 7365 676d 656e 7473  mber of segments
+00003250: 2077 6974 6869 6e29 2e0a 5468 6520 6041   within)..The `A
+00003260: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003270: 612e 7473 762e 6b73 642e 7064 662f 7376  a.tsv.ksd.pdf/sv
+00003280: 6760 2069 7320 7468 6520 2a4b 2a3c 7375  g` is the *K*<su
+00003290: 623e 533c 2f73 7562 3e20 706c 6f74 2077  b>S</sub> plot w
+000032a0: 6974 6820 616e 6368 6f72 2070 6169 7273  ith anchor pairs
+000032b0: 2061 6e6e 6f74 6174 6564 2e0a 5468 6520   annotated..The 
+000032c0: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
+000032d0: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
+000032e0: 5f63 6f65 7275 6c65 612e 646f 742e 7064  _coerulea.dot.pd
+000032f0: 662f 706e 672f 7376 6760 2069 7320 7468  f/png/svg` is th
+00003300: 6520 646f 7420 706c 6f74 2077 6974 686f  e dot plot witho
+00003310: 7574 202a 4b2a 3c73 7562 3e53 3c2f 7375  ut *K*<sub>S</su
+00003320: 623e 2061 6e6e 6f74 6174 696f 6e20 616e  b> annotation an
+00003330: 6420 7769 7468 2063 6f6f 7264 696e 6174  d with coordinat
+00003340: 6573 2069 6e20 7468 6520 6e75 6d62 6572  es in the number
+00003350: 206f 6620 6261 7365 2e0a 5468 6520 6041   of base..The `A
+00003360: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003370: 612d 7673 2d41 7175 696c 6567 6961 5f63  a-vs-Aquilegia_c
+00003380: 6f65 7275 6c65 612e 646f 745f 756e 6974  oerulea.dot_unit
+00003390: 5f67 656e 652e 7064 662f 706e 672f 7376  _gene.pdf/png/sv
+000033a0: 6760 2069 7320 7468 6520 646f 7420 706c  g` is the dot pl
+000033b0: 6f74 2077 6974 686f 7574 202a 4b2a 3c73  ot without *K*<s
+000033c0: 7562 3e53 3c2f 7375 623e 2061 6e6e 6f74  ub>S</sub> annot
+000033d0: 6174 696f 6e20 616e 6420 7769 7468 2063  ation and with c
+000033e0: 6f6f 7264 696e 6174 6573 2069 6e20 7468  oordinates in th
+000033f0: 6520 6e75 6d62 6572 206f 6620 6765 6e65  e number of gene
+00003400: 2e0a 5468 6520 6041 7175 696c 6567 6961  ..The `Aquilegia
+00003410: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
+00003420: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+00003430: 4b73 2e64 6f74 2e70 6466 2f70 6e67 2f73  Ks.dot.pdf/png/s
+00003440: 7667 6020 6973 2074 6865 2064 6f74 2070  vg` is the dot p
+00003450: 6c6f 7420 7769 7468 202a 4b2a 3c73 7562  lot with *K*<sub
+00003460: 3e53 3c2f 7375 623e 2061 6e6e 6f74 6174  >S</sub> annotat
+00003470: 696f 6e20 616e 6420 7769 7468 2063 6f6f  ion and with coo
+00003480: 7264 696e 6174 6573 2069 6e20 7468 6520  rdinates in the 
+00003490: 6e75 6d62 6572 206f 6620 6261 7365 2e0a  number of base..
+000034a0: 5468 6520 6041 7175 696c 6567 6961 5f63  The `Aquilegia_c
+000034b0: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
+000034c0: 6567 6961 5f63 6f65 7275 6c65 615f 4b73  egia_coerulea_Ks
+000034d0: 2e64 6f74 5f75 6e69 745f 6765 6e65 2e70  .dot_unit_gene.p
+000034e0: 6466 2f70 6e67 2f73 7667 6020 6973 2074  df/png/svg` is t
+000034f0: 6865 2064 6f74 2070 6c6f 7420 7769 7468  he dot plot with
+00003500: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00003510: 2061 6e6e 6f74 6174 696f 6e20 616e 6420   annotation and 
+00003520: 7769 7468 2063 6f6f 7264 696e 6174 6573  with coordinates
+00003530: 2069 6e20 7468 6520 6e75 6d62 6572 206f   in the number o
+00003540: 6620 6765 6e65 2e0a 5468 6520 6069 6164  f gene..The `iad
+00003550: 686f 7265 2e63 6f6e 6660 2069 7320 7468  hore.conf` is th
+00003560: 6520 636f 6e66 6967 7572 6174 696f 6e20  e configuration 
+00003570: 6669 6c65 2066 6f72 2060 692d 6164 686f  file for `i-adho
+00003580: 7265 602e 0a54 6865 2060 6961 6468 6f72  re`..The `iadhor
+00003590: 652d 6f75 7460 2073 7562 666f 6c64 6572  e-out` subfolder
+000035a0: 2063 6f6e 7461 696e 7320 7468 6520 6f72   contains the or
+000035b0: 6967 696e 616c 2063 6f6c 6c69 6e65 6172  iginal collinear
+000035c0: 2072 6573 756c 7473 2066 726f 6d20 6069   results from `i
+000035d0: 2d61 6468 6f72 6560 2028 706c 6561 7365  -adhore` (please
+000035e0: 2072 6566 6572 2074 6f20 7468 6520 6d61   refer to the ma
+000035f0: 6e75 616c 2074 6865 7265 696e 2066 6f72  nual therein for
+00003600: 2064 6574 6169 6c65 6420 6465 7363 7269   detailed descri
+00003610: 7074 696f 6e29 2e0a 0a23 2323 2050 6970  ption)...### Pip
+00003620: 656c 696e 6520 330a 2a2a 436f 6e73 7472  eline 3.**Constr
+00003630: 7563 7469 6f6e 206f 6620 6167 6520 6469  uction of age di
+00003640: 7374 7269 6275 7469 6f6e 2061 6e64 2045  stribution and E
+00003650: 4c4d 4d20 616e 616c 7973 6973 2a2a 0a60  LMM analysis**.`
+00003660: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
+00003670: 6567 6961 5f63 6f65 7275 6c65 6120 2d6f  egia_coerulea -o
+00003680: 2077 6764 5f64 6d64 0a77 6764 206b 7364   wgd_dmd.wgd ksd
+00003690: 2077 6764 5f64 6d64 2f41 7175 696c 6567   wgd_dmd/Aquileg
+000036a0: 6961 5f63 6f65 7275 6c65 612e 7473 7620  ia_coerulea.tsv 
+000036b0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000036c0: 6561 202d 6f20 7767 645f 6b73 640a 7767  ea -o wgd_ksd.wg
+000036d0: 6420 7669 7a20 2d64 2077 6764 5f6b 7364  d viz -d wgd_ksd
+000036e0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000036f0: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00003700: 6f20 7767 645f 454c 4d4d 0a60 6060 0a60  o wgd_ELMM.```.`
+00003710: 6060 0a23 2077 6861 7420 7769 6c6c 2062  ``.# what will b
+00003720: 6520 696e 2074 6865 2077 6764 5f45 4c4d  e in the wgd_ELM
+00003730: 4d20 7265 7375 6c74 2064 6972 6563 746f  M result directo
+00003740: 7279 0a2d 7767 645f 454c 4d4d 0a2d 2d41  ry.-wgd_ELMM.--A
+00003750: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003760: 612e 7473 762e 6b73 2e74 7376 2e6b 7364  a.tsv.ks.tsv.ksd
+00003770: 2e70 6466 2f73 7667 0a2d 2d41 7175 696c  .pdf/svg.--Aquil
+00003780: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00003790: 762e 6b73 2e74 7376 2e73 706c 696e 655f  v.ks.tsv.spline_
+000037a0: 6e6f 6465 5f61 7665 7261 6765 642e 7064  node_averaged.pd
+000037b0: 662f 7376 670a 2d2d 4171 7569 6c65 6769  f/svg.--Aquilegi
+000037c0: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+000037d0: 732e 7473 762e 7370 6c69 6e65 5f77 6569  s.tsv.spline_wei
+000037e0: 6768 7465 642e 7064 662f 7376 670a 2d2d  ghted.pdf/svg.--
+000037f0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00003800: 6561 2e74 7376 2e6b 732e 7473 765f 7065  ea.tsv.ks.tsv_pe
+00003810: 616b 5f64 6574 6563 7469 6f6e 5f6e 6f64  ak_detection_nod
+00003820: 655f 6176 6572 6167 6564 2e70 6466 2f73  e_averaged.pdf/s
+00003830: 7667 0a2d 2d41 7175 696c 6567 6961 5f63  vg.--Aquilegia_c
+00003840: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00003850: 7376 5f70 6561 6b5f 6465 7465 6374 696f  sv_peak_detectio
+00003860: 6e5f 7765 6967 6874 6564 2e70 6466 2f73  n_weighted.pdf/s
+00003870: 7667 0a2d 2d65 6c6d 6d5f 4171 7569 6c65  vg.--elmm_Aquile
+00003880: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00003890: 2e6b 732e 7473 765f 6265 7374 5f6d 6f64  .ks.tsv_best_mod
+000038a0: 656c 735f 6e6f 6465 5f61 7665 7261 6765  els_node_average
+000038b0: 642e 7064 662f 7376 670a 2d2d 656c 6d6d  d.pdf/svg.--elmm
+000038c0: 5f41 7175 696c 6567 6961 5f63 6f65 7275  _Aquilegia_coeru
+000038d0: 6c65 612e 7473 762e 6b73 2e74 7376 5f62  lea.tsv.ks.tsv_b
+000038e0: 6573 745f 6d6f 6465 6c73 5f77 6569 6768  est_models_weigh
+000038f0: 7465 642e 7064 662f 7376 670a 2d2d 656c  ted.pdf/svg.--el
+00003900: 6d6d 5f41 7175 696c 6567 6961 5f63 6f65  mm_Aquilegia_coe
+00003910: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00003920: 5f6d 6f64 656c 735f 6461 7461 5f64 7269  _models_data_dri
+00003930: 7665 6e5f 6e6f 6465 5f61 7665 7261 6765  ven_node_average
+00003940: 642e 7064 662f 7376 670a 2d2d 656c 6d6d  d.pdf/svg.--elmm
+00003950: 5f41 7175 696c 6567 6961 5f63 6f65 7275  _Aquilegia_coeru
+00003960: 6c65 612e 7473 762e 6b73 2e74 7376 5f6d  lea.tsv.ks.tsv_m
+00003970: 6f64 656c 735f 6461 7461 5f64 7269 7665  odels_data_drive
+00003980: 6e5f 7765 6967 6874 6564 2e70 6466 2f73  n_weighted.pdf/s
+00003990: 7667 0a2d 2d65 6c6d 6d5f 4171 7569 6c65  vg.--elmm_Aquile
+000039a0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+000039b0: 2e6b 732e 7473 765f 6d6f 6465 6c73 5f72  .ks.tsv_models_r
+000039c0: 616e 646f 6d5f 6e6f 6465 5f61 7665 7261  andom_node_avera
+000039d0: 6765 642e 7064 662f 7376 670a 2d2d 656c  ged.pdf/svg.--el
+000039e0: 6d6d 5f41 7175 696c 6567 6961 5f63 6f65  mm_Aquilegia_coe
+000039f0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00003a00: 5f6d 6f64 656c 735f 7261 6e64 6f6d 5f77  _models_random_w
+00003a10: 6569 6768 7465 642e 7064 662f 7376 670a  eighted.pdf/svg.
+00003a20: 2d2d 656c 6d6d 5f42 4943 5f41 7175 696c  --elmm_BIC_Aquil
+00003a30: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00003a40: 762e 6b73 2e74 7376 5f6e 6f64 655f 6176  v.ks.tsv_node_av
+00003a50: 6572 6167 6564 2e70 6466 2f73 7667 0a2d  eraged.pdf/svg.-
+00003a60: 2d65 6c6d 6d5f 4249 435f 4171 7569 6c65  -elmm_BIC_Aquile
+00003a70: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00003a80: 2e6b 732e 7473 765f 7765 6967 6874 6564  .ks.tsv_weighted
+00003a90: 2e70 6466 2f73 7667 0a60 6060 0a54 6865  .pdf/svg.```.The
+00003aa0: 2060 4171 7569 6c65 6769 615f 636f 6572   `Aquilegia_coer
+00003ab0: 756c 6561 2e74 7376 2e6b 732e 7473 762e  ulea.tsv.ks.tsv.
+00003ac0: 6b73 642e 7064 662f 7376 6760 2069 7320  ksd.pdf/svg` is 
+00003ad0: 7468 6520 6261 7369 6320 2a4b 2a3c 7375  the basic *K*<su
+00003ae0: 623e 533c 2f73 7562 3e20 706c 6f74 2e20  b>S</sub> plot. 
+00003af0: 5468 6520 6041 7175 696c 6567 6961 5f63  The `Aquilegia_c
+00003b00: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00003b10: 7376 2e73 706c 696e 655f 6e6f 6465 5f61  sv.spline_node_a
+00003b20: 7665 7261 6765 642e 7064 662f 7376 6760  veraged.pdf/svg`
+00003b30: 2061 6e64 2060 4171 7569 6c65 6769 615f   and `Aquilegia_
+00003b40: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00003b50: 7473 762e 7370 6c69 6e65 5f77 6569 6768  tsv.spline_weigh
+00003b60: 7465 642e 7064 662f 7376 6760 2061 7265  ted.pdf/svg` are
+00003b70: 2074 6865 2073 706c 696e 6520 706c 6f74   the spline plot
+00003b80: 7320 7573 6564 2066 6f72 2070 6561 6b20  s used for peak 
+00003b90: 6465 7465 6374 696f 6e2e 2054 6865 2060  detection. The `
+00003ba0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00003bb0: 6561 2e74 7376 2e6b 732e 7473 765f 7065  ea.tsv.ks.tsv_pe
+00003bc0: 616b 5f64 6574 6563 7469 6f6e 5f6e 6f64  ak_detection_nod
+00003bd0: 655f 6176 6572 6167 6564 2e70 6466 2f73  e_averaged.pdf/s
+00003be0: 7667 6020 616e 6420 6041 7175 696c 6567  vg` and `Aquileg
+00003bf0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00003c00: 6b73 2e74 7376 5f70 6561 6b5f 6465 7465  ks.tsv_peak_dete
+00003c10: 6374 696f 6e5f 7765 6967 6874 6564 2e70  ction_weighted.p
+00003c20: 6466 2f73 7667 6020 6172 6520 7468 6520  df/svg` are the 
+00003c30: 7265 7375 6c74 7320 6f66 2070 6561 6b20  results of peak 
+00003c40: 6465 7465 6374 696f 6e2e 2054 6865 2060  detection. The `
+00003c50: 656c 6d6d 5f41 7175 696c 6567 6961 5f63  elmm_Aquilegia_c
+00003c60: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00003c70: 7376 5f62 6573 745f 6d6f 6465 6c73 5f6e  sv_best_models_n
+00003c80: 6f64 655f 6176 6572 6167 6564 2e70 6466  ode_averaged.pdf
+00003c90: 2f73 7667 6020 616e 6420 6065 6c6d 6d5f  /svg` and `elmm_
+00003ca0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00003cb0: 6561 2e74 7376 2e6b 732e 7473 765f 6265  ea.tsv.ks.tsv_be
+00003cc0: 7374 5f6d 6f64 656c 735f 7765 6967 6874  st_models_weight
+00003cd0: 6564 2e70 6466 2f73 7667 6020 6172 6520  ed.pdf/svg` are 
+00003ce0: 7468 6520 7265 7375 6c74 7320 6672 6f6d  the results from
+00003cf0: 2074 6865 2062 6573 7420 6d6f 6465 6c20   the best model 
+00003d00: 696e 666f 726d 6564 2062 7920 6042 4943  informed by `BIC
+00003d10: 602e 2054 6865 2060 656c 6d6d 5f41 7175  `. The `elmm_Aqu
+00003d20: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00003d30: 7473 762e 6b73 2e74 7376 5f6d 6f64 656c  tsv.ks.tsv_model
+00003d40: 735f 6461 7461 5f64 7269 7665 6e5f 6e6f  s_data_driven_no
+00003d50: 6465 5f61 7665 7261 6765 642e 7064 662f  de_averaged.pdf/
+00003d60: 7376 6760 2061 6e64 2060 656c 6d6d 5f41  svg` and `elmm_A
+00003d70: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003d80: 612e 7473 762e 6b73 2e74 7376 5f6d 6f64  a.tsv.ks.tsv_mod
+00003d90: 656c 735f 6461 7461 5f64 7269 7665 6e5f  els_data_driven_
+00003da0: 7765 6967 6874 6564 2e70 6466 2f73 7667  weighted.pdf/svg
+00003db0: 6020 6172 6520 7468 6520 6d6f 6465 6c20  ` are the model 
+00003dc0: 7265 7375 6c74 7320 6672 6f6d 2064 6174  results from dat
+00003dd0: 612d 6472 6976 656e 2069 6e69 7469 6174  a-driven initiat
+00003de0: 696f 6e2e 2054 6865 2060 656c 6d6d 5f41  ion. The `elmm_A
+00003df0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003e00: 612e 7473 762e 6b73 2e74 7376 5f6d 6f64  a.tsv.ks.tsv_mod
+00003e10: 656c 735f 7261 6e64 6f6d 5f6e 6f64 655f  els_random_node_
+00003e20: 6176 6572 6167 6564 2e70 6466 2f73 7667  averaged.pdf/svg
+00003e30: 6020 616e 6420 6065 6c6d 6d5f 4171 7569  ` and `elmm_Aqui
+00003e40: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00003e50: 7376 2e6b 732e 7473 765f 6d6f 6465 6c73  sv.ks.tsv_models
+00003e60: 5f72 616e 646f 6d5f 7765 6967 6874 6564  _random_weighted
+00003e70: 2e70 6466 2f73 7667 6020 6172 6520 7468  .pdf/svg` are th
+00003e80: 6520 6d6f 6465 6c20 7265 7375 6c74 7320  e model results 
+00003e90: 6672 6f6d 2072 616e 646f 6d20 696e 6974  from random init
+00003ea0: 6961 7469 6f6e 2e20 5468 6520 6065 6c6d  iation. The `elm
+00003eb0: 6d5f 4249 435f 4171 7569 6c65 6769 615f  m_BIC_Aquilegia_
+00003ec0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00003ed0: 7473 765f 6e6f 6465 5f61 7665 7261 6765  tsv_node_average
+00003ee0: 642e 7064 662f 7376 6760 2061 6e64 2060  d.pdf/svg` and `
+00003ef0: 656c 6d6d 5f42 4943 5f41 7175 696c 6567  elmm_BIC_Aquileg
+00003f00: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00003f10: 6b73 2e74 7376 5f77 6569 6768 7465 642e  ks.tsv_weighted.
+00003f20: 7064 662f 7376 6760 2061 7265 2074 6865  pdf/svg` are the
+00003f30: 2042 4943 2070 6c6f 7473 2066 6f72 2065   BIC plots for e
+00003f40: 6163 6820 6d6f 6465 6c2e 0a0a 2323 2320  ach model...### 
+00003f50: 5069 7065 6c69 6e65 2034 0a2a 2a43 6f6e  Pipeline 4.**Con
+00003f60: 7374 7275 6374 696f 6e20 6f66 2061 6765  struction of age
+00003f70: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
+00003f80: 7468 2063 6f6c 6c69 6e65 6172 6974 7920  th collinearity 
+00003f90: 616e 6420 7065 616b 2066 696e 6469 6e67  and peak finding
+00003fa0: 2061 6e61 6c79 7369 732a 2a0a 6060 600a   analysis**.```.
+00003fb0: 7767 6420 646d 6420 4171 7569 6c65 6769  wgd dmd Aquilegi
+00003fc0: 615f 636f 6572 756c 6561 202d 6f20 7767  a_coerulea -o wg
+00003fd0: 645f 646d 640a 7767 6420 6b73 6420 7767  d_dmd.wgd ksd wg
+00003fe0: 645f 646d 642f 4171 7569 6c65 6769 615f  d_dmd/Aquilegia_
+00003ff0: 636f 6572 756c 6561 2e74 7376 2041 7175  coerulea.tsv Aqu
+00004000: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+00004010: 2d6f 2077 6764 5f6b 7364 0a77 6764 2073  -o wgd_ksd.wgd s
+00004020: 796e 202d 6620 6d52 4e41 202d 6120 4e61  yn -f mRNA -a Na
+00004030: 6d65 2077 6764 5f64 6d64 2f41 7175 696c  me wgd_dmd/Aquil
+00004040: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00004050: 7620 4171 7569 6c65 6769 615f 636f 6572  v Aquilegia_coer
+00004060: 756c 6561 2e67 6666 3320 2d6b 7320 7767  ulea.gff3 -ks wg
+00004070: 645f 6b73 642f 4171 7569 6c65 6769 615f  d_ksd/Aquilegia_
+00004080: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00004090: 7473 7620 2d6f 2077 6764 5f73 796e 0a77  tsv -o wgd_syn.w
+000040a0: 6764 2070 6561 6b20 2d2d 6865 7572 6973  gd peak --heuris
+000040b0: 7469 6320 7767 645f 6b73 642f 4171 7569  tic wgd_ksd/Aqui
+000040c0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000040d0: 7376 2e6b 732e 7473 7620 2d61 7020 7767  sv.ks.tsv -ap wg
+000040e0: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+000040f0: 742f 616e 6368 6f72 706f 696e 7473 2e74  t/anchorpoints.t
+00004100: 7874 202d 736d 2077 6764 5f73 796e 2f69  xt -sm wgd_syn/i
+00004110: 6164 686f 7265 2d6f 7574 2f73 6567 6d65  adhore-out/segme
+00004120: 6e74 732e 7478 7420 2d6c 6520 7767 645f  nts.txt -le wgd_
+00004130: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+00004140: 6c69 7374 5f65 6c65 6d65 6e74 732e 7478  list_elements.tx
+00004150: 7420 2d6d 7020 7767 645f 7379 6e2f 6961  t -mp wgd_syn/ia
+00004160: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
+00004170: 6c69 636f 6e5f 7061 6972 732e 7478 7420  licon_pairs.txt 
+00004180: 2d6e 2031 2034 202d 6b63 2033 202d 6f20  -n 1 4 -kc 3 -o 
+00004190: 7767 645f 7065 616b 0a60 6060 0a60 6060  wgd_peak.```.```
+000041a0: 0a23 2077 6861 7420 7769 6c6c 2062 6520  .# what will be 
+000041b0: 696e 2074 6865 2077 6764 5f70 6561 6b20  in the wgd_peak 
+000041c0: 7265 7375 6c74 2064 6972 6563 746f 7279  result directory
+000041d0: 0a2d 7767 645f 7065 616b 0a2d 2d41 6e63  .-wgd_peak.--Anc
+000041e0: 686f 724b 735f 4669 6e64 5065 616b 0a2d  horKs_FindPeak.-
+000041f0: 2d2d 416e 6368 6f72 4b73 5f50 6561 6b43  --AnchorKs_PeakC
+00004200: 495f 4171 7569 6c65 6769 615f 636f 6572  I_Aquilegia_coer
+00004210: 756c 6561 2e74 7376 2e6b 732e 7473 765f  ulea.tsv.ks.tsv_
+00004220: 6e6f 6465 5f61 7665 7261 6765 642f 7765  node_averaged/we
+00004230: 6967 6874 6564 2e70 6466 0a2d 2d2d 4171  ighted.pdf.---Aq
+00004240: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00004250: 2e74 7376 2e6b 732e 7473 765f 3935 2543  .tsv.ks.tsv_95%C
+00004260: 495f 4150 5f66 6f72 5f64 6174 696e 675f  I_AP_for_dating_
+00004270: 7765 6967 6874 6564 5f66 6f72 6d61 742e  weighted_format.
+00004280: 7473 760a 2d2d 2d41 7175 696c 6567 6961  tsv.---Aquilegia
+00004290: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+000042a0: 2e74 7376 5f39 3525 4349 5f41 505f 666f  .tsv_95%CI_AP_fo
+000042b0: 725f 6461 7469 6e67 5f77 6569 6768 7465  r_dating_weighte
+000042c0: 642e 7473 760a 2d2d 2d41 7175 696c 6567  d.tsv.---Aquileg
+000042d0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+000042e0: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
+000042f0: 666f 725f 6461 7469 6e67 5f6e 6f64 655f  for_dating_node_
+00004300: 6176 6572 6167 6564 5f66 6f72 6d61 742e  averaged_format.
+00004310: 7473 760a 2d2d 2d41 7175 696c 6567 6961  tsv.---Aquilegia
+00004320: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00004330: 2e74 7376 5f39 3525 4349 5f41 505f 666f  .tsv_95%CI_AP_fo
+00004340: 725f 6461 7469 6e67 5f6e 6f64 655f 6176  r_dating_node_av
+00004350: 6572 6167 6564 2e74 7376 0a2d 2d41 6e63  eraged.tsv.--Anc
+00004360: 686f 724b 735f 474d 4d0a 2d2d 2d47 4d4d  horKs_GMM.---GMM
+00004370: 5f45 6c62 6f77 2d4c 6f73 735f 6f72 6967  _Elbow-Loss_orig
+00004380: 696e 616c 5f4b 732e 7064 660a 2d2d 2d47  inal_Ks.pdf.---G
+00004390: 4d4d 5f4f 7269 6769 6e61 6c5f 416e 6368  MM_Original_Anch
+000043a0: 6f72 4b73 5f43 6c75 7374 6572 696e 675f  orKs_Clustering_
+000043b0: 5369 6c68 6f75 6574 7465 5f43 6f65 6666  Silhouette_Coeff
+000043c0: 6963 6965 6e74 2e70 6466 0a2d 2d2d 4f72  icient.pdf.---Or
+000043d0: 6967 696e 616c 5f41 6e63 686f 724b 735f  iginal_AnchorKs_
+000043e0: 474d 4d5f 312f 322f 332f 3463 6f6d 706f  GMM_1/2/3/4compo
+000043f0: 6e65 6e74 735f 7072 6564 6963 7469 6f6e  nents_prediction
+00004400: 2e74 7376 0a2d 2d2d 4f72 6967 696e 616c  .tsv.---Original
+00004410: 5f41 6e63 686f 724b 735f 474d 4d5f 4149  _AnchorKs_GMM_AI
+00004420: 435f 4249 432e 7064 660a 2d2d 2d4f 7269  C_BIC.pdf.---Ori
+00004430: 6769 6e61 6c5f 416e 6368 6f72 4b73 5f47  ginal_AnchorKs_G
+00004440: 4d4d 5f43 6f6d 706f 6e65 6e74 312f 322f  MM_Component1/2/
+00004450: 332f 345f 6e6f 6465 5f61 7665 7261 6765  3/4_node_average
+00004460: 645f 4c6f 676e 6f72 6d61 6c2e 7064 660a  d_Lognormal.pdf.
+00004470: 2d2d 2d4f 7269 6769 6e61 6c5f 416e 6368  ---Original_Anch
+00004480: 6f72 4b73 5f47 4d4d 5f43 6f6d 706f 6e65  orKs_GMM_Compone
+00004490: 6e74 312f 322f 332f 345f 6e6f 6465 5f61  nt1/2/3/4_node_a
+000044a0: 7665 7261 6765 642e 7064 660a 2d2d 2d4c  veraged.pdf.---L
+000044b0: 6f67 474d 4d5f 4349 0a2d 2d2d 2d47 4d4d  ogGMM_CI.----GMM
+000044c0: 5f31 2f32 2f33 2f34 636f 6d70 6f6e 656e  _1/2/3/4componen
+000044d0: 7473 5f43 302f 312f 322f 335f 3935 2543  ts_C0/1/2/3_95%C
+000044e0: 492e 7473 760a 2d2d 2d2d 474d 4d5f 436f  I.tsv.----GMM_Co
+000044f0: 6d70 6f6e 656e 7431 2f32 2f33 2f34 5f6e  mponent1/2/3/4_n
+00004500: 6f64 655f 6176 6572 6167 6564 5f4c 6f67  ode_averaged_Log
+00004510: 6e6f 726d 616c 2e70 6466 0a2d 2d2d 4869  normal.pdf.---Hi
+00004520: 6768 4d61 7373 5f43 490a 2d2d 2d2d 474d  ghMass_CI.----GM
+00004530: 4d5f 312f 322f 332f 3463 6f6d 706f 6e65  M_1/2/3/4compone
+00004540: 6e74 735f 4330 2f31 2f32 2f33 5f48 6967  nts_C0/1/2/3_Hig
+00004550: 684d 6173 735f 3935 2543 492e 7473 760a  hMass_95%CI.tsv.
+00004560: 2d2d 2d2d 474d 4d5f 312f 322f 332f 3463  ----GMM_1/2/3/4c
+00004570: 6f6d 706f 6e65 6e74 735f 4869 6768 4d61  omponents_HighMa
+00004580: 7373 5f39 3525 4349 2e70 6466 0a2d 2d53  ss_95%CI.pdf.--S
+00004590: 6567 6d65 6e74 4775 6964 654b 735f 474d  egmentGuideKs_GM
+000045a0: 4d0a 2d2d 2d47 4d4d 5f45 6c62 6f77 2d4c  M.---GMM_Elbow-L
+000045b0: 6f73 735f 5365 676d 656e 745f 4b73 2e70  oss_Segment_Ks.p
+000045c0: 6466 0a2d 2d2d 474d 4d5f 5365 676d 656e  df.---GMM_Segmen
+000045d0: 745f 4b73 5f43 6c75 7374 6572 696e 675f  t_Ks_Clustering_
+000045e0: 5369 6c68 6f75 6574 7465 5f43 6f65 6666  Silhouette_Coeff
+000045f0: 6963 6965 6e74 2e70 6466 0a2d 2d2d 5365  icient.pdf.---Se
+00004600: 676d 656e 742d 6775 6964 6564 5f41 6e63  gment-guided_Anc
+00004610: 686f 724b 735f 474d 4d5f 312f 322f 332f  horKs_GMM_1/2/3/
+00004620: 3463 6f6d 706f 6e65 6e74 735f 7072 6564  4components_pred
+00004630: 6963 7469 6f6e 2e74 7376 0a2d 2d2d 5365  iction.tsv.---Se
+00004640: 676d 656e 745f 4b73 5f43 6c75 7374 6572  gment_Ks_Cluster
+00004650: 735f 474d 4d5f 436f 6d70 6f6e 656e 7431  s_GMM_Component1
+00004660: 2f32 2f33 2f34 2e70 6466 0a2d 2d2d 5365  /2/3/4.pdf.---Se
+00004670: 676d 656e 745f 4b73 5f43 6c75 7374 6572  gment_Ks_Cluster
+00004680: 735f 4c6f 676e 6f72 6d61 6c5f 474d 4d5f  s_Lognormal_GMM_
+00004690: 436f 6d70 6f6e 656e 7431 2f32 2f33 2f34  Component1/2/3/4
+000046a0: 2e70 6466 0a2d 2d2d 5365 676d 656e 745f  .pdf.---Segment_
+000046b0: 4b73 5f47 4d4d 5f41 4943 5f42 4943 2e70  Ks_GMM_AIC_BIC.p
+000046c0: 6466 0a2d 2d2d 5365 676d 656e 745f 4b73  df.---Segment_Ks
+000046d0: 2e74 7376 0a2d 2d2d 4869 6768 4d61 7373  .tsv.---HighMass
+000046e0: 5f43 490a 2d2d 2d2d 5365 676d 656e 745f  _CI.----Segment_
+000046f0: 6775 6964 6564 5f31 2f32 2f33 2f34 636f  guided_1/2/3/4co
+00004700: 6d70 6f6e 656e 7473 5f43 302f 312f 322f  mponents_C0/1/2/
+00004710: 335f 4869 6768 4d61 7373 5f39 3525 4349  3_HighMass_95%CI
+00004720: 2e74 7376 0a2d 2d2d 2d53 6567 6d65 6e74  .tsv.----Segment
+00004730: 5f67 7569 6465 645f 312f 322f 332f 3463  _guided_1/2/3/4c
+00004740: 6f6d 706f 6e65 6e74 735f 4869 6768 4d61  omponents_HighMa
+00004750: 7373 5f39 3525 4349 2e70 6466 0a2d 2d2d  ss_95%CI.pdf.---
+00004760: 4844 525f 4349 0a2d 2d2d 2d53 6567 6d65  HDR_CI.----Segme
+00004770: 6e74 5f67 7569 6465 645f 3935 2548 4452  nt_guided_95%HDR
+00004780: 5f41 505f 312f 322f 332f 3463 6f6d 706f  _AP_1/2/3/4compo
+00004790: 6e65 6e74 735f 4330 2f31 2f32 2f33 2e74  nents_C0/1/2/3.t
+000047a0: 7376 0a2d 2d2d 2d53 6567 6d65 6e74 5f67  sv.----Segment_g
+000047b0: 7569 6465 645f 416e 6368 6f72 4b73 5f47  uided_AnchorKs_G
+000047c0: 4d4d 5f43 6f6d 706f 6e65 6e74 312f 322f  MM_Component1/2/
+000047d0: 332f 345f 6e6f 6465 5f61 7665 7261 6765  3/4_node_average
+000047e0: 645f 6b64 652e 7064 660a 2d2d 2d2d 5365  d_kde.pdf.----Se
+000047f0: 676d 656e 745f 6775 6964 6564 5f41 6e63  gment_guided_Anc
+00004800: 686f 724b 735f 474d 4d5f 436f 6d70 6f6e  horKs_GMM_Compon
+00004810: 656e 7431 2f32 2f33 2f34 5f6e 6f64 655f  ent1/2/3/4_node_
+00004820: 6176 6572 6167 6564 2e70 6466 0a2d 2d53  averaged.pdf.--S
+00004830: 6567 6d65 6e74 4b73 5f46 696e 6450 6561  egmentKs_FindPea
+00004840: 6b0a 2d2d 2d53 6567 6d65 6e74 4b73 5f50  k.---SegmentKs_P
+00004850: 6561 6b43 495f 4171 7569 6c65 6769 615f  eakCI_Aquilegia_
+00004860: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00004870: 7473 762e 7064 660a 2d2d 2d50 6561 6b5f  tsv.pdf.---Peak_
+00004880: 312f 325f 5365 676d 656e 745f 6775 6964  1/2_Segment_guid
+00004890: 6564 5f41 7175 696c 6567 6961 5f63 6f65  ed_Aquilegia_coe
+000048a0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+000048b0: 5f39 3525 4349 5f4d 505f 666f 725f 6461  _95%CI_MP_for_da
+000048c0: 7469 6e67 5f66 6f72 6d61 742e 7473 760a  ting_format.tsv.
+000048d0: 2d2d 2d50 6561 6b5f 312f 325f 5365 676d  ---Peak_1/2_Segm
+000048e0: 656e 745f 6775 6964 6564 5f41 7175 696c  ent_guided_Aquil
+000048f0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00004900: 762e 6b73 2e74 7376 5f39 3525 4349 5f4d  v.ks.tsv_95%CI_M
+00004910: 505f 666f 725f 6461 7469 6e67 2e74 7376  P_for_dating.tsv
+00004920: 0a60 6060 0a46 6f75 7220 7265 7375 6c74  .```.Four result
+00004930: 2073 7562 666c 6f64 6572 7320 7769 6c6c   subfloders will
+00004940: 2062 6520 7072 6f64 7563 6564 2c20 6e61   be produced, na
+00004950: 6d65 6c79 2060 416e 6368 6f72 4b73 5f46  mely `AnchorKs_F
+00004960: 696e 6450 6561 6b60 2c20 6041 6e63 686f  indPeak`, `Ancho
+00004970: 724b 735f 474d 4d60 2c20 6053 6567 6d65  rKs_GMM`, `Segme
+00004980: 6e74 4775 6964 654b 735f 474d 4d60 2061  ntGuideKs_GMM` a
+00004990: 6e64 2060 5365 676d 656e 744b 735f 4669  nd `SegmentKs_Fi
+000049a0: 6e64 5065 616b 602e 2054 6865 2060 416e  ndPeak`. The `An
+000049b0: 6368 6f72 4b73 5f46 696e 6450 6561 6b60  chorKs_FindPeak`
+000049c0: 2073 7562 666c 6f64 6572 2063 6f6e 7461   subfloder conta
+000049d0: 696e 7320 7265 7375 6c74 7320 6f66 2074  ins results of t
+000049e0: 6865 2064 6574 6563 7465 6420 7065 616b  he detected peak
+000049f0: 7320 6279 2074 6865 2060 7369 676e 616c  s by the `signal
+00004a00: 6020 6d6f 6475 6c65 206f 6620 6053 6369  ` module of `Sci
+00004a10: 5079 6020 6c69 6272 6172 7920 616e 6420  Py` library and 
+00004a20: 7468 6520 6173 7375 6d65 6420 6869 6768  the assumed high
+00004a30: 6573 7420 6d61 7373 2070 6172 7420 2872  est mass part (r
+00004a40: 6566 6572 7265 6420 746f 2061 7320 4869  eferred to as Hi
+00004a50: 6768 4d61 7373 2068 6572 6561 6674 6572  ghMass hereafter
+00004a60: 2920 6f66 2065 6163 6820 7065 616b 2c20  ) of each peak, 
+00004a70: 7768 6963 6820 6361 6e20 6265 2075 7365  which can be use
+00004a80: 6420 666f 7220 6675 7274 6865 7220 5747  d for further WG
+00004a90: 4420 6461 7469 6e67 2e20 5468 6520 6041  D dating. The `A
+00004aa0: 6e63 686f 724b 735f 474d 4d60 2073 686f  nchorKs_GMM` sho
+00004ab0: 7773 2074 6865 2047 4d4d 2072 6573 756c  ws the GMM resul
+00004ac0: 7473 2075 706f 6e20 7468 6520 6f72 6967  ts upon the orig
+00004ad0: 696e 616c 2061 6e63 686f 7220 2a4b 2a3c  inal anchor *K*<
+00004ae0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
+00004af0: 7269 6275 7469 6f6e 2062 7920 7468 6520  ribution by the 
+00004b00: 606d 6978 7475 7265 6020 6d6f 6475 6c65  `mixture` module
+00004b10: 206f 6620 6073 6369 6b69 742d 6c65 6172   of `scikit-lear
+00004b20: 6e60 206c 6962 7261 7279 2061 6e64 2074  n` library and t
+00004b30: 776f 2073 7562 666c 6f64 6572 732c 2060  wo subfloders, `
+00004b40: 4c6f 6747 4d4d 5f43 4960 2063 6f6e 7461  LogGMM_CI` conta
+00004b50: 696e 696e 6720 7468 6520 7265 7375 6c74  ining the result
+00004b60: 7320 6f66 2039 3525 2043 4920 6f66 2065  s of 95% CI of e
+00004b70: 6163 6820 636f 6d70 6f6e 656e 742c 2060  ach component, `
+00004b80: 4869 6768 4d61 7373 5f43 4960 2063 6f6e  HighMass_CI` con
+00004b90: 7461 696e 696e 6720 7468 6520 4869 6768  taining the High
+00004ba0: 4d61 7373 206f 6620 6561 6368 2063 6f6d  Mass of each com
+00004bb0: 706f 6e65 6e74 2c20 7768 6963 6820 6361  ponent, which ca
+00004bc0: 6e20 6265 2075 7365 6420 666f 7220 6675  n be used for fu
+00004bd0: 7274 6865 7220 5747 4420 6461 7469 6e67  rther WGD dating
+00004be0: 2e20 5468 6520 6053 6567 6d65 6e74 4775  . The `SegmentGu
+00004bf0: 6964 654b 735f 474d 4d60 2073 7562 666f  ideKs_GMM` subfo
+00004c00: 6c64 6572 2070 7265 7365 6e74 7320 7265  lder presents re
+00004c10: 7375 6c74 7320 6f66 2073 6567 6d65 6e74  sults of segment
+00004c20: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00004c30: 2047 4d4d 2077 6869 6368 2061 7265 206d   GMM which are m
+00004c40: 6170 7065 6420 6261 636b 2074 6f20 7468  apped back to th
+00004c50: 6520 7265 7369 6469 6e67 2061 6e63 686f  e residing ancho
+00004c60: 7220 7061 6972 7320 616e 6420 7468 6520  r pairs and the 
+00004c70: 6173 736f 6369 6174 6564 2039 3525 2048  associated 95% H
+00004c80: 4452 2061 6e64 2048 6967 684d 6173 7320  DR and HighMass 
+00004c90: 6f66 2065 6163 6820 7365 676d 656e 7420  of each segment 
+00004ca0: 636c 7573 7465 7220 696e 2073 7562 666c  cluster in subfl
+00004cb0: 6f64 6572 7320 6f66 2060 4844 525f 4349  oders of `HDR_CI
+00004cc0: 6020 616e 6420 6048 6967 684d 6173 735f  ` and `HighMass_
+00004cd0: 4349 602e 2054 6865 2060 5365 676d 656e  CI`. The `Segmen
+00004ce0: 744b 735f 4669 6e64 5065 616b 6020 7375  tKs_FindPeak` su
+00004cf0: 6266 6f6c 6465 7220 6973 2073 696d 696c  bfolder is simil
+00004d00: 6172 2074 6f20 6041 6e63 686f 724b 735f  ar to `AnchorKs_
+00004d10: 4669 6e64 5065 616b 6020 6275 7420 7769  FindPeak` but wi
+00004d20: 7468 2073 6567 6d65 6e74 202a 4b2a 3c73  th segment *K*<s
+00004d30: 7562 3e53 3c2f 7375 623e 2069 6e73 7465  ub>S</sub> inste
+00004d40: 6164 2e20 5468 6520 2a4b 2a3c 7375 623e  ad. The *K*<sub>
+00004d50: 533c 2f73 7562 3e20 696e 2060 4d75 6c74  S</sub> in `Mult
+00004d60: 6970 6c69 636f 6e60 2063 616e 2061 6c73  iplicon` can als
+00004d70: 6f20 6265 2063 616c 6375 6c61 7465 6420  o be calculated 
+00004d80: 696e 2070 6c61 6365 206f 6620 6053 6567  in place of `Seg
+00004d90: 6d65 6e74 6020 7573 696e 6720 7468 6520  ment` using the 
+00004da0: 6f70 7469 6f6e 2060 2d2d 6775 6964 6560  option `--guide`
+00004db0: 2061 7320 7375 6368 2074 6865 2072 6573   as such the res
+00004dc0: 756c 7420 7469 746c 652c 206c 6162 656c  ult title, label
+00004dd0: 2c20 6669 6c65 2061 6e64 2066 6f6c 6465  , file and folde
+00004de0: 7220 6e61 6d65 7320 7769 6c6c 2062 6520  r names will be 
+00004df0: 6368 616e 6765 6420 6163 636f 7264 696e  changed accordin
+00004e00: 676c 792e 0a0a 2323 2320 5069 7065 6c69  gly...### Pipeli
+00004e10: 6e65 2035 0a2a 2a43 6f6e 7374 7275 6374  ne 5.**Construct
+00004e20: 696f 6e20 6f66 2061 6765 2064 6973 7472  ion of age distr
+00004e30: 6962 7574 696f 6e20 7769 7468 2063 6f6c  ibution with col
+00004e40: 6c69 6e65 6172 6974 7920 616e 6420 7261  linearity and ra
+00004e50: 7465 2063 6f72 7265 6374 696f 6e2a 2a0a  te correction**.
+00004e60: 6060 600a 7767 6420 646d 6420 4171 7569  ```.wgd dmd Aqui
+00004e70: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
+00004e80: 6f20 7767 645f 646d 640a 7767 6420 6b73  o wgd_dmd.wgd ks
+00004e90: 6420 7767 645f 646d 642f 4171 7569 6c65  d wgd_dmd/Aquile
+00004ea0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00004eb0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+00004ec0: 6c65 6120 2d6f 2077 6764 5f6b 7364 0a77  lea -o wgd_ksd.w
+00004ed0: 6764 2073 796e 202d 6620 6d52 4e41 202d  gd syn -f mRNA -
+00004ee0: 6120 4e61 6d65 2077 6764 5f64 6d64 2f41  a Name wgd_dmd/A
+00004ef0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00004f00: 612e 7473 7620 4171 7569 6c65 6769 615f  a.tsv Aquilegia_
+00004f10: 636f 6572 756c 6561 2e67 6666 3320 2d6b  coerulea.gff3 -k
+00004f20: 7320 7767 645f 6b73 642f 4171 7569 6c65  s wgd_ksd/Aquile
+00004f30: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00004f40: 2e6b 732e 7473 7620 2d6f 2077 6764 5f73  .ks.tsv -o wgd_s
+00004f50: 796e 0a77 6764 2064 6d64 202d 2d67 6c6f  yn.wgd dmd --glo
+00004f60: 6261 6c6d 7262 6820 4171 7569 6c65 6769  balmrbh Aquilegi
+00004f70: 615f 636f 6572 756c 6561 2050 726f 7465  a_coerulea Prote
+00004f80: 615f 6379 6e61 726f 6964 6573 2041 636f  a_cynaroides Aco
+00004f90: 7275 735f 616d 6572 6963 616e 7573 2056  rus_americanus V
+00004fa0: 6974 6973 5f76 696e 6966 6572 6120 2d6f  itis_vinifera -o
+00004fb0: 2077 6764 5f67 6c6f 6261 6c6d 7262 680a   wgd_globalmrbh.
+00004fc0: 7767 6420 6b73 6420 7767 645f 676c 6f62  wgd ksd wgd_glob
+00004fd0: 616c 6d72 6268 2f67 6c6f 6261 6c5f 4d52  almrbh/global_MR
+00004fe0: 4248 2e74 7376 2041 7175 696c 6567 6961  BH.tsv Aquilegia
+00004ff0: 5f63 6f65 7275 6c65 6120 5072 6f74 6561  _coerulea Protea
+00005000: 5f63 796e 6172 6f69 6465 7320 4163 6f72  _cynaroides Acor
+00005010: 7573 5f61 6d65 7269 6361 6e75 7320 5669  us_americanus Vi
+00005020: 7469 735f 7669 6e69 6665 7261 202d 6f20  tis_vinifera -o 
+00005030: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
+00005040: 730a 7767 6420 7669 7a20 2d64 2077 6764  s.wgd viz -d wgd
+00005050: 5f67 6c6f 6261 6c6d 7262 685f 6b73 2f67  _globalmrbh_ks/g
+00005060: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
+00005070: 732e 7473 7620 2d66 6120 4171 7569 6c65  s.tsv -fa Aquile
+00005080: 6769 615f 636f 6572 756c 6561 202d 6570  gia_coerulea -ep
+00005090: 6b20 7767 645f 6b73 642f 4171 7569 6c65  k wgd_ksd/Aquile
+000050a0: 6769 615f 636f 6572 756c 6561 2e6b 732e  gia_coerulea.ks.
+000050b0: 7473 7620 2d61 7020 7767 645f 7379 6e2f  tsv -ap wgd_syn/
+000050c0: 6961 6468 6f72 652d 6f75 742f 616e 6368  iadhore-out/anch
+000050d0: 6f72 706f 696e 7473 2e74 7874 202d 7370  orpoints.txt -sp
+000050e0: 2073 7065 6369 6573 7472 6565 2e6e 7720   speciestree.nw 
+000050f0: 2d6f 2077 6764 5f76 697a 5f6d 6978 6564  -o wgd_viz_mixed
+00005100: 5f4b 7320 2d2d 706c 6f74 656c 6d6d 202d  _Ks --plotelmm -
+00005110: 2d70 6c6f 7461 7067 6d6d 202d 2d72 6577  -plotapgmm --rew
+00005120: 6569 6768 740a 6060 600a 6060 600a 2320  eight.```.```.# 
+00005130: 7768 6174 2077 696c 6c20 6265 2069 6e20  what will be in 
+00005140: 7468 6520 7767 645f 7669 7a5f 6d69 7865  the wgd_viz_mixe
+00005150: 645f 4b73 2072 6573 756c 7420 6469 7265  d_Ks result dire
+00005160: 6374 6f72 790a 2d77 6764 5f76 697a 5f6d  ctory.-wgd_viz_m
+00005170: 6978 6564 5f4b 730a 2d2d 416c 6c5f 7061  ixed_Ks.--All_pa
+00005180: 6972 732e 6b73 2e6e 6f64 652e 7765 6967  irs.ks.node.weig
+00005190: 6874 6564 2e70 6466 0a2d 2d46 6f63 7573  hted.pdf.--Focus
+000051a0: 5f73 6973 7465 725f 7061 6972 732e 6b73  _sister_pairs.ks
+000051b0: 2e6e 6f64 652e 7765 6967 6874 6564 2e70  .node.weighted.p
+000051c0: 6466 0a2d 2d67 6c6f 6261 6c5f 4d52 4248  df.--global_MRBH
+000051d0: 2e74 7376 2e6b 732e 7473 762e 6b73 642e  .tsv.ks.tsv.ksd.
+000051e0: 7064 662f 7376 670a 2d2d 4d69 7865 642e  pdf/svg.--Mixed.
+000051f0: 6b73 2e41 7175 696c 6567 6961 5f63 6f65  ks.Aquilegia_coe
+00005200: 7275 6c65 612e 6e6f 6465 2e77 6569 6768  rulea.node.weigh
+00005210: 7465 642e 7064 660a 2d2d 7370 6169 722e  ted.pdf.--spair.
+00005220: 636f 7272 6563 7465 642e 6b73 2e69 6e66  corrected.ks.inf
+00005230: 6f2e 7473 760a 2d2d 7370 6169 722e 6f72  o.tsv.--spair.or
+00005240: 6967 696e 616c 2e6b 732e 696e 666f 2e74  iginal.ks.info.t
+00005250: 7376 0a2d 2d53 696d 706c 655f 4b73 5f44  sv.--Simple_Ks_D
+00005260: 6973 7472 6962 7574 696f 6e73 0a2d 2d2d  istributions.---
+00005270: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+00005280: 732f 4171 7569 6c65 6769 615f 636f 6572  s/Aquilegia_coer
+00005290: 756c 6561 2f50 726f 7465 615f 6379 6e61  ulea/Protea_cyna
+000052a0: 726f 6964 6573 2f56 6974 6973 5f76 696e  roides/Vitis_vin
+000052b0: 6966 6572 615f 5f41 7175 696c 6567 6961  ifera__Aquilegia
+000052c0: 5f63 6f65 7275 6c65 612f 5072 6f74 6561  _coerulea/Protea
+000052d0: 5f63 796e 6172 6f69 6465 732f 5669 7469  _cynaroides/Viti
+000052e0: 735f 7669 6e69 6665 7261 2e6b 732e 6e6f  s_vinifera.ks.no
+000052f0: 6465 2e77 6569 6768 7465 642e 7064 660a  de.weighted.pdf.
+00005300: 6060 600a 5468 6520 6041 6c6c 5f70 6169  ```.The `All_pai
+00005310: 7273 2e6b 732e 6e6f 6465 2e77 6569 6768  rs.ks.node.weigh
+00005320: 7465 642e 7064 6660 2069 7320 7468 6520  ted.pdf` is the 
+00005330: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00005340: 706c 6f74 206f 6620 616c 6c20 7370 6563  plot of all spec
+00005350: 6965 7320 7061 6972 732e 2054 6865 2060  ies pairs. The `
+00005360: 466f 6375 735f 7369 7374 6572 5f70 6169  Focus_sister_pai
+00005370: 7273 2e6b 732e 6e6f 6465 2e77 6569 6768  rs.ks.node.weigh
+00005380: 7465 642e 7064 6660 2069 7320 7468 6520  ted.pdf` is the 
+00005390: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000053a0: 706c 6f74 206f 6620 616c 6c20 666f 6361  plot of all foca
+000053b0: 6c2d 7369 7374 6572 2073 7065 6369 6573  l-sister species
+000053c0: 2070 6169 7273 2e20 5468 6520 6067 6c6f   pairs. The `glo
+000053d0: 6261 6c5f 4d52 4248 2e74 7376 2e6b 732e  bal_MRBH.tsv.ks.
+000053e0: 7473 762e 6b73 642e 7064 662f 7376 6760  tsv.ksd.pdf/svg`
+000053f0: 2069 7320 7468 6520 2a4b 2a3c 7375 623e   is the *K*<sub>
+00005400: 533c 2f73 7562 3e20 706c 6f74 206f 6620  S</sub> plot of 
+00005410: 7468 6520 6461 7461 6669 6c65 2060 676c  the datafile `gl
+00005420: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
+00005430: 2e74 7376 602e 2054 6865 2060 4d69 7865  .tsv`. The `Mixe
+00005440: 642e 6b73 2e41 7175 696c 6567 6961 5f63  d.ks.Aquilegia_c
+00005450: 6f65 7275 6c65 612e 6e6f 6465 2e77 6569  oerulea.node.wei
+00005460: 6768 7465 642e 7064 6660 2069 7320 7468  ghted.pdf` is th
+00005470: 6520 2a2a 6669 6e61 6c20 7265 7375 6c74  e **final result
+00005480: 2a2a 206f 6620 7261 7465 2063 6f72 7265  ** of rate corre
+00005490: 6374 696f 6e20 2877 6974 6820 6d69 7874  ction (with mixt
+000054a0: 7572 6520 6d6f 6465 6c69 6e67 2072 6573  ure modeling res
+000054b0: 756c 7473 2069 6620 7365 7429 2e20 5468  ults if set). Th
+000054c0: 6520 6073 7061 6972 2e63 6f72 7265 6374  e `spair.correct
+000054d0: 6564 2e6b 732e 696e 666f 2e74 7376 6020  ed.ks.info.tsv` 
+000054e0: 616e 6420 6073 7061 6972 2e6f 7269 6769  and `spair.origi
+000054f0: 6e61 6c2e 6b73 2e69 6e66 6f2e 7473 7660  nal.ks.info.tsv`
+00005500: 2064 6f63 756d 656e 7420 7468 6520 2a4b   document the *K
+00005510: 2a3c 7375 623e 533c 2f73 7562 3e20 696e  *<sub>S</sub> in
+00005520: 666f 726d 6174 696f 6e20 6f66 2061 6c6c  formation of all
+00005530: 2073 7065 6369 6573 2070 6169 7273 2062   species pairs b
+00005540: 6566 6f72 6520 616e 6420 6166 7465 7220  efore and after 
+00005550: 7261 7465 2063 6f72 7265 6374 696f 6e2e  rate correction.
+00005560: 2054 6865 2073 7562 666f 6c64 6572 2060   The subfolder `
+00005570: 5369 6d70 6c65 5f4b 735f 4469 7374 7269  Simple_Ks_Distri
+00005580: 6275 7469 6f6e 7360 2063 6f6e 7461 696e  butions` contain
+00005590: 7320 7468 6520 7369 6e67 6c65 202a 4b2a  s the single *K*
+000055a0: 3c73 7562 3e53 3c2f 7375 623e 2070 6c6f  <sub>S</sub> plo
+000055b0: 7473 206f 6620 616c 6c20 7370 6563 6965  ts of all specie
+000055c0: 7320 7061 6972 732e 0a0a 2323 2320 5069  s pairs...### Pi
+000055d0: 7065 6c69 6e65 2036 0a2a 2a43 6f6e 7374  peline 6.**Const
+000055e0: 7275 6374 696f 6e20 6f66 2061 6765 2064  ruction of age d
+000055f0: 6973 7472 6962 7574 696f 6e20 7769 7468  istribution with
+00005600: 2063 6f6c 6c69 6e65 6172 6974 7920 616e   collinearity an
+00005610: 6420 5747 4420 6461 7469 6e67 2a2a 0a60  d WGD dating**.`
+00005620: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
+00005630: 6567 6961 5f63 6f65 7275 6c65 6120 2d6f  egia_coerulea -o
+00005640: 2077 6764 5f64 6d64 0a77 6764 206b 7364   wgd_dmd.wgd ksd
+00005650: 2077 6764 5f64 6d64 2f41 7175 696c 6567   wgd_dmd/Aquileg
+00005660: 6961 5f63 6f65 7275 6c65 612e 7473 7620  ia_coerulea.tsv 
+00005670: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00005680: 6561 202d 6f20 7767 645f 6b73 640a 7767  ea -o wgd_ksd.wg
+00005690: 6420 7379 6e20 2d66 206d 524e 4120 2d61  d syn -f mRNA -a
+000056a0: 204e 616d 6520 7767 645f 646d 642f 4171   Name wgd_dmd/Aq
+000056b0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000056c0: 2e74 7376 2041 7175 696c 6567 6961 5f63  .tsv Aquilegia_c
+000056d0: 6f65 7275 6c65 612e 6766 6633 202d 6b73  oerulea.gff3 -ks
+000056e0: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
+000056f0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00005700: 6b73 2e74 7376 202d 6f20 7767 645f 7379  ks.tsv -o wgd_sy
+00005710: 6e0a 7767 6420 7065 616b 202d 2d68 6575  n.wgd peak --heu
+00005720: 7269 7374 6963 2077 6764 5f6b 7364 2f41  ristic wgd_ksd/A
+00005730: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00005740: 612e 7473 762e 6b73 2e74 7376 202d 6170  a.tsv.ks.tsv -ap
+00005750: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+00005760: 2d6f 7574 2f61 6e63 686f 7270 6f69 6e74  -out/anchorpoint
+00005770: 732e 7478 7420 2d73 6d20 7767 645f 7379  s.txt -sm wgd_sy
+00005780: 6e2f 6961 6468 6f72 652d 6f75 742f 7365  n/iadhore-out/se
+00005790: 676d 656e 7473 2e74 7874 202d 6c65 2077  gments.txt -le w
+000057a0: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+000057b0: 7574 2f6c 6973 745f 656c 656d 656e 7473  ut/list_elements
+000057c0: 2e74 7874 202d 6d70 2077 6764 5f73 796e  .txt -mp wgd_syn
+000057d0: 2f69 6164 686f 7265 2d6f 7574 2f6d 756c  /iadhore-out/mul
+000057e0: 7469 706c 6963 6f6e 5f70 6169 7273 2e74  tiplicon_pairs.t
+000057f0: 7874 202d 6e20 3120 3420 2d6b 6320 3320  xt -n 1 4 -kc 3 
+00005800: 2d6f 2077 6764 5f70 6561 6b0a 7767 6420  -o wgd_peak.wgd 
+00005810: 646d 6420 2d66 2041 7175 696c 6567 6961  dmd -f Aquilegia
+00005820: 5f63 6f65 7275 6c65 6120 2d61 7020 7767  _coerulea -ap wg
+00005830: 645f 7065 616b 2f41 6e63 686f 724b 735f  d_peak/AnchorKs_
+00005840: 4669 6e64 5065 616b 2f41 7175 696c 6567  FindPeak/Aquileg
+00005850: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00005860: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
+00005870: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
+00005880: 7465 645f 666f 726d 6174 2e74 7376 202d  ted_format.tsv -
+00005890: 6f20 7767 645f 646d 645f 6f72 7468 6f20  o wgd_dmd_ortho 
+000058a0: 506f 7461 6d6f 6765 746f 6e5f 6163 7574  Potamogeton_acut
+000058b0: 6966 6f6c 6975 7320 5370 6972 6f64 656c  ifolius Spirodel
+000058c0: 615f 696e 7465 726d 6564 6961 2041 6d6f  a_intermedia Amo
+000058d0: 7270 686f 7068 616c 6c75 735f 6b6f 6e6a  rphophallus_konj
+000058e0: 6163 2041 6361 6e74 686f 6368 6c61 6d79  ac Acanthochlamy
+000058f0: 735f 6272 6163 7465 6174 6120 4469 6f73  s_bracteata Dios
+00005900: 636f 7265 615f 616c 6174 6120 4469 6f73  corea_alata Dios
+00005910: 636f 7265 615f 726f 7475 6e64 6174 6120  corea_rotundata 
+00005920: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+00005930: 7320 4163 6f72 7573 5f74 6174 6172 696e  s Acorus_tatarin
+00005940: 6f77 6969 2054 6574 7261 6365 6e74 726f  owii Tetracentro
+00005950: 6e5f 7369 6e65 6e73 6520 5472 6f63 686f  n_sinense Trocho
+00005960: 6465 6e64 726f 6e5f 6172 616c 696f 6964  dendron_aralioid
+00005970: 6573 2042 7578 7573 5f61 7573 7472 6f79  es Buxus_austroy
+00005980: 756e 6e61 6e65 6e73 6973 2042 7578 7573  unnanensis Buxus
+00005990: 5f73 696e 6963 6120 4e65 6c75 6d62 6f5f  _sinica Nelumbo_
+000059a0: 6e75 6369 6665 7261 2054 656c 6f70 6561  nucifera Telopea
+000059b0: 5f73 7065 6369 6f73 6973 7369 6d61 2050  _speciosissima P
+000059c0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
+000059d0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+000059e0: 6c65 610a 7767 6420 666f 6375 7320 2d2d  lea.wgd focus --
+000059f0: 7072 6f74 636f 6364 6174 696e 6720 2d2d  protcocdating --
+00005a00: 6161 6d6f 6465 6c20 6c67 2077 6764 5f64  aamodel lg wgd_d
+00005a10: 6d64 5f6f 7274 686f 2f6d 6572 6765 5f66  md_ortho/merge_f
+00005a20: 6f63 7573 5f61 702e 7473 7620 2d73 7020  ocus_ap.tsv -sp 
+00005a30: 6461 7469 6e67 5f74 7265 652e 6e77 202d  dating_tree.nw -
+00005a40: 6f20 7767 645f 6461 7469 6e67 202d 6420  o wgd_dating -d 
+00005a50: 6d63 6d63 7472 6565 202d 6473 2027 6275  mcmctree -ds 'bu
+00005a60: 726e 696e 203d 2032 3030 3027 202d 6473  rnin = 2000' -ds
+00005a70: 2027 7361 6d70 6672 6571 203d 2031 3030   'sampfreq = 100
+00005a80: 3027 202d 6473 2027 6e73 616d 706c 6520  0' -ds 'nsample 
+00005a90: 3d20 3230 3030 3027 2050 6f74 616d 6f67  = 20000' Potamog
+00005aa0: 6574 6f6e 5f61 6375 7469 666f 6c69 7573  eton_acutifolius
+00005ab0: 2053 7069 726f 6465 6c61 5f69 6e74 6572   Spirodela_inter
+00005ac0: 6d65 6469 6120 416d 6f72 7068 6f70 6861  media Amorphopha
+00005ad0: 6c6c 7573 5f6b 6f6e 6a61 6320 4163 616e  llus_konjac Acan
+00005ae0: 7468 6f63 686c 616d 7973 5f62 7261 6374  thochlamys_bract
+00005af0: 6561 7461 2044 696f 7363 6f72 6561 5f61  eata Dioscorea_a
+00005b00: 6c61 7461 2044 696f 7363 6f72 6561 5f72  lata Dioscorea_r
+00005b10: 6f74 756e 6461 7461 2041 636f 7275 735f  otundata Acorus_
+00005b20: 616d 6572 6963 616e 7573 2041 636f 7275  americanus Acoru
+00005b30: 735f 7461 7461 7269 6e6f 7769 6920 5465  s_tatarinowii Te
+00005b40: 7472 6163 656e 7472 6f6e 5f73 696e 656e  tracentron_sinen
+00005b50: 7365 2054 726f 6368 6f64 656e 6472 6f6e  se Trochodendron
+00005b60: 5f61 7261 6c69 6f69 6465 7320 4275 7875  _aralioides Buxu
+00005b70: 735f 6175 7374 726f 7975 6e6e 616e 656e  s_austroyunnanen
+00005b80: 7369 7320 4275 7875 735f 7369 6e69 6361  sis Buxus_sinica
+00005b90: 204e 656c 756d 626f 5f6e 7563 6966 6572   Nelumbo_nucifer
+00005ba0: 6120 5465 6c6f 7065 615f 7370 6563 696f  a Telopea_specio
+00005bb0: 7369 7373 696d 6120 5072 6f74 6561 5f63  sissima Protea_c
+00005bc0: 796e 6172 6f69 6465 7320 4171 7569 6c65  ynaroides Aquile
+00005bd0: 6769 615f 636f 6572 756c 6561 0a60 6060  gia_coerulea.```
+00005be0: 0a60 6060 0a23 2077 6861 7420 7769 6c6c  .```.# what will
+00005bf0: 2062 6520 696e 2074 6865 2077 6764 5f64   be in the wgd_d
+00005c00: 6174 696e 6720 7265 7375 6c74 2064 6972  ating result dir
+00005c10: 6563 746f 7279 0a2d 7767 645f 6461 7469  ectory.-wgd_dati
+00005c20: 6e67 0a2d 2d43 6f6e 6361 7465 6e61 7465  ng.--Concatenate
+00005c30: 642e 7061 6c6e 0a2d 2d43 6f6e 6361 7465  d.paln.--Concate
+00005c40: 6e61 7465 642e 7061 6c6e 2e70 616d 6c0a  nated.paln.paml.
+00005c50: 2d2d 4732 532e 4d61 700a 2d2d 4746 3030  --G2S.Map.--GF00
+00005c60: 3030 3030 3031 2e70 616c 6e0a 2d2d 2e2e  000001.paln.--..
+00005c70: 0a2d 2d47 4630 3030 3030 3138 372e 7061  .--GF00000187.pa
+00005c80: 6c6e 0a2d 2d6d 636d 6374 7265 650a 2d2d  ln.--mcmctree.--
+00005c90: 2d43 6f6e 6361 7465 6e61 7465 640a 2d2d  -Concatenated.--
+00005ca0: 2d2d 7065 700a 2d2d 2d2d 2d43 6f6e 6361  --pep.-----Conca
+00005cb0: 7465 6e61 7465 642e 7061 6c6e 2e70 616d  tenated.paln.pam
+00005cc0: 6c0a 2d2d 2d2d 2d64 6174 696e 675f 7472  l.-----dating_tr
+00005cd0: 6565 2e6e 770a 2d2d 2d2d 2d46 6967 5472  ee.nw.-----FigTr
+00005ce0: 6565 2e74 7265 0a2d 2d2d 2d2d 696e 2e42  ee.tre.-----in.B
+00005cf0: 560a 2d2d 2d2d 2d6c 672e 6461 740a 2d2d  V.-----lg.dat.--
+00005d00: 2d2d 2d6c 6e66 0a2d 2d2d 2d2d 6d63 6d63  ---lnf.-----mcmc
+00005d10: 7472 6565 2e63 7472 6c0a 2d2d 2d2d 2d6d  tree.ctrl.-----m
+00005d20: 636d 6374 7265 652e 6f75 740a 2d2d 2d2d  cmctree.out.----
+00005d30: 2d6d 636d 632e 7478 740a 2d2d 2d2d 2d72  -mcmc.txt.-----r
+00005d40: 6174 6573 0a2d 2d2d 2d2d 7273 740a 2d2d  ates.-----rst.--
+00005d50: 2d2d 2d72 7374 310a 2d2d 2d2d 2d72 7562  ---rst1.-----rub
+00005d60: 0a2d 2d2d 2d2d 746d 7030 3030 312e 6374  .-----tmp0001.ct
+00005d70: 6c0a 2d2d 2d2d 2d74 6d70 3030 3031 2e6f  l.-----tmp0001.o
+00005d80: 7574 0a2d 2d2d 2d2d 746d 7030 3030 312e  ut.-----tmp0001.
+00005d90: 7472 6565 730a 2d2d 2d2d 2d74 6d70 3030  trees.-----tmp00
+00005da0: 3031 2e74 7874 0a60 6060 0a54 6865 2060  01.txt.```.The `
+00005db0: 436f 6e63 6174 656e 6174 6564 2e70 616c  Concatenated.pal
+00005dc0: 6e60 2061 6e64 2060 436f 6e63 6174 656e  n` and `Concaten
+00005dd0: 6174 6564 2e70 616c 6e2e 7061 6d6c 6020  ated.paln.paml` 
+00005de0: 6172 6520 7468 6520 636f 6e63 6174 656e  are the concaten
+00005df0: 6174 6564 2070 726f 7465 696e 2061 6c69  ated protein ali
+00005e00: 676e 6d65 6e74 7320 696e 2066 6173 7461  gnments in fasta
+00005e10: 2061 6e64 2070 616d 6c20 666f 726d 6174   and paml format
+00005e20: 2e20 5468 6520 6047 3253 2e4d 6170 6020  . The `G2S.Map` 
+00005e30: 6973 2074 6865 206d 6170 2062 6574 7765  is the map betwe
+00005e40: 656e 2067 656e 6520 616e 6420 7370 6563  en gene and spec
+00005e50: 6965 7320 6e61 6d65 732e 2054 6865 2060  ies names. The `
+00005e60: 4746 3030 3030 3030 3031 2e70 616c 6e60  GF00000001.paln`
+00005e70: 2c2e 2e20 616e 6420 6047 4630 3030 3030  ,.. and `GF00000
+00005e80: 3138 372e 7061 6c6e 6020 6172 6520 7468  187.paln` are th
+00005e90: 6520 7072 6f74 6569 6e20 616c 6967 6e6d  e protein alignm
+00005ea0: 656e 7473 2066 6f72 2065 6163 6820 6765  ents for each ge
+00005eb0: 6e65 2066 616d 696c 792e 2054 6865 2060  ne family. The `
+00005ec0: 6d63 6d63 7472 6565 6020 7375 6266 6f6c  mcmctree` subfol
+00005ed0: 6465 7220 636f 6e74 6169 6e73 2074 6865  der contains the
+00005ee0: 2064 6174 696e 6720 7265 7375 6c74 7320   dating results 
+00005ef0: 666f 7220 7468 6520 636f 6e63 6174 656e  for the concaten
+00005f00: 6174 6564 2066 616d 696c 7920 2861 6e64  ated family (and
+00005f10: 2070 6572 2067 656e 6520 6661 6d69 6c79   per gene family
+00005f20: 2069 6620 7365 7429 2e20 5468 6520 6465   if set). The de
+00005f30: 6570 6572 2060 436f 6e63 6174 656e 6174  eper `Concatenat
+00005f40: 6564 6020 7375 6266 6f6c 6465 7220 636f  ed` subfolder co
+00005f50: 6e74 6169 6e73 2074 6865 2064 6174 696e  ntains the datin
+00005f60: 6720 7265 7375 6c74 7320 6f66 2063 6f6e  g results of con
+00005f70: 6361 7465 6e61 7465 6420 7072 6f74 6569  catenated protei
+00005f80: 6e20 616c 6967 6e6d 656e 7420 286f 7220  n alignment (or 
+00005f90: 6e75 636c 656f 7469 6465 2061 6c69 676e  nucleotide align
+00005fa0: 6d65 6e74 2069 6620 7365 7429 2e20 5468  ment if set). Th
+00005fb0: 6520 6465 6570 6573 7420 6070 6570 6020  e deepest `pep` 
+00005fc0: 7375 6266 6f6c 6465 7220 636f 6e74 6169  subfolder contai
+00005fd0: 6e73 2074 6865 2066 696e 616c 2064 6174  ns the final dat
+00005fe0: 696e 6720 7265 7375 6c74 7320 286f 6620  ing results (of 
+00005ff0: 636f 6e63 6174 656e 6174 6564 2070 726f  concatenated pro
+00006000: 7465 696e 2061 6c69 676e 6d65 6e74 2069  tein alignment i
+00006010: 6e20 7468 6973 2063 6173 6529 2e20 506c  n this case). Pl
+00006020: 6561 7365 2072 6566 6572 2074 6f20 606d  ease refer to `m
+00006030: 636d 6374 7265 6560 206d 616e 7561 6c20  cmctree` manual 
+00006040: 666f 7220 6465 7461 696c 6564 2064 6573  for detailed des
+00006050: 6372 6970 7469 6f6e 206f 6620 6561 6368  cription of each
+00006060: 2066 696c 6520 7072 6f64 7563 6564 2062   file produced b
+00006070: 7920 606d 636d 6374 7265 6560 2e20 5468  y `mcmctree`. Th
+00006080: 6520 696d 706f 7274 616e 7420 7265 7375  e important resu
+00006090: 6c74 2066 696c 6573 2061 7265 2060 4669  lt files are `Fi
+000060a0: 6754 7265 652e 7472 6560 2c20 606d 636d  gTree.tre`, `mcm
+000060b0: 6374 7265 652e 6f75 7460 2061 6e64 2060  ctree.out` and `
+000060c0: 6d63 6d63 2e74 7874 6020 7768 6963 6820  mcmc.txt` which 
+000060d0: 646f 6375 6d65 6e74 2074 6865 2066 696e  document the fin
+000060e0: 616c 2064 6174 6520 6573 7469 6d61 7469  al date estimati
+000060f0: 6f6e 2c20 6c6f 6720 696e 666f 726d 6174  on, log informat
+00006100: 696f 6e20 616e 6420 706f 7374 6572 696f  ion and posterio
+00006110: 7220 7361 6d70 6c65 7320 666f 7220 6561  r samples for ea
+00006120: 6368 206e 6f64 6520 7265 7370 6563 7469  ch node respecti
+00006130: 7665 6c79 2e0a 200a 2323 2050 6172 616d  vely.. .## Param
+00006140: 6574 6572 730a 0a54 6865 7265 2061 7265  eters..There are
+00006150: 2037 206d 6169 6e20 7072 6f67 7261 6d73   7 main programs
+00006160: 2069 6e20 6077 6764 2076 3260 3a20 6064   in `wgd v2`: `d
+00006170: 6d64 602c 6066 6f63 7573 602c 606b 7364  md`,`focus`,`ksd
+00006180: 602c 606d 6978 602c 6070 6561 6b60 2c60  `,`mix`,`peak`,`
+00006190: 7379 6e60 2c60 7669 7a60 2e20 4865 7265  syn`,`viz`. Here
+000061a0: 6166 7465 7220 7765 2077 696c 6c20 7072  after we will pr
+000061b0: 6f76 6964 6520 6120 6465 7461 696c 6564  ovide a detailed
+000061c0: 2065 6c75 6369 6461 7469 6f6e 206f 6e20   elucidation on 
+000061d0: 6561 6368 206f 6620 7468 6520 7072 6f67  each of the prog
+000061e0: 7261 6d20 616e 6420 6974 7320 6173 736f  ram and its asso
+000061f0: 6369 6174 6564 2070 6172 616d 6574 6572  ciated parameter
+00006200: 732e 2050 6c65 6173 6520 7265 6665 7220  s. Please refer 
+00006210: 746f 2074 6865 205b 5573 6167 655d 2823  to the [Usage](#
+00006220: 7573 6167 6529 2066 6f72 2074 6865 2073  usage) for the s
+00006230: 6365 6e61 7269 6f73 2074 6f20 7768 6963  cenarios to whic
+00006240: 6820 6561 6368 2070 6172 616d 6574 6572  h each parameter
+00006250: 2061 7070 6c69 6573 2e0a 0a54 6865 2070   applies...The p
+00006260: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
+00006270: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+00006280: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
+00006290: 7768 6f6c 6520 7061 7261 6e6f 6d65 2c20  whole paranome, 
+000062a0: 5242 4873 2028 5265 6369 7072 6f63 616c  RBHs (Reciprocal
+000062b0: 2042 6573 7420 4869 7473 292c 204d 5242   Best Hits), MRB
+000062c0: 4873 2028 4d75 6c74 6970 6c65 2052 6563  Hs (Multiple Rec
+000062d0: 6970 726f 6361 6c20 4265 7374 2048 6974  iprocal Best Hit
+000062e0: 7329 2c20 6f72 7468 6f67 726f 7570 7320  s), orthogroups 
+000062f0: 616e 6420 736f 6d65 206f 7468 6572 206f  and some other o
+00006300: 7274 686f 6772 6f75 702d 7265 6c61 7465  rthogroup-relate
+00006310: 6420 6675 6e63 7469 6f6e 732c 2069 6e63  d functions, inc
+00006320: 6c75 6469 6e67 2063 6972 6375 6d73 6372  luding circumscr
+00006330: 6970 7469 6f6e 206f 6620 6e65 7374 6564  iption of nested
+00006340: 2073 696e 676c 652d 636f 7079 206f 7274   single-copy ort
+00006350: 686f 6772 6f75 7073 2028 4e53 4f47 7329  hogroups (NSOGs)
+00006360: 2c20 756e 6269 6173 6564 2074 6573 7420  , unbiased test 
+00006370: 6f66 2073 696e 676c 652d 636f 7079 206f  of single-copy o
+00006380: 7274 686f 6772 6f75 7073 2028 534f 4773  rthogroups (SOGs
+00006390: 2920 6f76 6572 206d 6973 7369 6e67 2069  ) over missing i
+000063a0: 6e70 6172 616c 6f67 732c 2063 6f6e 7374  nparalogs, const
+000063b0: 7275 6374 696f 6e20 6f66 2042 5553 434f  ruction of BUSCO
+000063c0: 2d67 7569 6465 6420 7369 6e67 6c65 2d63  -guided single-c
+000063d0: 6f70 7920 6f72 7468 6f67 726f 7570 7320  opy orthogroups 
+000063e0: 2853 4f47 7329 2c61 6e64 2074 6865 2063  (SOGs),and the c
+000063f0: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
+00006400: 656e 6365 2069 6e66 6572 656e 6365 206f  ence inference o
+00006410: 6620 7068 796c 6f67 656e 792e 0a60 6060  f phylogeny..```
+00006420: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
+00006430: 6573 2028 6f70 7469 6f6e 290a 2d2d 2d2d  es (option).----
+00006440: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006450: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006460: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006470: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006480: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c  ------------.-o,
+00006490: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+000064a0: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+000064b0: 2064 6566 6175 6c74 2077 6764 5f64 6d64   default wgd_dmd
+000064c0: 0a2d 742c 202d 2d74 6d70 6469 722c 2074  .-t, --tmpdir, t
+000064d0: 6865 2074 656d 706f 7261 7279 2077 6f72  he temporary wor
+000064e0: 6b69 6e67 2064 6972 6563 746f 7279 2c20  king directory, 
+000064f0: 6465 6661 756c 7420 4e6f 6e65 2c20 6966  default None, if
+00006500: 204e 6f6e 6520 7761 7320 6769 7665 6e2c   None was given,
+00006510: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
+00006520: 6420 6265 2061 7373 6967 6e65 6420 7261  d be assigned ra
+00006530: 6e64 6f6d 206e 616d 6573 2069 6e20 6375  ndom names in cu
+00006540: 7272 656e 7420 6469 7265 6374 6f72 7920  rrent directory 
+00006550: 616e 6420 6175 746f 6d61 7465 6c79 2072  and automately r
+00006560: 656d 6f76 6564 2061 7420 7468 6520 636f  emoved at the co
+00006570: 6d70 6c65 7469 6f6e 206f 6620 7072 6f67  mpletion of prog
+00006580: 7261 6d2c 2065 6c73 6520 7468 6520 746d  ram, else the tm
+00006590: 7064 6972 2077 6f75 6c64 2062 6520 6b65  pdir would be ke
+000065a0: 7074 0a2d 702c 202d 2d70 726f 742c 2066  pt.-p, --prot, f
+000065b0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+000065c0: 6865 7220 7573 696e 6720 7072 6f74 6569  her using protei
+000065d0: 6e20 6f72 206e 7563 6c65 6f74 6964 6520  n or nucleotide 
+000065e0: 7365 7175 656e 6365 730a 2d63 2c20 2d2d  sequences.-c, --
+000065f0: 6373 636f 7265 2c20 7468 6520 632d 7363  cscore, the c-sc
+00006600: 6f72 6520 746f 2072 6573 7472 6963 7420  ore to restrict 
+00006610: 7468 6520 686f 6d6f 6c6f 6720 7369 6d69  the homolog simi
+00006620: 6c61 7269 7479 206f 6620 4d52 4248 732c  larity of MRBHs,
+00006630: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
+00006640: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
+00006650: 2c20 7468 6520 632d 7363 6f72 6520 6675  , the c-score fu
+00006660: 6e63 696f 6e20 776f 756c 646e 2774 2062  ncion wouldn't b
+00006670: 6520 6163 7469 7661 7465 642c 2065 6c73  e activated, els
+00006680: 6520 6578 7065 6374 696e 6720 6120 6465  e expecting a de
+00006690: 6369 6d61 6c20 7769 7468 696e 2074 6865  cimal within the
+000066a0: 2072 616e 6765 206f 6620 3020 616e 6420   range of 0 and 
+000066b0: 310a 2d49 2c20 2d2d 696e 666c 6174 696f  1.-I, --inflatio
+000066c0: 6e2c 2074 6865 2069 6e66 6c61 7469 6f6e  n, the inflation
+000066d0: 2066 6163 746f 7220 666f 7220 4d43 4c20   factor for MCL 
+000066e0: 7072 6f67 7261 6d2c 2064 6566 6175 6c74  program, default
+000066f0: 2032 2e30 2c20 7769 7468 2068 6967 6865   2.0, with highe
+00006700: 7220 7661 6c75 6520 6c65 6164 696e 6720  r value leading 
+00006710: 746f 206d 6f72 6520 6275 7420 736d 616c  to more but smal
+00006720: 6c65 7220 636c 7573 7465 7273 0a2d 652c  ler clusters.-e,
+00006730: 202d 2d65 7661 6c2c 2074 6865 2065 2d76   --eval, the e-v
+00006740: 616c 7565 2063 7574 2d6f 6666 2066 6f72  alue cut-off for
+00006750: 2073 696d 696c 6172 6974 7920 696e 2064   similarity in d
+00006760: 6961 6d6f 6e64 2061 6e64 2f6f 7220 686d  iamond and/or hm
+00006770: 6d65 722c 2064 6566 6175 6c74 2031 652d  mer, default 1e-
+00006780: 3130 0a2d 2d74 6f5f 7374 6f70 2c20 666c  10.--to_stop, fl
+00006790: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+000067a0: 6572 2074 6f20 7472 616e 736c 6174 6520  er to translate 
+000067b0: 7468 726f 7567 6820 5354 4f50 2063 6f64  through STOP cod
+000067c0: 6f6e 732c 2069 6620 7468 6520 666c 6167  ons, if the flag
+000067d0: 2077 6173 2073 6574 2c20 7472 616e 736c   was set, transl
+000067e0: 6174 696f 6e20 7769 6c6c 2062 6520 7465  ation will be te
+000067f0: 726d 696e 6174 6564 2061 7420 7468 6520  rminated at the 
+00006800: 6669 7273 7420 696e 2d66 7261 6d65 2073  first in-frame s
+00006810: 746f 7020 636f 646f 6e2c 2065 6c73 6520  top codon, else 
+00006820: 6120 6675 6c6c 2074 7261 6e73 6c61 7469  a full translati
+00006830: 6f6e 2063 6f6e 7469 6e75 696e 6720 6f6e  on continuing on
+00006840: 2070 6173 7369 6e67 2061 6e79 2073 746f   passing any sto
+00006850: 7020 636f 646f 6e73 2077 6f75 6c64 2062  p codons would b
+00006860: 6520 696e 6974 6961 7465 640a 2d2d 6364  e initiated.--cd
+00006870: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
+00006880: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
+00006890: 7472 616e 736c 6174 6520 7468 6520 636f  translate the co
+000068a0: 6d70 6c65 7465 2043 4453 2074 6861 7420  mplete CDS that 
+000068b0: 7374 6172 7473 2077 6974 6820 6120 7661  starts with a va
+000068c0: 6c69 6420 7374 6172 7420 636f 646f 6e20  lid start codon 
+000068d0: 616e 6420 6f6e 6c79 2063 6f6e 7461 696e  and only contain
+000068e0: 7320 6120 7369 6e67 6c65 2069 6e2d 6672  s a single in-fr
+000068f0: 616d 6520 7374 6f70 2063 6f64 6f6e 2061  ame stop codon a
+00006900: 7420 7468 6520 656e 6420 616e 6420 6d75  t the end and mu
+00006910: 7374 2062 6520 6469 7669 6461 626c 6520  st be dividable 
+00006920: 6279 2074 6872 6565 2c20 6966 2074 6865  by three, if the
+00006930: 2066 6c61 6720 7761 7320 7365 742c 206f   flag was set, o
+00006940: 6e6c 7920 7468 6520 636f 6d70 6c65 7465  nly the complete
+00006950: 2043 4453 2077 6f75 6c64 2062 6520 7472   CDS would be tr
+00006960: 616e 736c 6174 6564 0a2d 662c 202d 2d66  anslated.-f, --f
+00006970: 6f63 7573 2c20 7468 6520 7370 6563 6965  ocus, the specie
+00006980: 7320 746f 2062 6520 6d65 7267 6564 206f  s to be merged o
+00006990: 6e20 6c6f 6361 6c20 4d52 4248 732c 2064  n local MRBHs, d
+000069a0: 6566 6175 6c74 204e 6f6e 652c 2069 6620  efault None, if 
+000069b0: 4e6f 6e65 2077 6173 2067 6976 656e 2c20  None was given, 
+000069c0: 7468 6520 6c6f 6361 6c20 4d52 4248 7320  the local MRBHs 
+000069d0: 776f 756c 646e 2774 2062 6520 696e 6665  wouldn't be infe
+000069e0: 7272 6564 0a2d 6170 2c20 2d2d 616e 6368  rred.-ap, --anch
+000069f0: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
+00006a00: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
+00006a10: 2066 696c 6520 6672 6f6d 2069 2d61 6468   file from i-adh
+00006a20: 6f72 6520 666f 7220 636f 6e73 7472 7563  ore for construc
+00006a30: 7469 6e67 2074 6865 206f 7274 686f 6772  ting the orthogr
+00006a40: 6f75 7073 2077 6974 6820 616e 6368 6f72  oups with anchor
+00006a50: 2070 6169 7273 2c20 6465 6661 756c 7420   pairs, default 
+00006a60: 4e6f 6e65 0a2d 736d 2c20 2d2d 7365 676d  None.-sm, --segm
+00006a70: 656e 7473 2c20 7468 6520 7365 676d 656e  ents, the segmen
+00006a80: 7473 2064 6174 6166 696c 6520 7573 6564  ts datafile used
+00006a90: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
+00006aa0: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
+00006ab0: 6973 2069 6620 696e 6974 6961 7465 642c  is if initiated,
+00006ac0: 2064 6566 6175 6c74 204e 6f6e 650a 2d6c   default None.-l
+00006ad0: 652c 202d 2d6c 6973 7465 6c65 6d65 6e74  e, --listelement
+00006ae0: 732c 2074 6865 206c 6973 7473 6567 6d65  s, the listsegme
+00006af0: 6e74 7320 6461 7461 2066 696c 6520 7573  nts data file us
+00006b00: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
+00006b10: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+00006b20: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
+00006b30: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
+00006b40: 2d67 742c 202d 2d67 656e 6574 6162 6c65  -gt, --genetable
+00006b50: 2c20 7468 6520 6765 6e65 2074 6162 6c65  , the gene table
+00006b60: 2064 6174 6166 696c 6520 7573 6564 2069   datafile used i
+00006b70: 6e20 636f 6c6c 696e 6561 7220 636f 616c  n collinear coal
+00006b80: 6573 6365 6e63 6520 616e 616c 7973 6973  escence analysis
+00006b90: 2069 6620 696e 6974 6961 7465 642c 2064   if initiated, d
+00006ba0: 6566 6175 6c74 204e 6f6e 650a 2d63 6f63  efault None.-coc
+00006bb0: 2c20 2d2d 636f 6c6c 696e 6561 7263 6f61  , --collinearcoa
+00006bc0: 6c65 7363 656e 6365 2c20 666c 6167 206f  lescence, flag o
+00006bd0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00006be0: 6f20 696e 6974 6961 7465 2074 6865 2063  o initiate the c
+00006bf0: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
+00006c00: 656e 6365 2061 6e61 6c79 7369 732c 2069  ence analysis, i
+00006c10: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00006c20: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
+00006c30: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00006c40: 7465 640a 2d6b 662c 202d 2d6b 6565 7066  ted.-kf, --keepf
+00006c50: 6173 7461 2c20 666c 6167 206f 7074 696f  asta, flag optio
+00006c60: 6e2c 2077 6865 7468 6572 2074 6f20 6f75  n, whether to ou
+00006c70: 7470 7574 2074 6865 2073 6571 7565 6e63  tput the sequenc
+00006c80: 6520 696e 666f 726d 6174 696f 6e20 6f66  e information of
+00006c90: 204d 5242 4873 2c20 6966 2074 6865 2066   MRBHs, if the f
+00006ca0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+00006cb0: 2073 6571 7565 6e63 6573 206f 6620 4d52   sequences of MR
+00006cc0: 4248 7320 776f 756c 6420 6265 2069 6e20  BHs would be in 
+00006cd0: 6f75 7470 7574 0a2d 6b64 2c20 2d2d 6b65  output.-kd, --ke
+00006ce0: 6570 6475 706c 6963 6174 6573 2c20 666c  epduplicates, fl
+00006cf0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+00006d00: 6572 2074 6f20 616c 6c6f 7720 7468 6520  er to allow the 
+00006d10: 7361 6d65 2067 656e 6520 746f 206f 6363  same gene to occ
+00006d20: 7572 2069 6e20 6469 6666 6572 656e 7420  ur in different 
+00006d30: 4d52 4248 7320 286f 6e6c 7920 6d65 616e  MRBHs (only mean
+00006d40: 696e 6766 756c 2077 6865 6e20 7468 6520  ingful when the 
+00006d50: 6373 636f 7265 2077 6173 2075 7365 6429  cscore was used)
+00006d60: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00006d70: 7320 7365 742c 2074 6865 2073 616d 6520  s set, the same 
+00006d80: 6765 6e65 2063 6f75 6c64 2062 6520 6173  gene could be as
+00006d90: 7369 676e 6564 2074 6f20 6469 6666 6572  signed to differ
+00006da0: 656e 7420 4d52 4248 730a 2d67 6d2c 202d  ent MRBHs.-gm, -
+00006db0: 2d67 6c6f 6261 6c6d 7262 682c 2066 6c61  -globalmrbh, fla
+00006dc0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00006dd0: 7220 746f 2069 6e69 7469 6174 6520 676c  r to initiate gl
+00006de0: 6f62 616c 204d 5242 4873 2063 6f6e 7374  obal MRBHs const
+00006df0: 7275 6374 696f 6e2c 2069 6620 7468 6520  ruction, if the 
 00006e00: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00006e10: 6520 6865 7572 6973 7469 6320 6d65 7468  e heuristic meth
-00006e20: 6f64 2077 6f75 6c64 2062 6520 696e 6974  od would be init
-00006e30: 6961 7465 640a 2d6b 632c 202d 2d6b 7363  iated.-kc, --ksc
-00006e40: 7574 6f66 662c 2074 6865 204b 7320 7361  utoff, the Ks sa
-00006e50: 7475 7261 7469 6f6e 2063 7574 6f66 6620  turation cutoff 
-00006e60: 696e 2064 6174 696e 672c 2064 6566 6175  in dating, defau
-00006e70: 6c74 2035 0a2d 2d6b 6565 7074 6d70 6669  lt 5.--keeptmpfi
-00006e80: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
-00006e90: 7768 6574 6865 7220 746f 206b 6565 7020  whether to keep 
-00006ea0: 7465 6d70 6f72 6172 7920 6669 6775 7265  temporary figure
-00006eb0: 7320 696e 2070 6561 6b20 6669 6e64 696e  s in peak findin
-00006ec0: 6720 7072 6f63 6573 732c 2069 6620 7468  g process, if th
-00006ed0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00006ee0: 7468 6f73 6520 6669 6775 7265 7320 776f  those figures wo
-00006ef0: 756c 6420 6265 206b 6570 740a 6060 600a  uld be kept.```.
-00006f00: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
-00006f10: 6420 7379 6e60 2063 616e 2072 6561 6c69  d syn` can reali
-00006f20: 7a65 2074 6865 2069 6e74 7261 2d20 616e  ze the intra- an
-00006f30: 6420 696e 7465 722d 7370 6563 6966 6963  d inter-specific
-00006f40: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
-00006f50: 6365 2e0a 6060 600a 7767 6420 7379 6e20  ce..```.wgd syn 
-00006f60: 6661 6d69 6c69 6573 2067 6666 7320 286f  families gffs (o
-00006f70: 7074 696f 6e29 0a2d 2d2d 2d2d 2d2d 2d2d  ption).---------
-00006f80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006f90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006fa0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006fb0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00006fc0: 2d2d 2d2d 2d2d 2d0a 2d6b 732c 202d 2d6b  -------.-ks, --k
-00006fd0: 735f 6469 7374 7269 6275 7469 6f6e 2c20  s_distribution, 
-00006fe0: 6b73 2064 6973 7472 6962 7574 696f 6e20  ks distribution 
-00006ff0: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
-00007000: 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574  t None.-o, --out
-00007010: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
-00007020: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
-00007030: 6c74 2077 6764 5f73 796e 0a2d 662c 202d  lt wgd_syn.-f, -
-00007040: 2d66 6561 7475 7265 2c20 7468 6520 6665  -feature, the fe
-00007050: 6174 7572 6520 666f 7220 7061 7273 696e  ature for parsin
-00007060: 6720 6765 6e65 2049 4473 2066 726f 6d20  g gene IDs from 
-00007070: 4746 4620 6669 6c65 732c 2064 6566 6175  GFF files, defau
-00007080: 6c74 2067 656e 650a 2d61 2c20 2d2d 6174  lt gene.-a, --at
-00007090: 7472 6962 7574 652c 2074 6865 2061 7474  tribute, the att
-000070a0: 7269 6275 7465 2066 6f72 2070 6172 7369  ribute for parsi
-000070b0: 6e67 2074 6865 2067 656e 6520 4944 7320  ng the gene IDs 
-000070c0: 6672 6f6d 2074 6865 2047 4646 2066 696c  from the GFF fil
-000070d0: 6573 2c20 6465 6661 756c 7420 4944 0a2d  es, default ID.-
-000070e0: 6174 672c 202d 2d61 6464 6974 696f 6e61  atg, --additiona
-000070f0: 6c67 6666 696e 666f 2c20 7468 6520 6665  lgffinfo, the fe
-00007100: 6174 7572 6520 616e 6420 6174 7472 6962  ature and attrib
-00007110: 7574 6520 696e 666f 726d 6174 696f 6e20  ute information 
-00007120: 6f66 2061 6464 6974 696f 6e61 6c20 6766  of additional gf
-00007130: 6633 2066 696c 6573 2069 6620 6469 6666  f3 files if diff
-00007140: 6572 656e 7420 696e 2074 6865 2066 6f72  erent in the for
-00007150: 6d61 7420 6f66 2028 6665 6174 7572 653b  mat of (feature;
-00007160: 6174 7472 6962 7574 6529 272c 2064 6566  attribute)', def
-00007170: 6175 6c74 204e 6f6e 650a 2d6d 6c2c 202d  ault None.-ml, -
-00007180: 2d6d 696e 6c65 6e2c 2074 6865 206d 696e  -minlen, the min
-00007190: 696d 756d 206c 656e 6774 6820 6f66 2061  imum length of a
-000071a0: 2073 6361 6666 6f6c 6420 746f 2062 6520   scaffold to be 
-000071b0: 696e 636c 7564 6564 2069 6e20 646f 7470  included in dotp
-000071c0: 6c6f 742c 2064 6566 6175 6c74 202d 312c  lot, default -1,
-000071d0: 2069 6620 2d31 2077 6173 2073 6574 2c20   if -1 was set, 
-000071e0: 7468 6520 3130 2520 6f66 2074 6865 206c  the 10% of the l
-000071f0: 6f6e 6765 7374 2073 6361 6666 6f6c 6420  ongest scaffold 
-00007200: 776f 756c 6420 6265 2073 6574 0a2d 6d73  would be set.-ms
-00007210: 2c20 2d2d 6d61 7873 697a 652c 2074 6865  , --maxsize, the
-00007220: 206d 6178 696d 756d 2066 616d 696c 7920   maximum family 
-00007230: 7369 7a65 2074 6f20 6265 2069 6e63 6c75  size to be inclu
-00007240: 6465 642c 2064 6566 6175 6c74 2032 3030  ded, default 200
-00007250: 0a2d 722c 202d 2d6b 735f 7261 6e67 652c  .-r, --ks_range,
-00007260: 2074 6865 204b 7320 7261 6e67 6520 696e   the Ks range in
-00007270: 2063 6f6c 6f72 6564 2064 6f74 706c 6f74   colored dotplot
-00007280: 2c20 6465 6661 756c 7420 2830 2c20 3529  , default (0, 5)
-00007290: 0a2d 2d69 6164 686f 7265 5f6f 7074 696f  .--iadhore_optio
-000072a0: 6e73 2c20 7468 6520 7061 7261 6d65 7465  ns, the paramete
-000072b0: 7220 7365 7474 696e 6720 696e 2069 6164  r setting in iad
-000072c0: 686f 7265 2c20 6465 6661 756c 7420 6173  hore, default as
-000072d0: 2061 2073 7472 696e 6720 6f66 206c 656e   a string of len
-000072e0: 6774 6820 7a65 726f 0a2d 6d67 2c20 2d2d  gth zero.-mg, --
-000072f0: 6d69 6e73 6567 6c65 6e2c 2074 6865 206d  minseglen, the m
-00007300: 696e 696d 756d 206c 656e 6774 6820 6f66  inimum length of
-00007310: 2073 6567 6d65 6e74 7320 746f 2069 6e63   segments to inc
-00007320: 6c75 6465 2069 6e20 7261 7469 6f20 6966  lude in ratio if
-00007330: 203c 3d20 312c 2064 6566 6175 6c74 2031   <= 1, default 1
-00007340: 3030 3030 0a2d 6b72 2c20 2d2d 6b65 6570  0000.-kr, --keep
-00007350: 7265 6475 6e2c 2066 6c61 6720 6f70 7469  redun, flag opti
-00007360: 6f6e 2c20 7768 6574 6865 7220 746f 206b  on, whether to k
-00007370: 6565 7020 7265 6475 6e64 616e 7420 6d75  eep redundant mu
-00007380: 6c74 6970 6c69 636f 6e73 2c20 6966 2074  ltiplicons, if t
-00007390: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000073a0: 2074 6865 2072 6564 756e 6461 6e74 206d   the redundant m
-000073b0: 756c 7469 706c 6963 6f6e 7320 776f 756c  ultiplicons woul
-000073c0: 6420 6265 206b 6570 740a 2d6d 676e 2c20  d be kept.-mgn, 
-000073d0: 2d2d 6d69 6e67 656e 656e 756d 2c20 7468  --mingenenum, th
-000073e0: 6520 6d69 6e69 6d75 6d20 6e75 6d62 6572  e minimum number
-000073f0: 206f 6620 6765 6e65 7320 666f 7220 6120   of genes for a 
-00007400: 7365 676d 656e 7420 746f 2062 6520 636f  segment to be co
-00007410: 6e73 6964 6572 6564 2c20 6465 6661 756c  nsidered, defaul
-00007420: 7420 3330 0a2d 6473 2c20 2d2d 646f 7473  t 30.-ds, --dots
-00007430: 697a 652c 2074 6865 2064 6f74 2073 697a  ize, the dot siz
-00007440: 6520 696e 2064 6f74 2070 6c6f 742c 2064  e in dot plot, d
-00007450: 6566 6175 6c74 2030 2e33 0a2d 6161 2c20  efault 0.3.-aa, 
-00007460: 2d2d 6170 616c 7068 612c 2074 6865 206f  --apalpha, the o
-00007470: 7061 6369 7479 206f 6620 616e 6368 6f72  pacity of anchor
-00007480: 2064 6f74 732c 2064 6566 6175 6c74 2031   dots, default 1
-00007490: 0a2d 6861 2c20 2d2d 686f 616c 7068 612c  .-ha, --hoalpha,
-000074a0: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
-000074b0: 686f 6d6f 6c6f 6720 646f 7473 2c20 6465  homolog dots, de
-000074c0: 6661 756c 7420 300a 2d73 7274 2c20 2d2d  fault 0.-srt, --
-000074d0: 7368 6f77 7265 616c 7469 636b 2c20 666c  showrealtick, fl
-000074e0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-000074f0: 6572 2074 6f20 7368 6f77 2074 6865 2072  er to show the r
-00007500: 6561 6c20 7469 636b 2069 6e20 6765 6e65  eal tick in gene
-00007510: 7320 6f72 2062 6173 6573 2c20 6966 2074  s or bases, if t
-00007520: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00007530: 2074 6865 2072 6561 6c20 7469 636b 2077   the real tick w
-00007540: 6f75 6c64 2062 6520 7368 6f77 6564 0a2d  ould be showed.-
-00007550: 746c 732c 202d 2d74 6963 6b6c 6162 656c  tls, --ticklabel
-00007560: 7369 7a65 2c20 7468 6520 6c61 6265 6c20  size, the label 
-00007570: 7369 7a65 206f 6620 7469 636b 2c20 6465  size of tick, de
-00007580: 6661 756c 7420 350a 2d67 722c 202d 2d67  fault 5.-gr, --g
-00007590: 6973 7472 622c 2066 6c61 6720 6f70 7469  istrb, flag opti
-000075a0: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
-000075b0: 7365 2067 6973 745f 7261 696e 626f 7720  se gist_rainbow 
-000075c0: 6173 2063 6f6c 6f72 206d 6170 206f 6620  as color map of 
-000075d0: 646f 7470 6c6f 740a 2d6e 2c20 2d2d 6e74  dotplot.-n, --nt
-000075e0: 6872 6561 6473 2c20 7468 6520 6e75 6d62  hreads, the numb
-000075f0: 6572 206f 6620 7468 7265 6164 7320 746f  er of threads to
-00007600: 2075 7365 2069 6e20 7379 6e74 656e 7920   use in synteny 
-00007610: 696e 6665 7265 6e63 652c 2064 6566 6175  inference, defau
-00007620: 6c74 2034 0a60 6060 0a0a 5468 6520 7072  lt 4.```..The pr
-00007630: 6f67 7261 6d20 6077 6764 2076 697a 6020  ogram `wgd viz` 
-00007640: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
-00007650: 7669 7375 616c 697a 6174 696f 6e20 6f66  visualization of
-00007660: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00007670: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-00007680: 6e20 616e 6420 7379 6e74 656e 792e 0a60  n and synteny..`
-00007690: 6060 0a77 6764 2076 697a 2028 6f70 7469  ``.wgd viz (opti
-000076a0: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
-000076b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000076c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000076d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000076e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000076f0: 2d2d 2d2d 0a2d 642c 202d 2d64 6174 6166  ----.-d, --dataf
-00007700: 696c 652c 2074 6865 204b 7320 6461 7461  ile, the Ks data
-00007710: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00007720: 6e65 0a2d 6f2c 202d 2d6f 7574 6469 722c  ne.-o, --outdir,
-00007730: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
-00007740: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
-00007750: 6764 5f76 697a 0a2d 7372 2c20 2d2d 7370  gd_viz.-sr, --sp
-00007760: 6169 722c 2074 6865 2073 7065 6369 6573  air, the species
-00007770: 2070 6169 7220 746f 2062 6520 706c 6f74   pair to be plot
-00007780: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
-00007790: 652c 2074 6869 7320 6f70 7469 6f6e 2063  e, this option c
-000077a0: 616e 2062 6520 7072 6f76 6964 6564 206d  an be provided m
-000077b0: 756c 7469 706c 6520 7469 6d65 730a 2d66  ultiple times.-f
-000077c0: 612c 202d 2d66 6f63 7573 3261 6c6c 2c20  a, --focus2all, 
-000077d0: 7365 7420 666f 6361 6c20 7370 6563 6965  set focal specie
-000077e0: 7320 616e 6420 6c65 7420 7370 6563 6965  s and let specie
-000077f0: 7320 7061 6972 2074 6f20 6265 2062 6574  s pair to be bet
-00007800: 7765 656e 2066 6f63 616c 2061 6e64 2061  ween focal and a
-00007810: 6c6c 2074 6865 2072 656d 6169 6e69 6e67  ll the remaining
-00007820: 2073 7065 6369 6573 2c20 6465 6661 756c   species, defaul
-00007830: 7420 4e6f 6e65 0a2d 6773 2c20 2d2d 6773  t None.-gs, --gs
-00007840: 6d61 702c 2074 6865 2067 656e 6520 6e61  map, the gene na
-00007850: 6d65 2d73 7065 6369 6573 206e 616d 6520  me-species name 
-00007860: 6d61 702c 2064 6566 6175 6c74 204e 6f6e  map, default Non
-00007870: 650a 2d73 702c 202d 2d73 7065 6369 6573  e.-sp, --species
-00007880: 7472 6565 2c20 7468 6520 7370 6563 6965  tree, the specie
-00007890: 7320 7472 6565 2074 6f20 7065 7266 6f72  s tree to perfor
-000078a0: 6d20 7261 7465 2063 6f72 7265 6374 696f  m rate correctio
-000078b0: 6e2c 2064 6566 6175 6c74 204e 6f6e 652c  n, default None,
-000078c0: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
-000078d0: 656e 2c20 7468 6520 7261 7465 2063 6f72  en, the rate cor
-000078e0: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
-000078f0: 2077 6f75 6c64 2062 6520 6361 6c6c 6564   would be called
-00007900: 206f 6666 0a2d 706b 2c20 2d2d 706c 6f74   off.-pk, --plot
-00007910: 6b64 652c 2066 6c61 6720 6f70 7469 6f6e  kde, flag option
-00007920: 2c20 7768 6574 6865 7220 746f 2070 6c6f  , whether to plo
-00007930: 7420 6b64 6520 6375 7276 6520 7570 6f6e  t kde curve upon
-00007940: 2068 6973 746f 6772 616d 2c20 6966 2074   histogram, if t
-00007950: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00007960: 206b 6465 2063 7572 7665 2077 6f75 6c64   kde curve would
-00007970: 2062 6520 6164 6465 640a 2d72 772c 202d   be added.-rw, -
-00007980: 2d72 6577 6569 6768 742c 2066 6c61 6720  -reweight, flag 
-00007990: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000079a0: 746f 2072 6563 616c 6375 6c61 7465 2074  to recalculate t
-000079b0: 6865 2077 6569 6768 7420 7065 7220 7370  he weight per sp
-000079c0: 6563 6965 7320 7061 6972 2c20 6966 2074  ecies pair, if t
-000079d0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000079e0: 2074 6865 2077 6569 6768 7420 776f 756c   the weight woul
-000079f0: 6420 6265 2072 6563 616c 6375 6c61 7465  d be recalculate
-00007a00: 640a 2d6f 722c 202d 2d6f 6e6c 7972 6f6f  d.-or, --onlyroo
-00007a10: 746f 7574 2c20 666c 6167 206f 7074 696f  tout, flag optio
-00007a20: 6e2c 2077 6865 7468 6572 2074 6f20 6f6e  n, whether to on
-00007a30: 6c79 2063 6f6e 6475 6374 2072 6174 6520  ly conduct rate 
-00007a40: 636f 7272 6563 7469 6f6e 2075 7369 6e67  correction using
-00007a50: 2074 6865 206f 7574 6772 6f75 7020 6174   the outgroup at
-00007a60: 2072 6f6f 7420 6173 206f 7574 6772 6f75   root as outgrou
-00007a70: 702c 2069 6620 7468 6520 666c 6167 2077  p, if the flag w
-00007a80: 6173 2073 6574 2c20 6f6e 6c79 2074 6865  as set, only the
-00007a90: 206f 7574 6772 6f75 7020 6174 2072 6f6f   outgroup at roo
-00007aa0: 7420 776f 756c 6420 6265 2075 7365 6420  t would be used 
-00007ab0: 6173 206f 7574 6772 6f75 700a 2d69 7465  as outgroup.-ite
-00007ac0: 722c 202d 2d65 6d5f 6974 6572 6174 696f  r, --em_iteratio
-00007ad0: 6e73 2c20 7468 6520 6d61 7869 6d75 6d20  ns, the maximum 
-00007ae0: 454d 2069 7465 7261 7469 6f6e 732c 2064  EM iterations, d
-00007af0: 6566 6175 6c74 2032 3030 0a2d 696e 6974  efault 200.-init
-00007b00: 2c20 2d2d 656d 5f69 6e69 7469 616c 697a  , --em_initializ
-00007b10: 6174 696f 6e73 2c20 7468 6520 6d61 7869  ations, the maxi
-00007b20: 6d75 6d20 454d 2069 6e69 7469 616c 697a  mum EM initializ
-00007b30: 6174 696f 6e73 2c20 6465 6661 756c 7420  ations, default 
-00007b40: 3230 300a 2d70 7263 742c 202d 2d70 726f  200.-prct, --pro
-00007b50: 6d69 6e65 6e63 655f 6375 746f 6666 2c20  minence_cutoff, 
-00007b60: 7468 6520 7072 6f6d 696e 656e 6365 2063  the prominence c
-00007b70: 7574 6f66 6620 6f66 2061 6363 6570 7461  utoff of accepta
-00007b80: 626c 6520 7065 616b 732c 2064 6566 6175  ble peaks, defau
-00007b90: 6c74 2030 2e31 0a2d 7268 2c20 2d2d 7265  lt 0.1.-rh, --re
-00007ba0: 6c5f 6865 6967 6874 2c20 7468 6520 7265  l_height, the re
-00007bb0: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
-00007bc0: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
-00007bd0: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
-00007be0: 642c 2064 6566 6175 6c74 2030 2e34 0a2d  d, default 0.4.-
-00007bf0: 736d 2c20 2d2d 7365 676d 656e 7473 2c20  sm, --segments, 
-00007c00: 7468 6520 7365 676d 656e 7473 2064 6174  the segments dat
-00007c10: 6166 696c 652c 2064 6566 6175 6c74 204e  afile, default N
-00007c20: 6f6e 650a 2d6d 6c2c 202d 2d6d 696e 6c65  one.-ml, --minle
-00007c30: 6e2c 2074 6865 206d 696e 696d 756d 206c  n, the minimum l
-00007c40: 656e 6774 6820 6f66 2061 2073 6361 6666  ength of a scaff
-00007c50: 6f6c 6420 746f 2062 6520 696e 636c 7564  old to be includ
-00007c60: 6564 2069 6e20 646f 7470 6c6f 742c 2064  ed in dotplot, d
-00007c70: 6566 6175 6c74 202d 312c 2069 6620 2d31  efault -1, if -1
-00007c80: 2077 6173 2073 6574 2c20 7468 6520 3130   was set, the 10
-00007c90: 2520 6f66 2074 6865 206c 6f6e 6765 7374  % of the longest
-00007ca0: 2073 6361 6666 6f6c 6473 2077 696c 6c20   scaffolds will 
-00007cb0: 6265 2073 6574 0a2d 6d73 2c20 2d2d 6d61  be set.-ms, --ma
-00007cc0: 7873 697a 652c 2074 6865 206d 6178 696d  xsize, the maxim
-00007cd0: 756d 2066 616d 696c 7920 7369 7a65 2074  um family size t
-00007ce0: 6f20 6265 2069 6e63 6c75 6465 642c 2064  o be included, d
-00007cf0: 6566 6175 6c74 2032 3030 0a2d 6170 2c20  efault 200.-ap, 
-00007d00: 2d2d 616e 6368 6f72 706f 696e 7473 2c20  --anchorpoints, 
-00007d10: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
-00007d20: 7320 6461 7461 6669 6c65 2c20 6465 6661  s datafile, defa
-00007d30: 756c 7420 4e6f 6e65 0a2d 6d74 2c20 2d2d  ult None.-mt, --
-00007d40: 6d75 6c74 6970 6c69 636f 6e2c 2074 6865  multiplicon, the
-00007d50: 206d 756c 7469 706c 6963 6f6e 7320 6461   multiplicons da
-00007d60: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-00007d70: 4e6f 6e65 0a2d 6774 2c20 2d2d 6765 6e65  None.-gt, --gene
-00007d80: 7461 626c 652c 2074 6865 2067 656e 6520  table, the gene 
-00007d90: 7461 626c 6520 6461 7461 6669 6c65 2c20  table datafile, 
-00007da0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6d67  default None.-mg
-00007db0: 2c20 2d2d 6d69 6e73 6567 6c65 6e2c 2074  , --minseglen, t
-00007dc0: 6865 206d 696e 696d 756d 206c 656e 6774  he minimum lengt
-00007dd0: 6820 6f66 2073 6567 6d65 6e74 7320 746f  h of segments to
-00007de0: 2069 6e63 6c75 6465 2c20 696e 2072 6174   include, in rat
-00007df0: 696f 2069 6620 3c3d 2031 2c20 6465 6661  io if <= 1, defa
-00007e00: 756c 7420 3130 3030 300a 2d6d 676e 2c20  ult 10000.-mgn, 
-00007e10: 2d2d 6d69 6e67 656e 656e 756d 2c20 7468  --mingenenum, th
-00007e20: 6520 6d69 6e69 6d75 6d20 6e75 6d62 6572  e minimum number
-00007e30: 206f 6620 6765 6e65 7320 666f 7220 6120   of genes for a 
-00007e40: 7365 676d 656e 7420 746f 2062 6520 636f  segment to be co
-00007e50: 6e73 6964 6572 6564 2c20 6465 6661 756c  nsidered, defaul
-00007e60: 7420 3330 0a2d 6b72 2c20 2d2d 6b65 6570  t 30.-kr, --keep
-00007e70: 7265 6475 6e2c 2066 6c61 6720 6f70 7469  redun, flag opti
-00007e80: 6f6e 2c20 7768 6574 6865 7220 746f 206b  on, whether to k
-00007e90: 6565 7020 7265 6475 6e64 616e 7420 6d75  eep redundant mu
-00007ea0: 6c74 6970 6c69 636f 6e73 2c20 6966 2074  ltiplicons, if t
-00007eb0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00007ec0: 2074 6865 2072 6564 756e 6461 6e74 206d   the redundant m
-00007ed0: 756c 7469 706c 6963 6f6e 7320 776f 756c  ultiplicons woul
-00007ee0: 6420 6265 206b 6570 740a 2d65 706b 2c20  d be kept.-epk, 
-00007ef0: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
-00007f00: 732c 2065 7874 7261 2070 6172 616e 6f6d  s, extra paranom
-00007f10: 6520 4b73 2064 6174 6120 746f 2070 6c6f  e Ks data to plo
-00007f20: 7420 696e 2074 6865 206d 6978 6564 204b  t in the mixed K
-00007f30: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
-00007f40: 6465 6661 756c 7420 4e6f 6e65 0a2d 7061  default None.-pa
-00007f50: 672c 202d 2d70 6c6f 7461 7067 6d6d 2c20  g, --plotapgmm, 
-00007f60: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
-00007f70: 7468 6572 2074 6f20 636f 6e64 7563 7420  ther to conduct 
-00007f80: 616e 6420 706c 6f74 206d 6978 7475 7265  and plot mixture
-00007f90: 206d 6f64 656c 696e 6720 6f66 2061 6e63   modeling of anc
-00007fa0: 686f 7220 4b73 2069 6e20 7468 6520 6d69  hor Ks in the mi
-00007fb0: 7865 6420 4b73 2064 6973 7472 6962 7574  xed Ks distribut
-00007fc0: 696f 6e2c 2069 6620 7468 6520 666c 6167  ion, if the flag
-00007fd0: 2077 6173 2073 6574 2c20 7468 6520 6d69   was set, the mi
-00007fe0: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
-00007ff0: 6620 616e 6368 6f72 204b 7320 776f 756c  f anchor Ks woul
-00008000: 6420 6265 2063 6f6e 6475 6374 6564 2061  d be conducted a
-00008010: 6e64 2070 6c6f 7474 6564 0a2d 7065 6d2c  nd plotted.-pem,
-00008020: 202d 2d70 6c6f 7465 6c6d 6d2c 2066 6c61   --plotelmm, fla
-00008030: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00008040: 7220 746f 2063 6f6e 6475 6374 2061 6e64  r to conduct and
-00008050: 2070 6c6f 7420 656c 6d6d 206d 6978 7475   plot elmm mixtu
-00008060: 7265 206d 6f64 656c 696e 6720 6f66 2070  re modeling of p
-00008070: 6172 616e 6f6d 6520 4b73 2069 6e20 7468  aranome Ks in th
-00008080: 6520 6d69 7865 6420 4b73 2064 6973 7472  e mixed Ks distr
-00008090: 6962 7574 696f 6e2c 2069 6620 7468 6520  ibution, if the 
-000080a0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-000080b0: 6520 656c 6d6d 206d 6978 7475 7265 206d  e elmm mixture m
-000080c0: 6f64 656c 696e 6720 6f66 2070 6172 616e  odeling of paran
-000080d0: 6f6d 6520 4b73 2077 6f75 6c64 2062 6520  ome Ks would be 
-000080e0: 636f 6e64 7563 7465 6420 616e 6420 706c  conducted and pl
-000080f0: 6f74 7465 640a 2d63 2c20 2d2d 636f 6d70  otted.-c, --comp
-00008100: 6f6e 656e 7473 2c20 7468 6520 7261 6e67  onents, the rang
-00008110: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
-00008120: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
-00008130: 2066 6974 2069 6e20 616e 6368 6f72 204b   fit in anchor K
-00008140: 7320 6d69 7874 7572 6520 6d6f 6465 6c69  s mixture modeli
-00008150: 6e67 2c20 6465 6661 756c 7420 2831 2c34  ng, default (1,4
-00008160: 290a 2d70 7379 2c20 2d2d 706c 6f74 7379  ).-psy, --plotsy
-00008170: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
-00008180: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-00008190: 6174 6520 7468 6520 7379 6e74 656e 7920  ate the synteny 
-000081a0: 706c 6f74 2c20 6f6e 6c79 2077 6865 6e20  plot, only when 
-000081b0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-000081c0: 2c20 7468 6520 7379 6e74 656e 7920 706c  , the synteny pl
-000081d0: 6f74 2077 6f75 6c64 2062 6520 7072 6f64  ot would be prod
-000081e0: 7563 6564 0a2d 6473 2c20 2d2d 646f 7473  uced.-ds, --dots
-000081f0: 697a 652c 2074 6865 2064 6f74 2073 697a  ize, the dot siz
-00008200: 6520 696e 2064 6f74 2070 6c6f 742c 2064  e in dot plot, d
-00008210: 6566 6175 6c74 2030 2e33 0a2d 6161 2c20  efault 0.3.-aa, 
-00008220: 2d2d 6170 616c 7068 612c 2074 6865 206f  --apalpha, the o
-00008230: 7061 6369 7479 206f 6620 616e 6368 6f72  pacity of anchor
-00008240: 2064 6f74 732c 2064 6566 6175 6c74 2031   dots, default 1
-00008250: 0a2d 6861 2c20 2d2d 686f 616c 7068 612c  .-ha, --hoalpha,
-00008260: 2074 6865 206f 7061 6369 7479 206f 6620   the opacity of 
-00008270: 686f 6d6f 6c6f 6720 646f 7473 2c20 6465  homolog dots, de
-00008280: 6661 756c 7420 300a 2d73 7274 2c20 2d2d  fault 0.-srt, --
-00008290: 7368 6f77 7265 616c 7469 636b 2c20 666c  showrealtick, fl
-000082a0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-000082b0: 6572 2074 6f20 7368 6f77 2074 6865 2072  er to show the r
-000082c0: 6561 6c20 7469 636b 2069 6e20 6765 6e65  eal tick in gene
-000082d0: 7320 6f72 2062 6173 6573 2c20 6966 2074  s or bases, if t
-000082e0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-000082f0: 2074 6865 2072 6561 6c20 7469 636b 2077   the real tick w
-00008300: 6f75 6c64 2062 6520 7368 6f77 6564 0a2d  ould be showed.-
-00008310: 746c 732c 202d 2d74 6963 6b6c 6162 656c  tls, --ticklabel
-00008320: 7369 7a65 2c20 7468 6520 6c61 6265 6c20  size, the label 
-00008330: 7369 7a65 206f 6620 7469 636b 2c20 6465  size of tick, de
-00008340: 6661 756c 7420 350a 2d78 6c2c 202d 2d78  fault 5.-xl, --x
-00008350: 6c69 6d2c 2074 6865 2078 2061 7869 7320  lim, the x axis 
-00008360: 6c69 6d69 7420 6f66 204b 7320 6469 7374  limit of Ks dist
-00008370: 7269 6275 7469 6f6e 0a2d 796c 2c20 2d2d  ribution.-yl, --
-00008380: 796c 696d 2c20 7468 6520 7920 6178 6973  ylim, the y axis
-00008390: 206c 696d 6974 206f 6620 4b73 2064 6973   limit of Ks dis
-000083a0: 7472 6962 7574 696f 6e0a 2d61 646f 2c20  tribution.-ado, 
-000083b0: 2d2d 6164 6a75 7374 6f72 7468 6f2c 2066  --adjustortho, f
-000083c0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-000083d0: 6865 7220 746f 2061 646a 7573 7420 7468  her to adjust th
-000083e0: 6520 6869 7374 6f67 7261 6d20 6865 6967  e histogram heig
-000083f0: 6874 206f 6620 6f72 7468 6f6c 6f67 6f75  ht of orthologou
-00008400: 7320 4b73 2061 7320 746f 206d 6174 6368  s Ks as to match
-00008410: 2074 6865 2068 6569 6768 7420 6f66 2070   the height of p
-00008420: 6172 616c 6f67 6f75 7320 4b73 2c20 6966  aralogous Ks, if
-00008430: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00008440: 742c 2074 6865 2061 646a 7573 746d 656e  t, the adjustmen
-00008450: 7420 776f 756c 6420 6265 2063 6f6e 6475  t would be condu
-00008460: 6374 6564 0a2d 6164 662c 202d 2d61 646a  cted.-adf, --adj
-00008470: 7573 7466 6163 746f 722c 2074 6865 2061  ustfactor, the a
-00008480: 646a 7573 746d 656e 7420 6661 6374 6f72  djustment factor
-00008490: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-000084a0: 4b73 2c20 6465 6661 756c 7420 302e 350a  Ks, default 0.5.
-000084b0: 2d6f 612c 202d 2d6f 6b61 6c70 6861 2c20  -oa, --okalpha, 
-000084c0: 7468 6520 6f70 6163 6974 7920 6f66 206f  the opacity of o
-000084d0: 7274 686f 6c6f 676f 7573 204b 7320 6469  rthologous Ks di
-000084e0: 7374 7269 6275 7469 6f6e 2069 6e20 6d69  stribution in mi
-000084f0: 7865 6420 706c 6f74 2c20 6465 6661 756c  xed plot, defaul
-00008500: 7420 302e 350a 2d63 732c 202d 2d63 6c61  t 0.5.-cs, --cla
-00008510: 7373 6963 2c20 666c 6167 206f 7074 696f  ssic, flag optio
-00008520: 6e2c 2077 6865 7468 6572 2074 6f20 6472  n, whether to dr
-00008530: 6177 206d 6978 6564 204b 7320 706c 6f74  aw mixed Ks plot
-00008540: 2069 6e20 6120 636c 6173 7369 6320 6d61   in a classic ma
-00008550: 6e6e 6572 2077 6865 7265 2074 6865 2066  nner where the f
-00008560: 756c 6c20 6f72 7468 6f6c 6f67 6f75 7320  ull orthologous 
-00008570: 4b73 2064 6973 7472 6962 7574 696f 6e20  Ks distribution 
-00008580: 6973 2064 7261 7765 642c 2069 6620 7468  is drawed, if th
-00008590: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000085a0: 7468 6520 636c 6173 7369 6320 6d69 7865  the classic mixe
-000085b0: 6420 4b73 2070 6c6f 7420 776f 756c 6420  d Ks plot would 
-000085c0: 6265 2064 7261 776e 0a2d 7461 2c20 2d2d  be drawn.-ta, --
-000085d0: 746f 7061 7272 6f77 2c20 666c 6167 206f  toparrow, flag o
-000085e0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000085f0: 6f20 6164 6a75 7374 2074 6865 2061 7272  o adjust the arr
-00008600: 6f77 2074 6f20 6265 2061 7420 7468 6520  ow to be at the 
-00008610: 746f 7020 6f66 2074 6865 2070 6c6f 742c  top of the plot,
-00008620: 2069 6e73 7465 6164 206f 6620 6265 696e   instead of bein
-00008630: 6720 636f 6f72 6469 6e61 7465 6420 6173  g coordinated as
-00008640: 2074 6865 204b 4445 206f 6620 7468 6520   the KDE of the 
-00008650: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2064  orthologous Ks d
-00008660: 6973 7472 6962 7574 696f 6e2c 2069 6620  istribution, if 
-00008670: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00008680: 2c20 7468 6520 6172 726f 7720 776f 756c  , the arrow woul
-00008690: 6420 6265 2073 6574 2061 7420 7468 6520  d be set at the 
-000086a0: 746f 700a 2d6e 612c 202d 2d6e 6f64 6561  top.-na, --nodea
-000086b0: 7665 7261 6765 642c 2066 6c61 6720 6f70  veraged, flag op
-000086c0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-000086d0: 2075 7365 206e 6f64 652d 6176 6572 6167   use node-averag
-000086e0: 6564 206d 6574 686f 6420 666f 7220 6465  ed method for de
-000086f0: 2d72 6564 756e 6461 6e63 792c 2069 6620  -redundancy, if 
-00008700: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00008710: 2c20 7468 6520 6e6f 6465 2d61 7665 7261  , the node-avera
-00008720: 6765 6420 6d65 7468 6f64 2077 6f75 6c64  ged method would
-00008730: 2062 6520 696e 6974 6961 7465 640a 2d62   be initiated.-b
-00008740: 732c 202d 2d62 6f6f 7473 7472 6170 2c20  s, --bootstrap, 
-00008750: 7468 6520 6e75 6d62 6572 206f 6620 626f  the number of bo
-00008760: 6f74 7374 7261 7020 7265 706c 6963 6174  otstrap replicat
-00008770: 6573 206f 6620 6f72 7468 6f6c 6f67 204b  es of ortholog K
-00008780: 7320 6469 7374 7269 6275 7469 6f6e 2069  s distribution i
-00008790: 6e20 6d69 7865 6420 706c 6f74 0a2d 6772  n mixed plot.-gr
-000087a0: 2c20 2d2d 6769 7374 7262 2c20 666c 6167  , --gistrb, flag
-000087b0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000087c0: 2074 6f20 7573 6520 6769 7374 5f72 6169   to use gist_rai
-000087d0: 6e62 6f77 2061 7320 636f 6c6f 7220 6d61  nbow as color ma
-000087e0: 7020 6f66 2064 6f74 706c 6f74 0a2d 6e2c  p of dotplot.-n,
-000087f0: 202d 2d6e 7468 7265 6164 732c 2074 6865   --nthreads, the
-00008800: 206e 756d 6265 7220 6f66 2074 6872 6561   number of threa
-00008810: 6473 2074 6f20 7573 6520 696e 2062 6f6f  ds to use in boo
-00008820: 7473 7472 6170 2073 616d 706c 696e 672c  tstrap sampling,
-00008830: 2064 6566 6175 6c74 2031 0a60 6060 0a0a   default 1.```..
-00008840: 2323 2055 7361 6765 0a0a 4865 7265 2077  ## Usage..Here w
-00008850: 6520 7072 6f76 6964 6564 2074 6865 2062  e provided the b
-00008860: 6173 6963 2075 7361 6765 2066 6f72 2065  asic usage for e
-00008870: 6163 6820 7072 6f67 7261 6d20 616e 6420  ach program and 
-00008880: 7468 6520 7265 6c65 7661 6e74 2070 6172  the relevant par
-00008890: 616d 6574 6572 7320 616e 6420 7375 6767  ameters and sugg
-000088a0: 6573 7469 6f6e 7320 6f6e 2070 6172 616d  estions on param
-000088b0: 6574 6572 697a 6174 696f 6e2e 2041 2072  eterization. A r
-000088c0: 656d 696e 6465 7220 7468 6174 2074 6865  eminder that the
-000088d0: 2067 6976 656e 2063 6f72 6573 2061 6e64   given cores and
-000088e0: 2074 6872 6561 6473 2063 616e 2073 6967   threads can sig
-000088f0: 6e69 6669 6361 6e74 6c79 2069 6d70 6163  nificantly impac
-00008900: 7420 7468 6520 7275 6e20 7469 6d65 2061  t the run time a
-00008910: 6e64 2074 6875 7320 7765 2061 6464 6564  nd thus we added
-00008920: 2073 6f6d 6520 7265 706f 7274 2069 6e66   some report inf
-00008930: 6f72 6d61 7469 6f6e 2070 6572 7461 696e  ormation pertain
-00008940: 696e 6720 746f 2074 6865 2073 7973 7465  ing to the syste
-00008950: 6d20 6f66 2075 7365 7273 2074 6f20 6661  m of users to fa
-00008960: 6369 6c69 7461 7465 2074 6865 2065 6666  cilitate the eff
-00008970: 6963 6965 6e74 2073 6574 7469 6e67 206f  icient setting o
-00008980: 6620 7468 7265 6164 7320 616e 6420 6d65  f threads and me
-00008990: 6d6f 7279 2e20 5468 6520 6c6f 6769 6361  mory. The logica
-000089a0: 6c20 4350 5573 2072 6570 6f72 7465 6420  l CPUs reported 
-000089b0: 7265 7072 6573 656e 7473 2074 6865 206e  represents the n
-000089c0: 756d 6265 7220 6f66 2070 6879 7369 6361  umber of physica
-000089d0: 6c20 636f 7265 7320 6d75 6c74 6970 6c69  l cores multipli
-000089e0: 6564 2062 7920 7468 6520 6e75 6d62 6572  ed by the number
-000089f0: 206f 6620 7468 7265 6164 7320 7468 6174   of threads that
-00008a00: 2063 616e 2072 756e 206f 6e20 6561 6368   can run on each
-00008a10: 2063 6f72 652c 2061 6c73 6f20 6b6e 6f77   core, also know
-00008a20: 6e20 6173 2048 7970 6572 2054 6872 6561  n as Hyper Threa
-00008a30: 6469 6e67 2e20 5468 6520 6e75 6d62 6572  ding. The number
-00008a40: 206f 6620 6c6f 6769 6361 6c20 4350 5573   of logical CPUs
-00008a50: 206d 6179 206e 6f74 206e 6563 6573 7361   may not necessa
-00008a60: 7269 6c79 2062 6520 6571 7569 7661 6c65  rily be equivale
-00008a70: 6e74 2074 6f20 7468 6520 6163 7475 616c  nt to the actual
-00008a80: 206e 756d 6265 7220 6f66 2043 5055 7320   number of CPUs 
-00008a90: 7468 6520 6375 7272 656e 7420 7072 6f63  the current proc
-00008aa0: 6573 7320 6361 6e20 7573 652e 2054 6865  ess can use. The
-00008ab0: 2061 7661 696c 6162 6c65 206d 656d 6f72   available memor
-00008ac0: 7920 7265 6665 7273 2074 6f20 7468 6520  y refers to the 
-00008ad0: 6d65 6d6f 7279 2074 6861 7420 6361 6e20  memory that can 
-00008ae0: 6265 2067 6976 656e 2069 6e73 7461 6e74  be given instant
-00008af0: 6c79 2074 6f20 7072 6f63 6573 7365 7320  ly to processes 
-00008b00: 7769 7468 6f75 7420 7468 6520 7379 7374  without the syst
-00008b10: 656d 2067 6f69 6e67 2069 6e74 6f20 7377  em going into sw
-00008b20: 6170 2061 6e64 2072 6566 6c65 6374 7320  ap and reflects 
-00008b30: 7468 6520 6163 7475 616c 206d 656d 6f72  the actual memor
-00008b40: 7920 6176 6169 6c61 626c 652e 2054 6865  y available. The
-00008b50: 2066 7265 6520 6d65 6d6f 7279 2072 6566   free memory ref
-00008b60: 6572 7320 746f 2074 6865 206d 656d 6f72  ers to the memor
-00008b70: 7920 6e6f 7420 6265 696e 6720 7573 6564  y not being used
-00008b80: 2061 7420 616c 6c20 287a 6572 6f65 6429   at all (zeroed)
-00008b90: 2074 6861 7420 6973 2072 6561 6469 6c79   that is readily
-00008ba0: 2061 7661 696c 6162 6c65 2e20 5468 6520   available. The 
-00008bb0: 6465 7363 7269 7074 696f 6e20 6162 6f76  description abov
-00008bc0: 6520 7265 6665 7273 2074 6f20 7468 6520  e refers to the 
-00008bd0: 646f 6375 6d65 6e74 6174 696f 6e20 6f66  documentation of
-00008be0: 2060 7073 7574 696c 602e 0a0a 2323 2320   `psutil`...### 
-00008bf0: 7767 6420 646d 640a 0a2a 2a54 6865 2064  wgd dmd..**The d
-00008c00: 656c 696e 6561 7469 6f6e 206f 6620 7768  elineation of wh
-00008c10: 6f6c 6520 7061 7261 6e6f 6d65 2a2a 0a60  ole paranome**.`
-00008c20: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
-00008c30: 6567 6961 5f63 6f65 7275 6c65 6120 2d49  egia_coerulea -I
-00008c40: 2032 202d 6520 3165 2d31 3020 2d62 7320   2 -e 1e-10 -bs 
-00008c50: 3130 3020 2d6e 7020 3520 282d 6e6e 2920  100 -np 5 (-nn) 
-00008c60: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
-00008c70: 6473 2920 282d 6e20 3429 2028 2d6f 2077  ds) (-n 4) (-o w
-00008c80: 6764 5f64 6d64 2920 282d 7420 776f 726b  gd_dmd) (-t work
-00008c90: 696e 675f 746d 7029 0a60 6060 200a 0a4e  ing_tmp).``` ..N
-00008ca0: 6f74 6520 7468 6174 2077 6520 646f 6e27  ote that we don'
-00008cb0: 7420 7072 6f76 6964 6520 7468 6520 6461  t provide the da
-00008cc0: 7461 206f 6620 7468 6973 2063 6f64 696e  ta of this codin
-00008cd0: 6720 7365 7175 656e 6365 2028 6364 7329  g sequence (cds)
-00008ce0: 2066 696c 6520 6041 7175 696c 6567 6961   file `Aquilegia
-00008cf0: 5f63 6f65 7275 6c65 6160 2062 7574 2069  _coerulea` but i
-00008d00: 7420 6361 6e20 6265 2065 6173 696c 7920  t can be easily 
-00008d10: 646f 776e 6c6f 6164 6564 2061 7420 5b50  downloaded at [P
-00008d20: 6879 746f 7a6f 6d65 5d28 6874 7470 733a  hytozome](https:
-00008d30: 2f2f 7068 7974 6f7a 6f6d 652d 6e65 7874  //phytozome-next
-00008d40: 2e6a 6769 2e64 6f65 2e67 6f76 2f69 6e66  .jgi.doe.gov/inf
-00008d50: 6f2f 4163 6f65 7275 6c65 615f 7633 5f31  o/Acoerulea_v3_1
-00008d60: 2920 2873 616d 6520 666f 7220 6f74 6865  ) (same for othe
-00008d70: 7220 5573 6167 6520 646f 6329 2e20 4120  r Usage doc). A 
-00008d80: 7265 6d69 6e64 6572 2074 6861 7420 696e  reminder that in
-00008d90: 2074 6865 2069 7373 7565 7320 736f 6d65   the issues some
-00008da0: 2075 7365 7273 2064 6964 6e27 7420 646f   users didn't do
-00008db0: 776e 6c6f 6164 2074 6865 2060 4163 6f65  wnload the `Acoe
-00008dc0: 7275 6c65 615f 3332 325f 7633 2e31 2e63  rulea_322_v3.1.c
-00008dd0: 6473 5f70 7269 6d61 7279 5472 616e 7363  ds_primaryTransc
-00008de0: 7269 7074 4f6e 6c79 2e66 612e 677a 6020  riptOnly.fa.gz` 
-00008df0: 6275 7420 696e 7374 6561 6420 7468 6520  but instead the 
-00008e00: 6041 636f 6572 756c 6561 5f33 3232 5f76  `Acoerulea_322_v
-00008e10: 332e 312e 6364 732e 6661 2e67 7a60 2066  3.1.cds.fa.gz` f
-00008e20: 696c 652e 2046 6f72 2074 6865 2063 6f6e  ile. For the con
-00008e30: 7374 7275 6374 696f 6e20 6f66 2077 686f  struction of who
-00008e40: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
-00008e50: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00008e60: 7269 6275 7469 6f6e 2c20 2a2a 6f6e 6c79  ribution, **only
-00008e70: 206f 6e65 2074 7261 6e73 6372 6970 7420   one transcript 
-00008e80: 2874 6865 2070 7269 6d61 7279 206f 6e65  (the primary one
-00008e90: 2920 7065 7220 6765 6e65 2073 686f 756c  ) per gene shoul
-00008ea0: 6420 6265 2069 6e63 6c75 6465 642a 2a20  d be included** 
-00008eb0: 7375 6368 2074 6861 7420 7468 6520 2a4b  such that the *K
-00008ec0: 2a3c 7375 623e 533c 2f73 7562 3e20 6973  *<sub>S</sub> is
-00008ed0: 2072 6561 6c6c 7920 696e 6469 6361 7469   really indicati
-00008ee0: 6e67 2074 6865 2061 6765 206f 6620 6765  ng the age of ge
-00008ef0: 6e65 2064 7570 6c69 6361 7469 6f6e 2065  ne duplication e
-00008f00: 7665 6e74 2c20 696e 7374 6561 6420 6f66  vent, instead of
-00008f10: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
-00008f20: 6963 696e 672e 2054 7261 6e73 6372 6970  icing. Transcrip
-00008f30: 746f 6d65 2064 6174 6120 7368 6f75 6c64  tome data should
-00008f40: 2062 6520 6361 7265 6675 6c6c 7920 7472   be carefully tr
-00008f50: 6561 7465 6420 7769 7468 2064 652d 7265  eated with de-re
-00008f60: 6475 6e64 616e 6379 2073 6f20 6173 2074  dundancy so as t
-00008f70: 6f20 7265 6475 6365 2074 6865 2066 616c  o reduce the fal
-00008f80: 7365 2070 6f73 6974 6976 6520 6475 706c  se positive dupl
-00008f90: 6963 6174 696f 6e20 6275 6d70 2063 6175  ication bump cau
-00008fa0: 7365 6420 6279 2070 6572 7661 7369 7665  sed by pervasive
-00008fb0: 2061 6c74 6572 6e61 7469 7665 2073 706c   alternative spl
-00008fc0: 6963 696e 672e 2048 6572 6520 7468 6520  icing. Here the 
-00008fd0: 696e 666c 6174 696f 6e20 6661 6374 6f72  inflation factor
-00008fe0: 2070 6172 616d 6574 6572 2c20 6769 7665   parameter, give
-00008ff0: 6e20 6279 2060 2d49 6020 6f72 2060 2d2d  n by `-I` or `--
-00009000: 696e 666c 6174 696f 6e60 2c20 6166 6665  inflation`, affe
-00009010: 6374 7320 7468 6520 6772 616e 756c 6172  cts the granular
-00009020: 6974 7920 6f72 2072 6573 6f6c 7574 696f  ity or resolutio
-00009030: 6e20 6f66 2074 6865 2063 6c75 7374 6572  n of the cluster
-00009040: 696e 6720 6f75 7463 6f6d 6520 616e 6420  ing outcome and 
-00009050: 696d 706c 6963 6974 6c79 2063 6f6e 7472  implicitly contr
-00009060: 6f6c 6c73 2074 6865 206e 756d 6265 7220  olls the number 
-00009070: 6f66 2063 6c75 7374 6572 732c 2077 6974  of clusters, wit
-00009080: 6820 6c6f 7720 7661 6c75 6573 2073 7563  h low values suc
-00009090: 6820 6173 2031 2e33 206f 7220 312e 3420  h as 1.3 or 1.4 
-000090a0: 6c65 6164 696e 6720 746f 2066 6577 6572  leading to fewer
-000090b0: 2062 7574 206c 6172 6765 7220 636c 7573   but larger clus
-000090c0: 7465 7273 2061 6e64 2068 6967 6820 7661  ters and high va
-000090d0: 6c75 6573 2073 7563 6820 6173 2035 206f  lues such as 5 o
-000090e0: 7220 3620 6c65 6164 696e 6720 746f 206d  r 6 leading to m
-000090f0: 6f72 6520 6275 7420 736d 616c 6c65 7220  ore but smaller 
-00009100: 636c 7573 7465 7273 2e20 5765 2073 6574  clusters. We set
-00009110: 2074 6865 2064 6566 6175 6c74 2076 616c   the default val
-00009120: 7565 2061 7320 3220 6173 2073 7567 6765  ue as 2 as sugge
-00009130: 7374 6564 2062 7920 5b4d 434c 5d28 6874  sted by [MCL](ht
-00009140: 7470 733a 2f2f 6d69 6361 6e73 2e6f 7267  tps://micans.org
-00009150: 2f6d 636c 2f29 2e20 5468 6520 652d 7661  /mcl/). The e-va
-00009160: 6c75 6520 6375 742d 6f66 6620 666f 7220  lue cut-off for 
-00009170: 7365 7175 656e 6365 2073 696d 696c 6172  sequence similar
-00009180: 6974 792c 2067 6976 656e 2062 7920 602d  ity, given by `-
-00009190: 6560 206f 7220 602d 2d65 7661 6c60 2c20  e` or `--eval`, 
-000091a0: 7768 6963 6820 6465 6e6f 7465 7320 7468  which denotes th
-000091b0: 6520 6578 7065 6374 6564 2076 616c 7565  e expected value
-000091c0: 206f 6620 7468 6520 6869 7420 7175 616e   of the hit quan
-000091d0: 7469 6669 6573 2074 6865 206e 756d 6265  tifies the numbe
-000091e0: 7220 6f66 2061 6c69 676e 6d65 6e74 7320  r of alignments 
-000091f0: 6f66 2073 696d 696c 6172 206f 7220 6265  of similar or be
-00009200: 7474 6572 2071 7561 6c69 7479 2074 6861  tter quality tha
-00009210: 7420 796f 7520 6578 7065 6374 2074 6f20  t you expect to 
-00009220: 6669 6e64 2073 6561 7263 6869 6e67 2074  find searching t
-00009230: 6869 7320 7175 6572 7920 6167 6169 6e73  his query agains
-00009240: 7420 6120 6461 7461 6261 7365 206f 6620  t a database of 
-00009250: 7261 6e64 6f6d 2073 6571 7565 6e63 6573  random sequences
-00009260: 2074 6865 2073 616d 6520 7369 7a65 2061   the same size a
-00009270: 7320 7468 6520 6163 7475 616c 2074 6172  s the actual tar
-00009280: 6765 7420 6461 7461 6261 7365 2c20 6973  get database, is
-00009290: 2074 6865 206b 6579 2070 6172 616d 6574   the key paramet
-000092a0: 6572 206d 6561 7375 7269 6e67 2074 6865  er measuring the
-000092b0: 2073 6967 6e69 6669 6361 6e63 6520 6f66   significance of
-000092c0: 2061 2068 6974 2c20 7768 6963 6820 6973   a hit, which is
-000092d0: 2073 6574 2068 6572 6520 6173 2064 6566   set here as def
-000092e0: 6175 6c74 2031 652d 3130 2e20 4e6f 7465  ault 1e-10. Note
-000092f0: 2074 6861 7420 5b44 4941 4d4f 4e44 5d28   that [DIAMOND](
-00009300: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00009310: 6f6d 2f62 6275 6368 6669 6e6b 2f64 6961  om/bbuchfink/dia
-00009320: 6d6f 6e64 2f77 696b 692f 312e 2d54 7574  mond/wiki/1.-Tut
-00009330: 6f72 6961 6c29 2069 7473 656c 6620 6279  orial) itself by
-00009340: 2064 6566 6175 6c74 206f 6e6c 7920 7265   default only re
-00009350: 706f 7274 7320 616c 6c20 616c 6967 6e6d  ports all alignm
-00009360: 656e 7473 2077 6974 6820 652d 7661 6c75  ents with e-valu
-00009370: 6520 3c20 302e 3030 312e 2054 6865 2070  e < 0.001. The p
-00009380: 6572 6365 6e74 6167 6520 6f66 2075 7070  ercentage of upp
-00009390: 6572 2068 6974 7320 7573 6564 2066 6f72  er hits used for
-000093a0: 2067 656e 6520 6c65 6e67 7468 206e 6f72   gene length nor
-000093b0: 6d61 6c69 7a61 7469 6f6e 2c20 6769 7665  malization, give
-000093c0: 6e20 6279 2060 2d6e 7060 206f 7220 602d  n by `-np` or `-
-000093d0: 2d6e 6f72 6d61 6c69 7a65 6470 6572 6365  -normalizedperce
-000093e0: 6e74 602c 2077 6869 6368 2064 6574 6572  nt`, which deter
-000093f0: 6d69 6e65 7320 7468 6520 7570 7065 7220  mines the upper 
-00009400: 7065 7263 656e 7469 6c65 206f 6620 6869  percentile of hi
-00009410: 7473 2070 6572 2062 696e 2028 6361 7465  ts per bin (cate
-00009420: 676f 7269 7a65 6420 6279 2067 656e 6520  gorized by gene 
-00009430: 6c65 6e67 7468 2920 7573 6564 2069 6e20  length) used in 
-00009440: 7468 6520 6669 7420 6f66 206c 696e 6561  the fit of linea
-00009450: 7220 7265 6772 6573 7369 6f6e 2c20 636f  r regression, co
-00009460: 6e73 6964 6572 696e 6720 7468 6174 206e  nsidering that n
-00009470: 6f74 2061 6c6c 2068 6974 7320 7065 7220  ot all hits per 
-00009480: 6269 6e20 7368 6f77 2061 7070 6172 656e  bin show apparen
-00009490: 7420 6c69 6e65 6172 2072 656c 6174 696f  t linear relatio
-000094a0: 6e73 6869 702c 2069 7320 7365 7420 6173  nship, is set as
-000094b0: 2064 6566 6175 6c74 2035 2c20 696e 6469   default 5, indi
-000094c0: 6361 7469 6e67 2074 6865 2075 7361 6765  cating the usage
-000094d0: 206f 6620 746f 7020 3525 2068 6974 7320   of top 5% hits 
-000094e0: 7065 7220 6269 6e2e 2054 6865 206e 756d  per bin. The num
-000094f0: 6265 7220 6f66 2062 696e 7320 6469 7669  ber of bins divi
-00009500: 6465 6420 696e 2067 656e 6520 6c65 6e67  ded in gene leng
-00009510: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
-00009520: 2c20 6769 7665 6e20 6279 2060 2d62 7360  , given by `-bs`
-00009530: 206f 7220 602d 2d62 696e 7360 2c20 6465   or `--bins`, de
-00009540: 7465 726d 696e 6573 2074 6865 206e 756d  termines the num
-00009550: 6265 7220 6f66 2062 696e 7320 746f 2063  ber of bins to c
-00009560: 6174 6567 6f72 697a 6520 7468 6520 6765  ategorize the ge
-00009570: 6e65 206c 656e 6774 682c 2069 7320 7365  ne length, is se
-00009580: 7420 6173 2064 6566 6175 6c74 2031 3030  t as default 100
-00009590: 2e20 5468 6520 7061 7261 6d65 7465 7220  . The parameter 
-000095a0: 602d 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f  `-nn` or `--nono
-000095b0: 726d 616c 697a 6174 696f 6e60 2063 616e  rmalization` can
-000095c0: 2062 6520 7365 7420 746f 2063 616c 6c20   be set to call 
-000095d0: 6f66 6620 7468 6520 6e6f 726d 616c 697a  off the normaliz
-000095e0: 6174 696f 6e20 7072 6f63 6573 732c 2061  ation process, a
-000095f0: 6c74 686f 7567 6820 6974 2773 2073 7567  lthough it's sug
-00009600: 6765 7374 6564 2074 6f20 636f 6e64 7563  gested to conduc
-00009610: 7420 7468 6520 6e6f 726d 616c 697a 6174  t the normalizat
-00009620: 696f 6e20 746f 2061 6371 7569 7265 206d  ion to acquire m
-00009630: 6f72 6520 6163 6375 7261 7465 2067 656e  ore accurate gen
-00009640: 6520 6661 6d69 6c79 2063 6c75 7374 6572  e family cluster
-00009650: 696e 6720 7265 7375 6c74 2e20 5468 6520  ing result. The 
-00009660: 7061 7261 6d65 7465 7273 2060 2d2d 746f  parameters `--to
-00009670: 5f73 746f 7060 2061 6e64 2060 2d2d 6364  _stop` and `--cd
-00009680: 7360 2063 6f6e 7472 6f6c 2074 6865 2062  s` control the b
-00009690: 6568 6176 696f 7572 206f 6620 7472 616e  ehaviour of tran
-000096a0: 736c 6174 696e 6720 636f 6469 6e67 2073  slating coding s
-000096b0: 6571 7565 6e63 6520 696e 746f 2061 6d69  equence into ami
-000096c0: 6e6f 2061 6369 6420 7365 7175 656e 6365  no acid sequence
-000096d0: 2e20 4966 2074 6865 2060 2d2d 746f 5f73  . If the `--to_s
-000096e0: 746f 7060 2077 6173 2073 6574 2c20 7468  top` was set, th
-000096f0: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
-00009700: 756c 6420 6265 2074 6572 6d69 6e61 7465  uld be terminate
-00009710: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
-00009720: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
-00009730: 6f6e 2c20 6f74 6865 7277 6973 6520 7468  on, otherwise th
-00009740: 6520 7472 616e 736c 6174 696f 6e20 776f  e translation wo
-00009750: 756c 6420 7369 6d70 6c79 2073 6b69 7020  uld simply skip 
-00009760: 616e 7920 7374 6f70 2063 6f64 6f6e 732e  any stop codons.
-00009770: 2049 6620 7468 6520 602d 2d63 6473 6020   If the `--cds` 
-00009780: 7761 7320 7365 742c 2073 6571 7565 6e63  was set, sequenc
-00009790: 6573 2074 6861 7420 646f 6573 6e27 7420  es that doesn't 
-000097a0: 7374 6172 7420 7769 7468 2061 2076 616c  start with a val
-000097b0: 6964 2073 7461 7274 2063 6f64 6f6e 2c20  id start codon, 
-000097c0: 6f72 2063 6f6e 7461 696e 7320 6d6f 7265  or contains more
-000097d0: 2074 6861 6e20 6f6e 6520 696e 2d66 7261   than one in-fra
-000097e0: 6d65 2073 746f 7020 636f 646f 6e2c 206f  me stop codon, o
-000097f0: 7220 6973 206e 6f74 2064 6976 6964 6162  r is not dividab
-00009800: 6c65 2062 7920 332c 2077 6f75 6c64 2062  le by 3, would b
-00009810: 6520 7369 6d70 6c79 2064 726f 7070 6564  e simply dropped
-00009820: 2c20 7375 6368 2074 6861 7420 6f6e 6c79  , such that only
-00009830: 2073 7472 6963 7420 636f 6d70 6c65 7465   strict complete
-00009840: 2063 6f64 696e 6720 7365 7175 656e 6365   coding sequence
-00009850: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
-00009860: 6465 6420 696e 2074 6865 2073 7562 7365  ded in the subse
-00009870: 7175 656e 7420 616e 616c 7973 6973 2e20  quent analysis. 
-00009880: 5468 6520 6578 6163 7420 6265 6861 7669  The exact behavi
-00009890: 6f75 7220 6f66 2060 2d2d 746f 5f73 746f  our of `--to_sto
-000098a0: 7060 2061 6e64 2060 2d2d 6364 7360 2069  p` and `--cds` i
-000098b0: 7320 6465 6669 6e65 6420 616e 6420 6465  s defined and de
-000098c0: 7363 7269 6265 6420 696e 2074 6865 205b  scribed in the [
-000098d0: 6269 6f70 7974 686f 6e5d 2868 7474 7073  biopython](https
-000098e0: 3a2f 2f62 696f 7079 7468 6f6e 2e6f 7267  ://biopython.org
-000098f0: 2f64 6f63 732f 312e 3735 2f61 7069 2f42  /docs/1.75/api/B
-00009900: 696f 2e53 6571 2e68 746d 6c23 4269 6f2e  io.Seq.html#Bio.
-00009910: 5365 712e 5365 712e 7472 616e 736c 6174  Seq.Seq.translat
-00009920: 6529 206c 6962 7261 7279 2e20 5468 6520  e) library. The 
-00009930: 6e75 6d62 6572 206f 6620 7061 7261 6c6c  number of parall
-00009940: 656c 2074 6872 6561 6473 2062 7920 7468  el threads by th
-00009950: 6520 6f70 7469 6f6e 2060 2d6e 6020 6f72  e option `-n` or
-00009960: 2060 2d2d 6e74 6872 6561 6473 6020 6361   `--nthreads` ca
-00009970: 6e20 6265 2073 6574 2074 6f20 6163 6365  n be set to acce
-00009980: 6c65 7261 7465 2074 6865 2063 616c 6375  lerate the calcu
-00009990: 6c61 7469 6f6e 2077 6974 6869 6e20 6064  lation within `d
-000099a0: 6961 6d6f 6e64 602e 2054 6865 2064 6972  iamond`. The dir
-000099b0: 6563 746f 7279 206f 6620 6f75 7470 7574  ectory of output
-000099c0: 206f 7220 696e 7465 726d 6564 6961 7465   or intermediate
-000099d0: 2066 696c 6573 2069 7320 6465 7465 726d   files is determ
-000099e0: 696e 6564 2062 7920 7468 6520 7061 7261  ined by the para
-000099f0: 6d65 7465 7220 602d 6f60 206f 7220 602d  meter `-o` or `-
-00009a00: 2d6f 7574 6469 7260 2c20 616e 6420 602d  -outdir`, and `-
-00009a10: 7460 206f 7220 602d 2d74 6d70 6469 7260  t` or `--tmpdir`
-00009a20: 2c20 7768 6963 6820 7769 6c6c 2062 6520  , which will be 
-00009a30: 6372 6561 7465 6420 6279 2074 6865 2070  created by the p
-00009a40: 726f 6772 616d 2069 7473 656c 6620 616e  rogram itself an
-00009a50: 6420 6265 206f 7665 7277 7269 7474 656e  d be overwritten
-00009a60: 2069 6620 7468 6520 666f 6c64 6572 2068   if the folder h
-00009a70: 6173 2061 6c72 6561 6479 2062 6565 6e20  as already been 
-00009a80: 6372 6561 7465 642e 204e 6f74 6520 7468  created. Note th
-00009a90: 6174 2074 6865 2073 6f66 7477 6172 6520  at the software 
-00009aa0: 6064 6961 6d6f 6e64 6020 7368 6f75 6c64  `diamond` should
-00009ab0: 2062 6520 7072 652d 696e 7374 616c 6c65   be pre-installe
-00009ac0: 6420 616e 6420 7365 7420 696e 2074 6865  d and set in the
-00009ad0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
-00009ae0: 6820 696e 2061 6c6c 2074 6865 2061 6e61  h in all the ana
-00009af0: 6c79 7369 7320 7065 7266 6f72 6d65 6420  lysis performed 
-00009b00: 6279 2060 7767 6420 646d 6460 2065 7863  by `wgd dmd` exc
-00009b10: 6570 7420 666f 7220 7468 6520 636f 6c6c  ept for the coll
-00009b20: 696e 6561 7220 636f 616c 6573 6365 6e63  inear coalescenc
-00009b30: 6520 616e 616c 7973 6973 2e0a 0a2a 2a57  e analysis...**W
-00009b40: 6520 7375 6767 6573 7420 7468 6174 2074  e suggest that t
-00009b50: 6865 2064 6566 6175 6c74 2073 6574 7469  he default setti
-00009b60: 6e67 2069 6e20 7768 6963 6820 7468 6520  ng in which the 
-00009b70: 696e 666c 6174 696f 6e20 6661 6374 6f72  inflation factor
-00009b80: 2069 7320 7365 7420 6173 2032 2c20 652d   is set as 2, e-
-00009b90: 7661 6c75 6520 6375 742d 6f66 6620 6173  value cut-off as
-00009ba0: 2031 652d 3130 2061 6e64 206f 7468 6572   1e-10 and other
-00009bb0: 2070 6172 616d 6574 6572 7320 696e 2064   parameters in d
-00009bc0: 6566 6175 6c74 2069 7320 6120 676f 6f64  efault is a good
-00009bd0: 2073 7461 7274 696e 6720 706f 696e 742c   starting point,
-00009be0: 2075 6e6c 6573 7320 796f 7520 7370 6563   unless you spec
-00009bf0: 6966 6963 616c 6c79 2077 616e 7420 746f  ifically want to
-00009c00: 2065 7870 6c6f 7265 2074 6865 2065 6666   explore the eff
-00009c10: 6563 7473 206f 6620 6469 6666 6572 656e  ects of differen
-00009c20: 7420 7061 7261 6d65 7465 7273 2e2a 2a20  t parameters.** 
-00009c30: 5375 6368 2074 6861 7420 7468 6520 636f  Such that the co
-00009c40: 6d6d 616e 6420 666f 7220 7468 6520 6465  mmand for the de
-00009c50: 6c69 6e65 6174 696f 6e20 6f66 2077 686f  lineation of who
-00009c60: 6c65 2070 6172 616e 6f6d 6520 6973 2073  le paranome is s
-00009c70: 696d 706c 7920 6173 2062 656c 6f77 2e0a  imply as below..
-00009c80: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
-00009c90: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
-00009ca0: 6060 600a 0a2a 2a54 6865 2064 656c 696e  ```..**The delin
-00009cb0: 6561 7469 6f6e 206f 6620 5242 4873 2a2a  eation of RBHs**
-00009cc0: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
-00009cd0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
-00009ce0: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-00009cf0: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-00009d00: 6e20 3429 2028 2d63 2030 2e39 2920 282d  n 4) (-c 0.9) (-
-00009d10: 2d6f 6766 6f72 6d61 7429 2028 2d2d 746f  -ogformat) (--to
-00009d20: 5f73 746f 7029 2028 2d2d 6364 7329 2028  _stop) (--cds) (
-00009d30: 2d6f 2077 6764 5f64 6d64 2920 282d 7420  -o wgd_dmd) (-t 
-00009d40: 776f 726b 696e 675f 746d 7029 0a60 6060  working_tmp).```
-00009d50: 0a0a 546f 2064 656c 696e 6561 7465 2052  ..To delineate R
-00009d60: 4248 7320 6265 7477 6565 6e20 7477 6f20  BHs between two 
-00009d70: 6364 7320 7365 7175 656e 6365 2066 696c  cds sequence fil
-00009d80: 6573 2c20 7468 6520 7265 6c65 7661 6e74  es, the relevant
-00009d90: 2070 6172 616d 6574 6572 2069 7320 6d6f   parameter is mo
-00009da0: 7374 6c79 2074 6865 2073 616d 6520 6173  stly the same as
-00009db0: 2077 686f 6c65 2070 6172 616e 6f6d 6520   whole paranome 
-00009dc0: 696e 6665 7265 6e63 652c 2065 7863 6570  inference, excep
-00009dd0: 7420 666f 7220 7468 6520 7061 7261 6d65  t for the parame
-00009de0: 7465 7220 602d 6360 206f 7220 602d 2d63  ter `-c` or `--c
-00009df0: 7363 6f72 6560 2c20 7768 6963 6820 7261  score`, which ra
-00009e00: 6e67 6573 2062 6574 7765 656e 2030 2061  nges between 0 a
-00009e10: 6e64 2031 2061 6e64 2069 7320 7573 6564  nd 1 and is used
-00009e20: 2074 6f20 7265 6c61 7820 7468 6520 7369   to relax the si
-00009e30: 6d69 6c61 7269 7479 2063 7574 6f66 6620  milarity cutoff 
-00009e40: 6672 6f6d 2074 6865 2065 7863 6c75 7369  from the exclusi
-00009e50: 7665 2072 6563 6970 726f 6361 6c20 6265  ve reciprocal be
-00009e60: 7374 2068 6974 7320 746f 2061 2063 6572  st hits to a cer
-00009e70: 7461 696e 2072 6174 696f 2061 7320 746f  tain ratio as to
-00009e80: 2074 6865 2062 6573 7420 6869 7473 2e20   the best hits. 
-00009e90: 466f 7220 696e 7374 616e 6365 2c20 6966  For instance, if
-00009ea0: 2074 6865 2067 656e 6520 6231 2066 726f   the gene b1 fro
-00009eb0: 6d20 6765 6e6f 6d65 2042 2068 6173 2074  m genome B has t
-00009ec0: 6865 2062 6573 7420 6869 7420 6765 6e65  he best hit gene
-00009ed0: 2061 3120 6672 6f6d 2067 656e 6f6d 6520   a1 from genome 
-00009ee0: 4120 7769 7468 2074 6865 2062 6974 2073  A with the bit s
-00009ef0: 636f 7265 2061 7320 3130 302c 2077 6869  core as 100, whi
-00009f00: 6368 2069 7320 6120 7363 6f72 696e 6720  ch is a scoring 
-00009f10: 6d61 7472 6978 2069 6e64 6570 656e 6465  matrix independe
-00009f20: 6e74 206d 6561 7375 7265 206f 6620 7468  nt measure of th
-00009f30: 6520 286c 6f63 616c 2920 7369 6d69 6c61  e (local) simila
-00009f40: 7269 7479 206f 6620 7468 6520 7477 6f20  rity of the two 
-00009f50: 616c 6967 6e65 6420 7365 7175 656e 6365  aligned sequence
-00009f60: 732c 2077 6974 6820 6c61 7267 6572 2076  s, with larger v
-00009f70: 616c 7565 7320 696e 6469 6361 7469 6e67  alues indicating
-00009f80: 2068 6967 6865 7220 7369 6d69 6c61 7269   higher similari
-00009f90: 7469 6573 2c20 6769 7665 6e20 7468 6520  ties, given the 
-00009fa0: 602d 6320 302e 3960 2c20 6765 6e65 7320  `-c 0.9`, genes 
-00009fb0: 6672 6f6d 2067 656e 6f6d 6520 4120 7768  from genome A wh
-00009fc0: 6963 6820 6861 7320 7468 6520 6269 7420  ich has the bit 
-00009fd0: 7363 6f72 6520 7769 7468 2067 656e 6520  score with gene 
-00009fe0: 6231 2068 6967 6865 7220 7468 616e 2030  b1 higher than 0
-00009ff0: 2e39 7831 3030 2077 696c 6c20 616c 736f  .9x100 will also
-0000a000: 2062 6520 7772 6974 7465 6e20 696e 2074   be written in t
-0000a010: 6865 2072 6573 756c 7420 6669 6c65 2c20  he result file, 
-0000a020: 7768 6963 6820 696e 2061 2073 656e 7365  which in a sense
-0000a030: 2061 7265 206e 6f74 2052 4248 7320 616e   are not RBHs an
-0000a040: 796d 6f72 6520 6f66 2063 6f75 7273 652c  ymore of course,
-0000a050: 2062 7574 2074 6865 2068 6967 686c 7920   but the highly 
-0000a060: 7369 6d69 6c61 7220 686f 6d6f 6c6f 6775  similar homologu
-0000a070: 6520 7061 6972 732e 2049 6620 6d6f 7265  e pairs. If more
-0000a080: 2074 6861 6e20 3220 7365 7175 656e 6365   than 2 sequence
-0000a090: 2066 696c 6573 2077 6572 6520 7072 6f76   files were prov
-0000a0a0: 6964 6564 2c20 6576 6572 7920 7061 6972  ided, every pair
-0000a0b0: 2d77 6973 6520 5242 4873 2077 6f75 6c64  -wise RBHs would
-0000a0c0: 2062 6520 6361 6c63 756c 6174 6564 2065   be calculated e
-0000a0d0: 7863 6570 7420 666f 7220 7175 6572 7969  xcept for queryi
-0000a0e0: 6e67 2074 6865 2073 616d 6520 7365 7175  ng the same sequ
-0000a0f0: 656e 6365 2069 7473 656c 662e 2054 6865  ence itself. The
-0000a100: 206e 756d 6265 7220 6f66 2070 6172 616c   number of paral
-0000a110: 6c65 6c20 7468 7265 6164 7320 746f 2062  lel threads to b
-0000a120: 6f6f 7374 6572 2074 6865 2072 756e 6e69  ooster the runni
-0000a130: 6e67 2073 7065 6564 2063 616e 2062 6520  ng speed can be 
-0000a140: 7365 7420 6279 2074 6865 206f 7074 696f  set by the optio
-0000a150: 6e20 602d 6e60 206f 7220 602d 2d6e 7468  n `-n` or `--nth
-0000a160: 7265 6164 7360 2077 6869 6368 2069 7320  reads` which is 
-0000a170: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
-0000a180: 7365 7420 6173 2028 4e2d 3129 4e2f 3220  set as (N-1)N/2 
-0000a190: 7768 6572 6520 4e20 6973 2074 6865 206e  where N is the n
-0000a1a0: 756d 6265 7220 6f66 2063 6473 2066 696c  umber of cds fil
-0000a1b0: 6573 2074 6f20 6163 6869 6576 6520 7468  es to achieve th
-0000a1c0: 6520 6869 6768 6573 7420 6566 6669 6369  e highest effici
-0000a1d0: 656e 6379 2e20 5468 6520 6f70 7469 6f6e  ency. The option
-0000a1e0: 2060 2d2d 6f67 666f 726d 6174 6020 6361   `--ogformat` ca
-0000a1f0: 6e20 6265 2073 6574 2074 6f20 6164 6420  n be set to add 
-0000a200: 696e 6465 7820 2866 6f72 2069 6e73 7461  index (for insta
-0000a210: 6e63 6520 4746 3030 3030 3030 3031 2920  nce GF00000001) 
-0000a220: 746f 2074 6865 206f 7574 7075 7420 5242  to the output RB
-0000a230: 4820 6765 6e65 2066 616d 696c 6965 7320  H gene families 
-0000a240: 7768 6963 6820 6361 6e20 6265 2066 7572  which can be fur
-0000a250: 7468 6572 2075 7365 6420 696e 2074 6865  ther used in the
-0000a260: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000a270: 2063 616c 6375 6c61 7469 6f6e 2062 7920   calculation by 
-0000a280: 6077 6764 206b 7364 602e 0a0a 2a2a 5468  `wgd ksd`...**Th
-0000a290: 6520 7375 6767 6573 7465 6420 636f 6d6d  e suggested comm
-0000a2a0: 616e 6420 746f 2073 7461 7274 2077 6974  and to start wit
-0000a2b0: 6820 6973 2061 6c73 6f20 756e 6465 7220  h is also under 
-0000a2c0: 7468 6520 6465 6661 756c 7420 7365 7474  the default sett
-0000a2d0: 696e 6720 7769 7468 2074 6865 2063 6f6d  ing with the com
-0000a2e0: 6d61 6e64 2073 686f 776e 2062 656c 6f77  mand shown below
-0000a2f0: 2a2a 0a0a 6060 600a 7767 6420 646d 6420  **..```.wgd dmd 
-0000a300: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
-0000a310: 6365 320a 6060 600a 0a2a 2a54 6865 2064  ce2.```..**The d
-0000a320: 656c 696e 6561 7469 6f6e 206f 6620 6c6f  elineation of lo
-0000a330: 6361 6c20 4d52 4248 732a 2a0a 6060 600a  cal MRBHs**.```.
-0000a340: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
-0000a350: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
-0000a360: 656e 6365 3320 2d66 2073 6571 7565 6e63  ence3 -f sequenc
-0000a370: 6531 202d 6520 3165 2d31 3020 2d62 7320  e1 -e 1e-10 -bs 
-0000a380: 3130 3020 2d6e 7020 3520 282d 6e6e 2920  100 -np 5 (-nn) 
-0000a390: 282d 6e20 3429 2028 2d6b 6629 2028 2d6b  (-n 4) (-kf) (-k
-0000a3a0: 6429 2028 2d63 2030 2e39 2920 282d 2d74  d) (-c 0.9) (--t
-0000a3b0: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
-0000a3c0: 282d 6f20 7767 645f 646d 6429 2028 2d74  (-o wgd_dmd) (-t
-0000a3d0: 2077 6f72 6b69 6e67 5f74 6d70 290a 6060   working_tmp).``
-0000a3e0: 600a 0a2a 2a54 6865 2064 6973 7469 6e63  `..**The distinc
-0000a3f0: 7469 6f6e 2062 6574 7765 656e 206c 6f63  tion between loc
-0000a400: 616c 2061 6e64 2067 6c6f 6261 6c20 2868  al and global (h
-0000a410: 6572 6575 6e64 6572 2920 4d52 4248 7320  ereunder) MRBHs 
-0000a420: 6973 2074 6861 7420 6c6f 6361 6c20 4d52  is that local MR
-0000a430: 4248 7320 6172 6520 7468 6520 7265 7375  BHs are the resu
-0000a440: 6c74 7320 6f66 206d 6572 6765 6420 5242  lts of merged RB
-0000a450: 4873 206f 6e20 6120 6a6f 696e 7420 666f  Hs on a joint fo
-0000a460: 6361 6c20 7370 6563 6965 732c 2066 6f72  cal species, for
-0000a470: 2069 6e73 7461 6e63 6520 696e 2061 2074   instance in a t
-0000a480: 6872 6565 2073 7065 6369 6573 2073 7973  hree species sys
-0000a490: 7465 6d20 2841 2c28 422c 4329 292c 2074  tem (A,(B,C)), t
-0000a4a0: 6865 206c 6f63 616c 204d 5242 4873 206f  he local MRBHs o
-0000a4b0: 6620 4120 6f6e 6c79 2072 6571 7569 7265  f A only require
-0000a4c0: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
-0000a4d0: 206f 6620 5242 4873 2062 6574 7765 656e   of RBHs between
-0000a4e0: 2041 2061 6e64 2043 2028 6465 6e6f 7465   A and C (denote
-0000a4f0: 6420 6173 2041 4329 2061 6e64 2041 4220  d as AC) and AB 
-0000a500: 616e 6420 7468 656e 2074 6865 206d 6572  and then the mer
-0000a510: 6769 6e67 206f 6620 4142 2061 6e64 2041  ging of AB and A
-0000a520: 4320 6174 2074 6865 2061 7869 7320 6f66  C at the axis of
-0000a530: 2041 2c20 7768 696c 6520 7468 6520 676c   A, while the gl
-0000a540: 6f61 626c 204d 5242 4873 2061 7265 2069  oabl MRBHs are i
-0000a550: 6e64 6570 656e 6465 6e74 206f 6620 666f  ndependent of fo
-0000a560: 6361 6c20 7370 6563 6965 7320 696e 2074  cal species in t
-0000a570: 6861 7420 6974 206a 7573 7420 6361 6c6c  hat it just call
-0000a580: 7320 7468 6520 6361 6c63 756c 6174 696f  s the calculatio
-0000a590: 6e73 206f 6620 616c 6c20 706f 7373 6962  ns of all possib
-0000a5a0: 6c65 2073 7065 6369 6573 2070 6169 7220  le species pair 
-0000a5b0: 286e 6f74 2073 656c 6620 746f 2073 656c  (not self to sel
-0000a5c0: 6620 7370 6563 6965 7320 7061 6972 292c  f species pair),
-0000a5d0: 2073 7563 6820 7468 6174 2041 422c 2041   such that AB, A
-0000a5e0: 432c 2061 6e64 2042 4320 6172 6520 616c  C, and BC are al
-0000a5f0: 6c20 746f 2062 6520 6361 6c63 756c 6174  l to be calculat
-0000a600: 6564 2061 6e64 206d 6572 6765 642e 2a2a  ed and merged.**
-0000a610: 0a0a 5477 6f20 7479 7065 7320 6f66 204d  ..Two types of M
-0000a620: 5242 4873 2061 7320 696e 7465 7072 6574  RBHs as intepret
-0000a630: 6174 6564 2061 626f 7665 2063 616e 2062  ated above can b
-0000a640: 6520 6465 6c69 6e65 6174 6564 2062 7920  e delineated by 
-0000a650: 6077 6764 2064 6d64 602c 2074 6865 206c  `wgd dmd`, the l
-0000a660: 6f63 616c 204d 5242 4873 2061 6e64 2074  ocal MRBHs and t
-0000a670: 6865 2067 6c6f 6261 6c20 4d52 4248 732e  he global MRBHs.
-0000a680: 2054 6865 206c 6f63 616c 204d 5242 4873   The local MRBHs
-0000a690: 2061 7265 2063 6f6e 7374 7275 6374 6564   are constructed
-0000a6a0: 2062 7920 6d65 7267 696e 6720 616c 6c20   by merging all 
-0000a6b0: 7468 6520 7265 6c65 7661 6e74 2052 4248  the relevant RBH
-0000a6c0: 7320 6f6e 6c79 2077 6974 6820 7468 6520  s only with the 
-0000a6d0: 666f 6361 6c20 7370 6563 6965 732c 2077  focal species, w
-0000a6e0: 6869 6368 2069 7320 7365 7420 6279 2060  hich is set by `
-0000a6f0: 2d66 6020 6f72 2060 2d2d 666f 6375 7360  -f` or `--focus`
-0000a700: 2e20 5468 6520 7061 7261 6d65 7465 7220  . The parameter 
-0000a710: 602d 6b66 6020 6f72 2060 2d2d 6b65 6570  `-kf` or `--keep
-0000a720: 6661 7374 6160 2063 616e 2062 6520 7365  fasta` can be se
-0000a730: 7420 746f 2072 6574 6169 6e20 7468 6520  t to retain the 
-0000a740: 7365 7175 656e 6365 2069 6e66 6f72 6d61  sequence informa
-0000a750: 7469 6f6e 206f 6620 6561 6368 204d 5242  tion of each MRB
-0000a760: 482e 2054 6865 2070 6172 616d 6574 6572  H. The parameter
-0000a770: 2060 2d6b 6460 206f 7220 602d 2d6b 6565   `-kd` or `--kee
-0000a780: 7064 7570 6c69 6361 7465 7360 2064 6574  pduplicates` det
-0000a790: 6572 6d69 6e65 7320 7768 6574 6865 7220  ermines whether 
-0000a7a0: 7468 6520 7361 6d65 2067 656e 6573 2063  the same genes c
-0000a7b0: 616e 2061 7070 6561 7220 696e 2064 6966  an appear in dif
-0000a7c0: 6665 7265 6e74 206c 6f63 616c 204d 5242  ferent local MRB
-0000a7d0: 4873 2e20 4e6f 726d 616c 6c79 2074 6865  Hs. Normally the
-0000a7e0: 7265 2077 696c 6c20 6265 206e 6f20 6475  re will be no du
-0000a7f0: 706c 6963 6174 6573 2069 6e20 7468 6520  plicates in the 
-0000a800: 6c6f 6361 6c20 4d52 4248 7320 6275 7420  local MRBHs but 
-0000a810: 6966 2075 7365 7273 2073 6574 2074 6865  if users set the
-0000a820: 2060 2d63 6020 6173 2030 2e39 2028 6f72   `-c` as 0.9 (or
-0000a830: 2061 6e79 2076 616c 7565 2073 6d61 6c6c   any value small
-0000a840: 6572 2074 6861 6e20 3129 2c20 6974 2773  er than 1), it's
-0000a850: 206c 696b 656c 7920 7468 6174 2074 6865   likely that the
-0000a860: 2073 616d 6520 6765 6e65 2077 696c 6c20   same gene will 
-0000a870: 6861 7665 2063 6861 6e63 6520 746f 2061  have chance to a
-0000a880: 7070 6561 7220 6d75 6c74 6970 6c65 2074  ppear multiple t
-0000a890: 696d 6573 2069 6e20 6469 6666 6572 656e  imes in differen
-0000a8a0: 7420 6c6f 6361 6c20 4d52 4248 732e 2054  t local MRBHs. T
-0000a8b0: 6861 7420 6973 2074 6f20 7361 792c 2074  hat is to say, t
-0000a8c0: 6865 2070 6172 616d 6574 6572 2060 2d6b  he parameter `-k
-0000a8d0: 6460 2069 7320 6d65 616e 696e 6766 756c  d` is meaningful
-0000a8e0: 206f 6e6c 7920 7768 656e 2069 7427 7320   only when it's 
-0000a8f0: 7365 7420 746f 6765 7468 6572 2077 6974  set together wit
-0000a900: 6820 7468 6520 7061 7261 6d65 7465 7220  h the parameter 
-0000a910: 602d 6360 2e20 5468 6520 6e75 6d62 6572  `-c`. The number
-0000a920: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
-0000a930: 6561 6473 2069 7320 7375 6767 6573 7465  eads is suggeste
-0000a940: 6420 746f 2062 6520 7365 7420 6173 2074  d to be set as t
-0000a950: 6865 206e 756d 6265 7220 6f66 2063 6473  he number of cds
-0000a960: 2066 696c 6573 206d 696e 7573 2031 2e0a   files minus 1..
-0000a970: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
-0000a980: 7461 7274 696e 6720 7275 6e20 6973 2075  tarting run is u
-0000a990: 6e64 6572 2074 6865 2064 6566 6175 6c74  nder the default
-0000a9a0: 2070 6172 616d 6574 6572 2077 6974 6820   parameter with 
-0000a9b0: 7468 6520 636f 6d6d 616e 6420 7368 6f77  the command show
-0000a9c0: 6e20 6173 2062 656c 6f77 2e2a 2a0a 0a60  n as below.**..`
-0000a9d0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-0000a9e0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000a9f0: 6571 7565 6e63 6533 202d 6620 7365 7175  equence3 -f sequ
-0000aa00: 656e 6365 310a 6060 600a 0a2a 2a54 6865  ence1.```..**The
-0000aa10: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
-0000aa20: 676c 6f62 616c 204d 5242 4873 2a2a 0a60  global MRBHs**.`
-0000aa30: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-0000aa40: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000aa50: 6571 7565 6e63 6533 202d 676d 202d 6520  equence3 -gm -e 
-0000aa60: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
-0000aa70: 7020 3520 282d 6e6e 2920 282d 6e20 3429  p 5 (-nn) (-n 4)
-0000aa80: 2028 2d6b 6629 2028 2d6b 6429 2028 2d63   (-kf) (-kd) (-c
-0000aa90: 2030 2e39 2920 282d 2d74 6f5f 7374 6f70   0.9) (--to_stop
-0000aaa0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
-0000aab0: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
-0000aac0: 6e67 5f74 6d70 290a 6060 600a 0a54 6865  ng_tmp).```..The
-0000aad0: 2067 6c6f 6261 6c20 4d52 4248 7320 6973   global MRBHs is
-0000aae0: 2063 6f6e 7374 7275 6374 6564 2062 7920   constructed by 
-0000aaf0: 6578 6861 7573 7469 7665 6c79 206d 6572  exhaustively mer
-0000ab00: 6769 6e67 2061 6c6c 2074 6865 2070 6f73  ging all the pos
-0000ab10: 7369 626c 6520 7061 6972 2d77 6973 6520  sible pair-wise 
-0000ab20: 5242 4873 2065 7863 6570 7420 666f 7220  RBHs except for 
-0000ab30: 7175 6572 7969 6e67 2074 6865 2073 6571  querying the seq
-0000ab40: 7565 6e63 6520 6974 7365 6c66 2c20 7768  uence itself, wh
-0000ab50: 6963 6820 6361 6e20 6265 2069 6e69 7469  ich can be initi
-0000ab60: 6174 6564 2062 7920 6164 6420 7468 6520  ated by add the 
-0000ab70: 666c 6167 2060 2d67 6d60 206f 7220 602d  flag `-gm` or `-
-0000ab80: 2d67 6c6f 6261 6c6d 7262 6860 2e20 5468  -globalmrbh`. Th
-0000ab90: 6520 7265 7374 206f 6620 7265 6c65 7661  e rest of releva
-0000aba0: 6e74 2070 6172 616d 6574 6572 7320 7374  nt parameters st
-0000abb0: 6179 7320 7468 6520 7361 6d65 2061 7320  ays the same as 
-0000abc0: 7468 6520 6c6f 6361 6c20 4d52 4248 732e  the local MRBHs.
-0000abd0: 2054 6865 206e 756d 6265 7220 6f66 2070   The number of p
-0000abe0: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-0000abf0: 6973 2073 7567 6765 7374 6564 2074 6f20  is suggested to 
-0000ac00: 6265 2073 6574 2061 7320 284e 2d31 294e  be set as (N-1)N
-0000ac10: 2f32 2074 6f6f 2077 6865 7265 204e 2069  /2 too where N i
-0000ac20: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
-0000ac30: 6364 7320 6669 6c65 7320 746f 2061 6368  cds files to ach
-0000ac40: 6965 7665 2074 6865 2068 6967 6865 7374  ieve the highest
-0000ac50: 2065 6666 6963 6965 6e63 792e 0a0a 2a2a   efficiency...**
-0000ac60: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000ac70: 7469 6e67 2072 756e 2069 7320 756e 6465  ting run is unde
-0000ac80: 7220 7468 6520 6465 6661 756c 7420 7061  r the default pa
-0000ac90: 7261 6d65 7465 7220 7769 7468 2074 6865  rameter with the
-0000aca0: 2063 6f6d 6d61 6e64 2073 686f 776e 2061   command shown a
-0000acb0: 7320 6265 6c6f 772e 2a2a 0a0a 6060 600a  s below.**..```.
-0000acc0: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
-0000acd0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
-0000ace0: 656e 6365 3320 2d67 6d0a 6060 600a 0a2a  ence3 -gm.```..*
-0000acf0: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
-0000ad00: 206f 6620 6f72 7468 6f67 726f 7570 732a   of orthogroups*
-0000ad10: 2a0a 6060 600a 7767 6420 646d 6420 7365  *.```.wgd dmd se
-0000ad20: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000ad30: 3220 7365 7175 656e 6365 3320 2d6f 6920  2 sequence3 -oi 
-0000ad40: 2d6f 6f20 2d65 2031 652d 3130 202d 6273  -oo -e 1e-10 -bs
-0000ad50: 2031 3030 202d 6e70 2035 2028 2d6e 6e29   100 -np 5 (-nn)
-0000ad60: 2028 2d63 6329 2028 2d74 6529 2028 2d6d   (-cc) (-te) (-m
-0000ad70: 6320 302e 3829 2028 2d67 6e29 2028 2d74  c 0.8) (-gn) (-t
-0000ad80: 7265 6520 2766 6173 7474 7265 6527 2920  ree 'fasttree') 
-0000ad90: 282d 7473 2027 2d66 6173 7465 7374 2729  (-ts '-fastest')
-0000ada0: 2028 2d6e 2034 2920 282d 2d74 6f5f 7374   (-n 4) (--to_st
-0000adb0: 6f70 2920 282d 2d63 6473 2920 282d 6f20  op) (--cds) (-o 
-0000adc0: 7767 645f 646d 6429 2028 2d74 2077 6f72  wgd_dmd) (-t wor
-0000add0: 6b69 6e67 5f74 6d70 290a 6060 600a 0a49  king_tmp).```..I
-0000ade0: 6e20 6077 6764 2076 3260 2c20 7765 2061  n `wgd v2`, we a
-0000adf0: 6c73 6f20 696d 706c 656d 656e 7465 6420  lso implemented 
-0000ae00: 616e 2061 6c67 6f72 6974 686d 206f 6620  an algorithm of 
-0000ae10: 6465 6c69 6e65 6174 696e 6720 6f72 7468  delineating orth
-0000ae20: 6f67 726f 7570 732c 2077 6869 6368 2063  ogroups, which c
-0000ae30: 616e 2062 6520 696e 6974 6961 7465 6420  an be initiated 
-0000ae40: 7769 7468 2074 6865 2070 6172 616d 6574  with the paramet
-0000ae50: 6572 2060 2d6f 6960 206f 7220 602d 2d6f  er `-oi` or `--o
-0000ae60: 7274 686f 696e 6665 7260 2e20 5477 6f20  rthoinfer`. Two 
-0000ae70: 7761 7973 206f 6620 6465 6c69 6e65 6174  ways of delineat
-0000ae80: 696f 6e20 6361 6e20 6265 2063 686f 7365  ion can be chose
-0000ae90: 6e2c 2074 6865 2063 6f6e 6361 7465 6e61  n, the concatena
-0000aea0: 7469 6f6e 2077 6179 2028 7365 7420 6279  tion way (set by
-0000aeb0: 2074 6865 2070 6172 616d 6574 6572 2060   the parameter `
-0000aec0: 2d63 6360 206f 7220 602d 2d63 6f6e 6361  -cc` or `--conca
-0000aed0: 7460 2920 6f72 2074 6865 206e 6f6e 2d63  t`) or the non-c
-0000aee0: 6f6e 6361 7465 6e61 7469 6f6e 2028 6465  oncatenation (de
-0000aef0: 6661 756c 7429 2077 6179 2e20 496e 2062  fault) way. In b
-0000af00: 7269 6566 2c20 7468 6520 636f 6e63 6174  rief, the concat
-0000af10: 656e 6174 696f 6e20 7761 7920 6f66 2064  enation way of d
-0000af20: 656c 696e 6561 7469 6e67 206f 7274 686f  elineating ortho
-0000af30: 6772 6f75 7073 2073 7461 7274 7320 7769  groups starts wi
-0000af40: 7468 2063 6f6e 6361 7465 6e61 7469 6e67  th concatenating
-0000af50: 2061 6c6c 2074 6865 2073 6571 7565 6e63   all the sequenc
-0000af60: 6573 2069 6e74 6f20 6120 7369 6e67 6c65  es into a single
-0000af70: 2073 6571 7565 6e63 6520 6669 6c65 2061   sequence file a
-0000af80: 6e64 2074 6865 6e20 696e 6665 7272 696e  nd then inferrin
-0000af90: 6720 7468 6520 7768 6f6c 6520 7061 7261  g the whole para
-0000afa0: 6e6f 6d65 206f 6620 7468 6973 2073 696e  nome of this sin
-0000afb0: 676c 6520 7365 7175 656e 6365 2066 696c  gle sequence fil
-0000afc0: 6520 7769 7468 2074 6865 2063 6c75 7374  e with the clust
-0000afd0: 6572 696e 6720 7265 7375 6c74 7320 6d61  ering results ma
-0000afe0: 7070 6564 2062 6163 6b20 746f 2074 6865  pped back to the
-0000aff0: 2062 656c 6f6e 6769 6e67 2073 7065 6369   belonging speci
-0000b000: 6573 2e20 5768 696c 6520 7468 6520 6e6f  es. While the no
-0000b010: 6e2d 636f 6e63 6174 656e 6174 696f 6e20  n-concatenation 
-0000b020: 7761 7920 7374 6172 7473 2077 6974 6820  way starts with 
-0000b030: 7265 7370 6563 7469 7665 2070 6169 722d  respective pair-
-0000b040: 7769 7365 2064 6961 6d6f 6e64 2073 6561  wise diamond sea
-0000b050: 7263 6820 2869 6e63 6c75 6469 6e67 2071  rch (including q
-0000b060: 7565 7279 696e 6720 7468 6520 7361 6d65  uerying the same
-0000b070: 2073 6571 7565 6e63 6520 6974 7365 6c66   sequence itself
-0000b080: 2920 616e 6420 7468 656e 2061 6c6c 2074  ) and then all t
-0000b090: 6865 2073 6571 7565 6e63 6520 7369 6d69  he sequence simi
-0000b0a0: 6c61 7269 7479 2074 6162 6c65 7320 7769  larity tables wi
-0000b0b0: 6c6c 2062 6520 636f 6e63 6174 656e 6174  ll be concatenat
-0000b0c0: 6564 2061 6e64 2063 6c75 7374 6572 6564  ed and clustered
-0000b0d0: 2069 6e74 6f20 6f72 7468 6f67 726f 7570   into orthogroup
-0000b0e0: 732e 2053 6f6d 6520 6f74 6865 7220 706f  s. Some other po
-0000b0f0: 7373 6962 6c79 2075 7365 6675 6c20 706f  ssibly useful po
-0000b100: 7374 2d63 6c75 7374 6572 696e 6720 6675  st-clustering fu
-0000b110: 6e63 7469 6f6e 7320 6361 6e20 6265 2069  nctions can be i
-0000b120: 6e69 7469 6174 6564 2c20 696e 636c 7564  nitiated, includ
-0000b130: 696e 6720 7468 6520 7061 7261 6d65 7465  ing the paramete
-0000b140: 7220 602d 7465 6020 6f72 2060 2d2d 7465  r `-te` or `--te
-0000b150: 7374 736f 6760 2c20 7768 6963 6820 6361  stsog`, which ca
-0000b160: 6e20 6265 2073 6574 2074 6f20 7374 6172  n be set to star
-0000b170: 7420 7468 6520 756e 6269 6173 6564 2074  t the unbiased t
-0000b180: 6573 7420 6f66 2073 696e 676c 652d 636f  est of single-co
-0000b190: 7079 2067 656e 6520 6661 6d69 6c69 6573  py gene families
-0000b1a0: 2028 6e6f 7465 2074 6861 7420 7468 6973   (note that this
-0000b1b0: 2066 756e 6374 696f 6e20 6e65 6564 7320   function needs 
-0000b1c0: 6068 6d6d 6572 6020 2876 332e 3162 3229  `hmmer` (v3.1b2)
-0000b1d0: 2074 6f20 6265 2069 6e73 7461 6c6c 6564   to be installed
-0000b1e0: 2069 6e20 7468 6520 656e 7669 726f 6e6d   in the environm
-0000b1f0: 656e 7420 7061 7468 292c 2074 6865 2070  ent path), the p
-0000b200: 6172 616d 6574 6572 2060 2d6d 6360 206f  arameter `-mc` o
-0000b210: 7220 602d 2d6d 736f 6763 7574 602c 2072  r `--msogcut`, r
-0000b220: 616e 6769 6e67 2062 6574 7765 656e 2030  anging between 0
-0000b230: 2074 6f20 312c 2077 6869 6368 2063 616e   to 1, which can
-0000b240: 2062 6520 7365 7420 746f 2073 6561 7263   be set to searc
-0000b250: 6820 7468 6520 736f 2d63 616c 6c65 6420  h the so-called 
-0000b260: 6d6f 7374 6c79 2073 696e 676c 652d 636f  mostly single-co
-0000b270: 7079 2066 616d 696c 7920 7768 6963 6820  py family which 
-0000b280: 6861 7320 6869 6768 6572 2074 6861 6e20  has higher than 
-0000b290: 6365 7274 6169 6e20 6375 742d 6f66 6620  certain cut-off 
-0000b2a0: 7065 7263 656e 7461 6765 206f 6620 7370  percentage of sp
-0000b2b0: 6563 6965 7320 636f 7665 7261 6765 2c20  ecies coverage, 
-0000b2c0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
-0000b2d0: 676e 6020 6f72 2060 2d2d 6765 746e 736f  gn` or `--getnso
-0000b2e0: 6760 2c20 7768 6963 6820 6361 6e20 6265  g`, which can be
-0000b2f0: 2073 6574 2074 6f20 7365 6172 6368 2066   set to search f
-0000b300: 6f72 206e 6573 7465 6420 7369 6e67 6c65  or nested single
-0000b310: 2d63 6f70 7920 6765 6e65 2066 616d 696c  -copy gene famil
-0000b320: 6965 7320 284e 534f 4773 2920 7768 6963  ies (NSOGs) whic
-0000b330: 6820 6973 206f 7269 6769 6e61 6c6c 7920  h is originally 
-0000b340: 6d75 6c74 6979 2d63 6f70 7920 6275 7420  multiy-copy but 
-0000b350: 6861 7320 6120 286d 6f73 746c 7929 2073  has a (mostly) s
-0000b360: 696e 676c 652d 636f 7079 2062 7261 6e63  ingle-copy branc
-0000b370: 6820 2877 6869 6368 2072 6571 7569 7265  h (which require
-0000b380: 7320 7468 6520 6368 6f73 656e 2074 7265  s the chosen tre
-0000b390: 652d 696e 6665 7265 6e63 6520 7072 6f67  e-inference prog
-0000b3a0: 7261 6d20 7365 7420 6279 2060 2d74 7265  ram set by `-tre
-0000b3b0: 6560 206f 7220 602d 2d74 7265 655f 6d65  e` or `--tree_me
-0000b3c0: 7468 6f64 6020 746f 2062 6520 7072 652d  thod` to be pre-
-0000b3d0: 696e 7374 616c 6c65 6420 696e 2074 6865  installed in the
-0000b3e0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
-0000b3f0: 6820 7769 7468 2074 6865 2070 6172 616d  h with the param
-0000b400: 6574 6572 7320 7365 7474 696e 6720 666f  eters setting fo
-0000b410: 7220 6765 6e65 2074 7265 6520 696e 6665  r gene tree infe
-0000b420: 7265 6e63 6520 636f 6e74 726f 6c6c 6564  rence controlled
-0000b430: 2062 7920 602d 7473 6020 6f72 2060 2d2d   by `-ts` or `--
-0000b440: 7472 6565 7365 7460 292e 2054 6865 2070  treeset`). The p
-0000b450: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
-0000b460: 2077 6f75 6c64 2073 7469 6c6c 2063 6f6e   would still con
-0000b470: 6475 6374 2074 6865 2052 4248 7320 6361  duct the RBHs ca
-0000b480: 6c63 756c 6174 696f 6e20 756e 6c65 7373  lculation unless
-0000b490: 2074 6865 2070 6172 616d 6574 6572 2060   the parameter `
-0000b4a0: 2d6f 6f60 206f 7220 602d 2d6f 6e6c 796f  -oo` or `--onlyo
-0000b4b0: 7274 686f 6020 7761 7320 7365 742e 2049  rtho` was set. I
-0000b4c0: 6620 6f6e 6520 6f6e 6c79 2077 616e 7473  f one only wants
-0000b4d0: 2074 6f20 696e 6665 7220 7468 6520 6f72   to infer the or
-0000b4e0: 7468 6f67 726f 7570 732c 2069 7427 7320  thogroups, it's 
-0000b4f0: 7375 6767 6573 7465 6420 746f 2061 6464  suggested to add
-0000b500: 2074 6865 2066 6c61 6720 602d 6f6f 6020   the flag `-oo` 
-0000b510: 746f 206a 7573 7420 696d 706c 656d 656e  to just implemen
-0000b520: 7420 7468 6520 6f72 7468 6f67 726f 7570  t the orthogroup
-0000b530: 7320 6465 6c69 6e65 6174 696f 6e20 616e  s delineation an
-0000b540: 616c 7973 6973 2e20 5468 6520 6e75 6d62  alysis. The numb
-0000b550: 6572 206f 6620 7061 7261 6c6c 656c 2074  er of parallel t
-0000b560: 6872 6561 6473 2069 7320 7375 6767 6573  hreads is sugges
-0000b570: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
-0000b580: 2028 4e2b 3129 4e2f 3220 7768 6572 6520   (N+1)N/2 where 
-0000b590: 4e20 6973 2074 6865 206e 756d 6265 7220  N is the number 
-0000b5a0: 6f66 2063 6473 2066 696c 6573 2074 6f20  of cds files to 
-0000b5b0: 6163 6869 6576 6520 7468 6520 6869 6768  achieve the high
-0000b5c0: 6573 7420 6566 6669 6369 656e 6379 2073  est efficiency s
-0000b5d0: 696e 6365 2074 6865 2073 656c 662d 636f  ince the self-co
-0000b5e0: 6d70 6172 6973 6f6e 2069 7320 616c 736f  mparison is also
-0000b5f0: 2069 6e63 6c75 6465 642e 0a0a 2a2a 5468   included...**Th
-0000b600: 6520 6465 6661 756c 7420 7365 7474 696e  e default settin
-0000b610: 6720 6f66 2070 6172 616d 6574 6572 7320  g of parameters 
-0000b620: 6973 2061 2072 6561 736f 6e61 626c 6520  is a reasonable 
-0000b630: 7374 6172 7469 6e67 2070 6f69 6e74 2077  starting point w
-0000b640: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
-0000b650: 6173 2062 656c 6f77 2e2a 2a0a 0a60 6060  as below.**..```
-0000b660: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000b670: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000b680: 7565 6e63 6533 202d 6f69 202d 6f6f 0a60  uence3 -oi -oo.`
-0000b690: 6060 0a0a 2a2a 5468 6520 636f 6c6c 696e  ``..**The collin
-0000b6a0: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-0000b6b0: 696e 6665 7265 6e63 6520 6f66 2070 6879  inference of phy
-0000b6c0: 6c6f 6765 6e79 2a2a 0a60 6060 0a77 6764  logeny**.```.wgd
-0000b6d0: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
-0000b6e0: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
-0000b6f0: 6533 202d 6170 2061 7064 6174 6120 2d73  e3 -ap apdata -s
-0000b700: 6d20 736d 6461 7461 202d 6c65 206c 6564  m smdata -le led
-0000b710: 6174 6120 2d67 7420 6774 6461 7461 202d  ata -gt gtdata -
-0000b720: 636f 6320 282d 7472 6565 2027 6661 7374  coc (-tree 'fast
-0000b730: 7472 6565 2729 2028 2d74 7320 272d 6661  tree') (-ts '-fa
-0000b740: 7374 6573 7427 2920 282d 6e20 3429 2028  stest') (-n 4) (
-0000b750: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
-0000b760: 7329 2028 2d6f 2077 6764 5f64 6d64 2920  s) (-o wgd_dmd) 
-0000b770: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
-0000b780: 0a60 6060 0a0a 4120 6e6f 7665 6c20 7068  .```..A novel ph
-0000b790: 796c 6f67 656e 6574 6963 2069 6e66 6572  ylogenetic infer
-0000b7a0: 656e 6365 206d 6574 686f 6420 6e61 6d65  ence method name
-0000b7b0: 6420 2263 6f6c 6c69 6e65 6172 2063 6f61  d "collinear coa
-0000b7c0: 6c65 7363 656e 6365 2069 6e66 6572 656e  lescence inferen
-0000b7d0: 6365 2220 6973 2061 6c73 6f20 696d 706c  ce" is also impl
-0000b7e0: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
-0000b7f0: 7632 602e 2046 6f72 2074 6869 7320 616e  v2`. For this an
-0000b800: 616c 7973 6973 2c20 7573 6572 7320 6e65  alysis, users ne
-0000b810: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
-0000b820: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-0000b830: 6669 6c65 2062 7920 602d 6170 6020 6f72  file by `-ap` or
-0000b840: 2060 2d2d 616e 6368 6f72 706f 696e 7473   `--anchorpoints
-0000b850: 602c 2074 6865 2063 6f6c 6c69 6e65 6172  `, the collinear
-0000b860: 2073 6567 6d65 6e74 7320 6669 6c65 2062   segments file b
-0000b870: 7920 602d 736d 6020 6f72 2060 2d2d 7365  y `-sm` or `--se
-0000b880: 676d 656e 7473 602c 2074 6865 206c 6973  gments`, the lis
-0000b890: 7473 6567 6d65 6e74 7320 6669 6c65 2062  tsegments file b
-0000b8a0: 7920 602d 6c65 6020 6f72 2060 2d2d 6c69  y `-le` or `--li
-0000b8b0: 7374 656c 656d 656e 7473 602c 2061 6e64  stelements`, and
-0000b8c0: 2074 6865 2067 656e 6520 7461 626c 6520   the gene table 
-0000b8d0: 6669 6c65 2062 7920 602d 6774 6020 6f72  file by `-gt` or
-0000b8e0: 2060 2d2d 6765 6e65 7461 626c 6560 2c20   `--genetable`, 
-0000b8f0: 616c 6c20 6f66 2077 6869 6368 2063 616e  all of which can
-0000b900: 2062 6520 7072 6f64 7563 6564 2069 6e20   be produced in 
-0000b910: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
-0000b920: 2073 796e 602e 2054 6865 2070 6172 616d   syn`. The param
-0000b930: 6574 6572 2060 2d63 6f63 6020 6f72 2060  eter `-coc` or `
-0000b940: 2d2d 636f 6c6c 696e 6561 7263 6f61 6c65  --collinearcoale
-0000b950: 7363 656e 6365 6020 6e65 6564 7320 746f  scence` needs to
-0000b960: 2062 6520 7365 7420 746f 2073 7461 7274   be set to start
-0000b970: 2074 6869 7320 616e 616c 7973 6973 2e20   this analysis. 
-0000b980: 5468 6520 7472 6565 2d69 6e66 6572 656e  The tree-inferen
-0000b990: 6365 2070 726f 6772 616d 2061 6e64 2074  ce program and t
-0000b9a0: 6865 2061 7373 6f63 6961 7465 6420 7061  he associated pa
-0000b9b0: 7261 6d65 7465 7273 2063 616e 2062 6520  rameters can be 
-0000b9c0: 7365 7420 6a75 7374 2061 7320 6162 6f76  set just as abov
-0000b9d0: 6520 6279 2060 2d74 7265 6560 206f 7220  e by `-tree` or 
-0000b9e0: 602d 2d74 7265 655f 6d65 7468 6f64 6020  `--tree_method` 
-0000b9f0: 616e 6420 602d 7473 6020 6f72 2060 2d2d  and `-ts` or `--
-0000ba00: 7472 6565 7365 7460 2e20 506c 6561 7365  treeset`. Please
-0000ba10: 2061 6c73 6f20 6d61 6b65 2073 7572 6520   also make sure 
-0000ba20: 7468 6520 6368 6f73 656e 2074 7265 652d  the chosen tree-
-0000ba30: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-0000ba40: 6d20 6973 2069 6e73 7461 6c6c 6564 2069  m is installed i
-0000ba50: 6e20 7468 6520 656e 7669 726f 6e6d 656e  n the environmen
-0000ba60: 7420 7061 7468 2e20 5468 6520 7072 6f67  t path. The prog
-0000ba70: 7261 6d20 6061 7374 7261 6c2d 7072 6f60  ram `astral-pro`
-0000ba80: 2069 7320 7265 7175 6972 6564 2074 6f20   is required to 
-0000ba90: 6265 2069 6e73 7461 6c6c 6564 2069 6e20  be installed in 
-0000baa0: 7468 6520 656e 7669 726f 6e6d 656e 7420  the environment 
-0000bab0: 7061 7468 2074 6f6f 2e20 4e6f 7465 2074  path too. Note t
-0000bac0: 6861 7420 7468 6572 6520 7368 6f75 6c64  hat there should
-0000bad0: 2062 6520 6e6f 2064 7570 6c69 6361 7465   be no duplicate
-0000bae0: 6420 6765 6e65 2049 4473 2069 6e20 7468  d gene IDs in th
-0000baf0: 6520 7365 7175 656e 6365 2066 696c 652e  e sequence file.
-0000bb00: 2054 6865 2070 6172 616c 6c65 6c20 7468   The parallel th
-0000bb10: 7265 6164 7320 6865 7265 2061 7265 2074  reads here are t
-0000bb20: 6f20 6163 6365 6c65 7261 7465 2074 6865  o accelerate the
-0000bb30: 2073 6571 7565 6e63 6520 616c 6967 6e6d   sequence alignm
-0000bb40: 656e 7420 616e 6420 6765 6e65 2074 7265  ent and gene tre
-0000bb50: 6520 696e 6665 7265 6e63 6520 666f 7220  e inference for 
-0000bb60: 6561 6368 2067 656e 6520 6661 6d69 6c79  each gene family
-0000bb70: 2061 6e64 2074 6875 7320 7375 6767 6573   and thus sugges
-0000bb80: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
-0000bb90: 206d 7563 6820 6173 2074 6865 206e 756d   much as the num
-0000bba0: 6265 7220 6f66 2067 656e 6520 6661 6d69  ber of gene fami
-0000bbb0: 6c69 6573 2e0a 0a2a 2a41 2073 7567 6765  lies...**A sugge
-0000bbc0: 7374 6564 2073 7461 7274 696e 6720 7275  sted starting ru
-0000bbd0: 6e20 6f66 2074 6869 7320 616e 616c 7973  n of this analys
-0000bbe0: 6973 2069 7320 7769 7468 2074 6865 2073  is is with the s
-0000bbf0: 696d 706c 6520 636f 6d6d 616e 6420 6265  imple command be
-0000bc00: 6c6f 772e 2a2a 0a0a 6060 600a 7767 6420  low.**..```.wgd 
-0000bc10: 646d 6420 7365 7175 656e 6365 3120 7365  dmd sequence1 se
-0000bc20: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000bc30: 3320 2d61 7020 6170 6461 7461 202d 736d  3 -ap apdata -sm
-0000bc40: 2073 6d64 6174 6120 2d6c 6520 6c65 6461   smdata -le leda
-0000bc50: 7461 202d 6774 2067 7464 6174 6120 2d63  ta -gt gtdata -c
-0000bc60: 6f63 0a60 6060 0a0a 2323 2320 7767 6420  oc.```..### wgd 
-0000bc70: 666f 6375 730a 0a2a 2a54 6865 2063 6f6e  focus..**The con
-0000bc80: 6361 7465 6e61 7469 6f6e 2d62 6173 6564  catenation-based
-0000bc90: 2f63 6f61 6c65 7363 656e 6365 2d62 6173  /coalescence-bas
-0000bca0: 6564 2070 6879 6c6f 6765 6e65 7469 6320  ed phylogenetic 
-0000bcb0: 696e 6665 7265 6e63 652a 2a0a 6060 600a  inference**.```.
-0000bcc0: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000bcd0: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000bce0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000bcf0: 2028 2d2d 636f 6e63 6174 656e 6174 696f   (--concatenatio
-0000bd00: 6e29 2028 2d2d 636f 616c 6573 6365 6e63  n) (--coalescenc
-0000bd10: 6529 2028 2d74 7265 6520 2766 6173 7474  e) (-tree 'fastt
-0000bd20: 7265 6527 2920 282d 7473 2027 2d66 6173  ree') (-ts '-fas
-0000bd30: 7465 7374 2729 2028 2d6e 2034 2920 282d  test') (-n 4) (-
-0000bd40: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
-0000bd50: 2920 282d 6f20 7767 645f 666f 6375 7329  ) (-o wgd_focus)
-0000bd60: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
-0000bd70: 290a 6060 600a 0a54 6865 2070 726f 6772  ).```..The progr
-0000bd80: 616d 2060 7767 6420 666f 6375 7360 2069  am `wgd focus` i
-0000bd90: 6d70 6c65 6d65 6e74 6564 2074 776f 2062  mplemented two b
-0000bda0: 6173 6963 2070 6879 6c6f 6765 6e65 7469  asic phylogeneti
-0000bdb0: 6320 696e 6665 7265 6e63 6520 6d65 7468  c inference meth
-0000bdc0: 6f64 732c 2069 2e65 2e2c 2063 6f6e 6361  ods, i.e., conca
-0000bdd0: 7465 6e61 7469 6f6e 2d62 6173 6564 2061  tenation-based a
-0000bde0: 6e64 2063 6f61 6c65 7363 656e 6365 2d62  nd coalescence-b
-0000bdf0: 6173 6564 206d 6574 686f 6473 2e20 546f  ased methods. To
-0000be00: 2069 6e69 7469 6174 6520 7468 6573 6520   initiate these 
-0000be10: 616e 616c 7973 6973 2c20 7573 6572 7320  analysis, users 
-0000be20: 6e65 6564 2074 6f20 7365 7420 7468 6520  need to set the 
-0000be30: 666c 6167 206f 7074 696f 6e20 602d 2d63  flag option `--c
-0000be40: 6f6e 6361 7465 6e61 7469 6f6e 6020 6f72  oncatenation` or
-0000be50: 2060 2d2d 636f 616c 6573 6365 6e63 6560   `--coalescence`
-0000be60: 2e20 5468 6520 636f 6e63 6174 656e 6174  . The concatenat
-0000be70: 696f 6e2d 6261 7365 6420 6d65 7468 6f64  ion-based method
-0000be80: 2069 6e63 6c75 6465 7320 6120 6665 7720   includes a few 
-0000be90: 6d61 6a6f 7220 7374 6570 732c 2069 2e65  major steps, i.e
-0000bea0: 2e2c 2074 6865 206d 756c 7469 706c 6520  ., the multiple 
-0000beb0: 7365 7175 656e 6365 2061 6c69 676e 6d65  sequence alignme
-0000bec0: 6e74 2028 4d53 4129 206f 6620 6561 6368  nt (MSA) of each
-0000bed0: 2067 656e 6520 6661 6d69 6c79 2c20 7468   gene family, th
-0000bee0: 6520 636f 6e63 6174 656e 6174 696f 6e20  e concatenation 
-0000bef0: 6f66 2061 6c6c 2067 656e 6520 6661 6d69  of all gene fami
-0000bf00: 6c69 6573 2061 6e64 2074 6865 6e20 7468  lies and then th
-0000bf10: 6520 6765 6e65 2074 7265 6520 2861 6c73  e gene tree (als
-0000bf20: 6f20 7370 6563 6965 7320 7472 6565 2069  o species tree i
-0000bf30: 6e20 7468 6973 2063 6173 6529 2069 6e66  n this case) inf
-0000bf40: 6572 656e 6365 2e20 5468 6520 636f 616c  erence. The coal
-0000bf50: 6573 6365 6e63 652d 6261 7365 6420 6d65  escence-based me
-0000bf60: 7468 6f64 2077 696c 6c20 696e 7374 6561  thod will instea
-0000bf70: 6420 7065 7266 6f72 6d20 7468 6520 4d53  d perform the MS
-0000bf80: 4120 6f66 2065 6163 6820 6765 6e65 2066  A of each gene f
-0000bf90: 616d 696c 7920 616e 6420 7468 6520 6765  amily and the ge
-0000bfa0: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
-0000bfb0: 6520 6261 7365 6420 6f6e 2065 6163 6820  e based on each 
-0000bfc0: 4d53 412c 2061 6e64 2074 6865 6e20 696e  MSA, and then in
-0000bfd0: 6665 7220 7468 6520 7370 6563 6965 7320  fer the species 
-0000bfe0: 7472 6565 2062 6173 6564 206f 6e20 7468  tree based on th
-0000bff0: 6573 6520 696e 6469 7669 6475 616c 2067  ese individual g
-0000c000: 656e 6520 7472 6565 732e 2054 6865 2074  ene trees. The t
-0000c010: 7265 652d 696e 6665 7265 6e63 6520 7072  ree-inference pr
-0000c020: 6f67 7261 6d20 616e 6420 7468 6520 6173  ogram and the as
-0000c030: 736f 6369 6174 6564 2070 6172 616d 6574  sociated paramet
-0000c040: 6572 7320 6361 6e20 6265 2073 6574 206a  ers can be set j
-0000c050: 7573 7420 6173 2061 626f 7665 2062 7920  ust as above by 
-0000c060: 602d 7472 6565 6020 6f72 2060 2d2d 7472  `-tree` or `--tr
-0000c070: 6565 5f6d 6574 686f 6460 2061 6e64 2060  ee_method` and `
-0000c080: 2d74 7360 206f 7220 602d 2d74 7265 6573  -ts` or `--trees
-0000c090: 6574 602e 2050 6c65 6173 6520 616c 736f  et`. Please also
-0000c0a0: 206d 616b 6520 7375 7265 2074 6865 2063   make sure the c
-0000c0b0: 686f 7365 6e20 7472 6565 2d69 6e66 6572  hosen tree-infer
-0000c0c0: 656e 6365 2070 726f 6772 616d 2069 7320  ence program is 
-0000c0d0: 696e 7374 616c 6c65 6420 696e 2074 6865  installed in the
-0000c0e0: 2065 6e76 6972 6f6e 6d65 6e74 2070 6174   environment pat
-0000c0f0: 682e 2054 6865 2070 726f 6772 616d 2060  h. The program `
-0000c100: 6173 7472 616c 2d70 726f 6020 6973 2072  astral-pro` is r
-0000c110: 6571 7569 7265 6420 746f 2062 6520 696e  equired to be in
-0000c120: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
-0000c130: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
-0000c140: 6966 2074 6865 2063 6f61 6c65 7363 656e  if the coalescen
-0000c150: 6365 2d62 6173 6564 206d 6574 686f 6420  ce-based method 
-0000c160: 6973 2063 686f 7365 6e2e 2054 6865 2070  is chosen. The p
-0000c170: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-0000c180: 6865 7265 2061 7265 2074 6f20 6163 6365  here are to acce
-0000c190: 6c65 7261 7465 2074 6865 2073 6571 7565  lerate the seque
-0000c1a0: 6e63 6520 616c 6967 6e6d 656e 7420 616e  nce alignment an
-0000c1b0: 6420 6765 6e65 2074 7265 6520 696e 6665  d gene tree infe
-0000c1c0: 7265 6e63 6520 666f 7220 6561 6368 2067  rence for each g
-0000c1d0: 656e 6520 6661 6d69 6c79 2074 6f6f 2061  ene family too a
-0000c1e0: 6e64 2074 6875 7320 7375 6767 6573 7465  nd thus suggeste
-0000c1f0: 6420 746f 2062 6520 7365 7420 6173 206d  d to be set as m
-0000c200: 7563 6820 6173 2074 6865 206e 756d 6265  uch as the numbe
-0000c210: 7220 6f66 2067 656e 6520 6661 6d69 6c69  r of gene famili
-0000c220: 6573 2e0a 0a2a 2a41 2073 7567 6765 7374  es...**A suggest
-0000c230: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-0000c240: 6f66 2074 6869 7320 616e 616c 7973 6973  of this analysis
-0000c250: 2069 7320 7769 7468 2074 6865 2073 696d   is with the sim
-0000c260: 706c 6520 636f 6d6d 616e 6420 6265 6c6f  ple command belo
-0000c270: 772e 2a2a 0a0a 6060 600a 7767 6420 666f  w.**..```.wgd fo
-0000c280: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
-0000c290: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
-0000c2a0: 2073 6571 7565 6e63 6533 202d 2d63 6f6e   sequence3 --con
-0000c2b0: 6361 7465 6e61 7469 6f6e 0a77 6764 2066  catenation.wgd f
-0000c2c0: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
-0000c2d0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000c2e0: 3220 7365 7175 656e 6365 3320 2d2d 636f  2 sequence3 --co
-0000c2f0: 616c 6573 6365 6e63 650a 6060 600a 0a2a  alescence.```..*
-0000c300: 2a54 6865 2066 756e 6374 696f 6e61 6c20  *The functional 
-0000c310: 616e 6e6f 7461 7469 6f6e 206f 6620 6765  annotation of ge
-0000c320: 6e65 2066 616d 696c 6965 732a 2a0a 6060  ne families**.``
-0000c330: 600a 7767 6420 666f 6375 7320 6661 6d69  `.wgd focus fami
-0000c340: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
-0000c350: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
-0000c360: 6533 202d 2d61 6e6e 6f74 6174 696f 6e20  e3 --annotation 
-0000c370: 6567 676e 6f67 202d 6564 2065 6464 6174  eggnog -ed eddat
-0000c380: 6120 2d2d 646d 6e62 2064 6264 6174 610a  a --dmnb dbdata.
-0000c390: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
-0000c3a0: 2060 7767 6420 666f 6375 7360 2061 6c73   `wgd focus` als
-0000c3b0: 6f20 6164 6465 6420 736f 6d65 2077 7261  o added some wra
-0000c3c0: 7070 696e 6720 6675 6e63 7469 6f6e 7320  pping functions 
-0000c3d0: 666f 7220 6675 6e63 7469 6f6e 616c 2061  for functional a
-0000c3e0: 6e6e 6f74 6174 696f 6e20 6f66 2067 656e  nnotation of gen
-0000c3f0: 6520 6661 6d69 6c69 6573 206f 6e20 7468  e families on th
-0000c400: 6520 686f 6f64 206f 6620 6461 7461 6261  e hood of databa
-0000c410: 7365 7320 616e 6420 736f 6674 7761 7265  ses and software
-0000c420: 7320 666f 7220 696e 7374 616e 6365 2060  s for instance `
-0000c430: 4567 674e 4f47 6020 616e 6420 6045 6767  EggNOG` and `Egg
-0000c440: 4e4f 472d 6d61 7070 6572 602e 2046 6f72  NOG-mapper`. For
-0000c450: 2074 6865 2061 6e6e 6f74 6174 696f 6e20   the annotation 
-0000c460: 7573 696e 6720 6045 6767 4e4f 472d 6d61  using `EggNOG-ma
-0000c470: 7070 6572 602c 2075 7365 7273 206e 6565  pper`, users nee
-0000c480: 6420 746f 2070 726f 7669 6465 2074 6865  d to provide the
-0000c490: 2070 6174 6820 746f 2074 6865 2065 6767   path to the egg
-0000c4a0: 4e4f 4720 616e 6e6f 7461 7469 6f6e 2064  NOG annotation d
-0000c4b0: 6174 6162 6173 6520 7669 6120 7468 6520  atabase via the 
-0000c4c0: 6f70 7469 6f6e 2060 2d65 6460 2c20 7468  option `-ed`, th
-0000c4d0: 6520 7061 7468 2074 6f20 7468 6520 6469  e path to the di
-0000c4e0: 616d 6f6e 642d 636f 6d70 6174 6962 6c65  amond-compatible
-0000c4f0: 2064 6174 6162 6173 6520 7669 6120 7468   database via th
-0000c500: 6520 6f70 7469 6f6e 2060 2d2d 646d 6e64  e option `--dmnd
-0000c510: 5f64 6260 2061 6e64 2074 6865 206f 7074  _db` and the opt
-0000c520: 696f 6e20 602d 2d61 6e6e 6f74 6174 696f  ion `--annotatio
-0000c530: 6e60 2073 6574 2061 7320 2265 6767 6e6f  n` set as "eggno
-0000c540: 6722 2e20 5468 6520 6d61 6e6e 6572 2068  g". The manner h
-0000c550: 6f77 2050 4641 4d20 616e 6e6f 7461 7469  ow PFAM annotati
-0000c560: 6f6e 2077 696c 6c20 6265 2070 6572 666f  on will be perfo
-0000c570: 726d 6564 2063 616e 2062 6520 636f 6e74  rmed can be cont
-0000c580: 726f 6c6c 6564 2076 6961 2074 6865 206f  rolled via the o
-0000c590: 7074 696f 6e20 6070 6661 6d60 2c20 6569  ption `pfam`, ei
-0000c5a0: 7468 6572 2022 6e6f 6e65 222c 2022 7265  ther "none", "re
-0000c5b0: 616c 6967 6e22 206f 7220 2264 656e 6f76  align" or "denov
-0000c5c0: 6f22 2c20 7468 6520 6465 7461 696c 6564  o", the detailed
-0000c5d0: 2065 7870 6c61 6e61 7469 6f6e 2063 616e   explanation can
-0000c5e0: 2062 6520 666f 756e 6420 6174 2074 6865   be found at the
-0000c5f0: 2077 696b 6920 6f66 205b 4567 674e 4f47   wiki of [EggNOG
-0000c600: 2d6d 6170 7065 725d 2868 7474 7073 3a2f  -mapper](https:/
-0000c610: 2f67 6974 6875 622e 636f 6d2f 6567 676e  /github.com/eggn
-0000c620: 6f67 6462 2f65 6767 6e6f 672d 6d61 7070  ogdb/eggnog-mapp
-0000c630: 6572 2f77 696b 6929 2e20 506c 6561 7365  er/wiki). Please
-0000c640: 2070 7265 2d69 6e73 7461 6c6c 2074 6865   pre-install the
-0000c650: 2060 4567 674e 4f47 2d6d 6170 7065 7260   `EggNOG-mapper`
-0000c660: 2070 7974 686f 6e20 7061 636b 6167 6520   python package 
-0000c670: 6966 2075 7369 6e67 2074 6869 7320 6675  if using this fu
-0000c680: 6e63 7469 6f6e 2e20 466f 7220 7468 6520  nction. For the 
-0000c690: 616e 6e6f 7461 7469 6f6e 2075 7369 6e67  annotation using
-0000c6a0: 2060 686d 6d73 6361 6e60 2c20 7768 6174   `hmmscan`, what
-0000c6b0: 2069 7320 696d 706c 656d 656e 7465 6420   is implemented 
-0000c6c0: 696e 2060 7767 6420 7632 6020 6973 2061  in `wgd v2` is a
-0000c6d0: 2073 696d 706c 6520 6275 6e64 6c65 2066   simple bundle f
-0000c6e0: 756e 6374 696f 6e20 746f 2070 6572 666f  unction to perfo
-0000c6f0: 726d 2060 686d 6d73 6361 6e60 2061 6e61  rm `hmmscan` ana
-0000c700: 6c79 7369 7320 666f 7220 6120 6769 7665  lysis for a give
-0000c710: 6e20 686d 6d20 7072 6f66 696c 6520 616e  n hmm profile an
-0000c720: 6420 7365 7420 6f66 2067 656e 6520 6661  d set of gene fa
-0000c730: 6d69 6c69 6573 2073 7563 6820 7468 6174  milies such that
-0000c740: 2075 7365 7273 206e 6565 6420 746f 2073   users need to s
-0000c750: 6574 2074 6865 206f 7074 696f 6e20 602d  et the option `-
-0000c760: 2d61 6e6e 6f74 6174 696f 6e60 2061 7320  -annotation` as 
-0000c770: 2268 6d6d 7066 616d 2220 616e 6420 7072  "hmmpfam" and pr
-0000c780: 6f76 6964 6520 7468 6520 686d 6d70 726f  ovide the hmmpro
-0000c790: 6669 6c65 2076 6961 2074 6865 206f 7074  file via the opt
-0000c7a0: 696f 6e20 602d 2d68 6d6d 602e 2046 6f72  ion `--hmm`. For
-0000c7b0: 2074 6865 2061 6e6e 6f74 6174 696f 6e20   the annotation 
-0000c7c0: 7573 696e 6720 6069 6e74 6572 7072 6f73  using `interpros
-0000c7d0: 6361 6e60 2c20 7573 6572 7320 6e65 6564  can`, users need
-0000c7e0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-0000c7f0: 7061 7468 2074 6f20 7468 6520 696e 7465  path to the inte
-0000c800: 7270 726f 7363 616e 2069 6e73 7461 6c6c  rproscan install
-0000c810: 6174 696f 6e20 666f 6c64 6572 2077 6865  ation folder whe
-0000c820: 7265 2074 6865 7265 2069 7320 6120 6069  re there is a `i
-0000c830: 6e74 6572 7072 6f73 6361 6e2e 7368 6020  nterproscan.sh` 
-0000c840: 6669 6c65 2076 6961 2074 6865 206f 7074  file via the opt
-0000c850: 696f 6e20 602d 2d65 7865 7061 7468 6020  ion `--exepath` 
-0000c860: 616e 6420 7365 7420 7468 6520 7468 6520  and set the the 
-0000c870: 6f70 7469 6f6e 2060 2d2d 616e 6e6f 7461  option `--annota
-0000c880: 7469 6f6e 6020 6173 2022 696e 7465 7270  tion` as "interp
-0000c890: 726f 7363 616e 222e 2054 6865 2070 6172  roscan". The par
-0000c8a0: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000c8b0: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
-0000c8c0: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
-0000c8d0: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
-0000c8e0: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
-0000c8f0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000c900: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
-0000c910: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
-0000c920: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
-0000c930: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000c940: 7469 6e67 2072 756e 206f 6620 7468 6973  ting run of this
-0000c950: 2061 6e61 6c79 7369 7320 6361 6e20 6265   analysis can be
-0000c960: 2077 6974 6820 7468 6520 636f 6d6d 616e   with the comman
-0000c970: 6420 6265 6c6f 772e 2a2a 0a0a 6060 600a  d below.**..```.
-0000c980: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000c990: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000c9a0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000c9b0: 202d 2d61 6e6e 6f74 6174 696f 6e20 6567   --annotation eg
-0000c9c0: 676e 6f67 202d 6564 2065 6464 6174 6120  gnog -ed eddata 
-0000c9d0: 2d2d 646d 6e62 2064 6264 6174 610a 7767  --dmnb dbdata.wg
-0000c9e0: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000c9f0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000ca00: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000ca10: 2d61 6e6e 6f74 6174 696f 6e20 686d 6d70  -annotation hmmp
-0000ca20: 6661 6d20 2d2d 686d 6d20 686d 6d64 6174  fam --hmm hmmdat
-0000ca30: 610a 7767 6420 666f 6375 7320 6661 6d69  a.wgd focus fami
-0000ca40: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
-0000ca50: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
-0000ca60: 6533 202d 2d61 6e6e 6f74 6174 696f 6e20  e3 --annotation 
-0000ca70: 696e 7465 7270 726f 7363 616e 202d 2d65  interproscan --e
-0000ca80: 7865 7061 7468 2024 5041 5448 0a60 6060  xepath $PATH.```
-0000ca90: 0a0a 2a2a 5468 6520 7068 796c 6f67 656e  ..**The phylogen
-0000caa0: 6574 6963 2064 6174 696e 6720 6f66 2057  etic dating of W
-0000cab0: 4744 732a 2a0a 6060 600a 7767 6420 666f  GDs**.```.wgd fo
-0000cac0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
-0000cad0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
-0000cae0: 2073 6571 7565 6e63 6533 202d 6420 6d63   sequence3 -d mc
-0000caf0: 6d63 7472 6565 202d 7370 2073 7064 6174  mctree -sp spdat
-0000cb00: 6120 282d 2d70 726f 7463 6f63 6461 7469  a (--protcocdati
-0000cb10: 6e67 2920 282d 2d70 6172 7469 7469 6f6e  ng) (--partition
-0000cb20: 2920 282d 2d61 616d 6f64 656c 206c 6729  ) (--aamodel lg)
-0000cb30: 2028 2d64 7320 2762 7572 6e69 6e20 3d20   (-ds 'burnin = 
-0000cb40: 3230 3030 2729 2028 2d64 7320 2773 616d  2000') (-ds 'sam
-0000cb50: 7066 7265 7120 3d20 3130 3030 2729 2028  pfreq = 1000') (
-0000cb60: 2d64 7320 276e 7361 6d70 6c65 203d 2032  -ds 'nsample = 2
-0000cb70: 3030 3030 2729 2020 282d 6e20 3429 2028  0000')  (-n 4) (
-0000cb80: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
-0000cb90: 7329 2028 2d6f 2077 6764 5f66 6f63 7573  s) (-o wgd_focus
-0000cba0: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
-0000cbb0: 7029 0a60 6060 0a0a 5468 6520 6162 736f  p).```..The abso
-0000cbc0: 6c75 7465 2064 6174 696e 6720 6f66 2057  lute dating of W
-0000cbd0: 4744 7320 6973 2061 2073 7065 6369 6669  GDs is a specifi
-0000cbe0: 6320 7069 7065 6c69 6e65 2069 6d70 6c65  c pipeline imple
-0000cbf0: 6d65 6e74 6564 2069 6e20 6077 6764 2076  mented in `wgd v
-0000cc00: 3260 2075 7369 6e67 2074 6865 206d 6574  2` using the met
-0000cc10: 686f 6420 6f66 2070 6879 6c6f 6765 6e65  hod of phylogene
-0000cc20: 7469 6320 6461 7469 6e67 2e20 5468 6520  tic dating. The 
-0000cc30: 6661 6d69 6c69 6573 2075 7365 6420 696e  families used in
-0000cc40: 2074 6869 7320 7374 6570 2063 616e 2062   this step can b
-0000cc50: 6520 7072 6f64 7563 6564 2066 726f 6d20  e produced from 
-0000cc60: 6077 6764 2064 6d64 6020 616e 6420 6077  `wgd dmd` and `w
-0000cc70: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
-0000cc80: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
-0000cc90: 2064 6973 6375 7373 2068 6f77 2074 6f20   discuss how to 
-0000cca0: 6461 7465 2057 4744 7320 7769 7468 2067  date WGDs with g
-0000ccb0: 656e 6f6d 6520 6173 7365 6d62 6c79 2c20  enome assembly, 
-0000ccc0: 696e 7374 6561 6420 6f66 2074 7261 6e73  instead of trans
-0000ccd0: 6372 6970 746f 6d65 2061 7373 656d 626c  criptome assembl
-0000cce0: 7920 2877 6869 6368 2077 696c 6c20 6265  y (which will be
-0000ccf0: 2064 6973 6375 7373 6564 2069 6e20 6120   discussed in a 
-0000cd00: 7365 7061 7261 7465 2073 6563 7469 6f6e  separate section
-0000cd10: 2068 6572 6575 6e64 6572 292e 2054 6865   hereunder). The
-0000cd20: 2061 7373 756d 7074 696f 6e20 7765 206d   assumption we m
-0000cd30: 6164 6520 6865 7265 2069 7320 7468 6174  ade here is that
-0000cd40: 206e 6f74 2061 6c6c 2061 6e63 686f 7220   not all anchor 
-0000cd50: 7061 6972 7320 2863 6f6c 6c69 6e65 6172  pairs (collinear
-0000cd60: 2064 7570 6c69 6361 7465 7329 2061 7265   duplicates) are
-0000cd70: 2073 7569 7461 626c 6520 666f 7220 7068   suitable for ph
-0000cd80: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-0000cd90: 672c 2066 6f72 2069 6e73 7461 6e63 6520  g, for instance 
-0000cda0: 7468 6f73 6520 6661 7374 6c79 206f 7220  those fastly or 
-0000cdb0: 736c 6f77 6c79 2065 766f 6c76 696e 6720  slowly evolving 
-0000cdc0: 6765 6e65 2064 7570 6c69 6361 7465 732c  gene duplicates,
-0000cdd0: 2062 6563 6175 7365 2074 6865 7927 7265   because they're
-0000cde0: 2070 726f 6e65 2074 6f20 6769 7665 2062   prone to give b
-0000cdf0: 6961 7365 6420 6461 7469 6e67 2065 7374  iased dating est
-0000ce00: 696d 6174 696f 6e2e 2053 6f20 6173 2074  imation. So as t
-0000ce10: 6f20 7265 7472 6965 7665 2074 6865 2022  o retrieve the "
-0000ce20: 7265 6c69 6162 6c65 2220 616e 6368 6f72  reliable" anchor
-0000ce30: 2070 6169 7273 2c20 7765 2069 6d70 6c65   pairs, we imple
-0000ce40: 6d65 6e74 6564 2073 6f6d 6520 6d65 7468  mented some meth
-0000ce50: 6f64 7320 6f66 2069 6465 6e74 6966 7969  ods of identifyi
-0000ce60: 6e67 2063 7265 6469 6162 6c65 2061 6e63  ng crediable anc
-0000ce70: 686f 7220 7061 6972 7320 6261 7365 6420  hor pairs based 
-0000ce80: 6f6e 2074 6865 6972 204b 7320 7661 6c75  on their Ks valu
-0000ce90: 6573 2061 6e64 2f6f 7220 7265 7369 6469  es and/or residi
-0000cea0: 6e67 2063 6f6c 6c69 6e65 6172 2073 6567  ng collinear seg
-0000ceb0: 6d65 6e74 732e 2046 6f72 2061 2067 656e  ments. For a gen
-0000cec0: 6f6d 6520 7769 7468 2063 6c65 6172 2073  ome with clear s
-0000ced0: 6967 6e61 6c73 206f 6620 7075 7461 7469  ignals of putati
-0000cee0: 7665 2057 4744 732c 2073 7563 6820 6173  ve WGDs, such as
-0000cef0: 2074 6865 202a 4171 7569 6c65 6769 6120   the *Aquilegia 
-0000cf00: 636f 6572 756c 6561 2a20 696e 2074 6865  coerulea* in the
-0000cf10: 2065 7861 6d70 6c65 2c20 6120 6865 7572   example, a heur
-0000cf20: 6973 7469 6320 6d65 7468 6f64 2074 6861  istic method tha
-0000cf30: 7420 6170 706c 6965 7320 7468 6520 7072  t applies the pr
-0000cf40: 696e 6369 706c 6520 6f66 2068 6f77 205b  inciple of how [
-0000cf50: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
-0000cf60: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
-0000cf70: 5053 422f 6b73 7261 7465 7329 2066 696e  PSB/ksrates) fin
-0000cf80: 6420 7468 6520 696e 6974 6961 6c20 7065  d the initial pe
-0000cf90: 616b 7320 616e 6420 7468 6569 7220 7061  aks and their pa
-0000cfa0: 7261 6d65 7465 7273 2077 6173 2069 6d70  rameters was imp
-0000cfb0: 6c65 6d65 6e74 6564 2074 6f20 6669 6e64  lemented to find
-0000cfc0: 2074 6865 2039 3525 2063 6f6e 6669 6465   the 95% confide
-0000cfd0: 6e63 6520 6c65 7665 6c20 6f66 2074 6865  nce level of the
-0000cfe0: 2061 7373 756d 6564 206c 6f67 6e6f 726d   assumed lognorm
-0000cff0: 616c 2064 6973 7472 6962 7574 696f 6e20  al distribution 
-0000d000: 6f66 2074 6865 2061 6e63 686f 7220 7061  of the anchor pa
-0000d010: 6972 202a 4b2a 3c73 7562 3e53 3c2f 7375  ir *K*<sub>S</su
-0000d020: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000d030: 696f 6e20 746f 2066 696c 7465 7220 616e  ion to filter an
-0000d040: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
-0000d050: 746f 6f20 6869 6768 206f 7220 6c6f 7720  too high or low 
-0000d060: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000d070: 6167 6573 2c20 7468 6520 6578 616d 706c  ages, the exampl
-0000d080: 6172 2063 6f6d 6d61 6e64 206f 6620 7768  ar command of wh
-0000d090: 6963 6820 6973 2073 686f 7765 6420 696e  ich is showed in
-0000d0a0: 2074 6865 2060 7767 6420 7065 616b 6020   the `wgd peak` 
-0000d0b0: 7365 6374 696f 6e20 6f66 205b 496c 6c75  section of [Illu
-0000d0c0: 7374 7261 7469 6f6e 5d28 6874 7470 733a  stration](https:
-0000d0d0: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
-0000d0e0: 6865 2d70 7362 2f77 6764 3f74 6162 3d72  he-psb/wgd?tab=r
-0000d0f0: 6561 646d 652d 6f76 2d66 696c 6523 696c  eadme-ov-file#il
-0000d100: 6c75 7374 7261 7469 6f6e 292e 2049 6620  lustration). If 
-0000d110: 7468 6973 2068 6575 7269 7374 6963 206d  this heuristic m
-0000d120: 6574 686f 6420 6661 696c 6564 2074 6f20  ethod failed to 
-0000d130: 6769 7665 2072 6561 736f 6e61 626c 6520  give reasonable 
-0000d140: 7265 7375 6c74 732c 2077 6869 6368 2069  results, which i
-0000d150: 7320 7573 7561 6c6c 7920 6475 6520 746f  s usually due to
-0000d160: 2074 6865 2065 6666 6563 7420 6f66 206d   the effect of m
-0000d170: 756c 7469 706c 6520 6164 6a61 6365 6e74  ultiple adjacent
-0000d180: 2057 4744 7320 7468 6174 2062 6c75 7273   WGDs that blurs
-0000d190: 2074 6865 2070 6561 6b2d 6669 6e64 696e   the peak-findin
-0000d1a0: 6720 7072 6f63 6573 732c 2075 7365 7273  g process, users
-0000d1b0: 2063 616e 2074 7572 6e20 746f 2074 6865   can turn to the
-0000d1c0: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
-0000d1d0: 6e74 732d 6775 6964 6564 2061 6e63 686f  nts-guided ancho
-0000d1e0: 7220 7061 6972 2063 6c75 7374 6572 696e  r pair clusterin
-0000d1f0: 6720 696d 706c 656d 656e 7465 6420 696e  g implemented in
-0000d200: 2060 7767 6420 7632 602c 2069 6e20 7768   `wgd v2`, in wh
-0000d210: 6963 6820 6120 636f 6c6c 696e 6561 7220  ich a collinear 
-0000d220: 7365 676d 656e 742d 7769 7365 2047 4d4d  segment-wise GMM
-0000d230: 2063 6c75 7374 6572 696e 6720 7769 6c6c   clustering will
-0000d240: 2062 6520 6669 7273 7420 636f 6e64 7563   be first conduc
-0000d250: 7465 6420 6261 7365 6420 6f6e 2074 6865  ted based on the
-0000d260: 2022 736f 2d63 616c 6c65 6422 2073 6567   "so-called" seg
-0000d270: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
-0000d280: 7375 623e 2061 6765 2072 6570 7265 7365  sub> age represe
-0000d290: 6e74 6564 2062 7920 7468 6520 6d65 6469  nted by the medi
-0000d2a0: 616e 202a 4b2a 3c73 7562 3e53 3c2f 7375  an *K*<sub>S</su
-0000d2b0: 623e 2061 6765 206f 6620 616c 6c20 7468  b> age of all th
-0000d2c0: 6520 7265 7369 6469 6e67 2073 796e 7465  e residing synte
-0000d2d0: 6c6f 6773 2028 6e6f 7465 2074 6861 7420  logs (note that 
-0000d2e0: 7468 6520 6765 6e65 2064 7570 6c69 6361  the gene duplica
-0000d2f0: 7465 2070 6169 7273 2061 646f 7074 6564  te pairs adopted
-0000d300: 2069 6e20 7468 6973 2073 7465 7020 6973   in this step is
-0000d310: 2066 726f 6d20 7468 6520 6669 6c65 2060   from the file `
-0000d320: 6d75 6c74 6970 6c69 636f 6e5f 7061 6972  multiplicon_pair
-0000d330: 732e 7478 7460 2077 6869 6368 2063 6f6e  s.txt` which con
-0000d340: 7461 696e 7320 7468 6520 6675 6c6c 2073  tains the full s
-0000d350: 6574 206f 6620 7379 6e74 656c 6f67 7329  et of syntelogs)
-0000d360: 2c20 696e 7374 6561 6420 6f66 2074 6865  , instead of the
-0000d370: 2073 6d61 6c6c 6572 2067 656e 6520 7365   smaller gene se
-0000d380: 7420 6f66 2061 6e63 686f 7220 7061 6972  t of anchor pair
-0000d390: 732e 2054 6865 2064 6973 7469 6e63 7469  s. The distincti
-0000d3a0: 6f6e 2062 6574 7765 656e 2061 6e63 686f  on between ancho
-0000d3b0: 7220 7061 6972 7320 616e 6420 7379 6e74  r pairs and synt
-0000d3c0: 656c 6f67 7320 6973 2074 6861 7420 7468  elogs is that th
-0000d3d0: 6520 6c61 7474 6572 2072 6566 6572 7320  e latter refers 
-0000d3e0: 746f 206d 756c 7469 706c 6520 7365 7473  to multiple sets
-0000d3f0: 206f 6620 6765 6e65 7320 6465 7269 7665   of genes derive
-0000d400: 6420 6672 6f6d 2074 6865 2073 616d 6520  d from the same 
-0000d410: 616e 6365 7374 7261 6c20 6765 6e6f 6d69  ancestral genomi
-0000d420: 6320 7265 6769 6f6e 2077 6869 6c65 2074  c region while t
-0000d430: 6865 2066 6f72 6d65 7220 696d 706c 6965  he former implie
-0000d440: 7320 7468 6520 6c61 7474 6572 2062 7574  s the latter but
-0000d450: 2072 6571 7569 7265 7320 6164 6469 7469   requires additi
-0000d460: 6f6e 616c 6c79 2074 6865 2063 6f6e 7365  onally the conse
-0000d470: 7276 6564 2067 656e 6520 6f72 6465 722c  rved gene order,
-0000d480: 2062 6f74 6820 6f66 2077 6869 6368 2061   both of which a
-0000d490: 7265 2c20 7769 7468 696e 2061 2067 656e  re, within a gen
-0000d4a0: 6f6d 652c 2061 7373 756d 6564 2074 6f20  ome, assumed to 
-0000d4b0: 6265 206f 7269 6769 6e61 7465 6420 6672  be originated fr
-0000d4c0: 6f6d 2074 6865 2064 7570 6c69 6361 7469  om the duplicati
-0000d4d0: 6f6e 206f 6620 6120 636f 6d6d 6f6e 2061  on of a common a
-0000d4e0: 6e63 6573 7472 616c 2067 656e 6f6d 6963  ncestral genomic
-0000d4f0: 2072 6567 696f 6e20 616e 6420 6173 2073   region and as s
-0000d500: 7563 6820 6465 656d 6564 2065 7669 6465  uch deemed evide
-0000d510: 6e63 6520 666f 7220 5747 442e 2054 6865  nce for WGD. The
-0000d520: 6e20 7468 6520 7379 6e74 656c 6f67 7320  n the syntelogs 
-0000d530: 7769 6c6c 2062 6520 6d61 7070 6564 2062  will be mapped b
-0000d540: 6163 6b20 6163 636f 7264 696e 6720 746f  ack according to
-0000d550: 2074 6865 2063 6c75 7374 6572 696e 6720   the clustering 
-0000d560: 7265 7375 6c74 7320 6f66 2074 6865 6972  results of their
-0000d570: 2061 6666 696c 6961 7465 6420 7365 676d   affiliated segm
-0000d580: 656e 7473 2e20 5369 6e63 6520 7468 6520  ents. Since the 
-0000d590: 4761 7573 7369 616e 2073 6861 7065 206f  Gaussian shape o
-0000d5a0: 6620 7468 6520 7365 676d 656e 7420 636c  f the segment cl
-0000d5b0: 7573 7465 7220 646f 6573 6e27 7420 6e65  uster doesn't ne
-0000d5c0: 6365 7373 6172 696c 7920 696d 6974 6174  cessarily imitat
-0000d5d0: 6520 7468 6520 7368 6170 6520 6f66 2074  e the shape of t
-0000d5e0: 6865 2072 6573 6964 696e 6720 7379 6e74  he residing synt
-0000d5f0: 656c 6f67 732c 2073 6f20 6173 2074 6f20  elogs, so as to 
-0000d600: 7265 7472 6965 7665 2074 6865 2022 7265  retrieve the "re
-0000d610: 6c69 6162 6c65 2220 6765 6e65 2073 6574  liable" gene set
-0000d620: 7320 666f 7220 7068 796c 6f67 656e 6574  s for phylogenet
-0000d630: 6963 2064 6174 696e 672c 2077 6520 6164  ic dating, we ad
-0000d640: 706f 7465 6420 7468 6520 2839 3525 2920  poted the (95%) 
-0000d650: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
-0000d660: 7265 6769 6f6e 2028 4844 5229 206f 6620  region (HDR) of 
-0000d670: 7468 6520 7379 6e74 656c 6f67 202a 4b2a  the syntelog *K*
-0000d680: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-0000d690: 2064 6973 7472 6962 7574 696f 6e20 666f   distribution fo
-0000d6a0: 7220 7468 6520 7068 796c 6f67 656e 6574  r the phylogenet
-0000d6b0: 6963 2064 6174 696e 672c 2074 6865 2063  ic dating, the c
-0000d6c0: 616c 6375 6c61 7469 6f6e 206f 6620 7468  alculation of th
-0000d6d0: 6520 2839 3525 2920 4844 5220 696e 2074  e (95%) HDR in t
-0000d6e0: 6865 2066 756e 6374 696f 6e20 5b63 616c  he function [cal
-0000d6f0: 6375 6c61 7465 4850 445d 2868 7474 7073  culateHPD](https
-0000d700: 3a2f 2f67 6974 6875 622e 636f 6d2f 6865  ://github.com/he
-0000d710: 6368 652d 7073 622f 7767 642f 626c 6f62  che-psb/wgd/blob
-0000d720: 2f70 6879 6c6f 6461 7469 6e67 2f77 6764  /phylodating/wgd
-0000d730: 2f70 6561 6b2e 7079 234c 3137 3537 2920  /peak.py#L1757) 
-0000d740: 7365 656b 7320 7468 6520 7368 6f72 7465  seeks the shorte
-0000d750: 7374 202a 4b2a 3c73 7562 3e53 3c2f 7375  st *K*<sub>S</su
-0000d760: 623e 2072 616e 6765 2028 612c 6229 2077  b> range (a,b) w
-0000d770: 6869 6368 2073 6174 6973 6669 6573 2074  hich satisfies t
-0000d780: 6865 2072 6571 7569 7265 6d65 6e74 206f  he requirement o
-0000d790: 6620 7370 616e 6e69 6e67 206d 6f72 6520  f spanning more 
-0000d7a0: 7468 616e 2028 3935 2529 206f 6620 616c  than (95%) of al
-0000d7b0: 6c20 7468 6520 2a4b 2a3c 7375 623e 533c  l the *K*<sub>S<
-0000d7c0: 2f73 7562 3e20 7661 6c75 6573 2e20 4f6e  /sub> values. On
-0000d7d0: 2074 6865 2070 7265 6d69 7365 206f 6620   the premise of 
-0000d7e0: 6964 656e 7469 6669 6564 2061 6e63 686f  identified ancho
-0000d7f0: 7220 7061 6972 7320 286e 6f74 6520 7468  r pairs (note th
-0000d800: 6174 2074 6865 2073 796e 7465 6c6f 6773  at the syntelogs
-0000d810: 2061 7265 2061 6c73 6f20 7265 6665 7272   are also referr
-0000d820: 6564 2061 7320 616e 6368 6f72 2070 6169  ed as anchor pai
-0000d830: 7273 2068 6572 6561 6674 6572 292c 2077  rs hereafter), w
-0000d840: 6520 696d 706c 656d 656e 7465 6420 696e  e implemented in
-0000d850: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
-0000d860: 6420 646d 6460 2074 6865 2073 6f2d 6361  d dmd` the so-ca
-0000d870: 6c6c 6564 2061 6e63 686f 722d 6177 6172  lled anchor-awar
-0000d880: 6520 6c6f 6361 6c20 4d52 4248 7320 6f72  e local MRBHs or
-0000d890: 206f 7274 686f 6772 6f75 7073 2c20 696e   orthogroups, in
-0000d8a0: 2077 6869 6368 2074 6865 206f 7269 6769   which the origi
-0000d8b0: 6e61 6c20 6c6f 6361 6c20 4d52 4248 7320  nal local MRBHs 
-0000d8c0: 6172 6520 6675 7274 6865 7220 6d65 7267  are further merg
-0000d8d0: 6564 2077 6974 6820 616e 6368 6f72 2070  ed with anchor p
-0000d8e0: 6169 7273 2073 7563 6820 7468 6174 2065  airs such that e
-0000d8f0: 6163 6820 6f72 7468 6f67 726f 7570 2063  ach orthogroup c
-0000d900: 6f6e 7461 696e 7320 7468 6520 616e 6368  ontains the anch
-0000d910: 6f72 2070 6169 7220 616e 6420 7468 6520  or pair and the 
-0000d920: 6f72 7468 6f6c 6f67 7565 732e 2057 6974  orthologues. Wit
-0000d930: 6820 7468 6973 206f 7274 686f 6772 6f75  h this orthogrou
-0000d940: 702c 2075 7365 7273 2074 6865 6e20 6e65  p, users then ne
-0000d950: 6564 206f 6e65 2073 7461 7274 696e 6720  ed one starting 
-0000d960: 7472 6565 2066 696c 6520 2861 7320 7368  tree file (as sh
-0000d970: 6f77 6e20 696e 2074 6865 205b 496c 6c75  own in the [Illu
-0000d980: 7374 7261 7469 6f6e 5d28 6874 7470 733a  stration](https:
-0000d990: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
-0000d9a0: 6865 2d70 7362 2f77 6764 2f74 7265 652f  he-psb/wgd/tree/
-0000d9b0: 7068 796c 6f64 6174 696e 673f 7461 623d  phylodating?tab=
-0000d9c0: 7265 6164 6d65 2d6f 762d 6669 6c65 2369  readme-ov-file#i
-0000d9d0: 6c6c 7573 7472 6174 696f 6e29 2070 6172  llustration) par
-0000d9e0: 7429 2069 6e64 6963 6174 696e 6720 7468  t) indicating th
-0000d9f0: 6520 7472 6565 2074 6f70 6f6c 6f67 7920  e tree topology 
-0000da00: 616e 6420 666f 7373 696c 2063 616c 6962  and fossil calib
-0000da10: 7261 7469 6f6e 2069 6e66 6f72 6d61 7469  ration informati
-0000da20: 6f6e 2066 6f72 2074 6865 2066 696e 616c  on for the final
-0000da30: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
-0000da40: 7469 6e67 2075 7369 6e67 2074 6865 2070  ting using the p
-0000da50: 726f 6772 616d 2060 7767 6420 666f 6375  rogram `wgd focu
-0000da60: 7360 2e20 5573 6572 7320 6361 6e20 7365  s`. Users can se
-0000da70: 7420 7468 6520 666c 6167 206f 7074 696f  t the flag optio
-0000da80: 6e20 602d 2d70 726f 7463 6f63 6461 7469  n `--protcocdati
-0000da90: 6e67 6020 746f 206f 6e6c 7920 636f 6e64  ng` to only cond
-0000daa0: 7563 7420 7468 6520 6461 7469 6e67 206f  uct the dating o
-0000dab0: 6620 7468 6520 636f 6e63 6174 656e 6174  f the concatenat
-0000dac0: 696f 6e20 7072 6f74 6569 6e20 4d53 4120  ion protein MSA 
-0000dad0: 6f72 2074 6865 2066 6c61 6720 6f70 7469  or the flag opti
-0000dae0: 6f6e 2060 2d2d 7072 6f74 6461 7469 6e67  on `--protdating
-0000daf0: 6020 746f 206f 6e6c 7920 636f 6e64 7563  ` to only conduc
-0000db00: 7420 7468 6520 6461 7469 6e67 206f 6620  t the dating of 
-0000db10: 7468 6520 7072 6f74 6569 6e20 4d53 4173  the protein MSAs
-0000db20: 2c20 6e6f 7465 6420 7468 6174 2074 6865  , noted that the
-0000db30: 7365 2074 776f 206f 7074 696f 6e73 206f  se two options o
-0000db40: 6e6c 7920 776f 726b 2066 6f72 2074 6865  nly work for the
-0000db50: 2060 6d63 6d63 7472 6565 6020 6f70 7469   `mcmctree` opti
-0000db60: 6f6e 2073 6f20 6661 722e 2054 6865 2066  on so far. The f
-0000db70: 6c61 6720 6f70 7469 6f6e 2060 2d2d 7061  lag option `--pa
-0000db80: 7274 6974 696f 6e60 2063 616e 2062 6520  rtition` can be 
-0000db90: 7365 7420 746f 2070 6572 666f 726d 2060  set to perform `
-0000dba0: 6d63 6d63 7472 6565 6020 616e 616c 7973  mcmctree` analys
-0000dbb0: 6973 2075 7369 6e67 2074 6865 2070 6172  is using the par
-0000dbc0: 7469 7469 6f6e 6564 2064 6174 6120 2869  titioned data (i
-0000dbd0: 2e65 2e2c 2031 7374 2c20 326e 6420 616e  .e., 1st, 2nd an
-0000dbe0: 6420 3372 6420 706f 7369 7469 6f6e 206f  d 3rd position o
-0000dbf0: 6620 636f 646f 6e29 2069 6e73 7465 6164  f codon) instead
-0000dc00: 206f 6620 7573 696e 6720 7468 6520 636f   of using the co
-0000dc10: 646f 6e20 6173 2061 2077 686f 6c65 2e20  don as a whole. 
-0000dc20: 5468 6520 6f70 7469 6f6e 2060 2d2d 6161  The option `--aa
-0000dc30: 6d6f 6465 6c60 2063 616e 2062 6520 7365  model` can be se
-0000dc40: 7420 746f 2064 6574 6572 6d69 6e65 2074  t to determine t
-0000dc50: 6865 2061 6d69 6e6f 2061 6369 6420 6d6f  he amino acid mo
-0000dc60: 6465 6c20 6170 706c 6965 6420 696e 2060  del applied in `
-0000dc70: 6d63 6d63 7472 6565 6020 616e 616c 7973  mcmctree` analys
-0000dc80: 6973 2e20 5468 6520 6f70 7469 6f6e 2060  is. The option `
-0000dc90: 2d64 7360 2063 616e 2062 6520 7573 6564  -ds` can be used
-0000dca0: 2074 6f20 7365 7420 7061 7261 6d65 7465   to set paramete
-0000dcb0: 7273 2066 6f72 2074 6865 206d 6f6c 6563  rs for the molec
-0000dcc0: 756c 6172 2064 6174 696e 6720 7072 6f67  ular dating prog
-0000dcd0: 7261 6d2e 2054 6865 2070 6172 616c 6c65  ram. The paralle
-0000dce0: 6c20 7468 7265 6164 7320 6865 7265 2061  l threads here a
-0000dcf0: 7265 2074 6f20 7061 7261 6c6c 656c 697a  re to paralleliz
-0000dd00: 6520 7468 6520 616e 616c 7973 6973 2066  e the analysis f
-0000dd10: 6f72 2065 6163 6820 6765 6e65 2066 616d  or each gene fam
-0000dd20: 696c 7920 616e 6420 7468 7573 2073 7567  ily and thus sug
-0000dd30: 6765 7374 6564 2074 6f20 6265 2073 6574  gested to be set
-0000dd40: 2061 7320 6d75 6368 2061 7320 7468 6520   as much as the 
-0000dd50: 6e75 6d62 6572 206f 6620 6765 6e65 2066  number of gene f
-0000dd60: 616d 696c 6965 732e 0a0a 2a2a 4120 7375  amilies...**A su
-0000dd70: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-0000dd80: 2072 756e 2063 616e 2062 6520 7769 7468   run can be with
-0000dd90: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-0000dda0: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2066  ow.**..```.wgd f
-0000ddb0: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
-0000ddc0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000ddd0: 3220 7365 7175 656e 6365 3320 2d64 206d  2 sequence3 -d m
-0000dde0: 636d 6374 7265 6520 2d73 7020 7370 6461  cmctree -sp spda
-0000ddf0: 7461 0a77 6764 2066 6f63 7573 2066 616d  ta.wgd focus fam
-0000de00: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
-0000de10: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000de20: 6365 3320 2d64 2072 3873 202d 7370 2073  ce3 -d r8s -sp s
-0000de30: 7064 6174 6120 2d2d 6e73 6974 6573 206e  pdata --nsites n
-0000de40: 7369 7465 696e 666f 0a77 6764 2066 6f63  siteinfo.wgd foc
-0000de50: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000de60: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000de70: 7365 7175 656e 6365 3320 2d64 2062 6561  sequence3 -d bea
-0000de80: 7374 202d 2d66 6f73 7369 6c20 666f 7373  st --fossil foss
-0000de90: 696c 696e 666f 202d 2d72 6f6f 7468 6569  ilinfo --roothei
-0000dea0: 6768 7420 726f 6f74 6865 6967 6874 696e  ght rootheightin
-0000deb0: 666f 202d 2d63 6861 696e 7365 7420 6368  fo --chainset ch
-0000dec0: 6169 6e73 6574 696e 666f 202d 2d62 6561  ainsetinfo --bea
-0000ded0: 7374 6c67 6a61 7220 2450 4154 480a 6060  stlgjar $PATH.``
-0000dee0: 600a 0a23 2323 2077 6764 206b 7364 0a0a  `..### wgd ksd..
-0000def0: 2a2a 5468 6520 636f 6e73 7472 7563 7469  **The constructi
-0000df00: 6f6e 206f 6620 7768 6f6c 6520 7061 7261  on of whole para
-0000df10: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-0000df20: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-0000df30: 7574 696f 6e2a 2a0a 6060 600a 7767 6420  ution**.```.wgd 
-0000df40: 6b73 6420 6661 6d69 6c69 6573 2073 6571  ksd families seq
-0000df50: 7565 6e63 6520 282d 6f20 7767 645f 6b73  uence (-o wgd_ks
-0000df60: 6420 2d74 2077 6764 5f6b 7364 5f74 6d70  d -t wgd_ksd_tmp
-0000df70: 202d 2d6e 7468 7265 6164 7320 3420 2d2d   --nthreads 4 --
-0000df80: 746f 5f73 746f 7020 2d2d 6364 7320 2d2d  to_stop --cds --
-0000df90: 7061 6972 7769 7365 202d 2d73 7472 6970  pairwise --strip
-0000dfa0: 5f67 6170 7320 2d2d 616c 6967 6e65 7220  _gaps --aligner 
-0000dfb0: 6d61 6666 7420 2d2d 7472 6565 5f6d 6574  mafft --tree_met
-0000dfc0: 686f 6420 6661 7374 7472 6565 202d 2d6e  hod fasttree --n
-0000dfd0: 6f64 655f 6176 6572 6167 6529 0a60 6060  ode_average).```
-0000dfe0: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
-0000dff0: 6764 206b 7364 602c 2061 7320 696d 7069  gd ksd`, as impi
-0000e000: 6564 2062 7920 6974 7320 6e61 6d65 2c20  ed by its name, 
-0000e010: 6973 2066 6f72 2074 6865 2063 6f6e 7374  is for the const
-0000e020: 7275 6374 696f 6e20 6f66 202a 4b2a 3c73  ruction of *K*<s
-0000e030: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000e040: 6973 7472 6962 7574 696f 6e2e 2045 7863  istribution. Exc
-0000e050: 6570 7420 666f 7220 7468 6520 6166 6f72  ept for the afor
-0000e060: 656d 656e 7469 6f6e 6564 2070 6172 616d  ementioned param
-0000e070: 6574 6572 7320 7375 6368 2061 7320 602d  eters such as `-
-0000e080: 2d74 6f5f 7374 6f70 6020 616e 6420 602d  -to_stop` and `-
-0000e090: 2d63 6473 602c 2074 6865 7265 2061 7265  -cds`, there are
-0000e0a0: 2073 6f6d 6520 696d 706f 7274 616e 7420   some important 
-0000e0b0: 7061 7261 6d65 7465 7273 2074 6861 7420  parameters that 
-0000e0c0: 6861 7665 2063 7275 6369 616c 2069 6d70  have crucial imp
-0000e0d0: 6163 7420 6f6e 2074 6865 202a 4b2a 3c73  act on the *K*<s
-0000e0e0: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
-0000e0f0: 6174 696f 6e2e 2054 6865 206f 7074 696f  ation. The optio
-0000e100: 6e20 602d 2d70 6169 7277 6973 6560 2069  n `--pairwise` i
-0000e110: 7320 6120 7665 7279 2069 6d70 6f72 7461  s a very importa
-0000e120: 6e74 2070 6172 616d 6574 6572 2066 6f72  nt parameter for
-0000e130: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-0000e140: 7375 623e 2065 7374 696d 6174 696f 6e2c  sub> estimation,
-0000e150: 2077 6974 6820 7768 6963 6820 7468 6520   with which the 
-0000e160: 6043 4f44 454d 4c60 2077 696c 6c20 6361  `CODEML` will ca
-0000e170: 6c63 756c 6174 6520 7468 6520 2a4b 2a3c  lculate the *K*<
-0000e180: 7375 623e 533c 2f73 7562 3e20 666f 7220  sub>S</sub> for 
-0000e190: 6561 6368 2067 656e 6520 7061 6972 2073  each gene pair s
-0000e1a0: 6570 6172 6174 656c 7920 6261 7365 6420  eparately based 
-0000e1b0: 6f6e 2074 6865 2061 6c69 676e 6d65 6e74  on the alignment
-0000e1c0: 206f 6620 6f6e 6c79 2074 6865 7365 2074   of only these t
-0000e1d0: 776f 2067 656e 6573 2069 6e73 7465 6164  wo genes instead
-0000e1e0: 206f 6620 7468 6520 7768 6f6c 6520 616c   of the whole al
-0000e1f0: 6967 6e6d 656e 7420 6f66 2074 6865 2066  ignment of the f
-0000e200: 616d 696c 792c 2073 7563 6820 7468 6174  amily, such that
-0000e210: 206c 6573 7320 6761 7073 2061 7265 2065   less gaps are e
-0000e220: 7870 6563 7465 6420 616e 6420 7468 7573  xpected and thus
-0000e230: 2074 6865 2061 6c69 676e 6d65 6e74 2069   the alignment i
-0000e240: 6e20 7468 6520 636f 6e73 6964 6572 6174  n the considerat
-0000e250: 696f 6e20 6f66 2060 434f 4445 4d4c 6020  ion of `CODEML` 
-0000e260: 7769 6c6c 2062 6520 6c6f 6e67 6572 2028  will be longer (
-0000e270: 6265 6361 7573 6520 6043 4f44 454d 4c60  because `CODEML`
-0000e280: 2077 696c 6c20 6175 746f 6d61 7465 6c79   will automately
-0000e290: 2073 6b69 7020 6576 6572 7920 636f 6c75   skip every colu
-0000e2a0: 6d6e 2077 6974 6820 6761 702c 2072 6567  mn with gap, reg
-0000e2b0: 6172 646c 6573 7320 6f66 2077 6865 7468  ardless of wheth
-0000e2c0: 6572 2069 7427 7320 616e 206f 7665 7261  er it's an overa
-0000e2d0: 6c6c 206f 7220 7061 7274 6961 6c20 6761  ll or partial ga
-0000e2e0: 7029 2c20 7769 7468 6f75 7420 7768 6963  p), without whic
-0000e2f0: 6820 7468 6520 6043 4f44 454d 4c60 2077  h the `CODEML` w
-0000e300: 696c 6c20 6361 6c63 756c 6174 6520 7468  ill calculate th
-0000e310: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-0000e320: 3e20 6261 7365 6420 6f6e 2074 6865 2077  > based on the w
-0000e330: 686f 6c65 2061 6c69 676e 6d65 6e74 206f  hole alignment o
-0000e340: 6620 7468 6520 6661 6d69 6c79 2c20 7768  f the family, wh
-0000e350: 6963 6820 6d69 6768 7420 6861 7665 206e  ich might have n
-0000e360: 6f20 2a4b 2a3c 7375 623e 533c 2f73 7562  o *K*<sub>S</sub
-0000e370: 3e20 7265 7375 6c74 2061 7420 616c 6c20  > result at all 
-0000e380: 6966 2074 6865 2073 7472 6970 7065 6420  if the stripped 
-0000e390: 616c 6967 6e6d 656e 7420 6c65 6e67 7468  alignment length
-0000e3a0: 2028 7265 6d6f 7669 6e67 2061 6c6c 2067   (removing all g
-0000e3b0: 6170 2d63 6f6e 7461 696e 696e 6720 636f  ap-containing co
-0000e3c0: 6c75 6d6e 7329 2077 6173 207a 6572 6f2c  lumns) was zero,
-0000e3d0: 2061 2063 6175 7365 206f 6620 6469 6666   a cause of diff
-0000e3e0: 6572 656e 7420 6e75 6d62 6572 206f 6620  erent number of 
-0000e3f0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e400: 6573 7469 6d61 7465 7320 6265 7477 6565  estimates betwee
-0000e410: 6e20 2270 6169 7277 6973 6520 6d6f 6465  n "pairwise mode
-0000e420: 2220 616e 6420 226e 6f6e 2d70 6169 7277  " and "non-pairw
-0000e430: 6973 6520 6d6f 6465 222e 2049 7427 7320  ise mode". It's 
-0000e440: 6469 6666 6963 756c 7420 746f 2073 6179  difficult to say
-0000e450: 2077 6869 6368 206d 6f64 6520 6973 206d   which mode is m
-0000e460: 6f72 6520 6964 6561 6c2c 2061 6c74 686f  ore ideal, altho
-0000e470: 7567 6820 7468 6520 226e 6f6e 2d70 6169  ugh the "non-pai
-0000e480: 7277 6973 6520 6d6f 6465 2220 2864 6566  rwise mode" (def
-0000e490: 6175 6c74 2073 6574 7469 6e67 2920 7768  ault setting) wh
-0000e4a0: 6963 6820 7275 6e73 206f 6e20 7468 6520  ich runs on the 
-0000e4b0: 7768 6f6c 6520 616c 6967 6e6d 656e 7420  whole alignment 
-0000e4c0: 696e 7374 6561 6420 6f66 2061 206c 6f63  instead of a loc
-0000e4d0: 616c 2061 6c69 676e 6d65 6e74 2c20 6d69  al alignment, mi
-0000e4e0: 6768 7420 6265 206d 6f72 6520 6269 6f6c  ght be more biol
-0000e4f0: 6f67 6963 616c 6c79 2063 6f6e 7365 7276  ogically conserv
-0000e500: 6564 2069 6e20 7468 6174 2069 7420 6173  ed in that it as
-0000e510: 7375 7265 7320 7468 6520 6576 6f6c 7574  sures the evolut
-0000e520: 696f 6e20 6f66 2065 6163 6820 636f 6c75  ion of each colu
-0000e530: 6d6e 2074 6f20 6265 2073 7461 7274 6564  mn to be started
-0000e540: 2066 726f 6d20 7468 6520 726f 6f74 206f   from the root o
-0000e550: 6620 7468 6520 6661 6d69 6c79 2061 6e64  f the family and
-0000e560: 2061 6c6c 2074 6865 2067 656e 6520 6475   all the gene du
-0000e570: 706c 6963 6174 6573 2061 7265 2074 616b  plicates are tak
-0000e580: 656e 2069 6e74 6f20 6163 636f 756e 7420  en into account 
-0000e590: 696e 2074 6865 202a 4b2a 3c73 7562 3e53  in the *K*<sub>S
-0000e5a0: 3c2f 7375 623e 2065 7374 696d 6174 696f  </sub> estimatio
-0000e5b0: 6e20 7072 6f63 6573 732e 2054 6865 206f  n process. The o
-0000e5c0: 7074 696f 6e20 602d 2d73 7472 6970 5f67  ption `--strip_g
-0000e5d0: 6170 7360 2063 616e 2072 656d 6f76 6520  aps` can remove 
-0000e5e0: 616c 6c20 7468 6520 6761 702d 636f 6e74  all the gap-cont
-0000e5f0: 6169 6e69 6e67 2063 6f6c 756d 6e73 2c20  aining columns, 
-0000e600: 7769 7468 206f 7220 7769 7468 6f75 7420  with or without 
-0000e610: 7768 6963 6820 7468 6520 7265 7375 6c74  which the result
-0000e620: 206f 6620 226e 6f6e 2d70 6169 7277 6973   of "non-pairwis
-0000e630: 6520 6d6f 6465 2220 776f 6e27 7420 6265  e mode" won't be
-0000e640: 2061 6666 6563 7465 642c 2077 6869 6c65   affected, while
-0000e650: 2077 6974 6820 7768 6963 6820 7468 6520   with which the 
-0000e660: 7265 7375 6c74 206f 6620 2270 6169 7277  result of "pairw
-0000e670: 6973 6520 6d6f 6465 2220 7769 6c6c 2062  ise mode" will b
-0000e680: 6520 616c 7465 7265 642e 2054 6865 206f  e altered. The o
-0000e690: 7074 696f 6e20 602d 2d61 6c69 676e 6572  ption `--aligner
-0000e6a0: 6020 616e 6420 602d 2d61 6c6e 5f6f 7074  ` and `--aln_opt
-0000e6b0: 696f 6e73 6020 7768 6963 6820 6465 6369  ions` which deci
-0000e6c0: 6465 2077 6869 6368 2061 6c69 676e 6d65  de which alignme
-0000e6d0: 6e74 2070 726f 6772 616d 2074 6f20 6265  nt program to be
-0000e6e0: 2075 7365 6420 616e 6420 7768 6963 6820   used and which 
-0000e6f0: 7061 7261 6d65 7465 7220 746f 2062 6520  parameter to be 
-0000e700: 7365 742c 2077 696c 6c20 6861 7665 2069  set, will have i
-0000e710: 6d70 6163 7420 6f6e 2074 6865 202a 4b2a  mpact on the *K*
-0000e720: 3c73 7562 3e53 3c2f 7375 623e 2072 6573  <sub>S</sub> res
-0000e730: 756c 7473 2c20 6e6f 7465 6420 7468 6174  ults, noted that
-0000e740: 2074 6865 2064 6566 6175 6c74 2070 726f   the default pro
-0000e750: 6772 616d 2069 7320 606d 6166 6674 6020  gram is `mafft` 
-0000e760: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
-0000e770: 7220 6973 2060 2d2d 6175 746f 602e 2054  r is `--auto`. T
-0000e780: 6865 206f 7074 696f 6e20 602d 2d74 7265  he option `--tre
-0000e790: 655f 6d65 7468 6f64 6020 616e 6420 602d  e_method` and `-
-0000e7a0: 2d74 7265 655f 6f70 7469 6f6e 7360 2064  -tree_options` d
-0000e7b0: 6563 6964 6520 7768 6963 6820 6765 6e65  ecide which gene
-0000e7c0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-0000e7d0: 7072 6f67 7261 6d20 746f 2062 6520 7573  program to be us
-0000e7e0: 6564 2061 6e64 2077 6869 6368 2070 6172  ed and which par
-0000e7f0: 616d 6574 6572 2074 6f20 6265 2073 6574  ameter to be set
-0000e800: 2c20 776f 6e27 7420 6166 6665 6374 2074  , won't affect t
-0000e810: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-0000e820: 623e 2065 7374 696d 6174 696f 6e20 6974  b> estimation it
-0000e830: 7365 6c66 2062 7574 2074 6865 2072 6573  self but the res
-0000e840: 756c 7420 6f66 2064 652d 7265 6475 6e64  ult of de-redund
-0000e850: 616e 6379 2c20 6e6f 7465 6420 7468 6174  ancy, noted that
-0000e860: 2077 6520 696d 706c 656d 656e 7465 6420   we implemented 
-0000e870: 6120 6275 696c 742d 696e 2067 656e 6520  a built-in gene 
-0000e880: 7472 6565 2069 6e66 6572 656e 6365 206d  tree inference m
-0000e890: 6574 686f 6420 6261 7365 6420 6f6e 2074  ethod based on t
-0000e8a0: 6865 2041 7665 7261 6765 204c 696e 6b61  he Average Linka
-0000e8b0: 6765 2043 6c75 7374 6572 696e 6720 2841  ge Clustering (A
-0000e8c0: 4c43 2920 2874 6875 7320 6120 6469 7374  LC) (thus a dist
-0000e8d0: 616e 6365 2d62 6173 6564 2074 7265 6529  ance-based tree)
-0000e8e0: 2077 6974 6820 7468 6520 602d 2d74 7265   with the `--tre
-0000e8f0: 655f 6d65 7468 6f64 6020 7365 7420 6173  e_method` set as
-0000e900: 2022 636c 7573 7465 7222 2e20 5477 6f20   "cluster". Two 
-0000e910: 6d65 7468 6f64 7320 6f66 2064 652d 7265  methods of de-re
-0000e920: 6475 6e64 616e 6379 2061 7265 2069 6d70  dundancy are imp
-0000e930: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
-0000e940: 2076 3260 2c20 6e61 6d65 6c79 206e 6f64   v2`, namely nod
-0000e950: 652d 7765 6967 6874 6564 2061 6e64 206e  e-weighted and n
-0000e960: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
-0000e970: 686f 6473 2e20 5468 6520 6e6f 6465 2d77  hods. The node-w
-0000e980: 6569 6768 7465 6420 6d65 7468 6f64 2061  eighted method a
-0000e990: 6368 6965 7665 7320 7468 6520 6465 2d72  chieves the de-r
-0000e9a0: 6564 756e 6461 6e63 7920 7669 6120 7765  edundancy via we
-0000e9b0: 6967 6869 6e67 2074 6865 202a 4b2a 3c73  ighing the *K*<s
-0000e9c0: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
-0000e9d0: 2061 7373 6f63 6961 7465 6420 7769 7468   associated with
-0000e9e0: 2065 6163 6820 6765 6e65 2070 6169 7220   each gene pair 
-0000e9f0: 7375 6368 2074 6861 7420 7468 6520 7765  such that the we
-0000ea00: 6967 6874 206f 6620 6120 7369 6e67 6c65  ight of a single
-0000ea10: 2064 7570 6c69 6361 7469 6f6e 2065 7665   duplication eve
-0000ea20: 6e74 2073 756d 7320 7570 2074 6f20 3120  nt sums up to 1 
-0000ea30: 286e 6f74 6564 2074 6861 7420 7468 6520  (noted that the 
-0000ea40: 6e75 6d62 6572 206f 6620 2a4b 2a3c 7375  number of *K*<su
-0000ea50: 623e 533c 2f73 7562 3e20 6573 7469 6d61  b>S</sub> estima
-0000ea60: 7469 6f6e 2072 656d 6169 6e73 2074 6865  tion remains the
-0000ea70: 2073 616d 6529 2077 6869 6c65 2074 6865   same) while the
-0000ea80: 206e 6f64 652d 6176 6572 6167 6564 206d   node-averaged m
-0000ea90: 6574 686f 6420 7265 616c 697a 6573 2074  ethod realizes t
-0000eaa0: 6865 2064 652d 7265 6475 6e64 616e 6379  he de-redundancy
-0000eab0: 2076 6961 2063 616c 6375 6c61 7469 6e67   via calculating
-0000eac0: 2070 6572 2067 656e 6520 7472 6565 206e   per gene tree n
-0000ead0: 6f64 6520 6f6e 6520 6176 6572 6167 6564  ode one averaged
-0000eae0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000eaf0: 2076 616c 7565 2074 6f20 7265 7072 6573   value to repres
-0000eb00: 656e 7420 7468 6520 6167 6520 6f66 2065  ent the age of e
-0000eb10: 6163 6820 6475 706c 6963 6174 696f 6e20  ach duplication 
-0000eb20: 6576 656e 742e 2054 6865 206f 7074 696f  event. The optio
-0000eb30: 6e20 602d 2d6e 6f64 655f 6176 6572 6167  n `--node_averag
-0000eb40: 6560 2063 616e 2062 6520 7365 7420 746f  e` can be set to
-0000eb50: 2063 686f 6f73 6520 7468 6520 6e6f 6465   choose the node
-0000eb60: 2d61 7665 7261 6765 6420 7761 7920 6f66  -averaged way of
-0000eb70: 2064 652d 7265 6475 6e64 616e 6379 2e20   de-redundancy. 
-0000eb80: 4469 6666 6572 656e 7420 6d65 7468 6f64  Different method
-0000eb90: 7320 6f66 2064 652d 7265 6475 6e64 616e  s of de-redundan
-0000eba0: 6379 2077 696c 6c20 6861 7665 2069 6d70  cy will have imp
-0000ebb0: 6163 7420 6f6e 2074 6865 2064 6574 6563  act on the detec
-0000ebc0: 7469 6f6e 206f 6620 5747 4420 7369 676e  tion of WGD sign
-0000ebd0: 616c 732c 2077 6869 6368 2068 6173 2062  als, which has b
-0000ebe0: 6565 6e20 696e 7665 7374 6967 6174 6564  een investigated
-0000ebf0: 2069 6e20 7468 6973 205b 6c69 7465 7261   in this [litera
-0000ec00: 7475 7265 5d28 6874 7470 733a 2f2f 646f  ture](https://do
-0000ec10: 692e 6f72 672f 3130 2e31 3039 332f 6762  i.org/10.1093/gb
-0000ec20: 652f 6576 7932 3030 292e 2054 6865 2070  e/evy200). The p
-0000ec30: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
-0000ec40: 6865 7265 2061 7265 2074 6f20 7061 7261  here are to para
-0000ec50: 6c6c 656c 697a 6520 7468 6520 616e 616c  llelize the anal
-0000ec60: 7973 6973 2066 6f72 2065 6163 6820 6765  ysis for each ge
-0000ec70: 6e65 2066 616d 696c 7920 616e 6420 7468  ne family and th
-0000ec80: 7573 2073 7567 6765 7374 6564 2074 6f20  us suggested to 
-0000ec90: 6265 2073 6574 2061 7320 6d75 6368 2061  be set as much a
-0000eca0: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
-0000ecb0: 6765 6e65 2066 616d 696c 6965 732e 0a0a  gene families...
-0000ecc0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-0000ecd0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-0000ece0: 7365 2063 6f6d 6d61 6e64 2061 7320 6265  se command as be
-0000ecf0: 6c6f 772a 2a0a 0a60 6060 0a77 6764 206b  low**..```.wgd k
-0000ed00: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
-0000ed10: 656e 6365 0a60 6060 0a0a 2a2a 5468 6520  ence.```..**The 
-0000ed20: 636f 6e73 7472 7563 7469 6f6e 206f 6620  construction of 
-0000ed30: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-0000ed40: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-0000ed50: 6469 7374 7269 6275 7469 6f6e 2a2a 0a60  distribution**.`
-0000ed60: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
-0000ed70: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000ed80: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000ed90: 3320 282d 2d72 6577 6569 6768 7429 0a60  3 (--reweight).`
-0000eda0: 6060 0a0a 4672 6f6d 2070 6172 616c 6f67  ``..From paralog
-0000edb0: 6f75 7320 746f 206f 7274 686f 6c6f 676f  ous to orthologo
-0000edc0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-0000edd0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000ede0: 696f 6e2c 2075 7365 7273 206f 6e6c 7920  ion, users only 
-0000edf0: 6e65 6564 2074 6f20 7072 6f76 6964 6520  need to provide 
-0000ee00: 6d6f 7265 2063 6473 2066 696c 6573 2e20  more cds files. 
-0000ee10: 4e6f 7465 2074 6861 7420 7769 7468 206f  Note that with o
-0000ee20: 7274 686f 6c6f 676f 7573 2067 656e 6520  rthologous gene 
-0000ee30: 6661 6d69 6c69 6573 2074 6865 2077 6569  families the wei
-0000ee40: 6768 7469 6e67 206d 6574 686f 6420 6361  ghting method ca
-0000ee50: 6e20 6265 2073 6574 2074 6f20 6265 2063  n be set to be c
-0000ee60: 616c 6375 6c61 7465 6420 7065 7220 7370  alculated per sp
-0000ee70: 6563 6965 7320 7061 6972 2069 6e73 7465  ecies pair inste
-0000ee80: 6164 206f 6620 636f 6e73 6964 6572 696e  ad of considerin
-0000ee90: 6720 7468 6520 7768 6f6c 6520 6661 6d69  g the whole fami
-0000eea0: 6c79 2062 6563 6175 7365 2077 6865 6e20  ly because when 
-0000eeb0: 706c 6f74 7469 6e67 206f 7274 686f 6c6f  plotting ortholo
-0000eec0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
-0000eed0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-0000eee0: 7574 696f 6e20 6265 7477 6565 6e20 7477  ution between tw
-0000eef0: 6f20 7370 6563 6965 7320 7468 6520 7765  o species the we
-0000ef00: 6967 6874 2063 616c 6375 6c61 7465 6420  ight calculated 
-0000ef10: 6672 6f6d 2074 6869 7320 7370 6563 6966  from this specif
-0000ef20: 6963 2073 7065 6369 6573 2070 6169 7220  ic species pair 
-0000ef30: 7368 6f75 6c64 2062 6520 636f 6e73 6572  should be conser
-0000ef40: 7665 6420 7768 696c 6520 7468 6520 6f6e  ved while the on
-0000ef50: 6520 6361 6c63 756c 6174 6564 2066 726f  e calculated fro
-0000ef60: 6d20 7468 6520 7768 6f6c 6520 6661 6d69  m the whole fami
-0000ef70: 6c79 2077 696c 6c20 7661 7279 2077 6974  ly will vary wit
-0000ef80: 6820 7468 6520 6e75 6d62 6572 206f 6620  h the number of 
-0000ef90: 7370 6563 6965 732e 2054 6f20 696e 6974  species. To init
-0000efa0: 6961 7465 2074 6865 2077 6569 6768 7469  iate the weighti
-0000efb0: 6e67 2070 6572 2073 7065 6369 6573 2070  ng per species p
-0000efc0: 6169 722c 2074 6865 206f 7074 696f 6e20  air, the option 
-0000efd0: 602d 2d72 6577 6569 6768 7460 2063 616e  `--reweight` can
-0000efe0: 2062 6520 7365 742e 0a0a 2a2a 4120 7375   be set...**A su
-0000eff0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-0000f000: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
-0000f010: 6d61 6e64 2061 7320 6265 6c6f 772a 2a0a  mand as below**.
-0000f020: 0a60 6060 0a77 6764 206b 7364 2066 616d  .```.wgd ksd fam
-0000f030: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
-0000f040: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000f050: 6365 330a 6060 600a 0a2a 2a54 6865 2063  ce3.```..**The c
-0000f060: 6f6e 7374 7275 6374 696f 6e20 6f66 202a  onstruction of *
-0000f070: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-0000f080: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-0000f090: 7769 7468 2072 6174 6520 636f 7272 6563  with rate correc
-0000f0a0: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206b  tion**.```.wgd k
-0000f0b0: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
-0000f0c0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000f0d0: 7365 7175 656e 6365 3320 2d2d 666f 6375  sequence3 --focu
-0000f0e0: 7332 616c 6c20 7365 7175 656e 6365 3120  s2all sequence1 
-0000f0f0: 2d73 7020 7370 6461 7461 202d 2d65 7874  -sp spdata --ext
-0000f100: 7261 7061 7261 6e6f 6d65 6b73 2070 6172  raparanomeks par
-0000f110: 616e 6f6d 654b 7364 6174 6120 282d 2d70  anomeKsdata (--p
-0000f120: 6c6f 7465 6c6d 6d20 2d2d 706c 6f74 6170  lotelmm --plotap
-0000f130: 676d 6d20 2d2d 616e 6368 6f72 706f 696e  gmm --anchorpoin
-0000f140: 7473 2061 7064 6174 6120 2d2d 7265 7765  ts apdata --rewe
-0000f150: 6967 6874 202d 2d6f 6e6c 7972 6f6f 746f  ight --onlyrooto
-0000f160: 7574 290a 6060 600a 0a49 6e73 7069 7265  ut).```..Inspire
-0000f170: 6420 6279 2074 6865 2072 6174 6520 636f  d by the rate co
-0000f180: 7272 6563 7469 6f6e 2061 6c67 6f72 6974  rrection algorit
-0000f190: 686d 2069 6e20 5b6b 7372 6174 6573 5d28  hm in [ksrates](
-0000f1a0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-0000f1b0: 6f6d 2f56 4942 2d50 5342 2f6b 7372 6174  om/VIB-PSB/ksrat
-0000f1c0: 6573 292c 2077 6520 696d 706c 656d 656e  es), we implemen
-0000f1d0: 7465 6420 7468 6520 7261 7465 2063 6f72  ted the rate cor
-0000f1e0: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
-0000f1f0: 2061 6c73 6f20 696e 2060 7767 6420 7632   also in `wgd v2
-0000f200: 602c 2077 6869 6368 2069 7320 6d6f 7374  `, which is most
-0000f210: 6c79 2074 6865 2073 616d 6520 6173 205b  ly the same as [
-0000f220: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
-0000f230: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
-0000f240: 5053 422f 6b73 7261 7465 7329 2062 7574  PSB/ksrates) but
-0000f250: 2064 6966 6665 7273 2069 6e20 7468 6520   differs in the 
-0000f260: 6361 6c63 756c 6174 696f 6e20 6f66 2074  calculation of t
-0000f270: 6865 2073 7461 6e64 6172 6420 6465 7669  he standard devi
-0000f280: 6174 696f 6e20 6f66 2072 6573 6361 6c65  ation of rescale
-0000f290: 6420 2a4b 2a3c 7375 623e 533c 2f73 7562  d *K*<sub>S</sub
-0000f2a0: 3e20 6167 6573 2e20 546f 2070 6572 666f  > ages. To perfo
-0000f2b0: 726d 2074 6865 2072 6174 6520 636f 7272  rm the rate corr
-0000f2c0: 6563 7469 6f6e 2061 6e61 6c79 7369 732c  ection analysis,
-0000f2d0: 2075 7365 7273 2063 616e 2075 7365 2062   users can use b
-0000f2e0: 6f74 6820 7468 6520 6077 6764 206b 7364  oth the `wgd ksd
-0000f2f0: 6020 616e 6420 6077 6764 2076 697a 6020  ` and `wgd viz` 
-0000f300: 7072 6f67 7261 6d2e 2046 6f72 2074 6865  program. For the
-0000f310: 2060 7767 6420 6b73 6460 2070 726f 6772   `wgd ksd` progr
-0000f320: 616d 2c20 7573 6572 7320 6e65 6564 2074  am, users need t
-0000f330: 6f20 7072 6f76 6964 6520 6120 7370 6563  o provide a spec
-0000f340: 6965 7320 7472 6565 2076 6961 2074 6865  ies tree via the
-0000f350: 206f 7074 696f 6e20 602d 2d73 7065 6369   option `--speci
-0000f360: 6573 7472 6565 6020 6f6e 2077 6869 6368  estree` on which
-0000f370: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
-0000f380: 2063 616e 2062 6520 636f 6e64 7563 7465   can be conducte
-0000f390: 642e 204e 6f74 6520 7468 6174 2075 6e6e  d. Note that unn
-0000f3a0: 6563 6573 7361 7279 2062 7261 636b 6574  ecessary bracket
-0000f3b0: 7320 6d69 6768 7420 6c65 6164 2074 6f20  s might lead to 
-0000f3c0: 756e 6578 7065 6374 6564 2065 7272 6f72  unexpected error
-0000f3d0: 732c 2066 6f72 2069 6e73 7461 6e63 6520  s, for instance 
-0000f3e0: 6120 7472 6565 2060 2841 2c28 422c 4329  a tree `(A,(B,C)
-0000f3f0: 293b 6020 7368 6f75 6c64 206e 6f74 2062  );` should not b
-0000f400: 6520 7265 7072 6573 656e 7465 6420 6173  e represented as
-0000f410: 2060 2841 2c28 2842 2c43 2929 293b 602e   `(A,((B,C)));`.
-0000f420: 2054 6865 2073 6574 206f 6620 7370 6563   The set of spec
-0000f430: 6965 7320 7061 6972 7320 746f 2062 6520  ies pairs to be 
-0000f440: 7368 6f77 6e20 6973 2066 6c65 7869 626c  shown is flexibl
-0000f450: 6520 7468 6174 2074 6865 206d 6f73 7420  e that the most 
-0000f460: 636f 6e76 656e 6965 6e74 206f 7074 696f  convenient optio
-0000f470: 6e20 6973 2060 2d2d 666f 6375 7332 616c  n is `--focus2al
-0000f480: 6c60 2077 6869 6368 2073 696d 706c 7920  l` which simply 
-0000f490: 7368 6f77 7320 616c 6c20 7468 6520 706f  shows all the po
-0000f4a0: 7373 6962 6c65 2066 6f63 616c 2d73 6973  ssible focal-sis
-0000f4b0: 7465 7220 7370 6563 6965 7320 7061 6972  ter species pair
-0000f4c0: 732c 206f 7220 7573 6572 7320 6361 6e20  s, or users can 
-0000f4d0: 6d61 6e75 616c 6c79 2073 6574 2074 6865  manually set the
-0000f4e0: 2073 7065 6369 6573 2070 6169 7273 2076   species pairs v
-0000f4f0: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-0000f500: 2d73 7061 6972 602e 204e 6f74 6520 7468  -spair`. Note th
-0000f510: 6174 2069 6620 7468 6520 7370 6563 6965  at if the specie
-0000f520: 7320 7061 6972 7320 7765 7265 206d 616e  s pairs were man
-0000f530: 7561 6c6c 7920 7365 742c 2069 7420 776f  ually set, it wo
-0000f540: 756c 6420 6265 206e 6565 6465 6420 746f  uld be needed to
-0000f550: 2063 6f2d 7365 7420 7468 6520 6f70 7469   co-set the opti
-0000f560: 6f6e 2060 2d2d 636c 6173 7369 6360 2e20  on `--classic`. 
-0000f570: 5468 6520 6f70 7469 6f6e 2060 2d2d 6f6e  The option `--on
-0000f580: 6c79 726f 6f74 6f75 7460 2063 616e 2062  lyrootout` can b
-0000f590: 6520 7365 7420 746f 206f 6e6c 7920 636f  e set to only co
-0000f5a0: 6e73 6964 6572 206f 7574 6772 6f75 7020  nsider outgroup 
-0000f5b0: 6174 2074 6865 2072 6f6f 742c 2069 6e73  at the root, ins
-0000f5c0: 7465 6164 206f 6620 616c 6c20 7468 6520  tead of all the 
-0000f5d0: 706f 7373 6962 6c65 206f 7574 6772 6f75  possible outgrou
-0000f5e0: 7073 2070 6572 2066 6f63 616c 2d73 6973  ps per focal-sis
-0000f5f0: 7465 7220 7370 6563 6965 7320 7061 6972  ter species pair
-0000f600: 2c20 7768 6963 6820 6861 7320 696d 7061  , which has impa
-0000f610: 6374 206f 6e20 7468 6520 6669 6e61 6c20  ct on the final 
-0000f620: 636f 7272 6563 7465 6420 2a4b 2a3c 7375  corrected *K*<su
-0000f630: 623e 533c 2f73 7562 3e20 6167 6573 2e20  b>S</sub> ages. 
-0000f640: 5765 2073 7567 6765 7374 206f 6620 7573  We suggest of us
-0000f650: 696e 6720 616c 6c20 7468 6520 706f 7373  ing all the poss
-0000f660: 6962 6c65 206f 7574 6772 6f75 7073 2070  ible outgroups p
-0000f670: 6572 2066 6f63 616c 2d73 6973 7465 7220  er focal-sister 
-0000f680: 7370 6563 6965 7320 7061 6972 2061 7320  species pair as 
-0000f690: 666f 7220 6120 6c65 7373 2062 6961 7365  for a less biase
-0000f6a0: 6420 7265 7375 6c74 2e20 5468 6520 7061  d result. The pa
-0000f6b0: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
-0000f6c0: 3c2f 7375 623e 2064 6174 6120 7368 6f75  </sub> data shou
-0000f6d0: 6c64 2062 6520 7072 6f76 6964 6564 2076  ld be provided v
-0000f6e0: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-0000f6f0: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-0000f700: 602e 200a 0a53 6f6d 6520 6f74 6865 7220  `. ..Some other 
-0000f710: 6f70 7469 6f6e 7320 7768 6963 6820 6861  options which ha
-0000f720: 7665 206e 6f20 696d 7061 6374 206f 6e20  ve no impact on 
-0000f730: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
-0000f740: 696f 6e20 6275 7420 6164 6420 6d6f 7265  ion but add more
-0000f750: 206c 6179 6572 7320 6f72 2063 6861 6e67   layers or chang
-0000f760: 6520 7468 6520 6170 7065 6172 616e 6365  e the appearance
-0000f770: 206f 6e20 7468 6520 6d69 7865 6420 2a4b   on the mixed *K
-0000f780: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-0000f790: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
-0000f7a0: 696e 636c 7564 6520 602d 2d70 6c6f 7461  include `--plota
-0000f7b0: 7067 6d6d 6020 616e 6420 602d 2d70 6c6f  pgmm` and `--plo
-0000f7c0: 7465 6c6d 6d60 2065 7463 2e20 5468 6520  telmm` etc. The 
-0000f7d0: 6f70 7469 6f6e 2060 2d2d 706c 6f74 6170  option `--plotap
-0000f7e0: 676d 6d60 2063 616e 2062 6520 7365 7420  gmm` can be set 
-0000f7f0: 746f 2063 616c 6c20 7468 6520 474d 4d20  to call the GMM 
-0000f800: 616e 616c 7973 6973 2075 706f 6e20 7468  analysis upon th
-0000f810: 6520 616e 6368 6f72 2070 6169 7220 2a4b  e anchor pair *K
-0000f820: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-0000f830: 6520 6469 7374 7269 6275 7469 6f6e 2061  e distribution a
-0000f840: 6e64 2070 6c6f 7420 7468 6520 636c 7573  nd plot the clus
-0000f850: 7465 7269 6e67 2072 6573 756c 7420 7570  tering result up
-0000f860: 6f6e 2074 6865 206d 6978 6564 202a 4b2a  on the mixed *K*
-0000f870: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-0000f880: 2064 6973 7472 6962 7574 696f 6e2c 2077   distribution, w
-0000f890: 6869 6368 2068 6173 2074 6f20 6265 2063  hich has to be c
-0000f8a0: 6f2d 7365 7420 7769 7468 2074 6865 206f  o-set with the o
-0000f8b0: 7074 696f 6e20 602d 2d61 6e63 686f 7270  ption `--anchorp
-0000f8c0: 6f69 6e74 7360 2070 726f 7669 6469 6e67  oints` providing
-0000f8d0: 2074 6865 2061 6e63 686f 7220 7061 6972   the anchor pair
-0000f8e0: 7320 696e 666f 726d 6174 696f 6e2e 2054  s information. T
-0000f8f0: 6865 206f 7074 696f 6e20 602d 2d70 6c6f  he option `--plo
-0000f900: 7465 6c6d 6d60 2063 616e 2062 6520 7365  telmm` can be se
-0000f910: 7420 746f 2063 616c 6c20 7468 6520 454c  t to call the EL
-0000f920: 4d4d 2061 6e61 6c79 7369 7320 7570 6f6e  MM analysis upon
-0000f930: 2074 6865 2077 686f 6c65 2070 6172 616e   the whole paran
-0000f940: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
-0000f950: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
-0000f960: 7469 6f6e 2e20 2a2a 4e6f 7465 2074 6861  tion. **Note tha
-0000f970: 7420 7468 6520 7370 6563 6965 7320 6e61  t the species na
-0000f980: 6d65 7320 7072 6573 656e 7420 696e 2074  mes present in t
-0000f990: 6865 2073 7065 6369 6573 2074 7265 6520  he species tree 
-0000f9a0: 6669 6c65 2073 686f 756c 6420 6d61 7463  file should matc
-0000f9b0: 6820 7468 6520 6e61 6d65 7320 6f66 2074  h the names of t
-0000f9c0: 6865 2063 6f72 7265 7370 6f6e 6469 6e67  he corresponding
-0000f9d0: 2073 6571 7565 6e63 6520 6669 6c65 732a   sequence files*
-0000f9e0: 2a20 466f 7220 696e 7374 616e 6365 2c20  * For instance, 
-0000f9f0: 6769 7665 6e20 7468 6520 6364 7320 6669  given the cds fi
-0000fa00: 6c65 206e 616d 6573 2027 412e 6364 7327  le names 'A.cds'
-0000fa10: 2c27 422e 6364 7327 2c27 432e 6364 7327  ,'B.cds','C.cds'
-0000fa20: 2c20 7468 6520 7370 6563 6965 7320 7472  , the species tr
-0000fa30: 6565 2063 6f75 6c64 2062 6520 2728 412e  ee could be '(A.
-0000fa40: 6364 732c 2842 2e63 6473 2c43 2e63 6473  cds,(B.cds,C.cds
-0000fa50: 2929 3b27 2072 6174 6865 7220 7468 616e  ));' rather than
-0000fa60: 2027 2841 2c28 422c 4329 293b 272e 2054   '(A,(B,C));'. T
-0000fa70: 6865 7265 2069 7320 6e6f 2072 6571 7569  here is no requi
-0000fa80: 7265 6d65 6e74 2066 6f72 2074 6865 206e  rement for the n
-0000fa90: 616d 6520 6f66 2074 6865 2070 6172 616e  ame of the paran
-0000faa0: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
-0000fab0: 7562 3e20 6461 7461 6669 6c65 2077 6869  ub> datafile whi
-0000fac0: 6368 2063 616e 2062 6520 6e61 6d65 6420  ch can be named 
-0000fad0: 696e 2077 6861 7465 7665 7220 6d61 6e6e  in whatever mann
-0000fae0: 6572 2075 7365 7273 2070 7265 6665 722e  er users prefer.
-0000faf0: 2054 6865 2027 474d 4d27 2069 7320 7468   The 'GMM' is th
-0000fb00: 6520 6162 6272 6576 6961 7469 6f6e 206f  e abbreviation o
-0000fb10: 6620 4761 7573 7369 616e 204d 6978 7475  f Gaussian Mixtu
-0000fb20: 7265 204d 6f64 656c 696e 6720 7768 696c  re Modeling whil
-0000fb30: 6520 7468 6520 2745 4c4d 4d27 2072 6566  e the 'ELMM' ref
-0000fb40: 6572 7320 746f 2045 7870 6f6e 656e 7469  ers to Exponenti
-0000fb50: 616c 2d4c 6f67 6e6f 726d 616c 204d 6978  al-Lognormal Mix
-0000fb60: 7475 7265 204d 6f64 656c 696e 6720 6173  ture Modeling as
-0000fb70: 205b 6b73 7261 7465 735d 2868 7474 7073   [ksrates](https
-0000fb80: 3a2f 2f67 6974 6875 622e 636f 6d2f 5649  ://github.com/VI
-0000fb90: 422d 5053 422f 6b73 7261 7465 7329 2069  B-PSB/ksrates) i
-0000fba0: 6e74 6572 7072 6574 732e 0a0a 5468 6572  nterprets...Ther
-0000fbb0: 6520 6172 6520 3231 2063 6f6c 756d 6e73  e are 21 columns
-0000fbc0: 2069 6e20 7468 6520 7265 7375 6c74 2060   in the result `
-0000fbd0: 2e6b 732e 7473 7660 2066 696c 6520 6265  .ks.tsv` file be
-0000fbe0: 7369 6465 7320 7468 6520 696e 6465 7820  sides the index 
-0000fbf0: 636f 6c75 6d6e 7320 6070 6169 7260 2061  columns `pair` a
-0000fc00: 7320 7468 6520 756e 6971 7565 2069 6465  s the unique ide
-0000fc10: 6e74 6966 6965 7220 666f 7220 6561 6368  ntifier for each
-0000fc20: 2067 656e 6520 7061 6972 2e20 5468 6520   gene pair. The 
-0000fc30: 604e 602c 2060 5360 2c20 6064 4e60 2c20  `N`, `S`, `dN`, 
-0000fc40: 6064 4e2f 6453 602c 2060 6453 602c 2060  `dN/dS`, `dS`, `
-0000fc50: 6c60 2061 6e64 2060 7460 2061 7265 2066  l` and `t` are f
-0000fc60: 726f 6d20 7468 6520 636f 6465 6d6c 2072  rom the codeml r
-0000fc70: 6573 756c 7473 2c20 7265 7072 6573 656e  esults, represen
-0000fc80: 7469 6e67 2074 6865 204e 2065 7374 696d  ting the N estim
-0000fc90: 6174 652c 2074 6865 2053 2065 7374 696d  ate, the S estim
-0000fca0: 6174 652c 2074 6865 2064 4e20 6573 7469  ate, the dN esti
-0000fcb0: 6d61 7465 2c20 7468 6520 644e 2f64 5320  mate, the dN/dS 
-0000fcc0: 286f 6d65 6761 2920 6573 7469 6d61 7465  (omega) estimate
-0000fcd0: 2c20 7468 6520 6453 2065 7374 696d 6174  , the dS estimat
-0000fce0: 652c 2074 6865 206c 6f67 2d6c 696b 656c  e, the log-likel
-0000fcf0: 6968 6f6f 6420 616e 6420 7468 6520 7420  ihood and the t 
-0000fd00: 6573 7469 6d61 7465 2c20 7265 7370 6563  estimate, respec
-0000fd10: 7469 7665 6c79 2e20 5468 6520 6061 6c69  tively. The `ali
-0000fd20: 676e 6d65 6e74 636f 7665 7261 6765 602c  gnmentcoverage`,
-0000fd30: 2060 616c 6967 6e6d 656e 7469 6465 6e74   `alignmentident
-0000fd40: 6974 7960 2061 6e64 2060 616c 6967 6e6d  ity` and `alignm
-0000fd50: 656e 746c 656e 6774 6860 2061 7265 2074  entlength` are t
-0000fd60: 6865 2069 6e66 6f72 6d61 7469 6f6e 2070  he information p
-0000fd70: 6572 7461 696e 696e 6720 746f 2074 6865  ertaining to the
-0000fd80: 2061 6c69 676e 6d65 6e74 2066 6f72 2065   alignment for e
-0000fd90: 6163 6820 6661 6d69 6c79 2c20 7265 7072  ach family, repr
-0000fda0: 6573 656e 7469 6e67 2074 6865 2072 6174  esenting the rat
-0000fdb0: 696f 206f 6620 7468 6520 7374 7269 7070  io of the stripp
-0000fdc0: 6564 2061 6c69 676e 6d65 6e74 206c 656e  ed alignment len
-0000fdd0: 6774 6820 636f 6d70 6172 6564 2074 6f20  gth compared to 
-0000fde0: 7468 6520 6675 6c6c 2061 6c69 676e 6d65  the full alignme
-0000fdf0: 6e74 206c 656e 6774 682c 2074 6865 2072  nt length, the r
-0000fe00: 6174 696f 206f 6620 636f 6c75 6d6e 7320  atio of columns 
-0000fe10: 7769 7468 2069 6465 6e74 6963 616c 206e  with identical n
-0000fe20: 7563 6c65 6f74 6964 6573 2063 6f6d 7061  ucleotides compa
-0000fe30: 7265 6420 746f 2074 6865 206f 7665 7261  red to the overa
-0000fe40: 6c6c 2063 6f6c 756d 6e73 206f 6620 7468  ll columns of th
-0000fe50: 6520 7374 7269 7070 6564 2061 6c69 676e  e stripped align
-0000fe60: 6d65 6e74 2c20 616e 6420 7468 6520 6c65  ment, and the le
-0000fe70: 6e67 7468 206f 6620 7468 6520 6675 6c6c  ngth of the full
-0000fe80: 2061 6c69 676e 6d65 6e74 2c20 7265 7370   alignment, resp
-0000fe90: 6563 7469 7665 6c79 2e0a 2323 2320 7767  ectively..### wg
-0000fea0: 6420 6d69 780a 0a2a 2a54 6865 206d 6978  d mix..**The mix
-0000feb0: 7475 7265 206d 6f64 656c 2063 6c75 7374  ture model clust
-0000fec0: 6572 696e 6720 616e 616c 7973 6973 206f  ering analysis o
-0000fed0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
-0000fee0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
-0000fef0: 6f6e 2a2a 0a60 6060 0a77 6764 206d 6978  on**.```.wgd mix
-0000ff00: 206b 7364 6174 6120 282d 2d6e 5f69 6e69   ksdata (--n_ini
-0000ff10: 7420 3230 3020 2d2d 6d61 785f 6974 6572  t 200 --max_iter
-0000ff20: 2032 3030 202d 2d6b 735f 7261 6e67 6520   200 --ks_range 
-0000ff30: 3020 3520 2d2d 6669 6c74 6572 7320 3330  0 5 --filters 30
-0000ff40: 3020 2d2d 6269 6e73 2035 3020 2d2d 636f  0 --bins 50 --co
-0000ff50: 6d70 6f6e 656e 7473 2031 2034 202d 2d67  mponents 1 4 --g
-0000ff60: 616d 6d61 2030 2e30 3031 290a 6060 600a  amma 0.001).```.
-0000ff70: 0a54 6869 7320 7061 7274 206f 6620 4761  .This part of Ga
-0000ff80: 7573 7369 616e 206d 6978 7475 7265 206d  ussian mixture m
-0000ff90: 6f64 656c 696e 6720 2847 4d4d 2920 616e  odeling (GMM) an
-0000ffa0: 616c 7973 6973 2069 7320 696e 6865 7269  alysis is inheri
-0000ffb0: 7465 6420 6672 6f6d 2074 6865 206f 7269  ted from the ori
-0000ffc0: 6769 6e61 6c20 6077 6764 6020 7072 6f67  ginal `wgd` prog
-0000ffd0: 7261 6d2c 2062 7574 2077 7269 7465 7320  ram, but writes 
-0000ffe0: 6164 6469 7469 6f6e 616c 6c79 2074 6865  additionally the
-0000fff0: 2070 726f 6261 6269 6c69 7479 206f 6620   probability of 
-00010000: 6561 6368 202a 4b2a 3c73 7562 3e53 3c2f  each *K*<sub>S</
-00010010: 7375 623e 2076 616c 7565 2069 6e74 6f20  sub> value into 
-00010020: 7468 6520 6669 6e61 6c20 6461 7461 6672  the final datafr
-00010030: 616d 652e 2042 6173 6963 616c 6c79 2c20  ame. Basically, 
-00010040: 7573 6572 7320 6e65 6564 2074 6f20 7072  users need to pr
-00010050: 6f76 6964 6520 7769 7468 2061 2028 6e6f  ovide with a (no
-00010060: 726d 616c 6c79 2066 726f 6d20 7768 6f6c  rmally from whol
-00010070: 652d 7061 7261 6e6f 6d65 206f 7220 616e  e-paranome or an
-00010080: 6368 6f72 2d70 6169 7273 2920 2a4b 2a3c  chor-pairs) *K*<
-00010090: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
-000100a0: 6669 6c65 2061 6e64 2074 6865 2047 4d4d  file and the GMM
-000100b0: 2061 6e61 6c79 7369 7320 7769 6c6c 2062   analysis will b
-000100c0: 6520 636f 6e64 7563 7465 6420 7570 6f6e  e conducted upon
-000100d0: 2074 6865 2064 6174 6166 696c 652e 2053   the datafile. S
-000100e0: 6f6d 6520 7061 7261 6d65 7465 7273 2063  ome parameters c
-000100f0: 616e 2061 6666 6563 7420 7468 6520 7265  an affect the re
-00010100: 7375 6c74 732c 2069 6e63 6c75 6469 6e67  sults, including
-00010110: 2060 2d2d 6e5f 696e 6974 602c 2077 6869   `--n_init`, whi
-00010120: 6368 2073 6574 7320 7468 6520 6e75 6d62  ch sets the numb
-00010130: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
-00010140: 6974 6961 6c69 7a61 7469 6f6e 7320 2864  itializations (d
-00010150: 6566 6175 6c74 2032 3030 292c 2060 2d2d  efault 200), `--
-00010160: 6d61 785f 6974 6572 602c 2077 6869 6368  max_iter`, which
-00010170: 2073 6574 7320 7468 6520 6d61 7869 6d75   sets the maximu
-00010180: 6d20 6e75 6d62 6572 206f 6620 6974 6572  m number of iter
-00010190: 6174 696f 6e73 2028 6465 6661 756c 7420  ations (default 
-000101a0: 3230 3029 2c20 602d 2d6d 6574 686f 6460  200), `--method`
-000101b0: 2c20 7768 6963 6820 6465 7465 726d 696e  , which determin
-000101c0: 6573 2077 6869 6368 2063 6c75 7374 6572  es which cluster
-000101d0: 696e 6720 6d65 7468 6f64 2074 6f20 7573  ing method to us
-000101e0: 6520 2864 6566 6175 6c74 2067 6d6d 292c  e (default gmm),
-000101f0: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
-00010200: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
-00010210: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-00010220: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
-00010230: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
-00010240: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
-00010250: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
-00010260: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-00010270: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-00010280: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
-00010290: 3120 3429 2c20 616e 6420 7468 6520 6461  1 4), and the da
-000102a0: 7461 2066 696c 7465 7269 6e67 2070 6172  ta filtering par
-000102b0: 616d 6574 6572 7320 602d 2d66 696c 7465  ameters `--filte
-000102c0: 7273 6020 7768 6963 6820 6669 6c74 6572  rs` which filter
-000102d0: 7320 6461 7461 2062 6173 6564 206f 6e20  s data based on 
-000102e0: 616c 6967 6e6d 656e 7420 6c65 6e67 7468  alignment length
-000102f0: 2c20 602d 2d6b 735f 7261 6e67 6560 2077  , `--ks_range` w
-00010300: 6869 6368 2066 696c 7465 7273 2064 6174  hich filters dat
-00010310: 6120 6261 7365 6420 6f6e 202a 4b2a 3c73  a based on *K*<s
-00010320: 7562 3e53 3c2f 7375 623e 2076 616c 7565  ub>S</sub> value
-00010330: 7320 616e 6420 7468 6520 7061 7261 6d65  s and the parame
-00010340: 7465 7220 602d 2d62 696e 7360 2077 6869  ter `--bins` whi
-00010350: 6368 2073 6574 7320 7468 6520 6e75 6d62  ch sets the numb
-00010360: 6572 206f 6620 6269 6e73 2069 6e20 2a4b  er of bins in *K
-00010370: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00010380: 7374 7269 6275 7469 6f6e 2028 6465 6661  stribution (defa
-00010390: 756c 7420 3530 292e 0a0a 2a2a 4120 7375  ult 50)...**A su
-000103a0: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
-000103b0: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
-000103c0: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
-000103d0: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
-000103e0: 6d69 7820 6b73 6461 7461 0a60 6060 0a0a  mix ksdata.```..
-000103f0: 2323 2320 7767 6420 7065 616b 0a0a 2a2a  ### wgd peak..**
-00010400: 5468 6520 7365 6172 6368 206f 6620 6372  The search of cr
-00010410: 6564 6961 626c 6520 2a4b 2a3c 7375 623e  ediable *K*<sub>
-00010420: 533c 2f73 7562 3e20 7261 6e67 6520 7573  S</sub> range us
-00010430: 6564 2069 6e20 5747 4420 6461 7469 6e67  ed in WGD dating
-00010440: 2a2a 0a60 6060 0a77 6764 2070 6561 6b20  **.```.wgd peak 
-00010450: 6b73 6461 7461 202d 6170 2061 7064 6174  ksdata -ap apdat
-00010460: 6120 2d73 6d20 736d 6461 7461 202d 6c65  a -sm smdata -le
-00010470: 206c 6564 6174 6120 2d6d 7020 6d70 6461   ledata -mp mpda
-00010480: 7461 202d 2d68 6575 7269 7374 6963 2028  ta --heuristic (
-00010490: 2d2d 616c 6967 6e66 696c 7465 7220 302e  --alignfilter 0.
-000104a0: 3020 3020 302e 3020 2d2d 6b73 7261 6e67  0 0 0.0 --ksrang
-000104b0: 6520 3020 3520 2d2d 6269 6e5f 7769 6474  e 0 5 --bin_widt
-000104c0: 6820 302e 3120 2d2d 6775 6964 6520 7365  h 0.1 --guide se
-000104d0: 676d 656e 7420 2d2d 7072 6f6d 696e 656e  gment --prominen
-000104e0: 6365 5f63 7574 6f66 6620 302e 3120 2d2d  ce_cutoff 0.1 --
-000104f0: 7265 6c5f 6865 6967 6874 2030 2e34 202d  rel_height 0.4 -
-00010500: 2d63 6920 3935 202d 2d68 6472 2039 3520  -ci 95 --hdr 95 
-00010510: 2d2d 6b73 6375 746f 6666 2035 290a 6060  --kscutoff 5).``
-00010520: 600a 0a41 7320 6d65 6e74 696f 6e65 6420  `..As mentioned 
-00010530: 7072 6576 696f 7573 6c79 2c20 6120 6865  previously, a he
-00010540: 7572 6973 7469 6320 6d65 7468 6f64 2061  uristic method a
-00010550: 6e64 2061 2063 6f6c 6c69 6e65 6172 2073  nd a collinear s
-00010560: 6567 6d65 6e74 732d 6775 6964 6564 2061  egments-guided a
-00010570: 6e63 686f 7220 7061 6972 2063 6c75 7374  nchor pair clust
-00010580: 6572 696e 6720 666f 7220 7468 6520 7365  ering for the se
-00010590: 6172 6368 206f 6620 6372 6564 6961 626c  arch of crediabl
-000105a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000105b0: 3e20 7261 6e67 6520 7573 6564 2069 6e20  > range used in 
-000105c0: 5747 4420 6461 7469 6e67 2061 7265 2069  WGD dating are i
-000105d0: 6d70 6c65 6d65 6e74 6564 2069 6e20 6077  mplemented in `w
-000105e0: 6764 2076 3260 2e20 5573 6572 7320 6e65  gd v2`. Users ne
-000105f0: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
-00010600: 6520 616e 6368 6f72 2070 6f69 6e74 732c  e anchor points,
-00010610: 2073 6567 6d65 6e74 732c 206c 6973 7473   segments, lists
-00010620: 6567 6d65 6e74 732c 206d 756c 7469 706c  egments, multipl
-00010630: 6963 6f6e 7061 6972 7320 6461 7461 6669  iconpairs datafi
-00010640: 6c65 2066 726f 6d20 6069 2d61 6468 6f72  le from `i-adhor
-00010650: 6560 2074 6f20 6163 6869 6576 6520 7468  e` to achieve th
-00010660: 6520 636c 7573 7465 7269 6e67 2066 756e  e clustering fun
-00010670: 6374 696f 6e2e 2053 6f6d 6520 7061 7261  ction. Some para
-00010680: 6d65 7465 7273 2074 6861 7420 6361 6e20  meters that can 
-00010690: 696d 7061 6374 2074 6865 2072 6573 756c  impact the resul
-000106a0: 7473 2069 6e63 6c75 6465 2060 2d2d 616c  ts include `--al
-000106b0: 6967 6e66 696c 7465 7260 2c20 7768 6963  ignfilter`, whic
-000106c0: 6820 6669 6c74 6572 7320 7468 6520 6461  h filters the da
-000106d0: 7461 2062 6173 6564 206f 6e20 616c 6967  ta based on alig
-000106e0: 6e6d 656e 7420 6964 656e 7469 7479 2c20  nment identity, 
-000106f0: 6c65 6e67 7468 2061 6e64 2063 6f76 6572  length and cover
-00010700: 6167 652c 2060 2d2d 6b73 7261 6e67 6560  age, `--ksrange`
-00010710: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
-00010720: 2072 616e 6765 206f 6620 4b73 2074 6f20   range of Ks to 
-00010730: 6265 2061 6e61 6c79 7a65 642c 2060 2d2d  be analyzed, `--
-00010740: 6269 6e5f 7769 6474 6860 2c20 7768 6963  bin_width`, whic
-00010750: 6820 7365 7473 2074 6865 2062 616e 6477  h sets the bandw
-00010760: 6964 7468 206f 6620 2a4b 2a3c 7375 623e  idth of *K*<sub>
-00010770: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
-00010780: 7469 6f6e 2c20 602d 2d77 6569 6768 7473  tion, `--weights
-00010790: 5f6f 7574 6c69 6572 735f 696e 636c 7564  _outliers_includ
-000107a0: 6564 6020 7768 6963 6820 6465 7465 726d  ed` which determ
-000107b0: 696e 6573 2077 6865 7468 6572 2074 6f20  ines whether to 
-000107c0: 696e 636c 7564 6520 2a4b 2a3c 7375 623e  include *K*<sub>
-000107d0: 533c 2f73 7562 3e20 6f75 746c 6965 7273  S</sub> outliers
-000107e0: 2028 7768 6f73 6520 7661 6c75 6520 6973   (whose value is
-000107f0: 206f 7665 7220 3529 2069 6e20 616e 616c   over 5) in anal
-00010800: 7973 6973 2c20 602d 2d6d 6574 686f 6460  ysis, `--method`
-00010810: 2077 6869 6368 2064 6574 6572 6d69 6e65   which determine
-00010820: 7320 7768 6963 6820 636c 7573 7465 7269  s which clusteri
-00010830: 6e67 206d 6574 686f 6420 746f 2075 7365  ng method to use
-00010840: 2028 6465 6661 756c 7420 676d 6d29 2c20   (default gmm), 
-00010850: 602d 2d73 6565 6460 2077 6869 6368 2073  `--seed` which s
-00010860: 6574 7320 7468 6520 7261 6e64 6f6d 2073  ets the random s
-00010870: 6565 6420 6769 7665 6e20 746f 2069 6e69  eed given to ini
-00010880: 7469 616c 697a 6174 696f 6e20 2864 6566  tialization (def
-00010890: 6175 6c74 2032 3335 3238 3930 292c 2060  ault 2352890), `
-000108a0: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
-000108b0: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-000108c0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-000108d0: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
-000108e0: 6175 6c74 2032 3030 292c 2060 2d2d 656d  ault 200), `--em
-000108f0: 5f69 7465 7260 2c20 7768 6963 6820 7365  _iter`, which se
-00010900: 7473 2074 6865 206d 6178 696d 756d 206e  ts the maximum n
-00010910: 756d 6265 7220 6f66 2069 7465 7261 7469  umber of iterati
-00010920: 6f6e 7320 2864 6566 6175 6c74 2032 3030  ons (default 200
-00010930: 292c 2060 2d2d 6761 6d6d 6160 2c20 7768  ), `--gamma`, wh
-00010940: 6963 6820 7365 7473 2074 6865 2067 616d  ich sets the gam
-00010950: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
-00010960: 2074 6865 2062 676d 6d20 6d6f 6465 6c20   the bgmm model 
-00010970: 2864 6566 6175 6c74 2030 2e30 3031 292c  (default 0.001),
-00010980: 2060 2d2d 636f 6d70 6f6e 656e 7473 602c   `--components`,
-00010990: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
-000109a0: 7261 6e67 6520 6f66 2074 6865 206e 756d  range of the num
-000109b0: 6265 7220 6f66 2063 6f6d 706f 6e65 6e74  ber of component
-000109c0: 7320 746f 2066 6974 2028 6465 6661 756c  s to fit (defaul
-000109d0: 7420 3120 3429 2c20 602d 2d77 6569 6768  t 1 4), `--weigh
-000109e0: 7465 6460 2077 6869 6368 2064 6574 6572  ted` which deter
-000109f0: 6d69 6e65 7320 7768 6574 6865 7220 746f  mines whether to
-00010a00: 2075 7365 206e 6f64 652d 7765 6967 6874   use node-weight
-00010a10: 6564 206d 6574 686f 6420 666f 7220 6465  ed method for de
-00010a20: 2d72 6564 756e 6461 6e63 792c 2060 2d2d  -redundancy, `--
-00010a30: 6775 6964 6560 2077 6869 6368 2064 6574  guide` which det
-00010a40: 6572 6d69 6e65 7320 7768 6963 6820 7265  ermines which re
-00010a50: 6769 6d65 2072 6573 6964 696e 6720 616e  gime residing an
-00010a60: 6368 6f72 7320 746f 2062 6520 7573 6564  chors to be used
-00010a70: 2028 6465 6661 756c 7420 5365 676d 656e   (default Segmen
-00010a80: 7429 2c20 602d 2d70 726f 6d69 6e65 6e63  t), `--prominenc
-00010a90: 655f 6375 746f 6666 6020 7768 6963 6820  e_cutoff` which 
-00010aa0: 7365 7473 2074 6865 2070 726f 6d69 6e65  sets the promine
-00010ab0: 6e63 6520 6375 746f 6666 206f 6620 6163  nce cutoff of ac
-00010ac0: 6365 7074 6162 6c65 2070 6561 6b73 2069  ceptable peaks i
-00010ad0: 6e20 7065 616b 2066 696e 6469 6e67 2070  n peak finding p
-00010ae0: 726f 6365 7373 2c20 602d 2d72 656c 5f68  rocess, `--rel_h
-00010af0: 6569 6768 7460 2077 6869 6368 2073 6574  eight` which set
-00010b00: 7320 7468 6520 7265 6c61 7469 7665 2068  s the relative h
-00010b10: 6569 6768 7420 6174 2077 6869 6368 2074  eight at which t
-00010b20: 6865 2070 6561 6b20 7769 6474 6820 6973  he peak width is
-00010b30: 206d 6561 7375 7265 642c 2060 2d2d 6b73   measured, `--ks
-00010b40: 746f 6461 7465 6020 7768 6963 6820 6d61  todate` which ma
-00010b50: 6e75 616c 6c79 2073 6574 7320 7468 6520  nually sets the 
-00010b60: 7261 6e67 6520 6f66 202a 4b2a 3c73 7562  range of *K*<sub
-00010b70: 3e53 3c2f 7375 623e 2074 6f20 6265 2064  >S</sub> to be d
-00010b80: 6174 6564 2069 6e20 6865 7572 6973 7469  ated in heuristi
-00010b90: 6320 7365 6172 6368 2061 6e64 206e 6565  c search and nee
-00010ba0: 6473 2074 6f20 6265 2063 6f2d 7365 7420  ds to be co-set 
-00010bb0: 7769 7468 206f 7074 696f 6e20 602d 2d6d  with option `--m
-00010bc0: 616e 7561 6c73 6574 602c 2060 2d2d 786c  anualset`, `--xl
-00010bd0: 696d 6020 616e 6420 602d 2d79 6c69 6d60  im` and `--ylim`
-00010be0: 2064 6574 6572 6d69 6e69 6e67 2074 6865   determining the
-00010bf0: 2078 2061 6e64 2079 2061 7869 7320 6c69   x and y axis li
-00010c00: 6d69 7420 6f66 2047 4d4d 202a 4b2a 3c73  mit of GMM *K*<s
-00010c10: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
-00010c20: 6962 7574 696f 6e2c 2060 2d2d 6369 6020  ibution, `--ci` 
-00010c30: 7365 7474 696e 6720 7468 6520 636f 6e66  setting the conf
-00010c40: 6964 656e 6365 206c 6576 656c 206f 6620  idence level of 
-00010c50: 6c6f 672d 6e6f 726d 616c 2064 6973 7472  log-normal distr
-00010c60: 6962 7574 696f 6e20 746f 2064 6174 6520  ibution to date 
-00010c70: 2864 6566 6175 6c74 2039 3529 2c20 602d  (default 95), `-
-00010c80: 2d68 6472 6020 7365 7474 696e 6720 7468  -hdr` setting th
-00010c90: 6520 6869 6768 6573 7420 6465 6e73 6974  e highest densit
-00010ca0: 7920 7265 6769 6f6e 2028 4844 5229 2061  y region (HDR) a
-00010cb0: 7070 6c69 6564 2069 6e20 7468 6520 7365  pplied in the se
-00010cc0: 676d 656e 742d 6775 6964 6564 2061 6e63  gment-guided anc
-00010cd0: 686f 7220 7061 6972 202a 4b2a 3c73 7562  hor pair *K*<sub
-00010ce0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-00010cf0: 7574 696f 6e20 2864 6566 6175 6c74 2039  ution (default 9
-00010d00: 3529 2c20 602d 2d68 6575 7269 7374 6963  5), `--heuristic
-00010d10: 6020 6465 7465 726d 696e 696e 6720 7768  ` determining wh
-00010d20: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00010d30: 6520 6865 7572 6973 7469 6320 6d65 7468  e heuristic meth
-00010d40: 6f64 206f 6620 6465 6669 6e69 6e67 2043  od of defining C
-00010d50: 4920 666f 7220 6461 7469 6e67 2c20 602d  I for dating, `-
-00010d60: 2d6b 7363 7574 6f66 6660 2073 6574 7469  -kscutoff` setti
-00010d70: 6e67 2074 6865 202a 4b2a 3c73 7562 3e53  ng the *K*<sub>S
-00010d80: 3c2f 7375 623e 2073 6174 7572 6174 696f  </sub> saturatio
-00010d90: 6e20 6375 746f 6666 2069 6e20 6461 7469  n cutoff in dati
-00010da0: 6e67 2028 6465 6661 756c 7420 3529 2e0a  ng (default 5)..
-00010db0: 0a46 6f75 7220 7265 7375 6c74 2073 7562  .Four result sub
-00010dc0: 666c 6f64 6572 7320 7769 6c6c 2062 6520  floders will be 
-00010dd0: 7072 6f64 7563 6564 2c20 6e61 6d65 6c79  produced, namely
-00010de0: 2060 416e 6368 6f72 4b73 5f46 696e 6450   `AnchorKs_FindP
-00010df0: 6561 6b60 2c20 6041 6e63 686f 724b 735f  eak`, `AnchorKs_
-00010e00: 474d 4d60 2c20 6053 6567 6d65 6e74 4775  GMM`, `SegmentGu
-00010e10: 6964 654b 735f 474d 4d60 2061 6e64 2060  ideKs_GMM` and `
-00010e20: 5365 676d 656e 744b 735f 4669 6e64 5065  SegmentKs_FindPe
-00010e30: 616b 602e 2054 6865 2060 416e 6368 6f72  ak`. The `Anchor
-00010e40: 4b73 5f46 696e 6450 6561 6b60 2073 7562  Ks_FindPeak` sub
-00010e50: 666c 6f64 6572 2063 6f6e 7461 696e 7320  floder contains 
-00010e60: 7265 7375 6c74 7320 6f66 2074 6865 2064  results of the d
-00010e70: 6574 6563 7465 6420 7065 616b 7320 6279  etected peaks by
-00010e80: 2074 6865 2060 7369 676e 616c 6020 6d6f   the `signal` mo
-00010e90: 6475 6c65 206f 6620 6053 6369 5079 6020  dule of `SciPy` 
-00010ea0: 6c69 6272 6172 7920 616e 6420 7468 6520  library and the 
-00010eb0: 6173 7375 6d65 6420 6869 6768 6573 7420  assumed highest 
-00010ec0: 6d61 7373 2070 6172 7420 2872 6566 6572  mass part (refer
-00010ed0: 7265 6420 746f 2061 7320 4869 6768 4d61  red to as HighMa
-00010ee0: 7373 2068 6572 6561 6674 6572 2920 6f66  ss hereafter) of
-00010ef0: 2065 6163 6820 7065 616b 2c20 7768 6963   each peak, whic
-00010f00: 6820 6361 6e20 6265 2075 7365 6420 666f  h can be used fo
-00010f10: 7220 6675 7274 6865 7220 5747 4420 6461  r further WGD da
-00010f20: 7469 6e67 2e20 5468 6520 6041 6e63 686f  ting. The `Ancho
-00010f30: 724b 735f 474d 4d60 2073 686f 7773 2074  rKs_GMM` shows t
-00010f40: 6865 2047 4d4d 2072 6573 756c 7473 2075  he GMM results u
-00010f50: 706f 6e20 7468 6520 6f72 6967 696e 616c  pon the original
-00010f60: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
-00010f70: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
-00010f80: 7469 6f6e 2062 7920 7468 6520 606d 6978  tion by the `mix
-00010f90: 7475 7265 6020 6d6f 6475 6c65 206f 6620  ture` module of 
-00010fa0: 6073 6369 6b69 742d 6c65 6172 6e60 206c  `scikit-learn` l
-00010fb0: 6962 7261 7279 2061 6e64 2074 776f 2073  ibrary and two s
-00010fc0: 7562 666c 6f64 6572 732c 2060 4c6f 6747  ubfloders, `LogG
-00010fd0: 4d4d 5f43 4960 2063 6f6e 7461 696e 696e  MM_CI` containin
-00010fe0: 6720 7468 6520 7265 7375 6c74 7320 6f66  g the results of
-00010ff0: 2039 3525 2043 4920 6f66 2065 6163 6820   95% CI of each 
-00011000: 636f 6d70 6f6e 656e 742c 2060 4869 6768  component, `High
-00011010: 4d61 7373 5f43 4960 2063 6f6e 7461 696e  Mass_CI` contain
-00011020: 696e 6720 7468 6520 4869 6768 4d61 7373  ing the HighMass
-00011030: 206f 6620 6561 6368 2063 6f6d 706f 6e65   of each compone
-00011040: 6e74 2c20 7768 6963 6820 6361 6e20 6265  nt, which can be
-00011050: 2075 7365 6420 666f 7220 6675 7274 6865   used for furthe
-00011060: 7220 5747 4420 6461 7469 6e67 2e20 5468  r WGD dating. Th
-00011070: 6520 6053 6567 6d65 6e74 4775 6964 654b  e `SegmentGuideK
-00011080: 735f 474d 4d60 2073 7562 666f 6c64 6572  s_GMM` subfolder
-00011090: 2070 7265 7365 6e74 7320 7265 7375 6c74   presents result
-000110a0: 7320 6f66 2073 6567 6d65 6e74 202a 4b2a  s of segment *K*
-000110b0: 3c73 7562 3e53 3c2f 7375 623e 2047 4d4d  <sub>S</sub> GMM
-000110c0: 2077 6869 6368 2061 7265 206d 6170 7065   which are mappe
-000110d0: 6420 6261 636b 2074 6f20 7468 6520 7265  d back to the re
-000110e0: 7369 6469 6e67 2061 6e63 686f 7220 7061  siding anchor pa
-000110f0: 6972 7320 616e 6420 7468 6520 6173 736f  irs and the asso
-00011100: 6369 6174 6564 2039 3525 2048 4452 2061  ciated 95% HDR a
-00011110: 6e64 2048 6967 684d 6173 7320 6f66 2065  nd HighMass of e
-00011120: 6163 6820 7365 676d 656e 7420 636c 7573  ach segment clus
-00011130: 7465 7220 696e 2073 7562 666c 6f64 6572  ter in subfloder
-00011140: 7320 6f66 2060 4844 525f 4349 6020 616e  s of `HDR_CI` an
-00011150: 6420 6048 6967 684d 6173 735f 4349 602e  d `HighMass_CI`.
-00011160: 2054 6865 2060 5365 676d 656e 744b 735f   The `SegmentKs_
-00011170: 4669 6e64 5065 616b 6020 7375 6266 6f6c  FindPeak` subfol
-00011180: 6465 7220 6973 2073 696d 696c 6172 2074  der is similar t
-00011190: 6f20 6041 6e63 686f 724b 735f 4669 6e64  o `AnchorKs_Find
-000111a0: 5065 616b 6020 6275 7420 7769 7468 2073  Peak` but with s
-000111b0: 6567 6d65 6e74 202a 4b2a 3c73 7562 3e53  egment *K*<sub>S
-000111c0: 3c2f 7375 623e 2069 6e73 7465 6164 2e20  </sub> instead. 
-000111d0: 5468 6520 2a4b 2a3c 7375 623e 533c 2f73  The *K*<sub>S</s
-000111e0: 7562 3e20 696e 2060 4d75 6c74 6970 6c69  ub> in `Multipli
-000111f0: 636f 6e60 2063 616e 2061 6c73 6f20 6265  con` can also be
-00011200: 2063 616c 6375 6c61 7465 6420 696e 2070   calculated in p
-00011210: 6c61 6365 206f 6620 6053 6567 6d65 6e74  lace of `Segment
-00011220: 6020 7573 696e 6720 7468 6520 6f70 7469  ` using the opti
-00011230: 6f6e 2060 2d2d 6775 6964 6560 2061 7320  on `--guide` as 
-00011240: 7375 6368 2074 6865 2072 6573 756c 7420  such the result 
-00011250: 7469 746c 652c 206c 6162 656c 2c20 6669  title, label, fi
-00011260: 6c65 2061 6e64 2066 6f6c 6465 7220 6e61  le and folder na
-00011270: 6d65 7320 7769 6c6c 2062 6520 6368 616e  mes will be chan
-00011280: 6765 6420 6163 636f 7264 696e 676c 792e  ged accordingly.
-00011290: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-000112a0: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
-000112b0: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
-000112c0: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
-000112d0: 6060 600a 7767 6420 7065 616b 206b 7364  ```.wgd peak ksd
-000112e0: 6174 6120 2d61 7020 6170 6461 7461 202d  ata -ap apdata -
-000112f0: 736d 2073 6d64 6174 6120 2d6c 6520 6c65  sm smdata -le le
-00011300: 6461 7461 202d 6d70 206d 7064 6174 6120  data -mp mpdata 
-00011310: 2d2d 6865 7572 6973 7469 630a 6060 600a  --heuristic.```.
-00011320: 0a23 2323 2077 6764 2073 796e 0a0a 2a2a  .### wgd syn..**
-00011330: 5468 6520 696e 7472 612d 7370 6563 6966  The intra-specif
-00011340: 6963 2073 796e 7465 6e79 2069 6e66 6572  ic synteny infer
-00011350: 656e 6365 2a2a 0a60 6060 0a77 6764 2073  ence**.```.wgd s
-00011360: 796e 2066 616d 696c 6965 7320 6766 6620  yn families gff 
-00011370: 282d 2d6b 735f 6469 7374 7269 6275 7469  (--ks_distributi
-00011380: 6f6e 206b 7364 6174 6120 2d66 2067 656e  on ksdata -f gen
-00011390: 6520 2d61 2049 4420 2d2d 6d69 6e6c 656e  e -a ID --minlen
-000113a0: 202d 3120 2d2d 6d69 6e73 6567 6c65 6e20   -1 --minseglen 
-000113b0: 3130 3030 3020 2d2d 6d69 6e67 656e 656e  10000 --mingenen
-000113c0: 756d 2033 3029 0a60 6060 0a0a 5468 6520  um 30).```..The 
-000113d0: 7072 6f67 7261 6d20 6077 6764 2073 796e  program `wgd syn
-000113e0: 6020 6973 206d 6169 6e6c 7920 6465 616c  ` is mainly deal
-000113f0: 696e 6720 7769 7468 2063 6f6c 6c69 6e65  ing with colline
-00011400: 6172 6974 7920 6f72 2073 796e 7465 6e79  arity or synteny
-00011410: 2028 626f 7468 2072 6566 6572 7265 6420   (both referred 
-00011420: 746f 2061 7320 7379 6e74 656e 7920 6865  to as synteny he
-00011430: 7265 6166 7465 7229 2061 6e61 6c79 7369  reafter) analysi
-00011440: 732e 2054 776f 2069 6e70 7574 2066 696c  s. Two input fil
-00011450: 6573 2061 7265 2065 7373 656e 7469 616c  es are essential
-00011460: 2c20 7468 6520 6765 6e65 2066 616d 696c  , the gene famil
-00011470: 7920 6669 6c65 2061 6e64 2074 6865 2067  y file and the g
-00011480: 6666 3320 6669 6c65 2e20 5468 6520 6765  ff3 file. The ge
-00011490: 6e65 2066 616d 696c 7920 6669 6c65 2069  ne family file i
-000114a0: 7320 696e 2074 6865 2066 6f72 6d61 7420  s in the format 
-000114b0: 6173 2060 4f72 7468 6f46 696e 6465 7260  as `OrthoFinder`
-000114c0: 2e20 5468 6520 736f 6674 7761 7265 2060  . The software `
-000114d0: 692d 6164 686f 7265 6020 6973 2061 2070  i-adhore` is a p
-000114e0: 7265 7265 7175 6973 6974 652e 2057 6974  rerequisite. Wit
-000114f0: 6820 6465 6661 756c 7420 7061 7261 6d65  h default parame
-00011500: 7465 7273 2c20 7468 6520 7072 6f67 7261  ters, the progra
-00011510: 6d20 6261 7369 6361 6c6c 7920 636f 6e64  m basically cond
-00011520: 7563 7473 2031 2920 6669 6c74 6572 696e  ucts 1) filterin
-00011530: 6720 6765 6e65 2066 616d 696c 6965 7320  g gene families 
-00011540: 6261 7365 6420 6f6e 206d 6178 696d 756d  based on maximum
-00011550: 2066 616d 696c 7920 7369 7a65 2032 2920   family size 2) 
-00011560: 7265 7472 6965 7669 6e67 2067 656e 6520  retrieving gene 
-00011570: 706f 7369 7469 6f6e 2061 6e64 2073 6361  position and sca
-00011580: 6666 6f6c 6420 696e 666f 726d 6174 696f  ffold informatio
-00011590: 6e20 6672 6f6d 2067 6666 3320 6669 6c65  n from gff3 file
-000115a0: 2033 2920 7072 6f64 7563 696e 6720 7468   3) producing th
-000115b0: 6520 636f 6e66 6967 7572 6174 696f 6e20  e configuration 
-000115c0: 6669 6c65 2061 6e64 2061 7373 6f63 6961  file and associa
-000115d0: 7465 6420 6461 7461 6669 6c65 7320 666f  ted datafiles fo
-000115e0: 7220 6069 2d61 6468 6f72 6560 2034 2920  r `i-adhore` 4) 
-000115f0: 6361 6c6c 696e 6720 6069 2d61 6468 6f72  calling `i-adhor
-00011600: 6560 2067 6976 656e 2074 6865 2070 6172  e` given the par
-00011610: 616d 6574 6572 7320 7365 7420 746f 2069  ameters set to i
-00011620: 6e66 6572 2073 796e 7465 6e79 2035 2920  nfer synteny 5) 
-00011630: 7669 7375 616c 697a 696e 6720 7468 6520  visualizing the 
-00011640: 7379 6e74 656e 7920 696e 2022 646f 7470  synteny in "dotp
-00011650: 6c6f 7422 2069 6e20 7468 6520 756e 6974  lot" in the unit
-00011660: 206f 6620 6765 6e65 7320 616e 6420 6261   of genes and ba
-00011670: 7365 732c 2069 6e20 2253 796e 6465 7074  ses, in "Syndept
-00011680: 6822 2070 6c6f 7420 7368 6f77 696e 6720  h" plot showing 
-00011690: 7468 6520 6469 7374 7269 6275 7469 6f6e  the distribution
-000116a0: 206f 6620 6469 6666 6572 656e 7420 6361   of different ca
-000116b0: 7465 676f 7269 6573 206f 6620 636f 6c6c  tegories of coll
-000116c0: 696e 6561 7269 7479 2072 6174 696f 7320  inearity ratios 
-000116d0: 7769 7468 696e 2061 6e64 2062 6574 7765  within and betwe
-000116e0: 656e 2073 7065 6369 6573 2c20 696e 2022  en species, in "
-000116f0: 6475 7053 7461 636b 2220 706c 6f74 2073  dupStack" plot s
-00011700: 686f 7769 6e67 206d 756c 7469 706c 6963  howing multiplic
-00011710: 6f6e 7320 7769 7468 2064 6966 6665 7265  ons with differe
-00011720: 6e74 206d 756c 7469 706c 6963 6174 696f  nt multiplicatio
-00011730: 6e20 6c65 7665 6c73 2e20 3629 2069 6620  n levels. 6) if 
-00011740: 7769 7468 202a 4b2a 3c73 7562 3e53 3c2f  with *K*<sub>S</
-00011750: 7375 623e 2064 6174 612c 2061 2022 2a4b  sub> data, a "*K
-00011760: 2a3c 7375 623e 533c 2f73 7562 3e20 646f  *<sub>S</sub> do
-00011770: 7470 6c6f 7422 2077 6974 6820 646f 7473  tplot" with dots
-00011780: 2061 6e6e 6f74 6174 6564 2069 6e20 2a4b   annotated in *K
-00011790: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-000117a0: 6c75 6573 2061 6e64 2061 202a 4b2a 3c73  lues and a *K*<s
-000117b0: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
-000117c0: 6962 7574 696f 6e20 7769 7468 2061 6e63  ibution with anc
-000117d0: 686f 7220 7061 6972 7320 6465 6e6f 7465  hor pairs denote
-000117e0: 6420 7769 6c6c 2062 6520 7072 6f64 7563  d will be produc
-000117f0: 6564 2e20 5468 6520 6765 6e65 2069 6e66  ed. The gene inf
-00011800: 6f72 6d61 7469 6f6e 2069 6e20 7468 6520  ormation in the 
-00011810: 6765 6e65 2066 616d 696c 7920 6669 6c65  gene family file
-00011820: 2061 6e64 2067 6666 3320 6669 6c65 2073   and gff3 file s
-00011830: 686f 756c 6420 6265 206d 6174 6368 6564  hould be matched
-00011840: 2077 6869 6368 2072 6571 7569 7265 7320   which requires 
-00011850: 7573 6572 7320 746f 2073 6574 2070 726f  users to set pro
-00011860: 7065 7220 602d 2d66 6561 7475 7265 6020  per `--feature` 
-00011870: 616e 6420 602d 2d61 7474 7269 6275 7465  and `--attribute
-00011880: 602e 2054 6865 206d 6178 696d 756d 2066  `. The maximum f
-00011890: 616d 696c 7920 7369 7a65 2074 6f20 6265  amily size to be
-000118a0: 2069 6e63 6c75 6465 6420 6361 6e20 6265   included can be
-000118b0: 2073 6574 2076 6961 2074 6865 206f 7074   set via the opt
-000118c0: 696f 6e20 602d 2d6d 6178 7369 7a65 602c  ion `--maxsize`,
-000118d0: 206e 6f74 6564 2074 6861 7420 7468 6973   noted that this
-000118e0: 2066 696c 7465 7269 6e67 2069 7320 6d61   filtering is ma
-000118f0: 696e 6c79 2074 6f20 6472 6f70 2074 686f  inly to drop tho
-00011900: 7365 2068 7567 6520 7461 6e64 656d 2064  se huge tandem d
-00011910: 7570 6c69 6361 7465 7320 6661 6d69 6c79  uplicates family
-00011920: 2061 6e64 2074 7261 6e73 706f 7361 626c   and transposabl
-00011930: 6520 656c 656d 656e 7473 2028 5445 7329  e elements (TEs)
-00011940: 2066 616d 696c 792c 2061 6e64 206e 6f74   family, and not
-00011950: 206d 616e 6461 746f 7279 2e20 5573 6572   mandatory. User
-00011960: 7320 6361 6e20 6669 6c74 6572 2074 686f  s can filter tho
-00011970: 7365 2066 7261 676d 656e 7461 7279 2073  se fragmentary s
-00011980: 6361 6666 6f6c 6473 2076 6961 2074 6865  caffolds via the
-00011990: 206f 7074 696f 6e20 602d 2d6d 696e 6c65   option `--minle
-000119a0: 6e60 2e20 5468 6520 6d69 6e69 6d75 6d20  n`. The minimum 
-000119b0: 6c65 6e67 7468 2061 6e64 206e 756d 6265  length and numbe
-000119c0: 7220 6f66 2067 656e 6573 2066 6f72 2061  r of genes for a
-000119d0: 2073 6567 6d65 6e74 2074 6f20 6265 2063   segment to be c
-000119e0: 6f6e 7369 6465 7265 6420 6361 6e20 6265  onsidered can be
-000119f0: 2073 6574 2076 6961 2074 6865 206f 7074   set via the opt
-00011a00: 696f 6e20 602d 2d6d 696e 7365 676c 656e  ion `--minseglen
-00011a10: 6020 616e 6420 602d 2d6d 696e 6765 6e65  ` and `--mingene
-00011a20: 6e75 6d60 2e20 5265 6475 6e64 616e 7420  num`. Redundant 
-00011a30: 6d75 6c74 6970 6c69 636f 6e73 2063 616e  multiplicons can
-00011a40: 2062 6520 6b65 7074 2062 7920 7365 7420   be kept by set 
-00011a50: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-00011a60: 602d 2d6b 6565 7072 6564 756e 602e 0a0a  `--keepredun`...
-00011a70: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-00011a80: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-00011a90: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
-00011aa0: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
-00011ab0: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
-00011ac0: 6573 2067 6666 0a60 6060 0a0a 2a2a 5468  es gff.```..**Th
-00011ad0: 6520 696e 7465 722d 7370 6563 6966 6963  e inter-specific
-00011ae0: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
-00011af0: 6365 2a2a 0a60 6060 0a77 6764 2073 796e  ce**.```.wgd syn
-00011b00: 2066 616d 696c 6965 7320 6766 6631 2067   families gff1 g
-00011b10: 6666 3220 282d 2d61 6464 6974 696f 6e61  ff2 (--additiona
-00011b20: 6c67 6666 696e 666f 2022 6d52 4e41 3b4e  lgffinfo "mRNA;N
-00011b30: 616d 6522 202d 2d61 6464 6974 696f 6e61  ame" --additiona
-00011b40: 6c67 6666 696e 666f 2022 6765 6e65 3b49  lgffinfo "gene;I
-00011b50: 4422 290a 6060 600a 0a46 6f72 206d 756c  D").```..For mul
-00011b60: 7469 2d73 7065 6369 6573 2073 796e 7465  ti-species synte
-00011b70: 6e79 2069 6e66 6572 656e 6365 2c20 6966  ny inference, if
-00011b80: 2075 7365 7273 2068 6176 6520 6766 6633   users have gff3
-00011b90: 2066 696c 6573 2077 6869 6368 2068 6176   files which hav
-00011ba0: 6520 6469 6666 6572 656e 7420 6665 6174  e different feat
-00011bb0: 7572 6573 206f 7220 6174 7472 6962 7574  ures or attribut
-00011bc0: 6573 2066 6f72 2067 656e 6520 706f 7369  es for gene posi
-00011bd0: 7469 6f6e 2069 6e66 6f72 6d61 7469 6f6e  tion information
-00011be0: 2072 6574 7269 6576 616c 2c20 7468 6520   retrieval, the 
-00011bf0: 6f70 7469 6f6e 2060 2d2d 6164 6469 7469  option `--additi
-00011c00: 6f6e 616c 6766 6669 6e66 6f60 2063 616e  onalgffinfo` can
-00011c10: 2062 6520 7365 7420 746f 2070 726f 7669   be set to provi
-00011c20: 6465 2074 6865 2061 6464 6974 696f 6e61  de the additiona
-00011c30: 6c20 696e 666f 726d 6174 696f 6e2e 2054  l information. T
-00011c40: 6865 2072 656d 6169 6e69 6e67 2070 6172  he remaining par
-00011c50: 616d 6574 6572 2073 6574 7469 6e67 2069  ameter setting i
-00011c60: 7320 7468 6520 7361 6d65 2061 7320 7468  s the same as th
-00011c70: 6520 696e 7472 612d 7370 6563 6966 6963  e intra-specific
-00011c80: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
-00011c90: 6365 2e0a 0a2a 2a41 2073 7567 6765 7374  ce...**A suggest
-00011ca0: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-00011cb0: 6361 6e20 7573 6520 636f 6d6d 616e 6420  can use command 
-00011cc0: 7369 6d70 6c79 2061 7320 6265 6c6f 772a  simply as below*
-00011cd0: 2a0a 0a60 6060 0a77 6764 2073 796e 2066  *..```.wgd syn f
-00011ce0: 616d 696c 6965 7320 6766 6631 2067 6666  amilies gff1 gff
-00011cf0: 320a 6060 600a 0a23 2323 2077 6764 2076  2.```..### wgd v
-00011d00: 697a 0a0a 2a2a 5468 6520 7669 7375 616c  iz..**The visual
-00011d10: 697a 6174 696f 6e20 6f66 202a 4b2a 3c73  ization of *K*<s
-00011d20: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-00011d30: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
-00011d40: 454c 4d4d 2061 6e61 6c79 7369 732a 2a0a  ELMM analysis**.
-00011d50: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
-00011d60: 7364 6174 610a 6060 600a 0a54 6865 2070  sdata.```..The p
-00011d70: 726f 6772 616d 2060 7767 6420 7669 7a60  rogram `wgd viz`
-00011d80: 2069 7320 6d61 696e 6c79 2066 6f72 2074   is mainly for t
-00011d90: 6865 2070 7572 706f 7365 206f 6620 2a4b  he purpose of *K
-00011da0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00011db0: 7374 7269 6275 7469 6f6e 2061 6e64 2073  stribution and s
-00011dc0: 796e 7465 6e79 2076 6973 7561 6c69 7a61  ynteny visualiza
-00011dd0: 7469 6f6e 2c20 7769 7468 2073 6f6d 6520  tion, with some 
-00011de0: 6f70 7469 6f6e 616c 206d 6978 7475 7265  optional mixture
-00011df0: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
-00011e00: 6973 2e20 5468 6520 6261 7369 6320 6675  is. The basic fu
-00011e10: 6e63 7469 6f6e 2069 7320 6a75 7374 2074  nction is just t
-00011e20: 6f20 706c 6f74 2074 6865 202a 4b2a 3c73  o plot the *K*<s
-00011e30: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
-00011e40: 6962 7574 696f 6e20 616e 6420 636f 6e64  ibution and cond
-00011e50: 7563 7420 616e 2045 4c4d 4d20 616e 616c  uct an ELMM anal
-00011e60: 7973 6973 2069 6e20 7365 6172 6368 206f  ysis in search o
-00011e70: 6620 706f 7465 6e74 6961 6c20 5747 4420  f potential WGD 
-00011e80: 636f 6d70 6f6e 656e 7473 2e20 536f 6d65  components. Some
-00011e90: 206b 6579 2070 6172 616d 6574 6572 7320   key parameters 
-00011ea0: 6166 6665 6374 696e 6720 7468 6520 454c  affecting the EL
-00011eb0: 4d4d 2072 6573 756c 7420 696e 636c 7564  MM result includ
-00011ec0: 6520 602d 2d70 726f 6d69 6e65 6e63 655f  e `--prominence_
-00011ed0: 6375 746f 6666 6020 616e 6420 602d 2d72  cutoff` and `--r
-00011ee0: 656c 5f68 6569 6768 7460 2c20 7768 6963  el_height`, whic
-00011ef0: 6820 6861 7665 2062 6565 6e20 6578 706c  h have been expl
-00011f00: 6169 6e65 6420 6168 6561 642c 2060 2d2d  ained ahead, `--
-00011f10: 656d 5f69 7465 7261 7469 6f6e 7360 2061  em_iterations` a
-00011f20: 6e64 2060 2d2d 656d 5f69 6e69 7469 616c  nd `--em_initial
-00011f30: 697a 6174 696f 6e73 6020 6465 7465 726d  izations` determ
-00011f40: 696e 696e 6720 7468 6520 6d61 7869 6d75  ining the maximu
-00011f50: 6d20 6974 6572 6174 696f 6e73 2061 6e64  m iterations and
-00011f60: 2069 6e69 7469 616c 697a 6174 696f 6e73   initializations
-00011f70: 2069 6e20 7468 6520 454d 2061 6c67 6f72   in the EM algor
-00011f80: 6974 686d 2e0a 0a2a 2a41 2073 7567 6765  ithm...**A sugge
-00011f90: 7374 6564 2073 7461 7274 696e 6720 7275  sted starting ru
-00011fa0: 6e20 6361 6e20 7573 6520 636f 6d6d 616e  n can use comman
-00011fb0: 6420 7369 6d70 6c79 2061 7320 6265 6c6f  d simply as belo
-00011fc0: 772a 2a0a 0a60 6060 0a77 6764 2076 697a  w**..```.wgd viz
-00011fd0: 202d 6420 6b73 6461 7461 0a60 6060 0a0a   -d ksdata.```..
-00011fe0: 2a2a 5468 6520 7669 7375 616c 697a 6174  **The visualizat
-00011ff0: 696f 6e20 6f66 202a 4b2a 3c73 7562 3e53  ion of *K*<sub>S
-00012000: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
-00012010: 6962 7574 696f 6e20 7769 7468 2072 6174  ibution with rat
-00012020: 6520 636f 7272 6563 7469 6f6e 2a2a 0a60  e correction**.`
-00012030: 6060 0a77 6764 2076 697a 202d 6420 6b73  ``.wgd viz -d ks
-00012040: 6461 7461 202d 7370 2073 7064 6174 6120  data -sp spdata 
-00012050: 2d2d 666f 6375 7332 616c 6c20 666f 6361  --focus2all foca
-00012060: 6c5f 7370 6563 6965 7320 2d2d 6578 7472  l_species --extr
-00012070: 6170 6172 616e 6f6d 656b 7320 6b73 6461  aparanomeks ksda
-00012080: 7461 2028 2d2d 616e 6368 6f72 706f 696e  ta (--anchorpoin
-00012090: 7473 2061 7064 6174 6120 2d2d 706c 6f74  ts apdata --plot
-000120a0: 6170 676d 6d20 2d2d 706c 6f74 656c 6d6d  apgmm --plotelmm
-000120b0: 290a 6060 600a 0a42 6573 6964 6573 2074  ).```..Besides t
-000120c0: 6865 2062 6173 6963 202a 4b2a 3c73 7562  he basic *K*<sub
-000120d0: 3e53 3c2f 7375 623e 2070 6c6f 742c 2073  >S</sub> plot, s
-000120e0: 7562 7374 6974 7574 696f 6e20 7261 7465  ubstitution rate
-000120f0: 2063 6f72 7265 6374 696f 6e20 6361 6e20   correction can 
-00012100: 616c 736f 2062 6520 6163 6869 6576 6564  also be achieved
-00012110: 2067 6976 656e 2061 7420 6c65 6173 7420   given at least 
-00012120: 6120 7370 6563 6965 7320 7472 6565 2028  a species tree (
-00012130: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
-00012140: 2d2d 7370 6563 6965 7374 7265 6560 2920  --speciestree`) 
-00012150: 616e 6420 6120 666f 6361 6c20 7370 6563  and a focal spec
-00012160: 6965 7320 2865 6974 6865 7220 7669 6120  ies (either via 
-00012170: 7468 6520 6f70 7469 6f6e 2060 2d2d 666f  the option `--fo
-00012180: 6375 7332 616c 6c60 206f 7220 7669 6120  cus2all` or via 
-00012190: 7468 6520 6f70 7469 6f6e 2060 2d2d 7370  the option `--sp
-000121a0: 6169 7260 2069 6e20 7468 6520 666f 726d  air` in the form
-000121b0: 206f 6620 2224 666f 6361 6c5f 7370 6563   of "$focal_spec
-000121c0: 6965 733b 2466 6f63 616c 5f73 7065 6369  ies;$focal_speci
-000121d0: 6573 2229 2e20 4974 2773 2073 7567 6765  es"). It's sugge
-000121e0: 7374 6564 2074 6861 7420 7468 6520 6f72  sted that the or
-000121f0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-00012200: 623e 533c 2f73 7562 3e20 6461 7461 2069  b>S</sub> data i
-00012210: 7320 7072 6f76 6964 6564 2062 7920 7468  s provided by th
-00012220: 6520 602d 2d64 6174 6166 696c 6560 206f  e `--datafile` o
-00012230: 7074 696f 6e20 616e 6420 7468 6520 7061  ption and the pa
-00012240: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
-00012250: 3e53 3c2f 7375 623e 2064 6174 6120 6973  >S</sub> data is
-00012260: 2070 726f 7669 6465 6420 6279 2074 6865   provided by the
-00012270: 2060 2d2d 6578 7472 6170 6172 616e 6f6d   `--extraparanom
-00012280: 656b 7360 206f 7074 696f 6e2c 2061 6c74  eks` option, alt
-00012290: 686f 7567 6820 6974 2773 2061 6c6c 6f77  hough it's allow
-000122a0: 6564 2074 6f20 6f6e 6c79 2070 726f 7669  ed to only provi
-000122b0: 6465 202a 4b2a 3c73 7562 3e53 3c2f 7375  de *K*<sub>S</su
-000122c0: 623e 2064 6174 6120 7669 6120 7468 6520  b> data via the 
-000122d0: 602d 2d64 6174 6166 696c 6560 206f 7074  `--datafile` opt
-000122e0: 696f 6e20 616e 6420 6465 706f 7369 7420  ion and deposit 
-000122f0: 626f 7468 206f 7274 686f 6c6f 676f 7573  both orthologous
-00012300: 2061 6e64 2070 6172 616c 6f67 6f75 7320   and paralogous 
-00012310: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00012320: 6461 7461 2074 6865 7265 6f6e 2e20 5468  data thereon. Th
-00012330: 6572 6520 6172 6520 7477 6f20 7479 7065  ere are two type
-00012340: 7320 6f66 206d 6978 6564 2070 6c6f 7473  s of mixed plots
-00012350: 2028 7468 6520 226d 6978 6564 2220 6865   (the "mixed" he
-00012360: 7265 2072 6566 6572 7320 746f 206d 6978  re refers to mix
-00012370: 6564 206f 7274 686f 6c6f 676f 7573 2061  ed orthologous a
-00012380: 6e64 2070 6172 616c 6f67 6f75 7320 2a4b  nd paralogous *K
-00012390: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-000123a0: 7374 7269 6275 7469 6f6e 7329 2c20 6f6e  stributions), on
-000123b0: 6520 6f66 2077 6869 6368 2069 7320 7369  e of which is si
-000123c0: 6d69 6c61 7220 746f 2077 6861 7420 606b  milar to what `k
-000123d0: 7372 6174 6573 6020 706c 6f74 732c 2077  srates` plots, w
-000123e0: 6869 6c65 2074 6865 206f 7468 6572 206f  hile the other o
-000123f0: 6620 7768 6963 6820 6973 206c 696b 6520  f which is like 
-00012400: 7468 6520 636f 6e76 656e 7469 6f6e 616c  the conventional
-00012410: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00012420: 2070 6c6f 7420 7468 6174 2062 6f74 6820   plot that both 
-00012430: 7468 6520 6f72 6967 696e 616c 206f 7274  the original ort
-00012440: 686f 6c6f 676f 7573 2061 6e64 2070 6172  hologous and par
-00012450: 616c 6f67 6f75 7320 2a4b 2a3c 7375 623e  alogous *K*<sub>
-00012460: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
-00012470: 7469 6f6e 7320 6172 6520 7472 7574 6866  tions are truthf
-00012480: 756c 6c79 2070 6c6f 7474 6564 2069 6e73  ully plotted ins
-00012490: 7465 6164 206f 6620 6a75 7374 2062 6569  tead of just bei
-000124a0: 6e67 2072 6570 7265 7365 6e74 6564 2062  ng represented b
-000124b0: 7920 736f 6d65 2076 6572 7469 6361 6c20  y some vertical 
-000124c0: 6c69 6e65 732e 2057 6520 7375 6767 6573  lines. We sugges
-000124d0: 7420 7573 6572 7320 746f 2061 646f 7074  t users to adopt
-000124e0: 2074 6865 2060 6b73 7261 7465 7360 2d6c   the `ksrates`-l
-000124f0: 696b 6520 706c 6f74 732c 2077 6869 6368  ike plots, which
-00012500: 2069 7320 7468 6520 6465 6661 756c 742e   is the default.
-00012510: 204f 7468 6572 7769 7365 2074 6865 206f   Otherwise the o
-00012520: 7074 696f 6e20 602d 2d63 6c61 7373 6963  ption `--classic
-00012530: 6020 7769 6c6c 2062 6520 6e65 6564 6564  ` will be needed
-00012540: 2074 6f20 7365 742e 2045 7874 7261 206d   to set. Extra m
-00012550: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-00012560: 616e 616c 7973 6973 2063 616e 2062 6520  analysis can be 
-00012570: 696e 6974 6961 7465 6420 7669 6120 7468  initiated via th
-00012580: 6520 6f70 7469 6f6e 2060 2d2d 706c 6f74  e option `--plot
-00012590: 656c 6d6d 6020 616e 6420 602d 2d70 6c6f  elmm` and `--plo
-000125a0: 7461 7067 6d6d 6020 7769 7468 2074 6865  tapgmm` with the
-000125b0: 2061 6e63 686f 7220 706f 696e 7473 2064   anchor points d
-000125c0: 6174 6166 696c 6520 7072 6f76 6964 6564  atafile provided
-000125d0: 2062 7920 7468 6520 602d 2d61 6e63 686f   by the `--ancho
-000125e0: 7270 6f69 6e74 7360 206f 7074 696f 6e2e  rpoints` option.
-000125f0: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
-00012600: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
-00012610: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
-00012620: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
-00012630: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
-00012640: 7364 6174 6120 2d73 7020 7370 6461 7461  sdata -sp spdata
-00012650: 202d 2d66 6f63 7573 3261 6c6c 2066 6f63   --focus2all foc
-00012660: 616c 5f73 7065 6369 6573 202d 2d65 7874  al_species --ext
-00012670: 7261 7061 7261 6e6f 6d65 6b73 206b 7364  raparanomeks ksd
-00012680: 6174 610a 6060 600a 0a2a 2a54 6865 2076  ata.```..**The v
-00012690: 6973 7561 6c69 7a61 7469 6f6e 206f 6620  isualization of 
-000126a0: 7379 6e74 656e 792a 2a0a 6060 600a 7767  synteny**.```.wg
-000126b0: 6420 7669 7a20 2d61 7020 6170 6461 7461  d viz -ap apdata
-000126c0: 202d 736d 2073 6d64 6174 6120 2d6d 7420   -sm smdata -mt 
-000126d0: 6d74 6461 7461 202d 6774 2067 7464 6174  mtdata -gt gtdat
-000126e0: 6120 2d2d 706c 6f74 7379 6e20 282d 2d6d  a --plotsyn (--m
-000126f0: 696e 6c65 6e20 2d31 202d 2d6d 696e 7365  inlen -1 --minse
-00012700: 676c 656e 2031 3030 3030 202d 2d6d 696e  glen 10000 --min
-00012710: 6765 6e65 6e75 6d20 3330 290a 6060 600a  genenum 30).```.
-00012720: 0a43 6f6d 7061 7265 6420 746f 2074 6865  .Compared to the
-00012730: 206f 7269 6769 6e61 6c20 6077 6764 602c   original `wgd`,
-00012740: 2074 6865 2060 7767 6420 7669 7a60 2070   the `wgd viz` p
-00012750: 726f 6772 616d 2061 6464 6564 2073 796e  rogram added syn
-00012760: 7465 6e79 2076 6973 7561 6c69 7a61 7469  teny visualizati
-00012770: 6f6e 2070 6970 656c 696e 652e 2055 7365  on pipeline. Use
-00012780: 7273 206e 6565 6420 746f 2070 726f 7669  rs need to provi
-00012790: 6465 2074 6865 2066 6c61 6720 6f70 7469  de the flag opti
-000127a0: 6f6e 2060 2d2d 706c 6f74 7379 6e60 2074  on `--plotsyn` t
-000127b0: 6f20 696e 6974 6961 7465 2074 6869 7320  o initiate this 
-000127c0: 7061 7274 206f 6620 7069 7065 6c69 6e65  part of pipeline
-000127d0: 2e20 5468 6973 2073 7465 7020 6173 7375  . This step assu
-000127e0: 6d65 7320 7468 6174 2075 7365 7273 2068  mes that users h
-000127f0: 6176 6520 6f62 7461 696e 6564 2061 6c72  ave obtained alr
-00012800: 6561 6479 2074 6865 2073 796e 7465 6e69  eady the synteni
-00012810: 6320 7265 7375 6c74 7320 6672 6f6d 2060  c results from `
-00012820: 692d 6164 686f 7265 6020 616e 6420 7573  i-adhore` and us
-00012830: 6573 2074 686f 7365 2072 6573 756c 7420  es those result 
-00012840: 6669 6c65 7320 746f 2072 6561 6c69 7a65  files to realize
-00012850: 2074 6865 2073 796e 7465 6e79 2076 6973   the synteny vis
-00012860: 7561 6c69 7a61 7469 6f6e 2e20 5369 6d6c  ualization. Siml
-00012870: 6961 7220 746f 2060 7767 6420 7379 6e60  iar to `wgd syn`
-00012880: 2c20 7468 6520 6578 7472 6120 2a4b 2a3c  , the extra *K*<
-00012890: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
-000128a0: 2066 696c 6520 6361 6e20 6265 2074 7261   file can be tra
-000128b0: 6e73 6d69 7474 6564 2076 6961 2074 6865  nsmitted via the
-000128c0: 206f 7074 696f 6e20 602d 2d64 6174 6166   option `--dataf
-000128d0: 696c 6560 2028 696e 7374 6561 6420 6f66  ile` (instead of
-000128e0: 2060 2d2d 6578 7472 6170 6172 616e 6f6d   `--extraparanom
-000128f0: 656b 7360 206f 7074 696f 6e29 2e20 4261  eks` option). Ba
-00012900: 7369 6361 6c6c 792c 2074 6865 2073 796e  sically, the syn
-00012910: 7465 6e69 6320 7265 7375 6c74 2066 696c  tenic result fil
-00012920: 6573 2072 6571 7569 7265 6420 6172 6520  es required are 
-00012930: 7468 6520 616e 6368 6f72 2070 6f69 6e74  the anchor point
-00012940: 7320 6461 7461 6669 6c65 2c20 6d75 6c74  s datafile, mult
-00012950: 6970 6c69 636f 6e73 2064 6174 6166 696c  iplicons datafil
-00012960: 652c 2067 656e 6520 7461 626c 6520 6461  e, gene table da
-00012970: 7461 6669 6c65 2028 6175 746f 6d61 7465  tafile (automate
-00012980: 6c79 2070 726f 6475 6365 6420 6279 2060  ly produced by `
-00012990: 7767 6420 7379 6e60 292c 2061 6e64 2073  wgd syn`), and s
-000129a0: 6567 6d65 6e74 7320 6461 7461 6669 6c65  egments datafile
-000129b0: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
-000129c0: 2073 7461 7274 696e 6720 7275 6e20 6361   starting run ca
-000129d0: 6e20 7573 6520 636f 6d6d 616e 6420 7369  n use command si
-000129e0: 6d70 6c79 2061 7320 6265 6c6f 772a 2a0a  mply as below**.
-000129f0: 0a60 6060 0a77 6764 2076 697a 202d 6170  .```.wgd viz -ap
-00012a00: 2061 7064 6174 6120 2d73 6d20 736d 6461   apdata -sm smda
-00012a10: 7461 202d 6d74 206d 7464 6174 6120 2d67  ta -mt mtdata -g
-00012a20: 7420 6774 6461 7461 202d 2d70 6c6f 7473  t gtdata --plots
-00012a30: 796e 0a60 6060 0a0a 2323 2049 6c6c 7573  yn.```..## Illus
-00012a40: 7472 6174 696f 6e0a 0a57 6520 696c 6c75  tration..We illu
-00012a50: 7374 7261 7465 206f 7572 2070 726f 6772  strate our progr
-00012a60: 616d 206f 6e20 616e 2065 7865 6d70 6c61  am on an exempla
-00012a70: 7279 2057 4744 2069 6e66 6572 656e 6365  ry WGD inference
-00012a80: 2061 6e64 2064 6174 696e 6720 7570 6f6e   and dating upon
-00012a90: 2073 7065 6369 6573 202a 4171 7569 6c65   species *Aquile
-00012aa0: 6769 6120 636f 6572 756c 6561 2a2e 0a0a  gia coerulea*...
-00012ab0: 5468 6520 2a41 7175 696c 6567 6961 2063  The *Aquilegia c
-00012ac0: 6f65 7275 6c65 612a 2077 6173 2072 6570  oerulea* was rep
-00012ad0: 6f72 7465 6420 746f 2065 7870 6572 6965  orted to experie
-00012ae0: 6e63 6520 616e 2070 616c 656f 2d70 6f6c  nce an paleo-pol
-00012af0: 7970 6c6f 6964 697a 6174 696f 6e20 6576  yploidization ev
-00012b00: 656e 7420 6166 7465 7220 7468 6520 6469  ent after the di
-00012b10: 7665 7267 656e 6365 206f 6620 636f 7265  vergence of core
-00012b20: 2065 7564 6963 6f74 732c 2077 6869 6368   eudicots, which
-00012b30: 2069 7320 6c69 6b65 6c79 2073 6861 7265   is likely share
-00012b40: 6420 6279 2061 6c6c 2052 616e 756e 6375  d by all Ranuncu
-00012b50: 6c61 6c65 732e 0a0a 4669 7273 7420 6162  lales...First ab
-00012b60: 6f76 6520 616c 6c2c 206c 6574 2773 2064  ove all, let's d
-00012b70: 656c 696e 6561 7465 2074 6865 2077 686f  elineate the who
-00012b80: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
-00012b90: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-00012ba0: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
-00012bb0: 2068 6176 6520 6120 6261 7369 6320 6f62   have a basic ob
-00012bc0: 7365 7276 6174 696f 6e20 666f 7220 706f  servation for po
-00012bd0: 7465 6e74 6961 6c6c 7920 636f 6e63 6569  tentially concei
-00012be0: 7661 626c 6520 5747 4473 2c20 7573 696e  vable WGDs, usin
-00012bf0: 6720 7468 6520 636f 6d6d 616e 6420 6c69  g the command li
-00012c00: 6e65 2062 656c 6f77 2e0a 0a60 6060 0a77  ne below...```.w
-00012c10: 6764 2064 6d64 2041 7175 696c 6567 6961  gd dmd Aquilegia
-00012c20: 5f63 6f65 7275 6c65 610a 7767 6420 6b73  _coerulea.wgd ks
-00012c30: 6420 7767 645f 646d 642f 4171 7569 6c65  d wgd_dmd/Aquile
-00012c40: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
-00012c50: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00012c60: 6c65 610a 6060 600a 0a54 6865 2063 6f6e  lea.```..The con
-00012c70: 7374 7275 6374 6564 2077 686f 6c65 2070  structed whole p
-00012c80: 6172 616e 6f6d 6520 2a4b 2a3c 7375 623e  aranome *K*<sub>
-00012c90: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
-00012ca0: 7269 6275 7469 6f6e 206f 6620 2a41 7175  ribution of *Aqu
-00012cb0: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
-00012cc0: 2069 7320 6173 2062 656c 6f77 2c20 7765   is as below, we
-00012cd0: 2063 616e 2073 6565 2074 6861 7420 7468   can see that th
-00012ce0: 6572 6520 7365 656d 7320 746f 2062 6520  ere seems to be 
-00012cf0: 6120 6875 6d70 2061 7420 2a4b 2a3c 7375  a hump at *K*<su
-00012d00: 623e 533c 2f73 7562 3e20 3120 6275 7420  b>S</sub> 1 but 
-00012d10: 6e6f 7420 636c 6561 722e 0a0a 215b 5d28  not clear...![](
-00012d20: 6461 7461 2f41 7175 696c 6567 6961 5f63  data/Aquilegia_c
-00012d30: 6f65 7275 6c65 612e 7473 762e 6b73 645f  oerulea.tsv.ksd_
-00012d40: 7770 2e73 7667 290a 0a57 6520 7468 656e  wp.svg)..We then
-00012d50: 2063 6f6e 7374 7275 6374 2074 6865 2061   construct the a
-00012d60: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
-00012d70: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
-00012d80: 6275 7469 6f6e 2075 7369 6e67 2074 6865  bution using the
-00012d90: 2063 6f6d 6d61 6e64 206c 696e 6520 6265   command line be
-00012da0: 6c6f 772e 0a0a 6060 600a 7767 6420 7379  low...```.wgd sy
-00012db0: 6e20 2d66 206d 524e 4120 2d61 204e 616d  n -f mRNA -a Nam
-00012dc0: 6520 7767 645f 646d 642f 4171 7569 6c65  e wgd_dmd/Aquile
-00012dd0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
-00012de0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
-00012df0: 6c65 612e 6766 6633 202d 6b73 2077 6764  lea.gff3 -ks wgd
-00012e00: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
-00012e10: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
-00012e20: 7376 0a60 6060 0a0a 4173 2073 686f 776e  sv.```..As shown
-00012e30: 2062 656c 6f77 2c20 7468 6572 6520 6172   below, there ar
-00012e40: 6520 736f 6d65 2072 6574 6169 6e65 6420  e some retained 
-00012e50: 616e 6368 6f72 2070 6169 7273 2077 6974  anchor pairs wit
-00012e60: 6820 2a4b 2a3c 7375 623e 533c 2f73 7562  h *K*<sub>S</sub
-00012e70: 3e20 6265 7477 6565 6e20 3120 616e 6420  > between 1 and 
-00012e80: 322c 2077 6869 6368 2073 6565 6d73 2074  2, which seems t
-00012e90: 6f20 7375 6767 6573 7420 6120 5747 4420  o suggest a WGD 
-00012ea0: 6576 656e 742e 0a0a 215b 5d28 6461 7461  event...![](data
-00012eb0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
-00012ec0: 6c65 612e 7473 762e 6b73 645f 7770 5f61  lea.tsv.ksd_wp_a
-00012ed0: 702e 7376 6729 0a0a 5468 6520 6173 736f  p.svg)..The asso
-00012ee0: 6369 6174 6564 2060 6475 7053 7461 636b  ciated `dupStack
-00012ef0: 6020 706c 6f74 2073 686f 7773 2074 6861  ` plot shows tha
-00012f00: 7420 7468 6572 6520 6172 6520 6e75 6d65  t there are nume
-00012f10: 726f 7573 2064 7570 6c69 6361 7465 6420  rous duplicated 
-00012f20: 7365 676d 656e 7473 2061 6372 6f73 7320  segments across 
-00012f30: 6d6f 7374 206f 6620 7468 6520 6368 726f  most of the chro
-00012f40: 6d6f 736f 6d65 732e 0a0a 215b 5d28 6461  mosomes...![](da
-00012f50: 7461 2f73 796e 5f72 6573 756c 7473 2f41  ta/syn_results/A
-00012f60: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00012f70: 615f 4171 7569 6c65 6769 615f 636f 6572  a_Aquilegia_coer
-00012f80: 756c 6561 5f6d 756c 7469 706c 6963 6f6e  ulea_multiplicon
-00012f90: 735f 6c65 7665 6c2e 706e 6729 0a0a 5765  s_level.png)..We
-00012fa0: 2069 6d70 6c65 6d65 6e74 6564 2074 776f   implemented two
-00012fb0: 2074 7970 6573 206f 6620 646f 7420 706c   types of dot pl
-00012fc0: 6f74 7320 696e 206f 7866 6f72 6420 6772  ots in oxford gr
-00012fd0: 6964 3a20 6f6e 6520 696e 2074 6865 2075  id: one in the u
-00012fe0: 6e69 7420 6f66 2062 6173 6573 2061 6e64  nit of bases and
-00012ff0: 2074 6865 206f 7468 6572 2069 6e20 7468   the other in th
-00013000: 6520 756e 6974 206f 6620 6765 6e65 732c  e unit of genes,
-00013010: 2077 6869 6368 2063 616e 2062 6520 636f   which can be co
-00013020: 6c6f 7265 6420 6279 202a 4b2a 3c73 7562  lored by *K*<sub
-00013030: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
-00013040: 6769 7665 6e20 2a4b 2a3c 7375 623e 533c  given *K*<sub>S<
-00013050: 2f73 7562 3e20 6461 7461 2e0a 0a21 5b5d  /sub> data...![]
-00013060: 2864 6174 612f 7379 6e5f 7265 7375 6c74  (data/syn_result
-00013070: 732f 4171 7569 6c65 6769 615f 636f 6572  s/Aquilegia_coer
-00013080: 756c 6561 2d76 732d 4171 7569 6c65 6769  ulea-vs-Aquilegi
-00013090: 615f 636f 6572 756c 6561 5f4b 732e 646f  a_coerulea_Ks.do
-000130a0: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
-000130b0: 0a0a 4173 2073 686f 776e 2061 626f 7665  ..As shown above
-000130c0: 2c20 7468 6520 646f 7420 706c 6f74 2069  , the dot plot i
-000130d0: 6e20 7468 6520 756e 6974 206f 6620 6765  n the unit of ge
-000130e0: 6e65 7320 7072 6573 656e 7473 206e 756d  nes presents num
-000130f0: 6572 6f75 7320 6465 6e73 656c 7920 6167  erous densely ag
-00013100: 6772 6567 6174 6564 2028 6c69 6e65 2d6c  gregated (line-l
-00013110: 696b 6529 2061 6e63 686f 7220 706f 696e  ike) anchor poin
-00013120: 7473 2061 7420 6d6f 7374 206f 6620 7468  ts at most of th
-00013130: 6520 6368 726f 6d6f 736f 6d65 7320 7769  e chromosomes wi
-00013140: 7468 2063 6f6e 7369 7374 656e 7420 2a4b  th consistent *K
-00013150: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-00013160: 6520 6265 7477 6565 6e20 3120 616e 6420  e between 1 and 
-00013170: 322e 2054 6865 2064 6f74 2070 6c6f 7420  2. The dot plot 
-00013180: 696e 2074 6865 2075 6e69 7420 6f66 2062  in the unit of b
-00013190: 6173 6573 2073 686f 7773 2074 6865 2073  ases shows the s
-000131a0: 616d 6520 7061 7474 6572 6e2c 2061 7320  ame pattern, as 
-000131b0: 6d61 6e69 6665 7374 6564 2062 656c 6f77  manifested below
-000131c0: 2e0a 0a21 5b5d 2864 6174 612f 7379 6e5f  ...![](data/syn_
-000131d0: 7265 7375 6c74 732f 4171 7569 6c65 6769  results/Aquilegi
-000131e0: 615f 636f 6572 756c 6561 2d76 732d 4171  a_coerulea-vs-Aq
-000131f0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00013200: 5f4b 732e 646f 742e 706e 6729 0a0a 5468  _Ks.dot.png)..Th
-00013210: 6520 646f 7420 706c 6f74 7320 7769 7468  e dot plots with
-00013220: 6f75 7420 2a4b 2a3c 7375 623e 533c 2f73  out *K*<sub>S</s
-00013230: 7562 3e20 616e 6e6f 7461 7469 6f6e 2077  ub> annotation w
-00013240: 696c 6c20 616c 736f 2062 6520 6175 746f  ill also be auto
-00013250: 6d61 7465 6c79 2070 726f 6475 6365 642c  mately produced,
-00013260: 2061 7320 7368 6f77 6e20 6265 6c6f 772e   as shown below.
-00013270: 0a0a 215b 5d28 6461 7461 2f73 796e 5f72  ..![](data/syn_r
-00013280: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
-00013290: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
-000132a0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-000132b0: 646f 745f 756e 6974 5f67 656e 652e 706e  dot_unit_gene.pn
-000132c0: 6729 0a0a 215b 5d28 6461 7461 2f73 796e  g)..![](data/syn
-000132d0: 5f72 6573 756c 7473 2f41 7175 696c 6567  _results/Aquileg
-000132e0: 6961 5f63 6f65 7275 6c65 612d 7673 2d41  ia_coerulea-vs-A
-000132f0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00013300: 612e 646f 742e 706e 6729 0a0a 4e6f 7465  a.dot.png)..Note
-00013310: 2074 6861 7420 7468 6520 6f70 6163 6974   that the opacit
-00013320: 7920 6f66 2061 6e63 686f 7220 646f 7473  y of anchor dots
-00013330: 2061 6e64 2061 6c6c 2068 6f6d 6f6c 6f67   and all homolog
-00013340: 2064 6f74 7320 6361 6e20 6265 2073 6574   dots can be set
-00013350: 2062 7920 7468 6520 6f70 7469 6f6e 2060   by the option `
-00013360: 2d2d 6170 616c 7068 6160 2061 6e64 2060  --apalpha` and `
-00013370: 2d2d 686f 616c 7068 6160 2073 6570 6172  --hoalpha` separ
-00013380: 6174 656c 792e 2049 6620 6f6e 6520 6a75  ately. If one ju
-00013390: 7374 2077 616e 7473 2074 6f20 7365 6520  st wants to see 
-000133a0: 7468 6520 616e 6368 6f72 2064 6f74 732c  the anchor dots,
-000133b0: 2073 6574 7469 6e67 2074 6865 2060 686f   setting the `ho
-000133c0: 616c 7068 6160 2061 7320 3020 286f 7220  alpha` as 0 (or 
-000133d0: 6f74 6865 7220 6d69 6e75 7363 756c 6520  other minuscule 
-000133e0: 7661 6c75 6573 2920 7769 6c6c 2064 6f2e  values) will do.
-000133f0: 2049 6620 6f6e 6520 7761 6e74 7320 746f   If one wants to
-00013400: 2073 6565 2074 6865 2064 6973 7472 6962   see the distrib
-00013410: 7574 696f 6e20 6f66 2077 686f 6c65 2064  ution of whole d
-00013420: 6f74 7320 6265 7474 6572 2c20 7365 7474  ots better, sett
-00013430: 696e 6720 7468 6520 6068 6f61 6c70 6861  ing the `hoalpha
-00013440: 6020 6869 6768 6572 2028 616e 6420 6061  ` higher (and `a
-00013450: 7061 6c70 6861 6020 6c6f 7765 7229 2077  palpha` lower) w
-00013460: 696c 6c20 646f 2e20 5468 6520 6064 6f74  ill do. The `dot
-00013470: 7369 7a65 6020 6f70 7469 6f6e 2063 616e  size` option can
-00013480: 2062 6520 6361 6c6c 6564 2074 6f20 6164   be called to ad
-00013490: 6a75 7374 2074 6865 2073 697a 6520 6f66  just the size of
-000134a0: 2064 6f74 732e 0a0a 4120 6675 7274 6865   dots...A furthe
-000134b0: 7220 6173 736f 6369 6174 6564 2053 796e  r associated Syn
-000134c0: 6465 7074 6820 706c 6f74 2073 686f 7773  depth plot shows
-000134d0: 2074 6861 7420 7468 6572 6520 6172 6520   that there are 
-000134e0: 6d6f 7265 2074 6861 6e20 3530 2064 7570  more than 50 dup
-000134f0: 6c69 6361 7465 6420 7365 676d 656e 7473  licated segments
-00013500: 206c 6f6e 6765 7220 7468 616e 2031 3030   longer than 100
-00013510: 3030 2062 7020 616e 6420 3330 2067 656e  00 bp and 30 gen
-00013520: 6573 2028 736f 2061 7320 746f 2064 726f  es (so as to dro
-00013530: 7020 6672 6167 6d65 6e74 6172 7920 7365  p fragmentary se
-00013540: 676d 656e 7473 292c 2077 6869 6368 2064  gments), which d
-00013550: 6f6d 696e 6174 6573 2074 6865 2077 686f  ominates the who
-00013560: 6c65 2063 6f6c 6c69 6e65 6172 2072 6174  le collinear rat
-00013570: 696f 2063 6174 6567 6f72 792e 0a0a 215b  io category...![
-00013580: 5d28 6461 7461 2f73 796e 5f72 6573 756c  ](data/syn_resul
-00013590: 7473 2f53 796e 6465 7074 682e 7376 6729  ts/Syndepth.svg)
-000135a0: 0a0a 4d6f 7265 2065 7871 7569 7369 7465  ..More exquisite
-000135b0: 2070 6c6f 7473 2069 6e63 6c75 6469 6e67   plots including
-000135c0: 2062 6f74 6820 696e 7472 612d 7370 6563   both intra-spec
-000135d0: 6966 6963 2061 6e64 2069 6e74 6572 2d73  ific and inter-s
-000135e0: 7065 6369 6669 6320 636f 6d70 6172 6973  pecific comparis
-000135f0: 6f6e 7320 7573 696e 6720 7468 6520 6f72  ons using the or
-00013600: 7468 6f67 726f 7570 7320 2863 6f6d 706f  thogroups (compo
-00013610: 7365 6420 6f66 202a 4171 7569 6c65 6769  sed of *Aquilegi
-00013620: 6120 636f 6572 756c 6561 2a2c 202a 5072  a coerulea*, *Pr
-00013630: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
-00013640: 2c20 2a41 636f 7275 7320 616d 6572 6963  , *Acorus americ
-00013650: 616e 7573 2a20 616e 6420 2a56 6974 6973  anus* and *Vitis
-00013660: 2076 696e 6966 6572 612a 2c20 7365 6520   vinifera*, see 
-00013670: 6675 7274 6865 7220 666f 7220 7468 6520  further for the 
-00013680: 636f 6e74 6578 7429 2069 6e66 6572 7265  context) inferre
-00013690: 6420 6865 7265 696e 6166 7465 7220 6361  d hereinafter ca
-000136a0: 6e20 6265 2061 6c73 6f20 7072 6f64 7563  n be also produc
-000136b0: 6564 2075 7369 6e67 2060 7767 6420 7379  ed using `wgd sy
-000136c0: 6e60 2e20 4e6f 7465 2074 6861 7420 6469  n`. Note that di
-000136d0: 6666 6572 656e 7420 6765 6e6f 6d65 2061  fferent genome a
-000136e0: 7373 656d 626c 6965 7320 6d69 6768 7420  ssemblies might 
-000136f0: 6861 7665 2064 6966 6665 7265 6e74 2066  have different f
-00013700: 6561 7475 7265 7320 616e 6420 6174 7472  eatures and attr
-00013710: 6962 7574 6573 2077 6869 6368 2063 616e  ibutes which can
-00013720: 2062 6520 6163 636f 6d6d 6f64 6174 6564   be accommodated
-00013730: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
-00013740: 602d 2d61 6464 6974 696f 6e61 6c67 6666  `--additionalgff
-00013750: 696e 666f 6020 666f 7220 6561 6368 2067  info` for each g
-00013760: 656e 6f6d 6520 6173 7365 6d62 6c79 2077  enome assembly w
-00013770: 686f 7365 206f 7264 6572 206e 6565 6473  hose order needs
-00013780: 2074 6f20 666f 6c6c 6f77 2074 6865 206f   to follow the o
-00013790: 7264 6572 206f 6620 6766 6633 2066 696c  rder of gff3 fil
-000137a0: 6573 2c20 666f 7220 696e 7374 616e 6365  es, for instance
-000137b0: 2027 6d4e 5241 3b4e 616d 6527 2066 6f72   'mNRA;Name' for
-000137c0: 2060 4171 7569 6c65 6769 615f 636f 6572   `Aquilegia_coer
-000137d0: 756c 6561 2e67 6666 3360 2c20 276d 4e52  ulea.gff3`, 'mNR
-000137e0: 413b 4944 2720 666f 7220 6050 726f 7465  A;ID' for `Prote
-000137f0: 615f 6379 6e61 726f 6964 6573 2e67 6666  a_cynaroides.gff
-00013800: 3360 2c20 276d 4e52 413b 4e61 6d65 2720  3`, 'mNRA;Name' 
-00013810: 666f 7220 6041 636f 7275 735f 616d 6572  for `Acorus_amer
-00013820: 6963 616e 7573 2e67 6666 3360 2061 6e64  icanus.gff3` and
-00013830: 2027 6d4e 5241 3b4e 616d 6527 2066 6f72   'mNRA;Name' for
-00013840: 2060 5669 7469 735f 7669 6e69 6665 7261   `Vitis_vinifera
-00013850: 2e67 6666 3360 2e0a 0a60 6060 0a77 6764  .gff3`...```.wgd
-00013860: 2073 796e 2077 6764 5f6f 7274 686f 2f4f   syn wgd_ortho/O
-00013870: 7274 686f 6772 6f75 7073 2e73 702e 7473  rthogroups.sp.ts
-00013880: 7620 2d6b 7320 7767 645f 6f72 7468 6f5f  v -ks wgd_ortho_
-00013890: 6b73 2f4f 7274 686f 6772 6f75 7073 2e73  ks/Orthogroups.s
-000138a0: 702e 7473 762e 6b73 2e74 7376 2041 7175  p.tsv.ks.tsv Aqu
-000138b0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-000138c0: 6766 6633 202d 2d61 6464 6974 696f 6e61  gff3 --additiona
-000138d0: 6c67 6666 696e 666f 2027 6d4e 5241 3b4e  lgffinfo 'mNRA;N
-000138e0: 616d 6527 2050 726f 7465 615f 6379 6e61  ame' Protea_cyna
-000138f0: 726f 6964 6573 2e67 6666 3320 2d2d 6164  roides.gff3 --ad
-00013900: 6469 7469 6f6e 616c 6766 6669 6e66 6f20  ditionalgffinfo 
-00013910: 276d 4e52 413b 4944 2720 4163 6f72 7573  'mNRA;ID' Acorus
-00013920: 5f61 6d65 7269 6361 6e75 732e 6766 6633  _americanus.gff3
-00013930: 202d 2d61 6464 6974 696f 6e61 6c67 6666   --additionalgff
-00013940: 696e 666f 2027 6d4e 5241 3b4e 616d 6527  info 'mNRA;Name'
-00013950: 2056 6974 6973 5f76 696e 6966 6572 612e   Vitis_vinifera.
-00013960: 6766 6633 202d 2d61 6464 6974 696f 6e61  gff3 --additiona
-00013970: 6c67 6666 696e 666f 2027 6d4e 5241 3b4e  lgffinfo 'mNRA;N
-00013980: 616d 6527 202d 6f20 7767 645f 6f72 7468  ame' -o wgd_orth
-00013990: 6f5f 7379 6e0a 6060 600a 0a55 706f 6e20  o_syn.```..Upon 
-000139a0: 7468 6520 6163 7175 6973 6974 696f 6e20  the acquisition 
-000139b0: 6f66 2074 6865 2063 6f6c 6c69 6e65 6172  of the collinear
-000139c0: 2072 6573 756c 7473 2075 7369 6e67 2060   results using `
-000139d0: 7767 6420 7379 6e60 2c20 7468 6520 7361  wgd syn`, the sa
-000139e0: 6d65 2063 6f6c 6c69 6e65 6172 2070 6c6f  me collinear plo
-000139f0: 7473 2063 616e 2062 6520 616c 736f 2070  ts can be also p
-00013a00: 726f 6475 6365 6420 6279 2060 7767 6420  roduced by `wgd 
-00013a10: 7669 7a60 2075 7369 6e67 2074 6865 2063  viz` using the c
-00013a20: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 6060  ommand below..``
-00013a30: 600a 7767 6420 7669 7a20 2d2d 706c 6f74  `.wgd viz --plot
-00013a40: 7379 6e20 2d73 6d20 7767 645f 6f72 7468  syn -sm wgd_orth
-00013a50: 6f5f 7379 6e2f 6961 6468 6f72 652d 6f75  o_syn/iadhore-ou
-00013a60: 742f 7365 676d 656e 7473 2e74 7874 202d  t/segments.txt -
-00013a70: 6170 2077 6764 5f6f 7274 686f 5f73 796e  ap wgd_ortho_syn
-00013a80: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
-00013a90: 686f 7270 6f69 6e74 732e 7478 7420 2d6d  horpoints.txt -m
-00013aa0: 7420 7767 645f 6f72 7468 6f5f 7379 6e2f  t wgd_ortho_syn/
-00013ab0: 6961 6468 6f72 652d 6f75 742f 6d75 6c74  iadhore-out/mult
-00013ac0: 6970 6c69 636f 6e73 2e74 7874 202d 6774  iplicons.txt -gt
-00013ad0: 2077 6764 5f6f 7274 686f 5f73 796e 2f67   wgd_ortho_syn/g
-00013ae0: 656e 652d 7461 626c 652e 6373 7620 2d64  ene-table.csv -d
-00013af0: 2077 6764 5f6f 7274 686f 5f6b 732f 4f72   wgd_ortho_ks/Or
-00013b00: 7468 6f67 726f 7570 732e 7370 2e74 7376  thogroups.sp.tsv
-00013b10: 2e6b 732e 7473 7620 2d6f 2077 6764 5f6f  .ks.tsv -o wgd_o
-00013b20: 7274 686f 5f76 697a 0a60 6060 0a0a 215b  rtho_viz.```..![
-00013b30: 5d28 6461 7461 2f6f 7274 686f 5f73 796e  ](data/ortho_syn
-00013b40: 5f72 6573 756c 7473 2f41 7175 696c 6567  _results/Aquileg
-00013b50: 6961 5f63 6f65 7275 6c65 615f 5669 7469  ia_coerulea_Viti
-00013b60: 735f 7669 6e69 6665 7261 5f6d 756c 7469  s_vinifera_multi
-00013b70: 706c 6963 6f6e 735f 6c65 7665 6c2e 706e  plicons_level.pn
-00013b80: 6729 0a54 6865 2061 626f 7665 2060 6475  g).The above `du
-00013b90: 7053 7461 636b 6020 706c 6f74 2073 686f  pStack` plot sho
-00013ba0: 7773 2074 6865 2064 6973 7472 6962 7574  ws the distribut
-00013bb0: 696f 6e20 6f66 2064 7570 6c69 6361 7465  ion of duplicate
-00013bc0: 6420 7365 676d 656e 7473 206f 6620 2a41  d segments of *A
-00013bd0: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
-00013be0: 612a 2063 6f6d 7061 7265 6420 746f 2069  a* compared to i
-00013bf0: 7473 656c 6620 2869 6e20 6772 6565 6e29  tself (in green)
-00013c00: 2061 6e64 2063 6f6d 7061 7265 6420 746f   and compared to
-00013c10: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
-00013c20: 2a20 2869 6e20 626c 7565 2920 6f76 6572  * (in blue) over
-00013c30: 2074 6865 2063 6872 6f6d 6f73 6f6d 6573   the chromosomes
-00013c40: 206f 6620 2a41 2e20 636f 6572 756c 6561   of *A. coerulea
-00013c50: 2a2e 0a21 5b5d 2864 6174 612f 6f72 7468  *..![](data/orth
-00013c60: 6f5f 7379 6e5f 7265 7375 6c74 732f 4f76  o_syn_results/Ov
-00013c70: 6572 616c 6c73 7065 6369 6573 5f4b 732e  erallspecies_Ks.
-00013c80: 646f 745f 756e 6974 5f67 656e 652e 706e  dot_unit_gene.pn
-00013c90: 6729 0a54 6865 2061 626f 7665 202a 4b2a  g).The above *K*
-00013ca0: 3c73 7562 3e53 3c2f 7375 623e 2064 6f74  <sub>S</sub> dot
-00013cb0: 706c 6f74 2069 6e20 756e 6974 206f 6620  plot in unit of 
-00013cc0: 6765 6e65 2073 686f 7773 2074 6865 206f  gene shows the o
-00013cd0: 7665 7261 6c6c 2064 6973 7472 6962 7574  verall distribut
-00013ce0: 696f 6e20 6f66 2063 6f6c 6c69 6e65 6172  ion of collinear
-00013cf0: 6974 7920 6163 6f72 7373 2074 6865 2066  ity acorss the f
-00013d00: 6f75 7220 7370 6563 6965 7320 696e 766f  our species invo
-00013d10: 6c76 6564 2e0a 215b 5d28 6461 7461 2f6f  lved..![](data/o
-00013d20: 7274 686f 5f73 796e 5f72 6573 756c 7473  rtho_syn_results
-00013d30: 2f4f 7665 7261 6c6c 7370 6563 6965 732e  /Overallspecies.
-00013d40: 646f 745f 756e 6974 5f67 656e 652e 706e  dot_unit_gene.pn
-00013d50: 6729 0a54 6865 2061 626f 7665 2064 6f74  g).The above dot
-00013d60: 706c 6f74 2069 7320 7769 7468 6f75 7420  plot is without 
-00013d70: 7468 6520 616e 6e6f 7461 7469 6f6e 206f  the annotation o
-00013d80: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
-00013d90: 3e20 6167 6573 2063 6f6d 7061 7265 6420  > ages compared 
-00013da0: 746f 2074 6865 206c 6173 7420 6f6e 652e  to the last one.
-00013db0: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
-00013dc0: 7379 6e5f 7265 7375 6c74 732f 5379 6e64  syn_results/Synd
-00013dd0: 6570 7468 2e73 7667 290a 5468 6520 6162  epth.svg).The ab
-00013de0: 6f76 6520 5379 6e64 6570 7468 2070 6c6f  ove Syndepth plo
-00013df0: 7420 7368 6f77 7320 7468 6520 636f 6c6c  t shows the coll
-00013e00: 696e 6561 7220 7261 7469 6f20 6163 6f72  inear ratio acor
-00013e10: 7373 2061 6c6c 2073 7065 6369 6573 2070  ss all species p
-00013e20: 6169 7273 2028 696e 7472 612d 7370 6563  airs (intra-spec
-00013e30: 6966 6963 2063 6f6d 7061 7269 736f 6e20  ific comparison 
-00013e40: 696e 2067 7265 656e 2077 6869 6c65 2069  in green while i
-00013e50: 6e74 6572 2d73 7065 6369 6669 6320 636f  nter-specific co
-00013e60: 6d70 6172 6973 6f6e 2069 6e20 626c 7565  mparison in blue
-00013e70: 292e 0a0a 5765 2063 616e 2066 6974 2061  )...We can fit a
-00013e80: 6e20 454c 4d4d 206d 6978 7475 7265 206d  n ELMM mixture m
-00013e90: 6f64 656c 2075 706f 6e20 7468 6520 7768  odel upon the wh
-00013ea0: 6f6c 6520 7061 7261 6e6f 6d65 202a 4b2a  ole paranome *K*
-00013eb0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-00013ec0: 2064 6973 7472 6962 7574 696f 6e20 746f   distribution to
-00013ed0: 2073 6565 206d 6f72 6520 6163 6375 7261   see more accura
-00013ee0: 7465 6c79 2074 6865 2073 6967 6e69 6669  tely the signifi
-00013ef0: 6361 6e63 6520 616e 6420 6c6f 6361 7469  cance and locati
-00013f00: 6f6e 206f 6620 706f 7465 6e74 6961 6c20  on of potential 
-00013f10: 5747 4473 2c20 7573 696e 6720 7468 6520  WGDs, using the 
-00013f20: 636f 6d6d 616e 6420 6c69 6e65 2062 656c  command line bel
-00013f30: 6f77 2e0a 0a60 6060 0a77 6764 2076 697a  ow...```.wgd viz
-00013f40: 202d 6420 7767 645f 6b73 642f 4171 7569   -d wgd_ksd/Aqui
-00013f50: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
-00013f60: 7376 2e6b 732e 7473 760a 6060 600a 0a54  sv.ks.tsv.```..T
-00013f70: 6865 2072 6573 756c 7420 6f66 2045 4c4d  he result of ELM
-00013f80: 4d20 6d69 7874 7572 6520 6d6f 6465 6c20  M mixture model 
-00013f90: 636c 7573 7465 7269 6e67 2073 686f 7773  clustering shows
-00013fa0: 2074 6861 7420 7468 6572 6520 6973 2061   that there is a
-00013fb0: 206c 696b 656c 7920 5747 4420 636f 6d70   likely WGD comp
-00013fc0: 6f6e 656e 7420 6174 202a 4b2a 3c73 7562  onent at *K*<sub
-00013fd0: 3e53 3c2f 7375 623e 2031 2e31 392e 0a0a  >S</sub> 1.19...
-00013fe0: 215b 5d28 6461 7461 2f65 6c6d 6d5f 4171  ![](data/elmm_Aq
-00013ff0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00014000: 2e74 7376 2e6b 732e 7473 765f 6265 7374  .tsv.ks.tsv_best
-00014010: 5f6d 6f64 656c 735f 7765 6967 6874 6564  _models_weighted
-00014020: 2e73 7667 290a 0a4c 6574 2773 2064 6f20  .svg)..Let's do 
-00014030: 6120 6d69 7874 7572 6520 6d6f 6465 6c20  a mixture model 
-00014040: 636c 7573 7465 7269 6e67 2066 6f72 2061  clustering for a
-00014050: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
-00014060: 2f73 7562 3e20 746f 6f2c 2075 7369 6e67  /sub> too, using
-00014070: 2074 6865 2063 6f6d 6d61 6e64 206c 696e   the command lin
-00014080: 6520 6265 6c6f 772e 204e 6f74 6520 7468  e below. Note th
-00014090: 6174 2074 6869 7320 7374 6570 2077 696c  at this step wil
-000140a0: 6c20 6175 746f 6d61 7465 6c79 2063 616c  l automately cal
-000140b0: 6c20 7468 6520 7365 676d 656e 7420 2a4b  l the segment *K
-000140c0: 2a3c 7375 623e 533c 2f73 7562 3e20 636c  *<sub>S</sub> cl
-000140d0: 7573 7465 7269 6e67 2061 6e61 6c79 7369  ustering analysi
-000140e0: 7320 746f 6f2e 0a0a 6060 600a 7767 6420  s too...```.wgd 
-000140f0: 7065 616b 2077 6764 5f6b 7364 2f41 7175  peak wgd_ksd/Aqu
-00014100: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-00014110: 7473 762e 6b73 2e74 7376 202d 2d61 6e63  tsv.ks.tsv --anc
-00014120: 686f 7270 6f69 6e74 7320 7767 645f 7379  horpoints wgd_sy
-00014130: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
-00014140: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
-00014150: 2d73 6567 6d65 6e74 7320 7767 645f 7379  -segments wgd_sy
-00014160: 6e2f 6961 6468 6f72 652d 6f75 742f 7365  n/iadhore-out/se
-00014170: 676d 656e 7473 2e74 7874 202d 2d6c 6973  gments.txt --lis
-00014180: 7465 6c65 6d65 6e74 7320 7767 645f 7379  telements wgd_sy
-00014190: 6e2f 6961 6468 6f72 652d 6f75 742f 6c69  n/iadhore-out/li
-000141a0: 7374 5f65 6c65 6d65 6e74 732e 7478 7420  st_elements.txt 
-000141b0: 2d2d 6d75 6c74 6970 6c69 636f 6e70 6169  --multipliconpai
-000141c0: 7273 2077 6764 5f73 796e 2f69 6164 686f  rs wgd_syn/iadho
-000141d0: 7265 2d6f 7574 2f6d 756c 7469 706c 6963  re-out/multiplic
-000141e0: 6f6e 5f70 6169 7273 2e74 7874 2028 2d2d  on_pairs.txt (--
-000141f0: 7765 6967 6874 6564 290a 6060 600a 0a54  weighted).```..T
-00014200: 6865 2061 6e63 686f 7220 2a4b 2a3c 7375  he anchor *K*<su
-00014210: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00014220: 7374 7269 6275 7469 6f6e 2061 6c73 6f20  stribution also 
-00014230: 6861 7320 6120 6c69 6b65 6c79 2057 4744  has a likely WGD
-00014240: 2063 6f6d 706f 6e65 6e74 2077 6974 6820   component with 
-00014250: 6d6f 6465 2031 2e32 382e 0a0a 215b 5d28  mode 1.28...![](
-00014260: 6461 7461 2f4f 7269 6769 6e61 6c5f 416e  data/Original_An
-00014270: 6368 6f72 4b73 5f47 4d4d 5f43 6f6d 706f  chorKs_GMM_Compo
-00014280: 6e65 6e74 335f 6e6f 6465 5f61 7665 7261  nent3_node_avera
-00014290: 6765 645f 4c6f 676e 6f72 6d61 6c2e 7376  ged_Lognormal.sv
-000142a0: 6729 0a0a 4e6f 7720 7468 6174 2077 6520  g)..Now that we 
-000142b0: 6861 7665 2073 6565 6e20 7468 6520 6576  have seen the ev
-000142c0: 6964 656e 6365 206f 6620 6e75 6d65 726f  idence of numero
-000142d0: 7573 2064 7570 6c69 6361 7465 6420 7365  us duplicated se
-000142e0: 676d 656e 7473 2061 6e64 2074 6865 2061  gments and the a
-000142f0: 6767 7265 6761 7469 6f6e 206f 6620 6475  ggregation of du
-00014300: 706c 6963 6174 6573 2061 6765 2061 7420  plicates age at 
-00014310: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00014320: 312e 3238 206f 7220 312e 3139 2066 6f72  1.28 or 1.19 for
-00014330: 2061 6e63 686f 7220 7061 6972 7320 616e   anchor pairs an
-00014340: 6420 6e6f 6e2d 616e 6368 6f72 2070 6169  d non-anchor pai
-00014350: 7273 2074 6872 6f75 6768 6f75 7420 7468  rs throughout th
-00014360: 6520 7768 6f6c 6520 6765 6e6f 6d65 2e20  e whole genome. 
-00014370: 5765 2063 616e 2063 6c61 696d 2077 6974  We can claim wit
-00014380: 6820 736f 6d65 2063 6f6e 6669 6465 6e63  h some confidenc
-00014390: 6520 7468 6174 202a 4171 7569 6c65 6769  e that *Aquilegi
-000143a0: 6120 636f 6572 756c 6561 2a20 6d69 6768  a coerulea* migh
-000143b0: 7420 6861 7665 2065 7870 6572 6965 6e63  t have experienc
-000143c0: 6564 2061 2070 616c 656f 2d70 6f6c 7970  ed a paleo-polyp
-000143d0: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
-000143e0: 742e 204e 6578 742c 204c 6574 2773 2068  t. Next, Let's h
-000143f0: 6176 6520 6120 6675 7274 6865 7220 6c6f  ave a further lo
-00014400: 6f6b 2061 626f 7574 2069 7473 2070 6879  ok about its phy
-00014410: 6c6f 6765 6e65 7469 6320 6c6f 6361 7469  logenetic locati
-00014420: 6f6e 2e20 5765 206b 6e6f 7720 7468 6174  on. We know that
-00014430: 2074 6865 7265 2061 7265 2075 6e63 6572   there are uncer
-00014440: 7461 696e 7469 6573 2061 626f 7574 2077  tainties about w
-00014450: 6865 7468 6572 2074 6869 7320 7075 7461  hether this puta
-00014460: 7469 7665 2070 616c 656f 2d70 6f6c 7970  tive paleo-polyp
-00014470: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
-00014480: 7420 6973 2073 6861 7265 6420 7769 7468  t is shared with
-00014490: 2061 6c6c 2065 7564 6963 6f74 7320 6f72   all eudicots or
-000144a0: 206e 6f74 2e20 5765 2063 616e 2063 686f   not. We can cho
-000144b0: 6f73 6520 736f 6d65 206f 7468 6572 2065  ose some other e
-000144c0: 7564 6963 6f74 2067 656e 6f6d 6573 2074  udicot genomes t
-000144d0: 6f20 7365 6520 7468 6520 6f72 6465 7269  o see the orderi
-000144e0: 6e67 206f 6620 7370 6563 6961 7469 6f6e  ng of speciation
-000144f0: 2061 6e64 2070 6f6c 7970 6c6f 6964 697a   and polyploidiz
-00014500: 6174 696f 6e20 6576 656e 7473 2e20 4865  ation events. He
-00014510: 7265 2077 6520 6368 6f6f 7365 202a 5669  re we choose *Vi
-00014520: 7469 7320 7669 6e69 6665 7261 2a2c 202a  tis vinifera*, *
-00014530: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
-00014540: 732a 2061 6e64 202a 4163 6f72 7573 2061  s* and *Acorus a
-00014550: 6d65 7269 6361 6e75 732a 2069 6e20 7468  mericanus* in th
-00014560: 6520 666f 6c6c 6f77 696e 6720 2a4b 2a3c  e following *K*<
-00014570: 7375 623e 533c 2f73 7562 3e20 616e 616c  sub>S</sub> anal
-00014580: 7973 6973 2e20 4669 7273 742c 2077 6520  ysis. First, we 
-00014590: 6275 696c 7420 6120 676c 6f62 616c 204d  built a global M
-000145a0: 5242 4820 6661 6d69 6c79 2075 7369 6e67  RBH family using
-000145b0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-000145c0: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
-000145d0: 202d 2d67 6c6f 6261 6c6d 7262 6820 4171   --globalmrbh Aq
-000145e0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000145f0: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
-00014600: 6573 2041 636f 7275 735f 616d 6572 6963  es Acorus_americ
-00014610: 616e 7573 2056 6974 6973 5f76 696e 6966  anus Vitis_vinif
-00014620: 6572 6120 2d6f 2077 6764 5f67 6c6f 6261  era -o wgd_globa
-00014630: 6c6d 7262 680a 6060 600a 0a49 6e20 7468  lmrbh.```..In th
-00014640: 6520 676c 6f62 616c 204d 5242 4820 6661  e global MRBH fa
-00014650: 6d69 6c79 2c20 6576 6572 7920 7061 6972  mily, every pair
-00014660: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-00014670: 6765 6e65 7320 6973 2074 6865 2072 6563  genes is the rec
-00014680: 6970 726f 6361 6c20 6265 7374 2068 6974  iprocal best hit
-00014690: 2c20 7375 6767 6573 7469 6e67 2074 7275  , suggesting tru
-000146a0: 6520 6f72 7468 6f6c 6f67 6f75 7320 7265  e orthologous re
-000146b0: 6c61 7469 6f6e 7368 6970 732e 2057 6520  lationships. We 
-000146c0: 776f 756c 6420 7573 6520 7468 6520 2a4b  would use the *K
-000146d0: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-000146e0: 6c75 6573 2061 7373 6f63 6961 7465 6420  lues associated 
-000146f0: 7769 7468 2074 6865 7365 206f 7274 686f  with these ortho
-00014700: 6c6f 676f 7573 2070 6169 7273 2074 6f20  logous pairs to 
-00014710: 6465 6c69 6d69 7420 7468 6520 6469 7665  delimit the dive
-00014720: 7267 656e 6365 202a 4b2a 3c73 7562 3e53  rgence *K*<sub>S
-00014730: 3c2f 7375 623e 2070 6561 6b2e 2054 6f67  </sub> peak. Tog
-00014740: 6574 6865 7220 7769 7468 2074 6865 2077  ether with the w
-00014750: 686f 6c65 2070 6172 616e 6f6d 6520 2a4b  hole paranome *K
-00014760: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00014770: 7374 7269 6275 7469 6f6e 2c20 7765 2063  stribution, we c
-00014780: 6f6e 6475 6374 2074 6865 2072 6174 6520  onduct the rate 
-00014790: 636f 7272 6563 7469 6f6e 2075 7369 6e67  correction using
-000147a0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-000147b0: 6f77 2e0a 0a21 2153 696e 6365 2060 7767  ow...!!Since `wg
-000147c0: 6460 2076 6572 7369 6f6e 2032 2e30 2e32  d` version 2.0.2
-000147d0: 342c 2077 6520 7265 7772 6f74 6520 6120  4, we rewrote a 
-000147e0: 636c 6561 6e65 7220 616e 6420 7175 6963  cleaner and quic
-000147f0: 6b65 7220 7761 7920 6f66 2064 6f69 6e67  ker way of doing
-00014800: 2073 7562 7374 6974 7574 696f 6e20 7261   substitution ra
-00014810: 7465 2063 6f72 7265 6374 696f 6e2e 2049  te correction. I
-00014820: 7427 7320 6e6f 7420 7265 7175 6972 6564  t's not required
-00014830: 2074 6f20 7479 7065 2069 6e20 616e 7920   to type in any 
-00014840: 7370 6569 6365 7320 7061 6972 2061 6e64  speices pair and
-00014850: 2061 2073 6572 6965 7320 6f66 202a 4b2a   a series of *K*
-00014860: 3c73 7562 3e53 3c2f 7375 623e 2070 6c6f  <sub>S</sub> plo
-00014870: 7473 2077 696c 6c20 6265 2070 726f 6475  ts will be produ
-00014880: 6365 642e 2054 6865 2072 6571 7569 7265  ced. The require
-00014890: 6420 6669 6c65 7320 6172 6520 6f72 7468  d files are orth
-000148a0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-000148b0: 533c 2f73 7562 3e20 6461 7461 6669 6c65  S</sub> datafile
-000148c0: 2c20 7061 7261 6c6f 676f 7573 202a 4b2a  , paralogous *K*
-000148d0: 3c73 7562 3e53 3c2f 7375 623e 2064 6174  <sub>S</sub> dat
-000148e0: 6166 696c 652c 2061 2073 7065 6369 6573  afile, a species
-000148f0: 2074 7265 6520 616e 6420 6120 666f 6361   tree and a foca
-00014900: 6c20 7370 6563 6965 7320 2874 6865 206f  l species (the o
-00014910: 6e65 2069 6e70 7574 7465 6420 7769 7468  ne inputted with
-00014920: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
-00014930: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
-00014940: 292e 2055 7365 7273 2063 616e 2063 686f  ). Users can cho
-00014950: 6f73 6520 746f 2061 6464 206f 6e65 206d  ose to add one m
-00014960: 6f72 6520 6c61 7965 7220 6f66 2045 4c4d  ore layer of ELM
-00014970: 4d20 616e 616c 7973 6973 206f 6e20 7061  M analysis on pa
-00014980: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
-00014990: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
-000149a0: 616e 642f 6f72 2047 4d4d 2061 6e61 6c79  and/or GMM analy
-000149b0: 7369 7320 6f6e 2061 6e63 686f 7220 2a4b  sis on anchor *K
-000149c0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-000149d0: 7374 7269 6275 7469 6f6e 2e20 5468 6520  stribution. The 
-000149e0: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-000149f0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00014a00: 7269 6275 7469 6f6e 2063 616e 2062 6520  ribution can be 
-00014a10: 6361 6c63 756c 6174 6564 2075 7369 6e67  calculated using
-00014a20: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-00014a30: 6f77 2e0a 0a60 6060 0a77 6764 206b 7364  ow...```.wgd ksd
-00014a40: 2077 6764 5f67 6c6f 6261 6c6d 7262 682f   wgd_globalmrbh/
-00014a50: 676c 6f62 616c 5f4d 5242 482e 7473 7620  global_MRBH.tsv 
-00014a60: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00014a70: 6561 2050 726f 7465 615f 6379 6e61 726f  ea Protea_cynaro
-00014a80: 6964 6573 2041 636f 7275 735f 616d 6572  ides Acorus_amer
-00014a90: 6963 616e 7573 2056 6974 6973 5f76 696e  icanus Vitis_vin
-00014aa0: 6966 6572 6120 2d6f 2077 6764 5f67 6c6f  ifera -o wgd_glo
-00014ab0: 6261 6c6d 7262 685f 6b73 0a60 6060 0a0a  balmrbh_ks.```..
-00014ac0: 5769 7468 2074 6865 2063 616c 6375 6c61  With the calcula
-00014ad0: 7465 6420 6f72 7468 6f6c 6f67 6f75 7320  ted orthologous 
-00014ae0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00014af0: 6469 7374 7269 6275 7469 6f6e 2c20 7765  distribution, we
-00014b00: 2063 616e 2075 7365 2074 6865 2063 6f6d   can use the com
-00014b10: 6d61 6e64 2062 656c 6f77 2074 6f20 636f  mand below to co
-00014b20: 6e64 7563 7420 7468 6520 7261 7465 2063  nduct the rate c
-00014b30: 6f72 7265 6374 696f 6e20 616e 642f 6f72  orrection and/or
-00014b40: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
-00014b50: 6720 616e 616c 7973 6973 2e0a 0a60 6060  g analysis...```
-00014b60: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
-00014b70: 676c 6f62 616c 6d72 6268 5f6b 732f 676c  globalmrbh_ks/gl
-00014b80: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
-00014b90: 2e74 7376 202d 6661 2041 7175 696c 6567  .tsv -fa Aquileg
-00014ba0: 6961 5f63 6f65 7275 6c65 6120 2d65 706b  ia_coerulea -epk
-00014bb0: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
-00014bc0: 6961 5f63 6f65 7275 6c65 612e 6b73 2e74  ia_coerulea.ks.t
-00014bd0: 7376 202d 6170 2077 6764 5f73 796e 2f69  sv -ap wgd_syn/i
-00014be0: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00014bf0: 7270 6f69 6e74 732e 7478 7420 2d73 7020  rpoints.txt -sp 
-00014c00: 7370 6563 6965 7374 7265 652e 6e77 202d  speciestree.nw -
-00014c10: 6f20 7767 645f 7669 7a5f 6d69 7865 645f  o wgd_viz_mixed_
-00014c20: 4b73 202d 2d70 6c6f 7465 6c6d 6d20 2d2d  Ks --plotelmm --
-00014c30: 706c 6f74 6170 676d 6d20 2d2d 7265 7765  plotapgmm --rewe
-00014c40: 6967 6874 0a60 6060 0a0a 6f72 2075 7369  ight.```..or usi
-00014c50: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00014c60: 656c 6f77 2c20 7768 6963 6820 636f 6d62  elow, which comb
-00014c70: 696e 6573 2074 6865 2074 776f 2073 7465  ines the two ste
-00014c80: 7073 2061 626f 7665 2069 6e20 6f6e 652e  ps above in one.
-00014c90: 204e 6f74 6520 7468 6174 2077 6520 7375   Note that we su
-00014ca0: 6767 6573 7420 6f66 2074 616b 696e 6720  ggest of taking 
-00014cb0: 7477 6f20 7365 7061 7261 7465 2073 7465  two separate ste
-00014cc0: 7073 2069 6e20 7768 6963 6820 6077 6764  ps in which `wgd
-00014cd0: 206b 7364 6020 756e 6465 7274 616b 6573   ksd` undertakes
-00014ce0: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
-00014cf0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-00014d00: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00014d10: 6469 7374 7269 6275 7469 6f6e 2077 6869  distribution whi
-00014d20: 6c65 2060 7767 6420 7669 7a60 2063 6172  le `wgd viz` car
-00014d30: 7269 6573 206f 7574 2074 6865 2072 6174  ries out the rat
-00014d40: 6520 636f 7272 6563 7469 6f6e 2061 6e64  e correction and
-00014d50: 2047 4d4d 2061 6e61 6c79 7369 7320 7375   GMM analysis su
-00014d60: 6368 2074 6861 7420 6974 2773 2065 6173  ch that it's eas
-00014d70: 6965 7220 746f 2064 6562 7567 2e0a 6060  ier to debug..``
-00014d80: 600a 7767 6420 6b73 6420 7767 645f 676c  `.wgd ksd wgd_gl
-00014d90: 6f62 616c 6d72 6268 2f67 6c6f 6261 6c5f  obalmrbh/global_
-00014da0: 4d52 4248 2e74 7376 2041 7175 696c 6567  MRBH.tsv Aquileg
-00014db0: 6961 5f63 6f65 7275 6c65 6120 5072 6f74  ia_coerulea Prot
-00014dc0: 6561 5f63 796e 6172 6f69 6465 7320 4163  ea_cynaroides Ac
-00014dd0: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
-00014de0: 5669 7469 735f 7669 6e69 6665 7261 202d  Vitis_vinifera -
-00014df0: 2d65 7874 7261 7061 7261 6e6f 6d65 6b73  -extraparanomeks
-00014e00: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
-00014e10: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-00014e20: 6b73 2e74 7376 202d 7370 2073 7065 6369  ks.tsv -sp speci
-00014e30: 6573 7472 6565 2e6e 7720 2d2d 7265 7765  estree.nw --rewe
-00014e40: 6967 6874 202d 6f20 7767 645f 676c 6f62  ight -o wgd_glob
-00014e50: 616c 6d72 6268 5f6b 735f 7261 7465 5f63  almrbh_ks_rate_c
-00014e60: 6f72 7265 6374 696f 6e20 2d66 6120 4171  orrection -fa Aq
-00014e70: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00014e80: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
-00014e90: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00014ea0: 6f69 6e74 732e 7478 7420 2d2d 706c 6f74  oints.txt --plot
-00014eb0: 656c 6d6d 202d 2d70 6c6f 7461 7067 6d6d  elmm --plotapgmm
-00014ec0: 0a60 6060 0a0a 5468 6520 6669 6c65 2060  .```..The file `
-00014ed0: 7370 6563 6965 7374 7265 652e 6e77 6020  speciestree.nw` 
-00014ee0: 6973 2074 6865 2074 6578 7420 6669 6c65  is the text file
-00014ef0: 206f 6620 7370 6563 6965 7320 7472 6565   of species tree
-00014f00: 2069 6e20 6e65 7769 636b 2074 6861 7420   in newick that 
-00014f10: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
-00014f20: 776f 756c 6420 6265 2063 6f6e 6475 6374  would be conduct
-00014f30: 6564 206f 6e2e 2049 7473 2063 6f6e 7465  ed on. Its conte
-00014f40: 6e74 2069 7320 6173 2062 656c 6f77 2e20  nt is as below. 
-00014f50: 5573 6572 7320 6361 6e20 6f70 7469 6f6e  Users can option
-00014f60: 616c 6c79 2070 726f 7669 6465 2074 6865  ally provide the
-00014f70: 2073 7065 6369 6573 2070 6169 7273 2074   species pairs t
-00014f80: 6f20 6265 2070 6c6f 7474 6564 2062 7574  o be plotted but
-00014f90: 2077 6520 7375 6767 6573 7420 6f66 206a   we suggest of j
-00014fa0: 7573 7420 7573 696e 6720 602d 6661 2041  ust using `-fa A
-00014fb0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00014fc0: 6160 2074 6f20 706c 6f74 2061 6c6c 2070  a` to plot all p
-00014fd0: 6f73 7369 626c 6520 666f 6361 6c2d 7369  ossible focal-si
-00014fe0: 7374 6572 2073 7065 6369 6573 2070 6169  ster species pai
-00014ff0: 7273 2e20 5765 2073 7567 6765 7374 2061  rs. We suggest a
-00015000: 6464 696e 6720 7468 6520 6f70 7469 6f6e  dding the option
-00015010: 2060 2d2d 7265 7765 6967 6874 6020 746f   `--reweight` to
-00015020: 2072 6563 616c 6375 6c61 7465 2074 6865   recalculate the
-00015030: 2077 6569 6768 7420 7065 7220 7370 6563   weight per spec
-00015040: 6965 7320 7061 6972 2073 7563 6820 7468  ies pair such th
-00015050: 6174 2074 6865 2077 6569 6768 7420 6f66  at the weight of
-00015060: 206f 7274 686f 6c6f 676f 7573 2067 656e   orthologous gen
-00015070: 6520 7061 6972 7320 7769 6c6c 2062 6563  e pairs will bec
-00015080: 6f6d 6520 312e 2045 7874 7261 2063 6f6c  ome 1. Extra col
-00015090: 6c69 6e65 6172 2064 6174 6120 6361 6e20  linear data can 
-000150a0: 6265 2061 6464 6564 2062 7920 7468 6520  be added by the 
-000150b0: 6f70 7469 6f6e 2060 2d61 7060 2061 6e64  option `-ap` and
-000150c0: 2061 6464 6974 696f 6e61 6c20 636c 7573   additional clus
-000150d0: 7465 7269 6e67 2061 6e61 6c79 7369 7320  tering analysis 
-000150e0: 6361 6e20 6265 2069 6e69 7469 6174 6564  can be initiated
-000150f0: 2062 7920 7365 7474 696e 6720 7468 6520   by setting the 
-00015100: 6f70 7469 6f6e 2060 2d2d 706c 6f74 6170  option `--plotap
-00015110: 676d 6d60 2061 6e64 2060 2d2d 706c 6f74  gmm` and `--plot
-00015120: 656c 6d6d 602e 0a0a 6060 600a 2828 2856  elmm`...```.(((V
-00015130: 6974 6973 5f76 696e 6966 6572 612c 5072  itis_vinifera,Pr
-00015140: 6f74 6561 5f63 796e 6172 6f69 6465 7329  otea_cynaroides)
-00015150: 2c41 7175 696c 6567 6961 5f63 6f65 7275  ,Aquilegia_coeru
-00015160: 6c65 6129 2c41 636f 7275 735f 616d 6572  lea),Acorus_amer
-00015170: 6963 616e 7573 293b 0a60 6060 0a0a 215b  icanus);.```..![
-00015180: 5d28 6461 7461 2f4d 6978 6564 2e6b 732e  ](data/Mixed.ks.
-00015190: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000151a0: 6561 2e6e 6f64 652e 7765 6967 6874 6564  ea.node.weighted
-000151b0: 2e70 6e67 290a 0a54 6865 206d 6978 6564  .png)..The mixed
-000151c0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000151d0: 2064 6973 7472 6962 7574 696f 6e20 7368   distribution sh
-000151e0: 6f77 6e20 6162 6f76 6520 6973 2061 2070  own above is a p
-000151f0: 7562 6c69 6361 7469 6f6e 2d72 6561 6479  ublication-ready
-00015200: 2066 6967 7572 6520 7468 6174 2061 7373   figure that ass
-00015210: 656d 626c 6573 2074 6865 2072 6573 756c  embles the resul
-00015220: 7473 206f 6620 454c 4d4d 2c20 474d 4d20  ts of ELMM, GMM 
-00015230: 616e 6420 7261 7465 2063 6f72 7265 6374  and rate correct
-00015240: 696f 6e2e 2054 6865 206f 6e65 2d76 732d  ion. The one-vs-
-00015250: 6f6e 6520 6f72 7468 6f6c 6f67 6f75 7320  one orthologous 
-00015260: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00015270: 6469 7374 7269 6275 7469 6f6e 7320 6973  distributions is
-00015280: 2061 6c73 6f20 6175 746f 6d61 7469 6361   also automatica
-00015290: 6c6c 7920 7072 6f64 7563 6564 2077 6974  lly produced wit
-000152a0: 6820 7261 7465 2063 6f72 7265 6374 696f  h rate correctio
-000152b0: 6e20 7265 7375 6c74 7320 7375 7065 7269  n results superi
-000152c0: 6d70 6f73 6564 2077 6865 7265 2061 7661  mposed where ava
-000152d0: 696c 6162 6c65 2061 7320 7368 6f77 6e20  ilable as shown 
-000152e0: 6265 6c6f 772e 0a0a 215b 5d28 6461 7461  below...![](data
-000152f0: 2f46 6f63 7573 5f73 6973 7465 725f 7061  /Focus_sister_pa
-00015300: 6972 732e 6b73 2e6e 6f64 652e 7765 6967  irs.ks.node.weig
-00015310: 6874 6564 2e70 6e67 290a 215b 5d28 6461  hted.png).![](da
-00015320: 7461 2f41 6c6c 5f70 6169 7273 2e6b 732e  ta/All_pairs.ks.
-00015330: 6e6f 6465 2e77 6569 6768 7465 642e 706e  node.weighted.pn
-00015340: 6729 0a0a 4265 7369 6465 7320 7468 6520  g)..Besides the 
-00015350: 6162 6f76 6520 6b73 7261 7465 732d 6c69  above ksrates-li
-00015360: 6b65 202a 4b2a 3c73 7562 3e53 3c2f 7375  ke *K*<sub>S</su
-00015370: 623e 2070 6c6f 742c 2061 206d 6f72 6520  b> plot, a more 
-00015380: 636c 6173 7369 6320 2a4b 2a3c 7375 623e  classic *K*<sub>
-00015390: 533c 2f73 7562 3e20 706c 6f74 2063 616e  S</sub> plot can
-000153a0: 2062 6520 6d61 6465 2062 7920 6164 6469   be made by addi
-000153b0: 6e67 2074 6865 206f 7074 696f 6e20 602d  ng the option `-
-000153c0: 2d63 6c61 7373 6963 6020 746f 2074 6170  -classic` to tap
-000153d0: 206d 6f72 6520 6465 7461 696c 6564 6c79   more detailedly
-000153e0: 2069 6e74 6f20 7468 6520 7661 7269 6174   into the variat
-000153f0: 696f 6e20 6f66 2073 796e 6f6e 796d 6f75  ion of synonymou
-00015400: 7320 7375 6273 7469 7475 7469 6f6e 2072  s substitution r
-00015410: 6174 652e 0a0a 5573 696e 6720 7468 6520  ate...Using the 
-00015420: 636f 6d6d 616e 6420 6265 6c6f 772c 2074  command below, t
-00015430: 6865 2064 6972 6563 7469 6f6e 206f 6620  he direction of 
-00015440: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
-00015450: 616e 6420 6465 6772 6565 206f 6620 7261  and degree of ra
-00015460: 7465 2076 6172 6961 7469 6f6e 2063 616e  te variation can
-00015470: 2062 6520 6f62 7365 7276 6564 206d 6f72   be observed mor
-00015480: 6520 6469 7265 6374 6c79 2e0a 0a60 6060  e directly...```
-00015490: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
-000154a0: 676c 6f62 616c 6d72 6268 5f6b 732f 676c  globalmrbh_ks/gl
-000154b0: 6f62 616c 5f4d 5242 482e 7473 762e 6b73  obal_MRBH.tsv.ks
-000154c0: 2e74 7376 202d 6661 2041 7175 696c 6567  .tsv -fa Aquileg
-000154d0: 6961 5f63 6f65 7275 6c65 6120 2d65 706b  ia_coerulea -epk
-000154e0: 2077 6764 5f6b 7364 2f41 7175 696c 6567   wgd_ksd/Aquileg
-000154f0: 6961 5f63 6f65 7275 6c65 612e 6b73 2e74  ia_coerulea.ks.t
-00015500: 7376 202d 6170 2077 6764 5f73 796e 2f69  sv -ap wgd_syn/i
-00015510: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00015520: 7270 6f69 6e74 732e 7478 7420 2d73 7020  rpoints.txt -sp 
-00015530: 7370 6563 6965 7374 7265 652e 6e77 202d  speciestree.nw -
-00015540: 6f20 7767 645f 7669 7a5f 6d69 7865 645f  o wgd_viz_mixed_
-00015550: 4b73 202d 2d70 6c6f 7465 6c6d 6d20 2d2d  Ks --plotelmm --
-00015560: 706c 6f74 6170 676d 6d20 2d2d 7265 7765  plotapgmm --rewe
-00015570: 6967 6874 202d 2d70 6c6f 746b 6465 202d  ight --plotkde -
-00015580: 2d63 6c61 7373 6963 0a60 6060 0a0a 215b  -classic.```..![
-00015590: 5d28 6461 7461 2f41 7175 696c 6567 6961  ](data/Aquilegia
-000155a0: 5f63 6f65 7275 6c65 615f 476c 6f62 616c  _coerulea_Global
-000155b0: 6d72 6268 4b73 5f45 6c6d 6d5f 4170 676d  mrbhKs_Elmm_Apgm
-000155c0: 6d5f 436f 7272 6563 7465 642e 6b73 642e  m_Corrected.ksd.
-000155d0: 7376 6729 0a0a 4173 2073 686f 776e 2061  svg)..As shown a
-000155e0: 626f 7665 2c20 6265 6361 7573 6520 6f66  bove, because of
-000155f0: 2074 6865 2068 6967 6865 7220 7375 6273   the higher subs
-00015600: 7469 7475 7469 6f6e 2072 6174 6520 6f66  titution rate of
-00015610: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
-00015620: 756c 6561 2a2c 2074 6865 206f 7269 6769  ulea*, the origi
-00015630: 6e61 6c20 6f72 7468 6f6c 6f67 6f75 7320  nal orthologous 
-00015640: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00015650: 7661 6c75 6573 2077 6572 6520 6163 7475  values were actu
-00015660: 616c 6c79 2075 6e64 6572 6573 7469 6d61  ally underestima
-00015670: 7465 6420 696e 2074 6865 2074 696d 652d  ted in the time-
-00015680: 6672 616d 6520 6f66 202a 4171 7569 6c65  frame of *Aquile
-00015690: 6769 6120 636f 6572 756c 6561 2a2e 2057  gia coerulea*. W
-000156a0: 6865 6e20 7765 2072 6563 6f76 6572 6564  hen we recovered
-000156b0: 2074 6865 2064 6976 6572 6765 6e63 6520   the divergence 
-000156c0: 7375 6273 7469 7475 7469 6f6e 2064 6973  substitution dis
-000156d0: 7461 6e63 6520 696e 2074 6572 6d73 206f  tance in terms o
-000156e0: 6620 7477 6f20 7469 6d65 7320 6f66 2074  f two times of t
-000156f0: 6865 2062 7261 6e63 682d 7370 6563 6966  he branch-specif
-00015700: 6963 2063 6f6e 7472 6962 7574 696f 6e20  ic contribution 
-00015710: 6f66 202a 412e 2063 6f65 7275 6c65 612a  of *A. coerulea*
-00015720: 2073 696e 6365 2069 7473 2064 6976 6572   since its diver
-00015730: 6765 6e63 6520 7769 7468 2074 6865 2073  gence with the s
-00015740: 6973 7465 7220 7370 6563 6965 7320 706c  ister species pl
-00015750: 7573 2074 6865 2073 6861 7265 6420 7375  us the shared su
-00015760: 6273 7469 7475 7469 6f6e 2064 6973 7461  bstitution dista
-00015770: 6e63 6520 6265 666f 7265 2064 6976 6572  nce before diver
-00015780: 6765 6e63 6520 2869 6e20 7265 6c61 7469  gence (in relati
-00015790: 7665 2074 6f20 7468 6520 6f75 7467 726f  ve to the outgro
-000157a0: 7570 292c 2074 6865 2063 6f72 7265 6374  up), the correct
-000157b0: 6564 202a 4b2a 3c73 7562 3e53 3c2f 7375  ed *K*<sub>S</su
-000157c0: 623e 206d 6f64 6520 6265 6361 6d65 206c  b> mode became l
-000157d0: 6172 6765 722e 0a0a 4e6f 7465 2074 6861  arger...Note tha
-000157e0: 7420 7765 2063 616e 2065 6173 696c 7920  t we can easily 
-000157f0: 7368 6f77 2074 6861 7420 2a41 7175 696c  show that *Aquil
-00015800: 6567 6961 2063 6f65 7275 6c65 612a 2068  egia coerulea* h
-00015810: 6173 2068 6967 6865 7220 7375 6273 7469  as higher substi
-00015820: 7475 7469 6f6e 2072 6174 6520 7468 616e  tution rate than
-00015830: 202a 5072 6f74 6561 2063 796e 6172 6f69   *Protea cynaroi
-00015840: 6465 732a 2061 6e64 202a 5669 7469 7320  des* and *Vitis 
-00015850: 7669 6e69 6665 7261 2a20 6279 2063 6f6d  vinifera* by com
-00015860: 7061 7269 6e67 2074 6865 6972 2073 7562  paring their sub
-00015870: 7374 6974 7574 696f 6e20 6469 7374 616e  stitution distan
-00015880: 6365 2069 6e20 7265 6761 7264 2074 6f20  ce in regard to 
-00015890: 7468 6520 7361 6d65 2064 6976 6572 6765  the same diverge
-000158a0: 6e63 6520 6576 656e 7420 7769 7468 206f  nce event with o
-000158b0: 7574 6772 6f75 7020 7370 6563 6965 7320  utgroup species 
-000158c0: 2a41 636f 7275 735f 616d 6572 6963 616e  *Acorus_american
-000158d0: 7573 2a2c 2075 7369 6e67 2063 6f6d 6d61  us*, using comma
-000158e0: 6e64 2062 656c 6f77 2e0a 0a60 6060 0a77  nd below...```.w
-000158f0: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
-00015900: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
-00015910: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
-00015920: 7376 202d 7370 2073 7065 6369 6573 7472  sv -sp speciestr
-00015930: 6565 2e6e 7720 2d2d 7265 7765 6967 6874  ee.nw --reweight
-00015940: 202d 6f20 7767 645f 7669 7a5f 436f 6d70   -o wgd_viz_Comp
-00015950: 6172 655f 7261 7465 202d 2d73 7061 6972  are_rate --spair
-00015960: 2022 4163 6f72 7573 5f61 6d65 7269 6361   "Acorus_america
-00015970: 6e75 733b 5072 6f74 6561 5f63 796e 6172  nus;Protea_cynar
-00015980: 6f69 6465 7322 202d 2d73 7061 6972 2022  oides" --spair "
-00015990: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000159a0: 6561 3b41 636f 7275 735f 616d 6572 6963  ea;Acorus_americ
-000159b0: 616e 7573 2220 2d2d 7370 6169 7220 2256  anus" --spair "V
-000159c0: 6974 6973 5f76 696e 6966 6572 613b 4163  itis_vinifera;Ac
-000159d0: 6f72 7573 5f61 6d65 7269 6361 6e75 7322  orus_americanus"
-000159e0: 202d 2d70 6c6f 746b 6465 202d 2d63 6c61   --plotkde --cla
-000159f0: 7373 6963 0a60 6060 0a0a 215b 5d28 6461  ssic.```..![](da
-00015a00: 7461 2f52 6177 5f4f 7274 686f 6c6f 6775  ta/Raw_Orthologu
-00015a10: 6573 5f43 6f6d 7061 7265 5f72 6174 652e  es_Compare_rate.
-00015a20: 6b73 642e 7376 6729 0a0a 4173 2064 6973  ksd.svg)..As dis
-00015a30: 706c 6179 6564 2061 626f 7665 2c20 7468  played above, th
-00015a40: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00015a50: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-00015a60: 6c75 6573 2062 6577 7465 656e 202a 4171  lues bewteen *Aq
-00015a70: 7569 6c65 6769 6120 636f 6572 756c 6561  uilegia coerulea
-00015a80: 2a20 616e 6420 2a41 636f 7275 7320 616d  * and *Acorus am
-00015a90: 6572 6963 616e 7573 2a20 6861 7320 7468  ericanus* has th
-00015aa0: 6520 6869 6768 6573 7420 6d6f 6465 2c20  e highest mode, 
-00015ab0: 696e 6469 6361 7469 6e67 2074 6865 2066  indicating the f
-00015ac0: 6173 7465 7220 7375 6273 7469 7475 7469  aster substituti
-00015ad0: 6f6e 2072 6174 6520 6f66 202a 412e 2063  on rate of *A. c
-00015ae0: 6f65 7275 6c65 612a 2063 6f6d 7061 7265  oerulea* compare
-00015af0: 6420 746f 202a 5072 6f74 6561 2063 796e  d to *Protea cyn
-00015b00: 6172 6f69 6465 732a 2061 6e64 202a 5669  aroides* and *Vi
-00015b10: 7469 7320 7669 6e69 6665 7261 2a2e 0a0a  tis vinifera*...
-00015b20: 4265 666f 7265 2076 322e 302e 3231 2c20  Before v2.0.21, 
-00015b30: 7468 6520 6765 6e65 2d73 7065 6369 6573  the gene-species
-00015b40: 206d 6170 2066 696c 6520 6973 206e 6563   map file is nec
-00015b50: 6365 7373 6172 696c 7920 6e65 6564 6564  cessarily needed
-00015b60: 2066 6f72 2069 7473 2069 6d70 6c65 6d65   for its impleme
-00015b70: 6e74 6174 696f 6e20 696e 2060 7767 6420  ntation in `wgd 
-00015b80: 7669 7a60 2c20 7768 6963 6820 7368 6f75  viz`, which shou
-00015b90: 6c64 2062 6520 6175 746f 6d61 7465 6c79  ld be automately
-00015ba0: 2070 726f 6475 6365 6420 6279 2074 6865   produced by the
-00015bb0: 206c 6173 7420 6077 6764 206b 7364 6020   last `wgd ksd` 
-00015bc0: 7374 6570 2067 6976 656e 2074 6865 2060  step given the `
-00015bd0: 7370 6169 7260 2061 6e64 2060 7370 6563  spair` and `spec
-00015be0: 6965 7374 7265 6560 2070 6172 616d 6574  iestree` paramet
-00015bf0: 6572 732e 2054 6865 2060 6765 6e65 5f73  ers. The `gene_s
-00015c00: 7065 6369 6573 2e6d 6170 6020 6861 7320  pecies.map` has 
-00015c10: 636f 6e74 656e 7473 2061 7320 6265 6c6f  contents as belo
-00015c20: 7720 696e 2077 6869 6368 2065 6163 6820  w in which each 
-00015c30: 6c69 6e65 2069 7320 7468 6520 6a6f 696e  line is the join
-00015c40: 6564 2073 7472 696e 6720 6f66 2067 656e  ed string of gen
-00015c50: 6520 6e61 6d65 2061 6e64 2073 7065 6369  e name and speci
-00015c60: 6573 206e 616d 6520 6279 2073 7061 6365  es name by space
-00015c70: 2e20 4166 7465 7220 7632 2e30 2e32 3120  . After v2.0.21 
-00015c80: 2869 6e63 6c75 6465 6429 2c20 7468 6520  (included), the 
-00015c90: 6765 6e65 2d73 7065 6369 6573 206d 6170  gene-species map
-00015ca0: 2066 696c 6520 6973 206e 6f74 206e 6563   file is not nec
-00015cb0: 6365 7373 6172 696c 7920 6e65 6564 6564  cessarily needed
-00015cc0: 2061 6e79 6d6f 7265 2e0a 0a60 6060 0a41   anymore...```.A
-00015cd0: 7163 6f65 3647 3035 3738 3030 2e31 2041  qcoe6G057800.1 A
-00015ce0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015cf0: 610a 5676 695f 5649 545f 3230 3173 3030  a.Vvi_VIT_201s00
-00015d00: 3131 6730 3135 3330 2e31 2056 6974 6973  11g01530.1 Vitis
-00015d10: 5f76 696e 6966 6572 610a 5063 795f 5072  _vinifera.Pcy_Pr
-00015d20: 6f63 7930 3167 3038 3531 3020 5072 6f74  ocy01g08510 Prot
-00015d30: 6561 5f63 796e 6172 6f69 6465 730a 4161  ea_cynaroides.Aa
-00015d40: 6d5f 4163 6f72 612e 3034 4731 3432 3930  m_Acora.04G14290
-00015d50: 302e 3120 4163 6f72 7573 5f61 6d65 7269  0.1 Acorus_ameri
-00015d60: 6361 6e75 730a 6060 600a 0a41 6e20 616c  canus.```..An al
-00015d70: 7465 726e 6174 6976 6520 7761 7920 746f  ternative way to
-00015d80: 2063 616c 6375 6c61 7465 2074 6865 206f   calculate the o
-00015d90: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00015da0: 7562 3e53 3c2f 7375 623e 2069 7320 746f  ub>S</sub> is to
-00015db0: 2064 6972 6563 746c 7920 7573 6520 7468   directly use th
-00015dc0: 6520 6f72 7468 6f67 726f 7570 7320 696e  e orthogroups in
-00015dd0: 7374 6561 6420 6f66 2067 6c6f 6261 6c20  stead of global 
-00015de0: 4d52 4248 2066 616d 696c 792e 2054 6861  MRBH family. Tha
-00015df0: 7420 7761 7920 7765 2064 6f6e 2774 2075  t way we don't u
-00015e00: 7365 2074 6865 2070 7265 2d69 6e66 6572  se the pre-infer
-00015e10: 7265 6420 7061 7261 6e6f 6d65 202a 4b2a  red paranome *K*
-00015e20: 3c73 7562 3e53 3c2f 7375 623e 2062 7574  <sub>S</sub> but
-00015e30: 2074 6865 2070 6172 616c 6f67 6f75 7320   the paralogous 
-00015e40: 6765 6e65 2070 6169 7273 2069 6e73 6964  gene pairs insid
-00015e50: 6520 6561 6368 206f 7274 686f 6772 6f75  e each orthogrou
-00015e60: 7020 696e 7374 6561 642e 2054 6f20 6163  p instead. To ac
-00015e70: 6869 6576 6520 7468 6174 2c20 7765 2066  hieve that, we f
-00015e80: 6972 7374 206e 6565 6420 746f 2069 6e66  irst need to inf
-00015e90: 6572 206f 7274 686f 6772 6f75 7073 2075  er orthogroups u
-00015ea0: 7369 6e67 2074 6865 2063 6f6d 6d61 6e64  sing the command
-00015eb0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00015ec0: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
-00015ed0: 6f65 7275 6c65 6120 5072 6f74 6561 5f63  oerulea Protea_c
-00015ee0: 796e 6172 6f69 6465 7320 4163 6f72 7573  ynaroides Acorus
-00015ef0: 5f61 6d65 7269 6361 6e75 7320 5669 7469  _americanus Viti
-00015f00: 735f 7669 6e69 6665 7261 202d 2d6f 7274  s_vinifera --ort
-00015f10: 686f 696e 6665 7220 2d6f 2077 6764 5f6f  hoinfer -o wgd_o
-00015f20: 7274 686f 2028 2d2d 6f6e 6c79 6f72 7468  rtho (--onlyorth
-00015f30: 6f29 200a 6060 600a 0a55 7365 7273 2063  o) .```..Users c
-00015f40: 616e 2064 6563 6964 6520 746f 206f 6e6c  an decide to onl
-00015f50: 7920 636f 6e64 7563 7420 7468 6520 6f72  y conduct the or
-00015f60: 7468 6f67 726f 7570 2061 6e61 6c79 7369  thogroup analysi
-00015f70: 7320 7768 696c 6520 736b 6970 7069 6e67  s while skipping
-00015f80: 206f 7468 6572 2061 6e61 6c79 7369 7320   other analysis 
-00015f90: 6279 2061 6464 696e 6720 7468 6520 666c  by adding the fl
-00015fa0: 6167 2060 2d2d 6f6e 6c79 6f72 7468 6f60  ag `--onlyortho`
-00015fb0: 2e20 4e65 7874 2073 7465 7020 6973 2074  . Next step is t
-00015fc0: 6865 2073 616d 6520 7769 7468 2067 6c6f  he same with glo
-00015fd0: 6261 6c20 4d52 4248 2066 616d 696c 792e  bal MRBH family.
-00015fe0: 0a0a 6060 600a 7767 6420 6b73 6420 7767  ..```.wgd ksd wg
-00015ff0: 645f 6f72 7468 6f2f 4f72 7468 6f67 726f  d_ortho/Orthogro
-00016000: 7570 732e 7370 2e74 7376 2041 7175 696c  ups.sp.tsv Aquil
-00016010: 6567 6961 5f63 6f65 7275 6c65 6120 5072  egia_coerulea Pr
-00016020: 6f74 6561 5f63 796e 6172 6f69 6465 7320  otea_cynaroides 
-00016030: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-00016040: 7320 5669 7469 735f 7669 6e69 6665 7261  s Vitis_vinifera
-00016050: 202d 6f20 7767 645f 6f72 7468 6f5f 6b73   -o wgd_ortho_ks
-00016060: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
-00016070: 6f72 7468 6f5f 6b73 2f4f 7274 686f 6772  ortho_ks/Orthogr
-00016080: 6f75 7073 2e73 702e 7473 762e 6b73 2e74  oups.sp.tsv.ks.t
-00016090: 7376 202d 6570 6b20 7767 645f 6b73 642f  sv -epk wgd_ksd/
-000160a0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000160b0: 6561 2e6b 732e 7473 7620 2d73 7020 7370  ea.ks.tsv -sp sp
-000160c0: 6563 6965 7374 7265 652e 6e77 202d 2d72  eciestree.nw --r
-000160d0: 6577 6569 6768 7420 2d61 7020 7767 645f  eweight -ap wgd_
-000160e0: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
-000160f0: 616e 6368 6f72 706f 696e 7473 2e74 7874  anchorpoints.txt
-00016100: 202d 706c 6f74 656c 6d6d 202d 2d70 6c6f   -plotelmm --plo
-00016110: 7461 7067 6d6d 202d 6f20 7767 645f 6f72  tapgmm -o wgd_or
-00016120: 7468 6f5f 6b73 5f72 6174 655f 636f 7272  tho_ks_rate_corr
-00016130: 6563 7469 6f6e 0a60 6060 0a0a 215b 5d28  ection.```..![](
-00016140: 6461 7461 2f4d 6978 6564 2e6b 732e 4171  data/Mixed.ks.Aq
-00016150: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00016160: 2e6e 6f64 652e 7765 6967 6874 6564 5f4f  .node.weighted_O
-00016170: 7274 686f 5f4b 732e 706e 6729 0a21 5b5d  rtho_Ks.png).![]
-00016180: 2864 6174 612f 466f 6375 735f 7369 7374  (data/Focus_sist
-00016190: 6572 5f70 6169 7273 2e6b 732e 6e6f 6465  er_pairs.ks.node
-000161a0: 2e77 6569 6768 7465 645f 4f72 7468 6f5f  .weighted_Ortho_
-000161b0: 4b73 2e70 6e67 290a 215b 5d28 6461 7461  Ks.png).![](data
-000161c0: 2f41 6c6c 5f70 6169 7273 2e6b 732e 6e6f  /All_pairs.ks.no
-000161d0: 6465 2e77 6569 6768 7465 645f 4f72 7468  de.weighted_Orth
-000161e0: 6f5f 4b73 2e70 6e67 290a 0a41 7320 7368  o_Ks.png)..As sh
-000161f0: 6f77 6e20 6162 6f76 652c 2074 6865 206e  own above, the n
-00016200: 756d 6265 7220 6f66 206f 7274 686f 6c6f  umber of ortholo
-00016210: 676f 7573 2067 656e 6520 7061 6972 7320  gous gene pairs 
-00016220: 6973 2064 6966 6665 7265 6e74 2074 6861  is different tha
-00016230: 6e20 7468 6520 6f6e 6520 6672 6f6d 2067  n the one from g
-00016240: 6c6f 6261 6c20 4d52 4248 2066 616d 696c  lobal MRBH famil
-00016250: 6965 7320 696e 2074 6861 7420 6865 7265  ies in that here
-00016260: 2077 6520 706c 6f74 7465 6420 616c 6c20   we plotted all 
-00016270: 6f72 7468 6f6c 6f67 6f75 7320 6765 6e65  orthologous gene
-00016280: 2070 6169 7273 2069 6e73 7465 6164 206f   pairs instead o
-00016290: 6620 6f6e 6c79 2067 6c6f 6261 6c20 4d52  f only global MR
-000162a0: 4248 2066 616d 696c 6965 732c 2074 6f67  BH families, tog
-000162b0: 6574 6865 7220 7769 7468 2064 6966 6665  ether with diffe
-000162c0: 7265 6e74 2072 6563 616c 6375 6c61 7465  rent recalculate
-000162d0: 6420 7765 6967 6874 732e 0a0a 4166 7465  d weights...Afte
-000162e0: 7220 7468 6520 7068 796c 6f67 656e 6574  r the phylogenet
-000162f0: 6963 2074 696d 696e 6720 6f66 2074 6865  ic timing of the
-00016300: 2052 616e 756e 6375 6c61 6c65 7320 5747   Ranunculales WG
-00016310: 442c 2077 6520 6361 6e20 6675 7274 6865  D, we can furthe
-00016320: 7220 696e 6665 7220 6974 7320 6162 736f  r infer its abso
-00016330: 6c75 7465 2061 6765 2e20 4669 7273 7420  lute age. First 
-00016340: 7765 2069 6e66 6572 2074 6865 2063 7265  we infer the cre
-00016350: 6469 626c 6520 7261 6e67 6520 6f66 2061  dible range of a
-00016360: 6e63 686f 7220 7061 6972 7320 6279 202a  nchor pairs by *
-00016370: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2068  K*<sub>S</sub> h
-00016380: 6575 7269 7374 6963 616c 6c79 2075 7369  euristically usi
-00016390: 6e67 2074 6865 2070 726f 6772 616d 2060  ng the program `
-000163a0: 7767 6420 7065 616b 602e 0a0a 6060 600a  wgd peak`...```.
-000163b0: 7767 6420 7065 616b 202d 2d68 6575 7269  wgd peak --heuri
-000163c0: 7374 6963 2077 6764 5f6b 7364 2f41 7175  stic wgd_ksd/Aqu
-000163d0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-000163e0: 7473 762e 6b73 2e74 7376 202d 6170 2077  tsv.ks.tsv -ap w
-000163f0: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
-00016400: 7574 2f61 6e63 686f 7270 6f69 6e74 732e  ut/anchorpoints.
-00016410: 7478 7420 2d73 6d20 7767 645f 7379 6e2f  txt -sm wgd_syn/
-00016420: 6961 6468 6f72 652d 6f75 742f 7365 676d  iadhore-out/segm
-00016430: 656e 7473 2e74 7874 202d 6c65 2077 6764  ents.txt -le wgd
-00016440: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-00016450: 2f6c 6973 745f 656c 656d 656e 7473 2e74  /list_elements.t
-00016460: 7874 202d 6d70 2077 6764 5f73 796e 2f69  xt -mp wgd_syn/i
-00016470: 6164 686f 7265 2d6f 7574 2f6d 756c 7469  adhore-out/multi
-00016480: 706c 6963 6f6e 5f70 6169 7273 2e74 7874  plicon_pairs.txt
-00016490: 202d 6f20 7767 645f 7065 616b 0a60 6060   -o wgd_peak.```
-000164a0: 0a0a 215b 5d28 6461 7461 2f41 6e63 686f  ..![](data/Ancho
-000164b0: 724b 735f 5065 616b 4349 5f41 7175 696c  rKs_PeakCI_Aquil
-000164c0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-000164d0: 762e 6b73 2e74 7376 5f6e 6f64 655f 7765  v.ks.tsv_node_we
-000164e0: 6967 6874 6564 2e73 7667 290a 0a41 7320  ighted.svg)..As 
-000164f0: 7368 6f77 6e20 6162 6f76 652c 2077 6520  shown above, we 
-00016500: 6173 7375 6d65 6420 6120 6c6f 676e 6f72  assumed a lognor
-00016510: 6d61 6c20 6469 7374 7269 6275 7469 6f6e  mal distribution
-00016520: 2061 7420 7468 6520 7065 616b 206c 6f63   at the peak loc
-00016530: 6174 696f 6e20 6465 7465 6374 6564 2062  ation detected b
-00016540: 7920 7468 6520 6073 6967 6e61 6c60 206d  y the `signal` m
-00016550: 6f64 756c 6520 6f66 2060 7363 6970 7960  odule of `scipy`
-00016560: 206c 6962 7261 7279 2e20 5468 6520 3935   library. The 95
-00016570: 2520 636f 6e66 6964 656e 6365 206c 6576  % confidence lev
-00016580: 656c 206f 6620 7468 6520 6c6f 676e 6f72  el of the lognor
-00016590: 6d61 6c20 6469 7374 7269 6275 7469 6f6e  mal distribution
-000165a0: 2077 6173 2061 7070 6c69 6564 2c20 692e   was applied, i.
-000165b0: 652e 2c20 302e 3638 2d32 2e37 342c 2069  e., 0.68-2.74, i
-000165c0: 6e20 6675 7274 6865 7220 6d6f 6c65 6375  n further molecu
-000165d0: 6c61 7220 6461 7469 6e67 2e20 5468 6520  lar dating. The 
-000165e0: 6669 6c65 2060 4171 7569 6c65 6769 615f  file `Aquilegia_
-000165f0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
-00016600: 7473 765f 3935 2543 495f 4150 5f66 6f72  tsv_95%CI_AP_for
-00016610: 5f64 6174 696e 675f 7765 6967 6874 6564  _dating_weighted
-00016620: 5f66 6f72 6d61 742e 7473 7660 2069 7320  _format.tsv` is 
-00016630: 7768 6174 2077 6520 6e65 6564 2066 6f72  what we need for
-00016640: 206e 6578 7420 7374 6570 2e20 546f 2062   next step. To b
-00016650: 7569 6c64 2074 6865 206f 7274 686f 6772  uild the orthogr
-00016660: 6f75 7073 2075 7365 6420 696e 2070 6879  oups used in phy
-00016670: 6c6f 6765 6e65 7469 6320 6461 7469 6e67  logenetic dating
-00016680: 2c20 7765 206e 6565 6420 746f 2073 656c  , we need to sel
-00016690: 6563 7420 736f 6d65 2073 7065 6369 6573  ect some species
-000166a0: 2061 6e64 2066 6f72 6d20 6120 7374 6172   and form a star
-000166b0: 7469 6e67 2074 7265 6520 7769 7468 2070  ting tree with p
-000166c0: 726f 7065 7220 666f 7373 696c 2063 616c  roper fossil cal
-000166d0: 6962 7261 7469 6f6e 732e 2057 6520 7072  ibrations. We pr
-000166e0: 6f76 6964 6520 6f6e 6520 696e 206d 636d  ovide one in mcm
-000166f0: 6374 7265 6520 666f 726d 6174 2061 7320  ctree format as 
-00016700: 6265 6c6f 772e 0a0a 6060 600a 3137 2031  below...```.17 1
-00016710: 0a28 2828 2850 6f74 616d 6f67 6574 6f6e  .((((Potamogeton
-00016720: 5f61 6375 7469 666f 6c69 7573 2c28 5370  _acutifolius,(Sp
-00016730: 6972 6f64 656c 615f 696e 7465 726d 6564  irodela_intermed
-00016740: 6961 2c41 6d6f 7270 686f 7068 616c 6c75  ia,Amorphophallu
-00016750: 735f 6b6f 6e6a 6163 2929 2c28 4163 616e  s_konjac)),(Acan
-00016760: 7468 6f63 686c 616d 7973 5f62 7261 6374  thochlamys_bract
-00016770: 6561 7461 2c28 4469 6f73 636f 7265 615f  eata,(Dioscorea_
-00016780: 616c 6174 612c 4469 6f73 636f 7265 615f  alata,Dioscorea_
-00016790: 726f 7475 6e64 6174 6129 2927 3e30 2e35  rotundata))'>0.5
-000167a0: 3630 303c 312e 3238 3633 2729 273e 302e  600<1.2863')'>0.
-000167b0: 3833 3630 3c31 2e32 3836 3327 2c28 4163  8360<1.2863',(Ac
-000167c0: 6f72 7573 5f61 6d65 7269 6361 6e75 732c  orus_americanus,
-000167d0: 4163 6f72 7573 5f74 6174 6172 696e 6f77  Acorus_tatarinow
-000167e0: 6969 2929 273e 302e 3833 3630 3c31 2e32  ii))'>0.8360<1.2
-000167f0: 3836 3327 2c28 2828 2854 6574 7261 6365  863',((((Tetrace
-00016800: 6e74 726f 6e5f 7369 6e65 6e73 652c 5472  ntron_sinense,Tr
-00016810: 6f63 686f 6465 6e64 726f 6e5f 6172 616c  ochodendron_aral
-00016820: 696f 6964 6573 292c 2842 7578 7573 5f61  ioides),(Buxus_a
-00016830: 7573 7472 6f79 756e 6e61 6e65 6e73 6973  ustroyunnanensis
-00016840: 2c42 7578 7573 5f73 696e 6963 6129 2927  ,Buxus_sinica))'
-00016850: 3e31 2e31 3038 303c 312e 3238 3633 272c  >1.1080<1.2863',
-00016860: 284e 656c 756d 626f 5f6e 7563 6966 6572  (Nelumbo_nucifer
-00016870: 612c 2854 656c 6f70 6561 5f73 7065 6369  a,(Telopea_speci
-00016880: 6f73 6973 7369 6d61 2c50 726f 7465 615f  osissima,Protea_
-00016890: 6379 6e61 726f 6964 6573 2929 2927 3e31  cynaroides)))'>1
-000168a0: 2e31 3038 303c 312e 3238 3633 272c 2841  .1080<1.2863',(A
-000168b0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-000168c0: 615f 6170 312c 4171 7569 6c65 6769 615f  a_ap1,Aquilegia_
-000168d0: 636f 6572 756c 6561 5f61 7032 2929 273e  coerulea_ap2))'>
-000168e0: 312e 3130 3830 3c31 2e32 3836 3327 2927  1.1080<1.2863')'
-000168f0: 3e31 2e32 3732 303c 322e 3437 3230 273b  >1.2720<2.4720';
-00016900: 0a60 6060 0a0a 4173 2070 7265 7365 6e74  .```..As present
-00016910: 6564 2061 626f 7665 2c20 7468 6520 666f  ed above, the fo
-00016920: 6375 7320 7370 6563 6965 7320 7468 6174  cus species that
-00016930: 2069 7320 6162 6f75 7420 746f 2062 6520   is about to be 
-00016940: 6461 7465 6420 6e65 6564 7320 746f 2062  dated needs to b
-00016950: 6520 7265 706c 6163 6564 2077 6974 6820  e replaced with 
-00016960: 6028 4171 7569 6c65 6769 615f 636f 6572  `(Aquilegia_coer
-00016970: 756c 6561 5f61 7031 2c41 7175 696c 6567  ulea_ap1,Aquileg
-00016980: 6961 5f63 6f65 7275 6c65 615f 6170 3229  ia_coerulea_ap2)
-00016990: 602e 2057 6974 6820 7468 6973 2073 7461  `. With this sta
-000169a0: 7274 696e 6720 7472 6565 2061 6e64 2070  rting tree and p
-000169b0: 7265 646f 776e 6c6f 6164 6564 2063 6473  redownloaded cds
-000169c0: 2066 696c 6573 206f 6620 616c 6c20 7468   files of all th
-000169d0: 6520 7370 6563 6965 732c 2077 6520 6361  e species, we ca
-000169e0: 6e20 6275 696c 6420 7468 6520 6f72 7468  n build the orth
-000169f0: 6f67 726f 7570 2075 7365 6420 696e 2074  ogroup used in t
-00016a00: 6865 2066 696e 616c 206d 6f6c 6563 756c  he final molecul
-00016a10: 6172 2064 6174 696e 6720 7573 696e 6720  ar dating using 
-00016a20: 7468 6520 636f 6d6d 616e 6420 6173 2062  the command as b
-00016a30: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
-00016a40: 6d64 202d 6620 4171 7569 6c65 6769 615f  md -f Aquilegia_
-00016a50: 636f 6572 756c 6561 202d 6170 2077 6764  coerulea -ap wgd
-00016a60: 5f70 6561 6b2f 4171 7569 6c65 6769 615f  _peak/Aquilegia_
-00016a70: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
-00016a80: 7473 765f 3935 2543 495f 4150 5f66 6f72  tsv_95%CI_AP_for
-00016a90: 5f64 6174 696e 675f 7765 6967 6874 6564  _dating_weighted
-00016aa0: 5f66 6f72 6d61 742e 7473 7620 2d6f 2077  _format.tsv -o w
-00016ab0: 6764 5f64 6d64 5f6f 7274 686f 2050 6f74  gd_dmd_ortho Pot
-00016ac0: 616d 6f67 6574 6f6e 5f61 6375 7469 666f  amogeton_acutifo
-00016ad0: 6c69 7573 2053 7069 726f 6465 6c61 5f69  lius Spirodela_i
-00016ae0: 6e74 6572 6d65 6469 6120 416d 6f72 7068  ntermedia Amorph
-00016af0: 6f70 6861 6c6c 7573 5f6b 6f6e 6a61 6320  ophallus_konjac 
-00016b00: 4163 616e 7468 6f63 686c 616d 7973 5f62  Acanthochlamys_b
-00016b10: 7261 6374 6561 7461 2044 696f 7363 6f72  racteata Dioscor
-00016b20: 6561 5f61 6c61 7461 2044 696f 7363 6f72  ea_alata Dioscor
-00016b30: 6561 5f72 6f74 756e 6461 7461 2041 636f  ea_rotundata Aco
-00016b40: 7275 735f 616d 6572 6963 616e 7573 2041  rus_americanus A
-00016b50: 636f 7275 735f 7461 7461 7269 6e6f 7769  corus_tatarinowi
-00016b60: 6920 5465 7472 6163 656e 7472 6f6e 5f73  i Tetracentron_s
-00016b70: 696e 656e 7365 2054 726f 6368 6f64 656e  inense Trochoden
-00016b80: 6472 6f6e 5f61 7261 6c69 6f69 6465 7320  dron_aralioides 
-00016b90: 4275 7875 735f 6175 7374 726f 7975 6e6e  Buxus_austroyunn
-00016ba0: 616e 656e 7369 7320 4275 7875 735f 7369  anensis Buxus_si
-00016bb0: 6e69 6361 204e 656c 756d 626f 5f6e 7563  nica Nelumbo_nuc
-00016bc0: 6966 6572 6120 5465 6c6f 7065 615f 7370  ifera Telopea_sp
-00016bd0: 6563 696f 7369 7373 696d 6120 5072 6f74  eciosissima Prot
-00016be0: 6561 5f63 796e 6172 6f69 6465 7320 4171  ea_cynaroides Aq
-00016bf0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00016c00: 0a60 6060 0a0a 5468 6520 7265 7375 6c74  .```..The result
-00016c10: 2066 696c 6520 606d 6572 6765 5f66 6f63   file `merge_foc
-00016c20: 7573 5f61 702e 7473 7660 2069 7320 7768  us_ap.tsv` is wh
-00016c30: 6174 2077 6520 6e65 6564 2066 6f72 2074  at we need for t
-00016c40: 6865 2066 696e 616c 2073 7465 7020 6f66  he final step of
-00016c50: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
-00016c60: 6720 696e 2070 726f 6772 616d 2060 7767  g in program `wg
-00016c70: 6420 666f 6375 7360 2e0a 0a60 6060 0a77  d focus`...```.w
-00016c80: 6764 2066 6f63 7573 202d 2d70 726f 7464  gd focus --protd
-00016c90: 6174 696e 6720 2d2d 6161 6d6f 6465 6c20  ating --aamodel 
-00016ca0: 6c67 2077 6764 5f64 6d64 5f6f 7274 686f  lg wgd_dmd_ortho
-00016cb0: 2f6d 6572 6765 5f66 6f63 7573 5f61 702e  /merge_focus_ap.
-00016cc0: 7473 7620 2d73 7020 6461 7469 6e67 5f74  tsv -sp dating_t
-00016cd0: 7265 652e 6e77 202d 6f20 7767 645f 6461  ree.nw -o wgd_da
-00016ce0: 7469 6e67 202d 6420 6d63 6d63 7472 6565  ting -d mcmctree
-00016cf0: 202d 6473 2027 6275 726e 696e 203d 2032   -ds 'burnin = 2
-00016d00: 3030 3027 202d 6473 2027 7361 6d70 6672  000' -ds 'sampfr
-00016d10: 6571 203d 2031 3030 3027 202d 6473 2027  eq = 1000' -ds '
-00016d20: 6e73 616d 706c 6520 3d20 3230 3030 3027  nsample = 20000'
-00016d30: 2050 6f74 616d 6f67 6574 6f6e 5f61 6375   Potamogeton_acu
-00016d40: 7469 666f 6c69 7573 2053 7069 726f 6465  tifolius Spirode
-00016d50: 6c61 5f69 6e74 6572 6d65 6469 6120 416d  la_intermedia Am
-00016d60: 6f72 7068 6f70 6861 6c6c 7573 5f6b 6f6e  orphophallus_kon
-00016d70: 6a61 6320 4163 616e 7468 6f63 686c 616d  jac Acanthochlam
-00016d80: 7973 5f62 7261 6374 6561 7461 2044 696f  ys_bracteata Dio
-00016d90: 7363 6f72 6561 5f61 6c61 7461 2044 696f  scorea_alata Dio
-00016da0: 7363 6f72 6561 5f72 6f74 756e 6461 7461  scorea_rotundata
-00016db0: 2041 636f 7275 735f 616d 6572 6963 616e   Acorus_american
-00016dc0: 7573 2041 636f 7275 735f 7461 7461 7269  us Acorus_tatari
-00016dd0: 6e6f 7769 6920 5465 7472 6163 656e 7472  nowii Tetracentr
-00016de0: 6f6e 5f73 696e 656e 7365 2054 726f 6368  on_sinense Troch
-00016df0: 6f64 656e 6472 6f6e 5f61 7261 6c69 6f69  odendron_aralioi
-00016e00: 6465 7320 4275 7875 735f 6175 7374 726f  des Buxus_austro
-00016e10: 7975 6e6e 616e 656e 7369 7320 4275 7875  yunnanensis Buxu
-00016e20: 735f 7369 6e69 6361 204e 656c 756d 626f  s_sinica Nelumbo
-00016e30: 5f6e 7563 6966 6572 6120 5465 6c6f 7065  _nucifera Telope
-00016e40: 615f 7370 6563 696f 7369 7373 696d 6120  a_speciosissima 
-00016e50: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
-00016e60: 7320 4171 7569 6c65 6769 615f 636f 6572  s Aquilegia_coer
-00016e70: 756c 6561 0a60 6060 0a0a 4865 7265 2077  ulea.```..Here w
-00016e80: 6520 6f6e 6c79 2069 6d70 6c65 6d65 6e74  e only implement
-00016e90: 6564 2074 6865 2063 6f6e 6361 7465 6e61  ed the concatena
-00016ea0: 7469 6f6e 2061 6e61 6c79 7369 7320 7573  tion analysis us
-00016eb0: 696e 6720 7072 6f74 6569 6e20 7365 7175  ing protein sequ
-00016ec0: 656e 6365 2062 7920 6164 6469 6e67 2074  ence by adding t
-00016ed0: 6865 2066 6c61 6720 602d 2d70 726f 7464  he flag `--protd
-00016ee0: 6174 696e 6760 2061 6e64 2077 6520 7365  ating` and we se
-00016ef0: 7420 7468 6520 7061 7261 6d65 7465 7220  t the parameter 
-00016f00: 666f 7220 606d 636d 6374 7265 6560 2076  for `mcmctree` v
-00016f10: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00016f20: 6473 602e 204e 6f74 6520 7468 6174 206f  ds`. Note that o
-00016f30: 7468 6572 2064 6174 696e 6720 7072 6f67  ther dating prog
-00016f40: 7261 6d20 7375 6368 2061 7320 6072 3873  ram such as `r8s
-00016f50: 6020 616e 6420 6062 6561 7374 6020 6172  ` and `beast` ar
-00016f60: 6520 616c 736f 2061 7661 696c 6162 6c65  e also available
-00016f70: 2067 6976 656e 2073 6f6d 6520 6d61 6e64   given some mand
-00016f80: 6174 6f72 7920 7061 7261 6d65 7465 7273  atory parameters
-00016f90: 2e20 5468 6520 6669 6e61 6c20 6c6f 6720  . The final log 
-00016fa0: 6f66 2074 6865 2073 7563 6365 7373 6675  of the successfu
-00016fb0: 6c20 7275 6e20 6973 2061 7320 6265 6c6f  l run is as belo
-00016fc0: 772e 0a0a 6060 600a 3136 3a30 343a 3235  w...```.16:04:25
-00016fd0: 2049 4e46 4f20 2020 2020 5275 6e6e 696e   INFO     Runnin
-00016fe0: 6720 6d63 6d63 7472 6565 2075 7369 6e67  g mcmctree using
-00016ff0: 2048 6573 7369 616e 206d 6174 7269 7820   Hessian matrix 
-00017000: 6f66 204c 472b 4761 6d6d 6120 2063 6f72  of LG+Gamma  cor
-00017010: 652e 7079 3a39 3637 0a20 2020 2020 2020  e.py:967.       
-00017020: 2020 2020 2020 2020 2020 2066 6f72 2070             for p
-00017030: 726f 7465 696e 206d 6f64 656c 0a32 333a  rotein model.23:
-00017040: 3439 3a33 3720 494e 464f 2020 2020 2050  49:37 INFO     P
-00017050: 6f73 7465 7269 6f72 206d 6561 6e20 666f  osterior mean fo
-00017060: 7220 7468 6520 6167 6573 206f 6620 7767  r the ages of wg
-00017070: 6420 6973 2031 3132 2e38 3934 3520 6d63  d is 112.8945 mc
-00017080: 6d63 7472 6565 2e70 793a 3239 360a 2020  mctree.py:296.  
-00017090: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000170a0: 6d69 6c6c 696f 6e20 7965 6172 7320 6672  million years fr
-000170b0: 6f6d 2043 6f6e 6361 7465 6e61 7465 6420  om Concatenated 
-000170c0: 7065 7074 6964 650a 2020 2020 2020 2020  peptide.        
-000170d0: 2020 2020 2020 2020 2020 616c 6967 6e6d            alignm
-000170e0: 656e 7420 616e 6420 3935 2520 6372 6564  ent and 95% cred
-000170f0: 6962 696c 6974 7920 696e 7465 7276 616c  ibility interval
-00017100: 7320 2843 4929 0a20 2020 2020 2020 2020  s (CI).         
-00017110: 2020 2020 2020 2020 2069 7320 3130 312e           is 101.
-00017120: 3232 342d 3132 332e 3132 3120 6d69 6c6c  224-123.121 mill
-00017130: 696f 6e20 7965 6172 730a 2020 2020 2020  ion years.      
-00017140: 2020 2049 4e46 4f20 2020 2020 546f 7461     INFO     Tota
-00017150: 6c20 7275 6e20 7469 6d65 3a20 3239 3137  l run time: 2917
-00017160: 3573 2020 2020 2020 2020 2020 2020 2020  5s              
-00017170: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017180: 636c 692e 7079 3a32 3431 0a20 2020 2020  cli.py:241.     
-00017190: 2020 2020 494e 464f 2020 2020 2044 6f6e      INFO     Don
-000171a0: 6520 2020 2020 2020 2020 2020 2020 2020  e               
-000171b0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000171c0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000171d0: 2063 6c69 2e70 793a 3234 320a 6060 600a   cli.py:242.```.
-000171e0: 0a54 6f20 7669 7375 616c 697a 6520 7468  .To visualize th
-000171f0: 6520 6461 7465 2c20 7765 2061 6c73 6f20  e date, we also 
-00017200: 7072 6f76 6964 6564 2061 2070 7974 686f  provided a pytho
-00017210: 6e20 7363 7269 7074 2074 6f20 706c 6f74  n script to plot
-00017220: 2074 6865 2057 4744 2064 6174 6573 2069   the WGD dates i
-00017230: 6e20 7468 6520 6077 6764 6020 666f 6c64  n the `wgd` fold
-00017240: 6572 2e20 5573 6572 7320 6e65 6564 2074  er. Users need t
-00017250: 6f20 6578 7472 6163 7420 7468 6520 7261  o extract the ra
-00017260: 7720 6461 7465 7320 6672 6f6d 2074 6865  w dates from the
-00017270: 2060 6d63 6d63 2e74 7874 6020 666f 7220   `mcmc.txt` for 
-00017280: 7468 6520 5747 4420 6e6f 6465 2066 6972  the WGD node fir
-00017290: 7374 2061 6e64 2073 6176 6520 6974 2061  st and save it a
-000172a0: 7320 6669 6c65 2060 6461 7465 732e 7478  s file `dates.tx
-000172b0: 7460 2028 6f72 2077 6861 7465 7665 7220  t` (or whatever 
-000172c0: 7072 6566 6572 7265 6420 6e61 6d65 292e  preferred name).
-000172d0: 2041 6e20 6578 616d 706c 6520 636f 6d6d   An example comm
-000172e0: 616e 6420 6973 2061 7320 6265 6c6f 772e  and is as below.
-000172f0: 0a0a 6060 600a 7079 7468 6f6e 2024 5041  ..```.python $PA
-00017300: 5448 2f70 6f73 7470 6c6f 742e 7079 2070  TH/postplot.py p
-00017310: 6f73 7464 6973 2064 6174 6573 2e74 7874  ostdis dates.txt
-00017320: 202d 2d70 6572 6365 6e74 696c 6520 3930   --percentile 90
-00017330: 202d 2d74 6974 6c65 2022 5747 4420 6461   --title "WGD da
-00017340: 7465 2220 2d2d 6870 6420 2d6f 2022 5261  te" --hpd -o "Ra
-00017350: 6e75 6e63 756c 616c 6573 5f57 4744 5f64  nunculales_WGD_d
-00017360: 6174 652e 7376 6722 0a60 6060 0a0a 215b  ate.svg".```..![
-00017370: 5d28 6461 7461 2f52 616e 756e 6375 6c61  ](data/Ranuncula
-00017380: 6c65 735f 5747 445f 6461 7465 2e73 7667  les_WGD_date.svg
-00017390: 290a 0a54 6865 2070 6f73 7465 7269 6f72  )..The posterior
-000173a0: 206d 6561 6e2c 206d 6564 6961 6e20 616e   mean, median an
-000173b0: 6420 6d6f 6465 206f 6620 7468 6520 5261  d mode of the Ra
-000173c0: 6e75 6e63 756c 616c 6573 2057 4744 2061  nunculales WGD a
-000173d0: 6765 2069 7320 3131 322e 3932 2c20 3131  ge is 112.92, 11
-000173e0: 332e 3434 2061 6e64 2031 3132 2e35 3420  3.44 and 112.54 
-000173f0: 6d79 612c 2077 6974 6820 3930 2520 4850  mya, with 90% HP
-00017400: 4420 3130 352e 3037 202d 2031 3232 2e33  D 105.07 - 122.3
-00017410: 3220 6d79 6120 6173 206d 616e 6966 6573  2 mya as manifes
-00017420: 7465 6420 6162 6f76 652e 0a0a 2323 2320  ted above...### 
-00017430: 4b73 7472 6565 0a0a 496e 2061 6464 6974  Kstree..In addit
-00017440: 696f 6e20 746f 2070 6169 7277 6973 6520  ion to pairwise 
-00017450: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00017460: 6573 7469 6d61 7469 6f6e 2c20 6120 2a4b  estimation, a *K
-00017470: 2a3c 7375 623e 533c 2f73 7562 3e20 7472  *<sub>S</sub> tr
-00017480: 6565 2077 6974 6820 6272 616e 6368 206c  ee with branch l
-00017490: 656e 6774 6820 696e 202a 4b2a 3c73 7562  ength in *K*<sub
-000174a0: 3e53 3c2f 7375 623e 2075 6e69 7420 6361  >S</sub> unit ca
-000174b0: 6e20 616c 736f 2062 6520 6465 7269 7665  n also be derive
-000174c0: 6420 6672 6f6d 2074 6865 2070 726f 6772  d from the progr
-000174d0: 616d 2060 7767 6420 6b73 6460 2067 6976  am `wgd ksd` giv
-000174e0: 656e 2074 6865 206f 7074 696f 6e20 602d  en the option `-
-000174f0: 2d6b 7374 7265 6560 2061 6e64 2060 2d2d  -kstree` and `--
-00017500: 7370 6563 6965 7374 7265 6560 2e20 4e6f  speciestree`. No
-00017510: 7465 2074 6861 7420 7468 6520 6164 6469  te that the addi
-00017520: 7469 6f6e 616c 206f 7074 696f 6e20 602d  tional option `-
-00017530: 2d6f 6e6c 7963 6f6e 6361 746b 7374 7265  -onlyconcatkstre
-00017540: 6560 2077 696c 6c20 6f6e 6c79 2063 616c  e` will only cal
-00017550: 6c20 7468 6520 2a4b 2a3c 7375 623e 533c  l the *K*<sub>S<
-00017560: 2f73 7562 3e20 6573 7469 6d61 7469 6f6e  /sub> estimation
-00017570: 2066 6f72 2074 6865 2063 6f6e 6361 7465   for the concate
-00017580: 6e61 7465 6420 616c 6967 6e6d 656e 7420  nated alignment 
-00017590: 7261 7468 6572 2074 6861 6e20 616c 6c20  rather than all 
-000175a0: 7468 6520 616c 6967 6e6d 656e 7473 2e20  the alignments. 
-000175b0: 5573 6572 7320 6e65 6564 2074 6f20 7072  Users need to pr
-000175c0: 6f76 6964 6520 6120 7072 6573 6574 2073  ovide a preset s
-000175d0: 7065 6369 6573 2074 7265 6520 666f 7220  pecies tree for 
-000175e0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-000175f0: 7562 3e20 7472 6565 2069 6e66 6572 656e  ub> tree inferen
-00017600: 6365 206f 6620 7468 6520 636f 6e63 6174  ce of the concat
-00017610: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
-00017620: 2077 6869 6c65 2074 6865 2072 656d 6169   while the remai
-00017630: 6e69 6e67 2061 6c69 676e 6d65 6e74 7320  ning alignments 
-00017640: 7769 6c6c 2062 6520 6167 6169 6e73 7420  will be against 
-00017650: 616e 2061 7574 6f6d 6174 656c 7920 696e  an automately in
-00017660: 6665 7272 6564 2074 7265 6520 6672 6f6d  ferred tree from
-00017670: 2060 6661 7374 7472 6565 6020 6f72 2060   `fasttree` or `
-00017680: 6971 7472 6565 602e 2049 6e20 7468 6520  iqtree`. In the 
-00017690: 656e 642c 2075 7365 7273 2077 696c 6c20  end, users will 
-000176a0: 6765 7420 6120 2a4b 2a3c 7375 623e 533c  get a *K*<sub>S<
-000176b0: 2f73 7562 3e20 7472 6565 2c20 6120 2a4b  /sub> tree, a *K
-000176c0: 2a3c 7375 623e 413c 2f73 7562 3e20 7472  *<sub>A</sub> tr
-000176d0: 6565 2061 6e64 2061 20cf 8920 7472 6565  ee and a .. tree
-000176e0: 2070 6572 2066 616d 2061 6e64 2066 6f72   per fam and for
-000176f0: 2074 6865 2063 6f6e 6361 7465 6e61 7465   the concatenate
-00017700: 6420 616c 6967 6e6d 656e 742e 0a0a 6060  d alignment...``
-00017710: 600a 7767 6420 6b73 6420 6461 7461 2f6b  `.wgd ksd data/k
-00017720: 7374 7265 655f 6461 7461 2f66 616d 2e74  stree_data/fam.t
-00017730: 7376 2064 6174 612f 6b73 7472 6565 5f64  sv data/kstree_d
-00017740: 6174 612f 4163 6f72 7573 5f74 6174 6172  ata/Acorus_tatar
-00017750: 696e 6f77 6969 2064 6174 612f 6b73 7472  inowii data/kstr
-00017760: 6565 5f64 6174 612f 416d 626f 7265 6c6c  ee_data/Amborell
-00017770: 615f 7472 6963 686f 706f 6461 2064 6174  a_trichopoda dat
-00017780: 612f 6b73 7472 6565 5f64 6174 612f 4171  a/kstree_data/Aq
-00017790: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000177a0: 2064 6174 612f 6b73 7472 6565 5f64 6174   data/kstree_dat
-000177b0: 612f 4172 6973 746f 6c6f 6368 6961 5f66  a/Aristolochia_f
-000177c0: 696d 6272 6961 7461 2064 6174 612f 6b73  imbriata data/ks
-000177d0: 7472 6565 5f64 6174 612f 4379 6361 735f  tree_data/Cycas_
-000177e0: 7061 6e7a 6869 6875 6165 6e73 6920 2d2d  panzhihuaensi --
-000177f0: 6b73 7472 6565 202d 2d73 7065 6369 6573  kstree --species
-00017800: 7472 6565 2064 6174 612f 6b73 7472 6565  tree data/kstree
-00017810: 5f64 6174 612f 7370 6563 6965 735f 7472  _data/species_tr
-00017820: 6565 312e 6e77 202d 2d6f 6e6c 7963 6f6e  ee1.nw --onlycon
-00017830: 6361 746b 7374 7265 6520 2d6f 2077 6764  catkstree -o wgd
-00017840: 5f6b 7374 7265 655f 746f 706f 6c6f 6779  _kstree_topology
-00017850: 310a 6060 600a 0a21 5b5d 2864 6174 612f  1.```..![](data/
-00017860: 6b73 7472 6565 5f72 6573 756c 7473 2f6b  kstree_results/k
-00017870: 7374 7265 652e 7376 6729 0a0a 4162 6f76  stree.svg)..Abov
-00017880: 6520 7765 2075 7365 6420 7468 7265 6520  e we used three 
-00017890: 616c 7465 726e 6174 6976 6520 746f 706f  alternative topo
-000178a0: 6c6f 6769 6573 2074 6f20 696e 6665 7220  logies to infer 
-000178b0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-000178c0: 7562 3e20 7472 6565 2077 6869 6368 206c  ub> tree which l
-000178d0: 6564 2074 6f20 6469 6666 6572 656e 7420  ed to different 
-000178e0: 6272 616e 6368 206c 656e 6774 6820 6573  branch length es
-000178f0: 7469 6d61 7469 6f6e 2e20 4e6f 7465 2074  timation. Note t
-00017900: 6861 7420 7468 6520 6661 6d69 6c69 6573  hat the families
-00017910: 2077 6520 7573 6564 2077 6572 6520 6f6e   we used were on
-00017920: 6c79 2074 776f 2067 6c6f 6261 6c20 4d52  ly two global MR
-00017930: 4248 2066 616d 696c 6965 7320 666f 7220  BH families for 
-00017940: 7468 6520 7075 7270 6f73 6520 6f66 2069  the purpose of i
-00017950: 6c6c 7573 7472 6174 696f 6e2e 2054 6f20  llustration. To 
-00017960: 6163 7175 6972 6520 616e 2061 6363 7572  acquire an accur
-00017970: 6174 6520 7072 6f66 696c 6520 6f66 2074  ate profile of t
-00017980: 6865 2073 7562 7374 6974 7574 696f 6e20  he substitution 
-00017990: 7261 7465 2076 6172 6961 7469 6f6e 2c20  rate variation, 
-000179a0: 6f72 7468 6f6c 6f67 7565 7320 6174 2074  orthologues at t
-000179b0: 6865 2077 686f 6c65 2067 656e 6f6d 6520  he whole genome 
-000179c0: 7363 616c 6520 7368 6f75 6c64 2062 6520  scale should be 
-000179d0: 7573 6564 2e0a 0a23 2320 4369 7461 7469  used...## Citati
-000179e0: 6f6e 0a20 0a50 6c65 6173 6520 6369 7465  on. .Please cite
-000179f0: 2075 7320 6174 2068 7474 7073 3a2f 2f64   us at https://d
-00017a00: 6f69 2e6f 7267 2f31 302e 3130 3037 2f39  oi.org/10.1007/9
-00017a10: 3738 2d31 2d30 3731 362d 3235 3631 2d33  78-1-0716-2561-3
-00017a20: 5f31 2061 6e64 2068 7474 7073 3a2f 2f64  _1 and https://d
-00017a30: 6f69 2e6f 7267 2f31 302e 3130 3933 2f62  oi.org/10.1093/b
-00017a40: 696f 696e 666f 726d 6174 6963 732f 6274  ioinformatics/bt
-00017a50: 6165 3237 322e 0a0a 6060 600a 4865 6e67  ae272...```.Heng
-00017a60: 6368 6920 4368 656e 2c20 4172 7468 7572  chi Chen, Arthur
-00017a70: 205a 7761 656e 6570 6f65 6c20 2832 3032   Zwaenepoel (202
-00017a80: 3329 2e20 496e 6665 7265 6e63 6520 6f66  3). Inference of
-00017a90: 2041 6e63 6965 6e74 2050 6f6c 7970 6c6f   Ancient Polyplo
-00017aa0: 6964 7920 6672 6f6d 2047 656e 6f6d 6963  idy from Genomic
-00017ab0: 2044 6174 612e 2049 6e3a 2056 616e 2064   Data. In: Van d
-00017ac0: 6520 5065 6572 2c20 592e 2028 6564 7329  e Peer, Y. (eds)
-00017ad0: 2050 6f6c 7970 6c6f 6964 792e 204d 6574   Polyploidy. Met
-00017ae0: 686f 6473 2069 6e20 4d6f 6c65 6375 6c61  hods in Molecula
-00017af0: 7220 4269 6f6c 6f67 792c 2076 6f6c 2032  r Biology, vol 2
-00017b00: 3534 352e 2048 756d 616e 612c 204e 6577  545. Humana, New
-00017b10: 2059 6f72 6b2c 204e 592e 2068 7474 7073   York, NY. https
-00017b20: 3a2f 2f64 6f69 2e6f 7267 2f31 302e 3130  ://doi.org/10.10
-00017b30: 3037 2f39 3738 2d31 2d30 3731 362d 3235  07/978-1-0716-25
-00017b40: 3631 2d33 5f31 0a48 656e 6763 6869 2043  61-3_1.Hengchi C
-00017b50: 6865 6e2c 2041 7274 6875 7220 5a77 6165  hen, Arthur Zwae
-00017b60: 6e65 706f 656c 2c20 5976 6573 2056 616e  nepoel, Yves Van
-00017b70: 2064 6520 5065 6572 2028 3230 3234 292e   de Peer (2024).
-00017b80: 2077 6764 2076 323a 2061 2073 7569 7465   wgd v2: a suite
-00017b90: 206f 6620 746f 6f6c 7320 746f 2075 6e63   of tools to unc
-00017ba0: 6f76 6572 2061 6e64 2064 6174 6520 616e  over and date an
-00017bb0: 6369 656e 7420 706f 6c79 706c 6f69 6479  cient polyploidy
-00017bc0: 2061 6e64 2077 686f 6c65 2d67 656e 6f6d   and whole-genom
-00017bd0: 6520 6475 706c 6963 6174 696f 6e2e 2042  e duplication. B
-00017be0: 696f 696e 666f 726d 6174 6963 732c 2068  ioinformatics, h
-00017bf0: 7474 7073 3a2f 2f64 6f69 2e6f 7267 2f31  ttps://doi.org/1
-00017c00: 302e 3130 3933 2f62 696f 696e 666f 726d  0.1093/bioinform
-00017c10: 6174 6963 732f 6274 6165 3237 320a 6060  atics/btae272.``
-00017c20: 600a 0a46 6f72 2063 6974 6174 696f 6e20  `..For citation 
-00017c30: 6f66 2074 6865 2074 6f6f 6c73 2075 7365  of the tools use
-00017c40: 6420 696e 2077 6764 2c20 706c 6561 7365  d in wgd, please
-00017c50: 2063 6f6e 7375 6c74 2074 6865 2064 6f63   consult the doc
-00017c60: 756d 656e 7461 7469 6f6e 2061 740a 6874  umentation at.ht
-00017c70: 7470 733a 2f2f 7767 6476 322e 7265 6164  tps://wgdv2.read
-00017c80: 7468 6564 6f63 732e 696f 2f65 6e2f 6c61  thedocs.io/en/la
-00017c90: 7465 7374 2f63 6974 6174 696f 6e2e 6874  test/citation.ht
-00017ca0: 6d6c 2e0a 0a                             ml...
+00006e10: 6520 2d2d 666f 6375 7320 6f70 7469 6f6e  e --focus option
+00006e20: 2077 6f75 6c64 2062 6520 6967 6e6f 7265   would be ignore
+00006e30: 6420 616e 6420 6f6e 6c79 2067 6c6f 6261  d and only globa
+00006e40: 6c20 4d52 4248 7320 776f 756c 6420 6265  l MRBHs would be
+00006e50: 2062 7569 6c74 0a2d 6e2c 202d 2d6e 7468   built.-n, --nth
+00006e60: 7265 6164 732c 2074 6865 206e 756d 6265  reads, the numbe
+00006e70: 7220 6f66 2074 6872 6561 6473 2074 6f20  r of threads to 
+00006e80: 7573 652c 2064 6566 6175 6c74 2034 0a2d  use, default 4.-
+00006e90: 6f69 2c20 2d2d 6f72 7468 6f69 6e66 6572  oi, --orthoinfer
+00006ea0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00006eb0: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
+00006ec0: 7465 206f 7274 686f 6772 6f75 7020 696e  te orthogroup in
+00006ed0: 6665 726e 6563 652c 2069 6620 7468 6520  fernece, if the 
+00006ee0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00006ef0: 6520 6f72 7468 6f67 726f 7570 2069 6e66  e orthogroup inf
+00006f00: 6572 6e65 6365 2070 726f 6772 616d 2077  ernece program w
+00006f10: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00006f20: 640a 2d6f 6f2c 202d 2d6f 6e6c 796f 7274  d.-oo, --onlyort
+00006f30: 686f 2c20 666c 6167 206f 7074 696f 6e2c  ho, flag option,
+00006f40: 2077 6865 7468 6572 2074 6f20 6f6e 6c79   whether to only
+00006f50: 2063 6f6e 6475 6374 206f 7274 686f 6772   conduct orthogr
+00006f60: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
+00006f70: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00006f80: 6574 2c20 6f6e 6c79 2074 6865 206f 7274  et, only the ort
+00006f90: 686f 6772 6f75 7020 696e 6665 726e 6563  hogroup infernec
+00006fa0: 6520 7069 7065 6c69 6e65 2077 6f75 6c64  e pipeline would
+00006fb0: 2062 6520 7065 7266 6f72 6d65 6420 7768   be performed wh
+00006fc0: 696c 6520 7468 6520 6f74 6865 7220 616e  ile the other an
+00006fd0: 616c 7973 6973 2077 6f75 6c64 6e27 7420  alysis wouldn't 
+00006fe0: 6265 2069 6e69 7469 6174 6564 0a2d 676e  be initiated.-gn
+00006ff0: 2c20 2d2d 6765 746e 736f 672c 2066 6c61  , --getnsog, fla
+00007000: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00007010: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+00007020: 6520 7365 6172 6368 696e 6720 666f 7220  e searching for 
+00007030: 6e65 7374 6564 2073 696e 676c 652d 636f  nested single-co
+00007040: 7079 2067 656e 6520 6661 6d69 6c69 6573  py gene families
+00007050: 2028 4e53 4f47 7329 2028 6f6e 6c79 206d   (NSOGs) (only m
+00007060: 6561 6e69 6e67 6675 6c20 7768 656e 2074  eaningful when t
+00007070: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+00007080: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
+00007090: 2077 6173 2061 6374 6976 6174 6564 292c   was activated),
+000070a0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+000070b0: 2073 6574 2c20 6164 6469 7469 6f6e 616c   set, additional
+000070c0: 204e 534f 4773 2061 6e61 6c79 7369 7320   NSOGs analysis 
+000070d0: 776f 756c 6420 6265 2070 6572 666f 726d  would be perform
+000070e0: 6564 2062 6573 6964 6573 2074 6865 2062  ed besides the b
+000070f0: 6173 6963 206f 7274 686f 6772 6f75 7020  asic orthogroup 
+00007100: 696e 6665 726e 6563 650a 2d74 7265 652c  infernece.-tree,
+00007110: 202d 2d74 7265 655f 6d65 7468 6f64 2c20   --tree_method, 
+00007120: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
+00007130: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
+00007140: 6d20 746f 2069 6e76 6f6b 6520 286f 6e6c  m to invoke (onl
+00007150: 7920 6d65 616e 696e 6766 756c 2077 6865  y meaningful whe
+00007160: 6e20 7468 6520 636f 6c6c 696e 6561 7220  n the collinear 
+00007170: 636f 616c 6573 6365 6e63 652c 2067 656e  coalescence, gen
+00007180: 652d 746f 2d66 616d 696c 7920 6173 7369  e-to-family assi
+00007190: 676e 6d65 6e74 206f 7220 4e53 4f47 7320  gnment or NSOGs 
+000071a0: 616e 616c 7973 6973 2077 6572 6520 6163  analysis were ac
+000071b0: 7469 7661 7465 6429 2c20 6465 6661 756c  tivated), defaul
+000071c0: 7420 6661 7374 7472 6565 0a2d 7473 2c20  t fasttree.-ts, 
+000071d0: 2d2d 7472 6565 7365 742c 2074 6865 2070  --treeset, the p
+000071e0: 6172 616d 6574 6572 7320 7365 7474 696e  arameters settin
+000071f0: 6720 666f 7220 6765 6e65 2074 7265 6520  g for gene tree 
+00007200: 696e 6665 7265 6e63 652c 2064 6566 6175  inference, defau
+00007210: 6c74 204e 6f6e 652c 2074 6869 7320 6f70  lt None, this op
+00007220: 7469 6f6e 2063 616e 2062 6520 7072 6f76  tion can be prov
+00007230: 6964 6564 206d 756c 7469 706c 6520 7469  ided multiple ti
+00007240: 6d65 730a 2d6d 632c 202d 2d6d 736f 6763  mes.-mc, --msogc
+00007250: 7574 2c20 7468 6520 7261 7469 6f20 6375  ut, the ratio cu
+00007260: 746f 6666 2066 6f72 206d 6f73 746c 7920  toff for mostly 
+00007270: 7369 6e67 6c65 2d63 6f70 7920 6661 6d69  single-copy fami
+00007280: 6c79 2028 6d65 616e 696e 6766 756c 2077  ly (meaningful w
+00007290: 6865 6e20 6163 7469 7661 7469 6e67 2074  hen activating t
+000072a0: 6865 206f 7274 686f 6772 6f75 7020 696e  he orthogroup in
+000072b0: 6665 726e 6563 6520 7069 7065 6c69 6e65  fernece pipeline
+000072c0: 2920 616e 6420 7370 6563 6965 7320 7265  ) and species re
+000072d0: 7072 6573 656e 7461 7469 6f6e 2069 6e20  presentation in 
+000072e0: 636f 6c6c 696e 6561 7220 636f 616c 6573  collinear coales
+000072f0: 6365 6e63 6520 616e 616c 7973 6973 2c20  cence analysis, 
+00007300: 6465 6661 756c 7420 302e 382e 0a2d 6761  default 0.8..-ga
+00007310: 2c20 2d2d 6765 6e65 6173 7369 676e 2c20  , --geneassign, 
+00007320: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00007330: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+00007340: 2074 6865 2067 656e 652d 746f 2d66 616d   the gene-to-fam
+00007350: 696c 7920 6173 7369 676e 6d65 6e74 2061  ily assignment a
+00007360: 6e61 6c79 7369 732c 2069 6620 7468 6520  nalysis, if the 
+00007370: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+00007380: 6520 616e 616c 7973 6973 2077 6f75 6c64  e analysis would
+00007390: 2062 6520 696e 6974 6961 7465 640a 2d73   be initiated.-s
+000073a0: 612c 202d 2d73 6571 3261 7373 6967 6e2c  a, --seq2assign,
+000073b0: 2074 6865 2071 7565 7269 6564 2073 6571   the queried seq
+000073c0: 7565 6e63 6573 2064 6174 6120 6669 6c65  uences data file
+000073d0: 2069 6e20 6765 6e65 2d74 6f2d 6661 6d69   in gene-to-fami
+000073e0: 6c79 2061 7373 6967 6e6d 656e 7420 616e  ly assignment an
+000073f0: 616c 7973 6973 2c20 6465 6661 756c 7420  alysis, default 
+00007400: 4e6f 6e65 2c20 7468 6973 206f 7074 696f  None, this optio
+00007410: 6e20 6361 6e20 6265 2070 726f 7669 6465  n can be provide
+00007420: 6420 6d75 6c74 6970 6c65 2074 696d 6573  d multiple times
+00007430: 0a2d 6661 2c20 2d2d 6661 6d32 6173 7369  .-fa, --fam2assi
+00007440: 676e 2c20 7468 6520 7175 6572 6965 6420  gn, the queried 
+00007450: 6661 6d69 6c69 7920 6461 7461 2066 696c  familiy data fil
+00007460: 6520 696e 2067 656e 652d 746f 2d66 616d  e in gene-to-fam
+00007470: 696c 7920 6173 7369 676e 6d65 6e74 2061  ily assignment a
+00007480: 6e61 6c79 7369 732c 2064 6566 6175 6c74  nalysis, default
+00007490: 204e 6f6e 650a 2d63 632c 202d 2d63 6f6e   None.-cc, --con
+000074a0: 6361 742c 2066 6c61 6720 6f70 7469 6f6e  cat, flag option
+000074b0: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
+000074c0: 7469 6174 6520 7468 6520 636f 6e63 6174  tiate the concat
+000074d0: 656e 6174 696f 6e20 7069 7065 6c69 6e65  enation pipeline
+000074e0: 2066 6f72 206f 7274 686f 6772 6f75 7020   for orthogroup 
+000074f0: 696e 6665 726e 6563 652c 2069 6620 7468  infernece, if th
+00007500: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00007510: 7468 6520 616e 616c 7973 6973 2077 6f75  the analysis wou
+00007520: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+00007530: 2d74 652c 202d 2d74 6573 7473 6f67 2c20  -te, --testsog, 
+00007540: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00007550: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+00007560: 2074 6865 2075 6e62 6961 7365 6420 7465   the unbiased te
+00007570: 7374 206f 6620 7369 6e67 6c65 2d63 6f70  st of single-cop
+00007580: 7920 6765 6e65 2066 616d 696c 6965 732c  y gene families,
+00007590: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+000075a0: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
+000075b0: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
+000075c0: 6961 7465 640a 2d62 732c 202d 2d62 696e  iated.-bs, --bin
+000075d0: 732c 2074 6865 206e 756d 6265 7220 6f66  s, the number of
+000075e0: 2062 696e 7320 6469 7669 6465 6420 696e   bins divided in
+000075f0: 2074 6865 2067 656e 6520 6c65 6e67 7468   the gene length
+00007600: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
+00007610: 6465 6661 756c 7420 3130 300a 2d6e 702c  default 100.-np,
+00007620: 202d 2d6e 6f72 6d61 6c69 7a65 6470 6572   --normalizedper
+00007630: 6365 6e74 2c20 7468 6520 7065 7263 656e  cent, the percen
+00007640: 7461 6765 206f 6620 7570 7065 7220 6869  tage of upper hi
+00007650: 7473 2075 7365 6420 666f 7220 6765 6e65  ts used for gene
+00007660: 206c 656e 6774 6820 6e6f 726d 616c 697a   length normaliz
+00007670: 6174 696f 6e2c 2064 6566 6175 6c74 2035  ation, default 5
+00007680: 0a2d 6e6e 2c20 2d2d 6e6f 6e6f 726d 616c  .-nn, --nonormal
+00007690: 697a 6174 696f 6e2c 2066 6c61 6720 6f70  ization, flag op
+000076a0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+000076b0: 2063 616c 6c20 6f66 6620 7468 6520 6e6f   call off the no
+000076c0: 726d 616c 697a 6174 696f 6e2c 2069 6620  rmalization, if 
+000076d0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+000076e0: 2c20 6e6f 206e 6f72 6d61 6c69 7a61 7469  , no normalizati
+000076f0: 6f6e 2077 6f75 6c64 2062 6520 636f 6e64  on would be cond
+00007700: 7563 7465 640a 2d62 736f 672c 202d 2d62  ucted.-bsog, --b
+00007710: 7573 636f 736f 672c 2066 6c61 6720 6f70  uscosog, flag op
+00007720: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00007730: 2069 6e69 7469 6174 6520 7468 6520 6275   initiate the bu
+00007740: 7363 6f2d 6775 6964 6564 2073 696e 676c  sco-guided singl
+00007750: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
+00007760: 6c79 2061 6e61 6c79 7369 732c 2069 6620  ly analysis, if 
+00007770: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00007780: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
+00007790: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+000077a0: 640a 2d62 686d 6d2c 202d 2d62 7573 636f  d.-bhmm, --busco
+000077b0: 686d 6d2c 2074 6865 2048 4d4d 2070 726f  hmm, the HMM pro
+000077c0: 6669 6c65 2064 6174 6166 696c 6520 696e  file datafile in
+000077d0: 2074 6865 2062 7573 636f 2d67 7569 6465   the busco-guide
+000077e0: 6420 7369 6e67 6c65 2d63 6f70 7920 6765  d single-copy ge
+000077f0: 6e65 2066 616d 696c 7920 616e 616c 7973  ne family analys
+00007800: 6973 2c20 6465 6661 756c 7420 4e6f 6e65  is, default None
+00007810: 0a2d 6263 7466 2c20 2d2d 6275 7363 6f63  .-bctf, --buscoc
+00007820: 7574 6f66 662c 2074 6865 2048 4d4d 2073  utoff, the HMM s
+00007830: 636f 7265 2063 7574 6f66 6620 6461 7461  core cutoff data
+00007840: 6669 6c65 2069 6e20 7468 6520 6275 7363  file in the busc
+00007850: 6f2d 6775 6964 6564 2073 696e 676c 652d  o-guided single-
+00007860: 636f 7079 2067 656e 6520 6661 6d69 6c79  copy gene family
+00007870: 2061 6e61 6c79 7369 732c 2064 6566 6175   analysis, defau
+00007880: 6c74 204e 6f6e 650a 2d6f 6620 2c2d 2d6f  lt None.-of ,--o
+00007890: 6766 6f72 6d61 742c 2066 6c61 6720 6f70  gformat, flag op
+000078a0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+000078b0: 2061 6464 2069 6e64 6578 2074 6f20 7468   add index to th
+000078c0: 6520 5242 4820 6661 6d69 6c69 6573 0a60  e RBH families.`
+000078d0: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+000078e0: 6077 6764 2066 6f63 7573 6020 6361 6e20  `wgd focus` can 
+000078f0: 7265 616c 697a 6520 7468 6520 636f 6e63  realize the conc
+00007900: 6174 656e 6174 696f 6e2d 6261 7365 6420  atenation-based 
+00007910: 616e 6420 636f 616c 6573 6365 6e63 652d  and coalescence-
+00007920: 6261 7365 6420 7068 796c 6f67 656e 6574  based phylogenet
+00007930: 6963 2069 6e66 6572 656e 6365 2061 6e64  ic inference and
+00007940: 2070 6879 6c6f 6765 6e65 7469 6320 6461   phylogenetic da
+00007950: 7469 6e67 206f 6620 5747 4473 2065 7463  ting of WGDs etc
+00007960: 2e0a 6060 600a 7767 6420 666f 6375 7320  ..```.wgd focus 
+00007970: 6661 6d69 6c69 6573 2073 6571 7565 6e63  families sequenc
+00007980: 6573 2028 6f70 7469 6f6e 290a 2d2d 2d2d  es (option).----
+00007990: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000079a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000079b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000079c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000079d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c  ------------.-o,
+000079e0: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+000079f0: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+00007a00: 2064 6566 6175 6c74 2077 6764 5f66 6f63   default wgd_foc
+00007a10: 7573 0a2d 742c 202d 2d74 6d70 6469 722c  us.-t, --tmpdir,
+00007a20: 2074 6865 2074 656d 706f 7261 7279 2077   the temporary w
+00007a30: 6f72 6b69 6e67 2064 6972 6563 746f 7279  orking directory
+00007a40: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+00007a50: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
+00007a60: 6e2c 2074 6865 2074 6d70 6469 7220 7769  n, the tmpdir wi
+00007a70: 6c6c 2062 6520 6173 7369 676e 6564 2072  ll be assigned r
+00007a80: 616e 646f 6d20 6e61 6d65 7320 696e 2063  andom names in c
+00007a90: 7572 7265 6e74 2064 6972 6563 746f 7279  urrent directory
+00007aa0: 2061 6e64 2061 7574 6f6d 6174 656c 7920   and automately 
+00007ab0: 7265 6d6f 7665 6420 6174 2074 6865 2063  removed at the c
+00007ac0: 6f6d 706c 6574 696f 6e20 6f66 2070 726f  ompletion of pro
+00007ad0: 6772 616d 2c20 656c 7365 2074 6865 2074  gram, else the t
+00007ae0: 6d70 6469 7220 776f 756c 6420 6265 206b  mpdir would be k
+00007af0: 6570 740a 2d6e 2c20 2d2d 6e74 6872 6561  ept.-n, --nthrea
+00007b00: 6473 2c20 7468 6520 6e75 6d62 6572 206f  ds, the number o
+00007b10: 6620 7468 7265 6164 7320 746f 2075 7365  f threads to use
+00007b20: 2c20 6465 6661 756c 7420 340a 2d2d 746f  , default 4.--to
+00007b30: 5f73 746f 702c 2066 6c61 6720 6f70 7469  _stop, flag opti
+00007b40: 6f6e 2c20 7768 6574 6865 7220 746f 2074  on, whether to t
+00007b50: 7261 6e73 6c61 7465 2074 6872 6f75 6768  ranslate through
+00007b60: 2053 544f 5020 636f 646f 6e73 2c20 6966   STOP codons, if
+00007b70: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00007b80: 742c 2074 7261 6e73 6c61 7469 6f6e 2077  t, translation w
+00007b90: 696c 6c20 6265 2074 6572 6d69 6e61 7465  ill be terminate
+00007ba0: 6420 6174 2074 6865 2066 6972 7374 2069  d at the first i
+00007bb0: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
+00007bc0: 6f6e 2c20 656c 7365 2061 2066 756c 6c20  on, else a full 
+00007bd0: 7472 616e 736c 6174 696f 6e20 636f 6e74  translation cont
+00007be0: 696e 7569 6e67 206f 6e20 7061 7374 2061  inuing on past a
+00007bf0: 6e79 2073 746f 7020 636f 646f 6e73 2077  ny stop codons w
+00007c00: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00007c10: 640a 2d2d 6364 732c 2066 6c61 6720 6f70  d.--cds, flag op
+00007c20: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00007c30: 206f 6e6c 7920 7472 616e 736c 6174 6520   only translate 
+00007c40: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
+00007c50: 2074 6861 7420 7374 6172 7473 2077 6974   that starts wit
+00007c60: 6820 6120 7661 6c69 6420 7374 6172 7420  h a valid start 
+00007c70: 636f 646f 6e20 616e 6420 6f6e 6c79 2063  codon and only c
+00007c80: 6f6e 7461 696e 7320 6120 7369 6e67 6c65  ontains a single
+00007c90: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
+00007ca0: 6f64 6f6e 2061 7420 7468 6520 656e 6420  odon at the end 
+00007cb0: 616e 6420 6d75 7374 2062 6520 6469 7669  and must be divi
+00007cc0: 6461 626c 6520 6279 2074 6872 6565 2c20  dable by three, 
+00007cd0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00007ce0: 7365 742c 206f 6e6c 7920 7468 6520 636f  set, only the co
+00007cf0: 6d70 6c65 7465 2043 4453 2077 6f75 6c64  mplete CDS would
+00007d00: 2062 6520 7472 616e 736c 6174 6564 0a2d   be translated.-
+00007d10: 2d73 7472 6970 5f67 6170 732c 2066 6c61  -strip_gaps, fla
+00007d20: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00007d30: 7220 746f 2064 726f 7020 616c 6c20 6761  r to drop all ga
+00007d40: 7073 2069 6e20 6d75 6c74 6970 6c65 2073  ps in multiple s
+00007d50: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
+00007d60: 742c 2069 6620 7468 6520 666c 6167 2077  t, if the flag w
+00007d70: 6173 2073 6574 2c20 616c 6c20 6761 7073  as set, all gaps
+00007d80: 2077 6f75 6c64 2062 6520 6472 6f70 7065   would be droppe
+00007d90: 640a 2d61 2c20 2d2d 616c 6967 6e65 722c  d.-a, --aligner,
+00007da0: 2077 6869 6368 2061 6c69 676e 6d65 6e74   which alignment
+00007db0: 2070 726f 6772 616d 2074 6f20 7573 652c   program to use,
+00007dc0: 2064 6566 6175 6c74 206d 6166 6674 0a2d   default mafft.-
+00007dd0: 7472 6565 2c20 2d2d 7472 6565 5f6d 6574  tree, --tree_met
+00007de0: 686f 642c 2077 6869 6368 2067 656e 6520  hod, which gene 
+00007df0: 7472 6565 2069 6e66 6572 656e 6365 2070  tree inference p
+00007e00: 726f 6772 616d 2074 6f20 696e 766f 6b65  rogram to invoke
+00007e10: 2c20 6465 6661 756c 7420 6661 7374 7472  , default fasttr
+00007e20: 6565 0a2d 7473 2c20 2d2d 7472 6565 7365  ee.-ts, --treese
+00007e30: 742c 2074 6865 2070 6172 616d 6574 6572  t, the parameter
+00007e40: 7320 7365 7474 696e 6720 666f 7220 6765  s setting for ge
+00007e50: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
+00007e60: 652c 2064 6566 6175 6c74 204e 6f6e 652c  e, default None,
+00007e70: 2074 6869 7320 6f70 7469 6f6e 2063 616e   this option can
+00007e80: 2062 6520 7072 6f76 6964 6564 206d 756c   be provided mul
+00007e90: 7469 706c 6520 7469 6d65 730a 2d2d 636f  tiple times.--co
+00007ea0: 6e63 6174 656e 6174 696f 6e2c 2066 6c61  ncatenation, fla
+00007eb0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00007ec0: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+00007ed0: 6520 636f 6e63 6174 656e 6174 696f 6e2d  e concatenation-
+00007ee0: 6261 7365 6420 7370 6563 6965 7320 7472  based species tr
+00007ef0: 6565 2069 6e66 6572 656e 6365 2c20 6966  ee inference, if
+00007f00: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00007f10: 742c 2063 6f6e 6361 7465 6e61 7469 6f6e  t, concatenation
+00007f20: 2d62 6173 6564 2073 7065 6369 6573 2074  -based species t
+00007f30: 7265 6520 776f 756c 6420 6265 2069 6e66  ree would be inf
+00007f40: 6572 6564 0a2d 2d63 6f61 6c65 7363 656e  ered.--coalescen
+00007f50: 6365 2c20 666c 6167 206f 7074 696f 6e2c  ce, flag option,
+00007f60: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
+00007f70: 6961 7465 2074 6865 2063 6f61 6c65 7363  iate the coalesc
+00007f80: 656e 6365 2d62 6173 6564 2073 7065 6369  ence-based speci
+00007f90: 6573 2074 7265 6520 696e 6665 7265 6e63  es tree inferenc
+00007fa0: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
+00007fb0: 6173 2073 6574 2c20 636f 616c 6573 6365  as set, coalesce
+00007fc0: 6e63 652d 6261 7365 6420 7370 6563 6965  nce-based specie
+00007fd0: 7320 7472 6565 2077 6f75 6c64 2062 6520  s tree would be 
+00007fe0: 696e 6665 7265 640a 2d73 702c 202d 2d73  infered.-sp, --s
+00007ff0: 7065 6369 6573 7472 6565 2c20 7370 6563  peciestree, spec
+00008000: 6965 7320 7472 6565 2064 6174 6166 696c  ies tree datafil
+00008010: 6520 666f 7220 6461 7469 6e67 2c20 6465  e for dating, de
+00008020: 6661 756c 7420 4e6f 6e65 0a2d 642c 202d  fault None.-d, -
+00008030: 2d64 6174 696e 672c 2077 6869 6368 206d  -dating, which m
+00008040: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
+00008050: 7072 6f67 7261 6d20 746f 2075 7365 2c20  program to use, 
+00008060: 6465 6661 756c 7420 6e6f 6e65 0a2d 6473  default none.-ds
+00008070: 2c20 2d2d 6461 7469 6e67 7365 742c 2074  , --datingset, t
+00008080: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
+00008090: 7474 696e 6720 666f 7220 6461 7469 6e67  tting for dating
+000080a0: 2070 726f 6772 616d 2c20 6465 6661 756c   program, defaul
+000080b0: 7420 4e6f 6e65 2c20 7468 6973 206f 7074  t None, this opt
+000080c0: 696f 6e20 6361 6e20 6265 2070 726f 7669  ion can be provi
+000080d0: 6465 6420 6d75 6c74 6970 6c65 2074 696d  ded multiple tim
+000080e0: 6573 0a2d 6e73 2c20 2d2d 6e73 6974 6573  es.-ns, --nsites
+000080f0: 2c20 7468 6520 6e73 6974 6573 2069 6e66  , the nsites inf
+00008100: 6f72 6d61 7469 6f6e 2066 6f72 2072 3873  ormation for r8s
+00008110: 2064 6174 696e 672c 2064 6566 6175 6c74   dating, default
+00008120: 204e 6f6e 650a 2d6f 742c 202d 2d6f 7574   None.-ot, --out
+00008130: 6772 6f75 702c 2074 6865 206f 7574 6772  group, the outgr
+00008140: 6f75 7020 696e 666f 726d 6174 696f 6e20  oup information 
+00008150: 666f 7220 7238 7320 6461 7469 6e67 2c20  for r8s dating, 
+00008160: 6465 6661 756c 7420 4e6f 6e65 0a2d 7074  default None.-pt
+00008170: 2c20 2d2d 7061 7274 6974 696f 6e2c 2066  , --partition, f
+00008180: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00008190: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+000081a0: 7061 7274 6974 696f 6e20 6461 7469 6e67  partition dating
+000081b0: 2061 6e61 6c79 7369 7320 666f 7220 636f   analysis for co
+000081c0: 646f 6e2c 2069 6620 7468 6520 666c 6167  don, if the flag
+000081d0: 2077 6173 2073 6574 2c20 616e 2061 6464   was set, an add
+000081e0: 6974 696f 6e61 6c20 7061 7274 6974 696f  itional partitio
+000081f0: 6e20 6461 7469 6e67 2061 6e61 6c79 7369  n dating analysi
+00008200: 7320 776f 756c 6420 6265 2069 6e69 7469  s would be initi
+00008210: 6174 6564 0a2d 616d 2c20 2d2d 6161 6d6f  ated.-am, --aamo
+00008220: 6465 6c2c 2077 6869 6368 2070 726f 7465  del, which prote
+00008230: 696e 206d 6f64 656c 2074 6f20 6265 2075  in model to be u
+00008240: 7365 6420 696e 206d 636d 6374 7265 652c  sed in mcmctree,
+00008250: 2064 6566 6175 6c74 2070 6f69 7373 6f6e   default poisson
+00008260: 0a2d 6b73 2c20 666c 6167 206f 7074 696f  .-ks, flag optio
+00008270: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00008280: 6974 6961 7465 204b 7320 6361 6c63 756c  itiate Ks calcul
+00008290: 6174 696f 6e20 666f 7220 686f 6d6f 6c6f  ation for homolo
+000082a0: 6775 6573 2069 6e20 7468 6520 7072 6f76  gues in the prov
+000082b0: 6964 6564 206f 7274 686f 6c6f 676f 7573  ided orthologous
+000082c0: 2067 656e 6520 6661 6d69 6c79 0a2d 2d61   gene family.--a
+000082d0: 6e6e 6f74 6174 696f 6e2c 2077 6869 6368  nnotation, which
+000082e0: 2061 6e6e 6f74 6174 696f 6e20 7072 6f67   annotation prog
+000082f0: 7261 6d20 746f 2075 7365 2c20 6465 6661  ram to use, defa
+00008300: 756c 7420 4e6f 6e65 0a2d 2d70 6169 7277  ult None.--pairw
+00008310: 6973 652c 2066 6c61 6720 6f70 7469 6f6e  ise, flag option
+00008320: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
+00008330: 7469 6174 6520 7061 6972 7769 7365 204b  tiate pairwise K
+00008340: 7320 6573 7469 6d61 7469 6f6e 2c20 6966  s estimation, if
+00008350: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00008360: 742c 2070 6169 7277 6973 6520 4b73 2076  t, pairwise Ks v
+00008370: 616c 7565 7320 776f 756c 6420 6265 2065  alues would be e
+00008380: 7374 696d 6174 6564 0a2d 6564 2c20 2d2d  stimated.-ed, --
+00008390: 6567 676e 6f67 6461 7461 2c20 7468 6520  eggnogdata, the 
+000083a0: 6567 676e 6f67 2061 6e6e 6f74 6174 696f  eggnog annotatio
+000083b0: 6e20 6461 7461 6669 6c65 2c20 6465 6661  n datafile, defa
+000083c0: 756c 7420 4e6f 6e65 0a2d 2d70 6661 6d2c  ult None.--pfam,
+000083d0: 2077 6869 6368 206f 7074 696f 6e20 746f   which option to
+000083e0: 2075 7365 2066 6f72 2070 6661 6d20 616e   use for pfam an
+000083f0: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
+00008400: 7420 4e6f 6e65 0a2d 2d64 6d6e 622c 2074  t None.--dmnb, t
+00008410: 6865 2064 6961 6d6f 6e64 2064 6174 6162  he diamond datab
+00008420: 6173 6520 666f 7220 616e 6e6f 7461 7469  ase for annotati
+00008430: 6f6e 2c20 6465 6661 756c 7420 4e6f 6e65  on, default None
+00008440: 0a2d 2d68 6d6d 2c20 7468 6520 484d 4d20  .--hmm, the HMM 
+00008450: 7072 6f66 696c 6520 666f 7220 616e 6e6f  profile for anno
+00008460: 7461 7469 6f6e 2c20 6465 6661 756c 7420  tation, default 
+00008470: 4e6f 6e65 0a2d 2d65 7661 6c75 652c 2074  None.--evalue, t
+00008480: 6865 2065 2d76 616c 7565 2063 7574 2d6f  he e-value cut-o
+00008490: 6666 2066 6f72 2061 6e6e 6f74 6174 696f  ff for annotatio
+000084a0: 6e2c 2064 6566 6175 6c74 2031 652d 3130  n, default 1e-10
+000084b0: 0a2d 2d65 7865 7061 7468 2c20 7468 6520  .--exepath, the 
+000084c0: 7061 7468 2074 6f20 7468 6520 696e 7465  path to the inte
+000084d0: 7270 726f 7363 616e 2065 7865 6375 7461  rproscan executa
+000084e0: 626c 652c 2064 6566 6175 6c74 204e 6f6e  ble, default Non
+000084f0: 650a 2d66 2c20 2d2d 666f 7373 696c 2c20  e.-f, --fossil, 
+00008500: 7468 6520 666f 7373 696c 2063 616c 6962  the fossil calib
+00008510: 7261 7469 6f6e 2069 6e66 6f72 6d61 7469  ration informati
+00008520: 6f6e 2069 6e20 4265 6173 742c 2064 6566  on in Beast, def
+00008530: 6175 6c74 2028 2763 6c61 6465 313b 636c  ault ('clade1;cl
+00008540: 6164 6532 272c 2027 7461 7861 312c 7461  ade2', 'taxa1,ta
+00008550: 7861 323b 7461 7861 332c 7461 7861 3427  xa2;taxa3,taxa4'
+00008560: 2c20 2734 3b35 272c 2027 302e 353b 302e  , '4;5', '0.5;0.
+00008570: 3627 2c20 2734 3030 3b35 3030 2729 0a2d  6', '400;500').-
+00008580: 7268 2c20 2d2d 726f 6f74 6865 6967 6874  rh, --rootheight
+00008590: 2c20 7468 6520 726f 6f74 2068 6569 6768  , the root heigh
+000085a0: 7420 6361 6c69 6272 6174 696f 6e20 696e  t calibration in
+000085b0: 666f 2069 6e20 4265 6173 742c 2064 6566  fo in Beast, def
+000085c0: 6175 6c74 2028 342c 302e 352c 3430 3029  ault (4,0.5,400)
+000085d0: 0a2d 6373 2c20 2d2d 6368 6169 6e73 6574  .-cs, --chainset
+000085e0: 2c20 7468 6520 7061 7261 6d65 7465 7273  , the parameters
+000085f0: 206f 6620 4d43 4d43 2063 6861 696e 2069   of MCMC chain i
+00008600: 6e20 4265 6173 742c 2064 6566 6175 6c74  n Beast, default
+00008610: 2028 3130 3030 302c 3130 3029 0a2d 2d62   (10000,100).--b
+00008620: 6561 7374 6c67 6a61 722c 2074 6865 2070  eastlgjar, the p
+00008630: 6174 6820 746f 2062 6561 7374 4c47 2e6a  ath to beastLG.j
+00008640: 6172 2c20 6465 6661 756c 7420 4e6f 6e65  ar, default None
+00008650: 0a2d 2d62 6561 676c 652c 2066 6c61 6720  .--beagle, flag 
+00008660: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00008670: 746f 2075 7365 2062 6561 676c 6520 696e  to use beagle in
+00008680: 2042 6561 7374 2c20 6966 2074 6865 2066   Beast, if the f
+00008690: 6c61 6720 7761 7320 7365 742c 2062 6561  lag was set, bea
+000086a0: 676c 6520 776f 756c 6420 6265 2075 7365  gle would be use
+000086b0: 640a 2d2d 7072 6f74 636f 6364 6174 696e  d.--protcocdatin
+000086c0: 672c 2066 6c61 6720 6f70 7469 6f6e 2c20  g, flag option, 
+000086d0: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
+000086e0: 696e 6974 6961 7465 2074 6865 2070 726f  initiate the pro
+000086f0: 7465 696e 2d63 6f6e 6361 7465 6e61 7469  tein-concatenati
+00008700: 6f6e 2d62 6173 6564 2064 6174 696e 6720  on-based dating 
+00008710: 616e 616c 7973 6973 2c20 6966 2074 6865  analysis, if the
+00008720: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00008730: 6865 2061 6e61 6c79 7369 7320 776f 756c  he analysis woul
+00008740: 6420 6265 2069 6e69 7469 6174 6564 0a2d  d be initiated.-
+00008750: 2d70 726f 7464 6174 696e 672c 2066 6c61  -protdating, fla
+00008760: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00008770: 7220 746f 206f 6e6c 7920 696e 6974 6961  r to only initia
+00008780: 7465 2074 6865 2070 726f 7465 696e 2d62  te the protein-b
+00008790: 6173 6564 2064 6174 696e 6720 616e 616c  ased dating anal
+000087a0: 7973 6973 2c20 6966 2074 6865 2066 6c61  ysis, if the fla
+000087b0: 6720 7761 7320 7365 742c 2074 6865 2061  g was set, the a
+000087c0: 6e61 6c79 7369 7320 776f 756c 6420 6265  nalysis would be
+000087d0: 2069 6e69 7469 6174 6564 0a60 6060 0a0a   initiated.```..
+000087e0: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
+000087f0: 206b 7364 6020 6361 6e20 7265 616c 697a   ksd` can realiz
+00008800: 6520 7468 6520 636f 6e73 7472 7563 7469  e the constructi
+00008810: 6f6e 206f 6620 2a4b 2a3c 7375 623e 533c  on of *K*<sub>S<
+00008820: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00008830: 6275 7469 6f6e 2061 6e64 2072 6174 6520  bution and rate 
+00008840: 636f 7272 6563 7469 6f6e 2e0a 6060 600a  correction..```.
+00008850: 7767 6420 6b73 6420 6661 6d69 6c69 6573  wgd ksd families
+00008860: 2073 6571 7565 6e63 6573 2028 6f70 7469   sequences (opti
+00008870: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
+00008880: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00008890: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000088a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000088b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000088c0: 2d2d 2d2d 0a2d 6f2c 202d 2d6f 7574 6469  ----.-o, --outdi
+000088d0: 722c 2074 6865 206f 7574 7075 7420 6469  r, the output di
+000088e0: 7265 6374 6f72 792c 2064 6566 6175 6c74  rectory, default
+000088f0: 2077 6764 5f6b 7364 0a2d 742c 202d 2d74   wgd_ksd.-t, --t
+00008900: 6d70 6469 722c 2074 6865 2074 656d 706f  mpdir, the tempo
+00008910: 7261 7279 2077 6f72 6b69 6e67 2064 6972  rary working dir
+00008920: 6563 746f 7279 2c20 6465 6661 756c 7420  ectory, default 
+00008930: 4e6f 6e65 2c20 6966 204e 6f6e 6520 7761  None, if None wa
+00008940: 7320 6769 7665 6e2c 2074 6865 2074 6d70  s given, the tmp
+00008950: 6469 7220 7769 6c6c 2062 6520 6173 7369  dir will be assi
+00008960: 676e 6564 2072 616e 646f 6d20 6e61 6d65  gned random name
+00008970: 7320 696e 2063 7572 7265 6e74 2064 6972  s in current dir
+00008980: 6563 746f 7279 2061 6e64 2061 7574 6f6d  ectory and autom
+00008990: 6174 656c 7920 7265 6d6f 7665 6420 6174  ately removed at
+000089a0: 2074 6865 2063 6f6d 706c 6574 696f 6e20   the completion 
+000089b0: 6f66 2070 726f 6772 616d 2c20 656c 7365  of program, else
+000089c0: 2074 6865 2074 6d70 6469 7220 776f 756c   the tmpdir woul
+000089d0: 6420 6265 206b 6570 740a 2d6e 2c20 2d2d  d be kept.-n, --
+000089e0: 6e74 6872 6561 6473 2c20 7468 6520 6e75  nthreads, the nu
+000089f0: 6d62 6572 206f 6620 7468 7265 6164 7320  mber of threads 
+00008a00: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
+00008a10: 340a 2d2d 746f 5f73 746f 702c 2066 6c61  4.--to_stop, fla
+00008a20: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00008a30: 7220 746f 2074 7261 6e73 6c61 7465 2074  r to translate t
+00008a40: 6872 6f75 6768 2053 544f 5020 636f 646f  hrough STOP codo
+00008a50: 6e73 2c20 6966 2074 6865 2066 6c61 6720  ns, if the flag 
+00008a60: 7761 7320 7365 742c 2074 7261 6e73 6c61  was set, transla
+00008a70: 7469 6f6e 2077 696c 6c20 6265 2074 6572  tion will be ter
+00008a80: 6d69 6e61 7465 6420 6174 2074 6865 2066  minated at the f
+00008a90: 6972 7374 2069 6e2d 6672 616d 6520 7374  irst in-frame st
+00008aa0: 6f70 2063 6f64 6f6e 2c20 656c 7365 2061  op codon, else a
+00008ab0: 2066 756c 6c20 7472 616e 736c 6174 696f   full translatio
+00008ac0: 6e20 636f 6e74 696e 7569 6e67 206f 6e20  n continuing on 
+00008ad0: 7061 7374 2061 6e79 2073 746f 7020 636f  past any stop co
+00008ae0: 646f 6e73 2077 6f75 6c64 2062 6520 696e  dons would be in
+00008af0: 6974 6961 7465 640a 2d2d 6364 732c 2066  itiated.--cds, f
+00008b00: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00008b10: 6865 7220 746f 206f 6e6c 7920 7472 616e  her to only tran
+00008b20: 736c 6174 6520 7468 6520 636f 6d70 6c65  slate the comple
+00008b30: 7465 2043 4453 2074 6861 7420 7374 6172  te CDS that star
+00008b40: 7473 2077 6974 6820 6120 7661 6c69 6420  ts with a valid 
+00008b50: 7374 6172 7420 636f 646f 6e20 616e 6420  start codon and 
+00008b60: 6f6e 6c79 2063 6f6e 7461 696e 7320 6120  only contains a 
+00008b70: 7369 6e67 6c65 2069 6e2d 6672 616d 6520  single in-frame 
+00008b80: 7374 6f70 2063 6f64 6f6e 2061 7420 7468  stop codon at th
+00008b90: 6520 656e 6420 616e 6420 6d75 7374 2062  e end and must b
+00008ba0: 6520 6469 7669 6461 626c 6520 6279 2074  e dividable by t
+00008bb0: 6872 6565 2c20 6966 2074 6865 2066 6c61  hree, if the fla
+00008bc0: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
+00008bd0: 7468 6520 636f 6d70 6c65 7465 2043 4453  the complete CDS
+00008be0: 2077 6f75 6c64 2062 6520 7472 616e 736c   would be transl
+00008bf0: 6174 6564 0a2d 2d70 6169 7277 6973 652c  ated.--pairwise,
+00008c00: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00008c10: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+00008c20: 6520 7061 6972 7769 7365 204b 7320 6573  e pairwise Ks es
+00008c30: 7469 6d61 7469 6f6e 2c20 6966 2074 6865  timation, if the
+00008c40: 2066 6c61 6720 7761 7320 7365 742c 2070   flag was set, p
+00008c50: 6169 7277 6973 6520 4b73 2076 616c 7565  airwise Ks value
+00008c60: 7320 776f 756c 6420 6265 2065 7374 696d  s would be estim
+00008c70: 6174 6564 0a2d 2d73 7472 6970 5f67 6170  ated.--strip_gap
+00008c80: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
+00008c90: 7768 6574 6865 7220 746f 2064 726f 7020  whether to drop 
+00008ca0: 616c 6c20 6761 7073 2069 6e20 6d75 6c74  all gaps in mult
+00008cb0: 6970 6c65 2073 6571 7565 6e63 6520 616c  iple sequence al
+00008cc0: 6967 6e6d 656e 742c 2069 6620 7468 6520  ignment, if the 
+00008cd0: 666c 6167 2077 6173 2073 6574 2c20 616c  flag was set, al
+00008ce0: 6c20 6761 7073 2077 6f75 6c64 2062 6520  l gaps would be 
+00008cf0: 6472 6f70 7065 640a 2d61 2c20 2d2d 616c  dropped.-a, --al
+00008d00: 6967 6e65 722c 2077 6869 6368 2061 6c69  igner, which ali
+00008d10: 676e 6d65 6e74 2070 726f 6772 616d 2074  gnment program t
+00008d20: 6f20 7573 652c 2064 6566 6175 6c74 206d  o use, default m
+00008d30: 6166 6674 200a 2d74 7265 652c 202d 2d74  afft .-tree, --t
+00008d40: 7265 655f 6d65 7468 6f64 2c20 7768 6963  ree_method, whic
+00008d50: 6820 6765 6e65 2074 7265 6520 696e 6665  h gene tree infe
+00008d60: 7265 6e63 6520 7072 6f67 7261 6d20 746f  rence program to
+00008d70: 2069 6e76 6f6b 652c 2064 6566 6175 6c74   invoke, default
+00008d80: 2066 6173 7474 7265 650a 2d2d 7472 6565   fasttree.--tree
+00008d90: 5f6f 7074 696f 6e73 2c20 6f70 7469 6f6e  _options, option
+00008da0: 7320 696e 2074 7265 6520 696e 6665 7265  s in tree infere
+00008db0: 6e63 6520 6173 2061 2063 6f6d 6d61 2073  nce as a comma s
+00008dc0: 6570 6172 6174 6564 2073 7472 696e 672c  eparated string,
+00008dd0: 2064 6566 6175 6c74 204e 6f6e 650a 2d2d   default None.--
+00008de0: 6e6f 6465 5f61 7665 7261 6765 2c20 666c  node_average, fl
+00008df0: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
+00008e00: 6572 2074 6f20 696e 6974 6961 7465 206e  er to initiate n
+00008e10: 6f64 652d 6176 6572 6167 6520 7761 7920  ode-average way 
+00008e20: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
+00008e30: 2069 6e73 7465 6164 206f 6620 6e6f 6465   instead of node
+00008e40: 2d77 6569 6768 7465 642c 2069 6620 7468  -weighted, if th
+00008e50: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00008e60: 7468 6520 6e6f 6465 2d61 7665 7261 6769  the node-averagi
+00008e70: 6e67 2064 652d 7265 6475 6e64 616e 6379  ng de-redundancy
+00008e80: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00008e90: 7465 640a 2d73 722c 202d 2d73 7061 6972  ted.-sr, --spair
+00008ea0: 2c20 7468 6520 7370 6563 6965 7320 7061  , the species pa
+00008eb0: 6972 2074 6f20 6265 2070 6c6f 7474 6564  ir to be plotted
+00008ec0: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+00008ed0: 7468 6973 206f 7074 696f 6e20 6361 6e20  this option can 
+00008ee0: 6265 2070 726f 7669 6465 6420 6d75 6c74  be provided mult
+00008ef0: 6970 6c65 2074 696d 6573 0a2d 7370 2c20  iple times.-sp, 
+00008f00: 2d2d 7370 6563 6965 7374 7265 652c 2074  --speciestree, t
+00008f10: 6865 2073 7065 6369 6573 2074 7265 6520  he species tree 
+00008f20: 746f 2070 6572 666f 726d 2072 6174 6520  to perform rate 
+00008f30: 636f 7272 6563 7469 6f6e 2c20 6465 6661  correction, defa
+00008f40: 756c 7420 4e6f 6e65 2c20 6966 204e 6f6e  ult None, if Non
+00008f50: 6520 7761 7320 6769 7665 6e2c 2074 6865  e was given, the
+00008f60: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
+00008f70: 2061 6e61 6c79 7369 7320 776f 756c 6420   analysis would 
+00008f80: 6265 2063 616c 6c65 6420 6f66 660a 2d72  be called off.-r
+00008f90: 772c 202d 2d72 6577 6569 6768 742c 2066  w, --reweight, f
+00008fa0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00008fb0: 6865 7220 746f 2072 6563 616c 6375 6c61  her to recalcula
+00008fc0: 7465 2074 6865 2077 6569 6768 7420 7065  te the weight pe
+00008fd0: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
+00008fe0: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00008ff0: 7365 742c 2074 6865 2077 6569 6768 7420  set, the weight 
+00009000: 776f 756c 6420 6265 2072 6563 616c 6375  would be recalcu
+00009010: 6c61 7465 640a 2d6f 722c 202d 2d6f 6e6c  lated.-or, --onl
+00009020: 7972 6f6f 746f 7574 2c20 666c 6167 206f  yrootout, flag o
+00009030: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+00009040: 6f20 6f6e 6c79 2063 6f6e 6475 6374 2072  o only conduct r
+00009050: 6174 6520 636f 7272 6563 7469 6f6e 2075  ate correction u
+00009060: 7369 6e67 2074 6865 206f 7574 6772 6f75  sing the outgrou
+00009070: 7020 6174 2072 6f6f 7420 6173 206f 7574  p at root as out
+00009080: 6772 6f75 702c 2069 6620 7468 6520 666c  group, if the fl
+00009090: 6167 2077 6173 2073 6574 2c20 6f6e 6c79  ag was set, only
+000090a0: 2074 6865 206f 7574 6772 6f75 7020 6174   the outgroup at
+000090b0: 2072 6f6f 7420 776f 756c 6420 6265 2075   root would be u
+000090c0: 7365 6420 6173 206f 7574 6772 6f75 700a  sed as outgroup.
+000090d0: 2d65 706b 2c20 2d2d 6578 7472 6170 6172  -epk, --extrapar
+000090e0: 616e 6f6d 656b 732c 2065 7874 7261 2070  anomeks, extra p
+000090f0: 6172 616e 6f6d 6520 4b73 2064 6174 6120  aranome Ks data 
+00009100: 746f 2070 6c6f 7420 696e 2074 6865 206d  to plot in the m
+00009110: 6978 6564 204b 7320 6469 7374 7269 6275  ixed Ks distribu
+00009120: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
+00009130: 6e65 0a2d 6170 2c20 2d2d 616e 6368 6f72  ne.-ap, --anchor
+00009140: 706f 696e 7473 2c20 616e 6368 6f72 706f  points, anchorpo
+00009150: 696e 7473 2e74 7874 2066 696c 6520 746f  ints.txt file to
+00009160: 2070 6c6f 7420 616e 6368 6f72 204b 7320   plot anchor Ks 
+00009170: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
+00009180: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
+00009190: 6661 756c 7420 4e6f 6e65 0a2d 706b 2c20  fault None.-pk, 
+000091a0: 2d2d 706c 6f74 6b64 652c 2066 6c61 6720  --plotkde, flag 
+000091b0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000091c0: 746f 2070 6c6f 7420 6b64 6520 6375 7276  to plot kde curv
+000091d0: 6520 6f66 206f 7274 686f 6c6f 676f 7573  e of orthologous
+000091e0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+000091f0: 206f 7665 7220 6869 7374 6f67 7261 6d20   over histogram 
+00009200: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
+00009210: 6469 7374 7269 6275 7469 6f6e 2c20 6966  distribution, if
+00009220: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00009230: 742c 2074 6865 206b 6465 2063 7572 7665  t, the kde curve
+00009240: 2077 6f75 6c64 2062 6520 706c 6f74 7465   would be plotte
+00009250: 640a 2d70 6167 2c20 2d2d 706c 6f74 6170  d.-pag, --plotap
+00009260: 676d 6d2c 2066 6c61 6720 6f70 7469 6f6e  gmm, flag option
+00009270: 2c20 7768 6574 6865 7220 746f 2070 6572  , whether to per
+00009280: 666f 726d 2061 6e64 2070 6c6f 7420 6d69  form and plot mi
+00009290: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
+000092a0: 6620 616e 6368 6f72 204b 7320 696e 2074  f anchor Ks in t
+000092b0: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
+000092c0: 7269 6275 7469 6f6e 2c20 6966 2074 6865  ribution, if the
+000092d0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+000092e0: 6865 206d 6978 7475 7265 206d 6f64 656c  he mixture model
+000092f0: 696e 6720 6f66 2061 6e63 686f 7220 4b73  ing of anchor Ks
+00009300: 2077 6f75 6c64 2062 6520 706c 6f74 7465   would be plotte
+00009310: 640a 2d70 656d 2c20 2d2d 706c 6f74 656c  d.-pem, --plotel
+00009320: 6d6d 2c20 666c 6167 206f 7074 696f 6e2c  mm, flag option,
+00009330: 2077 6865 7468 6572 2074 6f20 7065 7266   whether to perf
+00009340: 6f72 6d20 616e 6420 706c 6f74 2065 6c6d  orm and plot elm
+00009350: 6d20 6d69 7874 7572 6520 6d6f 6465 6c69  m mixture modeli
+00009360: 6e67 206f 6620 7061 7261 6e6f 6d65 204b  ng of paranome K
+00009370: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
+00009380: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00009390: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+000093a0: 7365 742c 2074 6865 2065 6c6d 6d20 6d69  set, the elmm mi
+000093b0: 7874 7572 6520 6d6f 6465 6c69 6e67 206f  xture modeling o
+000093c0: 6620 7061 7261 6e6f 6d65 204b 7320 776f  f paranome Ks wo
+000093d0: 756c 6420 6265 2070 6c6f 7474 6564 0a2d  uld be plotted.-
+000093e0: 632c 202d 2d63 6f6d 706f 6e65 6e74 732c  c, --components,
+000093f0: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
+00009400: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
+00009410: 6f6e 656e 7473 2074 6f20 6669 7420 696e  onents to fit in
+00009420: 2061 6e63 686f 7220 4b73 206d 6978 7475   anchor Ks mixtu
+00009430: 7265 206d 6f64 656c 696e 672c 2064 6566  re modeling, def
+00009440: 6175 6c74 2028 312c 3429 0a2d 786c 2c20  ault (1,4).-xl, 
+00009450: 2d2d 786c 696d 2c20 7468 6520 7820 6178  --xlim, the x ax
+00009460: 6973 206c 696d 6974 206f 6620 4b73 2064  is limit of Ks d
+00009470: 6973 7472 6962 7574 696f 6e0a 2d79 6c2c  istribution.-yl,
+00009480: 202d 2d79 6c69 6d2c 2074 6865 2079 2061   --ylim, the y a
+00009490: 7869 7320 6c69 6d69 7420 6f66 204b 7320  xis limit of Ks 
+000094a0: 6469 7374 7269 6275 7469 6f6e 0a2d 6164  distribution.-ad
+000094b0: 6f2c 202d 2d61 646a 7573 746f 7274 686f  o, --adjustortho
+000094c0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+000094d0: 6865 7468 6572 2074 6f20 6164 6a75 7374  hether to adjust
+000094e0: 2074 6865 2068 6973 746f 6772 616d 2068   the histogram h
+000094f0: 6569 6768 7420 6f66 206f 7274 686f 6c6f  eight of ortholo
+00009500: 676f 7573 204b 7320 6173 2074 6f20 6d61  gous Ks as to ma
+00009510: 7463 6820 7468 6520 6865 6967 6874 206f  tch the height o
+00009520: 6620 7061 7261 6c6f 676f 7573 204b 732c  f paralogous Ks,
+00009530: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00009540: 2073 6574 2c20 7468 6520 6164 6a75 7374   set, the adjust
+00009550: 6d65 6e74 2077 6f75 6c64 2062 6520 636f  ment would be co
+00009560: 6e64 7563 7465 640a 2d61 6466 2c20 2d2d  nducted.-adf, --
+00009570: 6164 6a75 7374 6661 6374 6f72 2c20 7468  adjustfactor, th
+00009580: 6520 6164 6a75 7374 6d65 6e74 2066 6163  e adjustment fac
+00009590: 746f 7220 6f66 206f 7274 686f 6c6f 676f  tor of orthologo
+000095a0: 7573 204b 732c 2064 6566 6175 6c74 2030  us Ks, default 0
+000095b0: 2e35 0a2d 6f61 2c20 2d2d 6f6b 616c 7068  .5.-oa, --okalph
+000095c0: 612c 2074 6865 206f 7061 6369 7479 206f  a, the opacity o
+000095d0: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
+000095e0: 2064 6973 7472 6962 7574 696f 6e20 696e   distribution in
+000095f0: 206d 6978 6564 2070 6c6f 742c 2064 6566   mixed plot, def
+00009600: 6175 6c74 2030 2e35 0a2d 6661 2c20 2d2d  ault 0.5.-fa, --
+00009610: 666f 6375 7332 616c 6c2c 2073 6574 2066  focus2all, set f
+00009620: 6f63 616c 2073 7065 6369 6573 2061 6e64  ocal species and
+00009630: 206c 6574 2073 7065 6369 6573 2070 6169   let species pai
+00009640: 7220 746f 2062 6520 6265 7477 6565 6e20  r to be between 
+00009650: 666f 6361 6c20 616e 6420 616c 6c20 7468  focal and all th
+00009660: 6520 7265 6d61 696e 696e 6720 7370 6563  e remaining spec
+00009670: 6965 732c 2064 6566 6175 6c74 204e 6f6e  ies, default Non
+00009680: 650a 2d6b 732c 202d 2d6b 7374 7265 652c  e.-ks, --kstree,
+00009690: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+000096a0: 6574 6865 7220 746f 2069 6e66 6572 204b  ether to infer K
+000096b0: 7320 7472 6565 2c20 6966 2074 6865 2066  s tree, if the f
+000096c0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+000096d0: 204b 7320 7472 6565 2069 6e66 6572 656e   Ks tree inferen
+000096e0: 6365 2061 6e61 6c79 7369 7320 776f 756c  ce analysis woul
+000096f0: 6420 6265 2069 6e69 7469 6174 6564 0a2d  d be initiated.-
+00009700: 6f63 6b2c 202d 2d6f 6e6c 7963 6f6e 6361  ock, --onlyconca
+00009710: 746b 7374 7265 652c 2066 6c61 6720 6f70  tkstree, flag op
+00009720: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00009730: 206f 6e6c 7920 696e 6665 7220 4b73 2074   only infer Ks t
+00009740: 7265 6520 756e 6465 7220 636f 6e63 6174  ree under concat
+00009750: 656e 6174 6564 2061 6c69 676e 6d65 6e74  enated alignment
+00009760: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00009770: 7320 7365 742c 206f 6e6c 7920 7468 6520  s set, only the 
+00009780: 4b73 2074 7265 6520 756e 6465 7220 636f  Ks tree under co
+00009790: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
+000097a0: 6d65 6e74 2077 6f75 6c64 2062 6520 6361  ment would be ca
+000097b0: 6c63 756c 6174 6564 0a2d 6373 2c20 2d2d  lculated.-cs, --
+000097c0: 636c 6173 7369 632c 2066 6c61 6720 6f70  classic, flag op
+000097d0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+000097e0: 2064 7261 7720 6d69 7865 6420 4b73 2070   draw mixed Ks p
+000097f0: 6c6f 7420 696e 2061 2063 6c61 7373 6963  lot in a classic
+00009800: 206d 616e 6e65 7220 7768 6572 6520 7468   manner where th
+00009810: 6520 6675 6c6c 206f 7274 686f 6c6f 676f  e full orthologo
+00009820: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
+00009830: 6f6e 2069 7320 6472 6177 6564 2c20 6966  on is drawed, if
+00009840: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
+00009850: 742c 2074 6865 2063 6c61 7373 6963 206d  t, the classic m
+00009860: 6978 6564 204b 7320 706c 6f74 2077 6f75  ixed Ks plot wou
+00009870: 6c64 2062 6520 6472 6177 6e0a 2d74 612c  ld be drawn.-ta,
+00009880: 202d 2d74 6f70 6172 726f 772c 2066 6c61   --toparrow, fla
+00009890: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+000098a0: 7220 746f 2061 646a 7573 7420 7468 6520  r to adjust the 
+000098b0: 6172 726f 7720 746f 2062 6520 6174 2074  arrow to be at t
+000098c0: 6865 2074 6f70 206f 6620 7468 6520 706c  he top of the pl
+000098d0: 6f74 2c20 696e 7374 6561 6420 6f66 2062  ot, instead of b
+000098e0: 6569 6e67 2063 6f6f 7264 696e 6174 6564  eing coordinated
+000098f0: 2061 7320 7468 6520 4b44 4520 6f66 2074   as the KDE of t
+00009900: 6865 206f 7274 686f 6c6f 676f 7573 204b  he orthologous K
+00009910: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00009920: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00009930: 7365 742c 2074 6865 2061 7272 6f77 2077  set, the arrow w
+00009940: 6f75 6c64 2062 6520 7365 7420 6174 2074  ould be set at t
+00009950: 6865 2074 6f70 0a2d 6273 2c20 2d2d 626f  he top.-bs, --bo
+00009960: 6f74 7374 7261 702c 2074 6865 206e 756d  otstrap, the num
+00009970: 6265 7220 6f66 2062 6f6f 7473 7472 6170  ber of bootstrap
+00009980: 2072 6570 6c69 6361 7465 7320 6f66 206f   replicates of o
+00009990: 7274 686f 6c6f 6720 4b73 2064 6973 7472  rtholog Ks distr
+000099a0: 6962 7574 696f 6e20 696e 206d 6978 6564  ibution in mixed
+000099b0: 2070 6c6f 740a 6060 600a 0a54 6865 2070   plot.```..The p
+000099c0: 726f 6772 616d 2060 7767 6420 6d69 7860  rogram `wgd mix`
+000099d0: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+000099e0: 206d 6978 7475 7265 206d 6f64 656c 2063   mixture model c
+000099f0: 6c75 7374 6572 696e 6720 616e 616c 7973  lustering analys
+00009a00: 6973 206f 6620 2a4b 2a3c 7375 623e 533c  is of *K*<sub>S<
+00009a10: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00009a20: 6275 7469 6f6e 2e0a 6060 600a 7767 6420  bution..```.wgd 
+00009a30: 6d69 7820 6b73 5f64 6174 6166 696c 6520  mix ks_datafile 
+00009a40: 286f 7074 696f 6e29 0a2d 2d2d 2d2d 2d2d  (option).-------
+00009a50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009a60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009a70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009a80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009a90: 2d2d 2d2d 2d2d 2d2d 2d0a 2d66 2c20 2d2d  ---------.-f, --
+00009aa0: 6669 6c74 6572 732c 2074 6865 2063 7574  filters, the cut
+00009ab0: 6f66 6620 616c 6967 6e6d 656e 7420 6c65  off alignment le
+00009ac0: 6e67 7468 2c20 6465 6661 756c 7420 3330  ngth, default 30
+00009ad0: 300a 2d72 2c20 2d2d 6b73 5f72 616e 6765  0.-r, --ks_range
+00009ae0: 2c20 7468 6520 4b73 2072 616e 6765 2074  , the Ks range t
+00009af0: 6f20 6265 2063 6f6e 7369 6465 7265 642c  o be considered,
+00009b00: 2064 6566 6175 6c74 2028 302c 2035 290a   default (0, 5).
+00009b10: 2d62 2c20 2d2d 6269 6e73 2c20 7468 6520  -b, --bins, the 
+00009b20: 6e75 6d62 6572 206f 6620 6269 6e73 2069  number of bins i
+00009b30: 6e20 4b73 2064 6973 7472 6962 7574 696f  n Ks distributio
+00009b40: 6e2c 2064 6566 6175 6c74 2035 300a 2d6f  n, default 50.-o
+00009b50: 2c20 2d2d 6f75 7464 6972 2c20 7468 6520  , --outdir, the 
+00009b60: 6f75 7470 7574 2064 6972 6563 746f 7279  output directory
+00009b70: 2c20 6465 6661 756c 7420 7767 645f 6d69  , default wgd_mi
+00009b80: 780a 2d2d 6d65 7468 6f64 2c20 7768 6963  x.--method, whic
+00009b90: 6820 6d69 7874 7572 6520 6d6f 6465 6c20  h mixture model 
+00009ba0: 746f 2075 7365 2c20 6465 6661 756c 7420  to use, default 
+00009bb0: 676d 6d0a 2d6e 2c20 2d2d 636f 6d70 6f6e  gmm.-n, --compon
+00009bc0: 656e 7473 2c20 7468 6520 7261 6e67 6520  ents, the range 
+00009bd0: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
+00009be0: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
+00009bf0: 6974 2c20 6465 6661 756c 7420 2831 2c20  it, default (1, 
+00009c00: 3429 0a2d 672c 202d 2d67 616d 6d61 2c20  4).-g, --gamma, 
+00009c10: 7468 6520 6761 6d6d 6120 7061 7261 6d65  the gamma parame
+00009c20: 7465 7220 666f 7220 6267 6d6d 206d 6f64  ter for bgmm mod
+00009c30: 656c 732c 2064 6566 6175 6c74 2030 2e30  els, default 0.0
+00009c40: 3031 0a2d 6e69 2c20 2d2d 6e5f 696e 6974  01.-ni, --n_init
+00009c50: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+00009c60: 6b2d 6d65 616e 7320 696e 6974 6961 6c69  k-means initiali
+00009c70: 7a61 7469 6f6e 732c 2064 6566 6175 6c74  zations, default
+00009c80: 2032 3030 0a2d 6d69 2c20 2d2d 6d61 785f   200.-mi, --max_
+00009c90: 6974 6572 2c20 7468 6520 6d61 7869 6d75  iter, the maximu
+00009ca0: 6d20 6e75 6d62 6572 206f 6620 6974 6572  m number of iter
+00009cb0: 6174 696f 6e73 2c20 6465 6661 756c 7420  ations, default 
+00009cc0: 3230 300a 6060 600a 0a54 6865 2070 726f  200.```..The pro
+00009cd0: 6772 616d 2060 7767 6420 7065 616b 6020  gram `wgd peak` 
+00009ce0: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
+00009cf0: 7365 6172 6368 206f 6620 6372 6564 6961  search of credia
+00009d00: 626c 6520 2a4b 2a3c 7375 623e 533c 2f73  ble *K*<sub>S</s
+00009d10: 7562 3e20 7261 6e67 6520 7573 6564 2069  ub> range used i
+00009d20: 6e20 5747 4420 6461 7469 6e67 2e0a 6060  n WGD dating..``
+00009d30: 600a 7767 6420 7065 616b 206b 735f 6461  `.wgd peak ks_da
+00009d40: 7461 6669 6c65 2028 6f70 7469 6f6e 290a  tafile (option).
+00009d50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009d60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009d70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009d80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009d90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009da0: 0a2d 6170 2c20 2d2d 616e 6368 6f72 706f  .-ap, --anchorpo
+00009db0: 696e 7473 2c20 7468 6520 616e 6368 6f72  ints, the anchor
+00009dc0: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
+00009dd0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00009de0: 736d 2c20 2d2d 7365 676d 656e 7473 2c20  sm, --segments, 
+00009df0: 7468 6520 7365 676d 656e 7473 2064 6174  the segments dat
+00009e00: 6166 696c 652c 2064 6566 6175 6c74 204e  afile, default N
+00009e10: 6f6e 650a 2d6c 652c 202d 2d6c 6973 7465  one.-le, --liste
+00009e20: 6c65 6d65 6e74 732c 2074 6865 206c 6973  lements, the lis
+00009e30: 7473 6567 6d65 6e74 7320 6461 7461 6669  tsegments datafi
+00009e40: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+00009e50: 200a 2d6d 702c 202d 2d6d 756c 7469 706c   .-mp, --multipl
+00009e60: 6963 6f6e 7061 6972 732c 2074 6865 206d  iconpairs, the m
+00009e70: 756c 7469 706c 6963 6f6e 7061 6972 7320  ultipliconpairs 
+00009e80: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+00009e90: 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574  t None.-o, --out
+00009ea0: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
+00009eb0: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
+00009ec0: 6c74 2077 6764 5f70 6561 6b0a 2d61 662c  lt wgd_peak.-af,
+00009ed0: 202d 2d61 6c69 676e 6669 6c74 6572 2c20   --alignfilter, 
+00009ee0: 6375 746f 6666 2066 6f72 2061 6c69 676e  cutoff for align
+00009ef0: 6d65 6e74 2069 6465 6e74 6974 792c 206c  ment identity, l
+00009f00: 656e 6774 6820 616e 6420 636f 7665 7261  ength and covera
+00009f10: 6765 2c20 6465 6661 756c 7420 302e 302c  ge, default 0.0,
+00009f20: 2030 2c20 302e 300a 2d72 2c20 2d2d 6b73   0, 0.0.-r, --ks
+00009f30: 7261 6e67 652c 2072 616e 6765 206f 6620  range, range of 
+00009f40: 4b73 2074 6f20 6265 2061 6e61 6c79 7a65  Ks to be analyze
+00009f50: 642c 2064 6566 6175 6c74 2028 302c 2035  d, default (0, 5
+00009f60: 290a 2d62 772c 202d 2d62 696e 5f77 6964  ).-bw, --bin_wid
+00009f70: 7468 2c20 6261 6e64 7769 6474 6820 6f66  th, bandwidth of
+00009f80: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00009f90: 2c20 6465 6661 756c 7420 302e 310a 2d69  , default 0.1.-i
+00009fa0: 632c 202d 2d77 6569 6768 7473 5f6f 7574  c, --weights_out
+00009fb0: 6c69 6572 735f 696e 636c 7564 6564 2c20  liers_included, 
+00009fc0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00009fd0: 7468 6572 2074 6f20 696e 636c 7564 6520  ther to include 
+00009fe0: 4b73 206f 7574 6c69 6572 732c 2069 6620  Ks outliers, if 
+00009ff0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+0000a000: 2c20 4b73 206f 7574 6c69 6572 7320 776f  , Ks outliers wo
+0000a010: 756c 6420 6265 2069 6e63 6c75 6465 6420  uld be included 
+0000a020: 696e 2074 6865 2061 6e61 6c79 7369 730a  in the analysis.
+0000a030: 2d6d 2c20 2d2d 6d65 7468 6f64 2c20 7768  -m, --method, wh
+0000a040: 6963 6820 6d69 7874 7572 6520 6d6f 6465  ich mixture mode
+0000a050: 6c20 746f 2075 7365 2c20 6465 6661 756c  l to use, defaul
+0000a060: 7420 676d 6d0a 2d2d 7365 6564 2c20 7261  t gmm.--seed, ra
+0000a070: 6e64 6f6d 2073 6565 6420 6769 7665 6e20  ndom seed given 
+0000a080: 746f 2069 6e69 7469 616c 697a 6174 696f  to initializatio
+0000a090: 6e2c 2064 6566 6175 6c74 2032 3335 3238  n, default 23528
+0000a0a0: 3930 0a2d 6569 2c20 2d2d 656d 5f69 7465  90.-ei, --em_ite
+0000a0b0: 722c 2074 6865 206e 756d 6265 7220 6f66  r, the number of
+0000a0c0: 2045 4d20 6974 6572 6174 696f 6e73 2074   EM iterations t
+0000a0d0: 6f20 7065 7266 6f72 6d2c 2064 6566 6175  o perform, defau
+0000a0e0: 6c74 2032 3030 0a2d 6e69 2c20 2d2d 6e5f  lt 200.-ni, --n_
+0000a0f0: 696e 6974 2c20 7468 6520 6e75 6d62 6572  init, the number
+0000a100: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
+0000a110: 6961 6c69 7a61 7469 6f6e 732c 2064 6566  ializations, def
+0000a120: 6175 6c74 2032 3030 0a2d 6e2c 202d 2d63  ault 200.-n, --c
+0000a130: 6f6d 706f 6e65 6e74 732c 2074 6865 2072  omponents, the r
+0000a140: 616e 6765 206f 6620 7468 6520 6e75 6d62  ange of the numb
+0000a150: 6572 206f 6620 636f 6d70 6f6e 656e 7473  er of components
+0000a160: 2074 6f20 6669 742c 2064 6566 6175 6c74   to fit, default
+0000a170: 2028 312c 2034 290a 2d67 2c20 2d2d 6761   (1, 4).-g, --ga
+0000a180: 6d6d 612c 2074 6865 2067 616d 6d61 2070  mma, the gamma p
+0000a190: 6172 616d 6574 6572 2066 6f72 2062 676d  arameter for bgm
+0000a1a0: 6d20 6d6f 6465 6c73 2c20 6465 6661 756c  m models, defaul
+0000a1b0: 7420 3165 2d33 0a2d 2d62 6f6f 7473 2c20  t 1e-3.--boots, 
+0000a1c0: 7468 6520 6e75 6d62 6572 206f 6620 626f  the number of bo
+0000a1d0: 6f74 7374 7261 7020 7265 706c 6963 6174  otstrap replicat
+0000a1e0: 6573 206f 6620 6b64 652c 2064 6566 6175  es of kde, defau
+0000a1f0: 6c74 2032 3030 0a2d 2d77 6569 6768 7465  lt 200.--weighte
+0000a200: 642c 2066 6c61 6720 6f70 7469 6f6e 2c20  d, flag option, 
+0000a210: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
+0000a220: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
+0000a230: 686f 6420 6f66 2064 652d 7265 6475 6e64  hod of de-redund
+0000a240: 616e 6379 2c20 6966 2074 6865 2066 6c61  ancy, if the fla
+0000a250: 6720 7761 7320 7365 742c 2074 6865 206e  g was set, the n
+0000a260: 6f64 652d 7765 6967 6874 6564 206d 6574  ode-weighted met
+0000a270: 686f 6420 776f 756c 6420 6265 2075 7365  hod would be use
+0000a280: 640a 2d70 2c20 2d2d 706c 6f74 2c20 7468  d.-p, --plot, th
+0000a290: 6520 706c 6f74 7469 6e67 206d 6574 686f  e plotting metho
+0000a2a0: 6420 746f 2062 6520 7573 6564 2c20 6465  d to be used, de
+0000a2b0: 6661 756c 7420 6964 656e 7469 6361 6c0a  fault identical.
+0000a2c0: 2d62 6d2c 202d 2d62 775f 6d65 7468 6f64  -bm, --bw_method
+0000a2d0: 2c20 7468 6520 6261 6e64 7769 6474 6820  , the bandwidth 
+0000a2e0: 6d65 7468 6f64 2074 6f20 6265 2075 7365  method to be use
+0000a2f0: 6420 696e 2061 6e61 6c79 7a69 6e67 2074  d in analyzing t
+0000a300: 6865 2070 6561 6b20 6f66 2057 4744 2064  he peak of WGD d
+0000a310: 6174 6573 2c20 6465 6661 756c 7420 7369  ates, default si
+0000a320: 6c76 6572 6d61 6e0a 2d2d 6e5f 6d65 646f  lverman.--n_medo
+0000a330: 6964 732c 2074 6865 206e 756d 6265 7220  ids, the number 
+0000a340: 6f66 206d 6564 6f69 6473 2074 6f20 6669  of medoids to fi
+0000a350: 742c 2064 6566 6175 6c74 2032 0a2d 6b6d  t, default 2.-km
+0000a360: 2c20 2d2d 6b64 656d 6574 686f 642c 2074  , --kdemethod, t
+0000a370: 6865 206b 6465 206d 6574 686f 6420 746f  he kde method to
+0000a380: 2062 6520 7573 6564 2069 6e20 616e 616c   be used in anal
+0000a390: 797a 696e 6720 7468 6520 7065 616b 206f  yzing the peak o
+0000a3a0: 6620 5747 4420 6461 7465 732c 206b 6d65  f WGD dates, kme
+0000a3b0: 646f 6964 7320 616e 616c 7973 6973 206f  doids analysis o
+0000a3c0: 7220 7468 6520 6261 7369 6320 4b73 2070  r the basic Ks p
+0000a3d0: 6c6f 7474 696e 672c 2064 6566 6175 6c74  lotting, default
+0000a3e0: 2073 6369 7079 0a2d 2d6e 5f63 6c75 7374   scipy.--n_clust
+0000a3f0: 6572 732c 2074 6865 206e 756d 6265 7220  ers, the number 
+0000a400: 6f66 2063 6c75 7374 6572 7320 746f 2070  of clusters to p
+0000a410: 6c6f 7420 456c 626f 7720 6c6f 7373 2066  lot Elbow loss f
+0000a420: 756e 6374 696f 6e2c 2064 6566 6175 6c74  unction, default
+0000a430: 2035 0a2d 6764 2c20 2d2d 6775 6964 652c   5.-gd, --guide,
+0000a440: 2074 6865 2072 6567 696d 6520 7265 7369   the regime resi
+0000a450: 6469 6e67 2061 6e63 686f 7273 2c20 6465  ding anchors, de
+0000a460: 6661 756c 7420 5365 676d 656e 740a 2d70  fault Segment.-p
+0000a470: 7263 742c 202d 2d70 726f 6d69 6e65 6e63  rct, --prominenc
+0000a480: 655f 6375 746f 6666 2c20 7468 6520 7072  e_cutoff, the pr
+0000a490: 6f6d 696e 656e 6365 2063 7574 6f66 6620  ominence cutoff 
+0000a4a0: 6f66 2061 6363 6570 7461 626c 6520 7065  of acceptable pe
+0000a4b0: 616b 7320 696e 2070 6561 6b20 6669 6e64  aks in peak find
+0000a4c0: 696e 6720 7374 6570 732c 2064 6566 6175  ing steps, defau
+0000a4d0: 6c74 2030 2e31 0a2d 7268 2c20 2d2d 7265  lt 0.1.-rh, --re
+0000a4e0: 6c5f 6865 6967 6874 2c20 7468 6520 7265  l_height, the re
+0000a4f0: 6c61 7469 7665 2068 6569 6768 7420 6174  lative height at
+0000a500: 2077 6869 6368 2074 6865 2070 6561 6b20   which the peak 
+0000a510: 7769 6474 6820 6973 206d 6561 7375 7265  width is measure
+0000a520: 642c 2064 6566 6175 6c74 2030 2e34 0a2d  d, default 0.4.-
+0000a530: 6b64 2c20 2d2d 6b73 746f 6461 7465 2c20  kd, --kstodate, 
+0000a540: 7468 6520 7261 6e67 6520 6f66 204b 7320  the range of Ks 
+0000a550: 746f 2062 6520 6461 7465 6420 696e 2068  to be dated in h
+0000a560: 6575 7269 7374 6963 2073 6561 7263 682c  euristic search,
+0000a570: 2064 6566 6175 6c74 2028 302e 352c 2031   default (0.5, 1
+0000a580: 2e35 290a 2d78 6c2c 202d 2d78 6c69 6d2c  .5).-xl, --xlim,
+0000a590: 2074 6865 2078 2061 7869 7320 6c69 6d69   the x axis limi
+0000a5a0: 7420 6f66 2047 4d4d 204b 7320 6469 7374  t of GMM Ks dist
+0000a5b0: 7269 6275 7469 6f6e 0a2d 796c 2c20 2d2d  ribution.-yl, --
+0000a5c0: 796c 696d 2c20 7468 6520 7920 6178 6973  ylim, the y axis
+0000a5d0: 206c 696d 6974 206f 6620 474d 4d20 4b73   limit of GMM Ks
+0000a5e0: 2064 6973 7472 6962 7574 696f 6e0a 2d2d   distribution.--
+0000a5f0: 6d61 6e75 616c 7365 742c 2066 6c61 6720  manualset, flag 
+0000a600: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+0000a610: 746f 206f 7574 7075 7420 616e 6368 6f72  to output anchor
+0000a620: 2070 6169 7273 2077 6974 6820 6d61 6e75   pairs with manu
+0000a630: 616c 6c79 2073 6574 204b 7320 7261 6e67  ally set Ks rang
+0000a640: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
+0000a650: 6173 2073 6574 2c20 6d61 6e75 616c 6c79  as set, manually
+0000a660: 2073 6574 204b 7320 7261 6e67 6520 776f   set Ks range wo
+0000a670: 756c 6420 6265 206f 7574 7075 7474 6564  uld be outputted
+0000a680: 0a2d 2d63 692c 2074 6865 2063 6f6e 6669  .--ci, the confi
+0000a690: 6465 6e63 6520 6c65 7665 6c20 6f66 206c  dence level of l
+0000a6a0: 6f67 2d6e 6f72 6d61 6c20 6469 7374 7269  og-normal distri
+0000a6b0: 6275 7469 6f6e 2074 6f20 6461 7465 2c20  bution to date, 
+0000a6c0: 6465 6661 756c 7420 3935 0a2d 2d68 6472  default 95.--hdr
+0000a6d0: 2c20 7468 6520 6869 6768 6573 7420 6465  , the highest de
+0000a6e0: 6e73 6974 7920 7265 6769 6f6e 2028 4844  nsity region (HD
+0000a6f0: 5229 2061 7070 6c69 6564 2069 6e20 7468  R) applied in th
+0000a700: 6520 7365 676d 656e 742d 6775 6964 6564  e segment-guided
+0000a710: 2061 6e63 686f 7220 7061 6972 204b 7320   anchor pair Ks 
+0000a720: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
+0000a730: 6661 756c 7420 3935 0a2d 2d68 6575 7269  fault 95.--heuri
+0000a740: 7374 6963 2c20 666c 6167 206f 7074 696f  stic, flag optio
+0000a750: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+0000a760: 6974 6961 7465 2068 6575 7269 7374 6963  itiate heuristic
+0000a770: 206d 6574 686f 6420 6f66 2064 6566 696e   method of defin
+0000a780: 696e 6720 4349 2066 6f72 2064 6174 696e  ing CI for datin
+0000a790: 672c 2069 6620 7468 6520 666c 6167 2077  g, if the flag w
+0000a7a0: 6173 2073 6574 2c20 7468 6520 6865 7572  as set, the heur
+0000a7b0: 6973 7469 6320 6d65 7468 6f64 2077 6f75  istic method wou
+0000a7c0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+0000a7d0: 2d6b 632c 202d 2d6b 7363 7574 6f66 662c  -kc, --kscutoff,
+0000a7e0: 2074 6865 204b 7320 7361 7475 7261 7469   the Ks saturati
+0000a7f0: 6f6e 2063 7574 6f66 6620 696e 2064 6174  on cutoff in dat
+0000a800: 696e 672c 2064 6566 6175 6c74 2035 0a2d  ing, default 5.-
+0000a810: 2d6b 6565 7074 6d70 6669 672c 2066 6c61  -keeptmpfig, fla
+0000a820: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+0000a830: 7220 746f 206b 6565 7020 7465 6d70 6f72  r to keep tempor
+0000a840: 6172 7920 6669 6775 7265 7320 696e 2070  ary figures in p
+0000a850: 6561 6b20 6669 6e64 696e 6720 7072 6f63  eak finding proc
+0000a860: 6573 732c 2069 6620 7468 6520 666c 6167  ess, if the flag
+0000a870: 2077 6173 2073 6574 2c20 7468 6f73 6520   was set, those 
+0000a880: 6669 6775 7265 7320 776f 756c 6420 6265  figures would be
+0000a890: 206b 6570 740a 6060 600a 0a54 6865 2070   kept.```..The p
+0000a8a0: 726f 6772 616d 2060 7767 6420 7379 6e60  rogram `wgd syn`
+0000a8b0: 2063 616e 2072 6561 6c69 7a65 2074 6865   can realize the
+0000a8c0: 2069 6e74 7261 2d20 616e 6420 696e 7465   intra- and inte
+0000a8d0: 722d 7370 6563 6966 6963 2073 796e 7465  r-specific synte
+0000a8e0: 6e79 2069 6e66 6572 656e 6365 2e0a 6060  ny inference..``
+0000a8f0: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
+0000a900: 6573 2067 6666 7320 286f 7074 696f 6e29  es gffs (option)
+0000a910: 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  .---------------
+0000a920: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a930: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a940: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a950: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a960: 2d0a 2d6b 732c 202d 2d6b 735f 6469 7374  -.-ks, --ks_dist
+0000a970: 7269 6275 7469 6f6e 2c20 6b73 2064 6973  ribution, ks dis
+0000a980: 7472 6962 7574 696f 6e20 6461 7461 6669  tribution datafi
+0000a990: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+0000a9a0: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
+0000a9b0: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
+0000a9c0: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
+0000a9d0: 5f73 796e 0a2d 662c 202d 2d66 6561 7475  _syn.-f, --featu
+0000a9e0: 7265 2c20 7468 6520 6665 6174 7572 6520  re, the feature 
+0000a9f0: 666f 7220 7061 7273 696e 6720 6765 6e65  for parsing gene
+0000aa00: 2049 4473 2066 726f 6d20 4746 4620 6669   IDs from GFF fi
+0000aa10: 6c65 732c 2064 6566 6175 6c74 2067 656e  les, default gen
+0000aa20: 650a 2d61 2c20 2d2d 6174 7472 6962 7574  e.-a, --attribut
+0000aa30: 652c 2074 6865 2061 7474 7269 6275 7465  e, the attribute
+0000aa40: 2066 6f72 2070 6172 7369 6e67 2074 6865   for parsing the
+0000aa50: 2067 656e 6520 4944 7320 6672 6f6d 2074   gene IDs from t
+0000aa60: 6865 2047 4646 2066 696c 6573 2c20 6465  he GFF files, de
+0000aa70: 6661 756c 7420 4944 0a2d 6174 672c 202d  fault ID.-atg, -
+0000aa80: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+0000aa90: 666f 2c20 7468 6520 6665 6174 7572 6520  fo, the feature 
+0000aaa0: 616e 6420 6174 7472 6962 7574 6520 696e  and attribute in
+0000aab0: 666f 726d 6174 696f 6e20 6f66 2061 6464  formation of add
+0000aac0: 6974 696f 6e61 6c20 6766 6633 2066 696c  itional gff3 fil
+0000aad0: 6573 2069 6620 6469 6666 6572 656e 7420  es if different 
+0000aae0: 696e 2074 6865 2066 6f72 6d61 7420 6f66  in the format of
+0000aaf0: 2028 6665 6174 7572 653b 6174 7472 6962   (feature;attrib
+0000ab00: 7574 6529 272c 2064 6566 6175 6c74 204e  ute)', default N
+0000ab10: 6f6e 650a 2d6d 6c2c 202d 2d6d 696e 6c65  one.-ml, --minle
+0000ab20: 6e2c 2074 6865 206d 696e 696d 756d 206c  n, the minimum l
+0000ab30: 656e 6774 6820 6f66 2061 2073 6361 6666  ength of a scaff
+0000ab40: 6f6c 6420 746f 2062 6520 696e 636c 7564  old to be includ
+0000ab50: 6564 2069 6e20 646f 7470 6c6f 742c 2064  ed in dotplot, d
+0000ab60: 6566 6175 6c74 202d 312c 2069 6620 2d31  efault -1, if -1
+0000ab70: 2077 6173 2073 6574 2c20 7468 6520 3130   was set, the 10
+0000ab80: 2520 6f66 2074 6865 206c 6f6e 6765 7374  % of the longest
+0000ab90: 2073 6361 6666 6f6c 6420 776f 756c 6420   scaffold would 
+0000aba0: 6265 2073 6574 0a2d 6d73 2c20 2d2d 6d61  be set.-ms, --ma
+0000abb0: 7873 697a 652c 2074 6865 206d 6178 696d  xsize, the maxim
+0000abc0: 756d 2066 616d 696c 7920 7369 7a65 2074  um family size t
+0000abd0: 6f20 6265 2069 6e63 6c75 6465 642c 2064  o be included, d
+0000abe0: 6566 6175 6c74 2032 3030 0a2d 722c 202d  efault 200.-r, -
+0000abf0: 2d6b 735f 7261 6e67 652c 2074 6865 204b  -ks_range, the K
+0000ac00: 7320 7261 6e67 6520 696e 2063 6f6c 6f72  s range in color
+0000ac10: 6564 2064 6f74 706c 6f74 2c20 6465 6661  ed dotplot, defa
+0000ac20: 756c 7420 2830 2c20 3529 0a2d 2d69 6164  ult (0, 5).--iad
+0000ac30: 686f 7265 5f6f 7074 696f 6e73 2c20 7468  hore_options, th
+0000ac40: 6520 7061 7261 6d65 7465 7220 7365 7474  e parameter sett
+0000ac50: 696e 6720 696e 2069 6164 686f 7265 2c20  ing in iadhore, 
+0000ac60: 6465 6661 756c 7420 6173 2061 2073 7472  default as a str
+0000ac70: 696e 6720 6f66 206c 656e 6774 6820 7a65  ing of length ze
+0000ac80: 726f 0a2d 6d67 2c20 2d2d 6d69 6e73 6567  ro.-mg, --minseg
+0000ac90: 6c65 6e2c 2074 6865 206d 696e 696d 756d  len, the minimum
+0000aca0: 206c 656e 6774 6820 6f66 2073 6567 6d65   length of segme
+0000acb0: 6e74 7320 746f 2069 6e63 6c75 6465 2069  nts to include i
+0000acc0: 6e20 7261 7469 6f20 6966 203c 3d20 312c  n ratio if <= 1,
+0000acd0: 2064 6566 6175 6c74 2031 3030 3030 0a2d   default 10000.-
+0000ace0: 6b72 2c20 2d2d 6b65 6570 7265 6475 6e2c  kr, --keepredun,
+0000acf0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+0000ad00: 6574 6865 7220 746f 206b 6565 7020 7265  ether to keep re
+0000ad10: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
+0000ad20: 636f 6e73 2c20 6966 2074 6865 2066 6c61  cons, if the fla
+0000ad30: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+0000ad40: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+0000ad50: 6963 6f6e 7320 776f 756c 6420 6265 206b  icons would be k
+0000ad60: 6570 740a 2d6d 676e 2c20 2d2d 6d69 6e67  ept.-mgn, --ming
+0000ad70: 656e 656e 756d 2c20 7468 6520 6d69 6e69  enenum, the mini
+0000ad80: 6d75 6d20 6e75 6d62 6572 206f 6620 6765  mum number of ge
+0000ad90: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+0000ada0: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+0000adb0: 6564 2c20 6465 6661 756c 7420 3330 0a2d  ed, default 30.-
+0000adc0: 6473 2c20 2d2d 646f 7473 697a 652c 2074  ds, --dotsize, t
+0000add0: 6865 2064 6f74 2073 697a 6520 696e 2064  he dot size in d
+0000ade0: 6f74 2070 6c6f 742c 2064 6566 6175 6c74  ot plot, default
+0000adf0: 2030 2e33 0a2d 6161 2c20 2d2d 6170 616c   0.3.-aa, --apal
+0000ae00: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+0000ae10: 206f 6620 616e 6368 6f72 2064 6f74 732c   of anchor dots,
+0000ae20: 2064 6566 6175 6c74 2031 0a2d 6861 2c20   default 1.-ha, 
+0000ae30: 2d2d 686f 616c 7068 612c 2074 6865 206f  --hoalpha, the o
+0000ae40: 7061 6369 7479 206f 6620 686f 6d6f 6c6f  pacity of homolo
+0000ae50: 6720 646f 7473 2c20 6465 6661 756c 7420  g dots, default 
+0000ae60: 300a 2d73 7274 2c20 2d2d 7368 6f77 7265  0.-srt, --showre
+0000ae70: 616c 7469 636b 2c20 666c 6167 206f 7074  altick, flag opt
+0000ae80: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+0000ae90: 7368 6f77 2074 6865 2072 6561 6c20 7469  show the real ti
+0000aea0: 636b 2069 6e20 6765 6e65 7320 6f72 2062  ck in genes or b
+0000aeb0: 6173 6573 2c20 6966 2074 6865 2066 6c61  ases, if the fla
+0000aec0: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+0000aed0: 6561 6c20 7469 636b 2077 6f75 6c64 2062  eal tick would b
+0000aee0: 6520 7368 6f77 6564 0a2d 746c 732c 202d  e showed.-tls, -
+0000aef0: 2d74 6963 6b6c 6162 656c 7369 7a65 2c20  -ticklabelsize, 
+0000af00: 7468 6520 6c61 6265 6c20 7369 7a65 206f  the label size o
+0000af10: 6620 7469 636b 2c20 6465 6661 756c 7420  f tick, default 
+0000af20: 350a 2d67 722c 202d 2d67 6973 7472 622c  5.-gr, --gistrb,
+0000af30: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+0000af40: 6574 6865 7220 746f 2075 7365 2067 6973  ether to use gis
+0000af50: 745f 7261 696e 626f 7720 6173 2063 6f6c  t_rainbow as col
+0000af60: 6f72 206d 6170 206f 6620 646f 7470 6c6f  or map of dotplo
+0000af70: 740a 2d6e 2c20 2d2d 6e74 6872 6561 6473  t.-n, --nthreads
+0000af80: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+0000af90: 7468 7265 6164 7320 746f 2075 7365 2069  threads to use i
+0000afa0: 6e20 7379 6e74 656e 7920 696e 6665 7265  n synteny infere
+0000afb0: 6e63 652c 2064 6566 6175 6c74 2034 0a60  nce, default 4.`
+0000afc0: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+0000afd0: 6077 6764 2076 697a 6020 6361 6e20 7265  `wgd viz` can re
+0000afe0: 616c 697a 6520 7468 6520 7669 7375 616c  alize the visual
+0000aff0: 697a 6174 696f 6e20 6f66 202a 4b2a 3c73  ization of *K*<s
+0000b000: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+0000b010: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
+0000b020: 7379 6e74 656e 792e 0a60 6060 0a77 6764  synteny..```.wgd
+0000b030: 2076 697a 2028 6f70 7469 6f6e 290a 2d2d   viz (option).--
+0000b040: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000b050: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000b060: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000b070: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000b080: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
+0000b090: 642c 202d 2d64 6174 6166 696c 652c 2074  d, --datafile, t
+0000b0a0: 6865 204b 7320 6461 7461 6669 6c65 2c20  he Ks datafile, 
+0000b0b0: 6465 6661 756c 7420 4e6f 6e65 0a2d 6f2c  default None.-o,
+0000b0c0: 202d 2d6f 7574 6469 722c 2074 6865 206f   --outdir, the o
+0000b0d0: 7574 7075 7420 6469 7265 6374 6f72 792c  utput directory,
+0000b0e0: 2064 6566 6175 6c74 2077 6764 5f76 697a   default wgd_viz
+0000b0f0: 0a2d 7372 2c20 2d2d 7370 6169 722c 2074  .-sr, --spair, t
+0000b100: 6865 2073 7065 6369 6573 2070 6169 7220  he species pair 
+0000b110: 746f 2062 6520 706c 6f74 7465 642c 2064  to be plotted, d
+0000b120: 6566 6175 6c74 204e 6f6e 652c 2074 6869  efault None, thi
+0000b130: 7320 6f70 7469 6f6e 2063 616e 2062 6520  s option can be 
+0000b140: 7072 6f76 6964 6564 206d 756c 7469 706c  provided multipl
+0000b150: 6520 7469 6d65 730a 2d66 612c 202d 2d66  e times.-fa, --f
+0000b160: 6f63 7573 3261 6c6c 2c20 7365 7420 666f  ocus2all, set fo
+0000b170: 6361 6c20 7370 6563 6965 7320 616e 6420  cal species and 
+0000b180: 6c65 7420 7370 6563 6965 7320 7061 6972  let species pair
+0000b190: 2074 6f20 6265 2062 6574 7765 656e 2066   to be between f
+0000b1a0: 6f63 616c 2061 6e64 2061 6c6c 2074 6865  ocal and all the
+0000b1b0: 2072 656d 6169 6e69 6e67 2073 7065 6369   remaining speci
+0000b1c0: 6573 2c20 6465 6661 756c 7420 4e6f 6e65  es, default None
+0000b1d0: 0a2d 6773 2c20 2d2d 6773 6d61 702c 2074  .-gs, --gsmap, t
+0000b1e0: 6865 2067 656e 6520 6e61 6d65 2d73 7065  he gene name-spe
+0000b1f0: 6369 6573 206e 616d 6520 6d61 702c 2064  cies name map, d
+0000b200: 6566 6175 6c74 204e 6f6e 650a 2d73 702c  efault None.-sp,
+0000b210: 202d 2d73 7065 6369 6573 7472 6565 2c20   --speciestree, 
+0000b220: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
+0000b230: 2074 6f20 7065 7266 6f72 6d20 7261 7465   to perform rate
+0000b240: 2063 6f72 7265 6374 696f 6e2c 2064 6566   correction, def
+0000b250: 6175 6c74 204e 6f6e 652c 2069 6620 4e6f  ault None, if No
+0000b260: 6e65 2077 6173 2067 6976 656e 2c20 7468  ne was given, th
+0000b270: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+0000b280: 6e20 616e 616c 7973 6973 2077 6f75 6c64  n analysis would
+0000b290: 2062 6520 6361 6c6c 6564 206f 6666 0a2d   be called off.-
+0000b2a0: 706b 2c20 2d2d 706c 6f74 6b64 652c 2066  pk, --plotkde, f
+0000b2b0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+0000b2c0: 6865 7220 746f 2070 6c6f 7420 6b64 6520  her to plot kde 
+0000b2d0: 6375 7276 6520 7570 6f6e 2068 6973 746f  curve upon histo
+0000b2e0: 6772 616d 2c20 6966 2074 6865 2066 6c61  gram, if the fla
+0000b2f0: 6720 7761 7320 7365 742c 206b 6465 2063  g was set, kde c
+0000b300: 7572 7665 2077 6f75 6c64 2062 6520 6164  urve would be ad
+0000b310: 6465 640a 2d72 772c 202d 2d72 6577 6569  ded.-rw, --rewei
+0000b320: 6768 742c 2066 6c61 6720 6f70 7469 6f6e  ght, flag option
+0000b330: 2c20 7768 6574 6865 7220 746f 2072 6563  , whether to rec
+0000b340: 616c 6375 6c61 7465 2074 6865 2077 6569  alculate the wei
+0000b350: 6768 7420 7065 7220 7370 6563 6965 7320  ght per species 
+0000b360: 7061 6972 2c20 6966 2074 6865 2066 6c61  pair, if the fla
+0000b370: 6720 7761 7320 7365 742c 2074 6865 2077  g was set, the w
+0000b380: 6569 6768 7420 776f 756c 6420 6265 2072  eight would be r
+0000b390: 6563 616c 6375 6c61 7465 640a 2d6f 722c  ecalculated.-or,
+0000b3a0: 202d 2d6f 6e6c 7972 6f6f 746f 7574 2c20   --onlyrootout, 
+0000b3b0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+0000b3c0: 7468 6572 2074 6f20 6f6e 6c79 2063 6f6e  ther to only con
+0000b3d0: 6475 6374 2072 6174 6520 636f 7272 6563  duct rate correc
+0000b3e0: 7469 6f6e 2075 7369 6e67 2074 6865 206f  tion using the o
+0000b3f0: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
+0000b400: 6173 206f 7574 6772 6f75 702c 2069 6620  as outgroup, if 
+0000b410: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+0000b420: 2c20 6f6e 6c79 2074 6865 206f 7574 6772  , only the outgr
+0000b430: 6f75 7020 6174 2072 6f6f 7420 776f 756c  oup at root woul
+0000b440: 6420 6265 2075 7365 6420 6173 206f 7574  d be used as out
+0000b450: 6772 6f75 700a 2d69 7465 722c 202d 2d65  group.-iter, --e
+0000b460: 6d5f 6974 6572 6174 696f 6e73 2c20 7468  m_iterations, th
+0000b470: 6520 6d61 7869 6d75 6d20 454d 2069 7465  e maximum EM ite
+0000b480: 7261 7469 6f6e 732c 2064 6566 6175 6c74  rations, default
+0000b490: 2032 3030 0a2d 696e 6974 2c20 2d2d 656d   200.-init, --em
+0000b4a0: 5f69 6e69 7469 616c 697a 6174 696f 6e73  _initializations
+0000b4b0: 2c20 7468 6520 6d61 7869 6d75 6d20 454d  , the maximum EM
+0000b4c0: 2069 6e69 7469 616c 697a 6174 696f 6e73   initializations
+0000b4d0: 2c20 6465 6661 756c 7420 3230 300a 2d70  , default 200.-p
+0000b4e0: 7263 742c 202d 2d70 726f 6d69 6e65 6e63  rct, --prominenc
+0000b4f0: 655f 6375 746f 6666 2c20 7468 6520 7072  e_cutoff, the pr
+0000b500: 6f6d 696e 656e 6365 2063 7574 6f66 6620  ominence cutoff 
+0000b510: 6f66 2061 6363 6570 7461 626c 6520 7065  of acceptable pe
+0000b520: 616b 732c 2064 6566 6175 6c74 2030 2e31  aks, default 0.1
+0000b530: 0a2d 7268 2c20 2d2d 7265 6c5f 6865 6967  .-rh, --rel_heig
+0000b540: 6874 2c20 7468 6520 7265 6c61 7469 7665  ht, the relative
+0000b550: 2068 6569 6768 7420 6174 2077 6869 6368   height at which
+0000b560: 2074 6865 2070 6561 6b20 7769 6474 6820   the peak width 
+0000b570: 6973 206d 6561 7375 7265 642c 2064 6566  is measured, def
+0000b580: 6175 6c74 2030 2e34 0a2d 736d 2c20 2d2d  ault 0.4.-sm, --
+0000b590: 7365 676d 656e 7473 2c20 7468 6520 7365  segments, the se
+0000b5a0: 676d 656e 7473 2064 6174 6166 696c 652c  gments datafile,
+0000b5b0: 2064 6566 6175 6c74 204e 6f6e 650a 2d6d   default None.-m
+0000b5c0: 6c2c 202d 2d6d 696e 6c65 6e2c 2074 6865  l, --minlen, the
+0000b5d0: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
+0000b5e0: 6f66 2061 2073 6361 6666 6f6c 6420 746f  of a scaffold to
+0000b5f0: 2062 6520 696e 636c 7564 6564 2069 6e20   be included in 
+0000b600: 646f 7470 6c6f 742c 2064 6566 6175 6c74  dotplot, default
+0000b610: 202d 312c 2069 6620 2d31 2077 6173 2073   -1, if -1 was s
+0000b620: 6574 2c20 7468 6520 3130 2520 6f66 2074  et, the 10% of t
+0000b630: 6865 206c 6f6e 6765 7374 2073 6361 6666  he longest scaff
+0000b640: 6f6c 6473 2077 696c 6c20 6265 2073 6574  olds will be set
+0000b650: 0a2d 6d73 2c20 2d2d 6d61 7873 697a 652c  .-ms, --maxsize,
+0000b660: 2074 6865 206d 6178 696d 756d 2066 616d   the maximum fam
+0000b670: 696c 7920 7369 7a65 2074 6f20 6265 2069  ily size to be i
+0000b680: 6e63 6c75 6465 642c 2064 6566 6175 6c74  ncluded, default
+0000b690: 2032 3030 0a2d 6170 2c20 2d2d 616e 6368   200.-ap, --anch
+0000b6a0: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
+0000b6b0: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
+0000b6c0: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+0000b6d0: 6e65 0a2d 6d74 2c20 2d2d 6d75 6c74 6970  ne.-mt, --multip
+0000b6e0: 6c69 636f 6e2c 2074 6865 206d 756c 7469  licon, the multi
+0000b6f0: 706c 6963 6f6e 7320 6461 7461 6669 6c65  plicons datafile
+0000b700: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+0000b710: 6774 2c20 2d2d 6765 6e65 7461 626c 652c  gt, --genetable,
+0000b720: 2074 6865 2067 656e 6520 7461 626c 6520   the gene table 
+0000b730: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
+0000b740: 7420 4e6f 6e65 0a2d 6d67 2c20 2d2d 6d69  t None.-mg, --mi
+0000b750: 6e73 6567 6c65 6e2c 2074 6865 206d 696e  nseglen, the min
+0000b760: 696d 756d 206c 656e 6774 6820 6f66 2073  imum length of s
+0000b770: 6567 6d65 6e74 7320 746f 2069 6e63 6c75  egments to inclu
+0000b780: 6465 2c20 696e 2072 6174 696f 2069 6620  de, in ratio if 
+0000b790: 3c3d 2031 2c20 6465 6661 756c 7420 3130  <= 1, default 10
+0000b7a0: 3030 300a 2d6d 676e 2c20 2d2d 6d69 6e67  000.-mgn, --ming
+0000b7b0: 656e 656e 756d 2c20 7468 6520 6d69 6e69  enenum, the mini
+0000b7c0: 6d75 6d20 6e75 6d62 6572 206f 6620 6765  mum number of ge
+0000b7d0: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+0000b7e0: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+0000b7f0: 6564 2c20 6465 6661 756c 7420 3330 0a2d  ed, default 30.-
+0000b800: 6b72 2c20 2d2d 6b65 6570 7265 6475 6e2c  kr, --keepredun,
+0000b810: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+0000b820: 6574 6865 7220 746f 206b 6565 7020 7265  ether to keep re
+0000b830: 6475 6e64 616e 7420 6d75 6c74 6970 6c69  dundant multipli
+0000b840: 636f 6e73 2c20 6966 2074 6865 2066 6c61  cons, if the fla
+0000b850: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+0000b860: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+0000b870: 6963 6f6e 7320 776f 756c 6420 6265 206b  icons would be k
+0000b880: 6570 740a 2d65 706b 2c20 2d2d 6578 7472  ept.-epk, --extr
+0000b890: 6170 6172 616e 6f6d 656b 732c 2065 7874  aparanomeks, ext
+0000b8a0: 7261 2070 6172 616e 6f6d 6520 4b73 2064  ra paranome Ks d
+0000b8b0: 6174 6120 746f 2070 6c6f 7420 696e 2074  ata to plot in t
+0000b8c0: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
+0000b8d0: 7269 6275 7469 6f6e 2c20 6465 6661 756c  ribution, defaul
+0000b8e0: 7420 4e6f 6e65 0a2d 7061 672c 202d 2d70  t None.-pag, --p
+0000b8f0: 6c6f 7461 7067 6d6d 2c20 666c 6167 206f  lotapgmm, flag o
+0000b900: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+0000b910: 6f20 636f 6e64 7563 7420 616e 6420 706c  o conduct and pl
+0000b920: 6f74 206d 6978 7475 7265 206d 6f64 656c  ot mixture model
+0000b930: 696e 6720 6f66 2061 6e63 686f 7220 4b73  ing of anchor Ks
+0000b940: 2069 6e20 7468 6520 6d69 7865 6420 4b73   in the mixed Ks
+0000b950: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
+0000b960: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+0000b970: 6574 2c20 7468 6520 6d69 7874 7572 6520  et, the mixture 
+0000b980: 6d6f 6465 6c69 6e67 206f 6620 616e 6368  modeling of anch
+0000b990: 6f72 204b 7320 776f 756c 6420 6265 2063  or Ks would be c
+0000b9a0: 6f6e 6475 6374 6564 2061 6e64 2070 6c6f  onducted and plo
+0000b9b0: 7474 6564 0a2d 7065 6d2c 202d 2d70 6c6f  tted.-pem, --plo
+0000b9c0: 7465 6c6d 6d2c 2066 6c61 6720 6f70 7469  telmm, flag opti
+0000b9d0: 6f6e 2c20 7768 6574 6865 7220 746f 2063  on, whether to c
+0000b9e0: 6f6e 6475 6374 2061 6e64 2070 6c6f 7420  onduct and plot 
+0000b9f0: 656c 6d6d 206d 6978 7475 7265 206d 6f64  elmm mixture mod
+0000ba00: 656c 696e 6720 6f66 2070 6172 616e 6f6d  eling of paranom
+0000ba10: 6520 4b73 2069 6e20 7468 6520 6d69 7865  e Ks in the mixe
+0000ba20: 6420 4b73 2064 6973 7472 6962 7574 696f  d Ks distributio
+0000ba30: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
+0000ba40: 6173 2073 6574 2c20 7468 6520 656c 6d6d  as set, the elmm
+0000ba50: 206d 6978 7475 7265 206d 6f64 656c 696e   mixture modelin
+0000ba60: 6720 6f66 2070 6172 616e 6f6d 6520 4b73  g of paranome Ks
+0000ba70: 2077 6f75 6c64 2062 6520 636f 6e64 7563   would be conduc
+0000ba80: 7465 6420 616e 6420 706c 6f74 7465 640a  ted and plotted.
+0000ba90: 2d63 2c20 2d2d 636f 6d70 6f6e 656e 7473  -c, --components
+0000baa0: 2c20 7468 6520 7261 6e67 6520 6f66 2074  , the range of t
+0000bab0: 6865 206e 756d 6265 7220 6f66 2063 6f6d  he number of com
+0000bac0: 706f 6e65 6e74 7320 746f 2066 6974 2069  ponents to fit i
+0000bad0: 6e20 616e 6368 6f72 204b 7320 6d69 7874  n anchor Ks mixt
+0000bae0: 7572 6520 6d6f 6465 6c69 6e67 2c20 6465  ure modeling, de
+0000baf0: 6661 756c 7420 2831 2c34 290a 2d70 7379  fault (1,4).-psy
+0000bb00: 2c20 2d2d 706c 6f74 7379 6e2c 2066 6c61  , --plotsyn, fla
+0000bb10: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+0000bb20: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
+0000bb30: 6520 7379 6e74 656e 7920 706c 6f74 2c20  e synteny plot, 
+0000bb40: 6f6e 6c79 2077 6865 6e20 7468 6520 666c  only when the fl
+0000bb50: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+0000bb60: 7379 6e74 656e 7920 706c 6f74 2077 6f75  synteny plot wou
+0000bb70: 6c64 2062 6520 7072 6f64 7563 6564 0a2d  ld be produced.-
+0000bb80: 6473 2c20 2d2d 646f 7473 697a 652c 2074  ds, --dotsize, t
+0000bb90: 6865 2064 6f74 2073 697a 6520 696e 2064  he dot size in d
+0000bba0: 6f74 2070 6c6f 742c 2064 6566 6175 6c74  ot plot, default
+0000bbb0: 2030 2e33 0a2d 6161 2c20 2d2d 6170 616c   0.3.-aa, --apal
+0000bbc0: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+0000bbd0: 206f 6620 616e 6368 6f72 2064 6f74 732c   of anchor dots,
+0000bbe0: 2064 6566 6175 6c74 2031 0a2d 6861 2c20   default 1.-ha, 
+0000bbf0: 2d2d 686f 616c 7068 612c 2074 6865 206f  --hoalpha, the o
+0000bc00: 7061 6369 7479 206f 6620 686f 6d6f 6c6f  pacity of homolo
+0000bc10: 6720 646f 7473 2c20 6465 6661 756c 7420  g dots, default 
+0000bc20: 300a 2d73 7274 2c20 2d2d 7368 6f77 7265  0.-srt, --showre
+0000bc30: 616c 7469 636b 2c20 666c 6167 206f 7074  altick, flag opt
+0000bc40: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+0000bc50: 7368 6f77 2074 6865 2072 6561 6c20 7469  show the real ti
+0000bc60: 636b 2069 6e20 6765 6e65 7320 6f72 2062  ck in genes or b
+0000bc70: 6173 6573 2c20 6966 2074 6865 2066 6c61  ases, if the fla
+0000bc80: 6720 7761 7320 7365 742c 2074 6865 2072  g was set, the r
+0000bc90: 6561 6c20 7469 636b 2077 6f75 6c64 2062  eal tick would b
+0000bca0: 6520 7368 6f77 6564 0a2d 746c 732c 202d  e showed.-tls, -
+0000bcb0: 2d74 6963 6b6c 6162 656c 7369 7a65 2c20  -ticklabelsize, 
+0000bcc0: 7468 6520 6c61 6265 6c20 7369 7a65 206f  the label size o
+0000bcd0: 6620 7469 636b 2c20 6465 6661 756c 7420  f tick, default 
+0000bce0: 350a 2d78 6c2c 202d 2d78 6c69 6d2c 2074  5.-xl, --xlim, t
+0000bcf0: 6865 2078 2061 7869 7320 6c69 6d69 7420  he x axis limit 
+0000bd00: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
+0000bd10: 6f6e 0a2d 796c 2c20 2d2d 796c 696d 2c20  on.-yl, --ylim, 
+0000bd20: 7468 6520 7920 6178 6973 206c 696d 6974  the y axis limit
+0000bd30: 206f 6620 4b73 2064 6973 7472 6962 7574   of Ks distribut
+0000bd40: 696f 6e0a 2d61 646f 2c20 2d2d 6164 6a75  ion.-ado, --adju
+0000bd50: 7374 6f72 7468 6f2c 2066 6c61 6720 6f70  stortho, flag op
+0000bd60: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+0000bd70: 2061 646a 7573 7420 7468 6520 6869 7374   adjust the hist
+0000bd80: 6f67 7261 6d20 6865 6967 6874 206f 6620  ogram height of 
+0000bd90: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2061  orthologous Ks a
+0000bda0: 7320 746f 206d 6174 6368 2074 6865 2068  s to match the h
+0000bdb0: 6569 6768 7420 6f66 2070 6172 616c 6f67  eight of paralog
+0000bdc0: 6f75 7320 4b73 2c20 6966 2074 6865 2066  ous Ks, if the f
+0000bdd0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+0000bde0: 2061 646a 7573 746d 656e 7420 776f 756c   adjustment woul
+0000bdf0: 6420 6265 2063 6f6e 6475 6374 6564 0a2d  d be conducted.-
+0000be00: 6164 662c 202d 2d61 646a 7573 7466 6163  adf, --adjustfac
+0000be10: 746f 722c 2074 6865 2061 646a 7573 746d  tor, the adjustm
+0000be20: 656e 7420 6661 6374 6f72 206f 6620 6f72  ent factor of or
+0000be30: 7468 6f6c 6f67 6f75 7320 4b73 2c20 6465  thologous Ks, de
+0000be40: 6661 756c 7420 302e 350a 2d6f 612c 202d  fault 0.5.-oa, -
+0000be50: 2d6f 6b61 6c70 6861 2c20 7468 6520 6f70  -okalpha, the op
+0000be60: 6163 6974 7920 6f66 206f 7274 686f 6c6f  acity of ortholo
+0000be70: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
+0000be80: 7469 6f6e 2069 6e20 6d69 7865 6420 706c  tion in mixed pl
+0000be90: 6f74 2c20 6465 6661 756c 7420 302e 350a  ot, default 0.5.
+0000bea0: 2d63 732c 202d 2d63 6c61 7373 6963 2c20  -cs, --classic, 
+0000beb0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+0000bec0: 7468 6572 2074 6f20 6472 6177 206d 6978  ther to draw mix
+0000bed0: 6564 204b 7320 706c 6f74 2069 6e20 6120  ed Ks plot in a 
+0000bee0: 636c 6173 7369 6320 6d61 6e6e 6572 2077  classic manner w
+0000bef0: 6865 7265 2074 6865 2066 756c 6c20 6f72  here the full or
+0000bf00: 7468 6f6c 6f67 6f75 7320 4b73 2064 6973  thologous Ks dis
+0000bf10: 7472 6962 7574 696f 6e20 6973 2064 7261  tribution is dra
+0000bf20: 7765 642c 2069 6620 7468 6520 666c 6167  wed, if the flag
+0000bf30: 2077 6173 2073 6574 2c20 7468 6520 636c   was set, the cl
+0000bf40: 6173 7369 6320 6d69 7865 6420 4b73 2070  assic mixed Ks p
+0000bf50: 6c6f 7420 776f 756c 6420 6265 2064 7261  lot would be dra
+0000bf60: 776e 0a2d 7461 2c20 2d2d 746f 7061 7272  wn.-ta, --toparr
+0000bf70: 6f77 2c20 666c 6167 206f 7074 696f 6e2c  ow, flag option,
+0000bf80: 2077 6865 7468 6572 2074 6f20 6164 6a75   whether to adju
+0000bf90: 7374 2074 6865 2061 7272 6f77 2074 6f20  st the arrow to 
+0000bfa0: 6265 2061 7420 7468 6520 746f 7020 6f66  be at the top of
+0000bfb0: 2074 6865 2070 6c6f 742c 2069 6e73 7465   the plot, inste
+0000bfc0: 6164 206f 6620 6265 696e 6720 636f 6f72  ad of being coor
+0000bfd0: 6469 6e61 7465 6420 6173 2074 6865 204b  dinated as the K
+0000bfe0: 4445 206f 6620 7468 6520 6f72 7468 6f6c  DE of the orthol
+0000bff0: 6f67 6f75 7320 4b73 2064 6973 7472 6962  ogous Ks distrib
+0000c000: 7574 696f 6e2c 2069 6620 7468 6520 666c  ution, if the fl
+0000c010: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+0000c020: 6172 726f 7720 776f 756c 6420 6265 2073  arrow would be s
+0000c030: 6574 2061 7420 7468 6520 746f 700a 2d6e  et at the top.-n
+0000c040: 612c 202d 2d6e 6f64 6561 7665 7261 6765  a, --nodeaverage
+0000c050: 642c 2066 6c61 6720 6f70 7469 6f6e 2c20  d, flag option, 
+0000c060: 7768 6574 6865 7220 746f 2075 7365 206e  whether to use n
+0000c070: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
+0000c080: 686f 6420 666f 7220 6465 2d72 6564 756e  hod for de-redun
+0000c090: 6461 6e63 792c 2069 6620 7468 6520 666c  dancy, if the fl
+0000c0a0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+0000c0b0: 6e6f 6465 2d61 7665 7261 6765 6420 6d65  node-averaged me
+0000c0c0: 7468 6f64 2077 6f75 6c64 2062 6520 696e  thod would be in
+0000c0d0: 6974 6961 7465 640a 2d62 732c 202d 2d62  itiated.-bs, --b
+0000c0e0: 6f6f 7473 7472 6170 2c20 7468 6520 6e75  ootstrap, the nu
+0000c0f0: 6d62 6572 206f 6620 626f 6f74 7374 7261  mber of bootstra
+0000c100: 7020 7265 706c 6963 6174 6573 206f 6620  p replicates of 
+0000c110: 6f72 7468 6f6c 6f67 204b 7320 6469 7374  ortholog Ks dist
+0000c120: 7269 6275 7469 6f6e 2069 6e20 6d69 7865  ribution in mixe
+0000c130: 6420 706c 6f74 0a2d 6772 2c20 2d2d 6769  d plot.-gr, --gi
+0000c140: 7374 7262 2c20 666c 6167 206f 7074 696f  strb, flag optio
+0000c150: 6e2c 2077 6865 7468 6572 2074 6f20 7573  n, whether to us
+0000c160: 6520 6769 7374 5f72 6169 6e62 6f77 2061  e gist_rainbow a
+0000c170: 7320 636f 6c6f 7220 6d61 7020 6f66 2064  s color map of d
+0000c180: 6f74 706c 6f74 0a2d 6e2c 202d 2d6e 7468  otplot.-n, --nth
+0000c190: 7265 6164 732c 2074 6865 206e 756d 6265  reads, the numbe
+0000c1a0: 7220 6f66 2074 6872 6561 6473 2074 6f20  r of threads to 
+0000c1b0: 7573 6520 696e 2062 6f6f 7473 7472 6170  use in bootstrap
+0000c1c0: 2073 616d 706c 696e 672c 2064 6566 6175   sampling, defau
+0000c1d0: 6c74 2031 0a60 6060 0a0a 2323 2055 7361  lt 1.```..## Usa
+0000c1e0: 6765 0a0a 4865 7265 2077 6520 7072 6f76  ge..Here we prov
+0000c1f0: 6964 6564 2074 6865 2062 6173 6963 2075  ided the basic u
+0000c200: 7361 6765 2066 6f72 2065 6163 6820 7072  sage for each pr
+0000c210: 6f67 7261 6d20 616e 6420 7468 6520 7265  ogram and the re
+0000c220: 6c65 7661 6e74 2070 6172 616d 6574 6572  levant parameter
+0000c230: 7320 616e 6420 7375 6767 6573 7469 6f6e  s and suggestion
+0000c240: 7320 6f6e 2070 6172 616d 6574 6572 697a  s on parameteriz
+0000c250: 6174 696f 6e2e 2041 2072 656d 696e 6465  ation. A reminde
+0000c260: 7220 7468 6174 2074 6865 2067 6976 656e  r that the given
+0000c270: 2063 6f72 6573 2061 6e64 2074 6872 6561   cores and threa
+0000c280: 6473 2063 616e 2073 6967 6e69 6669 6361  ds can significa
+0000c290: 6e74 6c79 2069 6d70 6163 7420 7468 6520  ntly impact the 
+0000c2a0: 7275 6e20 7469 6d65 2061 6e64 2074 6875  run time and thu
+0000c2b0: 7320 7765 2061 6464 6564 2073 6f6d 6520  s we added some 
+0000c2c0: 7265 706f 7274 2069 6e66 6f72 6d61 7469  report informati
+0000c2d0: 6f6e 2070 6572 7461 696e 696e 6720 746f  on pertaining to
+0000c2e0: 2074 6865 2073 7973 7465 6d20 6f66 2075   the system of u
+0000c2f0: 7365 7273 2074 6f20 6661 6369 6c69 7461  sers to facilita
+0000c300: 7465 2074 6865 2065 6666 6963 6965 6e74  te the efficient
+0000c310: 2073 6574 7469 6e67 206f 6620 7468 7265   setting of thre
+0000c320: 6164 7320 616e 6420 6d65 6d6f 7279 2e20  ads and memory. 
+0000c330: 5468 6520 6c6f 6769 6361 6c20 4350 5573  The logical CPUs
+0000c340: 2072 6570 6f72 7465 6420 7265 7072 6573   reported repres
+0000c350: 656e 7473 2074 6865 206e 756d 6265 7220  ents the number 
+0000c360: 6f66 2070 6879 7369 6361 6c20 636f 7265  of physical core
+0000c370: 7320 6d75 6c74 6970 6c69 6564 2062 7920  s multiplied by 
+0000c380: 7468 6520 6e75 6d62 6572 206f 6620 7468  the number of th
+0000c390: 7265 6164 7320 7468 6174 2063 616e 2072  reads that can r
+0000c3a0: 756e 206f 6e20 6561 6368 2063 6f72 652c  un on each core,
+0000c3b0: 2061 6c73 6f20 6b6e 6f77 6e20 6173 2048   also known as H
+0000c3c0: 7970 6572 2054 6872 6561 6469 6e67 2e20  yper Threading. 
+0000c3d0: 5468 6520 6e75 6d62 6572 206f 6620 6c6f  The number of lo
+0000c3e0: 6769 6361 6c20 4350 5573 206d 6179 206e  gical CPUs may n
+0000c3f0: 6f74 206e 6563 6573 7361 7269 6c79 2062  ot necessarily b
+0000c400: 6520 6571 7569 7661 6c65 6e74 2074 6f20  e equivalent to 
+0000c410: 7468 6520 6163 7475 616c 206e 756d 6265  the actual numbe
+0000c420: 7220 6f66 2043 5055 7320 7468 6520 6375  r of CPUs the cu
+0000c430: 7272 656e 7420 7072 6f63 6573 7320 6361  rrent process ca
+0000c440: 6e20 7573 652e 2054 6865 2061 7661 696c  n use. The avail
+0000c450: 6162 6c65 206d 656d 6f72 7920 7265 6665  able memory refe
+0000c460: 7273 2074 6f20 7468 6520 6d65 6d6f 7279  rs to the memory
+0000c470: 2074 6861 7420 6361 6e20 6265 2067 6976   that can be giv
+0000c480: 656e 2069 6e73 7461 6e74 6c79 2074 6f20  en instantly to 
+0000c490: 7072 6f63 6573 7365 7320 7769 7468 6f75  processes withou
+0000c4a0: 7420 7468 6520 7379 7374 656d 2067 6f69  t the system goi
+0000c4b0: 6e67 2069 6e74 6f20 7377 6170 2061 6e64  ng into swap and
+0000c4c0: 2072 6566 6c65 6374 7320 7468 6520 6163   reflects the ac
+0000c4d0: 7475 616c 206d 656d 6f72 7920 6176 6169  tual memory avai
+0000c4e0: 6c61 626c 652e 2054 6865 2066 7265 6520  lable. The free 
+0000c4f0: 6d65 6d6f 7279 2072 6566 6572 7320 746f  memory refers to
+0000c500: 2074 6865 206d 656d 6f72 7920 6e6f 7420   the memory not 
+0000c510: 6265 696e 6720 7573 6564 2061 7420 616c  being used at al
+0000c520: 6c20 287a 6572 6f65 6429 2074 6861 7420  l (zeroed) that 
+0000c530: 6973 2072 6561 6469 6c79 2061 7661 696c  is readily avail
+0000c540: 6162 6c65 2e20 5468 6520 6465 7363 7269  able. The descri
+0000c550: 7074 696f 6e20 6162 6f76 6520 7265 6665  ption above refe
+0000c560: 7273 2074 6f20 7468 6520 646f 6375 6d65  rs to the docume
+0000c570: 6e74 6174 696f 6e20 6f66 2060 7073 7574  ntation of `psut
+0000c580: 696c 602e 0a0a 2323 2320 7767 6420 646d  il`...### wgd dm
+0000c590: 640a 0a2a 2a54 6865 2064 656c 696e 6561  d..**The delinea
+0000c5a0: 7469 6f6e 206f 6620 7768 6f6c 6520 7061  tion of whole pa
+0000c5b0: 7261 6e6f 6d65 2a2a 0a60 6060 0a77 6764  ranome**.```.wgd
+0000c5c0: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
+0000c5d0: 6f65 7275 6c65 6120 2d49 2032 202d 6520  oerulea -I 2 -e 
+0000c5e0: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+0000c5f0: 7020 3520 282d 6e6e 2920 282d 2d74 6f5f  p 5 (-nn) (--to_
+0000c600: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
+0000c610: 6e20 3429 2028 2d6f 2077 6764 5f64 6d64  n 4) (-o wgd_dmd
+0000c620: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
+0000c630: 7029 0a60 6060 200a 0a4e 6f74 6520 7468  p).``` ..Note th
+0000c640: 6174 2077 6520 646f 6e27 7420 7072 6f76  at we don't prov
+0000c650: 6964 6520 7468 6520 6461 7461 206f 6620  ide the data of 
+0000c660: 7468 6973 2063 6f64 696e 6720 7365 7175  this coding sequ
+0000c670: 656e 6365 2028 6364 7329 2066 696c 6520  ence (cds) file 
+0000c680: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
+0000c690: 6c65 6160 2062 7574 2069 7420 6361 6e20  lea` but it can 
+0000c6a0: 6265 2065 6173 696c 7920 646f 776e 6c6f  be easily downlo
+0000c6b0: 6164 6564 2061 7420 5b50 6879 746f 7a6f  aded at [Phytozo
+0000c6c0: 6d65 5d28 6874 7470 733a 2f2f 7068 7974  me](https://phyt
+0000c6d0: 6f7a 6f6d 652d 6e65 7874 2e6a 6769 2e64  ozome-next.jgi.d
+0000c6e0: 6f65 2e67 6f76 2f69 6e66 6f2f 4163 6f65  oe.gov/info/Acoe
+0000c6f0: 7275 6c65 615f 7633 5f31 2920 2873 616d  rulea_v3_1) (sam
+0000c700: 6520 666f 7220 6f74 6865 7220 5573 6167  e for other Usag
+0000c710: 6520 646f 6329 2e20 4120 7265 6d69 6e64  e doc). A remind
+0000c720: 6572 2074 6861 7420 696e 2074 6865 2069  er that in the i
+0000c730: 7373 7565 7320 736f 6d65 2075 7365 7273  ssues some users
+0000c740: 2064 6964 6e27 7420 646f 776e 6c6f 6164   didn't download
+0000c750: 2074 6865 2060 4163 6f65 7275 6c65 615f   the `Acoerulea_
+0000c760: 3332 325f 7633 2e31 2e63 6473 5f70 7269  322_v3.1.cds_pri
+0000c770: 6d61 7279 5472 616e 7363 7269 7074 4f6e  maryTranscriptOn
+0000c780: 6c79 2e66 612e 677a 6020 6275 7420 696e  ly.fa.gz` but in
+0000c790: 7374 6561 6420 7468 6520 6041 636f 6572  stead the `Acoer
+0000c7a0: 756c 6561 5f33 3232 5f76 332e 312e 6364  ulea_322_v3.1.cd
+0000c7b0: 732e 6661 2e67 7a60 2066 696c 652e 2046  s.fa.gz` file. F
+0000c7c0: 6f72 2074 6865 2063 6f6e 7374 7275 6374  or the construct
+0000c7d0: 696f 6e20 6f66 2077 686f 6c65 2070 6172  ion of whole par
+0000c7e0: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+0000c7f0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+0000c800: 6f6e 2c20 2a2a 6f6e 6c79 206f 6e65 2074  on, **only one t
+0000c810: 7261 6e73 6372 6970 7420 2874 6865 2070  ranscript (the p
+0000c820: 7269 6d61 7279 206f 6e65 2920 7065 7220  rimary one) per 
+0000c830: 6765 6e65 2073 686f 756c 6420 6265 2069  gene should be i
+0000c840: 6e63 6c75 6465 642a 2a20 7375 6368 2074  ncluded** such t
+0000c850: 6861 7420 7468 6520 2a4b 2a3c 7375 623e  hat the *K*<sub>
+0000c860: 533c 2f73 7562 3e20 6973 2072 6561 6c6c  S</sub> is reall
+0000c870: 7920 696e 6469 6361 7469 6e67 2074 6865  y indicating the
+0000c880: 2061 6765 206f 6620 6765 6e65 2064 7570   age of gene dup
+0000c890: 6c69 6361 7469 6f6e 2065 7665 6e74 2c20  lication event, 
+0000c8a0: 696e 7374 6561 6420 6f66 2061 6c74 6572  instead of alter
+0000c8b0: 6e61 7469 7665 2073 706c 6963 696e 672e  native splicing.
+0000c8c0: 2054 7261 6e73 6372 6970 746f 6d65 2064   Transcriptome d
+0000c8d0: 6174 6120 7368 6f75 6c64 2062 6520 6361  ata should be ca
+0000c8e0: 7265 6675 6c6c 7920 7472 6561 7465 6420  refully treated 
+0000c8f0: 7769 7468 2064 652d 7265 6475 6e64 616e  with de-redundan
+0000c900: 6379 2073 6f20 6173 2074 6f20 7265 6475  cy so as to redu
+0000c910: 6365 2074 6865 2066 616c 7365 2070 6f73  ce the false pos
+0000c920: 6974 6976 6520 6475 706c 6963 6174 696f  itive duplicatio
+0000c930: 6e20 6275 6d70 2063 6175 7365 6420 6279  n bump caused by
+0000c940: 2070 6572 7661 7369 7665 2061 6c74 6572   pervasive alter
+0000c950: 6e61 7469 7665 2073 706c 6963 696e 672e  native splicing.
+0000c960: 2048 6572 6520 7468 6520 696e 666c 6174   Here the inflat
+0000c970: 696f 6e20 6661 6374 6f72 2070 6172 616d  ion factor param
+0000c980: 6574 6572 2c20 6769 7665 6e20 6279 2060  eter, given by `
+0000c990: 2d49 6020 6f72 2060 2d2d 696e 666c 6174  -I` or `--inflat
+0000c9a0: 696f 6e60 2c20 6166 6665 6374 7320 7468  ion`, affects th
+0000c9b0: 6520 6772 616e 756c 6172 6974 7920 6f72  e granularity or
+0000c9c0: 2072 6573 6f6c 7574 696f 6e20 6f66 2074   resolution of t
+0000c9d0: 6865 2063 6c75 7374 6572 696e 6720 6f75  he clustering ou
+0000c9e0: 7463 6f6d 6520 616e 6420 696d 706c 6963  tcome and implic
+0000c9f0: 6974 6c79 2063 6f6e 7472 6f6c 6c73 2074  itly controlls t
+0000ca00: 6865 206e 756d 6265 7220 6f66 2063 6c75  he number of clu
+0000ca10: 7374 6572 732c 2077 6974 6820 6c6f 7720  sters, with low 
+0000ca20: 7661 6c75 6573 2073 7563 6820 6173 2031  values such as 1
+0000ca30: 2e33 206f 7220 312e 3420 6c65 6164 696e  .3 or 1.4 leadin
+0000ca40: 6720 746f 2066 6577 6572 2062 7574 206c  g to fewer but l
+0000ca50: 6172 6765 7220 636c 7573 7465 7273 2061  arger clusters a
+0000ca60: 6e64 2068 6967 6820 7661 6c75 6573 2073  nd high values s
+0000ca70: 7563 6820 6173 2035 206f 7220 3620 6c65  uch as 5 or 6 le
+0000ca80: 6164 696e 6720 746f 206d 6f72 6520 6275  ading to more bu
+0000ca90: 7420 736d 616c 6c65 7220 636c 7573 7465  t smaller cluste
+0000caa0: 7273 2e20 5765 2073 6574 2074 6865 2064  rs. We set the d
+0000cab0: 6566 6175 6c74 2076 616c 7565 2061 7320  efault value as 
+0000cac0: 3220 6173 2073 7567 6765 7374 6564 2062  2 as suggested b
+0000cad0: 7920 5b4d 434c 5d28 6874 7470 733a 2f2f  y [MCL](https://
+0000cae0: 6d69 6361 6e73 2e6f 7267 2f6d 636c 2f29  micans.org/mcl/)
+0000caf0: 2e20 5468 6520 652d 7661 6c75 6520 6375  . The e-value cu
+0000cb00: 742d 6f66 6620 666f 7220 7365 7175 656e  t-off for sequen
+0000cb10: 6365 2073 696d 696c 6172 6974 792c 2067  ce similarity, g
+0000cb20: 6976 656e 2062 7920 602d 6560 206f 7220  iven by `-e` or 
+0000cb30: 602d 2d65 7661 6c60 2c20 7768 6963 6820  `--eval`, which 
+0000cb40: 6465 6e6f 7465 7320 7468 6520 6578 7065  denotes the expe
+0000cb50: 6374 6564 2076 616c 7565 206f 6620 7468  cted value of th
+0000cb60: 6520 6869 7420 7175 616e 7469 6669 6573  e hit quantifies
+0000cb70: 2074 6865 206e 756d 6265 7220 6f66 2061   the number of a
+0000cb80: 6c69 676e 6d65 6e74 7320 6f66 2073 696d  lignments of sim
+0000cb90: 696c 6172 206f 7220 6265 7474 6572 2071  ilar or better q
+0000cba0: 7561 6c69 7479 2074 6861 7420 796f 7520  uality that you 
+0000cbb0: 6578 7065 6374 2074 6f20 6669 6e64 2073  expect to find s
+0000cbc0: 6561 7263 6869 6e67 2074 6869 7320 7175  earching this qu
+0000cbd0: 6572 7920 6167 6169 6e73 7420 6120 6461  ery against a da
+0000cbe0: 7461 6261 7365 206f 6620 7261 6e64 6f6d  tabase of random
+0000cbf0: 2073 6571 7565 6e63 6573 2074 6865 2073   sequences the s
+0000cc00: 616d 6520 7369 7a65 2061 7320 7468 6520  ame size as the 
+0000cc10: 6163 7475 616c 2074 6172 6765 7420 6461  actual target da
+0000cc20: 7461 6261 7365 2c20 6973 2074 6865 206b  tabase, is the k
+0000cc30: 6579 2070 6172 616d 6574 6572 206d 6561  ey parameter mea
+0000cc40: 7375 7269 6e67 2074 6865 2073 6967 6e69  suring the signi
+0000cc50: 6669 6361 6e63 6520 6f66 2061 2068 6974  ficance of a hit
+0000cc60: 2c20 7768 6963 6820 6973 2073 6574 2068  , which is set h
+0000cc70: 6572 6520 6173 2064 6566 6175 6c74 2031  ere as default 1
+0000cc80: 652d 3130 2e20 4e6f 7465 2074 6861 7420  e-10. Note that 
+0000cc90: 5b44 4941 4d4f 4e44 5d28 6874 7470 733a  [DIAMOND](https:
+0000cca0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 6275  //github.com/bbu
+0000ccb0: 6368 6669 6e6b 2f64 6961 6d6f 6e64 2f77  chfink/diamond/w
+0000ccc0: 696b 692f 312e 2d54 7574 6f72 6961 6c29  iki/1.-Tutorial)
+0000ccd0: 2069 7473 656c 6620 6279 2064 6566 6175   itself by defau
+0000cce0: 6c74 206f 6e6c 7920 7265 706f 7274 7320  lt only reports 
+0000ccf0: 616c 6c20 616c 6967 6e6d 656e 7473 2077  all alignments w
+0000cd00: 6974 6820 652d 7661 6c75 6520 3c20 302e  ith e-value < 0.
+0000cd10: 3030 312e 2054 6865 2070 6572 6365 6e74  001. The percent
+0000cd20: 6167 6520 6f66 2075 7070 6572 2068 6974  age of upper hit
+0000cd30: 7320 7573 6564 2066 6f72 2067 656e 6520  s used for gene 
+0000cd40: 6c65 6e67 7468 206e 6f72 6d61 6c69 7a61  length normaliza
+0000cd50: 7469 6f6e 2c20 6769 7665 6e20 6279 2060  tion, given by `
+0000cd60: 2d6e 7060 206f 7220 602d 2d6e 6f72 6d61  -np` or `--norma
+0000cd70: 6c69 7a65 6470 6572 6365 6e74 602c 2077  lizedpercent`, w
+0000cd80: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
+0000cd90: 7468 6520 7570 7065 7220 7065 7263 656e  the upper percen
+0000cda0: 7469 6c65 206f 6620 6869 7473 2070 6572  tile of hits per
+0000cdb0: 2062 696e 2028 6361 7465 676f 7269 7a65   bin (categorize
+0000cdc0: 6420 6279 2067 656e 6520 6c65 6e67 7468  d by gene length
+0000cdd0: 2920 7573 6564 2069 6e20 7468 6520 6669  ) used in the fi
+0000cde0: 7420 6f66 206c 696e 6561 7220 7265 6772  t of linear regr
+0000cdf0: 6573 7369 6f6e 2c20 636f 6e73 6964 6572  ession, consider
+0000ce00: 696e 6720 7468 6174 206e 6f74 2061 6c6c  ing that not all
+0000ce10: 2068 6974 7320 7065 7220 6269 6e20 7368   hits per bin sh
+0000ce20: 6f77 2061 7070 6172 656e 7420 6c69 6e65  ow apparent line
+0000ce30: 6172 2072 656c 6174 696f 6e73 6869 702c  ar relationship,
+0000ce40: 2069 7320 7365 7420 6173 2064 6566 6175   is set as defau
+0000ce50: 6c74 2035 2c20 696e 6469 6361 7469 6e67  lt 5, indicating
+0000ce60: 2074 6865 2075 7361 6765 206f 6620 746f   the usage of to
+0000ce70: 7020 3525 2068 6974 7320 7065 7220 6269  p 5% hits per bi
+0000ce80: 6e2e 2054 6865 206e 756d 6265 7220 6f66  n. The number of
+0000ce90: 2062 696e 7320 6469 7669 6465 6420 696e   bins divided in
+0000cea0: 2067 656e 6520 6c65 6e67 7468 206e 6f72   gene length nor
+0000ceb0: 6d61 6c69 7a61 7469 6f6e 2c20 6769 7665  malization, give
+0000cec0: 6e20 6279 2060 2d62 7360 206f 7220 602d  n by `-bs` or `-
+0000ced0: 2d62 696e 7360 2c20 6465 7465 726d 696e  -bins`, determin
+0000cee0: 6573 2074 6865 206e 756d 6265 7220 6f66  es the number of
+0000cef0: 2062 696e 7320 746f 2063 6174 6567 6f72   bins to categor
+0000cf00: 697a 6520 7468 6520 6765 6e65 206c 656e  ize the gene len
+0000cf10: 6774 682c 2069 7320 7365 7420 6173 2064  gth, is set as d
+0000cf20: 6566 6175 6c74 2031 3030 2e20 5468 6520  efault 100. The 
+0000cf30: 7061 7261 6d65 7465 7220 602d 6e6e 6020  parameter `-nn` 
+0000cf40: 6f72 2060 2d2d 6e6f 6e6f 726d 616c 697a  or `--nonormaliz
+0000cf50: 6174 696f 6e60 2063 616e 2062 6520 7365  ation` can be se
+0000cf60: 7420 746f 2063 616c 6c20 6f66 6620 7468  t to call off th
+0000cf70: 6520 6e6f 726d 616c 697a 6174 696f 6e20  e normalization 
+0000cf80: 7072 6f63 6573 732c 2061 6c74 686f 7567  process, althoug
+0000cf90: 6820 6974 2773 2073 7567 6765 7374 6564  h it's suggested
+0000cfa0: 2074 6f20 636f 6e64 7563 7420 7468 6520   to conduct the 
+0000cfb0: 6e6f 726d 616c 697a 6174 696f 6e20 746f  normalization to
+0000cfc0: 2061 6371 7569 7265 206d 6f72 6520 6163   acquire more ac
+0000cfd0: 6375 7261 7465 2067 656e 6520 6661 6d69  curate gene fami
+0000cfe0: 6c79 2063 6c75 7374 6572 696e 6720 7265  ly clustering re
+0000cff0: 7375 6c74 2e20 5468 6520 7061 7261 6d65  sult. The parame
+0000d000: 7465 7273 2060 2d2d 746f 5f73 746f 7060  ters `--to_stop`
+0000d010: 2061 6e64 2060 2d2d 6364 7360 2063 6f6e   and `--cds` con
+0000d020: 7472 6f6c 2074 6865 2062 6568 6176 696f  trol the behavio
+0000d030: 7572 206f 6620 7472 616e 736c 6174 696e  ur of translatin
+0000d040: 6720 636f 6469 6e67 2073 6571 7565 6e63  g coding sequenc
+0000d050: 6520 696e 746f 2061 6d69 6e6f 2061 6369  e into amino aci
+0000d060: 6420 7365 7175 656e 6365 2e20 4966 2074  d sequence. If t
+0000d070: 6865 2060 2d2d 746f 5f73 746f 7060 2077  he `--to_stop` w
+0000d080: 6173 2073 6574 2c20 7468 6520 7472 616e  as set, the tran
+0000d090: 736c 6174 696f 6e20 776f 756c 6420 6265  slation would be
+0000d0a0: 2074 6572 6d69 6e61 7465 6420 6174 2074   terminated at t
+0000d0b0: 6865 2066 6972 7374 2069 6e2d 6672 616d  he first in-fram
+0000d0c0: 6520 7374 6f70 2063 6f64 6f6e 2c20 6f74  e stop codon, ot
+0000d0d0: 6865 7277 6973 6520 7468 6520 7472 616e  herwise the tran
+0000d0e0: 736c 6174 696f 6e20 776f 756c 6420 7369  slation would si
+0000d0f0: 6d70 6c79 2073 6b69 7020 616e 7920 7374  mply skip any st
+0000d100: 6f70 2063 6f64 6f6e 732e 2049 6620 7468  op codons. If th
+0000d110: 6520 602d 2d63 6473 6020 7761 7320 7365  e `--cds` was se
+0000d120: 742c 2073 6571 7565 6e63 6573 2074 6861  t, sequences tha
+0000d130: 7420 646f 6573 6e27 7420 7374 6172 7420  t doesn't start 
+0000d140: 7769 7468 2061 2076 616c 6964 2073 7461  with a valid sta
+0000d150: 7274 2063 6f64 6f6e 2c20 6f72 2063 6f6e  rt codon, or con
+0000d160: 7461 696e 7320 6d6f 7265 2074 6861 6e20  tains more than 
+0000d170: 6f6e 6520 696e 2d66 7261 6d65 2073 746f  one in-frame sto
+0000d180: 7020 636f 646f 6e2c 206f 7220 6973 206e  p codon, or is n
+0000d190: 6f74 2064 6976 6964 6162 6c65 2062 7920  ot dividable by 
+0000d1a0: 332c 2077 6f75 6c64 2062 6520 7369 6d70  3, would be simp
+0000d1b0: 6c79 2064 726f 7070 6564 2c20 7375 6368  ly dropped, such
+0000d1c0: 2074 6861 7420 6f6e 6c79 2073 7472 6963   that only stric
+0000d1d0: 7420 636f 6d70 6c65 7465 2063 6f64 696e  t complete codin
+0000d1e0: 6720 7365 7175 656e 6365 7320 776f 756c  g sequences woul
+0000d1f0: 6420 6265 2069 6e63 6c75 6465 6420 696e  d be included in
+0000d200: 2074 6865 2073 7562 7365 7175 656e 7420   the subsequent 
+0000d210: 616e 616c 7973 6973 2e20 5468 6520 6578  analysis. The ex
+0000d220: 6163 7420 6265 6861 7669 6f75 7220 6f66  act behaviour of
+0000d230: 2060 2d2d 746f 5f73 746f 7060 2061 6e64   `--to_stop` and
+0000d240: 2060 2d2d 6364 7360 2069 7320 6465 6669   `--cds` is defi
+0000d250: 6e65 6420 616e 6420 6465 7363 7269 6265  ned and describe
+0000d260: 6420 696e 2074 6865 205b 6269 6f70 7974  d in the [biopyt
+0000d270: 686f 6e5d 2868 7474 7073 3a2f 2f62 696f  hon](https://bio
+0000d280: 7079 7468 6f6e 2e6f 7267 2f64 6f63 732f  python.org/docs/
+0000d290: 312e 3735 2f61 7069 2f42 696f 2e53 6571  1.75/api/Bio.Seq
+0000d2a0: 2e68 746d 6c23 4269 6f2e 5365 712e 5365  .html#Bio.Seq.Se
+0000d2b0: 712e 7472 616e 736c 6174 6529 206c 6962  q.translate) lib
+0000d2c0: 7261 7279 2e20 5468 6520 6e75 6d62 6572  rary. The number
+0000d2d0: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
+0000d2e0: 6561 6473 2062 7920 7468 6520 6f70 7469  eads by the opti
+0000d2f0: 6f6e 2060 2d6e 6020 6f72 2060 2d2d 6e74  on `-n` or `--nt
+0000d300: 6872 6561 6473 6020 6361 6e20 6265 2073  hreads` can be s
+0000d310: 6574 2074 6f20 6163 6365 6c65 7261 7465  et to accelerate
+0000d320: 2074 6865 2063 616c 6375 6c61 7469 6f6e   the calculation
+0000d330: 2077 6974 6869 6e20 6064 6961 6d6f 6e64   within `diamond
+0000d340: 602e 2054 6865 2064 6972 6563 746f 7279  `. The directory
+0000d350: 206f 6620 6f75 7470 7574 206f 7220 696e   of output or in
+0000d360: 7465 726d 6564 6961 7465 2066 696c 6573  termediate files
+0000d370: 2069 7320 6465 7465 726d 696e 6564 2062   is determined b
+0000d380: 7920 7468 6520 7061 7261 6d65 7465 7220  y the parameter 
+0000d390: 602d 6f60 206f 7220 602d 2d6f 7574 6469  `-o` or `--outdi
+0000d3a0: 7260 2c20 616e 6420 602d 7460 206f 7220  r`, and `-t` or 
+0000d3b0: 602d 2d74 6d70 6469 7260 2c20 7768 6963  `--tmpdir`, whic
+0000d3c0: 6820 7769 6c6c 2062 6520 6372 6561 7465  h will be create
+0000d3d0: 6420 6279 2074 6865 2070 726f 6772 616d  d by the program
+0000d3e0: 2069 7473 656c 6620 616e 6420 6265 206f   itself and be o
+0000d3f0: 7665 7277 7269 7474 656e 2069 6620 7468  verwritten if th
+0000d400: 6520 666f 6c64 6572 2068 6173 2061 6c72  e folder has alr
+0000d410: 6561 6479 2062 6565 6e20 6372 6561 7465  eady been create
+0000d420: 642e 204e 6f74 6520 7468 6174 2074 6865  d. Note that the
+0000d430: 2073 6f66 7477 6172 6520 6064 6961 6d6f   software `diamo
+0000d440: 6e64 6020 7368 6f75 6c64 2062 6520 7072  nd` should be pr
+0000d450: 652d 696e 7374 616c 6c65 6420 616e 6420  e-installed and 
+0000d460: 7365 7420 696e 2074 6865 2065 6e76 6972  set in the envir
+0000d470: 6f6e 6d65 6e74 2070 6174 6820 696e 2061  onment path in a
+0000d480: 6c6c 2074 6865 2061 6e61 6c79 7369 7320  ll the analysis 
+0000d490: 7065 7266 6f72 6d65 6420 6279 2060 7767  performed by `wg
+0000d4a0: 6420 646d 6460 2065 7863 6570 7420 666f  d dmd` except fo
+0000d4b0: 7220 7468 6520 636f 6c6c 696e 6561 7220  r the collinear 
+0000d4c0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+0000d4d0: 7973 6973 2e0a 0a2a 2a57 6520 7375 6767  ysis...**We sugg
+0000d4e0: 6573 7420 7468 6174 2074 6865 2064 6566  est that the def
+0000d4f0: 6175 6c74 2073 6574 7469 6e67 2069 6e20  ault setting in 
+0000d500: 7768 6963 6820 7468 6520 696e 666c 6174  which the inflat
+0000d510: 696f 6e20 6661 6374 6f72 2069 7320 7365  ion factor is se
+0000d520: 7420 6173 2032 2c20 652d 7661 6c75 6520  t as 2, e-value 
+0000d530: 6375 742d 6f66 6620 6173 2031 652d 3130  cut-off as 1e-10
+0000d540: 2061 6e64 206f 7468 6572 2070 6172 616d   and other param
+0000d550: 6574 6572 7320 696e 2064 6566 6175 6c74  eters in default
+0000d560: 2069 7320 6120 676f 6f64 2073 7461 7274   is a good start
+0000d570: 696e 6720 706f 696e 742c 2075 6e6c 6573  ing point, unles
+0000d580: 7320 796f 7520 7370 6563 6966 6963 616c  s you specifical
+0000d590: 6c79 2077 616e 7420 746f 2065 7870 6c6f  ly want to explo
+0000d5a0: 7265 2074 6865 2065 6666 6563 7473 206f  re the effects o
+0000d5b0: 6620 6469 6666 6572 656e 7420 7061 7261  f different para
+0000d5c0: 6d65 7465 7273 2e2a 2a20 5375 6368 2074  meters.** Such t
+0000d5d0: 6861 7420 7468 6520 636f 6d6d 616e 6420  hat the command 
+0000d5e0: 666f 7220 7468 6520 6465 6c69 6e65 6174  for the delineat
+0000d5f0: 696f 6e20 6f66 2077 686f 6c65 2070 6172  ion of whole par
+0000d600: 616e 6f6d 6520 6973 2073 696d 706c 7920  anome is simply 
+0000d610: 6173 2062 656c 6f77 2e0a 0a60 6060 0a77  as below...```.w
+0000d620: 6764 2064 6d64 2041 7175 696c 6567 6961  gd dmd Aquilegia
+0000d630: 5f63 6f65 7275 6c65 610a 6060 600a 0a2a  _coerulea.```..*
+0000d640: 2a54 6865 2064 656c 696e 6561 7469 6f6e  *The delineation
+0000d650: 206f 6620 5242 4873 2a2a 0a60 6060 0a77   of RBHs**.```.w
+0000d660: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
+0000d670: 2073 6571 7565 6e63 6532 202d 6520 3165   sequence2 -e 1e
+0000d680: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
+0000d690: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
+0000d6a0: 2d63 2030 2e39 2920 282d 2d6f 6766 6f72  -c 0.9) (--ogfor
+0000d6b0: 6d61 7429 2028 2d2d 746f 5f73 746f 7029  mat) (--to_stop)
+0000d6c0: 2028 2d2d 6364 7329 2028 2d6f 2077 6764   (--cds) (-o wgd
+0000d6d0: 5f64 6d64 2920 282d 7420 776f 726b 696e  _dmd) (-t workin
+0000d6e0: 675f 746d 7029 0a60 6060 0a0a 546f 2064  g_tmp).```..To d
+0000d6f0: 656c 696e 6561 7465 2052 4248 7320 6265  elineate RBHs be
+0000d700: 7477 6565 6e20 7477 6f20 6364 7320 7365  tween two cds se
+0000d710: 7175 656e 6365 2066 696c 6573 2c20 7468  quence files, th
+0000d720: 6520 7265 6c65 7661 6e74 2070 6172 616d  e relevant param
+0000d730: 6574 6572 2069 7320 6d6f 7374 6c79 2074  eter is mostly t
+0000d740: 6865 2073 616d 6520 6173 2077 686f 6c65  he same as whole
+0000d750: 2070 6172 616e 6f6d 6520 696e 6665 7265   paranome infere
+0000d760: 6e63 652c 2065 7863 6570 7420 666f 7220  nce, except for 
+0000d770: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
+0000d780: 6360 206f 7220 602d 2d63 7363 6f72 6560  c` or `--cscore`
+0000d790: 2c20 7768 6963 6820 7261 6e67 6573 2062  , which ranges b
+0000d7a0: 6574 7765 656e 2030 2061 6e64 2031 2061  etween 0 and 1 a
+0000d7b0: 6e64 2069 7320 7573 6564 2074 6f20 7265  nd is used to re
+0000d7c0: 6c61 7820 7468 6520 7369 6d69 6c61 7269  lax the similari
+0000d7d0: 7479 2063 7574 6f66 6620 6672 6f6d 2074  ty cutoff from t
+0000d7e0: 6865 2065 7863 6c75 7369 7665 2072 6563  he exclusive rec
+0000d7f0: 6970 726f 6361 6c20 6265 7374 2068 6974  iprocal best hit
+0000d800: 7320 746f 2061 2063 6572 7461 696e 2072  s to a certain r
+0000d810: 6174 696f 2061 7320 746f 2074 6865 2062  atio as to the b
+0000d820: 6573 7420 6869 7473 2e20 466f 7220 696e  est hits. For in
+0000d830: 7374 616e 6365 2c20 6966 2074 6865 2067  stance, if the g
+0000d840: 656e 6520 6231 2066 726f 6d20 6765 6e6f  ene b1 from geno
+0000d850: 6d65 2042 2068 6173 2074 6865 2062 6573  me B has the bes
+0000d860: 7420 6869 7420 6765 6e65 2061 3120 6672  t hit gene a1 fr
+0000d870: 6f6d 2067 656e 6f6d 6520 4120 7769 7468  om genome A with
+0000d880: 2074 6865 2062 6974 2073 636f 7265 2061   the bit score a
+0000d890: 7320 3130 302c 2077 6869 6368 2069 7320  s 100, which is 
+0000d8a0: 6120 7363 6f72 696e 6720 6d61 7472 6978  a scoring matrix
+0000d8b0: 2069 6e64 6570 656e 6465 6e74 206d 6561   independent mea
+0000d8c0: 7375 7265 206f 6620 7468 6520 286c 6f63  sure of the (loc
+0000d8d0: 616c 2920 7369 6d69 6c61 7269 7479 206f  al) similarity o
+0000d8e0: 6620 7468 6520 7477 6f20 616c 6967 6e65  f the two aligne
+0000d8f0: 6420 7365 7175 656e 6365 732c 2077 6974  d sequences, wit
+0000d900: 6820 6c61 7267 6572 2076 616c 7565 7320  h larger values 
+0000d910: 696e 6469 6361 7469 6e67 2068 6967 6865  indicating highe
+0000d920: 7220 7369 6d69 6c61 7269 7469 6573 2c20  r similarities, 
+0000d930: 6769 7665 6e20 7468 6520 602d 6320 302e  given the `-c 0.
+0000d940: 3960 2c20 6765 6e65 7320 6672 6f6d 2067  9`, genes from g
+0000d950: 656e 6f6d 6520 4120 7768 6963 6820 6861  enome A which ha
+0000d960: 7320 7468 6520 6269 7420 7363 6f72 6520  s the bit score 
+0000d970: 7769 7468 2067 656e 6520 6231 2068 6967  with gene b1 hig
+0000d980: 6865 7220 7468 616e 2030 2e39 7831 3030  her than 0.9x100
+0000d990: 2077 696c 6c20 616c 736f 2062 6520 7772   will also be wr
+0000d9a0: 6974 7465 6e20 696e 2074 6865 2072 6573  itten in the res
+0000d9b0: 756c 7420 6669 6c65 2c20 7768 6963 6820  ult file, which 
+0000d9c0: 696e 2061 2073 656e 7365 2061 7265 206e  in a sense are n
+0000d9d0: 6f74 2052 4248 7320 616e 796d 6f72 6520  ot RBHs anymore 
+0000d9e0: 6f66 2063 6f75 7273 652c 2062 7574 2074  of course, but t
+0000d9f0: 6865 2068 6967 686c 7920 7369 6d69 6c61  he highly simila
+0000da00: 7220 686f 6d6f 6c6f 6775 6520 7061 6972  r homologue pair
+0000da10: 732e 2049 6620 6d6f 7265 2074 6861 6e20  s. If more than 
+0000da20: 3220 7365 7175 656e 6365 2066 696c 6573  2 sequence files
+0000da30: 2077 6572 6520 7072 6f76 6964 6564 2c20   were provided, 
+0000da40: 6576 6572 7920 7061 6972 2d77 6973 6520  every pair-wise 
+0000da50: 5242 4873 2077 6f75 6c64 2062 6520 6361  RBHs would be ca
+0000da60: 6c63 756c 6174 6564 2065 7863 6570 7420  lculated except 
+0000da70: 666f 7220 7175 6572 7969 6e67 2074 6865  for querying the
+0000da80: 2073 616d 6520 7365 7175 656e 6365 2069   same sequence i
+0000da90: 7473 656c 662e 2054 6865 206e 756d 6265  tself. The numbe
+0000daa0: 7220 6f66 2070 6172 616c 6c65 6c20 7468  r of parallel th
+0000dab0: 7265 6164 7320 746f 2062 6f6f 7374 6572  reads to booster
+0000dac0: 2074 6865 2072 756e 6e69 6e67 2073 7065   the running spe
+0000dad0: 6564 2063 616e 2062 6520 7365 7420 6279  ed can be set by
+0000dae0: 2074 6865 206f 7074 696f 6e20 602d 6e60   the option `-n`
+0000daf0: 206f 7220 602d 2d6e 7468 7265 6164 7360   or `--nthreads`
+0000db00: 2077 6869 6368 2069 7320 7375 6767 6573   which is sugges
+0000db10: 7465 6420 746f 2062 6520 7365 7420 6173  ted to be set as
+0000db20: 2028 4e2d 3129 4e2f 3220 7768 6572 6520   (N-1)N/2 where 
+0000db30: 4e20 6973 2074 6865 206e 756d 6265 7220  N is the number 
+0000db40: 6f66 2063 6473 2066 696c 6573 2074 6f20  of cds files to 
+0000db50: 6163 6869 6576 6520 7468 6520 6869 6768  achieve the high
+0000db60: 6573 7420 6566 6669 6369 656e 6379 2e20  est efficiency. 
+0000db70: 5468 6520 6f70 7469 6f6e 2060 2d2d 6f67  The option `--og
+0000db80: 666f 726d 6174 6020 6361 6e20 6265 2073  format` can be s
+0000db90: 6574 2074 6f20 6164 6420 696e 6465 7820  et to add index 
+0000dba0: 2866 6f72 2069 6e73 7461 6e63 6520 4746  (for instance GF
+0000dbb0: 3030 3030 3030 3031 2920 746f 2074 6865  00000001) to the
+0000dbc0: 206f 7574 7075 7420 5242 4820 6765 6e65   output RBH gene
+0000dbd0: 2066 616d 696c 6965 7320 7768 6963 6820   families which 
+0000dbe0: 6361 6e20 6265 2066 7572 7468 6572 2075  can be further u
+0000dbf0: 7365 6420 696e 2074 6865 202a 4b2a 3c73  sed in the *K*<s
+0000dc00: 7562 3e53 3c2f 7375 623e 2063 616c 6375  ub>S</sub> calcu
+0000dc10: 6c61 7469 6f6e 2062 7920 6077 6764 206b  lation by `wgd k
+0000dc20: 7364 602e 0a0a 2a2a 5468 6520 7375 6767  sd`...**The sugg
+0000dc30: 6573 7465 6420 636f 6d6d 616e 6420 746f  ested command to
+0000dc40: 2073 7461 7274 2077 6974 6820 6973 2061   start with is a
+0000dc50: 6c73 6f20 756e 6465 7220 7468 6520 6465  lso under the de
+0000dc60: 6661 756c 7420 7365 7474 696e 6720 7769  fault setting wi
+0000dc70: 7468 2074 6865 2063 6f6d 6d61 6e64 2073  th the command s
+0000dc80: 686f 776e 2062 656c 6f77 2a2a 0a0a 6060  hown below**..``
+0000dc90: 600a 7767 6420 646d 6420 7365 7175 656e  `.wgd dmd sequen
+0000dca0: 6365 3120 7365 7175 656e 6365 320a 6060  ce1 sequence2.``
+0000dcb0: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
+0000dcc0: 7469 6f6e 206f 6620 6c6f 6361 6c20 4d52  tion of local MR
+0000dcd0: 4248 732a 2a0a 6060 600a 7767 6420 646d  BHs**.```.wgd dm
+0000dce0: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
+0000dcf0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000dd00: 2d66 2073 6571 7565 6e63 6531 202d 6520  -f sequence1 -e 
+0000dd10: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+0000dd20: 7020 3520 282d 6e6e 2920 282d 6e20 3429  p 5 (-nn) (-n 4)
+0000dd30: 2028 2d6b 6629 2028 2d6b 6429 2028 2d63   (-kf) (-kd) (-c
+0000dd40: 2030 2e39 2920 282d 2d74 6f5f 7374 6f70   0.9) (--to_stop
+0000dd50: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
+0000dd60: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
+0000dd70: 6e67 5f74 6d70 290a 6060 600a 0a2a 2a54  ng_tmp).```..**T
+0000dd80: 6865 2064 6973 7469 6e63 7469 6f6e 2062  he distinction b
+0000dd90: 6574 7765 656e 206c 6f63 616c 2061 6e64  etween local and
+0000dda0: 2067 6c6f 6261 6c20 2868 6572 6575 6e64   global (hereund
+0000ddb0: 6572 2920 4d52 4248 7320 6973 2074 6861  er) MRBHs is tha
+0000ddc0: 7420 6c6f 6361 6c20 4d52 4248 7320 6172  t local MRBHs ar
+0000ddd0: 6520 7468 6520 7265 7375 6c74 7320 6f66  e the results of
+0000dde0: 206d 6572 6765 6420 5242 4873 206f 6e20   merged RBHs on 
+0000ddf0: 6120 6a6f 696e 7420 666f 6361 6c20 7370  a joint focal sp
+0000de00: 6563 6965 732c 2066 6f72 2069 6e73 7461  ecies, for insta
+0000de10: 6e63 6520 696e 2061 2074 6872 6565 2073  nce in a three s
+0000de20: 7065 6369 6573 2073 7973 7465 6d20 2841  pecies system (A
+0000de30: 2c28 422c 4329 292c 2074 6865 206c 6f63  ,(B,C)), the loc
+0000de40: 616c 204d 5242 4873 206f 6620 4120 6f6e  al MRBHs of A on
+0000de50: 6c79 2072 6571 7569 7265 2074 6865 2063  ly require the c
+0000de60: 616c 6375 6c61 7469 6f6e 206f 6620 5242  alculation of RB
+0000de70: 4873 2062 6574 7765 656e 2041 2061 6e64  Hs between A and
+0000de80: 2043 2028 6465 6e6f 7465 6420 6173 2041   C (denoted as A
+0000de90: 4329 2061 6e64 2041 4220 616e 6420 7468  C) and AB and th
+0000dea0: 656e 2074 6865 206d 6572 6769 6e67 206f  en the merging o
+0000deb0: 6620 4142 2061 6e64 2041 4320 6174 2074  f AB and AC at t
+0000dec0: 6865 2061 7869 7320 6f66 2041 2c20 7768  he axis of A, wh
+0000ded0: 696c 6520 7468 6520 676c 6f61 626c 204d  ile the gloabl M
+0000dee0: 5242 4873 2061 7265 2069 6e64 6570 656e  RBHs are indepen
+0000def0: 6465 6e74 206f 6620 666f 6361 6c20 7370  dent of focal sp
+0000df00: 6563 6965 7320 696e 2074 6861 7420 6974  ecies in that it
+0000df10: 206a 7573 7420 6361 6c6c 7320 7468 6520   just calls the 
+0000df20: 6361 6c63 756c 6174 696f 6e73 206f 6620  calculations of 
+0000df30: 616c 6c20 706f 7373 6962 6c65 2073 7065  all possible spe
+0000df40: 6369 6573 2070 6169 7220 286e 6f74 2073  cies pair (not s
+0000df50: 656c 6620 746f 2073 656c 6620 7370 6563  elf to self spec
+0000df60: 6965 7320 7061 6972 292c 2073 7563 6820  ies pair), such 
+0000df70: 7468 6174 2041 422c 2041 432c 2061 6e64  that AB, AC, and
+0000df80: 2042 4320 6172 6520 616c 6c20 746f 2062   BC are all to b
+0000df90: 6520 6361 6c63 756c 6174 6564 2061 6e64  e calculated and
+0000dfa0: 206d 6572 6765 642e 2a2a 0a0a 5477 6f20   merged.**..Two 
+0000dfb0: 7479 7065 7320 6f66 204d 5242 4873 2061  types of MRBHs a
+0000dfc0: 7320 696e 7465 7072 6574 6174 6564 2061  s intepretated a
+0000dfd0: 626f 7665 2063 616e 2062 6520 6465 6c69  bove can be deli
+0000dfe0: 6e65 6174 6564 2062 7920 6077 6764 2064  neated by `wgd d
+0000dff0: 6d64 602c 2074 6865 206c 6f63 616c 204d  md`, the local M
+0000e000: 5242 4873 2061 6e64 2074 6865 2067 6c6f  RBHs and the glo
+0000e010: 6261 6c20 4d52 4248 732e 2054 6865 206c  bal MRBHs. The l
+0000e020: 6f63 616c 204d 5242 4873 2061 7265 2063  ocal MRBHs are c
+0000e030: 6f6e 7374 7275 6374 6564 2062 7920 6d65  onstructed by me
+0000e040: 7267 696e 6720 616c 6c20 7468 6520 7265  rging all the re
+0000e050: 6c65 7661 6e74 2052 4248 7320 6f6e 6c79  levant RBHs only
+0000e060: 2077 6974 6820 7468 6520 666f 6361 6c20   with the focal 
+0000e070: 7370 6563 6965 732c 2077 6869 6368 2069  species, which i
+0000e080: 7320 7365 7420 6279 2060 2d66 6020 6f72  s set by `-f` or
+0000e090: 2060 2d2d 666f 6375 7360 2e20 5468 6520   `--focus`. The 
+0000e0a0: 7061 7261 6d65 7465 7220 602d 6b66 6020  parameter `-kf` 
+0000e0b0: 6f72 2060 2d2d 6b65 6570 6661 7374 6160  or `--keepfasta`
+0000e0c0: 2063 616e 2062 6520 7365 7420 746f 2072   can be set to r
+0000e0d0: 6574 6169 6e20 7468 6520 7365 7175 656e  etain the sequen
+0000e0e0: 6365 2069 6e66 6f72 6d61 7469 6f6e 206f  ce information o
+0000e0f0: 6620 6561 6368 204d 5242 482e 2054 6865  f each MRBH. The
+0000e100: 2070 6172 616d 6574 6572 2060 2d6b 6460   parameter `-kd`
+0000e110: 206f 7220 602d 2d6b 6565 7064 7570 6c69   or `--keepdupli
+0000e120: 6361 7465 7360 2064 6574 6572 6d69 6e65  cates` determine
+0000e130: 7320 7768 6574 6865 7220 7468 6520 7361  s whether the sa
+0000e140: 6d65 2067 656e 6573 2063 616e 2061 7070  me genes can app
+0000e150: 6561 7220 696e 2064 6966 6665 7265 6e74  ear in different
+0000e160: 206c 6f63 616c 204d 5242 4873 2e20 4e6f   local MRBHs. No
+0000e170: 726d 616c 6c79 2074 6865 7265 2077 696c  rmally there wil
+0000e180: 6c20 6265 206e 6f20 6475 706c 6963 6174  l be no duplicat
+0000e190: 6573 2069 6e20 7468 6520 6c6f 6361 6c20  es in the local 
+0000e1a0: 4d52 4248 7320 6275 7420 6966 2075 7365  MRBHs but if use
+0000e1b0: 7273 2073 6574 2074 6865 2060 2d63 6020  rs set the `-c` 
+0000e1c0: 6173 2030 2e39 2028 6f72 2061 6e79 2076  as 0.9 (or any v
+0000e1d0: 616c 7565 2073 6d61 6c6c 6572 2074 6861  alue smaller tha
+0000e1e0: 6e20 3129 2c20 6974 2773 206c 696b 656c  n 1), it's likel
+0000e1f0: 7920 7468 6174 2074 6865 2073 616d 6520  y that the same 
+0000e200: 6765 6e65 2077 696c 6c20 6861 7665 2063  gene will have c
+0000e210: 6861 6e63 6520 746f 2061 7070 6561 7220  hance to appear 
+0000e220: 6d75 6c74 6970 6c65 2074 696d 6573 2069  multiple times i
+0000e230: 6e20 6469 6666 6572 656e 7420 6c6f 6361  n different loca
+0000e240: 6c20 4d52 4248 732e 2054 6861 7420 6973  l MRBHs. That is
+0000e250: 2074 6f20 7361 792c 2074 6865 2070 6172   to say, the par
+0000e260: 616d 6574 6572 2060 2d6b 6460 2069 7320  ameter `-kd` is 
+0000e270: 6d65 616e 696e 6766 756c 206f 6e6c 7920  meaningful only 
+0000e280: 7768 656e 2069 7427 7320 7365 7420 746f  when it's set to
+0000e290: 6765 7468 6572 2077 6974 6820 7468 6520  gether with the 
+0000e2a0: 7061 7261 6d65 7465 7220 602d 6360 2e20  parameter `-c`. 
+0000e2b0: 5468 6520 6e75 6d62 6572 206f 6620 7061  The number of pa
+0000e2c0: 7261 6c6c 656c 2074 6872 6561 6473 2069  rallel threads i
+0000e2d0: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+0000e2e0: 6520 7365 7420 6173 2074 6865 206e 756d  e set as the num
+0000e2f0: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
+0000e300: 206d 696e 7573 2031 2e0a 0a2a 2a41 2073   minus 1...**A s
+0000e310: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+0000e320: 6720 7275 6e20 6973 2075 6e64 6572 2074  g run is under t
+0000e330: 6865 2064 6566 6175 6c74 2070 6172 616d  he default param
+0000e340: 6574 6572 2077 6974 6820 7468 6520 636f  eter with the co
+0000e350: 6d6d 616e 6420 7368 6f77 6e20 6173 2062  mmand shown as b
+0000e360: 656c 6f77 2e2a 2a0a 0a60 6060 0a77 6764  elow.**..```.wgd
+0000e370: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+0000e380: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000e390: 6533 202d 6620 7365 7175 656e 6365 310a  e3 -f sequence1.
+0000e3a0: 6060 600a 0a2a 2a54 6865 2064 656c 696e  ```..**The delin
+0000e3b0: 6561 7469 6f6e 206f 6620 676c 6f62 616c  eation of global
+0000e3c0: 204d 5242 4873 2a2a 0a60 6060 0a77 6764   MRBHs**.```.wgd
+0000e3d0: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+0000e3e0: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000e3f0: 6533 202d 676d 202d 6520 3165 2d31 3020  e3 -gm -e 1e-10 
+0000e400: 2d62 7320 3130 3020 2d6e 7020 3520 282d  -bs 100 -np 5 (-
+0000e410: 6e6e 2920 282d 6e20 3429 2028 2d6b 6629  nn) (-n 4) (-kf)
+0000e420: 2028 2d6b 6429 2028 2d63 2030 2e39 2920   (-kd) (-c 0.9) 
+0000e430: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
+0000e440: 6473 2920 282d 6f20 7767 645f 646d 6429  ds) (-o wgd_dmd)
+0000e450: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
+0000e460: 290a 6060 600a 0a54 6865 2067 6c6f 6261  ).```..The globa
+0000e470: 6c20 4d52 4248 7320 6973 2063 6f6e 7374  l MRBHs is const
+0000e480: 7275 6374 6564 2062 7920 6578 6861 7573  ructed by exhaus
+0000e490: 7469 7665 6c79 206d 6572 6769 6e67 2061  tively merging a
+0000e4a0: 6c6c 2074 6865 2070 6f73 7369 626c 6520  ll the possible 
+0000e4b0: 7061 6972 2d77 6973 6520 5242 4873 2065  pair-wise RBHs e
+0000e4c0: 7863 6570 7420 666f 7220 7175 6572 7969  xcept for queryi
+0000e4d0: 6e67 2074 6865 2073 6571 7565 6e63 6520  ng the sequence 
+0000e4e0: 6974 7365 6c66 2c20 7768 6963 6820 6361  itself, which ca
+0000e4f0: 6e20 6265 2069 6e69 7469 6174 6564 2062  n be initiated b
+0000e500: 7920 6164 6420 7468 6520 666c 6167 2060  y add the flag `
+0000e510: 2d67 6d60 206f 7220 602d 2d67 6c6f 6261  -gm` or `--globa
+0000e520: 6c6d 7262 6860 2e20 5468 6520 7265 7374  lmrbh`. The rest
+0000e530: 206f 6620 7265 6c65 7661 6e74 2070 6172   of relevant par
+0000e540: 616d 6574 6572 7320 7374 6179 7320 7468  ameters stays th
+0000e550: 6520 7361 6d65 2061 7320 7468 6520 6c6f  e same as the lo
+0000e560: 6361 6c20 4d52 4248 732e 2054 6865 206e  cal MRBHs. The n
+0000e570: 756d 6265 7220 6f66 2070 6172 616c 6c65  umber of paralle
+0000e580: 6c20 7468 7265 6164 7320 6973 2073 7567  l threads is sug
+0000e590: 6765 7374 6564 2074 6f20 6265 2073 6574  gested to be set
+0000e5a0: 2061 7320 284e 2d31 294e 2f32 2074 6f6f   as (N-1)N/2 too
+0000e5b0: 2077 6865 7265 204e 2069 7320 7468 6520   where N is the 
+0000e5c0: 6e75 6d62 6572 206f 6620 6364 7320 6669  number of cds fi
+0000e5d0: 6c65 7320 746f 2061 6368 6965 7665 2074  les to achieve t
+0000e5e0: 6865 2068 6967 6865 7374 2065 6666 6963  he highest effic
+0000e5f0: 6965 6e63 792e 0a0a 2a2a 4120 7375 6767  iency...**A sugg
+0000e600: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+0000e610: 756e 2069 7320 756e 6465 7220 7468 6520  un is under the 
+0000e620: 6465 6661 756c 7420 7061 7261 6d65 7465  default paramete
+0000e630: 7220 7769 7468 2074 6865 2063 6f6d 6d61  r with the comma
+0000e640: 6e64 2073 686f 776e 2061 7320 6265 6c6f  nd shown as belo
+0000e650: 772e 2a2a 0a0a 6060 600a 7767 6420 646d  w.**..```.wgd dm
+0000e660: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
+0000e670: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+0000e680: 2d67 6d0a 6060 600a 0a2a 2a54 6865 2064  -gm.```..**The d
+0000e690: 656c 696e 6561 7469 6f6e 206f 6620 6f72  elineation of or
+0000e6a0: 7468 6f67 726f 7570 732a 2a0a 6060 600a  thogroups**.```.
+0000e6b0: 7767 6420 646d 6420 7365 7175 656e 6365  wgd dmd sequence
+0000e6c0: 3120 7365 7175 656e 6365 3220 7365 7175  1 sequence2 sequ
+0000e6d0: 656e 6365 3320 2d6f 6920 2d6f 6f20 2d65  ence3 -oi -oo -e
+0000e6e0: 2031 652d 3130 202d 6273 2031 3030 202d   1e-10 -bs 100 -
+0000e6f0: 6e70 2035 2028 2d6e 6e29 2028 2d63 6329  np 5 (-nn) (-cc)
+0000e700: 2028 2d74 6529 2028 2d6d 6320 302e 3829   (-te) (-mc 0.8)
+0000e710: 2028 2d67 6e29 2028 2d74 7265 6520 2766   (-gn) (-tree 'f
+0000e720: 6173 7474 7265 6527 2920 282d 7473 2027  asttree') (-ts '
+0000e730: 2d66 6173 7465 7374 2729 2028 2d6e 2034  -fastest') (-n 4
+0000e740: 2920 282d 2d74 6f5f 7374 6f70 2920 282d  ) (--to_stop) (-
+0000e750: 2d63 6473 2920 282d 6f20 7767 645f 646d  -cds) (-o wgd_dm
+0000e760: 6429 2028 2d74 2077 6f72 6b69 6e67 5f74  d) (-t working_t
+0000e770: 6d70 290a 6060 600a 0a49 6e20 6077 6764  mp).```..In `wgd
+0000e780: 2076 3260 2c20 7765 2061 6c73 6f20 696d   v2`, we also im
+0000e790: 706c 656d 656e 7465 6420 616e 2061 6c67  plemented an alg
+0000e7a0: 6f72 6974 686d 206f 6620 6465 6c69 6e65  orithm of deline
+0000e7b0: 6174 696e 6720 6f72 7468 6f67 726f 7570  ating orthogroup
+0000e7c0: 732c 2077 6869 6368 2063 616e 2062 6520  s, which can be 
+0000e7d0: 696e 6974 6961 7465 6420 7769 7468 2074  initiated with t
+0000e7e0: 6865 2070 6172 616d 6574 6572 2060 2d6f  he parameter `-o
+0000e7f0: 6960 206f 7220 602d 2d6f 7274 686f 696e  i` or `--orthoin
+0000e800: 6665 7260 2e20 5477 6f20 7761 7973 206f  fer`. Two ways o
+0000e810: 6620 6465 6c69 6e65 6174 696f 6e20 6361  f delineation ca
+0000e820: 6e20 6265 2063 686f 7365 6e2c 2074 6865  n be chosen, the
+0000e830: 2063 6f6e 6361 7465 6e61 7469 6f6e 2077   concatenation w
+0000e840: 6179 2028 7365 7420 6279 2074 6865 2070  ay (set by the p
+0000e850: 6172 616d 6574 6572 2060 2d63 6360 206f  arameter `-cc` o
+0000e860: 7220 602d 2d63 6f6e 6361 7460 2920 6f72  r `--concat`) or
+0000e870: 2074 6865 206e 6f6e 2d63 6f6e 6361 7465   the non-concate
+0000e880: 6e61 7469 6f6e 2028 6465 6661 756c 7429  nation (default)
+0000e890: 2077 6179 2e20 496e 2062 7269 6566 2c20   way. In brief, 
+0000e8a0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+0000e8b0: 6e20 7761 7920 6f66 2064 656c 696e 6561  n way of delinea
+0000e8c0: 7469 6e67 206f 7274 686f 6772 6f75 7073  ting orthogroups
+0000e8d0: 2073 7461 7274 7320 7769 7468 2063 6f6e   starts with con
+0000e8e0: 6361 7465 6e61 7469 6e67 2061 6c6c 2074  catenating all t
+0000e8f0: 6865 2073 6571 7565 6e63 6573 2069 6e74  he sequences int
+0000e900: 6f20 6120 7369 6e67 6c65 2073 6571 7565  o a single seque
+0000e910: 6e63 6520 6669 6c65 2061 6e64 2074 6865  nce file and the
+0000e920: 6e20 696e 6665 7272 696e 6720 7468 6520  n inferring the 
+0000e930: 7768 6f6c 6520 7061 7261 6e6f 6d65 206f  whole paranome o
+0000e940: 6620 7468 6973 2073 696e 676c 6520 7365  f this single se
+0000e950: 7175 656e 6365 2066 696c 6520 7769 7468  quence file with
+0000e960: 2074 6865 2063 6c75 7374 6572 696e 6720   the clustering 
+0000e970: 7265 7375 6c74 7320 6d61 7070 6564 2062  results mapped b
+0000e980: 6163 6b20 746f 2074 6865 2062 656c 6f6e  ack to the belon
+0000e990: 6769 6e67 2073 7065 6369 6573 2e20 5768  ging species. Wh
+0000e9a0: 696c 6520 7468 6520 6e6f 6e2d 636f 6e63  ile the non-conc
+0000e9b0: 6174 656e 6174 696f 6e20 7761 7920 7374  atenation way st
+0000e9c0: 6172 7473 2077 6974 6820 7265 7370 6563  arts with respec
+0000e9d0: 7469 7665 2070 6169 722d 7769 7365 2064  tive pair-wise d
+0000e9e0: 6961 6d6f 6e64 2073 6561 7263 6820 2869  iamond search (i
+0000e9f0: 6e63 6c75 6469 6e67 2071 7565 7279 696e  ncluding queryin
+0000ea00: 6720 7468 6520 7361 6d65 2073 6571 7565  g the same seque
+0000ea10: 6e63 6520 6974 7365 6c66 2920 616e 6420  nce itself) and 
+0000ea20: 7468 656e 2061 6c6c 2074 6865 2073 6571  then all the seq
+0000ea30: 7565 6e63 6520 7369 6d69 6c61 7269 7479  uence similarity
+0000ea40: 2074 6162 6c65 7320 7769 6c6c 2062 6520   tables will be 
+0000ea50: 636f 6e63 6174 656e 6174 6564 2061 6e64  concatenated and
+0000ea60: 2063 6c75 7374 6572 6564 2069 6e74 6f20   clustered into 
+0000ea70: 6f72 7468 6f67 726f 7570 732e 2053 6f6d  orthogroups. Som
+0000ea80: 6520 6f74 6865 7220 706f 7373 6962 6c79  e other possibly
+0000ea90: 2075 7365 6675 6c20 706f 7374 2d63 6c75   useful post-clu
+0000eaa0: 7374 6572 696e 6720 6675 6e63 7469 6f6e  stering function
+0000eab0: 7320 6361 6e20 6265 2069 6e69 7469 6174  s can be initiat
+0000eac0: 6564 2c20 696e 636c 7564 696e 6720 7468  ed, including th
+0000ead0: 6520 7061 7261 6d65 7465 7220 602d 7465  e parameter `-te
+0000eae0: 6020 6f72 2060 2d2d 7465 7374 736f 6760  ` or `--testsog`
+0000eaf0: 2c20 7768 6963 6820 6361 6e20 6265 2073  , which can be s
+0000eb00: 6574 2074 6f20 7374 6172 7420 7468 6520  et to start the 
+0000eb10: 756e 6269 6173 6564 2074 6573 7420 6f66  unbiased test of
+0000eb20: 2073 696e 676c 652d 636f 7079 2067 656e   single-copy gen
+0000eb30: 6520 6661 6d69 6c69 6573 2028 6e6f 7465  e families (note
+0000eb40: 2074 6861 7420 7468 6973 2066 756e 6374   that this funct
+0000eb50: 696f 6e20 6e65 6564 7320 6068 6d6d 6572  ion needs `hmmer
+0000eb60: 6020 2876 332e 3162 3229 2074 6f20 6265  ` (v3.1b2) to be
+0000eb70: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
+0000eb80: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
+0000eb90: 7468 292c 2074 6865 2070 6172 616d 6574  th), the paramet
+0000eba0: 6572 2060 2d6d 6360 206f 7220 602d 2d6d  er `-mc` or `--m
+0000ebb0: 736f 6763 7574 602c 2072 616e 6769 6e67  sogcut`, ranging
+0000ebc0: 2062 6574 7765 656e 2030 2074 6f20 312c   between 0 to 1,
+0000ebd0: 2077 6869 6368 2063 616e 2062 6520 7365   which can be se
+0000ebe0: 7420 746f 2073 6561 7263 6820 7468 6520  t to search the 
+0000ebf0: 736f 2d63 616c 6c65 6420 6d6f 7374 6c79  so-called mostly
+0000ec00: 2073 696e 676c 652d 636f 7079 2066 616d   single-copy fam
+0000ec10: 696c 7920 7768 6963 6820 6861 7320 6869  ily which has hi
+0000ec20: 6768 6572 2074 6861 6e20 6365 7274 6169  gher than certai
+0000ec30: 6e20 6375 742d 6f66 6620 7065 7263 656e  n cut-off percen
+0000ec40: 7461 6765 206f 6620 7370 6563 6965 7320  tage of species 
+0000ec50: 636f 7665 7261 6765 2c20 7468 6520 7061  coverage, the pa
+0000ec60: 7261 6d65 7465 7220 602d 676e 6020 6f72  rameter `-gn` or
+0000ec70: 2060 2d2d 6765 746e 736f 6760 2c20 7768   `--getnsog`, wh
+0000ec80: 6963 6820 6361 6e20 6265 2073 6574 2074  ich can be set t
+0000ec90: 6f20 7365 6172 6368 2066 6f72 206e 6573  o search for nes
+0000eca0: 7465 6420 7369 6e67 6c65 2d63 6f70 7920  ted single-copy 
+0000ecb0: 6765 6e65 2066 616d 696c 6965 7320 284e  gene families (N
+0000ecc0: 534f 4773 2920 7768 6963 6820 6973 206f  SOGs) which is o
+0000ecd0: 7269 6769 6e61 6c6c 7920 6d75 6c74 6979  riginally multiy
+0000ece0: 2d63 6f70 7920 6275 7420 6861 7320 6120  -copy but has a 
+0000ecf0: 286d 6f73 746c 7929 2073 696e 676c 652d  (mostly) single-
+0000ed00: 636f 7079 2062 7261 6e63 6820 2877 6869  copy branch (whi
+0000ed10: 6368 2072 6571 7569 7265 7320 7468 6520  ch requires the 
+0000ed20: 6368 6f73 656e 2074 7265 652d 696e 6665  chosen tree-infe
+0000ed30: 7265 6e63 6520 7072 6f67 7261 6d20 7365  rence program se
+0000ed40: 7420 6279 2060 2d74 7265 6560 206f 7220  t by `-tree` or 
+0000ed50: 602d 2d74 7265 655f 6d65 7468 6f64 6020  `--tree_method` 
+0000ed60: 746f 2062 6520 7072 652d 696e 7374 616c  to be pre-instal
+0000ed70: 6c65 6420 696e 2074 6865 2065 6e76 6972  led in the envir
+0000ed80: 6f6e 6d65 6e74 2070 6174 6820 7769 7468  onment path with
+0000ed90: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
+0000eda0: 7365 7474 696e 6720 666f 7220 6765 6e65  setting for gene
+0000edb0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0000edc0: 636f 6e74 726f 6c6c 6564 2062 7920 602d  controlled by `-
+0000edd0: 7473 6020 6f72 2060 2d2d 7472 6565 7365  ts` or `--treese
+0000ede0: 7460 292e 2054 6865 2070 726f 6772 616d  t`). The program
+0000edf0: 2060 7767 6420 646d 6460 2077 6f75 6c64   `wgd dmd` would
+0000ee00: 2073 7469 6c6c 2063 6f6e 6475 6374 2074   still conduct t
+0000ee10: 6865 2052 4248 7320 6361 6c63 756c 6174  he RBHs calculat
+0000ee20: 696f 6e20 756e 6c65 7373 2074 6865 2070  ion unless the p
+0000ee30: 6172 616d 6574 6572 2060 2d6f 6f60 206f  arameter `-oo` o
+0000ee40: 7220 602d 2d6f 6e6c 796f 7274 686f 6020  r `--onlyortho` 
+0000ee50: 7761 7320 7365 742e 2049 6620 6f6e 6520  was set. If one 
+0000ee60: 6f6e 6c79 2077 616e 7473 2074 6f20 696e  only wants to in
+0000ee70: 6665 7220 7468 6520 6f72 7468 6f67 726f  fer the orthogro
+0000ee80: 7570 732c 2069 7427 7320 7375 6767 6573  ups, it's sugges
+0000ee90: 7465 6420 746f 2061 6464 2074 6865 2066  ted to add the f
+0000eea0: 6c61 6720 602d 6f6f 6020 746f 206a 7573  lag `-oo` to jus
+0000eeb0: 7420 696d 706c 656d 656e 7420 7468 6520  t implement the 
+0000eec0: 6f72 7468 6f67 726f 7570 7320 6465 6c69  orthogroups deli
+0000eed0: 6e65 6174 696f 6e20 616e 616c 7973 6973  neation analysis
+0000eee0: 2e20 5468 6520 6e75 6d62 6572 206f 6620  . The number of 
+0000eef0: 7061 7261 6c6c 656c 2074 6872 6561 6473  parallel threads
+0000ef00: 2069 7320 7375 6767 6573 7465 6420 746f   is suggested to
+0000ef10: 2062 6520 7365 7420 6173 2028 4e2b 3129   be set as (N+1)
+0000ef20: 4e2f 3220 7768 6572 6520 4e20 6973 2074  N/2 where N is t
+0000ef30: 6865 206e 756d 6265 7220 6f66 2063 6473  he number of cds
+0000ef40: 2066 696c 6573 2074 6f20 6163 6869 6576   files to achiev
+0000ef50: 6520 7468 6520 6869 6768 6573 7420 6566  e the highest ef
+0000ef60: 6669 6369 656e 6379 2073 696e 6365 2074  ficiency since t
+0000ef70: 6865 2073 656c 662d 636f 6d70 6172 6973  he self-comparis
+0000ef80: 6f6e 2069 7320 616c 736f 2069 6e63 6c75  on is also inclu
+0000ef90: 6465 642e 0a0a 2a2a 5468 6520 6465 6661  ded...**The defa
+0000efa0: 756c 7420 7365 7474 696e 6720 6f66 2070  ult setting of p
+0000efb0: 6172 616d 6574 6572 7320 6973 2061 2072  arameters is a r
+0000efc0: 6561 736f 6e61 626c 6520 7374 6172 7469  easonable starti
+0000efd0: 6e67 2070 6f69 6e74 2077 6974 6820 7468  ng point with th
+0000efe0: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
+0000eff0: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2064  ow.**..```.wgd d
+0000f000: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
+0000f010: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+0000f020: 202d 6f69 202d 6f6f 0a60 6060 0a0a 2a2a   -oi -oo.```..**
+0000f030: 5468 6520 636f 6c6c 696e 6561 7220 636f  The collinear co
+0000f040: 616c 6573 6365 6e63 6520 696e 6665 7265  alescence infere
+0000f050: 6e63 6520 6f66 2070 6879 6c6f 6765 6e79  nce of phylogeny
+0000f060: 2a2a 0a60 6060 0a77 6764 2064 6d64 2073  **.```.wgd dmd s
+0000f070: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
+0000f080: 6532 2073 6571 7565 6e63 6533 202d 6170  e2 sequence3 -ap
+0000f090: 2061 7064 6174 6120 2d73 6d20 736d 6461   apdata -sm smda
+0000f0a0: 7461 202d 6c65 206c 6564 6174 6120 2d67  ta -le ledata -g
+0000f0b0: 7420 6774 6461 7461 202d 636f 6320 282d  t gtdata -coc (-
+0000f0c0: 7472 6565 2027 6661 7374 7472 6565 2729  tree 'fasttree')
+0000f0d0: 2028 2d74 7320 272d 6661 7374 6573 7427   (-ts '-fastest'
+0000f0e0: 2920 282d 6e20 3429 2028 2d2d 746f 5f73  ) (-n 4) (--to_s
+0000f0f0: 746f 7029 2028 2d2d 6364 7329 2028 2d6f  top) (--cds) (-o
+0000f100: 2077 6764 5f64 6d64 2920 282d 7420 776f   wgd_dmd) (-t wo
+0000f110: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
+0000f120: 4120 6e6f 7665 6c20 7068 796c 6f67 656e  A novel phylogen
+0000f130: 6574 6963 2069 6e66 6572 656e 6365 206d  etic inference m
+0000f140: 6574 686f 6420 6e61 6d65 6420 2263 6f6c  ethod named "col
+0000f150: 6c69 6e65 6172 2063 6f61 6c65 7363 656e  linear coalescen
+0000f160: 6365 2069 6e66 6572 656e 6365 2220 6973  ce inference" is
+0000f170: 2061 6c73 6f20 696d 706c 656d 656e 7465   also implemente
+0000f180: 6420 696e 2060 7767 6420 7632 602e 2046  d in `wgd v2`. F
+0000f190: 6f72 2074 6869 7320 616e 616c 7973 6973  or this analysis
+0000f1a0: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
+0000f1b0: 7072 6f76 6964 6520 7468 6520 616e 6368  provide the anch
+0000f1c0: 6f72 2070 6f69 6e74 7320 6669 6c65 2062  or points file b
+0000f1d0: 7920 602d 6170 6020 6f72 2060 2d2d 616e  y `-ap` or `--an
+0000f1e0: 6368 6f72 706f 696e 7473 602c 2074 6865  chorpoints`, the
+0000f1f0: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
+0000f200: 6e74 7320 6669 6c65 2062 7920 602d 736d  nts file by `-sm
+0000f210: 6020 6f72 2060 2d2d 7365 676d 656e 7473  ` or `--segments
+0000f220: 602c 2074 6865 206c 6973 7473 6567 6d65  `, the listsegme
+0000f230: 6e74 7320 6669 6c65 2062 7920 602d 6c65  nts file by `-le
+0000f240: 6020 6f72 2060 2d2d 6c69 7374 656c 656d  ` or `--listelem
+0000f250: 656e 7473 602c 2061 6e64 2074 6865 2067  ents`, and the g
+0000f260: 656e 6520 7461 626c 6520 6669 6c65 2062  ene table file b
+0000f270: 7920 602d 6774 6020 6f72 2060 2d2d 6765  y `-gt` or `--ge
+0000f280: 6e65 7461 626c 6560 2c20 616c 6c20 6f66  netable`, all of
+0000f290: 2077 6869 6368 2063 616e 2062 6520 7072   which can be pr
+0000f2a0: 6f64 7563 6564 2069 6e20 7468 6520 7072  oduced in the pr
+0000f2b0: 6f67 7261 6d20 6077 6764 2073 796e 602e  ogram `wgd syn`.
+0000f2c0: 2054 6865 2070 6172 616d 6574 6572 2060   The parameter `
+0000f2d0: 2d63 6f63 6020 6f72 2060 2d2d 636f 6c6c  -coc` or `--coll
+0000f2e0: 696e 6561 7263 6f61 6c65 7363 656e 6365  inearcoalescence
+0000f2f0: 6020 6e65 6564 7320 746f 2062 6520 7365  ` needs to be se
+0000f300: 7420 746f 2073 7461 7274 2074 6869 7320  t to start this 
+0000f310: 616e 616c 7973 6973 2e20 5468 6520 7472  analysis. The tr
+0000f320: 6565 2d69 6e66 6572 656e 6365 2070 726f  ee-inference pro
+0000f330: 6772 616d 2061 6e64 2074 6865 2061 7373  gram and the ass
+0000f340: 6f63 6961 7465 6420 7061 7261 6d65 7465  ociated paramete
+0000f350: 7273 2063 616e 2062 6520 7365 7420 6a75  rs can be set ju
+0000f360: 7374 2061 7320 6162 6f76 6520 6279 2060  st as above by `
+0000f370: 2d74 7265 6560 206f 7220 602d 2d74 7265  -tree` or `--tre
+0000f380: 655f 6d65 7468 6f64 6020 616e 6420 602d  e_method` and `-
+0000f390: 7473 6020 6f72 2060 2d2d 7472 6565 7365  ts` or `--treese
+0000f3a0: 7460 2e20 506c 6561 7365 2061 6c73 6f20  t`. Please also 
+0000f3b0: 6d61 6b65 2073 7572 6520 7468 6520 6368  make sure the ch
+0000f3c0: 6f73 656e 2074 7265 652d 696e 6665 7265  osen tree-infere
+0000f3d0: 6e63 6520 7072 6f67 7261 6d20 6973 2069  nce program is i
+0000f3e0: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+0000f3f0: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+0000f400: 2e20 5468 6520 7072 6f67 7261 6d20 6061  . The program `a
+0000f410: 7374 7261 6c2d 7072 6f60 2069 7320 7265  stral-pro` is re
+0000f420: 7175 6972 6564 2074 6f20 6265 2069 6e73  quired to be ins
+0000f430: 7461 6c6c 6564 2069 6e20 7468 6520 656e  talled in the en
+0000f440: 7669 726f 6e6d 656e 7420 7061 7468 2074  vironment path t
+0000f450: 6f6f 2e20 4e6f 7465 2074 6861 7420 7468  oo. Note that th
+0000f460: 6572 6520 7368 6f75 6c64 2062 6520 6e6f  ere should be no
+0000f470: 2064 7570 6c69 6361 7465 6420 6765 6e65   duplicated gene
+0000f480: 2049 4473 2069 6e20 7468 6520 7365 7175   IDs in the sequ
+0000f490: 656e 6365 2066 696c 652e 2054 6865 2070  ence file. The p
+0000f4a0: 6172 616c 6c65 6c20 7468 7265 6164 7320  arallel threads 
+0000f4b0: 6865 7265 2061 7265 2074 6f20 6163 6365  here are to acce
+0000f4c0: 6c65 7261 7465 2074 6865 2073 6571 7565  lerate the seque
+0000f4d0: 6e63 6520 616c 6967 6e6d 656e 7420 616e  nce alignment an
+0000f4e0: 6420 6765 6e65 2074 7265 6520 696e 6665  d gene tree infe
+0000f4f0: 7265 6e63 6520 666f 7220 6561 6368 2067  rence for each g
+0000f500: 656e 6520 6661 6d69 6c79 2061 6e64 2074  ene family and t
+0000f510: 6875 7320 7375 6767 6573 7465 6420 746f  hus suggested to
+0000f520: 2062 6520 7365 7420 6173 206d 7563 6820   be set as much 
+0000f530: 6173 2074 6865 206e 756d 6265 7220 6f66  as the number of
+0000f540: 2067 656e 6520 6661 6d69 6c69 6573 2e0a   gene families..
+0000f550: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+0000f560: 7461 7274 696e 6720 7275 6e20 6f66 2074  tarting run of t
+0000f570: 6869 7320 616e 616c 7973 6973 2069 7320  his analysis is 
+0000f580: 7769 7468 2074 6865 2073 696d 706c 6520  with the simple 
+0000f590: 636f 6d6d 616e 6420 6265 6c6f 772e 2a2a  command below.**
+0000f5a0: 0a0a 6060 600a 7767 6420 646d 6420 7365  ..```.wgd dmd se
+0000f5b0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+0000f5c0: 3220 7365 7175 656e 6365 3320 2d61 7020  2 sequence3 -ap 
+0000f5d0: 6170 6461 7461 202d 736d 2073 6d64 6174  apdata -sm smdat
+0000f5e0: 6120 2d6c 6520 6c65 6461 7461 202d 6774  a -le ledata -gt
+0000f5f0: 2067 7464 6174 6120 2d63 6f63 0a60 6060   gtdata -coc.```
+0000f600: 0a0a 2323 2320 7767 6420 6b73 640a 0a2a  ..### wgd ksd..*
+0000f610: 2a54 6865 2063 6f6e 7374 7275 6374 696f  *The constructio
+0000f620: 6e20 6f66 2077 686f 6c65 2070 6172 616e  n of whole paran
+0000f630: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
+0000f640: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+0000f650: 7469 6f6e 2a2a 0a60 6060 0a77 6764 206b  tion**.```.wgd k
+0000f660: 7364 2066 616d 696c 6965 7320 7365 7175  sd families sequ
+0000f670: 656e 6365 2028 2d6f 2077 6764 5f6b 7364  ence (-o wgd_ksd
+0000f680: 202d 7420 7767 645f 6b73 645f 746d 7020   -t wgd_ksd_tmp 
+0000f690: 2d2d 6e74 6872 6561 6473 2034 202d 2d74  --nthreads 4 --t
+0000f6a0: 6f5f 7374 6f70 202d 2d63 6473 202d 2d70  o_stop --cds --p
+0000f6b0: 6169 7277 6973 6520 2d2d 7374 7269 705f  airwise --strip_
+0000f6c0: 6761 7073 202d 2d61 6c69 676e 6572 206d  gaps --aligner m
+0000f6d0: 6166 6674 202d 2d74 7265 655f 6d65 7468  afft --tree_meth
+0000f6e0: 6f64 2066 6173 7474 7265 6520 2d2d 6e6f  od fasttree --no
+0000f6f0: 6465 5f61 7665 7261 6765 290a 6060 600a  de_average).```.
+0000f700: 0a54 6865 2070 726f 6772 616d 2060 7767  .The program `wg
+0000f710: 6420 6b73 6460 2c20 6173 2069 6d70 6965  d ksd`, as impie
+0000f720: 6420 6279 2069 7473 206e 616d 652c 2069  d by its name, i
+0000f730: 7320 666f 7220 7468 6520 636f 6e73 7472  s for the constr
+0000f740: 7563 7469 6f6e 206f 6620 2a4b 2a3c 7375  uction of *K*<su
+0000f750: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+0000f760: 7374 7269 6275 7469 6f6e 2e20 4578 6365  stribution. Exce
+0000f770: 7074 2066 6f72 2074 6865 2061 666f 7265  pt for the afore
+0000f780: 6d65 6e74 696f 6e65 6420 7061 7261 6d65  mentioned parame
+0000f790: 7465 7273 2073 7563 6820 6173 2060 2d2d  ters such as `--
+0000f7a0: 746f 5f73 746f 7060 2061 6e64 2060 2d2d  to_stop` and `--
+0000f7b0: 6364 7360 2c20 7468 6572 6520 6172 6520  cds`, there are 
+0000f7c0: 736f 6d65 2069 6d70 6f72 7461 6e74 2070  some important p
+0000f7d0: 6172 616d 6574 6572 7320 7468 6174 2068  arameters that h
+0000f7e0: 6176 6520 6372 7563 6961 6c20 696d 7061  ave crucial impa
+0000f7f0: 6374 206f 6e20 7468 6520 2a4b 2a3c 7375  ct on the *K*<su
+0000f800: 623e 533c 2f73 7562 3e20 6573 7469 6d61  b>S</sub> estima
+0000f810: 7469 6f6e 2e20 5468 6520 6f70 7469 6f6e  tion. The option
+0000f820: 2060 2d2d 7061 6972 7769 7365 6020 6973   `--pairwise` is
+0000f830: 2061 2076 6572 7920 696d 706f 7274 616e   a very importan
+0000f840: 7420 7061 7261 6d65 7465 7220 666f 7220  t parameter for 
+0000f850: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+0000f860: 7562 3e20 6573 7469 6d61 7469 6f6e 2c20  ub> estimation, 
+0000f870: 7769 7468 2077 6869 6368 2074 6865 2060  with which the `
+0000f880: 434f 4445 4d4c 6020 7769 6c6c 2063 616c  CODEML` will cal
+0000f890: 6375 6c61 7465 2074 6865 202a 4b2a 3c73  culate the *K*<s
+0000f8a0: 7562 3e53 3c2f 7375 623e 2066 6f72 2065  ub>S</sub> for e
+0000f8b0: 6163 6820 6765 6e65 2070 6169 7220 7365  ach gene pair se
+0000f8c0: 7061 7261 7465 6c79 2062 6173 6564 206f  parately based o
+0000f8d0: 6e20 7468 6520 616c 6967 6e6d 656e 7420  n the alignment 
+0000f8e0: 6f66 206f 6e6c 7920 7468 6573 6520 7477  of only these tw
+0000f8f0: 6f20 6765 6e65 7320 696e 7374 6561 6420  o genes instead 
+0000f900: 6f66 2074 6865 2077 686f 6c65 2061 6c69  of the whole ali
+0000f910: 676e 6d65 6e74 206f 6620 7468 6520 6661  gnment of the fa
+0000f920: 6d69 6c79 2c20 7375 6368 2074 6861 7420  mily, such that 
+0000f930: 6c65 7373 2067 6170 7320 6172 6520 6578  less gaps are ex
+0000f940: 7065 6374 6564 2061 6e64 2074 6875 7320  pected and thus 
+0000f950: 7468 6520 616c 6967 6e6d 656e 7420 696e  the alignment in
+0000f960: 2074 6865 2063 6f6e 7369 6465 7261 7469   the considerati
+0000f970: 6f6e 206f 6620 6043 4f44 454d 4c60 2077  on of `CODEML` w
+0000f980: 696c 6c20 6265 206c 6f6e 6765 7220 2862  ill be longer (b
+0000f990: 6563 6175 7365 2060 434f 4445 4d4c 6020  ecause `CODEML` 
+0000f9a0: 7769 6c6c 2061 7574 6f6d 6174 656c 7920  will automately 
+0000f9b0: 736b 6970 2065 7665 7279 2063 6f6c 756d  skip every colum
+0000f9c0: 6e20 7769 7468 2067 6170 2c20 7265 6761  n with gap, rega
+0000f9d0: 7264 6c65 7373 206f 6620 7768 6574 6865  rdless of whethe
+0000f9e0: 7220 6974 2773 2061 6e20 6f76 6572 616c  r it's an overal
+0000f9f0: 6c20 6f72 2070 6172 7469 616c 2067 6170  l or partial gap
+0000fa00: 292c 2077 6974 686f 7574 2077 6869 6368  ), without which
+0000fa10: 2074 6865 2060 434f 4445 4d4c 6020 7769   the `CODEML` wi
+0000fa20: 6c6c 2063 616c 6375 6c61 7465 2074 6865  ll calculate the
+0000fa30: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000fa40: 2062 6173 6564 206f 6e20 7468 6520 7768   based on the wh
+0000fa50: 6f6c 6520 616c 6967 6e6d 656e 7420 6f66  ole alignment of
+0000fa60: 2074 6865 2066 616d 696c 792c 2077 6869   the family, whi
+0000fa70: 6368 206d 6967 6874 2068 6176 6520 6e6f  ch might have no
+0000fa80: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000fa90: 2072 6573 756c 7420 6174 2061 6c6c 2069   result at all i
+0000faa0: 6620 7468 6520 7374 7269 7070 6564 2061  f the stripped a
+0000fab0: 6c69 676e 6d65 6e74 206c 656e 6774 6820  lignment length 
+0000fac0: 2872 656d 6f76 696e 6720 616c 6c20 6761  (removing all ga
+0000fad0: 702d 636f 6e74 6169 6e69 6e67 2063 6f6c  p-containing col
+0000fae0: 756d 6e73 2920 7761 7320 7a65 726f 2c20  umns) was zero, 
+0000faf0: 6120 6361 7573 6520 6f66 2064 6966 6665  a cause of diffe
+0000fb00: 7265 6e74 206e 756d 6265 7220 6f66 202a  rent number of *
+0000fb10: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
+0000fb20: 7374 696d 6174 6573 2062 6574 7765 656e  stimates between
+0000fb30: 2022 7061 6972 7769 7365 206d 6f64 6522   "pairwise mode"
+0000fb40: 2061 6e64 2022 6e6f 6e2d 7061 6972 7769   and "non-pairwi
+0000fb50: 7365 206d 6f64 6522 2e20 4974 2773 2064  se mode". It's d
+0000fb60: 6966 6669 6375 6c74 2074 6f20 7361 7920  ifficult to say 
+0000fb70: 7768 6963 6820 6d6f 6465 2069 7320 6d6f  which mode is mo
+0000fb80: 7265 2069 6465 616c 2c20 616c 7468 6f75  re ideal, althou
+0000fb90: 6768 2074 6865 2022 6e6f 6e2d 7061 6972  gh the "non-pair
+0000fba0: 7769 7365 206d 6f64 6522 2028 6465 6661  wise mode" (defa
+0000fbb0: 756c 7420 7365 7474 696e 6729 2077 6869  ult setting) whi
+0000fbc0: 6368 2072 756e 7320 6f6e 2074 6865 2077  ch runs on the w
+0000fbd0: 686f 6c65 2061 6c69 676e 6d65 6e74 2069  hole alignment i
+0000fbe0: 6e73 7465 6164 206f 6620 6120 6c6f 6361  nstead of a loca
+0000fbf0: 6c20 616c 6967 6e6d 656e 742c 206d 6967  l alignment, mig
+0000fc00: 6874 2062 6520 6d6f 7265 2062 696f 6c6f  ht be more biolo
+0000fc10: 6769 6361 6c6c 7920 636f 6e73 6572 7665  gically conserve
+0000fc20: 6420 696e 2074 6861 7420 6974 2061 7373  d in that it ass
+0000fc30: 7572 6573 2074 6865 2065 766f 6c75 7469  ures the evoluti
+0000fc40: 6f6e 206f 6620 6561 6368 2063 6f6c 756d  on of each colum
+0000fc50: 6e20 746f 2062 6520 7374 6172 7465 6420  n to be started 
+0000fc60: 6672 6f6d 2074 6865 2072 6f6f 7420 6f66  from the root of
+0000fc70: 2074 6865 2066 616d 696c 7920 616e 6420   the family and 
+0000fc80: 616c 6c20 7468 6520 6765 6e65 2064 7570  all the gene dup
+0000fc90: 6c69 6361 7465 7320 6172 6520 7461 6b65  licates are take
+0000fca0: 6e20 696e 746f 2061 6363 6f75 6e74 2069  n into account i
+0000fcb0: 6e20 7468 6520 2a4b 2a3c 7375 623e 533c  n the *K*<sub>S<
+0000fcc0: 2f73 7562 3e20 6573 7469 6d61 7469 6f6e  /sub> estimation
+0000fcd0: 2070 726f 6365 7373 2e20 5468 6520 6f70   process. The op
+0000fce0: 7469 6f6e 2060 2d2d 7374 7269 705f 6761  tion `--strip_ga
+0000fcf0: 7073 6020 6361 6e20 7265 6d6f 7665 2061  ps` can remove a
+0000fd00: 6c6c 2074 6865 2067 6170 2d63 6f6e 7461  ll the gap-conta
+0000fd10: 696e 696e 6720 636f 6c75 6d6e 732c 2077  ining columns, w
+0000fd20: 6974 6820 6f72 2077 6974 686f 7574 2077  ith or without w
+0000fd30: 6869 6368 2074 6865 2072 6573 756c 7420  hich the result 
+0000fd40: 6f66 2022 6e6f 6e2d 7061 6972 7769 7365  of "non-pairwise
+0000fd50: 206d 6f64 6522 2077 6f6e 2774 2062 6520   mode" won't be 
+0000fd60: 6166 6665 6374 6564 2c20 7768 696c 6520  affected, while 
+0000fd70: 7769 7468 2077 6869 6368 2074 6865 2072  with which the r
+0000fd80: 6573 756c 7420 6f66 2022 7061 6972 7769  esult of "pairwi
+0000fd90: 7365 206d 6f64 6522 2077 696c 6c20 6265  se mode" will be
+0000fda0: 2061 6c74 6572 6564 2e20 5468 6520 6f70   altered. The op
+0000fdb0: 7469 6f6e 2060 2d2d 616c 6967 6e65 7260  tion `--aligner`
+0000fdc0: 2061 6e64 2060 2d2d 616c 6e5f 6f70 7469   and `--aln_opti
+0000fdd0: 6f6e 7360 2077 6869 6368 2064 6563 6964  ons` which decid
+0000fde0: 6520 7768 6963 6820 616c 6967 6e6d 656e  e which alignmen
+0000fdf0: 7420 7072 6f67 7261 6d20 746f 2062 6520  t program to be 
+0000fe00: 7573 6564 2061 6e64 2077 6869 6368 2070  used and which p
+0000fe10: 6172 616d 6574 6572 2074 6f20 6265 2073  arameter to be s
+0000fe20: 6574 2c20 7769 6c6c 2068 6176 6520 696d  et, will have im
+0000fe30: 7061 6374 206f 6e20 7468 6520 2a4b 2a3c  pact on the *K*<
+0000fe40: 7375 623e 533c 2f73 7562 3e20 7265 7375  sub>S</sub> resu
+0000fe50: 6c74 732c 206e 6f74 6564 2074 6861 7420  lts, noted that 
+0000fe60: 7468 6520 6465 6661 756c 7420 7072 6f67  the default prog
+0000fe70: 7261 6d20 6973 2060 6d61 6666 7460 2061  ram is `mafft` a
+0000fe80: 6e64 2074 6865 2070 6172 616d 6574 6572  nd the parameter
+0000fe90: 2069 7320 602d 2d61 7574 6f60 2e20 5468   is `--auto`. Th
+0000fea0: 6520 6f70 7469 6f6e 2060 2d2d 7472 6565  e option `--tree
+0000feb0: 5f6d 6574 686f 6460 2061 6e64 2060 2d2d  _method` and `--
+0000fec0: 7472 6565 5f6f 7074 696f 6e73 6020 6465  tree_options` de
+0000fed0: 6369 6465 2077 6869 6368 2067 656e 6520  cide which gene 
+0000fee0: 7472 6565 2069 6e66 6572 656e 6365 2070  tree inference p
+0000fef0: 726f 6772 616d 2074 6f20 6265 2075 7365  rogram to be use
+0000ff00: 6420 616e 6420 7768 6963 6820 7061 7261  d and which para
+0000ff10: 6d65 7465 7220 746f 2062 6520 7365 742c  meter to be set,
+0000ff20: 2077 6f6e 2774 2061 6666 6563 7420 7468   won't affect th
+0000ff30: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+0000ff40: 3e20 6573 7469 6d61 7469 6f6e 2069 7473  > estimation its
+0000ff50: 656c 6620 6275 7420 7468 6520 7265 7375  elf but the resu
+0000ff60: 6c74 206f 6620 6465 2d72 6564 756e 6461  lt of de-redunda
+0000ff70: 6e63 792c 206e 6f74 6564 2074 6861 7420  ncy, noted that 
+0000ff80: 7765 2069 6d70 6c65 6d65 6e74 6564 2061  we implemented a
+0000ff90: 2062 7569 6c74 2d69 6e20 6765 6e65 2074   built-in gene t
+0000ffa0: 7265 6520 696e 6665 7265 6e63 6520 6d65  ree inference me
+0000ffb0: 7468 6f64 2062 6173 6564 206f 6e20 7468  thod based on th
+0000ffc0: 6520 4176 6572 6167 6520 4c69 6e6b 6167  e Average Linkag
+0000ffd0: 6520 436c 7573 7465 7269 6e67 2028 414c  e Clustering (AL
+0000ffe0: 4329 2028 7468 7573 2061 2064 6973 7461  C) (thus a dista
+0000fff0: 6e63 652d 6261 7365 6420 7472 6565 2920  nce-based tree) 
+00010000: 7769 7468 2074 6865 2060 2d2d 7472 6565  with the `--tree
+00010010: 5f6d 6574 686f 6460 2073 6574 2061 7320  _method` set as 
+00010020: 2263 6c75 7374 6572 222e 2054 776f 206d  "cluster". Two m
+00010030: 6574 686f 6473 206f 6620 6465 2d72 6564  ethods of de-red
+00010040: 756e 6461 6e63 7920 6172 6520 696d 706c  undancy are impl
+00010050: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
+00010060: 7632 602c 206e 616d 656c 7920 6e6f 6465  v2`, namely node
+00010070: 2d77 6569 6768 7465 6420 616e 6420 6e6f  -weighted and no
+00010080: 6465 2d61 7665 7261 6765 6420 6d65 7468  de-averaged meth
+00010090: 6f64 732e 2054 6865 206e 6f64 652d 7765  ods. The node-we
+000100a0: 6967 6874 6564 206d 6574 686f 6420 6163  ighted method ac
+000100b0: 6869 6576 6573 2074 6865 2064 652d 7265  hieves the de-re
+000100c0: 6475 6e64 616e 6379 2076 6961 2077 6569  dundancy via wei
+000100d0: 6768 696e 6720 7468 6520 2a4b 2a3c 7375  ghing the *K*<su
+000100e0: 623e 533c 2f73 7562 3e20 7661 6c75 6520  b>S</sub> value 
+000100f0: 6173 736f 6369 6174 6564 2077 6974 6820  associated with 
+00010100: 6561 6368 2067 656e 6520 7061 6972 2073  each gene pair s
+00010110: 7563 6820 7468 6174 2074 6865 2077 6569  uch that the wei
+00010120: 6768 7420 6f66 2061 2073 696e 676c 6520  ght of a single 
+00010130: 6475 706c 6963 6174 696f 6e20 6576 656e  duplication even
+00010140: 7420 7375 6d73 2075 7020 746f 2031 2028  t sums up to 1 (
+00010150: 6e6f 7465 6420 7468 6174 2074 6865 206e  noted that the n
+00010160: 756d 6265 7220 6f66 202a 4b2a 3c73 7562  umber of *K*<sub
+00010170: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
+00010180: 696f 6e20 7265 6d61 696e 7320 7468 6520  ion remains the 
+00010190: 7361 6d65 2920 7768 696c 6520 7468 6520  same) while the 
+000101a0: 6e6f 6465 2d61 7665 7261 6765 6420 6d65  node-averaged me
+000101b0: 7468 6f64 2072 6561 6c69 7a65 7320 7468  thod realizes th
+000101c0: 6520 6465 2d72 6564 756e 6461 6e63 7920  e de-redundancy 
+000101d0: 7669 6120 6361 6c63 756c 6174 696e 6720  via calculating 
+000101e0: 7065 7220 6765 6e65 2074 7265 6520 6e6f  per gene tree no
+000101f0: 6465 206f 6e65 2061 7665 7261 6765 6420  de one averaged 
+00010200: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00010210: 7661 6c75 6520 746f 2072 6570 7265 7365  value to represe
+00010220: 6e74 2074 6865 2061 6765 206f 6620 6561  nt the age of ea
+00010230: 6368 2064 7570 6c69 6361 7469 6f6e 2065  ch duplication e
+00010240: 7665 6e74 2e20 5468 6520 6f70 7469 6f6e  vent. The option
+00010250: 2060 2d2d 6e6f 6465 5f61 7665 7261 6765   `--node_average
+00010260: 6020 6361 6e20 6265 2073 6574 2074 6f20  ` can be set to 
+00010270: 6368 6f6f 7365 2074 6865 206e 6f64 652d  choose the node-
+00010280: 6176 6572 6167 6564 2077 6179 206f 6620  averaged way of 
+00010290: 6465 2d72 6564 756e 6461 6e63 792e 2044  de-redundancy. D
+000102a0: 6966 6665 7265 6e74 206d 6574 686f 6473  ifferent methods
+000102b0: 206f 6620 6465 2d72 6564 756e 6461 6e63   of de-redundanc
+000102c0: 7920 7769 6c6c 2068 6176 6520 696d 7061  y will have impa
+000102d0: 6374 206f 6e20 7468 6520 6465 7465 6374  ct on the detect
+000102e0: 696f 6e20 6f66 2057 4744 2073 6967 6e61  ion of WGD signa
+000102f0: 6c73 2c20 7768 6963 6820 6861 7320 6265  ls, which has be
+00010300: 656e 2069 6e76 6573 7469 6761 7465 6420  en investigated 
+00010310: 696e 2074 6869 7320 5b6c 6974 6572 6174  in this [literat
+00010320: 7572 655d 2868 7474 7073 3a2f 2f64 6f69  ure](https://doi
+00010330: 2e6f 7267 2f31 302e 3130 3933 2f67 6265  .org/10.1093/gbe
+00010340: 2f65 7679 3230 3029 2e20 5468 6520 7061  /evy200). The pa
+00010350: 7261 6c6c 656c 2074 6872 6561 6473 2068  rallel threads h
+00010360: 6572 6520 6172 6520 746f 2070 6172 616c  ere are to paral
+00010370: 6c65 6c69 7a65 2074 6865 2061 6e61 6c79  lelize the analy
+00010380: 7369 7320 666f 7220 6561 6368 2067 656e  sis for each gen
+00010390: 6520 6661 6d69 6c79 2061 6e64 2074 6875  e family and thu
+000103a0: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+000103b0: 6520 7365 7420 6173 206d 7563 6820 6173  e set as much as
+000103c0: 2074 6865 206e 756d 6265 7220 6f66 2067   the number of g
+000103d0: 656e 6520 6661 6d69 6c69 6573 2e0a 0a2a  ene families...*
+000103e0: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
+000103f0: 7274 696e 6720 7275 6e20 6361 6e20 7573  rting run can us
+00010400: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
+00010410: 6f77 2a2a 0a0a 6060 600a 7767 6420 6b73  ow**..```.wgd ks
+00010420: 6420 6661 6d69 6c69 6573 2073 6571 7565  d families seque
+00010430: 6e63 650a 6060 600a 0a2a 2a54 6865 2063  nce.```..**The c
+00010440: 6f6e 7374 7275 6374 696f 6e20 6f66 206f  onstruction of o
+00010450: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
+00010460: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+00010470: 6973 7472 6962 7574 696f 6e2a 2a0a 6060  istribution**.``
+00010480: 600a 7767 6420 6b73 6420 6661 6d69 6c69  `.wgd ksd famili
+00010490: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
+000104a0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
+000104b0: 2028 2d2d 7265 7765 6967 6874 290a 6060   (--reweight).``
+000104c0: 600a 0a46 726f 6d20 7061 7261 6c6f 676f  `..From paralogo
+000104d0: 7573 2074 6f20 6f72 7468 6f6c 6f67 6f75  us to orthologou
+000104e0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+000104f0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00010500: 6f6e 2c20 7573 6572 7320 6f6e 6c79 206e  on, users only n
+00010510: 6565 6420 746f 2070 726f 7669 6465 206d  eed to provide m
+00010520: 6f72 6520 6364 7320 6669 6c65 732e 204e  ore cds files. N
+00010530: 6f74 6520 7468 6174 2077 6974 6820 6f72  ote that with or
+00010540: 7468 6f6c 6f67 6f75 7320 6765 6e65 2066  thologous gene f
+00010550: 616d 696c 6965 7320 7468 6520 7765 6967  amilies the weig
+00010560: 6874 696e 6720 6d65 7468 6f64 2063 616e  hting method can
+00010570: 2062 6520 7365 7420 746f 2062 6520 6361   be set to be ca
+00010580: 6c63 756c 6174 6564 2070 6572 2073 7065  lculated per spe
+00010590: 6369 6573 2070 6169 7220 696e 7374 6561  cies pair instea
+000105a0: 6420 6f66 2063 6f6e 7369 6465 7269 6e67  d of considering
+000105b0: 2074 6865 2077 686f 6c65 2066 616d 696c   the whole famil
+000105c0: 7920 6265 6361 7573 6520 7768 656e 2070  y because when p
+000105d0: 6c6f 7474 696e 6720 6f72 7468 6f6c 6f67  lotting ortholog
+000105e0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+000105f0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00010600: 7469 6f6e 2062 6574 7765 656e 2074 776f  tion between two
+00010610: 2073 7065 6369 6573 2074 6865 2077 6569   species the wei
+00010620: 6768 7420 6361 6c63 756c 6174 6564 2066  ght calculated f
+00010630: 726f 6d20 7468 6973 2073 7065 6369 6669  rom this specifi
+00010640: 6320 7370 6563 6965 7320 7061 6972 2073  c species pair s
+00010650: 686f 756c 6420 6265 2063 6f6e 7365 7276  hould be conserv
+00010660: 6564 2077 6869 6c65 2074 6865 206f 6e65  ed while the one
+00010670: 2063 616c 6375 6c61 7465 6420 6672 6f6d   calculated from
+00010680: 2074 6865 2077 686f 6c65 2066 616d 696c   the whole famil
+00010690: 7920 7769 6c6c 2076 6172 7920 7769 7468  y will vary with
+000106a0: 2074 6865 206e 756d 6265 7220 6f66 2073   the number of s
+000106b0: 7065 6369 6573 2e20 546f 2069 6e69 7469  pecies. To initi
+000106c0: 6174 6520 7468 6520 7765 6967 6874 696e  ate the weightin
+000106d0: 6720 7065 7220 7370 6563 6965 7320 7061  g per species pa
+000106e0: 6972 2c20 7468 6520 6f70 7469 6f6e 2060  ir, the option `
+000106f0: 2d2d 7265 7765 6967 6874 6020 6361 6e20  --reweight` can 
+00010700: 6265 2073 6574 2e0a 0a2a 2a41 2073 7567  be set...**A sug
+00010710: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+00010720: 7275 6e20 6361 6e20 7573 6520 636f 6d6d  run can use comm
+00010730: 616e 6420 6173 2062 656c 6f77 2a2a 0a0a  and as below**..
+00010740: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
+00010750: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+00010760: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+00010770: 6533 0a60 6060 0a0a 2a2a 5468 6520 636f  e3.```..**The co
+00010780: 6e73 7472 7563 7469 6f6e 206f 6620 2a4b  nstruction of *K
+00010790: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+000107a0: 6520 6469 7374 7269 6275 7469 6f6e 2077  e distribution w
+000107b0: 6974 6820 7261 7465 2063 6f72 7265 6374  ith rate correct
+000107c0: 696f 6e2a 2a0a 6060 600a 7767 6420 6b73  ion**.```.wgd ks
+000107d0: 6420 6661 6d69 6c69 6573 2073 6571 7565  d families seque
+000107e0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+000107f0: 6571 7565 6e63 6533 202d 2d66 6f63 7573  equence3 --focus
+00010800: 3261 6c6c 2073 6571 7565 6e63 6531 202d  2all sequence1 -
+00010810: 7370 2073 7064 6174 6120 2d2d 6578 7472  sp spdata --extr
+00010820: 6170 6172 616e 6f6d 656b 7320 7061 7261  aparanomeks para
+00010830: 6e6f 6d65 4b73 6461 7461 2028 2d2d 706c  nomeKsdata (--pl
+00010840: 6f74 656c 6d6d 202d 2d70 6c6f 7461 7067  otelmm --plotapg
+00010850: 6d6d 202d 2d61 6e63 686f 7270 6f69 6e74  mm --anchorpoint
+00010860: 7320 6170 6461 7461 202d 2d72 6577 6569  s apdata --rewei
+00010870: 6768 7420 2d2d 6f6e 6c79 726f 6f74 6f75  ght --onlyrootou
+00010880: 7429 0a60 6060 0a0a 496e 7370 6972 6564  t).```..Inspired
+00010890: 2062 7920 7468 6520 7261 7465 2063 6f72   by the rate cor
+000108a0: 7265 6374 696f 6e20 616c 676f 7269 7468  rection algorith
+000108b0: 6d20 696e 205b 6b73 7261 7465 735d 2868  m in [ksrates](h
+000108c0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+000108d0: 6d2f 5649 422d 5053 422f 6b73 7261 7465  m/VIB-PSB/ksrate
+000108e0: 7329 2c20 7765 2069 6d70 6c65 6d65 6e74  s), we implement
+000108f0: 6564 2074 6865 2072 6174 6520 636f 7272  ed the rate corr
+00010900: 6563 7469 6f6e 2061 6e61 6c79 7369 7320  ection analysis 
+00010910: 616c 736f 2069 6e20 6077 6764 2076 3260  also in `wgd v2`
+00010920: 2c20 7768 6963 6820 6973 206d 6f73 746c  , which is mostl
+00010930: 7920 7468 6520 7361 6d65 2061 7320 5b6b  y the same as [k
+00010940: 7372 6174 6573 5d28 6874 7470 733a 2f2f  srates](https://
+00010950: 6769 7468 7562 2e63 6f6d 2f56 4942 2d50  github.com/VIB-P
+00010960: 5342 2f6b 7372 6174 6573 2920 6275 7420  SB/ksrates) but 
+00010970: 6469 6666 6572 7320 696e 2074 6865 2063  differs in the c
+00010980: 616c 6375 6c61 7469 6f6e 206f 6620 7468  alculation of th
+00010990: 6520 7374 616e 6461 7264 2064 6576 6961  e standard devia
+000109a0: 7469 6f6e 206f 6620 7265 7363 616c 6564  tion of rescaled
+000109b0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000109c0: 2061 6765 732e 2054 6f20 7065 7266 6f72   ages. To perfor
+000109d0: 6d20 7468 6520 7261 7465 2063 6f72 7265  m the rate corre
+000109e0: 6374 696f 6e20 616e 616c 7973 6973 2c20  ction analysis, 
+000109f0: 7573 6572 7320 6361 6e20 7573 6520 626f  users can use bo
+00010a00: 7468 2074 6865 2060 7767 6420 6b73 6460  th the `wgd ksd`
+00010a10: 2061 6e64 2060 7767 6420 7669 7a60 2070   and `wgd viz` p
+00010a20: 726f 6772 616d 2e20 466f 7220 7468 6520  rogram. For the 
+00010a30: 6077 6764 206b 7364 6020 7072 6f67 7261  `wgd ksd` progra
+00010a40: 6d2c 2075 7365 7273 206e 6565 6420 746f  m, users need to
+00010a50: 2070 726f 7669 6465 2061 2073 7065 6369   provide a speci
+00010a60: 6573 2074 7265 6520 7669 6120 7468 6520  es tree via the 
+00010a70: 6f70 7469 6f6e 2060 2d2d 7370 6563 6965  option `--specie
+00010a80: 7374 7265 6560 206f 6e20 7768 6963 6820  stree` on which 
+00010a90: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
+00010aa0: 6361 6e20 6265 2063 6f6e 6475 6374 6564  can be conducted
+00010ab0: 2e20 4e6f 7465 2074 6861 7420 756e 6e65  . Note that unne
+00010ac0: 6365 7373 6172 7920 6272 6163 6b65 7473  cessary brackets
+00010ad0: 206d 6967 6874 206c 6561 6420 746f 2075   might lead to u
+00010ae0: 6e65 7870 6563 7465 6420 6572 726f 7273  nexpected errors
+00010af0: 2c20 666f 7220 696e 7374 616e 6365 2061  , for instance a
+00010b00: 2074 7265 6520 6028 412c 2842 2c43 2929   tree `(A,(B,C))
+00010b10: 3b60 2073 686f 756c 6420 6e6f 7420 6265  ;` should not be
+00010b20: 2072 6570 7265 7365 6e74 6564 2061 7320   represented as 
+00010b30: 6028 412c 2828 422c 4329 2929 3b60 2e20  `(A,((B,C)));`. 
+00010b40: 5468 6520 7365 7420 6f66 2073 7065 6369  The set of speci
+00010b50: 6573 2070 6169 7273 2074 6f20 6265 2073  es pairs to be s
+00010b60: 686f 776e 2069 7320 666c 6578 6962 6c65  hown is flexible
+00010b70: 2074 6861 7420 7468 6520 6d6f 7374 2063   that the most c
+00010b80: 6f6e 7665 6e69 656e 7420 6f70 7469 6f6e  onvenient option
+00010b90: 2069 7320 602d 2d66 6f63 7573 3261 6c6c   is `--focus2all
+00010ba0: 6020 7768 6963 6820 7369 6d70 6c79 2073  ` which simply s
+00010bb0: 686f 7773 2061 6c6c 2074 6865 2070 6f73  hows all the pos
+00010bc0: 7369 626c 6520 666f 6361 6c2d 7369 7374  sible focal-sist
+00010bd0: 6572 2073 7065 6369 6573 2070 6169 7273  er species pairs
+00010be0: 2c20 6f72 2075 7365 7273 2063 616e 206d  , or users can m
+00010bf0: 616e 7561 6c6c 7920 7365 7420 7468 6520  anually set the 
+00010c00: 7370 6563 6965 7320 7061 6972 7320 7669  species pairs vi
+00010c10: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00010c20: 7370 6169 7260 2e20 4e6f 7465 2074 6861  spair`. Note tha
+00010c30: 7420 6966 2074 6865 2073 7065 6369 6573  t if the species
+00010c40: 2070 6169 7273 2077 6572 6520 6d61 6e75   pairs were manu
+00010c50: 616c 6c79 2073 6574 2c20 6974 2077 6f75  ally set, it wou
+00010c60: 6c64 2062 6520 6e65 6564 6564 2074 6f20  ld be needed to 
+00010c70: 636f 2d73 6574 2074 6865 206f 7074 696f  co-set the optio
+00010c80: 6e20 602d 2d63 6c61 7373 6963 602e 2054  n `--classic`. T
+00010c90: 6865 206f 7074 696f 6e20 602d 2d6f 6e6c  he option `--onl
+00010ca0: 7972 6f6f 746f 7574 6020 6361 6e20 6265  yrootout` can be
+00010cb0: 2073 6574 2074 6f20 6f6e 6c79 2063 6f6e   set to only con
+00010cc0: 7369 6465 7220 6f75 7467 726f 7570 2061  sider outgroup a
+00010cd0: 7420 7468 6520 726f 6f74 2c20 696e 7374  t the root, inst
+00010ce0: 6561 6420 6f66 2061 6c6c 2074 6865 2070  ead of all the p
+00010cf0: 6f73 7369 626c 6520 6f75 7467 726f 7570  ossible outgroup
+00010d00: 7320 7065 7220 666f 6361 6c2d 7369 7374  s per focal-sist
+00010d10: 6572 2073 7065 6369 6573 2070 6169 722c  er species pair,
+00010d20: 2077 6869 6368 2068 6173 2069 6d70 6163   which has impac
+00010d30: 7420 6f6e 2074 6865 2066 696e 616c 2063  t on the final c
+00010d40: 6f72 7265 6374 6564 202a 4b2a 3c73 7562  orrected *K*<sub
+00010d50: 3e53 3c2f 7375 623e 2061 6765 732e 2057  >S</sub> ages. W
+00010d60: 6520 7375 6767 6573 7420 6f66 2075 7369  e suggest of usi
+00010d70: 6e67 2061 6c6c 2074 6865 2070 6f73 7369  ng all the possi
+00010d80: 626c 6520 6f75 7467 726f 7570 7320 7065  ble outgroups pe
+00010d90: 7220 666f 6361 6c2d 7369 7374 6572 2073  r focal-sister s
+00010da0: 7065 6369 6573 2070 6169 7220 6173 2066  pecies pair as f
+00010db0: 6f72 2061 206c 6573 7320 6269 6173 6564  or a less biased
+00010dc0: 2072 6573 756c 742e 2054 6865 2070 6172   result. The par
+00010dd0: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00010de0: 2f73 7562 3e20 6461 7461 2073 686f 756c  /sub> data shoul
+00010df0: 6420 6265 2070 726f 7669 6465 6420 7669  d be provided vi
+00010e00: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00010e10: 6578 7472 6170 6172 616e 6f6d 656b 7360  extraparanomeks`
+00010e20: 2e20 0a0a 536f 6d65 206f 7468 6572 206f  . ..Some other o
+00010e30: 7074 696f 6e73 2077 6869 6368 2068 6176  ptions which hav
+00010e40: 6520 6e6f 2069 6d70 6163 7420 6f6e 2074  e no impact on t
+00010e50: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+00010e60: 6f6e 2062 7574 2061 6464 206d 6f72 6520  on but add more 
+00010e70: 6c61 7965 7273 206f 7220 6368 616e 6765  layers or change
+00010e80: 2074 6865 2061 7070 6561 7261 6e63 6520   the appearance 
+00010e90: 6f6e 2074 6865 206d 6978 6564 202a 4b2a  on the mixed *K*
+00010ea0: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00010eb0: 2064 6973 7472 6962 7574 696f 6e2c 2069   distribution, i
+00010ec0: 6e63 6c75 6465 2060 2d2d 706c 6f74 6170  nclude `--plotap
+00010ed0: 676d 6d60 2061 6e64 2060 2d2d 706c 6f74  gmm` and `--plot
+00010ee0: 656c 6d6d 6020 6574 632e 2054 6865 206f  elmm` etc. The o
+00010ef0: 7074 696f 6e20 602d 2d70 6c6f 7461 7067  ption `--plotapg
+00010f00: 6d6d 6020 6361 6e20 6265 2073 6574 2074  mm` can be set t
+00010f10: 6f20 6361 6c6c 2074 6865 2047 4d4d 2061  o call the GMM a
+00010f20: 6e61 6c79 7369 7320 7570 6f6e 2074 6865  nalysis upon the
+00010f30: 2061 6e63 686f 7220 7061 6972 202a 4b2a   anchor pair *K*
+00010f40: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00010f50: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
+00010f60: 6420 706c 6f74 2074 6865 2063 6c75 7374  d plot the clust
+00010f70: 6572 696e 6720 7265 7375 6c74 2075 706f  ering result upo
+00010f80: 6e20 7468 6520 6d69 7865 6420 2a4b 2a3c  n the mixed *K*<
+00010f90: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+00010fa0: 6469 7374 7269 6275 7469 6f6e 2c20 7768  distribution, wh
+00010fb0: 6963 6820 6861 7320 746f 2062 6520 636f  ich has to be co
+00010fc0: 2d73 6574 2077 6974 6820 7468 6520 6f70  -set with the op
+00010fd0: 7469 6f6e 2060 2d2d 616e 6368 6f72 706f  tion `--anchorpo
+00010fe0: 696e 7473 6020 7072 6f76 6964 696e 6720  ints` providing 
+00010ff0: 7468 6520 616e 6368 6f72 2070 6169 7273  the anchor pairs
+00011000: 2069 6e66 6f72 6d61 7469 6f6e 2e20 5468   information. Th
+00011010: 6520 6f70 7469 6f6e 2060 2d2d 706c 6f74  e option `--plot
+00011020: 656c 6d6d 6020 6361 6e20 6265 2073 6574  elmm` can be set
+00011030: 2074 6f20 6361 6c6c 2074 6865 2045 4c4d   to call the ELM
+00011040: 4d20 616e 616c 7973 6973 2075 706f 6e20  M analysis upon 
+00011050: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
+00011060: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
+00011070: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
+00011080: 696f 6e2e 202a 2a4e 6f74 6520 7468 6174  ion. **Note that
+00011090: 2074 6865 2073 7065 6369 6573 206e 616d   the species nam
+000110a0: 6573 2070 7265 7365 6e74 2069 6e20 7468  es present in th
+000110b0: 6520 7370 6563 6965 7320 7472 6565 2066  e species tree f
+000110c0: 696c 6520 7368 6f75 6c64 206d 6174 6368  ile should match
+000110d0: 2074 6865 206e 616d 6573 206f 6620 7468   the names of th
+000110e0: 6520 636f 7272 6573 706f 6e64 696e 6720  e corresponding 
+000110f0: 7365 7175 656e 6365 2066 696c 6573 2a2a  sequence files**
+00011100: 2046 6f72 2069 6e73 7461 6e63 652c 2067   For instance, g
+00011110: 6976 656e 2074 6865 2063 6473 2066 696c  iven the cds fil
+00011120: 6520 6e61 6d65 7320 2741 2e63 6473 272c  e names 'A.cds',
+00011130: 2742 2e63 6473 272c 2743 2e63 6473 272c  'B.cds','C.cds',
+00011140: 2074 6865 2073 7065 6369 6573 2074 7265   the species tre
+00011150: 6520 636f 756c 6420 6265 2027 2841 2e63  e could be '(A.c
+00011160: 6473 2c28 422e 6364 732c 432e 6364 7329  ds,(B.cds,C.cds)
+00011170: 293b 2720 7261 7468 6572 2074 6861 6e20  );' rather than 
+00011180: 2728 412c 2842 2c43 2929 3b27 2e20 5468  '(A,(B,C));'. Th
+00011190: 6572 6520 6973 206e 6f20 7265 7175 6972  ere is no requir
+000111a0: 656d 656e 7420 666f 7220 7468 6520 6e61  ement for the na
+000111b0: 6d65 206f 6620 7468 6520 7061 7261 6e6f  me of the parano
+000111c0: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
+000111d0: 623e 2064 6174 6166 696c 6520 7768 6963  b> datafile whic
+000111e0: 6820 6361 6e20 6265 206e 616d 6564 2069  h can be named i
+000111f0: 6e20 7768 6174 6576 6572 206d 616e 6e65  n whatever manne
+00011200: 7220 7573 6572 7320 7072 6566 6572 2e20  r users prefer. 
+00011210: 5468 6520 2747 4d4d 2720 6973 2074 6865  The 'GMM' is the
+00011220: 2061 6262 7265 7669 6174 696f 6e20 6f66   abbreviation of
+00011230: 2047 6175 7373 6961 6e20 4d69 7874 7572   Gaussian Mixtur
+00011240: 6520 4d6f 6465 6c69 6e67 2077 6869 6c65  e Modeling while
+00011250: 2074 6865 2027 454c 4d4d 2720 7265 6665   the 'ELMM' refe
+00011260: 7273 2074 6f20 4578 706f 6e65 6e74 6961  rs to Exponentia
+00011270: 6c2d 4c6f 676e 6f72 6d61 6c20 4d69 7874  l-Lognormal Mixt
+00011280: 7572 6520 4d6f 6465 6c69 6e67 2061 7320  ure Modeling as 
+00011290: 5b6b 7372 6174 6573 5d28 6874 7470 733a  [ksrates](https:
+000112a0: 2f2f 6769 7468 7562 2e63 6f6d 2f56 4942  //github.com/VIB
+000112b0: 2d50 5342 2f6b 7372 6174 6573 2920 696e  -PSB/ksrates) in
+000112c0: 7465 7270 7265 7473 2e0a 0a54 6865 7265  terprets...There
+000112d0: 2061 7265 2032 3120 636f 6c75 6d6e 7320   are 21 columns 
+000112e0: 696e 2074 6865 2072 6573 756c 7420 602e  in the result `.
+000112f0: 6b73 2e74 7376 6020 6669 6c65 2062 6573  ks.tsv` file bes
+00011300: 6964 6573 2074 6865 2069 6e64 6578 2063  ides the index c
+00011310: 6f6c 756d 6e73 2060 7061 6972 6020 6173  olumns `pair` as
+00011320: 2074 6865 2075 6e69 7175 6520 6964 656e   the unique iden
+00011330: 7469 6669 6572 2066 6f72 2065 6163 6820  tifier for each 
+00011340: 6765 6e65 2070 6169 722e 2054 6865 2060  gene pair. The `
+00011350: 4e60 2c20 6053 602c 2060 644e 602c 2060  N`, `S`, `dN`, `
+00011360: 644e 2f64 5360 2c20 6064 5360 2c20 606c  dN/dS`, `dS`, `l
+00011370: 6020 616e 6420 6074 6020 6172 6520 6672  ` and `t` are fr
+00011380: 6f6d 2074 6865 2063 6f64 656d 6c20 7265  om the codeml re
+00011390: 7375 6c74 732c 2072 6570 7265 7365 6e74  sults, represent
+000113a0: 696e 6720 7468 6520 4e20 6573 7469 6d61  ing the N estima
+000113b0: 7465 2c20 7468 6520 5320 6573 7469 6d61  te, the S estima
+000113c0: 7465 2c20 7468 6520 644e 2065 7374 696d  te, the dN estim
+000113d0: 6174 652c 2074 6865 2064 4e2f 6453 2028  ate, the dN/dS (
+000113e0: 6f6d 6567 6129 2065 7374 696d 6174 652c  omega) estimate,
+000113f0: 2074 6865 2064 5320 6573 7469 6d61 7465   the dS estimate
+00011400: 2c20 7468 6520 6c6f 672d 6c69 6b65 6c69  , the log-likeli
+00011410: 686f 6f64 2061 6e64 2074 6865 2074 2065  hood and the t e
+00011420: 7374 696d 6174 652c 2072 6573 7065 6374  stimate, respect
+00011430: 6976 656c 792e 2054 6865 2060 616c 6967  ively. The `alig
+00011440: 6e6d 656e 7463 6f76 6572 6167 6560 2c20  nmentcoverage`, 
+00011450: 6061 6c69 676e 6d65 6e74 6964 656e 7469  `alignmentidenti
+00011460: 7479 6020 616e 6420 6061 6c69 676e 6d65  ty` and `alignme
+00011470: 6e74 6c65 6e67 7468 6020 6172 6520 7468  ntlength` are th
+00011480: 6520 696e 666f 726d 6174 696f 6e20 7065  e information pe
+00011490: 7274 6169 6e69 6e67 2074 6f20 7468 6520  rtaining to the 
+000114a0: 616c 6967 6e6d 656e 7420 666f 7220 6561  alignment for ea
+000114b0: 6368 2066 616d 696c 792c 2072 6570 7265  ch family, repre
+000114c0: 7365 6e74 696e 6720 7468 6520 7261 7469  senting the rati
+000114d0: 6f20 6f66 2074 6865 2073 7472 6970 7065  o of the strippe
+000114e0: 6420 616c 6967 6e6d 656e 7420 6c65 6e67  d alignment leng
+000114f0: 7468 2063 6f6d 7061 7265 6420 746f 2074  th compared to t
+00011500: 6865 2066 756c 6c20 616c 6967 6e6d 656e  he full alignmen
+00011510: 7420 6c65 6e67 7468 2c20 7468 6520 7261  t length, the ra
+00011520: 7469 6f20 6f66 2063 6f6c 756d 6e73 2077  tio of columns w
+00011530: 6974 6820 6964 656e 7469 6361 6c20 6e75  ith identical nu
+00011540: 636c 656f 7469 6465 7320 636f 6d70 6172  cleotides compar
+00011550: 6564 2074 6f20 7468 6520 6f76 6572 616c  ed to the overal
+00011560: 6c20 636f 6c75 6d6e 7320 6f66 2074 6865  l columns of the
+00011570: 2073 7472 6970 7065 6420 616c 6967 6e6d   stripped alignm
+00011580: 656e 742c 2061 6e64 2074 6865 206c 656e  ent, and the len
+00011590: 6774 6820 6f66 2074 6865 2066 756c 6c20  gth of the full 
+000115a0: 616c 6967 6e6d 656e 742c 2072 6573 7065  alignment, respe
+000115b0: 6374 6976 656c 792e 0a0a 2323 2320 7767  ctively...### wg
+000115c0: 6420 7379 6e0a 0a2a 2a54 6865 2069 6e74  d syn..**The int
+000115d0: 7261 2d73 7065 6369 6669 6320 7379 6e74  ra-specific synt
+000115e0: 656e 7920 696e 6665 7265 6e63 652a 2a0a  eny inference**.
+000115f0: 6060 600a 7767 6420 7379 6e20 6661 6d69  ```.wgd syn fami
+00011600: 6c69 6573 2067 6666 2028 2d2d 6b73 5f64  lies gff (--ks_d
+00011610: 6973 7472 6962 7574 696f 6e20 6b73 6461  istribution ksda
+00011620: 7461 202d 6620 6765 6e65 202d 6120 4944  ta -f gene -a ID
+00011630: 202d 2d6d 696e 6c65 6e20 2d31 202d 2d6d   --minlen -1 --m
+00011640: 696e 7365 676c 656e 2031 3030 3030 202d  inseglen 10000 -
+00011650: 2d6d 696e 6765 6e65 6e75 6d20 3330 290a  -mingenenum 30).
+00011660: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
+00011670: 2060 7767 6420 7379 6e60 2069 7320 6d61   `wgd syn` is ma
+00011680: 696e 6c79 2064 6561 6c69 6e67 2077 6974  inly dealing wit
+00011690: 6820 636f 6c6c 696e 6561 7269 7479 206f  h collinearity o
+000116a0: 7220 7379 6e74 656e 7920 2862 6f74 6820  r synteny (both 
+000116b0: 7265 6665 7272 6564 2074 6f20 6173 2073  referred to as s
+000116c0: 796e 7465 6e79 2068 6572 6561 6674 6572  ynteny hereafter
+000116d0: 2920 616e 616c 7973 6973 2e20 5477 6f20  ) analysis. Two 
+000116e0: 696e 7075 7420 6669 6c65 7320 6172 6520  input files are 
+000116f0: 6573 7365 6e74 6961 6c2c 2074 6865 2067  essential, the g
+00011700: 656e 6520 6661 6d69 6c79 2066 696c 6520  ene family file 
+00011710: 616e 6420 7468 6520 6766 6633 2066 696c  and the gff3 fil
+00011720: 652e 2054 6865 2067 656e 6520 6661 6d69  e. The gene fami
+00011730: 6c79 2066 696c 6520 6973 2069 6e20 7468  ly file is in th
+00011740: 6520 666f 726d 6174 2061 7320 604f 7274  e format as `Ort
+00011750: 686f 4669 6e64 6572 602e 2054 6865 2073  hoFinder`. The s
+00011760: 6f66 7477 6172 6520 6069 2d61 6468 6f72  oftware `i-adhor
+00011770: 6560 2069 7320 6120 7072 6572 6571 7569  e` is a prerequi
+00011780: 7369 7465 2e20 5769 7468 2064 6566 6175  site. With defau
+00011790: 6c74 2070 6172 616d 6574 6572 732c 2074  lt parameters, t
+000117a0: 6865 2070 726f 6772 616d 2062 6173 6963  he program basic
+000117b0: 616c 6c79 2063 6f6e 6475 6374 7320 3129  ally conducts 1)
+000117c0: 2066 696c 7465 7269 6e67 2067 656e 6520   filtering gene 
+000117d0: 6661 6d69 6c69 6573 2062 6173 6564 206f  families based o
+000117e0: 6e20 6d61 7869 6d75 6d20 6661 6d69 6c79  n maximum family
+000117f0: 2073 697a 6520 3229 2072 6574 7269 6576   size 2) retriev
+00011800: 696e 6720 6765 6e65 2070 6f73 6974 696f  ing gene positio
+00011810: 6e20 616e 6420 7363 6166 666f 6c64 2069  n and scaffold i
+00011820: 6e66 6f72 6d61 7469 6f6e 2066 726f 6d20  nformation from 
+00011830: 6766 6633 2066 696c 6520 3329 2070 726f  gff3 file 3) pro
+00011840: 6475 6369 6e67 2074 6865 2063 6f6e 6669  ducing the confi
+00011850: 6775 7261 7469 6f6e 2066 696c 6520 616e  guration file an
+00011860: 6420 6173 736f 6369 6174 6564 2064 6174  d associated dat
+00011870: 6166 696c 6573 2066 6f72 2060 692d 6164  afiles for `i-ad
+00011880: 686f 7265 6020 3429 2063 616c 6c69 6e67  hore` 4) calling
+00011890: 2060 692d 6164 686f 7265 6020 6769 7665   `i-adhore` give
+000118a0: 6e20 7468 6520 7061 7261 6d65 7465 7273  n the parameters
+000118b0: 2073 6574 2074 6f20 696e 6665 7220 7379   set to infer sy
+000118c0: 6e74 656e 7920 3529 2076 6973 7561 6c69  nteny 5) visuali
+000118d0: 7a69 6e67 2074 6865 2073 796e 7465 6e79  zing the synteny
+000118e0: 2069 6e20 2264 6f74 706c 6f74 2220 696e   in "dotplot" in
+000118f0: 2074 6865 2075 6e69 7420 6f66 2067 656e   the unit of gen
+00011900: 6573 2061 6e64 2062 6173 6573 2c20 696e  es and bases, in
+00011910: 2022 5379 6e64 6570 7468 2220 706c 6f74   "Syndepth" plot
+00011920: 2073 686f 7769 6e67 2074 6865 2064 6973   showing the dis
+00011930: 7472 6962 7574 696f 6e20 6f66 2064 6966  tribution of dif
+00011940: 6665 7265 6e74 2063 6174 6567 6f72 6965  ferent categorie
+00011950: 7320 6f66 2063 6f6c 6c69 6e65 6172 6974  s of collinearit
+00011960: 7920 7261 7469 6f73 2077 6974 6869 6e20  y ratios within 
+00011970: 616e 6420 6265 7477 6565 6e20 7370 6563  and between spec
+00011980: 6965 732c 2069 6e20 2264 7570 5374 6163  ies, in "dupStac
+00011990: 6b22 2070 6c6f 7420 7368 6f77 696e 6720  k" plot showing 
+000119a0: 6d75 6c74 6970 6c69 636f 6e73 2077 6974  multiplicons wit
+000119b0: 6820 6469 6666 6572 656e 7420 6d75 6c74  h different mult
+000119c0: 6970 6c69 6361 7469 6f6e 206c 6576 656c  iplication level
+000119d0: 732e 2036 2920 6966 2077 6974 6820 2a4b  s. 6) if with *K
+000119e0: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+000119f0: 7461 2c20 6120 222a 4b2a 3c73 7562 3e53  ta, a "*K*<sub>S
+00011a00: 3c2f 7375 623e 2064 6f74 706c 6f74 2220  </sub> dotplot" 
+00011a10: 7769 7468 2064 6f74 7320 616e 6e6f 7461  with dots annota
+00011a20: 7465 6420 696e 202a 4b2a 3c73 7562 3e53  ted in *K*<sub>S
+00011a30: 3c2f 7375 623e 2076 616c 7565 7320 616e  </sub> values an
+00011a40: 6420 6120 2a4b 2a3c 7375 623e 533c 2f73  d a *K*<sub>S</s
+00011a50: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
+00011a60: 2077 6974 6820 616e 6368 6f72 2070 6169   with anchor pai
+00011a70: 7273 2064 656e 6f74 6564 2077 696c 6c20  rs denoted will 
+00011a80: 6265 2070 726f 6475 6365 642e 2054 6865  be produced. The
+00011a90: 2067 656e 6520 696e 666f 726d 6174 696f   gene informatio
+00011aa0: 6e20 696e 2074 6865 2067 656e 6520 6661  n in the gene fa
+00011ab0: 6d69 6c79 2066 696c 6520 616e 6420 6766  mily file and gf
+00011ac0: 6633 2066 696c 6520 7368 6f75 6c64 2062  f3 file should b
+00011ad0: 6520 6d61 7463 6865 6420 7768 6963 6820  e matched which 
+00011ae0: 7265 7175 6972 6573 2075 7365 7273 2074  requires users t
+00011af0: 6f20 7365 7420 7072 6f70 6572 2060 2d2d  o set proper `--
+00011b00: 6665 6174 7572 6560 2061 6e64 2060 2d2d  feature` and `--
+00011b10: 6174 7472 6962 7574 6560 2e20 5468 6520  attribute`. The 
+00011b20: 6d61 7869 6d75 6d20 6661 6d69 6c79 2073  maximum family s
+00011b30: 697a 6520 746f 2062 6520 696e 636c 7564  ize to be includ
+00011b40: 6564 2063 616e 2062 6520 7365 7420 7669  ed can be set vi
+00011b50: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00011b60: 6d61 7873 697a 6560 2c20 6e6f 7465 6420  maxsize`, noted 
+00011b70: 7468 6174 2074 6869 7320 6669 6c74 6572  that this filter
+00011b80: 696e 6720 6973 206d 6169 6e6c 7920 746f  ing is mainly to
+00011b90: 2064 726f 7020 7468 6f73 6520 6875 6765   drop those huge
+00011ba0: 2074 616e 6465 6d20 6475 706c 6963 6174   tandem duplicat
+00011bb0: 6573 2066 616d 696c 7920 616e 6420 7472  es family and tr
+00011bc0: 616e 7370 6f73 6162 6c65 2065 6c65 6d65  ansposable eleme
+00011bd0: 6e74 7320 2854 4573 2920 6661 6d69 6c79  nts (TEs) family
+00011be0: 2c20 616e 6420 6e6f 7420 6d61 6e64 6174  , and not mandat
+00011bf0: 6f72 792e 2055 7365 7273 2063 616e 2066  ory. Users can f
+00011c00: 696c 7465 7220 7468 6f73 6520 6672 6167  ilter those frag
+00011c10: 6d65 6e74 6172 7920 7363 6166 666f 6c64  mentary scaffold
+00011c20: 7320 7669 6120 7468 6520 6f70 7469 6f6e  s via the option
+00011c30: 2060 2d2d 6d69 6e6c 656e 602e 2054 6865   `--minlen`. The
+00011c40: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
+00011c50: 616e 6420 6e75 6d62 6572 206f 6620 6765  and number of ge
+00011c60: 6e65 7320 666f 7220 6120 7365 676d 656e  nes for a segmen
+00011c70: 7420 746f 2062 6520 636f 6e73 6964 6572  t to be consider
+00011c80: 6564 2063 616e 2062 6520 7365 7420 7669  ed can be set vi
+00011c90: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
+00011ca0: 6d69 6e73 6567 6c65 6e60 2061 6e64 2060  minseglen` and `
+00011cb0: 2d2d 6d69 6e67 656e 656e 756d 602e 2052  --mingenenum`. R
+00011cc0: 6564 756e 6461 6e74 206d 756c 7469 706c  edundant multipl
+00011cd0: 6963 6f6e 7320 6361 6e20 6265 206b 6570  icons can be kep
+00011ce0: 7420 6279 2073 6574 2074 6865 2066 6c61  t by set the fla
+00011cf0: 6720 6f70 7469 6f6e 2060 2d2d 6b65 6570  g option `--keep
+00011d00: 7265 6475 6e60 2e0a 0a2a 2a41 2073 7567  redun`...**A sug
+00011d10: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+00011d20: 7275 6e20 6361 6e20 7573 6520 636f 6d6d  run can use comm
+00011d30: 616e 6420 7369 6d70 6c79 2061 7320 6265  and simply as be
+00011d40: 6c6f 772a 2a0a 0a60 6060 0a77 6764 2073  low**..```.wgd s
+00011d50: 796e 2066 616d 696c 6965 7320 6766 660a  yn families gff.
+00011d60: 6060 600a 0a2a 2a54 6865 2069 6e74 6572  ```..**The inter
+00011d70: 2d73 7065 6369 6669 6320 7379 6e74 656e  -specific synten
+00011d80: 7920 696e 6665 7265 6e63 652a 2a0a 6060  y inference**.``
+00011d90: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
+00011da0: 6573 2067 6666 3120 6766 6632 2028 2d2d  es gff1 gff2 (--
+00011db0: 6164 6469 7469 6f6e 616c 6766 6669 6e66  additionalgffinf
+00011dc0: 6f20 226d 524e 413b 4e61 6d65 2220 2d2d  o "mRNA;Name" --
+00011dd0: 6164 6469 7469 6f6e 616c 6766 6669 6e66  additionalgffinf
+00011de0: 6f20 2267 656e 653b 4944 2229 0a60 6060  o "gene;ID").```
+00011df0: 0a0a 466f 7220 6d75 6c74 692d 7370 6563  ..For multi-spec
+00011e00: 6965 7320 7379 6e74 656e 7920 696e 6665  ies synteny infe
+00011e10: 7265 6e63 652c 2069 6620 7573 6572 7320  rence, if users 
+00011e20: 6861 7665 2067 6666 3320 6669 6c65 7320  have gff3 files 
+00011e30: 7768 6963 6820 6861 7665 2064 6966 6665  which have diffe
+00011e40: 7265 6e74 2066 6561 7475 7265 7320 6f72  rent features or
+00011e50: 2061 7474 7269 6275 7465 7320 666f 7220   attributes for 
+00011e60: 6765 6e65 2070 6f73 6974 696f 6e20 696e  gene position in
+00011e70: 666f 726d 6174 696f 6e20 7265 7472 6965  formation retrie
+00011e80: 7661 6c2c 2074 6865 206f 7074 696f 6e20  val, the option 
+00011e90: 602d 2d61 6464 6974 696f 6e61 6c67 6666  `--additionalgff
+00011ea0: 696e 666f 6020 6361 6e20 6265 2073 6574  info` can be set
+00011eb0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+00011ec0: 6164 6469 7469 6f6e 616c 2069 6e66 6f72  additional infor
+00011ed0: 6d61 7469 6f6e 2e20 5468 6520 7265 6d61  mation. The rema
+00011ee0: 696e 696e 6720 7061 7261 6d65 7465 7220  ining parameter 
+00011ef0: 7365 7474 696e 6720 6973 2074 6865 2073  setting is the s
+00011f00: 616d 6520 6173 2074 6865 2069 6e74 7261  ame as the intra
+00011f10: 2d73 7065 6369 6669 6320 7379 6e74 656e  -specific synten
+00011f20: 7920 696e 6665 7265 6e63 652e 0a0a 2a2a  y inference...**
+00011f30: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+00011f40: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
+00011f50: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
+00011f60: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
+00011f70: 7767 6420 7379 6e20 6661 6d69 6c69 6573  wgd syn families
+00011f80: 2067 6666 3120 6766 6632 0a60 6060 0a0a   gff1 gff2.```..
+00011f90: 2323 2320 7767 6420 7669 7a0a 0a2a 2a54  ### wgd viz..**T
+00011fa0: 6865 2076 6973 7561 6c69 7a61 7469 6f6e  he visualization
+00011fb0: 206f 6620 2a4b 2a3c 7375 623e 533c 2f73   of *K*<sub>S</s
+00011fc0: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00011fd0: 7469 6f6e 2061 6e64 2045 4c4d 4d20 616e  tion and ELMM an
+00011fe0: 616c 7973 6973 2a2a 0a60 6060 0a77 6764  alysis**.```.wgd
+00011ff0: 2076 697a 202d 6420 6b73 6461 7461 0a60   viz -d ksdata.`
+00012000: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+00012010: 6077 6764 2076 697a 6020 6973 206d 6169  `wgd viz` is mai
+00012020: 6e6c 7920 666f 7220 7468 6520 7075 7270  nly for the purp
+00012030: 6f73 6520 6f66 202a 4b2a 3c73 7562 3e53  ose of *K*<sub>S
+00012040: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
+00012050: 696f 6e20 616e 6420 7379 6e74 656e 7920  ion and synteny 
+00012060: 7669 7375 616c 697a 6174 696f 6e2c 2077  visualization, w
+00012070: 6974 6820 736f 6d65 206f 7074 696f 6e61  ith some optiona
+00012080: 6c20 6d69 7874 7572 6520 6d6f 6465 6c69  l mixture modeli
+00012090: 6e67 2061 6e61 6c79 7369 732e 2054 6865  ng analysis. The
+000120a0: 2062 6173 6963 2066 756e 6374 696f 6e20   basic function 
+000120b0: 6973 206a 7573 7420 746f 2070 6c6f 7420  is just to plot 
+000120c0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+000120d0: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
+000120e0: 2061 6e64 2063 6f6e 6475 6374 2061 6e20   and conduct an 
+000120f0: 454c 4d4d 2061 6e61 6c79 7369 7320 696e  ELMM analysis in
+00012100: 2073 6561 7263 6820 6f66 2070 6f74 656e   search of poten
+00012110: 7469 616c 2057 4744 2063 6f6d 706f 6e65  tial WGD compone
+00012120: 6e74 732e 2053 6f6d 6520 6b65 7920 7061  nts. Some key pa
+00012130: 7261 6d65 7465 7273 2061 6666 6563 7469  rameters affecti
+00012140: 6e67 2074 6865 2045 4c4d 4d20 7265 7375  ng the ELMM resu
+00012150: 6c74 2069 6e63 6c75 6465 2060 2d2d 7072  lt include `--pr
+00012160: 6f6d 696e 656e 6365 5f63 7574 6f66 6660  ominence_cutoff`
+00012170: 2061 6e64 2060 2d2d 7265 6c5f 6865 6967   and `--rel_heig
+00012180: 6874 602c 2077 6869 6368 2068 6176 6520  ht`, which have 
+00012190: 6265 656e 2065 7870 6c61 696e 6564 2061  been explained a
+000121a0: 6865 6164 2c20 602d 2d65 6d5f 6974 6572  head, `--em_iter
+000121b0: 6174 696f 6e73 6020 616e 6420 602d 2d65  ations` and `--e
+000121c0: 6d5f 696e 6974 6961 6c69 7a61 7469 6f6e  m_initialization
+000121d0: 7360 2064 6574 6572 6d69 6e69 6e67 2074  s` determining t
+000121e0: 6865 206d 6178 696d 756d 2069 7465 7261  he maximum itera
+000121f0: 7469 6f6e 7320 616e 6420 696e 6974 6961  tions and initia
+00012200: 6c69 7a61 7469 6f6e 7320 696e 2074 6865  lizations in the
+00012210: 2045 4d20 616c 676f 7269 7468 6d2e 0a0a   EM algorithm...
+00012220: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+00012230: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+00012240: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
+00012250: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
+00012260: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
+00012270: 6174 610a 6060 600a 0a2a 2a54 6865 2076  ata.```..**The v
+00012280: 6973 7561 6c69 7a61 7469 6f6e 206f 6620  isualization of 
+00012290: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000122a0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+000122b0: 2077 6974 6820 7261 7465 2063 6f72 7265   with rate corre
+000122c0: 6374 696f 6e2a 2a0a 6060 600a 7767 6420  ction**.```.wgd 
+000122d0: 7669 7a20 2d64 206b 7364 6174 6120 2d73  viz -d ksdata -s
+000122e0: 7020 7370 6461 7461 202d 2d66 6f63 7573  p spdata --focus
+000122f0: 3261 6c6c 2066 6f63 616c 5f73 7065 6369  2all focal_speci
+00012300: 6573 202d 2d65 7874 7261 7061 7261 6e6f  es --extraparano
+00012310: 6d65 6b73 206b 7364 6174 6120 282d 2d61  meks ksdata (--a
+00012320: 6e63 686f 7270 6f69 6e74 7320 6170 6461  nchorpoints apda
+00012330: 7461 202d 2d70 6c6f 7461 7067 6d6d 202d  ta --plotapgmm -
+00012340: 2d70 6c6f 7465 6c6d 6d29 0a60 6060 0a0a  -plotelmm).```..
+00012350: 4265 7369 6465 7320 7468 6520 6261 7369  Besides the basi
+00012360: 6320 2a4b 2a3c 7375 623e 533c 2f73 7562  c *K*<sub>S</sub
+00012370: 3e20 706c 6f74 2c20 7375 6273 7469 7475  > plot, substitu
+00012380: 7469 6f6e 2072 6174 6520 636f 7272 6563  tion rate correc
+00012390: 7469 6f6e 2063 616e 2061 6c73 6f20 6265  tion can also be
+000123a0: 2061 6368 6965 7665 6420 6769 7665 6e20   achieved given 
+000123b0: 6174 206c 6561 7374 2061 2073 7065 6369  at least a speci
+000123c0: 6573 2074 7265 6520 2876 6961 2074 6865  es tree (via the
+000123d0: 206f 7074 696f 6e20 602d 2d73 7065 6369   option `--speci
+000123e0: 6573 7472 6565 6029 2061 6e64 2061 2066  estree`) and a f
+000123f0: 6f63 616c 2073 7065 6369 6573 2028 6569  ocal species (ei
+00012400: 7468 6572 2076 6961 2074 6865 206f 7074  ther via the opt
+00012410: 696f 6e20 602d 2d66 6f63 7573 3261 6c6c  ion `--focus2all
+00012420: 6020 6f72 2076 6961 2074 6865 206f 7074  ` or via the opt
+00012430: 696f 6e20 602d 2d73 7061 6972 6020 696e  ion `--spair` in
+00012440: 2074 6865 2066 6f72 6d20 6f66 2022 2466   the form of "$f
+00012450: 6f63 616c 5f73 7065 6369 6573 3b24 666f  ocal_species;$fo
+00012460: 6361 6c5f 7370 6563 6965 7322 292e 2049  cal_species"). I
+00012470: 7427 7320 7375 6767 6573 7465 6420 7468  t's suggested th
+00012480: 6174 2074 6865 206f 7274 686f 6c6f 676f  at the orthologo
+00012490: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+000124a0: 623e 2064 6174 6120 6973 2070 726f 7669  b> data is provi
+000124b0: 6465 6420 6279 2074 6865 2060 2d2d 6461  ded by the `--da
+000124c0: 7461 6669 6c65 6020 6f70 7469 6f6e 2061  tafile` option a
+000124d0: 6e64 2074 6865 2070 6172 616c 6f67 6f75  nd the paralogou
+000124e0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+000124f0: 3e20 6461 7461 2069 7320 7072 6f76 6964  > data is provid
+00012500: 6564 2062 7920 7468 6520 602d 2d65 7874  ed by the `--ext
+00012510: 7261 7061 7261 6e6f 6d65 6b73 6020 6f70  raparanomeks` op
+00012520: 7469 6f6e 2c20 616c 7468 6f75 6768 2069  tion, although i
+00012530: 7427 7320 616c 6c6f 7765 6420 746f 206f  t's allowed to o
+00012540: 6e6c 7920 7072 6f76 6964 6520 2a4b 2a3c  nly provide *K*<
+00012550: 7375 623e 533c 2f73 7562 3e20 6461 7461  sub>S</sub> data
+00012560: 2076 6961 2074 6865 2060 2d2d 6461 7461   via the `--data
+00012570: 6669 6c65 6020 6f70 7469 6f6e 2061 6e64  file` option and
+00012580: 2064 6570 6f73 6974 2062 6f74 6820 6f72   deposit both or
+00012590: 7468 6f6c 6f67 6f75 7320 616e 6420 7061  thologous and pa
+000125a0: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
+000125b0: 3e53 3c2f 7375 623e 2064 6174 6120 7468  >S</sub> data th
+000125c0: 6572 656f 6e2e 2054 6865 7265 2061 7265  ereon. There are
+000125d0: 2074 776f 2074 7970 6573 206f 6620 6d69   two types of mi
+000125e0: 7865 6420 706c 6f74 7320 2874 6865 2022  xed plots (the "
+000125f0: 6d69 7865 6422 2068 6572 6520 7265 6665  mixed" here refe
+00012600: 7273 2074 6f20 6d69 7865 6420 6f72 7468  rs to mixed orth
+00012610: 6f6c 6f67 6f75 7320 616e 6420 7061 7261  ologous and para
+00012620: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+00012630: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
+00012640: 696f 6e73 292c 206f 6e65 206f 6620 7768  ions), one of wh
+00012650: 6963 6820 6973 2073 696d 696c 6172 2074  ich is similar t
+00012660: 6f20 7768 6174 2060 6b73 7261 7465 7360  o what `ksrates`
+00012670: 2070 6c6f 7473 2c20 7768 696c 6520 7468   plots, while th
+00012680: 6520 6f74 6865 7220 6f66 2077 6869 6368  e other of which
+00012690: 2069 7320 6c69 6b65 2074 6865 2063 6f6e   is like the con
+000126a0: 7665 6e74 696f 6e61 6c20 2a4b 2a3c 7375  ventional *K*<su
+000126b0: 623e 533c 2f73 7562 3e20 706c 6f74 2074  b>S</sub> plot t
+000126c0: 6861 7420 626f 7468 2074 6865 206f 7269  hat both the ori
+000126d0: 6769 6e61 6c20 6f72 7468 6f6c 6f67 6f75  ginal orthologou
+000126e0: 7320 616e 6420 7061 7261 6c6f 676f 7573  s and paralogous
+000126f0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00012700: 2064 6973 7472 6962 7574 696f 6e73 2061   distributions a
+00012710: 7265 2074 7275 7468 6675 6c6c 7920 706c  re truthfully pl
+00012720: 6f74 7465 6420 696e 7374 6561 6420 6f66  otted instead of
+00012730: 206a 7573 7420 6265 696e 6720 7265 7072   just being repr
+00012740: 6573 656e 7465 6420 6279 2073 6f6d 6520  esented by some 
+00012750: 7665 7274 6963 616c 206c 696e 6573 2e20  vertical lines. 
+00012760: 5765 2073 7567 6765 7374 2075 7365 7273  We suggest users
+00012770: 2074 6f20 6164 6f70 7420 7468 6520 606b   to adopt the `k
+00012780: 7372 6174 6573 602d 6c69 6b65 2070 6c6f  srates`-like plo
+00012790: 7473 2c20 7768 6963 6820 6973 2074 6865  ts, which is the
+000127a0: 2064 6566 6175 6c74 2e20 4f74 6865 7277   default. Otherw
+000127b0: 6973 6520 7468 6520 6f70 7469 6f6e 2060  ise the option `
+000127c0: 2d2d 636c 6173 7369 6360 2077 696c 6c20  --classic` will 
+000127d0: 6265 206e 6565 6465 6420 746f 2073 6574  be needed to set
+000127e0: 2e20 4578 7472 6120 6d69 7874 7572 6520  . Extra mixture 
+000127f0: 6d6f 6465 6c69 6e67 2061 6e61 6c79 7369  modeling analysi
+00012800: 7320 6361 6e20 6265 2069 6e69 7469 6174  s can be initiat
+00012810: 6564 2076 6961 2074 6865 206f 7074 696f  ed via the optio
+00012820: 6e20 602d 2d70 6c6f 7465 6c6d 6d60 2061  n `--plotelmm` a
+00012830: 6e64 2060 2d2d 706c 6f74 6170 676d 6d60  nd `--plotapgmm`
+00012840: 2077 6974 6820 7468 6520 616e 6368 6f72   with the anchor
+00012850: 2070 6f69 6e74 7320 6461 7461 6669 6c65   points datafile
+00012860: 2070 726f 7669 6465 6420 6279 2074 6865   provided by the
+00012870: 2060 2d2d 616e 6368 6f72 706f 696e 7473   `--anchorpoints
+00012880: 6020 6f70 7469 6f6e 2e0a 0a2a 2a41 2073  ` option...**A s
+00012890: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+000128a0: 6720 7275 6e20 6361 6e20 7573 6520 636f  g run can use co
+000128b0: 6d6d 616e 6420 7369 6d70 6c79 2061 7320  mmand simply as 
+000128c0: 6265 6c6f 772a 2a0a 0a60 6060 0a77 6764  below**..```.wgd
+000128d0: 2076 697a 202d 6420 6b73 6461 7461 202d   viz -d ksdata -
+000128e0: 7370 2073 7064 6174 6120 2d2d 666f 6375  sp spdata --focu
+000128f0: 7332 616c 6c20 666f 6361 6c5f 7370 6563  s2all focal_spec
+00012900: 6965 7320 2d2d 6578 7472 6170 6172 616e  ies --extraparan
+00012910: 6f6d 656b 7320 6b73 6461 7461 0a60 6060  omeks ksdata.```
+00012920: 0a0a 2a2a 5468 6520 7669 7375 616c 697a  ..**The visualiz
+00012930: 6174 696f 6e20 6f66 2073 796e 7465 6e79  ation of synteny
+00012940: 2a2a 0a60 6060 0a77 6764 2076 697a 202d  **.```.wgd viz -
+00012950: 6170 2061 7064 6174 6120 2d73 6d20 736d  ap apdata -sm sm
+00012960: 6461 7461 202d 6d74 206d 7464 6174 6120  data -mt mtdata 
+00012970: 2d67 7420 6774 6461 7461 202d 2d70 6c6f  -gt gtdata --plo
+00012980: 7473 796e 2028 2d2d 6d69 6e6c 656e 202d  tsyn (--minlen -
+00012990: 3120 2d2d 6d69 6e73 6567 6c65 6e20 3130  1 --minseglen 10
+000129a0: 3030 3020 2d2d 6d69 6e67 656e 656e 756d  000 --mingenenum
+000129b0: 2033 3029 0a60 6060 0a0a 436f 6d70 6172   30).```..Compar
+000129c0: 6564 2074 6f20 7468 6520 6f72 6967 696e  ed to the origin
+000129d0: 616c 2060 7767 6460 2c20 7468 6520 6077  al `wgd`, the `w
+000129e0: 6764 2076 697a 6020 7072 6f67 7261 6d20  gd viz` program 
+000129f0: 6164 6465 6420 7379 6e74 656e 7920 7669  added synteny vi
+00012a00: 7375 616c 697a 6174 696f 6e20 7069 7065  sualization pipe
+00012a10: 6c69 6e65 2e20 5573 6572 7320 6e65 6564  line. Users need
+00012a20: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+00012a30: 666c 6167 206f 7074 696f 6e20 602d 2d70  flag option `--p
+00012a40: 6c6f 7473 796e 6020 746f 2069 6e69 7469  lotsyn` to initi
+00012a50: 6174 6520 7468 6973 2070 6172 7420 6f66  ate this part of
+00012a60: 2070 6970 656c 696e 652e 2054 6869 7320   pipeline. This 
+00012a70: 7374 6570 2061 7373 756d 6573 2074 6861  step assumes tha
+00012a80: 7420 7573 6572 7320 6861 7665 206f 6274  t users have obt
+00012a90: 6169 6e65 6420 616c 7265 6164 7920 7468  ained already th
+00012aa0: 6520 7379 6e74 656e 6963 2072 6573 756c  e syntenic resul
+00012ab0: 7473 2066 726f 6d20 6069 2d61 6468 6f72  ts from `i-adhor
+00012ac0: 6560 2061 6e64 2075 7365 7320 7468 6f73  e` and uses thos
+00012ad0: 6520 7265 7375 6c74 2066 696c 6573 2074  e result files t
+00012ae0: 6f20 7265 616c 697a 6520 7468 6520 7379  o realize the sy
+00012af0: 6e74 656e 7920 7669 7375 616c 697a 6174  nteny visualizat
+00012b00: 696f 6e2e 2053 696d 6c69 6172 2074 6f20  ion. Simliar to 
+00012b10: 6077 6764 2073 796e 602c 2074 6865 2065  `wgd syn`, the e
+00012b20: 7874 7261 202a 4b2a 3c73 7562 3e53 3c2f  xtra *K*<sub>S</
+00012b30: 7375 623e 2064 6174 6120 6669 6c65 2063  sub> data file c
+00012b40: 616e 2062 6520 7472 616e 736d 6974 7465  an be transmitte
+00012b50: 6420 7669 6120 7468 6520 6f70 7469 6f6e  d via the option
+00012b60: 2060 2d2d 6461 7461 6669 6c65 6020 2869   `--datafile` (i
+00012b70: 6e73 7465 6164 206f 6620 602d 2d65 7874  nstead of `--ext
+00012b80: 7261 7061 7261 6e6f 6d65 6b73 6020 6f70  raparanomeks` op
+00012b90: 7469 6f6e 292e 2042 6173 6963 616c 6c79  tion). Basically
+00012ba0: 2c20 7468 6520 7379 6e74 656e 6963 2072  , the syntenic r
+00012bb0: 6573 756c 7420 6669 6c65 7320 7265 7175  esult files requ
+00012bc0: 6972 6564 2061 7265 2074 6865 2061 6e63  ired are the anc
+00012bd0: 686f 7220 706f 696e 7473 2064 6174 6166  hor points dataf
+00012be0: 696c 652c 206d 756c 7469 706c 6963 6f6e  ile, multiplicon
+00012bf0: 7320 6461 7461 6669 6c65 2c20 6765 6e65  s datafile, gene
+00012c00: 2074 6162 6c65 2064 6174 6166 696c 6520   table datafile 
+00012c10: 2861 7574 6f6d 6174 656c 7920 7072 6f64  (automately prod
+00012c20: 7563 6564 2062 7920 6077 6764 2073 796e  uced by `wgd syn
+00012c30: 6029 2c20 616e 6420 7365 676d 656e 7473  `), and segments
+00012c40: 2064 6174 6166 696c 652e 0a0a 2a2a 4120   datafile...**A 
+00012c50: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
+00012c60: 6e67 2072 756e 2063 616e 2075 7365 2063  ng run can use c
+00012c70: 6f6d 6d61 6e64 2073 696d 706c 7920 6173  ommand simply as
+00012c80: 2062 656c 6f77 2a2a 0a0a 6060 600a 7767   below**..```.wg
+00012c90: 6420 7669 7a20 2d61 7020 6170 6461 7461  d viz -ap apdata
+00012ca0: 202d 736d 2073 6d64 6174 6120 2d6d 7420   -sm smdata -mt 
+00012cb0: 6d74 6461 7461 202d 6774 2067 7464 6174  mtdata -gt gtdat
+00012cc0: 6120 2d2d 706c 6f74 7379 6e0a 6060 600a  a --plotsyn.```.
+00012cd0: 0a23 2323 2077 6764 206d 6978 0a0a 2a2a  .### wgd mix..**
+00012ce0: 5468 6520 6d69 7874 7572 6520 6d6f 6465  The mixture mode
+00012cf0: 6c20 636c 7573 7465 7269 6e67 2061 6e61  l clustering ana
+00012d00: 6c79 7369 7320 6f66 202a 4b2a 3c73 7562  lysis of *K*<sub
+00012d10: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
+00012d20: 7472 6962 7574 696f 6e2a 2a0a 6060 600a  tribution**.```.
+00012d30: 7767 6420 6d69 7820 6b73 6461 7461 2028  wgd mix ksdata (
+00012d40: 2d2d 6e5f 696e 6974 2032 3030 202d 2d6d  --n_init 200 --m
+00012d50: 6178 5f69 7465 7220 3230 3020 2d2d 6b73  ax_iter 200 --ks
+00012d60: 5f72 616e 6765 2030 2035 202d 2d66 696c  _range 0 5 --fil
+00012d70: 7465 7273 2033 3030 202d 2d62 696e 7320  ters 300 --bins 
+00012d80: 3530 202d 2d63 6f6d 706f 6e65 6e74 7320  50 --components 
+00012d90: 3120 3420 2d2d 6761 6d6d 6120 302e 3030  1 4 --gamma 0.00
+00012da0: 3129 0a60 6060 0a0a 5468 6973 2070 6172  1).```..This par
+00012db0: 7420 6f66 2047 6175 7373 6961 6e20 6d69  t of Gaussian mi
+00012dc0: 7874 7572 6520 6d6f 6465 6c69 6e67 2028  xture modeling (
+00012dd0: 474d 4d29 2061 6e61 6c79 7369 7320 6973  GMM) analysis is
+00012de0: 2069 6e68 6572 6974 6564 2066 726f 6d20   inherited from 
+00012df0: 7468 6520 6f72 6967 696e 616c 2060 7767  the original `wg
+00012e00: 6460 2070 726f 6772 616d 2c20 6275 7420  d` program, but 
+00012e10: 7772 6974 6573 2061 6464 6974 696f 6e61  writes additiona
+00012e20: 6c6c 7920 7468 6520 7072 6f62 6162 696c  lly the probabil
+00012e30: 6974 7920 6f66 2065 6163 6820 2a4b 2a3c  ity of each *K*<
+00012e40: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+00012e50: 6520 696e 746f 2074 6865 2066 696e 616c  e into the final
+00012e60: 2064 6174 6166 7261 6d65 2e20 4261 7369   dataframe. Basi
+00012e70: 6361 6c6c 792c 2075 7365 7273 206e 6565  cally, users nee
+00012e80: 6420 746f 2070 726f 7669 6465 2077 6974  d to provide wit
+00012e90: 6820 6120 286e 6f72 6d61 6c6c 7920 6672  h a (normally fr
+00012ea0: 6f6d 2077 686f 6c65 2d70 6172 616e 6f6d  om whole-paranom
+00012eb0: 6520 6f72 2061 6e63 686f 722d 7061 6972  e or anchor-pair
+00012ec0: 7329 202a 4b2a 3c73 7562 3e53 3c2f 7375  s) *K*<sub>S</su
+00012ed0: 623e 2064 6174 6166 696c 6520 616e 6420  b> datafile and 
+00012ee0: 7468 6520 474d 4d20 616e 616c 7973 6973  the GMM analysis
+00012ef0: 2077 696c 6c20 6265 2063 6f6e 6475 6374   will be conduct
+00012f00: 6564 2075 706f 6e20 7468 6520 6461 7461  ed upon the data
+00012f10: 6669 6c65 2e20 536f 6d65 2070 6172 616d  file. Some param
+00012f20: 6574 6572 7320 6361 6e20 6166 6665 6374  eters can affect
+00012f30: 2074 6865 2072 6573 756c 7473 2c20 696e   the results, in
+00012f40: 636c 7564 696e 6720 602d 2d6e 5f69 6e69  cluding `--n_ini
+00012f50: 7460 2c20 7768 6963 6820 7365 7473 2074  t`, which sets t
+00012f60: 6865 206e 756d 6265 7220 6f66 206b 2d6d  he number of k-m
+00012f70: 6561 6e73 2069 6e69 7469 616c 697a 6174  eans initializat
+00012f80: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
+00012f90: 3029 2c20 602d 2d6d 6178 5f69 7465 7260  0), `--max_iter`
+00012fa0: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
+00012fb0: 206d 6178 696d 756d 206e 756d 6265 7220   maximum number 
+00012fc0: 6f66 2069 7465 7261 7469 6f6e 7320 2864  of iterations (d
+00012fd0: 6566 6175 6c74 2032 3030 292c 2060 2d2d  efault 200), `--
+00012fe0: 6d65 7468 6f64 602c 2077 6869 6368 2064  method`, which d
+00012ff0: 6574 6572 6d69 6e65 7320 7768 6963 6820  etermines which 
+00013000: 636c 7573 7465 7269 6e67 206d 6574 686f  clustering metho
+00013010: 6420 746f 2075 7365 2028 6465 6661 756c  d to use (defaul
+00013020: 7420 676d 6d29 2c20 602d 2d67 616d 6d61  t gmm), `--gamma
+00013030: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
+00013040: 6520 6761 6d6d 6120 7061 7261 6d65 7465  e gamma paramete
+00013050: 7220 666f 7220 7468 6520 6267 6d6d 206d  r for the bgmm m
+00013060: 6f64 656c 2028 6465 6661 756c 7420 302e  odel (default 0.
+00013070: 3030 3129 2c20 602d 2d63 6f6d 706f 6e65  001), `--compone
+00013080: 6e74 7360 2c20 7768 6963 6820 7365 7473  nts`, which sets
+00013090: 2074 6865 2072 616e 6765 206f 6620 7468   the range of th
+000130a0: 6520 6e75 6d62 6572 206f 6620 636f 6d70  e number of comp
+000130b0: 6f6e 656e 7473 2074 6f20 6669 7420 2864  onents to fit (d
+000130c0: 6566 6175 6c74 2031 2034 292c 2061 6e64  efault 1 4), and
+000130d0: 2074 6865 2064 6174 6120 6669 6c74 6572   the data filter
+000130e0: 696e 6720 7061 7261 6d65 7465 7273 2060  ing parameters `
+000130f0: 2d2d 6669 6c74 6572 7360 2077 6869 6368  --filters` which
+00013100: 2066 696c 7465 7273 2064 6174 6120 6261   filters data ba
+00013110: 7365 6420 6f6e 2061 6c69 676e 6d65 6e74  sed on alignment
+00013120: 206c 656e 6774 682c 2060 2d2d 6b73 5f72   length, `--ks_r
+00013130: 616e 6765 6020 7768 6963 6820 6669 6c74  ange` which filt
+00013140: 6572 7320 6461 7461 2062 6173 6564 206f  ers data based o
+00013150: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+00013160: 3e20 7661 6c75 6573 2061 6e64 2074 6865  > values and the
+00013170: 2070 6172 616d 6574 6572 2060 2d2d 6269   parameter `--bi
+00013180: 6e73 6020 7768 6963 6820 7365 7473 2074  ns` which sets t
+00013190: 6865 206e 756d 6265 7220 6f66 2062 696e  he number of bin
+000131a0: 7320 696e 202a 4b2a 3c73 7562 3e53 3c2f  s in *K*<sub>S</
+000131b0: 7375 623e 2064 6973 7472 6962 7574 696f  sub> distributio
+000131c0: 6e20 2864 6566 6175 6c74 2035 3029 2e0a  n (default 50)..
+000131d0: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+000131e0: 7461 7274 696e 6720 7275 6e20 6361 6e20  tarting run can 
+000131f0: 7573 6520 636f 6d6d 616e 6420 7369 6d70  use command simp
+00013200: 6c79 2061 7320 6265 6c6f 772a 2a0a 0a60  ly as below**..`
+00013210: 6060 0a77 6764 206d 6978 206b 7364 6174  ``.wgd mix ksdat
+00013220: 610a 6060 600a 0a23 2323 2077 6764 2070  a.```..### wgd p
+00013230: 6561 6b0a 0a2a 2a54 6865 2073 6561 7263  eak..**The searc
+00013240: 6820 6f66 2063 7265 6469 6162 6c65 202a  h of crediable *
+00013250: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2072  K*<sub>S</sub> r
+00013260: 616e 6765 2075 7365 6420 696e 2057 4744  ange used in WGD
+00013270: 2064 6174 696e 672a 2a0a 6060 600a 7767   dating**.```.wg
+00013280: 6420 7065 616b 206b 7364 6174 6120 2d61  d peak ksdata -a
+00013290: 7020 6170 6461 7461 202d 736d 2073 6d64  p apdata -sm smd
+000132a0: 6174 6120 2d6c 6520 6c65 6461 7461 202d  ata -le ledata -
+000132b0: 6d70 206d 7064 6174 6120 2d2d 6865 7572  mp mpdata --heur
+000132c0: 6973 7469 6320 282d 2d61 6c69 676e 6669  istic (--alignfi
+000132d0: 6c74 6572 2030 2e30 2030 2030 2e30 202d  lter 0.0 0 0.0 -
+000132e0: 2d6b 7372 616e 6765 2030 2035 202d 2d62  -ksrange 0 5 --b
+000132f0: 696e 5f77 6964 7468 2030 2e31 202d 2d67  in_width 0.1 --g
+00013300: 7569 6465 2073 6567 6d65 6e74 202d 2d70  uide segment --p
+00013310: 726f 6d69 6e65 6e63 655f 6375 746f 6666  rominence_cutoff
+00013320: 2030 2e31 202d 2d72 656c 5f68 6569 6768   0.1 --rel_heigh
+00013330: 7420 302e 3420 2d2d 6369 2039 3520 2d2d  t 0.4 --ci 95 --
+00013340: 6864 7220 3935 202d 2d6b 7363 7574 6f66  hdr 95 --kscutof
+00013350: 6620 3529 0a60 6060 0a0a 4173 206d 656e  f 5).```..As men
+00013360: 7469 6f6e 6564 2070 7265 7669 6f75 736c  tioned previousl
+00013370: 792c 2061 2068 6575 7269 7374 6963 206d  y, a heuristic m
+00013380: 6574 686f 6420 616e 6420 6120 636f 6c6c  ethod and a coll
+00013390: 696e 6561 7220 7365 676d 656e 7473 2d67  inear segments-g
+000133a0: 7569 6465 6420 616e 6368 6f72 2070 6169  uided anchor pai
+000133b0: 7220 636c 7573 7465 7269 6e67 2066 6f72  r clustering for
+000133c0: 2074 6865 2073 6561 7263 6820 6f66 2063   the search of c
+000133d0: 7265 6469 6162 6c65 202a 4b2a 3c73 7562  rediable *K*<sub
+000133e0: 3e53 3c2f 7375 623e 2072 616e 6765 2075  >S</sub> range u
+000133f0: 7365 6420 696e 2057 4744 2064 6174 696e  sed in WGD datin
+00013400: 6720 6172 6520 696d 706c 656d 656e 7465  g are implemente
+00013410: 6420 696e 2060 7767 6420 7632 602e 2055  d in `wgd v2`. U
+00013420: 7365 7273 206e 6565 6420 746f 2070 726f  sers need to pro
+00013430: 7669 6465 2074 6865 2061 6e63 686f 7220  vide the anchor 
+00013440: 706f 696e 7473 2c20 7365 676d 656e 7473  points, segments
+00013450: 2c20 6c69 7374 7365 676d 656e 7473 2c20  , listsegments, 
+00013460: 6d75 6c74 6970 6c69 636f 6e70 6169 7273  multipliconpairs
+00013470: 2064 6174 6166 696c 6520 6672 6f6d 2060   datafile from `
+00013480: 692d 6164 686f 7265 6020 746f 2061 6368  i-adhore` to ach
+00013490: 6965 7665 2074 6865 2063 6c75 7374 6572  ieve the cluster
+000134a0: 696e 6720 6675 6e63 7469 6f6e 2e20 536f  ing function. So
+000134b0: 6d65 2070 6172 616d 6574 6572 7320 7468  me parameters th
+000134c0: 6174 2063 616e 2069 6d70 6163 7420 7468  at can impact th
+000134d0: 6520 7265 7375 6c74 7320 696e 636c 7564  e results includ
+000134e0: 6520 602d 2d61 6c69 676e 6669 6c74 6572  e `--alignfilter
+000134f0: 602c 2077 6869 6368 2066 696c 7465 7273  `, which filters
+00013500: 2074 6865 2064 6174 6120 6261 7365 6420   the data based 
+00013510: 6f6e 2061 6c69 676e 6d65 6e74 2069 6465  on alignment ide
+00013520: 6e74 6974 792c 206c 656e 6774 6820 616e  ntity, length an
+00013530: 6420 636f 7665 7261 6765 2c20 602d 2d6b  d coverage, `--k
+00013540: 7372 616e 6765 602c 2077 6869 6368 2073  srange`, which s
+00013550: 6574 7320 7468 6520 7261 6e67 6520 6f66  ets the range of
+00013560: 204b 7320 746f 2062 6520 616e 616c 797a   Ks to be analyz
+00013570: 6564 2c20 602d 2d62 696e 5f77 6964 7468  ed, `--bin_width
+00013580: 602c 2077 6869 6368 2073 6574 7320 7468  `, which sets th
+00013590: 6520 6261 6e64 7769 6474 6820 6f66 202a  e bandwidth of *
+000135a0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
+000135b0: 6973 7472 6962 7574 696f 6e2c 2060 2d2d  istribution, `--
+000135c0: 7765 6967 6874 735f 6f75 746c 6965 7273  weights_outliers
+000135d0: 5f69 6e63 6c75 6465 6460 2077 6869 6368  _included` which
+000135e0: 2064 6574 6572 6d69 6e65 7320 7768 6574   determines whet
+000135f0: 6865 7220 746f 2069 6e63 6c75 6465 202a  her to include *
+00013600: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206f  K*<sub>S</sub> o
+00013610: 7574 6c69 6572 7320 2877 686f 7365 2076  utliers (whose v
+00013620: 616c 7565 2069 7320 6f76 6572 2035 2920  alue is over 5) 
+00013630: 696e 2061 6e61 6c79 7369 732c 2060 2d2d  in analysis, `--
+00013640: 6d65 7468 6f64 6020 7768 6963 6820 6465  method` which de
+00013650: 7465 726d 696e 6573 2077 6869 6368 2063  termines which c
+00013660: 6c75 7374 6572 696e 6720 6d65 7468 6f64  lustering method
+00013670: 2074 6f20 7573 6520 2864 6566 6175 6c74   to use (default
+00013680: 2067 6d6d 292c 2060 2d2d 7365 6564 6020   gmm), `--seed` 
+00013690: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
+000136a0: 616e 646f 6d20 7365 6564 2067 6976 656e  andom seed given
+000136b0: 2074 6f20 696e 6974 6961 6c69 7a61 7469   to initializati
+000136c0: 6f6e 2028 6465 6661 756c 7420 3233 3532  on (default 2352
+000136d0: 3839 3029 2c20 602d 2d6e 5f69 6e69 7460  890), `--n_init`
+000136e0: 2c20 7768 6963 6820 7365 7473 2074 6865  , which sets the
+000136f0: 206e 756d 6265 7220 6f66 206b 2d6d 6561   number of k-mea
+00013700: 6e73 2069 6e69 7469 616c 697a 6174 696f  ns initializatio
+00013710: 6e73 2028 6465 6661 756c 7420 3230 3029  ns (default 200)
+00013720: 2c20 602d 2d65 6d5f 6974 6572 602c 2077  , `--em_iter`, w
+00013730: 6869 6368 2073 6574 7320 7468 6520 6d61  hich sets the ma
+00013740: 7869 6d75 6d20 6e75 6d62 6572 206f 6620  ximum number of 
+00013750: 6974 6572 6174 696f 6e73 2028 6465 6661  iterations (defa
+00013760: 756c 7420 3230 3029 2c20 602d 2d67 616d  ult 200), `--gam
+00013770: 6d61 602c 2077 6869 6368 2073 6574 7320  ma`, which sets 
+00013780: 7468 6520 6761 6d6d 6120 7061 7261 6d65  the gamma parame
+00013790: 7465 7220 666f 7220 7468 6520 6267 6d6d  ter for the bgmm
+000137a0: 206d 6f64 656c 2028 6465 6661 756c 7420   model (default 
+000137b0: 302e 3030 3129 2c20 602d 2d63 6f6d 706f  0.001), `--compo
+000137c0: 6e65 6e74 7360 2c20 7768 6963 6820 7365  nents`, which se
+000137d0: 7473 2074 6865 2072 616e 6765 206f 6620  ts the range of 
+000137e0: 7468 6520 6e75 6d62 6572 206f 6620 636f  the number of co
+000137f0: 6d70 6f6e 656e 7473 2074 6f20 6669 7420  mponents to fit 
+00013800: 2864 6566 6175 6c74 2031 2034 292c 2060  (default 1 4), `
+00013810: 2d2d 7765 6967 6874 6564 6020 7768 6963  --weighted` whic
+00013820: 6820 6465 7465 726d 696e 6573 2077 6865  h determines whe
+00013830: 7468 6572 2074 6f20 7573 6520 6e6f 6465  ther to use node
+00013840: 2d77 6569 6768 7465 6420 6d65 7468 6f64  -weighted method
+00013850: 2066 6f72 2064 652d 7265 6475 6e64 616e   for de-redundan
+00013860: 6379 2c20 602d 2d67 7569 6465 6020 7768  cy, `--guide` wh
+00013870: 6963 6820 6465 7465 726d 696e 6573 2077  ich determines w
+00013880: 6869 6368 2072 6567 696d 6520 7265 7369  hich regime resi
+00013890: 6469 6e67 2061 6e63 686f 7273 2074 6f20  ding anchors to 
+000138a0: 6265 2075 7365 6420 2864 6566 6175 6c74  be used (default
+000138b0: 2053 6567 6d65 6e74 292c 2060 2d2d 7072   Segment), `--pr
+000138c0: 6f6d 696e 656e 6365 5f63 7574 6f66 6660  ominence_cutoff`
+000138d0: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
+000138e0: 7072 6f6d 696e 656e 6365 2063 7574 6f66  prominence cutof
+000138f0: 6620 6f66 2061 6363 6570 7461 626c 6520  f of acceptable 
+00013900: 7065 616b 7320 696e 2070 6561 6b20 6669  peaks in peak fi
+00013910: 6e64 696e 6720 7072 6f63 6573 732c 2060  nding process, `
+00013920: 2d2d 7265 6c5f 6865 6967 6874 6020 7768  --rel_height` wh
+00013930: 6963 6820 7365 7473 2074 6865 2072 656c  ich sets the rel
+00013940: 6174 6976 6520 6865 6967 6874 2061 7420  ative height at 
+00013950: 7768 6963 6820 7468 6520 7065 616b 2077  which the peak w
+00013960: 6964 7468 2069 7320 6d65 6173 7572 6564  idth is measured
+00013970: 2c20 602d 2d6b 7374 6f64 6174 6560 2077  , `--kstodate` w
+00013980: 6869 6368 206d 616e 7561 6c6c 7920 7365  hich manually se
+00013990: 7473 2074 6865 2072 616e 6765 206f 6620  ts the range of 
+000139a0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000139b0: 746f 2062 6520 6461 7465 6420 696e 2068  to be dated in h
+000139c0: 6575 7269 7374 6963 2073 6561 7263 6820  euristic search 
+000139d0: 616e 6420 6e65 6564 7320 746f 2062 6520  and needs to be 
+000139e0: 636f 2d73 6574 2077 6974 6820 6f70 7469  co-set with opti
+000139f0: 6f6e 2060 2d2d 6d61 6e75 616c 7365 7460  on `--manualset`
+00013a00: 2c20 602d 2d78 6c69 6d60 2061 6e64 2060  , `--xlim` and `
+00013a10: 2d2d 796c 696d 6020 6465 7465 726d 696e  --ylim` determin
+00013a20: 696e 6720 7468 6520 7820 616e 6420 7920  ing the x and y 
+00013a30: 6178 6973 206c 696d 6974 206f 6620 474d  axis limit of GM
+00013a40: 4d20 2a4b 2a3c 7375 623e 533c 2f73 7562  M *K*<sub>S</sub
+00013a50: 3e20 6469 7374 7269 6275 7469 6f6e 2c20  > distribution, 
+00013a60: 602d 2d63 6960 2073 6574 7469 6e67 2074  `--ci` setting t
+00013a70: 6865 2063 6f6e 6669 6465 6e63 6520 6c65  he confidence le
+00013a80: 7665 6c20 6f66 206c 6f67 2d6e 6f72 6d61  vel of log-norma
+00013a90: 6c20 6469 7374 7269 6275 7469 6f6e 2074  l distribution t
+00013aa0: 6f20 6461 7465 2028 6465 6661 756c 7420  o date (default 
+00013ab0: 3935 292c 2060 2d2d 6864 7260 2073 6574  95), `--hdr` set
+00013ac0: 7469 6e67 2074 6865 2068 6967 6865 7374  ting the highest
+00013ad0: 2064 656e 7369 7479 2072 6567 696f 6e20   density region 
+00013ae0: 2848 4452 2920 6170 706c 6965 6420 696e  (HDR) applied in
+00013af0: 2074 6865 2073 6567 6d65 6e74 2d67 7569   the segment-gui
+00013b00: 6465 6420 616e 6368 6f72 2070 6169 7220  ded anchor pair 
+00013b10: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00013b20: 6469 7374 7269 6275 7469 6f6e 2028 6465  distribution (de
+00013b30: 6661 756c 7420 3935 292c 2060 2d2d 6865  fault 95), `--he
+00013b40: 7572 6973 7469 6360 2064 6574 6572 6d69  uristic` determi
+00013b50: 6e69 6e67 2077 6865 7468 6572 2074 6f20  ning whether to 
+00013b60: 696e 6974 6961 7465 2068 6575 7269 7374  initiate heurist
+00013b70: 6963 206d 6574 686f 6420 6f66 2064 6566  ic method of def
+00013b80: 696e 696e 6720 4349 2066 6f72 2064 6174  ining CI for dat
+00013b90: 696e 672c 2060 2d2d 6b73 6375 746f 6666  ing, `--kscutoff
+00013ba0: 6020 7365 7474 696e 6720 7468 6520 2a4b  ` setting the *K
+00013bb0: 2a3c 7375 623e 533c 2f73 7562 3e20 7361  *<sub>S</sub> sa
+00013bc0: 7475 7261 7469 6f6e 2063 7574 6f66 6620  turation cutoff 
+00013bd0: 696e 2064 6174 696e 6720 2864 6566 6175  in dating (defau
+00013be0: 6c74 2035 292e 0a0a 2a2a 4120 7375 6767  lt 5)...**A sugg
+00013bf0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
+00013c00: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
+00013c10: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
+00013c20: 6f77 2a2a 0a0a 6060 600a 7767 6420 7065  ow**..```.wgd pe
+00013c30: 616b 206b 7364 6174 6120 2d61 7020 6170  ak ksdata -ap ap
+00013c40: 6461 7461 202d 736d 2073 6d64 6174 6120  data -sm smdata 
+00013c50: 2d6c 6520 6c65 6461 7461 202d 6d70 206d  -le ledata -mp m
+00013c60: 7064 6174 6120 2d2d 6865 7572 6973 7469  pdata --heuristi
+00013c70: 630a 6060 600a 0a23 2323 2077 6764 2066  c.```..### wgd f
+00013c80: 6f63 7573 0a0a 2a2a 5468 6520 636f 6e63  ocus..**The conc
+00013c90: 6174 656e 6174 696f 6e2d 6261 7365 642f  atenation-based/
+00013ca0: 636f 616c 6573 6365 6e63 652d 6261 7365  coalescence-base
+00013cb0: 6420 7068 796c 6f67 656e 6574 6963 2069  d phylogenetic i
+00013cc0: 6e66 6572 656e 6365 2a2a 0a60 6060 0a77  nference**.```.w
+00013cd0: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
+00013ce0: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
+00013cf0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+00013d00: 282d 2d63 6f6e 6361 7465 6e61 7469 6f6e  (--concatenation
+00013d10: 2920 282d 2d63 6f61 6c65 7363 656e 6365  ) (--coalescence
+00013d20: 2920 282d 7472 6565 2027 6661 7374 7472  ) (-tree 'fasttr
+00013d30: 6565 2729 2028 2d74 7320 272d 6661 7374  ee') (-ts '-fast
+00013d40: 6573 7427 2920 282d 6e20 3429 2028 2d2d  est') (-n 4) (--
+00013d50: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
+00013d60: 2028 2d6f 2077 6764 5f66 6f63 7573 2920   (-o wgd_focus) 
+00013d70: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
+00013d80: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+00013d90: 6d20 6077 6764 2066 6f63 7573 6020 696d  m `wgd focus` im
+00013da0: 706c 656d 656e 7465 6420 7477 6f20 6261  plemented two ba
+00013db0: 7369 6320 7068 796c 6f67 656e 6574 6963  sic phylogenetic
+00013dc0: 2069 6e66 6572 656e 6365 206d 6574 686f   inference metho
+00013dd0: 6473 2c20 692e 652e 2c20 636f 6e63 6174  ds, i.e., concat
+00013de0: 656e 6174 696f 6e2d 6261 7365 6420 616e  enation-based an
+00013df0: 6420 636f 616c 6573 6365 6e63 652d 6261  d coalescence-ba
+00013e00: 7365 6420 6d65 7468 6f64 732e 2054 6f20  sed methods. To 
+00013e10: 696e 6974 6961 7465 2074 6865 7365 2061  initiate these a
+00013e20: 6e61 6c79 7369 732c 2075 7365 7273 206e  nalysis, users n
+00013e30: 6565 6420 746f 2073 6574 2074 6865 2066  eed to set the f
+00013e40: 6c61 6720 6f70 7469 6f6e 2060 2d2d 636f  lag option `--co
+00013e50: 6e63 6174 656e 6174 696f 6e60 206f 7220  ncatenation` or 
+00013e60: 602d 2d63 6f61 6c65 7363 656e 6365 602e  `--coalescence`.
+00013e70: 2054 6865 2063 6f6e 6361 7465 6e61 7469   The concatenati
+00013e80: 6f6e 2d62 6173 6564 206d 6574 686f 6420  on-based method 
+00013e90: 696e 636c 7564 6573 2061 2066 6577 206d  includes a few m
+00013ea0: 616a 6f72 2073 7465 7073 2c20 692e 652e  ajor steps, i.e.
+00013eb0: 2c20 7468 6520 6d75 6c74 6970 6c65 2073  , the multiple s
+00013ec0: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
+00013ed0: 7420 284d 5341 2920 6f66 2065 6163 6820  t (MSA) of each 
+00013ee0: 6765 6e65 2066 616d 696c 792c 2074 6865  gene family, the
+00013ef0: 2063 6f6e 6361 7465 6e61 7469 6f6e 206f   concatenation o
+00013f00: 6620 616c 6c20 6765 6e65 2066 616d 696c  f all gene famil
+00013f10: 6965 7320 616e 6420 7468 656e 2074 6865  ies and then the
+00013f20: 2067 656e 6520 7472 6565 2028 616c 736f   gene tree (also
+00013f30: 2073 7065 6369 6573 2074 7265 6520 696e   species tree in
+00013f40: 2074 6869 7320 6361 7365 2920 696e 6665   this case) infe
+00013f50: 7265 6e63 652e 2054 6865 2063 6f61 6c65  rence. The coale
+00013f60: 7363 656e 6365 2d62 6173 6564 206d 6574  scence-based met
+00013f70: 686f 6420 7769 6c6c 2069 6e73 7465 6164  hod will instead
+00013f80: 2070 6572 666f 726d 2074 6865 204d 5341   perform the MSA
+00013f90: 206f 6620 6561 6368 2067 656e 6520 6661   of each gene fa
+00013fa0: 6d69 6c79 2061 6e64 2074 6865 2067 656e  mily and the gen
+00013fb0: 6520 7472 6565 2069 6e66 6572 656e 6365  e tree inference
+00013fc0: 2062 6173 6564 206f 6e20 6561 6368 204d   based on each M
+00013fd0: 5341 2c20 616e 6420 7468 656e 2069 6e66  SA, and then inf
+00013fe0: 6572 2074 6865 2073 7065 6369 6573 2074  er the species t
+00013ff0: 7265 6520 6261 7365 6420 6f6e 2074 6865  ree based on the
+00014000: 7365 2069 6e64 6976 6964 7561 6c20 6765  se individual ge
+00014010: 6e65 2074 7265 6573 2e20 5468 6520 7472  ne trees. The tr
+00014020: 6565 2d69 6e66 6572 656e 6365 2070 726f  ee-inference pro
+00014030: 6772 616d 2061 6e64 2074 6865 2061 7373  gram and the ass
+00014040: 6f63 6961 7465 6420 7061 7261 6d65 7465  ociated paramete
+00014050: 7273 2063 616e 2062 6520 7365 7420 6a75  rs can be set ju
+00014060: 7374 2061 7320 6162 6f76 6520 6279 2060  st as above by `
+00014070: 2d74 7265 6560 206f 7220 602d 2d74 7265  -tree` or `--tre
+00014080: 655f 6d65 7468 6f64 6020 616e 6420 602d  e_method` and `-
+00014090: 7473 6020 6f72 2060 2d2d 7472 6565 7365  ts` or `--treese
+000140a0: 7460 2e20 506c 6561 7365 2061 6c73 6f20  t`. Please also 
+000140b0: 6d61 6b65 2073 7572 6520 7468 6520 6368  make sure the ch
+000140c0: 6f73 656e 2074 7265 652d 696e 6665 7265  osen tree-infere
+000140d0: 6e63 6520 7072 6f67 7261 6d20 6973 2069  nce program is i
+000140e0: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+000140f0: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+00014100: 2e20 5468 6520 7072 6f67 7261 6d20 6061  . The program `a
+00014110: 7374 7261 6c2d 7072 6f60 2069 7320 7265  stral-pro` is re
+00014120: 7175 6972 6564 2074 6f20 6265 2069 6e73  quired to be ins
+00014130: 7461 6c6c 6564 2069 6e20 7468 6520 656e  talled in the en
+00014140: 7669 726f 6e6d 656e 7420 7061 7468 2069  vironment path i
+00014150: 6620 7468 6520 636f 616c 6573 6365 6e63  f the coalescenc
+00014160: 652d 6261 7365 6420 6d65 7468 6f64 2069  e-based method i
+00014170: 7320 6368 6f73 656e 2e20 5468 6520 7061  s chosen. The pa
+00014180: 7261 6c6c 656c 2074 6872 6561 6473 2068  rallel threads h
+00014190: 6572 6520 6172 6520 746f 2061 6363 656c  ere are to accel
+000141a0: 6572 6174 6520 7468 6520 7365 7175 656e  erate the sequen
+000141b0: 6365 2061 6c69 676e 6d65 6e74 2061 6e64  ce alignment and
+000141c0: 2067 656e 6520 7472 6565 2069 6e66 6572   gene tree infer
+000141d0: 656e 6365 2066 6f72 2065 6163 6820 6765  ence for each ge
+000141e0: 6e65 2066 616d 696c 7920 746f 6f20 616e  ne family too an
+000141f0: 6420 7468 7573 2073 7567 6765 7374 6564  d thus suggested
+00014200: 2074 6f20 6265 2073 6574 2061 7320 6d75   to be set as mu
+00014210: 6368 2061 7320 7468 6520 6e75 6d62 6572  ch as the number
+00014220: 206f 6620 6765 6e65 2066 616d 696c 6965   of gene familie
+00014230: 732e 0a0a 2a2a 4120 7375 6767 6573 7465  s...**A suggeste
+00014240: 6420 7374 6172 7469 6e67 2072 756e 206f  d starting run o
+00014250: 6620 7468 6973 2061 6e61 6c79 7369 7320  f this analysis 
+00014260: 6973 2077 6974 6820 7468 6520 7369 6d70  is with the simp
+00014270: 6c65 2063 6f6d 6d61 6e64 2062 656c 6f77  le command below
+00014280: 2e2a 2a0a 0a60 6060 0a77 6764 2066 6f63  .**..```.wgd foc
+00014290: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
+000142a0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+000142b0: 7365 7175 656e 6365 3320 2d2d 636f 6e63  sequence3 --conc
+000142c0: 6174 656e 6174 696f 6e0a 7767 6420 666f  atenation.wgd fo
+000142d0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+000142e0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+000142f0: 2073 6571 7565 6e63 6533 202d 2d63 6f61   sequence3 --coa
+00014300: 6c65 7363 656e 6365 0a60 6060 0a0a 2a2a  lescence.```..**
+00014310: 5468 6520 6675 6e63 7469 6f6e 616c 2061  The functional a
+00014320: 6e6e 6f74 6174 696f 6e20 6f66 2067 656e  nnotation of gen
+00014330: 6520 6661 6d69 6c69 6573 2a2a 0a60 6060  e families**.```
+00014340: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+00014350: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+00014360: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+00014370: 3320 2d2d 616e 6e6f 7461 7469 6f6e 2065  3 --annotation e
+00014380: 6767 6e6f 6720 2d65 6420 6564 6461 7461  ggnog -ed eddata
+00014390: 202d 2d64 6d6e 6220 6462 6461 7461 0a60   --dmnb dbdata.`
+000143a0: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
+000143b0: 6077 6764 2066 6f63 7573 6020 616c 736f  `wgd focus` also
+000143c0: 2061 6464 6564 2073 6f6d 6520 7772 6170   added some wrap
+000143d0: 7069 6e67 2066 756e 6374 696f 6e73 2066  ping functions f
+000143e0: 6f72 2066 756e 6374 696f 6e61 6c20 616e  or functional an
+000143f0: 6e6f 7461 7469 6f6e 206f 6620 6765 6e65  notation of gene
+00014400: 2066 616d 696c 6965 7320 6f6e 2074 6865   families on the
+00014410: 2068 6f6f 6420 6f66 2064 6174 6162 6173   hood of databas
+00014420: 6573 2061 6e64 2073 6f66 7477 6172 6573  es and softwares
+00014430: 2066 6f72 2069 6e73 7461 6e63 6520 6045   for instance `E
+00014440: 6767 4e4f 4760 2061 6e64 2060 4567 674e  ggNOG` and `EggN
+00014450: 4f47 2d6d 6170 7065 7260 2e20 466f 7220  OG-mapper`. For 
+00014460: 7468 6520 616e 6e6f 7461 7469 6f6e 2075  the annotation u
+00014470: 7369 6e67 2060 4567 674e 4f47 2d6d 6170  sing `EggNOG-map
+00014480: 7065 7260 2c20 7573 6572 7320 6e65 6564  per`, users need
+00014490: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
+000144a0: 7061 7468 2074 6f20 7468 6520 6567 674e  path to the eggN
+000144b0: 4f47 2061 6e6e 6f74 6174 696f 6e20 6461  OG annotation da
+000144c0: 7461 6261 7365 2076 6961 2074 6865 206f  tabase via the o
+000144d0: 7074 696f 6e20 602d 6564 602c 2074 6865  ption `-ed`, the
+000144e0: 2070 6174 6820 746f 2074 6865 2064 6961   path to the dia
+000144f0: 6d6f 6e64 2d63 6f6d 7061 7469 626c 6520  mond-compatible 
+00014500: 6461 7461 6261 7365 2076 6961 2074 6865  database via the
+00014510: 206f 7074 696f 6e20 602d 2d64 6d6e 645f   option `--dmnd_
+00014520: 6462 6020 616e 6420 7468 6520 6f70 7469  db` and the opti
+00014530: 6f6e 2060 2d2d 616e 6e6f 7461 7469 6f6e  on `--annotation
+00014540: 6020 7365 7420 6173 2022 6567 676e 6f67  ` set as "eggnog
+00014550: 222e 2054 6865 206d 616e 6e65 7220 686f  ". The manner ho
+00014560: 7720 5046 414d 2061 6e6e 6f74 6174 696f  w PFAM annotatio
+00014570: 6e20 7769 6c6c 2062 6520 7065 7266 6f72  n will be perfor
+00014580: 6d65 6420 6361 6e20 6265 2063 6f6e 7472  med can be contr
+00014590: 6f6c 6c65 6420 7669 6120 7468 6520 6f70  olled via the op
+000145a0: 7469 6f6e 2060 7066 616d 602c 2065 6974  tion `pfam`, eit
+000145b0: 6865 7220 226e 6f6e 6522 2c20 2272 6561  her "none", "rea
+000145c0: 6c69 676e 2220 6f72 2022 6465 6e6f 766f  lign" or "denovo
+000145d0: 222c 2074 6865 2064 6574 6169 6c65 6420  ", the detailed 
+000145e0: 6578 706c 616e 6174 696f 6e20 6361 6e20  explanation can 
+000145f0: 6265 2066 6f75 6e64 2061 7420 7468 6520  be found at the 
+00014600: 7769 6b69 206f 6620 5b45 6767 4e4f 472d  wiki of [EggNOG-
+00014610: 6d61 7070 6572 5d28 6874 7470 733a 2f2f  mapper](https://
+00014620: 6769 7468 7562 2e63 6f6d 2f65 6767 6e6f  github.com/eggno
+00014630: 6764 622f 6567 676e 6f67 2d6d 6170 7065  gdb/eggnog-mappe
+00014640: 722f 7769 6b69 292e 2050 6c65 6173 6520  r/wiki). Please 
+00014650: 7072 652d 696e 7374 616c 6c20 7468 6520  pre-install the 
+00014660: 6045 6767 4e4f 472d 6d61 7070 6572 6020  `EggNOG-mapper` 
+00014670: 7079 7468 6f6e 2070 6163 6b61 6765 2069  python package i
+00014680: 6620 7573 696e 6720 7468 6973 2066 756e  f using this fun
+00014690: 6374 696f 6e2e 2046 6f72 2074 6865 2061  ction. For the a
+000146a0: 6e6e 6f74 6174 696f 6e20 7573 696e 6720  nnotation using 
+000146b0: 6068 6d6d 7363 616e 602c 2077 6861 7420  `hmmscan`, what 
+000146c0: 6973 2069 6d70 6c65 6d65 6e74 6564 2069  is implemented i
+000146d0: 6e20 6077 6764 2076 3260 2069 7320 6120  n `wgd v2` is a 
+000146e0: 7369 6d70 6c65 2062 756e 646c 6520 6675  simple bundle fu
+000146f0: 6e63 7469 6f6e 2074 6f20 7065 7266 6f72  nction to perfor
+00014700: 6d20 6068 6d6d 7363 616e 6020 616e 616c  m `hmmscan` anal
+00014710: 7973 6973 2066 6f72 2061 2067 6976 656e  ysis for a given
+00014720: 2068 6d6d 2070 726f 6669 6c65 2061 6e64   hmm profile and
+00014730: 2073 6574 206f 6620 6765 6e65 2066 616d   set of gene fam
+00014740: 696c 6965 7320 7375 6368 2074 6861 7420  ilies such that 
+00014750: 7573 6572 7320 6e65 6564 2074 6f20 7365  users need to se
+00014760: 7420 7468 6520 6f70 7469 6f6e 2060 2d2d  t the option `--
+00014770: 616e 6e6f 7461 7469 6f6e 6020 6173 2022  annotation` as "
+00014780: 686d 6d70 6661 6d22 2061 6e64 2070 726f  hmmpfam" and pro
+00014790: 7669 6465 2074 6865 2068 6d6d 7072 6f66  vide the hmmprof
+000147a0: 696c 6520 7669 6120 7468 6520 6f70 7469  ile via the opti
+000147b0: 6f6e 2060 2d2d 686d 6d60 2e20 466f 7220  on `--hmm`. For 
+000147c0: 7468 6520 616e 6e6f 7461 7469 6f6e 2075  the annotation u
+000147d0: 7369 6e67 2060 696e 7465 7270 726f 7363  sing `interprosc
+000147e0: 616e 602c 2075 7365 7273 206e 6565 6420  an`, users need 
+000147f0: 746f 2070 726f 7669 6465 2074 6865 2070  to provide the p
+00014800: 6174 6820 746f 2074 6865 2069 6e74 6572  ath to the inter
+00014810: 7072 6f73 6361 6e20 696e 7374 616c 6c61  proscan installa
+00014820: 7469 6f6e 2066 6f6c 6465 7220 7768 6572  tion folder wher
+00014830: 6520 7468 6572 6520 6973 2061 2060 696e  e there is a `in
+00014840: 7465 7270 726f 7363 616e 2e73 6860 2066  terproscan.sh` f
+00014850: 696c 6520 7669 6120 7468 6520 6f70 7469  ile via the opti
+00014860: 6f6e 2060 2d2d 6578 6570 6174 6860 2061  on `--exepath` a
+00014870: 6e64 2073 6574 2074 6865 2074 6865 206f  nd set the the o
+00014880: 7074 696f 6e20 602d 2d61 6e6e 6f74 6174  ption `--annotat
+00014890: 696f 6e60 2061 7320 2269 6e74 6572 7072  ion` as "interpr
+000148a0: 6f73 6361 6e22 2e20 5468 6520 7061 7261  oscan". The para
+000148b0: 6c6c 656c 2074 6872 6561 6473 2068 6572  llel threads her
+000148c0: 6520 6172 6520 746f 2070 6172 616c 6c65  e are to paralle
+000148d0: 6c69 7a65 2074 6865 2061 6e61 6c79 7369  lize the analysi
+000148e0: 7320 666f 7220 6561 6368 2067 656e 6520  s for each gene 
+000148f0: 6661 6d69 6c79 2061 6e64 2074 6875 7320  family and thus 
+00014900: 7375 6767 6573 7465 6420 746f 2062 6520  suggested to be 
+00014910: 7365 7420 6173 206d 7563 6820 6173 2074  set as much as t
+00014920: 6865 206e 756d 6265 7220 6f66 2067 656e  he number of gen
+00014930: 6520 6661 6d69 6c69 6573 2e0a 0a2a 2a41  e families...**A
+00014940: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
+00014950: 696e 6720 7275 6e20 6f66 2074 6869 7320  ing run of this 
+00014960: 616e 616c 7973 6973 2063 616e 2062 6520  analysis can be 
+00014970: 7769 7468 2074 6865 2063 6f6d 6d61 6e64  with the command
+00014980: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
+00014990: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
+000149a0: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
+000149b0: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+000149c0: 2d2d 616e 6e6f 7461 7469 6f6e 2065 6767  --annotation egg
+000149d0: 6e6f 6720 2d65 6420 6564 6461 7461 202d  nog -ed eddata -
+000149e0: 2d64 6d6e 6220 6462 6461 7461 0a77 6764  -dmnb dbdata.wgd
+000149f0: 2066 6f63 7573 2066 616d 696c 6965 7320   focus families 
+00014a00: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+00014a10: 6365 3220 7365 7175 656e 6365 3320 2d2d  ce2 sequence3 --
+00014a20: 616e 6e6f 7461 7469 6f6e 2068 6d6d 7066  annotation hmmpf
+00014a30: 616d 202d 2d68 6d6d 2068 6d6d 6461 7461  am --hmm hmmdata
+00014a40: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+00014a50: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+00014a60: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+00014a70: 3320 2d2d 616e 6e6f 7461 7469 6f6e 2069  3 --annotation i
+00014a80: 6e74 6572 7072 6f73 6361 6e20 2d2d 6578  nterproscan --ex
+00014a90: 6570 6174 6820 2450 4154 480a 6060 600a  epath $PATH.```.
+00014aa0: 0a2a 2a54 6865 2070 6879 6c6f 6765 6e65  .**The phylogene
+00014ab0: 7469 6320 6461 7469 6e67 206f 6620 5747  tic dating of WG
+00014ac0: 4473 2a2a 0a60 6060 0a77 6764 2066 6f63  Ds**.```.wgd foc
+00014ad0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
+00014ae0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
+00014af0: 7365 7175 656e 6365 3320 2d64 206d 636d  sequence3 -d mcm
+00014b00: 6374 7265 6520 2d73 7020 7370 6461 7461  ctree -sp spdata
+00014b10: 2028 2d2d 7072 6f74 636f 6364 6174 696e   (--protcocdatin
+00014b20: 6729 2028 2d2d 7061 7274 6974 696f 6e29  g) (--partition)
+00014b30: 2028 2d2d 6161 6d6f 6465 6c20 6c67 2920   (--aamodel lg) 
+00014b40: 282d 6473 2027 6275 726e 696e 203d 2032  (-ds 'burnin = 2
+00014b50: 3030 3027 2920 282d 6473 2027 7361 6d70  000') (-ds 'samp
+00014b60: 6672 6571 203d 2031 3030 3027 2920 282d  freq = 1000') (-
+00014b70: 6473 2027 6e73 616d 706c 6520 3d20 3230  ds 'nsample = 20
+00014b80: 3030 3027 2920 2028 2d6e 2034 2920 282d  000')  (-n 4) (-
+00014b90: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
+00014ba0: 2920 282d 6f20 7767 645f 666f 6375 7329  ) (-o wgd_focus)
+00014bb0: 2028 2d74 2077 6f72 6b69 6e67 5f74 6d70   (-t working_tmp
+00014bc0: 290a 6060 600a 0a54 6865 2061 6273 6f6c  ).```..The absol
+00014bd0: 7574 6520 6461 7469 6e67 206f 6620 5747  ute dating of WG
+00014be0: 4473 2069 7320 6120 7370 6563 6966 6963  Ds is a specific
+00014bf0: 2070 6970 656c 696e 6520 696d 706c 656d   pipeline implem
+00014c00: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
+00014c10: 6020 7573 696e 6720 7468 6520 6d65 7468  ` using the meth
+00014c20: 6f64 206f 6620 7068 796c 6f67 656e 6574  od of phylogenet
+00014c30: 6963 2064 6174 696e 672e 2054 6865 2066  ic dating. The f
+00014c40: 616d 696c 6965 7320 7573 6564 2069 6e20  amilies used in 
+00014c50: 7468 6973 2073 7465 7020 6361 6e20 6265  this step can be
+00014c60: 2070 726f 6475 6365 6420 6672 6f6d 2060   produced from `
+00014c70: 7767 6420 646d 6460 2061 6e64 2060 7767  wgd dmd` and `wg
+00014c80: 6420 7065 616b 602e 204e 6f74 6520 7468  d peak`. Note th
+00014c90: 6174 2068 6572 6520 7765 206f 6e6c 7920  at here we only 
+00014ca0: 6469 7363 7573 7320 686f 7720 746f 2064  discuss how to d
+00014cb0: 6174 6520 5747 4473 2077 6974 6820 6765  ate WGDs with ge
+00014cc0: 6e6f 6d65 2061 7373 656d 626c 792c 2069  nome assembly, i
+00014cd0: 6e73 7465 6164 206f 6620 7472 616e 7363  nstead of transc
+00014ce0: 7269 7074 6f6d 6520 6173 7365 6d62 6c79  riptome assembly
+00014cf0: 2028 7768 6963 6820 7769 6c6c 2062 6520   (which will be 
+00014d00: 6469 7363 7573 7365 6420 696e 2061 2073  discussed in a s
+00014d10: 6570 6172 6174 6520 7365 6374 696f 6e20  eparate section 
+00014d20: 6865 7265 756e 6465 7229 2e20 5468 6520  hereunder). The 
+00014d30: 6173 7375 6d70 7469 6f6e 2077 6520 6d61  assumption we ma
+00014d40: 6465 2068 6572 6520 6973 2074 6861 7420  de here is that 
+00014d50: 6e6f 7420 616c 6c20 616e 6368 6f72 2070  not all anchor p
+00014d60: 6169 7273 2028 636f 6c6c 696e 6561 7220  airs (collinear 
+00014d70: 6475 706c 6963 6174 6573 2920 6172 6520  duplicates) are 
+00014d80: 7375 6974 6162 6c65 2066 6f72 2070 6879  suitable for phy
+00014d90: 6c6f 6765 6e65 7469 6320 6461 7469 6e67  logenetic dating
+00014da0: 2c20 666f 7220 696e 7374 616e 6365 2074  , for instance t
+00014db0: 686f 7365 2066 6173 746c 7920 6f72 2073  hose fastly or s
+00014dc0: 6c6f 776c 7920 6576 6f6c 7669 6e67 2067  lowly evolving g
+00014dd0: 656e 6520 6475 706c 6963 6174 6573 2c20  ene duplicates, 
+00014de0: 6265 6361 7573 6520 7468 6579 2772 6520  because they're 
+00014df0: 7072 6f6e 6520 746f 2067 6976 6520 6269  prone to give bi
+00014e00: 6173 6564 2064 6174 696e 6720 6573 7469  ased dating esti
+00014e10: 6d61 7469 6f6e 2e20 536f 2061 7320 746f  mation. So as to
+00014e20: 2072 6574 7269 6576 6520 7468 6520 2272   retrieve the "r
+00014e30: 656c 6961 626c 6522 2061 6e63 686f 7220  eliable" anchor 
+00014e40: 7061 6972 732c 2077 6520 696d 706c 656d  pairs, we implem
+00014e50: 656e 7465 6420 736f 6d65 206d 6574 686f  ented some metho
+00014e60: 6473 206f 6620 6964 656e 7469 6679 696e  ds of identifyin
+00014e70: 6720 6372 6564 6961 626c 6520 616e 6368  g crediable anch
+00014e80: 6f72 2070 6169 7273 2062 6173 6564 206f  or pairs based o
+00014e90: 6e20 7468 6569 7220 4b73 2076 616c 7565  n their Ks value
+00014ea0: 7320 616e 642f 6f72 2072 6573 6964 696e  s and/or residin
+00014eb0: 6720 636f 6c6c 696e 6561 7220 7365 676d  g collinear segm
+00014ec0: 656e 7473 2e20 466f 7220 6120 6765 6e6f  ents. For a geno
+00014ed0: 6d65 2077 6974 6820 636c 6561 7220 7369  me with clear si
+00014ee0: 676e 616c 7320 6f66 2070 7574 6174 6976  gnals of putativ
+00014ef0: 6520 5747 4473 2c20 7375 6368 2061 7320  e WGDs, such as 
+00014f00: 7468 6520 2a41 7175 696c 6567 6961 2063  the *Aquilegia c
+00014f10: 6f65 7275 6c65 612a 2069 6e20 7468 6520  oerulea* in the 
+00014f20: 6578 616d 706c 652c 2061 2068 6575 7269  example, a heuri
+00014f30: 7374 6963 206d 6574 686f 6420 7468 6174  stic method that
+00014f40: 2061 7070 6c69 6573 2074 6865 2070 7269   applies the pri
+00014f50: 6e63 6970 6c65 206f 6620 686f 7720 5b6b  nciple of how [k
+00014f60: 7372 6174 6573 5d28 6874 7470 733a 2f2f  srates](https://
+00014f70: 6769 7468 7562 2e63 6f6d 2f56 4942 2d50  github.com/VIB-P
+00014f80: 5342 2f6b 7372 6174 6573 2920 6669 6e64  SB/ksrates) find
+00014f90: 2074 6865 2069 6e69 7469 616c 2070 6561   the initial pea
+00014fa0: 6b73 2061 6e64 2074 6865 6972 2070 6172  ks and their par
+00014fb0: 616d 6574 6572 7320 7761 7320 696d 706c  ameters was impl
+00014fc0: 656d 656e 7465 6420 746f 2066 696e 6420  emented to find 
+00014fd0: 7468 6520 3935 2520 636f 6e66 6964 656e  the 95% confiden
+00014fe0: 6365 206c 6576 656c 206f 6620 7468 6520  ce level of the 
+00014ff0: 6173 7375 6d65 6420 6c6f 676e 6f72 6d61  assumed lognorma
+00015000: 6c20 6469 7374 7269 6275 7469 6f6e 206f  l distribution o
+00015010: 6620 7468 6520 616e 6368 6f72 2070 6169  f the anchor pai
+00015020: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
+00015030: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00015040: 6f6e 2074 6f20 6669 6c74 6572 2061 6e63  on to filter anc
+00015050: 686f 7220 7061 6972 7320 7769 7468 2074  hor pairs with t
+00015060: 6f6f 2068 6967 6820 6f72 206c 6f77 202a  oo high or low *
+00015070: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00015080: 6765 732c 2074 6865 2065 7861 6d70 6c61  ges, the exampla
+00015090: 7220 636f 6d6d 616e 6420 6f66 2077 6869  r command of whi
+000150a0: 6368 2069 7320 7368 6f77 6564 2069 6e20  ch is showed in 
+000150b0: 7468 6520 6077 6764 2070 6561 6b60 2073  the `wgd peak` s
+000150c0: 6563 7469 6f6e 206f 6620 5b49 6c6c 7573  ection of [Illus
+000150d0: 7472 6174 696f 6e5d 2868 7474 7073 3a2f  tration](https:/
+000150e0: 2f67 6974 6875 622e 636f 6d2f 6865 6368  /github.com/hech
+000150f0: 652d 7073 622f 7767 643f 7461 623d 7265  e-psb/wgd?tab=re
+00015100: 6164 6d65 2d6f 762d 6669 6c65 2369 6c6c  adme-ov-file#ill
+00015110: 7573 7472 6174 696f 6e29 2e20 4966 2074  ustration). If t
+00015120: 6869 7320 6865 7572 6973 7469 6320 6d65  his heuristic me
+00015130: 7468 6f64 2066 6169 6c65 6420 746f 2067  thod failed to g
+00015140: 6976 6520 7265 6173 6f6e 6162 6c65 2072  ive reasonable r
+00015150: 6573 756c 7473 2c20 7768 6963 6820 6973  esults, which is
+00015160: 2075 7375 616c 6c79 2064 7565 2074 6f20   usually due to 
+00015170: 7468 6520 6566 6665 6374 206f 6620 6d75  the effect of mu
+00015180: 6c74 6970 6c65 2061 646a 6163 656e 7420  ltiple adjacent 
+00015190: 5747 4473 2074 6861 7420 626c 7572 7320  WGDs that blurs 
+000151a0: 7468 6520 7065 616b 2d66 696e 6469 6e67  the peak-finding
+000151b0: 2070 726f 6365 7373 2c20 7573 6572 7320   process, users 
+000151c0: 6361 6e20 7475 726e 2074 6f20 7468 6520  can turn to the 
+000151d0: 636f 6c6c 696e 6561 7220 7365 676d 656e  collinear segmen
+000151e0: 7473 2d67 7569 6465 6420 616e 6368 6f72  ts-guided anchor
+000151f0: 2070 6169 7220 636c 7573 7465 7269 6e67   pair clustering
+00015200: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
+00015210: 6077 6764 2076 3260 2c20 696e 2077 6869  `wgd v2`, in whi
+00015220: 6368 2061 2063 6f6c 6c69 6e65 6172 2073  ch a collinear s
+00015230: 6567 6d65 6e74 2d77 6973 6520 474d 4d20  egment-wise GMM 
+00015240: 636c 7573 7465 7269 6e67 2077 696c 6c20  clustering will 
+00015250: 6265 2066 6972 7374 2063 6f6e 6475 6374  be first conduct
+00015260: 6564 2062 6173 6564 206f 6e20 7468 6520  ed based on the 
+00015270: 2273 6f2d 6361 6c6c 6564 2220 7365 676d  "so-called" segm
+00015280: 656e 7420 2a4b 2a3c 7375 623e 533c 2f73  ent *K*<sub>S</s
+00015290: 7562 3e20 6167 6520 7265 7072 6573 656e  ub> age represen
+000152a0: 7465 6420 6279 2074 6865 206d 6564 6961  ted by the media
+000152b0: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+000152c0: 3e20 6167 6520 6f66 2061 6c6c 2074 6865  > age of all the
+000152d0: 2072 6573 6964 696e 6720 7379 6e74 656c   residing syntel
+000152e0: 6f67 7320 286e 6f74 6520 7468 6174 2074  ogs (note that t
+000152f0: 6865 2067 656e 6520 6475 706c 6963 6174  he gene duplicat
+00015300: 6520 7061 6972 7320 6164 6f70 7465 6420  e pairs adopted 
+00015310: 696e 2074 6869 7320 7374 6570 2069 7320  in this step is 
+00015320: 6672 6f6d 2074 6865 2066 696c 6520 606d  from the file `m
+00015330: 756c 7469 706c 6963 6f6e 5f70 6169 7273  ultiplicon_pairs
+00015340: 2e74 7874 6020 7768 6963 6820 636f 6e74  .txt` which cont
+00015350: 6169 6e73 2074 6865 2066 756c 6c20 7365  ains the full se
+00015360: 7420 6f66 2073 796e 7465 6c6f 6773 292c  t of syntelogs),
+00015370: 2069 6e73 7465 6164 206f 6620 7468 6520   instead of the 
+00015380: 736d 616c 6c65 7220 6765 6e65 2073 6574  smaller gene set
+00015390: 206f 6620 616e 6368 6f72 2070 6169 7273   of anchor pairs
+000153a0: 2e20 5468 6520 6469 7374 696e 6374 696f  . The distinctio
+000153b0: 6e20 6265 7477 6565 6e20 616e 6368 6f72  n between anchor
+000153c0: 2070 6169 7273 2061 6e64 2073 796e 7465   pairs and synte
+000153d0: 6c6f 6773 2069 7320 7468 6174 2074 6865  logs is that the
+000153e0: 206c 6174 7465 7220 7265 6665 7273 2074   latter refers t
+000153f0: 6f20 6d75 6c74 6970 6c65 2073 6574 7320  o multiple sets 
+00015400: 6f66 2067 656e 6573 2064 6572 6976 6564  of genes derived
+00015410: 2066 726f 6d20 7468 6520 7361 6d65 2061   from the same a
+00015420: 6e63 6573 7472 616c 2067 656e 6f6d 6963  ncestral genomic
+00015430: 2072 6567 696f 6e20 7768 696c 6520 7468   region while th
+00015440: 6520 666f 726d 6572 2069 6d70 6c69 6573  e former implies
+00015450: 2074 6865 206c 6174 7465 7220 6275 7420   the latter but 
+00015460: 7265 7175 6972 6573 2061 6464 6974 696f  requires additio
+00015470: 6e61 6c6c 7920 7468 6520 636f 6e73 6572  nally the conser
+00015480: 7665 6420 6765 6e65 206f 7264 6572 2c20  ved gene order, 
+00015490: 626f 7468 206f 6620 7768 6963 6820 6172  both of which ar
+000154a0: 652c 2077 6974 6869 6e20 6120 6765 6e6f  e, within a geno
+000154b0: 6d65 2c20 6173 7375 6d65 6420 746f 2062  me, assumed to b
+000154c0: 6520 6f72 6967 696e 6174 6564 2066 726f  e originated fro
+000154d0: 6d20 7468 6520 6475 706c 6963 6174 696f  m the duplicatio
+000154e0: 6e20 6f66 2061 2063 6f6d 6d6f 6e20 616e  n of a common an
+000154f0: 6365 7374 7261 6c20 6765 6e6f 6d69 6320  cestral genomic 
+00015500: 7265 6769 6f6e 2061 6e64 2061 7320 7375  region and as su
+00015510: 6368 2064 6565 6d65 6420 6576 6964 656e  ch deemed eviden
+00015520: 6365 2066 6f72 2057 4744 2e20 5468 656e  ce for WGD. Then
+00015530: 2074 6865 2073 796e 7465 6c6f 6773 2077   the syntelogs w
+00015540: 696c 6c20 6265 206d 6170 7065 6420 6261  ill be mapped ba
+00015550: 636b 2061 6363 6f72 6469 6e67 2074 6f20  ck according to 
+00015560: 7468 6520 636c 7573 7465 7269 6e67 2072  the clustering r
+00015570: 6573 756c 7473 206f 6620 7468 6569 7220  esults of their 
+00015580: 6166 6669 6c69 6174 6564 2073 6567 6d65  affiliated segme
+00015590: 6e74 732e 2053 696e 6365 2074 6865 2047  nts. Since the G
+000155a0: 6175 7373 6961 6e20 7368 6170 6520 6f66  aussian shape of
+000155b0: 2074 6865 2073 6567 6d65 6e74 2063 6c75   the segment clu
+000155c0: 7374 6572 2064 6f65 736e 2774 206e 6563  ster doesn't nec
+000155d0: 6573 7361 7269 6c79 2069 6d69 7461 7465  essarily imitate
+000155e0: 2074 6865 2073 6861 7065 206f 6620 7468   the shape of th
+000155f0: 6520 7265 7369 6469 6e67 2073 796e 7465  e residing synte
+00015600: 6c6f 6773 2c20 736f 2061 7320 746f 2072  logs, so as to r
+00015610: 6574 7269 6576 6520 7468 6520 2272 656c  etrieve the "rel
+00015620: 6961 626c 6522 2067 656e 6520 7365 7473  iable" gene sets
+00015630: 2066 6f72 2070 6879 6c6f 6765 6e65 7469   for phylogeneti
+00015640: 6320 6461 7469 6e67 2c20 7765 2061 6470  c dating, we adp
+00015650: 6f74 6564 2074 6865 2028 3935 2529 2068  oted the (95%) h
+00015660: 6967 6865 7374 2064 656e 7369 7479 2072  ighest density r
+00015670: 6567 696f 6e20 2848 4452 2920 6f66 2074  egion (HDR) of t
+00015680: 6865 2073 796e 7465 6c6f 6720 2a4b 2a3c  he syntelog *K*<
+00015690: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+000156a0: 6469 7374 7269 6275 7469 6f6e 2066 6f72  distribution for
+000156b0: 2074 6865 2070 6879 6c6f 6765 6e65 7469   the phylogeneti
+000156c0: 6320 6461 7469 6e67 2c20 7468 6520 6361  c dating, the ca
+000156d0: 6c63 756c 6174 696f 6e20 6f66 2074 6865  lculation of the
+000156e0: 2028 3935 2529 2048 4452 2069 6e20 7468   (95%) HDR in th
+000156f0: 6520 6675 6e63 7469 6f6e 205b 6361 6c63  e function [calc
+00015700: 756c 6174 6548 5044 5d28 6874 7470 733a  ulateHPD](https:
+00015710: 2f2f 6769 7468 7562 2e63 6f6d 2f68 6563  //github.com/hec
+00015720: 6865 2d70 7362 2f77 6764 2f62 6c6f 622f  he-psb/wgd/blob/
+00015730: 7068 796c 6f64 6174 696e 672f 7767 642f  phylodating/wgd/
+00015740: 7065 616b 2e70 7923 4c31 3735 3729 2073  peak.py#L1757) s
+00015750: 6565 6b73 2074 6865 2073 686f 7274 6573  eeks the shortes
+00015760: 7420 2a4b 2a3c 7375 623e 533c 2f73 7562  t *K*<sub>S</sub
+00015770: 3e20 7261 6e67 6520 2861 2c62 2920 7768  > range (a,b) wh
+00015780: 6963 6820 7361 7469 7366 6965 7320 7468  ich satisfies th
+00015790: 6520 7265 7175 6972 656d 656e 7420 6f66  e requirement of
+000157a0: 2073 7061 6e6e 696e 6720 6d6f 7265 2074   spanning more t
+000157b0: 6861 6e20 2839 3525 2920 6f66 2061 6c6c  han (95%) of all
+000157c0: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+000157d0: 7375 623e 2076 616c 7565 732e 204f 6e20  sub> values. On 
+000157e0: 7468 6520 7072 656d 6973 6520 6f66 2069  the premise of i
+000157f0: 6465 6e74 6966 6965 6420 616e 6368 6f72  dentified anchor
+00015800: 2070 6169 7273 2028 6e6f 7465 2074 6861   pairs (note tha
+00015810: 7420 7468 6520 7379 6e74 656c 6f67 7320  t the syntelogs 
+00015820: 6172 6520 616c 736f 2072 6566 6572 7265  are also referre
+00015830: 6420 6173 2061 6e63 686f 7220 7061 6972  d as anchor pair
+00015840: 7320 6865 7265 6166 7465 7229 2c20 7765  s hereafter), we
+00015850: 2069 6d70 6c65 6d65 6e74 6564 2069 6e20   implemented in 
+00015860: 7468 6520 7072 6f67 7261 6d20 6077 6764  the program `wgd
+00015870: 2064 6d64 6020 7468 6520 736f 2d63 616c   dmd` the so-cal
+00015880: 6c65 6420 616e 6368 6f72 2d61 7761 7265  led anchor-aware
+00015890: 206c 6f63 616c 204d 5242 4873 206f 7220   local MRBHs or 
+000158a0: 6f72 7468 6f67 726f 7570 732c 2069 6e20  orthogroups, in 
+000158b0: 7768 6963 6820 7468 6520 6f72 6967 696e  which the origin
+000158c0: 616c 206c 6f63 616c 204d 5242 4873 2061  al local MRBHs a
+000158d0: 7265 2066 7572 7468 6572 206d 6572 6765  re further merge
+000158e0: 6420 7769 7468 2061 6e63 686f 7220 7061  d with anchor pa
+000158f0: 6972 7320 7375 6368 2074 6861 7420 6561  irs such that ea
+00015900: 6368 206f 7274 686f 6772 6f75 7020 636f  ch orthogroup co
+00015910: 6e74 6169 6e73 2074 6865 2061 6e63 686f  ntains the ancho
+00015920: 7220 7061 6972 2061 6e64 2074 6865 206f  r pair and the o
+00015930: 7274 686f 6c6f 6775 6573 2e20 5769 7468  rthologues. With
+00015940: 2074 6869 7320 6f72 7468 6f67 726f 7570   this orthogroup
+00015950: 2c20 7573 6572 7320 7468 656e 206e 6565  , users then nee
+00015960: 6420 6f6e 6520 7374 6172 7469 6e67 2074  d one starting t
+00015970: 7265 6520 6669 6c65 2028 6173 2073 686f  ree file (as sho
+00015980: 776e 2069 6e20 7468 6520 5b49 6c6c 7573  wn in the [Illus
+00015990: 7472 6174 696f 6e5d 2868 7474 7073 3a2f  tration](https:/
+000159a0: 2f67 6974 6875 622e 636f 6d2f 6865 6368  /github.com/hech
+000159b0: 652d 7073 622f 7767 642f 7472 6565 2f70  e-psb/wgd/tree/p
+000159c0: 6879 6c6f 6461 7469 6e67 3f74 6162 3d72  hylodating?tab=r
+000159d0: 6561 646d 652d 6f76 2d66 696c 6523 696c  eadme-ov-file#il
+000159e0: 6c75 7374 7261 7469 6f6e 2920 7061 7274  lustration) part
+000159f0: 2920 696e 6469 6361 7469 6e67 2074 6865  ) indicating the
+00015a00: 2074 7265 6520 746f 706f 6c6f 6779 2061   tree topology a
+00015a10: 6e64 2066 6f73 7369 6c20 6361 6c69 6272  nd fossil calibr
+00015a20: 6174 696f 6e20 696e 666f 726d 6174 696f  ation informatio
+00015a30: 6e20 666f 7220 7468 6520 6669 6e61 6c20  n for the final 
+00015a40: 7068 796c 6f67 656e 6574 6963 2064 6174  phylogenetic dat
+00015a50: 696e 6720 7573 696e 6720 7468 6520 7072  ing using the pr
+00015a60: 6f67 7261 6d20 6077 6764 2066 6f63 7573  ogram `wgd focus
+00015a70: 602e 2055 7365 7273 2063 616e 2073 6574  `. Users can set
+00015a80: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
+00015a90: 2060 2d2d 7072 6f74 636f 6364 6174 696e   `--protcocdatin
+00015aa0: 6760 2074 6f20 6f6e 6c79 2063 6f6e 6475  g` to only condu
+00015ab0: 6374 2074 6865 2064 6174 696e 6720 6f66  ct the dating of
+00015ac0: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
+00015ad0: 6f6e 2070 726f 7465 696e 204d 5341 206f  on protein MSA o
+00015ae0: 7220 7468 6520 666c 6167 206f 7074 696f  r the flag optio
+00015af0: 6e20 602d 2d70 726f 7464 6174 696e 6760  n `--protdating`
+00015b00: 2074 6f20 6f6e 6c79 2063 6f6e 6475 6374   to only conduct
+00015b10: 2074 6865 2064 6174 696e 6720 6f66 2074   the dating of t
+00015b20: 6865 2070 726f 7465 696e 204d 5341 732c  he protein MSAs,
+00015b30: 206e 6f74 6564 2074 6861 7420 7468 6573   noted that thes
+00015b40: 6520 7477 6f20 6f70 7469 6f6e 7320 6f6e  e two options on
+00015b50: 6c79 2077 6f72 6b20 666f 7220 7468 6520  ly work for the 
+00015b60: 606d 636d 6374 7265 6560 206f 7074 696f  `mcmctree` optio
+00015b70: 6e20 736f 2066 6172 2e20 5468 6520 666c  n so far. The fl
+00015b80: 6167 206f 7074 696f 6e20 602d 2d70 6172  ag option `--par
+00015b90: 7469 7469 6f6e 6020 6361 6e20 6265 2073  tition` can be s
+00015ba0: 6574 2074 6f20 7065 7266 6f72 6d20 606d  et to perform `m
+00015bb0: 636d 6374 7265 6560 2061 6e61 6c79 7369  cmctree` analysi
+00015bc0: 7320 7573 696e 6720 7468 6520 7061 7274  s using the part
+00015bd0: 6974 696f 6e65 6420 6461 7461 2028 692e  itioned data (i.
+00015be0: 652e 2c20 3173 742c 2032 6e64 2061 6e64  e., 1st, 2nd and
+00015bf0: 2033 7264 2070 6f73 6974 696f 6e20 6f66   3rd position of
+00015c00: 2063 6f64 6f6e 2920 696e 7374 6561 6420   codon) instead 
+00015c10: 6f66 2075 7369 6e67 2074 6865 2063 6f64  of using the cod
+00015c20: 6f6e 2061 7320 6120 7768 6f6c 652e 2054  on as a whole. T
+00015c30: 6865 206f 7074 696f 6e20 602d 2d61 616d  he option `--aam
+00015c40: 6f64 656c 6020 6361 6e20 6265 2073 6574  odel` can be set
+00015c50: 2074 6f20 6465 7465 726d 696e 6520 7468   to determine th
+00015c60: 6520 616d 696e 6f20 6163 6964 206d 6f64  e amino acid mod
+00015c70: 656c 2061 7070 6c69 6564 2069 6e20 606d  el applied in `m
+00015c80: 636d 6374 7265 6560 2061 6e61 6c79 7369  cmctree` analysi
+00015c90: 732e 2054 6865 206f 7074 696f 6e20 602d  s. The option `-
+00015ca0: 6473 6020 6361 6e20 6265 2075 7365 6420  ds` can be used 
+00015cb0: 746f 2073 6574 2070 6172 616d 6574 6572  to set parameter
+00015cc0: 7320 666f 7220 7468 6520 6d6f 6c65 6375  s for the molecu
+00015cd0: 6c61 7220 6461 7469 6e67 2070 726f 6772  lar dating progr
+00015ce0: 616d 2e20 5468 6520 7061 7261 6c6c 656c  am. The parallel
+00015cf0: 2074 6872 6561 6473 2068 6572 6520 6172   threads here ar
+00015d00: 6520 746f 2070 6172 616c 6c65 6c69 7a65  e to parallelize
+00015d10: 2074 6865 2061 6e61 6c79 7369 7320 666f   the analysis fo
+00015d20: 7220 6561 6368 2067 656e 6520 6661 6d69  r each gene fami
+00015d30: 6c79 2061 6e64 2074 6875 7320 7375 6767  ly and thus sugg
+00015d40: 6573 7465 6420 746f 2062 6520 7365 7420  ested to be set 
+00015d50: 6173 206d 7563 6820 6173 2074 6865 206e  as much as the n
+00015d60: 756d 6265 7220 6f66 2067 656e 6520 6661  umber of gene fa
+00015d70: 6d69 6c69 6573 2e0a 0a2a 2a41 2073 7567  milies...**A sug
+00015d80: 6765 7374 6564 2073 7461 7274 696e 6720  gested starting 
+00015d90: 7275 6e20 6361 6e20 6265 2077 6974 6820  run can be with 
+00015da0: 7468 6520 636f 6d6d 616e 6420 6265 6c6f  the command belo
+00015db0: 772e 2a2a 0a0a 6060 600a 7767 6420 666f  w.**..```.wgd fo
+00015dc0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+00015dd0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+00015de0: 2073 6571 7565 6e63 6533 202d 6420 6d63   sequence3 -d mc
+00015df0: 6d63 7472 6565 202d 7370 2073 7064 6174  mctree -sp spdat
+00015e00: 610a 7767 6420 666f 6375 7320 6661 6d69  a.wgd focus fami
+00015e10: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+00015e20: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+00015e30: 6533 202d 6420 7238 7320 2d73 7020 7370  e3 -d r8s -sp sp
+00015e40: 6461 7461 202d 2d6e 7369 7465 7320 6e73  data --nsites ns
+00015e50: 6974 6569 6e66 6f0a 7767 6420 666f 6375  iteinfo.wgd focu
+00015e60: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+00015e70: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
+00015e80: 6571 7565 6e63 6533 202d 6420 6265 6173  equence3 -d beas
+00015e90: 7420 2d2d 666f 7373 696c 2066 6f73 7369  t --fossil fossi
+00015ea0: 6c69 6e66 6f20 2d2d 726f 6f74 6865 6967  linfo --rootheig
+00015eb0: 6874 2072 6f6f 7468 6569 6768 7469 6e66  ht rootheightinf
+00015ec0: 6f20 2d2d 6368 6169 6e73 6574 2063 6861  o --chainset cha
+00015ed0: 696e 7365 7469 6e66 6f20 2d2d 6265 6173  insetinfo --beas
+00015ee0: 746c 676a 6172 2024 5041 5448 0a60 6060  tlgjar $PATH.```
+00015ef0: 0a0a 2323 2049 6c6c 7573 7472 6174 696f  ..## Illustratio
+00015f00: 6e0a 0a57 6520 696c 6c75 7374 7261 7465  n..We illustrate
+00015f10: 206f 7572 2070 726f 6772 616d 206f 6e20   our program on 
+00015f20: 616e 2065 7865 6d70 6c61 7279 2057 4744  an exemplary WGD
+00015f30: 2069 6e66 6572 656e 6365 2061 6e64 2064   inference and d
+00015f40: 6174 696e 6720 7570 6f6e 2073 7065 6369  ating upon speci
+00015f50: 6573 202a 4171 7569 6c65 6769 6120 636f  es *Aquilegia co
+00015f60: 6572 756c 6561 2a2e 0a0a 5468 6520 2a41  erulea*...The *A
+00015f70: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
+00015f80: 612a 2077 6173 2072 6570 6f72 7465 6420  a* was reported 
+00015f90: 746f 2065 7870 6572 6965 6e63 6520 616e  to experience an
+00015fa0: 2070 616c 656f 2d70 6f6c 7970 6c6f 6964   paleo-polyploid
+00015fb0: 697a 6174 696f 6e20 6576 656e 7420 6166  ization event af
+00015fc0: 7465 7220 7468 6520 6469 7665 7267 656e  ter the divergen
+00015fd0: 6365 206f 6620 636f 7265 2065 7564 6963  ce of core eudic
+00015fe0: 6f74 732c 2077 6869 6368 2069 7320 6c69  ots, which is li
+00015ff0: 6b65 6c79 2073 6861 7265 6420 6279 2061  kely shared by a
+00016000: 6c6c 2052 616e 756e 6375 6c61 6c65 732e  ll Ranunculales.
+00016010: 0a0a 4669 7273 7420 6162 6f76 6520 616c  ..First above al
+00016020: 6c2c 206c 6574 2773 2064 656c 696e 6561  l, let's delinea
+00016030: 7465 2074 6865 2077 686f 6c65 2070 6172  te the whole par
+00016040: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00016050: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00016060: 6275 7469 6f6e 2061 6e64 2068 6176 6520  bution and have 
+00016070: 6120 6261 7369 6320 6f62 7365 7276 6174  a basic observat
+00016080: 696f 6e20 666f 7220 706f 7465 6e74 6961  ion for potentia
+00016090: 6c6c 7920 636f 6e63 6569 7661 626c 6520  lly conceivable 
+000160a0: 5747 4473 2c20 7573 696e 6720 7468 6520  WGDs, using the 
+000160b0: 636f 6d6d 616e 6420 6c69 6e65 2062 656c  command line bel
+000160c0: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
+000160d0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+000160e0: 6c65 610a 7767 6420 6b73 6420 7767 645f  lea.wgd ksd wgd_
+000160f0: 646d 642f 4171 7569 6c65 6769 615f 636f  dmd/Aquilegia_co
+00016100: 6572 756c 6561 2e74 7376 2041 7175 696c  erulea.tsv Aquil
+00016110: 6567 6961 5f63 6f65 7275 6c65 610a 6060  egia_coerulea.``
+00016120: 600a 0a54 6865 2063 6f6e 7374 7275 6374  `..The construct
+00016130: 6564 2077 686f 6c65 2070 6172 616e 6f6d  ed whole paranom
+00016140: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00016150: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00016160: 6f6e 206f 6620 2a41 7175 696c 6567 6961  on of *Aquilegia
+00016170: 2063 6f65 7275 6c65 612a 2069 7320 6173   coerulea* is as
+00016180: 2062 656c 6f77 2c20 7765 2063 616e 2073   below, we can s
+00016190: 6565 2074 6861 7420 7468 6572 6520 7365  ee that there se
+000161a0: 656d 7320 746f 2062 6520 6120 6875 6d70  ems to be a hump
+000161b0: 2061 7420 2a4b 2a3c 7375 623e 533c 2f73   at *K*<sub>S</s
+000161c0: 7562 3e20 3120 6275 7420 6e6f 7420 636c  ub> 1 but not cl
+000161d0: 6561 722e 0a0a 215b 5d28 6461 7461 2f41  ear...![](data/A
+000161e0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+000161f0: 612e 7473 762e 6b73 645f 7770 2e73 7667  a.tsv.ksd_wp.svg
+00016200: 290a 0a57 6520 7468 656e 2063 6f6e 7374  )..We then const
+00016210: 7275 6374 2074 6865 2061 6e63 686f 7220  ruct the anchor 
+00016220: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00016230: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+00016240: 2075 7369 6e67 2074 6865 2063 6f6d 6d61   using the comma
+00016250: 6e64 206c 696e 6520 6265 6c6f 772e 0a0a  nd line below...
+00016260: 6060 600a 7767 6420 7379 6e20 2d66 206d  ```.wgd syn -f m
+00016270: 524e 4120 2d61 204e 616d 6520 7767 645f  RNA -a Name wgd_
+00016280: 646d 642f 4171 7569 6c65 6769 615f 636f  dmd/Aquilegia_co
+00016290: 6572 756c 6561 2e74 7376 2041 7175 696c  erulea.tsv Aquil
+000162a0: 6567 6961 5f63 6f65 7275 6c65 612e 6766  egia_coerulea.gf
+000162b0: 6633 202d 6b73 2077 6764 5f6b 7364 2f41  f3 -ks wgd_ksd/A
+000162c0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+000162d0: 612e 7473 762e 6b73 2e74 7376 0a60 6060  a.tsv.ks.tsv.```
+000162e0: 0a0a 4173 2073 686f 776e 2062 656c 6f77  ..As shown below
+000162f0: 2c20 7468 6572 6520 6172 6520 736f 6d65  , there are some
+00016300: 2072 6574 6169 6e65 6420 616e 6368 6f72   retained anchor
+00016310: 2070 6169 7273 2077 6974 6820 2a4b 2a3c   pairs with *K*<
+00016320: 7375 623e 533c 2f73 7562 3e20 6265 7477  sub>S</sub> betw
+00016330: 6565 6e20 3120 616e 6420 322c 2077 6869  een 1 and 2, whi
+00016340: 6368 2073 6565 6d73 2074 6f20 7375 6767  ch seems to sugg
+00016350: 6573 7420 6120 5747 4420 6576 656e 742e  est a WGD event.
+00016360: 0a0a 215b 5d28 6461 7461 2f41 7175 696c  ..![](data/Aquil
+00016370: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00016380: 762e 6b73 645f 7770 5f61 702e 7376 6729  v.ksd_wp_ap.svg)
+00016390: 0a0a 5468 6520 6173 736f 6369 6174 6564  ..The associated
+000163a0: 2060 6475 7053 7461 636b 6020 706c 6f74   `dupStack` plot
+000163b0: 2073 686f 7773 2074 6861 7420 7468 6572   shows that ther
+000163c0: 6520 6172 6520 6e75 6d65 726f 7573 2064  e are numerous d
+000163d0: 7570 6c69 6361 7465 6420 7365 676d 656e  uplicated segmen
+000163e0: 7473 2061 6372 6f73 7320 6d6f 7374 206f  ts across most o
+000163f0: 6620 7468 6520 6368 726f 6d6f 736f 6d65  f the chromosome
+00016400: 732e 0a0a 215b 5d28 6461 7461 2f73 796e  s...![](data/syn
+00016410: 5f72 6573 756c 7473 2f41 7175 696c 6567  _results/Aquileg
+00016420: 6961 5f63 6f65 7275 6c65 615f 4171 7569  ia_coerulea_Aqui
+00016430: 6c65 6769 615f 636f 6572 756c 6561 5f6d  legia_coerulea_m
+00016440: 756c 7469 706c 6963 6f6e 735f 6c65 7665  ultiplicons_leve
+00016450: 6c2e 706e 6729 0a0a 5765 2069 6d70 6c65  l.png)..We imple
+00016460: 6d65 6e74 6564 2074 776f 2074 7970 6573  mented two types
+00016470: 206f 6620 646f 7420 706c 6f74 7320 696e   of dot plots in
+00016480: 206f 7866 6f72 6420 6772 6964 3a20 6f6e   oxford grid: on
+00016490: 6520 696e 2074 6865 2075 6e69 7420 6f66  e in the unit of
+000164a0: 2062 6173 6573 2061 6e64 2074 6865 206f   bases and the o
+000164b0: 7468 6572 2069 6e20 7468 6520 756e 6974  ther in the unit
+000164c0: 206f 6620 6765 6e65 732c 2077 6869 6368   of genes, which
+000164d0: 2063 616e 2062 6520 636f 6c6f 7265 6420   can be colored 
+000164e0: 6279 202a 4b2a 3c73 7562 3e53 3c2f 7375  by *K*<sub>S</su
+000164f0: 623e 2076 616c 7565 7320 6769 7665 6e20  b> values given 
+00016500: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00016510: 6461 7461 2e0a 0a21 5b5d 2864 6174 612f  data...![](data/
+00016520: 7379 6e5f 7265 7375 6c74 732f 4171 7569  syn_results/Aqui
+00016530: 6c65 6769 615f 636f 6572 756c 6561 2d76  legia_coerulea-v
+00016540: 732d 4171 7569 6c65 6769 615f 636f 6572  s-Aquilegia_coer
+00016550: 756c 6561 5f4b 732e 646f 745f 756e 6974  ulea_Ks.dot_unit
+00016560: 5f67 656e 652e 706e 6729 0a0a 4173 2073  _gene.png)..As s
+00016570: 686f 776e 2061 626f 7665 2c20 7468 6520  hown above, the 
+00016580: 646f 7420 706c 6f74 2069 6e20 7468 6520  dot plot in the 
+00016590: 756e 6974 206f 6620 6765 6e65 7320 7072  unit of genes pr
+000165a0: 6573 656e 7473 206e 756d 6572 6f75 7320  esents numerous 
+000165b0: 6465 6e73 656c 7920 6167 6772 6567 6174  densely aggregat
+000165c0: 6564 2028 6c69 6e65 2d6c 696b 6529 2061  ed (line-like) a
+000165d0: 6e63 686f 7220 706f 696e 7473 2061 7420  nchor points at 
+000165e0: 6d6f 7374 206f 6620 7468 6520 6368 726f  most of the chro
+000165f0: 6d6f 736f 6d65 7320 7769 7468 2063 6f6e  mosomes with con
+00016600: 7369 7374 656e 7420 2a4b 2a3c 7375 623e  sistent *K*<sub>
+00016610: 533c 2f73 7562 3e20 6167 6520 6265 7477  S</sub> age betw
+00016620: 6565 6e20 3120 616e 6420 322e 2054 6865  een 1 and 2. The
+00016630: 2064 6f74 2070 6c6f 7420 696e 2074 6865   dot plot in the
+00016640: 2075 6e69 7420 6f66 2062 6173 6573 2073   unit of bases s
+00016650: 686f 7773 2074 6865 2073 616d 6520 7061  hows the same pa
+00016660: 7474 6572 6e2c 2061 7320 6d61 6e69 6665  ttern, as manife
+00016670: 7374 6564 2062 656c 6f77 2e0a 0a21 5b5d  sted below...![]
+00016680: 2864 6174 612f 7379 6e5f 7265 7375 6c74  (data/syn_result
+00016690: 732f 4171 7569 6c65 6769 615f 636f 6572  s/Aquilegia_coer
+000166a0: 756c 6561 2d76 732d 4171 7569 6c65 6769  ulea-vs-Aquilegi
+000166b0: 615f 636f 6572 756c 6561 5f4b 732e 646f  a_coerulea_Ks.do
+000166c0: 742e 706e 6729 0a0a 5468 6520 646f 7420  t.png)..The dot 
+000166d0: 706c 6f74 7320 7769 7468 6f75 7420 2a4b  plots without *K
+000166e0: 2a3c 7375 623e 533c 2f73 7562 3e20 616e  *<sub>S</sub> an
+000166f0: 6e6f 7461 7469 6f6e 2077 696c 6c20 616c  notation will al
+00016700: 736f 2062 6520 6175 746f 6d61 7465 6c79  so be automately
+00016710: 2070 726f 6475 6365 642c 2061 7320 7368   produced, as sh
+00016720: 6f77 6e20 6265 6c6f 772e 0a0a 215b 5d28  own below...![](
+00016730: 6461 7461 2f73 796e 5f72 6573 756c 7473  data/syn_results
+00016740: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00016750: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
+00016760: 5f63 6f65 7275 6c65 612e 646f 745f 756e  _coerulea.dot_un
+00016770: 6974 5f67 656e 652e 706e 6729 0a0a 215b  it_gene.png)..![
+00016780: 5d28 6461 7461 2f73 796e 5f72 6573 756c  ](data/syn_resul
+00016790: 7473 2f41 7175 696c 6567 6961 5f63 6f65  ts/Aquilegia_coe
+000167a0: 7275 6c65 612d 7673 2d41 7175 696c 6567  rulea-vs-Aquileg
+000167b0: 6961 5f63 6f65 7275 6c65 612e 646f 742e  ia_coerulea.dot.
+000167c0: 706e 6729 0a0a 4e6f 7465 2074 6861 7420  png)..Note that 
+000167d0: 7468 6520 6f70 6163 6974 7920 6f66 2061  the opacity of a
+000167e0: 6e63 686f 7220 646f 7473 2061 6e64 2061  nchor dots and a
+000167f0: 6c6c 2068 6f6d 6f6c 6f67 2064 6f74 7320  ll homolog dots 
+00016800: 6361 6e20 6265 2073 6574 2062 7920 7468  can be set by th
+00016810: 6520 6f70 7469 6f6e 2060 2d2d 6170 616c  e option `--apal
+00016820: 7068 6160 2061 6e64 2060 2d2d 686f 616c  pha` and `--hoal
+00016830: 7068 6160 2073 6570 6172 6174 656c 792e  pha` separately.
+00016840: 2049 6620 6f6e 6520 6a75 7374 2077 616e   If one just wan
+00016850: 7473 2074 6f20 7365 6520 7468 6520 616e  ts to see the an
+00016860: 6368 6f72 2064 6f74 732c 2073 6574 7469  chor dots, setti
+00016870: 6e67 2074 6865 2060 686f 616c 7068 6160  ng the `hoalpha`
+00016880: 2061 7320 3020 286f 7220 6f74 6865 7220   as 0 (or other 
+00016890: 6d69 6e75 7363 756c 6520 7661 6c75 6573  minuscule values
+000168a0: 2920 7769 6c6c 2064 6f2e 2049 6620 6f6e  ) will do. If on
+000168b0: 6520 7761 6e74 7320 746f 2073 6565 2074  e wants to see t
+000168c0: 6865 2064 6973 7472 6962 7574 696f 6e20  he distribution 
+000168d0: 6f66 2077 686f 6c65 2064 6f74 7320 6265  of whole dots be
+000168e0: 7474 6572 2c20 7365 7474 696e 6720 7468  tter, setting th
+000168f0: 6520 6068 6f61 6c70 6861 6020 6869 6768  e `hoalpha` high
+00016900: 6572 2028 616e 6420 6061 7061 6c70 6861  er (and `apalpha
+00016910: 6020 6c6f 7765 7229 2077 696c 6c20 646f  ` lower) will do
+00016920: 2e20 5468 6520 6064 6f74 7369 7a65 6020  . The `dotsize` 
+00016930: 6f70 7469 6f6e 2063 616e 2062 6520 6361  option can be ca
+00016940: 6c6c 6564 2074 6f20 6164 6a75 7374 2074  lled to adjust t
+00016950: 6865 2073 697a 6520 6f66 2064 6f74 732e  he size of dots.
+00016960: 0a0a 4120 6675 7274 6865 7220 6173 736f  ..A further asso
+00016970: 6369 6174 6564 2053 796e 6465 7074 6820  ciated Syndepth 
+00016980: 706c 6f74 2073 686f 7773 2074 6861 7420  plot shows that 
+00016990: 7468 6572 6520 6172 6520 6d6f 7265 2074  there are more t
+000169a0: 6861 6e20 3530 2064 7570 6c69 6361 7465  han 50 duplicate
+000169b0: 6420 7365 676d 656e 7473 206c 6f6e 6765  d segments longe
+000169c0: 7220 7468 616e 2031 3030 3030 2062 7020  r than 10000 bp 
+000169d0: 616e 6420 3330 2067 656e 6573 2028 736f  and 30 genes (so
+000169e0: 2061 7320 746f 2064 726f 7020 6672 6167   as to drop frag
+000169f0: 6d65 6e74 6172 7920 7365 676d 656e 7473  mentary segments
+00016a00: 292c 2077 6869 6368 2064 6f6d 696e 6174  ), which dominat
+00016a10: 6573 2074 6865 2077 686f 6c65 2063 6f6c  es the whole col
+00016a20: 6c69 6e65 6172 2072 6174 696f 2063 6174  linear ratio cat
+00016a30: 6567 6f72 792e 0a0a 215b 5d28 6461 7461  egory...![](data
+00016a40: 2f73 796e 5f72 6573 756c 7473 2f53 796e  /syn_results/Syn
+00016a50: 6465 7074 682e 7376 6729 0a0a 5765 2063  depth.svg)..We c
+00016a60: 616e 2066 6974 2061 6e20 454c 4d4d 206d  an fit an ELMM m
+00016a70: 6978 7475 7265 206d 6f64 656c 2075 706f  ixture model upo
+00016a80: 6e20 7468 6520 7768 6f6c 6520 7061 7261  n the whole para
+00016a90: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+00016aa0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+00016ab0: 7574 696f 6e20 746f 2073 6565 206d 6f72  ution to see mor
+00016ac0: 6520 6163 6375 7261 7465 6c79 2074 6865  e accurately the
+00016ad0: 2073 6967 6e69 6669 6361 6e63 6520 616e   significance an
+00016ae0: 6420 6c6f 6361 7469 6f6e 206f 6620 706f  d location of po
+00016af0: 7465 6e74 6961 6c20 5747 4473 2c20 7573  tential WGDs, us
+00016b00: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
+00016b10: 6c69 6e65 2062 656c 6f77 2e0a 0a60 6060  line below...```
+00016b20: 0a77 6764 2076 697a 202d 6420 7767 645f  .wgd viz -d wgd_
+00016b30: 6b73 642f 4171 7569 6c65 6769 615f 636f  ksd/Aquilegia_co
+00016b40: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
+00016b50: 760a 6060 600a 0a54 6865 2072 6573 756c  v.```..The resul
+00016b60: 7420 6f66 2045 4c4d 4d20 6d69 7874 7572  t of ELMM mixtur
+00016b70: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
+00016b80: 6e67 2073 686f 7773 2074 6861 7420 7468  ng shows that th
+00016b90: 6572 6520 6973 2061 206c 696b 656c 7920  ere is a likely 
+00016ba0: 5747 4420 636f 6d70 6f6e 656e 7420 6174  WGD component at
+00016bb0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00016bc0: 2031 2e31 392e 0a0a 215b 5d28 6461 7461   1.19...![](data
+00016bd0: 2f65 6c6d 6d5f 4171 7569 6c65 6769 615f  /elmm_Aquilegia_
+00016be0: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00016bf0: 7473 765f 6265 7374 5f6d 6f64 656c 735f  tsv_best_models_
+00016c00: 7765 6967 6874 6564 2e73 7667 290a 0a4c  weighted.svg)..L
+00016c10: 6574 2773 2064 6f20 6120 6d69 7874 7572  et's do a mixtur
+00016c20: 6520 6d6f 6465 6c20 636c 7573 7465 7269  e model clusteri
+00016c30: 6e67 2066 6f72 2061 6e63 686f 7220 2a4b  ng for anchor *K
+00016c40: 2a3c 7375 623e 533c 2f73 7562 3e20 746f  *<sub>S</sub> to
+00016c50: 6f2c 2075 7369 6e67 2074 6865 2063 6f6d  o, using the com
+00016c60: 6d61 6e64 206c 696e 6520 6265 6c6f 772e  mand line below.
+00016c70: 204e 6f74 6520 7468 6174 2074 6869 7320   Note that this 
+00016c80: 7374 6570 2077 696c 6c20 6175 746f 6d61  step will automa
+00016c90: 7465 6c79 2063 616c 6c20 7468 6520 7365  tely call the se
+00016ca0: 676d 656e 7420 2a4b 2a3c 7375 623e 533c  gment *K*<sub>S<
+00016cb0: 2f73 7562 3e20 636c 7573 7465 7269 6e67  /sub> clustering
+00016cc0: 2061 6e61 6c79 7369 7320 746f 6f2e 0a0a   analysis too...
+00016cd0: 6060 600a 7767 6420 7065 616b 2077 6764  ```.wgd peak wgd
+00016ce0: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
+00016cf0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00016d00: 7376 202d 2d61 6e63 686f 7270 6f69 6e74  sv --anchorpoint
+00016d10: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
+00016d20: 652d 6f75 742f 616e 6368 6f72 706f 696e  e-out/anchorpoin
+00016d30: 7473 2e74 7874 202d 2d73 6567 6d65 6e74  ts.txt --segment
+00016d40: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
+00016d50: 652d 6f75 742f 7365 676d 656e 7473 2e74  e-out/segments.t
+00016d60: 7874 202d 2d6c 6973 7465 6c65 6d65 6e74  xt --listelement
+00016d70: 7320 7767 645f 7379 6e2f 6961 6468 6f72  s wgd_syn/iadhor
+00016d80: 652d 6f75 742f 6c69 7374 5f65 6c65 6d65  e-out/list_eleme
+00016d90: 6e74 732e 7478 7420 2d2d 6d75 6c74 6970  nts.txt --multip
+00016da0: 6c69 636f 6e70 6169 7273 2077 6764 5f73  liconpairs wgd_s
+00016db0: 796e 2f69 6164 686f 7265 2d6f 7574 2f6d  yn/iadhore-out/m
+00016dc0: 756c 7469 706c 6963 6f6e 5f70 6169 7273  ultiplicon_pairs
+00016dd0: 2e74 7874 2028 2d2d 7765 6967 6874 6564  .txt (--weighted
+00016de0: 290a 6060 600a 0a54 6865 2061 6e63 686f  ).```..The ancho
+00016df0: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
+00016e00: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00016e10: 6f6e 2061 6c73 6f20 6861 7320 6120 6c69  on also has a li
+00016e20: 6b65 6c79 2057 4744 2063 6f6d 706f 6e65  kely WGD compone
+00016e30: 6e74 2077 6974 6820 6d6f 6465 2031 2e32  nt with mode 1.2
+00016e40: 382e 0a0a 215b 5d28 6461 7461 2f4f 7269  8...![](data/Ori
+00016e50: 6769 6e61 6c5f 416e 6368 6f72 4b73 5f47  ginal_AnchorKs_G
+00016e60: 4d4d 5f43 6f6d 706f 6e65 6e74 335f 6e6f  MM_Component3_no
+00016e70: 6465 5f61 7665 7261 6765 645f 4c6f 676e  de_averaged_Logn
+00016e80: 6f72 6d61 6c2e 7376 6729 0a0a 4e6f 7720  ormal.svg)..Now 
+00016e90: 7468 6174 2077 6520 6861 7665 2073 6565  that we have see
+00016ea0: 6e20 7468 6520 6576 6964 656e 6365 206f  n the evidence o
+00016eb0: 6620 6e75 6d65 726f 7573 2064 7570 6c69  f numerous dupli
+00016ec0: 6361 7465 6420 7365 676d 656e 7473 2061  cated segments a
+00016ed0: 6e64 2074 6865 2061 6767 7265 6761 7469  nd the aggregati
+00016ee0: 6f6e 206f 6620 6475 706c 6963 6174 6573  on of duplicates
+00016ef0: 2061 6765 2061 7420 2a4b 2a3c 7375 623e   age at *K*<sub>
+00016f00: 533c 2f73 7562 3e20 312e 3238 206f 7220  S</sub> 1.28 or 
+00016f10: 312e 3139 2066 6f72 2061 6e63 686f 7220  1.19 for anchor 
+00016f20: 7061 6972 7320 616e 6420 6e6f 6e2d 616e  pairs and non-an
+00016f30: 6368 6f72 2070 6169 7273 2074 6872 6f75  chor pairs throu
+00016f40: 6768 6f75 7420 7468 6520 7768 6f6c 6520  ghout the whole 
+00016f50: 6765 6e6f 6d65 2e20 5765 2063 616e 2063  genome. We can c
+00016f60: 6c61 696d 2077 6974 6820 736f 6d65 2063  laim with some c
+00016f70: 6f6e 6669 6465 6e63 6520 7468 6174 202a  onfidence that *
+00016f80: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
+00016f90: 6561 2a20 6d69 6768 7420 6861 7665 2065  ea* might have e
+00016fa0: 7870 6572 6965 6e63 6564 2061 2070 616c  xperienced a pal
+00016fb0: 656f 2d70 6f6c 7970 6c6f 6964 697a 6174  eo-polyploidizat
+00016fc0: 696f 6e20 6576 656e 742e 204e 6578 742c  ion event. Next,
+00016fd0: 204c 6574 2773 2068 6176 6520 6120 6675   Let's have a fu
+00016fe0: 7274 6865 7220 6c6f 6f6b 2061 626f 7574  rther look about
+00016ff0: 2069 7473 2070 6879 6c6f 6765 6e65 7469   its phylogeneti
+00017000: 6320 6c6f 6361 7469 6f6e 2e20 5765 206b  c location. We k
+00017010: 6e6f 7720 7468 6174 2074 6865 7265 2061  now that there a
+00017020: 7265 2075 6e63 6572 7461 696e 7469 6573  re uncertainties
+00017030: 2061 626f 7574 2077 6865 7468 6572 2074   about whether t
+00017040: 6869 7320 7075 7461 7469 7665 2070 616c  his putative pal
+00017050: 656f 2d70 6f6c 7970 6c6f 6964 697a 6174  eo-polyploidizat
+00017060: 696f 6e20 6576 656e 7420 6973 2073 6861  ion event is sha
+00017070: 7265 6420 7769 7468 2061 6c6c 2065 7564  red with all eud
+00017080: 6963 6f74 7320 6f72 206e 6f74 2e20 5765  icots or not. We
+00017090: 2063 616e 2063 686f 6f73 6520 736f 6d65   can choose some
+000170a0: 206f 7468 6572 2065 7564 6963 6f74 2067   other eudicot g
+000170b0: 656e 6f6d 6573 2074 6f20 7365 6520 7468  enomes to see th
+000170c0: 6520 6f72 6465 7269 6e67 206f 6620 7370  e ordering of sp
+000170d0: 6563 6961 7469 6f6e 2061 6e64 2070 6f6c  eciation and pol
+000170e0: 7970 6c6f 6964 697a 6174 696f 6e20 6576  yploidization ev
+000170f0: 656e 7473 2e20 4865 7265 2077 6520 6368  ents. Here we ch
+00017100: 6f6f 7365 202a 5669 7469 7320 7669 6e69  oose *Vitis vini
+00017110: 6665 7261 2a2c 202a 5072 6f74 6561 2063  fera*, *Protea c
+00017120: 796e 6172 6f69 6465 732a 2061 6e64 202a  ynaroides* and *
+00017130: 4163 6f72 7573 2061 6d65 7269 6361 6e75  Acorus americanu
+00017140: 732a 2069 6e20 7468 6520 666f 6c6c 6f77  s* in the follow
+00017150: 696e 6720 2a4b 2a3c 7375 623e 533c 2f73  ing *K*<sub>S</s
+00017160: 7562 3e20 616e 616c 7973 6973 2e20 4669  ub> analysis. Fi
+00017170: 7273 742c 2077 6520 6275 696c 7420 6120  rst, we built a 
+00017180: 676c 6f62 616c 204d 5242 4820 6661 6d69  global MRBH fami
+00017190: 6c79 2075 7369 6e67 2074 6865 2063 6f6d  ly using the com
+000171a0: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
+000171b0: 0a77 6764 2064 6d64 202d 2d67 6c6f 6261  .wgd dmd --globa
+000171c0: 6c6d 7262 6820 4171 7569 6c65 6769 615f  lmrbh Aquilegia_
+000171d0: 636f 6572 756c 6561 2050 726f 7465 615f  coerulea Protea_
+000171e0: 6379 6e61 726f 6964 6573 2041 636f 7275  cynaroides Acoru
+000171f0: 735f 616d 6572 6963 616e 7573 2056 6974  s_americanus Vit
+00017200: 6973 5f76 696e 6966 6572 6120 2d6f 2077  is_vinifera -o w
+00017210: 6764 5f67 6c6f 6261 6c6d 7262 680a 6060  gd_globalmrbh.``
+00017220: 600a 0a49 6e20 7468 6520 676c 6f62 616c  `..In the global
+00017230: 204d 5242 4820 6661 6d69 6c79 2c20 6576   MRBH family, ev
+00017240: 6572 7920 7061 6972 206f 6620 6f72 7468  ery pair of orth
+00017250: 6f6c 6f67 6f75 7320 6765 6e65 7320 6973  ologous genes is
+00017260: 2074 6865 2072 6563 6970 726f 6361 6c20   the reciprocal 
+00017270: 6265 7374 2068 6974 2c20 7375 6767 6573  best hit, sugges
+00017280: 7469 6e67 2074 7275 6520 6f72 7468 6f6c  ting true orthol
+00017290: 6f67 6f75 7320 7265 6c61 7469 6f6e 7368  ogous relationsh
+000172a0: 6970 732e 2057 6520 776f 756c 6420 7573  ips. We would us
+000172b0: 6520 7468 6520 2a4b 2a3c 7375 623e 533c  e the *K*<sub>S<
+000172c0: 2f73 7562 3e20 7661 6c75 6573 2061 7373  /sub> values ass
+000172d0: 6f63 6961 7465 6420 7769 7468 2074 6865  ociated with the
+000172e0: 7365 206f 7274 686f 6c6f 676f 7573 2070  se orthologous p
+000172f0: 6169 7273 2074 6f20 6465 6c69 6d69 7420  airs to delimit 
+00017300: 7468 6520 6469 7665 7267 656e 6365 202a  the divergence *
+00017310: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
+00017320: 6561 6b2e 2054 6f67 6574 6865 7220 7769  eak. Together wi
+00017330: 7468 2074 6865 2077 686f 6c65 2070 6172  th the whole par
+00017340: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+00017350: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00017360: 6f6e 2c20 7765 2063 6f6e 6475 6374 2074  on, we conduct t
+00017370: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+00017380: 6f6e 2075 7369 6e67 2074 6865 2063 6f6d  on using the com
+00017390: 6d61 6e64 2062 656c 6f77 2e0a 0a21 2153  mand below...!!S
+000173a0: 696e 6365 2060 7767 6460 2076 6572 7369  ince `wgd` versi
+000173b0: 6f6e 2032 2e30 2e32 342c 2077 6520 7265  on 2.0.24, we re
+000173c0: 7772 6f74 6520 6120 636c 6561 6e65 7220  wrote a cleaner 
+000173d0: 616e 6420 7175 6963 6b65 7220 7761 7920  and quicker way 
+000173e0: 6f66 2064 6f69 6e67 2073 7562 7374 6974  of doing substit
+000173f0: 7574 696f 6e20 7261 7465 2063 6f72 7265  ution rate corre
+00017400: 6374 696f 6e2e 2049 7427 7320 6e6f 7420  ction. It's not 
+00017410: 7265 7175 6972 6564 2074 6f20 7479 7065  required to type
+00017420: 2069 6e20 616e 7920 7370 6569 6365 7320   in any speices 
+00017430: 7061 6972 2061 6e64 2061 2073 6572 6965  pair and a serie
+00017440: 7320 6f66 202a 4b2a 3c73 7562 3e53 3c2f  s of *K*<sub>S</
+00017450: 7375 623e 2070 6c6f 7473 2077 696c 6c20  sub> plots will 
+00017460: 6265 2070 726f 6475 6365 642e 2054 6865  be produced. The
+00017470: 2072 6571 7569 7265 6420 6669 6c65 7320   required files 
+00017480: 6172 6520 6f72 7468 6f6c 6f67 6f75 7320  are orthologous 
+00017490: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000174a0: 6461 7461 6669 6c65 2c20 7061 7261 6c6f  datafile, paralo
+000174b0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+000174c0: 7375 623e 2064 6174 6166 696c 652c 2061  sub> datafile, a
+000174d0: 2073 7065 6369 6573 2074 7265 6520 616e   species tree an
+000174e0: 6420 6120 666f 6361 6c20 7370 6563 6965  d a focal specie
+000174f0: 7320 2874 6865 206f 6e65 2069 6e70 7574  s (the one input
+00017500: 7465 6420 7769 7468 2070 6172 616c 6f67  ted with paralog
+00017510: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00017520: 7562 3e20 6461 7461 292e 2055 7365 7273  ub> data). Users
+00017530: 2063 616e 2063 686f 6f73 6520 746f 2061   can choose to a
+00017540: 6464 206f 6e65 206d 6f72 6520 6c61 7965  dd one more laye
+00017550: 7220 6f66 2045 4c4d 4d20 616e 616c 7973  r of ELMM analys
+00017560: 6973 206f 6e20 7061 7261 6c6f 676f 7573  is on paralogous
+00017570: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00017580: 2076 616c 7565 7320 616e 642f 6f72 2047   values and/or G
+00017590: 4d4d 2061 6e61 6c79 7369 7320 6f6e 2061  MM analysis on a
+000175a0: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
+000175b0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+000175c0: 6f6e 2e20 5468 6520 6f72 7468 6f6c 6f67  on. The ortholog
+000175d0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+000175e0: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
+000175f0: 2063 616e 2062 6520 6361 6c63 756c 6174   can be calculat
+00017600: 6564 2075 7369 6e67 2074 6865 2063 6f6d  ed using the com
+00017610: 6d61 6e64 2062 656c 6f77 2e0a 0a60 6060  mand below...```
+00017620: 0a77 6764 206b 7364 2077 6764 5f67 6c6f  .wgd ksd wgd_glo
+00017630: 6261 6c6d 7262 682f 676c 6f62 616c 5f4d  balmrbh/global_M
+00017640: 5242 482e 7473 7620 4171 7569 6c65 6769  RBH.tsv Aquilegi
+00017650: 615f 636f 6572 756c 6561 2050 726f 7465  a_coerulea Prote
+00017660: 615f 6379 6e61 726f 6964 6573 2041 636f  a_cynaroides Aco
+00017670: 7275 735f 616d 6572 6963 616e 7573 2056  rus_americanus V
+00017680: 6974 6973 5f76 696e 6966 6572 6120 2d6f  itis_vinifera -o
+00017690: 2077 6764 5f67 6c6f 6261 6c6d 7262 685f   wgd_globalmrbh_
+000176a0: 6b73 0a60 6060 0a0a 5769 7468 2074 6865  ks.```..With the
+000176b0: 2063 616c 6375 6c61 7465 6420 6f72 7468   calculated orth
+000176c0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000176d0: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+000176e0: 7469 6f6e 2c20 7765 2063 616e 2075 7365  tion, we can use
+000176f0: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
+00017700: 6f77 2074 6f20 636f 6e64 7563 7420 7468  ow to conduct th
+00017710: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
+00017720: 6e20 616e 642f 6f72 206d 6978 7475 7265  n and/or mixture
+00017730: 206d 6f64 656c 696e 6720 616e 616c 7973   modeling analys
+00017740: 6973 2e0a 0a60 6060 0a77 6764 2076 697a  is...```.wgd viz
+00017750: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
+00017760: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
+00017770: 482e 7473 762e 6b73 2e74 7376 202d 6661  H.tsv.ks.tsv -fa
+00017780: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+00017790: 6c65 6120 2d65 706b 2077 6764 5f6b 7364  lea -epk wgd_ksd
+000177a0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000177b0: 6c65 612e 6b73 2e74 7376 202d 6170 2077  lea.ks.tsv -ap w
+000177c0: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+000177d0: 7574 2f61 6e63 686f 7270 6f69 6e74 732e  ut/anchorpoints.
+000177e0: 7478 7420 2d73 7020 7370 6563 6965 7374  txt -sp speciest
+000177f0: 7265 652e 6e77 202d 6f20 7767 645f 7669  ree.nw -o wgd_vi
+00017800: 7a5f 6d69 7865 645f 4b73 202d 2d70 6c6f  z_mixed_Ks --plo
+00017810: 7465 6c6d 6d20 2d2d 706c 6f74 6170 676d  telmm --plotapgm
+00017820: 6d20 2d2d 7265 7765 6967 6874 0a60 6060  m --reweight.```
+00017830: 0a0a 6f72 2075 7369 6e67 2074 6865 2063  ..or using the c
+00017840: 6f6d 6d61 6e64 2062 656c 6f77 2c20 7768  ommand below, wh
+00017850: 6963 6820 636f 6d62 696e 6573 2074 6865  ich combines the
+00017860: 2074 776f 2073 7465 7073 2061 626f 7665   two steps above
+00017870: 2069 6e20 6f6e 652e 204e 6f74 6520 7468   in one. Note th
+00017880: 6174 2077 6520 7375 6767 6573 7420 6f66  at we suggest of
+00017890: 2074 616b 696e 6720 7477 6f20 7365 7061   taking two sepa
+000178a0: 7261 7465 2073 7465 7073 2069 6e20 7768  rate steps in wh
+000178b0: 6963 6820 6077 6764 206b 7364 6020 756e  ich `wgd ksd` un
+000178c0: 6465 7274 616b 6573 2074 6865 2063 616c  dertakes the cal
+000178d0: 6375 6c61 7469 6f6e 206f 6620 6f72 7468  culation of orth
+000178e0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+000178f0: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00017900: 7469 6f6e 2077 6869 6c65 2060 7767 6420  tion while `wgd 
+00017910: 7669 7a60 2063 6172 7269 6573 206f 7574  viz` carries out
+00017920: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+00017930: 7469 6f6e 2061 6e64 2047 4d4d 2061 6e61  tion and GMM ana
+00017940: 6c79 7369 7320 7375 6368 2074 6861 7420  lysis such that 
+00017950: 6974 2773 2065 6173 6965 7220 746f 2064  it's easier to d
+00017960: 6562 7567 2e0a 6060 600a 7767 6420 6b73  ebug..```.wgd ks
+00017970: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
+00017980: 2f67 6c6f 6261 6c5f 4d52 4248 2e74 7376  /global_MRBH.tsv
+00017990: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+000179a0: 6c65 6120 5072 6f74 6561 5f63 796e 6172  lea Protea_cynar
+000179b0: 6f69 6465 7320 4163 6f72 7573 5f61 6d65  oides Acorus_ame
+000179c0: 7269 6361 6e75 7320 5669 7469 735f 7669  ricanus Vitis_vi
+000179d0: 6e69 6665 7261 202d 2d65 7874 7261 7061  nifera --extrapa
+000179e0: 7261 6e6f 6d65 6b73 2077 6764 5f6b 7364  ranomeks wgd_ksd
+000179f0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00017a00: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00017a10: 7370 2073 7065 6369 6573 7472 6565 2e6e  sp speciestree.n
+00017a20: 7720 2d2d 7265 7765 6967 6874 202d 6f20  w --reweight -o 
+00017a30: 7767 645f 676c 6f62 616c 6d72 6268 5f6b  wgd_globalmrbh_k
+00017a40: 735f 7261 7465 5f63 6f72 7265 6374 696f  s_rate_correctio
+00017a50: 6e20 2d66 6120 4171 7569 6c65 6769 615f  n -fa Aquilegia_
+00017a60: 636f 6572 756c 6561 202d 6170 2077 6764  coerulea -ap wgd
+00017a70: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
+00017a80: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
+00017a90: 7420 2d2d 706c 6f74 656c 6d6d 202d 2d70  t --plotelmm --p
+00017aa0: 6c6f 7461 7067 6d6d 0a60 6060 0a0a 5468  lotapgmm.```..Th
+00017ab0: 6520 6669 6c65 2060 7370 6563 6965 7374  e file `speciest
+00017ac0: 7265 652e 6e77 6020 6973 2074 6865 2074  ree.nw` is the t
+00017ad0: 6578 7420 6669 6c65 206f 6620 7370 6563  ext file of spec
+00017ae0: 6965 7320 7472 6565 2069 6e20 6e65 7769  ies tree in newi
+00017af0: 636b 2074 6861 7420 7261 7465 2063 6f72  ck that rate cor
+00017b00: 7265 6374 696f 6e20 776f 756c 6420 6265  rection would be
+00017b10: 2063 6f6e 6475 6374 6564 206f 6e2e 2049   conducted on. I
+00017b20: 7473 2063 6f6e 7465 6e74 2069 7320 6173  ts content is as
+00017b30: 2062 656c 6f77 2e20 5573 6572 7320 6361   below. Users ca
+00017b40: 6e20 6f70 7469 6f6e 616c 6c79 2070 726f  n optionally pro
+00017b50: 7669 6465 2074 6865 2073 7065 6369 6573  vide the species
+00017b60: 2070 6169 7273 2074 6f20 6265 2070 6c6f   pairs to be plo
+00017b70: 7474 6564 2062 7574 2077 6520 7375 6767  tted but we sugg
+00017b80: 6573 7420 6f66 206a 7573 7420 7573 696e  est of just usin
+00017b90: 6720 602d 6661 2041 7175 696c 6567 6961  g `-fa Aquilegia
+00017ba0: 5f63 6f65 7275 6c65 6160 2074 6f20 706c  _coerulea` to pl
+00017bb0: 6f74 2061 6c6c 2070 6f73 7369 626c 6520  ot all possible 
+00017bc0: 666f 6361 6c2d 7369 7374 6572 2073 7065  focal-sister spe
+00017bd0: 6369 6573 2070 6169 7273 2e20 5765 2073  cies pairs. We s
+00017be0: 7567 6765 7374 2061 6464 696e 6720 7468  uggest adding th
+00017bf0: 6520 6f70 7469 6f6e 2060 2d2d 7265 7765  e option `--rewe
+00017c00: 6967 6874 6020 746f 2072 6563 616c 6375  ight` to recalcu
+00017c10: 6c61 7465 2074 6865 2077 6569 6768 7420  late the weight 
+00017c20: 7065 7220 7370 6563 6965 7320 7061 6972  per species pair
+00017c30: 2073 7563 6820 7468 6174 2074 6865 2077   such that the w
+00017c40: 6569 6768 7420 6f66 206f 7274 686f 6c6f  eight of ortholo
+00017c50: 676f 7573 2067 656e 6520 7061 6972 7320  gous gene pairs 
+00017c60: 7769 6c6c 2062 6563 6f6d 6520 312e 2045  will become 1. E
+00017c70: 7874 7261 2063 6f6c 6c69 6e65 6172 2064  xtra collinear d
+00017c80: 6174 6120 6361 6e20 6265 2061 6464 6564  ata can be added
+00017c90: 2062 7920 7468 6520 6f70 7469 6f6e 2060   by the option `
+00017ca0: 2d61 7060 2061 6e64 2061 6464 6974 696f  -ap` and additio
+00017cb0: 6e61 6c20 636c 7573 7465 7269 6e67 2061  nal clustering a
+00017cc0: 6e61 6c79 7369 7320 6361 6e20 6265 2069  nalysis can be i
+00017cd0: 6e69 7469 6174 6564 2062 7920 7365 7474  nitiated by sett
+00017ce0: 696e 6720 7468 6520 6f70 7469 6f6e 2060  ing the option `
+00017cf0: 2d2d 706c 6f74 6170 676d 6d60 2061 6e64  --plotapgmm` and
+00017d00: 2060 2d2d 706c 6f74 656c 6d6d 602e 0a0a   `--plotelmm`...
+00017d10: 6060 600a 2828 2856 6974 6973 5f76 696e  ```.(((Vitis_vin
+00017d20: 6966 6572 612c 5072 6f74 6561 5f63 796e  ifera,Protea_cyn
+00017d30: 6172 6f69 6465 7329 2c41 7175 696c 6567  aroides),Aquileg
+00017d40: 6961 5f63 6f65 7275 6c65 6129 2c41 636f  ia_coerulea),Aco
+00017d50: 7275 735f 616d 6572 6963 616e 7573 293b  rus_americanus);
+00017d60: 0a60 6060 0a0a 215b 5d28 6461 7461 2f4d  .```..![](data/M
+00017d70: 6978 6564 2e6b 732e 4171 7569 6c65 6769  ixed.ks.Aquilegi
+00017d80: 615f 636f 6572 756c 6561 2e6e 6f64 652e  a_coerulea.node.
+00017d90: 7765 6967 6874 6564 2e70 6e67 290a 0a54  weighted.png)..T
+00017da0: 6865 206d 6978 6564 202a 4b2a 3c73 7562  he mixed *K*<sub
+00017db0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00017dc0: 7574 696f 6e20 7368 6f77 6e20 6162 6f76  ution shown abov
+00017dd0: 6520 6973 2061 2070 7562 6c69 6361 7469  e is a publicati
+00017de0: 6f6e 2d72 6561 6479 2066 6967 7572 6520  on-ready figure 
+00017df0: 7468 6174 2061 7373 656d 626c 6573 2074  that assembles t
+00017e00: 6865 2072 6573 756c 7473 206f 6620 454c  he results of EL
+00017e10: 4d4d 2c20 474d 4d20 616e 6420 7261 7465  MM, GMM and rate
+00017e20: 2063 6f72 7265 6374 696f 6e2e 2054 6865   correction. The
+00017e30: 206f 6e65 2d76 732d 6f6e 6520 6f72 7468   one-vs-one orth
+00017e40: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00017e50: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00017e60: 7469 6f6e 7320 6973 2061 6c73 6f20 6175  tions is also au
+00017e70: 746f 6d61 7469 6361 6c6c 7920 7072 6f64  tomatically prod
+00017e80: 7563 6564 2077 6974 6820 7261 7465 2063  uced with rate c
+00017e90: 6f72 7265 6374 696f 6e20 7265 7375 6c74  orrection result
+00017ea0: 7320 7375 7065 7269 6d70 6f73 6564 2077  s superimposed w
+00017eb0: 6865 7265 2061 7661 696c 6162 6c65 2061  here available a
+00017ec0: 7320 7368 6f77 6e20 6265 6c6f 772e 0a0a  s shown below...
+00017ed0: 215b 5d28 6461 7461 2f46 6f63 7573 5f73  ![](data/Focus_s
+00017ee0: 6973 7465 725f 7061 6972 732e 6b73 2e6e  ister_pairs.ks.n
+00017ef0: 6f64 652e 7765 6967 6874 6564 2e70 6e67  ode.weighted.png
+00017f00: 290a 215b 5d28 6461 7461 2f41 6c6c 5f70  ).![](data/All_p
+00017f10: 6169 7273 2e6b 732e 6e6f 6465 2e77 6569  airs.ks.node.wei
+00017f20: 6768 7465 642e 706e 6729 0a0a 4265 7369  ghted.png)..Besi
+00017f30: 6465 7320 7468 6520 6162 6f76 6520 6b73  des the above ks
+00017f40: 7261 7465 732d 6c69 6b65 202a 4b2a 3c73  rates-like *K*<s
+00017f50: 7562 3e53 3c2f 7375 623e 2070 6c6f 742c  ub>S</sub> plot,
+00017f60: 2061 206d 6f72 6520 636c 6173 7369 6320   a more classic 
+00017f70: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00017f80: 706c 6f74 2063 616e 2062 6520 6d61 6465  plot can be made
+00017f90: 2062 7920 6164 6469 6e67 2074 6865 206f   by adding the o
+00017fa0: 7074 696f 6e20 602d 2d63 6c61 7373 6963  ption `--classic
+00017fb0: 6020 746f 2074 6170 206d 6f72 6520 6465  ` to tap more de
+00017fc0: 7461 696c 6564 6c79 2069 6e74 6f20 7468  tailedly into th
+00017fd0: 6520 7661 7269 6174 696f 6e20 6f66 2073  e variation of s
+00017fe0: 796e 6f6e 796d 6f75 7320 7375 6273 7469  ynonymous substi
+00017ff0: 7475 7469 6f6e 2072 6174 652e 0a0a 5573  tution rate...Us
+00018000: 696e 6720 7468 6520 636f 6d6d 616e 6420  ing the command 
+00018010: 6265 6c6f 772c 2074 6865 2064 6972 6563  below, the direc
+00018020: 7469 6f6e 206f 6620 7261 7465 2063 6f72  tion of rate cor
+00018030: 7265 6374 696f 6e20 616e 6420 6465 6772  rection and degr
+00018040: 6565 206f 6620 7261 7465 2076 6172 6961  ee of rate varia
+00018050: 7469 6f6e 2063 616e 2062 6520 6f62 7365  tion can be obse
+00018060: 7276 6564 206d 6f72 6520 6469 7265 6374  rved more direct
+00018070: 6c79 2e0a 0a60 6060 0a77 6764 2076 697a  ly...```.wgd viz
+00018080: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
+00018090: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
+000180a0: 482e 7473 762e 6b73 2e74 7376 202d 6661  H.tsv.ks.tsv -fa
+000180b0: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+000180c0: 6c65 6120 2d65 706b 2077 6764 5f6b 7364  lea -epk wgd_ksd
+000180d0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000180e0: 6c65 612e 6b73 2e74 7376 202d 6170 2077  lea.ks.tsv -ap w
+000180f0: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+00018100: 7574 2f61 6e63 686f 7270 6f69 6e74 732e  ut/anchorpoints.
+00018110: 7478 7420 2d73 7020 7370 6563 6965 7374  txt -sp speciest
+00018120: 7265 652e 6e77 202d 6f20 7767 645f 7669  ree.nw -o wgd_vi
+00018130: 7a5f 6d69 7865 645f 4b73 202d 2d70 6c6f  z_mixed_Ks --plo
+00018140: 7465 6c6d 6d20 2d2d 706c 6f74 6170 676d  telmm --plotapgm
+00018150: 6d20 2d2d 7265 7765 6967 6874 202d 2d70  m --reweight --p
+00018160: 6c6f 746b 6465 202d 2d63 6c61 7373 6963  lotkde --classic
+00018170: 0a60 6060 0a0a 215b 5d28 6461 7461 2f41  .```..![](data/A
+00018180: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00018190: 615f 476c 6f62 616c 6d72 6268 4b73 5f45  a_GlobalmrbhKs_E
+000181a0: 6c6d 6d5f 4170 676d 6d5f 436f 7272 6563  lmm_Apgmm_Correc
+000181b0: 7465 642e 6b73 642e 7376 6729 0a0a 4173  ted.ksd.svg)..As
+000181c0: 2073 686f 776e 2061 626f 7665 2c20 6265   shown above, be
+000181d0: 6361 7573 6520 6f66 2074 6865 2068 6967  cause of the hig
+000181e0: 6865 7220 7375 6273 7469 7475 7469 6f6e  her substitution
+000181f0: 2072 6174 6520 6f66 202a 4171 7569 6c65   rate of *Aquile
+00018200: 6769 6120 636f 6572 756c 6561 2a2c 2074  gia coerulea*, t
+00018210: 6865 206f 7269 6769 6e61 6c20 6f72 7468  he original orth
+00018220: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00018230: 533c 2f73 7562 3e20 7661 6c75 6573 2077  S</sub> values w
+00018240: 6572 6520 6163 7475 616c 6c79 2075 6e64  ere actually und
+00018250: 6572 6573 7469 6d61 7465 6420 696e 2074  erestimated in t
+00018260: 6865 2074 696d 652d 6672 616d 6520 6f66  he time-frame of
+00018270: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
+00018280: 756c 6561 2a2e 2057 6865 6e20 7765 2072  ulea*. When we r
+00018290: 6563 6f76 6572 6564 2074 6865 2064 6976  ecovered the div
+000182a0: 6572 6765 6e63 6520 7375 6273 7469 7475  ergence substitu
+000182b0: 7469 6f6e 2064 6973 7461 6e63 6520 696e  tion distance in
+000182c0: 2074 6572 6d73 206f 6620 7477 6f20 7469   terms of two ti
+000182d0: 6d65 7320 6f66 2074 6865 2062 7261 6e63  mes of the branc
+000182e0: 682d 7370 6563 6966 6963 2063 6f6e 7472  h-specific contr
+000182f0: 6962 7574 696f 6e20 6f66 202a 412e 2063  ibution of *A. c
+00018300: 6f65 7275 6c65 612a 2073 696e 6365 2069  oerulea* since i
+00018310: 7473 2064 6976 6572 6765 6e63 6520 7769  ts divergence wi
+00018320: 7468 2074 6865 2073 6973 7465 7220 7370  th the sister sp
+00018330: 6563 6965 7320 706c 7573 2074 6865 2073  ecies plus the s
+00018340: 6861 7265 6420 7375 6273 7469 7475 7469  hared substituti
+00018350: 6f6e 2064 6973 7461 6e63 6520 6265 666f  on distance befo
+00018360: 7265 2064 6976 6572 6765 6e63 6520 2869  re divergence (i
+00018370: 6e20 7265 6c61 7469 7665 2074 6f20 7468  n relative to th
+00018380: 6520 6f75 7467 726f 7570 292c 2074 6865  e outgroup), the
+00018390: 2063 6f72 7265 6374 6564 202a 4b2a 3c73   corrected *K*<s
+000183a0: 7562 3e53 3c2f 7375 623e 206d 6f64 6520  ub>S</sub> mode 
+000183b0: 6265 6361 6d65 206c 6172 6765 722e 0a0a  became larger...
+000183c0: 4e6f 7465 2074 6861 7420 7765 2063 616e  Note that we can
+000183d0: 2065 6173 696c 7920 7368 6f77 2074 6861   easily show tha
+000183e0: 7420 2a41 7175 696c 6567 6961 2063 6f65  t *Aquilegia coe
+000183f0: 7275 6c65 612a 2068 6173 2068 6967 6865  rulea* has highe
+00018400: 7220 7375 6273 7469 7475 7469 6f6e 2072  r substitution r
+00018410: 6174 6520 7468 616e 202a 5072 6f74 6561  ate than *Protea
+00018420: 2063 796e 6172 6f69 6465 732a 2061 6e64   cynaroides* and
+00018430: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
+00018440: 2a20 6279 2063 6f6d 7061 7269 6e67 2074  * by comparing t
+00018450: 6865 6972 2073 7562 7374 6974 7574 696f  heir substitutio
+00018460: 6e20 6469 7374 616e 6365 2069 6e20 7265  n distance in re
+00018470: 6761 7264 2074 6f20 7468 6520 7361 6d65  gard to the same
+00018480: 2064 6976 6572 6765 6e63 6520 6576 656e   divergence even
+00018490: 7420 7769 7468 206f 7574 6772 6f75 7020  t with outgroup 
+000184a0: 7370 6563 6965 7320 2a41 636f 7275 735f  species *Acorus_
+000184b0: 616d 6572 6963 616e 7573 2a2c 2075 7369  americanus*, usi
+000184c0: 6e67 2063 6f6d 6d61 6e64 2062 656c 6f77  ng command below
+000184d0: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
+000184e0: 6420 7767 645f 676c 6f62 616c 6d72 6268  d wgd_globalmrbh
+000184f0: 5f6b 732f 676c 6f62 616c 5f4d 5242 482e  _ks/global_MRBH.
+00018500: 7473 762e 6b73 2e74 7376 202d 7370 2073  tsv.ks.tsv -sp s
+00018510: 7065 6369 6573 7472 6565 2e6e 7720 2d2d  peciestree.nw --
+00018520: 7265 7765 6967 6874 202d 6f20 7767 645f  reweight -o wgd_
+00018530: 7669 7a5f 436f 6d70 6172 655f 7261 7465  viz_Compare_rate
+00018540: 202d 2d73 7061 6972 2022 4163 6f72 7573   --spair "Acorus
+00018550: 5f61 6d65 7269 6361 6e75 733b 5072 6f74  _americanus;Prot
+00018560: 6561 5f63 796e 6172 6f69 6465 7322 202d  ea_cynaroides" -
+00018570: 2d73 7061 6972 2022 4171 7569 6c65 6769  -spair "Aquilegi
+00018580: 615f 636f 6572 756c 6561 3b41 636f 7275  a_coerulea;Acoru
+00018590: 735f 616d 6572 6963 616e 7573 2220 2d2d  s_americanus" --
+000185a0: 7370 6169 7220 2256 6974 6973 5f76 696e  spair "Vitis_vin
+000185b0: 6966 6572 613b 4163 6f72 7573 5f61 6d65  ifera;Acorus_ame
+000185c0: 7269 6361 6e75 7322 202d 2d70 6c6f 746b  ricanus" --plotk
+000185d0: 6465 202d 2d63 6c61 7373 6963 0a60 6060  de --classic.```
+000185e0: 0a0a 215b 5d28 6461 7461 2f52 6177 5f4f  ..![](data/Raw_O
+000185f0: 7274 686f 6c6f 6775 6573 5f43 6f6d 7061  rthologues_Compa
+00018600: 7265 5f72 6174 652e 6b73 642e 7376 6729  re_rate.ksd.svg)
+00018610: 0a0a 4173 2064 6973 706c 6179 6564 2061  ..As displayed a
+00018620: 626f 7665 2c20 7468 6520 6f72 7468 6f6c  bove, the orthol
+00018630: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+00018640: 2f73 7562 3e20 7661 6c75 6573 2062 6577  /sub> values bew
+00018650: 7465 656e 202a 4171 7569 6c65 6769 6120  teen *Aquilegia 
+00018660: 636f 6572 756c 6561 2a20 616e 6420 2a41  coerulea* and *A
+00018670: 636f 7275 7320 616d 6572 6963 616e 7573  corus americanus
+00018680: 2a20 6861 7320 7468 6520 6869 6768 6573  * has the highes
+00018690: 7420 6d6f 6465 2c20 696e 6469 6361 7469  t mode, indicati
+000186a0: 6e67 2074 6865 2066 6173 7465 7220 7375  ng the faster su
+000186b0: 6273 7469 7475 7469 6f6e 2072 6174 6520  bstitution rate 
+000186c0: 6f66 202a 412e 2063 6f65 7275 6c65 612a  of *A. coerulea*
+000186d0: 2063 6f6d 7061 7265 6420 746f 202a 5072   compared to *Pr
+000186e0: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
+000186f0: 2061 6e64 202a 5669 7469 7320 7669 6e69   and *Vitis vini
+00018700: 6665 7261 2a2e 0a0a 4265 666f 7265 2076  fera*...Before v
+00018710: 322e 302e 3231 2c20 7468 6520 6765 6e65  2.0.21, the gene
+00018720: 2d73 7065 6369 6573 206d 6170 2066 696c  -species map fil
+00018730: 6520 6973 206e 6563 6365 7373 6172 696c  e is neccessaril
+00018740: 7920 6e65 6564 6564 2066 6f72 2069 7473  y needed for its
+00018750: 2069 6d70 6c65 6d65 6e74 6174 696f 6e20   implementation 
+00018760: 696e 2060 7767 6420 7669 7a60 2c20 7768  in `wgd viz`, wh
+00018770: 6963 6820 7368 6f75 6c64 2062 6520 6175  ich should be au
+00018780: 746f 6d61 7465 6c79 2070 726f 6475 6365  tomately produce
+00018790: 6420 6279 2074 6865 206c 6173 7420 6077  d by the last `w
+000187a0: 6764 206b 7364 6020 7374 6570 2067 6976  gd ksd` step giv
+000187b0: 656e 2074 6865 2060 7370 6169 7260 2061  en the `spair` a
+000187c0: 6e64 2060 7370 6563 6965 7374 7265 6560  nd `speciestree`
+000187d0: 2070 6172 616d 6574 6572 732e 2054 6865   parameters. The
+000187e0: 2060 6765 6e65 5f73 7065 6369 6573 2e6d   `gene_species.m
+000187f0: 6170 6020 6861 7320 636f 6e74 656e 7473  ap` has contents
+00018800: 2061 7320 6265 6c6f 7720 696e 2077 6869   as below in whi
+00018810: 6368 2065 6163 6820 6c69 6e65 2069 7320  ch each line is 
+00018820: 7468 6520 6a6f 696e 6564 2073 7472 696e  the joined strin
+00018830: 6720 6f66 2067 656e 6520 6e61 6d65 2061  g of gene name a
+00018840: 6e64 2073 7065 6369 6573 206e 616d 6520  nd species name 
+00018850: 6279 2073 7061 6365 2e20 4166 7465 7220  by space. After 
+00018860: 7632 2e30 2e32 3120 2869 6e63 6c75 6465  v2.0.21 (include
+00018870: 6429 2c20 7468 6520 6765 6e65 2d73 7065  d), the gene-spe
+00018880: 6369 6573 206d 6170 2066 696c 6520 6973  cies map file is
+00018890: 206e 6f74 206e 6563 6365 7373 6172 696c   not neccessaril
+000188a0: 7920 6e65 6564 6564 2061 6e79 6d6f 7265  y needed anymore
+000188b0: 2e0a 0a60 6060 0a41 7163 6f65 3647 3035  ...```.Aqcoe6G05
+000188c0: 3738 3030 2e31 2041 7175 696c 6567 6961  7800.1 Aquilegia
+000188d0: 5f63 6f65 7275 6c65 610a 5676 695f 5649  _coerulea.Vvi_VI
+000188e0: 545f 3230 3173 3030 3131 6730 3135 3330  T_201s0011g01530
+000188f0: 2e31 2056 6974 6973 5f76 696e 6966 6572  .1 Vitis_vinifer
+00018900: 610a 5063 795f 5072 6f63 7930 3167 3038  a.Pcy_Procy01g08
+00018910: 3531 3020 5072 6f74 6561 5f63 796e 6172  510 Protea_cynar
+00018920: 6f69 6465 730a 4161 6d5f 4163 6f72 612e  oides.Aam_Acora.
+00018930: 3034 4731 3432 3930 302e 3120 4163 6f72  04G142900.1 Acor
+00018940: 7573 5f61 6d65 7269 6361 6e75 730a 6060  us_americanus.``
+00018950: 600a 0a41 6e20 616c 7465 726e 6174 6976  `..An alternativ
+00018960: 6520 7761 7920 746f 2063 616c 6375 6c61  e way to calcula
+00018970: 7465 2074 6865 206f 7274 686f 6c6f 676f  te the orthologo
+00018980: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+00018990: 623e 2069 7320 746f 2064 6972 6563 746c  b> is to directl
+000189a0: 7920 7573 6520 7468 6520 6f72 7468 6f67  y use the orthog
+000189b0: 726f 7570 7320 696e 7374 6561 6420 6f66  roups instead of
+000189c0: 2067 6c6f 6261 6c20 4d52 4248 2066 616d   global MRBH fam
+000189d0: 696c 792e 2054 6861 7420 7761 7920 7765  ily. That way we
+000189e0: 2064 6f6e 2774 2075 7365 2074 6865 2070   don't use the p
+000189f0: 7265 2d69 6e66 6572 7265 6420 7061 7261  re-inferred para
+00018a00: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+00018a10: 7375 623e 2062 7574 2074 6865 2070 6172  sub> but the par
+00018a20: 616c 6f67 6f75 7320 6765 6e65 2070 6169  alogous gene pai
+00018a30: 7273 2069 6e73 6964 6520 6561 6368 206f  rs inside each o
+00018a40: 7274 686f 6772 6f75 7020 696e 7374 6561  rthogroup instea
+00018a50: 642e 2054 6f20 6163 6869 6576 6520 7468  d. To achieve th
+00018a60: 6174 2c20 7765 2066 6972 7374 206e 6565  at, we first nee
+00018a70: 6420 746f 2069 6e66 6572 206f 7274 686f  d to infer ortho
+00018a80: 6772 6f75 7073 2075 7369 6e67 2074 6865  groups using the
+00018a90: 2063 6f6d 6d61 6e64 2062 656c 6f77 2e0a   command below..
+00018aa0: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
+00018ab0: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+00018ac0: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
+00018ad0: 7320 4163 6f72 7573 5f61 6d65 7269 6361  s Acorus_america
+00018ae0: 6e75 7320 5669 7469 735f 7669 6e69 6665  nus Vitis_vinife
+00018af0: 7261 202d 2d6f 7274 686f 696e 6665 7220  ra --orthoinfer 
+00018b00: 2d6f 2077 6764 5f6f 7274 686f 2028 2d2d  -o wgd_ortho (--
+00018b10: 6f6e 6c79 6f72 7468 6f29 200a 6060 600a  onlyortho) .```.
+00018b20: 0a55 7365 7273 2063 616e 2064 6563 6964  .Users can decid
+00018b30: 6520 746f 206f 6e6c 7920 636f 6e64 7563  e to only conduc
+00018b40: 7420 7468 6520 6f72 7468 6f67 726f 7570  t the orthogroup
+00018b50: 2061 6e61 6c79 7369 7320 7768 696c 6520   analysis while 
+00018b60: 736b 6970 7069 6e67 206f 7468 6572 2061  skipping other a
+00018b70: 6e61 6c79 7369 7320 6279 2061 6464 696e  nalysis by addin
+00018b80: 6720 7468 6520 666c 6167 2060 2d2d 6f6e  g the flag `--on
+00018b90: 6c79 6f72 7468 6f60 2e20 4e65 7874 2073  lyortho`. Next s
+00018ba0: 7465 7020 6973 2074 6865 2073 616d 6520  tep is the same 
+00018bb0: 7769 7468 2067 6c6f 6261 6c20 4d52 4248  with global MRBH
+00018bc0: 2066 616d 696c 792e 0a0a 6060 600a 7767   family...```.wg
+00018bd0: 6420 6b73 6420 7767 645f 6f72 7468 6f2f  d ksd wgd_ortho/
+00018be0: 4f72 7468 6f67 726f 7570 732e 7370 2e74  Orthogroups.sp.t
+00018bf0: 7376 2041 7175 696c 6567 6961 5f63 6f65  sv Aquilegia_coe
+00018c00: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
+00018c10: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
+00018c20: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
+00018c30: 7669 6e69 6665 7261 202d 6f20 7767 645f  vinifera -o wgd_
+00018c40: 6f72 7468 6f5f 6b73 0a77 6764 2076 697a  ortho_ks.wgd viz
+00018c50: 202d 6420 7767 645f 6f72 7468 6f5f 6b73   -d wgd_ortho_ks
+00018c60: 2f4f 7274 686f 6772 6f75 7073 2e73 702e  /Orthogroups.sp.
+00018c70: 7473 762e 6b73 2e74 7376 202d 6570 6b20  tsv.ks.tsv -epk 
+00018c80: 7767 645f 6b73 642f 4171 7569 6c65 6769  wgd_ksd/Aquilegi
+00018c90: 615f 636f 6572 756c 6561 2e6b 732e 7473  a_coerulea.ks.ts
+00018ca0: 7620 2d73 7020 7370 6563 6965 7374 7265  v -sp speciestre
+00018cb0: 652e 6e77 202d 2d72 6577 6569 6768 7420  e.nw --reweight 
+00018cc0: 2d61 7020 7767 645f 7379 6e2f 6961 6468  -ap wgd_syn/iadh
+00018cd0: 6f72 652d 6f75 742f 616e 6368 6f72 706f  ore-out/anchorpo
+00018ce0: 696e 7473 2e74 7874 202d 706c 6f74 656c  ints.txt -plotel
+00018cf0: 6d6d 202d 2d70 6c6f 7461 7067 6d6d 202d  mm --plotapgmm -
+00018d00: 6f20 7767 645f 6f72 7468 6f5f 6b73 5f72  o wgd_ortho_ks_r
+00018d10: 6174 655f 636f 7272 6563 7469 6f6e 0a60  ate_correction.`
+00018d20: 6060 0a0a 215b 5d28 6461 7461 2f4d 6978  ``..![](data/Mix
+00018d30: 6564 2e6b 732e 4171 7569 6c65 6769 615f  ed.ks.Aquilegia_
+00018d40: 636f 6572 756c 6561 2e6e 6f64 652e 7765  coerulea.node.we
+00018d50: 6967 6874 6564 5f4f 7274 686f 5f4b 732e  ighted_Ortho_Ks.
+00018d60: 706e 6729 0a21 5b5d 2864 6174 612f 466f  png).![](data/Fo
+00018d70: 6375 735f 7369 7374 6572 5f70 6169 7273  cus_sister_pairs
+00018d80: 2e6b 732e 6e6f 6465 2e77 6569 6768 7465  .ks.node.weighte
+00018d90: 645f 4f72 7468 6f5f 4b73 2e70 6e67 290a  d_Ortho_Ks.png).
+00018da0: 215b 5d28 6461 7461 2f41 6c6c 5f70 6169  ![](data/All_pai
+00018db0: 7273 2e6b 732e 6e6f 6465 2e77 6569 6768  rs.ks.node.weigh
+00018dc0: 7465 645f 4f72 7468 6f5f 4b73 2e70 6e67  ted_Ortho_Ks.png
+00018dd0: 290a 0a41 7320 7368 6f77 6e20 6162 6f76  )..As shown abov
+00018de0: 652c 2074 6865 206e 756d 6265 7220 6f66  e, the number of
+00018df0: 206f 7274 686f 6c6f 676f 7573 2067 656e   orthologous gen
+00018e00: 6520 7061 6972 7320 6973 2064 6966 6665  e pairs is diffe
+00018e10: 7265 6e74 2074 6861 6e20 7468 6520 6f6e  rent than the on
+00018e20: 6520 6672 6f6d 2067 6c6f 6261 6c20 4d52  e from global MR
+00018e30: 4248 2066 616d 696c 6965 7320 696e 2074  BH families in t
+00018e40: 6861 7420 6865 7265 2077 6520 706c 6f74  hat here we plot
+00018e50: 7465 6420 616c 6c20 6f72 7468 6f6c 6f67  ted all ortholog
+00018e60: 6f75 7320 6765 6e65 2070 6169 7273 2069  ous gene pairs i
+00018e70: 6e73 7465 6164 206f 6620 6f6e 6c79 2067  nstead of only g
+00018e80: 6c6f 6261 6c20 4d52 4248 2066 616d 696c  lobal MRBH famil
+00018e90: 6965 732c 2074 6f67 6574 6865 7220 7769  ies, together wi
+00018ea0: 7468 2064 6966 6665 7265 6e74 2072 6563  th different rec
+00018eb0: 616c 6375 6c61 7465 6420 7765 6967 6874  alculated weight
+00018ec0: 732e 0a0a 4166 7465 7220 7468 6520 7068  s...After the ph
+00018ed0: 796c 6f67 656e 6574 6963 2074 696d 696e  ylogenetic timin
+00018ee0: 6720 6f66 2074 6865 2052 616e 756e 6375  g of the Ranuncu
+00018ef0: 6c61 6c65 7320 5747 442c 2077 6520 6361  lales WGD, we ca
+00018f00: 6e20 6675 7274 6865 7220 696e 6665 7220  n further infer 
+00018f10: 6974 7320 6162 736f 6c75 7465 2061 6765  its absolute age
+00018f20: 2e20 4669 7273 7420 7765 2069 6e66 6572  . First we infer
+00018f30: 2074 6865 2063 7265 6469 626c 6520 7261   the credible ra
+00018f40: 6e67 6520 6f66 2061 6e63 686f 7220 7061  nge of anchor pa
+00018f50: 6972 7320 6279 202a 4b2a 3c73 7562 3e53  irs by *K*<sub>S
+00018f60: 3c2f 7375 623e 2068 6575 7269 7374 6963  </sub> heuristic
+00018f70: 616c 6c79 2075 7369 6e67 2074 6865 2070  ally using the p
+00018f80: 726f 6772 616d 2060 7767 6420 7065 616b  rogram `wgd peak
+00018f90: 602e 0a0a 6060 600a 7767 6420 7065 616b  `...```.wgd peak
+00018fa0: 202d 2d68 6575 7269 7374 6963 2077 6764   --heuristic wgd
+00018fb0: 5f6b 7364 2f41 7175 696c 6567 6961 5f63  _ksd/Aquilegia_c
+00018fc0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00018fd0: 7376 202d 6170 2077 6764 5f73 796e 2f69  sv -ap wgd_syn/i
+00018fe0: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+00018ff0: 7270 6f69 6e74 732e 7478 7420 2d73 6d20  rpoints.txt -sm 
+00019000: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
+00019010: 6f75 742f 7365 676d 656e 7473 2e74 7874  out/segments.txt
+00019020: 202d 6c65 2077 6764 5f73 796e 2f69 6164   -le wgd_syn/iad
+00019030: 686f 7265 2d6f 7574 2f6c 6973 745f 656c  hore-out/list_el
+00019040: 656d 656e 7473 2e74 7874 202d 6d70 2077  ements.txt -mp w
+00019050: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+00019060: 7574 2f6d 756c 7469 706c 6963 6f6e 5f70  ut/multiplicon_p
+00019070: 6169 7273 2e74 7874 202d 6f20 7767 645f  airs.txt -o wgd_
+00019080: 7065 616b 0a60 6060 0a0a 215b 5d28 6461  peak.```..![](da
+00019090: 7461 2f41 6e63 686f 724b 735f 5065 616b  ta/AnchorKs_Peak
+000190a0: 4349 5f41 7175 696c 6567 6961 5f63 6f65  CI_Aquilegia_coe
+000190b0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+000190c0: 5f6e 6f64 655f 7765 6967 6874 6564 2e73  _node_weighted.s
+000190d0: 7667 290a 0a41 7320 7368 6f77 6e20 6162  vg)..As shown ab
+000190e0: 6f76 652c 2077 6520 6173 7375 6d65 6420  ove, we assumed 
+000190f0: 6120 6c6f 676e 6f72 6d61 6c20 6469 7374  a lognormal dist
+00019100: 7269 6275 7469 6f6e 2061 7420 7468 6520  ribution at the 
+00019110: 7065 616b 206c 6f63 6174 696f 6e20 6465  peak location de
+00019120: 7465 6374 6564 2062 7920 7468 6520 6073  tected by the `s
+00019130: 6967 6e61 6c60 206d 6f64 756c 6520 6f66  ignal` module of
+00019140: 2060 7363 6970 7960 206c 6962 7261 7279   `scipy` library
+00019150: 2e20 5468 6520 3935 2520 636f 6e66 6964  . The 95% confid
+00019160: 656e 6365 206c 6576 656c 206f 6620 7468  ence level of th
+00019170: 6520 6c6f 676e 6f72 6d61 6c20 6469 7374  e lognormal dist
+00019180: 7269 6275 7469 6f6e 2077 6173 2061 7070  ribution was app
+00019190: 6c69 6564 2c20 692e 652e 2c20 302e 3638  lied, i.e., 0.68
+000191a0: 2d32 2e37 342c 2069 6e20 6675 7274 6865  -2.74, in furthe
+000191b0: 7220 6d6f 6c65 6375 6c61 7220 6461 7469  r molecular dati
+000191c0: 6e67 2e20 5468 6520 6669 6c65 2060 4171  ng. The file `Aq
+000191d0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000191e0: 2e74 7376 2e6b 732e 7473 765f 3935 2543  .tsv.ks.tsv_95%C
+000191f0: 495f 4150 5f66 6f72 5f64 6174 696e 675f  I_AP_for_dating_
+00019200: 7765 6967 6874 6564 5f66 6f72 6d61 742e  weighted_format.
+00019210: 7473 7660 2069 7320 7768 6174 2077 6520  tsv` is what we 
+00019220: 6e65 6564 2066 6f72 206e 6578 7420 7374  need for next st
+00019230: 6570 2e20 546f 2062 7569 6c64 2074 6865  ep. To build the
+00019240: 206f 7274 686f 6772 6f75 7073 2075 7365   orthogroups use
+00019250: 6420 696e 2070 6879 6c6f 6765 6e65 7469  d in phylogeneti
+00019260: 6320 6461 7469 6e67 2c20 7765 206e 6565  c dating, we nee
+00019270: 6420 746f 2073 656c 6563 7420 736f 6d65  d to select some
+00019280: 2073 7065 6369 6573 2061 6e64 2066 6f72   species and for
+00019290: 6d20 6120 7374 6172 7469 6e67 2074 7265  m a starting tre
+000192a0: 6520 7769 7468 2070 726f 7065 7220 666f  e with proper fo
+000192b0: 7373 696c 2063 616c 6962 7261 7469 6f6e  ssil calibration
+000192c0: 732e 2057 6520 7072 6f76 6964 6520 6f6e  s. We provide on
+000192d0: 6520 696e 206d 636d 6374 7265 6520 666f  e in mcmctree fo
+000192e0: 726d 6174 2061 7320 6265 6c6f 772e 0a0a  rmat as below...
+000192f0: 6060 600a 3137 2031 0a28 2828 2850 6f74  ```.17 1.((((Pot
+00019300: 616d 6f67 6574 6f6e 5f61 6375 7469 666f  amogeton_acutifo
+00019310: 6c69 7573 2c28 5370 6972 6f64 656c 615f  lius,(Spirodela_
+00019320: 696e 7465 726d 6564 6961 2c41 6d6f 7270  intermedia,Amorp
+00019330: 686f 7068 616c 6c75 735f 6b6f 6e6a 6163  hophallus_konjac
+00019340: 2929 2c28 4163 616e 7468 6f63 686c 616d  )),(Acanthochlam
+00019350: 7973 5f62 7261 6374 6561 7461 2c28 4469  ys_bracteata,(Di
+00019360: 6f73 636f 7265 615f 616c 6174 612c 4469  oscorea_alata,Di
+00019370: 6f73 636f 7265 615f 726f 7475 6e64 6174  oscorea_rotundat
+00019380: 6129 2927 3e30 2e35 3630 303c 312e 3238  a))'>0.5600<1.28
+00019390: 3633 2729 273e 302e 3833 3630 3c31 2e32  63')'>0.8360<1.2
+000193a0: 3836 3327 2c28 4163 6f72 7573 5f61 6d65  863',(Acorus_ame
+000193b0: 7269 6361 6e75 732c 4163 6f72 7573 5f74  ricanus,Acorus_t
+000193c0: 6174 6172 696e 6f77 6969 2929 273e 302e  atarinowii))'>0.
+000193d0: 3833 3630 3c31 2e32 3836 3327 2c28 2828  8360<1.2863',(((
+000193e0: 2854 6574 7261 6365 6e74 726f 6e5f 7369  (Tetracentron_si
+000193f0: 6e65 6e73 652c 5472 6f63 686f 6465 6e64  nense,Trochodend
+00019400: 726f 6e5f 6172 616c 696f 6964 6573 292c  ron_aralioides),
+00019410: 2842 7578 7573 5f61 7573 7472 6f79 756e  (Buxus_austroyun
+00019420: 6e61 6e65 6e73 6973 2c42 7578 7573 5f73  nanensis,Buxus_s
+00019430: 696e 6963 6129 2927 3e31 2e31 3038 303c  inica))'>1.1080<
+00019440: 312e 3238 3633 272c 284e 656c 756d 626f  1.2863',(Nelumbo
+00019450: 5f6e 7563 6966 6572 612c 2854 656c 6f70  _nucifera,(Telop
+00019460: 6561 5f73 7065 6369 6f73 6973 7369 6d61  ea_speciosissima
+00019470: 2c50 726f 7465 615f 6379 6e61 726f 6964  ,Protea_cynaroid
+00019480: 6573 2929 2927 3e31 2e31 3038 303c 312e  es)))'>1.1080<1.
+00019490: 3238 3633 272c 2841 7175 696c 6567 6961  2863',(Aquilegia
+000194a0: 5f63 6f65 7275 6c65 615f 6170 312c 4171  _coerulea_ap1,Aq
+000194b0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+000194c0: 5f61 7032 2929 273e 312e 3130 3830 3c31  _ap2))'>1.1080<1
+000194d0: 2e32 3836 3327 2927 3e31 2e32 3732 303c  .2863')'>1.2720<
+000194e0: 322e 3437 3230 273b 0a60 6060 0a0a 4173  2.4720';.```..As
+000194f0: 2070 7265 7365 6e74 6564 2061 626f 7665   presented above
+00019500: 2c20 7468 6520 666f 6375 7320 7370 6563  , the focus spec
+00019510: 6965 7320 7468 6174 2069 7320 6162 6f75  ies that is abou
+00019520: 7420 746f 2062 6520 6461 7465 6420 6e65  t to be dated ne
+00019530: 6564 7320 746f 2062 6520 7265 706c 6163  eds to be replac
+00019540: 6564 2077 6974 6820 6028 4171 7569 6c65  ed with `(Aquile
+00019550: 6769 615f 636f 6572 756c 6561 5f61 7031  gia_coerulea_ap1
+00019560: 2c41 7175 696c 6567 6961 5f63 6f65 7275  ,Aquilegia_coeru
+00019570: 6c65 615f 6170 3229 602e 2057 6974 6820  lea_ap2)`. With 
+00019580: 7468 6973 2073 7461 7274 696e 6720 7472  this starting tr
+00019590: 6565 2061 6e64 2070 7265 646f 776e 6c6f  ee and predownlo
+000195a0: 6164 6564 2063 6473 2066 696c 6573 206f  aded cds files o
+000195b0: 6620 616c 6c20 7468 6520 7370 6563 6965  f all the specie
+000195c0: 732c 2077 6520 6361 6e20 6275 696c 6420  s, we can build 
+000195d0: 7468 6520 6f72 7468 6f67 726f 7570 2075  the orthogroup u
+000195e0: 7365 6420 696e 2074 6865 2066 696e 616c  sed in the final
+000195f0: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
+00019600: 6720 7573 696e 6720 7468 6520 636f 6d6d  g using the comm
+00019610: 616e 6420 6173 2062 656c 6f77 2e0a 0a60  and as below...`
+00019620: 6060 0a77 6764 2064 6d64 202d 6620 4171  ``.wgd dmd -f Aq
+00019630: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00019640: 202d 6170 2077 6764 5f70 6561 6b2f 416e   -ap wgd_peak/An
+00019650: 6368 6f72 4b73 5f46 696e 6450 6561 6b2f  chorKs_FindPeak/
+00019660: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00019670: 6561 2e74 7376 2e6b 732e 7473 765f 3935  ea.tsv.ks.tsv_95
+00019680: 2543 495f 4150 5f66 6f72 5f64 6174 696e  %CI_AP_for_datin
+00019690: 675f 7765 6967 6874 6564 5f66 6f72 6d61  g_weighted_forma
+000196a0: 742e 7473 7620 2d6f 2077 6764 5f64 6d64  t.tsv -o wgd_dmd
+000196b0: 5f6f 7274 686f 2050 6f74 616d 6f67 6574  _ortho Potamoget
+000196c0: 6f6e 5f61 6375 7469 666f 6c69 7573 2053  on_acutifolius S
+000196d0: 7069 726f 6465 6c61 5f69 6e74 6572 6d65  pirodela_interme
+000196e0: 6469 6120 416d 6f72 7068 6f70 6861 6c6c  dia Amorphophall
+000196f0: 7573 5f6b 6f6e 6a61 6320 4163 616e 7468  us_konjac Acanth
+00019700: 6f63 686c 616d 7973 5f62 7261 6374 6561  ochlamys_bractea
+00019710: 7461 2044 696f 7363 6f72 6561 5f61 6c61  ta Dioscorea_ala
+00019720: 7461 2044 696f 7363 6f72 6561 5f72 6f74  ta Dioscorea_rot
+00019730: 756e 6461 7461 2041 636f 7275 735f 616d  undata Acorus_am
+00019740: 6572 6963 616e 7573 2041 636f 7275 735f  ericanus Acorus_
+00019750: 7461 7461 7269 6e6f 7769 6920 5465 7472  tatarinowii Tetr
+00019760: 6163 656e 7472 6f6e 5f73 696e 656e 7365  acentron_sinense
+00019770: 2054 726f 6368 6f64 656e 6472 6f6e 5f61   Trochodendron_a
+00019780: 7261 6c69 6f69 6465 7320 4275 7875 735f  ralioides Buxus_
+00019790: 6175 7374 726f 7975 6e6e 616e 656e 7369  austroyunnanensi
+000197a0: 7320 4275 7875 735f 7369 6e69 6361 204e  s Buxus_sinica N
+000197b0: 656c 756d 626f 5f6e 7563 6966 6572 6120  elumbo_nucifera 
+000197c0: 5465 6c6f 7065 615f 7370 6563 696f 7369  Telopea_speciosi
+000197d0: 7373 696d 6120 5072 6f74 6561 5f63 796e  ssima Protea_cyn
+000197e0: 6172 6f69 6465 7320 4171 7569 6c65 6769  aroides Aquilegi
+000197f0: 615f 636f 6572 756c 6561 0a60 6060 0a0a  a_coerulea.```..
+00019800: 5468 6520 7265 7375 6c74 2066 696c 6520  The result file 
+00019810: 606d 6572 6765 5f66 6f63 7573 5f61 702e  `merge_focus_ap.
+00019820: 7473 7660 2069 7320 7768 6174 2077 6520  tsv` is what we 
+00019830: 6e65 6564 2066 6f72 2074 6865 2066 696e  need for the fin
+00019840: 616c 2073 7465 7020 6f66 206d 6f6c 6563  al step of molec
+00019850: 756c 6172 2064 6174 696e 6720 696e 2070  ular dating in p
+00019860: 726f 6772 616d 2060 7767 6420 666f 6375  rogram `wgd focu
+00019870: 7360 2e0a 0a60 6060 0a77 6764 2066 6f63  s`...```.wgd foc
+00019880: 7573 202d 2d70 726f 7463 6f63 6461 7469  us --protcocdati
+00019890: 6e67 202d 2d61 616d 6f64 656c 206c 6720  ng --aamodel lg 
+000198a0: 7767 645f 646d 645f 6f72 7468 6f2f 6d65  wgd_dmd_ortho/me
+000198b0: 7267 655f 666f 6375 735f 6170 2e74 7376  rge_focus_ap.tsv
+000198c0: 202d 7370 2064 6174 696e 675f 7472 6565   -sp dating_tree
+000198d0: 2e6e 7720 2d6f 2077 6764 5f64 6174 696e  .nw -o wgd_datin
+000198e0: 6720 2d64 206d 636d 6374 7265 6520 2d64  g -d mcmctree -d
+000198f0: 7320 2762 7572 6e69 6e20 3d20 3230 3030  s 'burnin = 2000
+00019900: 2720 2d64 7320 2773 616d 7066 7265 7120  ' -ds 'sampfreq 
+00019910: 3d20 3130 3030 2720 2d64 7320 276e 7361  = 1000' -ds 'nsa
+00019920: 6d70 6c65 203d 2032 3030 3030 2720 506f  mple = 20000' Po
+00019930: 7461 6d6f 6765 746f 6e5f 6163 7574 6966  tamogeton_acutif
+00019940: 6f6c 6975 7320 5370 6972 6f64 656c 615f  olius Spirodela_
+00019950: 696e 7465 726d 6564 6961 2041 6d6f 7270  intermedia Amorp
+00019960: 686f 7068 616c 6c75 735f 6b6f 6e6a 6163  hophallus_konjac
+00019970: 2041 6361 6e74 686f 6368 6c61 6d79 735f   Acanthochlamys_
+00019980: 6272 6163 7465 6174 6120 4469 6f73 636f  bracteata Diosco
+00019990: 7265 615f 616c 6174 6120 4469 6f73 636f  rea_alata Diosco
+000199a0: 7265 615f 726f 7475 6e64 6174 6120 4163  rea_rotundata Ac
+000199b0: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
+000199c0: 4163 6f72 7573 5f74 6174 6172 696e 6f77  Acorus_tatarinow
+000199d0: 6969 2054 6574 7261 6365 6e74 726f 6e5f  ii Tetracentron_
+000199e0: 7369 6e65 6e73 6520 5472 6f63 686f 6465  sinense Trochode
+000199f0: 6e64 726f 6e5f 6172 616c 696f 6964 6573  ndron_aralioides
+00019a00: 2042 7578 7573 5f61 7573 7472 6f79 756e   Buxus_austroyun
+00019a10: 6e61 6e65 6e73 6973 2042 7578 7573 5f73  nanensis Buxus_s
+00019a20: 696e 6963 6120 4e65 6c75 6d62 6f5f 6e75  inica Nelumbo_nu
+00019a30: 6369 6665 7261 2054 656c 6f70 6561 5f73  cifera Telopea_s
+00019a40: 7065 6369 6f73 6973 7369 6d61 2050 726f  peciosissima Pro
+00019a50: 7465 615f 6379 6e61 726f 6964 6573 2041  tea_cynaroides A
+00019a60: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00019a70: 610a 6060 600a 0a48 6572 6520 7765 206f  a.```..Here we o
+00019a80: 6e6c 7920 696d 706c 656d 656e 7465 6420  nly implemented 
+00019a90: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+00019aa0: 6e20 616e 616c 7973 6973 2075 7369 6e67  n analysis using
+00019ab0: 2070 726f 7465 696e 2073 6571 7565 6e63   protein sequenc
+00019ac0: 6520 6279 2061 6464 696e 6720 7468 6520  e by adding the 
+00019ad0: 666c 6167 2060 2d2d 7072 6f74 6461 7469  flag `--protdati
+00019ae0: 6e67 6020 616e 6420 7765 2073 6574 2074  ng` and we set t
+00019af0: 6865 2070 6172 616d 6574 6572 2066 6f72  he parameter for
+00019b00: 2060 6d63 6d63 7472 6565 6020 7669 6120   `mcmctree` via 
+00019b10: 7468 6520 6f70 7469 6f6e 2060 2d64 7360  the option `-ds`
+00019b20: 2e20 4e6f 7465 2074 6861 7420 6f74 6865  . Note that othe
+00019b30: 7220 6461 7469 6e67 2070 726f 6772 616d  r dating program
+00019b40: 2073 7563 6820 6173 2060 7238 7360 2061   such as `r8s` a
+00019b50: 6e64 2060 6265 6173 7460 2061 7265 2061  nd `beast` are a
+00019b60: 6c73 6f20 6176 6169 6c61 626c 6520 6769  lso available gi
+00019b70: 7665 6e20 736f 6d65 206d 616e 6461 746f  ven some mandato
+00019b80: 7279 2070 6172 616d 6574 6572 732e 2054  ry parameters. T
+00019b90: 6865 2066 696e 616c 206c 6f67 206f 6620  he final log of 
+00019ba0: 7468 6520 7375 6363 6573 7366 756c 2072  the successful r
+00019bb0: 756e 2069 7320 6173 2062 656c 6f77 2e0a  un is as below..
+00019bc0: 0a60 6060 0a31 363a 3034 3a32 3520 494e  .```.16:04:25 IN
+00019bd0: 464f 2020 2020 2052 756e 6e69 6e67 206d  FO     Running m
+00019be0: 636d 6374 7265 6520 7573 696e 6720 4865  cmctree using He
+00019bf0: 7373 6961 6e20 6d61 7472 6978 206f 6620  ssian matrix of 
+00019c00: 4c47 2b47 616d 6d61 2020 636f 7265 2e70  LG+Gamma  core.p
+00019c10: 793a 3936 370a 2020 2020 2020 2020 2020  y:967.          
+00019c20: 2020 2020 2020 2020 666f 7220 7072 6f74          for prot
+00019c30: 6569 6e20 6d6f 6465 6c0a 3233 3a34 393a  ein model.23:49:
+00019c40: 3337 2049 4e46 4f20 2020 2020 506f 7374  37 INFO     Post
+00019c50: 6572 696f 7220 6d65 616e 2066 6f72 2074  erior mean for t
+00019c60: 6865 2061 6765 7320 6f66 2077 6764 2069  he ages of wgd i
+00019c70: 7320 3131 322e 3839 3435 206d 636d 6374  s 112.8945 mcmct
+00019c80: 7265 652e 7079 3a32 3936 0a20 2020 2020  ree.py:296.     
+00019c90: 2020 2020 2020 2020 2020 2020 206d 696c               mil
+00019ca0: 6c69 6f6e 2079 6561 7273 2066 726f 6d20  lion years from 
+00019cb0: 436f 6e63 6174 656e 6174 6564 2070 6570  Concatenated pep
+00019cc0: 7469 6465 0a20 2020 2020 2020 2020 2020  tide.           
+00019cd0: 2020 2020 2020 2061 6c69 676e 6d65 6e74         alignment
+00019ce0: 2061 6e64 2039 3525 2063 7265 6469 6269   and 95% credibi
+00019cf0: 6c69 7479 2069 6e74 6572 7661 6c73 2028  lity intervals (
+00019d00: 4349 290a 2020 2020 2020 2020 2020 2020  CI).            
+00019d10: 2020 2020 2020 6973 2031 3031 2e32 3234        is 101.224
+00019d20: 2d31 3233 2e31 3231 206d 696c 6c69 6f6e  -123.121 million
+00019d30: 2079 6561 7273 0a20 2020 2020 2020 2020   years.         
+00019d40: 494e 464f 2020 2020 2054 6f74 616c 2072  INFO     Total r
+00019d50: 756e 2074 696d 653a 2032 3931 3735 7320  un time: 29175s 
+00019d60: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019d70: 2020 2020 2020 2020 2020 2020 2063 6c69               cli
+00019d80: 2e70 793a 3234 310a 2020 2020 2020 2020  .py:241.        
+00019d90: 2049 4e46 4f20 2020 2020 446f 6e65 2020   INFO     Done  
+00019da0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019db0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019dc0: 2020 2020 2020 2020 2020 2020 2020 636c                cl
+00019dd0: 692e 7079 3a32 3432 0a60 6060 0a0a 546f  i.py:242.```..To
+00019de0: 2076 6973 7561 6c69 7a65 2074 6865 2064   visualize the d
+00019df0: 6174 652c 2077 6520 616c 736f 2070 726f  ate, we also pro
+00019e00: 7669 6465 6420 6120 7079 7468 6f6e 2073  vided a python s
+00019e10: 6372 6970 7420 746f 2070 6c6f 7420 7468  cript to plot th
+00019e20: 6520 5747 4420 6461 7465 7320 696e 2074  e WGD dates in t
+00019e30: 6865 2060 7767 6460 2066 6f6c 6465 722e  he `wgd` folder.
+00019e40: 2055 7365 7273 206e 6565 6420 746f 2065   Users need to e
+00019e50: 7874 7261 6374 2074 6865 2072 6177 2064  xtract the raw d
+00019e60: 6174 6573 2066 726f 6d20 7468 6520 606d  ates from the `m
+00019e70: 636d 632e 7478 7460 2066 6f72 2074 6865  cmc.txt` for the
+00019e80: 2057 4744 206e 6f64 6520 6669 7273 7420   WGD node first 
+00019e90: 616e 6420 7361 7665 2069 7420 6173 2066  and save it as f
+00019ea0: 696c 6520 6064 6174 6573 2e74 7874 6020  ile `dates.txt` 
+00019eb0: 286f 7220 7768 6174 6576 6572 2070 7265  (or whatever pre
+00019ec0: 6665 7272 6564 206e 616d 6529 2e20 416e  ferred name). An
+00019ed0: 2065 7861 6d70 6c65 2063 6f6d 6d61 6e64   example command
+00019ee0: 2069 7320 6173 2062 656c 6f77 2e0a 0a60   is as below...`
+00019ef0: 6060 0a70 7974 686f 6e20 2450 4154 482f  ``.python $PATH/
+00019f00: 706f 7374 706c 6f74 2e70 7920 706f 7374  postplot.py post
+00019f10: 6469 7320 6461 7465 732e 7478 7420 2d2d  dis dates.txt --
+00019f20: 7065 7263 656e 7469 6c65 2039 3020 2d2d  percentile 90 --
+00019f30: 7469 746c 6520 2257 4744 2064 6174 6522  title "WGD date"
+00019f40: 202d 2d68 7064 202d 6f20 2252 616e 756e   --hpd -o "Ranun
+00019f50: 6375 6c61 6c65 735f 5747 445f 6461 7465  culales_WGD_date
+00019f60: 2e73 7667 220a 6060 600a 0a21 5b5d 2864  .svg".```..![](d
+00019f70: 6174 612f 5261 6e75 6e63 756c 616c 6573  ata/Ranunculales
+00019f80: 5f57 4744 5f64 6174 652e 7376 6729 0a0a  _WGD_date.svg)..
+00019f90: 5468 6520 706f 7374 6572 696f 7220 6d65  The posterior me
+00019fa0: 616e 2c20 6d65 6469 616e 2061 6e64 206d  an, median and m
+00019fb0: 6f64 6520 6f66 2074 6865 2052 616e 756e  ode of the Ranun
+00019fc0: 6375 6c61 6c65 7320 5747 4420 6167 6520  culales WGD age 
+00019fd0: 6973 2031 3132 2e39 322c 2031 3133 2e34  is 112.92, 113.4
+00019fe0: 3420 616e 6420 3131 322e 3534 206d 7961  4 and 112.54 mya
+00019ff0: 2c20 7769 7468 2039 3025 2048 5044 2031  , with 90% HPD 1
+0001a000: 3035 2e30 3720 2d20 3132 322e 3332 206d  05.07 - 122.32 m
+0001a010: 7961 2061 7320 6d61 6e69 6665 7374 6564  ya as manifested
+0001a020: 2061 626f 7665 2e0a 0a23 2323 204b 7374   above...### Kst
+0001a030: 7265 650a 0a49 6e20 6164 6469 7469 6f6e  ree..In addition
+0001a040: 2074 6f20 7061 6972 7769 7365 202a 4b2a   to pairwise *K*
+0001a050: 3c73 7562 3e53 3c2f 7375 623e 2065 7374  <sub>S</sub> est
+0001a060: 696d 6174 696f 6e2c 2061 202a 4b2a 3c73  imation, a *K*<s
+0001a070: 7562 3e53 3c2f 7375 623e 2074 7265 6520  ub>S</sub> tree 
+0001a080: 7769 7468 2062 7261 6e63 6820 6c65 6e67  with branch leng
+0001a090: 7468 2069 6e20 2a4b 2a3c 7375 623e 533c  th in *K*<sub>S<
+0001a0a0: 2f73 7562 3e20 756e 6974 2063 616e 2061  /sub> unit can a
+0001a0b0: 6c73 6f20 6265 2064 6572 6976 6564 2066  lso be derived f
+0001a0c0: 726f 6d20 7468 6520 7072 6f67 7261 6d20  rom the program 
+0001a0d0: 6077 6764 206b 7364 6020 6769 7665 6e20  `wgd ksd` given 
+0001a0e0: 7468 6520 6f70 7469 6f6e 2060 2d2d 6b73  the option `--ks
+0001a0f0: 7472 6565 6020 616e 6420 602d 2d73 7065  tree` and `--spe
+0001a100: 6369 6573 7472 6565 602e 204e 6f74 6520  ciestree`. Note 
+0001a110: 7468 6174 2074 6865 2061 6464 6974 696f  that the additio
+0001a120: 6e61 6c20 6f70 7469 6f6e 2060 2d2d 6f6e  nal option `--on
+0001a130: 6c79 636f 6e63 6174 6b73 7472 6565 6020  lyconcatkstree` 
+0001a140: 7769 6c6c 206f 6e6c 7920 6361 6c6c 2074  will only call t
+0001a150: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0001a160: 623e 2065 7374 696d 6174 696f 6e20 666f  b> estimation fo
+0001a170: 7220 7468 6520 636f 6e63 6174 656e 6174  r the concatenat
+0001a180: 6564 2061 6c69 676e 6d65 6e74 2072 6174  ed alignment rat
+0001a190: 6865 7220 7468 616e 2061 6c6c 2074 6865  her than all the
+0001a1a0: 2061 6c69 676e 6d65 6e74 732e 2055 7365   alignments. Use
+0001a1b0: 7273 206e 6565 6420 746f 2070 726f 7669  rs need to provi
+0001a1c0: 6465 2061 2070 7265 7365 7420 7370 6563  de a preset spec
+0001a1d0: 6965 7320 7472 6565 2066 6f72 2074 6865  ies tree for the
+0001a1e0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0001a1f0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0001a200: 6f66 2074 6865 2063 6f6e 6361 7465 6e61  of the concatena
+0001a210: 7465 6420 616c 6967 6e6d 656e 7420 7768  ted alignment wh
+0001a220: 696c 6520 7468 6520 7265 6d61 696e 696e  ile the remainin
+0001a230: 6720 616c 6967 6e6d 656e 7473 2077 696c  g alignments wil
+0001a240: 6c20 6265 2061 6761 696e 7374 2061 6e20  l be against an 
+0001a250: 6175 746f 6d61 7465 6c79 2069 6e66 6572  automately infer
+0001a260: 7265 6420 7472 6565 2066 726f 6d20 6066  red tree from `f
+0001a270: 6173 7474 7265 6560 206f 7220 6069 7174  asttree` or `iqt
+0001a280: 7265 6560 2e20 496e 2074 6865 2065 6e64  ree`. In the end
+0001a290: 2c20 7573 6572 7320 7769 6c6c 2067 6574  , users will get
+0001a2a0: 2061 202a 4b2a 3c73 7562 3e53 3c2f 7375   a *K*<sub>S</su
+0001a2b0: 623e 2074 7265 652c 2061 202a 4b2a 3c73  b> tree, a *K*<s
+0001a2c0: 7562 3e41 3c2f 7375 623e 2074 7265 6520  ub>A</sub> tree 
+0001a2d0: 616e 6420 6120 cf89 2074 7265 6520 7065  and a .. tree pe
+0001a2e0: 7220 6661 6d20 616e 6420 666f 7220 7468  r fam and for th
+0001a2f0: 6520 636f 6e63 6174 656e 6174 6564 2061  e concatenated a
+0001a300: 6c69 676e 6d65 6e74 2e0a 0a60 6060 0a77  lignment...```.w
+0001a310: 6764 206b 7364 2064 6174 612f 6b73 7472  gd ksd data/kstr
+0001a320: 6565 5f64 6174 612f 6661 6d2e 7473 7620  ee_data/fam.tsv 
+0001a330: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
+0001a340: 2f41 636f 7275 735f 7461 7461 7269 6e6f  /Acorus_tatarino
+0001a350: 7769 6920 6461 7461 2f6b 7374 7265 655f  wii data/kstree_
+0001a360: 6461 7461 2f41 6d62 6f72 656c 6c61 5f74  data/Amborella_t
+0001a370: 7269 6368 6f70 6f64 6120 6461 7461 2f6b  richopoda data/k
+0001a380: 7374 7265 655f 6461 7461 2f41 7175 696c  stree_data/Aquil
+0001a390: 6567 6961 5f63 6f65 7275 6c65 6120 6461  egia_coerulea da
+0001a3a0: 7461 2f6b 7374 7265 655f 6461 7461 2f41  ta/kstree_data/A
+0001a3b0: 7269 7374 6f6c 6f63 6869 615f 6669 6d62  ristolochia_fimb
+0001a3c0: 7269 6174 6120 6461 7461 2f6b 7374 7265  riata data/kstre
+0001a3d0: 655f 6461 7461 2f43 7963 6173 5f70 616e  e_data/Cycas_pan
+0001a3e0: 7a68 6968 7561 656e 7369 202d 2d6b 7374  zhihuaensi --kst
+0001a3f0: 7265 6520 2d2d 7370 6563 6965 7374 7265  ree --speciestre
+0001a400: 6520 6461 7461 2f6b 7374 7265 655f 6461  e data/kstree_da
+0001a410: 7461 2f73 7065 6369 6573 5f74 7265 6531  ta/species_tree1
+0001a420: 2e6e 7720 2d2d 6f6e 6c79 636f 6e63 6174  .nw --onlyconcat
+0001a430: 6b73 7472 6565 202d 6f20 7767 645f 6b73  kstree -o wgd_ks
+0001a440: 7472 6565 5f74 6f70 6f6c 6f67 7931 0a60  tree_topology1.`
+0001a450: 6060 0a0a 215b 5d28 6461 7461 2f6b 7374  ``..![](data/kst
+0001a460: 7265 655f 7265 7375 6c74 732f 6b73 7472  ree_results/kstr
+0001a470: 6565 2e73 7667 290a 0a41 626f 7665 2077  ee.svg)..Above w
+0001a480: 6520 7573 6564 2074 6872 6565 2061 6c74  e used three alt
+0001a490: 6572 6e61 7469 7665 2074 6f70 6f6c 6f67  ernative topolog
+0001a4a0: 6965 7320 746f 2069 6e66 6572 2074 6865  ies to infer the
+0001a4b0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0001a4c0: 2074 7265 6520 7768 6963 6820 6c65 6420   tree which led 
+0001a4d0: 746f 2064 6966 6665 7265 6e74 2062 7261  to different bra
+0001a4e0: 6e63 6820 6c65 6e67 7468 2065 7374 696d  nch length estim
+0001a4f0: 6174 696f 6e2e 204e 6f74 6520 7468 6174  ation. Note that
+0001a500: 2074 6865 2066 616d 696c 6965 7320 7765   the families we
+0001a510: 2075 7365 6420 7765 7265 206f 6e6c 7920   used were only 
+0001a520: 7477 6f20 676c 6f62 616c 204d 5242 4820  two global MRBH 
+0001a530: 6661 6d69 6c69 6573 2066 6f72 2074 6865  families for the
+0001a540: 2070 7572 706f 7365 206f 6620 696c 6c75   purpose of illu
+0001a550: 7374 7261 7469 6f6e 2e20 546f 2061 6371  stration. To acq
+0001a560: 7569 7265 2061 6e20 6163 6375 7261 7465  uire an accurate
+0001a570: 2070 726f 6669 6c65 206f 6620 7468 6520   profile of the 
+0001a580: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
+0001a590: 6520 7661 7269 6174 696f 6e2c 206f 7274  e variation, ort
+0001a5a0: 686f 6c6f 6775 6573 2061 7420 7468 6520  hologues at the 
+0001a5b0: 7768 6f6c 6520 6765 6e6f 6d65 2073 6361  whole genome sca
+0001a5c0: 6c65 2073 686f 756c 6420 6265 2075 7365  le should be use
+0001a5d0: 642e 0a0a 496e 2061 6464 6974 696f 6e2c  d...In addition,
+0001a5e0: 206d 6f72 6520 6578 7175 6973 6974 6520   more exquisite 
+0001a5f0: 636f 6c6c 696e 6561 7220 706c 6f74 7320  collinear plots 
+0001a600: 696e 636c 7564 696e 6720 626f 7468 2069  including both i
+0001a610: 6e74 7261 2d73 7065 6369 6669 6320 616e  ntra-specific an
+0001a620: 6420 696e 7465 722d 7370 6563 6966 6963  d inter-specific
+0001a630: 2063 6f6d 7061 7269 736f 6e73 2075 7369   comparisons usi
+0001a640: 6e67 2074 6865 206f 7274 686f 6772 6f75  ng the orthogrou
+0001a650: 7073 2028 636f 6d70 6f73 6564 206f 6620  ps (composed of 
+0001a660: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
+0001a670: 6c65 612a 2c20 2a50 726f 7465 6120 6379  lea*, *Protea cy
+0001a680: 6e61 726f 6964 6573 2a2c 202a 4163 6f72  naroides*, *Acor
+0001a690: 7573 2061 6d65 7269 6361 6e75 732a 2061  us americanus* a
+0001a6a0: 6e64 202a 5669 7469 7320 7669 6e69 6665  nd *Vitis vinife
+0001a6b0: 7261 2a29 2069 6e66 6572 7265 6420 6361  ra*) inferred ca
+0001a6c0: 6e20 6265 2061 6c73 6f20 7072 6f64 7563  n be also produc
+0001a6d0: 6564 2075 7369 6e67 2060 7767 6420 7379  ed using `wgd sy
+0001a6e0: 6e60 2e20 4e6f 7465 2074 6861 7420 6469  n`. Note that di
+0001a6f0: 6666 6572 656e 7420 6765 6e6f 6d65 2061  fferent genome a
+0001a700: 7373 656d 626c 6965 7320 6d69 6768 7420  ssemblies might 
+0001a710: 6861 7665 2064 6966 6665 7265 6e74 2066  have different f
+0001a720: 6561 7475 7265 7320 616e 6420 6174 7472  eatures and attr
+0001a730: 6962 7574 6573 2077 6869 6368 2063 616e  ibutes which can
+0001a740: 2062 6520 6163 636f 6d6d 6f64 6174 6564   be accommodated
+0001a750: 2076 6961 2074 6865 206f 7074 696f 6e20   via the option 
+0001a760: 602d 2d61 6464 6974 696f 6e61 6c67 6666  `--additionalgff
+0001a770: 696e 666f 6020 666f 7220 6561 6368 2067  info` for each g
+0001a780: 656e 6f6d 6520 6173 7365 6d62 6c79 2077  enome assembly w
+0001a790: 686f 7365 206f 7264 6572 206e 6565 6473  hose order needs
+0001a7a0: 2074 6f20 666f 6c6c 6f77 2074 6865 206f   to follow the o
+0001a7b0: 7264 6572 206f 6620 6766 6633 2066 696c  rder of gff3 fil
+0001a7c0: 6573 2c20 666f 7220 696e 7374 616e 6365  es, for instance
+0001a7d0: 2027 6d4e 5241 3b4e 616d 6527 2066 6f72   'mNRA;Name' for
+0001a7e0: 2060 4171 7569 6c65 6769 615f 636f 6572   `Aquilegia_coer
+0001a7f0: 756c 6561 2e67 6666 3360 2c20 276d 4e52  ulea.gff3`, 'mNR
+0001a800: 413b 4944 2720 666f 7220 6050 726f 7465  A;ID' for `Prote
+0001a810: 615f 6379 6e61 726f 6964 6573 2e67 6666  a_cynaroides.gff
+0001a820: 3360 2c20 276d 4e52 413b 4e61 6d65 2720  3`, 'mNRA;Name' 
+0001a830: 666f 7220 6041 636f 7275 735f 616d 6572  for `Acorus_amer
+0001a840: 6963 616e 7573 2e67 6666 3360 2061 6e64  icanus.gff3` and
+0001a850: 2027 6d4e 5241 3b4e 616d 6527 2066 6f72   'mNRA;Name' for
+0001a860: 2060 5669 7469 735f 7669 6e69 6665 7261   `Vitis_vinifera
+0001a870: 2e67 6666 3360 2e0a 0a60 6060 0a77 6764  .gff3`...```.wgd
+0001a880: 2064 6d64 202d 6f6f 202d 6f69 2041 7175   dmd -oo -oi Aqu
+0001a890: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+0001a8a0: 5072 6f74 6561 5f63 796e 6172 6f69 6465  Protea_cynaroide
+0001a8b0: 7320 4163 6f72 7573 5f61 6d65 7269 6361  s Acorus_america
+0001a8c0: 6e75 7320 5669 7469 735f 7669 6e69 6665  nus Vitis_vinife
+0001a8d0: 7261 202d 6f20 7767 645f 6f72 7468 6f0a  ra -o wgd_ortho.
+0001a8e0: 7767 6420 6b73 6420 7767 645f 6f72 7468  wgd ksd wgd_orth
+0001a8f0: 6f2f 4f72 7468 6f67 726f 7570 732e 7370  o/Orthogroups.sp
+0001a900: 2e74 7376 2041 7175 696c 6567 6961 5f63  .tsv Aquilegia_c
+0001a910: 6f65 7275 6c65 6120 5072 6f74 6561 5f63  oerulea Protea_c
+0001a920: 796e 6172 6f69 6465 7320 4163 6f72 7573  ynaroides Acorus
+0001a930: 5f61 6d65 7269 6361 6e75 7320 5669 7469  _americanus Viti
+0001a940: 735f 7669 6e69 6665 7261 202d 6f20 7767  s_vinifera -o wg
+0001a950: 645f 6f72 7468 6f5f 6b73 0a77 6764 2073  d_ortho_ks.wgd s
+0001a960: 796e 2077 6764 5f6f 7274 686f 2f4f 7274  yn wgd_ortho/Ort
+0001a970: 686f 6772 6f75 7073 2e73 702e 7473 7620  hogroups.sp.tsv 
+0001a980: 2d6b 7320 7767 645f 6f72 7468 6f5f 6b73  -ks wgd_ortho_ks
+0001a990: 2f4f 7274 686f 6772 6f75 7073 2e73 702e  /Orthogroups.sp.
+0001a9a0: 7473 762e 6b73 2e74 7376 2041 7175 696c  tsv.ks.tsv Aquil
+0001a9b0: 6567 6961 5f63 6f65 7275 6c65 612e 6766  egia_coerulea.gf
+0001a9c0: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
+0001a9d0: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
+0001a9e0: 6527 2050 726f 7465 615f 6379 6e61 726f  e' Protea_cynaro
+0001a9f0: 6964 6573 2e67 6666 3320 2d2d 6164 6469  ides.gff3 --addi
+0001aa00: 7469 6f6e 616c 6766 6669 6e66 6f20 276d  tionalgffinfo 'm
+0001aa10: 4e52 413b 4944 2720 4163 6f72 7573 5f61  NRA;ID' Acorus_a
+0001aa20: 6d65 7269 6361 6e75 732e 6766 6633 202d  mericanus.gff3 -
+0001aa30: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
+0001aa40: 666f 2027 6d4e 5241 3b4e 616d 6527 2056  fo 'mNRA;Name' V
+0001aa50: 6974 6973 5f76 696e 6966 6572 612e 6766  itis_vinifera.gf
+0001aa60: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
+0001aa70: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
+0001aa80: 6527 202d 6f20 7767 645f 6f72 7468 6f5f  e' -o wgd_ortho_
+0001aa90: 7379 6e0a 6060 600a 0a55 706f 6e20 7468  syn.```..Upon th
+0001aaa0: 6520 6163 7175 6973 6974 696f 6e20 6f66  e acquisition of
+0001aab0: 2074 6865 2063 6f6c 6c69 6e65 6172 2072   the collinear r
+0001aac0: 6573 756c 7473 2075 7369 6e67 2060 7767  esults using `wg
+0001aad0: 6420 7379 6e60 2c20 7468 6520 7361 6d65  d syn`, the same
+0001aae0: 2063 6f6c 6c69 6e65 6172 2070 6c6f 7473   collinear plots
+0001aaf0: 2063 616e 2062 6520 616c 736f 2070 726f   can be also pro
+0001ab00: 6475 6365 6420 6279 2060 7767 6420 7669  duced by `wgd vi
+0001ab10: 7a60 2075 7369 6e67 2074 6865 2063 6f6d  z` using the com
+0001ab20: 6d61 6e64 2062 656c 6f77 2e0a 6060 600a  mand below..```.
+0001ab30: 7767 6420 7669 7a20 2d2d 706c 6f74 7379  wgd viz --plotsy
+0001ab40: 6e20 2d73 6d20 7767 645f 6f72 7468 6f5f  n -sm wgd_ortho_
+0001ab50: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+0001ab60: 7365 676d 656e 7473 2e74 7874 202d 6170  segments.txt -ap
+0001ab70: 2077 6764 5f6f 7274 686f 5f73 796e 2f69   wgd_ortho_syn/i
+0001ab80: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
+0001ab90: 7270 6f69 6e74 732e 7478 7420 2d6d 7420  rpoints.txt -mt 
+0001aba0: 7767 645f 6f72 7468 6f5f 7379 6e2f 6961  wgd_ortho_syn/ia
+0001abb0: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
+0001abc0: 6c69 636f 6e73 2e74 7874 202d 6774 2077  licons.txt -gt w
+0001abd0: 6764 5f6f 7274 686f 5f73 796e 2f67 656e  gd_ortho_syn/gen
+0001abe0: 652d 7461 626c 652e 6373 7620 2d64 2077  e-table.csv -d w
+0001abf0: 6764 5f6f 7274 686f 5f6b 732f 4f72 7468  gd_ortho_ks/Orth
+0001ac00: 6f67 726f 7570 732e 7370 2e74 7376 2e6b  ogroups.sp.tsv.k
+0001ac10: 732e 7473 7620 2d6f 2077 6764 5f6f 7274  s.tsv -o wgd_ort
+0001ac20: 686f 5f76 697a 0a60 6060 0a0a 215b 5d28  ho_viz.```..![](
+0001ac30: 6461 7461 2f6f 7274 686f 5f73 796e 5f72  data/ortho_syn_r
+0001ac40: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
+0001ac50: 5f63 6f65 7275 6c65 615f 5669 7469 735f  _coerulea_Vitis_
+0001ac60: 7669 6e69 6665 7261 5f6d 756c 7469 706c  vinifera_multipl
+0001ac70: 6963 6f6e 735f 6c65 7665 6c2e 706e 6729  icons_level.png)
+0001ac80: 0a54 6865 2061 626f 7665 2060 6475 7053  .The above `dupS
+0001ac90: 7461 636b 6020 706c 6f74 2073 686f 7773  tack` plot shows
+0001aca0: 2074 6865 2064 6973 7472 6962 7574 696f   the distributio
+0001acb0: 6e20 6f66 2064 7570 6c69 6361 7465 6420  n of duplicated 
+0001acc0: 7365 676d 656e 7473 206f 6620 2a41 7175  segments of *Aqu
+0001acd0: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
+0001ace0: 2063 6f6d 7061 7265 6420 746f 2069 7473   compared to its
+0001acf0: 656c 6620 2869 6e20 6772 6565 6e29 2061  elf (in green) a
+0001ad00: 6e64 2063 6f6d 7061 7265 6420 746f 202a  nd compared to *
+0001ad10: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
+0001ad20: 2869 6e20 626c 7565 2920 6f76 6572 2074  (in blue) over t
+0001ad30: 6865 2063 6872 6f6d 6f73 6f6d 6573 206f  he chromosomes o
+0001ad40: 6620 2a41 2e20 636f 6572 756c 6561 2a2e  f *A. coerulea*.
+0001ad50: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
+0001ad60: 7379 6e5f 7265 7375 6c74 732f 4f76 6572  syn_results/Over
+0001ad70: 616c 6c73 7065 6369 6573 5f4b 732e 646f  allspecies_Ks.do
+0001ad80: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+0001ad90: 0a54 6865 2061 626f 7665 202a 4b2a 3c73  .The above *K*<s
+0001ada0: 7562 3e53 3c2f 7375 623e 2064 6f74 706c  ub>S</sub> dotpl
+0001adb0: 6f74 2069 6e20 756e 6974 206f 6620 6765  ot in unit of ge
+0001adc0: 6e65 2073 686f 7773 2074 6865 206f 7665  ne shows the ove
+0001add0: 7261 6c6c 2064 6973 7472 6962 7574 696f  rall distributio
+0001ade0: 6e20 6f66 2063 6f6c 6c69 6e65 6172 6974  n of collinearit
+0001adf0: 7920 6163 6f72 7373 2074 6865 2066 6f75  y acorss the fou
+0001ae00: 7220 7370 6563 6965 7320 696e 766f 6c76  r species involv
+0001ae10: 6564 2e0a 215b 5d28 6461 7461 2f6f 7274  ed..![](data/ort
+0001ae20: 686f 5f73 796e 5f72 6573 756c 7473 2f4f  ho_syn_results/O
+0001ae30: 7665 7261 6c6c 7370 6563 6965 732e 646f  verallspecies.do
+0001ae40: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
+0001ae50: 0a54 6865 2061 626f 7665 2064 6f74 706c  .The above dotpl
+0001ae60: 6f74 2069 7320 7769 7468 6f75 7420 7468  ot is without th
+0001ae70: 6520 616e 6e6f 7461 7469 6f6e 206f 6620  e annotation of 
+0001ae80: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+0001ae90: 6167 6573 2063 6f6d 7061 7265 6420 746f  ages compared to
+0001aea0: 2074 6865 206c 6173 7420 6f6e 652e 0a21   the last one..!
+0001aeb0: 5b5d 2864 6174 612f 6f72 7468 6f5f 7379  [](data/ortho_sy
+0001aec0: 6e5f 7265 7375 6c74 732f 5379 6e64 6570  n_results/Syndep
+0001aed0: 7468 2e73 7667 290a 5468 6520 6162 6f76  th.svg).The abov
+0001aee0: 6520 5379 6e64 6570 7468 2070 6c6f 7420  e Syndepth plot 
+0001aef0: 7368 6f77 7320 7468 6520 636f 6c6c 696e  shows the collin
+0001af00: 6561 7220 7261 7469 6f20 6163 6f72 7373  ear ratio acorss
+0001af10: 2061 6c6c 2073 7065 6369 6573 2070 6169   all species pai
+0001af20: 7273 2028 696e 7472 612d 7370 6563 6966  rs (intra-specif
+0001af30: 6963 2063 6f6d 7061 7269 736f 6e20 696e  ic comparison in
+0001af40: 2067 7265 656e 2077 6869 6c65 2069 6e74   green while int
+0001af50: 6572 2d73 7065 6369 6669 6320 636f 6d70  er-specific comp
+0001af60: 6172 6973 6f6e 2069 6e20 626c 7565 292e  arison in blue).
+0001af70: 0a0a 2323 2043 6974 6174 696f 6e0a 200a  ..## Citation. .
+0001af80: 506c 6561 7365 2063 6974 6520 7573 2061  Please cite us a
+0001af90: 7420 6874 7470 733a 2f2f 646f 692e 6f72  t https://doi.or
+0001afa0: 672f 3130 2e31 3030 372f 3937 382d 312d  g/10.1007/978-1-
+0001afb0: 3037 3136 2d32 3536 312d 335f 3120 616e  0716-2561-3_1 an
+0001afc0: 6420 6874 7470 733a 2f2f 646f 692e 6f72  d https://doi.or
+0001afd0: 672f 3130 2e31 3039 332f 6269 6f69 6e66  g/10.1093/bioinf
+0001afe0: 6f72 6d61 7469 6373 2f62 7461 6532 3732  ormatics/btae272
+0001aff0: 2e0a 0a60 6060 0a48 656e 6763 6869 2043  ...```.Hengchi C
+0001b000: 6865 6e2c 2041 7274 6875 7220 5a77 6165  hen, Arthur Zwae
+0001b010: 6e65 706f 656c 2028 3230 3233 292e 2049  nepoel (2023). I
+0001b020: 6e66 6572 656e 6365 206f 6620 416e 6369  nference of Anci
+0001b030: 656e 7420 506f 6c79 706c 6f69 6479 2066  ent Polyploidy f
+0001b040: 726f 6d20 4765 6e6f 6d69 6320 4461 7461  rom Genomic Data
+0001b050: 2e20 496e 3a20 5661 6e20 6465 2050 6565  . In: Van de Pee
+0001b060: 722c 2059 2e20 2865 6473 2920 506f 6c79  r, Y. (eds) Poly
+0001b070: 706c 6f69 6479 2e20 4d65 7468 6f64 7320  ploidy. Methods 
+0001b080: 696e 204d 6f6c 6563 756c 6172 2042 696f  in Molecular Bio
+0001b090: 6c6f 6779 2c20 766f 6c20 3235 3435 2e20  logy, vol 2545. 
+0001b0a0: 4875 6d61 6e61 2c20 4e65 7720 596f 726b  Humana, New York
+0001b0b0: 2c20 4e59 2e20 6874 7470 733a 2f2f 646f  , NY. https://do
+0001b0c0: 692e 6f72 672f 3130 2e31 3030 372f 3937  i.org/10.1007/97
+0001b0d0: 382d 312d 3037 3136 2d32 3536 312d 335f  8-1-0716-2561-3_
+0001b0e0: 310a 4865 6e67 6368 6920 4368 656e 2c20  1.Hengchi Chen, 
+0001b0f0: 4172 7468 7572 205a 7761 656e 6570 6f65  Arthur Zwaenepoe
+0001b100: 6c2c 2059 7665 7320 5661 6e20 6465 2050  l, Yves Van de P
+0001b110: 6565 7220 2832 3032 3429 2e20 7767 6420  eer (2024). wgd 
+0001b120: 7632 3a20 6120 7375 6974 6520 6f66 2074  v2: a suite of t
+0001b130: 6f6f 6c73 2074 6f20 756e 636f 7665 7220  ools to uncover 
+0001b140: 616e 6420 6461 7465 2061 6e63 6965 6e74  and date ancient
+0001b150: 2070 6f6c 7970 6c6f 6964 7920 616e 6420   polyploidy and 
+0001b160: 7768 6f6c 652d 6765 6e6f 6d65 2064 7570  whole-genome dup
+0001b170: 6c69 6361 7469 6f6e 2e20 4269 6f69 6e66  lication. Bioinf
+0001b180: 6f72 6d61 7469 6373 2c20 566f 6c75 6d65  ormatics, Volume
+0001b190: 2034 302c 2049 7373 7565 2035 2c20 4d61   40, Issue 5, Ma
+0001b1a0: 7920 3230 3234 2c20 6274 6165 3237 322c  y 2024, btae272,
+0001b1b0: 2068 7474 7073 3a2f 2f64 6f69 2e6f 7267   https://doi.org
+0001b1c0: 2f31 302e 3130 3933 2f62 696f 696e 666f  /10.1093/bioinfo
+0001b1d0: 726d 6174 6963 732f 6274 6165 3237 320a  rmatics/btae272.
+0001b1e0: 6060 600a 0a46 6f72 2063 6974 6174 696f  ```..For citatio
+0001b1f0: 6e20 6f66 2074 6865 2074 6f6f 6c73 2075  n of the tools u
+0001b200: 7365 6420 696e 2077 6764 2c20 706c 6561  sed in wgd, plea
+0001b210: 7365 2063 6f6e 7375 6c74 2074 6865 2064  se consult the d
+0001b220: 6f63 756d 656e 7461 7469 6f6e 2061 740a  ocumentation at.
+0001b230: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
+0001b240: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
+0001b250: 6c61 7465 7374 2f63 6974 6174 696f 6e2e  latest/citation.
+0001b260: 6874 6d6c 2e0a 0a                        html...
```

### Comparing `wgd-2.0.37/cli.py` & `wgd-2.0.38/cli.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/setup.py` & `wgd-2.0.38/setup.py`

 * *Files 0% similar despite different names*

```diff
@@ -5,15 +5,15 @@
 from setuptools import setup
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
 setup(
     name='wgd',
-    version='2.0.37',
+    version='2.0.38',
     packages=['wgd'],
     url='http://github.com/heche-psb/wgd',
     license='GPL',
     author='Hengchi Chen',
     author_email='heche@psb.vib-ugent.be',
     description='wgd',
     long_description=long_description,
```

### Comparing `wgd-2.0.37/test/test_core.py` & `wgd-2.0.38/test/test_core.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/__init__.py` & `wgd-2.0.38/wgd/__init__.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/beast.py` & `wgd-2.0.38/wgd/beast.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/cluster.py` & `wgd-2.0.38/wgd/cluster.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/codeml.py` & `wgd-2.0.38/wgd/codeml.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/core.py` & `wgd-2.0.38/wgd/core.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/mcmctree.py` & `wgd-2.0.38/wgd/mcmctree.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/mix.py` & `wgd-2.0.38/wgd/mix.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/peak.py` & `wgd-2.0.38/wgd/peak.py`

 * *Files 0% similar despite different names*

```diff
@@ -1328,15 +1328,15 @@
     else: init_means, init_stdevs, good_prominences = onlyfind_peak_init_parameters(spl_x,spl_y,peak_threshold=peak_threshold,rel_height=rel_height)
     if withfig:
         lower95CI,upper95CI = plot_95CI_lognorm_hist(init_means, init_stdevs, ks_or, w, outdir, na, sp, ci=ci, keeptmp=True)
         if user: get95CIap(user_low,user_upp,anchor,gs_ks,outdir,na,sp,ci,user=user,kscutoff=kscutoff)
         else: get95CIap(lower95CI,upper95CI,anchor,gs_ks,outdir,na,sp,ci,user=user,kscutoff=kscutoff)
     else:
         if len(init_means)==0 or len(init_stdevs) ==0: lower95CI,upper95CI = None,None
-        else: lower95CI,upper95CI = noplot_95CI_lognorm_hist(init_means, init_stdevs, ci=ci)
+        else: lower95CI,upper95CI = noplot_95CI_lognorm_hist(init_means, init_stdevs, good_prominences, ci=ci)
         return lower95CI,upper95CI
 
 def get95CIap(lower,upper,anchor,gs_ks,outdir,na,sp,ci,user=False,kscutoff=5):
     if type(lower) == float:
         upper = min([upper,kscutoff])
         ap_95CI = gs_ks.loc[(gs_ks['dS']<=upper) & (gs_ks['dS']>=lower),:]
         sp_m = '{}'.format(sp)
@@ -1478,22 +1478,23 @@
     sns.despine(offset=1)
     plt.title('Anchor $K_\mathrm{S}$'+' distribution of {}'.format(sp))
     plt.tight_layout()
     plt.savefig(fname,format ='pdf', bbox_inches='tight')
     plt.close()
     return np.exp(mean)-std*2,np.exp(mean)+std*2
 
-def noplot_95CI_lognorm_hist(init_means, init_stdevs, ci=95):
+def noplot_95CI_lognorm_hist(init_means, init_stdevs, good_prominences, ci=95):
     ci_l = (1-ci/100)/2
     ci_u = 1-(1-ci/100)/2
     CI_95s = []
-    for mean,std,i in zip(init_means, init_stdevs,range(len(init_means))):
+    best = np.argmax(good_prominences)
+    for mean,std,i in zip(init_means, init_stdevs, range(len(init_means))):
         CI_95 = stats.lognorm.ppf([ci_l, ci_u], scale=np.exp(mean), s=std)
         CI_95s.append(CI_95) # There is supposed to be only one peak detected provided the GMM component
-    return CI_95s[0][0],CI_95s[0][1]
+    return CI_95s[best][0],CI_95s[best][1]
 
 def plot_95CI_lognorm_hist(init_means, init_stdevs, ks_or, w, outdir, na, sp, guide = None, ci=95, keeptmp=False):
     if guide != None: fname = os.path.join(outdir, "{}Ks_PeakCI_{}.pdf".format(guide,sp))
     elif na: fname = os.path.join(outdir, "AnchorKs_PeakCI_{}_node_averaged.pdf".format(sp))
     else: fname = os.path.join(outdir, "AnchorKs_PeakCI_{}_node_weighted.pdf".format(sp))
     f, ax = plt.subplots()
     x_points_strictly_positive = np.linspace(0, 5, int(5 * 100))
```

### Comparing `wgd-2.0.37/wgd/postplot.py` & `wgd-2.0.38/wgd/postplot.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/ratecorrect.py` & `wgd-2.0.38/wgd/ratecorrect.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/syn.py` & `wgd-2.0.38/wgd/syn.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/tree.py` & `wgd-2.0.38/wgd/tree.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/utils.py` & `wgd-2.0.38/wgd/utils.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd/viz.py` & `wgd-2.0.38/wgd/viz.py`

 * *Files identical despite different names*

### Comparing `wgd-2.0.37/wgd.egg-info/PKG-INFO` & `wgd-2.0.38/README.md`

 * *Files 10% similar despite different names*

```diff
@@ -1,6106 +1,6936 @@
-00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
-00000010: 3a20 322e 310a 4e61 6d65 3a20 7767 640a  : 2.1.Name: wgd.
-00000020: 5665 7273 696f 6e3a 2032 2e30 2e33 370a  Version: 2.0.37.
-00000030: 5375 6d6d 6172 793a 2077 6764 0a48 6f6d  Summary: wgd.Hom
-00000040: 652d 7061 6765 3a20 6874 7470 3a2f 2f67  e-page: http://g
-00000050: 6974 6875 622e 636f 6d2f 6865 6368 652d  ithub.com/heche-
-00000060: 7073 622f 7767 640a 4175 7468 6f72 3a20  psb/wgd.Author: 
-00000070: 4865 6e67 6368 6920 4368 656e 0a41 7574  Hengchi Chen.Aut
-00000080: 686f 722d 656d 6169 6c3a 2068 6563 6865  hor-email: heche
-00000090: 4070 7362 2e76 6962 2d75 6765 6e74 2e62  @psb.vib-ugent.b
-000000a0: 650a 4c69 6365 6e73 653a 2047 504c 0a44  e.License: GPL.D
-000000b0: 6573 6372 6970 7469 6f6e 2d43 6f6e 7465  escription-Conte
-000000c0: 6e74 2d54 7970 653a 2074 6578 742f 6d61  nt-Type: text/ma
-000000d0: 726b 646f 776e 0a4c 6963 656e 7365 2d46  rkdown.License-F
-000000e0: 696c 653a 204c 4943 454e 5345 0a0a 3c64  ile: LICENSE..<d
-000000f0: 6976 2061 6c69 676e 3d22 6365 6e74 6572  iv align="center
-00000100: 223e 0a0a 2320 6077 6764 2076 3260 203a  ">..# `wgd v2` :
-00000110: 2061 2073 7569 7465 2074 6f6f 6c20 6f66   a suite tool of
-00000120: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
-00000130: 6e64 2074 696d 696e 670a 0a5b 215b 4275  nd timing..[![Bu
-00000140: 696c 6420 5374 6174 7573 5d28 6874 7470  ild Status](http
-00000150: 733a 2f2f 6170 702e 7472 6176 6973 2d63  s://app.travis-c
-00000160: 692e 636f 6d2f 6865 6368 652d 7073 622f  i.com/heche-psb/
-00000170: 7767 642e 7376 673f 6272 616e 6368 3d70  wgd.svg?branch=p
-00000180: 6879 6c6f 6461 7469 6e67 295d 2868 7474  hylodating)](htt
-00000190: 7073 3a2f 2f61 7070 2e74 7261 7669 732d  ps://app.travis-
-000001a0: 6369 2e63 6f6d 2f68 6563 6865 2d70 7362  ci.com/heche-psb
-000001b0: 2f77 6764 290a 5b21 5b44 6f63 756d 656e  /wgd).[![Documen
-000001c0: 7461 7469 6f6e 2053 7461 7475 735d 2868  tation Status](h
-000001d0: 7474 7073 3a2f 2f72 6561 6474 6865 646f  ttps://readthedo
-000001e0: 6373 2e6f 7267 2f70 726f 6a65 6374 732f  cs.org/projects/
-000001f0: 7767 6476 322f 6261 6467 652f 3f76 6572  wgdv2/badge/?ver
-00000200: 7369 6f6e 3d6c 6174 6573 7429 5d28 6874  sion=latest)](ht
-00000210: 7470 733a 2f2f 7767 6476 322e 7265 6164  tps://wgdv2.read
-00000220: 7468 6564 6f63 732e 696f 2f65 6e2f 6c61  thedocs.io/en/la
-00000230: 7465 7374 2f3f 6261 6467 653d 6c61 7465  test/?badge=late
-00000240: 7374 290a 5b21 5b6c 6963 656e 7365 5d28  st).[![license](
-00000250: 6874 7470 733a 2f2f 696d 672e 7368 6965  https://img.shie
-00000260: 6c64 732e 696f 2f70 7970 692f 6c2f 7767  lds.io/pypi/l/wg
-00000270: 642e 7376 6729 5d28 6874 7470 733a 2f2f  d.svg)](https://
-00000280: 7079 7069 2e70 7974 686f 6e2e 6f72 672f  pypi.python.org/
-00000290: 7079 7069 2f77 6764 290a 5b21 5b4c 6174  pypi/wgd).[![Lat
-000002a0: 6573 7420 5079 5049 2076 6572 7369 6f6e  est PyPI version
-000002b0: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
-000002c0: 6965 6c64 732e 696f 2f70 7970 692f 762f  ields.io/pypi/v/
-000002d0: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
-000002e0: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
-000002f0: 672f 7079 7069 2f77 6764 290a 5b21 5b69  g/pypi/wgd).[![i
-00000300: 6e73 7461 6c6c 2077 6974 6820 6269 6f63  nstall with bioc
-00000310: 6f6e 6461 5d28 6874 7470 733a 2f2f 696d  onda](https://im
-00000320: 672e 7368 6965 6c64 732e 696f 2f62 6164  g.shields.io/bad
-00000330: 6765 2f69 6e73 7461 6c6c 2532 3077 6974  ge/install%20wit
-00000340: 682d 6269 6f63 6f6e 6461 2d62 7269 6768  h-bioconda-brigh
-00000350: 7467 7265 656e 2e73 7667 3f73 7479 6c65  tgreen.svg?style
-00000360: 3d66 6c61 7429 5d28 6874 7470 3a2f 2f62  =flat)](http://b
-00000370: 696f 636f 6e64 612e 6769 7468 7562 2e69  ioconda.github.i
-00000380: 6f2f 7265 6369 7065 732f 7767 642f 5245  o/recipes/wgd/RE
-00000390: 4144 4d45 2e68 746d 6c29 0a5b 215b 446f  ADME.html).[![Do
-000003a0: 776e 6c6f 6164 735d 2868 7474 7073 3a2f  wnloads](https:/
-000003b0: 2f70 6570 792e 7465 6368 2f62 6164 6765  /pepy.tech/badge
-000003c0: 2f77 6764 295d 2868 7474 7073 3a2f 2f70  /wgd)](https://p
-000003d0: 6570 792e 7465 6368 2f70 726f 6a65 6374  epy.tech/project
-000003e0: 2f77 6764 290a 0a2a 2a48 656e 6763 6869  /wgd)..**Hengchi
-000003f0: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
-00000400: 6165 6e65 706f 656c 2c20 5976 6573 2056  aenepoel, Yves V
-00000410: 616e 2064 6520 5065 6572 2a2a 0a0a 5b2a  an de Peer**..[*
-00000420: 2a42 696f 696e 666f 726d 6174 6963 7320  *Bioinformatics 
-00000430: 2620 4576 6f6c 7574 696f 6e61 7279 2047  & Evolutionary G
-00000440: 656e 6f6d 6963 7320 4772 6f75 702a 2a5d  enomics Group**]
-00000450: 2868 7474 7073 3a2f 2f77 7777 2e76 616e  (https://www.van
-00000460: 6465 7065 6572 6c61 622e 6f72 672f 292a  depeerlab.org/)*
-00000470: 2a2c 2056 4942 2d55 4765 6e74 2043 656e  *, VIB-UGent Cen
-00000480: 7465 7220 666f 7220 506c 616e 7420 5379  ter for Plant Sy
-00000490: 7374 656d 7320 4269 6f6c 6f67 792a 2a0a  stems Biology**.
-000004a0: 0a5b 2a2a 496e 7472 6f64 7563 7469 6f6e  .[**Introduction
-000004b0: 2a2a 5d28 2369 6e74 726f 6475 6374 696f  **](#introductio
-000004c0: 6e29 207c 200a 5b2a 2a49 6e73 7461 6c6c  n) | .[**Install
-000004d0: 6174 696f 6e2a 2a5d 2823 696e 7374 616c  ation**](#instal
-000004e0: 6c61 7469 6f6e 2920 7c20 0a5b 2a2a 5061  lation) | .[**Pa
-000004f0: 7261 6d65 7465 7273 2a2a 5d28 2370 6172  rameters**](#par
-00000500: 616d 6574 6572 7329 207c 200a 5b2a 2a55  ameters) | .[**U
-00000510: 7361 6765 2a2a 5d28 2375 7361 6765 2920  sage**](#usage) 
-00000520: 7c20 0a5b 2a2a 496c 6c75 7374 7261 7469  | .[**Illustrati
-00000530: 6f6e 2a2a 5d28 2369 6c6c 7573 7472 6174  on**](#illustrat
-00000540: 696f 6e29 207c 0a5b 2a2a 446f 6375 6d65  ion) |.[**Docume
-00000550: 6e74 6174 696f 6e2a 2a5d 2868 7474 7073  ntation**](https
-00000560: 3a2f 2f77 6764 7632 2e72 6561 6474 6865  ://wgdv2.readthe
-00000570: 646f 6373 2e69 6f2f 656e 2f6c 6174 6573  docs.io/en/lates
-00000580: 742f 3f62 6164 6765 3d6c 6174 6573 7429  t/?badge=latest)
-00000590: 207c 0a5b 2a2a 4369 7461 7469 6f6e 2a2a   |.[**Citation**
-000005a0: 5d28 2363 6974 6174 696f 6e29 0a0a 3c2f  ](#citation)..</
-000005b0: 6469 763e 0a0a 6077 6764 2076 3260 2069  div>..`wgd v2` i
-000005c0: 7320 6120 7079 7468 6f6e 2070 6163 6b61  s a python packa
-000005d0: 6765 2075 7067 7261 6465 6420 6672 6f6d  ge upgraded from
-000005e0: 2074 6865 206f 7269 6769 6e61 6c20 6077   the original `w
-000005f0: 6764 6020 7061 636b 6167 6520 6169 6d69  gd` package aimi
-00000600: 6e67 2066 6f72 2074 6865 2069 6e66 6572  ng for the infer
-00000610: 656e 6365 2061 6e64 2074 696d 696e 6720  ence and timing 
-00000620: 6f66 2061 6e63 6965 6e74 2077 686f 6c65  of ancient whole
-00000630: 2d67 656e 6f6d 6520 6475 706c 6963 6174  -genome duplicat
-00000640: 696f 6e20 2857 4744 2920 6576 656e 7473  ion (WGD) events
-00000650: 2e20 466f 7220 7468 6520 7072 6f70 6f73  . For the propos
-00000660: 6520 6f66 2069 6c6c 7573 7472 6174 696e  e of illustratin
-00000670: 6720 7468 6520 7072 696e 6369 706c 6520  g the principle 
-00000680: 616e 6420 7573 6167 6520 6f66 2060 7767  and usage of `wg
-00000690: 6420 7632 602c 2077 6520 636f 6d70 696c  d v2`, we compil
-000006a0: 6564 2074 6869 7320 646f 6375 6d65 6e74  ed this document
-000006b0: 6174 696f 6e2e 2042 656c 6f77 2077 6520  ation. Below we 
-000006c0: 6669 7273 7420 6761 7665 2061 6e20 696e  first gave an in
-000006d0: 7472 6f64 7563 7469 6f6e 206f 7665 7220  troduction over 
-000006e0: 7468 6520 7363 6f70 6520 616e 6420 6d65  the scope and me
-000006f0: 6368 616e 6973 6d20 6f66 2060 7767 6420  chanism of `wgd 
-00000700: 7632 6020 616e 6420 7468 656e 2074 6865  v2` and then the
-00000710: 2070 7261 6374 6963 616c 2069 6e66 6f72   practical infor
-00000720: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-00000730: 6720 746f 2074 6865 2069 6e73 7461 6c6c  g to the install
-00000740: 6174 696f 6e20 616e 6420 7573 6167 652e  ation and usage.
-00000750: 2041 6e20 6578 616d 706c 6172 2077 6f72   An examplar wor
-00000760: 6b66 6c6f 7720 6973 2070 726f 7669 6465  kflow is provide
-00000770: 6420 696e 2074 6865 2074 7574 6f72 6961  d in the tutoria
-00000780: 6c20 7365 6374 696f 6e20 6f6e 2068 6f77  l section on how
-00000790: 2074 6f20 7365 656b 2065 7669 6465 6e63   to seek evidenc
-000007a0: 6520 666f 7220 6120 7075 7461 7469 7665  e for a putative
-000007b0: 2057 4744 2065 7665 6e74 2061 6e64 2070   WGD event and p
-000007c0: 6572 666f 726d 2070 726f 7065 7220 7469  erform proper ti
-000007d0: 6d69 6e67 2077 6974 6820 6120 6672 6573  ming with a fres
-000007e0: 686c 7920 6f62 7461 696e 6564 2067 656e  hly obtained gen
-000007f0: 6f6d 6520 6173 7365 6d62 6c79 2069 6e20  ome assembly in 
-00000800: 6861 6e64 2e20 466f 7220 7468 6f73 6520  hand. For those 
-00000810: 7768 6f20 6172 6520 696e 7465 7265 7374  who are interest
-00000820: 6564 2069 6e20 6d6f 7265 2074 6865 6f72  ed in more theor
-00000830: 6574 6963 616c 2064 6574 6169 6c73 2c20  etical details, 
-00000840: 7765 2072 6563 6f6d 6d65 6e64 2074 7572  we recommend tur
-00000850: 6e69 6e67 2074 6f20 6f75 7220 7061 7065  ning to our pape
-00000860: 7220 616e 6420 626f 6f6b 2063 6861 7074  r and book chapt
-00000870: 6572 2066 6f72 206d 6f72 6520 6465 7461  er for more deta
-00000880: 696c 6564 2064 6573 6372 6970 7469 6f6e  iled description
-00000890: 2061 6e64 2069 6e73 6967 6874 6675 6c20   and insightful 
-000008a0: 6469 7363 7573 7369 6f6e 732e 2054 6865  discussions. The
-000008b0: 206b 6579 2069 6d70 726f 7665 6420 6665   key improved fe
-000008c0: 6174 7572 6573 206f 6620 6077 6764 2076  atures of `wgd v
-000008d0: 3260 2061 7265 2064 656d 6f6e 7374 7261  2` are demonstra
-000008e0: 7465 6420 696e 206f 7572 206c 6174 6573  ted in our lates
-000008f0: 7420 6d61 6e75 7363 7269 7074 206f 6e20  t manuscript on 
-00000900: 5b42 696f 696e 666f 726d 6174 6963 735d  [Bioinformatics]
-00000910: 2868 7474 7073 3a2f 2f64 6f69 2e6f 7267  (https://doi.org
-00000920: 2f31 302e 3130 3933 2f62 696f 696e 666f  /10.1093/bioinfo
-00000930: 726d 6174 6963 732f 6274 6165 3237 3229  rmatics/btae272)
-00000940: 2e20 4966 2079 6f75 2075 7365 2060 7767  . If you use `wg
-00000950: 6420 7632 6020 696e 2079 6f75 7220 7265  d v2` in your re
-00000960: 7365 6172 6368 2c20 706c 6561 7365 2063  search, please c
-00000970: 6974 6520 7573 2061 7320 7375 6767 6573  ite us as sugges
-00000980: 7465 6420 696e 205b 4369 7461 7469 6f6e  ted in [Citation
-00000990: 5d28 2363 6974 6174 696f 6e29 2073 6563  ](#citation) sec
-000009a0: 7469 6f6e 2e0a 0a23 2320 496e 7472 6f64  tion...## Introd
-000009b0: 7563 7469 6f6e 0a0a 506f 6c79 706c 6f69  uction..Polyploi
-000009c0: 6469 7a61 7469 6f6e 732c 2074 6865 2065  dizations, the e
-000009d0: 766f 6c75 7469 6f6e 6172 7920 7072 6f63  volutionary proc
-000009e0: 6573 7320 7468 6174 2074 6865 2065 6e74  ess that the ent
-000009f0: 6972 6520 6765 6e6f 6d65 206f 6620 616e  ire genome of an
-00000a00: 206f 7267 616e 6973 6d20 6973 2064 7570   organism is dup
-00000a10: 6c69 6361 7465 642c 2061 6c73 6f20 6e61  licated, also na
-00000a20: 6d65 6420 6173 2077 686f 6c65 2d67 656e  med as whole-gen
-00000a30: 6f6d 6520 6475 706c 6963 6174 696f 6e73  ome duplications
-00000a40: 2028 5747 4473 292c 206f 6363 7572 2072   (WGDs), occur r
-00000a50: 6563 7572 7265 6e74 6c79 2061 6372 6f73  ecurrently acros
-00000a60: 7320 7468 6520 7472 6565 206f 6620 6c69  s the tree of li
-00000a70: 6665 2e20 5468 6572 6520 6172 6520 7477  fe. There are tw
-00000a80: 6f20 6d6f 6465 7320 6f66 2070 6f6c 7970  o modes of polyp
-00000a90: 6c6f 6964 697a 6174 696f 6e73 2c20 6175  loidizations, au
-00000aa0: 746f 706f 6c79 706c 6f69 6469 7a61 7469  topolyploidizati
-00000ab0: 6f6e 7320 616e 6420 616c 6c6f 706f 6c79  ons and allopoly
-00000ac0: 706c 6f69 6469 7a61 7469 6f6e 732e 2041  ploidizations. A
-00000ad0: 7574 6f70 6f6c 7970 6c6f 6964 697a 6174  utopolyploidizat
-00000ae0: 696f 6e73 2061 7265 2074 6865 2064 7570  ions are the dup
-00000af0: 6c69 6361 7469 6f6e 206f 6620 7468 6520  lication of the 
-00000b00: 7361 6d65 2067 656e 6f6d 652c 2072 6573  same genome, res
-00000b10: 756c 7469 6e67 2069 6e20 7477 6f20 6964  ulting in two id
-00000b20: 656e 7469 6361 6c20 7375 6267 656e 6f6d  entical subgenom
-00000b30: 6573 2061 7420 7468 6520 7469 6d65 2069  es at the time i
-00000b40: 7420 656d 6572 6765 642e 2057 6869 6c65  t emerged. While
-00000b50: 2074 6865 2061 6c6c 6f70 6f6c 7970 6c6f   the allopolyplo
-00000b60: 6964 697a 6174 696f 6e73 2061 7265 206e  idizations are n
-00000b70: 6f72 6d61 6c6c 7920 6163 6869 6576 6564  ormally achieved
-00000b80: 2069 6e20 7477 6f20 7374 6570 732c 2066   in two steps, f
-00000b90: 6972 7374 2074 6865 2068 7962 7269 6469  irst the hybridi
-00000ba0: 7a61 7469 6f6e 2062 6574 7765 656e 2074  zation between t
-00000bb0: 776f 2064 6966 6665 7265 6e74 2073 7065  wo different spe
-00000bc0: 6369 6573 2c20 7265 7375 6c74 696e 6720  cies, resulting 
-00000bd0: 696e 2074 6865 2061 7269 7369 6e67 206f  in the arising o
-00000be0: 6620 7472 616e 7369 656e 7420 686f 6d6f  f transient homo
-00000bf0: 706c 6f69 6479 2c73 6563 6f6e 6420 7468  ploidy,second th
-00000c00: 6520 6475 706c 6963 6174 696f 6e20 6f66  e duplication of
-00000c10: 2074 6865 2068 6f6d 6f70 6c6f 6964 792c   the homoploidy,
-00000c20: 2072 6573 756c 7469 6e67 2069 6e20 7468   resulting in th
-00000c30: 6520 656d 6572 6765 6e63 6520 6f66 2061  e emergence of a
-00000c40: 6c6c 6f70 6f6c 7970 6c6f 6964 792e 2044  llopolyploidy. D
-00000c50: 7565 2074 6f20 7468 6520 756e 7374 6162  ue to the unstab
-00000c60: 696c 6974 7920 616e 6420 756e 6261 6c61  ility and unbala
-00000c70: 6e63 6564 2074 6574 7261 736f 6d69 6320  nced tetrasomic 
-00000c80: 696e 6865 7269 7461 6e63 652c 2066 6f72  inheritance, for
-00000c90: 2069 6e73 7461 6e63 6520 7468 6520 6e75   instance the nu
-00000ca0: 636c 6561 722d 6379 746f 706c 6173 6d69  clear-cytoplasmi
-00000cb0: 6320 696e 636f 6d70 6174 6962 696c 6974  c incompatibilit
-00000cc0: 792c 2074 6865 2070 6f6c 7970 6c6f 6964  y, the polyploid
-00000cd0: 7920 6765 6e6f 6d65 2077 696c 6c20 7468  y genome will th
-00000ce0: 656e 2065 7870 6572 6965 6e63 6520 6120  en experience a 
-00000cf0: 7072 6f63 6573 7320 6361 6c6c 6564 2064  process called d
-00000d00: 6970 6c6f 6964 697a 6174 696f 6e2c 2061  iploidization, a
-00000d10: 6c73 6f20 6e61 6d65 6420 6173 2066 7261  lso named as fra
-00000d20: 6374 696f 6e61 7469 6f6e 2c20 6475 7269  ctionation, duri
-00000d30: 6e67 2077 6869 6368 2061 206c 6172 6765  ng which a large
-00000d40: 2070 6f72 7469 6f6e 206f 6620 6765 6e65   portion of gene
-00000d50: 2064 7570 6c69 6361 7465 7320 7769 6c6c   duplicates will
-00000d60: 2067 6574 206c 6f73 7420 616e 6420 6f6e   get lost and on
-00000d70: 6c79 2061 2066 7261 6374 696f 6e20 6361  ly a fraction ca
-00000d80: 6e20 6265 2072 6574 6169 6e65 642e 2054  n be retained. T
-00000d90: 6865 2074 7261 6365 7320 6f66 2070 6f6c  he traces of pol
-00000da0: 7970 6c6f 6964 697a 6174 696f 6e73 2063  yploidizations c
-00000db0: 616e 2062 6520 7468 7573 2075 6e65 6172  an be thus unear
-00000dc0: 7468 6564 2066 726f 6d20 7468 6573 6520  thed from these 
-00000dd0: 7265 7461 696e 6564 2067 656e 6520 6475  retained gene du
-00000de0: 706c 6963 6174 6573 2e20 5468 7265 6520  plicates. Three 
-00000df0: 6170 7072 6f61 6368 6573 2062 6173 6564  approaches based
-00000e00: 206f 6e20 6765 6e65 2064 7570 6c69 6361   on gene duplica
-00000e10: 7465 732c 206e 616d 656c 792c 202a 4b2a  tes, namely, *K*
-00000e20: 3c73 7562 3e53 3c2f 7375 623e 206d 6574  <sub>S</sub> met
-00000e30: 686f 642c 2067 656e 6520 7472 6565 202d  hod, gene tree -
-00000e40: 2073 7065 6369 6573 2074 7265 6520 7265   species tree re
-00000e50: 636f 6e63 696c 6961 7469 6f6e 206d 6574  conciliation met
-00000e60: 686f 6420 616e 6420 7379 6e74 656e 7920  hod and synteny 
-00000e70: 6d65 7468 6f64 2c20 6172 6520 636f 6d6d  method, are comm
-00000e80: 6f6e 6c79 2075 7365 6420 696e 2064 6574  only used in det
-00000e90: 6563 7469 6e67 2065 7669 6465 6e63 6520  ecting evidence 
-00000ea0: 666f 7220 5747 4473 2e20 5468 6520 6765  for WGDs. The ge
-00000eb0: 6e65 2074 7265 6520 2d20 7370 6563 6965  ne tree - specie
-00000ec0: 7320 7472 6565 2072 6563 6f6e 6369 6c69  s tree reconcili
-00000ed0: 6174 696f 6e20 6d65 7468 6f64 2069 7320  ation method is 
-00000ee0: 6e6f 7420 7769 7468 696e 2074 6865 2073  not within the s
-00000ef0: 636f 7065 206f 6620 6077 6764 2076 3260  cope of `wgd v2`
-00000f00: 2c20 6275 7420 7765 206b 696e 646c 7920  , but we kindly 
-00000f10: 7265 6665 7220 7265 6164 6572 7320 7768  refer readers wh
-00000f20: 6f20 6172 6520 696e 7465 7265 7374 6564  o are interested
-00000f30: 2074 6f20 7468 6520 7068 796c 6f67 656e   to the phylogen
-00000f40: 6f6d 6963 2070 726f 6772 616d 2064 6576  omic program dev
-00000f50: 656c 6f70 6564 2062 7920 4172 7468 7572  eloped by Arthur
-00000f60: 205a 7761 656e 6570 6f65 6c20 6e61 6d65   Zwaenepoel name
-00000f70: 6420 5b57 4841 4c45 5d28 6874 7470 733a  d [WHALE](https:
-00000f80: 2f2f 6769 7468 7562 2e63 6f6d 2f61 727a  //github.com/arz
-00000f90: 7761 2f57 6861 6c65 2e6a 6c29 2061 6e64  wa/Whale.jl) and
-00000fa0: 2074 6865 2061 7373 6f63 6961 7465 6420   the associated 
-00000fb0: 5b70 6170 6572 5d28 6874 7470 733a 2f2f  [paper](https://
-00000fc0: 646f 692e 6f72 672f 3130 2e31 3039 332f  doi.org/10.1093/
-00000fd0: 6d6f 6c62 6576 2f6d 737a 3038 3829 2066  molbev/msz088) f
-00000fe0: 6f72 206d 6f72 6520 7465 6368 6e69 6361  or more technica
-00000ff0: 6c20 616e 6420 7468 656f 7265 7469 6361  l and theoretica
-00001000: 6c20 6465 7461 696c 732e 0a0a 5468 6520  l details...The 
-00001010: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00001020: 6d65 7468 6f64 2069 7320 6573 7461 626c  method is establ
-00001030: 6973 6865 6420 6f6e 2061 206d 6f64 656c  ished on a model
-00001040: 206f 6620 6765 6e65 2066 616d 696c 7920   of gene family 
-00001050: 6576 6f6c 7574 696f 6e20 7468 6174 2065  evolution that e
-00001060: 6163 6820 6765 6e65 2066 616d 696c 7920  ach gene family 
-00001070: 6973 2061 6c6c 6f77 6564 2074 6f20 6576  is allowed to ev
-00001080: 6f6c 7665 2076 6961 2067 656e 6520 6475  olve via gene du
-00001090: 706c 6963 6174 696f 6e20 616e 6420 6c6f  plication and lo
-000010a0: 7373 2e20 4e6f 7465 2074 6861 7420 7468  ss. Note that th
-000010b0: 6520 6765 6e65 2066 616d 696c 7920 6865  e gene family he
-000010c0: 7265 2069 7320 6173 7375 6d65 6420 746f  re is assumed to
-000010d0: 2062 6520 7468 6520 636c 7573 7465 7220   be the cluster 
-000010e0: 6f66 2061 6c6c 2067 656e 6573 2064 6573  of all genes des
-000010f0: 6365 6e64 6564 2066 726f 6d20 616e 2061  cended from an a
-00001100: 6e63 6573 7472 616c 2067 656e 6520 696e  ncestral gene in
-00001110: 2061 2073 696e 676c 6520 6765 6e6f 6d65   a single genome
-00001120: 2e20 5265 636f 7665 7269 6e67 2074 6865  . Recovering the
-00001130: 2067 656e 6520 7472 6565 206f 6620 7375   gene tree of su
-00001140: 6368 2067 656e 6520 6661 6d69 6c79 2069  ch gene family i
-00001150: 6e66 6f72 6d73 2074 6865 2074 696d 696e  nforms the timin
-00001160: 672c 2073 6369 6c69 6365 7420 7468 6520  g, scilicet the 
-00001170: 6167 652c 206f 6620 6765 6e65 2064 7570  age, of gene dup
-00001180: 6c69 6361 7469 6f6e 2065 7665 6e74 732e  lication events.
-00001190: 2054 6865 2061 6765 2072 6566 6572 6564   The age refered
-000011a0: 2068 6572 652c 2069 7320 6e6f 7420 696e   here, is not in
-000011b0: 2072 6561 6c20 6765 6f6c 6f67 6963 616c   real geological
-000011c0: 2074 696d 652c 2062 7574 2069 6e20 7468   time, but in th
-000011d0: 6520 756e 6974 206f 6620 6576 6f6c 7574  e unit of evolut
-000011e0: 696f 6e61 7279 2064 6973 7461 6e63 652c  ionary distance,
-000011f0: 2069 2e65 2e2c 2074 6865 206e 756d 6265   i.e., the numbe
-00001200: 7220 6f66 2073 7562 7374 6974 7574 696f  r of substitutio
-00001210: 6e73 2070 6572 2073 6974 652e 2057 6865  ns per site. Whe
-00001220: 6e20 7468 6520 6576 6f6c 7574 696f 6e61  n the evolutiona
-00001230: 7279 2072 6174 6520 7265 6d61 696e 7320  ry rate remains 
-00001240: 6170 7072 6f78 696d 6174 656c 7920 636f  approximately co
-00001250: 6e73 7461 6e74 2c20 7468 6520 6576 6f6c  nstant, the evol
-00001260: 7574 696f 6e61 7279 2064 6973 7461 6e63  utionary distanc
-00001270: 6520 6973 2074 6865 6e20 7375 7070 6f73  e is then suppos
-00001280: 6564 2074 6f20 6265 2070 726f 706f 7274  ed to be proport
-00001290: 696f 6e61 6c20 746f 2074 6865 2072 6561  ional to the rea
-000012a0: 6c20 6576 6f6c 7574 696f 6e61 7279 2074  l evolutionary t
-000012b0: 696d 652e 2054 6865 2073 796e 6f6e 796d  ime. The synonym
-000012c0: 6f75 7320 6469 7374 616e 6365 202a 4b2a  ous distance *K*
-000012d0: 3c73 7562 3e53 3c2f 7375 623e 2c20 7468  <sub>S</sub>, th
-000012e0: 6520 6e75 6d62 6572 206f 6620 7379 6e6f  e number of syno
-000012f0: 6e79 6d6f 7573 2073 7562 7374 6974 7574  nymous substitut
-00001300: 696f 6e73 2070 6572 2073 796e 6f6e 796d  ions per synonym
-00001310: 6f75 7320 7369 7465 2c20 6973 2073 7563  ous site, is suc
-00001320: 6820 6361 6e64 6964 6174 6520 7468 6174  h candidate that
-00001330: 2073 796e 6f6e 796d 6f75 7320 7375 6273   synonymous subs
-00001340: 7469 7475 7469 6f6e 7320 776f 756c 6420  titutions would 
-00001350: 6e6f 7420 696e 6375 7220 7468 6520 6368  not incur the ch
-00001360: 616e 6765 206f 6620 616d 696e 6f20 6163  ange of amino ac
-00001370: 6964 2061 6e64 2061 7265 2074 6875 7320  id and are thus 
-00001380: 7265 6761 7264 6564 2061 7320 6e65 7574  regarded as neut
-00001390: 7261 6c2c 2077 6869 6368 2061 6363 6f72  ral, which accor
-000013a0: 6469 6e67 2074 6f20 7468 6520 6e65 7574  ding to the neut
-000013b0: 7261 6c20 7468 656f 7279 2073 686f 756c  ral theory shoul
-000013c0: 6420 6f63 6375 7220 696e 2063 6f6e 7374  d occur in const
-000013d0: 616e 7420 7261 7465 2e20 4769 7665 6e20  ant rate. Given 
-000013e0: 6120 6d6f 6465 6c20 6f66 2067 656e 6520  a model of gene 
-000013f0: 6661 6d69 6c79 2074 6861 7420 616c 6c6f  family that allo
-00001400: 7773 2074 6865 2067 656e 6520 746f 2064  ws the gene to d
-00001410: 7570 6c69 6361 7465 2061 6e64 2067 6574  uplicate and get
-00001420: 206c 6f73 7420 696e 2061 2066 6978 6564   lost in a fixed
-00001430: 2072 6174 652c 206f 6e65 2063 616e 2064   rate, one can d
-00001440: 6572 6976 6520 7468 6174 2074 6865 2070  erive that the p
-00001450: 726f 6261 6269 6c69 7479 2064 656e 7369  robability densi
-00001460: 7479 2066 756e 6374 696f 6e20 6f66 2074  ty function of t
-00001470: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-00001480: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-00001490: 696f 6e20 6f66 2072 6574 6169 6e65 6420  ion of retained 
-000014a0: 6765 6e65 2064 7570 6c69 6361 7465 7320  gene duplicates 
-000014b0: 6973 2061 2071 7561 7369 2d65 7870 6f6e  is a quasi-expon
-000014c0: 656e 7469 616c 2066 756e 6374 696f 6e20  ential function 
-000014d0: 7468 6174 206d 6f73 7420 7265 7461 696e  that most retain
-000014e0: 6564 2067 656e 6520 6475 706c 6963 6174  ed gene duplicat
-000014f0: 6573 2061 7265 2072 6563 656e 746c 7920  es are recently 
-00001500: 626f 726e 6564 2077 6974 6820 7e30 2061  borned with ~0 a
-00001510: 6765 2077 6869 6c65 2061 7320 7468 6520  ge while as the 
-00001520: 6167 6520 676f 696e 6720 6f6c 6465 7220  age going older 
-00001530: 7468 6520 6173 736f 6369 6174 6564 206e  the associated n
-00001540: 756d 6265 7220 6f66 2072 6574 6169 6e65  umber of retaine
-00001550: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
-00001560: 7320 6465 6361 7920 7175 6173 692d 6578  s decay quasi-ex
-00001570: 706f 6e65 6e74 6961 6c6c 792e 2054 6865  ponentially. The
-00001580: 7265 666f 7265 2c20 7468 6520 6f63 6375  refore, the occu
-00001590: 7261 6e63 6520 6f66 206c 6172 6765 2d73  rance of large-s
-000015a0: 6361 6c65 2067 656e 6520 6475 706c 6963  cale gene duplic
-000015b0: 6174 696f 6e20 6576 656e 7473 2c20 666f  ation events, fo
-000015c0: 7220 696e 7374 616e 6520 5747 4473 2c20  r instane WGDs, 
-000015d0: 7769 7468 2076 6172 6965 6420 7265 7465  with varied rete
-000015e0: 6e74 696f 6e20 7261 7465 2c20 7769 6c6c  ntion rate, will
-000015f0: 206c 6561 7665 2061 6e20 6167 6520 7065   leave an age pe
-00001600: 616b 2066 726f 6d20 7468 6520 6275 7273  ak from the burs
-00001610: 7420 6f66 2067 656e 6520 6475 706c 6963  t of gene duplic
-00001620: 6174 6573 2069 6e20 6120 7368 6f72 7420  ates in a short 
-00001630: 7469 6d65 2d66 7261 6d65 2075 706f 6e20  time-frame upon 
-00001640: 7468 6520 696e 6974 6961 6c20 6167 6520  the initial age 
-00001650: 6469 7374 7269 6275 7469 6f6e 2c20 616e  distribution, an
-00001660: 6420 6361 6e20 6265 2075 6e76 6569 6c65  d can be unveile
-00001670: 6420 6672 6f6d 206d 6978 7475 7265 206d  d from mixture m
-00001680: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
-00001690: 2e20 486f 7765 7665 722c 2057 4744 7320  . However, WGDs 
-000016a0: 6964 656e 7469 6669 6564 2066 726f 6d20  identified from 
-000016b0: 7468 6520 7061 7261 6c6f 676f 7573 202a  the paralogous *
-000016c0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-000016d0: 6765 2064 6973 7472 6962 7574 696f 6e73  ge distributions
-000016e0: 2063 616e 206f 6e6c 7920 696e 666f 726d   can only inform
-000016f0: 2074 6865 2057 4744 2074 696d 696e 6720   the WGD timing 
-00001700: 696e 2074 6865 2074 696d 652d 7363 616c  in the time-scal
-00001710: 6520 6f66 2074 6861 7420 7370 6563 6966  e of that specif
-00001720: 6963 2073 7065 6369 6573 2c20 7768 6963  ic species, whic
-00001730: 6820 6973 206e 6f74 2063 6f6d 7061 7261  h is not compara
-00001740: 626c 6520 696e 2074 6865 2070 6879 6c6f  ble in the phylo
-00001750: 6765 6e65 7469 6320 636f 6e74 6578 742e  genetic context.
-00001760: 204f 6e6c 7920 7769 7468 2074 6865 206f   Only with the o
-00001770: 7274 686f 6c6f 676f 7573 202a 4b2a 3c73  rthologous *K*<s
-00001780: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-00001790: 6973 7472 6962 7574 696f 6e73 2c20 7768  istributions, wh
-000017a0: 6963 6820 636f 6e76 6572 7420 7468 6520  ich convert the 
-000017b0: 6573 7469 6d61 7465 6420 626f 6479 2066  estimated body f
-000017c0: 726f 6d20 7061 7261 6c6f 6775 6573 2074  rom paralogues t
-000017d0: 6f20 6f72 7468 6f6c 6f67 7565 7320 616e  o orthologues an
-000017e0: 6420 696e 666f 726d 2074 6865 2072 656c  d inform the rel
-000017f0: 6174 6976 6520 7469 6d69 6e67 206f 6620  ative timing of 
-00001800: 7370 6563 6961 7469 6f6e 2065 7665 6e74  speciation event
-00001810: 732c 2063 616e 2077 6520 6465 6369 7068  s, can we deciph
-00001820: 6572 2074 6865 2070 6879 6c6f 6765 6e65  er the phylogene
-00001830: 7469 6320 706c 6163 656d 656e 7420 6f66  tic placement of
-00001840: 2057 4744 7320 6166 7465 7220 7072 6f70   WGDs after prop
-00001850: 6572 2072 6174 6520 636f 7272 6563 7469  er rate correcti
-00001860: 6f6e 2e20 6077 6764 2076 3260 2069 7320  on. `wgd v2` is 
-00001870: 7375 6368 2070 726f 6772 616d 2074 6861  such program tha
-00001880: 7420 6865 6c70 7320 7573 6572 7320 636f  t helps users co
-00001890: 6e73 7472 7563 7420 7061 7261 6c6f 676f  nstruct paralogo
-000018a0: 7573 2061 6e64 206f 7274 686f 6c6f 676f  us and orthologo
-000018b0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-000018c0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-000018d0: 696f 6e73 2061 6e64 2072 6561 6c69 7a65  ions and realize
-000018e0: 2062 6f74 6820 7468 6520 6964 656e 7469   both the identi
-000018f0: 6669 6361 7469 6f6e 2061 6e64 2070 6c61  fication and pla
-00001900: 6365 6d65 6e74 206f 6620 5747 4473 2e0a  cement of WGDs..
-00001910: 0a49 6e20 7468 6520 7072 656d 6973 6520  .In the premise 
-00001920: 6f66 2070 6879 6c6f 6765 6e65 7469 6361  of phylogenetica
-00001930: 6c6c 7920 6c6f 6361 7465 6420 5747 4473  lly located WGDs
-00001940: 2c20 7468 6520 6162 736f 6c75 7465 2061  , the absolute a
-00001950: 6765 2028 696e 2067 656f 6c6f 6769 6361  ge (in geologica
-00001960: 6c20 7469 6d65 2920 6f66 2057 4744 7320  l time) of WGDs 
-00001970: 6361 6e20 616c 736f 2062 6520 696e 6665  can also be infe
-00001980: 7272 6564 2066 726f 6d20 7468 6f73 6520  rred from those 
-00001990: 5747 442d 7265 7461 696e 6564 2067 656e  WGD-retained gen
-000019a0: 6520 6475 706c 6963 6174 6573 2c20 616c  e duplicates, al
-000019b0: 7468 6f75 6768 2074 6865 7265 2068 6173  though there has
-000019c0: 2062 6565 6e20 6e6f 2065 6173 7920 6f72   been no easy or
-000019d0: 2073 7472 6169 6768 7466 6f72 7761 7264   straightforward
-000019e0: 2070 6970 656c 696e 6520 666f 7220 7468   pipeline for th
-000019f0: 6973 206a 6f62 2073 6f20 6661 722e 2049  is job so far. I
-00001a00: 6e20 6077 6764 2076 3260 2c20 7765 2064  n `wgd v2`, we d
-00001a10: 6576 656c 6f70 6564 2061 2066 6561 7369  eveloped a feasi
-00001a20: 626c 6520 696e 7465 6772 6174 6564 2070  ble integrated p
-00001a30: 6970 6c65 6c69 6e65 2066 6f72 2061 6273  ipleline for abs
-00001a40: 6f6c 7574 6520 6461 7469 6e67 206f 6620  olute dating of 
-00001a50: 5747 4473 2e20 5468 6520 7069 7065 6c69  WGDs. The pipeli
-00001a60: 6e65 2063 616e 2062 6520 726f 7567 686c  ne can be roughl
-00001a70: 7920 6469 7669 6465 6420 696e 746f 2074  y divided into t
-00001a80: 6872 6565 206d 6169 6e20 7374 6570 732e  hree main steps.
-00001a90: 2031 2920 5468 6520 636f 6e73 7472 7563   1) The construc
-00001aa0: 7469 6f6e 206f 6620 616e 6368 6f72 202a  tion of anchor *
-00001ab0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
-00001ac0: 6973 7472 6962 7574 696f 6e20 616e 6420  istribution and 
-00001ad0: 7468 6520 6465 6c69 6e65 6174 696f 6e20  the delineation 
-00001ae0: 6f66 2063 7265 6469 6162 6c65 202a 4b2a  of crediable *K*
-00001af0: 3c73 7562 3e53 3c2f 7375 623e 2072 616e  <sub>S</sub> ran
-00001b00: 6765 2061 646f 7074 6564 2066 6f72 2070  ge adopted for p
-00001b10: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
-00001b20: 6e67 2c20 7573 696e 6720 6077 6764 2064  ng, using `wgd d
-00001b30: 6d64 602c 2060 7767 6420 6b73 6460 2c20  md`, `wgd ksd`, 
-00001b40: 6077 6764 2073 796e 6020 616e 6420 6077  `wgd syn` and `w
-00001b50: 6764 2070 6561 6b60 2e20 4e6f 7465 2074  gd peak`. Note t
-00001b60: 6861 7420 6865 7265 2077 6520 6f6e 6c79  hat here we only
-00001b70: 2063 6f6e 7369 6465 7220 6765 6e6f 6d65   consider genome
-00001b80: 2061 7373 656d 626c 7920 6265 6361 7573   assembly becaus
-00001b90: 6520 666f 7220 7472 616e 7363 7269 7074  e for transcript
-00001ba0: 6f6d 6520 6173 7365 6d62 6c79 2069 7427  ome assembly it'
-00001bb0: 7320 696d 706f 7373 6962 6c65 2074 6f20  s impossible to 
-00001bc0: 6469 7374 696e 6775 6973 6820 5747 442d  distinguish WGD-
-00001bd0: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
-00001be0: 6573 2066 726f 6d20 736d 616c 6c2d 7363  es from small-sc
-00001bf0: 616c 6520 6475 706c 6963 6174 696f 6e2d  ale duplication-
-00001c00: 6465 7269 7665 6420 6475 706c 6963 6174  derived duplicat
-00001c10: 6573 2c20 7768 6963 6820 6861 7070 656e  es, which happen
-00001c20: 6564 2069 6e20 6120 636f 6e74 696e 756f  ed in a continuo
-00001c30: 7573 2074 696d 652d 6672 616d 6520 696e  us time-frame in
-00001c40: 7374 6561 6420 6f66 206f 6e6c 7920 6120  stead of only a 
-00001c50: 7365 7061 7261 7465 2073 686f 7274 2074  separate short t
-00001c60: 696d 652d 6672 616d 6520 616e 6420 7468  ime-frame and th
-00001c70: 7573 2072 6566 6c65 6374 7320 7468 6520  us reflects the 
-00001c80: 6475 7261 7469 6f6e 206f 6620 7468 6174  duration of that
-00001c90: 2062 7261 6e63 6820 7261 7468 6572 2074   branch rather t
-00001ca0: 6861 6e20 7468 6520 7469 6d65 2061 7420  han the time at 
-00001cb0: 7768 6963 6820 5747 4420 6f63 6375 7272  which WGD occurr
-00001cc0: 6564 2e20 3229 2054 6865 2066 6f72 6d75  ed. 2) The formu
-00001cd0: 6c61 7469 6f6e 206f 6620 6120 7374 6172  lation of a star
-00001ce0: 7469 6e67 2074 7265 6520 7573 6564 2069  ting tree used i
-00001cf0: 6e20 7468 6520 7068 796c 6f67 656e 6574  n the phylogenet
-00001d00: 6963 2064 6174 696e 672c 2063 6f6d 706f  ic dating, compo
-00001d10: 7365 6420 6f66 2061 2066 6577 2073 7065  sed of a few spe
-00001d20: 6369 6573 2061 6e64 2061 6e6e 6f74 6174  cies and annotat
-00001d30: 6564 2077 6974 6820 666f 7373 696c 2063  ed with fossil c
-00001d40: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-00001d50: 6d61 7469 6f6e 2e20 5468 6973 2073 7465  mation. This ste
-00001d60: 7020 6973 2065 7373 656e 7469 616c 2066  p is essential f
-00001d70: 6f72 2074 6865 2072 6573 756c 7420 6f66  or the result of
-00001d80: 2061 6273 6f6c 7574 6520 5747 4420 6461   absolute WGD da
-00001d90: 7469 6e67 2074 6861 7420 7765 2073 7567  ting that we sug
-00001da0: 6765 7374 2075 7365 7273 206f 6620 7461  gest users of ta
-00001db0: 6b69 6e67 2067 7265 6174 2063 6175 7469  king great cauti
-00001dc0: 6f6e 2074 6f20 6173 7375 7265 2074 6865  on to assure the
-00001dd0: 2074 7265 6520 746f 706f 6c6f 6779 2061   tree topology a
-00001de0: 6e64 2070 726f 7065 7220 626f 756e 6473  nd proper bounds
-00001df0: 2066 6f72 2066 6f73 7369 6c20 6361 6c69   for fossil cali
-00001e00: 6272 6174 696f 6e73 2e20 3329 2054 6865  brations. 3) The
-00001e10: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
-00001e20: 206f 7274 686f 6772 6f75 7073 2063 6f6e   orthogroups con
-00001e30: 7369 7374 696e 6720 6f66 2063 6f6c 6c69  sisting of colli
-00001e40: 6e65 6172 2064 7570 6c69 6361 7465 7320  near duplicates 
-00001e50: 6f66 2074 6865 2066 6f63 616c 2073 7065  of the focal spe
-00001e60: 6369 6573 2061 6e64 2074 6865 6972 2072  cies and their r
-00001e70: 6563 6970 726f 6361 6c20 6265 7374 2068  eciprocal best h
-00001e80: 6974 7320 2852 4248 7329 2061 6761 696e  its (RBHs) again
-00001e90: 7374 206f 7468 6572 2073 7065 6369 6573  st other species
-00001ea0: 2069 6e20 7468 6520 7374 6172 7469 6e67   in the starting
-00001eb0: 2074 7265 6520 616e 6420 7468 6520 7068   tree and the ph
-00001ec0: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
-00001ed0: 6720 7573 696e 6720 6120 6d6f 6c65 6375  g using a molecu
-00001ee0: 6c61 7220 6461 7469 6e67 2070 726f 6772  lar dating progr
-00001ef0: 616d 2066 6f72 2069 6e73 7461 6e63 6520  am for instance 
-00001f00: 606d 636d 6374 7265 6560 2c20 7669 6120  `mcmctree`, via 
-00001f10: 7072 6f67 7261 6d20 6077 6764 2064 6d64  program `wgd dmd
-00001f20: 6020 616e 6420 6077 6764 2066 6f63 7573  ` and `wgd focus
-00001f30: 602e 2057 6520 7265 636f 6d6d 656e 6420  `. We recommend 
-00001f40: 7468 6520 7573 6167 6520 6f66 2042 6179  the usage of Bay
-00001f50: 6573 6961 6e20 6d6f 6c65 6375 6c61 7220  esian molecular 
-00001f60: 6461 7469 6e67 2070 726f 6772 616d 2060  dating program `
-00001f70: 6d63 6d63 7472 6565 602c 2077 6869 6368  mcmctree`, which
-00001f80: 2070 726f 7669 6465 7320 6120 7661 7269   provides a vari
-00001f90: 6574 7920 6f66 2064 6966 6665 7265 6e74  ety of different
-00001fa0: 2073 7562 7374 6974 7574 696f 6e20 616e   substitution an
-00001fb0: 6420 7261 7465 206d 6f64 656c 732e 204e  d rate models. N
-00001fc0: 6f6e 6574 6865 6c65 7373 2c20 7765 2075  onetheless, we u
-00001fd0: 7267 6520 7573 6572 7320 746f 2073 6574  rge users to set
-00001fe0: 2070 7269 6f72 2064 6973 7472 6962 7574   prior distribut
-00001ff0: 696f 6e20 6f66 2064 6966 6665 7265 6e74  ion of different
-00002000: 2070 6172 616d 6574 6572 7320 7769 7468   parameters with
-00002010: 2063 6175 7469 6f6e 2061 6e64 2061 7373   caution and ass
-00002020: 7572 6520 6164 6571 7561 7465 2073 616d  ure adequate sam
-00002030: 706c 696e 6720 6f66 2064 6966 6665 7265  pling of differe
-00002040: 6e74 2070 6172 616d 6574 6572 732e 0a0a  nt parameters...
-00002050: 2323 2049 6e73 7461 6c6c 6174 696f 6e0a  ## Installation.
-00002060: 0a54 6865 2065 6173 6965 7374 2077 6179  .The easiest way
-00002070: 2074 6f20 696e 7374 616c 6c20 6077 6764   to install `wgd
-00002080: 2076 3260 2069 7320 7573 696e 6720 6050   v2` is using `P
-00002090: 5950 4960 2e20 4e6f 7465 2074 6861 7420  YPI`. Note that 
-000020a0: 6966 2079 6f75 2077 616e 7420 746f 2067  if you want to g
-000020b0: 6574 2074 6865 206c 6174 6573 7420 7570  et the latest up
-000020c0: 6461 7465 2c20 7765 2073 7567 6765 7374  date, we suggest
-000020d0: 2069 6e73 7461 6c6c 696e 6720 6672 6f6d   installing from
-000020e0: 2074 6865 2073 6f75 7263 652c 2073 696e   the source, sin
-000020f0: 6365 2074 6865 2075 7064 6174 6520 6f6e  ce the update on
-00002100: 2060 5059 5049 6020 7769 6c6c 2062 6520   `PYPI` will be 
-00002110: 6465 6c61 7965 6420 636f 6d70 6172 6564  delayed compared
-00002120: 2074 6f20 6865 7265 206f 6620 736f 7572   to here of sour
-00002130: 6365 2e0a 0a60 6060 0a70 6970 2069 6e73  ce...```.pip ins
-00002140: 7461 6c6c 2077 6764 0a60 6060 0a0a 546f  tall wgd.```..To
-00002150: 2069 6e73 7461 6c6c 2060 7767 6420 7632   install `wgd v2
-00002160: 6020 696e 2061 2076 6972 7475 616c 2065  ` in a virtual e
-00002170: 6e76 6972 6f6e 6d65 6e74 2028 7768 6963  nvironment (whic
-00002180: 6820 7765 2073 7472 6f6e 676c 7920 7265  h we strongly re
-00002190: 636f 6d6d 656e 642c 206e 6f20 6d61 7474  commend, no matt
-000021a0: 6572 2079 6f75 2069 6e73 7461 6c6c 2066  er you install f
-000021b0: 726f 6d20 736f 7572 6365 2c20 6050 5950  rom source, `PYP
-000021c0: 4960 206f 7220 6062 696f 636f 6e64 6160  I` or `bioconda`
-000021d0: 292c 2074 6865 2066 6f6c 6c6f 7769 6e67  ), the following
-000021e0: 2063 6f6d 6d61 6e64 206c 696e 6573 2063   command lines c
-000021f0: 6f75 6c64 2062 6520 7573 6564 2e0a 0a60  ould be used...`
-00002200: 6060 0a67 6974 2063 6c6f 6e65 2068 7474  ``.git clone htt
-00002210: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00002220: 6865 6368 652d 7073 622f 7767 640a 6364  heche-psb/wgd.cd
-00002230: 2077 6764 0a76 6972 7475 616c 656e 7620   wgd.virtualenv 
-00002240: 2d70 3d70 7974 686f 6e33 2045 4e56 2028  -p=python3 ENV (
-00002250: 6f72 2070 7974 686f 6e33 202d 6d20 7665  or python3 -m ve
-00002260: 6e76 2045 4e56 290a 736f 7572 6365 2045  nv ENV).source E
-00002270: 4e56 2f62 696e 2f61 6374 6976 6174 650a  NV/bin/activate.
-00002280: 7069 7020 696e 7374 616c 6c20 2d72 2072  pip install -r r
-00002290: 6571 7569 7265 6d65 6e74 732e 7478 740a  equirements.txt.
-000022a0: 7069 7020 696e 7374 616c 6c20 2e0a 6060  pip install ..``
-000022b0: 600a 0a57 6865 6e20 6d65 7420 7769 7468  `..When met with
-000022c0: 2070 6572 6d69 7373 696f 6e20 7072 6f62   permission prob
-000022d0: 6c65 6d20 696e 2069 6e73 7461 6c6c 6174  lem in installat
-000022e0: 696f 6e2c 2070 6c65 6173 6520 7472 7920  ion, please try 
-000022f0: 7468 6520 666f 6c6c 6f77 696e 6720 636f  the following co
-00002300: 6d6d 616e 6420 6c69 6e65 2e0a 0a60 6060  mmand line...```
-00002310: 0a70 6970 2069 6e73 7461 6c6c 202d 6520  .pip install -e 
-00002320: 2e0a 6060 600a 0a49 6620 6d75 6c74 6970  ..```..If multip
-00002330: 6c79 2076 6572 7369 6f6e 7320 6f66 2060  ly versions of `
-00002340: 7767 6460 2077 6572 6520 696e 7374 616c  wgd` were instal
-00002350: 6c65 6420 696e 2074 6865 2073 7973 7465  led in the syste
-00002360: 6d2c 2070 6c65 6173 6520 6164 6420 7468  m, please add th
-00002370: 6520 7269 6768 7420 7061 7468 206f 6620  e right path of 
-00002380: 696e 7465 7265 7374 6564 2076 6572 7369  interested versi
-00002390: 6f6e 2069 6e74 6f20 7468 6520 656e 7669  on into the envi
-000023a0: 726f 6e6d 656e 7420 7661 7269 6162 6c65  ronment variable
-000023b0: 732c 2066 6f72 2065 7861 6d70 6c65 0a0a  s, for example..
-000023c0: 6060 600a 6578 706f 7274 2050 4154 483d  ```.export PATH=
-000023d0: 2224 5041 5448 3a7e 2f2e 6c6f 6361 6c2f  "$PATH:~/.local/
-000023e0: 6269 6e2f 7767 6422 0a60 6060 0a0a 4e6f  bin/wgd".```..No
-000023f0: 7465 2074 6861 7420 7468 6520 7665 7273  te that the vers
-00002400: 696f 6e20 6f66 2060 6e75 6d70 7960 2069  ion of `numpy` i
-00002410: 7320 696d 706f 7274 616e 7420 2866 6f72  s important (for
-00002420: 206d 616e 7920 6f74 6865 7220 7061 636b   many other pack
-00002430: 6167 6573 2061 7265 2074 6865 2073 616d  ages are the sam
-00002440: 6520 6f66 2063 6f75 7273 6529 2c20 6573  e of course), es
-00002450: 7065 6369 616c 6c79 2066 6f72 2060 6661  pecially for `fa
-00002460: 7374 636c 7573 7465 7260 2070 6163 6b61  stcluster` packa
-00002470: 6765 2e20 496e 206f 7572 2074 6573 742c  ge. In our test,
-00002480: 2074 6865 2060 6e75 6d70 7960 2031 2e31   the `numpy` 1.1
-00002490: 392e 3020 776f 726b 7320 6669 6e65 206f  9.0 works fine o
-000024a0: 6e20 6070 7974 686f 6e33 2e36 2f38 602e  n `python3.6/8`.
-000024b0: 2049 6620 796f 7520 6d65 7420 736f 6d65   If you met some
-000024c0: 2065 7272 6f72 7320 6f72 2077 6172 6e69   errors or warni
-000024d0: 6e67 7320 6162 6f75 7420 606e 756d 7079  ngs about `numpy
-000024e0: 602c 206d 6179 6265 2063 6f6e 7369 6465  `, maybe conside
-000024f0: 7269 6e67 2070 7265 2d69 6e73 7461 6c6c  ring pre-install
-00002500: 2060 6e75 6d70 7960 2061 7320 312e 3139   `numpy` as 1.19
-00002510: 2e30 206f 7220 6f74 6865 7220 636c 6f73  .0 or other clos
-00002520: 652d 6279 2076 6572 7369 6f6e 7320 6265  e-by versions be
-00002530: 666f 7265 2079 6f75 2069 6e73 7461 6c6c  fore you install
-00002540: 2060 7767 6460 2e20 6077 6764 6020 7265   `wgd`. `wgd` re
-00002550: 6c69 6573 206f 6e20 6578 7465 726e 616c  lies on external
-00002560: 2073 6f66 7477 6172 6573 2069 6e63 6c75   softwares inclu
-00002570: 6469 6e67 2060 6469 616d 6f6e 6460 2061  ding `diamond` a
-00002580: 6e64 2060 6d63 6c60 2066 6f72 2060 7767  nd `mcl` for `wg
-00002590: 6420 646d 6460 2c20 6070 616d 6c20 7634  d dmd`, `paml v4
-000025a0: 2e39 6a60 2c20 606d 6166 6674 6020 2860  .9j`, `mafft` (`
-000025b0: 6d75 7363 6c65 6020 616e 6420 6070 7261  muscle` and `pra
-000025c0: 6e6b 6020 6966 2073 6574 292c 2060 6661  nk` if set), `fa
-000025d0: 7374 7472 6565 6020 286f 7220 6069 7174  sttree` (or `iqt
-000025e0: 7265 6560 2069 6620 7365 7429 2066 6f72  ree` if set) for
-000025f0: 2060 7767 6420 6b73 6460 2061 6e64 206f   `wgd ksd` and o
-00002600: 7074 696f 6e61 6c6c 7920 606d 7262 6179  ptionally `mrbay
-00002610: 6573 6020 666f 7220 7468 6520 7068 796c  es` for the phyl
-00002620: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
-00002630: 6365 2066 756e 6374 696f 6e20 696e 2060  ce function in `
-00002640: 7767 6420 646d 6460 2061 6e64 2060 7767  wgd dmd` and `wg
-00002650: 6420 666f 6375 7360 2028 606d 6166 6674  d focus` (`mafft
-00002660: 602c 2060 6d75 7363 6c65 6020 616e 6420  `, `muscle` and 
-00002670: 6070 7261 6e6b 6020 6173 2077 656c 6c20  `prank` as well 
-00002680: 7768 656e 2074 6865 2061 6e61 6c79 7369  when the analysi
-00002690: 7320 7265 7175 6972 6573 2073 6571 7565  s requires seque
-000026a0: 6e63 6520 616c 6967 6e6d 656e 7429 2e20  nce alignment). 
-000026b0: 536f 6d65 206f 7468 6572 206f 7074 696f  Some other optio
-000026c0: 6e61 6c20 736f 6674 7761 7265 7320 696e  nal softwares in
-000026d0: 636c 7564 696e 6720 6070 616d 6c20 7634  cluding `paml v4
-000026e0: 2e39 6a60 2c20 6072 3873 602c 2060 6265  .9j`, `r8s`, `be
-000026f0: 6173 7460 2c20 6065 6767 6e6f 6760 2c20  ast`, `eggnog`, 
-00002700: 6064 6961 6d6f 6e64 602c 2060 696e 7465  `diamond`, `inte
-00002710: 7270 726f 7363 616e 602c 2060 686d 6d65  rproscan`, `hmme
-00002720: 7220 7633 2e31 6232 6020 616e 6420 6061  r v3.1b2` and `a
-00002730: 7374 7261 6c2d 7072 6f60 2028 6068 6d6d  stral-pro` (`hmm
-00002740: 6572 2076 332e 3162 3260 2069 7320 616c  er v3.1b2` is al
-00002750: 736f 2066 6f72 2074 6865 205b 6f72 7468  so for the [orth
-00002760: 6f67 726f 7570 2061 7373 6967 6e6d 656e  ogroup assignmen
-00002770: 745d 2868 7474 7073 3a2f 2f77 6764 7632  t](https://wgdv2
-00002780: 2e72 6561 6474 6865 646f 6373 2e69 6f2f  .readthedocs.io/
-00002790: 656e 2f6c 6174 6573 742f 7573 6167 652e  en/latest/usage.
-000027a0: 6874 6d6c 236f 7274 686f 6772 6f75 702d  html#orthogroup-
-000027b0: 6173 7369 676e 6d65 6e74 2920 6675 6e63  assignment) func
-000027c0: 7469 6f6e 2069 6e20 6077 6764 2064 6d64  tion in `wgd dmd
-000027d0: 6020 616e 6420 6061 7374 7261 6c2d 7072  ` and `astral-pr
-000027e0: 6f60 2069 7320 616c 736f 2066 6f72 2074  o` is also for t
-000027f0: 6865 2066 756e 6374 696f 6e20 602d 2d63  he function `--c
-00002800: 6f6c 6c69 6e65 6172 636f 616c 6573 6365  ollinearcoalesce
-00002810: 6e63 6560 2069 6e20 6077 6764 2064 6d64  nce` in `wgd dmd
-00002820: 6029 2061 7265 2066 6f72 2074 6865 206d  `) are for the m
-00002830: 6f6c 6563 756c 6172 2064 6174 696e 672c  olecular dating,
-00002840: 2067 656e 6520 6661 6d69 6c79 2066 756e   gene family fun
-00002850: 6374 696f 6e20 616e 6e6f 7461 7469 6f6e  ction annotation
-00002860: 206f 7220 7068 796c 6f67 656e 6574 6963   or phylogenetic
-00002870: 2069 6e66 6572 656e 6365 2069 6e20 6077   inference in `w
-00002880: 6764 2066 6f63 7573 602e 0a0a 2323 2050  gd focus`...## P
-00002890: 6172 616d 6574 6572 730a 0a54 6865 7265  arameters..There
-000028a0: 2061 7265 2037 206d 6169 6e20 7072 6f67   are 7 main prog
-000028b0: 7261 6d73 2069 6e20 6077 6764 2076 3260  rams in `wgd v2`
-000028c0: 3a20 6064 6d64 602c 6066 6f63 7573 602c  : `dmd`,`focus`,
-000028d0: 606b 7364 602c 606d 6978 602c 6070 6561  `ksd`,`mix`,`pea
-000028e0: 6b60 2c60 7379 6e60 2c60 7669 7a60 2e20  k`,`syn`,`viz`. 
-000028f0: 4865 7265 6166 7465 7220 7765 2077 696c  Hereafter we wil
-00002900: 6c20 7072 6f76 6964 6520 6120 6465 7461  l provide a deta
-00002910: 696c 6564 2065 6c75 6369 6461 7469 6f6e  iled elucidation
-00002920: 206f 6e20 6561 6368 206f 6620 7468 6520   on each of the 
-00002930: 7072 6f67 7261 6d20 616e 6420 6974 7320  program and its 
-00002940: 6173 736f 6369 6174 6564 2070 6172 616d  associated param
-00002950: 6574 6572 732e 2050 6c65 6173 6520 7265  eters. Please re
-00002960: 6665 7220 746f 2074 6865 205b 5573 6167  fer to the [Usag
-00002970: 655d 2823 7573 6167 6529 2066 6f72 2074  e](#usage) for t
-00002980: 6865 2073 6365 6e61 7269 6f73 2074 6f20  he scenarios to 
-00002990: 7768 6963 6820 6561 6368 2070 6172 616d  which each param
-000029a0: 6574 6572 2061 7070 6c69 6573 2e0a 0a54  eter applies...T
-000029b0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-000029c0: 646d 6460 2063 616e 2072 6561 6c69 7a65  dmd` can realize
-000029d0: 2074 6865 2064 656c 696e 6561 7469 6f6e   the delineation
-000029e0: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
-000029f0: 6d65 2c20 5242 4873 2028 5265 6369 7072  me, RBHs (Recipr
-00002a00: 6f63 616c 2042 6573 7420 4869 7473 292c  ocal Best Hits),
-00002a10: 204d 5242 4873 2028 4d75 6c74 6970 6c65   MRBHs (Multiple
-00002a20: 2052 6563 6970 726f 6361 6c20 4265 7374   Reciprocal Best
-00002a30: 2048 6974 7329 2c20 6f72 7468 6f67 726f   Hits), orthogro
-00002a40: 7570 7320 616e 6420 736f 6d65 206f 7468  ups and some oth
-00002a50: 6572 206f 7274 686f 6772 6f75 702d 7265  er orthogroup-re
-00002a60: 6c61 7465 6420 6675 6e63 7469 6f6e 732c  lated functions,
-00002a70: 2069 6e63 6c75 6469 6e67 2063 6972 6375   including circu
-00002a80: 6d73 6372 6970 7469 6f6e 206f 6620 6e65  mscription of ne
-00002a90: 7374 6564 2073 696e 676c 652d 636f 7079  sted single-copy
-00002aa0: 206f 7274 686f 6772 6f75 7073 2028 4e53   orthogroups (NS
-00002ab0: 4f47 7329 2c20 756e 6269 6173 6564 2074  OGs), unbiased t
-00002ac0: 6573 7420 6f66 2073 696e 676c 652d 636f  est of single-co
-00002ad0: 7079 206f 7274 686f 6772 6f75 7073 2028  py orthogroups (
-00002ae0: 534f 4773 2920 6f76 6572 206d 6973 7369  SOGs) over missi
-00002af0: 6e67 2069 6e70 6172 616c 6f67 732c 2063  ng inparalogs, c
-00002b00: 6f6e 7374 7275 6374 696f 6e20 6f66 2042  onstruction of B
-00002b10: 5553 434f 2d67 7569 6465 6420 7369 6e67  USCO-guided sing
-00002b20: 6c65 2d63 6f70 7920 6f72 7468 6f67 726f  le-copy orthogro
-00002b30: 7570 7320 2853 4f47 7329 2c61 6e64 2074  ups (SOGs),and t
-00002b40: 6865 2063 6f6c 6c69 6e65 6172 2063 6f61  he collinear coa
-00002b50: 6c65 7363 656e 6365 2069 6e66 6572 656e  lescence inferen
-00002b60: 6365 206f 6620 7068 796c 6f67 656e 792e  ce of phylogeny.
-00002b70: 0a60 6060 0a77 6764 2064 6d64 2073 6571  .```.wgd dmd seq
-00002b80: 7565 6e63 6573 2028 6f70 7469 6f6e 290a  uences (option).
-00002b90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002ba0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bb0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bc0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002bd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00002be0: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00002bf0: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00002c00: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00002c10: 5f64 6d64 0a2d 742c 202d 2d74 6d70 6469  _dmd.-t, --tmpdi
-00002c20: 722c 2074 6865 2074 656d 706f 7261 7279  r, the temporary
-00002c30: 2077 6f72 6b69 6e67 2064 6972 6563 746f   working directo
-00002c40: 7279 2c20 6465 6661 756c 7420 4e6f 6e65  ry, default None
-00002c50: 2c20 6966 204e 6f6e 6520 7761 7320 6769  , if None was gi
-00002c60: 7665 6e2c 2074 6865 2074 6d70 6469 7220  ven, the tmpdir 
-00002c70: 776f 756c 6420 6265 2061 7373 6967 6e65  would be assigne
-00002c80: 6420 7261 6e64 6f6d 206e 616d 6573 2069  d random names i
-00002c90: 6e20 6375 7272 656e 7420 6469 7265 6374  n current direct
-00002ca0: 6f72 7920 616e 6420 6175 746f 6d61 7465  ory and automate
-00002cb0: 6c79 2072 656d 6f76 6564 2061 7420 7468  ly removed at th
-00002cc0: 6520 636f 6d70 6c65 7469 6f6e 206f 6620  e completion of 
-00002cd0: 7072 6f67 7261 6d2c 2065 6c73 6520 7468  program, else th
-00002ce0: 6520 746d 7064 6972 2077 6f75 6c64 2062  e tmpdir would b
-00002cf0: 6520 6b65 7074 0a2d 702c 202d 2d70 726f  e kept.-p, --pro
-00002d00: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
-00002d10: 7768 6574 6865 7220 7573 696e 6720 7072  whether using pr
-00002d20: 6f74 6569 6e20 6f72 206e 7563 6c65 6f74  otein or nucleot
-00002d30: 6964 6520 7365 7175 656e 6365 730a 2d63  ide sequences.-c
-00002d40: 2c20 2d2d 6373 636f 7265 2c20 7468 6520  , --cscore, the 
-00002d50: 632d 7363 6f72 6520 746f 2072 6573 7472  c-score to restr
-00002d60: 6963 7420 7468 6520 686f 6d6f 6c6f 6720  ict the homolog 
-00002d70: 7369 6d69 6c61 7269 7479 206f 6620 4d52  similarity of MR
-00002d80: 4248 732c 2064 6566 6175 6c74 204e 6f6e  BHs, default Non
-00002d90: 652c 2069 6620 4e6f 6e65 2077 6173 2067  e, if None was g
-00002da0: 6976 656e 2c20 7468 6520 632d 7363 6f72  iven, the c-scor
-00002db0: 6520 6675 6e63 696f 6e20 776f 756c 646e  e funcion wouldn
-00002dc0: 2774 2062 6520 6163 7469 7661 7465 642c  't be activated,
-00002dd0: 2065 6c73 6520 6578 7065 6374 696e 6720   else expecting 
-00002de0: 6120 6465 6369 6d61 6c20 7769 7468 696e  a decimal within
-00002df0: 2074 6865 2072 616e 6765 206f 6620 3020   the range of 0 
-00002e00: 616e 6420 310a 2d49 2c20 2d2d 696e 666c  and 1.-I, --infl
-00002e10: 6174 696f 6e2c 2074 6865 2069 6e66 6c61  ation, the infla
-00002e20: 7469 6f6e 2066 6163 746f 7220 666f 7220  tion factor for 
-00002e30: 4d43 4c20 7072 6f67 7261 6d2c 2064 6566  MCL program, def
-00002e40: 6175 6c74 2032 2e30 2c20 7769 7468 2068  ault 2.0, with h
-00002e50: 6967 6865 7220 7661 6c75 6520 6c65 6164  igher value lead
-00002e60: 696e 6720 746f 206d 6f72 6520 6275 7420  ing to more but 
-00002e70: 736d 616c 6c65 7220 636c 7573 7465 7273  smaller clusters
-00002e80: 0a2d 652c 202d 2d65 7661 6c2c 2074 6865  .-e, --eval, the
-00002e90: 2065 2d76 616c 7565 2063 7574 2d6f 6666   e-value cut-off
-00002ea0: 2066 6f72 2073 696d 696c 6172 6974 7920   for similarity 
-00002eb0: 696e 2064 6961 6d6f 6e64 2061 6e64 2f6f  in diamond and/o
-00002ec0: 7220 686d 6d65 722c 2064 6566 6175 6c74  r hmmer, default
-00002ed0: 2031 652d 3130 0a2d 2d74 6f5f 7374 6f70   1e-10.--to_stop
-00002ee0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00002ef0: 6865 7468 6572 2074 6f20 7472 616e 736c  hether to transl
-00002f00: 6174 6520 7468 726f 7567 6820 5354 4f50  ate through STOP
-00002f10: 2063 6f64 6f6e 732c 2069 6620 7468 6520   codons, if the 
-00002f20: 666c 6167 2077 6173 2073 6574 2c20 7472  flag was set, tr
-00002f30: 616e 736c 6174 696f 6e20 7769 6c6c 2062  anslation will b
-00002f40: 6520 7465 726d 696e 6174 6564 2061 7420  e terminated at 
-00002f50: 7468 6520 6669 7273 7420 696e 2d66 7261  the first in-fra
-00002f60: 6d65 2073 746f 7020 636f 646f 6e2c 2065  me stop codon, e
-00002f70: 6c73 6520 6120 6675 6c6c 2074 7261 6e73  lse a full trans
-00002f80: 6c61 7469 6f6e 2063 6f6e 7469 6e75 696e  lation continuin
-00002f90: 6720 6f6e 2070 6173 7369 6e67 2061 6e79  g on passing any
-00002fa0: 2073 746f 7020 636f 646f 6e73 2077 6f75   stop codons wou
-00002fb0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
-00002fc0: 2d2d 6364 732c 2066 6c61 6720 6f70 7469  --cds, flag opti
-00002fd0: 6f6e 2c20 7768 6574 6865 7220 746f 206f  on, whether to o
-00002fe0: 6e6c 7920 7472 616e 736c 6174 6520 7468  nly translate th
-00002ff0: 6520 636f 6d70 6c65 7465 2043 4453 2074  e complete CDS t
-00003000: 6861 7420 7374 6172 7473 2077 6974 6820  hat starts with 
-00003010: 6120 7661 6c69 6420 7374 6172 7420 636f  a valid start co
-00003020: 646f 6e20 616e 6420 6f6e 6c79 2063 6f6e  don and only con
-00003030: 7461 696e 7320 6120 7369 6e67 6c65 2069  tains a single i
-00003040: 6e2d 6672 616d 6520 7374 6f70 2063 6f64  n-frame stop cod
-00003050: 6f6e 2061 7420 7468 6520 656e 6420 616e  on at the end an
-00003060: 6420 6d75 7374 2062 6520 6469 7669 6461  d must be divida
-00003070: 626c 6520 6279 2074 6872 6565 2c20 6966  ble by three, if
-00003080: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00003090: 742c 206f 6e6c 7920 7468 6520 636f 6d70  t, only the comp
-000030a0: 6c65 7465 2043 4453 2077 6f75 6c64 2062  lete CDS would b
-000030b0: 6520 7472 616e 736c 6174 6564 0a2d 662c  e translated.-f,
-000030c0: 202d 2d66 6f63 7573 2c20 7468 6520 7370   --focus, the sp
-000030d0: 6563 6965 7320 746f 2062 6520 6d65 7267  ecies to be merg
-000030e0: 6564 206f 6e20 6c6f 6361 6c20 4d52 4248  ed on local MRBH
-000030f0: 732c 2064 6566 6175 6c74 204e 6f6e 652c  s, default None,
-00003100: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
-00003110: 656e 2c20 7468 6520 6c6f 6361 6c20 4d52  en, the local MR
-00003120: 4248 7320 776f 756c 646e 2774 2062 6520  BHs wouldn't be 
-00003130: 696e 6665 7272 6564 0a2d 6170 2c20 2d2d  inferred.-ap, --
-00003140: 616e 6368 6f72 706f 696e 7473 2c20 7468  anchorpoints, th
-00003150: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-00003160: 6461 7461 2066 696c 6520 6672 6f6d 2069  data file from i
-00003170: 2d61 6468 6f72 6520 666f 7220 636f 6e73  -adhore for cons
-00003180: 7472 7563 7469 6e67 2074 6865 206f 7274  tructing the ort
-00003190: 686f 6772 6f75 7073 2077 6974 6820 616e  hogroups with an
-000031a0: 6368 6f72 2070 6169 7273 2c20 6465 6661  chor pairs, defa
-000031b0: 756c 7420 4e6f 6e65 0a2d 736d 2c20 2d2d  ult None.-sm, --
-000031c0: 7365 676d 656e 7473 2c20 7468 6520 7365  segments, the se
-000031d0: 676d 656e 7473 2064 6174 6166 696c 6520  gments datafile 
-000031e0: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
-000031f0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
-00003200: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
-00003210: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
-00003220: 650a 2d6c 652c 202d 2d6c 6973 7465 6c65  e.-le, --listele
-00003230: 6d65 6e74 732c 2074 6865 206c 6973 7473  ments, the lists
-00003240: 6567 6d65 6e74 7320 6461 7461 2066 696c  egments data fil
-00003250: 6520 7573 6564 2069 6e20 636f 6c6c 696e  e used in collin
-00003260: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-00003270: 616e 616c 7973 6973 2069 6620 696e 6974  analysis if init
-00003280: 6961 7465 642c 2064 6566 6175 6c74 204e  iated, default N
-00003290: 6f6e 650a 2d67 742c 202d 2d67 656e 6574  one.-gt, --genet
-000032a0: 6162 6c65 2c20 7468 6520 6765 6e65 2074  able, the gene t
-000032b0: 6162 6c65 2064 6174 6166 696c 6520 7573  able datafile us
-000032c0: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
-000032d0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
-000032e0: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
-000032f0: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
-00003300: 2d63 6f63 2c20 2d2d 636f 6c6c 696e 6561  -coc, --collinea
-00003310: 7263 6f61 6c65 7363 656e 6365 2c20 666c  rcoalescence, fl
-00003320: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00003330: 6572 2074 6f20 696e 6974 6961 7465 2074  er to initiate t
-00003340: 6865 2063 6f6c 6c69 6e65 6172 2063 6f61  he collinear coa
-00003350: 6c65 7363 656e 6365 2061 6e61 6c79 7369  lescence analysi
-00003360: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
-00003370: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
-00003380: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
-00003390: 6974 6961 7465 640a 2d6b 662c 202d 2d6b  itiated.-kf, --k
-000033a0: 6565 7066 6173 7461 2c20 666c 6167 206f  eepfasta, flag o
-000033b0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000033c0: 6f20 6f75 7470 7574 2074 6865 2073 6571  o output the seq
-000033d0: 7565 6e63 6520 696e 666f 726d 6174 696f  uence informatio
-000033e0: 6e20 6f66 204d 5242 4873 2c20 6966 2074  n of MRBHs, if t
-000033f0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00003400: 2074 6865 2073 6571 7565 6e63 6573 206f   the sequences o
-00003410: 6620 4d52 4248 7320 776f 756c 6420 6265  f MRBHs would be
-00003420: 2069 6e20 6f75 7470 7574 0a2d 6b64 2c20   in output.-kd, 
-00003430: 2d2d 6b65 6570 6475 706c 6963 6174 6573  --keepduplicates
-00003440: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00003450: 6865 7468 6572 2074 6f20 616c 6c6f 7720  hether to allow 
-00003460: 7468 6520 7361 6d65 2067 656e 6520 746f  the same gene to
-00003470: 206f 6363 7572 2069 6e20 6469 6666 6572   occur in differ
-00003480: 656e 7420 4d52 4248 7320 286f 6e6c 7920  ent MRBHs (only 
-00003490: 6d65 616e 696e 6766 756c 2077 6865 6e20  meaningful when 
-000034a0: 7468 6520 6373 636f 7265 2077 6173 2075  the cscore was u
-000034b0: 7365 6429 2c20 6966 2074 6865 2066 6c61  sed), if the fla
-000034c0: 6720 7761 7320 7365 742c 2074 6865 2073  g was set, the s
-000034d0: 616d 6520 6765 6e65 2063 6f75 6c64 2062  ame gene could b
-000034e0: 6520 6173 7369 676e 6564 2074 6f20 6469  e assigned to di
-000034f0: 6666 6572 656e 7420 4d52 4248 730a 2d67  fferent MRBHs.-g
-00003500: 6d2c 202d 2d67 6c6f 6261 6c6d 7262 682c  m, --globalmrbh,
-00003510: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00003520: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00003530: 6520 676c 6f62 616c 204d 5242 4873 2063  e global MRBHs c
-00003540: 6f6e 7374 7275 6374 696f 6e2c 2069 6620  onstruction, if 
-00003550: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003560: 2c20 7468 6520 2d2d 666f 6375 7320 6f70  , the --focus op
-00003570: 7469 6f6e 2077 6f75 6c64 2062 6520 6967  tion would be ig
-00003580: 6e6f 7265 6420 616e 6420 6f6e 6c79 2067  nored and only g
-00003590: 6c6f 6261 6c20 4d52 4248 7320 776f 756c  lobal MRBHs woul
-000035a0: 6420 6265 2062 7569 6c74 0a2d 6e2c 202d  d be built.-n, -
-000035b0: 2d6e 7468 7265 6164 732c 2074 6865 206e  -nthreads, the n
-000035c0: 756d 6265 7220 6f66 2074 6872 6561 6473  umber of threads
-000035d0: 2074 6f20 7573 652c 2064 6566 6175 6c74   to use, default
-000035e0: 2034 0a2d 6f69 2c20 2d2d 6f72 7468 6f69   4.-oi, --orthoi
-000035f0: 6e66 6572 2c20 666c 6167 206f 7074 696f  nfer, flag optio
-00003600: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
-00003610: 6974 6961 7465 206f 7274 686f 6772 6f75  itiate orthogrou
-00003620: 7020 696e 6665 726e 6563 652c 2069 6620  p infernece, if 
-00003630: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003640: 2c20 7468 6520 6f72 7468 6f67 726f 7570  , the orthogroup
-00003650: 2069 6e66 6572 6e65 6365 2070 726f 6772   infernece progr
-00003660: 616d 2077 6f75 6c64 2062 6520 696e 6974  am would be init
-00003670: 6961 7465 640a 2d6f 6f2c 202d 2d6f 6e6c  iated.-oo, --onl
-00003680: 796f 7274 686f 2c20 666c 6167 206f 7074  yortho, flag opt
-00003690: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000036a0: 6f6e 6c79 2063 6f6e 6475 6374 206f 7274  only conduct ort
-000036b0: 686f 6772 6f75 7020 696e 6665 726e 6563  hogroup infernec
-000036c0: 652c 2069 6620 7468 6520 666c 6167 2077  e, if the flag w
-000036d0: 6173 2073 6574 2c20 6f6e 6c79 2074 6865  as set, only the
-000036e0: 206f 7274 686f 6772 6f75 7020 696e 6665   orthogroup infe
-000036f0: 726e 6563 6520 7069 7065 6c69 6e65 2077  rnece pipeline w
-00003700: 6f75 6c64 2062 6520 7065 7266 6f72 6d65  ould be performe
-00003710: 6420 7768 696c 6520 7468 6520 6f74 6865  d while the othe
-00003720: 7220 616e 616c 7973 6973 2077 6f75 6c64  r analysis would
-00003730: 6e27 7420 6265 2069 6e69 7469 6174 6564  n't be initiated
-00003740: 0a2d 676e 2c20 2d2d 6765 746e 736f 672c  .-gn, --getnsog,
-00003750: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00003760: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00003770: 6520 7468 6520 7365 6172 6368 696e 6720  e the searching 
-00003780: 666f 7220 6e65 7374 6564 2073 696e 676c  for nested singl
-00003790: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
-000037a0: 6c69 6573 2028 4e53 4f47 7329 2028 6f6e  lies (NSOGs) (on
-000037b0: 6c79 206d 6561 6e69 6e67 6675 6c20 7768  ly meaningful wh
-000037c0: 656e 2074 6865 206f 7274 686f 6772 6f75  en the orthogrou
-000037d0: 7020 696e 6665 726e 6563 6520 7069 7065  p infernece pipe
-000037e0: 6c69 6e65 2077 6173 2061 6374 6976 6174  line was activat
-000037f0: 6564 292c 2069 6620 7468 6520 666c 6167  ed), if the flag
-00003800: 2077 6173 2073 6574 2c20 6164 6469 7469   was set, additi
-00003810: 6f6e 616c 204e 534f 4773 2061 6e61 6c79  onal NSOGs analy
-00003820: 7369 7320 776f 756c 6420 6265 2070 6572  sis would be per
-00003830: 666f 726d 6564 2062 6573 6964 6573 2074  formed besides t
-00003840: 6865 2062 6173 6963 206f 7274 686f 6772  he basic orthogr
-00003850: 6f75 7020 696e 6665 726e 6563 650a 2d74  oup infernece.-t
-00003860: 7265 652c 202d 2d74 7265 655f 6d65 7468  ree, --tree_meth
-00003870: 6f64 2c20 7768 6963 6820 6765 6e65 2074  od, which gene t
-00003880: 7265 6520 696e 6665 7265 6e63 6520 7072  ree inference pr
-00003890: 6f67 7261 6d20 746f 2069 6e76 6f6b 6520  ogram to invoke 
-000038a0: 286f 6e6c 7920 6d65 616e 696e 6766 756c  (only meaningful
-000038b0: 2077 6865 6e20 7468 6520 636f 6c6c 696e   when the collin
-000038c0: 6561 7220 636f 616c 6573 6365 6e63 652c  ear coalescence,
-000038d0: 2067 656e 652d 746f 2d66 616d 696c 7920   gene-to-family 
-000038e0: 6173 7369 676e 6d65 6e74 206f 7220 4e53  assignment or NS
-000038f0: 4f47 7320 616e 616c 7973 6973 2077 6572  OGs analysis wer
-00003900: 6520 6163 7469 7661 7465 6429 2c20 6465  e activated), de
-00003910: 6661 756c 7420 6661 7374 7472 6565 0a2d  fault fasttree.-
-00003920: 7473 2c20 2d2d 7472 6565 7365 742c 2074  ts, --treeset, t
-00003930: 6865 2070 6172 616d 6574 6572 7320 7365  he parameters se
-00003940: 7474 696e 6720 666f 7220 6765 6e65 2074  tting for gene t
-00003950: 7265 6520 696e 6665 7265 6e63 652c 2064  ree inference, d
-00003960: 6566 6175 6c74 204e 6f6e 652c 2074 6869  efault None, thi
-00003970: 7320 6f70 7469 6f6e 2063 616e 2062 6520  s option can be 
-00003980: 7072 6f76 6964 6564 206d 756c 7469 706c  provided multipl
-00003990: 6520 7469 6d65 730a 2d6d 632c 202d 2d6d  e times.-mc, --m
-000039a0: 736f 6763 7574 2c20 7468 6520 7261 7469  sogcut, the rati
-000039b0: 6f20 6375 746f 6666 2066 6f72 206d 6f73  o cutoff for mos
-000039c0: 746c 7920 7369 6e67 6c65 2d63 6f70 7920  tly single-copy 
-000039d0: 6661 6d69 6c79 2028 6d65 616e 696e 6766  family (meaningf
-000039e0: 756c 2077 6865 6e20 6163 7469 7661 7469  ul when activati
-000039f0: 6e67 2074 6865 206f 7274 686f 6772 6f75  ng the orthogrou
-00003a00: 7020 696e 6665 726e 6563 6520 7069 7065  p infernece pipe
-00003a10: 6c69 6e65 2920 616e 6420 7370 6563 6965  line) and specie
-00003a20: 7320 7265 7072 6573 656e 7461 7469 6f6e  s representation
-00003a30: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
-00003a40: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
-00003a50: 6973 2c20 6465 6661 756c 7420 302e 382e  is, default 0.8.
-00003a60: 0a2d 6761 2c20 2d2d 6765 6e65 6173 7369  .-ga, --geneassi
-00003a70: 676e 2c20 666c 6167 206f 7074 696f 6e2c  gn, flag option,
-00003a80: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00003a90: 6961 7465 2074 6865 2067 656e 652d 746f  iate the gene-to
-00003aa0: 2d66 616d 696c 7920 6173 7369 676e 6d65  -family assignme
-00003ab0: 6e74 2061 6e61 6c79 7369 732c 2069 6620  nt analysis, if 
-00003ac0: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00003ad0: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
-00003ae0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
-00003af0: 640a 2d73 612c 202d 2d73 6571 3261 7373  d.-sa, --seq2ass
-00003b00: 6967 6e2c 2074 6865 2071 7565 7269 6564  ign, the queried
-00003b10: 2073 6571 7565 6e63 6573 2064 6174 6120   sequences data 
-00003b20: 6669 6c65 2069 6e20 6765 6e65 2d74 6f2d  file in gene-to-
-00003b30: 6661 6d69 6c79 2061 7373 6967 6e6d 656e  family assignmen
-00003b40: 7420 616e 616c 7973 6973 2c20 6465 6661  t analysis, defa
-00003b50: 756c 7420 4e6f 6e65 2c20 7468 6973 206f  ult None, this o
-00003b60: 7074 696f 6e20 6361 6e20 6265 2070 726f  ption can be pro
-00003b70: 7669 6465 6420 6d75 6c74 6970 6c65 2074  vided multiple t
-00003b80: 696d 6573 0a2d 6661 2c20 2d2d 6661 6d32  imes.-fa, --fam2
-00003b90: 6173 7369 676e 2c20 7468 6520 7175 6572  assign, the quer
-00003ba0: 6965 6420 6661 6d69 6c69 7920 6461 7461  ied familiy data
-00003bb0: 2066 696c 6520 696e 2067 656e 652d 746f   file in gene-to
-00003bc0: 2d66 616d 696c 7920 6173 7369 676e 6d65  -family assignme
-00003bd0: 6e74 2061 6e61 6c79 7369 732c 2064 6566  nt analysis, def
-00003be0: 6175 6c74 204e 6f6e 650a 2d63 632c 202d  ault None.-cc, -
-00003bf0: 2d63 6f6e 6361 742c 2066 6c61 6720 6f70  -concat, flag op
-00003c00: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00003c10: 2069 6e69 7469 6174 6520 7468 6520 636f   initiate the co
-00003c20: 6e63 6174 656e 6174 696f 6e20 7069 7065  ncatenation pipe
-00003c30: 6c69 6e65 2066 6f72 206f 7274 686f 6772  line for orthogr
-00003c40: 6f75 7020 696e 6665 726e 6563 652c 2069  oup infernece, i
-00003c50: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-00003c60: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
-00003c70: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00003c80: 7465 640a 2d74 652c 202d 2d74 6573 7473  ted.-te, --tests
-00003c90: 6f67 2c20 666c 6167 206f 7074 696f 6e2c  og, flag option,
-00003ca0: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
-00003cb0: 6961 7465 2074 6865 2075 6e62 6961 7365  iate the unbiase
-00003cc0: 6420 7465 7374 206f 6620 7369 6e67 6c65  d test of single
-00003cd0: 2d63 6f70 7920 6765 6e65 2066 616d 696c  -copy gene famil
-00003ce0: 6965 732c 2069 6620 7468 6520 666c 6167  ies, if the flag
-00003cf0: 2077 6173 2073 6574 2c20 7468 6520 616e   was set, the an
-00003d00: 616c 7973 6973 2077 6f75 6c64 2062 6520  alysis would be 
-00003d10: 696e 6974 6961 7465 640a 2d62 732c 202d  initiated.-bs, -
-00003d20: 2d62 696e 732c 2074 6865 206e 756d 6265  -bins, the numbe
-00003d30: 7220 6f66 2062 696e 7320 6469 7669 6465  r of bins divide
-00003d40: 6420 696e 2074 6865 2067 656e 6520 6c65  d in the gene le
-00003d50: 6e67 7468 206e 6f72 6d61 6c69 7a61 7469  ngth normalizati
-00003d60: 6f6e 2c20 6465 6661 756c 7420 3130 300a  on, default 100.
-00003d70: 2d6e 702c 202d 2d6e 6f72 6d61 6c69 7a65  -np, --normalize
-00003d80: 6470 6572 6365 6e74 2c20 7468 6520 7065  dpercent, the pe
-00003d90: 7263 656e 7461 6765 206f 6620 7570 7065  rcentage of uppe
-00003da0: 7220 6869 7473 2075 7365 6420 666f 7220  r hits used for 
-00003db0: 6765 6e65 206c 656e 6774 6820 6e6f 726d  gene length norm
-00003dc0: 616c 697a 6174 696f 6e2c 2064 6566 6175  alization, defau
-00003dd0: 6c74 2035 0a2d 6e6e 2c20 2d2d 6e6f 6e6f  lt 5.-nn, --nono
-00003de0: 726d 616c 697a 6174 696f 6e2c 2066 6c61  rmalization, fla
-00003df0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00003e00: 7220 746f 2063 616c 6c20 6f66 6620 7468  r to call off th
-00003e10: 6520 6e6f 726d 616c 697a 6174 696f 6e2c  e normalization,
-00003e20: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003e30: 2073 6574 2c20 6e6f 206e 6f72 6d61 6c69   set, no normali
-00003e40: 7a61 7469 6f6e 2077 6f75 6c64 2062 6520  zation would be 
-00003e50: 636f 6e64 7563 7465 640a 2d62 736f 672c  conducted.-bsog,
-00003e60: 202d 2d62 7573 636f 736f 672c 2066 6c61   --buscosog, fla
-00003e70: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00003e80: 7220 746f 2069 6e69 7469 6174 6520 7468  r to initiate th
-00003e90: 6520 6275 7363 6f2d 6775 6964 6564 2073  e busco-guided s
-00003ea0: 696e 676c 652d 636f 7079 2067 656e 6520  ingle-copy gene 
-00003eb0: 6661 6d69 6c79 2061 6e61 6c79 7369 732c  family analysis,
-00003ec0: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00003ed0: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
-00003ee0: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
-00003ef0: 6961 7465 640a 2d62 686d 6d2c 202d 2d62  iated.-bhmm, --b
-00003f00: 7573 636f 686d 6d2c 2074 6865 2048 4d4d  uscohmm, the HMM
-00003f10: 2070 726f 6669 6c65 2064 6174 6166 696c   profile datafil
-00003f20: 6520 696e 2074 6865 2062 7573 636f 2d67  e in the busco-g
-00003f30: 7569 6465 6420 7369 6e67 6c65 2d63 6f70  uided single-cop
-00003f40: 7920 6765 6e65 2066 616d 696c 7920 616e  y gene family an
-00003f50: 616c 7973 6973 2c20 6465 6661 756c 7420  alysis, default 
-00003f60: 4e6f 6e65 0a2d 6263 7466 2c20 2d2d 6275  None.-bctf, --bu
-00003f70: 7363 6f63 7574 6f66 662c 2074 6865 2048  scocutoff, the H
-00003f80: 4d4d 2073 636f 7265 2063 7574 6f66 6620  MM score cutoff 
-00003f90: 6461 7461 6669 6c65 2069 6e20 7468 6520  datafile in the 
-00003fa0: 6275 7363 6f2d 6775 6964 6564 2073 696e  busco-guided sin
-00003fb0: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
-00003fc0: 6d69 6c79 2061 6e61 6c79 7369 732c 2064  mily analysis, d
-00003fd0: 6566 6175 6c74 204e 6f6e 650a 2d6f 6620  efault None.-of 
-00003fe0: 2c2d 2d6f 6766 6f72 6d61 742c 2066 6c61  ,--ogformat, fla
-00003ff0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004000: 7220 746f 2061 6464 2069 6e64 6578 2074  r to add index t
-00004010: 6f20 7468 6520 5242 4820 6661 6d69 6c69  o the RBH famili
-00004020: 6573 0a60 6060 0a0a 5468 6520 7072 6f67  es.```..The prog
-00004030: 7261 6d20 6077 6764 2066 6f63 7573 6020  ram `wgd focus` 
-00004040: 6361 6e20 7265 616c 697a 6520 7468 6520  can realize the 
-00004050: 636f 6e63 6174 656e 6174 696f 6e2d 6261  concatenation-ba
-00004060: 7365 6420 616e 6420 636f 616c 6573 6365  sed and coalesce
-00004070: 6e63 652d 6261 7365 6420 7068 796c 6f67  nce-based phylog
-00004080: 656e 6574 6963 2069 6e66 6572 656e 6365  enetic inference
-00004090: 2061 6e64 2070 6879 6c6f 6765 6e65 7469   and phylogeneti
-000040a0: 6320 6461 7469 6e67 206f 6620 5747 4473  c dating of WGDs
-000040b0: 2065 7463 2e0a 6060 600a 7767 6420 666f   etc..```.wgd fo
-000040c0: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
-000040d0: 7565 6e63 6573 2028 6f70 7469 6f6e 290a  uences (option).
-000040e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000040f0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004100: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004110: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004120: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004130: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00004140: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00004150: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00004160: 5f66 6f63 7573 0a2d 742c 202d 2d74 6d70  _focus.-t, --tmp
-00004170: 6469 722c 2074 6865 2074 656d 706f 7261  dir, the tempora
-00004180: 7279 2077 6f72 6b69 6e67 2064 6972 6563  ry working direc
-00004190: 746f 7279 2c20 6465 6661 756c 7420 4e6f  tory, default No
-000041a0: 6e65 2c20 6966 204e 6f6e 6520 7761 7320  ne, if None was 
-000041b0: 6769 7665 6e2c 2074 6865 2074 6d70 6469  given, the tmpdi
-000041c0: 7220 7769 6c6c 2062 6520 6173 7369 676e  r will be assign
-000041d0: 6564 2072 616e 646f 6d20 6e61 6d65 7320  ed random names 
-000041e0: 696e 2063 7572 7265 6e74 2064 6972 6563  in current direc
-000041f0: 746f 7279 2061 6e64 2061 7574 6f6d 6174  tory and automat
-00004200: 656c 7920 7265 6d6f 7665 6420 6174 2074  ely removed at t
-00004210: 6865 2063 6f6d 706c 6574 696f 6e20 6f66  he completion of
-00004220: 2070 726f 6772 616d 2c20 656c 7365 2074   program, else t
-00004230: 6865 2074 6d70 6469 7220 776f 756c 6420  he tmpdir would 
-00004240: 6265 206b 6570 740a 2d6e 2c20 2d2d 6e74  be kept.-n, --nt
-00004250: 6872 6561 6473 2c20 7468 6520 6e75 6d62  hreads, the numb
-00004260: 6572 206f 6620 7468 7265 6164 7320 746f  er of threads to
-00004270: 2075 7365 2c20 6465 6661 756c 7420 340a   use, default 4.
-00004280: 2d2d 746f 5f73 746f 702c 2066 6c61 6720  --to_stop, flag 
-00004290: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-000042a0: 746f 2074 7261 6e73 6c61 7465 2074 6872  to translate thr
-000042b0: 6f75 6768 2053 544f 5020 636f 646f 6e73  ough STOP codons
-000042c0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-000042d0: 7320 7365 742c 2074 7261 6e73 6c61 7469  s set, translati
-000042e0: 6f6e 2077 696c 6c20 6265 2074 6572 6d69  on will be termi
-000042f0: 6e61 7465 6420 6174 2074 6865 2066 6972  nated at the fir
-00004300: 7374 2069 6e2d 6672 616d 6520 7374 6f70  st in-frame stop
-00004310: 2063 6f64 6f6e 2c20 656c 7365 2061 2066   codon, else a f
-00004320: 756c 6c20 7472 616e 736c 6174 696f 6e20  ull translation 
-00004330: 636f 6e74 696e 7569 6e67 206f 6e20 7061  continuing on pa
-00004340: 7374 2061 6e79 2073 746f 7020 636f 646f  st any stop codo
-00004350: 6e73 2077 6f75 6c64 2062 6520 696e 6974  ns would be init
-00004360: 6961 7465 640a 2d2d 6364 732c 2066 6c61  iated.--cds, fla
-00004370: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00004380: 7220 746f 206f 6e6c 7920 7472 616e 736c  r to only transl
-00004390: 6174 6520 7468 6520 636f 6d70 6c65 7465  ate the complete
-000043a0: 2043 4453 2074 6861 7420 7374 6172 7473   CDS that starts
-000043b0: 2077 6974 6820 6120 7661 6c69 6420 7374   with a valid st
-000043c0: 6172 7420 636f 646f 6e20 616e 6420 6f6e  art codon and on
-000043d0: 6c79 2063 6f6e 7461 696e 7320 6120 7369  ly contains a si
-000043e0: 6e67 6c65 2069 6e2d 6672 616d 6520 7374  ngle in-frame st
-000043f0: 6f70 2063 6f64 6f6e 2061 7420 7468 6520  op codon at the 
-00004400: 656e 6420 616e 6420 6d75 7374 2062 6520  end and must be 
-00004410: 6469 7669 6461 626c 6520 6279 2074 6872  dividable by thr
-00004420: 6565 2c20 6966 2074 6865 2066 6c61 6720  ee, if the flag 
-00004430: 7761 7320 7365 742c 206f 6e6c 7920 7468  was set, only th
-00004440: 6520 636f 6d70 6c65 7465 2043 4453 2077  e complete CDS w
-00004450: 6f75 6c64 2062 6520 7472 616e 736c 6174  ould be translat
-00004460: 6564 0a2d 2d73 7472 6970 5f67 6170 732c  ed.--strip_gaps,
-00004470: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004480: 6574 6865 7220 746f 2064 726f 7020 616c  ether to drop al
-00004490: 6c20 6761 7073 2069 6e20 6d75 6c74 6970  l gaps in multip
-000044a0: 6c65 2073 6571 7565 6e63 6520 616c 6967  le sequence alig
-000044b0: 6e6d 656e 742c 2069 6620 7468 6520 666c  nment, if the fl
-000044c0: 6167 2077 6173 2073 6574 2c20 616c 6c20  ag was set, all 
-000044d0: 6761 7073 2077 6f75 6c64 2062 6520 6472  gaps would be dr
-000044e0: 6f70 7065 640a 2d61 2c20 2d2d 616c 6967  opped.-a, --alig
-000044f0: 6e65 722c 2077 6869 6368 2061 6c69 676e  ner, which align
-00004500: 6d65 6e74 2070 726f 6772 616d 2074 6f20  ment program to 
-00004510: 7573 652c 2064 6566 6175 6c74 206d 6166  use, default maf
-00004520: 6674 0a2d 7472 6565 2c20 2d2d 7472 6565  ft.-tree, --tree
-00004530: 5f6d 6574 686f 642c 2077 6869 6368 2067  _method, which g
-00004540: 656e 6520 7472 6565 2069 6e66 6572 656e  ene tree inferen
-00004550: 6365 2070 726f 6772 616d 2074 6f20 696e  ce program to in
-00004560: 766f 6b65 2c20 6465 6661 756c 7420 6661  voke, default fa
-00004570: 7374 7472 6565 0a2d 7473 2c20 2d2d 7472  sttree.-ts, --tr
-00004580: 6565 7365 742c 2074 6865 2070 6172 616d  eeset, the param
-00004590: 6574 6572 7320 7365 7474 696e 6720 666f  eters setting fo
-000045a0: 7220 6765 6e65 2074 7265 6520 696e 6665  r gene tree infe
-000045b0: 7265 6e63 652c 2064 6566 6175 6c74 204e  rence, default N
-000045c0: 6f6e 652c 2074 6869 7320 6f70 7469 6f6e  one, this option
-000045d0: 2063 616e 2062 6520 7072 6f76 6964 6564   can be provided
-000045e0: 206d 756c 7469 706c 6520 7469 6d65 730a   multiple times.
-000045f0: 2d2d 636f 6e63 6174 656e 6174 696f 6e2c  --concatenation,
-00004600: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004610: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00004620: 6520 7468 6520 636f 6e63 6174 656e 6174  e the concatenat
-00004630: 696f 6e2d 6261 7365 6420 7370 6563 6965  ion-based specie
-00004640: 7320 7472 6565 2069 6e66 6572 656e 6365  s tree inference
-00004650: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00004660: 7320 7365 742c 2063 6f6e 6361 7465 6e61  s set, concatena
-00004670: 7469 6f6e 2d62 6173 6564 2073 7065 6369  tion-based speci
-00004680: 6573 2074 7265 6520 776f 756c 6420 6265  es tree would be
-00004690: 2069 6e66 6572 6564 0a2d 2d63 6f61 6c65   infered.--coale
-000046a0: 7363 656e 6365 2c20 666c 6167 206f 7074  scence, flag opt
-000046b0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000046c0: 696e 6974 6961 7465 2074 6865 2063 6f61  initiate the coa
-000046d0: 6c65 7363 656e 6365 2d62 6173 6564 2073  lescence-based s
-000046e0: 7065 6369 6573 2074 7265 6520 696e 6665  pecies tree infe
-000046f0: 7265 6e63 652c 2069 6620 7468 6520 666c  rence, if the fl
-00004700: 6167 2077 6173 2073 6574 2c20 636f 616c  ag was set, coal
-00004710: 6573 6365 6e63 652d 6261 7365 6420 7370  escence-based sp
-00004720: 6563 6965 7320 7472 6565 2077 6f75 6c64  ecies tree would
-00004730: 2062 6520 696e 6665 7265 640a 2d73 702c   be infered.-sp,
-00004740: 202d 2d73 7065 6369 6573 7472 6565 2c20   --speciestree, 
-00004750: 7370 6563 6965 7320 7472 6565 2064 6174  species tree dat
-00004760: 6166 696c 6520 666f 7220 6461 7469 6e67  afile for dating
-00004770: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-00004780: 642c 202d 2d64 6174 696e 672c 2077 6869  d, --dating, whi
-00004790: 6368 206d 6f6c 6563 756c 6172 2064 6174  ch molecular dat
-000047a0: 696e 6720 7072 6f67 7261 6d20 746f 2075  ing program to u
-000047b0: 7365 2c20 6465 6661 756c 7420 6e6f 6e65  se, default none
-000047c0: 0a2d 6473 2c20 2d2d 6461 7469 6e67 7365  .-ds, --datingse
-000047d0: 742c 2074 6865 2070 6172 616d 6574 6572  t, the parameter
-000047e0: 7320 7365 7474 696e 6720 666f 7220 6461  s setting for da
-000047f0: 7469 6e67 2070 726f 6772 616d 2c20 6465  ting program, de
-00004800: 6661 756c 7420 4e6f 6e65 2c20 7468 6973  fault None, this
-00004810: 206f 7074 696f 6e20 6361 6e20 6265 2070   option can be p
-00004820: 726f 7669 6465 6420 6d75 6c74 6970 6c65  rovided multiple
-00004830: 2074 696d 6573 0a2d 6e73 2c20 2d2d 6e73   times.-ns, --ns
-00004840: 6974 6573 2c20 7468 6520 6e73 6974 6573  ites, the nsites
-00004850: 2069 6e66 6f72 6d61 7469 6f6e 2066 6f72   information for
-00004860: 2072 3873 2064 6174 696e 672c 2064 6566   r8s dating, def
-00004870: 6175 6c74 204e 6f6e 650a 2d6f 742c 202d  ault None.-ot, -
-00004880: 2d6f 7574 6772 6f75 702c 2074 6865 206f  -outgroup, the o
-00004890: 7574 6772 6f75 7020 696e 666f 726d 6174  utgroup informat
-000048a0: 696f 6e20 666f 7220 7238 7320 6461 7469  ion for r8s dati
-000048b0: 6e67 2c20 6465 6661 756c 7420 4e6f 6e65  ng, default None
-000048c0: 0a2d 7074 2c20 2d2d 7061 7274 6974 696f  .-pt, --partitio
-000048d0: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
-000048e0: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
-000048f0: 6174 6520 7061 7274 6974 696f 6e20 6461  ate partition da
-00004900: 7469 6e67 2061 6e61 6c79 7369 7320 666f  ting analysis fo
-00004910: 7220 636f 646f 6e2c 2069 6620 7468 6520  r codon, if the 
-00004920: 666c 6167 2077 6173 2073 6574 2c20 616e  flag was set, an
-00004930: 2061 6464 6974 696f 6e61 6c20 7061 7274   additional part
-00004940: 6974 696f 6e20 6461 7469 6e67 2061 6e61  ition dating ana
-00004950: 6c79 7369 7320 776f 756c 6420 6265 2069  lysis would be i
-00004960: 6e69 7469 6174 6564 0a2d 616d 2c20 2d2d  nitiated.-am, --
-00004970: 6161 6d6f 6465 6c2c 2077 6869 6368 2070  aamodel, which p
-00004980: 726f 7465 696e 206d 6f64 656c 2074 6f20  rotein model to 
-00004990: 6265 2075 7365 6420 696e 206d 636d 6374  be used in mcmct
-000049a0: 7265 652c 2064 6566 6175 6c74 2070 6f69  ree, default poi
-000049b0: 7373 6f6e 0a2d 6b73 2c20 666c 6167 206f  sson.-ks, flag o
-000049c0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000049d0: 6f20 696e 6974 6961 7465 204b 7320 6361  o initiate Ks ca
-000049e0: 6c63 756c 6174 696f 6e20 666f 7220 686f  lculation for ho
-000049f0: 6d6f 6c6f 6775 6573 2069 6e20 7468 6520  mologues in the 
-00004a00: 7072 6f76 6964 6564 206f 7274 686f 6c6f  provided ortholo
-00004a10: 676f 7573 2067 656e 6520 6661 6d69 6c79  gous gene family
-00004a20: 0a2d 2d61 6e6e 6f74 6174 696f 6e2c 2077  .--annotation, w
-00004a30: 6869 6368 2061 6e6e 6f74 6174 696f 6e20  hich annotation 
-00004a40: 7072 6f67 7261 6d20 746f 2075 7365 2c20  program to use, 
-00004a50: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d70  default None.--p
-00004a60: 6169 7277 6973 652c 2066 6c61 6720 6f70  airwise, flag op
-00004a70: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00004a80: 2069 6e69 7469 6174 6520 7061 6972 7769   initiate pairwi
-00004a90: 7365 204b 7320 6573 7469 6d61 7469 6f6e  se Ks estimation
-00004aa0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00004ab0: 7320 7365 742c 2070 6169 7277 6973 6520  s set, pairwise 
-00004ac0: 4b73 2076 616c 7565 7320 776f 756c 6420  Ks values would 
-00004ad0: 6265 2065 7374 696d 6174 6564 0a2d 6564  be estimated.-ed
-00004ae0: 2c20 2d2d 6567 676e 6f67 6461 7461 2c20  , --eggnogdata, 
-00004af0: 7468 6520 6567 676e 6f67 2061 6e6e 6f74  the eggnog annot
-00004b00: 6174 696f 6e20 6461 7461 6669 6c65 2c20  ation datafile, 
-00004b10: 6465 6661 756c 7420 4e6f 6e65 0a2d 2d70  default None.--p
-00004b20: 6661 6d2c 2077 6869 6368 206f 7074 696f  fam, which optio
-00004b30: 6e20 746f 2075 7365 2066 6f72 2070 6661  n to use for pfa
-00004b40: 6d20 616e 6e6f 7461 7469 6f6e 2c20 6465  m annotation, de
-00004b50: 6661 756c 7420 4e6f 6e65 0a2d 2d64 6d6e  fault None.--dmn
-00004b60: 622c 2074 6865 2064 6961 6d6f 6e64 2064  b, the diamond d
-00004b70: 6174 6162 6173 6520 666f 7220 616e 6e6f  atabase for anno
-00004b80: 7461 7469 6f6e 2c20 6465 6661 756c 7420  tation, default 
-00004b90: 4e6f 6e65 0a2d 2d68 6d6d 2c20 7468 6520  None.--hmm, the 
-00004ba0: 484d 4d20 7072 6f66 696c 6520 666f 7220  HMM profile for 
-00004bb0: 616e 6e6f 7461 7469 6f6e 2c20 6465 6661  annotation, defa
-00004bc0: 756c 7420 4e6f 6e65 0a2d 2d65 7661 6c75  ult None.--evalu
-00004bd0: 652c 2074 6865 2065 2d76 616c 7565 2063  e, the e-value c
-00004be0: 7574 2d6f 6666 2066 6f72 2061 6e6e 6f74  ut-off for annot
-00004bf0: 6174 696f 6e2c 2064 6566 6175 6c74 2031  ation, default 1
-00004c00: 652d 3130 0a2d 2d65 7865 7061 7468 2c20  e-10.--exepath, 
-00004c10: 7468 6520 7061 7468 2074 6f20 7468 6520  the path to the 
-00004c20: 696e 7465 7270 726f 7363 616e 2065 7865  interproscan exe
-00004c30: 6375 7461 626c 652c 2064 6566 6175 6c74  cutable, default
-00004c40: 204e 6f6e 650a 2d66 2c20 2d2d 666f 7373   None.-f, --foss
-00004c50: 696c 2c20 7468 6520 666f 7373 696c 2063  il, the fossil c
-00004c60: 616c 6962 7261 7469 6f6e 2069 6e66 6f72  alibration infor
-00004c70: 6d61 7469 6f6e 2069 6e20 4265 6173 742c  mation in Beast,
-00004c80: 2064 6566 6175 6c74 2028 2763 6c61 6465   default ('clade
-00004c90: 313b 636c 6164 6532 272c 2027 7461 7861  1;clade2', 'taxa
-00004ca0: 312c 7461 7861 323b 7461 7861 332c 7461  1,taxa2;taxa3,ta
-00004cb0: 7861 3427 2c20 2734 3b35 272c 2027 302e  xa4', '4;5', '0.
-00004cc0: 353b 302e 3627 2c20 2734 3030 3b35 3030  5;0.6', '400;500
-00004cd0: 2729 0a2d 7268 2c20 2d2d 726f 6f74 6865  ').-rh, --roothe
-00004ce0: 6967 6874 2c20 7468 6520 726f 6f74 2068  ight, the root h
-00004cf0: 6569 6768 7420 6361 6c69 6272 6174 696f  eight calibratio
-00004d00: 6e20 696e 666f 2069 6e20 4265 6173 742c  n info in Beast,
-00004d10: 2064 6566 6175 6c74 2028 342c 302e 352c   default (4,0.5,
-00004d20: 3430 3029 0a2d 6373 2c20 2d2d 6368 6169  400).-cs, --chai
-00004d30: 6e73 6574 2c20 7468 6520 7061 7261 6d65  nset, the parame
-00004d40: 7465 7273 206f 6620 4d43 4d43 2063 6861  ters of MCMC cha
-00004d50: 696e 2069 6e20 4265 6173 742c 2064 6566  in in Beast, def
-00004d60: 6175 6c74 2028 3130 3030 302c 3130 3029  ault (10000,100)
-00004d70: 0a2d 2d62 6561 7374 6c67 6a61 722c 2074  .--beastlgjar, t
-00004d80: 6865 2070 6174 6820 746f 2062 6561 7374  he path to beast
-00004d90: 4c47 2e6a 6172 2c20 6465 6661 756c 7420  LG.jar, default 
-00004da0: 4e6f 6e65 0a2d 2d62 6561 676c 652c 2066  None.--beagle, f
-00004db0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00004dc0: 6865 7220 746f 2075 7365 2062 6561 676c  her to use beagl
-00004dd0: 6520 696e 2042 6561 7374 2c20 6966 2074  e in Beast, if t
-00004de0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00004df0: 2062 6561 676c 6520 776f 756c 6420 6265   beagle would be
-00004e00: 2075 7365 640a 2d2d 7072 6f74 636f 6364   used.--protcocd
-00004e10: 6174 696e 672c 2066 6c61 6720 6f70 7469  ating, flag opti
-00004e20: 6f6e 2c20 7768 6574 6865 7220 746f 206f  on, whether to o
-00004e30: 6e6c 7920 696e 6974 6961 7465 2074 6865  nly initiate the
-00004e40: 2070 726f 7465 696e 2d63 6f6e 6361 7465   protein-concate
-00004e50: 6e61 7469 6f6e 2d62 6173 6564 2064 6174  nation-based dat
-00004e60: 696e 6720 616e 616c 7973 6973 2c20 6966  ing analysis, if
-00004e70: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00004e80: 742c 2074 6865 2061 6e61 6c79 7369 7320  t, the analysis 
-00004e90: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
-00004ea0: 6564 0a2d 2d70 726f 7464 6174 696e 672c  ed.--protdating,
-00004eb0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00004ec0: 6574 6865 7220 746f 206f 6e6c 7920 696e  ether to only in
-00004ed0: 6974 6961 7465 2074 6865 2070 726f 7465  itiate the prote
-00004ee0: 696e 2d62 6173 6564 2064 6174 696e 6720  in-based dating 
-00004ef0: 616e 616c 7973 6973 2c20 6966 2074 6865  analysis, if the
-00004f00: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00004f10: 6865 2061 6e61 6c79 7369 7320 776f 756c  he analysis woul
-00004f20: 6420 6265 2069 6e69 7469 6174 6564 0a60  d be initiated.`
-00004f30: 6060 0a0a 5468 6520 7072 6f67 7261 6d20  ``..The program 
-00004f40: 6077 6764 206b 7364 6020 6361 6e20 7265  `wgd ksd` can re
-00004f50: 616c 697a 6520 7468 6520 636f 6e73 7472  alize the constr
-00004f60: 7563 7469 6f6e 206f 6620 2a4b 2a3c 7375  uction of *K*<su
-00004f70: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00004f80: 7374 7269 6275 7469 6f6e 2061 6e64 2072  stribution and r
-00004f90: 6174 6520 636f 7272 6563 7469 6f6e 2e0a  ate correction..
-00004fa0: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
-00004fb0: 6c69 6573 2073 6571 7565 6e63 6573 2028  lies sequences (
-00004fc0: 6f70 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d  option).--------
-00004fd0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004fe0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00004ff0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005000: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00005010: 2d2d 2d2d 2d2d 2d2d 0a2d 6f2c 202d 2d6f  --------.-o, --o
-00005020: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
-00005030: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
-00005040: 6175 6c74 2077 6764 5f6b 7364 0a2d 742c  ault wgd_ksd.-t,
-00005050: 202d 2d74 6d70 6469 722c 2074 6865 2074   --tmpdir, the t
-00005060: 656d 706f 7261 7279 2077 6f72 6b69 6e67  emporary working
-00005070: 2064 6972 6563 746f 7279 2c20 6465 6661   directory, defa
-00005080: 756c 7420 4e6f 6e65 2c20 6966 204e 6f6e  ult None, if Non
-00005090: 6520 7761 7320 6769 7665 6e2c 2074 6865  e was given, the
-000050a0: 2074 6d70 6469 7220 7769 6c6c 2062 6520   tmpdir will be 
-000050b0: 6173 7369 676e 6564 2072 616e 646f 6d20  assigned random 
-000050c0: 6e61 6d65 7320 696e 2063 7572 7265 6e74  names in current
-000050d0: 2064 6972 6563 746f 7279 2061 6e64 2061   directory and a
-000050e0: 7574 6f6d 6174 656c 7920 7265 6d6f 7665  utomately remove
-000050f0: 6420 6174 2074 6865 2063 6f6d 706c 6574  d at the complet
-00005100: 696f 6e20 6f66 2070 726f 6772 616d 2c20  ion of program, 
-00005110: 656c 7365 2074 6865 2074 6d70 6469 7220  else the tmpdir 
-00005120: 776f 756c 6420 6265 206b 6570 740a 2d6e  would be kept.-n
-00005130: 2c20 2d2d 6e74 6872 6561 6473 2c20 7468  , --nthreads, th
-00005140: 6520 6e75 6d62 6572 206f 6620 7468 7265  e number of thre
-00005150: 6164 7320 746f 2075 7365 2c20 6465 6661  ads to use, defa
-00005160: 756c 7420 340a 2d2d 746f 5f73 746f 702c  ult 4.--to_stop,
-00005170: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005180: 6574 6865 7220 746f 2074 7261 6e73 6c61  ether to transla
-00005190: 7465 2074 6872 6f75 6768 2053 544f 5020  te through STOP 
-000051a0: 636f 646f 6e73 2c20 6966 2074 6865 2066  codons, if the f
-000051b0: 6c61 6720 7761 7320 7365 742c 2074 7261  lag was set, tra
-000051c0: 6e73 6c61 7469 6f6e 2077 696c 6c20 6265  nslation will be
-000051d0: 2074 6572 6d69 6e61 7465 6420 6174 2074   terminated at t
-000051e0: 6865 2066 6972 7374 2069 6e2d 6672 616d  he first in-fram
-000051f0: 6520 7374 6f70 2063 6f64 6f6e 2c20 656c  e stop codon, el
-00005200: 7365 2061 2066 756c 6c20 7472 616e 736c  se a full transl
-00005210: 6174 696f 6e20 636f 6e74 696e 7569 6e67  ation continuing
-00005220: 206f 6e20 7061 7374 2061 6e79 2073 746f   on past any sto
-00005230: 7020 636f 646f 6e73 2077 6f75 6c64 2062  p codons would b
-00005240: 6520 696e 6974 6961 7465 640a 2d2d 6364  e initiated.--cd
-00005250: 732c 2066 6c61 6720 6f70 7469 6f6e 2c20  s, flag option, 
-00005260: 7768 6574 6865 7220 746f 206f 6e6c 7920  whether to only 
-00005270: 7472 616e 736c 6174 6520 7468 6520 636f  translate the co
-00005280: 6d70 6c65 7465 2043 4453 2074 6861 7420  mplete CDS that 
-00005290: 7374 6172 7473 2077 6974 6820 6120 7661  starts with a va
-000052a0: 6c69 6420 7374 6172 7420 636f 646f 6e20  lid start codon 
-000052b0: 616e 6420 6f6e 6c79 2063 6f6e 7461 696e  and only contain
-000052c0: 7320 6120 7369 6e67 6c65 2069 6e2d 6672  s a single in-fr
-000052d0: 616d 6520 7374 6f70 2063 6f64 6f6e 2061  ame stop codon a
-000052e0: 7420 7468 6520 656e 6420 616e 6420 6d75  t the end and mu
-000052f0: 7374 2062 6520 6469 7669 6461 626c 6520  st be dividable 
-00005300: 6279 2074 6872 6565 2c20 6966 2074 6865  by three, if the
-00005310: 2066 6c61 6720 7761 7320 7365 742c 206f   flag was set, o
-00005320: 6e6c 7920 7468 6520 636f 6d70 6c65 7465  nly the complete
-00005330: 2043 4453 2077 6f75 6c64 2062 6520 7472   CDS would be tr
-00005340: 616e 736c 6174 6564 0a2d 2d70 6169 7277  anslated.--pairw
-00005350: 6973 652c 2066 6c61 6720 6f70 7469 6f6e  ise, flag option
-00005360: 2c20 7768 6574 6865 7220 746f 2069 6e69  , whether to ini
-00005370: 7469 6174 6520 7061 6972 7769 7365 204b  tiate pairwise K
-00005380: 7320 6573 7469 6d61 7469 6f6e 2c20 6966  s estimation, if
-00005390: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-000053a0: 742c 2070 6169 7277 6973 6520 4b73 2076  t, pairwise Ks v
-000053b0: 616c 7565 7320 776f 756c 6420 6265 2065  alues would be e
-000053c0: 7374 696d 6174 6564 0a2d 2d73 7472 6970  stimated.--strip
-000053d0: 5f67 6170 732c 2066 6c61 6720 6f70 7469  _gaps, flag opti
-000053e0: 6f6e 2c20 7768 6574 6865 7220 746f 2064  on, whether to d
-000053f0: 726f 7020 616c 6c20 6761 7073 2069 6e20  rop all gaps in 
-00005400: 6d75 6c74 6970 6c65 2073 6571 7565 6e63  multiple sequenc
-00005410: 6520 616c 6967 6e6d 656e 742c 2069 6620  e alignment, if 
-00005420: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
-00005430: 2c20 616c 6c20 6761 7073 2077 6f75 6c64  , all gaps would
-00005440: 2062 6520 6472 6f70 7065 640a 2d61 2c20   be dropped.-a, 
-00005450: 2d2d 616c 6967 6e65 722c 2077 6869 6368  --aligner, which
-00005460: 2061 6c69 676e 6d65 6e74 2070 726f 6772   alignment progr
-00005470: 616d 2074 6f20 7573 652c 2064 6566 6175  am to use, defau
-00005480: 6c74 206d 6166 6674 200a 2d74 7265 652c  lt mafft .-tree,
-00005490: 202d 2d74 7265 655f 6d65 7468 6f64 2c20   --tree_method, 
-000054a0: 7768 6963 6820 6765 6e65 2074 7265 6520  which gene tree 
-000054b0: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-000054c0: 6d20 746f 2069 6e76 6f6b 652c 2064 6566  m to invoke, def
-000054d0: 6175 6c74 2066 6173 7474 7265 650a 2d2d  ault fasttree.--
-000054e0: 7472 6565 5f6f 7074 696f 6e73 2c20 6f70  tree_options, op
-000054f0: 7469 6f6e 7320 696e 2074 7265 6520 696e  tions in tree in
-00005500: 6665 7265 6e63 6520 6173 2061 2063 6f6d  ference as a com
-00005510: 6d61 2073 6570 6172 6174 6564 2073 7472  ma separated str
-00005520: 696e 672c 2064 6566 6175 6c74 204e 6f6e  ing, default Non
-00005530: 650a 2d2d 6e6f 6465 5f61 7665 7261 6765  e.--node_average
-00005540: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
-00005550: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
-00005560: 7465 206e 6f64 652d 6176 6572 6167 6520  te node-average 
-00005570: 7761 7920 6f66 2064 652d 7265 6475 6e64  way of de-redund
-00005580: 616e 6379 2069 6e73 7465 6164 206f 6620  ancy instead of 
-00005590: 6e6f 6465 2d77 6569 6768 7465 642c 2069  node-weighted, i
-000055a0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
-000055b0: 6574 2c20 7468 6520 6e6f 6465 2d61 7665  et, the node-ave
-000055c0: 7261 6769 6e67 2064 652d 7265 6475 6e64  raging de-redund
-000055d0: 616e 6379 2077 6f75 6c64 2062 6520 696e  ancy would be in
-000055e0: 6974 6961 7465 640a 2d73 722c 202d 2d73  itiated.-sr, --s
-000055f0: 7061 6972 2c20 7468 6520 7370 6563 6965  pair, the specie
-00005600: 7320 7061 6972 2074 6f20 6265 2070 6c6f  s pair to be plo
-00005610: 7474 6564 2c20 6465 6661 756c 7420 4e6f  tted, default No
-00005620: 6e65 2c20 7468 6973 206f 7074 696f 6e20  ne, this option 
-00005630: 6361 6e20 6265 2070 726f 7669 6465 6420  can be provided 
-00005640: 6d75 6c74 6970 6c65 2074 696d 6573 0a2d  multiple times.-
-00005650: 7370 2c20 2d2d 7370 6563 6965 7374 7265  sp, --speciestre
-00005660: 652c 2074 6865 2073 7065 6369 6573 2074  e, the species t
-00005670: 7265 6520 746f 2070 6572 666f 726d 2072  ree to perform r
-00005680: 6174 6520 636f 7272 6563 7469 6f6e 2c20  ate correction, 
-00005690: 6465 6661 756c 7420 4e6f 6e65 2c20 6966  default None, if
-000056a0: 204e 6f6e 6520 7761 7320 6769 7665 6e2c   None was given,
-000056b0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
-000056c0: 7469 6f6e 2061 6e61 6c79 7369 7320 776f  tion analysis wo
-000056d0: 756c 6420 6265 2063 616c 6c65 6420 6f66  uld be called of
-000056e0: 660a 2d72 772c 202d 2d72 6577 6569 6768  f.-rw, --reweigh
-000056f0: 742c 2066 6c61 6720 6f70 7469 6f6e 2c20  t, flag option, 
-00005700: 7768 6574 6865 7220 746f 2072 6563 616c  whether to recal
-00005710: 6375 6c61 7465 2074 6865 2077 6569 6768  culate the weigh
-00005720: 7420 7065 7220 7370 6563 6965 7320 7061  t per species pa
-00005730: 6972 2c20 6966 2074 6865 2066 6c61 6720  ir, if the flag 
-00005740: 7761 7320 7365 742c 2074 6865 2077 6569  was set, the wei
-00005750: 6768 7420 776f 756c 6420 6265 2072 6563  ght would be rec
-00005760: 616c 6375 6c61 7465 640a 2d6f 722c 202d  alculated.-or, -
-00005770: 2d6f 6e6c 7972 6f6f 746f 7574 2c20 666c  -onlyrootout, fl
-00005780: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00005790: 6572 2074 6f20 6f6e 6c79 2063 6f6e 6475  er to only condu
-000057a0: 6374 2072 6174 6520 636f 7272 6563 7469  ct rate correcti
-000057b0: 6f6e 2075 7369 6e67 2074 6865 206f 7574  on using the out
-000057c0: 6772 6f75 7020 6174 2072 6f6f 7420 6173  group at root as
-000057d0: 206f 7574 6772 6f75 702c 2069 6620 7468   outgroup, if th
-000057e0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000057f0: 6f6e 6c79 2074 6865 206f 7574 6772 6f75  only the outgrou
-00005800: 7020 6174 2072 6f6f 7420 776f 756c 6420  p at root would 
-00005810: 6265 2075 7365 6420 6173 206f 7574 6772  be used as outgr
-00005820: 6f75 700a 2d65 706b 2c20 2d2d 6578 7472  oup.-epk, --extr
-00005830: 6170 6172 616e 6f6d 656b 732c 2065 7874  aparanomeks, ext
-00005840: 7261 2070 6172 616e 6f6d 6520 4b73 2064  ra paranome Ks d
-00005850: 6174 6120 746f 2070 6c6f 7420 696e 2074  ata to plot in t
-00005860: 6865 206d 6978 6564 204b 7320 6469 7374  he mixed Ks dist
-00005870: 7269 6275 7469 6f6e 2c20 6465 6661 756c  ribution, defaul
-00005880: 7420 4e6f 6e65 0a2d 6170 2c20 2d2d 616e  t None.-ap, --an
-00005890: 6368 6f72 706f 696e 7473 2c20 616e 6368  chorpoints, anch
-000058a0: 6f72 706f 696e 7473 2e74 7874 2066 696c  orpoints.txt fil
-000058b0: 6520 746f 2070 6c6f 7420 616e 6368 6f72  e to plot anchor
-000058c0: 204b 7320 696e 2074 6865 206d 6978 6564   Ks in the mixed
-000058d0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-000058e0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
-000058f0: 706b 2c20 2d2d 706c 6f74 6b64 652c 2066  pk, --plotkde, f
-00005900: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00005910: 6865 7220 746f 2070 6c6f 7420 6b64 6520  her to plot kde 
-00005920: 6375 7276 6520 6f66 206f 7274 686f 6c6f  curve of ortholo
-00005930: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
-00005940: 7469 6f6e 206f 7665 7220 6869 7374 6f67  tion over histog
-00005950: 7261 6d20 696e 2074 6865 206d 6978 6564  ram in the mixed
-00005960: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005970: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005980: 7320 7365 742c 2074 6865 206b 6465 2063  s set, the kde c
-00005990: 7572 7665 2077 6f75 6c64 2062 6520 706c  urve would be pl
-000059a0: 6f74 7465 640a 2d70 6167 2c20 2d2d 706c  otted.-pag, --pl
-000059b0: 6f74 6170 676d 6d2c 2066 6c61 6720 6f70  otapgmm, flag op
-000059c0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-000059d0: 2070 6572 666f 726d 2061 6e64 2070 6c6f   perform and plo
-000059e0: 7420 6d69 7874 7572 6520 6d6f 6465 6c69  t mixture modeli
-000059f0: 6e67 206f 6620 616e 6368 6f72 204b 7320  ng of anchor Ks 
-00005a00: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
-00005a10: 6469 7374 7269 6275 7469 6f6e 2c20 6966  distribution, if
-00005a20: 2074 6865 2066 6c61 6720 7761 7320 7365   the flag was se
-00005a30: 742c 2074 6865 206d 6978 7475 7265 206d  t, the mixture m
-00005a40: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
-00005a50: 7220 4b73 2077 6f75 6c64 2062 6520 706c  r Ks would be pl
-00005a60: 6f74 7465 640a 2d70 656d 2c20 2d2d 706c  otted.-pem, --pl
-00005a70: 6f74 656c 6d6d 2c20 666c 6167 206f 7074  otelmm, flag opt
-00005a80: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-00005a90: 7065 7266 6f72 6d20 616e 6420 706c 6f74  perform and plot
-00005aa0: 2065 6c6d 6d20 6d69 7874 7572 6520 6d6f   elmm mixture mo
-00005ab0: 6465 6c69 6e67 206f 6620 7061 7261 6e6f  deling of parano
-00005ac0: 6d65 204b 7320 696e 2074 6865 206d 6978  me Ks in the mix
-00005ad0: 6564 204b 7320 6469 7374 7269 6275 7469  ed Ks distributi
-00005ae0: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-00005af0: 7761 7320 7365 742c 2074 6865 2065 6c6d  was set, the elm
-00005b00: 6d20 6d69 7874 7572 6520 6d6f 6465 6c69  m mixture modeli
-00005b10: 6e67 206f 6620 7061 7261 6e6f 6d65 204b  ng of paranome K
-00005b20: 7320 776f 756c 6420 6265 2070 6c6f 7474  s would be plott
-00005b30: 6564 0a2d 632c 202d 2d63 6f6d 706f 6e65  ed.-c, --compone
-00005b40: 6e74 732c 2074 6865 2072 616e 6765 206f  nts, the range o
-00005b50: 6620 7468 6520 6e75 6d62 6572 206f 6620  f the number of 
-00005b60: 636f 6d70 6f6e 656e 7473 2074 6f20 6669  components to fi
-00005b70: 7420 696e 2061 6e63 686f 7220 4b73 206d  t in anchor Ks m
-00005b80: 6978 7475 7265 206d 6f64 656c 696e 672c  ixture modeling,
-00005b90: 2064 6566 6175 6c74 2028 312c 3429 0a2d   default (1,4).-
-00005ba0: 786c 2c20 2d2d 786c 696d 2c20 7468 6520  xl, --xlim, the 
-00005bb0: 7820 6178 6973 206c 696d 6974 206f 6620  x axis limit of 
-00005bc0: 4b73 2064 6973 7472 6962 7574 696f 6e0a  Ks distribution.
-00005bd0: 2d79 6c2c 202d 2d79 6c69 6d2c 2074 6865  -yl, --ylim, the
-00005be0: 2079 2061 7869 7320 6c69 6d69 7420 6f66   y axis limit of
-00005bf0: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00005c00: 0a2d 6164 6f2c 202d 2d61 646a 7573 746f  .-ado, --adjusto
-00005c10: 7274 686f 2c20 666c 6167 206f 7074 696f  rtho, flag optio
-00005c20: 6e2c 2077 6865 7468 6572 2074 6f20 6164  n, whether to ad
-00005c30: 6a75 7374 2074 6865 2068 6973 746f 6772  just the histogr
-00005c40: 616d 2068 6569 6768 7420 6f66 206f 7274  am height of ort
-00005c50: 686f 6c6f 676f 7573 204b 7320 6173 2074  hologous Ks as t
-00005c60: 6f20 6d61 7463 6820 7468 6520 6865 6967  o match the heig
-00005c70: 6874 206f 6620 7061 7261 6c6f 676f 7573  ht of paralogous
-00005c80: 204b 732c 2069 6620 7468 6520 666c 6167   Ks, if the flag
-00005c90: 2077 6173 2073 6574 2c20 7468 6520 6164   was set, the ad
-00005ca0: 6a75 7374 6d65 6e74 2077 6f75 6c64 2062  justment would b
-00005cb0: 6520 636f 6e64 7563 7465 640a 2d61 6466  e conducted.-adf
-00005cc0: 2c20 2d2d 6164 6a75 7374 6661 6374 6f72  , --adjustfactor
-00005cd0: 2c20 7468 6520 6164 6a75 7374 6d65 6e74  , the adjustment
-00005ce0: 2066 6163 746f 7220 6f66 206f 7274 686f   factor of ortho
-00005cf0: 6c6f 676f 7573 204b 732c 2064 6566 6175  logous Ks, defau
-00005d00: 6c74 2030 2e35 0a2d 6f61 2c20 2d2d 6f6b  lt 0.5.-oa, --ok
-00005d10: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
-00005d20: 7479 206f 6620 6f72 7468 6f6c 6f67 6f75  ty of orthologou
-00005d30: 7320 4b73 2064 6973 7472 6962 7574 696f  s Ks distributio
-00005d40: 6e20 696e 206d 6978 6564 2070 6c6f 742c  n in mixed plot,
-00005d50: 2064 6566 6175 6c74 2030 2e35 0a2d 6661   default 0.5.-fa
-00005d60: 2c20 2d2d 666f 6375 7332 616c 6c2c 2073  , --focus2all, s
-00005d70: 6574 2066 6f63 616c 2073 7065 6369 6573  et focal species
-00005d80: 2061 6e64 206c 6574 2073 7065 6369 6573   and let species
-00005d90: 2070 6169 7220 746f 2062 6520 6265 7477   pair to be betw
-00005da0: 6565 6e20 666f 6361 6c20 616e 6420 616c  een focal and al
-00005db0: 6c20 7468 6520 7265 6d61 696e 696e 6720  l the remaining 
-00005dc0: 7370 6563 6965 732c 2064 6566 6175 6c74  species, default
-00005dd0: 204e 6f6e 650a 2d6b 732c 202d 2d6b 7374   None.-ks, --kst
-00005de0: 7265 652c 2066 6c61 6720 6f70 7469 6f6e  ree, flag option
-00005df0: 2c20 7768 6574 6865 7220 746f 2069 6e66  , whether to inf
-00005e00: 6572 204b 7320 7472 6565 2c20 6966 2074  er Ks tree, if t
-00005e10: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00005e20: 2074 6865 204b 7320 7472 6565 2069 6e66   the Ks tree inf
-00005e30: 6572 656e 6365 2061 6e61 6c79 7369 7320  erence analysis 
-00005e40: 776f 756c 6420 6265 2069 6e69 7469 6174  would be initiat
-00005e50: 6564 0a2d 6f63 6b2c 202d 2d6f 6e6c 7963  ed.-ock, --onlyc
-00005e60: 6f6e 6361 746b 7374 7265 652c 2066 6c61  oncatkstree, fla
-00005e70: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00005e80: 7220 746f 206f 6e6c 7920 696e 6665 7220  r to only infer 
-00005e90: 4b73 2074 7265 6520 756e 6465 7220 636f  Ks tree under co
-00005ea0: 6e63 6174 656e 6174 6564 2061 6c69 676e  ncatenated align
-00005eb0: 6d65 6e74 2c20 6966 2074 6865 2066 6c61  ment, if the fla
-00005ec0: 6720 7761 7320 7365 742c 206f 6e6c 7920  g was set, only 
-00005ed0: 7468 6520 4b73 2074 7265 6520 756e 6465  the Ks tree unde
-00005ee0: 7220 636f 6e63 6174 656e 6174 6564 2061  r concatenated a
-00005ef0: 6c69 676e 6d65 6e74 2077 6f75 6c64 2062  lignment would b
-00005f00: 6520 6361 6c63 756c 6174 6564 0a2d 6373  e calculated.-cs
-00005f10: 2c20 2d2d 636c 6173 7369 632c 2066 6c61  , --classic, fla
-00005f20: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-00005f30: 7220 746f 2064 7261 7720 6d69 7865 6420  r to draw mixed 
-00005f40: 4b73 2070 6c6f 7420 696e 2061 2063 6c61  Ks plot in a cla
-00005f50: 7373 6963 206d 616e 6e65 7220 7768 6572  ssic manner wher
-00005f60: 6520 7468 6520 6675 6c6c 206f 7274 686f  e the full ortho
-00005f70: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
-00005f80: 6275 7469 6f6e 2069 7320 6472 6177 6564  bution is drawed
-00005f90: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
-00005fa0: 7320 7365 742c 2074 6865 2063 6c61 7373  s set, the class
-00005fb0: 6963 206d 6978 6564 204b 7320 706c 6f74  ic mixed Ks plot
-00005fc0: 2077 6f75 6c64 2062 6520 6472 6177 6e0a   would be drawn.
-00005fd0: 2d74 612c 202d 2d74 6f70 6172 726f 772c  -ta, --toparrow,
-00005fe0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00005ff0: 6574 6865 7220 746f 2061 646a 7573 7420  ether to adjust 
-00006000: 7468 6520 6172 726f 7720 746f 2062 6520  the arrow to be 
-00006010: 6174 2074 6865 2074 6f70 206f 6620 7468  at the top of th
-00006020: 6520 706c 6f74 2c20 696e 7374 6561 6420  e plot, instead 
-00006030: 6f66 2062 6569 6e67 2063 6f6f 7264 696e  of being coordin
-00006040: 6174 6564 2061 7320 7468 6520 4b44 4520  ated as the KDE 
-00006050: 6f66 2074 6865 206f 7274 686f 6c6f 676f  of the orthologo
-00006060: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
-00006070: 6f6e 2c20 6966 2074 6865 2066 6c61 6720  on, if the flag 
-00006080: 7761 7320 7365 742c 2074 6865 2061 7272  was set, the arr
-00006090: 6f77 2077 6f75 6c64 2062 6520 7365 7420  ow would be set 
-000060a0: 6174 2074 6865 2074 6f70 0a2d 6273 2c20  at the top.-bs, 
-000060b0: 2d2d 626f 6f74 7374 7261 702c 2074 6865  --bootstrap, the
-000060c0: 206e 756d 6265 7220 6f66 2062 6f6f 7473   number of boots
-000060d0: 7472 6170 2072 6570 6c69 6361 7465 7320  trap replicates 
-000060e0: 6f66 206f 7274 686f 6c6f 6720 4b73 2064  of ortholog Ks d
-000060f0: 6973 7472 6962 7574 696f 6e20 696e 206d  istribution in m
-00006100: 6978 6564 2070 6c6f 740a 6060 600a 0a54  ixed plot.```..T
-00006110: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-00006120: 6d69 7860 2063 616e 2072 6561 6c69 7a65  mix` can realize
-00006130: 2074 6865 206d 6978 7475 7265 206d 6f64   the mixture mod
-00006140: 656c 2063 6c75 7374 6572 696e 6720 616e  el clustering an
-00006150: 616c 7973 6973 206f 6620 2a4b 2a3c 7375  alysis of *K*<su
-00006160: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00006170: 7374 7269 6275 7469 6f6e 2e0a 6060 600a  stribution..```.
-00006180: 7767 6420 6d69 7820 6b73 5f64 6174 6166  wgd mix ks_dataf
-00006190: 696c 6520 286f 7074 696f 6e29 0a2d 2d2d  ile (option).---
-000061a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000061e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d66  -------------.-f
-000061f0: 2c20 2d2d 6669 6c74 6572 732c 2074 6865  , --filters, the
-00006200: 2063 7574 6f66 6620 616c 6967 6e6d 656e   cutoff alignmen
-00006210: 7420 6c65 6e67 7468 2c20 6465 6661 756c  t length, defaul
-00006220: 7420 3330 300a 2d72 2c20 2d2d 6b73 5f72  t 300.-r, --ks_r
-00006230: 616e 6765 2c20 7468 6520 4b73 2072 616e  ange, the Ks ran
-00006240: 6765 2074 6f20 6265 2063 6f6e 7369 6465  ge to be conside
-00006250: 7265 642c 2064 6566 6175 6c74 2028 302c  red, default (0,
-00006260: 2035 290a 2d62 2c20 2d2d 6269 6e73 2c20   5).-b, --bins, 
-00006270: 7468 6520 6e75 6d62 6572 206f 6620 6269  the number of bi
-00006280: 6e73 2069 6e20 4b73 2064 6973 7472 6962  ns in Ks distrib
-00006290: 7574 696f 6e2c 2064 6566 6175 6c74 2035  ution, default 5
-000062a0: 300a 2d6f 2c20 2d2d 6f75 7464 6972 2c20  0.-o, --outdir, 
-000062b0: 7468 6520 6f75 7470 7574 2064 6972 6563  the output direc
-000062c0: 746f 7279 2c20 6465 6661 756c 7420 7767  tory, default wg
-000062d0: 645f 6d69 780a 2d2d 6d65 7468 6f64 2c20  d_mix.--method, 
-000062e0: 7768 6963 6820 6d69 7874 7572 6520 6d6f  which mixture mo
-000062f0: 6465 6c20 746f 2075 7365 2c20 6465 6661  del to use, defa
-00006300: 756c 7420 676d 6d0a 2d6e 2c20 2d2d 636f  ult gmm.-n, --co
-00006310: 6d70 6f6e 656e 7473 2c20 7468 6520 7261  mponents, the ra
-00006320: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-00006330: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-00006340: 746f 2066 6974 2c20 6465 6661 756c 7420  to fit, default 
-00006350: 2831 2c20 3429 0a2d 672c 202d 2d67 616d  (1, 4).-g, --gam
-00006360: 6d61 2c20 7468 6520 6761 6d6d 6120 7061  ma, the gamma pa
-00006370: 7261 6d65 7465 7220 666f 7220 6267 6d6d  rameter for bgmm
-00006380: 206d 6f64 656c 732c 2064 6566 6175 6c74   models, default
-00006390: 2030 2e30 3031 0a2d 6e69 2c20 2d2d 6e5f   0.001.-ni, --n_
-000063a0: 696e 6974 2c20 7468 6520 6e75 6d62 6572  init, the number
-000063b0: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-000063c0: 6961 6c69 7a61 7469 6f6e 732c 2064 6566  ializations, def
-000063d0: 6175 6c74 2032 3030 0a2d 6d69 2c20 2d2d  ault 200.-mi, --
-000063e0: 6d61 785f 6974 6572 2c20 7468 6520 6d61  max_iter, the ma
-000063f0: 7869 6d75 6d20 6e75 6d62 6572 206f 6620  ximum number of 
-00006400: 6974 6572 6174 696f 6e73 2c20 6465 6661  iterations, defa
-00006410: 756c 7420 3230 300a 6060 600a 0a54 6865  ult 200.```..The
-00006420: 2070 726f 6772 616d 2060 7767 6420 7065   program `wgd pe
-00006430: 616b 6020 6361 6e20 7265 616c 697a 6520  ak` can realize 
-00006440: 7468 6520 7365 6172 6368 206f 6620 6372  the search of cr
-00006450: 6564 6961 626c 6520 2a4b 2a3c 7375 623e  ediable *K*<sub>
-00006460: 533c 2f73 7562 3e20 7261 6e67 6520 7573  S</sub> range us
-00006470: 6564 2069 6e20 5747 4420 6461 7469 6e67  ed in WGD dating
-00006480: 2e0a 6060 600a 7767 6420 7065 616b 206b  ..```.wgd peak k
-00006490: 735f 6461 7461 6669 6c65 2028 6f70 7469  s_datafile (opti
-000064a0: 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  on).------------
-000064b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000064f0: 2d2d 2d2d 0a2d 6170 2c20 2d2d 616e 6368  ----.-ap, --anch
-00006500: 6f72 706f 696e 7473 2c20 7468 6520 616e  orpoints, the an
-00006510: 6368 6f72 2070 6f69 6e74 7320 6461 7461  chor points data
-00006520: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00006530: 6e65 0a2d 736d 2c20 2d2d 7365 676d 656e  ne.-sm, --segmen
-00006540: 7473 2c20 7468 6520 7365 676d 656e 7473  ts, the segments
-00006550: 2064 6174 6166 696c 652c 2064 6566 6175   datafile, defau
-00006560: 6c74 204e 6f6e 650a 2d6c 652c 202d 2d6c  lt None.-le, --l
-00006570: 6973 7465 6c65 6d65 6e74 732c 2074 6865  istelements, the
-00006580: 206c 6973 7473 6567 6d65 6e74 7320 6461   listsegments da
-00006590: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-000065a0: 4e6f 6e65 200a 2d6d 702c 202d 2d6d 756c  None .-mp, --mul
-000065b0: 7469 706c 6963 6f6e 7061 6972 732c 2074  tipliconpairs, t
-000065c0: 6865 206d 756c 7469 706c 6963 6f6e 7061  he multipliconpa
-000065d0: 6972 7320 6461 7461 6669 6c65 2c20 6465  irs datafile, de
-000065e0: 6661 756c 7420 4e6f 6e65 0a2d 6f2c 202d  fault None.-o, -
-000065f0: 2d6f 7574 6469 722c 2074 6865 206f 7574  -outdir, the out
-00006600: 7075 7420 6469 7265 6374 6f72 792c 2064  put directory, d
-00006610: 6566 6175 6c74 2077 6764 5f70 6561 6b0a  efault wgd_peak.
-00006620: 2d61 662c 202d 2d61 6c69 676e 6669 6c74  -af, --alignfilt
-00006630: 6572 2c20 6375 746f 6666 2066 6f72 2061  er, cutoff for a
-00006640: 6c69 676e 6d65 6e74 2069 6465 6e74 6974  lignment identit
-00006650: 792c 206c 656e 6774 6820 616e 6420 636f  y, length and co
-00006660: 7665 7261 6765 2c20 6465 6661 756c 7420  verage, default 
-00006670: 302e 302c 2030 2c20 302e 300a 2d72 2c20  0.0, 0, 0.0.-r, 
-00006680: 2d2d 6b73 7261 6e67 652c 2072 616e 6765  --ksrange, range
-00006690: 206f 6620 4b73 2074 6f20 6265 2061 6e61   of Ks to be ana
-000066a0: 6c79 7a65 642c 2064 6566 6175 6c74 2028  lyzed, default (
-000066b0: 302c 2035 290a 2d62 772c 202d 2d62 696e  0, 5).-bw, --bin
-000066c0: 5f77 6964 7468 2c20 6261 6e64 7769 6474  _width, bandwidt
-000066d0: 6820 6f66 204b 7320 6469 7374 7269 6275  h of Ks distribu
-000066e0: 7469 6f6e 2c20 6465 6661 756c 7420 302e  tion, default 0.
-000066f0: 310a 2d69 632c 202d 2d77 6569 6768 7473  1.-ic, --weights
-00006700: 5f6f 7574 6c69 6572 735f 696e 636c 7564  _outliers_includ
-00006710: 6564 2c20 666c 6167 206f 7074 696f 6e2c  ed, flag option,
-00006720: 2077 6865 7468 6572 2074 6f20 696e 636c   whether to incl
-00006730: 7564 6520 4b73 206f 7574 6c69 6572 732c  ude Ks outliers,
-00006740: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00006750: 2073 6574 2c20 4b73 206f 7574 6c69 6572   set, Ks outlier
-00006760: 7320 776f 756c 6420 6265 2069 6e63 6c75  s would be inclu
-00006770: 6465 6420 696e 2074 6865 2061 6e61 6c79  ded in the analy
-00006780: 7369 730a 2d6d 2c20 2d2d 6d65 7468 6f64  sis.-m, --method
-00006790: 2c20 7768 6963 6820 6d69 7874 7572 6520  , which mixture 
-000067a0: 6d6f 6465 6c20 746f 2075 7365 2c20 6465  model to use, de
-000067b0: 6661 756c 7420 676d 6d0a 2d2d 7365 6564  fault gmm.--seed
-000067c0: 2c20 7261 6e64 6f6d 2073 6565 6420 6769  , random seed gi
-000067d0: 7665 6e20 746f 2069 6e69 7469 616c 697a  ven to initializ
-000067e0: 6174 696f 6e2c 2064 6566 6175 6c74 2032  ation, default 2
-000067f0: 3335 3238 3930 0a2d 6569 2c20 2d2d 656d  352890.-ei, --em
-00006800: 5f69 7465 722c 2074 6865 206e 756d 6265  _iter, the numbe
-00006810: 7220 6f66 2045 4d20 6974 6572 6174 696f  r of EM iteratio
-00006820: 6e73 2074 6f20 7065 7266 6f72 6d2c 2064  ns to perform, d
-00006830: 6566 6175 6c74 2032 3030 0a2d 6e69 2c20  efault 200.-ni, 
-00006840: 2d2d 6e5f 696e 6974 2c20 7468 6520 6e75  --n_init, the nu
-00006850: 6d62 6572 206f 6620 6b2d 6d65 616e 7320  mber of k-means 
-00006860: 696e 6974 6961 6c69 7a61 7469 6f6e 732c  initializations,
-00006870: 2064 6566 6175 6c74 2032 3030 0a2d 6e2c   default 200.-n,
-00006880: 202d 2d63 6f6d 706f 6e65 6e74 732c 2074   --components, t
-00006890: 6865 2072 616e 6765 206f 6620 7468 6520  he range of the 
-000068a0: 6e75 6d62 6572 206f 6620 636f 6d70 6f6e  number of compon
-000068b0: 656e 7473 2074 6f20 6669 742c 2064 6566  ents to fit, def
-000068c0: 6175 6c74 2028 312c 2034 290a 2d67 2c20  ault (1, 4).-g, 
-000068d0: 2d2d 6761 6d6d 612c 2074 6865 2067 616d  --gamma, the gam
-000068e0: 6d61 2070 6172 616d 6574 6572 2066 6f72  ma parameter for
-000068f0: 2062 676d 6d20 6d6f 6465 6c73 2c20 6465   bgmm models, de
-00006900: 6661 756c 7420 3165 2d33 0a2d 2d62 6f6f  fault 1e-3.--boo
-00006910: 7473 2c20 7468 6520 6e75 6d62 6572 206f  ts, the number o
-00006920: 6620 626f 6f74 7374 7261 7020 7265 706c  f bootstrap repl
-00006930: 6963 6174 6573 206f 6620 6b64 652c 2064  icates of kde, d
-00006940: 6566 6175 6c74 2032 3030 0a2d 2d77 6569  efault 200.--wei
-00006950: 6768 7465 642c 2066 6c61 6720 6f70 7469  ghted, flag opti
-00006960: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
-00006970: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
-00006980: 206d 6574 686f 6420 6f66 2064 652d 7265   method of de-re
-00006990: 6475 6e64 616e 6379 2c20 6966 2074 6865  dundancy, if the
-000069a0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-000069b0: 6865 206e 6f64 652d 7765 6967 6874 6564  he node-weighted
-000069c0: 206d 6574 686f 6420 776f 756c 6420 6265   method would be
-000069d0: 2075 7365 640a 2d70 2c20 2d2d 706c 6f74   used.-p, --plot
-000069e0: 2c20 7468 6520 706c 6f74 7469 6e67 206d  , the plotting m
-000069f0: 6574 686f 6420 746f 2062 6520 7573 6564  ethod to be used
-00006a00: 2c20 6465 6661 756c 7420 6964 656e 7469  , default identi
-00006a10: 6361 6c0a 2d62 6d2c 202d 2d62 775f 6d65  cal.-bm, --bw_me
-00006a20: 7468 6f64 2c20 7468 6520 6261 6e64 7769  thod, the bandwi
-00006a30: 6474 6820 6d65 7468 6f64 2074 6f20 6265  dth method to be
-00006a40: 2075 7365 6420 696e 2061 6e61 6c79 7a69   used in analyzi
-00006a50: 6e67 2074 6865 2070 6561 6b20 6f66 2057  ng the peak of W
-00006a60: 4744 2064 6174 6573 2c20 6465 6661 756c  GD dates, defaul
-00006a70: 7420 7369 6c76 6572 6d61 6e0a 2d2d 6e5f  t silverman.--n_
-00006a80: 6d65 646f 6964 732c 2074 6865 206e 756d  medoids, the num
-00006a90: 6265 7220 6f66 206d 6564 6f69 6473 2074  ber of medoids t
-00006aa0: 6f20 6669 742c 2064 6566 6175 6c74 2032  o fit, default 2
-00006ab0: 0a2d 6b6d 2c20 2d2d 6b64 656d 6574 686f  .-km, --kdemetho
-00006ac0: 642c 2074 6865 206b 6465 206d 6574 686f  d, the kde metho
-00006ad0: 6420 746f 2062 6520 7573 6564 2069 6e20  d to be used in 
-00006ae0: 616e 616c 797a 696e 6720 7468 6520 7065  analyzing the pe
-00006af0: 616b 206f 6620 5747 4420 6461 7465 732c  ak of WGD dates,
-00006b00: 206b 6d65 646f 6964 7320 616e 616c 7973   kmedoids analys
-00006b10: 6973 206f 7220 7468 6520 6261 7369 6320  is or the basic 
-00006b20: 4b73 2070 6c6f 7474 696e 672c 2064 6566  Ks plotting, def
-00006b30: 6175 6c74 2073 6369 7079 0a2d 2d6e 5f63  ault scipy.--n_c
-00006b40: 6c75 7374 6572 732c 2074 6865 206e 756d  lusters, the num
-00006b50: 6265 7220 6f66 2063 6c75 7374 6572 7320  ber of clusters 
-00006b60: 746f 2070 6c6f 7420 456c 626f 7720 6c6f  to plot Elbow lo
-00006b70: 7373 2066 756e 6374 696f 6e2c 2064 6566  ss function, def
-00006b80: 6175 6c74 2035 0a2d 6764 2c20 2d2d 6775  ault 5.-gd, --gu
-00006b90: 6964 652c 2074 6865 2072 6567 696d 6520  ide, the regime 
-00006ba0: 7265 7369 6469 6e67 2061 6e63 686f 7273  residing anchors
-00006bb0: 2c20 6465 6661 756c 7420 5365 676d 656e  , default Segmen
-00006bc0: 740a 2d70 7263 742c 202d 2d70 726f 6d69  t.-prct, --promi
-00006bd0: 6e65 6e63 655f 6375 746f 6666 2c20 7468  nence_cutoff, th
-00006be0: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
-00006bf0: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
-00006c00: 6520 7065 616b 7320 696e 2070 6561 6b20  e peaks in peak 
-00006c10: 6669 6e64 696e 6720 7374 6570 732c 2064  finding steps, d
-00006c20: 6566 6175 6c74 2030 2e31 0a2d 7268 2c20  efault 0.1.-rh, 
-00006c30: 2d2d 7265 6c5f 6865 6967 6874 2c20 7468  --rel_height, th
-00006c40: 6520 7265 6c61 7469 7665 2068 6569 6768  e relative heigh
-00006c50: 7420 6174 2077 6869 6368 2074 6865 2070  t at which the p
-00006c60: 6561 6b20 7769 6474 6820 6973 206d 6561  eak width is mea
-00006c70: 7375 7265 642c 2064 6566 6175 6c74 2030  sured, default 0
-00006c80: 2e34 0a2d 6b64 2c20 2d2d 6b73 746f 6461  .4.-kd, --kstoda
-00006c90: 7465 2c20 7468 6520 7261 6e67 6520 6f66  te, the range of
-00006ca0: 204b 7320 746f 2062 6520 6461 7465 6420   Ks to be dated 
-00006cb0: 696e 2068 6575 7269 7374 6963 2073 6561  in heuristic sea
-00006cc0: 7263 682c 2064 6566 6175 6c74 2028 302e  rch, default (0.
-00006cd0: 352c 2031 2e35 290a 2d78 6c2c 202d 2d78  5, 1.5).-xl, --x
-00006ce0: 6c69 6d2c 2074 6865 2078 2061 7869 7320  lim, the x axis 
-00006cf0: 6c69 6d69 7420 6f66 2047 4d4d 204b 7320  limit of GMM Ks 
-00006d00: 6469 7374 7269 6275 7469 6f6e 0a2d 796c  distribution.-yl
-00006d10: 2c20 2d2d 796c 696d 2c20 7468 6520 7920  , --ylim, the y 
-00006d20: 6178 6973 206c 696d 6974 206f 6620 474d  axis limit of GM
-00006d30: 4d20 4b73 2064 6973 7472 6962 7574 696f  M Ks distributio
-00006d40: 6e0a 2d2d 6d61 6e75 616c 7365 742c 2066  n.--manualset, f
-00006d50: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
-00006d60: 6865 7220 746f 206f 7574 7075 7420 616e  her to output an
-00006d70: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
-00006d80: 6d61 6e75 616c 6c79 2073 6574 204b 7320  manually set Ks 
-00006d90: 7261 6e67 652c 2069 6620 7468 6520 666c  range, if the fl
-00006da0: 6167 2077 6173 2073 6574 2c20 6d61 6e75  ag was set, manu
-00006db0: 616c 6c79 2073 6574 204b 7320 7261 6e67  ally set Ks rang
-00006dc0: 6520 776f 756c 6420 6265 206f 7574 7075  e would be outpu
-00006dd0: 7474 6564 0a2d 2d63 692c 2074 6865 2063  tted.--ci, the c
-00006de0: 6f6e 6669 6465 6e63 6520 6c65 7665 6c20  onfidence level 
-00006df0: 6f66 206c 6f67 2d6e 6f72 6d61 6c20 6469  of log-normal di
-00006e00: 7374 7269 6275 7469 6f6e 2074 6f20 6461  stribution to da
-00006e10: 7465 2c20 6465 6661 756c 7420 3935 0a2d  te, default 95.-
-00006e20: 2d68 6472 2c20 7468 6520 6869 6768 6573  -hdr, the highes
-00006e30: 7420 6465 6e73 6974 7920 7265 6769 6f6e  t density region
-00006e40: 2028 4844 5229 2061 7070 6c69 6564 2069   (HDR) applied i
-00006e50: 6e20 7468 6520 7365 676d 656e 742d 6775  n the segment-gu
-00006e60: 6964 6564 2061 6e63 686f 7220 7061 6972  ided anchor pair
-00006e70: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
-00006e80: 2c20 6465 6661 756c 7420 3935 0a2d 2d68  , default 95.--h
-00006e90: 6575 7269 7374 6963 2c20 666c 6167 206f  euristic, flag o
-00006ea0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-00006eb0: 6f20 696e 6974 6961 7465 2068 6575 7269  o initiate heuri
-00006ec0: 7374 6963 206d 6574 686f 6420 6f66 2064  stic method of d
-00006ed0: 6566 696e 696e 6720 4349 2066 6f72 2064  efining CI for d
-00006ee0: 6174 696e 672c 2069 6620 7468 6520 666c  ating, if the fl
-00006ef0: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-00006f00: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
-00006f10: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
-00006f20: 7465 640a 2d6b 632c 202d 2d6b 7363 7574  ted.-kc, --kscut
-00006f30: 6f66 662c 2074 6865 204b 7320 7361 7475  off, the Ks satu
-00006f40: 7261 7469 6f6e 2063 7574 6f66 6620 696e  ration cutoff in
-00006f50: 2064 6174 696e 672c 2064 6566 6175 6c74   dating, default
-00006f60: 2035 0a2d 2d6b 6565 7074 6d70 6669 672c   5.--keeptmpfig,
-00006f70: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00006f80: 6574 6865 7220 746f 206b 6565 7020 7465  ether to keep te
-00006f90: 6d70 6f72 6172 7920 6669 6775 7265 7320  mporary figures 
-00006fa0: 696e 2070 6561 6b20 6669 6e64 696e 6720  in peak finding 
-00006fb0: 7072 6f63 6573 732c 2069 6620 7468 6520  process, if the 
-00006fc0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00006fd0: 6f73 6520 6669 6775 7265 7320 776f 756c  ose figures woul
-00006fe0: 6420 6265 206b 6570 740a 6060 600a 0a54  d be kept.```..T
-00006ff0: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-00007000: 7379 6e60 2063 616e 2072 6561 6c69 7a65  syn` can realize
-00007010: 2074 6865 2069 6e74 7261 2d20 616e 6420   the intra- and 
-00007020: 696e 7465 722d 7370 6563 6966 6963 2073  inter-specific s
-00007030: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
-00007040: 2e0a 6060 600a 7767 6420 7379 6e20 6661  ..```.wgd syn fa
-00007050: 6d69 6c69 6573 2067 6666 7320 286f 7074  milies gffs (opt
-00007060: 696f 6e29 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d  ion).-----------
-00007070: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007080: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-00007090: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000070a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000070b0: 2d2d 2d2d 2d0a 2d6b 732c 202d 2d6b 735f  -----.-ks, --ks_
-000070c0: 6469 7374 7269 6275 7469 6f6e 2c20 6b73  distribution, ks
-000070d0: 2064 6973 7472 6962 7574 696f 6e20 6461   distribution da
-000070e0: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
-000070f0: 4e6f 6e65 0a2d 6f2c 202d 2d6f 7574 6469  None.-o, --outdi
-00007100: 722c 2074 6865 206f 7574 7075 7420 6469  r, the output di
-00007110: 7265 6374 6f72 792c 2064 6566 6175 6c74  rectory, default
-00007120: 2077 6764 5f73 796e 0a2d 662c 202d 2d66   wgd_syn.-f, --f
-00007130: 6561 7475 7265 2c20 7468 6520 6665 6174  eature, the feat
-00007140: 7572 6520 666f 7220 7061 7273 696e 6720  ure for parsing 
-00007150: 6765 6e65 2049 4473 2066 726f 6d20 4746  gene IDs from GF
-00007160: 4620 6669 6c65 732c 2064 6566 6175 6c74  F files, default
-00007170: 2067 656e 650a 2d61 2c20 2d2d 6174 7472   gene.-a, --attr
-00007180: 6962 7574 652c 2074 6865 2061 7474 7269  ibute, the attri
-00007190: 6275 7465 2066 6f72 2070 6172 7369 6e67  bute for parsing
-000071a0: 2074 6865 2067 656e 6520 4944 7320 6672   the gene IDs fr
-000071b0: 6f6d 2074 6865 2047 4646 2066 696c 6573  om the GFF files
-000071c0: 2c20 6465 6661 756c 7420 4944 0a2d 6174  , default ID.-at
-000071d0: 672c 202d 2d61 6464 6974 696f 6e61 6c67  g, --additionalg
-000071e0: 6666 696e 666f 2c20 7468 6520 6665 6174  ffinfo, the feat
-000071f0: 7572 6520 616e 6420 6174 7472 6962 7574  ure and attribut
-00007200: 6520 696e 666f 726d 6174 696f 6e20 6f66  e information of
-00007210: 2061 6464 6974 696f 6e61 6c20 6766 6633   additional gff3
-00007220: 2066 696c 6573 2069 6620 6469 6666 6572   files if differ
-00007230: 656e 7420 696e 2074 6865 2066 6f72 6d61  ent in the forma
-00007240: 7420 6f66 2028 6665 6174 7572 653b 6174  t of (feature;at
-00007250: 7472 6962 7574 6529 272c 2064 6566 6175  tribute)', defau
-00007260: 6c74 204e 6f6e 650a 2d6d 6c2c 202d 2d6d  lt None.-ml, --m
-00007270: 696e 6c65 6e2c 2074 6865 206d 696e 696d  inlen, the minim
-00007280: 756d 206c 656e 6774 6820 6f66 2061 2073  um length of a s
-00007290: 6361 6666 6f6c 6420 746f 2062 6520 696e  caffold to be in
-000072a0: 636c 7564 6564 2069 6e20 646f 7470 6c6f  cluded in dotplo
-000072b0: 742c 2064 6566 6175 6c74 202d 312c 2069  t, default -1, i
-000072c0: 6620 2d31 2077 6173 2073 6574 2c20 7468  f -1 was set, th
-000072d0: 6520 3130 2520 6f66 2074 6865 206c 6f6e  e 10% of the lon
-000072e0: 6765 7374 2073 6361 6666 6f6c 6420 776f  gest scaffold wo
-000072f0: 756c 6420 6265 2073 6574 0a2d 6d73 2c20  uld be set.-ms, 
-00007300: 2d2d 6d61 7873 697a 652c 2074 6865 206d  --maxsize, the m
-00007310: 6178 696d 756d 2066 616d 696c 7920 7369  aximum family si
-00007320: 7a65 2074 6f20 6265 2069 6e63 6c75 6465  ze to be include
-00007330: 642c 2064 6566 6175 6c74 2032 3030 0a2d  d, default 200.-
-00007340: 722c 202d 2d6b 735f 7261 6e67 652c 2074  r, --ks_range, t
-00007350: 6865 204b 7320 7261 6e67 6520 696e 2063  he Ks range in c
-00007360: 6f6c 6f72 6564 2064 6f74 706c 6f74 2c20  olored dotplot, 
-00007370: 6465 6661 756c 7420 2830 2c20 3529 0a2d  default (0, 5).-
-00007380: 2d69 6164 686f 7265 5f6f 7074 696f 6e73  -iadhore_options
-00007390: 2c20 7468 6520 7061 7261 6d65 7465 7220  , the parameter 
-000073a0: 7365 7474 696e 6720 696e 2069 6164 686f  setting in iadho
-000073b0: 7265 2c20 6465 6661 756c 7420 6173 2061  re, default as a
-000073c0: 2073 7472 696e 6720 6f66 206c 656e 6774   string of lengt
-000073d0: 6820 7a65 726f 0a2d 6d67 2c20 2d2d 6d69  h zero.-mg, --mi
-000073e0: 6e73 6567 6c65 6e2c 2074 6865 206d 696e  nseglen, the min
-000073f0: 696d 756d 206c 656e 6774 6820 6f66 2073  imum length of s
-00007400: 6567 6d65 6e74 7320 746f 2069 6e63 6c75  egments to inclu
-00007410: 6465 2069 6e20 7261 7469 6f20 6966 203c  de in ratio if <
-00007420: 3d20 312c 2064 6566 6175 6c74 2031 3030  = 1, default 100
-00007430: 3030 0a2d 6b72 2c20 2d2d 6b65 6570 7265  00.-kr, --keepre
-00007440: 6475 6e2c 2066 6c61 6720 6f70 7469 6f6e  dun, flag option
-00007450: 2c20 7768 6574 6865 7220 746f 206b 6565  , whether to kee
-00007460: 7020 7265 6475 6e64 616e 7420 6d75 6c74  p redundant mult
-00007470: 6970 6c69 636f 6e73 2c20 6966 2074 6865  iplicons, if the
-00007480: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007490: 6865 2072 6564 756e 6461 6e74 206d 756c  he redundant mul
-000074a0: 7469 706c 6963 6f6e 7320 776f 756c 6420  tiplicons would 
-000074b0: 6265 206b 6570 740a 2d6d 676e 2c20 2d2d  be kept.-mgn, --
-000074c0: 6d69 6e67 656e 656e 756d 2c20 7468 6520  mingenenum, the 
-000074d0: 6d69 6e69 6d75 6d20 6e75 6d62 6572 206f  minimum number o
-000074e0: 6620 6765 6e65 7320 666f 7220 6120 7365  f genes for a se
-000074f0: 676d 656e 7420 746f 2062 6520 636f 6e73  gment to be cons
-00007500: 6964 6572 6564 2c20 6465 6661 756c 7420  idered, default 
-00007510: 3330 0a2d 6473 2c20 2d2d 646f 7473 697a  30.-ds, --dotsiz
-00007520: 652c 2074 6865 2064 6f74 2073 697a 6520  e, the dot size 
-00007530: 696e 2064 6f74 2070 6c6f 742c 2064 6566  in dot plot, def
-00007540: 6175 6c74 2030 2e33 0a2d 6161 2c20 2d2d  ault 0.3.-aa, --
-00007550: 6170 616c 7068 612c 2074 6865 206f 7061  apalpha, the opa
-00007560: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
-00007570: 6f74 732c 2064 6566 6175 6c74 2031 0a2d  ots, default 1.-
-00007580: 6861 2c20 2d2d 686f 616c 7068 612c 2074  ha, --hoalpha, t
-00007590: 6865 206f 7061 6369 7479 206f 6620 686f  he opacity of ho
-000075a0: 6d6f 6c6f 6720 646f 7473 2c20 6465 6661  molog dots, defa
-000075b0: 756c 7420 300a 2d73 7274 2c20 2d2d 7368  ult 0.-srt, --sh
-000075c0: 6f77 7265 616c 7469 636b 2c20 666c 6167  owrealtick, flag
-000075d0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000075e0: 2074 6f20 7368 6f77 2074 6865 2072 6561   to show the rea
-000075f0: 6c20 7469 636b 2069 6e20 6765 6e65 7320  l tick in genes 
-00007600: 6f72 2062 6173 6573 2c20 6966 2074 6865  or bases, if the
-00007610: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007620: 6865 2072 6561 6c20 7469 636b 2077 6f75  he real tick wou
-00007630: 6c64 2062 6520 7368 6f77 6564 0a2d 746c  ld be showed.-tl
-00007640: 732c 202d 2d74 6963 6b6c 6162 656c 7369  s, --ticklabelsi
-00007650: 7a65 2c20 7468 6520 6c61 6265 6c20 7369  ze, the label si
-00007660: 7a65 206f 6620 7469 636b 2c20 6465 6661  ze of tick, defa
-00007670: 756c 7420 350a 2d67 722c 202d 2d67 6973  ult 5.-gr, --gis
-00007680: 7472 622c 2066 6c61 6720 6f70 7469 6f6e  trb, flag option
-00007690: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
-000076a0: 2067 6973 745f 7261 696e 626f 7720 6173   gist_rainbow as
-000076b0: 2063 6f6c 6f72 206d 6170 206f 6620 646f   color map of do
-000076c0: 7470 6c6f 740a 2d6e 2c20 2d2d 6e74 6872  tplot.-n, --nthr
-000076d0: 6561 6473 2c20 7468 6520 6e75 6d62 6572  eads, the number
-000076e0: 206f 6620 7468 7265 6164 7320 746f 2075   of threads to u
-000076f0: 7365 2069 6e20 7379 6e74 656e 7920 696e  se in synteny in
-00007700: 6665 7265 6e63 652c 2064 6566 6175 6c74  ference, default
-00007710: 2034 0a60 6060 0a0a 5468 6520 7072 6f67   4.```..The prog
-00007720: 7261 6d20 6077 6764 2076 697a 6020 6361  ram `wgd viz` ca
-00007730: 6e20 7265 616c 697a 6520 7468 6520 7669  n realize the vi
-00007740: 7375 616c 697a 6174 696f 6e20 6f66 202a  sualization of *
-00007750: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
-00007760: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
-00007770: 616e 6420 7379 6e74 656e 792e 0a60 6060  and synteny..```
-00007780: 0a77 6764 2076 697a 2028 6f70 7469 6f6e  .wgd viz (option
-00007790: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
-000077a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
-000077e0: 2d2d 0a2d 642c 202d 2d64 6174 6166 696c  --.-d, --datafil
-000077f0: 652c 2074 6865 204b 7320 6461 7461 6669  e, the Ks datafi
-00007800: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
-00007810: 0a2d 6f2c 202d 2d6f 7574 6469 722c 2074  .-o, --outdir, t
-00007820: 6865 206f 7574 7075 7420 6469 7265 6374  he output direct
-00007830: 6f72 792c 2064 6566 6175 6c74 2077 6764  ory, default wgd
-00007840: 5f76 697a 0a2d 7372 2c20 2d2d 7370 6169  _viz.-sr, --spai
-00007850: 722c 2074 6865 2073 7065 6369 6573 2070  r, the species p
-00007860: 6169 7220 746f 2062 6520 706c 6f74 7465  air to be plotte
-00007870: 642c 2064 6566 6175 6c74 204e 6f6e 652c  d, default None,
-00007880: 2074 6869 7320 6f70 7469 6f6e 2063 616e   this option can
-00007890: 2062 6520 7072 6f76 6964 6564 206d 756c   be provided mul
-000078a0: 7469 706c 6520 7469 6d65 730a 2d66 612c  tiple times.-fa,
-000078b0: 202d 2d66 6f63 7573 3261 6c6c 2c20 7365   --focus2all, se
-000078c0: 7420 666f 6361 6c20 7370 6563 6965 7320  t focal species 
-000078d0: 616e 6420 6c65 7420 7370 6563 6965 7320  and let species 
-000078e0: 7061 6972 2074 6f20 6265 2062 6574 7765  pair to be betwe
-000078f0: 656e 2066 6f63 616c 2061 6e64 2061 6c6c  en focal and all
-00007900: 2074 6865 2072 656d 6169 6e69 6e67 2073   the remaining s
-00007910: 7065 6369 6573 2c20 6465 6661 756c 7420  pecies, default 
-00007920: 4e6f 6e65 0a2d 6773 2c20 2d2d 6773 6d61  None.-gs, --gsma
-00007930: 702c 2074 6865 2067 656e 6520 6e61 6d65  p, the gene name
-00007940: 2d73 7065 6369 6573 206e 616d 6520 6d61  -species name ma
-00007950: 702c 2064 6566 6175 6c74 204e 6f6e 650a  p, default None.
-00007960: 2d73 702c 202d 2d73 7065 6369 6573 7472  -sp, --speciestr
-00007970: 6565 2c20 7468 6520 7370 6563 6965 7320  ee, the species 
-00007980: 7472 6565 2074 6f20 7065 7266 6f72 6d20  tree to perform 
-00007990: 7261 7465 2063 6f72 7265 6374 696f 6e2c  rate correction,
-000079a0: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
-000079b0: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
-000079c0: 2c20 7468 6520 7261 7465 2063 6f72 7265  , the rate corre
-000079d0: 6374 696f 6e20 616e 616c 7973 6973 2077  ction analysis w
-000079e0: 6f75 6c64 2062 6520 6361 6c6c 6564 206f  ould be called o
-000079f0: 6666 0a2d 706b 2c20 2d2d 706c 6f74 6b64  ff.-pk, --plotkd
-00007a00: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
-00007a10: 7768 6574 6865 7220 746f 2070 6c6f 7420  whether to plot 
-00007a20: 6b64 6520 6375 7276 6520 7570 6f6e 2068  kde curve upon h
-00007a30: 6973 746f 6772 616d 2c20 6966 2074 6865  istogram, if the
-00007a40: 2066 6c61 6720 7761 7320 7365 742c 206b   flag was set, k
-00007a50: 6465 2063 7572 7665 2077 6f75 6c64 2062  de curve would b
-00007a60: 6520 6164 6465 640a 2d72 772c 202d 2d72  e added.-rw, --r
-00007a70: 6577 6569 6768 742c 2066 6c61 6720 6f70  eweight, flag op
-00007a80: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
-00007a90: 2072 6563 616c 6375 6c61 7465 2074 6865   recalculate the
-00007aa0: 2077 6569 6768 7420 7065 7220 7370 6563   weight per spec
-00007ab0: 6965 7320 7061 6972 2c20 6966 2074 6865  ies pair, if the
-00007ac0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007ad0: 6865 2077 6569 6768 7420 776f 756c 6420  he weight would 
-00007ae0: 6265 2072 6563 616c 6375 6c61 7465 640a  be recalculated.
-00007af0: 2d6f 722c 202d 2d6f 6e6c 7972 6f6f 746f  -or, --onlyrooto
-00007b00: 7574 2c20 666c 6167 206f 7074 696f 6e2c  ut, flag option,
-00007b10: 2077 6865 7468 6572 2074 6f20 6f6e 6c79   whether to only
-00007b20: 2063 6f6e 6475 6374 2072 6174 6520 636f   conduct rate co
-00007b30: 7272 6563 7469 6f6e 2075 7369 6e67 2074  rrection using t
-00007b40: 6865 206f 7574 6772 6f75 7020 6174 2072  he outgroup at r
-00007b50: 6f6f 7420 6173 206f 7574 6772 6f75 702c  oot as outgroup,
-00007b60: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
-00007b70: 2073 6574 2c20 6f6e 6c79 2074 6865 206f   set, only the o
-00007b80: 7574 6772 6f75 7020 6174 2072 6f6f 7420  utgroup at root 
-00007b90: 776f 756c 6420 6265 2075 7365 6420 6173  would be used as
-00007ba0: 206f 7574 6772 6f75 700a 2d69 7465 722c   outgroup.-iter,
-00007bb0: 202d 2d65 6d5f 6974 6572 6174 696f 6e73   --em_iterations
-00007bc0: 2c20 7468 6520 6d61 7869 6d75 6d20 454d  , the maximum EM
-00007bd0: 2069 7465 7261 7469 6f6e 732c 2064 6566   iterations, def
-00007be0: 6175 6c74 2032 3030 0a2d 696e 6974 2c20  ault 200.-init, 
-00007bf0: 2d2d 656d 5f69 6e69 7469 616c 697a 6174  --em_initializat
-00007c00: 696f 6e73 2c20 7468 6520 6d61 7869 6d75  ions, the maximu
-00007c10: 6d20 454d 2069 6e69 7469 616c 697a 6174  m EM initializat
-00007c20: 696f 6e73 2c20 6465 6661 756c 7420 3230  ions, default 20
-00007c30: 300a 2d70 7263 742c 202d 2d70 726f 6d69  0.-prct, --promi
-00007c40: 6e65 6e63 655f 6375 746f 6666 2c20 7468  nence_cutoff, th
-00007c50: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
-00007c60: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
-00007c70: 6520 7065 616b 732c 2064 6566 6175 6c74  e peaks, default
-00007c80: 2030 2e31 0a2d 7268 2c20 2d2d 7265 6c5f   0.1.-rh, --rel_
-00007c90: 6865 6967 6874 2c20 7468 6520 7265 6c61  height, the rela
-00007ca0: 7469 7665 2068 6569 6768 7420 6174 2077  tive height at w
-00007cb0: 6869 6368 2074 6865 2070 6561 6b20 7769  hich the peak wi
-00007cc0: 6474 6820 6973 206d 6561 7375 7265 642c  dth is measured,
-00007cd0: 2064 6566 6175 6c74 2030 2e34 0a2d 736d   default 0.4.-sm
-00007ce0: 2c20 2d2d 7365 676d 656e 7473 2c20 7468  , --segments, th
-00007cf0: 6520 7365 676d 656e 7473 2064 6174 6166  e segments dataf
-00007d00: 696c 652c 2064 6566 6175 6c74 204e 6f6e  ile, default Non
-00007d10: 650a 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c  e.-ml, --minlen,
-00007d20: 2074 6865 206d 696e 696d 756d 206c 656e   the minimum len
-00007d30: 6774 6820 6f66 2061 2073 6361 6666 6f6c  gth of a scaffol
-00007d40: 6420 746f 2062 6520 696e 636c 7564 6564  d to be included
-00007d50: 2069 6e20 646f 7470 6c6f 742c 2064 6566   in dotplot, def
-00007d60: 6175 6c74 202d 312c 2069 6620 2d31 2077  ault -1, if -1 w
-00007d70: 6173 2073 6574 2c20 7468 6520 3130 2520  as set, the 10% 
-00007d80: 6f66 2074 6865 206c 6f6e 6765 7374 2073  of the longest s
-00007d90: 6361 6666 6f6c 6473 2077 696c 6c20 6265  caffolds will be
-00007da0: 2073 6574 0a2d 6d73 2c20 2d2d 6d61 7873   set.-ms, --maxs
-00007db0: 697a 652c 2074 6865 206d 6178 696d 756d  ize, the maximum
-00007dc0: 2066 616d 696c 7920 7369 7a65 2074 6f20   family size to 
-00007dd0: 6265 2069 6e63 6c75 6465 642c 2064 6566  be included, def
-00007de0: 6175 6c74 2032 3030 0a2d 6170 2c20 2d2d  ault 200.-ap, --
-00007df0: 616e 6368 6f72 706f 696e 7473 2c20 7468  anchorpoints, th
-00007e00: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-00007e10: 6461 7461 6669 6c65 2c20 6465 6661 756c  datafile, defaul
-00007e20: 7420 4e6f 6e65 0a2d 6d74 2c20 2d2d 6d75  t None.-mt, --mu
-00007e30: 6c74 6970 6c69 636f 6e2c 2074 6865 206d  ltiplicon, the m
-00007e40: 756c 7469 706c 6963 6f6e 7320 6461 7461  ultiplicons data
-00007e50: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
-00007e60: 6e65 0a2d 6774 2c20 2d2d 6765 6e65 7461  ne.-gt, --geneta
-00007e70: 626c 652c 2074 6865 2067 656e 6520 7461  ble, the gene ta
-00007e80: 626c 6520 6461 7461 6669 6c65 2c20 6465  ble datafile, de
-00007e90: 6661 756c 7420 4e6f 6e65 0a2d 6d67 2c20  fault None.-mg, 
-00007ea0: 2d2d 6d69 6e73 6567 6c65 6e2c 2074 6865  --minseglen, the
-00007eb0: 206d 696e 696d 756d 206c 656e 6774 6820   minimum length 
-00007ec0: 6f66 2073 6567 6d65 6e74 7320 746f 2069  of segments to i
-00007ed0: 6e63 6c75 6465 2c20 696e 2072 6174 696f  nclude, in ratio
-00007ee0: 2069 6620 3c3d 2031 2c20 6465 6661 756c   if <= 1, defaul
-00007ef0: 7420 3130 3030 300a 2d6d 676e 2c20 2d2d  t 10000.-mgn, --
-00007f00: 6d69 6e67 656e 656e 756d 2c20 7468 6520  mingenenum, the 
-00007f10: 6d69 6e69 6d75 6d20 6e75 6d62 6572 206f  minimum number o
-00007f20: 6620 6765 6e65 7320 666f 7220 6120 7365  f genes for a se
-00007f30: 676d 656e 7420 746f 2062 6520 636f 6e73  gment to be cons
-00007f40: 6964 6572 6564 2c20 6465 6661 756c 7420  idered, default 
-00007f50: 3330 0a2d 6b72 2c20 2d2d 6b65 6570 7265  30.-kr, --keepre
-00007f60: 6475 6e2c 2066 6c61 6720 6f70 7469 6f6e  dun, flag option
-00007f70: 2c20 7768 6574 6865 7220 746f 206b 6565  , whether to kee
-00007f80: 7020 7265 6475 6e64 616e 7420 6d75 6c74  p redundant mult
-00007f90: 6970 6c69 636f 6e73 2c20 6966 2074 6865  iplicons, if the
-00007fa0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-00007fb0: 6865 2072 6564 756e 6461 6e74 206d 756c  he redundant mul
-00007fc0: 7469 706c 6963 6f6e 7320 776f 756c 6420  tiplicons would 
-00007fd0: 6265 206b 6570 740a 2d65 706b 2c20 2d2d  be kept.-epk, --
-00007fe0: 6578 7472 6170 6172 616e 6f6d 656b 732c  extraparanomeks,
-00007ff0: 2065 7874 7261 2070 6172 616e 6f6d 6520   extra paranome 
-00008000: 4b73 2064 6174 6120 746f 2070 6c6f 7420  Ks data to plot 
-00008010: 696e 2074 6865 206d 6978 6564 204b 7320  in the mixed Ks 
-00008020: 6469 7374 7269 6275 7469 6f6e 2c20 6465  distribution, de
-00008030: 6661 756c 7420 4e6f 6e65 0a2d 7061 672c  fault None.-pag,
-00008040: 202d 2d70 6c6f 7461 7067 6d6d 2c20 666c   --plotapgmm, fl
-00008050: 6167 206f 7074 696f 6e2c 2077 6865 7468  ag option, wheth
-00008060: 6572 2074 6f20 636f 6e64 7563 7420 616e  er to conduct an
-00008070: 6420 706c 6f74 206d 6978 7475 7265 206d  d plot mixture m
-00008080: 6f64 656c 696e 6720 6f66 2061 6e63 686f  odeling of ancho
-00008090: 7220 4b73 2069 6e20 7468 6520 6d69 7865  r Ks in the mixe
-000080a0: 6420 4b73 2064 6973 7472 6962 7574 696f  d Ks distributio
-000080b0: 6e2c 2069 6620 7468 6520 666c 6167 2077  n, if the flag w
-000080c0: 6173 2073 6574 2c20 7468 6520 6d69 7874  as set, the mixt
-000080d0: 7572 6520 6d6f 6465 6c69 6e67 206f 6620  ure modeling of 
-000080e0: 616e 6368 6f72 204b 7320 776f 756c 6420  anchor Ks would 
-000080f0: 6265 2063 6f6e 6475 6374 6564 2061 6e64  be conducted and
-00008100: 2070 6c6f 7474 6564 0a2d 7065 6d2c 202d   plotted.-pem, -
-00008110: 2d70 6c6f 7465 6c6d 6d2c 2066 6c61 6720  -plotelmm, flag 
-00008120: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
-00008130: 746f 2063 6f6e 6475 6374 2061 6e64 2070  to conduct and p
-00008140: 6c6f 7420 656c 6d6d 206d 6978 7475 7265  lot elmm mixture
-00008150: 206d 6f64 656c 696e 6720 6f66 2070 6172   modeling of par
-00008160: 616e 6f6d 6520 4b73 2069 6e20 7468 6520  anome Ks in the 
-00008170: 6d69 7865 6420 4b73 2064 6973 7472 6962  mixed Ks distrib
-00008180: 7574 696f 6e2c 2069 6620 7468 6520 666c  ution, if the fl
-00008190: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
-000081a0: 656c 6d6d 206d 6978 7475 7265 206d 6f64  elmm mixture mod
-000081b0: 656c 696e 6720 6f66 2070 6172 616e 6f6d  eling of paranom
-000081c0: 6520 4b73 2077 6f75 6c64 2062 6520 636f  e Ks would be co
-000081d0: 6e64 7563 7465 6420 616e 6420 706c 6f74  nducted and plot
-000081e0: 7465 640a 2d63 2c20 2d2d 636f 6d70 6f6e  ted.-c, --compon
-000081f0: 656e 7473 2c20 7468 6520 7261 6e67 6520  ents, the range 
-00008200: 6f66 2074 6865 206e 756d 6265 7220 6f66  of the number of
-00008210: 2063 6f6d 706f 6e65 6e74 7320 746f 2066   components to f
-00008220: 6974 2069 6e20 616e 6368 6f72 204b 7320  it in anchor Ks 
-00008230: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
-00008240: 2c20 6465 6661 756c 7420 2831 2c34 290a  , default (1,4).
-00008250: 2d70 7379 2c20 2d2d 706c 6f74 7379 6e2c  -psy, --plotsyn,
-00008260: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
-00008270: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
-00008280: 6520 7468 6520 7379 6e74 656e 7920 706c  e the synteny pl
-00008290: 6f74 2c20 6f6e 6c79 2077 6865 6e20 7468  ot, only when th
-000082a0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-000082b0: 7468 6520 7379 6e74 656e 7920 706c 6f74  the synteny plot
-000082c0: 2077 6f75 6c64 2062 6520 7072 6f64 7563   would be produc
-000082d0: 6564 0a2d 6473 2c20 2d2d 646f 7473 697a  ed.-ds, --dotsiz
-000082e0: 652c 2074 6865 2064 6f74 2073 697a 6520  e, the dot size 
-000082f0: 696e 2064 6f74 2070 6c6f 742c 2064 6566  in dot plot, def
-00008300: 6175 6c74 2030 2e33 0a2d 6161 2c20 2d2d  ault 0.3.-aa, --
-00008310: 6170 616c 7068 612c 2074 6865 206f 7061  apalpha, the opa
-00008320: 6369 7479 206f 6620 616e 6368 6f72 2064  city of anchor d
-00008330: 6f74 732c 2064 6566 6175 6c74 2031 0a2d  ots, default 1.-
-00008340: 6861 2c20 2d2d 686f 616c 7068 612c 2074  ha, --hoalpha, t
-00008350: 6865 206f 7061 6369 7479 206f 6620 686f  he opacity of ho
-00008360: 6d6f 6c6f 6720 646f 7473 2c20 6465 6661  molog dots, defa
-00008370: 756c 7420 300a 2d73 7274 2c20 2d2d 7368  ult 0.-srt, --sh
-00008380: 6f77 7265 616c 7469 636b 2c20 666c 6167  owrealtick, flag
-00008390: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
-000083a0: 2074 6f20 7368 6f77 2074 6865 2072 6561   to show the rea
-000083b0: 6c20 7469 636b 2069 6e20 6765 6e65 7320  l tick in genes 
-000083c0: 6f72 2062 6173 6573 2c20 6966 2074 6865  or bases, if the
-000083d0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
-000083e0: 6865 2072 6561 6c20 7469 636b 2077 6f75  he real tick wou
-000083f0: 6c64 2062 6520 7368 6f77 6564 0a2d 746c  ld be showed.-tl
-00008400: 732c 202d 2d74 6963 6b6c 6162 656c 7369  s, --ticklabelsi
-00008410: 7a65 2c20 7468 6520 6c61 6265 6c20 7369  ze, the label si
-00008420: 7a65 206f 6620 7469 636b 2c20 6465 6661  ze of tick, defa
-00008430: 756c 7420 350a 2d78 6c2c 202d 2d78 6c69  ult 5.-xl, --xli
-00008440: 6d2c 2074 6865 2078 2061 7869 7320 6c69  m, the x axis li
-00008450: 6d69 7420 6f66 204b 7320 6469 7374 7269  mit of Ks distri
-00008460: 6275 7469 6f6e 0a2d 796c 2c20 2d2d 796c  bution.-yl, --yl
-00008470: 696d 2c20 7468 6520 7920 6178 6973 206c  im, the y axis l
-00008480: 696d 6974 206f 6620 4b73 2064 6973 7472  imit of Ks distr
-00008490: 6962 7574 696f 6e0a 2d61 646f 2c20 2d2d  ibution.-ado, --
-000084a0: 6164 6a75 7374 6f72 7468 6f2c 2066 6c61  adjustortho, fla
-000084b0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
-000084c0: 7220 746f 2061 646a 7573 7420 7468 6520  r to adjust the 
-000084d0: 6869 7374 6f67 7261 6d20 6865 6967 6874  histogram height
-000084e0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
-000084f0: 4b73 2061 7320 746f 206d 6174 6368 2074  Ks as to match t
-00008500: 6865 2068 6569 6768 7420 6f66 2070 6172  he height of par
-00008510: 616c 6f67 6f75 7320 4b73 2c20 6966 2074  alogous Ks, if t
-00008520: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
-00008530: 2074 6865 2061 646a 7573 746d 656e 7420   the adjustment 
-00008540: 776f 756c 6420 6265 2063 6f6e 6475 6374  would be conduct
-00008550: 6564 0a2d 6164 662c 202d 2d61 646a 7573  ed.-adf, --adjus
-00008560: 7466 6163 746f 722c 2074 6865 2061 646a  tfactor, the adj
-00008570: 7573 746d 656e 7420 6661 6374 6f72 206f  ustment factor o
-00008580: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
-00008590: 2c20 6465 6661 756c 7420 302e 350a 2d6f  , default 0.5.-o
-000085a0: 612c 202d 2d6f 6b61 6c70 6861 2c20 7468  a, --okalpha, th
-000085b0: 6520 6f70 6163 6974 7920 6f66 206f 7274  e opacity of ort
-000085c0: 686f 6c6f 676f 7573 204b 7320 6469 7374  hologous Ks dist
-000085d0: 7269 6275 7469 6f6e 2069 6e20 6d69 7865  ribution in mixe
-000085e0: 6420 706c 6f74 2c20 6465 6661 756c 7420  d plot, default 
-000085f0: 302e 350a 2d63 732c 202d 2d63 6c61 7373  0.5.-cs, --class
-00008600: 6963 2c20 666c 6167 206f 7074 696f 6e2c  ic, flag option,
-00008610: 2077 6865 7468 6572 2074 6f20 6472 6177   whether to draw
-00008620: 206d 6978 6564 204b 7320 706c 6f74 2069   mixed Ks plot i
-00008630: 6e20 6120 636c 6173 7369 6320 6d61 6e6e  n a classic mann
-00008640: 6572 2077 6865 7265 2074 6865 2066 756c  er where the ful
-00008650: 6c20 6f72 7468 6f6c 6f67 6f75 7320 4b73  l orthologous Ks
-00008660: 2064 6973 7472 6962 7574 696f 6e20 6973   distribution is
-00008670: 2064 7261 7765 642c 2069 6620 7468 6520   drawed, if the 
-00008680: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
-00008690: 6520 636c 6173 7369 6320 6d69 7865 6420  e classic mixed 
-000086a0: 4b73 2070 6c6f 7420 776f 756c 6420 6265  Ks plot would be
-000086b0: 2064 7261 776e 0a2d 7461 2c20 2d2d 746f   drawn.-ta, --to
-000086c0: 7061 7272 6f77 2c20 666c 6167 206f 7074  parrow, flag opt
-000086d0: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
-000086e0: 6164 6a75 7374 2074 6865 2061 7272 6f77  adjust the arrow
-000086f0: 2074 6f20 6265 2061 7420 7468 6520 746f   to be at the to
-00008700: 7020 6f66 2074 6865 2070 6c6f 742c 2069  p of the plot, i
-00008710: 6e73 7465 6164 206f 6620 6265 696e 6720  nstead of being 
-00008720: 636f 6f72 6469 6e61 7465 6420 6173 2074  coordinated as t
-00008730: 6865 204b 4445 206f 6620 7468 6520 6f72  he KDE of the or
-00008740: 7468 6f6c 6f67 6f75 7320 4b73 2064 6973  thologous Ks dis
-00008750: 7472 6962 7574 696f 6e2c 2069 6620 7468  tribution, if th
-00008760: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00008770: 7468 6520 6172 726f 7720 776f 756c 6420  the arrow would 
-00008780: 6265 2073 6574 2061 7420 7468 6520 746f  be set at the to
-00008790: 700a 2d6e 612c 202d 2d6e 6f64 6561 7665  p.-na, --nodeave
-000087a0: 7261 6765 642c 2066 6c61 6720 6f70 7469  raged, flag opti
-000087b0: 6f6e 2c20 7768 6574 6865 7220 746f 2075  on, whether to u
-000087c0: 7365 206e 6f64 652d 6176 6572 6167 6564  se node-averaged
-000087d0: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
-000087e0: 6564 756e 6461 6e63 792c 2069 6620 7468  edundancy, if th
-000087f0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
-00008800: 7468 6520 6e6f 6465 2d61 7665 7261 6765  the node-average
-00008810: 6420 6d65 7468 6f64 2077 6f75 6c64 2062  d method would b
-00008820: 6520 696e 6974 6961 7465 640a 2d62 732c  e initiated.-bs,
-00008830: 202d 2d62 6f6f 7473 7472 6170 2c20 7468   --bootstrap, th
-00008840: 6520 6e75 6d62 6572 206f 6620 626f 6f74  e number of boot
-00008850: 7374 7261 7020 7265 706c 6963 6174 6573  strap replicates
-00008860: 206f 6620 6f72 7468 6f6c 6f67 204b 7320   of ortholog Ks 
-00008870: 6469 7374 7269 6275 7469 6f6e 2069 6e20  distribution in 
-00008880: 6d69 7865 6420 706c 6f74 0a2d 6772 2c20  mixed plot.-gr, 
-00008890: 2d2d 6769 7374 7262 2c20 666c 6167 206f  --gistrb, flag o
-000088a0: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
-000088b0: 6f20 7573 6520 6769 7374 5f72 6169 6e62  o use gist_rainb
-000088c0: 6f77 2061 7320 636f 6c6f 7220 6d61 7020  ow as color map 
-000088d0: 6f66 2064 6f74 706c 6f74 0a2d 6e2c 202d  of dotplot.-n, -
-000088e0: 2d6e 7468 7265 6164 732c 2074 6865 206e  -nthreads, the n
-000088f0: 756d 6265 7220 6f66 2074 6872 6561 6473  umber of threads
-00008900: 2074 6f20 7573 6520 696e 2062 6f6f 7473   to use in boots
-00008910: 7472 6170 2073 616d 706c 696e 672c 2064  trap sampling, d
-00008920: 6566 6175 6c74 2031 0a60 6060 0a0a 2323  efault 1.```..##
-00008930: 2055 7361 6765 0a0a 4865 7265 2077 6520   Usage..Here we 
-00008940: 7072 6f76 6964 6564 2074 6865 2062 6173  provided the bas
-00008950: 6963 2075 7361 6765 2066 6f72 2065 6163  ic usage for eac
-00008960: 6820 7072 6f67 7261 6d20 616e 6420 7468  h program and th
-00008970: 6520 7265 6c65 7661 6e74 2070 6172 616d  e relevant param
-00008980: 6574 6572 7320 616e 6420 7375 6767 6573  eters and sugges
-00008990: 7469 6f6e 7320 6f6e 2070 6172 616d 6574  tions on paramet
-000089a0: 6572 697a 6174 696f 6e2e 2041 2072 656d  erization. A rem
-000089b0: 696e 6465 7220 7468 6174 2074 6865 2067  inder that the g
-000089c0: 6976 656e 2063 6f72 6573 2061 6e64 2074  iven cores and t
-000089d0: 6872 6561 6473 2063 616e 2073 6967 6e69  hreads can signi
-000089e0: 6669 6361 6e74 6c79 2069 6d70 6163 7420  ficantly impact 
-000089f0: 7468 6520 7275 6e20 7469 6d65 2061 6e64  the run time and
-00008a00: 2074 6875 7320 7765 2061 6464 6564 2073   thus we added s
-00008a10: 6f6d 6520 7265 706f 7274 2069 6e66 6f72  ome report infor
-00008a20: 6d61 7469 6f6e 2070 6572 7461 696e 696e  mation pertainin
-00008a30: 6720 746f 2074 6865 2073 7973 7465 6d20  g to the system 
-00008a40: 6f66 2075 7365 7273 2074 6f20 6661 6369  of users to faci
-00008a50: 6c69 7461 7465 2074 6865 2065 6666 6963  litate the effic
-00008a60: 6965 6e74 2073 6574 7469 6e67 206f 6620  ient setting of 
-00008a70: 7468 7265 6164 7320 616e 6420 6d65 6d6f  threads and memo
-00008a80: 7279 2e20 5468 6520 6c6f 6769 6361 6c20  ry. The logical 
-00008a90: 4350 5573 2072 6570 6f72 7465 6420 7265  CPUs reported re
-00008aa0: 7072 6573 656e 7473 2074 6865 206e 756d  presents the num
-00008ab0: 6265 7220 6f66 2070 6879 7369 6361 6c20  ber of physical 
-00008ac0: 636f 7265 7320 6d75 6c74 6970 6c69 6564  cores multiplied
-00008ad0: 2062 7920 7468 6520 6e75 6d62 6572 206f   by the number o
-00008ae0: 6620 7468 7265 6164 7320 7468 6174 2063  f threads that c
-00008af0: 616e 2072 756e 206f 6e20 6561 6368 2063  an run on each c
-00008b00: 6f72 652c 2061 6c73 6f20 6b6e 6f77 6e20  ore, also known 
-00008b10: 6173 2048 7970 6572 2054 6872 6561 6469  as Hyper Threadi
-00008b20: 6e67 2e20 5468 6520 6e75 6d62 6572 206f  ng. The number o
-00008b30: 6620 6c6f 6769 6361 6c20 4350 5573 206d  f logical CPUs m
-00008b40: 6179 206e 6f74 206e 6563 6573 7361 7269  ay not necessari
-00008b50: 6c79 2062 6520 6571 7569 7661 6c65 6e74  ly be equivalent
-00008b60: 2074 6f20 7468 6520 6163 7475 616c 206e   to the actual n
-00008b70: 756d 6265 7220 6f66 2043 5055 7320 7468  umber of CPUs th
-00008b80: 6520 6375 7272 656e 7420 7072 6f63 6573  e current proces
-00008b90: 7320 6361 6e20 7573 652e 2054 6865 2061  s can use. The a
-00008ba0: 7661 696c 6162 6c65 206d 656d 6f72 7920  vailable memory 
-00008bb0: 7265 6665 7273 2074 6f20 7468 6520 6d65  refers to the me
-00008bc0: 6d6f 7279 2074 6861 7420 6361 6e20 6265  mory that can be
-00008bd0: 2067 6976 656e 2069 6e73 7461 6e74 6c79   given instantly
-00008be0: 2074 6f20 7072 6f63 6573 7365 7320 7769   to processes wi
-00008bf0: 7468 6f75 7420 7468 6520 7379 7374 656d  thout the system
-00008c00: 2067 6f69 6e67 2069 6e74 6f20 7377 6170   going into swap
-00008c10: 2061 6e64 2072 6566 6c65 6374 7320 7468   and reflects th
-00008c20: 6520 6163 7475 616c 206d 656d 6f72 7920  e actual memory 
-00008c30: 6176 6169 6c61 626c 652e 2054 6865 2066  available. The f
-00008c40: 7265 6520 6d65 6d6f 7279 2072 6566 6572  ree memory refer
-00008c50: 7320 746f 2074 6865 206d 656d 6f72 7920  s to the memory 
-00008c60: 6e6f 7420 6265 696e 6720 7573 6564 2061  not being used a
-00008c70: 7420 616c 6c20 287a 6572 6f65 6429 2074  t all (zeroed) t
-00008c80: 6861 7420 6973 2072 6561 6469 6c79 2061  hat is readily a
-00008c90: 7661 696c 6162 6c65 2e20 5468 6520 6465  vailable. The de
-00008ca0: 7363 7269 7074 696f 6e20 6162 6f76 6520  scription above 
-00008cb0: 7265 6665 7273 2074 6f20 7468 6520 646f  refers to the do
-00008cc0: 6375 6d65 6e74 6174 696f 6e20 6f66 2060  cumentation of `
-00008cd0: 7073 7574 696c 602e 0a0a 2323 2320 7767  psutil`...### wg
-00008ce0: 6420 646d 640a 0a2a 2a54 6865 2064 656c  d dmd..**The del
-00008cf0: 696e 6561 7469 6f6e 206f 6620 7768 6f6c  ineation of whol
-00008d00: 6520 7061 7261 6e6f 6d65 2a2a 0a60 6060  e paranome**.```
-00008d10: 0a77 6764 2064 6d64 2041 7175 696c 6567  .wgd dmd Aquileg
-00008d20: 6961 5f63 6f65 7275 6c65 6120 2d49 2032  ia_coerulea -I 2
-00008d30: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-00008d40: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-00008d50: 2d74 6f5f 7374 6f70 2920 282d 2d63 6473  -to_stop) (--cds
-00008d60: 2920 282d 6e20 3429 2028 2d6f 2077 6764  ) (-n 4) (-o wgd
-00008d70: 5f64 6d64 2920 282d 7420 776f 726b 696e  _dmd) (-t workin
-00008d80: 675f 746d 7029 0a60 6060 200a 0a4e 6f74  g_tmp).``` ..Not
-00008d90: 6520 7468 6174 2077 6520 646f 6e27 7420  e that we don't 
-00008da0: 7072 6f76 6964 6520 7468 6520 6461 7461  provide the data
-00008db0: 206f 6620 7468 6973 2063 6f64 696e 6720   of this coding 
-00008dc0: 7365 7175 656e 6365 2028 6364 7329 2066  sequence (cds) f
-00008dd0: 696c 6520 6041 7175 696c 6567 6961 5f63  ile `Aquilegia_c
-00008de0: 6f65 7275 6c65 6160 2062 7574 2069 7420  oerulea` but it 
-00008df0: 6361 6e20 6265 2065 6173 696c 7920 646f  can be easily do
-00008e00: 776e 6c6f 6164 6564 2061 7420 5b50 6879  wnloaded at [Phy
-00008e10: 746f 7a6f 6d65 5d28 6874 7470 733a 2f2f  tozome](https://
-00008e20: 7068 7974 6f7a 6f6d 652d 6e65 7874 2e6a  phytozome-next.j
-00008e30: 6769 2e64 6f65 2e67 6f76 2f69 6e66 6f2f  gi.doe.gov/info/
-00008e40: 4163 6f65 7275 6c65 615f 7633 5f31 2920  Acoerulea_v3_1) 
-00008e50: 2873 616d 6520 666f 7220 6f74 6865 7220  (same for other 
-00008e60: 5573 6167 6520 646f 6329 2e20 4120 7265  Usage doc). A re
-00008e70: 6d69 6e64 6572 2074 6861 7420 696e 2074  minder that in t
-00008e80: 6865 2069 7373 7565 7320 736f 6d65 2075  he issues some u
-00008e90: 7365 7273 2064 6964 6e27 7420 646f 776e  sers didn't down
-00008ea0: 6c6f 6164 2074 6865 2060 4163 6f65 7275  load the `Acoeru
-00008eb0: 6c65 615f 3332 325f 7633 2e31 2e63 6473  lea_322_v3.1.cds
-00008ec0: 5f70 7269 6d61 7279 5472 616e 7363 7269  _primaryTranscri
-00008ed0: 7074 4f6e 6c79 2e66 612e 677a 6020 6275  ptOnly.fa.gz` bu
-00008ee0: 7420 696e 7374 6561 6420 7468 6520 6041  t instead the `A
-00008ef0: 636f 6572 756c 6561 5f33 3232 5f76 332e  coerulea_322_v3.
-00008f00: 312e 6364 732e 6661 2e67 7a60 2066 696c  1.cds.fa.gz` fil
-00008f10: 652e 2046 6f72 2074 6865 2063 6f6e 7374  e. For the const
-00008f20: 7275 6374 696f 6e20 6f66 2077 686f 6c65  ruction of whole
-00008f30: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
-00008f40: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
-00008f50: 6275 7469 6f6e 2c20 2a2a 6f6e 6c79 206f  bution, **only o
-00008f60: 6e65 2074 7261 6e73 6372 6970 7420 2874  ne transcript (t
-00008f70: 6865 2070 7269 6d61 7279 206f 6e65 2920  he primary one) 
-00008f80: 7065 7220 6765 6e65 2073 686f 756c 6420  per gene should 
-00008f90: 6265 2069 6e63 6c75 6465 642a 2a20 7375  be included** su
-00008fa0: 6368 2074 6861 7420 7468 6520 2a4b 2a3c  ch that the *K*<
-00008fb0: 7375 623e 533c 2f73 7562 3e20 6973 2072  sub>S</sub> is r
-00008fc0: 6561 6c6c 7920 696e 6469 6361 7469 6e67  eally indicating
-00008fd0: 2074 6865 2061 6765 206f 6620 6765 6e65   the age of gene
-00008fe0: 2064 7570 6c69 6361 7469 6f6e 2065 7665   duplication eve
-00008ff0: 6e74 2c20 696e 7374 6561 6420 6f66 2061  nt, instead of a
-00009000: 6c74 6572 6e61 7469 7665 2073 706c 6963  lternative splic
-00009010: 696e 672e 2054 7261 6e73 6372 6970 746f  ing. Transcripto
-00009020: 6d65 2064 6174 6120 7368 6f75 6c64 2062  me data should b
-00009030: 6520 6361 7265 6675 6c6c 7920 7472 6561  e carefully trea
-00009040: 7465 6420 7769 7468 2064 652d 7265 6475  ted with de-redu
-00009050: 6e64 616e 6379 2073 6f20 6173 2074 6f20  ndancy so as to 
-00009060: 7265 6475 6365 2074 6865 2066 616c 7365  reduce the false
-00009070: 2070 6f73 6974 6976 6520 6475 706c 6963   positive duplic
-00009080: 6174 696f 6e20 6275 6d70 2063 6175 7365  ation bump cause
-00009090: 6420 6279 2070 6572 7661 7369 7665 2061  d by pervasive a
-000090a0: 6c74 6572 6e61 7469 7665 2073 706c 6963  lternative splic
-000090b0: 696e 672e 2048 6572 6520 7468 6520 696e  ing. Here the in
-000090c0: 666c 6174 696f 6e20 6661 6374 6f72 2070  flation factor p
-000090d0: 6172 616d 6574 6572 2c20 6769 7665 6e20  arameter, given 
-000090e0: 6279 2060 2d49 6020 6f72 2060 2d2d 696e  by `-I` or `--in
-000090f0: 666c 6174 696f 6e60 2c20 6166 6665 6374  flation`, affect
-00009100: 7320 7468 6520 6772 616e 756c 6172 6974  s the granularit
-00009110: 7920 6f72 2072 6573 6f6c 7574 696f 6e20  y or resolution 
-00009120: 6f66 2074 6865 2063 6c75 7374 6572 696e  of the clusterin
-00009130: 6720 6f75 7463 6f6d 6520 616e 6420 696d  g outcome and im
-00009140: 706c 6963 6974 6c79 2063 6f6e 7472 6f6c  plicitly control
-00009150: 6c73 2074 6865 206e 756d 6265 7220 6f66  ls the number of
-00009160: 2063 6c75 7374 6572 732c 2077 6974 6820   clusters, with 
-00009170: 6c6f 7720 7661 6c75 6573 2073 7563 6820  low values such 
-00009180: 6173 2031 2e33 206f 7220 312e 3420 6c65  as 1.3 or 1.4 le
-00009190: 6164 696e 6720 746f 2066 6577 6572 2062  ading to fewer b
-000091a0: 7574 206c 6172 6765 7220 636c 7573 7465  ut larger cluste
-000091b0: 7273 2061 6e64 2068 6967 6820 7661 6c75  rs and high valu
-000091c0: 6573 2073 7563 6820 6173 2035 206f 7220  es such as 5 or 
-000091d0: 3620 6c65 6164 696e 6720 746f 206d 6f72  6 leading to mor
-000091e0: 6520 6275 7420 736d 616c 6c65 7220 636c  e but smaller cl
-000091f0: 7573 7465 7273 2e20 5765 2073 6574 2074  usters. We set t
-00009200: 6865 2064 6566 6175 6c74 2076 616c 7565  he default value
-00009210: 2061 7320 3220 6173 2073 7567 6765 7374   as 2 as suggest
-00009220: 6564 2062 7920 5b4d 434c 5d28 6874 7470  ed by [MCL](http
-00009230: 733a 2f2f 6d69 6361 6e73 2e6f 7267 2f6d  s://micans.org/m
-00009240: 636c 2f29 2e20 5468 6520 652d 7661 6c75  cl/). The e-valu
-00009250: 6520 6375 742d 6f66 6620 666f 7220 7365  e cut-off for se
-00009260: 7175 656e 6365 2073 696d 696c 6172 6974  quence similarit
-00009270: 792c 2067 6976 656e 2062 7920 602d 6560  y, given by `-e`
-00009280: 206f 7220 602d 2d65 7661 6c60 2c20 7768   or `--eval`, wh
-00009290: 6963 6820 6465 6e6f 7465 7320 7468 6520  ich denotes the 
-000092a0: 6578 7065 6374 6564 2076 616c 7565 206f  expected value o
-000092b0: 6620 7468 6520 6869 7420 7175 616e 7469  f the hit quanti
-000092c0: 6669 6573 2074 6865 206e 756d 6265 7220  fies the number 
-000092d0: 6f66 2061 6c69 676e 6d65 6e74 7320 6f66  of alignments of
-000092e0: 2073 696d 696c 6172 206f 7220 6265 7474   similar or bett
-000092f0: 6572 2071 7561 6c69 7479 2074 6861 7420  er quality that 
-00009300: 796f 7520 6578 7065 6374 2074 6f20 6669  you expect to fi
-00009310: 6e64 2073 6561 7263 6869 6e67 2074 6869  nd searching thi
-00009320: 7320 7175 6572 7920 6167 6169 6e73 7420  s query against 
-00009330: 6120 6461 7461 6261 7365 206f 6620 7261  a database of ra
-00009340: 6e64 6f6d 2073 6571 7565 6e63 6573 2074  ndom sequences t
-00009350: 6865 2073 616d 6520 7369 7a65 2061 7320  he same size as 
-00009360: 7468 6520 6163 7475 616c 2074 6172 6765  the actual targe
-00009370: 7420 6461 7461 6261 7365 2c20 6973 2074  t database, is t
-00009380: 6865 206b 6579 2070 6172 616d 6574 6572  he key parameter
-00009390: 206d 6561 7375 7269 6e67 2074 6865 2073   measuring the s
-000093a0: 6967 6e69 6669 6361 6e63 6520 6f66 2061  ignificance of a
-000093b0: 2068 6974 2c20 7768 6963 6820 6973 2073   hit, which is s
-000093c0: 6574 2068 6572 6520 6173 2064 6566 6175  et here as defau
-000093d0: 6c74 2031 652d 3130 2e20 4e6f 7465 2074  lt 1e-10. Note t
-000093e0: 6861 7420 5b44 4941 4d4f 4e44 5d28 6874  hat [DIAMOND](ht
-000093f0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-00009400: 2f62 6275 6368 6669 6e6b 2f64 6961 6d6f  /bbuchfink/diamo
-00009410: 6e64 2f77 696b 692f 312e 2d54 7574 6f72  nd/wiki/1.-Tutor
-00009420: 6961 6c29 2069 7473 656c 6620 6279 2064  ial) itself by d
-00009430: 6566 6175 6c74 206f 6e6c 7920 7265 706f  efault only repo
-00009440: 7274 7320 616c 6c20 616c 6967 6e6d 656e  rts all alignmen
-00009450: 7473 2077 6974 6820 652d 7661 6c75 6520  ts with e-value 
-00009460: 3c20 302e 3030 312e 2054 6865 2070 6572  < 0.001. The per
-00009470: 6365 6e74 6167 6520 6f66 2075 7070 6572  centage of upper
-00009480: 2068 6974 7320 7573 6564 2066 6f72 2067   hits used for g
-00009490: 656e 6520 6c65 6e67 7468 206e 6f72 6d61  ene length norma
-000094a0: 6c69 7a61 7469 6f6e 2c20 6769 7665 6e20  lization, given 
-000094b0: 6279 2060 2d6e 7060 206f 7220 602d 2d6e  by `-np` or `--n
-000094c0: 6f72 6d61 6c69 7a65 6470 6572 6365 6e74  ormalizedpercent
-000094d0: 602c 2077 6869 6368 2064 6574 6572 6d69  `, which determi
-000094e0: 6e65 7320 7468 6520 7570 7065 7220 7065  nes the upper pe
-000094f0: 7263 656e 7469 6c65 206f 6620 6869 7473  rcentile of hits
-00009500: 2070 6572 2062 696e 2028 6361 7465 676f   per bin (catego
-00009510: 7269 7a65 6420 6279 2067 656e 6520 6c65  rized by gene le
-00009520: 6e67 7468 2920 7573 6564 2069 6e20 7468  ngth) used in th
-00009530: 6520 6669 7420 6f66 206c 696e 6561 7220  e fit of linear 
-00009540: 7265 6772 6573 7369 6f6e 2c20 636f 6e73  regression, cons
-00009550: 6964 6572 696e 6720 7468 6174 206e 6f74  idering that not
-00009560: 2061 6c6c 2068 6974 7320 7065 7220 6269   all hits per bi
-00009570: 6e20 7368 6f77 2061 7070 6172 656e 7420  n show apparent 
-00009580: 6c69 6e65 6172 2072 656c 6174 696f 6e73  linear relations
-00009590: 6869 702c 2069 7320 7365 7420 6173 2064  hip, is set as d
-000095a0: 6566 6175 6c74 2035 2c20 696e 6469 6361  efault 5, indica
-000095b0: 7469 6e67 2074 6865 2075 7361 6765 206f  ting the usage o
-000095c0: 6620 746f 7020 3525 2068 6974 7320 7065  f top 5% hits pe
-000095d0: 7220 6269 6e2e 2054 6865 206e 756d 6265  r bin. The numbe
-000095e0: 7220 6f66 2062 696e 7320 6469 7669 6465  r of bins divide
-000095f0: 6420 696e 2067 656e 6520 6c65 6e67 7468  d in gene length
-00009600: 206e 6f72 6d61 6c69 7a61 7469 6f6e 2c20   normalization, 
-00009610: 6769 7665 6e20 6279 2060 2d62 7360 206f  given by `-bs` o
-00009620: 7220 602d 2d62 696e 7360 2c20 6465 7465  r `--bins`, dete
-00009630: 726d 696e 6573 2074 6865 206e 756d 6265  rmines the numbe
-00009640: 7220 6f66 2062 696e 7320 746f 2063 6174  r of bins to cat
-00009650: 6567 6f72 697a 6520 7468 6520 6765 6e65  egorize the gene
-00009660: 206c 656e 6774 682c 2069 7320 7365 7420   length, is set 
-00009670: 6173 2064 6566 6175 6c74 2031 3030 2e20  as default 100. 
-00009680: 5468 6520 7061 7261 6d65 7465 7220 602d  The parameter `-
-00009690: 6e6e 6020 6f72 2060 2d2d 6e6f 6e6f 726d  nn` or `--nonorm
-000096a0: 616c 697a 6174 696f 6e60 2063 616e 2062  alization` can b
-000096b0: 6520 7365 7420 746f 2063 616c 6c20 6f66  e set to call of
-000096c0: 6620 7468 6520 6e6f 726d 616c 697a 6174  f the normalizat
-000096d0: 696f 6e20 7072 6f63 6573 732c 2061 6c74  ion process, alt
-000096e0: 686f 7567 6820 6974 2773 2073 7567 6765  hough it's sugge
-000096f0: 7374 6564 2074 6f20 636f 6e64 7563 7420  sted to conduct 
-00009700: 7468 6520 6e6f 726d 616c 697a 6174 696f  the normalizatio
-00009710: 6e20 746f 2061 6371 7569 7265 206d 6f72  n to acquire mor
-00009720: 6520 6163 6375 7261 7465 2067 656e 6520  e accurate gene 
-00009730: 6661 6d69 6c79 2063 6c75 7374 6572 696e  family clusterin
-00009740: 6720 7265 7375 6c74 2e20 5468 6520 7061  g result. The pa
-00009750: 7261 6d65 7465 7273 2060 2d2d 746f 5f73  rameters `--to_s
-00009760: 746f 7060 2061 6e64 2060 2d2d 6364 7360  top` and `--cds`
-00009770: 2063 6f6e 7472 6f6c 2074 6865 2062 6568   control the beh
-00009780: 6176 696f 7572 206f 6620 7472 616e 736c  aviour of transl
-00009790: 6174 696e 6720 636f 6469 6e67 2073 6571  ating coding seq
-000097a0: 7565 6e63 6520 696e 746f 2061 6d69 6e6f  uence into amino
-000097b0: 2061 6369 6420 7365 7175 656e 6365 2e20   acid sequence. 
-000097c0: 4966 2074 6865 2060 2d2d 746f 5f73 746f  If the `--to_sto
-000097d0: 7060 2077 6173 2073 6574 2c20 7468 6520  p` was set, the 
-000097e0: 7472 616e 736c 6174 696f 6e20 776f 756c  translation woul
-000097f0: 6420 6265 2074 6572 6d69 6e61 7465 6420  d be terminated 
-00009800: 6174 2074 6865 2066 6972 7374 2069 6e2d  at the first in-
-00009810: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
-00009820: 2c20 6f74 6865 7277 6973 6520 7468 6520  , otherwise the 
-00009830: 7472 616e 736c 6174 696f 6e20 776f 756c  translation woul
-00009840: 6420 7369 6d70 6c79 2073 6b69 7020 616e  d simply skip an
-00009850: 7920 7374 6f70 2063 6f64 6f6e 732e 2049  y stop codons. I
-00009860: 6620 7468 6520 602d 2d63 6473 6020 7761  f the `--cds` wa
-00009870: 7320 7365 742c 2073 6571 7565 6e63 6573  s set, sequences
-00009880: 2074 6861 7420 646f 6573 6e27 7420 7374   that doesn't st
-00009890: 6172 7420 7769 7468 2061 2076 616c 6964  art with a valid
-000098a0: 2073 7461 7274 2063 6f64 6f6e 2c20 6f72   start codon, or
-000098b0: 2063 6f6e 7461 696e 7320 6d6f 7265 2074   contains more t
-000098c0: 6861 6e20 6f6e 6520 696e 2d66 7261 6d65  han one in-frame
-000098d0: 2073 746f 7020 636f 646f 6e2c 206f 7220   stop codon, or 
-000098e0: 6973 206e 6f74 2064 6976 6964 6162 6c65  is not dividable
-000098f0: 2062 7920 332c 2077 6f75 6c64 2062 6520   by 3, would be 
-00009900: 7369 6d70 6c79 2064 726f 7070 6564 2c20  simply dropped, 
-00009910: 7375 6368 2074 6861 7420 6f6e 6c79 2073  such that only s
-00009920: 7472 6963 7420 636f 6d70 6c65 7465 2063  trict complete c
-00009930: 6f64 696e 6720 7365 7175 656e 6365 7320  oding sequences 
-00009940: 776f 756c 6420 6265 2069 6e63 6c75 6465  would be include
-00009950: 6420 696e 2074 6865 2073 7562 7365 7175  d in the subsequ
-00009960: 656e 7420 616e 616c 7973 6973 2e20 5468  ent analysis. Th
-00009970: 6520 6578 6163 7420 6265 6861 7669 6f75  e exact behaviou
-00009980: 7220 6f66 2060 2d2d 746f 5f73 746f 7060  r of `--to_stop`
-00009990: 2061 6e64 2060 2d2d 6364 7360 2069 7320   and `--cds` is 
-000099a0: 6465 6669 6e65 6420 616e 6420 6465 7363  defined and desc
-000099b0: 7269 6265 6420 696e 2074 6865 205b 6269  ribed in the [bi
-000099c0: 6f70 7974 686f 6e5d 2868 7474 7073 3a2f  opython](https:/
-000099d0: 2f62 696f 7079 7468 6f6e 2e6f 7267 2f64  /biopython.org/d
-000099e0: 6f63 732f 312e 3735 2f61 7069 2f42 696f  ocs/1.75/api/Bio
-000099f0: 2e53 6571 2e68 746d 6c23 4269 6f2e 5365  .Seq.html#Bio.Se
-00009a00: 712e 5365 712e 7472 616e 736c 6174 6529  q.Seq.translate)
-00009a10: 206c 6962 7261 7279 2e20 5468 6520 6e75   library. The nu
-00009a20: 6d62 6572 206f 6620 7061 7261 6c6c 656c  mber of parallel
-00009a30: 2074 6872 6561 6473 2062 7920 7468 6520   threads by the 
-00009a40: 6f70 7469 6f6e 2060 2d6e 6020 6f72 2060  option `-n` or `
-00009a50: 2d2d 6e74 6872 6561 6473 6020 6361 6e20  --nthreads` can 
-00009a60: 6265 2073 6574 2074 6f20 6163 6365 6c65  be set to accele
-00009a70: 7261 7465 2074 6865 2063 616c 6375 6c61  rate the calcula
-00009a80: 7469 6f6e 2077 6974 6869 6e20 6064 6961  tion within `dia
-00009a90: 6d6f 6e64 602e 2054 6865 2064 6972 6563  mond`. The direc
-00009aa0: 746f 7279 206f 6620 6f75 7470 7574 206f  tory of output o
-00009ab0: 7220 696e 7465 726d 6564 6961 7465 2066  r intermediate f
-00009ac0: 696c 6573 2069 7320 6465 7465 726d 696e  iles is determin
-00009ad0: 6564 2062 7920 7468 6520 7061 7261 6d65  ed by the parame
-00009ae0: 7465 7220 602d 6f60 206f 7220 602d 2d6f  ter `-o` or `--o
-00009af0: 7574 6469 7260 2c20 616e 6420 602d 7460  utdir`, and `-t`
-00009b00: 206f 7220 602d 2d74 6d70 6469 7260 2c20   or `--tmpdir`, 
-00009b10: 7768 6963 6820 7769 6c6c 2062 6520 6372  which will be cr
-00009b20: 6561 7465 6420 6279 2074 6865 2070 726f  eated by the pro
-00009b30: 6772 616d 2069 7473 656c 6620 616e 6420  gram itself and 
-00009b40: 6265 206f 7665 7277 7269 7474 656e 2069  be overwritten i
-00009b50: 6620 7468 6520 666f 6c64 6572 2068 6173  f the folder has
-00009b60: 2061 6c72 6561 6479 2062 6565 6e20 6372   already been cr
-00009b70: 6561 7465 642e 204e 6f74 6520 7468 6174  eated. Note that
-00009b80: 2074 6865 2073 6f66 7477 6172 6520 6064   the software `d
-00009b90: 6961 6d6f 6e64 6020 7368 6f75 6c64 2062  iamond` should b
-00009ba0: 6520 7072 652d 696e 7374 616c 6c65 6420  e pre-installed 
-00009bb0: 616e 6420 7365 7420 696e 2074 6865 2065  and set in the e
-00009bc0: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
-00009bd0: 696e 2061 6c6c 2074 6865 2061 6e61 6c79  in all the analy
-00009be0: 7369 7320 7065 7266 6f72 6d65 6420 6279  sis performed by
-00009bf0: 2060 7767 6420 646d 6460 2065 7863 6570   `wgd dmd` excep
-00009c00: 7420 666f 7220 7468 6520 636f 6c6c 696e  t for the collin
-00009c10: 6561 7220 636f 616c 6573 6365 6e63 6520  ear coalescence 
-00009c20: 616e 616c 7973 6973 2e0a 0a2a 2a57 6520  analysis...**We 
-00009c30: 7375 6767 6573 7420 7468 6174 2074 6865  suggest that the
-00009c40: 2064 6566 6175 6c74 2073 6574 7469 6e67   default setting
-00009c50: 2069 6e20 7768 6963 6820 7468 6520 696e   in which the in
-00009c60: 666c 6174 696f 6e20 6661 6374 6f72 2069  flation factor i
-00009c70: 7320 7365 7420 6173 2032 2c20 652d 7661  s set as 2, e-va
-00009c80: 6c75 6520 6375 742d 6f66 6620 6173 2031  lue cut-off as 1
-00009c90: 652d 3130 2061 6e64 206f 7468 6572 2070  e-10 and other p
-00009ca0: 6172 616d 6574 6572 7320 696e 2064 6566  arameters in def
-00009cb0: 6175 6c74 2069 7320 6120 676f 6f64 2073  ault is a good s
-00009cc0: 7461 7274 696e 6720 706f 696e 742c 2075  tarting point, u
-00009cd0: 6e6c 6573 7320 796f 7520 7370 6563 6966  nless you specif
-00009ce0: 6963 616c 6c79 2077 616e 7420 746f 2065  ically want to e
-00009cf0: 7870 6c6f 7265 2074 6865 2065 6666 6563  xplore the effec
-00009d00: 7473 206f 6620 6469 6666 6572 656e 7420  ts of different 
-00009d10: 7061 7261 6d65 7465 7273 2e2a 2a20 5375  parameters.** Su
-00009d20: 6368 2074 6861 7420 7468 6520 636f 6d6d  ch that the comm
-00009d30: 616e 6420 666f 7220 7468 6520 6465 6c69  and for the deli
-00009d40: 6e65 6174 696f 6e20 6f66 2077 686f 6c65  neation of whole
-00009d50: 2070 6172 616e 6f6d 6520 6973 2073 696d   paranome is sim
-00009d60: 706c 7920 6173 2062 656c 6f77 2e0a 0a60  ply as below...`
-00009d70: 6060 0a77 6764 2064 6d64 2041 7175 696c  ``.wgd dmd Aquil
-00009d80: 6567 6961 5f63 6f65 7275 6c65 610a 6060  egia_coerulea.``
-00009d90: 600a 0a2a 2a54 6865 2064 656c 696e 6561  `..**The delinea
-00009da0: 7469 6f6e 206f 6620 5242 4873 2a2a 0a60  tion of RBHs**.`
-00009db0: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
-00009dc0: 6e63 6531 2073 6571 7565 6e63 6532 202d  nce1 sequence2 -
-00009dd0: 6520 3165 2d31 3020 2d62 7320 3130 3020  e 1e-10 -bs 100 
-00009de0: 2d6e 7020 3520 282d 6e6e 2920 282d 6e20  -np 5 (-nn) (-n 
-00009df0: 3429 2028 2d63 2030 2e39 2920 282d 2d6f  4) (-c 0.9) (--o
-00009e00: 6766 6f72 6d61 7429 2028 2d2d 746f 5f73  gformat) (--to_s
-00009e10: 746f 7029 2028 2d2d 6364 7329 2028 2d6f  top) (--cds) (-o
-00009e20: 2077 6764 5f64 6d64 2920 282d 7420 776f   wgd_dmd) (-t wo
-00009e30: 726b 696e 675f 746d 7029 0a60 6060 0a0a  rking_tmp).```..
-00009e40: 546f 2064 656c 696e 6561 7465 2052 4248  To delineate RBH
-00009e50: 7320 6265 7477 6565 6e20 7477 6f20 6364  s between two cd
-00009e60: 7320 7365 7175 656e 6365 2066 696c 6573  s sequence files
-00009e70: 2c20 7468 6520 7265 6c65 7661 6e74 2070  , the relevant p
-00009e80: 6172 616d 6574 6572 2069 7320 6d6f 7374  arameter is most
-00009e90: 6c79 2074 6865 2073 616d 6520 6173 2077  ly the same as w
-00009ea0: 686f 6c65 2070 6172 616e 6f6d 6520 696e  hole paranome in
-00009eb0: 6665 7265 6e63 652c 2065 7863 6570 7420  ference, except 
-00009ec0: 666f 7220 7468 6520 7061 7261 6d65 7465  for the paramete
-00009ed0: 7220 602d 6360 206f 7220 602d 2d63 7363  r `-c` or `--csc
-00009ee0: 6f72 6560 2c20 7768 6963 6820 7261 6e67  ore`, which rang
-00009ef0: 6573 2062 6574 7765 656e 2030 2061 6e64  es between 0 and
-00009f00: 2031 2061 6e64 2069 7320 7573 6564 2074   1 and is used t
-00009f10: 6f20 7265 6c61 7820 7468 6520 7369 6d69  o relax the simi
-00009f20: 6c61 7269 7479 2063 7574 6f66 6620 6672  larity cutoff fr
-00009f30: 6f6d 2074 6865 2065 7863 6c75 7369 7665  om the exclusive
-00009f40: 2072 6563 6970 726f 6361 6c20 6265 7374   reciprocal best
-00009f50: 2068 6974 7320 746f 2061 2063 6572 7461   hits to a certa
-00009f60: 696e 2072 6174 696f 2061 7320 746f 2074  in ratio as to t
-00009f70: 6865 2062 6573 7420 6869 7473 2e20 466f  he best hits. Fo
-00009f80: 7220 696e 7374 616e 6365 2c20 6966 2074  r instance, if t
-00009f90: 6865 2067 656e 6520 6231 2066 726f 6d20  he gene b1 from 
-00009fa0: 6765 6e6f 6d65 2042 2068 6173 2074 6865  genome B has the
-00009fb0: 2062 6573 7420 6869 7420 6765 6e65 2061   best hit gene a
-00009fc0: 3120 6672 6f6d 2067 656e 6f6d 6520 4120  1 from genome A 
-00009fd0: 7769 7468 2074 6865 2062 6974 2073 636f  with the bit sco
-00009fe0: 7265 2061 7320 3130 302c 2077 6869 6368  re as 100, which
-00009ff0: 2069 7320 6120 7363 6f72 696e 6720 6d61   is a scoring ma
-0000a000: 7472 6978 2069 6e64 6570 656e 6465 6e74  trix independent
-0000a010: 206d 6561 7375 7265 206f 6620 7468 6520   measure of the 
-0000a020: 286c 6f63 616c 2920 7369 6d69 6c61 7269  (local) similari
-0000a030: 7479 206f 6620 7468 6520 7477 6f20 616c  ty of the two al
-0000a040: 6967 6e65 6420 7365 7175 656e 6365 732c  igned sequences,
-0000a050: 2077 6974 6820 6c61 7267 6572 2076 616c   with larger val
-0000a060: 7565 7320 696e 6469 6361 7469 6e67 2068  ues indicating h
-0000a070: 6967 6865 7220 7369 6d69 6c61 7269 7469  igher similariti
-0000a080: 6573 2c20 6769 7665 6e20 7468 6520 602d  es, given the `-
-0000a090: 6320 302e 3960 2c20 6765 6e65 7320 6672  c 0.9`, genes fr
-0000a0a0: 6f6d 2067 656e 6f6d 6520 4120 7768 6963  om genome A whic
-0000a0b0: 6820 6861 7320 7468 6520 6269 7420 7363  h has the bit sc
-0000a0c0: 6f72 6520 7769 7468 2067 656e 6520 6231  ore with gene b1
-0000a0d0: 2068 6967 6865 7220 7468 616e 2030 2e39   higher than 0.9
-0000a0e0: 7831 3030 2077 696c 6c20 616c 736f 2062  x100 will also b
-0000a0f0: 6520 7772 6974 7465 6e20 696e 2074 6865  e written in the
-0000a100: 2072 6573 756c 7420 6669 6c65 2c20 7768   result file, wh
-0000a110: 6963 6820 696e 2061 2073 656e 7365 2061  ich in a sense a
-0000a120: 7265 206e 6f74 2052 4248 7320 616e 796d  re not RBHs anym
-0000a130: 6f72 6520 6f66 2063 6f75 7273 652c 2062  ore of course, b
-0000a140: 7574 2074 6865 2068 6967 686c 7920 7369  ut the highly si
-0000a150: 6d69 6c61 7220 686f 6d6f 6c6f 6775 6520  milar homologue 
-0000a160: 7061 6972 732e 2049 6620 6d6f 7265 2074  pairs. If more t
-0000a170: 6861 6e20 3220 7365 7175 656e 6365 2066  han 2 sequence f
-0000a180: 696c 6573 2077 6572 6520 7072 6f76 6964  iles were provid
-0000a190: 6564 2c20 6576 6572 7920 7061 6972 2d77  ed, every pair-w
-0000a1a0: 6973 6520 5242 4873 2077 6f75 6c64 2062  ise RBHs would b
-0000a1b0: 6520 6361 6c63 756c 6174 6564 2065 7863  e calculated exc
-0000a1c0: 6570 7420 666f 7220 7175 6572 7969 6e67  ept for querying
-0000a1d0: 2074 6865 2073 616d 6520 7365 7175 656e   the same sequen
-0000a1e0: 6365 2069 7473 656c 662e 2054 6865 206e  ce itself. The n
-0000a1f0: 756d 6265 7220 6f66 2070 6172 616c 6c65  umber of paralle
-0000a200: 6c20 7468 7265 6164 7320 746f 2062 6f6f  l threads to boo
-0000a210: 7374 6572 2074 6865 2072 756e 6e69 6e67  ster the running
-0000a220: 2073 7065 6564 2063 616e 2062 6520 7365   speed can be se
-0000a230: 7420 6279 2074 6865 206f 7074 696f 6e20  t by the option 
-0000a240: 602d 6e60 206f 7220 602d 2d6e 7468 7265  `-n` or `--nthre
-0000a250: 6164 7360 2077 6869 6368 2069 7320 7375  ads` which is su
-0000a260: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
-0000a270: 7420 6173 2028 4e2d 3129 4e2f 3220 7768  t as (N-1)N/2 wh
-0000a280: 6572 6520 4e20 6973 2074 6865 206e 756d  ere N is the num
-0000a290: 6265 7220 6f66 2063 6473 2066 696c 6573  ber of cds files
-0000a2a0: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-0000a2b0: 6869 6768 6573 7420 6566 6669 6369 656e  highest efficien
-0000a2c0: 6379 2e20 5468 6520 6f70 7469 6f6e 2060  cy. The option `
-0000a2d0: 2d2d 6f67 666f 726d 6174 6020 6361 6e20  --ogformat` can 
-0000a2e0: 6265 2073 6574 2074 6f20 6164 6420 696e  be set to add in
-0000a2f0: 6465 7820 2866 6f72 2069 6e73 7461 6e63  dex (for instanc
-0000a300: 6520 4746 3030 3030 3030 3031 2920 746f  e GF00000001) to
-0000a310: 2074 6865 206f 7574 7075 7420 5242 4820   the output RBH 
-0000a320: 6765 6e65 2066 616d 696c 6965 7320 7768  gene families wh
-0000a330: 6963 6820 6361 6e20 6265 2066 7572 7468  ich can be furth
-0000a340: 6572 2075 7365 6420 696e 2074 6865 202a  er used in the *
-0000a350: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2063  K*<sub>S</sub> c
-0000a360: 616c 6375 6c61 7469 6f6e 2062 7920 6077  alculation by `w
-0000a370: 6764 206b 7364 602e 0a0a 2a2a 5468 6520  gd ksd`...**The 
-0000a380: 7375 6767 6573 7465 6420 636f 6d6d 616e  suggested comman
-0000a390: 6420 746f 2073 7461 7274 2077 6974 6820  d to start with 
-0000a3a0: 6973 2061 6c73 6f20 756e 6465 7220 7468  is also under th
-0000a3b0: 6520 6465 6661 756c 7420 7365 7474 696e  e default settin
-0000a3c0: 6720 7769 7468 2074 6865 2063 6f6d 6d61  g with the comma
-0000a3d0: 6e64 2073 686f 776e 2062 656c 6f77 2a2a  nd shown below**
-0000a3e0: 0a0a 6060 600a 7767 6420 646d 6420 7365  ..```.wgd dmd se
-0000a3f0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
-0000a400: 320a 6060 600a 0a2a 2a54 6865 2064 656c  2.```..**The del
-0000a410: 696e 6561 7469 6f6e 206f 6620 6c6f 6361  ineation of loca
-0000a420: 6c20 4d52 4248 732a 2a0a 6060 600a 7767  l MRBHs**.```.wg
-0000a430: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
-0000a440: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000a450: 6365 3320 2d66 2073 6571 7565 6e63 6531  ce3 -f sequence1
-0000a460: 202d 6520 3165 2d31 3020 2d62 7320 3130   -e 1e-10 -bs 10
-0000a470: 3020 2d6e 7020 3520 282d 6e6e 2920 282d  0 -np 5 (-nn) (-
-0000a480: 6e20 3429 2028 2d6b 6629 2028 2d6b 6429  n 4) (-kf) (-kd)
-0000a490: 2028 2d63 2030 2e39 2920 282d 2d74 6f5f   (-c 0.9) (--to_
-0000a4a0: 7374 6f70 2920 282d 2d63 6473 2920 282d  stop) (--cds) (-
-0000a4b0: 6f20 7767 645f 646d 6429 2028 2d74 2077  o wgd_dmd) (-t w
-0000a4c0: 6f72 6b69 6e67 5f74 6d70 290a 6060 600a  orking_tmp).```.
-0000a4d0: 0a2a 2a54 6865 2064 6973 7469 6e63 7469  .**The distincti
-0000a4e0: 6f6e 2062 6574 7765 656e 206c 6f63 616c  on between local
-0000a4f0: 2061 6e64 2067 6c6f 6261 6c20 2868 6572   and global (her
-0000a500: 6575 6e64 6572 2920 4d52 4248 7320 6973  eunder) MRBHs is
-0000a510: 2074 6861 7420 6c6f 6361 6c20 4d52 4248   that local MRBH
-0000a520: 7320 6172 6520 7468 6520 7265 7375 6c74  s are the result
-0000a530: 7320 6f66 206d 6572 6765 6420 5242 4873  s of merged RBHs
-0000a540: 206f 6e20 6120 6a6f 696e 7420 666f 6361   on a joint foca
-0000a550: 6c20 7370 6563 6965 732c 2066 6f72 2069  l species, for i
-0000a560: 6e73 7461 6e63 6520 696e 2061 2074 6872  nstance in a thr
-0000a570: 6565 2073 7065 6369 6573 2073 7973 7465  ee species syste
-0000a580: 6d20 2841 2c28 422c 4329 292c 2074 6865  m (A,(B,C)), the
-0000a590: 206c 6f63 616c 204d 5242 4873 206f 6620   local MRBHs of 
-0000a5a0: 4120 6f6e 6c79 2072 6571 7569 7265 2074  A only require t
-0000a5b0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
-0000a5c0: 6620 5242 4873 2062 6574 7765 656e 2041  f RBHs between A
-0000a5d0: 2061 6e64 2043 2028 6465 6e6f 7465 6420   and C (denoted 
-0000a5e0: 6173 2041 4329 2061 6e64 2041 4220 616e  as AC) and AB an
-0000a5f0: 6420 7468 656e 2074 6865 206d 6572 6769  d then the mergi
-0000a600: 6e67 206f 6620 4142 2061 6e64 2041 4320  ng of AB and AC 
-0000a610: 6174 2074 6865 2061 7869 7320 6f66 2041  at the axis of A
-0000a620: 2c20 7768 696c 6520 7468 6520 676c 6f61  , while the gloa
-0000a630: 626c 204d 5242 4873 2061 7265 2069 6e64  bl MRBHs are ind
-0000a640: 6570 656e 6465 6e74 206f 6620 666f 6361  ependent of foca
-0000a650: 6c20 7370 6563 6965 7320 696e 2074 6861  l species in tha
-0000a660: 7420 6974 206a 7573 7420 6361 6c6c 7320  t it just calls 
-0000a670: 7468 6520 6361 6c63 756c 6174 696f 6e73  the calculations
-0000a680: 206f 6620 616c 6c20 706f 7373 6962 6c65   of all possible
-0000a690: 2073 7065 6369 6573 2070 6169 7220 286e   species pair (n
-0000a6a0: 6f74 2073 656c 6620 746f 2073 656c 6620  ot self to self 
-0000a6b0: 7370 6563 6965 7320 7061 6972 292c 2073  species pair), s
-0000a6c0: 7563 6820 7468 6174 2041 422c 2041 432c  uch that AB, AC,
-0000a6d0: 2061 6e64 2042 4320 6172 6520 616c 6c20   and BC are all 
-0000a6e0: 746f 2062 6520 6361 6c63 756c 6174 6564  to be calculated
-0000a6f0: 2061 6e64 206d 6572 6765 642e 2a2a 0a0a   and merged.**..
-0000a700: 5477 6f20 7479 7065 7320 6f66 204d 5242  Two types of MRB
-0000a710: 4873 2061 7320 696e 7465 7072 6574 6174  Hs as intepretat
-0000a720: 6564 2061 626f 7665 2063 616e 2062 6520  ed above can be 
-0000a730: 6465 6c69 6e65 6174 6564 2062 7920 6077  delineated by `w
-0000a740: 6764 2064 6d64 602c 2074 6865 206c 6f63  gd dmd`, the loc
-0000a750: 616c 204d 5242 4873 2061 6e64 2074 6865  al MRBHs and the
-0000a760: 2067 6c6f 6261 6c20 4d52 4248 732e 2054   global MRBHs. T
-0000a770: 6865 206c 6f63 616c 204d 5242 4873 2061  he local MRBHs a
-0000a780: 7265 2063 6f6e 7374 7275 6374 6564 2062  re constructed b
-0000a790: 7920 6d65 7267 696e 6720 616c 6c20 7468  y merging all th
-0000a7a0: 6520 7265 6c65 7661 6e74 2052 4248 7320  e relevant RBHs 
-0000a7b0: 6f6e 6c79 2077 6974 6820 7468 6520 666f  only with the fo
-0000a7c0: 6361 6c20 7370 6563 6965 732c 2077 6869  cal species, whi
-0000a7d0: 6368 2069 7320 7365 7420 6279 2060 2d66  ch is set by `-f
-0000a7e0: 6020 6f72 2060 2d2d 666f 6375 7360 2e20  ` or `--focus`. 
-0000a7f0: 5468 6520 7061 7261 6d65 7465 7220 602d  The parameter `-
-0000a800: 6b66 6020 6f72 2060 2d2d 6b65 6570 6661  kf` or `--keepfa
-0000a810: 7374 6160 2063 616e 2062 6520 7365 7420  sta` can be set 
-0000a820: 746f 2072 6574 6169 6e20 7468 6520 7365  to retain the se
-0000a830: 7175 656e 6365 2069 6e66 6f72 6d61 7469  quence informati
-0000a840: 6f6e 206f 6620 6561 6368 204d 5242 482e  on of each MRBH.
-0000a850: 2054 6865 2070 6172 616d 6574 6572 2060   The parameter `
-0000a860: 2d6b 6460 206f 7220 602d 2d6b 6565 7064  -kd` or `--keepd
-0000a870: 7570 6c69 6361 7465 7360 2064 6574 6572  uplicates` deter
-0000a880: 6d69 6e65 7320 7768 6574 6865 7220 7468  mines whether th
-0000a890: 6520 7361 6d65 2067 656e 6573 2063 616e  e same genes can
-0000a8a0: 2061 7070 6561 7220 696e 2064 6966 6665   appear in diffe
-0000a8b0: 7265 6e74 206c 6f63 616c 204d 5242 4873  rent local MRBHs
-0000a8c0: 2e20 4e6f 726d 616c 6c79 2074 6865 7265  . Normally there
-0000a8d0: 2077 696c 6c20 6265 206e 6f20 6475 706c   will be no dupl
-0000a8e0: 6963 6174 6573 2069 6e20 7468 6520 6c6f  icates in the lo
-0000a8f0: 6361 6c20 4d52 4248 7320 6275 7420 6966  cal MRBHs but if
-0000a900: 2075 7365 7273 2073 6574 2074 6865 2060   users set the `
-0000a910: 2d63 6020 6173 2030 2e39 2028 6f72 2061  -c` as 0.9 (or a
-0000a920: 6e79 2076 616c 7565 2073 6d61 6c6c 6572  ny value smaller
-0000a930: 2074 6861 6e20 3129 2c20 6974 2773 206c   than 1), it's l
-0000a940: 696b 656c 7920 7468 6174 2074 6865 2073  ikely that the s
-0000a950: 616d 6520 6765 6e65 2077 696c 6c20 6861  ame gene will ha
-0000a960: 7665 2063 6861 6e63 6520 746f 2061 7070  ve chance to app
-0000a970: 6561 7220 6d75 6c74 6970 6c65 2074 696d  ear multiple tim
-0000a980: 6573 2069 6e20 6469 6666 6572 656e 7420  es in different 
-0000a990: 6c6f 6361 6c20 4d52 4248 732e 2054 6861  local MRBHs. Tha
-0000a9a0: 7420 6973 2074 6f20 7361 792c 2074 6865  t is to say, the
-0000a9b0: 2070 6172 616d 6574 6572 2060 2d6b 6460   parameter `-kd`
-0000a9c0: 2069 7320 6d65 616e 696e 6766 756c 206f   is meaningful o
-0000a9d0: 6e6c 7920 7768 656e 2069 7427 7320 7365  nly when it's se
-0000a9e0: 7420 746f 6765 7468 6572 2077 6974 6820  t together with 
-0000a9f0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
-0000aa00: 6360 2e20 5468 6520 6e75 6d62 6572 206f  c`. The number o
-0000aa10: 6620 7061 7261 6c6c 656c 2074 6872 6561  f parallel threa
-0000aa20: 6473 2069 7320 7375 6767 6573 7465 6420  ds is suggested 
-0000aa30: 746f 2062 6520 7365 7420 6173 2074 6865  to be set as the
-0000aa40: 206e 756d 6265 7220 6f66 2063 6473 2066   number of cds f
-0000aa50: 696c 6573 206d 696e 7573 2031 2e0a 0a2a  iles minus 1...*
-0000aa60: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
-0000aa70: 7274 696e 6720 7275 6e20 6973 2075 6e64  rting run is und
-0000aa80: 6572 2074 6865 2064 6566 6175 6c74 2070  er the default p
-0000aa90: 6172 616d 6574 6572 2077 6974 6820 7468  arameter with th
-0000aaa0: 6520 636f 6d6d 616e 6420 7368 6f77 6e20  e command shown 
-0000aab0: 6173 2062 656c 6f77 2e2a 2a0a 0a60 6060  as below.**..```
-0000aac0: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000aad0: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000aae0: 7565 6e63 6533 202d 6620 7365 7175 656e  uence3 -f sequen
-0000aaf0: 6365 310a 6060 600a 0a2a 2a54 6865 2064  ce1.```..**The d
-0000ab00: 656c 696e 6561 7469 6f6e 206f 6620 676c  elineation of gl
-0000ab10: 6f62 616c 204d 5242 4873 2a2a 0a60 6060  obal MRBHs**.```
-0000ab20: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
-0000ab30: 6531 2073 6571 7565 6e63 6532 2073 6571  e1 sequence2 seq
-0000ab40: 7565 6e63 6533 202d 676d 202d 6520 3165  uence3 -gm -e 1e
-0000ab50: 2d31 3020 2d62 7320 3130 3020 2d6e 7020  -10 -bs 100 -np 
-0000ab60: 3520 282d 6e6e 2920 282d 6e20 3429 2028  5 (-nn) (-n 4) (
-0000ab70: 2d6b 6629 2028 2d6b 6429 2028 2d63 2030  -kf) (-kd) (-c 0
-0000ab80: 2e39 2920 282d 2d74 6f5f 7374 6f70 2920  .9) (--to_stop) 
-0000ab90: 282d 2d63 6473 2920 282d 6f20 7767 645f  (--cds) (-o wgd_
-0000aba0: 646d 6429 2028 2d74 2077 6f72 6b69 6e67  dmd) (-t working
-0000abb0: 5f74 6d70 290a 6060 600a 0a54 6865 2067  _tmp).```..The g
-0000abc0: 6c6f 6261 6c20 4d52 4248 7320 6973 2063  lobal MRBHs is c
-0000abd0: 6f6e 7374 7275 6374 6564 2062 7920 6578  onstructed by ex
-0000abe0: 6861 7573 7469 7665 6c79 206d 6572 6769  haustively mergi
-0000abf0: 6e67 2061 6c6c 2074 6865 2070 6f73 7369  ng all the possi
-0000ac00: 626c 6520 7061 6972 2d77 6973 6520 5242  ble pair-wise RB
-0000ac10: 4873 2065 7863 6570 7420 666f 7220 7175  Hs except for qu
-0000ac20: 6572 7969 6e67 2074 6865 2073 6571 7565  erying the seque
-0000ac30: 6e63 6520 6974 7365 6c66 2c20 7768 6963  nce itself, whic
-0000ac40: 6820 6361 6e20 6265 2069 6e69 7469 6174  h can be initiat
-0000ac50: 6564 2062 7920 6164 6420 7468 6520 666c  ed by add the fl
-0000ac60: 6167 2060 2d67 6d60 206f 7220 602d 2d67  ag `-gm` or `--g
-0000ac70: 6c6f 6261 6c6d 7262 6860 2e20 5468 6520  lobalmrbh`. The 
-0000ac80: 7265 7374 206f 6620 7265 6c65 7661 6e74  rest of relevant
-0000ac90: 2070 6172 616d 6574 6572 7320 7374 6179   parameters stay
-0000aca0: 7320 7468 6520 7361 6d65 2061 7320 7468  s the same as th
-0000acb0: 6520 6c6f 6361 6c20 4d52 4248 732e 2054  e local MRBHs. T
-0000acc0: 6865 206e 756d 6265 7220 6f66 2070 6172  he number of par
-0000acd0: 616c 6c65 6c20 7468 7265 6164 7320 6973  allel threads is
-0000ace0: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000acf0: 2073 6574 2061 7320 284e 2d31 294e 2f32   set as (N-1)N/2
-0000ad00: 2074 6f6f 2077 6865 7265 204e 2069 7320   too where N is 
-0000ad10: 7468 6520 6e75 6d62 6572 206f 6620 6364  the number of cd
-0000ad20: 7320 6669 6c65 7320 746f 2061 6368 6965  s files to achie
-0000ad30: 7665 2074 6865 2068 6967 6865 7374 2065  ve the highest e
-0000ad40: 6666 6963 6965 6e63 792e 0a0a 2a2a 4120  fficiency...**A 
-0000ad50: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
-0000ad60: 6e67 2072 756e 2069 7320 756e 6465 7220  ng run is under 
-0000ad70: 7468 6520 6465 6661 756c 7420 7061 7261  the default para
-0000ad80: 6d65 7465 7220 7769 7468 2074 6865 2063  meter with the c
-0000ad90: 6f6d 6d61 6e64 2073 686f 776e 2061 7320  ommand shown as 
-0000ada0: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
-0000adb0: 6420 646d 6420 7365 7175 656e 6365 3120  d dmd sequence1 
-0000adc0: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
-0000add0: 6365 3320 2d67 6d0a 6060 600a 0a2a 2a54  ce3 -gm.```..**T
-0000ade0: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
-0000adf0: 6620 6f72 7468 6f67 726f 7570 732a 2a0a  f orthogroups**.
-0000ae00: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
-0000ae10: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000ae20: 7365 7175 656e 6365 3320 2d6f 6920 2d6f  sequence3 -oi -o
-0000ae30: 6f20 2d65 2031 652d 3130 202d 6273 2031  o -e 1e-10 -bs 1
-0000ae40: 3030 202d 6e70 2035 2028 2d6e 6e29 2028  00 -np 5 (-nn) (
-0000ae50: 2d63 6329 2028 2d74 6529 2028 2d6d 6320  -cc) (-te) (-mc 
-0000ae60: 302e 3829 2028 2d67 6e29 2028 2d74 7265  0.8) (-gn) (-tre
-0000ae70: 6520 2766 6173 7474 7265 6527 2920 282d  e 'fasttree') (-
-0000ae80: 7473 2027 2d66 6173 7465 7374 2729 2028  ts '-fastest') (
-0000ae90: 2d6e 2034 2920 282d 2d74 6f5f 7374 6f70  -n 4) (--to_stop
-0000aea0: 2920 282d 2d63 6473 2920 282d 6f20 7767  ) (--cds) (-o wg
-0000aeb0: 645f 646d 6429 2028 2d74 2077 6f72 6b69  d_dmd) (-t worki
-0000aec0: 6e67 5f74 6d70 290a 6060 600a 0a49 6e20  ng_tmp).```..In 
-0000aed0: 6077 6764 2076 3260 2c20 7765 2061 6c73  `wgd v2`, we als
-0000aee0: 6f20 696d 706c 656d 656e 7465 6420 616e  o implemented an
-0000aef0: 2061 6c67 6f72 6974 686d 206f 6620 6465   algorithm of de
-0000af00: 6c69 6e65 6174 696e 6720 6f72 7468 6f67  lineating orthog
-0000af10: 726f 7570 732c 2077 6869 6368 2063 616e  roups, which can
-0000af20: 2062 6520 696e 6974 6961 7465 6420 7769   be initiated wi
-0000af30: 7468 2074 6865 2070 6172 616d 6574 6572  th the parameter
-0000af40: 2060 2d6f 6960 206f 7220 602d 2d6f 7274   `-oi` or `--ort
-0000af50: 686f 696e 6665 7260 2e20 5477 6f20 7761  hoinfer`. Two wa
-0000af60: 7973 206f 6620 6465 6c69 6e65 6174 696f  ys of delineatio
-0000af70: 6e20 6361 6e20 6265 2063 686f 7365 6e2c  n can be chosen,
-0000af80: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
-0000af90: 6f6e 2077 6179 2028 7365 7420 6279 2074  on way (set by t
-0000afa0: 6865 2070 6172 616d 6574 6572 2060 2d63  he parameter `-c
-0000afb0: 6360 206f 7220 602d 2d63 6f6e 6361 7460  c` or `--concat`
-0000afc0: 2920 6f72 2074 6865 206e 6f6e 2d63 6f6e  ) or the non-con
-0000afd0: 6361 7465 6e61 7469 6f6e 2028 6465 6661  catenation (defa
-0000afe0: 756c 7429 2077 6179 2e20 496e 2062 7269  ult) way. In bri
-0000aff0: 6566 2c20 7468 6520 636f 6e63 6174 656e  ef, the concaten
-0000b000: 6174 696f 6e20 7761 7920 6f66 2064 656c  ation way of del
-0000b010: 696e 6561 7469 6e67 206f 7274 686f 6772  ineating orthogr
-0000b020: 6f75 7073 2073 7461 7274 7320 7769 7468  oups starts with
-0000b030: 2063 6f6e 6361 7465 6e61 7469 6e67 2061   concatenating a
-0000b040: 6c6c 2074 6865 2073 6571 7565 6e63 6573  ll the sequences
-0000b050: 2069 6e74 6f20 6120 7369 6e67 6c65 2073   into a single s
-0000b060: 6571 7565 6e63 6520 6669 6c65 2061 6e64  equence file and
-0000b070: 2074 6865 6e20 696e 6665 7272 696e 6720   then inferring 
-0000b080: 7468 6520 7768 6f6c 6520 7061 7261 6e6f  the whole parano
-0000b090: 6d65 206f 6620 7468 6973 2073 696e 676c  me of this singl
-0000b0a0: 6520 7365 7175 656e 6365 2066 696c 6520  e sequence file 
-0000b0b0: 7769 7468 2074 6865 2063 6c75 7374 6572  with the cluster
-0000b0c0: 696e 6720 7265 7375 6c74 7320 6d61 7070  ing results mapp
-0000b0d0: 6564 2062 6163 6b20 746f 2074 6865 2062  ed back to the b
-0000b0e0: 656c 6f6e 6769 6e67 2073 7065 6369 6573  elonging species
-0000b0f0: 2e20 5768 696c 6520 7468 6520 6e6f 6e2d  . While the non-
-0000b100: 636f 6e63 6174 656e 6174 696f 6e20 7761  concatenation wa
-0000b110: 7920 7374 6172 7473 2077 6974 6820 7265  y starts with re
-0000b120: 7370 6563 7469 7665 2070 6169 722d 7769  spective pair-wi
-0000b130: 7365 2064 6961 6d6f 6e64 2073 6561 7263  se diamond searc
-0000b140: 6820 2869 6e63 6c75 6469 6e67 2071 7565  h (including que
-0000b150: 7279 696e 6720 7468 6520 7361 6d65 2073  rying the same s
-0000b160: 6571 7565 6e63 6520 6974 7365 6c66 2920  equence itself) 
-0000b170: 616e 6420 7468 656e 2061 6c6c 2074 6865  and then all the
-0000b180: 2073 6571 7565 6e63 6520 7369 6d69 6c61   sequence simila
-0000b190: 7269 7479 2074 6162 6c65 7320 7769 6c6c  rity tables will
-0000b1a0: 2062 6520 636f 6e63 6174 656e 6174 6564   be concatenated
-0000b1b0: 2061 6e64 2063 6c75 7374 6572 6564 2069   and clustered i
-0000b1c0: 6e74 6f20 6f72 7468 6f67 726f 7570 732e  nto orthogroups.
-0000b1d0: 2053 6f6d 6520 6f74 6865 7220 706f 7373   Some other poss
-0000b1e0: 6962 6c79 2075 7365 6675 6c20 706f 7374  ibly useful post
-0000b1f0: 2d63 6c75 7374 6572 696e 6720 6675 6e63  -clustering func
-0000b200: 7469 6f6e 7320 6361 6e20 6265 2069 6e69  tions can be ini
-0000b210: 7469 6174 6564 2c20 696e 636c 7564 696e  tiated, includin
-0000b220: 6720 7468 6520 7061 7261 6d65 7465 7220  g the parameter 
-0000b230: 602d 7465 6020 6f72 2060 2d2d 7465 7374  `-te` or `--test
-0000b240: 736f 6760 2c20 7768 6963 6820 6361 6e20  sog`, which can 
-0000b250: 6265 2073 6574 2074 6f20 7374 6172 7420  be set to start 
-0000b260: 7468 6520 756e 6269 6173 6564 2074 6573  the unbiased tes
-0000b270: 7420 6f66 2073 696e 676c 652d 636f 7079  t of single-copy
-0000b280: 2067 656e 6520 6661 6d69 6c69 6573 2028   gene families (
-0000b290: 6e6f 7465 2074 6861 7420 7468 6973 2066  note that this f
-0000b2a0: 756e 6374 696f 6e20 6e65 6564 7320 6068  unction needs `h
-0000b2b0: 6d6d 6572 6020 2876 332e 3162 3229 2074  mmer` (v3.1b2) t
-0000b2c0: 6f20 6265 2069 6e73 7461 6c6c 6564 2069  o be installed i
-0000b2d0: 6e20 7468 6520 656e 7669 726f 6e6d 656e  n the environmen
-0000b2e0: 7420 7061 7468 292c 2074 6865 2070 6172  t path), the par
-0000b2f0: 616d 6574 6572 2060 2d6d 6360 206f 7220  ameter `-mc` or 
-0000b300: 602d 2d6d 736f 6763 7574 602c 2072 616e  `--msogcut`, ran
-0000b310: 6769 6e67 2062 6574 7765 656e 2030 2074  ging between 0 t
-0000b320: 6f20 312c 2077 6869 6368 2063 616e 2062  o 1, which can b
-0000b330: 6520 7365 7420 746f 2073 6561 7263 6820  e set to search 
-0000b340: 7468 6520 736f 2d63 616c 6c65 6420 6d6f  the so-called mo
-0000b350: 7374 6c79 2073 696e 676c 652d 636f 7079  stly single-copy
-0000b360: 2066 616d 696c 7920 7768 6963 6820 6861   family which ha
-0000b370: 7320 6869 6768 6572 2074 6861 6e20 6365  s higher than ce
-0000b380: 7274 6169 6e20 6375 742d 6f66 6620 7065  rtain cut-off pe
-0000b390: 7263 656e 7461 6765 206f 6620 7370 6563  rcentage of spec
-0000b3a0: 6965 7320 636f 7665 7261 6765 2c20 7468  ies coverage, th
-0000b3b0: 6520 7061 7261 6d65 7465 7220 602d 676e  e parameter `-gn
-0000b3c0: 6020 6f72 2060 2d2d 6765 746e 736f 6760  ` or `--getnsog`
-0000b3d0: 2c20 7768 6963 6820 6361 6e20 6265 2073  , which can be s
-0000b3e0: 6574 2074 6f20 7365 6172 6368 2066 6f72  et to search for
-0000b3f0: 206e 6573 7465 6420 7369 6e67 6c65 2d63   nested single-c
-0000b400: 6f70 7920 6765 6e65 2066 616d 696c 6965  opy gene familie
-0000b410: 7320 284e 534f 4773 2920 7768 6963 6820  s (NSOGs) which 
-0000b420: 6973 206f 7269 6769 6e61 6c6c 7920 6d75  is originally mu
-0000b430: 6c74 6979 2d63 6f70 7920 6275 7420 6861  ltiy-copy but ha
-0000b440: 7320 6120 286d 6f73 746c 7929 2073 696e  s a (mostly) sin
-0000b450: 676c 652d 636f 7079 2062 7261 6e63 6820  gle-copy branch 
-0000b460: 2877 6869 6368 2072 6571 7569 7265 7320  (which requires 
-0000b470: 7468 6520 6368 6f73 656e 2074 7265 652d  the chosen tree-
-0000b480: 696e 6665 7265 6e63 6520 7072 6f67 7261  inference progra
-0000b490: 6d20 7365 7420 6279 2060 2d74 7265 6560  m set by `-tree`
-0000b4a0: 206f 7220 602d 2d74 7265 655f 6d65 7468   or `--tree_meth
-0000b4b0: 6f64 6020 746f 2062 6520 7072 652d 696e  od` to be pre-in
-0000b4c0: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
-0000b4d0: 6e76 6972 6f6e 6d65 6e74 2070 6174 6820  nvironment path 
-0000b4e0: 7769 7468 2074 6865 2070 6172 616d 6574  with the paramet
-0000b4f0: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
-0000b500: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-0000b510: 6e63 6520 636f 6e74 726f 6c6c 6564 2062  nce controlled b
-0000b520: 7920 602d 7473 6020 6f72 2060 2d2d 7472  y `-ts` or `--tr
-0000b530: 6565 7365 7460 292e 2054 6865 2070 726f  eeset`). The pro
-0000b540: 6772 616d 2060 7767 6420 646d 6460 2077  gram `wgd dmd` w
-0000b550: 6f75 6c64 2073 7469 6c6c 2063 6f6e 6475  ould still condu
-0000b560: 6374 2074 6865 2052 4248 7320 6361 6c63  ct the RBHs calc
-0000b570: 756c 6174 696f 6e20 756e 6c65 7373 2074  ulation unless t
-0000b580: 6865 2070 6172 616d 6574 6572 2060 2d6f  he parameter `-o
-0000b590: 6f60 206f 7220 602d 2d6f 6e6c 796f 7274  o` or `--onlyort
-0000b5a0: 686f 6020 7761 7320 7365 742e 2049 6620  ho` was set. If 
-0000b5b0: 6f6e 6520 6f6e 6c79 2077 616e 7473 2074  one only wants t
-0000b5c0: 6f20 696e 6665 7220 7468 6520 6f72 7468  o infer the orth
-0000b5d0: 6f67 726f 7570 732c 2069 7427 7320 7375  ogroups, it's su
-0000b5e0: 6767 6573 7465 6420 746f 2061 6464 2074  ggested to add t
-0000b5f0: 6865 2066 6c61 6720 602d 6f6f 6020 746f  he flag `-oo` to
-0000b600: 206a 7573 7420 696d 706c 656d 656e 7420   just implement 
-0000b610: 7468 6520 6f72 7468 6f67 726f 7570 7320  the orthogroups 
-0000b620: 6465 6c69 6e65 6174 696f 6e20 616e 616c  delineation anal
-0000b630: 7973 6973 2e20 5468 6520 6e75 6d62 6572  ysis. The number
-0000b640: 206f 6620 7061 7261 6c6c 656c 2074 6872   of parallel thr
-0000b650: 6561 6473 2069 7320 7375 6767 6573 7465  eads is suggeste
-0000b660: 6420 746f 2062 6520 7365 7420 6173 2028  d to be set as (
-0000b670: 4e2b 3129 4e2f 3220 7768 6572 6520 4e20  N+1)N/2 where N 
-0000b680: 6973 2074 6865 206e 756d 6265 7220 6f66  is the number of
-0000b690: 2063 6473 2066 696c 6573 2074 6f20 6163   cds files to ac
-0000b6a0: 6869 6576 6520 7468 6520 6869 6768 6573  hieve the highes
-0000b6b0: 7420 6566 6669 6369 656e 6379 2073 696e  t efficiency sin
-0000b6c0: 6365 2074 6865 2073 656c 662d 636f 6d70  ce the self-comp
-0000b6d0: 6172 6973 6f6e 2069 7320 616c 736f 2069  arison is also i
-0000b6e0: 6e63 6c75 6465 642e 0a0a 2a2a 5468 6520  ncluded...**The 
-0000b6f0: 6465 6661 756c 7420 7365 7474 696e 6720  default setting 
-0000b700: 6f66 2070 6172 616d 6574 6572 7320 6973  of parameters is
-0000b710: 2061 2072 6561 736f 6e61 626c 6520 7374   a reasonable st
-0000b720: 6172 7469 6e67 2070 6f69 6e74 2077 6974  arting point wit
-0000b730: 6820 7468 6520 636f 6d6d 616e 6420 6173  h the command as
-0000b740: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
-0000b750: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
-0000b760: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
-0000b770: 6e63 6533 202d 6f69 202d 6f6f 0a60 6060  nce3 -oi -oo.```
-0000b780: 0a0a 2a2a 5468 6520 636f 6c6c 696e 6561  ..**The collinea
-0000b790: 7220 636f 616c 6573 6365 6e63 6520 696e  r coalescence in
-0000b7a0: 6665 7265 6e63 6520 6f66 2070 6879 6c6f  ference of phylo
-0000b7b0: 6765 6e79 2a2a 0a60 6060 0a77 6764 2064  geny**.```.wgd d
-0000b7c0: 6d64 2073 6571 7565 6e63 6531 2073 6571  md sequence1 seq
-0000b7d0: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000b7e0: 202d 6170 2061 7064 6174 6120 2d73 6d20   -ap apdata -sm 
-0000b7f0: 736d 6461 7461 202d 6c65 206c 6564 6174  smdata -le ledat
-0000b800: 6120 2d67 7420 6774 6461 7461 202d 636f  a -gt gtdata -co
-0000b810: 6320 282d 7472 6565 2027 6661 7374 7472  c (-tree 'fasttr
-0000b820: 6565 2729 2028 2d74 7320 272d 6661 7374  ee') (-ts '-fast
-0000b830: 6573 7427 2920 282d 6e20 3429 2028 2d2d  est') (-n 4) (--
-0000b840: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
-0000b850: 2028 2d6f 2077 6764 5f64 6d64 2920 282d   (-o wgd_dmd) (-
-0000b860: 7420 776f 726b 696e 675f 746d 7029 0a60  t working_tmp).`
-0000b870: 6060 0a0a 4120 6e6f 7665 6c20 7068 796c  ``..A novel phyl
-0000b880: 6f67 656e 6574 6963 2069 6e66 6572 656e  ogenetic inferen
-0000b890: 6365 206d 6574 686f 6420 6e61 6d65 6420  ce method named 
-0000b8a0: 2263 6f6c 6c69 6e65 6172 2063 6f61 6c65  "collinear coale
-0000b8b0: 7363 656e 6365 2069 6e66 6572 656e 6365  scence inference
-0000b8c0: 2220 6973 2061 6c73 6f20 696d 706c 656d  " is also implem
-0000b8d0: 656e 7465 6420 696e 2060 7767 6420 7632  ented in `wgd v2
-0000b8e0: 602e 2046 6f72 2074 6869 7320 616e 616c  `. For this anal
-0000b8f0: 7973 6973 2c20 7573 6572 7320 6e65 6564  ysis, users need
-0000b900: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-0000b910: 616e 6368 6f72 2070 6f69 6e74 7320 6669  anchor points fi
-0000b920: 6c65 2062 7920 602d 6170 6020 6f72 2060  le by `-ap` or `
-0000b930: 2d2d 616e 6368 6f72 706f 696e 7473 602c  --anchorpoints`,
-0000b940: 2074 6865 2063 6f6c 6c69 6e65 6172 2073   the collinear s
-0000b950: 6567 6d65 6e74 7320 6669 6c65 2062 7920  egments file by 
-0000b960: 602d 736d 6020 6f72 2060 2d2d 7365 676d  `-sm` or `--segm
-0000b970: 656e 7473 602c 2074 6865 206c 6973 7473  ents`, the lists
-0000b980: 6567 6d65 6e74 7320 6669 6c65 2062 7920  egments file by 
-0000b990: 602d 6c65 6020 6f72 2060 2d2d 6c69 7374  `-le` or `--list
-0000b9a0: 656c 656d 656e 7473 602c 2061 6e64 2074  elements`, and t
-0000b9b0: 6865 2067 656e 6520 7461 626c 6520 6669  he gene table fi
-0000b9c0: 6c65 2062 7920 602d 6774 6020 6f72 2060  le by `-gt` or `
-0000b9d0: 2d2d 6765 6e65 7461 626c 6560 2c20 616c  --genetable`, al
-0000b9e0: 6c20 6f66 2077 6869 6368 2063 616e 2062  l of which can b
-0000b9f0: 6520 7072 6f64 7563 6564 2069 6e20 7468  e produced in th
-0000ba00: 6520 7072 6f67 7261 6d20 6077 6764 2073  e program `wgd s
-0000ba10: 796e 602e 2054 6865 2070 6172 616d 6574  yn`. The paramet
-0000ba20: 6572 2060 2d63 6f63 6020 6f72 2060 2d2d  er `-coc` or `--
-0000ba30: 636f 6c6c 696e 6561 7263 6f61 6c65 7363  collinearcoalesc
-0000ba40: 656e 6365 6020 6e65 6564 7320 746f 2062  ence` needs to b
-0000ba50: 6520 7365 7420 746f 2073 7461 7274 2074  e set to start t
-0000ba60: 6869 7320 616e 616c 7973 6973 2e20 5468  his analysis. Th
-0000ba70: 6520 7472 6565 2d69 6e66 6572 656e 6365  e tree-inference
-0000ba80: 2070 726f 6772 616d 2061 6e64 2074 6865   program and the
-0000ba90: 2061 7373 6f63 6961 7465 6420 7061 7261   associated para
-0000baa0: 6d65 7465 7273 2063 616e 2062 6520 7365  meters can be se
-0000bab0: 7420 6a75 7374 2061 7320 6162 6f76 6520  t just as above 
-0000bac0: 6279 2060 2d74 7265 6560 206f 7220 602d  by `-tree` or `-
-0000bad0: 2d74 7265 655f 6d65 7468 6f64 6020 616e  -tree_method` an
-0000bae0: 6420 602d 7473 6020 6f72 2060 2d2d 7472  d `-ts` or `--tr
-0000baf0: 6565 7365 7460 2e20 506c 6561 7365 2061  eeset`. Please a
-0000bb00: 6c73 6f20 6d61 6b65 2073 7572 6520 7468  lso make sure th
-0000bb10: 6520 6368 6f73 656e 2074 7265 652d 696e  e chosen tree-in
-0000bb20: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
-0000bb30: 6973 2069 6e73 7461 6c6c 6564 2069 6e20  is installed in 
-0000bb40: 7468 6520 656e 7669 726f 6e6d 656e 7420  the environment 
-0000bb50: 7061 7468 2e20 5468 6520 7072 6f67 7261  path. The progra
-0000bb60: 6d20 6061 7374 7261 6c2d 7072 6f60 2069  m `astral-pro` i
-0000bb70: 7320 7265 7175 6972 6564 2074 6f20 6265  s required to be
-0000bb80: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
-0000bb90: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
-0000bba0: 7468 2074 6f6f 2e20 4e6f 7465 2074 6861  th too. Note tha
-0000bbb0: 7420 7468 6572 6520 7368 6f75 6c64 2062  t there should b
-0000bbc0: 6520 6e6f 2064 7570 6c69 6361 7465 6420  e no duplicated 
-0000bbd0: 6765 6e65 2049 4473 2069 6e20 7468 6520  gene IDs in the 
-0000bbe0: 7365 7175 656e 6365 2066 696c 652e 2054  sequence file. T
-0000bbf0: 6865 2070 6172 616c 6c65 6c20 7468 7265  he parallel thre
-0000bc00: 6164 7320 6865 7265 2061 7265 2074 6f20  ads here are to 
-0000bc10: 6163 6365 6c65 7261 7465 2074 6865 2073  accelerate the s
-0000bc20: 6571 7565 6e63 6520 616c 6967 6e6d 656e  equence alignmen
-0000bc30: 7420 616e 6420 6765 6e65 2074 7265 6520  t and gene tree 
-0000bc40: 696e 6665 7265 6e63 6520 666f 7220 6561  inference for ea
-0000bc50: 6368 2067 656e 6520 6661 6d69 6c79 2061  ch gene family a
-0000bc60: 6e64 2074 6875 7320 7375 6767 6573 7465  nd thus suggeste
-0000bc70: 6420 746f 2062 6520 7365 7420 6173 206d  d to be set as m
-0000bc80: 7563 6820 6173 2074 6865 206e 756d 6265  uch as the numbe
-0000bc90: 7220 6f66 2067 656e 6520 6661 6d69 6c69  r of gene famili
-0000bca0: 6573 2e0a 0a2a 2a41 2073 7567 6765 7374  es...**A suggest
-0000bcb0: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-0000bcc0: 6f66 2074 6869 7320 616e 616c 7973 6973  of this analysis
-0000bcd0: 2069 7320 7769 7468 2074 6865 2073 696d   is with the sim
-0000bce0: 706c 6520 636f 6d6d 616e 6420 6265 6c6f  ple command belo
-0000bcf0: 772e 2a2a 0a0a 6060 600a 7767 6420 646d  w.**..```.wgd dm
-0000bd00: 6420 7365 7175 656e 6365 3120 7365 7175  d sequence1 sequ
-0000bd10: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000bd20: 2d61 7020 6170 6461 7461 202d 736d 2073  -ap apdata -sm s
-0000bd30: 6d64 6174 6120 2d6c 6520 6c65 6461 7461  mdata -le ledata
-0000bd40: 202d 6774 2067 7464 6174 6120 2d63 6f63   -gt gtdata -coc
-0000bd50: 0a60 6060 0a0a 2323 2320 7767 6420 666f  .```..### wgd fo
-0000bd60: 6375 730a 0a2a 2a54 6865 2063 6f6e 6361  cus..**The conca
-0000bd70: 7465 6e61 7469 6f6e 2d62 6173 6564 2f63  tenation-based/c
-0000bd80: 6f61 6c65 7363 656e 6365 2d62 6173 6564  oalescence-based
-0000bd90: 2070 6879 6c6f 6765 6e65 7469 6320 696e   phylogenetic in
-0000bda0: 6665 7265 6e63 652a 2a0a 6060 600a 7767  ference**.```.wg
-0000bdb0: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000bdc0: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000bdd0: 6e63 6532 2073 6571 7565 6e63 6533 2028  nce2 sequence3 (
-0000bde0: 2d2d 636f 6e63 6174 656e 6174 696f 6e29  --concatenation)
-0000bdf0: 2028 2d2d 636f 616c 6573 6365 6e63 6529   (--coalescence)
-0000be00: 2028 2d74 7265 6520 2766 6173 7474 7265   (-tree 'fasttre
-0000be10: 6527 2920 282d 7473 2027 2d66 6173 7465  e') (-ts '-faste
-0000be20: 7374 2729 2028 2d6e 2034 2920 282d 2d74  st') (-n 4) (--t
-0000be30: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
-0000be40: 282d 6f20 7767 645f 666f 6375 7329 2028  (-o wgd_focus) (
-0000be50: 2d74 2077 6f72 6b69 6e67 5f74 6d70 290a  -t working_tmp).
-0000be60: 6060 600a 0a54 6865 2070 726f 6772 616d  ```..The program
-0000be70: 2060 7767 6420 666f 6375 7360 2069 6d70   `wgd focus` imp
-0000be80: 6c65 6d65 6e74 6564 2074 776f 2062 6173  lemented two bas
-0000be90: 6963 2070 6879 6c6f 6765 6e65 7469 6320  ic phylogenetic 
-0000bea0: 696e 6665 7265 6e63 6520 6d65 7468 6f64  inference method
-0000beb0: 732c 2069 2e65 2e2c 2063 6f6e 6361 7465  s, i.e., concate
-0000bec0: 6e61 7469 6f6e 2d62 6173 6564 2061 6e64  nation-based and
-0000bed0: 2063 6f61 6c65 7363 656e 6365 2d62 6173   coalescence-bas
-0000bee0: 6564 206d 6574 686f 6473 2e20 546f 2069  ed methods. To i
-0000bef0: 6e69 7469 6174 6520 7468 6573 6520 616e  nitiate these an
-0000bf00: 616c 7973 6973 2c20 7573 6572 7320 6e65  alysis, users ne
-0000bf10: 6564 2074 6f20 7365 7420 7468 6520 666c  ed to set the fl
-0000bf20: 6167 206f 7074 696f 6e20 602d 2d63 6f6e  ag option `--con
-0000bf30: 6361 7465 6e61 7469 6f6e 6020 6f72 2060  catenation` or `
-0000bf40: 2d2d 636f 616c 6573 6365 6e63 6560 2e20  --coalescence`. 
-0000bf50: 5468 6520 636f 6e63 6174 656e 6174 696f  The concatenatio
-0000bf60: 6e2d 6261 7365 6420 6d65 7468 6f64 2069  n-based method i
-0000bf70: 6e63 6c75 6465 7320 6120 6665 7720 6d61  ncludes a few ma
-0000bf80: 6a6f 7220 7374 6570 732c 2069 2e65 2e2c  jor steps, i.e.,
-0000bf90: 2074 6865 206d 756c 7469 706c 6520 7365   the multiple se
-0000bfa0: 7175 656e 6365 2061 6c69 676e 6d65 6e74  quence alignment
-0000bfb0: 2028 4d53 4129 206f 6620 6561 6368 2067   (MSA) of each g
-0000bfc0: 656e 6520 6661 6d69 6c79 2c20 7468 6520  ene family, the 
-0000bfd0: 636f 6e63 6174 656e 6174 696f 6e20 6f66  concatenation of
-0000bfe0: 2061 6c6c 2067 656e 6520 6661 6d69 6c69   all gene famili
-0000bff0: 6573 2061 6e64 2074 6865 6e20 7468 6520  es and then the 
-0000c000: 6765 6e65 2074 7265 6520 2861 6c73 6f20  gene tree (also 
-0000c010: 7370 6563 6965 7320 7472 6565 2069 6e20  species tree in 
-0000c020: 7468 6973 2063 6173 6529 2069 6e66 6572  this case) infer
-0000c030: 656e 6365 2e20 5468 6520 636f 616c 6573  ence. The coales
-0000c040: 6365 6e63 652d 6261 7365 6420 6d65 7468  cence-based meth
-0000c050: 6f64 2077 696c 6c20 696e 7374 6561 6420  od will instead 
-0000c060: 7065 7266 6f72 6d20 7468 6520 4d53 4120  perform the MSA 
-0000c070: 6f66 2065 6163 6820 6765 6e65 2066 616d  of each gene fam
-0000c080: 696c 7920 616e 6420 7468 6520 6765 6e65  ily and the gene
-0000c090: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
-0000c0a0: 6261 7365 6420 6f6e 2065 6163 6820 4d53  based on each MS
-0000c0b0: 412c 2061 6e64 2074 6865 6e20 696e 6665  A, and then infe
-0000c0c0: 7220 7468 6520 7370 6563 6965 7320 7472  r the species tr
-0000c0d0: 6565 2062 6173 6564 206f 6e20 7468 6573  ee based on thes
-0000c0e0: 6520 696e 6469 7669 6475 616c 2067 656e  e individual gen
-0000c0f0: 6520 7472 6565 732e 2054 6865 2074 7265  e trees. The tre
-0000c100: 652d 696e 6665 7265 6e63 6520 7072 6f67  e-inference prog
-0000c110: 7261 6d20 616e 6420 7468 6520 6173 736f  ram and the asso
-0000c120: 6369 6174 6564 2070 6172 616d 6574 6572  ciated parameter
-0000c130: 7320 6361 6e20 6265 2073 6574 206a 7573  s can be set jus
-0000c140: 7420 6173 2061 626f 7665 2062 7920 602d  t as above by `-
-0000c150: 7472 6565 6020 6f72 2060 2d2d 7472 6565  tree` or `--tree
-0000c160: 5f6d 6574 686f 6460 2061 6e64 2060 2d74  _method` and `-t
-0000c170: 7360 206f 7220 602d 2d74 7265 6573 6574  s` or `--treeset
-0000c180: 602e 2050 6c65 6173 6520 616c 736f 206d  `. Please also m
-0000c190: 616b 6520 7375 7265 2074 6865 2063 686f  ake sure the cho
-0000c1a0: 7365 6e20 7472 6565 2d69 6e66 6572 656e  sen tree-inferen
-0000c1b0: 6365 2070 726f 6772 616d 2069 7320 696e  ce program is in
-0000c1c0: 7374 616c 6c65 6420 696e 2074 6865 2065  stalled in the e
-0000c1d0: 6e76 6972 6f6e 6d65 6e74 2070 6174 682e  nvironment path.
-0000c1e0: 2054 6865 2070 726f 6772 616d 2060 6173   The program `as
-0000c1f0: 7472 616c 2d70 726f 6020 6973 2072 6571  tral-pro` is req
-0000c200: 7569 7265 6420 746f 2062 6520 696e 7374  uired to be inst
-0000c210: 616c 6c65 6420 696e 2074 6865 2065 6e76  alled in the env
-0000c220: 6972 6f6e 6d65 6e74 2070 6174 6820 6966  ironment path if
-0000c230: 2074 6865 2063 6f61 6c65 7363 656e 6365   the coalescence
-0000c240: 2d62 6173 6564 206d 6574 686f 6420 6973  -based method is
-0000c250: 2063 686f 7365 6e2e 2054 6865 2070 6172   chosen. The par
-0000c260: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000c270: 7265 2061 7265 2074 6f20 6163 6365 6c65  re are to accele
-0000c280: 7261 7465 2074 6865 2073 6571 7565 6e63  rate the sequenc
-0000c290: 6520 616c 6967 6e6d 656e 7420 616e 6420  e alignment and 
-0000c2a0: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
-0000c2b0: 6e63 6520 666f 7220 6561 6368 2067 656e  nce for each gen
-0000c2c0: 6520 6661 6d69 6c79 2074 6f6f 2061 6e64  e family too and
-0000c2d0: 2074 6875 7320 7375 6767 6573 7465 6420   thus suggested 
-0000c2e0: 746f 2062 6520 7365 7420 6173 206d 7563  to be set as muc
-0000c2f0: 6820 6173 2074 6865 206e 756d 6265 7220  h as the number 
-0000c300: 6f66 2067 656e 6520 6661 6d69 6c69 6573  of gene families
-0000c310: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
-0000c320: 2073 7461 7274 696e 6720 7275 6e20 6f66   starting run of
-0000c330: 2074 6869 7320 616e 616c 7973 6973 2069   this analysis i
-0000c340: 7320 7769 7468 2074 6865 2073 696d 706c  s with the simpl
-0000c350: 6520 636f 6d6d 616e 6420 6265 6c6f 772e  e command below.
-0000c360: 2a2a 0a0a 6060 600a 7767 6420 666f 6375  **..```.wgd focu
-0000c370: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-0000c380: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000c390: 6571 7565 6e63 6533 202d 2d63 6f6e 6361  equence3 --conca
-0000c3a0: 7465 6e61 7469 6f6e 0a77 6764 2066 6f63  tenation.wgd foc
-0000c3b0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000c3c0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000c3d0: 7365 7175 656e 6365 3320 2d2d 636f 616c  sequence3 --coal
-0000c3e0: 6573 6365 6e63 650a 6060 600a 0a2a 2a54  escence.```..**T
-0000c3f0: 6865 2066 756e 6374 696f 6e61 6c20 616e  he functional an
-0000c400: 6e6f 7461 7469 6f6e 206f 6620 6765 6e65  notation of gene
-0000c410: 2066 616d 696c 6965 732a 2a0a 6060 600a   families**.```.
-0000c420: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000c430: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000c440: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000c450: 202d 2d61 6e6e 6f74 6174 696f 6e20 6567   --annotation eg
-0000c460: 676e 6f67 202d 6564 2065 6464 6174 6120  gnog -ed eddata 
-0000c470: 2d2d 646d 6e62 2064 6264 6174 610a 6060  --dmnb dbdata.``
-0000c480: 600a 0a54 6865 2070 726f 6772 616d 2060  `..The program `
-0000c490: 7767 6420 666f 6375 7360 2061 6c73 6f20  wgd focus` also 
-0000c4a0: 6164 6465 6420 736f 6d65 2077 7261 7070  added some wrapp
-0000c4b0: 696e 6720 6675 6e63 7469 6f6e 7320 666f  ing functions fo
-0000c4c0: 7220 6675 6e63 7469 6f6e 616c 2061 6e6e  r functional ann
-0000c4d0: 6f74 6174 696f 6e20 6f66 2067 656e 6520  otation of gene 
-0000c4e0: 6661 6d69 6c69 6573 206f 6e20 7468 6520  families on the 
-0000c4f0: 686f 6f64 206f 6620 6461 7461 6261 7365  hood of database
-0000c500: 7320 616e 6420 736f 6674 7761 7265 7320  s and softwares 
-0000c510: 666f 7220 696e 7374 616e 6365 2060 4567  for instance `Eg
-0000c520: 674e 4f47 6020 616e 6420 6045 6767 4e4f  gNOG` and `EggNO
-0000c530: 472d 6d61 7070 6572 602e 2046 6f72 2074  G-mapper`. For t
-0000c540: 6865 2061 6e6e 6f74 6174 696f 6e20 7573  he annotation us
-0000c550: 696e 6720 6045 6767 4e4f 472d 6d61 7070  ing `EggNOG-mapp
-0000c560: 6572 602c 2075 7365 7273 206e 6565 6420  er`, users need 
-0000c570: 746f 2070 726f 7669 6465 2074 6865 2070  to provide the p
-0000c580: 6174 6820 746f 2074 6865 2065 6767 4e4f  ath to the eggNO
-0000c590: 4720 616e 6e6f 7461 7469 6f6e 2064 6174  G annotation dat
-0000c5a0: 6162 6173 6520 7669 6120 7468 6520 6f70  abase via the op
-0000c5b0: 7469 6f6e 2060 2d65 6460 2c20 7468 6520  tion `-ed`, the 
-0000c5c0: 7061 7468 2074 6f20 7468 6520 6469 616d  path to the diam
-0000c5d0: 6f6e 642d 636f 6d70 6174 6962 6c65 2064  ond-compatible d
-0000c5e0: 6174 6162 6173 6520 7669 6120 7468 6520  atabase via the 
-0000c5f0: 6f70 7469 6f6e 2060 2d2d 646d 6e64 5f64  option `--dmnd_d
-0000c600: 6260 2061 6e64 2074 6865 206f 7074 696f  b` and the optio
-0000c610: 6e20 602d 2d61 6e6e 6f74 6174 696f 6e60  n `--annotation`
-0000c620: 2073 6574 2061 7320 2265 6767 6e6f 6722   set as "eggnog"
-0000c630: 2e20 5468 6520 6d61 6e6e 6572 2068 6f77  . The manner how
-0000c640: 2050 4641 4d20 616e 6e6f 7461 7469 6f6e   PFAM annotation
-0000c650: 2077 696c 6c20 6265 2070 6572 666f 726d   will be perform
-0000c660: 6564 2063 616e 2062 6520 636f 6e74 726f  ed can be contro
-0000c670: 6c6c 6564 2076 6961 2074 6865 206f 7074  lled via the opt
-0000c680: 696f 6e20 6070 6661 6d60 2c20 6569 7468  ion `pfam`, eith
-0000c690: 6572 2022 6e6f 6e65 222c 2022 7265 616c  er "none", "real
-0000c6a0: 6967 6e22 206f 7220 2264 656e 6f76 6f22  ign" or "denovo"
-0000c6b0: 2c20 7468 6520 6465 7461 696c 6564 2065  , the detailed e
-0000c6c0: 7870 6c61 6e61 7469 6f6e 2063 616e 2062  xplanation can b
-0000c6d0: 6520 666f 756e 6420 6174 2074 6865 2077  e found at the w
-0000c6e0: 696b 6920 6f66 205b 4567 674e 4f47 2d6d  iki of [EggNOG-m
-0000c6f0: 6170 7065 725d 2868 7474 7073 3a2f 2f67  apper](https://g
-0000c700: 6974 6875 622e 636f 6d2f 6567 676e 6f67  ithub.com/eggnog
-0000c710: 6462 2f65 6767 6e6f 672d 6d61 7070 6572  db/eggnog-mapper
-0000c720: 2f77 696b 6929 2e20 506c 6561 7365 2070  /wiki). Please p
-0000c730: 7265 2d69 6e73 7461 6c6c 2074 6865 2060  re-install the `
-0000c740: 4567 674e 4f47 2d6d 6170 7065 7260 2070  EggNOG-mapper` p
-0000c750: 7974 686f 6e20 7061 636b 6167 6520 6966  ython package if
-0000c760: 2075 7369 6e67 2074 6869 7320 6675 6e63   using this func
-0000c770: 7469 6f6e 2e20 466f 7220 7468 6520 616e  tion. For the an
-0000c780: 6e6f 7461 7469 6f6e 2075 7369 6e67 2060  notation using `
-0000c790: 686d 6d73 6361 6e60 2c20 7768 6174 2069  hmmscan`, what i
-0000c7a0: 7320 696d 706c 656d 656e 7465 6420 696e  s implemented in
-0000c7b0: 2060 7767 6420 7632 6020 6973 2061 2073   `wgd v2` is a s
-0000c7c0: 696d 706c 6520 6275 6e64 6c65 2066 756e  imple bundle fun
-0000c7d0: 6374 696f 6e20 746f 2070 6572 666f 726d  ction to perform
-0000c7e0: 2060 686d 6d73 6361 6e60 2061 6e61 6c79   `hmmscan` analy
-0000c7f0: 7369 7320 666f 7220 6120 6769 7665 6e20  sis for a given 
-0000c800: 686d 6d20 7072 6f66 696c 6520 616e 6420  hmm profile and 
-0000c810: 7365 7420 6f66 2067 656e 6520 6661 6d69  set of gene fami
-0000c820: 6c69 6573 2073 7563 6820 7468 6174 2075  lies such that u
-0000c830: 7365 7273 206e 6565 6420 746f 2073 6574  sers need to set
-0000c840: 2074 6865 206f 7074 696f 6e20 602d 2d61   the option `--a
-0000c850: 6e6e 6f74 6174 696f 6e60 2061 7320 2268  nnotation` as "h
-0000c860: 6d6d 7066 616d 2220 616e 6420 7072 6f76  mmpfam" and prov
-0000c870: 6964 6520 7468 6520 686d 6d70 726f 6669  ide the hmmprofi
-0000c880: 6c65 2076 6961 2074 6865 206f 7074 696f  le via the optio
-0000c890: 6e20 602d 2d68 6d6d 602e 2046 6f72 2074  n `--hmm`. For t
-0000c8a0: 6865 2061 6e6e 6f74 6174 696f 6e20 7573  he annotation us
-0000c8b0: 696e 6720 6069 6e74 6572 7072 6f73 6361  ing `interprosca
-0000c8c0: 6e60 2c20 7573 6572 7320 6e65 6564 2074  n`, users need t
-0000c8d0: 6f20 7072 6f76 6964 6520 7468 6520 7061  o provide the pa
-0000c8e0: 7468 2074 6f20 7468 6520 696e 7465 7270  th to the interp
-0000c8f0: 726f 7363 616e 2069 6e73 7461 6c6c 6174  roscan installat
-0000c900: 696f 6e20 666f 6c64 6572 2077 6865 7265  ion folder where
-0000c910: 2074 6865 7265 2069 7320 6120 6069 6e74   there is a `int
-0000c920: 6572 7072 6f73 6361 6e2e 7368 6020 6669  erproscan.sh` fi
-0000c930: 6c65 2076 6961 2074 6865 206f 7074 696f  le via the optio
-0000c940: 6e20 602d 2d65 7865 7061 7468 6020 616e  n `--exepath` an
-0000c950: 6420 7365 7420 7468 6520 7468 6520 6f70  d set the the op
-0000c960: 7469 6f6e 2060 2d2d 616e 6e6f 7461 7469  tion `--annotati
-0000c970: 6f6e 6020 6173 2022 696e 7465 7270 726f  on` as "interpro
-0000c980: 7363 616e 222e 2054 6865 2070 6172 616c  scan". The paral
-0000c990: 6c65 6c20 7468 7265 6164 7320 6865 7265  lel threads here
-0000c9a0: 2061 7265 2074 6f20 7061 7261 6c6c 656c   are to parallel
-0000c9b0: 697a 6520 7468 6520 616e 616c 7973 6973  ize the analysis
-0000c9c0: 2066 6f72 2065 6163 6820 6765 6e65 2066   for each gene f
-0000c9d0: 616d 696c 7920 616e 6420 7468 7573 2073  amily and thus s
-0000c9e0: 7567 6765 7374 6564 2074 6f20 6265 2073  uggested to be s
-0000c9f0: 6574 2061 7320 6d75 6368 2061 7320 7468  et as much as th
-0000ca00: 6520 6e75 6d62 6572 206f 6620 6765 6e65  e number of gene
-0000ca10: 2066 616d 696c 6965 732e 0a0a 2a2a 4120   families...**A 
-0000ca20: 7375 6767 6573 7465 6420 7374 6172 7469  suggested starti
-0000ca30: 6e67 2072 756e 206f 6620 7468 6973 2061  ng run of this a
-0000ca40: 6e61 6c79 7369 7320 6361 6e20 6265 2077  nalysis can be w
-0000ca50: 6974 6820 7468 6520 636f 6d6d 616e 6420  ith the command 
-0000ca60: 6265 6c6f 772e 2a2a 0a0a 6060 600a 7767  below.**..```.wg
-0000ca70: 6420 666f 6375 7320 6661 6d69 6c69 6573  d focus families
-0000ca80: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
-0000ca90: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
-0000caa0: 2d61 6e6e 6f74 6174 696f 6e20 6567 676e  -annotation eggn
-0000cab0: 6f67 202d 6564 2065 6464 6174 6120 2d2d  og -ed eddata --
-0000cac0: 646d 6e62 2064 6264 6174 610a 7767 6420  dmnb dbdata.wgd 
-0000cad0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
-0000cae0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
-0000caf0: 6532 2073 6571 7565 6e63 6533 202d 2d61  e2 sequence3 --a
-0000cb00: 6e6e 6f74 6174 696f 6e20 686d 6d70 6661  nnotation hmmpfa
-0000cb10: 6d20 2d2d 686d 6d20 686d 6d64 6174 610a  m --hmm hmmdata.
-0000cb20: 7767 6420 666f 6375 7320 6661 6d69 6c69  wgd focus famili
-0000cb30: 6573 2073 6571 7565 6e63 6531 2073 6571  es sequence1 seq
-0000cb40: 7565 6e63 6532 2073 6571 7565 6e63 6533  uence2 sequence3
-0000cb50: 202d 2d61 6e6e 6f74 6174 696f 6e20 696e   --annotation in
-0000cb60: 7465 7270 726f 7363 616e 202d 2d65 7865  terproscan --exe
-0000cb70: 7061 7468 2024 5041 5448 0a60 6060 0a0a  path $PATH.```..
-0000cb80: 2a2a 5468 6520 7068 796c 6f67 656e 6574  **The phylogenet
-0000cb90: 6963 2064 6174 696e 6720 6f66 2057 4744  ic dating of WGD
-0000cba0: 732a 2a0a 6060 600a 7767 6420 666f 6375  s**.```.wgd focu
-0000cbb0: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
-0000cbc0: 6e63 6531 2073 6571 7565 6e63 6532 2073  nce1 sequence2 s
-0000cbd0: 6571 7565 6e63 6533 202d 6420 6d63 6d63  equence3 -d mcmc
-0000cbe0: 7472 6565 202d 7370 2073 7064 6174 6120  tree -sp spdata 
-0000cbf0: 282d 2d70 726f 7463 6f63 6461 7469 6e67  (--protcocdating
-0000cc00: 2920 282d 2d70 6172 7469 7469 6f6e 2920  ) (--partition) 
-0000cc10: 282d 2d61 616d 6f64 656c 206c 6729 2028  (--aamodel lg) (
-0000cc20: 2d64 7320 2762 7572 6e69 6e20 3d20 3230  -ds 'burnin = 20
-0000cc30: 3030 2729 2028 2d64 7320 2773 616d 7066  00') (-ds 'sampf
-0000cc40: 7265 7120 3d20 3130 3030 2729 2028 2d64  req = 1000') (-d
-0000cc50: 7320 276e 7361 6d70 6c65 203d 2032 3030  s 'nsample = 200
-0000cc60: 3030 2729 2020 282d 6e20 3429 2028 2d2d  00')  (-n 4) (--
-0000cc70: 746f 5f73 746f 7029 2028 2d2d 6364 7329  to_stop) (--cds)
-0000cc80: 2028 2d6f 2077 6764 5f66 6f63 7573 2920   (-o wgd_focus) 
-0000cc90: 282d 7420 776f 726b 696e 675f 746d 7029  (-t working_tmp)
-0000cca0: 0a60 6060 0a0a 5468 6520 6162 736f 6c75  .```..The absolu
-0000ccb0: 7465 2064 6174 696e 6720 6f66 2057 4744  te dating of WGD
-0000ccc0: 7320 6973 2061 2073 7065 6369 6669 6320  s is a specific 
-0000ccd0: 7069 7065 6c69 6e65 2069 6d70 6c65 6d65  pipeline impleme
-0000cce0: 6e74 6564 2069 6e20 6077 6764 2076 3260  nted in `wgd v2`
-0000ccf0: 2075 7369 6e67 2074 6865 206d 6574 686f   using the metho
-0000cd00: 6420 6f66 2070 6879 6c6f 6765 6e65 7469  d of phylogeneti
-0000cd10: 6320 6461 7469 6e67 2e20 5468 6520 6661  c dating. The fa
-0000cd20: 6d69 6c69 6573 2075 7365 6420 696e 2074  milies used in t
-0000cd30: 6869 7320 7374 6570 2063 616e 2062 6520  his step can be 
-0000cd40: 7072 6f64 7563 6564 2066 726f 6d20 6077  produced from `w
-0000cd50: 6764 2064 6d64 6020 616e 6420 6077 6764  gd dmd` and `wgd
-0000cd60: 2070 6561 6b60 2e20 4e6f 7465 2074 6861   peak`. Note tha
-0000cd70: 7420 6865 7265 2077 6520 6f6e 6c79 2064  t here we only d
-0000cd80: 6973 6375 7373 2068 6f77 2074 6f20 6461  iscuss how to da
-0000cd90: 7465 2057 4744 7320 7769 7468 2067 656e  te WGDs with gen
-0000cda0: 6f6d 6520 6173 7365 6d62 6c79 2c20 696e  ome assembly, in
-0000cdb0: 7374 6561 6420 6f66 2074 7261 6e73 6372  stead of transcr
-0000cdc0: 6970 746f 6d65 2061 7373 656d 626c 7920  iptome assembly 
-0000cdd0: 2877 6869 6368 2077 696c 6c20 6265 2064  (which will be d
-0000cde0: 6973 6375 7373 6564 2069 6e20 6120 7365  iscussed in a se
-0000cdf0: 7061 7261 7465 2073 6563 7469 6f6e 2068  parate section h
-0000ce00: 6572 6575 6e64 6572 292e 2054 6865 2061  ereunder). The a
-0000ce10: 7373 756d 7074 696f 6e20 7765 206d 6164  ssumption we mad
-0000ce20: 6520 6865 7265 2069 7320 7468 6174 206e  e here is that n
-0000ce30: 6f74 2061 6c6c 2061 6e63 686f 7220 7061  ot all anchor pa
-0000ce40: 6972 7320 2863 6f6c 6c69 6e65 6172 2064  irs (collinear d
-0000ce50: 7570 6c69 6361 7465 7329 2061 7265 2073  uplicates) are s
-0000ce60: 7569 7461 626c 6520 666f 7220 7068 796c  uitable for phyl
-0000ce70: 6f67 656e 6574 6963 2064 6174 696e 672c  ogenetic dating,
-0000ce80: 2066 6f72 2069 6e73 7461 6e63 6520 7468   for instance th
-0000ce90: 6f73 6520 6661 7374 6c79 206f 7220 736c  ose fastly or sl
-0000cea0: 6f77 6c79 2065 766f 6c76 696e 6720 6765  owly evolving ge
-0000ceb0: 6e65 2064 7570 6c69 6361 7465 732c 2062  ne duplicates, b
-0000cec0: 6563 6175 7365 2074 6865 7927 7265 2070  ecause they're p
-0000ced0: 726f 6e65 2074 6f20 6769 7665 2062 6961  rone to give bia
-0000cee0: 7365 6420 6461 7469 6e67 2065 7374 696d  sed dating estim
-0000cef0: 6174 696f 6e2e 2053 6f20 6173 2074 6f20  ation. So as to 
-0000cf00: 7265 7472 6965 7665 2074 6865 2022 7265  retrieve the "re
-0000cf10: 6c69 6162 6c65 2220 616e 6368 6f72 2070  liable" anchor p
-0000cf20: 6169 7273 2c20 7765 2069 6d70 6c65 6d65  airs, we impleme
-0000cf30: 6e74 6564 2073 6f6d 6520 6d65 7468 6f64  nted some method
-0000cf40: 7320 6f66 2069 6465 6e74 6966 7969 6e67  s of identifying
-0000cf50: 2063 7265 6469 6162 6c65 2061 6e63 686f   crediable ancho
-0000cf60: 7220 7061 6972 7320 6261 7365 6420 6f6e  r pairs based on
-0000cf70: 2074 6865 6972 204b 7320 7661 6c75 6573   their Ks values
-0000cf80: 2061 6e64 2f6f 7220 7265 7369 6469 6e67   and/or residing
-0000cf90: 2063 6f6c 6c69 6e65 6172 2073 6567 6d65   collinear segme
-0000cfa0: 6e74 732e 2046 6f72 2061 2067 656e 6f6d  nts. For a genom
-0000cfb0: 6520 7769 7468 2063 6c65 6172 2073 6967  e with clear sig
-0000cfc0: 6e61 6c73 206f 6620 7075 7461 7469 7665  nals of putative
-0000cfd0: 2057 4744 732c 2073 7563 6820 6173 2074   WGDs, such as t
-0000cfe0: 6865 202a 4171 7569 6c65 6769 6120 636f  he *Aquilegia co
-0000cff0: 6572 756c 6561 2a20 696e 2074 6865 2065  erulea* in the e
-0000d000: 7861 6d70 6c65 2c20 6120 6865 7572 6973  xample, a heuris
-0000d010: 7469 6320 6d65 7468 6f64 2074 6861 7420  tic method that 
-0000d020: 6170 706c 6965 7320 7468 6520 7072 696e  applies the prin
-0000d030: 6369 706c 6520 6f66 2068 6f77 205b 6b73  ciple of how [ks
-0000d040: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
-0000d050: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
-0000d060: 422f 6b73 7261 7465 7329 2066 696e 6420  B/ksrates) find 
-0000d070: 7468 6520 696e 6974 6961 6c20 7065 616b  the initial peak
-0000d080: 7320 616e 6420 7468 6569 7220 7061 7261  s and their para
-0000d090: 6d65 7465 7273 2077 6173 2069 6d70 6c65  meters was imple
-0000d0a0: 6d65 6e74 6564 2074 6f20 6669 6e64 2074  mented to find t
-0000d0b0: 6865 2039 3525 2063 6f6e 6669 6465 6e63  he 95% confidenc
-0000d0c0: 6520 6c65 7665 6c20 6f66 2074 6865 2061  e level of the a
-0000d0d0: 7373 756d 6564 206c 6f67 6e6f 726d 616c  ssumed lognormal
-0000d0e0: 2064 6973 7472 6962 7574 696f 6e20 6f66   distribution of
-0000d0f0: 2074 6865 2061 6e63 686f 7220 7061 6972   the anchor pair
-0000d100: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d110: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000d120: 6e20 746f 2066 696c 7465 7220 616e 6368  n to filter anch
-0000d130: 6f72 2070 6169 7273 2077 6974 6820 746f  or pairs with to
-0000d140: 6f20 6869 6768 206f 7220 6c6f 7720 2a4b  o high or low *K
-0000d150: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
-0000d160: 6573 2c20 7468 6520 6578 616d 706c 6172  es, the examplar
-0000d170: 2063 6f6d 6d61 6e64 206f 6620 7768 6963   command of whic
-0000d180: 6820 6973 2073 686f 7765 6420 696e 2074  h is showed in t
-0000d190: 6865 2060 7767 6420 7065 616b 6020 7365  he `wgd peak` se
-0000d1a0: 6374 696f 6e20 6f66 205b 496c 6c75 7374  ction of [Illust
-0000d1b0: 7261 7469 6f6e 5d28 6874 7470 733a 2f2f  ration](https://
-0000d1c0: 6769 7468 7562 2e63 6f6d 2f68 6563 6865  github.com/heche
-0000d1d0: 2d70 7362 2f77 6764 3f74 6162 3d72 6561  -psb/wgd?tab=rea
-0000d1e0: 646d 652d 6f76 2d66 696c 6523 696c 6c75  dme-ov-file#illu
-0000d1f0: 7374 7261 7469 6f6e 292e 2049 6620 7468  stration). If th
-0000d200: 6973 2068 6575 7269 7374 6963 206d 6574  is heuristic met
-0000d210: 686f 6420 6661 696c 6564 2074 6f20 6769  hod failed to gi
-0000d220: 7665 2072 6561 736f 6e61 626c 6520 7265  ve reasonable re
-0000d230: 7375 6c74 732c 2077 6869 6368 2069 7320  sults, which is 
-0000d240: 7573 7561 6c6c 7920 6475 6520 746f 2074  usually due to t
-0000d250: 6865 2065 6666 6563 7420 6f66 206d 756c  he effect of mul
-0000d260: 7469 706c 6520 6164 6a61 6365 6e74 2057  tiple adjacent W
-0000d270: 4744 7320 7468 6174 2062 6c75 7273 2074  GDs that blurs t
-0000d280: 6865 2070 6561 6b2d 6669 6e64 696e 6720  he peak-finding 
-0000d290: 7072 6f63 6573 732c 2075 7365 7273 2063  process, users c
-0000d2a0: 616e 2074 7572 6e20 746f 2074 6865 2063  an turn to the c
-0000d2b0: 6f6c 6c69 6e65 6172 2073 6567 6d65 6e74  ollinear segment
-0000d2c0: 732d 6775 6964 6564 2061 6e63 686f 7220  s-guided anchor 
-0000d2d0: 7061 6972 2063 6c75 7374 6572 696e 6720  pair clustering 
-0000d2e0: 696d 706c 656d 656e 7465 6420 696e 2060  implemented in `
-0000d2f0: 7767 6420 7632 602c 2069 6e20 7768 6963  wgd v2`, in whic
-0000d300: 6820 6120 636f 6c6c 696e 6561 7220 7365  h a collinear se
-0000d310: 676d 656e 742d 7769 7365 2047 4d4d 2063  gment-wise GMM c
-0000d320: 6c75 7374 6572 696e 6720 7769 6c6c 2062  lustering will b
-0000d330: 6520 6669 7273 7420 636f 6e64 7563 7465  e first conducte
-0000d340: 6420 6261 7365 6420 6f6e 2074 6865 2022  d based on the "
-0000d350: 736f 2d63 616c 6c65 6422 2073 6567 6d65  so-called" segme
-0000d360: 6e74 202a 4b2a 3c73 7562 3e53 3c2f 7375  nt *K*<sub>S</su
-0000d370: 623e 2061 6765 2072 6570 7265 7365 6e74  b> age represent
-0000d380: 6564 2062 7920 7468 6520 6d65 6469 616e  ed by the median
-0000d390: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d3a0: 2061 6765 206f 6620 616c 6c20 7468 6520   age of all the 
-0000d3b0: 7265 7369 6469 6e67 2073 796e 7465 6c6f  residing syntelo
-0000d3c0: 6773 2028 6e6f 7465 2074 6861 7420 7468  gs (note that th
-0000d3d0: 6520 6765 6e65 2064 7570 6c69 6361 7465  e gene duplicate
-0000d3e0: 2070 6169 7273 2061 646f 7074 6564 2069   pairs adopted i
-0000d3f0: 6e20 7468 6973 2073 7465 7020 6973 2066  n this step is f
-0000d400: 726f 6d20 7468 6520 6669 6c65 2060 6d75  rom the file `mu
-0000d410: 6c74 6970 6c69 636f 6e5f 7061 6972 732e  ltiplicon_pairs.
-0000d420: 7478 7460 2077 6869 6368 2063 6f6e 7461  txt` which conta
-0000d430: 696e 7320 7468 6520 6675 6c6c 2073 6574  ins the full set
-0000d440: 206f 6620 7379 6e74 656c 6f67 7329 2c20   of syntelogs), 
-0000d450: 696e 7374 6561 6420 6f66 2074 6865 2073  instead of the s
-0000d460: 6d61 6c6c 6572 2067 656e 6520 7365 7420  maller gene set 
-0000d470: 6f66 2061 6e63 686f 7220 7061 6972 732e  of anchor pairs.
-0000d480: 2054 6865 2064 6973 7469 6e63 7469 6f6e   The distinction
-0000d490: 2062 6574 7765 656e 2061 6e63 686f 7220   between anchor 
-0000d4a0: 7061 6972 7320 616e 6420 7379 6e74 656c  pairs and syntel
-0000d4b0: 6f67 7320 6973 2074 6861 7420 7468 6520  ogs is that the 
-0000d4c0: 6c61 7474 6572 2072 6566 6572 7320 746f  latter refers to
-0000d4d0: 206d 756c 7469 706c 6520 7365 7473 206f   multiple sets o
-0000d4e0: 6620 6765 6e65 7320 6465 7269 7665 6420  f genes derived 
-0000d4f0: 6672 6f6d 2074 6865 2073 616d 6520 616e  from the same an
-0000d500: 6365 7374 7261 6c20 6765 6e6f 6d69 6320  cestral genomic 
-0000d510: 7265 6769 6f6e 2077 6869 6c65 2074 6865  region while the
-0000d520: 2066 6f72 6d65 7220 696d 706c 6965 7320   former implies 
-0000d530: 7468 6520 6c61 7474 6572 2062 7574 2072  the latter but r
-0000d540: 6571 7569 7265 7320 6164 6469 7469 6f6e  equires addition
-0000d550: 616c 6c79 2074 6865 2063 6f6e 7365 7276  ally the conserv
-0000d560: 6564 2067 656e 6520 6f72 6465 722c 2062  ed gene order, b
-0000d570: 6f74 6820 6f66 2077 6869 6368 2061 7265  oth of which are
-0000d580: 2c20 7769 7468 696e 2061 2067 656e 6f6d  , within a genom
-0000d590: 652c 2061 7373 756d 6564 2074 6f20 6265  e, assumed to be
-0000d5a0: 206f 7269 6769 6e61 7465 6420 6672 6f6d   originated from
-0000d5b0: 2074 6865 2064 7570 6c69 6361 7469 6f6e   the duplication
-0000d5c0: 206f 6620 6120 636f 6d6d 6f6e 2061 6e63   of a common anc
-0000d5d0: 6573 7472 616c 2067 656e 6f6d 6963 2072  estral genomic r
-0000d5e0: 6567 696f 6e20 616e 6420 6173 2073 7563  egion and as suc
-0000d5f0: 6820 6465 656d 6564 2065 7669 6465 6e63  h deemed evidenc
-0000d600: 6520 666f 7220 5747 442e 2054 6865 6e20  e for WGD. Then 
-0000d610: 7468 6520 7379 6e74 656c 6f67 7320 7769  the syntelogs wi
-0000d620: 6c6c 2062 6520 6d61 7070 6564 2062 6163  ll be mapped bac
-0000d630: 6b20 6163 636f 7264 696e 6720 746f 2074  k according to t
-0000d640: 6865 2063 6c75 7374 6572 696e 6720 7265  he clustering re
-0000d650: 7375 6c74 7320 6f66 2074 6865 6972 2061  sults of their a
-0000d660: 6666 696c 6961 7465 6420 7365 676d 656e  ffiliated segmen
-0000d670: 7473 2e20 5369 6e63 6520 7468 6520 4761  ts. Since the Ga
-0000d680: 7573 7369 616e 2073 6861 7065 206f 6620  ussian shape of 
-0000d690: 7468 6520 7365 676d 656e 7420 636c 7573  the segment clus
-0000d6a0: 7465 7220 646f 6573 6e27 7420 6e65 6365  ter doesn't nece
-0000d6b0: 7373 6172 696c 7920 696d 6974 6174 6520  ssarily imitate 
-0000d6c0: 7468 6520 7368 6170 6520 6f66 2074 6865  the shape of the
-0000d6d0: 2072 6573 6964 696e 6720 7379 6e74 656c   residing syntel
-0000d6e0: 6f67 732c 2073 6f20 6173 2074 6f20 7265  ogs, so as to re
-0000d6f0: 7472 6965 7665 2074 6865 2022 7265 6c69  trieve the "reli
-0000d700: 6162 6c65 2220 6765 6e65 2073 6574 7320  able" gene sets 
-0000d710: 666f 7220 7068 796c 6f67 656e 6574 6963  for phylogenetic
-0000d720: 2064 6174 696e 672c 2077 6520 6164 706f   dating, we adpo
-0000d730: 7465 6420 7468 6520 2839 3525 2920 6869  ted the (95%) hi
-0000d740: 6768 6573 7420 6465 6e73 6974 7920 7265  ghest density re
-0000d750: 6769 6f6e 2028 4844 5229 206f 6620 7468  gion (HDR) of th
-0000d760: 6520 7379 6e74 656c 6f67 202a 4b2a 3c73  e syntelog *K*<s
-0000d770: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000d780: 6973 7472 6962 7574 696f 6e20 666f 7220  istribution for 
-0000d790: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
-0000d7a0: 2064 6174 696e 672c 2074 6865 2063 616c   dating, the cal
-0000d7b0: 6375 6c61 7469 6f6e 206f 6620 7468 6520  culation of the 
-0000d7c0: 2839 3525 2920 4844 5220 696e 2074 6865  (95%) HDR in the
-0000d7d0: 2066 756e 6374 696f 6e20 5b63 616c 6375   function [calcu
-0000d7e0: 6c61 7465 4850 445d 2868 7474 7073 3a2f  lateHPD](https:/
-0000d7f0: 2f67 6974 6875 622e 636f 6d2f 6865 6368  /github.com/hech
-0000d800: 652d 7073 622f 7767 642f 626c 6f62 2f70  e-psb/wgd/blob/p
-0000d810: 6879 6c6f 6461 7469 6e67 2f77 6764 2f70  hylodating/wgd/p
-0000d820: 6561 6b2e 7079 234c 3137 3537 2920 7365  eak.py#L1757) se
-0000d830: 656b 7320 7468 6520 7368 6f72 7465 7374  eks the shortest
-0000d840: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000d850: 2072 616e 6765 2028 612c 6229 2077 6869   range (a,b) whi
-0000d860: 6368 2073 6174 6973 6669 6573 2074 6865  ch satisfies the
-0000d870: 2072 6571 7569 7265 6d65 6e74 206f 6620   requirement of 
-0000d880: 7370 616e 6e69 6e67 206d 6f72 6520 7468  spanning more th
-0000d890: 616e 2028 3935 2529 206f 6620 616c 6c20  an (95%) of all 
-0000d8a0: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-0000d8b0: 7562 3e20 7661 6c75 6573 2e20 4f6e 2074  ub> values. On t
-0000d8c0: 6865 2070 7265 6d69 7365 206f 6620 6964  he premise of id
-0000d8d0: 656e 7469 6669 6564 2061 6e63 686f 7220  entified anchor 
-0000d8e0: 7061 6972 7320 286e 6f74 6520 7468 6174  pairs (note that
-0000d8f0: 2074 6865 2073 796e 7465 6c6f 6773 2061   the syntelogs a
-0000d900: 7265 2061 6c73 6f20 7265 6665 7272 6564  re also referred
-0000d910: 2061 7320 616e 6368 6f72 2070 6169 7273   as anchor pairs
-0000d920: 2068 6572 6561 6674 6572 292c 2077 6520   hereafter), we 
-0000d930: 696d 706c 656d 656e 7465 6420 696e 2074  implemented in t
-0000d940: 6865 2070 726f 6772 616d 2060 7767 6420  he program `wgd 
-0000d950: 646d 6460 2074 6865 2073 6f2d 6361 6c6c  dmd` the so-call
-0000d960: 6564 2061 6e63 686f 722d 6177 6172 6520  ed anchor-aware 
-0000d970: 6c6f 6361 6c20 4d52 4248 7320 6f72 206f  local MRBHs or o
-0000d980: 7274 686f 6772 6f75 7073 2c20 696e 2077  rthogroups, in w
-0000d990: 6869 6368 2074 6865 206f 7269 6769 6e61  hich the origina
-0000d9a0: 6c20 6c6f 6361 6c20 4d52 4248 7320 6172  l local MRBHs ar
-0000d9b0: 6520 6675 7274 6865 7220 6d65 7267 6564  e further merged
-0000d9c0: 2077 6974 6820 616e 6368 6f72 2070 6169   with anchor pai
-0000d9d0: 7273 2073 7563 6820 7468 6174 2065 6163  rs such that eac
-0000d9e0: 6820 6f72 7468 6f67 726f 7570 2063 6f6e  h orthogroup con
-0000d9f0: 7461 696e 7320 7468 6520 616e 6368 6f72  tains the anchor
-0000da00: 2070 6169 7220 616e 6420 7468 6520 6f72   pair and the or
-0000da10: 7468 6f6c 6f67 7565 732e 2057 6974 6820  thologues. With 
-0000da20: 7468 6973 206f 7274 686f 6772 6f75 702c  this orthogroup,
-0000da30: 2075 7365 7273 2074 6865 6e20 6e65 6564   users then need
-0000da40: 206f 6e65 2073 7461 7274 696e 6720 7472   one starting tr
-0000da50: 6565 2066 696c 6520 2861 7320 7368 6f77  ee file (as show
-0000da60: 6e20 696e 2074 6865 205b 496c 6c75 7374  n in the [Illust
-0000da70: 7261 7469 6f6e 5d28 6874 7470 733a 2f2f  ration](https://
-0000da80: 6769 7468 7562 2e63 6f6d 2f68 6563 6865  github.com/heche
-0000da90: 2d70 7362 2f77 6764 2f74 7265 652f 7068  -psb/wgd/tree/ph
-0000daa0: 796c 6f64 6174 696e 673f 7461 623d 7265  ylodating?tab=re
-0000dab0: 6164 6d65 2d6f 762d 6669 6c65 2369 6c6c  adme-ov-file#ill
-0000dac0: 7573 7472 6174 696f 6e29 2070 6172 7429  ustration) part)
-0000dad0: 2069 6e64 6963 6174 696e 6720 7468 6520   indicating the 
-0000dae0: 7472 6565 2074 6f70 6f6c 6f67 7920 616e  tree topology an
-0000daf0: 6420 666f 7373 696c 2063 616c 6962 7261  d fossil calibra
-0000db00: 7469 6f6e 2069 6e66 6f72 6d61 7469 6f6e  tion information
-0000db10: 2066 6f72 2074 6865 2066 696e 616c 2070   for the final p
-0000db20: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
-0000db30: 6e67 2075 7369 6e67 2074 6865 2070 726f  ng using the pro
-0000db40: 6772 616d 2060 7767 6420 666f 6375 7360  gram `wgd focus`
-0000db50: 2e20 5573 6572 7320 6361 6e20 7365 7420  . Users can set 
-0000db60: 7468 6520 666c 6167 206f 7074 696f 6e20  the flag option 
-0000db70: 602d 2d70 726f 7463 6f63 6461 7469 6e67  `--protcocdating
-0000db80: 6020 746f 206f 6e6c 7920 636f 6e64 7563  ` to only conduc
-0000db90: 7420 7468 6520 6461 7469 6e67 206f 6620  t the dating of 
-0000dba0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
-0000dbb0: 6e20 7072 6f74 6569 6e20 4d53 4120 6f72  n protein MSA or
-0000dbc0: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
-0000dbd0: 2060 2d2d 7072 6f74 6461 7469 6e67 6020   `--protdating` 
-0000dbe0: 746f 206f 6e6c 7920 636f 6e64 7563 7420  to only conduct 
-0000dbf0: 7468 6520 6461 7469 6e67 206f 6620 7468  the dating of th
-0000dc00: 6520 7072 6f74 6569 6e20 4d53 4173 2c20  e protein MSAs, 
-0000dc10: 6e6f 7465 6420 7468 6174 2074 6865 7365  noted that these
-0000dc20: 2074 776f 206f 7074 696f 6e73 206f 6e6c   two options onl
-0000dc30: 7920 776f 726b 2066 6f72 2074 6865 2060  y work for the `
-0000dc40: 6d63 6d63 7472 6565 6020 6f70 7469 6f6e  mcmctree` option
-0000dc50: 2073 6f20 6661 722e 2054 6865 2066 6c61   so far. The fla
-0000dc60: 6720 6f70 7469 6f6e 2060 2d2d 7061 7274  g option `--part
-0000dc70: 6974 696f 6e60 2063 616e 2062 6520 7365  ition` can be se
-0000dc80: 7420 746f 2070 6572 666f 726d 2060 6d63  t to perform `mc
-0000dc90: 6d63 7472 6565 6020 616e 616c 7973 6973  mctree` analysis
-0000dca0: 2075 7369 6e67 2074 6865 2070 6172 7469   using the parti
-0000dcb0: 7469 6f6e 6564 2064 6174 6120 2869 2e65  tioned data (i.e
-0000dcc0: 2e2c 2031 7374 2c20 326e 6420 616e 6420  ., 1st, 2nd and 
-0000dcd0: 3372 6420 706f 7369 7469 6f6e 206f 6620  3rd position of 
-0000dce0: 636f 646f 6e29 2069 6e73 7465 6164 206f  codon) instead o
-0000dcf0: 6620 7573 696e 6720 7468 6520 636f 646f  f using the codo
-0000dd00: 6e20 6173 2061 2077 686f 6c65 2e20 5468  n as a whole. Th
-0000dd10: 6520 6f70 7469 6f6e 2060 2d2d 6161 6d6f  e option `--aamo
-0000dd20: 6465 6c60 2063 616e 2062 6520 7365 7420  del` can be set 
-0000dd30: 746f 2064 6574 6572 6d69 6e65 2074 6865  to determine the
-0000dd40: 2061 6d69 6e6f 2061 6369 6420 6d6f 6465   amino acid mode
-0000dd50: 6c20 6170 706c 6965 6420 696e 2060 6d63  l applied in `mc
-0000dd60: 6d63 7472 6565 6020 616e 616c 7973 6973  mctree` analysis
-0000dd70: 2e20 5468 6520 6f70 7469 6f6e 2060 2d64  . The option `-d
-0000dd80: 7360 2063 616e 2062 6520 7573 6564 2074  s` can be used t
-0000dd90: 6f20 7365 7420 7061 7261 6d65 7465 7273  o set parameters
-0000dda0: 2066 6f72 2074 6865 206d 6f6c 6563 756c   for the molecul
-0000ddb0: 6172 2064 6174 696e 6720 7072 6f67 7261  ar dating progra
-0000ddc0: 6d2e 2054 6865 2070 6172 616c 6c65 6c20  m. The parallel 
-0000ddd0: 7468 7265 6164 7320 6865 7265 2061 7265  threads here are
-0000dde0: 2074 6f20 7061 7261 6c6c 656c 697a 6520   to parallelize 
-0000ddf0: 7468 6520 616e 616c 7973 6973 2066 6f72  the analysis for
-0000de00: 2065 6163 6820 6765 6e65 2066 616d 696c   each gene famil
-0000de10: 7920 616e 6420 7468 7573 2073 7567 6765  y and thus sugge
-0000de20: 7374 6564 2074 6f20 6265 2073 6574 2061  sted to be set a
-0000de30: 7320 6d75 6368 2061 7320 7468 6520 6e75  s much as the nu
-0000de40: 6d62 6572 206f 6620 6765 6e65 2066 616d  mber of gene fam
-0000de50: 696c 6965 732e 0a0a 2a2a 4120 7375 6767  ilies...**A sugg
-0000de60: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-0000de70: 756e 2063 616e 2062 6520 7769 7468 2074  un can be with t
-0000de80: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-0000de90: 2e2a 2a0a 0a60 6060 0a77 6764 2066 6f63  .**..```.wgd foc
-0000dea0: 7573 2066 616d 696c 6965 7320 7365 7175  us families sequ
-0000deb0: 656e 6365 3120 7365 7175 656e 6365 3220  ence1 sequence2 
-0000dec0: 7365 7175 656e 6365 3320 2d64 206d 636d  sequence3 -d mcm
-0000ded0: 6374 7265 6520 2d73 7020 7370 6461 7461  ctree -sp spdata
-0000dee0: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
-0000def0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000df00: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000df10: 3320 2d64 2072 3873 202d 7370 2073 7064  3 -d r8s -sp spd
-0000df20: 6174 6120 2d2d 6e73 6974 6573 206e 7369  ata --nsites nsi
-0000df30: 7465 696e 666f 0a77 6764 2066 6f63 7573  teinfo.wgd focus
-0000df40: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000df50: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
-0000df60: 7175 656e 6365 3320 2d64 2062 6561 7374  quence3 -d beast
-0000df70: 202d 2d66 6f73 7369 6c20 666f 7373 696c   --fossil fossil
-0000df80: 696e 666f 202d 2d72 6f6f 7468 6569 6768  info --rootheigh
-0000df90: 7420 726f 6f74 6865 6967 6874 696e 666f  t rootheightinfo
-0000dfa0: 202d 2d63 6861 696e 7365 7420 6368 6169   --chainset chai
-0000dfb0: 6e73 6574 696e 666f 202d 2d62 6561 7374  nsetinfo --beast
-0000dfc0: 6c67 6a61 7220 2450 4154 480a 6060 600a  lgjar $PATH.```.
-0000dfd0: 0a23 2323 2077 6764 206b 7364 0a0a 2a2a  .### wgd ksd..**
-0000dfe0: 5468 6520 636f 6e73 7472 7563 7469 6f6e  The construction
-0000dff0: 206f 6620 7768 6f6c 6520 7061 7261 6e6f   of whole parano
-0000e000: 6d65 202a 4b2a 3c73 7562 3e53 3c2f 7375  me *K*<sub>S</su
-0000e010: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000e020: 696f 6e2a 2a0a 6060 600a 7767 6420 6b73  ion**.```.wgd ks
-0000e030: 6420 6661 6d69 6c69 6573 2073 6571 7565  d families seque
-0000e040: 6e63 6520 282d 6f20 7767 645f 6b73 6420  nce (-o wgd_ksd 
-0000e050: 2d74 2077 6764 5f6b 7364 5f74 6d70 202d  -t wgd_ksd_tmp -
-0000e060: 2d6e 7468 7265 6164 7320 3420 2d2d 746f  -nthreads 4 --to
-0000e070: 5f73 746f 7020 2d2d 6364 7320 2d2d 7061  _stop --cds --pa
-0000e080: 6972 7769 7365 202d 2d73 7472 6970 5f67  irwise --strip_g
-0000e090: 6170 7320 2d2d 616c 6967 6e65 7220 6d61  aps --aligner ma
-0000e0a0: 6666 7420 2d2d 7472 6565 5f6d 6574 686f  fft --tree_metho
-0000e0b0: 6420 6661 7374 7472 6565 202d 2d6e 6f64  d fasttree --nod
-0000e0c0: 655f 6176 6572 6167 6529 0a60 6060 0a0a  e_average).```..
-0000e0d0: 5468 6520 7072 6f67 7261 6d20 6077 6764  The program `wgd
-0000e0e0: 206b 7364 602c 2061 7320 696d 7069 6564   ksd`, as impied
-0000e0f0: 2062 7920 6974 7320 6e61 6d65 2c20 6973   by its name, is
-0000e100: 2066 6f72 2074 6865 2063 6f6e 7374 7275   for the constru
-0000e110: 6374 696f 6e20 6f66 202a 4b2a 3c73 7562  ction of *K*<sub
-0000e120: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-0000e130: 7472 6962 7574 696f 6e2e 2045 7863 6570  tribution. Excep
-0000e140: 7420 666f 7220 7468 6520 6166 6f72 656d  t for the aforem
-0000e150: 656e 7469 6f6e 6564 2070 6172 616d 6574  entioned paramet
-0000e160: 6572 7320 7375 6368 2061 7320 602d 2d74  ers such as `--t
-0000e170: 6f5f 7374 6f70 6020 616e 6420 602d 2d63  o_stop` and `--c
-0000e180: 6473 602c 2074 6865 7265 2061 7265 2073  ds`, there are s
-0000e190: 6f6d 6520 696d 706f 7274 616e 7420 7061  ome important pa
-0000e1a0: 7261 6d65 7465 7273 2074 6861 7420 6861  rameters that ha
-0000e1b0: 7665 2063 7275 6369 616c 2069 6d70 6163  ve crucial impac
-0000e1c0: 7420 6f6e 2074 6865 202a 4b2a 3c73 7562  t on the *K*<sub
-0000e1d0: 3e53 3c2f 7375 623e 2065 7374 696d 6174  >S</sub> estimat
-0000e1e0: 696f 6e2e 2054 6865 206f 7074 696f 6e20  ion. The option 
-0000e1f0: 602d 2d70 6169 7277 6973 6560 2069 7320  `--pairwise` is 
-0000e200: 6120 7665 7279 2069 6d70 6f72 7461 6e74  a very important
-0000e210: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-0000e220: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
-0000e230: 623e 2065 7374 696d 6174 696f 6e2c 2077  b> estimation, w
-0000e240: 6974 6820 7768 6963 6820 7468 6520 6043  ith which the `C
-0000e250: 4f44 454d 4c60 2077 696c 6c20 6361 6c63  ODEML` will calc
-0000e260: 756c 6174 6520 7468 6520 2a4b 2a3c 7375  ulate the *K*<su
-0000e270: 623e 533c 2f73 7562 3e20 666f 7220 6561  b>S</sub> for ea
-0000e280: 6368 2067 656e 6520 7061 6972 2073 6570  ch gene pair sep
-0000e290: 6172 6174 656c 7920 6261 7365 6420 6f6e  arately based on
-0000e2a0: 2074 6865 2061 6c69 676e 6d65 6e74 206f   the alignment o
-0000e2b0: 6620 6f6e 6c79 2074 6865 7365 2074 776f  f only these two
-0000e2c0: 2067 656e 6573 2069 6e73 7465 6164 206f   genes instead o
-0000e2d0: 6620 7468 6520 7768 6f6c 6520 616c 6967  f the whole alig
-0000e2e0: 6e6d 656e 7420 6f66 2074 6865 2066 616d  nment of the fam
-0000e2f0: 696c 792c 2073 7563 6820 7468 6174 206c  ily, such that l
-0000e300: 6573 7320 6761 7073 2061 7265 2065 7870  ess gaps are exp
-0000e310: 6563 7465 6420 616e 6420 7468 7573 2074  ected and thus t
-0000e320: 6865 2061 6c69 676e 6d65 6e74 2069 6e20  he alignment in 
-0000e330: 7468 6520 636f 6e73 6964 6572 6174 696f  the consideratio
-0000e340: 6e20 6f66 2060 434f 4445 4d4c 6020 7769  n of `CODEML` wi
-0000e350: 6c6c 2062 6520 6c6f 6e67 6572 2028 6265  ll be longer (be
-0000e360: 6361 7573 6520 6043 4f44 454d 4c60 2077  cause `CODEML` w
-0000e370: 696c 6c20 6175 746f 6d61 7465 6c79 2073  ill automately s
-0000e380: 6b69 7020 6576 6572 7920 636f 6c75 6d6e  kip every column
-0000e390: 2077 6974 6820 6761 702c 2072 6567 6172   with gap, regar
-0000e3a0: 646c 6573 7320 6f66 2077 6865 7468 6572  dless of whether
-0000e3b0: 2069 7427 7320 616e 206f 7665 7261 6c6c   it's an overall
-0000e3c0: 206f 7220 7061 7274 6961 6c20 6761 7029   or partial gap)
-0000e3d0: 2c20 7769 7468 6f75 7420 7768 6963 6820  , without which 
-0000e3e0: 7468 6520 6043 4f44 454d 4c60 2077 696c  the `CODEML` wil
-0000e3f0: 6c20 6361 6c63 756c 6174 6520 7468 6520  l calculate the 
-0000e400: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e410: 6261 7365 6420 6f6e 2074 6865 2077 686f  based on the who
-0000e420: 6c65 2061 6c69 676e 6d65 6e74 206f 6620  le alignment of 
-0000e430: 7468 6520 6661 6d69 6c79 2c20 7768 6963  the family, whic
-0000e440: 6820 6d69 6768 7420 6861 7665 206e 6f20  h might have no 
-0000e450: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000e460: 7265 7375 6c74 2061 7420 616c 6c20 6966  result at all if
-0000e470: 2074 6865 2073 7472 6970 7065 6420 616c   the stripped al
-0000e480: 6967 6e6d 656e 7420 6c65 6e67 7468 2028  ignment length (
-0000e490: 7265 6d6f 7669 6e67 2061 6c6c 2067 6170  removing all gap
-0000e4a0: 2d63 6f6e 7461 696e 696e 6720 636f 6c75  -containing colu
-0000e4b0: 6d6e 7329 2077 6173 207a 6572 6f2c 2061  mns) was zero, a
-0000e4c0: 2063 6175 7365 206f 6620 6469 6666 6572   cause of differ
-0000e4d0: 656e 7420 6e75 6d62 6572 206f 6620 2a4b  ent number of *K
-0000e4e0: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
-0000e4f0: 7469 6d61 7465 7320 6265 7477 6565 6e20  timates between 
-0000e500: 2270 6169 7277 6973 6520 6d6f 6465 2220  "pairwise mode" 
-0000e510: 616e 6420 226e 6f6e 2d70 6169 7277 6973  and "non-pairwis
-0000e520: 6520 6d6f 6465 222e 2049 7427 7320 6469  e mode". It's di
-0000e530: 6666 6963 756c 7420 746f 2073 6179 2077  fficult to say w
-0000e540: 6869 6368 206d 6f64 6520 6973 206d 6f72  hich mode is mor
-0000e550: 6520 6964 6561 6c2c 2061 6c74 686f 7567  e ideal, althoug
-0000e560: 6820 7468 6520 226e 6f6e 2d70 6169 7277  h the "non-pairw
-0000e570: 6973 6520 6d6f 6465 2220 2864 6566 6175  ise mode" (defau
-0000e580: 6c74 2073 6574 7469 6e67 2920 7768 6963  lt setting) whic
-0000e590: 6820 7275 6e73 206f 6e20 7468 6520 7768  h runs on the wh
-0000e5a0: 6f6c 6520 616c 6967 6e6d 656e 7420 696e  ole alignment in
-0000e5b0: 7374 6561 6420 6f66 2061 206c 6f63 616c  stead of a local
-0000e5c0: 2061 6c69 676e 6d65 6e74 2c20 6d69 6768   alignment, migh
-0000e5d0: 7420 6265 206d 6f72 6520 6269 6f6c 6f67  t be more biolog
-0000e5e0: 6963 616c 6c79 2063 6f6e 7365 7276 6564  ically conserved
-0000e5f0: 2069 6e20 7468 6174 2069 7420 6173 7375   in that it assu
-0000e600: 7265 7320 7468 6520 6576 6f6c 7574 696f  res the evolutio
-0000e610: 6e20 6f66 2065 6163 6820 636f 6c75 6d6e  n of each column
-0000e620: 2074 6f20 6265 2073 7461 7274 6564 2066   to be started f
-0000e630: 726f 6d20 7468 6520 726f 6f74 206f 6620  rom the root of 
-0000e640: 7468 6520 6661 6d69 6c79 2061 6e64 2061  the family and a
-0000e650: 6c6c 2074 6865 2067 656e 6520 6475 706c  ll the gene dupl
-0000e660: 6963 6174 6573 2061 7265 2074 616b 656e  icates are taken
-0000e670: 2069 6e74 6f20 6163 636f 756e 7420 696e   into account in
-0000e680: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-0000e690: 7375 623e 2065 7374 696d 6174 696f 6e20  sub> estimation 
-0000e6a0: 7072 6f63 6573 732e 2054 6865 206f 7074  process. The opt
-0000e6b0: 696f 6e20 602d 2d73 7472 6970 5f67 6170  ion `--strip_gap
-0000e6c0: 7360 2063 616e 2072 656d 6f76 6520 616c  s` can remove al
-0000e6d0: 6c20 7468 6520 6761 702d 636f 6e74 6169  l the gap-contai
-0000e6e0: 6e69 6e67 2063 6f6c 756d 6e73 2c20 7769  ning columns, wi
-0000e6f0: 7468 206f 7220 7769 7468 6f75 7420 7768  th or without wh
-0000e700: 6963 6820 7468 6520 7265 7375 6c74 206f  ich the result o
-0000e710: 6620 226e 6f6e 2d70 6169 7277 6973 6520  f "non-pairwise 
-0000e720: 6d6f 6465 2220 776f 6e27 7420 6265 2061  mode" won't be a
-0000e730: 6666 6563 7465 642c 2077 6869 6c65 2077  ffected, while w
-0000e740: 6974 6820 7768 6963 6820 7468 6520 7265  ith which the re
-0000e750: 7375 6c74 206f 6620 2270 6169 7277 6973  sult of "pairwis
-0000e760: 6520 6d6f 6465 2220 7769 6c6c 2062 6520  e mode" will be 
-0000e770: 616c 7465 7265 642e 2054 6865 206f 7074  altered. The opt
-0000e780: 696f 6e20 602d 2d61 6c69 676e 6572 6020  ion `--aligner` 
-0000e790: 616e 6420 602d 2d61 6c6e 5f6f 7074 696f  and `--aln_optio
-0000e7a0: 6e73 6020 7768 6963 6820 6465 6369 6465  ns` which decide
-0000e7b0: 2077 6869 6368 2061 6c69 676e 6d65 6e74   which alignment
-0000e7c0: 2070 726f 6772 616d 2074 6f20 6265 2075   program to be u
-0000e7d0: 7365 6420 616e 6420 7768 6963 6820 7061  sed and which pa
-0000e7e0: 7261 6d65 7465 7220 746f 2062 6520 7365  rameter to be se
-0000e7f0: 742c 2077 696c 6c20 6861 7665 2069 6d70  t, will have imp
-0000e800: 6163 7420 6f6e 2074 6865 202a 4b2a 3c73  act on the *K*<s
-0000e810: 7562 3e53 3c2f 7375 623e 2072 6573 756c  ub>S</sub> resul
-0000e820: 7473 2c20 6e6f 7465 6420 7468 6174 2074  ts, noted that t
-0000e830: 6865 2064 6566 6175 6c74 2070 726f 6772  he default progr
-0000e840: 616d 2069 7320 606d 6166 6674 6020 616e  am is `mafft` an
-0000e850: 6420 7468 6520 7061 7261 6d65 7465 7220  d the parameter 
-0000e860: 6973 2060 2d2d 6175 746f 602e 2054 6865  is `--auto`. The
-0000e870: 206f 7074 696f 6e20 602d 2d74 7265 655f   option `--tree_
-0000e880: 6d65 7468 6f64 6020 616e 6420 602d 2d74  method` and `--t
-0000e890: 7265 655f 6f70 7469 6f6e 7360 2064 6563  ree_options` dec
-0000e8a0: 6964 6520 7768 6963 6820 6765 6e65 2074  ide which gene t
-0000e8b0: 7265 6520 696e 6665 7265 6e63 6520 7072  ree inference pr
-0000e8c0: 6f67 7261 6d20 746f 2062 6520 7573 6564  ogram to be used
-0000e8d0: 2061 6e64 2077 6869 6368 2070 6172 616d   and which param
-0000e8e0: 6574 6572 2074 6f20 6265 2073 6574 2c20  eter to be set, 
-0000e8f0: 776f 6e27 7420 6166 6665 6374 2074 6865  won't affect the
-0000e900: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000e910: 2065 7374 696d 6174 696f 6e20 6974 7365   estimation itse
-0000e920: 6c66 2062 7574 2074 6865 2072 6573 756c  lf but the resul
-0000e930: 7420 6f66 2064 652d 7265 6475 6e64 616e  t of de-redundan
-0000e940: 6379 2c20 6e6f 7465 6420 7468 6174 2077  cy, noted that w
-0000e950: 6520 696d 706c 656d 656e 7465 6420 6120  e implemented a 
-0000e960: 6275 696c 742d 696e 2067 656e 6520 7472  built-in gene tr
-0000e970: 6565 2069 6e66 6572 656e 6365 206d 6574  ee inference met
-0000e980: 686f 6420 6261 7365 6420 6f6e 2074 6865  hod based on the
-0000e990: 2041 7665 7261 6765 204c 696e 6b61 6765   Average Linkage
-0000e9a0: 2043 6c75 7374 6572 696e 6720 2841 4c43   Clustering (ALC
-0000e9b0: 2920 2874 6875 7320 6120 6469 7374 616e  ) (thus a distan
-0000e9c0: 6365 2d62 6173 6564 2074 7265 6529 2077  ce-based tree) w
-0000e9d0: 6974 6820 7468 6520 602d 2d74 7265 655f  ith the `--tree_
-0000e9e0: 6d65 7468 6f64 6020 7365 7420 6173 2022  method` set as "
-0000e9f0: 636c 7573 7465 7222 2e20 5477 6f20 6d65  cluster". Two me
-0000ea00: 7468 6f64 7320 6f66 2064 652d 7265 6475  thods of de-redu
-0000ea10: 6e64 616e 6379 2061 7265 2069 6d70 6c65  ndancy are imple
-0000ea20: 6d65 6e74 6564 2069 6e20 6077 6764 2076  mented in `wgd v
-0000ea30: 3260 2c20 6e61 6d65 6c79 206e 6f64 652d  2`, namely node-
-0000ea40: 7765 6967 6874 6564 2061 6e64 206e 6f64  weighted and nod
-0000ea50: 652d 6176 6572 6167 6564 206d 6574 686f  e-averaged metho
-0000ea60: 6473 2e20 5468 6520 6e6f 6465 2d77 6569  ds. The node-wei
-0000ea70: 6768 7465 6420 6d65 7468 6f64 2061 6368  ghted method ach
-0000ea80: 6965 7665 7320 7468 6520 6465 2d72 6564  ieves the de-red
-0000ea90: 756e 6461 6e63 7920 7669 6120 7765 6967  undancy via weig
-0000eaa0: 6869 6e67 2074 6865 202a 4b2a 3c73 7562  hing the *K*<sub
-0000eab0: 3e53 3c2f 7375 623e 2076 616c 7565 2061  >S</sub> value a
-0000eac0: 7373 6f63 6961 7465 6420 7769 7468 2065  ssociated with e
-0000ead0: 6163 6820 6765 6e65 2070 6169 7220 7375  ach gene pair su
-0000eae0: 6368 2074 6861 7420 7468 6520 7765 6967  ch that the weig
-0000eaf0: 6874 206f 6620 6120 7369 6e67 6c65 2064  ht of a single d
-0000eb00: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
-0000eb10: 2073 756d 7320 7570 2074 6f20 3120 286e   sums up to 1 (n
-0000eb20: 6f74 6564 2074 6861 7420 7468 6520 6e75  oted that the nu
-0000eb30: 6d62 6572 206f 6620 2a4b 2a3c 7375 623e  mber of *K*<sub>
-0000eb40: 533c 2f73 7562 3e20 6573 7469 6d61 7469  S</sub> estimati
-0000eb50: 6f6e 2072 656d 6169 6e73 2074 6865 2073  on remains the s
-0000eb60: 616d 6529 2077 6869 6c65 2074 6865 206e  ame) while the n
-0000eb70: 6f64 652d 6176 6572 6167 6564 206d 6574  ode-averaged met
-0000eb80: 686f 6420 7265 616c 697a 6573 2074 6865  hod realizes the
-0000eb90: 2064 652d 7265 6475 6e64 616e 6379 2076   de-redundancy v
-0000eba0: 6961 2063 616c 6375 6c61 7469 6e67 2070  ia calculating p
-0000ebb0: 6572 2067 656e 6520 7472 6565 206e 6f64  er gene tree nod
-0000ebc0: 6520 6f6e 6520 6176 6572 6167 6564 202a  e one averaged *
-0000ebd0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2076  K*<sub>S</sub> v
-0000ebe0: 616c 7565 2074 6f20 7265 7072 6573 656e  alue to represen
-0000ebf0: 7420 7468 6520 6167 6520 6f66 2065 6163  t the age of eac
-0000ec00: 6820 6475 706c 6963 6174 696f 6e20 6576  h duplication ev
-0000ec10: 656e 742e 2054 6865 206f 7074 696f 6e20  ent. The option 
-0000ec20: 602d 2d6e 6f64 655f 6176 6572 6167 6560  `--node_average`
-0000ec30: 2063 616e 2062 6520 7365 7420 746f 2063   can be set to c
-0000ec40: 686f 6f73 6520 7468 6520 6e6f 6465 2d61  hoose the node-a
-0000ec50: 7665 7261 6765 6420 7761 7920 6f66 2064  veraged way of d
-0000ec60: 652d 7265 6475 6e64 616e 6379 2e20 4469  e-redundancy. Di
-0000ec70: 6666 6572 656e 7420 6d65 7468 6f64 7320  fferent methods 
-0000ec80: 6f66 2064 652d 7265 6475 6e64 616e 6379  of de-redundancy
-0000ec90: 2077 696c 6c20 6861 7665 2069 6d70 6163   will have impac
-0000eca0: 7420 6f6e 2074 6865 2064 6574 6563 7469  t on the detecti
-0000ecb0: 6f6e 206f 6620 5747 4420 7369 676e 616c  on of WGD signal
-0000ecc0: 732c 2077 6869 6368 2068 6173 2062 6565  s, which has bee
-0000ecd0: 6e20 696e 7665 7374 6967 6174 6564 2069  n investigated i
-0000ece0: 6e20 7468 6973 205b 6c69 7465 7261 7475  n this [literatu
-0000ecf0: 7265 5d28 6874 7470 733a 2f2f 646f 692e  re](https://doi.
-0000ed00: 6f72 672f 3130 2e31 3039 332f 6762 652f  org/10.1093/gbe/
-0000ed10: 6576 7932 3030 292e 2054 6865 2070 6172  evy200). The par
-0000ed20: 616c 6c65 6c20 7468 7265 6164 7320 6865  allel threads he
-0000ed30: 7265 2061 7265 2074 6f20 7061 7261 6c6c  re are to parall
-0000ed40: 656c 697a 6520 7468 6520 616e 616c 7973  elize the analys
-0000ed50: 6973 2066 6f72 2065 6163 6820 6765 6e65  is for each gene
-0000ed60: 2066 616d 696c 7920 616e 6420 7468 7573   family and thus
-0000ed70: 2073 7567 6765 7374 6564 2074 6f20 6265   suggested to be
-0000ed80: 2073 6574 2061 7320 6d75 6368 2061 7320   set as much as 
-0000ed90: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
-0000eda0: 6e65 2066 616d 696c 6965 732e 0a0a 2a2a  ne families...**
-0000edb0: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-0000edc0: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-0000edd0: 2063 6f6d 6d61 6e64 2061 7320 6265 6c6f   command as belo
-0000ede0: 772a 2a0a 0a60 6060 0a77 6764 206b 7364  w**..```.wgd ksd
-0000edf0: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000ee00: 6365 0a60 6060 0a0a 2a2a 5468 6520 636f  ce.```..**The co
-0000ee10: 6e73 7472 7563 7469 6f6e 206f 6620 6f72  nstruction of or
-0000ee20: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-0000ee30: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-0000ee40: 7374 7269 6275 7469 6f6e 2a2a 0a60 6060  stribution**.```
-0000ee50: 0a77 6764 206b 7364 2066 616d 696c 6965  .wgd ksd familie
-0000ee60: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
-0000ee70: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
-0000ee80: 282d 2d72 6577 6569 6768 7429 0a60 6060  (--reweight).```
-0000ee90: 0a0a 4672 6f6d 2070 6172 616c 6f67 6f75  ..From paralogou
-0000eea0: 7320 746f 206f 7274 686f 6c6f 676f 7573  s to orthologous
-0000eeb0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-0000eec0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
-0000eed0: 6e2c 2075 7365 7273 206f 6e6c 7920 6e65  n, users only ne
-0000eee0: 6564 2074 6f20 7072 6f76 6964 6520 6d6f  ed to provide mo
-0000eef0: 7265 2063 6473 2066 696c 6573 2e20 4e6f  re cds files. No
-0000ef00: 7465 2074 6861 7420 7769 7468 206f 7274  te that with ort
-0000ef10: 686f 6c6f 676f 7573 2067 656e 6520 6661  hologous gene fa
-0000ef20: 6d69 6c69 6573 2074 6865 2077 6569 6768  milies the weigh
-0000ef30: 7469 6e67 206d 6574 686f 6420 6361 6e20  ting method can 
-0000ef40: 6265 2073 6574 2074 6f20 6265 2063 616c  be set to be cal
-0000ef50: 6375 6c61 7465 6420 7065 7220 7370 6563  culated per spec
-0000ef60: 6965 7320 7061 6972 2069 6e73 7465 6164  ies pair instead
-0000ef70: 206f 6620 636f 6e73 6964 6572 696e 6720   of considering 
-0000ef80: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
-0000ef90: 2062 6563 6175 7365 2077 6865 6e20 706c   because when pl
-0000efa0: 6f74 7469 6e67 206f 7274 686f 6c6f 676f  otting orthologo
-0000efb0: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
-0000efc0: 623e 2061 6765 2064 6973 7472 6962 7574  b> age distribut
-0000efd0: 696f 6e20 6265 7477 6565 6e20 7477 6f20  ion between two 
-0000efe0: 7370 6563 6965 7320 7468 6520 7765 6967  species the weig
-0000eff0: 6874 2063 616c 6375 6c61 7465 6420 6672  ht calculated fr
-0000f000: 6f6d 2074 6869 7320 7370 6563 6966 6963  om this specific
-0000f010: 2073 7065 6369 6573 2070 6169 7220 7368   species pair sh
-0000f020: 6f75 6c64 2062 6520 636f 6e73 6572 7665  ould be conserve
-0000f030: 6420 7768 696c 6520 7468 6520 6f6e 6520  d while the one 
-0000f040: 6361 6c63 756c 6174 6564 2066 726f 6d20  calculated from 
-0000f050: 7468 6520 7768 6f6c 6520 6661 6d69 6c79  the whole family
-0000f060: 2077 696c 6c20 7661 7279 2077 6974 6820   will vary with 
-0000f070: 7468 6520 6e75 6d62 6572 206f 6620 7370  the number of sp
-0000f080: 6563 6965 732e 2054 6f20 696e 6974 6961  ecies. To initia
-0000f090: 7465 2074 6865 2077 6569 6768 7469 6e67  te the weighting
-0000f0a0: 2070 6572 2073 7065 6369 6573 2070 6169   per species pai
-0000f0b0: 722c 2074 6865 206f 7074 696f 6e20 602d  r, the option `-
-0000f0c0: 2d72 6577 6569 6768 7460 2063 616e 2062  -reweight` can b
-0000f0d0: 6520 7365 742e 0a0a 2a2a 4120 7375 6767  e set...**A sugg
-0000f0e0: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-0000f0f0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-0000f100: 6e64 2061 7320 6265 6c6f 772a 2a0a 0a60  nd as below**..`
-0000f110: 6060 0a77 6764 206b 7364 2066 616d 696c  ``.wgd ksd famil
-0000f120: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
-0000f130: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
-0000f140: 330a 6060 600a 0a2a 2a54 6865 2063 6f6e  3.```..**The con
-0000f150: 7374 7275 6374 696f 6e20 6f66 202a 4b2a  struction of *K*
-0000f160: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
-0000f170: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
-0000f180: 7468 2072 6174 6520 636f 7272 6563 7469  th rate correcti
-0000f190: 6f6e 2a2a 0a60 6060 0a77 6764 206b 7364  on**.```.wgd ksd
-0000f1a0: 2066 616d 696c 6965 7320 7365 7175 656e   families sequen
-0000f1b0: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
-0000f1c0: 7175 656e 6365 3320 2d2d 666f 6375 7332  quence3 --focus2
-0000f1d0: 616c 6c20 7365 7175 656e 6365 3120 2d73  all sequence1 -s
-0000f1e0: 7020 7370 6461 7461 202d 2d65 7874 7261  p spdata --extra
-0000f1f0: 7061 7261 6e6f 6d65 6b73 2070 6172 616e  paranomeks paran
-0000f200: 6f6d 654b 7364 6174 6120 282d 2d70 6c6f  omeKsdata (--plo
-0000f210: 7465 6c6d 6d20 2d2d 706c 6f74 6170 676d  telmm --plotapgm
-0000f220: 6d20 2d2d 616e 6368 6f72 706f 696e 7473  m --anchorpoints
-0000f230: 2061 7064 6174 6120 2d2d 7265 7765 6967   apdata --reweig
-0000f240: 6874 202d 2d6f 6e6c 7972 6f6f 746f 7574  ht --onlyrootout
-0000f250: 290a 6060 600a 0a49 6e73 7069 7265 6420  ).```..Inspired 
-0000f260: 6279 2074 6865 2072 6174 6520 636f 7272  by the rate corr
-0000f270: 6563 7469 6f6e 2061 6c67 6f72 6974 686d  ection algorithm
-0000f280: 2069 6e20 5b6b 7372 6174 6573 5d28 6874   in [ksrates](ht
-0000f290: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-0000f2a0: 2f56 4942 2d50 5342 2f6b 7372 6174 6573  /VIB-PSB/ksrates
-0000f2b0: 292c 2077 6520 696d 706c 656d 656e 7465  ), we implemente
-0000f2c0: 6420 7468 6520 7261 7465 2063 6f72 7265  d the rate corre
-0000f2d0: 6374 696f 6e20 616e 616c 7973 6973 2061  ction analysis a
-0000f2e0: 6c73 6f20 696e 2060 7767 6420 7632 602c  lso in `wgd v2`,
-0000f2f0: 2077 6869 6368 2069 7320 6d6f 7374 6c79   which is mostly
-0000f300: 2074 6865 2073 616d 6520 6173 205b 6b73   the same as [ks
-0000f310: 7261 7465 735d 2868 7474 7073 3a2f 2f67  rates](https://g
-0000f320: 6974 6875 622e 636f 6d2f 5649 422d 5053  ithub.com/VIB-PS
-0000f330: 422f 6b73 7261 7465 7329 2062 7574 2064  B/ksrates) but d
-0000f340: 6966 6665 7273 2069 6e20 7468 6520 6361  iffers in the ca
-0000f350: 6c63 756c 6174 696f 6e20 6f66 2074 6865  lculation of the
-0000f360: 2073 7461 6e64 6172 6420 6465 7669 6174   standard deviat
-0000f370: 696f 6e20 6f66 2072 6573 6361 6c65 6420  ion of rescaled 
-0000f380: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000f390: 6167 6573 2e20 546f 2070 6572 666f 726d  ages. To perform
-0000f3a0: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
-0000f3b0: 7469 6f6e 2061 6e61 6c79 7369 732c 2075  tion analysis, u
-0000f3c0: 7365 7273 2063 616e 2075 7365 2062 6f74  sers can use bot
-0000f3d0: 6820 7468 6520 6077 6764 206b 7364 6020  h the `wgd ksd` 
-0000f3e0: 616e 6420 6077 6764 2076 697a 6020 7072  and `wgd viz` pr
-0000f3f0: 6f67 7261 6d2e 2046 6f72 2074 6865 2060  ogram. For the `
-0000f400: 7767 6420 6b73 6460 2070 726f 6772 616d  wgd ksd` program
-0000f410: 2c20 7573 6572 7320 6e65 6564 2074 6f20  , users need to 
-0000f420: 7072 6f76 6964 6520 6120 7370 6563 6965  provide a specie
-0000f430: 7320 7472 6565 2076 6961 2074 6865 206f  s tree via the o
-0000f440: 7074 696f 6e20 602d 2d73 7065 6369 6573  ption `--species
-0000f450: 7472 6565 6020 6f6e 2077 6869 6368 2072  tree` on which r
-0000f460: 6174 6520 636f 7272 6563 7469 6f6e 2063  ate correction c
-0000f470: 616e 2062 6520 636f 6e64 7563 7465 642e  an be conducted.
-0000f480: 204e 6f74 6520 7468 6174 2075 6e6e 6563   Note that unnec
-0000f490: 6573 7361 7279 2062 7261 636b 6574 7320  essary brackets 
-0000f4a0: 6d69 6768 7420 6c65 6164 2074 6f20 756e  might lead to un
-0000f4b0: 6578 7065 6374 6564 2065 7272 6f72 732c  expected errors,
-0000f4c0: 2066 6f72 2069 6e73 7461 6e63 6520 6120   for instance a 
-0000f4d0: 7472 6565 2060 2841 2c28 422c 4329 293b  tree `(A,(B,C));
-0000f4e0: 6020 7368 6f75 6c64 206e 6f74 2062 6520  ` should not be 
-0000f4f0: 7265 7072 6573 656e 7465 6420 6173 2060  represented as `
-0000f500: 2841 2c28 2842 2c43 2929 293b 602e 2054  (A,((B,C)));`. T
-0000f510: 6865 2073 6574 206f 6620 7370 6563 6965  he set of specie
-0000f520: 7320 7061 6972 7320 746f 2062 6520 7368  s pairs to be sh
-0000f530: 6f77 6e20 6973 2066 6c65 7869 626c 6520  own is flexible 
-0000f540: 7468 6174 2074 6865 206d 6f73 7420 636f  that the most co
-0000f550: 6e76 656e 6965 6e74 206f 7074 696f 6e20  nvenient option 
-0000f560: 6973 2060 2d2d 666f 6375 7332 616c 6c60  is `--focus2all`
-0000f570: 2077 6869 6368 2073 696d 706c 7920 7368   which simply sh
-0000f580: 6f77 7320 616c 6c20 7468 6520 706f 7373  ows all the poss
-0000f590: 6962 6c65 2066 6f63 616c 2d73 6973 7465  ible focal-siste
-0000f5a0: 7220 7370 6563 6965 7320 7061 6972 732c  r species pairs,
-0000f5b0: 206f 7220 7573 6572 7320 6361 6e20 6d61   or users can ma
-0000f5c0: 6e75 616c 6c79 2073 6574 2074 6865 2073  nually set the s
-0000f5d0: 7065 6369 6573 2070 6169 7273 2076 6961  pecies pairs via
-0000f5e0: 2074 6865 206f 7074 696f 6e20 602d 2d73   the option `--s
-0000f5f0: 7061 6972 602e 204e 6f74 6520 7468 6174  pair`. Note that
-0000f600: 2069 6620 7468 6520 7370 6563 6965 7320   if the species 
-0000f610: 7061 6972 7320 7765 7265 206d 616e 7561  pairs were manua
-0000f620: 6c6c 7920 7365 742c 2069 7420 776f 756c  lly set, it woul
-0000f630: 6420 6265 206e 6565 6465 6420 746f 2063  d be needed to c
-0000f640: 6f2d 7365 7420 7468 6520 6f70 7469 6f6e  o-set the option
-0000f650: 2060 2d2d 636c 6173 7369 6360 2e20 5468   `--classic`. Th
-0000f660: 6520 6f70 7469 6f6e 2060 2d2d 6f6e 6c79  e option `--only
-0000f670: 726f 6f74 6f75 7460 2063 616e 2062 6520  rootout` can be 
-0000f680: 7365 7420 746f 206f 6e6c 7920 636f 6e73  set to only cons
-0000f690: 6964 6572 206f 7574 6772 6f75 7020 6174  ider outgroup at
-0000f6a0: 2074 6865 2072 6f6f 742c 2069 6e73 7465   the root, inste
-0000f6b0: 6164 206f 6620 616c 6c20 7468 6520 706f  ad of all the po
-0000f6c0: 7373 6962 6c65 206f 7574 6772 6f75 7073  ssible outgroups
-0000f6d0: 2070 6572 2066 6f63 616c 2d73 6973 7465   per focal-siste
-0000f6e0: 7220 7370 6563 6965 7320 7061 6972 2c20  r species pair, 
-0000f6f0: 7768 6963 6820 6861 7320 696d 7061 6374  which has impact
-0000f700: 206f 6e20 7468 6520 6669 6e61 6c20 636f   on the final co
-0000f710: 7272 6563 7465 6420 2a4b 2a3c 7375 623e  rrected *K*<sub>
-0000f720: 533c 2f73 7562 3e20 6167 6573 2e20 5765  S</sub> ages. We
-0000f730: 2073 7567 6765 7374 206f 6620 7573 696e   suggest of usin
-0000f740: 6720 616c 6c20 7468 6520 706f 7373 6962  g all the possib
-0000f750: 6c65 206f 7574 6772 6f75 7073 2070 6572  le outgroups per
-0000f760: 2066 6f63 616c 2d73 6973 7465 7220 7370   focal-sister sp
-0000f770: 6563 6965 7320 7061 6972 2061 7320 666f  ecies pair as fo
-0000f780: 7220 6120 6c65 7373 2062 6961 7365 6420  r a less biased 
-0000f790: 7265 7375 6c74 2e20 5468 6520 7061 7261  result. The para
-0000f7a0: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
-0000f7b0: 7375 623e 2064 6174 6120 7368 6f75 6c64  sub> data should
-0000f7c0: 2062 6520 7072 6f76 6964 6564 2076 6961   be provided via
-0000f7d0: 2074 6865 206f 7074 696f 6e20 602d 2d65   the option `--e
-0000f7e0: 7874 7261 7061 7261 6e6f 6d65 6b73 602e  xtraparanomeks`.
-0000f7f0: 200a 0a53 6f6d 6520 6f74 6865 7220 6f70   ..Some other op
-0000f800: 7469 6f6e 7320 7768 6963 6820 6861 7665  tions which have
-0000f810: 206e 6f20 696d 7061 6374 206f 6e20 7468   no impact on th
-0000f820: 6520 7261 7465 2063 6f72 7265 6374 696f  e rate correctio
-0000f830: 6e20 6275 7420 6164 6420 6d6f 7265 206c  n but add more l
-0000f840: 6179 6572 7320 6f72 2063 6861 6e67 6520  ayers or change 
-0000f850: 7468 6520 6170 7065 6172 616e 6365 206f  the appearance o
-0000f860: 6e20 7468 6520 6d69 7865 6420 2a4b 2a3c  n the mixed *K*<
-0000f870: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-0000f880: 6469 7374 7269 6275 7469 6f6e 2c20 696e  distribution, in
-0000f890: 636c 7564 6520 602d 2d70 6c6f 7461 7067  clude `--plotapg
-0000f8a0: 6d6d 6020 616e 6420 602d 2d70 6c6f 7465  mm` and `--plote
-0000f8b0: 6c6d 6d60 2065 7463 2e20 5468 6520 6f70  lmm` etc. The op
-0000f8c0: 7469 6f6e 2060 2d2d 706c 6f74 6170 676d  tion `--plotapgm
-0000f8d0: 6d60 2063 616e 2062 6520 7365 7420 746f  m` can be set to
-0000f8e0: 2063 616c 6c20 7468 6520 474d 4d20 616e   call the GMM an
-0000f8f0: 616c 7973 6973 2075 706f 6e20 7468 6520  alysis upon the 
-0000f900: 616e 6368 6f72 2070 6169 7220 2a4b 2a3c  anchor pair *K*<
-0000f910: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-0000f920: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
-0000f930: 2070 6c6f 7420 7468 6520 636c 7573 7465   plot the cluste
-0000f940: 7269 6e67 2072 6573 756c 7420 7570 6f6e  ring result upon
-0000f950: 2074 6865 206d 6978 6564 202a 4b2a 3c73   the mixed *K*<s
-0000f960: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-0000f970: 6973 7472 6962 7574 696f 6e2c 2077 6869  istribution, whi
-0000f980: 6368 2068 6173 2074 6f20 6265 2063 6f2d  ch has to be co-
-0000f990: 7365 7420 7769 7468 2074 6865 206f 7074  set with the opt
-0000f9a0: 696f 6e20 602d 2d61 6e63 686f 7270 6f69  ion `--anchorpoi
-0000f9b0: 6e74 7360 2070 726f 7669 6469 6e67 2074  nts` providing t
-0000f9c0: 6865 2061 6e63 686f 7220 7061 6972 7320  he anchor pairs 
-0000f9d0: 696e 666f 726d 6174 696f 6e2e 2054 6865  information. The
-0000f9e0: 206f 7074 696f 6e20 602d 2d70 6c6f 7465   option `--plote
-0000f9f0: 6c6d 6d60 2063 616e 2062 6520 7365 7420  lmm` can be set 
-0000fa00: 746f 2063 616c 6c20 7468 6520 454c 4d4d  to call the ELMM
-0000fa10: 2061 6e61 6c79 7369 7320 7570 6f6e 2074   analysis upon t
-0000fa20: 6865 2077 686f 6c65 2070 6172 616e 6f6d  he whole paranom
-0000fa30: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-0000fa40: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
-0000fa50: 6f6e 2e20 2a2a 4e6f 7465 2074 6861 7420  on. **Note that 
-0000fa60: 7468 6520 7370 6563 6965 7320 6e61 6d65  the species name
-0000fa70: 7320 7072 6573 656e 7420 696e 2074 6865  s present in the
-0000fa80: 2073 7065 6369 6573 2074 7265 6520 6669   species tree fi
-0000fa90: 6c65 2073 686f 756c 6420 6d61 7463 6820  le should match 
-0000faa0: 7468 6520 6e61 6d65 7320 6f66 2074 6865  the names of the
-0000fab0: 2063 6f72 7265 7370 6f6e 6469 6e67 2073   corresponding s
-0000fac0: 6571 7565 6e63 6520 6669 6c65 732a 2a20  equence files** 
-0000fad0: 466f 7220 696e 7374 616e 6365 2c20 6769  For instance, gi
-0000fae0: 7665 6e20 7468 6520 6364 7320 6669 6c65  ven the cds file
-0000faf0: 206e 616d 6573 2027 412e 6364 7327 2c27   names 'A.cds','
-0000fb00: 422e 6364 7327 2c27 432e 6364 7327 2c20  B.cds','C.cds', 
-0000fb10: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
-0000fb20: 2063 6f75 6c64 2062 6520 2728 412e 6364   could be '(A.cd
-0000fb30: 732c 2842 2e63 6473 2c43 2e63 6473 2929  s,(B.cds,C.cds))
-0000fb40: 3b27 2072 6174 6865 7220 7468 616e 2027  ;' rather than '
-0000fb50: 2841 2c28 422c 4329 293b 272e 2054 6865  (A,(B,C));'. The
-0000fb60: 7265 2069 7320 6e6f 2072 6571 7569 7265  re is no require
-0000fb70: 6d65 6e74 2066 6f72 2074 6865 206e 616d  ment for the nam
-0000fb80: 6520 6f66 2074 6865 2070 6172 616e 6f6d  e of the paranom
-0000fb90: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-0000fba0: 3e20 6461 7461 6669 6c65 2077 6869 6368  > datafile which
-0000fbb0: 2063 616e 2062 6520 6e61 6d65 6420 696e   can be named in
-0000fbc0: 2077 6861 7465 7665 7220 6d61 6e6e 6572   whatever manner
-0000fbd0: 2075 7365 7273 2070 7265 6665 722e 2054   users prefer. T
-0000fbe0: 6865 2027 474d 4d27 2069 7320 7468 6520  he 'GMM' is the 
-0000fbf0: 6162 6272 6576 6961 7469 6f6e 206f 6620  abbreviation of 
-0000fc00: 4761 7573 7369 616e 204d 6978 7475 7265  Gaussian Mixture
-0000fc10: 204d 6f64 656c 696e 6720 7768 696c 6520   Modeling while 
-0000fc20: 7468 6520 2745 4c4d 4d27 2072 6566 6572  the 'ELMM' refer
-0000fc30: 7320 746f 2045 7870 6f6e 656e 7469 616c  s to Exponential
-0000fc40: 2d4c 6f67 6e6f 726d 616c 204d 6978 7475  -Lognormal Mixtu
-0000fc50: 7265 204d 6f64 656c 696e 6720 6173 205b  re Modeling as [
-0000fc60: 6b73 7261 7465 735d 2868 7474 7073 3a2f  ksrates](https:/
-0000fc70: 2f67 6974 6875 622e 636f 6d2f 5649 422d  /github.com/VIB-
-0000fc80: 5053 422f 6b73 7261 7465 7329 2069 6e74  PSB/ksrates) int
-0000fc90: 6572 7072 6574 732e 0a0a 5468 6572 6520  erprets...There 
-0000fca0: 6172 6520 3231 2063 6f6c 756d 6e73 2069  are 21 columns i
-0000fcb0: 6e20 7468 6520 7265 7375 6c74 2060 2e6b  n the result `.k
-0000fcc0: 732e 7473 7660 2066 696c 6520 6265 7369  s.tsv` file besi
-0000fcd0: 6465 7320 7468 6520 696e 6465 7820 636f  des the index co
-0000fce0: 6c75 6d6e 7320 6070 6169 7260 2061 7320  lumns `pair` as 
-0000fcf0: 7468 6520 756e 6971 7565 2069 6465 6e74  the unique ident
-0000fd00: 6966 6965 7220 666f 7220 6561 6368 2067  ifier for each g
-0000fd10: 656e 6520 7061 6972 2e20 5468 6520 604e  ene pair. The `N
-0000fd20: 602c 2060 5360 2c20 6064 4e60 2c20 6064  `, `S`, `dN`, `d
-0000fd30: 4e2f 6453 602c 2060 6453 602c 2060 6c60  N/dS`, `dS`, `l`
-0000fd40: 2061 6e64 2060 7460 2061 7265 2066 726f   and `t` are fro
-0000fd50: 6d20 7468 6520 636f 6465 6d6c 2072 6573  m the codeml res
-0000fd60: 756c 7473 2c20 7265 7072 6573 656e 7469  ults, representi
-0000fd70: 6e67 2074 6865 204e 2065 7374 696d 6174  ng the N estimat
-0000fd80: 652c 2074 6865 2053 2065 7374 696d 6174  e, the S estimat
-0000fd90: 652c 2074 6865 2064 4e20 6573 7469 6d61  e, the dN estima
-0000fda0: 7465 2c20 7468 6520 644e 2f64 5320 286f  te, the dN/dS (o
-0000fdb0: 6d65 6761 2920 6573 7469 6d61 7465 2c20  mega) estimate, 
-0000fdc0: 7468 6520 6453 2065 7374 696d 6174 652c  the dS estimate,
-0000fdd0: 2074 6865 206c 6f67 2d6c 696b 656c 6968   the log-likelih
-0000fde0: 6f6f 6420 616e 6420 7468 6520 7420 6573  ood and the t es
-0000fdf0: 7469 6d61 7465 2c20 7265 7370 6563 7469  timate, respecti
-0000fe00: 7665 6c79 2e20 5468 6520 6061 6c69 676e  vely. The `align
-0000fe10: 6d65 6e74 636f 7665 7261 6765 602c 2060  mentcoverage`, `
-0000fe20: 616c 6967 6e6d 656e 7469 6465 6e74 6974  alignmentidentit
-0000fe30: 7960 2061 6e64 2060 616c 6967 6e6d 656e  y` and `alignmen
-0000fe40: 746c 656e 6774 6860 2061 7265 2074 6865  tlength` are the
-0000fe50: 2069 6e66 6f72 6d61 7469 6f6e 2070 6572   information per
-0000fe60: 7461 696e 696e 6720 746f 2074 6865 2061  taining to the a
-0000fe70: 6c69 676e 6d65 6e74 2066 6f72 2065 6163  lignment for eac
-0000fe80: 6820 6661 6d69 6c79 2c20 7265 7072 6573  h family, repres
-0000fe90: 656e 7469 6e67 2074 6865 2072 6174 696f  enting the ratio
-0000fea0: 206f 6620 7468 6520 7374 7269 7070 6564   of the stripped
-0000feb0: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
-0000fec0: 6820 636f 6d70 6172 6564 2074 6f20 7468  h compared to th
-0000fed0: 6520 6675 6c6c 2061 6c69 676e 6d65 6e74  e full alignment
-0000fee0: 206c 656e 6774 682c 2074 6865 2072 6174   length, the rat
-0000fef0: 696f 206f 6620 636f 6c75 6d6e 7320 7769  io of columns wi
-0000ff00: 7468 2069 6465 6e74 6963 616c 206e 7563  th identical nuc
-0000ff10: 6c65 6f74 6964 6573 2063 6f6d 7061 7265  leotides compare
-0000ff20: 6420 746f 2074 6865 206f 7665 7261 6c6c  d to the overall
-0000ff30: 2063 6f6c 756d 6e73 206f 6620 7468 6520   columns of the 
-0000ff40: 7374 7269 7070 6564 2061 6c69 676e 6d65  stripped alignme
-0000ff50: 6e74 2c20 616e 6420 7468 6520 6c65 6e67  nt, and the leng
-0000ff60: 7468 206f 6620 7468 6520 6675 6c6c 2061  th of the full a
-0000ff70: 6c69 676e 6d65 6e74 2c20 7265 7370 6563  lignment, respec
-0000ff80: 7469 7665 6c79 2e0a 2323 2320 7767 6420  tively..### wgd 
-0000ff90: 6d69 780a 0a2a 2a54 6865 206d 6978 7475  mix..**The mixtu
-0000ffa0: 7265 206d 6f64 656c 2063 6c75 7374 6572  re model cluster
-0000ffb0: 696e 6720 616e 616c 7973 6973 206f 6620  ing analysis of 
-0000ffc0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-0000ffd0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
-0000ffe0: 2a2a 0a60 6060 0a77 6764 206d 6978 206b  **.```.wgd mix k
-0000fff0: 7364 6174 6120 282d 2d6e 5f69 6e69 7420  sdata (--n_init 
-00010000: 3230 3020 2d2d 6d61 785f 6974 6572 2032  200 --max_iter 2
-00010010: 3030 202d 2d6b 735f 7261 6e67 6520 3020  00 --ks_range 0 
-00010020: 3520 2d2d 6669 6c74 6572 7320 3330 3020  5 --filters 300 
-00010030: 2d2d 6269 6e73 2035 3020 2d2d 636f 6d70  --bins 50 --comp
-00010040: 6f6e 656e 7473 2031 2034 202d 2d67 616d  onents 1 4 --gam
-00010050: 6d61 2030 2e30 3031 290a 6060 600a 0a54  ma 0.001).```..T
-00010060: 6869 7320 7061 7274 206f 6620 4761 7573  his part of Gaus
-00010070: 7369 616e 206d 6978 7475 7265 206d 6f64  sian mixture mod
-00010080: 656c 696e 6720 2847 4d4d 2920 616e 616c  eling (GMM) anal
-00010090: 7973 6973 2069 7320 696e 6865 7269 7465  ysis is inherite
-000100a0: 6420 6672 6f6d 2074 6865 206f 7269 6769  d from the origi
-000100b0: 6e61 6c20 6077 6764 6020 7072 6f67 7261  nal `wgd` progra
-000100c0: 6d2c 2062 7574 2077 7269 7465 7320 6164  m, but writes ad
-000100d0: 6469 7469 6f6e 616c 6c79 2074 6865 2070  ditionally the p
-000100e0: 726f 6261 6269 6c69 7479 206f 6620 6561  robability of ea
-000100f0: 6368 202a 4b2a 3c73 7562 3e53 3c2f 7375  ch *K*<sub>S</su
-00010100: 623e 2076 616c 7565 2069 6e74 6f20 7468  b> value into th
-00010110: 6520 6669 6e61 6c20 6461 7461 6672 616d  e final datafram
-00010120: 652e 2042 6173 6963 616c 6c79 2c20 7573  e. Basically, us
-00010130: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
-00010140: 6964 6520 7769 7468 2061 2028 6e6f 726d  ide with a (norm
-00010150: 616c 6c79 2066 726f 6d20 7768 6f6c 652d  ally from whole-
-00010160: 7061 7261 6e6f 6d65 206f 7220 616e 6368  paranome or anch
-00010170: 6f72 2d70 6169 7273 2920 2a4b 2a3c 7375  or-pairs) *K*<su
-00010180: 623e 533c 2f73 7562 3e20 6461 7461 6669  b>S</sub> datafi
-00010190: 6c65 2061 6e64 2074 6865 2047 4d4d 2061  le and the GMM a
-000101a0: 6e61 6c79 7369 7320 7769 6c6c 2062 6520  nalysis will be 
-000101b0: 636f 6e64 7563 7465 6420 7570 6f6e 2074  conducted upon t
-000101c0: 6865 2064 6174 6166 696c 652e 2053 6f6d  he datafile. Som
-000101d0: 6520 7061 7261 6d65 7465 7273 2063 616e  e parameters can
-000101e0: 2061 6666 6563 7420 7468 6520 7265 7375   affect the resu
-000101f0: 6c74 732c 2069 6e63 6c75 6469 6e67 2060  lts, including `
-00010200: 2d2d 6e5f 696e 6974 602c 2077 6869 6368  --n_init`, which
-00010210: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-00010220: 206f 6620 6b2d 6d65 616e 7320 696e 6974   of k-means init
-00010230: 6961 6c69 7a61 7469 6f6e 7320 2864 6566  ializations (def
-00010240: 6175 6c74 2032 3030 292c 2060 2d2d 6d61  ault 200), `--ma
-00010250: 785f 6974 6572 602c 2077 6869 6368 2073  x_iter`, which s
-00010260: 6574 7320 7468 6520 6d61 7869 6d75 6d20  ets the maximum 
-00010270: 6e75 6d62 6572 206f 6620 6974 6572 6174  number of iterat
-00010280: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
-00010290: 3029 2c20 602d 2d6d 6574 686f 6460 2c20  0), `--method`, 
-000102a0: 7768 6963 6820 6465 7465 726d 696e 6573  which determines
-000102b0: 2077 6869 6368 2063 6c75 7374 6572 696e   which clusterin
-000102c0: 6720 6d65 7468 6f64 2074 6f20 7573 6520  g method to use 
-000102d0: 2864 6566 6175 6c74 2067 6d6d 292c 2060  (default gmm), `
-000102e0: 2d2d 6761 6d6d 6160 2c20 7768 6963 6820  --gamma`, which 
-000102f0: 7365 7473 2074 6865 2067 616d 6d61 2070  sets the gamma p
-00010300: 6172 616d 6574 6572 2066 6f72 2074 6865  arameter for the
-00010310: 2062 676d 6d20 6d6f 6465 6c20 2864 6566   bgmm model (def
-00010320: 6175 6c74 2030 2e30 3031 292c 2060 2d2d  ault 0.001), `--
-00010330: 636f 6d70 6f6e 656e 7473 602c 2077 6869  components`, whi
-00010340: 6368 2073 6574 7320 7468 6520 7261 6e67  ch sets the rang
-00010350: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
-00010360: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
-00010370: 2066 6974 2028 6465 6661 756c 7420 3120   fit (default 1 
-00010380: 3429 2c20 616e 6420 7468 6520 6461 7461  4), and the data
-00010390: 2066 696c 7465 7269 6e67 2070 6172 616d   filtering param
-000103a0: 6574 6572 7320 602d 2d66 696c 7465 7273  eters `--filters
-000103b0: 6020 7768 6963 6820 6669 6c74 6572 7320  ` which filters 
-000103c0: 6461 7461 2062 6173 6564 206f 6e20 616c  data based on al
-000103d0: 6967 6e6d 656e 7420 6c65 6e67 7468 2c20  ignment length, 
-000103e0: 602d 2d6b 735f 7261 6e67 6560 2077 6869  `--ks_range` whi
-000103f0: 6368 2066 696c 7465 7273 2064 6174 6120  ch filters data 
-00010400: 6261 7365 6420 6f6e 202a 4b2a 3c73 7562  based on *K*<sub
-00010410: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
-00010420: 616e 6420 7468 6520 7061 7261 6d65 7465  and the paramete
-00010430: 7220 602d 2d62 696e 7360 2077 6869 6368  r `--bins` which
-00010440: 2073 6574 7320 7468 6520 6e75 6d62 6572   sets the number
-00010450: 206f 6620 6269 6e73 2069 6e20 2a4b 2a3c   of bins in *K*<
-00010460: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00010470: 7269 6275 7469 6f6e 2028 6465 6661 756c  ribution (defaul
-00010480: 7420 3530 292e 0a0a 2a2a 4120 7375 6767  t 50)...**A sugg
-00010490: 6573 7465 6420 7374 6172 7469 6e67 2072  ested starting r
-000104a0: 756e 2063 616e 2075 7365 2063 6f6d 6d61  un can use comma
-000104b0: 6e64 2073 696d 706c 7920 6173 2062 656c  nd simply as bel
-000104c0: 6f77 2a2a 0a0a 6060 600a 7767 6420 6d69  ow**..```.wgd mi
-000104d0: 7820 6b73 6461 7461 0a60 6060 0a0a 2323  x ksdata.```..##
-000104e0: 2320 7767 6420 7065 616b 0a0a 2a2a 5468  # wgd peak..**Th
-000104f0: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
-00010500: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
-00010510: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
-00010520: 2069 6e20 5747 4420 6461 7469 6e67 2a2a   in WGD dating**
-00010530: 0a60 6060 0a77 6764 2070 6561 6b20 6b73  .```.wgd peak ks
-00010540: 6461 7461 202d 6170 2061 7064 6174 6120  data -ap apdata 
-00010550: 2d73 6d20 736d 6461 7461 202d 6c65 206c  -sm smdata -le l
-00010560: 6564 6174 6120 2d6d 7020 6d70 6461 7461  edata -mp mpdata
-00010570: 202d 2d68 6575 7269 7374 6963 2028 2d2d   --heuristic (--
-00010580: 616c 6967 6e66 696c 7465 7220 302e 3020  alignfilter 0.0 
-00010590: 3020 302e 3020 2d2d 6b73 7261 6e67 6520  0 0.0 --ksrange 
-000105a0: 3020 3520 2d2d 6269 6e5f 7769 6474 6820  0 5 --bin_width 
-000105b0: 302e 3120 2d2d 6775 6964 6520 7365 676d  0.1 --guide segm
-000105c0: 656e 7420 2d2d 7072 6f6d 696e 656e 6365  ent --prominence
-000105d0: 5f63 7574 6f66 6620 302e 3120 2d2d 7265  _cutoff 0.1 --re
-000105e0: 6c5f 6865 6967 6874 2030 2e34 202d 2d63  l_height 0.4 --c
-000105f0: 6920 3935 202d 2d68 6472 2039 3520 2d2d  i 95 --hdr 95 --
-00010600: 6b73 6375 746f 6666 2035 290a 6060 600a  kscutoff 5).```.
-00010610: 0a41 7320 6d65 6e74 696f 6e65 6420 7072  .As mentioned pr
-00010620: 6576 696f 7573 6c79 2c20 6120 6865 7572  eviously, a heur
-00010630: 6973 7469 6320 6d65 7468 6f64 2061 6e64  istic method and
-00010640: 2061 2063 6f6c 6c69 6e65 6172 2073 6567   a collinear seg
-00010650: 6d65 6e74 732d 6775 6964 6564 2061 6e63  ments-guided anc
-00010660: 686f 7220 7061 6972 2063 6c75 7374 6572  hor pair cluster
-00010670: 696e 6720 666f 7220 7468 6520 7365 6172  ing for the sear
-00010680: 6368 206f 6620 6372 6564 6961 626c 6520  ch of crediable 
-00010690: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-000106a0: 7261 6e67 6520 7573 6564 2069 6e20 5747  range used in WG
-000106b0: 4420 6461 7469 6e67 2061 7265 2069 6d70  D dating are imp
-000106c0: 6c65 6d65 6e74 6564 2069 6e20 6077 6764  lemented in `wgd
-000106d0: 2076 3260 2e20 5573 6572 7320 6e65 6564   v2`. Users need
-000106e0: 2074 6f20 7072 6f76 6964 6520 7468 6520   to provide the 
-000106f0: 616e 6368 6f72 2070 6f69 6e74 732c 2073  anchor points, s
-00010700: 6567 6d65 6e74 732c 206c 6973 7473 6567  egments, listseg
-00010710: 6d65 6e74 732c 206d 756c 7469 706c 6963  ments, multiplic
-00010720: 6f6e 7061 6972 7320 6461 7461 6669 6c65  onpairs datafile
-00010730: 2066 726f 6d20 6069 2d61 6468 6f72 6560   from `i-adhore`
-00010740: 2074 6f20 6163 6869 6576 6520 7468 6520   to achieve the 
-00010750: 636c 7573 7465 7269 6e67 2066 756e 6374  clustering funct
-00010760: 696f 6e2e 2053 6f6d 6520 7061 7261 6d65  ion. Some parame
-00010770: 7465 7273 2074 6861 7420 6361 6e20 696d  ters that can im
-00010780: 7061 6374 2074 6865 2072 6573 756c 7473  pact the results
-00010790: 2069 6e63 6c75 6465 2060 2d2d 616c 6967   include `--alig
-000107a0: 6e66 696c 7465 7260 2c20 7768 6963 6820  nfilter`, which 
-000107b0: 6669 6c74 6572 7320 7468 6520 6461 7461  filters the data
-000107c0: 2062 6173 6564 206f 6e20 616c 6967 6e6d   based on alignm
-000107d0: 656e 7420 6964 656e 7469 7479 2c20 6c65  ent identity, le
-000107e0: 6e67 7468 2061 6e64 2063 6f76 6572 6167  ngth and coverag
-000107f0: 652c 2060 2d2d 6b73 7261 6e67 6560 2c20  e, `--ksrange`, 
-00010800: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
-00010810: 616e 6765 206f 6620 4b73 2074 6f20 6265  ange of Ks to be
-00010820: 2061 6e61 6c79 7a65 642c 2060 2d2d 6269   analyzed, `--bi
-00010830: 6e5f 7769 6474 6860 2c20 7768 6963 6820  n_width`, which 
-00010840: 7365 7473 2074 6865 2062 616e 6477 6964  sets the bandwid
-00010850: 7468 206f 6620 2a4b 2a3c 7375 623e 533c  th of *K*<sub>S<
-00010860: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00010870: 6f6e 2c20 602d 2d77 6569 6768 7473 5f6f  on, `--weights_o
-00010880: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
-00010890: 6020 7768 6963 6820 6465 7465 726d 696e  ` which determin
-000108a0: 6573 2077 6865 7468 6572 2074 6f20 696e  es whether to in
-000108b0: 636c 7564 6520 2a4b 2a3c 7375 623e 533c  clude *K*<sub>S<
-000108c0: 2f73 7562 3e20 6f75 746c 6965 7273 2028  /sub> outliers (
-000108d0: 7768 6f73 6520 7661 6c75 6520 6973 206f  whose value is o
-000108e0: 7665 7220 3529 2069 6e20 616e 616c 7973  ver 5) in analys
-000108f0: 6973 2c20 602d 2d6d 6574 686f 6460 2077  is, `--method` w
-00010900: 6869 6368 2064 6574 6572 6d69 6e65 7320  hich determines 
-00010910: 7768 6963 6820 636c 7573 7465 7269 6e67  which clustering
-00010920: 206d 6574 686f 6420 746f 2075 7365 2028   method to use (
-00010930: 6465 6661 756c 7420 676d 6d29 2c20 602d  default gmm), `-
-00010940: 2d73 6565 6460 2077 6869 6368 2073 6574  -seed` which set
-00010950: 7320 7468 6520 7261 6e64 6f6d 2073 6565  s the random see
-00010960: 6420 6769 7665 6e20 746f 2069 6e69 7469  d given to initi
-00010970: 616c 697a 6174 696f 6e20 2864 6566 6175  alization (defau
-00010980: 6c74 2032 3335 3238 3930 292c 2060 2d2d  lt 2352890), `--
-00010990: 6e5f 696e 6974 602c 2077 6869 6368 2073  n_init`, which s
-000109a0: 6574 7320 7468 6520 6e75 6d62 6572 206f  ets the number o
-000109b0: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
-000109c0: 6c69 7a61 7469 6f6e 7320 2864 6566 6175  lizations (defau
-000109d0: 6c74 2032 3030 292c 2060 2d2d 656d 5f69  lt 200), `--em_i
-000109e0: 7465 7260 2c20 7768 6963 6820 7365 7473  ter`, which sets
-000109f0: 2074 6865 206d 6178 696d 756d 206e 756d   the maximum num
-00010a00: 6265 7220 6f66 2069 7465 7261 7469 6f6e  ber of iteration
-00010a10: 7320 2864 6566 6175 6c74 2032 3030 292c  s (default 200),
-00010a20: 2060 2d2d 6761 6d6d 6160 2c20 7768 6963   `--gamma`, whic
-00010a30: 6820 7365 7473 2074 6865 2067 616d 6d61  h sets the gamma
-00010a40: 2070 6172 616d 6574 6572 2066 6f72 2074   parameter for t
-00010a50: 6865 2062 676d 6d20 6d6f 6465 6c20 2864  he bgmm model (d
-00010a60: 6566 6175 6c74 2030 2e30 3031 292c 2060  efault 0.001), `
-00010a70: 2d2d 636f 6d70 6f6e 656e 7473 602c 2077  --components`, w
-00010a80: 6869 6368 2073 6574 7320 7468 6520 7261  hich sets the ra
-00010a90: 6e67 6520 6f66 2074 6865 206e 756d 6265  nge of the numbe
-00010aa0: 7220 6f66 2063 6f6d 706f 6e65 6e74 7320  r of components 
-00010ab0: 746f 2066 6974 2028 6465 6661 756c 7420  to fit (default 
-00010ac0: 3120 3429 2c20 602d 2d77 6569 6768 7465  1 4), `--weighte
-00010ad0: 6460 2077 6869 6368 2064 6574 6572 6d69  d` which determi
-00010ae0: 6e65 7320 7768 6574 6865 7220 746f 2075  nes whether to u
-00010af0: 7365 206e 6f64 652d 7765 6967 6874 6564  se node-weighted
-00010b00: 206d 6574 686f 6420 666f 7220 6465 2d72   method for de-r
-00010b10: 6564 756e 6461 6e63 792c 2060 2d2d 6775  edundancy, `--gu
-00010b20: 6964 6560 2077 6869 6368 2064 6574 6572  ide` which deter
-00010b30: 6d69 6e65 7320 7768 6963 6820 7265 6769  mines which regi
-00010b40: 6d65 2072 6573 6964 696e 6720 616e 6368  me residing anch
-00010b50: 6f72 7320 746f 2062 6520 7573 6564 2028  ors to be used (
-00010b60: 6465 6661 756c 7420 5365 676d 656e 7429  default Segment)
-00010b70: 2c20 602d 2d70 726f 6d69 6e65 6e63 655f  , `--prominence_
-00010b80: 6375 746f 6666 6020 7768 6963 6820 7365  cutoff` which se
-00010b90: 7473 2074 6865 2070 726f 6d69 6e65 6e63  ts the prominenc
-00010ba0: 6520 6375 746f 6666 206f 6620 6163 6365  e cutoff of acce
-00010bb0: 7074 6162 6c65 2070 6561 6b73 2069 6e20  ptable peaks in 
-00010bc0: 7065 616b 2066 696e 6469 6e67 2070 726f  peak finding pro
-00010bd0: 6365 7373 2c20 602d 2d72 656c 5f68 6569  cess, `--rel_hei
-00010be0: 6768 7460 2077 6869 6368 2073 6574 7320  ght` which sets 
-00010bf0: 7468 6520 7265 6c61 7469 7665 2068 6569  the relative hei
-00010c00: 6768 7420 6174 2077 6869 6368 2074 6865  ght at which the
-00010c10: 2070 6561 6b20 7769 6474 6820 6973 206d   peak width is m
-00010c20: 6561 7375 7265 642c 2060 2d2d 6b73 746f  easured, `--ksto
-00010c30: 6461 7465 6020 7768 6963 6820 6d61 6e75  date` which manu
-00010c40: 616c 6c79 2073 6574 7320 7468 6520 7261  ally sets the ra
-00010c50: 6e67 6520 6f66 202a 4b2a 3c73 7562 3e53  nge of *K*<sub>S
-00010c60: 3c2f 7375 623e 2074 6f20 6265 2064 6174  </sub> to be dat
-00010c70: 6564 2069 6e20 6865 7572 6973 7469 6320  ed in heuristic 
-00010c80: 7365 6172 6368 2061 6e64 206e 6565 6473  search and needs
-00010c90: 2074 6f20 6265 2063 6f2d 7365 7420 7769   to be co-set wi
-00010ca0: 7468 206f 7074 696f 6e20 602d 2d6d 616e  th option `--man
-00010cb0: 7561 6c73 6574 602c 2060 2d2d 786c 696d  ualset`, `--xlim
-00010cc0: 6020 616e 6420 602d 2d79 6c69 6d60 2064  ` and `--ylim` d
-00010cd0: 6574 6572 6d69 6e69 6e67 2074 6865 2078  etermining the x
-00010ce0: 2061 6e64 2079 2061 7869 7320 6c69 6d69   and y axis limi
-00010cf0: 7420 6f66 2047 4d4d 202a 4b2a 3c73 7562  t of GMM *K*<sub
-00010d00: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-00010d10: 7574 696f 6e2c 2060 2d2d 6369 6020 7365  ution, `--ci` se
-00010d20: 7474 696e 6720 7468 6520 636f 6e66 6964  tting the confid
-00010d30: 656e 6365 206c 6576 656c 206f 6620 6c6f  ence level of lo
-00010d40: 672d 6e6f 726d 616c 2064 6973 7472 6962  g-normal distrib
-00010d50: 7574 696f 6e20 746f 2064 6174 6520 2864  ution to date (d
-00010d60: 6566 6175 6c74 2039 3529 2c20 602d 2d68  efault 95), `--h
-00010d70: 6472 6020 7365 7474 696e 6720 7468 6520  dr` setting the 
-00010d80: 6869 6768 6573 7420 6465 6e73 6974 7920  highest density 
-00010d90: 7265 6769 6f6e 2028 4844 5229 2061 7070  region (HDR) app
-00010da0: 6c69 6564 2069 6e20 7468 6520 7365 676d  lied in the segm
-00010db0: 656e 742d 6775 6964 6564 2061 6e63 686f  ent-guided ancho
-00010dc0: 7220 7061 6972 202a 4b2a 3c73 7562 3e53  r pair *K*<sub>S
-00010dd0: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
-00010de0: 696f 6e20 2864 6566 6175 6c74 2039 3529  ion (default 95)
-00010df0: 2c20 602d 2d68 6575 7269 7374 6963 6020  , `--heuristic` 
-00010e00: 6465 7465 726d 696e 696e 6720 7768 6574  determining whet
-00010e10: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
-00010e20: 6865 7572 6973 7469 6320 6d65 7468 6f64  heuristic method
-00010e30: 206f 6620 6465 6669 6e69 6e67 2043 4920   of defining CI 
-00010e40: 666f 7220 6461 7469 6e67 2c20 602d 2d6b  for dating, `--k
-00010e50: 7363 7574 6f66 6660 2073 6574 7469 6e67  scutoff` setting
-00010e60: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
-00010e70: 7375 623e 2073 6174 7572 6174 696f 6e20  sub> saturation 
-00010e80: 6375 746f 6666 2069 6e20 6461 7469 6e67  cutoff in dating
-00010e90: 2028 6465 6661 756c 7420 3529 2e0a 0a46   (default 5)...F
-00010ea0: 6f75 7220 7265 7375 6c74 2073 7562 666c  our result subfl
-00010eb0: 6f64 6572 7320 7769 6c6c 2062 6520 7072  oders will be pr
-00010ec0: 6f64 7563 6564 2c20 6e61 6d65 6c79 2060  oduced, namely `
-00010ed0: 416e 6368 6f72 4b73 5f46 696e 6450 6561  AnchorKs_FindPea
-00010ee0: 6b60 2c20 6041 6e63 686f 724b 735f 474d  k`, `AnchorKs_GM
-00010ef0: 4d60 2c20 6053 6567 6d65 6e74 4775 6964  M`, `SegmentGuid
-00010f00: 654b 735f 474d 4d60 2061 6e64 2060 5365  eKs_GMM` and `Se
-00010f10: 676d 656e 744b 735f 4669 6e64 5065 616b  gmentKs_FindPeak
-00010f20: 602e 2054 6865 2060 416e 6368 6f72 4b73  `. The `AnchorKs
-00010f30: 5f46 696e 6450 6561 6b60 2073 7562 666c  _FindPeak` subfl
-00010f40: 6f64 6572 2063 6f6e 7461 696e 7320 7265  oder contains re
-00010f50: 7375 6c74 7320 6f66 2074 6865 2064 6574  sults of the det
-00010f60: 6563 7465 6420 7065 616b 7320 6279 2074  ected peaks by t
-00010f70: 6865 2060 7369 676e 616c 6020 6d6f 6475  he `signal` modu
-00010f80: 6c65 206f 6620 6053 6369 5079 6020 6c69  le of `SciPy` li
-00010f90: 6272 6172 7920 616e 6420 7468 6520 6173  brary and the as
-00010fa0: 7375 6d65 6420 6869 6768 6573 7420 6d61  sumed highest ma
-00010fb0: 7373 2070 6172 7420 2872 6566 6572 7265  ss part (referre
-00010fc0: 6420 746f 2061 7320 4869 6768 4d61 7373  d to as HighMass
-00010fd0: 2068 6572 6561 6674 6572 2920 6f66 2065   hereafter) of e
-00010fe0: 6163 6820 7065 616b 2c20 7768 6963 6820  ach peak, which 
-00010ff0: 6361 6e20 6265 2075 7365 6420 666f 7220  can be used for 
-00011000: 6675 7274 6865 7220 5747 4420 6461 7469  further WGD dati
-00011010: 6e67 2e20 5468 6520 6041 6e63 686f 724b  ng. The `AnchorK
-00011020: 735f 474d 4d60 2073 686f 7773 2074 6865  s_GMM` shows the
-00011030: 2047 4d4d 2072 6573 756c 7473 2075 706f   GMM results upo
-00011040: 6e20 7468 6520 6f72 6967 696e 616c 2061  n the original a
-00011050: 6e63 686f 7220 2a4b 2a3c 7375 623e 533c  nchor *K*<sub>S<
-00011060: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00011070: 6f6e 2062 7920 7468 6520 606d 6978 7475  on by the `mixtu
-00011080: 7265 6020 6d6f 6475 6c65 206f 6620 6073  re` module of `s
-00011090: 6369 6b69 742d 6c65 6172 6e60 206c 6962  cikit-learn` lib
-000110a0: 7261 7279 2061 6e64 2074 776f 2073 7562  rary and two sub
-000110b0: 666c 6f64 6572 732c 2060 4c6f 6747 4d4d  floders, `LogGMM
-000110c0: 5f43 4960 2063 6f6e 7461 696e 696e 6720  _CI` containing 
-000110d0: 7468 6520 7265 7375 6c74 7320 6f66 2039  the results of 9
-000110e0: 3525 2043 4920 6f66 2065 6163 6820 636f  5% CI of each co
-000110f0: 6d70 6f6e 656e 742c 2060 4869 6768 4d61  mponent, `HighMa
-00011100: 7373 5f43 4960 2063 6f6e 7461 696e 696e  ss_CI` containin
-00011110: 6720 7468 6520 4869 6768 4d61 7373 206f  g the HighMass o
-00011120: 6620 6561 6368 2063 6f6d 706f 6e65 6e74  f each component
-00011130: 2c20 7768 6963 6820 6361 6e20 6265 2075  , which can be u
-00011140: 7365 6420 666f 7220 6675 7274 6865 7220  sed for further 
-00011150: 5747 4420 6461 7469 6e67 2e20 5468 6520  WGD dating. The 
-00011160: 6053 6567 6d65 6e74 4775 6964 654b 735f  `SegmentGuideKs_
-00011170: 474d 4d60 2073 7562 666f 6c64 6572 2070  GMM` subfolder p
-00011180: 7265 7365 6e74 7320 7265 7375 6c74 7320  resents results 
-00011190: 6f66 2073 6567 6d65 6e74 202a 4b2a 3c73  of segment *K*<s
-000111a0: 7562 3e53 3c2f 7375 623e 2047 4d4d 2077  ub>S</sub> GMM w
-000111b0: 6869 6368 2061 7265 206d 6170 7065 6420  hich are mapped 
-000111c0: 6261 636b 2074 6f20 7468 6520 7265 7369  back to the resi
-000111d0: 6469 6e67 2061 6e63 686f 7220 7061 6972  ding anchor pair
-000111e0: 7320 616e 6420 7468 6520 6173 736f 6369  s and the associ
-000111f0: 6174 6564 2039 3525 2048 4452 2061 6e64  ated 95% HDR and
-00011200: 2048 6967 684d 6173 7320 6f66 2065 6163   HighMass of eac
-00011210: 6820 7365 676d 656e 7420 636c 7573 7465  h segment cluste
-00011220: 7220 696e 2073 7562 666c 6f64 6572 7320  r in subfloders 
-00011230: 6f66 2060 4844 525f 4349 6020 616e 6420  of `HDR_CI` and 
-00011240: 6048 6967 684d 6173 735f 4349 602e 2054  `HighMass_CI`. T
-00011250: 6865 2060 5365 676d 656e 744b 735f 4669  he `SegmentKs_Fi
-00011260: 6e64 5065 616b 6020 7375 6266 6f6c 6465  ndPeak` subfolde
-00011270: 7220 6973 2073 696d 696c 6172 2074 6f20  r is similar to 
-00011280: 6041 6e63 686f 724b 735f 4669 6e64 5065  `AnchorKs_FindPe
-00011290: 616b 6020 6275 7420 7769 7468 2073 6567  ak` but with seg
-000112a0: 6d65 6e74 202a 4b2a 3c73 7562 3e53 3c2f  ment *K*<sub>S</
-000112b0: 7375 623e 2069 6e73 7465 6164 2e20 5468  sub> instead. Th
-000112c0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000112d0: 3e20 696e 2060 4d75 6c74 6970 6c69 636f  > in `Multiplico
-000112e0: 6e60 2063 616e 2061 6c73 6f20 6265 2063  n` can also be c
-000112f0: 616c 6375 6c61 7465 6420 696e 2070 6c61  alculated in pla
-00011300: 6365 206f 6620 6053 6567 6d65 6e74 6020  ce of `Segment` 
-00011310: 7573 696e 6720 7468 6520 6f70 7469 6f6e  using the option
-00011320: 2060 2d2d 6775 6964 6560 2061 7320 7375   `--guide` as su
-00011330: 6368 2074 6865 2072 6573 756c 7420 7469  ch the result ti
-00011340: 746c 652c 206c 6162 656c 2c20 6669 6c65  tle, label, file
-00011350: 2061 6e64 2066 6f6c 6465 7220 6e61 6d65   and folder name
-00011360: 7320 7769 6c6c 2062 6520 6368 616e 6765  s will be change
-00011370: 6420 6163 636f 7264 696e 676c 792e 0a0a  d accordingly...
-00011380: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-00011390: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-000113a0: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
-000113b0: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
-000113c0: 600a 7767 6420 7065 616b 206b 7364 6174  `.wgd peak ksdat
-000113d0: 6120 2d61 7020 6170 6461 7461 202d 736d  a -ap apdata -sm
-000113e0: 2073 6d64 6174 6120 2d6c 6520 6c65 6461   smdata -le leda
-000113f0: 7461 202d 6d70 206d 7064 6174 6120 2d2d  ta -mp mpdata --
-00011400: 6865 7572 6973 7469 630a 6060 600a 0a23  heuristic.```..#
-00011410: 2323 2077 6764 2073 796e 0a0a 2a2a 5468  ## wgd syn..**Th
-00011420: 6520 696e 7472 612d 7370 6563 6966 6963  e intra-specific
-00011430: 2073 796e 7465 6e79 2069 6e66 6572 656e   synteny inferen
-00011440: 6365 2a2a 0a60 6060 0a77 6764 2073 796e  ce**.```.wgd syn
-00011450: 2066 616d 696c 6965 7320 6766 6620 282d   families gff (-
-00011460: 2d6b 735f 6469 7374 7269 6275 7469 6f6e  -ks_distribution
-00011470: 206b 7364 6174 6120 2d66 2067 656e 6520   ksdata -f gene 
-00011480: 2d61 2049 4420 2d2d 6d69 6e6c 656e 202d  -a ID --minlen -
-00011490: 3120 2d2d 6d69 6e73 6567 6c65 6e20 3130  1 --minseglen 10
-000114a0: 3030 3020 2d2d 6d69 6e67 656e 656e 756d  000 --mingenenum
-000114b0: 2033 3029 0a60 6060 0a0a 5468 6520 7072   30).```..The pr
-000114c0: 6f67 7261 6d20 6077 6764 2073 796e 6020  ogram `wgd syn` 
-000114d0: 6973 206d 6169 6e6c 7920 6465 616c 696e  is mainly dealin
-000114e0: 6720 7769 7468 2063 6f6c 6c69 6e65 6172  g with collinear
-000114f0: 6974 7920 6f72 2073 796e 7465 6e79 2028  ity or synteny (
-00011500: 626f 7468 2072 6566 6572 7265 6420 746f  both referred to
-00011510: 2061 7320 7379 6e74 656e 7920 6865 7265   as synteny here
-00011520: 6166 7465 7229 2061 6e61 6c79 7369 732e  after) analysis.
-00011530: 2054 776f 2069 6e70 7574 2066 696c 6573   Two input files
-00011540: 2061 7265 2065 7373 656e 7469 616c 2c20   are essential, 
-00011550: 7468 6520 6765 6e65 2066 616d 696c 7920  the gene family 
-00011560: 6669 6c65 2061 6e64 2074 6865 2067 6666  file and the gff
-00011570: 3320 6669 6c65 2e20 5468 6520 6765 6e65  3 file. The gene
-00011580: 2066 616d 696c 7920 6669 6c65 2069 7320   family file is 
-00011590: 696e 2074 6865 2066 6f72 6d61 7420 6173  in the format as
-000115a0: 2060 4f72 7468 6f46 696e 6465 7260 2e20   `OrthoFinder`. 
-000115b0: 5468 6520 736f 6674 7761 7265 2060 692d  The software `i-
-000115c0: 6164 686f 7265 6020 6973 2061 2070 7265  adhore` is a pre
-000115d0: 7265 7175 6973 6974 652e 2057 6974 6820  requisite. With 
-000115e0: 6465 6661 756c 7420 7061 7261 6d65 7465  default paramete
-000115f0: 7273 2c20 7468 6520 7072 6f67 7261 6d20  rs, the program 
-00011600: 6261 7369 6361 6c6c 7920 636f 6e64 7563  basically conduc
-00011610: 7473 2031 2920 6669 6c74 6572 696e 6720  ts 1) filtering 
-00011620: 6765 6e65 2066 616d 696c 6965 7320 6261  gene families ba
-00011630: 7365 6420 6f6e 206d 6178 696d 756d 2066  sed on maximum f
-00011640: 616d 696c 7920 7369 7a65 2032 2920 7265  amily size 2) re
-00011650: 7472 6965 7669 6e67 2067 656e 6520 706f  trieving gene po
-00011660: 7369 7469 6f6e 2061 6e64 2073 6361 6666  sition and scaff
-00011670: 6f6c 6420 696e 666f 726d 6174 696f 6e20  old information 
-00011680: 6672 6f6d 2067 6666 3320 6669 6c65 2033  from gff3 file 3
-00011690: 2920 7072 6f64 7563 696e 6720 7468 6520  ) producing the 
-000116a0: 636f 6e66 6967 7572 6174 696f 6e20 6669  configuration fi
-000116b0: 6c65 2061 6e64 2061 7373 6f63 6961 7465  le and associate
-000116c0: 6420 6461 7461 6669 6c65 7320 666f 7220  d datafiles for 
-000116d0: 6069 2d61 6468 6f72 6560 2034 2920 6361  `i-adhore` 4) ca
-000116e0: 6c6c 696e 6720 6069 2d61 6468 6f72 6560  lling `i-adhore`
-000116f0: 2067 6976 656e 2074 6865 2070 6172 616d   given the param
-00011700: 6574 6572 7320 7365 7420 746f 2069 6e66  eters set to inf
-00011710: 6572 2073 796e 7465 6e79 2035 2920 7669  er synteny 5) vi
-00011720: 7375 616c 697a 696e 6720 7468 6520 7379  sualizing the sy
-00011730: 6e74 656e 7920 696e 2022 646f 7470 6c6f  nteny in "dotplo
-00011740: 7422 2069 6e20 7468 6520 756e 6974 206f  t" in the unit o
-00011750: 6620 6765 6e65 7320 616e 6420 6261 7365  f genes and base
-00011760: 732c 2069 6e20 2253 796e 6465 7074 6822  s, in "Syndepth"
-00011770: 2070 6c6f 7420 7368 6f77 696e 6720 7468   plot showing th
-00011780: 6520 6469 7374 7269 6275 7469 6f6e 206f  e distribution o
-00011790: 6620 6469 6666 6572 656e 7420 6361 7465  f different cate
-000117a0: 676f 7269 6573 206f 6620 636f 6c6c 696e  gories of collin
-000117b0: 6561 7269 7479 2072 6174 696f 7320 7769  earity ratios wi
-000117c0: 7468 696e 2061 6e64 2062 6574 7765 656e  thin and between
-000117d0: 2073 7065 6369 6573 2c20 696e 2022 6475   species, in "du
-000117e0: 7053 7461 636b 2220 706c 6f74 2073 686f  pStack" plot sho
-000117f0: 7769 6e67 206d 756c 7469 706c 6963 6f6e  wing multiplicon
-00011800: 7320 7769 7468 2064 6966 6665 7265 6e74  s with different
-00011810: 206d 756c 7469 706c 6963 6174 696f 6e20   multiplication 
-00011820: 6c65 7665 6c73 2e20 3629 2069 6620 7769  levels. 6) if wi
-00011830: 7468 202a 4b2a 3c73 7562 3e53 3c2f 7375  th *K*<sub>S</su
-00011840: 623e 2064 6174 612c 2061 2022 2a4b 2a3c  b> data, a "*K*<
-00011850: 7375 623e 533c 2f73 7562 3e20 646f 7470  sub>S</sub> dotp
-00011860: 6c6f 7422 2077 6974 6820 646f 7473 2061  lot" with dots a
-00011870: 6e6e 6f74 6174 6564 2069 6e20 2a4b 2a3c  nnotated in *K*<
-00011880: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
-00011890: 6573 2061 6e64 2061 202a 4b2a 3c73 7562  es and a *K*<sub
-000118a0: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-000118b0: 7574 696f 6e20 7769 7468 2061 6e63 686f  ution with ancho
-000118c0: 7220 7061 6972 7320 6465 6e6f 7465 6420  r pairs denoted 
-000118d0: 7769 6c6c 2062 6520 7072 6f64 7563 6564  will be produced
-000118e0: 2e20 5468 6520 6765 6e65 2069 6e66 6f72  . The gene infor
-000118f0: 6d61 7469 6f6e 2069 6e20 7468 6520 6765  mation in the ge
-00011900: 6e65 2066 616d 696c 7920 6669 6c65 2061  ne family file a
-00011910: 6e64 2067 6666 3320 6669 6c65 2073 686f  nd gff3 file sho
-00011920: 756c 6420 6265 206d 6174 6368 6564 2077  uld be matched w
-00011930: 6869 6368 2072 6571 7569 7265 7320 7573  hich requires us
-00011940: 6572 7320 746f 2073 6574 2070 726f 7065  ers to set prope
-00011950: 7220 602d 2d66 6561 7475 7265 6020 616e  r `--feature` an
-00011960: 6420 602d 2d61 7474 7269 6275 7465 602e  d `--attribute`.
-00011970: 2054 6865 206d 6178 696d 756d 2066 616d   The maximum fam
-00011980: 696c 7920 7369 7a65 2074 6f20 6265 2069  ily size to be i
-00011990: 6e63 6c75 6465 6420 6361 6e20 6265 2073  ncluded can be s
-000119a0: 6574 2076 6961 2074 6865 206f 7074 696f  et via the optio
-000119b0: 6e20 602d 2d6d 6178 7369 7a65 602c 206e  n `--maxsize`, n
-000119c0: 6f74 6564 2074 6861 7420 7468 6973 2066  oted that this f
-000119d0: 696c 7465 7269 6e67 2069 7320 6d61 696e  iltering is main
-000119e0: 6c79 2074 6f20 6472 6f70 2074 686f 7365  ly to drop those
-000119f0: 2068 7567 6520 7461 6e64 656d 2064 7570   huge tandem dup
-00011a00: 6c69 6361 7465 7320 6661 6d69 6c79 2061  licates family a
-00011a10: 6e64 2074 7261 6e73 706f 7361 626c 6520  nd transposable 
-00011a20: 656c 656d 656e 7473 2028 5445 7329 2066  elements (TEs) f
-00011a30: 616d 696c 792c 2061 6e64 206e 6f74 206d  amily, and not m
-00011a40: 616e 6461 746f 7279 2e20 5573 6572 7320  andatory. Users 
-00011a50: 6361 6e20 6669 6c74 6572 2074 686f 7365  can filter those
-00011a60: 2066 7261 676d 656e 7461 7279 2073 6361   fragmentary sca
-00011a70: 6666 6f6c 6473 2076 6961 2074 6865 206f  ffolds via the o
-00011a80: 7074 696f 6e20 602d 2d6d 696e 6c65 6e60  ption `--minlen`
-00011a90: 2e20 5468 6520 6d69 6e69 6d75 6d20 6c65  . The minimum le
-00011aa0: 6e67 7468 2061 6e64 206e 756d 6265 7220  ngth and number 
-00011ab0: 6f66 2067 656e 6573 2066 6f72 2061 2073  of genes for a s
-00011ac0: 6567 6d65 6e74 2074 6f20 6265 2063 6f6e  egment to be con
-00011ad0: 7369 6465 7265 6420 6361 6e20 6265 2073  sidered can be s
-00011ae0: 6574 2076 6961 2074 6865 206f 7074 696f  et via the optio
-00011af0: 6e20 602d 2d6d 696e 7365 676c 656e 6020  n `--minseglen` 
-00011b00: 616e 6420 602d 2d6d 696e 6765 6e65 6e75  and `--mingenenu
-00011b10: 6d60 2e20 5265 6475 6e64 616e 7420 6d75  m`. Redundant mu
-00011b20: 6c74 6970 6c69 636f 6e73 2063 616e 2062  ltiplicons can b
-00011b30: 6520 6b65 7074 2062 7920 7365 7420 7468  e kept by set th
-00011b40: 6520 666c 6167 206f 7074 696f 6e20 602d  e flag option `-
-00011b50: 2d6b 6565 7072 6564 756e 602e 0a0a 2a2a  -keepredun`...**
-00011b60: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
-00011b70: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
-00011b80: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
-00011b90: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
-00011ba0: 7767 6420 7379 6e20 6661 6d69 6c69 6573  wgd syn families
-00011bb0: 2067 6666 0a60 6060 0a0a 2a2a 5468 6520   gff.```..**The 
-00011bc0: 696e 7465 722d 7370 6563 6966 6963 2073  inter-specific s
-00011bd0: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
-00011be0: 2a2a 0a60 6060 0a77 6764 2073 796e 2066  **.```.wgd syn f
-00011bf0: 616d 696c 6965 7320 6766 6631 2067 6666  amilies gff1 gff
-00011c00: 3220 282d 2d61 6464 6974 696f 6e61 6c67  2 (--additionalg
-00011c10: 6666 696e 666f 2022 6d52 4e41 3b4e 616d  ffinfo "mRNA;Nam
-00011c20: 6522 202d 2d61 6464 6974 696f 6e61 6c67  e" --additionalg
-00011c30: 6666 696e 666f 2022 6765 6e65 3b49 4422  ffinfo "gene;ID"
-00011c40: 290a 6060 600a 0a46 6f72 206d 756c 7469  ).```..For multi
-00011c50: 2d73 7065 6369 6573 2073 796e 7465 6e79  -species synteny
-00011c60: 2069 6e66 6572 656e 6365 2c20 6966 2075   inference, if u
-00011c70: 7365 7273 2068 6176 6520 6766 6633 2066  sers have gff3 f
-00011c80: 696c 6573 2077 6869 6368 2068 6176 6520  iles which have 
-00011c90: 6469 6666 6572 656e 7420 6665 6174 7572  different featur
-00011ca0: 6573 206f 7220 6174 7472 6962 7574 6573  es or attributes
-00011cb0: 2066 6f72 2067 656e 6520 706f 7369 7469   for gene positi
-00011cc0: 6f6e 2069 6e66 6f72 6d61 7469 6f6e 2072  on information r
-00011cd0: 6574 7269 6576 616c 2c20 7468 6520 6f70  etrieval, the op
-00011ce0: 7469 6f6e 2060 2d2d 6164 6469 7469 6f6e  tion `--addition
-00011cf0: 616c 6766 6669 6e66 6f60 2063 616e 2062  algffinfo` can b
-00011d00: 6520 7365 7420 746f 2070 726f 7669 6465  e set to provide
-00011d10: 2074 6865 2061 6464 6974 696f 6e61 6c20   the additional 
-00011d20: 696e 666f 726d 6174 696f 6e2e 2054 6865  information. The
-00011d30: 2072 656d 6169 6e69 6e67 2070 6172 616d   remaining param
-00011d40: 6574 6572 2073 6574 7469 6e67 2069 7320  eter setting is 
-00011d50: 7468 6520 7361 6d65 2061 7320 7468 6520  the same as the 
-00011d60: 696e 7472 612d 7370 6563 6966 6963 2073  intra-specific s
-00011d70: 796e 7465 6e79 2069 6e66 6572 656e 6365  ynteny inference
-00011d80: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
-00011d90: 2073 7461 7274 696e 6720 7275 6e20 6361   starting run ca
-00011da0: 6e20 7573 6520 636f 6d6d 616e 6420 7369  n use command si
-00011db0: 6d70 6c79 2061 7320 6265 6c6f 772a 2a0a  mply as below**.
-00011dc0: 0a60 6060 0a77 6764 2073 796e 2066 616d  .```.wgd syn fam
-00011dd0: 696c 6965 7320 6766 6631 2067 6666 320a  ilies gff1 gff2.
-00011de0: 6060 600a 0a23 2323 2077 6764 2076 697a  ```..### wgd viz
-00011df0: 0a0a 2a2a 5468 6520 7669 7375 616c 697a  ..**The visualiz
-00011e00: 6174 696f 6e20 6f66 202a 4b2a 3c73 7562  ation of *K*<sub
-00011e10: 3e53 3c2f 7375 623e 2061 6765 2064 6973  >S</sub> age dis
-00011e20: 7472 6962 7574 696f 6e20 616e 6420 454c  tribution and EL
-00011e30: 4d4d 2061 6e61 6c79 7369 732a 2a0a 6060  MM analysis**.``
-00011e40: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
-00011e50: 6174 610a 6060 600a 0a54 6865 2070 726f  ata.```..The pro
-00011e60: 6772 616d 2060 7767 6420 7669 7a60 2069  gram `wgd viz` i
-00011e70: 7320 6d61 696e 6c79 2066 6f72 2074 6865  s mainly for the
-00011e80: 2070 7572 706f 7365 206f 6620 2a4b 2a3c   purpose of *K*<
-00011e90: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00011ea0: 7269 6275 7469 6f6e 2061 6e64 2073 796e  ribution and syn
-00011eb0: 7465 6e79 2076 6973 7561 6c69 7a61 7469  teny visualizati
-00011ec0: 6f6e 2c20 7769 7468 2073 6f6d 6520 6f70  on, with some op
-00011ed0: 7469 6f6e 616c 206d 6978 7475 7265 206d  tional mixture m
-00011ee0: 6f64 656c 696e 6720 616e 616c 7973 6973  odeling analysis
-00011ef0: 2e20 5468 6520 6261 7369 6320 6675 6e63  . The basic func
-00011f00: 7469 6f6e 2069 7320 6a75 7374 2074 6f20  tion is just to 
-00011f10: 706c 6f74 2074 6865 202a 4b2a 3c73 7562  plot the *K*<sub
-00011f20: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
-00011f30: 7574 696f 6e20 616e 6420 636f 6e64 7563  ution and conduc
-00011f40: 7420 616e 2045 4c4d 4d20 616e 616c 7973  t an ELMM analys
-00011f50: 6973 2069 6e20 7365 6172 6368 206f 6620  is in search of 
-00011f60: 706f 7465 6e74 6961 6c20 5747 4420 636f  potential WGD co
-00011f70: 6d70 6f6e 656e 7473 2e20 536f 6d65 206b  mponents. Some k
-00011f80: 6579 2070 6172 616d 6574 6572 7320 6166  ey parameters af
-00011f90: 6665 6374 696e 6720 7468 6520 454c 4d4d  fecting the ELMM
-00011fa0: 2072 6573 756c 7420 696e 636c 7564 6520   result include 
-00011fb0: 602d 2d70 726f 6d69 6e65 6e63 655f 6375  `--prominence_cu
-00011fc0: 746f 6666 6020 616e 6420 602d 2d72 656c  toff` and `--rel
-00011fd0: 5f68 6569 6768 7460 2c20 7768 6963 6820  _height`, which 
-00011fe0: 6861 7665 2062 6565 6e20 6578 706c 6169  have been explai
-00011ff0: 6e65 6420 6168 6561 642c 2060 2d2d 656d  ned ahead, `--em
-00012000: 5f69 7465 7261 7469 6f6e 7360 2061 6e64  _iterations` and
-00012010: 2060 2d2d 656d 5f69 6e69 7469 616c 697a   `--em_initializ
-00012020: 6174 696f 6e73 6020 6465 7465 726d 696e  ations` determin
-00012030: 696e 6720 7468 6520 6d61 7869 6d75 6d20  ing the maximum 
-00012040: 6974 6572 6174 696f 6e73 2061 6e64 2069  iterations and i
-00012050: 6e69 7469 616c 697a 6174 696f 6e73 2069  nitializations i
-00012060: 6e20 7468 6520 454d 2061 6c67 6f72 6974  n the EM algorit
-00012070: 686d 2e0a 0a2a 2a41 2073 7567 6765 7374  hm...**A suggest
-00012080: 6564 2073 7461 7274 696e 6720 7275 6e20  ed starting run 
-00012090: 6361 6e20 7573 6520 636f 6d6d 616e 6420  can use command 
-000120a0: 7369 6d70 6c79 2061 7320 6265 6c6f 772a  simply as below*
-000120b0: 2a0a 0a60 6060 0a77 6764 2076 697a 202d  *..```.wgd viz -
-000120c0: 6420 6b73 6461 7461 0a60 6060 0a0a 2a2a  d ksdata.```..**
-000120d0: 5468 6520 7669 7375 616c 697a 6174 696f  The visualizatio
-000120e0: 6e20 6f66 202a 4b2a 3c73 7562 3e53 3c2f  n of *K*<sub>S</
-000120f0: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
-00012100: 7574 696f 6e20 7769 7468 2072 6174 6520  ution with rate 
-00012110: 636f 7272 6563 7469 6f6e 2a2a 0a60 6060  correction**.```
-00012120: 0a77 6764 2076 697a 202d 6420 6b73 6461  .wgd viz -d ksda
-00012130: 7461 202d 7370 2073 7064 6174 6120 2d2d  ta -sp spdata --
-00012140: 666f 6375 7332 616c 6c20 666f 6361 6c5f  focus2all focal_
-00012150: 7370 6563 6965 7320 2d2d 6578 7472 6170  species --extrap
-00012160: 6172 616e 6f6d 656b 7320 6b73 6461 7461  aranomeks ksdata
-00012170: 2028 2d2d 616e 6368 6f72 706f 696e 7473   (--anchorpoints
-00012180: 2061 7064 6174 6120 2d2d 706c 6f74 6170   apdata --plotap
-00012190: 676d 6d20 2d2d 706c 6f74 656c 6d6d 290a  gmm --plotelmm).
-000121a0: 6060 600a 0a42 6573 6964 6573 2074 6865  ```..Besides the
-000121b0: 2062 6173 6963 202a 4b2a 3c73 7562 3e53   basic *K*<sub>S
-000121c0: 3c2f 7375 623e 2070 6c6f 742c 2073 7562  </sub> plot, sub
-000121d0: 7374 6974 7574 696f 6e20 7261 7465 2063  stitution rate c
-000121e0: 6f72 7265 6374 696f 6e20 6361 6e20 616c  orrection can al
-000121f0: 736f 2062 6520 6163 6869 6576 6564 2067  so be achieved g
-00012200: 6976 656e 2061 7420 6c65 6173 7420 6120  iven at least a 
-00012210: 7370 6563 6965 7320 7472 6565 2028 7669  species tree (vi
-00012220: 6120 7468 6520 6f70 7469 6f6e 2060 2d2d  a the option `--
-00012230: 7370 6563 6965 7374 7265 6560 2920 616e  speciestree`) an
-00012240: 6420 6120 666f 6361 6c20 7370 6563 6965  d a focal specie
-00012250: 7320 2865 6974 6865 7220 7669 6120 7468  s (either via th
-00012260: 6520 6f70 7469 6f6e 2060 2d2d 666f 6375  e option `--focu
-00012270: 7332 616c 6c60 206f 7220 7669 6120 7468  s2all` or via th
-00012280: 6520 6f70 7469 6f6e 2060 2d2d 7370 6169  e option `--spai
-00012290: 7260 2069 6e20 7468 6520 666f 726d 206f  r` in the form o
-000122a0: 6620 2224 666f 6361 6c5f 7370 6563 6965  f "$focal_specie
-000122b0: 733b 2466 6f63 616c 5f73 7065 6369 6573  s;$focal_species
-000122c0: 2229 2e20 4974 2773 2073 7567 6765 7374  "). It's suggest
-000122d0: 6564 2074 6861 7420 7468 6520 6f72 7468  ed that the orth
-000122e0: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
-000122f0: 533c 2f73 7562 3e20 6461 7461 2069 7320  S</sub> data is 
-00012300: 7072 6f76 6964 6564 2062 7920 7468 6520  provided by the 
-00012310: 602d 2d64 6174 6166 696c 6560 206f 7074  `--datafile` opt
-00012320: 696f 6e20 616e 6420 7468 6520 7061 7261  ion and the para
-00012330: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00012340: 3c2f 7375 623e 2064 6174 6120 6973 2070  </sub> data is p
-00012350: 726f 7669 6465 6420 6279 2074 6865 2060  rovided by the `
-00012360: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
-00012370: 7360 206f 7074 696f 6e2c 2061 6c74 686f  s` option, altho
-00012380: 7567 6820 6974 2773 2061 6c6c 6f77 6564  ugh it's allowed
-00012390: 2074 6f20 6f6e 6c79 2070 726f 7669 6465   to only provide
-000123a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000123b0: 2064 6174 6120 7669 6120 7468 6520 602d   data via the `-
-000123c0: 2d64 6174 6166 696c 6560 206f 7074 696f  -datafile` optio
-000123d0: 6e20 616e 6420 6465 706f 7369 7420 626f  n and deposit bo
-000123e0: 7468 206f 7274 686f 6c6f 676f 7573 2061  th orthologous a
-000123f0: 6e64 2070 6172 616c 6f67 6f75 7320 2a4b  nd paralogous *K
-00012400: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
-00012410: 7461 2074 6865 7265 6f6e 2e20 5468 6572  ta thereon. Ther
-00012420: 6520 6172 6520 7477 6f20 7479 7065 7320  e are two types 
-00012430: 6f66 206d 6978 6564 2070 6c6f 7473 2028  of mixed plots (
-00012440: 7468 6520 226d 6978 6564 2220 6865 7265  the "mixed" here
-00012450: 2072 6566 6572 7320 746f 206d 6978 6564   refers to mixed
-00012460: 206f 7274 686f 6c6f 676f 7573 2061 6e64   orthologous and
-00012470: 2070 6172 616c 6f67 6f75 7320 2a4b 2a3c   paralogous *K*<
-00012480: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00012490: 7269 6275 7469 6f6e 7329 2c20 6f6e 6520  ributions), one 
-000124a0: 6f66 2077 6869 6368 2069 7320 7369 6d69  of which is simi
-000124b0: 6c61 7220 746f 2077 6861 7420 606b 7372  lar to what `ksr
-000124c0: 6174 6573 6020 706c 6f74 732c 2077 6869  ates` plots, whi
-000124d0: 6c65 2074 6865 206f 7468 6572 206f 6620  le the other of 
-000124e0: 7768 6963 6820 6973 206c 696b 6520 7468  which is like th
-000124f0: 6520 636f 6e76 656e 7469 6f6e 616c 202a  e conventional *
-00012500: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
-00012510: 6c6f 7420 7468 6174 2062 6f74 6820 7468  lot that both th
-00012520: 6520 6f72 6967 696e 616c 206f 7274 686f  e original ortho
-00012530: 6c6f 676f 7573 2061 6e64 2070 6172 616c  logous and paral
-00012540: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-00012550: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
-00012560: 6f6e 7320 6172 6520 7472 7574 6866 756c  ons are truthful
-00012570: 6c79 2070 6c6f 7474 6564 2069 6e73 7465  ly plotted inste
-00012580: 6164 206f 6620 6a75 7374 2062 6569 6e67  ad of just being
-00012590: 2072 6570 7265 7365 6e74 6564 2062 7920   represented by 
-000125a0: 736f 6d65 2076 6572 7469 6361 6c20 6c69  some vertical li
-000125b0: 6e65 732e 2057 6520 7375 6767 6573 7420  nes. We suggest 
-000125c0: 7573 6572 7320 746f 2061 646f 7074 2074  users to adopt t
-000125d0: 6865 2060 6b73 7261 7465 7360 2d6c 696b  he `ksrates`-lik
-000125e0: 6520 706c 6f74 732c 2077 6869 6368 2069  e plots, which i
-000125f0: 7320 7468 6520 6465 6661 756c 742e 204f  s the default. O
-00012600: 7468 6572 7769 7365 2074 6865 206f 7074  therwise the opt
-00012610: 696f 6e20 602d 2d63 6c61 7373 6963 6020  ion `--classic` 
-00012620: 7769 6c6c 2062 6520 6e65 6564 6564 2074  will be needed t
-00012630: 6f20 7365 742e 2045 7874 7261 206d 6978  o set. Extra mix
-00012640: 7475 7265 206d 6f64 656c 696e 6720 616e  ture modeling an
-00012650: 616c 7973 6973 2063 616e 2062 6520 696e  alysis can be in
-00012660: 6974 6961 7465 6420 7669 6120 7468 6520  itiated via the 
-00012670: 6f70 7469 6f6e 2060 2d2d 706c 6f74 656c  option `--plotel
-00012680: 6d6d 6020 616e 6420 602d 2d70 6c6f 7461  mm` and `--plota
-00012690: 7067 6d6d 6020 7769 7468 2074 6865 2061  pgmm` with the a
-000126a0: 6e63 686f 7220 706f 696e 7473 2064 6174  nchor points dat
-000126b0: 6166 696c 6520 7072 6f76 6964 6564 2062  afile provided b
-000126c0: 7920 7468 6520 602d 2d61 6e63 686f 7270  y the `--anchorp
-000126d0: 6f69 6e74 7360 206f 7074 696f 6e2e 0a0a  oints` option...
-000126e0: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
-000126f0: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
-00012700: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
-00012710: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
-00012720: 600a 7767 6420 7669 7a20 2d64 206b 7364  `.wgd viz -d ksd
-00012730: 6174 6120 2d73 7020 7370 6461 7461 202d  ata -sp spdata -
-00012740: 2d66 6f63 7573 3261 6c6c 2066 6f63 616c  -focus2all focal
-00012750: 5f73 7065 6369 6573 202d 2d65 7874 7261  _species --extra
-00012760: 7061 7261 6e6f 6d65 6b73 206b 7364 6174  paranomeks ksdat
-00012770: 610a 6060 600a 0a2a 2a54 6865 2076 6973  a.```..**The vis
-00012780: 7561 6c69 7a61 7469 6f6e 206f 6620 7379  ualization of sy
-00012790: 6e74 656e 792a 2a0a 6060 600a 7767 6420  nteny**.```.wgd 
-000127a0: 7669 7a20 2d61 7020 6170 6461 7461 202d  viz -ap apdata -
-000127b0: 736d 2073 6d64 6174 6120 2d6d 7420 6d74  sm smdata -mt mt
-000127c0: 6461 7461 202d 6774 2067 7464 6174 6120  data -gt gtdata 
-000127d0: 2d2d 706c 6f74 7379 6e20 282d 2d6d 696e  --plotsyn (--min
-000127e0: 6c65 6e20 2d31 202d 2d6d 696e 7365 676c  len -1 --minsegl
-000127f0: 656e 2031 3030 3030 202d 2d6d 696e 6765  en 10000 --minge
-00012800: 6e65 6e75 6d20 3330 290a 6060 600a 0a43  nenum 30).```..C
-00012810: 6f6d 7061 7265 6420 746f 2074 6865 206f  ompared to the o
-00012820: 7269 6769 6e61 6c20 6077 6764 602c 2074  riginal `wgd`, t
-00012830: 6865 2060 7767 6420 7669 7a60 2070 726f  he `wgd viz` pro
-00012840: 6772 616d 2061 6464 6564 2073 796e 7465  gram added synte
-00012850: 6e79 2076 6973 7561 6c69 7a61 7469 6f6e  ny visualization
-00012860: 2070 6970 656c 696e 652e 2055 7365 7273   pipeline. Users
-00012870: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
-00012880: 2074 6865 2066 6c61 6720 6f70 7469 6f6e   the flag option
-00012890: 2060 2d2d 706c 6f74 7379 6e60 2074 6f20   `--plotsyn` to 
-000128a0: 696e 6974 6961 7465 2074 6869 7320 7061  initiate this pa
-000128b0: 7274 206f 6620 7069 7065 6c69 6e65 2e20  rt of pipeline. 
-000128c0: 5468 6973 2073 7465 7020 6173 7375 6d65  This step assume
-000128d0: 7320 7468 6174 2075 7365 7273 2068 6176  s that users hav
-000128e0: 6520 6f62 7461 696e 6564 2061 6c72 6561  e obtained alrea
-000128f0: 6479 2074 6865 2073 796e 7465 6e69 6320  dy the syntenic 
-00012900: 7265 7375 6c74 7320 6672 6f6d 2060 692d  results from `i-
-00012910: 6164 686f 7265 6020 616e 6420 7573 6573  adhore` and uses
-00012920: 2074 686f 7365 2072 6573 756c 7420 6669   those result fi
-00012930: 6c65 7320 746f 2072 6561 6c69 7a65 2074  les to realize t
-00012940: 6865 2073 796e 7465 6e79 2076 6973 7561  he synteny visua
-00012950: 6c69 7a61 7469 6f6e 2e20 5369 6d6c 6961  lization. Simlia
-00012960: 7220 746f 2060 7767 6420 7379 6e60 2c20  r to `wgd syn`, 
-00012970: 7468 6520 6578 7472 6120 2a4b 2a3c 7375  the extra *K*<su
-00012980: 623e 533c 2f73 7562 3e20 6461 7461 2066  b>S</sub> data f
-00012990: 696c 6520 6361 6e20 6265 2074 7261 6e73  ile can be trans
-000129a0: 6d69 7474 6564 2076 6961 2074 6865 206f  mitted via the o
-000129b0: 7074 696f 6e20 602d 2d64 6174 6166 696c  ption `--datafil
-000129c0: 6560 2028 696e 7374 6561 6420 6f66 2060  e` (instead of `
-000129d0: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
-000129e0: 7360 206f 7074 696f 6e29 2e20 4261 7369  s` option). Basi
-000129f0: 6361 6c6c 792c 2074 6865 2073 796e 7465  cally, the synte
-00012a00: 6e69 6320 7265 7375 6c74 2066 696c 6573  nic result files
-00012a10: 2072 6571 7569 7265 6420 6172 6520 7468   required are th
-00012a20: 6520 616e 6368 6f72 2070 6f69 6e74 7320  e anchor points 
-00012a30: 6461 7461 6669 6c65 2c20 6d75 6c74 6970  datafile, multip
-00012a40: 6c69 636f 6e73 2064 6174 6166 696c 652c  licons datafile,
-00012a50: 2067 656e 6520 7461 626c 6520 6461 7461   gene table data
-00012a60: 6669 6c65 2028 6175 746f 6d61 7465 6c79  file (automately
-00012a70: 2070 726f 6475 6365 6420 6279 2060 7767   produced by `wg
-00012a80: 6420 7379 6e60 292c 2061 6e64 2073 6567  d syn`), and seg
-00012a90: 6d65 6e74 7320 6461 7461 6669 6c65 2e0a  ments datafile..
-00012aa0: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
-00012ab0: 7461 7274 696e 6720 7275 6e20 6361 6e20  tarting run can 
-00012ac0: 7573 6520 636f 6d6d 616e 6420 7369 6d70  use command simp
-00012ad0: 6c79 2061 7320 6265 6c6f 772a 2a0a 0a60  ly as below**..`
-00012ae0: 6060 0a77 6764 2076 697a 202d 6170 2061  ``.wgd viz -ap a
-00012af0: 7064 6174 6120 2d73 6d20 736d 6461 7461  pdata -sm smdata
-00012b00: 202d 6d74 206d 7464 6174 6120 2d67 7420   -mt mtdata -gt 
-00012b10: 6774 6461 7461 202d 2d70 6c6f 7473 796e  gtdata --plotsyn
-00012b20: 0a60 6060 0a0a 2323 2049 6c6c 7573 7472  .```..## Illustr
-00012b30: 6174 696f 6e0a 0a57 6520 696c 6c75 7374  ation..We illust
-00012b40: 7261 7465 206f 7572 2070 726f 6772 616d  rate our program
-00012b50: 206f 6e20 616e 2065 7865 6d70 6c61 7279   on an exemplary
-00012b60: 2057 4744 2069 6e66 6572 656e 6365 2061   WGD inference a
-00012b70: 6e64 2064 6174 696e 6720 7570 6f6e 2073  nd dating upon s
-00012b80: 7065 6369 6573 202a 4171 7569 6c65 6769  pecies *Aquilegi
-00012b90: 6120 636f 6572 756c 6561 2a2e 0a0a 5468  a coerulea*...Th
-00012ba0: 6520 2a41 7175 696c 6567 6961 2063 6f65  e *Aquilegia coe
-00012bb0: 7275 6c65 612a 2077 6173 2072 6570 6f72  rulea* was repor
-00012bc0: 7465 6420 746f 2065 7870 6572 6965 6e63  ted to experienc
-00012bd0: 6520 616e 2070 616c 656f 2d70 6f6c 7970  e an paleo-polyp
-00012be0: 6c6f 6964 697a 6174 696f 6e20 6576 656e  loidization even
-00012bf0: 7420 6166 7465 7220 7468 6520 6469 7665  t after the dive
-00012c00: 7267 656e 6365 206f 6620 636f 7265 2065  rgence of core e
-00012c10: 7564 6963 6f74 732c 2077 6869 6368 2069  udicots, which i
-00012c20: 7320 6c69 6b65 6c79 2073 6861 7265 6420  s likely shared 
-00012c30: 6279 2061 6c6c 2052 616e 756e 6375 6c61  by all Ranuncula
-00012c40: 6c65 732e 0a0a 4669 7273 7420 6162 6f76  les...First abov
-00012c50: 6520 616c 6c2c 206c 6574 2773 2064 656c  e all, let's del
-00012c60: 696e 6561 7465 2074 6865 2077 686f 6c65  ineate the whole
-00012c70: 2070 6172 616e 6f6d 6520 2a4b 2a3c 7375   paranome *K*<su
-00012c80: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
-00012c90: 7374 7269 6275 7469 6f6e 2061 6e64 2068  stribution and h
-00012ca0: 6176 6520 6120 6261 7369 6320 6f62 7365  ave a basic obse
-00012cb0: 7276 6174 696f 6e20 666f 7220 706f 7465  rvation for pote
-00012cc0: 6e74 6961 6c6c 7920 636f 6e63 6569 7661  ntially conceiva
-00012cd0: 626c 6520 5747 4473 2c20 7573 696e 6720  ble WGDs, using 
-00012ce0: 7468 6520 636f 6d6d 616e 6420 6c69 6e65  the command line
-00012cf0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-00012d00: 2064 6d64 2041 7175 696c 6567 6961 5f63   dmd Aquilegia_c
-00012d10: 6f65 7275 6c65 610a 7767 6420 6b73 6420  oerulea.wgd ksd 
-00012d20: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
-00012d30: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
-00012d40: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00012d50: 610a 6060 600a 0a54 6865 2063 6f6e 7374  a.```..The const
-00012d60: 7275 6374 6564 2077 686f 6c65 2070 6172  ructed whole par
-00012d70: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
-00012d80: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
-00012d90: 6275 7469 6f6e 206f 6620 2a41 7175 696c  bution of *Aquil
-00012da0: 6567 6961 2063 6f65 7275 6c65 612a 2069  egia coerulea* i
-00012db0: 7320 6173 2062 656c 6f77 2c20 7765 2063  s as below, we c
-00012dc0: 616e 2073 6565 2074 6861 7420 7468 6572  an see that ther
-00012dd0: 6520 7365 656d 7320 746f 2062 6520 6120  e seems to be a 
-00012de0: 6875 6d70 2061 7420 2a4b 2a3c 7375 623e  hump at *K*<sub>
-00012df0: 533c 2f73 7562 3e20 3120 6275 7420 6e6f  S</sub> 1 but no
-00012e00: 7420 636c 6561 722e 0a0a 215b 5d28 6461  t clear...![](da
-00012e10: 7461 2f41 7175 696c 6567 6961 5f63 6f65  ta/Aquilegia_coe
-00012e20: 7275 6c65 612e 7473 762e 6b73 645f 7770  rulea.tsv.ksd_wp
-00012e30: 2e73 7667 290a 0a57 6520 7468 656e 2063  .svg)..We then c
-00012e40: 6f6e 7374 7275 6374 2074 6865 2061 6e63  onstruct the anc
-00012e50: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
-00012e60: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
-00012e70: 7469 6f6e 2075 7369 6e67 2074 6865 2063  tion using the c
-00012e80: 6f6d 6d61 6e64 206c 696e 6520 6265 6c6f  ommand line belo
-00012e90: 772e 0a0a 6060 600a 7767 6420 7379 6e20  w...```.wgd syn 
-00012ea0: 2d66 206d 524e 4120 2d61 204e 616d 6520  -f mRNA -a Name 
-00012eb0: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
-00012ec0: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
-00012ed0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00012ee0: 612e 6766 6633 202d 6b73 2077 6764 5f6b  a.gff3 -ks wgd_k
-00012ef0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
-00012f00: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
-00012f10: 0a60 6060 0a0a 4173 2073 686f 776e 2062  .```..As shown b
-00012f20: 656c 6f77 2c20 7468 6572 6520 6172 6520  elow, there are 
-00012f30: 736f 6d65 2072 6574 6169 6e65 6420 616e  some retained an
-00012f40: 6368 6f72 2070 6169 7273 2077 6974 6820  chor pairs with 
-00012f50: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00012f60: 6265 7477 6565 6e20 3120 616e 6420 322c  between 1 and 2,
-00012f70: 2077 6869 6368 2073 6565 6d73 2074 6f20   which seems to 
-00012f80: 7375 6767 6573 7420 6120 5747 4420 6576  suggest a WGD ev
-00012f90: 656e 742e 0a0a 215b 5d28 6461 7461 2f41  ent...![](data/A
-00012fa0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00012fb0: 612e 7473 762e 6b73 645f 7770 5f61 702e  a.tsv.ksd_wp_ap.
-00012fc0: 7376 6729 0a0a 5468 6520 6173 736f 6369  svg)..The associ
-00012fd0: 6174 6564 2060 6475 7053 7461 636b 6020  ated `dupStack` 
-00012fe0: 706c 6f74 2073 686f 7773 2074 6861 7420  plot shows that 
-00012ff0: 7468 6572 6520 6172 6520 6e75 6d65 726f  there are numero
-00013000: 7573 2064 7570 6c69 6361 7465 6420 7365  us duplicated se
-00013010: 676d 656e 7473 2061 6372 6f73 7320 6d6f  gments across mo
-00013020: 7374 206f 6620 7468 6520 6368 726f 6d6f  st of the chromo
-00013030: 736f 6d65 732e 0a0a 215b 5d28 6461 7461  somes...![](data
-00013040: 2f73 796e 5f72 6573 756c 7473 2f41 7175  /syn_results/Aqu
-00013050: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
-00013060: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013070: 6561 5f6d 756c 7469 706c 6963 6f6e 735f  ea_multiplicons_
-00013080: 6c65 7665 6c2e 706e 6729 0a0a 5765 2069  level.png)..We i
-00013090: 6d70 6c65 6d65 6e74 6564 2074 776f 2074  mplemented two t
-000130a0: 7970 6573 206f 6620 646f 7420 706c 6f74  ypes of dot plot
-000130b0: 7320 696e 206f 7866 6f72 6420 6772 6964  s in oxford grid
-000130c0: 3a20 6f6e 6520 696e 2074 6865 2075 6e69  : one in the uni
-000130d0: 7420 6f66 2062 6173 6573 2061 6e64 2074  t of bases and t
-000130e0: 6865 206f 7468 6572 2069 6e20 7468 6520  he other in the 
-000130f0: 756e 6974 206f 6620 6765 6e65 732c 2077  unit of genes, w
-00013100: 6869 6368 2063 616e 2062 6520 636f 6c6f  hich can be colo
-00013110: 7265 6420 6279 202a 4b2a 3c73 7562 3e53  red by *K*<sub>S
-00013120: 3c2f 7375 623e 2076 616c 7565 7320 6769  </sub> values gi
-00013130: 7665 6e20 2a4b 2a3c 7375 623e 533c 2f73  ven *K*<sub>S</s
-00013140: 7562 3e20 6461 7461 2e0a 0a21 5b5d 2864  ub> data...![](d
-00013150: 6174 612f 7379 6e5f 7265 7375 6c74 732f  ata/syn_results/
-00013160: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00013170: 6561 2d76 732d 4171 7569 6c65 6769 615f  ea-vs-Aquilegia_
-00013180: 636f 6572 756c 6561 5f4b 732e 646f 745f  coerulea_Ks.dot_
-00013190: 756e 6974 5f67 656e 652e 706e 6729 0a0a  unit_gene.png)..
-000131a0: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
-000131b0: 7468 6520 646f 7420 706c 6f74 2069 6e20  the dot plot in 
-000131c0: 7468 6520 756e 6974 206f 6620 6765 6e65  the unit of gene
-000131d0: 7320 7072 6573 656e 7473 206e 756d 6572  s presents numer
-000131e0: 6f75 7320 6465 6e73 656c 7920 6167 6772  ous densely aggr
-000131f0: 6567 6174 6564 2028 6c69 6e65 2d6c 696b  egated (line-lik
-00013200: 6529 2061 6e63 686f 7220 706f 696e 7473  e) anchor points
-00013210: 2061 7420 6d6f 7374 206f 6620 7468 6520   at most of the 
-00013220: 6368 726f 6d6f 736f 6d65 7320 7769 7468  chromosomes with
-00013230: 2063 6f6e 7369 7374 656e 7420 2a4b 2a3c   consistent *K*<
-00013240: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
-00013250: 6265 7477 6565 6e20 3120 616e 6420 322e  between 1 and 2.
-00013260: 2054 6865 2064 6f74 2070 6c6f 7420 696e   The dot plot in
-00013270: 2074 6865 2075 6e69 7420 6f66 2062 6173   the unit of bas
-00013280: 6573 2073 686f 7773 2074 6865 2073 616d  es shows the sam
-00013290: 6520 7061 7474 6572 6e2c 2061 7320 6d61  e pattern, as ma
-000132a0: 6e69 6665 7374 6564 2062 656c 6f77 2e0a  nifested below..
-000132b0: 0a21 5b5d 2864 6174 612f 7379 6e5f 7265  .![](data/syn_re
-000132c0: 7375 6c74 732f 4171 7569 6c65 6769 615f  sults/Aquilegia_
-000132d0: 636f 6572 756c 6561 2d76 732d 4171 7569  coerulea-vs-Aqui
-000132e0: 6c65 6769 615f 636f 6572 756c 6561 5f4b  legia_coerulea_K
-000132f0: 732e 646f 742e 706e 6729 0a0a 5468 6520  s.dot.png)..The 
-00013300: 646f 7420 706c 6f74 7320 7769 7468 6f75  dot plots withou
-00013310: 7420 2a4b 2a3c 7375 623e 533c 2f73 7562  t *K*<sub>S</sub
-00013320: 3e20 616e 6e6f 7461 7469 6f6e 2077 696c  > annotation wil
-00013330: 6c20 616c 736f 2062 6520 6175 746f 6d61  l also be automa
-00013340: 7465 6c79 2070 726f 6475 6365 642c 2061  tely produced, a
-00013350: 7320 7368 6f77 6e20 6265 6c6f 772e 0a0a  s shown below...
-00013360: 215b 5d28 6461 7461 2f73 796e 5f72 6573  ![](data/syn_res
-00013370: 756c 7473 2f41 7175 696c 6567 6961 5f63  ults/Aquilegia_c
-00013380: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
-00013390: 6567 6961 5f63 6f65 7275 6c65 612e 646f  egia_coerulea.do
-000133a0: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
-000133b0: 0a0a 215b 5d28 6461 7461 2f73 796e 5f72  ..![](data/syn_r
-000133c0: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
-000133d0: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
-000133e0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
-000133f0: 646f 742e 706e 6729 0a0a 4e6f 7465 2074  dot.png)..Note t
-00013400: 6861 7420 7468 6520 6f70 6163 6974 7920  hat the opacity 
-00013410: 6f66 2061 6e63 686f 7220 646f 7473 2061  of anchor dots a
-00013420: 6e64 2061 6c6c 2068 6f6d 6f6c 6f67 2064  nd all homolog d
-00013430: 6f74 7320 6361 6e20 6265 2073 6574 2062  ots can be set b
-00013440: 7920 7468 6520 6f70 7469 6f6e 2060 2d2d  y the option `--
-00013450: 6170 616c 7068 6160 2061 6e64 2060 2d2d  apalpha` and `--
-00013460: 686f 616c 7068 6160 2073 6570 6172 6174  hoalpha` separat
-00013470: 656c 792e 2049 6620 6f6e 6520 6a75 7374  ely. If one just
-00013480: 2077 616e 7473 2074 6f20 7365 6520 7468   wants to see th
-00013490: 6520 616e 6368 6f72 2064 6f74 732c 2073  e anchor dots, s
-000134a0: 6574 7469 6e67 2074 6865 2060 686f 616c  etting the `hoal
-000134b0: 7068 6160 2061 7320 3020 286f 7220 6f74  pha` as 0 (or ot
-000134c0: 6865 7220 6d69 6e75 7363 756c 6520 7661  her minuscule va
-000134d0: 6c75 6573 2920 7769 6c6c 2064 6f2e 2049  lues) will do. I
-000134e0: 6620 6f6e 6520 7761 6e74 7320 746f 2073  f one wants to s
-000134f0: 6565 2074 6865 2064 6973 7472 6962 7574  ee the distribut
-00013500: 696f 6e20 6f66 2077 686f 6c65 2064 6f74  ion of whole dot
-00013510: 7320 6265 7474 6572 2c20 7365 7474 696e  s better, settin
-00013520: 6720 7468 6520 6068 6f61 6c70 6861 6020  g the `hoalpha` 
-00013530: 6869 6768 6572 2028 616e 6420 6061 7061  higher (and `apa
-00013540: 6c70 6861 6020 6c6f 7765 7229 2077 696c  lpha` lower) wil
-00013550: 6c20 646f 2e20 5468 6520 6064 6f74 7369  l do. The `dotsi
-00013560: 7a65 6020 6f70 7469 6f6e 2063 616e 2062  ze` option can b
-00013570: 6520 6361 6c6c 6564 2074 6f20 6164 6a75  e called to adju
-00013580: 7374 2074 6865 2073 697a 6520 6f66 2064  st the size of d
-00013590: 6f74 732e 0a0a 4120 6675 7274 6865 7220  ots...A further 
-000135a0: 6173 736f 6369 6174 6564 2053 796e 6465  associated Synde
-000135b0: 7074 6820 706c 6f74 2073 686f 7773 2074  pth plot shows t
-000135c0: 6861 7420 7468 6572 6520 6172 6520 6d6f  hat there are mo
-000135d0: 7265 2074 6861 6e20 3530 2064 7570 6c69  re than 50 dupli
-000135e0: 6361 7465 6420 7365 676d 656e 7473 206c  cated segments l
-000135f0: 6f6e 6765 7220 7468 616e 2031 3030 3030  onger than 10000
-00013600: 2062 7020 616e 6420 3330 2067 656e 6573   bp and 30 genes
-00013610: 2028 736f 2061 7320 746f 2064 726f 7020   (so as to drop 
-00013620: 6672 6167 6d65 6e74 6172 7920 7365 676d  fragmentary segm
-00013630: 656e 7473 292c 2077 6869 6368 2064 6f6d  ents), which dom
-00013640: 696e 6174 6573 2074 6865 2077 686f 6c65  inates the whole
-00013650: 2063 6f6c 6c69 6e65 6172 2072 6174 696f   collinear ratio
-00013660: 2063 6174 6567 6f72 792e 0a0a 215b 5d28   category...![](
-00013670: 6461 7461 2f73 796e 5f72 6573 756c 7473  data/syn_results
-00013680: 2f53 796e 6465 7074 682e 7376 6729 0a0a  /Syndepth.svg)..
-00013690: 4d6f 7265 2065 7871 7569 7369 7465 2070  More exquisite p
-000136a0: 6c6f 7473 2069 6e63 6c75 6469 6e67 2062  lots including b
-000136b0: 6f74 6820 696e 7472 612d 7370 6563 6966  oth intra-specif
-000136c0: 6963 2061 6e64 2069 6e74 6572 2d73 7065  ic and inter-spe
-000136d0: 6369 6669 6320 636f 6d70 6172 6973 6f6e  cific comparison
-000136e0: 7320 7573 696e 6720 7468 6520 6f72 7468  s using the orth
-000136f0: 6f67 726f 7570 7320 2863 6f6d 706f 7365  ogroups (compose
-00013700: 6420 6f66 202a 4171 7569 6c65 6769 6120  d of *Aquilegia 
-00013710: 636f 6572 756c 6561 2a2c 202a 5072 6f74  coerulea*, *Prot
-00013720: 6561 2063 796e 6172 6f69 6465 732a 2c20  ea cynaroides*, 
-00013730: 2a41 636f 7275 7320 616d 6572 6963 616e  *Acorus american
-00013740: 7573 2a20 616e 6420 2a56 6974 6973 2076  us* and *Vitis v
-00013750: 696e 6966 6572 612a 2c20 7365 6520 6675  inifera*, see fu
-00013760: 7274 6865 7220 666f 7220 7468 6520 636f  rther for the co
-00013770: 6e74 6578 7429 2069 6e66 6572 7265 6420  ntext) inferred 
-00013780: 6865 7265 696e 6166 7465 7220 6361 6e20  hereinafter can 
-00013790: 6265 2061 6c73 6f20 7072 6f64 7563 6564  be also produced
-000137a0: 2075 7369 6e67 2060 7767 6420 7379 6e60   using `wgd syn`
-000137b0: 2e20 4e6f 7465 2074 6861 7420 6469 6666  . Note that diff
-000137c0: 6572 656e 7420 6765 6e6f 6d65 2061 7373  erent genome ass
-000137d0: 656d 626c 6965 7320 6d69 6768 7420 6861  emblies might ha
-000137e0: 7665 2064 6966 6665 7265 6e74 2066 6561  ve different fea
-000137f0: 7475 7265 7320 616e 6420 6174 7472 6962  tures and attrib
-00013800: 7574 6573 2077 6869 6368 2063 616e 2062  utes which can b
-00013810: 6520 6163 636f 6d6d 6f64 6174 6564 2076  e accommodated v
-00013820: 6961 2074 6865 206f 7074 696f 6e20 602d  ia the option `-
-00013830: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
-00013840: 666f 6020 666f 7220 6561 6368 2067 656e  fo` for each gen
-00013850: 6f6d 6520 6173 7365 6d62 6c79 2077 686f  ome assembly who
-00013860: 7365 206f 7264 6572 206e 6565 6473 2074  se order needs t
-00013870: 6f20 666f 6c6c 6f77 2074 6865 206f 7264  o follow the ord
-00013880: 6572 206f 6620 6766 6633 2066 696c 6573  er of gff3 files
-00013890: 2c20 666f 7220 696e 7374 616e 6365 2027  , for instance '
-000138a0: 6d4e 5241 3b4e 616d 6527 2066 6f72 2060  mNRA;Name' for `
-000138b0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-000138c0: 6561 2e67 6666 3360 2c20 276d 4e52 413b  ea.gff3`, 'mNRA;
-000138d0: 4944 2720 666f 7220 6050 726f 7465 615f  ID' for `Protea_
-000138e0: 6379 6e61 726f 6964 6573 2e67 6666 3360  cynaroides.gff3`
-000138f0: 2c20 276d 4e52 413b 4e61 6d65 2720 666f  , 'mNRA;Name' fo
-00013900: 7220 6041 636f 7275 735f 616d 6572 6963  r `Acorus_americ
-00013910: 616e 7573 2e67 6666 3360 2061 6e64 2027  anus.gff3` and '
-00013920: 6d4e 5241 3b4e 616d 6527 2066 6f72 2060  mNRA;Name' for `
-00013930: 5669 7469 735f 7669 6e69 6665 7261 2e67  Vitis_vinifera.g
-00013940: 6666 3360 2e0a 0a60 6060 0a77 6764 2073  ff3`...```.wgd s
-00013950: 796e 2077 6764 5f6f 7274 686f 2f4f 7274  yn wgd_ortho/Ort
-00013960: 686f 6772 6f75 7073 2e73 702e 7473 7620  hogroups.sp.tsv 
-00013970: 2d6b 7320 7767 645f 6f72 7468 6f5f 6b73  -ks wgd_ortho_ks
-00013980: 2f4f 7274 686f 6772 6f75 7073 2e73 702e  /Orthogroups.sp.
-00013990: 7473 762e 6b73 2e74 7376 2041 7175 696c  tsv.ks.tsv Aquil
-000139a0: 6567 6961 5f63 6f65 7275 6c65 612e 6766  egia_coerulea.gf
-000139b0: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
-000139c0: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
-000139d0: 6527 2050 726f 7465 615f 6379 6e61 726f  e' Protea_cynaro
-000139e0: 6964 6573 2e67 6666 3320 2d2d 6164 6469  ides.gff3 --addi
-000139f0: 7469 6f6e 616c 6766 6669 6e66 6f20 276d  tionalgffinfo 'm
-00013a00: 4e52 413b 4944 2720 4163 6f72 7573 5f61  NRA;ID' Acorus_a
-00013a10: 6d65 7269 6361 6e75 732e 6766 6633 202d  mericanus.gff3 -
-00013a20: 2d61 6464 6974 696f 6e61 6c67 6666 696e  -additionalgffin
-00013a30: 666f 2027 6d4e 5241 3b4e 616d 6527 2056  fo 'mNRA;Name' V
-00013a40: 6974 6973 5f76 696e 6966 6572 612e 6766  itis_vinifera.gf
-00013a50: 6633 202d 2d61 6464 6974 696f 6e61 6c67  f3 --additionalg
-00013a60: 6666 696e 666f 2027 6d4e 5241 3b4e 616d  ffinfo 'mNRA;Nam
-00013a70: 6527 202d 6f20 7767 645f 6f72 7468 6f5f  e' -o wgd_ortho_
-00013a80: 7379 6e0a 6060 600a 0a55 706f 6e20 7468  syn.```..Upon th
-00013a90: 6520 6163 7175 6973 6974 696f 6e20 6f66  e acquisition of
-00013aa0: 2074 6865 2063 6f6c 6c69 6e65 6172 2072   the collinear r
-00013ab0: 6573 756c 7473 2075 7369 6e67 2060 7767  esults using `wg
-00013ac0: 6420 7379 6e60 2c20 7468 6520 7361 6d65  d syn`, the same
-00013ad0: 2063 6f6c 6c69 6e65 6172 2070 6c6f 7473   collinear plots
-00013ae0: 2063 616e 2062 6520 616c 736f 2070 726f   can be also pro
-00013af0: 6475 6365 6420 6279 2060 7767 6420 7669  duced by `wgd vi
-00013b00: 7a60 2075 7369 6e67 2074 6865 2063 6f6d  z` using the com
-00013b10: 6d61 6e64 2062 656c 6f77 2e0a 6060 600a  mand below..```.
-00013b20: 7767 6420 7669 7a20 2d2d 706c 6f74 7379  wgd viz --plotsy
-00013b30: 6e20 2d73 6d20 7767 645f 6f72 7468 6f5f  n -sm wgd_ortho_
-00013b40: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
-00013b50: 7365 676d 656e 7473 2e74 7874 202d 6170  segments.txt -ap
-00013b60: 2077 6764 5f6f 7274 686f 5f73 796e 2f69   wgd_ortho_syn/i
-00013b70: 6164 686f 7265 2d6f 7574 2f61 6e63 686f  adhore-out/ancho
-00013b80: 7270 6f69 6e74 732e 7478 7420 2d6d 7420  rpoints.txt -mt 
-00013b90: 7767 645f 6f72 7468 6f5f 7379 6e2f 6961  wgd_ortho_syn/ia
-00013ba0: 6468 6f72 652d 6f75 742f 6d75 6c74 6970  dhore-out/multip
-00013bb0: 6c69 636f 6e73 2e74 7874 202d 6774 2077  licons.txt -gt w
-00013bc0: 6764 5f6f 7274 686f 5f73 796e 2f67 656e  gd_ortho_syn/gen
-00013bd0: 652d 7461 626c 652e 6373 7620 2d64 2077  e-table.csv -d w
-00013be0: 6764 5f6f 7274 686f 5f6b 732f 4f72 7468  gd_ortho_ks/Orth
-00013bf0: 6f67 726f 7570 732e 7370 2e74 7376 2e6b  ogroups.sp.tsv.k
-00013c00: 732e 7473 7620 2d6f 2077 6764 5f6f 7274  s.tsv -o wgd_ort
-00013c10: 686f 5f76 697a 0a60 6060 0a0a 215b 5d28  ho_viz.```..![](
-00013c20: 6461 7461 2f6f 7274 686f 5f73 796e 5f72  data/ortho_syn_r
-00013c30: 6573 756c 7473 2f41 7175 696c 6567 6961  esults/Aquilegia
-00013c40: 5f63 6f65 7275 6c65 615f 5669 7469 735f  _coerulea_Vitis_
-00013c50: 7669 6e69 6665 7261 5f6d 756c 7469 706c  vinifera_multipl
-00013c60: 6963 6f6e 735f 6c65 7665 6c2e 706e 6729  icons_level.png)
-00013c70: 0a54 6865 2061 626f 7665 2060 6475 7053  .The above `dupS
-00013c80: 7461 636b 6020 706c 6f74 2073 686f 7773  tack` plot shows
-00013c90: 2074 6865 2064 6973 7472 6962 7574 696f   the distributio
-00013ca0: 6e20 6f66 2064 7570 6c69 6361 7465 6420  n of duplicated 
-00013cb0: 7365 676d 656e 7473 206f 6620 2a41 7175  segments of *Aqu
-00013cc0: 696c 6567 6961 2063 6f65 7275 6c65 612a  ilegia coerulea*
-00013cd0: 2063 6f6d 7061 7265 6420 746f 2069 7473   compared to its
-00013ce0: 656c 6620 2869 6e20 6772 6565 6e29 2061  elf (in green) a
-00013cf0: 6e64 2063 6f6d 7061 7265 6420 746f 202a  nd compared to *
-00013d00: 5669 7469 7320 7669 6e69 6665 7261 2a20  Vitis vinifera* 
-00013d10: 2869 6e20 626c 7565 2920 6f76 6572 2074  (in blue) over t
-00013d20: 6865 2063 6872 6f6d 6f73 6f6d 6573 206f  he chromosomes o
-00013d30: 6620 2a41 2e20 636f 6572 756c 6561 2a2e  f *A. coerulea*.
-00013d40: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
-00013d50: 7379 6e5f 7265 7375 6c74 732f 4f76 6572  syn_results/Over
-00013d60: 616c 6c73 7065 6369 6573 5f4b 732e 646f  allspecies_Ks.do
-00013d70: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
-00013d80: 0a54 6865 2061 626f 7665 202a 4b2a 3c73  .The above *K*<s
-00013d90: 7562 3e53 3c2f 7375 623e 2064 6f74 706c  ub>S</sub> dotpl
-00013da0: 6f74 2069 6e20 756e 6974 206f 6620 6765  ot in unit of ge
-00013db0: 6e65 2073 686f 7773 2074 6865 206f 7665  ne shows the ove
-00013dc0: 7261 6c6c 2064 6973 7472 6962 7574 696f  rall distributio
-00013dd0: 6e20 6f66 2063 6f6c 6c69 6e65 6172 6974  n of collinearit
-00013de0: 7920 6163 6f72 7373 2074 6865 2066 6f75  y acorss the fou
-00013df0: 7220 7370 6563 6965 7320 696e 766f 6c76  r species involv
-00013e00: 6564 2e0a 215b 5d28 6461 7461 2f6f 7274  ed..![](data/ort
-00013e10: 686f 5f73 796e 5f72 6573 756c 7473 2f4f  ho_syn_results/O
-00013e20: 7665 7261 6c6c 7370 6563 6965 732e 646f  verallspecies.do
-00013e30: 745f 756e 6974 5f67 656e 652e 706e 6729  t_unit_gene.png)
-00013e40: 0a54 6865 2061 626f 7665 2064 6f74 706c  .The above dotpl
-00013e50: 6f74 2069 7320 7769 7468 6f75 7420 7468  ot is without th
-00013e60: 6520 616e 6e6f 7461 7469 6f6e 206f 6620  e annotation of 
-00013e70: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
-00013e80: 6167 6573 2063 6f6d 7061 7265 6420 746f  ages compared to
-00013e90: 2074 6865 206c 6173 7420 6f6e 652e 0a21   the last one..!
-00013ea0: 5b5d 2864 6174 612f 6f72 7468 6f5f 7379  [](data/ortho_sy
-00013eb0: 6e5f 7265 7375 6c74 732f 5379 6e64 6570  n_results/Syndep
-00013ec0: 7468 2e73 7667 290a 5468 6520 6162 6f76  th.svg).The abov
-00013ed0: 6520 5379 6e64 6570 7468 2070 6c6f 7420  e Syndepth plot 
-00013ee0: 7368 6f77 7320 7468 6520 636f 6c6c 696e  shows the collin
-00013ef0: 6561 7220 7261 7469 6f20 6163 6f72 7373  ear ratio acorss
-00013f00: 2061 6c6c 2073 7065 6369 6573 2070 6169   all species pai
-00013f10: 7273 2028 696e 7472 612d 7370 6563 6966  rs (intra-specif
-00013f20: 6963 2063 6f6d 7061 7269 736f 6e20 696e  ic comparison in
-00013f30: 2067 7265 656e 2077 6869 6c65 2069 6e74   green while int
-00013f40: 6572 2d73 7065 6369 6669 6320 636f 6d70  er-specific comp
-00013f50: 6172 6973 6f6e 2069 6e20 626c 7565 292e  arison in blue).
-00013f60: 0a0a 5765 2063 616e 2066 6974 2061 6e20  ..We can fit an 
-00013f70: 454c 4d4d 206d 6978 7475 7265 206d 6f64  ELMM mixture mod
-00013f80: 656c 2075 706f 6e20 7468 6520 7768 6f6c  el upon the whol
-00013f90: 6520 7061 7261 6e6f 6d65 202a 4b2a 3c73  e paranome *K*<s
-00013fa0: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
-00013fb0: 6973 7472 6962 7574 696f 6e20 746f 2073  istribution to s
-00013fc0: 6565 206d 6f72 6520 6163 6375 7261 7465  ee more accurate
-00013fd0: 6c79 2074 6865 2073 6967 6e69 6669 6361  ly the significa
-00013fe0: 6e63 6520 616e 6420 6c6f 6361 7469 6f6e  nce and location
-00013ff0: 206f 6620 706f 7465 6e74 6961 6c20 5747   of potential WG
-00014000: 4473 2c20 7573 696e 6720 7468 6520 636f  Ds, using the co
-00014010: 6d6d 616e 6420 6c69 6e65 2062 656c 6f77  mmand line below
-00014020: 2e0a 0a60 6060 0a77 6764 2076 697a 202d  ...```.wgd viz -
-00014030: 6420 7767 645f 6b73 642f 4171 7569 6c65  d wgd_ksd/Aquile
-00014040: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
-00014050: 2e6b 732e 7473 760a 6060 600a 0a54 6865  .ks.tsv.```..The
-00014060: 2072 6573 756c 7420 6f66 2045 4c4d 4d20   result of ELMM 
-00014070: 6d69 7874 7572 6520 6d6f 6465 6c20 636c  mixture model cl
-00014080: 7573 7465 7269 6e67 2073 686f 7773 2074  ustering shows t
-00014090: 6861 7420 7468 6572 6520 6973 2061 206c  hat there is a l
-000140a0: 696b 656c 7920 5747 4420 636f 6d70 6f6e  ikely WGD compon
-000140b0: 656e 7420 6174 202a 4b2a 3c73 7562 3e53  ent at *K*<sub>S
-000140c0: 3c2f 7375 623e 2031 2e31 392e 0a0a 215b  </sub> 1.19...![
-000140d0: 5d28 6461 7461 2f65 6c6d 6d5f 4171 7569  ](data/elmm_Aqui
-000140e0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
-000140f0: 7376 2e6b 732e 7473 765f 6265 7374 5f6d  sv.ks.tsv_best_m
-00014100: 6f64 656c 735f 7765 6967 6874 6564 2e73  odels_weighted.s
-00014110: 7667 290a 0a4c 6574 2773 2064 6f20 6120  vg)..Let's do a 
-00014120: 6d69 7874 7572 6520 6d6f 6465 6c20 636c  mixture model cl
-00014130: 7573 7465 7269 6e67 2066 6f72 2061 6e63  ustering for anc
-00014140: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
-00014150: 7562 3e20 746f 6f2c 2075 7369 6e67 2074  ub> too, using t
-00014160: 6865 2063 6f6d 6d61 6e64 206c 696e 6520  he command line 
-00014170: 6265 6c6f 772e 204e 6f74 6520 7468 6174  below. Note that
-00014180: 2074 6869 7320 7374 6570 2077 696c 6c20   this step will 
-00014190: 6175 746f 6d61 7465 6c79 2063 616c 6c20  automately call 
-000141a0: 7468 6520 7365 676d 656e 7420 2a4b 2a3c  the segment *K*<
-000141b0: 7375 623e 533c 2f73 7562 3e20 636c 7573  sub>S</sub> clus
-000141c0: 7465 7269 6e67 2061 6e61 6c79 7369 7320  tering analysis 
-000141d0: 746f 6f2e 0a0a 6060 600a 7767 6420 7065  too...```.wgd pe
-000141e0: 616b 2077 6764 5f6b 7364 2f41 7175 696c  ak wgd_ksd/Aquil
-000141f0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-00014200: 762e 6b73 2e74 7376 202d 2d61 6e63 686f  v.ks.tsv --ancho
-00014210: 7270 6f69 6e74 7320 7767 645f 7379 6e2f  rpoints wgd_syn/
-00014220: 6961 6468 6f72 652d 6f75 742f 616e 6368  iadhore-out/anch
-00014230: 6f72 706f 696e 7473 2e74 7874 202d 2d73  orpoints.txt --s
-00014240: 6567 6d65 6e74 7320 7767 645f 7379 6e2f  egments wgd_syn/
-00014250: 6961 6468 6f72 652d 6f75 742f 7365 676d  iadhore-out/segm
-00014260: 656e 7473 2e74 7874 202d 2d6c 6973 7465  ents.txt --liste
-00014270: 6c65 6d65 6e74 7320 7767 645f 7379 6e2f  lements wgd_syn/
-00014280: 6961 6468 6f72 652d 6f75 742f 6c69 7374  iadhore-out/list
-00014290: 5f65 6c65 6d65 6e74 732e 7478 7420 2d2d  _elements.txt --
-000142a0: 6d75 6c74 6970 6c69 636f 6e70 6169 7273  multipliconpairs
-000142b0: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
-000142c0: 2d6f 7574 2f6d 756c 7469 706c 6963 6f6e  -out/multiplicon
-000142d0: 5f70 6169 7273 2e74 7874 2028 2d2d 7765  _pairs.txt (--we
-000142e0: 6967 6874 6564 290a 6060 600a 0a54 6865  ighted).```..The
-000142f0: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
-00014300: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
-00014310: 7269 6275 7469 6f6e 2061 6c73 6f20 6861  ribution also ha
-00014320: 7320 6120 6c69 6b65 6c79 2057 4744 2063  s a likely WGD c
-00014330: 6f6d 706f 6e65 6e74 2077 6974 6820 6d6f  omponent with mo
-00014340: 6465 2031 2e32 382e 0a0a 215b 5d28 6461  de 1.28...![](da
-00014350: 7461 2f4f 7269 6769 6e61 6c5f 416e 6368  ta/Original_Anch
-00014360: 6f72 4b73 5f47 4d4d 5f43 6f6d 706f 6e65  orKs_GMM_Compone
-00014370: 6e74 335f 6e6f 6465 5f61 7665 7261 6765  nt3_node_average
-00014380: 645f 4c6f 676e 6f72 6d61 6c2e 7376 6729  d_Lognormal.svg)
-00014390: 0a0a 4e6f 7720 7468 6174 2077 6520 6861  ..Now that we ha
-000143a0: 7665 2073 6565 6e20 7468 6520 6576 6964  ve seen the evid
-000143b0: 656e 6365 206f 6620 6e75 6d65 726f 7573  ence of numerous
-000143c0: 2064 7570 6c69 6361 7465 6420 7365 676d   duplicated segm
-000143d0: 656e 7473 2061 6e64 2074 6865 2061 6767  ents and the agg
-000143e0: 7265 6761 7469 6f6e 206f 6620 6475 706c  regation of dupl
-000143f0: 6963 6174 6573 2061 6765 2061 7420 2a4b  icates age at *K
-00014400: 2a3c 7375 623e 533c 2f73 7562 3e20 312e  *<sub>S</sub> 1.
-00014410: 3238 206f 7220 312e 3139 2066 6f72 2061  28 or 1.19 for a
-00014420: 6e63 686f 7220 7061 6972 7320 616e 6420  nchor pairs and 
-00014430: 6e6f 6e2d 616e 6368 6f72 2070 6169 7273  non-anchor pairs
-00014440: 2074 6872 6f75 6768 6f75 7420 7468 6520   throughout the 
-00014450: 7768 6f6c 6520 6765 6e6f 6d65 2e20 5765  whole genome. We
-00014460: 2063 616e 2063 6c61 696d 2077 6974 6820   can claim with 
-00014470: 736f 6d65 2063 6f6e 6669 6465 6e63 6520  some confidence 
-00014480: 7468 6174 202a 4171 7569 6c65 6769 6120  that *Aquilegia 
-00014490: 636f 6572 756c 6561 2a20 6d69 6768 7420  coerulea* might 
-000144a0: 6861 7665 2065 7870 6572 6965 6e63 6564  have experienced
-000144b0: 2061 2070 616c 656f 2d70 6f6c 7970 6c6f   a paleo-polyplo
-000144c0: 6964 697a 6174 696f 6e20 6576 656e 742e  idization event.
-000144d0: 204e 6578 742c 204c 6574 2773 2068 6176   Next, Let's hav
-000144e0: 6520 6120 6675 7274 6865 7220 6c6f 6f6b  e a further look
-000144f0: 2061 626f 7574 2069 7473 2070 6879 6c6f   about its phylo
-00014500: 6765 6e65 7469 6320 6c6f 6361 7469 6f6e  genetic location
-00014510: 2e20 5765 206b 6e6f 7720 7468 6174 2074  . We know that t
-00014520: 6865 7265 2061 7265 2075 6e63 6572 7461  here are uncerta
-00014530: 696e 7469 6573 2061 626f 7574 2077 6865  inties about whe
-00014540: 7468 6572 2074 6869 7320 7075 7461 7469  ther this putati
-00014550: 7665 2070 616c 656f 2d70 6f6c 7970 6c6f  ve paleo-polyplo
-00014560: 6964 697a 6174 696f 6e20 6576 656e 7420  idization event 
-00014570: 6973 2073 6861 7265 6420 7769 7468 2061  is shared with a
-00014580: 6c6c 2065 7564 6963 6f74 7320 6f72 206e  ll eudicots or n
-00014590: 6f74 2e20 5765 2063 616e 2063 686f 6f73  ot. We can choos
-000145a0: 6520 736f 6d65 206f 7468 6572 2065 7564  e some other eud
-000145b0: 6963 6f74 2067 656e 6f6d 6573 2074 6f20  icot genomes to 
-000145c0: 7365 6520 7468 6520 6f72 6465 7269 6e67  see the ordering
-000145d0: 206f 6620 7370 6563 6961 7469 6f6e 2061   of speciation a
-000145e0: 6e64 2070 6f6c 7970 6c6f 6964 697a 6174  nd polyploidizat
-000145f0: 696f 6e20 6576 656e 7473 2e20 4865 7265  ion events. Here
-00014600: 2077 6520 6368 6f6f 7365 202a 5669 7469   we choose *Viti
-00014610: 7320 7669 6e69 6665 7261 2a2c 202a 5072  s vinifera*, *Pr
-00014620: 6f74 6561 2063 796e 6172 6f69 6465 732a  otea cynaroides*
-00014630: 2061 6e64 202a 4163 6f72 7573 2061 6d65   and *Acorus ame
-00014640: 7269 6361 6e75 732a 2069 6e20 7468 6520  ricanus* in the 
-00014650: 666f 6c6c 6f77 696e 6720 2a4b 2a3c 7375  following *K*<su
-00014660: 623e 533c 2f73 7562 3e20 616e 616c 7973  b>S</sub> analys
-00014670: 6973 2e20 4669 7273 742c 2077 6520 6275  is. First, we bu
-00014680: 696c 7420 6120 676c 6f62 616c 204d 5242  ilt a global MRB
-00014690: 4820 6661 6d69 6c79 2075 7369 6e67 2074  H family using t
-000146a0: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-000146b0: 2e0a 0a60 6060 0a77 6764 2064 6d64 202d  ...```.wgd dmd -
-000146c0: 2d67 6c6f 6261 6c6d 7262 6820 4171 7569  -globalmrbh Aqui
-000146d0: 6c65 6769 615f 636f 6572 756c 6561 2050  legia_coerulea P
-000146e0: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
-000146f0: 2041 636f 7275 735f 616d 6572 6963 616e   Acorus_american
-00014700: 7573 2056 6974 6973 5f76 696e 6966 6572  us Vitis_vinifer
-00014710: 6120 2d6f 2077 6764 5f67 6c6f 6261 6c6d  a -o wgd_globalm
-00014720: 7262 680a 6060 600a 0a49 6e20 7468 6520  rbh.```..In the 
-00014730: 676c 6f62 616c 204d 5242 4820 6661 6d69  global MRBH fami
-00014740: 6c79 2c20 6576 6572 7920 7061 6972 206f  ly, every pair o
-00014750: 6620 6f72 7468 6f6c 6f67 6f75 7320 6765  f orthologous ge
-00014760: 6e65 7320 6973 2074 6865 2072 6563 6970  nes is the recip
-00014770: 726f 6361 6c20 6265 7374 2068 6974 2c20  rocal best hit, 
-00014780: 7375 6767 6573 7469 6e67 2074 7275 6520  suggesting true 
-00014790: 6f72 7468 6f6c 6f67 6f75 7320 7265 6c61  orthologous rela
-000147a0: 7469 6f6e 7368 6970 732e 2057 6520 776f  tionships. We wo
-000147b0: 756c 6420 7573 6520 7468 6520 2a4b 2a3c  uld use the *K*<
-000147c0: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
-000147d0: 6573 2061 7373 6f63 6961 7465 6420 7769  es associated wi
-000147e0: 7468 2074 6865 7365 206f 7274 686f 6c6f  th these ortholo
-000147f0: 676f 7573 2070 6169 7273 2074 6f20 6465  gous pairs to de
-00014800: 6c69 6d69 7420 7468 6520 6469 7665 7267  limit the diverg
-00014810: 656e 6365 202a 4b2a 3c73 7562 3e53 3c2f  ence *K*<sub>S</
-00014820: 7375 623e 2070 6561 6b2e 2054 6f67 6574  sub> peak. Toget
-00014830: 6865 7220 7769 7468 2074 6865 2077 686f  her with the who
-00014840: 6c65 2070 6172 616e 6f6d 6520 2a4b 2a3c  le paranome *K*<
-00014850: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00014860: 7269 6275 7469 6f6e 2c20 7765 2063 6f6e  ribution, we con
-00014870: 6475 6374 2074 6865 2072 6174 6520 636f  duct the rate co
-00014880: 7272 6563 7469 6f6e 2075 7369 6e67 2074  rrection using t
-00014890: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-000148a0: 2e0a 0a21 2153 696e 6365 2060 7767 6460  ...!!Since `wgd`
-000148b0: 2076 6572 7369 6f6e 2032 2e30 2e32 342c   version 2.0.24,
-000148c0: 2077 6520 7265 7772 6f74 6520 6120 636c   we rewrote a cl
-000148d0: 6561 6e65 7220 616e 6420 7175 6963 6b65  eaner and quicke
-000148e0: 7220 7761 7920 6f66 2064 6f69 6e67 2073  r way of doing s
-000148f0: 7562 7374 6974 7574 696f 6e20 7261 7465  ubstitution rate
-00014900: 2063 6f72 7265 6374 696f 6e2e 2049 7427   correction. It'
-00014910: 7320 6e6f 7420 7265 7175 6972 6564 2074  s not required t
-00014920: 6f20 7479 7065 2069 6e20 616e 7920 7370  o type in any sp
-00014930: 6569 6365 7320 7061 6972 2061 6e64 2061  eices pair and a
-00014940: 2073 6572 6965 7320 6f66 202a 4b2a 3c73   series of *K*<s
-00014950: 7562 3e53 3c2f 7375 623e 2070 6c6f 7473  ub>S</sub> plots
-00014960: 2077 696c 6c20 6265 2070 726f 6475 6365   will be produce
-00014970: 642e 2054 6865 2072 6571 7569 7265 6420  d. The required 
-00014980: 6669 6c65 7320 6172 6520 6f72 7468 6f6c  files are orthol
-00014990: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
-000149a0: 2f73 7562 3e20 6461 7461 6669 6c65 2c20  /sub> datafile, 
-000149b0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
-000149c0: 7562 3e53 3c2f 7375 623e 2064 6174 6166  ub>S</sub> dataf
-000149d0: 696c 652c 2061 2073 7065 6369 6573 2074  ile, a species t
-000149e0: 7265 6520 616e 6420 6120 666f 6361 6c20  ree and a focal 
-000149f0: 7370 6563 6965 7320 2874 6865 206f 6e65  species (the one
-00014a00: 2069 6e70 7574 7465 6420 7769 7468 2070   inputted with p
-00014a10: 6172 616c 6f67 6f75 7320 2a4b 2a3c 7375  aralogous *K*<su
-00014a20: 623e 533c 2f73 7562 3e20 6461 7461 292e  b>S</sub> data).
-00014a30: 2055 7365 7273 2063 616e 2063 686f 6f73   Users can choos
-00014a40: 6520 746f 2061 6464 206f 6e65 206d 6f72  e to add one mor
-00014a50: 6520 6c61 7965 7220 6f66 2045 4c4d 4d20  e layer of ELMM 
-00014a60: 616e 616c 7973 6973 206f 6e20 7061 7261  analysis on para
-00014a70: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
-00014a80: 3c2f 7375 623e 2076 616c 7565 7320 616e  </sub> values an
-00014a90: 642f 6f72 2047 4d4d 2061 6e61 6c79 7369  d/or GMM analysi
-00014aa0: 7320 6f6e 2061 6e63 686f 7220 2a4b 2a3c  s on anchor *K*<
-00014ab0: 7375 623e 533c 2f73 7562 3e20 6469 7374  sub>S</sub> dist
-00014ac0: 7269 6275 7469 6f6e 2e20 5468 6520 6f72  ribution. The or
-00014ad0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
-00014ae0: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
-00014af0: 6275 7469 6f6e 2063 616e 2062 6520 6361  bution can be ca
-00014b00: 6c63 756c 6174 6564 2075 7369 6e67 2074  lculated using t
-00014b10: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
-00014b20: 2e0a 0a60 6060 0a77 6764 206b 7364 2077  ...```.wgd ksd w
-00014b30: 6764 5f67 6c6f 6261 6c6d 7262 682f 676c  gd_globalmrbh/gl
-00014b40: 6f62 616c 5f4d 5242 482e 7473 7620 4171  obal_MRBH.tsv Aq
-00014b50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00014b60: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
-00014b70: 6573 2041 636f 7275 735f 616d 6572 6963  es Acorus_americ
-00014b80: 616e 7573 2056 6974 6973 5f76 696e 6966  anus Vitis_vinif
-00014b90: 6572 6120 2d6f 2077 6764 5f67 6c6f 6261  era -o wgd_globa
-00014ba0: 6c6d 7262 685f 6b73 0a60 6060 0a0a 5769  lmrbh_ks.```..Wi
-00014bb0: 7468 2074 6865 2063 616c 6375 6c61 7465  th the calculate
-00014bc0: 6420 6f72 7468 6f6c 6f67 6f75 7320 2a4b  d orthologous *K
-00014bd0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00014be0: 7374 7269 6275 7469 6f6e 2c20 7765 2063  stribution, we c
-00014bf0: 616e 2075 7365 2074 6865 2063 6f6d 6d61  an use the comma
-00014c00: 6e64 2062 656c 6f77 2074 6f20 636f 6e64  nd below to cond
-00014c10: 7563 7420 7468 6520 7261 7465 2063 6f72  uct the rate cor
-00014c20: 7265 6374 696f 6e20 616e 642f 6f72 206d  rection and/or m
-00014c30: 6978 7475 7265 206d 6f64 656c 696e 6720  ixture modeling 
-00014c40: 616e 616c 7973 6973 2e0a 0a60 6060 0a77  analysis...```.w
-00014c50: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
-00014c60: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
-00014c70: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
-00014c80: 7376 202d 6661 2041 7175 696c 6567 6961  sv -fa Aquilegia
-00014c90: 5f63 6f65 7275 6c65 6120 2d65 706b 2077  _coerulea -epk w
-00014ca0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00014cb0: 5f63 6f65 7275 6c65 612e 6b73 2e74 7376  _coerulea.ks.tsv
-00014cc0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
-00014cd0: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00014ce0: 6f69 6e74 732e 7478 7420 2d73 7020 7370  oints.txt -sp sp
-00014cf0: 6563 6965 7374 7265 652e 6e77 202d 6f20  eciestree.nw -o 
-00014d00: 7767 645f 7669 7a5f 6d69 7865 645f 4b73  wgd_viz_mixed_Ks
-00014d10: 202d 2d70 6c6f 7465 6c6d 6d20 2d2d 706c   --plotelmm --pl
-00014d20: 6f74 6170 676d 6d20 2d2d 7265 7765 6967  otapgmm --reweig
-00014d30: 6874 0a60 6060 0a0a 6f72 2075 7369 6e67  ht.```..or using
-00014d40: 2074 6865 2063 6f6d 6d61 6e64 2062 656c   the command bel
-00014d50: 6f77 2c20 7768 6963 6820 636f 6d62 696e  ow, which combin
-00014d60: 6573 2074 6865 2074 776f 2073 7465 7073  es the two steps
-00014d70: 2061 626f 7665 2069 6e20 6f6e 652e 204e   above in one. N
-00014d80: 6f74 6520 7468 6174 2077 6520 7375 6767  ote that we sugg
-00014d90: 6573 7420 6f66 2074 616b 696e 6720 7477  est of taking tw
-00014da0: 6f20 7365 7061 7261 7465 2073 7465 7073  o separate steps
-00014db0: 2069 6e20 7768 6963 6820 6077 6764 206b   in which `wgd k
-00014dc0: 7364 6020 756e 6465 7274 616b 6573 2074  sd` undertakes t
-00014dd0: 6865 2063 616c 6375 6c61 7469 6f6e 206f  he calculation o
-00014de0: 6620 6f72 7468 6f6c 6f67 6f75 7320 2a4b  f orthologous *K
-00014df0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00014e00: 7374 7269 6275 7469 6f6e 2077 6869 6c65  stribution while
-00014e10: 2060 7767 6420 7669 7a60 2063 6172 7269   `wgd viz` carri
-00014e20: 6573 206f 7574 2074 6865 2072 6174 6520  es out the rate 
-00014e30: 636f 7272 6563 7469 6f6e 2061 6e64 2047  correction and G
-00014e40: 4d4d 2061 6e61 6c79 7369 7320 7375 6368  MM analysis such
-00014e50: 2074 6861 7420 6974 2773 2065 6173 6965   that it's easie
-00014e60: 7220 746f 2064 6562 7567 2e0a 6060 600a  r to debug..```.
-00014e70: 7767 6420 6b73 6420 7767 645f 676c 6f62  wgd ksd wgd_glob
-00014e80: 616c 6d72 6268 2f67 6c6f 6261 6c5f 4d52  almrbh/global_MR
-00014e90: 4248 2e74 7376 2041 7175 696c 6567 6961  BH.tsv Aquilegia
-00014ea0: 5f63 6f65 7275 6c65 6120 5072 6f74 6561  _coerulea Protea
-00014eb0: 5f63 796e 6172 6f69 6465 7320 4163 6f72  _cynaroides Acor
-00014ec0: 7573 5f61 6d65 7269 6361 6e75 7320 5669  us_americanus Vi
-00014ed0: 7469 735f 7669 6e69 6665 7261 202d 2d65  tis_vinifera --e
-00014ee0: 7874 7261 7061 7261 6e6f 6d65 6b73 2077  xtraparanomeks w
-00014ef0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-00014f00: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
-00014f10: 2e74 7376 202d 7370 2073 7065 6369 6573  .tsv -sp species
-00014f20: 7472 6565 2e6e 7720 2d2d 7265 7765 6967  tree.nw --reweig
-00014f30: 6874 202d 6f20 7767 645f 676c 6f62 616c  ht -o wgd_global
-00014f40: 6d72 6268 5f6b 735f 7261 7465 5f63 6f72  mrbh_ks_rate_cor
-00014f50: 7265 6374 696f 6e20 2d66 6120 4171 7569  rection -fa Aqui
-00014f60: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
-00014f70: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
-00014f80: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
-00014f90: 6e74 732e 7478 7420 2d2d 706c 6f74 656c  nts.txt --plotel
-00014fa0: 6d6d 202d 2d70 6c6f 7461 7067 6d6d 0a60  mm --plotapgmm.`
-00014fb0: 6060 0a0a 5468 6520 6669 6c65 2060 7370  ``..The file `sp
-00014fc0: 6563 6965 7374 7265 652e 6e77 6020 6973  eciestree.nw` is
-00014fd0: 2074 6865 2074 6578 7420 6669 6c65 206f   the text file o
-00014fe0: 6620 7370 6563 6965 7320 7472 6565 2069  f species tree i
-00014ff0: 6e20 6e65 7769 636b 2074 6861 7420 7261  n newick that ra
-00015000: 7465 2063 6f72 7265 6374 696f 6e20 776f  te correction wo
-00015010: 756c 6420 6265 2063 6f6e 6475 6374 6564  uld be conducted
-00015020: 206f 6e2e 2049 7473 2063 6f6e 7465 6e74   on. Its content
-00015030: 2069 7320 6173 2062 656c 6f77 2e20 5573   is as below. Us
-00015040: 6572 7320 6361 6e20 6f70 7469 6f6e 616c  ers can optional
-00015050: 6c79 2070 726f 7669 6465 2074 6865 2073  ly provide the s
-00015060: 7065 6369 6573 2070 6169 7273 2074 6f20  pecies pairs to 
-00015070: 6265 2070 6c6f 7474 6564 2062 7574 2077  be plotted but w
-00015080: 6520 7375 6767 6573 7420 6f66 206a 7573  e suggest of jus
-00015090: 7420 7573 696e 6720 602d 6661 2041 7175  t using `-fa Aqu
-000150a0: 696c 6567 6961 5f63 6f65 7275 6c65 6160  ilegia_coerulea`
-000150b0: 2074 6f20 706c 6f74 2061 6c6c 2070 6f73   to plot all pos
-000150c0: 7369 626c 6520 666f 6361 6c2d 7369 7374  sible focal-sist
-000150d0: 6572 2073 7065 6369 6573 2070 6169 7273  er species pairs
-000150e0: 2e20 5765 2073 7567 6765 7374 2061 6464  . We suggest add
-000150f0: 696e 6720 7468 6520 6f70 7469 6f6e 2060  ing the option `
-00015100: 2d2d 7265 7765 6967 6874 6020 746f 2072  --reweight` to r
-00015110: 6563 616c 6375 6c61 7465 2074 6865 2077  ecalculate the w
-00015120: 6569 6768 7420 7065 7220 7370 6563 6965  eight per specie
-00015130: 7320 7061 6972 2073 7563 6820 7468 6174  s pair such that
-00015140: 2074 6865 2077 6569 6768 7420 6f66 206f   the weight of o
-00015150: 7274 686f 6c6f 676f 7573 2067 656e 6520  rthologous gene 
-00015160: 7061 6972 7320 7769 6c6c 2062 6563 6f6d  pairs will becom
-00015170: 6520 312e 2045 7874 7261 2063 6f6c 6c69  e 1. Extra colli
-00015180: 6e65 6172 2064 6174 6120 6361 6e20 6265  near data can be
-00015190: 2061 6464 6564 2062 7920 7468 6520 6f70   added by the op
-000151a0: 7469 6f6e 2060 2d61 7060 2061 6e64 2061  tion `-ap` and a
-000151b0: 6464 6974 696f 6e61 6c20 636c 7573 7465  dditional cluste
-000151c0: 7269 6e67 2061 6e61 6c79 7369 7320 6361  ring analysis ca
-000151d0: 6e20 6265 2069 6e69 7469 6174 6564 2062  n be initiated b
-000151e0: 7920 7365 7474 696e 6720 7468 6520 6f70  y setting the op
-000151f0: 7469 6f6e 2060 2d2d 706c 6f74 6170 676d  tion `--plotapgm
-00015200: 6d60 2061 6e64 2060 2d2d 706c 6f74 656c  m` and `--plotel
-00015210: 6d6d 602e 0a0a 6060 600a 2828 2856 6974  mm`...```.(((Vit
-00015220: 6973 5f76 696e 6966 6572 612c 5072 6f74  is_vinifera,Prot
-00015230: 6561 5f63 796e 6172 6f69 6465 7329 2c41  ea_cynaroides),A
-00015240: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
-00015250: 6129 2c41 636f 7275 735f 616d 6572 6963  a),Acorus_americ
-00015260: 616e 7573 293b 0a60 6060 0a0a 215b 5d28  anus);.```..![](
-00015270: 6461 7461 2f4d 6978 6564 2e6b 732e 4171  data/Mixed.ks.Aq
-00015280: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015290: 2e6e 6f64 652e 7765 6967 6874 6564 2e70  .node.weighted.p
-000152a0: 6e67 290a 0a54 6865 206d 6978 6564 202a  ng)..The mixed *
-000152b0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2064  K*<sub>S</sub> d
-000152c0: 6973 7472 6962 7574 696f 6e20 7368 6f77  istribution show
-000152d0: 6e20 6162 6f76 6520 6973 2061 2070 7562  n above is a pub
-000152e0: 6c69 6361 7469 6f6e 2d72 6561 6479 2066  lication-ready f
-000152f0: 6967 7572 6520 7468 6174 2061 7373 656d  igure that assem
-00015300: 626c 6573 2074 6865 2072 6573 756c 7473  bles the results
-00015310: 206f 6620 454c 4d4d 2c20 474d 4d20 616e   of ELMM, GMM an
-00015320: 6420 7261 7465 2063 6f72 7265 6374 696f  d rate correctio
-00015330: 6e2e 2054 6865 206f 6e65 2d76 732d 6f6e  n. The one-vs-on
-00015340: 6520 6f72 7468 6f6c 6f67 6f75 7320 2a4b  e orthologous *K
-00015350: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
-00015360: 7374 7269 6275 7469 6f6e 7320 6973 2061  stributions is a
-00015370: 6c73 6f20 6175 746f 6d61 7469 6361 6c6c  lso automaticall
-00015380: 7920 7072 6f64 7563 6564 2077 6974 6820  y produced with 
-00015390: 7261 7465 2063 6f72 7265 6374 696f 6e20  rate correction 
-000153a0: 7265 7375 6c74 7320 7375 7065 7269 6d70  results superimp
-000153b0: 6f73 6564 2077 6865 7265 2061 7661 696c  osed where avail
-000153c0: 6162 6c65 2061 7320 7368 6f77 6e20 6265  able as shown be
-000153d0: 6c6f 772e 0a0a 215b 5d28 6461 7461 2f46  low...![](data/F
-000153e0: 6f63 7573 5f73 6973 7465 725f 7061 6972  ocus_sister_pair
-000153f0: 732e 6b73 2e6e 6f64 652e 7765 6967 6874  s.ks.node.weight
-00015400: 6564 2e70 6e67 290a 215b 5d28 6461 7461  ed.png).![](data
-00015410: 2f41 6c6c 5f70 6169 7273 2e6b 732e 6e6f  /All_pairs.ks.no
-00015420: 6465 2e77 6569 6768 7465 642e 706e 6729  de.weighted.png)
-00015430: 0a0a 4265 7369 6465 7320 7468 6520 6162  ..Besides the ab
-00015440: 6f76 6520 6b73 7261 7465 732d 6c69 6b65  ove ksrates-like
-00015450: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-00015460: 2070 6c6f 742c 2061 206d 6f72 6520 636c   plot, a more cl
-00015470: 6173 7369 6320 2a4b 2a3c 7375 623e 533c  assic *K*<sub>S<
-00015480: 2f73 7562 3e20 706c 6f74 2063 616e 2062  /sub> plot can b
-00015490: 6520 6d61 6465 2062 7920 6164 6469 6e67  e made by adding
-000154a0: 2074 6865 206f 7074 696f 6e20 602d 2d63   the option `--c
-000154b0: 6c61 7373 6963 6020 746f 2074 6170 206d  lassic` to tap m
-000154c0: 6f72 6520 6465 7461 696c 6564 6c79 2069  ore detailedly i
-000154d0: 6e74 6f20 7468 6520 7661 7269 6174 696f  nto the variatio
-000154e0: 6e20 6f66 2073 796e 6f6e 796d 6f75 7320  n of synonymous 
-000154f0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
-00015500: 652e 0a0a 5573 696e 6720 7468 6520 636f  e...Using the co
-00015510: 6d6d 616e 6420 6265 6c6f 772c 2074 6865  mmand below, the
-00015520: 2064 6972 6563 7469 6f6e 206f 6620 7261   direction of ra
-00015530: 7465 2063 6f72 7265 6374 696f 6e20 616e  te correction an
-00015540: 6420 6465 6772 6565 206f 6620 7261 7465  d degree of rate
-00015550: 2076 6172 6961 7469 6f6e 2063 616e 2062   variation can b
-00015560: 6520 6f62 7365 7276 6564 206d 6f72 6520  e observed more 
-00015570: 6469 7265 6374 6c79 2e0a 0a60 6060 0a77  directly...```.w
-00015580: 6764 2076 697a 202d 6420 7767 645f 676c  gd viz -d wgd_gl
-00015590: 6f62 616c 6d72 6268 5f6b 732f 676c 6f62  obalmrbh_ks/glob
-000155a0: 616c 5f4d 5242 482e 7473 762e 6b73 2e74  al_MRBH.tsv.ks.t
-000155b0: 7376 202d 6661 2041 7175 696c 6567 6961  sv -fa Aquilegia
-000155c0: 5f63 6f65 7275 6c65 6120 2d65 706b 2077  _coerulea -epk w
-000155d0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
-000155e0: 5f63 6f65 7275 6c65 612e 6b73 2e74 7376  _coerulea.ks.tsv
-000155f0: 202d 6170 2077 6764 5f73 796e 2f69 6164   -ap wgd_syn/iad
-00015600: 686f 7265 2d6f 7574 2f61 6e63 686f 7270  hore-out/anchorp
-00015610: 6f69 6e74 732e 7478 7420 2d73 7020 7370  oints.txt -sp sp
-00015620: 6563 6965 7374 7265 652e 6e77 202d 6f20  eciestree.nw -o 
-00015630: 7767 645f 7669 7a5f 6d69 7865 645f 4b73  wgd_viz_mixed_Ks
-00015640: 202d 2d70 6c6f 7465 6c6d 6d20 2d2d 706c   --plotelmm --pl
-00015650: 6f74 6170 676d 6d20 2d2d 7265 7765 6967  otapgmm --reweig
-00015660: 6874 202d 2d70 6c6f 746b 6465 202d 2d63  ht --plotkde --c
-00015670: 6c61 7373 6963 0a60 6060 0a0a 215b 5d28  lassic.```..![](
-00015680: 6461 7461 2f41 7175 696c 6567 6961 5f63  data/Aquilegia_c
-00015690: 6f65 7275 6c65 615f 476c 6f62 616c 6d72  oerulea_Globalmr
-000156a0: 6268 4b73 5f45 6c6d 6d5f 4170 676d 6d5f  bhKs_Elmm_Apgmm_
-000156b0: 436f 7272 6563 7465 642e 6b73 642e 7376  Corrected.ksd.sv
-000156c0: 6729 0a0a 4173 2073 686f 776e 2061 626f  g)..As shown abo
-000156d0: 7665 2c20 6265 6361 7573 6520 6f66 2074  ve, because of t
-000156e0: 6865 2068 6967 6865 7220 7375 6273 7469  he higher substi
-000156f0: 7475 7469 6f6e 2072 6174 6520 6f66 202a  tution rate of *
-00015700: 4171 7569 6c65 6769 6120 636f 6572 756c  Aquilegia coerul
-00015710: 6561 2a2c 2074 6865 206f 7269 6769 6e61  ea*, the origina
-00015720: 6c20 6f72 7468 6f6c 6f67 6f75 7320 2a4b  l orthologous *K
-00015730: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
-00015740: 6c75 6573 2077 6572 6520 6163 7475 616c  lues were actual
-00015750: 6c79 2075 6e64 6572 6573 7469 6d61 7465  ly underestimate
-00015760: 6420 696e 2074 6865 2074 696d 652d 6672  d in the time-fr
-00015770: 616d 6520 6f66 202a 4171 7569 6c65 6769  ame of *Aquilegi
-00015780: 6120 636f 6572 756c 6561 2a2e 2057 6865  a coerulea*. Whe
-00015790: 6e20 7765 2072 6563 6f76 6572 6564 2074  n we recovered t
-000157a0: 6865 2064 6976 6572 6765 6e63 6520 7375  he divergence su
-000157b0: 6273 7469 7475 7469 6f6e 2064 6973 7461  bstitution dista
-000157c0: 6e63 6520 696e 2074 6572 6d73 206f 6620  nce in terms of 
-000157d0: 7477 6f20 7469 6d65 7320 6f66 2074 6865  two times of the
-000157e0: 2062 7261 6e63 682d 7370 6563 6966 6963   branch-specific
-000157f0: 2063 6f6e 7472 6962 7574 696f 6e20 6f66   contribution of
-00015800: 202a 412e 2063 6f65 7275 6c65 612a 2073   *A. coerulea* s
-00015810: 696e 6365 2069 7473 2064 6976 6572 6765  ince its diverge
-00015820: 6e63 6520 7769 7468 2074 6865 2073 6973  nce with the sis
-00015830: 7465 7220 7370 6563 6965 7320 706c 7573  ter species plus
-00015840: 2074 6865 2073 6861 7265 6420 7375 6273   the shared subs
-00015850: 7469 7475 7469 6f6e 2064 6973 7461 6e63  titution distanc
-00015860: 6520 6265 666f 7265 2064 6976 6572 6765  e before diverge
-00015870: 6e63 6520 2869 6e20 7265 6c61 7469 7665  nce (in relative
-00015880: 2074 6f20 7468 6520 6f75 7467 726f 7570   to the outgroup
-00015890: 292c 2074 6865 2063 6f72 7265 6374 6564  ), the corrected
-000158a0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
-000158b0: 206d 6f64 6520 6265 6361 6d65 206c 6172   mode became lar
-000158c0: 6765 722e 0a0a 4e6f 7465 2074 6861 7420  ger...Note that 
-000158d0: 7765 2063 616e 2065 6173 696c 7920 7368  we can easily sh
-000158e0: 6f77 2074 6861 7420 2a41 7175 696c 6567  ow that *Aquileg
-000158f0: 6961 2063 6f65 7275 6c65 612a 2068 6173  ia coerulea* has
-00015900: 2068 6967 6865 7220 7375 6273 7469 7475   higher substitu
-00015910: 7469 6f6e 2072 6174 6520 7468 616e 202a  tion rate than *
-00015920: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
-00015930: 732a 2061 6e64 202a 5669 7469 7320 7669  s* and *Vitis vi
-00015940: 6e69 6665 7261 2a20 6279 2063 6f6d 7061  nifera* by compa
-00015950: 7269 6e67 2074 6865 6972 2073 7562 7374  ring their subst
-00015960: 6974 7574 696f 6e20 6469 7374 616e 6365  itution distance
-00015970: 2069 6e20 7265 6761 7264 2074 6f20 7468   in regard to th
-00015980: 6520 7361 6d65 2064 6976 6572 6765 6e63  e same divergenc
-00015990: 6520 6576 656e 7420 7769 7468 206f 7574  e event with out
-000159a0: 6772 6f75 7020 7370 6563 6965 7320 2a41  group species *A
-000159b0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-000159c0: 2a2c 2075 7369 6e67 2063 6f6d 6d61 6e64  *, using command
-000159d0: 2062 656c 6f77 2e0a 0a60 6060 0a77 6764   below...```.wgd
-000159e0: 2076 697a 202d 6420 7767 645f 676c 6f62   viz -d wgd_glob
-000159f0: 616c 6d72 6268 5f6b 732f 676c 6f62 616c  almrbh_ks/global
-00015a00: 5f4d 5242 482e 7473 762e 6b73 2e74 7376  _MRBH.tsv.ks.tsv
-00015a10: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
-00015a20: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
-00015a30: 6f20 7767 645f 7669 7a5f 436f 6d70 6172  o wgd_viz_Compar
-00015a40: 655f 7261 7465 202d 2d73 7061 6972 2022  e_rate --spair "
-00015a50: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
-00015a60: 733b 5072 6f74 6561 5f63 796e 6172 6f69  s;Protea_cynaroi
-00015a70: 6465 7322 202d 2d73 7061 6972 2022 4171  des" --spair "Aq
-00015a80: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-00015a90: 3b41 636f 7275 735f 616d 6572 6963 616e  ;Acorus_american
-00015aa0: 7573 2220 2d2d 7370 6169 7220 2256 6974  us" --spair "Vit
-00015ab0: 6973 5f76 696e 6966 6572 613b 4163 6f72  is_vinifera;Acor
-00015ac0: 7573 5f61 6d65 7269 6361 6e75 7322 202d  us_americanus" -
-00015ad0: 2d70 6c6f 746b 6465 202d 2d63 6c61 7373  -plotkde --class
-00015ae0: 6963 0a60 6060 0a0a 215b 5d28 6461 7461  ic.```..![](data
-00015af0: 2f52 6177 5f4f 7274 686f 6c6f 6775 6573  /Raw_Orthologues
-00015b00: 5f43 6f6d 7061 7265 5f72 6174 652e 6b73  _Compare_rate.ks
-00015b10: 642e 7376 6729 0a0a 4173 2064 6973 706c  d.svg)..As displ
-00015b20: 6179 6564 2061 626f 7665 2c20 7468 6520  ayed above, the 
-00015b30: 6f72 7468 6f6c 6f67 6f75 7320 2a4b 2a3c  orthologous *K*<
-00015b40: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
-00015b50: 6573 2062 6577 7465 656e 202a 4171 7569  es bewteen *Aqui
-00015b60: 6c65 6769 6120 636f 6572 756c 6561 2a20  legia coerulea* 
-00015b70: 616e 6420 2a41 636f 7275 7320 616d 6572  and *Acorus amer
-00015b80: 6963 616e 7573 2a20 6861 7320 7468 6520  icanus* has the 
-00015b90: 6869 6768 6573 7420 6d6f 6465 2c20 696e  highest mode, in
-00015ba0: 6469 6361 7469 6e67 2074 6865 2066 6173  dicating the fas
-00015bb0: 7465 7220 7375 6273 7469 7475 7469 6f6e  ter substitution
-00015bc0: 2072 6174 6520 6f66 202a 412e 2063 6f65   rate of *A. coe
-00015bd0: 7275 6c65 612a 2063 6f6d 7061 7265 6420  rulea* compared 
-00015be0: 746f 202a 5072 6f74 6561 2063 796e 6172  to *Protea cynar
-00015bf0: 6f69 6465 732a 2061 6e64 202a 5669 7469  oides* and *Viti
-00015c00: 7320 7669 6e69 6665 7261 2a2e 0a0a 4265  s vinifera*...Be
-00015c10: 666f 7265 2076 322e 302e 3231 2c20 7468  fore v2.0.21, th
-00015c20: 6520 6765 6e65 2d73 7065 6369 6573 206d  e gene-species m
-00015c30: 6170 2066 696c 6520 6973 206e 6563 6365  ap file is necce
-00015c40: 7373 6172 696c 7920 6e65 6564 6564 2066  ssarily needed f
-00015c50: 6f72 2069 7473 2069 6d70 6c65 6d65 6e74  or its implement
-00015c60: 6174 696f 6e20 696e 2060 7767 6420 7669  ation in `wgd vi
-00015c70: 7a60 2c20 7768 6963 6820 7368 6f75 6c64  z`, which should
-00015c80: 2062 6520 6175 746f 6d61 7465 6c79 2070   be automately p
-00015c90: 726f 6475 6365 6420 6279 2074 6865 206c  roduced by the l
-00015ca0: 6173 7420 6077 6764 206b 7364 6020 7374  ast `wgd ksd` st
-00015cb0: 6570 2067 6976 656e 2074 6865 2060 7370  ep given the `sp
-00015cc0: 6169 7260 2061 6e64 2060 7370 6563 6965  air` and `specie
-00015cd0: 7374 7265 6560 2070 6172 616d 6574 6572  stree` parameter
-00015ce0: 732e 2054 6865 2060 6765 6e65 5f73 7065  s. The `gene_spe
-00015cf0: 6369 6573 2e6d 6170 6020 6861 7320 636f  cies.map` has co
-00015d00: 6e74 656e 7473 2061 7320 6265 6c6f 7720  ntents as below 
-00015d10: 696e 2077 6869 6368 2065 6163 6820 6c69  in which each li
-00015d20: 6e65 2069 7320 7468 6520 6a6f 696e 6564  ne is the joined
-00015d30: 2073 7472 696e 6720 6f66 2067 656e 6520   string of gene 
-00015d40: 6e61 6d65 2061 6e64 2073 7065 6369 6573  name and species
-00015d50: 206e 616d 6520 6279 2073 7061 6365 2e20   name by space. 
-00015d60: 4166 7465 7220 7632 2e30 2e32 3120 2869  After v2.0.21 (i
-00015d70: 6e63 6c75 6465 6429 2c20 7468 6520 6765  ncluded), the ge
-00015d80: 6e65 2d73 7065 6369 6573 206d 6170 2066  ne-species map f
-00015d90: 696c 6520 6973 206e 6f74 206e 6563 6365  ile is not necce
-00015da0: 7373 6172 696c 7920 6e65 6564 6564 2061  ssarily needed a
-00015db0: 6e79 6d6f 7265 2e0a 0a60 6060 0a41 7163  nymore...```.Aqc
-00015dc0: 6f65 3647 3035 3738 3030 2e31 2041 7175  oe6G057800.1 Aqu
-00015dd0: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
-00015de0: 5676 695f 5649 545f 3230 3173 3030 3131  Vvi_VIT_201s0011
-00015df0: 6730 3135 3330 2e31 2056 6974 6973 5f76  g01530.1 Vitis_v
-00015e00: 696e 6966 6572 610a 5063 795f 5072 6f63  inifera.Pcy_Proc
-00015e10: 7930 3167 3038 3531 3020 5072 6f74 6561  y01g08510 Protea
-00015e20: 5f63 796e 6172 6f69 6465 730a 4161 6d5f  _cynaroides.Aam_
-00015e30: 4163 6f72 612e 3034 4731 3432 3930 302e  Acora.04G142900.
-00015e40: 3120 4163 6f72 7573 5f61 6d65 7269 6361  1 Acorus_america
-00015e50: 6e75 730a 6060 600a 0a41 6e20 616c 7465  nus.```..An alte
-00015e60: 726e 6174 6976 6520 7761 7920 746f 2063  rnative way to c
-00015e70: 616c 6375 6c61 7465 2074 6865 206f 7274  alculate the ort
-00015e80: 686f 6c6f 676f 7573 202a 4b2a 3c73 7562  hologous *K*<sub
-00015e90: 3e53 3c2f 7375 623e 2069 7320 746f 2064  >S</sub> is to d
-00015ea0: 6972 6563 746c 7920 7573 6520 7468 6520  irectly use the 
-00015eb0: 6f72 7468 6f67 726f 7570 7320 696e 7374  orthogroups inst
-00015ec0: 6561 6420 6f66 2067 6c6f 6261 6c20 4d52  ead of global MR
-00015ed0: 4248 2066 616d 696c 792e 2054 6861 7420  BH family. That 
-00015ee0: 7761 7920 7765 2064 6f6e 2774 2075 7365  way we don't use
-00015ef0: 2074 6865 2070 7265 2d69 6e66 6572 7265   the pre-inferre
-00015f00: 6420 7061 7261 6e6f 6d65 202a 4b2a 3c73  d paranome *K*<s
-00015f10: 7562 3e53 3c2f 7375 623e 2062 7574 2074  ub>S</sub> but t
-00015f20: 6865 2070 6172 616c 6f67 6f75 7320 6765  he paralogous ge
-00015f30: 6e65 2070 6169 7273 2069 6e73 6964 6520  ne pairs inside 
-00015f40: 6561 6368 206f 7274 686f 6772 6f75 7020  each orthogroup 
-00015f50: 696e 7374 6561 642e 2054 6f20 6163 6869  instead. To achi
-00015f60: 6576 6520 7468 6174 2c20 7765 2066 6972  eve that, we fir
-00015f70: 7374 206e 6565 6420 746f 2069 6e66 6572  st need to infer
-00015f80: 206f 7274 686f 6772 6f75 7073 2075 7369   orthogroups usi
-00015f90: 6e67 2074 6865 2063 6f6d 6d61 6e64 2062  ng the command b
-00015fa0: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
-00015fb0: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
-00015fc0: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
-00015fd0: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
-00015fe0: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
-00015ff0: 7669 6e69 6665 7261 202d 2d6f 7274 686f  vinifera --ortho
-00016000: 696e 6665 7220 2d6f 2077 6764 5f6f 7274  infer -o wgd_ort
-00016010: 686f 2028 2d2d 6f6e 6c79 6f72 7468 6f29  ho (--onlyortho)
-00016020: 200a 6060 600a 0a55 7365 7273 2063 616e   .```..Users can
-00016030: 2064 6563 6964 6520 746f 206f 6e6c 7920   decide to only 
-00016040: 636f 6e64 7563 7420 7468 6520 6f72 7468  conduct the orth
-00016050: 6f67 726f 7570 2061 6e61 6c79 7369 7320  ogroup analysis 
-00016060: 7768 696c 6520 736b 6970 7069 6e67 206f  while skipping o
-00016070: 7468 6572 2061 6e61 6c79 7369 7320 6279  ther analysis by
-00016080: 2061 6464 696e 6720 7468 6520 666c 6167   adding the flag
-00016090: 2060 2d2d 6f6e 6c79 6f72 7468 6f60 2e20   `--onlyortho`. 
-000160a0: 4e65 7874 2073 7465 7020 6973 2074 6865  Next step is the
-000160b0: 2073 616d 6520 7769 7468 2067 6c6f 6261   same with globa
-000160c0: 6c20 4d52 4248 2066 616d 696c 792e 0a0a  l MRBH family...
-000160d0: 6060 600a 7767 6420 6b73 6420 7767 645f  ```.wgd ksd wgd_
-000160e0: 6f72 7468 6f2f 4f72 7468 6f67 726f 7570  ortho/Orthogroup
-000160f0: 732e 7370 2e74 7376 2041 7175 696c 6567  s.sp.tsv Aquileg
-00016100: 6961 5f63 6f65 7275 6c65 6120 5072 6f74  ia_coerulea Prot
-00016110: 6561 5f63 796e 6172 6f69 6465 7320 4163  ea_cynaroides Ac
-00016120: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
-00016130: 5669 7469 735f 7669 6e69 6665 7261 202d  Vitis_vinifera -
-00016140: 6f20 7767 645f 6f72 7468 6f5f 6b73 0a77  o wgd_ortho_ks.w
-00016150: 6764 2076 697a 202d 6420 7767 645f 6f72  gd viz -d wgd_or
-00016160: 7468 6f5f 6b73 2f4f 7274 686f 6772 6f75  tho_ks/Orthogrou
-00016170: 7073 2e73 702e 7473 762e 6b73 2e74 7376  ps.sp.tsv.ks.tsv
-00016180: 202d 6570 6b20 7767 645f 6b73 642f 4171   -epk wgd_ksd/Aq
-00016190: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
-000161a0: 2e6b 732e 7473 7620 2d73 7020 7370 6563  .ks.tsv -sp spec
-000161b0: 6965 7374 7265 652e 6e77 202d 2d72 6577  iestree.nw --rew
-000161c0: 6569 6768 7420 2d61 7020 7767 645f 7379  eight -ap wgd_sy
-000161d0: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
-000161e0: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
-000161f0: 706c 6f74 656c 6d6d 202d 2d70 6c6f 7461  plotelmm --plota
-00016200: 7067 6d6d 202d 6f20 7767 645f 6f72 7468  pgmm -o wgd_orth
-00016210: 6f5f 6b73 5f72 6174 655f 636f 7272 6563  o_ks_rate_correc
-00016220: 7469 6f6e 0a60 6060 0a0a 215b 5d28 6461  tion.```..![](da
-00016230: 7461 2f4d 6978 6564 2e6b 732e 4171 7569  ta/Mixed.ks.Aqui
-00016240: 6c65 6769 615f 636f 6572 756c 6561 2e6e  legia_coerulea.n
-00016250: 6f64 652e 7765 6967 6874 6564 5f4f 7274  ode.weighted_Ort
-00016260: 686f 5f4b 732e 706e 6729 0a21 5b5d 2864  ho_Ks.png).![](d
-00016270: 6174 612f 466f 6375 735f 7369 7374 6572  ata/Focus_sister
-00016280: 5f70 6169 7273 2e6b 732e 6e6f 6465 2e77  _pairs.ks.node.w
-00016290: 6569 6768 7465 645f 4f72 7468 6f5f 4b73  eighted_Ortho_Ks
-000162a0: 2e70 6e67 290a 215b 5d28 6461 7461 2f41  .png).![](data/A
-000162b0: 6c6c 5f70 6169 7273 2e6b 732e 6e6f 6465  ll_pairs.ks.node
-000162c0: 2e77 6569 6768 7465 645f 4f72 7468 6f5f  .weighted_Ortho_
-000162d0: 4b73 2e70 6e67 290a 0a41 7320 7368 6f77  Ks.png)..As show
-000162e0: 6e20 6162 6f76 652c 2074 6865 206e 756d  n above, the num
-000162f0: 6265 7220 6f66 206f 7274 686f 6c6f 676f  ber of orthologo
-00016300: 7573 2067 656e 6520 7061 6972 7320 6973  us gene pairs is
-00016310: 2064 6966 6665 7265 6e74 2074 6861 6e20   different than 
-00016320: 7468 6520 6f6e 6520 6672 6f6d 2067 6c6f  the one from glo
-00016330: 6261 6c20 4d52 4248 2066 616d 696c 6965  bal MRBH familie
-00016340: 7320 696e 2074 6861 7420 6865 7265 2077  s in that here w
-00016350: 6520 706c 6f74 7465 6420 616c 6c20 6f72  e plotted all or
-00016360: 7468 6f6c 6f67 6f75 7320 6765 6e65 2070  thologous gene p
-00016370: 6169 7273 2069 6e73 7465 6164 206f 6620  airs instead of 
-00016380: 6f6e 6c79 2067 6c6f 6261 6c20 4d52 4248  only global MRBH
-00016390: 2066 616d 696c 6965 732c 2074 6f67 6574   families, toget
-000163a0: 6865 7220 7769 7468 2064 6966 6665 7265  her with differe
-000163b0: 6e74 2072 6563 616c 6375 6c61 7465 6420  nt recalculated 
-000163c0: 7765 6967 6874 732e 0a0a 4166 7465 7220  weights...After 
-000163d0: 7468 6520 7068 796c 6f67 656e 6574 6963  the phylogenetic
-000163e0: 2074 696d 696e 6720 6f66 2074 6865 2052   timing of the R
-000163f0: 616e 756e 6375 6c61 6c65 7320 5747 442c  anunculales WGD,
-00016400: 2077 6520 6361 6e20 6675 7274 6865 7220   we can further 
-00016410: 696e 6665 7220 6974 7320 6162 736f 6c75  infer its absolu
-00016420: 7465 2061 6765 2e20 4669 7273 7420 7765  te age. First we
-00016430: 2069 6e66 6572 2074 6865 2063 7265 6469   infer the credi
-00016440: 626c 6520 7261 6e67 6520 6f66 2061 6e63  ble range of anc
-00016450: 686f 7220 7061 6972 7320 6279 202a 4b2a  hor pairs by *K*
-00016460: 3c73 7562 3e53 3c2f 7375 623e 2068 6575  <sub>S</sub> heu
-00016470: 7269 7374 6963 616c 6c79 2075 7369 6e67  ristically using
-00016480: 2074 6865 2070 726f 6772 616d 2060 7767   the program `wg
-00016490: 6420 7065 616b 602e 0a0a 6060 600a 7767  d peak`...```.wg
-000164a0: 6420 7065 616b 202d 2d68 6575 7269 7374  d peak --heurist
-000164b0: 6963 2077 6764 5f6b 7364 2f41 7175 696c  ic wgd_ksd/Aquil
-000164c0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
-000164d0: 762e 6b73 2e74 7376 202d 6170 2077 6764  v.ks.tsv -ap wgd
-000164e0: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
-000164f0: 2f61 6e63 686f 7270 6f69 6e74 732e 7478  /anchorpoints.tx
-00016500: 7420 2d73 6d20 7767 645f 7379 6e2f 6961  t -sm wgd_syn/ia
-00016510: 6468 6f72 652d 6f75 742f 7365 676d 656e  dhore-out/segmen
-00016520: 7473 2e74 7874 202d 6c65 2077 6764 5f73  ts.txt -le wgd_s
-00016530: 796e 2f69 6164 686f 7265 2d6f 7574 2f6c  yn/iadhore-out/l
-00016540: 6973 745f 656c 656d 656e 7473 2e74 7874  ist_elements.txt
-00016550: 202d 6d70 2077 6764 5f73 796e 2f69 6164   -mp wgd_syn/iad
-00016560: 686f 7265 2d6f 7574 2f6d 756c 7469 706c  hore-out/multipl
-00016570: 6963 6f6e 5f70 6169 7273 2e74 7874 202d  icon_pairs.txt -
-00016580: 6f20 7767 645f 7065 616b 0a60 6060 0a0a  o wgd_peak.```..
-00016590: 215b 5d28 6461 7461 2f41 6e63 686f 724b  ![](data/AnchorK
-000165a0: 735f 5065 616b 4349 5f41 7175 696c 6567  s_PeakCI_Aquileg
-000165b0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
-000165c0: 6b73 2e74 7376 5f6e 6f64 655f 7765 6967  ks.tsv_node_weig
-000165d0: 6874 6564 2e73 7667 290a 0a41 7320 7368  hted.svg)..As sh
-000165e0: 6f77 6e20 6162 6f76 652c 2077 6520 6173  own above, we as
-000165f0: 7375 6d65 6420 6120 6c6f 676e 6f72 6d61  sumed a lognorma
-00016600: 6c20 6469 7374 7269 6275 7469 6f6e 2061  l distribution a
-00016610: 7420 7468 6520 7065 616b 206c 6f63 6174  t the peak locat
-00016620: 696f 6e20 6465 7465 6374 6564 2062 7920  ion detected by 
-00016630: 7468 6520 6073 6967 6e61 6c60 206d 6f64  the `signal` mod
-00016640: 756c 6520 6f66 2060 7363 6970 7960 206c  ule of `scipy` l
-00016650: 6962 7261 7279 2e20 5468 6520 3935 2520  ibrary. The 95% 
-00016660: 636f 6e66 6964 656e 6365 206c 6576 656c  confidence level
-00016670: 206f 6620 7468 6520 6c6f 676e 6f72 6d61   of the lognorma
-00016680: 6c20 6469 7374 7269 6275 7469 6f6e 2077  l distribution w
-00016690: 6173 2061 7070 6c69 6564 2c20 692e 652e  as applied, i.e.
-000166a0: 2c20 302e 3638 2d32 2e37 342c 2069 6e20  , 0.68-2.74, in 
-000166b0: 6675 7274 6865 7220 6d6f 6c65 6375 6c61  further molecula
-000166c0: 7220 6461 7469 6e67 2e20 5468 6520 6669  r dating. The fi
-000166d0: 6c65 2060 4171 7569 6c65 6769 615f 636f  le `Aquilegia_co
-000166e0: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-000166f0: 765f 3935 2543 495f 4150 5f66 6f72 5f64  v_95%CI_AP_for_d
-00016700: 6174 696e 675f 7765 6967 6874 6564 5f66  ating_weighted_f
-00016710: 6f72 6d61 742e 7473 7660 2069 7320 7768  ormat.tsv` is wh
-00016720: 6174 2077 6520 6e65 6564 2066 6f72 206e  at we need for n
-00016730: 6578 7420 7374 6570 2e20 546f 2062 7569  ext step. To bui
-00016740: 6c64 2074 6865 206f 7274 686f 6772 6f75  ld the orthogrou
-00016750: 7073 2075 7365 6420 696e 2070 6879 6c6f  ps used in phylo
-00016760: 6765 6e65 7469 6320 6461 7469 6e67 2c20  genetic dating, 
-00016770: 7765 206e 6565 6420 746f 2073 656c 6563  we need to selec
-00016780: 7420 736f 6d65 2073 7065 6369 6573 2061  t some species a
-00016790: 6e64 2066 6f72 6d20 6120 7374 6172 7469  nd form a starti
-000167a0: 6e67 2074 7265 6520 7769 7468 2070 726f  ng tree with pro
-000167b0: 7065 7220 666f 7373 696c 2063 616c 6962  per fossil calib
-000167c0: 7261 7469 6f6e 732e 2057 6520 7072 6f76  rations. We prov
-000167d0: 6964 6520 6f6e 6520 696e 206d 636d 6374  ide one in mcmct
-000167e0: 7265 6520 666f 726d 6174 2061 7320 6265  ree format as be
-000167f0: 6c6f 772e 0a0a 6060 600a 3137 2031 0a28  low...```.17 1.(
-00016800: 2828 2850 6f74 616d 6f67 6574 6f6e 5f61  (((Potamogeton_a
-00016810: 6375 7469 666f 6c69 7573 2c28 5370 6972  cutifolius,(Spir
-00016820: 6f64 656c 615f 696e 7465 726d 6564 6961  odela_intermedia
-00016830: 2c41 6d6f 7270 686f 7068 616c 6c75 735f  ,Amorphophallus_
-00016840: 6b6f 6e6a 6163 2929 2c28 4163 616e 7468  konjac)),(Acanth
-00016850: 6f63 686c 616d 7973 5f62 7261 6374 6561  ochlamys_bractea
-00016860: 7461 2c28 4469 6f73 636f 7265 615f 616c  ta,(Dioscorea_al
-00016870: 6174 612c 4469 6f73 636f 7265 615f 726f  ata,Dioscorea_ro
-00016880: 7475 6e64 6174 6129 2927 3e30 2e35 3630  tundata))'>0.560
-00016890: 303c 312e 3238 3633 2729 273e 302e 3833  0<1.2863')'>0.83
-000168a0: 3630 3c31 2e32 3836 3327 2c28 4163 6f72  60<1.2863',(Acor
-000168b0: 7573 5f61 6d65 7269 6361 6e75 732c 4163  us_americanus,Ac
-000168c0: 6f72 7573 5f74 6174 6172 696e 6f77 6969  orus_tatarinowii
-000168d0: 2929 273e 302e 3833 3630 3c31 2e32 3836  ))'>0.8360<1.286
-000168e0: 3327 2c28 2828 2854 6574 7261 6365 6e74  3',((((Tetracent
-000168f0: 726f 6e5f 7369 6e65 6e73 652c 5472 6f63  ron_sinense,Troc
-00016900: 686f 6465 6e64 726f 6e5f 6172 616c 696f  hodendron_aralio
-00016910: 6964 6573 292c 2842 7578 7573 5f61 7573  ides),(Buxus_aus
-00016920: 7472 6f79 756e 6e61 6e65 6e73 6973 2c42  troyunnanensis,B
-00016930: 7578 7573 5f73 696e 6963 6129 2927 3e31  uxus_sinica))'>1
-00016940: 2e31 3038 303c 312e 3238 3633 272c 284e  .1080<1.2863',(N
-00016950: 656c 756d 626f 5f6e 7563 6966 6572 612c  elumbo_nucifera,
-00016960: 2854 656c 6f70 6561 5f73 7065 6369 6f73  (Telopea_specios
-00016970: 6973 7369 6d61 2c50 726f 7465 615f 6379  issima,Protea_cy
-00016980: 6e61 726f 6964 6573 2929 2927 3e31 2e31  naroides)))'>1.1
-00016990: 3038 303c 312e 3238 3633 272c 2841 7175  080<1.2863',(Aqu
-000169a0: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
-000169b0: 6170 312c 4171 7569 6c65 6769 615f 636f  ap1,Aquilegia_co
-000169c0: 6572 756c 6561 5f61 7032 2929 273e 312e  erulea_ap2))'>1.
-000169d0: 3130 3830 3c31 2e32 3836 3327 2927 3e31  1080<1.2863')'>1
-000169e0: 2e32 3732 303c 322e 3437 3230 273b 0a60  .2720<2.4720';.`
-000169f0: 6060 0a0a 4173 2070 7265 7365 6e74 6564  ``..As presented
-00016a00: 2061 626f 7665 2c20 7468 6520 666f 6375   above, the focu
-00016a10: 7320 7370 6563 6965 7320 7468 6174 2069  s species that i
-00016a20: 7320 6162 6f75 7420 746f 2062 6520 6461  s about to be da
-00016a30: 7465 6420 6e65 6564 7320 746f 2062 6520  ted needs to be 
-00016a40: 7265 706c 6163 6564 2077 6974 6820 6028  replaced with `(
-00016a50: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016a60: 6561 5f61 7031 2c41 7175 696c 6567 6961  ea_ap1,Aquilegia
-00016a70: 5f63 6f65 7275 6c65 615f 6170 3229 602e  _coerulea_ap2)`.
-00016a80: 2057 6974 6820 7468 6973 2073 7461 7274   With this start
-00016a90: 696e 6720 7472 6565 2061 6e64 2070 7265  ing tree and pre
-00016aa0: 646f 776e 6c6f 6164 6564 2063 6473 2066  downloaded cds f
-00016ab0: 696c 6573 206f 6620 616c 6c20 7468 6520  iles of all the 
-00016ac0: 7370 6563 6965 732c 2077 6520 6361 6e20  species, we can 
-00016ad0: 6275 696c 6420 7468 6520 6f72 7468 6f67  build the orthog
-00016ae0: 726f 7570 2075 7365 6420 696e 2074 6865  roup used in the
-00016af0: 2066 696e 616c 206d 6f6c 6563 756c 6172   final molecular
-00016b00: 2064 6174 696e 6720 7573 696e 6720 7468   dating using th
-00016b10: 6520 636f 6d6d 616e 6420 6173 2062 656c  e command as bel
-00016b20: 6f77 2e0a 0a60 6060 0a77 6764 2064 6d64  ow...```.wgd dmd
-00016b30: 202d 6620 4171 7569 6c65 6769 615f 636f   -f Aquilegia_co
-00016b40: 6572 756c 6561 202d 6170 2077 6764 5f70  erulea -ap wgd_p
-00016b50: 6561 6b2f 4171 7569 6c65 6769 615f 636f  eak/Aquilegia_co
-00016b60: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
-00016b70: 765f 3935 2543 495f 4150 5f66 6f72 5f64  v_95%CI_AP_for_d
-00016b80: 6174 696e 675f 7765 6967 6874 6564 5f66  ating_weighted_f
-00016b90: 6f72 6d61 742e 7473 7620 2d6f 2077 6764  ormat.tsv -o wgd
-00016ba0: 5f64 6d64 5f6f 7274 686f 2050 6f74 616d  _dmd_ortho Potam
-00016bb0: 6f67 6574 6f6e 5f61 6375 7469 666f 6c69  ogeton_acutifoli
-00016bc0: 7573 2053 7069 726f 6465 6c61 5f69 6e74  us Spirodela_int
-00016bd0: 6572 6d65 6469 6120 416d 6f72 7068 6f70  ermedia Amorphop
-00016be0: 6861 6c6c 7573 5f6b 6f6e 6a61 6320 4163  hallus_konjac Ac
-00016bf0: 616e 7468 6f63 686c 616d 7973 5f62 7261  anthochlamys_bra
-00016c00: 6374 6561 7461 2044 696f 7363 6f72 6561  cteata Dioscorea
-00016c10: 5f61 6c61 7461 2044 696f 7363 6f72 6561  _alata Dioscorea
-00016c20: 5f72 6f74 756e 6461 7461 2041 636f 7275  _rotundata Acoru
-00016c30: 735f 616d 6572 6963 616e 7573 2041 636f  s_americanus Aco
-00016c40: 7275 735f 7461 7461 7269 6e6f 7769 6920  rus_tatarinowii 
-00016c50: 5465 7472 6163 656e 7472 6f6e 5f73 696e  Tetracentron_sin
-00016c60: 656e 7365 2054 726f 6368 6f64 656e 6472  ense Trochodendr
-00016c70: 6f6e 5f61 7261 6c69 6f69 6465 7320 4275  on_aralioides Bu
-00016c80: 7875 735f 6175 7374 726f 7975 6e6e 616e  xus_austroyunnan
-00016c90: 656e 7369 7320 4275 7875 735f 7369 6e69  ensis Buxus_sini
-00016ca0: 6361 204e 656c 756d 626f 5f6e 7563 6966  ca Nelumbo_nucif
-00016cb0: 6572 6120 5465 6c6f 7065 615f 7370 6563  era Telopea_spec
-00016cc0: 696f 7369 7373 696d 6120 5072 6f74 6561  iosissima Protea
-00016cd0: 5f63 796e 6172 6f69 6465 7320 4171 7569  _cynaroides Aqui
-00016ce0: 6c65 6769 615f 636f 6572 756c 6561 0a60  legia_coerulea.`
-00016cf0: 6060 0a0a 5468 6520 7265 7375 6c74 2066  ``..The result f
-00016d00: 696c 6520 606d 6572 6765 5f66 6f63 7573  ile `merge_focus
-00016d10: 5f61 702e 7473 7660 2069 7320 7768 6174  _ap.tsv` is what
-00016d20: 2077 6520 6e65 6564 2066 6f72 2074 6865   we need for the
-00016d30: 2066 696e 616c 2073 7465 7020 6f66 206d   final step of m
-00016d40: 6f6c 6563 756c 6172 2064 6174 696e 6720  olecular dating 
-00016d50: 696e 2070 726f 6772 616d 2060 7767 6420  in program `wgd 
-00016d60: 666f 6375 7360 2e0a 0a60 6060 0a77 6764  focus`...```.wgd
-00016d70: 2066 6f63 7573 202d 2d70 726f 7464 6174   focus --protdat
-00016d80: 696e 6720 2d2d 6161 6d6f 6465 6c20 6c67  ing --aamodel lg
-00016d90: 2077 6764 5f64 6d64 5f6f 7274 686f 2f6d   wgd_dmd_ortho/m
-00016da0: 6572 6765 5f66 6f63 7573 5f61 702e 7473  erge_focus_ap.ts
-00016db0: 7620 2d73 7020 6461 7469 6e67 5f74 7265  v -sp dating_tre
-00016dc0: 652e 6e77 202d 6f20 7767 645f 6461 7469  e.nw -o wgd_dati
-00016dd0: 6e67 202d 6420 6d63 6d63 7472 6565 202d  ng -d mcmctree -
-00016de0: 6473 2027 6275 726e 696e 203d 2032 3030  ds 'burnin = 200
-00016df0: 3027 202d 6473 2027 7361 6d70 6672 6571  0' -ds 'sampfreq
-00016e00: 203d 2031 3030 3027 202d 6473 2027 6e73   = 1000' -ds 'ns
-00016e10: 616d 706c 6520 3d20 3230 3030 3027 2050  ample = 20000' P
-00016e20: 6f74 616d 6f67 6574 6f6e 5f61 6375 7469  otamogeton_acuti
-00016e30: 666f 6c69 7573 2053 7069 726f 6465 6c61  folius Spirodela
-00016e40: 5f69 6e74 6572 6d65 6469 6120 416d 6f72  _intermedia Amor
-00016e50: 7068 6f70 6861 6c6c 7573 5f6b 6f6e 6a61  phophallus_konja
-00016e60: 6320 4163 616e 7468 6f63 686c 616d 7973  c Acanthochlamys
-00016e70: 5f62 7261 6374 6561 7461 2044 696f 7363  _bracteata Diosc
-00016e80: 6f72 6561 5f61 6c61 7461 2044 696f 7363  orea_alata Diosc
-00016e90: 6f72 6561 5f72 6f74 756e 6461 7461 2041  orea_rotundata A
-00016ea0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
-00016eb0: 2041 636f 7275 735f 7461 7461 7269 6e6f   Acorus_tatarino
-00016ec0: 7769 6920 5465 7472 6163 656e 7472 6f6e  wii Tetracentron
-00016ed0: 5f73 696e 656e 7365 2054 726f 6368 6f64  _sinense Trochod
-00016ee0: 656e 6472 6f6e 5f61 7261 6c69 6f69 6465  endron_aralioide
-00016ef0: 7320 4275 7875 735f 6175 7374 726f 7975  s Buxus_austroyu
-00016f00: 6e6e 616e 656e 7369 7320 4275 7875 735f  nnanensis Buxus_
-00016f10: 7369 6e69 6361 204e 656c 756d 626f 5f6e  sinica Nelumbo_n
-00016f20: 7563 6966 6572 6120 5465 6c6f 7065 615f  ucifera Telopea_
-00016f30: 7370 6563 696f 7369 7373 696d 6120 5072  speciosissima Pr
-00016f40: 6f74 6561 5f63 796e 6172 6f69 6465 7320  otea_cynaroides 
-00016f50: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
-00016f60: 6561 0a60 6060 0a0a 4865 7265 2077 6520  ea.```..Here we 
-00016f70: 6f6e 6c79 2069 6d70 6c65 6d65 6e74 6564  only implemented
-00016f80: 2074 6865 2063 6f6e 6361 7465 6e61 7469   the concatenati
-00016f90: 6f6e 2061 6e61 6c79 7369 7320 7573 696e  on analysis usin
-00016fa0: 6720 7072 6f74 6569 6e20 7365 7175 656e  g protein sequen
-00016fb0: 6365 2062 7920 6164 6469 6e67 2074 6865  ce by adding the
-00016fc0: 2066 6c61 6720 602d 2d70 726f 7464 6174   flag `--protdat
-00016fd0: 696e 6760 2061 6e64 2077 6520 7365 7420  ing` and we set 
-00016fe0: 7468 6520 7061 7261 6d65 7465 7220 666f  the parameter fo
-00016ff0: 7220 606d 636d 6374 7265 6560 2076 6961  r `mcmctree` via
-00017000: 2074 6865 206f 7074 696f 6e20 602d 6473   the option `-ds
-00017010: 602e 204e 6f74 6520 7468 6174 206f 7468  `. Note that oth
-00017020: 6572 2064 6174 696e 6720 7072 6f67 7261  er dating progra
-00017030: 6d20 7375 6368 2061 7320 6072 3873 6020  m such as `r8s` 
-00017040: 616e 6420 6062 6561 7374 6020 6172 6520  and `beast` are 
-00017050: 616c 736f 2061 7661 696c 6162 6c65 2067  also available g
-00017060: 6976 656e 2073 6f6d 6520 6d61 6e64 6174  iven some mandat
-00017070: 6f72 7920 7061 7261 6d65 7465 7273 2e20  ory parameters. 
-00017080: 5468 6520 6669 6e61 6c20 6c6f 6720 6f66  The final log of
-00017090: 2074 6865 2073 7563 6365 7373 6675 6c20   the successful 
-000170a0: 7275 6e20 6973 2061 7320 6265 6c6f 772e  run is as below.
-000170b0: 0a0a 6060 600a 3136 3a30 343a 3235 2049  ..```.16:04:25 I
-000170c0: 4e46 4f20 2020 2020 5275 6e6e 696e 6720  NFO     Running 
-000170d0: 6d63 6d63 7472 6565 2075 7369 6e67 2048  mcmctree using H
-000170e0: 6573 7369 616e 206d 6174 7269 7820 6f66  essian matrix of
-000170f0: 204c 472b 4761 6d6d 6120 2063 6f72 652e   LG+Gamma  core.
-00017100: 7079 3a39 3637 0a20 2020 2020 2020 2020  py:967.         
-00017110: 2020 2020 2020 2020 2066 6f72 2070 726f           for pro
-00017120: 7465 696e 206d 6f64 656c 0a32 333a 3439  tein model.23:49
-00017130: 3a33 3720 494e 464f 2020 2020 2050 6f73  :37 INFO     Pos
-00017140: 7465 7269 6f72 206d 6561 6e20 666f 7220  terior mean for 
-00017150: 7468 6520 6167 6573 206f 6620 7767 6420  the ages of wgd 
-00017160: 6973 2031 3132 2e38 3934 3520 6d63 6d63  is 112.8945 mcmc
-00017170: 7472 6565 2e70 793a 3239 360a 2020 2020  tree.py:296.    
-00017180: 2020 2020 2020 2020 2020 2020 2020 6d69                mi
-00017190: 6c6c 696f 6e20 7965 6172 7320 6672 6f6d  llion years from
-000171a0: 2043 6f6e 6361 7465 6e61 7465 6420 7065   Concatenated pe
-000171b0: 7074 6964 650a 2020 2020 2020 2020 2020  ptide.          
-000171c0: 2020 2020 2020 2020 616c 6967 6e6d 656e          alignmen
-000171d0: 7420 616e 6420 3935 2520 6372 6564 6962  t and 95% credib
-000171e0: 696c 6974 7920 696e 7465 7276 616c 7320  ility intervals 
-000171f0: 2843 4929 0a20 2020 2020 2020 2020 2020  (CI).           
-00017200: 2020 2020 2020 2069 7320 3130 312e 3232         is 101.22
-00017210: 342d 3132 332e 3132 3120 6d69 6c6c 696f  4-123.121 millio
-00017220: 6e20 7965 6172 730a 2020 2020 2020 2020  n years.        
-00017230: 2049 4e46 4f20 2020 2020 546f 7461 6c20   INFO     Total 
-00017240: 7275 6e20 7469 6d65 3a20 3239 3137 3573  run time: 29175s
-00017250: 2020 2020 2020 2020 2020 2020 2020 2020                  
-00017260: 2020 2020 2020 2020 2020 2020 2020 636c                cl
-00017270: 692e 7079 3a32 3431 0a20 2020 2020 2020  i.py:241.       
-00017280: 2020 494e 464f 2020 2020 2044 6f6e 6520    INFO     Done 
-00017290: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000172a0: 2020 2020 2020 2020 2020 2020 2020 2020                  
-000172b0: 2020 2020 2020 2020 2020 2020 2020 2063                 c
-000172c0: 6c69 2e70 793a 3234 320a 6060 600a 0a54  li.py:242.```..T
-000172d0: 6f20 7669 7375 616c 697a 6520 7468 6520  o visualize the 
-000172e0: 6461 7465 2c20 7765 2061 6c73 6f20 7072  date, we also pr
-000172f0: 6f76 6964 6564 2061 2070 7974 686f 6e20  ovided a python 
-00017300: 7363 7269 7074 2074 6f20 706c 6f74 2074  script to plot t
-00017310: 6865 2057 4744 2064 6174 6573 2069 6e20  he WGD dates in 
-00017320: 7468 6520 6077 6764 6020 666f 6c64 6572  the `wgd` folder
-00017330: 2e20 5573 6572 7320 6e65 6564 2074 6f20  . Users need to 
-00017340: 6578 7472 6163 7420 7468 6520 7261 7720  extract the raw 
-00017350: 6461 7465 7320 6672 6f6d 2074 6865 2060  dates from the `
-00017360: 6d63 6d63 2e74 7874 6020 666f 7220 7468  mcmc.txt` for th
-00017370: 6520 5747 4420 6e6f 6465 2066 6972 7374  e WGD node first
-00017380: 2061 6e64 2073 6176 6520 6974 2061 7320   and save it as 
-00017390: 6669 6c65 2060 6461 7465 732e 7478 7460  file `dates.txt`
-000173a0: 2028 6f72 2077 6861 7465 7665 7220 7072   (or whatever pr
-000173b0: 6566 6572 7265 6420 6e61 6d65 292e 2041  eferred name). A
-000173c0: 6e20 6578 616d 706c 6520 636f 6d6d 616e  n example comman
-000173d0: 6420 6973 2061 7320 6265 6c6f 772e 0a0a  d is as below...
-000173e0: 6060 600a 7079 7468 6f6e 2024 5041 5448  ```.python $PATH
-000173f0: 2f70 6f73 7470 6c6f 742e 7079 2070 6f73  /postplot.py pos
-00017400: 7464 6973 2064 6174 6573 2e74 7874 202d  tdis dates.txt -
-00017410: 2d70 6572 6365 6e74 696c 6520 3930 202d  -percentile 90 -
-00017420: 2d74 6974 6c65 2022 5747 4420 6461 7465  -title "WGD date
-00017430: 2220 2d2d 6870 6420 2d6f 2022 5261 6e75  " --hpd -o "Ranu
-00017440: 6e63 756c 616c 6573 5f57 4744 5f64 6174  nculales_WGD_dat
-00017450: 652e 7376 6722 0a60 6060 0a0a 215b 5d28  e.svg".```..![](
-00017460: 6461 7461 2f52 616e 756e 6375 6c61 6c65  data/Ranunculale
-00017470: 735f 5747 445f 6461 7465 2e73 7667 290a  s_WGD_date.svg).
-00017480: 0a54 6865 2070 6f73 7465 7269 6f72 206d  .The posterior m
-00017490: 6561 6e2c 206d 6564 6961 6e20 616e 6420  ean, median and 
-000174a0: 6d6f 6465 206f 6620 7468 6520 5261 6e75  mode of the Ranu
-000174b0: 6e63 756c 616c 6573 2057 4744 2061 6765  nculales WGD age
-000174c0: 2069 7320 3131 322e 3932 2c20 3131 332e   is 112.92, 113.
-000174d0: 3434 2061 6e64 2031 3132 2e35 3420 6d79  44 and 112.54 my
-000174e0: 612c 2077 6974 6820 3930 2520 4850 4420  a, with 90% HPD 
-000174f0: 3130 352e 3037 202d 2031 3232 2e33 3220  105.07 - 122.32 
-00017500: 6d79 6120 6173 206d 616e 6966 6573 7465  mya as manifeste
-00017510: 6420 6162 6f76 652e 0a0a 2323 2320 4b73  d above...### Ks
-00017520: 7472 6565 0a0a 496e 2061 6464 6974 696f  tree..In additio
-00017530: 6e20 746f 2070 6169 7277 6973 6520 2a4b  n to pairwise *K
-00017540: 2a3c 7375 623e 533c 2f73 7562 3e20 6573  *<sub>S</sub> es
-00017550: 7469 6d61 7469 6f6e 2c20 6120 2a4b 2a3c  timation, a *K*<
-00017560: 7375 623e 533c 2f73 7562 3e20 7472 6565  sub>S</sub> tree
-00017570: 2077 6974 6820 6272 616e 6368 206c 656e   with branch len
-00017580: 6774 6820 696e 202a 4b2a 3c73 7562 3e53  gth in *K*<sub>S
-00017590: 3c2f 7375 623e 2075 6e69 7420 6361 6e20  </sub> unit can 
-000175a0: 616c 736f 2062 6520 6465 7269 7665 6420  also be derived 
-000175b0: 6672 6f6d 2074 6865 2070 726f 6772 616d  from the program
-000175c0: 2060 7767 6420 6b73 6460 2067 6976 656e   `wgd ksd` given
-000175d0: 2074 6865 206f 7074 696f 6e20 602d 2d6b   the option `--k
-000175e0: 7374 7265 6560 2061 6e64 2060 2d2d 7370  stree` and `--sp
-000175f0: 6563 6965 7374 7265 6560 2e20 4e6f 7465  eciestree`. Note
-00017600: 2074 6861 7420 7468 6520 6164 6469 7469   that the additi
-00017610: 6f6e 616c 206f 7074 696f 6e20 602d 2d6f  onal option `--o
-00017620: 6e6c 7963 6f6e 6361 746b 7374 7265 6560  nlyconcatkstree`
-00017630: 2077 696c 6c20 6f6e 6c79 2063 616c 6c20   will only call 
-00017640: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
-00017650: 7562 3e20 6573 7469 6d61 7469 6f6e 2066  ub> estimation f
-00017660: 6f72 2074 6865 2063 6f6e 6361 7465 6e61  or the concatena
-00017670: 7465 6420 616c 6967 6e6d 656e 7420 7261  ted alignment ra
-00017680: 7468 6572 2074 6861 6e20 616c 6c20 7468  ther than all th
-00017690: 6520 616c 6967 6e6d 656e 7473 2e20 5573  e alignments. Us
-000176a0: 6572 7320 6e65 6564 2074 6f20 7072 6f76  ers need to prov
-000176b0: 6964 6520 6120 7072 6573 6574 2073 7065  ide a preset spe
-000176c0: 6369 6573 2074 7265 6520 666f 7220 7468  cies tree for th
-000176d0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000176e0: 3e20 7472 6565 2069 6e66 6572 656e 6365  > tree inference
-000176f0: 206f 6620 7468 6520 636f 6e63 6174 656e   of the concaten
-00017700: 6174 6564 2061 6c69 676e 6d65 6e74 2077  ated alignment w
-00017710: 6869 6c65 2074 6865 2072 656d 6169 6e69  hile the remaini
-00017720: 6e67 2061 6c69 676e 6d65 6e74 7320 7769  ng alignments wi
-00017730: 6c6c 2062 6520 6167 6169 6e73 7420 616e  ll be against an
-00017740: 2061 7574 6f6d 6174 656c 7920 696e 6665   automately infe
-00017750: 7272 6564 2074 7265 6520 6672 6f6d 2060  rred tree from `
-00017760: 6661 7374 7472 6565 6020 6f72 2060 6971  fasttree` or `iq
-00017770: 7472 6565 602e 2049 6e20 7468 6520 656e  tree`. In the en
-00017780: 642c 2075 7365 7273 2077 696c 6c20 6765  d, users will ge
-00017790: 7420 6120 2a4b 2a3c 7375 623e 533c 2f73  t a *K*<sub>S</s
-000177a0: 7562 3e20 7472 6565 2c20 6120 2a4b 2a3c  ub> tree, a *K*<
-000177b0: 7375 623e 413c 2f73 7562 3e20 7472 6565  sub>A</sub> tree
-000177c0: 2061 6e64 2061 20cf 8920 7472 6565 2070   and a .. tree p
-000177d0: 6572 2066 616d 2061 6e64 2066 6f72 2074  er fam and for t
-000177e0: 6865 2063 6f6e 6361 7465 6e61 7465 6420  he concatenated 
-000177f0: 616c 6967 6e6d 656e 742e 0a0a 6060 600a  alignment...```.
-00017800: 7767 6420 6b73 6420 6461 7461 2f6b 7374  wgd ksd data/kst
-00017810: 7265 655f 6461 7461 2f66 616d 2e74 7376  ree_data/fam.tsv
-00017820: 2064 6174 612f 6b73 7472 6565 5f64 6174   data/kstree_dat
-00017830: 612f 4163 6f72 7573 5f74 6174 6172 696e  a/Acorus_tatarin
-00017840: 6f77 6969 2064 6174 612f 6b73 7472 6565  owii data/kstree
-00017850: 5f64 6174 612f 416d 626f 7265 6c6c 615f  _data/Amborella_
-00017860: 7472 6963 686f 706f 6461 2064 6174 612f  trichopoda data/
-00017870: 6b73 7472 6565 5f64 6174 612f 4171 7569  kstree_data/Aqui
-00017880: 6c65 6769 615f 636f 6572 756c 6561 2064  legia_coerulea d
-00017890: 6174 612f 6b73 7472 6565 5f64 6174 612f  ata/kstree_data/
-000178a0: 4172 6973 746f 6c6f 6368 6961 5f66 696d  Aristolochia_fim
-000178b0: 6272 6961 7461 2064 6174 612f 6b73 7472  briata data/kstr
-000178c0: 6565 5f64 6174 612f 4379 6361 735f 7061  ee_data/Cycas_pa
-000178d0: 6e7a 6869 6875 6165 6e73 6920 2d2d 6b73  nzhihuaensi --ks
-000178e0: 7472 6565 202d 2d73 7065 6369 6573 7472  tree --speciestr
-000178f0: 6565 2064 6174 612f 6b73 7472 6565 5f64  ee data/kstree_d
-00017900: 6174 612f 7370 6563 6965 735f 7472 6565  ata/species_tree
-00017910: 312e 6e77 202d 2d6f 6e6c 7963 6f6e 6361  1.nw --onlyconca
-00017920: 746b 7374 7265 6520 2d6f 2077 6764 5f6b  tkstree -o wgd_k
-00017930: 7374 7265 655f 746f 706f 6c6f 6779 310a  stree_topology1.
-00017940: 6060 600a 0a21 5b5d 2864 6174 612f 6b73  ```..![](data/ks
-00017950: 7472 6565 5f72 6573 756c 7473 2f6b 7374  tree_results/kst
-00017960: 7265 652e 7376 6729 0a0a 4162 6f76 6520  ree.svg)..Above 
-00017970: 7765 2075 7365 6420 7468 7265 6520 616c  we used three al
-00017980: 7465 726e 6174 6976 6520 746f 706f 6c6f  ternative topolo
-00017990: 6769 6573 2074 6f20 696e 6665 7220 7468  gies to infer th
-000179a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
-000179b0: 3e20 7472 6565 2077 6869 6368 206c 6564  > tree which led
-000179c0: 2074 6f20 6469 6666 6572 656e 7420 6272   to different br
-000179d0: 616e 6368 206c 656e 6774 6820 6573 7469  anch length esti
-000179e0: 6d61 7469 6f6e 2e20 4e6f 7465 2074 6861  mation. Note tha
-000179f0: 7420 7468 6520 6661 6d69 6c69 6573 2077  t the families w
-00017a00: 6520 7573 6564 2077 6572 6520 6f6e 6c79  e used were only
-00017a10: 2074 776f 2067 6c6f 6261 6c20 4d52 4248   two global MRBH
-00017a20: 2066 616d 696c 6965 7320 666f 7220 7468   families for th
-00017a30: 6520 7075 7270 6f73 6520 6f66 2069 6c6c  e purpose of ill
-00017a40: 7573 7472 6174 696f 6e2e 2054 6f20 6163  ustration. To ac
-00017a50: 7175 6972 6520 616e 2061 6363 7572 6174  quire an accurat
-00017a60: 6520 7072 6f66 696c 6520 6f66 2074 6865  e profile of the
-00017a70: 2073 7562 7374 6974 7574 696f 6e20 7261   substitution ra
-00017a80: 7465 2076 6172 6961 7469 6f6e 2c20 6f72  te variation, or
-00017a90: 7468 6f6c 6f67 7565 7320 6174 2074 6865  thologues at the
-00017aa0: 2077 686f 6c65 2067 656e 6f6d 6520 7363   whole genome sc
-00017ab0: 616c 6520 7368 6f75 6c64 2062 6520 7573  ale should be us
-00017ac0: 6564 2e0a 0a23 2320 4369 7461 7469 6f6e  ed...## Citation
-00017ad0: 0a20 0a50 6c65 6173 6520 6369 7465 2075  . .Please cite u
-00017ae0: 7320 6174 2068 7474 7073 3a2f 2f64 6f69  s at https://doi
-00017af0: 2e6f 7267 2f31 302e 3130 3037 2f39 3738  .org/10.1007/978
-00017b00: 2d31 2d30 3731 362d 3235 3631 2d33 5f31  -1-0716-2561-3_1
-00017b10: 2061 6e64 2068 7474 7073 3a2f 2f64 6f69   and https://doi
-00017b20: 2e6f 7267 2f31 302e 3130 3933 2f62 696f  .org/10.1093/bio
-00017b30: 696e 666f 726d 6174 6963 732f 6274 6165  informatics/btae
-00017b40: 3237 322e 0a0a 6060 600a 4865 6e67 6368  272...```.Hengch
-00017b50: 6920 4368 656e 2c20 4172 7468 7572 205a  i Chen, Arthur Z
-00017b60: 7761 656e 6570 6f65 6c20 2832 3032 3329  waenepoel (2023)
-00017b70: 2e20 496e 6665 7265 6e63 6520 6f66 2041  . Inference of A
-00017b80: 6e63 6965 6e74 2050 6f6c 7970 6c6f 6964  ncient Polyploid
-00017b90: 7920 6672 6f6d 2047 656e 6f6d 6963 2044  y from Genomic D
-00017ba0: 6174 612e 2049 6e3a 2056 616e 2064 6520  ata. In: Van de 
-00017bb0: 5065 6572 2c20 592e 2028 6564 7329 2050  Peer, Y. (eds) P
-00017bc0: 6f6c 7970 6c6f 6964 792e 204d 6574 686f  olyploidy. Metho
-00017bd0: 6473 2069 6e20 4d6f 6c65 6375 6c61 7220  ds in Molecular 
-00017be0: 4269 6f6c 6f67 792c 2076 6f6c 2032 3534  Biology, vol 254
-00017bf0: 352e 2048 756d 616e 612c 204e 6577 2059  5. Humana, New Y
-00017c00: 6f72 6b2c 204e 592e 2068 7474 7073 3a2f  ork, NY. https:/
-00017c10: 2f64 6f69 2e6f 7267 2f31 302e 3130 3037  /doi.org/10.1007
-00017c20: 2f39 3738 2d31 2d30 3731 362d 3235 3631  /978-1-0716-2561
-00017c30: 2d33 5f31 0a48 656e 6763 6869 2043 6865  -3_1.Hengchi Che
-00017c40: 6e2c 2041 7274 6875 7220 5a77 6165 6e65  n, Arthur Zwaene
-00017c50: 706f 656c 2c20 5976 6573 2056 616e 2064  poel, Yves Van d
-00017c60: 6520 5065 6572 2028 3230 3234 292e 2077  e Peer (2024). w
-00017c70: 6764 2076 323a 2061 2073 7569 7465 206f  gd v2: a suite o
-00017c80: 6620 746f 6f6c 7320 746f 2075 6e63 6f76  f tools to uncov
-00017c90: 6572 2061 6e64 2064 6174 6520 616e 6369  er and date anci
-00017ca0: 656e 7420 706f 6c79 706c 6f69 6479 2061  ent polyploidy a
-00017cb0: 6e64 2077 686f 6c65 2d67 656e 6f6d 6520  nd whole-genome 
-00017cc0: 6475 706c 6963 6174 696f 6e2e 2042 696f  duplication. Bio
-00017cd0: 696e 666f 726d 6174 6963 732c 2068 7474  informatics, htt
-00017ce0: 7073 3a2f 2f64 6f69 2e6f 7267 2f31 302e  ps://doi.org/10.
-00017cf0: 3130 3933 2f62 696f 696e 666f 726d 6174  1093/bioinformat
-00017d00: 6963 732f 6274 6165 3237 320a 6060 600a  ics/btae272.```.
-00017d10: 0a46 6f72 2063 6974 6174 696f 6e20 6f66  .For citation of
-00017d20: 2074 6865 2074 6f6f 6c73 2075 7365 6420   the tools used 
-00017d30: 696e 2077 6764 2c20 706c 6561 7365 2063  in wgd, please c
-00017d40: 6f6e 7375 6c74 2074 6865 2064 6f63 756d  onsult the docum
-00017d50: 656e 7461 7469 6f6e 2061 740a 6874 7470  entation at.http
-00017d60: 733a 2f2f 7767 6476 322e 7265 6164 7468  s://wgdv2.readth
-00017d70: 6564 6f63 732e 696f 2f65 6e2f 6c61 7465  edocs.io/en/late
-00017d80: 7374 2f63 6974 6174 696f 6e2e 6874 6d6c  st/citation.html
-00017d90: 2e0a 0a                                  ...
+00000000: 3c64 6976 2061 6c69 676e 3d22 6365 6e74  <div align="cent
+00000010: 6572 223e 0a0a 2320 6077 6764 2076 3260  er">..# `wgd v2`
+00000020: 203a 2061 2073 7569 7465 2074 6f6f 6c20   : a suite tool 
+00000030: 6f66 2057 4744 2069 6e66 6572 656e 6365  of WGD inference
+00000040: 2061 6e64 2074 696d 696e 670a 0a5b 215b   and timing..[![
+00000050: 4275 696c 6420 5374 6174 7573 5d28 6874  Build Status](ht
+00000060: 7470 733a 2f2f 6170 702e 7472 6176 6973  tps://app.travis
+00000070: 2d63 692e 636f 6d2f 6865 6368 652d 7073  -ci.com/heche-ps
+00000080: 622f 7767 642e 7376 673f 6272 616e 6368  b/wgd.svg?branch
+00000090: 3d70 6879 6c6f 6461 7469 6e67 295d 2868  =phylodating)](h
+000000a0: 7474 7073 3a2f 2f61 7070 2e74 7261 7669  ttps://app.travi
+000000b0: 732d 6369 2e63 6f6d 2f68 6563 6865 2d70  s-ci.com/heche-p
+000000c0: 7362 2f77 6764 290a 5b21 5b44 6f63 756d  sb/wgd).[![Docum
+000000d0: 656e 7461 7469 6f6e 2053 7461 7475 735d  entation Status]
+000000e0: 2868 7474 7073 3a2f 2f72 6561 6474 6865  (https://readthe
+000000f0: 646f 6373 2e6f 7267 2f70 726f 6a65 6374  docs.org/project
+00000100: 732f 7767 6476 322f 6261 6467 652f 3f76  s/wgdv2/badge/?v
+00000110: 6572 7369 6f6e 3d6c 6174 6573 7429 5d28  ersion=latest)](
+00000120: 6874 7470 733a 2f2f 7767 6476 322e 7265  https://wgdv2.re
+00000130: 6164 7468 6564 6f63 732e 696f 2f65 6e2f  adthedocs.io/en/
+00000140: 6c61 7465 7374 2f3f 6261 6467 653d 6c61  latest/?badge=la
+00000150: 7465 7374 290a 5b21 5b6c 6963 656e 7365  test).[![license
+00000160: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
+00000170: 6965 6c64 732e 696f 2f70 7970 692f 6c2f  ields.io/pypi/l/
+00000180: 7767 642e 7376 6729 5d28 6874 7470 733a  wgd.svg)](https:
+00000190: 2f2f 7079 7069 2e70 7974 686f 6e2e 6f72  //pypi.python.or
+000001a0: 672f 7079 7069 2f77 6764 290a 5b21 5b4c  g/pypi/wgd).[![L
+000001b0: 6174 6573 7420 5079 5049 2076 6572 7369  atest PyPI versi
+000001c0: 6f6e 5d28 6874 7470 733a 2f2f 696d 672e  on](https://img.
+000001d0: 7368 6965 6c64 732e 696f 2f70 7970 692f  shields.io/pypi/
+000001e0: 762f 7767 642e 7376 6729 5d28 6874 7470  v/wgd.svg)](http
+000001f0: 733a 2f2f 7079 7069 2e70 7974 686f 6e2e  s://pypi.python.
+00000200: 6f72 672f 7079 7069 2f77 6764 290a 5b21  org/pypi/wgd).[!
+00000210: 5b69 6e73 7461 6c6c 2077 6974 6820 6269  [install with bi
+00000220: 6f63 6f6e 6461 5d28 6874 7470 733a 2f2f  oconda](https://
+00000230: 696d 672e 7368 6965 6c64 732e 696f 2f62  img.shields.io/b
+00000240: 6164 6765 2f69 6e73 7461 6c6c 2532 3077  adge/install%20w
+00000250: 6974 682d 6269 6f63 6f6e 6461 2d62 7269  ith-bioconda-bri
+00000260: 6768 7467 7265 656e 2e73 7667 3f73 7479  ghtgreen.svg?sty
+00000270: 6c65 3d66 6c61 7429 5d28 6874 7470 3a2f  le=flat)](http:/
+00000280: 2f62 696f 636f 6e64 612e 6769 7468 7562  /bioconda.github
+00000290: 2e69 6f2f 7265 6369 7065 732f 7767 642f  .io/recipes/wgd/
+000002a0: 5245 4144 4d45 2e68 746d 6c29 0a5b 215b  README.html).[![
+000002b0: 446f 776e 6c6f 6164 735d 2868 7474 7073  Downloads](https
+000002c0: 3a2f 2f70 6570 792e 7465 6368 2f62 6164  ://pepy.tech/bad
+000002d0: 6765 2f77 6764 295d 2868 7474 7073 3a2f  ge/wgd)](https:/
+000002e0: 2f70 6570 792e 7465 6368 2f70 726f 6a65  /pepy.tech/proje
+000002f0: 6374 2f77 6764 290a 0a2a 2a48 656e 6763  ct/wgd)..**Hengc
+00000300: 6869 2043 6865 6e2c 2041 7274 6875 7220  hi Chen, Arthur 
+00000310: 5a77 6165 6e65 706f 656c 2c20 5976 6573  Zwaenepoel, Yves
+00000320: 2056 616e 2064 6520 5065 6572 2a2a 0a0a   Van de Peer**..
+00000330: 5b2a 2a42 696f 696e 666f 726d 6174 6963  [**Bioinformatic
+00000340: 7320 2620 4576 6f6c 7574 696f 6e61 7279  s & Evolutionary
+00000350: 2047 656e 6f6d 6963 7320 4772 6f75 702a   Genomics Group*
+00000360: 2a5d 2868 7474 7073 3a2f 2f77 7777 2e76  *](https://www.v
+00000370: 616e 6465 7065 6572 6c61 622e 6f72 672f  andepeerlab.org/
+00000380: 292a 2a2c 2056 4942 2d55 4765 6e74 2043  )**, VIB-UGent C
+00000390: 656e 7465 7220 666f 7220 506c 616e 7420  enter for Plant 
+000003a0: 5379 7374 656d 7320 4269 6f6c 6f67 792a  Systems Biology*
+000003b0: 2a0a 0a5b 2a2a 496e 7472 6f64 7563 7469  *..[**Introducti
+000003c0: 6f6e 2a2a 5d28 2369 6e74 726f 6475 6374  on**](#introduct
+000003d0: 696f 6e29 207c 200a 5b2a 2a49 6e73 7461  ion) | .[**Insta
+000003e0: 6c6c 6174 696f 6e2a 2a5d 2823 696e 7374  llation**](#inst
+000003f0: 616c 6c61 7469 6f6e 2920 7c0a 5b2a 2a50  allation) |.[**P
+00000400: 6970 656c 696e 6573 2a2a 5d28 2370 6970  ipelines**](#pip
+00000410: 656c 696e 6573 2920 7c0a 5b2a 2a50 6172  elines) |.[**Par
+00000420: 616d 6574 6572 732a 2a5d 2823 7061 7261  ameters**](#para
+00000430: 6d65 7465 7273 2920 7c20 0a5b 2a2a 5573  meters) | .[**Us
+00000440: 6167 652a 2a5d 2823 7573 6167 6529 207c  age**](#usage) |
+00000450: 200a 5b2a 2a49 6c6c 7573 7472 6174 696f   .[**Illustratio
+00000460: 6e2a 2a5d 2823 696c 6c75 7374 7261 7469  n**](#illustrati
+00000470: 6f6e 2920 7c0a 5b2a 2a44 6f63 756d 656e  on) |.[**Documen
+00000480: 7461 7469 6f6e 2a2a 5d28 6874 7470 733a  tation**](https:
+00000490: 2f2f 7767 6476 322e 7265 6164 7468 6564  //wgdv2.readthed
+000004a0: 6f63 732e 696f 2f65 6e2f 6c61 7465 7374  ocs.io/en/latest
+000004b0: 2f3f 6261 6467 653d 6c61 7465 7374 2920  /?badge=latest) 
+000004c0: 7c0a 5b2a 2a43 6974 6174 696f 6e2a 2a5d  |.[**Citation**]
+000004d0: 2823 6369 7461 7469 6f6e 290a 0a3c 2f64  (#citation)..</d
+000004e0: 6976 3e0a 0a60 7767 6420 7632 6020 6973  iv>..`wgd v2` is
+000004f0: 2061 2070 7974 686f 6e20 7061 636b 6167   a python packag
+00000500: 6520 7570 6772 6164 6564 2066 726f 6d20  e upgraded from 
+00000510: 7468 6520 6f72 6967 696e 616c 2060 7767  the original `wg
+00000520: 6460 2070 6163 6b61 6765 2061 696d 696e  d` package aimin
+00000530: 6720 666f 7220 7468 6520 696e 6665 7265  g for the infere
+00000540: 6e63 6520 616e 6420 7469 6d69 6e67 206f  nce and timing o
+00000550: 6620 616e 6369 656e 7420 7768 6f6c 652d  f ancient whole-
+00000560: 6765 6e6f 6d65 2064 7570 6c69 6361 7469  genome duplicati
+00000570: 6f6e 2028 5747 4429 2065 7665 6e74 732e  on (WGD) events.
+00000580: 2046 6f72 2074 6865 2070 726f 706f 7365   For the propose
+00000590: 206f 6620 696c 6c75 7374 7261 7469 6e67   of illustrating
+000005a0: 2074 6865 2070 7269 6e63 6970 6c65 2061   the principle a
+000005b0: 6e64 2075 7361 6765 206f 6620 6077 6764  nd usage of `wgd
+000005c0: 2076 3260 2c20 7765 2063 6f6d 7069 6c65   v2`, we compile
+000005d0: 6420 7468 6973 2064 6f63 756d 656e 7461  d this documenta
+000005e0: 7469 6f6e 2e20 4265 6c6f 7720 7765 2066  tion. Below we f
+000005f0: 6972 7374 2067 6176 6520 616e 2069 6e74  irst gave an int
+00000600: 726f 6475 6374 696f 6e20 6f76 6572 2074  roduction over t
+00000610: 6865 2073 636f 7065 2061 6e64 206d 6563  he scope and mec
+00000620: 6861 6e69 736d 206f 6620 6077 6764 2076  hanism of `wgd v
+00000630: 3260 2061 6e64 2074 6865 6e20 7468 6520  2` and then the 
+00000640: 7072 6163 7469 6361 6c20 696e 666f 726d  practical inform
+00000650: 6174 696f 6e20 7065 7274 6169 6e69 6e67  ation pertaining
+00000660: 2074 6f20 7468 6520 696e 7374 616c 6c61   to the installa
+00000670: 7469 6f6e 2061 6e64 2075 7361 6765 2e20  tion and usage. 
+00000680: 416e 2065 7861 6d70 6c61 7220 776f 726b  An examplar work
+00000690: 666c 6f77 2069 7320 7072 6f76 6964 6564  flow is provided
+000006a0: 2069 6e20 7468 6520 7475 746f 7269 616c   in the tutorial
+000006b0: 2073 6563 7469 6f6e 206f 6e20 686f 7720   section on how 
+000006c0: 746f 2073 6565 6b20 6576 6964 656e 6365  to seek evidence
+000006d0: 2066 6f72 2061 2070 7574 6174 6976 6520   for a putative 
+000006e0: 5747 4420 6576 656e 7420 616e 6420 7065  WGD event and pe
+000006f0: 7266 6f72 6d20 7072 6f70 6572 2074 696d  rform proper tim
+00000700: 696e 6720 7769 7468 2061 2066 7265 7368  ing with a fresh
+00000710: 6c79 206f 6274 6169 6e65 6420 6765 6e6f  ly obtained geno
+00000720: 6d65 2061 7373 656d 626c 7920 696e 2068  me assembly in h
+00000730: 616e 642e 2046 6f72 2074 686f 7365 2077  and. For those w
+00000740: 686f 2061 7265 2069 6e74 6572 6573 7465  ho are intereste
+00000750: 6420 696e 206d 6f72 6520 7468 656f 7265  d in more theore
+00000760: 7469 6361 6c20 6465 7461 696c 732c 2077  tical details, w
+00000770: 6520 7265 636f 6d6d 656e 6420 7475 726e  e recommend turn
+00000780: 696e 6720 746f 206f 7572 2070 6170 6572  ing to our paper
+00000790: 2061 6e64 2062 6f6f 6b20 6368 6170 7465   and book chapte
+000007a0: 7220 666f 7220 6d6f 7265 2064 6574 6169  r for more detai
+000007b0: 6c65 6420 6465 7363 7269 7074 696f 6e20  led description 
+000007c0: 616e 6420 696e 7369 6768 7466 756c 2064  and insightful d
+000007d0: 6973 6375 7373 696f 6e73 2e20 5468 6520  iscussions. The 
+000007e0: 6b65 7920 696d 7072 6f76 6564 2066 6561  key improved fea
+000007f0: 7475 7265 7320 6f66 2060 7767 6420 7632  tures of `wgd v2
+00000800: 6020 6172 6520 6465 6d6f 6e73 7472 6174  ` are demonstrat
+00000810: 6564 2069 6e20 6f75 7220 6c61 7465 7374  ed in our latest
+00000820: 206d 616e 7573 6372 6970 7420 6f6e 205b   manuscript on [
+00000830: 4269 6f69 6e66 6f72 6d61 7469 6373 5d28  Bioinformatics](
+00000840: 6874 7470 733a 2f2f 646f 692e 6f72 672f  https://doi.org/
+00000850: 3130 2e31 3039 332f 6269 6f69 6e66 6f72  10.1093/bioinfor
+00000860: 6d61 7469 6373 2f62 7461 6532 3732 292e  matics/btae272).
+00000870: 2049 6620 796f 7520 7573 6520 6077 6764   If you use `wgd
+00000880: 2076 3260 2069 6e20 796f 7572 2072 6573   v2` in your res
+00000890: 6561 7263 682c 2070 6c65 6173 6520 6369  earch, please ci
+000008a0: 7465 2075 7320 6173 2073 7567 6765 7374  te us as suggest
+000008b0: 6564 2069 6e20 5b43 6974 6174 696f 6e5d  ed in [Citation]
+000008c0: 2823 6369 7461 7469 6f6e 2920 7365 6374  (#citation) sect
+000008d0: 696f 6e2e 0a0a 2323 2049 6e74 726f 6475  ion...## Introdu
+000008e0: 6374 696f 6e0a 0a50 6f6c 7970 6c6f 6964  ction..Polyploid
+000008f0: 697a 6174 696f 6e73 2c20 7468 6520 6576  izations, the ev
+00000900: 6f6c 7574 696f 6e61 7279 2070 726f 6365  olutionary proce
+00000910: 7373 2074 6861 7420 7468 6520 656e 7469  ss that the enti
+00000920: 7265 2067 656e 6f6d 6520 6f66 2061 6e20  re genome of an 
+00000930: 6f72 6761 6e69 736d 2069 7320 6475 706c  organism is dupl
+00000940: 6963 6174 6564 2c20 616c 736f 206e 616d  icated, also nam
+00000950: 6564 2061 7320 7768 6f6c 652d 6765 6e6f  ed as whole-geno
+00000960: 6d65 2064 7570 6c69 6361 7469 6f6e 7320  me duplications 
+00000970: 2857 4744 7329 2c20 6f63 6375 7220 7265  (WGDs), occur re
+00000980: 6375 7272 656e 746c 7920 6163 726f 7373  currently across
+00000990: 2074 6865 2074 7265 6520 6f66 206c 6966   the tree of lif
+000009a0: 652e 2054 6865 7265 2061 7265 2074 776f  e. There are two
+000009b0: 206d 6f64 6573 206f 6620 706f 6c79 706c   modes of polypl
+000009c0: 6f69 6469 7a61 7469 6f6e 732c 2061 7574  oidizations, aut
+000009d0: 6f70 6f6c 7970 6c6f 6964 697a 6174 696f  opolyploidizatio
+000009e0: 6e73 2061 6e64 2061 6c6c 6f70 6f6c 7970  ns and allopolyp
+000009f0: 6c6f 6964 697a 6174 696f 6e73 2e20 4175  loidizations. Au
+00000a00: 746f 706f 6c79 706c 6f69 6469 7a61 7469  topolyploidizati
+00000a10: 6f6e 7320 6172 6520 7468 6520 6475 706c  ons are the dupl
+00000a20: 6963 6174 696f 6e20 6f66 2074 6865 2073  ication of the s
+00000a30: 616d 6520 6765 6e6f 6d65 2c20 7265 7375  ame genome, resu
+00000a40: 6c74 696e 6720 696e 2074 776f 2069 6465  lting in two ide
+00000a50: 6e74 6963 616c 2073 7562 6765 6e6f 6d65  ntical subgenome
+00000a60: 7320 6174 2074 6865 2074 696d 6520 6974  s at the time it
+00000a70: 2065 6d65 7267 6564 2e20 5768 696c 6520   emerged. While 
+00000a80: 7468 6520 616c 6c6f 706f 6c79 706c 6f69  the allopolyploi
+00000a90: 6469 7a61 7469 6f6e 7320 6172 6520 6e6f  dizations are no
+00000aa0: 726d 616c 6c79 2061 6368 6965 7665 6420  rmally achieved 
+00000ab0: 696e 2074 776f 2073 7465 7073 2c20 6669  in two steps, fi
+00000ac0: 7273 7420 7468 6520 6879 6272 6964 697a  rst the hybridiz
+00000ad0: 6174 696f 6e20 6265 7477 6565 6e20 7477  ation between tw
+00000ae0: 6f20 6469 6666 6572 656e 7420 7370 6563  o different spec
+00000af0: 6965 732c 2072 6573 756c 7469 6e67 2069  ies, resulting i
+00000b00: 6e20 7468 6520 6172 6973 696e 6720 6f66  n the arising of
+00000b10: 2074 7261 6e73 6965 6e74 2068 6f6d 6f70   transient homop
+00000b20: 6c6f 6964 792c 7365 636f 6e64 2074 6865  loidy,second the
+00000b30: 2064 7570 6c69 6361 7469 6f6e 206f 6620   duplication of 
+00000b40: 7468 6520 686f 6d6f 706c 6f69 6479 2c20  the homoploidy, 
+00000b50: 7265 7375 6c74 696e 6720 696e 2074 6865  resulting in the
+00000b60: 2065 6d65 7267 656e 6365 206f 6620 616c   emergence of al
+00000b70: 6c6f 706f 6c79 706c 6f69 6479 2e20 4475  lopolyploidy. Du
+00000b80: 6520 746f 2074 6865 2075 6e73 7461 6269  e to the unstabi
+00000b90: 6c69 7479 2061 6e64 2075 6e62 616c 616e  lity and unbalan
+00000ba0: 6365 6420 7465 7472 6173 6f6d 6963 2069  ced tetrasomic i
+00000bb0: 6e68 6572 6974 616e 6365 2c20 666f 7220  nheritance, for 
+00000bc0: 696e 7374 616e 6365 2074 6865 206e 7563  instance the nuc
+00000bd0: 6c65 6172 2d63 7974 6f70 6c61 736d 6963  lear-cytoplasmic
+00000be0: 2069 6e63 6f6d 7061 7469 6269 6c69 7479   incompatibility
+00000bf0: 2c20 7468 6520 706f 6c79 706c 6f69 6479  , the polyploidy
+00000c00: 2067 656e 6f6d 6520 7769 6c6c 2074 6865   genome will the
+00000c10: 6e20 6578 7065 7269 656e 6365 2061 2070  n experience a p
+00000c20: 726f 6365 7373 2063 616c 6c65 6420 6469  rocess called di
+00000c30: 706c 6f69 6469 7a61 7469 6f6e 2c20 616c  ploidization, al
+00000c40: 736f 206e 616d 6564 2061 7320 6672 6163  so named as frac
+00000c50: 7469 6f6e 6174 696f 6e2c 2064 7572 696e  tionation, durin
+00000c60: 6720 7768 6963 6820 6120 6c61 7267 6520  g which a large 
+00000c70: 706f 7274 696f 6e20 6f66 2067 656e 6520  portion of gene 
+00000c80: 6475 706c 6963 6174 6573 2077 696c 6c20  duplicates will 
+00000c90: 6765 7420 6c6f 7374 2061 6e64 206f 6e6c  get lost and onl
+00000ca0: 7920 6120 6672 6163 7469 6f6e 2063 616e  y a fraction can
+00000cb0: 2062 6520 7265 7461 696e 6564 2e20 5468   be retained. Th
+00000cc0: 6520 7472 6163 6573 206f 6620 706f 6c79  e traces of poly
+00000cd0: 706c 6f69 6469 7a61 7469 6f6e 7320 6361  ploidizations ca
+00000ce0: 6e20 6265 2074 6875 7320 756e 6561 7274  n be thus uneart
+00000cf0: 6865 6420 6672 6f6d 2074 6865 7365 2072  hed from these r
+00000d00: 6574 6169 6e65 6420 6765 6e65 2064 7570  etained gene dup
+00000d10: 6c69 6361 7465 732e 2054 6872 6565 2061  licates. Three a
+00000d20: 7070 726f 6163 6865 7320 6261 7365 6420  pproaches based 
+00000d30: 6f6e 2067 656e 6520 6475 706c 6963 6174  on gene duplicat
+00000d40: 6573 2c20 6e61 6d65 6c79 2c20 2a4b 2a3c  es, namely, *K*<
+00000d50: 7375 623e 533c 2f73 7562 3e20 6d65 7468  sub>S</sub> meth
+00000d60: 6f64 2c20 6765 6e65 2074 7265 6520 2d20  od, gene tree - 
+00000d70: 7370 6563 6965 7320 7472 6565 2072 6563  species tree rec
+00000d80: 6f6e 6369 6c69 6174 696f 6e20 6d65 7468  onciliation meth
+00000d90: 6f64 2061 6e64 2073 796e 7465 6e79 206d  od and synteny m
+00000da0: 6574 686f 642c 2061 7265 2063 6f6d 6d6f  ethod, are commo
+00000db0: 6e6c 7920 7573 6564 2069 6e20 6465 7465  nly used in dete
+00000dc0: 6374 696e 6720 6576 6964 656e 6365 2066  cting evidence f
+00000dd0: 6f72 2057 4744 732e 2054 6865 2067 656e  or WGDs. The gen
+00000de0: 6520 7472 6565 202d 2073 7065 6369 6573  e tree - species
+00000df0: 2074 7265 6520 7265 636f 6e63 696c 6961   tree reconcilia
+00000e00: 7469 6f6e 206d 6574 686f 6420 6973 206e  tion method is n
+00000e10: 6f74 2077 6974 6869 6e20 7468 6520 7363  ot within the sc
+00000e20: 6f70 6520 6f66 2060 7767 6420 7632 602c  ope of `wgd v2`,
+00000e30: 2062 7574 2077 6520 6b69 6e64 6c79 2072   but we kindly r
+00000e40: 6566 6572 2072 6561 6465 7273 2077 686f  efer readers who
+00000e50: 2061 7265 2069 6e74 6572 6573 7465 6420   are interested 
+00000e60: 746f 2074 6865 2070 6879 6c6f 6765 6e6f  to the phylogeno
+00000e70: 6d69 6320 7072 6f67 7261 6d20 6465 7665  mic program deve
+00000e80: 6c6f 7065 6420 6279 2041 7274 6875 7220  loped by Arthur 
+00000e90: 5a77 6165 6e65 706f 656c 206e 616d 6564  Zwaenepoel named
+00000ea0: 205b 5748 414c 455d 2868 7474 7073 3a2f   [WHALE](https:/
+00000eb0: 2f67 6974 6875 622e 636f 6d2f 6172 7a77  /github.com/arzw
+00000ec0: 612f 5768 616c 652e 6a6c 2920 616e 6420  a/Whale.jl) and 
+00000ed0: 7468 6520 6173 736f 6369 6174 6564 205b  the associated [
+00000ee0: 7061 7065 725d 2868 7474 7073 3a2f 2f64  paper](https://d
+00000ef0: 6f69 2e6f 7267 2f31 302e 3130 3933 2f6d  oi.org/10.1093/m
+00000f00: 6f6c 6265 762f 6d73 7a30 3838 2920 666f  olbev/msz088) fo
+00000f10: 7220 6d6f 7265 2074 6563 686e 6963 616c  r more technical
+00000f20: 2061 6e64 2074 6865 6f72 6574 6963 616c   and theoretical
+00000f30: 2064 6574 6169 6c73 2e0a 0a54 6865 202a   details...The *
+00000f40: 4b2a 3c73 7562 3e53 3c2f 7375 623e 206d  K*<sub>S</sub> m
+00000f50: 6574 686f 6420 6973 2065 7374 6162 6c69  ethod is establi
+00000f60: 7368 6564 206f 6e20 6120 6d6f 6465 6c20  shed on a model 
+00000f70: 6f66 2067 656e 6520 6661 6d69 6c79 2065  of gene family e
+00000f80: 766f 6c75 7469 6f6e 2074 6861 7420 6561  volution that ea
+00000f90: 6368 2067 656e 6520 6661 6d69 6c79 2069  ch gene family i
+00000fa0: 7320 616c 6c6f 7765 6420 746f 2065 766f  s allowed to evo
+00000fb0: 6c76 6520 7669 6120 6765 6e65 2064 7570  lve via gene dup
+00000fc0: 6c69 6361 7469 6f6e 2061 6e64 206c 6f73  lication and los
+00000fd0: 732e 204e 6f74 6520 7468 6174 2074 6865  s. Note that the
+00000fe0: 2067 656e 6520 6661 6d69 6c79 2068 6572   gene family her
+00000ff0: 6520 6973 2061 7373 756d 6564 2074 6f20  e is assumed to 
+00001000: 6265 2074 6865 2063 6c75 7374 6572 206f  be the cluster o
+00001010: 6620 616c 6c20 6765 6e65 7320 6465 7363  f all genes desc
+00001020: 656e 6465 6420 6672 6f6d 2061 6e20 616e  ended from an an
+00001030: 6365 7374 7261 6c20 6765 6e65 2069 6e20  cestral gene in 
+00001040: 6120 7369 6e67 6c65 2067 656e 6f6d 652e  a single genome.
+00001050: 2052 6563 6f76 6572 696e 6720 7468 6520   Recovering the 
+00001060: 6765 6e65 2074 7265 6520 6f66 2073 7563  gene tree of suc
+00001070: 6820 6765 6e65 2066 616d 696c 7920 696e  h gene family in
+00001080: 666f 726d 7320 7468 6520 7469 6d69 6e67  forms the timing
+00001090: 2c20 7363 696c 6963 6574 2074 6865 2061  , scilicet the a
+000010a0: 6765 2c20 6f66 2067 656e 6520 6475 706c  ge, of gene dupl
+000010b0: 6963 6174 696f 6e20 6576 656e 7473 2e20  ication events. 
+000010c0: 5468 6520 6167 6520 7265 6665 7265 6420  The age refered 
+000010d0: 6865 7265 2c20 6973 206e 6f74 2069 6e20  here, is not in 
+000010e0: 7265 616c 2067 656f 6c6f 6769 6361 6c20  real geological 
+000010f0: 7469 6d65 2c20 6275 7420 696e 2074 6865  time, but in the
+00001100: 2075 6e69 7420 6f66 2065 766f 6c75 7469   unit of evoluti
+00001110: 6f6e 6172 7920 6469 7374 616e 6365 2c20  onary distance, 
+00001120: 692e 652e 2c20 7468 6520 6e75 6d62 6572  i.e., the number
+00001130: 206f 6620 7375 6273 7469 7475 7469 6f6e   of substitution
+00001140: 7320 7065 7220 7369 7465 2e20 5768 656e  s per site. When
+00001150: 2074 6865 2065 766f 6c75 7469 6f6e 6172   the evolutionar
+00001160: 7920 7261 7465 2072 656d 6169 6e73 2061  y rate remains a
+00001170: 7070 726f 7869 6d61 7465 6c79 2063 6f6e  pproximately con
+00001180: 7374 616e 742c 2074 6865 2065 766f 6c75  stant, the evolu
+00001190: 7469 6f6e 6172 7920 6469 7374 616e 6365  tionary distance
+000011a0: 2069 7320 7468 656e 2073 7570 706f 7365   is then suppose
+000011b0: 6420 746f 2062 6520 7072 6f70 6f72 7469  d to be proporti
+000011c0: 6f6e 616c 2074 6f20 7468 6520 7265 616c  onal to the real
+000011d0: 2065 766f 6c75 7469 6f6e 6172 7920 7469   evolutionary ti
+000011e0: 6d65 2e20 5468 6520 7379 6e6f 6e79 6d6f  me. The synonymo
+000011f0: 7573 2064 6973 7461 6e63 6520 2a4b 2a3c  us distance *K*<
+00001200: 7375 623e 533c 2f73 7562 3e2c 2074 6865  sub>S</sub>, the
+00001210: 206e 756d 6265 7220 6f66 2073 796e 6f6e   number of synon
+00001220: 796d 6f75 7320 7375 6273 7469 7475 7469  ymous substituti
+00001230: 6f6e 7320 7065 7220 7379 6e6f 6e79 6d6f  ons per synonymo
+00001240: 7573 2073 6974 652c 2069 7320 7375 6368  us site, is such
+00001250: 2063 616e 6469 6461 7465 2074 6861 7420   candidate that 
+00001260: 7379 6e6f 6e79 6d6f 7573 2073 7562 7374  synonymous subst
+00001270: 6974 7574 696f 6e73 2077 6f75 6c64 206e  itutions would n
+00001280: 6f74 2069 6e63 7572 2074 6865 2063 6861  ot incur the cha
+00001290: 6e67 6520 6f66 2061 6d69 6e6f 2061 6369  nge of amino aci
+000012a0: 6420 616e 6420 6172 6520 7468 7573 2072  d and are thus r
+000012b0: 6567 6172 6465 6420 6173 206e 6575 7472  egarded as neutr
+000012c0: 616c 2c20 7768 6963 6820 6163 636f 7264  al, which accord
+000012d0: 696e 6720 746f 2074 6865 206e 6575 7472  ing to the neutr
+000012e0: 616c 2074 6865 6f72 7920 7368 6f75 6c64  al theory should
+000012f0: 206f 6363 7572 2069 6e20 636f 6e73 7461   occur in consta
+00001300: 6e74 2072 6174 652e 2047 6976 656e 2061  nt rate. Given a
+00001310: 206d 6f64 656c 206f 6620 6765 6e65 2066   model of gene f
+00001320: 616d 696c 7920 7468 6174 2061 6c6c 6f77  amily that allow
+00001330: 7320 7468 6520 6765 6e65 2074 6f20 6475  s the gene to du
+00001340: 706c 6963 6174 6520 616e 6420 6765 7420  plicate and get 
+00001350: 6c6f 7374 2069 6e20 6120 6669 7865 6420  lost in a fixed 
+00001360: 7261 7465 2c20 6f6e 6520 6361 6e20 6465  rate, one can de
+00001370: 7269 7665 2074 6861 7420 7468 6520 7072  rive that the pr
+00001380: 6f62 6162 696c 6974 7920 6465 6e73 6974  obability densit
+00001390: 7920 6675 6e63 7469 6f6e 206f 6620 7468  y function of th
+000013a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000013b0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+000013c0: 6f6e 206f 6620 7265 7461 696e 6564 2067  on of retained g
+000013d0: 656e 6520 6475 706c 6963 6174 6573 2069  ene duplicates i
+000013e0: 7320 6120 7175 6173 692d 6578 706f 6e65  s a quasi-expone
+000013f0: 6e74 6961 6c20 6675 6e63 7469 6f6e 2074  ntial function t
+00001400: 6861 7420 6d6f 7374 2072 6574 6169 6e65  hat most retaine
+00001410: 6420 6765 6e65 2064 7570 6c69 6361 7465  d gene duplicate
+00001420: 7320 6172 6520 7265 6365 6e74 6c79 2062  s are recently b
+00001430: 6f72 6e65 6420 7769 7468 207e 3020 6167  orned with ~0 ag
+00001440: 6520 7768 696c 6520 6173 2074 6865 2061  e while as the a
+00001450: 6765 2067 6f69 6e67 206f 6c64 6572 2074  ge going older t
+00001460: 6865 2061 7373 6f63 6961 7465 6420 6e75  he associated nu
+00001470: 6d62 6572 206f 6620 7265 7461 696e 6564  mber of retained
+00001480: 2067 656e 6520 6475 706c 6963 6174 6573   gene duplicates
+00001490: 2064 6563 6179 2071 7561 7369 2d65 7870   decay quasi-exp
+000014a0: 6f6e 656e 7469 616c 6c79 2e20 5468 6572  onentially. Ther
+000014b0: 6566 6f72 652c 2074 6865 206f 6363 7572  efore, the occur
+000014c0: 616e 6365 206f 6620 6c61 7267 652d 7363  ance of large-sc
+000014d0: 616c 6520 6765 6e65 2064 7570 6c69 6361  ale gene duplica
+000014e0: 7469 6f6e 2065 7665 6e74 732c 2066 6f72  tion events, for
+000014f0: 2069 6e73 7461 6e65 2057 4744 732c 2077   instane WGDs, w
+00001500: 6974 6820 7661 7269 6564 2072 6574 656e  ith varied reten
+00001510: 7469 6f6e 2072 6174 652c 2077 696c 6c20  tion rate, will 
+00001520: 6c65 6176 6520 616e 2061 6765 2070 6561  leave an age pea
+00001530: 6b20 6672 6f6d 2074 6865 2062 7572 7374  k from the burst
+00001540: 206f 6620 6765 6e65 2064 7570 6c69 6361   of gene duplica
+00001550: 7465 7320 696e 2061 2073 686f 7274 2074  tes in a short t
+00001560: 696d 652d 6672 616d 6520 7570 6f6e 2074  ime-frame upon t
+00001570: 6865 2069 6e69 7469 616c 2061 6765 2064  he initial age d
+00001580: 6973 7472 6962 7574 696f 6e2c 2061 6e64  istribution, and
+00001590: 2063 616e 2062 6520 756e 7665 696c 6564   can be unveiled
+000015a0: 2066 726f 6d20 6d69 7874 7572 6520 6d6f   from mixture mo
+000015b0: 6465 6c69 6e67 2061 6e61 6c79 7369 732e  deling analysis.
+000015c0: 2048 6f77 6576 6572 2c20 5747 4473 2069   However, WGDs i
+000015d0: 6465 6e74 6966 6965 6420 6672 6f6d 2074  dentified from t
+000015e0: 6865 2070 6172 616c 6f67 6f75 7320 2a4b  he paralogous *K
+000015f0: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+00001600: 6520 6469 7374 7269 6275 7469 6f6e 7320  e distributions 
+00001610: 6361 6e20 6f6e 6c79 2069 6e66 6f72 6d20  can only inform 
+00001620: 7468 6520 5747 4420 7469 6d69 6e67 2069  the WGD timing i
+00001630: 6e20 7468 6520 7469 6d65 2d73 6361 6c65  n the time-scale
+00001640: 206f 6620 7468 6174 2073 7065 6369 6669   of that specifi
+00001650: 6320 7370 6563 6965 732c 2077 6869 6368  c species, which
+00001660: 2069 7320 6e6f 7420 636f 6d70 6172 6162   is not comparab
+00001670: 6c65 2069 6e20 7468 6520 7068 796c 6f67  le in the phylog
+00001680: 656e 6574 6963 2063 6f6e 7465 7874 2e20  enetic context. 
+00001690: 4f6e 6c79 2077 6974 6820 7468 6520 6f72  Only with the or
+000016a0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+000016b0: 623e 533c 2f73 7562 3e20 6167 6520 6469  b>S</sub> age di
+000016c0: 7374 7269 6275 7469 6f6e 732c 2077 6869  stributions, whi
+000016d0: 6368 2063 6f6e 7665 7274 2074 6865 2065  ch convert the e
+000016e0: 7374 696d 6174 6564 2062 6f64 7920 6672  stimated body fr
+000016f0: 6f6d 2070 6172 616c 6f67 7565 7320 746f  om paralogues to
+00001700: 206f 7274 686f 6c6f 6775 6573 2061 6e64   orthologues and
+00001710: 2069 6e66 6f72 6d20 7468 6520 7265 6c61   inform the rela
+00001720: 7469 7665 2074 696d 696e 6720 6f66 2073  tive timing of s
+00001730: 7065 6369 6174 696f 6e20 6576 656e 7473  peciation events
+00001740: 2c20 6361 6e20 7765 2064 6563 6970 6865  , can we deciphe
+00001750: 7220 7468 6520 7068 796c 6f67 656e 6574  r the phylogenet
+00001760: 6963 2070 6c61 6365 6d65 6e74 206f 6620  ic placement of 
+00001770: 5747 4473 2061 6674 6572 2070 726f 7065  WGDs after prope
+00001780: 7220 7261 7465 2063 6f72 7265 6374 696f  r rate correctio
+00001790: 6e2e 2060 7767 6420 7632 6020 6973 2073  n. `wgd v2` is s
+000017a0: 7563 6820 7072 6f67 7261 6d20 7468 6174  uch program that
+000017b0: 2068 656c 7073 2075 7365 7273 2063 6f6e   helps users con
+000017c0: 7374 7275 6374 2070 6172 616c 6f67 6f75  struct paralogou
+000017d0: 7320 616e 6420 6f72 7468 6f6c 6f67 6f75  s and orthologou
+000017e0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+000017f0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00001800: 6f6e 7320 616e 6420 7265 616c 697a 6520  ons and realize 
+00001810: 626f 7468 2074 6865 2069 6465 6e74 6966  both the identif
+00001820: 6963 6174 696f 6e20 616e 6420 706c 6163  ication and plac
+00001830: 656d 656e 7420 6f66 2057 4744 732e 0a0a  ement of WGDs...
+00001840: 496e 2074 6865 2070 7265 6d69 7365 206f  In the premise o
+00001850: 6620 7068 796c 6f67 656e 6574 6963 616c  f phylogenetical
+00001860: 6c79 206c 6f63 6174 6564 2057 4744 732c  ly located WGDs,
+00001870: 2074 6865 2061 6273 6f6c 7574 6520 6167   the absolute ag
+00001880: 6520 2869 6e20 6765 6f6c 6f67 6963 616c  e (in geological
+00001890: 2074 696d 6529 206f 6620 5747 4473 2063   time) of WGDs c
+000018a0: 616e 2061 6c73 6f20 6265 2069 6e66 6572  an also be infer
+000018b0: 7265 6420 6672 6f6d 2074 686f 7365 2057  red from those W
+000018c0: 4744 2d72 6574 6169 6e65 6420 6765 6e65  GD-retained gene
+000018d0: 2064 7570 6c69 6361 7465 732c 2061 6c74   duplicates, alt
+000018e0: 686f 7567 6820 7468 6572 6520 6861 7320  hough there has 
+000018f0: 6265 656e 206e 6f20 6561 7379 206f 7220  been no easy or 
+00001900: 7374 7261 6967 6874 666f 7277 6172 6420  straightforward 
+00001910: 7069 7065 6c69 6e65 2066 6f72 2074 6869  pipeline for thi
+00001920: 7320 6a6f 6220 736f 2066 6172 2e20 496e  s job so far. In
+00001930: 2060 7767 6420 7632 602c 2077 6520 6465   `wgd v2`, we de
+00001940: 7665 6c6f 7065 6420 6120 6665 6173 6962  veloped a feasib
+00001950: 6c65 2069 6e74 6567 7261 7465 6420 7069  le integrated pi
+00001960: 706c 656c 696e 6520 666f 7220 6162 736f  pleline for abso
+00001970: 6c75 7465 2064 6174 696e 6720 6f66 2057  lute dating of W
+00001980: 4744 732e 2054 6865 2070 6970 656c 696e  GDs. The pipelin
+00001990: 6520 6361 6e20 6265 2072 6f75 6768 6c79  e can be roughly
+000019a0: 2064 6976 6964 6564 2069 6e74 6f20 7468   divided into th
+000019b0: 7265 6520 6d61 696e 2073 7465 7073 2e20  ree main steps. 
+000019c0: 3129 2054 6865 2063 6f6e 7374 7275 6374  1) The construct
+000019d0: 696f 6e20 6f66 2061 6e63 686f 7220 2a4b  ion of anchor *K
+000019e0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+000019f0: 7374 7269 6275 7469 6f6e 2061 6e64 2074  stribution and t
+00001a00: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
+00001a10: 6620 6372 6564 6961 626c 6520 2a4b 2a3c  f crediable *K*<
+00001a20: 7375 623e 533c 2f73 7562 3e20 7261 6e67  sub>S</sub> rang
+00001a30: 6520 6164 6f70 7465 6420 666f 7220 7068  e adopted for ph
+00001a40: 796c 6f67 656e 6574 6963 2064 6174 696e  ylogenetic datin
+00001a50: 672c 2075 7369 6e67 2060 7767 6420 646d  g, using `wgd dm
+00001a60: 6460 2c20 6077 6764 206b 7364 602c 2060  d`, `wgd ksd`, `
+00001a70: 7767 6420 7379 6e60 2061 6e64 2060 7767  wgd syn` and `wg
+00001a80: 6420 7065 616b 602e 204e 6f74 6520 7468  d peak`. Note th
+00001a90: 6174 2068 6572 6520 7765 206f 6e6c 7920  at here we only 
+00001aa0: 636f 6e73 6964 6572 2067 656e 6f6d 6520  consider genome 
+00001ab0: 6173 7365 6d62 6c79 2062 6563 6175 7365  assembly because
+00001ac0: 2066 6f72 2074 7261 6e73 6372 6970 746f   for transcripto
+00001ad0: 6d65 2061 7373 656d 626c 7920 6974 2773  me assembly it's
+00001ae0: 2069 6d70 6f73 7369 626c 6520 746f 2064   impossible to d
+00001af0: 6973 7469 6e67 7569 7368 2057 4744 2d64  istinguish WGD-d
+00001b00: 6572 6976 6564 2064 7570 6c69 6361 7465  erived duplicate
+00001b10: 7320 6672 6f6d 2073 6d61 6c6c 2d73 6361  s from small-sca
+00001b20: 6c65 2064 7570 6c69 6361 7469 6f6e 2d64  le duplication-d
+00001b30: 6572 6976 6564 2064 7570 6c69 6361 7465  erived duplicate
+00001b40: 732c 2077 6869 6368 2068 6170 7065 6e65  s, which happene
+00001b50: 6420 696e 2061 2063 6f6e 7469 6e75 6f75  d in a continuou
+00001b60: 7320 7469 6d65 2d66 7261 6d65 2069 6e73  s time-frame ins
+00001b70: 7465 6164 206f 6620 6f6e 6c79 2061 2073  tead of only a s
+00001b80: 6570 6172 6174 6520 7368 6f72 7420 7469  eparate short ti
+00001b90: 6d65 2d66 7261 6d65 2061 6e64 2074 6875  me-frame and thu
+00001ba0: 7320 7265 666c 6563 7473 2074 6865 2064  s reflects the d
+00001bb0: 7572 6174 696f 6e20 6f66 2074 6861 7420  uration of that 
+00001bc0: 6272 616e 6368 2072 6174 6865 7220 7468  branch rather th
+00001bd0: 616e 2074 6865 2074 696d 6520 6174 2077  an the time at w
+00001be0: 6869 6368 2057 4744 206f 6363 7572 7265  hich WGD occurre
+00001bf0: 642e 2032 2920 5468 6520 666f 726d 756c  d. 2) The formul
+00001c00: 6174 696f 6e20 6f66 2061 2073 7461 7274  ation of a start
+00001c10: 696e 6720 7472 6565 2075 7365 6420 696e  ing tree used in
+00001c20: 2074 6865 2070 6879 6c6f 6765 6e65 7469   the phylogeneti
+00001c30: 6320 6461 7469 6e67 2c20 636f 6d70 6f73  c dating, compos
+00001c40: 6564 206f 6620 6120 6665 7720 7370 6563  ed of a few spec
+00001c50: 6965 7320 616e 6420 616e 6e6f 7461 7465  ies and annotate
+00001c60: 6420 7769 7468 2066 6f73 7369 6c20 6361  d with fossil ca
+00001c70: 6c69 6272 6174 696f 6e20 696e 666f 726d  libration inform
+00001c80: 6174 696f 6e2e 2054 6869 7320 7374 6570  ation. This step
+00001c90: 2069 7320 6573 7365 6e74 6961 6c20 666f   is essential fo
+00001ca0: 7220 7468 6520 7265 7375 6c74 206f 6620  r the result of 
+00001cb0: 6162 736f 6c75 7465 2057 4744 2064 6174  absolute WGD dat
+00001cc0: 696e 6720 7468 6174 2077 6520 7375 6767  ing that we sugg
+00001cd0: 6573 7420 7573 6572 7320 6f66 2074 616b  est users of tak
+00001ce0: 696e 6720 6772 6561 7420 6361 7574 696f  ing great cautio
+00001cf0: 6e20 746f 2061 7373 7572 6520 7468 6520  n to assure the 
+00001d00: 7472 6565 2074 6f70 6f6c 6f67 7920 616e  tree topology an
+00001d10: 6420 7072 6f70 6572 2062 6f75 6e64 7320  d proper bounds 
+00001d20: 666f 7220 666f 7373 696c 2063 616c 6962  for fossil calib
+00001d30: 7261 7469 6f6e 732e 2033 2920 5468 6520  rations. 3) The 
+00001d40: 636f 6e73 7472 7563 7469 6f6e 206f 6620  construction of 
+00001d50: 6f72 7468 6f67 726f 7570 7320 636f 6e73  orthogroups cons
+00001d60: 6973 7469 6e67 206f 6620 636f 6c6c 696e  isting of collin
+00001d70: 6561 7220 6475 706c 6963 6174 6573 206f  ear duplicates o
+00001d80: 6620 7468 6520 666f 6361 6c20 7370 6563  f the focal spec
+00001d90: 6965 7320 616e 6420 7468 6569 7220 7265  ies and their re
+00001da0: 6369 7072 6f63 616c 2062 6573 7420 6869  ciprocal best hi
+00001db0: 7473 2028 5242 4873 2920 6167 6169 6e73  ts (RBHs) agains
+00001dc0: 7420 6f74 6865 7220 7370 6563 6965 7320  t other species 
+00001dd0: 696e 2074 6865 2073 7461 7274 696e 6720  in the starting 
+00001de0: 7472 6565 2061 6e64 2074 6865 2070 6879  tree and the phy
+00001df0: 6c6f 6765 6e65 7469 6320 6461 7469 6e67  logenetic dating
+00001e00: 2075 7369 6e67 2061 206d 6f6c 6563 756c   using a molecul
+00001e10: 6172 2064 6174 696e 6720 7072 6f67 7261  ar dating progra
+00001e20: 6d20 666f 7220 696e 7374 616e 6365 2060  m for instance `
+00001e30: 6d63 6d63 7472 6565 602c 2076 6961 2070  mcmctree`, via p
+00001e40: 726f 6772 616d 2060 7767 6420 646d 6460  rogram `wgd dmd`
+00001e50: 2061 6e64 2060 7767 6420 666f 6375 7360   and `wgd focus`
+00001e60: 2e20 5765 2072 6563 6f6d 6d65 6e64 2074  . We recommend t
+00001e70: 6865 2075 7361 6765 206f 6620 4261 7965  he usage of Baye
+00001e80: 7369 616e 206d 6f6c 6563 756c 6172 2064  sian molecular d
+00001e90: 6174 696e 6720 7072 6f67 7261 6d20 606d  ating program `m
+00001ea0: 636d 6374 7265 6560 2c20 7768 6963 6820  cmctree`, which 
+00001eb0: 7072 6f76 6964 6573 2061 2076 6172 6965  provides a varie
+00001ec0: 7479 206f 6620 6469 6666 6572 656e 7420  ty of different 
+00001ed0: 7375 6273 7469 7475 7469 6f6e 2061 6e64  substitution and
+00001ee0: 2072 6174 6520 6d6f 6465 6c73 2e20 4e6f   rate models. No
+00001ef0: 6e65 7468 656c 6573 732c 2077 6520 7572  netheless, we ur
+00001f00: 6765 2075 7365 7273 2074 6f20 7365 7420  ge users to set 
+00001f10: 7072 696f 7220 6469 7374 7269 6275 7469  prior distributi
+00001f20: 6f6e 206f 6620 6469 6666 6572 656e 7420  on of different 
+00001f30: 7061 7261 6d65 7465 7273 2077 6974 6820  parameters with 
+00001f40: 6361 7574 696f 6e20 616e 6420 6173 7375  caution and assu
+00001f50: 7265 2061 6465 7175 6174 6520 7361 6d70  re adequate samp
+00001f60: 6c69 6e67 206f 6620 6469 6666 6572 656e  ling of differen
+00001f70: 7420 7061 7261 6d65 7465 7273 2e0a 0a23  t parameters...#
+00001f80: 2320 496e 7374 616c 6c61 7469 6f6e 0a0a  # Installation..
+00001f90: 5468 6520 6561 7369 6573 7420 7761 7920  The easiest way 
+00001fa0: 746f 2069 6e73 7461 6c6c 2060 7767 6420  to install `wgd 
+00001fb0: 7632 6020 6973 2075 7369 6e67 2060 5059  v2` is using `PY
+00001fc0: 5049 602e 204e 6f74 6520 7468 6174 2069  PI`. Note that i
+00001fd0: 6620 796f 7520 7761 6e74 2074 6f20 6765  f you want to ge
+00001fe0: 7420 7468 6520 6c61 7465 7374 2075 7064  t the latest upd
+00001ff0: 6174 652c 2077 6520 7375 6767 6573 7420  ate, we suggest 
+00002000: 696e 7374 616c 6c69 6e67 2066 726f 6d20  installing from 
+00002010: 7468 6520 736f 7572 6365 2c20 7369 6e63  the source, sinc
+00002020: 6520 7468 6520 7570 6461 7465 206f 6e20  e the update on 
+00002030: 6050 5950 4960 2077 696c 6c20 6265 2064  `PYPI` will be d
+00002040: 656c 6179 6564 2063 6f6d 7061 7265 6420  elayed compared 
+00002050: 746f 2068 6572 6520 6f66 2073 6f75 7263  to here of sourc
+00002060: 652e 0a0a 6060 600a 7069 7020 696e 7374  e...```.pip inst
+00002070: 616c 6c20 7767 640a 6060 600a 0a54 6f20  all wgd.```..To 
+00002080: 696e 7374 616c 6c20 6077 6764 2076 3260  install `wgd v2`
+00002090: 2069 6e20 6120 7669 7274 7561 6c20 656e   in a virtual en
+000020a0: 7669 726f 6e6d 656e 7420 2877 6869 6368  vironment (which
+000020b0: 2077 6520 7374 726f 6e67 6c79 2072 6563   we strongly rec
+000020c0: 6f6d 6d65 6e64 2c20 6e6f 206d 6174 7465  ommend, no matte
+000020d0: 7220 796f 7520 696e 7374 616c 6c20 6672  r you install fr
+000020e0: 6f6d 2073 6f75 7263 652c 2060 5059 5049  om source, `PYPI
+000020f0: 6020 6f72 2060 6269 6f63 6f6e 6461 6029  ` or `bioconda`)
+00002100: 2c20 7468 6520 666f 6c6c 6f77 696e 6720  , the following 
+00002110: 636f 6d6d 616e 6420 6c69 6e65 7320 636f  command lines co
+00002120: 756c 6420 6265 2075 7365 642e 0a0a 6060  uld be used...``
+00002130: 600a 6769 7420 636c 6f6e 6520 6874 7470  `.git clone http
+00002140: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f68  s://github.com/h
+00002150: 6563 6865 2d70 7362 2f77 6764 0a63 6420  eche-psb/wgd.cd 
+00002160: 7767 640a 7669 7274 7561 6c65 6e76 202d  wgd.virtualenv -
+00002170: 703d 7079 7468 6f6e 3320 454e 5620 286f  p=python3 ENV (o
+00002180: 7220 7079 7468 6f6e 3320 2d6d 2076 656e  r python3 -m ven
+00002190: 7620 454e 5629 0a73 6f75 7263 6520 454e  v ENV).source EN
+000021a0: 562f 6269 6e2f 6163 7469 7661 7465 0a70  V/bin/activate.p
+000021b0: 6970 2069 6e73 7461 6c6c 202d 7220 7265  ip install -r re
+000021c0: 7175 6972 656d 656e 7473 2e74 7874 0a70  quirements.txt.p
+000021d0: 6970 2069 6e73 7461 6c6c 202e 0a60 6060  ip install ..```
+000021e0: 0a0a 5768 656e 206d 6574 2077 6974 6820  ..When met with 
+000021f0: 7065 726d 6973 7369 6f6e 2070 726f 626c  permission probl
+00002200: 656d 2069 6e20 696e 7374 616c 6c61 7469  em in installati
+00002210: 6f6e 2c20 706c 6561 7365 2074 7279 2074  on, please try t
+00002220: 6865 2066 6f6c 6c6f 7769 6e67 2063 6f6d  he following com
+00002230: 6d61 6e64 206c 696e 652e 0a0a 6060 600a  mand line...```.
+00002240: 7069 7020 696e 7374 616c 6c20 2d65 202e  pip install -e .
+00002250: 0a60 6060 0a0a 4966 206d 756c 7469 706c  .```..If multipl
+00002260: 7920 7665 7273 696f 6e73 206f 6620 6077  y versions of `w
+00002270: 6764 6020 7765 7265 2069 6e73 7461 6c6c  gd` were install
+00002280: 6564 2069 6e20 7468 6520 7379 7374 656d  ed in the system
+00002290: 2c20 706c 6561 7365 2061 6464 2074 6865  , please add the
+000022a0: 2072 6967 6874 2070 6174 6820 6f66 2069   right path of i
+000022b0: 6e74 6572 6573 7465 6420 7665 7273 696f  nterested versio
+000022c0: 6e20 696e 746f 2074 6865 2065 6e76 6972  n into the envir
+000022d0: 6f6e 6d65 6e74 2076 6172 6961 626c 6573  onment variables
+000022e0: 2c20 666f 7220 6578 616d 706c 650a 0a60  , for example..`
+000022f0: 6060 0a65 7870 6f72 7420 5041 5448 3d22  ``.export PATH="
+00002300: 2450 4154 483a 7e2f 2e6c 6f63 616c 2f62  $PATH:~/.local/b
+00002310: 696e 2f77 6764 220a 6060 600a 0a4e 6f74  in/wgd".```..Not
+00002320: 6520 7468 6174 2074 6865 2076 6572 7369  e that the versi
+00002330: 6f6e 206f 6620 606e 756d 7079 6020 6973  on of `numpy` is
+00002340: 2069 6d70 6f72 7461 6e74 2028 666f 7220   important (for 
+00002350: 6d61 6e79 206f 7468 6572 2070 6163 6b61  many other packa
+00002360: 6765 7320 6172 6520 7468 6520 7361 6d65  ges are the same
+00002370: 206f 6620 636f 7572 7365 292c 2065 7370   of course), esp
+00002380: 6563 6961 6c6c 7920 666f 7220 6066 6173  ecially for `fas
+00002390: 7463 6c75 7374 6572 6020 7061 636b 6167  tcluster` packag
+000023a0: 652e 2049 6e20 6f75 7220 7465 7374 2c20  e. In our test, 
+000023b0: 7468 6520 606e 756d 7079 6020 312e 3139  the `numpy` 1.19
+000023c0: 2e30 2077 6f72 6b73 2066 696e 6520 6f6e  .0 works fine on
+000023d0: 2060 7079 7468 6f6e 332e 362f 3860 2e20   `python3.6/8`. 
+000023e0: 4966 2079 6f75 206d 6574 2073 6f6d 6520  If you met some 
+000023f0: 6572 726f 7273 206f 7220 7761 726e 696e  errors or warnin
+00002400: 6773 2061 626f 7574 2060 6e75 6d70 7960  gs about `numpy`
+00002410: 2c20 6d61 7962 6520 636f 6e73 6964 6572  , maybe consider
+00002420: 696e 6720 7072 652d 696e 7374 616c 6c20  ing pre-install 
+00002430: 606e 756d 7079 6020 6173 2031 2e31 392e  `numpy` as 1.19.
+00002440: 3020 6f72 206f 7468 6572 2063 6c6f 7365  0 or other close
+00002450: 2d62 7920 7665 7273 696f 6e73 2062 6566  -by versions bef
+00002460: 6f72 6520 796f 7520 696e 7374 616c 6c20  ore you install 
+00002470: 6077 6764 602e 2060 7767 6460 2072 656c  `wgd`. `wgd` rel
+00002480: 6965 7320 6f6e 2065 7874 6572 6e61 6c20  ies on external 
+00002490: 736f 6674 7761 7265 7320 696e 636c 7564  softwares includ
+000024a0: 696e 6720 6064 6961 6d6f 6e64 6020 616e  ing `diamond` an
+000024b0: 6420 606d 636c 6020 666f 7220 6077 6764  d `mcl` for `wgd
+000024c0: 2064 6d64 602c 2060 7061 6d6c 2076 342e   dmd`, `paml v4.
+000024d0: 396a 602c 2060 6d61 6666 7460 2028 606d  9j`, `mafft` (`m
+000024e0: 7573 636c 6560 2061 6e64 2060 7072 616e  uscle` and `pran
+000024f0: 6b60 2069 6620 7365 7429 2c20 6066 6173  k` if set), `fas
+00002500: 7474 7265 6560 2028 6f72 2060 6971 7472  ttree` (or `iqtr
+00002510: 6565 6020 6966 2073 6574 2920 666f 7220  ee` if set) for 
+00002520: 6077 6764 206b 7364 6020 616e 6420 6f70  `wgd ksd` and op
+00002530: 7469 6f6e 616c 6c79 2060 6d72 6261 7965  tionally `mrbaye
+00002540: 7360 2066 6f72 2074 6865 2070 6879 6c6f  s` for the phylo
+00002550: 6765 6e65 7469 6320 696e 6665 7265 6e63  genetic inferenc
+00002560: 6520 6675 6e63 7469 6f6e 2069 6e20 6077  e function in `w
+00002570: 6764 2064 6d64 6020 616e 6420 6077 6764  gd dmd` and `wgd
+00002580: 2066 6f63 7573 6020 2860 6d61 6666 7460   focus` (`mafft`
+00002590: 2c20 606d 7573 636c 6560 2061 6e64 2060  , `muscle` and `
+000025a0: 7072 616e 6b60 2061 7320 7765 6c6c 2077  prank` as well w
+000025b0: 6865 6e20 7468 6520 616e 616c 7973 6973  hen the analysis
+000025c0: 2072 6571 7569 7265 7320 7365 7175 656e   requires sequen
+000025d0: 6365 2061 6c69 676e 6d65 6e74 292e 2053  ce alignment). S
+000025e0: 6f6d 6520 6f74 6865 7220 6f70 7469 6f6e  ome other option
+000025f0: 616c 2073 6f66 7477 6172 6573 2069 6e63  al softwares inc
+00002600: 6c75 6469 6e67 2060 7061 6d6c 2076 342e  luding `paml v4.
+00002610: 396a 602c 2060 7238 7360 2c20 6062 6561  9j`, `r8s`, `bea
+00002620: 7374 602c 2060 6567 676e 6f67 602c 2060  st`, `eggnog`, `
+00002630: 6469 616d 6f6e 6460 2c20 6069 6e74 6572  diamond`, `inter
+00002640: 7072 6f73 6361 6e60 2c20 6068 6d6d 6572  proscan`, `hmmer
+00002650: 2076 332e 3162 3260 2061 6e64 2060 6173   v3.1b2` and `as
+00002660: 7472 616c 2d70 726f 6020 2860 686d 6d65  tral-pro` (`hmme
+00002670: 7220 7633 2e31 6232 6020 6973 2061 6c73  r v3.1b2` is als
+00002680: 6f20 666f 7220 7468 6520 5b6f 7274 686f  o for the [ortho
+00002690: 6772 6f75 7020 6173 7369 676e 6d65 6e74  group assignment
+000026a0: 5d28 6874 7470 733a 2f2f 7767 6476 322e  ](https://wgdv2.
+000026b0: 7265 6164 7468 6564 6f63 732e 696f 2f65  readthedocs.io/e
+000026c0: 6e2f 6c61 7465 7374 2f75 7361 6765 2e68  n/latest/usage.h
+000026d0: 746d 6c23 6f72 7468 6f67 726f 7570 2d61  tml#orthogroup-a
+000026e0: 7373 6967 6e6d 656e 7429 2066 756e 6374  ssignment) funct
+000026f0: 696f 6e20 696e 2060 7767 6420 646d 6460  ion in `wgd dmd`
+00002700: 2061 6e64 2060 6173 7472 616c 2d70 726f   and `astral-pro
+00002710: 6020 6973 2061 6c73 6f20 666f 7220 7468  ` is also for th
+00002720: 6520 6675 6e63 7469 6f6e 2060 2d2d 636f  e function `--co
+00002730: 6c6c 696e 6561 7263 6f61 6c65 7363 656e  llinearcoalescen
+00002740: 6365 6020 696e 2060 7767 6420 646d 6460  ce` in `wgd dmd`
+00002750: 2920 6172 6520 666f 7220 7468 6520 6d6f  ) are for the mo
+00002760: 6c65 6375 6c61 7220 6461 7469 6e67 2c20  lecular dating, 
+00002770: 6765 6e65 2066 616d 696c 7920 6675 6e63  gene family func
+00002780: 7469 6f6e 2061 6e6e 6f74 6174 696f 6e20  tion annotation 
+00002790: 6f72 2070 6879 6c6f 6765 6e65 7469 6320  or phylogenetic 
+000027a0: 696e 6665 7265 6e63 6520 696e 2060 7767  inference in `wg
+000027b0: 6420 666f 6375 7360 2e0a 0a23 2320 5069  d focus`...## Pi
+000027c0: 7065 6c69 6e65 730a 0a54 6f20 7175 6963  pelines..To quic
+000027d0: 6b6c 7920 6765 7420 6661 6d69 6c69 6172  kly get familiar
+000027e0: 2077 6974 6820 6077 6764 2076 3260 2c20   with `wgd v2`, 
+000027f0: 7765 2070 726f 7669 6465 6420 736f 6d65  we provided some
+00002800: 2063 6f6d 6d6f 6e20 7069 7065 6c69 6e65   common pipeline
+00002810: 7320 666f 7220 7573 6572 7320 7769 7468  s for users with
+00002820: 2061 2066 7265 7368 2067 656e 6f6d 6520   a fresh genome 
+00002830: 6173 7365 6d62 6c79 2069 6e20 6861 6e64  assembly in hand
+00002840: 2e0a 0a23 2323 2050 6970 656c 696e 6520  ...### Pipeline 
+00002850: 310a 2a2a 5369 6d70 6c65 2063 6f6e 7374  1.**Simple const
+00002860: 7275 6374 696f 6e20 6f66 2061 6765 2064  ruction of age d
+00002870: 6973 7472 6962 7574 696f 6e2a 2a0a 6060  istribution**.``
+00002880: 600a 7767 6420 646d 6420 4171 7569 6c65  `.wgd dmd Aquile
+00002890: 6769 615f 636f 6572 756c 6561 202d 6f20  gia_coerulea -o 
+000028a0: 7767 645f 646d 640a 7767 6420 6b73 6420  wgd_dmd.wgd ksd 
+000028b0: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
+000028c0: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
+000028d0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+000028e0: 6120 2d6f 2077 6764 5f6b 7364 0a60 6060  a -o wgd_ksd.```
+000028f0: 0a60 6060 0a23 7768 6174 2077 696c 6c20  .```.#what will 
+00002900: 6265 2069 6e20 7468 6520 7265 7375 6c74  be in the result
+00002910: 2064 6972 6563 746f 7279 0a2d 7767 645f   directory.-wgd_
+00002920: 646d 640a 2d2d 4171 7569 6c65 6769 615f  dmd.--Aquilegia_
+00002930: 636f 6572 756c 6561 2e74 7376 0a2d 7767  coerulea.tsv.-wg
+00002940: 645f 6b73 640a 2d2d 4171 7569 6c65 6769  d_ksd.--Aquilegi
+00002950: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00002960: 732e 7473 7620 4171 7569 6c65 6769 615f  s.tsv Aquilegia_
+00002970: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00002980: 7376 672f 7064 660a 6060 600a 5468 6520  svg/pdf.```.The 
+00002990: 7265 7375 6c74 696e 6720 6041 7175 696c  resulting `Aquil
+000029a0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+000029b0: 7660 2069 7320 7468 6520 7768 6f6c 6520  v` is the whole 
+000029c0: 7061 7261 6e6f 6d65 2066 616d 696c 7920  paranome family 
+000029d0: 6669 6c65 2c20 6041 7175 696c 6567 6961  file, `Aquilegia
+000029e0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+000029f0: 2e74 7376 6020 6669 6c65 2069 7320 7468  .tsv` file is th
+00002a00: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00002a10: 3e20 6469 7374 7269 6275 6974 6f6e 2066  > distribuiton f
+00002a20: 696c 652c 2060 4171 7569 6c65 6769 615f  ile, `Aquilegia_
+00002a30: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00002a40: 7376 6760 2061 6e64 2060 4171 7569 6c65  svg` and `Aquile
+00002a50: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+00002a60: 2e6b 732e 7064 6660 2061 7265 2074 6865  .ks.pdf` are the
+00002a70: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00002a80: 2070 6c6f 7473 2e0a 0a23 2323 2050 6970   plots...### Pip
+00002a90: 656c 696e 6520 320a 2a2a 5369 6d70 6c65  eline 2.**Simple
+00002aa0: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
+00002ab0: 2061 6765 2064 6973 7472 6962 7574 696f   age distributio
+00002ac0: 6e20 7769 7468 2063 6f6c 6c69 6e65 6172  n with collinear
+00002ad0: 6974 792a 2a0a 6060 600a 7767 6420 646d  ity**.```.wgd dm
+00002ae0: 6420 4171 7569 6c65 6769 615f 636f 6572  d Aquilegia_coer
+00002af0: 756c 6561 202d 6f20 7767 645f 646d 640a  ulea -o wgd_dmd.
+00002b00: 7767 6420 6b73 6420 7767 645f 646d 642f  wgd ksd wgd_dmd/
+00002b10: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00002b20: 6561 2e74 7376 2041 7175 696c 6567 6961  ea.tsv Aquilegia
+00002b30: 5f63 6f65 7275 6c65 6120 2d6f 2077 6764  _coerulea -o wgd
+00002b40: 5f6b 7364 0a77 6764 2073 796e 202d 6620  _ksd.wgd syn -f 
+00002b50: 6d52 4e41 202d 6120 4e61 6d65 2077 6764  mRNA -a Name wgd
+00002b60: 5f64 6d64 2f41 7175 696c 6567 6961 5f63  _dmd/Aquilegia_c
+00002b70: 6f65 7275 6c65 612e 7473 7620 4171 7569  oerulea.tsv Aqui
+00002b80: 6c65 6769 615f 636f 6572 756c 6561 2e67  legia_coerulea.g
+00002b90: 6666 3320 2d6b 7320 7767 645f 6b73 642f  ff3 -ks wgd_ksd/
+00002ba0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00002bb0: 6561 2e74 7376 2e6b 732e 7473 7620 2d6f  ea.tsv.ks.tsv -o
+00002bc0: 2077 6764 5f73 796e 0a60 6060 0a60 6060   wgd_syn.```.```
+00002bd0: 0a23 2077 6861 7420 7769 6c6c 2062 6520  .# what will be 
+00002be0: 696e 2074 6865 2077 6764 5f73 796e 2072  in the wgd_syn r
+00002bf0: 6573 756c 7420 6469 7265 6374 6f72 790a  esult directory.
+00002c00: 2d77 6764 5f73 796e 0a2d 2d61 6e63 686f  -wgd_syn.--ancho
+00002c10: 7273 2e63 7376 0a2d 2d66 616d 696c 6965  rs.csv.--familie
+00002c20: 732e 7473 760a 2d2d 6765 6e65 2d74 6162  s.tsv.--gene-tab
+00002c30: 6c65 2e63 7376 0a2d 2d69 6164 686f 7265  le.csv.--iadhore
+00002c40: 2e63 6f6e 660a 2d2d 7363 6166 666f 6c64  .conf.--scaffold
+00002c50: 5f6c 656e 6774 682e 7473 760a 2d2d 7365  _length.tsv.--se
+00002c60: 676d 656e 7473 5f63 6f6f 7264 696e 6174  gments_coordinat
+00002c70: 6573 2e74 7376 0a2d 2d53 6567 7072 6f66  es.tsv.--Segprof
+00002c80: 696c 652e 6373 760a 2d2d 5379 6e64 6570  ile.csv.--Syndep
+00002c90: 7468 2e70 6466 2f73 7667 0a2d 2d41 7175  th.pdf/svg.--Aqu
+00002ca0: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+00002cb0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00002cc0: 6561 5f6d 756c 7469 706c 6963 6f6e 735f  ea_multiplicons_
+00002cd0: 6c65 7665 6c2e 7064 662f 706e 672f 7376  level.pdf/png/sv
+00002ce0: 670a 2d2d 4171 7569 6c65 6769 615f 636f  g.--Aquilegia_co
+00002cf0: 6572 756c 6561 5f67 656e 655f 6f72 6465  erulea_gene_orde
+00002d00: 725f 7065 7263 6872 6f6d 2e74 7376 0a2d  r_perchrom.tsv.-
+00002d10: 2d41 7175 696c 6567 6961 5f63 6f65 7275  -Aquilegia_coeru
+00002d20: 6c65 612e 7473 762e 616e 6368 6f72 732e  lea.tsv.anchors.
+00002d30: 6b73 2e74 7376 0a2d 2d41 7175 696c 6567  ks.tsv.--Aquileg
+00002d40: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00002d50: 6b73 642e 7064 662f 7376 670a 2d2d 4171  ksd.pdf/svg.--Aq
+00002d60: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00002d70: 2d76 732d 4171 7569 6c65 6769 615f 636f  -vs-Aquilegia_co
+00002d80: 6572 756c 6561 2e64 6f74 2e70 6466 2f70  erulea.dot.pdf/p
+00002d90: 6e67 2f73 7667 0a2d 2d41 7175 696c 6567  ng/svg.--Aquileg
+00002da0: 6961 5f63 6f65 7275 6c65 612d 7673 2d41  ia_coerulea-vs-A
+00002db0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00002dc0: 612e 646f 745f 756e 6974 5f67 656e 652e  a.dot_unit_gene.
+00002dd0: 7064 662f 706e 672f 7376 670a 2d2d 4171  pdf/png/svg.--Aq
+00002de0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00002df0: 2d76 732d 4171 7569 6c65 6769 615f 636f  -vs-Aquilegia_co
+00002e00: 6572 756c 6561 5f4b 732e 646f 742e 7064  erulea_Ks.dot.pd
+00002e10: 662f 706e 672f 7376 670a 2d2d 4171 7569  f/png/svg.--Aqui
+00002e20: 6c65 6769 615f 636f 6572 756c 6561 2d76  legia_coerulea-v
+00002e30: 732d 4171 7569 6c65 6769 615f 636f 6572  s-Aquilegia_coer
+00002e40: 756c 6561 5f4b 732e 646f 745f 756e 6974  ulea_Ks.dot_unit
+00002e50: 5f67 656e 652e 7064 662f 706e 672f 7376  _gene.pdf/png/sv
+00002e60: 670a 2d2d 6961 6468 6f72 652d 6f75 740a  g.--iadhore-out.
+00002e70: 2d2d 2d61 6c69 676e 6d65 6e74 2e74 7874  ---alignment.txt
+00002e80: 0a2d 2d2d 616e 6368 6f72 706f 696e 7473  .---anchorpoints
+00002e90: 2e74 7874 0a2d 2d2d 6261 7365 636c 7573  .txt.---baseclus
+00002ea0: 7465 7273 2e74 7874 0a2d 2d2d 6765 6e65  ters.txt.---gene
+00002eb0: 732e 7478 740a 2d2d 2d6c 6973 745f 656c  s.txt.---list_el
+00002ec0: 656d 656e 7473 2e74 7874 0a2d 2d2d 6d75  ements.txt.---mu
+00002ed0: 6c74 6970 6c69 636f 6e5f 7061 6972 732e  ltiplicon_pairs.
+00002ee0: 7478 740a 2d2d 2d6d 756c 7469 706c 6963  txt.---multiplic
+00002ef0: 6f6e 732e 7478 740a 2d2d 2d73 6567 6d65  ons.txt.---segme
+00002f00: 6e74 732e 7478 740a 6060 600a 5468 6520  nts.txt.```.The 
+00002f10: 6061 6e63 686f 7273 2e63 7376 602c 2060  `anchors.csv`, `
+00002f20: 6661 6d69 6c69 6573 2e74 7376 602c 2060  families.tsv`, `
+00002f30: 6765 6e65 2d74 6162 6c65 2e63 7376 602c  gene-table.csv`,
+00002f40: 2060 7363 6166 666f 6c64 5f6c 656e 6774   `scaffold_lengt
+00002f50: 682e 7473 7660 2c20 6073 6567 6d65 6e74  h.tsv`, `segment
+00002f60: 735f 636f 6f72 6469 6e61 7465 732e 7473  s_coordinates.ts
+00002f70: 7660 2c20 6053 6567 7072 6f66 696c 652e  v`, `Segprofile.
+00002f80: 6373 7660 2c20 6041 7175 696c 6567 6961  csv`, `Aquilegia
+00002f90: 5f63 6f65 7275 6c65 615f 6765 6e65 5f6f  _coerulea_gene_o
+00002fa0: 7264 6572 5f70 6572 6368 726f 6d2e 7473  rder_perchrom.ts
+00002fb0: 7660 2061 7265 2074 6865 2062 6173 6963  v` are the basic
+00002fc0: 2066 696c 6573 2073 756d 6d61 7269 7a69   files summarizi
+00002fd0: 6e67 2074 6865 2067 656e 6520 6f72 6465  ng the gene orde
+00002fe0: 722c 2066 616d 696c 792c 2061 6e63 686f  r, family, ancho
+00002ff0: 7273 2061 6e64 2063 6f6c 6c69 6e65 6172  rs and collinear
+00003000: 2073 6567 6d65 6e74 732e 0a54 6865 2060   segments..The `
+00003010: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00003020: 6561 2e74 7376 2e61 6e63 686f 7273 2e6b  ea.tsv.anchors.k
+00003030: 732e 7473 7660 2069 7320 7468 6520 2a4b  s.tsv` is the *K
+00003040: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00003050: 7374 7269 6275 7469 6f6e 2066 696c 6520  stribution file 
+00003060: 6f66 2061 6e63 686f 7220 7061 6972 732e  of anchor pairs.
+00003070: 0a54 6865 2060 5379 6e64 6570 7468 2e70  .The `Syndepth.p
+00003080: 6466 2f73 7667 6020 7368 6f77 7320 7468  df/svg` shows th
+00003090: 6520 636f 6c6c 696e 6561 7220 7261 7469  e collinear rati
+000030a0: 6f20 6f66 206d 756c 7469 706c 6963 6f6e  o of multiplicon
+000030b0: 732e 0a54 6865 2060 4171 7569 6c65 6769  s..The `Aquilegi
+000030c0: 615f 636f 6572 756c 6561 5f41 7175 696c  a_coerulea_Aquil
+000030d0: 6567 6961 5f63 6f65 7275 6c65 615f 6d75  egia_coerulea_mu
+000030e0: 6c74 6970 6c69 636f 6e73 5f6c 6576 656c  ltiplicons_level
+000030f0: 2e70 6466 2f70 6e67 2f73 7667 6020 6973  .pdf/png/svg` is
+00003100: 2074 6865 2022 6475 7053 7461 636b 2220   the "dupStack" 
+00003110: 706c 6f74 2073 686f 7769 6e67 206d 756c  plot showing mul
+00003120: 7469 706c 6963 6f6e 7320 7769 7468 2064  tiplicons with d
+00003130: 6966 6665 7265 6e74 206c 6576 656c 7320  ifferent levels 
+00003140: 2864 6566 696e 6564 2061 7320 7468 6520  (defined as the 
+00003150: 6e75 6d62 6572 206f 6620 7365 676d 656e  number of segmen
+00003160: 7473 2077 6974 6869 6e29 2e0a 5468 6520  ts within)..The 
+00003170: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
+00003180: 6c65 612e 7473 762e 6b73 642e 7064 662f  lea.tsv.ksd.pdf/
+00003190: 7376 6760 2069 7320 7468 6520 2a4b 2a3c  svg` is the *K*<
+000031a0: 7375 623e 533c 2f73 7562 3e20 706c 6f74  sub>S</sub> plot
+000031b0: 2077 6974 6820 616e 6368 6f72 2070 6169   with anchor pai
+000031c0: 7273 2061 6e6e 6f74 6174 6564 2e0a 5468  rs annotated..Th
+000031d0: 6520 6041 7175 696c 6567 6961 5f63 6f65  e `Aquilegia_coe
+000031e0: 7275 6c65 612d 7673 2d41 7175 696c 6567  rulea-vs-Aquileg
+000031f0: 6961 5f63 6f65 7275 6c65 612e 646f 742e  ia_coerulea.dot.
+00003200: 7064 662f 706e 672f 7376 6760 2069 7320  pdf/png/svg` is 
+00003210: 7468 6520 646f 7420 706c 6f74 2077 6974  the dot plot wit
+00003220: 686f 7574 202a 4b2a 3c73 7562 3e53 3c2f  hout *K*<sub>S</
+00003230: 7375 623e 2061 6e6e 6f74 6174 696f 6e20  sub> annotation 
+00003240: 616e 6420 7769 7468 2063 6f6f 7264 696e  and with coordin
+00003250: 6174 6573 2069 6e20 7468 6520 6e75 6d62  ates in the numb
+00003260: 6572 206f 6620 6261 7365 2e0a 5468 6520  er of base..The 
+00003270: 6041 7175 696c 6567 6961 5f63 6f65 7275  `Aquilegia_coeru
+00003280: 6c65 612d 7673 2d41 7175 696c 6567 6961  lea-vs-Aquilegia
+00003290: 5f63 6f65 7275 6c65 612e 646f 745f 756e  _coerulea.dot_un
+000032a0: 6974 5f67 656e 652e 7064 662f 706e 672f  it_gene.pdf/png/
+000032b0: 7376 6760 2069 7320 7468 6520 646f 7420  svg` is the dot 
+000032c0: 706c 6f74 2077 6974 686f 7574 202a 4b2a  plot without *K*
+000032d0: 3c73 7562 3e53 3c2f 7375 623e 2061 6e6e  <sub>S</sub> ann
+000032e0: 6f74 6174 696f 6e20 616e 6420 7769 7468  otation and with
+000032f0: 2063 6f6f 7264 696e 6174 6573 2069 6e20   coordinates in 
+00003300: 7468 6520 6e75 6d62 6572 206f 6620 6765  the number of ge
+00003310: 6e65 2e0a 5468 6520 6041 7175 696c 6567  ne..The `Aquileg
+00003320: 6961 5f63 6f65 7275 6c65 612d 7673 2d41  ia_coerulea-vs-A
+00003330: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003340: 615f 4b73 2e64 6f74 2e70 6466 2f70 6e67  a_Ks.dot.pdf/png
+00003350: 2f73 7667 6020 6973 2074 6865 2064 6f74  /svg` is the dot
+00003360: 2070 6c6f 7420 7769 7468 202a 4b2a 3c73   plot with *K*<s
+00003370: 7562 3e53 3c2f 7375 623e 2061 6e6e 6f74  ub>S</sub> annot
+00003380: 6174 696f 6e20 616e 6420 7769 7468 2063  ation and with c
+00003390: 6f6f 7264 696e 6174 6573 2069 6e20 7468  oordinates in th
+000033a0: 6520 6e75 6d62 6572 206f 6620 6261 7365  e number of base
+000033b0: 2e0a 5468 6520 6041 7175 696c 6567 6961  ..The `Aquilegia
+000033c0: 5f63 6f65 7275 6c65 612d 7673 2d41 7175  _coerulea-vs-Aqu
+000033d0: 696c 6567 6961 5f63 6f65 7275 6c65 615f  ilegia_coerulea_
+000033e0: 4b73 2e64 6f74 5f75 6e69 745f 6765 6e65  Ks.dot_unit_gene
+000033f0: 2e70 6466 2f70 6e67 2f73 7667 6020 6973  .pdf/png/svg` is
+00003400: 2074 6865 2064 6f74 2070 6c6f 7420 7769   the dot plot wi
+00003410: 7468 202a 4b2a 3c73 7562 3e53 3c2f 7375  th *K*<sub>S</su
+00003420: 623e 2061 6e6e 6f74 6174 696f 6e20 616e  b> annotation an
+00003430: 6420 7769 7468 2063 6f6f 7264 696e 6174  d with coordinat
+00003440: 6573 2069 6e20 7468 6520 6e75 6d62 6572  es in the number
+00003450: 206f 6620 6765 6e65 2e0a 5468 6520 6069   of gene..The `i
+00003460: 6164 686f 7265 2e63 6f6e 6660 2069 7320  adhore.conf` is 
+00003470: 7468 6520 636f 6e66 6967 7572 6174 696f  the configuratio
+00003480: 6e20 6669 6c65 2066 6f72 2060 692d 6164  n file for `i-ad
+00003490: 686f 7265 602e 0a54 6865 2060 6961 6468  hore`..The `iadh
+000034a0: 6f72 652d 6f75 7460 2073 7562 666f 6c64  ore-out` subfold
+000034b0: 6572 2063 6f6e 7461 696e 7320 7468 6520  er contains the 
+000034c0: 6f72 6967 696e 616c 2063 6f6c 6c69 6e65  original colline
+000034d0: 6172 2072 6573 756c 7473 2066 726f 6d20  ar results from 
+000034e0: 6069 2d61 6468 6f72 6560 2028 706c 6561  `i-adhore` (plea
+000034f0: 7365 2072 6566 6572 2074 6f20 7468 6520  se refer to the 
+00003500: 6d61 6e75 616c 2074 6865 7265 696e 2066  manual therein f
+00003510: 6f72 2064 6574 6169 6c65 6420 6465 7363  or detailed desc
+00003520: 7269 7074 696f 6e29 2e0a 0a23 2323 2050  ription)...### P
+00003530: 6970 656c 696e 6520 330a 2a2a 436f 6e73  ipeline 3.**Cons
+00003540: 7472 7563 7469 6f6e 206f 6620 6167 6520  truction of age 
+00003550: 6469 7374 7269 6275 7469 6f6e 2061 6e64  distribution and
+00003560: 2045 4c4d 4d20 616e 616c 7973 6973 2a2a   ELMM analysis**
+00003570: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
+00003580: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+00003590: 2d6f 2077 6764 5f64 6d64 0a77 6764 206b  -o wgd_dmd.wgd k
+000035a0: 7364 2077 6764 5f64 6d64 2f41 7175 696c  sd wgd_dmd/Aquil
+000035b0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+000035c0: 7620 4171 7569 6c65 6769 615f 636f 6572  v Aquilegia_coer
+000035d0: 756c 6561 202d 6f20 7767 645f 6b73 640a  ulea -o wgd_ksd.
+000035e0: 7767 6420 7669 7a20 2d64 2077 6764 5f6b  wgd viz -d wgd_k
+000035f0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+00003600: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00003610: 202d 6f20 7767 645f 454c 4d4d 0a60 6060   -o wgd_ELMM.```
+00003620: 0a60 6060 0a23 2077 6861 7420 7769 6c6c  .```.# what will
+00003630: 2062 6520 696e 2074 6865 2077 6764 5f45   be in the wgd_E
+00003640: 4c4d 4d20 7265 7375 6c74 2064 6972 6563  LMM result direc
+00003650: 746f 7279 0a2d 7767 645f 454c 4d4d 0a2d  tory.-wgd_ELMM.-
+00003660: 2d41 7175 696c 6567 6961 5f63 6f65 7275  -Aquilegia_coeru
+00003670: 6c65 612e 7473 762e 6b73 2e74 7376 2e6b  lea.tsv.ks.tsv.k
+00003680: 7364 2e70 6466 2f73 7667 0a2d 2d41 7175  sd.pdf/svg.--Aqu
+00003690: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+000036a0: 7473 762e 6b73 2e74 7376 2e73 706c 696e  tsv.ks.tsv.splin
+000036b0: 655f 6e6f 6465 5f61 7665 7261 6765 642e  e_node_averaged.
+000036c0: 7064 662f 7376 670a 2d2d 4171 7569 6c65  pdf/svg.--Aquile
+000036d0: 6769 615f 636f 6572 756c 6561 2e74 7376  gia_coerulea.tsv
+000036e0: 2e6b 732e 7473 762e 7370 6c69 6e65 5f77  .ks.tsv.spline_w
+000036f0: 6569 6768 7465 642e 7064 662f 7376 670a  eighted.pdf/svg.
+00003700: 2d2d 4171 7569 6c65 6769 615f 636f 6572  --Aquilegia_coer
+00003710: 756c 6561 2e74 7376 2e6b 732e 7473 765f  ulea.tsv.ks.tsv_
+00003720: 7065 616b 5f64 6574 6563 7469 6f6e 5f6e  peak_detection_n
+00003730: 6f64 655f 6176 6572 6167 6564 2e70 6466  ode_averaged.pdf
+00003740: 2f73 7667 0a2d 2d41 7175 696c 6567 6961  /svg.--Aquilegia
+00003750: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00003760: 2e74 7376 5f70 6561 6b5f 6465 7465 6374  .tsv_peak_detect
+00003770: 696f 6e5f 7765 6967 6874 6564 2e70 6466  ion_weighted.pdf
+00003780: 2f73 7667 0a2d 2d65 6c6d 6d5f 4171 7569  /svg.--elmm_Aqui
+00003790: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000037a0: 7376 2e6b 732e 7473 765f 6265 7374 5f6d  sv.ks.tsv_best_m
+000037b0: 6f64 656c 735f 6e6f 6465 5f61 7665 7261  odels_node_avera
+000037c0: 6765 642e 7064 662f 7376 670a 2d2d 656c  ged.pdf/svg.--el
+000037d0: 6d6d 5f41 7175 696c 6567 6961 5f63 6f65  mm_Aquilegia_coe
+000037e0: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+000037f0: 5f62 6573 745f 6d6f 6465 6c73 5f77 6569  _best_models_wei
+00003800: 6768 7465 642e 7064 662f 7376 670a 2d2d  ghted.pdf/svg.--
+00003810: 656c 6d6d 5f41 7175 696c 6567 6961 5f63  elmm_Aquilegia_c
+00003820: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00003830: 7376 5f6d 6f64 656c 735f 6461 7461 5f64  sv_models_data_d
+00003840: 7269 7665 6e5f 6e6f 6465 5f61 7665 7261  riven_node_avera
+00003850: 6765 642e 7064 662f 7376 670a 2d2d 656c  ged.pdf/svg.--el
+00003860: 6d6d 5f41 7175 696c 6567 6961 5f63 6f65  mm_Aquilegia_coe
+00003870: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00003880: 5f6d 6f64 656c 735f 6461 7461 5f64 7269  _models_data_dri
+00003890: 7665 6e5f 7765 6967 6874 6564 2e70 6466  ven_weighted.pdf
+000038a0: 2f73 7667 0a2d 2d65 6c6d 6d5f 4171 7569  /svg.--elmm_Aqui
+000038b0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+000038c0: 7376 2e6b 732e 7473 765f 6d6f 6465 6c73  sv.ks.tsv_models
+000038d0: 5f72 616e 646f 6d5f 6e6f 6465 5f61 7665  _random_node_ave
+000038e0: 7261 6765 642e 7064 662f 7376 670a 2d2d  raged.pdf/svg.--
+000038f0: 656c 6d6d 5f41 7175 696c 6567 6961 5f63  elmm_Aquilegia_c
+00003900: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00003910: 7376 5f6d 6f64 656c 735f 7261 6e64 6f6d  sv_models_random
+00003920: 5f77 6569 6768 7465 642e 7064 662f 7376  _weighted.pdf/sv
+00003930: 670a 2d2d 656c 6d6d 5f42 4943 5f41 7175  g.--elmm_BIC_Aqu
+00003940: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00003950: 7473 762e 6b73 2e74 7376 5f6e 6f64 655f  tsv.ks.tsv_node_
+00003960: 6176 6572 6167 6564 2e70 6466 2f73 7667  averaged.pdf/svg
+00003970: 0a2d 2d65 6c6d 6d5f 4249 435f 4171 7569  .--elmm_BIC_Aqui
+00003980: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00003990: 7376 2e6b 732e 7473 765f 7765 6967 6874  sv.ks.tsv_weight
+000039a0: 6564 2e70 6466 2f73 7667 0a60 6060 0a54  ed.pdf/svg.```.T
+000039b0: 6865 2060 4171 7569 6c65 6769 615f 636f  he `Aquilegia_co
+000039c0: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
+000039d0: 762e 6b73 642e 7064 662f 7376 6760 2069  v.ksd.pdf/svg` i
+000039e0: 7320 7468 6520 6261 7369 6320 2a4b 2a3c  s the basic *K*<
+000039f0: 7375 623e 533c 2f73 7562 3e20 706c 6f74  sub>S</sub> plot
+00003a00: 2e20 5468 6520 6041 7175 696c 6567 6961  . The `Aquilegia
+00003a10: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00003a20: 2e74 7376 2e73 706c 696e 655f 6e6f 6465  .tsv.spline_node
+00003a30: 5f61 7665 7261 6765 642e 7064 662f 7376  _averaged.pdf/sv
+00003a40: 6760 2061 6e64 2060 4171 7569 6c65 6769  g` and `Aquilegi
+00003a50: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00003a60: 732e 7473 762e 7370 6c69 6e65 5f77 6569  s.tsv.spline_wei
+00003a70: 6768 7465 642e 7064 662f 7376 6760 2061  ghted.pdf/svg` a
+00003a80: 7265 2074 6865 2073 706c 696e 6520 706c  re the spline pl
+00003a90: 6f74 7320 7573 6564 2066 6f72 2070 6561  ots used for pea
+00003aa0: 6b20 6465 7465 6374 696f 6e2e 2054 6865  k detection. The
+00003ab0: 2060 4171 7569 6c65 6769 615f 636f 6572   `Aquilegia_coer
+00003ac0: 756c 6561 2e74 7376 2e6b 732e 7473 765f  ulea.tsv.ks.tsv_
+00003ad0: 7065 616b 5f64 6574 6563 7469 6f6e 5f6e  peak_detection_n
+00003ae0: 6f64 655f 6176 6572 6167 6564 2e70 6466  ode_averaged.pdf
+00003af0: 2f73 7667 6020 616e 6420 6041 7175 696c  /svg` and `Aquil
+00003b00: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00003b10: 762e 6b73 2e74 7376 5f70 6561 6b5f 6465  v.ks.tsv_peak_de
+00003b20: 7465 6374 696f 6e5f 7765 6967 6874 6564  tection_weighted
+00003b30: 2e70 6466 2f73 7667 6020 6172 6520 7468  .pdf/svg` are th
+00003b40: 6520 7265 7375 6c74 7320 6f66 2070 6561  e results of pea
+00003b50: 6b20 6465 7465 6374 696f 6e2e 2054 6865  k detection. The
+00003b60: 2060 656c 6d6d 5f41 7175 696c 6567 6961   `elmm_Aquilegia
+00003b70: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00003b80: 2e74 7376 5f62 6573 745f 6d6f 6465 6c73  .tsv_best_models
+00003b90: 5f6e 6f64 655f 6176 6572 6167 6564 2e70  _node_averaged.p
+00003ba0: 6466 2f73 7667 6020 616e 6420 6065 6c6d  df/svg` and `elm
+00003bb0: 6d5f 4171 7569 6c65 6769 615f 636f 6572  m_Aquilegia_coer
+00003bc0: 756c 6561 2e74 7376 2e6b 732e 7473 765f  ulea.tsv.ks.tsv_
+00003bd0: 6265 7374 5f6d 6f64 656c 735f 7765 6967  best_models_weig
+00003be0: 6874 6564 2e70 6466 2f73 7667 6020 6172  hted.pdf/svg` ar
+00003bf0: 6520 7468 6520 7265 7375 6c74 7320 6672  e the results fr
+00003c00: 6f6d 2074 6865 2062 6573 7420 6d6f 6465  om the best mode
+00003c10: 6c20 696e 666f 726d 6564 2062 7920 6042  l informed by `B
+00003c20: 4943 602e 2054 6865 2060 656c 6d6d 5f41  IC`. The `elmm_A
+00003c30: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003c40: 612e 7473 762e 6b73 2e74 7376 5f6d 6f64  a.tsv.ks.tsv_mod
+00003c50: 656c 735f 6461 7461 5f64 7269 7665 6e5f  els_data_driven_
+00003c60: 6e6f 6465 5f61 7665 7261 6765 642e 7064  node_averaged.pd
+00003c70: 662f 7376 6760 2061 6e64 2060 656c 6d6d  f/svg` and `elmm
+00003c80: 5f41 7175 696c 6567 6961 5f63 6f65 7275  _Aquilegia_coeru
+00003c90: 6c65 612e 7473 762e 6b73 2e74 7376 5f6d  lea.tsv.ks.tsv_m
+00003ca0: 6f64 656c 735f 6461 7461 5f64 7269 7665  odels_data_drive
+00003cb0: 6e5f 7765 6967 6874 6564 2e70 6466 2f73  n_weighted.pdf/s
+00003cc0: 7667 6020 6172 6520 7468 6520 6d6f 6465  vg` are the mode
+00003cd0: 6c20 7265 7375 6c74 7320 6672 6f6d 2064  l results from d
+00003ce0: 6174 612d 6472 6976 656e 2069 6e69 7469  ata-driven initi
+00003cf0: 6174 696f 6e2e 2054 6865 2060 656c 6d6d  ation. The `elmm
+00003d00: 5f41 7175 696c 6567 6961 5f63 6f65 7275  _Aquilegia_coeru
+00003d10: 6c65 612e 7473 762e 6b73 2e74 7376 5f6d  lea.tsv.ks.tsv_m
+00003d20: 6f64 656c 735f 7261 6e64 6f6d 5f6e 6f64  odels_random_nod
+00003d30: 655f 6176 6572 6167 6564 2e70 6466 2f73  e_averaged.pdf/s
+00003d40: 7667 6020 616e 6420 6065 6c6d 6d5f 4171  vg` and `elmm_Aq
+00003d50: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00003d60: 2e74 7376 2e6b 732e 7473 765f 6d6f 6465  .tsv.ks.tsv_mode
+00003d70: 6c73 5f72 616e 646f 6d5f 7765 6967 6874  ls_random_weight
+00003d80: 6564 2e70 6466 2f73 7667 6020 6172 6520  ed.pdf/svg` are 
+00003d90: 7468 6520 6d6f 6465 6c20 7265 7375 6c74  the model result
+00003da0: 7320 6672 6f6d 2072 616e 646f 6d20 696e  s from random in
+00003db0: 6974 6961 7469 6f6e 2e20 5468 6520 6065  itiation. The `e
+00003dc0: 6c6d 6d5f 4249 435f 4171 7569 6c65 6769  lmm_BIC_Aquilegi
+00003dd0: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00003de0: 732e 7473 765f 6e6f 6465 5f61 7665 7261  s.tsv_node_avera
+00003df0: 6765 642e 7064 662f 7376 6760 2061 6e64  ged.pdf/svg` and
+00003e00: 2060 656c 6d6d 5f42 4943 5f41 7175 696c   `elmm_BIC_Aquil
+00003e10: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00003e20: 762e 6b73 2e74 7376 5f77 6569 6768 7465  v.ks.tsv_weighte
+00003e30: 642e 7064 662f 7376 6760 2061 7265 2074  d.pdf/svg` are t
+00003e40: 6865 2042 4943 2070 6c6f 7473 2066 6f72  he BIC plots for
+00003e50: 2065 6163 6820 6d6f 6465 6c2e 0a0a 2323   each model...##
+00003e60: 2320 5069 7065 6c69 6e65 2034 0a2a 2a43  # Pipeline 4.**C
+00003e70: 6f6e 7374 7275 6374 696f 6e20 6f66 2061  onstruction of a
+00003e80: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00003e90: 7769 7468 2063 6f6c 6c69 6e65 6172 6974  with collinearit
+00003ea0: 7920 616e 6420 7065 616b 2066 696e 6469  y and peak findi
+00003eb0: 6e67 2061 6e61 6c79 7369 732a 2a0a 6060  ng analysis**.``
+00003ec0: 600a 7767 6420 646d 6420 4171 7569 6c65  `.wgd dmd Aquile
+00003ed0: 6769 615f 636f 6572 756c 6561 202d 6f20  gia_coerulea -o 
+00003ee0: 7767 645f 646d 640a 7767 6420 6b73 6420  wgd_dmd.wgd ksd 
+00003ef0: 7767 645f 646d 642f 4171 7569 6c65 6769  wgd_dmd/Aquilegi
+00003f00: 615f 636f 6572 756c 6561 2e74 7376 2041  a_coerulea.tsv A
+00003f10: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+00003f20: 6120 2d6f 2077 6764 5f6b 7364 0a77 6764  a -o wgd_ksd.wgd
+00003f30: 2073 796e 202d 6620 6d52 4e41 202d 6120   syn -f mRNA -a 
+00003f40: 4e61 6d65 2077 6764 5f64 6d64 2f41 7175  Name wgd_dmd/Aqu
+00003f50: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00003f60: 7473 7620 4171 7569 6c65 6769 615f 636f  tsv Aquilegia_co
+00003f70: 6572 756c 6561 2e67 6666 3320 2d6b 7320  erulea.gff3 -ks 
+00003f80: 7767 645f 6b73 642f 4171 7569 6c65 6769  wgd_ksd/Aquilegi
+00003f90: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00003fa0: 732e 7473 7620 2d6f 2077 6764 5f73 796e  s.tsv -o wgd_syn
+00003fb0: 0a77 6764 2070 6561 6b20 2d2d 6865 7572  .wgd peak --heur
+00003fc0: 6973 7469 6320 7767 645f 6b73 642f 4171  istic wgd_ksd/Aq
+00003fd0: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00003fe0: 2e74 7376 2e6b 732e 7473 7620 2d61 7020  .tsv.ks.tsv -ap 
+00003ff0: 7767 645f 7379 6e2f 6961 6468 6f72 652d  wgd_syn/iadhore-
+00004000: 6f75 742f 616e 6368 6f72 706f 696e 7473  out/anchorpoints
+00004010: 2e74 7874 202d 736d 2077 6764 5f73 796e  .txt -sm wgd_syn
+00004020: 2f69 6164 686f 7265 2d6f 7574 2f73 6567  /iadhore-out/seg
+00004030: 6d65 6e74 732e 7478 7420 2d6c 6520 7767  ments.txt -le wg
+00004040: 645f 7379 6e2f 6961 6468 6f72 652d 6f75  d_syn/iadhore-ou
+00004050: 742f 6c69 7374 5f65 6c65 6d65 6e74 732e  t/list_elements.
+00004060: 7478 7420 2d6d 7020 7767 645f 7379 6e2f  txt -mp wgd_syn/
+00004070: 6961 6468 6f72 652d 6f75 742f 6d75 6c74  iadhore-out/mult
+00004080: 6970 6c69 636f 6e5f 7061 6972 732e 7478  iplicon_pairs.tx
+00004090: 7420 2d6e 2031 2034 202d 6b63 2033 202d  t -n 1 4 -kc 3 -
+000040a0: 6f20 7767 645f 7065 616b 0a60 6060 0a60  o wgd_peak.```.`
+000040b0: 6060 0a23 2077 6861 7420 7769 6c6c 2062  ``.# what will b
+000040c0: 6520 696e 2074 6865 2077 6764 5f70 6561  e in the wgd_pea
+000040d0: 6b20 7265 7375 6c74 2064 6972 6563 746f  k result directo
+000040e0: 7279 0a2d 7767 645f 7065 616b 0a2d 2d41  ry.-wgd_peak.--A
+000040f0: 6e63 686f 724b 735f 4669 6e64 5065 616b  nchorKs_FindPeak
+00004100: 0a2d 2d2d 416e 6368 6f72 4b73 5f50 6561  .---AnchorKs_Pea
+00004110: 6b43 495f 4171 7569 6c65 6769 615f 636f  kCI_Aquilegia_co
+00004120: 6572 756c 6561 2e74 7376 2e6b 732e 7473  erulea.tsv.ks.ts
+00004130: 765f 6e6f 6465 5f61 7665 7261 6765 642f  v_node_averaged/
+00004140: 7765 6967 6874 6564 2e70 6466 0a2d 2d2d  weighted.pdf.---
+00004150: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00004160: 6561 2e74 7376 2e6b 732e 7473 765f 3935  ea.tsv.ks.tsv_95
+00004170: 2543 495f 4150 5f66 6f72 5f64 6174 696e  %CI_AP_for_datin
+00004180: 675f 7765 6967 6874 6564 5f66 6f72 6d61  g_weighted_forma
+00004190: 742e 7473 760a 2d2d 2d41 7175 696c 6567  t.tsv.---Aquileg
+000041a0: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+000041b0: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
+000041c0: 666f 725f 6461 7469 6e67 5f77 6569 6768  for_dating_weigh
+000041d0: 7465 642e 7473 760a 2d2d 2d41 7175 696c  ted.tsv.---Aquil
+000041e0: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+000041f0: 762e 6b73 2e74 7376 5f39 3525 4349 5f41  v.ks.tsv_95%CI_A
+00004200: 505f 666f 725f 6461 7469 6e67 5f6e 6f64  P_for_dating_nod
+00004210: 655f 6176 6572 6167 6564 5f66 6f72 6d61  e_averaged_forma
+00004220: 742e 7473 760a 2d2d 2d41 7175 696c 6567  t.tsv.---Aquileg
+00004230: 6961 5f63 6f65 7275 6c65 612e 7473 762e  ia_coerulea.tsv.
+00004240: 6b73 2e74 7376 5f39 3525 4349 5f41 505f  ks.tsv_95%CI_AP_
+00004250: 666f 725f 6461 7469 6e67 5f6e 6f64 655f  for_dating_node_
+00004260: 6176 6572 6167 6564 2e74 7376 0a2d 2d41  averaged.tsv.--A
+00004270: 6e63 686f 724b 735f 474d 4d0a 2d2d 2d47  nchorKs_GMM.---G
+00004280: 4d4d 5f45 6c62 6f77 2d4c 6f73 735f 6f72  MM_Elbow-Loss_or
+00004290: 6967 696e 616c 5f4b 732e 7064 660a 2d2d  iginal_Ks.pdf.--
+000042a0: 2d47 4d4d 5f4f 7269 6769 6e61 6c5f 416e  -GMM_Original_An
+000042b0: 6368 6f72 4b73 5f43 6c75 7374 6572 696e  chorKs_Clusterin
+000042c0: 675f 5369 6c68 6f75 6574 7465 5f43 6f65  g_Silhouette_Coe
+000042d0: 6666 6963 6965 6e74 2e70 6466 0a2d 2d2d  fficient.pdf.---
+000042e0: 4f72 6967 696e 616c 5f41 6e63 686f 724b  Original_AnchorK
+000042f0: 735f 474d 4d5f 312f 322f 332f 3463 6f6d  s_GMM_1/2/3/4com
+00004300: 706f 6e65 6e74 735f 7072 6564 6963 7469  ponents_predicti
+00004310: 6f6e 2e74 7376 0a2d 2d2d 4f72 6967 696e  on.tsv.---Origin
+00004320: 616c 5f41 6e63 686f 724b 735f 474d 4d5f  al_AnchorKs_GMM_
+00004330: 4149 435f 4249 432e 7064 660a 2d2d 2d4f  AIC_BIC.pdf.---O
+00004340: 7269 6769 6e61 6c5f 416e 6368 6f72 4b73  riginal_AnchorKs
+00004350: 5f47 4d4d 5f43 6f6d 706f 6e65 6e74 312f  _GMM_Component1/
+00004360: 322f 332f 345f 6e6f 6465 5f61 7665 7261  2/3/4_node_avera
+00004370: 6765 645f 4c6f 676e 6f72 6d61 6c2e 7064  ged_Lognormal.pd
+00004380: 660a 2d2d 2d4f 7269 6769 6e61 6c5f 416e  f.---Original_An
+00004390: 6368 6f72 4b73 5f47 4d4d 5f43 6f6d 706f  chorKs_GMM_Compo
+000043a0: 6e65 6e74 312f 322f 332f 345f 6e6f 6465  nent1/2/3/4_node
+000043b0: 5f61 7665 7261 6765 642e 7064 660a 2d2d  _averaged.pdf.--
+000043c0: 2d4c 6f67 474d 4d5f 4349 0a2d 2d2d 2d47  -LogGMM_CI.----G
+000043d0: 4d4d 5f31 2f32 2f33 2f34 636f 6d70 6f6e  MM_1/2/3/4compon
+000043e0: 656e 7473 5f43 302f 312f 322f 335f 3935  ents_C0/1/2/3_95
+000043f0: 2543 492e 7473 760a 2d2d 2d2d 474d 4d5f  %CI.tsv.----GMM_
+00004400: 436f 6d70 6f6e 656e 7431 2f32 2f33 2f34  Component1/2/3/4
+00004410: 5f6e 6f64 655f 6176 6572 6167 6564 5f4c  _node_averaged_L
+00004420: 6f67 6e6f 726d 616c 2e70 6466 0a2d 2d2d  ognormal.pdf.---
+00004430: 4869 6768 4d61 7373 5f43 490a 2d2d 2d2d  HighMass_CI.----
+00004440: 474d 4d5f 312f 322f 332f 3463 6f6d 706f  GMM_1/2/3/4compo
+00004450: 6e65 6e74 735f 4330 2f31 2f32 2f33 5f48  nents_C0/1/2/3_H
+00004460: 6967 684d 6173 735f 3935 2543 492e 7473  ighMass_95%CI.ts
+00004470: 760a 2d2d 2d2d 474d 4d5f 312f 322f 332f  v.----GMM_1/2/3/
+00004480: 3463 6f6d 706f 6e65 6e74 735f 4869 6768  4components_High
+00004490: 4d61 7373 5f39 3525 4349 2e70 6466 0a2d  Mass_95%CI.pdf.-
+000044a0: 2d53 6567 6d65 6e74 4775 6964 654b 735f  -SegmentGuideKs_
+000044b0: 474d 4d0a 2d2d 2d47 4d4d 5f45 6c62 6f77  GMM.---GMM_Elbow
+000044c0: 2d4c 6f73 735f 5365 676d 656e 745f 4b73  -Loss_Segment_Ks
+000044d0: 2e70 6466 0a2d 2d2d 474d 4d5f 5365 676d  .pdf.---GMM_Segm
+000044e0: 656e 745f 4b73 5f43 6c75 7374 6572 696e  ent_Ks_Clusterin
+000044f0: 675f 5369 6c68 6f75 6574 7465 5f43 6f65  g_Silhouette_Coe
+00004500: 6666 6963 6965 6e74 2e70 6466 0a2d 2d2d  fficient.pdf.---
+00004510: 5365 676d 656e 742d 6775 6964 6564 5f41  Segment-guided_A
+00004520: 6e63 686f 724b 735f 474d 4d5f 312f 322f  nchorKs_GMM_1/2/
+00004530: 332f 3463 6f6d 706f 6e65 6e74 735f 7072  3/4components_pr
+00004540: 6564 6963 7469 6f6e 2e74 7376 0a2d 2d2d  ediction.tsv.---
+00004550: 5365 676d 656e 745f 4b73 5f43 6c75 7374  Segment_Ks_Clust
+00004560: 6572 735f 474d 4d5f 436f 6d70 6f6e 656e  ers_GMM_Componen
+00004570: 7431 2f32 2f33 2f34 2e70 6466 0a2d 2d2d  t1/2/3/4.pdf.---
+00004580: 5365 676d 656e 745f 4b73 5f43 6c75 7374  Segment_Ks_Clust
+00004590: 6572 735f 4c6f 676e 6f72 6d61 6c5f 474d  ers_Lognormal_GM
+000045a0: 4d5f 436f 6d70 6f6e 656e 7431 2f32 2f33  M_Component1/2/3
+000045b0: 2f34 2e70 6466 0a2d 2d2d 5365 676d 656e  /4.pdf.---Segmen
+000045c0: 745f 4b73 5f47 4d4d 5f41 4943 5f42 4943  t_Ks_GMM_AIC_BIC
+000045d0: 2e70 6466 0a2d 2d2d 5365 676d 656e 745f  .pdf.---Segment_
+000045e0: 4b73 2e74 7376 0a2d 2d2d 4869 6768 4d61  Ks.tsv.---HighMa
+000045f0: 7373 5f43 490a 2d2d 2d2d 5365 676d 656e  ss_CI.----Segmen
+00004600: 745f 6775 6964 6564 5f31 2f32 2f33 2f34  t_guided_1/2/3/4
+00004610: 636f 6d70 6f6e 656e 7473 5f43 302f 312f  components_C0/1/
+00004620: 322f 335f 4869 6768 4d61 7373 5f39 3525  2/3_HighMass_95%
+00004630: 4349 2e74 7376 0a2d 2d2d 2d53 6567 6d65  CI.tsv.----Segme
+00004640: 6e74 5f67 7569 6465 645f 312f 322f 332f  nt_guided_1/2/3/
+00004650: 3463 6f6d 706f 6e65 6e74 735f 4869 6768  4components_High
+00004660: 4d61 7373 5f39 3525 4349 2e70 6466 0a2d  Mass_95%CI.pdf.-
+00004670: 2d2d 4844 525f 4349 0a2d 2d2d 2d53 6567  --HDR_CI.----Seg
+00004680: 6d65 6e74 5f67 7569 6465 645f 3935 2548  ment_guided_95%H
+00004690: 4452 5f41 505f 312f 322f 332f 3463 6f6d  DR_AP_1/2/3/4com
+000046a0: 706f 6e65 6e74 735f 4330 2f31 2f32 2f33  ponents_C0/1/2/3
+000046b0: 2e74 7376 0a2d 2d2d 2d53 6567 6d65 6e74  .tsv.----Segment
+000046c0: 5f67 7569 6465 645f 416e 6368 6f72 4b73  _guided_AnchorKs
+000046d0: 5f47 4d4d 5f43 6f6d 706f 6e65 6e74 312f  _GMM_Component1/
+000046e0: 322f 332f 345f 6e6f 6465 5f61 7665 7261  2/3/4_node_avera
+000046f0: 6765 645f 6b64 652e 7064 660a 2d2d 2d2d  ged_kde.pdf.----
+00004700: 5365 676d 656e 745f 6775 6964 6564 5f41  Segment_guided_A
+00004710: 6e63 686f 724b 735f 474d 4d5f 436f 6d70  nchorKs_GMM_Comp
+00004720: 6f6e 656e 7431 2f32 2f33 2f34 5f6e 6f64  onent1/2/3/4_nod
+00004730: 655f 6176 6572 6167 6564 2e70 6466 0a2d  e_averaged.pdf.-
+00004740: 2d53 6567 6d65 6e74 4b73 5f46 696e 6450  -SegmentKs_FindP
+00004750: 6561 6b0a 2d2d 2d53 6567 6d65 6e74 4b73  eak.---SegmentKs
+00004760: 5f50 6561 6b43 495f 4171 7569 6c65 6769  _PeakCI_Aquilegi
+00004770: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00004780: 732e 7473 762e 7064 660a 2d2d 2d50 6561  s.tsv.pdf.---Pea
+00004790: 6b5f 312f 325f 5365 676d 656e 745f 6775  k_1/2_Segment_gu
+000047a0: 6964 6564 5f41 7175 696c 6567 6961 5f63  ided_Aquilegia_c
+000047b0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+000047c0: 7376 5f39 3525 4349 5f4d 505f 666f 725f  sv_95%CI_MP_for_
+000047d0: 6461 7469 6e67 5f66 6f72 6d61 742e 7473  dating_format.ts
+000047e0: 760a 2d2d 2d50 6561 6b5f 312f 325f 5365  v.---Peak_1/2_Se
+000047f0: 676d 656e 745f 6775 6964 6564 5f41 7175  gment_guided_Aqu
+00004800: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00004810: 7473 762e 6b73 2e74 7376 5f39 3525 4349  tsv.ks.tsv_95%CI
+00004820: 5f4d 505f 666f 725f 6461 7469 6e67 2e74  _MP_for_dating.t
+00004830: 7376 0a60 6060 0a46 6f75 7220 7265 7375  sv.```.Four resu
+00004840: 6c74 2073 7562 666c 6f64 6572 7320 7769  lt subfloders wi
+00004850: 6c6c 2062 6520 7072 6f64 7563 6564 2c20  ll be produced, 
+00004860: 6e61 6d65 6c79 2060 416e 6368 6f72 4b73  namely `AnchorKs
+00004870: 5f46 696e 6450 6561 6b60 2c20 6041 6e63  _FindPeak`, `Anc
+00004880: 686f 724b 735f 474d 4d60 2c20 6053 6567  horKs_GMM`, `Seg
+00004890: 6d65 6e74 4775 6964 654b 735f 474d 4d60  mentGuideKs_GMM`
+000048a0: 2061 6e64 2060 5365 676d 656e 744b 735f   and `SegmentKs_
+000048b0: 4669 6e64 5065 616b 602e 2054 6865 2060  FindPeak`. The `
+000048c0: 416e 6368 6f72 4b73 5f46 696e 6450 6561  AnchorKs_FindPea
+000048d0: 6b60 2073 7562 666c 6f64 6572 2063 6f6e  k` subfloder con
+000048e0: 7461 696e 7320 7265 7375 6c74 7320 6f66  tains results of
+000048f0: 2074 6865 2064 6574 6563 7465 6420 7065   the detected pe
+00004900: 616b 7320 6279 2074 6865 2060 7369 676e  aks by the `sign
+00004910: 616c 6020 6d6f 6475 6c65 206f 6620 6053  al` module of `S
+00004920: 6369 5079 6020 6c69 6272 6172 7920 616e  ciPy` library an
+00004930: 6420 7468 6520 6173 7375 6d65 6420 6869  d the assumed hi
+00004940: 6768 6573 7420 6d61 7373 2070 6172 7420  ghest mass part 
+00004950: 2872 6566 6572 7265 6420 746f 2061 7320  (referred to as 
+00004960: 4869 6768 4d61 7373 2068 6572 6561 6674  HighMass hereaft
+00004970: 6572 2920 6f66 2065 6163 6820 7065 616b  er) of each peak
+00004980: 2c20 7768 6963 6820 6361 6e20 6265 2075  , which can be u
+00004990: 7365 6420 666f 7220 6675 7274 6865 7220  sed for further 
+000049a0: 5747 4420 6461 7469 6e67 2e20 5468 6520  WGD dating. The 
+000049b0: 6041 6e63 686f 724b 735f 474d 4d60 2073  `AnchorKs_GMM` s
+000049c0: 686f 7773 2074 6865 2047 4d4d 2072 6573  hows the GMM res
+000049d0: 756c 7473 2075 706f 6e20 7468 6520 6f72  ults upon the or
+000049e0: 6967 696e 616c 2061 6e63 686f 7220 2a4b  iginal anchor *K
+000049f0: 2a3c 7375 623e 533c 2f73 7562 3e20 6469  *<sub>S</sub> di
+00004a00: 7374 7269 6275 7469 6f6e 2062 7920 7468  stribution by th
+00004a10: 6520 606d 6978 7475 7265 6020 6d6f 6475  e `mixture` modu
+00004a20: 6c65 206f 6620 6073 6369 6b69 742d 6c65  le of `scikit-le
+00004a30: 6172 6e60 206c 6962 7261 7279 2061 6e64  arn` library and
+00004a40: 2074 776f 2073 7562 666c 6f64 6572 732c   two subfloders,
+00004a50: 2060 4c6f 6747 4d4d 5f43 4960 2063 6f6e   `LogGMM_CI` con
+00004a60: 7461 696e 696e 6720 7468 6520 7265 7375  taining the resu
+00004a70: 6c74 7320 6f66 2039 3525 2043 4920 6f66  lts of 95% CI of
+00004a80: 2065 6163 6820 636f 6d70 6f6e 656e 742c   each component,
+00004a90: 2060 4869 6768 4d61 7373 5f43 4960 2063   `HighMass_CI` c
+00004aa0: 6f6e 7461 696e 696e 6720 7468 6520 4869  ontaining the Hi
+00004ab0: 6768 4d61 7373 206f 6620 6561 6368 2063  ghMass of each c
+00004ac0: 6f6d 706f 6e65 6e74 2c20 7768 6963 6820  omponent, which 
+00004ad0: 6361 6e20 6265 2075 7365 6420 666f 7220  can be used for 
+00004ae0: 6675 7274 6865 7220 5747 4420 6461 7469  further WGD dati
+00004af0: 6e67 2e20 5468 6520 6053 6567 6d65 6e74  ng. The `Segment
+00004b00: 4775 6964 654b 735f 474d 4d60 2073 7562  GuideKs_GMM` sub
+00004b10: 666f 6c64 6572 2070 7265 7365 6e74 7320  folder presents 
+00004b20: 7265 7375 6c74 7320 6f66 2073 6567 6d65  results of segme
+00004b30: 6e74 202a 4b2a 3c73 7562 3e53 3c2f 7375  nt *K*<sub>S</su
+00004b40: 623e 2047 4d4d 2077 6869 6368 2061 7265  b> GMM which are
+00004b50: 206d 6170 7065 6420 6261 636b 2074 6f20   mapped back to 
+00004b60: 7468 6520 7265 7369 6469 6e67 2061 6e63  the residing anc
+00004b70: 686f 7220 7061 6972 7320 616e 6420 7468  hor pairs and th
+00004b80: 6520 6173 736f 6369 6174 6564 2039 3525  e associated 95%
+00004b90: 2048 4452 2061 6e64 2048 6967 684d 6173   HDR and HighMas
+00004ba0: 7320 6f66 2065 6163 6820 7365 676d 656e  s of each segmen
+00004bb0: 7420 636c 7573 7465 7220 696e 2073 7562  t cluster in sub
+00004bc0: 666c 6f64 6572 7320 6f66 2060 4844 525f  floders of `HDR_
+00004bd0: 4349 6020 616e 6420 6048 6967 684d 6173  CI` and `HighMas
+00004be0: 735f 4349 602e 2054 6865 2060 5365 676d  s_CI`. The `Segm
+00004bf0: 656e 744b 735f 4669 6e64 5065 616b 6020  entKs_FindPeak` 
+00004c00: 7375 6266 6f6c 6465 7220 6973 2073 696d  subfolder is sim
+00004c10: 696c 6172 2074 6f20 6041 6e63 686f 724b  ilar to `AnchorK
+00004c20: 735f 4669 6e64 5065 616b 6020 6275 7420  s_FindPeak` but 
+00004c30: 7769 7468 2073 6567 6d65 6e74 202a 4b2a  with segment *K*
+00004c40: 3c73 7562 3e53 3c2f 7375 623e 2069 6e73  <sub>S</sub> ins
+00004c50: 7465 6164 2e20 5468 6520 2a4b 2a3c 7375  tead. The *K*<su
+00004c60: 623e 533c 2f73 7562 3e20 696e 2060 4d75  b>S</sub> in `Mu
+00004c70: 6c74 6970 6c69 636f 6e60 2063 616e 2061  ltiplicon` can a
+00004c80: 6c73 6f20 6265 2063 616c 6375 6c61 7465  lso be calculate
+00004c90: 6420 696e 2070 6c61 6365 206f 6620 6053  d in place of `S
+00004ca0: 6567 6d65 6e74 6020 7573 696e 6720 7468  egment` using th
+00004cb0: 6520 6f70 7469 6f6e 2060 2d2d 6775 6964  e option `--guid
+00004cc0: 6560 2061 7320 7375 6368 2074 6865 2072  e` as such the r
+00004cd0: 6573 756c 7420 7469 746c 652c 206c 6162  esult title, lab
+00004ce0: 656c 2c20 6669 6c65 2061 6e64 2066 6f6c  el, file and fol
+00004cf0: 6465 7220 6e61 6d65 7320 7769 6c6c 2062  der names will b
+00004d00: 6520 6368 616e 6765 6420 6163 636f 7264  e changed accord
+00004d10: 696e 676c 792e 0a0a 2323 2320 5069 7065  ingly...### Pipe
+00004d20: 6c69 6e65 2035 0a2a 2a43 6f6e 7374 7275  line 5.**Constru
+00004d30: 6374 696f 6e20 6f66 2061 6765 2064 6973  ction of age dis
+00004d40: 7472 6962 7574 696f 6e20 7769 7468 2063  tribution with c
+00004d50: 6f6c 6c69 6e65 6172 6974 7920 616e 6420  ollinearity and 
+00004d60: 7261 7465 2063 6f72 7265 6374 696f 6e2a  rate correction*
+00004d70: 2a0a 6060 600a 7767 6420 646d 6420 4171  *.```.wgd dmd Aq
+00004d80: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00004d90: 202d 6f20 7767 645f 646d 640a 7767 6420   -o wgd_dmd.wgd 
+00004da0: 6b73 6420 7767 645f 646d 642f 4171 7569  ksd wgd_dmd/Aqui
+00004db0: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00004dc0: 7376 2041 7175 696c 6567 6961 5f63 6f65  sv Aquilegia_coe
+00004dd0: 7275 6c65 6120 2d6f 2077 6764 5f6b 7364  rulea -o wgd_ksd
+00004de0: 0a77 6764 2073 796e 202d 6620 6d52 4e41  .wgd syn -f mRNA
+00004df0: 202d 6120 4e61 6d65 2077 6764 5f64 6d64   -a Name wgd_dmd
+00004e00: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00004e10: 6c65 612e 7473 7620 4171 7569 6c65 6769  lea.tsv Aquilegi
+00004e20: 615f 636f 6572 756c 6561 2e67 6666 3320  a_coerulea.gff3 
+00004e30: 2d6b 7320 7767 645f 6b73 642f 4171 7569  -ks wgd_ksd/Aqui
+00004e40: 6c65 6769 615f 636f 6572 756c 6561 2e74  legia_coerulea.t
+00004e50: 7376 2e6b 732e 7473 7620 2d6f 2077 6764  sv.ks.tsv -o wgd
+00004e60: 5f73 796e 0a77 6764 2064 6d64 202d 2d67  _syn.wgd dmd --g
+00004e70: 6c6f 6261 6c6d 7262 6820 4171 7569 6c65  lobalmrbh Aquile
+00004e80: 6769 615f 636f 6572 756c 6561 2050 726f  gia_coerulea Pro
+00004e90: 7465 615f 6379 6e61 726f 6964 6573 2041  tea_cynaroides A
+00004ea0: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00004eb0: 2056 6974 6973 5f76 696e 6966 6572 6120   Vitis_vinifera 
+00004ec0: 2d6f 2077 6764 5f67 6c6f 6261 6c6d 7262  -o wgd_globalmrb
+00004ed0: 680a 7767 6420 6b73 6420 7767 645f 676c  h.wgd ksd wgd_gl
+00004ee0: 6f62 616c 6d72 6268 2f67 6c6f 6261 6c5f  obalmrbh/global_
+00004ef0: 4d52 4248 2e74 7376 2041 7175 696c 6567  MRBH.tsv Aquileg
+00004f00: 6961 5f63 6f65 7275 6c65 6120 5072 6f74  ia_coerulea Prot
+00004f10: 6561 5f63 796e 6172 6f69 6465 7320 4163  ea_cynaroides Ac
+00004f20: 6f72 7573 5f61 6d65 7269 6361 6e75 7320  orus_americanus 
+00004f30: 5669 7469 735f 7669 6e69 6665 7261 202d  Vitis_vinifera -
+00004f40: 6f20 7767 645f 676c 6f62 616c 6d72 6268  o wgd_globalmrbh
+00004f50: 5f6b 730a 7767 6420 7669 7a20 2d64 2077  _ks.wgd viz -d w
+00004f60: 6764 5f67 6c6f 6261 6c6d 7262 685f 6b73  gd_globalmrbh_ks
+00004f70: 2f67 6c6f 6261 6c5f 4d52 4248 2e74 7376  /global_MRBH.tsv
+00004f80: 2e6b 732e 7473 7620 2d66 6120 4171 7569  .ks.tsv -fa Aqui
+00004f90: 6c65 6769 615f 636f 6572 756c 6561 202d  legia_coerulea -
+00004fa0: 6570 6b20 7767 645f 6b73 642f 4171 7569  epk wgd_ksd/Aqui
+00004fb0: 6c65 6769 615f 636f 6572 756c 6561 2e6b  legia_coerulea.k
+00004fc0: 732e 7473 7620 2d61 7020 7767 645f 7379  s.tsv -ap wgd_sy
+00004fd0: 6e2f 6961 6468 6f72 652d 6f75 742f 616e  n/iadhore-out/an
+00004fe0: 6368 6f72 706f 696e 7473 2e74 7874 202d  chorpoints.txt -
+00004ff0: 7370 2073 7065 6369 6573 7472 6565 2e6e  sp speciestree.n
+00005000: 7720 2d6f 2077 6764 5f76 697a 5f6d 6978  w -o wgd_viz_mix
+00005010: 6564 5f4b 7320 2d2d 706c 6f74 656c 6d6d  ed_Ks --plotelmm
+00005020: 202d 2d70 6c6f 7461 7067 6d6d 202d 2d72   --plotapgmm --r
+00005030: 6577 6569 6768 740a 6060 600a 6060 600a  eweight.```.```.
+00005040: 2320 7768 6174 2077 696c 6c20 6265 2069  # what will be i
+00005050: 6e20 7468 6520 7767 645f 7669 7a5f 6d69  n the wgd_viz_mi
+00005060: 7865 645f 4b73 2072 6573 756c 7420 6469  xed_Ks result di
+00005070: 7265 6374 6f72 790a 2d77 6764 5f76 697a  rectory.-wgd_viz
+00005080: 5f6d 6978 6564 5f4b 730a 2d2d 416c 6c5f  _mixed_Ks.--All_
+00005090: 7061 6972 732e 6b73 2e6e 6f64 652e 7765  pairs.ks.node.we
+000050a0: 6967 6874 6564 2e70 6466 0a2d 2d46 6f63  ighted.pdf.--Foc
+000050b0: 7573 5f73 6973 7465 725f 7061 6972 732e  us_sister_pairs.
+000050c0: 6b73 2e6e 6f64 652e 7765 6967 6874 6564  ks.node.weighted
+000050d0: 2e70 6466 0a2d 2d67 6c6f 6261 6c5f 4d52  .pdf.--global_MR
+000050e0: 4248 2e74 7376 2e6b 732e 7473 762e 6b73  BH.tsv.ks.tsv.ks
+000050f0: 642e 7064 662f 7376 670a 2d2d 4d69 7865  d.pdf/svg.--Mixe
+00005100: 642e 6b73 2e41 7175 696c 6567 6961 5f63  d.ks.Aquilegia_c
+00005110: 6f65 7275 6c65 612e 6e6f 6465 2e77 6569  oerulea.node.wei
+00005120: 6768 7465 642e 7064 660a 2d2d 7370 6169  ghted.pdf.--spai
+00005130: 722e 636f 7272 6563 7465 642e 6b73 2e69  r.corrected.ks.i
+00005140: 6e66 6f2e 7473 760a 2d2d 7370 6169 722e  nfo.tsv.--spair.
+00005150: 6f72 6967 696e 616c 2e6b 732e 696e 666f  original.ks.info
+00005160: 2e74 7376 0a2d 2d53 696d 706c 655f 4b73  .tsv.--Simple_Ks
+00005170: 5f44 6973 7472 6962 7574 696f 6e73 0a2d  _Distributions.-
+00005180: 2d2d 4163 6f72 7573 5f61 6d65 7269 6361  --Acorus_america
+00005190: 6e75 732f 4171 7569 6c65 6769 615f 636f  nus/Aquilegia_co
+000051a0: 6572 756c 6561 2f50 726f 7465 615f 6379  erulea/Protea_cy
+000051b0: 6e61 726f 6964 6573 2f56 6974 6973 5f76  naroides/Vitis_v
+000051c0: 696e 6966 6572 615f 5f41 7175 696c 6567  inifera__Aquileg
+000051d0: 6961 5f63 6f65 7275 6c65 612f 5072 6f74  ia_coerulea/Prot
+000051e0: 6561 5f63 796e 6172 6f69 6465 732f 5669  ea_cynaroides/Vi
+000051f0: 7469 735f 7669 6e69 6665 7261 2e6b 732e  tis_vinifera.ks.
+00005200: 6e6f 6465 2e77 6569 6768 7465 642e 7064  node.weighted.pd
+00005210: 660a 6060 600a 5468 6520 6041 6c6c 5f70  f.```.The `All_p
+00005220: 6169 7273 2e6b 732e 6e6f 6465 2e77 6569  airs.ks.node.wei
+00005230: 6768 7465 642e 7064 6660 2069 7320 7468  ghted.pdf` is th
+00005240: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+00005250: 3e20 706c 6f74 206f 6620 616c 6c20 7370  > plot of all sp
+00005260: 6563 6965 7320 7061 6972 732e 2054 6865  ecies pairs. The
+00005270: 2060 466f 6375 735f 7369 7374 6572 5f70   `Focus_sister_p
+00005280: 6169 7273 2e6b 732e 6e6f 6465 2e77 6569  airs.ks.node.wei
+00005290: 6768 7465 642e 7064 6660 2069 7320 7468  ghted.pdf` is th
+000052a0: 6520 2a4b 2a3c 7375 623e 533c 2f73 7562  e *K*<sub>S</sub
+000052b0: 3e20 706c 6f74 206f 6620 616c 6c20 666f  > plot of all fo
+000052c0: 6361 6c2d 7369 7374 6572 2073 7065 6369  cal-sister speci
+000052d0: 6573 2070 6169 7273 2e20 5468 6520 6067  es pairs. The `g
+000052e0: 6c6f 6261 6c5f 4d52 4248 2e74 7376 2e6b  lobal_MRBH.tsv.k
+000052f0: 732e 7473 762e 6b73 642e 7064 662f 7376  s.tsv.ksd.pdf/sv
+00005300: 6760 2069 7320 7468 6520 2a4b 2a3c 7375  g` is the *K*<su
+00005310: 623e 533c 2f73 7562 3e20 706c 6f74 206f  b>S</sub> plot o
+00005320: 6620 7468 6520 6461 7461 6669 6c65 2060  f the datafile `
+00005330: 676c 6f62 616c 5f4d 5242 482e 7473 762e  global_MRBH.tsv.
+00005340: 6b73 2e74 7376 602e 2054 6865 2060 4d69  ks.tsv`. The `Mi
+00005350: 7865 642e 6b73 2e41 7175 696c 6567 6961  xed.ks.Aquilegia
+00005360: 5f63 6f65 7275 6c65 612e 6e6f 6465 2e77  _coerulea.node.w
+00005370: 6569 6768 7465 642e 7064 6660 2069 7320  eighted.pdf` is 
+00005380: 7468 6520 2a2a 6669 6e61 6c20 7265 7375  the **final resu
+00005390: 6c74 2a2a 206f 6620 7261 7465 2063 6f72  lt** of rate cor
+000053a0: 7265 6374 696f 6e20 2877 6974 6820 6d69  rection (with mi
+000053b0: 7874 7572 6520 6d6f 6465 6c69 6e67 2072  xture modeling r
+000053c0: 6573 756c 7473 2069 6620 7365 7429 2e20  esults if set). 
+000053d0: 5468 6520 6073 7061 6972 2e63 6f72 7265  The `spair.corre
+000053e0: 6374 6564 2e6b 732e 696e 666f 2e74 7376  cted.ks.info.tsv
+000053f0: 6020 616e 6420 6073 7061 6972 2e6f 7269  ` and `spair.ori
+00005400: 6769 6e61 6c2e 6b73 2e69 6e66 6f2e 7473  ginal.ks.info.ts
+00005410: 7660 2064 6f63 756d 656e 7420 7468 6520  v` document the 
+00005420: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00005430: 696e 666f 726d 6174 696f 6e20 6f66 2061  information of a
+00005440: 6c6c 2073 7065 6369 6573 2070 6169 7273  ll species pairs
+00005450: 2062 6566 6f72 6520 616e 6420 6166 7465   before and afte
+00005460: 7220 7261 7465 2063 6f72 7265 6374 696f  r rate correctio
+00005470: 6e2e 2054 6865 2073 7562 666f 6c64 6572  n. The subfolder
+00005480: 2060 5369 6d70 6c65 5f4b 735f 4469 7374   `Simple_Ks_Dist
+00005490: 7269 6275 7469 6f6e 7360 2063 6f6e 7461  ributions` conta
+000054a0: 696e 7320 7468 6520 7369 6e67 6c65 202a  ins the single *
+000054b0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2070  K*<sub>S</sub> p
+000054c0: 6c6f 7473 206f 6620 616c 6c20 7370 6563  lots of all spec
+000054d0: 6965 7320 7061 6972 732e 0a0a 2323 2320  ies pairs...### 
+000054e0: 5069 7065 6c69 6e65 2036 0a2a 2a43 6f6e  Pipeline 6.**Con
+000054f0: 7374 7275 6374 696f 6e20 6f66 2061 6765  struction of age
+00005500: 2064 6973 7472 6962 7574 696f 6e20 7769   distribution wi
+00005510: 7468 2063 6f6c 6c69 6e65 6172 6974 7920  th collinearity 
+00005520: 616e 6420 5747 4420 6461 7469 6e67 2a2a  and WGD dating**
+00005530: 0a60 6060 0a77 6764 2064 6d64 2041 7175  .```.wgd dmd Aqu
+00005540: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+00005550: 2d6f 2077 6764 5f64 6d64 0a77 6764 206b  -o wgd_dmd.wgd k
+00005560: 7364 2077 6764 5f64 6d64 2f41 7175 696c  sd wgd_dmd/Aquil
+00005570: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00005580: 7620 4171 7569 6c65 6769 615f 636f 6572  v Aquilegia_coer
+00005590: 756c 6561 202d 6f20 7767 645f 6b73 640a  ulea -o wgd_ksd.
+000055a0: 7767 6420 7379 6e20 2d66 206d 524e 4120  wgd syn -f mRNA 
+000055b0: 2d61 204e 616d 6520 7767 645f 646d 642f  -a Name wgd_dmd/
+000055c0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000055d0: 6561 2e74 7376 2041 7175 696c 6567 6961  ea.tsv Aquilegia
+000055e0: 5f63 6f65 7275 6c65 612e 6766 6633 202d  _coerulea.gff3 -
+000055f0: 6b73 2077 6764 5f6b 7364 2f41 7175 696c  ks wgd_ksd/Aquil
+00005600: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00005610: 762e 6b73 2e74 7376 202d 6f20 7767 645f  v.ks.tsv -o wgd_
+00005620: 7379 6e0a 7767 6420 7065 616b 202d 2d68  syn.wgd peak --h
+00005630: 6575 7269 7374 6963 2077 6764 5f6b 7364  euristic wgd_ksd
+00005640: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00005650: 6c65 612e 7473 762e 6b73 2e74 7376 202d  lea.tsv.ks.tsv -
+00005660: 6170 2077 6764 5f73 796e 2f69 6164 686f  ap wgd_syn/iadho
+00005670: 7265 2d6f 7574 2f61 6e63 686f 7270 6f69  re-out/anchorpoi
+00005680: 6e74 732e 7478 7420 2d73 6d20 7767 645f  nts.txt -sm wgd_
+00005690: 7379 6e2f 6961 6468 6f72 652d 6f75 742f  syn/iadhore-out/
+000056a0: 7365 676d 656e 7473 2e74 7874 202d 6c65  segments.txt -le
+000056b0: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+000056c0: 2d6f 7574 2f6c 6973 745f 656c 656d 656e  -out/list_elemen
+000056d0: 7473 2e74 7874 202d 6d70 2077 6764 5f73  ts.txt -mp wgd_s
+000056e0: 796e 2f69 6164 686f 7265 2d6f 7574 2f6d  yn/iadhore-out/m
+000056f0: 756c 7469 706c 6963 6f6e 5f70 6169 7273  ultiplicon_pairs
+00005700: 2e74 7874 202d 6e20 3120 3420 2d6b 6320  .txt -n 1 4 -kc 
+00005710: 3320 2d6f 2077 6764 5f70 6561 6b0a 7767  3 -o wgd_peak.wg
+00005720: 6420 646d 6420 2d66 2041 7175 696c 6567  d dmd -f Aquileg
+00005730: 6961 5f63 6f65 7275 6c65 6120 2d61 7020  ia_coerulea -ap 
+00005740: 7767 645f 7065 616b 2f41 6e63 686f 724b  wgd_peak/AnchorK
+00005750: 735f 4669 6e64 5065 616b 2f41 7175 696c  s_FindPeak/Aquil
+00005760: 6567 6961 5f63 6f65 7275 6c65 612e 7473  egia_coerulea.ts
+00005770: 762e 6b73 2e74 7376 5f39 3525 4349 5f41  v.ks.tsv_95%CI_A
+00005780: 505f 666f 725f 6461 7469 6e67 5f77 6569  P_for_dating_wei
+00005790: 6768 7465 645f 666f 726d 6174 2e74 7376  ghted_format.tsv
+000057a0: 202d 6f20 7767 645f 646d 645f 6f72 7468   -o wgd_dmd_orth
+000057b0: 6f20 506f 7461 6d6f 6765 746f 6e5f 6163  o Potamogeton_ac
+000057c0: 7574 6966 6f6c 6975 7320 5370 6972 6f64  utifolius Spirod
+000057d0: 656c 615f 696e 7465 726d 6564 6961 2041  ela_intermedia A
+000057e0: 6d6f 7270 686f 7068 616c 6c75 735f 6b6f  morphophallus_ko
+000057f0: 6e6a 6163 2041 6361 6e74 686f 6368 6c61  njac Acanthochla
+00005800: 6d79 735f 6272 6163 7465 6174 6120 4469  mys_bracteata Di
+00005810: 6f73 636f 7265 615f 616c 6174 6120 4469  oscorea_alata Di
+00005820: 6f73 636f 7265 615f 726f 7475 6e64 6174  oscorea_rotundat
+00005830: 6120 4163 6f72 7573 5f61 6d65 7269 6361  a Acorus_america
+00005840: 6e75 7320 4163 6f72 7573 5f74 6174 6172  nus Acorus_tatar
+00005850: 696e 6f77 6969 2054 6574 7261 6365 6e74  inowii Tetracent
+00005860: 726f 6e5f 7369 6e65 6e73 6520 5472 6f63  ron_sinense Troc
+00005870: 686f 6465 6e64 726f 6e5f 6172 616c 696f  hodendron_aralio
+00005880: 6964 6573 2042 7578 7573 5f61 7573 7472  ides Buxus_austr
+00005890: 6f79 756e 6e61 6e65 6e73 6973 2042 7578  oyunnanensis Bux
+000058a0: 7573 5f73 696e 6963 6120 4e65 6c75 6d62  us_sinica Nelumb
+000058b0: 6f5f 6e75 6369 6665 7261 2054 656c 6f70  o_nucifera Telop
+000058c0: 6561 5f73 7065 6369 6f73 6973 7369 6d61  ea_speciosissima
+000058d0: 2050 726f 7465 615f 6379 6e61 726f 6964   Protea_cynaroid
+000058e0: 6573 2041 7175 696c 6567 6961 5f63 6f65  es Aquilegia_coe
+000058f0: 7275 6c65 610a 7767 6420 666f 6375 7320  rulea.wgd focus 
+00005900: 2d2d 7072 6f74 636f 6364 6174 696e 6720  --protcocdating 
+00005910: 2d2d 6161 6d6f 6465 6c20 6c67 2077 6764  --aamodel lg wgd
+00005920: 5f64 6d64 5f6f 7274 686f 2f6d 6572 6765  _dmd_ortho/merge
+00005930: 5f66 6f63 7573 5f61 702e 7473 7620 2d73  _focus_ap.tsv -s
+00005940: 7020 6461 7469 6e67 5f74 7265 652e 6e77  p dating_tree.nw
+00005950: 202d 6f20 7767 645f 6461 7469 6e67 202d   -o wgd_dating -
+00005960: 6420 6d63 6d63 7472 6565 202d 6473 2027  d mcmctree -ds '
+00005970: 6275 726e 696e 203d 2032 3030 3027 202d  burnin = 2000' -
+00005980: 6473 2027 7361 6d70 6672 6571 203d 2031  ds 'sampfreq = 1
+00005990: 3030 3027 202d 6473 2027 6e73 616d 706c  000' -ds 'nsampl
+000059a0: 6520 3d20 3230 3030 3027 2050 6f74 616d  e = 20000' Potam
+000059b0: 6f67 6574 6f6e 5f61 6375 7469 666f 6c69  ogeton_acutifoli
+000059c0: 7573 2053 7069 726f 6465 6c61 5f69 6e74  us Spirodela_int
+000059d0: 6572 6d65 6469 6120 416d 6f72 7068 6f70  ermedia Amorphop
+000059e0: 6861 6c6c 7573 5f6b 6f6e 6a61 6320 4163  hallus_konjac Ac
+000059f0: 616e 7468 6f63 686c 616d 7973 5f62 7261  anthochlamys_bra
+00005a00: 6374 6561 7461 2044 696f 7363 6f72 6561  cteata Dioscorea
+00005a10: 5f61 6c61 7461 2044 696f 7363 6f72 6561  _alata Dioscorea
+00005a20: 5f72 6f74 756e 6461 7461 2041 636f 7275  _rotundata Acoru
+00005a30: 735f 616d 6572 6963 616e 7573 2041 636f  s_americanus Aco
+00005a40: 7275 735f 7461 7461 7269 6e6f 7769 6920  rus_tatarinowii 
+00005a50: 5465 7472 6163 656e 7472 6f6e 5f73 696e  Tetracentron_sin
+00005a60: 656e 7365 2054 726f 6368 6f64 656e 6472  ense Trochodendr
+00005a70: 6f6e 5f61 7261 6c69 6f69 6465 7320 4275  on_aralioides Bu
+00005a80: 7875 735f 6175 7374 726f 7975 6e6e 616e  xus_austroyunnan
+00005a90: 656e 7369 7320 4275 7875 735f 7369 6e69  ensis Buxus_sini
+00005aa0: 6361 204e 656c 756d 626f 5f6e 7563 6966  ca Nelumbo_nucif
+00005ab0: 6572 6120 5465 6c6f 7065 615f 7370 6563  era Telopea_spec
+00005ac0: 696f 7369 7373 696d 6120 5072 6f74 6561  iosissima Protea
+00005ad0: 5f63 796e 6172 6f69 6465 7320 4171 7569  _cynaroides Aqui
+00005ae0: 6c65 6769 615f 636f 6572 756c 6561 0a60  legia_coerulea.`
+00005af0: 6060 0a60 6060 0a23 2077 6861 7420 7769  ``.```.# what wi
+00005b00: 6c6c 2062 6520 696e 2074 6865 2077 6764  ll be in the wgd
+00005b10: 5f64 6174 696e 6720 7265 7375 6c74 2064  _dating result d
+00005b20: 6972 6563 746f 7279 0a2d 7767 645f 6461  irectory.-wgd_da
+00005b30: 7469 6e67 0a2d 2d43 6f6e 6361 7465 6e61  ting.--Concatena
+00005b40: 7465 642e 7061 6c6e 0a2d 2d43 6f6e 6361  ted.paln.--Conca
+00005b50: 7465 6e61 7465 642e 7061 6c6e 2e70 616d  tenated.paln.pam
+00005b60: 6c0a 2d2d 4732 532e 4d61 700a 2d2d 4746  l.--G2S.Map.--GF
+00005b70: 3030 3030 3030 3031 2e70 616c 6e0a 2d2d  00000001.paln.--
+00005b80: 2e2e 0a2d 2d47 4630 3030 3030 3138 372e  ...--GF00000187.
+00005b90: 7061 6c6e 0a2d 2d6d 636d 6374 7265 650a  paln.--mcmctree.
+00005ba0: 2d2d 2d43 6f6e 6361 7465 6e61 7465 640a  ---Concatenated.
+00005bb0: 2d2d 2d2d 7065 700a 2d2d 2d2d 2d43 6f6e  ----pep.-----Con
+00005bc0: 6361 7465 6e61 7465 642e 7061 6c6e 2e70  catenated.paln.p
+00005bd0: 616d 6c0a 2d2d 2d2d 2d64 6174 696e 675f  aml.-----dating_
+00005be0: 7472 6565 2e6e 770a 2d2d 2d2d 2d46 6967  tree.nw.-----Fig
+00005bf0: 5472 6565 2e74 7265 0a2d 2d2d 2d2d 696e  Tree.tre.-----in
+00005c00: 2e42 560a 2d2d 2d2d 2d6c 672e 6461 740a  .BV.-----lg.dat.
+00005c10: 2d2d 2d2d 2d6c 6e66 0a2d 2d2d 2d2d 6d63  -----lnf.-----mc
+00005c20: 6d63 7472 6565 2e63 7472 6c0a 2d2d 2d2d  mctree.ctrl.----
+00005c30: 2d6d 636d 6374 7265 652e 6f75 740a 2d2d  -mcmctree.out.--
+00005c40: 2d2d 2d6d 636d 632e 7478 740a 2d2d 2d2d  ---mcmc.txt.----
+00005c50: 2d72 6174 6573 0a2d 2d2d 2d2d 7273 740a  -rates.-----rst.
+00005c60: 2d2d 2d2d 2d72 7374 310a 2d2d 2d2d 2d72  -----rst1.-----r
+00005c70: 7562 0a2d 2d2d 2d2d 746d 7030 3030 312e  ub.-----tmp0001.
+00005c80: 6374 6c0a 2d2d 2d2d 2d74 6d70 3030 3031  ctl.-----tmp0001
+00005c90: 2e6f 7574 0a2d 2d2d 2d2d 746d 7030 3030  .out.-----tmp000
+00005ca0: 312e 7472 6565 730a 2d2d 2d2d 2d74 6d70  1.trees.-----tmp
+00005cb0: 3030 3031 2e74 7874 0a60 6060 0a54 6865  0001.txt.```.The
+00005cc0: 2060 436f 6e63 6174 656e 6174 6564 2e70   `Concatenated.p
+00005cd0: 616c 6e60 2061 6e64 2060 436f 6e63 6174  aln` and `Concat
+00005ce0: 656e 6174 6564 2e70 616c 6e2e 7061 6d6c  enated.paln.paml
+00005cf0: 6020 6172 6520 7468 6520 636f 6e63 6174  ` are the concat
+00005d00: 656e 6174 6564 2070 726f 7465 696e 2061  enated protein a
+00005d10: 6c69 676e 6d65 6e74 7320 696e 2066 6173  lignments in fas
+00005d20: 7461 2061 6e64 2070 616d 6c20 666f 726d  ta and paml form
+00005d30: 6174 2e20 5468 6520 6047 3253 2e4d 6170  at. The `G2S.Map
+00005d40: 6020 6973 2074 6865 206d 6170 2062 6574  ` is the map bet
+00005d50: 7765 656e 2067 656e 6520 616e 6420 7370  ween gene and sp
+00005d60: 6563 6965 7320 6e61 6d65 732e 2054 6865  ecies names. The
+00005d70: 2060 4746 3030 3030 3030 3031 2e70 616c   `GF00000001.pal
+00005d80: 6e60 2c2e 2e20 616e 6420 6047 4630 3030  n`,.. and `GF000
+00005d90: 3030 3138 372e 7061 6c6e 6020 6172 6520  00187.paln` are 
+00005da0: 7468 6520 7072 6f74 6569 6e20 616c 6967  the protein alig
+00005db0: 6e6d 656e 7473 2066 6f72 2065 6163 6820  nments for each 
+00005dc0: 6765 6e65 2066 616d 696c 792e 2054 6865  gene family. The
+00005dd0: 2060 6d63 6d63 7472 6565 6020 7375 6266   `mcmctree` subf
+00005de0: 6f6c 6465 7220 636f 6e74 6169 6e73 2074  older contains t
+00005df0: 6865 2064 6174 696e 6720 7265 7375 6c74  he dating result
+00005e00: 7320 666f 7220 7468 6520 636f 6e63 6174  s for the concat
+00005e10: 656e 6174 6564 2066 616d 696c 7920 2861  enated family (a
+00005e20: 6e64 2070 6572 2067 656e 6520 6661 6d69  nd per gene fami
+00005e30: 6c79 2069 6620 7365 7429 2e20 5468 6520  ly if set). The 
+00005e40: 6465 6570 6572 2060 436f 6e63 6174 656e  deeper `Concaten
+00005e50: 6174 6564 6020 7375 6266 6f6c 6465 7220  ated` subfolder 
+00005e60: 636f 6e74 6169 6e73 2074 6865 2064 6174  contains the dat
+00005e70: 696e 6720 7265 7375 6c74 7320 6f66 2063  ing results of c
+00005e80: 6f6e 6361 7465 6e61 7465 6420 7072 6f74  oncatenated prot
+00005e90: 6569 6e20 616c 6967 6e6d 656e 7420 286f  ein alignment (o
+00005ea0: 7220 6e75 636c 656f 7469 6465 2061 6c69  r nucleotide ali
+00005eb0: 676e 6d65 6e74 2069 6620 7365 7429 2e20  gnment if set). 
+00005ec0: 5468 6520 6465 6570 6573 7420 6070 6570  The deepest `pep
+00005ed0: 6020 7375 6266 6f6c 6465 7220 636f 6e74  ` subfolder cont
+00005ee0: 6169 6e73 2074 6865 2066 696e 616c 2064  ains the final d
+00005ef0: 6174 696e 6720 7265 7375 6c74 7320 286f  ating results (o
+00005f00: 6620 636f 6e63 6174 656e 6174 6564 2070  f concatenated p
+00005f10: 726f 7465 696e 2061 6c69 676e 6d65 6e74  rotein alignment
+00005f20: 2069 6e20 7468 6973 2063 6173 6529 2e20   in this case). 
+00005f30: 506c 6561 7365 2072 6566 6572 2074 6f20  Please refer to 
+00005f40: 606d 636d 6374 7265 6560 206d 616e 7561  `mcmctree` manua
+00005f50: 6c20 666f 7220 6465 7461 696c 6564 2064  l for detailed d
+00005f60: 6573 6372 6970 7469 6f6e 206f 6620 6561  escription of ea
+00005f70: 6368 2066 696c 6520 7072 6f64 7563 6564  ch file produced
+00005f80: 2062 7920 606d 636d 6374 7265 6560 2e20   by `mcmctree`. 
+00005f90: 5468 6520 696d 706f 7274 616e 7420 7265  The important re
+00005fa0: 7375 6c74 2066 696c 6573 2061 7265 2060  sult files are `
+00005fb0: 4669 6754 7265 652e 7472 6560 2c20 606d  FigTree.tre`, `m
+00005fc0: 636d 6374 7265 652e 6f75 7460 2061 6e64  cmctree.out` and
+00005fd0: 2060 6d63 6d63 2e74 7874 6020 7768 6963   `mcmc.txt` whic
+00005fe0: 6820 646f 6375 6d65 6e74 2074 6865 2066  h document the f
+00005ff0: 696e 616c 2064 6174 6520 6573 7469 6d61  inal date estima
+00006000: 7469 6f6e 2c20 6c6f 6720 696e 666f 726d  tion, log inform
+00006010: 6174 696f 6e20 616e 6420 706f 7374 6572  ation and poster
+00006020: 696f 7220 7361 6d70 6c65 7320 666f 7220  ior samples for 
+00006030: 6561 6368 206e 6f64 6520 7265 7370 6563  each node respec
+00006040: 7469 7665 6c79 2e0a 200a 2323 2050 6172  tively.. .## Par
+00006050: 616d 6574 6572 730a 0a54 6865 7265 2061  ameters..There a
+00006060: 7265 2037 206d 6169 6e20 7072 6f67 7261  re 7 main progra
+00006070: 6d73 2069 6e20 6077 6764 2076 3260 3a20  ms in `wgd v2`: 
+00006080: 6064 6d64 602c 6066 6f63 7573 602c 606b  `dmd`,`focus`,`k
+00006090: 7364 602c 606d 6978 602c 6070 6561 6b60  sd`,`mix`,`peak`
+000060a0: 2c60 7379 6e60 2c60 7669 7a60 2e20 4865  ,`syn`,`viz`. He
+000060b0: 7265 6166 7465 7220 7765 2077 696c 6c20  reafter we will 
+000060c0: 7072 6f76 6964 6520 6120 6465 7461 696c  provide a detail
+000060d0: 6564 2065 6c75 6369 6461 7469 6f6e 206f  ed elucidation o
+000060e0: 6e20 6561 6368 206f 6620 7468 6520 7072  n each of the pr
+000060f0: 6f67 7261 6d20 616e 6420 6974 7320 6173  ogram and its as
+00006100: 736f 6369 6174 6564 2070 6172 616d 6574  sociated paramet
+00006110: 6572 732e 2050 6c65 6173 6520 7265 6665  ers. Please refe
+00006120: 7220 746f 2074 6865 205b 5573 6167 655d  r to the [Usage]
+00006130: 2823 7573 6167 6529 2066 6f72 2074 6865  (#usage) for the
+00006140: 2073 6365 6e61 7269 6f73 2074 6f20 7768   scenarios to wh
+00006150: 6963 6820 6561 6368 2070 6172 616d 6574  ich each paramet
+00006160: 6572 2061 7070 6c69 6573 2e0a 0a54 6865  er applies...The
+00006170: 2070 726f 6772 616d 2060 7767 6420 646d   program `wgd dm
+00006180: 6460 2063 616e 2072 6561 6c69 7a65 2074  d` can realize t
+00006190: 6865 2064 656c 696e 6561 7469 6f6e 206f  he delineation o
+000061a0: 6620 7768 6f6c 6520 7061 7261 6e6f 6d65  f whole paranome
+000061b0: 2c20 5242 4873 2028 5265 6369 7072 6f63  , RBHs (Reciproc
+000061c0: 616c 2042 6573 7420 4869 7473 292c 204d  al Best Hits), M
+000061d0: 5242 4873 2028 4d75 6c74 6970 6c65 2052  RBHs (Multiple R
+000061e0: 6563 6970 726f 6361 6c20 4265 7374 2048  eciprocal Best H
+000061f0: 6974 7329 2c20 6f72 7468 6f67 726f 7570  its), orthogroup
+00006200: 7320 616e 6420 736f 6d65 206f 7468 6572  s and some other
+00006210: 206f 7274 686f 6772 6f75 702d 7265 6c61   orthogroup-rela
+00006220: 7465 6420 6675 6e63 7469 6f6e 732c 2069  ted functions, i
+00006230: 6e63 6c75 6469 6e67 2063 6972 6375 6d73  ncluding circums
+00006240: 6372 6970 7469 6f6e 206f 6620 6e65 7374  cription of nest
+00006250: 6564 2073 696e 676c 652d 636f 7079 206f  ed single-copy o
+00006260: 7274 686f 6772 6f75 7073 2028 4e53 4f47  rthogroups (NSOG
+00006270: 7329 2c20 756e 6269 6173 6564 2074 6573  s), unbiased tes
+00006280: 7420 6f66 2073 696e 676c 652d 636f 7079  t of single-copy
+00006290: 206f 7274 686f 6772 6f75 7073 2028 534f   orthogroups (SO
+000062a0: 4773 2920 6f76 6572 206d 6973 7369 6e67  Gs) over missing
+000062b0: 2069 6e70 6172 616c 6f67 732c 2063 6f6e   inparalogs, con
+000062c0: 7374 7275 6374 696f 6e20 6f66 2042 5553  struction of BUS
+000062d0: 434f 2d67 7569 6465 6420 7369 6e67 6c65  CO-guided single
+000062e0: 2d63 6f70 7920 6f72 7468 6f67 726f 7570  -copy orthogroup
+000062f0: 7320 2853 4f47 7329 2c61 6e64 2074 6865  s (SOGs),and the
+00006300: 2063 6f6c 6c69 6e65 6172 2063 6f61 6c65   collinear coale
+00006310: 7363 656e 6365 2069 6e66 6572 656e 6365  scence inference
+00006320: 206f 6620 7068 796c 6f67 656e 792e 0a60   of phylogeny..`
+00006330: 6060 0a77 6764 2064 6d64 2073 6571 7565  ``.wgd dmd seque
+00006340: 6e63 6573 2028 6f70 7469 6f6e 290a 2d2d  nces (option).--
+00006350: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006360: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006370: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006380: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00006390: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
+000063a0: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
+000063b0: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
+000063c0: 792c 2064 6566 6175 6c74 2077 6764 5f64  y, default wgd_d
+000063d0: 6d64 0a2d 742c 202d 2d74 6d70 6469 722c  md.-t, --tmpdir,
+000063e0: 2074 6865 2074 656d 706f 7261 7279 2077   the temporary w
+000063f0: 6f72 6b69 6e67 2064 6972 6563 746f 7279  orking directory
+00006400: 2c20 6465 6661 756c 7420 4e6f 6e65 2c20  , default None, 
+00006410: 6966 204e 6f6e 6520 7761 7320 6769 7665  if None was give
+00006420: 6e2c 2074 6865 2074 6d70 6469 7220 776f  n, the tmpdir wo
+00006430: 756c 6420 6265 2061 7373 6967 6e65 6420  uld be assigned 
+00006440: 7261 6e64 6f6d 206e 616d 6573 2069 6e20  random names in 
+00006450: 6375 7272 656e 7420 6469 7265 6374 6f72  current director
+00006460: 7920 616e 6420 6175 746f 6d61 7465 6c79  y and automately
+00006470: 2072 656d 6f76 6564 2061 7420 7468 6520   removed at the 
+00006480: 636f 6d70 6c65 7469 6f6e 206f 6620 7072  completion of pr
+00006490: 6f67 7261 6d2c 2065 6c73 6520 7468 6520  ogram, else the 
+000064a0: 746d 7064 6972 2077 6f75 6c64 2062 6520  tmpdir would be 
+000064b0: 6b65 7074 0a2d 702c 202d 2d70 726f 742c  kept.-p, --prot,
+000064c0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+000064d0: 6574 6865 7220 7573 696e 6720 7072 6f74  ether using prot
+000064e0: 6569 6e20 6f72 206e 7563 6c65 6f74 6964  ein or nucleotid
+000064f0: 6520 7365 7175 656e 6365 730a 2d63 2c20  e sequences.-c, 
+00006500: 2d2d 6373 636f 7265 2c20 7468 6520 632d  --cscore, the c-
+00006510: 7363 6f72 6520 746f 2072 6573 7472 6963  score to restric
+00006520: 7420 7468 6520 686f 6d6f 6c6f 6720 7369  t the homolog si
+00006530: 6d69 6c61 7269 7479 206f 6620 4d52 4248  milarity of MRBH
+00006540: 732c 2064 6566 6175 6c74 204e 6f6e 652c  s, default None,
+00006550: 2069 6620 4e6f 6e65 2077 6173 2067 6976   if None was giv
+00006560: 656e 2c20 7468 6520 632d 7363 6f72 6520  en, the c-score 
+00006570: 6675 6e63 696f 6e20 776f 756c 646e 2774  funcion wouldn't
+00006580: 2062 6520 6163 7469 7661 7465 642c 2065   be activated, e
+00006590: 6c73 6520 6578 7065 6374 696e 6720 6120  lse expecting a 
+000065a0: 6465 6369 6d61 6c20 7769 7468 696e 2074  decimal within t
+000065b0: 6865 2072 616e 6765 206f 6620 3020 616e  he range of 0 an
+000065c0: 6420 310a 2d49 2c20 2d2d 696e 666c 6174  d 1.-I, --inflat
+000065d0: 696f 6e2c 2074 6865 2069 6e66 6c61 7469  ion, the inflati
+000065e0: 6f6e 2066 6163 746f 7220 666f 7220 4d43  on factor for MC
+000065f0: 4c20 7072 6f67 7261 6d2c 2064 6566 6175  L program, defau
+00006600: 6c74 2032 2e30 2c20 7769 7468 2068 6967  lt 2.0, with hig
+00006610: 6865 7220 7661 6c75 6520 6c65 6164 696e  her value leadin
+00006620: 6720 746f 206d 6f72 6520 6275 7420 736d  g to more but sm
+00006630: 616c 6c65 7220 636c 7573 7465 7273 0a2d  aller clusters.-
+00006640: 652c 202d 2d65 7661 6c2c 2074 6865 2065  e, --eval, the e
+00006650: 2d76 616c 7565 2063 7574 2d6f 6666 2066  -value cut-off f
+00006660: 6f72 2073 696d 696c 6172 6974 7920 696e  or similarity in
+00006670: 2064 6961 6d6f 6e64 2061 6e64 2f6f 7220   diamond and/or 
+00006680: 686d 6d65 722c 2064 6566 6175 6c74 2031  hmmer, default 1
+00006690: 652d 3130 0a2d 2d74 6f5f 7374 6f70 2c20  e-10.--to_stop, 
+000066a0: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+000066b0: 7468 6572 2074 6f20 7472 616e 736c 6174  ther to translat
+000066c0: 6520 7468 726f 7567 6820 5354 4f50 2063  e through STOP c
+000066d0: 6f64 6f6e 732c 2069 6620 7468 6520 666c  odons, if the fl
+000066e0: 6167 2077 6173 2073 6574 2c20 7472 616e  ag was set, tran
+000066f0: 736c 6174 696f 6e20 7769 6c6c 2062 6520  slation will be 
+00006700: 7465 726d 696e 6174 6564 2061 7420 7468  terminated at th
+00006710: 6520 6669 7273 7420 696e 2d66 7261 6d65  e first in-frame
+00006720: 2073 746f 7020 636f 646f 6e2c 2065 6c73   stop codon, els
+00006730: 6520 6120 6675 6c6c 2074 7261 6e73 6c61  e a full transla
+00006740: 7469 6f6e 2063 6f6e 7469 6e75 696e 6720  tion continuing 
+00006750: 6f6e 2070 6173 7369 6e67 2061 6e79 2073  on passing any s
+00006760: 746f 7020 636f 646f 6e73 2077 6f75 6c64  top codons would
+00006770: 2062 6520 696e 6974 6961 7465 640a 2d2d   be initiated.--
+00006780: 6364 732c 2066 6c61 6720 6f70 7469 6f6e  cds, flag option
+00006790: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
+000067a0: 7920 7472 616e 736c 6174 6520 7468 6520  y translate the 
+000067b0: 636f 6d70 6c65 7465 2043 4453 2074 6861  complete CDS tha
+000067c0: 7420 7374 6172 7473 2077 6974 6820 6120  t starts with a 
+000067d0: 7661 6c69 6420 7374 6172 7420 636f 646f  valid start codo
+000067e0: 6e20 616e 6420 6f6e 6c79 2063 6f6e 7461  n and only conta
+000067f0: 696e 7320 6120 7369 6e67 6c65 2069 6e2d  ins a single in-
+00006800: 6672 616d 6520 7374 6f70 2063 6f64 6f6e  frame stop codon
+00006810: 2061 7420 7468 6520 656e 6420 616e 6420   at the end and 
+00006820: 6d75 7374 2062 6520 6469 7669 6461 626c  must be dividabl
+00006830: 6520 6279 2074 6872 6565 2c20 6966 2074  e by three, if t
+00006840: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00006850: 206f 6e6c 7920 7468 6520 636f 6d70 6c65   only the comple
+00006860: 7465 2043 4453 2077 6f75 6c64 2062 6520  te CDS would be 
+00006870: 7472 616e 736c 6174 6564 0a2d 662c 202d  translated.-f, -
+00006880: 2d66 6f63 7573 2c20 7468 6520 7370 6563  -focus, the spec
+00006890: 6965 7320 746f 2062 6520 6d65 7267 6564  ies to be merged
+000068a0: 206f 6e20 6c6f 6361 6c20 4d52 4248 732c   on local MRBHs,
+000068b0: 2064 6566 6175 6c74 204e 6f6e 652c 2069   default None, i
+000068c0: 6620 4e6f 6e65 2077 6173 2067 6976 656e  f None was given
+000068d0: 2c20 7468 6520 6c6f 6361 6c20 4d52 4248  , the local MRBH
+000068e0: 7320 776f 756c 646e 2774 2062 6520 696e  s wouldn't be in
+000068f0: 6665 7272 6564 0a2d 6170 2c20 2d2d 616e  ferred.-ap, --an
+00006900: 6368 6f72 706f 696e 7473 2c20 7468 6520  chorpoints, the 
+00006910: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
+00006920: 7461 2066 696c 6520 6672 6f6d 2069 2d61  ta file from i-a
+00006930: 6468 6f72 6520 666f 7220 636f 6e73 7472  dhore for constr
+00006940: 7563 7469 6e67 2074 6865 206f 7274 686f  ucting the ortho
+00006950: 6772 6f75 7073 2077 6974 6820 616e 6368  groups with anch
+00006960: 6f72 2070 6169 7273 2c20 6465 6661 756c  or pairs, defaul
+00006970: 7420 4e6f 6e65 0a2d 736d 2c20 2d2d 7365  t None.-sm, --se
+00006980: 676d 656e 7473 2c20 7468 6520 7365 676d  gments, the segm
+00006990: 656e 7473 2064 6174 6166 696c 6520 7573  ents datafile us
+000069a0: 6564 2069 6e20 636f 6c6c 696e 6561 7220  ed in collinear 
+000069b0: 636f 616c 6573 6365 6e63 6520 616e 616c  coalescence anal
+000069c0: 7973 6973 2069 6620 696e 6974 6961 7465  ysis if initiate
+000069d0: 642c 2064 6566 6175 6c74 204e 6f6e 650a  d, default None.
+000069e0: 2d6c 652c 202d 2d6c 6973 7465 6c65 6d65  -le, --listeleme
+000069f0: 6e74 732c 2074 6865 206c 6973 7473 6567  nts, the listseg
+00006a00: 6d65 6e74 7320 6461 7461 2066 696c 6520  ments data file 
+00006a10: 7573 6564 2069 6e20 636f 6c6c 696e 6561  used in collinea
+00006a20: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
+00006a30: 616c 7973 6973 2069 6620 696e 6974 6961  alysis if initia
+00006a40: 7465 642c 2064 6566 6175 6c74 204e 6f6e  ted, default Non
+00006a50: 650a 2d67 742c 202d 2d67 656e 6574 6162  e.-gt, --genetab
+00006a60: 6c65 2c20 7468 6520 6765 6e65 2074 6162  le, the gene tab
+00006a70: 6c65 2064 6174 6166 696c 6520 7573 6564  le datafile used
+00006a80: 2069 6e20 636f 6c6c 696e 6561 7220 636f   in collinear co
+00006a90: 616c 6573 6365 6e63 6520 616e 616c 7973  alescence analys
+00006aa0: 6973 2069 6620 696e 6974 6961 7465 642c  is if initiated,
+00006ab0: 2064 6566 6175 6c74 204e 6f6e 650a 2d63   default None.-c
+00006ac0: 6f63 2c20 2d2d 636f 6c6c 696e 6561 7263  oc, --collinearc
+00006ad0: 6f61 6c65 7363 656e 6365 2c20 666c 6167  oalescence, flag
+00006ae0: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+00006af0: 2074 6f20 696e 6974 6961 7465 2074 6865   to initiate the
+00006b00: 2063 6f6c 6c69 6e65 6172 2063 6f61 6c65   collinear coale
+00006b10: 7363 656e 6365 2061 6e61 6c79 7369 732c  scence analysis,
+00006b20: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00006b30: 2073 6574 2c20 7468 6520 616e 616c 7973   set, the analys
+00006b40: 6973 2077 6f75 6c64 2062 6520 696e 6974  is would be init
+00006b50: 6961 7465 640a 2d6b 662c 202d 2d6b 6565  iated.-kf, --kee
+00006b60: 7066 6173 7461 2c20 666c 6167 206f 7074  pfasta, flag opt
+00006b70: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00006b80: 6f75 7470 7574 2074 6865 2073 6571 7565  output the seque
+00006b90: 6e63 6520 696e 666f 726d 6174 696f 6e20  nce information 
+00006ba0: 6f66 204d 5242 4873 2c20 6966 2074 6865  of MRBHs, if the
+00006bb0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+00006bc0: 6865 2073 6571 7565 6e63 6573 206f 6620  he sequences of 
+00006bd0: 4d52 4248 7320 776f 756c 6420 6265 2069  MRBHs would be i
+00006be0: 6e20 6f75 7470 7574 0a2d 6b64 2c20 2d2d  n output.-kd, --
+00006bf0: 6b65 6570 6475 706c 6963 6174 6573 2c20  keepduplicates, 
+00006c00: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00006c10: 7468 6572 2074 6f20 616c 6c6f 7720 7468  ther to allow th
+00006c20: 6520 7361 6d65 2067 656e 6520 746f 206f  e same gene to o
+00006c30: 6363 7572 2069 6e20 6469 6666 6572 656e  ccur in differen
+00006c40: 7420 4d52 4248 7320 286f 6e6c 7920 6d65  t MRBHs (only me
+00006c50: 616e 696e 6766 756c 2077 6865 6e20 7468  aningful when th
+00006c60: 6520 6373 636f 7265 2077 6173 2075 7365  e cscore was use
+00006c70: 6429 2c20 6966 2074 6865 2066 6c61 6720  d), if the flag 
+00006c80: 7761 7320 7365 742c 2074 6865 2073 616d  was set, the sam
+00006c90: 6520 6765 6e65 2063 6f75 6c64 2062 6520  e gene could be 
+00006ca0: 6173 7369 676e 6564 2074 6f20 6469 6666  assigned to diff
+00006cb0: 6572 656e 7420 4d52 4248 730a 2d67 6d2c  erent MRBHs.-gm,
+00006cc0: 202d 2d67 6c6f 6261 6c6d 7262 682c 2066   --globalmrbh, f
+00006cd0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00006ce0: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00006cf0: 676c 6f62 616c 204d 5242 4873 2063 6f6e  global MRBHs con
+00006d00: 7374 7275 6374 696f 6e2c 2069 6620 7468  struction, if th
+00006d10: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00006d20: 7468 6520 2d2d 666f 6375 7320 6f70 7469  the --focus opti
+00006d30: 6f6e 2077 6f75 6c64 2062 6520 6967 6e6f  on would be igno
+00006d40: 7265 6420 616e 6420 6f6e 6c79 2067 6c6f  red and only glo
+00006d50: 6261 6c20 4d52 4248 7320 776f 756c 6420  bal MRBHs would 
+00006d60: 6265 2062 7569 6c74 0a2d 6e2c 202d 2d6e  be built.-n, --n
+00006d70: 7468 7265 6164 732c 2074 6865 206e 756d  threads, the num
+00006d80: 6265 7220 6f66 2074 6872 6561 6473 2074  ber of threads t
+00006d90: 6f20 7573 652c 2064 6566 6175 6c74 2034  o use, default 4
+00006da0: 0a2d 6f69 2c20 2d2d 6f72 7468 6f69 6e66  .-oi, --orthoinf
+00006db0: 6572 2c20 666c 6167 206f 7074 696f 6e2c  er, flag option,
+00006dc0: 2077 6865 7468 6572 2074 6f20 696e 6974   whether to init
+00006dd0: 6961 7465 206f 7274 686f 6772 6f75 7020  iate orthogroup 
+00006de0: 696e 6665 726e 6563 652c 2069 6620 7468  infernece, if th
+00006df0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00006e00: 7468 6520 6f72 7468 6f67 726f 7570 2069  the orthogroup i
+00006e10: 6e66 6572 6e65 6365 2070 726f 6772 616d  nfernece program
+00006e20: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00006e30: 7465 640a 2d6f 6f2c 202d 2d6f 6e6c 796f  ted.-oo, --onlyo
+00006e40: 7274 686f 2c20 666c 6167 206f 7074 696f  rtho, flag optio
+00006e50: 6e2c 2077 6865 7468 6572 2074 6f20 6f6e  n, whether to on
+00006e60: 6c79 2063 6f6e 6475 6374 206f 7274 686f  ly conduct ortho
+00006e70: 6772 6f75 7020 696e 6665 726e 6563 652c  group infernece,
+00006e80: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+00006e90: 2073 6574 2c20 6f6e 6c79 2074 6865 206f   set, only the o
+00006ea0: 7274 686f 6772 6f75 7020 696e 6665 726e  rthogroup infern
+00006eb0: 6563 6520 7069 7065 6c69 6e65 2077 6f75  ece pipeline wou
+00006ec0: 6c64 2062 6520 7065 7266 6f72 6d65 6420  ld be performed 
+00006ed0: 7768 696c 6520 7468 6520 6f74 6865 7220  while the other 
+00006ee0: 616e 616c 7973 6973 2077 6f75 6c64 6e27  analysis wouldn'
+00006ef0: 7420 6265 2069 6e69 7469 6174 6564 0a2d  t be initiated.-
+00006f00: 676e 2c20 2d2d 6765 746e 736f 672c 2066  gn, --getnsog, f
+00006f10: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00006f20: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00006f30: 7468 6520 7365 6172 6368 696e 6720 666f  the searching fo
+00006f40: 7220 6e65 7374 6564 2073 696e 676c 652d  r nested single-
+00006f50: 636f 7079 2067 656e 6520 6661 6d69 6c69  copy gene famili
+00006f60: 6573 2028 4e53 4f47 7329 2028 6f6e 6c79  es (NSOGs) (only
+00006f70: 206d 6561 6e69 6e67 6675 6c20 7768 656e   meaningful when
+00006f80: 2074 6865 206f 7274 686f 6772 6f75 7020   the orthogroup 
+00006f90: 696e 6665 726e 6563 6520 7069 7065 6c69  infernece pipeli
+00006fa0: 6e65 2077 6173 2061 6374 6976 6174 6564  ne was activated
+00006fb0: 292c 2069 6620 7468 6520 666c 6167 2077  ), if the flag w
+00006fc0: 6173 2073 6574 2c20 6164 6469 7469 6f6e  as set, addition
+00006fd0: 616c 204e 534f 4773 2061 6e61 6c79 7369  al NSOGs analysi
+00006fe0: 7320 776f 756c 6420 6265 2070 6572 666f  s would be perfo
+00006ff0: 726d 6564 2062 6573 6964 6573 2074 6865  rmed besides the
+00007000: 2062 6173 6963 206f 7274 686f 6772 6f75   basic orthogrou
+00007010: 7020 696e 6665 726e 6563 650a 2d74 7265  p infernece.-tre
+00007020: 652c 202d 2d74 7265 655f 6d65 7468 6f64  e, --tree_method
+00007030: 2c20 7768 6963 6820 6765 6e65 2074 7265  , which gene tre
+00007040: 6520 696e 6665 7265 6e63 6520 7072 6f67  e inference prog
+00007050: 7261 6d20 746f 2069 6e76 6f6b 6520 286f  ram to invoke (o
+00007060: 6e6c 7920 6d65 616e 696e 6766 756c 2077  nly meaningful w
+00007070: 6865 6e20 7468 6520 636f 6c6c 696e 6561  hen the collinea
+00007080: 7220 636f 616c 6573 6365 6e63 652c 2067  r coalescence, g
+00007090: 656e 652d 746f 2d66 616d 696c 7920 6173  ene-to-family as
+000070a0: 7369 676e 6d65 6e74 206f 7220 4e53 4f47  signment or NSOG
+000070b0: 7320 616e 616c 7973 6973 2077 6572 6520  s analysis were 
+000070c0: 6163 7469 7661 7465 6429 2c20 6465 6661  activated), defa
+000070d0: 756c 7420 6661 7374 7472 6565 0a2d 7473  ult fasttree.-ts
+000070e0: 2c20 2d2d 7472 6565 7365 742c 2074 6865  , --treeset, the
+000070f0: 2070 6172 616d 6574 6572 7320 7365 7474   parameters sett
+00007100: 696e 6720 666f 7220 6765 6e65 2074 7265  ing for gene tre
+00007110: 6520 696e 6665 7265 6e63 652c 2064 6566  e inference, def
+00007120: 6175 6c74 204e 6f6e 652c 2074 6869 7320  ault None, this 
+00007130: 6f70 7469 6f6e 2063 616e 2062 6520 7072  option can be pr
+00007140: 6f76 6964 6564 206d 756c 7469 706c 6520  ovided multiple 
+00007150: 7469 6d65 730a 2d6d 632c 202d 2d6d 736f  times.-mc, --mso
+00007160: 6763 7574 2c20 7468 6520 7261 7469 6f20  gcut, the ratio 
+00007170: 6375 746f 6666 2066 6f72 206d 6f73 746c  cutoff for mostl
+00007180: 7920 7369 6e67 6c65 2d63 6f70 7920 6661  y single-copy fa
+00007190: 6d69 6c79 2028 6d65 616e 696e 6766 756c  mily (meaningful
+000071a0: 2077 6865 6e20 6163 7469 7661 7469 6e67   when activating
+000071b0: 2074 6865 206f 7274 686f 6772 6f75 7020   the orthogroup 
+000071c0: 696e 6665 726e 6563 6520 7069 7065 6c69  infernece pipeli
+000071d0: 6e65 2920 616e 6420 7370 6563 6965 7320  ne) and species 
+000071e0: 7265 7072 6573 656e 7461 7469 6f6e 2069  representation i
+000071f0: 6e20 636f 6c6c 696e 6561 7220 636f 616c  n collinear coal
+00007200: 6573 6365 6e63 6520 616e 616c 7973 6973  escence analysis
+00007210: 2c20 6465 6661 756c 7420 302e 382e 0a2d  , default 0.8..-
+00007220: 6761 2c20 2d2d 6765 6e65 6173 7369 676e  ga, --geneassign
+00007230: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00007240: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
+00007250: 7465 2074 6865 2067 656e 652d 746f 2d66  te the gene-to-f
+00007260: 616d 696c 7920 6173 7369 676e 6d65 6e74  amily assignment
+00007270: 2061 6e61 6c79 7369 732c 2069 6620 7468   analysis, if th
+00007280: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00007290: 7468 6520 616e 616c 7973 6973 2077 6f75  the analysis wou
+000072a0: 6c64 2062 6520 696e 6974 6961 7465 640a  ld be initiated.
+000072b0: 2d73 612c 202d 2d73 6571 3261 7373 6967  -sa, --seq2assig
+000072c0: 6e2c 2074 6865 2071 7565 7269 6564 2073  n, the queried s
+000072d0: 6571 7565 6e63 6573 2064 6174 6120 6669  equences data fi
+000072e0: 6c65 2069 6e20 6765 6e65 2d74 6f2d 6661  le in gene-to-fa
+000072f0: 6d69 6c79 2061 7373 6967 6e6d 656e 7420  mily assignment 
+00007300: 616e 616c 7973 6973 2c20 6465 6661 756c  analysis, defaul
+00007310: 7420 4e6f 6e65 2c20 7468 6973 206f 7074  t None, this opt
+00007320: 696f 6e20 6361 6e20 6265 2070 726f 7669  ion can be provi
+00007330: 6465 6420 6d75 6c74 6970 6c65 2074 696d  ded multiple tim
+00007340: 6573 0a2d 6661 2c20 2d2d 6661 6d32 6173  es.-fa, --fam2as
+00007350: 7369 676e 2c20 7468 6520 7175 6572 6965  sign, the querie
+00007360: 6420 6661 6d69 6c69 7920 6461 7461 2066  d familiy data f
+00007370: 696c 6520 696e 2067 656e 652d 746f 2d66  ile in gene-to-f
+00007380: 616d 696c 7920 6173 7369 676e 6d65 6e74  amily assignment
+00007390: 2061 6e61 6c79 7369 732c 2064 6566 6175   analysis, defau
+000073a0: 6c74 204e 6f6e 650a 2d63 632c 202d 2d63  lt None.-cc, --c
+000073b0: 6f6e 6361 742c 2066 6c61 6720 6f70 7469  oncat, flag opti
+000073c0: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
+000073d0: 6e69 7469 6174 6520 7468 6520 636f 6e63  nitiate the conc
+000073e0: 6174 656e 6174 696f 6e20 7069 7065 6c69  atenation pipeli
+000073f0: 6e65 2066 6f72 206f 7274 686f 6772 6f75  ne for orthogrou
+00007400: 7020 696e 6665 726e 6563 652c 2069 6620  p infernece, if 
+00007410: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00007420: 2c20 7468 6520 616e 616c 7973 6973 2077  , the analysis w
+00007430: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+00007440: 640a 2d74 652c 202d 2d74 6573 7473 6f67  d.-te, --testsog
+00007450: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00007460: 6865 7468 6572 2074 6f20 696e 6974 6961  hether to initia
+00007470: 7465 2074 6865 2075 6e62 6961 7365 6420  te the unbiased 
+00007480: 7465 7374 206f 6620 7369 6e67 6c65 2d63  test of single-c
+00007490: 6f70 7920 6765 6e65 2066 616d 696c 6965  opy gene familie
+000074a0: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
+000074b0: 6173 2073 6574 2c20 7468 6520 616e 616c  as set, the anal
+000074c0: 7973 6973 2077 6f75 6c64 2062 6520 696e  ysis would be in
+000074d0: 6974 6961 7465 640a 2d62 732c 202d 2d62  itiated.-bs, --b
+000074e0: 696e 732c 2074 6865 206e 756d 6265 7220  ins, the number 
+000074f0: 6f66 2062 696e 7320 6469 7669 6465 6420  of bins divided 
+00007500: 696e 2074 6865 2067 656e 6520 6c65 6e67  in the gene leng
+00007510: 7468 206e 6f72 6d61 6c69 7a61 7469 6f6e  th normalization
+00007520: 2c20 6465 6661 756c 7420 3130 300a 2d6e  , default 100.-n
+00007530: 702c 202d 2d6e 6f72 6d61 6c69 7a65 6470  p, --normalizedp
+00007540: 6572 6365 6e74 2c20 7468 6520 7065 7263  ercent, the perc
+00007550: 656e 7461 6765 206f 6620 7570 7065 7220  entage of upper 
+00007560: 6869 7473 2075 7365 6420 666f 7220 6765  hits used for ge
+00007570: 6e65 206c 656e 6774 6820 6e6f 726d 616c  ne length normal
+00007580: 697a 6174 696f 6e2c 2064 6566 6175 6c74  ization, default
+00007590: 2035 0a2d 6e6e 2c20 2d2d 6e6f 6e6f 726d   5.-nn, --nonorm
+000075a0: 616c 697a 6174 696f 6e2c 2066 6c61 6720  alization, flag 
+000075b0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000075c0: 746f 2063 616c 6c20 6f66 6620 7468 6520  to call off the 
+000075d0: 6e6f 726d 616c 697a 6174 696f 6e2c 2069  normalization, i
+000075e0: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+000075f0: 6574 2c20 6e6f 206e 6f72 6d61 6c69 7a61  et, no normaliza
+00007600: 7469 6f6e 2077 6f75 6c64 2062 6520 636f  tion would be co
+00007610: 6e64 7563 7465 640a 2d62 736f 672c 202d  nducted.-bsog, -
+00007620: 2d62 7573 636f 736f 672c 2066 6c61 6720  -buscosog, flag 
+00007630: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00007640: 746f 2069 6e69 7469 6174 6520 7468 6520  to initiate the 
+00007650: 6275 7363 6f2d 6775 6964 6564 2073 696e  busco-guided sin
+00007660: 676c 652d 636f 7079 2067 656e 6520 6661  gle-copy gene fa
+00007670: 6d69 6c79 2061 6e61 6c79 7369 732c 2069  mily analysis, i
+00007680: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00007690: 6574 2c20 7468 6520 616e 616c 7973 6973  et, the analysis
+000076a0: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+000076b0: 7465 640a 2d62 686d 6d2c 202d 2d62 7573  ted.-bhmm, --bus
+000076c0: 636f 686d 6d2c 2074 6865 2048 4d4d 2070  cohmm, the HMM p
+000076d0: 726f 6669 6c65 2064 6174 6166 696c 6520  rofile datafile 
+000076e0: 696e 2074 6865 2062 7573 636f 2d67 7569  in the busco-gui
+000076f0: 6465 6420 7369 6e67 6c65 2d63 6f70 7920  ded single-copy 
+00007700: 6765 6e65 2066 616d 696c 7920 616e 616c  gene family anal
+00007710: 7973 6973 2c20 6465 6661 756c 7420 4e6f  ysis, default No
+00007720: 6e65 0a2d 6263 7466 2c20 2d2d 6275 7363  ne.-bctf, --busc
+00007730: 6f63 7574 6f66 662c 2074 6865 2048 4d4d  ocutoff, the HMM
+00007740: 2073 636f 7265 2063 7574 6f66 6620 6461   score cutoff da
+00007750: 7461 6669 6c65 2069 6e20 7468 6520 6275  tafile in the bu
+00007760: 7363 6f2d 6775 6964 6564 2073 696e 676c  sco-guided singl
+00007770: 652d 636f 7079 2067 656e 6520 6661 6d69  e-copy gene fami
+00007780: 6c79 2061 6e61 6c79 7369 732c 2064 6566  ly analysis, def
+00007790: 6175 6c74 204e 6f6e 650a 2d6f 6620 2c2d  ault None.-of ,-
+000077a0: 2d6f 6766 6f72 6d61 742c 2066 6c61 6720  -ogformat, flag 
+000077b0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000077c0: 746f 2061 6464 2069 6e64 6578 2074 6f20  to add index to 
+000077d0: 7468 6520 5242 4820 6661 6d69 6c69 6573  the RBH families
+000077e0: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+000077f0: 6d20 6077 6764 2066 6f63 7573 6020 6361  m `wgd focus` ca
+00007800: 6e20 7265 616c 697a 6520 7468 6520 636f  n realize the co
+00007810: 6e63 6174 656e 6174 696f 6e2d 6261 7365  ncatenation-base
+00007820: 6420 616e 6420 636f 616c 6573 6365 6e63  d and coalescenc
+00007830: 652d 6261 7365 6420 7068 796c 6f67 656e  e-based phylogen
+00007840: 6574 6963 2069 6e66 6572 656e 6365 2061  etic inference a
+00007850: 6e64 2070 6879 6c6f 6765 6e65 7469 6320  nd phylogenetic 
+00007860: 6461 7469 6e67 206f 6620 5747 4473 2065  dating of WGDs e
+00007870: 7463 2e0a 6060 600a 7767 6420 666f 6375  tc..```.wgd focu
+00007880: 7320 6661 6d69 6c69 6573 2073 6571 7565  s families seque
+00007890: 6e63 6573 2028 6f70 7469 6f6e 290a 2d2d  nces (option).--
+000078a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000078b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000078c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000078d0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000078e0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 0a2d  --------------.-
+000078f0: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
+00007900: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
+00007910: 792c 2064 6566 6175 6c74 2077 6764 5f66  y, default wgd_f
+00007920: 6f63 7573 0a2d 742c 202d 2d74 6d70 6469  ocus.-t, --tmpdi
+00007930: 722c 2074 6865 2074 656d 706f 7261 7279  r, the temporary
+00007940: 2077 6f72 6b69 6e67 2064 6972 6563 746f   working directo
+00007950: 7279 2c20 6465 6661 756c 7420 4e6f 6e65  ry, default None
+00007960: 2c20 6966 204e 6f6e 6520 7761 7320 6769  , if None was gi
+00007970: 7665 6e2c 2074 6865 2074 6d70 6469 7220  ven, the tmpdir 
+00007980: 7769 6c6c 2062 6520 6173 7369 676e 6564  will be assigned
+00007990: 2072 616e 646f 6d20 6e61 6d65 7320 696e   random names in
+000079a0: 2063 7572 7265 6e74 2064 6972 6563 746f   current directo
+000079b0: 7279 2061 6e64 2061 7574 6f6d 6174 656c  ry and automatel
+000079c0: 7920 7265 6d6f 7665 6420 6174 2074 6865  y removed at the
+000079d0: 2063 6f6d 706c 6574 696f 6e20 6f66 2070   completion of p
+000079e0: 726f 6772 616d 2c20 656c 7365 2074 6865  rogram, else the
+000079f0: 2074 6d70 6469 7220 776f 756c 6420 6265   tmpdir would be
+00007a00: 206b 6570 740a 2d6e 2c20 2d2d 6e74 6872   kept.-n, --nthr
+00007a10: 6561 6473 2c20 7468 6520 6e75 6d62 6572  eads, the number
+00007a20: 206f 6620 7468 7265 6164 7320 746f 2075   of threads to u
+00007a30: 7365 2c20 6465 6661 756c 7420 340a 2d2d  se, default 4.--
+00007a40: 746f 5f73 746f 702c 2066 6c61 6720 6f70  to_stop, flag op
+00007a50: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+00007a60: 2074 7261 6e73 6c61 7465 2074 6872 6f75   translate throu
+00007a70: 6768 2053 544f 5020 636f 646f 6e73 2c20  gh STOP codons, 
+00007a80: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00007a90: 7365 742c 2074 7261 6e73 6c61 7469 6f6e  set, translation
+00007aa0: 2077 696c 6c20 6265 2074 6572 6d69 6e61   will be termina
+00007ab0: 7465 6420 6174 2074 6865 2066 6972 7374  ted at the first
+00007ac0: 2069 6e2d 6672 616d 6520 7374 6f70 2063   in-frame stop c
+00007ad0: 6f64 6f6e 2c20 656c 7365 2061 2066 756c  odon, else a ful
+00007ae0: 6c20 7472 616e 736c 6174 696f 6e20 636f  l translation co
+00007af0: 6e74 696e 7569 6e67 206f 6e20 7061 7374  ntinuing on past
+00007b00: 2061 6e79 2073 746f 7020 636f 646f 6e73   any stop codons
+00007b10: 2077 6f75 6c64 2062 6520 696e 6974 6961   would be initia
+00007b20: 7465 640a 2d2d 6364 732c 2066 6c61 6720  ted.--cds, flag 
+00007b30: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00007b40: 746f 206f 6e6c 7920 7472 616e 736c 6174  to only translat
+00007b50: 6520 7468 6520 636f 6d70 6c65 7465 2043  e the complete C
+00007b60: 4453 2074 6861 7420 7374 6172 7473 2077  DS that starts w
+00007b70: 6974 6820 6120 7661 6c69 6420 7374 6172  ith a valid star
+00007b80: 7420 636f 646f 6e20 616e 6420 6f6e 6c79  t codon and only
+00007b90: 2063 6f6e 7461 696e 7320 6120 7369 6e67   contains a sing
+00007ba0: 6c65 2069 6e2d 6672 616d 6520 7374 6f70  le in-frame stop
+00007bb0: 2063 6f64 6f6e 2061 7420 7468 6520 656e   codon at the en
+00007bc0: 6420 616e 6420 6d75 7374 2062 6520 6469  d and must be di
+00007bd0: 7669 6461 626c 6520 6279 2074 6872 6565  vidable by three
+00007be0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00007bf0: 7320 7365 742c 206f 6e6c 7920 7468 6520  s set, only the 
+00007c00: 636f 6d70 6c65 7465 2043 4453 2077 6f75  complete CDS wou
+00007c10: 6c64 2062 6520 7472 616e 736c 6174 6564  ld be translated
+00007c20: 0a2d 2d73 7472 6970 5f67 6170 732c 2066  .--strip_gaps, f
+00007c30: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00007c40: 6865 7220 746f 2064 726f 7020 616c 6c20  her to drop all 
+00007c50: 6761 7073 2069 6e20 6d75 6c74 6970 6c65  gaps in multiple
+00007c60: 2073 6571 7565 6e63 6520 616c 6967 6e6d   sequence alignm
+00007c70: 656e 742c 2069 6620 7468 6520 666c 6167  ent, if the flag
+00007c80: 2077 6173 2073 6574 2c20 616c 6c20 6761   was set, all ga
+00007c90: 7073 2077 6f75 6c64 2062 6520 6472 6f70  ps would be drop
+00007ca0: 7065 640a 2d61 2c20 2d2d 616c 6967 6e65  ped.-a, --aligne
+00007cb0: 722c 2077 6869 6368 2061 6c69 676e 6d65  r, which alignme
+00007cc0: 6e74 2070 726f 6772 616d 2074 6f20 7573  nt program to us
+00007cd0: 652c 2064 6566 6175 6c74 206d 6166 6674  e, default mafft
+00007ce0: 0a2d 7472 6565 2c20 2d2d 7472 6565 5f6d  .-tree, --tree_m
+00007cf0: 6574 686f 642c 2077 6869 6368 2067 656e  ethod, which gen
+00007d00: 6520 7472 6565 2069 6e66 6572 656e 6365  e tree inference
+00007d10: 2070 726f 6772 616d 2074 6f20 696e 766f   program to invo
+00007d20: 6b65 2c20 6465 6661 756c 7420 6661 7374  ke, default fast
+00007d30: 7472 6565 0a2d 7473 2c20 2d2d 7472 6565  tree.-ts, --tree
+00007d40: 7365 742c 2074 6865 2070 6172 616d 6574  set, the paramet
+00007d50: 6572 7320 7365 7474 696e 6720 666f 7220  ers setting for 
+00007d60: 6765 6e65 2074 7265 6520 696e 6665 7265  gene tree infere
+00007d70: 6e63 652c 2064 6566 6175 6c74 204e 6f6e  nce, default Non
+00007d80: 652c 2074 6869 7320 6f70 7469 6f6e 2063  e, this option c
+00007d90: 616e 2062 6520 7072 6f76 6964 6564 206d  an be provided m
+00007da0: 756c 7469 706c 6520 7469 6d65 730a 2d2d  ultiple times.--
+00007db0: 636f 6e63 6174 656e 6174 696f 6e2c 2066  concatenation, f
+00007dc0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00007dd0: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+00007de0: 7468 6520 636f 6e63 6174 656e 6174 696f  the concatenatio
+00007df0: 6e2d 6261 7365 6420 7370 6563 6965 7320  n-based species 
+00007e00: 7472 6565 2069 6e66 6572 656e 6365 2c20  tree inference, 
+00007e10: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00007e20: 7365 742c 2063 6f6e 6361 7465 6e61 7469  set, concatenati
+00007e30: 6f6e 2d62 6173 6564 2073 7065 6369 6573  on-based species
+00007e40: 2074 7265 6520 776f 756c 6420 6265 2069   tree would be i
+00007e50: 6e66 6572 6564 0a2d 2d63 6f61 6c65 7363  nfered.--coalesc
+00007e60: 656e 6365 2c20 666c 6167 206f 7074 696f  ence, flag optio
+00007e70: 6e2c 2077 6865 7468 6572 2074 6f20 696e  n, whether to in
+00007e80: 6974 6961 7465 2074 6865 2063 6f61 6c65  itiate the coale
+00007e90: 7363 656e 6365 2d62 6173 6564 2073 7065  scence-based spe
+00007ea0: 6369 6573 2074 7265 6520 696e 6665 7265  cies tree infere
+00007eb0: 6e63 652c 2069 6620 7468 6520 666c 6167  nce, if the flag
+00007ec0: 2077 6173 2073 6574 2c20 636f 616c 6573   was set, coales
+00007ed0: 6365 6e63 652d 6261 7365 6420 7370 6563  cence-based spec
+00007ee0: 6965 7320 7472 6565 2077 6f75 6c64 2062  ies tree would b
+00007ef0: 6520 696e 6665 7265 640a 2d73 702c 202d  e infered.-sp, -
+00007f00: 2d73 7065 6369 6573 7472 6565 2c20 7370  -speciestree, sp
+00007f10: 6563 6965 7320 7472 6565 2064 6174 6166  ecies tree dataf
+00007f20: 696c 6520 666f 7220 6461 7469 6e67 2c20  ile for dating, 
+00007f30: 6465 6661 756c 7420 4e6f 6e65 0a2d 642c  default None.-d,
+00007f40: 202d 2d64 6174 696e 672c 2077 6869 6368   --dating, which
+00007f50: 206d 6f6c 6563 756c 6172 2064 6174 696e   molecular datin
+00007f60: 6720 7072 6f67 7261 6d20 746f 2075 7365  g program to use
+00007f70: 2c20 6465 6661 756c 7420 6e6f 6e65 0a2d  , default none.-
+00007f80: 6473 2c20 2d2d 6461 7469 6e67 7365 742c  ds, --datingset,
+00007f90: 2074 6865 2070 6172 616d 6574 6572 7320   the parameters 
+00007fa0: 7365 7474 696e 6720 666f 7220 6461 7469  setting for dati
+00007fb0: 6e67 2070 726f 6772 616d 2c20 6465 6661  ng program, defa
+00007fc0: 756c 7420 4e6f 6e65 2c20 7468 6973 206f  ult None, this o
+00007fd0: 7074 696f 6e20 6361 6e20 6265 2070 726f  ption can be pro
+00007fe0: 7669 6465 6420 6d75 6c74 6970 6c65 2074  vided multiple t
+00007ff0: 696d 6573 0a2d 6e73 2c20 2d2d 6e73 6974  imes.-ns, --nsit
+00008000: 6573 2c20 7468 6520 6e73 6974 6573 2069  es, the nsites i
+00008010: 6e66 6f72 6d61 7469 6f6e 2066 6f72 2072  nformation for r
+00008020: 3873 2064 6174 696e 672c 2064 6566 6175  8s dating, defau
+00008030: 6c74 204e 6f6e 650a 2d6f 742c 202d 2d6f  lt None.-ot, --o
+00008040: 7574 6772 6f75 702c 2074 6865 206f 7574  utgroup, the out
+00008050: 6772 6f75 7020 696e 666f 726d 6174 696f  group informatio
+00008060: 6e20 666f 7220 7238 7320 6461 7469 6e67  n for r8s dating
+00008070: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+00008080: 7074 2c20 2d2d 7061 7274 6974 696f 6e2c  pt, --partition,
+00008090: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+000080a0: 6574 6865 7220 746f 2069 6e69 7469 6174  ether to initiat
+000080b0: 6520 7061 7274 6974 696f 6e20 6461 7469  e partition dati
+000080c0: 6e67 2061 6e61 6c79 7369 7320 666f 7220  ng analysis for 
+000080d0: 636f 646f 6e2c 2069 6620 7468 6520 666c  codon, if the fl
+000080e0: 6167 2077 6173 2073 6574 2c20 616e 2061  ag was set, an a
+000080f0: 6464 6974 696f 6e61 6c20 7061 7274 6974  dditional partit
+00008100: 696f 6e20 6461 7469 6e67 2061 6e61 6c79  ion dating analy
+00008110: 7369 7320 776f 756c 6420 6265 2069 6e69  sis would be ini
+00008120: 7469 6174 6564 0a2d 616d 2c20 2d2d 6161  tiated.-am, --aa
+00008130: 6d6f 6465 6c2c 2077 6869 6368 2070 726f  model, which pro
+00008140: 7465 696e 206d 6f64 656c 2074 6f20 6265  tein model to be
+00008150: 2075 7365 6420 696e 206d 636d 6374 7265   used in mcmctre
+00008160: 652c 2064 6566 6175 6c74 2070 6f69 7373  e, default poiss
+00008170: 6f6e 0a2d 6b73 2c20 666c 6167 206f 7074  on.-ks, flag opt
+00008180: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+00008190: 696e 6974 6961 7465 204b 7320 6361 6c63  initiate Ks calc
+000081a0: 756c 6174 696f 6e20 666f 7220 686f 6d6f  ulation for homo
+000081b0: 6c6f 6775 6573 2069 6e20 7468 6520 7072  logues in the pr
+000081c0: 6f76 6964 6564 206f 7274 686f 6c6f 676f  ovided orthologo
+000081d0: 7573 2067 656e 6520 6661 6d69 6c79 0a2d  us gene family.-
+000081e0: 2d61 6e6e 6f74 6174 696f 6e2c 2077 6869  -annotation, whi
+000081f0: 6368 2061 6e6e 6f74 6174 696f 6e20 7072  ch annotation pr
+00008200: 6f67 7261 6d20 746f 2075 7365 2c20 6465  ogram to use, de
+00008210: 6661 756c 7420 4e6f 6e65 0a2d 2d70 6169  fault None.--pai
+00008220: 7277 6973 652c 2066 6c61 6720 6f70 7469  rwise, flag opti
+00008230: 6f6e 2c20 7768 6574 6865 7220 746f 2069  on, whether to i
+00008240: 6e69 7469 6174 6520 7061 6972 7769 7365  nitiate pairwise
+00008250: 204b 7320 6573 7469 6d61 7469 6f6e 2c20   Ks estimation, 
+00008260: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00008270: 7365 742c 2070 6169 7277 6973 6520 4b73  set, pairwise Ks
+00008280: 2076 616c 7565 7320 776f 756c 6420 6265   values would be
+00008290: 2065 7374 696d 6174 6564 0a2d 6564 2c20   estimated.-ed, 
+000082a0: 2d2d 6567 676e 6f67 6461 7461 2c20 7468  --eggnogdata, th
+000082b0: 6520 6567 676e 6f67 2061 6e6e 6f74 6174  e eggnog annotat
+000082c0: 696f 6e20 6461 7461 6669 6c65 2c20 6465  ion datafile, de
+000082d0: 6661 756c 7420 4e6f 6e65 0a2d 2d70 6661  fault None.--pfa
+000082e0: 6d2c 2077 6869 6368 206f 7074 696f 6e20  m, which option 
+000082f0: 746f 2075 7365 2066 6f72 2070 6661 6d20  to use for pfam 
+00008300: 616e 6e6f 7461 7469 6f6e 2c20 6465 6661  annotation, defa
+00008310: 756c 7420 4e6f 6e65 0a2d 2d64 6d6e 622c  ult None.--dmnb,
+00008320: 2074 6865 2064 6961 6d6f 6e64 2064 6174   the diamond dat
+00008330: 6162 6173 6520 666f 7220 616e 6e6f 7461  abase for annota
+00008340: 7469 6f6e 2c20 6465 6661 756c 7420 4e6f  tion, default No
+00008350: 6e65 0a2d 2d68 6d6d 2c20 7468 6520 484d  ne.--hmm, the HM
+00008360: 4d20 7072 6f66 696c 6520 666f 7220 616e  M profile for an
+00008370: 6e6f 7461 7469 6f6e 2c20 6465 6661 756c  notation, defaul
+00008380: 7420 4e6f 6e65 0a2d 2d65 7661 6c75 652c  t None.--evalue,
+00008390: 2074 6865 2065 2d76 616c 7565 2063 7574   the e-value cut
+000083a0: 2d6f 6666 2066 6f72 2061 6e6e 6f74 6174  -off for annotat
+000083b0: 696f 6e2c 2064 6566 6175 6c74 2031 652d  ion, default 1e-
+000083c0: 3130 0a2d 2d65 7865 7061 7468 2c20 7468  10.--exepath, th
+000083d0: 6520 7061 7468 2074 6f20 7468 6520 696e  e path to the in
+000083e0: 7465 7270 726f 7363 616e 2065 7865 6375  terproscan execu
+000083f0: 7461 626c 652c 2064 6566 6175 6c74 204e  table, default N
+00008400: 6f6e 650a 2d66 2c20 2d2d 666f 7373 696c  one.-f, --fossil
+00008410: 2c20 7468 6520 666f 7373 696c 2063 616c  , the fossil cal
+00008420: 6962 7261 7469 6f6e 2069 6e66 6f72 6d61  ibration informa
+00008430: 7469 6f6e 2069 6e20 4265 6173 742c 2064  tion in Beast, d
+00008440: 6566 6175 6c74 2028 2763 6c61 6465 313b  efault ('clade1;
+00008450: 636c 6164 6532 272c 2027 7461 7861 312c  clade2', 'taxa1,
+00008460: 7461 7861 323b 7461 7861 332c 7461 7861  taxa2;taxa3,taxa
+00008470: 3427 2c20 2734 3b35 272c 2027 302e 353b  4', '4;5', '0.5;
+00008480: 302e 3627 2c20 2734 3030 3b35 3030 2729  0.6', '400;500')
+00008490: 0a2d 7268 2c20 2d2d 726f 6f74 6865 6967  .-rh, --rootheig
+000084a0: 6874 2c20 7468 6520 726f 6f74 2068 6569  ht, the root hei
+000084b0: 6768 7420 6361 6c69 6272 6174 696f 6e20  ght calibration 
+000084c0: 696e 666f 2069 6e20 4265 6173 742c 2064  info in Beast, d
+000084d0: 6566 6175 6c74 2028 342c 302e 352c 3430  efault (4,0.5,40
+000084e0: 3029 0a2d 6373 2c20 2d2d 6368 6169 6e73  0).-cs, --chains
+000084f0: 6574 2c20 7468 6520 7061 7261 6d65 7465  et, the paramete
+00008500: 7273 206f 6620 4d43 4d43 2063 6861 696e  rs of MCMC chain
+00008510: 2069 6e20 4265 6173 742c 2064 6566 6175   in Beast, defau
+00008520: 6c74 2028 3130 3030 302c 3130 3029 0a2d  lt (10000,100).-
+00008530: 2d62 6561 7374 6c67 6a61 722c 2074 6865  -beastlgjar, the
+00008540: 2070 6174 6820 746f 2062 6561 7374 4c47   path to beastLG
+00008550: 2e6a 6172 2c20 6465 6661 756c 7420 4e6f  .jar, default No
+00008560: 6e65 0a2d 2d62 6561 676c 652c 2066 6c61  ne.--beagle, fla
+00008570: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+00008580: 7220 746f 2075 7365 2062 6561 676c 6520  r to use beagle 
+00008590: 696e 2042 6561 7374 2c20 6966 2074 6865  in Beast, if the
+000085a0: 2066 6c61 6720 7761 7320 7365 742c 2062   flag was set, b
+000085b0: 6561 676c 6520 776f 756c 6420 6265 2075  eagle would be u
+000085c0: 7365 640a 2d2d 7072 6f74 636f 6364 6174  sed.--protcocdat
+000085d0: 696e 672c 2066 6c61 6720 6f70 7469 6f6e  ing, flag option
+000085e0: 2c20 7768 6574 6865 7220 746f 206f 6e6c  , whether to onl
+000085f0: 7920 696e 6974 6961 7465 2074 6865 2070  y initiate the p
+00008600: 726f 7465 696e 2d63 6f6e 6361 7465 6e61  rotein-concatena
+00008610: 7469 6f6e 2d62 6173 6564 2064 6174 696e  tion-based datin
+00008620: 6720 616e 616c 7973 6973 2c20 6966 2074  g analysis, if t
+00008630: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00008640: 2074 6865 2061 6e61 6c79 7369 7320 776f   the analysis wo
+00008650: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
+00008660: 0a2d 2d70 726f 7464 6174 696e 672c 2066  .--protdating, f
+00008670: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00008680: 6865 7220 746f 206f 6e6c 7920 696e 6974  her to only init
+00008690: 6961 7465 2074 6865 2070 726f 7465 696e  iate the protein
+000086a0: 2d62 6173 6564 2064 6174 696e 6720 616e  -based dating an
+000086b0: 616c 7973 6973 2c20 6966 2074 6865 2066  alysis, if the f
+000086c0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+000086d0: 2061 6e61 6c79 7369 7320 776f 756c 6420   analysis would 
+000086e0: 6265 2069 6e69 7469 6174 6564 0a60 6060  be initiated.```
+000086f0: 0a0a 5468 6520 7072 6f67 7261 6d20 6077  ..The program `w
+00008700: 6764 206b 7364 6020 6361 6e20 7265 616c  gd ksd` can real
+00008710: 697a 6520 7468 6520 636f 6e73 7472 7563  ize the construc
+00008720: 7469 6f6e 206f 6620 2a4b 2a3c 7375 623e  tion of *K*<sub>
+00008730: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+00008740: 7269 6275 7469 6f6e 2061 6e64 2072 6174  ribution and rat
+00008750: 6520 636f 7272 6563 7469 6f6e 2e0a 6060  e correction..``
+00008760: 600a 7767 6420 6b73 6420 6661 6d69 6c69  `.wgd ksd famili
+00008770: 6573 2073 6571 7565 6e63 6573 2028 6f70  es sequences (op
+00008780: 7469 6f6e 290a 2d2d 2d2d 2d2d 2d2d 2d2d  tion).----------
+00008790: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000087a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000087b0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000087c0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000087d0: 2d2d 2d2d 2d2d 0a2d 6f2c 202d 2d6f 7574  ------.-o, --out
+000087e0: 6469 722c 2074 6865 206f 7574 7075 7420  dir, the output 
+000087f0: 6469 7265 6374 6f72 792c 2064 6566 6175  directory, defau
+00008800: 6c74 2077 6764 5f6b 7364 0a2d 742c 202d  lt wgd_ksd.-t, -
+00008810: 2d74 6d70 6469 722c 2074 6865 2074 656d  -tmpdir, the tem
+00008820: 706f 7261 7279 2077 6f72 6b69 6e67 2064  porary working d
+00008830: 6972 6563 746f 7279 2c20 6465 6661 756c  irectory, defaul
+00008840: 7420 4e6f 6e65 2c20 6966 204e 6f6e 6520  t None, if None 
+00008850: 7761 7320 6769 7665 6e2c 2074 6865 2074  was given, the t
+00008860: 6d70 6469 7220 7769 6c6c 2062 6520 6173  mpdir will be as
+00008870: 7369 676e 6564 2072 616e 646f 6d20 6e61  signed random na
+00008880: 6d65 7320 696e 2063 7572 7265 6e74 2064  mes in current d
+00008890: 6972 6563 746f 7279 2061 6e64 2061 7574  irectory and aut
+000088a0: 6f6d 6174 656c 7920 7265 6d6f 7665 6420  omately removed 
+000088b0: 6174 2074 6865 2063 6f6d 706c 6574 696f  at the completio
+000088c0: 6e20 6f66 2070 726f 6772 616d 2c20 656c  n of program, el
+000088d0: 7365 2074 6865 2074 6d70 6469 7220 776f  se the tmpdir wo
+000088e0: 756c 6420 6265 206b 6570 740a 2d6e 2c20  uld be kept.-n, 
+000088f0: 2d2d 6e74 6872 6561 6473 2c20 7468 6520  --nthreads, the 
+00008900: 6e75 6d62 6572 206f 6620 7468 7265 6164  number of thread
+00008910: 7320 746f 2075 7365 2c20 6465 6661 756c  s to use, defaul
+00008920: 7420 340a 2d2d 746f 5f73 746f 702c 2066  t 4.--to_stop, f
+00008930: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+00008940: 6865 7220 746f 2074 7261 6e73 6c61 7465  her to translate
+00008950: 2074 6872 6f75 6768 2053 544f 5020 636f   through STOP co
+00008960: 646f 6e73 2c20 6966 2074 6865 2066 6c61  dons, if the fla
+00008970: 6720 7761 7320 7365 742c 2074 7261 6e73  g was set, trans
+00008980: 6c61 7469 6f6e 2077 696c 6c20 6265 2074  lation will be t
+00008990: 6572 6d69 6e61 7465 6420 6174 2074 6865  erminated at the
+000089a0: 2066 6972 7374 2069 6e2d 6672 616d 6520   first in-frame 
+000089b0: 7374 6f70 2063 6f64 6f6e 2c20 656c 7365  stop codon, else
+000089c0: 2061 2066 756c 6c20 7472 616e 736c 6174   a full translat
+000089d0: 696f 6e20 636f 6e74 696e 7569 6e67 206f  ion continuing o
+000089e0: 6e20 7061 7374 2061 6e79 2073 746f 7020  n past any stop 
+000089f0: 636f 646f 6e73 2077 6f75 6c64 2062 6520  codons would be 
+00008a00: 696e 6974 6961 7465 640a 2d2d 6364 732c  initiated.--cds,
+00008a10: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00008a20: 6574 6865 7220 746f 206f 6e6c 7920 7472  ether to only tr
+00008a30: 616e 736c 6174 6520 7468 6520 636f 6d70  anslate the comp
+00008a40: 6c65 7465 2043 4453 2074 6861 7420 7374  lete CDS that st
+00008a50: 6172 7473 2077 6974 6820 6120 7661 6c69  arts with a vali
+00008a60: 6420 7374 6172 7420 636f 646f 6e20 616e  d start codon an
+00008a70: 6420 6f6e 6c79 2063 6f6e 7461 696e 7320  d only contains 
+00008a80: 6120 7369 6e67 6c65 2069 6e2d 6672 616d  a single in-fram
+00008a90: 6520 7374 6f70 2063 6f64 6f6e 2061 7420  e stop codon at 
+00008aa0: 7468 6520 656e 6420 616e 6420 6d75 7374  the end and must
+00008ab0: 2062 6520 6469 7669 6461 626c 6520 6279   be dividable by
+00008ac0: 2074 6872 6565 2c20 6966 2074 6865 2066   three, if the f
+00008ad0: 6c61 6720 7761 7320 7365 742c 206f 6e6c  lag was set, onl
+00008ae0: 7920 7468 6520 636f 6d70 6c65 7465 2043  y the complete C
+00008af0: 4453 2077 6f75 6c64 2062 6520 7472 616e  DS would be tran
+00008b00: 736c 6174 6564 0a2d 2d70 6169 7277 6973  slated.--pairwis
+00008b10: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
+00008b20: 7768 6574 6865 7220 746f 2069 6e69 7469  whether to initi
+00008b30: 6174 6520 7061 6972 7769 7365 204b 7320  ate pairwise Ks 
+00008b40: 6573 7469 6d61 7469 6f6e 2c20 6966 2074  estimation, if t
+00008b50: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+00008b60: 2070 6169 7277 6973 6520 4b73 2076 616c   pairwise Ks val
+00008b70: 7565 7320 776f 756c 6420 6265 2065 7374  ues would be est
+00008b80: 696d 6174 6564 0a2d 2d73 7472 6970 5f67  imated.--strip_g
+00008b90: 6170 732c 2066 6c61 6720 6f70 7469 6f6e  aps, flag option
+00008ba0: 2c20 7768 6574 6865 7220 746f 2064 726f  , whether to dro
+00008bb0: 7020 616c 6c20 6761 7073 2069 6e20 6d75  p all gaps in mu
+00008bc0: 6c74 6970 6c65 2073 6571 7565 6e63 6520  ltiple sequence 
+00008bd0: 616c 6967 6e6d 656e 742c 2069 6620 7468  alignment, if th
+00008be0: 6520 666c 6167 2077 6173 2073 6574 2c20  e flag was set, 
+00008bf0: 616c 6c20 6761 7073 2077 6f75 6c64 2062  all gaps would b
+00008c00: 6520 6472 6f70 7065 640a 2d61 2c20 2d2d  e dropped.-a, --
+00008c10: 616c 6967 6e65 722c 2077 6869 6368 2061  aligner, which a
+00008c20: 6c69 676e 6d65 6e74 2070 726f 6772 616d  lignment program
+00008c30: 2074 6f20 7573 652c 2064 6566 6175 6c74   to use, default
+00008c40: 206d 6166 6674 200a 2d74 7265 652c 202d   mafft .-tree, -
+00008c50: 2d74 7265 655f 6d65 7468 6f64 2c20 7768  -tree_method, wh
+00008c60: 6963 6820 6765 6e65 2074 7265 6520 696e  ich gene tree in
+00008c70: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
+00008c80: 746f 2069 6e76 6f6b 652c 2064 6566 6175  to invoke, defau
+00008c90: 6c74 2066 6173 7474 7265 650a 2d2d 7472  lt fasttree.--tr
+00008ca0: 6565 5f6f 7074 696f 6e73 2c20 6f70 7469  ee_options, opti
+00008cb0: 6f6e 7320 696e 2074 7265 6520 696e 6665  ons in tree infe
+00008cc0: 7265 6e63 6520 6173 2061 2063 6f6d 6d61  rence as a comma
+00008cd0: 2073 6570 6172 6174 6564 2073 7472 696e   separated strin
+00008ce0: 672c 2064 6566 6175 6c74 204e 6f6e 650a  g, default None.
+00008cf0: 2d2d 6e6f 6465 5f61 7665 7261 6765 2c20  --node_average, 
+00008d00: 666c 6167 206f 7074 696f 6e2c 2077 6865  flag option, whe
+00008d10: 7468 6572 2074 6f20 696e 6974 6961 7465  ther to initiate
+00008d20: 206e 6f64 652d 6176 6572 6167 6520 7761   node-average wa
+00008d30: 7920 6f66 2064 652d 7265 6475 6e64 616e  y of de-redundan
+00008d40: 6379 2069 6e73 7465 6164 206f 6620 6e6f  cy instead of no
+00008d50: 6465 2d77 6569 6768 7465 642c 2069 6620  de-weighted, if 
+00008d60: 7468 6520 666c 6167 2077 6173 2073 6574  the flag was set
+00008d70: 2c20 7468 6520 6e6f 6465 2d61 7665 7261  , the node-avera
+00008d80: 6769 6e67 2064 652d 7265 6475 6e64 616e  ging de-redundan
+00008d90: 6379 2077 6f75 6c64 2062 6520 696e 6974  cy would be init
+00008da0: 6961 7465 640a 2d73 722c 202d 2d73 7061  iated.-sr, --spa
+00008db0: 6972 2c20 7468 6520 7370 6563 6965 7320  ir, the species 
+00008dc0: 7061 6972 2074 6f20 6265 2070 6c6f 7474  pair to be plott
+00008dd0: 6564 2c20 6465 6661 756c 7420 4e6f 6e65  ed, default None
+00008de0: 2c20 7468 6973 206f 7074 696f 6e20 6361  , this option ca
+00008df0: 6e20 6265 2070 726f 7669 6465 6420 6d75  n be provided mu
+00008e00: 6c74 6970 6c65 2074 696d 6573 0a2d 7370  ltiple times.-sp
+00008e10: 2c20 2d2d 7370 6563 6965 7374 7265 652c  , --speciestree,
+00008e20: 2074 6865 2073 7065 6369 6573 2074 7265   the species tre
+00008e30: 6520 746f 2070 6572 666f 726d 2072 6174  e to perform rat
+00008e40: 6520 636f 7272 6563 7469 6f6e 2c20 6465  e correction, de
+00008e50: 6661 756c 7420 4e6f 6e65 2c20 6966 204e  fault None, if N
+00008e60: 6f6e 6520 7761 7320 6769 7665 6e2c 2074  one was given, t
+00008e70: 6865 2072 6174 6520 636f 7272 6563 7469  he rate correcti
+00008e80: 6f6e 2061 6e61 6c79 7369 7320 776f 756c  on analysis woul
+00008e90: 6420 6265 2063 616c 6c65 6420 6f66 660a  d be called off.
+00008ea0: 2d72 772c 202d 2d72 6577 6569 6768 742c  -rw, --reweight,
+00008eb0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+00008ec0: 6574 6865 7220 746f 2072 6563 616c 6375  ether to recalcu
+00008ed0: 6c61 7465 2074 6865 2077 6569 6768 7420  late the weight 
+00008ee0: 7065 7220 7370 6563 6965 7320 7061 6972  per species pair
+00008ef0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00008f00: 7320 7365 742c 2074 6865 2077 6569 6768  s set, the weigh
+00008f10: 7420 776f 756c 6420 6265 2072 6563 616c  t would be recal
+00008f20: 6375 6c61 7465 640a 2d6f 722c 202d 2d6f  culated.-or, --o
+00008f30: 6e6c 7972 6f6f 746f 7574 2c20 666c 6167  nlyrootout, flag
+00008f40: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+00008f50: 2074 6f20 6f6e 6c79 2063 6f6e 6475 6374   to only conduct
+00008f60: 2072 6174 6520 636f 7272 6563 7469 6f6e   rate correction
+00008f70: 2075 7369 6e67 2074 6865 206f 7574 6772   using the outgr
+00008f80: 6f75 7020 6174 2072 6f6f 7420 6173 206f  oup at root as o
+00008f90: 7574 6772 6f75 702c 2069 6620 7468 6520  utgroup, if the 
+00008fa0: 666c 6167 2077 6173 2073 6574 2c20 6f6e  flag was set, on
+00008fb0: 6c79 2074 6865 206f 7574 6772 6f75 7020  ly the outgroup 
+00008fc0: 6174 2072 6f6f 7420 776f 756c 6420 6265  at root would be
+00008fd0: 2075 7365 6420 6173 206f 7574 6772 6f75   used as outgrou
+00008fe0: 700a 2d65 706b 2c20 2d2d 6578 7472 6170  p.-epk, --extrap
+00008ff0: 6172 616e 6f6d 656b 732c 2065 7874 7261  aranomeks, extra
+00009000: 2070 6172 616e 6f6d 6520 4b73 2064 6174   paranome Ks dat
+00009010: 6120 746f 2070 6c6f 7420 696e 2074 6865  a to plot in the
+00009020: 206d 6978 6564 204b 7320 6469 7374 7269   mixed Ks distri
+00009030: 6275 7469 6f6e 2c20 6465 6661 756c 7420  bution, default 
+00009040: 4e6f 6e65 0a2d 6170 2c20 2d2d 616e 6368  None.-ap, --anch
+00009050: 6f72 706f 696e 7473 2c20 616e 6368 6f72  orpoints, anchor
+00009060: 706f 696e 7473 2e74 7874 2066 696c 6520  points.txt file 
+00009070: 746f 2070 6c6f 7420 616e 6368 6f72 204b  to plot anchor K
+00009080: 7320 696e 2074 6865 206d 6978 6564 204b  s in the mixed K
+00009090: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+000090a0: 6465 6661 756c 7420 4e6f 6e65 0a2d 706b  default None.-pk
+000090b0: 2c20 2d2d 706c 6f74 6b64 652c 2066 6c61  , --plotkde, fla
+000090c0: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+000090d0: 7220 746f 2070 6c6f 7420 6b64 6520 6375  r to plot kde cu
+000090e0: 7276 6520 6f66 206f 7274 686f 6c6f 676f  rve of orthologo
+000090f0: 7573 204b 7320 6469 7374 7269 6275 7469  us Ks distributi
+00009100: 6f6e 206f 7665 7220 6869 7374 6f67 7261  on over histogra
+00009110: 6d20 696e 2074 6865 206d 6978 6564 204b  m in the mixed K
+00009120: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+00009130: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00009140: 7365 742c 2074 6865 206b 6465 2063 7572  set, the kde cur
+00009150: 7665 2077 6f75 6c64 2062 6520 706c 6f74  ve would be plot
+00009160: 7465 640a 2d70 6167 2c20 2d2d 706c 6f74  ted.-pag, --plot
+00009170: 6170 676d 6d2c 2066 6c61 6720 6f70 7469  apgmm, flag opti
+00009180: 6f6e 2c20 7768 6574 6865 7220 746f 2070  on, whether to p
+00009190: 6572 666f 726d 2061 6e64 2070 6c6f 7420  erform and plot 
+000091a0: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
+000091b0: 206f 6620 616e 6368 6f72 204b 7320 696e   of anchor Ks in
+000091c0: 2074 6865 206d 6978 6564 204b 7320 6469   the mixed Ks di
+000091d0: 7374 7269 6275 7469 6f6e 2c20 6966 2074  stribution, if t
+000091e0: 6865 2066 6c61 6720 7761 7320 7365 742c  he flag was set,
+000091f0: 2074 6865 206d 6978 7475 7265 206d 6f64   the mixture mod
+00009200: 656c 696e 6720 6f66 2061 6e63 686f 7220  eling of anchor 
+00009210: 4b73 2077 6f75 6c64 2062 6520 706c 6f74  Ks would be plot
+00009220: 7465 640a 2d70 656d 2c20 2d2d 706c 6f74  ted.-pem, --plot
+00009230: 656c 6d6d 2c20 666c 6167 206f 7074 696f  elmm, flag optio
+00009240: 6e2c 2077 6865 7468 6572 2074 6f20 7065  n, whether to pe
+00009250: 7266 6f72 6d20 616e 6420 706c 6f74 2065  rform and plot e
+00009260: 6c6d 6d20 6d69 7874 7572 6520 6d6f 6465  lmm mixture mode
+00009270: 6c69 6e67 206f 6620 7061 7261 6e6f 6d65  ling of paranome
+00009280: 204b 7320 696e 2074 6865 206d 6978 6564   Ks in the mixed
+00009290: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+000092a0: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+000092b0: 7320 7365 742c 2074 6865 2065 6c6d 6d20  s set, the elmm 
+000092c0: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
+000092d0: 206f 6620 7061 7261 6e6f 6d65 204b 7320   of paranome Ks 
+000092e0: 776f 756c 6420 6265 2070 6c6f 7474 6564  would be plotted
+000092f0: 0a2d 632c 202d 2d63 6f6d 706f 6e65 6e74  .-c, --component
+00009300: 732c 2074 6865 2072 616e 6765 206f 6620  s, the range of 
+00009310: 7468 6520 6e75 6d62 6572 206f 6620 636f  the number of co
+00009320: 6d70 6f6e 656e 7473 2074 6f20 6669 7420  mponents to fit 
+00009330: 696e 2061 6e63 686f 7220 4b73 206d 6978  in anchor Ks mix
+00009340: 7475 7265 206d 6f64 656c 696e 672c 2064  ture modeling, d
+00009350: 6566 6175 6c74 2028 312c 3429 0a2d 786c  efault (1,4).-xl
+00009360: 2c20 2d2d 786c 696d 2c20 7468 6520 7820  , --xlim, the x 
+00009370: 6178 6973 206c 696d 6974 206f 6620 4b73  axis limit of Ks
+00009380: 2064 6973 7472 6962 7574 696f 6e0a 2d79   distribution.-y
+00009390: 6c2c 202d 2d79 6c69 6d2c 2074 6865 2079  l, --ylim, the y
+000093a0: 2061 7869 7320 6c69 6d69 7420 6f66 204b   axis limit of K
+000093b0: 7320 6469 7374 7269 6275 7469 6f6e 0a2d  s distribution.-
+000093c0: 6164 6f2c 202d 2d61 646a 7573 746f 7274  ado, --adjustort
+000093d0: 686f 2c20 666c 6167 206f 7074 696f 6e2c  ho, flag option,
+000093e0: 2077 6865 7468 6572 2074 6f20 6164 6a75   whether to adju
+000093f0: 7374 2074 6865 2068 6973 746f 6772 616d  st the histogram
+00009400: 2068 6569 6768 7420 6f66 206f 7274 686f   height of ortho
+00009410: 6c6f 676f 7573 204b 7320 6173 2074 6f20  logous Ks as to 
+00009420: 6d61 7463 6820 7468 6520 6865 6967 6874  match the height
+00009430: 206f 6620 7061 7261 6c6f 676f 7573 204b   of paralogous K
+00009440: 732c 2069 6620 7468 6520 666c 6167 2077  s, if the flag w
+00009450: 6173 2073 6574 2c20 7468 6520 6164 6a75  as set, the adju
+00009460: 7374 6d65 6e74 2077 6f75 6c64 2062 6520  stment would be 
+00009470: 636f 6e64 7563 7465 640a 2d61 6466 2c20  conducted.-adf, 
+00009480: 2d2d 6164 6a75 7374 6661 6374 6f72 2c20  --adjustfactor, 
+00009490: 7468 6520 6164 6a75 7374 6d65 6e74 2066  the adjustment f
+000094a0: 6163 746f 7220 6f66 206f 7274 686f 6c6f  actor of ortholo
+000094b0: 676f 7573 204b 732c 2064 6566 6175 6c74  gous Ks, default
+000094c0: 2030 2e35 0a2d 6f61 2c20 2d2d 6f6b 616c   0.5.-oa, --okal
+000094d0: 7068 612c 2074 6865 206f 7061 6369 7479  pha, the opacity
+000094e0: 206f 6620 6f72 7468 6f6c 6f67 6f75 7320   of orthologous 
+000094f0: 4b73 2064 6973 7472 6962 7574 696f 6e20  Ks distribution 
+00009500: 696e 206d 6978 6564 2070 6c6f 742c 2064  in mixed plot, d
+00009510: 6566 6175 6c74 2030 2e35 0a2d 6661 2c20  efault 0.5.-fa, 
+00009520: 2d2d 666f 6375 7332 616c 6c2c 2073 6574  --focus2all, set
+00009530: 2066 6f63 616c 2073 7065 6369 6573 2061   focal species a
+00009540: 6e64 206c 6574 2073 7065 6369 6573 2070  nd let species p
+00009550: 6169 7220 746f 2062 6520 6265 7477 6565  air to be betwee
+00009560: 6e20 666f 6361 6c20 616e 6420 616c 6c20  n focal and all 
+00009570: 7468 6520 7265 6d61 696e 696e 6720 7370  the remaining sp
+00009580: 6563 6965 732c 2064 6566 6175 6c74 204e  ecies, default N
+00009590: 6f6e 650a 2d6b 732c 202d 2d6b 7374 7265  one.-ks, --kstre
+000095a0: 652c 2066 6c61 6720 6f70 7469 6f6e 2c20  e, flag option, 
+000095b0: 7768 6574 6865 7220 746f 2069 6e66 6572  whether to infer
+000095c0: 204b 7320 7472 6565 2c20 6966 2074 6865   Ks tree, if the
+000095d0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+000095e0: 6865 204b 7320 7472 6565 2069 6e66 6572  he Ks tree infer
+000095f0: 656e 6365 2061 6e61 6c79 7369 7320 776f  ence analysis wo
+00009600: 756c 6420 6265 2069 6e69 7469 6174 6564  uld be initiated
+00009610: 0a2d 6f63 6b2c 202d 2d6f 6e6c 7963 6f6e  .-ock, --onlycon
+00009620: 6361 746b 7374 7265 652c 2066 6c61 6720  catkstree, flag 
+00009630: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+00009640: 746f 206f 6e6c 7920 696e 6665 7220 4b73  to only infer Ks
+00009650: 2074 7265 6520 756e 6465 7220 636f 6e63   tree under conc
+00009660: 6174 656e 6174 6564 2061 6c69 676e 6d65  atenated alignme
+00009670: 6e74 2c20 6966 2074 6865 2066 6c61 6720  nt, if the flag 
+00009680: 7761 7320 7365 742c 206f 6e6c 7920 7468  was set, only th
+00009690: 6520 4b73 2074 7265 6520 756e 6465 7220  e Ks tree under 
+000096a0: 636f 6e63 6174 656e 6174 6564 2061 6c69  concatenated ali
+000096b0: 676e 6d65 6e74 2077 6f75 6c64 2062 6520  gnment would be 
+000096c0: 6361 6c63 756c 6174 6564 0a2d 6373 2c20  calculated.-cs, 
+000096d0: 2d2d 636c 6173 7369 632c 2066 6c61 6720  --classic, flag 
+000096e0: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+000096f0: 746f 2064 7261 7720 6d69 7865 6420 4b73  to draw mixed Ks
+00009700: 2070 6c6f 7420 696e 2061 2063 6c61 7373   plot in a class
+00009710: 6963 206d 616e 6e65 7220 7768 6572 6520  ic manner where 
+00009720: 7468 6520 6675 6c6c 206f 7274 686f 6c6f  the full ortholo
+00009730: 676f 7573 204b 7320 6469 7374 7269 6275  gous Ks distribu
+00009740: 7469 6f6e 2069 7320 6472 6177 6564 2c20  tion is drawed, 
+00009750: 6966 2074 6865 2066 6c61 6720 7761 7320  if the flag was 
+00009760: 7365 742c 2074 6865 2063 6c61 7373 6963  set, the classic
+00009770: 206d 6978 6564 204b 7320 706c 6f74 2077   mixed Ks plot w
+00009780: 6f75 6c64 2062 6520 6472 6177 6e0a 2d74  ould be drawn.-t
+00009790: 612c 202d 2d74 6f70 6172 726f 772c 2066  a, --toparrow, f
+000097a0: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+000097b0: 6865 7220 746f 2061 646a 7573 7420 7468  her to adjust th
+000097c0: 6520 6172 726f 7720 746f 2062 6520 6174  e arrow to be at
+000097d0: 2074 6865 2074 6f70 206f 6620 7468 6520   the top of the 
+000097e0: 706c 6f74 2c20 696e 7374 6561 6420 6f66  plot, instead of
+000097f0: 2062 6569 6e67 2063 6f6f 7264 696e 6174   being coordinat
+00009800: 6564 2061 7320 7468 6520 4b44 4520 6f66  ed as the KDE of
+00009810: 2074 6865 206f 7274 686f 6c6f 676f 7573   the orthologous
+00009820: 204b 7320 6469 7374 7269 6275 7469 6f6e   Ks distribution
+00009830: 2c20 6966 2074 6865 2066 6c61 6720 7761  , if the flag wa
+00009840: 7320 7365 742c 2074 6865 2061 7272 6f77  s set, the arrow
+00009850: 2077 6f75 6c64 2062 6520 7365 7420 6174   would be set at
+00009860: 2074 6865 2074 6f70 0a2d 6273 2c20 2d2d   the top.-bs, --
+00009870: 626f 6f74 7374 7261 702c 2074 6865 206e  bootstrap, the n
+00009880: 756d 6265 7220 6f66 2062 6f6f 7473 7472  umber of bootstr
+00009890: 6170 2072 6570 6c69 6361 7465 7320 6f66  ap replicates of
+000098a0: 206f 7274 686f 6c6f 6720 4b73 2064 6973   ortholog Ks dis
+000098b0: 7472 6962 7574 696f 6e20 696e 206d 6978  tribution in mix
+000098c0: 6564 2070 6c6f 740a 6060 600a 0a54 6865  ed plot.```..The
+000098d0: 2070 726f 6772 616d 2060 7767 6420 6d69   program `wgd mi
+000098e0: 7860 2063 616e 2072 6561 6c69 7a65 2074  x` can realize t
+000098f0: 6865 206d 6978 7475 7265 206d 6f64 656c  he mixture model
+00009900: 2063 6c75 7374 6572 696e 6720 616e 616c   clustering anal
+00009910: 7973 6973 206f 6620 2a4b 2a3c 7375 623e  ysis of *K*<sub>
+00009920: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+00009930: 7269 6275 7469 6f6e 2e0a 6060 600a 7767  ribution..```.wg
+00009940: 6420 6d69 7820 6b73 5f64 6174 6166 696c  d mix ks_datafil
+00009950: 6520 286f 7074 696f 6e29 0a2d 2d2d 2d2d  e (option).-----
+00009960: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009970: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009980: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009990: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+000099a0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d0a 2d66 2c20  -----------.-f, 
+000099b0: 2d2d 6669 6c74 6572 732c 2074 6865 2063  --filters, the c
+000099c0: 7574 6f66 6620 616c 6967 6e6d 656e 7420  utoff alignment 
+000099d0: 6c65 6e67 7468 2c20 6465 6661 756c 7420  length, default 
+000099e0: 3330 300a 2d72 2c20 2d2d 6b73 5f72 616e  300.-r, --ks_ran
+000099f0: 6765 2c20 7468 6520 4b73 2072 616e 6765  ge, the Ks range
+00009a00: 2074 6f20 6265 2063 6f6e 7369 6465 7265   to be considere
+00009a10: 642c 2064 6566 6175 6c74 2028 302c 2035  d, default (0, 5
+00009a20: 290a 2d62 2c20 2d2d 6269 6e73 2c20 7468  ).-b, --bins, th
+00009a30: 6520 6e75 6d62 6572 206f 6620 6269 6e73  e number of bins
+00009a40: 2069 6e20 4b73 2064 6973 7472 6962 7574   in Ks distribut
+00009a50: 696f 6e2c 2064 6566 6175 6c74 2035 300a  ion, default 50.
+00009a60: 2d6f 2c20 2d2d 6f75 7464 6972 2c20 7468  -o, --outdir, th
+00009a70: 6520 6f75 7470 7574 2064 6972 6563 746f  e output directo
+00009a80: 7279 2c20 6465 6661 756c 7420 7767 645f  ry, default wgd_
+00009a90: 6d69 780a 2d2d 6d65 7468 6f64 2c20 7768  mix.--method, wh
+00009aa0: 6963 6820 6d69 7874 7572 6520 6d6f 6465  ich mixture mode
+00009ab0: 6c20 746f 2075 7365 2c20 6465 6661 756c  l to use, defaul
+00009ac0: 7420 676d 6d0a 2d6e 2c20 2d2d 636f 6d70  t gmm.-n, --comp
+00009ad0: 6f6e 656e 7473 2c20 7468 6520 7261 6e67  onents, the rang
+00009ae0: 6520 6f66 2074 6865 206e 756d 6265 7220  e of the number 
+00009af0: 6f66 2063 6f6d 706f 6e65 6e74 7320 746f  of components to
+00009b00: 2066 6974 2c20 6465 6661 756c 7420 2831   fit, default (1
+00009b10: 2c20 3429 0a2d 672c 202d 2d67 616d 6d61  , 4).-g, --gamma
+00009b20: 2c20 7468 6520 6761 6d6d 6120 7061 7261  , the gamma para
+00009b30: 6d65 7465 7220 666f 7220 6267 6d6d 206d  meter for bgmm m
+00009b40: 6f64 656c 732c 2064 6566 6175 6c74 2030  odels, default 0
+00009b50: 2e30 3031 0a2d 6e69 2c20 2d2d 6e5f 696e  .001.-ni, --n_in
+00009b60: 6974 2c20 7468 6520 6e75 6d62 6572 206f  it, the number o
+00009b70: 6620 6b2d 6d65 616e 7320 696e 6974 6961  f k-means initia
+00009b80: 6c69 7a61 7469 6f6e 732c 2064 6566 6175  lizations, defau
+00009b90: 6c74 2032 3030 0a2d 6d69 2c20 2d2d 6d61  lt 200.-mi, --ma
+00009ba0: 785f 6974 6572 2c20 7468 6520 6d61 7869  x_iter, the maxi
+00009bb0: 6d75 6d20 6e75 6d62 6572 206f 6620 6974  mum number of it
+00009bc0: 6572 6174 696f 6e73 2c20 6465 6661 756c  erations, defaul
+00009bd0: 7420 3230 300a 6060 600a 0a54 6865 2070  t 200.```..The p
+00009be0: 726f 6772 616d 2060 7767 6420 7065 616b  rogram `wgd peak
+00009bf0: 6020 6361 6e20 7265 616c 697a 6520 7468  ` can realize th
+00009c00: 6520 7365 6172 6368 206f 6620 6372 6564  e search of cred
+00009c10: 6961 626c 6520 2a4b 2a3c 7375 623e 533c  iable *K*<sub>S<
+00009c20: 2f73 7562 3e20 7261 6e67 6520 7573 6564  /sub> range used
+00009c30: 2069 6e20 5747 4420 6461 7469 6e67 2e0a   in WGD dating..
+00009c40: 6060 600a 7767 6420 7065 616b 206b 735f  ```.wgd peak ks_
+00009c50: 6461 7461 6669 6c65 2028 6f70 7469 6f6e  datafile (option
+00009c60: 290a 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ).--------------
+00009c70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009c80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009c90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009ca0: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00009cb0: 2d2d 0a2d 6170 2c20 2d2d 616e 6368 6f72  --.-ap, --anchor
+00009cc0: 706f 696e 7473 2c20 7468 6520 616e 6368  points, the anch
+00009cd0: 6f72 2070 6f69 6e74 7320 6461 7461 6669  or points datafi
+00009ce0: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+00009cf0: 0a2d 736d 2c20 2d2d 7365 676d 656e 7473  .-sm, --segments
+00009d00: 2c20 7468 6520 7365 676d 656e 7473 2064  , the segments d
+00009d10: 6174 6166 696c 652c 2064 6566 6175 6c74  atafile, default
+00009d20: 204e 6f6e 650a 2d6c 652c 202d 2d6c 6973   None.-le, --lis
+00009d30: 7465 6c65 6d65 6e74 732c 2074 6865 206c  telements, the l
+00009d40: 6973 7473 6567 6d65 6e74 7320 6461 7461  istsegments data
+00009d50: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+00009d60: 6e65 200a 2d6d 702c 202d 2d6d 756c 7469  ne .-mp, --multi
+00009d70: 706c 6963 6f6e 7061 6972 732c 2074 6865  pliconpairs, the
+00009d80: 206d 756c 7469 706c 6963 6f6e 7061 6972   multipliconpair
+00009d90: 7320 6461 7461 6669 6c65 2c20 6465 6661  s datafile, defa
+00009da0: 756c 7420 4e6f 6e65 0a2d 6f2c 202d 2d6f  ult None.-o, --o
+00009db0: 7574 6469 722c 2074 6865 206f 7574 7075  utdir, the outpu
+00009dc0: 7420 6469 7265 6374 6f72 792c 2064 6566  t directory, def
+00009dd0: 6175 6c74 2077 6764 5f70 6561 6b0a 2d61  ault wgd_peak.-a
+00009de0: 662c 202d 2d61 6c69 676e 6669 6c74 6572  f, --alignfilter
+00009df0: 2c20 6375 746f 6666 2066 6f72 2061 6c69  , cutoff for ali
+00009e00: 676e 6d65 6e74 2069 6465 6e74 6974 792c  gnment identity,
+00009e10: 206c 656e 6774 6820 616e 6420 636f 7665   length and cove
+00009e20: 7261 6765 2c20 6465 6661 756c 7420 302e  rage, default 0.
+00009e30: 302c 2030 2c20 302e 300a 2d72 2c20 2d2d  0, 0, 0.0.-r, --
+00009e40: 6b73 7261 6e67 652c 2072 616e 6765 206f  ksrange, range o
+00009e50: 6620 4b73 2074 6f20 6265 2061 6e61 6c79  f Ks to be analy
+00009e60: 7a65 642c 2064 6566 6175 6c74 2028 302c  zed, default (0,
+00009e70: 2035 290a 2d62 772c 202d 2d62 696e 5f77   5).-bw, --bin_w
+00009e80: 6964 7468 2c20 6261 6e64 7769 6474 6820  idth, bandwidth 
+00009e90: 6f66 204b 7320 6469 7374 7269 6275 7469  of Ks distributi
+00009ea0: 6f6e 2c20 6465 6661 756c 7420 302e 310a  on, default 0.1.
+00009eb0: 2d69 632c 202d 2d77 6569 6768 7473 5f6f  -ic, --weights_o
+00009ec0: 7574 6c69 6572 735f 696e 636c 7564 6564  utliers_included
+00009ed0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+00009ee0: 6865 7468 6572 2074 6f20 696e 636c 7564  hether to includ
+00009ef0: 6520 4b73 206f 7574 6c69 6572 732c 2069  e Ks outliers, i
+00009f00: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+00009f10: 6574 2c20 4b73 206f 7574 6c69 6572 7320  et, Ks outliers 
+00009f20: 776f 756c 6420 6265 2069 6e63 6c75 6465  would be include
+00009f30: 6420 696e 2074 6865 2061 6e61 6c79 7369  d in the analysi
+00009f40: 730a 2d6d 2c20 2d2d 6d65 7468 6f64 2c20  s.-m, --method, 
+00009f50: 7768 6963 6820 6d69 7874 7572 6520 6d6f  which mixture mo
+00009f60: 6465 6c20 746f 2075 7365 2c20 6465 6661  del to use, defa
+00009f70: 756c 7420 676d 6d0a 2d2d 7365 6564 2c20  ult gmm.--seed, 
+00009f80: 7261 6e64 6f6d 2073 6565 6420 6769 7665  random seed give
+00009f90: 6e20 746f 2069 6e69 7469 616c 697a 6174  n to initializat
+00009fa0: 696f 6e2c 2064 6566 6175 6c74 2032 3335  ion, default 235
+00009fb0: 3238 3930 0a2d 6569 2c20 2d2d 656d 5f69  2890.-ei, --em_i
+00009fc0: 7465 722c 2074 6865 206e 756d 6265 7220  ter, the number 
+00009fd0: 6f66 2045 4d20 6974 6572 6174 696f 6e73  of EM iterations
+00009fe0: 2074 6f20 7065 7266 6f72 6d2c 2064 6566   to perform, def
+00009ff0: 6175 6c74 2032 3030 0a2d 6e69 2c20 2d2d  ault 200.-ni, --
+0000a000: 6e5f 696e 6974 2c20 7468 6520 6e75 6d62  n_init, the numb
+0000a010: 6572 206f 6620 6b2d 6d65 616e 7320 696e  er of k-means in
+0000a020: 6974 6961 6c69 7a61 7469 6f6e 732c 2064  itializations, d
+0000a030: 6566 6175 6c74 2032 3030 0a2d 6e2c 202d  efault 200.-n, -
+0000a040: 2d63 6f6d 706f 6e65 6e74 732c 2074 6865  -components, the
+0000a050: 2072 616e 6765 206f 6620 7468 6520 6e75   range of the nu
+0000a060: 6d62 6572 206f 6620 636f 6d70 6f6e 656e  mber of componen
+0000a070: 7473 2074 6f20 6669 742c 2064 6566 6175  ts to fit, defau
+0000a080: 6c74 2028 312c 2034 290a 2d67 2c20 2d2d  lt (1, 4).-g, --
+0000a090: 6761 6d6d 612c 2074 6865 2067 616d 6d61  gamma, the gamma
+0000a0a0: 2070 6172 616d 6574 6572 2066 6f72 2062   parameter for b
+0000a0b0: 676d 6d20 6d6f 6465 6c73 2c20 6465 6661  gmm models, defa
+0000a0c0: 756c 7420 3165 2d33 0a2d 2d62 6f6f 7473  ult 1e-3.--boots
+0000a0d0: 2c20 7468 6520 6e75 6d62 6572 206f 6620  , the number of 
+0000a0e0: 626f 6f74 7374 7261 7020 7265 706c 6963  bootstrap replic
+0000a0f0: 6174 6573 206f 6620 6b64 652c 2064 6566  ates of kde, def
+0000a100: 6175 6c74 2032 3030 0a2d 2d77 6569 6768  ault 200.--weigh
+0000a110: 7465 642c 2066 6c61 6720 6f70 7469 6f6e  ted, flag option
+0000a120: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
+0000a130: 206e 6f64 652d 7765 6967 6874 6564 206d   node-weighted m
+0000a140: 6574 686f 6420 6f66 2064 652d 7265 6475  ethod of de-redu
+0000a150: 6e64 616e 6379 2c20 6966 2074 6865 2066  ndancy, if the f
+0000a160: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+0000a170: 206e 6f64 652d 7765 6967 6874 6564 206d   node-weighted m
+0000a180: 6574 686f 6420 776f 756c 6420 6265 2075  ethod would be u
+0000a190: 7365 640a 2d70 2c20 2d2d 706c 6f74 2c20  sed.-p, --plot, 
+0000a1a0: 7468 6520 706c 6f74 7469 6e67 206d 6574  the plotting met
+0000a1b0: 686f 6420 746f 2062 6520 7573 6564 2c20  hod to be used, 
+0000a1c0: 6465 6661 756c 7420 6964 656e 7469 6361  default identica
+0000a1d0: 6c0a 2d62 6d2c 202d 2d62 775f 6d65 7468  l.-bm, --bw_meth
+0000a1e0: 6f64 2c20 7468 6520 6261 6e64 7769 6474  od, the bandwidt
+0000a1f0: 6820 6d65 7468 6f64 2074 6f20 6265 2075  h method to be u
+0000a200: 7365 6420 696e 2061 6e61 6c79 7a69 6e67  sed in analyzing
+0000a210: 2074 6865 2070 6561 6b20 6f66 2057 4744   the peak of WGD
+0000a220: 2064 6174 6573 2c20 6465 6661 756c 7420   dates, default 
+0000a230: 7369 6c76 6572 6d61 6e0a 2d2d 6e5f 6d65  silverman.--n_me
+0000a240: 646f 6964 732c 2074 6865 206e 756d 6265  doids, the numbe
+0000a250: 7220 6f66 206d 6564 6f69 6473 2074 6f20  r of medoids to 
+0000a260: 6669 742c 2064 6566 6175 6c74 2032 0a2d  fit, default 2.-
+0000a270: 6b6d 2c20 2d2d 6b64 656d 6574 686f 642c  km, --kdemethod,
+0000a280: 2074 6865 206b 6465 206d 6574 686f 6420   the kde method 
+0000a290: 746f 2062 6520 7573 6564 2069 6e20 616e  to be used in an
+0000a2a0: 616c 797a 696e 6720 7468 6520 7065 616b  alyzing the peak
+0000a2b0: 206f 6620 5747 4420 6461 7465 732c 206b   of WGD dates, k
+0000a2c0: 6d65 646f 6964 7320 616e 616c 7973 6973  medoids analysis
+0000a2d0: 206f 7220 7468 6520 6261 7369 6320 4b73   or the basic Ks
+0000a2e0: 2070 6c6f 7474 696e 672c 2064 6566 6175   plotting, defau
+0000a2f0: 6c74 2073 6369 7079 0a2d 2d6e 5f63 6c75  lt scipy.--n_clu
+0000a300: 7374 6572 732c 2074 6865 206e 756d 6265  sters, the numbe
+0000a310: 7220 6f66 2063 6c75 7374 6572 7320 746f  r of clusters to
+0000a320: 2070 6c6f 7420 456c 626f 7720 6c6f 7373   plot Elbow loss
+0000a330: 2066 756e 6374 696f 6e2c 2064 6566 6175   function, defau
+0000a340: 6c74 2035 0a2d 6764 2c20 2d2d 6775 6964  lt 5.-gd, --guid
+0000a350: 652c 2074 6865 2072 6567 696d 6520 7265  e, the regime re
+0000a360: 7369 6469 6e67 2061 6e63 686f 7273 2c20  siding anchors, 
+0000a370: 6465 6661 756c 7420 5365 676d 656e 740a  default Segment.
+0000a380: 2d70 7263 742c 202d 2d70 726f 6d69 6e65  -prct, --promine
+0000a390: 6e63 655f 6375 746f 6666 2c20 7468 6520  nce_cutoff, the 
+0000a3a0: 7072 6f6d 696e 656e 6365 2063 7574 6f66  prominence cutof
+0000a3b0: 6620 6f66 2061 6363 6570 7461 626c 6520  f of acceptable 
+0000a3c0: 7065 616b 7320 696e 2070 6561 6b20 6669  peaks in peak fi
+0000a3d0: 6e64 696e 6720 7374 6570 732c 2064 6566  nding steps, def
+0000a3e0: 6175 6c74 2030 2e31 0a2d 7268 2c20 2d2d  ault 0.1.-rh, --
+0000a3f0: 7265 6c5f 6865 6967 6874 2c20 7468 6520  rel_height, the 
+0000a400: 7265 6c61 7469 7665 2068 6569 6768 7420  relative height 
+0000a410: 6174 2077 6869 6368 2074 6865 2070 6561  at which the pea
+0000a420: 6b20 7769 6474 6820 6973 206d 6561 7375  k width is measu
+0000a430: 7265 642c 2064 6566 6175 6c74 2030 2e34  red, default 0.4
+0000a440: 0a2d 6b64 2c20 2d2d 6b73 746f 6461 7465  .-kd, --kstodate
+0000a450: 2c20 7468 6520 7261 6e67 6520 6f66 204b  , the range of K
+0000a460: 7320 746f 2062 6520 6461 7465 6420 696e  s to be dated in
+0000a470: 2068 6575 7269 7374 6963 2073 6561 7263   heuristic searc
+0000a480: 682c 2064 6566 6175 6c74 2028 302e 352c  h, default (0.5,
+0000a490: 2031 2e35 290a 2d78 6c2c 202d 2d78 6c69   1.5).-xl, --xli
+0000a4a0: 6d2c 2074 6865 2078 2061 7869 7320 6c69  m, the x axis li
+0000a4b0: 6d69 7420 6f66 2047 4d4d 204b 7320 6469  mit of GMM Ks di
+0000a4c0: 7374 7269 6275 7469 6f6e 0a2d 796c 2c20  stribution.-yl, 
+0000a4d0: 2d2d 796c 696d 2c20 7468 6520 7920 6178  --ylim, the y ax
+0000a4e0: 6973 206c 696d 6974 206f 6620 474d 4d20  is limit of GMM 
+0000a4f0: 4b73 2064 6973 7472 6962 7574 696f 6e0a  Ks distribution.
+0000a500: 2d2d 6d61 6e75 616c 7365 742c 2066 6c61  --manualset, fla
+0000a510: 6720 6f70 7469 6f6e 2c20 7768 6574 6865  g option, whethe
+0000a520: 7220 746f 206f 7574 7075 7420 616e 6368  r to output anch
+0000a530: 6f72 2070 6169 7273 2077 6974 6820 6d61  or pairs with ma
+0000a540: 6e75 616c 6c79 2073 6574 204b 7320 7261  nually set Ks ra
+0000a550: 6e67 652c 2069 6620 7468 6520 666c 6167  nge, if the flag
+0000a560: 2077 6173 2073 6574 2c20 6d61 6e75 616c   was set, manual
+0000a570: 6c79 2073 6574 204b 7320 7261 6e67 6520  ly set Ks range 
+0000a580: 776f 756c 6420 6265 206f 7574 7075 7474  would be outputt
+0000a590: 6564 0a2d 2d63 692c 2074 6865 2063 6f6e  ed.--ci, the con
+0000a5a0: 6669 6465 6e63 6520 6c65 7665 6c20 6f66  fidence level of
+0000a5b0: 206c 6f67 2d6e 6f72 6d61 6c20 6469 7374   log-normal dist
+0000a5c0: 7269 6275 7469 6f6e 2074 6f20 6461 7465  ribution to date
+0000a5d0: 2c20 6465 6661 756c 7420 3935 0a2d 2d68  , default 95.--h
+0000a5e0: 6472 2c20 7468 6520 6869 6768 6573 7420  dr, the highest 
+0000a5f0: 6465 6e73 6974 7920 7265 6769 6f6e 2028  density region (
+0000a600: 4844 5229 2061 7070 6c69 6564 2069 6e20  HDR) applied in 
+0000a610: 7468 6520 7365 676d 656e 742d 6775 6964  the segment-guid
+0000a620: 6564 2061 6e63 686f 7220 7061 6972 204b  ed anchor pair K
+0000a630: 7320 6469 7374 7269 6275 7469 6f6e 2c20  s distribution, 
+0000a640: 6465 6661 756c 7420 3935 0a2d 2d68 6575  default 95.--heu
+0000a650: 7269 7374 6963 2c20 666c 6167 206f 7074  ristic, flag opt
+0000a660: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+0000a670: 696e 6974 6961 7465 2068 6575 7269 7374  initiate heurist
+0000a680: 6963 206d 6574 686f 6420 6f66 2064 6566  ic method of def
+0000a690: 696e 696e 6720 4349 2066 6f72 2064 6174  ining CI for dat
+0000a6a0: 696e 672c 2069 6620 7468 6520 666c 6167  ing, if the flag
+0000a6b0: 2077 6173 2073 6574 2c20 7468 6520 6865   was set, the he
+0000a6c0: 7572 6973 7469 6320 6d65 7468 6f64 2077  uristic method w
+0000a6d0: 6f75 6c64 2062 6520 696e 6974 6961 7465  ould be initiate
+0000a6e0: 640a 2d6b 632c 202d 2d6b 7363 7574 6f66  d.-kc, --kscutof
+0000a6f0: 662c 2074 6865 204b 7320 7361 7475 7261  f, the Ks satura
+0000a700: 7469 6f6e 2063 7574 6f66 6620 696e 2064  tion cutoff in d
+0000a710: 6174 696e 672c 2064 6566 6175 6c74 2035  ating, default 5
+0000a720: 0a2d 2d6b 6565 7074 6d70 6669 672c 2066  .--keeptmpfig, f
+0000a730: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+0000a740: 6865 7220 746f 206b 6565 7020 7465 6d70  her to keep temp
+0000a750: 6f72 6172 7920 6669 6775 7265 7320 696e  orary figures in
+0000a760: 2070 6561 6b20 6669 6e64 696e 6720 7072   peak finding pr
+0000a770: 6f63 6573 732c 2069 6620 7468 6520 666c  ocess, if the fl
+0000a780: 6167 2077 6173 2073 6574 2c20 7468 6f73  ag was set, thos
+0000a790: 6520 6669 6775 7265 7320 776f 756c 6420  e figures would 
+0000a7a0: 6265 206b 6570 740a 6060 600a 0a54 6865  be kept.```..The
+0000a7b0: 2070 726f 6772 616d 2060 7767 6420 7379   program `wgd sy
+0000a7c0: 6e60 2063 616e 2072 6561 6c69 7a65 2074  n` can realize t
+0000a7d0: 6865 2069 6e74 7261 2d20 616e 6420 696e  he intra- and in
+0000a7e0: 7465 722d 7370 6563 6966 6963 2073 796e  ter-specific syn
+0000a7f0: 7465 6e79 2069 6e66 6572 656e 6365 2e0a  teny inference..
+0000a800: 6060 600a 7767 6420 7379 6e20 6661 6d69  ```.wgd syn fami
+0000a810: 6c69 6573 2067 6666 7320 286f 7074 696f  lies gffs (optio
+0000a820: 6e29 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  n).-------------
+0000a830: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a840: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a850: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a860: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000a870: 2d2d 2d0a 2d6b 732c 202d 2d6b 735f 6469  ---.-ks, --ks_di
+0000a880: 7374 7269 6275 7469 6f6e 2c20 6b73 2064  stribution, ks d
+0000a890: 6973 7472 6962 7574 696f 6e20 6461 7461  istribution data
+0000a8a0: 6669 6c65 2c20 6465 6661 756c 7420 4e6f  file, default No
+0000a8b0: 6e65 0a2d 6f2c 202d 2d6f 7574 6469 722c  ne.-o, --outdir,
+0000a8c0: 2074 6865 206f 7574 7075 7420 6469 7265   the output dire
+0000a8d0: 6374 6f72 792c 2064 6566 6175 6c74 2077  ctory, default w
+0000a8e0: 6764 5f73 796e 0a2d 662c 202d 2d66 6561  gd_syn.-f, --fea
+0000a8f0: 7475 7265 2c20 7468 6520 6665 6174 7572  ture, the featur
+0000a900: 6520 666f 7220 7061 7273 696e 6720 6765  e for parsing ge
+0000a910: 6e65 2049 4473 2066 726f 6d20 4746 4620  ne IDs from GFF 
+0000a920: 6669 6c65 732c 2064 6566 6175 6c74 2067  files, default g
+0000a930: 656e 650a 2d61 2c20 2d2d 6174 7472 6962  ene.-a, --attrib
+0000a940: 7574 652c 2074 6865 2061 7474 7269 6275  ute, the attribu
+0000a950: 7465 2066 6f72 2070 6172 7369 6e67 2074  te for parsing t
+0000a960: 6865 2067 656e 6520 4944 7320 6672 6f6d  he gene IDs from
+0000a970: 2074 6865 2047 4646 2066 696c 6573 2c20   the GFF files, 
+0000a980: 6465 6661 756c 7420 4944 0a2d 6174 672c  default ID.-atg,
+0000a990: 202d 2d61 6464 6974 696f 6e61 6c67 6666   --additionalgff
+0000a9a0: 696e 666f 2c20 7468 6520 6665 6174 7572  info, the featur
+0000a9b0: 6520 616e 6420 6174 7472 6962 7574 6520  e and attribute 
+0000a9c0: 696e 666f 726d 6174 696f 6e20 6f66 2061  information of a
+0000a9d0: 6464 6974 696f 6e61 6c20 6766 6633 2066  dditional gff3 f
+0000a9e0: 696c 6573 2069 6620 6469 6666 6572 656e  iles if differen
+0000a9f0: 7420 696e 2074 6865 2066 6f72 6d61 7420  t in the format 
+0000aa00: 6f66 2028 6665 6174 7572 653b 6174 7472  of (feature;attr
+0000aa10: 6962 7574 6529 272c 2064 6566 6175 6c74  ibute)', default
+0000aa20: 204e 6f6e 650a 2d6d 6c2c 202d 2d6d 696e   None.-ml, --min
+0000aa30: 6c65 6e2c 2074 6865 206d 696e 696d 756d  len, the minimum
+0000aa40: 206c 656e 6774 6820 6f66 2061 2073 6361   length of a sca
+0000aa50: 6666 6f6c 6420 746f 2062 6520 696e 636c  ffold to be incl
+0000aa60: 7564 6564 2069 6e20 646f 7470 6c6f 742c  uded in dotplot,
+0000aa70: 2064 6566 6175 6c74 202d 312c 2069 6620   default -1, if 
+0000aa80: 2d31 2077 6173 2073 6574 2c20 7468 6520  -1 was set, the 
+0000aa90: 3130 2520 6f66 2074 6865 206c 6f6e 6765  10% of the longe
+0000aaa0: 7374 2073 6361 6666 6f6c 6420 776f 756c  st scaffold woul
+0000aab0: 6420 6265 2073 6574 0a2d 6d73 2c20 2d2d  d be set.-ms, --
+0000aac0: 6d61 7873 697a 652c 2074 6865 206d 6178  maxsize, the max
+0000aad0: 696d 756d 2066 616d 696c 7920 7369 7a65  imum family size
+0000aae0: 2074 6f20 6265 2069 6e63 6c75 6465 642c   to be included,
+0000aaf0: 2064 6566 6175 6c74 2032 3030 0a2d 722c   default 200.-r,
+0000ab00: 202d 2d6b 735f 7261 6e67 652c 2074 6865   --ks_range, the
+0000ab10: 204b 7320 7261 6e67 6520 696e 2063 6f6c   Ks range in col
+0000ab20: 6f72 6564 2064 6f74 706c 6f74 2c20 6465  ored dotplot, de
+0000ab30: 6661 756c 7420 2830 2c20 3529 0a2d 2d69  fault (0, 5).--i
+0000ab40: 6164 686f 7265 5f6f 7074 696f 6e73 2c20  adhore_options, 
+0000ab50: 7468 6520 7061 7261 6d65 7465 7220 7365  the parameter se
+0000ab60: 7474 696e 6720 696e 2069 6164 686f 7265  tting in iadhore
+0000ab70: 2c20 6465 6661 756c 7420 6173 2061 2073  , default as a s
+0000ab80: 7472 696e 6720 6f66 206c 656e 6774 6820  tring of length 
+0000ab90: 7a65 726f 0a2d 6d67 2c20 2d2d 6d69 6e73  zero.-mg, --mins
+0000aba0: 6567 6c65 6e2c 2074 6865 206d 696e 696d  eglen, the minim
+0000abb0: 756d 206c 656e 6774 6820 6f66 2073 6567  um length of seg
+0000abc0: 6d65 6e74 7320 746f 2069 6e63 6c75 6465  ments to include
+0000abd0: 2069 6e20 7261 7469 6f20 6966 203c 3d20   in ratio if <= 
+0000abe0: 312c 2064 6566 6175 6c74 2031 3030 3030  1, default 10000
+0000abf0: 0a2d 6b72 2c20 2d2d 6b65 6570 7265 6475  .-kr, --keepredu
+0000ac00: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
+0000ac10: 7768 6574 6865 7220 746f 206b 6565 7020  whether to keep 
+0000ac20: 7265 6475 6e64 616e 7420 6d75 6c74 6970  redundant multip
+0000ac30: 6c69 636f 6e73 2c20 6966 2074 6865 2066  licons, if the f
+0000ac40: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+0000ac50: 2072 6564 756e 6461 6e74 206d 756c 7469   redundant multi
+0000ac60: 706c 6963 6f6e 7320 776f 756c 6420 6265  plicons would be
+0000ac70: 206b 6570 740a 2d6d 676e 2c20 2d2d 6d69   kept.-mgn, --mi
+0000ac80: 6e67 656e 656e 756d 2c20 7468 6520 6d69  ngenenum, the mi
+0000ac90: 6e69 6d75 6d20 6e75 6d62 6572 206f 6620  nimum number of 
+0000aca0: 6765 6e65 7320 666f 7220 6120 7365 676d  genes for a segm
+0000acb0: 656e 7420 746f 2062 6520 636f 6e73 6964  ent to be consid
+0000acc0: 6572 6564 2c20 6465 6661 756c 7420 3330  ered, default 30
+0000acd0: 0a2d 6473 2c20 2d2d 646f 7473 697a 652c  .-ds, --dotsize,
+0000ace0: 2074 6865 2064 6f74 2073 697a 6520 696e   the dot size in
+0000acf0: 2064 6f74 2070 6c6f 742c 2064 6566 6175   dot plot, defau
+0000ad00: 6c74 2030 2e33 0a2d 6161 2c20 2d2d 6170  lt 0.3.-aa, --ap
+0000ad10: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
+0000ad20: 7479 206f 6620 616e 6368 6f72 2064 6f74  ty of anchor dot
+0000ad30: 732c 2064 6566 6175 6c74 2031 0a2d 6861  s, default 1.-ha
+0000ad40: 2c20 2d2d 686f 616c 7068 612c 2074 6865  , --hoalpha, the
+0000ad50: 206f 7061 6369 7479 206f 6620 686f 6d6f   opacity of homo
+0000ad60: 6c6f 6720 646f 7473 2c20 6465 6661 756c  log dots, defaul
+0000ad70: 7420 300a 2d73 7274 2c20 2d2d 7368 6f77  t 0.-srt, --show
+0000ad80: 7265 616c 7469 636b 2c20 666c 6167 206f  realtick, flag o
+0000ad90: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+0000ada0: 6f20 7368 6f77 2074 6865 2072 6561 6c20  o show the real 
+0000adb0: 7469 636b 2069 6e20 6765 6e65 7320 6f72  tick in genes or
+0000adc0: 2062 6173 6573 2c20 6966 2074 6865 2066   bases, if the f
+0000add0: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+0000ade0: 2072 6561 6c20 7469 636b 2077 6f75 6c64   real tick would
+0000adf0: 2062 6520 7368 6f77 6564 0a2d 746c 732c   be showed.-tls,
+0000ae00: 202d 2d74 6963 6b6c 6162 656c 7369 7a65   --ticklabelsize
+0000ae10: 2c20 7468 6520 6c61 6265 6c20 7369 7a65  , the label size
+0000ae20: 206f 6620 7469 636b 2c20 6465 6661 756c   of tick, defaul
+0000ae30: 7420 350a 2d67 722c 202d 2d67 6973 7472  t 5.-gr, --gistr
+0000ae40: 622c 2066 6c61 6720 6f70 7469 6f6e 2c20  b, flag option, 
+0000ae50: 7768 6574 6865 7220 746f 2075 7365 2067  whether to use g
+0000ae60: 6973 745f 7261 696e 626f 7720 6173 2063  ist_rainbow as c
+0000ae70: 6f6c 6f72 206d 6170 206f 6620 646f 7470  olor map of dotp
+0000ae80: 6c6f 740a 2d6e 2c20 2d2d 6e74 6872 6561  lot.-n, --nthrea
+0000ae90: 6473 2c20 7468 6520 6e75 6d62 6572 206f  ds, the number o
+0000aea0: 6620 7468 7265 6164 7320 746f 2075 7365  f threads to use
+0000aeb0: 2069 6e20 7379 6e74 656e 7920 696e 6665   in synteny infe
+0000aec0: 7265 6e63 652c 2064 6566 6175 6c74 2034  rence, default 4
+0000aed0: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+0000aee0: 6d20 6077 6764 2076 697a 6020 6361 6e20  m `wgd viz` can 
+0000aef0: 7265 616c 697a 6520 7468 6520 7669 7375  realize the visu
+0000af00: 616c 697a 6174 696f 6e20 6f66 202a 4b2a  alization of *K*
+0000af10: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+0000af20: 2064 6973 7472 6962 7574 696f 6e20 616e   distribution an
+0000af30: 6420 7379 6e74 656e 792e 0a60 6060 0a77  d synteny..```.w
+0000af40: 6764 2076 697a 2028 6f70 7469 6f6e 290a  gd viz (option).
+0000af50: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000af60: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000af70: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000af80: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000af90: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+0000afa0: 0a2d 642c 202d 2d64 6174 6166 696c 652c  .-d, --datafile,
+0000afb0: 2074 6865 204b 7320 6461 7461 6669 6c65   the Ks datafile
+0000afc0: 2c20 6465 6661 756c 7420 4e6f 6e65 0a2d  , default None.-
+0000afd0: 6f2c 202d 2d6f 7574 6469 722c 2074 6865  o, --outdir, the
+0000afe0: 206f 7574 7075 7420 6469 7265 6374 6f72   output director
+0000aff0: 792c 2064 6566 6175 6c74 2077 6764 5f76  y, default wgd_v
+0000b000: 697a 0a2d 7372 2c20 2d2d 7370 6169 722c  iz.-sr, --spair,
+0000b010: 2074 6865 2073 7065 6369 6573 2070 6169   the species pai
+0000b020: 7220 746f 2062 6520 706c 6f74 7465 642c  r to be plotted,
+0000b030: 2064 6566 6175 6c74 204e 6f6e 652c 2074   default None, t
+0000b040: 6869 7320 6f70 7469 6f6e 2063 616e 2062  his option can b
+0000b050: 6520 7072 6f76 6964 6564 206d 756c 7469  e provided multi
+0000b060: 706c 6520 7469 6d65 730a 2d66 612c 202d  ple times.-fa, -
+0000b070: 2d66 6f63 7573 3261 6c6c 2c20 7365 7420  -focus2all, set 
+0000b080: 666f 6361 6c20 7370 6563 6965 7320 616e  focal species an
+0000b090: 6420 6c65 7420 7370 6563 6965 7320 7061  d let species pa
+0000b0a0: 6972 2074 6f20 6265 2062 6574 7765 656e  ir to be between
+0000b0b0: 2066 6f63 616c 2061 6e64 2061 6c6c 2074   focal and all t
+0000b0c0: 6865 2072 656d 6169 6e69 6e67 2073 7065  he remaining spe
+0000b0d0: 6369 6573 2c20 6465 6661 756c 7420 4e6f  cies, default No
+0000b0e0: 6e65 0a2d 6773 2c20 2d2d 6773 6d61 702c  ne.-gs, --gsmap,
+0000b0f0: 2074 6865 2067 656e 6520 6e61 6d65 2d73   the gene name-s
+0000b100: 7065 6369 6573 206e 616d 6520 6d61 702c  pecies name map,
+0000b110: 2064 6566 6175 6c74 204e 6f6e 650a 2d73   default None.-s
+0000b120: 702c 202d 2d73 7065 6369 6573 7472 6565  p, --speciestree
+0000b130: 2c20 7468 6520 7370 6563 6965 7320 7472  , the species tr
+0000b140: 6565 2074 6f20 7065 7266 6f72 6d20 7261  ee to perform ra
+0000b150: 7465 2063 6f72 7265 6374 696f 6e2c 2064  te correction, d
+0000b160: 6566 6175 6c74 204e 6f6e 652c 2069 6620  efault None, if 
+0000b170: 4e6f 6e65 2077 6173 2067 6976 656e 2c20  None was given, 
+0000b180: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+0000b190: 696f 6e20 616e 616c 7973 6973 2077 6f75  ion analysis wou
+0000b1a0: 6c64 2062 6520 6361 6c6c 6564 206f 6666  ld be called off
+0000b1b0: 0a2d 706b 2c20 2d2d 706c 6f74 6b64 652c  .-pk, --plotkde,
+0000b1c0: 2066 6c61 6720 6f70 7469 6f6e 2c20 7768   flag option, wh
+0000b1d0: 6574 6865 7220 746f 2070 6c6f 7420 6b64  ether to plot kd
+0000b1e0: 6520 6375 7276 6520 7570 6f6e 2068 6973  e curve upon his
+0000b1f0: 746f 6772 616d 2c20 6966 2074 6865 2066  togram, if the f
+0000b200: 6c61 6720 7761 7320 7365 742c 206b 6465  lag was set, kde
+0000b210: 2063 7572 7665 2077 6f75 6c64 2062 6520   curve would be 
+0000b220: 6164 6465 640a 2d72 772c 202d 2d72 6577  added.-rw, --rew
+0000b230: 6569 6768 742c 2066 6c61 6720 6f70 7469  eight, flag opti
+0000b240: 6f6e 2c20 7768 6574 6865 7220 746f 2072  on, whether to r
+0000b250: 6563 616c 6375 6c61 7465 2074 6865 2077  ecalculate the w
+0000b260: 6569 6768 7420 7065 7220 7370 6563 6965  eight per specie
+0000b270: 7320 7061 6972 2c20 6966 2074 6865 2066  s pair, if the f
+0000b280: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+0000b290: 2077 6569 6768 7420 776f 756c 6420 6265   weight would be
+0000b2a0: 2072 6563 616c 6375 6c61 7465 640a 2d6f   recalculated.-o
+0000b2b0: 722c 202d 2d6f 6e6c 7972 6f6f 746f 7574  r, --onlyrootout
+0000b2c0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+0000b2d0: 6865 7468 6572 2074 6f20 6f6e 6c79 2063  hether to only c
+0000b2e0: 6f6e 6475 6374 2072 6174 6520 636f 7272  onduct rate corr
+0000b2f0: 6563 7469 6f6e 2075 7369 6e67 2074 6865  ection using the
+0000b300: 206f 7574 6772 6f75 7020 6174 2072 6f6f   outgroup at roo
+0000b310: 7420 6173 206f 7574 6772 6f75 702c 2069  t as outgroup, i
+0000b320: 6620 7468 6520 666c 6167 2077 6173 2073  f the flag was s
+0000b330: 6574 2c20 6f6e 6c79 2074 6865 206f 7574  et, only the out
+0000b340: 6772 6f75 7020 6174 2072 6f6f 7420 776f  group at root wo
+0000b350: 756c 6420 6265 2075 7365 6420 6173 206f  uld be used as o
+0000b360: 7574 6772 6f75 700a 2d69 7465 722c 202d  utgroup.-iter, -
+0000b370: 2d65 6d5f 6974 6572 6174 696f 6e73 2c20  -em_iterations, 
+0000b380: 7468 6520 6d61 7869 6d75 6d20 454d 2069  the maximum EM i
+0000b390: 7465 7261 7469 6f6e 732c 2064 6566 6175  terations, defau
+0000b3a0: 6c74 2032 3030 0a2d 696e 6974 2c20 2d2d  lt 200.-init, --
+0000b3b0: 656d 5f69 6e69 7469 616c 697a 6174 696f  em_initializatio
+0000b3c0: 6e73 2c20 7468 6520 6d61 7869 6d75 6d20  ns, the maximum 
+0000b3d0: 454d 2069 6e69 7469 616c 697a 6174 696f  EM initializatio
+0000b3e0: 6e73 2c20 6465 6661 756c 7420 3230 300a  ns, default 200.
+0000b3f0: 2d70 7263 742c 202d 2d70 726f 6d69 6e65  -prct, --promine
+0000b400: 6e63 655f 6375 746f 6666 2c20 7468 6520  nce_cutoff, the 
+0000b410: 7072 6f6d 696e 656e 6365 2063 7574 6f66  prominence cutof
+0000b420: 6620 6f66 2061 6363 6570 7461 626c 6520  f of acceptable 
+0000b430: 7065 616b 732c 2064 6566 6175 6c74 2030  peaks, default 0
+0000b440: 2e31 0a2d 7268 2c20 2d2d 7265 6c5f 6865  .1.-rh, --rel_he
+0000b450: 6967 6874 2c20 7468 6520 7265 6c61 7469  ight, the relati
+0000b460: 7665 2068 6569 6768 7420 6174 2077 6869  ve height at whi
+0000b470: 6368 2074 6865 2070 6561 6b20 7769 6474  ch the peak widt
+0000b480: 6820 6973 206d 6561 7375 7265 642c 2064  h is measured, d
+0000b490: 6566 6175 6c74 2030 2e34 0a2d 736d 2c20  efault 0.4.-sm, 
+0000b4a0: 2d2d 7365 676d 656e 7473 2c20 7468 6520  --segments, the 
+0000b4b0: 7365 676d 656e 7473 2064 6174 6166 696c  segments datafil
+0000b4c0: 652c 2064 6566 6175 6c74 204e 6f6e 650a  e, default None.
+0000b4d0: 2d6d 6c2c 202d 2d6d 696e 6c65 6e2c 2074  -ml, --minlen, t
+0000b4e0: 6865 206d 696e 696d 756d 206c 656e 6774  he minimum lengt
+0000b4f0: 6820 6f66 2061 2073 6361 6666 6f6c 6420  h of a scaffold 
+0000b500: 746f 2062 6520 696e 636c 7564 6564 2069  to be included i
+0000b510: 6e20 646f 7470 6c6f 742c 2064 6566 6175  n dotplot, defau
+0000b520: 6c74 202d 312c 2069 6620 2d31 2077 6173  lt -1, if -1 was
+0000b530: 2073 6574 2c20 7468 6520 3130 2520 6f66   set, the 10% of
+0000b540: 2074 6865 206c 6f6e 6765 7374 2073 6361   the longest sca
+0000b550: 6666 6f6c 6473 2077 696c 6c20 6265 2073  ffolds will be s
+0000b560: 6574 0a2d 6d73 2c20 2d2d 6d61 7873 697a  et.-ms, --maxsiz
+0000b570: 652c 2074 6865 206d 6178 696d 756d 2066  e, the maximum f
+0000b580: 616d 696c 7920 7369 7a65 2074 6f20 6265  amily size to be
+0000b590: 2069 6e63 6c75 6465 642c 2064 6566 6175   included, defau
+0000b5a0: 6c74 2032 3030 0a2d 6170 2c20 2d2d 616e  lt 200.-ap, --an
+0000b5b0: 6368 6f72 706f 696e 7473 2c20 7468 6520  chorpoints, the 
+0000b5c0: 616e 6368 6f72 2070 6f69 6e74 7320 6461  anchor points da
+0000b5d0: 7461 6669 6c65 2c20 6465 6661 756c 7420  tafile, default 
+0000b5e0: 4e6f 6e65 0a2d 6d74 2c20 2d2d 6d75 6c74  None.-mt, --mult
+0000b5f0: 6970 6c69 636f 6e2c 2074 6865 206d 756c  iplicon, the mul
+0000b600: 7469 706c 6963 6f6e 7320 6461 7461 6669  tiplicons datafi
+0000b610: 6c65 2c20 6465 6661 756c 7420 4e6f 6e65  le, default None
+0000b620: 0a2d 6774 2c20 2d2d 6765 6e65 7461 626c  .-gt, --genetabl
+0000b630: 652c 2074 6865 2067 656e 6520 7461 626c  e, the gene tabl
+0000b640: 6520 6461 7461 6669 6c65 2c20 6465 6661  e datafile, defa
+0000b650: 756c 7420 4e6f 6e65 0a2d 6d67 2c20 2d2d  ult None.-mg, --
+0000b660: 6d69 6e73 6567 6c65 6e2c 2074 6865 206d  minseglen, the m
+0000b670: 696e 696d 756d 206c 656e 6774 6820 6f66  inimum length of
+0000b680: 2073 6567 6d65 6e74 7320 746f 2069 6e63   segments to inc
+0000b690: 6c75 6465 2c20 696e 2072 6174 696f 2069  lude, in ratio i
+0000b6a0: 6620 3c3d 2031 2c20 6465 6661 756c 7420  f <= 1, default 
+0000b6b0: 3130 3030 300a 2d6d 676e 2c20 2d2d 6d69  10000.-mgn, --mi
+0000b6c0: 6e67 656e 656e 756d 2c20 7468 6520 6d69  ngenenum, the mi
+0000b6d0: 6e69 6d75 6d20 6e75 6d62 6572 206f 6620  nimum number of 
+0000b6e0: 6765 6e65 7320 666f 7220 6120 7365 676d  genes for a segm
+0000b6f0: 656e 7420 746f 2062 6520 636f 6e73 6964  ent to be consid
+0000b700: 6572 6564 2c20 6465 6661 756c 7420 3330  ered, default 30
+0000b710: 0a2d 6b72 2c20 2d2d 6b65 6570 7265 6475  .-kr, --keepredu
+0000b720: 6e2c 2066 6c61 6720 6f70 7469 6f6e 2c20  n, flag option, 
+0000b730: 7768 6574 6865 7220 746f 206b 6565 7020  whether to keep 
+0000b740: 7265 6475 6e64 616e 7420 6d75 6c74 6970  redundant multip
+0000b750: 6c69 636f 6e73 2c20 6966 2074 6865 2066  licons, if the f
+0000b760: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+0000b770: 2072 6564 756e 6461 6e74 206d 756c 7469   redundant multi
+0000b780: 706c 6963 6f6e 7320 776f 756c 6420 6265  plicons would be
+0000b790: 206b 6570 740a 2d65 706b 2c20 2d2d 6578   kept.-epk, --ex
+0000b7a0: 7472 6170 6172 616e 6f6d 656b 732c 2065  traparanomeks, e
+0000b7b0: 7874 7261 2070 6172 616e 6f6d 6520 4b73  xtra paranome Ks
+0000b7c0: 2064 6174 6120 746f 2070 6c6f 7420 696e   data to plot in
+0000b7d0: 2074 6865 206d 6978 6564 204b 7320 6469   the mixed Ks di
+0000b7e0: 7374 7269 6275 7469 6f6e 2c20 6465 6661  stribution, defa
+0000b7f0: 756c 7420 4e6f 6e65 0a2d 7061 672c 202d  ult None.-pag, -
+0000b800: 2d70 6c6f 7461 7067 6d6d 2c20 666c 6167  -plotapgmm, flag
+0000b810: 206f 7074 696f 6e2c 2077 6865 7468 6572   option, whether
+0000b820: 2074 6f20 636f 6e64 7563 7420 616e 6420   to conduct and 
+0000b830: 706c 6f74 206d 6978 7475 7265 206d 6f64  plot mixture mod
+0000b840: 656c 696e 6720 6f66 2061 6e63 686f 7220  eling of anchor 
+0000b850: 4b73 2069 6e20 7468 6520 6d69 7865 6420  Ks in the mixed 
+0000b860: 4b73 2064 6973 7472 6962 7574 696f 6e2c  Ks distribution,
+0000b870: 2069 6620 7468 6520 666c 6167 2077 6173   if the flag was
+0000b880: 2073 6574 2c20 7468 6520 6d69 7874 7572   set, the mixtur
+0000b890: 6520 6d6f 6465 6c69 6e67 206f 6620 616e  e modeling of an
+0000b8a0: 6368 6f72 204b 7320 776f 756c 6420 6265  chor Ks would be
+0000b8b0: 2063 6f6e 6475 6374 6564 2061 6e64 2070   conducted and p
+0000b8c0: 6c6f 7474 6564 0a2d 7065 6d2c 202d 2d70  lotted.-pem, --p
+0000b8d0: 6c6f 7465 6c6d 6d2c 2066 6c61 6720 6f70  lotelmm, flag op
+0000b8e0: 7469 6f6e 2c20 7768 6574 6865 7220 746f  tion, whether to
+0000b8f0: 2063 6f6e 6475 6374 2061 6e64 2070 6c6f   conduct and plo
+0000b900: 7420 656c 6d6d 206d 6978 7475 7265 206d  t elmm mixture m
+0000b910: 6f64 656c 696e 6720 6f66 2070 6172 616e  odeling of paran
+0000b920: 6f6d 6520 4b73 2069 6e20 7468 6520 6d69  ome Ks in the mi
+0000b930: 7865 6420 4b73 2064 6973 7472 6962 7574  xed Ks distribut
+0000b940: 696f 6e2c 2069 6620 7468 6520 666c 6167  ion, if the flag
+0000b950: 2077 6173 2073 6574 2c20 7468 6520 656c   was set, the el
+0000b960: 6d6d 206d 6978 7475 7265 206d 6f64 656c  mm mixture model
+0000b970: 696e 6720 6f66 2070 6172 616e 6f6d 6520  ing of paranome 
+0000b980: 4b73 2077 6f75 6c64 2062 6520 636f 6e64  Ks would be cond
+0000b990: 7563 7465 6420 616e 6420 706c 6f74 7465  ucted and plotte
+0000b9a0: 640a 2d63 2c20 2d2d 636f 6d70 6f6e 656e  d.-c, --componen
+0000b9b0: 7473 2c20 7468 6520 7261 6e67 6520 6f66  ts, the range of
+0000b9c0: 2074 6865 206e 756d 6265 7220 6f66 2063   the number of c
+0000b9d0: 6f6d 706f 6e65 6e74 7320 746f 2066 6974  omponents to fit
+0000b9e0: 2069 6e20 616e 6368 6f72 204b 7320 6d69   in anchor Ks mi
+0000b9f0: 7874 7572 6520 6d6f 6465 6c69 6e67 2c20  xture modeling, 
+0000ba00: 6465 6661 756c 7420 2831 2c34 290a 2d70  default (1,4).-p
+0000ba10: 7379 2c20 2d2d 706c 6f74 7379 6e2c 2066  sy, --plotsyn, f
+0000ba20: 6c61 6720 6f70 7469 6f6e 2c20 7768 6574  lag option, whet
+0000ba30: 6865 7220 746f 2069 6e69 7469 6174 6520  her to initiate 
+0000ba40: 7468 6520 7379 6e74 656e 7920 706c 6f74  the synteny plot
+0000ba50: 2c20 6f6e 6c79 2077 6865 6e20 7468 6520  , only when the 
+0000ba60: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+0000ba70: 6520 7379 6e74 656e 7920 706c 6f74 2077  e synteny plot w
+0000ba80: 6f75 6c64 2062 6520 7072 6f64 7563 6564  ould be produced
+0000ba90: 0a2d 6473 2c20 2d2d 646f 7473 697a 652c  .-ds, --dotsize,
+0000baa0: 2074 6865 2064 6f74 2073 697a 6520 696e   the dot size in
+0000bab0: 2064 6f74 2070 6c6f 742c 2064 6566 6175   dot plot, defau
+0000bac0: 6c74 2030 2e33 0a2d 6161 2c20 2d2d 6170  lt 0.3.-aa, --ap
+0000bad0: 616c 7068 612c 2074 6865 206f 7061 6369  alpha, the opaci
+0000bae0: 7479 206f 6620 616e 6368 6f72 2064 6f74  ty of anchor dot
+0000baf0: 732c 2064 6566 6175 6c74 2031 0a2d 6861  s, default 1.-ha
+0000bb00: 2c20 2d2d 686f 616c 7068 612c 2074 6865  , --hoalpha, the
+0000bb10: 206f 7061 6369 7479 206f 6620 686f 6d6f   opacity of homo
+0000bb20: 6c6f 6720 646f 7473 2c20 6465 6661 756c  log dots, defaul
+0000bb30: 7420 300a 2d73 7274 2c20 2d2d 7368 6f77  t 0.-srt, --show
+0000bb40: 7265 616c 7469 636b 2c20 666c 6167 206f  realtick, flag o
+0000bb50: 7074 696f 6e2c 2077 6865 7468 6572 2074  ption, whether t
+0000bb60: 6f20 7368 6f77 2074 6865 2072 6561 6c20  o show the real 
+0000bb70: 7469 636b 2069 6e20 6765 6e65 7320 6f72  tick in genes or
+0000bb80: 2062 6173 6573 2c20 6966 2074 6865 2066   bases, if the f
+0000bb90: 6c61 6720 7761 7320 7365 742c 2074 6865  lag was set, the
+0000bba0: 2072 6561 6c20 7469 636b 2077 6f75 6c64   real tick would
+0000bbb0: 2062 6520 7368 6f77 6564 0a2d 746c 732c   be showed.-tls,
+0000bbc0: 202d 2d74 6963 6b6c 6162 656c 7369 7a65   --ticklabelsize
+0000bbd0: 2c20 7468 6520 6c61 6265 6c20 7369 7a65  , the label size
+0000bbe0: 206f 6620 7469 636b 2c20 6465 6661 756c   of tick, defaul
+0000bbf0: 7420 350a 2d78 6c2c 202d 2d78 6c69 6d2c  t 5.-xl, --xlim,
+0000bc00: 2074 6865 2078 2061 7869 7320 6c69 6d69   the x axis limi
+0000bc10: 7420 6f66 204b 7320 6469 7374 7269 6275  t of Ks distribu
+0000bc20: 7469 6f6e 0a2d 796c 2c20 2d2d 796c 696d  tion.-yl, --ylim
+0000bc30: 2c20 7468 6520 7920 6178 6973 206c 696d  , the y axis lim
+0000bc40: 6974 206f 6620 4b73 2064 6973 7472 6962  it of Ks distrib
+0000bc50: 7574 696f 6e0a 2d61 646f 2c20 2d2d 6164  ution.-ado, --ad
+0000bc60: 6a75 7374 6f72 7468 6f2c 2066 6c61 6720  justortho, flag 
+0000bc70: 6f70 7469 6f6e 2c20 7768 6574 6865 7220  option, whether 
+0000bc80: 746f 2061 646a 7573 7420 7468 6520 6869  to adjust the hi
+0000bc90: 7374 6f67 7261 6d20 6865 6967 6874 206f  stogram height o
+0000bca0: 6620 6f72 7468 6f6c 6f67 6f75 7320 4b73  f orthologous Ks
+0000bcb0: 2061 7320 746f 206d 6174 6368 2074 6865   as to match the
+0000bcc0: 2068 6569 6768 7420 6f66 2070 6172 616c   height of paral
+0000bcd0: 6f67 6f75 7320 4b73 2c20 6966 2074 6865  ogous Ks, if the
+0000bce0: 2066 6c61 6720 7761 7320 7365 742c 2074   flag was set, t
+0000bcf0: 6865 2061 646a 7573 746d 656e 7420 776f  he adjustment wo
+0000bd00: 756c 6420 6265 2063 6f6e 6475 6374 6564  uld be conducted
+0000bd10: 0a2d 6164 662c 202d 2d61 646a 7573 7466  .-adf, --adjustf
+0000bd20: 6163 746f 722c 2074 6865 2061 646a 7573  actor, the adjus
+0000bd30: 746d 656e 7420 6661 6374 6f72 206f 6620  tment factor of 
+0000bd40: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2c20  orthologous Ks, 
+0000bd50: 6465 6661 756c 7420 302e 350a 2d6f 612c  default 0.5.-oa,
+0000bd60: 202d 2d6f 6b61 6c70 6861 2c20 7468 6520   --okalpha, the 
+0000bd70: 6f70 6163 6974 7920 6f66 206f 7274 686f  opacity of ortho
+0000bd80: 6c6f 676f 7573 204b 7320 6469 7374 7269  logous Ks distri
+0000bd90: 6275 7469 6f6e 2069 6e20 6d69 7865 6420  bution in mixed 
+0000bda0: 706c 6f74 2c20 6465 6661 756c 7420 302e  plot, default 0.
+0000bdb0: 350a 2d63 732c 202d 2d63 6c61 7373 6963  5.-cs, --classic
+0000bdc0: 2c20 666c 6167 206f 7074 696f 6e2c 2077  , flag option, w
+0000bdd0: 6865 7468 6572 2074 6f20 6472 6177 206d  hether to draw m
+0000bde0: 6978 6564 204b 7320 706c 6f74 2069 6e20  ixed Ks plot in 
+0000bdf0: 6120 636c 6173 7369 6320 6d61 6e6e 6572  a classic manner
+0000be00: 2077 6865 7265 2074 6865 2066 756c 6c20   where the full 
+0000be10: 6f72 7468 6f6c 6f67 6f75 7320 4b73 2064  orthologous Ks d
+0000be20: 6973 7472 6962 7574 696f 6e20 6973 2064  istribution is d
+0000be30: 7261 7765 642c 2069 6620 7468 6520 666c  rawed, if the fl
+0000be40: 6167 2077 6173 2073 6574 2c20 7468 6520  ag was set, the 
+0000be50: 636c 6173 7369 6320 6d69 7865 6420 4b73  classic mixed Ks
+0000be60: 2070 6c6f 7420 776f 756c 6420 6265 2064   plot would be d
+0000be70: 7261 776e 0a2d 7461 2c20 2d2d 746f 7061  rawn.-ta, --topa
+0000be80: 7272 6f77 2c20 666c 6167 206f 7074 696f  rrow, flag optio
+0000be90: 6e2c 2077 6865 7468 6572 2074 6f20 6164  n, whether to ad
+0000bea0: 6a75 7374 2074 6865 2061 7272 6f77 2074  just the arrow t
+0000beb0: 6f20 6265 2061 7420 7468 6520 746f 7020  o be at the top 
+0000bec0: 6f66 2074 6865 2070 6c6f 742c 2069 6e73  of the plot, ins
+0000bed0: 7465 6164 206f 6620 6265 696e 6720 636f  tead of being co
+0000bee0: 6f72 6469 6e61 7465 6420 6173 2074 6865  ordinated as the
+0000bef0: 204b 4445 206f 6620 7468 6520 6f72 7468   KDE of the orth
+0000bf00: 6f6c 6f67 6f75 7320 4b73 2064 6973 7472  ologous Ks distr
+0000bf10: 6962 7574 696f 6e2c 2069 6620 7468 6520  ibution, if the 
+0000bf20: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+0000bf30: 6520 6172 726f 7720 776f 756c 6420 6265  e arrow would be
+0000bf40: 2073 6574 2061 7420 7468 6520 746f 700a   set at the top.
+0000bf50: 2d6e 612c 202d 2d6e 6f64 6561 7665 7261  -na, --nodeavera
+0000bf60: 6765 642c 2066 6c61 6720 6f70 7469 6f6e  ged, flag option
+0000bf70: 2c20 7768 6574 6865 7220 746f 2075 7365  , whether to use
+0000bf80: 206e 6f64 652d 6176 6572 6167 6564 206d   node-averaged m
+0000bf90: 6574 686f 6420 666f 7220 6465 2d72 6564  ethod for de-red
+0000bfa0: 756e 6461 6e63 792c 2069 6620 7468 6520  undancy, if the 
+0000bfb0: 666c 6167 2077 6173 2073 6574 2c20 7468  flag was set, th
+0000bfc0: 6520 6e6f 6465 2d61 7665 7261 6765 6420  e node-averaged 
+0000bfd0: 6d65 7468 6f64 2077 6f75 6c64 2062 6520  method would be 
+0000bfe0: 696e 6974 6961 7465 640a 2d62 732c 202d  initiated.-bs, -
+0000bff0: 2d62 6f6f 7473 7472 6170 2c20 7468 6520  -bootstrap, the 
+0000c000: 6e75 6d62 6572 206f 6620 626f 6f74 7374  number of bootst
+0000c010: 7261 7020 7265 706c 6963 6174 6573 206f  rap replicates o
+0000c020: 6620 6f72 7468 6f6c 6f67 204b 7320 6469  f ortholog Ks di
+0000c030: 7374 7269 6275 7469 6f6e 2069 6e20 6d69  stribution in mi
+0000c040: 7865 6420 706c 6f74 0a2d 6772 2c20 2d2d  xed plot.-gr, --
+0000c050: 6769 7374 7262 2c20 666c 6167 206f 7074  gistrb, flag opt
+0000c060: 696f 6e2c 2077 6865 7468 6572 2074 6f20  ion, whether to 
+0000c070: 7573 6520 6769 7374 5f72 6169 6e62 6f77  use gist_rainbow
+0000c080: 2061 7320 636f 6c6f 7220 6d61 7020 6f66   as color map of
+0000c090: 2064 6f74 706c 6f74 0a2d 6e2c 202d 2d6e   dotplot.-n, --n
+0000c0a0: 7468 7265 6164 732c 2074 6865 206e 756d  threads, the num
+0000c0b0: 6265 7220 6f66 2074 6872 6561 6473 2074  ber of threads t
+0000c0c0: 6f20 7573 6520 696e 2062 6f6f 7473 7472  o use in bootstr
+0000c0d0: 6170 2073 616d 706c 696e 672c 2064 6566  ap sampling, def
+0000c0e0: 6175 6c74 2031 0a60 6060 0a0a 2323 2055  ault 1.```..## U
+0000c0f0: 7361 6765 0a0a 4865 7265 2077 6520 7072  sage..Here we pr
+0000c100: 6f76 6964 6564 2074 6865 2062 6173 6963  ovided the basic
+0000c110: 2075 7361 6765 2066 6f72 2065 6163 6820   usage for each 
+0000c120: 7072 6f67 7261 6d20 616e 6420 7468 6520  program and the 
+0000c130: 7265 6c65 7661 6e74 2070 6172 616d 6574  relevant paramet
+0000c140: 6572 7320 616e 6420 7375 6767 6573 7469  ers and suggesti
+0000c150: 6f6e 7320 6f6e 2070 6172 616d 6574 6572  ons on parameter
+0000c160: 697a 6174 696f 6e2e 2041 2072 656d 696e  ization. A remin
+0000c170: 6465 7220 7468 6174 2074 6865 2067 6976  der that the giv
+0000c180: 656e 2063 6f72 6573 2061 6e64 2074 6872  en cores and thr
+0000c190: 6561 6473 2063 616e 2073 6967 6e69 6669  eads can signifi
+0000c1a0: 6361 6e74 6c79 2069 6d70 6163 7420 7468  cantly impact th
+0000c1b0: 6520 7275 6e20 7469 6d65 2061 6e64 2074  e run time and t
+0000c1c0: 6875 7320 7765 2061 6464 6564 2073 6f6d  hus we added som
+0000c1d0: 6520 7265 706f 7274 2069 6e66 6f72 6d61  e report informa
+0000c1e0: 7469 6f6e 2070 6572 7461 696e 696e 6720  tion pertaining 
+0000c1f0: 746f 2074 6865 2073 7973 7465 6d20 6f66  to the system of
+0000c200: 2075 7365 7273 2074 6f20 6661 6369 6c69   users to facili
+0000c210: 7461 7465 2074 6865 2065 6666 6963 6965  tate the efficie
+0000c220: 6e74 2073 6574 7469 6e67 206f 6620 7468  nt setting of th
+0000c230: 7265 6164 7320 616e 6420 6d65 6d6f 7279  reads and memory
+0000c240: 2e20 5468 6520 6c6f 6769 6361 6c20 4350  . The logical CP
+0000c250: 5573 2072 6570 6f72 7465 6420 7265 7072  Us reported repr
+0000c260: 6573 656e 7473 2074 6865 206e 756d 6265  esents the numbe
+0000c270: 7220 6f66 2070 6879 7369 6361 6c20 636f  r of physical co
+0000c280: 7265 7320 6d75 6c74 6970 6c69 6564 2062  res multiplied b
+0000c290: 7920 7468 6520 6e75 6d62 6572 206f 6620  y the number of 
+0000c2a0: 7468 7265 6164 7320 7468 6174 2063 616e  threads that can
+0000c2b0: 2072 756e 206f 6e20 6561 6368 2063 6f72   run on each cor
+0000c2c0: 652c 2061 6c73 6f20 6b6e 6f77 6e20 6173  e, also known as
+0000c2d0: 2048 7970 6572 2054 6872 6561 6469 6e67   Hyper Threading
+0000c2e0: 2e20 5468 6520 6e75 6d62 6572 206f 6620  . The number of 
+0000c2f0: 6c6f 6769 6361 6c20 4350 5573 206d 6179  logical CPUs may
+0000c300: 206e 6f74 206e 6563 6573 7361 7269 6c79   not necessarily
+0000c310: 2062 6520 6571 7569 7661 6c65 6e74 2074   be equivalent t
+0000c320: 6f20 7468 6520 6163 7475 616c 206e 756d  o the actual num
+0000c330: 6265 7220 6f66 2043 5055 7320 7468 6520  ber of CPUs the 
+0000c340: 6375 7272 656e 7420 7072 6f63 6573 7320  current process 
+0000c350: 6361 6e20 7573 652e 2054 6865 2061 7661  can use. The ava
+0000c360: 696c 6162 6c65 206d 656d 6f72 7920 7265  ilable memory re
+0000c370: 6665 7273 2074 6f20 7468 6520 6d65 6d6f  fers to the memo
+0000c380: 7279 2074 6861 7420 6361 6e20 6265 2067  ry that can be g
+0000c390: 6976 656e 2069 6e73 7461 6e74 6c79 2074  iven instantly t
+0000c3a0: 6f20 7072 6f63 6573 7365 7320 7769 7468  o processes with
+0000c3b0: 6f75 7420 7468 6520 7379 7374 656d 2067  out the system g
+0000c3c0: 6f69 6e67 2069 6e74 6f20 7377 6170 2061  oing into swap a
+0000c3d0: 6e64 2072 6566 6c65 6374 7320 7468 6520  nd reflects the 
+0000c3e0: 6163 7475 616c 206d 656d 6f72 7920 6176  actual memory av
+0000c3f0: 6169 6c61 626c 652e 2054 6865 2066 7265  ailable. The fre
+0000c400: 6520 6d65 6d6f 7279 2072 6566 6572 7320  e memory refers 
+0000c410: 746f 2074 6865 206d 656d 6f72 7920 6e6f  to the memory no
+0000c420: 7420 6265 696e 6720 7573 6564 2061 7420  t being used at 
+0000c430: 616c 6c20 287a 6572 6f65 6429 2074 6861  all (zeroed) tha
+0000c440: 7420 6973 2072 6561 6469 6c79 2061 7661  t is readily ava
+0000c450: 696c 6162 6c65 2e20 5468 6520 6465 7363  ilable. The desc
+0000c460: 7269 7074 696f 6e20 6162 6f76 6520 7265  ription above re
+0000c470: 6665 7273 2074 6f20 7468 6520 646f 6375  fers to the docu
+0000c480: 6d65 6e74 6174 696f 6e20 6f66 2060 7073  mentation of `ps
+0000c490: 7574 696c 602e 0a0a 2323 2320 7767 6420  util`...### wgd 
+0000c4a0: 646d 640a 0a2a 2a54 6865 2064 656c 696e  dmd..**The delin
+0000c4b0: 6561 7469 6f6e 206f 6620 7768 6f6c 6520  eation of whole 
+0000c4c0: 7061 7261 6e6f 6d65 2a2a 0a60 6060 0a77  paranome**.```.w
+0000c4d0: 6764 2064 6d64 2041 7175 696c 6567 6961  gd dmd Aquilegia
+0000c4e0: 5f63 6f65 7275 6c65 6120 2d49 2032 202d  _coerulea -I 2 -
+0000c4f0: 6520 3165 2d31 3020 2d62 7320 3130 3020  e 1e-10 -bs 100 
+0000c500: 2d6e 7020 3520 282d 6e6e 2920 282d 2d74  -np 5 (-nn) (--t
+0000c510: 6f5f 7374 6f70 2920 282d 2d63 6473 2920  o_stop) (--cds) 
+0000c520: 282d 6e20 3429 2028 2d6f 2077 6764 5f64  (-n 4) (-o wgd_d
+0000c530: 6d64 2920 282d 7420 776f 726b 696e 675f  md) (-t working_
+0000c540: 746d 7029 0a60 6060 200a 0a4e 6f74 6520  tmp).``` ..Note 
+0000c550: 7468 6174 2077 6520 646f 6e27 7420 7072  that we don't pr
+0000c560: 6f76 6964 6520 7468 6520 6461 7461 206f  ovide the data o
+0000c570: 6620 7468 6973 2063 6f64 696e 6720 7365  f this coding se
+0000c580: 7175 656e 6365 2028 6364 7329 2066 696c  quence (cds) fil
+0000c590: 6520 6041 7175 696c 6567 6961 5f63 6f65  e `Aquilegia_coe
+0000c5a0: 7275 6c65 6160 2062 7574 2069 7420 6361  rulea` but it ca
+0000c5b0: 6e20 6265 2065 6173 696c 7920 646f 776e  n be easily down
+0000c5c0: 6c6f 6164 6564 2061 7420 5b50 6879 746f  loaded at [Phyto
+0000c5d0: 7a6f 6d65 5d28 6874 7470 733a 2f2f 7068  zome](https://ph
+0000c5e0: 7974 6f7a 6f6d 652d 6e65 7874 2e6a 6769  ytozome-next.jgi
+0000c5f0: 2e64 6f65 2e67 6f76 2f69 6e66 6f2f 4163  .doe.gov/info/Ac
+0000c600: 6f65 7275 6c65 615f 7633 5f31 2920 2873  oerulea_v3_1) (s
+0000c610: 616d 6520 666f 7220 6f74 6865 7220 5573  ame for other Us
+0000c620: 6167 6520 646f 6329 2e20 4120 7265 6d69  age doc). A remi
+0000c630: 6e64 6572 2074 6861 7420 696e 2074 6865  nder that in the
+0000c640: 2069 7373 7565 7320 736f 6d65 2075 7365   issues some use
+0000c650: 7273 2064 6964 6e27 7420 646f 776e 6c6f  rs didn't downlo
+0000c660: 6164 2074 6865 2060 4163 6f65 7275 6c65  ad the `Acoerule
+0000c670: 615f 3332 325f 7633 2e31 2e63 6473 5f70  a_322_v3.1.cds_p
+0000c680: 7269 6d61 7279 5472 616e 7363 7269 7074  rimaryTranscript
+0000c690: 4f6e 6c79 2e66 612e 677a 6020 6275 7420  Only.fa.gz` but 
+0000c6a0: 696e 7374 6561 6420 7468 6520 6041 636f  instead the `Aco
+0000c6b0: 6572 756c 6561 5f33 3232 5f76 332e 312e  erulea_322_v3.1.
+0000c6c0: 6364 732e 6661 2e67 7a60 2066 696c 652e  cds.fa.gz` file.
+0000c6d0: 2046 6f72 2074 6865 2063 6f6e 7374 7275   For the constru
+0000c6e0: 6374 696f 6e20 6f66 2077 686f 6c65 2070  ction of whole p
+0000c6f0: 6172 616e 6f6d 6520 2a4b 2a3c 7375 623e  aranome *K*<sub>
+0000c700: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+0000c710: 7469 6f6e 2c20 2a2a 6f6e 6c79 206f 6e65  tion, **only one
+0000c720: 2074 7261 6e73 6372 6970 7420 2874 6865   transcript (the
+0000c730: 2070 7269 6d61 7279 206f 6e65 2920 7065   primary one) pe
+0000c740: 7220 6765 6e65 2073 686f 756c 6420 6265  r gene should be
+0000c750: 2069 6e63 6c75 6465 642a 2a20 7375 6368   included** such
+0000c760: 2074 6861 7420 7468 6520 2a4b 2a3c 7375   that the *K*<su
+0000c770: 623e 533c 2f73 7562 3e20 6973 2072 6561  b>S</sub> is rea
+0000c780: 6c6c 7920 696e 6469 6361 7469 6e67 2074  lly indicating t
+0000c790: 6865 2061 6765 206f 6620 6765 6e65 2064  he age of gene d
+0000c7a0: 7570 6c69 6361 7469 6f6e 2065 7665 6e74  uplication event
+0000c7b0: 2c20 696e 7374 6561 6420 6f66 2061 6c74  , instead of alt
+0000c7c0: 6572 6e61 7469 7665 2073 706c 6963 696e  ernative splicin
+0000c7d0: 672e 2054 7261 6e73 6372 6970 746f 6d65  g. Transcriptome
+0000c7e0: 2064 6174 6120 7368 6f75 6c64 2062 6520   data should be 
+0000c7f0: 6361 7265 6675 6c6c 7920 7472 6561 7465  carefully treate
+0000c800: 6420 7769 7468 2064 652d 7265 6475 6e64  d with de-redund
+0000c810: 616e 6379 2073 6f20 6173 2074 6f20 7265  ancy so as to re
+0000c820: 6475 6365 2074 6865 2066 616c 7365 2070  duce the false p
+0000c830: 6f73 6974 6976 6520 6475 706c 6963 6174  ositive duplicat
+0000c840: 696f 6e20 6275 6d70 2063 6175 7365 6420  ion bump caused 
+0000c850: 6279 2070 6572 7661 7369 7665 2061 6c74  by pervasive alt
+0000c860: 6572 6e61 7469 7665 2073 706c 6963 696e  ernative splicin
+0000c870: 672e 2048 6572 6520 7468 6520 696e 666c  g. Here the infl
+0000c880: 6174 696f 6e20 6661 6374 6f72 2070 6172  ation factor par
+0000c890: 616d 6574 6572 2c20 6769 7665 6e20 6279  ameter, given by
+0000c8a0: 2060 2d49 6020 6f72 2060 2d2d 696e 666c   `-I` or `--infl
+0000c8b0: 6174 696f 6e60 2c20 6166 6665 6374 7320  ation`, affects 
+0000c8c0: 7468 6520 6772 616e 756c 6172 6974 7920  the granularity 
+0000c8d0: 6f72 2072 6573 6f6c 7574 696f 6e20 6f66  or resolution of
+0000c8e0: 2074 6865 2063 6c75 7374 6572 696e 6720   the clustering 
+0000c8f0: 6f75 7463 6f6d 6520 616e 6420 696d 706c  outcome and impl
+0000c900: 6963 6974 6c79 2063 6f6e 7472 6f6c 6c73  icitly controlls
+0000c910: 2074 6865 206e 756d 6265 7220 6f66 2063   the number of c
+0000c920: 6c75 7374 6572 732c 2077 6974 6820 6c6f  lusters, with lo
+0000c930: 7720 7661 6c75 6573 2073 7563 6820 6173  w values such as
+0000c940: 2031 2e33 206f 7220 312e 3420 6c65 6164   1.3 or 1.4 lead
+0000c950: 696e 6720 746f 2066 6577 6572 2062 7574  ing to fewer but
+0000c960: 206c 6172 6765 7220 636c 7573 7465 7273   larger clusters
+0000c970: 2061 6e64 2068 6967 6820 7661 6c75 6573   and high values
+0000c980: 2073 7563 6820 6173 2035 206f 7220 3620   such as 5 or 6 
+0000c990: 6c65 6164 696e 6720 746f 206d 6f72 6520  leading to more 
+0000c9a0: 6275 7420 736d 616c 6c65 7220 636c 7573  but smaller clus
+0000c9b0: 7465 7273 2e20 5765 2073 6574 2074 6865  ters. We set the
+0000c9c0: 2064 6566 6175 6c74 2076 616c 7565 2061   default value a
+0000c9d0: 7320 3220 6173 2073 7567 6765 7374 6564  s 2 as suggested
+0000c9e0: 2062 7920 5b4d 434c 5d28 6874 7470 733a   by [MCL](https:
+0000c9f0: 2f2f 6d69 6361 6e73 2e6f 7267 2f6d 636c  //micans.org/mcl
+0000ca00: 2f29 2e20 5468 6520 652d 7661 6c75 6520  /). The e-value 
+0000ca10: 6375 742d 6f66 6620 666f 7220 7365 7175  cut-off for sequ
+0000ca20: 656e 6365 2073 696d 696c 6172 6974 792c  ence similarity,
+0000ca30: 2067 6976 656e 2062 7920 602d 6560 206f   given by `-e` o
+0000ca40: 7220 602d 2d65 7661 6c60 2c20 7768 6963  r `--eval`, whic
+0000ca50: 6820 6465 6e6f 7465 7320 7468 6520 6578  h denotes the ex
+0000ca60: 7065 6374 6564 2076 616c 7565 206f 6620  pected value of 
+0000ca70: 7468 6520 6869 7420 7175 616e 7469 6669  the hit quantifi
+0000ca80: 6573 2074 6865 206e 756d 6265 7220 6f66  es the number of
+0000ca90: 2061 6c69 676e 6d65 6e74 7320 6f66 2073   alignments of s
+0000caa0: 696d 696c 6172 206f 7220 6265 7474 6572  imilar or better
+0000cab0: 2071 7561 6c69 7479 2074 6861 7420 796f   quality that yo
+0000cac0: 7520 6578 7065 6374 2074 6f20 6669 6e64  u expect to find
+0000cad0: 2073 6561 7263 6869 6e67 2074 6869 7320   searching this 
+0000cae0: 7175 6572 7920 6167 6169 6e73 7420 6120  query against a 
+0000caf0: 6461 7461 6261 7365 206f 6620 7261 6e64  database of rand
+0000cb00: 6f6d 2073 6571 7565 6e63 6573 2074 6865  om sequences the
+0000cb10: 2073 616d 6520 7369 7a65 2061 7320 7468   same size as th
+0000cb20: 6520 6163 7475 616c 2074 6172 6765 7420  e actual target 
+0000cb30: 6461 7461 6261 7365 2c20 6973 2074 6865  database, is the
+0000cb40: 206b 6579 2070 6172 616d 6574 6572 206d   key parameter m
+0000cb50: 6561 7375 7269 6e67 2074 6865 2073 6967  easuring the sig
+0000cb60: 6e69 6669 6361 6e63 6520 6f66 2061 2068  nificance of a h
+0000cb70: 6974 2c20 7768 6963 6820 6973 2073 6574  it, which is set
+0000cb80: 2068 6572 6520 6173 2064 6566 6175 6c74   here as default
+0000cb90: 2031 652d 3130 2e20 4e6f 7465 2074 6861   1e-10. Note tha
+0000cba0: 7420 5b44 4941 4d4f 4e44 5d28 6874 7470  t [DIAMOND](http
+0000cbb0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+0000cbc0: 6275 6368 6669 6e6b 2f64 6961 6d6f 6e64  buchfink/diamond
+0000cbd0: 2f77 696b 692f 312e 2d54 7574 6f72 6961  /wiki/1.-Tutoria
+0000cbe0: 6c29 2069 7473 656c 6620 6279 2064 6566  l) itself by def
+0000cbf0: 6175 6c74 206f 6e6c 7920 7265 706f 7274  ault only report
+0000cc00: 7320 616c 6c20 616c 6967 6e6d 656e 7473  s all alignments
+0000cc10: 2077 6974 6820 652d 7661 6c75 6520 3c20   with e-value < 
+0000cc20: 302e 3030 312e 2054 6865 2070 6572 6365  0.001. The perce
+0000cc30: 6e74 6167 6520 6f66 2075 7070 6572 2068  ntage of upper h
+0000cc40: 6974 7320 7573 6564 2066 6f72 2067 656e  its used for gen
+0000cc50: 6520 6c65 6e67 7468 206e 6f72 6d61 6c69  e length normali
+0000cc60: 7a61 7469 6f6e 2c20 6769 7665 6e20 6279  zation, given by
+0000cc70: 2060 2d6e 7060 206f 7220 602d 2d6e 6f72   `-np` or `--nor
+0000cc80: 6d61 6c69 7a65 6470 6572 6365 6e74 602c  malizedpercent`,
+0000cc90: 2077 6869 6368 2064 6574 6572 6d69 6e65   which determine
+0000cca0: 7320 7468 6520 7570 7065 7220 7065 7263  s the upper perc
+0000ccb0: 656e 7469 6c65 206f 6620 6869 7473 2070  entile of hits p
+0000ccc0: 6572 2062 696e 2028 6361 7465 676f 7269  er bin (categori
+0000ccd0: 7a65 6420 6279 2067 656e 6520 6c65 6e67  zed by gene leng
+0000cce0: 7468 2920 7573 6564 2069 6e20 7468 6520  th) used in the 
+0000ccf0: 6669 7420 6f66 206c 696e 6561 7220 7265  fit of linear re
+0000cd00: 6772 6573 7369 6f6e 2c20 636f 6e73 6964  gression, consid
+0000cd10: 6572 696e 6720 7468 6174 206e 6f74 2061  ering that not a
+0000cd20: 6c6c 2068 6974 7320 7065 7220 6269 6e20  ll hits per bin 
+0000cd30: 7368 6f77 2061 7070 6172 656e 7420 6c69  show apparent li
+0000cd40: 6e65 6172 2072 656c 6174 696f 6e73 6869  near relationshi
+0000cd50: 702c 2069 7320 7365 7420 6173 2064 6566  p, is set as def
+0000cd60: 6175 6c74 2035 2c20 696e 6469 6361 7469  ault 5, indicati
+0000cd70: 6e67 2074 6865 2075 7361 6765 206f 6620  ng the usage of 
+0000cd80: 746f 7020 3525 2068 6974 7320 7065 7220  top 5% hits per 
+0000cd90: 6269 6e2e 2054 6865 206e 756d 6265 7220  bin. The number 
+0000cda0: 6f66 2062 696e 7320 6469 7669 6465 6420  of bins divided 
+0000cdb0: 696e 2067 656e 6520 6c65 6e67 7468 206e  in gene length n
+0000cdc0: 6f72 6d61 6c69 7a61 7469 6f6e 2c20 6769  ormalization, gi
+0000cdd0: 7665 6e20 6279 2060 2d62 7360 206f 7220  ven by `-bs` or 
+0000cde0: 602d 2d62 696e 7360 2c20 6465 7465 726d  `--bins`, determ
+0000cdf0: 696e 6573 2074 6865 206e 756d 6265 7220  ines the number 
+0000ce00: 6f66 2062 696e 7320 746f 2063 6174 6567  of bins to categ
+0000ce10: 6f72 697a 6520 7468 6520 6765 6e65 206c  orize the gene l
+0000ce20: 656e 6774 682c 2069 7320 7365 7420 6173  ength, is set as
+0000ce30: 2064 6566 6175 6c74 2031 3030 2e20 5468   default 100. Th
+0000ce40: 6520 7061 7261 6d65 7465 7220 602d 6e6e  e parameter `-nn
+0000ce50: 6020 6f72 2060 2d2d 6e6f 6e6f 726d 616c  ` or `--nonormal
+0000ce60: 697a 6174 696f 6e60 2063 616e 2062 6520  ization` can be 
+0000ce70: 7365 7420 746f 2063 616c 6c20 6f66 6620  set to call off 
+0000ce80: 7468 6520 6e6f 726d 616c 697a 6174 696f  the normalizatio
+0000ce90: 6e20 7072 6f63 6573 732c 2061 6c74 686f  n process, altho
+0000cea0: 7567 6820 6974 2773 2073 7567 6765 7374  ugh it's suggest
+0000ceb0: 6564 2074 6f20 636f 6e64 7563 7420 7468  ed to conduct th
+0000cec0: 6520 6e6f 726d 616c 697a 6174 696f 6e20  e normalization 
+0000ced0: 746f 2061 6371 7569 7265 206d 6f72 6520  to acquire more 
+0000cee0: 6163 6375 7261 7465 2067 656e 6520 6661  accurate gene fa
+0000cef0: 6d69 6c79 2063 6c75 7374 6572 696e 6720  mily clustering 
+0000cf00: 7265 7375 6c74 2e20 5468 6520 7061 7261  result. The para
+0000cf10: 6d65 7465 7273 2060 2d2d 746f 5f73 746f  meters `--to_sto
+0000cf20: 7060 2061 6e64 2060 2d2d 6364 7360 2063  p` and `--cds` c
+0000cf30: 6f6e 7472 6f6c 2074 6865 2062 6568 6176  ontrol the behav
+0000cf40: 696f 7572 206f 6620 7472 616e 736c 6174  iour of translat
+0000cf50: 696e 6720 636f 6469 6e67 2073 6571 7565  ing coding seque
+0000cf60: 6e63 6520 696e 746f 2061 6d69 6e6f 2061  nce into amino a
+0000cf70: 6369 6420 7365 7175 656e 6365 2e20 4966  cid sequence. If
+0000cf80: 2074 6865 2060 2d2d 746f 5f73 746f 7060   the `--to_stop`
+0000cf90: 2077 6173 2073 6574 2c20 7468 6520 7472   was set, the tr
+0000cfa0: 616e 736c 6174 696f 6e20 776f 756c 6420  anslation would 
+0000cfb0: 6265 2074 6572 6d69 6e61 7465 6420 6174  be terminated at
+0000cfc0: 2074 6865 2066 6972 7374 2069 6e2d 6672   the first in-fr
+0000cfd0: 616d 6520 7374 6f70 2063 6f64 6f6e 2c20  ame stop codon, 
+0000cfe0: 6f74 6865 7277 6973 6520 7468 6520 7472  otherwise the tr
+0000cff0: 616e 736c 6174 696f 6e20 776f 756c 6420  anslation would 
+0000d000: 7369 6d70 6c79 2073 6b69 7020 616e 7920  simply skip any 
+0000d010: 7374 6f70 2063 6f64 6f6e 732e 2049 6620  stop codons. If 
+0000d020: 7468 6520 602d 2d63 6473 6020 7761 7320  the `--cds` was 
+0000d030: 7365 742c 2073 6571 7565 6e63 6573 2074  set, sequences t
+0000d040: 6861 7420 646f 6573 6e27 7420 7374 6172  hat doesn't star
+0000d050: 7420 7769 7468 2061 2076 616c 6964 2073  t with a valid s
+0000d060: 7461 7274 2063 6f64 6f6e 2c20 6f72 2063  tart codon, or c
+0000d070: 6f6e 7461 696e 7320 6d6f 7265 2074 6861  ontains more tha
+0000d080: 6e20 6f6e 6520 696e 2d66 7261 6d65 2073  n one in-frame s
+0000d090: 746f 7020 636f 646f 6e2c 206f 7220 6973  top codon, or is
+0000d0a0: 206e 6f74 2064 6976 6964 6162 6c65 2062   not dividable b
+0000d0b0: 7920 332c 2077 6f75 6c64 2062 6520 7369  y 3, would be si
+0000d0c0: 6d70 6c79 2064 726f 7070 6564 2c20 7375  mply dropped, su
+0000d0d0: 6368 2074 6861 7420 6f6e 6c79 2073 7472  ch that only str
+0000d0e0: 6963 7420 636f 6d70 6c65 7465 2063 6f64  ict complete cod
+0000d0f0: 696e 6720 7365 7175 656e 6365 7320 776f  ing sequences wo
+0000d100: 756c 6420 6265 2069 6e63 6c75 6465 6420  uld be included 
+0000d110: 696e 2074 6865 2073 7562 7365 7175 656e  in the subsequen
+0000d120: 7420 616e 616c 7973 6973 2e20 5468 6520  t analysis. The 
+0000d130: 6578 6163 7420 6265 6861 7669 6f75 7220  exact behaviour 
+0000d140: 6f66 2060 2d2d 746f 5f73 746f 7060 2061  of `--to_stop` a
+0000d150: 6e64 2060 2d2d 6364 7360 2069 7320 6465  nd `--cds` is de
+0000d160: 6669 6e65 6420 616e 6420 6465 7363 7269  fined and descri
+0000d170: 6265 6420 696e 2074 6865 205b 6269 6f70  bed in the [biop
+0000d180: 7974 686f 6e5d 2868 7474 7073 3a2f 2f62  ython](https://b
+0000d190: 696f 7079 7468 6f6e 2e6f 7267 2f64 6f63  iopython.org/doc
+0000d1a0: 732f 312e 3735 2f61 7069 2f42 696f 2e53  s/1.75/api/Bio.S
+0000d1b0: 6571 2e68 746d 6c23 4269 6f2e 5365 712e  eq.html#Bio.Seq.
+0000d1c0: 5365 712e 7472 616e 736c 6174 6529 206c  Seq.translate) l
+0000d1d0: 6962 7261 7279 2e20 5468 6520 6e75 6d62  ibrary. The numb
+0000d1e0: 6572 206f 6620 7061 7261 6c6c 656c 2074  er of parallel t
+0000d1f0: 6872 6561 6473 2062 7920 7468 6520 6f70  hreads by the op
+0000d200: 7469 6f6e 2060 2d6e 6020 6f72 2060 2d2d  tion `-n` or `--
+0000d210: 6e74 6872 6561 6473 6020 6361 6e20 6265  nthreads` can be
+0000d220: 2073 6574 2074 6f20 6163 6365 6c65 7261   set to accelera
+0000d230: 7465 2074 6865 2063 616c 6375 6c61 7469  te the calculati
+0000d240: 6f6e 2077 6974 6869 6e20 6064 6961 6d6f  on within `diamo
+0000d250: 6e64 602e 2054 6865 2064 6972 6563 746f  nd`. The directo
+0000d260: 7279 206f 6620 6f75 7470 7574 206f 7220  ry of output or 
+0000d270: 696e 7465 726d 6564 6961 7465 2066 696c  intermediate fil
+0000d280: 6573 2069 7320 6465 7465 726d 696e 6564  es is determined
+0000d290: 2062 7920 7468 6520 7061 7261 6d65 7465   by the paramete
+0000d2a0: 7220 602d 6f60 206f 7220 602d 2d6f 7574  r `-o` or `--out
+0000d2b0: 6469 7260 2c20 616e 6420 602d 7460 206f  dir`, and `-t` o
+0000d2c0: 7220 602d 2d74 6d70 6469 7260 2c20 7768  r `--tmpdir`, wh
+0000d2d0: 6963 6820 7769 6c6c 2062 6520 6372 6561  ich will be crea
+0000d2e0: 7465 6420 6279 2074 6865 2070 726f 6772  ted by the progr
+0000d2f0: 616d 2069 7473 656c 6620 616e 6420 6265  am itself and be
+0000d300: 206f 7665 7277 7269 7474 656e 2069 6620   overwritten if 
+0000d310: 7468 6520 666f 6c64 6572 2068 6173 2061  the folder has a
+0000d320: 6c72 6561 6479 2062 6565 6e20 6372 6561  lready been crea
+0000d330: 7465 642e 204e 6f74 6520 7468 6174 2074  ted. Note that t
+0000d340: 6865 2073 6f66 7477 6172 6520 6064 6961  he software `dia
+0000d350: 6d6f 6e64 6020 7368 6f75 6c64 2062 6520  mond` should be 
+0000d360: 7072 652d 696e 7374 616c 6c65 6420 616e  pre-installed an
+0000d370: 6420 7365 7420 696e 2074 6865 2065 6e76  d set in the env
+0000d380: 6972 6f6e 6d65 6e74 2070 6174 6820 696e  ironment path in
+0000d390: 2061 6c6c 2074 6865 2061 6e61 6c79 7369   all the analysi
+0000d3a0: 7320 7065 7266 6f72 6d65 6420 6279 2060  s performed by `
+0000d3b0: 7767 6420 646d 6460 2065 7863 6570 7420  wgd dmd` except 
+0000d3c0: 666f 7220 7468 6520 636f 6c6c 696e 6561  for the collinea
+0000d3d0: 7220 636f 616c 6573 6365 6e63 6520 616e  r coalescence an
+0000d3e0: 616c 7973 6973 2e0a 0a2a 2a57 6520 7375  alysis...**We su
+0000d3f0: 6767 6573 7420 7468 6174 2074 6865 2064  ggest that the d
+0000d400: 6566 6175 6c74 2073 6574 7469 6e67 2069  efault setting i
+0000d410: 6e20 7768 6963 6820 7468 6520 696e 666c  n which the infl
+0000d420: 6174 696f 6e20 6661 6374 6f72 2069 7320  ation factor is 
+0000d430: 7365 7420 6173 2032 2c20 652d 7661 6c75  set as 2, e-valu
+0000d440: 6520 6375 742d 6f66 6620 6173 2031 652d  e cut-off as 1e-
+0000d450: 3130 2061 6e64 206f 7468 6572 2070 6172  10 and other par
+0000d460: 616d 6574 6572 7320 696e 2064 6566 6175  ameters in defau
+0000d470: 6c74 2069 7320 6120 676f 6f64 2073 7461  lt is a good sta
+0000d480: 7274 696e 6720 706f 696e 742c 2075 6e6c  rting point, unl
+0000d490: 6573 7320 796f 7520 7370 6563 6966 6963  ess you specific
+0000d4a0: 616c 6c79 2077 616e 7420 746f 2065 7870  ally want to exp
+0000d4b0: 6c6f 7265 2074 6865 2065 6666 6563 7473  lore the effects
+0000d4c0: 206f 6620 6469 6666 6572 656e 7420 7061   of different pa
+0000d4d0: 7261 6d65 7465 7273 2e2a 2a20 5375 6368  rameters.** Such
+0000d4e0: 2074 6861 7420 7468 6520 636f 6d6d 616e   that the comman
+0000d4f0: 6420 666f 7220 7468 6520 6465 6c69 6e65  d for the deline
+0000d500: 6174 696f 6e20 6f66 2077 686f 6c65 2070  ation of whole p
+0000d510: 6172 616e 6f6d 6520 6973 2073 696d 706c  aranome is simpl
+0000d520: 7920 6173 2062 656c 6f77 2e0a 0a60 6060  y as below...```
+0000d530: 0a77 6764 2064 6d64 2041 7175 696c 6567  .wgd dmd Aquileg
+0000d540: 6961 5f63 6f65 7275 6c65 610a 6060 600a  ia_coerulea.```.
+0000d550: 0a2a 2a54 6865 2064 656c 696e 6561 7469  .**The delineati
+0000d560: 6f6e 206f 6620 5242 4873 2a2a 0a60 6060  on of RBHs**.```
+0000d570: 0a77 6764 2064 6d64 2073 6571 7565 6e63  .wgd dmd sequenc
+0000d580: 6531 2073 6571 7565 6e63 6532 202d 6520  e1 sequence2 -e 
+0000d590: 3165 2d31 3020 2d62 7320 3130 3020 2d6e  1e-10 -bs 100 -n
+0000d5a0: 7020 3520 282d 6e6e 2920 282d 6e20 3429  p 5 (-nn) (-n 4)
+0000d5b0: 2028 2d63 2030 2e39 2920 282d 2d6f 6766   (-c 0.9) (--ogf
+0000d5c0: 6f72 6d61 7429 2028 2d2d 746f 5f73 746f  ormat) (--to_sto
+0000d5d0: 7029 2028 2d2d 6364 7329 2028 2d6f 2077  p) (--cds) (-o w
+0000d5e0: 6764 5f64 6d64 2920 282d 7420 776f 726b  gd_dmd) (-t work
+0000d5f0: 696e 675f 746d 7029 0a60 6060 0a0a 546f  ing_tmp).```..To
+0000d600: 2064 656c 696e 6561 7465 2052 4248 7320   delineate RBHs 
+0000d610: 6265 7477 6565 6e20 7477 6f20 6364 7320  between two cds 
+0000d620: 7365 7175 656e 6365 2066 696c 6573 2c20  sequence files, 
+0000d630: 7468 6520 7265 6c65 7661 6e74 2070 6172  the relevant par
+0000d640: 616d 6574 6572 2069 7320 6d6f 7374 6c79  ameter is mostly
+0000d650: 2074 6865 2073 616d 6520 6173 2077 686f   the same as who
+0000d660: 6c65 2070 6172 616e 6f6d 6520 696e 6665  le paranome infe
+0000d670: 7265 6e63 652c 2065 7863 6570 7420 666f  rence, except fo
+0000d680: 7220 7468 6520 7061 7261 6d65 7465 7220  r the parameter 
+0000d690: 602d 6360 206f 7220 602d 2d63 7363 6f72  `-c` or `--cscor
+0000d6a0: 6560 2c20 7768 6963 6820 7261 6e67 6573  e`, which ranges
+0000d6b0: 2062 6574 7765 656e 2030 2061 6e64 2031   between 0 and 1
+0000d6c0: 2061 6e64 2069 7320 7573 6564 2074 6f20   and is used to 
+0000d6d0: 7265 6c61 7820 7468 6520 7369 6d69 6c61  relax the simila
+0000d6e0: 7269 7479 2063 7574 6f66 6620 6672 6f6d  rity cutoff from
+0000d6f0: 2074 6865 2065 7863 6c75 7369 7665 2072   the exclusive r
+0000d700: 6563 6970 726f 6361 6c20 6265 7374 2068  eciprocal best h
+0000d710: 6974 7320 746f 2061 2063 6572 7461 696e  its to a certain
+0000d720: 2072 6174 696f 2061 7320 746f 2074 6865   ratio as to the
+0000d730: 2062 6573 7420 6869 7473 2e20 466f 7220   best hits. For 
+0000d740: 696e 7374 616e 6365 2c20 6966 2074 6865  instance, if the
+0000d750: 2067 656e 6520 6231 2066 726f 6d20 6765   gene b1 from ge
+0000d760: 6e6f 6d65 2042 2068 6173 2074 6865 2062  nome B has the b
+0000d770: 6573 7420 6869 7420 6765 6e65 2061 3120  est hit gene a1 
+0000d780: 6672 6f6d 2067 656e 6f6d 6520 4120 7769  from genome A wi
+0000d790: 7468 2074 6865 2062 6974 2073 636f 7265  th the bit score
+0000d7a0: 2061 7320 3130 302c 2077 6869 6368 2069   as 100, which i
+0000d7b0: 7320 6120 7363 6f72 696e 6720 6d61 7472  s a scoring matr
+0000d7c0: 6978 2069 6e64 6570 656e 6465 6e74 206d  ix independent m
+0000d7d0: 6561 7375 7265 206f 6620 7468 6520 286c  easure of the (l
+0000d7e0: 6f63 616c 2920 7369 6d69 6c61 7269 7479  ocal) similarity
+0000d7f0: 206f 6620 7468 6520 7477 6f20 616c 6967   of the two alig
+0000d800: 6e65 6420 7365 7175 656e 6365 732c 2077  ned sequences, w
+0000d810: 6974 6820 6c61 7267 6572 2076 616c 7565  ith larger value
+0000d820: 7320 696e 6469 6361 7469 6e67 2068 6967  s indicating hig
+0000d830: 6865 7220 7369 6d69 6c61 7269 7469 6573  her similarities
+0000d840: 2c20 6769 7665 6e20 7468 6520 602d 6320  , given the `-c 
+0000d850: 302e 3960 2c20 6765 6e65 7320 6672 6f6d  0.9`, genes from
+0000d860: 2067 656e 6f6d 6520 4120 7768 6963 6820   genome A which 
+0000d870: 6861 7320 7468 6520 6269 7420 7363 6f72  has the bit scor
+0000d880: 6520 7769 7468 2067 656e 6520 6231 2068  e with gene b1 h
+0000d890: 6967 6865 7220 7468 616e 2030 2e39 7831  igher than 0.9x1
+0000d8a0: 3030 2077 696c 6c20 616c 736f 2062 6520  00 will also be 
+0000d8b0: 7772 6974 7465 6e20 696e 2074 6865 2072  written in the r
+0000d8c0: 6573 756c 7420 6669 6c65 2c20 7768 6963  esult file, whic
+0000d8d0: 6820 696e 2061 2073 656e 7365 2061 7265  h in a sense are
+0000d8e0: 206e 6f74 2052 4248 7320 616e 796d 6f72   not RBHs anymor
+0000d8f0: 6520 6f66 2063 6f75 7273 652c 2062 7574  e of course, but
+0000d900: 2074 6865 2068 6967 686c 7920 7369 6d69   the highly simi
+0000d910: 6c61 7220 686f 6d6f 6c6f 6775 6520 7061  lar homologue pa
+0000d920: 6972 732e 2049 6620 6d6f 7265 2074 6861  irs. If more tha
+0000d930: 6e20 3220 7365 7175 656e 6365 2066 696c  n 2 sequence fil
+0000d940: 6573 2077 6572 6520 7072 6f76 6964 6564  es were provided
+0000d950: 2c20 6576 6572 7920 7061 6972 2d77 6973  , every pair-wis
+0000d960: 6520 5242 4873 2077 6f75 6c64 2062 6520  e RBHs would be 
+0000d970: 6361 6c63 756c 6174 6564 2065 7863 6570  calculated excep
+0000d980: 7420 666f 7220 7175 6572 7969 6e67 2074  t for querying t
+0000d990: 6865 2073 616d 6520 7365 7175 656e 6365  he same sequence
+0000d9a0: 2069 7473 656c 662e 2054 6865 206e 756d   itself. The num
+0000d9b0: 6265 7220 6f66 2070 6172 616c 6c65 6c20  ber of parallel 
+0000d9c0: 7468 7265 6164 7320 746f 2062 6f6f 7374  threads to boost
+0000d9d0: 6572 2074 6865 2072 756e 6e69 6e67 2073  er the running s
+0000d9e0: 7065 6564 2063 616e 2062 6520 7365 7420  peed can be set 
+0000d9f0: 6279 2074 6865 206f 7074 696f 6e20 602d  by the option `-
+0000da00: 6e60 206f 7220 602d 2d6e 7468 7265 6164  n` or `--nthread
+0000da10: 7360 2077 6869 6368 2069 7320 7375 6767  s` which is sugg
+0000da20: 6573 7465 6420 746f 2062 6520 7365 7420  ested to be set 
+0000da30: 6173 2028 4e2d 3129 4e2f 3220 7768 6572  as (N-1)N/2 wher
+0000da40: 6520 4e20 6973 2074 6865 206e 756d 6265  e N is the numbe
+0000da50: 7220 6f66 2063 6473 2066 696c 6573 2074  r of cds files t
+0000da60: 6f20 6163 6869 6576 6520 7468 6520 6869  o achieve the hi
+0000da70: 6768 6573 7420 6566 6669 6369 656e 6379  ghest efficiency
+0000da80: 2e20 5468 6520 6f70 7469 6f6e 2060 2d2d  . The option `--
+0000da90: 6f67 666f 726d 6174 6020 6361 6e20 6265  ogformat` can be
+0000daa0: 2073 6574 2074 6f20 6164 6420 696e 6465   set to add inde
+0000dab0: 7820 2866 6f72 2069 6e73 7461 6e63 6520  x (for instance 
+0000dac0: 4746 3030 3030 3030 3031 2920 746f 2074  GF00000001) to t
+0000dad0: 6865 206f 7574 7075 7420 5242 4820 6765  he output RBH ge
+0000dae0: 6e65 2066 616d 696c 6965 7320 7768 6963  ne families whic
+0000daf0: 6820 6361 6e20 6265 2066 7572 7468 6572  h can be further
+0000db00: 2075 7365 6420 696e 2074 6865 202a 4b2a   used in the *K*
+0000db10: 3c73 7562 3e53 3c2f 7375 623e 2063 616c  <sub>S</sub> cal
+0000db20: 6375 6c61 7469 6f6e 2062 7920 6077 6764  culation by `wgd
+0000db30: 206b 7364 602e 0a0a 2a2a 5468 6520 7375   ksd`...**The su
+0000db40: 6767 6573 7465 6420 636f 6d6d 616e 6420  ggested command 
+0000db50: 746f 2073 7461 7274 2077 6974 6820 6973  to start with is
+0000db60: 2061 6c73 6f20 756e 6465 7220 7468 6520   also under the 
+0000db70: 6465 6661 756c 7420 7365 7474 696e 6720  default setting 
+0000db80: 7769 7468 2074 6865 2063 6f6d 6d61 6e64  with the command
+0000db90: 2073 686f 776e 2062 656c 6f77 2a2a 0a0a   shown below**..
+0000dba0: 6060 600a 7767 6420 646d 6420 7365 7175  ```.wgd dmd sequ
+0000dbb0: 656e 6365 3120 7365 7175 656e 6365 320a  ence1 sequence2.
+0000dbc0: 6060 600a 0a2a 2a54 6865 2064 656c 696e  ```..**The delin
+0000dbd0: 6561 7469 6f6e 206f 6620 6c6f 6361 6c20  eation of local 
+0000dbe0: 4d52 4248 732a 2a0a 6060 600a 7767 6420  MRBHs**.```.wgd 
+0000dbf0: 646d 6420 7365 7175 656e 6365 3120 7365  dmd sequence1 se
+0000dc00: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000dc10: 3320 2d66 2073 6571 7565 6e63 6531 202d  3 -f sequence1 -
+0000dc20: 6520 3165 2d31 3020 2d62 7320 3130 3020  e 1e-10 -bs 100 
+0000dc30: 2d6e 7020 3520 282d 6e6e 2920 282d 6e20  -np 5 (-nn) (-n 
+0000dc40: 3429 2028 2d6b 6629 2028 2d6b 6429 2028  4) (-kf) (-kd) (
+0000dc50: 2d63 2030 2e39 2920 282d 2d74 6f5f 7374  -c 0.9) (--to_st
+0000dc60: 6f70 2920 282d 2d63 6473 2920 282d 6f20  op) (--cds) (-o 
+0000dc70: 7767 645f 646d 6429 2028 2d74 2077 6f72  wgd_dmd) (-t wor
+0000dc80: 6b69 6e67 5f74 6d70 290a 6060 600a 0a2a  king_tmp).```..*
+0000dc90: 2a54 6865 2064 6973 7469 6e63 7469 6f6e  *The distinction
+0000dca0: 2062 6574 7765 656e 206c 6f63 616c 2061   between local a
+0000dcb0: 6e64 2067 6c6f 6261 6c20 2868 6572 6575  nd global (hereu
+0000dcc0: 6e64 6572 2920 4d52 4248 7320 6973 2074  nder) MRBHs is t
+0000dcd0: 6861 7420 6c6f 6361 6c20 4d52 4248 7320  hat local MRBHs 
+0000dce0: 6172 6520 7468 6520 7265 7375 6c74 7320  are the results 
+0000dcf0: 6f66 206d 6572 6765 6420 5242 4873 206f  of merged RBHs o
+0000dd00: 6e20 6120 6a6f 696e 7420 666f 6361 6c20  n a joint focal 
+0000dd10: 7370 6563 6965 732c 2066 6f72 2069 6e73  species, for ins
+0000dd20: 7461 6e63 6520 696e 2061 2074 6872 6565  tance in a three
+0000dd30: 2073 7065 6369 6573 2073 7973 7465 6d20   species system 
+0000dd40: 2841 2c28 422c 4329 292c 2074 6865 206c  (A,(B,C)), the l
+0000dd50: 6f63 616c 204d 5242 4873 206f 6620 4120  ocal MRBHs of A 
+0000dd60: 6f6e 6c79 2072 6571 7569 7265 2074 6865  only require the
+0000dd70: 2063 616c 6375 6c61 7469 6f6e 206f 6620   calculation of 
+0000dd80: 5242 4873 2062 6574 7765 656e 2041 2061  RBHs between A a
+0000dd90: 6e64 2043 2028 6465 6e6f 7465 6420 6173  nd C (denoted as
+0000dda0: 2041 4329 2061 6e64 2041 4220 616e 6420   AC) and AB and 
+0000ddb0: 7468 656e 2074 6865 206d 6572 6769 6e67  then the merging
+0000ddc0: 206f 6620 4142 2061 6e64 2041 4320 6174   of AB and AC at
+0000ddd0: 2074 6865 2061 7869 7320 6f66 2041 2c20   the axis of A, 
+0000dde0: 7768 696c 6520 7468 6520 676c 6f61 626c  while the gloabl
+0000ddf0: 204d 5242 4873 2061 7265 2069 6e64 6570   MRBHs are indep
+0000de00: 656e 6465 6e74 206f 6620 666f 6361 6c20  endent of focal 
+0000de10: 7370 6563 6965 7320 696e 2074 6861 7420  species in that 
+0000de20: 6974 206a 7573 7420 6361 6c6c 7320 7468  it just calls th
+0000de30: 6520 6361 6c63 756c 6174 696f 6e73 206f  e calculations o
+0000de40: 6620 616c 6c20 706f 7373 6962 6c65 2073  f all possible s
+0000de50: 7065 6369 6573 2070 6169 7220 286e 6f74  pecies pair (not
+0000de60: 2073 656c 6620 746f 2073 656c 6620 7370   self to self sp
+0000de70: 6563 6965 7320 7061 6972 292c 2073 7563  ecies pair), suc
+0000de80: 6820 7468 6174 2041 422c 2041 432c 2061  h that AB, AC, a
+0000de90: 6e64 2042 4320 6172 6520 616c 6c20 746f  nd BC are all to
+0000dea0: 2062 6520 6361 6c63 756c 6174 6564 2061   be calculated a
+0000deb0: 6e64 206d 6572 6765 642e 2a2a 0a0a 5477  nd merged.**..Tw
+0000dec0: 6f20 7479 7065 7320 6f66 204d 5242 4873  o types of MRBHs
+0000ded0: 2061 7320 696e 7465 7072 6574 6174 6564   as intepretated
+0000dee0: 2061 626f 7665 2063 616e 2062 6520 6465   above can be de
+0000def0: 6c69 6e65 6174 6564 2062 7920 6077 6764  lineated by `wgd
+0000df00: 2064 6d64 602c 2074 6865 206c 6f63 616c   dmd`, the local
+0000df10: 204d 5242 4873 2061 6e64 2074 6865 2067   MRBHs and the g
+0000df20: 6c6f 6261 6c20 4d52 4248 732e 2054 6865  lobal MRBHs. The
+0000df30: 206c 6f63 616c 204d 5242 4873 2061 7265   local MRBHs are
+0000df40: 2063 6f6e 7374 7275 6374 6564 2062 7920   constructed by 
+0000df50: 6d65 7267 696e 6720 616c 6c20 7468 6520  merging all the 
+0000df60: 7265 6c65 7661 6e74 2052 4248 7320 6f6e  relevant RBHs on
+0000df70: 6c79 2077 6974 6820 7468 6520 666f 6361  ly with the foca
+0000df80: 6c20 7370 6563 6965 732c 2077 6869 6368  l species, which
+0000df90: 2069 7320 7365 7420 6279 2060 2d66 6020   is set by `-f` 
+0000dfa0: 6f72 2060 2d2d 666f 6375 7360 2e20 5468  or `--focus`. Th
+0000dfb0: 6520 7061 7261 6d65 7465 7220 602d 6b66  e parameter `-kf
+0000dfc0: 6020 6f72 2060 2d2d 6b65 6570 6661 7374  ` or `--keepfast
+0000dfd0: 6160 2063 616e 2062 6520 7365 7420 746f  a` can be set to
+0000dfe0: 2072 6574 6169 6e20 7468 6520 7365 7175   retain the sequ
+0000dff0: 656e 6365 2069 6e66 6f72 6d61 7469 6f6e  ence information
+0000e000: 206f 6620 6561 6368 204d 5242 482e 2054   of each MRBH. T
+0000e010: 6865 2070 6172 616d 6574 6572 2060 2d6b  he parameter `-k
+0000e020: 6460 206f 7220 602d 2d6b 6565 7064 7570  d` or `--keepdup
+0000e030: 6c69 6361 7465 7360 2064 6574 6572 6d69  licates` determi
+0000e040: 6e65 7320 7768 6574 6865 7220 7468 6520  nes whether the 
+0000e050: 7361 6d65 2067 656e 6573 2063 616e 2061  same genes can a
+0000e060: 7070 6561 7220 696e 2064 6966 6665 7265  ppear in differe
+0000e070: 6e74 206c 6f63 616c 204d 5242 4873 2e20  nt local MRBHs. 
+0000e080: 4e6f 726d 616c 6c79 2074 6865 7265 2077  Normally there w
+0000e090: 696c 6c20 6265 206e 6f20 6475 706c 6963  ill be no duplic
+0000e0a0: 6174 6573 2069 6e20 7468 6520 6c6f 6361  ates in the loca
+0000e0b0: 6c20 4d52 4248 7320 6275 7420 6966 2075  l MRBHs but if u
+0000e0c0: 7365 7273 2073 6574 2074 6865 2060 2d63  sers set the `-c
+0000e0d0: 6020 6173 2030 2e39 2028 6f72 2061 6e79  ` as 0.9 (or any
+0000e0e0: 2076 616c 7565 2073 6d61 6c6c 6572 2074   value smaller t
+0000e0f0: 6861 6e20 3129 2c20 6974 2773 206c 696b  han 1), it's lik
+0000e100: 656c 7920 7468 6174 2074 6865 2073 616d  ely that the sam
+0000e110: 6520 6765 6e65 2077 696c 6c20 6861 7665  e gene will have
+0000e120: 2063 6861 6e63 6520 746f 2061 7070 6561   chance to appea
+0000e130: 7220 6d75 6c74 6970 6c65 2074 696d 6573  r multiple times
+0000e140: 2069 6e20 6469 6666 6572 656e 7420 6c6f   in different lo
+0000e150: 6361 6c20 4d52 4248 732e 2054 6861 7420  cal MRBHs. That 
+0000e160: 6973 2074 6f20 7361 792c 2074 6865 2070  is to say, the p
+0000e170: 6172 616d 6574 6572 2060 2d6b 6460 2069  arameter `-kd` i
+0000e180: 7320 6d65 616e 696e 6766 756c 206f 6e6c  s meaningful onl
+0000e190: 7920 7768 656e 2069 7427 7320 7365 7420  y when it's set 
+0000e1a0: 746f 6765 7468 6572 2077 6974 6820 7468  together with th
+0000e1b0: 6520 7061 7261 6d65 7465 7220 602d 6360  e parameter `-c`
+0000e1c0: 2e20 5468 6520 6e75 6d62 6572 206f 6620  . The number of 
+0000e1d0: 7061 7261 6c6c 656c 2074 6872 6561 6473  parallel threads
+0000e1e0: 2069 7320 7375 6767 6573 7465 6420 746f   is suggested to
+0000e1f0: 2062 6520 7365 7420 6173 2074 6865 206e   be set as the n
+0000e200: 756d 6265 7220 6f66 2063 6473 2066 696c  umber of cds fil
+0000e210: 6573 206d 696e 7573 2031 2e0a 0a2a 2a41  es minus 1...**A
+0000e220: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
+0000e230: 696e 6720 7275 6e20 6973 2075 6e64 6572  ing run is under
+0000e240: 2074 6865 2064 6566 6175 6c74 2070 6172   the default par
+0000e250: 616d 6574 6572 2077 6974 6820 7468 6520  ameter with the 
+0000e260: 636f 6d6d 616e 6420 7368 6f77 6e20 6173  command shown as
+0000e270: 2062 656c 6f77 2e2a 2a0a 0a60 6060 0a77   below.**..```.w
+0000e280: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
+0000e290: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
+0000e2a0: 6e63 6533 202d 6620 7365 7175 656e 6365  nce3 -f sequence
+0000e2b0: 310a 6060 600a 0a2a 2a54 6865 2064 656c  1.```..**The del
+0000e2c0: 696e 6561 7469 6f6e 206f 6620 676c 6f62  ineation of glob
+0000e2d0: 616c 204d 5242 4873 2a2a 0a60 6060 0a77  al MRBHs**.```.w
+0000e2e0: 6764 2064 6d64 2073 6571 7565 6e63 6531  gd dmd sequence1
+0000e2f0: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
+0000e300: 6e63 6533 202d 676d 202d 6520 3165 2d31  nce3 -gm -e 1e-1
+0000e310: 3020 2d62 7320 3130 3020 2d6e 7020 3520  0 -bs 100 -np 5 
+0000e320: 282d 6e6e 2920 282d 6e20 3429 2028 2d6b  (-nn) (-n 4) (-k
+0000e330: 6629 2028 2d6b 6429 2028 2d63 2030 2e39  f) (-kd) (-c 0.9
+0000e340: 2920 282d 2d74 6f5f 7374 6f70 2920 282d  ) (--to_stop) (-
+0000e350: 2d63 6473 2920 282d 6f20 7767 645f 646d  -cds) (-o wgd_dm
+0000e360: 6429 2028 2d74 2077 6f72 6b69 6e67 5f74  d) (-t working_t
+0000e370: 6d70 290a 6060 600a 0a54 6865 2067 6c6f  mp).```..The glo
+0000e380: 6261 6c20 4d52 4248 7320 6973 2063 6f6e  bal MRBHs is con
+0000e390: 7374 7275 6374 6564 2062 7920 6578 6861  structed by exha
+0000e3a0: 7573 7469 7665 6c79 206d 6572 6769 6e67  ustively merging
+0000e3b0: 2061 6c6c 2074 6865 2070 6f73 7369 626c   all the possibl
+0000e3c0: 6520 7061 6972 2d77 6973 6520 5242 4873  e pair-wise RBHs
+0000e3d0: 2065 7863 6570 7420 666f 7220 7175 6572   except for quer
+0000e3e0: 7969 6e67 2074 6865 2073 6571 7565 6e63  ying the sequenc
+0000e3f0: 6520 6974 7365 6c66 2c20 7768 6963 6820  e itself, which 
+0000e400: 6361 6e20 6265 2069 6e69 7469 6174 6564  can be initiated
+0000e410: 2062 7920 6164 6420 7468 6520 666c 6167   by add the flag
+0000e420: 2060 2d67 6d60 206f 7220 602d 2d67 6c6f   `-gm` or `--glo
+0000e430: 6261 6c6d 7262 6860 2e20 5468 6520 7265  balmrbh`. The re
+0000e440: 7374 206f 6620 7265 6c65 7661 6e74 2070  st of relevant p
+0000e450: 6172 616d 6574 6572 7320 7374 6179 7320  arameters stays 
+0000e460: 7468 6520 7361 6d65 2061 7320 7468 6520  the same as the 
+0000e470: 6c6f 6361 6c20 4d52 4248 732e 2054 6865  local MRBHs. The
+0000e480: 206e 756d 6265 7220 6f66 2070 6172 616c   number of paral
+0000e490: 6c65 6c20 7468 7265 6164 7320 6973 2073  lel threads is s
+0000e4a0: 7567 6765 7374 6564 2074 6f20 6265 2073  uggested to be s
+0000e4b0: 6574 2061 7320 284e 2d31 294e 2f32 2074  et as (N-1)N/2 t
+0000e4c0: 6f6f 2077 6865 7265 204e 2069 7320 7468  oo where N is th
+0000e4d0: 6520 6e75 6d62 6572 206f 6620 6364 7320  e number of cds 
+0000e4e0: 6669 6c65 7320 746f 2061 6368 6965 7665  files to achieve
+0000e4f0: 2074 6865 2068 6967 6865 7374 2065 6666   the highest eff
+0000e500: 6963 6965 6e63 792e 0a0a 2a2a 4120 7375  iciency...**A su
+0000e510: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+0000e520: 2072 756e 2069 7320 756e 6465 7220 7468   run is under th
+0000e530: 6520 6465 6661 756c 7420 7061 7261 6d65  e default parame
+0000e540: 7465 7220 7769 7468 2074 6865 2063 6f6d  ter with the com
+0000e550: 6d61 6e64 2073 686f 776e 2061 7320 6265  mand shown as be
+0000e560: 6c6f 772e 2a2a 0a0a 6060 600a 7767 6420  low.**..```.wgd 
+0000e570: 646d 6420 7365 7175 656e 6365 3120 7365  dmd sequence1 se
+0000e580: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+0000e590: 3320 2d67 6d0a 6060 600a 0a2a 2a54 6865  3 -gm.```..**The
+0000e5a0: 2064 656c 696e 6561 7469 6f6e 206f 6620   delineation of 
+0000e5b0: 6f72 7468 6f67 726f 7570 732a 2a0a 6060  orthogroups**.``
+0000e5c0: 600a 7767 6420 646d 6420 7365 7175 656e  `.wgd dmd sequen
+0000e5d0: 6365 3120 7365 7175 656e 6365 3220 7365  ce1 sequence2 se
+0000e5e0: 7175 656e 6365 3320 2d6f 6920 2d6f 6f20  quence3 -oi -oo 
+0000e5f0: 2d65 2031 652d 3130 202d 6273 2031 3030  -e 1e-10 -bs 100
+0000e600: 202d 6e70 2035 2028 2d6e 6e29 2028 2d63   -np 5 (-nn) (-c
+0000e610: 6329 2028 2d74 6529 2028 2d6d 6320 302e  c) (-te) (-mc 0.
+0000e620: 3829 2028 2d67 6e29 2028 2d74 7265 6520  8) (-gn) (-tree 
+0000e630: 2766 6173 7474 7265 6527 2920 282d 7473  'fasttree') (-ts
+0000e640: 2027 2d66 6173 7465 7374 2729 2028 2d6e   '-fastest') (-n
+0000e650: 2034 2920 282d 2d74 6f5f 7374 6f70 2920   4) (--to_stop) 
+0000e660: 282d 2d63 6473 2920 282d 6f20 7767 645f  (--cds) (-o wgd_
+0000e670: 646d 6429 2028 2d74 2077 6f72 6b69 6e67  dmd) (-t working
+0000e680: 5f74 6d70 290a 6060 600a 0a49 6e20 6077  _tmp).```..In `w
+0000e690: 6764 2076 3260 2c20 7765 2061 6c73 6f20  gd v2`, we also 
+0000e6a0: 696d 706c 656d 656e 7465 6420 616e 2061  implemented an a
+0000e6b0: 6c67 6f72 6974 686d 206f 6620 6465 6c69  lgorithm of deli
+0000e6c0: 6e65 6174 696e 6720 6f72 7468 6f67 726f  neating orthogro
+0000e6d0: 7570 732c 2077 6869 6368 2063 616e 2062  ups, which can b
+0000e6e0: 6520 696e 6974 6961 7465 6420 7769 7468  e initiated with
+0000e6f0: 2074 6865 2070 6172 616d 6574 6572 2060   the parameter `
+0000e700: 2d6f 6960 206f 7220 602d 2d6f 7274 686f  -oi` or `--ortho
+0000e710: 696e 6665 7260 2e20 5477 6f20 7761 7973  infer`. Two ways
+0000e720: 206f 6620 6465 6c69 6e65 6174 696f 6e20   of delineation 
+0000e730: 6361 6e20 6265 2063 686f 7365 6e2c 2074  can be chosen, t
+0000e740: 6865 2063 6f6e 6361 7465 6e61 7469 6f6e  he concatenation
+0000e750: 2077 6179 2028 7365 7420 6279 2074 6865   way (set by the
+0000e760: 2070 6172 616d 6574 6572 2060 2d63 6360   parameter `-cc`
+0000e770: 206f 7220 602d 2d63 6f6e 6361 7460 2920   or `--concat`) 
+0000e780: 6f72 2074 6865 206e 6f6e 2d63 6f6e 6361  or the non-conca
+0000e790: 7465 6e61 7469 6f6e 2028 6465 6661 756c  tenation (defaul
+0000e7a0: 7429 2077 6179 2e20 496e 2062 7269 6566  t) way. In brief
+0000e7b0: 2c20 7468 6520 636f 6e63 6174 656e 6174  , the concatenat
+0000e7c0: 696f 6e20 7761 7920 6f66 2064 656c 696e  ion way of delin
+0000e7d0: 6561 7469 6e67 206f 7274 686f 6772 6f75  eating orthogrou
+0000e7e0: 7073 2073 7461 7274 7320 7769 7468 2063  ps starts with c
+0000e7f0: 6f6e 6361 7465 6e61 7469 6e67 2061 6c6c  oncatenating all
+0000e800: 2074 6865 2073 6571 7565 6e63 6573 2069   the sequences i
+0000e810: 6e74 6f20 6120 7369 6e67 6c65 2073 6571  nto a single seq
+0000e820: 7565 6e63 6520 6669 6c65 2061 6e64 2074  uence file and t
+0000e830: 6865 6e20 696e 6665 7272 696e 6720 7468  hen inferring th
+0000e840: 6520 7768 6f6c 6520 7061 7261 6e6f 6d65  e whole paranome
+0000e850: 206f 6620 7468 6973 2073 696e 676c 6520   of this single 
+0000e860: 7365 7175 656e 6365 2066 696c 6520 7769  sequence file wi
+0000e870: 7468 2074 6865 2063 6c75 7374 6572 696e  th the clusterin
+0000e880: 6720 7265 7375 6c74 7320 6d61 7070 6564  g results mapped
+0000e890: 2062 6163 6b20 746f 2074 6865 2062 656c   back to the bel
+0000e8a0: 6f6e 6769 6e67 2073 7065 6369 6573 2e20  onging species. 
+0000e8b0: 5768 696c 6520 7468 6520 6e6f 6e2d 636f  While the non-co
+0000e8c0: 6e63 6174 656e 6174 696f 6e20 7761 7920  ncatenation way 
+0000e8d0: 7374 6172 7473 2077 6974 6820 7265 7370  starts with resp
+0000e8e0: 6563 7469 7665 2070 6169 722d 7769 7365  ective pair-wise
+0000e8f0: 2064 6961 6d6f 6e64 2073 6561 7263 6820   diamond search 
+0000e900: 2869 6e63 6c75 6469 6e67 2071 7565 7279  (including query
+0000e910: 696e 6720 7468 6520 7361 6d65 2073 6571  ing the same seq
+0000e920: 7565 6e63 6520 6974 7365 6c66 2920 616e  uence itself) an
+0000e930: 6420 7468 656e 2061 6c6c 2074 6865 2073  d then all the s
+0000e940: 6571 7565 6e63 6520 7369 6d69 6c61 7269  equence similari
+0000e950: 7479 2074 6162 6c65 7320 7769 6c6c 2062  ty tables will b
+0000e960: 6520 636f 6e63 6174 656e 6174 6564 2061  e concatenated a
+0000e970: 6e64 2063 6c75 7374 6572 6564 2069 6e74  nd clustered int
+0000e980: 6f20 6f72 7468 6f67 726f 7570 732e 2053  o orthogroups. S
+0000e990: 6f6d 6520 6f74 6865 7220 706f 7373 6962  ome other possib
+0000e9a0: 6c79 2075 7365 6675 6c20 706f 7374 2d63  ly useful post-c
+0000e9b0: 6c75 7374 6572 696e 6720 6675 6e63 7469  lustering functi
+0000e9c0: 6f6e 7320 6361 6e20 6265 2069 6e69 7469  ons can be initi
+0000e9d0: 6174 6564 2c20 696e 636c 7564 696e 6720  ated, including 
+0000e9e0: 7468 6520 7061 7261 6d65 7465 7220 602d  the parameter `-
+0000e9f0: 7465 6020 6f72 2060 2d2d 7465 7374 736f  te` or `--testso
+0000ea00: 6760 2c20 7768 6963 6820 6361 6e20 6265  g`, which can be
+0000ea10: 2073 6574 2074 6f20 7374 6172 7420 7468   set to start th
+0000ea20: 6520 756e 6269 6173 6564 2074 6573 7420  e unbiased test 
+0000ea30: 6f66 2073 696e 676c 652d 636f 7079 2067  of single-copy g
+0000ea40: 656e 6520 6661 6d69 6c69 6573 2028 6e6f  ene families (no
+0000ea50: 7465 2074 6861 7420 7468 6973 2066 756e  te that this fun
+0000ea60: 6374 696f 6e20 6e65 6564 7320 6068 6d6d  ction needs `hmm
+0000ea70: 6572 6020 2876 332e 3162 3229 2074 6f20  er` (v3.1b2) to 
+0000ea80: 6265 2069 6e73 7461 6c6c 6564 2069 6e20  be installed in 
+0000ea90: 7468 6520 656e 7669 726f 6e6d 656e 7420  the environment 
+0000eaa0: 7061 7468 292c 2074 6865 2070 6172 616d  path), the param
+0000eab0: 6574 6572 2060 2d6d 6360 206f 7220 602d  eter `-mc` or `-
+0000eac0: 2d6d 736f 6763 7574 602c 2072 616e 6769  -msogcut`, rangi
+0000ead0: 6e67 2062 6574 7765 656e 2030 2074 6f20  ng between 0 to 
+0000eae0: 312c 2077 6869 6368 2063 616e 2062 6520  1, which can be 
+0000eaf0: 7365 7420 746f 2073 6561 7263 6820 7468  set to search th
+0000eb00: 6520 736f 2d63 616c 6c65 6420 6d6f 7374  e so-called most
+0000eb10: 6c79 2073 696e 676c 652d 636f 7079 2066  ly single-copy f
+0000eb20: 616d 696c 7920 7768 6963 6820 6861 7320  amily which has 
+0000eb30: 6869 6768 6572 2074 6861 6e20 6365 7274  higher than cert
+0000eb40: 6169 6e20 6375 742d 6f66 6620 7065 7263  ain cut-off perc
+0000eb50: 656e 7461 6765 206f 6620 7370 6563 6965  entage of specie
+0000eb60: 7320 636f 7665 7261 6765 2c20 7468 6520  s coverage, the 
+0000eb70: 7061 7261 6d65 7465 7220 602d 676e 6020  parameter `-gn` 
+0000eb80: 6f72 2060 2d2d 6765 746e 736f 6760 2c20  or `--getnsog`, 
+0000eb90: 7768 6963 6820 6361 6e20 6265 2073 6574  which can be set
+0000eba0: 2074 6f20 7365 6172 6368 2066 6f72 206e   to search for n
+0000ebb0: 6573 7465 6420 7369 6e67 6c65 2d63 6f70  ested single-cop
+0000ebc0: 7920 6765 6e65 2066 616d 696c 6965 7320  y gene families 
+0000ebd0: 284e 534f 4773 2920 7768 6963 6820 6973  (NSOGs) which is
+0000ebe0: 206f 7269 6769 6e61 6c6c 7920 6d75 6c74   originally mult
+0000ebf0: 6979 2d63 6f70 7920 6275 7420 6861 7320  iy-copy but has 
+0000ec00: 6120 286d 6f73 746c 7929 2073 696e 676c  a (mostly) singl
+0000ec10: 652d 636f 7079 2062 7261 6e63 6820 2877  e-copy branch (w
+0000ec20: 6869 6368 2072 6571 7569 7265 7320 7468  hich requires th
+0000ec30: 6520 6368 6f73 656e 2074 7265 652d 696e  e chosen tree-in
+0000ec40: 6665 7265 6e63 6520 7072 6f67 7261 6d20  ference program 
+0000ec50: 7365 7420 6279 2060 2d74 7265 6560 206f  set by `-tree` o
+0000ec60: 7220 602d 2d74 7265 655f 6d65 7468 6f64  r `--tree_method
+0000ec70: 6020 746f 2062 6520 7072 652d 696e 7374  ` to be pre-inst
+0000ec80: 616c 6c65 6420 696e 2074 6865 2065 6e76  alled in the env
+0000ec90: 6972 6f6e 6d65 6e74 2070 6174 6820 7769  ironment path wi
+0000eca0: 7468 2074 6865 2070 6172 616d 6574 6572  th the parameter
+0000ecb0: 7320 7365 7474 696e 6720 666f 7220 6765  s setting for ge
+0000ecc0: 6e65 2074 7265 6520 696e 6665 7265 6e63  ne tree inferenc
+0000ecd0: 6520 636f 6e74 726f 6c6c 6564 2062 7920  e controlled by 
+0000ece0: 602d 7473 6020 6f72 2060 2d2d 7472 6565  `-ts` or `--tree
+0000ecf0: 7365 7460 292e 2054 6865 2070 726f 6772  set`). The progr
+0000ed00: 616d 2060 7767 6420 646d 6460 2077 6f75  am `wgd dmd` wou
+0000ed10: 6c64 2073 7469 6c6c 2063 6f6e 6475 6374  ld still conduct
+0000ed20: 2074 6865 2052 4248 7320 6361 6c63 756c   the RBHs calcul
+0000ed30: 6174 696f 6e20 756e 6c65 7373 2074 6865  ation unless the
+0000ed40: 2070 6172 616d 6574 6572 2060 2d6f 6f60   parameter `-oo`
+0000ed50: 206f 7220 602d 2d6f 6e6c 796f 7274 686f   or `--onlyortho
+0000ed60: 6020 7761 7320 7365 742e 2049 6620 6f6e  ` was set. If on
+0000ed70: 6520 6f6e 6c79 2077 616e 7473 2074 6f20  e only wants to 
+0000ed80: 696e 6665 7220 7468 6520 6f72 7468 6f67  infer the orthog
+0000ed90: 726f 7570 732c 2069 7427 7320 7375 6767  roups, it's sugg
+0000eda0: 6573 7465 6420 746f 2061 6464 2074 6865  ested to add the
+0000edb0: 2066 6c61 6720 602d 6f6f 6020 746f 206a   flag `-oo` to j
+0000edc0: 7573 7420 696d 706c 656d 656e 7420 7468  ust implement th
+0000edd0: 6520 6f72 7468 6f67 726f 7570 7320 6465  e orthogroups de
+0000ede0: 6c69 6e65 6174 696f 6e20 616e 616c 7973  lineation analys
+0000edf0: 6973 2e20 5468 6520 6e75 6d62 6572 206f  is. The number o
+0000ee00: 6620 7061 7261 6c6c 656c 2074 6872 6561  f parallel threa
+0000ee10: 6473 2069 7320 7375 6767 6573 7465 6420  ds is suggested 
+0000ee20: 746f 2062 6520 7365 7420 6173 2028 4e2b  to be set as (N+
+0000ee30: 3129 4e2f 3220 7768 6572 6520 4e20 6973  1)N/2 where N is
+0000ee40: 2074 6865 206e 756d 6265 7220 6f66 2063   the number of c
+0000ee50: 6473 2066 696c 6573 2074 6f20 6163 6869  ds files to achi
+0000ee60: 6576 6520 7468 6520 6869 6768 6573 7420  eve the highest 
+0000ee70: 6566 6669 6369 656e 6379 2073 696e 6365  efficiency since
+0000ee80: 2074 6865 2073 656c 662d 636f 6d70 6172   the self-compar
+0000ee90: 6973 6f6e 2069 7320 616c 736f 2069 6e63  ison is also inc
+0000eea0: 6c75 6465 642e 0a0a 2a2a 5468 6520 6465  luded...**The de
+0000eeb0: 6661 756c 7420 7365 7474 696e 6720 6f66  fault setting of
+0000eec0: 2070 6172 616d 6574 6572 7320 6973 2061   parameters is a
+0000eed0: 2072 6561 736f 6e61 626c 6520 7374 6172   reasonable star
+0000eee0: 7469 6e67 2070 6f69 6e74 2077 6974 6820  ting point with 
+0000eef0: 7468 6520 636f 6d6d 616e 6420 6173 2062  the command as b
+0000ef00: 656c 6f77 2e2a 2a0a 0a60 6060 0a77 6764  elow.**..```.wgd
+0000ef10: 2064 6d64 2073 6571 7565 6e63 6531 2073   dmd sequence1 s
+0000ef20: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+0000ef30: 6533 202d 6f69 202d 6f6f 0a60 6060 0a0a  e3 -oi -oo.```..
+0000ef40: 2a2a 5468 6520 636f 6c6c 696e 6561 7220  **The collinear 
+0000ef50: 636f 616c 6573 6365 6e63 6520 696e 6665  coalescence infe
+0000ef60: 7265 6e63 6520 6f66 2070 6879 6c6f 6765  rence of phyloge
+0000ef70: 6e79 2a2a 0a60 6060 0a77 6764 2064 6d64  ny**.```.wgd dmd
+0000ef80: 2073 6571 7565 6e63 6531 2073 6571 7565   sequence1 seque
+0000ef90: 6e63 6532 2073 6571 7565 6e63 6533 202d  nce2 sequence3 -
+0000efa0: 6170 2061 7064 6174 6120 2d73 6d20 736d  ap apdata -sm sm
+0000efb0: 6461 7461 202d 6c65 206c 6564 6174 6120  data -le ledata 
+0000efc0: 2d67 7420 6774 6461 7461 202d 636f 6320  -gt gtdata -coc 
+0000efd0: 282d 7472 6565 2027 6661 7374 7472 6565  (-tree 'fasttree
+0000efe0: 2729 2028 2d74 7320 272d 6661 7374 6573  ') (-ts '-fastes
+0000eff0: 7427 2920 282d 6e20 3429 2028 2d2d 746f  t') (-n 4) (--to
+0000f000: 5f73 746f 7029 2028 2d2d 6364 7329 2028  _stop) (--cds) (
+0000f010: 2d6f 2077 6764 5f64 6d64 2920 282d 7420  -o wgd_dmd) (-t 
+0000f020: 776f 726b 696e 675f 746d 7029 0a60 6060  working_tmp).```
+0000f030: 0a0a 4120 6e6f 7665 6c20 7068 796c 6f67  ..A novel phylog
+0000f040: 656e 6574 6963 2069 6e66 6572 656e 6365  enetic inference
+0000f050: 206d 6574 686f 6420 6e61 6d65 6420 2263   method named "c
+0000f060: 6f6c 6c69 6e65 6172 2063 6f61 6c65 7363  ollinear coalesc
+0000f070: 656e 6365 2069 6e66 6572 656e 6365 2220  ence inference" 
+0000f080: 6973 2061 6c73 6f20 696d 706c 656d 656e  is also implemen
+0000f090: 7465 6420 696e 2060 7767 6420 7632 602e  ted in `wgd v2`.
+0000f0a0: 2046 6f72 2074 6869 7320 616e 616c 7973   For this analys
+0000f0b0: 6973 2c20 7573 6572 7320 6e65 6564 2074  is, users need t
+0000f0c0: 6f20 7072 6f76 6964 6520 7468 6520 616e  o provide the an
+0000f0d0: 6368 6f72 2070 6f69 6e74 7320 6669 6c65  chor points file
+0000f0e0: 2062 7920 602d 6170 6020 6f72 2060 2d2d   by `-ap` or `--
+0000f0f0: 616e 6368 6f72 706f 696e 7473 602c 2074  anchorpoints`, t
+0000f100: 6865 2063 6f6c 6c69 6e65 6172 2073 6567  he collinear seg
+0000f110: 6d65 6e74 7320 6669 6c65 2062 7920 602d  ments file by `-
+0000f120: 736d 6020 6f72 2060 2d2d 7365 676d 656e  sm` or `--segmen
+0000f130: 7473 602c 2074 6865 206c 6973 7473 6567  ts`, the listseg
+0000f140: 6d65 6e74 7320 6669 6c65 2062 7920 602d  ments file by `-
+0000f150: 6c65 6020 6f72 2060 2d2d 6c69 7374 656c  le` or `--listel
+0000f160: 656d 656e 7473 602c 2061 6e64 2074 6865  ements`, and the
+0000f170: 2067 656e 6520 7461 626c 6520 6669 6c65   gene table file
+0000f180: 2062 7920 602d 6774 6020 6f72 2060 2d2d   by `-gt` or `--
+0000f190: 6765 6e65 7461 626c 6560 2c20 616c 6c20  genetable`, all 
+0000f1a0: 6f66 2077 6869 6368 2063 616e 2062 6520  of which can be 
+0000f1b0: 7072 6f64 7563 6564 2069 6e20 7468 6520  produced in the 
+0000f1c0: 7072 6f67 7261 6d20 6077 6764 2073 796e  program `wgd syn
+0000f1d0: 602e 2054 6865 2070 6172 616d 6574 6572  `. The parameter
+0000f1e0: 2060 2d63 6f63 6020 6f72 2060 2d2d 636f   `-coc` or `--co
+0000f1f0: 6c6c 696e 6561 7263 6f61 6c65 7363 656e  llinearcoalescen
+0000f200: 6365 6020 6e65 6564 7320 746f 2062 6520  ce` needs to be 
+0000f210: 7365 7420 746f 2073 7461 7274 2074 6869  set to start thi
+0000f220: 7320 616e 616c 7973 6973 2e20 5468 6520  s analysis. The 
+0000f230: 7472 6565 2d69 6e66 6572 656e 6365 2070  tree-inference p
+0000f240: 726f 6772 616d 2061 6e64 2074 6865 2061  rogram and the a
+0000f250: 7373 6f63 6961 7465 6420 7061 7261 6d65  ssociated parame
+0000f260: 7465 7273 2063 616e 2062 6520 7365 7420  ters can be set 
+0000f270: 6a75 7374 2061 7320 6162 6f76 6520 6279  just as above by
+0000f280: 2060 2d74 7265 6560 206f 7220 602d 2d74   `-tree` or `--t
+0000f290: 7265 655f 6d65 7468 6f64 6020 616e 6420  ree_method` and 
+0000f2a0: 602d 7473 6020 6f72 2060 2d2d 7472 6565  `-ts` or `--tree
+0000f2b0: 7365 7460 2e20 506c 6561 7365 2061 6c73  set`. Please als
+0000f2c0: 6f20 6d61 6b65 2073 7572 6520 7468 6520  o make sure the 
+0000f2d0: 6368 6f73 656e 2074 7265 652d 696e 6665  chosen tree-infe
+0000f2e0: 7265 6e63 6520 7072 6f67 7261 6d20 6973  rence program is
+0000f2f0: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
+0000f300: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
+0000f310: 7468 2e20 5468 6520 7072 6f67 7261 6d20  th. The program 
+0000f320: 6061 7374 7261 6c2d 7072 6f60 2069 7320  `astral-pro` is 
+0000f330: 7265 7175 6972 6564 2074 6f20 6265 2069  required to be i
+0000f340: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+0000f350: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+0000f360: 2074 6f6f 2e20 4e6f 7465 2074 6861 7420   too. Note that 
+0000f370: 7468 6572 6520 7368 6f75 6c64 2062 6520  there should be 
+0000f380: 6e6f 2064 7570 6c69 6361 7465 6420 6765  no duplicated ge
+0000f390: 6e65 2049 4473 2069 6e20 7468 6520 7365  ne IDs in the se
+0000f3a0: 7175 656e 6365 2066 696c 652e 2054 6865  quence file. The
+0000f3b0: 2070 6172 616c 6c65 6c20 7468 7265 6164   parallel thread
+0000f3c0: 7320 6865 7265 2061 7265 2074 6f20 6163  s here are to ac
+0000f3d0: 6365 6c65 7261 7465 2074 6865 2073 6571  celerate the seq
+0000f3e0: 7565 6e63 6520 616c 6967 6e6d 656e 7420  uence alignment 
+0000f3f0: 616e 6420 6765 6e65 2074 7265 6520 696e  and gene tree in
+0000f400: 6665 7265 6e63 6520 666f 7220 6561 6368  ference for each
+0000f410: 2067 656e 6520 6661 6d69 6c79 2061 6e64   gene family and
+0000f420: 2074 6875 7320 7375 6767 6573 7465 6420   thus suggested 
+0000f430: 746f 2062 6520 7365 7420 6173 206d 7563  to be set as muc
+0000f440: 6820 6173 2074 6865 206e 756d 6265 7220  h as the number 
+0000f450: 6f66 2067 656e 6520 6661 6d69 6c69 6573  of gene families
+0000f460: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
+0000f470: 2073 7461 7274 696e 6720 7275 6e20 6f66   starting run of
+0000f480: 2074 6869 7320 616e 616c 7973 6973 2069   this analysis i
+0000f490: 7320 7769 7468 2074 6865 2073 696d 706c  s with the simpl
+0000f4a0: 6520 636f 6d6d 616e 6420 6265 6c6f 772e  e command below.
+0000f4b0: 2a2a 0a0a 6060 600a 7767 6420 646d 6420  **..```.wgd dmd 
+0000f4c0: 7365 7175 656e 6365 3120 7365 7175 656e  sequence1 sequen
+0000f4d0: 6365 3220 7365 7175 656e 6365 3320 2d61  ce2 sequence3 -a
+0000f4e0: 7020 6170 6461 7461 202d 736d 2073 6d64  p apdata -sm smd
+0000f4f0: 6174 6120 2d6c 6520 6c65 6461 7461 202d  ata -le ledata -
+0000f500: 6774 2067 7464 6174 6120 2d63 6f63 0a60  gt gtdata -coc.`
+0000f510: 6060 0a0a 2323 2320 7767 6420 6b73 640a  ``..### wgd ksd.
+0000f520: 0a2a 2a54 6865 2063 6f6e 7374 7275 6374  .**The construct
+0000f530: 696f 6e20 6f66 2077 686f 6c65 2070 6172  ion of whole par
+0000f540: 616e 6f6d 6520 2a4b 2a3c 7375 623e 533c  anome *K*<sub>S<
+0000f550: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+0000f560: 6275 7469 6f6e 2a2a 0a60 6060 0a77 6764  bution**.```.wgd
+0000f570: 206b 7364 2066 616d 696c 6965 7320 7365   ksd families se
+0000f580: 7175 656e 6365 2028 2d6f 2077 6764 5f6b  quence (-o wgd_k
+0000f590: 7364 202d 7420 7767 645f 6b73 645f 746d  sd -t wgd_ksd_tm
+0000f5a0: 7020 2d2d 6e74 6872 6561 6473 2034 202d  p --nthreads 4 -
+0000f5b0: 2d74 6f5f 7374 6f70 202d 2d63 6473 202d  -to_stop --cds -
+0000f5c0: 2d70 6169 7277 6973 6520 2d2d 7374 7269  -pairwise --stri
+0000f5d0: 705f 6761 7073 202d 2d61 6c69 676e 6572  p_gaps --aligner
+0000f5e0: 206d 6166 6674 202d 2d74 7265 655f 6d65   mafft --tree_me
+0000f5f0: 7468 6f64 2066 6173 7474 7265 6520 2d2d  thod fasttree --
+0000f600: 6e6f 6465 5f61 7665 7261 6765 290a 6060  node_average).``
+0000f610: 600a 0a54 6865 2070 726f 6772 616d 2060  `..The program `
+0000f620: 7767 6420 6b73 6460 2c20 6173 2069 6d70  wgd ksd`, as imp
+0000f630: 6965 6420 6279 2069 7473 206e 616d 652c  ied by its name,
+0000f640: 2069 7320 666f 7220 7468 6520 636f 6e73   is for the cons
+0000f650: 7472 7563 7469 6f6e 206f 6620 2a4b 2a3c  truction of *K*<
+0000f660: 7375 623e 533c 2f73 7562 3e20 6167 6520  sub>S</sub> age 
+0000f670: 6469 7374 7269 6275 7469 6f6e 2e20 4578  distribution. Ex
+0000f680: 6365 7074 2066 6f72 2074 6865 2061 666f  cept for the afo
+0000f690: 7265 6d65 6e74 696f 6e65 6420 7061 7261  rementioned para
+0000f6a0: 6d65 7465 7273 2073 7563 6820 6173 2060  meters such as `
+0000f6b0: 2d2d 746f 5f73 746f 7060 2061 6e64 2060  --to_stop` and `
+0000f6c0: 2d2d 6364 7360 2c20 7468 6572 6520 6172  --cds`, there ar
+0000f6d0: 6520 736f 6d65 2069 6d70 6f72 7461 6e74  e some important
+0000f6e0: 2070 6172 616d 6574 6572 7320 7468 6174   parameters that
+0000f6f0: 2068 6176 6520 6372 7563 6961 6c20 696d   have crucial im
+0000f700: 7061 6374 206f 6e20 7468 6520 2a4b 2a3c  pact on the *K*<
+0000f710: 7375 623e 533c 2f73 7562 3e20 6573 7469  sub>S</sub> esti
+0000f720: 6d61 7469 6f6e 2e20 5468 6520 6f70 7469  mation. The opti
+0000f730: 6f6e 2060 2d2d 7061 6972 7769 7365 6020  on `--pairwise` 
+0000f740: 6973 2061 2076 6572 7920 696d 706f 7274  is a very import
+0000f750: 616e 7420 7061 7261 6d65 7465 7220 666f  ant parameter fo
+0000f760: 7220 7468 6520 2a4b 2a3c 7375 623e 533c  r the *K*<sub>S<
+0000f770: 2f73 7562 3e20 6573 7469 6d61 7469 6f6e  /sub> estimation
+0000f780: 2c20 7769 7468 2077 6869 6368 2074 6865  , with which the
+0000f790: 2060 434f 4445 4d4c 6020 7769 6c6c 2063   `CODEML` will c
+0000f7a0: 616c 6375 6c61 7465 2074 6865 202a 4b2a  alculate the *K*
+0000f7b0: 3c73 7562 3e53 3c2f 7375 623e 2066 6f72  <sub>S</sub> for
+0000f7c0: 2065 6163 6820 6765 6e65 2070 6169 7220   each gene pair 
+0000f7d0: 7365 7061 7261 7465 6c79 2062 6173 6564  separately based
+0000f7e0: 206f 6e20 7468 6520 616c 6967 6e6d 656e   on the alignmen
+0000f7f0: 7420 6f66 206f 6e6c 7920 7468 6573 6520  t of only these 
+0000f800: 7477 6f20 6765 6e65 7320 696e 7374 6561  two genes instea
+0000f810: 6420 6f66 2074 6865 2077 686f 6c65 2061  d of the whole a
+0000f820: 6c69 676e 6d65 6e74 206f 6620 7468 6520  lignment of the 
+0000f830: 6661 6d69 6c79 2c20 7375 6368 2074 6861  family, such tha
+0000f840: 7420 6c65 7373 2067 6170 7320 6172 6520  t less gaps are 
+0000f850: 6578 7065 6374 6564 2061 6e64 2074 6875  expected and thu
+0000f860: 7320 7468 6520 616c 6967 6e6d 656e 7420  s the alignment 
+0000f870: 696e 2074 6865 2063 6f6e 7369 6465 7261  in the considera
+0000f880: 7469 6f6e 206f 6620 6043 4f44 454d 4c60  tion of `CODEML`
+0000f890: 2077 696c 6c20 6265 206c 6f6e 6765 7220   will be longer 
+0000f8a0: 2862 6563 6175 7365 2060 434f 4445 4d4c  (because `CODEML
+0000f8b0: 6020 7769 6c6c 2061 7574 6f6d 6174 656c  ` will automatel
+0000f8c0: 7920 736b 6970 2065 7665 7279 2063 6f6c  y skip every col
+0000f8d0: 756d 6e20 7769 7468 2067 6170 2c20 7265  umn with gap, re
+0000f8e0: 6761 7264 6c65 7373 206f 6620 7768 6574  gardless of whet
+0000f8f0: 6865 7220 6974 2773 2061 6e20 6f76 6572  her it's an over
+0000f900: 616c 6c20 6f72 2070 6172 7469 616c 2067  all or partial g
+0000f910: 6170 292c 2077 6974 686f 7574 2077 6869  ap), without whi
+0000f920: 6368 2074 6865 2060 434f 4445 4d4c 6020  ch the `CODEML` 
+0000f930: 7769 6c6c 2063 616c 6375 6c61 7465 2074  will calculate t
+0000f940: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0000f950: 623e 2062 6173 6564 206f 6e20 7468 6520  b> based on the 
+0000f960: 7768 6f6c 6520 616c 6967 6e6d 656e 7420  whole alignment 
+0000f970: 6f66 2074 6865 2066 616d 696c 792c 2077  of the family, w
+0000f980: 6869 6368 206d 6967 6874 2068 6176 6520  hich might have 
+0000f990: 6e6f 202a 4b2a 3c73 7562 3e53 3c2f 7375  no *K*<sub>S</su
+0000f9a0: 623e 2072 6573 756c 7420 6174 2061 6c6c  b> result at all
+0000f9b0: 2069 6620 7468 6520 7374 7269 7070 6564   if the stripped
+0000f9c0: 2061 6c69 676e 6d65 6e74 206c 656e 6774   alignment lengt
+0000f9d0: 6820 2872 656d 6f76 696e 6720 616c 6c20  h (removing all 
+0000f9e0: 6761 702d 636f 6e74 6169 6e69 6e67 2063  gap-containing c
+0000f9f0: 6f6c 756d 6e73 2920 7761 7320 7a65 726f  olumns) was zero
+0000fa00: 2c20 6120 6361 7573 6520 6f66 2064 6966  , a cause of dif
+0000fa10: 6665 7265 6e74 206e 756d 6265 7220 6f66  ferent number of
+0000fa20: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+0000fa30: 2065 7374 696d 6174 6573 2062 6574 7765   estimates betwe
+0000fa40: 656e 2022 7061 6972 7769 7365 206d 6f64  en "pairwise mod
+0000fa50: 6522 2061 6e64 2022 6e6f 6e2d 7061 6972  e" and "non-pair
+0000fa60: 7769 7365 206d 6f64 6522 2e20 4974 2773  wise mode". It's
+0000fa70: 2064 6966 6669 6375 6c74 2074 6f20 7361   difficult to sa
+0000fa80: 7920 7768 6963 6820 6d6f 6465 2069 7320  y which mode is 
+0000fa90: 6d6f 7265 2069 6465 616c 2c20 616c 7468  more ideal, alth
+0000faa0: 6f75 6768 2074 6865 2022 6e6f 6e2d 7061  ough the "non-pa
+0000fab0: 6972 7769 7365 206d 6f64 6522 2028 6465  irwise mode" (de
+0000fac0: 6661 756c 7420 7365 7474 696e 6729 2077  fault setting) w
+0000fad0: 6869 6368 2072 756e 7320 6f6e 2074 6865  hich runs on the
+0000fae0: 2077 686f 6c65 2061 6c69 676e 6d65 6e74   whole alignment
+0000faf0: 2069 6e73 7465 6164 206f 6620 6120 6c6f   instead of a lo
+0000fb00: 6361 6c20 616c 6967 6e6d 656e 742c 206d  cal alignment, m
+0000fb10: 6967 6874 2062 6520 6d6f 7265 2062 696f  ight be more bio
+0000fb20: 6c6f 6769 6361 6c6c 7920 636f 6e73 6572  logically conser
+0000fb30: 7665 6420 696e 2074 6861 7420 6974 2061  ved in that it a
+0000fb40: 7373 7572 6573 2074 6865 2065 766f 6c75  ssures the evolu
+0000fb50: 7469 6f6e 206f 6620 6561 6368 2063 6f6c  tion of each col
+0000fb60: 756d 6e20 746f 2062 6520 7374 6172 7465  umn to be starte
+0000fb70: 6420 6672 6f6d 2074 6865 2072 6f6f 7420  d from the root 
+0000fb80: 6f66 2074 6865 2066 616d 696c 7920 616e  of the family an
+0000fb90: 6420 616c 6c20 7468 6520 6765 6e65 2064  d all the gene d
+0000fba0: 7570 6c69 6361 7465 7320 6172 6520 7461  uplicates are ta
+0000fbb0: 6b65 6e20 696e 746f 2061 6363 6f75 6e74  ken into account
+0000fbc0: 2069 6e20 7468 6520 2a4b 2a3c 7375 623e   in the *K*<sub>
+0000fbd0: 533c 2f73 7562 3e20 6573 7469 6d61 7469  S</sub> estimati
+0000fbe0: 6f6e 2070 726f 6365 7373 2e20 5468 6520  on process. The 
+0000fbf0: 6f70 7469 6f6e 2060 2d2d 7374 7269 705f  option `--strip_
+0000fc00: 6761 7073 6020 6361 6e20 7265 6d6f 7665  gaps` can remove
+0000fc10: 2061 6c6c 2074 6865 2067 6170 2d63 6f6e   all the gap-con
+0000fc20: 7461 696e 696e 6720 636f 6c75 6d6e 732c  taining columns,
+0000fc30: 2077 6974 6820 6f72 2077 6974 686f 7574   with or without
+0000fc40: 2077 6869 6368 2074 6865 2072 6573 756c   which the resul
+0000fc50: 7420 6f66 2022 6e6f 6e2d 7061 6972 7769  t of "non-pairwi
+0000fc60: 7365 206d 6f64 6522 2077 6f6e 2774 2062  se mode" won't b
+0000fc70: 6520 6166 6665 6374 6564 2c20 7768 696c  e affected, whil
+0000fc80: 6520 7769 7468 2077 6869 6368 2074 6865  e with which the
+0000fc90: 2072 6573 756c 7420 6f66 2022 7061 6972   result of "pair
+0000fca0: 7769 7365 206d 6f64 6522 2077 696c 6c20  wise mode" will 
+0000fcb0: 6265 2061 6c74 6572 6564 2e20 5468 6520  be altered. The 
+0000fcc0: 6f70 7469 6f6e 2060 2d2d 616c 6967 6e65  option `--aligne
+0000fcd0: 7260 2061 6e64 2060 2d2d 616c 6e5f 6f70  r` and `--aln_op
+0000fce0: 7469 6f6e 7360 2077 6869 6368 2064 6563  tions` which dec
+0000fcf0: 6964 6520 7768 6963 6820 616c 6967 6e6d  ide which alignm
+0000fd00: 656e 7420 7072 6f67 7261 6d20 746f 2062  ent program to b
+0000fd10: 6520 7573 6564 2061 6e64 2077 6869 6368  e used and which
+0000fd20: 2070 6172 616d 6574 6572 2074 6f20 6265   parameter to be
+0000fd30: 2073 6574 2c20 7769 6c6c 2068 6176 6520   set, will have 
+0000fd40: 696d 7061 6374 206f 6e20 7468 6520 2a4b  impact on the *K
+0000fd50: 2a3c 7375 623e 533c 2f73 7562 3e20 7265  *<sub>S</sub> re
+0000fd60: 7375 6c74 732c 206e 6f74 6564 2074 6861  sults, noted tha
+0000fd70: 7420 7468 6520 6465 6661 756c 7420 7072  t the default pr
+0000fd80: 6f67 7261 6d20 6973 2060 6d61 6666 7460  ogram is `mafft`
+0000fd90: 2061 6e64 2074 6865 2070 6172 616d 6574   and the paramet
+0000fda0: 6572 2069 7320 602d 2d61 7574 6f60 2e20  er is `--auto`. 
+0000fdb0: 5468 6520 6f70 7469 6f6e 2060 2d2d 7472  The option `--tr
+0000fdc0: 6565 5f6d 6574 686f 6460 2061 6e64 2060  ee_method` and `
+0000fdd0: 2d2d 7472 6565 5f6f 7074 696f 6e73 6020  --tree_options` 
+0000fde0: 6465 6369 6465 2077 6869 6368 2067 656e  decide which gen
+0000fdf0: 6520 7472 6565 2069 6e66 6572 656e 6365  e tree inference
+0000fe00: 2070 726f 6772 616d 2074 6f20 6265 2075   program to be u
+0000fe10: 7365 6420 616e 6420 7768 6963 6820 7061  sed and which pa
+0000fe20: 7261 6d65 7465 7220 746f 2062 6520 7365  rameter to be se
+0000fe30: 742c 2077 6f6e 2774 2061 6666 6563 7420  t, won't affect 
+0000fe40: 7468 6520 2a4b 2a3c 7375 623e 533c 2f73  the *K*<sub>S</s
+0000fe50: 7562 3e20 6573 7469 6d61 7469 6f6e 2069  ub> estimation i
+0000fe60: 7473 656c 6620 6275 7420 7468 6520 7265  tself but the re
+0000fe70: 7375 6c74 206f 6620 6465 2d72 6564 756e  sult of de-redun
+0000fe80: 6461 6e63 792c 206e 6f74 6564 2074 6861  dancy, noted tha
+0000fe90: 7420 7765 2069 6d70 6c65 6d65 6e74 6564  t we implemented
+0000fea0: 2061 2062 7569 6c74 2d69 6e20 6765 6e65   a built-in gene
+0000feb0: 2074 7265 6520 696e 6665 7265 6e63 6520   tree inference 
+0000fec0: 6d65 7468 6f64 2062 6173 6564 206f 6e20  method based on 
+0000fed0: 7468 6520 4176 6572 6167 6520 4c69 6e6b  the Average Link
+0000fee0: 6167 6520 436c 7573 7465 7269 6e67 2028  age Clustering (
+0000fef0: 414c 4329 2028 7468 7573 2061 2064 6973  ALC) (thus a dis
+0000ff00: 7461 6e63 652d 6261 7365 6420 7472 6565  tance-based tree
+0000ff10: 2920 7769 7468 2074 6865 2060 2d2d 7472  ) with the `--tr
+0000ff20: 6565 5f6d 6574 686f 6460 2073 6574 2061  ee_method` set a
+0000ff30: 7320 2263 6c75 7374 6572 222e 2054 776f  s "cluster". Two
+0000ff40: 206d 6574 686f 6473 206f 6620 6465 2d72   methods of de-r
+0000ff50: 6564 756e 6461 6e63 7920 6172 6520 696d  edundancy are im
+0000ff60: 706c 656d 656e 7465 6420 696e 2060 7767  plemented in `wg
+0000ff70: 6420 7632 602c 206e 616d 656c 7920 6e6f  d v2`, namely no
+0000ff80: 6465 2d77 6569 6768 7465 6420 616e 6420  de-weighted and 
+0000ff90: 6e6f 6465 2d61 7665 7261 6765 6420 6d65  node-averaged me
+0000ffa0: 7468 6f64 732e 2054 6865 206e 6f64 652d  thods. The node-
+0000ffb0: 7765 6967 6874 6564 206d 6574 686f 6420  weighted method 
+0000ffc0: 6163 6869 6576 6573 2074 6865 2064 652d  achieves the de-
+0000ffd0: 7265 6475 6e64 616e 6379 2076 6961 2077  redundancy via w
+0000ffe0: 6569 6768 696e 6720 7468 6520 2a4b 2a3c  eighing the *K*<
+0000fff0: 7375 623e 533c 2f73 7562 3e20 7661 6c75  sub>S</sub> valu
+00010000: 6520 6173 736f 6369 6174 6564 2077 6974  e associated wit
+00010010: 6820 6561 6368 2067 656e 6520 7061 6972  h each gene pair
+00010020: 2073 7563 6820 7468 6174 2074 6865 2077   such that the w
+00010030: 6569 6768 7420 6f66 2061 2073 696e 676c  eight of a singl
+00010040: 6520 6475 706c 6963 6174 696f 6e20 6576  e duplication ev
+00010050: 656e 7420 7375 6d73 2075 7020 746f 2031  ent sums up to 1
+00010060: 2028 6e6f 7465 6420 7468 6174 2074 6865   (noted that the
+00010070: 206e 756d 6265 7220 6f66 202a 4b2a 3c73   number of *K*<s
+00010080: 7562 3e53 3c2f 7375 623e 2065 7374 696d  ub>S</sub> estim
+00010090: 6174 696f 6e20 7265 6d61 696e 7320 7468  ation remains th
+000100a0: 6520 7361 6d65 2920 7768 696c 6520 7468  e same) while th
+000100b0: 6520 6e6f 6465 2d61 7665 7261 6765 6420  e node-averaged 
+000100c0: 6d65 7468 6f64 2072 6561 6c69 7a65 7320  method realizes 
+000100d0: 7468 6520 6465 2d72 6564 756e 6461 6e63  the de-redundanc
+000100e0: 7920 7669 6120 6361 6c63 756c 6174 696e  y via calculatin
+000100f0: 6720 7065 7220 6765 6e65 2074 7265 6520  g per gene tree 
+00010100: 6e6f 6465 206f 6e65 2061 7665 7261 6765  node one average
+00010110: 6420 2a4b 2a3c 7375 623e 533c 2f73 7562  d *K*<sub>S</sub
+00010120: 3e20 7661 6c75 6520 746f 2072 6570 7265  > value to repre
+00010130: 7365 6e74 2074 6865 2061 6765 206f 6620  sent the age of 
+00010140: 6561 6368 2064 7570 6c69 6361 7469 6f6e  each duplication
+00010150: 2065 7665 6e74 2e20 5468 6520 6f70 7469   event. The opti
+00010160: 6f6e 2060 2d2d 6e6f 6465 5f61 7665 7261  on `--node_avera
+00010170: 6765 6020 6361 6e20 6265 2073 6574 2074  ge` can be set t
+00010180: 6f20 6368 6f6f 7365 2074 6865 206e 6f64  o choose the nod
+00010190: 652d 6176 6572 6167 6564 2077 6179 206f  e-averaged way o
+000101a0: 6620 6465 2d72 6564 756e 6461 6e63 792e  f de-redundancy.
+000101b0: 2044 6966 6665 7265 6e74 206d 6574 686f   Different metho
+000101c0: 6473 206f 6620 6465 2d72 6564 756e 6461  ds of de-redunda
+000101d0: 6e63 7920 7769 6c6c 2068 6176 6520 696d  ncy will have im
+000101e0: 7061 6374 206f 6e20 7468 6520 6465 7465  pact on the dete
+000101f0: 6374 696f 6e20 6f66 2057 4744 2073 6967  ction of WGD sig
+00010200: 6e61 6c73 2c20 7768 6963 6820 6861 7320  nals, which has 
+00010210: 6265 656e 2069 6e76 6573 7469 6761 7465  been investigate
+00010220: 6420 696e 2074 6869 7320 5b6c 6974 6572  d in this [liter
+00010230: 6174 7572 655d 2868 7474 7073 3a2f 2f64  ature](https://d
+00010240: 6f69 2e6f 7267 2f31 302e 3130 3933 2f67  oi.org/10.1093/g
+00010250: 6265 2f65 7679 3230 3029 2e20 5468 6520  be/evy200). The 
+00010260: 7061 7261 6c6c 656c 2074 6872 6561 6473  parallel threads
+00010270: 2068 6572 6520 6172 6520 746f 2070 6172   here are to par
+00010280: 616c 6c65 6c69 7a65 2074 6865 2061 6e61  allelize the ana
+00010290: 6c79 7369 7320 666f 7220 6561 6368 2067  lysis for each g
+000102a0: 656e 6520 6661 6d69 6c79 2061 6e64 2074  ene family and t
+000102b0: 6875 7320 7375 6767 6573 7465 6420 746f  hus suggested to
+000102c0: 2062 6520 7365 7420 6173 206d 7563 6820   be set as much 
+000102d0: 6173 2074 6865 206e 756d 6265 7220 6f66  as the number of
+000102e0: 2067 656e 6520 6661 6d69 6c69 6573 2e0a   gene families..
+000102f0: 0a2a 2a41 2073 7567 6765 7374 6564 2073  .**A suggested s
+00010300: 7461 7274 696e 6720 7275 6e20 6361 6e20  tarting run can 
+00010310: 7573 6520 636f 6d6d 616e 6420 6173 2062  use command as b
+00010320: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
+00010330: 6b73 6420 6661 6d69 6c69 6573 2073 6571  ksd families seq
+00010340: 7565 6e63 650a 6060 600a 0a2a 2a54 6865  uence.```..**The
+00010350: 2063 6f6e 7374 7275 6374 696f 6e20 6f66   construction of
+00010360: 206f 7274 686f 6c6f 676f 7573 202a 4b2a   orthologous *K*
+00010370: 3c73 7562 3e53 3c2f 7375 623e 2061 6765  <sub>S</sub> age
+00010380: 2064 6973 7472 6962 7574 696f 6e2a 2a0a   distribution**.
+00010390: 6060 600a 7767 6420 6b73 6420 6661 6d69  ```.wgd ksd fami
+000103a0: 6c69 6573 2073 6571 7565 6e63 6531 2073  lies sequence1 s
+000103b0: 6571 7565 6e63 6532 2073 6571 7565 6e63  equence2 sequenc
+000103c0: 6533 2028 2d2d 7265 7765 6967 6874 290a  e3 (--reweight).
+000103d0: 6060 600a 0a46 726f 6d20 7061 7261 6c6f  ```..From paralo
+000103e0: 676f 7573 2074 6f20 6f72 7468 6f6c 6f67  gous to ortholog
+000103f0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00010400: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00010410: 7469 6f6e 2c20 7573 6572 7320 6f6e 6c79  tion, users only
+00010420: 206e 6565 6420 746f 2070 726f 7669 6465   need to provide
+00010430: 206d 6f72 6520 6364 7320 6669 6c65 732e   more cds files.
+00010440: 204e 6f74 6520 7468 6174 2077 6974 6820   Note that with 
+00010450: 6f72 7468 6f6c 6f67 6f75 7320 6765 6e65  orthologous gene
+00010460: 2066 616d 696c 6965 7320 7468 6520 7765   families the we
+00010470: 6967 6874 696e 6720 6d65 7468 6f64 2063  ighting method c
+00010480: 616e 2062 6520 7365 7420 746f 2062 6520  an be set to be 
+00010490: 6361 6c63 756c 6174 6564 2070 6572 2073  calculated per s
+000104a0: 7065 6369 6573 2070 6169 7220 696e 7374  pecies pair inst
+000104b0: 6561 6420 6f66 2063 6f6e 7369 6465 7269  ead of consideri
+000104c0: 6e67 2074 6865 2077 686f 6c65 2066 616d  ng the whole fam
+000104d0: 696c 7920 6265 6361 7573 6520 7768 656e  ily because when
+000104e0: 2070 6c6f 7474 696e 6720 6f72 7468 6f6c   plotting orthol
+000104f0: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+00010500: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00010510: 6275 7469 6f6e 2062 6574 7765 656e 2074  bution between t
+00010520: 776f 2073 7065 6369 6573 2074 6865 2077  wo species the w
+00010530: 6569 6768 7420 6361 6c63 756c 6174 6564  eight calculated
+00010540: 2066 726f 6d20 7468 6973 2073 7065 6369   from this speci
+00010550: 6669 6320 7370 6563 6965 7320 7061 6972  fic species pair
+00010560: 2073 686f 756c 6420 6265 2063 6f6e 7365   should be conse
+00010570: 7276 6564 2077 6869 6c65 2074 6865 206f  rved while the o
+00010580: 6e65 2063 616c 6375 6c61 7465 6420 6672  ne calculated fr
+00010590: 6f6d 2074 6865 2077 686f 6c65 2066 616d  om the whole fam
+000105a0: 696c 7920 7769 6c6c 2076 6172 7920 7769  ily will vary wi
+000105b0: 7468 2074 6865 206e 756d 6265 7220 6f66  th the number of
+000105c0: 2073 7065 6369 6573 2e20 546f 2069 6e69   species. To ini
+000105d0: 7469 6174 6520 7468 6520 7765 6967 6874  tiate the weight
+000105e0: 696e 6720 7065 7220 7370 6563 6965 7320  ing per species 
+000105f0: 7061 6972 2c20 7468 6520 6f70 7469 6f6e  pair, the option
+00010600: 2060 2d2d 7265 7765 6967 6874 6020 6361   `--reweight` ca
+00010610: 6e20 6265 2073 6574 2e0a 0a2a 2a41 2073  n be set...**A s
+00010620: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+00010630: 6720 7275 6e20 6361 6e20 7573 6520 636f  g run can use co
+00010640: 6d6d 616e 6420 6173 2062 656c 6f77 2a2a  mmand as below**
+00010650: 0a0a 6060 600a 7767 6420 6b73 6420 6661  ..```.wgd ksd fa
+00010660: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
+00010670: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
+00010680: 6e63 6533 0a60 6060 0a0a 2a2a 5468 6520  nce3.```..**The 
+00010690: 636f 6e73 7472 7563 7469 6f6e 206f 6620  construction of 
+000106a0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+000106b0: 6167 6520 6469 7374 7269 6275 7469 6f6e  age distribution
+000106c0: 2077 6974 6820 7261 7465 2063 6f72 7265   with rate corre
+000106d0: 6374 696f 6e2a 2a0a 6060 600a 7767 6420  ction**.```.wgd 
+000106e0: 6b73 6420 6661 6d69 6c69 6573 2073 6571  ksd families seq
+000106f0: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+00010700: 2073 6571 7565 6e63 6533 202d 2d66 6f63   sequence3 --foc
+00010710: 7573 3261 6c6c 2073 6571 7565 6e63 6531  us2all sequence1
+00010720: 202d 7370 2073 7064 6174 6120 2d2d 6578   -sp spdata --ex
+00010730: 7472 6170 6172 616e 6f6d 656b 7320 7061  traparanomeks pa
+00010740: 7261 6e6f 6d65 4b73 6461 7461 2028 2d2d  ranomeKsdata (--
+00010750: 706c 6f74 656c 6d6d 202d 2d70 6c6f 7461  plotelmm --plota
+00010760: 7067 6d6d 202d 2d61 6e63 686f 7270 6f69  pgmm --anchorpoi
+00010770: 6e74 7320 6170 6461 7461 202d 2d72 6577  nts apdata --rew
+00010780: 6569 6768 7420 2d2d 6f6e 6c79 726f 6f74  eight --onlyroot
+00010790: 6f75 7429 0a60 6060 0a0a 496e 7370 6972  out).```..Inspir
+000107a0: 6564 2062 7920 7468 6520 7261 7465 2063  ed by the rate c
+000107b0: 6f72 7265 6374 696f 6e20 616c 676f 7269  orrection algori
+000107c0: 7468 6d20 696e 205b 6b73 7261 7465 735d  thm in [ksrates]
+000107d0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+000107e0: 636f 6d2f 5649 422d 5053 422f 6b73 7261  com/VIB-PSB/ksra
+000107f0: 7465 7329 2c20 7765 2069 6d70 6c65 6d65  tes), we impleme
+00010800: 6e74 6564 2074 6865 2072 6174 6520 636f  nted the rate co
+00010810: 7272 6563 7469 6f6e 2061 6e61 6c79 7369  rrection analysi
+00010820: 7320 616c 736f 2069 6e20 6077 6764 2076  s also in `wgd v
+00010830: 3260 2c20 7768 6963 6820 6973 206d 6f73  2`, which is mos
+00010840: 746c 7920 7468 6520 7361 6d65 2061 7320  tly the same as 
+00010850: 5b6b 7372 6174 6573 5d28 6874 7470 733a  [ksrates](https:
+00010860: 2f2f 6769 7468 7562 2e63 6f6d 2f56 4942  //github.com/VIB
+00010870: 2d50 5342 2f6b 7372 6174 6573 2920 6275  -PSB/ksrates) bu
+00010880: 7420 6469 6666 6572 7320 696e 2074 6865  t differs in the
+00010890: 2063 616c 6375 6c61 7469 6f6e 206f 6620   calculation of 
+000108a0: 7468 6520 7374 616e 6461 7264 2064 6576  the standard dev
+000108b0: 6961 7469 6f6e 206f 6620 7265 7363 616c  iation of rescal
+000108c0: 6564 202a 4b2a 3c73 7562 3e53 3c2f 7375  ed *K*<sub>S</su
+000108d0: 623e 2061 6765 732e 2054 6f20 7065 7266  b> ages. To perf
+000108e0: 6f72 6d20 7468 6520 7261 7465 2063 6f72  orm the rate cor
+000108f0: 7265 6374 696f 6e20 616e 616c 7973 6973  rection analysis
+00010900: 2c20 7573 6572 7320 6361 6e20 7573 6520  , users can use 
+00010910: 626f 7468 2074 6865 2060 7767 6420 6b73  both the `wgd ks
+00010920: 6460 2061 6e64 2060 7767 6420 7669 7a60  d` and `wgd viz`
+00010930: 2070 726f 6772 616d 2e20 466f 7220 7468   program. For th
+00010940: 6520 6077 6764 206b 7364 6020 7072 6f67  e `wgd ksd` prog
+00010950: 7261 6d2c 2075 7365 7273 206e 6565 6420  ram, users need 
+00010960: 746f 2070 726f 7669 6465 2061 2073 7065  to provide a spe
+00010970: 6369 6573 2074 7265 6520 7669 6120 7468  cies tree via th
+00010980: 6520 6f70 7469 6f6e 2060 2d2d 7370 6563  e option `--spec
+00010990: 6965 7374 7265 6560 206f 6e20 7768 6963  iestree` on whic
+000109a0: 6820 7261 7465 2063 6f72 7265 6374 696f  h rate correctio
+000109b0: 6e20 6361 6e20 6265 2063 6f6e 6475 6374  n can be conduct
+000109c0: 6564 2e20 4e6f 7465 2074 6861 7420 756e  ed. Note that un
+000109d0: 6e65 6365 7373 6172 7920 6272 6163 6b65  necessary bracke
+000109e0: 7473 206d 6967 6874 206c 6561 6420 746f  ts might lead to
+000109f0: 2075 6e65 7870 6563 7465 6420 6572 726f   unexpected erro
+00010a00: 7273 2c20 666f 7220 696e 7374 616e 6365  rs, for instance
+00010a10: 2061 2074 7265 6520 6028 412c 2842 2c43   a tree `(A,(B,C
+00010a20: 2929 3b60 2073 686f 756c 6420 6e6f 7420  ));` should not 
+00010a30: 6265 2072 6570 7265 7365 6e74 6564 2061  be represented a
+00010a40: 7320 6028 412c 2828 422c 4329 2929 3b60  s `(A,((B,C)));`
+00010a50: 2e20 5468 6520 7365 7420 6f66 2073 7065  . The set of spe
+00010a60: 6369 6573 2070 6169 7273 2074 6f20 6265  cies pairs to be
+00010a70: 2073 686f 776e 2069 7320 666c 6578 6962   shown is flexib
+00010a80: 6c65 2074 6861 7420 7468 6520 6d6f 7374  le that the most
+00010a90: 2063 6f6e 7665 6e69 656e 7420 6f70 7469   convenient opti
+00010aa0: 6f6e 2069 7320 602d 2d66 6f63 7573 3261  on is `--focus2a
+00010ab0: 6c6c 6020 7768 6963 6820 7369 6d70 6c79  ll` which simply
+00010ac0: 2073 686f 7773 2061 6c6c 2074 6865 2070   shows all the p
+00010ad0: 6f73 7369 626c 6520 666f 6361 6c2d 7369  ossible focal-si
+00010ae0: 7374 6572 2073 7065 6369 6573 2070 6169  ster species pai
+00010af0: 7273 2c20 6f72 2075 7365 7273 2063 616e  rs, or users can
+00010b00: 206d 616e 7561 6c6c 7920 7365 7420 7468   manually set th
+00010b10: 6520 7370 6563 6965 7320 7061 6972 7320  e species pairs 
+00010b20: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+00010b30: 2d2d 7370 6169 7260 2e20 4e6f 7465 2074  --spair`. Note t
+00010b40: 6861 7420 6966 2074 6865 2073 7065 6369  hat if the speci
+00010b50: 6573 2070 6169 7273 2077 6572 6520 6d61  es pairs were ma
+00010b60: 6e75 616c 6c79 2073 6574 2c20 6974 2077  nually set, it w
+00010b70: 6f75 6c64 2062 6520 6e65 6564 6564 2074  ould be needed t
+00010b80: 6f20 636f 2d73 6574 2074 6865 206f 7074  o co-set the opt
+00010b90: 696f 6e20 602d 2d63 6c61 7373 6963 602e  ion `--classic`.
+00010ba0: 2054 6865 206f 7074 696f 6e20 602d 2d6f   The option `--o
+00010bb0: 6e6c 7972 6f6f 746f 7574 6020 6361 6e20  nlyrootout` can 
+00010bc0: 6265 2073 6574 2074 6f20 6f6e 6c79 2063  be set to only c
+00010bd0: 6f6e 7369 6465 7220 6f75 7467 726f 7570  onsider outgroup
+00010be0: 2061 7420 7468 6520 726f 6f74 2c20 696e   at the root, in
+00010bf0: 7374 6561 6420 6f66 2061 6c6c 2074 6865  stead of all the
+00010c00: 2070 6f73 7369 626c 6520 6f75 7467 726f   possible outgro
+00010c10: 7570 7320 7065 7220 666f 6361 6c2d 7369  ups per focal-si
+00010c20: 7374 6572 2073 7065 6369 6573 2070 6169  ster species pai
+00010c30: 722c 2077 6869 6368 2068 6173 2069 6d70  r, which has imp
+00010c40: 6163 7420 6f6e 2074 6865 2066 696e 616c  act on the final
+00010c50: 2063 6f72 7265 6374 6564 202a 4b2a 3c73   corrected *K*<s
+00010c60: 7562 3e53 3c2f 7375 623e 2061 6765 732e  ub>S</sub> ages.
+00010c70: 2057 6520 7375 6767 6573 7420 6f66 2075   We suggest of u
+00010c80: 7369 6e67 2061 6c6c 2074 6865 2070 6f73  sing all the pos
+00010c90: 7369 626c 6520 6f75 7467 726f 7570 7320  sible outgroups 
+00010ca0: 7065 7220 666f 6361 6c2d 7369 7374 6572  per focal-sister
+00010cb0: 2073 7065 6369 6573 2070 6169 7220 6173   species pair as
+00010cc0: 2066 6f72 2061 206c 6573 7320 6269 6173   for a less bias
+00010cd0: 6564 2072 6573 756c 742e 2054 6865 2070  ed result. The p
+00010ce0: 6172 616e 6f6d 6520 2a4b 2a3c 7375 623e  aranome *K*<sub>
+00010cf0: 533c 2f73 7562 3e20 6461 7461 2073 686f  S</sub> data sho
+00010d00: 756c 6420 6265 2070 726f 7669 6465 6420  uld be provided 
+00010d10: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+00010d20: 2d2d 6578 7472 6170 6172 616e 6f6d 656b  --extraparanomek
+00010d30: 7360 2e20 0a0a 536f 6d65 206f 7468 6572  s`. ..Some other
+00010d40: 206f 7074 696f 6e73 2077 6869 6368 2068   options which h
+00010d50: 6176 6520 6e6f 2069 6d70 6163 7420 6f6e  ave no impact on
+00010d60: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+00010d70: 7469 6f6e 2062 7574 2061 6464 206d 6f72  tion but add mor
+00010d80: 6520 6c61 7965 7273 206f 7220 6368 616e  e layers or chan
+00010d90: 6765 2074 6865 2061 7070 6561 7261 6e63  ge the appearanc
+00010da0: 6520 6f6e 2074 6865 206d 6978 6564 202a  e on the mixed *
+00010db0: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00010dc0: 6765 2064 6973 7472 6962 7574 696f 6e2c  ge distribution,
+00010dd0: 2069 6e63 6c75 6465 2060 2d2d 706c 6f74   include `--plot
+00010de0: 6170 676d 6d60 2061 6e64 2060 2d2d 706c  apgmm` and `--pl
+00010df0: 6f74 656c 6d6d 6020 6574 632e 2054 6865  otelmm` etc. The
+00010e00: 206f 7074 696f 6e20 602d 2d70 6c6f 7461   option `--plota
+00010e10: 7067 6d6d 6020 6361 6e20 6265 2073 6574  pgmm` can be set
+00010e20: 2074 6f20 6361 6c6c 2074 6865 2047 4d4d   to call the GMM
+00010e30: 2061 6e61 6c79 7369 7320 7570 6f6e 2074   analysis upon t
+00010e40: 6865 2061 6e63 686f 7220 7061 6972 202a  he anchor pair *
+00010e50: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2061  K*<sub>S</sub> a
+00010e60: 6765 2064 6973 7472 6962 7574 696f 6e20  ge distribution 
+00010e70: 616e 6420 706c 6f74 2074 6865 2063 6c75  and plot the clu
+00010e80: 7374 6572 696e 6720 7265 7375 6c74 2075  stering result u
+00010e90: 706f 6e20 7468 6520 6d69 7865 6420 2a4b  pon the mixed *K
+00010ea0: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+00010eb0: 6520 6469 7374 7269 6275 7469 6f6e 2c20  e distribution, 
+00010ec0: 7768 6963 6820 6861 7320 746f 2062 6520  which has to be 
+00010ed0: 636f 2d73 6574 2077 6974 6820 7468 6520  co-set with the 
+00010ee0: 6f70 7469 6f6e 2060 2d2d 616e 6368 6f72  option `--anchor
+00010ef0: 706f 696e 7473 6020 7072 6f76 6964 696e  points` providin
+00010f00: 6720 7468 6520 616e 6368 6f72 2070 6169  g the anchor pai
+00010f10: 7273 2069 6e66 6f72 6d61 7469 6f6e 2e20  rs information. 
+00010f20: 5468 6520 6f70 7469 6f6e 2060 2d2d 706c  The option `--pl
+00010f30: 6f74 656c 6d6d 6020 6361 6e20 6265 2073  otelmm` can be s
+00010f40: 6574 2074 6f20 6361 6c6c 2074 6865 2045  et to call the E
+00010f50: 4c4d 4d20 616e 616c 7973 6973 2075 706f  LMM analysis upo
+00010f60: 6e20 7468 6520 7768 6f6c 6520 7061 7261  n the whole para
+00010f70: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+00010f80: 7375 623e 2061 6765 2064 6973 7472 6962  sub> age distrib
+00010f90: 7574 696f 6e2e 202a 2a4e 6f74 6520 7468  ution. **Note th
+00010fa0: 6174 2074 6865 2073 7065 6369 6573 206e  at the species n
+00010fb0: 616d 6573 2070 7265 7365 6e74 2069 6e20  ames present in 
+00010fc0: 7468 6520 7370 6563 6965 7320 7472 6565  the species tree
+00010fd0: 2066 696c 6520 7368 6f75 6c64 206d 6174   file should mat
+00010fe0: 6368 2074 6865 206e 616d 6573 206f 6620  ch the names of 
+00010ff0: 7468 6520 636f 7272 6573 706f 6e64 696e  the correspondin
+00011000: 6720 7365 7175 656e 6365 2066 696c 6573  g sequence files
+00011010: 2a2a 2046 6f72 2069 6e73 7461 6e63 652c  ** For instance,
+00011020: 2067 6976 656e 2074 6865 2063 6473 2066   given the cds f
+00011030: 696c 6520 6e61 6d65 7320 2741 2e63 6473  ile names 'A.cds
+00011040: 272c 2742 2e63 6473 272c 2743 2e63 6473  ','B.cds','C.cds
+00011050: 272c 2074 6865 2073 7065 6369 6573 2074  ', the species t
+00011060: 7265 6520 636f 756c 6420 6265 2027 2841  ree could be '(A
+00011070: 2e63 6473 2c28 422e 6364 732c 432e 6364  .cds,(B.cds,C.cd
+00011080: 7329 293b 2720 7261 7468 6572 2074 6861  s));' rather tha
+00011090: 6e20 2728 412c 2842 2c43 2929 3b27 2e20  n '(A,(B,C));'. 
+000110a0: 5468 6572 6520 6973 206e 6f20 7265 7175  There is no requ
+000110b0: 6972 656d 656e 7420 666f 7220 7468 6520  irement for the 
+000110c0: 6e61 6d65 206f 6620 7468 6520 7061 7261  name of the para
+000110d0: 6e6f 6d65 202a 4b2a 3c73 7562 3e53 3c2f  nome *K*<sub>S</
+000110e0: 7375 623e 2064 6174 6166 696c 6520 7768  sub> datafile wh
+000110f0: 6963 6820 6361 6e20 6265 206e 616d 6564  ich can be named
+00011100: 2069 6e20 7768 6174 6576 6572 206d 616e   in whatever man
+00011110: 6e65 7220 7573 6572 7320 7072 6566 6572  ner users prefer
+00011120: 2e20 5468 6520 2747 4d4d 2720 6973 2074  . The 'GMM' is t
+00011130: 6865 2061 6262 7265 7669 6174 696f 6e20  he abbreviation 
+00011140: 6f66 2047 6175 7373 6961 6e20 4d69 7874  of Gaussian Mixt
+00011150: 7572 6520 4d6f 6465 6c69 6e67 2077 6869  ure Modeling whi
+00011160: 6c65 2074 6865 2027 454c 4d4d 2720 7265  le the 'ELMM' re
+00011170: 6665 7273 2074 6f20 4578 706f 6e65 6e74  fers to Exponent
+00011180: 6961 6c2d 4c6f 676e 6f72 6d61 6c20 4d69  ial-Lognormal Mi
+00011190: 7874 7572 6520 4d6f 6465 6c69 6e67 2061  xture Modeling a
+000111a0: 7320 5b6b 7372 6174 6573 5d28 6874 7470  s [ksrates](http
+000111b0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f56  s://github.com/V
+000111c0: 4942 2d50 5342 2f6b 7372 6174 6573 2920  IB-PSB/ksrates) 
+000111d0: 696e 7465 7270 7265 7473 2e0a 0a54 6865  interprets...The
+000111e0: 7265 2061 7265 2032 3120 636f 6c75 6d6e  re are 21 column
+000111f0: 7320 696e 2074 6865 2072 6573 756c 7420  s in the result 
+00011200: 602e 6b73 2e74 7376 6020 6669 6c65 2062  `.ks.tsv` file b
+00011210: 6573 6964 6573 2074 6865 2069 6e64 6578  esides the index
+00011220: 2063 6f6c 756d 6e73 2060 7061 6972 6020   columns `pair` 
+00011230: 6173 2074 6865 2075 6e69 7175 6520 6964  as the unique id
+00011240: 656e 7469 6669 6572 2066 6f72 2065 6163  entifier for eac
+00011250: 6820 6765 6e65 2070 6169 722e 2054 6865  h gene pair. The
+00011260: 2060 4e60 2c20 6053 602c 2060 644e 602c   `N`, `S`, `dN`,
+00011270: 2060 644e 2f64 5360 2c20 6064 5360 2c20   `dN/dS`, `dS`, 
+00011280: 606c 6020 616e 6420 6074 6020 6172 6520  `l` and `t` are 
+00011290: 6672 6f6d 2074 6865 2063 6f64 656d 6c20  from the codeml 
+000112a0: 7265 7375 6c74 732c 2072 6570 7265 7365  results, represe
+000112b0: 6e74 696e 6720 7468 6520 4e20 6573 7469  nting the N esti
+000112c0: 6d61 7465 2c20 7468 6520 5320 6573 7469  mate, the S esti
+000112d0: 6d61 7465 2c20 7468 6520 644e 2065 7374  mate, the dN est
+000112e0: 696d 6174 652c 2074 6865 2064 4e2f 6453  imate, the dN/dS
+000112f0: 2028 6f6d 6567 6129 2065 7374 696d 6174   (omega) estimat
+00011300: 652c 2074 6865 2064 5320 6573 7469 6d61  e, the dS estima
+00011310: 7465 2c20 7468 6520 6c6f 672d 6c69 6b65  te, the log-like
+00011320: 6c69 686f 6f64 2061 6e64 2074 6865 2074  lihood and the t
+00011330: 2065 7374 696d 6174 652c 2072 6573 7065   estimate, respe
+00011340: 6374 6976 656c 792e 2054 6865 2060 616c  ctively. The `al
+00011350: 6967 6e6d 656e 7463 6f76 6572 6167 6560  ignmentcoverage`
+00011360: 2c20 6061 6c69 676e 6d65 6e74 6964 656e  , `alignmentiden
+00011370: 7469 7479 6020 616e 6420 6061 6c69 676e  tity` and `align
+00011380: 6d65 6e74 6c65 6e67 7468 6020 6172 6520  mentlength` are 
+00011390: 7468 6520 696e 666f 726d 6174 696f 6e20  the information 
+000113a0: 7065 7274 6169 6e69 6e67 2074 6f20 7468  pertaining to th
+000113b0: 6520 616c 6967 6e6d 656e 7420 666f 7220  e alignment for 
+000113c0: 6561 6368 2066 616d 696c 792c 2072 6570  each family, rep
+000113d0: 7265 7365 6e74 696e 6720 7468 6520 7261  resenting the ra
+000113e0: 7469 6f20 6f66 2074 6865 2073 7472 6970  tio of the strip
+000113f0: 7065 6420 616c 6967 6e6d 656e 7420 6c65  ped alignment le
+00011400: 6e67 7468 2063 6f6d 7061 7265 6420 746f  ngth compared to
+00011410: 2074 6865 2066 756c 6c20 616c 6967 6e6d   the full alignm
+00011420: 656e 7420 6c65 6e67 7468 2c20 7468 6520  ent length, the 
+00011430: 7261 7469 6f20 6f66 2063 6f6c 756d 6e73  ratio of columns
+00011440: 2077 6974 6820 6964 656e 7469 6361 6c20   with identical 
+00011450: 6e75 636c 656f 7469 6465 7320 636f 6d70  nucleotides comp
+00011460: 6172 6564 2074 6f20 7468 6520 6f76 6572  ared to the over
+00011470: 616c 6c20 636f 6c75 6d6e 7320 6f66 2074  all columns of t
+00011480: 6865 2073 7472 6970 7065 6420 616c 6967  he stripped alig
+00011490: 6e6d 656e 742c 2061 6e64 2074 6865 206c  nment, and the l
+000114a0: 656e 6774 6820 6f66 2074 6865 2066 756c  ength of the ful
+000114b0: 6c20 616c 6967 6e6d 656e 742c 2072 6573  l alignment, res
+000114c0: 7065 6374 6976 656c 792e 0a0a 2323 2320  pectively...### 
+000114d0: 7767 6420 7379 6e0a 0a2a 2a54 6865 2069  wgd syn..**The i
+000114e0: 6e74 7261 2d73 7065 6369 6669 6320 7379  ntra-specific sy
+000114f0: 6e74 656e 7920 696e 6665 7265 6e63 652a  nteny inference*
+00011500: 2a0a 6060 600a 7767 6420 7379 6e20 6661  *.```.wgd syn fa
+00011510: 6d69 6c69 6573 2067 6666 2028 2d2d 6b73  milies gff (--ks
+00011520: 5f64 6973 7472 6962 7574 696f 6e20 6b73  _distribution ks
+00011530: 6461 7461 202d 6620 6765 6e65 202d 6120  data -f gene -a 
+00011540: 4944 202d 2d6d 696e 6c65 6e20 2d31 202d  ID --minlen -1 -
+00011550: 2d6d 696e 7365 676c 656e 2031 3030 3030  -minseglen 10000
+00011560: 202d 2d6d 696e 6765 6e65 6e75 6d20 3330   --mingenenum 30
+00011570: 290a 6060 600a 0a54 6865 2070 726f 6772  ).```..The progr
+00011580: 616d 2060 7767 6420 7379 6e60 2069 7320  am `wgd syn` is 
+00011590: 6d61 696e 6c79 2064 6561 6c69 6e67 2077  mainly dealing w
+000115a0: 6974 6820 636f 6c6c 696e 6561 7269 7479  ith collinearity
+000115b0: 206f 7220 7379 6e74 656e 7920 2862 6f74   or synteny (bot
+000115c0: 6820 7265 6665 7272 6564 2074 6f20 6173  h referred to as
+000115d0: 2073 796e 7465 6e79 2068 6572 6561 6674   synteny hereaft
+000115e0: 6572 2920 616e 616c 7973 6973 2e20 5477  er) analysis. Tw
+000115f0: 6f20 696e 7075 7420 6669 6c65 7320 6172  o input files ar
+00011600: 6520 6573 7365 6e74 6961 6c2c 2074 6865  e essential, the
+00011610: 2067 656e 6520 6661 6d69 6c79 2066 696c   gene family fil
+00011620: 6520 616e 6420 7468 6520 6766 6633 2066  e and the gff3 f
+00011630: 696c 652e 2054 6865 2067 656e 6520 6661  ile. The gene fa
+00011640: 6d69 6c79 2066 696c 6520 6973 2069 6e20  mily file is in 
+00011650: 7468 6520 666f 726d 6174 2061 7320 604f  the format as `O
+00011660: 7274 686f 4669 6e64 6572 602e 2054 6865  rthoFinder`. The
+00011670: 2073 6f66 7477 6172 6520 6069 2d61 6468   software `i-adh
+00011680: 6f72 6560 2069 7320 6120 7072 6572 6571  ore` is a prereq
+00011690: 7569 7369 7465 2e20 5769 7468 2064 6566  uisite. With def
+000116a0: 6175 6c74 2070 6172 616d 6574 6572 732c  ault parameters,
+000116b0: 2074 6865 2070 726f 6772 616d 2062 6173   the program bas
+000116c0: 6963 616c 6c79 2063 6f6e 6475 6374 7320  ically conducts 
+000116d0: 3129 2066 696c 7465 7269 6e67 2067 656e  1) filtering gen
+000116e0: 6520 6661 6d69 6c69 6573 2062 6173 6564  e families based
+000116f0: 206f 6e20 6d61 7869 6d75 6d20 6661 6d69   on maximum fami
+00011700: 6c79 2073 697a 6520 3229 2072 6574 7269  ly size 2) retri
+00011710: 6576 696e 6720 6765 6e65 2070 6f73 6974  eving gene posit
+00011720: 696f 6e20 616e 6420 7363 6166 666f 6c64  ion and scaffold
+00011730: 2069 6e66 6f72 6d61 7469 6f6e 2066 726f   information fro
+00011740: 6d20 6766 6633 2066 696c 6520 3329 2070  m gff3 file 3) p
+00011750: 726f 6475 6369 6e67 2074 6865 2063 6f6e  roducing the con
+00011760: 6669 6775 7261 7469 6f6e 2066 696c 6520  figuration file 
+00011770: 616e 6420 6173 736f 6369 6174 6564 2064  and associated d
+00011780: 6174 6166 696c 6573 2066 6f72 2060 692d  atafiles for `i-
+00011790: 6164 686f 7265 6020 3429 2063 616c 6c69  adhore` 4) calli
+000117a0: 6e67 2060 692d 6164 686f 7265 6020 6769  ng `i-adhore` gi
+000117b0: 7665 6e20 7468 6520 7061 7261 6d65 7465  ven the paramete
+000117c0: 7273 2073 6574 2074 6f20 696e 6665 7220  rs set to infer 
+000117d0: 7379 6e74 656e 7920 3529 2076 6973 7561  synteny 5) visua
+000117e0: 6c69 7a69 6e67 2074 6865 2073 796e 7465  lizing the synte
+000117f0: 6e79 2069 6e20 2264 6f74 706c 6f74 2220  ny in "dotplot" 
+00011800: 696e 2074 6865 2075 6e69 7420 6f66 2067  in the unit of g
+00011810: 656e 6573 2061 6e64 2062 6173 6573 2c20  enes and bases, 
+00011820: 696e 2022 5379 6e64 6570 7468 2220 706c  in "Syndepth" pl
+00011830: 6f74 2073 686f 7769 6e67 2074 6865 2064  ot showing the d
+00011840: 6973 7472 6962 7574 696f 6e20 6f66 2064  istribution of d
+00011850: 6966 6665 7265 6e74 2063 6174 6567 6f72  ifferent categor
+00011860: 6965 7320 6f66 2063 6f6c 6c69 6e65 6172  ies of collinear
+00011870: 6974 7920 7261 7469 6f73 2077 6974 6869  ity ratios withi
+00011880: 6e20 616e 6420 6265 7477 6565 6e20 7370  n and between sp
+00011890: 6563 6965 732c 2069 6e20 2264 7570 5374  ecies, in "dupSt
+000118a0: 6163 6b22 2070 6c6f 7420 7368 6f77 696e  ack" plot showin
+000118b0: 6720 6d75 6c74 6970 6c69 636f 6e73 2077  g multiplicons w
+000118c0: 6974 6820 6469 6666 6572 656e 7420 6d75  ith different mu
+000118d0: 6c74 6970 6c69 6361 7469 6f6e 206c 6576  ltiplication lev
+000118e0: 656c 732e 2036 2920 6966 2077 6974 6820  els. 6) if with 
+000118f0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00011900: 6461 7461 2c20 6120 222a 4b2a 3c73 7562  data, a "*K*<sub
+00011910: 3e53 3c2f 7375 623e 2064 6f74 706c 6f74  >S</sub> dotplot
+00011920: 2220 7769 7468 2064 6f74 7320 616e 6e6f  " with dots anno
+00011930: 7461 7465 6420 696e 202a 4b2a 3c73 7562  tated in *K*<sub
+00011940: 3e53 3c2f 7375 623e 2076 616c 7565 7320  >S</sub> values 
+00011950: 616e 6420 6120 2a4b 2a3c 7375 623e 533c  and a *K*<sub>S<
+00011960: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00011970: 6f6e 2077 6974 6820 616e 6368 6f72 2070  on with anchor p
+00011980: 6169 7273 2064 656e 6f74 6564 2077 696c  airs denoted wil
+00011990: 6c20 6265 2070 726f 6475 6365 642e 2054  l be produced. T
+000119a0: 6865 2067 656e 6520 696e 666f 726d 6174  he gene informat
+000119b0: 696f 6e20 696e 2074 6865 2067 656e 6520  ion in the gene 
+000119c0: 6661 6d69 6c79 2066 696c 6520 616e 6420  family file and 
+000119d0: 6766 6633 2066 696c 6520 7368 6f75 6c64  gff3 file should
+000119e0: 2062 6520 6d61 7463 6865 6420 7768 6963   be matched whic
+000119f0: 6820 7265 7175 6972 6573 2075 7365 7273  h requires users
+00011a00: 2074 6f20 7365 7420 7072 6f70 6572 2060   to set proper `
+00011a10: 2d2d 6665 6174 7572 6560 2061 6e64 2060  --feature` and `
+00011a20: 2d2d 6174 7472 6962 7574 6560 2e20 5468  --attribute`. Th
+00011a30: 6520 6d61 7869 6d75 6d20 6661 6d69 6c79  e maximum family
+00011a40: 2073 697a 6520 746f 2062 6520 696e 636c   size to be incl
+00011a50: 7564 6564 2063 616e 2062 6520 7365 7420  uded can be set 
+00011a60: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+00011a70: 2d2d 6d61 7873 697a 6560 2c20 6e6f 7465  --maxsize`, note
+00011a80: 6420 7468 6174 2074 6869 7320 6669 6c74  d that this filt
+00011a90: 6572 696e 6720 6973 206d 6169 6e6c 7920  ering is mainly 
+00011aa0: 746f 2064 726f 7020 7468 6f73 6520 6875  to drop those hu
+00011ab0: 6765 2074 616e 6465 6d20 6475 706c 6963  ge tandem duplic
+00011ac0: 6174 6573 2066 616d 696c 7920 616e 6420  ates family and 
+00011ad0: 7472 616e 7370 6f73 6162 6c65 2065 6c65  transposable ele
+00011ae0: 6d65 6e74 7320 2854 4573 2920 6661 6d69  ments (TEs) fami
+00011af0: 6c79 2c20 616e 6420 6e6f 7420 6d61 6e64  ly, and not mand
+00011b00: 6174 6f72 792e 2055 7365 7273 2063 616e  atory. Users can
+00011b10: 2066 696c 7465 7220 7468 6f73 6520 6672   filter those fr
+00011b20: 6167 6d65 6e74 6172 7920 7363 6166 666f  agmentary scaffo
+00011b30: 6c64 7320 7669 6120 7468 6520 6f70 7469  lds via the opti
+00011b40: 6f6e 2060 2d2d 6d69 6e6c 656e 602e 2054  on `--minlen`. T
+00011b50: 6865 206d 696e 696d 756d 206c 656e 6774  he minimum lengt
+00011b60: 6820 616e 6420 6e75 6d62 6572 206f 6620  h and number of 
+00011b70: 6765 6e65 7320 666f 7220 6120 7365 676d  genes for a segm
+00011b80: 656e 7420 746f 2062 6520 636f 6e73 6964  ent to be consid
+00011b90: 6572 6564 2063 616e 2062 6520 7365 7420  ered can be set 
+00011ba0: 7669 6120 7468 6520 6f70 7469 6f6e 2060  via the option `
+00011bb0: 2d2d 6d69 6e73 6567 6c65 6e60 2061 6e64  --minseglen` and
+00011bc0: 2060 2d2d 6d69 6e67 656e 656e 756d 602e   `--mingenenum`.
+00011bd0: 2052 6564 756e 6461 6e74 206d 756c 7469   Redundant multi
+00011be0: 706c 6963 6f6e 7320 6361 6e20 6265 206b  plicons can be k
+00011bf0: 6570 7420 6279 2073 6574 2074 6865 2066  ept by set the f
+00011c00: 6c61 6720 6f70 7469 6f6e 2060 2d2d 6b65  lag option `--ke
+00011c10: 6570 7265 6475 6e60 2e0a 0a2a 2a41 2073  epredun`...**A s
+00011c20: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+00011c30: 6720 7275 6e20 6361 6e20 7573 6520 636f  g run can use co
+00011c40: 6d6d 616e 6420 7369 6d70 6c79 2061 7320  mmand simply as 
+00011c50: 6265 6c6f 772a 2a0a 0a60 6060 0a77 6764  below**..```.wgd
+00011c60: 2073 796e 2066 616d 696c 6965 7320 6766   syn families gf
+00011c70: 660a 6060 600a 0a2a 2a54 6865 2069 6e74  f.```..**The int
+00011c80: 6572 2d73 7065 6369 6669 6320 7379 6e74  er-specific synt
+00011c90: 656e 7920 696e 6665 7265 6e63 652a 2a0a  eny inference**.
+00011ca0: 6060 600a 7767 6420 7379 6e20 6661 6d69  ```.wgd syn fami
+00011cb0: 6c69 6573 2067 6666 3120 6766 6632 2028  lies gff1 gff2 (
+00011cc0: 2d2d 6164 6469 7469 6f6e 616c 6766 6669  --additionalgffi
+00011cd0: 6e66 6f20 226d 524e 413b 4e61 6d65 2220  nfo "mRNA;Name" 
+00011ce0: 2d2d 6164 6469 7469 6f6e 616c 6766 6669  --additionalgffi
+00011cf0: 6e66 6f20 2267 656e 653b 4944 2229 0a60  nfo "gene;ID").`
+00011d00: 6060 0a0a 466f 7220 6d75 6c74 692d 7370  ``..For multi-sp
+00011d10: 6563 6965 7320 7379 6e74 656e 7920 696e  ecies synteny in
+00011d20: 6665 7265 6e63 652c 2069 6620 7573 6572  ference, if user
+00011d30: 7320 6861 7665 2067 6666 3320 6669 6c65  s have gff3 file
+00011d40: 7320 7768 6963 6820 6861 7665 2064 6966  s which have dif
+00011d50: 6665 7265 6e74 2066 6561 7475 7265 7320  ferent features 
+00011d60: 6f72 2061 7474 7269 6275 7465 7320 666f  or attributes fo
+00011d70: 7220 6765 6e65 2070 6f73 6974 696f 6e20  r gene position 
+00011d80: 696e 666f 726d 6174 696f 6e20 7265 7472  information retr
+00011d90: 6965 7661 6c2c 2074 6865 206f 7074 696f  ieval, the optio
+00011da0: 6e20 602d 2d61 6464 6974 696f 6e61 6c67  n `--additionalg
+00011db0: 6666 696e 666f 6020 6361 6e20 6265 2073  ffinfo` can be s
+00011dc0: 6574 2074 6f20 7072 6f76 6964 6520 7468  et to provide th
+00011dd0: 6520 6164 6469 7469 6f6e 616c 2069 6e66  e additional inf
+00011de0: 6f72 6d61 7469 6f6e 2e20 5468 6520 7265  ormation. The re
+00011df0: 6d61 696e 696e 6720 7061 7261 6d65 7465  maining paramete
+00011e00: 7220 7365 7474 696e 6720 6973 2074 6865  r setting is the
+00011e10: 2073 616d 6520 6173 2074 6865 2069 6e74   same as the int
+00011e20: 7261 2d73 7065 6369 6669 6320 7379 6e74  ra-specific synt
+00011e30: 656e 7920 696e 6665 7265 6e63 652e 0a0a  eny inference...
+00011e40: 2a2a 4120 7375 6767 6573 7465 6420 7374  **A suggested st
+00011e50: 6172 7469 6e67 2072 756e 2063 616e 2075  arting run can u
+00011e60: 7365 2063 6f6d 6d61 6e64 2073 696d 706c  se command simpl
+00011e70: 7920 6173 2062 656c 6f77 2a2a 0a0a 6060  y as below**..``
+00011e80: 600a 7767 6420 7379 6e20 6661 6d69 6c69  `.wgd syn famili
+00011e90: 6573 2067 6666 3120 6766 6632 0a60 6060  es gff1 gff2.```
+00011ea0: 0a0a 2323 2320 7767 6420 7669 7a0a 0a2a  ..### wgd viz..*
+00011eb0: 2a54 6865 2076 6973 7561 6c69 7a61 7469  *The visualizati
+00011ec0: 6f6e 206f 6620 2a4b 2a3c 7375 623e 533c  on of *K*<sub>S<
+00011ed0: 2f73 7562 3e20 6167 6520 6469 7374 7269  /sub> age distri
+00011ee0: 6275 7469 6f6e 2061 6e64 2045 4c4d 4d20  bution and ELMM 
+00011ef0: 616e 616c 7973 6973 2a2a 0a60 6060 0a77  analysis**.```.w
+00011f00: 6764 2076 697a 202d 6420 6b73 6461 7461  gd viz -d ksdata
+00011f10: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+00011f20: 6d20 6077 6764 2076 697a 6020 6973 206d  m `wgd viz` is m
+00011f30: 6169 6e6c 7920 666f 7220 7468 6520 7075  ainly for the pu
+00011f40: 7270 6f73 6520 6f66 202a 4b2a 3c73 7562  rpose of *K*<sub
+00011f50: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00011f60: 7574 696f 6e20 616e 6420 7379 6e74 656e  ution and synten
+00011f70: 7920 7669 7375 616c 697a 6174 696f 6e2c  y visualization,
+00011f80: 2077 6974 6820 736f 6d65 206f 7074 696f   with some optio
+00011f90: 6e61 6c20 6d69 7874 7572 6520 6d6f 6465  nal mixture mode
+00011fa0: 6c69 6e67 2061 6e61 6c79 7369 732e 2054  ling analysis. T
+00011fb0: 6865 2062 6173 6963 2066 756e 6374 696f  he basic functio
+00011fc0: 6e20 6973 206a 7573 7420 746f 2070 6c6f  n is just to plo
+00011fd0: 7420 7468 6520 2a4b 2a3c 7375 623e 533c  t the *K*<sub>S<
+00011fe0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00011ff0: 6f6e 2061 6e64 2063 6f6e 6475 6374 2061  on and conduct a
+00012000: 6e20 454c 4d4d 2061 6e61 6c79 7369 7320  n ELMM analysis 
+00012010: 696e 2073 6561 7263 6820 6f66 2070 6f74  in search of pot
+00012020: 656e 7469 616c 2057 4744 2063 6f6d 706f  ential WGD compo
+00012030: 6e65 6e74 732e 2053 6f6d 6520 6b65 7920  nents. Some key 
+00012040: 7061 7261 6d65 7465 7273 2061 6666 6563  parameters affec
+00012050: 7469 6e67 2074 6865 2045 4c4d 4d20 7265  ting the ELMM re
+00012060: 7375 6c74 2069 6e63 6c75 6465 2060 2d2d  sult include `--
+00012070: 7072 6f6d 696e 656e 6365 5f63 7574 6f66  prominence_cutof
+00012080: 6660 2061 6e64 2060 2d2d 7265 6c5f 6865  f` and `--rel_he
+00012090: 6967 6874 602c 2077 6869 6368 2068 6176  ight`, which hav
+000120a0: 6520 6265 656e 2065 7870 6c61 696e 6564  e been explained
+000120b0: 2061 6865 6164 2c20 602d 2d65 6d5f 6974   ahead, `--em_it
+000120c0: 6572 6174 696f 6e73 6020 616e 6420 602d  erations` and `-
+000120d0: 2d65 6d5f 696e 6974 6961 6c69 7a61 7469  -em_initializati
+000120e0: 6f6e 7360 2064 6574 6572 6d69 6e69 6e67  ons` determining
+000120f0: 2074 6865 206d 6178 696d 756d 2069 7465   the maximum ite
+00012100: 7261 7469 6f6e 7320 616e 6420 696e 6974  rations and init
+00012110: 6961 6c69 7a61 7469 6f6e 7320 696e 2074  ializations in t
+00012120: 6865 2045 4d20 616c 676f 7269 7468 6d2e  he EM algorithm.
+00012130: 0a0a 2a2a 4120 7375 6767 6573 7465 6420  ..**A suggested 
+00012140: 7374 6172 7469 6e67 2072 756e 2063 616e  starting run can
+00012150: 2075 7365 2063 6f6d 6d61 6e64 2073 696d   use command sim
+00012160: 706c 7920 6173 2062 656c 6f77 2a2a 0a0a  ply as below**..
+00012170: 6060 600a 7767 6420 7669 7a20 2d64 206b  ```.wgd viz -d k
+00012180: 7364 6174 610a 6060 600a 0a2a 2a54 6865  sdata.```..**The
+00012190: 2076 6973 7561 6c69 7a61 7469 6f6e 206f   visualization o
+000121a0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+000121b0: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+000121c0: 6f6e 2077 6974 6820 7261 7465 2063 6f72  on with rate cor
+000121d0: 7265 6374 696f 6e2a 2a0a 6060 600a 7767  rection**.```.wg
+000121e0: 6420 7669 7a20 2d64 206b 7364 6174 6120  d viz -d ksdata 
+000121f0: 2d73 7020 7370 6461 7461 202d 2d66 6f63  -sp spdata --foc
+00012200: 7573 3261 6c6c 2066 6f63 616c 5f73 7065  us2all focal_spe
+00012210: 6369 6573 202d 2d65 7874 7261 7061 7261  cies --extrapara
+00012220: 6e6f 6d65 6b73 206b 7364 6174 6120 282d  nomeks ksdata (-
+00012230: 2d61 6e63 686f 7270 6f69 6e74 7320 6170  -anchorpoints ap
+00012240: 6461 7461 202d 2d70 6c6f 7461 7067 6d6d  data --plotapgmm
+00012250: 202d 2d70 6c6f 7465 6c6d 6d29 0a60 6060   --plotelmm).```
+00012260: 0a0a 4265 7369 6465 7320 7468 6520 6261  ..Besides the ba
+00012270: 7369 6320 2a4b 2a3c 7375 623e 533c 2f73  sic *K*<sub>S</s
+00012280: 7562 3e20 706c 6f74 2c20 7375 6273 7469  ub> plot, substi
+00012290: 7475 7469 6f6e 2072 6174 6520 636f 7272  tution rate corr
+000122a0: 6563 7469 6f6e 2063 616e 2061 6c73 6f20  ection can also 
+000122b0: 6265 2061 6368 6965 7665 6420 6769 7665  be achieved give
+000122c0: 6e20 6174 206c 6561 7374 2061 2073 7065  n at least a spe
+000122d0: 6369 6573 2074 7265 6520 2876 6961 2074  cies tree (via t
+000122e0: 6865 206f 7074 696f 6e20 602d 2d73 7065  he option `--spe
+000122f0: 6369 6573 7472 6565 6029 2061 6e64 2061  ciestree`) and a
+00012300: 2066 6f63 616c 2073 7065 6369 6573 2028   focal species (
+00012310: 6569 7468 6572 2076 6961 2074 6865 206f  either via the o
+00012320: 7074 696f 6e20 602d 2d66 6f63 7573 3261  ption `--focus2a
+00012330: 6c6c 6020 6f72 2076 6961 2074 6865 206f  ll` or via the o
+00012340: 7074 696f 6e20 602d 2d73 7061 6972 6020  ption `--spair` 
+00012350: 696e 2074 6865 2066 6f72 6d20 6f66 2022  in the form of "
+00012360: 2466 6f63 616c 5f73 7065 6369 6573 3b24  $focal_species;$
+00012370: 666f 6361 6c5f 7370 6563 6965 7322 292e  focal_species").
+00012380: 2049 7427 7320 7375 6767 6573 7465 6420   It's suggested 
+00012390: 7468 6174 2074 6865 206f 7274 686f 6c6f  that the ortholo
+000123a0: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+000123b0: 7375 623e 2064 6174 6120 6973 2070 726f  sub> data is pro
+000123c0: 7669 6465 6420 6279 2074 6865 2060 2d2d  vided by the `--
+000123d0: 6461 7461 6669 6c65 6020 6f70 7469 6f6e  datafile` option
+000123e0: 2061 6e64 2074 6865 2070 6172 616c 6f67   and the paralog
+000123f0: 6f75 7320 2a4b 2a3c 7375 623e 533c 2f73  ous *K*<sub>S</s
+00012400: 7562 3e20 6461 7461 2069 7320 7072 6f76  ub> data is prov
+00012410: 6964 6564 2062 7920 7468 6520 602d 2d65  ided by the `--e
+00012420: 7874 7261 7061 7261 6e6f 6d65 6b73 6020  xtraparanomeks` 
+00012430: 6f70 7469 6f6e 2c20 616c 7468 6f75 6768  option, although
+00012440: 2069 7427 7320 616c 6c6f 7765 6420 746f   it's allowed to
+00012450: 206f 6e6c 7920 7072 6f76 6964 6520 2a4b   only provide *K
+00012460: 2a3c 7375 623e 533c 2f73 7562 3e20 6461  *<sub>S</sub> da
+00012470: 7461 2076 6961 2074 6865 2060 2d2d 6461  ta via the `--da
+00012480: 7461 6669 6c65 6020 6f70 7469 6f6e 2061  tafile` option a
+00012490: 6e64 2064 6570 6f73 6974 2062 6f74 6820  nd deposit both 
+000124a0: 6f72 7468 6f6c 6f67 6f75 7320 616e 6420  orthologous and 
+000124b0: 7061 7261 6c6f 676f 7573 202a 4b2a 3c73  paralogous *K*<s
+000124c0: 7562 3e53 3c2f 7375 623e 2064 6174 6120  ub>S</sub> data 
+000124d0: 7468 6572 656f 6e2e 2054 6865 7265 2061  thereon. There a
+000124e0: 7265 2074 776f 2074 7970 6573 206f 6620  re two types of 
+000124f0: 6d69 7865 6420 706c 6f74 7320 2874 6865  mixed plots (the
+00012500: 2022 6d69 7865 6422 2068 6572 6520 7265   "mixed" here re
+00012510: 6665 7273 2074 6f20 6d69 7865 6420 6f72  fers to mixed or
+00012520: 7468 6f6c 6f67 6f75 7320 616e 6420 7061  thologous and pa
+00012530: 7261 6c6f 676f 7573 202a 4b2a 3c73 7562  ralogous *K*<sub
+00012540: 3e53 3c2f 7375 623e 2064 6973 7472 6962  >S</sub> distrib
+00012550: 7574 696f 6e73 292c 206f 6e65 206f 6620  utions), one of 
+00012560: 7768 6963 6820 6973 2073 696d 696c 6172  which is similar
+00012570: 2074 6f20 7768 6174 2060 6b73 7261 7465   to what `ksrate
+00012580: 7360 2070 6c6f 7473 2c20 7768 696c 6520  s` plots, while 
+00012590: 7468 6520 6f74 6865 7220 6f66 2077 6869  the other of whi
+000125a0: 6368 2069 7320 6c69 6b65 2074 6865 2063  ch is like the c
+000125b0: 6f6e 7665 6e74 696f 6e61 6c20 2a4b 2a3c  onventional *K*<
+000125c0: 7375 623e 533c 2f73 7562 3e20 706c 6f74  sub>S</sub> plot
+000125d0: 2074 6861 7420 626f 7468 2074 6865 206f   that both the o
+000125e0: 7269 6769 6e61 6c20 6f72 7468 6f6c 6f67  riginal ortholog
+000125f0: 6f75 7320 616e 6420 7061 7261 6c6f 676f  ous and paralogo
+00012600: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+00012610: 623e 2064 6973 7472 6962 7574 696f 6e73  b> distributions
+00012620: 2061 7265 2074 7275 7468 6675 6c6c 7920   are truthfully 
+00012630: 706c 6f74 7465 6420 696e 7374 6561 6420  plotted instead 
+00012640: 6f66 206a 7573 7420 6265 696e 6720 7265  of just being re
+00012650: 7072 6573 656e 7465 6420 6279 2073 6f6d  presented by som
+00012660: 6520 7665 7274 6963 616c 206c 696e 6573  e vertical lines
+00012670: 2e20 5765 2073 7567 6765 7374 2075 7365  . We suggest use
+00012680: 7273 2074 6f20 6164 6f70 7420 7468 6520  rs to adopt the 
+00012690: 606b 7372 6174 6573 602d 6c69 6b65 2070  `ksrates`-like p
+000126a0: 6c6f 7473 2c20 7768 6963 6820 6973 2074  lots, which is t
+000126b0: 6865 2064 6566 6175 6c74 2e20 4f74 6865  he default. Othe
+000126c0: 7277 6973 6520 7468 6520 6f70 7469 6f6e  rwise the option
+000126d0: 2060 2d2d 636c 6173 7369 6360 2077 696c   `--classic` wil
+000126e0: 6c20 6265 206e 6565 6465 6420 746f 2073  l be needed to s
+000126f0: 6574 2e20 4578 7472 6120 6d69 7874 7572  et. Extra mixtur
+00012700: 6520 6d6f 6465 6c69 6e67 2061 6e61 6c79  e modeling analy
+00012710: 7369 7320 6361 6e20 6265 2069 6e69 7469  sis can be initi
+00012720: 6174 6564 2076 6961 2074 6865 206f 7074  ated via the opt
+00012730: 696f 6e20 602d 2d70 6c6f 7465 6c6d 6d60  ion `--plotelmm`
+00012740: 2061 6e64 2060 2d2d 706c 6f74 6170 676d   and `--plotapgm
+00012750: 6d60 2077 6974 6820 7468 6520 616e 6368  m` with the anch
+00012760: 6f72 2070 6f69 6e74 7320 6461 7461 6669  or points datafi
+00012770: 6c65 2070 726f 7669 6465 6420 6279 2074  le provided by t
+00012780: 6865 2060 2d2d 616e 6368 6f72 706f 696e  he `--anchorpoin
+00012790: 7473 6020 6f70 7469 6f6e 2e0a 0a2a 2a41  ts` option...**A
+000127a0: 2073 7567 6765 7374 6564 2073 7461 7274   suggested start
+000127b0: 696e 6720 7275 6e20 6361 6e20 7573 6520  ing run can use 
+000127c0: 636f 6d6d 616e 6420 7369 6d70 6c79 2061  command simply a
+000127d0: 7320 6265 6c6f 772a 2a0a 0a60 6060 0a77  s below**..```.w
+000127e0: 6764 2076 697a 202d 6420 6b73 6461 7461  gd viz -d ksdata
+000127f0: 202d 7370 2073 7064 6174 6120 2d2d 666f   -sp spdata --fo
+00012800: 6375 7332 616c 6c20 666f 6361 6c5f 7370  cus2all focal_sp
+00012810: 6563 6965 7320 2d2d 6578 7472 6170 6172  ecies --extrapar
+00012820: 616e 6f6d 656b 7320 6b73 6461 7461 0a60  anomeks ksdata.`
+00012830: 6060 0a0a 2a2a 5468 6520 7669 7375 616c  ``..**The visual
+00012840: 697a 6174 696f 6e20 6f66 2073 796e 7465  ization of synte
+00012850: 6e79 2a2a 0a60 6060 0a77 6764 2076 697a  ny**.```.wgd viz
+00012860: 202d 6170 2061 7064 6174 6120 2d73 6d20   -ap apdata -sm 
+00012870: 736d 6461 7461 202d 6d74 206d 7464 6174  smdata -mt mtdat
+00012880: 6120 2d67 7420 6774 6461 7461 202d 2d70  a -gt gtdata --p
+00012890: 6c6f 7473 796e 2028 2d2d 6d69 6e6c 656e  lotsyn (--minlen
+000128a0: 202d 3120 2d2d 6d69 6e73 6567 6c65 6e20   -1 --minseglen 
+000128b0: 3130 3030 3020 2d2d 6d69 6e67 656e 656e  10000 --mingenen
+000128c0: 756d 2033 3029 0a60 6060 0a0a 436f 6d70  um 30).```..Comp
+000128d0: 6172 6564 2074 6f20 7468 6520 6f72 6967  ared to the orig
+000128e0: 696e 616c 2060 7767 6460 2c20 7468 6520  inal `wgd`, the 
+000128f0: 6077 6764 2076 697a 6020 7072 6f67 7261  `wgd viz` progra
+00012900: 6d20 6164 6465 6420 7379 6e74 656e 7920  m added synteny 
+00012910: 7669 7375 616c 697a 6174 696f 6e20 7069  visualization pi
+00012920: 7065 6c69 6e65 2e20 5573 6572 7320 6e65  peline. Users ne
+00012930: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
+00012940: 6520 666c 6167 206f 7074 696f 6e20 602d  e flag option `-
+00012950: 2d70 6c6f 7473 796e 6020 746f 2069 6e69  -plotsyn` to ini
+00012960: 7469 6174 6520 7468 6973 2070 6172 7420  tiate this part 
+00012970: 6f66 2070 6970 656c 696e 652e 2054 6869  of pipeline. Thi
+00012980: 7320 7374 6570 2061 7373 756d 6573 2074  s step assumes t
+00012990: 6861 7420 7573 6572 7320 6861 7665 206f  hat users have o
+000129a0: 6274 6169 6e65 6420 616c 7265 6164 7920  btained already 
+000129b0: 7468 6520 7379 6e74 656e 6963 2072 6573  the syntenic res
+000129c0: 756c 7473 2066 726f 6d20 6069 2d61 6468  ults from `i-adh
+000129d0: 6f72 6560 2061 6e64 2075 7365 7320 7468  ore` and uses th
+000129e0: 6f73 6520 7265 7375 6c74 2066 696c 6573  ose result files
+000129f0: 2074 6f20 7265 616c 697a 6520 7468 6520   to realize the 
+00012a00: 7379 6e74 656e 7920 7669 7375 616c 697a  synteny visualiz
+00012a10: 6174 696f 6e2e 2053 696d 6c69 6172 2074  ation. Simliar t
+00012a20: 6f20 6077 6764 2073 796e 602c 2074 6865  o `wgd syn`, the
+00012a30: 2065 7874 7261 202a 4b2a 3c73 7562 3e53   extra *K*<sub>S
+00012a40: 3c2f 7375 623e 2064 6174 6120 6669 6c65  </sub> data file
+00012a50: 2063 616e 2062 6520 7472 616e 736d 6974   can be transmit
+00012a60: 7465 6420 7669 6120 7468 6520 6f70 7469  ted via the opti
+00012a70: 6f6e 2060 2d2d 6461 7461 6669 6c65 6020  on `--datafile` 
+00012a80: 2869 6e73 7465 6164 206f 6620 602d 2d65  (instead of `--e
+00012a90: 7874 7261 7061 7261 6e6f 6d65 6b73 6020  xtraparanomeks` 
+00012aa0: 6f70 7469 6f6e 292e 2042 6173 6963 616c  option). Basical
+00012ab0: 6c79 2c20 7468 6520 7379 6e74 656e 6963  ly, the syntenic
+00012ac0: 2072 6573 756c 7420 6669 6c65 7320 7265   result files re
+00012ad0: 7175 6972 6564 2061 7265 2074 6865 2061  quired are the a
+00012ae0: 6e63 686f 7220 706f 696e 7473 2064 6174  nchor points dat
+00012af0: 6166 696c 652c 206d 756c 7469 706c 6963  afile, multiplic
+00012b00: 6f6e 7320 6461 7461 6669 6c65 2c20 6765  ons datafile, ge
+00012b10: 6e65 2074 6162 6c65 2064 6174 6166 696c  ne table datafil
+00012b20: 6520 2861 7574 6f6d 6174 656c 7920 7072  e (automately pr
+00012b30: 6f64 7563 6564 2062 7920 6077 6764 2073  oduced by `wgd s
+00012b40: 796e 6029 2c20 616e 6420 7365 676d 656e  yn`), and segmen
+00012b50: 7473 2064 6174 6166 696c 652e 0a0a 2a2a  ts datafile...**
+00012b60: 4120 7375 6767 6573 7465 6420 7374 6172  A suggested star
+00012b70: 7469 6e67 2072 756e 2063 616e 2075 7365  ting run can use
+00012b80: 2063 6f6d 6d61 6e64 2073 696d 706c 7920   command simply 
+00012b90: 6173 2062 656c 6f77 2a2a 0a0a 6060 600a  as below**..```.
+00012ba0: 7767 6420 7669 7a20 2d61 7020 6170 6461  wgd viz -ap apda
+00012bb0: 7461 202d 736d 2073 6d64 6174 6120 2d6d  ta -sm smdata -m
+00012bc0: 7420 6d74 6461 7461 202d 6774 2067 7464  t mtdata -gt gtd
+00012bd0: 6174 6120 2d2d 706c 6f74 7379 6e0a 6060  ata --plotsyn.``
+00012be0: 600a 0a23 2323 2077 6764 206d 6978 0a0a  `..### wgd mix..
+00012bf0: 2a2a 5468 6520 6d69 7874 7572 6520 6d6f  **The mixture mo
+00012c00: 6465 6c20 636c 7573 7465 7269 6e67 2061  del clustering a
+00012c10: 6e61 6c79 7369 7320 6f66 202a 4b2a 3c73  nalysis of *K*<s
+00012c20: 7562 3e53 3c2f 7375 623e 2061 6765 2064  ub>S</sub> age d
+00012c30: 6973 7472 6962 7574 696f 6e2a 2a0a 6060  istribution**.``
+00012c40: 600a 7767 6420 6d69 7820 6b73 6461 7461  `.wgd mix ksdata
+00012c50: 2028 2d2d 6e5f 696e 6974 2032 3030 202d   (--n_init 200 -
+00012c60: 2d6d 6178 5f69 7465 7220 3230 3020 2d2d  -max_iter 200 --
+00012c70: 6b73 5f72 616e 6765 2030 2035 202d 2d66  ks_range 0 5 --f
+00012c80: 696c 7465 7273 2033 3030 202d 2d62 696e  ilters 300 --bin
+00012c90: 7320 3530 202d 2d63 6f6d 706f 6e65 6e74  s 50 --component
+00012ca0: 7320 3120 3420 2d2d 6761 6d6d 6120 302e  s 1 4 --gamma 0.
+00012cb0: 3030 3129 0a60 6060 0a0a 5468 6973 2070  001).```..This p
+00012cc0: 6172 7420 6f66 2047 6175 7373 6961 6e20  art of Gaussian 
+00012cd0: 6d69 7874 7572 6520 6d6f 6465 6c69 6e67  mixture modeling
+00012ce0: 2028 474d 4d29 2061 6e61 6c79 7369 7320   (GMM) analysis 
+00012cf0: 6973 2069 6e68 6572 6974 6564 2066 726f  is inherited fro
+00012d00: 6d20 7468 6520 6f72 6967 696e 616c 2060  m the original `
+00012d10: 7767 6460 2070 726f 6772 616d 2c20 6275  wgd` program, bu
+00012d20: 7420 7772 6974 6573 2061 6464 6974 696f  t writes additio
+00012d30: 6e61 6c6c 7920 7468 6520 7072 6f62 6162  nally the probab
+00012d40: 696c 6974 7920 6f66 2065 6163 6820 2a4b  ility of each *K
+00012d50: 2a3c 7375 623e 533c 2f73 7562 3e20 7661  *<sub>S</sub> va
+00012d60: 6c75 6520 696e 746f 2074 6865 2066 696e  lue into the fin
+00012d70: 616c 2064 6174 6166 7261 6d65 2e20 4261  al dataframe. Ba
+00012d80: 7369 6361 6c6c 792c 2075 7365 7273 206e  sically, users n
+00012d90: 6565 6420 746f 2070 726f 7669 6465 2077  eed to provide w
+00012da0: 6974 6820 6120 286e 6f72 6d61 6c6c 7920  ith a (normally 
+00012db0: 6672 6f6d 2077 686f 6c65 2d70 6172 616e  from whole-paran
+00012dc0: 6f6d 6520 6f72 2061 6e63 686f 722d 7061  ome or anchor-pa
+00012dd0: 6972 7329 202a 4b2a 3c73 7562 3e53 3c2f  irs) *K*<sub>S</
+00012de0: 7375 623e 2064 6174 6166 696c 6520 616e  sub> datafile an
+00012df0: 6420 7468 6520 474d 4d20 616e 616c 7973  d the GMM analys
+00012e00: 6973 2077 696c 6c20 6265 2063 6f6e 6475  is will be condu
+00012e10: 6374 6564 2075 706f 6e20 7468 6520 6461  cted upon the da
+00012e20: 7461 6669 6c65 2e20 536f 6d65 2070 6172  tafile. Some par
+00012e30: 616d 6574 6572 7320 6361 6e20 6166 6665  ameters can affe
+00012e40: 6374 2074 6865 2072 6573 756c 7473 2c20  ct the results, 
+00012e50: 696e 636c 7564 696e 6720 602d 2d6e 5f69  including `--n_i
+00012e60: 6e69 7460 2c20 7768 6963 6820 7365 7473  nit`, which sets
+00012e70: 2074 6865 206e 756d 6265 7220 6f66 206b   the number of k
+00012e80: 2d6d 6561 6e73 2069 6e69 7469 616c 697a  -means initializ
+00012e90: 6174 696f 6e73 2028 6465 6661 756c 7420  ations (default 
+00012ea0: 3230 3029 2c20 602d 2d6d 6178 5f69 7465  200), `--max_ite
+00012eb0: 7260 2c20 7768 6963 6820 7365 7473 2074  r`, which sets t
+00012ec0: 6865 206d 6178 696d 756d 206e 756d 6265  he maximum numbe
+00012ed0: 7220 6f66 2069 7465 7261 7469 6f6e 7320  r of iterations 
+00012ee0: 2864 6566 6175 6c74 2032 3030 292c 2060  (default 200), `
+00012ef0: 2d2d 6d65 7468 6f64 602c 2077 6869 6368  --method`, which
+00012f00: 2064 6574 6572 6d69 6e65 7320 7768 6963   determines whic
+00012f10: 6820 636c 7573 7465 7269 6e67 206d 6574  h clustering met
+00012f20: 686f 6420 746f 2075 7365 2028 6465 6661  hod to use (defa
+00012f30: 756c 7420 676d 6d29 2c20 602d 2d67 616d  ult gmm), `--gam
+00012f40: 6d61 602c 2077 6869 6368 2073 6574 7320  ma`, which sets 
+00012f50: 7468 6520 6761 6d6d 6120 7061 7261 6d65  the gamma parame
+00012f60: 7465 7220 666f 7220 7468 6520 6267 6d6d  ter for the bgmm
+00012f70: 206d 6f64 656c 2028 6465 6661 756c 7420   model (default 
+00012f80: 302e 3030 3129 2c20 602d 2d63 6f6d 706f  0.001), `--compo
+00012f90: 6e65 6e74 7360 2c20 7768 6963 6820 7365  nents`, which se
+00012fa0: 7473 2074 6865 2072 616e 6765 206f 6620  ts the range of 
+00012fb0: 7468 6520 6e75 6d62 6572 206f 6620 636f  the number of co
+00012fc0: 6d70 6f6e 656e 7473 2074 6f20 6669 7420  mponents to fit 
+00012fd0: 2864 6566 6175 6c74 2031 2034 292c 2061  (default 1 4), a
+00012fe0: 6e64 2074 6865 2064 6174 6120 6669 6c74  nd the data filt
+00012ff0: 6572 696e 6720 7061 7261 6d65 7465 7273  ering parameters
+00013000: 2060 2d2d 6669 6c74 6572 7360 2077 6869   `--filters` whi
+00013010: 6368 2066 696c 7465 7273 2064 6174 6120  ch filters data 
+00013020: 6261 7365 6420 6f6e 2061 6c69 676e 6d65  based on alignme
+00013030: 6e74 206c 656e 6774 682c 2060 2d2d 6b73  nt length, `--ks
+00013040: 5f72 616e 6765 6020 7768 6963 6820 6669  _range` which fi
+00013050: 6c74 6572 7320 6461 7461 2062 6173 6564  lters data based
+00013060: 206f 6e20 2a4b 2a3c 7375 623e 533c 2f73   on *K*<sub>S</s
+00013070: 7562 3e20 7661 6c75 6573 2061 6e64 2074  ub> values and t
+00013080: 6865 2070 6172 616d 6574 6572 2060 2d2d  he parameter `--
+00013090: 6269 6e73 6020 7768 6963 6820 7365 7473  bins` which sets
+000130a0: 2074 6865 206e 756d 6265 7220 6f66 2062   the number of b
+000130b0: 696e 7320 696e 202a 4b2a 3c73 7562 3e53  ins in *K*<sub>S
+000130c0: 3c2f 7375 623e 2064 6973 7472 6962 7574  </sub> distribut
+000130d0: 696f 6e20 2864 6566 6175 6c74 2035 3029  ion (default 50)
+000130e0: 2e0a 0a2a 2a41 2073 7567 6765 7374 6564  ...**A suggested
+000130f0: 2073 7461 7274 696e 6720 7275 6e20 6361   starting run ca
+00013100: 6e20 7573 6520 636f 6d6d 616e 6420 7369  n use command si
+00013110: 6d70 6c79 2061 7320 6265 6c6f 772a 2a0a  mply as below**.
+00013120: 0a60 6060 0a77 6764 206d 6978 206b 7364  .```.wgd mix ksd
+00013130: 6174 610a 6060 600a 0a23 2323 2077 6764  ata.```..### wgd
+00013140: 2070 6561 6b0a 0a2a 2a54 6865 2073 6561   peak..**The sea
+00013150: 7263 6820 6f66 2063 7265 6469 6162 6c65  rch of crediable
+00013160: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00013170: 2072 616e 6765 2075 7365 6420 696e 2057   range used in W
+00013180: 4744 2064 6174 696e 672a 2a0a 6060 600a  GD dating**.```.
+00013190: 7767 6420 7065 616b 206b 7364 6174 6120  wgd peak ksdata 
+000131a0: 2d61 7020 6170 6461 7461 202d 736d 2073  -ap apdata -sm s
+000131b0: 6d64 6174 6120 2d6c 6520 6c65 6461 7461  mdata -le ledata
+000131c0: 202d 6d70 206d 7064 6174 6120 2d2d 6865   -mp mpdata --he
+000131d0: 7572 6973 7469 6320 282d 2d61 6c69 676e  uristic (--align
+000131e0: 6669 6c74 6572 2030 2e30 2030 2030 2e30  filter 0.0 0 0.0
+000131f0: 202d 2d6b 7372 616e 6765 2030 2035 202d   --ksrange 0 5 -
+00013200: 2d62 696e 5f77 6964 7468 2030 2e31 202d  -bin_width 0.1 -
+00013210: 2d67 7569 6465 2073 6567 6d65 6e74 202d  -guide segment -
+00013220: 2d70 726f 6d69 6e65 6e63 655f 6375 746f  -prominence_cuto
+00013230: 6666 2030 2e31 202d 2d72 656c 5f68 6569  ff 0.1 --rel_hei
+00013240: 6768 7420 302e 3420 2d2d 6369 2039 3520  ght 0.4 --ci 95 
+00013250: 2d2d 6864 7220 3935 202d 2d6b 7363 7574  --hdr 95 --kscut
+00013260: 6f66 6620 3529 0a60 6060 0a0a 4173 206d  off 5).```..As m
+00013270: 656e 7469 6f6e 6564 2070 7265 7669 6f75  entioned previou
+00013280: 736c 792c 2061 2068 6575 7269 7374 6963  sly, a heuristic
+00013290: 206d 6574 686f 6420 616e 6420 6120 636f   method and a co
+000132a0: 6c6c 696e 6561 7220 7365 676d 656e 7473  llinear segments
+000132b0: 2d67 7569 6465 6420 616e 6368 6f72 2070  -guided anchor p
+000132c0: 6169 7220 636c 7573 7465 7269 6e67 2066  air clustering f
+000132d0: 6f72 2074 6865 2073 6561 7263 6820 6f66  or the search of
+000132e0: 2063 7265 6469 6162 6c65 202a 4b2a 3c73   crediable *K*<s
+000132f0: 7562 3e53 3c2f 7375 623e 2072 616e 6765  ub>S</sub> range
+00013300: 2075 7365 6420 696e 2057 4744 2064 6174   used in WGD dat
+00013310: 696e 6720 6172 6520 696d 706c 656d 656e  ing are implemen
+00013320: 7465 6420 696e 2060 7767 6420 7632 602e  ted in `wgd v2`.
+00013330: 2055 7365 7273 206e 6565 6420 746f 2070   Users need to p
+00013340: 726f 7669 6465 2074 6865 2061 6e63 686f  rovide the ancho
+00013350: 7220 706f 696e 7473 2c20 7365 676d 656e  r points, segmen
+00013360: 7473 2c20 6c69 7374 7365 676d 656e 7473  ts, listsegments
+00013370: 2c20 6d75 6c74 6970 6c69 636f 6e70 6169  , multipliconpai
+00013380: 7273 2064 6174 6166 696c 6520 6672 6f6d  rs datafile from
+00013390: 2060 692d 6164 686f 7265 6020 746f 2061   `i-adhore` to a
+000133a0: 6368 6965 7665 2074 6865 2063 6c75 7374  chieve the clust
+000133b0: 6572 696e 6720 6675 6e63 7469 6f6e 2e20  ering function. 
+000133c0: 536f 6d65 2070 6172 616d 6574 6572 7320  Some parameters 
+000133d0: 7468 6174 2063 616e 2069 6d70 6163 7420  that can impact 
+000133e0: 7468 6520 7265 7375 6c74 7320 696e 636c  the results incl
+000133f0: 7564 6520 602d 2d61 6c69 676e 6669 6c74  ude `--alignfilt
+00013400: 6572 602c 2077 6869 6368 2066 696c 7465  er`, which filte
+00013410: 7273 2074 6865 2064 6174 6120 6261 7365  rs the data base
+00013420: 6420 6f6e 2061 6c69 676e 6d65 6e74 2069  d on alignment i
+00013430: 6465 6e74 6974 792c 206c 656e 6774 6820  dentity, length 
+00013440: 616e 6420 636f 7665 7261 6765 2c20 602d  and coverage, `-
+00013450: 2d6b 7372 616e 6765 602c 2077 6869 6368  -ksrange`, which
+00013460: 2073 6574 7320 7468 6520 7261 6e67 6520   sets the range 
+00013470: 6f66 204b 7320 746f 2062 6520 616e 616c  of Ks to be anal
+00013480: 797a 6564 2c20 602d 2d62 696e 5f77 6964  yzed, `--bin_wid
+00013490: 7468 602c 2077 6869 6368 2073 6574 7320  th`, which sets 
+000134a0: 7468 6520 6261 6e64 7769 6474 6820 6f66  the bandwidth of
+000134b0: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+000134c0: 2064 6973 7472 6962 7574 696f 6e2c 2060   distribution, `
+000134d0: 2d2d 7765 6967 6874 735f 6f75 746c 6965  --weights_outlie
+000134e0: 7273 5f69 6e63 6c75 6465 6460 2077 6869  rs_included` whi
+000134f0: 6368 2064 6574 6572 6d69 6e65 7320 7768  ch determines wh
+00013500: 6574 6865 7220 746f 2069 6e63 6c75 6465  ether to include
+00013510: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00013520: 206f 7574 6c69 6572 7320 2877 686f 7365   outliers (whose
+00013530: 2076 616c 7565 2069 7320 6f76 6572 2035   value is over 5
+00013540: 2920 696e 2061 6e61 6c79 7369 732c 2060  ) in analysis, `
+00013550: 2d2d 6d65 7468 6f64 6020 7768 6963 6820  --method` which 
+00013560: 6465 7465 726d 696e 6573 2077 6869 6368  determines which
+00013570: 2063 6c75 7374 6572 696e 6720 6d65 7468   clustering meth
+00013580: 6f64 2074 6f20 7573 6520 2864 6566 6175  od to use (defau
+00013590: 6c74 2067 6d6d 292c 2060 2d2d 7365 6564  lt gmm), `--seed
+000135a0: 6020 7768 6963 6820 7365 7473 2074 6865  ` which sets the
+000135b0: 2072 616e 646f 6d20 7365 6564 2067 6976   random seed giv
+000135c0: 656e 2074 6f20 696e 6974 6961 6c69 7a61  en to initializa
+000135d0: 7469 6f6e 2028 6465 6661 756c 7420 3233  tion (default 23
+000135e0: 3532 3839 3029 2c20 602d 2d6e 5f69 6e69  52890), `--n_ini
+000135f0: 7460 2c20 7768 6963 6820 7365 7473 2074  t`, which sets t
+00013600: 6865 206e 756d 6265 7220 6f66 206b 2d6d  he number of k-m
+00013610: 6561 6e73 2069 6e69 7469 616c 697a 6174  eans initializat
+00013620: 696f 6e73 2028 6465 6661 756c 7420 3230  ions (default 20
+00013630: 3029 2c20 602d 2d65 6d5f 6974 6572 602c  0), `--em_iter`,
+00013640: 2077 6869 6368 2073 6574 7320 7468 6520   which sets the 
+00013650: 6d61 7869 6d75 6d20 6e75 6d62 6572 206f  maximum number o
+00013660: 6620 6974 6572 6174 696f 6e73 2028 6465  f iterations (de
+00013670: 6661 756c 7420 3230 3029 2c20 602d 2d67  fault 200), `--g
+00013680: 616d 6d61 602c 2077 6869 6368 2073 6574  amma`, which set
+00013690: 7320 7468 6520 6761 6d6d 6120 7061 7261  s the gamma para
+000136a0: 6d65 7465 7220 666f 7220 7468 6520 6267  meter for the bg
+000136b0: 6d6d 206d 6f64 656c 2028 6465 6661 756c  mm model (defaul
+000136c0: 7420 302e 3030 3129 2c20 602d 2d63 6f6d  t 0.001), `--com
+000136d0: 706f 6e65 6e74 7360 2c20 7768 6963 6820  ponents`, which 
+000136e0: 7365 7473 2074 6865 2072 616e 6765 206f  sets the range o
+000136f0: 6620 7468 6520 6e75 6d62 6572 206f 6620  f the number of 
+00013700: 636f 6d70 6f6e 656e 7473 2074 6f20 6669  components to fi
+00013710: 7420 2864 6566 6175 6c74 2031 2034 292c  t (default 1 4),
+00013720: 2060 2d2d 7765 6967 6874 6564 6020 7768   `--weighted` wh
+00013730: 6963 6820 6465 7465 726d 696e 6573 2077  ich determines w
+00013740: 6865 7468 6572 2074 6f20 7573 6520 6e6f  hether to use no
+00013750: 6465 2d77 6569 6768 7465 6420 6d65 7468  de-weighted meth
+00013760: 6f64 2066 6f72 2064 652d 7265 6475 6e64  od for de-redund
+00013770: 616e 6379 2c20 602d 2d67 7569 6465 6020  ancy, `--guide` 
+00013780: 7768 6963 6820 6465 7465 726d 696e 6573  which determines
+00013790: 2077 6869 6368 2072 6567 696d 6520 7265   which regime re
+000137a0: 7369 6469 6e67 2061 6e63 686f 7273 2074  siding anchors t
+000137b0: 6f20 6265 2075 7365 6420 2864 6566 6175  o be used (defau
+000137c0: 6c74 2053 6567 6d65 6e74 292c 2060 2d2d  lt Segment), `--
+000137d0: 7072 6f6d 696e 656e 6365 5f63 7574 6f66  prominence_cutof
+000137e0: 6660 2077 6869 6368 2073 6574 7320 7468  f` which sets th
+000137f0: 6520 7072 6f6d 696e 656e 6365 2063 7574  e prominence cut
+00013800: 6f66 6620 6f66 2061 6363 6570 7461 626c  off of acceptabl
+00013810: 6520 7065 616b 7320 696e 2070 6561 6b20  e peaks in peak 
+00013820: 6669 6e64 696e 6720 7072 6f63 6573 732c  finding process,
+00013830: 2060 2d2d 7265 6c5f 6865 6967 6874 6020   `--rel_height` 
+00013840: 7768 6963 6820 7365 7473 2074 6865 2072  which sets the r
+00013850: 656c 6174 6976 6520 6865 6967 6874 2061  elative height a
+00013860: 7420 7768 6963 6820 7468 6520 7065 616b  t which the peak
+00013870: 2077 6964 7468 2069 7320 6d65 6173 7572   width is measur
+00013880: 6564 2c20 602d 2d6b 7374 6f64 6174 6560  ed, `--kstodate`
+00013890: 2077 6869 6368 206d 616e 7561 6c6c 7920   which manually 
+000138a0: 7365 7473 2074 6865 2072 616e 6765 206f  sets the range o
+000138b0: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+000138c0: 3e20 746f 2062 6520 6461 7465 6420 696e  > to be dated in
+000138d0: 2068 6575 7269 7374 6963 2073 6561 7263   heuristic searc
+000138e0: 6820 616e 6420 6e65 6564 7320 746f 2062  h and needs to b
+000138f0: 6520 636f 2d73 6574 2077 6974 6820 6f70  e co-set with op
+00013900: 7469 6f6e 2060 2d2d 6d61 6e75 616c 7365  tion `--manualse
+00013910: 7460 2c20 602d 2d78 6c69 6d60 2061 6e64  t`, `--xlim` and
+00013920: 2060 2d2d 796c 696d 6020 6465 7465 726d   `--ylim` determ
+00013930: 696e 696e 6720 7468 6520 7820 616e 6420  ining the x and 
+00013940: 7920 6178 6973 206c 696d 6974 206f 6620  y axis limit of 
+00013950: 474d 4d20 2a4b 2a3c 7375 623e 533c 2f73  GMM *K*<sub>S</s
+00013960: 7562 3e20 6469 7374 7269 6275 7469 6f6e  ub> distribution
+00013970: 2c20 602d 2d63 6960 2073 6574 7469 6e67  , `--ci` setting
+00013980: 2074 6865 2063 6f6e 6669 6465 6e63 6520   the confidence 
+00013990: 6c65 7665 6c20 6f66 206c 6f67 2d6e 6f72  level of log-nor
+000139a0: 6d61 6c20 6469 7374 7269 6275 7469 6f6e  mal distribution
+000139b0: 2074 6f20 6461 7465 2028 6465 6661 756c   to date (defaul
+000139c0: 7420 3935 292c 2060 2d2d 6864 7260 2073  t 95), `--hdr` s
+000139d0: 6574 7469 6e67 2074 6865 2068 6967 6865  etting the highe
+000139e0: 7374 2064 656e 7369 7479 2072 6567 696f  st density regio
+000139f0: 6e20 2848 4452 2920 6170 706c 6965 6420  n (HDR) applied 
+00013a00: 696e 2074 6865 2073 6567 6d65 6e74 2d67  in the segment-g
+00013a10: 7569 6465 6420 616e 6368 6f72 2070 6169  uided anchor pai
+00013a20: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
+00013a30: 3e20 6469 7374 7269 6275 7469 6f6e 2028  > distribution (
+00013a40: 6465 6661 756c 7420 3935 292c 2060 2d2d  default 95), `--
+00013a50: 6865 7572 6973 7469 6360 2064 6574 6572  heuristic` deter
+00013a60: 6d69 6e69 6e67 2077 6865 7468 6572 2074  mining whether t
+00013a70: 6f20 696e 6974 6961 7465 2068 6575 7269  o initiate heuri
+00013a80: 7374 6963 206d 6574 686f 6420 6f66 2064  stic method of d
+00013a90: 6566 696e 696e 6720 4349 2066 6f72 2064  efining CI for d
+00013aa0: 6174 696e 672c 2060 2d2d 6b73 6375 746f  ating, `--kscuto
+00013ab0: 6666 6020 7365 7474 696e 6720 7468 6520  ff` setting the 
+00013ac0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00013ad0: 7361 7475 7261 7469 6f6e 2063 7574 6f66  saturation cutof
+00013ae0: 6620 696e 2064 6174 696e 6720 2864 6566  f in dating (def
+00013af0: 6175 6c74 2035 292e 0a0a 2a2a 4120 7375  ault 5)...**A su
+00013b00: 6767 6573 7465 6420 7374 6172 7469 6e67  ggested starting
+00013b10: 2072 756e 2063 616e 2075 7365 2063 6f6d   run can use com
+00013b20: 6d61 6e64 2073 696d 706c 7920 6173 2062  mand simply as b
+00013b30: 656c 6f77 2a2a 0a0a 6060 600a 7767 6420  elow**..```.wgd 
+00013b40: 7065 616b 206b 7364 6174 6120 2d61 7020  peak ksdata -ap 
+00013b50: 6170 6461 7461 202d 736d 2073 6d64 6174  apdata -sm smdat
+00013b60: 6120 2d6c 6520 6c65 6461 7461 202d 6d70  a -le ledata -mp
+00013b70: 206d 7064 6174 6120 2d2d 6865 7572 6973   mpdata --heuris
+00013b80: 7469 630a 6060 600a 0a23 2323 2077 6764  tic.```..### wgd
+00013b90: 2066 6f63 7573 0a0a 2a2a 5468 6520 636f   focus..**The co
+00013ba0: 6e63 6174 656e 6174 696f 6e2d 6261 7365  ncatenation-base
+00013bb0: 642f 636f 616c 6573 6365 6e63 652d 6261  d/coalescence-ba
+00013bc0: 7365 6420 7068 796c 6f67 656e 6574 6963  sed phylogenetic
+00013bd0: 2069 6e66 6572 656e 6365 2a2a 0a60 6060   inference**.```
+00013be0: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+00013bf0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+00013c00: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+00013c10: 3320 282d 2d63 6f6e 6361 7465 6e61 7469  3 (--concatenati
+00013c20: 6f6e 2920 282d 2d63 6f61 6c65 7363 656e  on) (--coalescen
+00013c30: 6365 2920 282d 7472 6565 2027 6661 7374  ce) (-tree 'fast
+00013c40: 7472 6565 2729 2028 2d74 7320 272d 6661  tree') (-ts '-fa
+00013c50: 7374 6573 7427 2920 282d 6e20 3429 2028  stest') (-n 4) (
+00013c60: 2d2d 746f 5f73 746f 7029 2028 2d2d 6364  --to_stop) (--cd
+00013c70: 7329 2028 2d6f 2077 6764 5f66 6f63 7573  s) (-o wgd_focus
+00013c80: 2920 282d 7420 776f 726b 696e 675f 746d  ) (-t working_tm
+00013c90: 7029 0a60 6060 0a0a 5468 6520 7072 6f67  p).```..The prog
+00013ca0: 7261 6d20 6077 6764 2066 6f63 7573 6020  ram `wgd focus` 
+00013cb0: 696d 706c 656d 656e 7465 6420 7477 6f20  implemented two 
+00013cc0: 6261 7369 6320 7068 796c 6f67 656e 6574  basic phylogenet
+00013cd0: 6963 2069 6e66 6572 656e 6365 206d 6574  ic inference met
+00013ce0: 686f 6473 2c20 692e 652e 2c20 636f 6e63  hods, i.e., conc
+00013cf0: 6174 656e 6174 696f 6e2d 6261 7365 6420  atenation-based 
+00013d00: 616e 6420 636f 616c 6573 6365 6e63 652d  and coalescence-
+00013d10: 6261 7365 6420 6d65 7468 6f64 732e 2054  based methods. T
+00013d20: 6f20 696e 6974 6961 7465 2074 6865 7365  o initiate these
+00013d30: 2061 6e61 6c79 7369 732c 2075 7365 7273   analysis, users
+00013d40: 206e 6565 6420 746f 2073 6574 2074 6865   need to set the
+00013d50: 2066 6c61 6720 6f70 7469 6f6e 2060 2d2d   flag option `--
+00013d60: 636f 6e63 6174 656e 6174 696f 6e60 206f  concatenation` o
+00013d70: 7220 602d 2d63 6f61 6c65 7363 656e 6365  r `--coalescence
+00013d80: 602e 2054 6865 2063 6f6e 6361 7465 6e61  `. The concatena
+00013d90: 7469 6f6e 2d62 6173 6564 206d 6574 686f  tion-based metho
+00013da0: 6420 696e 636c 7564 6573 2061 2066 6577  d includes a few
+00013db0: 206d 616a 6f72 2073 7465 7073 2c20 692e   major steps, i.
+00013dc0: 652e 2c20 7468 6520 6d75 6c74 6970 6c65  e., the multiple
+00013dd0: 2073 6571 7565 6e63 6520 616c 6967 6e6d   sequence alignm
+00013de0: 656e 7420 284d 5341 2920 6f66 2065 6163  ent (MSA) of eac
+00013df0: 6820 6765 6e65 2066 616d 696c 792c 2074  h gene family, t
+00013e00: 6865 2063 6f6e 6361 7465 6e61 7469 6f6e  he concatenation
+00013e10: 206f 6620 616c 6c20 6765 6e65 2066 616d   of all gene fam
+00013e20: 696c 6965 7320 616e 6420 7468 656e 2074  ilies and then t
+00013e30: 6865 2067 656e 6520 7472 6565 2028 616c  he gene tree (al
+00013e40: 736f 2073 7065 6369 6573 2074 7265 6520  so species tree 
+00013e50: 696e 2074 6869 7320 6361 7365 2920 696e  in this case) in
+00013e60: 6665 7265 6e63 652e 2054 6865 2063 6f61  ference. The coa
+00013e70: 6c65 7363 656e 6365 2d62 6173 6564 206d  lescence-based m
+00013e80: 6574 686f 6420 7769 6c6c 2069 6e73 7465  ethod will inste
+00013e90: 6164 2070 6572 666f 726d 2074 6865 204d  ad perform the M
+00013ea0: 5341 206f 6620 6561 6368 2067 656e 6520  SA of each gene 
+00013eb0: 6661 6d69 6c79 2061 6e64 2074 6865 2067  family and the g
+00013ec0: 656e 6520 7472 6565 2069 6e66 6572 656e  ene tree inferen
+00013ed0: 6365 2062 6173 6564 206f 6e20 6561 6368  ce based on each
+00013ee0: 204d 5341 2c20 616e 6420 7468 656e 2069   MSA, and then i
+00013ef0: 6e66 6572 2074 6865 2073 7065 6369 6573  nfer the species
+00013f00: 2074 7265 6520 6261 7365 6420 6f6e 2074   tree based on t
+00013f10: 6865 7365 2069 6e64 6976 6964 7561 6c20  hese individual 
+00013f20: 6765 6e65 2074 7265 6573 2e20 5468 6520  gene trees. The 
+00013f30: 7472 6565 2d69 6e66 6572 656e 6365 2070  tree-inference p
+00013f40: 726f 6772 616d 2061 6e64 2074 6865 2061  rogram and the a
+00013f50: 7373 6f63 6961 7465 6420 7061 7261 6d65  ssociated parame
+00013f60: 7465 7273 2063 616e 2062 6520 7365 7420  ters can be set 
+00013f70: 6a75 7374 2061 7320 6162 6f76 6520 6279  just as above by
+00013f80: 2060 2d74 7265 6560 206f 7220 602d 2d74   `-tree` or `--t
+00013f90: 7265 655f 6d65 7468 6f64 6020 616e 6420  ree_method` and 
+00013fa0: 602d 7473 6020 6f72 2060 2d2d 7472 6565  `-ts` or `--tree
+00013fb0: 7365 7460 2e20 506c 6561 7365 2061 6c73  set`. Please als
+00013fc0: 6f20 6d61 6b65 2073 7572 6520 7468 6520  o make sure the 
+00013fd0: 6368 6f73 656e 2074 7265 652d 696e 6665  chosen tree-infe
+00013fe0: 7265 6e63 6520 7072 6f67 7261 6d20 6973  rence program is
+00013ff0: 2069 6e73 7461 6c6c 6564 2069 6e20 7468   installed in th
+00014000: 6520 656e 7669 726f 6e6d 656e 7420 7061  e environment pa
+00014010: 7468 2e20 5468 6520 7072 6f67 7261 6d20  th. The program 
+00014020: 6061 7374 7261 6c2d 7072 6f60 2069 7320  `astral-pro` is 
+00014030: 7265 7175 6972 6564 2074 6f20 6265 2069  required to be i
+00014040: 6e73 7461 6c6c 6564 2069 6e20 7468 6520  nstalled in the 
+00014050: 656e 7669 726f 6e6d 656e 7420 7061 7468  environment path
+00014060: 2069 6620 7468 6520 636f 616c 6573 6365   if the coalesce
+00014070: 6e63 652d 6261 7365 6420 6d65 7468 6f64  nce-based method
+00014080: 2069 7320 6368 6f73 656e 2e20 5468 6520   is chosen. The 
+00014090: 7061 7261 6c6c 656c 2074 6872 6561 6473  parallel threads
+000140a0: 2068 6572 6520 6172 6520 746f 2061 6363   here are to acc
+000140b0: 656c 6572 6174 6520 7468 6520 7365 7175  elerate the sequ
+000140c0: 656e 6365 2061 6c69 676e 6d65 6e74 2061  ence alignment a
+000140d0: 6e64 2067 656e 6520 7472 6565 2069 6e66  nd gene tree inf
+000140e0: 6572 656e 6365 2066 6f72 2065 6163 6820  erence for each 
+000140f0: 6765 6e65 2066 616d 696c 7920 746f 6f20  gene family too 
+00014100: 616e 6420 7468 7573 2073 7567 6765 7374  and thus suggest
+00014110: 6564 2074 6f20 6265 2073 6574 2061 7320  ed to be set as 
+00014120: 6d75 6368 2061 7320 7468 6520 6e75 6d62  much as the numb
+00014130: 6572 206f 6620 6765 6e65 2066 616d 696c  er of gene famil
+00014140: 6965 732e 0a0a 2a2a 4120 7375 6767 6573  ies...**A sugges
+00014150: 7465 6420 7374 6172 7469 6e67 2072 756e  ted starting run
+00014160: 206f 6620 7468 6973 2061 6e61 6c79 7369   of this analysi
+00014170: 7320 6973 2077 6974 6820 7468 6520 7369  s is with the si
+00014180: 6d70 6c65 2063 6f6d 6d61 6e64 2062 656c  mple command bel
+00014190: 6f77 2e2a 2a0a 0a60 6060 0a77 6764 2066  ow.**..```.wgd f
+000141a0: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
+000141b0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+000141c0: 3220 7365 7175 656e 6365 3320 2d2d 636f  2 sequence3 --co
+000141d0: 6e63 6174 656e 6174 696f 6e0a 7767 6420  ncatenation.wgd 
+000141e0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
+000141f0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
+00014200: 6532 2073 6571 7565 6e63 6533 202d 2d63  e2 sequence3 --c
+00014210: 6f61 6c65 7363 656e 6365 0a60 6060 0a0a  oalescence.```..
+00014220: 2a2a 5468 6520 6675 6e63 7469 6f6e 616c  **The functional
+00014230: 2061 6e6e 6f74 6174 696f 6e20 6f66 2067   annotation of g
+00014240: 656e 6520 6661 6d69 6c69 6573 2a2a 0a60  ene families**.`
+00014250: 6060 0a77 6764 2066 6f63 7573 2066 616d  ``.wgd focus fam
+00014260: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+00014270: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+00014280: 6365 3320 2d2d 616e 6e6f 7461 7469 6f6e  ce3 --annotation
+00014290: 2065 6767 6e6f 6720 2d65 6420 6564 6461   eggnog -ed edda
+000142a0: 7461 202d 2d64 6d6e 6220 6462 6461 7461  ta --dmnb dbdata
+000142b0: 0a60 6060 0a0a 5468 6520 7072 6f67 7261  .```..The progra
+000142c0: 6d20 6077 6764 2066 6f63 7573 6020 616c  m `wgd focus` al
+000142d0: 736f 2061 6464 6564 2073 6f6d 6520 7772  so added some wr
+000142e0: 6170 7069 6e67 2066 756e 6374 696f 6e73  apping functions
+000142f0: 2066 6f72 2066 756e 6374 696f 6e61 6c20   for functional 
+00014300: 616e 6e6f 7461 7469 6f6e 206f 6620 6765  annotation of ge
+00014310: 6e65 2066 616d 696c 6965 7320 6f6e 2074  ne families on t
+00014320: 6865 2068 6f6f 6420 6f66 2064 6174 6162  he hood of datab
+00014330: 6173 6573 2061 6e64 2073 6f66 7477 6172  ases and softwar
+00014340: 6573 2066 6f72 2069 6e73 7461 6e63 6520  es for instance 
+00014350: 6045 6767 4e4f 4760 2061 6e64 2060 4567  `EggNOG` and `Eg
+00014360: 674e 4f47 2d6d 6170 7065 7260 2e20 466f  gNOG-mapper`. Fo
+00014370: 7220 7468 6520 616e 6e6f 7461 7469 6f6e  r the annotation
+00014380: 2075 7369 6e67 2060 4567 674e 4f47 2d6d   using `EggNOG-m
+00014390: 6170 7065 7260 2c20 7573 6572 7320 6e65  apper`, users ne
+000143a0: 6564 2074 6f20 7072 6f76 6964 6520 7468  ed to provide th
+000143b0: 6520 7061 7468 2074 6f20 7468 6520 6567  e path to the eg
+000143c0: 674e 4f47 2061 6e6e 6f74 6174 696f 6e20  gNOG annotation 
+000143d0: 6461 7461 6261 7365 2076 6961 2074 6865  database via the
+000143e0: 206f 7074 696f 6e20 602d 6564 602c 2074   option `-ed`, t
+000143f0: 6865 2070 6174 6820 746f 2074 6865 2064  he path to the d
+00014400: 6961 6d6f 6e64 2d63 6f6d 7061 7469 626c  iamond-compatibl
+00014410: 6520 6461 7461 6261 7365 2076 6961 2074  e database via t
+00014420: 6865 206f 7074 696f 6e20 602d 2d64 6d6e  he option `--dmn
+00014430: 645f 6462 6020 616e 6420 7468 6520 6f70  d_db` and the op
+00014440: 7469 6f6e 2060 2d2d 616e 6e6f 7461 7469  tion `--annotati
+00014450: 6f6e 6020 7365 7420 6173 2022 6567 676e  on` set as "eggn
+00014460: 6f67 222e 2054 6865 206d 616e 6e65 7220  og". The manner 
+00014470: 686f 7720 5046 414d 2061 6e6e 6f74 6174  how PFAM annotat
+00014480: 696f 6e20 7769 6c6c 2062 6520 7065 7266  ion will be perf
+00014490: 6f72 6d65 6420 6361 6e20 6265 2063 6f6e  ormed can be con
+000144a0: 7472 6f6c 6c65 6420 7669 6120 7468 6520  trolled via the 
+000144b0: 6f70 7469 6f6e 2060 7066 616d 602c 2065  option `pfam`, e
+000144c0: 6974 6865 7220 226e 6f6e 6522 2c20 2272  ither "none", "r
+000144d0: 6561 6c69 676e 2220 6f72 2022 6465 6e6f  ealign" or "deno
+000144e0: 766f 222c 2074 6865 2064 6574 6169 6c65  vo", the detaile
+000144f0: 6420 6578 706c 616e 6174 696f 6e20 6361  d explanation ca
+00014500: 6e20 6265 2066 6f75 6e64 2061 7420 7468  n be found at th
+00014510: 6520 7769 6b69 206f 6620 5b45 6767 4e4f  e wiki of [EggNO
+00014520: 472d 6d61 7070 6572 5d28 6874 7470 733a  G-mapper](https:
+00014530: 2f2f 6769 7468 7562 2e63 6f6d 2f65 6767  //github.com/egg
+00014540: 6e6f 6764 622f 6567 676e 6f67 2d6d 6170  nogdb/eggnog-map
+00014550: 7065 722f 7769 6b69 292e 2050 6c65 6173  per/wiki). Pleas
+00014560: 6520 7072 652d 696e 7374 616c 6c20 7468  e pre-install th
+00014570: 6520 6045 6767 4e4f 472d 6d61 7070 6572  e `EggNOG-mapper
+00014580: 6020 7079 7468 6f6e 2070 6163 6b61 6765  ` python package
+00014590: 2069 6620 7573 696e 6720 7468 6973 2066   if using this f
+000145a0: 756e 6374 696f 6e2e 2046 6f72 2074 6865  unction. For the
+000145b0: 2061 6e6e 6f74 6174 696f 6e20 7573 696e   annotation usin
+000145c0: 6720 6068 6d6d 7363 616e 602c 2077 6861  g `hmmscan`, wha
+000145d0: 7420 6973 2069 6d70 6c65 6d65 6e74 6564  t is implemented
+000145e0: 2069 6e20 6077 6764 2076 3260 2069 7320   in `wgd v2` is 
+000145f0: 6120 7369 6d70 6c65 2062 756e 646c 6520  a simple bundle 
+00014600: 6675 6e63 7469 6f6e 2074 6f20 7065 7266  function to perf
+00014610: 6f72 6d20 6068 6d6d 7363 616e 6020 616e  orm `hmmscan` an
+00014620: 616c 7973 6973 2066 6f72 2061 2067 6976  alysis for a giv
+00014630: 656e 2068 6d6d 2070 726f 6669 6c65 2061  en hmm profile a
+00014640: 6e64 2073 6574 206f 6620 6765 6e65 2066  nd set of gene f
+00014650: 616d 696c 6965 7320 7375 6368 2074 6861  amilies such tha
+00014660: 7420 7573 6572 7320 6e65 6564 2074 6f20  t users need to 
+00014670: 7365 7420 7468 6520 6f70 7469 6f6e 2060  set the option `
+00014680: 2d2d 616e 6e6f 7461 7469 6f6e 6020 6173  --annotation` as
+00014690: 2022 686d 6d70 6661 6d22 2061 6e64 2070   "hmmpfam" and p
+000146a0: 726f 7669 6465 2074 6865 2068 6d6d 7072  rovide the hmmpr
+000146b0: 6f66 696c 6520 7669 6120 7468 6520 6f70  ofile via the op
+000146c0: 7469 6f6e 2060 2d2d 686d 6d60 2e20 466f  tion `--hmm`. Fo
+000146d0: 7220 7468 6520 616e 6e6f 7461 7469 6f6e  r the annotation
+000146e0: 2075 7369 6e67 2060 696e 7465 7270 726f   using `interpro
+000146f0: 7363 616e 602c 2075 7365 7273 206e 6565  scan`, users nee
+00014700: 6420 746f 2070 726f 7669 6465 2074 6865  d to provide the
+00014710: 2070 6174 6820 746f 2074 6865 2069 6e74   path to the int
+00014720: 6572 7072 6f73 6361 6e20 696e 7374 616c  erproscan instal
+00014730: 6c61 7469 6f6e 2066 6f6c 6465 7220 7768  lation folder wh
+00014740: 6572 6520 7468 6572 6520 6973 2061 2060  ere there is a `
+00014750: 696e 7465 7270 726f 7363 616e 2e73 6860  interproscan.sh`
+00014760: 2066 696c 6520 7669 6120 7468 6520 6f70   file via the op
+00014770: 7469 6f6e 2060 2d2d 6578 6570 6174 6860  tion `--exepath`
+00014780: 2061 6e64 2073 6574 2074 6865 2074 6865   and set the the
+00014790: 206f 7074 696f 6e20 602d 2d61 6e6e 6f74   option `--annot
+000147a0: 6174 696f 6e60 2061 7320 2269 6e74 6572  ation` as "inter
+000147b0: 7072 6f73 6361 6e22 2e20 5468 6520 7061  proscan". The pa
+000147c0: 7261 6c6c 656c 2074 6872 6561 6473 2068  rallel threads h
+000147d0: 6572 6520 6172 6520 746f 2070 6172 616c  ere are to paral
+000147e0: 6c65 6c69 7a65 2074 6865 2061 6e61 6c79  lelize the analy
+000147f0: 7369 7320 666f 7220 6561 6368 2067 656e  sis for each gen
+00014800: 6520 6661 6d69 6c79 2061 6e64 2074 6875  e family and thu
+00014810: 7320 7375 6767 6573 7465 6420 746f 2062  s suggested to b
+00014820: 6520 7365 7420 6173 206d 7563 6820 6173  e set as much as
+00014830: 2074 6865 206e 756d 6265 7220 6f66 2067   the number of g
+00014840: 656e 6520 6661 6d69 6c69 6573 2e0a 0a2a  ene families...*
+00014850: 2a41 2073 7567 6765 7374 6564 2073 7461  *A suggested sta
+00014860: 7274 696e 6720 7275 6e20 6f66 2074 6869  rting run of thi
+00014870: 7320 616e 616c 7973 6973 2063 616e 2062  s analysis can b
+00014880: 6520 7769 7468 2074 6865 2063 6f6d 6d61  e with the comma
+00014890: 6e64 2062 656c 6f77 2e2a 2a0a 0a60 6060  nd below.**..```
+000148a0: 0a77 6764 2066 6f63 7573 2066 616d 696c  .wgd focus famil
+000148b0: 6965 7320 7365 7175 656e 6365 3120 7365  ies sequence1 se
+000148c0: 7175 656e 6365 3220 7365 7175 656e 6365  quence2 sequence
+000148d0: 3320 2d2d 616e 6e6f 7461 7469 6f6e 2065  3 --annotation e
+000148e0: 6767 6e6f 6720 2d65 6420 6564 6461 7461  ggnog -ed eddata
+000148f0: 202d 2d64 6d6e 6220 6462 6461 7461 0a77   --dmnb dbdata.w
+00014900: 6764 2066 6f63 7573 2066 616d 696c 6965  gd focus familie
+00014910: 7320 7365 7175 656e 6365 3120 7365 7175  s sequence1 sequ
+00014920: 656e 6365 3220 7365 7175 656e 6365 3320  ence2 sequence3 
+00014930: 2d2d 616e 6e6f 7461 7469 6f6e 2068 6d6d  --annotation hmm
+00014940: 7066 616d 202d 2d68 6d6d 2068 6d6d 6461  pfam --hmm hmmda
+00014950: 7461 0a77 6764 2066 6f63 7573 2066 616d  ta.wgd focus fam
+00014960: 696c 6965 7320 7365 7175 656e 6365 3120  ilies sequence1 
+00014970: 7365 7175 656e 6365 3220 7365 7175 656e  sequence2 sequen
+00014980: 6365 3320 2d2d 616e 6e6f 7461 7469 6f6e  ce3 --annotation
+00014990: 2069 6e74 6572 7072 6f73 6361 6e20 2d2d   interproscan --
+000149a0: 6578 6570 6174 6820 2450 4154 480a 6060  exepath $PATH.``
+000149b0: 600a 0a2a 2a54 6865 2070 6879 6c6f 6765  `..**The phyloge
+000149c0: 6e65 7469 6320 6461 7469 6e67 206f 6620  netic dating of 
+000149d0: 5747 4473 2a2a 0a60 6060 0a77 6764 2066  WGDs**.```.wgd f
+000149e0: 6f63 7573 2066 616d 696c 6965 7320 7365  ocus families se
+000149f0: 7175 656e 6365 3120 7365 7175 656e 6365  quence1 sequence
+00014a00: 3220 7365 7175 656e 6365 3320 2d64 206d  2 sequence3 -d m
+00014a10: 636d 6374 7265 6520 2d73 7020 7370 6461  cmctree -sp spda
+00014a20: 7461 2028 2d2d 7072 6f74 636f 6364 6174  ta (--protcocdat
+00014a30: 696e 6729 2028 2d2d 7061 7274 6974 696f  ing) (--partitio
+00014a40: 6e29 2028 2d2d 6161 6d6f 6465 6c20 6c67  n) (--aamodel lg
+00014a50: 2920 282d 6473 2027 6275 726e 696e 203d  ) (-ds 'burnin =
+00014a60: 2032 3030 3027 2920 282d 6473 2027 7361   2000') (-ds 'sa
+00014a70: 6d70 6672 6571 203d 2031 3030 3027 2920  mpfreq = 1000') 
+00014a80: 282d 6473 2027 6e73 616d 706c 6520 3d20  (-ds 'nsample = 
+00014a90: 3230 3030 3027 2920 2028 2d6e 2034 2920  20000')  (-n 4) 
+00014aa0: 282d 2d74 6f5f 7374 6f70 2920 282d 2d63  (--to_stop) (--c
+00014ab0: 6473 2920 282d 6f20 7767 645f 666f 6375  ds) (-o wgd_focu
+00014ac0: 7329 2028 2d74 2077 6f72 6b69 6e67 5f74  s) (-t working_t
+00014ad0: 6d70 290a 6060 600a 0a54 6865 2061 6273  mp).```..The abs
+00014ae0: 6f6c 7574 6520 6461 7469 6e67 206f 6620  olute dating of 
+00014af0: 5747 4473 2069 7320 6120 7370 6563 6966  WGDs is a specif
+00014b00: 6963 2070 6970 656c 696e 6520 696d 706c  ic pipeline impl
+00014b10: 656d 656e 7465 6420 696e 2060 7767 6420  emented in `wgd 
+00014b20: 7632 6020 7573 696e 6720 7468 6520 6d65  v2` using the me
+00014b30: 7468 6f64 206f 6620 7068 796c 6f67 656e  thod of phylogen
+00014b40: 6574 6963 2064 6174 696e 672e 2054 6865  etic dating. The
+00014b50: 2066 616d 696c 6965 7320 7573 6564 2069   families used i
+00014b60: 6e20 7468 6973 2073 7465 7020 6361 6e20  n this step can 
+00014b70: 6265 2070 726f 6475 6365 6420 6672 6f6d  be produced from
+00014b80: 2060 7767 6420 646d 6460 2061 6e64 2060   `wgd dmd` and `
+00014b90: 7767 6420 7065 616b 602e 204e 6f74 6520  wgd peak`. Note 
+00014ba0: 7468 6174 2068 6572 6520 7765 206f 6e6c  that here we onl
+00014bb0: 7920 6469 7363 7573 7320 686f 7720 746f  y discuss how to
+00014bc0: 2064 6174 6520 5747 4473 2077 6974 6820   date WGDs with 
+00014bd0: 6765 6e6f 6d65 2061 7373 656d 626c 792c  genome assembly,
+00014be0: 2069 6e73 7465 6164 206f 6620 7472 616e   instead of tran
+00014bf0: 7363 7269 7074 6f6d 6520 6173 7365 6d62  scriptome assemb
+00014c00: 6c79 2028 7768 6963 6820 7769 6c6c 2062  ly (which will b
+00014c10: 6520 6469 7363 7573 7365 6420 696e 2061  e discussed in a
+00014c20: 2073 6570 6172 6174 6520 7365 6374 696f   separate sectio
+00014c30: 6e20 6865 7265 756e 6465 7229 2e20 5468  n hereunder). Th
+00014c40: 6520 6173 7375 6d70 7469 6f6e 2077 6520  e assumption we 
+00014c50: 6d61 6465 2068 6572 6520 6973 2074 6861  made here is tha
+00014c60: 7420 6e6f 7420 616c 6c20 616e 6368 6f72  t not all anchor
+00014c70: 2070 6169 7273 2028 636f 6c6c 696e 6561   pairs (collinea
+00014c80: 7220 6475 706c 6963 6174 6573 2920 6172  r duplicates) ar
+00014c90: 6520 7375 6974 6162 6c65 2066 6f72 2070  e suitable for p
+00014ca0: 6879 6c6f 6765 6e65 7469 6320 6461 7469  hylogenetic dati
+00014cb0: 6e67 2c20 666f 7220 696e 7374 616e 6365  ng, for instance
+00014cc0: 2074 686f 7365 2066 6173 746c 7920 6f72   those fastly or
+00014cd0: 2073 6c6f 776c 7920 6576 6f6c 7669 6e67   slowly evolving
+00014ce0: 2067 656e 6520 6475 706c 6963 6174 6573   gene duplicates
+00014cf0: 2c20 6265 6361 7573 6520 7468 6579 2772  , because they'r
+00014d00: 6520 7072 6f6e 6520 746f 2067 6976 6520  e prone to give 
+00014d10: 6269 6173 6564 2064 6174 696e 6720 6573  biased dating es
+00014d20: 7469 6d61 7469 6f6e 2e20 536f 2061 7320  timation. So as 
+00014d30: 746f 2072 6574 7269 6576 6520 7468 6520  to retrieve the 
+00014d40: 2272 656c 6961 626c 6522 2061 6e63 686f  "reliable" ancho
+00014d50: 7220 7061 6972 732c 2077 6520 696d 706c  r pairs, we impl
+00014d60: 656d 656e 7465 6420 736f 6d65 206d 6574  emented some met
+00014d70: 686f 6473 206f 6620 6964 656e 7469 6679  hods of identify
+00014d80: 696e 6720 6372 6564 6961 626c 6520 616e  ing crediable an
+00014d90: 6368 6f72 2070 6169 7273 2062 6173 6564  chor pairs based
+00014da0: 206f 6e20 7468 6569 7220 4b73 2076 616c   on their Ks val
+00014db0: 7565 7320 616e 642f 6f72 2072 6573 6964  ues and/or resid
+00014dc0: 696e 6720 636f 6c6c 696e 6561 7220 7365  ing collinear se
+00014dd0: 676d 656e 7473 2e20 466f 7220 6120 6765  gments. For a ge
+00014de0: 6e6f 6d65 2077 6974 6820 636c 6561 7220  nome with clear 
+00014df0: 7369 676e 616c 7320 6f66 2070 7574 6174  signals of putat
+00014e00: 6976 6520 5747 4473 2c20 7375 6368 2061  ive WGDs, such a
+00014e10: 7320 7468 6520 2a41 7175 696c 6567 6961  s the *Aquilegia
+00014e20: 2063 6f65 7275 6c65 612a 2069 6e20 7468   coerulea* in th
+00014e30: 6520 6578 616d 706c 652c 2061 2068 6575  e example, a heu
+00014e40: 7269 7374 6963 206d 6574 686f 6420 7468  ristic method th
+00014e50: 6174 2061 7070 6c69 6573 2074 6865 2070  at applies the p
+00014e60: 7269 6e63 6970 6c65 206f 6620 686f 7720  rinciple of how 
+00014e70: 5b6b 7372 6174 6573 5d28 6874 7470 733a  [ksrates](https:
+00014e80: 2f2f 6769 7468 7562 2e63 6f6d 2f56 4942  //github.com/VIB
+00014e90: 2d50 5342 2f6b 7372 6174 6573 2920 6669  -PSB/ksrates) fi
+00014ea0: 6e64 2074 6865 2069 6e69 7469 616c 2070  nd the initial p
+00014eb0: 6561 6b73 2061 6e64 2074 6865 6972 2070  eaks and their p
+00014ec0: 6172 616d 6574 6572 7320 7761 7320 696d  arameters was im
+00014ed0: 706c 656d 656e 7465 6420 746f 2066 696e  plemented to fin
+00014ee0: 6420 7468 6520 3935 2520 636f 6e66 6964  d the 95% confid
+00014ef0: 656e 6365 206c 6576 656c 206f 6620 7468  ence level of th
+00014f00: 6520 6173 7375 6d65 6420 6c6f 676e 6f72  e assumed lognor
+00014f10: 6d61 6c20 6469 7374 7269 6275 7469 6f6e  mal distribution
+00014f20: 206f 6620 7468 6520 616e 6368 6f72 2070   of the anchor p
+00014f30: 6169 7220 2a4b 2a3c 7375 623e 533c 2f73  air *K*<sub>S</s
+00014f40: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00014f50: 7469 6f6e 2074 6f20 6669 6c74 6572 2061  tion to filter a
+00014f60: 6e63 686f 7220 7061 6972 7320 7769 7468  nchor pairs with
+00014f70: 2074 6f6f 2068 6967 6820 6f72 206c 6f77   too high or low
+00014f80: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00014f90: 2061 6765 732c 2074 6865 2065 7861 6d70   ages, the examp
+00014fa0: 6c61 7220 636f 6d6d 616e 6420 6f66 2077  lar command of w
+00014fb0: 6869 6368 2069 7320 7368 6f77 6564 2069  hich is showed i
+00014fc0: 6e20 7468 6520 6077 6764 2070 6561 6b60  n the `wgd peak`
+00014fd0: 2073 6563 7469 6f6e 206f 6620 5b49 6c6c   section of [Ill
+00014fe0: 7573 7472 6174 696f 6e5d 2868 7474 7073  ustration](https
+00014ff0: 3a2f 2f67 6974 6875 622e 636f 6d2f 6865  ://github.com/he
+00015000: 6368 652d 7073 622f 7767 643f 7461 623d  che-psb/wgd?tab=
+00015010: 7265 6164 6d65 2d6f 762d 6669 6c65 2369  readme-ov-file#i
+00015020: 6c6c 7573 7472 6174 696f 6e29 2e20 4966  llustration). If
+00015030: 2074 6869 7320 6865 7572 6973 7469 6320   this heuristic 
+00015040: 6d65 7468 6f64 2066 6169 6c65 6420 746f  method failed to
+00015050: 2067 6976 6520 7265 6173 6f6e 6162 6c65   give reasonable
+00015060: 2072 6573 756c 7473 2c20 7768 6963 6820   results, which 
+00015070: 6973 2075 7375 616c 6c79 2064 7565 2074  is usually due t
+00015080: 6f20 7468 6520 6566 6665 6374 206f 6620  o the effect of 
+00015090: 6d75 6c74 6970 6c65 2061 646a 6163 656e  multiple adjacen
+000150a0: 7420 5747 4473 2074 6861 7420 626c 7572  t WGDs that blur
+000150b0: 7320 7468 6520 7065 616b 2d66 696e 6469  s the peak-findi
+000150c0: 6e67 2070 726f 6365 7373 2c20 7573 6572  ng process, user
+000150d0: 7320 6361 6e20 7475 726e 2074 6f20 7468  s can turn to th
+000150e0: 6520 636f 6c6c 696e 6561 7220 7365 676d  e collinear segm
+000150f0: 656e 7473 2d67 7569 6465 6420 616e 6368  ents-guided anch
+00015100: 6f72 2070 6169 7220 636c 7573 7465 7269  or pair clusteri
+00015110: 6e67 2069 6d70 6c65 6d65 6e74 6564 2069  ng implemented i
+00015120: 6e20 6077 6764 2076 3260 2c20 696e 2077  n `wgd v2`, in w
+00015130: 6869 6368 2061 2063 6f6c 6c69 6e65 6172  hich a collinear
+00015140: 2073 6567 6d65 6e74 2d77 6973 6520 474d   segment-wise GM
+00015150: 4d20 636c 7573 7465 7269 6e67 2077 696c  M clustering wil
+00015160: 6c20 6265 2066 6972 7374 2063 6f6e 6475  l be first condu
+00015170: 6374 6564 2062 6173 6564 206f 6e20 7468  cted based on th
+00015180: 6520 2273 6f2d 6361 6c6c 6564 2220 7365  e "so-called" se
+00015190: 676d 656e 7420 2a4b 2a3c 7375 623e 533c  gment *K*<sub>S<
+000151a0: 2f73 7562 3e20 6167 6520 7265 7072 6573  /sub> age repres
+000151b0: 656e 7465 6420 6279 2074 6865 206d 6564  ented by the med
+000151c0: 6961 6e20 2a4b 2a3c 7375 623e 533c 2f73  ian *K*<sub>S</s
+000151d0: 7562 3e20 6167 6520 6f66 2061 6c6c 2074  ub> age of all t
+000151e0: 6865 2072 6573 6964 696e 6720 7379 6e74  he residing synt
+000151f0: 656c 6f67 7320 286e 6f74 6520 7468 6174  elogs (note that
+00015200: 2074 6865 2067 656e 6520 6475 706c 6963   the gene duplic
+00015210: 6174 6520 7061 6972 7320 6164 6f70 7465  ate pairs adopte
+00015220: 6420 696e 2074 6869 7320 7374 6570 2069  d in this step i
+00015230: 7320 6672 6f6d 2074 6865 2066 696c 6520  s from the file 
+00015240: 606d 756c 7469 706c 6963 6f6e 5f70 6169  `multiplicon_pai
+00015250: 7273 2e74 7874 6020 7768 6963 6820 636f  rs.txt` which co
+00015260: 6e74 6169 6e73 2074 6865 2066 756c 6c20  ntains the full 
+00015270: 7365 7420 6f66 2073 796e 7465 6c6f 6773  set of syntelogs
+00015280: 292c 2069 6e73 7465 6164 206f 6620 7468  ), instead of th
+00015290: 6520 736d 616c 6c65 7220 6765 6e65 2073  e smaller gene s
+000152a0: 6574 206f 6620 616e 6368 6f72 2070 6169  et of anchor pai
+000152b0: 7273 2e20 5468 6520 6469 7374 696e 6374  rs. The distinct
+000152c0: 696f 6e20 6265 7477 6565 6e20 616e 6368  ion between anch
+000152d0: 6f72 2070 6169 7273 2061 6e64 2073 796e  or pairs and syn
+000152e0: 7465 6c6f 6773 2069 7320 7468 6174 2074  telogs is that t
+000152f0: 6865 206c 6174 7465 7220 7265 6665 7273  he latter refers
+00015300: 2074 6f20 6d75 6c74 6970 6c65 2073 6574   to multiple set
+00015310: 7320 6f66 2067 656e 6573 2064 6572 6976  s of genes deriv
+00015320: 6564 2066 726f 6d20 7468 6520 7361 6d65  ed from the same
+00015330: 2061 6e63 6573 7472 616c 2067 656e 6f6d   ancestral genom
+00015340: 6963 2072 6567 696f 6e20 7768 696c 6520  ic region while 
+00015350: 7468 6520 666f 726d 6572 2069 6d70 6c69  the former impli
+00015360: 6573 2074 6865 206c 6174 7465 7220 6275  es the latter bu
+00015370: 7420 7265 7175 6972 6573 2061 6464 6974  t requires addit
+00015380: 696f 6e61 6c6c 7920 7468 6520 636f 6e73  ionally the cons
+00015390: 6572 7665 6420 6765 6e65 206f 7264 6572  erved gene order
+000153a0: 2c20 626f 7468 206f 6620 7768 6963 6820  , both of which 
+000153b0: 6172 652c 2077 6974 6869 6e20 6120 6765  are, within a ge
+000153c0: 6e6f 6d65 2c20 6173 7375 6d65 6420 746f  nome, assumed to
+000153d0: 2062 6520 6f72 6967 696e 6174 6564 2066   be originated f
+000153e0: 726f 6d20 7468 6520 6475 706c 6963 6174  rom the duplicat
+000153f0: 696f 6e20 6f66 2061 2063 6f6d 6d6f 6e20  ion of a common 
+00015400: 616e 6365 7374 7261 6c20 6765 6e6f 6d69  ancestral genomi
+00015410: 6320 7265 6769 6f6e 2061 6e64 2061 7320  c region and as 
+00015420: 7375 6368 2064 6565 6d65 6420 6576 6964  such deemed evid
+00015430: 656e 6365 2066 6f72 2057 4744 2e20 5468  ence for WGD. Th
+00015440: 656e 2074 6865 2073 796e 7465 6c6f 6773  en the syntelogs
+00015450: 2077 696c 6c20 6265 206d 6170 7065 6420   will be mapped 
+00015460: 6261 636b 2061 6363 6f72 6469 6e67 2074  back according t
+00015470: 6f20 7468 6520 636c 7573 7465 7269 6e67  o the clustering
+00015480: 2072 6573 756c 7473 206f 6620 7468 6569   results of thei
+00015490: 7220 6166 6669 6c69 6174 6564 2073 6567  r affiliated seg
+000154a0: 6d65 6e74 732e 2053 696e 6365 2074 6865  ments. Since the
+000154b0: 2047 6175 7373 6961 6e20 7368 6170 6520   Gaussian shape 
+000154c0: 6f66 2074 6865 2073 6567 6d65 6e74 2063  of the segment c
+000154d0: 6c75 7374 6572 2064 6f65 736e 2774 206e  luster doesn't n
+000154e0: 6563 6573 7361 7269 6c79 2069 6d69 7461  ecessarily imita
+000154f0: 7465 2074 6865 2073 6861 7065 206f 6620  te the shape of 
+00015500: 7468 6520 7265 7369 6469 6e67 2073 796e  the residing syn
+00015510: 7465 6c6f 6773 2c20 736f 2061 7320 746f  telogs, so as to
+00015520: 2072 6574 7269 6576 6520 7468 6520 2272   retrieve the "r
+00015530: 656c 6961 626c 6522 2067 656e 6520 7365  eliable" gene se
+00015540: 7473 2066 6f72 2070 6879 6c6f 6765 6e65  ts for phylogene
+00015550: 7469 6320 6461 7469 6e67 2c20 7765 2061  tic dating, we a
+00015560: 6470 6f74 6564 2074 6865 2028 3935 2529  dpoted the (95%)
+00015570: 2068 6967 6865 7374 2064 656e 7369 7479   highest density
+00015580: 2072 6567 696f 6e20 2848 4452 2920 6f66   region (HDR) of
+00015590: 2074 6865 2073 796e 7465 6c6f 6720 2a4b   the syntelog *K
+000155a0: 2a3c 7375 623e 533c 2f73 7562 3e20 6167  *<sub>S</sub> ag
+000155b0: 6520 6469 7374 7269 6275 7469 6f6e 2066  e distribution f
+000155c0: 6f72 2074 6865 2070 6879 6c6f 6765 6e65  or the phylogene
+000155d0: 7469 6320 6461 7469 6e67 2c20 7468 6520  tic dating, the 
+000155e0: 6361 6c63 756c 6174 696f 6e20 6f66 2074  calculation of t
+000155f0: 6865 2028 3935 2529 2048 4452 2069 6e20  he (95%) HDR in 
+00015600: 7468 6520 6675 6e63 7469 6f6e 205b 6361  the function [ca
+00015610: 6c63 756c 6174 6548 5044 5d28 6874 7470  lculateHPD](http
+00015620: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f68  s://github.com/h
+00015630: 6563 6865 2d70 7362 2f77 6764 2f62 6c6f  eche-psb/wgd/blo
+00015640: 622f 7068 796c 6f64 6174 696e 672f 7767  b/phylodating/wg
+00015650: 642f 7065 616b 2e70 7923 4c31 3735 3729  d/peak.py#L1757)
+00015660: 2073 6565 6b73 2074 6865 2073 686f 7274   seeks the short
+00015670: 6573 7420 2a4b 2a3c 7375 623e 533c 2f73  est *K*<sub>S</s
+00015680: 7562 3e20 7261 6e67 6520 2861 2c62 2920  ub> range (a,b) 
+00015690: 7768 6963 6820 7361 7469 7366 6965 7320  which satisfies 
+000156a0: 7468 6520 7265 7175 6972 656d 656e 7420  the requirement 
+000156b0: 6f66 2073 7061 6e6e 696e 6720 6d6f 7265  of spanning more
+000156c0: 2074 6861 6e20 2839 3525 2920 6f66 2061   than (95%) of a
+000156d0: 6c6c 2074 6865 202a 4b2a 3c73 7562 3e53  ll the *K*<sub>S
+000156e0: 3c2f 7375 623e 2076 616c 7565 732e 204f  </sub> values. O
+000156f0: 6e20 7468 6520 7072 656d 6973 6520 6f66  n the premise of
+00015700: 2069 6465 6e74 6966 6965 6420 616e 6368   identified anch
+00015710: 6f72 2070 6169 7273 2028 6e6f 7465 2074  or pairs (note t
+00015720: 6861 7420 7468 6520 7379 6e74 656c 6f67  hat the syntelog
+00015730: 7320 6172 6520 616c 736f 2072 6566 6572  s are also refer
+00015740: 7265 6420 6173 2061 6e63 686f 7220 7061  red as anchor pa
+00015750: 6972 7320 6865 7265 6166 7465 7229 2c20  irs hereafter), 
+00015760: 7765 2069 6d70 6c65 6d65 6e74 6564 2069  we implemented i
+00015770: 6e20 7468 6520 7072 6f67 7261 6d20 6077  n the program `w
+00015780: 6764 2064 6d64 6020 7468 6520 736f 2d63  gd dmd` the so-c
+00015790: 616c 6c65 6420 616e 6368 6f72 2d61 7761  alled anchor-awa
+000157a0: 7265 206c 6f63 616c 204d 5242 4873 206f  re local MRBHs o
+000157b0: 7220 6f72 7468 6f67 726f 7570 732c 2069  r orthogroups, i
+000157c0: 6e20 7768 6963 6820 7468 6520 6f72 6967  n which the orig
+000157d0: 696e 616c 206c 6f63 616c 204d 5242 4873  inal local MRBHs
+000157e0: 2061 7265 2066 7572 7468 6572 206d 6572   are further mer
+000157f0: 6765 6420 7769 7468 2061 6e63 686f 7220  ged with anchor 
+00015800: 7061 6972 7320 7375 6368 2074 6861 7420  pairs such that 
+00015810: 6561 6368 206f 7274 686f 6772 6f75 7020  each orthogroup 
+00015820: 636f 6e74 6169 6e73 2074 6865 2061 6e63  contains the anc
+00015830: 686f 7220 7061 6972 2061 6e64 2074 6865  hor pair and the
+00015840: 206f 7274 686f 6c6f 6775 6573 2e20 5769   orthologues. Wi
+00015850: 7468 2074 6869 7320 6f72 7468 6f67 726f  th this orthogro
+00015860: 7570 2c20 7573 6572 7320 7468 656e 206e  up, users then n
+00015870: 6565 6420 6f6e 6520 7374 6172 7469 6e67  eed one starting
+00015880: 2074 7265 6520 6669 6c65 2028 6173 2073   tree file (as s
+00015890: 686f 776e 2069 6e20 7468 6520 5b49 6c6c  hown in the [Ill
+000158a0: 7573 7472 6174 696f 6e5d 2868 7474 7073  ustration](https
+000158b0: 3a2f 2f67 6974 6875 622e 636f 6d2f 6865  ://github.com/he
+000158c0: 6368 652d 7073 622f 7767 642f 7472 6565  che-psb/wgd/tree
+000158d0: 2f70 6879 6c6f 6461 7469 6e67 3f74 6162  /phylodating?tab
+000158e0: 3d72 6561 646d 652d 6f76 2d66 696c 6523  =readme-ov-file#
+000158f0: 696c 6c75 7374 7261 7469 6f6e 2920 7061  illustration) pa
+00015900: 7274 2920 696e 6469 6361 7469 6e67 2074  rt) indicating t
+00015910: 6865 2074 7265 6520 746f 706f 6c6f 6779  he tree topology
+00015920: 2061 6e64 2066 6f73 7369 6c20 6361 6c69   and fossil cali
+00015930: 6272 6174 696f 6e20 696e 666f 726d 6174  bration informat
+00015940: 696f 6e20 666f 7220 7468 6520 6669 6e61  ion for the fina
+00015950: 6c20 7068 796c 6f67 656e 6574 6963 2064  l phylogenetic d
+00015960: 6174 696e 6720 7573 696e 6720 7468 6520  ating using the 
+00015970: 7072 6f67 7261 6d20 6077 6764 2066 6f63  program `wgd foc
+00015980: 7573 602e 2055 7365 7273 2063 616e 2073  us`. Users can s
+00015990: 6574 2074 6865 2066 6c61 6720 6f70 7469  et the flag opti
+000159a0: 6f6e 2060 2d2d 7072 6f74 636f 6364 6174  on `--protcocdat
+000159b0: 696e 6760 2074 6f20 6f6e 6c79 2063 6f6e  ing` to only con
+000159c0: 6475 6374 2074 6865 2064 6174 696e 6720  duct the dating 
+000159d0: 6f66 2074 6865 2063 6f6e 6361 7465 6e61  of the concatena
+000159e0: 7469 6f6e 2070 726f 7465 696e 204d 5341  tion protein MSA
+000159f0: 206f 7220 7468 6520 666c 6167 206f 7074   or the flag opt
+00015a00: 696f 6e20 602d 2d70 726f 7464 6174 696e  ion `--protdatin
+00015a10: 6760 2074 6f20 6f6e 6c79 2063 6f6e 6475  g` to only condu
+00015a20: 6374 2074 6865 2064 6174 696e 6720 6f66  ct the dating of
+00015a30: 2074 6865 2070 726f 7465 696e 204d 5341   the protein MSA
+00015a40: 732c 206e 6f74 6564 2074 6861 7420 7468  s, noted that th
+00015a50: 6573 6520 7477 6f20 6f70 7469 6f6e 7320  ese two options 
+00015a60: 6f6e 6c79 2077 6f72 6b20 666f 7220 7468  only work for th
+00015a70: 6520 606d 636d 6374 7265 6560 206f 7074  e `mcmctree` opt
+00015a80: 696f 6e20 736f 2066 6172 2e20 5468 6520  ion so far. The 
+00015a90: 666c 6167 206f 7074 696f 6e20 602d 2d70  flag option `--p
+00015aa0: 6172 7469 7469 6f6e 6020 6361 6e20 6265  artition` can be
+00015ab0: 2073 6574 2074 6f20 7065 7266 6f72 6d20   set to perform 
+00015ac0: 606d 636d 6374 7265 6560 2061 6e61 6c79  `mcmctree` analy
+00015ad0: 7369 7320 7573 696e 6720 7468 6520 7061  sis using the pa
+00015ae0: 7274 6974 696f 6e65 6420 6461 7461 2028  rtitioned data (
+00015af0: 692e 652e 2c20 3173 742c 2032 6e64 2061  i.e., 1st, 2nd a
+00015b00: 6e64 2033 7264 2070 6f73 6974 696f 6e20  nd 3rd position 
+00015b10: 6f66 2063 6f64 6f6e 2920 696e 7374 6561  of codon) instea
+00015b20: 6420 6f66 2075 7369 6e67 2074 6865 2063  d of using the c
+00015b30: 6f64 6f6e 2061 7320 6120 7768 6f6c 652e  odon as a whole.
+00015b40: 2054 6865 206f 7074 696f 6e20 602d 2d61   The option `--a
+00015b50: 616d 6f64 656c 6020 6361 6e20 6265 2073  amodel` can be s
+00015b60: 6574 2074 6f20 6465 7465 726d 696e 6520  et to determine 
+00015b70: 7468 6520 616d 696e 6f20 6163 6964 206d  the amino acid m
+00015b80: 6f64 656c 2061 7070 6c69 6564 2069 6e20  odel applied in 
+00015b90: 606d 636d 6374 7265 6560 2061 6e61 6c79  `mcmctree` analy
+00015ba0: 7369 732e 2054 6865 206f 7074 696f 6e20  sis. The option 
+00015bb0: 602d 6473 6020 6361 6e20 6265 2075 7365  `-ds` can be use
+00015bc0: 6420 746f 2073 6574 2070 6172 616d 6574  d to set paramet
+00015bd0: 6572 7320 666f 7220 7468 6520 6d6f 6c65  ers for the mole
+00015be0: 6375 6c61 7220 6461 7469 6e67 2070 726f  cular dating pro
+00015bf0: 6772 616d 2e20 5468 6520 7061 7261 6c6c  gram. The parall
+00015c00: 656c 2074 6872 6561 6473 2068 6572 6520  el threads here 
+00015c10: 6172 6520 746f 2070 6172 616c 6c65 6c69  are to paralleli
+00015c20: 7a65 2074 6865 2061 6e61 6c79 7369 7320  ze the analysis 
+00015c30: 666f 7220 6561 6368 2067 656e 6520 6661  for each gene fa
+00015c40: 6d69 6c79 2061 6e64 2074 6875 7320 7375  mily and thus su
+00015c50: 6767 6573 7465 6420 746f 2062 6520 7365  ggested to be se
+00015c60: 7420 6173 206d 7563 6820 6173 2074 6865  t as much as the
+00015c70: 206e 756d 6265 7220 6f66 2067 656e 6520   number of gene 
+00015c80: 6661 6d69 6c69 6573 2e0a 0a2a 2a41 2073  families...**A s
+00015c90: 7567 6765 7374 6564 2073 7461 7274 696e  uggested startin
+00015ca0: 6720 7275 6e20 6361 6e20 6265 2077 6974  g run can be wit
+00015cb0: 6820 7468 6520 636f 6d6d 616e 6420 6265  h the command be
+00015cc0: 6c6f 772e 2a2a 0a0a 6060 600a 7767 6420  low.**..```.wgd 
+00015cd0: 666f 6375 7320 6661 6d69 6c69 6573 2073  focus families s
+00015ce0: 6571 7565 6e63 6531 2073 6571 7565 6e63  equence1 sequenc
+00015cf0: 6532 2073 6571 7565 6e63 6533 202d 6420  e2 sequence3 -d 
+00015d00: 6d63 6d63 7472 6565 202d 7370 2073 7064  mcmctree -sp spd
+00015d10: 6174 610a 7767 6420 666f 6375 7320 6661  ata.wgd focus fa
+00015d20: 6d69 6c69 6573 2073 6571 7565 6e63 6531  milies sequence1
+00015d30: 2073 6571 7565 6e63 6532 2073 6571 7565   sequence2 seque
+00015d40: 6e63 6533 202d 6420 7238 7320 2d73 7020  nce3 -d r8s -sp 
+00015d50: 7370 6461 7461 202d 2d6e 7369 7465 7320  spdata --nsites 
+00015d60: 6e73 6974 6569 6e66 6f0a 7767 6420 666f  nsiteinfo.wgd fo
+00015d70: 6375 7320 6661 6d69 6c69 6573 2073 6571  cus families seq
+00015d80: 7565 6e63 6531 2073 6571 7565 6e63 6532  uence1 sequence2
+00015d90: 2073 6571 7565 6e63 6533 202d 6420 6265   sequence3 -d be
+00015da0: 6173 7420 2d2d 666f 7373 696c 2066 6f73  ast --fossil fos
+00015db0: 7369 6c69 6e66 6f20 2d2d 726f 6f74 6865  silinfo --roothe
+00015dc0: 6967 6874 2072 6f6f 7468 6569 6768 7469  ight rootheighti
+00015dd0: 6e66 6f20 2d2d 6368 6169 6e73 6574 2063  nfo --chainset c
+00015de0: 6861 696e 7365 7469 6e66 6f20 2d2d 6265  hainsetinfo --be
+00015df0: 6173 746c 676a 6172 2024 5041 5448 0a60  astlgjar $PATH.`
+00015e00: 6060 0a0a 2323 2049 6c6c 7573 7472 6174  ``..## Illustrat
+00015e10: 696f 6e0a 0a57 6520 696c 6c75 7374 7261  ion..We illustra
+00015e20: 7465 206f 7572 2070 726f 6772 616d 206f  te our program o
+00015e30: 6e20 616e 2065 7865 6d70 6c61 7279 2057  n an exemplary W
+00015e40: 4744 2069 6e66 6572 656e 6365 2061 6e64  GD inference and
+00015e50: 2064 6174 696e 6720 7570 6f6e 2073 7065   dating upon spe
+00015e60: 6369 6573 202a 4171 7569 6c65 6769 6120  cies *Aquilegia 
+00015e70: 636f 6572 756c 6561 2a2e 0a0a 5468 6520  coerulea*...The 
+00015e80: 2a41 7175 696c 6567 6961 2063 6f65 7275  *Aquilegia coeru
+00015e90: 6c65 612a 2077 6173 2072 6570 6f72 7465  lea* was reporte
+00015ea0: 6420 746f 2065 7870 6572 6965 6e63 6520  d to experience 
+00015eb0: 616e 2070 616c 656f 2d70 6f6c 7970 6c6f  an paleo-polyplo
+00015ec0: 6964 697a 6174 696f 6e20 6576 656e 7420  idization event 
+00015ed0: 6166 7465 7220 7468 6520 6469 7665 7267  after the diverg
+00015ee0: 656e 6365 206f 6620 636f 7265 2065 7564  ence of core eud
+00015ef0: 6963 6f74 732c 2077 6869 6368 2069 7320  icots, which is 
+00015f00: 6c69 6b65 6c79 2073 6861 7265 6420 6279  likely shared by
+00015f10: 2061 6c6c 2052 616e 756e 6375 6c61 6c65   all Ranunculale
+00015f20: 732e 0a0a 4669 7273 7420 6162 6f76 6520  s...First above 
+00015f30: 616c 6c2c 206c 6574 2773 2064 656c 696e  all, let's delin
+00015f40: 6561 7465 2074 6865 2077 686f 6c65 2070  eate the whole p
+00015f50: 6172 616e 6f6d 6520 2a4b 2a3c 7375 623e  aranome *K*<sub>
+00015f60: 533c 2f73 7562 3e20 6167 6520 6469 7374  S</sub> age dist
+00015f70: 7269 6275 7469 6f6e 2061 6e64 2068 6176  ribution and hav
+00015f80: 6520 6120 6261 7369 6320 6f62 7365 7276  e a basic observ
+00015f90: 6174 696f 6e20 666f 7220 706f 7465 6e74  ation for potent
+00015fa0: 6961 6c6c 7920 636f 6e63 6569 7661 626c  ially conceivabl
+00015fb0: 6520 5747 4473 2c20 7573 696e 6720 7468  e WGDs, using th
+00015fc0: 6520 636f 6d6d 616e 6420 6c69 6e65 2062  e command line b
+00015fd0: 656c 6f77 2e0a 0a60 6060 0a77 6764 2064  elow...```.wgd d
+00015fe0: 6d64 2041 7175 696c 6567 6961 5f63 6f65  md Aquilegia_coe
+00015ff0: 7275 6c65 610a 7767 6420 6b73 6420 7767  rulea.wgd ksd wg
+00016000: 645f 646d 642f 4171 7569 6c65 6769 615f  d_dmd/Aquilegia_
+00016010: 636f 6572 756c 6561 2e74 7376 2041 7175  coerulea.tsv Aqu
+00016020: 696c 6567 6961 5f63 6f65 7275 6c65 610a  ilegia_coerulea.
+00016030: 6060 600a 0a54 6865 2063 6f6e 7374 7275  ```..The constru
+00016040: 6374 6564 2077 686f 6c65 2070 6172 616e  cted whole paran
+00016050: 6f6d 6520 2a4b 2a3c 7375 623e 533c 2f73  ome *K*<sub>S</s
+00016060: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00016070: 7469 6f6e 206f 6620 2a41 7175 696c 6567  tion of *Aquileg
+00016080: 6961 2063 6f65 7275 6c65 612a 2069 7320  ia coerulea* is 
+00016090: 6173 2062 656c 6f77 2c20 7765 2063 616e  as below, we can
+000160a0: 2073 6565 2074 6861 7420 7468 6572 6520   see that there 
+000160b0: 7365 656d 7320 746f 2062 6520 6120 6875  seems to be a hu
+000160c0: 6d70 2061 7420 2a4b 2a3c 7375 623e 533c  mp at *K*<sub>S<
+000160d0: 2f73 7562 3e20 3120 6275 7420 6e6f 7420  /sub> 1 but not 
+000160e0: 636c 6561 722e 0a0a 215b 5d28 6461 7461  clear...![](data
+000160f0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+00016100: 6c65 612e 7473 762e 6b73 645f 7770 2e73  lea.tsv.ksd_wp.s
+00016110: 7667 290a 0a57 6520 7468 656e 2063 6f6e  vg)..We then con
+00016120: 7374 7275 6374 2074 6865 2061 6e63 686f  struct the ancho
+00016130: 7220 2a4b 2a3c 7375 623e 533c 2f73 7562  r *K*<sub>S</sub
+00016140: 3e20 6167 6520 6469 7374 7269 6275 7469  > age distributi
+00016150: 6f6e 2075 7369 6e67 2074 6865 2063 6f6d  on using the com
+00016160: 6d61 6e64 206c 696e 6520 6265 6c6f 772e  mand line below.
+00016170: 0a0a 6060 600a 7767 6420 7379 6e20 2d66  ..```.wgd syn -f
+00016180: 206d 524e 4120 2d61 204e 616d 6520 7767   mRNA -a Name wg
+00016190: 645f 646d 642f 4171 7569 6c65 6769 615f  d_dmd/Aquilegia_
+000161a0: 636f 6572 756c 6561 2e74 7376 2041 7175  coerulea.tsv Aqu
+000161b0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+000161c0: 6766 6633 202d 6b73 2077 6764 5f6b 7364  gff3 -ks wgd_ksd
+000161d0: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000161e0: 6c65 612e 7473 762e 6b73 2e74 7376 0a60  lea.tsv.ks.tsv.`
+000161f0: 6060 0a0a 4173 2073 686f 776e 2062 656c  ``..As shown bel
+00016200: 6f77 2c20 7468 6572 6520 6172 6520 736f  ow, there are so
+00016210: 6d65 2072 6574 6169 6e65 6420 616e 6368  me retained anch
+00016220: 6f72 2070 6169 7273 2077 6974 6820 2a4b  or pairs with *K
+00016230: 2a3c 7375 623e 533c 2f73 7562 3e20 6265  *<sub>S</sub> be
+00016240: 7477 6565 6e20 3120 616e 6420 322c 2077  tween 1 and 2, w
+00016250: 6869 6368 2073 6565 6d73 2074 6f20 7375  hich seems to su
+00016260: 6767 6573 7420 6120 5747 4420 6576 656e  ggest a WGD even
+00016270: 742e 0a0a 215b 5d28 6461 7461 2f41 7175  t...![](data/Aqu
+00016280: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+00016290: 7473 762e 6b73 645f 7770 5f61 702e 7376  tsv.ksd_wp_ap.sv
+000162a0: 6729 0a0a 5468 6520 6173 736f 6369 6174  g)..The associat
+000162b0: 6564 2060 6475 7053 7461 636b 6020 706c  ed `dupStack` pl
+000162c0: 6f74 2073 686f 7773 2074 6861 7420 7468  ot shows that th
+000162d0: 6572 6520 6172 6520 6e75 6d65 726f 7573  ere are numerous
+000162e0: 2064 7570 6c69 6361 7465 6420 7365 676d   duplicated segm
+000162f0: 656e 7473 2061 6372 6f73 7320 6d6f 7374  ents across most
+00016300: 206f 6620 7468 6520 6368 726f 6d6f 736f   of the chromoso
+00016310: 6d65 732e 0a0a 215b 5d28 6461 7461 2f73  mes...![](data/s
+00016320: 796e 5f72 6573 756c 7473 2f41 7175 696c  yn_results/Aquil
+00016330: 6567 6961 5f63 6f65 7275 6c65 615f 4171  egia_coerulea_Aq
+00016340: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00016350: 5f6d 756c 7469 706c 6963 6f6e 735f 6c65  _multiplicons_le
+00016360: 7665 6c2e 706e 6729 0a0a 5765 2069 6d70  vel.png)..We imp
+00016370: 6c65 6d65 6e74 6564 2074 776f 2074 7970  lemented two typ
+00016380: 6573 206f 6620 646f 7420 706c 6f74 7320  es of dot plots 
+00016390: 696e 206f 7866 6f72 6420 6772 6964 3a20  in oxford grid: 
+000163a0: 6f6e 6520 696e 2074 6865 2075 6e69 7420  one in the unit 
+000163b0: 6f66 2062 6173 6573 2061 6e64 2074 6865  of bases and the
+000163c0: 206f 7468 6572 2069 6e20 7468 6520 756e   other in the un
+000163d0: 6974 206f 6620 6765 6e65 732c 2077 6869  it of genes, whi
+000163e0: 6368 2063 616e 2062 6520 636f 6c6f 7265  ch can be colore
+000163f0: 6420 6279 202a 4b2a 3c73 7562 3e53 3c2f  d by *K*<sub>S</
+00016400: 7375 623e 2076 616c 7565 7320 6769 7665  sub> values give
+00016410: 6e20 2a4b 2a3c 7375 623e 533c 2f73 7562  n *K*<sub>S</sub
+00016420: 3e20 6461 7461 2e0a 0a21 5b5d 2864 6174  > data...![](dat
+00016430: 612f 7379 6e5f 7265 7375 6c74 732f 4171  a/syn_results/Aq
+00016440: 7569 6c65 6769 615f 636f 6572 756c 6561  uilegia_coerulea
+00016450: 2d76 732d 4171 7569 6c65 6769 615f 636f  -vs-Aquilegia_co
+00016460: 6572 756c 6561 5f4b 732e 646f 745f 756e  erulea_Ks.dot_un
+00016470: 6974 5f67 656e 652e 706e 6729 0a0a 4173  it_gene.png)..As
+00016480: 2073 686f 776e 2061 626f 7665 2c20 7468   shown above, th
+00016490: 6520 646f 7420 706c 6f74 2069 6e20 7468  e dot plot in th
+000164a0: 6520 756e 6974 206f 6620 6765 6e65 7320  e unit of genes 
+000164b0: 7072 6573 656e 7473 206e 756d 6572 6f75  presents numerou
+000164c0: 7320 6465 6e73 656c 7920 6167 6772 6567  s densely aggreg
+000164d0: 6174 6564 2028 6c69 6e65 2d6c 696b 6529  ated (line-like)
+000164e0: 2061 6e63 686f 7220 706f 696e 7473 2061   anchor points a
+000164f0: 7420 6d6f 7374 206f 6620 7468 6520 6368  t most of the ch
+00016500: 726f 6d6f 736f 6d65 7320 7769 7468 2063  romosomes with c
+00016510: 6f6e 7369 7374 656e 7420 2a4b 2a3c 7375  onsistent *K*<su
+00016520: 623e 533c 2f73 7562 3e20 6167 6520 6265  b>S</sub> age be
+00016530: 7477 6565 6e20 3120 616e 6420 322e 2054  tween 1 and 2. T
+00016540: 6865 2064 6f74 2070 6c6f 7420 696e 2074  he dot plot in t
+00016550: 6865 2075 6e69 7420 6f66 2062 6173 6573  he unit of bases
+00016560: 2073 686f 7773 2074 6865 2073 616d 6520   shows the same 
+00016570: 7061 7474 6572 6e2c 2061 7320 6d61 6e69  pattern, as mani
+00016580: 6665 7374 6564 2062 656c 6f77 2e0a 0a21  fested below...!
+00016590: 5b5d 2864 6174 612f 7379 6e5f 7265 7375  [](data/syn_resu
+000165a0: 6c74 732f 4171 7569 6c65 6769 615f 636f  lts/Aquilegia_co
+000165b0: 6572 756c 6561 2d76 732d 4171 7569 6c65  erulea-vs-Aquile
+000165c0: 6769 615f 636f 6572 756c 6561 5f4b 732e  gia_coerulea_Ks.
+000165d0: 646f 742e 706e 6729 0a0a 5468 6520 646f  dot.png)..The do
+000165e0: 7420 706c 6f74 7320 7769 7468 6f75 7420  t plots without 
+000165f0: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00016600: 616e 6e6f 7461 7469 6f6e 2077 696c 6c20  annotation will 
+00016610: 616c 736f 2062 6520 6175 746f 6d61 7465  also be automate
+00016620: 6c79 2070 726f 6475 6365 642c 2061 7320  ly produced, as 
+00016630: 7368 6f77 6e20 6265 6c6f 772e 0a0a 215b  shown below...![
+00016640: 5d28 6461 7461 2f73 796e 5f72 6573 756c  ](data/syn_resul
+00016650: 7473 2f41 7175 696c 6567 6961 5f63 6f65  ts/Aquilegia_coe
+00016660: 7275 6c65 612d 7673 2d41 7175 696c 6567  rulea-vs-Aquileg
+00016670: 6961 5f63 6f65 7275 6c65 612e 646f 745f  ia_coerulea.dot_
+00016680: 756e 6974 5f67 656e 652e 706e 6729 0a0a  unit_gene.png)..
+00016690: 215b 5d28 6461 7461 2f73 796e 5f72 6573  ![](data/syn_res
+000166a0: 756c 7473 2f41 7175 696c 6567 6961 5f63  ults/Aquilegia_c
+000166b0: 6f65 7275 6c65 612d 7673 2d41 7175 696c  oerulea-vs-Aquil
+000166c0: 6567 6961 5f63 6f65 7275 6c65 612e 646f  egia_coerulea.do
+000166d0: 742e 706e 6729 0a0a 4e6f 7465 2074 6861  t.png)..Note tha
+000166e0: 7420 7468 6520 6f70 6163 6974 7920 6f66  t the opacity of
+000166f0: 2061 6e63 686f 7220 646f 7473 2061 6e64   anchor dots and
+00016700: 2061 6c6c 2068 6f6d 6f6c 6f67 2064 6f74   all homolog dot
+00016710: 7320 6361 6e20 6265 2073 6574 2062 7920  s can be set by 
+00016720: 7468 6520 6f70 7469 6f6e 2060 2d2d 6170  the option `--ap
+00016730: 616c 7068 6160 2061 6e64 2060 2d2d 686f  alpha` and `--ho
+00016740: 616c 7068 6160 2073 6570 6172 6174 656c  alpha` separatel
+00016750: 792e 2049 6620 6f6e 6520 6a75 7374 2077  y. If one just w
+00016760: 616e 7473 2074 6f20 7365 6520 7468 6520  ants to see the 
+00016770: 616e 6368 6f72 2064 6f74 732c 2073 6574  anchor dots, set
+00016780: 7469 6e67 2074 6865 2060 686f 616c 7068  ting the `hoalph
+00016790: 6160 2061 7320 3020 286f 7220 6f74 6865  a` as 0 (or othe
+000167a0: 7220 6d69 6e75 7363 756c 6520 7661 6c75  r minuscule valu
+000167b0: 6573 2920 7769 6c6c 2064 6f2e 2049 6620  es) will do. If 
+000167c0: 6f6e 6520 7761 6e74 7320 746f 2073 6565  one wants to see
+000167d0: 2074 6865 2064 6973 7472 6962 7574 696f   the distributio
+000167e0: 6e20 6f66 2077 686f 6c65 2064 6f74 7320  n of whole dots 
+000167f0: 6265 7474 6572 2c20 7365 7474 696e 6720  better, setting 
+00016800: 7468 6520 6068 6f61 6c70 6861 6020 6869  the `hoalpha` hi
+00016810: 6768 6572 2028 616e 6420 6061 7061 6c70  gher (and `apalp
+00016820: 6861 6020 6c6f 7765 7229 2077 696c 6c20  ha` lower) will 
+00016830: 646f 2e20 5468 6520 6064 6f74 7369 7a65  do. The `dotsize
+00016840: 6020 6f70 7469 6f6e 2063 616e 2062 6520  ` option can be 
+00016850: 6361 6c6c 6564 2074 6f20 6164 6a75 7374  called to adjust
+00016860: 2074 6865 2073 697a 6520 6f66 2064 6f74   the size of dot
+00016870: 732e 0a0a 4120 6675 7274 6865 7220 6173  s...A further as
+00016880: 736f 6369 6174 6564 2053 796e 6465 7074  sociated Syndept
+00016890: 6820 706c 6f74 2073 686f 7773 2074 6861  h plot shows tha
+000168a0: 7420 7468 6572 6520 6172 6520 6d6f 7265  t there are more
+000168b0: 2074 6861 6e20 3530 2064 7570 6c69 6361   than 50 duplica
+000168c0: 7465 6420 7365 676d 656e 7473 206c 6f6e  ted segments lon
+000168d0: 6765 7220 7468 616e 2031 3030 3030 2062  ger than 10000 b
+000168e0: 7020 616e 6420 3330 2067 656e 6573 2028  p and 30 genes (
+000168f0: 736f 2061 7320 746f 2064 726f 7020 6672  so as to drop fr
+00016900: 6167 6d65 6e74 6172 7920 7365 676d 656e  agmentary segmen
+00016910: 7473 292c 2077 6869 6368 2064 6f6d 696e  ts), which domin
+00016920: 6174 6573 2074 6865 2077 686f 6c65 2063  ates the whole c
+00016930: 6f6c 6c69 6e65 6172 2072 6174 696f 2063  ollinear ratio c
+00016940: 6174 6567 6f72 792e 0a0a 215b 5d28 6461  ategory...![](da
+00016950: 7461 2f73 796e 5f72 6573 756c 7473 2f53  ta/syn_results/S
+00016960: 796e 6465 7074 682e 7376 6729 0a0a 5765  yndepth.svg)..We
+00016970: 2063 616e 2066 6974 2061 6e20 454c 4d4d   can fit an ELMM
+00016980: 206d 6978 7475 7265 206d 6f64 656c 2075   mixture model u
+00016990: 706f 6e20 7468 6520 7768 6f6c 6520 7061  pon the whole pa
+000169a0: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
+000169b0: 3c2f 7375 623e 2061 6765 2064 6973 7472  </sub> age distr
+000169c0: 6962 7574 696f 6e20 746f 2073 6565 206d  ibution to see m
+000169d0: 6f72 6520 6163 6375 7261 7465 6c79 2074  ore accurately t
+000169e0: 6865 2073 6967 6e69 6669 6361 6e63 6520  he significance 
+000169f0: 616e 6420 6c6f 6361 7469 6f6e 206f 6620  and location of 
+00016a00: 706f 7465 6e74 6961 6c20 5747 4473 2c20  potential WGDs, 
+00016a10: 7573 696e 6720 7468 6520 636f 6d6d 616e  using the comman
+00016a20: 6420 6c69 6e65 2062 656c 6f77 2e0a 0a60  d line below...`
+00016a30: 6060 0a77 6764 2076 697a 202d 6420 7767  ``.wgd viz -d wg
+00016a40: 645f 6b73 642f 4171 7569 6c65 6769 615f  d_ksd/Aquilegia_
+00016a50: 636f 6572 756c 6561 2e74 7376 2e6b 732e  coerulea.tsv.ks.
+00016a60: 7473 760a 6060 600a 0a54 6865 2072 6573  tsv.```..The res
+00016a70: 756c 7420 6f66 2045 4c4d 4d20 6d69 7874  ult of ELMM mixt
+00016a80: 7572 6520 6d6f 6465 6c20 636c 7573 7465  ure model cluste
+00016a90: 7269 6e67 2073 686f 7773 2074 6861 7420  ring shows that 
+00016aa0: 7468 6572 6520 6973 2061 206c 696b 656c  there is a likel
+00016ab0: 7920 5747 4420 636f 6d70 6f6e 656e 7420  y WGD component 
+00016ac0: 6174 202a 4b2a 3c73 7562 3e53 3c2f 7375  at *K*<sub>S</su
+00016ad0: 623e 2031 2e31 392e 0a0a 215b 5d28 6461  b> 1.19...![](da
+00016ae0: 7461 2f65 6c6d 6d5f 4171 7569 6c65 6769  ta/elmm_Aquilegi
+00016af0: 615f 636f 6572 756c 6561 2e74 7376 2e6b  a_coerulea.tsv.k
+00016b00: 732e 7473 765f 6265 7374 5f6d 6f64 656c  s.tsv_best_model
+00016b10: 735f 7765 6967 6874 6564 2e73 7667 290a  s_weighted.svg).
+00016b20: 0a4c 6574 2773 2064 6f20 6120 6d69 7874  .Let's do a mixt
+00016b30: 7572 6520 6d6f 6465 6c20 636c 7573 7465  ure model cluste
+00016b40: 7269 6e67 2066 6f72 2061 6e63 686f 7220  ring for anchor 
+00016b50: 2a4b 2a3c 7375 623e 533c 2f73 7562 3e20  *K*<sub>S</sub> 
+00016b60: 746f 6f2c 2075 7369 6e67 2074 6865 2063  too, using the c
+00016b70: 6f6d 6d61 6e64 206c 696e 6520 6265 6c6f  ommand line belo
+00016b80: 772e 204e 6f74 6520 7468 6174 2074 6869  w. Note that thi
+00016b90: 7320 7374 6570 2077 696c 6c20 6175 746f  s step will auto
+00016ba0: 6d61 7465 6c79 2063 616c 6c20 7468 6520  mately call the 
+00016bb0: 7365 676d 656e 7420 2a4b 2a3c 7375 623e  segment *K*<sub>
+00016bc0: 533c 2f73 7562 3e20 636c 7573 7465 7269  S</sub> clusteri
+00016bd0: 6e67 2061 6e61 6c79 7369 7320 746f 6f2e  ng analysis too.
+00016be0: 0a0a 6060 600a 7767 6420 7065 616b 2077  ..```.wgd peak w
+00016bf0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00016c00: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00016c10: 2e74 7376 202d 2d61 6e63 686f 7270 6f69  .tsv --anchorpoi
+00016c20: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
+00016c30: 6f72 652d 6f75 742f 616e 6368 6f72 706f  ore-out/anchorpo
+00016c40: 696e 7473 2e74 7874 202d 2d73 6567 6d65  ints.txt --segme
+00016c50: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
+00016c60: 6f72 652d 6f75 742f 7365 676d 656e 7473  ore-out/segments
+00016c70: 2e74 7874 202d 2d6c 6973 7465 6c65 6d65  .txt --listeleme
+00016c80: 6e74 7320 7767 645f 7379 6e2f 6961 6468  nts wgd_syn/iadh
+00016c90: 6f72 652d 6f75 742f 6c69 7374 5f65 6c65  ore-out/list_ele
+00016ca0: 6d65 6e74 732e 7478 7420 2d2d 6d75 6c74  ments.txt --mult
+00016cb0: 6970 6c69 636f 6e70 6169 7273 2077 6764  ipliconpairs wgd
+00016cc0: 5f73 796e 2f69 6164 686f 7265 2d6f 7574  _syn/iadhore-out
+00016cd0: 2f6d 756c 7469 706c 6963 6f6e 5f70 6169  /multiplicon_pai
+00016ce0: 7273 2e74 7874 2028 2d2d 7765 6967 6874  rs.txt (--weight
+00016cf0: 6564 290a 6060 600a 0a54 6865 2061 6e63  ed).```..The anc
+00016d00: 686f 7220 2a4b 2a3c 7375 623e 533c 2f73  hor *K*<sub>S</s
+00016d10: 7562 3e20 6167 6520 6469 7374 7269 6275  ub> age distribu
+00016d20: 7469 6f6e 2061 6c73 6f20 6861 7320 6120  tion also has a 
+00016d30: 6c69 6b65 6c79 2057 4744 2063 6f6d 706f  likely WGD compo
+00016d40: 6e65 6e74 2077 6974 6820 6d6f 6465 2031  nent with mode 1
+00016d50: 2e32 382e 0a0a 215b 5d28 6461 7461 2f4f  .28...![](data/O
+00016d60: 7269 6769 6e61 6c5f 416e 6368 6f72 4b73  riginal_AnchorKs
+00016d70: 5f47 4d4d 5f43 6f6d 706f 6e65 6e74 335f  _GMM_Component3_
+00016d80: 6e6f 6465 5f61 7665 7261 6765 645f 4c6f  node_averaged_Lo
+00016d90: 676e 6f72 6d61 6c2e 7376 6729 0a0a 4e6f  gnormal.svg)..No
+00016da0: 7720 7468 6174 2077 6520 6861 7665 2073  w that we have s
+00016db0: 6565 6e20 7468 6520 6576 6964 656e 6365  een the evidence
+00016dc0: 206f 6620 6e75 6d65 726f 7573 2064 7570   of numerous dup
+00016dd0: 6c69 6361 7465 6420 7365 676d 656e 7473  licated segments
+00016de0: 2061 6e64 2074 6865 2061 6767 7265 6761   and the aggrega
+00016df0: 7469 6f6e 206f 6620 6475 706c 6963 6174  tion of duplicat
+00016e00: 6573 2061 6765 2061 7420 2a4b 2a3c 7375  es age at *K*<su
+00016e10: 623e 533c 2f73 7562 3e20 312e 3238 206f  b>S</sub> 1.28 o
+00016e20: 7220 312e 3139 2066 6f72 2061 6e63 686f  r 1.19 for ancho
+00016e30: 7220 7061 6972 7320 616e 6420 6e6f 6e2d  r pairs and non-
+00016e40: 616e 6368 6f72 2070 6169 7273 2074 6872  anchor pairs thr
+00016e50: 6f75 6768 6f75 7420 7468 6520 7768 6f6c  oughout the whol
+00016e60: 6520 6765 6e6f 6d65 2e20 5765 2063 616e  e genome. We can
+00016e70: 2063 6c61 696d 2077 6974 6820 736f 6d65   claim with some
+00016e80: 2063 6f6e 6669 6465 6e63 6520 7468 6174   confidence that
+00016e90: 202a 4171 7569 6c65 6769 6120 636f 6572   *Aquilegia coer
+00016ea0: 756c 6561 2a20 6d69 6768 7420 6861 7665  ulea* might have
+00016eb0: 2065 7870 6572 6965 6e63 6564 2061 2070   experienced a p
+00016ec0: 616c 656f 2d70 6f6c 7970 6c6f 6964 697a  aleo-polyploidiz
+00016ed0: 6174 696f 6e20 6576 656e 742e 204e 6578  ation event. Nex
+00016ee0: 742c 204c 6574 2773 2068 6176 6520 6120  t, Let's have a 
+00016ef0: 6675 7274 6865 7220 6c6f 6f6b 2061 626f  further look abo
+00016f00: 7574 2069 7473 2070 6879 6c6f 6765 6e65  ut its phylogene
+00016f10: 7469 6320 6c6f 6361 7469 6f6e 2e20 5765  tic location. We
+00016f20: 206b 6e6f 7720 7468 6174 2074 6865 7265   know that there
+00016f30: 2061 7265 2075 6e63 6572 7461 696e 7469   are uncertainti
+00016f40: 6573 2061 626f 7574 2077 6865 7468 6572  es about whether
+00016f50: 2074 6869 7320 7075 7461 7469 7665 2070   this putative p
+00016f60: 616c 656f 2d70 6f6c 7970 6c6f 6964 697a  aleo-polyploidiz
+00016f70: 6174 696f 6e20 6576 656e 7420 6973 2073  ation event is s
+00016f80: 6861 7265 6420 7769 7468 2061 6c6c 2065  hared with all e
+00016f90: 7564 6963 6f74 7320 6f72 206e 6f74 2e20  udicots or not. 
+00016fa0: 5765 2063 616e 2063 686f 6f73 6520 736f  We can choose so
+00016fb0: 6d65 206f 7468 6572 2065 7564 6963 6f74  me other eudicot
+00016fc0: 2067 656e 6f6d 6573 2074 6f20 7365 6520   genomes to see 
+00016fd0: 7468 6520 6f72 6465 7269 6e67 206f 6620  the ordering of 
+00016fe0: 7370 6563 6961 7469 6f6e 2061 6e64 2070  speciation and p
+00016ff0: 6f6c 7970 6c6f 6964 697a 6174 696f 6e20  olyploidization 
+00017000: 6576 656e 7473 2e20 4865 7265 2077 6520  events. Here we 
+00017010: 6368 6f6f 7365 202a 5669 7469 7320 7669  choose *Vitis vi
+00017020: 6e69 6665 7261 2a2c 202a 5072 6f74 6561  nifera*, *Protea
+00017030: 2063 796e 6172 6f69 6465 732a 2061 6e64   cynaroides* and
+00017040: 202a 4163 6f72 7573 2061 6d65 7269 6361   *Acorus america
+00017050: 6e75 732a 2069 6e20 7468 6520 666f 6c6c  nus* in the foll
+00017060: 6f77 696e 6720 2a4b 2a3c 7375 623e 533c  owing *K*<sub>S<
+00017070: 2f73 7562 3e20 616e 616c 7973 6973 2e20  /sub> analysis. 
+00017080: 4669 7273 742c 2077 6520 6275 696c 7420  First, we built 
+00017090: 6120 676c 6f62 616c 204d 5242 4820 6661  a global MRBH fa
+000170a0: 6d69 6c79 2075 7369 6e67 2074 6865 2063  mily using the c
+000170b0: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 0a60  ommand below...`
+000170c0: 6060 0a77 6764 2064 6d64 202d 2d67 6c6f  ``.wgd dmd --glo
+000170d0: 6261 6c6d 7262 6820 4171 7569 6c65 6769  balmrbh Aquilegi
+000170e0: 615f 636f 6572 756c 6561 2050 726f 7465  a_coerulea Prote
+000170f0: 615f 6379 6e61 726f 6964 6573 2041 636f  a_cynaroides Aco
+00017100: 7275 735f 616d 6572 6963 616e 7573 2056  rus_americanus V
+00017110: 6974 6973 5f76 696e 6966 6572 6120 2d6f  itis_vinifera -o
+00017120: 2077 6764 5f67 6c6f 6261 6c6d 7262 680a   wgd_globalmrbh.
+00017130: 6060 600a 0a49 6e20 7468 6520 676c 6f62  ```..In the glob
+00017140: 616c 204d 5242 4820 6661 6d69 6c79 2c20  al MRBH family, 
+00017150: 6576 6572 7920 7061 6972 206f 6620 6f72  every pair of or
+00017160: 7468 6f6c 6f67 6f75 7320 6765 6e65 7320  thologous genes 
+00017170: 6973 2074 6865 2072 6563 6970 726f 6361  is the reciproca
+00017180: 6c20 6265 7374 2068 6974 2c20 7375 6767  l best hit, sugg
+00017190: 6573 7469 6e67 2074 7275 6520 6f72 7468  esting true orth
+000171a0: 6f6c 6f67 6f75 7320 7265 6c61 7469 6f6e  ologous relation
+000171b0: 7368 6970 732e 2057 6520 776f 756c 6420  ships. We would 
+000171c0: 7573 6520 7468 6520 2a4b 2a3c 7375 623e  use the *K*<sub>
+000171d0: 533c 2f73 7562 3e20 7661 6c75 6573 2061  S</sub> values a
+000171e0: 7373 6f63 6961 7465 6420 7769 7468 2074  ssociated with t
+000171f0: 6865 7365 206f 7274 686f 6c6f 676f 7573  hese orthologous
+00017200: 2070 6169 7273 2074 6f20 6465 6c69 6d69   pairs to delimi
+00017210: 7420 7468 6520 6469 7665 7267 656e 6365  t the divergence
+00017220: 202a 4b2a 3c73 7562 3e53 3c2f 7375 623e   *K*<sub>S</sub>
+00017230: 2070 6561 6b2e 2054 6f67 6574 6865 7220   peak. Together 
+00017240: 7769 7468 2074 6865 2077 686f 6c65 2070  with the whole p
+00017250: 6172 616e 6f6d 6520 2a4b 2a3c 7375 623e  aranome *K*<sub>
+00017260: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+00017270: 7469 6f6e 2c20 7765 2063 6f6e 6475 6374  tion, we conduct
+00017280: 2074 6865 2072 6174 6520 636f 7272 6563   the rate correc
+00017290: 7469 6f6e 2075 7369 6e67 2074 6865 2063  tion using the c
+000172a0: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 0a21  ommand below...!
+000172b0: 2153 696e 6365 2060 7767 6460 2076 6572  !Since `wgd` ver
+000172c0: 7369 6f6e 2032 2e30 2e32 342c 2077 6520  sion 2.0.24, we 
+000172d0: 7265 7772 6f74 6520 6120 636c 6561 6e65  rewrote a cleane
+000172e0: 7220 616e 6420 7175 6963 6b65 7220 7761  r and quicker wa
+000172f0: 7920 6f66 2064 6f69 6e67 2073 7562 7374  y of doing subst
+00017300: 6974 7574 696f 6e20 7261 7465 2063 6f72  itution rate cor
+00017310: 7265 6374 696f 6e2e 2049 7427 7320 6e6f  rection. It's no
+00017320: 7420 7265 7175 6972 6564 2074 6f20 7479  t required to ty
+00017330: 7065 2069 6e20 616e 7920 7370 6569 6365  pe in any speice
+00017340: 7320 7061 6972 2061 6e64 2061 2073 6572  s pair and a ser
+00017350: 6965 7320 6f66 202a 4b2a 3c73 7562 3e53  ies of *K*<sub>S
+00017360: 3c2f 7375 623e 2070 6c6f 7473 2077 696c  </sub> plots wil
+00017370: 6c20 6265 2070 726f 6475 6365 642e 2054  l be produced. T
+00017380: 6865 2072 6571 7569 7265 6420 6669 6c65  he required file
+00017390: 7320 6172 6520 6f72 7468 6f6c 6f67 6f75  s are orthologou
+000173a0: 7320 2a4b 2a3c 7375 623e 533c 2f73 7562  s *K*<sub>S</sub
+000173b0: 3e20 6461 7461 6669 6c65 2c20 7061 7261  > datafile, para
+000173c0: 6c6f 676f 7573 202a 4b2a 3c73 7562 3e53  logous *K*<sub>S
+000173d0: 3c2f 7375 623e 2064 6174 6166 696c 652c  </sub> datafile,
+000173e0: 2061 2073 7065 6369 6573 2074 7265 6520   a species tree 
+000173f0: 616e 6420 6120 666f 6361 6c20 7370 6563  and a focal spec
+00017400: 6965 7320 2874 6865 206f 6e65 2069 6e70  ies (the one inp
+00017410: 7574 7465 6420 7769 7468 2070 6172 616c  utted with paral
+00017420: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+00017430: 2f73 7562 3e20 6461 7461 292e 2055 7365  /sub> data). Use
+00017440: 7273 2063 616e 2063 686f 6f73 6520 746f  rs can choose to
+00017450: 2061 6464 206f 6e65 206d 6f72 6520 6c61   add one more la
+00017460: 7965 7220 6f66 2045 4c4d 4d20 616e 616c  yer of ELMM anal
+00017470: 7973 6973 206f 6e20 7061 7261 6c6f 676f  ysis on paralogo
+00017480: 7573 202a 4b2a 3c73 7562 3e53 3c2f 7375  us *K*<sub>S</su
+00017490: 623e 2076 616c 7565 7320 616e 642f 6f72  b> values and/or
+000174a0: 2047 4d4d 2061 6e61 6c79 7369 7320 6f6e   GMM analysis on
+000174b0: 2061 6e63 686f 7220 2a4b 2a3c 7375 623e   anchor *K*<sub>
+000174c0: 533c 2f73 7562 3e20 6469 7374 7269 6275  S</sub> distribu
+000174d0: 7469 6f6e 2e20 5468 6520 6f72 7468 6f6c  tion. The orthol
+000174e0: 6f67 6f75 7320 2a4b 2a3c 7375 623e 533c  ogous *K*<sub>S<
+000174f0: 2f73 7562 3e20 6469 7374 7269 6275 7469  /sub> distributi
+00017500: 6f6e 2063 616e 2062 6520 6361 6c63 756c  on can be calcul
+00017510: 6174 6564 2075 7369 6e67 2074 6865 2063  ated using the c
+00017520: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 0a60  ommand below...`
+00017530: 6060 0a77 6764 206b 7364 2077 6764 5f67  ``.wgd ksd wgd_g
+00017540: 6c6f 6261 6c6d 7262 682f 676c 6f62 616c  lobalmrbh/global
+00017550: 5f4d 5242 482e 7473 7620 4171 7569 6c65  _MRBH.tsv Aquile
+00017560: 6769 615f 636f 6572 756c 6561 2050 726f  gia_coerulea Pro
+00017570: 7465 615f 6379 6e61 726f 6964 6573 2041  tea_cynaroides A
+00017580: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00017590: 2056 6974 6973 5f76 696e 6966 6572 6120   Vitis_vinifera 
+000175a0: 2d6f 2077 6764 5f67 6c6f 6261 6c6d 7262  -o wgd_globalmrb
+000175b0: 685f 6b73 0a60 6060 0a0a 5769 7468 2074  h_ks.```..With t
+000175c0: 6865 2063 616c 6375 6c61 7465 6420 6f72  he calculated or
+000175d0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+000175e0: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+000175f0: 6275 7469 6f6e 2c20 7765 2063 616e 2075  bution, we can u
+00017600: 7365 2074 6865 2063 6f6d 6d61 6e64 2062  se the command b
+00017610: 656c 6f77 2074 6f20 636f 6e64 7563 7420  elow to conduct 
+00017620: 7468 6520 7261 7465 2063 6f72 7265 6374  the rate correct
+00017630: 696f 6e20 616e 642f 6f72 206d 6978 7475  ion and/or mixtu
+00017640: 7265 206d 6f64 656c 696e 6720 616e 616c  re modeling anal
+00017650: 7973 6973 2e0a 0a60 6060 0a77 6764 2076  ysis...```.wgd v
+00017660: 697a 202d 6420 7767 645f 676c 6f62 616c  iz -d wgd_global
+00017670: 6d72 6268 5f6b 732f 676c 6f62 616c 5f4d  mrbh_ks/global_M
+00017680: 5242 482e 7473 762e 6b73 2e74 7376 202d  RBH.tsv.ks.tsv -
+00017690: 6661 2041 7175 696c 6567 6961 5f63 6f65  fa Aquilegia_coe
+000176a0: 7275 6c65 6120 2d65 706b 2077 6764 5f6b  rulea -epk wgd_k
+000176b0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+000176c0: 7275 6c65 612e 6b73 2e74 7376 202d 6170  rulea.ks.tsv -ap
+000176d0: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+000176e0: 2d6f 7574 2f61 6e63 686f 7270 6f69 6e74  -out/anchorpoint
+000176f0: 732e 7478 7420 2d73 7020 7370 6563 6965  s.txt -sp specie
+00017700: 7374 7265 652e 6e77 202d 6f20 7767 645f  stree.nw -o wgd_
+00017710: 7669 7a5f 6d69 7865 645f 4b73 202d 2d70  viz_mixed_Ks --p
+00017720: 6c6f 7465 6c6d 6d20 2d2d 706c 6f74 6170  lotelmm --plotap
+00017730: 676d 6d20 2d2d 7265 7765 6967 6874 0a60  gmm --reweight.`
+00017740: 6060 0a0a 6f72 2075 7369 6e67 2074 6865  ``..or using the
+00017750: 2063 6f6d 6d61 6e64 2062 656c 6f77 2c20   command below, 
+00017760: 7768 6963 6820 636f 6d62 696e 6573 2074  which combines t
+00017770: 6865 2074 776f 2073 7465 7073 2061 626f  he two steps abo
+00017780: 7665 2069 6e20 6f6e 652e 204e 6f74 6520  ve in one. Note 
+00017790: 7468 6174 2077 6520 7375 6767 6573 7420  that we suggest 
+000177a0: 6f66 2074 616b 696e 6720 7477 6f20 7365  of taking two se
+000177b0: 7061 7261 7465 2073 7465 7073 2069 6e20  parate steps in 
+000177c0: 7768 6963 6820 6077 6764 206b 7364 6020  which `wgd ksd` 
+000177d0: 756e 6465 7274 616b 6573 2074 6865 2063  undertakes the c
+000177e0: 616c 6375 6c61 7469 6f6e 206f 6620 6f72  alculation of or
+000177f0: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+00017800: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00017810: 6275 7469 6f6e 2077 6869 6c65 2060 7767  bution while `wg
+00017820: 6420 7669 7a60 2063 6172 7269 6573 206f  d viz` carries o
+00017830: 7574 2074 6865 2072 6174 6520 636f 7272  ut the rate corr
+00017840: 6563 7469 6f6e 2061 6e64 2047 4d4d 2061  ection and GMM a
+00017850: 6e61 6c79 7369 7320 7375 6368 2074 6861  nalysis such tha
+00017860: 7420 6974 2773 2065 6173 6965 7220 746f  t it's easier to
+00017870: 2064 6562 7567 2e0a 6060 600a 7767 6420   debug..```.wgd 
+00017880: 6b73 6420 7767 645f 676c 6f62 616c 6d72  ksd wgd_globalmr
+00017890: 6268 2f67 6c6f 6261 6c5f 4d52 4248 2e74  bh/global_MRBH.t
+000178a0: 7376 2041 7175 696c 6567 6961 5f63 6f65  sv Aquilegia_coe
+000178b0: 7275 6c65 6120 5072 6f74 6561 5f63 796e  rulea Protea_cyn
+000178c0: 6172 6f69 6465 7320 4163 6f72 7573 5f61  aroides Acorus_a
+000178d0: 6d65 7269 6361 6e75 7320 5669 7469 735f  mericanus Vitis_
+000178e0: 7669 6e69 6665 7261 202d 2d65 7874 7261  vinifera --extra
+000178f0: 7061 7261 6e6f 6d65 6b73 2077 6764 5f6b  paranomeks wgd_k
+00017900: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+00017910: 7275 6c65 612e 7473 762e 6b73 2e74 7376  rulea.tsv.ks.tsv
+00017920: 202d 7370 2073 7065 6369 6573 7472 6565   -sp speciestree
+00017930: 2e6e 7720 2d2d 7265 7765 6967 6874 202d  .nw --reweight -
+00017940: 6f20 7767 645f 676c 6f62 616c 6d72 6268  o wgd_globalmrbh
+00017950: 5f6b 735f 7261 7465 5f63 6f72 7265 6374  _ks_rate_correct
+00017960: 696f 6e20 2d66 6120 4171 7569 6c65 6769  ion -fa Aquilegi
+00017970: 615f 636f 6572 756c 6561 202d 6170 2077  a_coerulea -ap w
+00017980: 6764 5f73 796e 2f69 6164 686f 7265 2d6f  gd_syn/iadhore-o
+00017990: 7574 2f61 6e63 686f 7270 6f69 6e74 732e  ut/anchorpoints.
+000179a0: 7478 7420 2d2d 706c 6f74 656c 6d6d 202d  txt --plotelmm -
+000179b0: 2d70 6c6f 7461 7067 6d6d 0a60 6060 0a0a  -plotapgmm.```..
+000179c0: 5468 6520 6669 6c65 2060 7370 6563 6965  The file `specie
+000179d0: 7374 7265 652e 6e77 6020 6973 2074 6865  stree.nw` is the
+000179e0: 2074 6578 7420 6669 6c65 206f 6620 7370   text file of sp
+000179f0: 6563 6965 7320 7472 6565 2069 6e20 6e65  ecies tree in ne
+00017a00: 7769 636b 2074 6861 7420 7261 7465 2063  wick that rate c
+00017a10: 6f72 7265 6374 696f 6e20 776f 756c 6420  orrection would 
+00017a20: 6265 2063 6f6e 6475 6374 6564 206f 6e2e  be conducted on.
+00017a30: 2049 7473 2063 6f6e 7465 6e74 2069 7320   Its content is 
+00017a40: 6173 2062 656c 6f77 2e20 5573 6572 7320  as below. Users 
+00017a50: 6361 6e20 6f70 7469 6f6e 616c 6c79 2070  can optionally p
+00017a60: 726f 7669 6465 2074 6865 2073 7065 6369  rovide the speci
+00017a70: 6573 2070 6169 7273 2074 6f20 6265 2070  es pairs to be p
+00017a80: 6c6f 7474 6564 2062 7574 2077 6520 7375  lotted but we su
+00017a90: 6767 6573 7420 6f66 206a 7573 7420 7573  ggest of just us
+00017aa0: 696e 6720 602d 6661 2041 7175 696c 6567  ing `-fa Aquileg
+00017ab0: 6961 5f63 6f65 7275 6c65 6160 2074 6f20  ia_coerulea` to 
+00017ac0: 706c 6f74 2061 6c6c 2070 6f73 7369 626c  plot all possibl
+00017ad0: 6520 666f 6361 6c2d 7369 7374 6572 2073  e focal-sister s
+00017ae0: 7065 6369 6573 2070 6169 7273 2e20 5765  pecies pairs. We
+00017af0: 2073 7567 6765 7374 2061 6464 696e 6720   suggest adding 
+00017b00: 7468 6520 6f70 7469 6f6e 2060 2d2d 7265  the option `--re
+00017b10: 7765 6967 6874 6020 746f 2072 6563 616c  weight` to recal
+00017b20: 6375 6c61 7465 2074 6865 2077 6569 6768  culate the weigh
+00017b30: 7420 7065 7220 7370 6563 6965 7320 7061  t per species pa
+00017b40: 6972 2073 7563 6820 7468 6174 2074 6865  ir such that the
+00017b50: 2077 6569 6768 7420 6f66 206f 7274 686f   weight of ortho
+00017b60: 6c6f 676f 7573 2067 656e 6520 7061 6972  logous gene pair
+00017b70: 7320 7769 6c6c 2062 6563 6f6d 6520 312e  s will become 1.
+00017b80: 2045 7874 7261 2063 6f6c 6c69 6e65 6172   Extra collinear
+00017b90: 2064 6174 6120 6361 6e20 6265 2061 6464   data can be add
+00017ba0: 6564 2062 7920 7468 6520 6f70 7469 6f6e  ed by the option
+00017bb0: 2060 2d61 7060 2061 6e64 2061 6464 6974   `-ap` and addit
+00017bc0: 696f 6e61 6c20 636c 7573 7465 7269 6e67  ional clustering
+00017bd0: 2061 6e61 6c79 7369 7320 6361 6e20 6265   analysis can be
+00017be0: 2069 6e69 7469 6174 6564 2062 7920 7365   initiated by se
+00017bf0: 7474 696e 6720 7468 6520 6f70 7469 6f6e  tting the option
+00017c00: 2060 2d2d 706c 6f74 6170 676d 6d60 2061   `--plotapgmm` a
+00017c10: 6e64 2060 2d2d 706c 6f74 656c 6d6d 602e  nd `--plotelmm`.
+00017c20: 0a0a 6060 600a 2828 2856 6974 6973 5f76  ..```.(((Vitis_v
+00017c30: 696e 6966 6572 612c 5072 6f74 6561 5f63  inifera,Protea_c
+00017c40: 796e 6172 6f69 6465 7329 2c41 7175 696c  ynaroides),Aquil
+00017c50: 6567 6961 5f63 6f65 7275 6c65 6129 2c41  egia_coerulea),A
+00017c60: 636f 7275 735f 616d 6572 6963 616e 7573  corus_americanus
+00017c70: 293b 0a60 6060 0a0a 215b 5d28 6461 7461  );.```..![](data
+00017c80: 2f4d 6978 6564 2e6b 732e 4171 7569 6c65  /Mixed.ks.Aquile
+00017c90: 6769 615f 636f 6572 756c 6561 2e6e 6f64  gia_coerulea.nod
+00017ca0: 652e 7765 6967 6874 6564 2e70 6e67 290a  e.weighted.png).
+00017cb0: 0a54 6865 206d 6978 6564 202a 4b2a 3c73  .The mixed *K*<s
+00017cc0: 7562 3e53 3c2f 7375 623e 2064 6973 7472  ub>S</sub> distr
+00017cd0: 6962 7574 696f 6e20 7368 6f77 6e20 6162  ibution shown ab
+00017ce0: 6f76 6520 6973 2061 2070 7562 6c69 6361  ove is a publica
+00017cf0: 7469 6f6e 2d72 6561 6479 2066 6967 7572  tion-ready figur
+00017d00: 6520 7468 6174 2061 7373 656d 626c 6573  e that assembles
+00017d10: 2074 6865 2072 6573 756c 7473 206f 6620   the results of 
+00017d20: 454c 4d4d 2c20 474d 4d20 616e 6420 7261  ELMM, GMM and ra
+00017d30: 7465 2063 6f72 7265 6374 696f 6e2e 2054  te correction. T
+00017d40: 6865 206f 6e65 2d76 732d 6f6e 6520 6f72  he one-vs-one or
+00017d50: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+00017d60: 623e 533c 2f73 7562 3e20 6469 7374 7269  b>S</sub> distri
+00017d70: 6275 7469 6f6e 7320 6973 2061 6c73 6f20  butions is also 
+00017d80: 6175 746f 6d61 7469 6361 6c6c 7920 7072  automatically pr
+00017d90: 6f64 7563 6564 2077 6974 6820 7261 7465  oduced with rate
+00017da0: 2063 6f72 7265 6374 696f 6e20 7265 7375   correction resu
+00017db0: 6c74 7320 7375 7065 7269 6d70 6f73 6564  lts superimposed
+00017dc0: 2077 6865 7265 2061 7661 696c 6162 6c65   where available
+00017dd0: 2061 7320 7368 6f77 6e20 6265 6c6f 772e   as shown below.
+00017de0: 0a0a 215b 5d28 6461 7461 2f46 6f63 7573  ..![](data/Focus
+00017df0: 5f73 6973 7465 725f 7061 6972 732e 6b73  _sister_pairs.ks
+00017e00: 2e6e 6f64 652e 7765 6967 6874 6564 2e70  .node.weighted.p
+00017e10: 6e67 290a 215b 5d28 6461 7461 2f41 6c6c  ng).![](data/All
+00017e20: 5f70 6169 7273 2e6b 732e 6e6f 6465 2e77  _pairs.ks.node.w
+00017e30: 6569 6768 7465 642e 706e 6729 0a0a 4265  eighted.png)..Be
+00017e40: 7369 6465 7320 7468 6520 6162 6f76 6520  sides the above 
+00017e50: 6b73 7261 7465 732d 6c69 6b65 202a 4b2a  ksrates-like *K*
+00017e60: 3c73 7562 3e53 3c2f 7375 623e 2070 6c6f  <sub>S</sub> plo
+00017e70: 742c 2061 206d 6f72 6520 636c 6173 7369  t, a more classi
+00017e80: 6320 2a4b 2a3c 7375 623e 533c 2f73 7562  c *K*<sub>S</sub
+00017e90: 3e20 706c 6f74 2063 616e 2062 6520 6d61  > plot can be ma
+00017ea0: 6465 2062 7920 6164 6469 6e67 2074 6865  de by adding the
+00017eb0: 206f 7074 696f 6e20 602d 2d63 6c61 7373   option `--class
+00017ec0: 6963 6020 746f 2074 6170 206d 6f72 6520  ic` to tap more 
+00017ed0: 6465 7461 696c 6564 6c79 2069 6e74 6f20  detailedly into 
+00017ee0: 7468 6520 7661 7269 6174 696f 6e20 6f66  the variation of
+00017ef0: 2073 796e 6f6e 796d 6f75 7320 7375 6273   synonymous subs
+00017f00: 7469 7475 7469 6f6e 2072 6174 652e 0a0a  titution rate...
+00017f10: 5573 696e 6720 7468 6520 636f 6d6d 616e  Using the comman
+00017f20: 6420 6265 6c6f 772c 2074 6865 2064 6972  d below, the dir
+00017f30: 6563 7469 6f6e 206f 6620 7261 7465 2063  ection of rate c
+00017f40: 6f72 7265 6374 696f 6e20 616e 6420 6465  orrection and de
+00017f50: 6772 6565 206f 6620 7261 7465 2076 6172  gree of rate var
+00017f60: 6961 7469 6f6e 2063 616e 2062 6520 6f62  iation can be ob
+00017f70: 7365 7276 6564 206d 6f72 6520 6469 7265  served more dire
+00017f80: 6374 6c79 2e0a 0a60 6060 0a77 6764 2076  ctly...```.wgd v
+00017f90: 697a 202d 6420 7767 645f 676c 6f62 616c  iz -d wgd_global
+00017fa0: 6d72 6268 5f6b 732f 676c 6f62 616c 5f4d  mrbh_ks/global_M
+00017fb0: 5242 482e 7473 762e 6b73 2e74 7376 202d  RBH.tsv.ks.tsv -
+00017fc0: 6661 2041 7175 696c 6567 6961 5f63 6f65  fa Aquilegia_coe
+00017fd0: 7275 6c65 6120 2d65 706b 2077 6764 5f6b  rulea -epk wgd_k
+00017fe0: 7364 2f41 7175 696c 6567 6961 5f63 6f65  sd/Aquilegia_coe
+00017ff0: 7275 6c65 612e 6b73 2e74 7376 202d 6170  rulea.ks.tsv -ap
+00018000: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+00018010: 2d6f 7574 2f61 6e63 686f 7270 6f69 6e74  -out/anchorpoint
+00018020: 732e 7478 7420 2d73 7020 7370 6563 6965  s.txt -sp specie
+00018030: 7374 7265 652e 6e77 202d 6f20 7767 645f  stree.nw -o wgd_
+00018040: 7669 7a5f 6d69 7865 645f 4b73 202d 2d70  viz_mixed_Ks --p
+00018050: 6c6f 7465 6c6d 6d20 2d2d 706c 6f74 6170  lotelmm --plotap
+00018060: 676d 6d20 2d2d 7265 7765 6967 6874 202d  gmm --reweight -
+00018070: 2d70 6c6f 746b 6465 202d 2d63 6c61 7373  -plotkde --class
+00018080: 6963 0a60 6060 0a0a 215b 5d28 6461 7461  ic.```..![](data
+00018090: 2f41 7175 696c 6567 6961 5f63 6f65 7275  /Aquilegia_coeru
+000180a0: 6c65 615f 476c 6f62 616c 6d72 6268 4b73  lea_GlobalmrbhKs
+000180b0: 5f45 6c6d 6d5f 4170 676d 6d5f 436f 7272  _Elmm_Apgmm_Corr
+000180c0: 6563 7465 642e 6b73 642e 7376 6729 0a0a  ected.ksd.svg)..
+000180d0: 4173 2073 686f 776e 2061 626f 7665 2c20  As shown above, 
+000180e0: 6265 6361 7573 6520 6f66 2074 6865 2068  because of the h
+000180f0: 6967 6865 7220 7375 6273 7469 7475 7469  igher substituti
+00018100: 6f6e 2072 6174 6520 6f66 202a 4171 7569  on rate of *Aqui
+00018110: 6c65 6769 6120 636f 6572 756c 6561 2a2c  legia coerulea*,
+00018120: 2074 6865 206f 7269 6769 6e61 6c20 6f72   the original or
+00018130: 7468 6f6c 6f67 6f75 7320 2a4b 2a3c 7375  thologous *K*<su
+00018140: 623e 533c 2f73 7562 3e20 7661 6c75 6573  b>S</sub> values
+00018150: 2077 6572 6520 6163 7475 616c 6c79 2075   were actually u
+00018160: 6e64 6572 6573 7469 6d61 7465 6420 696e  nderestimated in
+00018170: 2074 6865 2074 696d 652d 6672 616d 6520   the time-frame 
+00018180: 6f66 202a 4171 7569 6c65 6769 6120 636f  of *Aquilegia co
+00018190: 6572 756c 6561 2a2e 2057 6865 6e20 7765  erulea*. When we
+000181a0: 2072 6563 6f76 6572 6564 2074 6865 2064   recovered the d
+000181b0: 6976 6572 6765 6e63 6520 7375 6273 7469  ivergence substi
+000181c0: 7475 7469 6f6e 2064 6973 7461 6e63 6520  tution distance 
+000181d0: 696e 2074 6572 6d73 206f 6620 7477 6f20  in terms of two 
+000181e0: 7469 6d65 7320 6f66 2074 6865 2062 7261  times of the bra
+000181f0: 6e63 682d 7370 6563 6966 6963 2063 6f6e  nch-specific con
+00018200: 7472 6962 7574 696f 6e20 6f66 202a 412e  tribution of *A.
+00018210: 2063 6f65 7275 6c65 612a 2073 696e 6365   coerulea* since
+00018220: 2069 7473 2064 6976 6572 6765 6e63 6520   its divergence 
+00018230: 7769 7468 2074 6865 2073 6973 7465 7220  with the sister 
+00018240: 7370 6563 6965 7320 706c 7573 2074 6865  species plus the
+00018250: 2073 6861 7265 6420 7375 6273 7469 7475   shared substitu
+00018260: 7469 6f6e 2064 6973 7461 6e63 6520 6265  tion distance be
+00018270: 666f 7265 2064 6976 6572 6765 6e63 6520  fore divergence 
+00018280: 2869 6e20 7265 6c61 7469 7665 2074 6f20  (in relative to 
+00018290: 7468 6520 6f75 7467 726f 7570 292c 2074  the outgroup), t
+000182a0: 6865 2063 6f72 7265 6374 6564 202a 4b2a  he corrected *K*
+000182b0: 3c73 7562 3e53 3c2f 7375 623e 206d 6f64  <sub>S</sub> mod
+000182c0: 6520 6265 6361 6d65 206c 6172 6765 722e  e became larger.
+000182d0: 0a0a 4e6f 7465 2074 6861 7420 7765 2063  ..Note that we c
+000182e0: 616e 2065 6173 696c 7920 7368 6f77 2074  an easily show t
+000182f0: 6861 7420 2a41 7175 696c 6567 6961 2063  hat *Aquilegia c
+00018300: 6f65 7275 6c65 612a 2068 6173 2068 6967  oerulea* has hig
+00018310: 6865 7220 7375 6273 7469 7475 7469 6f6e  her substitution
+00018320: 2072 6174 6520 7468 616e 202a 5072 6f74   rate than *Prot
+00018330: 6561 2063 796e 6172 6f69 6465 732a 2061  ea cynaroides* a
+00018340: 6e64 202a 5669 7469 7320 7669 6e69 6665  nd *Vitis vinife
+00018350: 7261 2a20 6279 2063 6f6d 7061 7269 6e67  ra* by comparing
+00018360: 2074 6865 6972 2073 7562 7374 6974 7574   their substitut
+00018370: 696f 6e20 6469 7374 616e 6365 2069 6e20  ion distance in 
+00018380: 7265 6761 7264 2074 6f20 7468 6520 7361  regard to the sa
+00018390: 6d65 2064 6976 6572 6765 6e63 6520 6576  me divergence ev
+000183a0: 656e 7420 7769 7468 206f 7574 6772 6f75  ent with outgrou
+000183b0: 7020 7370 6563 6965 7320 2a41 636f 7275  p species *Acoru
+000183c0: 735f 616d 6572 6963 616e 7573 2a2c 2075  s_americanus*, u
+000183d0: 7369 6e67 2063 6f6d 6d61 6e64 2062 656c  sing command bel
+000183e0: 6f77 2e0a 0a60 6060 0a77 6764 2076 697a  ow...```.wgd viz
+000183f0: 202d 6420 7767 645f 676c 6f62 616c 6d72   -d wgd_globalmr
+00018400: 6268 5f6b 732f 676c 6f62 616c 5f4d 5242  bh_ks/global_MRB
+00018410: 482e 7473 762e 6b73 2e74 7376 202d 7370  H.tsv.ks.tsv -sp
+00018420: 2073 7065 6369 6573 7472 6565 2e6e 7720   speciestree.nw 
+00018430: 2d2d 7265 7765 6967 6874 202d 6f20 7767  --reweight -o wg
+00018440: 645f 7669 7a5f 436f 6d70 6172 655f 7261  d_viz_Compare_ra
+00018450: 7465 202d 2d73 7061 6972 2022 4163 6f72  te --spair "Acor
+00018460: 7573 5f61 6d65 7269 6361 6e75 733b 5072  us_americanus;Pr
+00018470: 6f74 6561 5f63 796e 6172 6f69 6465 7322  otea_cynaroides"
+00018480: 202d 2d73 7061 6972 2022 4171 7569 6c65   --spair "Aquile
+00018490: 6769 615f 636f 6572 756c 6561 3b41 636f  gia_coerulea;Aco
+000184a0: 7275 735f 616d 6572 6963 616e 7573 2220  rus_americanus" 
+000184b0: 2d2d 7370 6169 7220 2256 6974 6973 5f76  --spair "Vitis_v
+000184c0: 696e 6966 6572 613b 4163 6f72 7573 5f61  inifera;Acorus_a
+000184d0: 6d65 7269 6361 6e75 7322 202d 2d70 6c6f  mericanus" --plo
+000184e0: 746b 6465 202d 2d63 6c61 7373 6963 0a60  tkde --classic.`
+000184f0: 6060 0a0a 215b 5d28 6461 7461 2f52 6177  ``..![](data/Raw
+00018500: 5f4f 7274 686f 6c6f 6775 6573 5f43 6f6d  _Orthologues_Com
+00018510: 7061 7265 5f72 6174 652e 6b73 642e 7376  pare_rate.ksd.sv
+00018520: 6729 0a0a 4173 2064 6973 706c 6179 6564  g)..As displayed
+00018530: 2061 626f 7665 2c20 7468 6520 6f72 7468   above, the orth
+00018540: 6f6c 6f67 6f75 7320 2a4b 2a3c 7375 623e  ologous *K*<sub>
+00018550: 533c 2f73 7562 3e20 7661 6c75 6573 2062  S</sub> values b
+00018560: 6577 7465 656e 202a 4171 7569 6c65 6769  ewteen *Aquilegi
+00018570: 6120 636f 6572 756c 6561 2a20 616e 6420  a coerulea* and 
+00018580: 2a41 636f 7275 7320 616d 6572 6963 616e  *Acorus american
+00018590: 7573 2a20 6861 7320 7468 6520 6869 6768  us* has the high
+000185a0: 6573 7420 6d6f 6465 2c20 696e 6469 6361  est mode, indica
+000185b0: 7469 6e67 2074 6865 2066 6173 7465 7220  ting the faster 
+000185c0: 7375 6273 7469 7475 7469 6f6e 2072 6174  substitution rat
+000185d0: 6520 6f66 202a 412e 2063 6f65 7275 6c65  e of *A. coerule
+000185e0: 612a 2063 6f6d 7061 7265 6420 746f 202a  a* compared to *
+000185f0: 5072 6f74 6561 2063 796e 6172 6f69 6465  Protea cynaroide
+00018600: 732a 2061 6e64 202a 5669 7469 7320 7669  s* and *Vitis vi
+00018610: 6e69 6665 7261 2a2e 0a0a 4265 666f 7265  nifera*...Before
+00018620: 2076 322e 302e 3231 2c20 7468 6520 6765   v2.0.21, the ge
+00018630: 6e65 2d73 7065 6369 6573 206d 6170 2066  ne-species map f
+00018640: 696c 6520 6973 206e 6563 6365 7373 6172  ile is neccessar
+00018650: 696c 7920 6e65 6564 6564 2066 6f72 2069  ily needed for i
+00018660: 7473 2069 6d70 6c65 6d65 6e74 6174 696f  ts implementatio
+00018670: 6e20 696e 2060 7767 6420 7669 7a60 2c20  n in `wgd viz`, 
+00018680: 7768 6963 6820 7368 6f75 6c64 2062 6520  which should be 
+00018690: 6175 746f 6d61 7465 6c79 2070 726f 6475  automately produ
+000186a0: 6365 6420 6279 2074 6865 206c 6173 7420  ced by the last 
+000186b0: 6077 6764 206b 7364 6020 7374 6570 2067  `wgd ksd` step g
+000186c0: 6976 656e 2074 6865 2060 7370 6169 7260  iven the `spair`
+000186d0: 2061 6e64 2060 7370 6563 6965 7374 7265   and `speciestre
+000186e0: 6560 2070 6172 616d 6574 6572 732e 2054  e` parameters. T
+000186f0: 6865 2060 6765 6e65 5f73 7065 6369 6573  he `gene_species
+00018700: 2e6d 6170 6020 6861 7320 636f 6e74 656e  .map` has conten
+00018710: 7473 2061 7320 6265 6c6f 7720 696e 2077  ts as below in w
+00018720: 6869 6368 2065 6163 6820 6c69 6e65 2069  hich each line i
+00018730: 7320 7468 6520 6a6f 696e 6564 2073 7472  s the joined str
+00018740: 696e 6720 6f66 2067 656e 6520 6e61 6d65  ing of gene name
+00018750: 2061 6e64 2073 7065 6369 6573 206e 616d   and species nam
+00018760: 6520 6279 2073 7061 6365 2e20 4166 7465  e by space. Afte
+00018770: 7220 7632 2e30 2e32 3120 2869 6e63 6c75  r v2.0.21 (inclu
+00018780: 6465 6429 2c20 7468 6520 6765 6e65 2d73  ded), the gene-s
+00018790: 7065 6369 6573 206d 6170 2066 696c 6520  pecies map file 
+000187a0: 6973 206e 6f74 206e 6563 6365 7373 6172  is not neccessar
+000187b0: 696c 7920 6e65 6564 6564 2061 6e79 6d6f  ily needed anymo
+000187c0: 7265 2e0a 0a60 6060 0a41 7163 6f65 3647  re...```.Aqcoe6G
+000187d0: 3035 3738 3030 2e31 2041 7175 696c 6567  057800.1 Aquileg
+000187e0: 6961 5f63 6f65 7275 6c65 610a 5676 695f  ia_coerulea.Vvi_
+000187f0: 5649 545f 3230 3173 3030 3131 6730 3135  VIT_201s0011g015
+00018800: 3330 2e31 2056 6974 6973 5f76 696e 6966  30.1 Vitis_vinif
+00018810: 6572 610a 5063 795f 5072 6f63 7930 3167  era.Pcy_Procy01g
+00018820: 3038 3531 3020 5072 6f74 6561 5f63 796e  08510 Protea_cyn
+00018830: 6172 6f69 6465 730a 4161 6d5f 4163 6f72  aroides.Aam_Acor
+00018840: 612e 3034 4731 3432 3930 302e 3120 4163  a.04G142900.1 Ac
+00018850: 6f72 7573 5f61 6d65 7269 6361 6e75 730a  orus_americanus.
+00018860: 6060 600a 0a41 6e20 616c 7465 726e 6174  ```..An alternat
+00018870: 6976 6520 7761 7920 746f 2063 616c 6375  ive way to calcu
+00018880: 6c61 7465 2074 6865 206f 7274 686f 6c6f  late the ortholo
+00018890: 676f 7573 202a 4b2a 3c73 7562 3e53 3c2f  gous *K*<sub>S</
+000188a0: 7375 623e 2069 7320 746f 2064 6972 6563  sub> is to direc
+000188b0: 746c 7920 7573 6520 7468 6520 6f72 7468  tly use the orth
+000188c0: 6f67 726f 7570 7320 696e 7374 6561 6420  ogroups instead 
+000188d0: 6f66 2067 6c6f 6261 6c20 4d52 4248 2066  of global MRBH f
+000188e0: 616d 696c 792e 2054 6861 7420 7761 7920  amily. That way 
+000188f0: 7765 2064 6f6e 2774 2075 7365 2074 6865  we don't use the
+00018900: 2070 7265 2d69 6e66 6572 7265 6420 7061   pre-inferred pa
+00018910: 7261 6e6f 6d65 202a 4b2a 3c73 7562 3e53  ranome *K*<sub>S
+00018920: 3c2f 7375 623e 2062 7574 2074 6865 2070  </sub> but the p
+00018930: 6172 616c 6f67 6f75 7320 6765 6e65 2070  aralogous gene p
+00018940: 6169 7273 2069 6e73 6964 6520 6561 6368  airs inside each
+00018950: 206f 7274 686f 6772 6f75 7020 696e 7374   orthogroup inst
+00018960: 6561 642e 2054 6f20 6163 6869 6576 6520  ead. To achieve 
+00018970: 7468 6174 2c20 7765 2066 6972 7374 206e  that, we first n
+00018980: 6565 6420 746f 2069 6e66 6572 206f 7274  eed to infer ort
+00018990: 686f 6772 6f75 7073 2075 7369 6e67 2074  hogroups using t
+000189a0: 6865 2063 6f6d 6d61 6e64 2062 656c 6f77  he command below
+000189b0: 2e0a 0a60 6060 0a77 6764 2064 6d64 2041  ...```.wgd dmd A
+000189c0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+000189d0: 6120 5072 6f74 6561 5f63 796e 6172 6f69  a Protea_cynaroi
+000189e0: 6465 7320 4163 6f72 7573 5f61 6d65 7269  des Acorus_ameri
+000189f0: 6361 6e75 7320 5669 7469 735f 7669 6e69  canus Vitis_vini
+00018a00: 6665 7261 202d 2d6f 7274 686f 696e 6665  fera --orthoinfe
+00018a10: 7220 2d6f 2077 6764 5f6f 7274 686f 2028  r -o wgd_ortho (
+00018a20: 2d2d 6f6e 6c79 6f72 7468 6f29 200a 6060  --onlyortho) .``
+00018a30: 600a 0a55 7365 7273 2063 616e 2064 6563  `..Users can dec
+00018a40: 6964 6520 746f 206f 6e6c 7920 636f 6e64  ide to only cond
+00018a50: 7563 7420 7468 6520 6f72 7468 6f67 726f  uct the orthogro
+00018a60: 7570 2061 6e61 6c79 7369 7320 7768 696c  up analysis whil
+00018a70: 6520 736b 6970 7069 6e67 206f 7468 6572  e skipping other
+00018a80: 2061 6e61 6c79 7369 7320 6279 2061 6464   analysis by add
+00018a90: 696e 6720 7468 6520 666c 6167 2060 2d2d  ing the flag `--
+00018aa0: 6f6e 6c79 6f72 7468 6f60 2e20 4e65 7874  onlyortho`. Next
+00018ab0: 2073 7465 7020 6973 2074 6865 2073 616d   step is the sam
+00018ac0: 6520 7769 7468 2067 6c6f 6261 6c20 4d52  e with global MR
+00018ad0: 4248 2066 616d 696c 792e 0a0a 6060 600a  BH family...```.
+00018ae0: 7767 6420 6b73 6420 7767 645f 6f72 7468  wgd ksd wgd_orth
+00018af0: 6f2f 4f72 7468 6f67 726f 7570 732e 7370  o/Orthogroups.sp
+00018b00: 2e74 7376 2041 7175 696c 6567 6961 5f63  .tsv Aquilegia_c
+00018b10: 6f65 7275 6c65 6120 5072 6f74 6561 5f63  oerulea Protea_c
+00018b20: 796e 6172 6f69 6465 7320 4163 6f72 7573  ynaroides Acorus
+00018b30: 5f61 6d65 7269 6361 6e75 7320 5669 7469  _americanus Viti
+00018b40: 735f 7669 6e69 6665 7261 202d 6f20 7767  s_vinifera -o wg
+00018b50: 645f 6f72 7468 6f5f 6b73 0a77 6764 2076  d_ortho_ks.wgd v
+00018b60: 697a 202d 6420 7767 645f 6f72 7468 6f5f  iz -d wgd_ortho_
+00018b70: 6b73 2f4f 7274 686f 6772 6f75 7073 2e73  ks/Orthogroups.s
+00018b80: 702e 7473 762e 6b73 2e74 7376 202d 6570  p.tsv.ks.tsv -ep
+00018b90: 6b20 7767 645f 6b73 642f 4171 7569 6c65  k wgd_ksd/Aquile
+00018ba0: 6769 615f 636f 6572 756c 6561 2e6b 732e  gia_coerulea.ks.
+00018bb0: 7473 7620 2d73 7020 7370 6563 6965 7374  tsv -sp speciest
+00018bc0: 7265 652e 6e77 202d 2d72 6577 6569 6768  ree.nw --reweigh
+00018bd0: 7420 2d61 7020 7767 645f 7379 6e2f 6961  t -ap wgd_syn/ia
+00018be0: 6468 6f72 652d 6f75 742f 616e 6368 6f72  dhore-out/anchor
+00018bf0: 706f 696e 7473 2e74 7874 202d 706c 6f74  points.txt -plot
+00018c00: 656c 6d6d 202d 2d70 6c6f 7461 7067 6d6d  elmm --plotapgmm
+00018c10: 202d 6f20 7767 645f 6f72 7468 6f5f 6b73   -o wgd_ortho_ks
+00018c20: 5f72 6174 655f 636f 7272 6563 7469 6f6e  _rate_correction
+00018c30: 0a60 6060 0a0a 215b 5d28 6461 7461 2f4d  .```..![](data/M
+00018c40: 6978 6564 2e6b 732e 4171 7569 6c65 6769  ixed.ks.Aquilegi
+00018c50: 615f 636f 6572 756c 6561 2e6e 6f64 652e  a_coerulea.node.
+00018c60: 7765 6967 6874 6564 5f4f 7274 686f 5f4b  weighted_Ortho_K
+00018c70: 732e 706e 6729 0a21 5b5d 2864 6174 612f  s.png).![](data/
+00018c80: 466f 6375 735f 7369 7374 6572 5f70 6169  Focus_sister_pai
+00018c90: 7273 2e6b 732e 6e6f 6465 2e77 6569 6768  rs.ks.node.weigh
+00018ca0: 7465 645f 4f72 7468 6f5f 4b73 2e70 6e67  ted_Ortho_Ks.png
+00018cb0: 290a 215b 5d28 6461 7461 2f41 6c6c 5f70  ).![](data/All_p
+00018cc0: 6169 7273 2e6b 732e 6e6f 6465 2e77 6569  airs.ks.node.wei
+00018cd0: 6768 7465 645f 4f72 7468 6f5f 4b73 2e70  ghted_Ortho_Ks.p
+00018ce0: 6e67 290a 0a41 7320 7368 6f77 6e20 6162  ng)..As shown ab
+00018cf0: 6f76 652c 2074 6865 206e 756d 6265 7220  ove, the number 
+00018d00: 6f66 206f 7274 686f 6c6f 676f 7573 2067  of orthologous g
+00018d10: 656e 6520 7061 6972 7320 6973 2064 6966  ene pairs is dif
+00018d20: 6665 7265 6e74 2074 6861 6e20 7468 6520  ferent than the 
+00018d30: 6f6e 6520 6672 6f6d 2067 6c6f 6261 6c20  one from global 
+00018d40: 4d52 4248 2066 616d 696c 6965 7320 696e  MRBH families in
+00018d50: 2074 6861 7420 6865 7265 2077 6520 706c   that here we pl
+00018d60: 6f74 7465 6420 616c 6c20 6f72 7468 6f6c  otted all orthol
+00018d70: 6f67 6f75 7320 6765 6e65 2070 6169 7273  ogous gene pairs
+00018d80: 2069 6e73 7465 6164 206f 6620 6f6e 6c79   instead of only
+00018d90: 2067 6c6f 6261 6c20 4d52 4248 2066 616d   global MRBH fam
+00018da0: 696c 6965 732c 2074 6f67 6574 6865 7220  ilies, together 
+00018db0: 7769 7468 2064 6966 6665 7265 6e74 2072  with different r
+00018dc0: 6563 616c 6375 6c61 7465 6420 7765 6967  ecalculated weig
+00018dd0: 6874 732e 0a0a 4166 7465 7220 7468 6520  hts...After the 
+00018de0: 7068 796c 6f67 656e 6574 6963 2074 696d  phylogenetic tim
+00018df0: 696e 6720 6f66 2074 6865 2052 616e 756e  ing of the Ranun
+00018e00: 6375 6c61 6c65 7320 5747 442c 2077 6520  culales WGD, we 
+00018e10: 6361 6e20 6675 7274 6865 7220 696e 6665  can further infe
+00018e20: 7220 6974 7320 6162 736f 6c75 7465 2061  r its absolute a
+00018e30: 6765 2e20 4669 7273 7420 7765 2069 6e66  ge. First we inf
+00018e40: 6572 2074 6865 2063 7265 6469 626c 6520  er the credible 
+00018e50: 7261 6e67 6520 6f66 2061 6e63 686f 7220  range of anchor 
+00018e60: 7061 6972 7320 6279 202a 4b2a 3c73 7562  pairs by *K*<sub
+00018e70: 3e53 3c2f 7375 623e 2068 6575 7269 7374  >S</sub> heurist
+00018e80: 6963 616c 6c79 2075 7369 6e67 2074 6865  ically using the
+00018e90: 2070 726f 6772 616d 2060 7767 6420 7065   program `wgd pe
+00018ea0: 616b 602e 0a0a 6060 600a 7767 6420 7065  ak`...```.wgd pe
+00018eb0: 616b 202d 2d68 6575 7269 7374 6963 2077  ak --heuristic w
+00018ec0: 6764 5f6b 7364 2f41 7175 696c 6567 6961  gd_ksd/Aquilegia
+00018ed0: 5f63 6f65 7275 6c65 612e 7473 762e 6b73  _coerulea.tsv.ks
+00018ee0: 2e74 7376 202d 6170 2077 6764 5f73 796e  .tsv -ap wgd_syn
+00018ef0: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
+00018f00: 686f 7270 6f69 6e74 732e 7478 7420 2d73  horpoints.txt -s
+00018f10: 6d20 7767 645f 7379 6e2f 6961 6468 6f72  m wgd_syn/iadhor
+00018f20: 652d 6f75 742f 7365 676d 656e 7473 2e74  e-out/segments.t
+00018f30: 7874 202d 6c65 2077 6764 5f73 796e 2f69  xt -le wgd_syn/i
+00018f40: 6164 686f 7265 2d6f 7574 2f6c 6973 745f  adhore-out/list_
+00018f50: 656c 656d 656e 7473 2e74 7874 202d 6d70  elements.txt -mp
+00018f60: 2077 6764 5f73 796e 2f69 6164 686f 7265   wgd_syn/iadhore
+00018f70: 2d6f 7574 2f6d 756c 7469 706c 6963 6f6e  -out/multiplicon
+00018f80: 5f70 6169 7273 2e74 7874 202d 6f20 7767  _pairs.txt -o wg
+00018f90: 645f 7065 616b 0a60 6060 0a0a 215b 5d28  d_peak.```..![](
+00018fa0: 6461 7461 2f41 6e63 686f 724b 735f 5065  data/AnchorKs_Pe
+00018fb0: 616b 4349 5f41 7175 696c 6567 6961 5f63  akCI_Aquilegia_c
+00018fc0: 6f65 7275 6c65 612e 7473 762e 6b73 2e74  oerulea.tsv.ks.t
+00018fd0: 7376 5f6e 6f64 655f 7765 6967 6874 6564  sv_node_weighted
+00018fe0: 2e73 7667 290a 0a41 7320 7368 6f77 6e20  .svg)..As shown 
+00018ff0: 6162 6f76 652c 2077 6520 6173 7375 6d65  above, we assume
+00019000: 6420 6120 6c6f 676e 6f72 6d61 6c20 6469  d a lognormal di
+00019010: 7374 7269 6275 7469 6f6e 2061 7420 7468  stribution at th
+00019020: 6520 7065 616b 206c 6f63 6174 696f 6e20  e peak location 
+00019030: 6465 7465 6374 6564 2062 7920 7468 6520  detected by the 
+00019040: 6073 6967 6e61 6c60 206d 6f64 756c 6520  `signal` module 
+00019050: 6f66 2060 7363 6970 7960 206c 6962 7261  of `scipy` libra
+00019060: 7279 2e20 5468 6520 3935 2520 636f 6e66  ry. The 95% conf
+00019070: 6964 656e 6365 206c 6576 656c 206f 6620  idence level of 
+00019080: 7468 6520 6c6f 676e 6f72 6d61 6c20 6469  the lognormal di
+00019090: 7374 7269 6275 7469 6f6e 2077 6173 2061  stribution was a
+000190a0: 7070 6c69 6564 2c20 692e 652e 2c20 302e  pplied, i.e., 0.
+000190b0: 3638 2d32 2e37 342c 2069 6e20 6675 7274  68-2.74, in furt
+000190c0: 6865 7220 6d6f 6c65 6375 6c61 7220 6461  her molecular da
+000190d0: 7469 6e67 2e20 5468 6520 6669 6c65 2060  ting. The file `
+000190e0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000190f0: 6561 2e74 7376 2e6b 732e 7473 765f 3935  ea.tsv.ks.tsv_95
+00019100: 2543 495f 4150 5f66 6f72 5f64 6174 696e  %CI_AP_for_datin
+00019110: 675f 7765 6967 6874 6564 5f66 6f72 6d61  g_weighted_forma
+00019120: 742e 7473 7660 2069 7320 7768 6174 2077  t.tsv` is what w
+00019130: 6520 6e65 6564 2066 6f72 206e 6578 7420  e need for next 
+00019140: 7374 6570 2e20 546f 2062 7569 6c64 2074  step. To build t
+00019150: 6865 206f 7274 686f 6772 6f75 7073 2075  he orthogroups u
+00019160: 7365 6420 696e 2070 6879 6c6f 6765 6e65  sed in phylogene
+00019170: 7469 6320 6461 7469 6e67 2c20 7765 206e  tic dating, we n
+00019180: 6565 6420 746f 2073 656c 6563 7420 736f  eed to select so
+00019190: 6d65 2073 7065 6369 6573 2061 6e64 2066  me species and f
+000191a0: 6f72 6d20 6120 7374 6172 7469 6e67 2074  orm a starting t
+000191b0: 7265 6520 7769 7468 2070 726f 7065 7220  ree with proper 
+000191c0: 666f 7373 696c 2063 616c 6962 7261 7469  fossil calibrati
+000191d0: 6f6e 732e 2057 6520 7072 6f76 6964 6520  ons. We provide 
+000191e0: 6f6e 6520 696e 206d 636d 6374 7265 6520  one in mcmctree 
+000191f0: 666f 726d 6174 2061 7320 6265 6c6f 772e  format as below.
+00019200: 0a0a 6060 600a 3137 2031 0a28 2828 2850  ..```.17 1.((((P
+00019210: 6f74 616d 6f67 6574 6f6e 5f61 6375 7469  otamogeton_acuti
+00019220: 666f 6c69 7573 2c28 5370 6972 6f64 656c  folius,(Spirodel
+00019230: 615f 696e 7465 726d 6564 6961 2c41 6d6f  a_intermedia,Amo
+00019240: 7270 686f 7068 616c 6c75 735f 6b6f 6e6a  rphophallus_konj
+00019250: 6163 2929 2c28 4163 616e 7468 6f63 686c  ac)),(Acanthochl
+00019260: 616d 7973 5f62 7261 6374 6561 7461 2c28  amys_bracteata,(
+00019270: 4469 6f73 636f 7265 615f 616c 6174 612c  Dioscorea_alata,
+00019280: 4469 6f73 636f 7265 615f 726f 7475 6e64  Dioscorea_rotund
+00019290: 6174 6129 2927 3e30 2e35 3630 303c 312e  ata))'>0.5600<1.
+000192a0: 3238 3633 2729 273e 302e 3833 3630 3c31  2863')'>0.8360<1
+000192b0: 2e32 3836 3327 2c28 4163 6f72 7573 5f61  .2863',(Acorus_a
+000192c0: 6d65 7269 6361 6e75 732c 4163 6f72 7573  mericanus,Acorus
+000192d0: 5f74 6174 6172 696e 6f77 6969 2929 273e  _tatarinowii))'>
+000192e0: 302e 3833 3630 3c31 2e32 3836 3327 2c28  0.8360<1.2863',(
+000192f0: 2828 2854 6574 7261 6365 6e74 726f 6e5f  (((Tetracentron_
+00019300: 7369 6e65 6e73 652c 5472 6f63 686f 6465  sinense,Trochode
+00019310: 6e64 726f 6e5f 6172 616c 696f 6964 6573  ndron_aralioides
+00019320: 292c 2842 7578 7573 5f61 7573 7472 6f79  ),(Buxus_austroy
+00019330: 756e 6e61 6e65 6e73 6973 2c42 7578 7573  unnanensis,Buxus
+00019340: 5f73 696e 6963 6129 2927 3e31 2e31 3038  _sinica))'>1.108
+00019350: 303c 312e 3238 3633 272c 284e 656c 756d  0<1.2863',(Nelum
+00019360: 626f 5f6e 7563 6966 6572 612c 2854 656c  bo_nucifera,(Tel
+00019370: 6f70 6561 5f73 7065 6369 6f73 6973 7369  opea_speciosissi
+00019380: 6d61 2c50 726f 7465 615f 6379 6e61 726f  ma,Protea_cynaro
+00019390: 6964 6573 2929 2927 3e31 2e31 3038 303c  ides)))'>1.1080<
+000193a0: 312e 3238 3633 272c 2841 7175 696c 6567  1.2863',(Aquileg
+000193b0: 6961 5f63 6f65 7275 6c65 615f 6170 312c  ia_coerulea_ap1,
+000193c0: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+000193d0: 6561 5f61 7032 2929 273e 312e 3130 3830  ea_ap2))'>1.1080
+000193e0: 3c31 2e32 3836 3327 2927 3e31 2e32 3732  <1.2863')'>1.272
+000193f0: 303c 322e 3437 3230 273b 0a60 6060 0a0a  0<2.4720';.```..
+00019400: 4173 2070 7265 7365 6e74 6564 2061 626f  As presented abo
+00019410: 7665 2c20 7468 6520 666f 6375 7320 7370  ve, the focus sp
+00019420: 6563 6965 7320 7468 6174 2069 7320 6162  ecies that is ab
+00019430: 6f75 7420 746f 2062 6520 6461 7465 6420  out to be dated 
+00019440: 6e65 6564 7320 746f 2062 6520 7265 706c  needs to be repl
+00019450: 6163 6564 2077 6974 6820 6028 4171 7569  aced with `(Aqui
+00019460: 6c65 6769 615f 636f 6572 756c 6561 5f61  legia_coerulea_a
+00019470: 7031 2c41 7175 696c 6567 6961 5f63 6f65  p1,Aquilegia_coe
+00019480: 7275 6c65 615f 6170 3229 602e 2057 6974  rulea_ap2)`. Wit
+00019490: 6820 7468 6973 2073 7461 7274 696e 6720  h this starting 
+000194a0: 7472 6565 2061 6e64 2070 7265 646f 776e  tree and predown
+000194b0: 6c6f 6164 6564 2063 6473 2066 696c 6573  loaded cds files
+000194c0: 206f 6620 616c 6c20 7468 6520 7370 6563   of all the spec
+000194d0: 6965 732c 2077 6520 6361 6e20 6275 696c  ies, we can buil
+000194e0: 6420 7468 6520 6f72 7468 6f67 726f 7570  d the orthogroup
+000194f0: 2075 7365 6420 696e 2074 6865 2066 696e   used in the fin
+00019500: 616c 206d 6f6c 6563 756c 6172 2064 6174  al molecular dat
+00019510: 696e 6720 7573 696e 6720 7468 6520 636f  ing using the co
+00019520: 6d6d 616e 6420 6173 2062 656c 6f77 2e0a  mmand as below..
+00019530: 0a60 6060 0a77 6764 2064 6d64 202d 6620  .```.wgd dmd -f 
+00019540: 4171 7569 6c65 6769 615f 636f 6572 756c  Aquilegia_coerul
+00019550: 6561 202d 6170 2077 6764 5f70 6561 6b2f  ea -ap wgd_peak/
+00019560: 416e 6368 6f72 4b73 5f46 696e 6450 6561  AnchorKs_FindPea
+00019570: 6b2f 4171 7569 6c65 6769 615f 636f 6572  k/Aquilegia_coer
+00019580: 756c 6561 2e74 7376 2e6b 732e 7473 765f  ulea.tsv.ks.tsv_
+00019590: 3935 2543 495f 4150 5f66 6f72 5f64 6174  95%CI_AP_for_dat
+000195a0: 696e 675f 7765 6967 6874 6564 5f66 6f72  ing_weighted_for
+000195b0: 6d61 742e 7473 7620 2d6f 2077 6764 5f64  mat.tsv -o wgd_d
+000195c0: 6d64 5f6f 7274 686f 2050 6f74 616d 6f67  md_ortho Potamog
+000195d0: 6574 6f6e 5f61 6375 7469 666f 6c69 7573  eton_acutifolius
+000195e0: 2053 7069 726f 6465 6c61 5f69 6e74 6572   Spirodela_inter
+000195f0: 6d65 6469 6120 416d 6f72 7068 6f70 6861  media Amorphopha
+00019600: 6c6c 7573 5f6b 6f6e 6a61 6320 4163 616e  llus_konjac Acan
+00019610: 7468 6f63 686c 616d 7973 5f62 7261 6374  thochlamys_bract
+00019620: 6561 7461 2044 696f 7363 6f72 6561 5f61  eata Dioscorea_a
+00019630: 6c61 7461 2044 696f 7363 6f72 6561 5f72  lata Dioscorea_r
+00019640: 6f74 756e 6461 7461 2041 636f 7275 735f  otundata Acorus_
+00019650: 616d 6572 6963 616e 7573 2041 636f 7275  americanus Acoru
+00019660: 735f 7461 7461 7269 6e6f 7769 6920 5465  s_tatarinowii Te
+00019670: 7472 6163 656e 7472 6f6e 5f73 696e 656e  tracentron_sinen
+00019680: 7365 2054 726f 6368 6f64 656e 6472 6f6e  se Trochodendron
+00019690: 5f61 7261 6c69 6f69 6465 7320 4275 7875  _aralioides Buxu
+000196a0: 735f 6175 7374 726f 7975 6e6e 616e 656e  s_austroyunnanen
+000196b0: 7369 7320 4275 7875 735f 7369 6e69 6361  sis Buxus_sinica
+000196c0: 204e 656c 756d 626f 5f6e 7563 6966 6572   Nelumbo_nucifer
+000196d0: 6120 5465 6c6f 7065 615f 7370 6563 696f  a Telopea_specio
+000196e0: 7369 7373 696d 6120 5072 6f74 6561 5f63  sissima Protea_c
+000196f0: 796e 6172 6f69 6465 7320 4171 7569 6c65  ynaroides Aquile
+00019700: 6769 615f 636f 6572 756c 6561 0a60 6060  gia_coerulea.```
+00019710: 0a0a 5468 6520 7265 7375 6c74 2066 696c  ..The result fil
+00019720: 6520 606d 6572 6765 5f66 6f63 7573 5f61  e `merge_focus_a
+00019730: 702e 7473 7660 2069 7320 7768 6174 2077  p.tsv` is what w
+00019740: 6520 6e65 6564 2066 6f72 2074 6865 2066  e need for the f
+00019750: 696e 616c 2073 7465 7020 6f66 206d 6f6c  inal step of mol
+00019760: 6563 756c 6172 2064 6174 696e 6720 696e  ecular dating in
+00019770: 2070 726f 6772 616d 2060 7767 6420 666f   program `wgd fo
+00019780: 6375 7360 2e0a 0a60 6060 0a77 6764 2066  cus`...```.wgd f
+00019790: 6f63 7573 202d 2d70 726f 7463 6f63 6461  ocus --protcocda
+000197a0: 7469 6e67 202d 2d61 616d 6f64 656c 206c  ting --aamodel l
+000197b0: 6720 7767 645f 646d 645f 6f72 7468 6f2f  g wgd_dmd_ortho/
+000197c0: 6d65 7267 655f 666f 6375 735f 6170 2e74  merge_focus_ap.t
+000197d0: 7376 202d 7370 2064 6174 696e 675f 7472  sv -sp dating_tr
+000197e0: 6565 2e6e 7720 2d6f 2077 6764 5f64 6174  ee.nw -o wgd_dat
+000197f0: 696e 6720 2d64 206d 636d 6374 7265 6520  ing -d mcmctree 
+00019800: 2d64 7320 2762 7572 6e69 6e20 3d20 3230  -ds 'burnin = 20
+00019810: 3030 2720 2d64 7320 2773 616d 7066 7265  00' -ds 'sampfre
+00019820: 7120 3d20 3130 3030 2720 2d64 7320 276e  q = 1000' -ds 'n
+00019830: 7361 6d70 6c65 203d 2032 3030 3030 2720  sample = 20000' 
+00019840: 506f 7461 6d6f 6765 746f 6e5f 6163 7574  Potamogeton_acut
+00019850: 6966 6f6c 6975 7320 5370 6972 6f64 656c  ifolius Spirodel
+00019860: 615f 696e 7465 726d 6564 6961 2041 6d6f  a_intermedia Amo
+00019870: 7270 686f 7068 616c 6c75 735f 6b6f 6e6a  rphophallus_konj
+00019880: 6163 2041 6361 6e74 686f 6368 6c61 6d79  ac Acanthochlamy
+00019890: 735f 6272 6163 7465 6174 6120 4469 6f73  s_bracteata Dios
+000198a0: 636f 7265 615f 616c 6174 6120 4469 6f73  corea_alata Dios
+000198b0: 636f 7265 615f 726f 7475 6e64 6174 6120  corea_rotundata 
+000198c0: 4163 6f72 7573 5f61 6d65 7269 6361 6e75  Acorus_americanu
+000198d0: 7320 4163 6f72 7573 5f74 6174 6172 696e  s Acorus_tatarin
+000198e0: 6f77 6969 2054 6574 7261 6365 6e74 726f  owii Tetracentro
+000198f0: 6e5f 7369 6e65 6e73 6520 5472 6f63 686f  n_sinense Trocho
+00019900: 6465 6e64 726f 6e5f 6172 616c 696f 6964  dendron_aralioid
+00019910: 6573 2042 7578 7573 5f61 7573 7472 6f79  es Buxus_austroy
+00019920: 756e 6e61 6e65 6e73 6973 2042 7578 7573  unnanensis Buxus
+00019930: 5f73 696e 6963 6120 4e65 6c75 6d62 6f5f  _sinica Nelumbo_
+00019940: 6e75 6369 6665 7261 2054 656c 6f70 6561  nucifera Telopea
+00019950: 5f73 7065 6369 6f73 6973 7369 6d61 2050  _speciosissima P
+00019960: 726f 7465 615f 6379 6e61 726f 6964 6573  rotea_cynaroides
+00019970: 2041 7175 696c 6567 6961 5f63 6f65 7275   Aquilegia_coeru
+00019980: 6c65 610a 6060 600a 0a48 6572 6520 7765  lea.```..Here we
+00019990: 206f 6e6c 7920 696d 706c 656d 656e 7465   only implemente
+000199a0: 6420 7468 6520 636f 6e63 6174 656e 6174  d the concatenat
+000199b0: 696f 6e20 616e 616c 7973 6973 2075 7369  ion analysis usi
+000199c0: 6e67 2070 726f 7465 696e 2073 6571 7565  ng protein seque
+000199d0: 6e63 6520 6279 2061 6464 696e 6720 7468  nce by adding th
+000199e0: 6520 666c 6167 2060 2d2d 7072 6f74 6461  e flag `--protda
+000199f0: 7469 6e67 6020 616e 6420 7765 2073 6574  ting` and we set
+00019a00: 2074 6865 2070 6172 616d 6574 6572 2066   the parameter f
+00019a10: 6f72 2060 6d63 6d63 7472 6565 6020 7669  or `mcmctree` vi
+00019a20: 6120 7468 6520 6f70 7469 6f6e 2060 2d64  a the option `-d
+00019a30: 7360 2e20 4e6f 7465 2074 6861 7420 6f74  s`. Note that ot
+00019a40: 6865 7220 6461 7469 6e67 2070 726f 6772  her dating progr
+00019a50: 616d 2073 7563 6820 6173 2060 7238 7360  am such as `r8s`
+00019a60: 2061 6e64 2060 6265 6173 7460 2061 7265   and `beast` are
+00019a70: 2061 6c73 6f20 6176 6169 6c61 626c 6520   also available 
+00019a80: 6769 7665 6e20 736f 6d65 206d 616e 6461  given some manda
+00019a90: 746f 7279 2070 6172 616d 6574 6572 732e  tory parameters.
+00019aa0: 2054 6865 2066 696e 616c 206c 6f67 206f   The final log o
+00019ab0: 6620 7468 6520 7375 6363 6573 7366 756c  f the successful
+00019ac0: 2072 756e 2069 7320 6173 2062 656c 6f77   run is as below
+00019ad0: 2e0a 0a60 6060 0a31 363a 3034 3a32 3520  ...```.16:04:25 
+00019ae0: 494e 464f 2020 2020 2052 756e 6e69 6e67  INFO     Running
+00019af0: 206d 636d 6374 7265 6520 7573 696e 6720   mcmctree using 
+00019b00: 4865 7373 6961 6e20 6d61 7472 6978 206f  Hessian matrix o
+00019b10: 6620 4c47 2b47 616d 6d61 2020 636f 7265  f LG+Gamma  core
+00019b20: 2e70 793a 3936 370a 2020 2020 2020 2020  .py:967.        
+00019b30: 2020 2020 2020 2020 2020 666f 7220 7072            for pr
+00019b40: 6f74 6569 6e20 6d6f 6465 6c0a 3233 3a34  otein model.23:4
+00019b50: 393a 3337 2049 4e46 4f20 2020 2020 506f  9:37 INFO     Po
+00019b60: 7374 6572 696f 7220 6d65 616e 2066 6f72  sterior mean for
+00019b70: 2074 6865 2061 6765 7320 6f66 2077 6764   the ages of wgd
+00019b80: 2069 7320 3131 322e 3839 3435 206d 636d   is 112.8945 mcm
+00019b90: 6374 7265 652e 7079 3a32 3936 0a20 2020  ctree.py:296.   
+00019ba0: 2020 2020 2020 2020 2020 2020 2020 206d                 m
+00019bb0: 696c 6c69 6f6e 2079 6561 7273 2066 726f  illion years fro
+00019bc0: 6d20 436f 6e63 6174 656e 6174 6564 2070  m Concatenated p
+00019bd0: 6570 7469 6465 0a20 2020 2020 2020 2020  eptide.         
+00019be0: 2020 2020 2020 2020 2061 6c69 676e 6d65           alignme
+00019bf0: 6e74 2061 6e64 2039 3525 2063 7265 6469  nt and 95% credi
+00019c00: 6269 6c69 7479 2069 6e74 6572 7661 6c73  bility intervals
+00019c10: 2028 4349 290a 2020 2020 2020 2020 2020   (CI).          
+00019c20: 2020 2020 2020 2020 6973 2031 3031 2e32          is 101.2
+00019c30: 3234 2d31 3233 2e31 3231 206d 696c 6c69  24-123.121 milli
+00019c40: 6f6e 2079 6561 7273 0a20 2020 2020 2020  on years.       
+00019c50: 2020 494e 464f 2020 2020 2054 6f74 616c    INFO     Total
+00019c60: 2072 756e 2074 696d 653a 2032 3931 3735   run time: 29175
+00019c70: 7320 2020 2020 2020 2020 2020 2020 2020  s               
+00019c80: 2020 2020 2020 2020 2020 2020 2020 2063                 c
+00019c90: 6c69 2e70 793a 3234 310a 2020 2020 2020  li.py:241.      
+00019ca0: 2020 2049 4e46 4f20 2020 2020 446f 6e65     INFO     Done
+00019cb0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019cc0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019cd0: 2020 2020 2020 2020 2020 2020 2020 2020                  
+00019ce0: 636c 692e 7079 3a32 3432 0a60 6060 0a0a  cli.py:242.```..
+00019cf0: 546f 2076 6973 7561 6c69 7a65 2074 6865  To visualize the
+00019d00: 2064 6174 652c 2077 6520 616c 736f 2070   date, we also p
+00019d10: 726f 7669 6465 6420 6120 7079 7468 6f6e  rovided a python
+00019d20: 2073 6372 6970 7420 746f 2070 6c6f 7420   script to plot 
+00019d30: 7468 6520 5747 4420 6461 7465 7320 696e  the WGD dates in
+00019d40: 2074 6865 2060 7767 6460 2066 6f6c 6465   the `wgd` folde
+00019d50: 722e 2055 7365 7273 206e 6565 6420 746f  r. Users need to
+00019d60: 2065 7874 7261 6374 2074 6865 2072 6177   extract the raw
+00019d70: 2064 6174 6573 2066 726f 6d20 7468 6520   dates from the 
+00019d80: 606d 636d 632e 7478 7460 2066 6f72 2074  `mcmc.txt` for t
+00019d90: 6865 2057 4744 206e 6f64 6520 6669 7273  he WGD node firs
+00019da0: 7420 616e 6420 7361 7665 2069 7420 6173  t and save it as
+00019db0: 2066 696c 6520 6064 6174 6573 2e74 7874   file `dates.txt
+00019dc0: 6020 286f 7220 7768 6174 6576 6572 2070  ` (or whatever p
+00019dd0: 7265 6665 7272 6564 206e 616d 6529 2e20  referred name). 
+00019de0: 416e 2065 7861 6d70 6c65 2063 6f6d 6d61  An example comma
+00019df0: 6e64 2069 7320 6173 2062 656c 6f77 2e0a  nd is as below..
+00019e00: 0a60 6060 0a70 7974 686f 6e20 2450 4154  .```.python $PAT
+00019e10: 482f 706f 7374 706c 6f74 2e70 7920 706f  H/postplot.py po
+00019e20: 7374 6469 7320 6461 7465 732e 7478 7420  stdis dates.txt 
+00019e30: 2d2d 7065 7263 656e 7469 6c65 2039 3020  --percentile 90 
+00019e40: 2d2d 7469 746c 6520 2257 4744 2064 6174  --title "WGD dat
+00019e50: 6522 202d 2d68 7064 202d 6f20 2252 616e  e" --hpd -o "Ran
+00019e60: 756e 6375 6c61 6c65 735f 5747 445f 6461  unculales_WGD_da
+00019e70: 7465 2e73 7667 220a 6060 600a 0a21 5b5d  te.svg".```..![]
+00019e80: 2864 6174 612f 5261 6e75 6e63 756c 616c  (data/Ranunculal
+00019e90: 6573 5f57 4744 5f64 6174 652e 7376 6729  es_WGD_date.svg)
+00019ea0: 0a0a 5468 6520 706f 7374 6572 696f 7220  ..The posterior 
+00019eb0: 6d65 616e 2c20 6d65 6469 616e 2061 6e64  mean, median and
+00019ec0: 206d 6f64 6520 6f66 2074 6865 2052 616e   mode of the Ran
+00019ed0: 756e 6375 6c61 6c65 7320 5747 4420 6167  unculales WGD ag
+00019ee0: 6520 6973 2031 3132 2e39 322c 2031 3133  e is 112.92, 113
+00019ef0: 2e34 3420 616e 6420 3131 322e 3534 206d  .44 and 112.54 m
+00019f00: 7961 2c20 7769 7468 2039 3025 2048 5044  ya, with 90% HPD
+00019f10: 2031 3035 2e30 3720 2d20 3132 322e 3332   105.07 - 122.32
+00019f20: 206d 7961 2061 7320 6d61 6e69 6665 7374   mya as manifest
+00019f30: 6564 2061 626f 7665 2e0a 0a23 2323 204b  ed above...### K
+00019f40: 7374 7265 650a 0a49 6e20 6164 6469 7469  stree..In additi
+00019f50: 6f6e 2074 6f20 7061 6972 7769 7365 202a  on to pairwise *
+00019f60: 4b2a 3c73 7562 3e53 3c2f 7375 623e 2065  K*<sub>S</sub> e
+00019f70: 7374 696d 6174 696f 6e2c 2061 202a 4b2a  stimation, a *K*
+00019f80: 3c73 7562 3e53 3c2f 7375 623e 2074 7265  <sub>S</sub> tre
+00019f90: 6520 7769 7468 2062 7261 6e63 6820 6c65  e with branch le
+00019fa0: 6e67 7468 2069 6e20 2a4b 2a3c 7375 623e  ngth in *K*<sub>
+00019fb0: 533c 2f73 7562 3e20 756e 6974 2063 616e  S</sub> unit can
+00019fc0: 2061 6c73 6f20 6265 2064 6572 6976 6564   also be derived
+00019fd0: 2066 726f 6d20 7468 6520 7072 6f67 7261   from the progra
+00019fe0: 6d20 6077 6764 206b 7364 6020 6769 7665  m `wgd ksd` give
+00019ff0: 6e20 7468 6520 6f70 7469 6f6e 2060 2d2d  n the option `--
+0001a000: 6b73 7472 6565 6020 616e 6420 602d 2d73  kstree` and `--s
+0001a010: 7065 6369 6573 7472 6565 602e 204e 6f74  peciestree`. Not
+0001a020: 6520 7468 6174 2074 6865 2061 6464 6974  e that the addit
+0001a030: 696f 6e61 6c20 6f70 7469 6f6e 2060 2d2d  ional option `--
+0001a040: 6f6e 6c79 636f 6e63 6174 6b73 7472 6565  onlyconcatkstree
+0001a050: 6020 7769 6c6c 206f 6e6c 7920 6361 6c6c  ` will only call
+0001a060: 2074 6865 202a 4b2a 3c73 7562 3e53 3c2f   the *K*<sub>S</
+0001a070: 7375 623e 2065 7374 696d 6174 696f 6e20  sub> estimation 
+0001a080: 666f 7220 7468 6520 636f 6e63 6174 656e  for the concaten
+0001a090: 6174 6564 2061 6c69 676e 6d65 6e74 2072  ated alignment r
+0001a0a0: 6174 6865 7220 7468 616e 2061 6c6c 2074  ather than all t
+0001a0b0: 6865 2061 6c69 676e 6d65 6e74 732e 2055  he alignments. U
+0001a0c0: 7365 7273 206e 6565 6420 746f 2070 726f  sers need to pro
+0001a0d0: 7669 6465 2061 2070 7265 7365 7420 7370  vide a preset sp
+0001a0e0: 6563 6965 7320 7472 6565 2066 6f72 2074  ecies tree for t
+0001a0f0: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0001a100: 623e 2074 7265 6520 696e 6665 7265 6e63  b> tree inferenc
+0001a110: 6520 6f66 2074 6865 2063 6f6e 6361 7465  e of the concate
+0001a120: 6e61 7465 6420 616c 6967 6e6d 656e 7420  nated alignment 
+0001a130: 7768 696c 6520 7468 6520 7265 6d61 696e  while the remain
+0001a140: 696e 6720 616c 6967 6e6d 656e 7473 2077  ing alignments w
+0001a150: 696c 6c20 6265 2061 6761 696e 7374 2061  ill be against a
+0001a160: 6e20 6175 746f 6d61 7465 6c79 2069 6e66  n automately inf
+0001a170: 6572 7265 6420 7472 6565 2066 726f 6d20  erred tree from 
+0001a180: 6066 6173 7474 7265 6560 206f 7220 6069  `fasttree` or `i
+0001a190: 7174 7265 6560 2e20 496e 2074 6865 2065  qtree`. In the e
+0001a1a0: 6e64 2c20 7573 6572 7320 7769 6c6c 2067  nd, users will g
+0001a1b0: 6574 2061 202a 4b2a 3c73 7562 3e53 3c2f  et a *K*<sub>S</
+0001a1c0: 7375 623e 2074 7265 652c 2061 202a 4b2a  sub> tree, a *K*
+0001a1d0: 3c73 7562 3e41 3c2f 7375 623e 2074 7265  <sub>A</sub> tre
+0001a1e0: 6520 616e 6420 6120 cf89 2074 7265 6520  e and a .. tree 
+0001a1f0: 7065 7220 6661 6d20 616e 6420 666f 7220  per fam and for 
+0001a200: 7468 6520 636f 6e63 6174 656e 6174 6564  the concatenated
+0001a210: 2061 6c69 676e 6d65 6e74 2e0a 0a60 6060   alignment...```
+0001a220: 0a77 6764 206b 7364 2064 6174 612f 6b73  .wgd ksd data/ks
+0001a230: 7472 6565 5f64 6174 612f 6661 6d2e 7473  tree_data/fam.ts
+0001a240: 7620 6461 7461 2f6b 7374 7265 655f 6461  v data/kstree_da
+0001a250: 7461 2f41 636f 7275 735f 7461 7461 7269  ta/Acorus_tatari
+0001a260: 6e6f 7769 6920 6461 7461 2f6b 7374 7265  nowii data/kstre
+0001a270: 655f 6461 7461 2f41 6d62 6f72 656c 6c61  e_data/Amborella
+0001a280: 5f74 7269 6368 6f70 6f64 6120 6461 7461  _trichopoda data
+0001a290: 2f6b 7374 7265 655f 6461 7461 2f41 7175  /kstree_data/Aqu
+0001a2a0: 696c 6567 6961 5f63 6f65 7275 6c65 6120  ilegia_coerulea 
+0001a2b0: 6461 7461 2f6b 7374 7265 655f 6461 7461  data/kstree_data
+0001a2c0: 2f41 7269 7374 6f6c 6f63 6869 615f 6669  /Aristolochia_fi
+0001a2d0: 6d62 7269 6174 6120 6461 7461 2f6b 7374  mbriata data/kst
+0001a2e0: 7265 655f 6461 7461 2f43 7963 6173 5f70  ree_data/Cycas_p
+0001a2f0: 616e 7a68 6968 7561 656e 7369 202d 2d6b  anzhihuaensi --k
+0001a300: 7374 7265 6520 2d2d 7370 6563 6965 7374  stree --speciest
+0001a310: 7265 6520 6461 7461 2f6b 7374 7265 655f  ree data/kstree_
+0001a320: 6461 7461 2f73 7065 6369 6573 5f74 7265  data/species_tre
+0001a330: 6531 2e6e 7720 2d2d 6f6e 6c79 636f 6e63  e1.nw --onlyconc
+0001a340: 6174 6b73 7472 6565 202d 6f20 7767 645f  atkstree -o wgd_
+0001a350: 6b73 7472 6565 5f74 6f70 6f6c 6f67 7931  kstree_topology1
+0001a360: 0a60 6060 0a0a 215b 5d28 6461 7461 2f6b  .```..![](data/k
+0001a370: 7374 7265 655f 7265 7375 6c74 732f 6b73  stree_results/ks
+0001a380: 7472 6565 2e73 7667 290a 0a41 626f 7665  tree.svg)..Above
+0001a390: 2077 6520 7573 6564 2074 6872 6565 2061   we used three a
+0001a3a0: 6c74 6572 6e61 7469 7665 2074 6f70 6f6c  lternative topol
+0001a3b0: 6f67 6965 7320 746f 2069 6e66 6572 2074  ogies to infer t
+0001a3c0: 6865 202a 4b2a 3c73 7562 3e53 3c2f 7375  he *K*<sub>S</su
+0001a3d0: 623e 2074 7265 6520 7768 6963 6820 6c65  b> tree which le
+0001a3e0: 6420 746f 2064 6966 6665 7265 6e74 2062  d to different b
+0001a3f0: 7261 6e63 6820 6c65 6e67 7468 2065 7374  ranch length est
+0001a400: 696d 6174 696f 6e2e 204e 6f74 6520 7468  imation. Note th
+0001a410: 6174 2074 6865 2066 616d 696c 6965 7320  at the families 
+0001a420: 7765 2075 7365 6420 7765 7265 206f 6e6c  we used were onl
+0001a430: 7920 7477 6f20 676c 6f62 616c 204d 5242  y two global MRB
+0001a440: 4820 6661 6d69 6c69 6573 2066 6f72 2074  H families for t
+0001a450: 6865 2070 7572 706f 7365 206f 6620 696c  he purpose of il
+0001a460: 6c75 7374 7261 7469 6f6e 2e20 546f 2061  lustration. To a
+0001a470: 6371 7569 7265 2061 6e20 6163 6375 7261  cquire an accura
+0001a480: 7465 2070 726f 6669 6c65 206f 6620 7468  te profile of th
+0001a490: 6520 7375 6273 7469 7475 7469 6f6e 2072  e substitution r
+0001a4a0: 6174 6520 7661 7269 6174 696f 6e2c 206f  ate variation, o
+0001a4b0: 7274 686f 6c6f 6775 6573 2061 7420 7468  rthologues at th
+0001a4c0: 6520 7768 6f6c 6520 6765 6e6f 6d65 2073  e whole genome s
+0001a4d0: 6361 6c65 2073 686f 756c 6420 6265 2075  cale should be u
+0001a4e0: 7365 642e 0a0a 496e 2061 6464 6974 696f  sed...In additio
+0001a4f0: 6e2c 206d 6f72 6520 6578 7175 6973 6974  n, more exquisit
+0001a500: 6520 636f 6c6c 696e 6561 7220 706c 6f74  e collinear plot
+0001a510: 7320 696e 636c 7564 696e 6720 626f 7468  s including both
+0001a520: 2069 6e74 7261 2d73 7065 6369 6669 6320   intra-specific 
+0001a530: 616e 6420 696e 7465 722d 7370 6563 6966  and inter-specif
+0001a540: 6963 2063 6f6d 7061 7269 736f 6e73 2075  ic comparisons u
+0001a550: 7369 6e67 2074 6865 206f 7274 686f 6772  sing the orthogr
+0001a560: 6f75 7073 2028 636f 6d70 6f73 6564 206f  oups (composed o
+0001a570: 6620 2a41 7175 696c 6567 6961 2063 6f65  f *Aquilegia coe
+0001a580: 7275 6c65 612a 2c20 2a50 726f 7465 6120  rulea*, *Protea 
+0001a590: 6379 6e61 726f 6964 6573 2a2c 202a 4163  cynaroides*, *Ac
+0001a5a0: 6f72 7573 2061 6d65 7269 6361 6e75 732a  orus americanus*
+0001a5b0: 2061 6e64 202a 5669 7469 7320 7669 6e69   and *Vitis vini
+0001a5c0: 6665 7261 2a29 2069 6e66 6572 7265 6420  fera*) inferred 
+0001a5d0: 6361 6e20 6265 2061 6c73 6f20 7072 6f64  can be also prod
+0001a5e0: 7563 6564 2075 7369 6e67 2060 7767 6420  uced using `wgd 
+0001a5f0: 7379 6e60 2e20 4e6f 7465 2074 6861 7420  syn`. Note that 
+0001a600: 6469 6666 6572 656e 7420 6765 6e6f 6d65  different genome
+0001a610: 2061 7373 656d 626c 6965 7320 6d69 6768   assemblies migh
+0001a620: 7420 6861 7665 2064 6966 6665 7265 6e74  t have different
+0001a630: 2066 6561 7475 7265 7320 616e 6420 6174   features and at
+0001a640: 7472 6962 7574 6573 2077 6869 6368 2063  tributes which c
+0001a650: 616e 2062 6520 6163 636f 6d6d 6f64 6174  an be accommodat
+0001a660: 6564 2076 6961 2074 6865 206f 7074 696f  ed via the optio
+0001a670: 6e20 602d 2d61 6464 6974 696f 6e61 6c67  n `--additionalg
+0001a680: 6666 696e 666f 6020 666f 7220 6561 6368  ffinfo` for each
+0001a690: 2067 656e 6f6d 6520 6173 7365 6d62 6c79   genome assembly
+0001a6a0: 2077 686f 7365 206f 7264 6572 206e 6565   whose order nee
+0001a6b0: 6473 2074 6f20 666f 6c6c 6f77 2074 6865  ds to follow the
+0001a6c0: 206f 7264 6572 206f 6620 6766 6633 2066   order of gff3 f
+0001a6d0: 696c 6573 2c20 666f 7220 696e 7374 616e  iles, for instan
+0001a6e0: 6365 2027 6d4e 5241 3b4e 616d 6527 2066  ce 'mNRA;Name' f
+0001a6f0: 6f72 2060 4171 7569 6c65 6769 615f 636f  or `Aquilegia_co
+0001a700: 6572 756c 6561 2e67 6666 3360 2c20 276d  erulea.gff3`, 'm
+0001a710: 4e52 413b 4944 2720 666f 7220 6050 726f  NRA;ID' for `Pro
+0001a720: 7465 615f 6379 6e61 726f 6964 6573 2e67  tea_cynaroides.g
+0001a730: 6666 3360 2c20 276d 4e52 413b 4e61 6d65  ff3`, 'mNRA;Name
+0001a740: 2720 666f 7220 6041 636f 7275 735f 616d  ' for `Acorus_am
+0001a750: 6572 6963 616e 7573 2e67 6666 3360 2061  ericanus.gff3` a
+0001a760: 6e64 2027 6d4e 5241 3b4e 616d 6527 2066  nd 'mNRA;Name' f
+0001a770: 6f72 2060 5669 7469 735f 7669 6e69 6665  or `Vitis_vinife
+0001a780: 7261 2e67 6666 3360 2e0a 0a60 6060 0a77  ra.gff3`...```.w
+0001a790: 6764 2064 6d64 202d 6f6f 202d 6f69 2041  gd dmd -oo -oi A
+0001a7a0: 7175 696c 6567 6961 5f63 6f65 7275 6c65  quilegia_coerule
+0001a7b0: 6120 5072 6f74 6561 5f63 796e 6172 6f69  a Protea_cynaroi
+0001a7c0: 6465 7320 4163 6f72 7573 5f61 6d65 7269  des Acorus_ameri
+0001a7d0: 6361 6e75 7320 5669 7469 735f 7669 6e69  canus Vitis_vini
+0001a7e0: 6665 7261 202d 6f20 7767 645f 6f72 7468  fera -o wgd_orth
+0001a7f0: 6f0a 7767 6420 6b73 6420 7767 645f 6f72  o.wgd ksd wgd_or
+0001a800: 7468 6f2f 4f72 7468 6f67 726f 7570 732e  tho/Orthogroups.
+0001a810: 7370 2e74 7376 2041 7175 696c 6567 6961  sp.tsv Aquilegia
+0001a820: 5f63 6f65 7275 6c65 6120 5072 6f74 6561  _coerulea Protea
+0001a830: 5f63 796e 6172 6f69 6465 7320 4163 6f72  _cynaroides Acor
+0001a840: 7573 5f61 6d65 7269 6361 6e75 7320 5669  us_americanus Vi
+0001a850: 7469 735f 7669 6e69 6665 7261 202d 6f20  tis_vinifera -o 
+0001a860: 7767 645f 6f72 7468 6f5f 6b73 0a77 6764  wgd_ortho_ks.wgd
+0001a870: 2073 796e 2077 6764 5f6f 7274 686f 2f4f   syn wgd_ortho/O
+0001a880: 7274 686f 6772 6f75 7073 2e73 702e 7473  rthogroups.sp.ts
+0001a890: 7620 2d6b 7320 7767 645f 6f72 7468 6f5f  v -ks wgd_ortho_
+0001a8a0: 6b73 2f4f 7274 686f 6772 6f75 7073 2e73  ks/Orthogroups.s
+0001a8b0: 702e 7473 762e 6b73 2e74 7376 2041 7175  p.tsv.ks.tsv Aqu
+0001a8c0: 696c 6567 6961 5f63 6f65 7275 6c65 612e  ilegia_coerulea.
+0001a8d0: 6766 6633 202d 2d61 6464 6974 696f 6e61  gff3 --additiona
+0001a8e0: 6c67 6666 696e 666f 2027 6d4e 5241 3b4e  lgffinfo 'mNRA;N
+0001a8f0: 616d 6527 2050 726f 7465 615f 6379 6e61  ame' Protea_cyna
+0001a900: 726f 6964 6573 2e67 6666 3320 2d2d 6164  roides.gff3 --ad
+0001a910: 6469 7469 6f6e 616c 6766 6669 6e66 6f20  ditionalgffinfo 
+0001a920: 276d 4e52 413b 4944 2720 4163 6f72 7573  'mNRA;ID' Acorus
+0001a930: 5f61 6d65 7269 6361 6e75 732e 6766 6633  _americanus.gff3
+0001a940: 202d 2d61 6464 6974 696f 6e61 6c67 6666   --additionalgff
+0001a950: 696e 666f 2027 6d4e 5241 3b4e 616d 6527  info 'mNRA;Name'
+0001a960: 2056 6974 6973 5f76 696e 6966 6572 612e   Vitis_vinifera.
+0001a970: 6766 6633 202d 2d61 6464 6974 696f 6e61  gff3 --additiona
+0001a980: 6c67 6666 696e 666f 2027 6d4e 5241 3b4e  lgffinfo 'mNRA;N
+0001a990: 616d 6527 202d 6f20 7767 645f 6f72 7468  ame' -o wgd_orth
+0001a9a0: 6f5f 7379 6e0a 6060 600a 0a55 706f 6e20  o_syn.```..Upon 
+0001a9b0: 7468 6520 6163 7175 6973 6974 696f 6e20  the acquisition 
+0001a9c0: 6f66 2074 6865 2063 6f6c 6c69 6e65 6172  of the collinear
+0001a9d0: 2072 6573 756c 7473 2075 7369 6e67 2060   results using `
+0001a9e0: 7767 6420 7379 6e60 2c20 7468 6520 7361  wgd syn`, the sa
+0001a9f0: 6d65 2063 6f6c 6c69 6e65 6172 2070 6c6f  me collinear plo
+0001aa00: 7473 2063 616e 2062 6520 616c 736f 2070  ts can be also p
+0001aa10: 726f 6475 6365 6420 6279 2060 7767 6420  roduced by `wgd 
+0001aa20: 7669 7a60 2075 7369 6e67 2074 6865 2063  viz` using the c
+0001aa30: 6f6d 6d61 6e64 2062 656c 6f77 2e0a 6060  ommand below..``
+0001aa40: 600a 7767 6420 7669 7a20 2d2d 706c 6f74  `.wgd viz --plot
+0001aa50: 7379 6e20 2d73 6d20 7767 645f 6f72 7468  syn -sm wgd_orth
+0001aa60: 6f5f 7379 6e2f 6961 6468 6f72 652d 6f75  o_syn/iadhore-ou
+0001aa70: 742f 7365 676d 656e 7473 2e74 7874 202d  t/segments.txt -
+0001aa80: 6170 2077 6764 5f6f 7274 686f 5f73 796e  ap wgd_ortho_syn
+0001aa90: 2f69 6164 686f 7265 2d6f 7574 2f61 6e63  /iadhore-out/anc
+0001aaa0: 686f 7270 6f69 6e74 732e 7478 7420 2d6d  horpoints.txt -m
+0001aab0: 7420 7767 645f 6f72 7468 6f5f 7379 6e2f  t wgd_ortho_syn/
+0001aac0: 6961 6468 6f72 652d 6f75 742f 6d75 6c74  iadhore-out/mult
+0001aad0: 6970 6c69 636f 6e73 2e74 7874 202d 6774  iplicons.txt -gt
+0001aae0: 2077 6764 5f6f 7274 686f 5f73 796e 2f67   wgd_ortho_syn/g
+0001aaf0: 656e 652d 7461 626c 652e 6373 7620 2d64  ene-table.csv -d
+0001ab00: 2077 6764 5f6f 7274 686f 5f6b 732f 4f72   wgd_ortho_ks/Or
+0001ab10: 7468 6f67 726f 7570 732e 7370 2e74 7376  thogroups.sp.tsv
+0001ab20: 2e6b 732e 7473 7620 2d6f 2077 6764 5f6f  .ks.tsv -o wgd_o
+0001ab30: 7274 686f 5f76 697a 0a60 6060 0a0a 215b  rtho_viz.```..![
+0001ab40: 5d28 6461 7461 2f6f 7274 686f 5f73 796e  ](data/ortho_syn
+0001ab50: 5f72 6573 756c 7473 2f41 7175 696c 6567  _results/Aquileg
+0001ab60: 6961 5f63 6f65 7275 6c65 615f 5669 7469  ia_coerulea_Viti
+0001ab70: 735f 7669 6e69 6665 7261 5f6d 756c 7469  s_vinifera_multi
+0001ab80: 706c 6963 6f6e 735f 6c65 7665 6c2e 706e  plicons_level.pn
+0001ab90: 6729 0a54 6865 2061 626f 7665 2060 6475  g).The above `du
+0001aba0: 7053 7461 636b 6020 706c 6f74 2073 686f  pStack` plot sho
+0001abb0: 7773 2074 6865 2064 6973 7472 6962 7574  ws the distribut
+0001abc0: 696f 6e20 6f66 2064 7570 6c69 6361 7465  ion of duplicate
+0001abd0: 6420 7365 676d 656e 7473 206f 6620 2a41  d segments of *A
+0001abe0: 7175 696c 6567 6961 2063 6f65 7275 6c65  quilegia coerule
+0001abf0: 612a 2063 6f6d 7061 7265 6420 746f 2069  a* compared to i
+0001ac00: 7473 656c 6620 2869 6e20 6772 6565 6e29  tself (in green)
+0001ac10: 2061 6e64 2063 6f6d 7061 7265 6420 746f   and compared to
+0001ac20: 202a 5669 7469 7320 7669 6e69 6665 7261   *Vitis vinifera
+0001ac30: 2a20 2869 6e20 626c 7565 2920 6f76 6572  * (in blue) over
+0001ac40: 2074 6865 2063 6872 6f6d 6f73 6f6d 6573   the chromosomes
+0001ac50: 206f 6620 2a41 2e20 636f 6572 756c 6561   of *A. coerulea
+0001ac60: 2a2e 0a21 5b5d 2864 6174 612f 6f72 7468  *..![](data/orth
+0001ac70: 6f5f 7379 6e5f 7265 7375 6c74 732f 4f76  o_syn_results/Ov
+0001ac80: 6572 616c 6c73 7065 6369 6573 5f4b 732e  erallspecies_Ks.
+0001ac90: 646f 745f 756e 6974 5f67 656e 652e 706e  dot_unit_gene.pn
+0001aca0: 6729 0a54 6865 2061 626f 7665 202a 4b2a  g).The above *K*
+0001acb0: 3c73 7562 3e53 3c2f 7375 623e 2064 6f74  <sub>S</sub> dot
+0001acc0: 706c 6f74 2069 6e20 756e 6974 206f 6620  plot in unit of 
+0001acd0: 6765 6e65 2073 686f 7773 2074 6865 206f  gene shows the o
+0001ace0: 7665 7261 6c6c 2064 6973 7472 6962 7574  verall distribut
+0001acf0: 696f 6e20 6f66 2063 6f6c 6c69 6e65 6172  ion of collinear
+0001ad00: 6974 7920 6163 6f72 7373 2074 6865 2066  ity acorss the f
+0001ad10: 6f75 7220 7370 6563 6965 7320 696e 766f  our species invo
+0001ad20: 6c76 6564 2e0a 215b 5d28 6461 7461 2f6f  lved..![](data/o
+0001ad30: 7274 686f 5f73 796e 5f72 6573 756c 7473  rtho_syn_results
+0001ad40: 2f4f 7665 7261 6c6c 7370 6563 6965 732e  /Overallspecies.
+0001ad50: 646f 745f 756e 6974 5f67 656e 652e 706e  dot_unit_gene.pn
+0001ad60: 6729 0a54 6865 2061 626f 7665 2064 6f74  g).The above dot
+0001ad70: 706c 6f74 2069 7320 7769 7468 6f75 7420  plot is without 
+0001ad80: 7468 6520 616e 6e6f 7461 7469 6f6e 206f  the annotation o
+0001ad90: 6620 2a4b 2a3c 7375 623e 533c 2f73 7562  f *K*<sub>S</sub
+0001ada0: 3e20 6167 6573 2063 6f6d 7061 7265 6420  > ages compared 
+0001adb0: 746f 2074 6865 206c 6173 7420 6f6e 652e  to the last one.
+0001adc0: 0a21 5b5d 2864 6174 612f 6f72 7468 6f5f  .![](data/ortho_
+0001add0: 7379 6e5f 7265 7375 6c74 732f 5379 6e64  syn_results/Synd
+0001ade0: 6570 7468 2e73 7667 290a 5468 6520 6162  epth.svg).The ab
+0001adf0: 6f76 6520 5379 6e64 6570 7468 2070 6c6f  ove Syndepth plo
+0001ae00: 7420 7368 6f77 7320 7468 6520 636f 6c6c  t shows the coll
+0001ae10: 696e 6561 7220 7261 7469 6f20 6163 6f72  inear ratio acor
+0001ae20: 7373 2061 6c6c 2073 7065 6369 6573 2070  ss all species p
+0001ae30: 6169 7273 2028 696e 7472 612d 7370 6563  airs (intra-spec
+0001ae40: 6966 6963 2063 6f6d 7061 7269 736f 6e20  ific comparison 
+0001ae50: 696e 2067 7265 656e 2077 6869 6c65 2069  in green while i
+0001ae60: 6e74 6572 2d73 7065 6369 6669 6320 636f  nter-specific co
+0001ae70: 6d70 6172 6973 6f6e 2069 6e20 626c 7565  mparison in blue
+0001ae80: 292e 0a0a 2323 2043 6974 6174 696f 6e0a  )...## Citation.
+0001ae90: 200a 506c 6561 7365 2063 6974 6520 7573   .Please cite us
+0001aea0: 2061 7420 6874 7470 733a 2f2f 646f 692e   at https://doi.
+0001aeb0: 6f72 672f 3130 2e31 3030 372f 3937 382d  org/10.1007/978-
+0001aec0: 312d 3037 3136 2d32 3536 312d 335f 3120  1-0716-2561-3_1 
+0001aed0: 616e 6420 6874 7470 733a 2f2f 646f 692e  and https://doi.
+0001aee0: 6f72 672f 3130 2e31 3039 332f 6269 6f69  org/10.1093/bioi
+0001aef0: 6e66 6f72 6d61 7469 6373 2f62 7461 6532  nformatics/btae2
+0001af00: 3732 2e0a 0a60 6060 0a48 656e 6763 6869  72...```.Hengchi
+0001af10: 2043 6865 6e2c 2041 7274 6875 7220 5a77   Chen, Arthur Zw
+0001af20: 6165 6e65 706f 656c 2028 3230 3233 292e  aenepoel (2023).
+0001af30: 2049 6e66 6572 656e 6365 206f 6620 416e   Inference of An
+0001af40: 6369 656e 7420 506f 6c79 706c 6f69 6479  cient Polyploidy
+0001af50: 2066 726f 6d20 4765 6e6f 6d69 6320 4461   from Genomic Da
+0001af60: 7461 2e20 496e 3a20 5661 6e20 6465 2050  ta. In: Van de P
+0001af70: 6565 722c 2059 2e20 2865 6473 2920 506f  eer, Y. (eds) Po
+0001af80: 6c79 706c 6f69 6479 2e20 4d65 7468 6f64  lyploidy. Method
+0001af90: 7320 696e 204d 6f6c 6563 756c 6172 2042  s in Molecular B
+0001afa0: 696f 6c6f 6779 2c20 766f 6c20 3235 3435  iology, vol 2545
+0001afb0: 2e20 4875 6d61 6e61 2c20 4e65 7720 596f  . Humana, New Yo
+0001afc0: 726b 2c20 4e59 2e20 6874 7470 733a 2f2f  rk, NY. https://
+0001afd0: 646f 692e 6f72 672f 3130 2e31 3030 372f  doi.org/10.1007/
+0001afe0: 3937 382d 312d 3037 3136 2d32 3536 312d  978-1-0716-2561-
+0001aff0: 335f 310a 4865 6e67 6368 6920 4368 656e  3_1.Hengchi Chen
+0001b000: 2c20 4172 7468 7572 205a 7761 656e 6570  , Arthur Zwaenep
+0001b010: 6f65 6c2c 2059 7665 7320 5661 6e20 6465  oel, Yves Van de
+0001b020: 2050 6565 7220 2832 3032 3429 2e20 7767   Peer (2024). wg
+0001b030: 6420 7632 3a20 6120 7375 6974 6520 6f66  d v2: a suite of
+0001b040: 2074 6f6f 6c73 2074 6f20 756e 636f 7665   tools to uncove
+0001b050: 7220 616e 6420 6461 7465 2061 6e63 6965  r and date ancie
+0001b060: 6e74 2070 6f6c 7970 6c6f 6964 7920 616e  nt polyploidy an
+0001b070: 6420 7768 6f6c 652d 6765 6e6f 6d65 2064  d whole-genome d
+0001b080: 7570 6c69 6361 7469 6f6e 2e20 4269 6f69  uplication. Bioi
+0001b090: 6e66 6f72 6d61 7469 6373 2c20 566f 6c75  nformatics, Volu
+0001b0a0: 6d65 2034 302c 2049 7373 7565 2035 2c20  me 40, Issue 5, 
+0001b0b0: 4d61 7920 3230 3234 2c20 6274 6165 3237  May 2024, btae27
+0001b0c0: 322c 2068 7474 7073 3a2f 2f64 6f69 2e6f  2, https://doi.o
+0001b0d0: 7267 2f31 302e 3130 3933 2f62 696f 696e  rg/10.1093/bioin
+0001b0e0: 666f 726d 6174 6963 732f 6274 6165 3237  formatics/btae27
+0001b0f0: 320a 6060 600a 0a46 6f72 2063 6974 6174  2.```..For citat
+0001b100: 696f 6e20 6f66 2074 6865 2074 6f6f 6c73  ion of the tools
+0001b110: 2075 7365 6420 696e 2077 6764 2c20 706c   used in wgd, pl
+0001b120: 6561 7365 2063 6f6e 7375 6c74 2074 6865  ease consult the
+0001b130: 2064 6f63 756d 656e 7461 7469 6f6e 2061   documentation a
+0001b140: 740a 6874 7470 733a 2f2f 7767 6476 322e  t.https://wgdv2.
+0001b150: 7265 6164 7468 6564 6f63 732e 696f 2f65  readthedocs.io/e
+0001b160: 6e2f 6c61 7465 7374 2f63 6974 6174 696f  n/latest/citatio
+0001b170: 6e2e 6874 6d6c 2e0a 0a                   n.html...
```

### Comparing `wgd-2.0.37/wgd.egg-info/requires.txt` & `wgd-2.0.38/wgd.egg-info/requires.txt`

 * *Files identical despite different names*

